<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.554">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>EMBER creates a unified space for independent breast cancer transcriptomic datasets enabling precision oncology. - 6&nbsp; Expanding embedding and removing batch effects</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../chapters/trying.html" rel="next">
<link href="../chapters/risk_score.html" rel="prev">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script src="../site_libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<link href="../site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="../site_libs/datatables-binding-0.28/datatables.js"></script>
<script src="../site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<link href="../site_libs/dt-core-1.13.4/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="../site_libs/dt-core-1.13.4/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="../site_libs/dt-core-1.13.4/js/jquery.dataTables.min.js"></script>
<link href="../site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet">
<script src="../site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="../site_libs/nouislider-7.0.10/jquery.nouislider.min.css" rel="stylesheet">
<script src="../site_libs/nouislider-7.0.10/jquery.nouislider.min.js"></script>
<link href="../site_libs/selectize-0.12.0/selectize.bootstrap3.css" rel="stylesheet">
<script src="../site_libs/selectize-0.12.0/selectize.min.js"></script>

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../chapters/expanding_ember.html"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Expanding embedding and removing batch effects</span></a></li></ol></nav>
        <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../">EMBER creates a unified space for independent breast cancer transcriptomic datasets enabling precision oncology.</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/chronchi/molecular_landscape" title="Source Code" class="quarto-navigation-tool px-1" aria-label="Source Code"><i class="bi bi-github"></i></a>
    <a href="https://twitter.com/intent/tweet?url=|url|" title="Twitter" class="quarto-navigation-tool px-1" aria-label="Twitter"><i class="bi bi-twitter"></i></a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Preface</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/intro.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Introduction</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/surv_analysis_estrogen.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Estrogen receptor status is continuous, not dichotomous</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/pca_merging.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">A new framework for personalized medicine: integrating</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/validation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Validating the molecular landscape</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/scoring.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Pathway activity in a personalized context</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/risk_score.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Risk score model</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/expanding_ember.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Expanding embedding and removing batch effects</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/trying.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Tests and tries</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/revision.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Revision</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/discussion.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Discussion</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/references.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">References</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#embedding-with-more-than-1000-genes" id="toc-embedding-with-more-than-1000-genes" class="nav-link active" data-scroll-target="#embedding-with-more-than-1000-genes"><span class="header-section-number">6.1</span> Embedding with more than 1000 genes</a></li>
  <li><a href="#removing-the-batch-effects" id="toc-removing-the-batch-effects" class="nav-link" data-scroll-target="#removing-the-batch-effects"><span class="header-section-number">6.2</span> Removing the batch effects</a></li>
  <li><a href="#scoring-strategy-with-regressed-data" id="toc-scoring-strategy-with-regressed-data" class="nav-link" data-scroll-target="#scoring-strategy-with-regressed-data"><span class="header-section-number">6.3</span> Scoring strategy with regressed data</a>
  <ul class="collapse">
  <li><a href="#checking-the-patients-from-poetic-with-new-scores" id="toc-checking-the-patients-from-poetic-with-new-scores" class="nav-link" data-scroll-target="#checking-the-patients-from-poetic-with-new-scores"><span class="header-section-number">6.3.1</span> Checking the patients from POETIC with new scores</a></li>
  <li><a href="#responder-vs-non-responder-comparison-of-new-scores" id="toc-responder-vs-non-responder-comparison-of-new-scores" class="nav-link" data-scroll-target="#responder-vs-non-responder-comparison-of-new-scores"><span class="header-section-number">6.3.2</span> Responder vs non responder comparison of new scores</a></li>
  <li><a href="#survival-analysis-with-the-new-scores" id="toc-survival-analysis-with-the-new-scores" class="nav-link" data-scroll-target="#survival-analysis-with-the-new-scores"><span class="header-section-number">6.3.3</span> Survival analysis with the new scores</a></li>
  </ul></li>
  <li><a href="#regressing-the-pdxs" id="toc-regressing-the-pdxs" class="nav-link" data-scroll-target="#regressing-the-pdxs"><span class="header-section-number">6.4</span> Regressing the PDXs</a></li>
  <li><a href="#singscore" id="toc-singscore" class="nav-link" data-scroll-target="#singscore"><span class="header-section-number">6.5</span> singscore</a></li>
  </ul>
<div class="toc-actions"><ul><li><a href="https://github.com/chronchi/molecular_landscape/issues/new" class="toc-action"><i class="bi bi-github"></i>Report an issue</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Expanding embedding and removing batch effects</span></h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<p>We have used some numbers of genes for doing the embedding. Here we try to perform the embedding with all the genes (~10000 genes) and then we try to regress out the components.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>* The library is already synchronized with the lockfile.</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>    ../../results/plots/expanding_ember ../../results/rds_files/expanding_ember 
                                  FALSE                                    TRUE 
   ../../results/tables/expanding_ember 
                                  FALSE </code></pre>
</div>
</div>
<section id="embedding-with-more-than-1000-genes" class="level2" data-number="6.1">
<h2 data-number="6.1" class="anchored" data-anchor-id="embedding-with-more-than-1000-genes"><span class="header-section-number">6.1</span> Embedding with more than 1000 genes</h2>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>Total number of stable genes: 39
Total number of genes: 8248
Number of samples: 1005
Total number of stable genes: 39
Total number of genes: 8248
Number of samples: 7216
Total number of stable genes: 39
Total number of genes: 8248
Number of samples: 1868
Total number of stable genes: 39
Total number of genes: 8248
Number of samples: 426</code></pre>
</div>
</div>
<p>And now we can visualize the results.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pca-embeddings-1" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-pca-embeddings-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="expanding_ember_files/figure-html/fig-pca-embeddings-1-1.png" class="img-fluid figure-img" width="1344">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-pca-embeddings-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;6.1: PCA projections colored by different factors and organized by different components. (A) Plot of the first two components colored by cohort. (B) Plot of the first two components colored by ER status. (C) Plot of the second and third components colored by cohort. (D) Plot of the second and third components colored by ER status. (E) Plot of the second and third components colored by PAM50. (F) Plot of the second and third components colored by INTCLUST, only METABRIC has an assigned value for this variable, NAs are TCGA samples.
</figcaption>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="expanding_ember_files/figure-html/unnamed-chunk-4-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>And now we can visualize the results.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pca-embeddings-2" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-pca-embeddings-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="expanding_ember_files/figure-html/fig-pca-embeddings-2-1.png" class="img-fluid figure-img" width="1344">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-pca-embeddings-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;6.2: PCA projections colored by different factors and organized by different components. (A) Plot of the first two components colored by cohort. (B) Plot of the first two components colored by ER status. (C) Plot of the second and third components colored by cohort. (D) Plot of the second and third components colored by ER status. (E) Plot of the second and third components colored by PAM50. (F) Plot of the second and third components colored by INTCLUST, only METABRIC has an assigned value for this variable, NAs are TCGA samples.
</figcaption>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="expanding_ember_files/figure-html/unnamed-chunk-8-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pca-scanb-pc1" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-pca-scanb-pc1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="expanding_ember_files/figure-html/fig-pca-scanb-pc1-1.png" class="img-fluid figure-img" width="672">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-pca-scanb-pc1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;6.3: PCA embedding colored by cohort. The components used are the first and second components.
</figcaption>
</figure>
</div>
</div>
</div>
<p><a href="#fig-pca-scanb-er-pam50-all-genes" class="quarto-xref">Figure&nbsp;<span>6.4</span></a> shows that the SCANB samples are also well mixed regarding the clinical factors, including the <span class="math inline">\(SET_{ER/PR}\)</span> signature.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pca-scanb-er-pam50-all-genes" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-pca-scanb-er-pam50-all-genes-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="expanding_ember_files/figure-html/fig-pca-scanb-er-pam50-all-genes-1.png" class="img-fluid figure-img" width="1536">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-pca-scanb-er-pam50-all-genes-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;6.4: PCA embedding of all samples from TCGA, SCANB and METABRIC. (A) Colored by cohort, (B) colored by ER status, (C) colored by PAM50 molecular subtype, (D) colored by the <span class="math inline">\(SET_{ER/PR}\)</span> signature and only SCANB samples.
</figcaption>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="expanding_ember_files/figure-html/unnamed-chunk-11-1.png" class="img-fluid figure-img" width="1152"></p>
</figure>
</div>
</div>
</div>
<p>Let us also check the other components in the biplots.</p>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="expanding_ember_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid figure-img" width="1536"></p>
</figure>
</div>
</div>
</div>
<p>We see that the pam50 subtypes are being separated using the components 3, 4 and also 8. In the component 8 there is some expression that is able to differentiate between HER2-like from all other samples.</p>
<p>We now plot by cohort.</p>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="expanding_ember_files/figure-html/unnamed-chunk-13-1.png" class="img-fluid figure-img" width="1536"></p>
</figure>
</div>
</div>
</div>
<p>In PC1, 2 and 5 there are some small differences in the differences of the distributions from the datasets. We will then remove these 3 components up to component 10 with exception of components 3, 4 and 5.</p>
</section>
<section id="removing-the-batch-effects" class="level2" data-number="6.2">
<h2 data-number="6.2" class="anchored" data-anchor-id="removing-the-batch-effects"><span class="header-section-number">6.2</span> Removing the batch effects</h2>
<p>We now try to remove the batch effects by removing all the PCs that are not of interest, so we remove the first two components and the fifth as well. We then calculate the scores for each cohort separately and then compare to what was calculated before. We can then try to calculate the scores using samples coming from all cohorts.</p>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="expanding_ember_files/figure-html/unnamed-chunk-15-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="expanding_ember_files/figure-html/unnamed-chunk-16-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>In general it looks good, the luminal B population is a bit compressed towards the luminal A population though.</p>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="expanding_ember_files/figure-html/unnamed-chunk-17-1.png" class="img-fluid figure-img" width="576"></p>
</figure>
</div>
</div>
</div>
<p>There is a very good merging of the datasets. We see mostly green points because they are the majority.</p>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="expanding_ember_files/figure-html/unnamed-chunk-18-1.png" class="img-fluid figure-img" width="960"></p>
</figure>
</div>
</div>
</div>
<p>Now we see that the first loadings has a tail in a similar fashio as the other loadings.</p>
<p>We now try to calculate the scores of the samples using these new embeddings only from TCGA samples as we will compare the results later on.</p>
<p>And we can now check and compare the scores.</p>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="expanding_ember_files/figure-html/unnamed-chunk-20-1.png" class="img-fluid figure-img" width="960"></p>
</figure>
</div>
</div>
</div>
<p>We see that for the three pathways the correlation is positive and strong. The difference is that it is a bit noisy, so the correlation is not perfect. What is interesting to see is that at the extremes the scores have less variation usually.</p>
<p>We now calculate the scores from SCANB and compare the results. Remember that SCANB was not used for the PCA training.</p>
<p>And we can now check and compare the scores.</p>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="expanding_ember_files/figure-html/unnamed-chunk-22-1.png" class="img-fluid figure-img" width="960"></p>
</figure>
</div>
</div>
</div>
<p>The same phenomenon happened. Samples in the endges have lower variability, but samples in the middle are the ones that have high variability. For example there are several samples from the regressed data with SET_ERPR score close to 0.25 and negative scores.</p>
<p>What happens now if we use another dataset and then use only one sample to calculate the score. Let us include SCANB samples in the pipeline. For this we select 50 scanb samples and 100 random samples from tcga and metabric. We then combine all these samples together and calculate the GSVA scores.</p>
<p>We now plot only the scores from the SCANB samples.</p>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="expanding_ember_files/figure-html/unnamed-chunk-24-1.png" class="img-fluid figure-img" width="960"></p>
</figure>
</div>
</div>
</div>
<p>So it seems there are still dataset related problems there when calculating the scores. The scores are positively correlated, but they are not quite the same as only using the SCANB dataset. Moreover, for G2M checkpoint the scores are concentrated around -0.4 from the regressed data but they are in a range going from - 0.4 to 0.1 in the unregressed data.</p>
<p>If we try to instead add some samples from the SCANB dataset as well randomly into the mix. How well does the scores get compared to the “ground truth”?</p>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="expanding_ember_files/figure-html/unnamed-chunk-26-1.png" class="img-fluid figure-img" width="960"></p>
</figure>
</div>
</div>
</div>
<p>It does not work so well when including some SCANB samples together with the METABRIC and TCGA samples. We are trying to achieve a correlation of almost 1 with very low variability since we want to use the scores clinically.</p>
<p>Let us check the distribution of some genes for SCANB and TCGA after regressing out the first two PCs.</p>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="expanding_ember_files/figure-html/unnamed-chunk-27-1.png" class="img-fluid figure-img" width="960"></p>
</figure>
</div>
</div>
</div>
<p>We see that even though the expression profile of the samples are overlapping there are discrepancies. That is why when using GSVA it does not work so well perhaps.</p>
</section>
<section id="scoring-strategy-with-regressed-data" class="level2" data-number="6.3">
<h2 data-number="6.3" class="anchored" data-anchor-id="scoring-strategy-with-regressed-data"><span class="header-section-number">6.3</span> Scoring strategy with regressed data</h2>
<p>We now try to use a slightly different scoring strategy. We can think of our data in a log scale, so what we do now instead is to sum all of our genes of interest and then subtract an average of the housekeeping genes expression levels.</p>
<p>First we check the distribution of this average for each cohort separately.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-hkg-regressed" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-hkg-regressed-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="expanding_ember_files/figure-html/fig-hkg-regressed-1.png" class="img-fluid figure-img" width="768">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-hkg-regressed-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;6.5: Distribution of the average of the housekeeping genes for distinct cohorts.
</figcaption>
</figure>
</div>
</div>
</div>
<p>Indeed the tcga and metabric cohorts have an overlapping distribution, but the SCANB does not overlap completely, it is actually shifted. One should still take into account that the distributions are very close to 0, meaning that the housekeeping genes have very close to 0 values, as expected here in this case. We now calculate the score by taking this into account.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-scores-regressed" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-scores-regressed-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="expanding_ember_files/figure-html/fig-scores-regressed-1.png" class="img-fluid figure-img" width="1344">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-scores-regressed-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;6.6: Scores obtained from regressed data of all the four cohorts TCGA, SCANB, METABRIC and POETIC compared to the original GSVA scores.
</figcaption>
</figure>
</div>
</div>
</div>
<p>The scores are actually highly correlated, specially the estrogen related pathways. Also the correlation between the scores in general follows the same trends.</p>
<p>The plot below is a subset of the scores above to include in the publication.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-scores-regressed-subset" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-scores-regressed-subset-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="expanding_ember_files/figure-html/fig-scores-regressed-subset-1.png" class="img-fluid figure-img" width="768">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-scores-regressed-subset-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;6.7: Scores obtained from regressed data of all the four cohorts TCGA, SCANB, METABRIC and POETIC compared to the original GSVA scores.
</figcaption>
</figure>
</div>
</div>
</div>
<p>We now compare the distributions of these scores across the different cohorts and pathways of interest.</p>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="expanding_ember_files/figure-html/unnamed-chunk-34-1.png" class="img-fluid figure-img" width="1536"></p>
</figure>
</div>
</div>
</div>
<p>And with all molecular subtypes together.</p>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="expanding_ember_files/figure-html/unnamed-chunk-35-1.png" class="img-fluid figure-img" width="960"></p>
</figure>
</div>
</div>
</div>
<p>We see that the overlap is actually pretty good in general, even for SCANB. The only part where there is a bigger discrepancy is for G2M checkpoint. We see below that by calculating the GSVA score on the whole scanb cohort we don’t have this problem.</p>
<p>And as a comparison we plot with the original scores using GSVA on the whole cohort individually.</p>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="expanding_ember_files/figure-html/unnamed-chunk-36-1.png" class="img-fluid figure-img" width="1536"></p>
</figure>
</div>
</div>
</div>
<p>And all the subtypes together for the original score:</p>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="expanding_ember_files/figure-html/unnamed-chunk-37-1.png" class="img-fluid figure-img" width="960"></p>
</figure>
</div>
</div>
</div>
<p>We see that by doing GSVA the distributions they overlap much more than using the regressed data and the simple sum of the values for those genes. Still for estrogen signaling it seems that the estrogen early signature is pretty robust.</p>
<section id="checking-the-patients-from-poetic-with-new-scores" class="level3" data-number="6.3.1">
<h3 data-number="6.3.1" class="anchored" data-anchor-id="checking-the-patients-from-poetic-with-new-scores"><span class="header-section-number">6.3.1</span> Checking the patients from POETIC with new scores</h3>
<p>We now use the new scores to evaluate the comparison with the neighboorhood, just as before in the previous chapter where we used the POETIC trial data.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1] "treated_nb63_baseline"
[1] "treated_nb236_baseline"</code></pre>
</div>
</div>
<p>Patient 63 <a href="#fig-pt63" class="quarto-xref">Figure&nbsp;<span>6.8</span></a> is considered a responder. According to the scores, when comparing the estrogen early signature to its average distribution in the neighborhood, the score is higher.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pt63" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-pt63-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="expanding_ember_files/figure-html/fig-pt63-1.png" class="img-fluid figure-img" width="960">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-pt63-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;6.8: Posterior distribution of the average scores in the neighborhood of patient 63 from POETIC trial. Each dot corresponds to a 1% quantile.
</figcaption>
</figure>
</div>
</div>
</div>
<p>Patient 236 <a href="#fig-pt236" class="quarto-xref">Figure&nbsp;<span>6.9</span></a> was considered a non responder. According to the scores, when comparing the estrogen early signature to its average distribution in the neighborhood, the score is lower, being in the 5% quantile.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pt236" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-pt236-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="expanding_ember_files/figure-html/fig-pt236-1.png" class="img-fluid figure-img" width="960">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-pt236-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;6.9: Posterior distribution of the average scores in the neighborhood of patient 236 from POETIC trial. Each dot corresponds to a 1% quantile.
</figcaption>
</figure>
</div>
</div>
</div>
<p>When comparing these two patients, there is also a difference in the androgen response score, which could be a reflection of different estrogen signaling.</p>
<p>It reflects what we saw previously as well.</p>
</section>
<section id="responder-vs-non-responder-comparison-of-new-scores" class="level3" data-number="6.3.2">
<h3 data-number="6.3.2" class="anchored" data-anchor-id="responder-vs-non-responder-comparison-of-new-scores"><span class="header-section-number">6.3.2</span> Responder vs non responder comparison of new scores</h3>
<p>We now compare the scores for responders and non responders.</p>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="expanding_ember_files/figure-html/unnamed-chunk-41-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
<p>The higher E2F targets in average the lower the estrogen early score is for non responders, whereas this correlation does not exist for responder patients. This difference is mostly drive by the high E2F target non responder tumors.</p>
<p>Now the plot below shows the correlation for baselien Ki67% and E2F targets for the score calculated in the regressed data in a single sample manner. We see a positive correlation between the two scores. Moreover, it is interesting to see that when the tumor has low Ki67, there is still a spectrum on the E2F targets score. Also there is no difference in the correlation between responders and non responders.</p>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="expanding_ember_files/figure-html/unnamed-chunk-42-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
<p>And lastly we compare the estrogen signaling signatures between responders and non-responder.</p>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="expanding_ember_files/figure-html/unnamed-chunk-43-1.png" class="img-fluid figure-img" width="960"></p>
</figure>
</div>
</div>
</div>
<p>We see that the estrogen response early has a wider range when compared to the SET ER/PR score. That makes sense as this signature has over 100 genes, whereas SET ER/PR has only 18. Also the responder was very good in discriminating the responders and non responders. What we can conclude from this figure is that the samples with very low scores are usually non responders, which reflect their position in the molecular landscape, also in general responders have higher ER signaling than non responders.</p>
<p>Below is the figure with the GSVA scores before regressing the data, we see that SET ER/PR is not able to really distinguish completely the non-responders with very low score similar to what was done with GSVA. Even estrogen early had some differences.</p>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="expanding_ember_files/figure-html/unnamed-chunk-44-1.png" class="img-fluid figure-img" width="960"></p>
</figure>
</div>
</div>
</div>
<p>And below is a figure with both GSVA and regressed scores together.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-comparison-scoring-strategy-poetic" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-comparison-scoring-strategy-poetic-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="expanding_ember_files/figure-html/fig-comparison-scoring-strategy-poetic-1.png" class="img-fluid figure-img" width="960">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-comparison-scoring-strategy-poetic-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;6.10: Comparing scoring strategies for responders and non responders in the POETIC dataset.
</figcaption>
</figure>
</div>
</div>
</div>
<p>Below we show the correlation between these scores for all cohorts.</p>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="expanding_ember_files/figure-html/unnamed-chunk-46-1.png" class="img-fluid figure-img" width="576"></p>
</figure>
</div>
</div>
</div>
<p>These scores are highly correlated but in different absolute scales. We now check the change in Ki67 for all patients and compare the scores with the original ones obtained with GSVA.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-comparison-og-regressed-er" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-comparison-og-regressed-er-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="expanding_ember_files/figure-html/fig-comparison-og-regressed-er-1.png" class="img-fluid figure-img" width="1152">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-comparison-og-regressed-er-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;6.11: Comparison of original GSVA and regressed scores between responders and non responders compared to change in Ki67.
</figcaption>
</figure>
</div>
</div>
</div>
<p>We see that the bigger the change the higher the score is when using the regressed data. Also the non responders tend to have scores mostly in the negative scale, whereas the responders are more shifted to the right. SET ER/PR is also a good measure here.</p>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="expanding_ember_files/figure-html/unnamed-chunk-48-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>There is a very good match between the two scores in general, meaning that the regression scoring strategy capture relatively well the scores. There is one patient that is a non-responder and has a low regressed estrogen early but a GSVA score close to 0, let us check this patient.</p>
<div class="cell">
<div class="cell-output-display">
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-4ae39b6aede5654482e0" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-4ae39b6aede5654482e0">{"x":{"filter":"none","vertical":false,"data":[["1"],["80"],[43.24122479462285],[-5.649422739144427],[26.83238127635096],["pos"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>patient_nb<\/th>\n      <th>ki67<\/th>\n      <th>PC3<\/th>\n      <th>change_ki67<\/th>\n      <th>er_status<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<p>It is a patient that is far to the left on the molecular landscape, among the basal like, and had an increase in Ki67 upon ET. So the new scoring system was able to capture this patient as with low score.</p>
</section>
<section id="survival-analysis-with-the-new-scores" class="level3" data-number="6.3.3">
<h3 data-number="6.3.3" class="anchored" data-anchor-id="survival-analysis-with-the-new-scores"><span class="header-section-number">6.3.3</span> Survival analysis with the new scores</h3>
<p>We can check how robust these scores are by performing survival analysis.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pca-mol-land-survival-regressed" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-pca-mol-land-survival-regressed-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="expanding_ember_files/figure-html/fig-pca-mol-land-survival-regressed-1.png" class="img-fluid figure-img" width="864">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-pca-mol-land-survival-regressed-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;6.12: Overall survival analysis from SCANB and METABRIC cohorts and their scores. The formulas used were the same as for the estrogen signaling analysis. Coefficients were scaled before performing cox regression
</figcaption>
</figure>
</div>
</div>
</div>
<p>The plot with TCGA is shown below.</p>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="expanding_ember_files/figure-html/unnamed-chunk-52-1.png" class="img-fluid figure-img" width="432"></p>
</figure>
</div>
</div>
</div>
<p>Scores for TCGA are noisier than SCANB and METABRIC. Though the scaled scores they have comparable hazard ratios and also they follow what we saw previously. The only pathway that is a bit different is the PI3K AKT MTOR signaling. This pathway had a noisier correlation with the GSVA score as well.</p>
<div class="cell">
<div class="cell-output-display">
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-52bb2d1b022e30491dca" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-52bb2d1b022e30491dca">{"x":{"filter":"top","vertical":false,"filterHTML":"<tr>\n  <td><\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"-0.398255102696566\" data-max=\"1.25129356321115\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0.00461050175932\" data-max=\"0.715250026338379\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"-6.76436087805681\" data-max=\"17.6290977009835\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0\" data-max=\"0.999099322836264\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0.671490706720718\" data-max=\"3.49486085817101\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n<\/tr>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114","115","116","117","118","119","120","121","122","123","124","125","126","127","128","129","130","131","132","133","134","135","136","137","138","139","140","141","142","143","144","145","146","147","148","149","150","151","152","153","154","155","156","157","158","159","160","161","162","163","164","165","166","167","168","169","170","171","172","173","174","175"],["os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs"],["tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","metabric","metabric","metabric","metabric","metabric","metabric","metabric","metabric","metabric","metabric","metabric","metabric","metabric","metabric","metabric","metabric","metabric","metabric","metabric","metabric","metabric","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","metabric","metabric","metabric","metabric","metabric","metabric","metabric","metabric","metabric","metabric","metabric","metabric","metabric","metabric","metabric","metabric","metabric","metabric","metabric","metabric","metabric"],["HALLMARK_G2M_CHECKPOINT_new","HALLMARK_G2M_CHECKPOINT_new","HALLMARK_G2M_CHECKPOINT_new","HALLMARK_G2M_CHECKPOINT_new","HALLMARK_G2M_CHECKPOINT_new","HALLMARK_G2M_CHECKPOINT_new","HALLMARK_G2M_CHECKPOINT_new","HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION_new","HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION_new","HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION_new","HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION_new","HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION_new","HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION_new","HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION_new","HALLMARK_DNA_REPAIR_new","HALLMARK_DNA_REPAIR_new","HALLMARK_DNA_REPAIR_new","HALLMARK_DNA_REPAIR_new","HALLMARK_DNA_REPAIR_new","HALLMARK_DNA_REPAIR_new","HALLMARK_DNA_REPAIR_new","HALLMARK_ANDROGEN_RESPONSE_new","HALLMARK_ANDROGEN_RESPONSE_new","HALLMARK_ANDROGEN_RESPONSE_new","HALLMARK_ANDROGEN_RESPONSE_new","HALLMARK_ANDROGEN_RESPONSE_new","HALLMARK_ANDROGEN_RESPONSE_new","HALLMARK_ANDROGEN_RESPONSE_new","HALLMARK_PI3K_AKT_MTOR_SIGNALING_new","HALLMARK_PI3K_AKT_MTOR_SIGNALING_new","HALLMARK_PI3K_AKT_MTOR_SIGNALING_new","HALLMARK_PI3K_AKT_MTOR_SIGNALING_new","HALLMARK_PI3K_AKT_MTOR_SIGNALING_new","HALLMARK_PI3K_AKT_MTOR_SIGNALING_new","HALLMARK_PI3K_AKT_MTOR_SIGNALING_new","SET_ERPR_new","SET_ERPR_new","SET_ERPR_new","SET_ERPR_new","SET_ERPR_new","SET_ERPR_new","SET_ERPR_new","HALLMARK_ESTROGEN_RESPONSE_EARLY_new","HALLMARK_ESTROGEN_RESPONSE_EARLY_new","HALLMARK_ESTROGEN_RESPONSE_EARLY_new","HALLMARK_ESTROGEN_RESPONSE_EARLY_new","HALLMARK_ESTROGEN_RESPONSE_EARLY_new","HALLMARK_ESTROGEN_RESPONSE_EARLY_new","HALLMARK_ESTROGEN_RESPONSE_EARLY_new","HALLMARK_G2M_CHECKPOINT_new","HALLMARK_G2M_CHECKPOINT_new","HALLMARK_G2M_CHECKPOINT_new","HALLMARK_G2M_CHECKPOINT_new","HALLMARK_G2M_CHECKPOINT_new","HALLMARK_G2M_CHECKPOINT_new","HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION_new","HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION_new","HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION_new","HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION_new","HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION_new","HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION_new","HALLMARK_DNA_REPAIR_new","HALLMARK_DNA_REPAIR_new","HALLMARK_DNA_REPAIR_new","HALLMARK_DNA_REPAIR_new","HALLMARK_DNA_REPAIR_new","HALLMARK_DNA_REPAIR_new","HALLMARK_ANDROGEN_RESPONSE_new","HALLMARK_ANDROGEN_RESPONSE_new","HALLMARK_ANDROGEN_RESPONSE_new","HALLMARK_ANDROGEN_RESPONSE_new","HALLMARK_ANDROGEN_RESPONSE_new","HALLMARK_ANDROGEN_RESPONSE_new","HALLMARK_PI3K_AKT_MTOR_SIGNALING_new","HALLMARK_PI3K_AKT_MTOR_SIGNALING_new","HALLMARK_PI3K_AKT_MTOR_SIGNALING_new","HALLMARK_PI3K_AKT_MTOR_SIGNALING_new","HALLMARK_PI3K_AKT_MTOR_SIGNALING_new","HALLMARK_PI3K_AKT_MTOR_SIGNALING_new","SET_ERPR_new","SET_ERPR_new","SET_ERPR_new","SET_ERPR_new","SET_ERPR_new","SET_ERPR_new","HALLMARK_ESTROGEN_RESPONSE_EARLY_new","HALLMARK_ESTROGEN_RESPONSE_EARLY_new","HALLMARK_ESTROGEN_RESPONSE_EARLY_new","HALLMARK_ESTROGEN_RESPONSE_EARLY_new","HALLMARK_ESTROGEN_RESPONSE_EARLY_new","HALLMARK_ESTROGEN_RESPONSE_EARLY_new","HALLMARK_G2M_CHECKPOINT_new","HALLMARK_G2M_CHECKPOINT_new","HALLMARK_G2M_CHECKPOINT_new","HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION_new","HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION_new","HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION_new","HALLMARK_DNA_REPAIR_new","HALLMARK_DNA_REPAIR_new","HALLMARK_DNA_REPAIR_new","HALLMARK_ANDROGEN_RESPONSE_new","HALLMARK_ANDROGEN_RESPONSE_new","HALLMARK_ANDROGEN_RESPONSE_new","HALLMARK_PI3K_AKT_MTOR_SIGNALING_new","HALLMARK_PI3K_AKT_MTOR_SIGNALING_new","HALLMARK_PI3K_AKT_MTOR_SIGNALING_new","SET_ERPR_new","SET_ERPR_new","SET_ERPR_new","HALLMARK_ESTROGEN_RESPONSE_EARLY_new","HALLMARK_ESTROGEN_RESPONSE_EARLY_new","HALLMARK_ESTROGEN_RESPONSE_EARLY_new","HALLMARK_G2M_CHECKPOINT_new","HALLMARK_G2M_CHECKPOINT_new","HALLMARK_G2M_CHECKPOINT_new","HALLMARK_G2M_CHECKPOINT_new","HALLMARK_G2M_CHECKPOINT_new","HALLMARK_G2M_CHECKPOINT_new","HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION_new","HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION_new","HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION_new","HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION_new","HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION_new","HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION_new","HALLMARK_DNA_REPAIR_new","HALLMARK_DNA_REPAIR_new","HALLMARK_DNA_REPAIR_new","HALLMARK_DNA_REPAIR_new","HALLMARK_DNA_REPAIR_new","HALLMARK_DNA_REPAIR_new","HALLMARK_ANDROGEN_RESPONSE_new","HALLMARK_ANDROGEN_RESPONSE_new","HALLMARK_ANDROGEN_RESPONSE_new","HALLMARK_ANDROGEN_RESPONSE_new","HALLMARK_ANDROGEN_RESPONSE_new","HALLMARK_ANDROGEN_RESPONSE_new","HALLMARK_PI3K_AKT_MTOR_SIGNALING_new","HALLMARK_PI3K_AKT_MTOR_SIGNALING_new","HALLMARK_PI3K_AKT_MTOR_SIGNALING_new","HALLMARK_PI3K_AKT_MTOR_SIGNALING_new","HALLMARK_PI3K_AKT_MTOR_SIGNALING_new","HALLMARK_PI3K_AKT_MTOR_SIGNALING_new","SET_ERPR_new","SET_ERPR_new","SET_ERPR_new","SET_ERPR_new","SET_ERPR_new","SET_ERPR_new","HALLMARK_ESTROGEN_RESPONSE_EARLY_new","HALLMARK_ESTROGEN_RESPONSE_EARLY_new","HALLMARK_ESTROGEN_RESPONSE_EARLY_new","HALLMARK_ESTROGEN_RESPONSE_EARLY_new","HALLMARK_ESTROGEN_RESPONSE_EARLY_new","HALLMARK_ESTROGEN_RESPONSE_EARLY_new","HALLMARK_G2M_CHECKPOINT_new","HALLMARK_G2M_CHECKPOINT_new","HALLMARK_G2M_CHECKPOINT_new","HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION_new","HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION_new","HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION_new","HALLMARK_DNA_REPAIR_new","HALLMARK_DNA_REPAIR_new","HALLMARK_DNA_REPAIR_new","HALLMARK_ANDROGEN_RESPONSE_new","HALLMARK_ANDROGEN_RESPONSE_new","HALLMARK_ANDROGEN_RESPONSE_new","HALLMARK_PI3K_AKT_MTOR_SIGNALING_new","HALLMARK_PI3K_AKT_MTOR_SIGNALING_new","HALLMARK_PI3K_AKT_MTOR_SIGNALING_new","SET_ERPR_new","SET_ERPR_new","SET_ERPR_new","HALLMARK_ESTROGEN_RESPONSE_EARLY_new","HALLMARK_ESTROGEN_RESPONSE_EARLY_new","HALLMARK_ESTROGEN_RESPONSE_EARLY_new"],["age","node_stageN1","node_stageN2","node_stageN3","tumor_stagestage_ii","tumor_stagestage_iii","HALLMARK_G2M_CHECKPOINT_new","age","node_stageN1","node_stageN2","node_stageN3","tumor_stagestage_ii","tumor_stagestage_iii","HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION_new","age","node_stageN1","node_stageN2","node_stageN3","tumor_stagestage_ii","tumor_stagestage_iii","HALLMARK_DNA_REPAIR_new","age","node_stageN1","node_stageN2","node_stageN3","tumor_stagestage_ii","tumor_stagestage_iii","HALLMARK_ANDROGEN_RESPONSE_new","age","node_stageN1","node_stageN2","node_stageN3","tumor_stagestage_ii","tumor_stagestage_iii","HALLMARK_PI3K_AKT_MTOR_SIGNALING_new","age","node_stageN1","node_stageN2","node_stageN3","tumor_stagestage_ii","tumor_stagestage_iii","SET_ERPR_new","age","node_stageN1","node_stageN2","node_stageN3","tumor_stagestage_ii","tumor_stagestage_iii","HALLMARK_ESTROGEN_RESPONSE_EARLY_new","age","node_stageN1","node_stageN2and3","tumor_stageT2","tumor_stageT3","HALLMARK_G2M_CHECKPOINT_new","age","node_stageN1","node_stageN2and3","tumor_stageT2","tumor_stageT3","HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION_new","age","node_stageN1","node_stageN2and3","tumor_stageT2","tumor_stageT3","HALLMARK_DNA_REPAIR_new","age","node_stageN1","node_stageN2and3","tumor_stageT2","tumor_stageT3","HALLMARK_ANDROGEN_RESPONSE_new","age","node_stageN1","node_stageN2and3","tumor_stageT2","tumor_stageT3","HALLMARK_PI3K_AKT_MTOR_SIGNALING_new","age","node_stageN1","node_stageN2and3","tumor_stageT2","tumor_stageT3","SET_ERPR_new","age","node_stageN1","node_stageN2and3","tumor_stageT2","tumor_stageT3","HALLMARK_ESTROGEN_RESPONSE_EARLY_new","age","npi","HALLMARK_G2M_CHECKPOINT_new","age","npi","HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION_new","age","npi","HALLMARK_DNA_REPAIR_new","age","npi","HALLMARK_ANDROGEN_RESPONSE_new","age","npi","HALLMARK_PI3K_AKT_MTOR_SIGNALING_new","age","npi","SET_ERPR_new","age","npi","HALLMARK_ESTROGEN_RESPONSE_EARLY_new","age","node_stageN1","node_stageN2and3","tumor_stageT2","tumor_stageT3","HALLMARK_G2M_CHECKPOINT_new","age","node_stageN1","node_stageN2and3","tumor_stageT2","tumor_stageT3","HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION_new","age","node_stageN1","node_stageN2and3","tumor_stageT2","tumor_stageT3","HALLMARK_DNA_REPAIR_new","age","node_stageN1","node_stageN2and3","tumor_stageT2","tumor_stageT3","HALLMARK_ANDROGEN_RESPONSE_new","age","node_stageN1","node_stageN2and3","tumor_stageT2","tumor_stageT3","HALLMARK_PI3K_AKT_MTOR_SIGNALING_new","age","node_stageN1","node_stageN2and3","tumor_stageT2","tumor_stageT3","SET_ERPR_new","age","node_stageN1","node_stageN2and3","tumor_stageT2","tumor_stageT3","HALLMARK_ESTROGEN_RESPONSE_EARLY_new","age","npi","HALLMARK_G2M_CHECKPOINT_new","age","npi","HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION_new","age","npi","HALLMARK_DNA_REPAIR_new","age","npi","HALLMARK_ANDROGEN_RESPONSE_new","age","npi","HALLMARK_PI3K_AKT_MTOR_SIGNALING_new","age","npi","SET_ERPR_new","age","npi","HALLMARK_ESTROGEN_RESPONSE_EARLY_new"],[0.0616377905840081,0.474306853167729,0.954042639241222,0.828955528976151,0.30423497669931,0.470034536811477,0.176153627760754,0.0618675885495185,0.387304547107589,1.01057220491273,0.690761668617657,0.399214859008978,0.505444029834159,0.130498921140544,0.0601662507909038,0.424311209876691,1.06766923308836,0.770167322699961,0.352492428430478,0.426883159333406,-0.0338415059102302,0.0606752369775765,0.427607873754539,1.07795451161204,0.764714585857082,0.357347071371586,0.429233939684253,0.049712265012628,0.062044296894465,0.491664232773412,1.17924009070085,0.937849378395496,0.395316611048624,0.403186797666766,0.180676610064079,0.0602291166978039,0.518466022382417,1.25129356321115,1.07208088242748,0.34162015562071,0.27266068577286,-0.398117441380964,0.0589267518980507,0.467215038072016,1.11839404761123,0.804951179735225,0.305224671782702,0.311986948544274,-0.153544815154153,0.0877546231962248,-0.00443048343256998,0.691749196715961,0.309124689172904,0.389256224976618,0.142924159656714,0.0909298696370283,0.0103778572918325,0.7099569643393771,0.353628557025232,0.402227020135157,0.0478910973264077,0.08822808043915779,0.0127461156425051,0.7311100029615339,0.323352005450829,0.380623469988135,0.0672133035593898,0.0901203101680311,0.0160285949029942,0.722822320956058,0.344403532933293,0.380981879247321,0.0216952214364992,0.08983741147324339,0.0135514064667183,0.715618434507387,0.339117875993968,0.386686328221643,0.0263567156502282,0.0899037945364161,0.0145799982530732,0.74544209198291,0.315132975551969,0.266770731378847,-0.189260056065385,0.08913234708588751,0.00442766686563733,0.733525548164808,0.322892628936288,0.406519586043661,-0.184846429087069,0.054796369073002,0.202884212883009,0.138193421517684,0.0545592933637651,0.226835099658034,4.73645807183829e-05,0.0545304687923254,0.225142427190362,0.0386085632262308,0.0538741584736043,0.228117685319517,0.0753258460397767,0.0543577423173955,0.21678892012922,0.0706450532302673,0.0553708476647072,0.214201716505237,-0.110047770769172,0.0551303773965216,0.221920253020793,-0.0843760067281521,0.0100627540567237,-0.127547982709042,1.16608745987939,0.734981661365082,1.03928630273352,0.341652474701783,0.0184001067077129,-0.141121299365367,1.16163588112556,0.849272110899908,0.98983392765093,0.165946725281887,0.0117416403950703,-0.110722062541095,1.23483040087918,0.775645315118437,0.968301109321701,0.14309699524062,0.0158250406458957,-0.125448460472345,1.20079283782718,0.8296158554936049,0.903381651078438,0.10430578486252,0.0148848685352247,-0.126313336346132,1.19306744975332,0.813340101325531,0.9529000081060059,0.0296628107181683,0.0144877737893204,-0.100932996230727,1.22973600362814,0.748424974574284,0.550319607017098,-0.398255102696566,0.0124810207453315,-0.105858755761456,1.21450697812095,0.768130700526372,0.934845017584215,-0.393978953395248,0.00713162732398492,0.335264730640554,0.285024422145762,0.00652840395830359,0.387769631918236,-0.0366441667297294,0.00663490084222653,0.37902119548141,0.139357411564226,0.00679052209098185,0.389871556151544,0.0607663626894693,0.0062291951380722,0.364100093398194,0.160537500194618,0.00776535216371423,0.371156901929141,-0.167246703185582,0.00771620673845139,0.378896708435262,-0.140954313892214],[0.0107199609781805,0.335128183823087,0.578836969575005,0.702473698806618,0.381934926386708,0.546724758918736,0.11858917411011,0.0106774977627038,0.337790951312884,0.580766851712013,0.707499557020563,0.386181403323284,0.553657907855745,0.121832192810099,0.0106341833842427,0.335078152263452,0.584765710377114,0.704332724224953,0.383469397907644,0.548503811543298,0.118901210661966,0.0107541130417584,0.334489329458643,0.585016015133896,0.704430579347454,0.383428076457855,0.548189339236448,0.121919705507839,0.0107978538970555,0.336667367342697,0.587149821419122,0.7136754478413621,0.383627093791693,0.547675690885109,0.118373797632042,0.0106619183004573,0.337155354245839,0.585851656364178,0.715250026338379,0.383824429425781,0.5510361276590729,0.11779764876355,0.0107832047430927,0.336449903237397,0.582968452237609,0.705280115745491,0.3835387564148,0.556022213499862,0.121921444471585,0.00513403870814099,0.102884644781799,0.14298806203984,0.0943708565437784,0.214050552593927,0.0399740615505735,0.00518302029841512,0.102598935411182,0.142994633856377,0.0943420027599613,0.214590054804199,0.0423542520548911,0.0051897740326235,0.102638844743173,0.142845341582877,0.0946587184244726,0.214138996515675,0.0421907249548449,0.00512914505098728,0.102687809280195,0.142807205543512,0.0939313711730355,0.21456945849856,0.0441160726286537,0.00509597331621979,0.102603947836456,0.143003527556099,0.0939706024274093,0.214349793568774,0.0415244429644247,0.00511253128948482,0.102540887055447,0.143536104884207,0.0938395147307023,0.216358070043775,0.0374531875895836,0.00511167984216038,0.102744052140275,0.14309255214418,0.09392947927132581,0.214444276124612,0.041031612449989,0.00463386455279242,0.0412516999447976,0.0441404039142059,0.00463807368630181,0.0402023404400431,0.0419589406500783,0.00461050175932014,0.0403485209901898,0.0402709518825079,0.00461992468633567,0.0399836142696403,0.0413240652523271,0.00462302074761757,0.0404172999464367,0.0379513142015799,0.00462680207492856,0.0404985121965301,0.0424577930714206,0.00461787453257413,0.0403665074484356,0.0426547700562495,0.0075667479508169,0.193878643012837,0.243959091568216,0.17048667439544,0.358439742322712,0.0701625364399097,0.00765501126281157,0.193018818654386,0.245944649558463,0.169718587872003,0.3604396516307,0.08133158495080239,0.00762699338191024,0.193380695615835,0.24574856765051,0.170423860878981,0.359884195118012,0.0748798456196226,0.00753575849277249,0.192970638099287,0.245779026718092,0.169204200291695,0.363077904994236,0.0786083781600346,0.00746914460777838,0.193032099431133,0.246369565583424,0.168948883007663,0.360837137066073,0.0739430373478525,0.00760254813151414,0.19307130030412,0.247283470832396,0.169517825192695,0.370995637272675,0.0588754961297943,0.00766291759829686,0.193969614194822,0.245949572845819,0.169132536726326,0.359516258248299,0.0635073274552802,0.00537611783369784,0.0543011581473659,0.0552141810825526,0.00540647443675992,0.052402044606779,0.0547038606792418,0.0053169540899735,0.0529000920904637,0.0511748995122641,0.00530446621124086,0.0522270257240905,0.0537943842840951,0.00535108171458861,0.0529017981884821,0.0481135255001357,0.00532367835040025,0.0527370313787425,0.0509516572264319,0.00531027131001208,0.0526262124769372,0.0516056047307676],[5.74981482763476,1.41529980485949,1.64820612605601,1.18005205089444,0.796562334787033,0.859727914537966,1.48541069690893,5.79420290450636,1.14658058660915,1.74006522915988,0.976342192391762,1.03374956839851,0.912917566357333,1.07113660298271,5.65781580182788,1.26630520972636,1.8258068387762,1.09347087847928,0.919219187642643,0.778268355387187,-0.284618682365152,5.64204939468029,1.278390178983,1.84260684105429,1.0855783497722,0.931979407123214,0.783003077517188,0.407745940703833,5.7459841081369,1.46038577083992,2.00841428828278,1.3141118714847,1.03047104192093,0.736178005299391,1.52632266327807,5.6489943929903,1.537765946331,2.13585393097075,1.49888967906208,0.890042762863711,0.494814535902,-3.37967222232159,5.4646789430384,1.38866153200333,1.9184469473751,1.14132124494164,0.7958118095700339,0.561105187831405,-1.25937496737859,17.092707746257,-0.0430626304048216,4.837810841322,3.27563720934865,1.81852473754212,3.57542251431951,17.5437996383756,0.101149756089003,4.96492032737712,3.74836813592972,1.87439730374348,1.13072702274003,17.0003703214333,0.124184130037696,5.11819282911199,3.415977004895,1.77745985636144,1.59308245192102,17.5702401223152,0.156090533193268,5.06152555961765,3.66654429326759,1.77556434132436,0.491775902608519,17.6290977009835,0.132074903085779,5.0042012720746,3.60876558449149,1.80399673721904,0.634727735488441,17.5849866623458,0.14218716720472,5.19341173835163,3.35821190525471,1.23300568971091,-5.05324294795195,17.4369971982081,0.0430941429056381,5.12623149963605,3.43760693065887,1.89568867675179,-4.50497599411593,11.825198697269,4.91820247782529,3.13076930121178,11.7633519978136,5.64233567437028,0.00112883166220486,11.8274477787785,5.57994250260381,0.958719906568705,11.6612633606231,5.70527926217835,1.82280822517903,11.7580571848846,5.36376552655722,1.86146526718504,11.9674122143127,5.28912557245967,-2.59193337213865,11.938474509785,5.49763323726421,-1.97811420895914,1.32986510481525,-0.657875363304437,4.77984834417752,4.31107981882684,2.89947285420663,4.8694430395114,2.40366814312887,-0.73112715303711,4.72315979717797,5.00400175106574,2.74618489717412,2.0403724504111,1.53948480182496,-0.5725600592576831,5.02477150806954,4.55127181791304,2.69059081353706,1.91102150460525,2.09999307449587,-0.650090924235837,4.88566031797533,4.90304527939266,2.48812069986131,1.32690417108173,1.99284781817232,-0.654364412542674,4.84259265923545,4.81411943569133,2.64080359314989,0.401157590790119,1.90564709867019,-0.522775762486402,4.97298100632707,4.41502227699966,1.48335870217423,-6.76436087805681,1.62875570371597,-0.54574916901743,4.9380324757966,4.54159037281683,2.60028578996438,-6.20367710596348,1.32653850689124,6.17417274472657,5.16215973066072,1.20751592089577,7.39989507714881,-0.669864361943189,1.24787627087816,7.16484944550288,2.72315945692924,1.28015182311688,7.46493890368544,1.12960420493995,1.16410017082894,6.88256554344246,3.33663971878688,1.45864412772616,7.03788006692294,-3.28245855561338,1.45307203492656,7.19977157013359,-2.73137607102153],[8.93412441127214e-09,0.156980612204637,0.0993103801666517,0.237979513539687,0.425705259648727,0.389939043486953,0.137435063773138,6.86464254171465e-09,0.251555003694479,0.0818475648797358,0.328894917910836,0.301253253593842,0.361285897987581,0.284108012208059,1.53311544401343e-08,0.205403832497456,0.0678793628322899,0.274187108351987,0.357980936922057,0.436410826761087,0.775936296543184,1.68037872360329e-08,0.201111885155347,0.06538643437351491,0.277665582620934,0.351347167889808,0.433625304834179,0.683460205661326,9.1387890172828e-09,0.144184081198564,0.0445992858111635,0.188808559690152,0.302788939172516,0.461622375768815,0.126929527217093,1.61389119169928e-08,0.124105852729266,0.0326913216465551,0.133902254314543,0.373442924786211,0.620731042723669,0.00072572323229897,4.63745365109287e-08,0.164935696889028,0.0550543630238187,0.253736264399819,0.426141426546249,0.574725831769763,0.20789492675888,1.6818001688998e-65,0.965651608266851,1.31276984489884e-06,0.00105423890168865,0.0689839693135163,0.000349662730068857,6.63397088903683e-69,0.919431580902718,6.87292790917023e-07,0.000177988874615348,0.0608756848540035,0.25817000393558,8.160279379587779e-65,0.901169487362546,3.08477131604852e-07,0.000635536134665469,0.0754926100798111,0.11114169542842,4.16406220081448e-69,0.875961660318148,4.1591496756996e-07,0.000245850351009487,0.0758047517698371,0.62287777348636,1.47298977698312e-69,0.894925046428713,5.60941247747605e-07,0.000307657448718158,0.07123181844745841,0.525606001290211,3.21057007486975e-69,0.8869321678884849,2.06474850656216e-07,0.00078448448774135,0.217573641405566,4.34370792729119e-07,4.32193306241412e-68,0.965626488248,2.95599109590522e-07,0.0005868789467009449,0.058001223489763,6.6380446183486e-06,2.89217570995343e-32,8.73425432634149e-07,0.00174349063497393,6.02921635622353e-32,1.67758622936334e-08,0.999099322836264,2.81573448532543e-32,2.40598104270752e-08,0.337699867316002,2.0103708442477e-31,1.16152494628369e-08,0.0683324415411018,6.41944523059606e-32,8.15046920263831e-08,0.0626785011218377,5.2645065594342e-33,1.22902496189732e-07,0.00954382531051436,7.45784405744468e-33,3.84922462436636e-08,0.0479158239830879,0.183562720476315,0.510618222733792,1.75427473174065e-06,1.62459244568604e-05,0.00373790710467898,1.11913236744137e-06,0.0162315005917338,0.464701490395904,2.32208249625826e-06,5.61522463953328e-07,0.00602927731015427,0.041313243695926,0.123685985419331,0.566942609571334,5.04032267685885e-07,5.33226063273654e-06,0.00713256192849929,0.0560018174044757,0.0357294503472146,0.515633491317331,1.03082787424326e-06,9.43622222326352e-07,0.0128420140821225,0.184540384426316,0.0462781137680327,0.512877056334737,1.2815583993527e-06,1.47850366659441e-06,0.00827096522031644,0.688304102089076,0.0566960070908614,0.6011303082609319,6.593112296805769e-07,1.00999588874723e-05,0.137979133564037,1.338989527249e-11,0.103364745381315,0.585238368432073,7.8914703199388e-07,5.58314429839566e-06,0.00931461517868535,5.51589007825469e-10,0.184661389115186,6.65107902803796e-10,2.44116845876432e-07,0.227233517465319,1.36292143302698e-13,0.502944259891563,0.212076371760296,7.78720866905022e-13,0.00646608401487375,0.200491745687089,8.3338272195096e-14,0.258643038377729,0.244383419432687,5.87841028298878e-12,0.000847977942838704,0.144663081444803,1.95186573461213e-12,0.00102906111367971,0.146203754592135,6.03135167910549e-13,0.00630704564046907],[1.06357703733371,1.60689999373656,2.59618390819077,2.29092468576945,1.35558755015702,1.60004945286942,1.19262126431184,1.06382147325741,1.47300502238927,2.74717251097207,1.9952346622603,1.49065386461438,1.65772143483858,1.13939671064398,1.06201309238607,1.52853721564951,2.90859234261705,2.16012766194083,1.42260888432578,1.53247359602665,0.966724712649293,1.06255377996974,1.53358460427492,2.93866239934253,2.14838110878918,1.42953193337302,1.53608034251781,1.05096865241433,1.06400947599961,1.63503503584894,3.25190211307095,2.55448178292098,1.48485423776325,1.49658642440693,1.19802768654417,1.06207985890091,1.67944943489955,3.49486085817101,2.92145237822941,1.40722566922191,1.31345449540746,0.671583151377701,1.06069754381279,1.59554446933606,3.05993614350592,2.23658730537014,1.35692983260613,1.36613686283834,0.857662327068153,1.09172020634455,0.995579316680731,1.99720598576046,1.36223221552254,1.47588266089071,1.1536423057197,1.09519219633893,1.01043189401953,2.03390372637275,1.42422606944959,1.49515072346963,1.0490564040563,1.09223721156377,1.01282769360675,2.07738523212742,1.38175164942303,1.46319656425477,1.06952358703349,1.09430593191629,1.01615774192272,2.06023967053956,1.41114796530604,1.4637210814417,1.02193227395154,1.09399639798189,1.01364364294908,2.04545126883409,1.40370879903094,1.47209466429462,1.02670712565892,1.0940690232244,1.01468680487644,2.107372881729,1.37044153406055,1.30574104656115,0.82757126378165,1.09322533193932,1.00443748346544,2.08240931556766,1.38111705093734,1.50158255113923,0.831231927096632,1.05632549223425,1.22493062885004,1.14819761492273,1.05607509280198,1.25462296312575,1.00004736570244,1.05604465232874,1.25250109370036,1.03936355889443,1.05535178671868,1.25623315693384,1.07823543239527,1.05586226121092,1.24208189597942,1.07320022930094,1.05693250295605,1.23887253062788,0.895791341632985,1.05667837267022,1.24847181203866,0.919085608743697,1.01011355381851,0.880251177984729,3.20941109201453,2.08544374796282,2.8271985302811,1.40727115019735,1.01857043173356,0.868383970884347,3.19515589838428,2.33794446813288,2.69078756982864,1.18051020869939,1.01181084404421,0.895187520500652,3.4377954241787,2.17199329514789,2.633466684804,1.1538417133931,1.01595091973895,0.8821012263360301,3.32275028079856,2.29243794365612,2.46793471001074,1.10993980581968,1.01499619988921,0.881338648082434,3.29717964412726,2.25542878023671,2.59321912177573,1.03010713430226,1.01459323024498,0.903993601835389,3.42032646331402,2.11366831220859,1.73380706622847,0.671490706720718,1.0125592337387,0.899551695597624,3.36863284515142,2.15573277492802,2.54681871503369,0.674368249253086,1.00715711793862,1.39831051186587,1.32979450416012,1.0065497604367,1.47369025398425,0.964019104396056,1.00665696055784,1.46085399891907,1.14953488365072,1.00681362996134,1.47679109696834,1.06265061035117,1.00624863692203,1.43921826319545,1.17414180286117,1.00779558070526,1.44941047084562,0.845990876946212,1.0077460533799,1.46067215283886,0.868528990500351]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>type_analysis<\/th>\n      <th>cohort<\/th>\n      <th>score<\/th>\n      <th>term<\/th>\n      <th>estimate<\/th>\n      <th>std.error<\/th>\n      <th>statistic<\/th>\n      <th>p.value<\/th>\n      <th>HR<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"columnDefs":[{"targets":8,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatSignif(data, 2, 3, \",\", \".\", null);\n  }"},{"targets":5,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 2, 3, \",\", \".\", null);\n  }"},{"targets":6,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 2, 3, \",\", \".\", null);\n  }"},{"targets":7,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 2, 3, \",\", \".\", null);\n  }"},{"targets":9,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 2, 3, \",\", \".\", null);\n  }"},{"className":"dt-right","targets":[5,6,7,8,9]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":["options.columnDefs.0.render","options.columnDefs.1.render","options.columnDefs.2.render","options.columnDefs.3.render","options.columnDefs.4.render"],"jsHooks":[]}</script>
</div>
</div>
</section>
</section>
<section id="regressing-the-pdxs" class="level2" data-number="6.4">
<h2 data-number="6.4" class="anchored" data-anchor-id="regressing-the-pdxs"><span class="header-section-number">6.4</span> Regressing the PDXs</h2>
<p>We can use the regressing strategy to score the PDXs and then evaluate their scores.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>Total number of stable genes: 39
Total number of genes: 8205
Number of samples: 76
[1] "Normalization done."</code></pre>
</div>
</div>
<p>We start by comparing the scores across the different control samples and their positions in the molecular landscape.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pca-pdx-control-only-regressed" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-pca-pdx-control-only-regressed-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="expanding_ember_files/figure-html/fig-pca-pdx-control-only-regressed-1.png" class="img-fluid figure-img" width="864">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-pca-pdx-control-only-regressed-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;6.13: Embedding of only regressed PDX control samples on top of the METABRIC, SCANB and TCGA samples
</figcaption>
</figure>
</div>
</div>
</div>
<p>The results are jus twhat we got previously as well, meaning that by using all these genes the information of the PDXs are still captured. The main differences is that the luminal B region is more compact compare to <a href="validation.html#fig-pca-pdx-control-only" class="quarto-xref">Figure&nbsp;<span>3.20</span></a>.</p>
<p>And now that we have the results we can compare the PDXs as well.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pdx-regressed-scores" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-pdx-regressed-scores-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="expanding_ember_files/figure-html/fig-pdx-regressed-scores-1.png" class="img-fluid figure-img" width="960">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-pdx-regressed-scores-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;6.14: Scores obtained from the regressed data of the PDXs.
</figcaption>
</figure>
</div>
</div>
</div>
<p>The random 200 is just a gene set with 200 random genes that works as a control and serves as a comparison to the other pathways.</p>
<p>And below we show the distribution of some of the scores calculated for all the other cohorts as a comparison.</p>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="expanding_ember_files/figure-html/unnamed-chunk-57-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
<p>We see that METS15 is the one with the highest proliferation and lowest EMT, it is a PDX that comes from a pleural efusion. Also it is the PDX with the lowest apoptosis and highest DNA repair, probably due to a higher replication rate. Moreover, all these PDXs they have a estrogen response early scores higher than 0 in average, which is about the average of the scores in the ER+ BC samples.</p>
<p>Now we check what happens with some of the PDXs when they are treated with P4, to compare with <a href="validation.html#fig-pdx-p4-control-pathways" class="quarto-xref">Figure&nbsp;<span>3.23</span></a>.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pdx-p4-control-pathways-regressed." class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-pdx-p4-control-pathways-regressed.-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="expanding_ember_files/figure-html/fig-pdx-p4-control-pathways-regressed.-1.png" class="img-fluid figure-img" width="1248">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-pdx-p4-control-pathways-regressed.-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;6.15: Comparison of position in the molecular landscape between CTRL, E2 and P4 treated samples with the regressed data
</figcaption>
</figure>
</div>
</div>
</div>
<p>The results are very similar to what was seen previously using the GSVA. What is interesting here is that we have an absolute scale, so we can compare the scores.</p>
</section>
<section id="singscore" class="level2" data-number="6.5">
<h2 data-number="6.5" class="anchored" data-anchor-id="singscore"><span class="header-section-number">6.5</span> singscore</h2>
<p>We can try to use singscore instead, it is a sample dependent measure so it does not estimate the distribution of the gene expression levels before calculating the enrichment score.</p>
<p>We start by comparing with the GSVA scores.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-all-pathways-gsva-singscores" class="quarto-figure quarto-figure-center quarto-float anchored" width="1920">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-all-pathways-gsva-singscores-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="expanding_ember_files/figure-html/fig-all-pathways-gsva-singscores-1.png" id="fig-all-pathways-gsva-singscores" class="img-fluid figure-img" width="1920">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig quarto-uncaptioned" id="fig-all-pathways-gsva-singscores-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;6.16
</figcaption>
</figure>
</div>
</div>
</div>
<p>We see that in all cases there is a correlation, but the interpretation changes. In some of the pathways there is difference in the singscores based on the cohort, this is not seen on GSVA.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-selected-pathways-gsva-singscores" class="quarto-figure quarto-figure-center quarto-float anchored" width="1152">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-selected-pathways-gsva-singscores-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="expanding_ember_files/figure-html/fig-selected-pathways-gsva-singscores-1.png" id="fig-selected-pathways-gsva-singscores" class="img-fluid figure-img" width="1152">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig quarto-uncaptioned" id="fig-selected-pathways-gsva-singscores-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;6.17
</figcaption>
</figure>
</div>
</div>
</div>
<p>The next figure shows the same comparison but instead of using singscore, we used ssGSEA.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Calculating ranks..."
[1] "Calculating absolute values from ranks..."
[1] "Calculating ranks..."
[1] "Calculating absolute values from ranks..."
[1] "Calculating ranks..."
[1] "Calculating absolute values from ranks..."
[1] "Calculating ranks..."
[1] "Calculating absolute values from ranks..."</code></pre>
</div>
</div>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="expanding_ember_files/figure-html/unnamed-chunk-63-1.png" class="img-fluid figure-img" width="1152"></p>
</figure>
</div>
</div>
</div>
<p>The picture is similar as before, but now we actually look at the distributions across the 3 big cohorts. We used ssGSEA without the normalization, as that would be necessary in the case of an individual sample.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-ssgsea-dists" class="quarto-figure quarto-figure-center quarto-float anchored" width="1152">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-ssgsea-dists-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="expanding_ember_files/figure-html/fig-ssgsea-dists-1.png" id="fig-ssgsea-dists" class="img-fluid figure-img" width="1152">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig quarto-uncaptioned" id="fig-ssgsea-dists-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;6.18
</figcaption>
</figure>
</div>
</div>
</div>
<p>We see that overall there is a discrepancy, sometimes big, depending on the pathway, including the random genes pathway. Next we show the same distribution for singscore. Similarly there is a big discrepancy in several cases.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-singscore-dists" class="quarto-figure quarto-figure-center quarto-float anchored" width="960">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-singscore-dists-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="expanding_ember_files/figure-html/fig-singscore-dists-1.png" id="fig-singscore-dists" class="img-fluid figure-img" width="960">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig quarto-uncaptioned" id="fig-singscore-dists-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;6.19
</figcaption>
</figure>
</div>
</div>
</div>
<p>But now, <a href="#fig-regressed-scores-dists" class="quarto-xref">Figure&nbsp;<span>6.20</span></a> shows that when we calculate the scores with the regressed data using our methodology the distributions are matched and adjusted, with the expection of G2M, which seems to be something more fundamental instead of batch effect. Note that all the distributions are overlapping significantly.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-regressed-scores-dists" class="quarto-figure quarto-figure-center quarto-float anchored" width="960">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-regressed-scores-dists-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="expanding_ember_files/figure-html/fig-regressed-scores-dists-1.png" id="fig-regressed-scores-dists" class="img-fluid figure-img" width="960">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig quarto-uncaptioned" id="fig-regressed-scores-dists-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;6.20
</figcaption>
</figure>
</div>
</div>
</div>
<p>The next two figures are the distributions of the singscore and GSVA filled by cohort.</p>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="expanding_ember_files/figure-html/unnamed-chunk-67-1.png" class="img-fluid figure-img" width="1920"></p>
</figure>
</div>
</div>
</div>
<p>For several pathways in the singscore the distributions are similar, but that is not always the case.</p>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="expanding_ember_files/figure-html/unnamed-chunk-68-1.png" class="img-fluid figure-img" width="1920"></p>
</figure>
</div>
</div>
</div>
<p>On the other hand GSVA gives very similar distributions for all the different cohorts. Therefore we actually need to use GSVA. It is important in the end to calculate scores based on the cohort.</p>
<p>We now move on and try the same analysis on the regressed data.</p>
<p>We start by comparing with the GSVA scores.</p>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="expanding_ember_files/figure-html/unnamed-chunk-70-1.png" class="img-fluid figure-img" width="1920"></p>
</figure>
</div>
</div>
</div>
<p>We see that in all cases there is a correlation, but the interpretation changes. In some of the pathways there is difference in the singscores based on the cohort, this is not seen on GSVA.</p>
<p>The next two figures are the distributions of the singscore and GSVA filled by cohort.</p>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="expanding_ember_files/figure-html/unnamed-chunk-71-1.png" class="img-fluid figure-img" width="1920"></p>
</figure>
</div>
</div>
</div>
<p>In the end the best way of scoring seems to be using GSVA on the unregressed data though it depends on the pathways. For example the hallmark estrogen response early and set ER/PR pathways seem to have been regressed quite well, so we can actually compare the results using the scores. On the other hand we can’t use the pathways G2M checkpoint, E2F targets and EMT. For the PI3K signaling score it seems it is comparable to SCANB only.</p>


<!-- -->

</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="../chapters/risk_score.html" class="pagination-link" aria-label="Risk score model">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Risk score model</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../chapters/trying.html" class="pagination-link" aria-label="Tests and tries">
        <span class="nav-page-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Tests and tries</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb7" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu"># Expanding embedding and removing batch effects </span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>We have used some numbers of genes for doing the embedding. Here we try to </span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>perform the embedding with all the genes (~10000 genes) and then</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>we try to regress out the components. </span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a><span class="in">```{r setup}</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a><span class="in">renv::restore()</span></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a><span class="in">library(tidyverse)</span></span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a><span class="in">library(ggplot2)</span></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a><span class="in">library(PCAtools)</span></span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a><span class="in">library(singscore)</span></span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a><span class="in">library(SummarizedExperiment)</span></span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a><span class="in">library(survival)</span></span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a><span class="in">library(genefu)</span></span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a><span class="in">library(caret)</span></span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-21"><a href="#cb7-21" aria-hidden="true" tabindex="-1"></a><span class="in">source("../R/utils.R")</span></span>
<span id="cb7-22"><a href="#cb7-22" aria-hidden="true" tabindex="-1"></a><span class="in">source("../R/first_run.R")</span></span>
<span id="cb7-23"><a href="#cb7-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-24"><a href="#cb7-24" aria-hidden="true" tabindex="-1"></a><span class="in"># the following script load all data necessary to run the chunks.</span></span>
<span id="cb7-25"><a href="#cb7-25" aria-hidden="true" tabindex="-1"></a><span class="in"># the data is generated from this quarto document itself, therefore</span></span>
<span id="cb7-26"><a href="#cb7-26" aria-hidden="true" tabindex="-1"></a><span class="in"># if you are running this documents the first time and don't have the</span></span>
<span id="cb7-27"><a href="#cb7-27" aria-hidden="true" tabindex="-1"></a><span class="in"># files, comment the following lines. Moreover, if this is your first</span></span>
<span id="cb7-28"><a href="#cb7-28" aria-hidden="true" tabindex="-1"></a><span class="in"># time running the document, you should run all chunks, to generate </span></span>
<span id="cb7-29"><a href="#cb7-29" aria-hidden="true" tabindex="-1"></a><span class="in"># all the necessary files, if you don't have them. Once all files </span></span>
<span id="cb7-30"><a href="#cb7-30" aria-hidden="true" tabindex="-1"></a><span class="in"># are saved and available in the respective folder, the following</span></span>
<span id="cb7-31"><a href="#cb7-31" aria-hidden="true" tabindex="-1"></a><span class="in"># lines can be executed. </span></span>
<span id="cb7-32"><a href="#cb7-32" aria-hidden="true" tabindex="-1"></a><span class="in">if (first_run){</span></span>
<span id="cb7-33"><a href="#cb7-33" aria-hidden="true" tabindex="-1"></a><span class="in">    load_at_setup &lt;- FALSE</span></span>
<span id="cb7-34"><a href="#cb7-34" aria-hidden="true" tabindex="-1"></a><span class="in">} else {</span></span>
<span id="cb7-35"><a href="#cb7-35" aria-hidden="true" tabindex="-1"></a><span class="in">    load_at_setup &lt;- TRUE</span></span>
<span id="cb7-36"><a href="#cb7-36" aria-hidden="true" tabindex="-1"></a><span class="in">}</span></span>
<span id="cb7-37"><a href="#cb7-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-38"><a href="#cb7-38" aria-hidden="true" tabindex="-1"></a><span class="in">name_document &lt;- "expanding_ember"</span></span>
<span id="cb7-39"><a href="#cb7-39" aria-hidden="true" tabindex="-1"></a><span class="in">sapply(</span></span>
<span id="cb7-40"><a href="#cb7-40" aria-hidden="true" tabindex="-1"></a><span class="in">    paste0("../../results/", c("plots", "rds_files", "tables"), "/", name_document),</span></span>
<span id="cb7-41"><a href="#cb7-41" aria-hidden="true" tabindex="-1"></a><span class="in">    dir.create,</span></span>
<span id="cb7-42"><a href="#cb7-42" aria-hidden="true" tabindex="-1"></a><span class="in">    showWarnings = FALSE,</span></span>
<span id="cb7-43"><a href="#cb7-43" aria-hidden="true" tabindex="-1"></a><span class="in">    recursive = TRUE</span></span>
<span id="cb7-44"><a href="#cb7-44" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-45"><a href="#cb7-45" aria-hidden="true" tabindex="-1"></a><span class="in">source("../R/load_rds_files.R")</span></span>
<span id="cb7-46"><a href="#cb7-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-47"><a href="#cb7-47" aria-hidden="true" tabindex="-1"></a><span class="in"># by setting the dev to png and pdf, this saves the figures in a specific</span></span>
<span id="cb7-48"><a href="#cb7-48" aria-hidden="true" tabindex="-1"></a><span class="in"># folder in both formats. moreover, since png is coming first, it shows</span></span>
<span id="cb7-49"><a href="#cb7-49" aria-hidden="true" tabindex="-1"></a><span class="in"># this figure when rendering the html. What is nice about this is that it</span></span>
<span id="cb7-50"><a href="#cb7-50" aria-hidden="true" tabindex="-1"></a><span class="in"># inherits the properties from the chunk to save the figure, so no need</span></span>
<span id="cb7-51"><a href="#cb7-51" aria-hidden="true" tabindex="-1"></a><span class="in"># to use ggplot2::ggsave to save the plots. this also works </span></span>
<span id="cb7-52"><a href="#cb7-52" aria-hidden="true" tabindex="-1"></a><span class="in">knitr::opts_chunk$set(dev = c('png', 'pdf'))</span></span>
<span id="cb7-53"><a href="#cb7-53" aria-hidden="true" tabindex="-1"></a><span class="in">options(bitmapType = 'cairo')</span></span>
<span id="cb7-54"><a href="#cb7-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-55"><a href="#cb7-55" aria-hidden="true" tabindex="-1"></a><span class="in">p4_pathways &lt;- msigdbr::msigdbr() %&gt;%</span></span>
<span id="cb7-56"><a href="#cb7-56" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::filter(gs_name %in% c(</span></span>
<span id="cb7-57"><a href="#cb7-57" aria-hidden="true" tabindex="-1"></a><span class="in">        "WILCOX_RESPONSE_TO_PROGESTERONE_UP",</span></span>
<span id="cb7-58"><a href="#cb7-58" aria-hidden="true" tabindex="-1"></a><span class="in">        "GOBP_RESPONSE_TO_PROGESTERONE",</span></span>
<span id="cb7-59"><a href="#cb7-59" aria-hidden="true" tabindex="-1"></a><span class="in">        "BIOCARTA_HER2_PATHWAY",</span></span>
<span id="cb7-60"><a href="#cb7-60" aria-hidden="true" tabindex="-1"></a><span class="in">        "KEGG_ERBB_SIGNALING_PATHWAY"</span></span>
<span id="cb7-61"><a href="#cb7-61" aria-hidden="true" tabindex="-1"></a><span class="in">    ))</span></span>
<span id="cb7-62"><a href="#cb7-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-63"><a href="#cb7-63" aria-hidden="true" tabindex="-1"></a><span class="in">gene_sets_prog &lt;- dplyr::bind_rows(</span></span>
<span id="cb7-64"><a href="#cb7-64" aria-hidden="true" tabindex="-1"></a><span class="in">    gene_sets, </span></span>
<span id="cb7-65"><a href="#cb7-65" aria-hidden="true" tabindex="-1"></a><span class="in">    p4_pathways</span></span>
<span id="cb7-66"><a href="#cb7-66" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-67"><a href="#cb7-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-68"><a href="#cb7-68" aria-hidden="true" tabindex="-1"></a><span class="in">gene_sets_ &lt;- sapply(</span></span>
<span id="cb7-69"><a href="#cb7-69" aria-hidden="true" tabindex="-1"></a><span class="in">    gene_sets_prog$gs_name %&gt;% unique,</span></span>
<span id="cb7-70"><a href="#cb7-70" aria-hidden="true" tabindex="-1"></a><span class="in">    function(x, gene_sets) gene_sets %&gt;% dplyr::filter(gs_name == x) %&gt;% </span></span>
<span id="cb7-71"><a href="#cb7-71" aria-hidden="true" tabindex="-1"></a><span class="in">        dplyr::pull(gene_symbol),</span></span>
<span id="cb7-72"><a href="#cb7-72" aria-hidden="true" tabindex="-1"></a><span class="in">    gene_sets = gene_sets_prog,</span></span>
<span id="cb7-73"><a href="#cb7-73" aria-hidden="true" tabindex="-1"></a><span class="in">    USE.NAMES = TRUE,</span></span>
<span id="cb7-74"><a href="#cb7-74" aria-hidden="true" tabindex="-1"></a><span class="in">    simplify = FALSE</span></span>
<span id="cb7-75"><a href="#cb7-75" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-76"><a href="#cb7-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-77"><a href="#cb7-77" aria-hidden="true" tabindex="-1"></a><span class="in">datasets$poetic$sample_name &lt;- colnames(datasets$poetic)</span></span>
<span id="cb7-78"><a href="#cb7-78" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-79"><a href="#cb7-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-80"><a href="#cb7-80" aria-hidden="true" tabindex="-1"></a><span class="fu">## Embedding with more than 1000 genes</span></span>
<span id="cb7-81"><a href="#cb7-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-82"><a href="#cb7-82" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb7-83"><a href="#cb7-83" aria-hidden="true" tabindex="-1"></a><span class="in"># we perform now the normalization for all genes</span></span>
<span id="cb7-84"><a href="#cb7-84" aria-hidden="true" tabindex="-1"></a><span class="in">common_genes &lt;- Reduce(intersect, lapply(datasets, rownames))</span></span>
<span id="cb7-85"><a href="#cb7-85" aria-hidden="true" tabindex="-1"></a><span class="in">stable_genes &lt;- intersect(stable_genes, common_genes)</span></span>
<span id="cb7-86"><a href="#cb7-86" aria-hidden="true" tabindex="-1"></a><span class="in">datasets_normalized &lt;- mapply(</span></span>
<span id="cb7-87"><a href="#cb7-87" aria-hidden="true" tabindex="-1"></a><span class="in">    get_final_ranking_values,</span></span>
<span id="cb7-88"><a href="#cb7-88" aria-hidden="true" tabindex="-1"></a><span class="in">    sum_exp = datasets,</span></span>
<span id="cb7-89"><a href="#cb7-89" aria-hidden="true" tabindex="-1"></a><span class="in">    assay_to_use = c(which_exp, list(poetic = "normalized_intensity")),</span></span>
<span id="cb7-90"><a href="#cb7-90" aria-hidden="true" tabindex="-1"></a><span class="in">    MoreArgs = list(</span></span>
<span id="cb7-91"><a href="#cb7-91" aria-hidden="true" tabindex="-1"></a><span class="in">        stable_genes = stable_genes,</span></span>
<span id="cb7-92"><a href="#cb7-92" aria-hidden="true" tabindex="-1"></a><span class="in">        most_variable_genes = common_genes</span></span>
<span id="cb7-93"><a href="#cb7-93" aria-hidden="true" tabindex="-1"></a><span class="in">    ),</span></span>
<span id="cb7-94"><a href="#cb7-94" aria-hidden="true" tabindex="-1"></a><span class="in">    USE.NAMES = TRUE,</span></span>
<span id="cb7-95"><a href="#cb7-95" aria-hidden="true" tabindex="-1"></a><span class="in">    SIMPLIFY = FALSE</span></span>
<span id="cb7-96"><a href="#cb7-96" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-97"><a href="#cb7-97" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-98"><a href="#cb7-98" aria-hidden="true" tabindex="-1"></a><span class="in">merged_col_data &lt;- lapply(datasets_normalized, colData) %&gt;% </span></span>
<span id="cb7-99"><a href="#cb7-99" aria-hidden="true" tabindex="-1"></a><span class="in">    lapply(., data.frame) %&gt;%</span></span>
<span id="cb7-100"><a href="#cb7-100" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::bind_rows(.id = "cohort")</span></span>
<span id="cb7-101"><a href="#cb7-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-102"><a href="#cb7-102" aria-hidden="true" tabindex="-1"></a><span class="in">set.seed(1329)</span></span>
<span id="cb7-103"><a href="#cb7-103" aria-hidden="true" tabindex="-1"></a><span class="in">samples_for_training &lt;- merged_col_data %&gt;%</span></span>
<span id="cb7-104"><a href="#cb7-104" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::filter(cohort %in% which_cohorts_training) %&gt;%</span></span>
<span id="cb7-105"><a href="#cb7-105" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::pull(sample_name) %&gt;%</span></span>
<span id="cb7-106"><a href="#cb7-106" aria-hidden="true" tabindex="-1"></a><span class="in">    sample(., size = 1000) </span></span>
<span id="cb7-107"><a href="#cb7-107" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-108"><a href="#cb7-108" aria-hidden="true" tabindex="-1"></a><span class="in"># and now we can perform the molecular embedding</span></span>
<span id="cb7-109"><a href="#cb7-109" aria-hidden="true" tabindex="-1"></a><span class="in">training_set &lt;- lapply(</span></span>
<span id="cb7-110"><a href="#cb7-110" aria-hidden="true" tabindex="-1"></a><span class="in">    datasets_normalized[which_cohorts_training], </span></span>
<span id="cb7-111"><a href="#cb7-111" aria-hidden="true" tabindex="-1"></a><span class="in">    function(sum_exp, i, genes_for_pca) </span></span>
<span id="cb7-112"><a href="#cb7-112" aria-hidden="true" tabindex="-1"></a><span class="in">        assay(sum_exp[genes_for_pca, ], i = i) %&gt;% </span></span>
<span id="cb7-113"><a href="#cb7-113" aria-hidden="true" tabindex="-1"></a><span class="in">        data.frame(check.names = FALSE), </span></span>
<span id="cb7-114"><a href="#cb7-114" aria-hidden="true" tabindex="-1"></a><span class="in">    i = "avg_ranking",</span></span>
<span id="cb7-115"><a href="#cb7-115" aria-hidden="true" tabindex="-1"></a><span class="in">    genes_for_pca = common_genes</span></span>
<span id="cb7-116"><a href="#cb7-116" aria-hidden="true" tabindex="-1"></a><span class="in">) %&gt;% </span></span>
<span id="cb7-117"><a href="#cb7-117" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::bind_cols() %&gt;%</span></span>
<span id="cb7-118"><a href="#cb7-118" aria-hidden="true" tabindex="-1"></a><span class="in">    .[, samples_for_training]</span></span>
<span id="cb7-119"><a href="#cb7-119" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-120"><a href="#cb7-120" aria-hidden="true" tabindex="-1"></a><span class="in">pca_fit_all_genes &lt;- PCAtools::pca(</span></span>
<span id="cb7-121"><a href="#cb7-121" aria-hidden="true" tabindex="-1"></a><span class="in">    training_set,</span></span>
<span id="cb7-122"><a href="#cb7-122" aria-hidden="true" tabindex="-1"></a><span class="in">    metadata = dplyr::bind_rows(</span></span>
<span id="cb7-123"><a href="#cb7-123" aria-hidden="true" tabindex="-1"></a><span class="in">        lapply(</span></span>
<span id="cb7-124"><a href="#cb7-124" aria-hidden="true" tabindex="-1"></a><span class="in">            datasets_normalized[which_cohorts_training],</span></span>
<span id="cb7-125"><a href="#cb7-125" aria-hidden="true" tabindex="-1"></a><span class="in">            function(df){</span></span>
<span id="cb7-126"><a href="#cb7-126" aria-hidden="true" tabindex="-1"></a><span class="in">                colData(df) %&gt;% data.frame %&gt;%</span></span>
<span id="cb7-127"><a href="#cb7-127" aria-hidden="true" tabindex="-1"></a><span class="in">                    dplyr::filter(sample_name %in% colnames(training_set))</span></span>
<span id="cb7-128"><a href="#cb7-128" aria-hidden="true" tabindex="-1"></a><span class="in">            }</span></span>
<span id="cb7-129"><a href="#cb7-129" aria-hidden="true" tabindex="-1"></a><span class="in">        ),</span></span>
<span id="cb7-130"><a href="#cb7-130" aria-hidden="true" tabindex="-1"></a><span class="in">        .id = "cohort"</span></span>
<span id="cb7-131"><a href="#cb7-131" aria-hidden="true" tabindex="-1"></a><span class="in">    ) %&gt;% .[colnames(training_set), ],</span></span>
<span id="cb7-132"><a href="#cb7-132" aria-hidden="true" tabindex="-1"></a><span class="in">    center = FALSE,</span></span>
<span id="cb7-133"><a href="#cb7-133" aria-hidden="true" tabindex="-1"></a><span class="in">    scale = FALSE</span></span>
<span id="cb7-134"><a href="#cb7-134" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-135"><a href="#cb7-135" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-136"><a href="#cb7-136" aria-hidden="true" tabindex="-1"></a><span class="in">saveRDS(</span></span>
<span id="cb7-137"><a href="#cb7-137" aria-hidden="true" tabindex="-1"></a><span class="in">    pca_fit_all_genes,</span></span>
<span id="cb7-138"><a href="#cb7-138" aria-hidden="true" tabindex="-1"></a><span class="in">    "../../results/rds_files/expanding_ember/pca_fit_all_genes.rds"</span></span>
<span id="cb7-139"><a href="#cb7-139" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-140"><a href="#cb7-140" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-141"><a href="#cb7-141" aria-hidden="true" tabindex="-1"></a><span class="in">saveRDS(</span></span>
<span id="cb7-142"><a href="#cb7-142" aria-hidden="true" tabindex="-1"></a><span class="in">    datasets_normalized,</span></span>
<span id="cb7-143"><a href="#cb7-143" aria-hidden="true" tabindex="-1"></a><span class="in">    "../../results/rds_files/expanding_ember/datasets_normalized_all_genes.rds"</span></span>
<span id="cb7-144"><a href="#cb7-144" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-145"><a href="#cb7-145" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-146"><a href="#cb7-146" aria-hidden="true" tabindex="-1"></a><span class="in">saveRDS(</span></span>
<span id="cb7-147"><a href="#cb7-147" aria-hidden="true" tabindex="-1"></a><span class="in">    merged_col_data,</span></span>
<span id="cb7-148"><a href="#cb7-148" aria-hidden="true" tabindex="-1"></a><span class="in">    "../../results/rds_files/expanding_ember/merged_col_data_trying.rds"</span></span>
<span id="cb7-149"><a href="#cb7-149" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-150"><a href="#cb7-150" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-151"><a href="#cb7-151" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-152"><a href="#cb7-152" aria-hidden="true" tabindex="-1"></a>And now we can visualize the results.</span>
<span id="cb7-153"><a href="#cb7-153" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-154"><a href="#cb7-154" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=14, fig.height=16}</span></span>
<span id="cb7-155"><a href="#cb7-155" aria-hidden="true" tabindex="-1"></a><span class="in">#| label: fig-pca-embeddings-1</span></span>
<span id="cb7-156"><a href="#cb7-156" aria-hidden="true" tabindex="-1"></a><span class="in">#| fig-cap: PCA projections colored by different factors and organized</span></span>
<span id="cb7-157"><a href="#cb7-157" aria-hidden="true" tabindex="-1"></a><span class="in">#|    by different components. (A) Plot of the first two components colored</span></span>
<span id="cb7-158"><a href="#cb7-158" aria-hidden="true" tabindex="-1"></a><span class="in">#|    by cohort. (B) Plot of the first two components colored by ER status.</span></span>
<span id="cb7-159"><a href="#cb7-159" aria-hidden="true" tabindex="-1"></a><span class="in">#|    (C) Plot of the second and third components colored by cohort. </span></span>
<span id="cb7-160"><a href="#cb7-160" aria-hidden="true" tabindex="-1"></a><span class="in">#|    (D) Plot of the second and third components colored by ER status.</span></span>
<span id="cb7-161"><a href="#cb7-161" aria-hidden="true" tabindex="-1"></a><span class="in">#|    (E) Plot of the second and third components colored by PAM50.</span></span>
<span id="cb7-162"><a href="#cb7-162" aria-hidden="true" tabindex="-1"></a><span class="in">#|    (F) Plot of the second and third components colored by INTCLUST, only</span></span>
<span id="cb7-163"><a href="#cb7-163" aria-hidden="true" tabindex="-1"></a><span class="in">#|    METABRIC has an assigned value for this variable, NAs are TCGA samples.</span></span>
<span id="cb7-164"><a href="#cb7-164" aria-hidden="true" tabindex="-1"></a><span class="in">plots_pca_fit &lt;- list()</span></span>
<span id="cb7-165"><a href="#cb7-165" aria-hidden="true" tabindex="-1"></a><span class="in">point_size &lt;- 2</span></span>
<span id="cb7-166"><a href="#cb7-166" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-167"><a href="#cb7-167" aria-hidden="true" tabindex="-1"></a><span class="in">plots_pca_fit$pc1_pc2_cohort &lt;- PCAtools::biplot(</span></span>
<span id="cb7-168"><a href="#cb7-168" aria-hidden="true" tabindex="-1"></a><span class="in">    pca_fit_all_genes,</span></span>
<span id="cb7-169"><a href="#cb7-169" aria-hidden="true" tabindex="-1"></a><span class="in">    colby = "cohort",</span></span>
<span id="cb7-170"><a href="#cb7-170" aria-hidden="true" tabindex="-1"></a><span class="in">    lab = NULL, </span></span>
<span id="cb7-171"><a href="#cb7-171" aria-hidden="true" tabindex="-1"></a><span class="in">    legendPosition = "right",</span></span>
<span id="cb7-172"><a href="#cb7-172" aria-hidden="true" tabindex="-1"></a><span class="in">    x = "PC1",</span></span>
<span id="cb7-173"><a href="#cb7-173" aria-hidden="true" tabindex="-1"></a><span class="in">    y = "PC2",</span></span>
<span id="cb7-174"><a href="#cb7-174" aria-hidden="true" tabindex="-1"></a><span class="in">    title = "First two components colored by cohort",</span></span>
<span id="cb7-175"><a href="#cb7-175" aria-hidden="true" tabindex="-1"></a><span class="in">    subtitle = "Only 1000 training samples",</span></span>
<span id="cb7-176"><a href="#cb7-176" aria-hidden="true" tabindex="-1"></a><span class="in">    pointSize = point_size</span></span>
<span id="cb7-177"><a href="#cb7-177" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-178"><a href="#cb7-178" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-179"><a href="#cb7-179" aria-hidden="true" tabindex="-1"></a><span class="in">plots_pca_fit$pc1_pc2_er_status &lt;- PCAtools::biplot(</span></span>
<span id="cb7-180"><a href="#cb7-180" aria-hidden="true" tabindex="-1"></a><span class="in">    pca_fit_all_genes,</span></span>
<span id="cb7-181"><a href="#cb7-181" aria-hidden="true" tabindex="-1"></a><span class="in">    colby = "er_status",</span></span>
<span id="cb7-182"><a href="#cb7-182" aria-hidden="true" tabindex="-1"></a><span class="in">    lab = NULL, </span></span>
<span id="cb7-183"><a href="#cb7-183" aria-hidden="true" tabindex="-1"></a><span class="in">    legendPosition = "right",</span></span>
<span id="cb7-184"><a href="#cb7-184" aria-hidden="true" tabindex="-1"></a><span class="in">    x = "PC1",</span></span>
<span id="cb7-185"><a href="#cb7-185" aria-hidden="true" tabindex="-1"></a><span class="in">    y = "PC2",</span></span>
<span id="cb7-186"><a href="#cb7-186" aria-hidden="true" tabindex="-1"></a><span class="in">    title = "First two components colored by ER status",</span></span>
<span id="cb7-187"><a href="#cb7-187" aria-hidden="true" tabindex="-1"></a><span class="in">    subtitle = "Only 1000 training samples",</span></span>
<span id="cb7-188"><a href="#cb7-188" aria-hidden="true" tabindex="-1"></a><span class="in">    pointSize = point_size</span></span>
<span id="cb7-189"><a href="#cb7-189" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-190"><a href="#cb7-190" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-191"><a href="#cb7-191" aria-hidden="true" tabindex="-1"></a><span class="in">plots_pca_fit$pc2_pc3_cohort &lt;- PCAtools::biplot(</span></span>
<span id="cb7-192"><a href="#cb7-192" aria-hidden="true" tabindex="-1"></a><span class="in">    pca_fit_all_genes,</span></span>
<span id="cb7-193"><a href="#cb7-193" aria-hidden="true" tabindex="-1"></a><span class="in">    colby = "cohort",</span></span>
<span id="cb7-194"><a href="#cb7-194" aria-hidden="true" tabindex="-1"></a><span class="in">    lab = NULL, </span></span>
<span id="cb7-195"><a href="#cb7-195" aria-hidden="true" tabindex="-1"></a><span class="in">    legendPosition = "right",</span></span>
<span id="cb7-196"><a href="#cb7-196" aria-hidden="true" tabindex="-1"></a><span class="in">    x = "PC2",</span></span>
<span id="cb7-197"><a href="#cb7-197" aria-hidden="true" tabindex="-1"></a><span class="in">    y = "PC3",</span></span>
<span id="cb7-198"><a href="#cb7-198" aria-hidden="true" tabindex="-1"></a><span class="in">    title = "Second and third components colored by cohort",</span></span>
<span id="cb7-199"><a href="#cb7-199" aria-hidden="true" tabindex="-1"></a><span class="in">    subtitle = "Only 1000 training samples",</span></span>
<span id="cb7-200"><a href="#cb7-200" aria-hidden="true" tabindex="-1"></a><span class="in">    pointSize = point_size</span></span>
<span id="cb7-201"><a href="#cb7-201" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-202"><a href="#cb7-202" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-203"><a href="#cb7-203" aria-hidden="true" tabindex="-1"></a><span class="in">plots_pca_fit$pc2_pc3_er_status &lt;- PCAtools::biplot(</span></span>
<span id="cb7-204"><a href="#cb7-204" aria-hidden="true" tabindex="-1"></a><span class="in">    pca_fit_all_genes,</span></span>
<span id="cb7-205"><a href="#cb7-205" aria-hidden="true" tabindex="-1"></a><span class="in">    colby = "er_status",</span></span>
<span id="cb7-206"><a href="#cb7-206" aria-hidden="true" tabindex="-1"></a><span class="in">    lab = NULL, </span></span>
<span id="cb7-207"><a href="#cb7-207" aria-hidden="true" tabindex="-1"></a><span class="in">    legendPosition = "right",</span></span>
<span id="cb7-208"><a href="#cb7-208" aria-hidden="true" tabindex="-1"></a><span class="in">    x = "PC2",</span></span>
<span id="cb7-209"><a href="#cb7-209" aria-hidden="true" tabindex="-1"></a><span class="in">    y = "PC3",</span></span>
<span id="cb7-210"><a href="#cb7-210" aria-hidden="true" tabindex="-1"></a><span class="in">    title = "Second and third components colored by ER status",</span></span>
<span id="cb7-211"><a href="#cb7-211" aria-hidden="true" tabindex="-1"></a><span class="in">    subtitle = "Only 1000 training samples",</span></span>
<span id="cb7-212"><a href="#cb7-212" aria-hidden="true" tabindex="-1"></a><span class="in">    pointSize = point_size</span></span>
<span id="cb7-213"><a href="#cb7-213" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-214"><a href="#cb7-214" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-215"><a href="#cb7-215" aria-hidden="true" tabindex="-1"></a><span class="in">plots_pca_fit$pc2_pc3_pam50 &lt;- PCAtools::biplot(</span></span>
<span id="cb7-216"><a href="#cb7-216" aria-hidden="true" tabindex="-1"></a><span class="in">    pca_fit_all_genes,</span></span>
<span id="cb7-217"><a href="#cb7-217" aria-hidden="true" tabindex="-1"></a><span class="in">    colby = "pam50",</span></span>
<span id="cb7-218"><a href="#cb7-218" aria-hidden="true" tabindex="-1"></a><span class="in">    lab = NULL, </span></span>
<span id="cb7-219"><a href="#cb7-219" aria-hidden="true" tabindex="-1"></a><span class="in">    legendPosition = "right",</span></span>
<span id="cb7-220"><a href="#cb7-220" aria-hidden="true" tabindex="-1"></a><span class="in">    x = "PC2",</span></span>
<span id="cb7-221"><a href="#cb7-221" aria-hidden="true" tabindex="-1"></a><span class="in">    y = "PC3",</span></span>
<span id="cb7-222"><a href="#cb7-222" aria-hidden="true" tabindex="-1"></a><span class="in">    title = "Second and third components colored by PAM50",</span></span>
<span id="cb7-223"><a href="#cb7-223" aria-hidden="true" tabindex="-1"></a><span class="in">    subtitle = "Only 1000 training samples",</span></span>
<span id="cb7-224"><a href="#cb7-224" aria-hidden="true" tabindex="-1"></a><span class="in">    pointSize = point_size</span></span>
<span id="cb7-225"><a href="#cb7-225" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-226"><a href="#cb7-226" aria-hidden="true" tabindex="-1"></a><span class="in">plots_pca_fit$pc2_pc3_pam50 &lt;- plots_pca_fit$pc2_pc3_pam50 +</span></span>
<span id="cb7-227"><a href="#cb7-227" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::scale_color_manual(</span></span>
<span id="cb7-228"><a href="#cb7-228" aria-hidden="true" tabindex="-1"></a><span class="in">        values = get_colors_pam50(pca_fit_all_genes$metadata)</span></span>
<span id="cb7-229"><a href="#cb7-229" aria-hidden="true" tabindex="-1"></a><span class="in">    )</span></span>
<span id="cb7-230"><a href="#cb7-230" aria-hidden="true" tabindex="-1"></a><span class="in">    </span></span>
<span id="cb7-231"><a href="#cb7-231" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-232"><a href="#cb7-232" aria-hidden="true" tabindex="-1"></a><span class="in">plots_pca_fit$pc2_pc3_intclust &lt;- PCAtools::biplot(</span></span>
<span id="cb7-233"><a href="#cb7-233" aria-hidden="true" tabindex="-1"></a><span class="in">    pca_fit_all_genes,</span></span>
<span id="cb7-234"><a href="#cb7-234" aria-hidden="true" tabindex="-1"></a><span class="in">    colby = "INTCLUST",</span></span>
<span id="cb7-235"><a href="#cb7-235" aria-hidden="true" tabindex="-1"></a><span class="in">    lab = NULL, </span></span>
<span id="cb7-236"><a href="#cb7-236" aria-hidden="true" tabindex="-1"></a><span class="in">    legendPosition = "right",</span></span>
<span id="cb7-237"><a href="#cb7-237" aria-hidden="true" tabindex="-1"></a><span class="in">    x = "PC2",</span></span>
<span id="cb7-238"><a href="#cb7-238" aria-hidden="true" tabindex="-1"></a><span class="in">    y = "PC3",</span></span>
<span id="cb7-239"><a href="#cb7-239" aria-hidden="true" tabindex="-1"></a><span class="in">    title = "Second and third components colored by INTCLUST",</span></span>
<span id="cb7-240"><a href="#cb7-240" aria-hidden="true" tabindex="-1"></a><span class="in">    subtitle = "Only 1000 training samples",</span></span>
<span id="cb7-241"><a href="#cb7-241" aria-hidden="true" tabindex="-1"></a><span class="in">    pointSize = point_size</span></span>
<span id="cb7-242"><a href="#cb7-242" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-243"><a href="#cb7-243" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-244"><a href="#cb7-244" aria-hidden="true" tabindex="-1"></a><span class="in">plots_pca_fit$pc3_pc4_pam50 &lt;- PCAtools::biplot(</span></span>
<span id="cb7-245"><a href="#cb7-245" aria-hidden="true" tabindex="-1"></a><span class="in">    pca_fit_all_genes,</span></span>
<span id="cb7-246"><a href="#cb7-246" aria-hidden="true" tabindex="-1"></a><span class="in">    colby = "pam50",</span></span>
<span id="cb7-247"><a href="#cb7-247" aria-hidden="true" tabindex="-1"></a><span class="in">    lab = NULL, </span></span>
<span id="cb7-248"><a href="#cb7-248" aria-hidden="true" tabindex="-1"></a><span class="in">    legendPosition = "right",</span></span>
<span id="cb7-249"><a href="#cb7-249" aria-hidden="true" tabindex="-1"></a><span class="in">    x = "PC3",</span></span>
<span id="cb7-250"><a href="#cb7-250" aria-hidden="true" tabindex="-1"></a><span class="in">    y = "PC4",</span></span>
<span id="cb7-251"><a href="#cb7-251" aria-hidden="true" tabindex="-1"></a><span class="in">    title = "Third and fourth components colored by PAM50",</span></span>
<span id="cb7-252"><a href="#cb7-252" aria-hidden="true" tabindex="-1"></a><span class="in">    subtitle = "Only 1000 training samples",</span></span>
<span id="cb7-253"><a href="#cb7-253" aria-hidden="true" tabindex="-1"></a><span class="in">    pointSize = point_size</span></span>
<span id="cb7-254"><a href="#cb7-254" aria-hidden="true" tabindex="-1"></a><span class="in">) </span></span>
<span id="cb7-255"><a href="#cb7-255" aria-hidden="true" tabindex="-1"></a><span class="in">plots_pca_fit$pc3_pc4_pam50 &lt;- plots_pca_fit$pc3_pc4_pam50 +</span></span>
<span id="cb7-256"><a href="#cb7-256" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::scale_color_manual(</span></span>
<span id="cb7-257"><a href="#cb7-257" aria-hidden="true" tabindex="-1"></a><span class="in">        values = get_colors_pam50(pca_fit_all_genes$metadata)</span></span>
<span id="cb7-258"><a href="#cb7-258" aria-hidden="true" tabindex="-1"></a><span class="in">    )</span></span>
<span id="cb7-259"><a href="#cb7-259" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-260"><a href="#cb7-260" aria-hidden="true" tabindex="-1"></a><span class="in">plots_pca_fit$pc3_pc4_cohort &lt;- PCAtools::biplot(</span></span>
<span id="cb7-261"><a href="#cb7-261" aria-hidden="true" tabindex="-1"></a><span class="in">    pca_fit_all_genes,</span></span>
<span id="cb7-262"><a href="#cb7-262" aria-hidden="true" tabindex="-1"></a><span class="in">    colby = "cohort",</span></span>
<span id="cb7-263"><a href="#cb7-263" aria-hidden="true" tabindex="-1"></a><span class="in">    lab = NULL, </span></span>
<span id="cb7-264"><a href="#cb7-264" aria-hidden="true" tabindex="-1"></a><span class="in">    legendPosition = "right",</span></span>
<span id="cb7-265"><a href="#cb7-265" aria-hidden="true" tabindex="-1"></a><span class="in">    x = "PC3",</span></span>
<span id="cb7-266"><a href="#cb7-266" aria-hidden="true" tabindex="-1"></a><span class="in">    y = "PC4",</span></span>
<span id="cb7-267"><a href="#cb7-267" aria-hidden="true" tabindex="-1"></a><span class="in">    title = "Third and fourth components colored by cohort",</span></span>
<span id="cb7-268"><a href="#cb7-268" aria-hidden="true" tabindex="-1"></a><span class="in">    subtitle = "Only 1000 training samples",</span></span>
<span id="cb7-269"><a href="#cb7-269" aria-hidden="true" tabindex="-1"></a><span class="in">    pointSize = point_size</span></span>
<span id="cb7-270"><a href="#cb7-270" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-271"><a href="#cb7-271" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-272"><a href="#cb7-272" aria-hidden="true" tabindex="-1"></a><span class="in">cowplot::plot_grid(plotlist = plots_pca_fit, ncol = 2, labels = "AUTO")</span></span>
<span id="cb7-273"><a href="#cb7-273" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-274"><a href="#cb7-274" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-275"><a href="#cb7-275" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb7-276"><a href="#cb7-276" aria-hidden="true" tabindex="-1"></a><span class="in">datasets_pca_coordinates &lt;- lapply(</span></span>
<span id="cb7-277"><a href="#cb7-277" aria-hidden="true" tabindex="-1"></a><span class="in">    datasets_normalized,</span></span>
<span id="cb7-278"><a href="#cb7-278" aria-hidden="true" tabindex="-1"></a><span class="in">    get_pca_coordinates,</span></span>
<span id="cb7-279"><a href="#cb7-279" aria-hidden="true" tabindex="-1"></a><span class="in">    pca_fit = pca_fit_all_genes,</span></span>
<span id="cb7-280"><a href="#cb7-280" aria-hidden="true" tabindex="-1"></a><span class="in">    genes_for_pca = rownames(pca_fit_all_genes$loadings)</span></span>
<span id="cb7-281"><a href="#cb7-281" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-282"><a href="#cb7-282" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-283"><a href="#cb7-283" aria-hidden="true" tabindex="-1"></a><span class="in">df_pca_coordinates &lt;- datasets_pca_coordinates %&gt;% </span></span>
<span id="cb7-284"><a href="#cb7-284" aria-hidden="true" tabindex="-1"></a><span class="in">    do.call(rbind, .) %&gt;%</span></span>
<span id="cb7-285"><a href="#cb7-285" aria-hidden="true" tabindex="-1"></a><span class="in">    data.frame %&gt;% </span></span>
<span id="cb7-286"><a href="#cb7-286" aria-hidden="true" tabindex="-1"></a><span class="in">    tibble::rownames_to_column(var = "sample_name") %&gt;%</span></span>
<span id="cb7-287"><a href="#cb7-287" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::inner_join(</span></span>
<span id="cb7-288"><a href="#cb7-288" aria-hidden="true" tabindex="-1"></a><span class="in">        .,</span></span>
<span id="cb7-289"><a href="#cb7-289" aria-hidden="true" tabindex="-1"></a><span class="in">        merged_col_data,</span></span>
<span id="cb7-290"><a href="#cb7-290" aria-hidden="true" tabindex="-1"></a><span class="in">        by = "sample_name"</span></span>
<span id="cb7-291"><a href="#cb7-291" aria-hidden="true" tabindex="-1"></a><span class="in">    )</span></span>
<span id="cb7-292"><a href="#cb7-292" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-293"><a href="#cb7-293" aria-hidden="true" tabindex="-1"></a><span class="in">saveRDS(</span></span>
<span id="cb7-294"><a href="#cb7-294" aria-hidden="true" tabindex="-1"></a><span class="in">    df_pca_coordinates,</span></span>
<span id="cb7-295"><a href="#cb7-295" aria-hidden="true" tabindex="-1"></a><span class="in">    "../../results/rds_files/expanding_ember/df_pca_coordinates_all_genes.rds"</span></span>
<span id="cb7-296"><a href="#cb7-296" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-297"><a href="#cb7-297" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-298"><a href="#cb7-298" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-301"><a href="#cb7-301" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb7-302"><a href="#cb7-302" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_pca_coordinates</span>(</span>
<span id="cb7-303"><a href="#cb7-303" aria-hidden="true" tabindex="-1"></a>    df_pca_coordinates <span class="sc">%&gt;%</span> </span>
<span id="cb7-304"><a href="#cb7-304" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">slice</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">n</span>()),</span>
<span id="cb7-305"><a href="#cb7-305" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"cohort"</span>,</span>
<span id="cb7-306"><a href="#cb7-306" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC3"</span>,</span>
<span id="cb7-307"><a href="#cb7-307" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC4"</span>,</span>
<span id="cb7-308"><a href="#cb7-308" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">1</span>,</span>
<span id="cb7-309"><a href="#cb7-309" aria-hidden="true" tabindex="-1"></a>    <span class="at">base_size =</span> <span class="dv">10</span>,</span>
<span id="cb7-310"><a href="#cb7-310" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="fu">paste0</span>(</span>
<span id="cb7-311"><a href="#cb7-311" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Embedding of all samples from TCGA,</span><span class="sc">\n</span><span class="st">"</span>,</span>
<span id="cb7-312"><a href="#cb7-312" aria-hidden="true" tabindex="-1"></a>        <span class="st">"METABRIC and SCANB"</span></span>
<span id="cb7-313"><a href="#cb7-313" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb7-314"><a href="#cb7-314" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span> ggplot2<span class="sc">::</span><span class="fu">scale_color_viridis_d</span>() <span class="sc">+</span></span>
<span id="cb7-315"><a href="#cb7-315" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">15</span>) <span class="sc">+</span></span>
<span id="cb7-316"><a href="#cb7-316" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span></span>
<span id="cb7-317"><a href="#cb7-317" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb7-318"><a href="#cb7-318" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-319"><a href="#cb7-319" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-320"><a href="#cb7-320" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb7-321"><a href="#cb7-321" aria-hidden="true" tabindex="-1"></a><span class="in">remove_metabric &lt;- df_pca_coordinates %&gt;%</span></span>
<span id="cb7-322"><a href="#cb7-322" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::filter(PC4 &gt; 10 &amp; cohort == "metabric") %&gt;%</span></span>
<span id="cb7-323"><a href="#cb7-323" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::pull(sample_name)</span></span>
<span id="cb7-324"><a href="#cb7-324" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-325"><a href="#cb7-325" aria-hidden="true" tabindex="-1"></a><span class="in">set.seed(1329)</span></span>
<span id="cb7-326"><a href="#cb7-326" aria-hidden="true" tabindex="-1"></a><span class="in">samples_for_training &lt;- merged_col_data %&gt;%</span></span>
<span id="cb7-327"><a href="#cb7-327" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::filter(cohort %in% which_cohorts_training) %&gt;%</span></span>
<span id="cb7-328"><a href="#cb7-328" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::pull(sample_name) %&gt;%</span></span>
<span id="cb7-329"><a href="#cb7-329" aria-hidden="true" tabindex="-1"></a><span class="in">    sample(., size = 1000) %&gt;%</span></span>
<span id="cb7-330"><a href="#cb7-330" aria-hidden="true" tabindex="-1"></a><span class="in">    setdiff(., remove_metabric)</span></span>
<span id="cb7-331"><a href="#cb7-331" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-332"><a href="#cb7-332" aria-hidden="true" tabindex="-1"></a><span class="in"># and now we can perform the molecular embedding</span></span>
<span id="cb7-333"><a href="#cb7-333" aria-hidden="true" tabindex="-1"></a><span class="in">training_set &lt;- lapply(</span></span>
<span id="cb7-334"><a href="#cb7-334" aria-hidden="true" tabindex="-1"></a><span class="in">    datasets_normalized[which_cohorts_training], </span></span>
<span id="cb7-335"><a href="#cb7-335" aria-hidden="true" tabindex="-1"></a><span class="in">    function(sum_exp, i, genes_for_pca) </span></span>
<span id="cb7-336"><a href="#cb7-336" aria-hidden="true" tabindex="-1"></a><span class="in">        assay(sum_exp[genes_for_pca, ], i = i) %&gt;% </span></span>
<span id="cb7-337"><a href="#cb7-337" aria-hidden="true" tabindex="-1"></a><span class="in">        data.frame(check.names = FALSE), </span></span>
<span id="cb7-338"><a href="#cb7-338" aria-hidden="true" tabindex="-1"></a><span class="in">    i = "avg_ranking",</span></span>
<span id="cb7-339"><a href="#cb7-339" aria-hidden="true" tabindex="-1"></a><span class="in">    genes_for_pca = common_genes</span></span>
<span id="cb7-340"><a href="#cb7-340" aria-hidden="true" tabindex="-1"></a><span class="in">) %&gt;% </span></span>
<span id="cb7-341"><a href="#cb7-341" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::bind_cols() %&gt;%</span></span>
<span id="cb7-342"><a href="#cb7-342" aria-hidden="true" tabindex="-1"></a><span class="in">    .[, samples_for_training]</span></span>
<span id="cb7-343"><a href="#cb7-343" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-344"><a href="#cb7-344" aria-hidden="true" tabindex="-1"></a><span class="in">pca_fit_all_genes_wo_outliers &lt;- PCAtools::pca(</span></span>
<span id="cb7-345"><a href="#cb7-345" aria-hidden="true" tabindex="-1"></a><span class="in">    training_set,</span></span>
<span id="cb7-346"><a href="#cb7-346" aria-hidden="true" tabindex="-1"></a><span class="in">    metadata = dplyr::bind_rows(</span></span>
<span id="cb7-347"><a href="#cb7-347" aria-hidden="true" tabindex="-1"></a><span class="in">        lapply(</span></span>
<span id="cb7-348"><a href="#cb7-348" aria-hidden="true" tabindex="-1"></a><span class="in">            datasets_normalized[which_cohorts_training],</span></span>
<span id="cb7-349"><a href="#cb7-349" aria-hidden="true" tabindex="-1"></a><span class="in">            function(df){</span></span>
<span id="cb7-350"><a href="#cb7-350" aria-hidden="true" tabindex="-1"></a><span class="in">                colData(df) %&gt;% data.frame %&gt;%</span></span>
<span id="cb7-351"><a href="#cb7-351" aria-hidden="true" tabindex="-1"></a><span class="in">                    dplyr::filter(sample_name %in% colnames(training_set))</span></span>
<span id="cb7-352"><a href="#cb7-352" aria-hidden="true" tabindex="-1"></a><span class="in">            }</span></span>
<span id="cb7-353"><a href="#cb7-353" aria-hidden="true" tabindex="-1"></a><span class="in">        ),</span></span>
<span id="cb7-354"><a href="#cb7-354" aria-hidden="true" tabindex="-1"></a><span class="in">        .id = "cohort"</span></span>
<span id="cb7-355"><a href="#cb7-355" aria-hidden="true" tabindex="-1"></a><span class="in">    ) %&gt;% .[colnames(training_set), ],</span></span>
<span id="cb7-356"><a href="#cb7-356" aria-hidden="true" tabindex="-1"></a><span class="in">    center = FALSE,</span></span>
<span id="cb7-357"><a href="#cb7-357" aria-hidden="true" tabindex="-1"></a><span class="in">    scale = FALSE</span></span>
<span id="cb7-358"><a href="#cb7-358" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-359"><a href="#cb7-359" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-360"><a href="#cb7-360" aria-hidden="true" tabindex="-1"></a><span class="in">saveRDS(</span></span>
<span id="cb7-361"><a href="#cb7-361" aria-hidden="true" tabindex="-1"></a><span class="in">    pca_fit_all_genes_wo_outliers, </span></span>
<span id="cb7-362"><a href="#cb7-362" aria-hidden="true" tabindex="-1"></a><span class="in">    "../../results/rds_files/expanding_ember/pca_fit_all_genes_wo_outliers.rds"</span></span>
<span id="cb7-363"><a href="#cb7-363" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-364"><a href="#cb7-364" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-365"><a href="#cb7-365" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-366"><a href="#cb7-366" aria-hidden="true" tabindex="-1"></a>And now we can visualize the results.</span>
<span id="cb7-367"><a href="#cb7-367" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-368"><a href="#cb7-368" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=14, fig.height=16}</span></span>
<span id="cb7-369"><a href="#cb7-369" aria-hidden="true" tabindex="-1"></a><span class="in">#| label: fig-pca-embeddings-2</span></span>
<span id="cb7-370"><a href="#cb7-370" aria-hidden="true" tabindex="-1"></a><span class="in">#| fig-cap: PCA projections colored by different factors and organized</span></span>
<span id="cb7-371"><a href="#cb7-371" aria-hidden="true" tabindex="-1"></a><span class="in">#|    by different components. (A) Plot of the first two components colored</span></span>
<span id="cb7-372"><a href="#cb7-372" aria-hidden="true" tabindex="-1"></a><span class="in">#|    by cohort. (B) Plot of the first two components colored by ER status.</span></span>
<span id="cb7-373"><a href="#cb7-373" aria-hidden="true" tabindex="-1"></a><span class="in">#|    (C) Plot of the second and third components colored by cohort. </span></span>
<span id="cb7-374"><a href="#cb7-374" aria-hidden="true" tabindex="-1"></a><span class="in">#|    (D) Plot of the second and third components colored by ER status.</span></span>
<span id="cb7-375"><a href="#cb7-375" aria-hidden="true" tabindex="-1"></a><span class="in">#|    (E) Plot of the second and third components colored by PAM50.</span></span>
<span id="cb7-376"><a href="#cb7-376" aria-hidden="true" tabindex="-1"></a><span class="in">#|    (F) Plot of the second and third components colored by INTCLUST, only</span></span>
<span id="cb7-377"><a href="#cb7-377" aria-hidden="true" tabindex="-1"></a><span class="in">#|    METABRIC has an assigned value for this variable, NAs are TCGA samples.</span></span>
<span id="cb7-378"><a href="#cb7-378" aria-hidden="true" tabindex="-1"></a><span class="in">plots_pca_fit &lt;- list()</span></span>
<span id="cb7-379"><a href="#cb7-379" aria-hidden="true" tabindex="-1"></a><span class="in">point_size &lt;- 2</span></span>
<span id="cb7-380"><a href="#cb7-380" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-381"><a href="#cb7-381" aria-hidden="true" tabindex="-1"></a><span class="in">plots_pca_fit$pc1_pc2_cohort &lt;- PCAtools::biplot(</span></span>
<span id="cb7-382"><a href="#cb7-382" aria-hidden="true" tabindex="-1"></a><span class="in">    pca_fit_all_genes_wo_outliers,</span></span>
<span id="cb7-383"><a href="#cb7-383" aria-hidden="true" tabindex="-1"></a><span class="in">    colby = "cohort",</span></span>
<span id="cb7-384"><a href="#cb7-384" aria-hidden="true" tabindex="-1"></a><span class="in">    lab = NULL, </span></span>
<span id="cb7-385"><a href="#cb7-385" aria-hidden="true" tabindex="-1"></a><span class="in">    legendPosition = "right",</span></span>
<span id="cb7-386"><a href="#cb7-386" aria-hidden="true" tabindex="-1"></a><span class="in">    x = "PC1",</span></span>
<span id="cb7-387"><a href="#cb7-387" aria-hidden="true" tabindex="-1"></a><span class="in">    y = "PC2",</span></span>
<span id="cb7-388"><a href="#cb7-388" aria-hidden="true" tabindex="-1"></a><span class="in">    title = "First two components colored by cohort",</span></span>
<span id="cb7-389"><a href="#cb7-389" aria-hidden="true" tabindex="-1"></a><span class="in">    subtitle = "Only 1000 training samples",</span></span>
<span id="cb7-390"><a href="#cb7-390" aria-hidden="true" tabindex="-1"></a><span class="in">    pointSize = point_size</span></span>
<span id="cb7-391"><a href="#cb7-391" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-392"><a href="#cb7-392" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-393"><a href="#cb7-393" aria-hidden="true" tabindex="-1"></a><span class="in">plots_pca_fit$pc1_pc2_er_status &lt;- PCAtools::biplot(</span></span>
<span id="cb7-394"><a href="#cb7-394" aria-hidden="true" tabindex="-1"></a><span class="in">    pca_fit_all_genes_wo_outliers,</span></span>
<span id="cb7-395"><a href="#cb7-395" aria-hidden="true" tabindex="-1"></a><span class="in">    colby = "er_status",</span></span>
<span id="cb7-396"><a href="#cb7-396" aria-hidden="true" tabindex="-1"></a><span class="in">    lab = NULL, </span></span>
<span id="cb7-397"><a href="#cb7-397" aria-hidden="true" tabindex="-1"></a><span class="in">    legendPosition = "right",</span></span>
<span id="cb7-398"><a href="#cb7-398" aria-hidden="true" tabindex="-1"></a><span class="in">    x = "PC1",</span></span>
<span id="cb7-399"><a href="#cb7-399" aria-hidden="true" tabindex="-1"></a><span class="in">    y = "PC2",</span></span>
<span id="cb7-400"><a href="#cb7-400" aria-hidden="true" tabindex="-1"></a><span class="in">    title = "First two components colored by ER status",</span></span>
<span id="cb7-401"><a href="#cb7-401" aria-hidden="true" tabindex="-1"></a><span class="in">    subtitle = "Only 1000 training samples",</span></span>
<span id="cb7-402"><a href="#cb7-402" aria-hidden="true" tabindex="-1"></a><span class="in">    pointSize = point_size</span></span>
<span id="cb7-403"><a href="#cb7-403" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-404"><a href="#cb7-404" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-405"><a href="#cb7-405" aria-hidden="true" tabindex="-1"></a><span class="in">plots_pca_fit$pc2_pc3_cohort &lt;- PCAtools::biplot(</span></span>
<span id="cb7-406"><a href="#cb7-406" aria-hidden="true" tabindex="-1"></a><span class="in">    pca_fit_all_genes_wo_outliers,</span></span>
<span id="cb7-407"><a href="#cb7-407" aria-hidden="true" tabindex="-1"></a><span class="in">    colby = "cohort",</span></span>
<span id="cb7-408"><a href="#cb7-408" aria-hidden="true" tabindex="-1"></a><span class="in">    lab = NULL, </span></span>
<span id="cb7-409"><a href="#cb7-409" aria-hidden="true" tabindex="-1"></a><span class="in">    legendPosition = "right",</span></span>
<span id="cb7-410"><a href="#cb7-410" aria-hidden="true" tabindex="-1"></a><span class="in">    x = "PC2",</span></span>
<span id="cb7-411"><a href="#cb7-411" aria-hidden="true" tabindex="-1"></a><span class="in">    y = "PC3",</span></span>
<span id="cb7-412"><a href="#cb7-412" aria-hidden="true" tabindex="-1"></a><span class="in">    title = "Second and third components colored by cohort",</span></span>
<span id="cb7-413"><a href="#cb7-413" aria-hidden="true" tabindex="-1"></a><span class="in">    subtitle = "Only 1000 training samples",</span></span>
<span id="cb7-414"><a href="#cb7-414" aria-hidden="true" tabindex="-1"></a><span class="in">    pointSize = point_size</span></span>
<span id="cb7-415"><a href="#cb7-415" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-416"><a href="#cb7-416" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-417"><a href="#cb7-417" aria-hidden="true" tabindex="-1"></a><span class="in">plots_pca_fit$pc2_pc3_er_status &lt;- PCAtools::biplot(</span></span>
<span id="cb7-418"><a href="#cb7-418" aria-hidden="true" tabindex="-1"></a><span class="in">    pca_fit_all_genes_wo_outliers,</span></span>
<span id="cb7-419"><a href="#cb7-419" aria-hidden="true" tabindex="-1"></a><span class="in">    colby = "er_status",</span></span>
<span id="cb7-420"><a href="#cb7-420" aria-hidden="true" tabindex="-1"></a><span class="in">    lab = NULL, </span></span>
<span id="cb7-421"><a href="#cb7-421" aria-hidden="true" tabindex="-1"></a><span class="in">    legendPosition = "right",</span></span>
<span id="cb7-422"><a href="#cb7-422" aria-hidden="true" tabindex="-1"></a><span class="in">    x = "PC2",</span></span>
<span id="cb7-423"><a href="#cb7-423" aria-hidden="true" tabindex="-1"></a><span class="in">    y = "PC3",</span></span>
<span id="cb7-424"><a href="#cb7-424" aria-hidden="true" tabindex="-1"></a><span class="in">    title = "Second and third components colored by ER status",</span></span>
<span id="cb7-425"><a href="#cb7-425" aria-hidden="true" tabindex="-1"></a><span class="in">    subtitle = "Only 1000 training samples",</span></span>
<span id="cb7-426"><a href="#cb7-426" aria-hidden="true" tabindex="-1"></a><span class="in">    pointSize = point_size</span></span>
<span id="cb7-427"><a href="#cb7-427" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-428"><a href="#cb7-428" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-429"><a href="#cb7-429" aria-hidden="true" tabindex="-1"></a><span class="in">plots_pca_fit$pc2_pc3_pam50 &lt;- PCAtools::biplot(</span></span>
<span id="cb7-430"><a href="#cb7-430" aria-hidden="true" tabindex="-1"></a><span class="in">    pca_fit_all_genes_wo_outliers,</span></span>
<span id="cb7-431"><a href="#cb7-431" aria-hidden="true" tabindex="-1"></a><span class="in">    colby = "pam50",</span></span>
<span id="cb7-432"><a href="#cb7-432" aria-hidden="true" tabindex="-1"></a><span class="in">    lab = NULL, </span></span>
<span id="cb7-433"><a href="#cb7-433" aria-hidden="true" tabindex="-1"></a><span class="in">    legendPosition = "right",</span></span>
<span id="cb7-434"><a href="#cb7-434" aria-hidden="true" tabindex="-1"></a><span class="in">    x = "PC2",</span></span>
<span id="cb7-435"><a href="#cb7-435" aria-hidden="true" tabindex="-1"></a><span class="in">    y = "PC3",</span></span>
<span id="cb7-436"><a href="#cb7-436" aria-hidden="true" tabindex="-1"></a><span class="in">    title = "Second and third components colored by PAM50",</span></span>
<span id="cb7-437"><a href="#cb7-437" aria-hidden="true" tabindex="-1"></a><span class="in">    subtitle = "Only 1000 training samples",</span></span>
<span id="cb7-438"><a href="#cb7-438" aria-hidden="true" tabindex="-1"></a><span class="in">    pointSize = point_size</span></span>
<span id="cb7-439"><a href="#cb7-439" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-440"><a href="#cb7-440" aria-hidden="true" tabindex="-1"></a><span class="in">plots_pca_fit$pc2_pc3_pam50 &lt;- plots_pca_fit$pc2_pc3_pam50 +</span></span>
<span id="cb7-441"><a href="#cb7-441" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::scale_color_manual(</span></span>
<span id="cb7-442"><a href="#cb7-442" aria-hidden="true" tabindex="-1"></a><span class="in">        values = get_colors_pam50(pca_fit_all_genes_wo_outliers$metadata)</span></span>
<span id="cb7-443"><a href="#cb7-443" aria-hidden="true" tabindex="-1"></a><span class="in">    )</span></span>
<span id="cb7-444"><a href="#cb7-444" aria-hidden="true" tabindex="-1"></a><span class="in">    </span></span>
<span id="cb7-445"><a href="#cb7-445" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-446"><a href="#cb7-446" aria-hidden="true" tabindex="-1"></a><span class="in">plots_pca_fit$pc2_pc3_intclust &lt;- PCAtools::biplot(</span></span>
<span id="cb7-447"><a href="#cb7-447" aria-hidden="true" tabindex="-1"></a><span class="in">    pca_fit_all_genes_wo_outliers,</span></span>
<span id="cb7-448"><a href="#cb7-448" aria-hidden="true" tabindex="-1"></a><span class="in">    colby = "INTCLUST",</span></span>
<span id="cb7-449"><a href="#cb7-449" aria-hidden="true" tabindex="-1"></a><span class="in">    lab = NULL, </span></span>
<span id="cb7-450"><a href="#cb7-450" aria-hidden="true" tabindex="-1"></a><span class="in">    legendPosition = "right",</span></span>
<span id="cb7-451"><a href="#cb7-451" aria-hidden="true" tabindex="-1"></a><span class="in">    x = "PC2",</span></span>
<span id="cb7-452"><a href="#cb7-452" aria-hidden="true" tabindex="-1"></a><span class="in">    y = "PC3",</span></span>
<span id="cb7-453"><a href="#cb7-453" aria-hidden="true" tabindex="-1"></a><span class="in">    title = "Second and third components colored by INTCLUST",</span></span>
<span id="cb7-454"><a href="#cb7-454" aria-hidden="true" tabindex="-1"></a><span class="in">    subtitle = "Only 1000 training samples",</span></span>
<span id="cb7-455"><a href="#cb7-455" aria-hidden="true" tabindex="-1"></a><span class="in">    pointSize = point_size</span></span>
<span id="cb7-456"><a href="#cb7-456" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-457"><a href="#cb7-457" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-458"><a href="#cb7-458" aria-hidden="true" tabindex="-1"></a><span class="in">plots_pca_fit$pc3_pc4_pam50 &lt;- PCAtools::biplot(</span></span>
<span id="cb7-459"><a href="#cb7-459" aria-hidden="true" tabindex="-1"></a><span class="in">    pca_fit_all_genes_wo_outliers,</span></span>
<span id="cb7-460"><a href="#cb7-460" aria-hidden="true" tabindex="-1"></a><span class="in">    colby = "pam50",</span></span>
<span id="cb7-461"><a href="#cb7-461" aria-hidden="true" tabindex="-1"></a><span class="in">    lab = NULL, </span></span>
<span id="cb7-462"><a href="#cb7-462" aria-hidden="true" tabindex="-1"></a><span class="in">    legendPosition = "right",</span></span>
<span id="cb7-463"><a href="#cb7-463" aria-hidden="true" tabindex="-1"></a><span class="in">    x = "PC3",</span></span>
<span id="cb7-464"><a href="#cb7-464" aria-hidden="true" tabindex="-1"></a><span class="in">    y = "PC4",</span></span>
<span id="cb7-465"><a href="#cb7-465" aria-hidden="true" tabindex="-1"></a><span class="in">    title = "Third and fourth components colored by PAM50",</span></span>
<span id="cb7-466"><a href="#cb7-466" aria-hidden="true" tabindex="-1"></a><span class="in">    subtitle = "Only 1000 training samples",</span></span>
<span id="cb7-467"><a href="#cb7-467" aria-hidden="true" tabindex="-1"></a><span class="in">    pointSize = point_size</span></span>
<span id="cb7-468"><a href="#cb7-468" aria-hidden="true" tabindex="-1"></a><span class="in">) </span></span>
<span id="cb7-469"><a href="#cb7-469" aria-hidden="true" tabindex="-1"></a><span class="in">plots_pca_fit$pc3_pc4_pam50 &lt;- plots_pca_fit$pc3_pc4_pam50 +</span></span>
<span id="cb7-470"><a href="#cb7-470" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::scale_color_manual(</span></span>
<span id="cb7-471"><a href="#cb7-471" aria-hidden="true" tabindex="-1"></a><span class="in">        values = get_colors_pam50(pca_fit_all_genes_wo_outliers$metadata)</span></span>
<span id="cb7-472"><a href="#cb7-472" aria-hidden="true" tabindex="-1"></a><span class="in">    )</span></span>
<span id="cb7-473"><a href="#cb7-473" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-474"><a href="#cb7-474" aria-hidden="true" tabindex="-1"></a><span class="in">plots_pca_fit$pc3_pc4_cohort &lt;- PCAtools::biplot(</span></span>
<span id="cb7-475"><a href="#cb7-475" aria-hidden="true" tabindex="-1"></a><span class="in">    pca_fit_all_genes_wo_outliers,</span></span>
<span id="cb7-476"><a href="#cb7-476" aria-hidden="true" tabindex="-1"></a><span class="in">    colby = "cohort",</span></span>
<span id="cb7-477"><a href="#cb7-477" aria-hidden="true" tabindex="-1"></a><span class="in">    lab = NULL, </span></span>
<span id="cb7-478"><a href="#cb7-478" aria-hidden="true" tabindex="-1"></a><span class="in">    legendPosition = "right",</span></span>
<span id="cb7-479"><a href="#cb7-479" aria-hidden="true" tabindex="-1"></a><span class="in">    x = "PC3",</span></span>
<span id="cb7-480"><a href="#cb7-480" aria-hidden="true" tabindex="-1"></a><span class="in">    y = "PC4",</span></span>
<span id="cb7-481"><a href="#cb7-481" aria-hidden="true" tabindex="-1"></a><span class="in">    title = "Third and fourth components colored by cohort",</span></span>
<span id="cb7-482"><a href="#cb7-482" aria-hidden="true" tabindex="-1"></a><span class="in">    subtitle = "Only 1000 training samples",</span></span>
<span id="cb7-483"><a href="#cb7-483" aria-hidden="true" tabindex="-1"></a><span class="in">    pointSize = point_size</span></span>
<span id="cb7-484"><a href="#cb7-484" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-485"><a href="#cb7-485" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-486"><a href="#cb7-486" aria-hidden="true" tabindex="-1"></a><span class="in">cowplot::plot_grid(plotlist = plots_pca_fit, ncol = 2, labels = "AUTO")</span></span>
<span id="cb7-487"><a href="#cb7-487" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-488"><a href="#cb7-488" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-489"><a href="#cb7-489" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb7-490"><a href="#cb7-490" aria-hidden="true" tabindex="-1"></a><span class="in">datasets_pca_coordinates &lt;- lapply(</span></span>
<span id="cb7-491"><a href="#cb7-491" aria-hidden="true" tabindex="-1"></a><span class="in">    datasets_normalized,</span></span>
<span id="cb7-492"><a href="#cb7-492" aria-hidden="true" tabindex="-1"></a><span class="in">    get_pca_coordinates,</span></span>
<span id="cb7-493"><a href="#cb7-493" aria-hidden="true" tabindex="-1"></a><span class="in">    pca_fit = pca_fit_all_genes_wo_outliers,</span></span>
<span id="cb7-494"><a href="#cb7-494" aria-hidden="true" tabindex="-1"></a><span class="in">    genes_for_pca = rownames(pca_fit_all_genes_wo_outliers$loadings)</span></span>
<span id="cb7-495"><a href="#cb7-495" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-496"><a href="#cb7-496" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-497"><a href="#cb7-497" aria-hidden="true" tabindex="-1"></a><span class="in">df_pca_coordinates_wo_outliers &lt;- datasets_pca_coordinates %&gt;% </span></span>
<span id="cb7-498"><a href="#cb7-498" aria-hidden="true" tabindex="-1"></a><span class="in">    do.call(rbind, .) %&gt;%</span></span>
<span id="cb7-499"><a href="#cb7-499" aria-hidden="true" tabindex="-1"></a><span class="in">    data.frame %&gt;% </span></span>
<span id="cb7-500"><a href="#cb7-500" aria-hidden="true" tabindex="-1"></a><span class="in">    tibble::rownames_to_column(var = "sample_name") %&gt;%</span></span>
<span id="cb7-501"><a href="#cb7-501" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::inner_join(</span></span>
<span id="cb7-502"><a href="#cb7-502" aria-hidden="true" tabindex="-1"></a><span class="in">        .,</span></span>
<span id="cb7-503"><a href="#cb7-503" aria-hidden="true" tabindex="-1"></a><span class="in">        merged_col_data,</span></span>
<span id="cb7-504"><a href="#cb7-504" aria-hidden="true" tabindex="-1"></a><span class="in">        by = "sample_name"</span></span>
<span id="cb7-505"><a href="#cb7-505" aria-hidden="true" tabindex="-1"></a><span class="in">    )</span></span>
<span id="cb7-506"><a href="#cb7-506" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-507"><a href="#cb7-507" aria-hidden="true" tabindex="-1"></a><span class="in">saveRDS(</span></span>
<span id="cb7-508"><a href="#cb7-508" aria-hidden="true" tabindex="-1"></a><span class="in">    df_pca_coordinates_wo_outliers,</span></span>
<span id="cb7-509"><a href="#cb7-509" aria-hidden="true" tabindex="-1"></a><span class="in">    "../../results/rds_files/expanding_ember/df_pca_coordinates_wo_outliers.rds"</span></span>
<span id="cb7-510"><a href="#cb7-510" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-511"><a href="#cb7-511" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-512"><a href="#cb7-512" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-515"><a href="#cb7-515" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb7-516"><a href="#cb7-516" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_pca_coordinates</span>(</span>
<span id="cb7-517"><a href="#cb7-517" aria-hidden="true" tabindex="-1"></a>    df_pca_coordinates_wo_outliers <span class="sc">%&gt;%</span> </span>
<span id="cb7-518"><a href="#cb7-518" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">slice</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">n</span>()),</span>
<span id="cb7-519"><a href="#cb7-519" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"cohort"</span>,</span>
<span id="cb7-520"><a href="#cb7-520" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC3"</span>,</span>
<span id="cb7-521"><a href="#cb7-521" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC4"</span>,</span>
<span id="cb7-522"><a href="#cb7-522" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">1</span>,</span>
<span id="cb7-523"><a href="#cb7-523" aria-hidden="true" tabindex="-1"></a>    <span class="at">base_size =</span> <span class="dv">10</span>,</span>
<span id="cb7-524"><a href="#cb7-524" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="fu">paste0</span>(</span>
<span id="cb7-525"><a href="#cb7-525" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Embedding of all samples from TCGA,</span><span class="sc">\n</span><span class="st">"</span>,</span>
<span id="cb7-526"><a href="#cb7-526" aria-hidden="true" tabindex="-1"></a>        <span class="st">"METABRIC and SCANB"</span></span>
<span id="cb7-527"><a href="#cb7-527" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb7-528"><a href="#cb7-528" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span> ggplot2<span class="sc">::</span><span class="fu">scale_color_viridis_d</span>() <span class="sc">+</span></span>
<span id="cb7-529"><a href="#cb7-529" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">15</span>) <span class="sc">+</span></span>
<span id="cb7-530"><a href="#cb7-530" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span></span>
<span id="cb7-531"><a href="#cb7-531" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb7-532"><a href="#cb7-532" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-533"><a href="#cb7-533" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-536"><a href="#cb7-536" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb7-537"><a href="#cb7-537" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-pca-scanb-pc1</span></span>
<span id="cb7-538"><a href="#cb7-538" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: PCA embedding colored by cohort. The components used are the</span></span>
<span id="cb7-539"><a href="#cb7-539" aria-hidden="true" tabindex="-1"></a><span class="co">#|     first and second components.</span></span>
<span id="cb7-540"><a href="#cb7-540" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_pca_coordinates</span>(</span>
<span id="cb7-541"><a href="#cb7-541" aria-hidden="true" tabindex="-1"></a>    df_pca_coordinates_wo_outliers,</span>
<span id="cb7-542"><a href="#cb7-542" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"cohort"</span>,</span>
<span id="cb7-543"><a href="#cb7-543" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC1"</span>,</span>
<span id="cb7-544"><a href="#cb7-544" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC2"</span>,</span>
<span id="cb7-545"><a href="#cb7-545" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">1</span>,</span>
<span id="cb7-546"><a href="#cb7-546" aria-hidden="true" tabindex="-1"></a>    <span class="at">base_size =</span> <span class="dv">10</span>,</span>
<span id="cb7-547"><a href="#cb7-547" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="fu">paste0</span>(</span>
<span id="cb7-548"><a href="#cb7-548" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Embedding of all samples from TCGA,</span><span class="sc">\n</span><span class="st">"</span>,</span>
<span id="cb7-549"><a href="#cb7-549" aria-hidden="true" tabindex="-1"></a>        <span class="st">"METABRIC and SCANB"</span></span>
<span id="cb7-550"><a href="#cb7-550" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb7-551"><a href="#cb7-551" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span> </span>
<span id="cb7-552"><a href="#cb7-552" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">15</span>) <span class="sc">+</span></span>
<span id="cb7-553"><a href="#cb7-553" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span></span>
<span id="cb7-554"><a href="#cb7-554" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb7-555"><a href="#cb7-555" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-556"><a href="#cb7-556" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-557"><a href="#cb7-557" aria-hidden="true" tabindex="-1"></a>@fig-pca-scanb-er-pam50-all-genes shows that the SCANB samples are also well mixed</span>
<span id="cb7-558"><a href="#cb7-558" aria-hidden="true" tabindex="-1"></a>regarding the clinical factors, including the $SET_{ER/PR}$ signature.</span>
<span id="cb7-559"><a href="#cb7-559" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-560"><a href="#cb7-560" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=16, fig.height=12}</span></span>
<span id="cb7-561"><a href="#cb7-561" aria-hidden="true" tabindex="-1"></a><span class="in">#| label: fig-pca-scanb-er-pam50-all-genes</span></span>
<span id="cb7-562"><a href="#cb7-562" aria-hidden="true" tabindex="-1"></a><span class="in">#| fig-cap: PCA embedding of all samples from TCGA, SCANB and METABRIC.</span></span>
<span id="cb7-563"><a href="#cb7-563" aria-hidden="true" tabindex="-1"></a><span class="in">#|     (A) Colored by cohort,</span></span>
<span id="cb7-564"><a href="#cb7-564" aria-hidden="true" tabindex="-1"></a><span class="in">#|     (B) colored by ER status, (C) colored by PAM50 molecular subtype,</span></span>
<span id="cb7-565"><a href="#cb7-565" aria-hidden="true" tabindex="-1"></a><span class="in">#|     (D) colored by the $SET_{ER/PR}$ signature and only SCANB samples.</span></span>
<span id="cb7-566"><a href="#cb7-566" aria-hidden="true" tabindex="-1"></a><span class="in">size &lt;- 2</span></span>
<span id="cb7-567"><a href="#cb7-567" aria-hidden="true" tabindex="-1"></a><span class="in">base_size &lt;- 20</span></span>
<span id="cb7-568"><a href="#cb7-568" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-569"><a href="#cb7-569" aria-hidden="true" tabindex="-1"></a><span class="in">plots_with_scanb &lt;- sapply(</span></span>
<span id="cb7-570"><a href="#cb7-570" aria-hidden="true" tabindex="-1"></a><span class="in">    c("cohort", "er_status", "pam50", "SET_ERPR"), </span></span>
<span id="cb7-571"><a href="#cb7-571" aria-hidden="true" tabindex="-1"></a><span class="in">    plot_pca_coordinates,</span></span>
<span id="cb7-572"><a href="#cb7-572" aria-hidden="true" tabindex="-1"></a><span class="in">    df_pca = df_pca_coordinates_wo_outliers %&gt;% dplyr::filter(</span></span>
<span id="cb7-573"><a href="#cb7-573" aria-hidden="true" tabindex="-1"></a><span class="in">        pam50 %in% c("luma", "lumb", "basal", "her2", "normal")</span></span>
<span id="cb7-574"><a href="#cb7-574" aria-hidden="true" tabindex="-1"></a><span class="in">    ) %&gt;%</span></span>
<span id="cb7-575"><a href="#cb7-575" aria-hidden="true" tabindex="-1"></a><span class="in">        dplyr::filter(!(cohort %in% c("poetic"))) %&gt;%</span></span>
<span id="cb7-576"><a href="#cb7-576" aria-hidden="true" tabindex="-1"></a><span class="in">        dplyr::slice(sample(1:n())),</span></span>
<span id="cb7-577"><a href="#cb7-577" aria-hidden="true" tabindex="-1"></a><span class="in">    x = "PC3", </span></span>
<span id="cb7-578"><a href="#cb7-578" aria-hidden="true" tabindex="-1"></a><span class="in">    y = "PC4",</span></span>
<span id="cb7-579"><a href="#cb7-579" aria-hidden="true" tabindex="-1"></a><span class="in">    size = size,</span></span>
<span id="cb7-580"><a href="#cb7-580" aria-hidden="true" tabindex="-1"></a><span class="in">    base_size = base_size,</span></span>
<span id="cb7-581"><a href="#cb7-581" aria-hidden="true" tabindex="-1"></a><span class="in">    title = paste0(</span></span>
<span id="cb7-582"><a href="#cb7-582" aria-hidden="true" tabindex="-1"></a><span class="in">        "Embedding of all samples from TCGA,\n",</span></span>
<span id="cb7-583"><a href="#cb7-583" aria-hidden="true" tabindex="-1"></a><span class="in">        "METABRIC and SCANB"</span></span>
<span id="cb7-584"><a href="#cb7-584" aria-hidden="true" tabindex="-1"></a><span class="in">    ),</span></span>
<span id="cb7-585"><a href="#cb7-585" aria-hidden="true" tabindex="-1"></a><span class="in">    USE.NAMES = TRUE, </span></span>
<span id="cb7-586"><a href="#cb7-586" aria-hidden="true" tabindex="-1"></a><span class="in">    simplify = FALSE</span></span>
<span id="cb7-587"><a href="#cb7-587" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-588"><a href="#cb7-588" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-589"><a href="#cb7-589" aria-hidden="true" tabindex="-1"></a><span class="in">plots_with_scanb$cohort &lt;- plots_with_scanb$cohort +</span></span>
<span id="cb7-590"><a href="#cb7-590" aria-hidden="true" tabindex="-1"></a><span class="in">    change_plot_aes_point() +</span></span>
<span id="cb7-591"><a href="#cb7-591" aria-hidden="true" tabindex="-1"></a><span class="in">    change_guides_point()</span></span>
<span id="cb7-592"><a href="#cb7-592" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-593"><a href="#cb7-593" aria-hidden="true" tabindex="-1"></a><span class="in">plots_with_scanb$pam50 &lt;- plots_with_scanb$pam50 +</span></span>
<span id="cb7-594"><a href="#cb7-594" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::scale_color_manual(</span></span>
<span id="cb7-595"><a href="#cb7-595" aria-hidden="true" tabindex="-1"></a><span class="in">        values = get_colors_pam50(plots_with_scanb$pam50$data)</span></span>
<span id="cb7-596"><a href="#cb7-596" aria-hidden="true" tabindex="-1"></a><span class="in">    ) +</span></span>
<span id="cb7-597"><a href="#cb7-597" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::labs(color = "Molecular\nsubtype") + </span></span>
<span id="cb7-598"><a href="#cb7-598" aria-hidden="true" tabindex="-1"></a><span class="in">    change_plot_aes_point() +</span></span>
<span id="cb7-599"><a href="#cb7-599" aria-hidden="true" tabindex="-1"></a><span class="in">    change_guides_point()</span></span>
<span id="cb7-600"><a href="#cb7-600" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-601"><a href="#cb7-601" aria-hidden="true" tabindex="-1"></a><span class="in">plots_with_scanb$er_status &lt;-  plots_with_scanb$er_status +</span></span>
<span id="cb7-602"><a href="#cb7-602" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::scale_color_viridis_d() +</span></span>
<span id="cb7-603"><a href="#cb7-603" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::labs(color = "ER status") +</span></span>
<span id="cb7-604"><a href="#cb7-604" aria-hidden="true" tabindex="-1"></a><span class="in">    change_plot_aes_point() +</span></span>
<span id="cb7-605"><a href="#cb7-605" aria-hidden="true" tabindex="-1"></a><span class="in">    change_guides_point()</span></span>
<span id="cb7-606"><a href="#cb7-606" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-607"><a href="#cb7-607" aria-hidden="true" tabindex="-1"></a><span class="in">plots_with_scanb$SET_ERPR &lt;- df_pca_coordinates_wo_outliers %&gt;% </span></span>
<span id="cb7-608"><a href="#cb7-608" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::filter(cohort %in% c("scanb")) %&gt;%</span></span>
<span id="cb7-609"><a href="#cb7-609" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::ggplot(aes_string(x = "PC3", y = "PC4", z = "SET_ERPR")) + </span></span>
<span id="cb7-610"><a href="#cb7-610" aria-hidden="true" tabindex="-1"></a><span class="in">        ggplot2::stat_summary_hex(bins = 25) +</span></span>
<span id="cb7-611"><a href="#cb7-611" aria-hidden="true" tabindex="-1"></a><span class="in">        ggplot2::scale_fill_viridis_c() +</span></span>
<span id="cb7-612"><a href="#cb7-612" aria-hidden="true" tabindex="-1"></a><span class="in">        ggplot2::labs(</span></span>
<span id="cb7-613"><a href="#cb7-613" aria-hidden="true" tabindex="-1"></a><span class="in">            title = "Embedding of SCANB only",</span></span>
<span id="cb7-614"><a href="#cb7-614" aria-hidden="true" tabindex="-1"></a><span class="in">            fill = expression(SET[ER/PR])</span></span>
<span id="cb7-615"><a href="#cb7-615" aria-hidden="true" tabindex="-1"></a><span class="in">        ) + </span></span>
<span id="cb7-616"><a href="#cb7-616" aria-hidden="true" tabindex="-1"></a><span class="in">        ggplot2::theme_bw(base_size = base_size) +</span></span>
<span id="cb7-617"><a href="#cb7-617" aria-hidden="true" tabindex="-1"></a><span class="in">        change_plot_aes_point()</span></span>
<span id="cb7-618"><a href="#cb7-618" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-619"><a href="#cb7-619" aria-hidden="true" tabindex="-1"></a><span class="in">cowplot::plot_grid(</span></span>
<span id="cb7-620"><a href="#cb7-620" aria-hidden="true" tabindex="-1"></a><span class="in">    plotlist = plots_with_scanb,</span></span>
<span id="cb7-621"><a href="#cb7-621" aria-hidden="true" tabindex="-1"></a><span class="in">    ncol = 2</span></span>
<span id="cb7-622"><a href="#cb7-622" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-623"><a href="#cb7-623" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-624"><a href="#cb7-624" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-625"><a href="#cb7-625" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=12, fig.height=8}</span></span>
<span id="cb7-626"><a href="#cb7-626" aria-hidden="true" tabindex="-1"></a><span class="in">select_pcs &lt;- paste0("PC", 1:30)</span></span>
<span id="cb7-627"><a href="#cb7-627" aria-hidden="true" tabindex="-1"></a><span class="in">pca_fit_all_genes_wo_outliers$loadings %&gt;%</span></span>
<span id="cb7-628"><a href="#cb7-628" aria-hidden="true" tabindex="-1"></a><span class="in">    data.frame %&gt;%</span></span>
<span id="cb7-629"><a href="#cb7-629" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::select(dplyr::all_of(select_pcs)) %&gt;%</span></span>
<span id="cb7-630"><a href="#cb7-630" aria-hidden="true" tabindex="-1"></a><span class="in">    tidyr::pivot_longer(</span></span>
<span id="cb7-631"><a href="#cb7-631" aria-hidden="true" tabindex="-1"></a><span class="in">        cols = dplyr::all_of(select_pcs),</span></span>
<span id="cb7-632"><a href="#cb7-632" aria-hidden="true" tabindex="-1"></a><span class="in">        names_to = "component", </span></span>
<span id="cb7-633"><a href="#cb7-633" aria-hidden="true" tabindex="-1"></a><span class="in">        values_to = "loadings"</span></span>
<span id="cb7-634"><a href="#cb7-634" aria-hidden="true" tabindex="-1"></a><span class="in">    ) %&gt;%</span></span>
<span id="cb7-635"><a href="#cb7-635" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::ggplot(</span></span>
<span id="cb7-636"><a href="#cb7-636" aria-hidden="true" tabindex="-1"></a><span class="in">        aes(x = abs(loadings))</span></span>
<span id="cb7-637"><a href="#cb7-637" aria-hidden="true" tabindex="-1"></a><span class="in">    ) + </span></span>
<span id="cb7-638"><a href="#cb7-638" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_histogram(bins = 30) +</span></span>
<span id="cb7-639"><a href="#cb7-639" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::facet_wrap(~component) + </span></span>
<span id="cb7-640"><a href="#cb7-640" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::labs(</span></span>
<span id="cb7-641"><a href="#cb7-641" aria-hidden="true" tabindex="-1"></a><span class="in">        title = "Histogram of the loadings from the first four components"</span></span>
<span id="cb7-642"><a href="#cb7-642" aria-hidden="true" tabindex="-1"></a><span class="in">    ) +</span></span>
<span id="cb7-643"><a href="#cb7-643" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme_bw(base_size = 20) +</span></span>
<span id="cb7-644"><a href="#cb7-644" aria-hidden="true" tabindex="-1"></a><span class="in">    change_plot_aes_point()</span></span>
<span id="cb7-645"><a href="#cb7-645" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-646"><a href="#cb7-646" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-647"><a href="#cb7-647" aria-hidden="true" tabindex="-1"></a>Let us also check the other components in the biplots.</span>
<span id="cb7-648"><a href="#cb7-648" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-649"><a href="#cb7-649" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=16, fig.height=12}</span></span>
<span id="cb7-650"><a href="#cb7-650" aria-hidden="true" tabindex="-1"></a><span class="in">color_by &lt;- "pam50"</span></span>
<span id="cb7-651"><a href="#cb7-651" aria-hidden="true" tabindex="-1"></a><span class="in">GGally::ggpairs(</span></span>
<span id="cb7-652"><a href="#cb7-652" aria-hidden="true" tabindex="-1"></a><span class="in">    df_pca_coordinates_wo_outliers %&gt;%</span></span>
<span id="cb7-653"><a href="#cb7-653" aria-hidden="true" tabindex="-1"></a><span class="in">        dplyr::filter(sample_name %in%</span></span>
<span id="cb7-654"><a href="#cb7-654" aria-hidden="true" tabindex="-1"></a><span class="in">                          pca_fit_all_genes_wo_outliers$metadata$sample_name</span></span>
<span id="cb7-655"><a href="#cb7-655" aria-hidden="true" tabindex="-1"></a><span class="in">        ) %&gt;%</span></span>
<span id="cb7-656"><a href="#cb7-656" aria-hidden="true" tabindex="-1"></a><span class="in">        dplyr::select(dplyr::all_of(c(paste0("PC", 1:10), color_by))),</span></span>
<span id="cb7-657"><a href="#cb7-657" aria-hidden="true" tabindex="-1"></a><span class="in">    aes(color = !!sym(color_by), alpha = 0.5)</span></span>
<span id="cb7-658"><a href="#cb7-658" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-659"><a href="#cb7-659" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-660"><a href="#cb7-660" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-661"><a href="#cb7-661" aria-hidden="true" tabindex="-1"></a>We see that the pam50 subtypes are being separated using the components 3, 4</span>
<span id="cb7-662"><a href="#cb7-662" aria-hidden="true" tabindex="-1"></a>and also 8. In the component 8 there is some expression that is able </span>
<span id="cb7-663"><a href="#cb7-663" aria-hidden="true" tabindex="-1"></a>to differentiate between HER2-like from all other samples.</span>
<span id="cb7-664"><a href="#cb7-664" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-665"><a href="#cb7-665" aria-hidden="true" tabindex="-1"></a>We now plot by cohort.</span>
<span id="cb7-666"><a href="#cb7-666" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-667"><a href="#cb7-667" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=16, fig.height=12}</span></span>
<span id="cb7-668"><a href="#cb7-668" aria-hidden="true" tabindex="-1"></a><span class="in">color_by &lt;- "cohort"</span></span>
<span id="cb7-669"><a href="#cb7-669" aria-hidden="true" tabindex="-1"></a><span class="in">GGally::ggpairs(</span></span>
<span id="cb7-670"><a href="#cb7-670" aria-hidden="true" tabindex="-1"></a><span class="in">    df_pca_coordinates_wo_outliers %&gt;%</span></span>
<span id="cb7-671"><a href="#cb7-671" aria-hidden="true" tabindex="-1"></a><span class="in">        dplyr::filter(sample_name %in% </span></span>
<span id="cb7-672"><a href="#cb7-672" aria-hidden="true" tabindex="-1"></a><span class="in">                          pca_fit_all_genes_wo_outliers$metadata$sample_name</span></span>
<span id="cb7-673"><a href="#cb7-673" aria-hidden="true" tabindex="-1"></a><span class="in">        ) %&gt;%</span></span>
<span id="cb7-674"><a href="#cb7-674" aria-hidden="true" tabindex="-1"></a><span class="in">        dplyr::select(dplyr::all_of(c(paste0("PC", 1:10), color_by))),</span></span>
<span id="cb7-675"><a href="#cb7-675" aria-hidden="true" tabindex="-1"></a><span class="in">    aes(color = !!sym(color_by), alpha = 0.5)</span></span>
<span id="cb7-676"><a href="#cb7-676" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-677"><a href="#cb7-677" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-678"><a href="#cb7-678" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-679"><a href="#cb7-679" aria-hidden="true" tabindex="-1"></a>In PC1, 2 and 5 there are some small differences in the differences of the </span>
<span id="cb7-680"><a href="#cb7-680" aria-hidden="true" tabindex="-1"></a>distributions from the datasets. We will then remove these 3 components</span>
<span id="cb7-681"><a href="#cb7-681" aria-hidden="true" tabindex="-1"></a>up to component 10 with exception of components 3, 4 and 5.</span>
<span id="cb7-682"><a href="#cb7-682" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-683"><a href="#cb7-683" aria-hidden="true" tabindex="-1"></a><span class="fu">## Removing the batch effects</span></span>
<span id="cb7-684"><a href="#cb7-684" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-685"><a href="#cb7-685" aria-hidden="true" tabindex="-1"></a>We now try to remove the batch effects by removing all the PCs that are not</span>
<span id="cb7-686"><a href="#cb7-686" aria-hidden="true" tabindex="-1"></a>of interest, so we remove the first two components and the fifth as well. We</span>
<span id="cb7-687"><a href="#cb7-687" aria-hidden="true" tabindex="-1"></a>then calculate the scores for each cohort separately and then compare</span>
<span id="cb7-688"><a href="#cb7-688" aria-hidden="true" tabindex="-1"></a>to what was calculated before. We can then try to calculate the scores</span>
<span id="cb7-689"><a href="#cb7-689" aria-hidden="true" tabindex="-1"></a>using samples coming from all cohorts.</span>
<span id="cb7-690"><a href="#cb7-690" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-691"><a href="#cb7-691" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb7-692"><a href="#cb7-692" aria-hidden="true" tabindex="-1"></a><span class="in">data_together &lt;- lapply(</span></span>
<span id="cb7-693"><a href="#cb7-693" aria-hidden="true" tabindex="-1"></a><span class="in">    datasets_normalized,</span></span>
<span id="cb7-694"><a href="#cb7-694" aria-hidden="true" tabindex="-1"></a><span class="in">    \(x, y) assay(x, y) %&gt;% as.matrix,</span></span>
<span id="cb7-695"><a href="#cb7-695" aria-hidden="true" tabindex="-1"></a><span class="in">    "avg_ranking"</span></span>
<span id="cb7-696"><a href="#cb7-696" aria-hidden="true" tabindex="-1"></a><span class="in">) %&gt;% Reduce(cbind, .)</span></span>
<span id="cb7-697"><a href="#cb7-697" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-698"><a href="#cb7-698" aria-hidden="true" tabindex="-1"></a><span class="in">samples_for_training &lt;- pca_fit_all_genes_wo_outliers$metadata$sample_name</span></span>
<span id="cb7-699"><a href="#cb7-699" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-700"><a href="#cb7-700" aria-hidden="true" tabindex="-1"></a><span class="in">rownames(df_pca_coordinates_wo_outliers) &lt;- df_pca_coordinates_wo_outliers$sample_name</span></span>
<span id="cb7-701"><a href="#cb7-701" aria-hidden="true" tabindex="-1"></a><span class="in">name_components &lt;- colnames(pca_fit_all_genes_wo_outliers$loadings)</span></span>
<span id="cb7-702"><a href="#cb7-702" aria-hidden="true" tabindex="-1"></a><span class="in"># from previous analysis we decided that the component 5 still included</span></span>
<span id="cb7-703"><a href="#cb7-703" aria-hidden="true" tabindex="-1"></a><span class="in"># differences between the two datasets, so we decided to remove it as well </span></span>
<span id="cb7-704"><a href="#cb7-704" aria-hidden="true" tabindex="-1"></a><span class="in"># besides component 1 and 2</span></span>
<span id="cb7-705"><a href="#cb7-705" aria-hidden="true" tabindex="-1"></a><span class="in">remove_components &lt;- paste0("PC", c(1, 2, 5))</span></span>
<span id="cb7-706"><a href="#cb7-706" aria-hidden="true" tabindex="-1"></a><span class="in">keep_components &lt;- setdiff(name_components, remove_components)</span></span>
<span id="cb7-707"><a href="#cb7-707" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-708"><a href="#cb7-708" aria-hidden="true" tabindex="-1"></a><span class="in"># the code below calculates the regressed dataframe for all samples from</span></span>
<span id="cb7-709"><a href="#cb7-709" aria-hidden="true" tabindex="-1"></a><span class="in"># scanb, tcga and metabric</span></span>
<span id="cb7-710"><a href="#cb7-710" aria-hidden="true" tabindex="-1"></a><span class="in">df_pcs_regressed &lt;- as.matrix(</span></span>
<span id="cb7-711"><a href="#cb7-711" aria-hidden="true" tabindex="-1"></a><span class="in">    df_pca_coordinates_wo_outliers[colnames(data_together), keep_components]</span></span>
<span id="cb7-712"><a href="#cb7-712" aria-hidden="true" tabindex="-1"></a><span class="in">) %*% t(pca_fit_all_genes_wo_outliers$loadings[, keep_components]) %&gt;%</span></span>
<span id="cb7-713"><a href="#cb7-713" aria-hidden="true" tabindex="-1"></a><span class="in">    t</span></span>
<span id="cb7-714"><a href="#cb7-714" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-715"><a href="#cb7-715" aria-hidden="true" tabindex="-1"></a><span class="in">pca_removed_pcs &lt;- PCAtools::pca(</span></span>
<span id="cb7-716"><a href="#cb7-716" aria-hidden="true" tabindex="-1"></a><span class="in">    mat = df_pcs_regressed[, samples_for_training],</span></span>
<span id="cb7-717"><a href="#cb7-717" aria-hidden="true" tabindex="-1"></a><span class="in">    metadata = pca_fit_all_genes_wo_outliers$metadata</span></span>
<span id="cb7-718"><a href="#cb7-718" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-719"><a href="#cb7-719" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-720"><a href="#cb7-720" aria-hidden="true" tabindex="-1"></a><span class="in">pca_all_samples_wo_pcs &lt;- t(df_pcs_regressed %&gt;% as.matrix) %*% </span></span>
<span id="cb7-721"><a href="#cb7-721" aria-hidden="true" tabindex="-1"></a><span class="in">    (pca_removed_pcs$loadings %&gt;% as.matrix)</span></span>
<span id="cb7-722"><a href="#cb7-722" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-723"><a href="#cb7-723" aria-hidden="true" tabindex="-1"></a><span class="in">df_pca_wo_pcs &lt;- pca_all_samples_wo_pcs %&gt;%</span></span>
<span id="cb7-724"><a href="#cb7-724" aria-hidden="true" tabindex="-1"></a><span class="in">    data.frame %&gt;%</span></span>
<span id="cb7-725"><a href="#cb7-725" aria-hidden="true" tabindex="-1"></a><span class="in">    tibble::rownames_to_column(var = "sample_name") %&gt;%</span></span>
<span id="cb7-726"><a href="#cb7-726" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::inner_join(</span></span>
<span id="cb7-727"><a href="#cb7-727" aria-hidden="true" tabindex="-1"></a><span class="in">        df_pca[, !grepl("PC", colnames(df_pca))],</span></span>
<span id="cb7-728"><a href="#cb7-728" aria-hidden="true" tabindex="-1"></a><span class="in">        by = "sample_name"</span></span>
<span id="cb7-729"><a href="#cb7-729" aria-hidden="true" tabindex="-1"></a><span class="in">    )</span></span>
<span id="cb7-730"><a href="#cb7-730" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-731"><a href="#cb7-731" aria-hidden="true" tabindex="-1"></a><span class="in">saveRDS(</span></span>
<span id="cb7-732"><a href="#cb7-732" aria-hidden="true" tabindex="-1"></a><span class="in">    df_pca_wo_pcs, </span></span>
<span id="cb7-733"><a href="#cb7-733" aria-hidden="true" tabindex="-1"></a><span class="in">    "../../results/rds_files/expanding_ember/df_pca_wo_pcs.rds"</span></span>
<span id="cb7-734"><a href="#cb7-734" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-735"><a href="#cb7-735" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-736"><a href="#cb7-736" aria-hidden="true" tabindex="-1"></a><span class="in">saveRDS(</span></span>
<span id="cb7-737"><a href="#cb7-737" aria-hidden="true" tabindex="-1"></a><span class="in">    pca_removed_pcs,</span></span>
<span id="cb7-738"><a href="#cb7-738" aria-hidden="true" tabindex="-1"></a><span class="in">    "../../results/rds_files/expanding_ember/pca_removed_pcs.rds"</span></span>
<span id="cb7-739"><a href="#cb7-739" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-740"><a href="#cb7-740" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-741"><a href="#cb7-741" aria-hidden="true" tabindex="-1"></a><span class="in">saveRDS(</span></span>
<span id="cb7-742"><a href="#cb7-742" aria-hidden="true" tabindex="-1"></a><span class="in">    df_pcs_regressed, </span></span>
<span id="cb7-743"><a href="#cb7-743" aria-hidden="true" tabindex="-1"></a><span class="in">    "../../results/rds_files/expanding_ember/df_pcs_regressed.rds"</span></span>
<span id="cb7-744"><a href="#cb7-744" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-745"><a href="#cb7-745" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-746"><a href="#cb7-746" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-749"><a href="#cb7-749" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb7-750"><a href="#cb7-750" aria-hidden="true" tabindex="-1"></a>PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb7-751"><a href="#cb7-751" aria-hidden="true" tabindex="-1"></a>    pca_removed_pcs,</span>
<span id="cb7-752"><a href="#cb7-752" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> <span class="cn">NULL</span>,</span>
<span id="cb7-753"><a href="#cb7-753" aria-hidden="true" tabindex="-1"></a>    <span class="at">colby =</span> <span class="st">"cohort"</span>,</span>
<span id="cb7-754"><a href="#cb7-754" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">"right"</span></span>
<span id="cb7-755"><a href="#cb7-755" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb7-756"><a href="#cb7-756" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-757"><a href="#cb7-757" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-760"><a href="#cb7-760" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb7-761"><a href="#cb7-761" aria-hidden="true" tabindex="-1"></a>PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb7-762"><a href="#cb7-762" aria-hidden="true" tabindex="-1"></a>    pca_removed_pcs,</span>
<span id="cb7-763"><a href="#cb7-763" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> <span class="cn">NULL</span>,</span>
<span id="cb7-764"><a href="#cb7-764" aria-hidden="true" tabindex="-1"></a>    <span class="at">colby =</span> <span class="st">"pam50"</span>,</span>
<span id="cb7-765"><a href="#cb7-765" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">"right"</span></span>
<span id="cb7-766"><a href="#cb7-766" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb7-767"><a href="#cb7-767" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-768"><a href="#cb7-768" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-769"><a href="#cb7-769" aria-hidden="true" tabindex="-1"></a>In general it looks good, the luminal B population is a bit compressed </span>
<span id="cb7-770"><a href="#cb7-770" aria-hidden="true" tabindex="-1"></a>towards the luminal A population though.</span>
<span id="cb7-771"><a href="#cb7-771" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-772"><a href="#cb7-772" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width = 6, fig.height = 4}</span></span>
<span id="cb7-773"><a href="#cb7-773" aria-hidden="true" tabindex="-1"></a><span class="in">size_dots &lt;- 1</span></span>
<span id="cb7-774"><a href="#cb7-774" aria-hidden="true" tabindex="-1"></a><span class="in">alpha_val &lt;- 0.8</span></span>
<span id="cb7-775"><a href="#cb7-775" aria-hidden="true" tabindex="-1"></a><span class="in">base_size &lt;- 10</span></span>
<span id="cb7-776"><a href="#cb7-776" aria-hidden="true" tabindex="-1"></a><span class="in">size_legend &lt;- 5</span></span>
<span id="cb7-777"><a href="#cb7-777" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-778"><a href="#cb7-778" aria-hidden="true" tabindex="-1"></a><span class="in">ggplot2::ggplot(df_pca_wo_pcs, aes(x = PC1, y = PC2, color = cohort)) + </span></span>
<span id="cb7-779"><a href="#cb7-779" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_point(size = size_dots, alpha = alpha_val) +</span></span>
<span id="cb7-780"><a href="#cb7-780" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::scale_alpha(guide = 'none') +</span></span>
<span id="cb7-781"><a href="#cb7-781" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::labs(</span></span>
<span id="cb7-782"><a href="#cb7-782" aria-hidden="true" tabindex="-1"></a><span class="in">        color = "PAM50"</span></span>
<span id="cb7-783"><a href="#cb7-783" aria-hidden="true" tabindex="-1"></a><span class="in">    ) + </span></span>
<span id="cb7-784"><a href="#cb7-784" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme_bw(base_size = base_size)</span></span>
<span id="cb7-785"><a href="#cb7-785" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-786"><a href="#cb7-786" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-787"><a href="#cb7-787" aria-hidden="true" tabindex="-1"></a>There is a very good merging of the datasets. We see mostly</span>
<span id="cb7-788"><a href="#cb7-788" aria-hidden="true" tabindex="-1"></a>green points because they are the majority. </span>
<span id="cb7-789"><a href="#cb7-789" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-790"><a href="#cb7-790" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=8}</span></span>
<span id="cb7-791"><a href="#cb7-791" aria-hidden="true" tabindex="-1"></a><span class="in">pcs_loadings_plot &lt;- paste0("PC", 1:10)</span></span>
<span id="cb7-792"><a href="#cb7-792" aria-hidden="true" tabindex="-1"></a><span class="in">pca_removed_pcs$loadings %&gt;%</span></span>
<span id="cb7-793"><a href="#cb7-793" aria-hidden="true" tabindex="-1"></a><span class="in">    data.frame %&gt;%</span></span>
<span id="cb7-794"><a href="#cb7-794" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::select(dplyr::all_of(pcs_loadings_plot)) %&gt;%</span></span>
<span id="cb7-795"><a href="#cb7-795" aria-hidden="true" tabindex="-1"></a><span class="in">    tidyr::pivot_longer(</span></span>
<span id="cb7-796"><a href="#cb7-796" aria-hidden="true" tabindex="-1"></a><span class="in">        cols = dplyr::all_of(pcs_loadings_plot),</span></span>
<span id="cb7-797"><a href="#cb7-797" aria-hidden="true" tabindex="-1"></a><span class="in">        names_to = "component", </span></span>
<span id="cb7-798"><a href="#cb7-798" aria-hidden="true" tabindex="-1"></a><span class="in">        values_to = "loadings"</span></span>
<span id="cb7-799"><a href="#cb7-799" aria-hidden="true" tabindex="-1"></a><span class="in">    ) %&gt;%</span></span>
<span id="cb7-800"><a href="#cb7-800" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::mutate(component = factor(component, levels = pcs_loadings_plot)) %&gt;%</span></span>
<span id="cb7-801"><a href="#cb7-801" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::ggplot(</span></span>
<span id="cb7-802"><a href="#cb7-802" aria-hidden="true" tabindex="-1"></a><span class="in">        aes(x = abs(loadings))</span></span>
<span id="cb7-803"><a href="#cb7-803" aria-hidden="true" tabindex="-1"></a><span class="in">    ) + </span></span>
<span id="cb7-804"><a href="#cb7-804" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_histogram(bins = 30) +</span></span>
<span id="cb7-805"><a href="#cb7-805" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::facet_wrap(~component) + </span></span>
<span id="cb7-806"><a href="#cb7-806" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::labs(</span></span>
<span id="cb7-807"><a href="#cb7-807" aria-hidden="true" tabindex="-1"></a><span class="in">        title = "Histogram of the loadings from the first four components"</span></span>
<span id="cb7-808"><a href="#cb7-808" aria-hidden="true" tabindex="-1"></a><span class="in">    ) +</span></span>
<span id="cb7-809"><a href="#cb7-809" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme_bw(base_size = 20)</span></span>
<span id="cb7-810"><a href="#cb7-810" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-811"><a href="#cb7-811" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-812"><a href="#cb7-812" aria-hidden="true" tabindex="-1"></a>Now we see that the first loadings has a tail in a similar fashio as the</span>
<span id="cb7-813"><a href="#cb7-813" aria-hidden="true" tabindex="-1"></a>other loadings. </span>
<span id="cb7-814"><a href="#cb7-814" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-815"><a href="#cb7-815" aria-hidden="true" tabindex="-1"></a>We now try to calculate the scores of the samples using these new embeddings</span>
<span id="cb7-816"><a href="#cb7-816" aria-hidden="true" tabindex="-1"></a>only from TCGA samples as we will compare the results later on. </span>
<span id="cb7-817"><a href="#cb7-817" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-818"><a href="#cb7-818" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb7-819"><a href="#cb7-819" aria-hidden="true" tabindex="-1"></a><span class="in">gsva_scores_embeddings_tcga &lt;- GSVA::gsva(</span></span>
<span id="cb7-820"><a href="#cb7-820" aria-hidden="true" tabindex="-1"></a><span class="in">    df_pcs_regressed[, merged_col_data %&gt;%</span></span>
<span id="cb7-821"><a href="#cb7-821" aria-hidden="true" tabindex="-1"></a><span class="in">                         dplyr::filter(cohort == "tcga") %&gt;%</span></span>
<span id="cb7-822"><a href="#cb7-822" aria-hidden="true" tabindex="-1"></a><span class="in">                         dplyr::pull(sample_name)],</span></span>
<span id="cb7-823"><a href="#cb7-823" aria-hidden="true" tabindex="-1"></a><span class="in">    gene_sets_, </span></span>
<span id="cb7-824"><a href="#cb7-824" aria-hidden="true" tabindex="-1"></a><span class="in">    parallel.sz = 10,</span></span>
<span id="cb7-825"><a href="#cb7-825" aria-hidden="true" tabindex="-1"></a><span class="in">    verbose = FALSE</span></span>
<span id="cb7-826"><a href="#cb7-826" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-827"><a href="#cb7-827" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-828"><a href="#cb7-828" aria-hidden="true" tabindex="-1"></a><span class="in">saveRDS(</span></span>
<span id="cb7-829"><a href="#cb7-829" aria-hidden="true" tabindex="-1"></a><span class="in">    gsva_scores_embeddings_tcga,</span></span>
<span id="cb7-830"><a href="#cb7-830" aria-hidden="true" tabindex="-1"></a><span class="in">    "../../results/rds_files/expanding_ember/gsva_scores_embeddings_tcga.rds"</span></span>
<span id="cb7-831"><a href="#cb7-831" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-832"><a href="#cb7-832" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-833"><a href="#cb7-833" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-834"><a href="#cb7-834" aria-hidden="true" tabindex="-1"></a>And we can now check and compare the scores.</span>
<span id="cb7-835"><a href="#cb7-835" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-836"><a href="#cb7-836" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=5}</span></span>
<span id="cb7-837"><a href="#cb7-837" aria-hidden="true" tabindex="-1"></a><span class="in">pathways_of_interest &lt;- c(</span></span>
<span id="cb7-838"><a href="#cb7-838" aria-hidden="true" tabindex="-1"></a><span class="in">    "HALLMARK_ESTROGEN_RESPONSE_EARLY",</span></span>
<span id="cb7-839"><a href="#cb7-839" aria-hidden="true" tabindex="-1"></a><span class="in">    "HALLMARK_G2M_CHECKPOINT", </span></span>
<span id="cb7-840"><a href="#cb7-840" aria-hidden="true" tabindex="-1"></a><span class="in">    "SET_ERPR"</span></span>
<span id="cb7-841"><a href="#cb7-841" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-842"><a href="#cb7-842" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-843"><a href="#cb7-843" aria-hidden="true" tabindex="-1"></a><span class="in">sapply(</span></span>
<span id="cb7-844"><a href="#cb7-844" aria-hidden="true" tabindex="-1"></a><span class="in">    pathways_of_interest, </span></span>
<span id="cb7-845"><a href="#cb7-845" aria-hidden="true" tabindex="-1"></a><span class="in">    get_merged_df_scores,</span></span>
<span id="cb7-846"><a href="#cb7-846" aria-hidden="true" tabindex="-1"></a><span class="in">    df1 = gsva_scores_embeddings_tcga,</span></span>
<span id="cb7-847"><a href="#cb7-847" aria-hidden="true" tabindex="-1"></a><span class="in">    df2 = df_pca,</span></span>
<span id="cb7-848"><a href="#cb7-848" aria-hidden="true" tabindex="-1"></a><span class="in">    USE.NAMES = TRUE, </span></span>
<span id="cb7-849"><a href="#cb7-849" aria-hidden="true" tabindex="-1"></a><span class="in">    simplify = FALSE</span></span>
<span id="cb7-850"><a href="#cb7-850" aria-hidden="true" tabindex="-1"></a><span class="in">) %&gt;% </span></span>
<span id="cb7-851"><a href="#cb7-851" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::bind_rows(.id = "pathway") %&gt;%</span></span>
<span id="cb7-852"><a href="#cb7-852" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::ggplot(aes(x = df1, y = df2)) +</span></span>
<span id="cb7-853"><a href="#cb7-853" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_point() + </span></span>
<span id="cb7-854"><a href="#cb7-854" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::facet_wrap(~pathway) +</span></span>
<span id="cb7-855"><a href="#cb7-855" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_abline(slope = 1, intercept = 0, color = "red") +</span></span>
<span id="cb7-856"><a href="#cb7-856" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::labs(</span></span>
<span id="cb7-857"><a href="#cb7-857" aria-hidden="true" tabindex="-1"></a><span class="in">        x = "GSVA scores from regressed data",</span></span>
<span id="cb7-858"><a href="#cb7-858" aria-hidden="true" tabindex="-1"></a><span class="in">        y = "GSVA scores from unregressed data",</span></span>
<span id="cb7-859"><a href="#cb7-859" aria-hidden="true" tabindex="-1"></a><span class="in">        title = "TCGA scores"</span></span>
<span id="cb7-860"><a href="#cb7-860" aria-hidden="true" tabindex="-1"></a><span class="in">    ) +</span></span>
<span id="cb7-861"><a href="#cb7-861" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme_bw()</span></span>
<span id="cb7-862"><a href="#cb7-862" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-863"><a href="#cb7-863" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-864"><a href="#cb7-864" aria-hidden="true" tabindex="-1"></a>We see that for the three pathways the correlation is positive and</span>
<span id="cb7-865"><a href="#cb7-865" aria-hidden="true" tabindex="-1"></a>strong. The difference is that it is a bit noisy, so the correlation</span>
<span id="cb7-866"><a href="#cb7-866" aria-hidden="true" tabindex="-1"></a>is not perfect. What is interesting to see is that at the extremes</span>
<span id="cb7-867"><a href="#cb7-867" aria-hidden="true" tabindex="-1"></a>the scores have less variation usually. </span>
<span id="cb7-868"><a href="#cb7-868" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-869"><a href="#cb7-869" aria-hidden="true" tabindex="-1"></a>We now calculate the scores from SCANB and compare the results. Remember</span>
<span id="cb7-870"><a href="#cb7-870" aria-hidden="true" tabindex="-1"></a>that SCANB was not used for the PCA training.</span>
<span id="cb7-871"><a href="#cb7-871" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-872"><a href="#cb7-872" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb7-873"><a href="#cb7-873" aria-hidden="true" tabindex="-1"></a><span class="in">gsva_scores_embeddings_scanb &lt;- GSVA::gsva(</span></span>
<span id="cb7-874"><a href="#cb7-874" aria-hidden="true" tabindex="-1"></a><span class="in">    df_pcs_regressed[, merged_col_data %&gt;%</span></span>
<span id="cb7-875"><a href="#cb7-875" aria-hidden="true" tabindex="-1"></a><span class="in">                         dplyr::filter(cohort == "scanb") %&gt;%</span></span>
<span id="cb7-876"><a href="#cb7-876" aria-hidden="true" tabindex="-1"></a><span class="in">                         dplyr::pull(sample_name)],</span></span>
<span id="cb7-877"><a href="#cb7-877" aria-hidden="true" tabindex="-1"></a><span class="in">    gene_sets_, </span></span>
<span id="cb7-878"><a href="#cb7-878" aria-hidden="true" tabindex="-1"></a><span class="in">    parallel.sz = 10,</span></span>
<span id="cb7-879"><a href="#cb7-879" aria-hidden="true" tabindex="-1"></a><span class="in">    verbose = FALSE</span></span>
<span id="cb7-880"><a href="#cb7-880" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-881"><a href="#cb7-881" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-882"><a href="#cb7-882" aria-hidden="true" tabindex="-1"></a><span class="in">saveRDS(</span></span>
<span id="cb7-883"><a href="#cb7-883" aria-hidden="true" tabindex="-1"></a><span class="in">    gsva_scores_embeddings_scanb,</span></span>
<span id="cb7-884"><a href="#cb7-884" aria-hidden="true" tabindex="-1"></a><span class="in">    "../../results/rds_files/expanding_ember/gsva_scores_embeddings_scanb.rds"</span></span>
<span id="cb7-885"><a href="#cb7-885" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-886"><a href="#cb7-886" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-887"><a href="#cb7-887" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-888"><a href="#cb7-888" aria-hidden="true" tabindex="-1"></a>And we can now check and compare the scores.</span>
<span id="cb7-889"><a href="#cb7-889" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-890"><a href="#cb7-890" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=5}</span></span>
<span id="cb7-891"><a href="#cb7-891" aria-hidden="true" tabindex="-1"></a><span class="in">sapply(</span></span>
<span id="cb7-892"><a href="#cb7-892" aria-hidden="true" tabindex="-1"></a><span class="in">    pathways_of_interest, </span></span>
<span id="cb7-893"><a href="#cb7-893" aria-hidden="true" tabindex="-1"></a><span class="in">    get_merged_df_scores,</span></span>
<span id="cb7-894"><a href="#cb7-894" aria-hidden="true" tabindex="-1"></a><span class="in">    df1 = gsva_scores_embeddings_scanb,</span></span>
<span id="cb7-895"><a href="#cb7-895" aria-hidden="true" tabindex="-1"></a><span class="in">    df2 = df_pca,</span></span>
<span id="cb7-896"><a href="#cb7-896" aria-hidden="true" tabindex="-1"></a><span class="in">    USE.NAMES = TRUE, </span></span>
<span id="cb7-897"><a href="#cb7-897" aria-hidden="true" tabindex="-1"></a><span class="in">    simplify = FALSE</span></span>
<span id="cb7-898"><a href="#cb7-898" aria-hidden="true" tabindex="-1"></a><span class="in">) %&gt;% </span></span>
<span id="cb7-899"><a href="#cb7-899" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::bind_rows(.id = "pathway") %&gt;%</span></span>
<span id="cb7-900"><a href="#cb7-900" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::ggplot(aes(x = df1, y = df2)) +</span></span>
<span id="cb7-901"><a href="#cb7-901" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_point() + </span></span>
<span id="cb7-902"><a href="#cb7-902" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::facet_wrap(~pathway) +</span></span>
<span id="cb7-903"><a href="#cb7-903" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_abline(slope = 1, intercept = 0, color = "red") +</span></span>
<span id="cb7-904"><a href="#cb7-904" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::labs(</span></span>
<span id="cb7-905"><a href="#cb7-905" aria-hidden="true" tabindex="-1"></a><span class="in">        x = "GSVA scores from regressed data",</span></span>
<span id="cb7-906"><a href="#cb7-906" aria-hidden="true" tabindex="-1"></a><span class="in">        y = "GSVA scores from unregressed data",</span></span>
<span id="cb7-907"><a href="#cb7-907" aria-hidden="true" tabindex="-1"></a><span class="in">        title = "TCGA scores"</span></span>
<span id="cb7-908"><a href="#cb7-908" aria-hidden="true" tabindex="-1"></a><span class="in">    ) +</span></span>
<span id="cb7-909"><a href="#cb7-909" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme_bw()</span></span>
<span id="cb7-910"><a href="#cb7-910" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-911"><a href="#cb7-911" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-912"><a href="#cb7-912" aria-hidden="true" tabindex="-1"></a>The same phenomenon happened. Samples in the endges have lower variability,</span>
<span id="cb7-913"><a href="#cb7-913" aria-hidden="true" tabindex="-1"></a>but samples in the middle are the ones that have high variability.</span>
<span id="cb7-914"><a href="#cb7-914" aria-hidden="true" tabindex="-1"></a>For example there are several samples from the regressed data</span>
<span id="cb7-915"><a href="#cb7-915" aria-hidden="true" tabindex="-1"></a>with SET_ERPR score close to 0.25 and negative scores.</span>
<span id="cb7-916"><a href="#cb7-916" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-917"><a href="#cb7-917" aria-hidden="true" tabindex="-1"></a>What happens now if we use another dataset and then use only one sample to </span>
<span id="cb7-918"><a href="#cb7-918" aria-hidden="true" tabindex="-1"></a>calculate the score. Let us include SCANB samples in the pipeline. For this </span>
<span id="cb7-919"><a href="#cb7-919" aria-hidden="true" tabindex="-1"></a>we select 50 scanb samples and 100 random samples from tcga and metabric.</span>
<span id="cb7-920"><a href="#cb7-920" aria-hidden="true" tabindex="-1"></a>We then combine all these samples together and calculate the GSVA scores.</span>
<span id="cb7-921"><a href="#cb7-921" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-922"><a href="#cb7-922" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb7-923"><a href="#cb7-923" aria-hidden="true" tabindex="-1"></a><span class="in">scanb_samples &lt;- df_pca %&gt;%</span></span>
<span id="cb7-924"><a href="#cb7-924" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::filter(cohort == "scanb") %&gt;%</span></span>
<span id="cb7-925"><a href="#cb7-925" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::slice_sample(n = 50) %&gt;%</span></span>
<span id="cb7-926"><a href="#cb7-926" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::pull(sample_name)</span></span>
<span id="cb7-927"><a href="#cb7-927" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-928"><a href="#cb7-928" aria-hidden="true" tabindex="-1"></a><span class="in">gsva_scores_embeddings_scanb_within_tcga &lt;- parallel::mclapply(</span></span>
<span id="cb7-929"><a href="#cb7-929" aria-hidden="true" tabindex="-1"></a><span class="in">    scanb_samples,</span></span>
<span id="cb7-930"><a href="#cb7-930" aria-hidden="true" tabindex="-1"></a><span class="in">    function(x){</span></span>
<span id="cb7-931"><a href="#cb7-931" aria-hidden="true" tabindex="-1"></a><span class="in">        GSVA::gsva(</span></span>
<span id="cb7-932"><a href="#cb7-932" aria-hidden="true" tabindex="-1"></a><span class="in">            df_pcs_regressed[</span></span>
<span id="cb7-933"><a href="#cb7-933" aria-hidden="true" tabindex="-1"></a><span class="in">                , </span></span>
<span id="cb7-934"><a href="#cb7-934" aria-hidden="true" tabindex="-1"></a><span class="in">                merged_col_data %&gt;%</span></span>
<span id="cb7-935"><a href="#cb7-935" aria-hidden="true" tabindex="-1"></a><span class="in">                    dplyr::filter(cohort %in% c("tcga", "metabric")) %&gt;%</span></span>
<span id="cb7-936"><a href="#cb7-936" aria-hidden="true" tabindex="-1"></a><span class="in">                    dplyr::slice_sample(n = 100) %&gt;%</span></span>
<span id="cb7-937"><a href="#cb7-937" aria-hidden="true" tabindex="-1"></a><span class="in">                    dplyr::pull(sample_name) %&gt;%</span></span>
<span id="cb7-938"><a href="#cb7-938" aria-hidden="true" tabindex="-1"></a><span class="in">                    c(., x)</span></span>
<span id="cb7-939"><a href="#cb7-939" aria-hidden="true" tabindex="-1"></a><span class="in">            ],</span></span>
<span id="cb7-940"><a href="#cb7-940" aria-hidden="true" tabindex="-1"></a><span class="in">            gene_sets_,</span></span>
<span id="cb7-941"><a href="#cb7-941" aria-hidden="true" tabindex="-1"></a><span class="in">            verbose = FALSE</span></span>
<span id="cb7-942"><a href="#cb7-942" aria-hidden="true" tabindex="-1"></a><span class="in">        )   </span></span>
<span id="cb7-943"><a href="#cb7-943" aria-hidden="true" tabindex="-1"></a><span class="in">    },</span></span>
<span id="cb7-944"><a href="#cb7-944" aria-hidden="true" tabindex="-1"></a><span class="in">    mc.cores = 10</span></span>
<span id="cb7-945"><a href="#cb7-945" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-946"><a href="#cb7-946" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-947"><a href="#cb7-947" aria-hidden="true" tabindex="-1"></a><span class="in">saveRDS(</span></span>
<span id="cb7-948"><a href="#cb7-948" aria-hidden="true" tabindex="-1"></a><span class="in">    gsva_scores_embeddings_scanb_within_tcga,</span></span>
<span id="cb7-949"><a href="#cb7-949" aria-hidden="true" tabindex="-1"></a><span class="in">    "../../results/rds_files/expanding_ember/gsva_scores_embeddings_scanb_within_tcga.rds"    </span></span>
<span id="cb7-950"><a href="#cb7-950" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-951"><a href="#cb7-951" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-952"><a href="#cb7-952" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-953"><a href="#cb7-953" aria-hidden="true" tabindex="-1"></a>We now plot only the scores from the SCANB samples.</span>
<span id="cb7-954"><a href="#cb7-954" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-955"><a href="#cb7-955" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=5}</span></span>
<span id="cb7-956"><a href="#cb7-956" aria-hidden="true" tabindex="-1"></a><span class="in">lapply(</span></span>
<span id="cb7-957"><a href="#cb7-957" aria-hidden="true" tabindex="-1"></a><span class="in">    gsva_scores_embeddings_scanb_within_tcga,</span></span>
<span id="cb7-958"><a href="#cb7-958" aria-hidden="true" tabindex="-1"></a><span class="in">    function(x){</span></span>
<span id="cb7-959"><a href="#cb7-959" aria-hidden="true" tabindex="-1"></a><span class="in">        sapply(</span></span>
<span id="cb7-960"><a href="#cb7-960" aria-hidden="true" tabindex="-1"></a><span class="in">            pathways_of_interest, </span></span>
<span id="cb7-961"><a href="#cb7-961" aria-hidden="true" tabindex="-1"></a><span class="in">            get_merged_df_scores,</span></span>
<span id="cb7-962"><a href="#cb7-962" aria-hidden="true" tabindex="-1"></a><span class="in">            df1 = x,</span></span>
<span id="cb7-963"><a href="#cb7-963" aria-hidden="true" tabindex="-1"></a><span class="in">            df2 = df_pca %&gt;%</span></span>
<span id="cb7-964"><a href="#cb7-964" aria-hidden="true" tabindex="-1"></a><span class="in">                dplyr::filter(cohort == "scanb"),</span></span>
<span id="cb7-965"><a href="#cb7-965" aria-hidden="true" tabindex="-1"></a><span class="in">            USE.NAMES = TRUE, </span></span>
<span id="cb7-966"><a href="#cb7-966" aria-hidden="true" tabindex="-1"></a><span class="in">            simplify = FALSE</span></span>
<span id="cb7-967"><a href="#cb7-967" aria-hidden="true" tabindex="-1"></a><span class="in">        ) %&gt;% </span></span>
<span id="cb7-968"><a href="#cb7-968" aria-hidden="true" tabindex="-1"></a><span class="in">            dplyr::bind_rows(.id = "pathway")</span></span>
<span id="cb7-969"><a href="#cb7-969" aria-hidden="true" tabindex="-1"></a><span class="in">    }</span></span>
<span id="cb7-970"><a href="#cb7-970" aria-hidden="true" tabindex="-1"></a><span class="in">) %&gt;%</span></span>
<span id="cb7-971"><a href="#cb7-971" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::bind_rows(.id = "sample_name") %&gt;%</span></span>
<span id="cb7-972"><a href="#cb7-972" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::ggplot(aes(x = df1, y = df2)) +</span></span>
<span id="cb7-973"><a href="#cb7-973" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_point() + </span></span>
<span id="cb7-974"><a href="#cb7-974" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::facet_wrap(~pathway) +</span></span>
<span id="cb7-975"><a href="#cb7-975" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_abline(slope = 1, intercept = 0, color = "red") +</span></span>
<span id="cb7-976"><a href="#cb7-976" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::labs(</span></span>
<span id="cb7-977"><a href="#cb7-977" aria-hidden="true" tabindex="-1"></a><span class="in">        x = "GSVA scores from regressed data",</span></span>
<span id="cb7-978"><a href="#cb7-978" aria-hidden="true" tabindex="-1"></a><span class="in">        y = "GSVA scores from unregressed data",</span></span>
<span id="cb7-979"><a href="#cb7-979" aria-hidden="true" tabindex="-1"></a><span class="in">        title = "TCGA scores"</span></span>
<span id="cb7-980"><a href="#cb7-980" aria-hidden="true" tabindex="-1"></a><span class="in">    ) +</span></span>
<span id="cb7-981"><a href="#cb7-981" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme_bw()</span></span>
<span id="cb7-982"><a href="#cb7-982" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-983"><a href="#cb7-983" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-984"><a href="#cb7-984" aria-hidden="true" tabindex="-1"></a>So it seems there are still dataset related problems there when calculating</span>
<span id="cb7-985"><a href="#cb7-985" aria-hidden="true" tabindex="-1"></a>the scores. The scores are positively correlated, but they are not quite</span>
<span id="cb7-986"><a href="#cb7-986" aria-hidden="true" tabindex="-1"></a>the same as only using the SCANB dataset. Moreover, for G2M checkpoint the </span>
<span id="cb7-987"><a href="#cb7-987" aria-hidden="true" tabindex="-1"></a>scores are concentrated around -0.4 from the regressed data but </span>
<span id="cb7-988"><a href="#cb7-988" aria-hidden="true" tabindex="-1"></a>they are in a range going from - 0.4 to 0.1 in the unregressed data. </span>
<span id="cb7-989"><a href="#cb7-989" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-990"><a href="#cb7-990" aria-hidden="true" tabindex="-1"></a>If we try to instead add some samples from the SCANB dataset as well </span>
<span id="cb7-991"><a href="#cb7-991" aria-hidden="true" tabindex="-1"></a>randomly into the mix. How well does the scores get compared to the </span>
<span id="cb7-992"><a href="#cb7-992" aria-hidden="true" tabindex="-1"></a>"ground truth"? </span>
<span id="cb7-993"><a href="#cb7-993" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-994"><a href="#cb7-994" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb7-995"><a href="#cb7-995" aria-hidden="true" tabindex="-1"></a><span class="in">scanb_samples &lt;- df_pca %&gt;%</span></span>
<span id="cb7-996"><a href="#cb7-996" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::filter(cohort == "scanb") %&gt;%</span></span>
<span id="cb7-997"><a href="#cb7-997" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::slice_sample(n = 30) %&gt;%</span></span>
<span id="cb7-998"><a href="#cb7-998" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::pull(sample_name)</span></span>
<span id="cb7-999"><a href="#cb7-999" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1000"><a href="#cb7-1000" aria-hidden="true" tabindex="-1"></a><span class="in">gsva_scores_embeddings_scanb_within_tcga_scanb &lt;- parallel::mclapply(</span></span>
<span id="cb7-1001"><a href="#cb7-1001" aria-hidden="true" tabindex="-1"></a><span class="in">    scanb_samples,</span></span>
<span id="cb7-1002"><a href="#cb7-1002" aria-hidden="true" tabindex="-1"></a><span class="in">    function(x){</span></span>
<span id="cb7-1003"><a href="#cb7-1003" aria-hidden="true" tabindex="-1"></a><span class="in">        GSVA::gsva(</span></span>
<span id="cb7-1004"><a href="#cb7-1004" aria-hidden="true" tabindex="-1"></a><span class="in">            df_pcs_regressed[, merged_col_data %&gt;%</span></span>
<span id="cb7-1005"><a href="#cb7-1005" aria-hidden="true" tabindex="-1"></a><span class="in">                                dplyr::filter(</span></span>
<span id="cb7-1006"><a href="#cb7-1006" aria-hidden="true" tabindex="-1"></a><span class="in">                                    cohort %in% c("tcga", "metabric", "scanb")</span></span>
<span id="cb7-1007"><a href="#cb7-1007" aria-hidden="true" tabindex="-1"></a><span class="in">                                ) %&gt;%</span></span>
<span id="cb7-1008"><a href="#cb7-1008" aria-hidden="true" tabindex="-1"></a><span class="in">                                dplyr::mutate(weights = dplyr::case_when(</span></span>
<span id="cb7-1009"><a href="#cb7-1009" aria-hidden="true" tabindex="-1"></a><span class="in">                                    cohort == "tcga" ~ 0.45,</span></span>
<span id="cb7-1010"><a href="#cb7-1010" aria-hidden="true" tabindex="-1"></a><span class="in">                                    cohort == "metabric" ~ 0.45,</span></span>
<span id="cb7-1011"><a href="#cb7-1011" aria-hidden="true" tabindex="-1"></a><span class="in">                                    cohort == "scanb" ~ 0.1</span></span>
<span id="cb7-1012"><a href="#cb7-1012" aria-hidden="true" tabindex="-1"></a><span class="in">                                )) %&gt;%</span></span>
<span id="cb7-1013"><a href="#cb7-1013" aria-hidden="true" tabindex="-1"></a><span class="in">                                dplyr::slice_sample(n = 150, weight_by = weights) %&gt;%</span></span>
<span id="cb7-1014"><a href="#cb7-1014" aria-hidden="true" tabindex="-1"></a><span class="in">                                dplyr::pull(sample_name) %&gt;%</span></span>
<span id="cb7-1015"><a href="#cb7-1015" aria-hidden="true" tabindex="-1"></a><span class="in">                                c(., x)],</span></span>
<span id="cb7-1016"><a href="#cb7-1016" aria-hidden="true" tabindex="-1"></a><span class="in">            gene_sets_,</span></span>
<span id="cb7-1017"><a href="#cb7-1017" aria-hidden="true" tabindex="-1"></a><span class="in">            verbose = FALSE</span></span>
<span id="cb7-1018"><a href="#cb7-1018" aria-hidden="true" tabindex="-1"></a><span class="in">        )   </span></span>
<span id="cb7-1019"><a href="#cb7-1019" aria-hidden="true" tabindex="-1"></a><span class="in">    },</span></span>
<span id="cb7-1020"><a href="#cb7-1020" aria-hidden="true" tabindex="-1"></a><span class="in">    mc.cores = 20</span></span>
<span id="cb7-1021"><a href="#cb7-1021" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-1022"><a href="#cb7-1022" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1023"><a href="#cb7-1023" aria-hidden="true" tabindex="-1"></a><span class="in">saveRDS(</span></span>
<span id="cb7-1024"><a href="#cb7-1024" aria-hidden="true" tabindex="-1"></a><span class="in">    gsva_scores_embeddings_scanb_within_tcga_scanb,</span></span>
<span id="cb7-1025"><a href="#cb7-1025" aria-hidden="true" tabindex="-1"></a><span class="in">    "../../results/rds_files/expanding_ember/gsva_scores_embeddings_scanb_within_tcga_scanb.rds"</span></span>
<span id="cb7-1026"><a href="#cb7-1026" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-1027"><a href="#cb7-1027" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-1028"><a href="#cb7-1028" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1029"><a href="#cb7-1029" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=5}</span></span>
<span id="cb7-1030"><a href="#cb7-1030" aria-hidden="true" tabindex="-1"></a><span class="in">lapply(</span></span>
<span id="cb7-1031"><a href="#cb7-1031" aria-hidden="true" tabindex="-1"></a><span class="in">    gsva_scores_embeddings_scanb_within_tcga_scanb,</span></span>
<span id="cb7-1032"><a href="#cb7-1032" aria-hidden="true" tabindex="-1"></a><span class="in">    function(x){</span></span>
<span id="cb7-1033"><a href="#cb7-1033" aria-hidden="true" tabindex="-1"></a><span class="in">        sapply(</span></span>
<span id="cb7-1034"><a href="#cb7-1034" aria-hidden="true" tabindex="-1"></a><span class="in">            pathways_of_interest, </span></span>
<span id="cb7-1035"><a href="#cb7-1035" aria-hidden="true" tabindex="-1"></a><span class="in">            get_merged_df_scores,</span></span>
<span id="cb7-1036"><a href="#cb7-1036" aria-hidden="true" tabindex="-1"></a><span class="in">            df1 = x,</span></span>
<span id="cb7-1037"><a href="#cb7-1037" aria-hidden="true" tabindex="-1"></a><span class="in">            df2 = df_pca %&gt;%</span></span>
<span id="cb7-1038"><a href="#cb7-1038" aria-hidden="true" tabindex="-1"></a><span class="in">                dplyr::filter(cohort == "scanb"),</span></span>
<span id="cb7-1039"><a href="#cb7-1039" aria-hidden="true" tabindex="-1"></a><span class="in">            USE.NAMES = TRUE, </span></span>
<span id="cb7-1040"><a href="#cb7-1040" aria-hidden="true" tabindex="-1"></a><span class="in">            simplify = FALSE</span></span>
<span id="cb7-1041"><a href="#cb7-1041" aria-hidden="true" tabindex="-1"></a><span class="in">        ) %&gt;% </span></span>
<span id="cb7-1042"><a href="#cb7-1042" aria-hidden="true" tabindex="-1"></a><span class="in">            dplyr::bind_rows(.id = "pathway")</span></span>
<span id="cb7-1043"><a href="#cb7-1043" aria-hidden="true" tabindex="-1"></a><span class="in">    }</span></span>
<span id="cb7-1044"><a href="#cb7-1044" aria-hidden="true" tabindex="-1"></a><span class="in">) %&gt;%</span></span>
<span id="cb7-1045"><a href="#cb7-1045" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::bind_rows(.id = "sample_name") %&gt;%</span></span>
<span id="cb7-1046"><a href="#cb7-1046" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::ggplot(aes(x = df1, y = df2)) +</span></span>
<span id="cb7-1047"><a href="#cb7-1047" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_point() + </span></span>
<span id="cb7-1048"><a href="#cb7-1048" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::facet_wrap(~pathway) +</span></span>
<span id="cb7-1049"><a href="#cb7-1049" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_abline(slope = 1, intercept = 0, color = "red") +</span></span>
<span id="cb7-1050"><a href="#cb7-1050" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::labs(</span></span>
<span id="cb7-1051"><a href="#cb7-1051" aria-hidden="true" tabindex="-1"></a><span class="in">        x = "GSVA scores from regressed data",</span></span>
<span id="cb7-1052"><a href="#cb7-1052" aria-hidden="true" tabindex="-1"></a><span class="in">        y = "GSVA scores from unregressed data",</span></span>
<span id="cb7-1053"><a href="#cb7-1053" aria-hidden="true" tabindex="-1"></a><span class="in">        title = "TCGA scores"</span></span>
<span id="cb7-1054"><a href="#cb7-1054" aria-hidden="true" tabindex="-1"></a><span class="in">    ) +</span></span>
<span id="cb7-1055"><a href="#cb7-1055" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme_bw()</span></span>
<span id="cb7-1056"><a href="#cb7-1056" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-1057"><a href="#cb7-1057" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1058"><a href="#cb7-1058" aria-hidden="true" tabindex="-1"></a>It does not work so well when including some SCANB samples together </span>
<span id="cb7-1059"><a href="#cb7-1059" aria-hidden="true" tabindex="-1"></a>with the METABRIC and TCGA samples. We are trying to achieve a</span>
<span id="cb7-1060"><a href="#cb7-1060" aria-hidden="true" tabindex="-1"></a>correlation of almost 1 with very low variability since we want to use</span>
<span id="cb7-1061"><a href="#cb7-1061" aria-hidden="true" tabindex="-1"></a>the scores clinically. </span>
<span id="cb7-1062"><a href="#cb7-1062" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1063"><a href="#cb7-1063" aria-hidden="true" tabindex="-1"></a>Let us check the distribution of some genes for SCANB and TCGA after</span>
<span id="cb7-1064"><a href="#cb7-1064" aria-hidden="true" tabindex="-1"></a>regressing out the first two PCs.</span>
<span id="cb7-1065"><a href="#cb7-1065" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1066"><a href="#cb7-1066" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=8}</span></span>
<span id="cb7-1067"><a href="#cb7-1067" aria-hidden="true" tabindex="-1"></a><span class="in">genes_histogram &lt;- gene_sets_$HALLMARK_G2M_CHECKPOINT</span></span>
<span id="cb7-1068"><a href="#cb7-1068" aria-hidden="true" tabindex="-1"></a><span class="in">genes_histogram &lt;- intersect(genes_histogram, rownames(df_pcs_regressed))</span></span>
<span id="cb7-1069"><a href="#cb7-1069" aria-hidden="true" tabindex="-1"></a><span class="in">genes_histogram &lt;- sample(genes_histogram, 30)</span></span>
<span id="cb7-1070"><a href="#cb7-1070" aria-hidden="true" tabindex="-1"></a><span class="in">genes_histogram &lt;- c(genes_histogram, "MKI67") %&gt;% unique</span></span>
<span id="cb7-1071"><a href="#cb7-1071" aria-hidden="true" tabindex="-1"></a><span class="in">df_pcs_regressed %&gt;% </span></span>
<span id="cb7-1072"><a href="#cb7-1072" aria-hidden="true" tabindex="-1"></a><span class="in">    t %&gt;%</span></span>
<span id="cb7-1073"><a href="#cb7-1073" aria-hidden="true" tabindex="-1"></a><span class="in">    data.frame %&gt;%</span></span>
<span id="cb7-1074"><a href="#cb7-1074" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::select(dplyr::all_of(genes_histogram)) %&gt;%</span></span>
<span id="cb7-1075"><a href="#cb7-1075" aria-hidden="true" tabindex="-1"></a><span class="in">    tibble::rownames_to_column(var = "sample_name") %&gt;%</span></span>
<span id="cb7-1076"><a href="#cb7-1076" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::inner_join(</span></span>
<span id="cb7-1077"><a href="#cb7-1077" aria-hidden="true" tabindex="-1"></a><span class="in">        .,</span></span>
<span id="cb7-1078"><a href="#cb7-1078" aria-hidden="true" tabindex="-1"></a><span class="in">        merged_col_data,</span></span>
<span id="cb7-1079"><a href="#cb7-1079" aria-hidden="true" tabindex="-1"></a><span class="in">        by = "sample_name"</span></span>
<span id="cb7-1080"><a href="#cb7-1080" aria-hidden="true" tabindex="-1"></a><span class="in">    ) %&gt;%</span></span>
<span id="cb7-1081"><a href="#cb7-1081" aria-hidden="true" tabindex="-1"></a><span class="in">    tidyr::pivot_longer(</span></span>
<span id="cb7-1082"><a href="#cb7-1082" aria-hidden="true" tabindex="-1"></a><span class="in">        cols = dplyr::all_of(genes_histogram),</span></span>
<span id="cb7-1083"><a href="#cb7-1083" aria-hidden="true" tabindex="-1"></a><span class="in">        values_to = "expression",</span></span>
<span id="cb7-1084"><a href="#cb7-1084" aria-hidden="true" tabindex="-1"></a><span class="in">        names_to = "gene"</span></span>
<span id="cb7-1085"><a href="#cb7-1085" aria-hidden="true" tabindex="-1"></a><span class="in">    ) %&gt;% </span></span>
<span id="cb7-1086"><a href="#cb7-1086" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::ggplot(aes(</span></span>
<span id="cb7-1087"><a href="#cb7-1087" aria-hidden="true" tabindex="-1"></a><span class="in">        x = expression,</span></span>
<span id="cb7-1088"><a href="#cb7-1088" aria-hidden="true" tabindex="-1"></a><span class="in">        fill = cohort,</span></span>
<span id="cb7-1089"><a href="#cb7-1089" aria-hidden="true" tabindex="-1"></a><span class="in">        alpha = 0.5</span></span>
<span id="cb7-1090"><a href="#cb7-1090" aria-hidden="true" tabindex="-1"></a><span class="in">    )) + </span></span>
<span id="cb7-1091"><a href="#cb7-1091" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_density(alpha = 0.5) +</span></span>
<span id="cb7-1092"><a href="#cb7-1092" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::facet_wrap(~gene, scales = "free") + </span></span>
<span id="cb7-1093"><a href="#cb7-1093" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme_bw()</span></span>
<span id="cb7-1094"><a href="#cb7-1094" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-1095"><a href="#cb7-1095" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1096"><a href="#cb7-1096" aria-hidden="true" tabindex="-1"></a>We see that even though the expression profile of the samples are </span>
<span id="cb7-1097"><a href="#cb7-1097" aria-hidden="true" tabindex="-1"></a>overlapping there are discrepancies. That is why when using GSVA </span>
<span id="cb7-1098"><a href="#cb7-1098" aria-hidden="true" tabindex="-1"></a>it does not work so well perhaps.</span>
<span id="cb7-1099"><a href="#cb7-1099" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1100"><a href="#cb7-1100" aria-hidden="true" tabindex="-1"></a><span class="fu">## Scoring strategy with regressed data</span></span>
<span id="cb7-1101"><a href="#cb7-1101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1102"><a href="#cb7-1102" aria-hidden="true" tabindex="-1"></a>We now try to use a slightly different scoring strategy. We can think of </span>
<span id="cb7-1103"><a href="#cb7-1103" aria-hidden="true" tabindex="-1"></a>our data in a log scale, so what we do now instead is to sum all of </span>
<span id="cb7-1104"><a href="#cb7-1104" aria-hidden="true" tabindex="-1"></a>our genes of interest and then subtract an average of the housekeeping</span>
<span id="cb7-1105"><a href="#cb7-1105" aria-hidden="true" tabindex="-1"></a>genes expression levels.</span>
<span id="cb7-1106"><a href="#cb7-1106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1107"><a href="#cb7-1107" aria-hidden="true" tabindex="-1"></a>First we check the distribution of this average for each cohort</span>
<span id="cb7-1108"><a href="#cb7-1108" aria-hidden="true" tabindex="-1"></a>separately.</span>
<span id="cb7-1109"><a href="#cb7-1109" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1112"><a href="#cb7-1112" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb7-1113"><a href="#cb7-1113" aria-hidden="true" tabindex="-1"></a>hk_genes <span class="ot">&lt;-</span> singscore<span class="sc">::</span><span class="fu">getStableGenes</span>(</span>
<span id="cb7-1114"><a href="#cb7-1114" aria-hidden="true" tabindex="-1"></a>    <span class="dv">50</span>, </span>
<span id="cb7-1115"><a href="#cb7-1115" aria-hidden="true" tabindex="-1"></a>    <span class="at">type =</span> <span class="st">"carcinoma"</span>, </span>
<span id="cb7-1116"><a href="#cb7-1116" aria-hidden="true" tabindex="-1"></a>    <span class="at">id =</span> <span class="st">"geneid"</span></span>
<span id="cb7-1117"><a href="#cb7-1117" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> </span>
<span id="cb7-1118"><a href="#cb7-1118" aria-hidden="true" tabindex="-1"></a>    <span class="fu">intersect</span>(., <span class="fu">rownames</span>(df_pcs_regressed))</span>
<span id="cb7-1119"><a href="#cb7-1119" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1120"><a href="#cb7-1120" aria-hidden="true" tabindex="-1"></a>avg_hkg_pcs_regressed <span class="ot">&lt;-</span> df_pcs_regressed <span class="sc">%&gt;%</span> </span>
<span id="cb7-1121"><a href="#cb7-1121" aria-hidden="true" tabindex="-1"></a>    t <span class="sc">%&gt;%</span></span>
<span id="cb7-1122"><a href="#cb7-1122" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span></span>
<span id="cb7-1123"><a href="#cb7-1123" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(dplyr<span class="sc">::</span><span class="fu">all_of</span>(hk_genes)) <span class="sc">%&gt;%</span></span>
<span id="cb7-1124"><a href="#cb7-1124" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">avg_hkg =</span> <span class="fu">rowMeans</span>(dplyr<span class="sc">::</span><span class="fu">select</span>(</span>
<span id="cb7-1125"><a href="#cb7-1125" aria-hidden="true" tabindex="-1"></a>        ., </span>
<span id="cb7-1126"><a href="#cb7-1126" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">all_of</span>(hk_genes)</span>
<span id="cb7-1127"><a href="#cb7-1127" aria-hidden="true" tabindex="-1"></a>    ))) <span class="sc">%&gt;%</span></span>
<span id="cb7-1128"><a href="#cb7-1128" aria-hidden="true" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"sample_name"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-1129"><a href="#cb7-1129" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">inner_join</span>(</span>
<span id="cb7-1130"><a href="#cb7-1130" aria-hidden="true" tabindex="-1"></a>        .,</span>
<span id="cb7-1131"><a href="#cb7-1131" aria-hidden="true" tabindex="-1"></a>        merged_col_data,</span>
<span id="cb7-1132"><a href="#cb7-1132" aria-hidden="true" tabindex="-1"></a>        <span class="at">by =</span> <span class="st">"sample_name"</span></span>
<span id="cb7-1133"><a href="#cb7-1133" aria-hidden="true" tabindex="-1"></a>    ) </span>
<span id="cb7-1134"><a href="#cb7-1134" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-1135"><a href="#cb7-1135" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1136"><a href="#cb7-1136" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width = 8, fig.height=6}</span></span>
<span id="cb7-1137"><a href="#cb7-1137" aria-hidden="true" tabindex="-1"></a><span class="in">#| label: fig-hkg-regressed</span></span>
<span id="cb7-1138"><a href="#cb7-1138" aria-hidden="true" tabindex="-1"></a><span class="in">#| fig-cap: Distribution of the average of the housekeeping genes</span></span>
<span id="cb7-1139"><a href="#cb7-1139" aria-hidden="true" tabindex="-1"></a><span class="in">#|     for distinct cohorts. </span></span>
<span id="cb7-1140"><a href="#cb7-1140" aria-hidden="true" tabindex="-1"></a><span class="in">avg_hkg_pcs_regressed %&gt;% </span></span>
<span id="cb7-1141"><a href="#cb7-1141" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::ggplot(aes(</span></span>
<span id="cb7-1142"><a href="#cb7-1142" aria-hidden="true" tabindex="-1"></a><span class="in">        x = avg_hkg,</span></span>
<span id="cb7-1143"><a href="#cb7-1143" aria-hidden="true" tabindex="-1"></a><span class="in">        fill = cohort,</span></span>
<span id="cb7-1144"><a href="#cb7-1144" aria-hidden="true" tabindex="-1"></a><span class="in">        alpha = 0.5</span></span>
<span id="cb7-1145"><a href="#cb7-1145" aria-hidden="true" tabindex="-1"></a><span class="in">    )) + </span></span>
<span id="cb7-1146"><a href="#cb7-1146" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_density(alpha = 0.5) +</span></span>
<span id="cb7-1147"><a href="#cb7-1147" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::labs(</span></span>
<span id="cb7-1148"><a href="#cb7-1148" aria-hidden="true" tabindex="-1"></a><span class="in">        x = "Average of housekeeping genes on regressed data",</span></span>
<span id="cb7-1149"><a href="#cb7-1149" aria-hidden="true" tabindex="-1"></a><span class="in">        y = "Density"</span></span>
<span id="cb7-1150"><a href="#cb7-1150" aria-hidden="true" tabindex="-1"></a><span class="in">    ) +</span></span>
<span id="cb7-1151"><a href="#cb7-1151" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme_bw(base_size = 30) +</span></span>
<span id="cb7-1152"><a href="#cb7-1152" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme(</span></span>
<span id="cb7-1153"><a href="#cb7-1153" aria-hidden="true" tabindex="-1"></a><span class="in">        legend.position = c(0.8, 0.6),</span></span>
<span id="cb7-1154"><a href="#cb7-1154" aria-hidden="true" tabindex="-1"></a><span class="in">        legend.background = element_blank()</span></span>
<span id="cb7-1155"><a href="#cb7-1155" aria-hidden="true" tabindex="-1"></a><span class="in">    ) +</span></span>
<span id="cb7-1156"><a href="#cb7-1156" aria-hidden="true" tabindex="-1"></a><span class="in">    change_guides_point() +</span></span>
<span id="cb7-1157"><a href="#cb7-1157" aria-hidden="true" tabindex="-1"></a><span class="in">    change_plot_aes_point()</span></span>
<span id="cb7-1158"><a href="#cb7-1158" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-1159"><a href="#cb7-1159" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1160"><a href="#cb7-1160" aria-hidden="true" tabindex="-1"></a>Indeed the tcga and metabric cohorts have an overlapping </span>
<span id="cb7-1161"><a href="#cb7-1161" aria-hidden="true" tabindex="-1"></a>distribution, but the SCANB does not overlap completely, it is</span>
<span id="cb7-1162"><a href="#cb7-1162" aria-hidden="true" tabindex="-1"></a>actually shifted. One should still take into account that </span>
<span id="cb7-1163"><a href="#cb7-1163" aria-hidden="true" tabindex="-1"></a>the distributions are very close to 0, meaning that the </span>
<span id="cb7-1164"><a href="#cb7-1164" aria-hidden="true" tabindex="-1"></a>housekeeping genes have very close to 0 values, as expected here in this case.</span>
<span id="cb7-1165"><a href="#cb7-1165" aria-hidden="true" tabindex="-1"></a>We now calculate the score by taking this into account. </span>
<span id="cb7-1166"><a href="#cb7-1166" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1169"><a href="#cb7-1169" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb7-1170"><a href="#cb7-1170" aria-hidden="true" tabindex="-1"></a>pathways_of_interest <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb7-1171"><a href="#cb7-1171" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_ESTROGEN_RESPONSE_EARLY"</span> <span class="ot">=</span> <span class="st">"Estrogen Early"</span>,</span>
<span id="cb7-1172"><a href="#cb7-1172" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_G2M_CHECKPOINT"</span> <span class="ot">=</span> <span class="st">"G2M"</span>, </span>
<span id="cb7-1173"><a href="#cb7-1173" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_E2F_TARGETS"</span> <span class="ot">=</span> <span class="st">"E2F"</span>, </span>
<span id="cb7-1174"><a href="#cb7-1174" aria-hidden="true" tabindex="-1"></a>    <span class="st">"SET_ERPR"</span> <span class="ot">=</span> <span class="st">"SET ER/PR"</span>,</span>
<span id="cb7-1175"><a href="#cb7-1175" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION"</span> <span class="ot">=</span> <span class="st">"EMT"</span>,</span>
<span id="cb7-1176"><a href="#cb7-1176" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_PI3K_AKT_MTOR_SIGNALING"</span> <span class="ot">=</span> <span class="st">"PI3K AKT MTOR"</span>,</span>
<span id="cb7-1177"><a href="#cb7-1177" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_DNA_REPAIR"</span> <span class="ot">=</span> <span class="st">"DNA Repair"</span> ,</span>
<span id="cb7-1178"><a href="#cb7-1178" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_INTERFERON_GAMMA_RESPONSE"</span> <span class="ot">=</span> <span class="st">"Interferon Gamma"</span>,</span>
<span id="cb7-1179"><a href="#cb7-1179" aria-hidden="true" tabindex="-1"></a>    <span class="st">"random_200"</span> <span class="ot">=</span> <span class="st">"Random 200"</span></span>
<span id="cb7-1180"><a href="#cb7-1180" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb7-1181"><a href="#cb7-1181" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-1182"><a href="#cb7-1182" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1183"><a href="#cb7-1183" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb7-1184"><a href="#cb7-1184" aria-hidden="true" tabindex="-1"></a><span class="in">scores &lt;- sapply(</span></span>
<span id="cb7-1185"><a href="#cb7-1185" aria-hidden="true" tabindex="-1"></a><span class="in">    gene_sets_[names(pathways_of_interest)],</span></span>
<span id="cb7-1186"><a href="#cb7-1186" aria-hidden="true" tabindex="-1"></a><span class="in">    function(gene_set, df_pcs_regressed, avg_hkg_pcs_regressed){</span></span>
<span id="cb7-1187"><a href="#cb7-1187" aria-hidden="true" tabindex="-1"></a><span class="in">        </span></span>
<span id="cb7-1188"><a href="#cb7-1188" aria-hidden="true" tabindex="-1"></a><span class="in">        proportion &lt;- 1</span></span>
<span id="cb7-1189"><a href="#cb7-1189" aria-hidden="true" tabindex="-1"></a><span class="in">        nb_genes &lt;- floor(proportion * length(gene_set))</span></span>
<span id="cb7-1190"><a href="#cb7-1190" aria-hidden="true" tabindex="-1"></a><span class="in">        gene_set &lt;- intersect(</span></span>
<span id="cb7-1191"><a href="#cb7-1191" aria-hidden="true" tabindex="-1"></a><span class="in">            sample(gene_set, nb_genes),</span></span>
<span id="cb7-1192"><a href="#cb7-1192" aria-hidden="true" tabindex="-1"></a><span class="in">            rownames(df_pcs_regressed)</span></span>
<span id="cb7-1193"><a href="#cb7-1193" aria-hidden="true" tabindex="-1"></a><span class="in">        )</span></span>
<span id="cb7-1194"><a href="#cb7-1194" aria-hidden="true" tabindex="-1"></a><span class="in">        </span></span>
<span id="cb7-1195"><a href="#cb7-1195" aria-hidden="true" tabindex="-1"></a><span class="in">        # the code below was tried and it didnt work</span></span>
<span id="cb7-1196"><a href="#cb7-1196" aria-hidden="true" tabindex="-1"></a><span class="in">        # 1) Mean of the genes and subtract the average of housekeeping genes</span></span>
<span id="cb7-1197"><a href="#cb7-1197" aria-hidden="true" tabindex="-1"></a><span class="in">        # colMeans(df_pcs_regressed[gene_set, ]) - avg_hkg_pcs_regressed$avg_hkg</span></span>
<span id="cb7-1198"><a href="#cb7-1198" aria-hidden="true" tabindex="-1"></a><span class="in">        # 2) top quantile. this one works surprisingly well considering </span></span>
<span id="cb7-1199"><a href="#cb7-1199" aria-hidden="true" tabindex="-1"></a><span class="in">        # what was done. It could be further developped</span></span>
<span id="cb7-1200"><a href="#cb7-1200" aria-hidden="true" tabindex="-1"></a><span class="in">        # apply(</span></span>
<span id="cb7-1201"><a href="#cb7-1201" aria-hidden="true" tabindex="-1"></a><span class="in">        #     X = df_pcs_regressed[gene_set, ], </span></span>
<span id="cb7-1202"><a href="#cb7-1202" aria-hidden="true" tabindex="-1"></a><span class="in">        #     MARGIN = 2,</span></span>
<span id="cb7-1203"><a href="#cb7-1203" aria-hidden="true" tabindex="-1"></a><span class="in">        #     quantile,</span></span>
<span id="cb7-1204"><a href="#cb7-1204" aria-hidden="true" tabindex="-1"></a><span class="in">        #     probs = 0.75</span></span>
<span id="cb7-1205"><a href="#cb7-1205" aria-hidden="true" tabindex="-1"></a><span class="in">        # )</span></span>
<span id="cb7-1206"><a href="#cb7-1206" aria-hidden="true" tabindex="-1"></a><span class="in">        # 3) pairwise differences of all genes. Small differences means genes</span></span>
<span id="cb7-1207"><a href="#cb7-1207" aria-hidden="true" tabindex="-1"></a><span class="in">        # are going in the same direction. Big differences means genes are</span></span>
<span id="cb7-1208"><a href="#cb7-1208" aria-hidden="true" tabindex="-1"></a><span class="in">        # going in opposite directions.</span></span>
<span id="cb7-1209"><a href="#cb7-1209" aria-hidden="true" tabindex="-1"></a><span class="in">        # apply(</span></span>
<span id="cb7-1210"><a href="#cb7-1210" aria-hidden="true" tabindex="-1"></a><span class="in">        #     X = df_pcs_regressed[gene_set, ],</span></span>
<span id="cb7-1211"><a href="#cb7-1211" aria-hidden="true" tabindex="-1"></a><span class="in">        #     MARGIN = 2,</span></span>
<span id="cb7-1212"><a href="#cb7-1212" aria-hidden="true" tabindex="-1"></a><span class="in">        #     function(v){</span></span>
<span id="cb7-1213"><a href="#cb7-1213" aria-hidden="true" tabindex="-1"></a><span class="in">        #         quantile(as.vector(dist(v)), probs = 0.95)</span></span>
<span id="cb7-1214"><a href="#cb7-1214" aria-hidden="true" tabindex="-1"></a><span class="in">        #     }</span></span>
<span id="cb7-1215"><a href="#cb7-1215" aria-hidden="true" tabindex="-1"></a><span class="in">        # )</span></span>
<span id="cb7-1216"><a href="#cb7-1216" aria-hidden="true" tabindex="-1"></a><span class="in">        # 4) Scale all genes for each sample and then take the average of </span></span>
<span id="cb7-1217"><a href="#cb7-1217" aria-hidden="true" tabindex="-1"></a><span class="in">        # the genes. Works quite well, still problem with SCANB</span></span>
<span id="cb7-1218"><a href="#cb7-1218" aria-hidden="true" tabindex="-1"></a><span class="in">        # apply(</span></span>
<span id="cb7-1219"><a href="#cb7-1219" aria-hidden="true" tabindex="-1"></a><span class="in">        #     X = df_pcs_regressed,</span></span>
<span id="cb7-1220"><a href="#cb7-1220" aria-hidden="true" tabindex="-1"></a><span class="in">        #     MARGIN = 2,</span></span>
<span id="cb7-1221"><a href="#cb7-1221" aria-hidden="true" tabindex="-1"></a><span class="in">        #     function(v, name_genes, gene_set){</span></span>
<span id="cb7-1222"><a href="#cb7-1222" aria-hidden="true" tabindex="-1"></a><span class="in">        #         mean(scale(v)[which(name_genes %in% gene_set)])</span></span>
<span id="cb7-1223"><a href="#cb7-1223" aria-hidden="true" tabindex="-1"></a><span class="in">        #     },</span></span>
<span id="cb7-1224"><a href="#cb7-1224" aria-hidden="true" tabindex="-1"></a><span class="in">        #     name_genes = rownames(df_pcs_regressed),</span></span>
<span id="cb7-1225"><a href="#cb7-1225" aria-hidden="true" tabindex="-1"></a><span class="in">        #     gene_set = gene_set</span></span>
<span id="cb7-1226"><a href="#cb7-1226" aria-hidden="true" tabindex="-1"></a><span class="in">        # )</span></span>
<span id="cb7-1227"><a href="#cb7-1227" aria-hidden="true" tabindex="-1"></a><span class="in">        # 5) Scale and then calculate 100 times the score with subset of </span></span>
<span id="cb7-1228"><a href="#cb7-1228" aria-hidden="true" tabindex="-1"></a><span class="in">        # genes and then take average value. </span></span>
<span id="cb7-1229"><a href="#cb7-1229" aria-hidden="true" tabindex="-1"></a><span class="in">        # apply(</span></span>
<span id="cb7-1230"><a href="#cb7-1230" aria-hidden="true" tabindex="-1"></a><span class="in">        #     X = df_pcs_regressed[, sample(colnames(df_pcs_regressed), size = 500)],</span></span>
<span id="cb7-1231"><a href="#cb7-1231" aria-hidden="true" tabindex="-1"></a><span class="in">        #     MARGIN = 2,</span></span>
<span id="cb7-1232"><a href="#cb7-1232" aria-hidden="true" tabindex="-1"></a><span class="in">        #     function(v, name_genes, gene_set){</span></span>
<span id="cb7-1233"><a href="#cb7-1233" aria-hidden="true" tabindex="-1"></a><span class="in">        #         </span></span>
<span id="cb7-1234"><a href="#cb7-1234" aria-hidden="true" tabindex="-1"></a><span class="in">        #         mean(sapply(1:100,</span></span>
<span id="cb7-1235"><a href="#cb7-1235" aria-hidden="true" tabindex="-1"></a><span class="in">        #             function(i, x, gene_set, name_genes){</span></span>
<span id="cb7-1236"><a href="#cb7-1236" aria-hidden="true" tabindex="-1"></a><span class="in">        #                 proportion &lt;- 0.2</span></span>
<span id="cb7-1237"><a href="#cb7-1237" aria-hidden="true" tabindex="-1"></a><span class="in">        #                 nb_genes &lt;- floor(proportion * length(gene_set))</span></span>
<span id="cb7-1238"><a href="#cb7-1238" aria-hidden="true" tabindex="-1"></a><span class="in">        #                 gene_set &lt;- sample(gene_set, nb_genes)</span></span>
<span id="cb7-1239"><a href="#cb7-1239" aria-hidden="true" tabindex="-1"></a><span class="in">        #                 mean(x[which(name_genes %in% gene_set)])</span></span>
<span id="cb7-1240"><a href="#cb7-1240" aria-hidden="true" tabindex="-1"></a><span class="in">        #             },</span></span>
<span id="cb7-1241"><a href="#cb7-1241" aria-hidden="true" tabindex="-1"></a><span class="in">        #             x = scale(v),</span></span>
<span id="cb7-1242"><a href="#cb7-1242" aria-hidden="true" tabindex="-1"></a><span class="in">        #             gene_set = gene_set,</span></span>
<span id="cb7-1243"><a href="#cb7-1243" aria-hidden="true" tabindex="-1"></a><span class="in">        #             name_genes = name_genes</span></span>
<span id="cb7-1244"><a href="#cb7-1244" aria-hidden="true" tabindex="-1"></a><span class="in">        #         ))</span></span>
<span id="cb7-1245"><a href="#cb7-1245" aria-hidden="true" tabindex="-1"></a><span class="in">        #     },</span></span>
<span id="cb7-1246"><a href="#cb7-1246" aria-hidden="true" tabindex="-1"></a><span class="in">        #     name_genes = rownames(df_pcs_regressed),</span></span>
<span id="cb7-1247"><a href="#cb7-1247" aria-hidden="true" tabindex="-1"></a><span class="in">        #     gene_set = gene_set</span></span>
<span id="cb7-1248"><a href="#cb7-1248" aria-hidden="true" tabindex="-1"></a><span class="in">        # )</span></span>
<span id="cb7-1249"><a href="#cb7-1249" aria-hidden="true" tabindex="-1"></a><span class="in">        </span></span>
<span id="cb7-1250"><a href="#cb7-1250" aria-hidden="true" tabindex="-1"></a><span class="in">        colSums(df_pcs_regressed[gene_set, ])</span></span>
<span id="cb7-1251"><a href="#cb7-1251" aria-hidden="true" tabindex="-1"></a><span class="in">            </span></span>
<span id="cb7-1252"><a href="#cb7-1252" aria-hidden="true" tabindex="-1"></a><span class="in">    },</span></span>
<span id="cb7-1253"><a href="#cb7-1253" aria-hidden="true" tabindex="-1"></a><span class="in">    df_pcs_regressed = df_pcs_regressed,</span></span>
<span id="cb7-1254"><a href="#cb7-1254" aria-hidden="true" tabindex="-1"></a><span class="in">    avg_hkg_pcs_regressed = avg_hkg_pcs_regressed</span></span>
<span id="cb7-1255"><a href="#cb7-1255" aria-hidden="true" tabindex="-1"></a><span class="in">) %&gt;% </span></span>
<span id="cb7-1256"><a href="#cb7-1256" aria-hidden="true" tabindex="-1"></a><span class="in">    data.frame %&gt;%</span></span>
<span id="cb7-1257"><a href="#cb7-1257" aria-hidden="true" tabindex="-1"></a><span class="in">    tibble::rownames_to_column(var = "sample_name")</span></span>
<span id="cb7-1258"><a href="#cb7-1258" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1259"><a href="#cb7-1259" aria-hidden="true" tabindex="-1"></a><span class="in">scores &lt;- dplyr::inner_join(</span></span>
<span id="cb7-1260"><a href="#cb7-1260" aria-hidden="true" tabindex="-1"></a><span class="in">    scores %&gt;% </span></span>
<span id="cb7-1261"><a href="#cb7-1261" aria-hidden="true" tabindex="-1"></a><span class="in">        tidyr::pivot_longer(</span></span>
<span id="cb7-1262"><a href="#cb7-1262" aria-hidden="true" tabindex="-1"></a><span class="in">            cols = dplyr::all_of(names(pathways_of_interest)),</span></span>
<span id="cb7-1263"><a href="#cb7-1263" aria-hidden="true" tabindex="-1"></a><span class="in">            values_to = "score_hkg",</span></span>
<span id="cb7-1264"><a href="#cb7-1264" aria-hidden="true" tabindex="-1"></a><span class="in">            names_to = "pathway"</span></span>
<span id="cb7-1265"><a href="#cb7-1265" aria-hidden="true" tabindex="-1"></a><span class="in">        ),</span></span>
<span id="cb7-1266"><a href="#cb7-1266" aria-hidden="true" tabindex="-1"></a><span class="in">    df_pca %&gt;%</span></span>
<span id="cb7-1267"><a href="#cb7-1267" aria-hidden="true" tabindex="-1"></a><span class="in">        tidyr::pivot_longer(</span></span>
<span id="cb7-1268"><a href="#cb7-1268" aria-hidden="true" tabindex="-1"></a><span class="in">            cols = dplyr::all_of(names(pathways_of_interest)),</span></span>
<span id="cb7-1269"><a href="#cb7-1269" aria-hidden="true" tabindex="-1"></a><span class="in">            values_to = "score_og",</span></span>
<span id="cb7-1270"><a href="#cb7-1270" aria-hidden="true" tabindex="-1"></a><span class="in">            names_to = "pathway"</span></span>
<span id="cb7-1271"><a href="#cb7-1271" aria-hidden="true" tabindex="-1"></a><span class="in">        ),</span></span>
<span id="cb7-1272"><a href="#cb7-1272" aria-hidden="true" tabindex="-1"></a><span class="in">    by = c("sample_name", "pathway")</span></span>
<span id="cb7-1273"><a href="#cb7-1273" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-1274"><a href="#cb7-1274" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1275"><a href="#cb7-1275" aria-hidden="true" tabindex="-1"></a><span class="in">saveRDS(</span></span>
<span id="cb7-1276"><a href="#cb7-1276" aria-hidden="true" tabindex="-1"></a><span class="in">    scores, </span></span>
<span id="cb7-1277"><a href="#cb7-1277" aria-hidden="true" tabindex="-1"></a><span class="in">    "../../results/rds_files/expanding_ember/scores_regressed_only.rds"</span></span>
<span id="cb7-1278"><a href="#cb7-1278" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-1279"><a href="#cb7-1279" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-1280"><a href="#cb7-1280" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1281"><a href="#cb7-1281" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=14, fig.height=7}</span></span>
<span id="cb7-1282"><a href="#cb7-1282" aria-hidden="true" tabindex="-1"></a><span class="in">#| label: fig-scores-regressed</span></span>
<span id="cb7-1283"><a href="#cb7-1283" aria-hidden="true" tabindex="-1"></a><span class="in">#| fig-cap: Scores obtained from regressed data</span></span>
<span id="cb7-1284"><a href="#cb7-1284" aria-hidden="true" tabindex="-1"></a><span class="in">#|     of all the four cohorts TCGA, SCANB, METABRIC and POETIC compared to the </span></span>
<span id="cb7-1285"><a href="#cb7-1285" aria-hidden="true" tabindex="-1"></a><span class="in">#|     original GSVA scores.</span></span>
<span id="cb7-1286"><a href="#cb7-1286" aria-hidden="true" tabindex="-1"></a><span class="in">scores %&gt;% </span></span>
<span id="cb7-1287"><a href="#cb7-1287" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::filter(pathway %in% names(pathways_of_interest)) %&gt;%</span></span>
<span id="cb7-1288"><a href="#cb7-1288" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::mutate(pathway = factor(pathway, levels = names(pathways_of_interest))) %&gt;%</span></span>
<span id="cb7-1289"><a href="#cb7-1289" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::ggplot(aes(x = score_og, y = score_hkg)) +</span></span>
<span id="cb7-1290"><a href="#cb7-1290" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_point() +</span></span>
<span id="cb7-1291"><a href="#cb7-1291" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::facet_wrap(</span></span>
<span id="cb7-1292"><a href="#cb7-1292" aria-hidden="true" tabindex="-1"></a><span class="in">        cohort ~ pathway, </span></span>
<span id="cb7-1293"><a href="#cb7-1293" aria-hidden="true" tabindex="-1"></a><span class="in">        scales = "free", </span></span>
<span id="cb7-1294"><a href="#cb7-1294" aria-hidden="true" tabindex="-1"></a><span class="in">        ncol = length(pathways_of_interest),</span></span>
<span id="cb7-1295"><a href="#cb7-1295" aria-hidden="true" tabindex="-1"></a><span class="in">        labeller = as_labeller(c(</span></span>
<span id="cb7-1296"><a href="#cb7-1296" aria-hidden="true" tabindex="-1"></a><span class="in">            pathways_of_interest, </span></span>
<span id="cb7-1297"><a href="#cb7-1297" aria-hidden="true" tabindex="-1"></a><span class="in">            "metabric" = "METABRIC",</span></span>
<span id="cb7-1298"><a href="#cb7-1298" aria-hidden="true" tabindex="-1"></a><span class="in">            "scanb" = "SCANB",</span></span>
<span id="cb7-1299"><a href="#cb7-1299" aria-hidden="true" tabindex="-1"></a><span class="in">            "tcga" = "TCGA",</span></span>
<span id="cb7-1300"><a href="#cb7-1300" aria-hidden="true" tabindex="-1"></a><span class="in">            "poetic" = "POETIC"</span></span>
<span id="cb7-1301"><a href="#cb7-1301" aria-hidden="true" tabindex="-1"></a><span class="in">        ))</span></span>
<span id="cb7-1302"><a href="#cb7-1302" aria-hidden="true" tabindex="-1"></a><span class="in">    ) +</span></span>
<span id="cb7-1303"><a href="#cb7-1303" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::labs(</span></span>
<span id="cb7-1304"><a href="#cb7-1304" aria-hidden="true" tabindex="-1"></a><span class="in">        x = "GSVA score",</span></span>
<span id="cb7-1305"><a href="#cb7-1305" aria-hidden="true" tabindex="-1"></a><span class="in">        y = "Regressed score"</span></span>
<span id="cb7-1306"><a href="#cb7-1306" aria-hidden="true" tabindex="-1"></a><span class="in">    ) + </span></span>
<span id="cb7-1307"><a href="#cb7-1307" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme_bw()</span></span>
<span id="cb7-1308"><a href="#cb7-1308" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-1309"><a href="#cb7-1309" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1310"><a href="#cb7-1310" aria-hidden="true" tabindex="-1"></a>The scores are actually highly correlated, specially the estrogen related </span>
<span id="cb7-1311"><a href="#cb7-1311" aria-hidden="true" tabindex="-1"></a>pathways. Also the correlation between the scores in general follows the </span>
<span id="cb7-1312"><a href="#cb7-1312" aria-hidden="true" tabindex="-1"></a>same trends.</span>
<span id="cb7-1313"><a href="#cb7-1313" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1314"><a href="#cb7-1314" aria-hidden="true" tabindex="-1"></a>The plot below is a subset of the scores above to include in the </span>
<span id="cb7-1315"><a href="#cb7-1315" aria-hidden="true" tabindex="-1"></a>publication.</span>
<span id="cb7-1316"><a href="#cb7-1316" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1317"><a href="#cb7-1317" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.height=11, fig.width=8}</span></span>
<span id="cb7-1318"><a href="#cb7-1318" aria-hidden="true" tabindex="-1"></a><span class="in">#| label: fig-scores-regressed-subset</span></span>
<span id="cb7-1319"><a href="#cb7-1319" aria-hidden="true" tabindex="-1"></a><span class="in">#| fig-cap: Scores obtained from regressed data</span></span>
<span id="cb7-1320"><a href="#cb7-1320" aria-hidden="true" tabindex="-1"></a><span class="in">#|     of all the four cohorts TCGA, SCANB, METABRIC and POETIC compared to the </span></span>
<span id="cb7-1321"><a href="#cb7-1321" aria-hidden="true" tabindex="-1"></a><span class="in">#|     original GSVA scores.</span></span>
<span id="cb7-1322"><a href="#cb7-1322" aria-hidden="true" tabindex="-1"></a><span class="in">pathways_of_interest_sub &lt;- c(</span></span>
<span id="cb7-1323"><a href="#cb7-1323" aria-hidden="true" tabindex="-1"></a><span class="in">    "HALLMARK_ESTROGEN_RESPONSE_EARLY" = "Estrogen Early",</span></span>
<span id="cb7-1324"><a href="#cb7-1324" aria-hidden="true" tabindex="-1"></a><span class="in">    "HALLMARK_G2M_CHECKPOINT" = "G2M", </span></span>
<span id="cb7-1325"><a href="#cb7-1325" aria-hidden="true" tabindex="-1"></a><span class="in">    "SET_ERPR" = "SET ER/PR",</span></span>
<span id="cb7-1326"><a href="#cb7-1326" aria-hidden="true" tabindex="-1"></a><span class="in">    "HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION" = "EMT",</span></span>
<span id="cb7-1327"><a href="#cb7-1327" aria-hidden="true" tabindex="-1"></a><span class="in">    "HALLMARK_DNA_REPAIR" = "DNA Repair"</span></span>
<span id="cb7-1328"><a href="#cb7-1328" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-1329"><a href="#cb7-1329" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1330"><a href="#cb7-1330" aria-hidden="true" tabindex="-1"></a><span class="in">correlation_gsva_regressed &lt;- scores %&gt;%</span></span>
<span id="cb7-1331"><a href="#cb7-1331" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::filter(cohort %in% c("tcga", "scanb", "metabric")) %&gt;%</span></span>
<span id="cb7-1332"><a href="#cb7-1332" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::group_by(cohort, pathway) %&gt;%</span></span>
<span id="cb7-1333"><a href="#cb7-1333" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::summarise(</span></span>
<span id="cb7-1334"><a href="#cb7-1334" aria-hidden="true" tabindex="-1"></a><span class="in">        correlation = cor(score_hkg, score_og, method = "spearman")</span></span>
<span id="cb7-1335"><a href="#cb7-1335" aria-hidden="true" tabindex="-1"></a><span class="in">    ) %&gt;%</span></span>
<span id="cb7-1336"><a href="#cb7-1336" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::inner_join(</span></span>
<span id="cb7-1337"><a href="#cb7-1337" aria-hidden="true" tabindex="-1"></a><span class="in">        .,</span></span>
<span id="cb7-1338"><a href="#cb7-1338" aria-hidden="true" tabindex="-1"></a><span class="in">        data.frame(pathway_name = pathways_of_interest_sub) %&gt;%</span></span>
<span id="cb7-1339"><a href="#cb7-1339" aria-hidden="true" tabindex="-1"></a><span class="in">            tibble::rownames_to_column(var = "pathway"),</span></span>
<span id="cb7-1340"><a href="#cb7-1340" aria-hidden="true" tabindex="-1"></a><span class="in">        by = "pathway"</span></span>
<span id="cb7-1341"><a href="#cb7-1341" aria-hidden="true" tabindex="-1"></a><span class="in">    ) %&gt;%</span></span>
<span id="cb7-1342"><a href="#cb7-1342" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::mutate(</span></span>
<span id="cb7-1343"><a href="#cb7-1343" aria-hidden="true" tabindex="-1"></a><span class="in">        corr_vals = paste0(</span></span>
<span id="cb7-1344"><a href="#cb7-1344" aria-hidden="true" tabindex="-1"></a><span class="in">            toupper(cohort), "\n",</span></span>
<span id="cb7-1345"><a href="#cb7-1345" aria-hidden="true" tabindex="-1"></a><span class="in">            pathway_name, ", rho: ", format(correlation, digits = 2)</span></span>
<span id="cb7-1346"><a href="#cb7-1346" aria-hidden="true" tabindex="-1"></a><span class="in">        )</span></span>
<span id="cb7-1347"><a href="#cb7-1347" aria-hidden="true" tabindex="-1"></a><span class="in">    ) %&gt;%</span></span>
<span id="cb7-1348"><a href="#cb7-1348" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::mutate(pathway = factor(pathway, names(pathways_of_interest_sub))) %&gt;%</span></span>
<span id="cb7-1349"><a href="#cb7-1349" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::arrange(cohort, pathway) %&gt;%</span></span>
<span id="cb7-1350"><a href="#cb7-1350" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::mutate(corr_vals = factor(corr_vals, levels = corr_vals)) %&gt;%</span></span>
<span id="cb7-1351"><a href="#cb7-1351" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::inner_join(</span></span>
<span id="cb7-1352"><a href="#cb7-1352" aria-hidden="true" tabindex="-1"></a><span class="in">        .,</span></span>
<span id="cb7-1353"><a href="#cb7-1353" aria-hidden="true" tabindex="-1"></a><span class="in">        scores,</span></span>
<span id="cb7-1354"><a href="#cb7-1354" aria-hidden="true" tabindex="-1"></a><span class="in">        by = c("cohort", "pathway")</span></span>
<span id="cb7-1355"><a href="#cb7-1355" aria-hidden="true" tabindex="-1"></a><span class="in">    )</span></span>
<span id="cb7-1356"><a href="#cb7-1356" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1357"><a href="#cb7-1357" aria-hidden="true" tabindex="-1"></a><span class="in">correlation_only &lt;- correlation_gsva_regressed %&gt;%</span></span>
<span id="cb7-1358"><a href="#cb7-1358" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::filter(cohort %in% c("scanb", "metabric")) %&gt;%</span></span>
<span id="cb7-1359"><a href="#cb7-1359" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::group_by(pathway, cohort) %&gt;%</span></span>
<span id="cb7-1360"><a href="#cb7-1360" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::summarise(</span></span>
<span id="cb7-1361"><a href="#cb7-1361" aria-hidden="true" tabindex="-1"></a><span class="in">        pathway = pathway[1],</span></span>
<span id="cb7-1362"><a href="#cb7-1362" aria-hidden="true" tabindex="-1"></a><span class="in">        correlation = correlation[1],</span></span>
<span id="cb7-1363"><a href="#cb7-1363" aria-hidden="true" tabindex="-1"></a><span class="in">        cohort = cohort[1]</span></span>
<span id="cb7-1364"><a href="#cb7-1364" aria-hidden="true" tabindex="-1"></a><span class="in">    ) %&gt;%</span></span>
<span id="cb7-1365"><a href="#cb7-1365" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::mutate(label = paste0(</span></span>
<span id="cb7-1366"><a href="#cb7-1366" aria-hidden="true" tabindex="-1"></a><span class="in">        "\u03c1", ": ", format(correlation, digits = 2)</span></span>
<span id="cb7-1367"><a href="#cb7-1367" aria-hidden="true" tabindex="-1"></a><span class="in">    ))</span></span>
<span id="cb7-1368"><a href="#cb7-1368" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1369"><a href="#cb7-1369" aria-hidden="true" tabindex="-1"></a><span class="in">correlation_gsva_regressed %&gt;%</span></span>
<span id="cb7-1370"><a href="#cb7-1370" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::filter(cohort %in% c("scanb", "metabric")) %&gt;%</span></span>
<span id="cb7-1371"><a href="#cb7-1371" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::ggplot(aes(x = score_og, y = score_hkg)) +</span></span>
<span id="cb7-1372"><a href="#cb7-1372" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_point() +</span></span>
<span id="cb7-1373"><a href="#cb7-1373" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_smooth(method = "loess", formula = y~x) + </span></span>
<span id="cb7-1374"><a href="#cb7-1374" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::facet_grid(</span></span>
<span id="cb7-1375"><a href="#cb7-1375" aria-hidden="true" tabindex="-1"></a><span class="in">        pathway ~ cohort,</span></span>
<span id="cb7-1376"><a href="#cb7-1376" aria-hidden="true" tabindex="-1"></a><span class="in">        labeller = as_labeller(c(</span></span>
<span id="cb7-1377"><a href="#cb7-1377" aria-hidden="true" tabindex="-1"></a><span class="in">            pathways_of_interest_sub,</span></span>
<span id="cb7-1378"><a href="#cb7-1378" aria-hidden="true" tabindex="-1"></a><span class="in">            c("metabric" = "METABRIC", "scanb" = "SCAN-B")</span></span>
<span id="cb7-1379"><a href="#cb7-1379" aria-hidden="true" tabindex="-1"></a><span class="in">        )),</span></span>
<span id="cb7-1380"><a href="#cb7-1380" aria-hidden="true" tabindex="-1"></a><span class="in">        scales = "free_y"</span></span>
<span id="cb7-1381"><a href="#cb7-1381" aria-hidden="true" tabindex="-1"></a><span class="in">    ) +</span></span>
<span id="cb7-1382"><a href="#cb7-1382" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_label(</span></span>
<span id="cb7-1383"><a href="#cb7-1383" aria-hidden="true" tabindex="-1"></a><span class="in">        data = correlation_only,</span></span>
<span id="cb7-1384"><a href="#cb7-1384" aria-hidden="true" tabindex="-1"></a><span class="in">        mapping = aes(x = -0.5, y = Inf, label = label),</span></span>
<span id="cb7-1385"><a href="#cb7-1385" aria-hidden="true" tabindex="-1"></a><span class="in">        hjust = 0.8,</span></span>
<span id="cb7-1386"><a href="#cb7-1386" aria-hidden="true" tabindex="-1"></a><span class="in">        vjust = 1.5,</span></span>
<span id="cb7-1387"><a href="#cb7-1387" aria-hidden="true" tabindex="-1"></a><span class="in">        size = 5,</span></span>
<span id="cb7-1388"><a href="#cb7-1388" aria-hidden="true" tabindex="-1"></a><span class="in">        label.padding = grid::unit(0.3, "lines")</span></span>
<span id="cb7-1389"><a href="#cb7-1389" aria-hidden="true" tabindex="-1"></a><span class="in">    ) + </span></span>
<span id="cb7-1390"><a href="#cb7-1390" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::labs(</span></span>
<span id="cb7-1391"><a href="#cb7-1391" aria-hidden="true" tabindex="-1"></a><span class="in">        x = "GSVA score",</span></span>
<span id="cb7-1392"><a href="#cb7-1392" aria-hidden="true" tabindex="-1"></a><span class="in">        y = "Regressed score"</span></span>
<span id="cb7-1393"><a href="#cb7-1393" aria-hidden="true" tabindex="-1"></a><span class="in">    ) + </span></span>
<span id="cb7-1394"><a href="#cb7-1394" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme_bw(base_size = 20) +</span></span>
<span id="cb7-1395"><a href="#cb7-1395" aria-hidden="true" tabindex="-1"></a><span class="in">    change_guides_point() +</span></span>
<span id="cb7-1396"><a href="#cb7-1396" aria-hidden="true" tabindex="-1"></a><span class="in">    change_plot_aes_point()</span></span>
<span id="cb7-1397"><a href="#cb7-1397" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-1398"><a href="#cb7-1398" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1399"><a href="#cb7-1399" aria-hidden="true" tabindex="-1"></a>We now compare the distributions</span>
<span id="cb7-1400"><a href="#cb7-1400" aria-hidden="true" tabindex="-1"></a>of these scores across the different cohorts and pathways of interest.</span>
<span id="cb7-1401"><a href="#cb7-1401" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1402"><a href="#cb7-1402" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=16, fig.height=8}</span></span>
<span id="cb7-1403"><a href="#cb7-1403" aria-hidden="true" tabindex="-1"></a><span class="in">scores %&gt;%</span></span>
<span id="cb7-1404"><a href="#cb7-1404" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::filter(pathway %in% names(pathways_of_interest)) %&gt;%</span></span>
<span id="cb7-1405"><a href="#cb7-1405" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::filter(cohort != "poetic") %&gt;%</span></span>
<span id="cb7-1406"><a href="#cb7-1406" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::filter(pam50 %in% c("luma", "lumb", "her2", "basal")) %&gt;%</span></span>
<span id="cb7-1407"><a href="#cb7-1407" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::ggplot(aes(x = score_hkg, fill = cohort)) +</span></span>
<span id="cb7-1408"><a href="#cb7-1408" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_density(alpha = 0.5) +</span></span>
<span id="cb7-1409"><a href="#cb7-1409" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::facet_wrap(</span></span>
<span id="cb7-1410"><a href="#cb7-1410" aria-hidden="true" tabindex="-1"></a><span class="in">        pam50~pathway, </span></span>
<span id="cb7-1411"><a href="#cb7-1411" aria-hidden="true" tabindex="-1"></a><span class="in">        scales = "free", </span></span>
<span id="cb7-1412"><a href="#cb7-1412" aria-hidden="true" tabindex="-1"></a><span class="in">        ncol = length(pathways_of_interest),</span></span>
<span id="cb7-1413"><a href="#cb7-1413" aria-hidden="true" tabindex="-1"></a><span class="in">        labeller = as_labeller(c(</span></span>
<span id="cb7-1414"><a href="#cb7-1414" aria-hidden="true" tabindex="-1"></a><span class="in">            pathways_of_interest, </span></span>
<span id="cb7-1415"><a href="#cb7-1415" aria-hidden="true" tabindex="-1"></a><span class="in">            "luma" = "Luminal A",</span></span>
<span id="cb7-1416"><a href="#cb7-1416" aria-hidden="true" tabindex="-1"></a><span class="in">            "lumb" = "Luminal B",</span></span>
<span id="cb7-1417"><a href="#cb7-1417" aria-hidden="true" tabindex="-1"></a><span class="in">            "her2" = "HER2",</span></span>
<span id="cb7-1418"><a href="#cb7-1418" aria-hidden="true" tabindex="-1"></a><span class="in">            "basal" = "Basal"</span></span>
<span id="cb7-1419"><a href="#cb7-1419" aria-hidden="true" tabindex="-1"></a><span class="in">        ))</span></span>
<span id="cb7-1420"><a href="#cb7-1420" aria-hidden="true" tabindex="-1"></a><span class="in">    ) + </span></span>
<span id="cb7-1421"><a href="#cb7-1421" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme_bw()</span></span>
<span id="cb7-1422"><a href="#cb7-1422" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-1423"><a href="#cb7-1423" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1424"><a href="#cb7-1424" aria-hidden="true" tabindex="-1"></a>And with all molecular subtypes together.</span>
<span id="cb7-1425"><a href="#cb7-1425" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1426"><a href="#cb7-1426" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=6}</span></span>
<span id="cb7-1427"><a href="#cb7-1427" aria-hidden="true" tabindex="-1"></a><span class="in">scores %&gt;%</span></span>
<span id="cb7-1428"><a href="#cb7-1428" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::filter(pathway %in% names(pathways_of_interest)) %&gt;%</span></span>
<span id="cb7-1429"><a href="#cb7-1429" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::ggplot(aes(x = score_hkg, fill = cohort)) +</span></span>
<span id="cb7-1430"><a href="#cb7-1430" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_density(alpha = 0.5) +</span></span>
<span id="cb7-1431"><a href="#cb7-1431" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::facet_wrap(</span></span>
<span id="cb7-1432"><a href="#cb7-1432" aria-hidden="true" tabindex="-1"></a><span class="in">        ~pathway, </span></span>
<span id="cb7-1433"><a href="#cb7-1433" aria-hidden="true" tabindex="-1"></a><span class="in">        scales = "free", </span></span>
<span id="cb7-1434"><a href="#cb7-1434" aria-hidden="true" tabindex="-1"></a><span class="in">        labeller = as_labeller(pathways_of_interest)</span></span>
<span id="cb7-1435"><a href="#cb7-1435" aria-hidden="true" tabindex="-1"></a><span class="in">    ) + </span></span>
<span id="cb7-1436"><a href="#cb7-1436" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme_bw()</span></span>
<span id="cb7-1437"><a href="#cb7-1437" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-1438"><a href="#cb7-1438" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1439"><a href="#cb7-1439" aria-hidden="true" tabindex="-1"></a>We see that the overlap is actually pretty good in general, even for </span>
<span id="cb7-1440"><a href="#cb7-1440" aria-hidden="true" tabindex="-1"></a>SCANB. The only part where there is a bigger discrepancy is for</span>
<span id="cb7-1441"><a href="#cb7-1441" aria-hidden="true" tabindex="-1"></a>G2M checkpoint. We see below that by calculating the GSVA score </span>
<span id="cb7-1442"><a href="#cb7-1442" aria-hidden="true" tabindex="-1"></a>on the whole scanb cohort we don't have this problem.</span>
<span id="cb7-1443"><a href="#cb7-1443" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1444"><a href="#cb7-1444" aria-hidden="true" tabindex="-1"></a>And as a comparison we plot with the original scores using GSVA on the </span>
<span id="cb7-1445"><a href="#cb7-1445" aria-hidden="true" tabindex="-1"></a>whole cohort individually.</span>
<span id="cb7-1446"><a href="#cb7-1446" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1447"><a href="#cb7-1447" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=16, fig.height=8}</span></span>
<span id="cb7-1448"><a href="#cb7-1448" aria-hidden="true" tabindex="-1"></a><span class="in">scores %&gt;%</span></span>
<span id="cb7-1449"><a href="#cb7-1449" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::filter(pathway %in% names(pathways_of_interest)) %&gt;%</span></span>
<span id="cb7-1450"><a href="#cb7-1450" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::filter(pam50 %in% c("luma", "lumb", "her2", "basal")) %&gt;%</span></span>
<span id="cb7-1451"><a href="#cb7-1451" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::ggplot(aes(x = score_og, fill = cohort)) +</span></span>
<span id="cb7-1452"><a href="#cb7-1452" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_density(alpha = 0.5) +</span></span>
<span id="cb7-1453"><a href="#cb7-1453" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::facet_wrap(</span></span>
<span id="cb7-1454"><a href="#cb7-1454" aria-hidden="true" tabindex="-1"></a><span class="in">        pam50~pathway, </span></span>
<span id="cb7-1455"><a href="#cb7-1455" aria-hidden="true" tabindex="-1"></a><span class="in">        scales = "free", </span></span>
<span id="cb7-1456"><a href="#cb7-1456" aria-hidden="true" tabindex="-1"></a><span class="in">        ncol = length(pathways_of_interest),</span></span>
<span id="cb7-1457"><a href="#cb7-1457" aria-hidden="true" tabindex="-1"></a><span class="in">        labeller = as_labeller(c(</span></span>
<span id="cb7-1458"><a href="#cb7-1458" aria-hidden="true" tabindex="-1"></a><span class="in">            pathways_of_interest, </span></span>
<span id="cb7-1459"><a href="#cb7-1459" aria-hidden="true" tabindex="-1"></a><span class="in">            "luma" = "Luminal A",</span></span>
<span id="cb7-1460"><a href="#cb7-1460" aria-hidden="true" tabindex="-1"></a><span class="in">            "lumb" = "Luminal B",</span></span>
<span id="cb7-1461"><a href="#cb7-1461" aria-hidden="true" tabindex="-1"></a><span class="in">            "her2" = "HER2",</span></span>
<span id="cb7-1462"><a href="#cb7-1462" aria-hidden="true" tabindex="-1"></a><span class="in">            "basal" = "Basal"</span></span>
<span id="cb7-1463"><a href="#cb7-1463" aria-hidden="true" tabindex="-1"></a><span class="in">        ))</span></span>
<span id="cb7-1464"><a href="#cb7-1464" aria-hidden="true" tabindex="-1"></a><span class="in">    ) + </span></span>
<span id="cb7-1465"><a href="#cb7-1465" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme_bw()</span></span>
<span id="cb7-1466"><a href="#cb7-1466" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-1467"><a href="#cb7-1467" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1468"><a href="#cb7-1468" aria-hidden="true" tabindex="-1"></a>And all the subtypes together for the original score:</span>
<span id="cb7-1469"><a href="#cb7-1469" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1470"><a href="#cb7-1470" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=6}</span></span>
<span id="cb7-1471"><a href="#cb7-1471" aria-hidden="true" tabindex="-1"></a><span class="in">scores %&gt;%</span></span>
<span id="cb7-1472"><a href="#cb7-1472" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::filter(pathway %in% names(pathways_of_interest)) %&gt;%</span></span>
<span id="cb7-1473"><a href="#cb7-1473" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::ggplot(aes(x = score_og, fill = cohort)) +</span></span>
<span id="cb7-1474"><a href="#cb7-1474" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_density(alpha = 0.5) +</span></span>
<span id="cb7-1475"><a href="#cb7-1475" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::facet_wrap(</span></span>
<span id="cb7-1476"><a href="#cb7-1476" aria-hidden="true" tabindex="-1"></a><span class="in">        ~pathway, </span></span>
<span id="cb7-1477"><a href="#cb7-1477" aria-hidden="true" tabindex="-1"></a><span class="in">        scales = "free", </span></span>
<span id="cb7-1478"><a href="#cb7-1478" aria-hidden="true" tabindex="-1"></a><span class="in">        labeller = as_labeller(pathways_of_interest)</span></span>
<span id="cb7-1479"><a href="#cb7-1479" aria-hidden="true" tabindex="-1"></a><span class="in">    ) + </span></span>
<span id="cb7-1480"><a href="#cb7-1480" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme_bw()</span></span>
<span id="cb7-1481"><a href="#cb7-1481" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-1482"><a href="#cb7-1482" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1483"><a href="#cb7-1483" aria-hidden="true" tabindex="-1"></a>We see that by doing GSVA the distributions they overlap much more than </span>
<span id="cb7-1484"><a href="#cb7-1484" aria-hidden="true" tabindex="-1"></a>using the regressed data and the simple sum of the values for those genes.</span>
<span id="cb7-1485"><a href="#cb7-1485" aria-hidden="true" tabindex="-1"></a>Still for estrogen signaling it seems that the estrogen early signature</span>
<span id="cb7-1486"><a href="#cb7-1486" aria-hidden="true" tabindex="-1"></a>is pretty robust.</span>
<span id="cb7-1487"><a href="#cb7-1487" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1488"><a href="#cb7-1488" aria-hidden="true" tabindex="-1"></a><span class="fu">### Checking the patients from POETIC with new scores</span></span>
<span id="cb7-1489"><a href="#cb7-1489" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1490"><a href="#cb7-1490" aria-hidden="true" tabindex="-1"></a>We now use the new scores to evaluate the comparison with the neighboorhood,</span>
<span id="cb7-1491"><a href="#cb7-1491" aria-hidden="true" tabindex="-1"></a>just as before in the previous chapter where we used the POETIC trial data.</span>
<span id="cb7-1492"><a href="#cb7-1492" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1493"><a href="#cb7-1493" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb7-1494"><a href="#cb7-1494" aria-hidden="true" tabindex="-1"></a><span class="in"># get scores in the wide format so we can join with the original df_pca</span></span>
<span id="cb7-1495"><a href="#cb7-1495" aria-hidden="true" tabindex="-1"></a><span class="in"># dataframe and use the same code as in the previous scoring chapter</span></span>
<span id="cb7-1496"><a href="#cb7-1496" aria-hidden="true" tabindex="-1"></a><span class="in">scores_to_use &lt;- c(</span></span>
<span id="cb7-1497"><a href="#cb7-1497" aria-hidden="true" tabindex="-1"></a><span class="in">    "Estrogen response early",</span></span>
<span id="cb7-1498"><a href="#cb7-1498" aria-hidden="true" tabindex="-1"></a><span class="in">    "E2F targets",</span></span>
<span id="cb7-1499"><a href="#cb7-1499" aria-hidden="true" tabindex="-1"></a><span class="in">    "G2M",</span></span>
<span id="cb7-1500"><a href="#cb7-1500" aria-hidden="true" tabindex="-1"></a><span class="in">    "P53 pathway",</span></span>
<span id="cb7-1501"><a href="#cb7-1501" aria-hidden="true" tabindex="-1"></a><span class="in">    "EMT",</span></span>
<span id="cb7-1502"><a href="#cb7-1502" aria-hidden="true" tabindex="-1"></a><span class="in">    "Androgen response",</span></span>
<span id="cb7-1503"><a href="#cb7-1503" aria-hidden="true" tabindex="-1"></a><span class="in">    "TGFb signaling",</span></span>
<span id="cb7-1504"><a href="#cb7-1504" aria-hidden="true" tabindex="-1"></a><span class="in">    "SET ER/PR"</span></span>
<span id="cb7-1505"><a href="#cb7-1505" aria-hidden="true" tabindex="-1"></a><span class="in">) %&gt;% `names&lt;-`(paste0(c(</span></span>
<span id="cb7-1506"><a href="#cb7-1506" aria-hidden="true" tabindex="-1"></a><span class="in">    "HALLMARK_ESTROGEN_RESPONSE_EARLY",</span></span>
<span id="cb7-1507"><a href="#cb7-1507" aria-hidden="true" tabindex="-1"></a><span class="in">    "HALLMARK_E2F_TARGETS",</span></span>
<span id="cb7-1508"><a href="#cb7-1508" aria-hidden="true" tabindex="-1"></a><span class="in">    "HALLMARK_G2M_CHECKPOINT",</span></span>
<span id="cb7-1509"><a href="#cb7-1509" aria-hidden="true" tabindex="-1"></a><span class="in">    "HALLMARK_P53_PATHWAY",</span></span>
<span id="cb7-1510"><a href="#cb7-1510" aria-hidden="true" tabindex="-1"></a><span class="in">    "HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION",</span></span>
<span id="cb7-1511"><a href="#cb7-1511" aria-hidden="true" tabindex="-1"></a><span class="in">    "HALLMARK_ANDROGEN_RESPONSE",</span></span>
<span id="cb7-1512"><a href="#cb7-1512" aria-hidden="true" tabindex="-1"></a><span class="in">    "HALLMARK_TGF_BETA_SIGNALING",</span></span>
<span id="cb7-1513"><a href="#cb7-1513" aria-hidden="true" tabindex="-1"></a><span class="in">    "SET_ERPR"</span></span>
<span id="cb7-1514"><a href="#cb7-1514" aria-hidden="true" tabindex="-1"></a><span class="in">)))</span></span>
<span id="cb7-1515"><a href="#cb7-1515" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1516"><a href="#cb7-1516" aria-hidden="true" tabindex="-1"></a><span class="in">scores_wide &lt;- sapply(</span></span>
<span id="cb7-1517"><a href="#cb7-1517" aria-hidden="true" tabindex="-1"></a><span class="in">    gene_sets_,</span></span>
<span id="cb7-1518"><a href="#cb7-1518" aria-hidden="true" tabindex="-1"></a><span class="in">    function(gene_set, df_pcs_regressed, avg_hkg_pcs_regressed){</span></span>
<span id="cb7-1519"><a href="#cb7-1519" aria-hidden="true" tabindex="-1"></a><span class="in">        </span></span>
<span id="cb7-1520"><a href="#cb7-1520" aria-hidden="true" tabindex="-1"></a><span class="in">        proportion &lt;- 1</span></span>
<span id="cb7-1521"><a href="#cb7-1521" aria-hidden="true" tabindex="-1"></a><span class="in">        nb_genes &lt;- floor(proportion * length(gene_set))</span></span>
<span id="cb7-1522"><a href="#cb7-1522" aria-hidden="true" tabindex="-1"></a><span class="in">        gene_set &lt;- intersect(</span></span>
<span id="cb7-1523"><a href="#cb7-1523" aria-hidden="true" tabindex="-1"></a><span class="in">            sample(gene_set, nb_genes),</span></span>
<span id="cb7-1524"><a href="#cb7-1524" aria-hidden="true" tabindex="-1"></a><span class="in">            rownames(df_pcs_regressed)</span></span>
<span id="cb7-1525"><a href="#cb7-1525" aria-hidden="true" tabindex="-1"></a><span class="in">        )</span></span>
<span id="cb7-1526"><a href="#cb7-1526" aria-hidden="true" tabindex="-1"></a><span class="in">        colSums(df_pcs_regressed[gene_set, ])</span></span>
<span id="cb7-1527"><a href="#cb7-1527" aria-hidden="true" tabindex="-1"></a><span class="in">            </span></span>
<span id="cb7-1528"><a href="#cb7-1528" aria-hidden="true" tabindex="-1"></a><span class="in">    },</span></span>
<span id="cb7-1529"><a href="#cb7-1529" aria-hidden="true" tabindex="-1"></a><span class="in">    df_pcs_regressed = df_pcs_regressed,</span></span>
<span id="cb7-1530"><a href="#cb7-1530" aria-hidden="true" tabindex="-1"></a><span class="in">    avg_hkg_pcs_regressed = avg_hkg_pcs_regressed</span></span>
<span id="cb7-1531"><a href="#cb7-1531" aria-hidden="true" tabindex="-1"></a><span class="in">) %&gt;% </span></span>
<span id="cb7-1532"><a href="#cb7-1532" aria-hidden="true" tabindex="-1"></a><span class="in">    data.frame %&gt;%</span></span>
<span id="cb7-1533"><a href="#cb7-1533" aria-hidden="true" tabindex="-1"></a><span class="in">    tibble::rownames_to_column(var = "sample_name")</span></span>
<span id="cb7-1534"><a href="#cb7-1534" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1535"><a href="#cb7-1535" aria-hidden="true" tabindex="-1"></a><span class="in">df_pca_scores &lt;- dplyr::inner_join(</span></span>
<span id="cb7-1536"><a href="#cb7-1536" aria-hidden="true" tabindex="-1"></a><span class="in">    df_pca, </span></span>
<span id="cb7-1537"><a href="#cb7-1537" aria-hidden="true" tabindex="-1"></a><span class="in">    scores_wide, </span></span>
<span id="cb7-1538"><a href="#cb7-1538" aria-hidden="true" tabindex="-1"></a><span class="in">    by = "sample_name", suffix = c("_og", "_new")</span></span>
<span id="cb7-1539"><a href="#cb7-1539" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-1540"><a href="#cb7-1540" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1541"><a href="#cb7-1541" aria-hidden="true" tabindex="-1"></a><span class="in"># arguments for getting the neighborhood score</span></span>
<span id="cb7-1542"><a href="#cb7-1542" aria-hidden="true" tabindex="-1"></a><span class="in">patients &lt;- c("treated_nb63_baseline", "treated_nb236_baseline")</span></span>
<span id="cb7-1543"><a href="#cb7-1543" aria-hidden="true" tabindex="-1"></a><span class="in">names(scores_to_use) &lt;- paste0(names(scores_to_use), "_new")</span></span>
<span id="cb7-1544"><a href="#cb7-1544" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1545"><a href="#cb7-1545" aria-hidden="true" tabindex="-1"></a><span class="in">pipeline_scores_plots_new_scores &lt;- sapply(</span></span>
<span id="cb7-1546"><a href="#cb7-1546" aria-hidden="true" tabindex="-1"></a><span class="in">    patients, </span></span>
<span id="cb7-1547"><a href="#cb7-1547" aria-hidden="true" tabindex="-1"></a><span class="in">    get_patient_scores_distributions_all,</span></span>
<span id="cb7-1548"><a href="#cb7-1548" aria-hidden="true" tabindex="-1"></a><span class="in">    df_pca = df_pca_scores,</span></span>
<span id="cb7-1549"><a href="#cb7-1549" aria-hidden="true" tabindex="-1"></a><span class="in">    scores_to_use = scores_to_use,</span></span>
<span id="cb7-1550"><a href="#cb7-1550" aria-hidden="true" tabindex="-1"></a><span class="in">    base_size = 20,</span></span>
<span id="cb7-1551"><a href="#cb7-1551" aria-hidden="true" tabindex="-1"></a><span class="in">    USE.NAMES = TRUE, </span></span>
<span id="cb7-1552"><a href="#cb7-1552" aria-hidden="true" tabindex="-1"></a><span class="in">    simplify = FALSE</span></span>
<span id="cb7-1553"><a href="#cb7-1553" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-1554"><a href="#cb7-1554" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1555"><a href="#cb7-1555" aria-hidden="true" tabindex="-1"></a><span class="in">saveRDS(</span></span>
<span id="cb7-1556"><a href="#cb7-1556" aria-hidden="true" tabindex="-1"></a><span class="in">    pipeline_scores_plots_new_scores,</span></span>
<span id="cb7-1557"><a href="#cb7-1557" aria-hidden="true" tabindex="-1"></a><span class="in">    "../../results/rds_files/expanding_ember/pipeline_scores_plots_new_scores.rds"</span></span>
<span id="cb7-1558"><a href="#cb7-1558" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-1559"><a href="#cb7-1559" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1560"><a href="#cb7-1560" aria-hidden="true" tabindex="-1"></a><span class="in">saveRDS(</span></span>
<span id="cb7-1561"><a href="#cb7-1561" aria-hidden="true" tabindex="-1"></a><span class="in">    df_pca_scores,</span></span>
<span id="cb7-1562"><a href="#cb7-1562" aria-hidden="true" tabindex="-1"></a><span class="in">    "../../results/rds_files/expanding_ember/df_pca_scores.rds"</span></span>
<span id="cb7-1563"><a href="#cb7-1563" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-1564"><a href="#cb7-1564" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-1565"><a href="#cb7-1565" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1566"><a href="#cb7-1566" aria-hidden="true" tabindex="-1"></a>Patient 63 @fig-pt63 is considered a responder. According to </span>
<span id="cb7-1567"><a href="#cb7-1567" aria-hidden="true" tabindex="-1"></a>the scores, when comparing the estrogen early signature to its average </span>
<span id="cb7-1568"><a href="#cb7-1568" aria-hidden="true" tabindex="-1"></a>distribution in the neighborhood, the score is higher.</span>
<span id="cb7-1569"><a href="#cb7-1569" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1570"><a href="#cb7-1570" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.height=8, fig.width=10}</span></span>
<span id="cb7-1571"><a href="#cb7-1571" aria-hidden="true" tabindex="-1"></a><span class="in">#| label: fig-pt63</span></span>
<span id="cb7-1572"><a href="#cb7-1572" aria-hidden="true" tabindex="-1"></a><span class="in">#| fig-cap: Posterior distribution of the average scores in the neighborhood</span></span>
<span id="cb7-1573"><a href="#cb7-1573" aria-hidden="true" tabindex="-1"></a><span class="in">#|     of patient 63 from POETIC trial. Each dot corresponds to a 1% quantile.</span></span>
<span id="cb7-1574"><a href="#cb7-1574" aria-hidden="true" tabindex="-1"></a><span class="in">pipeline_scores_plots_new_scores$treated_nb63_baseline$plot +</span></span>
<span id="cb7-1575"><a href="#cb7-1575" aria-hidden="true" tabindex="-1"></a><span class="in">    change_plot_aes_point()</span></span>
<span id="cb7-1576"><a href="#cb7-1576" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-1577"><a href="#cb7-1577" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1578"><a href="#cb7-1578" aria-hidden="true" tabindex="-1"></a>Patient 236 @fig-pt236 was considered a non responder. According to </span>
<span id="cb7-1579"><a href="#cb7-1579" aria-hidden="true" tabindex="-1"></a>the scores, when comparing the estrogen early signature to its average </span>
<span id="cb7-1580"><a href="#cb7-1580" aria-hidden="true" tabindex="-1"></a>distribution in the neighborhood, the score is lower, being in the</span>
<span id="cb7-1581"><a href="#cb7-1581" aria-hidden="true" tabindex="-1"></a>5% quantile.</span>
<span id="cb7-1582"><a href="#cb7-1582" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1583"><a href="#cb7-1583" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.height=8, fig.width=10}</span></span>
<span id="cb7-1584"><a href="#cb7-1584" aria-hidden="true" tabindex="-1"></a><span class="in">#| label: fig-pt236</span></span>
<span id="cb7-1585"><a href="#cb7-1585" aria-hidden="true" tabindex="-1"></a><span class="in">#| fig-cap: Posterior distribution of the average scores in the neighborhood</span></span>
<span id="cb7-1586"><a href="#cb7-1586" aria-hidden="true" tabindex="-1"></a><span class="in">#|     of patient 236 from POETIC trial. Each dot corresponds to a 1% quantile.</span></span>
<span id="cb7-1587"><a href="#cb7-1587" aria-hidden="true" tabindex="-1"></a><span class="in">pipeline_scores_plots_new_scores$treated_nb236_baseline$plot +</span></span>
<span id="cb7-1588"><a href="#cb7-1588" aria-hidden="true" tabindex="-1"></a><span class="in">    change_plot_aes_point()</span></span>
<span id="cb7-1589"><a href="#cb7-1589" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-1590"><a href="#cb7-1590" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1591"><a href="#cb7-1591" aria-hidden="true" tabindex="-1"></a>When comparing these two patients, there is also a difference in the androgen</span>
<span id="cb7-1592"><a href="#cb7-1592" aria-hidden="true" tabindex="-1"></a>response score, which could be a reflection of different estrogen signaling.</span>
<span id="cb7-1593"><a href="#cb7-1593" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1594"><a href="#cb7-1594" aria-hidden="true" tabindex="-1"></a>It reflects what we saw previously as well. </span>
<span id="cb7-1595"><a href="#cb7-1595" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1596"><a href="#cb7-1596" aria-hidden="true" tabindex="-1"></a><span class="fu">### Responder vs non responder comparison of new scores</span></span>
<span id="cb7-1597"><a href="#cb7-1597" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1598"><a href="#cb7-1598" aria-hidden="true" tabindex="-1"></a>We now compare the scores for responders and non responders.</span>
<span id="cb7-1599"><a href="#cb7-1599" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1600"><a href="#cb7-1600" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=8, fig.height=5}</span></span>
<span id="cb7-1601"><a href="#cb7-1601" aria-hidden="true" tabindex="-1"></a><span class="in">df_pca_scores %&gt;%</span></span>
<span id="cb7-1602"><a href="#cb7-1602" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::filter(</span></span>
<span id="cb7-1603"><a href="#cb7-1603" aria-hidden="true" tabindex="-1"></a><span class="in">        cohort == "poetic" &amp; </span></span>
<span id="cb7-1604"><a href="#cb7-1604" aria-hidden="true" tabindex="-1"></a><span class="in">            timepoint == "baseline" &amp; </span></span>
<span id="cb7-1605"><a href="#cb7-1605" aria-hidden="true" tabindex="-1"></a><span class="in">            is_responder != "not_available"</span></span>
<span id="cb7-1606"><a href="#cb7-1606" aria-hidden="true" tabindex="-1"></a><span class="in">    ) %&gt;%</span></span>
<span id="cb7-1607"><a href="#cb7-1607" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::ggplot(aes(</span></span>
<span id="cb7-1608"><a href="#cb7-1608" aria-hidden="true" tabindex="-1"></a><span class="in">        color = is_responder, </span></span>
<span id="cb7-1609"><a href="#cb7-1609" aria-hidden="true" tabindex="-1"></a><span class="in">        y = HALLMARK_ESTROGEN_RESPONSE_EARLY_new,</span></span>
<span id="cb7-1610"><a href="#cb7-1610" aria-hidden="true" tabindex="-1"></a><span class="in">        x = HALLMARK_E2F_TARGETS_new</span></span>
<span id="cb7-1611"><a href="#cb7-1611" aria-hidden="true" tabindex="-1"></a><span class="in">    )) +</span></span>
<span id="cb7-1612"><a href="#cb7-1612" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_point() +</span></span>
<span id="cb7-1613"><a href="#cb7-1613" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_smooth(method = "lm", formula = y~x) +</span></span>
<span id="cb7-1614"><a href="#cb7-1614" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::labs(</span></span>
<span id="cb7-1615"><a href="#cb7-1615" aria-hidden="true" tabindex="-1"></a><span class="in">        y = "Estrogen Early",</span></span>
<span id="cb7-1616"><a href="#cb7-1616" aria-hidden="true" tabindex="-1"></a><span class="in">        x = "E2F targets",</span></span>
<span id="cb7-1617"><a href="#cb7-1617" aria-hidden="true" tabindex="-1"></a><span class="in">        title = "Scores calculated after regressing the data"</span></span>
<span id="cb7-1618"><a href="#cb7-1618" aria-hidden="true" tabindex="-1"></a><span class="in">    ) +</span></span>
<span id="cb7-1619"><a href="#cb7-1619" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme_bw(base_size = 20) +</span></span>
<span id="cb7-1620"><a href="#cb7-1620" aria-hidden="true" tabindex="-1"></a><span class="in">    change_plot_aes_point()</span></span>
<span id="cb7-1621"><a href="#cb7-1621" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-1622"><a href="#cb7-1622" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1623"><a href="#cb7-1623" aria-hidden="true" tabindex="-1"></a>The higher E2F targets in average the lower the estrogen early score is</span>
<span id="cb7-1624"><a href="#cb7-1624" aria-hidden="true" tabindex="-1"></a>for non responders, whereas this correlation does not exist for responder</span>
<span id="cb7-1625"><a href="#cb7-1625" aria-hidden="true" tabindex="-1"></a>patients. This difference is mostly drive by the high E2F target non responder</span>
<span id="cb7-1626"><a href="#cb7-1626" aria-hidden="true" tabindex="-1"></a>tumors. </span>
<span id="cb7-1627"><a href="#cb7-1627" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1628"><a href="#cb7-1628" aria-hidden="true" tabindex="-1"></a>Now the plot below shows the correlation for baselien Ki67% and E2F targets</span>
<span id="cb7-1629"><a href="#cb7-1629" aria-hidden="true" tabindex="-1"></a>for the score calculated in the regressed data in a single sample manner. </span>
<span id="cb7-1630"><a href="#cb7-1630" aria-hidden="true" tabindex="-1"></a>We see a positive correlation between the two scores. Moreover, it is interesting</span>
<span id="cb7-1631"><a href="#cb7-1631" aria-hidden="true" tabindex="-1"></a>to see that when the tumor has low Ki67, there is still a spectrum on </span>
<span id="cb7-1632"><a href="#cb7-1632" aria-hidden="true" tabindex="-1"></a>the E2F targets score. Also there is no difference in the correlation </span>
<span id="cb7-1633"><a href="#cb7-1633" aria-hidden="true" tabindex="-1"></a>between responders and non responders.</span>
<span id="cb7-1634"><a href="#cb7-1634" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1635"><a href="#cb7-1635" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=8, fig.height=5}</span></span>
<span id="cb7-1636"><a href="#cb7-1636" aria-hidden="true" tabindex="-1"></a><span class="in">df_pca_scores %&gt;%</span></span>
<span id="cb7-1637"><a href="#cb7-1637" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::filter(</span></span>
<span id="cb7-1638"><a href="#cb7-1638" aria-hidden="true" tabindex="-1"></a><span class="in">        cohort == "poetic" &amp; </span></span>
<span id="cb7-1639"><a href="#cb7-1639" aria-hidden="true" tabindex="-1"></a><span class="in">            timepoint == "baseline" &amp; </span></span>
<span id="cb7-1640"><a href="#cb7-1640" aria-hidden="true" tabindex="-1"></a><span class="in">            is_responder != "not_available"</span></span>
<span id="cb7-1641"><a href="#cb7-1641" aria-hidden="true" tabindex="-1"></a><span class="in">    ) %&gt;%</span></span>
<span id="cb7-1642"><a href="#cb7-1642" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::ggplot(aes(</span></span>
<span id="cb7-1643"><a href="#cb7-1643" aria-hidden="true" tabindex="-1"></a><span class="in">        color = is_responder, </span></span>
<span id="cb7-1644"><a href="#cb7-1644" aria-hidden="true" tabindex="-1"></a><span class="in">        y = ki67,</span></span>
<span id="cb7-1645"><a href="#cb7-1645" aria-hidden="true" tabindex="-1"></a><span class="in">        x = HALLMARK_E2F_TARGETS_new</span></span>
<span id="cb7-1646"><a href="#cb7-1646" aria-hidden="true" tabindex="-1"></a><span class="in">    )) +</span></span>
<span id="cb7-1647"><a href="#cb7-1647" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_point() +</span></span>
<span id="cb7-1648"><a href="#cb7-1648" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_smooth(method = "gam", formula = y ~ s(x, bs = "cs")) +</span></span>
<span id="cb7-1649"><a href="#cb7-1649" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::labs(</span></span>
<span id="cb7-1650"><a href="#cb7-1650" aria-hidden="true" tabindex="-1"></a><span class="in">        y = "Ki67%",</span></span>
<span id="cb7-1651"><a href="#cb7-1651" aria-hidden="true" tabindex="-1"></a><span class="in">        x = "E2F targets",</span></span>
<span id="cb7-1652"><a href="#cb7-1652" aria-hidden="true" tabindex="-1"></a><span class="in">        title = "Scores calculated after regressing the data"</span></span>
<span id="cb7-1653"><a href="#cb7-1653" aria-hidden="true" tabindex="-1"></a><span class="in">    ) +</span></span>
<span id="cb7-1654"><a href="#cb7-1654" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme_bw(base_size = 20) +</span></span>
<span id="cb7-1655"><a href="#cb7-1655" aria-hidden="true" tabindex="-1"></a><span class="in">    change_plot_aes_point()</span></span>
<span id="cb7-1656"><a href="#cb7-1656" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-1657"><a href="#cb7-1657" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1658"><a href="#cb7-1658" aria-hidden="true" tabindex="-1"></a>And lastly we compare the estrogen signaling signatures between</span>
<span id="cb7-1659"><a href="#cb7-1659" aria-hidden="true" tabindex="-1"></a>responders and non-responder.</span>
<span id="cb7-1660"><a href="#cb7-1660" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1661"><a href="#cb7-1661" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=6}</span></span>
<span id="cb7-1662"><a href="#cb7-1662" aria-hidden="true" tabindex="-1"></a><span class="in">plot_responders_non_responder_comparison &lt;- function(</span></span>
<span id="cb7-1663"><a href="#cb7-1663" aria-hidden="true" tabindex="-1"></a><span class="in">    df_pca_scores, </span></span>
<span id="cb7-1664"><a href="#cb7-1664" aria-hidden="true" tabindex="-1"></a><span class="in">    og_or_new = "new",</span></span>
<span id="cb7-1665"><a href="#cb7-1665" aria-hidden="true" tabindex="-1"></a><span class="in">    pathways = c(</span></span>
<span id="cb7-1666"><a href="#cb7-1666" aria-hidden="true" tabindex="-1"></a><span class="in">        "SET_ERPR" = "SET ER/PR", </span></span>
<span id="cb7-1667"><a href="#cb7-1667" aria-hidden="true" tabindex="-1"></a><span class="in">        "HALLMARK_ESTROGEN_RESPONSE_EARLY" = "Estrogen early"</span></span>
<span id="cb7-1668"><a href="#cb7-1668" aria-hidden="true" tabindex="-1"></a><span class="in">    )</span></span>
<span id="cb7-1669"><a href="#cb7-1669" aria-hidden="true" tabindex="-1"></a><span class="in">){</span></span>
<span id="cb7-1670"><a href="#cb7-1670" aria-hidden="true" tabindex="-1"></a><span class="in">    </span></span>
<span id="cb7-1671"><a href="#cb7-1671" aria-hidden="true" tabindex="-1"></a><span class="in">    pathways_labeller &lt;- pathways</span></span>
<span id="cb7-1672"><a href="#cb7-1672" aria-hidden="true" tabindex="-1"></a><span class="in">    names(pathways_labeller) &lt;- paste0(</span></span>
<span id="cb7-1673"><a href="#cb7-1673" aria-hidden="true" tabindex="-1"></a><span class="in">        names(pathways), "_", og_or_new</span></span>
<span id="cb7-1674"><a href="#cb7-1674" aria-hidden="true" tabindex="-1"></a><span class="in">    )</span></span>
<span id="cb7-1675"><a href="#cb7-1675" aria-hidden="true" tabindex="-1"></a><span class="in">    </span></span>
<span id="cb7-1676"><a href="#cb7-1676" aria-hidden="true" tabindex="-1"></a><span class="in">    df_pca_scores %&gt;%</span></span>
<span id="cb7-1677"><a href="#cb7-1677" aria-hidden="true" tabindex="-1"></a><span class="in">        dplyr::filter(</span></span>
<span id="cb7-1678"><a href="#cb7-1678" aria-hidden="true" tabindex="-1"></a><span class="in">            cohort == "poetic" &amp; </span></span>
<span id="cb7-1679"><a href="#cb7-1679" aria-hidden="true" tabindex="-1"></a><span class="in">                timepoint == "baseline" &amp; </span></span>
<span id="cb7-1680"><a href="#cb7-1680" aria-hidden="true" tabindex="-1"></a><span class="in">                is_responder != "not_available"</span></span>
<span id="cb7-1681"><a href="#cb7-1681" aria-hidden="true" tabindex="-1"></a><span class="in">        ) %&gt;%</span></span>
<span id="cb7-1682"><a href="#cb7-1682" aria-hidden="true" tabindex="-1"></a><span class="in">        tidyr::pivot_longer(</span></span>
<span id="cb7-1683"><a href="#cb7-1683" aria-hidden="true" tabindex="-1"></a><span class="in">            cols = dplyr::all_of(</span></span>
<span id="cb7-1684"><a href="#cb7-1684" aria-hidden="true" tabindex="-1"></a><span class="in">                paste0(</span></span>
<span id="cb7-1685"><a href="#cb7-1685" aria-hidden="true" tabindex="-1"></a><span class="in">                    names(pathways),</span></span>
<span id="cb7-1686"><a href="#cb7-1686" aria-hidden="true" tabindex="-1"></a><span class="in">                    "_",</span></span>
<span id="cb7-1687"><a href="#cb7-1687" aria-hidden="true" tabindex="-1"></a><span class="in">                    og_or_new</span></span>
<span id="cb7-1688"><a href="#cb7-1688" aria-hidden="true" tabindex="-1"></a><span class="in">                )</span></span>
<span id="cb7-1689"><a href="#cb7-1689" aria-hidden="true" tabindex="-1"></a><span class="in">            ),</span></span>
<span id="cb7-1690"><a href="#cb7-1690" aria-hidden="true" tabindex="-1"></a><span class="in">            names_to = "pathway",</span></span>
<span id="cb7-1691"><a href="#cb7-1691" aria-hidden="true" tabindex="-1"></a><span class="in">            values_to = "score"</span></span>
<span id="cb7-1692"><a href="#cb7-1692" aria-hidden="true" tabindex="-1"></a><span class="in">        ) %&gt;%</span></span>
<span id="cb7-1693"><a href="#cb7-1693" aria-hidden="true" tabindex="-1"></a><span class="in">        ggplot2::ggplot(aes(</span></span>
<span id="cb7-1694"><a href="#cb7-1694" aria-hidden="true" tabindex="-1"></a><span class="in">            color = is_responder, </span></span>
<span id="cb7-1695"><a href="#cb7-1695" aria-hidden="true" tabindex="-1"></a><span class="in">            y = score,</span></span>
<span id="cb7-1696"><a href="#cb7-1696" aria-hidden="true" tabindex="-1"></a><span class="in">            x = is_responder</span></span>
<span id="cb7-1697"><a href="#cb7-1697" aria-hidden="true" tabindex="-1"></a><span class="in">        )) +</span></span>
<span id="cb7-1698"><a href="#cb7-1698" aria-hidden="true" tabindex="-1"></a><span class="in">        ggplot2::geom_boxplot(outlier.shape = NA) + </span></span>
<span id="cb7-1699"><a href="#cb7-1699" aria-hidden="true" tabindex="-1"></a><span class="in">        ggplot2::geom_jitter() +</span></span>
<span id="cb7-1700"><a href="#cb7-1700" aria-hidden="true" tabindex="-1"></a><span class="in">        ggplot2::labs(</span></span>
<span id="cb7-1701"><a href="#cb7-1701" aria-hidden="true" tabindex="-1"></a><span class="in">            x = "Is responder?",</span></span>
<span id="cb7-1702"><a href="#cb7-1702" aria-hidden="true" tabindex="-1"></a><span class="in">            y = ifelse(og_or_new == "new", </span></span>
<span id="cb7-1703"><a href="#cb7-1703" aria-hidden="true" tabindex="-1"></a><span class="in">                       "Scores with regressed data",</span></span>
<span id="cb7-1704"><a href="#cb7-1704" aria-hidden="true" tabindex="-1"></a><span class="in">                       "GSVA scores")</span></span>
<span id="cb7-1705"><a href="#cb7-1705" aria-hidden="true" tabindex="-1"></a><span class="in">        ) +</span></span>
<span id="cb7-1706"><a href="#cb7-1706" aria-hidden="true" tabindex="-1"></a><span class="in">        ggplot2::facet_wrap(</span></span>
<span id="cb7-1707"><a href="#cb7-1707" aria-hidden="true" tabindex="-1"></a><span class="in">            ~pathway,</span></span>
<span id="cb7-1708"><a href="#cb7-1708" aria-hidden="true" tabindex="-1"></a><span class="in">            labeller = as_labeller(pathways_labeller),</span></span>
<span id="cb7-1709"><a href="#cb7-1709" aria-hidden="true" tabindex="-1"></a><span class="in">            scales = "free_y"</span></span>
<span id="cb7-1710"><a href="#cb7-1710" aria-hidden="true" tabindex="-1"></a><span class="in">        ) + </span></span>
<span id="cb7-1711"><a href="#cb7-1711" aria-hidden="true" tabindex="-1"></a><span class="in">        ggplot2::theme_bw(base_size = 30) +</span></span>
<span id="cb7-1712"><a href="#cb7-1712" aria-hidden="true" tabindex="-1"></a><span class="in">        ggplot2::theme(legend.position = "none") + </span></span>
<span id="cb7-1713"><a href="#cb7-1713" aria-hidden="true" tabindex="-1"></a><span class="in">        change_plot_aes_point()</span></span>
<span id="cb7-1714"><a href="#cb7-1714" aria-hidden="true" tabindex="-1"></a><span class="in">}</span></span>
<span id="cb7-1715"><a href="#cb7-1715" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1716"><a href="#cb7-1716" aria-hidden="true" tabindex="-1"></a><span class="in">regressed_plot_er &lt;- plot_responders_non_responder_comparison(df_pca_scores)</span></span>
<span id="cb7-1717"><a href="#cb7-1717" aria-hidden="true" tabindex="-1"></a><span class="in">regressed_plot_er</span></span>
<span id="cb7-1718"><a href="#cb7-1718" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-1719"><a href="#cb7-1719" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1720"><a href="#cb7-1720" aria-hidden="true" tabindex="-1"></a>We see that the estrogen response early has a wider range when compared to the</span>
<span id="cb7-1721"><a href="#cb7-1721" aria-hidden="true" tabindex="-1"></a>SET ER/PR score. That makes sense as this signature has over 100 genes,</span>
<span id="cb7-1722"><a href="#cb7-1722" aria-hidden="true" tabindex="-1"></a>whereas SET ER/PR has only 18. Also the responder</span>
<span id="cb7-1723"><a href="#cb7-1723" aria-hidden="true" tabindex="-1"></a>was very good in discriminating the responders and non responders.</span>
<span id="cb7-1724"><a href="#cb7-1724" aria-hidden="true" tabindex="-1"></a>What we can conclude from this figure is that the samples</span>
<span id="cb7-1725"><a href="#cb7-1725" aria-hidden="true" tabindex="-1"></a>with very low scores are usually non responders, which reflect</span>
<span id="cb7-1726"><a href="#cb7-1726" aria-hidden="true" tabindex="-1"></a>their position in the molecular landscape, also in general </span>
<span id="cb7-1727"><a href="#cb7-1727" aria-hidden="true" tabindex="-1"></a>responders have higher ER signaling than non responders.</span>
<span id="cb7-1728"><a href="#cb7-1728" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1729"><a href="#cb7-1729" aria-hidden="true" tabindex="-1"></a>Below is the figure with the GSVA scores before regressing the data, </span>
<span id="cb7-1730"><a href="#cb7-1730" aria-hidden="true" tabindex="-1"></a>we see that SET ER/PR is not able to really distinguish </span>
<span id="cb7-1731"><a href="#cb7-1731" aria-hidden="true" tabindex="-1"></a>completely the non-responders with very low score similar</span>
<span id="cb7-1732"><a href="#cb7-1732" aria-hidden="true" tabindex="-1"></a>to what was done with GSVA. Even estrogen early had</span>
<span id="cb7-1733"><a href="#cb7-1733" aria-hidden="true" tabindex="-1"></a>some differences.</span>
<span id="cb7-1734"><a href="#cb7-1734" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1735"><a href="#cb7-1735" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=6}</span></span>
<span id="cb7-1736"><a href="#cb7-1736" aria-hidden="true" tabindex="-1"></a><span class="in">gsva_plot_er &lt;- plot_responders_non_responder_comparison(</span></span>
<span id="cb7-1737"><a href="#cb7-1737" aria-hidden="true" tabindex="-1"></a><span class="in">    df_pca_scores,</span></span>
<span id="cb7-1738"><a href="#cb7-1738" aria-hidden="true" tabindex="-1"></a><span class="in">    og_or_new = "og"</span></span>
<span id="cb7-1739"><a href="#cb7-1739" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-1740"><a href="#cb7-1740" aria-hidden="true" tabindex="-1"></a><span class="in">gsva_plot_er</span></span>
<span id="cb7-1741"><a href="#cb7-1741" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-1742"><a href="#cb7-1742" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1743"><a href="#cb7-1743" aria-hidden="true" tabindex="-1"></a>And below is a figure with both GSVA and regressed scores together.</span>
<span id="cb7-1744"><a href="#cb7-1744" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1745"><a href="#cb7-1745" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=10}</span></span>
<span id="cb7-1746"><a href="#cb7-1746" aria-hidden="true" tabindex="-1"></a><span class="in">#| label: fig-comparison-scoring-strategy-poetic</span></span>
<span id="cb7-1747"><a href="#cb7-1747" aria-hidden="true" tabindex="-1"></a><span class="in">#| fig-cap: Comparing scoring strategies for responders and non responders</span></span>
<span id="cb7-1748"><a href="#cb7-1748" aria-hidden="true" tabindex="-1"></a><span class="in">#|     in the POETIC dataset. </span></span>
<span id="cb7-1749"><a href="#cb7-1749" aria-hidden="true" tabindex="-1"></a><span class="in">cowplot::plot_grid(</span></span>
<span id="cb7-1750"><a href="#cb7-1750" aria-hidden="true" tabindex="-1"></a><span class="in">    regressed_plot_er,</span></span>
<span id="cb7-1751"><a href="#cb7-1751" aria-hidden="true" tabindex="-1"></a><span class="in">    gsva_plot_er,</span></span>
<span id="cb7-1752"><a href="#cb7-1752" aria-hidden="true" tabindex="-1"></a><span class="in">    ncol = 1</span></span>
<span id="cb7-1753"><a href="#cb7-1753" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-1754"><a href="#cb7-1754" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-1755"><a href="#cb7-1755" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1756"><a href="#cb7-1756" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1757"><a href="#cb7-1757" aria-hidden="true" tabindex="-1"></a>Below we show the correlation between</span>
<span id="cb7-1758"><a href="#cb7-1758" aria-hidden="true" tabindex="-1"></a>these scores for all cohorts.</span>
<span id="cb7-1759"><a href="#cb7-1759" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1760"><a href="#cb7-1760" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=6, fig.height=8}</span></span>
<span id="cb7-1761"><a href="#cb7-1761" aria-hidden="true" tabindex="-1"></a><span class="in">df_pca_scores %&gt;%</span></span>
<span id="cb7-1762"><a href="#cb7-1762" aria-hidden="true" tabindex="-1"></a><span class="in">    #dplyr::filter(cohort != "poetic") %&gt;%</span></span>
<span id="cb7-1763"><a href="#cb7-1763" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::ggplot(aes(</span></span>
<span id="cb7-1764"><a href="#cb7-1764" aria-hidden="true" tabindex="-1"></a><span class="in">        x = HALLMARK_ESTROGEN_RESPONSE_EARLY_new, </span></span>
<span id="cb7-1765"><a href="#cb7-1765" aria-hidden="true" tabindex="-1"></a><span class="in">        y = SET_ERPR_new,</span></span>
<span id="cb7-1766"><a href="#cb7-1766" aria-hidden="true" tabindex="-1"></a><span class="in">        color = pam50</span></span>
<span id="cb7-1767"><a href="#cb7-1767" aria-hidden="true" tabindex="-1"></a><span class="in">    )) +</span></span>
<span id="cb7-1768"><a href="#cb7-1768" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_point() + </span></span>
<span id="cb7-1769"><a href="#cb7-1769" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_smooth(</span></span>
<span id="cb7-1770"><a href="#cb7-1770" aria-hidden="true" tabindex="-1"></a><span class="in">        color = "black", </span></span>
<span id="cb7-1771"><a href="#cb7-1771" aria-hidden="true" tabindex="-1"></a><span class="in">        alpha = 0.5,</span></span>
<span id="cb7-1772"><a href="#cb7-1772" aria-hidden="true" tabindex="-1"></a><span class="in">        method = "gam",</span></span>
<span id="cb7-1773"><a href="#cb7-1773" aria-hidden="true" tabindex="-1"></a><span class="in">        formula = y ~ s(x, bs = "cs")</span></span>
<span id="cb7-1774"><a href="#cb7-1774" aria-hidden="true" tabindex="-1"></a><span class="in">    ) + </span></span>
<span id="cb7-1775"><a href="#cb7-1775" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::facet_wrap(~cohort, ncol = 1) +</span></span>
<span id="cb7-1776"><a href="#cb7-1776" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme_bw(base_size = 20) + </span></span>
<span id="cb7-1777"><a href="#cb7-1777" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::scale_color_manual(</span></span>
<span id="cb7-1778"><a href="#cb7-1778" aria-hidden="true" tabindex="-1"></a><span class="in">        values = get_colors_pam50(</span></span>
<span id="cb7-1779"><a href="#cb7-1779" aria-hidden="true" tabindex="-1"></a><span class="in">            df_pca_scores %&gt;%</span></span>
<span id="cb7-1780"><a href="#cb7-1780" aria-hidden="true" tabindex="-1"></a><span class="in">                dplyr::filter(cohort != "poetic")</span></span>
<span id="cb7-1781"><a href="#cb7-1781" aria-hidden="true" tabindex="-1"></a><span class="in">        )</span></span>
<span id="cb7-1782"><a href="#cb7-1782" aria-hidden="true" tabindex="-1"></a><span class="in">    ) +</span></span>
<span id="cb7-1783"><a href="#cb7-1783" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::labs(x = "Estrogen Early", y = "SET ER/PR") + </span></span>
<span id="cb7-1784"><a href="#cb7-1784" aria-hidden="true" tabindex="-1"></a><span class="in">    change_guides_point() + </span></span>
<span id="cb7-1785"><a href="#cb7-1785" aria-hidden="true" tabindex="-1"></a><span class="in">    change_plot_aes_point()</span></span>
<span id="cb7-1786"><a href="#cb7-1786" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-1787"><a href="#cb7-1787" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1788"><a href="#cb7-1788" aria-hidden="true" tabindex="-1"></a>These scores are highly correlated but in different absolute</span>
<span id="cb7-1789"><a href="#cb7-1789" aria-hidden="true" tabindex="-1"></a>scales. We now check the change in Ki67 for all patients and </span>
<span id="cb7-1790"><a href="#cb7-1790" aria-hidden="true" tabindex="-1"></a>compare the scores with the original ones obtained with GSVA.</span>
<span id="cb7-1791"><a href="#cb7-1791" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1792"><a href="#cb7-1792" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=12, fig.height=6}</span></span>
<span id="cb7-1793"><a href="#cb7-1793" aria-hidden="true" tabindex="-1"></a><span class="in">#| label: fig-comparison-og-regressed-er</span></span>
<span id="cb7-1794"><a href="#cb7-1794" aria-hidden="true" tabindex="-1"></a><span class="in">#| fig-cap: Comparison of original GSVA and regressed scores between responders</span></span>
<span id="cb7-1795"><a href="#cb7-1795" aria-hidden="true" tabindex="-1"></a><span class="in">#|     and non responders compared to change in Ki67.</span></span>
<span id="cb7-1796"><a href="#cb7-1796" aria-hidden="true" tabindex="-1"></a><span class="in">df_pca_scores %&gt;%</span></span>
<span id="cb7-1797"><a href="#cb7-1797" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::filter(</span></span>
<span id="cb7-1798"><a href="#cb7-1798" aria-hidden="true" tabindex="-1"></a><span class="in">        cohort == "poetic" &amp; </span></span>
<span id="cb7-1799"><a href="#cb7-1799" aria-hidden="true" tabindex="-1"></a><span class="in">            timepoint == "baseline" &amp; </span></span>
<span id="cb7-1800"><a href="#cb7-1800" aria-hidden="true" tabindex="-1"></a><span class="in">            is_responder != "not_available"</span></span>
<span id="cb7-1801"><a href="#cb7-1801" aria-hidden="true" tabindex="-1"></a><span class="in">    ) %&gt;%</span></span>
<span id="cb7-1802"><a href="#cb7-1802" aria-hidden="true" tabindex="-1"></a><span class="in">    tidyr::pivot_longer(</span></span>
<span id="cb7-1803"><a href="#cb7-1803" aria-hidden="true" tabindex="-1"></a><span class="in">        cols = c(</span></span>
<span id="cb7-1804"><a href="#cb7-1804" aria-hidden="true" tabindex="-1"></a><span class="in">            "HALLMARK_ESTROGEN_RESPONSE_EARLY_og",</span></span>
<span id="cb7-1805"><a href="#cb7-1805" aria-hidden="true" tabindex="-1"></a><span class="in">            "HALLMARK_ESTROGEN_RESPONSE_EARLY_new",</span></span>
<span id="cb7-1806"><a href="#cb7-1806" aria-hidden="true" tabindex="-1"></a><span class="in">            "SET_ERPR_og",</span></span>
<span id="cb7-1807"><a href="#cb7-1807" aria-hidden="true" tabindex="-1"></a><span class="in">            "SET_ERPR_new"</span></span>
<span id="cb7-1808"><a href="#cb7-1808" aria-hidden="true" tabindex="-1"></a><span class="in">        ),</span></span>
<span id="cb7-1809"><a href="#cb7-1809" aria-hidden="true" tabindex="-1"></a><span class="in">        names_to = "pathway",</span></span>
<span id="cb7-1810"><a href="#cb7-1810" aria-hidden="true" tabindex="-1"></a><span class="in">        values_to = "score"</span></span>
<span id="cb7-1811"><a href="#cb7-1811" aria-hidden="true" tabindex="-1"></a><span class="in">    ) %&gt;%</span></span>
<span id="cb7-1812"><a href="#cb7-1812" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::ggplot(aes(</span></span>
<span id="cb7-1813"><a href="#cb7-1813" aria-hidden="true" tabindex="-1"></a><span class="in">        x = score,</span></span>
<span id="cb7-1814"><a href="#cb7-1814" aria-hidden="true" tabindex="-1"></a><span class="in">        y = change_ki67</span></span>
<span id="cb7-1815"><a href="#cb7-1815" aria-hidden="true" tabindex="-1"></a><span class="in">    )) + </span></span>
<span id="cb7-1816"><a href="#cb7-1816" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_smooth(method = "lm", formula = "y~x") +</span></span>
<span id="cb7-1817"><a href="#cb7-1817" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_point() +</span></span>
<span id="cb7-1818"><a href="#cb7-1818" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::facet_wrap(</span></span>
<span id="cb7-1819"><a href="#cb7-1819" aria-hidden="true" tabindex="-1"></a><span class="in">        pathway~is_responder, </span></span>
<span id="cb7-1820"><a href="#cb7-1820" aria-hidden="true" tabindex="-1"></a><span class="in">        scales = "free",</span></span>
<span id="cb7-1821"><a href="#cb7-1821" aria-hidden="true" tabindex="-1"></a><span class="in">        ncol = 4,</span></span>
<span id="cb7-1822"><a href="#cb7-1822" aria-hidden="true" tabindex="-1"></a><span class="in">        labeller = as_labeller(c(</span></span>
<span id="cb7-1823"><a href="#cb7-1823" aria-hidden="true" tabindex="-1"></a><span class="in">            "non_responder" = "Non-responder",</span></span>
<span id="cb7-1824"><a href="#cb7-1824" aria-hidden="true" tabindex="-1"></a><span class="in">            "responder" = "Responder",</span></span>
<span id="cb7-1825"><a href="#cb7-1825" aria-hidden="true" tabindex="-1"></a><span class="in">            "HALLMARK_ESTROGEN_RESPONSE_EARLY_og" = "GSVA estrogen early",</span></span>
<span id="cb7-1826"><a href="#cb7-1826" aria-hidden="true" tabindex="-1"></a><span class="in">            "HALLMARK_ESTROGEN_RESPONSE_EARLY_new" = "Regressed estrogen early",</span></span>
<span id="cb7-1827"><a href="#cb7-1827" aria-hidden="true" tabindex="-1"></a><span class="in">            "SET_ERPR_og" = "GSVA SET ER/PR",</span></span>
<span id="cb7-1828"><a href="#cb7-1828" aria-hidden="true" tabindex="-1"></a><span class="in">            "SET_ERPR_new" = "Regressed SET ER/PR"</span></span>
<span id="cb7-1829"><a href="#cb7-1829" aria-hidden="true" tabindex="-1"></a><span class="in">        ))</span></span>
<span id="cb7-1830"><a href="#cb7-1830" aria-hidden="true" tabindex="-1"></a><span class="in">    ) +</span></span>
<span id="cb7-1831"><a href="#cb7-1831" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::labs(</span></span>
<span id="cb7-1832"><a href="#cb7-1832" aria-hidden="true" tabindex="-1"></a><span class="in">        x = "Scores",</span></span>
<span id="cb7-1833"><a href="#cb7-1833" aria-hidden="true" tabindex="-1"></a><span class="in">        y = "Change in Ki67 (%)"</span></span>
<span id="cb7-1834"><a href="#cb7-1834" aria-hidden="true" tabindex="-1"></a><span class="in">    ) + </span></span>
<span id="cb7-1835"><a href="#cb7-1835" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme_bw(base_size = 15) + </span></span>
<span id="cb7-1836"><a href="#cb7-1836" aria-hidden="true" tabindex="-1"></a><span class="in">    change_guides_point() + </span></span>
<span id="cb7-1837"><a href="#cb7-1837" aria-hidden="true" tabindex="-1"></a><span class="in">    change_plot_aes_point()</span></span>
<span id="cb7-1838"><a href="#cb7-1838" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-1839"><a href="#cb7-1839" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1840"><a href="#cb7-1840" aria-hidden="true" tabindex="-1"></a>We see that the bigger the change the higher the score is when using the</span>
<span id="cb7-1841"><a href="#cb7-1841" aria-hidden="true" tabindex="-1"></a>regressed data. Also the non responders tend to have scores mostly in</span>
<span id="cb7-1842"><a href="#cb7-1842" aria-hidden="true" tabindex="-1"></a>the negative scale, whereas the responders are more shifted to the</span>
<span id="cb7-1843"><a href="#cb7-1843" aria-hidden="true" tabindex="-1"></a>right. SET ER/PR is also a good measure here. </span>
<span id="cb7-1844"><a href="#cb7-1844" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1847"><a href="#cb7-1847" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb7-1848"><a href="#cb7-1848" aria-hidden="true" tabindex="-1"></a>df_pca_scores <span class="sc">%&gt;%</span></span>
<span id="cb7-1849"><a href="#cb7-1849" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb7-1850"><a href="#cb7-1850" aria-hidden="true" tabindex="-1"></a>        cohort <span class="sc">==</span> <span class="st">"poetic"</span> <span class="sc">&amp;</span> </span>
<span id="cb7-1851"><a href="#cb7-1851" aria-hidden="true" tabindex="-1"></a>            timepoint <span class="sc">==</span> <span class="st">"baseline"</span> <span class="sc">&amp;</span> </span>
<span id="cb7-1852"><a href="#cb7-1852" aria-hidden="true" tabindex="-1"></a>            is_responder <span class="sc">!=</span> <span class="st">"not_available"</span></span>
<span id="cb7-1853"><a href="#cb7-1853" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb7-1854"><a href="#cb7-1854" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(</span>
<span id="cb7-1855"><a href="#cb7-1855" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> HALLMARK_ESTROGEN_RESPONSE_EARLY_og,</span>
<span id="cb7-1856"><a href="#cb7-1856" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> HALLMARK_ESTROGEN_RESPONSE_EARLY_new,</span>
<span id="cb7-1857"><a href="#cb7-1857" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> is_responder</span>
<span id="cb7-1858"><a href="#cb7-1858" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">+</span> </span>
<span id="cb7-1859"><a href="#cb7-1859" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_smooth</span>(</span>
<span id="cb7-1860"><a href="#cb7-1860" aria-hidden="true" tabindex="-1"></a>        <span class="at">method =</span> <span class="st">"gam"</span>,</span>
<span id="cb7-1861"><a href="#cb7-1861" aria-hidden="true" tabindex="-1"></a>        <span class="at">formula =</span> y <span class="sc">~</span> <span class="fu">s</span>(x, <span class="at">bs =</span> <span class="st">"cs"</span>), </span>
<span id="cb7-1862"><a href="#cb7-1862" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">"black"</span></span>
<span id="cb7-1863"><a href="#cb7-1863" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb7-1864"><a href="#cb7-1864" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb7-1865"><a href="#cb7-1865" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb7-1866"><a href="#cb7-1866" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"GSVA Estrogen early"</span>,</span>
<span id="cb7-1867"><a href="#cb7-1867" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"Regressed Estrogen early"</span></span>
<span id="cb7-1868"><a href="#cb7-1868" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb7-1869"><a href="#cb7-1869" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">15</span>) <span class="sc">+</span> </span>
<span id="cb7-1870"><a href="#cb7-1870" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>() <span class="sc">+</span> </span>
<span id="cb7-1871"><a href="#cb7-1871" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>()</span>
<span id="cb7-1872"><a href="#cb7-1872" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-1873"><a href="#cb7-1873" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1874"><a href="#cb7-1874" aria-hidden="true" tabindex="-1"></a>There is a very good match between the two scores in general,</span>
<span id="cb7-1875"><a href="#cb7-1875" aria-hidden="true" tabindex="-1"></a>meaning that the regression scoring strategy capture relatively</span>
<span id="cb7-1876"><a href="#cb7-1876" aria-hidden="true" tabindex="-1"></a>well the scores. There is one patient that is a non-responder</span>
<span id="cb7-1877"><a href="#cb7-1877" aria-hidden="true" tabindex="-1"></a>and has a low regressed estrogen early but a GSVA score </span>
<span id="cb7-1878"><a href="#cb7-1878" aria-hidden="true" tabindex="-1"></a>close to 0, let us check this patient.</span>
<span id="cb7-1879"><a href="#cb7-1879" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1882"><a href="#cb7-1882" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb7-1883"><a href="#cb7-1883" aria-hidden="true" tabindex="-1"></a>df_pca_scores <span class="sc">%&gt;%</span></span>
<span id="cb7-1884"><a href="#cb7-1884" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb7-1885"><a href="#cb7-1885" aria-hidden="true" tabindex="-1"></a>        is_responder <span class="sc">==</span> <span class="st">"non_responder"</span> <span class="sc">&amp;</span> </span>
<span id="cb7-1886"><a href="#cb7-1886" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">between</span>(HALLMARK_ESTROGEN_RESPONSE_EARLY_og, <span class="sc">-</span><span class="fl">0.1</span>, <span class="fl">0.1</span>) <span class="sc">&amp;</span></span>
<span id="cb7-1887"><a href="#cb7-1887" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">between</span>(HALLMARK_ESTROGEN_RESPONSE_EARLY_new, <span class="sc">-</span><span class="dv">20</span>, <span class="sc">-</span><span class="dv">15</span>) </span>
<span id="cb7-1888"><a href="#cb7-1888" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb7-1889"><a href="#cb7-1889" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(patient_nb, ki67, PC3, change_ki67, er_status) <span class="sc">%&gt;%</span></span>
<span id="cb7-1890"><a href="#cb7-1890" aria-hidden="true" tabindex="-1"></a>    DT<span class="sc">::</span><span class="fu">datatable</span>()</span>
<span id="cb7-1891"><a href="#cb7-1891" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-1892"><a href="#cb7-1892" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1893"><a href="#cb7-1893" aria-hidden="true" tabindex="-1"></a>It is a patient that is far to the left on the molecular landscape, among the</span>
<span id="cb7-1894"><a href="#cb7-1894" aria-hidden="true" tabindex="-1"></a>basal like, and had an increase in Ki67 upon ET. So the </span>
<span id="cb7-1895"><a href="#cb7-1895" aria-hidden="true" tabindex="-1"></a>new scoring system was able to capture this patient as with low score. </span>
<span id="cb7-1896"><a href="#cb7-1896" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1897"><a href="#cb7-1897" aria-hidden="true" tabindex="-1"></a><span class="fu">### Survival analysis with the new scores</span></span>
<span id="cb7-1898"><a href="#cb7-1898" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1899"><a href="#cb7-1899" aria-hidden="true" tabindex="-1"></a>We can check how robust these scores are by performing survival analysis.</span>
<span id="cb7-1900"><a href="#cb7-1900" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1901"><a href="#cb7-1901" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb7-1902"><a href="#cb7-1902" aria-hidden="true" tabindex="-1"></a><span class="in">tcga_samples &lt;- colData(datasets$tcga) %&gt;% data.frame %&gt;%</span></span>
<span id="cb7-1903"><a href="#cb7-1903" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::filter(</span></span>
<span id="cb7-1904"><a href="#cb7-1904" aria-hidden="true" tabindex="-1"></a><span class="in">        pam50 %in% c("luma", "lumb") &amp;</span></span>
<span id="cb7-1905"><a href="#cb7-1905" aria-hidden="true" tabindex="-1"></a><span class="in">        node_stage != "NX" &amp;</span></span>
<span id="cb7-1906"><a href="#cb7-1906" aria-hidden="true" tabindex="-1"></a><span class="in">        !(tumor_stage %in% c("stage_iv", "na")) &amp; </span></span>
<span id="cb7-1907"><a href="#cb7-1907" aria-hidden="true" tabindex="-1"></a><span class="in">        er_status == "pos"</span></span>
<span id="cb7-1908"><a href="#cb7-1908" aria-hidden="true" tabindex="-1"></a><span class="in">    ) %&gt;% </span></span>
<span id="cb7-1909"><a href="#cb7-1909" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::pull(sample_name)</span></span>
<span id="cb7-1910"><a href="#cb7-1910" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1911"><a href="#cb7-1911" aria-hidden="true" tabindex="-1"></a><span class="in">scanb_samples &lt;- colData(datasets$scanb) %&gt;% data.frame %&gt;%</span></span>
<span id="cb7-1912"><a href="#cb7-1912" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::filter(</span></span>
<span id="cb7-1913"><a href="#cb7-1913" aria-hidden="true" tabindex="-1"></a><span class="in">        node_stage != "NX" &amp;</span></span>
<span id="cb7-1914"><a href="#cb7-1914" aria-hidden="true" tabindex="-1"></a><span class="in">        !(tumor_stage %in% c("T4", "Tis", "TX")) &amp; </span></span>
<span id="cb7-1915"><a href="#cb7-1915" aria-hidden="true" tabindex="-1"></a><span class="in">        er_status == "pos" &amp; </span></span>
<span id="cb7-1916"><a href="#cb7-1916" aria-hidden="true" tabindex="-1"></a><span class="in">        TreatGroup == "Endo"</span></span>
<span id="cb7-1917"><a href="#cb7-1917" aria-hidden="true" tabindex="-1"></a><span class="in">    ) %&gt;% </span></span>
<span id="cb7-1918"><a href="#cb7-1918" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::pull(sample_name)</span></span>
<span id="cb7-1919"><a href="#cb7-1919" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1920"><a href="#cb7-1920" aria-hidden="true" tabindex="-1"></a><span class="in">metabric_samples &lt;- colData(datasets$metabric) %&gt;% data.frame %&gt;%</span></span>
<span id="cb7-1921"><a href="#cb7-1921" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::filter(</span></span>
<span id="cb7-1922"><a href="#cb7-1922" aria-hidden="true" tabindex="-1"></a><span class="in">        CHEMOTHERAPY == "NO" &amp;</span></span>
<span id="cb7-1923"><a href="#cb7-1923" aria-hidden="true" tabindex="-1"></a><span class="in">        er_status == "pos" &amp; </span></span>
<span id="cb7-1924"><a href="#cb7-1924" aria-hidden="true" tabindex="-1"></a><span class="in">        HORMONE_THERAPY == "YES"</span></span>
<span id="cb7-1925"><a href="#cb7-1925" aria-hidden="true" tabindex="-1"></a><span class="in">    ) %&gt;% </span></span>
<span id="cb7-1926"><a href="#cb7-1926" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::pull(sample_name)</span></span>
<span id="cb7-1927"><a href="#cb7-1927" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1928"><a href="#cb7-1928" aria-hidden="true" tabindex="-1"></a><span class="in">formulas_survival &lt;- c(</span></span>
<span id="cb7-1929"><a href="#cb7-1929" aria-hidden="true" tabindex="-1"></a><span class="in">    "os_tcga" = "Surv(os_months, os_status) ~ age + node_stage + tumor_stage",</span></span>
<span id="cb7-1930"><a href="#cb7-1930" aria-hidden="true" tabindex="-1"></a><span class="in">    "os_scanb" = "Surv(os_months, os_status) ~ age + node_stage + tumor_stage",</span></span>
<span id="cb7-1931"><a href="#cb7-1931" aria-hidden="true" tabindex="-1"></a><span class="in">    "os_metabric" = "Surv(os_months, os_status) ~ age + npi",</span></span>
<span id="cb7-1932"><a href="#cb7-1932" aria-hidden="true" tabindex="-1"></a><span class="in">    "rfs_metabric" = "Surv(rfs_months, rfs_status) ~ age + npi",</span></span>
<span id="cb7-1933"><a href="#cb7-1933" aria-hidden="true" tabindex="-1"></a><span class="in">    "rfs_scanb" = "Surv(rfs_months, rfs_status) ~ age + node_stage + tumor_stage"</span></span>
<span id="cb7-1934"><a href="#cb7-1934" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-1935"><a href="#cb7-1935" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-1936"><a href="#cb7-1936" aria-hidden="true" tabindex="-1"></a><span class="in">type_survival &lt;- c("os", "rfs")</span></span>
<span id="cb7-1937"><a href="#cb7-1937" aria-hidden="true" tabindex="-1"></a><span class="in">which_scores &lt;- c(</span></span>
<span id="cb7-1938"><a href="#cb7-1938" aria-hidden="true" tabindex="-1"></a><span class="in">    paste0(c(</span></span>
<span id="cb7-1939"><a href="#cb7-1939" aria-hidden="true" tabindex="-1"></a><span class="in">        "HALLMARK_G2M_CHECKPOINT",</span></span>
<span id="cb7-1940"><a href="#cb7-1940" aria-hidden="true" tabindex="-1"></a><span class="in">        "HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION",</span></span>
<span id="cb7-1941"><a href="#cb7-1941" aria-hidden="true" tabindex="-1"></a><span class="in">        "HALLMARK_DNA_REPAIR",</span></span>
<span id="cb7-1942"><a href="#cb7-1942" aria-hidden="true" tabindex="-1"></a><span class="in">        "HALLMARK_ANDROGEN_RESPONSE",</span></span>
<span id="cb7-1943"><a href="#cb7-1943" aria-hidden="true" tabindex="-1"></a><span class="in">        "HALLMARK_PI3K_AKT_MTOR_SIGNALING",</span></span>
<span id="cb7-1944"><a href="#cb7-1944" aria-hidden="true" tabindex="-1"></a><span class="in">        "SET_ERPR",</span></span>
<span id="cb7-1945"><a href="#cb7-1945" aria-hidden="true" tabindex="-1"></a><span class="in">        "HALLMARK_ESTROGEN_RESPONSE_EARLY"</span></span>
<span id="cb7-1946"><a href="#cb7-1946" aria-hidden="true" tabindex="-1"></a><span class="in">    ), "_new")</span></span>
<span id="cb7-1947"><a href="#cb7-1947" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-1948"><a href="#cb7-1948" aria-hidden="true" tabindex="-1"></a><span class="in">survival_results &lt;- sapply(</span></span>
<span id="cb7-1949"><a href="#cb7-1949" aria-hidden="true" tabindex="-1"></a><span class="in">    c("os", "rfs"),</span></span>
<span id="cb7-1950"><a href="#cb7-1950" aria-hidden="true" tabindex="-1"></a><span class="in">    function(type_analysis, datasets, name_scores, formulas){</span></span>
<span id="cb7-1951"><a href="#cb7-1951" aria-hidden="true" tabindex="-1"></a><span class="in">        mapply(</span></span>
<span id="cb7-1952"><a href="#cb7-1952" aria-hidden="true" tabindex="-1"></a><span class="in">            function(</span></span>
<span id="cb7-1953"><a href="#cb7-1953" aria-hidden="true" tabindex="-1"></a><span class="in">                dataset, name_dataset, name_scores, formulas, type_analysis</span></span>
<span id="cb7-1954"><a href="#cb7-1954" aria-hidden="true" tabindex="-1"></a><span class="in">            ){</span></span>
<span id="cb7-1955"><a href="#cb7-1955" aria-hidden="true" tabindex="-1"></a><span class="in">                </span></span>
<span id="cb7-1956"><a href="#cb7-1956" aria-hidden="true" tabindex="-1"></a><span class="in">                if (type_analysis == "rfs" &amp; name_dataset == "tcga"){</span></span>
<span id="cb7-1957"><a href="#cb7-1957" aria-hidden="true" tabindex="-1"></a><span class="in">                    return()</span></span>
<span id="cb7-1958"><a href="#cb7-1958" aria-hidden="true" tabindex="-1"></a><span class="in">                } else { </span></span>
<span id="cb7-1959"><a href="#cb7-1959" aria-hidden="true" tabindex="-1"></a><span class="in">                </span></span>
<span id="cb7-1960"><a href="#cb7-1960" aria-hidden="true" tabindex="-1"></a><span class="in">                    sapply(</span></span>
<span id="cb7-1961"><a href="#cb7-1961" aria-hidden="true" tabindex="-1"></a><span class="in">                        name_scores,</span></span>
<span id="cb7-1962"><a href="#cb7-1962" aria-hidden="true" tabindex="-1"></a><span class="in">                        function(name_score, col_data, formula){</span></span>
<span id="cb7-1963"><a href="#cb7-1963" aria-hidden="true" tabindex="-1"></a><span class="in">                            </span></span>
<span id="cb7-1964"><a href="#cb7-1964" aria-hidden="true" tabindex="-1"></a><span class="in">                            col_data[, name_score] &lt;- base::scale(</span></span>
<span id="cb7-1965"><a href="#cb7-1965" aria-hidden="true" tabindex="-1"></a><span class="in">                                col_data[, name_score]</span></span>
<span id="cb7-1966"><a href="#cb7-1966" aria-hidden="true" tabindex="-1"></a><span class="in">                            )</span></span>
<span id="cb7-1967"><a href="#cb7-1967" aria-hidden="true" tabindex="-1"></a><span class="in">                            </span></span>
<span id="cb7-1968"><a href="#cb7-1968" aria-hidden="true" tabindex="-1"></a><span class="in">                            survival::coxph(</span></span>
<span id="cb7-1969"><a href="#cb7-1969" aria-hidden="true" tabindex="-1"></a><span class="in">                                as.formula(</span></span>
<span id="cb7-1970"><a href="#cb7-1970" aria-hidden="true" tabindex="-1"></a><span class="in">                                    paste(formula, name_score, sep = "+")</span></span>
<span id="cb7-1971"><a href="#cb7-1971" aria-hidden="true" tabindex="-1"></a><span class="in">                                ),</span></span>
<span id="cb7-1972"><a href="#cb7-1972" aria-hidden="true" tabindex="-1"></a><span class="in">                                data = col_data, </span></span>
<span id="cb7-1973"><a href="#cb7-1973" aria-hidden="true" tabindex="-1"></a><span class="in">                                y = FALSE,</span></span>
<span id="cb7-1974"><a href="#cb7-1974" aria-hidden="true" tabindex="-1"></a><span class="in">                                x = FALSE</span></span>
<span id="cb7-1975"><a href="#cb7-1975" aria-hidden="true" tabindex="-1"></a><span class="in">                            )</span></span>
<span id="cb7-1976"><a href="#cb7-1976" aria-hidden="true" tabindex="-1"></a><span class="in">                        },</span></span>
<span id="cb7-1977"><a href="#cb7-1977" aria-hidden="true" tabindex="-1"></a><span class="in">                        col_data = dataset,</span></span>
<span id="cb7-1978"><a href="#cb7-1978" aria-hidden="true" tabindex="-1"></a><span class="in">                        formula = formulas[</span></span>
<span id="cb7-1979"><a href="#cb7-1979" aria-hidden="true" tabindex="-1"></a><span class="in">                            grepl(</span></span>
<span id="cb7-1980"><a href="#cb7-1980" aria-hidden="true" tabindex="-1"></a><span class="in">                                paste(type_analysis, name_dataset, sep = "_"),</span></span>
<span id="cb7-1981"><a href="#cb7-1981" aria-hidden="true" tabindex="-1"></a><span class="in">                                names(formulas)</span></span>
<span id="cb7-1982"><a href="#cb7-1982" aria-hidden="true" tabindex="-1"></a><span class="in">                            )</span></span>
<span id="cb7-1983"><a href="#cb7-1983" aria-hidden="true" tabindex="-1"></a><span class="in">                        ],</span></span>
<span id="cb7-1984"><a href="#cb7-1984" aria-hidden="true" tabindex="-1"></a><span class="in">                        USE.NAMES = TRUE,</span></span>
<span id="cb7-1985"><a href="#cb7-1985" aria-hidden="true" tabindex="-1"></a><span class="in">                        simplify = FALSE</span></span>
<span id="cb7-1986"><a href="#cb7-1986" aria-hidden="true" tabindex="-1"></a><span class="in">                    )</span></span>
<span id="cb7-1987"><a href="#cb7-1987" aria-hidden="true" tabindex="-1"></a><span class="in">                }</span></span>
<span id="cb7-1988"><a href="#cb7-1988" aria-hidden="true" tabindex="-1"></a><span class="in">            },</span></span>
<span id="cb7-1989"><a href="#cb7-1989" aria-hidden="true" tabindex="-1"></a><span class="in">            dataset = datasets,</span></span>
<span id="cb7-1990"><a href="#cb7-1990" aria-hidden="true" tabindex="-1"></a><span class="in">            name_dataset = names(datasets),</span></span>
<span id="cb7-1991"><a href="#cb7-1991" aria-hidden="true" tabindex="-1"></a><span class="in">            MoreArgs = list(</span></span>
<span id="cb7-1992"><a href="#cb7-1992" aria-hidden="true" tabindex="-1"></a><span class="in">                formulas = formulas, name_scores = name_scores,</span></span>
<span id="cb7-1993"><a href="#cb7-1993" aria-hidden="true" tabindex="-1"></a><span class="in">                type_analysis = type_analysis</span></span>
<span id="cb7-1994"><a href="#cb7-1994" aria-hidden="true" tabindex="-1"></a><span class="in">            ),</span></span>
<span id="cb7-1995"><a href="#cb7-1995" aria-hidden="true" tabindex="-1"></a><span class="in">            USE.NAMES = TRUE,</span></span>
<span id="cb7-1996"><a href="#cb7-1996" aria-hidden="true" tabindex="-1"></a><span class="in">            SIMPLIFY = FALSE</span></span>
<span id="cb7-1997"><a href="#cb7-1997" aria-hidden="true" tabindex="-1"></a><span class="in">        )</span></span>
<span id="cb7-1998"><a href="#cb7-1998" aria-hidden="true" tabindex="-1"></a><span class="in">    },</span></span>
<span id="cb7-1999"><a href="#cb7-1999" aria-hidden="true" tabindex="-1"></a><span class="in">    datasets = lapply(</span></span>
<span id="cb7-2000"><a href="#cb7-2000" aria-hidden="true" tabindex="-1"></a><span class="in">        list(</span></span>
<span id="cb7-2001"><a href="#cb7-2001" aria-hidden="true" tabindex="-1"></a><span class="in">            "tcga" = tcga_samples, </span></span>
<span id="cb7-2002"><a href="#cb7-2002" aria-hidden="true" tabindex="-1"></a><span class="in">            "scanb" = scanb_samples, </span></span>
<span id="cb7-2003"><a href="#cb7-2003" aria-hidden="true" tabindex="-1"></a><span class="in">            "metabric" = metabric_samples</span></span>
<span id="cb7-2004"><a href="#cb7-2004" aria-hidden="true" tabindex="-1"></a><span class="in">        ),</span></span>
<span id="cb7-2005"><a href="#cb7-2005" aria-hidden="true" tabindex="-1"></a><span class="in">        function(x) df_pca_scores %&gt;% dplyr::filter(sample_name %in% x) </span></span>
<span id="cb7-2006"><a href="#cb7-2006" aria-hidden="true" tabindex="-1"></a><span class="in">    ),</span></span>
<span id="cb7-2007"><a href="#cb7-2007" aria-hidden="true" tabindex="-1"></a><span class="in">    name_scores = which_scores,</span></span>
<span id="cb7-2008"><a href="#cb7-2008" aria-hidden="true" tabindex="-1"></a><span class="in">    formulas = formulas_survival,</span></span>
<span id="cb7-2009"><a href="#cb7-2009" aria-hidden="true" tabindex="-1"></a><span class="in">    USE.NAMES = TRUE,</span></span>
<span id="cb7-2010"><a href="#cb7-2010" aria-hidden="true" tabindex="-1"></a><span class="in">    simplify = FALSE</span></span>
<span id="cb7-2011"><a href="#cb7-2011" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-2012"><a href="#cb7-2012" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2013"><a href="#cb7-2013" aria-hidden="true" tabindex="-1"></a><span class="in">survival_results$rfs &lt;- survival_results$rfs[</span></span>
<span id="cb7-2014"><a href="#cb7-2014" aria-hidden="true" tabindex="-1"></a><span class="in">    !sapply(survival_results$rfs, is.null)</span></span>
<span id="cb7-2015"><a href="#cb7-2015" aria-hidden="true" tabindex="-1"></a><span class="in">]</span></span>
<span id="cb7-2016"><a href="#cb7-2016" aria-hidden="true" tabindex="-1"></a><span class="in">survival_results$rfs$metabric &lt;- survival_results$rfs$metabric[</span></span>
<span id="cb7-2017"><a href="#cb7-2017" aria-hidden="true" tabindex="-1"></a><span class="in">    !sapply(survival_results$rfs$metabric, is.null)</span></span>
<span id="cb7-2018"><a href="#cb7-2018" aria-hidden="true" tabindex="-1"></a><span class="in">]</span></span>
<span id="cb7-2019"><a href="#cb7-2019" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2020"><a href="#cb7-2020" aria-hidden="true" tabindex="-1"></a><span class="in">survival_results$os &lt;- purrr::discard(</span></span>
<span id="cb7-2021"><a href="#cb7-2021" aria-hidden="true" tabindex="-1"></a><span class="in">    purrr::map(</span></span>
<span id="cb7-2022"><a href="#cb7-2022" aria-hidden="true" tabindex="-1"></a><span class="in">        survival_results$os, </span></span>
<span id="cb7-2023"><a href="#cb7-2023" aria-hidden="true" tabindex="-1"></a><span class="in">        ~ purrr::discard(.x, is.null),</span></span>
<span id="cb7-2024"><a href="#cb7-2024" aria-hidden="true" tabindex="-1"></a><span class="in">    ),</span></span>
<span id="cb7-2025"><a href="#cb7-2025" aria-hidden="true" tabindex="-1"></a><span class="in">    is.null</span></span>
<span id="cb7-2026"><a href="#cb7-2026" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-2027"><a href="#cb7-2027" aria-hidden="true" tabindex="-1"></a><span class="in"># we cannot simply save the models as each model has some </span></span>
<span id="cb7-2028"><a href="#cb7-2028" aria-hidden="true" tabindex="-1"></a><span class="in"># environment variables, which adds up to over 20GB. </span></span>
<span id="cb7-2029"><a href="#cb7-2029" aria-hidden="true" tabindex="-1"></a><span class="in"># check this stack exchange thread to read more on it:</span></span>
<span id="cb7-2030"><a href="#cb7-2030" aria-hidden="true" tabindex="-1"></a><span class="in"># https://stackoverflow.com/questions/42230920/saverds-inflating-size-of-object/52372480</span></span>
<span id="cb7-2031"><a href="#cb7-2031" aria-hidden="true" tabindex="-1"></a><span class="in"># the solution is to basically clear the environment from the terms </span></span>
<span id="cb7-2032"><a href="#cb7-2032" aria-hidden="true" tabindex="-1"></a><span class="in"># object. since it is pretty fast to run the survival analysis</span></span>
<span id="cb7-2033"><a href="#cb7-2033" aria-hidden="true" tabindex="-1"></a><span class="in"># we will not save the objects.</span></span>
<span id="cb7-2034"><a href="#cb7-2034" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2035"><a href="#cb7-2035" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2036"><a href="#cb7-2036" aria-hidden="true" tabindex="-1"></a><span class="in"># we now proceed to prepare the plots. since the survival result is </span></span>
<span id="cb7-2037"><a href="#cb7-2037" aria-hidden="true" tabindex="-1"></a><span class="in"># actually a large file, it is best to do all the plottings, save</span></span>
<span id="cb7-2038"><a href="#cb7-2038" aria-hidden="true" tabindex="-1"></a><span class="in"># them in other rds/pdf files and then use on the quarto markdown.</span></span>
<span id="cb7-2039"><a href="#cb7-2039" aria-hidden="true" tabindex="-1"></a><span class="in"># otherwise it is too slow to render the book. </span></span>
<span id="cb7-2040"><a href="#cb7-2040" aria-hidden="true" tabindex="-1"></a><span class="in"># the same is true for the statistics and coefficients available.</span></span>
<span id="cb7-2041"><a href="#cb7-2041" aria-hidden="true" tabindex="-1"></a><span class="in">names_signatures &lt;- c(</span></span>
<span id="cb7-2042"><a href="#cb7-2042" aria-hidden="true" tabindex="-1"></a><span class="in">    "HALLMARK_G2M_CHECKPOINT" = "G2M checkpoint",</span></span>
<span id="cb7-2043"><a href="#cb7-2043" aria-hidden="true" tabindex="-1"></a><span class="in">    "HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION" = "EMT",</span></span>
<span id="cb7-2044"><a href="#cb7-2044" aria-hidden="true" tabindex="-1"></a><span class="in">    "HALLMARK_DNA_REPAIR" = "DNA repair",</span></span>
<span id="cb7-2045"><a href="#cb7-2045" aria-hidden="true" tabindex="-1"></a><span class="in">    "HALLMARK_ANDROGEN_RESPONSE" = "Androgen response",</span></span>
<span id="cb7-2046"><a href="#cb7-2046" aria-hidden="true" tabindex="-1"></a><span class="in">    "HALLMARK_PI3K_AKT_MTOR_SIGNALING" = "PI3K AKT MTOR signaling",</span></span>
<span id="cb7-2047"><a href="#cb7-2047" aria-hidden="true" tabindex="-1"></a><span class="in">    "SET_ERPR" = "SET ER/PR",</span></span>
<span id="cb7-2048"><a href="#cb7-2048" aria-hidden="true" tabindex="-1"></a><span class="in">    "HALLMARK_ESTROGEN_RESPONSE_EARLY" = "Estrogen early"</span></span>
<span id="cb7-2049"><a href="#cb7-2049" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-2050"><a href="#cb7-2050" aria-hidden="true" tabindex="-1"></a><span class="in">names(names_signatures) &lt;- paste0(names(names_signatures), "_new")</span></span>
<span id="cb7-2051"><a href="#cb7-2051" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2052"><a href="#cb7-2052" aria-hidden="true" tabindex="-1"></a><span class="in">names_coefficients &lt;- list(</span></span>
<span id="cb7-2053"><a href="#cb7-2053" aria-hidden="true" tabindex="-1"></a><span class="in">    metabric = c(</span></span>
<span id="cb7-2054"><a href="#cb7-2054" aria-hidden="true" tabindex="-1"></a><span class="in">        "age" = "Age", </span></span>
<span id="cb7-2055"><a href="#cb7-2055" aria-hidden="true" tabindex="-1"></a><span class="in">        "npi" = "NPI", </span></span>
<span id="cb7-2056"><a href="#cb7-2056" aria-hidden="true" tabindex="-1"></a><span class="in">        names_signatures</span></span>
<span id="cb7-2057"><a href="#cb7-2057" aria-hidden="true" tabindex="-1"></a><span class="in">    ),</span></span>
<span id="cb7-2058"><a href="#cb7-2058" aria-hidden="true" tabindex="-1"></a><span class="in">    scanb = c(</span></span>
<span id="cb7-2059"><a href="#cb7-2059" aria-hidden="true" tabindex="-1"></a><span class="in">        "age" = "age",</span></span>
<span id="cb7-2060"><a href="#cb7-2060" aria-hidden="true" tabindex="-1"></a><span class="in">        names_signatures,</span></span>
<span id="cb7-2061"><a href="#cb7-2061" aria-hidden="true" tabindex="-1"></a><span class="in">        "node_stageN1" = "N1",</span></span>
<span id="cb7-2062"><a href="#cb7-2062" aria-hidden="true" tabindex="-1"></a><span class="in">        "node_stageN2and3" = "N2 and N3",</span></span>
<span id="cb7-2063"><a href="#cb7-2063" aria-hidden="true" tabindex="-1"></a><span class="in">        "tumor_stageT1" = "T1",</span></span>
<span id="cb7-2064"><a href="#cb7-2064" aria-hidden="true" tabindex="-1"></a><span class="in">        "tumor_stageT2" = "T2",</span></span>
<span id="cb7-2065"><a href="#cb7-2065" aria-hidden="true" tabindex="-1"></a><span class="in">        "tumor_stageT3" = "T3"</span></span>
<span id="cb7-2066"><a href="#cb7-2066" aria-hidden="true" tabindex="-1"></a><span class="in">    ),</span></span>
<span id="cb7-2067"><a href="#cb7-2067" aria-hidden="true" tabindex="-1"></a><span class="in">    tcga = c(</span></span>
<span id="cb7-2068"><a href="#cb7-2068" aria-hidden="true" tabindex="-1"></a><span class="in">        "age" = "age",</span></span>
<span id="cb7-2069"><a href="#cb7-2069" aria-hidden="true" tabindex="-1"></a><span class="in">        "tumor_stage" = "Tumor Stage",</span></span>
<span id="cb7-2070"><a href="#cb7-2070" aria-hidden="true" tabindex="-1"></a><span class="in">        names_signatures,</span></span>
<span id="cb7-2071"><a href="#cb7-2071" aria-hidden="true" tabindex="-1"></a><span class="in">        "node_stageN1" = "N1",</span></span>
<span id="cb7-2072"><a href="#cb7-2072" aria-hidden="true" tabindex="-1"></a><span class="in">        "node_stageN2" = "N2",</span></span>
<span id="cb7-2073"><a href="#cb7-2073" aria-hidden="true" tabindex="-1"></a><span class="in">        "node_stageN3" = "N3",</span></span>
<span id="cb7-2074"><a href="#cb7-2074" aria-hidden="true" tabindex="-1"></a><span class="in">        "tumor_stagestage_ii" = "T2",</span></span>
<span id="cb7-2075"><a href="#cb7-2075" aria-hidden="true" tabindex="-1"></a><span class="in">        "tumor_stagestage_iii" = "T3"</span></span>
<span id="cb7-2076"><a href="#cb7-2076" aria-hidden="true" tabindex="-1"></a><span class="in">    )</span></span>
<span id="cb7-2077"><a href="#cb7-2077" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-2078"><a href="#cb7-2078" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2079"><a href="#cb7-2079" aria-hidden="true" tabindex="-1"></a><span class="in">patients &lt;- list(</span></span>
<span id="cb7-2080"><a href="#cb7-2080" aria-hidden="true" tabindex="-1"></a><span class="in">    tcga = "Lum A/B, ER+ BC",</span></span>
<span id="cb7-2081"><a href="#cb7-2081" aria-hidden="true" tabindex="-1"></a><span class="in">    scanb = "Endo Only, ER+ BC",</span></span>
<span id="cb7-2082"><a href="#cb7-2082" aria-hidden="true" tabindex="-1"></a><span class="in">    metabric = "Endo Only, ER+ BC"</span></span>
<span id="cb7-2083"><a href="#cb7-2083" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-2084"><a href="#cb7-2084" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2085"><a href="#cb7-2085" aria-hidden="true" tabindex="-1"></a><span class="in">analysis_code &lt;- list(</span></span>
<span id="cb7-2086"><a href="#cb7-2086" aria-hidden="true" tabindex="-1"></a><span class="in">    os = "Overall survival",</span></span>
<span id="cb7-2087"><a href="#cb7-2087" aria-hidden="true" tabindex="-1"></a><span class="in">    rfs = "Recurrence free survival"</span></span>
<span id="cb7-2088"><a href="#cb7-2088" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-2089"><a href="#cb7-2089" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2090"><a href="#cb7-2090" aria-hidden="true" tabindex="-1"></a><span class="in">forest_plots &lt;- mapply(</span></span>
<span id="cb7-2091"><a href="#cb7-2091" aria-hidden="true" tabindex="-1"></a><span class="in">    function(fits, type_analysis, names_signatures){</span></span>
<span id="cb7-2092"><a href="#cb7-2092" aria-hidden="true" tabindex="-1"></a><span class="in">        mapply(</span></span>
<span id="cb7-2093"><a href="#cb7-2093" aria-hidden="true" tabindex="-1"></a><span class="in">            function(cohort_fits, name_cohort, type_analysis, names_signatures){</span></span>
<span id="cb7-2094"><a href="#cb7-2094" aria-hidden="true" tabindex="-1"></a><span class="in">                </span></span>
<span id="cb7-2095"><a href="#cb7-2095" aria-hidden="true" tabindex="-1"></a><span class="in">                tidy_survival &lt;- get_df_survival(</span></span>
<span id="cb7-2096"><a href="#cb7-2096" aria-hidden="true" tabindex="-1"></a><span class="in">                    cohort_fits,</span></span>
<span id="cb7-2097"><a href="#cb7-2097" aria-hidden="true" tabindex="-1"></a><span class="in">                    pathways_of_interest = names(names_signatures)</span></span>
<span id="cb7-2098"><a href="#cb7-2098" aria-hidden="true" tabindex="-1"></a><span class="in">                )</span></span>
<span id="cb7-2099"><a href="#cb7-2099" aria-hidden="true" tabindex="-1"></a><span class="in">                </span></span>
<span id="cb7-2100"><a href="#cb7-2100" aria-hidden="true" tabindex="-1"></a><span class="in">                plot_combined_scores(</span></span>
<span id="cb7-2101"><a href="#cb7-2101" aria-hidden="true" tabindex="-1"></a><span class="in">                    tidy_survival, </span></span>
<span id="cb7-2102"><a href="#cb7-2102" aria-hidden="true" tabindex="-1"></a><span class="in">                    range_min = 0.6, </span></span>
<span id="cb7-2103"><a href="#cb7-2103" aria-hidden="true" tabindex="-1"></a><span class="in">                    range_hr = 1.5,</span></span>
<span id="cb7-2104"><a href="#cb7-2104" aria-hidden="true" tabindex="-1"></a><span class="in">                    names_signatures = names_signatures</span></span>
<span id="cb7-2105"><a href="#cb7-2105" aria-hidden="true" tabindex="-1"></a><span class="in">                ) + </span></span>
<span id="cb7-2106"><a href="#cb7-2106" aria-hidden="true" tabindex="-1"></a><span class="in">                    ggplot2::labs(</span></span>
<span id="cb7-2107"><a href="#cb7-2107" aria-hidden="true" tabindex="-1"></a><span class="in">                        title = paste0(</span></span>
<span id="cb7-2108"><a href="#cb7-2108" aria-hidden="true" tabindex="-1"></a><span class="in">                            toupper(name_cohort), " ",</span></span>
<span id="cb7-2109"><a href="#cb7-2109" aria-hidden="true" tabindex="-1"></a><span class="in">                            patients[[name_cohort]], "\n",</span></span>
<span id="cb7-2110"><a href="#cb7-2110" aria-hidden="true" tabindex="-1"></a><span class="in">                            analysis_code[[type_analysis]]</span></span>
<span id="cb7-2111"><a href="#cb7-2111" aria-hidden="true" tabindex="-1"></a><span class="in">                        )</span></span>
<span id="cb7-2112"><a href="#cb7-2112" aria-hidden="true" tabindex="-1"></a><span class="in">                    )</span></span>
<span id="cb7-2113"><a href="#cb7-2113" aria-hidden="true" tabindex="-1"></a><span class="in">                </span></span>
<span id="cb7-2114"><a href="#cb7-2114" aria-hidden="true" tabindex="-1"></a><span class="in">            },</span></span>
<span id="cb7-2115"><a href="#cb7-2115" aria-hidden="true" tabindex="-1"></a><span class="in">            cohort_fits = fits,</span></span>
<span id="cb7-2116"><a href="#cb7-2116" aria-hidden="true" tabindex="-1"></a><span class="in">            name_cohort = names(fits),</span></span>
<span id="cb7-2117"><a href="#cb7-2117" aria-hidden="true" tabindex="-1"></a><span class="in">            MoreArgs = list(</span></span>
<span id="cb7-2118"><a href="#cb7-2118" aria-hidden="true" tabindex="-1"></a><span class="in">                type_analysis = type_analysis,</span></span>
<span id="cb7-2119"><a href="#cb7-2119" aria-hidden="true" tabindex="-1"></a><span class="in">                names_signatures = names_signatures</span></span>
<span id="cb7-2120"><a href="#cb7-2120" aria-hidden="true" tabindex="-1"></a><span class="in">            ),</span></span>
<span id="cb7-2121"><a href="#cb7-2121" aria-hidden="true" tabindex="-1"></a><span class="in">            USE.NAMES = TRUE,</span></span>
<span id="cb7-2122"><a href="#cb7-2122" aria-hidden="true" tabindex="-1"></a><span class="in">            SIMPLIFY = FALSE</span></span>
<span id="cb7-2123"><a href="#cb7-2123" aria-hidden="true" tabindex="-1"></a><span class="in">        )</span></span>
<span id="cb7-2124"><a href="#cb7-2124" aria-hidden="true" tabindex="-1"></a><span class="in">    },</span></span>
<span id="cb7-2125"><a href="#cb7-2125" aria-hidden="true" tabindex="-1"></a><span class="in">    survival_results,</span></span>
<span id="cb7-2126"><a href="#cb7-2126" aria-hidden="true" tabindex="-1"></a><span class="in">    names(survival_results),</span></span>
<span id="cb7-2127"><a href="#cb7-2127" aria-hidden="true" tabindex="-1"></a><span class="in">    MoreArgs = list(names_signatures = names_signatures),</span></span>
<span id="cb7-2128"><a href="#cb7-2128" aria-hidden="true" tabindex="-1"></a><span class="in">    USE.NAMES = TRUE,</span></span>
<span id="cb7-2129"><a href="#cb7-2129" aria-hidden="true" tabindex="-1"></a><span class="in">    SIMPLIFY = FALSE</span></span>
<span id="cb7-2130"><a href="#cb7-2130" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-2131"><a href="#cb7-2131" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2132"><a href="#cb7-2132" aria-hidden="true" tabindex="-1"></a><span class="in">saveRDS(</span></span>
<span id="cb7-2133"><a href="#cb7-2133" aria-hidden="true" tabindex="-1"></a><span class="in">    forest_plots,</span></span>
<span id="cb7-2134"><a href="#cb7-2134" aria-hidden="true" tabindex="-1"></a><span class="in">    "../../results/rds_files/expanding_ember/forest_plots.rds"</span></span>
<span id="cb7-2135"><a href="#cb7-2135" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-2136"><a href="#cb7-2136" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2137"><a href="#cb7-2137" aria-hidden="true" tabindex="-1"></a><span class="in"># and now we fetch the tables that will be used later on including the</span></span>
<span id="cb7-2138"><a href="#cb7-2138" aria-hidden="true" tabindex="-1"></a><span class="in"># confidence intervals and other parameters from the fit</span></span>
<span id="cb7-2139"><a href="#cb7-2139" aria-hidden="true" tabindex="-1"></a><span class="in">tables_survival &lt;- lapply(</span></span>
<span id="cb7-2140"><a href="#cb7-2140" aria-hidden="true" tabindex="-1"></a><span class="in">    survival_results, </span></span>
<span id="cb7-2141"><a href="#cb7-2141" aria-hidden="true" tabindex="-1"></a><span class="in">    function(x) </span></span>
<span id="cb7-2142"><a href="#cb7-2142" aria-hidden="true" tabindex="-1"></a><span class="in">        lapply(</span></span>
<span id="cb7-2143"><a href="#cb7-2143" aria-hidden="true" tabindex="-1"></a><span class="in">            x,</span></span>
<span id="cb7-2144"><a href="#cb7-2144" aria-hidden="true" tabindex="-1"></a><span class="in">            function(y)</span></span>
<span id="cb7-2145"><a href="#cb7-2145" aria-hidden="true" tabindex="-1"></a><span class="in">                lapply(</span></span>
<span id="cb7-2146"><a href="#cb7-2146" aria-hidden="true" tabindex="-1"></a><span class="in">                    y,</span></span>
<span id="cb7-2147"><a href="#cb7-2147" aria-hidden="true" tabindex="-1"></a><span class="in">                    function(z) broom::tidy(z)</span></span>
<span id="cb7-2148"><a href="#cb7-2148" aria-hidden="true" tabindex="-1"></a><span class="in">                ) %&gt;% dplyr::bind_rows(.id = "score")</span></span>
<span id="cb7-2149"><a href="#cb7-2149" aria-hidden="true" tabindex="-1"></a><span class="in">        ) %&gt;% dplyr::bind_rows(.id = "cohort")</span></span>
<span id="cb7-2150"><a href="#cb7-2150" aria-hidden="true" tabindex="-1"></a><span class="in">) %&gt;% dplyr::bind_rows(.id = "type_analysis") %&gt;%</span></span>
<span id="cb7-2151"><a href="#cb7-2151" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::mutate(HR = exp(estimate))</span></span>
<span id="cb7-2152"><a href="#cb7-2152" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2153"><a href="#cb7-2153" aria-hidden="true" tabindex="-1"></a><span class="in">write.csv(</span></span>
<span id="cb7-2154"><a href="#cb7-2154" aria-hidden="true" tabindex="-1"></a><span class="in">    tables_survival,</span></span>
<span id="cb7-2155"><a href="#cb7-2155" aria-hidden="true" tabindex="-1"></a><span class="in">    "../../results/tables/expanding_ember/survival_results.csv",</span></span>
<span id="cb7-2156"><a href="#cb7-2156" aria-hidden="true" tabindex="-1"></a><span class="in">    row.names = FALSE</span></span>
<span id="cb7-2157"><a href="#cb7-2157" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-2158"><a href="#cb7-2158" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-2159"><a href="#cb7-2159" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2160"><a href="#cb7-2160" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width = 9, fig.height=4, warning = FALSE}</span></span>
<span id="cb7-2161"><a href="#cb7-2161" aria-hidden="true" tabindex="-1"></a><span class="in">#| label: fig-pca-mol-land-survival-regressed</span></span>
<span id="cb7-2162"><a href="#cb7-2162" aria-hidden="true" tabindex="-1"></a><span class="in">#| fig-cap: Overall survival analysis from SCANB and METABRIC cohorts and their</span></span>
<span id="cb7-2163"><a href="#cb7-2163" aria-hidden="true" tabindex="-1"></a><span class="in">#|     scores. The formulas used were the same as for the estrogen</span></span>
<span id="cb7-2164"><a href="#cb7-2164" aria-hidden="true" tabindex="-1"></a><span class="in">#|     signaling analysis. Coefficients were scaled before performing</span></span>
<span id="cb7-2165"><a href="#cb7-2165" aria-hidden="true" tabindex="-1"></a><span class="in">#|     cox regression</span></span>
<span id="cb7-2166"><a href="#cb7-2166" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2167"><a href="#cb7-2167" aria-hidden="true" tabindex="-1"></a><span class="in">forest_plots &lt;- readRDS("../../results/rds_files/expanding_ember/forest_plots.rds")</span></span>
<span id="cb7-2168"><a href="#cb7-2168" aria-hidden="true" tabindex="-1"></a><span class="in"># we remove the TCGA plot so we have actually 4 plots and they are </span></span>
<span id="cb7-2169"><a href="#cb7-2169" aria-hidden="true" tabindex="-1"></a><span class="in"># ordered nicely. </span></span>
<span id="cb7-2170"><a href="#cb7-2170" aria-hidden="true" tabindex="-1"></a><span class="in">forest_plots$os_wo_tcga &lt;- forest_plots$os</span></span>
<span id="cb7-2171"><a href="#cb7-2171" aria-hidden="true" tabindex="-1"></a><span class="in">forest_plots$os_wo_tcga$tcga &lt;- NULL</span></span>
<span id="cb7-2172"><a href="#cb7-2172" aria-hidden="true" tabindex="-1"></a><span class="in">cowplot::plot_grid(plotlist = c(</span></span>
<span id="cb7-2173"><a href="#cb7-2173" aria-hidden="true" tabindex="-1"></a><span class="in">    forest_plots$os_wo_tcga,</span></span>
<span id="cb7-2174"><a href="#cb7-2174" aria-hidden="true" tabindex="-1"></a><span class="in">    forest_plots$rfs</span></span>
<span id="cb7-2175"><a href="#cb7-2175" aria-hidden="true" tabindex="-1"></a><span class="in">), nrow = 2)</span></span>
<span id="cb7-2176"><a href="#cb7-2176" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-2177"><a href="#cb7-2177" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2178"><a href="#cb7-2178" aria-hidden="true" tabindex="-1"></a>The plot with TCGA is shown below.</span>
<span id="cb7-2179"><a href="#cb7-2179" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2180"><a href="#cb7-2180" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=4.5, fig.height=10}</span></span>
<span id="cb7-2181"><a href="#cb7-2181" aria-hidden="true" tabindex="-1"></a><span class="in">cowplot::plot_grid(plotlist = c(</span></span>
<span id="cb7-2182"><a href="#cb7-2182" aria-hidden="true" tabindex="-1"></a><span class="in">    forest_plots$os,</span></span>
<span id="cb7-2183"><a href="#cb7-2183" aria-hidden="true" tabindex="-1"></a><span class="in">    forest_plots$rfs</span></span>
<span id="cb7-2184"><a href="#cb7-2184" aria-hidden="true" tabindex="-1"></a><span class="in">), ncol = 1)</span></span>
<span id="cb7-2185"><a href="#cb7-2185" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-2186"><a href="#cb7-2186" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2187"><a href="#cb7-2187" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2188"><a href="#cb7-2188" aria-hidden="true" tabindex="-1"></a>Scores for TCGA are noisier than SCANB and METABRIC. Though the</span>
<span id="cb7-2189"><a href="#cb7-2189" aria-hidden="true" tabindex="-1"></a>scaled scores they have comparable hazard ratios and also </span>
<span id="cb7-2190"><a href="#cb7-2190" aria-hidden="true" tabindex="-1"></a>they follow what we saw previously. The only pathway that is </span>
<span id="cb7-2191"><a href="#cb7-2191" aria-hidden="true" tabindex="-1"></a>a bit different is the PI3K AKT MTOR signaling. This pathway had a noisier</span>
<span id="cb7-2192"><a href="#cb7-2192" aria-hidden="true" tabindex="-1"></a>correlation with the GSVA score as well.</span>
<span id="cb7-2193"><a href="#cb7-2193" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2196"><a href="#cb7-2196" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb7-2197"><a href="#cb7-2197" aria-hidden="true" tabindex="-1"></a><span class="fu">read.csv</span>(<span class="st">"../../results/tables/expanding_ember/survival_results.csv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-2198"><a href="#cb7-2198" aria-hidden="true" tabindex="-1"></a>    DT<span class="sc">::</span><span class="fu">datatable</span>(<span class="at">options =</span> <span class="fu">list</span>(<span class="at">scrollX =</span> <span class="cn">TRUE</span>), <span class="at">filter =</span> <span class="st">'top'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb7-2199"><a href="#cb7-2199" aria-hidden="true" tabindex="-1"></a>    DT<span class="sc">::</span><span class="fu">formatRound</span>(</span>
<span id="cb7-2200"><a href="#cb7-2200" aria-hidden="true" tabindex="-1"></a>        <span class="at">columns=</span><span class="fu">c</span>(</span>
<span id="cb7-2201"><a href="#cb7-2201" aria-hidden="true" tabindex="-1"></a>            <span class="st">"estimate"</span>, <span class="st">"std.error"</span>, <span class="st">"statistic"</span>, <span class="st">"HR"</span></span>
<span id="cb7-2202"><a href="#cb7-2202" aria-hidden="true" tabindex="-1"></a>        ), </span>
<span id="cb7-2203"><a href="#cb7-2203" aria-hidden="true" tabindex="-1"></a>        <span class="at">digits =</span> <span class="dv">2</span></span>
<span id="cb7-2204"><a href="#cb7-2204" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span> </span>
<span id="cb7-2205"><a href="#cb7-2205" aria-hidden="true" tabindex="-1"></a>    DT<span class="sc">::</span><span class="fu">formatSignif</span>(<span class="at">columns =</span> <span class="fu">c</span>(<span class="st">"p.value"</span>))</span>
<span id="cb7-2206"><a href="#cb7-2206" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-2207"><a href="#cb7-2207" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2208"><a href="#cb7-2208" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2209"><a href="#cb7-2209" aria-hidden="true" tabindex="-1"></a><span class="fu">## Regressing the PDXs</span></span>
<span id="cb7-2210"><a href="#cb7-2210" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2211"><a href="#cb7-2211" aria-hidden="true" tabindex="-1"></a>We can use the regressing strategy to score the PDXs and then evaluate</span>
<span id="cb7-2212"><a href="#cb7-2212" aria-hidden="true" tabindex="-1"></a>their scores. </span>
<span id="cb7-2213"><a href="#cb7-2213" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2214"><a href="#cb7-2214" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb7-2215"><a href="#cb7-2215" aria-hidden="true" tabindex="-1"></a><span class="in"># first get the regressed data</span></span>
<span id="cb7-2216"><a href="#cb7-2216" aria-hidden="true" tabindex="-1"></a><span class="in">pdx_regressed &lt;- get_regressed_data(</span></span>
<span id="cb7-2217"><a href="#cb7-2217" aria-hidden="true" tabindex="-1"></a><span class="in">    sum_exp = pdx_tpm, </span></span>
<span id="cb7-2218"><a href="#cb7-2218" aria-hidden="true" tabindex="-1"></a><span class="in">    which_exp = "log2tpm",</span></span>
<span id="cb7-2219"><a href="#cb7-2219" aria-hidden="true" tabindex="-1"></a><span class="in">    pca_fit = pca_fit_all_genes_wo_outliers, </span></span>
<span id="cb7-2220"><a href="#cb7-2220" aria-hidden="true" tabindex="-1"></a><span class="in">    stable_genes = stable_genes</span></span>
<span id="cb7-2221"><a href="#cb7-2221" aria-hidden="true" tabindex="-1"></a><span class="in">) %&gt;%</span></span>
<span id="cb7-2222"><a href="#cb7-2222" aria-hidden="true" tabindex="-1"></a><span class="in">    # calculate the scores for the regressed data</span></span>
<span id="cb7-2223"><a href="#cb7-2223" aria-hidden="true" tabindex="-1"></a><span class="in">    get_scores_regressed(</span></span>
<span id="cb7-2224"><a href="#cb7-2224" aria-hidden="true" tabindex="-1"></a><span class="in">        sum_exp = .,</span></span>
<span id="cb7-2225"><a href="#cb7-2225" aria-hidden="true" tabindex="-1"></a><span class="in">        gene_sets = gene_sets_</span></span>
<span id="cb7-2226"><a href="#cb7-2226" aria-hidden="true" tabindex="-1"></a><span class="in">    )</span></span>
<span id="cb7-2227"><a href="#cb7-2227" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2228"><a href="#cb7-2228" aria-hidden="true" tabindex="-1"></a><span class="in">pdx_regressed$cohort &lt;- "pdx"</span></span>
<span id="cb7-2229"><a href="#cb7-2229" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2230"><a href="#cb7-2230" aria-hidden="true" tabindex="-1"></a><span class="in">saveRDS(</span></span>
<span id="cb7-2231"><a href="#cb7-2231" aria-hidden="true" tabindex="-1"></a><span class="in">    pdx_regressed,</span></span>
<span id="cb7-2232"><a href="#cb7-2232" aria-hidden="true" tabindex="-1"></a><span class="in">    "../../results/rds_files/expanding_ember/pdx_regressed.rds"</span></span>
<span id="cb7-2233"><a href="#cb7-2233" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-2234"><a href="#cb7-2234" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-2235"><a href="#cb7-2235" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2236"><a href="#cb7-2236" aria-hidden="true" tabindex="-1"></a>We start by comparing the scores across the different </span>
<span id="cb7-2237"><a href="#cb7-2237" aria-hidden="true" tabindex="-1"></a>control samples and their positions in the molecular landscape.</span>
<span id="cb7-2238"><a href="#cb7-2238" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2239"><a href="#cb7-2239" aria-hidden="true" tabindex="-1"></a><span class="in">```{r fig.width=9, fig.height=6}</span></span>
<span id="cb7-2240"><a href="#cb7-2240" aria-hidden="true" tabindex="-1"></a><span class="in">#| label: fig-pca-pdx-control-only-regressed</span></span>
<span id="cb7-2241"><a href="#cb7-2241" aria-hidden="true" tabindex="-1"></a><span class="in">#| fig-cap: Embedding of only regressed PDX control samples on top of the </span></span>
<span id="cb7-2242"><a href="#cb7-2242" aria-hidden="true" tabindex="-1"></a><span class="in">#|     METABRIC, SCANB and TCGA samples</span></span>
<span id="cb7-2243"><a href="#cb7-2243" aria-hidden="true" tabindex="-1"></a><span class="in">p &lt;- get_base_plot(</span></span>
<span id="cb7-2244"><a href="#cb7-2244" aria-hidden="true" tabindex="-1"></a><span class="in">    df_pca_coordinates_wo_outliers %&gt;% </span></span>
<span id="cb7-2245"><a href="#cb7-2245" aria-hidden="true" tabindex="-1"></a><span class="in">        dplyr::filter(cohort != "poetic")</span></span>
<span id="cb7-2246"><a href="#cb7-2246" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-2247"><a href="#cb7-2247" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2248"><a href="#cb7-2248" aria-hidden="true" tabindex="-1"></a><span class="in">pdx_df_pca_regressed_sub &lt;- colData(pdx_regressed) %&gt;%</span></span>
<span id="cb7-2249"><a href="#cb7-2249" aria-hidden="true" tabindex="-1"></a><span class="in">    data.frame %&gt;%</span></span>
<span id="cb7-2250"><a href="#cb7-2250" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::filter(cohort == "pdx" &amp; treatment == "CTRL") %&gt;%</span></span>
<span id="cb7-2251"><a href="#cb7-2251" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::mutate(pam50 = "nc")</span></span>
<span id="cb7-2252"><a href="#cb7-2252" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2253"><a href="#cb7-2253" aria-hidden="true" tabindex="-1"></a><span class="in">p +</span></span>
<span id="cb7-2254"><a href="#cb7-2254" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_point(</span></span>
<span id="cb7-2255"><a href="#cb7-2255" aria-hidden="true" tabindex="-1"></a><span class="in">        pdx_df_pca_regressed_sub,</span></span>
<span id="cb7-2256"><a href="#cb7-2256" aria-hidden="true" tabindex="-1"></a><span class="in">        mapping = aes(x = PC3, y = PC4, shape = pdx),</span></span>
<span id="cb7-2257"><a href="#cb7-2257" aria-hidden="true" tabindex="-1"></a><span class="in">        size = 5</span></span>
<span id="cb7-2258"><a href="#cb7-2258" aria-hidden="true" tabindex="-1"></a><span class="in">    ) + </span></span>
<span id="cb7-2259"><a href="#cb7-2259" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme_bw(base_size = 15) +</span></span>
<span id="cb7-2260"><a href="#cb7-2260" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::labs(</span></span>
<span id="cb7-2261"><a href="#cb7-2261" aria-hidden="true" tabindex="-1"></a><span class="in">        caption = "Grey points correspond to the PDX cohort",</span></span>
<span id="cb7-2262"><a href="#cb7-2262" aria-hidden="true" tabindex="-1"></a><span class="in">        title = "PDX cohort, only control samples"</span></span>
<span id="cb7-2263"><a href="#cb7-2263" aria-hidden="true" tabindex="-1"></a><span class="in">    ) + </span></span>
<span id="cb7-2264"><a href="#cb7-2264" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::scale_color_manual(</span></span>
<span id="cb7-2265"><a href="#cb7-2265" aria-hidden="true" tabindex="-1"></a><span class="in">        values = c(</span></span>
<span id="cb7-2266"><a href="#cb7-2266" aria-hidden="true" tabindex="-1"></a><span class="in">            get_colors_pam50(p$data),</span></span>
<span id="cb7-2267"><a href="#cb7-2267" aria-hidden="true" tabindex="-1"></a><span class="in">            "not available" = "grey"</span></span>
<span id="cb7-2268"><a href="#cb7-2268" aria-hidden="true" tabindex="-1"></a><span class="in">        )</span></span>
<span id="cb7-2269"><a href="#cb7-2269" aria-hidden="true" tabindex="-1"></a><span class="in">    ) +</span></span>
<span id="cb7-2270"><a href="#cb7-2270" aria-hidden="true" tabindex="-1"></a><span class="in">    change_plot_aes_point() +</span></span>
<span id="cb7-2271"><a href="#cb7-2271" aria-hidden="true" tabindex="-1"></a><span class="in">    change_guides_point(shape = TRUE)</span></span>
<span id="cb7-2272"><a href="#cb7-2272" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-2273"><a href="#cb7-2273" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2274"><a href="#cb7-2274" aria-hidden="true" tabindex="-1"></a>The results are jus twhat we got previously as well, meaning that</span>
<span id="cb7-2275"><a href="#cb7-2275" aria-hidden="true" tabindex="-1"></a>by using all these genes the information of the PDXs are still </span>
<span id="cb7-2276"><a href="#cb7-2276" aria-hidden="true" tabindex="-1"></a>captured. The main differences is that the luminal B region is more</span>
<span id="cb7-2277"><a href="#cb7-2277" aria-hidden="true" tabindex="-1"></a>compact compare to @fig-pca-pdx-control-only.</span>
<span id="cb7-2278"><a href="#cb7-2278" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2279"><a href="#cb7-2279" aria-hidden="true" tabindex="-1"></a>And now that we have the results we can compare the PDXs as well. </span>
<span id="cb7-2280"><a href="#cb7-2280" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2281"><a href="#cb7-2281" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=6}</span></span>
<span id="cb7-2282"><a href="#cb7-2282" aria-hidden="true" tabindex="-1"></a><span class="in">#| label: fig-pdx-regressed-scores</span></span>
<span id="cb7-2283"><a href="#cb7-2283" aria-hidden="true" tabindex="-1"></a><span class="in">#| fig-cap: Scores obtained from the regressed data of the PDXs.</span></span>
<span id="cb7-2284"><a href="#cb7-2284" aria-hidden="true" tabindex="-1"></a><span class="in">col_data_pdx_regressed &lt;- colData(pdx_regressed) %&gt;% data.frame</span></span>
<span id="cb7-2285"><a href="#cb7-2285" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2286"><a href="#cb7-2286" aria-hidden="true" tabindex="-1"></a><span class="in">pathways_pdx &lt;- paste0("regressed_", </span></span>
<span id="cb7-2287"><a href="#cb7-2287" aria-hidden="true" tabindex="-1"></a><span class="in">    c(paste0("HALLMARK_", c(</span></span>
<span id="cb7-2288"><a href="#cb7-2288" aria-hidden="true" tabindex="-1"></a><span class="in">        "ESTROGEN_RESPONSE_EARLY",</span></span>
<span id="cb7-2289"><a href="#cb7-2289" aria-hidden="true" tabindex="-1"></a><span class="in">        "G2M_CHECKPOINT",</span></span>
<span id="cb7-2290"><a href="#cb7-2290" aria-hidden="true" tabindex="-1"></a><span class="in">        "E2F_TARGETS",</span></span>
<span id="cb7-2291"><a href="#cb7-2291" aria-hidden="true" tabindex="-1"></a><span class="in">        "ANDROGEN_RESPONSE",</span></span>
<span id="cb7-2292"><a href="#cb7-2292" aria-hidden="true" tabindex="-1"></a><span class="in">        "APOPTOSIS",</span></span>
<span id="cb7-2293"><a href="#cb7-2293" aria-hidden="true" tabindex="-1"></a><span class="in">        "MTORC1_SIGNALING",</span></span>
<span id="cb7-2294"><a href="#cb7-2294" aria-hidden="true" tabindex="-1"></a><span class="in">        "EPITHELIAL_MESENCHYMAL_TRANSITION",</span></span>
<span id="cb7-2295"><a href="#cb7-2295" aria-hidden="true" tabindex="-1"></a><span class="in">        "DNA_REPAIR"</span></span>
<span id="cb7-2296"><a href="#cb7-2296" aria-hidden="true" tabindex="-1"></a><span class="in">    )),</span></span>
<span id="cb7-2297"><a href="#cb7-2297" aria-hidden="true" tabindex="-1"></a><span class="in">    "SET_ERPR",</span></span>
<span id="cb7-2298"><a href="#cb7-2298" aria-hidden="true" tabindex="-1"></a><span class="in">    "random_200")</span></span>
<span id="cb7-2299"><a href="#cb7-2299" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-2300"><a href="#cb7-2300" aria-hidden="true" tabindex="-1"></a><span class="in">labeller_names &lt;- c(</span></span>
<span id="cb7-2301"><a href="#cb7-2301" aria-hidden="true" tabindex="-1"></a><span class="in">    "Estrogen early",</span></span>
<span id="cb7-2302"><a href="#cb7-2302" aria-hidden="true" tabindex="-1"></a><span class="in">    "G2M",</span></span>
<span id="cb7-2303"><a href="#cb7-2303" aria-hidden="true" tabindex="-1"></a><span class="in">    "E2F",</span></span>
<span id="cb7-2304"><a href="#cb7-2304" aria-hidden="true" tabindex="-1"></a><span class="in">    "Androgen response",</span></span>
<span id="cb7-2305"><a href="#cb7-2305" aria-hidden="true" tabindex="-1"></a><span class="in">    "Apoptosis",</span></span>
<span id="cb7-2306"><a href="#cb7-2306" aria-hidden="true" tabindex="-1"></a><span class="in">    "MTORC1 signaling",</span></span>
<span id="cb7-2307"><a href="#cb7-2307" aria-hidden="true" tabindex="-1"></a><span class="in">    "EMT",</span></span>
<span id="cb7-2308"><a href="#cb7-2308" aria-hidden="true" tabindex="-1"></a><span class="in">    "DNA Repair",</span></span>
<span id="cb7-2309"><a href="#cb7-2309" aria-hidden="true" tabindex="-1"></a><span class="in">    "SET ER/PR",</span></span>
<span id="cb7-2310"><a href="#cb7-2310" aria-hidden="true" tabindex="-1"></a><span class="in">    "Random 200"</span></span>
<span id="cb7-2311"><a href="#cb7-2311" aria-hidden="true" tabindex="-1"></a><span class="in">) %&gt;%</span></span>
<span id="cb7-2312"><a href="#cb7-2312" aria-hidden="true" tabindex="-1"></a><span class="in">    `names&lt;-`(pathways_pdx)</span></span>
<span id="cb7-2313"><a href="#cb7-2313" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2314"><a href="#cb7-2314" aria-hidden="true" tabindex="-1"></a><span class="in">col_data_pdx_regressed %&gt;%</span></span>
<span id="cb7-2315"><a href="#cb7-2315" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::filter(treatment == "CTRL") %&gt;%</span></span>
<span id="cb7-2316"><a href="#cb7-2316" aria-hidden="true" tabindex="-1"></a><span class="in">    tidyr::pivot_longer(</span></span>
<span id="cb7-2317"><a href="#cb7-2317" aria-hidden="true" tabindex="-1"></a><span class="in">        cols = dplyr::all_of(pathways_pdx),</span></span>
<span id="cb7-2318"><a href="#cb7-2318" aria-hidden="true" tabindex="-1"></a><span class="in">        names_to = "pathway",</span></span>
<span id="cb7-2319"><a href="#cb7-2319" aria-hidden="true" tabindex="-1"></a><span class="in">        values_to = "score"</span></span>
<span id="cb7-2320"><a href="#cb7-2320" aria-hidden="true" tabindex="-1"></a><span class="in">    ) %&gt;%</span></span>
<span id="cb7-2321"><a href="#cb7-2321" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::ggplot(aes(y = pdx, x = score)) + </span></span>
<span id="cb7-2322"><a href="#cb7-2322" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_boxplot() +</span></span>
<span id="cb7-2323"><a href="#cb7-2323" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_jitter() + </span></span>
<span id="cb7-2324"><a href="#cb7-2324" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::labs(x = "Regressed score", y = "PDX") +</span></span>
<span id="cb7-2325"><a href="#cb7-2325" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::facet_wrap(~pathway, labeller = as_labeller(labeller_names)) + </span></span>
<span id="cb7-2326"><a href="#cb7-2326" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme_bw(base_size = 15)</span></span>
<span id="cb7-2327"><a href="#cb7-2327" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-2328"><a href="#cb7-2328" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2329"><a href="#cb7-2329" aria-hidden="true" tabindex="-1"></a>The random 200 is just a gene set with 200 random genes that works as a </span>
<span id="cb7-2330"><a href="#cb7-2330" aria-hidden="true" tabindex="-1"></a>control and serves as a comparison to the other pathways.</span>
<span id="cb7-2331"><a href="#cb7-2331" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2332"><a href="#cb7-2332" aria-hidden="true" tabindex="-1"></a>And below we show the distribution of some of the scores calculated </span>
<span id="cb7-2333"><a href="#cb7-2333" aria-hidden="true" tabindex="-1"></a>for all the other cohorts as a comparison. </span>
<span id="cb7-2334"><a href="#cb7-2334" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2335"><a href="#cb7-2335" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=8, fig.height=5}</span></span>
<span id="cb7-2336"><a href="#cb7-2336" aria-hidden="true" tabindex="-1"></a><span class="in">scores_wide &lt;- tidyr::pivot_wider(</span></span>
<span id="cb7-2337"><a href="#cb7-2337" aria-hidden="true" tabindex="-1"></a><span class="in">    scores,</span></span>
<span id="cb7-2338"><a href="#cb7-2338" aria-hidden="true" tabindex="-1"></a><span class="in">    id_cols = c("sample_name", "cohort"),</span></span>
<span id="cb7-2339"><a href="#cb7-2339" aria-hidden="true" tabindex="-1"></a><span class="in">    values_from = "score_hkg",</span></span>
<span id="cb7-2340"><a href="#cb7-2340" aria-hidden="true" tabindex="-1"></a><span class="in">    names_from = "pathway"</span></span>
<span id="cb7-2341"><a href="#cb7-2341" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-2342"><a href="#cb7-2342" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2343"><a href="#cb7-2343" aria-hidden="true" tabindex="-1"></a><span class="in">pathways_density &lt;- stringr::str_split(</span></span>
<span id="cb7-2344"><a href="#cb7-2344" aria-hidden="true" tabindex="-1"></a><span class="in">    pathways_pdx, </span></span>
<span id="cb7-2345"><a href="#cb7-2345" aria-hidden="true" tabindex="-1"></a><span class="in">    pattern = "_",</span></span>
<span id="cb7-2346"><a href="#cb7-2346" aria-hidden="true" tabindex="-1"></a><span class="in">    n = 2, </span></span>
<span id="cb7-2347"><a href="#cb7-2347" aria-hidden="true" tabindex="-1"></a><span class="in">    simplify = TRUE</span></span>
<span id="cb7-2348"><a href="#cb7-2348" aria-hidden="true" tabindex="-1"></a><span class="in">)[, 2]</span></span>
<span id="cb7-2349"><a href="#cb7-2349" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2350"><a href="#cb7-2350" aria-hidden="true" tabindex="-1"></a><span class="in">scores %&gt;%</span></span>
<span id="cb7-2351"><a href="#cb7-2351" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::filter(cohort != "poetic" &amp; pathway %in% pathways_density) %&gt;%</span></span>
<span id="cb7-2352"><a href="#cb7-2352" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::ggplot(aes(x=score_hkg, fill = cohort)) +</span></span>
<span id="cb7-2353"><a href="#cb7-2353" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_density(alpha = 0.5) +</span></span>
<span id="cb7-2354"><a href="#cb7-2354" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::facet_wrap(~pathway, scales = "free") +</span></span>
<span id="cb7-2355"><a href="#cb7-2355" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme_bw()</span></span>
<span id="cb7-2356"><a href="#cb7-2356" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-2357"><a href="#cb7-2357" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2358"><a href="#cb7-2358" aria-hidden="true" tabindex="-1"></a>We see that METS15 is the one with the highest proliferation and</span>
<span id="cb7-2359"><a href="#cb7-2359" aria-hidden="true" tabindex="-1"></a>lowest EMT, it is a PDX that comes from a pleural efusion. Also</span>
<span id="cb7-2360"><a href="#cb7-2360" aria-hidden="true" tabindex="-1"></a>it is the PDX with the lowest apoptosis and highest DNA repair,</span>
<span id="cb7-2361"><a href="#cb7-2361" aria-hidden="true" tabindex="-1"></a>probably due to a higher replication rate. Moreover, all these</span>
<span id="cb7-2362"><a href="#cb7-2362" aria-hidden="true" tabindex="-1"></a>PDXs they have a estrogen response early scores higher than 0</span>
<span id="cb7-2363"><a href="#cb7-2363" aria-hidden="true" tabindex="-1"></a>in average, which is about the average of the scores in the</span>
<span id="cb7-2364"><a href="#cb7-2364" aria-hidden="true" tabindex="-1"></a>ER+ BC samples.</span>
<span id="cb7-2365"><a href="#cb7-2365" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2366"><a href="#cb7-2366" aria-hidden="true" tabindex="-1"></a>Now we check what happens with some of the PDXs when they</span>
<span id="cb7-2367"><a href="#cb7-2367" aria-hidden="true" tabindex="-1"></a>are treated with P4, to compare with @fig-pdx-p4-control-pathways.</span>
<span id="cb7-2368"><a href="#cb7-2368" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2369"><a href="#cb7-2369" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=13, fig.height=15}</span></span>
<span id="cb7-2370"><a href="#cb7-2370" aria-hidden="true" tabindex="-1"></a><span class="in">#| label: fig-pdx-p4-control-pathways-regressed.</span></span>
<span id="cb7-2371"><a href="#cb7-2371" aria-hidden="true" tabindex="-1"></a><span class="in">#| fig-cap: Comparison of position in the molecular landscape between</span></span>
<span id="cb7-2372"><a href="#cb7-2372" aria-hidden="true" tabindex="-1"></a><span class="in">#|     CTRL, E2 and P4 treated samples with the regressed data</span></span>
<span id="cb7-2373"><a href="#cb7-2373" aria-hidden="true" tabindex="-1"></a><span class="in">pathways &lt;- c(</span></span>
<span id="cb7-2374"><a href="#cb7-2374" aria-hidden="true" tabindex="-1"></a><span class="in">    #"HALLMARK_ANDROGEN_RESPONSE" = "Androgen Response",</span></span>
<span id="cb7-2375"><a href="#cb7-2375" aria-hidden="true" tabindex="-1"></a><span class="in">    #"HALLMARK_E2F_TARGETS" = "E2F",</span></span>
<span id="cb7-2376"><a href="#cb7-2376" aria-hidden="true" tabindex="-1"></a><span class="in">    "HALLMARK_G2M_CHECKPOINT" = "G2M",</span></span>
<span id="cb7-2377"><a href="#cb7-2377" aria-hidden="true" tabindex="-1"></a><span class="in">    "HALLMARK_ESTROGEN_RESPONSE_EARLY" = "Estrogen Early",</span></span>
<span id="cb7-2378"><a href="#cb7-2378" aria-hidden="true" tabindex="-1"></a><span class="in">    #"SET_ERPR" = "SET ER/PR",</span></span>
<span id="cb7-2379"><a href="#cb7-2379" aria-hidden="true" tabindex="-1"></a><span class="in">    "HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION" = "EMT",</span></span>
<span id="cb7-2380"><a href="#cb7-2380" aria-hidden="true" tabindex="-1"></a><span class="in">    "GOBP_RESPONSE_TO_PROGESTERONE" = "GO Progesterone",</span></span>
<span id="cb7-2381"><a href="#cb7-2381" aria-hidden="true" tabindex="-1"></a><span class="in">    "WILCOX_RESPONSE_TO_PROGESTERONE_UP" = "W. Prog UP"</span></span>
<span id="cb7-2382"><a href="#cb7-2382" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-2383"><a href="#cb7-2383" aria-hidden="true" tabindex="-1"></a><span class="in">names(pathways) &lt;- paste0("regressed_", names(pathways))</span></span>
<span id="cb7-2384"><a href="#cb7-2384" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2385"><a href="#cb7-2385" aria-hidden="true" tabindex="-1"></a><span class="in">pdxs_to_use &lt;- c("METS15", "T113", "T111", "T105", "T109", "T110")</span></span>
<span id="cb7-2386"><a href="#cb7-2386" aria-hidden="true" tabindex="-1"></a><span class="in">names(pdxs_to_use) &lt;- pdxs_to_use</span></span>
<span id="cb7-2387"><a href="#cb7-2387" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2388"><a href="#cb7-2388" aria-hidden="true" tabindex="-1"></a><span class="in">pdx_df_pca_sub &lt;- col_data_pdx_regressed %&gt;%</span></span>
<span id="cb7-2389"><a href="#cb7-2389" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::filter(</span></span>
<span id="cb7-2390"><a href="#cb7-2390" aria-hidden="true" tabindex="-1"></a><span class="in">        cohort == "pdx" &amp; </span></span>
<span id="cb7-2391"><a href="#cb7-2391" aria-hidden="true" tabindex="-1"></a><span class="in">            treatment %in% c("CTRL", "P4", "E2")</span></span>
<span id="cb7-2392"><a href="#cb7-2392" aria-hidden="true" tabindex="-1"></a><span class="in">    ) %&gt;%</span></span>
<span id="cb7-2393"><a href="#cb7-2393" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::mutate(pam50 = "nc") %&gt;%</span></span>
<span id="cb7-2394"><a href="#cb7-2394" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::mutate(pdx = factor(pdx, levels = c(</span></span>
<span id="cb7-2395"><a href="#cb7-2395" aria-hidden="true" tabindex="-1"></a><span class="in">        "METS15", "T113", "T109",</span></span>
<span id="cb7-2396"><a href="#cb7-2396" aria-hidden="true" tabindex="-1"></a><span class="in">        "T105", "T111", "T110"</span></span>
<span id="cb7-2397"><a href="#cb7-2397" aria-hidden="true" tabindex="-1"></a><span class="in">    ))) %&gt;% </span></span>
<span id="cb7-2398"><a href="#cb7-2398" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::filter(pdx %in% pdxs_to_use)</span></span>
<span id="cb7-2399"><a href="#cb7-2399" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2400"><a href="#cb7-2400" aria-hidden="true" tabindex="-1"></a><span class="in">plots_pathways &lt;- mapply(</span></span>
<span id="cb7-2401"><a href="#cb7-2401" aria-hidden="true" tabindex="-1"></a><span class="in">    function(pathways, pathway_names, i){</span></span>
<span id="cb7-2402"><a href="#cb7-2402" aria-hidden="true" tabindex="-1"></a><span class="in">        names(pathways) &lt;- pathway_names </span></span>
<span id="cb7-2403"><a href="#cb7-2403" aria-hidden="true" tabindex="-1"></a><span class="in">        pdx_df_pca_sub %&gt;%</span></span>
<span id="cb7-2404"><a href="#cb7-2404" aria-hidden="true" tabindex="-1"></a><span class="in">            tidyr::pivot_longer(</span></span>
<span id="cb7-2405"><a href="#cb7-2405" aria-hidden="true" tabindex="-1"></a><span class="in">                cols = dplyr::all_of(pathway_names),</span></span>
<span id="cb7-2406"><a href="#cb7-2406" aria-hidden="true" tabindex="-1"></a><span class="in">                names_to = "pathway",</span></span>
<span id="cb7-2407"><a href="#cb7-2407" aria-hidden="true" tabindex="-1"></a><span class="in">                values_to = "score"</span></span>
<span id="cb7-2408"><a href="#cb7-2408" aria-hidden="true" tabindex="-1"></a><span class="in">            ) %&gt;%</span></span>
<span id="cb7-2409"><a href="#cb7-2409" aria-hidden="true" tabindex="-1"></a><span class="in">            dplyr::mutate(pathway = factor(pathway, levels = pathway_names)) %&gt;%</span></span>
<span id="cb7-2410"><a href="#cb7-2410" aria-hidden="true" tabindex="-1"></a><span class="in">            ggplot2::ggplot(aes(x = treatment, y = score)) +</span></span>
<span id="cb7-2411"><a href="#cb7-2411" aria-hidden="true" tabindex="-1"></a><span class="in">            ggplot2::geom_boxplot(outlier.shape = NA) + </span></span>
<span id="cb7-2412"><a href="#cb7-2412" aria-hidden="true" tabindex="-1"></a><span class="in">            ggplot2::geom_jitter(</span></span>
<span id="cb7-2413"><a href="#cb7-2413" aria-hidden="true" tabindex="-1"></a><span class="in">                size = 3, </span></span>
<span id="cb7-2414"><a href="#cb7-2414" aria-hidden="true" tabindex="-1"></a><span class="in">                alpha = 0.9, </span></span>
<span id="cb7-2415"><a href="#cb7-2415" aria-hidden="true" tabindex="-1"></a><span class="in">                mapping = aes(color = treatment)</span></span>
<span id="cb7-2416"><a href="#cb7-2416" aria-hidden="true" tabindex="-1"></a><span class="in">            ) + </span></span>
<span id="cb7-2417"><a href="#cb7-2417" aria-hidden="true" tabindex="-1"></a><span class="in">            ggplot2::scale_color_viridis_d(option = "H") +</span></span>
<span id="cb7-2418"><a href="#cb7-2418" aria-hidden="true" tabindex="-1"></a><span class="in">            ggplot2::facet_wrap(</span></span>
<span id="cb7-2419"><a href="#cb7-2419" aria-hidden="true" tabindex="-1"></a><span class="in">                ~ pdx + pathway, </span></span>
<span id="cb7-2420"><a href="#cb7-2420" aria-hidden="true" tabindex="-1"></a><span class="in">                scales = "fixed", </span></span>
<span id="cb7-2421"><a href="#cb7-2421" aria-hidden="true" tabindex="-1"></a><span class="in">                nrow = length(pathway_names),</span></span>
<span id="cb7-2422"><a href="#cb7-2422" aria-hidden="true" tabindex="-1"></a><span class="in">                labeller = as_labeller(c(pathways, pdxs_to_use))</span></span>
<span id="cb7-2423"><a href="#cb7-2423" aria-hidden="true" tabindex="-1"></a><span class="in">            ) +</span></span>
<span id="cb7-2424"><a href="#cb7-2424" aria-hidden="true" tabindex="-1"></a><span class="in">            ggplot2::labs(</span></span>
<span id="cb7-2425"><a href="#cb7-2425" aria-hidden="true" tabindex="-1"></a><span class="in">                x = ifelse(i == 4, "Treatment", ""),</span></span>
<span id="cb7-2426"><a href="#cb7-2426" aria-hidden="true" tabindex="-1"></a><span class="in">                y = "Regressed scores"</span></span>
<span id="cb7-2427"><a href="#cb7-2427" aria-hidden="true" tabindex="-1"></a><span class="in">            ) +</span></span>
<span id="cb7-2428"><a href="#cb7-2428" aria-hidden="true" tabindex="-1"></a><span class="in">            ggplot2::scale_y_continuous(labels = scales::number_format(</span></span>
<span id="cb7-2429"><a href="#cb7-2429" aria-hidden="true" tabindex="-1"></a><span class="in">                accuracy = 0.01,</span></span>
<span id="cb7-2430"><a href="#cb7-2430" aria-hidden="true" tabindex="-1"></a><span class="in">                decimal.mark = '.'</span></span>
<span id="cb7-2431"><a href="#cb7-2431" aria-hidden="true" tabindex="-1"></a><span class="in">            )) + </span></span>
<span id="cb7-2432"><a href="#cb7-2432" aria-hidden="true" tabindex="-1"></a><span class="in">            ggplot2::theme_bw(base_size = 20) +</span></span>
<span id="cb7-2433"><a href="#cb7-2433" aria-hidden="true" tabindex="-1"></a><span class="in">            ggplot2::theme(legend.position="none") +</span></span>
<span id="cb7-2434"><a href="#cb7-2434" aria-hidden="true" tabindex="-1"></a><span class="in">            change_guides_point() +</span></span>
<span id="cb7-2435"><a href="#cb7-2435" aria-hidden="true" tabindex="-1"></a><span class="in">            change_plot_aes_point()</span></span>
<span id="cb7-2436"><a href="#cb7-2436" aria-hidden="true" tabindex="-1"></a><span class="in">    },</span></span>
<span id="cb7-2437"><a href="#cb7-2437" aria-hidden="true" tabindex="-1"></a><span class="in">    pathways = pathways,</span></span>
<span id="cb7-2438"><a href="#cb7-2438" aria-hidden="true" tabindex="-1"></a><span class="in">    pathway_name = names(pathways),</span></span>
<span id="cb7-2439"><a href="#cb7-2439" aria-hidden="true" tabindex="-1"></a><span class="in">    i = 1:length(pathways),</span></span>
<span id="cb7-2440"><a href="#cb7-2440" aria-hidden="true" tabindex="-1"></a><span class="in">    SIMPLIFY = FALSE,</span></span>
<span id="cb7-2441"><a href="#cb7-2441" aria-hidden="true" tabindex="-1"></a><span class="in">    USE.NAMES = TRUE</span></span>
<span id="cb7-2442"><a href="#cb7-2442" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-2443"><a href="#cb7-2443" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2444"><a href="#cb7-2444" aria-hidden="true" tabindex="-1"></a><span class="in">cowplot::plot_grid(</span></span>
<span id="cb7-2445"><a href="#cb7-2445" aria-hidden="true" tabindex="-1"></a><span class="in">    plotlist = plots_pathways,</span></span>
<span id="cb7-2446"><a href="#cb7-2446" aria-hidden="true" tabindex="-1"></a><span class="in">    nrow = length(plots_pathways) </span></span>
<span id="cb7-2447"><a href="#cb7-2447" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-2448"><a href="#cb7-2448" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-2449"><a href="#cb7-2449" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2450"><a href="#cb7-2450" aria-hidden="true" tabindex="-1"></a>The results are very similar to what was seen previously using the GSVA. What is </span>
<span id="cb7-2451"><a href="#cb7-2451" aria-hidden="true" tabindex="-1"></a>interesting here is that we have an absolute scale, so we can compare the scores.</span>
<span id="cb7-2452"><a href="#cb7-2452" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2453"><a href="#cb7-2453" aria-hidden="true" tabindex="-1"></a><span class="fu">## singscore</span></span>
<span id="cb7-2454"><a href="#cb7-2454" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2455"><a href="#cb7-2455" aria-hidden="true" tabindex="-1"></a>We can try to use singscore instead, it is a sample dependent measure so </span>
<span id="cb7-2456"><a href="#cb7-2456" aria-hidden="true" tabindex="-1"></a>it does not estimate the distribution of the gene expression levels before</span>
<span id="cb7-2457"><a href="#cb7-2457" aria-hidden="true" tabindex="-1"></a>calculating the enrichment score.</span>
<span id="cb7-2458"><a href="#cb7-2458" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2459"><a href="#cb7-2459" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb7-2460"><a href="#cb7-2460" aria-hidden="true" tabindex="-1"></a><span class="in">gene_sets_names &lt;- setdiff(names(gene_sets_), p4_pathways$gs_name %&gt;% unique)</span></span>
<span id="cb7-2461"><a href="#cb7-2461" aria-hidden="true" tabindex="-1"></a><span class="in">gene_sets_gsea &lt;- mapply(</span></span>
<span id="cb7-2462"><a href="#cb7-2462" aria-hidden="true" tabindex="-1"></a><span class="in">    function(gene_set, name_gs){</span></span>
<span id="cb7-2463"><a href="#cb7-2463" aria-hidden="true" tabindex="-1"></a><span class="in">        GSEABase::GeneSet(</span></span>
<span id="cb7-2464"><a href="#cb7-2464" aria-hidden="true" tabindex="-1"></a><span class="in">            gene_set %&gt;% unique,</span></span>
<span id="cb7-2465"><a href="#cb7-2465" aria-hidden="true" tabindex="-1"></a><span class="in">            setName = name_gs</span></span>
<span id="cb7-2466"><a href="#cb7-2466" aria-hidden="true" tabindex="-1"></a><span class="in">        )</span></span>
<span id="cb7-2467"><a href="#cb7-2467" aria-hidden="true" tabindex="-1"></a><span class="in">    },</span></span>
<span id="cb7-2468"><a href="#cb7-2468" aria-hidden="true" tabindex="-1"></a><span class="in">    gene_set = gene_sets_[gene_sets_names],</span></span>
<span id="cb7-2469"><a href="#cb7-2469" aria-hidden="true" tabindex="-1"></a><span class="in">    name_gs = gene_sets_names</span></span>
<span id="cb7-2470"><a href="#cb7-2470" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-2471"><a href="#cb7-2471" aria-hidden="true" tabindex="-1"></a><span class="in">        </span></span>
<span id="cb7-2472"><a href="#cb7-2472" aria-hidden="true" tabindex="-1"></a><span class="in">singscore_dfs &lt;- mapply(</span></span>
<span id="cb7-2473"><a href="#cb7-2473" aria-hidden="true" tabindex="-1"></a><span class="in">    function(sum_exp, gene_sets, which_exp){</span></span>
<span id="cb7-2474"><a href="#cb7-2474" aria-hidden="true" tabindex="-1"></a><span class="in">        singscore::multiScore(</span></span>
<span id="cb7-2475"><a href="#cb7-2475" aria-hidden="true" tabindex="-1"></a><span class="in">            singscore::rankGenes(assay(sum_exp,which_exp)),</span></span>
<span id="cb7-2476"><a href="#cb7-2476" aria-hidden="true" tabindex="-1"></a><span class="in">            upSetColc = gene_sets</span></span>
<span id="cb7-2477"><a href="#cb7-2477" aria-hidden="true" tabindex="-1"></a><span class="in">        )[[1]]</span></span>
<span id="cb7-2478"><a href="#cb7-2478" aria-hidden="true" tabindex="-1"></a><span class="in">    },</span></span>
<span id="cb7-2479"><a href="#cb7-2479" aria-hidden="true" tabindex="-1"></a><span class="in">    sum_exp = datasets,</span></span>
<span id="cb7-2480"><a href="#cb7-2480" aria-hidden="true" tabindex="-1"></a><span class="in">    which_exp = c(which_exp, "poetic" = "normalized_intensity"),</span></span>
<span id="cb7-2481"><a href="#cb7-2481" aria-hidden="true" tabindex="-1"></a><span class="in">    MoreArgs = list(gene_sets = gene_sets_gsea)</span></span>
<span id="cb7-2482"><a href="#cb7-2482" aria-hidden="true" tabindex="-1"></a><span class="in">) %&gt;% Reduce(cbind, .)</span></span>
<span id="cb7-2483"><a href="#cb7-2483" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2484"><a href="#cb7-2484" aria-hidden="true" tabindex="-1"></a><span class="in">saveRDS(</span></span>
<span id="cb7-2485"><a href="#cb7-2485" aria-hidden="true" tabindex="-1"></a><span class="in">    singscore_dfs,</span></span>
<span id="cb7-2486"><a href="#cb7-2486" aria-hidden="true" tabindex="-1"></a><span class="in">    "../../results/rds_files/expanding_ember/singscore_dfs.rds"</span></span>
<span id="cb7-2487"><a href="#cb7-2487" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-2488"><a href="#cb7-2488" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2489"><a href="#cb7-2489" aria-hidden="true" tabindex="-1"></a><span class="in">saveRDS(</span></span>
<span id="cb7-2490"><a href="#cb7-2490" aria-hidden="true" tabindex="-1"></a><span class="in">    gene_sets_gsea,</span></span>
<span id="cb7-2491"><a href="#cb7-2491" aria-hidden="true" tabindex="-1"></a><span class="in">    "../../results/rds_files/expanding_ember/gene_sets_gsea.rds"</span></span>
<span id="cb7-2492"><a href="#cb7-2492" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-2493"><a href="#cb7-2493" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-2494"><a href="#cb7-2494" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2495"><a href="#cb7-2495" aria-hidden="true" tabindex="-1"></a>We start by comparing with the GSVA scores.</span>
<span id="cb7-2496"><a href="#cb7-2496" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2497"><a href="#cb7-2497" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=20, fig.height=16, dev='png'}</span></span>
<span id="cb7-2498"><a href="#cb7-2498" aria-hidden="true" tabindex="-1"></a><span class="in">#| label: fig-all-pathways-gsva-singscores</span></span>
<span id="cb7-2499"><a href="#cb7-2499" aria-hidden="true" tabindex="-1"></a><span class="in">df_all_scores &lt;- singscore_dfs %&gt;% </span></span>
<span id="cb7-2500"><a href="#cb7-2500" aria-hidden="true" tabindex="-1"></a><span class="in">    t %&gt;%</span></span>
<span id="cb7-2501"><a href="#cb7-2501" aria-hidden="true" tabindex="-1"></a><span class="in">    data.frame %&gt;%</span></span>
<span id="cb7-2502"><a href="#cb7-2502" aria-hidden="true" tabindex="-1"></a><span class="in">    tibble::rownames_to_column(var = "sample_name") %&gt;%</span></span>
<span id="cb7-2503"><a href="#cb7-2503" aria-hidden="true" tabindex="-1"></a><span class="in">    tidyr::pivot_longer(</span></span>
<span id="cb7-2504"><a href="#cb7-2504" aria-hidden="true" tabindex="-1"></a><span class="in">        cols = dplyr::all_of(names(gene_sets_gsea)),</span></span>
<span id="cb7-2505"><a href="#cb7-2505" aria-hidden="true" tabindex="-1"></a><span class="in">        names_to = "pathway",</span></span>
<span id="cb7-2506"><a href="#cb7-2506" aria-hidden="true" tabindex="-1"></a><span class="in">        values_to = "singscore"</span></span>
<span id="cb7-2507"><a href="#cb7-2507" aria-hidden="true" tabindex="-1"></a><span class="in">    ) %&gt;%</span></span>
<span id="cb7-2508"><a href="#cb7-2508" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::inner_join(</span></span>
<span id="cb7-2509"><a href="#cb7-2509" aria-hidden="true" tabindex="-1"></a><span class="in">        merged_col_data %&gt;%</span></span>
<span id="cb7-2510"><a href="#cb7-2510" aria-hidden="true" tabindex="-1"></a><span class="in">            tidyr::pivot_longer(</span></span>
<span id="cb7-2511"><a href="#cb7-2511" aria-hidden="true" tabindex="-1"></a><span class="in">                cols = dplyr::all_of(names(gene_sets_gsea)),</span></span>
<span id="cb7-2512"><a href="#cb7-2512" aria-hidden="true" tabindex="-1"></a><span class="in">                names_to = "pathway",</span></span>
<span id="cb7-2513"><a href="#cb7-2513" aria-hidden="true" tabindex="-1"></a><span class="in">                values_to = "gsva"</span></span>
<span id="cb7-2514"><a href="#cb7-2514" aria-hidden="true" tabindex="-1"></a><span class="in">            ),</span></span>
<span id="cb7-2515"><a href="#cb7-2515" aria-hidden="true" tabindex="-1"></a><span class="in">        .,</span></span>
<span id="cb7-2516"><a href="#cb7-2516" aria-hidden="true" tabindex="-1"></a><span class="in">        by = c("sample_name", "pathway")</span></span>
<span id="cb7-2517"><a href="#cb7-2517" aria-hidden="true" tabindex="-1"></a><span class="in">    ) </span></span>
<span id="cb7-2518"><a href="#cb7-2518" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2519"><a href="#cb7-2519" aria-hidden="true" tabindex="-1"></a><span class="in">df_all_scores %&gt;%</span></span>
<span id="cb7-2520"><a href="#cb7-2520" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::ggplot(aes(</span></span>
<span id="cb7-2521"><a href="#cb7-2521" aria-hidden="true" tabindex="-1"></a><span class="in">        x = singscore, y = gsva, color = cohort</span></span>
<span id="cb7-2522"><a href="#cb7-2522" aria-hidden="true" tabindex="-1"></a><span class="in">    )) +</span></span>
<span id="cb7-2523"><a href="#cb7-2523" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_point(alpha = 0.3) +</span></span>
<span id="cb7-2524"><a href="#cb7-2524" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::facet_wrap(~pathway, scales = "free") +</span></span>
<span id="cb7-2525"><a href="#cb7-2525" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme_bw()</span></span>
<span id="cb7-2526"><a href="#cb7-2526" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-2527"><a href="#cb7-2527" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2528"><a href="#cb7-2528" aria-hidden="true" tabindex="-1"></a>We see that in all cases there is a correlation, but the interpretation</span>
<span id="cb7-2529"><a href="#cb7-2529" aria-hidden="true" tabindex="-1"></a>changes. In some of the pathways there is difference in the singscores</span>
<span id="cb7-2530"><a href="#cb7-2530" aria-hidden="true" tabindex="-1"></a>based on the cohort, this is not seen on GSVA. </span>
<span id="cb7-2531"><a href="#cb7-2531" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2532"><a href="#cb7-2532" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=12, fig.height=5}</span></span>
<span id="cb7-2533"><a href="#cb7-2533" aria-hidden="true" tabindex="-1"></a><span class="in">#| label: fig-selected-pathways-gsva-singscores</span></span>
<span id="cb7-2534"><a href="#cb7-2534" aria-hidden="true" tabindex="-1"></a><span class="in">pathway_names &lt;- c(</span></span>
<span id="cb7-2535"><a href="#cb7-2535" aria-hidden="true" tabindex="-1"></a><span class="in">    "HALLMARK_ESTROGEN_RESPONSE_EARLY" = "Estrogen\nEarly", </span></span>
<span id="cb7-2536"><a href="#cb7-2536" aria-hidden="true" tabindex="-1"></a><span class="in">    "HALLMARK_G2M_CHECKPOINT" = "G2M",</span></span>
<span id="cb7-2537"><a href="#cb7-2537" aria-hidden="true" tabindex="-1"></a><span class="in">    "HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION" = "EMT",</span></span>
<span id="cb7-2538"><a href="#cb7-2538" aria-hidden="true" tabindex="-1"></a><span class="in">    #"SET_ERPR" = "SET ER/PR",</span></span>
<span id="cb7-2539"><a href="#cb7-2539" aria-hidden="true" tabindex="-1"></a><span class="in">    "HALLMARK_DNA_REPAIR" = "DNA Repair",</span></span>
<span id="cb7-2540"><a href="#cb7-2540" aria-hidden="true" tabindex="-1"></a><span class="in">    "HALLMARK_ANDROGEN_RESPONSE" = "Androgen\nResponse",</span></span>
<span id="cb7-2541"><a href="#cb7-2541" aria-hidden="true" tabindex="-1"></a><span class="in">    "HALLMARK_PI3K_AKT_MTOR_SIGNALING" = "PI3K AKT\nMTOR Signaling",</span></span>
<span id="cb7-2542"><a href="#cb7-2542" aria-hidden="true" tabindex="-1"></a><span class="in">    "HALLMARK_MYC_TARGETS_V1" = "MYC\ntargets V1",</span></span>
<span id="cb7-2543"><a href="#cb7-2543" aria-hidden="true" tabindex="-1"></a><span class="in">    "random_200" = "Random\n200 genes",</span></span>
<span id="cb7-2544"><a href="#cb7-2544" aria-hidden="true" tabindex="-1"></a><span class="in">    "scanb" = "SCAN-B",</span></span>
<span id="cb7-2545"><a href="#cb7-2545" aria-hidden="true" tabindex="-1"></a><span class="in">    "metabric" = "METABRIC",</span></span>
<span id="cb7-2546"><a href="#cb7-2546" aria-hidden="true" tabindex="-1"></a><span class="in">    "tcga" = "TCGA"</span></span>
<span id="cb7-2547"><a href="#cb7-2547" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-2548"><a href="#cb7-2548" aria-hidden="true" tabindex="-1"></a><span class="in">df_all_scores %&gt;%</span></span>
<span id="cb7-2549"><a href="#cb7-2549" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::filter(</span></span>
<span id="cb7-2550"><a href="#cb7-2550" aria-hidden="true" tabindex="-1"></a><span class="in">        pathway %in% names(pathway_names) &amp; </span></span>
<span id="cb7-2551"><a href="#cb7-2551" aria-hidden="true" tabindex="-1"></a><span class="in">            cohort %in% c("tcga", "metabric", "scanb")</span></span>
<span id="cb7-2552"><a href="#cb7-2552" aria-hidden="true" tabindex="-1"></a><span class="in">    ) %&gt;%</span></span>
<span id="cb7-2553"><a href="#cb7-2553" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::ggplot(aes(</span></span>
<span id="cb7-2554"><a href="#cb7-2554" aria-hidden="true" tabindex="-1"></a><span class="in">        x = singscore, y = gsva, color = cohort</span></span>
<span id="cb7-2555"><a href="#cb7-2555" aria-hidden="true" tabindex="-1"></a><span class="in">    )) +</span></span>
<span id="cb7-2556"><a href="#cb7-2556" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_point(alpha = 0.3) +</span></span>
<span id="cb7-2557"><a href="#cb7-2557" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::facet_grid(</span></span>
<span id="cb7-2558"><a href="#cb7-2558" aria-hidden="true" tabindex="-1"></a><span class="in">        cohort~pathway, </span></span>
<span id="cb7-2559"><a href="#cb7-2559" aria-hidden="true" tabindex="-1"></a><span class="in">        scales = "free",</span></span>
<span id="cb7-2560"><a href="#cb7-2560" aria-hidden="true" tabindex="-1"></a><span class="in">        labeller = as_labeller(pathway_names)</span></span>
<span id="cb7-2561"><a href="#cb7-2561" aria-hidden="true" tabindex="-1"></a><span class="in">    ) +</span></span>
<span id="cb7-2562"><a href="#cb7-2562" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme_bw(base_size = 15) +</span></span>
<span id="cb7-2563"><a href="#cb7-2563" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme(legend.position = "none")</span></span>
<span id="cb7-2564"><a href="#cb7-2564" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-2565"><a href="#cb7-2565" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2566"><a href="#cb7-2566" aria-hidden="true" tabindex="-1"></a>The next figure shows the same comparison but instead of using </span>
<span id="cb7-2567"><a href="#cb7-2567" aria-hidden="true" tabindex="-1"></a>singscore, we used ssGSEA.</span>
<span id="cb7-2568"><a href="#cb7-2568" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2569"><a href="#cb7-2569" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb7-2570"><a href="#cb7-2570" aria-hidden="true" tabindex="-1"></a><span class="in">ssgsea_dfs &lt;- mapply(</span></span>
<span id="cb7-2571"><a href="#cb7-2571" aria-hidden="true" tabindex="-1"></a><span class="in">    function(sum_exp, gene_sets, which_exp){</span></span>
<span id="cb7-2572"><a href="#cb7-2572" aria-hidden="true" tabindex="-1"></a><span class="in">        GSVA::gsva(</span></span>
<span id="cb7-2573"><a href="#cb7-2573" aria-hidden="true" tabindex="-1"></a><span class="in">            expr = assay(sum_exp,which_exp) %&gt;% as.matrix, </span></span>
<span id="cb7-2574"><a href="#cb7-2574" aria-hidden="true" tabindex="-1"></a><span class="in">            gset.idx.list = gene_sets, </span></span>
<span id="cb7-2575"><a href="#cb7-2575" aria-hidden="true" tabindex="-1"></a><span class="in">            method = "ssgsea", </span></span>
<span id="cb7-2576"><a href="#cb7-2576" aria-hidden="true" tabindex="-1"></a><span class="in">            ssgsea.norm = FALSE,</span></span>
<span id="cb7-2577"><a href="#cb7-2577" aria-hidden="true" tabindex="-1"></a><span class="in">            verbose = FALSE</span></span>
<span id="cb7-2578"><a href="#cb7-2578" aria-hidden="true" tabindex="-1"></a><span class="in">        )</span></span>
<span id="cb7-2579"><a href="#cb7-2579" aria-hidden="true" tabindex="-1"></a><span class="in">    },</span></span>
<span id="cb7-2580"><a href="#cb7-2580" aria-hidden="true" tabindex="-1"></a><span class="in">    sum_exp = datasets,</span></span>
<span id="cb7-2581"><a href="#cb7-2581" aria-hidden="true" tabindex="-1"></a><span class="in">    which_exp = c(which_exp, "poetic" = "normalized_intensity"),</span></span>
<span id="cb7-2582"><a href="#cb7-2582" aria-hidden="true" tabindex="-1"></a><span class="in">    MoreArgs = list(gene_sets = gene_sets_)</span></span>
<span id="cb7-2583"><a href="#cb7-2583" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-2584"><a href="#cb7-2584" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2585"><a href="#cb7-2585" aria-hidden="true" tabindex="-1"></a><span class="in">ssgsea_dfs &lt;- lapply(ssgsea_dfs, t)</span></span>
<span id="cb7-2586"><a href="#cb7-2586" aria-hidden="true" tabindex="-1"></a><span class="in">ssgsea_dfs &lt;- Reduce(rbind, x = ssgsea_dfs)</span></span>
<span id="cb7-2587"><a href="#cb7-2587" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2588"><a href="#cb7-2588" aria-hidden="true" tabindex="-1"></a><span class="in">saveRDS(</span></span>
<span id="cb7-2589"><a href="#cb7-2589" aria-hidden="true" tabindex="-1"></a><span class="in">    ssgsea_dfs,</span></span>
<span id="cb7-2590"><a href="#cb7-2590" aria-hidden="true" tabindex="-1"></a><span class="in">    "../../results/rds_files/expanding_ember/ssgsea_dfs.rds"</span></span>
<span id="cb7-2591"><a href="#cb7-2591" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-2592"><a href="#cb7-2592" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-2593"><a href="#cb7-2593" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2594"><a href="#cb7-2594" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=12, fig.height=5}</span></span>
<span id="cb7-2595"><a href="#cb7-2595" aria-hidden="true" tabindex="-1"></a><span class="in">ssgsea_dfs_gsva &lt;- ssgsea_dfs %&gt;% </span></span>
<span id="cb7-2596"><a href="#cb7-2596" aria-hidden="true" tabindex="-1"></a><span class="in">    data.frame %&gt;%</span></span>
<span id="cb7-2597"><a href="#cb7-2597" aria-hidden="true" tabindex="-1"></a><span class="in">    tibble::rownames_to_column(var = "sample_name") %&gt;%</span></span>
<span id="cb7-2598"><a href="#cb7-2598" aria-hidden="true" tabindex="-1"></a><span class="in">    tidyr::pivot_longer(</span></span>
<span id="cb7-2599"><a href="#cb7-2599" aria-hidden="true" tabindex="-1"></a><span class="in">        cols = dplyr::all_of(names(gene_sets_)),</span></span>
<span id="cb7-2600"><a href="#cb7-2600" aria-hidden="true" tabindex="-1"></a><span class="in">        names_to = "pathway",</span></span>
<span id="cb7-2601"><a href="#cb7-2601" aria-hidden="true" tabindex="-1"></a><span class="in">        values_to = "ssGSEA"</span></span>
<span id="cb7-2602"><a href="#cb7-2602" aria-hidden="true" tabindex="-1"></a><span class="in">    ) %&gt;%</span></span>
<span id="cb7-2603"><a href="#cb7-2603" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::inner_join(</span></span>
<span id="cb7-2604"><a href="#cb7-2604" aria-hidden="true" tabindex="-1"></a><span class="in">        merged_col_data %&gt;%</span></span>
<span id="cb7-2605"><a href="#cb7-2605" aria-hidden="true" tabindex="-1"></a><span class="in">            tidyr::pivot_longer(</span></span>
<span id="cb7-2606"><a href="#cb7-2606" aria-hidden="true" tabindex="-1"></a><span class="in">                cols = dplyr::all_of(names(gene_sets_gsea)),</span></span>
<span id="cb7-2607"><a href="#cb7-2607" aria-hidden="true" tabindex="-1"></a><span class="in">                names_to = "pathway",</span></span>
<span id="cb7-2608"><a href="#cb7-2608" aria-hidden="true" tabindex="-1"></a><span class="in">                values_to = "gsva"</span></span>
<span id="cb7-2609"><a href="#cb7-2609" aria-hidden="true" tabindex="-1"></a><span class="in">            ),</span></span>
<span id="cb7-2610"><a href="#cb7-2610" aria-hidden="true" tabindex="-1"></a><span class="in">        .,</span></span>
<span id="cb7-2611"><a href="#cb7-2611" aria-hidden="true" tabindex="-1"></a><span class="in">        by = c("sample_name", "pathway")</span></span>
<span id="cb7-2612"><a href="#cb7-2612" aria-hidden="true" tabindex="-1"></a><span class="in">    ) </span></span>
<span id="cb7-2613"><a href="#cb7-2613" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2614"><a href="#cb7-2614" aria-hidden="true" tabindex="-1"></a><span class="in">ssgsea_dfs_gsva %&gt;%</span></span>
<span id="cb7-2615"><a href="#cb7-2615" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::filter(</span></span>
<span id="cb7-2616"><a href="#cb7-2616" aria-hidden="true" tabindex="-1"></a><span class="in">        pathway %in% names(pathway_names) &amp; </span></span>
<span id="cb7-2617"><a href="#cb7-2617" aria-hidden="true" tabindex="-1"></a><span class="in">            cohort %in% c("tcga", "metabric", "scanb")</span></span>
<span id="cb7-2618"><a href="#cb7-2618" aria-hidden="true" tabindex="-1"></a><span class="in">    ) %&gt;%</span></span>
<span id="cb7-2619"><a href="#cb7-2619" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::ggplot(aes(</span></span>
<span id="cb7-2620"><a href="#cb7-2620" aria-hidden="true" tabindex="-1"></a><span class="in">        x = ssGSEA, y = gsva, color = cohort</span></span>
<span id="cb7-2621"><a href="#cb7-2621" aria-hidden="true" tabindex="-1"></a><span class="in">    )) +</span></span>
<span id="cb7-2622"><a href="#cb7-2622" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_point(alpha = 0.3) +</span></span>
<span id="cb7-2623"><a href="#cb7-2623" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::facet_grid(</span></span>
<span id="cb7-2624"><a href="#cb7-2624" aria-hidden="true" tabindex="-1"></a><span class="in">        cohort~pathway, </span></span>
<span id="cb7-2625"><a href="#cb7-2625" aria-hidden="true" tabindex="-1"></a><span class="in">        scales = "free_y",</span></span>
<span id="cb7-2626"><a href="#cb7-2626" aria-hidden="true" tabindex="-1"></a><span class="in">        labeller = as_labeller(pathway_names)</span></span>
<span id="cb7-2627"><a href="#cb7-2627" aria-hidden="true" tabindex="-1"></a><span class="in">    ) +</span></span>
<span id="cb7-2628"><a href="#cb7-2628" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme_bw(base_size = 15) +</span></span>
<span id="cb7-2629"><a href="#cb7-2629" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme(legend.position = "none")</span></span>
<span id="cb7-2630"><a href="#cb7-2630" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-2631"><a href="#cb7-2631" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2632"><a href="#cb7-2632" aria-hidden="true" tabindex="-1"></a>The picture is similar as before, but now we actually look at the distributions</span>
<span id="cb7-2633"><a href="#cb7-2633" aria-hidden="true" tabindex="-1"></a>across the 3 big cohorts. We used ssGSEA without the normalization, as that</span>
<span id="cb7-2634"><a href="#cb7-2634" aria-hidden="true" tabindex="-1"></a>would be necessary in the case of an individual sample. </span>
<span id="cb7-2635"><a href="#cb7-2635" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2636"><a href="#cb7-2636" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=12, fig.height=6}</span></span>
<span id="cb7-2637"><a href="#cb7-2637" aria-hidden="true" tabindex="-1"></a><span class="in">#| label: fig-ssgsea-dists</span></span>
<span id="cb7-2638"><a href="#cb7-2638" aria-hidden="true" tabindex="-1"></a><span class="in">unique_pathways &lt;- intersect(names(pathway_names), unique(scores$pathway))</span></span>
<span id="cb7-2639"><a href="#cb7-2639" aria-hidden="true" tabindex="-1"></a><span class="in">ssgsea_dfs_gsva %&gt;%</span></span>
<span id="cb7-2640"><a href="#cb7-2640" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::filter(</span></span>
<span id="cb7-2641"><a href="#cb7-2641" aria-hidden="true" tabindex="-1"></a><span class="in">        pathway %in% unique_pathways &amp; </span></span>
<span id="cb7-2642"><a href="#cb7-2642" aria-hidden="true" tabindex="-1"></a><span class="in">            cohort %in% c("tcga", "metabric", "scanb")</span></span>
<span id="cb7-2643"><a href="#cb7-2643" aria-hidden="true" tabindex="-1"></a><span class="in">    ) %&gt;%</span></span>
<span id="cb7-2644"><a href="#cb7-2644" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::mutate(cohort = dplyr::case_when(</span></span>
<span id="cb7-2645"><a href="#cb7-2645" aria-hidden="true" tabindex="-1"></a><span class="in">        cohort == "tcga" ~ "TCGA",</span></span>
<span id="cb7-2646"><a href="#cb7-2646" aria-hidden="true" tabindex="-1"></a><span class="in">        cohort == "metabric" ~ "METABRIC",</span></span>
<span id="cb7-2647"><a href="#cb7-2647" aria-hidden="true" tabindex="-1"></a><span class="in">        cohort == "scanb" ~ "SCAN-B"</span></span>
<span id="cb7-2648"><a href="#cb7-2648" aria-hidden="true" tabindex="-1"></a><span class="in">    )) %&gt;%</span></span>
<span id="cb7-2649"><a href="#cb7-2649" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::ggplot(aes(</span></span>
<span id="cb7-2650"><a href="#cb7-2650" aria-hidden="true" tabindex="-1"></a><span class="in">        x = ssGSEA, fill = cohort</span></span>
<span id="cb7-2651"><a href="#cb7-2651" aria-hidden="true" tabindex="-1"></a><span class="in">    )) +</span></span>
<span id="cb7-2652"><a href="#cb7-2652" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_density(alpha = 0.3) +</span></span>
<span id="cb7-2653"><a href="#cb7-2653" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::facet_wrap(</span></span>
<span id="cb7-2654"><a href="#cb7-2654" aria-hidden="true" tabindex="-1"></a><span class="in">        ~pathway, </span></span>
<span id="cb7-2655"><a href="#cb7-2655" aria-hidden="true" tabindex="-1"></a><span class="in">        scales = "free",</span></span>
<span id="cb7-2656"><a href="#cb7-2656" aria-hidden="true" tabindex="-1"></a><span class="in">        labeller = as_labeller(pathway_names)</span></span>
<span id="cb7-2657"><a href="#cb7-2657" aria-hidden="true" tabindex="-1"></a><span class="in">    ) + </span></span>
<span id="cb7-2658"><a href="#cb7-2658" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::scale_y_continuous(labels = NULL, breaks = NULL) + </span></span>
<span id="cb7-2659"><a href="#cb7-2659" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::scale_x_continuous(guide = guide_axis(angle = 45)) + </span></span>
<span id="cb7-2660"><a href="#cb7-2660" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::labs(</span></span>
<span id="cb7-2661"><a href="#cb7-2661" aria-hidden="true" tabindex="-1"></a><span class="in">        x = "ssGSEA",</span></span>
<span id="cb7-2662"><a href="#cb7-2662" aria-hidden="true" tabindex="-1"></a><span class="in">        y = "Density",</span></span>
<span id="cb7-2663"><a href="#cb7-2663" aria-hidden="true" tabindex="-1"></a><span class="in">        fill = "Cohort"</span></span>
<span id="cb7-2664"><a href="#cb7-2664" aria-hidden="true" tabindex="-1"></a><span class="in">    ) +</span></span>
<span id="cb7-2665"><a href="#cb7-2665" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme_bw(base_size = 20) +</span></span>
<span id="cb7-2666"><a href="#cb7-2666" aria-hidden="true" tabindex="-1"></a><span class="in">    change_guides_point() +</span></span>
<span id="cb7-2667"><a href="#cb7-2667" aria-hidden="true" tabindex="-1"></a><span class="in">    change_plot_aes_point()</span></span>
<span id="cb7-2668"><a href="#cb7-2668" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-2669"><a href="#cb7-2669" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2670"><a href="#cb7-2670" aria-hidden="true" tabindex="-1"></a>We see that overall there is a discrepancy, sometimes big,</span>
<span id="cb7-2671"><a href="#cb7-2671" aria-hidden="true" tabindex="-1"></a>depending on the pathway, including the random genes</span>
<span id="cb7-2672"><a href="#cb7-2672" aria-hidden="true" tabindex="-1"></a>pathway. Next we show the same distribution for </span>
<span id="cb7-2673"><a href="#cb7-2673" aria-hidden="true" tabindex="-1"></a>singscore. Similarly there is a big discrepancy in several cases. </span>
<span id="cb7-2674"><a href="#cb7-2674" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2675"><a href="#cb7-2675" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=6}</span></span>
<span id="cb7-2676"><a href="#cb7-2676" aria-hidden="true" tabindex="-1"></a><span class="in">#| label: fig-singscore-dists</span></span>
<span id="cb7-2677"><a href="#cb7-2677" aria-hidden="true" tabindex="-1"></a><span class="in">df_all_scores %&gt;%</span></span>
<span id="cb7-2678"><a href="#cb7-2678" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::filter(</span></span>
<span id="cb7-2679"><a href="#cb7-2679" aria-hidden="true" tabindex="-1"></a><span class="in">        pathway %in% unique_pathways &amp; </span></span>
<span id="cb7-2680"><a href="#cb7-2680" aria-hidden="true" tabindex="-1"></a><span class="in">            cohort %in% c("tcga", "metabric", "scanb")</span></span>
<span id="cb7-2681"><a href="#cb7-2681" aria-hidden="true" tabindex="-1"></a><span class="in">    ) %&gt;%</span></span>
<span id="cb7-2682"><a href="#cb7-2682" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::mutate(cohort = dplyr::case_when(</span></span>
<span id="cb7-2683"><a href="#cb7-2683" aria-hidden="true" tabindex="-1"></a><span class="in">        cohort == "tcga" ~ "TCGA",</span></span>
<span id="cb7-2684"><a href="#cb7-2684" aria-hidden="true" tabindex="-1"></a><span class="in">        cohort == "metabric" ~ "METABRIC",</span></span>
<span id="cb7-2685"><a href="#cb7-2685" aria-hidden="true" tabindex="-1"></a><span class="in">        cohort == "scanb" ~ "SCAN-B"</span></span>
<span id="cb7-2686"><a href="#cb7-2686" aria-hidden="true" tabindex="-1"></a><span class="in">    )) %&gt;%</span></span>
<span id="cb7-2687"><a href="#cb7-2687" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::ggplot(aes(x = singscore, fill = cohort)) +</span></span>
<span id="cb7-2688"><a href="#cb7-2688" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_density(alpha = 0.3) +</span></span>
<span id="cb7-2689"><a href="#cb7-2689" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::facet_wrap(</span></span>
<span id="cb7-2690"><a href="#cb7-2690" aria-hidden="true" tabindex="-1"></a><span class="in">        ~pathway, </span></span>
<span id="cb7-2691"><a href="#cb7-2691" aria-hidden="true" tabindex="-1"></a><span class="in">        scales = "free",</span></span>
<span id="cb7-2692"><a href="#cb7-2692" aria-hidden="true" tabindex="-1"></a><span class="in">        labeller = as_labeller(pathway_names)</span></span>
<span id="cb7-2693"><a href="#cb7-2693" aria-hidden="true" tabindex="-1"></a><span class="in">    ) + </span></span>
<span id="cb7-2694"><a href="#cb7-2694" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::scale_y_continuous(labels = NULL, breaks = NULL) + </span></span>
<span id="cb7-2695"><a href="#cb7-2695" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::scale_x_continuous(guide = guide_axis(angle = 45)) + </span></span>
<span id="cb7-2696"><a href="#cb7-2696" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::labs(</span></span>
<span id="cb7-2697"><a href="#cb7-2697" aria-hidden="true" tabindex="-1"></a><span class="in">        x = "singscore",</span></span>
<span id="cb7-2698"><a href="#cb7-2698" aria-hidden="true" tabindex="-1"></a><span class="in">        y = "Density",</span></span>
<span id="cb7-2699"><a href="#cb7-2699" aria-hidden="true" tabindex="-1"></a><span class="in">        fill = "Cohort"</span></span>
<span id="cb7-2700"><a href="#cb7-2700" aria-hidden="true" tabindex="-1"></a><span class="in">    ) +</span></span>
<span id="cb7-2701"><a href="#cb7-2701" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme_bw(base_size = 20) +</span></span>
<span id="cb7-2702"><a href="#cb7-2702" aria-hidden="true" tabindex="-1"></a><span class="in">    change_guides_point() +</span></span>
<span id="cb7-2703"><a href="#cb7-2703" aria-hidden="true" tabindex="-1"></a><span class="in">    change_plot_aes_point()</span></span>
<span id="cb7-2704"><a href="#cb7-2704" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-2705"><a href="#cb7-2705" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2706"><a href="#cb7-2706" aria-hidden="true" tabindex="-1"></a>But now, @fig-regressed-scores-dists shows that when we calculate</span>
<span id="cb7-2707"><a href="#cb7-2707" aria-hidden="true" tabindex="-1"></a>the scores with the regressed data using our methodology the distributions</span>
<span id="cb7-2708"><a href="#cb7-2708" aria-hidden="true" tabindex="-1"></a>are matched and adjusted, with the expection of G2M, which seems to be </span>
<span id="cb7-2709"><a href="#cb7-2709" aria-hidden="true" tabindex="-1"></a>something more fundamental instead of batch effect. Note that </span>
<span id="cb7-2710"><a href="#cb7-2710" aria-hidden="true" tabindex="-1"></a>all the distributions are overlapping significantly.</span>
<span id="cb7-2711"><a href="#cb7-2711" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2712"><a href="#cb7-2712" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=6}</span></span>
<span id="cb7-2713"><a href="#cb7-2713" aria-hidden="true" tabindex="-1"></a><span class="in">#| label: fig-regressed-scores-dists</span></span>
<span id="cb7-2714"><a href="#cb7-2714" aria-hidden="true" tabindex="-1"></a><span class="in">scores %&gt;%</span></span>
<span id="cb7-2715"><a href="#cb7-2715" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::filter(cohort %in% c("tcga", "scanb", "metabric")) %&gt;%</span></span>
<span id="cb7-2716"><a href="#cb7-2716" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::mutate(cohort = dplyr::case_when(</span></span>
<span id="cb7-2717"><a href="#cb7-2717" aria-hidden="true" tabindex="-1"></a><span class="in">        cohort == "tcga" ~ "TCGA",</span></span>
<span id="cb7-2718"><a href="#cb7-2718" aria-hidden="true" tabindex="-1"></a><span class="in">        cohort == "metabric" ~ "METABRIC",</span></span>
<span id="cb7-2719"><a href="#cb7-2719" aria-hidden="true" tabindex="-1"></a><span class="in">        cohort == "scanb" ~ "SCAN-B"</span></span>
<span id="cb7-2720"><a href="#cb7-2720" aria-hidden="true" tabindex="-1"></a><span class="in">    )) %&gt;%</span></span>
<span id="cb7-2721"><a href="#cb7-2721" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::filter(pathway %in% unique_pathways) %&gt;%</span></span>
<span id="cb7-2722"><a href="#cb7-2722" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::ggplot(aes(</span></span>
<span id="cb7-2723"><a href="#cb7-2723" aria-hidden="true" tabindex="-1"></a><span class="in">        x = score_hkg, fill = cohort</span></span>
<span id="cb7-2724"><a href="#cb7-2724" aria-hidden="true" tabindex="-1"></a><span class="in">    )) +</span></span>
<span id="cb7-2725"><a href="#cb7-2725" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_density(alpha = 0.3) +</span></span>
<span id="cb7-2726"><a href="#cb7-2726" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::facet_wrap(</span></span>
<span id="cb7-2727"><a href="#cb7-2727" aria-hidden="true" tabindex="-1"></a><span class="in">        ~pathway, </span></span>
<span id="cb7-2728"><a href="#cb7-2728" aria-hidden="true" tabindex="-1"></a><span class="in">        scales = "free",</span></span>
<span id="cb7-2729"><a href="#cb7-2729" aria-hidden="true" tabindex="-1"></a><span class="in">        labeller = as_labeller(pathway_names)</span></span>
<span id="cb7-2730"><a href="#cb7-2730" aria-hidden="true" tabindex="-1"></a><span class="in">    ) + </span></span>
<span id="cb7-2731"><a href="#cb7-2731" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::scale_y_continuous(labels = NULL, breaks = NULL) + </span></span>
<span id="cb7-2732"><a href="#cb7-2732" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::scale_x_continuous(guide = guide_axis(angle = 45)) + </span></span>
<span id="cb7-2733"><a href="#cb7-2733" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::labs(</span></span>
<span id="cb7-2734"><a href="#cb7-2734" aria-hidden="true" tabindex="-1"></a><span class="in">        x = "Regressed score",</span></span>
<span id="cb7-2735"><a href="#cb7-2735" aria-hidden="true" tabindex="-1"></a><span class="in">        y = "Density",</span></span>
<span id="cb7-2736"><a href="#cb7-2736" aria-hidden="true" tabindex="-1"></a><span class="in">        fill = "Cohort"</span></span>
<span id="cb7-2737"><a href="#cb7-2737" aria-hidden="true" tabindex="-1"></a><span class="in">    ) + </span></span>
<span id="cb7-2738"><a href="#cb7-2738" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme_bw(base_size = 20) +</span></span>
<span id="cb7-2739"><a href="#cb7-2739" aria-hidden="true" tabindex="-1"></a><span class="in">    change_guides_point() +</span></span>
<span id="cb7-2740"><a href="#cb7-2740" aria-hidden="true" tabindex="-1"></a><span class="in">    change_plot_aes_point()</span></span>
<span id="cb7-2741"><a href="#cb7-2741" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-2742"><a href="#cb7-2742" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2743"><a href="#cb7-2743" aria-hidden="true" tabindex="-1"></a>The next two figures are the distributions of the singscore and GSVA</span>
<span id="cb7-2744"><a href="#cb7-2744" aria-hidden="true" tabindex="-1"></a>filled by cohort.</span>
<span id="cb7-2745"><a href="#cb7-2745" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2746"><a href="#cb7-2746" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=20, fig.height=16, dev='png'}</span></span>
<span id="cb7-2747"><a href="#cb7-2747" aria-hidden="true" tabindex="-1"></a><span class="in">df_all_scores %&gt;%</span></span>
<span id="cb7-2748"><a href="#cb7-2748" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::ggplot(aes(</span></span>
<span id="cb7-2749"><a href="#cb7-2749" aria-hidden="true" tabindex="-1"></a><span class="in">        x = singscore, fill = cohort, alpha = 0.5</span></span>
<span id="cb7-2750"><a href="#cb7-2750" aria-hidden="true" tabindex="-1"></a><span class="in">    )) +</span></span>
<span id="cb7-2751"><a href="#cb7-2751" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_density(position = "identity", alpha = 0.5) +</span></span>
<span id="cb7-2752"><a href="#cb7-2752" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::facet_wrap(~pathway, scales = "free")</span></span>
<span id="cb7-2753"><a href="#cb7-2753" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-2754"><a href="#cb7-2754" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2755"><a href="#cb7-2755" aria-hidden="true" tabindex="-1"></a>For several pathways in the singscore the distributions are similar,</span>
<span id="cb7-2756"><a href="#cb7-2756" aria-hidden="true" tabindex="-1"></a>but that is not always the case.</span>
<span id="cb7-2757"><a href="#cb7-2757" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2758"><a href="#cb7-2758" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=20, fig.height=16, dev='png'}</span></span>
<span id="cb7-2759"><a href="#cb7-2759" aria-hidden="true" tabindex="-1"></a><span class="in">df_all_scores %&gt;%</span></span>
<span id="cb7-2760"><a href="#cb7-2760" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::ggplot(aes(</span></span>
<span id="cb7-2761"><a href="#cb7-2761" aria-hidden="true" tabindex="-1"></a><span class="in">        x = gsva, fill = cohort</span></span>
<span id="cb7-2762"><a href="#cb7-2762" aria-hidden="true" tabindex="-1"></a><span class="in">    )) +</span></span>
<span id="cb7-2763"><a href="#cb7-2763" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_density(position = "identity", alpha = 0.5) +</span></span>
<span id="cb7-2764"><a href="#cb7-2764" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::facet_wrap(~pathway, scales = "free")</span></span>
<span id="cb7-2765"><a href="#cb7-2765" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-2766"><a href="#cb7-2766" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2767"><a href="#cb7-2767" aria-hidden="true" tabindex="-1"></a>On the other hand GSVA gives very similar distributions for all the </span>
<span id="cb7-2768"><a href="#cb7-2768" aria-hidden="true" tabindex="-1"></a>different cohorts. Therefore we actually need to use GSVA. It is important in </span>
<span id="cb7-2769"><a href="#cb7-2769" aria-hidden="true" tabindex="-1"></a>the end to calculate scores based on the cohort. </span>
<span id="cb7-2770"><a href="#cb7-2770" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2771"><a href="#cb7-2771" aria-hidden="true" tabindex="-1"></a>We now move on and try the same analysis on the regressed data.</span>
<span id="cb7-2772"><a href="#cb7-2772" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2773"><a href="#cb7-2773" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb7-2774"><a href="#cb7-2774" aria-hidden="true" tabindex="-1"></a><span class="in">singscore_dfs_regressed &lt;- singscore::multiScore(</span></span>
<span id="cb7-2775"><a href="#cb7-2775" aria-hidden="true" tabindex="-1"></a><span class="in">    singscore::rankGenes(df_pcs_regressed),</span></span>
<span id="cb7-2776"><a href="#cb7-2776" aria-hidden="true" tabindex="-1"></a><span class="in">    upSetColc = gene_sets_gsea</span></span>
<span id="cb7-2777"><a href="#cb7-2777" aria-hidden="true" tabindex="-1"></a><span class="in">)[[1]]</span></span>
<span id="cb7-2778"><a href="#cb7-2778" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2779"><a href="#cb7-2779" aria-hidden="true" tabindex="-1"></a><span class="in">saveRDS(</span></span>
<span id="cb7-2780"><a href="#cb7-2780" aria-hidden="true" tabindex="-1"></a><span class="in">    singscore_dfs_regressed,</span></span>
<span id="cb7-2781"><a href="#cb7-2781" aria-hidden="true" tabindex="-1"></a><span class="in">    "../../results/rds_files/expanding_ember/singscore_dfs_regressed.rds"</span></span>
<span id="cb7-2782"><a href="#cb7-2782" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb7-2783"><a href="#cb7-2783" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-2784"><a href="#cb7-2784" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2785"><a href="#cb7-2785" aria-hidden="true" tabindex="-1"></a>We start by comparing with the GSVA scores.</span>
<span id="cb7-2786"><a href="#cb7-2786" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2787"><a href="#cb7-2787" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=20, fig.height=16, dev='png'}</span></span>
<span id="cb7-2788"><a href="#cb7-2788" aria-hidden="true" tabindex="-1"></a><span class="in">df_all_scores_regressed &lt;- singscore_dfs_regressed %&gt;% </span></span>
<span id="cb7-2789"><a href="#cb7-2789" aria-hidden="true" tabindex="-1"></a><span class="in">    t %&gt;%</span></span>
<span id="cb7-2790"><a href="#cb7-2790" aria-hidden="true" tabindex="-1"></a><span class="in">    data.frame %&gt;%</span></span>
<span id="cb7-2791"><a href="#cb7-2791" aria-hidden="true" tabindex="-1"></a><span class="in">    tibble::rownames_to_column(var = "sample_name") %&gt;%</span></span>
<span id="cb7-2792"><a href="#cb7-2792" aria-hidden="true" tabindex="-1"></a><span class="in">    tidyr::pivot_longer(</span></span>
<span id="cb7-2793"><a href="#cb7-2793" aria-hidden="true" tabindex="-1"></a><span class="in">        cols = dplyr::all_of(names(gene_sets_gsea)),</span></span>
<span id="cb7-2794"><a href="#cb7-2794" aria-hidden="true" tabindex="-1"></a><span class="in">        names_to = "pathway",</span></span>
<span id="cb7-2795"><a href="#cb7-2795" aria-hidden="true" tabindex="-1"></a><span class="in">        values_to = "singscore"</span></span>
<span id="cb7-2796"><a href="#cb7-2796" aria-hidden="true" tabindex="-1"></a><span class="in">    ) %&gt;%</span></span>
<span id="cb7-2797"><a href="#cb7-2797" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::inner_join(</span></span>
<span id="cb7-2798"><a href="#cb7-2798" aria-hidden="true" tabindex="-1"></a><span class="in">        merged_col_data %&gt;%</span></span>
<span id="cb7-2799"><a href="#cb7-2799" aria-hidden="true" tabindex="-1"></a><span class="in">            tidyr::pivot_longer(</span></span>
<span id="cb7-2800"><a href="#cb7-2800" aria-hidden="true" tabindex="-1"></a><span class="in">                cols = dplyr::all_of(names(gene_sets_gsea)),</span></span>
<span id="cb7-2801"><a href="#cb7-2801" aria-hidden="true" tabindex="-1"></a><span class="in">                names_to = "pathway",</span></span>
<span id="cb7-2802"><a href="#cb7-2802" aria-hidden="true" tabindex="-1"></a><span class="in">                values_to = "gsva"</span></span>
<span id="cb7-2803"><a href="#cb7-2803" aria-hidden="true" tabindex="-1"></a><span class="in">            ),</span></span>
<span id="cb7-2804"><a href="#cb7-2804" aria-hidden="true" tabindex="-1"></a><span class="in">        .,</span></span>
<span id="cb7-2805"><a href="#cb7-2805" aria-hidden="true" tabindex="-1"></a><span class="in">        by = c("sample_name", "pathway")</span></span>
<span id="cb7-2806"><a href="#cb7-2806" aria-hidden="true" tabindex="-1"></a><span class="in">    ) </span></span>
<span id="cb7-2807"><a href="#cb7-2807" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2808"><a href="#cb7-2808" aria-hidden="true" tabindex="-1"></a><span class="in">df_all_scores_regressed %&gt;%</span></span>
<span id="cb7-2809"><a href="#cb7-2809" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::ggplot(aes(</span></span>
<span id="cb7-2810"><a href="#cb7-2810" aria-hidden="true" tabindex="-1"></a><span class="in">        x = singscore, y = gsva, color = cohort</span></span>
<span id="cb7-2811"><a href="#cb7-2811" aria-hidden="true" tabindex="-1"></a><span class="in">    )) +</span></span>
<span id="cb7-2812"><a href="#cb7-2812" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_point() +</span></span>
<span id="cb7-2813"><a href="#cb7-2813" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::facet_wrap(~pathway, scales = "free") +</span></span>
<span id="cb7-2814"><a href="#cb7-2814" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme_bw()</span></span>
<span id="cb7-2815"><a href="#cb7-2815" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-2816"><a href="#cb7-2816" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2817"><a href="#cb7-2817" aria-hidden="true" tabindex="-1"></a>We see that in all cases there is a correlation, but the interpretation</span>
<span id="cb7-2818"><a href="#cb7-2818" aria-hidden="true" tabindex="-1"></a>changes. In some of the pathways there is difference in the singscores</span>
<span id="cb7-2819"><a href="#cb7-2819" aria-hidden="true" tabindex="-1"></a>based on the cohort, this is not seen on GSVA. </span>
<span id="cb7-2820"><a href="#cb7-2820" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2821"><a href="#cb7-2821" aria-hidden="true" tabindex="-1"></a>The next two figures are the distributions of the singscore and GSVA</span>
<span id="cb7-2822"><a href="#cb7-2822" aria-hidden="true" tabindex="-1"></a>filled by cohort.</span>
<span id="cb7-2823"><a href="#cb7-2823" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2824"><a href="#cb7-2824" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=20, fig.height=16, dev='png'}</span></span>
<span id="cb7-2825"><a href="#cb7-2825" aria-hidden="true" tabindex="-1"></a><span class="in">df_all_scores_regressed %&gt;%</span></span>
<span id="cb7-2826"><a href="#cb7-2826" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::ggplot(aes(</span></span>
<span id="cb7-2827"><a href="#cb7-2827" aria-hidden="true" tabindex="-1"></a><span class="in">        x = singscore, fill = cohort</span></span>
<span id="cb7-2828"><a href="#cb7-2828" aria-hidden="true" tabindex="-1"></a><span class="in">    )) +</span></span>
<span id="cb7-2829"><a href="#cb7-2829" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_density(position = "identity", alpha = 0.5) +</span></span>
<span id="cb7-2830"><a href="#cb7-2830" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::facet_wrap(~pathway, scales = "free")</span></span>
<span id="cb7-2831"><a href="#cb7-2831" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-2832"><a href="#cb7-2832" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2833"><a href="#cb7-2833" aria-hidden="true" tabindex="-1"></a>In the end the best way of scoring seems to be using GSVA on the </span>
<span id="cb7-2834"><a href="#cb7-2834" aria-hidden="true" tabindex="-1"></a>unregressed data though it depends on the pathways. For example the</span>
<span id="cb7-2835"><a href="#cb7-2835" aria-hidden="true" tabindex="-1"></a>hallmark estrogen response early and set ER/PR pathways seem to </span>
<span id="cb7-2836"><a href="#cb7-2836" aria-hidden="true" tabindex="-1"></a>have been regressed quite well, so we can actually compare the results using </span>
<span id="cb7-2837"><a href="#cb7-2837" aria-hidden="true" tabindex="-1"></a>the scores. On the other hand we can't use the pathways G2M checkpoint,</span>
<span id="cb7-2838"><a href="#cb7-2838" aria-hidden="true" tabindex="-1"></a>E2F targets and EMT. For the PI3K signaling score it seems it is </span>
<span id="cb7-2839"><a href="#cb7-2839" aria-hidden="true" tabindex="-1"></a>comparable to SCANB only.</span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
      &nbsp;
    </div>   
    <div class="nav-footer-center">
      &nbsp;
    <div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/chronchi/molecular_landscape/issues/new" class="toc-action"><i class="bi bi-github"></i>Report an issue</a></li></ul></div></div>
    <div class="nav-footer-right">
      <ul class="footer-items list-unstyled">
    <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/chronchi">
      <i class="bi bi-github" role="img">
</i> 
    </a>
  </li>  
    <li class="nav-item compact">
    <a class="nav-link" href="https://twitter.com/chronchi">
      <i class="bi bi-twitter" role="img">
</i> 
    </a>
  </li>  
</ul>
    </div>
  </div>
</footer>




</body></html>