<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.361">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>EMBER creates a unified space for independent breast cancer transcriptomic datasets enabling precision oncology. - 8&nbsp; Revision</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../chapters/discussion.html" rel="next">
<link href="../chapters/trying.html" rel="prev">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>

<script src="../site_libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<link href="../site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="../site_libs/datatables-binding-0.28/datatables.js"></script>
<script src="../site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<link href="../site_libs/dt-core-1.13.4/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="../site_libs/dt-core-1.13.4/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="../site_libs/dt-core-1.13.4/js/jquery.dataTables.min.js"></script>
<link href="../site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet">
<script src="../site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>


</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../chapters/revision.html"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Revision</span></a></li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
      <button type="button" class="btn quarto-search-button" aria-label="" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../">EMBER creates a unified space for independent breast cancer transcriptomic datasets enabling precision oncology.</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/chronchi/molecular_landscape" title="Source Code" class="quarto-navigation-tool px-1" aria-label="Source Code"><i class="bi bi-github"></i></a>
    <a href="https://twitter.com/intent/tweet?url=|url|" title="Twitter" class="quarto-navigation-tool px-1" aria-label="Twitter"><i class="bi bi-twitter"></i></a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Preface</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/intro.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Introduction</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/surv_analysis_estrogen.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Estrogen receptor status is continuous, not dichotomous</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/pca_merging.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">A new framework for personalized medicine: integrating</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/validation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Validating the molecular landscape</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/scoring.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Pathway activity in a personalized context</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/risk_score.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Risk score model</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/expanding_ember.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Expanding embedding and removing batch effects</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/trying.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Tests and tries</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/revision.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Revision</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/discussion.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Discussion</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/references.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">References</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#comparing-to-combats-performance" id="toc-comparing-to-combats-performance" class="nav-link active" data-scroll-target="#comparing-to-combats-performance"><span class="header-section-number">8.1</span> Comparing to combat’s performance</a></li>
  <li><a href="#autoencoders" id="toc-autoencoders" class="nav-link" data-scroll-target="#autoencoders"><span class="header-section-number">8.2</span> Autoencoders</a></li>
  <li><a href="#mixing-analysis" id="toc-mixing-analysis" class="nav-link" data-scroll-target="#mixing-analysis"><span class="header-section-number">8.3</span> Mixing analysis</a></li>
  <li><a href="#stable-genes" id="toc-stable-genes" class="nav-link" data-scroll-target="#stable-genes"><span class="header-section-number">8.4</span> Stable genes</a></li>
  <li><a href="#scaling" id="toc-scaling" class="nav-link" data-scroll-target="#scaling"><span class="header-section-number">8.5</span> Scaling</a></li>
  <li><a href="#tumor-purity-statistical-tests" id="toc-tumor-purity-statistical-tests" class="nav-link" data-scroll-target="#tumor-purity-statistical-tests"><span class="header-section-number">8.6</span> Tumor purity: statistical tests</a></li>
  <li><a href="#metastasis-and-ffpe-data" id="toc-metastasis-and-ffpe-data" class="nav-link" data-scroll-target="#metastasis-and-ffpe-data"><span class="header-section-number">8.7</span> Metastasis and FFPE data</a></li>
  <li><a href="#tnbc-and-er" id="toc-tnbc-and-er" class="nav-link" data-scroll-target="#tnbc-and-er"><span class="header-section-number">8.8</span> TNBC and ER+</a></li>
  <li><a href="#scan-b-library-protocol" id="toc-scan-b-library-protocol" class="nav-link" data-scroll-target="#scan-b-library-protocol"><span class="header-section-number">8.9</span> SCAN-B library protocol</a></li>
  <li><a href="#scaling-before-doing-pca" id="toc-scaling-before-doing-pca" class="nav-link" data-scroll-target="#scaling-before-doing-pca"><span class="header-section-number">8.10</span> Scaling before doing PCA</a></li>
  </ul>
<div class="toc-actions"><div><i class="bi bi-github"></i></div><div class="action-links"><p><a href="https://github.com/chronchi/molecular_landscape/issues/new" class="toc-action">Report an issue</a></p></div></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Revision</span></h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>* The library is already synchronized with the lockfile.</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>    ../../results/plots/revision ../../results/rds_files/revision 
                           FALSE                            FALSE 
   ../../results/tables/revision 
                           FALSE </code></pre>
</div>
</div>
<section id="comparing-to-combats-performance" class="level2" data-number="8.1">
<h2 data-number="8.1" class="anchored" data-anchor-id="comparing-to-combats-performance"><span class="header-section-number">8.1</span> Comparing to combat’s performance</h2>
<p>We know that the first two components of PCA correspond to cohort and platform specific effects. We can apply combat from the sva package on the unnormalized data.</p>
<p>We now check what is the relationship between the PC1 and PC2 from combat to the actual PCs from EMBER.</p>
<p>And the spearman correlation is for PC3 and PC1:</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-combat-comparison" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="revision_files/figure-html/fig-combat-comparison-1.png" class="img-fluid figure-img" width="960"></p>
<figcaption class="figure-caption">Figure&nbsp;8.1: <strong>?(caption)</strong></figcaption>
</figure>
</div>
</div>
</div>
<p>In a nutshell, the EMBER position corresponds to what is obtained by combat, with the advantage that we can actually use the embedding for new samples. Combat is restricted to the samples inputted.</p>
</section>
<section id="autoencoders" class="level2" data-number="8.2">
<h2 data-number="8.2" class="anchored" data-anchor-id="autoencoders"><span class="header-section-number">8.2</span> Autoencoders</h2>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
H2O is not running yet, starting it now...

Note:  In case of errors look at the following log files:
    /tmp/RtmpX13bqx/file1d3618472a9b32/h2o_ronchi_started_from_r.out
    /tmp/RtmpX13bqx/file1d361857c668d1/h2o_ronchi_started_from_r.err


Starting H2O JVM and connecting: .. Connection successful!

R is connected to the H2O cluster: 
    H2O cluster uptime:         1 seconds 274 milliseconds 
    H2O cluster timezone:       Europe/Zurich 
    H2O data parsing timezone:  UTC 
    H2O cluster version:        3.44.0.3 
    H2O cluster version age:    4 months and 19 days 
    H2O cluster name:           H2O_started_from_R_ronchi_xld624 
    H2O cluster total nodes:    1 
    H2O cluster total memory:   20.00 GB 
    H2O cluster total cores:    32 
    H2O cluster allowed cores:  32 
    H2O cluster healthy:        TRUE 
    H2O Connection ip:          localhost 
    H2O Connection port:        54321 
    H2O Connection proxy:       NA 
    H2O Internal Security:      FALSE 
    R Version:                  R version 4.2.0 (2022-04-22) </code></pre>
</div>
</div>
<p>Features are the columns and the rows are the samples. So for the genomic data we have to transpose it.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-first-autoencoder" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="revision_files/figure-html/fig-first-autoencoder-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Figure&nbsp;8.2: <strong>?(caption)</strong></figcaption>
</figure>
</div>
</div>
</div>
<p>Just using the autoencoder in the data does not work.</p>
<p>We now use on the normalized data instead on 4 units in the hidden layer.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-second-ae-cohort" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="revision_files/figure-html/fig-second-ae-cohort-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Figure&nbsp;8.3: <strong>?(caption)</strong></figcaption>
</figure>
</div>
</div>
</div>
<p>And now coloring by er status. It is not really able to completely distinguish the ER+ and ER- samples.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-second-ae-erstatus" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="revision_files/figure-html/fig-second-ae-erstatus-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Figure&nbsp;8.4: <strong>?(caption)</strong></figcaption>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="cell-output-display">
<div id="fig-ae-4-units-ae" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="revision_files/figure-html/fig-ae-4-units-ae-1.png" class="img-fluid figure-img" width="1152"></p>
<figcaption class="figure-caption">Figure&nbsp;8.5: <strong>?(caption)</strong></figcaption>
</figure>
</div>
</div>
</div>
<p>We do a pairsplot coloring by cohort to see where the distinction between cohorts is coming from.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-ggpairs-4-units-ae-cohort" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="revision_files/figure-html/fig-ggpairs-4-units-ae-cohort-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Figure&nbsp;8.6: <strong>?(caption)</strong></figcaption>
</figure>
</div>
</div>
</div>
<p>And now coloring by ER status instead.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-ggpairs-4-units-ae-erihc" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="revision_files/figure-html/fig-ggpairs-4-units-ae-erihc-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Figure&nbsp;8.7: <strong>?(caption)</strong></figcaption>
</figure>
</div>
</div>
</div>
<p>Perhaps now if we include more layers instead and use 4 components.</p>
<p>In terms of mean squared error all the models are very similar to each other. And based on the previous results it seems that by using</p>
</section>
<section id="mixing-analysis" class="level2" data-number="8.3">
<h2 data-number="8.3" class="anchored" data-anchor-id="mixing-analysis"><span class="header-section-number">8.3</span> Mixing analysis</h2>
<p>In this section we try to understand what is the minimum number of genes necessary to be able to distinguish ER+ and ER- BC samples. We select the top 10, 25, 50, 100, 200, 500, 750, 1000, 2500, 5000 and all the genes and try to classify the samples bewteen ER+ and ER- based on their position of the third component. This should give an idea of how good the embedding is based on the number of genes.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pca-nb-genes" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="revision_files/figure-html/fig-pca-nb-genes-1.png" class="img-fluid figure-img" width="2688"></p>
<figcaption class="figure-caption">Figure&nbsp;8.8: <strong>?(caption)</strong></figcaption>
</figure>
</div>
</div>
</div>
<p>Using less than 25 genes mixes well the cohorts in the first two components. When using more than 50 genes the mixing is already lost in the first two components. So what we actually want is a mixing in just one of the pairs of components, either PC1+PC2 or PC3+PC4 while there is no mixing in the other pair of component. By analysing visually this happens when using at least 1000 genes.</p>
<p>Due to the difference in cohort sizes, METABRIC has 648 samples and TCGA 352, the expected iLISI coefficient is:</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1] 1.84</code></pre>
</div>
</div>
<p>The expected value for the iLISI coefficient when calculating for ER status is:</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1] 1.59</code></pre>
</div>
</div>
<p>And the plot below shows the iLISI numbers for each combination of PC1+PC2 and PC3+PC4 annotated by the total number of genes. An integration by ER status means that the combination of components is not capturing the biology.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-lisi-scores" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="revision_files/figure-html/fig-lisi-scores-1.png" class="img-fluid figure-img" width="768"></p>
<figcaption class="figure-caption">Figure&nbsp;8.9: <strong>?(caption)</strong></figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="stable-genes" class="level2" data-number="8.4">
<h2 data-number="8.4" class="anchored" data-anchor-id="stable-genes"><span class="header-section-number">8.4</span> Stable genes</h2>
<p>One of the raised issues is that the average ranking of the stable genes might depend on phenotypes. Here we show that the distribution of average of stable genes is stable across different phenotypes, such as ER status and molecular subtype.</p>
<p>First we plot by ER status in both METABRIC and TCGA.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-avg-rank-er-ihc" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="revision_files/figure-html/fig-avg-rank-er-ihc-1.png" class="img-fluid figure-img" width="384"></p>
<figcaption class="figure-caption">Figure&nbsp;8.10: <strong>?(caption)</strong></figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="scaling" class="level2" data-number="8.5">
<h2 data-number="8.5" class="anchored" data-anchor-id="scaling"><span class="header-section-number">8.5</span> Scaling</h2>
<p>We now show the embedding using no scaling, i.e., just log FPKM and median intensity, using the qPCR like normalization and using a z-normalization on the sample level.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-comparison-plots" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="revision_files/figure-html/fig-comparison-plots-1.png" class="img-fluid figure-img" width="1920"></p>
<figcaption class="figure-caption">Figure&nbsp;8.11: <strong>?(caption)</strong></figcaption>
</figure>
</div>
</div>
</div>
<p>And now we compare the iLISI scores of the qPCR-like normalization and z-scaling as well.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-differences-scaling" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="revision_files/figure-html/fig-differences-scaling-1.png" class="img-fluid figure-img" width="480"></p>
<figcaption class="figure-caption">Figure&nbsp;8.12: <strong>?(caption)</strong></figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="tumor-purity-statistical-tests" class="level2" data-number="8.6">
<h2 data-number="8.6" class="anchored" data-anchor-id="tumor-purity-statistical-tests"><span class="header-section-number">8.6</span> Tumor purity: statistical tests</h2>
<p>One of the points raised by the reviewers was about the statistics on the differences between the principal components stratified by molecular subtype. We perform ANOVA followed by a Tukey post hoc test.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-tumor-purity-metabric" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="revision_files/figure-html/fig-tumor-purity-metabric-1.png" class="img-fluid figure-img" width="1344"></p>
<figcaption class="figure-caption">Figure&nbsp;8.13: <strong>?(caption)</strong></figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="metastasis-and-ffpe-data" class="level2" data-number="8.7">
<h2 data-number="8.7" class="anchored" data-anchor-id="metastasis-and-ffpe-data"><span class="header-section-number">8.7</span> Metastasis and FFPE data</h2>
<p>Accessed 20240425: https://www.cbioportal.org/study/summary?id=brca_mbcproject_2022</p>
<p>In the cBioportal one can check the clinical data directly with all the information. When hovering over the column names it is possible to obtain the column names from the data available when downloaded. For example, the ER percentage is available from the sample in the diagnostics. Ki67, similarly. In general the metastasis do not have ER percentage.</p>
<div class="cell">
<div class="cell-output-display">

<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-7ffa7b1a3ebc903a81b6" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-7ffa7b1a3ebc903a81b6">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13"],["ASCENDING COLON - FALLOPIAN TUBE","AXILLARY LYMPH NODE - AXILLARY LYMPH NODE - LYMPH NODE DISTANT","AXILLARY LYMPH NODE - BREAST","AXILLARY LYMPH NODE - BREAST - BREAST","AXILLARY LYMPH NODE - BREAST - PLEURAL FLUID","BONE - BREAST","BRAIN - BREAST","BREAST - BREAST","BREAST - BREAST - BREAST","BREAST - BREAST - LIVER","BREAST - BREAST SKIN","BREAST - CHEST WALL","BREAST - LIVER"],[1,1,2,1,1,2,1,11,4,1,1,2,1],[0.03448275862068965,0.03448275862068965,0.06896551724137931,0.03448275862068965,0.03448275862068965,0.06896551724137931,0.03448275862068965,0.3793103448275862,0.1379310344827586,0.03448275862068965,0.03448275862068965,0.06896551724137931,0.03448275862068965]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>matched_samples<\/th>\n      <th>n<\/th>\n      <th>percent<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<p>We now try to understand if samples are coming from core biopsy, surgical procedures or from a metastasis setting. To start with that, we look at a single patient ID that has two breast samples.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>  BX_TIME_DAYS CALC_TREATMENT_NAIVE    BX_TYPE
1            0                  YES       CORE
2          114                   NO MASTECTOMY</code></pre>
</div>
</div>
<p>For patient with ID <code>MBCProject_4MF1FlFQ</code>, there are two samples. One at time of diagnosis, core biopsy, and the other at a mastectomy after 114 days of treatment. The column CALC_TREATMENT_NAIVE tells if the sample is treatment naive or not, BX_TYPE corresponds to the type of surgery (PATH Procedure Type column on cBioPortal).</p>
<p>We now look at another patient (<code>MBCProject_5gHasou8</code>):</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>  BX_TIME_DAYS CALC_TREATMENT_NAIVE    BX_TYPE
1            0                  YES       CORE
2          199                   NO MASTECTOMY</code></pre>
</div>
</div>
<p>The results are similar for this patient too. We now move on and use EMBER on all available samples and then compare these matched samples.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>Total number of stable genes: 44
Total number of genes: 1037
Number of samples: 153</code></pre>
</div>
</div>
<p>We show all the samples from the project below on top of TCGA, SCAN-B and METABRIC.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-er-ihc-mets-project" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="revision_files/figure-html/fig-er-ihc-mets-project-1.png" class="img-fluid figure-img" width="864"></p>
<figcaption class="figure-caption">Figure&nbsp;8.14: <strong>?(caption)</strong></figcaption>
</figure>
</div>
</div>
</div>
<p>And now stratified by either local metastasis/primary sample or by distant metastasis.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-mets-local-distant" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="revision_files/figure-html/fig-mets-local-distant-1.png" class="img-fluid figure-img" width="1152"></p>
<figcaption class="figure-caption">Figure&nbsp;8.15: <strong>?(caption)</strong></figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="tnbc-and-er" class="level2" data-number="8.8">
<h2 data-number="8.8" class="anchored" data-anchor-id="tnbc-and-er"><span class="header-section-number">8.8</span> TNBC and ER+</h2>
<p>Here we use another cohort with both ER+ and ER- samples coming from FFPE with two different library preparation protocols: Illumina-Access and Nugen-Ovation.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>Total number of stable genes: 44
Total number of genes: 1022
Number of samples: 21</code></pre>
</div>
</div>
<p>We now show the samples highlighted on top of EMBER space with samples from TCGA and METABRIC.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-illumina-nugen" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="revision_files/figure-html/fig-illumina-nugen-1.png" class="img-fluid figure-img" width="1056"></p>
<figcaption class="figure-caption">Figure&nbsp;8.16: <strong>?(caption)</strong></figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="scan-b-library-protocol" class="level2" data-number="8.9">
<h2 data-number="8.9" class="anchored" data-anchor-id="scan-b-library-protocol"><span class="header-section-number">8.9</span> SCAN-B library protocol</h2>
<p>SCAN-B has 3 different types of samples based on their library protocol, namely: dUTP, NeoPrep and TruSeq.</p>
<p>Here we show that EMBER is capable of fully removing any effects associated to library protocol.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-scanb-libprep" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="revision_files/figure-html/fig-scanb-libprep-1.png" class="img-fluid figure-img" width="480"></p>
<figcaption class="figure-caption">Figure&nbsp;8.17: <strong>?(caption)</strong></figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="scaling-before-doing-pca" class="level2" data-number="8.10">
<h2 data-number="8.10" class="anchored" data-anchor-id="scaling-before-doing-pca"><span class="header-section-number">8.10</span> Scaling before doing PCA</h2>
<p>When developing EMBER we did not center or scale the data, as the gene expressions after normalization all had the same range. Here we compare what happens when doing the centering followed by the scaling and then applying PCA.</p>
<p>To understand the effect of centering and scaling the data prior to applying PCA, we compared the loadings of PC1 from EMBER to the average of the genes in the normalized data (a). The correlation between these two variables is -1, showing that PC1 corresponds to centering. When comparing the loadings with PC2 and the standard deviation of genes after centering, the higher loadings in absolute value are correlated with higher standard deviation (b).</p>
<p>Next, we show that PC1 and PC2 are correlated with PC1 obtained after centering and scaling the data (c), showing that EMBER’s PC1 and PC2 is automatically centering and scaling the data at the same time removing batch effects. In order to confirm that no biological information is lost when performing PCA on the uncentered and unscaled data, we compared EMBER’s PC3 and PC4 with PC2 and PC3 from the scaled data respectively (d and e). The correlation is 1 in both cases for both cohorts: METABRIC and TCGA. We show that there is no need to center and scale prior to applying PCA in EMBER’s context.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pca-center-scaled" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="revision_files/figure-html/fig-pca-center-scaled-1.png" class="img-fluid figure-img" width="1440"></p>
<figcaption class="figure-caption">Figure&nbsp;8.18: <strong>?(caption)</strong></figcaption>
</figure>
</div>
</div>
</div>


<!-- -->

</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="../chapters/trying.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Tests and tries</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../chapters/discussion.html" class="pagination-link">
        <span class="nav-page-text">Discussion</span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb10" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu"># Revision</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>renv<span class="sc">::</span><span class="fu">restore</span>()</span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggprism)</span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(PCAtools)</span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(singscore)</span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SummarizedExperiment)</span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(survival)</span>
<span id="cb10-17"><a href="#cb10-17" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(genefu)</span>
<span id="cb10-18"><a href="#cb10-18" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(caret)</span>
<span id="cb10-19"><a href="#cb10-19" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(sva)</span>
<span id="cb10-20"><a href="#cb10-20" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(lisi)</span>
<span id="cb10-21"><a href="#cb10-21" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(rstanarm)</span>
<span id="cb10-22"><a href="#cb10-22" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(rstatix)</span>
<span id="cb10-23"><a href="#cb10-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-24"><a href="#cb10-24" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb10-25"><a href="#cb10-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-26"><a href="#cb10-26" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(h2o)</span>
<span id="cb10-27"><a href="#cb10-27" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dslabs)</span>
<span id="cb10-28"><a href="#cb10-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-29"><a href="#cb10-29" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"../R/utils.R"</span>)</span>
<span id="cb10-30"><a href="#cb10-30" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"../R/first_run.R"</span>)</span>
<span id="cb10-31"><a href="#cb10-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-32"><a href="#cb10-32" aria-hidden="true" tabindex="-1"></a><span class="co"># the following script load all data necessary to run the chunks.</span></span>
<span id="cb10-33"><a href="#cb10-33" aria-hidden="true" tabindex="-1"></a><span class="co"># the data is generated from this quarto document itself, therefore</span></span>
<span id="cb10-34"><a href="#cb10-34" aria-hidden="true" tabindex="-1"></a><span class="co"># if you are running this documents the first time and don't have the</span></span>
<span id="cb10-35"><a href="#cb10-35" aria-hidden="true" tabindex="-1"></a><span class="co"># files, comment the following lines. Moreover, if this is your first</span></span>
<span id="cb10-36"><a href="#cb10-36" aria-hidden="true" tabindex="-1"></a><span class="co"># time running the document, you should run all chunks, to generate </span></span>
<span id="cb10-37"><a href="#cb10-37" aria-hidden="true" tabindex="-1"></a><span class="co"># all the necessary files, if you don't have them. Once all files </span></span>
<span id="cb10-38"><a href="#cb10-38" aria-hidden="true" tabindex="-1"></a><span class="co"># are saved and available in the respective folder, the following</span></span>
<span id="cb10-39"><a href="#cb10-39" aria-hidden="true" tabindex="-1"></a><span class="co"># lines can be executed. </span></span>
<span id="cb10-40"><a href="#cb10-40" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (first_run){</span>
<span id="cb10-41"><a href="#cb10-41" aria-hidden="true" tabindex="-1"></a>    load_at_setup <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb10-42"><a href="#cb10-42" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb10-43"><a href="#cb10-43" aria-hidden="true" tabindex="-1"></a>    load_at_setup <span class="ot">&lt;-</span> <span class="cn">TRUE</span></span>
<span id="cb10-44"><a href="#cb10-44" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb10-45"><a href="#cb10-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-46"><a href="#cb10-46" aria-hidden="true" tabindex="-1"></a>name_document <span class="ot">&lt;-</span> <span class="st">"revision"</span></span>
<span id="cb10-47"><a href="#cb10-47" aria-hidden="true" tabindex="-1"></a><span class="fu">sapply</span>(</span>
<span id="cb10-48"><a href="#cb10-48" aria-hidden="true" tabindex="-1"></a>    <span class="fu">paste0</span>(<span class="st">"../../results/"</span>, <span class="fu">c</span>(<span class="st">"plots"</span>, <span class="st">"rds_files"</span>, <span class="st">"tables"</span>), <span class="st">"/"</span>, name_document),</span>
<span id="cb10-49"><a href="#cb10-49" aria-hidden="true" tabindex="-1"></a>    dir.create,</span>
<span id="cb10-50"><a href="#cb10-50" aria-hidden="true" tabindex="-1"></a>    <span class="at">showWarnings =</span> <span class="cn">FALSE</span>,</span>
<span id="cb10-51"><a href="#cb10-51" aria-hidden="true" tabindex="-1"></a>    <span class="at">recursive =</span> <span class="cn">TRUE</span></span>
<span id="cb10-52"><a href="#cb10-52" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-53"><a href="#cb10-53" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"../R/load_rds_files.R"</span>)</span>
<span id="cb10-54"><a href="#cb10-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-55"><a href="#cb10-55" aria-hidden="true" tabindex="-1"></a><span class="co"># by setting the dev to png and pdf, this saves the figures in a specific</span></span>
<span id="cb10-56"><a href="#cb10-56" aria-hidden="true" tabindex="-1"></a><span class="co"># folder in both formats. moreover, since png is coming first, it shows</span></span>
<span id="cb10-57"><a href="#cb10-57" aria-hidden="true" tabindex="-1"></a><span class="co"># this figure when rendering the html. What is nice about this is that it</span></span>
<span id="cb10-58"><a href="#cb10-58" aria-hidden="true" tabindex="-1"></a><span class="co"># inherits the properties from the chunk to save the figure, so no need</span></span>
<span id="cb10-59"><a href="#cb10-59" aria-hidden="true" tabindex="-1"></a><span class="co"># to use ggplot2::ggsave to save the plots. this also works </span></span>
<span id="cb10-60"><a href="#cb10-60" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">dev =</span> <span class="fu">c</span>(<span class="st">'png'</span>, <span class="st">'pdf'</span>))</span>
<span id="cb10-61"><a href="#cb10-61" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">bitmapType =</span> <span class="st">'cairo'</span>)</span>
<span id="cb10-62"><a href="#cb10-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-63"><a href="#cb10-63" aria-hidden="true" tabindex="-1"></a>datasets<span class="sc">$</span>poetic<span class="sc">$</span>sample_name <span class="ot">&lt;-</span> <span class="fu">colnames</span>(datasets<span class="sc">$</span>poetic)</span>
<span id="cb10-64"><a href="#cb10-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-65"><a href="#cb10-65" aria-hidden="true" tabindex="-1"></a>mol_subs <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb10-66"><a href="#cb10-66" aria-hidden="true" tabindex="-1"></a>    <span class="st">"basal"</span> <span class="ot">=</span> <span class="st">"Basal-like"</span>, </span>
<span id="cb10-67"><a href="#cb10-67" aria-hidden="true" tabindex="-1"></a>    <span class="st">"her2"</span><span class="ot">=</span> <span class="st">"HER2-enriched"</span>, </span>
<span id="cb10-68"><a href="#cb10-68" aria-hidden="true" tabindex="-1"></a>    <span class="st">"lumb"</span> <span class="ot">=</span> <span class="st">"LumB"</span>, </span>
<span id="cb10-69"><a href="#cb10-69" aria-hidden="true" tabindex="-1"></a>    <span class="st">"luma"</span> <span class="ot">=</span> <span class="st">"LumA"</span>, </span>
<span id="cb10-70"><a href="#cb10-70" aria-hidden="true" tabindex="-1"></a>    <span class="st">"normal"</span> <span class="ot">=</span> <span class="st">"Normal-like"</span>,</span>
<span id="cb10-71"><a href="#cb10-71" aria-hidden="true" tabindex="-1"></a>    <span class="st">"claudin-low"</span> <span class="ot">=</span> <span class="st">"Claudin-low"</span></span>
<span id="cb10-72"><a href="#cb10-72" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-73"><a href="#cb10-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-74"><a href="#cb10-74" aria-hidden="true" tabindex="-1"></a>ggplot2<span class="sc">::</span><span class="fu">theme_set</span>(<span class="at">new =</span> ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>())</span>
<span id="cb10-75"><a href="#cb10-75" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-76"><a href="#cb10-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-77"><a href="#cb10-77" aria-hidden="true" tabindex="-1"></a><span class="fu">## Comparing to combat's performance</span></span>
<span id="cb10-78"><a href="#cb10-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-79"><a href="#cb10-79" aria-hidden="true" tabindex="-1"></a>We know that the first two components of PCA correspond to cohort and platform</span>
<span id="cb10-80"><a href="#cb10-80" aria-hidden="true" tabindex="-1"></a>specific effects. We can apply combat from the sva package on the unnormalized</span>
<span id="cb10-81"><a href="#cb10-81" aria-hidden="true" tabindex="-1"></a>data.</span>
<span id="cb10-82"><a href="#cb10-82" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-85"><a href="#cb10-85" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-86"><a href="#cb10-86" aria-hidden="true" tabindex="-1"></a>all_datasets <span class="ot">&lt;-</span> <span class="fu">mapply</span>(</span>
<span id="cb10-87"><a href="#cb10-87" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(sum_exp, which_exp){</span>
<span id="cb10-88"><a href="#cb10-88" aria-hidden="true" tabindex="-1"></a>        SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(sum_exp, which_exp)[</span>
<span id="cb10-89"><a href="#cb10-89" aria-hidden="true" tabindex="-1"></a>            <span class="fu">rownames</span>(pca_fit<span class="sc">$</span>loadings), </span>
<span id="cb10-90"><a href="#cb10-90" aria-hidden="true" tabindex="-1"></a>        ]</span>
<span id="cb10-91"><a href="#cb10-91" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb10-92"><a href="#cb10-92" aria-hidden="true" tabindex="-1"></a>    <span class="at">sum_exp =</span> datasets[<span class="fu">c</span>(<span class="st">"tcga"</span>, <span class="st">"metabric"</span>)],</span>
<span id="cb10-93"><a href="#cb10-93" aria-hidden="true" tabindex="-1"></a>    <span class="at">which_exp =</span> which_exp[<span class="fu">c</span>(<span class="st">"tcga"</span>, <span class="st">"metabric"</span>)]</span>
<span id="cb10-94"><a href="#cb10-94" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-95"><a href="#cb10-95" aria-hidden="true" tabindex="-1"></a>    <span class="fu">Reduce</span>(dplyr<span class="sc">::</span>bind_cols, <span class="at">x =</span> .)</span>
<span id="cb10-96"><a href="#cb10-96" aria-hidden="true" tabindex="-1"></a>training_dataset <span class="ot">&lt;-</span> all_datasets[, <span class="fu">rownames</span>(pca_fit<span class="sc">$</span>rotated)]</span>
<span id="cb10-97"><a href="#cb10-97" aria-hidden="true" tabindex="-1"></a>batch_training <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(</span>
<span id="cb10-98"><a href="#cb10-98" aria-hidden="true" tabindex="-1"></a>    stringr<span class="sc">::</span><span class="fu">str_starts</span>(<span class="fu">colnames</span>(training_dataset), <span class="st">"TCGA"</span>),</span>
<span id="cb10-99"><a href="#cb10-99" aria-hidden="true" tabindex="-1"></a>    <span class="st">"tcga"</span>,</span>
<span id="cb10-100"><a href="#cb10-100" aria-hidden="true" tabindex="-1"></a>    <span class="st">"metabric"</span></span>
<span id="cb10-101"><a href="#cb10-101" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-102"><a href="#cb10-102" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-103"><a href="#cb10-103" aria-hidden="true" tabindex="-1"></a>combat_training <span class="ot">&lt;-</span> sva<span class="sc">::</span><span class="fu">ComBat</span>(training_dataset, batch_training)</span>
<span id="cb10-104"><a href="#cb10-104" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-105"><a href="#cb10-105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-106"><a href="#cb10-106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-109"><a href="#cb10-109" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-110"><a href="#cb10-110" aria-hidden="true" tabindex="-1"></a>pca_out_combat <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">pca</span>(</span>
<span id="cb10-111"><a href="#cb10-111" aria-hidden="true" tabindex="-1"></a>    combat_training, </span>
<span id="cb10-112"><a href="#cb10-112" aria-hidden="true" tabindex="-1"></a>    <span class="at">metadata =</span> <span class="fu">data.frame</span>(</span>
<span id="cb10-113"><a href="#cb10-113" aria-hidden="true" tabindex="-1"></a>        <span class="at">sample_name =</span> <span class="fu">colnames</span>(combat_training)</span>
<span id="cb10-114"><a href="#cb10-114" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb10-115"><a href="#cb10-115" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">inner_join</span>(., df_pca, <span class="at">by =</span> <span class="st">"sample_name"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-116"><a href="#cb10-116" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">rownames&lt;-</span><span class="st">`</span>(.<span class="sc">$</span>sample_name) <span class="sc">%&gt;%</span></span>
<span id="cb10-117"><a href="#cb10-117" aria-hidden="true" tabindex="-1"></a>        .[<span class="fu">colnames</span>(combat_training), ]</span>
<span id="cb10-118"><a href="#cb10-118" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-119"><a href="#cb10-119" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-120"><a href="#cb10-120" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-121"><a href="#cb10-121" aria-hidden="true" tabindex="-1"></a>We now check what is the relationship between the PC1 and PC2 from </span>
<span id="cb10-122"><a href="#cb10-122" aria-hidden="true" tabindex="-1"></a>combat to the actual PCs from EMBER. </span>
<span id="cb10-123"><a href="#cb10-123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-126"><a href="#cb10-126" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-127"><a href="#cb10-127" aria-hidden="true" tabindex="-1"></a>combat_results <span class="ot">&lt;-</span> pca_out_combat<span class="sc">$</span>metadata</span>
<span id="cb10-128"><a href="#cb10-128" aria-hidden="true" tabindex="-1"></a>combat_results[, <span class="fu">paste0</span>(<span class="st">"combat_PC"</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">1000</span>)] <span class="ot">&lt;-</span> pca_out_combat<span class="sc">$</span>rotated</span>
<span id="cb10-129"><a href="#cb10-129" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-130"><a href="#cb10-130" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-131"><a href="#cb10-131" aria-hidden="true" tabindex="-1"></a>And the spearman correlation is for PC3 and PC1: </span>
<span id="cb10-132"><a href="#cb10-132" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-135"><a href="#cb10-135" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-136"><a href="#cb10-136" aria-hidden="true" tabindex="-1"></a>pc3_cor <span class="ot">&lt;-</span> <span class="fu">cor.test</span>(</span>
<span id="cb10-137"><a href="#cb10-137" aria-hidden="true" tabindex="-1"></a>    combat_results<span class="sc">$</span>PC3,</span>
<span id="cb10-138"><a href="#cb10-138" aria-hidden="true" tabindex="-1"></a>    combat_results<span class="sc">$</span>combat_PC1,</span>
<span id="cb10-139"><a href="#cb10-139" aria-hidden="true" tabindex="-1"></a>    <span class="at">method =</span> <span class="st">"spearman"</span></span>
<span id="cb10-140"><a href="#cb10-140" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-141"><a href="#cb10-141" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-142"><a href="#cb10-142" aria-hidden="true" tabindex="-1"></a>pc4_cor <span class="ot">&lt;-</span> <span class="fu">cor.test</span>(</span>
<span id="cb10-143"><a href="#cb10-143" aria-hidden="true" tabindex="-1"></a>    combat_results<span class="sc">$</span>PC4,</span>
<span id="cb10-144"><a href="#cb10-144" aria-hidden="true" tabindex="-1"></a>    combat_results<span class="sc">$</span>combat_PC2,</span>
<span id="cb10-145"><a href="#cb10-145" aria-hidden="true" tabindex="-1"></a>    <span class="at">method =</span> <span class="st">"spearman"</span></span>
<span id="cb10-146"><a href="#cb10-146" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-147"><a href="#cb10-147" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-148"><a href="#cb10-148" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-149"><a href="#cb10-149" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height = 4}</span></span>
<span id="cb10-150"><a href="#cb10-150" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-combat-comparison</span></span>
<span id="cb10-151"><a href="#cb10-151" aria-hidden="true" tabindex="-1"></a>pc3_comp <span class="ot">&lt;-</span> combat_results <span class="sc">%&gt;%</span></span>
<span id="cb10-152"><a href="#cb10-152" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> PC3, <span class="at">y =</span> combat_PC1, <span class="at">color =</span> cohort)) <span class="sc">+</span> </span>
<span id="cb10-153"><a href="#cb10-153" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span> </span>
<span id="cb10-154"><a href="#cb10-154" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_smooth</span>(</span>
<span id="cb10-155"><a href="#cb10-155" aria-hidden="true" tabindex="-1"></a>        <span class="at">method =</span> <span class="st">"lm"</span>,</span>
<span id="cb10-156"><a href="#cb10-156" aria-hidden="true" tabindex="-1"></a>        <span class="at">formula =</span> <span class="st">"y~x"</span></span>
<span id="cb10-157"><a href="#cb10-157" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb10-158"><a href="#cb10-158" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">15</span>) <span class="sc">+</span> </span>
<span id="cb10-159"><a href="#cb10-159" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">annotate</span>(</span>
<span id="cb10-160"><a href="#cb10-160" aria-hidden="true" tabindex="-1"></a>        <span class="at">geom =</span> <span class="st">"label"</span>, </span>
<span id="cb10-161"><a href="#cb10-161" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="dv">3</span>, </span>
<span id="cb10-162"><a href="#cb10-162" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="dv">30</span>, </span>
<span id="cb10-163"><a href="#cb10-163" aria-hidden="true" tabindex="-1"></a>        <span class="at">label =</span> <span class="fu">paste0</span>(<span class="st">"Cor: "</span>, <span class="fu">format</span>(pc3_cor<span class="sc">$</span>estimate, <span class="at">digits =</span> <span class="dv">2</span>))</span>
<span id="cb10-164"><a href="#cb10-164" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb10-165"><a href="#cb10-165" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb10-166"><a href="#cb10-166" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"EMBER PC3"</span>,</span>
<span id="cb10-167"><a href="#cb10-167" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"Combat PC1"</span></span>
<span id="cb10-168"><a href="#cb10-168" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb10-169"><a href="#cb10-169" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb10-170"><a href="#cb10-170" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-171"><a href="#cb10-171" aria-hidden="true" tabindex="-1"></a>pc4_comp <span class="ot">&lt;-</span> combat_results <span class="sc">%&gt;%</span></span>
<span id="cb10-172"><a href="#cb10-172" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">cohort =</span> <span class="fu">toupper</span>(cohort)) <span class="sc">%&gt;%</span> </span>
<span id="cb10-173"><a href="#cb10-173" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> PC4, <span class="at">y =</span> combat_PC2, <span class="at">color =</span> cohort)) <span class="sc">+</span> </span>
<span id="cb10-174"><a href="#cb10-174" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span> </span>
<span id="cb10-175"><a href="#cb10-175" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_smooth</span>(</span>
<span id="cb10-176"><a href="#cb10-176" aria-hidden="true" tabindex="-1"></a>        <span class="at">method =</span> <span class="st">"lm"</span>,</span>
<span id="cb10-177"><a href="#cb10-177" aria-hidden="true" tabindex="-1"></a>        <span class="at">formula =</span> <span class="st">"y~x"</span></span>
<span id="cb10-178"><a href="#cb10-178" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb10-179"><a href="#cb10-179" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">annotate</span>(</span>
<span id="cb10-180"><a href="#cb10-180" aria-hidden="true" tabindex="-1"></a>        <span class="at">geom =</span> <span class="st">"label"</span>, </span>
<span id="cb10-181"><a href="#cb10-181" aria-hidden="true" tabindex="-1"></a>        <span class="at">x=</span><span class="dv">3</span>, </span>
<span id="cb10-182"><a href="#cb10-182" aria-hidden="true" tabindex="-1"></a>        <span class="at">y=</span><span class="dv">30</span>, </span>
<span id="cb10-183"><a href="#cb10-183" aria-hidden="true" tabindex="-1"></a>        <span class="at">label =</span> <span class="fu">paste0</span>(<span class="st">"Cor: "</span>, <span class="fu">format</span>(pc4_cor<span class="sc">$</span>estimate, <span class="at">digits =</span> <span class="dv">2</span>))</span>
<span id="cb10-184"><a href="#cb10-184" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb10-185"><a href="#cb10-185" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb10-186"><a href="#cb10-186" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span>  <span class="st">"Cohort"</span>,</span>
<span id="cb10-187"><a href="#cb10-187" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"EMBER PC4"</span>,</span>
<span id="cb10-188"><a href="#cb10-188" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"Combat PC2"</span></span>
<span id="cb10-189"><a href="#cb10-189" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb10-190"><a href="#cb10-190" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">15</span>) </span>
<span id="cb10-191"><a href="#cb10-191" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-192"><a href="#cb10-192" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(pc3_comp, pc4_comp, <span class="at">rel_widths =</span> <span class="fu">c</span>(<span class="fl">0.75</span>, <span class="dv">1</span>))</span>
<span id="cb10-193"><a href="#cb10-193" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-194"><a href="#cb10-194" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-195"><a href="#cb10-195" aria-hidden="true" tabindex="-1"></a>In a nutshell, the EMBER position corresponds to what is obtained by combat,</span>
<span id="cb10-196"><a href="#cb10-196" aria-hidden="true" tabindex="-1"></a>with the advantage that we can actually use the embedding for new samples. </span>
<span id="cb10-197"><a href="#cb10-197" aria-hidden="true" tabindex="-1"></a>Combat is restricted to the samples inputted.</span>
<span id="cb10-198"><a href="#cb10-198" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-199"><a href="#cb10-199" aria-hidden="true" tabindex="-1"></a><span class="fu">## Autoencoders</span></span>
<span id="cb10-200"><a href="#cb10-200" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-203"><a href="#cb10-203" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-204"><a href="#cb10-204" aria-hidden="true" tabindex="-1"></a><span class="fu">h2o.no_progress</span>()  <span class="co"># turn off progress bars</span></span>
<span id="cb10-205"><a href="#cb10-205" aria-hidden="true" tabindex="-1"></a><span class="fu">h2o.init</span>(<span class="at">max_mem_size =</span> <span class="st">"20g"</span>)  <span class="co"># initialize H2O instance</span></span>
<span id="cb10-206"><a href="#cb10-206" aria-hidden="true" tabindex="-1"></a><span class="fu">h2o.removeAll</span>()</span>
<span id="cb10-207"><a href="#cb10-207" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-208"><a href="#cb10-208" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-209"><a href="#cb10-209" aria-hidden="true" tabindex="-1"></a>Features are the columns and the rows are the samples. So for the </span>
<span id="cb10-210"><a href="#cb10-210" aria-hidden="true" tabindex="-1"></a>genomic data we have to transpose it.</span>
<span id="cb10-211"><a href="#cb10-211" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-214"><a href="#cb10-214" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-215"><a href="#cb10-215" aria-hidden="true" tabindex="-1"></a>features <span class="ot">&lt;-</span> <span class="fu">as.h2o</span>(training_dataset <span class="sc">%&gt;%</span> t)</span>
<span id="cb10-216"><a href="#cb10-216" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-217"><a href="#cb10-217" aria-hidden="true" tabindex="-1"></a>ae1 <span class="ot">&lt;-</span> <span class="fu">h2o.deeplearning</span>(</span>
<span id="cb10-218"><a href="#cb10-218" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="fu">seq_along</span>(features),</span>
<span id="cb10-219"><a href="#cb10-219" aria-hidden="true" tabindex="-1"></a>    <span class="at">training_frame =</span> features,</span>
<span id="cb10-220"><a href="#cb10-220" aria-hidden="true" tabindex="-1"></a>    <span class="at">autoencoder =</span> <span class="cn">TRUE</span>,</span>
<span id="cb10-221"><a href="#cb10-221" aria-hidden="true" tabindex="-1"></a>    <span class="at">hidden =</span> <span class="dv">2</span>,</span>
<span id="cb10-222"><a href="#cb10-222" aria-hidden="true" tabindex="-1"></a>    <span class="at">activation =</span> <span class="st">'Tanh'</span>,</span>
<span id="cb10-223"><a href="#cb10-223" aria-hidden="true" tabindex="-1"></a>    <span class="at">sparse =</span> <span class="cn">TRUE</span>, </span>
<span id="cb10-224"><a href="#cb10-224" aria-hidden="true" tabindex="-1"></a>    <span class="at">seed =</span> <span class="dv">123</span>, </span>
<span id="cb10-225"><a href="#cb10-225" aria-hidden="true" tabindex="-1"></a>    <span class="at">reproducible =</span> <span class="cn">TRUE</span></span>
<span id="cb10-226"><a href="#cb10-226" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-227"><a href="#cb10-227" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-228"><a href="#cb10-228" aria-hidden="true" tabindex="-1"></a>ae1_codings <span class="ot">&lt;-</span> <span class="fu">h2o.deepfeatures</span>(ae1, features, <span class="at">layer =</span> <span class="dv">1</span>)</span>
<span id="cb10-229"><a href="#cb10-229" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-230"><a href="#cb10-230" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-233"><a href="#cb10-233" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-234"><a href="#cb10-234" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-first-autoencoder</span></span>
<span id="cb10-235"><a href="#cb10-235" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-236"><a href="#cb10-236" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> ae1_codings <span class="sc">%&gt;%</span></span>
<span id="cb10-237"><a href="#cb10-237" aria-hidden="true" tabindex="-1"></a>    <span class="fu">head</span>(<span class="at">n =</span> <span class="cn">Inf</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-238"><a href="#cb10-238" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span></span>
<span id="cb10-239"><a href="#cb10-239" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">cohort =</span> <span class="fu">toupper</span>(batch_training)) <span class="sc">%&gt;%</span></span>
<span id="cb10-240"><a href="#cb10-240" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> DF.L1.C1, <span class="at">y =</span> DF.L1.C2, <span class="at">color =</span> cohort)) <span class="sc">+</span></span>
<span id="cb10-241"><a href="#cb10-241" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb10-242"><a href="#cb10-242" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">color =</span> <span class="st">"Cohort"</span>) <span class="sc">+</span> </span>
<span id="cb10-243"><a href="#cb10-243" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">15</span>) <span class="sc">+</span> </span>
<span id="cb10-244"><a href="#cb10-244" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span> </span>
<span id="cb10-245"><a href="#cb10-245" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb10-246"><a href="#cb10-246" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-247"><a href="#cb10-247" aria-hidden="true" tabindex="-1"></a>p1</span>
<span id="cb10-248"><a href="#cb10-248" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-249"><a href="#cb10-249" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-250"><a href="#cb10-250" aria-hidden="true" tabindex="-1"></a>Just using the autoencoder in the data does not work. </span>
<span id="cb10-251"><a href="#cb10-251" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-252"><a href="#cb10-252" aria-hidden="true" tabindex="-1"></a>We now use on the normalized data instead on 4 units in the hidden</span>
<span id="cb10-253"><a href="#cb10-253" aria-hidden="true" tabindex="-1"></a>layer.</span>
<span id="cb10-254"><a href="#cb10-254" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-257"><a href="#cb10-257" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-258"><a href="#cb10-258" aria-hidden="true" tabindex="-1"></a><span class="fu">h2o.removeAll</span>()</span>
<span id="cb10-259"><a href="#cb10-259" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-260"><a href="#cb10-260" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-261"><a href="#cb10-261" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-264"><a href="#cb10-264" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-265"><a href="#cb10-265" aria-hidden="true" tabindex="-1"></a>all_datasets_normalized <span class="ot">&lt;-</span> <span class="fu">mapply</span>(</span>
<span id="cb10-266"><a href="#cb10-266" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(sum_exp, which_exp){</span>
<span id="cb10-267"><a href="#cb10-267" aria-hidden="true" tabindex="-1"></a>        SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(sum_exp, <span class="st">"avg_ranking"</span>)[</span>
<span id="cb10-268"><a href="#cb10-268" aria-hidden="true" tabindex="-1"></a>            <span class="fu">rownames</span>(pca_fit<span class="sc">$</span>loadings), </span>
<span id="cb10-269"><a href="#cb10-269" aria-hidden="true" tabindex="-1"></a>        ]</span>
<span id="cb10-270"><a href="#cb10-270" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb10-271"><a href="#cb10-271" aria-hidden="true" tabindex="-1"></a>    <span class="at">sum_exp =</span> datasets_normalized_og[<span class="fu">c</span>(<span class="st">"tcga"</span>, <span class="st">"metabric"</span>)],</span>
<span id="cb10-272"><a href="#cb10-272" aria-hidden="true" tabindex="-1"></a>    <span class="at">which_exp =</span> which_exp[<span class="fu">c</span>(<span class="st">"tcga"</span>, <span class="st">"metabric"</span>)]</span>
<span id="cb10-273"><a href="#cb10-273" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-274"><a href="#cb10-274" aria-hidden="true" tabindex="-1"></a>    <span class="fu">Reduce</span>(dplyr<span class="sc">::</span>bind_cols, <span class="at">x =</span> .)</span>
<span id="cb10-275"><a href="#cb10-275" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-276"><a href="#cb10-276" aria-hidden="true" tabindex="-1"></a>training_dataset_normalized <span class="ot">&lt;-</span> all_datasets_normalized[</span>
<span id="cb10-277"><a href="#cb10-277" aria-hidden="true" tabindex="-1"></a>    , </span>
<span id="cb10-278"><a href="#cb10-278" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames</span>(pca_fit<span class="sc">$</span>rotated)</span>
<span id="cb10-279"><a href="#cb10-279" aria-hidden="true" tabindex="-1"></a>]</span>
<span id="cb10-280"><a href="#cb10-280" aria-hidden="true" tabindex="-1"></a>batch_training_normalized <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(</span>
<span id="cb10-281"><a href="#cb10-281" aria-hidden="true" tabindex="-1"></a>    stringr<span class="sc">::</span><span class="fu">str_starts</span>(<span class="fu">colnames</span>(training_dataset_normalized), <span class="st">"TCGA"</span>),</span>
<span id="cb10-282"><a href="#cb10-282" aria-hidden="true" tabindex="-1"></a>    <span class="st">"tcga"</span>,</span>
<span id="cb10-283"><a href="#cb10-283" aria-hidden="true" tabindex="-1"></a>    <span class="st">"metabric"</span></span>
<span id="cb10-284"><a href="#cb10-284" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-285"><a href="#cb10-285" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-286"><a href="#cb10-286" aria-hidden="true" tabindex="-1"></a>features <span class="ot">&lt;-</span> <span class="fu">as.h2o</span>(training_dataset_normalized <span class="sc">%&gt;%</span> t)</span>
<span id="cb10-287"><a href="#cb10-287" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-288"><a href="#cb10-288" aria-hidden="true" tabindex="-1"></a>ae2 <span class="ot">&lt;-</span> <span class="fu">h2o.deeplearning</span>(</span>
<span id="cb10-289"><a href="#cb10-289" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="fu">seq_along</span>(features),</span>
<span id="cb10-290"><a href="#cb10-290" aria-hidden="true" tabindex="-1"></a>    <span class="at">training_frame =</span> features,</span>
<span id="cb10-291"><a href="#cb10-291" aria-hidden="true" tabindex="-1"></a>    <span class="at">autoencoder =</span> <span class="cn">TRUE</span>,</span>
<span id="cb10-292"><a href="#cb10-292" aria-hidden="true" tabindex="-1"></a>    <span class="at">hidden =</span> <span class="dv">4</span>,</span>
<span id="cb10-293"><a href="#cb10-293" aria-hidden="true" tabindex="-1"></a>    <span class="at">activation =</span> <span class="st">'Tanh'</span>,</span>
<span id="cb10-294"><a href="#cb10-294" aria-hidden="true" tabindex="-1"></a>    <span class="at">sparse =</span> <span class="cn">TRUE</span>, </span>
<span id="cb10-295"><a href="#cb10-295" aria-hidden="true" tabindex="-1"></a>    <span class="at">epochs =</span> <span class="dv">50</span>, </span>
<span id="cb10-296"><a href="#cb10-296" aria-hidden="true" tabindex="-1"></a>    <span class="at">seed =</span> <span class="dv">1000</span>,</span>
<span id="cb10-297"><a href="#cb10-297" aria-hidden="true" tabindex="-1"></a>    <span class="at">reproducible =</span> <span class="cn">TRUE</span></span>
<span id="cb10-298"><a href="#cb10-298" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-299"><a href="#cb10-299" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-300"><a href="#cb10-300" aria-hidden="true" tabindex="-1"></a>ae2_codings <span class="ot">&lt;-</span> <span class="fu">h2o.deepfeatures</span>(ae2, features, <span class="at">layer =</span> <span class="dv">1</span>)</span>
<span id="cb10-301"><a href="#cb10-301" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-302"><a href="#cb10-302" aria-hidden="true" tabindex="-1"></a>df_ae2_codings <span class="ot">&lt;-</span> ae2_codings <span class="sc">%&gt;%</span></span>
<span id="cb10-303"><a href="#cb10-303" aria-hidden="true" tabindex="-1"></a>    <span class="fu">head</span>(<span class="at">n =</span> <span class="cn">Inf</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-304"><a href="#cb10-304" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span></span>
<span id="cb10-305"><a href="#cb10-305" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb10-306"><a href="#cb10-306" aria-hidden="true" tabindex="-1"></a>        <span class="at">sample_name =</span> training_dataset_normalized <span class="sc">%&gt;%</span> colnames</span>
<span id="cb10-307"><a href="#cb10-307" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb10-308"><a href="#cb10-308" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">inner_join</span>(., df_pca, <span class="at">by =</span> <span class="st">"sample_name"</span>) </span>
<span id="cb10-309"><a href="#cb10-309" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-310"><a href="#cb10-310" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-313"><a href="#cb10-313" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-314"><a href="#cb10-314" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-second-ae-cohort</span></span>
<span id="cb10-315"><a href="#cb10-315" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-316"><a href="#cb10-316" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> df_ae2_codings <span class="sc">%&gt;%</span></span>
<span id="cb10-317"><a href="#cb10-317" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> DF.L1.C1, <span class="at">y =</span> DF.L1.C2, <span class="at">color =</span> <span class="fu">toupper</span>(cohort))) <span class="sc">+</span></span>
<span id="cb10-318"><a href="#cb10-318" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span> </span>
<span id="cb10-319"><a href="#cb10-319" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">color =</span> <span class="st">"Cohort"</span>)</span>
<span id="cb10-320"><a href="#cb10-320" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-321"><a href="#cb10-321" aria-hidden="true" tabindex="-1"></a>p2 <span class="sc">+</span> </span>
<span id="cb10-322"><a href="#cb10-322" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">15</span>) <span class="sc">+</span> </span>
<span id="cb10-323"><a href="#cb10-323" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span> </span>
<span id="cb10-324"><a href="#cb10-324" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb10-325"><a href="#cb10-325" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-326"><a href="#cb10-326" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-327"><a href="#cb10-327" aria-hidden="true" tabindex="-1"></a>And now coloring by er status. It is not really able to completely distinguish</span>
<span id="cb10-328"><a href="#cb10-328" aria-hidden="true" tabindex="-1"></a>the ER+ and ER- samples.</span>
<span id="cb10-329"><a href="#cb10-329" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-332"><a href="#cb10-332" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-333"><a href="#cb10-333" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-second-ae-erstatus</span></span>
<span id="cb10-334"><a href="#cb10-334" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-335"><a href="#cb10-335" aria-hidden="true" tabindex="-1"></a>p3 <span class="ot">&lt;-</span> df_ae2_codings <span class="sc">%&gt;%</span></span>
<span id="cb10-336"><a href="#cb10-336" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">er_status =</span> dplyr<span class="sc">::</span><span class="fu">case_when</span>(</span>
<span id="cb10-337"><a href="#cb10-337" aria-hidden="true" tabindex="-1"></a>        er_status <span class="sc">==</span> <span class="st">"neg"</span> <span class="sc">~</span> <span class="st">"Negative"</span>,</span>
<span id="cb10-338"><a href="#cb10-338" aria-hidden="true" tabindex="-1"></a>        er_status <span class="sc">==</span> <span class="st">"pos"</span> <span class="sc">~</span> <span class="st">"Positive"</span></span>
<span id="cb10-339"><a href="#cb10-339" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">%&gt;%</span></span>
<span id="cb10-340"><a href="#cb10-340" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> DF.L1.C1, <span class="at">y =</span> DF.L1.C2, <span class="at">color =</span> er_status)) <span class="sc">+</span></span>
<span id="cb10-341"><a href="#cb10-341" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span> </span>
<span id="cb10-342"><a href="#cb10-342" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">color =</span> <span class="st">"ER IHC"</span>)</span>
<span id="cb10-343"><a href="#cb10-343" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-344"><a href="#cb10-344" aria-hidden="true" tabindex="-1"></a>p3 <span class="sc">+</span> </span>
<span id="cb10-345"><a href="#cb10-345" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">15</span>) <span class="sc">+</span> </span>
<span id="cb10-346"><a href="#cb10-346" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span> </span>
<span id="cb10-347"><a href="#cb10-347" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb10-348"><a href="#cb10-348" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-349"><a href="#cb10-349" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-350"><a href="#cb10-350" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=12, fig.height=4}</span></span>
<span id="cb10-351"><a href="#cb10-351" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-ae-4-units-ae</span></span>
<span id="cb10-352"><a href="#cb10-352" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(p2, p3)</span>
<span id="cb10-353"><a href="#cb10-353" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-354"><a href="#cb10-354" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-355"><a href="#cb10-355" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-356"><a href="#cb10-356" aria-hidden="true" tabindex="-1"></a>We do a pairsplot coloring by cohort to see where the distinction between</span>
<span id="cb10-357"><a href="#cb10-357" aria-hidden="true" tabindex="-1"></a>cohorts is coming from. </span>
<span id="cb10-358"><a href="#cb10-358" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-361"><a href="#cb10-361" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-362"><a href="#cb10-362" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-ggpairs-4-units-ae-cohort</span></span>
<span id="cb10-363"><a href="#cb10-363" aria-hidden="true" tabindex="-1"></a>GGally<span class="sc">::</span><span class="fu">ggpairs</span>(</span>
<span id="cb10-364"><a href="#cb10-364" aria-hidden="true" tabindex="-1"></a>    df_ae2_codings, </span>
<span id="cb10-365"><a href="#cb10-365" aria-hidden="true" tabindex="-1"></a>    <span class="at">columns =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>,</span>
<span id="cb10-366"><a href="#cb10-366" aria-hidden="true" tabindex="-1"></a>    <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">color =</span> cohort, <span class="at">alpha =</span> <span class="fl">0.5</span>)</span>
<span id="cb10-367"><a href="#cb10-367" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-368"><a href="#cb10-368" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-369"><a href="#cb10-369" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-370"><a href="#cb10-370" aria-hidden="true" tabindex="-1"></a>And now coloring by ER status instead.</span>
<span id="cb10-371"><a href="#cb10-371" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-374"><a href="#cb10-374" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-375"><a href="#cb10-375" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-ggpairs-4-units-ae-erihc</span></span>
<span id="cb10-376"><a href="#cb10-376" aria-hidden="true" tabindex="-1"></a>GGally<span class="sc">::</span><span class="fu">ggpairs</span>(</span>
<span id="cb10-377"><a href="#cb10-377" aria-hidden="true" tabindex="-1"></a>    df_ae2_codings, </span>
<span id="cb10-378"><a href="#cb10-378" aria-hidden="true" tabindex="-1"></a>    <span class="at">columns =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>,</span>
<span id="cb10-379"><a href="#cb10-379" aria-hidden="true" tabindex="-1"></a>    <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">color =</span> er_status, <span class="at">alpha =</span> <span class="fl">0.5</span>)</span>
<span id="cb10-380"><a href="#cb10-380" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-381"><a href="#cb10-381" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-382"><a href="#cb10-382" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-383"><a href="#cb10-383" aria-hidden="true" tabindex="-1"></a>Perhaps now if we include more layers instead and use 4 components.</span>
<span id="cb10-384"><a href="#cb10-384" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-387"><a href="#cb10-387" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-388"><a href="#cb10-388" aria-hidden="true" tabindex="-1"></a>hyper_grid <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">hidden =</span> <span class="fu">list</span>(</span>
<span id="cb10-389"><a href="#cb10-389" aria-hidden="true" tabindex="-1"></a>    <span class="fu">c</span>(<span class="dv">2</span>),</span>
<span id="cb10-390"><a href="#cb10-390" aria-hidden="true" tabindex="-1"></a>    <span class="fu">c</span>(<span class="dv">4</span>),</span>
<span id="cb10-391"><a href="#cb10-391" aria-hidden="true" tabindex="-1"></a>    <span class="fu">c</span>(<span class="dv">50</span>, <span class="dv">2</span>, <span class="dv">50</span>),</span>
<span id="cb10-392"><a href="#cb10-392" aria-hidden="true" tabindex="-1"></a>    <span class="fu">c</span>(<span class="dv">100</span>, <span class="dv">2</span>, <span class="dv">100</span>), </span>
<span id="cb10-393"><a href="#cb10-393" aria-hidden="true" tabindex="-1"></a>    <span class="fu">c</span>(<span class="dv">300</span>, <span class="dv">2</span>, <span class="dv">300</span>),</span>
<span id="cb10-394"><a href="#cb10-394" aria-hidden="true" tabindex="-1"></a>    <span class="fu">c</span>(<span class="dv">250</span>, <span class="dv">100</span>, <span class="dv">2</span>, <span class="dv">100</span>, <span class="dv">250</span>),</span>
<span id="cb10-395"><a href="#cb10-395" aria-hidden="true" tabindex="-1"></a>    <span class="fu">c</span>(<span class="dv">50</span>, <span class="dv">4</span>, <span class="dv">50</span>),</span>
<span id="cb10-396"><a href="#cb10-396" aria-hidden="true" tabindex="-1"></a>    <span class="fu">c</span>(<span class="dv">100</span>, <span class="dv">4</span>, <span class="dv">100</span>), </span>
<span id="cb10-397"><a href="#cb10-397" aria-hidden="true" tabindex="-1"></a>    <span class="fu">c</span>(<span class="dv">300</span>, <span class="dv">4</span>, <span class="dv">300</span>),</span>
<span id="cb10-398"><a href="#cb10-398" aria-hidden="true" tabindex="-1"></a>    <span class="fu">c</span>(<span class="dv">250</span>, <span class="dv">100</span>, <span class="dv">4</span>, <span class="dv">100</span>, <span class="dv">250</span>)</span>
<span id="cb10-399"><a href="#cb10-399" aria-hidden="true" tabindex="-1"></a>))</span>
<span id="cb10-400"><a href="#cb10-400" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-401"><a href="#cb10-401" aria-hidden="true" tabindex="-1"></a><span class="co"># Execute grid search</span></span>
<span id="cb10-402"><a href="#cb10-402" aria-hidden="true" tabindex="-1"></a><span class="co"># Execute grid search</span></span>
<span id="cb10-403"><a href="#cb10-403" aria-hidden="true" tabindex="-1"></a>ae_grid <span class="ot">&lt;-</span> <span class="fu">h2o.grid</span>(</span>
<span id="cb10-404"><a href="#cb10-404" aria-hidden="true" tabindex="-1"></a>    <span class="at">algorithm =</span> <span class="st">'deeplearning'</span>,</span>
<span id="cb10-405"><a href="#cb10-405" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="fu">seq_along</span>(features),</span>
<span id="cb10-406"><a href="#cb10-406" aria-hidden="true" tabindex="-1"></a>    <span class="at">training_frame =</span> features,</span>
<span id="cb10-407"><a href="#cb10-407" aria-hidden="true" tabindex="-1"></a>    <span class="at">grid_id =</span> <span class="st">'autoencoder_grid'</span>,</span>
<span id="cb10-408"><a href="#cb10-408" aria-hidden="true" tabindex="-1"></a>    <span class="at">autoencoder =</span> <span class="cn">TRUE</span>,</span>
<span id="cb10-409"><a href="#cb10-409" aria-hidden="true" tabindex="-1"></a>    <span class="at">activation =</span> <span class="st">'Tanh'</span>,</span>
<span id="cb10-410"><a href="#cb10-410" aria-hidden="true" tabindex="-1"></a>    <span class="at">hyper_params =</span> hyper_grid,</span>
<span id="cb10-411"><a href="#cb10-411" aria-hidden="true" tabindex="-1"></a>    <span class="at">sparse =</span> <span class="cn">TRUE</span>,</span>
<span id="cb10-412"><a href="#cb10-412" aria-hidden="true" tabindex="-1"></a>    <span class="at">ignore_const_cols =</span> <span class="cn">FALSE</span>,</span>
<span id="cb10-413"><a href="#cb10-413" aria-hidden="true" tabindex="-1"></a>    <span class="at">seed =</span> <span class="dv">123</span>,</span>
<span id="cb10-414"><a href="#cb10-414" aria-hidden="true" tabindex="-1"></a>    <span class="at">epochs =</span> <span class="dv">10</span></span>
<span id="cb10-415"><a href="#cb10-415" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-416"><a href="#cb10-416" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-417"><a href="#cb10-417" aria-hidden="true" tabindex="-1"></a><span class="co"># Print grid details</span></span>
<span id="cb10-418"><a href="#cb10-418" aria-hidden="true" tabindex="-1"></a>grid_results <span class="ot">&lt;-</span> <span class="fu">h2o.getGrid</span>(</span>
<span id="cb10-419"><a href="#cb10-419" aria-hidden="true" tabindex="-1"></a>    <span class="st">'autoencoder_grid'</span>, </span>
<span id="cb10-420"><a href="#cb10-420" aria-hidden="true" tabindex="-1"></a>    <span class="at">sort_by =</span> <span class="st">'mse'</span>, </span>
<span id="cb10-421"><a href="#cb10-421" aria-hidden="true" tabindex="-1"></a>    <span class="at">decreasing =</span> <span class="cn">FALSE</span></span>
<span id="cb10-422"><a href="#cb10-422" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-423"><a href="#cb10-423" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-424"><a href="#cb10-424" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-425"><a href="#cb10-425" aria-hidden="true" tabindex="-1"></a>In terms of mean squared error all the models are very similar to each other.</span>
<span id="cb10-426"><a href="#cb10-426" aria-hidden="true" tabindex="-1"></a>And based on the previous results it seems that by using </span>
<span id="cb10-427"><a href="#cb10-427" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-430"><a href="#cb10-430" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-431"><a href="#cb10-431" aria-hidden="true" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">write_excel_csv</span>(</span>
<span id="cb10-432"><a href="#cb10-432" aria-hidden="true" tabindex="-1"></a>    grid_results<span class="sc">@</span>summary_table,</span>
<span id="cb10-433"><a href="#cb10-433" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../../results/tables/revision/grid_results_summary_table.csv"</span></span>
<span id="cb10-434"><a href="#cb10-434" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-435"><a href="#cb10-435" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-436"><a href="#cb10-436" aria-hidden="true" tabindex="-1"></a><span class="fu">h2o.shutdown</span>(<span class="at">prompt =</span> <span class="cn">FALSE</span>)</span>
<span id="cb10-437"><a href="#cb10-437" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-438"><a href="#cb10-438" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-439"><a href="#cb10-439" aria-hidden="true" tabindex="-1"></a><span class="fu">## Mixing analysis </span></span>
<span id="cb10-440"><a href="#cb10-440" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-441"><a href="#cb10-441" aria-hidden="true" tabindex="-1"></a>In this section we try to understand what is the minimum number of genes</span>
<span id="cb10-442"><a href="#cb10-442" aria-hidden="true" tabindex="-1"></a>necessary to be able to distinguish ER+ and ER- BC samples. We select</span>
<span id="cb10-443"><a href="#cb10-443" aria-hidden="true" tabindex="-1"></a>the top 10, 25, 50, 100, 200, 500, 750, 1000, 2500, 5000 and all the genes and</span>
<span id="cb10-444"><a href="#cb10-444" aria-hidden="true" tabindex="-1"></a>try to classify the samples bewteen ER+ and ER- based on their position of the</span>
<span id="cb10-445"><a href="#cb10-445" aria-hidden="true" tabindex="-1"></a>third component. This should give an idea of how good the embedding is based</span>
<span id="cb10-446"><a href="#cb10-446" aria-hidden="true" tabindex="-1"></a>on the number of genes.</span>
<span id="cb10-447"><a href="#cb10-447" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-450"><a href="#cb10-450" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-451"><a href="#cb10-451" aria-hidden="true" tabindex="-1"></a>nb_genes <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb10-452"><a href="#cb10-452" aria-hidden="true" tabindex="-1"></a>    <span class="dv">10</span>, <span class="dv">25</span>, <span class="dv">50</span>, <span class="dv">100</span>, <span class="dv">200</span>, <span class="dv">500</span>, </span>
<span id="cb10-453"><a href="#cb10-453" aria-hidden="true" tabindex="-1"></a>    <span class="dv">600</span>, <span class="dv">650</span>, <span class="dv">700</span>, <span class="dv">750</span>, <span class="dv">1000</span>, </span>
<span id="cb10-454"><a href="#cb10-454" aria-hidden="true" tabindex="-1"></a>    <span class="dv">2500</span>, <span class="dv">5000</span>, <span class="dv">8248</span></span>
<span id="cb10-455"><a href="#cb10-455" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-456"><a href="#cb10-456" aria-hidden="true" tabindex="-1"></a>which_cohorts_training <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"tcga"</span>, <span class="st">"metabric"</span>)</span>
<span id="cb10-457"><a href="#cb10-457" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-458"><a href="#cb10-458" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-459"><a href="#cb10-459" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb10-460"><a href="#cb10-460" aria-hidden="true" tabindex="-1"></a>pca_fits_nb_genes <span class="ot">&lt;-</span> <span class="fu">sapply</span>(</span>
<span id="cb10-461"><a href="#cb10-461" aria-hidden="true" tabindex="-1"></a>    nb_genes, </span>
<span id="cb10-462"><a href="#cb10-462" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(nb_gene){</span>
<span id="cb10-463"><a href="#cb10-463" aria-hidden="true" tabindex="-1"></a>        <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Nb of genes: "</span>, nb_gene))</span>
<span id="cb10-464"><a href="#cb10-464" aria-hidden="true" tabindex="-1"></a>        genes_for_pca <span class="ot">&lt;-</span> cv_genes <span class="sc">%&gt;%</span></span>
<span id="cb10-465"><a href="#cb10-465" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">slice_max</span>(<span class="at">n =</span> nb_gene, <span class="at">order_by =</span> average) <span class="sc">%&gt;%</span></span>
<span id="cb10-466"><a href="#cb10-466" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">pull</span>(symbol)</span>
<span id="cb10-467"><a href="#cb10-467" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb10-468"><a href="#cb10-468" aria-hidden="true" tabindex="-1"></a>        datasets_normalized <span class="ot">&lt;-</span> <span class="fu">mapply</span>(</span>
<span id="cb10-469"><a href="#cb10-469" aria-hidden="true" tabindex="-1"></a>            get_final_ranking_values,</span>
<span id="cb10-470"><a href="#cb10-470" aria-hidden="true" tabindex="-1"></a>            <span class="at">sum_exp =</span> datasets[which_cohorts_training],</span>
<span id="cb10-471"><a href="#cb10-471" aria-hidden="true" tabindex="-1"></a>            <span class="at">assay_to_use =</span> which_exp[which_cohorts_training],</span>
<span id="cb10-472"><a href="#cb10-472" aria-hidden="true" tabindex="-1"></a>            <span class="at">MoreArgs =</span> <span class="fu">list</span>(</span>
<span id="cb10-473"><a href="#cb10-473" aria-hidden="true" tabindex="-1"></a>                <span class="at">stable_genes =</span> stable_genes,</span>
<span id="cb10-474"><a href="#cb10-474" aria-hidden="true" tabindex="-1"></a>                <span class="at">most_variable_genes =</span> <span class="fu">setdiff</span>(genes_for_pca, stable_genes)</span>
<span id="cb10-475"><a href="#cb10-475" aria-hidden="true" tabindex="-1"></a>            ),</span>
<span id="cb10-476"><a href="#cb10-476" aria-hidden="true" tabindex="-1"></a>            <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>,</span>
<span id="cb10-477"><a href="#cb10-477" aria-hidden="true" tabindex="-1"></a>            <span class="at">SIMPLIFY =</span> <span class="cn">FALSE</span></span>
<span id="cb10-478"><a href="#cb10-478" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb10-479"><a href="#cb10-479" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb10-480"><a href="#cb10-480" aria-hidden="true" tabindex="-1"></a>        samples_for_training <span class="ot">&lt;-</span> pca_fit<span class="sc">$</span>rotated <span class="sc">%&gt;%</span> rownames</span>
<span id="cb10-481"><a href="#cb10-481" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb10-482"><a href="#cb10-482" aria-hidden="true" tabindex="-1"></a>        training_set <span class="ot">&lt;-</span> <span class="fu">lapply</span>(</span>
<span id="cb10-483"><a href="#cb10-483" aria-hidden="true" tabindex="-1"></a>            datasets_normalized[which_cohorts_training], </span>
<span id="cb10-484"><a href="#cb10-484" aria-hidden="true" tabindex="-1"></a>            <span class="cf">function</span>(sum_exp, i, genes_for_pca) </span>
<span id="cb10-485"><a href="#cb10-485" aria-hidden="true" tabindex="-1"></a>                <span class="fu">assay</span>(sum_exp[genes_for_pca, ], <span class="at">i =</span> i) <span class="sc">%&gt;%</span> </span>
<span id="cb10-486"><a href="#cb10-486" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">data.frame</span>(<span class="at">check.names =</span> <span class="cn">FALSE</span>), </span>
<span id="cb10-487"><a href="#cb10-487" aria-hidden="true" tabindex="-1"></a>                <span class="at">i =</span> <span class="st">"avg_ranking"</span>,</span>
<span id="cb10-488"><a href="#cb10-488" aria-hidden="true" tabindex="-1"></a>                <span class="at">genes_for_pca =</span> genes_for_pca</span>
<span id="cb10-489"><a href="#cb10-489" aria-hidden="true" tabindex="-1"></a>        ) <span class="sc">%&gt;%</span> </span>
<span id="cb10-490"><a href="#cb10-490" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">bind_cols</span>() <span class="sc">%&gt;%</span></span>
<span id="cb10-491"><a href="#cb10-491" aria-hidden="true" tabindex="-1"></a>            .[, samples_for_training]</span>
<span id="cb10-492"><a href="#cb10-492" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-493"><a href="#cb10-493" aria-hidden="true" tabindex="-1"></a>        pca_fit_sensitivity <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">pca</span>(</span>
<span id="cb10-494"><a href="#cb10-494" aria-hidden="true" tabindex="-1"></a>            training_set,</span>
<span id="cb10-495"><a href="#cb10-495" aria-hidden="true" tabindex="-1"></a>            <span class="at">metadata =</span> dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(</span>
<span id="cb10-496"><a href="#cb10-496" aria-hidden="true" tabindex="-1"></a>                <span class="fu">lapply</span>(</span>
<span id="cb10-497"><a href="#cb10-497" aria-hidden="true" tabindex="-1"></a>                    datasets_normalized[which_cohorts_training],</span>
<span id="cb10-498"><a href="#cb10-498" aria-hidden="true" tabindex="-1"></a>                    <span class="cf">function</span>(df){</span>
<span id="cb10-499"><a href="#cb10-499" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">colData</span>(df) <span class="sc">%&gt;%</span> </span>
<span id="cb10-500"><a href="#cb10-500" aria-hidden="true" tabindex="-1"></a>                            data.frame <span class="sc">%&gt;%</span></span>
<span id="cb10-501"><a href="#cb10-501" aria-hidden="true" tabindex="-1"></a>                            dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb10-502"><a href="#cb10-502" aria-hidden="true" tabindex="-1"></a>                                sample_name <span class="sc">%in%</span> <span class="fu">colnames</span>(training_set)</span>
<span id="cb10-503"><a href="#cb10-503" aria-hidden="true" tabindex="-1"></a>                            )</span>
<span id="cb10-504"><a href="#cb10-504" aria-hidden="true" tabindex="-1"></a>                    }</span>
<span id="cb10-505"><a href="#cb10-505" aria-hidden="true" tabindex="-1"></a>                ),</span>
<span id="cb10-506"><a href="#cb10-506" aria-hidden="true" tabindex="-1"></a>                <span class="at">.id =</span> <span class="st">"cohort"</span></span>
<span id="cb10-507"><a href="#cb10-507" aria-hidden="true" tabindex="-1"></a>            ) <span class="sc">%&gt;%</span> </span>
<span id="cb10-508"><a href="#cb10-508" aria-hidden="true" tabindex="-1"></a>                .[<span class="fu">colnames</span>(training_set), ],</span>
<span id="cb10-509"><a href="#cb10-509" aria-hidden="true" tabindex="-1"></a>            <span class="at">center =</span> <span class="cn">FALSE</span>,</span>
<span id="cb10-510"><a href="#cb10-510" aria-hidden="true" tabindex="-1"></a>            <span class="at">scale =</span> <span class="cn">FALSE</span></span>
<span id="cb10-511"><a href="#cb10-511" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb10-512"><a href="#cb10-512" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb10-513"><a href="#cb10-513" aria-hidden="true" tabindex="-1"></a>        pca_fit_sensitivity</span>
<span id="cb10-514"><a href="#cb10-514" aria-hidden="true" tabindex="-1"></a>    }, </span>
<span id="cb10-515"><a href="#cb10-515" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>, </span>
<span id="cb10-516"><a href="#cb10-516" aria-hidden="true" tabindex="-1"></a>    <span class="at">simplify =</span> <span class="cn">FALSE</span></span>
<span id="cb10-517"><a href="#cb10-517" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-518"><a href="#cb10-518" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-519"><a href="#cb10-519" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(pca_fits_nb_genes) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"nb_"</span>, nb_genes)</span>
<span id="cb10-520"><a href="#cb10-520" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-521"><a href="#cb10-521" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb10-522"><a href="#cb10-522" aria-hidden="true" tabindex="-1"></a>    pca_fits_nb_genes,</span>
<span id="cb10-523"><a href="#cb10-523" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../../results/rds_files/revision/pca_fits_nb_genes.rds"</span></span>
<span id="cb10-524"><a href="#cb10-524" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-525"><a href="#cb10-525" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-526"><a href="#cb10-526" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-529"><a href="#cb10-529" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-530"><a href="#cb10-530" aria-hidden="true" tabindex="-1"></a>plots_pca <span class="ot">&lt;-</span> <span class="fu">mapply</span>(</span>
<span id="cb10-531"><a href="#cb10-531" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(pca_fit_genes, nb_gene){</span>
<span id="cb10-532"><a href="#cb10-532" aria-hidden="true" tabindex="-1"></a>        pc1_pc2 <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb10-533"><a href="#cb10-533" aria-hidden="true" tabindex="-1"></a>            pca_fit_genes,</span>
<span id="cb10-534"><a href="#cb10-534" aria-hidden="true" tabindex="-1"></a>            <span class="at">x =</span> <span class="st">"PC1"</span>,</span>
<span id="cb10-535"><a href="#cb10-535" aria-hidden="true" tabindex="-1"></a>            <span class="at">y =</span> <span class="st">"PC2"</span>,</span>
<span id="cb10-536"><a href="#cb10-536" aria-hidden="true" tabindex="-1"></a>            <span class="at">colby =</span> <span class="st">"cohort"</span>,</span>
<span id="cb10-537"><a href="#cb10-537" aria-hidden="true" tabindex="-1"></a>            <span class="at">lab =</span> <span class="cn">NULL</span>,</span>
<span id="cb10-538"><a href="#cb10-538" aria-hidden="true" tabindex="-1"></a>            <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb10-539"><a href="#cb10-539" aria-hidden="true" tabindex="-1"></a>            <span class="at">title =</span> <span class="fu">paste0</span>(<span class="st">"Total number of genes: "</span>, nb_gene),</span>
<span id="cb10-540"><a href="#cb10-540" aria-hidden="true" tabindex="-1"></a>            <span class="at">pointSize =</span> <span class="dv">1</span></span>
<span id="cb10-541"><a href="#cb10-541" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb10-542"><a href="#cb10-542" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb10-543"><a href="#cb10-543" aria-hidden="true" tabindex="-1"></a>        pc3_pc4 <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb10-544"><a href="#cb10-544" aria-hidden="true" tabindex="-1"></a>            pca_fit_genes,</span>
<span id="cb10-545"><a href="#cb10-545" aria-hidden="true" tabindex="-1"></a>            <span class="at">x =</span> <span class="st">"PC3"</span>,</span>
<span id="cb10-546"><a href="#cb10-546" aria-hidden="true" tabindex="-1"></a>            <span class="at">y =</span> <span class="st">"PC4"</span>,</span>
<span id="cb10-547"><a href="#cb10-547" aria-hidden="true" tabindex="-1"></a>            <span class="at">colby =</span> <span class="st">"er_status"</span>,</span>
<span id="cb10-548"><a href="#cb10-548" aria-hidden="true" tabindex="-1"></a>            <span class="at">lab =</span> <span class="cn">NULL</span>,</span>
<span id="cb10-549"><a href="#cb10-549" aria-hidden="true" tabindex="-1"></a>            <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb10-550"><a href="#cb10-550" aria-hidden="true" tabindex="-1"></a>            <span class="at">pointSize =</span> <span class="dv">1</span></span>
<span id="cb10-551"><a href="#cb10-551" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb10-552"><a href="#cb10-552" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb10-553"><a href="#cb10-553" aria-hidden="true" tabindex="-1"></a>        cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(pc1_pc2, pc3_pc4)</span>
<span id="cb10-554"><a href="#cb10-554" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb10-555"><a href="#cb10-555" aria-hidden="true" tabindex="-1"></a>    pca_fits_nb_genes,</span>
<span id="cb10-556"><a href="#cb10-556" aria-hidden="true" tabindex="-1"></a>    nb_genes,</span>
<span id="cb10-557"><a href="#cb10-557" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>,</span>
<span id="cb10-558"><a href="#cb10-558" aria-hidden="true" tabindex="-1"></a>    <span class="at">SIMPLIFY =</span> <span class="cn">FALSE</span></span>
<span id="cb10-559"><a href="#cb10-559" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-560"><a href="#cb10-560" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-561"><a href="#cb10-561" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-564"><a href="#cb10-564" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-565"><a href="#cb10-565" aria-hidden="true" tabindex="-1"></a>plots_pca_swapped <span class="ot">&lt;-</span> <span class="fu">mapply</span>(</span>
<span id="cb10-566"><a href="#cb10-566" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(pca_fit_genes, nb_gene){</span>
<span id="cb10-567"><a href="#cb10-567" aria-hidden="true" tabindex="-1"></a>        pc1_pc2 <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb10-568"><a href="#cb10-568" aria-hidden="true" tabindex="-1"></a>            pca_fit_genes,</span>
<span id="cb10-569"><a href="#cb10-569" aria-hidden="true" tabindex="-1"></a>            <span class="at">x =</span> <span class="st">"PC1"</span>,</span>
<span id="cb10-570"><a href="#cb10-570" aria-hidden="true" tabindex="-1"></a>            <span class="at">y =</span> <span class="st">"PC2"</span>,</span>
<span id="cb10-571"><a href="#cb10-571" aria-hidden="true" tabindex="-1"></a>            <span class="at">colby =</span> <span class="st">"er_status"</span>,</span>
<span id="cb10-572"><a href="#cb10-572" aria-hidden="true" tabindex="-1"></a>            <span class="at">lab =</span> <span class="cn">NULL</span>,</span>
<span id="cb10-573"><a href="#cb10-573" aria-hidden="true" tabindex="-1"></a>            <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb10-574"><a href="#cb10-574" aria-hidden="true" tabindex="-1"></a>            <span class="at">title =</span> <span class="fu">paste0</span>(<span class="st">"Total number of genes: "</span>, nb_gene),</span>
<span id="cb10-575"><a href="#cb10-575" aria-hidden="true" tabindex="-1"></a>            <span class="at">pointSize =</span> <span class="dv">1</span></span>
<span id="cb10-576"><a href="#cb10-576" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb10-577"><a href="#cb10-577" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb10-578"><a href="#cb10-578" aria-hidden="true" tabindex="-1"></a>        pc1_pc2_pam50 <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb10-579"><a href="#cb10-579" aria-hidden="true" tabindex="-1"></a>            pca_fit_genes,</span>
<span id="cb10-580"><a href="#cb10-580" aria-hidden="true" tabindex="-1"></a>            <span class="at">x =</span> <span class="st">"PC1"</span>,</span>
<span id="cb10-581"><a href="#cb10-581" aria-hidden="true" tabindex="-1"></a>            <span class="at">y =</span> <span class="st">"PC2"</span>,</span>
<span id="cb10-582"><a href="#cb10-582" aria-hidden="true" tabindex="-1"></a>            <span class="at">colby =</span> <span class="st">"pam50"</span>,</span>
<span id="cb10-583"><a href="#cb10-583" aria-hidden="true" tabindex="-1"></a>            <span class="at">lab =</span> <span class="cn">NULL</span>,</span>
<span id="cb10-584"><a href="#cb10-584" aria-hidden="true" tabindex="-1"></a>            <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb10-585"><a href="#cb10-585" aria-hidden="true" tabindex="-1"></a>            <span class="at">pointSize =</span> <span class="dv">1</span></span>
<span id="cb10-586"><a href="#cb10-586" aria-hidden="true" tabindex="-1"></a>        ) <span class="sc">+</span> </span>
<span id="cb10-587"><a href="#cb10-587" aria-hidden="true" tabindex="-1"></a>            ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">color =</span> <span class="st">"Molecular</span><span class="sc">\n</span><span class="st">subtype"</span>) <span class="sc">+</span> </span>
<span id="cb10-588"><a href="#cb10-588" aria-hidden="true" tabindex="-1"></a>            ggplot2<span class="sc">::</span><span class="fu">scale_color_manual</span>(</span>
<span id="cb10-589"><a href="#cb10-589" aria-hidden="true" tabindex="-1"></a>                <span class="at">values =</span> <span class="fu">get_colors_pam50</span>(pca_fit_genes<span class="sc">$</span>metadata)</span>
<span id="cb10-590"><a href="#cb10-590" aria-hidden="true" tabindex="-1"></a>            )</span>
<span id="cb10-591"><a href="#cb10-591" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb10-592"><a href="#cb10-592" aria-hidden="true" tabindex="-1"></a>        pc1_pc2_cohort <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb10-593"><a href="#cb10-593" aria-hidden="true" tabindex="-1"></a>            pca_fit_genes,</span>
<span id="cb10-594"><a href="#cb10-594" aria-hidden="true" tabindex="-1"></a>            <span class="at">x =</span> <span class="st">"PC1"</span>,</span>
<span id="cb10-595"><a href="#cb10-595" aria-hidden="true" tabindex="-1"></a>            <span class="at">y =</span> <span class="st">"PC2"</span>,</span>
<span id="cb10-596"><a href="#cb10-596" aria-hidden="true" tabindex="-1"></a>            <span class="at">colby =</span> <span class="st">"cohort"</span>,</span>
<span id="cb10-597"><a href="#cb10-597" aria-hidden="true" tabindex="-1"></a>            <span class="at">lab =</span> <span class="cn">NULL</span>,</span>
<span id="cb10-598"><a href="#cb10-598" aria-hidden="true" tabindex="-1"></a>            <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb10-599"><a href="#cb10-599" aria-hidden="true" tabindex="-1"></a>            <span class="at">pointSize =</span> <span class="dv">1</span></span>
<span id="cb10-600"><a href="#cb10-600" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb10-601"><a href="#cb10-601" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb10-602"><a href="#cb10-602" aria-hidden="true" tabindex="-1"></a>        pc3_pc4 <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb10-603"><a href="#cb10-603" aria-hidden="true" tabindex="-1"></a>            pca_fit_genes,</span>
<span id="cb10-604"><a href="#cb10-604" aria-hidden="true" tabindex="-1"></a>            <span class="at">x =</span> <span class="st">"PC3"</span>,</span>
<span id="cb10-605"><a href="#cb10-605" aria-hidden="true" tabindex="-1"></a>            <span class="at">y =</span> <span class="st">"PC4"</span>,</span>
<span id="cb10-606"><a href="#cb10-606" aria-hidden="true" tabindex="-1"></a>            <span class="at">colby =</span> <span class="st">"cohort"</span>,</span>
<span id="cb10-607"><a href="#cb10-607" aria-hidden="true" tabindex="-1"></a>            <span class="at">lab =</span> <span class="cn">NULL</span>,</span>
<span id="cb10-608"><a href="#cb10-608" aria-hidden="true" tabindex="-1"></a>            <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb10-609"><a href="#cb10-609" aria-hidden="true" tabindex="-1"></a>            <span class="at">pointSize =</span> <span class="dv">1</span></span>
<span id="cb10-610"><a href="#cb10-610" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb10-611"><a href="#cb10-611" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb10-612"><a href="#cb10-612" aria-hidden="true" tabindex="-1"></a>        pc3_pc4_er_status <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb10-613"><a href="#cb10-613" aria-hidden="true" tabindex="-1"></a>            pca_fit_genes,</span>
<span id="cb10-614"><a href="#cb10-614" aria-hidden="true" tabindex="-1"></a>            <span class="at">x =</span> <span class="st">"PC3"</span>,</span>
<span id="cb10-615"><a href="#cb10-615" aria-hidden="true" tabindex="-1"></a>            <span class="at">y =</span> <span class="st">"PC4"</span>,</span>
<span id="cb10-616"><a href="#cb10-616" aria-hidden="true" tabindex="-1"></a>            <span class="at">colby =</span> <span class="st">"er_status"</span>,</span>
<span id="cb10-617"><a href="#cb10-617" aria-hidden="true" tabindex="-1"></a>            <span class="at">lab =</span> <span class="cn">NULL</span>,</span>
<span id="cb10-618"><a href="#cb10-618" aria-hidden="true" tabindex="-1"></a>            <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb10-619"><a href="#cb10-619" aria-hidden="true" tabindex="-1"></a>            <span class="at">pointSize =</span> <span class="dv">1</span></span>
<span id="cb10-620"><a href="#cb10-620" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb10-621"><a href="#cb10-621" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb10-622"><a href="#cb10-622" aria-hidden="true" tabindex="-1"></a>        pc3_pc4_pam50 <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb10-623"><a href="#cb10-623" aria-hidden="true" tabindex="-1"></a>            pca_fit_genes,</span>
<span id="cb10-624"><a href="#cb10-624" aria-hidden="true" tabindex="-1"></a>            <span class="at">x =</span> <span class="st">"PC3"</span>,</span>
<span id="cb10-625"><a href="#cb10-625" aria-hidden="true" tabindex="-1"></a>            <span class="at">y =</span> <span class="st">"PC4"</span>,</span>
<span id="cb10-626"><a href="#cb10-626" aria-hidden="true" tabindex="-1"></a>            <span class="at">colby =</span> <span class="st">"pam50"</span>,</span>
<span id="cb10-627"><a href="#cb10-627" aria-hidden="true" tabindex="-1"></a>            <span class="at">lab =</span> <span class="cn">NULL</span>,</span>
<span id="cb10-628"><a href="#cb10-628" aria-hidden="true" tabindex="-1"></a>            <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb10-629"><a href="#cb10-629" aria-hidden="true" tabindex="-1"></a>            <span class="at">pointSize =</span> <span class="dv">1</span></span>
<span id="cb10-630"><a href="#cb10-630" aria-hidden="true" tabindex="-1"></a>        ) <span class="sc">+</span> </span>
<span id="cb10-631"><a href="#cb10-631" aria-hidden="true" tabindex="-1"></a>            ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">color =</span> <span class="st">"Molecular</span><span class="sc">\n</span><span class="st">subtype"</span>) <span class="sc">+</span> </span>
<span id="cb10-632"><a href="#cb10-632" aria-hidden="true" tabindex="-1"></a>            ggplot2<span class="sc">::</span><span class="fu">scale_color_manual</span>(</span>
<span id="cb10-633"><a href="#cb10-633" aria-hidden="true" tabindex="-1"></a>                <span class="at">values =</span> <span class="fu">get_colors_pam50</span>(pca_fit_genes<span class="sc">$</span>metadata)</span>
<span id="cb10-634"><a href="#cb10-634" aria-hidden="true" tabindex="-1"></a>            )</span>
<span id="cb10-635"><a href="#cb10-635" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb10-636"><a href="#cb10-636" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb10-637"><a href="#cb10-637" aria-hidden="true" tabindex="-1"></a>        cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(</span>
<span id="cb10-638"><a href="#cb10-638" aria-hidden="true" tabindex="-1"></a>            pc1_pc2, pc1_pc2_cohort, pc1_pc2_pam50, </span>
<span id="cb10-639"><a href="#cb10-639" aria-hidden="true" tabindex="-1"></a>            pc3_pc4_er_status, pc3_pc4, pc3_pc4_pam50,</span>
<span id="cb10-640"><a href="#cb10-640" aria-hidden="true" tabindex="-1"></a>            <span class="at">nrow =</span> <span class="dv">1</span></span>
<span id="cb10-641"><a href="#cb10-641" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb10-642"><a href="#cb10-642" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb10-643"><a href="#cb10-643" aria-hidden="true" tabindex="-1"></a>    pca_fits_nb_genes,</span>
<span id="cb10-644"><a href="#cb10-644" aria-hidden="true" tabindex="-1"></a>    nb_genes,</span>
<span id="cb10-645"><a href="#cb10-645" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>,</span>
<span id="cb10-646"><a href="#cb10-646" aria-hidden="true" tabindex="-1"></a>    <span class="at">SIMPLIFY =</span> <span class="cn">FALSE</span></span>
<span id="cb10-647"><a href="#cb10-647" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-648"><a href="#cb10-648" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-649"><a href="#cb10-649" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-650"><a href="#cb10-650" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-651"><a href="#cb10-651" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=28, fig.height=45}</span></span>
<span id="cb10-652"><a href="#cb10-652" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-pca-nb-genes</span></span>
<span id="cb10-653"><a href="#cb10-653" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(</span>
<span id="cb10-654"><a href="#cb10-654" aria-hidden="true" tabindex="-1"></a>    <span class="at">plotlist =</span> plots_pca_swapped,</span>
<span id="cb10-655"><a href="#cb10-655" aria-hidden="true" tabindex="-1"></a>    <span class="at">ncol =</span> <span class="dv">1</span></span>
<span id="cb10-656"><a href="#cb10-656" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-657"><a href="#cb10-657" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-658"><a href="#cb10-658" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-659"><a href="#cb10-659" aria-hidden="true" tabindex="-1"></a>Using less than 25 genes mixes well the cohorts in the first two </span>
<span id="cb10-660"><a href="#cb10-660" aria-hidden="true" tabindex="-1"></a>components. When using more than 50 genes the mixing is already </span>
<span id="cb10-661"><a href="#cb10-661" aria-hidden="true" tabindex="-1"></a>lost in the first two components. So what we actually want is a mixing</span>
<span id="cb10-662"><a href="#cb10-662" aria-hidden="true" tabindex="-1"></a>in just one of the pairs of components, either PC1+PC2 or PC3+PC4 while there</span>
<span id="cb10-663"><a href="#cb10-663" aria-hidden="true" tabindex="-1"></a>is no mixing in the other pair of component. By analysing visually this happens</span>
<span id="cb10-664"><a href="#cb10-664" aria-hidden="true" tabindex="-1"></a>when using at least 1000 genes.</span>
<span id="cb10-665"><a href="#cb10-665" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-668"><a href="#cb10-668" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-669"><a href="#cb10-669" aria-hidden="true" tabindex="-1"></a>lisi_scores <span class="ot">&lt;-</span> <span class="fu">lapply</span>(</span>
<span id="cb10-670"><a href="#cb10-670" aria-hidden="true" tabindex="-1"></a>    pca_fits_nb_genes, </span>
<span id="cb10-671"><a href="#cb10-671" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(pca_fit_nb_gene){</span>
<span id="cb10-672"><a href="#cb10-672" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb10-673"><a href="#cb10-673" aria-hidden="true" tabindex="-1"></a>        perplexity_param <span class="ot">&lt;-</span> <span class="dv">30</span></span>
<span id="cb10-674"><a href="#cb10-674" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb10-675"><a href="#cb10-675" aria-hidden="true" tabindex="-1"></a>        lisi_scores_pc1_pc2 <span class="ot">&lt;-</span> lisi<span class="sc">::</span><span class="fu">compute_lisi</span>(</span>
<span id="cb10-676"><a href="#cb10-676" aria-hidden="true" tabindex="-1"></a>            pca_fit_nb_gene<span class="sc">$</span>rotated[, <span class="fu">c</span>(<span class="st">"PC1"</span>, <span class="st">"PC2"</span>)],</span>
<span id="cb10-677"><a href="#cb10-677" aria-hidden="true" tabindex="-1"></a>            pca_fit_nb_gene<span class="sc">$</span>metadata,</span>
<span id="cb10-678"><a href="#cb10-678" aria-hidden="true" tabindex="-1"></a>            <span class="fu">c</span>(<span class="st">"cohort"</span>, <span class="st">"er_status"</span>), </span>
<span id="cb10-679"><a href="#cb10-679" aria-hidden="true" tabindex="-1"></a>            <span class="at">perplexity =</span> perplexity_param</span>
<span id="cb10-680"><a href="#cb10-680" aria-hidden="true" tabindex="-1"></a>        ) <span class="sc">%&gt;%</span> </span>
<span id="cb10-681"><a href="#cb10-681" aria-hidden="true" tabindex="-1"></a>            <span class="st">`</span><span class="at">colnames&lt;-</span><span class="st">`</span>(<span class="fu">paste0</span>(<span class="st">"pc1_pc2_"</span>, <span class="fu">colnames</span>(.)))</span>
<span id="cb10-682"><a href="#cb10-682" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb10-683"><a href="#cb10-683" aria-hidden="true" tabindex="-1"></a>        lisi_scores_pc3_pc4 <span class="ot">&lt;-</span> lisi<span class="sc">::</span><span class="fu">compute_lisi</span>(</span>
<span id="cb10-684"><a href="#cb10-684" aria-hidden="true" tabindex="-1"></a>            pca_fit_nb_gene<span class="sc">$</span>rotated[, <span class="fu">c</span>(<span class="st">"PC3"</span>, <span class="st">"PC4"</span>)],</span>
<span id="cb10-685"><a href="#cb10-685" aria-hidden="true" tabindex="-1"></a>            pca_fit_nb_gene<span class="sc">$</span>metadata,</span>
<span id="cb10-686"><a href="#cb10-686" aria-hidden="true" tabindex="-1"></a>            <span class="fu">c</span>(<span class="st">"cohort"</span>, <span class="st">"er_status"</span>), </span>
<span id="cb10-687"><a href="#cb10-687" aria-hidden="true" tabindex="-1"></a>            <span class="at">perplexity =</span> perplexity_param</span>
<span id="cb10-688"><a href="#cb10-688" aria-hidden="true" tabindex="-1"></a>        ) <span class="sc">%&gt;%</span> </span>
<span id="cb10-689"><a href="#cb10-689" aria-hidden="true" tabindex="-1"></a>            <span class="st">`</span><span class="at">colnames&lt;-</span><span class="st">`</span>(<span class="fu">paste0</span>(<span class="st">"pc3_pc4_"</span>, <span class="fu">colnames</span>(.)))</span>
<span id="cb10-690"><a href="#cb10-690" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb10-691"><a href="#cb10-691" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">bind_cols</span>(</span>
<span id="cb10-692"><a href="#cb10-692" aria-hidden="true" tabindex="-1"></a>            lisi_scores_pc1_pc2,</span>
<span id="cb10-693"><a href="#cb10-693" aria-hidden="true" tabindex="-1"></a>            lisi_scores_pc3_pc4</span>
<span id="cb10-694"><a href="#cb10-694" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb10-695"><a href="#cb10-695" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb10-696"><a href="#cb10-696" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb10-697"><a href="#cb10-697" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-698"><a href="#cb10-698" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-699"><a href="#cb10-699" aria-hidden="true" tabindex="-1"></a>summary_lisi_scores <span class="ot">&lt;-</span> <span class="fu">lapply</span>(</span>
<span id="cb10-700"><a href="#cb10-700" aria-hidden="true" tabindex="-1"></a>    lisi_scores, </span>
<span id="cb10-701"><a href="#cb10-701" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(df){</span>
<span id="cb10-702"><a href="#cb10-702" aria-hidden="true" tabindex="-1"></a>        <span class="fu">apply</span>(df, <span class="dv">2</span>, mean)</span>
<span id="cb10-703"><a href="#cb10-703" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb10-704"><a href="#cb10-704" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb10-705"><a href="#cb10-705" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(<span class="at">.id =</span> <span class="st">"nb_genes"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-706"><a href="#cb10-706" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">nb =</span> stringr<span class="sc">::</span><span class="fu">str_replace_all</span>(nb_genes, <span class="st">"nb_"</span>, <span class="st">""</span>))</span>
<span id="cb10-707"><a href="#cb10-707" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-708"><a href="#cb10-708" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-709"><a href="#cb10-709" aria-hidden="true" tabindex="-1"></a>Due to the difference in cohort sizes, METABRIC has 648 samples and </span>
<span id="cb10-710"><a href="#cb10-710" aria-hidden="true" tabindex="-1"></a>TCGA 352, the expected iLISI coefficient is:</span>
<span id="cb10-711"><a href="#cb10-711" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-714"><a href="#cb10-714" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-715"><a href="#cb10-715" aria-hidden="true" tabindex="-1"></a>nb_tcga <span class="ot">&lt;-</span> <span class="fu">unname</span>(<span class="fu">table</span>(pca_fit<span class="sc">$</span>metadata<span class="sc">$</span>cohort)[<span class="st">"tcga"</span>])</span>
<span id="cb10-716"><a href="#cb10-716" aria-hidden="true" tabindex="-1"></a>nb_metabric <span class="ot">&lt;-</span> <span class="fu">unname</span>(<span class="fu">table</span>(pca_fit<span class="sc">$</span>metadata<span class="sc">$</span>cohort)[<span class="st">"metabric"</span>])</span>
<span id="cb10-717"><a href="#cb10-717" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-718"><a href="#cb10-718" aria-hidden="true" tabindex="-1"></a>ilisi_coefficient <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">/</span>((nb_tcga<span class="sc">/</span>(nb_tcga <span class="sc">+</span> nb_metabric))<span class="sc">^</span><span class="dv">2</span> <span class="sc">+</span></span>
<span id="cb10-719"><a href="#cb10-719" aria-hidden="true" tabindex="-1"></a>                            (nb_metabric<span class="sc">/</span>(nb_tcga<span class="sc">+</span>nb_metabric))<span class="sc">^</span><span class="dv">2</span>)</span>
<span id="cb10-720"><a href="#cb10-720" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-721"><a href="#cb10-721" aria-hidden="true" tabindex="-1"></a>expected_ilisi_vals <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">round</span>(ilisi_coefficient, <span class="at">digits =</span> <span class="dv">2</span>))</span>
<span id="cb10-722"><a href="#cb10-722" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-723"><a href="#cb10-723" aria-hidden="true" tabindex="-1"></a><span class="fu">round</span>(ilisi_coefficient, <span class="at">digits =</span> <span class="dv">2</span>)</span>
<span id="cb10-724"><a href="#cb10-724" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-725"><a href="#cb10-725" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-726"><a href="#cb10-726" aria-hidden="true" tabindex="-1"></a>The expected value for the iLISI coefficient when calculating for </span>
<span id="cb10-727"><a href="#cb10-727" aria-hidden="true" tabindex="-1"></a>ER status is: </span>
<span id="cb10-728"><a href="#cb10-728" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-731"><a href="#cb10-731" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-732"><a href="#cb10-732" aria-hidden="true" tabindex="-1"></a>nb_pos <span class="ot">&lt;-</span> <span class="fu">unname</span>(<span class="fu">table</span>(pca_fit<span class="sc">$</span>metadata<span class="sc">$</span>er_status)[<span class="st">"pos"</span>])</span>
<span id="cb10-733"><a href="#cb10-733" aria-hidden="true" tabindex="-1"></a>nb_neg <span class="ot">&lt;-</span> <span class="fu">unname</span>(<span class="fu">table</span>(pca_fit<span class="sc">$</span>metadata<span class="sc">$</span>er_status)[<span class="st">"neg"</span>])</span>
<span id="cb10-734"><a href="#cb10-734" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-735"><a href="#cb10-735" aria-hidden="true" tabindex="-1"></a>ilisi_coefficient <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">/</span>((nb_pos<span class="sc">/</span>(nb_pos <span class="sc">+</span> nb_neg))<span class="sc">^</span><span class="dv">2</span> <span class="sc">+</span></span>
<span id="cb10-736"><a href="#cb10-736" aria-hidden="true" tabindex="-1"></a>                            (nb_neg<span class="sc">/</span>(nb_pos<span class="sc">+</span>nb_neg))<span class="sc">^</span><span class="dv">2</span>)</span>
<span id="cb10-737"><a href="#cb10-737" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-738"><a href="#cb10-738" aria-hidden="true" tabindex="-1"></a>expected_ilisi_vals <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb10-739"><a href="#cb10-739" aria-hidden="true" tabindex="-1"></a>    expected_ilisi_vals, </span>
<span id="cb10-740"><a href="#cb10-740" aria-hidden="true" tabindex="-1"></a>    <span class="fu">round</span>(ilisi_coefficient, <span class="at">digits =</span> <span class="dv">2</span>)</span>
<span id="cb10-741"><a href="#cb10-741" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-742"><a href="#cb10-742" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-743"><a href="#cb10-743" aria-hidden="true" tabindex="-1"></a>expected_ilisi_vals <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb10-744"><a href="#cb10-744" aria-hidden="true" tabindex="-1"></a>    <span class="at">type_mixing =</span> <span class="fu">c</span>(<span class="st">"cohort"</span>, <span class="st">"er_status"</span>),</span>
<span id="cb10-745"><a href="#cb10-745" aria-hidden="true" tabindex="-1"></a>    <span class="at">expected_ilisi =</span> expected_ilisi_vals</span>
<span id="cb10-746"><a href="#cb10-746" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-747"><a href="#cb10-747" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-748"><a href="#cb10-748" aria-hidden="true" tabindex="-1"></a><span class="fu">round</span>(ilisi_coefficient, <span class="at">digits =</span> <span class="dv">2</span>)</span>
<span id="cb10-749"><a href="#cb10-749" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-750"><a href="#cb10-750" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-751"><a href="#cb10-751" aria-hidden="true" tabindex="-1"></a>And the plot below shows the iLISI numbers for each combination of PC1+PC2 and</span>
<span id="cb10-752"><a href="#cb10-752" aria-hidden="true" tabindex="-1"></a>PC3+PC4 annotated by the total number of genes. An integration by ER</span>
<span id="cb10-753"><a href="#cb10-753" aria-hidden="true" tabindex="-1"></a>status means that the combination of components is not capturing the biology.</span>
<span id="cb10-754"><a href="#cb10-754" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-755"><a href="#cb10-755" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-756"><a href="#cb10-756" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=8, fig.height=4}</span></span>
<span id="cb10-757"><a href="#cb10-757" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-lisi-scores</span></span>
<span id="cb10-758"><a href="#cb10-758" aria-hidden="true" tabindex="-1"></a>summary_lisi_scores_pc1_pc2 <span class="ot">&lt;-</span>  summary_lisi_scores <span class="sc">%&gt;%</span></span>
<span id="cb10-759"><a href="#cb10-759" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(</span>
<span id="cb10-760"><a href="#cb10-760" aria-hidden="true" tabindex="-1"></a>        <span class="at">cols =</span> <span class="fu">c</span>(pc1_pc2_cohort, pc1_pc2_er_status),</span>
<span id="cb10-761"><a href="#cb10-761" aria-hidden="true" tabindex="-1"></a>        <span class="at">names_to =</span> <span class="st">"type_mixing"</span>,</span>
<span id="cb10-762"><a href="#cb10-762" aria-hidden="true" tabindex="-1"></a>        <span class="at">values_to =</span> <span class="st">"pc1_pc2"</span></span>
<span id="cb10-763"><a href="#cb10-763" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb10-764"><a href="#cb10-764" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(pc1_pc2, type_mixing, nb) <span class="sc">%&gt;%</span></span>
<span id="cb10-765"><a href="#cb10-765" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb10-766"><a href="#cb10-766" aria-hidden="true" tabindex="-1"></a>        <span class="at">type_mixing =</span> stringr<span class="sc">::</span><span class="fu">str_replace</span>(type_mixing, <span class="st">"pc1_pc2_"</span>, <span class="st">""</span>)</span>
<span id="cb10-767"><a href="#cb10-767" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb10-768"><a href="#cb10-768" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-769"><a href="#cb10-769" aria-hidden="true" tabindex="-1"></a>summary_lisi_scores_pc3_pc4 <span class="ot">&lt;-</span>  summary_lisi_scores <span class="sc">%&gt;%</span></span>
<span id="cb10-770"><a href="#cb10-770" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(</span>
<span id="cb10-771"><a href="#cb10-771" aria-hidden="true" tabindex="-1"></a>        <span class="at">cols =</span> <span class="fu">c</span>(pc3_pc4_cohort, pc3_pc4_er_status),</span>
<span id="cb10-772"><a href="#cb10-772" aria-hidden="true" tabindex="-1"></a>        <span class="at">names_to =</span> <span class="st">"type_mixing"</span>,</span>
<span id="cb10-773"><a href="#cb10-773" aria-hidden="true" tabindex="-1"></a>        <span class="at">values_to =</span> <span class="st">"pc3_pc4"</span></span>
<span id="cb10-774"><a href="#cb10-774" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb10-775"><a href="#cb10-775" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(pc3_pc4, type_mixing, nb) <span class="sc">%&gt;%</span></span>
<span id="cb10-776"><a href="#cb10-776" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb10-777"><a href="#cb10-777" aria-hidden="true" tabindex="-1"></a>        <span class="at">type_mixing =</span> stringr<span class="sc">::</span><span class="fu">str_replace</span>(type_mixing, <span class="st">"pc3_pc4_"</span>, <span class="st">""</span>)</span>
<span id="cb10-778"><a href="#cb10-778" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb10-779"><a href="#cb10-779" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-780"><a href="#cb10-780" aria-hidden="true" tabindex="-1"></a>summary_lisi_scores_long <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">inner_join</span>(</span>
<span id="cb10-781"><a href="#cb10-781" aria-hidden="true" tabindex="-1"></a>    summary_lisi_scores_pc1_pc2,</span>
<span id="cb10-782"><a href="#cb10-782" aria-hidden="true" tabindex="-1"></a>    summary_lisi_scores_pc3_pc4,</span>
<span id="cb10-783"><a href="#cb10-783" aria-hidden="true" tabindex="-1"></a>    <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"type_mixing"</span>, <span class="st">"nb"</span>)</span>
<span id="cb10-784"><a href="#cb10-784" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb10-785"><a href="#cb10-785" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">is_above_1000 =</span> <span class="fu">as.numeric</span>(nb) <span class="sc">&gt;=</span> <span class="dv">1000</span>)</span>
<span id="cb10-786"><a href="#cb10-786" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-787"><a href="#cb10-787" aria-hidden="true" tabindex="-1"></a>summary_lisi_scores_long <span class="sc">%&gt;%</span> </span>
<span id="cb10-788"><a href="#cb10-788" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(</span>
<span id="cb10-789"><a href="#cb10-789" aria-hidden="true" tabindex="-1"></a>        <span class="fu">aes</span>(<span class="at">x =</span> pc1_pc2, <span class="at">y =</span> pc3_pc4, <span class="at">label =</span> nb, <span class="at">color =</span> is_above_1000),</span>
<span id="cb10-790"><a href="#cb10-790" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb10-791"><a href="#cb10-791" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_vline</span>(</span>
<span id="cb10-792"><a href="#cb10-792" aria-hidden="true" tabindex="-1"></a>        <span class="at">data =</span> expected_ilisi_vals,</span>
<span id="cb10-793"><a href="#cb10-793" aria-hidden="true" tabindex="-1"></a>        <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">xintercept =</span> expected_ilisi),</span>
<span id="cb10-794"><a href="#cb10-794" aria-hidden="true" tabindex="-1"></a>        <span class="at">linetype =</span> <span class="st">"dashed"</span></span>
<span id="cb10-795"><a href="#cb10-795" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb10-796"><a href="#cb10-796" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_hline</span>(</span>
<span id="cb10-797"><a href="#cb10-797" aria-hidden="true" tabindex="-1"></a>        <span class="at">data =</span> expected_ilisi_vals,</span>
<span id="cb10-798"><a href="#cb10-798" aria-hidden="true" tabindex="-1"></a>        <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">yintercept =</span> expected_ilisi),</span>
<span id="cb10-799"><a href="#cb10-799" aria-hidden="true" tabindex="-1"></a>        <span class="at">linetype =</span> <span class="st">"dashed"</span></span>
<span id="cb10-800"><a href="#cb10-800" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb10-801"><a href="#cb10-801" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span> </span>
<span id="cb10-802"><a href="#cb10-802" aria-hidden="true" tabindex="-1"></a>    ggrepel<span class="sc">::</span><span class="fu">geom_label_repel</span>(<span class="at">max.overlaps =</span> <span class="dv">12</span>) <span class="sc">+</span> </span>
<span id="cb10-803"><a href="#cb10-803" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb10-804"><a href="#cb10-804" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"LISI scores for PC1 and PC2"</span>,</span>
<span id="cb10-805"><a href="#cb10-805" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"LISI scores for PC3 and PC4"</span>,</span>
<span id="cb10-806"><a href="#cb10-806" aria-hidden="true" tabindex="-1"></a>        <span class="at">subtitle =</span> </span>
<span id="cb10-807"><a href="#cb10-807" aria-hidden="true" tabindex="-1"></a>            <span class="st">"Expected LISI for Cohort: 1.84</span><span class="sc">\n</span><span class="st">Expected LISI for ER status: 1.59"</span></span>
<span id="cb10-808"><a href="#cb10-808" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb10-809"><a href="#cb10-809" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(</span>
<span id="cb10-810"><a href="#cb10-810" aria-hidden="true" tabindex="-1"></a>        <span class="sc">~</span>type_mixing,</span>
<span id="cb10-811"><a href="#cb10-811" aria-hidden="true" tabindex="-1"></a>        <span class="at">labeller =</span> <span class="fu">as_labeller</span>(</span>
<span id="cb10-812"><a href="#cb10-812" aria-hidden="true" tabindex="-1"></a>            <span class="fu">c</span>(<span class="st">"cohort"</span> <span class="ot">=</span> <span class="st">"Cohort"</span>, <span class="st">"er_status"</span> <span class="ot">=</span> <span class="st">"ER IHC"</span>)</span>
<span id="cb10-813"><a href="#cb10-813" aria-hidden="true" tabindex="-1"></a>        ),</span>
<span id="cb10-814"><a href="#cb10-814" aria-hidden="true" tabindex="-1"></a>        <span class="at">scales =</span> <span class="st">"free"</span></span>
<span id="cb10-815"><a href="#cb10-815" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb10-816"><a href="#cb10-816" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_color_manual</span>(</span>
<span id="cb10-817"><a href="#cb10-817" aria-hidden="true" tabindex="-1"></a>        <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"black"</span>, viridis<span class="sc">::</span><span class="fu">viridis</span>(<span class="at">n =</span> <span class="dv">4</span>)[<span class="dv">3</span>])</span>
<span id="cb10-818"><a href="#cb10-818" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb10-819"><a href="#cb10-819" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">15</span>) <span class="sc">+</span> </span>
<span id="cb10-820"><a href="#cb10-820" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb10-821"><a href="#cb10-821" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-822"><a href="#cb10-822" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-823"><a href="#cb10-823" aria-hidden="true" tabindex="-1"></a><span class="fu">## Stable genes</span></span>
<span id="cb10-824"><a href="#cb10-824" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-825"><a href="#cb10-825" aria-hidden="true" tabindex="-1"></a>One of the raised issues is that the average ranking of the stable genes</span>
<span id="cb10-826"><a href="#cb10-826" aria-hidden="true" tabindex="-1"></a>might depend on phenotypes. Here we show that the distribution of average</span>
<span id="cb10-827"><a href="#cb10-827" aria-hidden="true" tabindex="-1"></a>of stable genes is stable across different phenotypes, such as ER status and</span>
<span id="cb10-828"><a href="#cb10-828" aria-hidden="true" tabindex="-1"></a>molecular subtype.</span>
<span id="cb10-829"><a href="#cb10-829" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-832"><a href="#cb10-832" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-833"><a href="#cb10-833" aria-hidden="true" tabindex="-1"></a>avg_rankings <span class="ot">&lt;-</span> <span class="fu">lapply</span>(</span>
<span id="cb10-834"><a href="#cb10-834" aria-hidden="true" tabindex="-1"></a>    datasets_normalized_og, </span>
<span id="cb10-835"><a href="#cb10-835" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(x) </span>
<span id="cb10-836"><a href="#cb10-836" aria-hidden="true" tabindex="-1"></a>        <span class="fu">colData</span>(x) <span class="sc">%&gt;%</span> </span>
<span id="cb10-837"><a href="#cb10-837" aria-hidden="true" tabindex="-1"></a>        data.frame <span class="sc">%&gt;%</span> </span>
<span id="cb10-838"><a href="#cb10-838" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">select</span>(avg_ranking, sample_name, er_status, pam50)</span>
<span id="cb10-839"><a href="#cb10-839" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-840"><a href="#cb10-840" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(<span class="at">.id =</span> <span class="st">"cohort"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-841"><a href="#cb10-841" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">cohort =</span> <span class="fu">toupper</span>(cohort))</span>
<span id="cb10-842"><a href="#cb10-842" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-843"><a href="#cb10-843" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-844"><a href="#cb10-844" aria-hidden="true" tabindex="-1"></a>First we plot by ER status in both METABRIC and TCGA.</span>
<span id="cb10-845"><a href="#cb10-845" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-846"><a href="#cb10-846" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=4, fig.height=5}</span></span>
<span id="cb10-847"><a href="#cb10-847" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-avg-rank-er-ihc</span></span>
<span id="cb10-848"><a href="#cb10-848" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-849"><a href="#cb10-849" aria-hidden="true" tabindex="-1"></a>er_status_labels <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"neg"</span> <span class="ot">=</span> <span class="st">"Negative"</span>, <span class="st">"pos"</span> <span class="ot">=</span> <span class="st">"Positive"</span>)</span>
<span id="cb10-850"><a href="#cb10-850" aria-hidden="true" tabindex="-1"></a>avg_rankings <span class="sc">%&gt;%</span> </span>
<span id="cb10-851"><a href="#cb10-851" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">!=</span> <span class="st">"SCANB"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-852"><a href="#cb10-852" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">er_status =</span> er_status_labels[er_status]) <span class="sc">%&gt;%</span> </span>
<span id="cb10-853"><a href="#cb10-853" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(</span>
<span id="cb10-854"><a href="#cb10-854" aria-hidden="true" tabindex="-1"></a>        <span class="fu">aes</span>(<span class="at">y =</span> er_status, <span class="at">x =</span> avg_ranking)</span>
<span id="cb10-855"><a href="#cb10-855" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb10-856"><a href="#cb10-856" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_jitter</span>() <span class="sc">+</span> </span>
<span id="cb10-857"><a href="#cb10-857" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_boxplot</span>(</span>
<span id="cb10-858"><a href="#cb10-858" aria-hidden="true" tabindex="-1"></a>        <span class="at">alpha =</span> <span class="fl">0.5</span>,</span>
<span id="cb10-859"><a href="#cb10-859" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">"black"</span></span>
<span id="cb10-860"><a href="#cb10-860" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb10-861"><a href="#cb10-861" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb10-862"><a href="#cb10-862" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"Average ranking of</span><span class="sc">\n</span><span class="st">stable genes"</span>,</span>
<span id="cb10-863"><a href="#cb10-863" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"ER IHC"</span></span>
<span id="cb10-864"><a href="#cb10-864" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb10-865"><a href="#cb10-865" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>cohort, <span class="at">ncol =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb10-866"><a href="#cb10-866" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">coord_cartesian</span>(<span class="at">xlim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>)) <span class="sc">+</span> </span>
<span id="cb10-867"><a href="#cb10-867" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>)</span>
<span id="cb10-868"><a href="#cb10-868" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-869"><a href="#cb10-869" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-870"><a href="#cb10-870" aria-hidden="true" tabindex="-1"></a><span class="fu">## Scaling</span></span>
<span id="cb10-871"><a href="#cb10-871" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-872"><a href="#cb10-872" aria-hidden="true" tabindex="-1"></a>We now show the embedding using no scaling, i.e., just log FPKM and median</span>
<span id="cb10-873"><a href="#cb10-873" aria-hidden="true" tabindex="-1"></a>intensity, using the qPCR like normalization and using a z-normalization</span>
<span id="cb10-874"><a href="#cb10-874" aria-hidden="true" tabindex="-1"></a>on the sample level.</span>
<span id="cb10-875"><a href="#cb10-875" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-878"><a href="#cb10-878" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-879"><a href="#cb10-879" aria-hidden="true" tabindex="-1"></a>samples_for_training <span class="ot">&lt;-</span> pca_fit<span class="sc">$</span>rotated <span class="sc">%&gt;%</span> rownames</span>
<span id="cb10-880"><a href="#cb10-880" aria-hidden="true" tabindex="-1"></a>training_set_no_norm <span class="ot">&lt;-</span> <span class="fu">mapply</span>( </span>
<span id="cb10-881"><a href="#cb10-881" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(sum_exp, i, genes_for_pca){</span>
<span id="cb10-882"><a href="#cb10-882" aria-hidden="true" tabindex="-1"></a>        <span class="fu">assay</span>(sum_exp[genes_for_pca, ], <span class="at">i =</span> i) <span class="sc">%&gt;%</span> </span>
<span id="cb10-883"><a href="#cb10-883" aria-hidden="true" tabindex="-1"></a>            <span class="fu">data.frame</span>(<span class="at">check.names =</span> <span class="cn">FALSE</span>)    </span>
<span id="cb10-884"><a href="#cb10-884" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb10-885"><a href="#cb10-885" aria-hidden="true" tabindex="-1"></a>    datasets_normalized_og[which_cohorts_training],</span>
<span id="cb10-886"><a href="#cb10-886" aria-hidden="true" tabindex="-1"></a>    <span class="at">i =</span> which_exp[which_cohorts_training],</span>
<span id="cb10-887"><a href="#cb10-887" aria-hidden="true" tabindex="-1"></a>    <span class="at">MoreArgs =</span> <span class="fu">list</span>(<span class="at">genes_for_pca =</span> genes_for_pca),</span>
<span id="cb10-888"><a href="#cb10-888" aria-hidden="true" tabindex="-1"></a>    <span class="at">SIMPLIFY =</span> <span class="cn">FALSE</span>,</span>
<span id="cb10-889"><a href="#cb10-889" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span></span>
<span id="cb10-890"><a href="#cb10-890" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-891"><a href="#cb10-891" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">bind_cols</span>() <span class="sc">%&gt;%</span></span>
<span id="cb10-892"><a href="#cb10-892" aria-hidden="true" tabindex="-1"></a>    .[, samples_for_training]</span>
<span id="cb10-893"><a href="#cb10-893" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-894"><a href="#cb10-894" aria-hidden="true" tabindex="-1"></a>pca_fit_no_norm <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">pca</span>(</span>
<span id="cb10-895"><a href="#cb10-895" aria-hidden="true" tabindex="-1"></a>    training_set_no_norm,</span>
<span id="cb10-896"><a href="#cb10-896" aria-hidden="true" tabindex="-1"></a>    <span class="at">metadata =</span> merged_col_data <span class="sc">%&gt;%</span> </span>
<span id="cb10-897"><a href="#cb10-897" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">filter</span>(sample_name <span class="sc">%in%</span> <span class="fu">colnames</span>(training_set_no_norm)) <span class="sc">%&gt;%</span></span>
<span id="cb10-898"><a href="#cb10-898" aria-hidden="true" tabindex="-1"></a>        .[<span class="fu">colnames</span>(training_set_no_norm), ],</span>
<span id="cb10-899"><a href="#cb10-899" aria-hidden="true" tabindex="-1"></a>    <span class="at">center =</span> <span class="cn">FALSE</span>,</span>
<span id="cb10-900"><a href="#cb10-900" aria-hidden="true" tabindex="-1"></a>    <span class="at">scale =</span> <span class="cn">FALSE</span></span>
<span id="cb10-901"><a href="#cb10-901" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-902"><a href="#cb10-902" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-903"><a href="#cb10-903" aria-hidden="true" tabindex="-1"></a>training_set_znorm <span class="ot">&lt;-</span> <span class="fu">mapply</span>( </span>
<span id="cb10-904"><a href="#cb10-904" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(sum_exp, genes_for_pca){</span>
<span id="cb10-905"><a href="#cb10-905" aria-hidden="true" tabindex="-1"></a>        <span class="fu">assay</span>(sum_exp[genes_for_pca, ], <span class="at">i =</span> <span class="st">"znorm"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-906"><a href="#cb10-906" aria-hidden="true" tabindex="-1"></a>            <span class="fu">data.frame</span>(<span class="at">check.names =</span> <span class="cn">FALSE</span>)    </span>
<span id="cb10-907"><a href="#cb10-907" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb10-908"><a href="#cb10-908" aria-hidden="true" tabindex="-1"></a>    datasets_normalized_og[which_cohorts_training],</span>
<span id="cb10-909"><a href="#cb10-909" aria-hidden="true" tabindex="-1"></a>    <span class="at">MoreArgs =</span> <span class="fu">list</span>(<span class="at">genes_for_pca =</span> genes_for_pca),</span>
<span id="cb10-910"><a href="#cb10-910" aria-hidden="true" tabindex="-1"></a>    <span class="at">SIMPLIFY =</span> <span class="cn">FALSE</span>,</span>
<span id="cb10-911"><a href="#cb10-911" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span></span>
<span id="cb10-912"><a href="#cb10-912" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-913"><a href="#cb10-913" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">bind_cols</span>() <span class="sc">%&gt;%</span></span>
<span id="cb10-914"><a href="#cb10-914" aria-hidden="true" tabindex="-1"></a>    .[, samples_for_training]</span>
<span id="cb10-915"><a href="#cb10-915" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-916"><a href="#cb10-916" aria-hidden="true" tabindex="-1"></a>pca_fit_znorm <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">pca</span>(</span>
<span id="cb10-917"><a href="#cb10-917" aria-hidden="true" tabindex="-1"></a>    training_set_znorm,</span>
<span id="cb10-918"><a href="#cb10-918" aria-hidden="true" tabindex="-1"></a>    <span class="at">metadata =</span> merged_col_data <span class="sc">%&gt;%</span> </span>
<span id="cb10-919"><a href="#cb10-919" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">filter</span>(sample_name <span class="sc">%in%</span> <span class="fu">colnames</span>(training_set_znorm)) <span class="sc">%&gt;%</span></span>
<span id="cb10-920"><a href="#cb10-920" aria-hidden="true" tabindex="-1"></a>        .[<span class="fu">colnames</span>(training_set_znorm), ],</span>
<span id="cb10-921"><a href="#cb10-921" aria-hidden="true" tabindex="-1"></a>    <span class="at">center =</span> <span class="cn">FALSE</span>,</span>
<span id="cb10-922"><a href="#cb10-922" aria-hidden="true" tabindex="-1"></a>    <span class="at">scale =</span> <span class="cn">FALSE</span></span>
<span id="cb10-923"><a href="#cb10-923" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-924"><a href="#cb10-924" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-925"><a href="#cb10-925" aria-hidden="true" tabindex="-1"></a>lisi_scores_normalizations <span class="ot">&lt;-</span> <span class="fu">lapply</span>(</span>
<span id="cb10-926"><a href="#cb10-926" aria-hidden="true" tabindex="-1"></a>    <span class="fu">list</span>(<span class="at">qpcr =</span> pca_fit, <span class="at">no_norm =</span> pca_fit_no_norm, <span class="at">znorm =</span> pca_fit_znorm), </span>
<span id="cb10-927"><a href="#cb10-927" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(pca_fit_nb_gene){</span>
<span id="cb10-928"><a href="#cb10-928" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb10-929"><a href="#cb10-929" aria-hidden="true" tabindex="-1"></a>        perplexity_param <span class="ot">&lt;-</span> <span class="dv">30</span></span>
<span id="cb10-930"><a href="#cb10-930" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb10-931"><a href="#cb10-931" aria-hidden="true" tabindex="-1"></a>        lisi_scores_pc1_pc2 <span class="ot">&lt;-</span> lisi<span class="sc">::</span><span class="fu">compute_lisi</span>(</span>
<span id="cb10-932"><a href="#cb10-932" aria-hidden="true" tabindex="-1"></a>            pca_fit_nb_gene<span class="sc">$</span>rotated[, <span class="fu">c</span>(<span class="st">"PC1"</span>, <span class="st">"PC2"</span>)],</span>
<span id="cb10-933"><a href="#cb10-933" aria-hidden="true" tabindex="-1"></a>            pca_fit_nb_gene<span class="sc">$</span>metadata,</span>
<span id="cb10-934"><a href="#cb10-934" aria-hidden="true" tabindex="-1"></a>            <span class="fu">c</span>(<span class="st">"cohort"</span>, <span class="st">"er_status"</span>), </span>
<span id="cb10-935"><a href="#cb10-935" aria-hidden="true" tabindex="-1"></a>            <span class="at">perplexity =</span> perplexity_param</span>
<span id="cb10-936"><a href="#cb10-936" aria-hidden="true" tabindex="-1"></a>        ) <span class="sc">%&gt;%</span> </span>
<span id="cb10-937"><a href="#cb10-937" aria-hidden="true" tabindex="-1"></a>            <span class="st">`</span><span class="at">colnames&lt;-</span><span class="st">`</span>(<span class="fu">paste0</span>(<span class="st">"pc1_pc2_"</span>, <span class="fu">colnames</span>(.)))</span>
<span id="cb10-938"><a href="#cb10-938" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb10-939"><a href="#cb10-939" aria-hidden="true" tabindex="-1"></a>        lisi_scores_pc3_pc4 <span class="ot">&lt;-</span> lisi<span class="sc">::</span><span class="fu">compute_lisi</span>(</span>
<span id="cb10-940"><a href="#cb10-940" aria-hidden="true" tabindex="-1"></a>            pca_fit_nb_gene<span class="sc">$</span>rotated[, <span class="fu">c</span>(<span class="st">"PC3"</span>, <span class="st">"PC4"</span>)],</span>
<span id="cb10-941"><a href="#cb10-941" aria-hidden="true" tabindex="-1"></a>            pca_fit_nb_gene<span class="sc">$</span>metadata,</span>
<span id="cb10-942"><a href="#cb10-942" aria-hidden="true" tabindex="-1"></a>            <span class="fu">c</span>(<span class="st">"cohort"</span>, <span class="st">"er_status"</span>), </span>
<span id="cb10-943"><a href="#cb10-943" aria-hidden="true" tabindex="-1"></a>            <span class="at">perplexity =</span> perplexity_param</span>
<span id="cb10-944"><a href="#cb10-944" aria-hidden="true" tabindex="-1"></a>        ) <span class="sc">%&gt;%</span> </span>
<span id="cb10-945"><a href="#cb10-945" aria-hidden="true" tabindex="-1"></a>            <span class="st">`</span><span class="at">colnames&lt;-</span><span class="st">`</span>(<span class="fu">paste0</span>(<span class="st">"pc3_pc4_"</span>, <span class="fu">colnames</span>(.)))</span>
<span id="cb10-946"><a href="#cb10-946" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb10-947"><a href="#cb10-947" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">bind_cols</span>(</span>
<span id="cb10-948"><a href="#cb10-948" aria-hidden="true" tabindex="-1"></a>            lisi_scores_pc1_pc2,</span>
<span id="cb10-949"><a href="#cb10-949" aria-hidden="true" tabindex="-1"></a>            lisi_scores_pc3_pc4</span>
<span id="cb10-950"><a href="#cb10-950" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb10-951"><a href="#cb10-951" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb10-952"><a href="#cb10-952" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb10-953"><a href="#cb10-953" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-954"><a href="#cb10-954" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-955"><a href="#cb10-955" aria-hidden="true" tabindex="-1"></a>summary_lisi_scores_normalizations <span class="ot">&lt;-</span> <span class="fu">lapply</span>(</span>
<span id="cb10-956"><a href="#cb10-956" aria-hidden="true" tabindex="-1"></a>    lisi_scores_normalizations, </span>
<span id="cb10-957"><a href="#cb10-957" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(df){</span>
<span id="cb10-958"><a href="#cb10-958" aria-hidden="true" tabindex="-1"></a>        <span class="fu">apply</span>(df, <span class="dv">2</span>, mean)</span>
<span id="cb10-959"><a href="#cb10-959" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb10-960"><a href="#cb10-960" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb10-961"><a href="#cb10-961" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(<span class="at">.id =</span> <span class="st">"norm_type"</span>)</span>
<span id="cb10-962"><a href="#cb10-962" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-963"><a href="#cb10-963" aria-hidden="true" tabindex="-1"></a>plots_pca_fit <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb10-964"><a href="#cb10-964" aria-hidden="true" tabindex="-1"></a>point_size <span class="ot">&lt;-</span> <span class="dv">2</span></span>
<span id="cb10-965"><a href="#cb10-965" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-966"><a href="#cb10-966" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>qpcr_norm <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb10-967"><a href="#cb10-967" aria-hidden="true" tabindex="-1"></a>    pca_fit,</span>
<span id="cb10-968"><a href="#cb10-968" aria-hidden="true" tabindex="-1"></a>    <span class="at">colby =</span> <span class="st">"cohort"</span>,</span>
<span id="cb10-969"><a href="#cb10-969" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> <span class="cn">NULL</span>, </span>
<span id="cb10-970"><a href="#cb10-970" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb10-971"><a href="#cb10-971" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC3"</span>,</span>
<span id="cb10-972"><a href="#cb10-972" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC4"</span>,</span>
<span id="cb10-973"><a href="#cb10-973" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> point_size,</span>
<span id="cb10-974"><a href="#cb10-974" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"qPCR-like normalization"</span>,</span>
<span id="cb10-975"><a href="#cb10-975" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="fu">paste0</span>(</span>
<span id="cb10-976"><a href="#cb10-976" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Cohort LISI score: "</span>, </span>
<span id="cb10-977"><a href="#cb10-977" aria-hidden="true" tabindex="-1"></a>        <span class="fu">round</span>(</span>
<span id="cb10-978"><a href="#cb10-978" aria-hidden="true" tabindex="-1"></a>            <span class="fu">as.numeric</span>(summary_lisi_scores_normalizations[<span class="dv">1</span>, <span class="st">"pc3_pc4_cohort"</span>]),</span>
<span id="cb10-979"><a href="#cb10-979" aria-hidden="true" tabindex="-1"></a>            <span class="at">digits =</span> <span class="dv">2</span></span>
<span id="cb10-980"><a href="#cb10-980" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb10-981"><a href="#cb10-981" aria-hidden="true" tabindex="-1"></a>    ), </span>
<span id="cb10-982"><a href="#cb10-982" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlab =</span> <span class="st">"PC3"</span>,</span>
<span id="cb10-983"><a href="#cb10-983" aria-hidden="true" tabindex="-1"></a>    <span class="at">ylab =</span> <span class="st">"PC4"</span></span>
<span id="cb10-984"><a href="#cb10-984" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-985"><a href="#cb10-985" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-986"><a href="#cb10-986" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>no_norm <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb10-987"><a href="#cb10-987" aria-hidden="true" tabindex="-1"></a>    pca_fit_no_norm,</span>
<span id="cb10-988"><a href="#cb10-988" aria-hidden="true" tabindex="-1"></a>    <span class="at">colby =</span> <span class="st">"cohort"</span>,</span>
<span id="cb10-989"><a href="#cb10-989" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> <span class="cn">NULL</span>, </span>
<span id="cb10-990"><a href="#cb10-990" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb10-991"><a href="#cb10-991" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC3"</span>,</span>
<span id="cb10-992"><a href="#cb10-992" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC4"</span>,</span>
<span id="cb10-993"><a href="#cb10-993" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> point_size,</span>
<span id="cb10-994"><a href="#cb10-994" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"No normalization"</span>,</span>
<span id="cb10-995"><a href="#cb10-995" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="fu">paste0</span>(</span>
<span id="cb10-996"><a href="#cb10-996" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Cohort LISI score: "</span>, </span>
<span id="cb10-997"><a href="#cb10-997" aria-hidden="true" tabindex="-1"></a>        <span class="fu">round</span>(</span>
<span id="cb10-998"><a href="#cb10-998" aria-hidden="true" tabindex="-1"></a>            <span class="fu">as.numeric</span>(summary_lisi_scores_normalizations[<span class="dv">2</span>, <span class="st">"pc3_pc4_cohort"</span>]),</span>
<span id="cb10-999"><a href="#cb10-999" aria-hidden="true" tabindex="-1"></a>            <span class="at">digits =</span> <span class="dv">2</span></span>
<span id="cb10-1000"><a href="#cb10-1000" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb10-1001"><a href="#cb10-1001" aria-hidden="true" tabindex="-1"></a>    ), </span>
<span id="cb10-1002"><a href="#cb10-1002" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlab =</span> <span class="st">"PC3"</span>,</span>
<span id="cb10-1003"><a href="#cb10-1003" aria-hidden="true" tabindex="-1"></a>    <span class="at">ylab =</span> <span class="st">"PC4"</span></span>
<span id="cb10-1004"><a href="#cb10-1004" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-1005"><a href="#cb10-1005" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1006"><a href="#cb10-1006" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>znorm <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb10-1007"><a href="#cb10-1007" aria-hidden="true" tabindex="-1"></a>    pca_fit_znorm,</span>
<span id="cb10-1008"><a href="#cb10-1008" aria-hidden="true" tabindex="-1"></a>    <span class="at">colby =</span> <span class="st">"cohort"</span>,</span>
<span id="cb10-1009"><a href="#cb10-1009" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> <span class="cn">NULL</span>, </span>
<span id="cb10-1010"><a href="#cb10-1010" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb10-1011"><a href="#cb10-1011" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC3"</span>,</span>
<span id="cb10-1012"><a href="#cb10-1012" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC4"</span>,</span>
<span id="cb10-1013"><a href="#cb10-1013" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> point_size,</span>
<span id="cb10-1014"><a href="#cb10-1014" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Z-scaling normalization"</span>,</span>
<span id="cb10-1015"><a href="#cb10-1015" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="fu">paste0</span>(</span>
<span id="cb10-1016"><a href="#cb10-1016" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Cohort LISI score: "</span>, </span>
<span id="cb10-1017"><a href="#cb10-1017" aria-hidden="true" tabindex="-1"></a>        <span class="fu">round</span>(</span>
<span id="cb10-1018"><a href="#cb10-1018" aria-hidden="true" tabindex="-1"></a>            <span class="fu">as.numeric</span>(summary_lisi_scores_normalizations[<span class="dv">3</span>, <span class="st">"pc3_pc4_cohort"</span>]),</span>
<span id="cb10-1019"><a href="#cb10-1019" aria-hidden="true" tabindex="-1"></a>            <span class="at">digits =</span> <span class="dv">2</span></span>
<span id="cb10-1020"><a href="#cb10-1020" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb10-1021"><a href="#cb10-1021" aria-hidden="true" tabindex="-1"></a>    ), </span>
<span id="cb10-1022"><a href="#cb10-1022" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlab =</span> <span class="st">"PC3"</span>,</span>
<span id="cb10-1023"><a href="#cb10-1023" aria-hidden="true" tabindex="-1"></a>    <span class="at">ylab =</span> <span class="st">"PC4"</span></span>
<span id="cb10-1024"><a href="#cb10-1024" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-1025"><a href="#cb10-1025" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-1026"><a href="#cb10-1026" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1027"><a href="#cb10-1027" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=20, fig.height=5}</span></span>
<span id="cb10-1028"><a href="#cb10-1028" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-comparison-plots</span></span>
<span id="cb10-1029"><a href="#cb10-1029" aria-hidden="true" tabindex="-1"></a>plots_pca_fit <span class="ot">&lt;-</span> <span class="fu">lapply</span>(</span>
<span id="cb10-1030"><a href="#cb10-1030" aria-hidden="true" tabindex="-1"></a>    plots_pca_fit,</span>
<span id="cb10-1031"><a href="#cb10-1031" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(p){</span>
<span id="cb10-1032"><a href="#cb10-1032" aria-hidden="true" tabindex="-1"></a>        p <span class="sc">+</span> </span>
<span id="cb10-1033"><a href="#cb10-1033" aria-hidden="true" tabindex="-1"></a>            ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">color =</span> <span class="st">"Cohort"</span>)</span>
<span id="cb10-1034"><a href="#cb10-1034" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb10-1035"><a href="#cb10-1035" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-1036"><a href="#cb10-1036" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1037"><a href="#cb10-1037" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(</span>
<span id="cb10-1038"><a href="#cb10-1038" aria-hidden="true" tabindex="-1"></a>    <span class="at">plotlist =</span> plots_pca_fit,</span>
<span id="cb10-1039"><a href="#cb10-1039" aria-hidden="true" tabindex="-1"></a>    <span class="at">nrow =</span> <span class="dv">1</span></span>
<span id="cb10-1040"><a href="#cb10-1040" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-1041"><a href="#cb10-1041" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-1042"><a href="#cb10-1042" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1043"><a href="#cb10-1043" aria-hidden="true" tabindex="-1"></a>And now we compare the iLISI scores of the qPCR-like normalization and z-scaling</span>
<span id="cb10-1044"><a href="#cb10-1044" aria-hidden="true" tabindex="-1"></a>as well.</span>
<span id="cb10-1045"><a href="#cb10-1045" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1046"><a href="#cb10-1046" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=5, fig.height=4}</span></span>
<span id="cb10-1047"><a href="#cb10-1047" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-differences-scaling</span></span>
<span id="cb10-1048"><a href="#cb10-1048" aria-hidden="true" tabindex="-1"></a>fit_lisi_norms <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(</span>
<span id="cb10-1049"><a href="#cb10-1049" aria-hidden="true" tabindex="-1"></a>    lisi_scores_normalizations[<span class="fu">c</span>(<span class="st">"qpcr"</span>, <span class="st">"znorm"</span>)],</span>
<span id="cb10-1050"><a href="#cb10-1050" aria-hidden="true" tabindex="-1"></a>    <span class="at">.id =</span> <span class="st">"norm_type"</span></span>
<span id="cb10-1051"><a href="#cb10-1051" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb10-1052"><a href="#cb10-1052" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span></span>
<span id="cb10-1053"><a href="#cb10-1053" aria-hidden="true" tabindex="-1"></a>    rstanarm<span class="sc">::</span><span class="fu">stan_lm</span>(</span>
<span id="cb10-1054"><a href="#cb10-1054" aria-hidden="true" tabindex="-1"></a>        pc3_pc4_cohort <span class="sc">~</span> norm_type, </span>
<span id="cb10-1055"><a href="#cb10-1055" aria-hidden="true" tabindex="-1"></a>        <span class="at">data =</span> .,</span>
<span id="cb10-1056"><a href="#cb10-1056" aria-hidden="true" tabindex="-1"></a>        <span class="at">prior=</span><span class="cn">NULL</span>,</span>
<span id="cb10-1057"><a href="#cb10-1057" aria-hidden="true" tabindex="-1"></a>        <span class="at">refresh =</span> <span class="dv">0</span></span>
<span id="cb10-1058"><a href="#cb10-1058" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb10-1059"><a href="#cb10-1059" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1060"><a href="#cb10-1060" aria-hidden="true" tabindex="-1"></a>dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(</span>
<span id="cb10-1061"><a href="#cb10-1061" aria-hidden="true" tabindex="-1"></a>    lisi_scores_normalizations[<span class="fu">c</span>(<span class="st">"qpcr"</span>, <span class="st">"znorm"</span>)],</span>
<span id="cb10-1062"><a href="#cb10-1062" aria-hidden="true" tabindex="-1"></a>    <span class="at">.id =</span> <span class="st">"norm_type"</span></span>
<span id="cb10-1063"><a href="#cb10-1063" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb10-1064"><a href="#cb10-1064" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span></span>
<span id="cb10-1065"><a href="#cb10-1065" aria-hidden="true" tabindex="-1"></a>    modelr<span class="sc">::</span><span class="fu">data_grid</span>(norm_type) <span class="sc">%&gt;%</span></span>
<span id="cb10-1066"><a href="#cb10-1066" aria-hidden="true" tabindex="-1"></a>    tidybayes<span class="sc">::</span><span class="fu">add_epred_draws</span>(fit_lisi_norms) <span class="sc">%&gt;%</span></span>
<span id="cb10-1067"><a href="#cb10-1067" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">pivot_wider</span>(</span>
<span id="cb10-1068"><a href="#cb10-1068" aria-hidden="true" tabindex="-1"></a>        <span class="at">id_cols =</span> <span class="fu">c</span>(<span class="st">".draw"</span>),</span>
<span id="cb10-1069"><a href="#cb10-1069" aria-hidden="true" tabindex="-1"></a>        <span class="at">names_from =</span> <span class="st">"norm_type"</span>, </span>
<span id="cb10-1070"><a href="#cb10-1070" aria-hidden="true" tabindex="-1"></a>        <span class="at">values_from =</span> <span class="st">".epred"</span></span>
<span id="cb10-1071"><a href="#cb10-1071" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb10-1072"><a href="#cb10-1072" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">difference =</span> <span class="st">`</span><span class="at">qpcr</span><span class="st">`</span> <span class="sc">-</span> <span class="st">`</span><span class="at">znorm</span><span class="st">`</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-1073"><a href="#cb10-1073" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> difference)) <span class="sc">+</span></span>
<span id="cb10-1074"><a href="#cb10-1074" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span> </span>
<span id="cb10-1075"><a href="#cb10-1075" aria-hidden="true" tabindex="-1"></a>    tidybayes<span class="sc">::</span><span class="fu">stat_dotsinterval</span>(<span class="at">quantiles =</span> <span class="dv">100</span>, <span class="at">.width =</span> <span class="fu">c</span>(.<span class="dv">66</span>, .<span class="dv">95</span>)) <span class="sc">+</span> </span>
<span id="cb10-1076"><a href="#cb10-1076" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb10-1077"><a href="#cb10-1077" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="fu">paste0</span>(</span>
<span id="cb10-1078"><a href="#cb10-1078" aria-hidden="true" tabindex="-1"></a>            <span class="st">"Difference from expection of</span><span class="sc">\n</span><span class="st">the posterior predictive"</span>,</span>
<span id="cb10-1079"><a href="#cb10-1079" aria-hidden="true" tabindex="-1"></a>            <span class="st">" distribution"</span></span>
<span id="cb10-1080"><a href="#cb10-1080" aria-hidden="true" tabindex="-1"></a>        ),</span>
<span id="cb10-1081"><a href="#cb10-1081" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="cn">NULL</span></span>
<span id="cb10-1082"><a href="#cb10-1082" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb10-1083"><a href="#cb10-1083" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">18</span>) <span class="sc">+</span> </span>
<span id="cb10-1084"><a href="#cb10-1084" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme</span>(</span>
<span id="cb10-1085"><a href="#cb10-1085" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb10-1086"><a href="#cb10-1086" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.ticks.y =</span> <span class="fu">element_blank</span>()</span>
<span id="cb10-1087"><a href="#cb10-1087" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb10-1088"><a href="#cb10-1088" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span> </span>
<span id="cb10-1089"><a href="#cb10-1089" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb10-1090"><a href="#cb10-1090" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-1091"><a href="#cb10-1091" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1092"><a href="#cb10-1092" aria-hidden="true" tabindex="-1"></a><span class="fu">## Tumor purity: statistical tests</span></span>
<span id="cb10-1093"><a href="#cb10-1093" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1094"><a href="#cb10-1094" aria-hidden="true" tabindex="-1"></a>One of the points raised by the reviewers was about the statistics</span>
<span id="cb10-1095"><a href="#cb10-1095" aria-hidden="true" tabindex="-1"></a>on the differences between the principal components stratified by </span>
<span id="cb10-1096"><a href="#cb10-1096" aria-hidden="true" tabindex="-1"></a>molecular subtype. We perform ANOVA followed by a Tukey post hoc </span>
<span id="cb10-1097"><a href="#cb10-1097" aria-hidden="true" tabindex="-1"></a>test.</span>
<span id="cb10-1098"><a href="#cb10-1098" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1101"><a href="#cb10-1101" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-1102"><a href="#cb10-1102" aria-hidden="true" tabindex="-1"></a>pc3_tests_purity <span class="ot">&lt;-</span> <span class="fu">sapply</span>(</span>
<span id="cb10-1103"><a href="#cb10-1103" aria-hidden="true" tabindex="-1"></a>    df_pca<span class="sc">$</span>pam50 <span class="sc">%&gt;%</span> unique <span class="sc">%&gt;%</span> .[<span class="dv">2</span><span class="sc">:</span><span class="fu">length</span>(.)], </span>
<span id="cb10-1104"><a href="#cb10-1104" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(mol_sub, df_pca){</span>
<span id="cb10-1105"><a href="#cb10-1105" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb10-1106"><a href="#cb10-1106" aria-hidden="true" tabindex="-1"></a>        df_pca <span class="sc">%&gt;%</span></span>
<span id="cb10-1107"><a href="#cb10-1107" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb10-1108"><a href="#cb10-1108" aria-hidden="true" tabindex="-1"></a>                cohort <span class="sc">==</span> <span class="st">"metabric"</span> <span class="sc">&amp;</span></span>
<span id="cb10-1109"><a href="#cb10-1109" aria-hidden="true" tabindex="-1"></a>                    CELLULARITY <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Low"</span>, <span class="st">"Moderate"</span>, <span class="st">"High"</span>) <span class="sc">&amp;</span></span>
<span id="cb10-1110"><a href="#cb10-1110" aria-hidden="true" tabindex="-1"></a>                    pam50 <span class="sc">==</span> mol_sub</span>
<span id="cb10-1111"><a href="#cb10-1111" aria-hidden="true" tabindex="-1"></a>            ) <span class="sc">%&gt;%</span></span>
<span id="cb10-1112"><a href="#cb10-1112" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb10-1113"><a href="#cb10-1113" aria-hidden="true" tabindex="-1"></a>                <span class="at">tumor_purity =</span> <span class="fu">factor</span>(</span>
<span id="cb10-1114"><a href="#cb10-1114" aria-hidden="true" tabindex="-1"></a>                    CELLULARITY,</span>
<span id="cb10-1115"><a href="#cb10-1115" aria-hidden="true" tabindex="-1"></a>                    <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Low"</span>, <span class="st">"Moderate"</span>, <span class="st">"High"</span>)</span>
<span id="cb10-1116"><a href="#cb10-1116" aria-hidden="true" tabindex="-1"></a>                )</span>
<span id="cb10-1117"><a href="#cb10-1117" aria-hidden="true" tabindex="-1"></a>            ) <span class="sc">%&gt;%</span></span>
<span id="cb10-1118"><a href="#cb10-1118" aria-hidden="true" tabindex="-1"></a>            <span class="fu">lm</span>(PC3 <span class="sc">~</span> CELLULARITY, <span class="at">data =</span> .) <span class="sc">%&gt;%</span></span>
<span id="cb10-1119"><a href="#cb10-1119" aria-hidden="true" tabindex="-1"></a>            aov <span class="sc">%&gt;%</span></span>
<span id="cb10-1120"><a href="#cb10-1120" aria-hidden="true" tabindex="-1"></a>            TukeyHSD <span class="sc">%&gt;%</span></span>
<span id="cb10-1121"><a href="#cb10-1121" aria-hidden="true" tabindex="-1"></a>            broom<span class="sc">::</span><span class="fu">tidy</span>(.)</span>
<span id="cb10-1122"><a href="#cb10-1122" aria-hidden="true" tabindex="-1"></a>    }, </span>
<span id="cb10-1123"><a href="#cb10-1123" aria-hidden="true" tabindex="-1"></a>    <span class="at">df_pca =</span> df_pca,</span>
<span id="cb10-1124"><a href="#cb10-1124" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>, </span>
<span id="cb10-1125"><a href="#cb10-1125" aria-hidden="true" tabindex="-1"></a>    <span class="at">simplify =</span> <span class="cn">FALSE</span></span>
<span id="cb10-1126"><a href="#cb10-1126" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb10-1127"><a href="#cb10-1127" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(<span class="at">.id =</span> <span class="st">"pam50"</span>)</span>
<span id="cb10-1128"><a href="#cb10-1128" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1129"><a href="#cb10-1129" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1130"><a href="#cb10-1130" aria-hidden="true" tabindex="-1"></a>pc4_tests_purity <span class="ot">&lt;-</span> <span class="fu">sapply</span>(</span>
<span id="cb10-1131"><a href="#cb10-1131" aria-hidden="true" tabindex="-1"></a>    df_pca<span class="sc">$</span>pam50 <span class="sc">%&gt;%</span> unique <span class="sc">%&gt;%</span> .[<span class="dv">2</span><span class="sc">:</span><span class="fu">length</span>(.)], </span>
<span id="cb10-1132"><a href="#cb10-1132" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(mol_sub, df_pca){</span>
<span id="cb10-1133"><a href="#cb10-1133" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb10-1134"><a href="#cb10-1134" aria-hidden="true" tabindex="-1"></a>        df_pca <span class="sc">%&gt;%</span></span>
<span id="cb10-1135"><a href="#cb10-1135" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb10-1136"><a href="#cb10-1136" aria-hidden="true" tabindex="-1"></a>                cohort <span class="sc">==</span> <span class="st">"metabric"</span> <span class="sc">&amp;</span></span>
<span id="cb10-1137"><a href="#cb10-1137" aria-hidden="true" tabindex="-1"></a>                    CELLULARITY <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Low"</span>, <span class="st">"Moderate"</span>, <span class="st">"High"</span>) <span class="sc">&amp;</span></span>
<span id="cb10-1138"><a href="#cb10-1138" aria-hidden="true" tabindex="-1"></a>                    pam50 <span class="sc">==</span> mol_sub</span>
<span id="cb10-1139"><a href="#cb10-1139" aria-hidden="true" tabindex="-1"></a>            ) <span class="sc">%&gt;%</span></span>
<span id="cb10-1140"><a href="#cb10-1140" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb10-1141"><a href="#cb10-1141" aria-hidden="true" tabindex="-1"></a>                <span class="at">tumor_purity =</span> <span class="fu">factor</span>(</span>
<span id="cb10-1142"><a href="#cb10-1142" aria-hidden="true" tabindex="-1"></a>                    CELLULARITY,</span>
<span id="cb10-1143"><a href="#cb10-1143" aria-hidden="true" tabindex="-1"></a>                    <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Low"</span>, <span class="st">"Moderate"</span>, <span class="st">"High"</span>)</span>
<span id="cb10-1144"><a href="#cb10-1144" aria-hidden="true" tabindex="-1"></a>                )</span>
<span id="cb10-1145"><a href="#cb10-1145" aria-hidden="true" tabindex="-1"></a>            ) <span class="sc">%&gt;%</span></span>
<span id="cb10-1146"><a href="#cb10-1146" aria-hidden="true" tabindex="-1"></a>            <span class="fu">lm</span>(PC4 <span class="sc">~</span> CELLULARITY, <span class="at">data =</span> .) <span class="sc">%&gt;%</span></span>
<span id="cb10-1147"><a href="#cb10-1147" aria-hidden="true" tabindex="-1"></a>            aov <span class="sc">%&gt;%</span></span>
<span id="cb10-1148"><a href="#cb10-1148" aria-hidden="true" tabindex="-1"></a>            TukeyHSD <span class="sc">%&gt;%</span></span>
<span id="cb10-1149"><a href="#cb10-1149" aria-hidden="true" tabindex="-1"></a>            broom<span class="sc">::</span><span class="fu">tidy</span>(.)</span>
<span id="cb10-1150"><a href="#cb10-1150" aria-hidden="true" tabindex="-1"></a>    }, </span>
<span id="cb10-1151"><a href="#cb10-1151" aria-hidden="true" tabindex="-1"></a>    <span class="at">df_pca =</span> df_pca,</span>
<span id="cb10-1152"><a href="#cb10-1152" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>, </span>
<span id="cb10-1153"><a href="#cb10-1153" aria-hidden="true" tabindex="-1"></a>    <span class="at">simplify =</span> <span class="cn">FALSE</span></span>
<span id="cb10-1154"><a href="#cb10-1154" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-1155"><a href="#cb10-1155" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(<span class="at">.id =</span> <span class="st">"pam50"</span>)</span>
<span id="cb10-1156"><a href="#cb10-1156" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1157"><a href="#cb10-1157" aria-hidden="true" tabindex="-1"></a>pcs_pvalues <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(</span>
<span id="cb10-1158"><a href="#cb10-1158" aria-hidden="true" tabindex="-1"></a>    <span class="fu">list</span>(<span class="at">PC3 =</span> pc3_tests_purity, <span class="at">PC4 =</span> pc4_tests_purity),</span>
<span id="cb10-1159"><a href="#cb10-1159" aria-hidden="true" tabindex="-1"></a>    <span class="at">.id =</span> <span class="st">"PC"</span></span>
<span id="cb10-1160"><a href="#cb10-1160" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-1161"><a href="#cb10-1161" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1162"><a href="#cb10-1162" aria-hidden="true" tabindex="-1"></a>pcs_pvalues_filtered <span class="ot">&lt;-</span> pcs_pvalues <span class="sc">%&gt;%</span></span>
<span id="cb10-1163"><a href="#cb10-1163" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb10-1164"><a href="#cb10-1164" aria-hidden="true" tabindex="-1"></a>        adj.p.value <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> </span>
<span id="cb10-1165"><a href="#cb10-1165" aria-hidden="true" tabindex="-1"></a>            contrast <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Low-Moderate"</span>, <span class="st">"Moderate-High"</span>)</span>
<span id="cb10-1166"><a href="#cb10-1166" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb10-1167"><a href="#cb10-1167" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb10-1168"><a href="#cb10-1168" aria-hidden="true" tabindex="-1"></a>        <span class="at">y.position =</span> <span class="fu">c</span>(<span class="dv">7</span>, <span class="dv">7</span>, <span class="dv">7</span>, <span class="dv">7</span>, <span class="dv">7</span>)</span>
<span id="cb10-1169"><a href="#cb10-1169" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb10-1170"><a href="#cb10-1170" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">separate</span>(</span>
<span id="cb10-1171"><a href="#cb10-1171" aria-hidden="true" tabindex="-1"></a>        <span class="at">col =</span> <span class="st">"contrast"</span>,</span>
<span id="cb10-1172"><a href="#cb10-1172" aria-hidden="true" tabindex="-1"></a>        <span class="at">into =</span> <span class="fu">c</span>(<span class="st">"group1"</span>, <span class="st">"group2"</span>),</span>
<span id="cb10-1173"><a href="#cb10-1173" aria-hidden="true" tabindex="-1"></a>        <span class="at">sep =</span> <span class="st">"-"</span>,</span>
<span id="cb10-1174"><a href="#cb10-1174" aria-hidden="true" tabindex="-1"></a>        <span class="at">remove =</span> <span class="cn">FALSE</span></span>
<span id="cb10-1175"><a href="#cb10-1175" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb10-1176"><a href="#cb10-1176" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb10-1177"><a href="#cb10-1177" aria-hidden="true" tabindex="-1"></a>        <span class="at">p.adj =</span> adj.p.value,</span>
<span id="cb10-1178"><a href="#cb10-1178" aria-hidden="true" tabindex="-1"></a>        <span class="at">pam50 =</span> <span class="fu">factor</span>(</span>
<span id="cb10-1179"><a href="#cb10-1179" aria-hidden="true" tabindex="-1"></a>            pam50, </span>
<span id="cb10-1180"><a href="#cb10-1180" aria-hidden="true" tabindex="-1"></a>            <span class="at">levels =</span> <span class="fu">c</span>(</span>
<span id="cb10-1181"><a href="#cb10-1181" aria-hidden="true" tabindex="-1"></a>                <span class="st">"basal"</span>, <span class="st">"her2"</span>, <span class="st">"luma"</span>, <span class="st">"lumb"</span>, <span class="st">"normal"</span>, <span class="st">"claudin-low"</span></span>
<span id="cb10-1182"><a href="#cb10-1182" aria-hidden="true" tabindex="-1"></a>            )</span>
<span id="cb10-1183"><a href="#cb10-1183" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb10-1184"><a href="#cb10-1184" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb10-1185"><a href="#cb10-1185" aria-hidden="true" tabindex="-1"></a>    rstatix<span class="sc">::</span><span class="fu">add_significance</span>(<span class="at">p.col =</span> <span class="st">"p.adj"</span>, <span class="at">output.col =</span> <span class="st">"p.adj.signif"</span>)</span>
<span id="cb10-1186"><a href="#cb10-1186" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-1187"><a href="#cb10-1187" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1188"><a href="#cb10-1188" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=14, fig.height=6}</span></span>
<span id="cb10-1189"><a href="#cb10-1189" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-tumor-purity-metabric</span></span>
<span id="cb10-1190"><a href="#cb10-1190" aria-hidden="true" tabindex="-1"></a>df_pca <span class="sc">%&gt;%</span></span>
<span id="cb10-1191"><a href="#cb10-1191" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb10-1192"><a href="#cb10-1192" aria-hidden="true" tabindex="-1"></a>        cohort <span class="sc">==</span> <span class="st">"metabric"</span> <span class="sc">&amp;</span></span>
<span id="cb10-1193"><a href="#cb10-1193" aria-hidden="true" tabindex="-1"></a>            CELLULARITY <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Low"</span>, <span class="st">"Moderate"</span>, <span class="st">"High"</span>)</span>
<span id="cb10-1194"><a href="#cb10-1194" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb10-1195"><a href="#cb10-1195" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb10-1196"><a href="#cb10-1196" aria-hidden="true" tabindex="-1"></a>        <span class="at">tumor_purity =</span> <span class="fu">factor</span>(</span>
<span id="cb10-1197"><a href="#cb10-1197" aria-hidden="true" tabindex="-1"></a>            CELLULARITY,</span>
<span id="cb10-1198"><a href="#cb10-1198" aria-hidden="true" tabindex="-1"></a>            <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Low"</span>, <span class="st">"Moderate"</span>, <span class="st">"High"</span>)</span>
<span id="cb10-1199"><a href="#cb10-1199" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb10-1200"><a href="#cb10-1200" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb10-1201"><a href="#cb10-1201" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(</span>
<span id="cb10-1202"><a href="#cb10-1202" aria-hidden="true" tabindex="-1"></a>        <span class="at">cols =</span> dplyr<span class="sc">::</span><span class="fu">all_of</span>(<span class="fu">c</span>(<span class="st">"PC3"</span>, <span class="st">"PC4"</span>)),</span>
<span id="cb10-1203"><a href="#cb10-1203" aria-hidden="true" tabindex="-1"></a>        <span class="at">names_to =</span> <span class="st">"PC"</span>,</span>
<span id="cb10-1204"><a href="#cb10-1204" aria-hidden="true" tabindex="-1"></a>        <span class="at">values_to =</span> <span class="st">"value"</span></span>
<span id="cb10-1205"><a href="#cb10-1205" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb10-1206"><a href="#cb10-1206" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(</span>
<span id="cb10-1207"><a href="#cb10-1207" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> tumor_purity, <span class="at">y =</span> value, <span class="at">color =</span> pam50</span>
<span id="cb10-1208"><a href="#cb10-1208" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">+</span> </span>
<span id="cb10-1209"><a href="#cb10-1209" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_jitter</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span> </span>
<span id="cb10-1210"><a href="#cb10-1210" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_violin</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb10-1211"><a href="#cb10-1211" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_x_discrete</span>(<span class="at">guide =</span> <span class="fu">guide_axis</span>(<span class="at">angle =</span> <span class="dv">45</span>)) <span class="sc">+</span> </span>
<span id="cb10-1212"><a href="#cb10-1212" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb10-1213"><a href="#cb10-1213" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">"Molecular</span><span class="sc">\n</span><span class="st">subtype"</span>,</span>
<span id="cb10-1214"><a href="#cb10-1214" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"PC"</span>,</span>
<span id="cb10-1215"><a href="#cb10-1215" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"Tumor purity"</span></span>
<span id="cb10-1216"><a href="#cb10-1216" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb10-1217"><a href="#cb10-1217" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_grid</span>(</span>
<span id="cb10-1218"><a href="#cb10-1218" aria-hidden="true" tabindex="-1"></a>        PC <span class="sc">~</span> pam50, </span>
<span id="cb10-1219"><a href="#cb10-1219" aria-hidden="true" tabindex="-1"></a>        <span class="at">scales =</span> <span class="st">"free_y"</span>,</span>
<span id="cb10-1220"><a href="#cb10-1220" aria-hidden="true" tabindex="-1"></a>        <span class="at">labeller =</span> <span class="fu">as_labeller</span>(<span class="fu">c</span>(</span>
<span id="cb10-1221"><a href="#cb10-1221" aria-hidden="true" tabindex="-1"></a>            <span class="fu">c</span>(<span class="st">"PC3"</span> <span class="ot">=</span> <span class="st">"PC3"</span>, <span class="st">"PC4"</span> <span class="ot">=</span> <span class="st">"PC4"</span>, mol_subs)</span>
<span id="cb10-1222"><a href="#cb10-1222" aria-hidden="true" tabindex="-1"></a>        ))</span>
<span id="cb10-1223"><a href="#cb10-1223" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb10-1224"><a href="#cb10-1224" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>) <span class="sc">+</span> </span>
<span id="cb10-1225"><a href="#cb10-1225" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">guides</span>(</span>
<span id="cb10-1226"><a href="#cb10-1226" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> ggplot2<span class="sc">::</span><span class="fu">guide_legend</span>(<span class="at">override.aes =</span> <span class="fu">list</span>(<span class="at">size =</span> <span class="dv">5</span>, <span class="at">alpha =</span> <span class="dv">1</span>))</span>
<span id="cb10-1227"><a href="#cb10-1227" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb10-1228"><a href="#cb10-1228" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span></span>
<span id="cb10-1229"><a href="#cb10-1229" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_color_manual</span>(</span>
<span id="cb10-1230"><a href="#cb10-1230" aria-hidden="true" tabindex="-1"></a>        <span class="at">labels =</span> mol_subs,</span>
<span id="cb10-1231"><a href="#cb10-1231" aria-hidden="true" tabindex="-1"></a>        <span class="at">values =</span> <span class="fu">get_colors_pam50</span>(df_pca)</span>
<span id="cb10-1232"><a href="#cb10-1232" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb10-1233"><a href="#cb10-1233" aria-hidden="true" tabindex="-1"></a>    ggprism<span class="sc">::</span><span class="fu">add_pvalue</span>(</span>
<span id="cb10-1234"><a href="#cb10-1234" aria-hidden="true" tabindex="-1"></a>        pcs_pvalues_filtered,</span>
<span id="cb10-1235"><a href="#cb10-1235" aria-hidden="true" tabindex="-1"></a>        <span class="at">bracket.nudge.y =</span> <span class="dv">1</span>, </span>
<span id="cb10-1236"><a href="#cb10-1236" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">"black"</span>,</span>
<span id="cb10-1237"><a href="#cb10-1237" aria-hidden="true" tabindex="-1"></a>        <span class="at">alpha =</span> <span class="fl">0.5</span>, </span>
<span id="cb10-1238"><a href="#cb10-1238" aria-hidden="true" tabindex="-1"></a>        <span class="at">label.size =</span> <span class="dv">6</span>, </span>
<span id="cb10-1239"><a href="#cb10-1239" aria-hidden="true" tabindex="-1"></a>        <span class="at">bracket.size =</span> <span class="dv">1</span></span>
<span id="cb10-1240"><a href="#cb10-1240" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb10-1241"><a href="#cb10-1241" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">coord_cartesian</span>(<span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">11</span>, <span class="dv">10</span>))</span>
<span id="cb10-1242"><a href="#cb10-1242" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-1243"><a href="#cb10-1243" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1244"><a href="#cb10-1244" aria-hidden="true" tabindex="-1"></a><span class="fu">## Metastasis and FFPE data</span></span>
<span id="cb10-1245"><a href="#cb10-1245" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1246"><a href="#cb10-1246" aria-hidden="true" tabindex="-1"></a>Accessed 20240425:</span>
<span id="cb10-1247"><a href="#cb10-1247" aria-hidden="true" tabindex="-1"></a>https://www.cbioportal.org/study/summary?id=brca_mbcproject_2022</span>
<span id="cb10-1248"><a href="#cb10-1248" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1249"><a href="#cb10-1249" aria-hidden="true" tabindex="-1"></a>In the cBioportal one can check the clinical data directly with</span>
<span id="cb10-1250"><a href="#cb10-1250" aria-hidden="true" tabindex="-1"></a>all the information. When hovering over the column names it is possible</span>
<span id="cb10-1251"><a href="#cb10-1251" aria-hidden="true" tabindex="-1"></a>to obtain the column names from the data available when downloaded. </span>
<span id="cb10-1252"><a href="#cb10-1252" aria-hidden="true" tabindex="-1"></a>For example, the ER percentage is available from the sample in the diagnostics.</span>
<span id="cb10-1253"><a href="#cb10-1253" aria-hidden="true" tabindex="-1"></a>Ki67, similarly. In general the metastasis do not have ER percentage.</span>
<span id="cb10-1254"><a href="#cb10-1254" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1257"><a href="#cb10-1257" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-1258"><a href="#cb10-1258" aria-hidden="true" tabindex="-1"></a>rsem_data <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(</span>
<span id="cb10-1259"><a href="#cb10-1259" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../../data/brca_mbcproject_2022/data_mrna_seq_v2_rsem.txt"</span>,</span>
<span id="cb10-1260"><a href="#cb10-1260" aria-hidden="true" tabindex="-1"></a>    <span class="at">check.names =</span> <span class="cn">FALSE</span></span>
<span id="cb10-1261"><a href="#cb10-1261" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-1262"><a href="#cb10-1262" aria-hidden="true" tabindex="-1"></a>rsem_data <span class="ot">&lt;-</span> rsem_data <span class="sc">%&gt;%</span></span>
<span id="cb10-1263"><a href="#cb10-1263" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">duplicated</span>(Hugo_Symbol))</span>
<span id="cb10-1264"><a href="#cb10-1264" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(rsem_data) <span class="ot">&lt;-</span> rsem_data<span class="sc">$</span>Hugo_Symbol</span>
<span id="cb10-1265"><a href="#cb10-1265" aria-hidden="true" tabindex="-1"></a>rsem_data<span class="sc">$</span>Hugo_Symbol <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb10-1266"><a href="#cb10-1266" aria-hidden="true" tabindex="-1"></a>rsem_data<span class="sc">$</span>Entrez_Gene_Id <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb10-1267"><a href="#cb10-1267" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1268"><a href="#cb10-1268" aria-hidden="true" tabindex="-1"></a>clin_data <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(</span>
<span id="cb10-1269"><a href="#cb10-1269" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../../data/brca_mbcproject_2022/data_clinical_sample.txt"</span>,</span>
<span id="cb10-1270"><a href="#cb10-1270" aria-hidden="true" tabindex="-1"></a>    <span class="at">check.names =</span> <span class="cn">FALSE</span>, </span>
<span id="cb10-1271"><a href="#cb10-1271" aria-hidden="true" tabindex="-1"></a>    <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, </span>
<span id="cb10-1272"><a href="#cb10-1272" aria-hidden="true" tabindex="-1"></a>    <span class="at">skip =</span> <span class="dv">4</span></span>
<span id="cb10-1273"><a href="#cb10-1273" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb10-1274"><a href="#cb10-1274" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(SAMPLE_ID <span class="sc">%in%</span> <span class="fu">colnames</span>(rsem_data))</span>
<span id="cb10-1275"><a href="#cb10-1275" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1276"><a href="#cb10-1276" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(clin_data) <span class="ot">&lt;-</span> clin_data<span class="sc">$</span>SAMPLE_ID</span>
<span id="cb10-1277"><a href="#cb10-1277" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1278"><a href="#cb10-1278" aria-hidden="true" tabindex="-1"></a>clin_data_patient <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(</span>
<span id="cb10-1279"><a href="#cb10-1279" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../../data/brca_mbcproject_2022/data_clinical_patient.txt"</span>,</span>
<span id="cb10-1280"><a href="#cb10-1280" aria-hidden="true" tabindex="-1"></a>    <span class="at">check.names =</span> <span class="cn">FALSE</span>, </span>
<span id="cb10-1281"><a href="#cb10-1281" aria-hidden="true" tabindex="-1"></a>    <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, </span>
<span id="cb10-1282"><a href="#cb10-1282" aria-hidden="true" tabindex="-1"></a>    <span class="at">skip =</span> <span class="dv">4</span></span>
<span id="cb10-1283"><a href="#cb10-1283" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-1284"><a href="#cb10-1284" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1285"><a href="#cb10-1285" aria-hidden="true" tabindex="-1"></a>timeline_specimen <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(</span>
<span id="cb10-1286"><a href="#cb10-1286" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../../data/brca_mbcproject_2022/data_timeline_specimen.txt"</span>,</span>
<span id="cb10-1287"><a href="#cb10-1287" aria-hidden="true" tabindex="-1"></a>    <span class="at">check.names =</span> <span class="cn">FALSE</span>, </span>
<span id="cb10-1288"><a href="#cb10-1288" aria-hidden="true" tabindex="-1"></a>    <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span></span>
<span id="cb10-1289"><a href="#cb10-1289" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-1290"><a href="#cb10-1290" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(</span>
<span id="cb10-1291"><a href="#cb10-1291" aria-hidden="true" tabindex="-1"></a>        PATIENT_ID, </span>
<span id="cb10-1292"><a href="#cb10-1292" aria-hidden="true" tabindex="-1"></a>        SPECIMEN_REFERENCE_NUMBER, </span>
<span id="cb10-1293"><a href="#cb10-1293" aria-hidden="true" tabindex="-1"></a>        START_DATE, </span>
<span id="cb10-1294"><a href="#cb10-1294" aria-hidden="true" tabindex="-1"></a>        SPECIMEN_SITE, </span>
<span id="cb10-1295"><a href="#cb10-1295" aria-hidden="true" tabindex="-1"></a>        SPECIMEN_TYPE</span>
<span id="cb10-1296"><a href="#cb10-1296" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span> </span>
<span id="cb10-1297"><a href="#cb10-1297" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">SAMPLE_TIMEPOINT =</span> SPECIMEN_REFERENCE_NUMBER)</span>
<span id="cb10-1298"><a href="#cb10-1298" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1299"><a href="#cb10-1299" aria-hidden="true" tabindex="-1"></a>clin_data <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">inner_join</span>(</span>
<span id="cb10-1300"><a href="#cb10-1300" aria-hidden="true" tabindex="-1"></a>    clin_data,</span>
<span id="cb10-1301"><a href="#cb10-1301" aria-hidden="true" tabindex="-1"></a>    timeline_specimen,</span>
<span id="cb10-1302"><a href="#cb10-1302" aria-hidden="true" tabindex="-1"></a>    <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"PATIENT_ID"</span>, <span class="st">"SAMPLE_TIMEPOINT"</span>)</span>
<span id="cb10-1303"><a href="#cb10-1303" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-1304"><a href="#cb10-1304" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1305"><a href="#cb10-1305" aria-hidden="true" tabindex="-1"></a>clin_data_all <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">inner_join</span>(</span>
<span id="cb10-1306"><a href="#cb10-1306" aria-hidden="true" tabindex="-1"></a>    clin_data,</span>
<span id="cb10-1307"><a href="#cb10-1307" aria-hidden="true" tabindex="-1"></a>    clin_data_patient,</span>
<span id="cb10-1308"><a href="#cb10-1308" aria-hidden="true" tabindex="-1"></a>    <span class="at">by =</span> <span class="st">"PATIENT_ID"</span></span>
<span id="cb10-1309"><a href="#cb10-1309" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb10-1310"><a href="#cb10-1310" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">separate</span>(</span>
<span id="cb10-1311"><a href="#cb10-1311" aria-hidden="true" tabindex="-1"></a>        AGE_AT_DIAGNOSIS,</span>
<span id="cb10-1312"><a href="#cb10-1312" aria-hidden="true" tabindex="-1"></a>        <span class="at">into =</span> <span class="fu">c</span>(<span class="st">"lower"</span>, <span class="st">"upper"</span>),</span>
<span id="cb10-1313"><a href="#cb10-1313" aria-hidden="true" tabindex="-1"></a>        <span class="at">remove =</span> <span class="cn">FALSE</span>,</span>
<span id="cb10-1314"><a href="#cb10-1314" aria-hidden="true" tabindex="-1"></a>        <span class="at">sep =</span> <span class="st">"-"</span></span>
<span id="cb10-1315"><a href="#cb10-1315" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span> </span>
<span id="cb10-1316"><a href="#cb10-1316" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb10-1317"><a href="#cb10-1317" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb10-1318"><a href="#cb10-1318" aria-hidden="true" tabindex="-1"></a>        <span class="at">AGE_AT_DIAGNOSIS_OG =</span> AGE_AT_DIAGNOSIS,</span>
<span id="cb10-1319"><a href="#cb10-1319" aria-hidden="true" tabindex="-1"></a>        <span class="at">AGE_AT_DIAGNOSIS =</span> <span class="fu">mean</span>(<span class="fu">c</span>(<span class="fu">as.numeric</span>(lower), <span class="fu">as.numeric</span>(upper))),</span>
<span id="cb10-1320"><a href="#cb10-1320" aria-hidden="true" tabindex="-1"></a>        <span class="at">lower =</span> <span class="cn">NULL</span>,</span>
<span id="cb10-1321"><a href="#cb10-1321" aria-hidden="true" tabindex="-1"></a>        <span class="at">upper =</span> <span class="cn">NULL</span></span>
<span id="cb10-1322"><a href="#cb10-1322" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb10-1323"><a href="#cb10-1323" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">ungroup</span>()</span>
<span id="cb10-1324"><a href="#cb10-1324" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1325"><a href="#cb10-1325" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(clin_data_all) <span class="ot">&lt;-</span> clin_data_all<span class="sc">$</span>SAMPLE_ID</span>
<span id="cb10-1326"><a href="#cb10-1326" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-1327"><a href="#cb10-1327" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1330"><a href="#cb10-1330" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-1331"><a href="#cb10-1331" aria-hidden="true" tabindex="-1"></a>matched_samples <span class="ot">&lt;-</span> clin_data <span class="sc">%&gt;%</span> </span>
<span id="cb10-1332"><a href="#cb10-1332" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb10-1333"><a href="#cb10-1333" aria-hidden="true" tabindex="-1"></a>        PATIENT_ID <span class="sc">%in%</span> <span class="fu">names</span>(<span class="fu">which</span>(<span class="fu">table</span>(clin_data<span class="sc">$</span>PATIENT_ID) <span class="sc">&gt;=</span> <span class="dv">2</span>))</span>
<span id="cb10-1334"><a href="#cb10-1334" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb10-1335"><a href="#cb10-1335" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">group_by</span>(PATIENT_ID) <span class="sc">%&gt;%</span></span>
<span id="cb10-1336"><a href="#cb10-1336" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">summarise</span>(</span>
<span id="cb10-1337"><a href="#cb10-1337" aria-hidden="true" tabindex="-1"></a>        <span class="at">matched_samples =</span> </span>
<span id="cb10-1338"><a href="#cb10-1338" aria-hidden="true" tabindex="-1"></a>            <span class="fu">paste0</span>(BX_LOCATION[<span class="fu">order</span>(BX_LOCATION)], <span class="at">collapse =</span> <span class="st">" - "</span>),</span>
<span id="cb10-1339"><a href="#cb10-1339" aria-hidden="true" tabindex="-1"></a>        <span class="at">CALC_MET_SETTING =</span> CALC_MET_SETTING[<span class="dv">1</span>]</span>
<span id="cb10-1340"><a href="#cb10-1340" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb10-1341"><a href="#cb10-1341" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1342"><a href="#cb10-1342" aria-hidden="true" tabindex="-1"></a>matched_samples <span class="sc">%&gt;%</span></span>
<span id="cb10-1343"><a href="#cb10-1343" aria-hidden="true" tabindex="-1"></a>    janitor<span class="sc">::</span><span class="fu">tabyl</span>(matched_samples) <span class="sc">%&gt;%</span></span>
<span id="cb10-1344"><a href="#cb10-1344" aria-hidden="true" tabindex="-1"></a>    DT<span class="sc">::</span><span class="fu">datatable</span>()</span>
<span id="cb10-1345"><a href="#cb10-1345" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-1346"><a href="#cb10-1346" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1347"><a href="#cb10-1347" aria-hidden="true" tabindex="-1"></a>We now try to understand if samples are coming from core biopsy, </span>
<span id="cb10-1348"><a href="#cb10-1348" aria-hidden="true" tabindex="-1"></a>surgical procedures or from a metastasis setting. To start with that, </span>
<span id="cb10-1349"><a href="#cb10-1349" aria-hidden="true" tabindex="-1"></a>we look at a single patient ID that has two breast samples.</span>
<span id="cb10-1350"><a href="#cb10-1350" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1353"><a href="#cb10-1353" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-1354"><a href="#cb10-1354" aria-hidden="true" tabindex="-1"></a>clin_data <span class="sc">%&gt;%</span> </span>
<span id="cb10-1355"><a href="#cb10-1355" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(PATIENT_ID <span class="sc">==</span> <span class="st">"MBCProject_4MF1FlFQ"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-1356"><a href="#cb10-1356" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(BX_TIME_DAYS, CALC_TREATMENT_NAIVE, BX_TYPE)</span>
<span id="cb10-1357"><a href="#cb10-1357" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-1358"><a href="#cb10-1358" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1359"><a href="#cb10-1359" aria-hidden="true" tabindex="-1"></a>For patient with ID <span class="in">`MBCProject_4MF1FlFQ`</span>, there are two samples. One</span>
<span id="cb10-1360"><a href="#cb10-1360" aria-hidden="true" tabindex="-1"></a>at time of diagnosis, core biopsy, and the other at a mastectomy after</span>
<span id="cb10-1361"><a href="#cb10-1361" aria-hidden="true" tabindex="-1"></a>114 days of treatment. The column CALC_TREATMENT_NAIVE tells if the sample</span>
<span id="cb10-1362"><a href="#cb10-1362" aria-hidden="true" tabindex="-1"></a>is treatment naive or not, BX_TYPE corresponds to the type of surgery </span>
<span id="cb10-1363"><a href="#cb10-1363" aria-hidden="true" tabindex="-1"></a>(PATH Procedure Type column on cBioPortal).</span>
<span id="cb10-1364"><a href="#cb10-1364" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1365"><a href="#cb10-1365" aria-hidden="true" tabindex="-1"></a>We now look at another patient (<span class="in">`MBCProject_5gHasou8`</span>):</span>
<span id="cb10-1366"><a href="#cb10-1366" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1369"><a href="#cb10-1369" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-1370"><a href="#cb10-1370" aria-hidden="true" tabindex="-1"></a>clin_data <span class="sc">%&gt;%</span> </span>
<span id="cb10-1371"><a href="#cb10-1371" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(PATIENT_ID <span class="sc">==</span> <span class="st">"MBCProject_5gHasou8"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-1372"><a href="#cb10-1372" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(BX_TIME_DAYS, CALC_TREATMENT_NAIVE, BX_TYPE)</span>
<span id="cb10-1373"><a href="#cb10-1373" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-1374"><a href="#cb10-1374" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1375"><a href="#cb10-1375" aria-hidden="true" tabindex="-1"></a>The results are similar for this patient too. We now move on and use</span>
<span id="cb10-1376"><a href="#cb10-1376" aria-hidden="true" tabindex="-1"></a>EMBER on all available samples and then compare these matched samples.</span>
<span id="cb10-1377"><a href="#cb10-1377" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1380"><a href="#cb10-1380" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-1381"><a href="#cb10-1381" aria-hidden="true" tabindex="-1"></a>sum_exp <span class="ot">&lt;-</span> DESeq2<span class="sc">::</span><span class="fu">DESeqDataSetFromMatrix</span>(</span>
<span id="cb10-1382"><a href="#cb10-1382" aria-hidden="true" tabindex="-1"></a>    <span class="at">countData =</span> <span class="fu">round</span>(rsem_data[, <span class="fu">rownames</span>(clin_data_all)])[</span>
<span id="cb10-1383"><a href="#cb10-1383" aria-hidden="true" tabindex="-1"></a>        <span class="fu">rowSums</span>(rsem_data) <span class="sc">&gt;</span> <span class="dv">10</span>,</span>
<span id="cb10-1384"><a href="#cb10-1384" aria-hidden="true" tabindex="-1"></a>    ],</span>
<span id="cb10-1385"><a href="#cb10-1385" aria-hidden="true" tabindex="-1"></a>    <span class="at">colData =</span> clin_data_all,</span>
<span id="cb10-1386"><a href="#cb10-1386" aria-hidden="true" tabindex="-1"></a>    <span class="at">design =</span> <span class="sc">~</span> <span class="dv">1</span></span>
<span id="cb10-1387"><a href="#cb10-1387" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-1388"><a href="#cb10-1388" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1389"><a href="#cb10-1389" aria-hidden="true" tabindex="-1"></a><span class="fu">assay</span>(sum_exp, <span class="st">"vst"</span>) <span class="ot">&lt;-</span> <span class="fu">assay</span>(DESeq2<span class="sc">::</span><span class="fu">vst</span>(sum_exp))</span>
<span id="cb10-1390"><a href="#cb10-1390" aria-hidden="true" tabindex="-1"></a><span class="fu">assay</span>(sum_exp, <span class="st">"logcpm"</span>) <span class="ot">&lt;-</span> edgeR<span class="sc">::</span><span class="fu">cpm</span>(</span>
<span id="cb10-1391"><a href="#cb10-1391" aria-hidden="true" tabindex="-1"></a>    <span class="fu">assay</span>(sum_exp, <span class="st">"counts"</span>) <span class="sc">%&gt;%</span> as.matrix,</span>
<span id="cb10-1392"><a href="#cb10-1392" aria-hidden="true" tabindex="-1"></a>    <span class="at">log =</span> <span class="cn">TRUE</span></span>
<span id="cb10-1393"><a href="#cb10-1393" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-1394"><a href="#cb10-1394" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1395"><a href="#cb10-1395" aria-hidden="true" tabindex="-1"></a>sum_exp_normalized <span class="ot">&lt;-</span> <span class="fu">get_final_ranking_values</span>(</span>
<span id="cb10-1396"><a href="#cb10-1396" aria-hidden="true" tabindex="-1"></a>    <span class="at">sum_exp =</span> sum_exp,</span>
<span id="cb10-1397"><a href="#cb10-1397" aria-hidden="true" tabindex="-1"></a>    <span class="at">assay_to_use =</span> <span class="st">"logcpm"</span>,</span>
<span id="cb10-1398"><a href="#cb10-1398" aria-hidden="true" tabindex="-1"></a>    <span class="at">stable_genes =</span> stable_genes,</span>
<span id="cb10-1399"><a href="#cb10-1399" aria-hidden="true" tabindex="-1"></a>    <span class="at">most_variable_genes =</span> <span class="fu">setdiff</span>(<span class="fu">rownames</span>(pca_fit<span class="sc">$</span>loadings), stable_genes)</span>
<span id="cb10-1400"><a href="#cb10-1400" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-1401"><a href="#cb10-1401" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1402"><a href="#cb10-1402" aria-hidden="true" tabindex="-1"></a><span class="co"># calculate the embedding</span></span>
<span id="cb10-1403"><a href="#cb10-1403" aria-hidden="true" tabindex="-1"></a>sum_exp_df_pca <span class="ot">&lt;-</span> <span class="fu">get_pca_coordinates</span>(sum_exp_normalized, pca_fit) <span class="sc">%&gt;%</span></span>
<span id="cb10-1404"><a href="#cb10-1404" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span> </span>
<span id="cb10-1405"><a href="#cb10-1405" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">bind_cols</span>(</span>
<span id="cb10-1406"><a href="#cb10-1406" aria-hidden="true" tabindex="-1"></a>        ., </span>
<span id="cb10-1407"><a href="#cb10-1407" aria-hidden="true" tabindex="-1"></a>        <span class="fu">colData</span>(sum_exp_normalized) <span class="sc">%&gt;%</span> </span>
<span id="cb10-1408"><a href="#cb10-1408" aria-hidden="true" tabindex="-1"></a>            data.frame <span class="sc">%&gt;%</span></span>
<span id="cb10-1409"><a href="#cb10-1409" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb10-1410"><a href="#cb10-1410" aria-hidden="true" tabindex="-1"></a>                <span class="at">sample_name =</span> SAMPLE_ID,</span>
<span id="cb10-1411"><a href="#cb10-1411" aria-hidden="true" tabindex="-1"></a>                <span class="at">cohort =</span> <span class="st">"mets_project"</span>,</span>
<span id="cb10-1412"><a href="#cb10-1412" aria-hidden="true" tabindex="-1"></a>                <span class="at">pam50 =</span> <span class="st">"not_available"</span>,</span>
<span id="cb10-1413"><a href="#cb10-1413" aria-hidden="true" tabindex="-1"></a>                <span class="at">er_status =</span> dplyr<span class="sc">::</span><span class="fu">case_when</span>(</span>
<span id="cb10-1414"><a href="#cb10-1414" aria-hidden="true" tabindex="-1"></a>                    BX_ER <span class="sc">==</span> <span class="st">"NEGATIVE"</span> <span class="sc">~</span> <span class="st">"neg"</span>,</span>
<span id="cb10-1415"><a href="#cb10-1415" aria-hidden="true" tabindex="-1"></a>                    BX_ER <span class="sc">==</span> <span class="st">"POSITIVE"</span> <span class="sc">~</span> <span class="st">"pos"</span>,</span>
<span id="cb10-1416"><a href="#cb10-1416" aria-hidden="true" tabindex="-1"></a>                    <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"not_available"</span></span>
<span id="cb10-1417"><a href="#cb10-1417" aria-hidden="true" tabindex="-1"></a>                ),</span>
<span id="cb10-1418"><a href="#cb10-1418" aria-hidden="true" tabindex="-1"></a>                <span class="at">is_distant_mets =</span> <span class="fu">ifelse</span>(</span>
<span id="cb10-1419"><a href="#cb10-1419" aria-hidden="true" tabindex="-1"></a>                    BX_TIME_DAYS <span class="sc">&gt;</span> <span class="dv">0</span> <span class="sc">&amp;</span> BX_LOCATION <span class="sc">!=</span> <span class="st">"BREAST"</span>,</span>
<span id="cb10-1420"><a href="#cb10-1420" aria-hidden="true" tabindex="-1"></a>                    <span class="st">"Distant Metastasis"</span>,</span>
<span id="cb10-1421"><a href="#cb10-1421" aria-hidden="true" tabindex="-1"></a>                    <span class="st">"Primary or local Mets"</span></span>
<span id="cb10-1422"><a href="#cb10-1422" aria-hidden="true" tabindex="-1"></a>                )</span>
<span id="cb10-1423"><a href="#cb10-1423" aria-hidden="true" tabindex="-1"></a>            )</span>
<span id="cb10-1424"><a href="#cb10-1424" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span> </span>
<span id="cb10-1425"><a href="#cb10-1425" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(., df_pca)</span>
<span id="cb10-1426"><a href="#cb10-1426" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-1427"><a href="#cb10-1427" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1430"><a href="#cb10-1430" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-1431"><a href="#cb10-1431" aria-hidden="true" tabindex="-1"></a>plots_ffpe <span class="ot">&lt;-</span> <span class="fu">sapply</span>(</span>
<span id="cb10-1432"><a href="#cb10-1432" aria-hidden="true" tabindex="-1"></a>    <span class="fu">c</span>(<span class="st">"cohort"</span>, <span class="st">"er_status"</span>), </span>
<span id="cb10-1433"><a href="#cb10-1433" aria-hidden="true" tabindex="-1"></a>    get_plot_new_samples,</span>
<span id="cb10-1434"><a href="#cb10-1434" aria-hidden="true" tabindex="-1"></a>    <span class="at">name_cohort =</span> <span class="st">"mets_project"</span>,</span>
<span id="cb10-1435"><a href="#cb10-1435" aria-hidden="true" tabindex="-1"></a>    <span class="at">df_pca =</span> sum_exp_df_pca <span class="sc">%&gt;%</span></span>
<span id="cb10-1436"><a href="#cb10-1436" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb10-1437"><a href="#cb10-1437" aria-hidden="true" tabindex="-1"></a>            cohort <span class="sc">%in%</span> </span>
<span id="cb10-1438"><a href="#cb10-1438" aria-hidden="true" tabindex="-1"></a>                <span class="fu">c</span>(<span class="st">"mets_project"</span>, <span class="st">"tcga"</span>, <span class="st">"scanb"</span>, <span class="st">"metabric"</span>)</span>
<span id="cb10-1439"><a href="#cb10-1439" aria-hidden="true" tabindex="-1"></a>        ) <span class="sc">%&gt;%</span></span>
<span id="cb10-1440"><a href="#cb10-1440" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">cohort =</span> <span class="fu">factor</span>(cohort, <span class="at">levels =</span> <span class="fu">c</span>(</span>
<span id="cb10-1441"><a href="#cb10-1441" aria-hidden="true" tabindex="-1"></a>            <span class="st">"tcga"</span>, <span class="st">"scanb"</span>, <span class="st">"metabric"</span>, <span class="st">"mets_project"</span></span>
<span id="cb10-1442"><a href="#cb10-1442" aria-hidden="true" tabindex="-1"></a>        ))) <span class="sc">%&gt;%</span></span>
<span id="cb10-1443"><a href="#cb10-1443" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">arrange</span>(cohort),</span>
<span id="cb10-1444"><a href="#cb10-1444" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="cn">NULL</span>,</span>
<span id="cb10-1445"><a href="#cb10-1445" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>, </span>
<span id="cb10-1446"><a href="#cb10-1446" aria-hidden="true" tabindex="-1"></a>    <span class="at">simplify =</span> <span class="cn">FALSE</span>,</span>
<span id="cb10-1447"><a href="#cb10-1447" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC3"</span>,</span>
<span id="cb10-1448"><a href="#cb10-1448" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC4"</span></span>
<span id="cb10-1449"><a href="#cb10-1449" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-1450"><a href="#cb10-1450" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1451"><a href="#cb10-1451" aria-hidden="true" tabindex="-1"></a>plots_ffpe<span class="sc">$</span>cohort <span class="ot">&lt;-</span> plots_ffpe<span class="sc">$</span>cohort <span class="sc">+</span></span>
<span id="cb10-1452"><a href="#cb10-1452" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_alpha_manual</span>(</span>
<span id="cb10-1453"><a href="#cb10-1453" aria-hidden="true" tabindex="-1"></a>        <span class="at">values =</span> <span class="fu">c</span>(.<span class="dv">2</span>, .<span class="dv">2</span>, .<span class="dv">2</span>, <span class="dv">1</span>),</span>
<span id="cb10-1454"><a href="#cb10-1454" aria-hidden="true" tabindex="-1"></a>        <span class="at">guide =</span> <span class="st">"none"</span></span>
<span id="cb10-1455"><a href="#cb10-1455" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb10-1456"><a href="#cb10-1456" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_color_manual</span>(</span>
<span id="cb10-1457"><a href="#cb10-1457" aria-hidden="true" tabindex="-1"></a>        <span class="at">values =</span> <span class="fu">c</span>(viridis<span class="sc">::</span><span class="fu">viridis</span>(<span class="at">n =</span> <span class="dv">3</span>), <span class="st">"black"</span>),</span>
<span id="cb10-1458"><a href="#cb10-1458" aria-hidden="true" tabindex="-1"></a>        <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"TCGA"</span>, <span class="st">"SCAN-B"</span>, <span class="st">"METABRIC"</span>, <span class="st">"METS PROJECT"</span>)</span>
<span id="cb10-1459"><a href="#cb10-1459" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb10-1460"><a href="#cb10-1460" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">title =</span> <span class="cn">NULL</span>, <span class="at">subtitle =</span> <span class="cn">NULL</span>, <span class="at">color =</span> <span class="st">"Cohort"</span>)</span>
<span id="cb10-1461"><a href="#cb10-1461" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-1462"><a href="#cb10-1462" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1463"><a href="#cb10-1463" aria-hidden="true" tabindex="-1"></a>We show all the samples from the project below on top of TCGA,</span>
<span id="cb10-1464"><a href="#cb10-1464" aria-hidden="true" tabindex="-1"></a>SCAN-B and METABRIC.</span>
<span id="cb10-1465"><a href="#cb10-1465" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1466"><a href="#cb10-1466" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=9, fig.height=5}</span></span>
<span id="cb10-1467"><a href="#cb10-1467" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-er-ihc-mets-project</span></span>
<span id="cb10-1468"><a href="#cb10-1468" aria-hidden="true" tabindex="-1"></a>plots_ffpe<span class="sc">$</span>er_status <span class="sc">+</span> </span>
<span id="cb10-1469"><a href="#cb10-1469" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb10-1470"><a href="#cb10-1470" aria-hidden="true" tabindex="-1"></a>        <span class="at">subtitle =</span> <span class="cn">NULL</span>,</span>
<span id="cb10-1471"><a href="#cb10-1471" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">"ER IHC"</span></span>
<span id="cb10-1472"><a href="#cb10-1472" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb10-1473"><a href="#cb10-1473" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_color_manual</span>(</span>
<span id="cb10-1474"><a href="#cb10-1474" aria-hidden="true" tabindex="-1"></a>        <span class="at">values =</span> viridis<span class="sc">::</span><span class="fu">viridis</span>(<span class="at">n=</span><span class="dv">3</span>),</span>
<span id="cb10-1475"><a href="#cb10-1475" aria-hidden="true" tabindex="-1"></a>        <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Negative"</span>, <span class="st">"Not available"</span>, <span class="st">"Positive"</span>)</span>
<span id="cb10-1476"><a href="#cb10-1476" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb10-1477"><a href="#cb10-1477" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span> </span>
<span id="cb10-1478"><a href="#cb10-1478" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb10-1479"><a href="#cb10-1479" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-1480"><a href="#cb10-1480" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1481"><a href="#cb10-1481" aria-hidden="true" tabindex="-1"></a>And now stratified by either local metastasis/primary sample</span>
<span id="cb10-1482"><a href="#cb10-1482" aria-hidden="true" tabindex="-1"></a>or by distant metastasis.</span>
<span id="cb10-1483"><a href="#cb10-1483" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1484"><a href="#cb10-1484" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=12, fig.height=5}</span></span>
<span id="cb10-1485"><a href="#cb10-1485" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-mets-local-distant</span></span>
<span id="cb10-1486"><a href="#cb10-1486" aria-hidden="true" tabindex="-1"></a>sum_exp_df_pca <span class="sc">%&gt;%</span></span>
<span id="cb10-1487"><a href="#cb10-1487" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"mets_project"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-1488"><a href="#cb10-1488" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(</span>
<span id="cb10-1489"><a href="#cb10-1489" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> PC3, <span class="at">y =</span> PC4, <span class="at">group =</span> PATIENT_ID</span>
<span id="cb10-1490"><a href="#cb10-1490" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">+</span> </span>
<span id="cb10-1491"><a href="#cb10-1491" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>(</span>
<span id="cb10-1492"><a href="#cb10-1492" aria-hidden="true" tabindex="-1"></a>        df_pca <span class="sc">%&gt;%</span></span>
<span id="cb10-1493"><a href="#cb10-1493" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"scanb"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb10-1494"><a href="#cb10-1494" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">BX_LOCATION =</span> <span class="st">"NOT AVAILABLE"</span>),</span>
<span id="cb10-1495"><a href="#cb10-1495" aria-hidden="true" tabindex="-1"></a>        <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> PC3, <span class="at">y =</span> PC4, <span class="at">color =</span> pam50),</span>
<span id="cb10-1496"><a href="#cb10-1496" aria-hidden="true" tabindex="-1"></a>        <span class="at">alpha =</span> <span class="fl">0.1</span>,</span>
<span id="cb10-1497"><a href="#cb10-1497" aria-hidden="true" tabindex="-1"></a>        <span class="at">size =</span> <span class="dv">1</span></span>
<span id="cb10-1498"><a href="#cb10-1498" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb10-1499"><a href="#cb10-1499" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>(<span class="at">color =</span> <span class="st">"black"</span>, <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb10-1500"><a href="#cb10-1500" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>is_distant_mets) <span class="sc">+</span> </span>
<span id="cb10-1501"><a href="#cb10-1501" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">coord_cartesian</span>(</span>
<span id="cb10-1502"><a href="#cb10-1502" aria-hidden="true" tabindex="-1"></a>        <span class="at">xlim =</span> <span class="fu">c</span>(<span class="fu">min</span>(sum_exp_df_pca<span class="sc">$</span>PC3), <span class="fu">max</span>(sum_exp_df_pca<span class="sc">$</span>PC3)),</span>
<span id="cb10-1503"><a href="#cb10-1503" aria-hidden="true" tabindex="-1"></a>        <span class="at">ylim =</span> <span class="fu">c</span>(<span class="fu">min</span>(sum_exp_df_pca<span class="sc">$</span>PC4), <span class="fu">max</span>(sum_exp_df_pca<span class="sc">$</span>PC4))</span>
<span id="cb10-1504"><a href="#cb10-1504" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb10-1505"><a href="#cb10-1505" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb10-1506"><a href="#cb10-1506" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">"Molecular</span><span class="sc">\n</span><span class="st">subtype"</span></span>
<span id="cb10-1507"><a href="#cb10-1507" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb10-1508"><a href="#cb10-1508" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_color_manual</span>(</span>
<span id="cb10-1509"><a href="#cb10-1509" aria-hidden="true" tabindex="-1"></a>        <span class="at">values =</span> <span class="fu">get_colors_pam50</span>(df_pca),</span>
<span id="cb10-1510"><a href="#cb10-1510" aria-hidden="true" tabindex="-1"></a>        <span class="at">labels =</span> mol_subs</span>
<span id="cb10-1511"><a href="#cb10-1511" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb10-1512"><a href="#cb10-1512" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>) <span class="sc">+</span> </span>
<span id="cb10-1513"><a href="#cb10-1513" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">guides</span>(</span>
<span id="cb10-1514"><a href="#cb10-1514" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> ggplot2<span class="sc">::</span><span class="fu">guide_legend</span>(</span>
<span id="cb10-1515"><a href="#cb10-1515" aria-hidden="true" tabindex="-1"></a>            <span class="at">override.aes =</span> <span class="fu">list</span>(<span class="at">size =</span> <span class="dv">5</span>, <span class="at">alpha =</span> <span class="dv">1</span>)</span>
<span id="cb10-1516"><a href="#cb10-1516" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb10-1517"><a href="#cb10-1517" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb10-1518"><a href="#cb10-1518" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>()</span>
<span id="cb10-1519"><a href="#cb10-1519" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-1520"><a href="#cb10-1520" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1521"><a href="#cb10-1521" aria-hidden="true" tabindex="-1"></a><span class="fu">## TNBC and ER+ </span></span>
<span id="cb10-1522"><a href="#cb10-1522" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1523"><a href="#cb10-1523" aria-hidden="true" tabindex="-1"></a>Here we use another cohort with both ER+ and ER- samples coming from</span>
<span id="cb10-1524"><a href="#cb10-1524" aria-hidden="true" tabindex="-1"></a>FFPE with two different library preparation protocols: Illumina-Access and</span>
<span id="cb10-1525"><a href="#cb10-1525" aria-hidden="true" tabindex="-1"></a>Nugen-Ovation.</span>
<span id="cb10-1526"><a href="#cb10-1526" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1527"><a href="#cb10-1527" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb10-1528"><a href="#cb10-1528" aria-hidden="true" tabindex="-1"></a>GEOquery<span class="sc">::</span><span class="fu">getGEOSuppFiles</span>(</span>
<span id="cb10-1529"><a href="#cb10-1529" aria-hidden="true" tabindex="-1"></a>    <span class="at">GEO =</span> <span class="st">"GSE130397"</span>,</span>
<span id="cb10-1530"><a href="#cb10-1530" aria-hidden="true" tabindex="-1"></a>    <span class="at">baseDir =</span> <span class="st">"../../data"</span>,</span>
<span id="cb10-1531"><a href="#cb10-1531" aria-hidden="true" tabindex="-1"></a>    <span class="at">makeDirectory =</span> <span class="cn">TRUE</span>, </span>
<span id="cb10-1532"><a href="#cb10-1532" aria-hidden="true" tabindex="-1"></a>    <span class="at">fetch_files =</span> <span class="cn">TRUE</span> </span>
<span id="cb10-1533"><a href="#cb10-1533" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-1534"><a href="#cb10-1534" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1535"><a href="#cb10-1535" aria-hidden="true" tabindex="-1"></a><span class="fu">system</span>(<span class="fu">paste0</span>(</span>
<span id="cb10-1536"><a href="#cb10-1536" aria-hidden="true" tabindex="-1"></a>    <span class="st">"tar xf ../../data/GSE130397/"</span>,</span>
<span id="cb10-1537"><a href="#cb10-1537" aria-hidden="true" tabindex="-1"></a>    <span class="st">"GSE130397_RAW.tar"</span></span>
<span id="cb10-1538"><a href="#cb10-1538" aria-hidden="true" tabindex="-1"></a>))</span>
<span id="cb10-1539"><a href="#cb10-1539" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1540"><a href="#cb10-1540" aria-hidden="true" tabindex="-1"></a><span class="fu">system</span>(<span class="fu">paste0</span>(<span class="st">"gunzip ../../data/GSE130397/*.gz"</span>))</span>
<span id="cb10-1541"><a href="#cb10-1541" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1542"><a href="#cb10-1542" aria-hidden="true" tabindex="-1"></a>col_data_gse <span class="ot">&lt;-</span> GEOquery<span class="sc">::</span><span class="fu">getGEO</span>(</span>
<span id="cb10-1543"><a href="#cb10-1543" aria-hidden="true" tabindex="-1"></a>    <span class="at">GEO =</span> <span class="st">"GSE130397"</span>,</span>
<span id="cb10-1544"><a href="#cb10-1544" aria-hidden="true" tabindex="-1"></a>    <span class="at">destdir =</span> <span class="st">"../../data"</span>, </span>
<span id="cb10-1545"><a href="#cb10-1545" aria-hidden="true" tabindex="-1"></a>    <span class="at">GSEMatrix =</span> <span class="cn">TRUE</span></span>
<span id="cb10-1546"><a href="#cb10-1546" aria-hidden="true" tabindex="-1"></a>)[[<span class="dv">1</span>]] <span class="sc">%&gt;%</span></span>
<span id="cb10-1547"><a href="#cb10-1547" aria-hidden="true" tabindex="-1"></a>    pData</span>
<span id="cb10-1548"><a href="#cb10-1548" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1549"><a href="#cb10-1549" aria-hidden="true" tabindex="-1"></a>gse_data <span class="ot">&lt;-</span> <span class="fu">sapply</span>(</span>
<span id="cb10-1550"><a href="#cb10-1550" aria-hidden="true" tabindex="-1"></a>    <span class="fu">list.files</span>(<span class="st">"../../data/GSE130397/"</span>, <span class="at">pattern =</span> <span class="st">"txt"</span>), </span>
<span id="cb10-1551"><a href="#cb10-1551" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(x){</span>
<span id="cb10-1552"><a href="#cb10-1552" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb10-1553"><a href="#cb10-1553" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(stringr<span class="sc">::</span><span class="fu">str_detect</span>(x, <span class="st">"ACC"</span>)){</span>
<span id="cb10-1554"><a href="#cb10-1554" aria-hidden="true" tabindex="-1"></a>            col_select <span class="ot">&lt;-</span> <span class="st">"rev"</span></span>
<span id="cb10-1555"><a href="#cb10-1555" aria-hidden="true" tabindex="-1"></a>        } <span class="cf">else</span> { </span>
<span id="cb10-1556"><a href="#cb10-1556" aria-hidden="true" tabindex="-1"></a>            col_select <span class="ot">&lt;-</span> <span class="st">"fwd"</span></span>
<span id="cb10-1557"><a href="#cb10-1557" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb10-1558"><a href="#cb10-1558" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb10-1559"><a href="#cb10-1559" aria-hidden="true" tabindex="-1"></a>        <span class="fu">read.table</span>(</span>
<span id="cb10-1560"><a href="#cb10-1560" aria-hidden="true" tabindex="-1"></a>            <span class="fu">paste0</span>(<span class="st">"../../data/GSE130397/"</span>, x),</span>
<span id="cb10-1561"><a href="#cb10-1561" aria-hidden="true" tabindex="-1"></a>            <span class="at">header =</span> <span class="cn">TRUE</span>,</span>
<span id="cb10-1562"><a href="#cb10-1562" aria-hidden="true" tabindex="-1"></a>            <span class="at">row.names =</span> <span class="dv">1</span>,</span>
<span id="cb10-1563"><a href="#cb10-1563" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,</span>
<span id="cb10-1564"><a href="#cb10-1564" aria-hidden="true" tabindex="-1"></a>            <span class="at">check.names =</span> <span class="cn">FALSE</span></span>
<span id="cb10-1565"><a href="#cb10-1565" aria-hidden="true" tabindex="-1"></a>        ) <span class="sc">%&gt;%</span></span>
<span id="cb10-1566"><a href="#cb10-1566" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">select</span>(dplyr<span class="sc">::</span><span class="fu">all_of</span>(<span class="fu">c</span>(col_select)))</span>
<span id="cb10-1567"><a href="#cb10-1567" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb10-1568"><a href="#cb10-1568" aria-hidden="true" tabindex="-1"></a>    <span class="at">simplify =</span> <span class="cn">FALSE</span>,</span>
<span id="cb10-1569"><a href="#cb10-1569" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span></span>
<span id="cb10-1570"><a href="#cb10-1570" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-1571"><a href="#cb10-1571" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(col_data_gse) <span class="ot">&lt;-</span> col_data_gse<span class="sc">$</span>geo_accession</span>
<span id="cb10-1572"><a href="#cb10-1572" aria-hidden="true" tabindex="-1"></a>gse_data_all <span class="ot">&lt;-</span> <span class="fu">sapply</span>(</span>
<span id="cb10-1573"><a href="#cb10-1573" aria-hidden="true" tabindex="-1"></a>    gse_data, </span>
<span id="cb10-1574"><a href="#cb10-1574" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(df){</span>
<span id="cb10-1575"><a href="#cb10-1575" aria-hidden="true" tabindex="-1"></a>        <span class="fu">as.vector</span>(df[, <span class="dv">1</span>])</span>
<span id="cb10-1576"><a href="#cb10-1576" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb10-1577"><a href="#cb10-1577" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-1578"><a href="#cb10-1578" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">rownames&lt;-</span><span class="st">`</span>(<span class="fu">rownames</span>(gse_data[[<span class="dv">1</span>]]))</span>
<span id="cb10-1579"><a href="#cb10-1579" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1580"><a href="#cb10-1580" aria-hidden="true" tabindex="-1"></a>gse_data_all <span class="ot">&lt;-</span> gse_data_all[<span class="fu">rowSums</span>(gse_data_all) <span class="sc">&gt;</span> <span class="dv">100</span>, ]</span>
<span id="cb10-1581"><a href="#cb10-1581" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1582"><a href="#cb10-1582" aria-hidden="true" tabindex="-1"></a>annot_erp_neg <span class="ot">&lt;-</span> annotables<span class="sc">::</span>grch38 <span class="sc">%&gt;%</span> </span>
<span id="cb10-1583"><a href="#cb10-1583" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(ensgene, symbol) <span class="sc">%&gt;%</span></span>
<span id="cb10-1584"><a href="#cb10-1584" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(ensgene <span class="sc">%in%</span> <span class="fu">rownames</span>(gse_data_all)) <span class="sc">%&gt;%</span></span>
<span id="cb10-1585"><a href="#cb10-1585" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">duplicated</span>(symbol)) <span class="sc">%&gt;%</span></span>
<span id="cb10-1586"><a href="#cb10-1586" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span></span>
<span id="cb10-1587"><a href="#cb10-1587" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">rownames&lt;-</span><span class="st">`</span>(.<span class="sc">$</span>ensgene)</span>
<span id="cb10-1588"><a href="#cb10-1588" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1589"><a href="#cb10-1589" aria-hidden="true" tabindex="-1"></a>gse_data_all <span class="ot">&lt;-</span> gse_data_all[<span class="fu">rownames</span>(annot_erp_neg), ]</span>
<span id="cb10-1590"><a href="#cb10-1590" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(gse_data_all) <span class="ot">&lt;-</span> annot_erp_neg<span class="sc">$</span>symbol</span>
<span id="cb10-1591"><a href="#cb10-1591" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1592"><a href="#cb10-1592" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(gse_data_all) <span class="ot">&lt;-</span> stringr<span class="sc">::</span><span class="fu">str_split</span>(</span>
<span id="cb10-1593"><a href="#cb10-1593" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(gse_data_all), </span>
<span id="cb10-1594"><a href="#cb10-1594" aria-hidden="true" tabindex="-1"></a>    <span class="at">pattern =</span> <span class="st">"_"</span>,</span>
<span id="cb10-1595"><a href="#cb10-1595" aria-hidden="true" tabindex="-1"></a>    <span class="at">simplify =</span> <span class="cn">TRUE</span></span>
<span id="cb10-1596"><a href="#cb10-1596" aria-hidden="true" tabindex="-1"></a>)[, <span class="dv">1</span>]</span>
<span id="cb10-1597"><a href="#cb10-1597" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1598"><a href="#cb10-1598" aria-hidden="true" tabindex="-1"></a>gse_sum_exp <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">SummarizedExperiment</span>(</span>
<span id="cb10-1599"><a href="#cb10-1599" aria-hidden="true" tabindex="-1"></a>    <span class="at">assays =</span> <span class="fu">list</span>(</span>
<span id="cb10-1600"><a href="#cb10-1600" aria-hidden="true" tabindex="-1"></a>        <span class="at">counts =</span> gse_data_all,</span>
<span id="cb10-1601"><a href="#cb10-1601" aria-hidden="true" tabindex="-1"></a>        <span class="at">logcpm =</span> edgeR<span class="sc">::</span><span class="fu">cpm</span>(gse_data_all, <span class="at">log =</span> <span class="cn">TRUE</span>)</span>
<span id="cb10-1602"><a href="#cb10-1602" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb10-1603"><a href="#cb10-1603" aria-hidden="true" tabindex="-1"></a>    <span class="at">colData =</span> col_data_gse[<span class="fu">colnames</span>(gse_data_all), ] <span class="sc">%&gt;%</span></span>
<span id="cb10-1604"><a href="#cb10-1604" aria-hidden="true" tabindex="-1"></a>        janitor<span class="sc">::</span><span class="fu">clean_names</span>()</span>
<span id="cb10-1605"><a href="#cb10-1605" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-1606"><a href="#cb10-1606" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1607"><a href="#cb10-1607" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb10-1608"><a href="#cb10-1608" aria-hidden="true" tabindex="-1"></a>    gse_sum_exp,</span>
<span id="cb10-1609"><a href="#cb10-1609" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../../results/rds_files/revision/gse_sum_exp.rds"</span></span>
<span id="cb10-1610"><a href="#cb10-1610" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-1611"><a href="#cb10-1611" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1612"><a href="#cb10-1612" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-1613"><a href="#cb10-1613" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1614"><a href="#cb10-1614" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1617"><a href="#cb10-1617" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-1618"><a href="#cb10-1618" aria-hidden="true" tabindex="-1"></a>gse_sum_exp_normalized <span class="ot">&lt;-</span> <span class="fu">get_final_ranking_values</span>(</span>
<span id="cb10-1619"><a href="#cb10-1619" aria-hidden="true" tabindex="-1"></a>    <span class="at">sum_exp =</span> gse_sum_exp,</span>
<span id="cb10-1620"><a href="#cb10-1620" aria-hidden="true" tabindex="-1"></a>    <span class="at">assay_to_use =</span> <span class="st">"logcpm"</span>,</span>
<span id="cb10-1621"><a href="#cb10-1621" aria-hidden="true" tabindex="-1"></a>    <span class="at">stable_genes =</span> stable_genes,</span>
<span id="cb10-1622"><a href="#cb10-1622" aria-hidden="true" tabindex="-1"></a>    <span class="at">most_variable_genes =</span> <span class="fu">setdiff</span>(<span class="fu">rownames</span>(pca_fit<span class="sc">$</span>loadings), stable_genes)</span>
<span id="cb10-1623"><a href="#cb10-1623" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-1624"><a href="#cb10-1624" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb10-1625"><a href="#cb10-1625" aria-hidden="true" tabindex="-1"></a><span class="co"># calculate the embedding</span></span>
<span id="cb10-1626"><a href="#cb10-1626" aria-hidden="true" tabindex="-1"></a>gse_sum_exp_df_pca <span class="ot">&lt;-</span> <span class="fu">get_pca_coordinates</span>(gse_sum_exp_normalized, pca_fit) <span class="sc">%&gt;%</span></span>
<span id="cb10-1627"><a href="#cb10-1627" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span> </span>
<span id="cb10-1628"><a href="#cb10-1628" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">bind_cols</span>(</span>
<span id="cb10-1629"><a href="#cb10-1629" aria-hidden="true" tabindex="-1"></a>        ., </span>
<span id="cb10-1630"><a href="#cb10-1630" aria-hidden="true" tabindex="-1"></a>        <span class="fu">colData</span>(gse_sum_exp_normalized) <span class="sc">%&gt;%</span> </span>
<span id="cb10-1631"><a href="#cb10-1631" aria-hidden="true" tabindex="-1"></a>            data.frame <span class="sc">%&gt;%</span></span>
<span id="cb10-1632"><a href="#cb10-1632" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>dplyr<span class="sc">::</span><span class="fu">all_of</span>(<span class="fu">c</span>(<span class="st">"status"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb10-1633"><a href="#cb10-1633" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb10-1634"><a href="#cb10-1634" aria-hidden="true" tabindex="-1"></a>                <span class="at">erstatus_by_ihc_ch1 =</span> <span class="fu">ifelse</span>(</span>
<span id="cb10-1635"><a href="#cb10-1635" aria-hidden="true" tabindex="-1"></a>                    erstatus_by_ihc_ch1 <span class="sc">==</span> <span class="st">"positive"</span>,</span>
<span id="cb10-1636"><a href="#cb10-1636" aria-hidden="true" tabindex="-1"></a>                    <span class="st">"pos"</span>,</span>
<span id="cb10-1637"><a href="#cb10-1637" aria-hidden="true" tabindex="-1"></a>                    <span class="st">"neg"</span></span>
<span id="cb10-1638"><a href="#cb10-1638" aria-hidden="true" tabindex="-1"></a>                ),</span>
<span id="cb10-1639"><a href="#cb10-1639" aria-hidden="true" tabindex="-1"></a>                <span class="at">sample_name =</span> geo_accession,</span>
<span id="cb10-1640"><a href="#cb10-1640" aria-hidden="true" tabindex="-1"></a>                <span class="at">cohort =</span> <span class="st">"erp_ern"</span>,</span>
<span id="cb10-1641"><a href="#cb10-1641" aria-hidden="true" tabindex="-1"></a>                <span class="at">pam50 =</span> <span class="st">"not_available"</span>,</span>
<span id="cb10-1642"><a href="#cb10-1642" aria-hidden="true" tabindex="-1"></a>                <span class="at">er_status =</span> erstatus_by_ihc_ch1</span>
<span id="cb10-1643"><a href="#cb10-1643" aria-hidden="true" tabindex="-1"></a>            )</span>
<span id="cb10-1644"><a href="#cb10-1644" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span> </span>
<span id="cb10-1645"><a href="#cb10-1645" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(., df_pca)</span>
<span id="cb10-1646"><a href="#cb10-1646" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-1647"><a href="#cb10-1647" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1648"><a href="#cb10-1648" aria-hidden="true" tabindex="-1"></a>We now show the samples highlighted on top of EMBER space with </span>
<span id="cb10-1649"><a href="#cb10-1649" aria-hidden="true" tabindex="-1"></a>samples from TCGA and METABRIC.</span>
<span id="cb10-1650"><a href="#cb10-1650" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1651"><a href="#cb10-1651" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=11, fig.height=5}</span></span>
<span id="cb10-1652"><a href="#cb10-1652" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-illumina-nugen</span></span>
<span id="cb10-1653"><a href="#cb10-1653" aria-hidden="true" tabindex="-1"></a>gse_sum_sub <span class="ot">&lt;-</span> gse_sum_exp_df_pca <span class="sc">%&gt;%</span></span>
<span id="cb10-1654"><a href="#cb10-1654" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"erp_ern"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-1655"><a href="#cb10-1655" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">separate</span>(</span>
<span id="cb10-1656"><a href="#cb10-1656" aria-hidden="true" tabindex="-1"></a>        <span class="at">col =</span> title,</span>
<span id="cb10-1657"><a href="#cb10-1657" aria-hidden="true" tabindex="-1"></a>        <span class="at">into =</span> <span class="fu">c</span>(<span class="st">"null"</span>, <span class="st">"lib_prep"</span>, <span class="st">"sample_1"</span>, <span class="st">"sample_2"</span>),</span>
<span id="cb10-1658"><a href="#cb10-1658" aria-hidden="true" tabindex="-1"></a>        <span class="at">remove =</span> <span class="cn">FALSE</span>, </span>
<span id="cb10-1659"><a href="#cb10-1659" aria-hidden="true" tabindex="-1"></a>        <span class="at">sep =</span> <span class="st">"_"</span></span>
<span id="cb10-1660"><a href="#cb10-1660" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb10-1661"><a href="#cb10-1661" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb10-1662"><a href="#cb10-1662" aria-hidden="true" tabindex="-1"></a>        <span class="at">sample_legend =</span> <span class="fu">ifelse</span>(</span>
<span id="cb10-1663"><a href="#cb10-1663" aria-hidden="true" tabindex="-1"></a>            sample_1 <span class="sc">!=</span> <span class="st">"ER"</span>,</span>
<span id="cb10-1664"><a href="#cb10-1664" aria-hidden="true" tabindex="-1"></a>            <span class="fu">paste0</span>(</span>
<span id="cb10-1665"><a href="#cb10-1665" aria-hidden="true" tabindex="-1"></a>                <span class="st">"S0"</span>, </span>
<span id="cb10-1666"><a href="#cb10-1666" aria-hidden="true" tabindex="-1"></a>                <span class="fu">as.integer</span>(stringr<span class="sc">::</span><span class="fu">str_sub</span>(sample_1, <span class="at">start =</span> <span class="dv">2</span>, <span class="at">end =</span> <span class="dv">2</span>)) <span class="sc">+</span> <span class="dv">6</span></span>
<span id="cb10-1667"><a href="#cb10-1667" aria-hidden="true" tabindex="-1"></a>            ),</span>
<span id="cb10-1668"><a href="#cb10-1668" aria-hidden="true" tabindex="-1"></a>            sample_2</span>
<span id="cb10-1669"><a href="#cb10-1669" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb10-1670"><a href="#cb10-1670" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb10-1671"><a href="#cb10-1671" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1672"><a href="#cb10-1672" aria-hidden="true" tabindex="-1"></a>gse_sum_sub <span class="sc">%&gt;%</span></span>
<span id="cb10-1673"><a href="#cb10-1673" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(</span>
<span id="cb10-1674"><a href="#cb10-1674" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> PC3, <span class="at">y =</span> PC4, <span class="at">fill =</span> er_status</span>
<span id="cb10-1675"><a href="#cb10-1675" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">+</span> </span>
<span id="cb10-1676"><a href="#cb10-1676" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>(</span>
<span id="cb10-1677"><a href="#cb10-1677" aria-hidden="true" tabindex="-1"></a>        df_pca <span class="sc">%&gt;%</span></span>
<span id="cb10-1678"><a href="#cb10-1678" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"tcga"</span>, <span class="st">"metabric"</span>)),</span>
<span id="cb10-1679"><a href="#cb10-1679" aria-hidden="true" tabindex="-1"></a>        <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> PC3, <span class="at">y =</span> PC4, <span class="at">fill =</span> er_status, <span class="at">color =</span> er_status),</span>
<span id="cb10-1680"><a href="#cb10-1680" aria-hidden="true" tabindex="-1"></a>        <span class="at">alpha =</span> <span class="fl">0.3</span>,</span>
<span id="cb10-1681"><a href="#cb10-1681" aria-hidden="true" tabindex="-1"></a>        <span class="at">size =</span> <span class="dv">1</span>,</span>
<span id="cb10-1682"><a href="#cb10-1682" aria-hidden="true" tabindex="-1"></a>        <span class="at">pch =</span> <span class="dv">21</span></span>
<span id="cb10-1683"><a href="#cb10-1683" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb10-1684"><a href="#cb10-1684" aria-hidden="true" tabindex="-1"></a>        ggplot2<span class="sc">::</span><span class="fu">geom_point</span>(</span>
<span id="cb10-1685"><a href="#cb10-1685" aria-hidden="true" tabindex="-1"></a>        <span class="at">size =</span> <span class="dv">5</span>, <span class="at">alpha =</span> <span class="fl">0.9</span>, <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">pch =</span> <span class="dv">21</span></span>
<span id="cb10-1686"><a href="#cb10-1686" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb10-1687"><a href="#cb10-1687" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(</span>
<span id="cb10-1688"><a href="#cb10-1688" aria-hidden="true" tabindex="-1"></a>        <span class="sc">~</span>lib_prep,</span>
<span id="cb10-1689"><a href="#cb10-1689" aria-hidden="true" tabindex="-1"></a>        <span class="at">labeller =</span> <span class="fu">as_labeller</span>(<span class="fu">c</span>(</span>
<span id="cb10-1690"><a href="#cb10-1690" aria-hidden="true" tabindex="-1"></a>            <span class="at">ACC =</span> <span class="st">"Illumina-Access"</span>,</span>
<span id="cb10-1691"><a href="#cb10-1691" aria-hidden="true" tabindex="-1"></a>            <span class="at">OVA =</span> <span class="st">"Nugen-Ovation"</span></span>
<span id="cb10-1692"><a href="#cb10-1692" aria-hidden="true" tabindex="-1"></a>        ))</span>
<span id="cb10-1693"><a href="#cb10-1693" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb10-1694"><a href="#cb10-1694" aria-hidden="true" tabindex="-1"></a>    ggrepel<span class="sc">::</span><span class="fu">geom_label_repel</span>(</span>
<span id="cb10-1695"><a href="#cb10-1695" aria-hidden="true" tabindex="-1"></a>        <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">label =</span> sample_legend), </span>
<span id="cb10-1696"><a href="#cb10-1696" aria-hidden="true" tabindex="-1"></a>        <span class="at">fill =</span> <span class="st">"white"</span></span>
<span id="cb10-1697"><a href="#cb10-1697" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb10-1698"><a href="#cb10-1698" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_color_manual</span>(</span>
<span id="cb10-1699"><a href="#cb10-1699" aria-hidden="true" tabindex="-1"></a>        <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Negative"</span>, <span class="st">"Positive"</span>),</span>
<span id="cb10-1700"><a href="#cb10-1700" aria-hidden="true" tabindex="-1"></a>        <span class="at">values =</span> viridis<span class="sc">::</span><span class="fu">viridis</span>(<span class="dv">2</span>)</span>
<span id="cb10-1701"><a href="#cb10-1701" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb10-1702"><a href="#cb10-1702" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_fill_manual</span>(</span>
<span id="cb10-1703"><a href="#cb10-1703" aria-hidden="true" tabindex="-1"></a>        <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Negative"</span>, <span class="st">"Positive"</span>),</span>
<span id="cb10-1704"><a href="#cb10-1704" aria-hidden="true" tabindex="-1"></a>        <span class="at">values =</span> viridis<span class="sc">::</span><span class="fu">viridis</span>(<span class="dv">2</span>)</span>
<span id="cb10-1705"><a href="#cb10-1705" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb10-1706"><a href="#cb10-1706" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">color =</span> <span class="st">"ER IHC"</span>, <span class="at">fill =</span> <span class="st">"ER IHC"</span>) <span class="sc">+</span> </span>
<span id="cb10-1707"><a href="#cb10-1707" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>) <span class="sc">+</span></span>
<span id="cb10-1708"><a href="#cb10-1708" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>() <span class="sc">+</span></span>
<span id="cb10-1709"><a href="#cb10-1709" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>()</span>
<span id="cb10-1710"><a href="#cb10-1710" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-1711"><a href="#cb10-1711" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1712"><a href="#cb10-1712" aria-hidden="true" tabindex="-1"></a><span class="fu">## SCAN-B library protocol</span></span>
<span id="cb10-1713"><a href="#cb10-1713" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1714"><a href="#cb10-1714" aria-hidden="true" tabindex="-1"></a>SCAN-B has 3 different types of samples based on their </span>
<span id="cb10-1715"><a href="#cb10-1715" aria-hidden="true" tabindex="-1"></a>library protocol, namely: dUTP, NeoPrep and TruSeq.</span>
<span id="cb10-1716"><a href="#cb10-1716" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1717"><a href="#cb10-1717" aria-hidden="true" tabindex="-1"></a>Here we show that EMBER is capable of fully removing any</span>
<span id="cb10-1718"><a href="#cb10-1718" aria-hidden="true" tabindex="-1"></a>effects associated to library protocol.</span>
<span id="cb10-1719"><a href="#cb10-1719" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1720"><a href="#cb10-1720" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=5, fig.height=13}</span></span>
<span id="cb10-1721"><a href="#cb10-1721" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-scanb-libprep</span></span>
<span id="cb10-1722"><a href="#cb10-1722" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1723"><a href="#cb10-1723" aria-hidden="true" tabindex="-1"></a>sub_scanb <span class="ot">&lt;-</span> df_pca <span class="sc">%&gt;%</span> </span>
<span id="cb10-1724"><a href="#cb10-1724" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"scanb"</span>)</span>
<span id="cb10-1725"><a href="#cb10-1725" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1726"><a href="#cb10-1726" aria-hidden="true" tabindex="-1"></a>lisi_scores_scanb <span class="ot">&lt;-</span> lisi<span class="sc">::</span><span class="fu">compute_lisi</span>(</span>
<span id="cb10-1727"><a href="#cb10-1727" aria-hidden="true" tabindex="-1"></a>    sub_scanb[, <span class="fu">paste0</span>(<span class="st">"PC"</span>, <span class="dv">3</span><span class="sc">:</span><span class="dv">4</span>)],</span>
<span id="cb10-1728"><a href="#cb10-1728" aria-hidden="true" tabindex="-1"></a>    sub_scanb,</span>
<span id="cb10-1729"><a href="#cb10-1729" aria-hidden="true" tabindex="-1"></a>    <span class="fu">c</span>(<span class="st">"LibraryProtocol"</span>), </span>
<span id="cb10-1730"><a href="#cb10-1730" aria-hidden="true" tabindex="-1"></a>    <span class="at">perplexity =</span> <span class="dv">30</span></span>
<span id="cb10-1731"><a href="#cb10-1731" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-1732"><a href="#cb10-1732" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1733"><a href="#cb10-1733" aria-hidden="true" tabindex="-1"></a>nb_protocols <span class="ot">&lt;-</span> <span class="fu">table</span>(sub_scanb<span class="sc">$</span>LibraryProtocol)</span>
<span id="cb10-1734"><a href="#cb10-1734" aria-hidden="true" tabindex="-1"></a>ilisi_coefficient <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">/</span>(</span>
<span id="cb10-1735"><a href="#cb10-1735" aria-hidden="true" tabindex="-1"></a>    (nb_protocols[<span class="dv">1</span>]<span class="sc">/</span><span class="fu">sum</span>(nb_protocols))<span class="sc">^</span><span class="dv">2</span> <span class="sc">+</span></span>
<span id="cb10-1736"><a href="#cb10-1736" aria-hidden="true" tabindex="-1"></a>    (nb_protocols[<span class="dv">2</span>]<span class="sc">/</span><span class="fu">sum</span>(nb_protocols))<span class="sc">^</span><span class="dv">2</span> <span class="sc">+</span></span>
<span id="cb10-1737"><a href="#cb10-1737" aria-hidden="true" tabindex="-1"></a>    (nb_protocols[<span class="dv">3</span>]<span class="sc">/</span><span class="fu">sum</span>(nb_protocols))<span class="sc">^</span><span class="dv">2</span> </span>
<span id="cb10-1738"><a href="#cb10-1738" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-1739"><a href="#cb10-1739" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1740"><a href="#cb10-1740" aria-hidden="true" tabindex="-1"></a>expected_ilisi_vals <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">round</span>(ilisi_coefficient, <span class="at">digits =</span> <span class="dv">2</span>))</span>
<span id="cb10-1741"><a href="#cb10-1741" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1742"><a href="#cb10-1742" aria-hidden="true" tabindex="-1"></a>df_pca <span class="sc">%&gt;%</span> </span>
<span id="cb10-1743"><a href="#cb10-1743" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"scanb"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-1744"><a href="#cb10-1744" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(</span>
<span id="cb10-1745"><a href="#cb10-1745" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> PC3,</span>
<span id="cb10-1746"><a href="#cb10-1746" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> PC4,</span>
<span id="cb10-1747"><a href="#cb10-1747" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> LibraryProtocol</span>
<span id="cb10-1748"><a href="#cb10-1748" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">+</span> </span>
<span id="cb10-1749"><a href="#cb10-1749" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span> </span>
<span id="cb10-1750"><a href="#cb10-1750" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>LibraryProtocol, <span class="at">ncol =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb10-1751"><a href="#cb10-1751" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb10-1752"><a href="#cb10-1752" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="fu">paste0</span>(</span>
<span id="cb10-1753"><a href="#cb10-1753" aria-hidden="true" tabindex="-1"></a>            <span class="st">"LISI score: "</span>, </span>
<span id="cb10-1754"><a href="#cb10-1754" aria-hidden="true" tabindex="-1"></a>            <span class="fu">round</span>(<span class="fu">mean</span>(lisi_scores_scanb[, <span class="dv">1</span>]), <span class="at">digits =</span> <span class="dv">2</span>)),</span>
<span id="cb10-1755"><a href="#cb10-1755" aria-hidden="true" tabindex="-1"></a>        <span class="at">subtitle =</span> <span class="fu">paste0</span>(</span>
<span id="cb10-1756"><a href="#cb10-1756" aria-hidden="true" tabindex="-1"></a>            <span class="st">"Expected maximum LISI score: "</span>, <span class="fu">unname</span>(expected_ilisi_vals)</span>
<span id="cb10-1757"><a href="#cb10-1757" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb10-1758"><a href="#cb10-1758" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb10-1759"><a href="#cb10-1759" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">19</span>) <span class="sc">+</span> </span>
<span id="cb10-1760"><a href="#cb10-1760" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb10-1761"><a href="#cb10-1761" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-1762"><a href="#cb10-1762" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1763"><a href="#cb10-1763" aria-hidden="true" tabindex="-1"></a><span class="fu">## Scaling before doing PCA</span></span>
<span id="cb10-1764"><a href="#cb10-1764" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1765"><a href="#cb10-1765" aria-hidden="true" tabindex="-1"></a>When developing EMBER we did not center or scale the data, as the gene</span>
<span id="cb10-1766"><a href="#cb10-1766" aria-hidden="true" tabindex="-1"></a>expressions after normalization all had the same range. Here we compare</span>
<span id="cb10-1767"><a href="#cb10-1767" aria-hidden="true" tabindex="-1"></a>what happens when doing the centering followed by the scaling and then </span>
<span id="cb10-1768"><a href="#cb10-1768" aria-hidden="true" tabindex="-1"></a>applying PCA.</span>
<span id="cb10-1769"><a href="#cb10-1769" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1772"><a href="#cb10-1772" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-1773"><a href="#cb10-1773" aria-hidden="true" tabindex="-1"></a>training_set <span class="ot">&lt;-</span> <span class="fu">lapply</span>(</span>
<span id="cb10-1774"><a href="#cb10-1774" aria-hidden="true" tabindex="-1"></a>    datasets_normalized_og[<span class="fu">c</span>(<span class="st">"tcga"</span>, <span class="st">"metabric"</span>)], </span>
<span id="cb10-1775"><a href="#cb10-1775" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(sum_exp, i, genes_for_pca) </span>
<span id="cb10-1776"><a href="#cb10-1776" aria-hidden="true" tabindex="-1"></a>        <span class="fu">assay</span>(sum_exp[genes_for_pca, ], <span class="at">i =</span> i) <span class="sc">%&gt;%</span> </span>
<span id="cb10-1777"><a href="#cb10-1777" aria-hidden="true" tabindex="-1"></a>        <span class="fu">data.frame</span>(<span class="at">check.names =</span> <span class="cn">FALSE</span>), </span>
<span id="cb10-1778"><a href="#cb10-1778" aria-hidden="true" tabindex="-1"></a>    <span class="at">i =</span> <span class="st">"avg_ranking"</span>,</span>
<span id="cb10-1779"><a href="#cb10-1779" aria-hidden="true" tabindex="-1"></a>    <span class="at">genes_for_pca =</span> <span class="fu">rownames</span>(pca_fit<span class="sc">$</span>loadings)</span>
<span id="cb10-1780"><a href="#cb10-1780" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-1781"><a href="#cb10-1781" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">bind_cols</span>() <span class="sc">%&gt;%</span></span>
<span id="cb10-1782"><a href="#cb10-1782" aria-hidden="true" tabindex="-1"></a>    .[, <span class="fu">rownames</span>(pca_fit<span class="sc">$</span>rotated)] <span class="sc">%&gt;%</span></span>
<span id="cb10-1783"><a href="#cb10-1783" aria-hidden="true" tabindex="-1"></a>    as.matrix</span>
<span id="cb10-1784"><a href="#cb10-1784" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1785"><a href="#cb10-1785" aria-hidden="true" tabindex="-1"></a>pca_fit_scaled <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">pca</span>(</span>
<span id="cb10-1786"><a href="#cb10-1786" aria-hidden="true" tabindex="-1"></a>    training_set,</span>
<span id="cb10-1787"><a href="#cb10-1787" aria-hidden="true" tabindex="-1"></a>    <span class="at">metadata =</span> pca_fit<span class="sc">$</span>metadata,</span>
<span id="cb10-1788"><a href="#cb10-1788" aria-hidden="true" tabindex="-1"></a>    <span class="at">center =</span> <span class="cn">TRUE</span>,</span>
<span id="cb10-1789"><a href="#cb10-1789" aria-hidden="true" tabindex="-1"></a>    <span class="at">scale =</span> <span class="cn">TRUE</span></span>
<span id="cb10-1790"><a href="#cb10-1790" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-1791"><a href="#cb10-1791" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1792"><a href="#cb10-1792" aria-hidden="true" tabindex="-1"></a>prcomp_training <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(<span class="fu">t</span>(training_set), <span class="at">center =</span> <span class="cn">TRUE</span>, <span class="at">scale =</span> <span class="cn">TRUE</span>)</span>
<span id="cb10-1793"><a href="#cb10-1793" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1794"><a href="#cb10-1794" aria-hidden="true" tabindex="-1"></a>df_pca_loadings_centers <span class="ot">&lt;-</span> pca_fit<span class="sc">$</span>loadings <span class="sc">%&gt;%</span></span>
<span id="cb10-1795"><a href="#cb10-1795" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span></span>
<span id="cb10-1796"><a href="#cb10-1796" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">bind_cols</span>(</span>
<span id="cb10-1797"><a href="#cb10-1797" aria-hidden="true" tabindex="-1"></a>        .,</span>
<span id="cb10-1798"><a href="#cb10-1798" aria-hidden="true" tabindex="-1"></a>        <span class="fu">data.frame</span>(</span>
<span id="cb10-1799"><a href="#cb10-1799" aria-hidden="true" tabindex="-1"></a>            <span class="at">scale_prcomp =</span> prcomp_training<span class="sc">$</span>scale,</span>
<span id="cb10-1800"><a href="#cb10-1800" aria-hidden="true" tabindex="-1"></a>            <span class="at">center_prcomp =</span> prcomp_training<span class="sc">$</span>center</span>
<span id="cb10-1801"><a href="#cb10-1801" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb10-1802"><a href="#cb10-1802" aria-hidden="true" tabindex="-1"></a>    ) </span>
<span id="cb10-1803"><a href="#cb10-1803" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1804"><a href="#cb10-1804" aria-hidden="true" tabindex="-1"></a>df_scaled_ember <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">bind_cols</span>(</span>
<span id="cb10-1805"><a href="#cb10-1805" aria-hidden="true" tabindex="-1"></a>    pca_fit_scaled<span class="sc">$</span>rotated <span class="sc">%&gt;%</span> <span class="st">`</span><span class="at">colnames&lt;-</span><span class="st">`</span>(<span class="fu">paste0</span>(<span class="st">"scaled_"</span>, <span class="fu">colnames</span>(.))),</span>
<span id="cb10-1806"><a href="#cb10-1806" aria-hidden="true" tabindex="-1"></a>    pca_fit<span class="sc">$</span>rotated <span class="sc">%&gt;%</span> <span class="st">`</span><span class="at">colnames&lt;-</span><span class="st">`</span>(<span class="fu">paste0</span>(<span class="st">"EMBER_"</span>, <span class="fu">colnames</span>(.)))</span>
<span id="cb10-1807"><a href="#cb10-1807" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb10-1808"><a href="#cb10-1808" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">sample_name =</span> pca_fit_scaled<span class="sc">$</span>rotated <span class="sc">%&gt;%</span> rownames) <span class="sc">%&gt;%</span></span>
<span id="cb10-1809"><a href="#cb10-1809" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">inner_join</span>(</span>
<span id="cb10-1810"><a href="#cb10-1810" aria-hidden="true" tabindex="-1"></a>        .,</span>
<span id="cb10-1811"><a href="#cb10-1811" aria-hidden="true" tabindex="-1"></a>        pca_fit<span class="sc">$</span>metadata,</span>
<span id="cb10-1812"><a href="#cb10-1812" aria-hidden="true" tabindex="-1"></a>        <span class="at">by =</span> <span class="st">"sample_name"</span></span>
<span id="cb10-1813"><a href="#cb10-1813" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb10-1814"><a href="#cb10-1814" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1815"><a href="#cb10-1815" aria-hidden="true" tabindex="-1"></a>p1_center <span class="ot">&lt;-</span> df_pca_loadings_centers <span class="sc">%&gt;%</span></span>
<span id="cb10-1816"><a href="#cb10-1816" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(</span>
<span id="cb10-1817"><a href="#cb10-1817" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> PC1,</span>
<span id="cb10-1818"><a href="#cb10-1818" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> center_prcomp</span>
<span id="cb10-1819"><a href="#cb10-1819" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">+</span> </span>
<span id="cb10-1820"><a href="#cb10-1820" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb10-1821"><a href="#cb10-1821" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb10-1822"><a href="#cb10-1822" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"Loadings of EMBER PC1"</span>,</span>
<span id="cb10-1823"><a href="#cb10-1823" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"Average of genes</span><span class="sc">\n</span><span class="st">in the normalized data"</span></span>
<span id="cb10-1824"><a href="#cb10-1824" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb10-1825"><a href="#cb10-1825" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>) <span class="sc">+</span></span>
<span id="cb10-1826"><a href="#cb10-1826" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>() <span class="sc">+</span> </span>
<span id="cb10-1827"><a href="#cb10-1827" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>()</span>
<span id="cb10-1828"><a href="#cb10-1828" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1829"><a href="#cb10-1829" aria-hidden="true" tabindex="-1"></a>p2_scale <span class="ot">&lt;-</span> df_pca_loadings_centers <span class="sc">%&gt;%</span></span>
<span id="cb10-1830"><a href="#cb10-1830" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(</span>
<span id="cb10-1831"><a href="#cb10-1831" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> PC2,</span>
<span id="cb10-1832"><a href="#cb10-1832" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> scale_prcomp</span>
<span id="cb10-1833"><a href="#cb10-1833" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">+</span> </span>
<span id="cb10-1834"><a href="#cb10-1834" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb10-1835"><a href="#cb10-1835" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb10-1836"><a href="#cb10-1836" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"Loadings of EMBER PC2"</span>,</span>
<span id="cb10-1837"><a href="#cb10-1837" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"SD of genes</span><span class="sc">\n</span><span class="st">after centering"</span></span>
<span id="cb10-1838"><a href="#cb10-1838" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb10-1839"><a href="#cb10-1839" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>) <span class="sc">+</span> </span>
<span id="cb10-1840"><a href="#cb10-1840" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>() <span class="sc">+</span> </span>
<span id="cb10-1841"><a href="#cb10-1841" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>()</span>
<span id="cb10-1842"><a href="#cb10-1842" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1843"><a href="#cb10-1843" aria-hidden="true" tabindex="-1"></a>p3_ember_pcs_scaled_pc <span class="ot">&lt;-</span> df_scaled_ember <span class="sc">%&gt;%</span></span>
<span id="cb10-1844"><a href="#cb10-1844" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(</span>
<span id="cb10-1845"><a href="#cb10-1845" aria-hidden="true" tabindex="-1"></a>        <span class="at">cols =</span> <span class="fu">c</span>(EMBER_PC1, EMBER_PC2),</span>
<span id="cb10-1846"><a href="#cb10-1846" aria-hidden="true" tabindex="-1"></a>        <span class="at">values_to =</span> <span class="st">"PC"</span>, </span>
<span id="cb10-1847"><a href="#cb10-1847" aria-hidden="true" tabindex="-1"></a>        <span class="at">names_to =</span> <span class="st">"ember_pcs"</span></span>
<span id="cb10-1848"><a href="#cb10-1848" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb10-1849"><a href="#cb10-1849" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb10-1850"><a href="#cb10-1850" aria-hidden="true" tabindex="-1"></a>        <span class="at">cohort =</span> <span class="fu">toupper</span>(cohort),</span>
<span id="cb10-1851"><a href="#cb10-1851" aria-hidden="true" tabindex="-1"></a>        <span class="at">ember_pcs =</span> stringr<span class="sc">::</span><span class="fu">str_replace</span>(ember_pcs, <span class="st">"_"</span>, <span class="st">" "</span>)</span>
<span id="cb10-1852"><a href="#cb10-1852" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb10-1853"><a href="#cb10-1853" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(</span>
<span id="cb10-1854"><a href="#cb10-1854" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> scaled_PC1,</span>
<span id="cb10-1855"><a href="#cb10-1855" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> PC,</span>
<span id="cb10-1856"><a href="#cb10-1856" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> cohort</span>
<span id="cb10-1857"><a href="#cb10-1857" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">+</span> </span>
<span id="cb10-1858"><a href="#cb10-1858" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span> </span>
<span id="cb10-1859"><a href="#cb10-1859" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">color =</span> <span class="st">"Cohort"</span>) <span class="sc">+</span></span>
<span id="cb10-1860"><a href="#cb10-1860" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>ember_pcs) <span class="sc">+</span> </span>
<span id="cb10-1861"><a href="#cb10-1861" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Scaled PC1"</span>) <span class="sc">+</span> </span>
<span id="cb10-1862"><a href="#cb10-1862" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>) <span class="sc">+</span></span>
<span id="cb10-1863"><a href="#cb10-1863" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>() <span class="sc">+</span> </span>
<span id="cb10-1864"><a href="#cb10-1864" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>()</span>
<span id="cb10-1865"><a href="#cb10-1865" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1866"><a href="#cb10-1866" aria-hidden="true" tabindex="-1"></a>p4_ember_pc3_scaled_pc2 <span class="ot">&lt;-</span> df_scaled_ember <span class="sc">%&gt;%</span></span>
<span id="cb10-1867"><a href="#cb10-1867" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb10-1868"><a href="#cb10-1868" aria-hidden="true" tabindex="-1"></a>        <span class="at">cohort =</span> <span class="fu">toupper</span>(cohort)</span>
<span id="cb10-1869"><a href="#cb10-1869" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb10-1870"><a href="#cb10-1870" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(</span>
<span id="cb10-1871"><a href="#cb10-1871" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> scaled_PC2,</span>
<span id="cb10-1872"><a href="#cb10-1872" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> EMBER_PC3, </span>
<span id="cb10-1873"><a href="#cb10-1873" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> cohort</span>
<span id="cb10-1874"><a href="#cb10-1874" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">+</span> </span>
<span id="cb10-1875"><a href="#cb10-1875" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span> </span>
<span id="cb10-1876"><a href="#cb10-1876" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">formula =</span> <span class="st">"y~x"</span>) <span class="sc">+</span> </span>
<span id="cb10-1877"><a href="#cb10-1877" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">color =</span> <span class="st">"Cohort"</span>) <span class="sc">+</span></span>
<span id="cb10-1878"><a href="#cb10-1878" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Scaled PC2"</span>, <span class="at">y =</span> <span class="st">"EMBER PC3"</span>) <span class="sc">+</span> </span>
<span id="cb10-1879"><a href="#cb10-1879" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>) <span class="sc">+</span></span>
<span id="cb10-1880"><a href="#cb10-1880" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>cohort)<span class="sc">+</span></span>
<span id="cb10-1881"><a href="#cb10-1881" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>() <span class="sc">+</span> </span>
<span id="cb10-1882"><a href="#cb10-1882" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span> </span>
<span id="cb10-1883"><a href="#cb10-1883" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb10-1884"><a href="#cb10-1884" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1885"><a href="#cb10-1885" aria-hidden="true" tabindex="-1"></a>p4_ember_pc4_scaled_pc3 <span class="ot">&lt;-</span> df_scaled_ember <span class="sc">%&gt;%</span></span>
<span id="cb10-1886"><a href="#cb10-1886" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb10-1887"><a href="#cb10-1887" aria-hidden="true" tabindex="-1"></a>        <span class="at">cohort =</span> <span class="fu">toupper</span>(cohort)</span>
<span id="cb10-1888"><a href="#cb10-1888" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb10-1889"><a href="#cb10-1889" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(</span>
<span id="cb10-1890"><a href="#cb10-1890" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> scaled_PC3,</span>
<span id="cb10-1891"><a href="#cb10-1891" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> EMBER_PC4, </span>
<span id="cb10-1892"><a href="#cb10-1892" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> cohort</span>
<span id="cb10-1893"><a href="#cb10-1893" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">+</span> </span>
<span id="cb10-1894"><a href="#cb10-1894" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span> </span>
<span id="cb10-1895"><a href="#cb10-1895" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">formula =</span> <span class="st">"y~x"</span>) <span class="sc">+</span> </span>
<span id="cb10-1896"><a href="#cb10-1896" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">color =</span> <span class="st">"Cohort"</span>) <span class="sc">+</span></span>
<span id="cb10-1897"><a href="#cb10-1897" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Scaled PC3"</span>, <span class="at">y =</span> <span class="st">"EMBER PC4"</span>) <span class="sc">+</span> </span>
<span id="cb10-1898"><a href="#cb10-1898" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>) <span class="sc">+</span></span>
<span id="cb10-1899"><a href="#cb10-1899" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>cohort)<span class="sc">+</span></span>
<span id="cb10-1900"><a href="#cb10-1900" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>() <span class="sc">+</span> </span>
<span id="cb10-1901"><a href="#cb10-1901" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span> </span>
<span id="cb10-1902"><a href="#cb10-1902" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb10-1903"><a href="#cb10-1903" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-1904"><a href="#cb10-1904" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1905"><a href="#cb10-1905" aria-hidden="true" tabindex="-1"></a>To understand the effect of centering and scaling </span>
<span id="cb10-1906"><a href="#cb10-1906" aria-hidden="true" tabindex="-1"></a>the data prior to applying PCA, we compared the loadings </span>
<span id="cb10-1907"><a href="#cb10-1907" aria-hidden="true" tabindex="-1"></a>of PC1 from EMBER to the average of the genes in the</span>
<span id="cb10-1908"><a href="#cb10-1908" aria-hidden="true" tabindex="-1"></a>normalized data (a). The correlation between these two</span>
<span id="cb10-1909"><a href="#cb10-1909" aria-hidden="true" tabindex="-1"></a>variables is -1, showing that PC1 corresponds to</span>
<span id="cb10-1910"><a href="#cb10-1910" aria-hidden="true" tabindex="-1"></a>centering. When comparing the loadings with PC2 and </span>
<span id="cb10-1911"><a href="#cb10-1911" aria-hidden="true" tabindex="-1"></a>the standard deviation of genes after centering, the</span>
<span id="cb10-1912"><a href="#cb10-1912" aria-hidden="true" tabindex="-1"></a>higher loadings in absolute value are correlated with</span>
<span id="cb10-1913"><a href="#cb10-1913" aria-hidden="true" tabindex="-1"></a>higher standard deviation (b). </span>
<span id="cb10-1914"><a href="#cb10-1914" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1915"><a href="#cb10-1915" aria-hidden="true" tabindex="-1"></a>Next, we show that PC1 and PC2 are correlated with</span>
<span id="cb10-1916"><a href="#cb10-1916" aria-hidden="true" tabindex="-1"></a>PC1 obtained after centering and scaling the data</span>
<span id="cb10-1917"><a href="#cb10-1917" aria-hidden="true" tabindex="-1"></a>(c), showing that EMBER’s PC1 and PC2 is</span>
<span id="cb10-1918"><a href="#cb10-1918" aria-hidden="true" tabindex="-1"></a>automatically centering and scaling the data </span>
<span id="cb10-1919"><a href="#cb10-1919" aria-hidden="true" tabindex="-1"></a>at the same time removing batch effects. </span>
<span id="cb10-1920"><a href="#cb10-1920" aria-hidden="true" tabindex="-1"></a>In order to confirm that no biological information </span>
<span id="cb10-1921"><a href="#cb10-1921" aria-hidden="true" tabindex="-1"></a>is lost when performing PCA on the </span>
<span id="cb10-1922"><a href="#cb10-1922" aria-hidden="true" tabindex="-1"></a>uncentered and unscaled data, we compared EMBER’s</span>
<span id="cb10-1923"><a href="#cb10-1923" aria-hidden="true" tabindex="-1"></a>PC3 and PC4 with PC2 and PC3 from the </span>
<span id="cb10-1924"><a href="#cb10-1924" aria-hidden="true" tabindex="-1"></a>scaled data respectively (d and e). </span>
<span id="cb10-1925"><a href="#cb10-1925" aria-hidden="true" tabindex="-1"></a>The correlation is 1 in both cases for </span>
<span id="cb10-1926"><a href="#cb10-1926" aria-hidden="true" tabindex="-1"></a>both cohorts: METABRIC and TCGA. We show that</span>
<span id="cb10-1927"><a href="#cb10-1927" aria-hidden="true" tabindex="-1"></a>there is no need to center and scale prior to</span>
<span id="cb10-1928"><a href="#cb10-1928" aria-hidden="true" tabindex="-1"></a>applying PCA in EMBER’s context.</span>
<span id="cb10-1929"><a href="#cb10-1929" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1930"><a href="#cb10-1930" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=15, fig.height=10}</span></span>
<span id="cb10-1931"><a href="#cb10-1931" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-pca-center-scaled</span></span>
<span id="cb10-1932"><a href="#cb10-1932" aria-hidden="true" tabindex="-1"></a>plots <span class="ot">&lt;-</span> cowplot<span class="sc">::</span><span class="fu">align_plots</span>(</span>
<span id="cb10-1933"><a href="#cb10-1933" aria-hidden="true" tabindex="-1"></a>    p3_ember_pcs_scaled_pc, </span>
<span id="cb10-1934"><a href="#cb10-1934" aria-hidden="true" tabindex="-1"></a>    p1_center, </span>
<span id="cb10-1935"><a href="#cb10-1935" aria-hidden="true" tabindex="-1"></a>    <span class="at">align =</span> <span class="st">'v'</span>,</span>
<span id="cb10-1936"><a href="#cb10-1936" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis =</span> <span class="st">'l'</span></span>
<span id="cb10-1937"><a href="#cb10-1937" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-1938"><a href="#cb10-1938" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1939"><a href="#cb10-1939" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(</span>
<span id="cb10-1940"><a href="#cb10-1940" aria-hidden="true" tabindex="-1"></a>    cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(</span>
<span id="cb10-1941"><a href="#cb10-1941" aria-hidden="true" tabindex="-1"></a>        p1_center,</span>
<span id="cb10-1942"><a href="#cb10-1942" aria-hidden="true" tabindex="-1"></a>        p2_scale,</span>
<span id="cb10-1943"><a href="#cb10-1943" aria-hidden="true" tabindex="-1"></a>        <span class="at">nrow =</span> <span class="dv">2</span>,</span>
<span id="cb10-1944"><a href="#cb10-1944" aria-hidden="true" tabindex="-1"></a>        <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"a"</span>, <span class="st">"b"</span>),</span>
<span id="cb10-1945"><a href="#cb10-1945" aria-hidden="true" tabindex="-1"></a>        <span class="at">label_size =</span> <span class="dv">20</span></span>
<span id="cb10-1946"><a href="#cb10-1946" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb10-1947"><a href="#cb10-1947" aria-hidden="true" tabindex="-1"></a>    cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(</span>
<span id="cb10-1948"><a href="#cb10-1948" aria-hidden="true" tabindex="-1"></a>        p3_ember_pcs_scaled_pc,</span>
<span id="cb10-1949"><a href="#cb10-1949" aria-hidden="true" tabindex="-1"></a>        p4_ember_pc3_scaled_pc2,</span>
<span id="cb10-1950"><a href="#cb10-1950" aria-hidden="true" tabindex="-1"></a>        p4_ember_pc4_scaled_pc3,</span>
<span id="cb10-1951"><a href="#cb10-1951" aria-hidden="true" tabindex="-1"></a>        <span class="at">nrow =</span> <span class="dv">3</span>,</span>
<span id="cb10-1952"><a href="#cb10-1952" aria-hidden="true" tabindex="-1"></a>        <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"c"</span>, <span class="st">"d"</span>, <span class="st">"e"</span>),</span>
<span id="cb10-1953"><a href="#cb10-1953" aria-hidden="true" tabindex="-1"></a>        <span class="at">label_size =</span> <span class="dv">20</span></span>
<span id="cb10-1954"><a href="#cb10-1954" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb10-1955"><a href="#cb10-1955" aria-hidden="true" tabindex="-1"></a>    <span class="at">ncol =</span> <span class="dv">2</span>, </span>
<span id="cb10-1956"><a href="#cb10-1956" aria-hidden="true" tabindex="-1"></a>    <span class="at">label_size =</span> <span class="dv">20</span></span>
<span id="cb10-1957"><a href="#cb10-1957" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-1958"><a href="#cb10-1958" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-1959"><a href="#cb10-1959" aria-hidden="true" tabindex="-1"></a></span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
      &nbsp;
    </div>   
    <div class="nav-footer-center">
      &nbsp;
    </div>
    <div class="nav-footer-right">
      <ul class="footer-items list-unstyled">
    <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/chronchi">
      <i class="bi bi-github" role="img">
</i> 
    </a>
  </li>  
    <li class="nav-item compact">
    <a class="nav-link" href="https://twitter.com/chronchi">
      <i class="bi bi-twitter" role="img">
</i> 
    </a>
  </li>  
</ul>
    </div>
  </div>
</footer>



</body></html>