<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.554">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>EMBER creates a unified space for independent breast cancer transcriptomic datasets enabling precision oncology. - 5&nbsp; Risk score model</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../chapters/expanding_ember.html" rel="next">
<link href="../chapters/scoring.html" rel="prev">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script src="../site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="../site_libs/lightable-0.0.1/lightable.css" rel="stylesheet">

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../chapters/risk_score.html"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Risk score model</span></a></li></ol></nav>
        <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../">EMBER creates a unified space for independent breast cancer transcriptomic datasets enabling precision oncology.</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/chronchi/molecular_landscape" title="Source Code" class="quarto-navigation-tool px-1" aria-label="Source Code"><i class="bi bi-github"></i></a>
    <a href="https://twitter.com/intent/tweet?url=|url|" title="Twitter" class="quarto-navigation-tool px-1" aria-label="Twitter"><i class="bi bi-twitter"></i></a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Preface</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/intro.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Introduction</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/surv_analysis_estrogen.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Estrogen receptor status is continuous, not dichotomous</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/pca_merging.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">A new framework for personalized medicine: integrating</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/validation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Validating the molecular landscape</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/scoring.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Pathway activity in a personalized context</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/risk_score.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Risk score model</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/expanding_ember.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Expanding embedding and removing batch effects</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/trying.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Tests and tries</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/revision.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Revision</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/discussion.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Discussion</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/references.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">References</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#generating-the-risk-score" id="toc-generating-the-risk-score" class="nav-link active" data-scroll-target="#generating-the-risk-score"><span class="header-section-number">5.1</span> Generating the risk score</a></li>
  <li><a href="#validating-the-risk-score-on-scanb" id="toc-validating-the-risk-score-on-scanb" class="nav-link" data-scroll-target="#validating-the-risk-score-on-scanb"><span class="header-section-number">5.2</span> Validating the risk score on SCANB</a></li>
  <li><a href="#standardized-survival-curves-for-scanb" id="toc-standardized-survival-curves-for-scanb" class="nav-link" data-scroll-target="#standardized-survival-curves-for-scanb"><span class="header-section-number">5.3</span> Standardized survival curves for SCANB</a></li>
  <li><a href="#risk-score-and-poetic-trial" id="toc-risk-score-and-poetic-trial" class="nav-link" data-scroll-target="#risk-score-and-poetic-trial"><span class="header-section-number">5.4</span> Risk score and POETIC trial</a></li>
  <li><a href="#risk-score-vs-pathways" id="toc-risk-score-vs-pathways" class="nav-link" data-scroll-target="#risk-score-vs-pathways"><span class="header-section-number">5.5</span> Risk score vs pathways</a></li>
  <li><a href="#abim-and-ncounter-data" id="toc-abim-and-ncounter-data" class="nav-link" data-scroll-target="#abim-and-ncounter-data"><span class="header-section-number">5.6</span> ABiM and nCounter data</a></li>
  </ul>
<div class="toc-actions"><ul><li><a href="https://github.com/chronchi/molecular_landscape/issues/new" class="toc-action"><i class="bi bi-github"></i>Report an issue</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Risk score model</span></h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<p>Several risk signatures were developed over the years using transcriptomic datasets. Some examples were given in the previous chapters, such as OncotypeDX, Prosigna and EndoPredict. They were all derived in a similar manner, subsetting genes related to usually estrogen and proliferation signaling and then using recurrence free survival data to calculate the risk scores.</p>
<p>In this chapter we introduce a new risk score that does not subset genes and uses the information of the molecular landscape. The concept is simple, we will use the third and fourth components as covariates in the survival modeling along with tumor size and node status. This way we avoid subsetting and selecting specific genes and we use all the information the molecular landscape is providing us. Also it avoids any kind of binning for patients based on molecular subtype, thus not being necessary calculating correlations with pre-defined molecular subtypes.</p>
<p>Later in the chapter we compare the risk score with the ROR binary categories: high and low/intermediate risk. We also compare the risk score with the ROR continuous value obtained by the SCANB team using nearest centroid classification (NCN). We use samples from patients that were treated with endocrine therapy only for the training of the risk score. Even though the ROR was trained on a cohort of patients that received no treatment, we can still compare the signatures. <span class="citation" data-cites="Buus2021">Buus et al. (<a href="references.html#ref-Buus2021" role="doc-biblioref">2021</a>)</span> showed that the EndoPredict signature is highly correlated to ROR (<span class="math inline">\(\rho = 0.68\)</span>), therefore we can to some extent take conclusions based on the ROR signature.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>* The library is already synchronized with the lockfile.</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>    ../../results/plots/risk_score ../../results/rds_files/risk_score 
                             FALSE                              FALSE 
   ../../results/tables/risk_score 
                             FALSE </code></pre>
</div>
</div>
<p>We develop a risk score for early stage breast cancer patients that received only endocrine therapy (ET). The idea is similar to what was done with the EndoPredict signatures, but here we extend it by using the molecular landscape information. This provides a full framework where we can analyse the risk of recurrence even for patients that have high ER+ percentage.</p>
<section id="generating-the-risk-score" class="level2" data-number="5.1">
<h2 data-number="5.1" class="anchored" data-anchor-id="generating-the-risk-score"><span class="header-section-number">5.1</span> Generating the risk score</h2>
<p>We focus only on the endocrine treated patients. For the training part of the algorithm we use a subset of patients from the METABRIC cohort. We then test the risk scores in the test set and we further validate the score on SCANB.</p>
<p>We will generate two risk scores: one including only the principal components and another one including the clinical features of the patients when valiable, namely tumor size, node status and age (either having one or more lymph node with breast cancer cells or none).</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>Call:
survival::coxph(formula = as.formula(paste0("Surv(time = rfs_months, event = rfs_status) ~ ", 
    paste(c(pcs_to_use, clin_vars), collapse = "+"))), data = training_set)

  n= 556, number of events= 216 

                    coef exp(coef)  se(coef)      z Pr(&gt;|z|)    
PC3            -0.091540  0.912525  0.027255 -3.359 0.000783 ***
PC4             0.080512  1.083842  0.022588  3.564 0.000365 ***
tumor_size      0.021809  1.022049  0.004292  5.082 3.74e-07 ***
node_statuspos  0.582681  1.790834  0.147602  3.948 7.89e-05 ***
age             0.003150  1.003155  0.007030  0.448 0.654061    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

               exp(coef) exp(-coef) lower .95 upper .95
PC3               0.9125     1.0959    0.8651    0.9626
PC4               1.0838     0.9226    1.0369    1.1329
tumor_size        1.0220     0.9784    1.0135    1.0307
node_statuspos    1.7908     0.5584    1.3410    2.3916
age               1.0032     0.9969    0.9894    1.0171

Concordance= 0.676  (se = 0.019 )
Likelihood ratio test= 66.18  on 5 df,   p=6e-13
Wald test            = 71.92  on 5 df,   p=4e-14
Score (logrank) test = 72.75  on 5 df,   p=3e-14</code></pre>
</div>
</div>
<p>We see that all coefficients have less variability than age. Now we calculate only the risk score based on the components.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>Call:
survival::coxph(formula = as.formula(paste0("Surv(time = rfs_months, event = rfs_status) ~ ", 
    paste(c(pcs_to_use), collapse = "+"))), data = training_set)

  n= 556, number of events= 216 

        coef exp(coef) se(coef)      z Pr(&gt;|z|)    
PC3 -0.08657   0.91707  0.02699 -3.207  0.00134 ** 
PC4  0.09068   1.09492  0.02216  4.091 4.29e-05 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

    exp(coef) exp(-coef) lower .95 upper .95
PC3    0.9171     1.0904    0.8698    0.9669
PC4    1.0949     0.9133    1.0484    1.1435

Concordance= 0.626  (se = 0.019 )
Likelihood ratio test= 22.5  on 2 df,   p=1e-05
Wald test            = 22.49  on 2 df,   p=1e-05
Score (logrank) test = 22.78  on 2 df,   p=1e-05</code></pre>
</div>
</div>
<p>We see that the coefficients are very similar to what was found before, it didn’t change much. Also when we include the principal components on the top of the clinical variables the log likelihood increases, which is a good indication:</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>Analysis of Deviance Table
 Cox model: response is  Surv(time = rfs_months, event = rfs_status)
 Model 1: ~ tumor_size + node_status + age
 Model 2: ~ PC3 + PC4 + tumor_size + node_status + age
   loglik Chisq Df Pr(&gt;|Chi|)    
1 -1231.1                        
2 -1221.2 19.88  2  4.821e-05 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
</div>
<p>Now we use the test set to calculate the score.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>Call:
survival::coxph(formula = Surv(time = rfs_months, event = rfs_status) ~ 
    risk_score, data = test_set)

  n= 371, number of events= 140 

             coef exp(coef) se(coef)     z Pr(&gt;|z|)    
risk_score 0.8632    2.3707   0.1583 5.453 4.96e-08 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

           exp(coef) exp(-coef) lower .95 upper .95
risk_score     2.371     0.4218     1.738     3.233

Concordance= 0.655  (se = 0.025 )
Likelihood ratio test= 28.56  on 1 df,   p=9e-08
Wald test            = 29.73  on 1 df,   p=5e-08
Score (logrank) test = 29.78  on 1 df,   p=5e-08</code></pre>
</div>
</div>
<p>We see that the higher the risk score the worse it is for the patients treated only with endocrine therapy.</p>
</section>
<section id="validating-the-risk-score-on-scanb" class="level2" data-number="5.2">
<h2 data-number="5.2" class="anchored" data-anchor-id="validating-the-risk-score-on-scanb"><span class="header-section-number">5.2</span> Validating the risk score on SCANB</h2>
<p>We can now validate the risk score on the SCANB dataset.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>Call:
survival::coxph(formula = as.formula(paste0("Surv(time = rfs_months, event = rfs_status) ~ ", 
    paste(c("risk_score"), collapse = "+"))), data = endo_only_scanb)

  n= 2442, number of events= 175 

             coef exp(coef) se(coef)     z Pr(&gt;|z|)    
risk_score 1.1248    3.0795   0.1138 9.885   &lt;2e-16 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

           exp(coef) exp(-coef) lower .95 upper .95
risk_score     3.079     0.3247     2.464     3.849

Concordance= 0.696  (se = 0.022 )
Likelihood ratio test= 78.2  on 1 df,   p=&lt;2e-16
Wald test            = 97.71  on 1 df,   p=&lt;2e-16
Score (logrank) test = 96.79  on 1 df,   p=&lt;2e-16</code></pre>
</div>
</div>
<p>Again the risk score is prognostic with a high coefficient, meaning that the higher the score the worse it is for the patient.</p>
<p>We now visualize a bit the risk scores developed here in terms of the molecular landscape. <a href="#fig-risk-endo-pam50" class="quarto-xref">Figure&nbsp;<span>5.1</span></a></p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-risk-endo-pam50" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-risk-endo-pam50-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="risk_score_files/figure-html/fig-risk-endo-pam50-1.png" class="img-fluid figure-img" width="672">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-risk-endo-pam50-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;5.1: Molecular landscape of SCANB patients treated with only endocrine therapy and that have ER+ breast cancer. Only patients with risk score less than 2 are plotted here. There are only 26 patients out of 2829 with risk score higher than 2 that would skew the color scale.
</figcaption>
</figure>
</div>
</div>
</div>
<p>We now investigate the risk score among patients that have high ER IHC percentage as well. We have seen that these patients have different outcomes based on the molecular estrogen signaling. We now investigate the effect of the risk score.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>Call:
survival::coxph(formula = as.formula(paste0("Surv(time = rfs_months, event = rfs_status) ~ ", 
    paste(c("risk_score"), collapse = "+"))), data = endo_only_scanb %&gt;% 
    dplyr::filter(ER.pct &gt;= 90 &amp; risk_score &lt; 2))

  n= 1832, number of events= 126 

             coef exp(coef) se(coef)     z Pr(&gt;|z|)    
risk_score 1.1989    3.3165   0.1652 7.257 3.96e-13 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

           exp(coef) exp(-coef) lower .95 upper .95
risk_score     3.317     0.3015     2.399     4.585

Concordance= 0.685  (se = 0.026 )
Likelihood ratio test= 48.9  on 1 df,   p=3e-12
Wald test            = 52.66  on 1 df,   p=4e-13
Score (logrank) test = 55.63  on 1 df,   p=9e-14</code></pre>
</div>
</div>
<p>We see that the score is still prognostic, meaning that even patients that have high ER IHC percentage, there are those with higher risk. Note though that the analysis still is a bit limited due to the followup time that in median is 5.3 years.</p>
<p>We also compare the risk score for the high risk group and low/intermediate risk group of the ROR signature (<a href="#fig-risk-ror" class="quarto-xref">Figure&nbsp;<span>5.2</span></a>) obtained by the SCANB team using single sample predictor models, where rule based algorithms are used to determine if a patient is high or low/intermediate risk. They validate this on prosigna’s own sequencing and showed good results <span class="citation" data-cites="SCANB2022">(<a href="references.html#ref-SCANB2022" role="doc-biblioref">Staaf et al. 2022</a>)</span>.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-risk-ror" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-risk-ror-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="risk_score_files/figure-html/fig-risk-ror-1.png" class="img-fluid figure-img" width="672">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-risk-ror-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;5.2: Risk score stratified by binary category risk as defined by the SCANB team at <span class="citation" data-cites="SCANB2022">(<a href="references.html#ref-SCANB2022" role="doc-biblioref">Staaf et al. 2022</a>)</span>. ER+ BC patients that received only ET were considered.
</figcaption>
</figure>
</div>
</div>
</div>
<p>As expected most of the patients in the high risk group are luminal B. The ROR signature was developed in such a way to get luminal B patients to be of high risk. But here we see that there is still an overlap between low/intermediate patients and high risk patients, suggesting that the new risk score has additional information. For assessing this hypothesis we perform recurrence free survival analysis with both variables and just the risk score and compare the likelihood of the two models using anova.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>Analysis of Deviance Table
 Cox model: response is  Surv(time = rfs_months, event = rfs_status)
 Model 1: ~ tumor_size + node_status + age + SSP.ROR.binary.risk.cat
 Model 2: ~ risk_score + SSP.ROR.binary.risk.cat + tumor_size + node_status + age
   loglik  Chisq Df Pr(&gt;|Chi|)  
1 -1229.7                       
2 -1228.0 3.4646  1     0.0627 .
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
</div>
<p>We see that by adding the risk score it provides aditional value on top of the binary category from prosigna. On top of that the SCANB team calculated also the ROR score based on the original paper. So we actually have the actual scores derived from the nearest centroid technique for all patients.</p>
<p><a href="#fig-ror-riskscore-comparison" class="quarto-xref">Figure&nbsp;<span>5.3</span></a> shows the comparison between risk score and the ROR score from prosigna calculated by using NC.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-ror-riskscore-comparison" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-ror-riskscore-comparison-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="risk_score_files/figure-html/fig-ror-riskscore-comparison-1.png" class="img-fluid figure-img" width="672">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-ror-riskscore-comparison-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;5.3: Comparison of the risk score developed here with the ROR from Prosigna.
</figcaption>
</figure>
</div>
</div>
</div>
<p>There is a correlation but we see that among the luminal A patients there are still those that have high risk score besides a ROR lower than 50. Below we show the results of the spearman test on these two variables. We see they are correlated.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
    Spearman's rank correlation rho

data:  endo_only_scanb$risk_score and endo_only_scanb$NCN.ROR.asT0
S = 1233308206, p-value &lt; 2.2e-16
alternative hypothesis: true rho is not equal to 0
sample estimates:
     rho 
0.491857 </code></pre>
</div>
</div>
<p>With a p-value of:</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1] 5.832521e-149</code></pre>
</div>
</div>
<p>We now evaluate the risk score only on luminal A patients that received just endocrine therapy.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>Call:
survival::coxph(formula = as.formula(paste0("Surv(time = rfs_months, event = rfs_status) ~ ", 
    paste(c("risk_score", clin_vars), collapse = "+"))), data = endo_only_scanb %&gt;% 
    dplyr::filter(pam50 == "luma"))

                    coef exp(coef)  se(coef)      z      p
risk_score      1.548555  4.704666  0.730935  2.119 0.0341
tumor_size      0.003568  1.003574  0.016252  0.220 0.8262
node_statuspos -0.673885  0.509724  0.486784 -1.384 0.1662
age            -0.014232  0.985869  0.012078 -1.178 0.2386

Likelihood ratio test=29.45  on 4 df, p=6.331e-06
n= 1502, number of events= 71 </code></pre>
</div>
</div>
<p>Even among these patients the score is prognostic, despite the number of events, the hazard ratio is relatively high here. If we do the survival analysis by using overall survival as endpoint the results are shown below.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>Call:
survival::coxph(formula = as.formula(paste0("Surv(time = os_months, event = os_status) ~ ", 
    paste(c("risk_score", clin_vars), collapse = "+"))), data = endo_only_scanb %&gt;% 
    dplyr::filter(pam50 == "luma"))

                     coef  exp(coef)   se(coef)      z      p
risk_score      5.046e-01  1.656e+00  4.293e-01  1.175  0.240
tumor_size     -4.809e-05  1.000e+00  1.103e-02 -0.004  0.997
node_statuspos -2.390e-01  7.874e-01  2.738e-01 -0.873  0.383
age             9.506e-02  1.100e+00  7.974e-03 11.921 &lt;2e-16

Likelihood ratio test=234.5  on 4 df, p=&lt; 2.2e-16
n= 1502, number of events= 227 </code></pre>
</div>
</div>
<p>Risk score still an estimated hazard ratio higher than 1, with more variability though, still in the expected direction. So we see that this signature provides aditional information to the PAM50 ROR signature.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>Call:
survival::coxph(formula = as.formula(paste0("Surv(time = rfs_months, event = rfs_status) ~ ", 
    paste(c("risk_score", "NCN.ROR.asT0", clin_vars), collapse = "+"))), 
    data = endo_only_scanb %&gt;% dplyr::filter(pam50 == "luma"))

                   coef exp(coef) se(coef)      z     p
risk_score      0.94883   2.58268  0.98705  0.961 0.336
NCN.ROR.asT0    0.01038   1.01043  0.01120  0.927 0.354
tumor_size      0.01689   1.01704  0.02187  0.772 0.440
node_statuspos -0.31774   0.72779  0.62635 -0.507 0.612
age            -0.01230   0.98778  0.01221 -1.007 0.314

Likelihood ratio test=30.31  on 5 df, p=1.283e-05
n= 1502, number of events= 71 </code></pre>
</div>
</div>
</section>
<section id="standardized-survival-curves-for-scanb" class="level2" data-number="5.3">
<h2 data-number="5.3" class="anchored" data-anchor-id="standardized-survival-curves-for-scanb"><span class="header-section-number">5.3</span> Standardized survival curves for SCANB</h2>
<p>We now use also the standardized survival curves as suggested in the paper <span class="citation" data-cites="Syriopoulou2022">(<a href="references.html#ref-Syriopoulou2022" role="doc-biblioref">Syriopoulou et al. 2022</a>)</span>. For this we use the package <code>flexsurv</code> along with the functions <code>flexsurvspline</code> to model the baseline hazard using splines and then to calculate the standardized survival curves we use the function <code>standsurv</code>.</p>
<p>For the sake of interpretation we categorize the risk score in 4 categories, a low, intermediate, high and ultra-high risk. The thresholds are based on the data, we use the quartiles. We understand that we loose power when doing this but we are able to communicate better the information. For comparison we first show the spline based model on the continous score.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 8 × 5
  term           estimate std.error statistic   p.value
  &lt;chr&gt;             &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;
1 gamma0         -7.84      0.643       NA    NA       
2 gamma1          0.971     0.299       NA    NA       
3 gamma2          0.0661    0.290       NA    NA       
4 gamma3         -0.384     0.901       NA    NA       
5 gamma4          0.405     0.840       NA    NA       
6 risk_score      1.66      0.263        6.32  1.30e-10
7 tumor_size     -0.00950   0.00735     -1.29  9.80e- 2
8 node_statuspos -0.718     0.224       -3.20  6.80e- 4</code></pre>
</div>
</div>
<p>We use the “hazard” scale to define the proportional hazards model. The estimate is above 0 (1.8), meaning that the higher the risk score the worse it is. Moreover, the p-value is 1.1e-14.</p>
<p>Before moving on we just compare the risk score distributions between METABRIC and SCANB.</p>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="risk_score_files/figure-html/unnamed-chunk-21-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>The distribution shift makes sense as the majority of patients in SCANB have lower tumor stage. Let us compare by molecular subtype.</p>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="risk_score_files/figure-html/unnamed-chunk-22-1.png" class="img-fluid figure-img" width="1536"></p>
</figure>
</div>
</div>
</div>
<p>In general they agree a bit more when looking by subtype.</p>
<p>We now proceed with the categorized data for visualization purposes. We also use the survival scale for the interpretation. For the categories we use METABRIC’s quantiles instead of SCANB. This way we used the training dataset for defining the risk scores and then we use SCAN-B to further validate the differences.</p>
<p>The quartiles are (0, 0.25, 0.5, 0.75 and 1):</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1] -0.2767315  0.5788431  0.9485916  1.3608617  3.8333981</code></pre>
</div>
</div>
<p>The patients are stratified in the categories low, intermediate, high and ultra-high. For the analysis the low category is set as the baseline.</p>
<p>First we show the marginal survival curves.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-rfs-marginal-curve" class="quarto-figure quarto-figure-center quarto-float anchored" width="672">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-rfs-marginal-curve-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="risk_score_files/figure-html/fig-rfs-marginal-curve-1.png" id="fig-rfs-marginal-curve" class="img-fluid figure-img" width="672">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig quarto-uncaptioned" id="fig-rfs-marginal-curve-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;5.4
</figcaption>
</figure>
</div>
</div>
</div>
<p>And now the marginal curves for the differences with the low risk as the baseline.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-rfs-difference-marginal-curve" class="quarto-figure quarto-figure-center quarto-float anchored" width="672">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-rfs-difference-marginal-curve-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="risk_score_files/figure-html/fig-rfs-difference-marginal-curve-1.png" id="fig-rfs-difference-marginal-curve" class="img-fluid figure-img" width="672">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig quarto-uncaptioned" id="fig-rfs-difference-marginal-curve-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;5.5
</figcaption>
</figure>
</div>
</div>
</div>
<p>Here we see that the difference between low and ultra-high can be as high as 40%. On the other hand the differences between intermediate and low stay around 3 to 10% after 5 years.</p>
<p>The confidence intervals are larger as you increase the risk groups due to the number of patients available in each category.</p>
<div class="cell">
<div class="cell-output-display">
<div>
<table class="lightable-classic table table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">risk_score_category</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">1</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">2</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">low</td>
<td style="text-align: left;">95% (1,539)</td>
<td style="text-align: left;">5% (79)</td>
</tr>
<tr class="even">
<td style="text-align: left;">intermediate</td>
<td style="text-align: left;">91% (375)</td>
<td style="text-align: left;">9% (39)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">high</td>
<td style="text-align: left;">89% (198)</td>
<td style="text-align: left;">11% (24)</td>
</tr>
<tr class="even">
<td style="text-align: left;">ultra-high</td>
<td style="text-align: left;">77% (105)</td>
<td style="text-align: left;">23% (31)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Total</td>
<td style="text-align: left;">93% (2,217)</td>
<td style="text-align: left;">7% (173)</td>
</tr>
</tbody>
</table>


</div>
</div>
</div>
<p>Still the number of events in percentage goes up as one goes up in the risk category, as expected.</p>
</section>
<section id="risk-score-and-poetic-trial" class="level2" data-number="5.4">
<h2 data-number="5.4" class="anchored" data-anchor-id="risk-score-and-poetic-trial"><span class="header-section-number">5.4</span> Risk score and POETIC trial</h2>
<p>We can evaluate one of the two risk scores, the one using only the principal components, in the POETIC trial data. The first question we can address is if there is difference in scores in the baseline tissue (<a href="#fig-poetic-risk-score" class="quarto-xref">Figure&nbsp;<span>5.6</span></a>).</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-poetic-risk-score" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-poetic-risk-score-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="risk_score_files/figure-html/fig-poetic-risk-score-1.png" class="img-fluid figure-img" width="672">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-poetic-risk-score-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;5.6: Risk scores for responders and non-responders in the baseline tumor tissue.
</figcaption>
</figure>
</div>
</div>
</div>
<p>We see in the <a href="#fig-poetic-risk-score" class="quarto-xref">Figure&nbsp;<span>5.6</span></a> that in average the non responders have higher risk scores. Now if we compare the risk scores between baseline and surgery, the slope of the line that fits for each group separately, responder and non responder, is different. The slope for the non responders is parallel to the identity line going through the origin with a slope of 1, meaning that there is not much change of risk upon the treatment. On other hand, for the responder group the slope is decreased and has a value lower than 1, meaning that the risk is decreased upon aromatase inhibition after two weeks (<a href="#fig-poetic-risk-base-surgery" class="quarto-xref">Figure&nbsp;<span>5.7</span></a>).</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-poetic-risk-base-surgery" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-poetic-risk-base-surgery-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="risk_score_files/figure-html/fig-poetic-risk-base-surgery-1.png" class="img-fluid figure-img" width="672">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-poetic-risk-base-surgery-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;5.7: Comparison of risk scores using the PCs only before and after two weeks of aromatase inhibitors in post-menopausal women. The colored lines correspond do the linear regression for each response status group separately.
</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="risk-score-vs-pathways" class="level2" data-number="5.5">
<h2 data-number="5.5" class="anchored" data-anchor-id="risk-score-vs-pathways"><span class="header-section-number">5.5</span> Risk score vs pathways</h2>
<p>We now compare the risk score with pathways of interest as discussed in the previous sections. Again we only look at the patients that received only endocrine therapy in both SCANB and METABRIC.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-cor-risk-pathways" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-cor-risk-pathways-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="risk_score_files/figure-html/fig-cor-risk-pathways-1.png" class="img-fluid figure-img" width="1152">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-cor-risk-pathways-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;5.8: Risk score versus pathways of interest. Each subplot corresponds to a different pathway.
</figcaption>
</figure>
</div>
</div>
</div>
<p><a href="#fig-cor-risk-pathways" class="quarto-xref">Figure&nbsp;<span>5.8</span></a> shows that in general the signatures are not strongly associated with one of the main pathways displayed here. On the other hand, <a href="#fig-cor-riskpcs-pathways" class="quarto-xref">Figure&nbsp;<span>5.9</span></a> shows the association between the signatures and the risk scores derived only from the principal components.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-cor-riskpcs-pathways" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-cor-riskpcs-pathways-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="risk_score_files/figure-html/fig-cor-riskpcs-pathways-1.png" class="img-fluid figure-img" width="1152">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-cor-riskpcs-pathways-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;5.9: Risk score derived from the principal components only versus pathways of interest. Each subplot corresponds to a different pathway.
</figcaption>
</figure>
</div>
</div>
</div>
<p>In this case we see some associations with G2M checkpoint in general and a bit with estrogen signaling. We now calculate the correlation plots for each one of these pathways with both risk scores.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-corr-risk-pathways" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-corr-risk-pathways-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="risk_score_files/figure-html/fig-corr-risk-pathways-1.png" class="img-fluid figure-img" width="960">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-corr-risk-pathways-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;5.10: Correlation matrix for all pathways of interest and the two risk scores, both including the clinical factors and without the clinical factors.
</figcaption>
</figure>
</div>
</div>
</div>
<p><a href="#fig-corr-risk-pathways" class="quarto-xref">Figure&nbsp;<span>5.10</span></a> shows that G2M checkpoint is highly correlated with the risk score using only the principal components (<span class="math inline">\(\rho = 0.72\)</span>). When you include the clinical factors this value decreases. Moreover, DNA repair is highly associated with RS PC. Estrogen signaling was not so important here in the signatures surprisingly.</p>
<p>If we subselect only the SCANB samples we can recalculate those values and compare with the NCN ROR score.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-corr-risk-pathways-scanb" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-corr-risk-pathways-scanb-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="risk_score_files/figure-html/fig-corr-risk-pathways-scanb-1.png" class="img-fluid figure-img" width="960">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-corr-risk-pathways-scanb-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;5.11: Correlation matrix for all pathways of interest and the two risk scores, both including the clinical factors and without the clinical factors.
</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="abim-and-ncounter-data" class="level2" data-number="5.6">
<h2 data-number="5.6" class="anchored" data-anchor-id="abim-and-ncounter-data"><span class="header-section-number">5.6</span> ABiM and nCounter data</h2>
<p>In <span class="citation" data-cites="SCANB2022">(<a href="references.html#ref-SCANB2022" role="doc-biblioref">Staaf et al. 2022</a>)</span>, the authors validated their approach of using RNA-seq instead of nCounter data by showing how one can get the ROR and compared to nCounter data sequenced at their facility and sent to prosigna for the calculation of the risk of recurrence. They have matched RNA-seq data as well. Here we will use the ROR obtained from the nCounter data and use the RNA-seq to calculate the risk score here developed.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>Total number of stable genes: 44
Total number of genes: 1044
Number of samples: 100</code></pre>
</div>
</div>
<p><a href="#fig-pca-abim100-cohort" class="quarto-xref">Figure&nbsp;<span>5.12</span></a> shows the embedding of the ABiM cohort, it matches perfectly the molecular landscape.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pca-abim100-cohort" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-pca-abim100-cohort-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="risk_score_files/figure-html/fig-pca-abim100-cohort-1.png" class="img-fluid figure-img" width="864">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-pca-abim100-cohort-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;5.12: Embedding of all samples from TCGA, SCNAB and METABRIC including the samples from ABiM 100 that have matched nCounter data.
</figcaption>
</figure>
</div>
</div>
</div>
<p><a href="#fig-abim100-cohort" class="quarto-xref">Figure&nbsp;<span>5.13</span></a> shows how well the ABiM cohort actually overlays with the SCANB dataset, which makes sense as the processing pipeline is all similar.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-abim100-cohort" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-abim100-cohort-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="risk_score_files/figure-html/fig-abim100-cohort-1.png" class="img-fluid figure-img" width="672">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-abim100-cohort-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;5.13: Biplot of first two PCA components from ABiM 100, TCGA, SCANB and METABRIC. The normal cohort is highlighted in the plot and has bigger dot size.
</figcaption>
</figure>
</div>
</div>
</div>
<p>And now we move on and calculate the risk scores for the ABiM cohort.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-rorft-riskscore-abim" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-rorft-riskscore-abim-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="risk_score_files/figure-html/fig-rorft-riskscore-abim-1.png" class="img-fluid figure-img" width="672">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-rorft-riskscore-abim-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;5.14: Correlation between risk score calculated based on the principal components and clinical factors versus the Prosigna ROR obtained from the nCounter data and evaluated by Prosigna. Cohort used is the ABiM 100 samples.
</figcaption>
</figure>
</div>
</div>
</div>
<p><a href="#fig-rorft-riskscore-abim" class="quarto-xref">Figure&nbsp;<span>5.14</span></a> shows that there is a positive correlation between the risk score and ROR obtained from Prosigna. We see that some of the luminal A patients have higher ROR but lower ROR. Also, among luminal B patients there were some with the same ROR but different risk score, showing that the risk. Also some of the basal samples have smaller ROR than some of the luminal B and A samples, and the risk score captures these differences, i.e., all basal samples have risk scores higher than 0.75.</p>


<!-- -->

<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list" style="display: none">
<div id="ref-Buus2021" class="csl-entry" role="listitem">
Buus, Richard, Ivana Sestak, Ralf Kronenwett, Sean Ferree, Catherine A. Schnabel, Frederick L. Baehner, Elizabeth A. Mallon, Jack Cuzick, and Mitch Dowsett. 2021. <span>“Molecular Drivers of Onco<span class="math inline">\(\less\)</span>i<span class="math inline">\(\greater\)</span>type<span class="math inline">\(\less\)</span>/i<span class="math inline">\(\greater\)</span> <span>DX</span>, Prosigna, <span>EndoPredict</span>, and the Breast Cancer Index: A <span>TransATAC</span> Study.”</span> <em>Journal of Clinical Oncology</em> 39 (2): 126–35. <a href="https://doi.org/10.1200/jco.20.00853">https://doi.org/10.1200/jco.20.00853</a>.
</div>
<div id="ref-SCANB2022" class="csl-entry" role="listitem">
Staaf, Johan, Jari Häkkinen, Cecilia Hegardt, Lao H. Saal, Siker Kimbung, Ingrid Hedenfalk, Tonje Lien, et al. 2022. <span>“<span>RNA</span> Sequencing-Based Single Sample Predictors of Molecular Subtype and Risk of Recurrence for Clinical Assessment of Early-Stage Breast Cancer.”</span> <em>Npj Breast Cancer</em> 8 (1). <a href="https://doi.org/10.1038/s41523-022-00465-3">https://doi.org/10.1038/s41523-022-00465-3</a>.
</div>
<div id="ref-Syriopoulou2022" class="csl-entry" role="listitem">
Syriopoulou, Elisavet, Tove Wästerlid, Paul C. Lambert, and Therese M.-L. Andersson. 2022. <span>“Standardised Survival Probabilities: A Useful and Informative Tool for Reporting Regression Models for Survival Data.”</span> <em>British Journal of Cancer</em> 127 (10): 1808–15. <a href="https://doi.org/10.1038/s41416-022-01949-6">https://doi.org/10.1038/s41416-022-01949-6</a>.
</div>
</div>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="../chapters/scoring.html" class="pagination-link" aria-label="Pathway activity in a personalized context">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Pathway activity in a personalized context</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../chapters/expanding_ember.html" class="pagination-link" aria-label="Expanding embedding and removing batch effects">
        <span class="nav-page-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Expanding embedding and removing batch effects</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb18" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu"># Risk score model</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>Several risk signatures were developed over the years using transcriptomic</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>datasets. Some examples were given in the previous chapters, such as </span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>OncotypeDX, Prosigna and EndoPredict. They were all derived in a similar</span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>manner, subsetting genes related to usually estrogen and proliferation </span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>signaling and then using recurrence free survival data to calculate</span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>the risk scores. </span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>In this chapter we introduce a new risk score that does not subset</span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>genes and uses the information of the molecular landscape. The concept</span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a>is simple, we will use the third and fourth components as covariates in</span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a>the survival modeling along with tumor size and node status. This</span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a>way we avoid subsetting and selecting specific genes and we use all</span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a>the information the molecular landscape is providing us. Also it avoids</span>
<span id="cb18-16"><a href="#cb18-16" aria-hidden="true" tabindex="-1"></a>any kind of binning for patients based on molecular subtype, thus not being</span>
<span id="cb18-17"><a href="#cb18-17" aria-hidden="true" tabindex="-1"></a>necessary calculating correlations with pre-defined molecular subtypes.</span>
<span id="cb18-18"><a href="#cb18-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-19"><a href="#cb18-19" aria-hidden="true" tabindex="-1"></a>Later in the chapter we compare the risk score with the ROR binary </span>
<span id="cb18-20"><a href="#cb18-20" aria-hidden="true" tabindex="-1"></a>categories: high and low/intermediate risk. We also compare the risk score</span>
<span id="cb18-21"><a href="#cb18-21" aria-hidden="true" tabindex="-1"></a>with the ROR continuous value obtained by the SCANB team using nearest</span>
<span id="cb18-22"><a href="#cb18-22" aria-hidden="true" tabindex="-1"></a>centroid classification (NCN). We use samples from patients that were treated</span>
<span id="cb18-23"><a href="#cb18-23" aria-hidden="true" tabindex="-1"></a>with endocrine therapy only for the training of the risk score. Even though </span>
<span id="cb18-24"><a href="#cb18-24" aria-hidden="true" tabindex="-1"></a>the ROR was trained on a cohort of patients that received no treatment, we </span>
<span id="cb18-25"><a href="#cb18-25" aria-hidden="true" tabindex="-1"></a>can still compare the signatures. @Buus2021 showed that the EndoPredict </span>
<span id="cb18-26"><a href="#cb18-26" aria-hidden="true" tabindex="-1"></a>signature is highly correlated to ROR ($\rho = 0.68$), therefore we can </span>
<span id="cb18-27"><a href="#cb18-27" aria-hidden="true" tabindex="-1"></a>to some extent take conclusions based on the ROR signature.</span>
<span id="cb18-28"><a href="#cb18-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-29"><a href="#cb18-29" aria-hidden="true" tabindex="-1"></a><span class="in">```{r setup}</span></span>
<span id="cb18-30"><a href="#cb18-30" aria-hidden="true" tabindex="-1"></a><span class="in">renv::restore()</span></span>
<span id="cb18-31"><a href="#cb18-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-32"><a href="#cb18-32" aria-hidden="true" tabindex="-1"></a><span class="in">library(tidyverse)</span></span>
<span id="cb18-33"><a href="#cb18-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-34"><a href="#cb18-34" aria-hidden="true" tabindex="-1"></a><span class="in">library(ggplot2)</span></span>
<span id="cb18-35"><a href="#cb18-35" aria-hidden="true" tabindex="-1"></a><span class="in">library(PCAtools)</span></span>
<span id="cb18-36"><a href="#cb18-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-37"><a href="#cb18-37" aria-hidden="true" tabindex="-1"></a><span class="in">library(singscore)</span></span>
<span id="cb18-38"><a href="#cb18-38" aria-hidden="true" tabindex="-1"></a><span class="in">library(SummarizedExperiment)</span></span>
<span id="cb18-39"><a href="#cb18-39" aria-hidden="true" tabindex="-1"></a><span class="in">library(survival)</span></span>
<span id="cb18-40"><a href="#cb18-40" aria-hidden="true" tabindex="-1"></a><span class="in">library(flexsurv)</span></span>
<span id="cb18-41"><a href="#cb18-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-42"><a href="#cb18-42" aria-hidden="true" tabindex="-1"></a><span class="in">source("../R/utils.R")</span></span>
<span id="cb18-43"><a href="#cb18-43" aria-hidden="true" tabindex="-1"></a><span class="in">source("../R/first_run.R")</span></span>
<span id="cb18-44"><a href="#cb18-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-45"><a href="#cb18-45" aria-hidden="true" tabindex="-1"></a><span class="in"># the following script load all data necessary to run the chunks.</span></span>
<span id="cb18-46"><a href="#cb18-46" aria-hidden="true" tabindex="-1"></a><span class="in"># the data is generated from this quarto document itself, therefore</span></span>
<span id="cb18-47"><a href="#cb18-47" aria-hidden="true" tabindex="-1"></a><span class="in"># if you are running this documents the first time and don't have the</span></span>
<span id="cb18-48"><a href="#cb18-48" aria-hidden="true" tabindex="-1"></a><span class="in"># files, comment the following lines. Moreover, if this is your first</span></span>
<span id="cb18-49"><a href="#cb18-49" aria-hidden="true" tabindex="-1"></a><span class="in"># time running the document, you should run all chunks, to generate </span></span>
<span id="cb18-50"><a href="#cb18-50" aria-hidden="true" tabindex="-1"></a><span class="in"># all the necessary files, if you don't have them. Once all files </span></span>
<span id="cb18-51"><a href="#cb18-51" aria-hidden="true" tabindex="-1"></a><span class="in"># are saved and available in the respective folder, the following</span></span>
<span id="cb18-52"><a href="#cb18-52" aria-hidden="true" tabindex="-1"></a><span class="in"># lines can be executed. </span></span>
<span id="cb18-53"><a href="#cb18-53" aria-hidden="true" tabindex="-1"></a><span class="in">if (first_run){</span></span>
<span id="cb18-54"><a href="#cb18-54" aria-hidden="true" tabindex="-1"></a><span class="in">    load_at_setup &lt;- FALSE</span></span>
<span id="cb18-55"><a href="#cb18-55" aria-hidden="true" tabindex="-1"></a><span class="in">} else {</span></span>
<span id="cb18-56"><a href="#cb18-56" aria-hidden="true" tabindex="-1"></a><span class="in">    load_at_setup &lt;- TRUE</span></span>
<span id="cb18-57"><a href="#cb18-57" aria-hidden="true" tabindex="-1"></a><span class="in">}</span></span>
<span id="cb18-58"><a href="#cb18-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-59"><a href="#cb18-59" aria-hidden="true" tabindex="-1"></a><span class="in">name_document &lt;- "risk_score"</span></span>
<span id="cb18-60"><a href="#cb18-60" aria-hidden="true" tabindex="-1"></a><span class="in">sapply(</span></span>
<span id="cb18-61"><a href="#cb18-61" aria-hidden="true" tabindex="-1"></a><span class="in">    paste0("../../results/", c("plots", "rds_files", "tables"), "/", name_document),</span></span>
<span id="cb18-62"><a href="#cb18-62" aria-hidden="true" tabindex="-1"></a><span class="in">    dir.create,</span></span>
<span id="cb18-63"><a href="#cb18-63" aria-hidden="true" tabindex="-1"></a><span class="in">    showWarnings = FALSE,</span></span>
<span id="cb18-64"><a href="#cb18-64" aria-hidden="true" tabindex="-1"></a><span class="in">    recursive = TRUE</span></span>
<span id="cb18-65"><a href="#cb18-65" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb18-66"><a href="#cb18-66" aria-hidden="true" tabindex="-1"></a><span class="in">source("../R/load_rds_files.R")</span></span>
<span id="cb18-67"><a href="#cb18-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-68"><a href="#cb18-68" aria-hidden="true" tabindex="-1"></a><span class="in"># by setting the dev to png and pdf, this saves the figures in a specific</span></span>
<span id="cb18-69"><a href="#cb18-69" aria-hidden="true" tabindex="-1"></a><span class="in"># folder in both formats. moreover, since png is coming first, it shows</span></span>
<span id="cb18-70"><a href="#cb18-70" aria-hidden="true" tabindex="-1"></a><span class="in"># this figure when rendering the html. What is nice about this is that it</span></span>
<span id="cb18-71"><a href="#cb18-71" aria-hidden="true" tabindex="-1"></a><span class="in"># inherits the properties from the chunk to save the figure, so no need</span></span>
<span id="cb18-72"><a href="#cb18-72" aria-hidden="true" tabindex="-1"></a><span class="in"># to use ggplot2::ggsave to save the plots. this also works </span></span>
<span id="cb18-73"><a href="#cb18-73" aria-hidden="true" tabindex="-1"></a><span class="in">knitr::opts_chunk$set(dev = c('png', 'pdf'))</span></span>
<span id="cb18-74"><a href="#cb18-74" aria-hidden="true" tabindex="-1"></a><span class="in">options(bitmapType = 'cairo')</span></span>
<span id="cb18-75"><a href="#cb18-75" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb18-76"><a href="#cb18-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-77"><a href="#cb18-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-78"><a href="#cb18-78" aria-hidden="true" tabindex="-1"></a>We develop a risk score for early stage breast cancer patients that </span>
<span id="cb18-79"><a href="#cb18-79" aria-hidden="true" tabindex="-1"></a>received only endocrine therapy (ET). The idea is similar to what was</span>
<span id="cb18-80"><a href="#cb18-80" aria-hidden="true" tabindex="-1"></a>done with the EndoPredict signatures, but here we extend it by using the</span>
<span id="cb18-81"><a href="#cb18-81" aria-hidden="true" tabindex="-1"></a>molecular landscape information. This provides a full framework where</span>
<span id="cb18-82"><a href="#cb18-82" aria-hidden="true" tabindex="-1"></a>we can analyse the risk of recurrence even for patients that have </span>
<span id="cb18-83"><a href="#cb18-83" aria-hidden="true" tabindex="-1"></a>high ER+ percentage.</span>
<span id="cb18-84"><a href="#cb18-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-85"><a href="#cb18-85" aria-hidden="true" tabindex="-1"></a><span class="fu">## Generating the risk score</span></span>
<span id="cb18-86"><a href="#cb18-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-87"><a href="#cb18-87" aria-hidden="true" tabindex="-1"></a>We focus only on the endocrine treated patients. For the training part</span>
<span id="cb18-88"><a href="#cb18-88" aria-hidden="true" tabindex="-1"></a>of the algorithm we use a subset of patients from the METABRIC cohort.</span>
<span id="cb18-89"><a href="#cb18-89" aria-hidden="true" tabindex="-1"></a>We then test the risk scores in the test set and we further validate</span>
<span id="cb18-90"><a href="#cb18-90" aria-hidden="true" tabindex="-1"></a>the score on SCANB. </span>
<span id="cb18-91"><a href="#cb18-91" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-92"><a href="#cb18-92" aria-hidden="true" tabindex="-1"></a>We will generate two risk scores: one including only the principal </span>
<span id="cb18-93"><a href="#cb18-93" aria-hidden="true" tabindex="-1"></a>components and another one including the clinical features of the </span>
<span id="cb18-94"><a href="#cb18-94" aria-hidden="true" tabindex="-1"></a>patients when valiable, namely tumor size, node status and age (either having</span>
<span id="cb18-95"><a href="#cb18-95" aria-hidden="true" tabindex="-1"></a>one or more lymph node with breast cancer cells or none). </span>
<span id="cb18-96"><a href="#cb18-96" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-99"><a href="#cb18-99" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb18-100"><a href="#cb18-100" aria-hidden="true" tabindex="-1"></a><span class="co"># start by preparing the datasets so it is easily used later</span></span>
<span id="cb18-101"><a href="#cb18-101" aria-hidden="true" tabindex="-1"></a>df_pca_metabric <span class="ot">&lt;-</span> df_pca <span class="sc">%&gt;%</span> </span>
<span id="cb18-102"><a href="#cb18-102" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"metabric"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb18-103"><a href="#cb18-103" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">TreatGroup =</span> dplyr<span class="sc">::</span><span class="fu">case_when</span>(</span>
<span id="cb18-104"><a href="#cb18-104" aria-hidden="true" tabindex="-1"></a>        HORMONE_THERAPY <span class="sc">==</span> <span class="st">"YES"</span> <span class="sc">&amp;</span> CHEMOTHERAPY <span class="sc">==</span> <span class="st">"YES"</span> <span class="sc">~</span> <span class="st">"ChemoEndo"</span>,</span>
<span id="cb18-105"><a href="#cb18-105" aria-hidden="true" tabindex="-1"></a>        HORMONE_THERAPY <span class="sc">==</span> <span class="st">"YES"</span> <span class="sc">&amp;</span> CHEMOTHERAPY <span class="sc">==</span> <span class="st">"NO"</span> <span class="sc">~</span> <span class="st">"Endo"</span>,</span>
<span id="cb18-106"><a href="#cb18-106" aria-hidden="true" tabindex="-1"></a>        <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"Other"</span></span>
<span id="cb18-107"><a href="#cb18-107" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">%&gt;%</span> </span>
<span id="cb18-108"><a href="#cb18-108" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">node_status =</span> <span class="fu">ifelse</span>(</span>
<span id="cb18-109"><a href="#cb18-109" aria-hidden="true" tabindex="-1"></a>        LYMPH_NODES_EXAMINED_POSITIVE <span class="sc">&gt;</span> <span class="dv">0</span>, </span>
<span id="cb18-110"><a href="#cb18-110" aria-hidden="true" tabindex="-1"></a>        <span class="st">"pos"</span>,</span>
<span id="cb18-111"><a href="#cb18-111" aria-hidden="true" tabindex="-1"></a>        <span class="st">"neg"</span></span>
<span id="cb18-112"><a href="#cb18-112" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">%&gt;%</span> </span>
<span id="cb18-113"><a href="#cb18-113" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb18-114"><a href="#cb18-114" aria-hidden="true" tabindex="-1"></a>        <span class="at">node_status =</span> <span class="fu">factor</span>(node_status, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"neg"</span>, <span class="st">"pos"</span>)),</span>
<span id="cb18-115"><a href="#cb18-115" aria-hidden="true" tabindex="-1"></a>        <span class="at">tumor_size =</span> TUMOR_SIZE</span>
<span id="cb18-116"><a href="#cb18-116" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb18-117"><a href="#cb18-117" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-118"><a href="#cb18-118" aria-hidden="true" tabindex="-1"></a>df_pca_scanb <span class="ot">&lt;-</span> df_pca <span class="sc">%&gt;%</span> </span>
<span id="cb18-119"><a href="#cb18-119" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"scanb"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb18-120"><a href="#cb18-120" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">node_status =</span> <span class="fu">ifelse</span>(</span>
<span id="cb18-121"><a href="#cb18-121" aria-hidden="true" tabindex="-1"></a>        node_group <span class="sc">==</span> <span class="st">"N0"</span>, </span>
<span id="cb18-122"><a href="#cb18-122" aria-hidden="true" tabindex="-1"></a>        <span class="st">"neg"</span>,</span>
<span id="cb18-123"><a href="#cb18-123" aria-hidden="true" tabindex="-1"></a>        <span class="st">"pos"</span></span>
<span id="cb18-124"><a href="#cb18-124" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">%&gt;%</span> </span>
<span id="cb18-125"><a href="#cb18-125" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">node_status =</span> <span class="fu">factor</span>(</span>
<span id="cb18-126"><a href="#cb18-126" aria-hidden="true" tabindex="-1"></a>        node_status, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"neg"</span>, <span class="st">"pos"</span>)</span>
<span id="cb18-127"><a href="#cb18-127" aria-hidden="true" tabindex="-1"></a>    ))</span>
<span id="cb18-128"><a href="#cb18-128" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-129"><a href="#cb18-129" aria-hidden="true" tabindex="-1"></a><span class="co"># select only the endocrine treated patients</span></span>
<span id="cb18-130"><a href="#cb18-130" aria-hidden="true" tabindex="-1"></a>endo_only_scanb <span class="ot">&lt;-</span> df_pca_scanb <span class="sc">%&gt;%</span> </span>
<span id="cb18-131"><a href="#cb18-131" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb18-132"><a href="#cb18-132" aria-hidden="true" tabindex="-1"></a>            TreatGroup <span class="sc">==</span> <span class="st">"Endo"</span> <span class="sc">&amp;</span> </span>
<span id="cb18-133"><a href="#cb18-133" aria-hidden="true" tabindex="-1"></a>            <span class="sc">!</span><span class="fu">is.na</span>(rfs_status) <span class="sc">&amp;</span> </span>
<span id="cb18-134"><a href="#cb18-134" aria-hidden="true" tabindex="-1"></a>            <span class="sc">!</span><span class="fu">is.na</span>(rfs_months) <span class="sc">&amp;</span></span>
<span id="cb18-135"><a href="#cb18-135" aria-hidden="true" tabindex="-1"></a>            er_status <span class="sc">==</span> <span class="st">"pos"</span> <span class="sc">&amp;</span></span>
<span id="cb18-136"><a href="#cb18-136" aria-hidden="true" tabindex="-1"></a>            <span class="sc">!</span><span class="fu">is.na</span>(age) <span class="sc">&amp;</span></span>
<span id="cb18-137"><a href="#cb18-137" aria-hidden="true" tabindex="-1"></a>            <span class="sc">!</span><span class="fu">is.na</span>(tumor_size) <span class="sc">&amp;</span></span>
<span id="cb18-138"><a href="#cb18-138" aria-hidden="true" tabindex="-1"></a>            <span class="sc">!</span><span class="fu">is.na</span>(node_status)</span>
<span id="cb18-139"><a href="#cb18-139" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb18-140"><a href="#cb18-140" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-141"><a href="#cb18-141" aria-hidden="true" tabindex="-1"></a>endo_only_metabric <span class="ot">&lt;-</span> df_pca_metabric <span class="sc">%&gt;%</span> </span>
<span id="cb18-142"><a href="#cb18-142" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb18-143"><a href="#cb18-143" aria-hidden="true" tabindex="-1"></a>            HORMONE_THERAPY <span class="sc">==</span> <span class="st">"YES"</span> <span class="sc">&amp;</span> </span>
<span id="cb18-144"><a href="#cb18-144" aria-hidden="true" tabindex="-1"></a>            CHEMOTHERAPY <span class="sc">==</span> <span class="st">"NO"</span> <span class="sc">&amp;</span></span>
<span id="cb18-145"><a href="#cb18-145" aria-hidden="true" tabindex="-1"></a>            <span class="sc">!</span><span class="fu">is.na</span>(rfs_status) <span class="sc">&amp;</span> </span>
<span id="cb18-146"><a href="#cb18-146" aria-hidden="true" tabindex="-1"></a>            <span class="sc">!</span><span class="fu">is.na</span>(rfs_months) <span class="sc">&amp;</span></span>
<span id="cb18-147"><a href="#cb18-147" aria-hidden="true" tabindex="-1"></a>            er_status <span class="sc">==</span> <span class="st">"pos"</span> <span class="sc">&amp;</span></span>
<span id="cb18-148"><a href="#cb18-148" aria-hidden="true" tabindex="-1"></a>            <span class="sc">!</span><span class="fu">is.na</span>(age) <span class="sc">&amp;</span></span>
<span id="cb18-149"><a href="#cb18-149" aria-hidden="true" tabindex="-1"></a>            <span class="sc">!</span><span class="fu">is.na</span>(tumor_size) <span class="sc">&amp;</span></span>
<span id="cb18-150"><a href="#cb18-150" aria-hidden="true" tabindex="-1"></a>            <span class="sc">!</span><span class="fu">is.na</span>(node_status)</span>
<span id="cb18-151"><a href="#cb18-151" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb18-152"><a href="#cb18-152" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-153"><a href="#cb18-153" aria-hidden="true" tabindex="-1"></a><span class="co"># set the seed so we can reproduce the results later and then</span></span>
<span id="cb18-154"><a href="#cb18-154" aria-hidden="true" tabindex="-1"></a><span class="co"># select the training cohort</span></span>
<span id="cb18-155"><a href="#cb18-155" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">12938</span>)</span>
<span id="cb18-156"><a href="#cb18-156" aria-hidden="true" tabindex="-1"></a>training_set <span class="ot">&lt;-</span> endo_only_metabric <span class="sc">%&gt;%</span></span>
<span id="cb18-157"><a href="#cb18-157" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">slice_sample</span>(<span class="at">prop =</span> <span class="fl">0.6</span>)</span>
<span id="cb18-158"><a href="#cb18-158" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-159"><a href="#cb18-159" aria-hidden="true" tabindex="-1"></a><span class="co"># we use the two PCs we have been investigating, but that does not mean</span></span>
<span id="cb18-160"><a href="#cb18-160" aria-hidden="true" tabindex="-1"></a><span class="co"># other PCs can't be used</span></span>
<span id="cb18-161"><a href="#cb18-161" aria-hidden="true" tabindex="-1"></a>pcs_to_use <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"PC"</span>, <span class="fu">c</span>(<span class="dv">3</span><span class="sc">:</span><span class="dv">4</span>))</span>
<span id="cb18-162"><a href="#cb18-162" aria-hidden="true" tabindex="-1"></a>clin_vars <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"tumor_size"</span>, <span class="st">"node_status"</span>, <span class="st">"age"</span>)</span>
<span id="cb18-163"><a href="#cb18-163" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-164"><a href="#cb18-164" aria-hidden="true" tabindex="-1"></a>risk_score_model <span class="ot">&lt;-</span> survival<span class="sc">::</span><span class="fu">coxph</span>(</span>
<span id="cb18-165"><a href="#cb18-165" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.formula</span>(<span class="fu">paste0</span>(</span>
<span id="cb18-166"><a href="#cb18-166" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Surv(time = rfs_months, event = rfs_status) ~ "</span>,</span>
<span id="cb18-167"><a href="#cb18-167" aria-hidden="true" tabindex="-1"></a>        <span class="fu">paste</span>(<span class="fu">c</span>(pcs_to_use, clin_vars), <span class="at">collapse =</span> <span class="st">"+"</span>)</span>
<span id="cb18-168"><a href="#cb18-168" aria-hidden="true" tabindex="-1"></a>    )),</span>
<span id="cb18-169"><a href="#cb18-169" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> training_set</span>
<span id="cb18-170"><a href="#cb18-170" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb18-171"><a href="#cb18-171" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-172"><a href="#cb18-172" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb18-173"><a href="#cb18-173" aria-hidden="true" tabindex="-1"></a>    risk_score_model,</span>
<span id="cb18-174"><a href="#cb18-174" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../../results/rds_files/risk_score/risk_score_model.rds"</span></span>
<span id="cb18-175"><a href="#cb18-175" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb18-176"><a href="#cb18-176" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-177"><a href="#cb18-177" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(</span>
<span id="cb18-178"><a href="#cb18-178" aria-hidden="true" tabindex="-1"></a>    broom<span class="sc">::</span><span class="fu">tidy</span>(risk_score_model, <span class="at">exponentiate =</span> <span class="cn">TRUE</span>, <span class="at">conf.int =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb18-179"><a href="#cb18-179" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">hazard_ratio =</span> estimate),</span>
<span id="cb18-180"><a href="#cb18-180" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../../results/tables/risk_score/risk_score_model.csv"</span></span>
<span id="cb18-181"><a href="#cb18-181" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb18-182"><a href="#cb18-182" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb18-183"><a href="#cb18-183" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-186"><a href="#cb18-186" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb18-187"><a href="#cb18-187" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(risk_score_model)</span>
<span id="cb18-188"><a href="#cb18-188" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb18-189"><a href="#cb18-189" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-190"><a href="#cb18-190" aria-hidden="true" tabindex="-1"></a>We see that all coefficients have less variability than age. </span>
<span id="cb18-191"><a href="#cb18-191" aria-hidden="true" tabindex="-1"></a>Now we calculate only the risk score based on the components. </span>
<span id="cb18-192"><a href="#cb18-192" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-195"><a href="#cb18-195" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb18-196"><a href="#cb18-196" aria-hidden="true" tabindex="-1"></a>risk_score_model_pcs <span class="ot">&lt;-</span> survival<span class="sc">::</span><span class="fu">coxph</span>(</span>
<span id="cb18-197"><a href="#cb18-197" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.formula</span>(<span class="fu">paste0</span>(</span>
<span id="cb18-198"><a href="#cb18-198" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Surv(time = rfs_months, event = rfs_status) ~ "</span>,</span>
<span id="cb18-199"><a href="#cb18-199" aria-hidden="true" tabindex="-1"></a>        <span class="fu">paste</span>(<span class="fu">c</span>(pcs_to_use), <span class="at">collapse =</span> <span class="st">"+"</span>)</span>
<span id="cb18-200"><a href="#cb18-200" aria-hidden="true" tabindex="-1"></a>    )),</span>
<span id="cb18-201"><a href="#cb18-201" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> training_set</span>
<span id="cb18-202"><a href="#cb18-202" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb18-203"><a href="#cb18-203" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-204"><a href="#cb18-204" aria-hidden="true" tabindex="-1"></a>risk_score_model_clins <span class="ot">&lt;-</span> survival<span class="sc">::</span><span class="fu">coxph</span>(</span>
<span id="cb18-205"><a href="#cb18-205" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.formula</span>(<span class="fu">paste0</span>(</span>
<span id="cb18-206"><a href="#cb18-206" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Surv(time = rfs_months, event = rfs_status) ~ "</span>,</span>
<span id="cb18-207"><a href="#cb18-207" aria-hidden="true" tabindex="-1"></a>        <span class="fu">paste</span>(<span class="fu">c</span>(clin_vars), <span class="at">collapse =</span> <span class="st">"+"</span>)</span>
<span id="cb18-208"><a href="#cb18-208" aria-hidden="true" tabindex="-1"></a>    )),</span>
<span id="cb18-209"><a href="#cb18-209" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> training_set</span>
<span id="cb18-210"><a href="#cb18-210" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb18-211"><a href="#cb18-211" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-212"><a href="#cb18-212" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb18-213"><a href="#cb18-213" aria-hidden="true" tabindex="-1"></a>    risk_score_model_pcs,</span>
<span id="cb18-214"><a href="#cb18-214" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../../results/rds_files/risk_score/risk_score_model_pcs.rds"</span></span>
<span id="cb18-215"><a href="#cb18-215" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb18-216"><a href="#cb18-216" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb18-217"><a href="#cb18-217" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-218"><a href="#cb18-218" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-221"><a href="#cb18-221" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb18-222"><a href="#cb18-222" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(risk_score_model_pcs)</span>
<span id="cb18-223"><a href="#cb18-223" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb18-224"><a href="#cb18-224" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-225"><a href="#cb18-225" aria-hidden="true" tabindex="-1"></a>We see that the coefficients are very similar to what was found before,</span>
<span id="cb18-226"><a href="#cb18-226" aria-hidden="true" tabindex="-1"></a>it didn't change much. Also when we include the principal components on the</span>
<span id="cb18-227"><a href="#cb18-227" aria-hidden="true" tabindex="-1"></a>top of the clinical variables the log likelihood increases, which is </span>
<span id="cb18-228"><a href="#cb18-228" aria-hidden="true" tabindex="-1"></a>a good indication: </span>
<span id="cb18-229"><a href="#cb18-229" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-232"><a href="#cb18-232" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb18-233"><a href="#cb18-233" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(risk_score_model_clins, risk_score_model)</span>
<span id="cb18-234"><a href="#cb18-234" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb18-235"><a href="#cb18-235" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-238"><a href="#cb18-238" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb18-239"><a href="#cb18-239" aria-hidden="true" tabindex="-1"></a><span class="co"># first get the coefficients </span></span>
<span id="cb18-240"><a href="#cb18-240" aria-hidden="true" tabindex="-1"></a>coefs_risk_model <span class="ot">&lt;-</span> <span class="fu">coef</span>(risk_score_model)</span>
<span id="cb18-241"><a href="#cb18-241" aria-hidden="true" tabindex="-1"></a>coefs_risk_model_pcs <span class="ot">&lt;-</span> <span class="fu">coef</span>(risk_score_model_pcs)</span>
<span id="cb18-242"><a href="#cb18-242" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-243"><a href="#cb18-243" aria-hidden="true" tabindex="-1"></a><span class="co"># let us save them so they can be used later as well</span></span>
<span id="cb18-244"><a href="#cb18-244" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb18-245"><a href="#cb18-245" aria-hidden="true" tabindex="-1"></a>    coefs_risk_model,</span>
<span id="cb18-246"><a href="#cb18-246" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../../results/rds_files/risk_score/coefs_risk_model.rds"</span></span>
<span id="cb18-247"><a href="#cb18-247" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb18-248"><a href="#cb18-248" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb18-249"><a href="#cb18-249" aria-hidden="true" tabindex="-1"></a>    coefs_risk_model_pcs,</span>
<span id="cb18-250"><a href="#cb18-250" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../../results/rds_files/risk_score/coefs_risk_model_pcs.rds"</span></span>
<span id="cb18-251"><a href="#cb18-251" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb18-252"><a href="#cb18-252" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-253"><a href="#cb18-253" aria-hidden="true" tabindex="-1"></a><span class="co"># now we can calculate the scores for all the patients. For TCGA and POETIC</span></span>
<span id="cb18-254"><a href="#cb18-254" aria-hidden="true" tabindex="-1"></a><span class="co"># we do not have tumor size, so for those we calculate the risk score based</span></span>
<span id="cb18-255"><a href="#cb18-255" aria-hidden="true" tabindex="-1"></a><span class="co"># on the components only.</span></span>
<span id="cb18-256"><a href="#cb18-256" aria-hidden="true" tabindex="-1"></a>endo_only_scanb<span class="sc">$</span>risk_score <span class="ot">&lt;-</span> <span class="fu">calculate_risk_score</span>(</span>
<span id="cb18-257"><a href="#cb18-257" aria-hidden="true" tabindex="-1"></a>    coefs_risk_model, endo_only_scanb</span>
<span id="cb18-258"><a href="#cb18-258" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb18-259"><a href="#cb18-259" aria-hidden="true" tabindex="-1"></a>endo_only_scanb<span class="sc">$</span>risk_score_pcs <span class="ot">&lt;-</span> <span class="fu">calculate_risk_score</span>(</span>
<span id="cb18-260"><a href="#cb18-260" aria-hidden="true" tabindex="-1"></a>    coefs_risk_model_pcs, endo_only_scanb, <span class="at">is_node =</span> <span class="cn">FALSE</span></span>
<span id="cb18-261"><a href="#cb18-261" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb18-262"><a href="#cb18-262" aria-hidden="true" tabindex="-1"></a>endo_only_metabric<span class="sc">$</span>risk_score <span class="ot">&lt;-</span> <span class="fu">calculate_risk_score</span>(</span>
<span id="cb18-263"><a href="#cb18-263" aria-hidden="true" tabindex="-1"></a>    coefs_risk_model, endo_only_metabric</span>
<span id="cb18-264"><a href="#cb18-264" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb18-265"><a href="#cb18-265" aria-hidden="true" tabindex="-1"></a>endo_only_metabric<span class="sc">$</span>risk_score_pcs <span class="ot">&lt;-</span> <span class="fu">calculate_risk_score</span>(</span>
<span id="cb18-266"><a href="#cb18-266" aria-hidden="true" tabindex="-1"></a>    coefs_risk_model_pcs, endo_only_metabric, <span class="at">is_node =</span> <span class="cn">FALSE</span></span>
<span id="cb18-267"><a href="#cb18-267" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb18-268"><a href="#cb18-268" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-269"><a href="#cb18-269" aria-hidden="true" tabindex="-1"></a>df_pca_metabric<span class="sc">$</span>risk_score <span class="ot">&lt;-</span> <span class="fu">calculate_risk_score</span>(</span>
<span id="cb18-270"><a href="#cb18-270" aria-hidden="true" tabindex="-1"></a>    coefs_risk_model, </span>
<span id="cb18-271"><a href="#cb18-271" aria-hidden="true" tabindex="-1"></a>    df_pca_metabric</span>
<span id="cb18-272"><a href="#cb18-272" aria-hidden="true" tabindex="-1"></a>) </span>
<span id="cb18-273"><a href="#cb18-273" aria-hidden="true" tabindex="-1"></a>df_pca_scanb<span class="sc">$</span>risk_score <span class="ot">&lt;-</span> <span class="fu">calculate_risk_score</span>(coefs_risk_model, df_pca_scanb)</span>
<span id="cb18-274"><a href="#cb18-274" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-275"><a href="#cb18-275" aria-hidden="true" tabindex="-1"></a><span class="co"># we can now calculate the score for all samples based on PC3 and PC4</span></span>
<span id="cb18-276"><a href="#cb18-276" aria-hidden="true" tabindex="-1"></a>df_pca<span class="sc">$</span>risk_score_pcs <span class="ot">&lt;-</span> <span class="fu">calculate_risk_score</span>(</span>
<span id="cb18-277"><a href="#cb18-277" aria-hidden="true" tabindex="-1"></a>    coefs_risk_model_pcs, </span>
<span id="cb18-278"><a href="#cb18-278" aria-hidden="true" tabindex="-1"></a>    df_pca, </span>
<span id="cb18-279"><a href="#cb18-279" aria-hidden="true" tabindex="-1"></a>    <span class="at">is_node =</span> <span class="cn">FALSE</span></span>
<span id="cb18-280"><a href="#cb18-280" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb18-281"><a href="#cb18-281" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-282"><a href="#cb18-282" aria-hidden="true" tabindex="-1"></a>test_set <span class="ot">&lt;-</span> endo_only_metabric <span class="sc">%&gt;%</span></span>
<span id="cb18-283"><a href="#cb18-283" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span>(sample_name <span class="sc">%in%</span> training_set<span class="sc">$</span>sample_name))</span>
<span id="cb18-284"><a href="#cb18-284" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-285"><a href="#cb18-285" aria-hidden="true" tabindex="-1"></a>risk_score_test <span class="ot">&lt;-</span> survival<span class="sc">::</span><span class="fu">coxph</span>(</span>
<span id="cb18-286"><a href="#cb18-286" aria-hidden="true" tabindex="-1"></a>    <span class="fu">Surv</span>(<span class="at">time =</span> rfs_months, <span class="at">event =</span> rfs_status) <span class="sc">~</span> risk_score,</span>
<span id="cb18-287"><a href="#cb18-287" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> test_set</span>
<span id="cb18-288"><a href="#cb18-288" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb18-289"><a href="#cb18-289" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-290"><a href="#cb18-290" aria-hidden="true" tabindex="-1"></a>risk_score_test_clins <span class="ot">&lt;-</span> survival<span class="sc">::</span><span class="fu">coxph</span>(</span>
<span id="cb18-291"><a href="#cb18-291" aria-hidden="true" tabindex="-1"></a>    <span class="fu">Surv</span>(<span class="at">time =</span> rfs_months, <span class="at">event =</span> rfs_status) <span class="sc">~</span> risk_score <span class="sc">+</span> </span>
<span id="cb18-292"><a href="#cb18-292" aria-hidden="true" tabindex="-1"></a>        tumor_size <span class="sc">+</span> npi,</span>
<span id="cb18-293"><a href="#cb18-293" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> test_set</span>
<span id="cb18-294"><a href="#cb18-294" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb18-295"><a href="#cb18-295" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb18-296"><a href="#cb18-296" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-297"><a href="#cb18-297" aria-hidden="true" tabindex="-1"></a>Now we use the test set to calculate the score.</span>
<span id="cb18-298"><a href="#cb18-298" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-301"><a href="#cb18-301" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb18-302"><a href="#cb18-302" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(risk_score_test)</span>
<span id="cb18-303"><a href="#cb18-303" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb18-304"><a href="#cb18-304" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-305"><a href="#cb18-305" aria-hidden="true" tabindex="-1"></a>We see that the higher the risk score the worse it is for the patients </span>
<span id="cb18-306"><a href="#cb18-306" aria-hidden="true" tabindex="-1"></a>treated only with endocrine therapy. </span>
<span id="cb18-307"><a href="#cb18-307" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-308"><a href="#cb18-308" aria-hidden="true" tabindex="-1"></a><span class="fu">## Validating the risk score on SCANB</span></span>
<span id="cb18-309"><a href="#cb18-309" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-310"><a href="#cb18-310" aria-hidden="true" tabindex="-1"></a>We can now validate the risk score on the SCANB dataset. </span>
<span id="cb18-311"><a href="#cb18-311" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-314"><a href="#cb18-314" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb18-315"><a href="#cb18-315" aria-hidden="true" tabindex="-1"></a>risk_score_scanb <span class="ot">&lt;-</span> survival<span class="sc">::</span><span class="fu">coxph</span>(</span>
<span id="cb18-316"><a href="#cb18-316" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.formula</span>(<span class="fu">paste0</span>(</span>
<span id="cb18-317"><a href="#cb18-317" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Surv(time = rfs_months, event = rfs_status) ~ "</span>,</span>
<span id="cb18-318"><a href="#cb18-318" aria-hidden="true" tabindex="-1"></a>        <span class="fu">paste</span>(<span class="fu">c</span>(<span class="st">"risk_score"</span>), <span class="at">collapse =</span> <span class="st">"+"</span>)</span>
<span id="cb18-319"><a href="#cb18-319" aria-hidden="true" tabindex="-1"></a>    )),</span>
<span id="cb18-320"><a href="#cb18-320" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> endo_only_scanb</span>
<span id="cb18-321"><a href="#cb18-321" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb18-322"><a href="#cb18-322" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-323"><a href="#cb18-323" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(risk_score_scanb)</span>
<span id="cb18-324"><a href="#cb18-324" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb18-325"><a href="#cb18-325" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-326"><a href="#cb18-326" aria-hidden="true" tabindex="-1"></a>Again the risk score is prognostic with a high coefficient, meaning</span>
<span id="cb18-327"><a href="#cb18-327" aria-hidden="true" tabindex="-1"></a>that the higher the score the worse it is for the patient. </span>
<span id="cb18-328"><a href="#cb18-328" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-329"><a href="#cb18-329" aria-hidden="true" tabindex="-1"></a>We now visualize a bit the risk scores developed here in terms </span>
<span id="cb18-330"><a href="#cb18-330" aria-hidden="true" tabindex="-1"></a>of the molecular landscape. @fig-risk-endo-pam50</span>
<span id="cb18-331"><a href="#cb18-331" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-332"><a href="#cb18-332" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=7, fig.height=5}</span></span>
<span id="cb18-333"><a href="#cb18-333" aria-hidden="true" tabindex="-1"></a><span class="in">#| label: fig-risk-endo-pam50</span></span>
<span id="cb18-334"><a href="#cb18-334" aria-hidden="true" tabindex="-1"></a><span class="in">#| fig-cap: Molecular landscape of SCANB patients treated with only</span></span>
<span id="cb18-335"><a href="#cb18-335" aria-hidden="true" tabindex="-1"></a><span class="in">#|     endocrine therapy and that have ER+ breast cancer. Only patients</span></span>
<span id="cb18-336"><a href="#cb18-336" aria-hidden="true" tabindex="-1"></a><span class="in">#|     with risk score less than 2 are plotted here. There are only 26</span></span>
<span id="cb18-337"><a href="#cb18-337" aria-hidden="true" tabindex="-1"></a><span class="in">#|     patients out of 2829 with risk score higher than 2 that would skew </span></span>
<span id="cb18-338"><a href="#cb18-338" aria-hidden="true" tabindex="-1"></a><span class="in">#|     the color scale.</span></span>
<span id="cb18-339"><a href="#cb18-339" aria-hidden="true" tabindex="-1"></a><span class="in">endo_only_scanb %&gt;%</span></span>
<span id="cb18-340"><a href="#cb18-340" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::filter(risk_score &lt;= 2) %&gt;%</span></span>
<span id="cb18-341"><a href="#cb18-341" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::ggplot(aes(x = PC3, y = PC4, z = risk_score)) +</span></span>
<span id="cb18-342"><a href="#cb18-342" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::stat_summary_hex(bins = 25) +</span></span>
<span id="cb18-343"><a href="#cb18-343" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::scale_fill_viridis_c() +</span></span>
<span id="cb18-344"><a href="#cb18-344" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme_bw(base_size = 15) +</span></span>
<span id="cb18-345"><a href="#cb18-345" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::labs(</span></span>
<span id="cb18-346"><a href="#cb18-346" aria-hidden="true" tabindex="-1"></a><span class="in">        fill = "Risk score",</span></span>
<span id="cb18-347"><a href="#cb18-347" aria-hidden="true" tabindex="-1"></a><span class="in">        title = paste0(</span></span>
<span id="cb18-348"><a href="#cb18-348" aria-hidden="true" tabindex="-1"></a><span class="in">            "Clinical risk score for SCANB patients treated\nwith ET only"</span></span>
<span id="cb18-349"><a href="#cb18-349" aria-hidden="true" tabindex="-1"></a><span class="in">        )</span></span>
<span id="cb18-350"><a href="#cb18-350" aria-hidden="true" tabindex="-1"></a><span class="in">    ) +</span></span>
<span id="cb18-351"><a href="#cb18-351" aria-hidden="true" tabindex="-1"></a><span class="in">    change_plot_aes_point()</span></span>
<span id="cb18-352"><a href="#cb18-352" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb18-353"><a href="#cb18-353" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-354"><a href="#cb18-354" aria-hidden="true" tabindex="-1"></a>We now investigate the risk score among patients that have high ER IHC</span>
<span id="cb18-355"><a href="#cb18-355" aria-hidden="true" tabindex="-1"></a>percentage as well. We have seen that these patients have different outcomes</span>
<span id="cb18-356"><a href="#cb18-356" aria-hidden="true" tabindex="-1"></a>based on the molecular estrogen signaling. We now investigate the effect</span>
<span id="cb18-357"><a href="#cb18-357" aria-hidden="true" tabindex="-1"></a>of the risk score. </span>
<span id="cb18-358"><a href="#cb18-358" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-361"><a href="#cb18-361" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb18-362"><a href="#cb18-362" aria-hidden="true" tabindex="-1"></a>risk_score_scanb_highER <span class="ot">&lt;-</span> survival<span class="sc">::</span><span class="fu">coxph</span>(</span>
<span id="cb18-363"><a href="#cb18-363" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.formula</span>(<span class="fu">paste0</span>(</span>
<span id="cb18-364"><a href="#cb18-364" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Surv(time = rfs_months, event = rfs_status) ~ "</span>,</span>
<span id="cb18-365"><a href="#cb18-365" aria-hidden="true" tabindex="-1"></a>        <span class="fu">paste</span>(<span class="fu">c</span>(<span class="st">"risk_score"</span>), <span class="at">collapse =</span> <span class="st">"+"</span>)</span>
<span id="cb18-366"><a href="#cb18-366" aria-hidden="true" tabindex="-1"></a>    )),</span>
<span id="cb18-367"><a href="#cb18-367" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> endo_only_scanb <span class="sc">%&gt;%</span></span>
<span id="cb18-368"><a href="#cb18-368" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">filter</span>(ER.pct <span class="sc">&gt;=</span> <span class="dv">90</span> <span class="sc">&amp;</span> risk_score <span class="sc">&lt;</span> <span class="dv">2</span>)</span>
<span id="cb18-369"><a href="#cb18-369" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb18-370"><a href="#cb18-370" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-371"><a href="#cb18-371" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(risk_score_scanb_highER)</span>
<span id="cb18-372"><a href="#cb18-372" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb18-373"><a href="#cb18-373" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-374"><a href="#cb18-374" aria-hidden="true" tabindex="-1"></a>We see that the score is still prognostic, meaning that even patients</span>
<span id="cb18-375"><a href="#cb18-375" aria-hidden="true" tabindex="-1"></a>that have high ER IHC percentage, there are those with higher risk. Note</span>
<span id="cb18-376"><a href="#cb18-376" aria-hidden="true" tabindex="-1"></a>though that the analysis still is a bit limited due to the followup time</span>
<span id="cb18-377"><a href="#cb18-377" aria-hidden="true" tabindex="-1"></a>that in median is <span class="in">`r format(median(endo_only_scanb$rfs_month)/12, digits = 2)`</span> </span>
<span id="cb18-378"><a href="#cb18-378" aria-hidden="true" tabindex="-1"></a>years.</span>
<span id="cb18-379"><a href="#cb18-379" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-380"><a href="#cb18-380" aria-hidden="true" tabindex="-1"></a>We also compare the risk score for the high risk group and low/intermediate</span>
<span id="cb18-381"><a href="#cb18-381" aria-hidden="true" tabindex="-1"></a>risk group of the ROR signature (@fig-risk-ror) </span>
<span id="cb18-382"><a href="#cb18-382" aria-hidden="true" tabindex="-1"></a>obtained by the SCANB team using single </span>
<span id="cb18-383"><a href="#cb18-383" aria-hidden="true" tabindex="-1"></a>sample predictor models, where rule based algorithms are used to determine</span>
<span id="cb18-384"><a href="#cb18-384" aria-hidden="true" tabindex="-1"></a>if a patient is high or low/intermediate risk. They validate this on</span>
<span id="cb18-385"><a href="#cb18-385" aria-hidden="true" tabindex="-1"></a>prosigna's own sequencing and showed good results <span class="co">[</span><span class="ot">@SCANB2022</span><span class="co">]</span>.</span>
<span id="cb18-386"><a href="#cb18-386" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-387"><a href="#cb18-387" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=7, fig.height=6}</span></span>
<span id="cb18-388"><a href="#cb18-388" aria-hidden="true" tabindex="-1"></a><span class="in">#| label: fig-risk-ror</span></span>
<span id="cb18-389"><a href="#cb18-389" aria-hidden="true" tabindex="-1"></a><span class="in">#| fig-cap: Risk score stratified by binary category risk as defined </span></span>
<span id="cb18-390"><a href="#cb18-390" aria-hidden="true" tabindex="-1"></a><span class="in">#|     by the SCANB team at [@SCANB2022]. ER+ BC patients that received </span></span>
<span id="cb18-391"><a href="#cb18-391" aria-hidden="true" tabindex="-1"></a><span class="in">#|     only ET were considered.</span></span>
<span id="cb18-392"><a href="#cb18-392" aria-hidden="true" tabindex="-1"></a><span class="in">endo_only_scanb %&gt;% </span></span>
<span id="cb18-393"><a href="#cb18-393" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::ggplot(</span></span>
<span id="cb18-394"><a href="#cb18-394" aria-hidden="true" tabindex="-1"></a><span class="in">        aes(x = SSP.ROR.binary.risk.cat, y = risk_score, color = pam50)</span></span>
<span id="cb18-395"><a href="#cb18-395" aria-hidden="true" tabindex="-1"></a><span class="in">    ) +</span></span>
<span id="cb18-396"><a href="#cb18-396" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_jitter() + </span></span>
<span id="cb18-397"><a href="#cb18-397" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_violin(color = "black", alpha = 0.3) +</span></span>
<span id="cb18-398"><a href="#cb18-398" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_boxplot(</span></span>
<span id="cb18-399"><a href="#cb18-399" aria-hidden="true" tabindex="-1"></a><span class="in">        width=0.1, </span></span>
<span id="cb18-400"><a href="#cb18-400" aria-hidden="true" tabindex="-1"></a><span class="in">        color = "black", </span></span>
<span id="cb18-401"><a href="#cb18-401" aria-hidden="true" tabindex="-1"></a><span class="in">        alpha = 0.3,</span></span>
<span id="cb18-402"><a href="#cb18-402" aria-hidden="true" tabindex="-1"></a><span class="in">        outlier.shape = NA</span></span>
<span id="cb18-403"><a href="#cb18-403" aria-hidden="true" tabindex="-1"></a><span class="in">    ) +</span></span>
<span id="cb18-404"><a href="#cb18-404" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme_bw(base_size = 15) +</span></span>
<span id="cb18-405"><a href="#cb18-405" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::labs(</span></span>
<span id="cb18-406"><a href="#cb18-406" aria-hidden="true" tabindex="-1"></a><span class="in">        x = "ROR binary risk category (SSP)",</span></span>
<span id="cb18-407"><a href="#cb18-407" aria-hidden="true" tabindex="-1"></a><span class="in">        title = paste0(</span></span>
<span id="cb18-408"><a href="#cb18-408" aria-hidden="true" tabindex="-1"></a><span class="in">            "ER+ BC patients from SCANB treated with ET only"</span></span>
<span id="cb18-409"><a href="#cb18-409" aria-hidden="true" tabindex="-1"></a><span class="in">        ),</span></span>
<span id="cb18-410"><a href="#cb18-410" aria-hidden="true" tabindex="-1"></a><span class="in">        y = "Risk score",</span></span>
<span id="cb18-411"><a href="#cb18-411" aria-hidden="true" tabindex="-1"></a><span class="in">        color = "Molecular\nsubtype"</span></span>
<span id="cb18-412"><a href="#cb18-412" aria-hidden="true" tabindex="-1"></a><span class="in">    ) + </span></span>
<span id="cb18-413"><a href="#cb18-413" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::scale_color_manual(</span></span>
<span id="cb18-414"><a href="#cb18-414" aria-hidden="true" tabindex="-1"></a><span class="in">       values = get_colors_pam50(endo_only_scanb)</span></span>
<span id="cb18-415"><a href="#cb18-415" aria-hidden="true" tabindex="-1"></a><span class="in">    ) +</span></span>
<span id="cb18-416"><a href="#cb18-416" aria-hidden="true" tabindex="-1"></a><span class="in">    change_plot_aes_point() + </span></span>
<span id="cb18-417"><a href="#cb18-417" aria-hidden="true" tabindex="-1"></a><span class="in">    change_guides_point()</span></span>
<span id="cb18-418"><a href="#cb18-418" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb18-419"><a href="#cb18-419" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-420"><a href="#cb18-420" aria-hidden="true" tabindex="-1"></a>As expected most of the patients in the high risk group are luminal B.</span>
<span id="cb18-421"><a href="#cb18-421" aria-hidden="true" tabindex="-1"></a>The ROR signature was developed in such a way to get luminal B patients to be</span>
<span id="cb18-422"><a href="#cb18-422" aria-hidden="true" tabindex="-1"></a>of high risk. But here we see that there is still an overlap between</span>
<span id="cb18-423"><a href="#cb18-423" aria-hidden="true" tabindex="-1"></a>low/intermediate patients and high risk patients, suggesting that the</span>
<span id="cb18-424"><a href="#cb18-424" aria-hidden="true" tabindex="-1"></a>new risk score has additional information. For assessing this hypothesis</span>
<span id="cb18-425"><a href="#cb18-425" aria-hidden="true" tabindex="-1"></a>we perform recurrence free survival analysis with both variables and just</span>
<span id="cb18-426"><a href="#cb18-426" aria-hidden="true" tabindex="-1"></a>the risk score and compare the likelihood of the two models using anova. </span>
<span id="cb18-427"><a href="#cb18-427" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-430"><a href="#cb18-430" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb18-431"><a href="#cb18-431" aria-hidden="true" tabindex="-1"></a>risk_score_scanb_ROR <span class="ot">&lt;-</span> survival<span class="sc">::</span><span class="fu">coxph</span>(</span>
<span id="cb18-432"><a href="#cb18-432" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.formula</span>(<span class="fu">paste0</span>(</span>
<span id="cb18-433"><a href="#cb18-433" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Surv(time = rfs_months, event = rfs_status) ~ "</span>,</span>
<span id="cb18-434"><a href="#cb18-434" aria-hidden="true" tabindex="-1"></a>        <span class="fu">paste</span>(</span>
<span id="cb18-435"><a href="#cb18-435" aria-hidden="true" tabindex="-1"></a>            <span class="fu">c</span>(<span class="st">"risk_score"</span>, <span class="st">"SSP.ROR.binary.risk.cat"</span>, clin_vars), </span>
<span id="cb18-436"><a href="#cb18-436" aria-hidden="true" tabindex="-1"></a>            <span class="at">collapse =</span> <span class="st">"+"</span></span>
<span id="cb18-437"><a href="#cb18-437" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb18-438"><a href="#cb18-438" aria-hidden="true" tabindex="-1"></a>    )),</span>
<span id="cb18-439"><a href="#cb18-439" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> endo_only_scanb</span>
<span id="cb18-440"><a href="#cb18-440" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb18-441"><a href="#cb18-441" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-442"><a href="#cb18-442" aria-hidden="true" tabindex="-1"></a>risk_score_scanb_onlyROR <span class="ot">&lt;-</span> survival<span class="sc">::</span><span class="fu">coxph</span>(</span>
<span id="cb18-443"><a href="#cb18-443" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.formula</span>(<span class="fu">paste0</span>(</span>
<span id="cb18-444"><a href="#cb18-444" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Surv(time = rfs_months, event = rfs_status) ~ "</span>,</span>
<span id="cb18-445"><a href="#cb18-445" aria-hidden="true" tabindex="-1"></a>        <span class="fu">paste</span>(</span>
<span id="cb18-446"><a href="#cb18-446" aria-hidden="true" tabindex="-1"></a>            <span class="fu">c</span>(clin_vars, <span class="st">"SSP.ROR.binary.risk.cat"</span>), </span>
<span id="cb18-447"><a href="#cb18-447" aria-hidden="true" tabindex="-1"></a>            <span class="at">collapse =</span> <span class="st">"+"</span></span>
<span id="cb18-448"><a href="#cb18-448" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb18-449"><a href="#cb18-449" aria-hidden="true" tabindex="-1"></a>    )),</span>
<span id="cb18-450"><a href="#cb18-450" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> endo_only_scanb</span>
<span id="cb18-451"><a href="#cb18-451" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb18-452"><a href="#cb18-452" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-453"><a href="#cb18-453" aria-hidden="true" tabindex="-1"></a>anova_out <span class="ot">&lt;-</span> <span class="fu">anova</span>(risk_score_scanb_onlyROR, risk_score_scanb_ROR)</span>
<span id="cb18-454"><a href="#cb18-454" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(</span>
<span id="cb18-455"><a href="#cb18-455" aria-hidden="true" tabindex="-1"></a>    broom<span class="sc">::</span><span class="fu">tidy</span>(anova_out),</span>
<span id="cb18-456"><a href="#cb18-456" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../../results/tables/risk_score/anova_ROR_RR.csv"</span></span>
<span id="cb18-457"><a href="#cb18-457" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb18-458"><a href="#cb18-458" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb18-459"><a href="#cb18-459" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-462"><a href="#cb18-462" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb18-463"><a href="#cb18-463" aria-hidden="true" tabindex="-1"></a>anova_out</span>
<span id="cb18-464"><a href="#cb18-464" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb18-465"><a href="#cb18-465" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-466"><a href="#cb18-466" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-467"><a href="#cb18-467" aria-hidden="true" tabindex="-1"></a>We see that by adding the risk score it provides aditional value on top of </span>
<span id="cb18-468"><a href="#cb18-468" aria-hidden="true" tabindex="-1"></a>the binary category from prosigna. On top of that the SCANB team calculated</span>
<span id="cb18-469"><a href="#cb18-469" aria-hidden="true" tabindex="-1"></a>also the ROR score based on the original paper. So we actually have the </span>
<span id="cb18-470"><a href="#cb18-470" aria-hidden="true" tabindex="-1"></a>actual scores derived from the nearest centroid technique for all patients.  </span>
<span id="cb18-471"><a href="#cb18-471" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-472"><a href="#cb18-472" aria-hidden="true" tabindex="-1"></a>@fig-ror-riskscore-comparison shows the comparison between risk score </span>
<span id="cb18-473"><a href="#cb18-473" aria-hidden="true" tabindex="-1"></a>and the ROR score from prosigna calculated by using NC. </span>
<span id="cb18-474"><a href="#cb18-474" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-477"><a href="#cb18-477" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb18-478"><a href="#cb18-478" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-ror-riskscore-comparison</span></span>
<span id="cb18-479"><a href="#cb18-479" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Comparison of the risk score developed here with the ROR from</span></span>
<span id="cb18-480"><a href="#cb18-480" aria-hidden="true" tabindex="-1"></a><span class="co">#|     Prosigna. </span></span>
<span id="cb18-481"><a href="#cb18-481" aria-hidden="true" tabindex="-1"></a>endo_only_scanb <span class="sc">%&gt;%</span> </span>
<span id="cb18-482"><a href="#cb18-482" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> risk_score, NCN.ROR.asT0, <span class="at">color =</span> pam50)) <span class="sc">+</span></span>
<span id="cb18-483"><a href="#cb18-483" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb18-484"><a href="#cb18-484" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">15</span>) <span class="sc">+</span></span>
<span id="cb18-485"><a href="#cb18-485" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb18-486"><a href="#cb18-486" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"Risk score"</span>,</span>
<span id="cb18-487"><a href="#cb18-487" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"ROR (by NC)"</span>,</span>
<span id="cb18-488"><a href="#cb18-488" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="fu">paste0</span>(</span>
<span id="cb18-489"><a href="#cb18-489" aria-hidden="true" tabindex="-1"></a>            <span class="st">"Clinical risk score vs ROR for SCANB"</span></span>
<span id="cb18-490"><a href="#cb18-490" aria-hidden="true" tabindex="-1"></a>        ),</span>
<span id="cb18-491"><a href="#cb18-491" aria-hidden="true" tabindex="-1"></a>        <span class="at">subtitle =</span> <span class="st">"Patients treated with ET only"</span>,</span>
<span id="cb18-492"><a href="#cb18-492" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">"Molecular</span><span class="sc">\n</span><span class="st">subtype"</span></span>
<span id="cb18-493"><a href="#cb18-493" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb18-494"><a href="#cb18-494" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_color_manual</span>(</span>
<span id="cb18-495"><a href="#cb18-495" aria-hidden="true" tabindex="-1"></a>        <span class="at">values =</span> <span class="fu">get_colors_pam50</span>(endo_only_scanb)</span>
<span id="cb18-496"><a href="#cb18-496" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb18-497"><a href="#cb18-497" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>() <span class="sc">+</span></span>
<span id="cb18-498"><a href="#cb18-498" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>()</span>
<span id="cb18-499"><a href="#cb18-499" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb18-500"><a href="#cb18-500" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-501"><a href="#cb18-501" aria-hidden="true" tabindex="-1"></a>There is a correlation but we see that among the luminal A patients </span>
<span id="cb18-502"><a href="#cb18-502" aria-hidden="true" tabindex="-1"></a>there are still those that have high risk score besides a ROR lower than</span>
<span id="cb18-503"><a href="#cb18-503" aria-hidden="true" tabindex="-1"></a><span class="ss">50. </span>Below we show the results of the spearman test on these two variables.</span>
<span id="cb18-504"><a href="#cb18-504" aria-hidden="true" tabindex="-1"></a>We see they are correlated.</span>
<span id="cb18-505"><a href="#cb18-505" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-508"><a href="#cb18-508" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb18-509"><a href="#cb18-509" aria-hidden="true" tabindex="-1"></a>spearman_riskscore_ror <span class="ot">&lt;-</span> <span class="fu">cor.test</span>(</span>
<span id="cb18-510"><a href="#cb18-510" aria-hidden="true" tabindex="-1"></a>    endo_only_scanb<span class="sc">$</span>risk_score,</span>
<span id="cb18-511"><a href="#cb18-511" aria-hidden="true" tabindex="-1"></a>    endo_only_scanb<span class="sc">$</span>NCN.ROR.asT0,</span>
<span id="cb18-512"><a href="#cb18-512" aria-hidden="true" tabindex="-1"></a>    <span class="at">method =</span> <span class="st">"spearman"</span>, </span>
<span id="cb18-513"><a href="#cb18-513" aria-hidden="true" tabindex="-1"></a>    <span class="at">exact =</span> <span class="cn">FALSE</span></span>
<span id="cb18-514"><a href="#cb18-514" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb18-515"><a href="#cb18-515" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-516"><a href="#cb18-516" aria-hidden="true" tabindex="-1"></a>spearman_riskscore_ror</span>
<span id="cb18-517"><a href="#cb18-517" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb18-518"><a href="#cb18-518" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-519"><a href="#cb18-519" aria-hidden="true" tabindex="-1"></a>With a p-value of:</span>
<span id="cb18-520"><a href="#cb18-520" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-523"><a href="#cb18-523" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb18-524"><a href="#cb18-524" aria-hidden="true" tabindex="-1"></a>spearman_riskscore_ror<span class="sc">$</span>p.value</span>
<span id="cb18-525"><a href="#cb18-525" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb18-526"><a href="#cb18-526" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-527"><a href="#cb18-527" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-528"><a href="#cb18-528" aria-hidden="true" tabindex="-1"></a>We now evaluate the risk score only on luminal A patients that </span>
<span id="cb18-529"><a href="#cb18-529" aria-hidden="true" tabindex="-1"></a>received just endocrine therapy.</span>
<span id="cb18-530"><a href="#cb18-530" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-533"><a href="#cb18-533" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb18-534"><a href="#cb18-534" aria-hidden="true" tabindex="-1"></a>risk_score_scanb_luma <span class="ot">&lt;-</span> survival<span class="sc">::</span><span class="fu">coxph</span>(</span>
<span id="cb18-535"><a href="#cb18-535" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.formula</span>(<span class="fu">paste0</span>(</span>
<span id="cb18-536"><a href="#cb18-536" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Surv(time = rfs_months, event = rfs_status) ~ "</span>,</span>
<span id="cb18-537"><a href="#cb18-537" aria-hidden="true" tabindex="-1"></a>        <span class="fu">paste</span>(<span class="fu">c</span>(<span class="st">"risk_score"</span>, clin_vars), <span class="at">collapse =</span> <span class="st">"+"</span>)</span>
<span id="cb18-538"><a href="#cb18-538" aria-hidden="true" tabindex="-1"></a>    )),</span>
<span id="cb18-539"><a href="#cb18-539" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> endo_only_scanb <span class="sc">%&gt;%</span></span>
<span id="cb18-540"><a href="#cb18-540" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">filter</span>(pam50 <span class="sc">==</span> <span class="st">"luma"</span>)</span>
<span id="cb18-541"><a href="#cb18-541" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb18-542"><a href="#cb18-542" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-543"><a href="#cb18-543" aria-hidden="true" tabindex="-1"></a>risk_score_scanb_luma</span>
<span id="cb18-544"><a href="#cb18-544" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb18-545"><a href="#cb18-545" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-546"><a href="#cb18-546" aria-hidden="true" tabindex="-1"></a>Even among these patients the score is prognostic, despite the number of </span>
<span id="cb18-547"><a href="#cb18-547" aria-hidden="true" tabindex="-1"></a>events, the hazard ratio is relatively high here. If we do the survival</span>
<span id="cb18-548"><a href="#cb18-548" aria-hidden="true" tabindex="-1"></a>analysis by using overall survival as endpoint the results are shown below.</span>
<span id="cb18-549"><a href="#cb18-549" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-552"><a href="#cb18-552" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb18-553"><a href="#cb18-553" aria-hidden="true" tabindex="-1"></a>risk_score_scanb_luma_os <span class="ot">&lt;-</span> survival<span class="sc">::</span><span class="fu">coxph</span>(</span>
<span id="cb18-554"><a href="#cb18-554" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.formula</span>(<span class="fu">paste0</span>(</span>
<span id="cb18-555"><a href="#cb18-555" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Surv(time = os_months, event = os_status) ~ "</span>,</span>
<span id="cb18-556"><a href="#cb18-556" aria-hidden="true" tabindex="-1"></a>        <span class="fu">paste</span>(<span class="fu">c</span>(<span class="st">"risk_score"</span>, clin_vars), <span class="at">collapse =</span> <span class="st">"+"</span>)</span>
<span id="cb18-557"><a href="#cb18-557" aria-hidden="true" tabindex="-1"></a>    )),</span>
<span id="cb18-558"><a href="#cb18-558" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> endo_only_scanb <span class="sc">%&gt;%</span></span>
<span id="cb18-559"><a href="#cb18-559" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">filter</span>(pam50 <span class="sc">==</span> <span class="st">"luma"</span>)</span>
<span id="cb18-560"><a href="#cb18-560" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb18-561"><a href="#cb18-561" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-562"><a href="#cb18-562" aria-hidden="true" tabindex="-1"></a>risk_score_scanb_luma_os</span>
<span id="cb18-563"><a href="#cb18-563" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb18-564"><a href="#cb18-564" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-565"><a href="#cb18-565" aria-hidden="true" tabindex="-1"></a>Risk score still an estimated hazard ratio higher than 1, with </span>
<span id="cb18-566"><a href="#cb18-566" aria-hidden="true" tabindex="-1"></a>more variability though, still in the expected direction. So we see that </span>
<span id="cb18-567"><a href="#cb18-567" aria-hidden="true" tabindex="-1"></a>this signature provides aditional information to the PAM50 ROR signature. </span>
<span id="cb18-568"><a href="#cb18-568" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-571"><a href="#cb18-571" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb18-572"><a href="#cb18-572" aria-hidden="true" tabindex="-1"></a>risk_score_scanb_luma_RORonly <span class="ot">&lt;-</span> survival<span class="sc">::</span><span class="fu">coxph</span>(</span>
<span id="cb18-573"><a href="#cb18-573" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.formula</span>(<span class="fu">paste0</span>(</span>
<span id="cb18-574"><a href="#cb18-574" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Surv(time = rfs_months, event = rfs_status) ~ "</span>,</span>
<span id="cb18-575"><a href="#cb18-575" aria-hidden="true" tabindex="-1"></a>        <span class="fu">paste</span>(<span class="fu">c</span>(<span class="st">"NCN.ROR.asT0"</span>, clin_vars), <span class="at">collapse =</span> <span class="st">"+"</span>)</span>
<span id="cb18-576"><a href="#cb18-576" aria-hidden="true" tabindex="-1"></a>    )),</span>
<span id="cb18-577"><a href="#cb18-577" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> endo_only_scanb <span class="sc">%&gt;%</span></span>
<span id="cb18-578"><a href="#cb18-578" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">filter</span>(pam50 <span class="sc">==</span> <span class="st">"luma"</span>)</span>
<span id="cb18-579"><a href="#cb18-579" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb18-580"><a href="#cb18-580" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-581"><a href="#cb18-581" aria-hidden="true" tabindex="-1"></a>risk_score_scanb_luma <span class="ot">&lt;-</span> survival<span class="sc">::</span><span class="fu">coxph</span>(</span>
<span id="cb18-582"><a href="#cb18-582" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.formula</span>(<span class="fu">paste0</span>(</span>
<span id="cb18-583"><a href="#cb18-583" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Surv(time = rfs_months, event = rfs_status) ~ "</span>,</span>
<span id="cb18-584"><a href="#cb18-584" aria-hidden="true" tabindex="-1"></a>        <span class="fu">paste</span>(<span class="fu">c</span>(<span class="st">"risk_score"</span>, </span>
<span id="cb18-585"><a href="#cb18-585" aria-hidden="true" tabindex="-1"></a>                <span class="st">"NCN.ROR.asT0"</span>, clin_vars), <span class="at">collapse =</span> <span class="st">"+"</span>)</span>
<span id="cb18-586"><a href="#cb18-586" aria-hidden="true" tabindex="-1"></a>    )),</span>
<span id="cb18-587"><a href="#cb18-587" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> endo_only_scanb <span class="sc">%&gt;%</span></span>
<span id="cb18-588"><a href="#cb18-588" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">filter</span>(pam50 <span class="sc">==</span> <span class="st">"luma"</span>)</span>
<span id="cb18-589"><a href="#cb18-589" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb18-590"><a href="#cb18-590" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-591"><a href="#cb18-591" aria-hidden="true" tabindex="-1"></a>risk_score_scanb_luma</span>
<span id="cb18-592"><a href="#cb18-592" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb18-593"><a href="#cb18-593" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-594"><a href="#cb18-594" aria-hidden="true" tabindex="-1"></a><span class="fu">## Standardized survival curves for SCANB</span></span>
<span id="cb18-595"><a href="#cb18-595" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-596"><a href="#cb18-596" aria-hidden="true" tabindex="-1"></a>We now use also the standardized survival curves as suggested in the </span>
<span id="cb18-597"><a href="#cb18-597" aria-hidden="true" tabindex="-1"></a>paper <span class="co">[</span><span class="ot">@Syriopoulou2022</span><span class="co">]</span>. For this we use the package</span>
<span id="cb18-598"><a href="#cb18-598" aria-hidden="true" tabindex="-1"></a><span class="in">`flexsurv`</span> along with the functions <span class="in">`flexsurvspline`</span> to model</span>
<span id="cb18-599"><a href="#cb18-599" aria-hidden="true" tabindex="-1"></a>the baseline hazard using splines and then to calculate the </span>
<span id="cb18-600"><a href="#cb18-600" aria-hidden="true" tabindex="-1"></a>standardized survival curves we use the function <span class="in">`standsurv`</span>.</span>
<span id="cb18-601"><a href="#cb18-601" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-602"><a href="#cb18-602" aria-hidden="true" tabindex="-1"></a>For the sake of interpretation we categorize the risk score in 4 categories,</span>
<span id="cb18-603"><a href="#cb18-603" aria-hidden="true" tabindex="-1"></a>a low, intermediate, high and ultra-high risk. The thresholds are </span>
<span id="cb18-604"><a href="#cb18-604" aria-hidden="true" tabindex="-1"></a>based on the data, we use the quartiles. We understand that we loose power</span>
<span id="cb18-605"><a href="#cb18-605" aria-hidden="true" tabindex="-1"></a>when doing this but we are able to communicate better the information. </span>
<span id="cb18-606"><a href="#cb18-606" aria-hidden="true" tabindex="-1"></a>For comparison we first show the spline based model on the continous </span>
<span id="cb18-607"><a href="#cb18-607" aria-hidden="true" tabindex="-1"></a>score.</span>
<span id="cb18-608"><a href="#cb18-608" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-611"><a href="#cb18-611" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb18-612"><a href="#cb18-612" aria-hidden="true" tabindex="-1"></a>fit_scanb_splines <span class="ot">&lt;-</span> flexsurv<span class="sc">::</span><span class="fu">flexsurvspline</span>(</span>
<span id="cb18-613"><a href="#cb18-613" aria-hidden="true" tabindex="-1"></a>    <span class="at">formula =</span> <span class="fu">Surv</span>(rfs_months, rfs_status) <span class="sc">~</span> </span>
<span id="cb18-614"><a href="#cb18-614" aria-hidden="true" tabindex="-1"></a>        risk_score <span class="sc">+</span> tumor_size <span class="sc">+</span> node_status,</span>
<span id="cb18-615"><a href="#cb18-615" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> endo_only_scanb,</span>
<span id="cb18-616"><a href="#cb18-616" aria-hidden="true" tabindex="-1"></a>    <span class="at">k =</span> <span class="dv">3</span>,</span>
<span id="cb18-617"><a href="#cb18-617" aria-hidden="true" tabindex="-1"></a>    <span class="at">scale =</span> <span class="st">"hazard"</span></span>
<span id="cb18-618"><a href="#cb18-618" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb18-619"><a href="#cb18-619" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-620"><a href="#cb18-620" aria-hidden="true" tabindex="-1"></a>broom<span class="sc">::</span><span class="fu">tidy</span>(fit_scanb_splines)</span>
<span id="cb18-621"><a href="#cb18-621" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb18-622"><a href="#cb18-622" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-623"><a href="#cb18-623" aria-hidden="true" tabindex="-1"></a>We use the "hazard" scale to define the proportional hazards model. The</span>
<span id="cb18-624"><a href="#cb18-624" aria-hidden="true" tabindex="-1"></a>estimate is above 0 (1.8), meaning that the higher the risk score the</span>
<span id="cb18-625"><a href="#cb18-625" aria-hidden="true" tabindex="-1"></a>worse it is. Moreover, the p-value is 1.1e-14. </span>
<span id="cb18-626"><a href="#cb18-626" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-627"><a href="#cb18-627" aria-hidden="true" tabindex="-1"></a>Before moving on we just compare the risk score distributions between METABRIC</span>
<span id="cb18-628"><a href="#cb18-628" aria-hidden="true" tabindex="-1"></a>and SCANB.</span>
<span id="cb18-629"><a href="#cb18-629" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-632"><a href="#cb18-632" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb18-633"><a href="#cb18-633" aria-hidden="true" tabindex="-1"></a>quartiles_risk_score_metabric <span class="ot">&lt;-</span> <span class="fu">quantile</span>(</span>
<span id="cb18-634"><a href="#cb18-634" aria-hidden="true" tabindex="-1"></a>    endo_only_metabric<span class="sc">$</span>risk_score, </span>
<span id="cb18-635"><a href="#cb18-635" aria-hidden="true" tabindex="-1"></a>    <span class="at">probs =</span> <span class="fu">c</span>(<span class="dv">0</span>, .<span class="dv">25</span>, .<span class="dv">5</span>, .<span class="dv">75</span>, <span class="dv">1</span>)</span>
<span id="cb18-636"><a href="#cb18-636" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb18-637"><a href="#cb18-637" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-638"><a href="#cb18-638" aria-hidden="true" tabindex="-1"></a>quartiles_risk_score_scanb <span class="ot">&lt;-</span> <span class="fu">quantile</span>(</span>
<span id="cb18-639"><a href="#cb18-639" aria-hidden="true" tabindex="-1"></a>    endo_only_scanb<span class="sc">$</span>risk_score, </span>
<span id="cb18-640"><a href="#cb18-640" aria-hidden="true" tabindex="-1"></a>    <span class="at">probs =</span> <span class="fu">c</span>(<span class="dv">0</span>, .<span class="dv">25</span>, .<span class="dv">5</span>, .<span class="dv">75</span>, <span class="dv">1</span>)</span>
<span id="cb18-641"><a href="#cb18-641" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb18-642"><a href="#cb18-642" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-643"><a href="#cb18-643" aria-hidden="true" tabindex="-1"></a>quartiles_risk_score_all <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb18-644"><a href="#cb18-644" aria-hidden="true" tabindex="-1"></a>    <span class="at">metabric =</span> quartiles_risk_score_metabric,</span>
<span id="cb18-645"><a href="#cb18-645" aria-hidden="true" tabindex="-1"></a>    <span class="at">scanb =</span> quartiles_risk_score_scanb</span>
<span id="cb18-646"><a href="#cb18-646" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb18-647"><a href="#cb18-647" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-648"><a href="#cb18-648" aria-hidden="true" tabindex="-1"></a>col_to_select <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"sample_name"</span>, <span class="st">"risk_score"</span>, <span class="st">"cohort"</span>, <span class="st">"pam50"</span>)</span>
<span id="cb18-649"><a href="#cb18-649" aria-hidden="true" tabindex="-1"></a>risk_score_all <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(</span>
<span id="cb18-650"><a href="#cb18-650" aria-hidden="true" tabindex="-1"></a>    endo_only_metabric[, col_to_select],</span>
<span id="cb18-651"><a href="#cb18-651" aria-hidden="true" tabindex="-1"></a>    endo_only_scanb[, col_to_select]</span>
<span id="cb18-652"><a href="#cb18-652" aria-hidden="true" tabindex="-1"></a>) </span>
<span id="cb18-653"><a href="#cb18-653" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-654"><a href="#cb18-654" aria-hidden="true" tabindex="-1"></a>risk_score_all <span class="sc">%&gt;%</span></span>
<span id="cb18-655"><a href="#cb18-655" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> risk_score, <span class="at">fill =</span> cohort)) <span class="sc">+</span></span>
<span id="cb18-656"><a href="#cb18-656" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_density</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb18-657"><a href="#cb18-657" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>)</span>
<span id="cb18-658"><a href="#cb18-658" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb18-659"><a href="#cb18-659" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-660"><a href="#cb18-660" aria-hidden="true" tabindex="-1"></a>The distribution shift makes sense as the majority of patients in SCANB have</span>
<span id="cb18-661"><a href="#cb18-661" aria-hidden="true" tabindex="-1"></a>lower tumor stage. Let us compare by molecular subtype.</span>
<span id="cb18-662"><a href="#cb18-662" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-663"><a href="#cb18-663" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=16, fig.height=4}</span></span>
<span id="cb18-664"><a href="#cb18-664" aria-hidden="true" tabindex="-1"></a><span class="in">risk_score_all %&gt;% </span></span>
<span id="cb18-665"><a href="#cb18-665" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::filter(pam50 != "claudin-low") %&gt;%</span></span>
<span id="cb18-666"><a href="#cb18-666" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::ggplot(aes(x = risk_score, fill = cohort)) +</span></span>
<span id="cb18-667"><a href="#cb18-667" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_density(alpha = 0.5) +</span></span>
<span id="cb18-668"><a href="#cb18-668" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::facet_wrap(~pam50, nrow = 1, scales = "free_y") + </span></span>
<span id="cb18-669"><a href="#cb18-669" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme_bw(base_size = 20)</span></span>
<span id="cb18-670"><a href="#cb18-670" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb18-671"><a href="#cb18-671" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-672"><a href="#cb18-672" aria-hidden="true" tabindex="-1"></a>In general they agree a bit more when looking by subtype. </span>
<span id="cb18-673"><a href="#cb18-673" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-674"><a href="#cb18-674" aria-hidden="true" tabindex="-1"></a>We now proceed with the categorized data for visualization purposes. We</span>
<span id="cb18-675"><a href="#cb18-675" aria-hidden="true" tabindex="-1"></a>also use the survival scale for the interpretation. For the categories we</span>
<span id="cb18-676"><a href="#cb18-676" aria-hidden="true" tabindex="-1"></a>use METABRIC's quantiles instead of SCANB. This way we used the training </span>
<span id="cb18-677"><a href="#cb18-677" aria-hidden="true" tabindex="-1"></a>dataset for defining the risk scores and then we use SCAN-B to further</span>
<span id="cb18-678"><a href="#cb18-678" aria-hidden="true" tabindex="-1"></a>validate the differences.</span>
<span id="cb18-679"><a href="#cb18-679" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-682"><a href="#cb18-682" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb18-683"><a href="#cb18-683" aria-hidden="true" tabindex="-1"></a>which_cohort_risk_strat <span class="ot">&lt;-</span> <span class="st">"metabric"</span></span>
<span id="cb18-684"><a href="#cb18-684" aria-hidden="true" tabindex="-1"></a>quartiles_risk_score <span class="ot">&lt;-</span> quartiles_risk_score_all[, which_cohort_risk_strat]</span>
<span id="cb18-685"><a href="#cb18-685" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb18-686"><a href="#cb18-686" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-687"><a href="#cb18-687" aria-hidden="true" tabindex="-1"></a>The quartiles are (0, 0.25, 0.5, 0.75 and 1):</span>
<span id="cb18-688"><a href="#cb18-688" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-691"><a href="#cb18-691" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb18-692"><a href="#cb18-692" aria-hidden="true" tabindex="-1"></a>quartiles_risk_score</span>
<span id="cb18-693"><a href="#cb18-693" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb18-694"><a href="#cb18-694" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-695"><a href="#cb18-695" aria-hidden="true" tabindex="-1"></a>The patients are stratified in the categories low, intermediate,</span>
<span id="cb18-696"><a href="#cb18-696" aria-hidden="true" tabindex="-1"></a>high and ultra-high. For the analysis the low category is set as the</span>
<span id="cb18-697"><a href="#cb18-697" aria-hidden="true" tabindex="-1"></a>baseline.</span>
<span id="cb18-698"><a href="#cb18-698" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-701"><a href="#cb18-701" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb18-702"><a href="#cb18-702" aria-hidden="true" tabindex="-1"></a>endo_only_scanb <span class="ot">&lt;-</span> endo_only_scanb <span class="sc">%&gt;%</span></span>
<span id="cb18-703"><a href="#cb18-703" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">risk_score_category =</span> dplyr<span class="sc">::</span><span class="fu">case_when</span>(</span>
<span id="cb18-704"><a href="#cb18-704" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">between</span>(</span>
<span id="cb18-705"><a href="#cb18-705" aria-hidden="true" tabindex="-1"></a>            risk_score, </span>
<span id="cb18-706"><a href="#cb18-706" aria-hidden="true" tabindex="-1"></a>            quartiles_risk_score[<span class="dv">1</span>], </span>
<span id="cb18-707"><a href="#cb18-707" aria-hidden="true" tabindex="-1"></a>            quartiles_risk_score[<span class="dv">2</span>]</span>
<span id="cb18-708"><a href="#cb18-708" aria-hidden="true" tabindex="-1"></a>        ) <span class="sc">~</span> <span class="st">"low"</span>,</span>
<span id="cb18-709"><a href="#cb18-709" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">between</span>(</span>
<span id="cb18-710"><a href="#cb18-710" aria-hidden="true" tabindex="-1"></a>            risk_score, </span>
<span id="cb18-711"><a href="#cb18-711" aria-hidden="true" tabindex="-1"></a>            quartiles_risk_score[<span class="dv">2</span>], </span>
<span id="cb18-712"><a href="#cb18-712" aria-hidden="true" tabindex="-1"></a>            quartiles_risk_score[<span class="dv">3</span>]</span>
<span id="cb18-713"><a href="#cb18-713" aria-hidden="true" tabindex="-1"></a>        ) <span class="sc">~</span> <span class="st">"intermediate"</span>,</span>
<span id="cb18-714"><a href="#cb18-714" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">between</span>(</span>
<span id="cb18-715"><a href="#cb18-715" aria-hidden="true" tabindex="-1"></a>            risk_score, </span>
<span id="cb18-716"><a href="#cb18-716" aria-hidden="true" tabindex="-1"></a>            quartiles_risk_score[<span class="dv">3</span>], </span>
<span id="cb18-717"><a href="#cb18-717" aria-hidden="true" tabindex="-1"></a>            quartiles_risk_score[<span class="dv">4</span>]</span>
<span id="cb18-718"><a href="#cb18-718" aria-hidden="true" tabindex="-1"></a>        ) <span class="sc">~</span> <span class="st">"high"</span>,</span>
<span id="cb18-719"><a href="#cb18-719" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">between</span>(</span>
<span id="cb18-720"><a href="#cb18-720" aria-hidden="true" tabindex="-1"></a>            risk_score, </span>
<span id="cb18-721"><a href="#cb18-721" aria-hidden="true" tabindex="-1"></a>            quartiles_risk_score[<span class="dv">4</span>], </span>
<span id="cb18-722"><a href="#cb18-722" aria-hidden="true" tabindex="-1"></a>            quartiles_risk_score[<span class="dv">5</span>]</span>
<span id="cb18-723"><a href="#cb18-723" aria-hidden="true" tabindex="-1"></a>        ) <span class="sc">~</span> <span class="st">"ultra-high"</span></span>
<span id="cb18-724"><a href="#cb18-724" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">%&gt;%</span></span>
<span id="cb18-725"><a href="#cb18-725" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb18-726"><a href="#cb18-726" aria-hidden="true" tabindex="-1"></a>        <span class="at">risk_score_category =</span> <span class="fu">factor</span>(</span>
<span id="cb18-727"><a href="#cb18-727" aria-hidden="true" tabindex="-1"></a>            risk_score_category, </span>
<span id="cb18-728"><a href="#cb18-728" aria-hidden="true" tabindex="-1"></a>            <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"low"</span>, <span class="st">"intermediate"</span>, <span class="st">"high"</span>, <span class="st">"ultra-high"</span>)</span>
<span id="cb18-729"><a href="#cb18-729" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb18-730"><a href="#cb18-730" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb18-731"><a href="#cb18-731" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-732"><a href="#cb18-732" aria-hidden="true" tabindex="-1"></a>fit_scanb_splines_category <span class="ot">&lt;-</span> flexsurv<span class="sc">::</span><span class="fu">flexsurvspline</span>(</span>
<span id="cb18-733"><a href="#cb18-733" aria-hidden="true" tabindex="-1"></a>    <span class="at">formula =</span> <span class="fu">Surv</span>(rfs_months, rfs_status) <span class="sc">~</span> </span>
<span id="cb18-734"><a href="#cb18-734" aria-hidden="true" tabindex="-1"></a>        risk_score_category <span class="sc">+</span> tumor_size <span class="sc">+</span> node_status,</span>
<span id="cb18-735"><a href="#cb18-735" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> endo_only_scanb,</span>
<span id="cb18-736"><a href="#cb18-736" aria-hidden="true" tabindex="-1"></a>    <span class="at">k =</span> <span class="dv">3</span></span>
<span id="cb18-737"><a href="#cb18-737" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb18-738"><a href="#cb18-738" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-739"><a href="#cb18-739" aria-hidden="true" tabindex="-1"></a>standardized_survival_curves <span class="ot">&lt;-</span> flexsurv<span class="sc">::</span><span class="fu">standsurv</span>(</span>
<span id="cb18-740"><a href="#cb18-740" aria-hidden="true" tabindex="-1"></a>    fit_scanb_splines_category,</span>
<span id="cb18-741"><a href="#cb18-741" aria-hidden="true" tabindex="-1"></a>    <span class="at">type =</span> <span class="st">"survival"</span>,</span>
<span id="cb18-742"><a href="#cb18-742" aria-hidden="true" tabindex="-1"></a>    <span class="at">at =</span> <span class="fu">list</span>(</span>
<span id="cb18-743"><a href="#cb18-743" aria-hidden="true" tabindex="-1"></a>        <span class="fu">list</span>(<span class="at">risk_score_category =</span> <span class="st">"low"</span>),</span>
<span id="cb18-744"><a href="#cb18-744" aria-hidden="true" tabindex="-1"></a>        <span class="fu">list</span>(<span class="at">risk_score_category =</span> <span class="st">"intermediate"</span>),</span>
<span id="cb18-745"><a href="#cb18-745" aria-hidden="true" tabindex="-1"></a>        <span class="fu">list</span>(<span class="at">risk_score_category =</span> <span class="st">"high"</span>),</span>
<span id="cb18-746"><a href="#cb18-746" aria-hidden="true" tabindex="-1"></a>        <span class="fu">list</span>(<span class="at">risk_score_category =</span> <span class="st">"ultra-high"</span>)</span>
<span id="cb18-747"><a href="#cb18-747" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb18-748"><a href="#cb18-748" aria-hidden="true" tabindex="-1"></a>    <span class="at">t =</span> <span class="fu">seq</span>(</span>
<span id="cb18-749"><a href="#cb18-749" aria-hidden="true" tabindex="-1"></a>        <span class="dv">0</span>,</span>
<span id="cb18-750"><a href="#cb18-750" aria-hidden="true" tabindex="-1"></a>        fit_scanb_splines_category<span class="sc">$</span>data<span class="sc">$</span>Y[, <span class="st">"time"</span>] <span class="sc">%&gt;%</span> max,</span>
<span id="cb18-751"><a href="#cb18-751" aria-hidden="true" tabindex="-1"></a>        <span class="at">length=</span><span class="dv">10</span></span>
<span id="cb18-752"><a href="#cb18-752" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb18-753"><a href="#cb18-753" aria-hidden="true" tabindex="-1"></a>    <span class="at">ci =</span> <span class="cn">TRUE</span>,</span>
<span id="cb18-754"><a href="#cb18-754" aria-hidden="true" tabindex="-1"></a>    <span class="at">boot =</span> <span class="cn">FALSE</span></span>
<span id="cb18-755"><a href="#cb18-755" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb18-756"><a href="#cb18-756" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-757"><a href="#cb18-757" aria-hidden="true" tabindex="-1"></a>standardized_survival_curves_contrast <span class="ot">&lt;-</span> flexsurv<span class="sc">::</span><span class="fu">standsurv</span>(</span>
<span id="cb18-758"><a href="#cb18-758" aria-hidden="true" tabindex="-1"></a>    fit_scanb_splines_category,</span>
<span id="cb18-759"><a href="#cb18-759" aria-hidden="true" tabindex="-1"></a>    <span class="at">type =</span> <span class="st">"survival"</span>,</span>
<span id="cb18-760"><a href="#cb18-760" aria-hidden="true" tabindex="-1"></a>    <span class="at">at =</span> <span class="fu">list</span>(</span>
<span id="cb18-761"><a href="#cb18-761" aria-hidden="true" tabindex="-1"></a>        <span class="fu">list</span>(<span class="at">risk_score_category =</span> <span class="st">"low"</span>),</span>
<span id="cb18-762"><a href="#cb18-762" aria-hidden="true" tabindex="-1"></a>        <span class="fu">list</span>(<span class="at">risk_score_category =</span> <span class="st">"intermediate"</span>),</span>
<span id="cb18-763"><a href="#cb18-763" aria-hidden="true" tabindex="-1"></a>        <span class="fu">list</span>(<span class="at">risk_score_category =</span> <span class="st">"high"</span>),</span>
<span id="cb18-764"><a href="#cb18-764" aria-hidden="true" tabindex="-1"></a>        <span class="fu">list</span>(<span class="at">risk_score_category =</span> <span class="st">"ultra-high"</span>)</span>
<span id="cb18-765"><a href="#cb18-765" aria-hidden="true" tabindex="-1"></a>    ), </span>
<span id="cb18-766"><a href="#cb18-766" aria-hidden="true" tabindex="-1"></a>    <span class="at">contrast =</span> <span class="st">"difference"</span>,</span>
<span id="cb18-767"><a href="#cb18-767" aria-hidden="true" tabindex="-1"></a>    <span class="at">t =</span> <span class="fu">seq</span>(</span>
<span id="cb18-768"><a href="#cb18-768" aria-hidden="true" tabindex="-1"></a>        <span class="dv">0</span>,</span>
<span id="cb18-769"><a href="#cb18-769" aria-hidden="true" tabindex="-1"></a>        fit_scanb_splines_category<span class="sc">$</span>data<span class="sc">$</span>Y[, <span class="st">"time"</span>] <span class="sc">%&gt;%</span> max,</span>
<span id="cb18-770"><a href="#cb18-770" aria-hidden="true" tabindex="-1"></a>        <span class="at">length=</span><span class="dv">10</span></span>
<span id="cb18-771"><a href="#cb18-771" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb18-772"><a href="#cb18-772" aria-hidden="true" tabindex="-1"></a>    <span class="at">ci =</span> <span class="cn">TRUE</span>,</span>
<span id="cb18-773"><a href="#cb18-773" aria-hidden="true" tabindex="-1"></a>    <span class="at">boot =</span> <span class="cn">FALSE</span></span>
<span id="cb18-774"><a href="#cb18-774" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb18-775"><a href="#cb18-775" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb18-776"><a href="#cb18-776" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-777"><a href="#cb18-777" aria-hidden="true" tabindex="-1"></a>First we show the marginal survival curves.</span>
<span id="cb18-778"><a href="#cb18-778" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-781"><a href="#cb18-781" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb18-782"><a href="#cb18-782" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-rfs-marginal-curve</span></span>
<span id="cb18-783"><a href="#cb18-783" aria-hidden="true" tabindex="-1"></a>labels_risk_score <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Low"</span>, <span class="st">"Intermediate"</span>, <span class="st">"High"</span>, <span class="st">"Ultra-high"</span>)</span>
<span id="cb18-784"><a href="#cb18-784" aria-hidden="true" tabindex="-1"></a>labels_risk_score_breaks <span class="ot">&lt;-</span> <span class="fu">paste0</span>(</span>
<span id="cb18-785"><a href="#cb18-785" aria-hidden="true" tabindex="-1"></a>    <span class="st">"risk_score_category="</span>,</span>
<span id="cb18-786"><a href="#cb18-786" aria-hidden="true" tabindex="-1"></a>    <span class="fu">c</span>(<span class="st">"low"</span>, <span class="st">"intermediate"</span>, <span class="st">"high"</span>, <span class="st">"ultra-high"</span>)</span>
<span id="cb18-787"><a href="#cb18-787" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb18-788"><a href="#cb18-788" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-789"><a href="#cb18-789" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(standardized_survival_curves, <span class="at">ci =</span> <span class="cn">TRUE</span>) <span class="sc">+</span></span>
<span id="cb18-790"><a href="#cb18-790" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">guides</span>(<span class="at">color=</span><span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb18-791"><a href="#cb18-791" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_color_manual</span>(</span>
<span id="cb18-792"><a href="#cb18-792" aria-hidden="true" tabindex="-1"></a>        <span class="at">labels =</span> labels_risk_score,</span>
<span id="cb18-793"><a href="#cb18-793" aria-hidden="true" tabindex="-1"></a>        <span class="at">values =</span> viridis<span class="sc">::</span><span class="fu">viridis</span>(<span class="at">n =</span> <span class="dv">4</span>),</span>
<span id="cb18-794"><a href="#cb18-794" aria-hidden="true" tabindex="-1"></a>        <span class="at">breaks =</span> labels_risk_score_breaks</span>
<span id="cb18-795"><a href="#cb18-795" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb18-796"><a href="#cb18-796" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_fill_manual</span>(</span>
<span id="cb18-797"><a href="#cb18-797" aria-hidden="true" tabindex="-1"></a>        <span class="at">labels =</span> labels_risk_score,</span>
<span id="cb18-798"><a href="#cb18-798" aria-hidden="true" tabindex="-1"></a>        <span class="at">values =</span> viridis<span class="sc">::</span><span class="fu">viridis</span>(<span class="at">n =</span> <span class="dv">4</span>),</span>
<span id="cb18-799"><a href="#cb18-799" aria-hidden="true" tabindex="-1"></a>        <span class="at">breaks =</span> labels_risk_score_breaks</span>
<span id="cb18-800"><a href="#cb18-800" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb18-801"><a href="#cb18-801" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_x_continuous</span>(</span>
<span id="cb18-802"><a href="#cb18-802" aria-hidden="true" tabindex="-1"></a>        <span class="at">labels =</span> <span class="cf">function</span>(x) x<span class="sc">/</span><span class="dv">12</span>,</span>
<span id="cb18-803"><a href="#cb18-803" aria-hidden="true" tabindex="-1"></a>        <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">200</span>, <span class="at">by =</span> <span class="dv">18</span>)</span>
<span id="cb18-804"><a href="#cb18-804" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb18-805"><a href="#cb18-805" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb18-806"><a href="#cb18-806" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"Time since diagnosis (years)"</span>,</span>
<span id="cb18-807"><a href="#cb18-807" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"Survival probability"</span>,</span>
<span id="cb18-808"><a href="#cb18-808" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">"Risk score</span><span class="sc">\n</span><span class="st">category"</span>,</span>
<span id="cb18-809"><a href="#cb18-809" aria-hidden="true" tabindex="-1"></a>        <span class="at">fill =</span> <span class="st">"Risk score</span><span class="sc">\n</span><span class="st">category"</span></span>
<span id="cb18-810"><a href="#cb18-810" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb18-811"><a href="#cb18-811" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">15</span>) <span class="sc">+</span></span>
<span id="cb18-812"><a href="#cb18-812" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"right"</span>)</span>
<span id="cb18-813"><a href="#cb18-813" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb18-814"><a href="#cb18-814" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-815"><a href="#cb18-815" aria-hidden="true" tabindex="-1"></a>And now the marginal curves for the differences with the low</span>
<span id="cb18-816"><a href="#cb18-816" aria-hidden="true" tabindex="-1"></a>risk as the baseline.</span>
<span id="cb18-817"><a href="#cb18-817" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-820"><a href="#cb18-820" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb18-821"><a href="#cb18-821" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-rfs-difference-marginal-curve</span></span>
<span id="cb18-822"><a href="#cb18-822" aria-hidden="true" tabindex="-1"></a>labels_risk_score <span class="ot">&lt;-</span> <span class="fu">paste0</span>(</span>
<span id="cb18-823"><a href="#cb18-823" aria-hidden="true" tabindex="-1"></a>    <span class="fu">c</span>(<span class="st">"Intermediate"</span>, <span class="st">"High"</span>, <span class="st">"Ultra-high"</span>),</span>
<span id="cb18-824"><a href="#cb18-824" aria-hidden="true" tabindex="-1"></a>    <span class="st">" - Low"</span></span>
<span id="cb18-825"><a href="#cb18-825" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb18-826"><a href="#cb18-826" aria-hidden="true" tabindex="-1"></a>labels_risk_score_breaks <span class="ot">&lt;-</span> <span class="fu">paste0</span>(</span>
<span id="cb18-827"><a href="#cb18-827" aria-hidden="true" tabindex="-1"></a>    <span class="st">"risk_score_category="</span>,</span>
<span id="cb18-828"><a href="#cb18-828" aria-hidden="true" tabindex="-1"></a>    <span class="fu">c</span>(<span class="st">"intermediate"</span>, <span class="st">"high"</span>, <span class="st">"ultra-high"</span>),</span>
<span id="cb18-829"><a href="#cb18-829" aria-hidden="true" tabindex="-1"></a>    <span class="st">" vs risk_score_category=low"</span></span>
<span id="cb18-830"><a href="#cb18-830" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb18-831"><a href="#cb18-831" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-832"><a href="#cb18-832" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(standardized_survival_curves_contrast, <span class="at">ci =</span> <span class="cn">TRUE</span>, <span class="at">contrast =</span> <span class="cn">TRUE</span>) <span class="sc">+</span></span>
<span id="cb18-833"><a href="#cb18-833" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">guides</span>(<span class="at">color=</span><span class="fu">guide_legend</span>(<span class="at">ncol=</span><span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb18-834"><a href="#cb18-834" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_color_manual</span>(</span>
<span id="cb18-835"><a href="#cb18-835" aria-hidden="true" tabindex="-1"></a>        <span class="at">labels =</span> labels_risk_score,</span>
<span id="cb18-836"><a href="#cb18-836" aria-hidden="true" tabindex="-1"></a>        <span class="at">values =</span> viridis<span class="sc">::</span><span class="fu">viridis</span>(<span class="at">n =</span> <span class="dv">3</span>),</span>
<span id="cb18-837"><a href="#cb18-837" aria-hidden="true" tabindex="-1"></a>        <span class="at">breaks =</span> labels_risk_score_breaks</span>
<span id="cb18-838"><a href="#cb18-838" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb18-839"><a href="#cb18-839" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_fill_manual</span>(</span>
<span id="cb18-840"><a href="#cb18-840" aria-hidden="true" tabindex="-1"></a>        <span class="at">labels =</span> labels_risk_score,</span>
<span id="cb18-841"><a href="#cb18-841" aria-hidden="true" tabindex="-1"></a>        <span class="at">values =</span> viridis<span class="sc">::</span><span class="fu">viridis</span>(<span class="at">n =</span> <span class="dv">3</span>),</span>
<span id="cb18-842"><a href="#cb18-842" aria-hidden="true" tabindex="-1"></a>        <span class="at">breaks =</span> labels_risk_score_breaks</span>
<span id="cb18-843"><a href="#cb18-843" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb18-844"><a href="#cb18-844" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_x_continuous</span>(</span>
<span id="cb18-845"><a href="#cb18-845" aria-hidden="true" tabindex="-1"></a>        <span class="at">labels =</span> <span class="cf">function</span>(x) x<span class="sc">/</span><span class="dv">12</span>,</span>
<span id="cb18-846"><a href="#cb18-846" aria-hidden="true" tabindex="-1"></a>        <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">200</span>, <span class="at">by =</span> <span class="dv">18</span>)</span>
<span id="cb18-847"><a href="#cb18-847" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb18-848"><a href="#cb18-848" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb18-849"><a href="#cb18-849" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"Time since diagnosis (years)"</span>,</span>
<span id="cb18-850"><a href="#cb18-850" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"Difference in survival probabilities"</span>,</span>
<span id="cb18-851"><a href="#cb18-851" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">"Difference of</span><span class="sc">\n</span><span class="st">risk category"</span>,</span>
<span id="cb18-852"><a href="#cb18-852" aria-hidden="true" tabindex="-1"></a>        <span class="at">fill =</span> <span class="st">"Difference of</span><span class="sc">\n</span><span class="st">risk category"</span></span>
<span id="cb18-853"><a href="#cb18-853" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb18-854"><a href="#cb18-854" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">15</span>) <span class="sc">+</span></span>
<span id="cb18-855"><a href="#cb18-855" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"right"</span>)</span>
<span id="cb18-856"><a href="#cb18-856" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb18-857"><a href="#cb18-857" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-858"><a href="#cb18-858" aria-hidden="true" tabindex="-1"></a>Here we see that the difference between low and ultra-high can be</span>
<span id="cb18-859"><a href="#cb18-859" aria-hidden="true" tabindex="-1"></a>as high as 40%. On the other hand the differences</span>
<span id="cb18-860"><a href="#cb18-860" aria-hidden="true" tabindex="-1"></a>between intermediate and low stay around 3 to 10% after 5 years.</span>
<span id="cb18-861"><a href="#cb18-861" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-862"><a href="#cb18-862" aria-hidden="true" tabindex="-1"></a>The confidence intervals are larger as you increase the risk groups</span>
<span id="cb18-863"><a href="#cb18-863" aria-hidden="true" tabindex="-1"></a>due to the number of patients available in each category.</span>
<span id="cb18-864"><a href="#cb18-864" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-867"><a href="#cb18-867" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb18-868"><a href="#cb18-868" aria-hidden="true" tabindex="-1"></a>endo_only_scanb <span class="sc">%&gt;%</span></span>
<span id="cb18-869"><a href="#cb18-869" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(risk_score_category)) <span class="sc">%&gt;%</span></span>
<span id="cb18-870"><a href="#cb18-870" aria-hidden="true" tabindex="-1"></a>    janitor<span class="sc">::</span><span class="fu">tabyl</span>(risk_score_category, rfs_status) <span class="sc">%&gt;%</span></span>
<span id="cb18-871"><a href="#cb18-871" aria-hidden="true" tabindex="-1"></a>    janitor<span class="sc">::</span><span class="fu">adorn_totals</span>(<span class="fu">c</span>(<span class="st">"row"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb18-872"><a href="#cb18-872" aria-hidden="true" tabindex="-1"></a>    janitor<span class="sc">::</span><span class="fu">adorn_percentages</span>(<span class="st">"row"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb18-873"><a href="#cb18-873" aria-hidden="true" tabindex="-1"></a>    janitor<span class="sc">::</span><span class="fu">adorn_pct_formatting</span>(<span class="at">rounding =</span> <span class="st">"half up"</span>, <span class="at">digits =</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb18-874"><a href="#cb18-874" aria-hidden="true" tabindex="-1"></a>    janitor<span class="sc">::</span><span class="fu">adorn_ns</span>() <span class="sc">%&gt;%</span></span>
<span id="cb18-875"><a href="#cb18-875" aria-hidden="true" tabindex="-1"></a>    kableExtra<span class="sc">::</span><span class="fu">kbl</span>() <span class="sc">%&gt;%</span></span>
<span id="cb18-876"><a href="#cb18-876" aria-hidden="true" tabindex="-1"></a>    kableExtra<span class="sc">::</span><span class="fu">kable_classic</span>(<span class="at">full_width =</span> <span class="cn">FALSE</span>)</span>
<span id="cb18-877"><a href="#cb18-877" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb18-878"><a href="#cb18-878" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-879"><a href="#cb18-879" aria-hidden="true" tabindex="-1"></a>Still the number of events in percentage goes up as one goes up in the</span>
<span id="cb18-880"><a href="#cb18-880" aria-hidden="true" tabindex="-1"></a>risk category, as expected.</span>
<span id="cb18-881"><a href="#cb18-881" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-882"><a href="#cb18-882" aria-hidden="true" tabindex="-1"></a><span class="fu">## Risk score and POETIC trial</span></span>
<span id="cb18-883"><a href="#cb18-883" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-884"><a href="#cb18-884" aria-hidden="true" tabindex="-1"></a>We can evaluate one of the two risk scores, the one using only the </span>
<span id="cb18-885"><a href="#cb18-885" aria-hidden="true" tabindex="-1"></a>principal components, in the POETIC trial data. The first question we</span>
<span id="cb18-886"><a href="#cb18-886" aria-hidden="true" tabindex="-1"></a>can address is if there is difference in scores in the baseline tissue </span>
<span id="cb18-887"><a href="#cb18-887" aria-hidden="true" tabindex="-1"></a>(@fig-poetic-risk-score).</span>
<span id="cb18-888"><a href="#cb18-888" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-891"><a href="#cb18-891" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb18-892"><a href="#cb18-892" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-poetic-risk-score</span></span>
<span id="cb18-893"><a href="#cb18-893" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Risk scores for responders and non-responders in the baseline</span></span>
<span id="cb18-894"><a href="#cb18-894" aria-hidden="true" tabindex="-1"></a><span class="co">#|     tumor tissue. </span></span>
<span id="cb18-895"><a href="#cb18-895" aria-hidden="true" tabindex="-1"></a>df_pca <span class="sc">%&gt;%</span></span>
<span id="cb18-896"><a href="#cb18-896" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb18-897"><a href="#cb18-897" aria-hidden="true" tabindex="-1"></a>        cohort <span class="sc">==</span> <span class="st">"poetic"</span> <span class="sc">&amp;</span> </span>
<span id="cb18-898"><a href="#cb18-898" aria-hidden="true" tabindex="-1"></a>            timepoint <span class="sc">==</span> <span class="st">"baseline"</span> <span class="sc">&amp;</span> </span>
<span id="cb18-899"><a href="#cb18-899" aria-hidden="true" tabindex="-1"></a>            is_responder <span class="sc">!=</span> <span class="st">"not_available"</span></span>
<span id="cb18-900"><a href="#cb18-900" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb18-901"><a href="#cb18-901" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> is_responder, <span class="at">y =</span> risk_score_pcs)) <span class="sc">+</span></span>
<span id="cb18-902"><a href="#cb18-902" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_violin</span>(<span class="at">alpha =</span> <span class="fl">0.3</span>) <span class="sc">+</span> </span>
<span id="cb18-903"><a href="#cb18-903" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_jitter</span>() <span class="sc">+</span> </span>
<span id="cb18-904"><a href="#cb18-904" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb18-905"><a href="#cb18-905" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"Response status"</span>,</span>
<span id="cb18-906"><a href="#cb18-906" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"Risk score (PCs only)"</span>,</span>
<span id="cb18-907"><a href="#cb18-907" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">"Risk score (PCs only) based on</span><span class="sc">\n</span><span class="st">response status"</span></span>
<span id="cb18-908"><a href="#cb18-908" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb18-909"><a href="#cb18-909" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>) <span class="sc">+</span></span>
<span id="cb18-910"><a href="#cb18-910" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>()</span>
<span id="cb18-911"><a href="#cb18-911" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb18-912"><a href="#cb18-912" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-913"><a href="#cb18-913" aria-hidden="true" tabindex="-1"></a>We see in the @fig-poetic-risk-score that in average the non responders</span>
<span id="cb18-914"><a href="#cb18-914" aria-hidden="true" tabindex="-1"></a>have higher risk scores. Now if we compare the risk scores between</span>
<span id="cb18-915"><a href="#cb18-915" aria-hidden="true" tabindex="-1"></a>baseline and surgery, the slope of the line that fits for each group</span>
<span id="cb18-916"><a href="#cb18-916" aria-hidden="true" tabindex="-1"></a>separately, responder and non responder, is different. The slope</span>
<span id="cb18-917"><a href="#cb18-917" aria-hidden="true" tabindex="-1"></a>for the non responders is parallel to the identity line going through</span>
<span id="cb18-918"><a href="#cb18-918" aria-hidden="true" tabindex="-1"></a>the origin with a slope of 1, meaning that there is not much change of risk</span>
<span id="cb18-919"><a href="#cb18-919" aria-hidden="true" tabindex="-1"></a>upon the treatment. On other hand, for the responder group the slope is</span>
<span id="cb18-920"><a href="#cb18-920" aria-hidden="true" tabindex="-1"></a>decreased and has a value lower than 1, meaning that the risk is decreased</span>
<span id="cb18-921"><a href="#cb18-921" aria-hidden="true" tabindex="-1"></a>upon aromatase inhibition after two weeks (@fig-poetic-risk-base-surgery).</span>
<span id="cb18-922"><a href="#cb18-922" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-925"><a href="#cb18-925" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb18-926"><a href="#cb18-926" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-poetic-risk-base-surgery</span></span>
<span id="cb18-927"><a href="#cb18-927" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Comparison of risk scores using the PCs only before and after</span></span>
<span id="cb18-928"><a href="#cb18-928" aria-hidden="true" tabindex="-1"></a><span class="co">#|     two weeks of aromatase inhibitors in post-menopausal women. The colored</span></span>
<span id="cb18-929"><a href="#cb18-929" aria-hidden="true" tabindex="-1"></a><span class="co">#|     lines correspond do the linear regression for each response status</span></span>
<span id="cb18-930"><a href="#cb18-930" aria-hidden="true" tabindex="-1"></a><span class="co">#|     group separately. </span></span>
<span id="cb18-931"><a href="#cb18-931" aria-hidden="true" tabindex="-1"></a>df_pca <span class="sc">%&gt;%</span></span>
<span id="cb18-932"><a href="#cb18-932" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb18-933"><a href="#cb18-933" aria-hidden="true" tabindex="-1"></a>        cohort <span class="sc">==</span> <span class="st">"poetic"</span> <span class="sc">&amp;</span> </span>
<span id="cb18-934"><a href="#cb18-934" aria-hidden="true" tabindex="-1"></a>            is_responder <span class="sc">!=</span> <span class="st">"not_available"</span></span>
<span id="cb18-935"><a href="#cb18-935" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb18-936"><a href="#cb18-936" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">pivot_wider</span>(</span>
<span id="cb18-937"><a href="#cb18-937" aria-hidden="true" tabindex="-1"></a>        <span class="at">id_cols =</span> <span class="fu">c</span>(<span class="st">"patient_nb"</span>, <span class="st">"is_responder"</span>),</span>
<span id="cb18-938"><a href="#cb18-938" aria-hidden="true" tabindex="-1"></a>        <span class="at">names_from =</span> <span class="st">"timepoint"</span>,</span>
<span id="cb18-939"><a href="#cb18-939" aria-hidden="true" tabindex="-1"></a>        <span class="at">values_from =</span> <span class="st">"risk_score_pcs"</span></span>
<span id="cb18-940"><a href="#cb18-940" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb18-941"><a href="#cb18-941" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> baseline, <span class="at">y =</span> surgery, <span class="at">color =</span> is_responder)) <span class="sc">+</span></span>
<span id="cb18-942"><a href="#cb18-942" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb18-943"><a href="#cb18-943" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_abline</span>(<span class="at">slope =</span> <span class="dv">1</span>, <span class="at">intercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span> </span>
<span id="cb18-944"><a href="#cb18-944" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">formula =</span> <span class="st">"y ~ x"</span>) <span class="sc">+</span></span>
<span id="cb18-945"><a href="#cb18-945" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb18-946"><a href="#cb18-946" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"Baseline risk score"</span>,</span>
<span id="cb18-947"><a href="#cb18-947" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"Surgery risk score"</span>,</span>
<span id="cb18-948"><a href="#cb18-948" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">"Risk score (PCs only)"</span>,</span>
<span id="cb18-949"><a href="#cb18-949" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">"Is responder?"</span></span>
<span id="cb18-950"><a href="#cb18-950" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb18-951"><a href="#cb18-951" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_color_viridis_d</span>() <span class="sc">+</span> </span>
<span id="cb18-952"><a href="#cb18-952" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>) <span class="sc">+</span></span>
<span id="cb18-953"><a href="#cb18-953" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>()</span>
<span id="cb18-954"><a href="#cb18-954" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb18-955"><a href="#cb18-955" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-956"><a href="#cb18-956" aria-hidden="true" tabindex="-1"></a><span class="fu">## Risk score vs pathways </span></span>
<span id="cb18-957"><a href="#cb18-957" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-958"><a href="#cb18-958" aria-hidden="true" tabindex="-1"></a>We now compare the risk score with pathways of interest as discussed in the</span>
<span id="cb18-959"><a href="#cb18-959" aria-hidden="true" tabindex="-1"></a>previous sections. Again we only look at the patients that received only</span>
<span id="cb18-960"><a href="#cb18-960" aria-hidden="true" tabindex="-1"></a>endocrine therapy in both SCANB and METABRIC.</span>
<span id="cb18-961"><a href="#cb18-961" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-964"><a href="#cb18-964" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb18-965"><a href="#cb18-965" aria-hidden="true" tabindex="-1"></a>pathways_of_interest <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb18-966"><a href="#cb18-966" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_G2M_CHECKPOINT"</span> <span class="ot">=</span> <span class="st">"G2M checkpoint"</span>,</span>
<span id="cb18-967"><a href="#cb18-967" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION"</span> <span class="ot">=</span> <span class="st">"EMT"</span>,</span>
<span id="cb18-968"><a href="#cb18-968" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_DNA_REPAIR"</span> <span class="ot">=</span> <span class="st">"DNA repair"</span>,</span>
<span id="cb18-969"><a href="#cb18-969" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_ANDROGEN_RESPONSE"</span> <span class="ot">=</span> <span class="st">"Androgen response"</span>,</span>
<span id="cb18-970"><a href="#cb18-970" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_PI3K_AKT_MTOR_SIGNALING"</span> <span class="ot">=</span> <span class="st">"PI3K AKT MTOR signaling"</span>,</span>
<span id="cb18-971"><a href="#cb18-971" aria-hidden="true" tabindex="-1"></a>    <span class="st">"SET_ERPR"</span> <span class="ot">=</span> <span class="st">"SET ER/PR"</span>,</span>
<span id="cb18-972"><a href="#cb18-972" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_ESTROGEN_RESPONSE_EARLY"</span> <span class="ot">=</span> <span class="st">"Estrogen early"</span></span>
<span id="cb18-973"><a href="#cb18-973" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb18-974"><a href="#cb18-974" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-975"><a href="#cb18-975" aria-hidden="true" tabindex="-1"></a>endo_only_sm <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(</span>
<span id="cb18-976"><a href="#cb18-976" aria-hidden="true" tabindex="-1"></a>    endo_only_scanb,</span>
<span id="cb18-977"><a href="#cb18-977" aria-hidden="true" tabindex="-1"></a>    endo_only_metabric</span>
<span id="cb18-978"><a href="#cb18-978" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> </span>
<span id="cb18-979"><a href="#cb18-979" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(dplyr<span class="sc">::</span><span class="fu">all_of</span>(<span class="fu">c</span>(</span>
<span id="cb18-980"><a href="#cb18-980" aria-hidden="true" tabindex="-1"></a>        <span class="fu">names</span>(pathways_of_interest),</span>
<span id="cb18-981"><a href="#cb18-981" aria-hidden="true" tabindex="-1"></a>        <span class="st">"cohort"</span>,</span>
<span id="cb18-982"><a href="#cb18-982" aria-hidden="true" tabindex="-1"></a>        <span class="st">"pam50"</span>,</span>
<span id="cb18-983"><a href="#cb18-983" aria-hidden="true" tabindex="-1"></a>        <span class="st">"risk_score"</span>,</span>
<span id="cb18-984"><a href="#cb18-984" aria-hidden="true" tabindex="-1"></a>        <span class="st">"risk_score_pcs"</span></span>
<span id="cb18-985"><a href="#cb18-985" aria-hidden="true" tabindex="-1"></a>    )))</span>
<span id="cb18-986"><a href="#cb18-986" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb18-987"><a href="#cb18-987" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-988"><a href="#cb18-988" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=12, fig.height=10}</span></span>
<span id="cb18-989"><a href="#cb18-989" aria-hidden="true" tabindex="-1"></a><span class="in">#| label: fig-cor-risk-pathways</span></span>
<span id="cb18-990"><a href="#cb18-990" aria-hidden="true" tabindex="-1"></a><span class="in">#| fig-cap: Risk score versus pathways of interest. Each subplot corresponds</span></span>
<span id="cb18-991"><a href="#cb18-991" aria-hidden="true" tabindex="-1"></a><span class="in">#|     to a different pathway. </span></span>
<span id="cb18-992"><a href="#cb18-992" aria-hidden="true" tabindex="-1"></a><span class="in">endo_only_sm %&gt;%</span></span>
<span id="cb18-993"><a href="#cb18-993" aria-hidden="true" tabindex="-1"></a><span class="in">    tidyr::pivot_longer(</span></span>
<span id="cb18-994"><a href="#cb18-994" aria-hidden="true" tabindex="-1"></a><span class="in">        cols = names(pathways_of_interest),</span></span>
<span id="cb18-995"><a href="#cb18-995" aria-hidden="true" tabindex="-1"></a><span class="in">        names_to = "pathway",</span></span>
<span id="cb18-996"><a href="#cb18-996" aria-hidden="true" tabindex="-1"></a><span class="in">        values_to = "score"</span></span>
<span id="cb18-997"><a href="#cb18-997" aria-hidden="true" tabindex="-1"></a><span class="in">    ) %&gt;%</span></span>
<span id="cb18-998"><a href="#cb18-998" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::ggplot(aes(x = risk_score, y = score)) +</span></span>
<span id="cb18-999"><a href="#cb18-999" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_point() +</span></span>
<span id="cb18-1000"><a href="#cb18-1000" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::scale_color_manual(</span></span>
<span id="cb18-1001"><a href="#cb18-1001" aria-hidden="true" tabindex="-1"></a><span class="in">        values = get_colors_pam50(endo_only_sm)</span></span>
<span id="cb18-1002"><a href="#cb18-1002" aria-hidden="true" tabindex="-1"></a><span class="in">    ) +</span></span>
<span id="cb18-1003"><a href="#cb18-1003" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::facet_wrap(</span></span>
<span id="cb18-1004"><a href="#cb18-1004" aria-hidden="true" tabindex="-1"></a><span class="in">        ~pathway,</span></span>
<span id="cb18-1005"><a href="#cb18-1005" aria-hidden="true" tabindex="-1"></a><span class="in">        labeller = as_labeller(pathways_of_interest),</span></span>
<span id="cb18-1006"><a href="#cb18-1006" aria-hidden="true" tabindex="-1"></a><span class="in">        scales = "free_y"</span></span>
<span id="cb18-1007"><a href="#cb18-1007" aria-hidden="true" tabindex="-1"></a><span class="in">    ) + </span></span>
<span id="cb18-1008"><a href="#cb18-1008" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::labs(</span></span>
<span id="cb18-1009"><a href="#cb18-1009" aria-hidden="true" tabindex="-1"></a><span class="in">        x = "Risk score",</span></span>
<span id="cb18-1010"><a href="#cb18-1010" aria-hidden="true" tabindex="-1"></a><span class="in">        y = "Pathway score"</span></span>
<span id="cb18-1011"><a href="#cb18-1011" aria-hidden="true" tabindex="-1"></a><span class="in">    ) +</span></span>
<span id="cb18-1012"><a href="#cb18-1012" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme_bw(base_size = 20) +</span></span>
<span id="cb18-1013"><a href="#cb18-1013" aria-hidden="true" tabindex="-1"></a><span class="in">    change_plot_aes_point()</span></span>
<span id="cb18-1014"><a href="#cb18-1014" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb18-1015"><a href="#cb18-1015" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-1016"><a href="#cb18-1016" aria-hidden="true" tabindex="-1"></a>@fig-cor-risk-pathways shows that in general the signatures are not </span>
<span id="cb18-1017"><a href="#cb18-1017" aria-hidden="true" tabindex="-1"></a>strongly associated with one of the main pathways displayed here.</span>
<span id="cb18-1018"><a href="#cb18-1018" aria-hidden="true" tabindex="-1"></a>On the other hand, @fig-cor-riskpcs-pathways shows the association between</span>
<span id="cb18-1019"><a href="#cb18-1019" aria-hidden="true" tabindex="-1"></a>the signatures and the risk scores derived only from the principal components.</span>
<span id="cb18-1020"><a href="#cb18-1020" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-1021"><a href="#cb18-1021" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=12, fig.height=10}</span></span>
<span id="cb18-1022"><a href="#cb18-1022" aria-hidden="true" tabindex="-1"></a><span class="in">#| label: fig-cor-riskpcs-pathways</span></span>
<span id="cb18-1023"><a href="#cb18-1023" aria-hidden="true" tabindex="-1"></a><span class="in">#| fig-cap: Risk score derived from the principal components only</span></span>
<span id="cb18-1024"><a href="#cb18-1024" aria-hidden="true" tabindex="-1"></a><span class="in">#|     versus pathways of interest. Each subplot corresponds</span></span>
<span id="cb18-1025"><a href="#cb18-1025" aria-hidden="true" tabindex="-1"></a><span class="in">#|     to a different pathway. </span></span>
<span id="cb18-1026"><a href="#cb18-1026" aria-hidden="true" tabindex="-1"></a><span class="in">endo_only_sm %&gt;%</span></span>
<span id="cb18-1027"><a href="#cb18-1027" aria-hidden="true" tabindex="-1"></a><span class="in">    tidyr::pivot_longer(</span></span>
<span id="cb18-1028"><a href="#cb18-1028" aria-hidden="true" tabindex="-1"></a><span class="in">        cols = c(names(pathways_of_interest)),</span></span>
<span id="cb18-1029"><a href="#cb18-1029" aria-hidden="true" tabindex="-1"></a><span class="in">        names_to = "pathway",</span></span>
<span id="cb18-1030"><a href="#cb18-1030" aria-hidden="true" tabindex="-1"></a><span class="in">        values_to = "score"</span></span>
<span id="cb18-1031"><a href="#cb18-1031" aria-hidden="true" tabindex="-1"></a><span class="in">    ) %&gt;%</span></span>
<span id="cb18-1032"><a href="#cb18-1032" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::ggplot(aes(x = risk_score_pcs, y = score)) +</span></span>
<span id="cb18-1033"><a href="#cb18-1033" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_point() +</span></span>
<span id="cb18-1034"><a href="#cb18-1034" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::scale_color_manual(</span></span>
<span id="cb18-1035"><a href="#cb18-1035" aria-hidden="true" tabindex="-1"></a><span class="in">        values = get_colors_pam50(endo_only_sm)</span></span>
<span id="cb18-1036"><a href="#cb18-1036" aria-hidden="true" tabindex="-1"></a><span class="in">    ) +</span></span>
<span id="cb18-1037"><a href="#cb18-1037" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::facet_wrap(</span></span>
<span id="cb18-1038"><a href="#cb18-1038" aria-hidden="true" tabindex="-1"></a><span class="in">        ~pathway,</span></span>
<span id="cb18-1039"><a href="#cb18-1039" aria-hidden="true" tabindex="-1"></a><span class="in">        labeller = as_labeller(pathways_of_interest),</span></span>
<span id="cb18-1040"><a href="#cb18-1040" aria-hidden="true" tabindex="-1"></a><span class="in">        scales = "free_y"</span></span>
<span id="cb18-1041"><a href="#cb18-1041" aria-hidden="true" tabindex="-1"></a><span class="in">    ) + </span></span>
<span id="cb18-1042"><a href="#cb18-1042" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::labs(</span></span>
<span id="cb18-1043"><a href="#cb18-1043" aria-hidden="true" tabindex="-1"></a><span class="in">        x = "Risk score (principal components only)",</span></span>
<span id="cb18-1044"><a href="#cb18-1044" aria-hidden="true" tabindex="-1"></a><span class="in">        y = "Pathway score"</span></span>
<span id="cb18-1045"><a href="#cb18-1045" aria-hidden="true" tabindex="-1"></a><span class="in">    ) +</span></span>
<span id="cb18-1046"><a href="#cb18-1046" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme_bw(base_size = 20) +</span></span>
<span id="cb18-1047"><a href="#cb18-1047" aria-hidden="true" tabindex="-1"></a><span class="in">    change_plot_aes_point()</span></span>
<span id="cb18-1048"><a href="#cb18-1048" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb18-1049"><a href="#cb18-1049" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-1050"><a href="#cb18-1050" aria-hidden="true" tabindex="-1"></a>In this case we see some associations with G2M checkpoint in general and a bit</span>
<span id="cb18-1051"><a href="#cb18-1051" aria-hidden="true" tabindex="-1"></a>with estrogen signaling. We now calculate the correlation plots </span>
<span id="cb18-1052"><a href="#cb18-1052" aria-hidden="true" tabindex="-1"></a>for each one of these pathways with both risk scores. </span>
<span id="cb18-1053"><a href="#cb18-1053" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-1056"><a href="#cb18-1056" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb18-1057"><a href="#cb18-1057" aria-hidden="true" tabindex="-1"></a>corr_values <span class="ot">&lt;-</span> <span class="fu">cor</span>(endo_only_sm[</span>
<span id="cb18-1058"><a href="#cb18-1058" aria-hidden="true" tabindex="-1"></a>    , </span>
<span id="cb18-1059"><a href="#cb18-1059" aria-hidden="true" tabindex="-1"></a>    <span class="fu">c</span>(<span class="st">"risk_score"</span>, <span class="st">"risk_score_pcs"</span>, <span class="fu">names</span>(pathways_of_interest))</span>
<span id="cb18-1060"><a href="#cb18-1060" aria-hidden="true" tabindex="-1"></a>], <span class="at">method =</span> <span class="st">"spearman"</span>) </span>
<span id="cb18-1061"><a href="#cb18-1061" aria-hidden="true" tabindex="-1"></a>names_corr <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb18-1062"><a href="#cb18-1062" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Risk Score"</span>, <span class="st">"Risk Score (PCs only)"</span>, </span>
<span id="cb18-1063"><a href="#cb18-1063" aria-hidden="true" tabindex="-1"></a>    pathways_of_interest[<span class="fu">rownames</span>(corr_values)[<span class="dv">3</span><span class="sc">:</span><span class="fu">nrow</span>(corr_values)]]</span>
<span id="cb18-1064"><a href="#cb18-1064" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb18-1065"><a href="#cb18-1065" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(corr_values) <span class="ot">&lt;-</span> names_corr</span>
<span id="cb18-1066"><a href="#cb18-1066" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(corr_values) <span class="ot">&lt;-</span> names_corr</span>
<span id="cb18-1067"><a href="#cb18-1067" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb18-1068"><a href="#cb18-1068" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-1069"><a href="#cb18-1069" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=8}</span></span>
<span id="cb18-1070"><a href="#cb18-1070" aria-hidden="true" tabindex="-1"></a><span class="in">#| label: fig-corr-risk-pathways</span></span>
<span id="cb18-1071"><a href="#cb18-1071" aria-hidden="true" tabindex="-1"></a><span class="in">#| fig-cap: Correlation matrix for all pathways of interest and the two </span></span>
<span id="cb18-1072"><a href="#cb18-1072" aria-hidden="true" tabindex="-1"></a><span class="in">#|     risk scores, both including the clinical factors and without the </span></span>
<span id="cb18-1073"><a href="#cb18-1073" aria-hidden="true" tabindex="-1"></a><span class="in">#|     clinical factors.</span></span>
<span id="cb18-1074"><a href="#cb18-1074" aria-hidden="true" tabindex="-1"></a><span class="in">ggcorrplot::ggcorrplot(</span></span>
<span id="cb18-1075"><a href="#cb18-1075" aria-hidden="true" tabindex="-1"></a><span class="in">    corr_values,</span></span>
<span id="cb18-1076"><a href="#cb18-1076" aria-hidden="true" tabindex="-1"></a><span class="in">    #hc.order = TRUE, </span></span>
<span id="cb18-1077"><a href="#cb18-1077" aria-hidden="true" tabindex="-1"></a><span class="in">    type = "lower",</span></span>
<span id="cb18-1078"><a href="#cb18-1078" aria-hidden="true" tabindex="-1"></a><span class="in">    lab = TRUE, </span></span>
<span id="cb18-1079"><a href="#cb18-1079" aria-hidden="true" tabindex="-1"></a><span class="in">    lab_size = 6, </span></span>
<span id="cb18-1080"><a href="#cb18-1080" aria-hidden="true" tabindex="-1"></a><span class="in">    ggtheme = ggplot2::theme_bw(base_size = 15), </span></span>
<span id="cb18-1081"><a href="#cb18-1081" aria-hidden="true" tabindex="-1"></a><span class="in">    tl.cex = 20</span></span>
<span id="cb18-1082"><a href="#cb18-1082" aria-hidden="true" tabindex="-1"></a><span class="in">) + </span></span>
<span id="cb18-1083"><a href="#cb18-1083" aria-hidden="true" tabindex="-1"></a><span class="in">    change_plot_aes_point() + </span></span>
<span id="cb18-1084"><a href="#cb18-1084" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::labs(x = "", y = "")</span></span>
<span id="cb18-1085"><a href="#cb18-1085" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb18-1086"><a href="#cb18-1086" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-1087"><a href="#cb18-1087" aria-hidden="true" tabindex="-1"></a>@fig-corr-risk-pathways shows that G2M checkpoint is highly correlated</span>
<span id="cb18-1088"><a href="#cb18-1088" aria-hidden="true" tabindex="-1"></a>with the risk score using only the principal components ($\rho = 0.72$). </span>
<span id="cb18-1089"><a href="#cb18-1089" aria-hidden="true" tabindex="-1"></a>When you include the clinical factors this value decreases. Moreover,</span>
<span id="cb18-1090"><a href="#cb18-1090" aria-hidden="true" tabindex="-1"></a>DNA repair is highly associated with RS PC. Estrogen signaling was not </span>
<span id="cb18-1091"><a href="#cb18-1091" aria-hidden="true" tabindex="-1"></a>so important here in the signatures surprisingly. </span>
<span id="cb18-1092"><a href="#cb18-1092" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-1093"><a href="#cb18-1093" aria-hidden="true" tabindex="-1"></a>If we subselect only the SCANB samples we can recalculate those values</span>
<span id="cb18-1094"><a href="#cb18-1094" aria-hidden="true" tabindex="-1"></a>and compare with the NCN ROR score. </span>
<span id="cb18-1095"><a href="#cb18-1095" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-1098"><a href="#cb18-1098" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb18-1099"><a href="#cb18-1099" aria-hidden="true" tabindex="-1"></a>corr_values <span class="ot">&lt;-</span> <span class="fu">cor</span>(</span>
<span id="cb18-1100"><a href="#cb18-1100" aria-hidden="true" tabindex="-1"></a>    endo_only_scanb <span class="sc">%&gt;%</span> </span>
<span id="cb18-1101"><a href="#cb18-1101" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">select</span>(dplyr<span class="sc">::</span><span class="fu">all_of</span>(<span class="fu">c</span>(</span>
<span id="cb18-1102"><a href="#cb18-1102" aria-hidden="true" tabindex="-1"></a>            <span class="st">"risk_score"</span>, <span class="st">"risk_score_pcs"</span>, <span class="st">"NCN.ROR.asT0"</span>,</span>
<span id="cb18-1103"><a href="#cb18-1103" aria-hidden="true" tabindex="-1"></a>            <span class="fu">names</span>(pathways_of_interest)</span>
<span id="cb18-1104"><a href="#cb18-1104" aria-hidden="true" tabindex="-1"></a>        ))), </span>
<span id="cb18-1105"><a href="#cb18-1105" aria-hidden="true" tabindex="-1"></a>    <span class="at">method =</span> <span class="st">"spearman"</span></span>
<span id="cb18-1106"><a href="#cb18-1106" aria-hidden="true" tabindex="-1"></a>) </span>
<span id="cb18-1107"><a href="#cb18-1107" aria-hidden="true" tabindex="-1"></a>names_corr <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb18-1108"><a href="#cb18-1108" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Risk Score"</span>, <span class="st">"Risk Score (PCs only)"</span>, <span class="st">"NCN ROR"</span>,</span>
<span id="cb18-1109"><a href="#cb18-1109" aria-hidden="true" tabindex="-1"></a>    pathways_of_interest[<span class="fu">rownames</span>(corr_values)[<span class="dv">4</span><span class="sc">:</span><span class="fu">nrow</span>(corr_values)]]</span>
<span id="cb18-1110"><a href="#cb18-1110" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb18-1111"><a href="#cb18-1111" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(corr_values) <span class="ot">&lt;-</span> names_corr</span>
<span id="cb18-1112"><a href="#cb18-1112" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(corr_values) <span class="ot">&lt;-</span> names_corr</span>
<span id="cb18-1113"><a href="#cb18-1113" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb18-1114"><a href="#cb18-1114" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-1115"><a href="#cb18-1115" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=8}</span></span>
<span id="cb18-1116"><a href="#cb18-1116" aria-hidden="true" tabindex="-1"></a><span class="in">#| label: fig-corr-risk-pathways-scanb</span></span>
<span id="cb18-1117"><a href="#cb18-1117" aria-hidden="true" tabindex="-1"></a><span class="in">#| fig-cap: Correlation matrix for all pathways of interest and the two </span></span>
<span id="cb18-1118"><a href="#cb18-1118" aria-hidden="true" tabindex="-1"></a><span class="in">#|     risk scores, both including the clinical factors and without the </span></span>
<span id="cb18-1119"><a href="#cb18-1119" aria-hidden="true" tabindex="-1"></a><span class="in">#|     clinical factors.</span></span>
<span id="cb18-1120"><a href="#cb18-1120" aria-hidden="true" tabindex="-1"></a><span class="in">ggcorrplot::ggcorrplot(</span></span>
<span id="cb18-1121"><a href="#cb18-1121" aria-hidden="true" tabindex="-1"></a><span class="in">    corr_values,</span></span>
<span id="cb18-1122"><a href="#cb18-1122" aria-hidden="true" tabindex="-1"></a><span class="in">    #hc.order = TRUE, </span></span>
<span id="cb18-1123"><a href="#cb18-1123" aria-hidden="true" tabindex="-1"></a><span class="in">    type = "lower",</span></span>
<span id="cb18-1124"><a href="#cb18-1124" aria-hidden="true" tabindex="-1"></a><span class="in">    lab = TRUE, </span></span>
<span id="cb18-1125"><a href="#cb18-1125" aria-hidden="true" tabindex="-1"></a><span class="in">    lab_size = 6, </span></span>
<span id="cb18-1126"><a href="#cb18-1126" aria-hidden="true" tabindex="-1"></a><span class="in">    ggtheme = ggplot2::theme_bw(base_size = 15), </span></span>
<span id="cb18-1127"><a href="#cb18-1127" aria-hidden="true" tabindex="-1"></a><span class="in">    tl.cex = 20</span></span>
<span id="cb18-1128"><a href="#cb18-1128" aria-hidden="true" tabindex="-1"></a><span class="in">) + </span></span>
<span id="cb18-1129"><a href="#cb18-1129" aria-hidden="true" tabindex="-1"></a><span class="in">    change_plot_aes_point() + </span></span>
<span id="cb18-1130"><a href="#cb18-1130" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::labs(x = "", y = "")</span></span>
<span id="cb18-1131"><a href="#cb18-1131" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb18-1132"><a href="#cb18-1132" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-1133"><a href="#cb18-1133" aria-hidden="true" tabindex="-1"></a><span class="fu">## ABiM and nCounter data</span></span>
<span id="cb18-1134"><a href="#cb18-1134" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-1135"><a href="#cb18-1135" aria-hidden="true" tabindex="-1"></a>In <span class="co">[</span><span class="ot">@SCANB2022</span><span class="co">]</span>, the authors validated their approach of using </span>
<span id="cb18-1136"><a href="#cb18-1136" aria-hidden="true" tabindex="-1"></a>RNA-seq instead of nCounter data by showing how one can get the </span>
<span id="cb18-1137"><a href="#cb18-1137" aria-hidden="true" tabindex="-1"></a>ROR and compared to nCounter data sequenced at their facility and</span>
<span id="cb18-1138"><a href="#cb18-1138" aria-hidden="true" tabindex="-1"></a>sent to prosigna for the calculation of the risk of recurrence. </span>
<span id="cb18-1139"><a href="#cb18-1139" aria-hidden="true" tabindex="-1"></a>They have matched RNA-seq data as well. Here we will use the </span>
<span id="cb18-1140"><a href="#cb18-1140" aria-hidden="true" tabindex="-1"></a>ROR obtained from the nCounter data and use the RNA-seq to </span>
<span id="cb18-1141"><a href="#cb18-1141" aria-hidden="true" tabindex="-1"></a>calculate the risk score here developed. </span>
<span id="cb18-1142"><a href="#cb18-1142" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-1145"><a href="#cb18-1145" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb18-1146"><a href="#cb18-1146" aria-hidden="true" tabindex="-1"></a><span class="co"># load the data and perform the normalization + embedding</span></span>
<span id="cb18-1147"><a href="#cb18-1147" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"../../data/scanb_2022/ABiM.100.mymatrix.Rdata"</span>)</span>
<span id="cb18-1148"><a href="#cb18-1148" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"../../data/scanb_2022/Gene.ID.ann.Rdata"</span>)</span>
<span id="cb18-1149"><a href="#cb18-1149" aria-hidden="true" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> readxl<span class="sc">::</span><span class="fu">read_excel</span>(</span>
<span id="cb18-1150"><a href="#cb18-1150" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../../data/scanb_2022/sup_data_20230113.xlsx"</span>, </span>
<span id="cb18-1151"><a href="#cb18-1151" aria-hidden="true" tabindex="-1"></a>    <span class="at">sheet =</span> <span class="st">"ABiM.100"</span></span>
<span id="cb18-1152"><a href="#cb18-1152" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb18-1153"><a href="#cb18-1153" aria-hidden="true" tabindex="-1"></a>abim_100 <span class="ot">&lt;-</span> ABiM.<span class="fl">100.</span>mymatrix</span>
<span id="cb18-1154"><a href="#cb18-1154" aria-hidden="true" tabindex="-1"></a>gene_id_ann <span class="ot">&lt;-</span> Gene.ID.ann</span>
<span id="cb18-1155"><a href="#cb18-1155" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(abim_100) <span class="ot">&lt;-</span> gene_id_ann[<span class="fu">rownames</span>(abim_100), <span class="st">"Gene.Name"</span>]</span>
<span id="cb18-1156"><a href="#cb18-1156" aria-hidden="true" tabindex="-1"></a>abim_100 <span class="ot">&lt;-</span> abim_100[<span class="sc">!</span><span class="fu">duplicated</span>(<span class="fu">rownames</span>(abim_100)), ]</span>
<span id="cb18-1157"><a href="#cb18-1157" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-1158"><a href="#cb18-1158" aria-hidden="true" tabindex="-1"></a><span class="co"># add information so the risk score can be calculated later on</span></span>
<span id="cb18-1159"><a href="#cb18-1159" aria-hidden="true" tabindex="-1"></a>metadata<span class="sc">$</span>age <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb18-1160"><a href="#cb18-1160" aria-hidden="true" tabindex="-1"></a>metadata<span class="sc">$</span>node_status <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(</span>
<span id="cb18-1161"><a href="#cb18-1161" aria-hidden="true" tabindex="-1"></a>    metadata<span class="sc">$</span>LN <span class="sc">&gt;</span> <span class="dv">1</span>,</span>
<span id="cb18-1162"><a href="#cb18-1162" aria-hidden="true" tabindex="-1"></a>    <span class="st">"pos"</span>,</span>
<span id="cb18-1163"><a href="#cb18-1163" aria-hidden="true" tabindex="-1"></a>    <span class="st">"neg"</span></span>
<span id="cb18-1164"><a href="#cb18-1164" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb18-1165"><a href="#cb18-1165" aria-hidden="true" tabindex="-1"></a>metadata<span class="sc">$</span>tumor_size <span class="ot">&lt;-</span> metadata<span class="sc">$</span>Size.mm</span>
<span id="cb18-1166"><a href="#cb18-1166" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-1167"><a href="#cb18-1167" aria-hidden="true" tabindex="-1"></a>abim_100 <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">SummarizedExperiment</span>(</span>
<span id="cb18-1168"><a href="#cb18-1168" aria-hidden="true" tabindex="-1"></a>    <span class="at">assays =</span> <span class="fu">list</span>(</span>
<span id="cb18-1169"><a href="#cb18-1169" aria-hidden="true" tabindex="-1"></a>            <span class="at">fpkm =</span> abim_100,</span>
<span id="cb18-1170"><a href="#cb18-1170" aria-hidden="true" tabindex="-1"></a>            <span class="at">log2fpkm =</span> <span class="fu">log2</span>(abim_100 <span class="sc">+</span> <span class="dv">1</span>)</span>
<span id="cb18-1171"><a href="#cb18-1171" aria-hidden="true" tabindex="-1"></a>        ),</span>
<span id="cb18-1172"><a href="#cb18-1172" aria-hidden="true" tabindex="-1"></a>    <span class="at">colData =</span> metadata <span class="sc">%&gt;%</span></span>
<span id="cb18-1173"><a href="#cb18-1173" aria-hidden="true" tabindex="-1"></a>        data.frame <span class="sc">%&gt;%</span></span>
<span id="cb18-1174"><a href="#cb18-1174" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">rownames&lt;-</span><span class="st">`</span>(.<span class="sc">$</span>GEX.assay) <span class="sc">%&gt;%</span></span>
<span id="cb18-1175"><a href="#cb18-1175" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb18-1176"><a href="#cb18-1176" aria-hidden="true" tabindex="-1"></a>            <span class="at">sample_name =</span> GEX.assay,</span>
<span id="cb18-1177"><a href="#cb18-1177" aria-hidden="true" tabindex="-1"></a>            <span class="at">pam50 =</span> <span class="fu">tolower</span>(SSP.PAM50),</span>
<span id="cb18-1178"><a href="#cb18-1178" aria-hidden="true" tabindex="-1"></a>            <span class="at">ER.pct =</span> <span class="cn">NULL</span></span>
<span id="cb18-1179"><a href="#cb18-1179" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb18-1180"><a href="#cb18-1180" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb18-1181"><a href="#cb18-1181" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-1182"><a href="#cb18-1182" aria-hidden="true" tabindex="-1"></a>abim_100_normalized <span class="ot">&lt;-</span> <span class="fu">get_final_ranking_values</span>(</span>
<span id="cb18-1183"><a href="#cb18-1183" aria-hidden="true" tabindex="-1"></a>    <span class="at">sum_exp =</span> abim_100,</span>
<span id="cb18-1184"><a href="#cb18-1184" aria-hidden="true" tabindex="-1"></a>    <span class="at">assay_to_use =</span> <span class="st">"log2fpkm"</span>,</span>
<span id="cb18-1185"><a href="#cb18-1185" aria-hidden="true" tabindex="-1"></a>    <span class="at">stable_genes =</span> stable_genes,</span>
<span id="cb18-1186"><a href="#cb18-1186" aria-hidden="true" tabindex="-1"></a>    <span class="at">most_variable_genes =</span> <span class="fu">setdiff</span>(<span class="fu">rownames</span>(pca_fit<span class="sc">$</span>loadings), stable_genes)</span>
<span id="cb18-1187"><a href="#cb18-1187" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb18-1188"><a href="#cb18-1188" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-1189"><a href="#cb18-1189" aria-hidden="true" tabindex="-1"></a><span class="co"># calculate the embedding</span></span>
<span id="cb18-1190"><a href="#cb18-1190" aria-hidden="true" tabindex="-1"></a>abim_100_df_pca <span class="ot">&lt;-</span> <span class="fu">get_pca_coordinates</span>(abim_100_normalized, pca_fit) <span class="sc">%&gt;%</span></span>
<span id="cb18-1191"><a href="#cb18-1191" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">bind_cols</span>(</span>
<span id="cb18-1192"><a href="#cb18-1192" aria-hidden="true" tabindex="-1"></a>        ., </span>
<span id="cb18-1193"><a href="#cb18-1193" aria-hidden="true" tabindex="-1"></a>        <span class="fu">data.frame</span>(<span class="at">cohort =</span> <span class="fu">rep</span>(<span class="st">"abim_100"</span>, <span class="fu">ncol</span>(abim_100_normalized)))</span>
<span id="cb18-1194"><a href="#cb18-1194" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb18-1195"><a href="#cb18-1195" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">sample_name =</span> <span class="fu">colnames</span>(abim_100_normalized)) <span class="sc">%&gt;%</span></span>
<span id="cb18-1196"><a href="#cb18-1196" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">inner_join</span>(</span>
<span id="cb18-1197"><a href="#cb18-1197" aria-hidden="true" tabindex="-1"></a>        .,</span>
<span id="cb18-1198"><a href="#cb18-1198" aria-hidden="true" tabindex="-1"></a>        <span class="fu">colData</span>(abim_100) <span class="sc">%&gt;%</span> data.frame,</span>
<span id="cb18-1199"><a href="#cb18-1199" aria-hidden="true" tabindex="-1"></a>        <span class="at">by =</span> <span class="st">"sample_name"</span></span>
<span id="cb18-1200"><a href="#cb18-1200" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb18-1201"><a href="#cb18-1201" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(., df_pca)</span>
<span id="cb18-1202"><a href="#cb18-1202" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-1203"><a href="#cb18-1203" aria-hidden="true" tabindex="-1"></a><span class="co"># save all the results</span></span>
<span id="cb18-1204"><a href="#cb18-1204" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb18-1205"><a href="#cb18-1205" aria-hidden="true" tabindex="-1"></a>    abim_100,</span>
<span id="cb18-1206"><a href="#cb18-1206" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../../results/rds_files/risk_score/abim_100.rds"</span></span>
<span id="cb18-1207"><a href="#cb18-1207" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb18-1208"><a href="#cb18-1208" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-1209"><a href="#cb18-1209" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb18-1210"><a href="#cb18-1210" aria-hidden="true" tabindex="-1"></a>    abim_100_df_pca <span class="sc">%&gt;%</span> data.frame,</span>
<span id="cb18-1211"><a href="#cb18-1211" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../../results/rds_files/risk_score/abim_100_df_pca.rds"</span></span>
<span id="cb18-1212"><a href="#cb18-1212" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb18-1213"><a href="#cb18-1213" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-1214"><a href="#cb18-1214" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb18-1215"><a href="#cb18-1215" aria-hidden="true" tabindex="-1"></a>    abim_100_normalized,</span>
<span id="cb18-1216"><a href="#cb18-1216" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../../results/rds_files/risk_score/abim_100_normalized.rds"</span></span>
<span id="cb18-1217"><a href="#cb18-1217" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb18-1218"><a href="#cb18-1218" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb18-1219"><a href="#cb18-1219" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-1220"><a href="#cb18-1220" aria-hidden="true" tabindex="-1"></a>@fig-pca-abim100-cohort shows the embedding of the ABiM cohort, it </span>
<span id="cb18-1221"><a href="#cb18-1221" aria-hidden="true" tabindex="-1"></a>matches perfectly the molecular landscape.</span>
<span id="cb18-1222"><a href="#cb18-1222" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-1223"><a href="#cb18-1223" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=9, fig.height=6}</span></span>
<span id="cb18-1224"><a href="#cb18-1224" aria-hidden="true" tabindex="-1"></a><span class="in">#| label: fig-pca-abim100-cohort</span></span>
<span id="cb18-1225"><a href="#cb18-1225" aria-hidden="true" tabindex="-1"></a><span class="in">#| fig-cap: Embedding of all samples from TCGA, SCNAB and METABRIC including</span></span>
<span id="cb18-1226"><a href="#cb18-1226" aria-hidden="true" tabindex="-1"></a><span class="in">#|     the samples from ABiM 100 that have matched nCounter data. </span></span>
<span id="cb18-1227"><a href="#cb18-1227" aria-hidden="true" tabindex="-1"></a><span class="in">abim_100_df_pca %&gt;%</span></span>
<span id="cb18-1228"><a href="#cb18-1228" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::ggplot(</span></span>
<span id="cb18-1229"><a href="#cb18-1229" aria-hidden="true" tabindex="-1"></a><span class="in">        aes(x = PC3, y = PC4, color = cohort)</span></span>
<span id="cb18-1230"><a href="#cb18-1230" aria-hidden="true" tabindex="-1"></a><span class="in">    ) +</span></span>
<span id="cb18-1231"><a href="#cb18-1231" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_point(aes(alpha = cohort, size = cohort)) +</span></span>
<span id="cb18-1232"><a href="#cb18-1232" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::scale_size_manual(values = c(</span></span>
<span id="cb18-1233"><a href="#cb18-1233" aria-hidden="true" tabindex="-1"></a><span class="in">        "abim_100" = 3, </span></span>
<span id="cb18-1234"><a href="#cb18-1234" aria-hidden="true" tabindex="-1"></a><span class="in">        "tcga" = 1,</span></span>
<span id="cb18-1235"><a href="#cb18-1235" aria-hidden="true" tabindex="-1"></a><span class="in">        "scanb" = 1,</span></span>
<span id="cb18-1236"><a href="#cb18-1236" aria-hidden="true" tabindex="-1"></a><span class="in">        "metabric" = 1</span></span>
<span id="cb18-1237"><a href="#cb18-1237" aria-hidden="true" tabindex="-1"></a><span class="in">    ), guide = "none") +</span></span>
<span id="cb18-1238"><a href="#cb18-1238" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::scale_alpha_manual(values = c(</span></span>
<span id="cb18-1239"><a href="#cb18-1239" aria-hidden="true" tabindex="-1"></a><span class="in">        "abim_100" = 1, </span></span>
<span id="cb18-1240"><a href="#cb18-1240" aria-hidden="true" tabindex="-1"></a><span class="in">        "tcga" = 0.1,</span></span>
<span id="cb18-1241"><a href="#cb18-1241" aria-hidden="true" tabindex="-1"></a><span class="in">        "scanb" = 0.1,</span></span>
<span id="cb18-1242"><a href="#cb18-1242" aria-hidden="true" tabindex="-1"></a><span class="in">        "metabric" = 0.1</span></span>
<span id="cb18-1243"><a href="#cb18-1243" aria-hidden="true" tabindex="-1"></a><span class="in">    ), guide = "none") +</span></span>
<span id="cb18-1244"><a href="#cb18-1244" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::labs(</span></span>
<span id="cb18-1245"><a href="#cb18-1245" aria-hidden="true" tabindex="-1"></a><span class="in">        color = "Cohort",</span></span>
<span id="cb18-1246"><a href="#cb18-1246" aria-hidden="true" tabindex="-1"></a><span class="in">        alpha = "Cohort",</span></span>
<span id="cb18-1247"><a href="#cb18-1247" aria-hidden="true" tabindex="-1"></a><span class="in">        size = "Cohort",</span></span>
<span id="cb18-1248"><a href="#cb18-1248" aria-hidden="true" tabindex="-1"></a><span class="in">        title = paste0(</span></span>
<span id="cb18-1249"><a href="#cb18-1249" aria-hidden="true" tabindex="-1"></a><span class="in">            "ABiM 100 cohort embedding"</span></span>
<span id="cb18-1250"><a href="#cb18-1250" aria-hidden="true" tabindex="-1"></a><span class="in">        ),</span></span>
<span id="cb18-1251"><a href="#cb18-1251" aria-hidden="true" tabindex="-1"></a><span class="in">        subtitle = "All samples from TCGA, SCANB and METABRIC are plotted"</span></span>
<span id="cb18-1252"><a href="#cb18-1252" aria-hidden="true" tabindex="-1"></a><span class="in">    ) + </span></span>
<span id="cb18-1253"><a href="#cb18-1253" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme_bw(base_size = 18) +</span></span>
<span id="cb18-1254"><a href="#cb18-1254" aria-hidden="true" tabindex="-1"></a><span class="in">    change_plot_aes_point() +</span></span>
<span id="cb18-1255"><a href="#cb18-1255" aria-hidden="true" tabindex="-1"></a><span class="in">    change_guides_point()</span></span>
<span id="cb18-1256"><a href="#cb18-1256" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb18-1257"><a href="#cb18-1257" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-1258"><a href="#cb18-1258" aria-hidden="true" tabindex="-1"></a>@fig-abim100-cohort shows how well the ABiM cohort actually overlays </span>
<span id="cb18-1259"><a href="#cb18-1259" aria-hidden="true" tabindex="-1"></a>with the SCANB dataset, which makes sense as the processing pipeline</span>
<span id="cb18-1260"><a href="#cb18-1260" aria-hidden="true" tabindex="-1"></a>is all similar. </span>
<span id="cb18-1261"><a href="#cb18-1261" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-1264"><a href="#cb18-1264" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb18-1265"><a href="#cb18-1265" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-abim100-cohort</span></span>
<span id="cb18-1266"><a href="#cb18-1266" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Biplot of first two PCA components from ABiM 100, </span></span>
<span id="cb18-1267"><a href="#cb18-1267" aria-hidden="true" tabindex="-1"></a><span class="co">#|     TCGA, SCANB and </span></span>
<span id="cb18-1268"><a href="#cb18-1268" aria-hidden="true" tabindex="-1"></a><span class="co">#|     METABRIC. The normal cohort is highlighted in the plot and</span></span>
<span id="cb18-1269"><a href="#cb18-1269" aria-hidden="true" tabindex="-1"></a><span class="co">#|     has bigger dot size.</span></span>
<span id="cb18-1270"><a href="#cb18-1270" aria-hidden="true" tabindex="-1"></a>abim_100_df_pca <span class="sc">%&gt;%</span> </span>
<span id="cb18-1271"><a href="#cb18-1271" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">!=</span> <span class="st">"poetic"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb18-1272"><a href="#cb18-1272" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">slice</span>(<span class="fu">sample</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">n</span>())) <span class="sc">%&gt;%</span></span>
<span id="cb18-1273"><a href="#cb18-1273" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(</span>
<span id="cb18-1274"><a href="#cb18-1274" aria-hidden="true" tabindex="-1"></a>        <span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> cohort)</span>
<span id="cb18-1275"><a href="#cb18-1275" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb18-1276"><a href="#cb18-1276" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">alpha =</span> cohort, <span class="at">size =</span> cohort)) <span class="sc">+</span></span>
<span id="cb18-1277"><a href="#cb18-1277" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_size_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(</span>
<span id="cb18-1278"><a href="#cb18-1278" aria-hidden="true" tabindex="-1"></a>        <span class="st">"abim_100"</span> <span class="ot">=</span> <span class="dv">2</span>, </span>
<span id="cb18-1279"><a href="#cb18-1279" aria-hidden="true" tabindex="-1"></a>        <span class="st">"tcga"</span> <span class="ot">=</span> <span class="dv">1</span>,</span>
<span id="cb18-1280"><a href="#cb18-1280" aria-hidden="true" tabindex="-1"></a>        <span class="st">"scanb"</span> <span class="ot">=</span> <span class="dv">1</span>,</span>
<span id="cb18-1281"><a href="#cb18-1281" aria-hidden="true" tabindex="-1"></a>        <span class="st">"metabric"</span> <span class="ot">=</span> <span class="dv">1</span></span>
<span id="cb18-1282"><a href="#cb18-1282" aria-hidden="true" tabindex="-1"></a>    ), <span class="at">guide =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb18-1283"><a href="#cb18-1283" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_alpha_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(</span>
<span id="cb18-1284"><a href="#cb18-1284" aria-hidden="true" tabindex="-1"></a>        <span class="st">"abim_100"</span> <span class="ot">=</span> <span class="dv">1</span>, </span>
<span id="cb18-1285"><a href="#cb18-1285" aria-hidden="true" tabindex="-1"></a>        <span class="st">"tcga"</span> <span class="ot">=</span> <span class="fl">0.1</span>,</span>
<span id="cb18-1286"><a href="#cb18-1286" aria-hidden="true" tabindex="-1"></a>        <span class="st">"scanb"</span> <span class="ot">=</span> <span class="fl">0.1</span>,</span>
<span id="cb18-1287"><a href="#cb18-1287" aria-hidden="true" tabindex="-1"></a>        <span class="st">"metabric"</span> <span class="ot">=</span> <span class="fl">0.1</span></span>
<span id="cb18-1288"><a href="#cb18-1288" aria-hidden="true" tabindex="-1"></a>    ), <span class="at">guide =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb18-1289"><a href="#cb18-1289" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb18-1290"><a href="#cb18-1290" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">"Cohort"</span>,</span>
<span id="cb18-1291"><a href="#cb18-1291" aria-hidden="true" tabindex="-1"></a>        <span class="at">alpha =</span> <span class="st">"Cohort"</span>,</span>
<span id="cb18-1292"><a href="#cb18-1292" aria-hidden="true" tabindex="-1"></a>        <span class="at">size =</span> <span class="st">"Cohort"</span>,</span>
<span id="cb18-1293"><a href="#cb18-1293" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="fu">paste0</span>(</span>
<span id="cb18-1294"><a href="#cb18-1294" aria-hidden="true" tabindex="-1"></a>            <span class="st">"First two components of ABiM 100"</span>,</span>
<span id="cb18-1295"><a href="#cb18-1295" aria-hidden="true" tabindex="-1"></a>            <span class="st">" cohort overlayed on all</span><span class="sc">\n</span><span class="st">the three "</span>,</span>
<span id="cb18-1296"><a href="#cb18-1296" aria-hidden="true" tabindex="-1"></a>            <span class="st">"big cohorts in the the molecular landscape"</span></span>
<span id="cb18-1297"><a href="#cb18-1297" aria-hidden="true" tabindex="-1"></a>        ),</span>
<span id="cb18-1298"><a href="#cb18-1298" aria-hidden="true" tabindex="-1"></a>        <span class="at">subtitle =</span> <span class="st">"All samples from TCGA, METABRIC and SCANB are plotted"</span></span>
<span id="cb18-1299"><a href="#cb18-1299" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb18-1300"><a href="#cb18-1300" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">14</span>) <span class="sc">+</span></span>
<span id="cb18-1301"><a href="#cb18-1301" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span></span>
<span id="cb18-1302"><a href="#cb18-1302" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb18-1303"><a href="#cb18-1303" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb18-1304"><a href="#cb18-1304" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-1305"><a href="#cb18-1305" aria-hidden="true" tabindex="-1"></a>And now we move on and calculate the risk scores for the ABiM </span>
<span id="cb18-1306"><a href="#cb18-1306" aria-hidden="true" tabindex="-1"></a>cohort. </span>
<span id="cb18-1307"><a href="#cb18-1307" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-1310"><a href="#cb18-1310" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb18-1311"><a href="#cb18-1311" aria-hidden="true" tabindex="-1"></a>abim_100<span class="sc">$</span>risk_score <span class="ot">&lt;-</span> <span class="fu">calculate_risk_score</span>(</span>
<span id="cb18-1312"><a href="#cb18-1312" aria-hidden="true" tabindex="-1"></a>    coefs_risk_model,</span>
<span id="cb18-1313"><a href="#cb18-1313" aria-hidden="true" tabindex="-1"></a>    abim_100_df_pca <span class="sc">%&gt;%</span></span>
<span id="cb18-1314"><a href="#cb18-1314" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"abim_100"</span>),</span>
<span id="cb18-1315"><a href="#cb18-1315" aria-hidden="true" tabindex="-1"></a>    <span class="at">is_node =</span> <span class="cn">TRUE</span></span>
<span id="cb18-1316"><a href="#cb18-1316" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb18-1317"><a href="#cb18-1317" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb18-1318"><a href="#cb18-1318" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-1321"><a href="#cb18-1321" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb18-1322"><a href="#cb18-1322" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-rorft-riskscore-abim</span></span>
<span id="cb18-1323"><a href="#cb18-1323" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Correlation between risk score calculated based on </span></span>
<span id="cb18-1324"><a href="#cb18-1324" aria-hidden="true" tabindex="-1"></a><span class="co">#|     the principal components and clinical factors versus the Prosigna</span></span>
<span id="cb18-1325"><a href="#cb18-1325" aria-hidden="true" tabindex="-1"></a><span class="co">#|     ROR obtained from the nCounter data and evaluated by Prosigna. Cohort</span></span>
<span id="cb18-1326"><a href="#cb18-1326" aria-hidden="true" tabindex="-1"></a><span class="co">#|     used is the ABiM 100 samples. </span></span>
<span id="cb18-1327"><a href="#cb18-1327" aria-hidden="true" tabindex="-1"></a>df_abim_100 <span class="ot">&lt;-</span> <span class="fu">colData</span>(abim_100) <span class="sc">%&gt;%</span></span>
<span id="cb18-1328"><a href="#cb18-1328" aria-hidden="true" tabindex="-1"></a>    data.frame</span>
<span id="cb18-1329"><a href="#cb18-1329" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-1330"><a href="#cb18-1330" aria-hidden="true" tabindex="-1"></a>df_abim_100 <span class="sc">%&gt;%</span></span>
<span id="cb18-1331"><a href="#cb18-1331" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(risk_score)) <span class="sc">%&gt;%</span></span>
<span id="cb18-1332"><a href="#cb18-1332" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(</span>
<span id="cb18-1333"><a href="#cb18-1333" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> risk_score,</span>
<span id="cb18-1334"><a href="#cb18-1334" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> ProsignaFT.ROR,</span>
<span id="cb18-1335"><a href="#cb18-1335" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> pam50</span>
<span id="cb18-1336"><a href="#cb18-1336" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">+</span> </span>
<span id="cb18-1337"><a href="#cb18-1337" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span> </span>
<span id="cb18-1338"><a href="#cb18-1338" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">get_colors_pam50</span>(df_abim_100)) <span class="sc">+</span></span>
<span id="cb18-1339"><a href="#cb18-1339" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_smooth</span>(</span>
<span id="cb18-1340"><a href="#cb18-1340" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">"black"</span>,</span>
<span id="cb18-1341"><a href="#cb18-1341" aria-hidden="true" tabindex="-1"></a>        <span class="at">method =</span> <span class="st">"gam"</span>,</span>
<span id="cb18-1342"><a href="#cb18-1342" aria-hidden="true" tabindex="-1"></a>        <span class="at">formula =</span> y <span class="sc">~</span> <span class="fu">s</span>(x, <span class="at">bs =</span> <span class="st">"cs"</span>)</span>
<span id="cb18-1343"><a href="#cb18-1343" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb18-1344"><a href="#cb18-1344" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">15</span>) <span class="sc">+</span></span>
<span id="cb18-1345"><a href="#cb18-1345" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb18-1346"><a href="#cb18-1346" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"Risk score"</span>,</span>
<span id="cb18-1347"><a href="#cb18-1347" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"ROR Prosigna FT"</span>, </span>
<span id="cb18-1348"><a href="#cb18-1348" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">"PAM50"</span>,</span>
<span id="cb18-1349"><a href="#cb18-1349" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">"ABiM 100 samples and risk scores"</span>,</span>
<span id="cb18-1350"><a href="#cb18-1350" aria-hidden="true" tabindex="-1"></a>        <span class="at">caption =</span> <span class="st">"FT: fresh-frozen tissue"</span></span>
<span id="cb18-1351"><a href="#cb18-1351" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb18-1352"><a href="#cb18-1352" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>() <span class="sc">+</span> </span>
<span id="cb18-1353"><a href="#cb18-1353" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>()</span>
<span id="cb18-1354"><a href="#cb18-1354" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb18-1355"><a href="#cb18-1355" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-1356"><a href="#cb18-1356" aria-hidden="true" tabindex="-1"></a>@fig-rorft-riskscore-abim shows that there is a positive correlation</span>
<span id="cb18-1357"><a href="#cb18-1357" aria-hidden="true" tabindex="-1"></a>between the risk score and ROR obtained from Prosigna. We see</span>
<span id="cb18-1358"><a href="#cb18-1358" aria-hidden="true" tabindex="-1"></a>that some of the luminal A patients have higher ROR but lower </span>
<span id="cb18-1359"><a href="#cb18-1359" aria-hidden="true" tabindex="-1"></a>ROR. Also, among luminal B patients there were some with the same </span>
<span id="cb18-1360"><a href="#cb18-1360" aria-hidden="true" tabindex="-1"></a>ROR but different risk score, showing that the risk. Also some</span>
<span id="cb18-1361"><a href="#cb18-1361" aria-hidden="true" tabindex="-1"></a>of the basal samples have smaller ROR than some of the luminal</span>
<span id="cb18-1362"><a href="#cb18-1362" aria-hidden="true" tabindex="-1"></a>B and A samples, and the risk score captures these differences, i.e.,</span>
<span id="cb18-1363"><a href="#cb18-1363" aria-hidden="true" tabindex="-1"></a>all basal samples have risk scores higher than 0.75.</span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
      &nbsp;
    </div>   
    <div class="nav-footer-center">
      &nbsp;
    <div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/chronchi/molecular_landscape/issues/new" class="toc-action"><i class="bi bi-github"></i>Report an issue</a></li></ul></div></div>
    <div class="nav-footer-right">
      <ul class="footer-items list-unstyled">
    <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/chronchi">
      <i class="bi bi-github" role="img">
</i> 
    </a>
  </li>  
    <li class="nav-item compact">
    <a class="nav-link" href="https://twitter.com/chronchi">
      <i class="bi bi-twitter" role="img">
</i> 
    </a>
  </li>  
</ul>
    </div>
  </div>
</footer>




</body></html>