<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.361">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>EMBER creates a unified space for independent breast cancer transcriptomic datasets enabling precision oncology. - 7&nbsp; Tests and tries</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../chapters/discussion.html" rel="next">
<link href="../chapters/expanding_ember.html" rel="prev">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>

<script src="../site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="../site_libs/lightable-0.0.1/lightable.css" rel="stylesheet">
<script src="../site_libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<link href="../site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="../site_libs/datatables-binding-0.28/datatables.js"></script>
<script src="../site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<link href="../site_libs/dt-core-1.13.4/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="../site_libs/dt-core-1.13.4/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="../site_libs/dt-core-1.13.4/js/jquery.dataTables.min.js"></script>
<link href="../site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet">
<script src="../site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../chapters/trying.html"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Tests and tries</span></a></li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
      <button type="button" class="btn quarto-search-button" aria-label="" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../">EMBER creates a unified space for independent breast cancer transcriptomic datasets enabling precision oncology.</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/chronchi/molecular_landscape" title="Source Code" class="quarto-navigation-tool px-1" aria-label="Source Code"><i class="bi bi-github"></i></a>
    <a href="https://twitter.com/intent/tweet?url=|url|" title="Twitter" class="quarto-navigation-tool px-1" aria-label="Twitter"><i class="bi bi-twitter"></i></a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Preface</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/intro.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Introduction</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/surv_analysis_estrogen.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Estrogen receptor status is continuous, not dichotomous</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/pca_merging.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">A new framework for personalized medicine: integrating</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/validation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Validating the molecular landscape</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/scoring.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Pathway activity in a personalized context</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/risk_score.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Risk score model</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/expanding_ember.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Expanding embedding and removing batch effects</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/trying.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Tests and tries</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/discussion.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Discussion</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/references.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">References</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#using-several-pathways-or-just-one-pathway-for-gsva" id="toc-using-several-pathways-or-just-one-pathway-for-gsva" class="nav-link active" data-scroll-target="#using-several-pathways-or-just-one-pathway-for-gsva"><span class="header-section-number">7.1</span> Using several pathways or just one pathway for GSVA</a>
  <ul class="collapse">
  <li><a href="#one-vs-all" id="toc-one-vs-all" class="nav-link" data-scroll-target="#one-vs-all"><span class="header-section-number">7.1.1</span> One vs all</a></li>
  </ul></li>
  <li><a href="#minimum-number-of-samples-for-gsva" id="toc-minimum-number-of-samples-for-gsva" class="nav-link" data-scroll-target="#minimum-number-of-samples-for-gsva"><span class="header-section-number">7.2</span> Minimum number of samples for GSVA</a></li>
  <li><a href="#normalization-strategy" id="toc-normalization-strategy" class="nav-link" data-scroll-target="#normalization-strategy"><span class="header-section-number">7.3</span> Normalization strategy</a></li>
  <li><a href="#genefu" id="toc-genefu" class="nav-link" data-scroll-target="#genefu"><span class="header-section-number">7.4</span> genefu</a>
  <ul class="collapse">
  <li><a href="#genefu-and-aims" id="toc-genefu-and-aims" class="nav-link" data-scroll-target="#genefu-and-aims"><span class="header-section-number">7.4.1</span> genefu and AIMS</a></li>
  </ul></li>
  <li><a href="#nanostring-signatures" id="toc-nanostring-signatures" class="nav-link" data-scroll-target="#nanostring-signatures"><span class="header-section-number">7.5</span> nanostring signatures</a></li>
  <li><a href="#risk-stratification-and-the-molecular-landscape" id="toc-risk-stratification-and-the-molecular-landscape" class="nav-link" data-scroll-target="#risk-stratification-and-the-molecular-landscape"><span class="header-section-number">7.6</span> Risk stratification and the molecular landscape</a></li>
  <li><a href="#late-distant-recurrence-trying-to-find-biomarkers" id="toc-late-distant-recurrence-trying-to-find-biomarkers" class="nav-link" data-scroll-target="#late-distant-recurrence-trying-to-find-biomarkers"><span class="header-section-number">7.7</span> Late distant recurrence: trying to find biomarkers</a></li>
  <li><a href="#correlation-of-pathways-with-proliferation" id="toc-correlation-of-pathways-with-proliferation" class="nav-link" data-scroll-target="#correlation-of-pathways-with-proliferation"><span class="header-section-number">7.8</span> Correlation of pathways with proliferation</a></li>
  <li><a href="#her2-and-their-categories" id="toc-her2-and-their-categories" class="nav-link" data-scroll-target="#her2-and-their-categories"><span class="header-section-number">7.9</span> HER2 and their categories</a>
  <ul class="collapse">
  <li><a href="#her2-amplicon-and-survival-analysis" id="toc-her2-amplicon-and-survival-analysis" class="nav-link" data-scroll-target="#her2-amplicon-and-survival-analysis"><span class="header-section-number">7.9.1</span> HER2 Amplicon and survival analysis</a></li>
  </ul></li>
  <li><a href="#ilc-and-position-in-the-molecular-landscape" id="toc-ilc-and-position-in-the-molecular-landscape" class="nav-link" data-scroll-target="#ilc-and-position-in-the-molecular-landscape"><span class="header-section-number">7.10</span> ILC and position in the molecular landscape</a></li>
  <li><a href="#robustness-to-library-protocol" id="toc-robustness-to-library-protocol" class="nav-link" data-scroll-target="#robustness-to-library-protocol"><span class="header-section-number">7.11</span> Robustness to library protocol</a></li>
  <li><a href="#distance-in-4th-component-for-poetic" id="toc-distance-in-4th-component-for-poetic" class="nav-link" data-scroll-target="#distance-in-4th-component-for-poetic"><span class="header-section-number">7.12</span> Distance in 4th component for POETIC</a></li>
  </ul>
<div class="toc-actions"><div><i class="bi bi-github"></i></div><div class="action-links"><p><a href="https://github.com/chronchi/molecular_landscape/issues/new" class="toc-action">Report an issue</a></p></div></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Tests and tries</span></h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<p>This is a file where we try different things using the molecular data. We keep it here in case anyone finds anything here interesting and would like to further discuss with us for potential collaborations.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>* The library is already synchronized with the lockfile.</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>    ../../results/plots/trying ../../results/rds_files/trying 
                         FALSE                          FALSE 
   ../../results/tables/trying 
                         FALSE </code></pre>
</div>
</div>
<section id="using-several-pathways-or-just-one-pathway-for-gsva" class="level2" data-number="7.1">
<h2 data-number="7.1" class="anchored" data-anchor-id="using-several-pathways-or-just-one-pathway-for-gsva"><span class="header-section-number">7.1</span> Using several pathways or just one pathway for GSVA</h2>
<p>We now evaluate the effect of the scores for each pathway when using all the 53 pathways combined and each one individually. We use only TCGA for this step.</p>
<section id="one-vs-all" class="level3" data-number="7.1.1">
<h3 data-number="7.1.1" class="anchored" data-anchor-id="one-vs-all"><span class="header-section-number">7.1.1</span> One vs all</h3>
<p>We now calculate the scores for each gene set individually.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-4-1.png" class="img-fluid" width="1920"></p>
</div>
</div>
<p>It seems they are all positively correlated, so one could calculate the scores basically using just one gene set with exception of some of the gene sets.</p>
</section>
</section>
<section id="minimum-number-of-samples-for-gsva" class="level2" data-number="7.2">
<h2 data-number="7.2" class="anchored" data-anchor-id="minimum-number-of-samples-for-gsva"><span class="header-section-number">7.2</span> Minimum number of samples for GSVA</h2>
<p>According to the original GSVA paper, a minimum of 10 samples are necessary to have a good power when using GSVA. Since here we have at least 5 molecular subgroups, we hypothesize we need at least 50 samples to get a good score so we can compare across cohorts. We will try with different numbers of samples using TCGA, we start with 10, 25, 50 and 100 samples and we compare with the scores using the full dataset.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-8-1.png" class="img-fluid" width="4800"></p>
</div>
</div>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid" width="4800"></p>
</div>
</div>
<p>The conclusion we can take here is that depending on the gene set not even 150 samples is enough. For some only 25 patients is already enough, such as SET ER/PR, androgen response, estrogen response early and E2F targets.</p>
<p>Could we mix around 20 samples of SCANB together with more samples from TCGA and METABRIC to get the scores? Let us try to adress this question again. We are using the regressed data now here as this is the only way.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid" width="1920"></p>
</div>
</div>
<p>There is still some correlation but the scores are shrinked in the end.</p>
</section>
<section id="normalization-strategy" class="level2" data-number="7.3">
<h2 data-number="7.3" class="anchored" data-anchor-id="normalization-strategy"><span class="header-section-number">7.3</span> Normalization strategy</h2>
<p>We can also try another way of normalizing, namely for each sample we can z scale it.</p>
<p>And now we can visualize the results.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pca-embeddings-3" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="trying_files/figure-html/fig-pca-embeddings-3-1.png" class="img-fluid figure-img" width="1344"></p>
<figcaption class="figure-caption">Figure&nbsp;7.1: PCA projections colored by different factors and organized by different components. (A) Plot of the first two components colored by cohort. (B) Plot of the first two components colored by ER status. (C) Plot of the second and third components colored by cohort. (D) Plot of the second and third components colored by ER status. (E) Plot of the second and third components colored by PAM50. (F) Plot of the second and third components colored by INTCLUST, only METABRIC has an assigned value for this variable, NAs are TCGA samples.</figcaption>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-16-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>We see that samples are well embedded as well.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pca-scanb-pc1-2" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="trying_files/figure-html/fig-pca-scanb-pc1-2-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Figure&nbsp;7.2: PCA embedding colored by cohort. The components used are the first and second components.</figcaption>
</figure>
</div>
</div>
</div>
<p>SCANB and POETIC are in between as expected and they are closer to their sequencing technology. Interestingly the variation for SCANB and TCGA are compressed in the PC1 vs PC2 axis compared to METABRIC and POETIC.</p>
<p><a href="#fig-pca-scanb-er-pam50-2">Figure&nbsp;<span>7.3</span></a> shows that the SCANB samples are also well mixed regarding the clinical factors, including the <span class="math inline">\(SET_{ER/PR}\)</span> signature.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pca-scanb-er-pam50-2" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="trying_files/figure-html/fig-pca-scanb-er-pam50-2-1.png" class="img-fluid figure-img" width="1536"></p>
<figcaption class="figure-caption">Figure&nbsp;7.3: PCA embedding of all samples from TCGA, SCANB and METABRIC. (A) Colored by cohort, (B) colored by ER status, (C) colored by PAM50 molecular subtype, (D) colored by the <span class="math inline">\(SET_{ER/PR}\)</span> signature and only SCANB samples.</figcaption>
</figure>
</div>
</div>
</div>
<p>In the end normalizing by scaling the sample before doing any embedding works as well. Though the embedding seems to be more compressed. I would argue that the qPCR-like normalization gives a bit better embedding.</p>
</section>
<section id="genefu" class="level2" data-number="7.4">
<h2 data-number="7.4" class="anchored" data-anchor-id="genefu"><span class="header-section-number">7.4</span> genefu</h2>
<p>With genefu it is possible to calculate scores from comercially available signatures. Here we try to use it on SCANB, METABRIC and TCGA. Moreover, we can start with ABiM, as this dataset has the gold standard ROR score obtained from nCounter data. This is a good way of validating the pipeline.</p>
<p>We start by comparing the molecular subtypes:</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
FALSE  TRUE 
   14    86 </code></pre>
</div>
</div>
<p>86 out of 100 were correctly called. Here we are using the gold standard wich is the molecular subtype called from FT samples. The normal like subtype is not available from the FT prosigna assay. So we now compare the results with the SSP obtained from the SCANB team.</p>
<p>Below is the confusion matrix to see which subtypes were misclassified.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>Confusion Matrix and Statistics

          Reference
Prediction Basal Her2 LumB LumA Normal
    Basal     13    0    0    0      0
    Her2       0    8    0    0      0
    LumB       1    1   33    1      0
    LumA       1    0    7   32      0
    Normal     1    0    0    2      0

Overall Statistics
                                          
               Accuracy : 0.86            
                 95% CI : (0.7763, 0.9213)
    No Information Rate : 0.4             
    P-Value [Acc &gt; NIR] : &lt; 2.2e-16       
                                          
                  Kappa : 0.7965          
                                          
 Mcnemar's Test P-Value : NA              

Statistics by Class:

                     Class: Basal Class: Her2 Class: LumB Class: LumA
Sensitivity                0.8125      0.8889      0.8250      0.9143
Specificity                1.0000      1.0000      0.9500      0.8769
Pos Pred Value             1.0000      1.0000      0.9167      0.8000
Neg Pred Value             0.9655      0.9891      0.8906      0.9500
Prevalence                 0.1600      0.0900      0.4000      0.3500
Detection Rate             0.1300      0.0800      0.3300      0.3200
Detection Prevalence       0.1300      0.0800      0.3600      0.4000
Balanced Accuracy          0.9062      0.9444      0.8875      0.8956
                     Class: Normal
Sensitivity                     NA
Specificity                   0.97
Pos Pred Value                  NA
Neg Pred Value                  NA
Prevalence                    0.00
Detection Rate                0.00
Detection Prevalence          0.03
Balanced Accuracy               NA</code></pre>
</div>
</div>
<p>All luminal A were predicted correctly. Some normal like were predicted to be luminal A or basal and Her2. There were some mismatches for the LumB, that were misclassified as LumA. Still it is ok.</p>
<p>Let us now calculate the ROR for the ABiM cohort.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-ror-genefu-abim" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="trying_files/figure-html/fig-ror-genefu-abim-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Figure&nbsp;7.4: Correlation between ROR obtained from prosigna assay and R package genefu</figcaption>
</figure>
</div>
</div>
</div>
<p>The correlation is relatively good and respects the molecular subtypes in general. The ROR obtained from genefu for some luminal B are not quite right as they are below 50.</p>
<p>We now move on to the other signatures: Mammaprint, EndoPredict (EP) and OncotypeDX Risk Score (RS).</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-ror-mammaprint-genefu-abim" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="trying_files/figure-html/fig-ror-mammaprint-genefu-abim-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Figure&nbsp;7.5: Correlation between ROR obtained from prosigna assay and the mammaprint signature obtained from the R package genefu</figcaption>
</figure>
</div>
</div>
</div>
<p><a href="#fig-ror-mammaprint-genefu-abim">Figure&nbsp;<span>7.5</span></a> shows the correlation between the mammaprint signature (named GENE70) and the ROR from prosigna.</p>
<p>Let us now compare the GENE70 signature with the risk score developed in the previous chapters. We use the ABiM 100 cohort to make the comparison. One side note, we are using only 51 out of the 70 probes available in the package. 15 probes don’t have any corresponding symbol. Out of the available genes, some are duplicated. For the duplicated genes, the correlation coefficients are all similar, so we can select just the first ocurrence in the list.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-24-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>There is a good correlation between the two signatures, suggesting that the Mammaprint signature is somehow based on the position of the samples in the molecular landscape.</p>
<p>We now proceed and calculate the mammaprint scores for SCANB and METABRIC as well and only for ER+ BC samples.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-met-scanb-mammaprint" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="trying_files/figure-html/fig-met-scanb-mammaprint-1.png" class="img-fluid figure-img" width="960"></p>
<figcaption class="figure-caption">Figure&nbsp;7.6: Embedding of ER+ BC samples colored by the mammaprint score.</figcaption>
</figure>
</div>
</div>
</div>
<p>We see that the pattern is very close to what we have from the risk score created in the previous chapter. The figure below shows the correlation between mammaprint and the risk score once again.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-28-1.png" class="img-fluid" width="1344"></p>
</div>
</div>
<p>Now for OncotypeDX:</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-ror-genefu-abim-oncotype" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="trying_files/figure-html/fig-ror-genefu-abim-oncotype-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Figure&nbsp;7.7: Correlation between oncotypeDX and R package genefu</figcaption>
</figure>
</div>
</div>
</div>
<p>There is also a correlation between the risk score and oncotypeDX. The paper https://www.nature.com/articles/s41523-022-00492-0#Sec8 also used the genefu for calculating the oncotypeDX.</p>
<section id="genefu-and-aims" class="level3" data-number="7.4.1">
<h3 data-number="7.4.1" class="anchored" data-anchor-id="genefu-and-aims"><span class="header-section-number">7.4.1</span> genefu and AIMS</h3>
<p>Here we try to use AIMS to calculate the PAM50 subtypes for the and compare with the results obtained previously. We want to compare the classifications with the molecular landscape.</p>
<p>The confusion matrix below show the results of the AIMS PAM50 as the predicted and the reference is the PAM50 subtypes obtained from the SCANB team.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>Confusion Matrix and Statistics

          Reference
Prediction luma her2 normal lumb basal
    luma   2301   61     51  664     0
    her2      9  627     16  200     4
    normal  724   37    830    6    11
    lumb     17   17      0  938     0
    basal     0   75     14    0   614

Overall Statistics
                                         
               Accuracy : 0.7359         
                 95% CI : (0.7255, 0.746)
    No Information Rate : 0.4228         
    P-Value [Acc &gt; NIR] : &lt; 2.2e-16      
                                         
                  Kappa : 0.6411         
                                         
 Mcnemar's Test P-Value : NA             

Statistics by Class:

                     Class: luma Class: her2 Class: normal Class: lumb
Sensitivity               0.7542     0.76744        0.9111      0.5188
Specificity               0.8137     0.96421        0.8766      0.9937
Pos Pred Value            0.7478     0.73248        0.5162      0.9650
Neg Pred Value            0.8188     0.97013        0.9856      0.8607
Prevalence                0.4228     0.11322        0.1262      0.2506
Detection Rate            0.3189     0.08689        0.1150      0.1300
Detection Prevalence      0.4264     0.11863        0.2228      0.1347
Balanced Accuracy         0.7839     0.86583        0.8938      0.7563
                     Class: basal
Sensitivity               0.97615
Specificity               0.98649
Pos Pred Value            0.87340
Neg Pred Value            0.99770
Prevalence                0.08717
Detection Rate            0.08509
Detection Prevalence      0.09742
Balanced Accuracy         0.98132</code></pre>
</div>
</div>
<p>We see that there is a big misclassification from luminal A samples to normal like samples and several luminal B are misclassified as luminal A.</p>
<p>The plot below shows the coloring by the reference PAM50 and the predicted by AIMS.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-scanb-aims" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="trying_files/figure-html/fig-scanb-aims-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Figure&nbsp;7.8: SCANB embedding stratified by PAM50 algorithms by both the ones provided by the SCANB (SSP) and the AIMS algorithm</figcaption>
</figure>
</div>
</div>
</div>
<p>On the left there is the coloring for the AIMS algorithm and on the right from the reference. We see the missclassified samples are not randomly misclassified, they are on the boundaries.</p>
<p>The plot below shows the maximum probability score for each sample individually.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-33-1.png" class="img-fluid" width="384"></p>
</div>
</div>
<p>The majority of samples have probabilities of either 1 or 0. Not so often in between.</p>
<p>And the figure below shows the confusion matrix for the ABIM cohort.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>Confusion Matrix and Statistics

          Reference
Prediction luma lumb her2 normal
    luma     28   17    1      3
    lumb      0   16    0      0
    her2      0    0   10      0
    normal    2    0    2      2

Overall Statistics
                                          
               Accuracy : 0.6914          
                 95% CI : (0.5789, 0.7893)
    No Information Rate : 0.4074          
    P-Value [Acc &gt; NIR] : 2.253e-07       
                                          
                  Kappa : 0.5401          
                                          
 Mcnemar's Test P-Value : NA              

Statistics by Class:

                     Class: luma Class: lumb Class: her2 Class: normal
Sensitivity               0.9333      0.4848      0.7692       0.40000
Specificity               0.5882      1.0000      1.0000       0.94737
Pos Pred Value            0.5714      1.0000      1.0000       0.33333
Neg Pred Value            0.9375      0.7385      0.9577       0.96000
Prevalence                0.3704      0.4074      0.1605       0.06173
Detection Rate            0.3457      0.1975      0.1235       0.02469
Detection Prevalence      0.6049      0.1975      0.1235       0.07407
Balanced Accuracy         0.7608      0.7424      0.8846       0.67368</code></pre>
</div>
</div>
<p>We see here that in this case that there are several luminal B samples missclassified as luminal A.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-35-1.png" class="img-fluid" width="768"></p>
</div>
</div>
<p>The misclassified samples from AIMS they really seem to be on the top region together with the luminal A. Some of the normal samples are even in the luminal A region.</p>
<p>Now we check the PDXs.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>$METS15
 treatment Basal Her2 LumA LumB Normal
      CTRL     0    0    0    3      0
       DHT     0    0    0    0      0
        E2     0    0    1    3      0
      E2P4     0    0    1    2      0
        P4     0    0    1    3      0

$T105
 treatment Basal Her2 LumA LumB Normal
      CTRL     0    0    0    0      5
       DHT     0    0    0    0      0
        E2     0    0    0    0      4
      E2P4     0    0    0    0      0
        P4     0    4    0    0      0

$T109
 treatment Basal Her2 LumA LumB Normal
      CTRL     0    0    3    0      1
       DHT     0    0    0    0      0
        E2     0    0    1    0      3
      E2P4     0    0    0    0      0
        P4     0    0    4    0      0

$T110
 treatment Basal Her2 LumA LumB Normal
      CTRL     0    1    0    0      3
       DHT     0    0    0    0      0
        E2     0    0    0    0      2
      E2P4     0    1    0    0      2
        P4     0    0    0    0      4

$T111
 treatment Basal Her2 LumA LumB Normal
      CTRL     0    0    0    0      3
       DHT     0    0    0    0      3
        E2     0    0    0    0      3
      E2P4     0    0    0    0      2
        P4     1    0    0    0      2

$T113
 treatment Basal Her2 LumA LumB Normal
      CTRL     0    0    2    0      1
       DHT     0    0    0    0      0
        E2     0    0    0    0      3
      E2P4     0    0    0    0      0
        P4     0    0    2    0      2</code></pre>
</div>
</div>
<p>These molecular subtyping do not make much sense. We know that by giving P4 estrogen signaling actually goes up, so a expected subtype change is from normal to either luminal A and B. Moreover, these PDXs are in an active state of proliferation due to the MIND model, so being in the normal like region is not a good indication. Also this results show that almost all PDXs would be normal like which does not go in hand with the results we get from the lab.</p>
<p>And the probabilities are available below.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-37-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>And another validation is using the normal cohort. Below is the number of all predicted molecular subtypes.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
Normal 
    66 </code></pre>
</div>
</div>
<p>All samples are normal-like as expected, since they are normal. So it could be that the AIMS algorithm is not suited for PDXs but for clinical samples it works quite well.</p>
</section>
</section>
<section id="nanostring-signatures" class="level2" data-number="7.5">
<h2 data-number="7.5" class="anchored" data-anchor-id="nanostring-signatures"><span class="header-section-number">7.5</span> nanostring signatures</h2>
<p>In this section we try the nanostring panel signatures provided by the Breast Cancer 360 test. We will compare some of the signatures by using GSVA on the original datasets and by calculating the sum of the gene expression levels of the genes when regressing out the 1st, 2nd and 5th components.</p>
<p>The table below shows the number of genes available for each signature and each dataset. In general it seems that all signatures have over 70% of the genes available, a good indication.</p>
<div class="cell">
<div class="cell-output-display">
<table class="lightable-classic table table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">pathway</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">tcga</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">scanb</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">metabric</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">poetic</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">n</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">average_percentage</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">adhesion_and_migration</td>
<td style="text-align: right;">65</td>
<td style="text-align: right;">61</td>
<td style="text-align: right;">68</td>
<td style="text-align: right;">83</td>
<td style="text-align: right;">83</td>
<td style="text-align: left;">0.83</td>
</tr>
<tr class="even">
<td style="text-align: left;">angiogenesis</td>
<td style="text-align: right;">27</td>
<td style="text-align: right;">27</td>
<td style="text-align: right;">29</td>
<td style="text-align: right;">33</td>
<td style="text-align: right;">34</td>
<td style="text-align: left;">0.85</td>
</tr>
<tr class="odd">
<td style="text-align: left;">antigen_presentation</td>
<td style="text-align: right;">20</td>
<td style="text-align: right;">19</td>
<td style="text-align: right;">19</td>
<td style="text-align: right;">19</td>
<td style="text-align: right;">21</td>
<td style="text-align: left;">0.92</td>
</tr>
<tr class="even">
<td style="text-align: left;">apoptosis</td>
<td style="text-align: right;">7</td>
<td style="text-align: right;">7</td>
<td style="text-align: right;">5</td>
<td style="text-align: right;">9</td>
<td style="text-align: right;">9</td>
<td style="text-align: left;">0.78</td>
</tr>
<tr class="odd">
<td style="text-align: left;">cytokine_and_chemokine_signaling</td>
<td style="text-align: right;">38</td>
<td style="text-align: right;">29</td>
<td style="text-align: right;">37</td>
<td style="text-align: right;">47</td>
<td style="text-align: right;">50</td>
<td style="text-align: left;">0.76</td>
</tr>
<tr class="even">
<td style="text-align: left;">dna_damage_repair</td>
<td style="text-align: right;">126</td>
<td style="text-align: right;">108</td>
<td style="text-align: right;">118</td>
<td style="text-align: right;">133</td>
<td style="text-align: right;">143</td>
<td style="text-align: left;">0.85</td>
</tr>
<tr class="odd">
<td style="text-align: left;">emt</td>
<td style="text-align: right;">72</td>
<td style="text-align: right;">71</td>
<td style="text-align: right;">63</td>
<td style="text-align: right;">82</td>
<td style="text-align: right;">85</td>
<td style="text-align: left;">0.85</td>
</tr>
<tr class="even">
<td style="text-align: left;">er_signaling</td>
<td style="text-align: right;">27</td>
<td style="text-align: right;">25</td>
<td style="text-align: right;">24</td>
<td style="text-align: right;">25</td>
<td style="text-align: right;">27</td>
<td style="text-align: left;">0.94</td>
</tr>
<tr class="odd">
<td style="text-align: left;">epigenetic_regulation</td>
<td style="text-align: right;">16</td>
<td style="text-align: right;">16</td>
<td style="text-align: right;">14</td>
<td style="text-align: right;">17</td>
<td style="text-align: right;">18</td>
<td style="text-align: left;">0.88</td>
</tr>
<tr class="even">
<td style="text-align: left;">hedgehog</td>
<td style="text-align: right;">14</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">17</td>
<td style="text-align: right;">20</td>
<td style="text-align: right;">20</td>
<td style="text-align: left;">0.76</td>
</tr>
<tr class="odd">
<td style="text-align: left;">immune_infiltration</td>
<td style="text-align: right;">24</td>
<td style="text-align: right;">21</td>
<td style="text-align: right;">23</td>
<td style="text-align: right;">34</td>
<td style="text-align: right;">34</td>
<td style="text-align: left;">0.75</td>
</tr>
<tr class="even">
<td style="text-align: left;">jak_stat</td>
<td style="text-align: right;">37</td>
<td style="text-align: right;">36</td>
<td style="text-align: right;">35</td>
<td style="text-align: right;">45</td>
<td style="text-align: right;">47</td>
<td style="text-align: left;">0.81</td>
</tr>
<tr class="odd">
<td style="text-align: left;">mapk</td>
<td style="text-align: right;">67</td>
<td style="text-align: right;">61</td>
<td style="text-align: right;">68</td>
<td style="text-align: right;">95</td>
<td style="text-align: right;">100</td>
<td style="text-align: left;">0.73</td>
</tr>
<tr class="even">
<td style="text-align: left;">notch</td>
<td style="text-align: right;">21</td>
<td style="text-align: right;">19</td>
<td style="text-align: right;">16</td>
<td style="text-align: right;">21</td>
<td style="text-align: right;">22</td>
<td style="text-align: left;">0.88</td>
</tr>
<tr class="odd">
<td style="text-align: left;">pi3k</td>
<td style="text-align: right;">72</td>
<td style="text-align: right;">67</td>
<td style="text-align: right;">63</td>
<td style="text-align: right;">91</td>
<td style="text-align: right;">96</td>
<td style="text-align: left;">0.76</td>
</tr>
<tr class="even">
<td style="text-align: left;">proliferation</td>
<td style="text-align: right;">121</td>
<td style="text-align: right;">113</td>
<td style="text-align: right;">114</td>
<td style="text-align: right;">136</td>
<td style="text-align: right;">144</td>
<td style="text-align: left;">0.84</td>
</tr>
<tr class="odd">
<td style="text-align: left;">stromal_markers</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">6</td>
<td style="text-align: left;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">subtypes</td>
<td style="text-align: right;">69</td>
<td style="text-align: right;">63</td>
<td style="text-align: right;">62</td>
<td style="text-align: right;">69</td>
<td style="text-align: right;">70</td>
<td style="text-align: left;">0.94</td>
</tr>
<tr class="odd">
<td style="text-align: left;">tgf_beta</td>
<td style="text-align: right;">47</td>
<td style="text-align: right;">47</td>
<td style="text-align: right;">42</td>
<td style="text-align: right;">53</td>
<td style="text-align: right;">57</td>
<td style="text-align: left;">0.83</td>
</tr>
<tr class="even">
<td style="text-align: left;">transcriptional_misregulation</td>
<td style="text-align: right;">44</td>
<td style="text-align: right;">37</td>
<td style="text-align: right;">44</td>
<td style="text-align: right;">61</td>
<td style="text-align: right;">63</td>
<td style="text-align: left;">0.74</td>
</tr>
<tr class="odd">
<td style="text-align: left;">triple_negative_biology</td>
<td style="text-align: right;">38</td>
<td style="text-align: right;">32</td>
<td style="text-align: right;">41</td>
<td style="text-align: right;">49</td>
<td style="text-align: right;">50</td>
<td style="text-align: left;">0.8</td>
</tr>
<tr class="even">
<td style="text-align: left;">tumor_metabolism</td>
<td style="text-align: right;">15</td>
<td style="text-align: right;">15</td>
<td style="text-align: right;">14</td>
<td style="text-align: right;">14</td>
<td style="text-align: right;">15</td>
<td style="text-align: left;">0.97</td>
</tr>
<tr class="odd">
<td style="text-align: left;">wnt</td>
<td style="text-align: right;">37</td>
<td style="text-align: right;">34</td>
<td style="text-align: right;">44</td>
<td style="text-align: right;">51</td>
<td style="text-align: right;">51</td>
<td style="text-align: left;">0.81</td>
</tr>
<tr class="even">
<td style="text-align: left;">internal_reference_gene</td>
<td style="text-align: right;">18</td>
<td style="text-align: right;">18</td>
<td style="text-align: right;">16</td>
<td style="text-align: right;">17</td>
<td style="text-align: right;">18</td>
<td style="text-align: left;">0.96</td>
</tr>
</tbody>
</table>


</div>
</div>
<p>We now compare the scores obtained by GSVA on the datasets and see how they relate with nanostring scores obtained with the regressed data.</p>
<p>First we compare GSVA SET ER/PR with ER signaling obtained by the scoring procedure when regressing the data.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-41-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Next we compare EMT, G2M with proliferation and PI3K signaling.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-42-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-43-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-44-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-45-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>In general there is a correlation but the scores are not so well correlated and they are noisy. One that works fairly well is the ER signaling score, but the others not so well. The proliferation score barely shows any difference between luminal A and B patients in any of the cohorts.</p>
<p>Perhaps if we calculate the scores using the hallmarks pathways the scores are better correlated, since they are the same gene sets.</p>
<p>Below we plot the correlation between the proliferation pathway using GSVA and the regressed scoring strategy for the TCGA cohort.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-46-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>It is perfectly correlated.</p>
<p>And for ER signaling:</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-47-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Also it is very much correlated. So the problem in the end is comparing the hallmarks signatures with the nanostring. But interestingly the nanostring proliferation signature was not good to pick up the differences between luminal A and B.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-48-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="risk-stratification-and-the-molecular-landscape" class="level2" data-number="7.6">
<h2 data-number="7.6" class="anchored" data-anchor-id="risk-stratification-and-the-molecular-landscape"><span class="header-section-number">7.6</span> Risk stratification and the molecular landscape</h2>
<p>One common problem in the clinics is what to do with patients that are in the so called intermediate risk group. Here we use the SSP classification from the SCANB team to check what is the position of these patients in the molecular landscape.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-scanb-risk-embedding" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="trying_files/figure-html/fig-scanb-risk-embedding-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Figure&nbsp;7.9: Embedding of the SCANB samples colored by their predicted ROR risk category.</figcaption>
</figure>
</div>
</div>
</div>
<p><a href="#fig-scanb-risk-embedding">Figure&nbsp;<span>7.9</span></a> shows that the embedding detects the positions where the intermediate risk patients are, namely they are in the intersection of the luminal A and B subtypes.</p>
<p>And <a href="#fig-intermediate-pca-position">Figure&nbsp;<span>7.10</span></a> below shows that there is a shift in PC4 for the intermediate to the high risk group, which makes sense as this is exactly where the distinction between luminal A and B are.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-intermediate-pca-position" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="trying_files/figure-html/fig-intermediate-pca-position-1.png" class="img-fluid figure-img" width="768"></p>
<figcaption class="figure-caption">Figure&nbsp;7.10: Distribution of the principal components for the different risk groups defined by the SSP ROR scores.</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="late-distant-recurrence-trying-to-find-biomarkers" class="level2" data-number="7.7">
<h2 data-number="7.7" class="anchored" data-anchor-id="late-distant-recurrence-trying-to-find-biomarkers"><span class="header-section-number">7.7</span> Late distant recurrence: trying to find biomarkers</h2>
<p>One of the key problems in breast cancer research is finding patients that will develop late recurrence. Here we use METABRIC as it has a long follow-up history. We start by plotting the embedding of all METABRIC patients that had a followup longer than 10 years and received only ET. We then try to predict which patients recurr or not based on the molecular data by using lasso regression. For this we use the package <code>glmnet</code> that provides such functionality.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-51-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>In general there is no distinction in terms of the molecular landscape among the patients that recurred, both in terms of molecular subtype and position in the molecular landscape.</p>
<p>We now compare their molecular scores.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-52-1.png" class="img-fluid" width="1920"></p>
</div>
</div>
<p>Even here we don’t see much difference. Unless there is some kind of linear combination of pathways, I don’t think it would be possible to classify these two groups by using only transcriptomic data and these pathways. In any case we proceed now with the lasso regression using the 50 pathways.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-53-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>We see that actually it does not work. And when we predict all the samples are predicted as there is no recurrence.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>.
 no 
407 </code></pre>
</div>
</div>
<p>This is probably because of the unbalanced dataset. Next we try with random forest as well.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
 randomForest(formula = as.factor(is_there_recurrence) ~ ., data = et_metabric_late_data[,      c("is_there_recurrence", pathways_to_compare, "AGE_AT_DIAGNOSIS",          "LYMPH_NODES_EXAMINED_POSITIVE", "npi")], ntree = 500) 
               Type of random forest: classification
                     Number of trees: 500
No. of variables tried at each split: 7

        OOB estimate of  error rate: 17.44%
Confusion matrix:
     no yes class.error
no  336   3 0.008849558
yes  68   0 1.000000000</code></pre>
</div>
</div>
<p>That is not the case, still we can’t predict if a patient will develop recurrence or not in this cohort.</p>
</section>
<section id="correlation-of-pathways-with-proliferation" class="level2" data-number="7.8">
<h2 data-number="7.8" class="anchored" data-anchor-id="correlation-of-pathways-with-proliferation"><span class="header-section-number">7.8</span> Correlation of pathways with proliferation</h2>
<p>One thing that is very common is defining a gene signature and then using this gene signature to calculate survival analysis. The thing is, proliferation related genes are pretty much present in almost any signature and hence all these pathways are correlated to G2M and E2F to some extent <span class="citation" data-cites="Venet2011">(<a href="references.html#ref-Venet2011" role="doc-biblioref">Venet, Dumont, and Detours 2011</a>)</span>.</p>
<p>Here we calculate the pathway scores by using the regressed data and do all pairwise comparison of some of the most important pathways to see their correlations.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-56-1.png" class="img-fluid" width="1632"></p>
</div>
</div>
<p>First we see that G2M and E2F are encoding basically the same information. PI3K AKT MTOR, AR and the Biocarta HER2 and SET ER/PR are all correlated to some extent to proliferation. Surprisingly the two HER2 signaling scores are not able to distinguish the HER2 subtype. The SET ER/PR correlation with G2M is possibly explained by the fact that the basal molecular subtype has higher G2M and lower ER signaling compared to luminal A and B.</p>
</section>
<section id="her2-and-their-categories" class="level2" data-number="7.9">
<h2 data-number="7.9" class="anchored" data-anchor-id="her2-and-their-categories"><span class="header-section-number">7.9</span> HER2 and their categories</h2>
<p>Research on the benefits of giving trastuzumab together with antibody drug conjugates (ADCs) has been shown recently. Patients that are either HER2+ or even considered negative but have some expression of HER2 (1+ and 2+) seem to benefit from ADC + trastuzumab.</p>
<p>Here we first see what are the genes that are leading the differences of the HER2 molecular subtype. We query 7 genes that are available in the list of 8+ regions and are in the amplicon region of ERBB2 when this gene is amplified (HER2+).</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-57-1.png" class="img-fluid" width="1728"></p>
</div>
</div>
<p>First we notice that ERBB2 is actually not the gene that is driving the differences, perhaps because when the embedding was learned during the PCA this gene was not used to make the differences between the molecular subtype. Instead, the genes that are mostly capturing the difference of the HER2-enriched PAM50 molecular subtype are FBXL20, GRB7, MED1 and STARD3.</p>
<p>And if we check by HER2 status according to the available datasets.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-her2-genes" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="trying_files/figure-html/fig-her2-genes-1.png" class="img-fluid figure-img" width="1536"></p>
<figcaption class="figure-caption">Figure&nbsp;7.11: Regressed gene expression levels and comparison between the HER2 groups.</figcaption>
</figure>
</div>
</div>
</div>
<p>Similar to what was seen before for ER status you see differences for the HER2+ patients. Note that several of the patients that have gain of HER2 have similar gene expression levels to those that have loss or are neutral. Moreover, the top genes that are differentiating the HER2 status are STARD3, GRB7 and MED1. The differences in HER2 are actually very small, but it looks like there is a consistent difference still.</p>
<p>Just before we move on we compare the expression levels before and after regressing the data on SCANB to see if we are not messing up too much with the gene expression levels and getting just noise.</p>
<p>The plot below shows the correlation between ERBB2 FPKM and regressed data and also for GRB7. We further show the correlation between GRB7 and ERBB2 in the regressed data.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-59-1.png" class="img-fluid" width="768"></p>
</div>
</div>
<p>We see a good correlation between FPKM and regressed data. Moreover there is a good correlation between GRB7 and HER2 in both regressed data and FPKM. We now define a new score based on these genes available. The score is calculated in the same way as before the regressed data, i.e., the sum of the expression levels.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-her2-amplicon-her2subgroups" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="trying_files/figure-html/fig-her2-amplicon-her2subgroups-1.png" class="img-fluid figure-img" width="768"></p>
<figcaption class="figure-caption">Figure&nbsp;7.12: HER2 AMPLICON signature stratified by the HER2 status</figcaption>
</figure>
</div>
</div>
</div>
<p>And now plotting for the molecular subtypes specifically.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pam50-her2-amplicon" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="trying_files/figure-html/fig-pam50-her2-amplicon-1.png" class="img-fluid figure-img" width="960"></p>
<figcaption class="figure-caption">Figure&nbsp;7.13: HER2 AMPLICON signature stratified by the PAM50 molecular subtypes</figcaption>
</figure>
</div>
</div>
</div>
<p>This is a signature that captures better the HER2 molecular subtype, but not in its entirety.</p>
<p>Similar to what we did previously, we check if this signature is correlated with proliferation.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-62-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>If there is any correlation it is too small and not clear. For comparison we do the same plot for the HER2 positive patients.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-63-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Again if there is any kind of correlation it is weak. But notice how the majority of the samples goes from 0 to 5. The maximum that a non HER2 positive sample goes is 2. Below is the density plot for the HER2+s.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-64-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Since when giving aromatase inhibitors to patients we are shutting down ER signaling or at least reducing it to some extent, so we expect to see alternative pathways taking the lead so the cells can keep proliferating. Here we calculate the difference between HER2 amplicon signature before and after treatment for responders and non-responders.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-65-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>There is a very small difference between the two groups with responders having a bit higher difference, meaning higher values of the HER2 AMPLICON signature after treatment in average.</p>
<section id="her2-amplicon-and-survival-analysis" class="level3" data-number="7.9.1">
<h3 data-number="7.9.1" class="anchored" data-anchor-id="her2-amplicon-and-survival-analysis"><span class="header-section-number">7.9.1</span> HER2 Amplicon and survival analysis</h3>
<p>Next we evaluate the HER2 amplicon signature in patients that received trastuzumab + chemo + endo or just trastuzumab + chemo, and have HER2+ and ER+ BC. The hypothesis here is that the higher the signalling the better the outcome. In the SCAN-B dataset we are making a big assumption in that the immunotherapy available on <code>TreatGroup</code> corresponds to herceptin in this context for patients with HER2+ BC.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>    TreatGroup Negative Positive
         Chemo        3        1
     ChemoEndo        2       19
          Endo        3       88
     ImmuChemo      209       12
 ImmuChemoEndo       14      461
          None       28       14
          &lt;NA&gt;       28       29</code></pre>
</div>
</div>
<p>Above the number of patients with HER2+ BC stratified by ER status and treatment group. Majority is ER+ and received endocrine therapy as well.</p>
<p>The table below shows the number of events (<span class="math inline">\(rfs\_status = 2\)</span>) for patients with HER2+ ER+ BC.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>    TreatGroup   1  2 NA_
         Chemo   0  1   0
     ChemoEndo  10  3   6
          Endo  57 15  16
     ImmuChemo   8  1   3
 ImmuChemoEndo 337 33  91
          None   8  4   2
          &lt;NA&gt;  16  3  10</code></pre>
</div>
</div>
<p>There is a considerable amount of patients with recurrence. The median recurrence free survival for those patients that received immu + chemo + endo is:</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>    median
1 67.91667</code></pre>
</div>
</div>
<p>i.e., 67 months or 5.6 years. We now calculate the survival analysis. And for the patients that have ER- BC and received only immuno + chemo:</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>    median
1 63.93333</code></pre>
</div>
</div>
<p>Similar as for the ER+ BC patients.</p>
<p>Below are the results from the analysis using the HER2 AMPLICON signature for both ER- and ER+ BC patients separately.</p>
<div class="cell">
<div class="cell-output-display">

<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-759ad7beef87b79255b4" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-759ad7beef87b79255b4">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14"],["Negative","Negative","Negative","Negative","Negative","Negative","Negative","Positive","Positive","Positive","Positive","Positive","Positive","Positive"],["HER2_AMPLICON   ","age             ","node_stageN1    ","node_stageN2and3","node_stageNX    ","tumor_stageT2   ","tumor_stageT3   ","HER2_AMPLICON   ","age             ","node_stageN1    ","node_stageN2and3","node_stageNX    ","tumor_stageT2   ","tumor_stageT3   "],["0.768","1.017","0.568","0.911","1.056","1.330","8.586","0.880","0.995","2.159","5.415","4.297","1.627","2.089"],["0.514","0.977","0.169","0.260","0.130","0.501","1.799","0.649","0.966","0.827","2.018","1.286","0.762","0.436"],[" 1.15"," 1.06"," 1.90"," 3.19"," 8.56"," 3.53","40.99"," 1.19"," 1.02"," 5.64","14.53","14.36"," 3.48","10.00"],["0.2046","0.0205","0.6170","0.6403","1.0673","0.4984","0.7975","0.1554","0.0147","0.4897","0.5035","0.6155","0.3873","0.7989"],["0.197657","0.410717","0.358885","0.884096","0.958936","0.567614","0.007018","0.412430","0.712021","0.116109","0.000794","0.017849","0.208622","0.356444"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>er_status<\/th>\n      <th>pathway<\/th>\n      <th>estimate<\/th>\n      <th>conf.low<\/th>\n      <th>conf.high<\/th>\n      <th>std.error<\/th>\n      <th>p.value<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div class="cell">
<div class="cell-output-display">
<div id="fig-her2-amplicon-rfs-scanb" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="trying_files/figure-html/fig-her2-amplicon-rfs-scanb-1.png" class="img-fluid figure-img" width="816"></p>
<figcaption class="figure-caption">Figure&nbsp;7.14: Hazard ratio and confidence interval for the HER2 AMPLICON signature.</figcaption>
</figure>
</div>
</div>
</div>
<p>For both ER- and ER+ BC patients, the higher the signature the better the outcome. The signature is noisier for ER+ tumors, I wonder if it is due to treatment selection or the fact we assumed that all immunotherapies are herceptin when in fact they are not.</p>
<p>Next we compare to using just HER2 FPKM as a measure. <a href="#fig-her2-amplicon-fpkm-rfs-scanb">Figure&nbsp;<span>7.15</span></a> shows that the HER2 amplicon improves in the estimation of the hazard ratio for such a signature and how including more genes might be important.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-her2-amplicon-fpkm-rfs-scanb" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="trying_files/figure-html/fig-her2-amplicon-fpkm-rfs-scanb-1.png" class="img-fluid figure-img" width="816"></p>
<figcaption class="figure-caption">Figure&nbsp;7.15: Hazard ratio and confidence interval for the HER2 AMPLICON signature.</figcaption>
</figure>
</div>
</div>
</div>
<p>These scores lead to similar hazard ratios and this is due to how correlated they are.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-corr-her2-fpkm-amplicon" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="trying_files/figure-html/fig-corr-her2-fpkm-amplicon-1.png" class="img-fluid figure-img" width="768"></p>
<figcaption class="figure-caption">Figure&nbsp;7.16: Correlation between HER2 FPKM and HER2 amplicon among HER2+ BC tumors</figcaption>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="ilc-and-position-in-the-molecular-landscape" class="level2" data-number="7.10">
<h2 data-number="7.10" class="anchored" data-anchor-id="ilc-and-position-in-the-molecular-landscape"><span class="header-section-number">7.10</span> ILC and position in the molecular landscape</h2>
<p>Invasive lobular carcinoma is a subtype of ER+ BC. They tend to respond better to ET in the short term but they recurr after 20 years. Here we just show where the ILC BC lie in the molecular landscape.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pca-lobular-scanb" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="trying_files/figure-html/fig-pca-lobular-scanb-1.png" class="img-fluid figure-img" width="864"></p>
<figcaption class="figure-caption">Figure&nbsp;7.17: Embedding of the lobular samples from SCANB on top METABRIC and TCGA</figcaption>
</figure>
</div>
</div>
</div>
<p>We see that they are mostly in the normal and luminal A region. Still there are several samples that are luminal B, meaning they are more proliferative.</p>
<p>Is there any difference in the SET ER/PR for lobulars when comparing to ER+ HER2- Ductal carcinomas?</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-75-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>They seem to be the same, also when checking the risk score calculated for all patients, the lobular patients don’t seem to have a decreased risk overall, it is mostly because they are in the same region of the molecular landscape. Interestingly there is way less luminal B patients in the lobular, but still their distributions are very similar.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-76-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>And we also plot by intrinsic molecular subtype.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-77-1.png" class="img-fluid" width="1344"></p>
</div>
</div>
<p>We also compare EMT.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-78-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>EMT is very similar across both histological subtypes. The plot below shows the densities for ER percentage, notice how the two overlap.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-79-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>We now check the lobular signature LobSig that is touted to be superior to OncotypeDX, Prosigna’s PAM50 and other signatures. In total, 63% of the genes are available here. Based on my experience this is already a good number to make comparisons. For example, G2M checkpoint has 71% of the genes available, estrogen response early has 66%.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-80-1.png" class="img-fluid" width="960"></p>
</div>
</div>
<p>We see that the LobSig is highgly correlated with G2M checkpoint and also the risk of recurrence score obtained from PAM50. The random 200 is a signature of 200 random genes that works as a negative control.</p>
<p>The figure below shows the distinction of the signature among the 3 different risk categories from ROR.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-81-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>There is an increase in the score as expected based on the previous results that show where the intermediate risk group is according to the molecular landscape.</p>
<p>Lastly we check the CDH1 expression levels on the regressed data to see if the differences are still captured upon the whole process.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-82-1.png" class="img-fluid" width="1344"></p>
</div>
</div>
<p>The differences are still captured, and CDH1 is decreased when comparing lobular vs NST. Note also that the difference seems to be the biggest in the luminal B subtype.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-83-1.png" class="img-fluid" width="1344"></p>
</div>
</div>
<p>We also check the PC3 and PC4 loadings for CDH1, to see how important this gene is in the molecular landscape.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>  gene          PC3        PC4
1 CDH1 0.0002872375 0.01509738</code></pre>
</div>
</div>
<p>PC4 has a relatively high value which puts it into the 95 percentile as shown below.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.9547769</code></pre>
</div>
</div>
<p>Next we can plot the lobular samples from Ciriello’s paper and color by the 3 subtypes they found. Note here that there is a mismatch between the cbioportal data and what is found in the original papers <span class="citation" data-cites="Curtis2012">(<a href="references.html#ref-Curtis2012" role="doc-biblioref">Curtis et al. 2012</a>)</span>. We will use the Curtis 2012 annotation as this is what matches Ciriello’s paper. Surprisingly the mismatch is not a systematic issue, rather 88 ILC samples are matching and the non-matching samples are NST but the other info matches usually, such as NPI.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pca-lobular-metabric" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="trying_files/figure-html/fig-pca-lobular-metabric-1.png" class="img-fluid figure-img" width="864"></p>
<figcaption class="figure-caption">Figure&nbsp;7.18: Embedding of the lobular samples from METABRIC on top SCANB and TCGA</figcaption>
</figure>
</div>
</div>
</div>
<p>The table below shows the number of subtypes for each molecular subtype. In general there are samples in each one of the molecular subtypes. The reactive-like is actually more enriched in the normal region.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>        subtype basal claudin-low her2 luma lumb normal
 Immune-Related     0           3    2   27   10      4
  Proliferative     3           3    5   11   15      4
  Reactive-like     0           5    1   22    2     20</code></pre>
</div>
</div>
<p>And below is a plot of the lobular samples from Ciriello/Curtis with the molecular subtypes defined by Ciriello.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-89-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>It seems that more proliferative saaples are in the luminal B region whereas the Reactive-like and Immune-related are not so distinguishable, they are in the luminal A region.</p>
<p>Some of the samples are NST on cbioportal and Lobular on Curtis. We selected the following samples and checked on cbioPortal (https://www.cbioportal.org/study/clinicalData?id=brca_metabric) their histologic subtype directly.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>  sample_name   npi
1     MB-0035 3.056
2     MB-0050 4.066
3     MB-0083 3.026
4     MB-0101 3.068
5     MB-0102 5.080
6     MB-0112 6.300</code></pre>
</div>
</div>
<p>All the 6 samples are Ductal/NST and not Lobular.</p>
</section>
<section id="robustness-to-library-protocol" class="level2" data-number="7.11">
<h2 data-number="7.11" class="anchored" data-anchor-id="robustness-to-library-protocol"><span class="header-section-number">7.11</span> Robustness to library protocol</h2>
<p>The figure below shows the first two components for SCANB stratified by the library protocol used in the study. We see that PC2 is partially capturing the differences due to library protocol.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-91-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="distance-in-4th-component-for-poetic" class="level2" data-number="7.12">
<h2 data-number="7.12" class="anchored" data-anchor-id="distance-in-4th-component-for-poetic"><span class="header-section-number">7.12</span> Distance in 4th component for POETIC</h2>
<p>We have seen previosuly that there is a bigger difference in between the matched samples in the responder than to non-responder patients. Hallmark G2M checkpoint is one of the signatures driving the PC4, meaning that the position wrt the fourth component might be used to define a new response status.</p>
<p>Below we plot the matched samples for all non responder patients with no change at all in Ki67 (change &gt;= 0%) and their respective positions in the molecular landscape.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-92-1.png" class="img-fluid" width="1536"></p>
</div>
</div>
<p>Now we do the same for the top responders (change in Ki67 <span class="math inline">\(\leq\)</span> -90%).</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-93-1.png" class="img-fluid" width="1536"></p>
</div>
</div>
<p>In general the distance for the responder samples seem to be higher and not only that we see that the surgery sample goes to the bottom part of the embedding, suggesting that the reduction of the proliferation is reflected in the molecular landscape.</p>


<!-- -->

<div id="refs" class="references csl-bib-body hanging-indent" role="list" style="display: none">
<div id="ref-Curtis2012" class="csl-entry" role="listitem">
Curtis, Christina, and Sohrab P. Shah, Suet-Feung Chin, Gulisa Turashvili, Oscar M. Rueda, Mark J. Dunning, Doug Speed, et al. 2012. <span>“The Genomic and Transcriptomic Architecture of 2, 000 Breast Tumours Reveals Novel Subgroups.”</span> <em>Nature</em> 486 (7403): 346–52. <a href="https://doi.org/10.1038/nature10983">https://doi.org/10.1038/nature10983</a>.
</div>
<div id="ref-Venet2011" class="csl-entry" role="listitem">
Venet, David, Jacques E. Dumont, and Vincent Detours. 2011. <span>“Most Random Gene Expression Signatures Are Significantly Associated with Breast Cancer Outcome.”</span> Edited by Isidore Rigoutsos. <em><span>PLoS</span> Computational Biology</em> 7 (10): e1002240. <a href="https://doi.org/10.1371/journal.pcbi.1002240">https://doi.org/10.1371/journal.pcbi.1002240</a>.
</div>
</div>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="../chapters/expanding_ember.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Expanding embedding and removing batch effects</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../chapters/discussion.html" class="pagination-link">
        <span class="nav-page-text">Discussion</span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb19" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="fu"># Tests and tries </span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>This is a file where we try different things using the molecular data.</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>We keep it here in case anyone finds anything here interesting and would like</span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>to further discuss with us for potential collaborations.</span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a><span class="in">```{r setup}</span></span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a>renv<span class="sc">::</span><span class="fu">restore</span>()</span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(PCAtools)</span>
<span id="cb19-14"><a href="#cb19-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-15"><a href="#cb19-15" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(singscore)</span>
<span id="cb19-16"><a href="#cb19-16" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SummarizedExperiment)</span>
<span id="cb19-17"><a href="#cb19-17" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(survival)</span>
<span id="cb19-18"><a href="#cb19-18" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(genefu)</span>
<span id="cb19-19"><a href="#cb19-19" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(caret)</span>
<span id="cb19-20"><a href="#cb19-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-21"><a href="#cb19-21" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"../R/utils.R"</span>)</span>
<span id="cb19-22"><a href="#cb19-22" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"../R/first_run.R"</span>)</span>
<span id="cb19-23"><a href="#cb19-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-24"><a href="#cb19-24" aria-hidden="true" tabindex="-1"></a><span class="co"># the following script load all data necessary to run the chunks.</span></span>
<span id="cb19-25"><a href="#cb19-25" aria-hidden="true" tabindex="-1"></a><span class="co"># the data is generated from this quarto document itself, therefore</span></span>
<span id="cb19-26"><a href="#cb19-26" aria-hidden="true" tabindex="-1"></a><span class="co"># if you are running this documents the first time and don't have the</span></span>
<span id="cb19-27"><a href="#cb19-27" aria-hidden="true" tabindex="-1"></a><span class="co"># files, comment the following lines. Moreover, if this is your first</span></span>
<span id="cb19-28"><a href="#cb19-28" aria-hidden="true" tabindex="-1"></a><span class="co"># time running the document, you should run all chunks, to generate </span></span>
<span id="cb19-29"><a href="#cb19-29" aria-hidden="true" tabindex="-1"></a><span class="co"># all the necessary files, if you don't have them. Once all files </span></span>
<span id="cb19-30"><a href="#cb19-30" aria-hidden="true" tabindex="-1"></a><span class="co"># are saved and available in the respective folder, the following</span></span>
<span id="cb19-31"><a href="#cb19-31" aria-hidden="true" tabindex="-1"></a><span class="co"># lines can be executed. </span></span>
<span id="cb19-32"><a href="#cb19-32" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (first_run){</span>
<span id="cb19-33"><a href="#cb19-33" aria-hidden="true" tabindex="-1"></a>    load_at_setup <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb19-34"><a href="#cb19-34" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb19-35"><a href="#cb19-35" aria-hidden="true" tabindex="-1"></a>    load_at_setup <span class="ot">&lt;-</span> <span class="cn">TRUE</span></span>
<span id="cb19-36"><a href="#cb19-36" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb19-37"><a href="#cb19-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-38"><a href="#cb19-38" aria-hidden="true" tabindex="-1"></a>name_document <span class="ot">&lt;-</span> <span class="st">"trying"</span></span>
<span id="cb19-39"><a href="#cb19-39" aria-hidden="true" tabindex="-1"></a><span class="fu">sapply</span>(</span>
<span id="cb19-40"><a href="#cb19-40" aria-hidden="true" tabindex="-1"></a>    <span class="fu">paste0</span>(<span class="st">"../../results/"</span>, <span class="fu">c</span>(<span class="st">"plots"</span>, <span class="st">"rds_files"</span>, <span class="st">"tables"</span>), <span class="st">"/"</span>, name_document),</span>
<span id="cb19-41"><a href="#cb19-41" aria-hidden="true" tabindex="-1"></a>    dir.create,</span>
<span id="cb19-42"><a href="#cb19-42" aria-hidden="true" tabindex="-1"></a>    <span class="at">showWarnings =</span> <span class="cn">FALSE</span>,</span>
<span id="cb19-43"><a href="#cb19-43" aria-hidden="true" tabindex="-1"></a>    <span class="at">recursive =</span> <span class="cn">TRUE</span></span>
<span id="cb19-44"><a href="#cb19-44" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-45"><a href="#cb19-45" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"../R/load_rds_files.R"</span>)</span>
<span id="cb19-46"><a href="#cb19-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-47"><a href="#cb19-47" aria-hidden="true" tabindex="-1"></a><span class="co"># by setting the dev to png and pdf, this saves the figures in a specific</span></span>
<span id="cb19-48"><a href="#cb19-48" aria-hidden="true" tabindex="-1"></a><span class="co"># folder in both formats. moreover, since png is coming first, it shows</span></span>
<span id="cb19-49"><a href="#cb19-49" aria-hidden="true" tabindex="-1"></a><span class="co"># this figure when rendering the html. What is nice about this is that it</span></span>
<span id="cb19-50"><a href="#cb19-50" aria-hidden="true" tabindex="-1"></a><span class="co"># inherits the properties from the chunk to save the figure, so no need</span></span>
<span id="cb19-51"><a href="#cb19-51" aria-hidden="true" tabindex="-1"></a><span class="co"># to use ggplot2::ggsave to save the plots. this also works </span></span>
<span id="cb19-52"><a href="#cb19-52" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">dev =</span> <span class="fu">c</span>(<span class="st">'png'</span>, <span class="st">'pdf'</span>))</span>
<span id="cb19-53"><a href="#cb19-53" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">bitmapType =</span> <span class="st">'cairo'</span>)</span>
<span id="cb19-54"><a href="#cb19-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-55"><a href="#cb19-55" aria-hidden="true" tabindex="-1"></a>p4_pathways <span class="ot">&lt;-</span> msigdbr<span class="sc">::</span><span class="fu">msigdbr</span>() <span class="sc">%&gt;%</span></span>
<span id="cb19-56"><a href="#cb19-56" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(gs_name <span class="sc">%in%</span> <span class="fu">c</span>(</span>
<span id="cb19-57"><a href="#cb19-57" aria-hidden="true" tabindex="-1"></a>        <span class="st">"WILCOX_RESPONSE_TO_PROGESTERONE_UP"</span>,</span>
<span id="cb19-58"><a href="#cb19-58" aria-hidden="true" tabindex="-1"></a>        <span class="st">"GOBP_RESPONSE_TO_PROGESTERONE"</span>,</span>
<span id="cb19-59"><a href="#cb19-59" aria-hidden="true" tabindex="-1"></a>        <span class="st">"BIOCARTA_HER2_PATHWAY"</span>,</span>
<span id="cb19-60"><a href="#cb19-60" aria-hidden="true" tabindex="-1"></a>        <span class="st">"KEGG_ERBB_SIGNALING_PATHWAY"</span></span>
<span id="cb19-61"><a href="#cb19-61" aria-hidden="true" tabindex="-1"></a>    ))</span>
<span id="cb19-62"><a href="#cb19-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-63"><a href="#cb19-63" aria-hidden="true" tabindex="-1"></a>gene_sets_prog <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(</span>
<span id="cb19-64"><a href="#cb19-64" aria-hidden="true" tabindex="-1"></a>    gene_sets, </span>
<span id="cb19-65"><a href="#cb19-65" aria-hidden="true" tabindex="-1"></a>    p4_pathways</span>
<span id="cb19-66"><a href="#cb19-66" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-67"><a href="#cb19-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-68"><a href="#cb19-68" aria-hidden="true" tabindex="-1"></a>gene_sets_ <span class="ot">&lt;-</span> <span class="fu">sapply</span>(</span>
<span id="cb19-69"><a href="#cb19-69" aria-hidden="true" tabindex="-1"></a>    gene_sets_prog<span class="sc">$</span>gs_name <span class="sc">%&gt;%</span> unique,</span>
<span id="cb19-70"><a href="#cb19-70" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(x, gene_sets) gene_sets <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(gs_name <span class="sc">==</span> x) <span class="sc">%&gt;%</span> </span>
<span id="cb19-71"><a href="#cb19-71" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">pull</span>(gene_symbol),</span>
<span id="cb19-72"><a href="#cb19-72" aria-hidden="true" tabindex="-1"></a>    <span class="at">gene_sets =</span> gene_sets_prog,</span>
<span id="cb19-73"><a href="#cb19-73" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>,</span>
<span id="cb19-74"><a href="#cb19-74" aria-hidden="true" tabindex="-1"></a>    <span class="at">simplify =</span> <span class="cn">FALSE</span></span>
<span id="cb19-75"><a href="#cb19-75" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-76"><a href="#cb19-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-77"><a href="#cb19-77" aria-hidden="true" tabindex="-1"></a>datasets<span class="sc">$</span>poetic<span class="sc">$</span>sample_name <span class="ot">&lt;-</span> <span class="fu">colnames</span>(datasets<span class="sc">$</span>poetic)</span>
<span id="cb19-78"><a href="#cb19-78" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-79"><a href="#cb19-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-80"><a href="#cb19-80" aria-hidden="true" tabindex="-1"></a><span class="fu">## Using several pathways or just one pathway for GSVA</span></span>
<span id="cb19-81"><a href="#cb19-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-82"><a href="#cb19-82" aria-hidden="true" tabindex="-1"></a>We now evaluate the effect of the scores for each pathway when using all the</span>
<span id="cb19-83"><a href="#cb19-83" aria-hidden="true" tabindex="-1"></a>53 pathways combined and each one individually. We use only TCGA for </span>
<span id="cb19-84"><a href="#cb19-84" aria-hidden="true" tabindex="-1"></a>this step.</span>
<span id="cb19-85"><a href="#cb19-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-88"><a href="#cb19-88" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-89"><a href="#cb19-89" aria-hidden="true" tabindex="-1"></a>gene_expression <span class="ot">&lt;-</span> <span class="fu">assay</span>(datasets<span class="sc">$</span>tcga, which_exp<span class="sc">$</span>tcga) <span class="sc">%&gt;%</span> as.matrix</span>
<span id="cb19-90"><a href="#cb19-90" aria-hidden="true" tabindex="-1"></a>all_scores <span class="ot">&lt;-</span> <span class="fu">colData</span>(datasets<span class="sc">$</span>tcga) <span class="sc">%&gt;%</span> data.frame</span>
<span id="cb19-91"><a href="#cb19-91" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-92"><a href="#cb19-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-93"><a href="#cb19-93" aria-hidden="true" tabindex="-1"></a><span class="fu">### One vs all</span></span>
<span id="cb19-94"><a href="#cb19-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-95"><a href="#cb19-95" aria-hidden="true" tabindex="-1"></a>We now calculate the scores for each gene set individually.</span>
<span id="cb19-96"><a href="#cb19-96" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-97"><a href="#cb19-97" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb19-98"><a href="#cb19-98" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">187293</span>)</span>
<span id="cb19-99"><a href="#cb19-99" aria-hidden="true" tabindex="-1"></a>samples_one_pathway <span class="ot">&lt;-</span> <span class="fu">sample</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(gene_expression), <span class="at">size =</span> <span class="dv">200</span>)</span>
<span id="cb19-100"><a href="#cb19-100" aria-hidden="true" tabindex="-1"></a>gsva_one_pathway <span class="ot">&lt;-</span> <span class="fu">lapply</span>(</span>
<span id="cb19-101"><a href="#cb19-101" aria-hidden="true" tabindex="-1"></a>    gene_sets_[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>],</span>
<span id="cb19-102"><a href="#cb19-102" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(x){</span>
<span id="cb19-103"><a href="#cb19-103" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb19-104"><a href="#cb19-104" aria-hidden="true" tabindex="-1"></a>        GSVA<span class="sc">::</span><span class="fu">gsva</span>(</span>
<span id="cb19-105"><a href="#cb19-105" aria-hidden="true" tabindex="-1"></a>            <span class="at">expr =</span> gene_expression[, samples_one_pathway], </span>
<span id="cb19-106"><a href="#cb19-106" aria-hidden="true" tabindex="-1"></a>            <span class="at">gset.idx.list =</span> <span class="fu">list</span>(x),</span>
<span id="cb19-107"><a href="#cb19-107" aria-hidden="true" tabindex="-1"></a>            <span class="at">verbose =</span> <span class="cn">FALSE</span></span>
<span id="cb19-108"><a href="#cb19-108" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb19-109"><a href="#cb19-109" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb19-110"><a href="#cb19-110" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-111"><a href="#cb19-111" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-112"><a href="#cb19-112" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb19-113"><a href="#cb19-113" aria-hidden="true" tabindex="-1"></a>    gsva_one_pathway,</span>
<span id="cb19-114"><a href="#cb19-114" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../../results/rds_files/trying/gsva_one_pathway.rds"</span></span>
<span id="cb19-115"><a href="#cb19-115" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-116"><a href="#cb19-116" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-117"><a href="#cb19-117" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-120"><a href="#cb19-120" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-121"><a href="#cb19-121" aria-hidden="true" tabindex="-1"></a>gsva_one_pathways <span class="ot">&lt;-</span> <span class="fu">Reduce</span>(rbind, gsva_one_pathway) <span class="sc">%&gt;%</span></span>
<span id="cb19-122"><a href="#cb19-122" aria-hidden="true" tabindex="-1"></a>    t <span class="sc">%&gt;%</span></span>
<span id="cb19-123"><a href="#cb19-123" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span> </span>
<span id="cb19-124"><a href="#cb19-124" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">colnames&lt;-</span><span class="st">`</span>(<span class="fu">names</span>(gsva_one_pathway)) <span class="sc">%&gt;%</span></span>
<span id="cb19-125"><a href="#cb19-125" aria-hidden="true" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"sample_name"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-126"><a href="#cb19-126" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(</span>
<span id="cb19-127"><a href="#cb19-127" aria-hidden="true" tabindex="-1"></a>        <span class="at">cols =</span> dplyr<span class="sc">::</span><span class="fu">all_of</span>(<span class="fu">names</span>(gsva_one_pathway)),</span>
<span id="cb19-128"><a href="#cb19-128" aria-hidden="true" tabindex="-1"></a>        <span class="at">names_to =</span> <span class="st">"pathway"</span>,</span>
<span id="cb19-129"><a href="#cb19-129" aria-hidden="true" tabindex="-1"></a>        <span class="at">values_to =</span> <span class="st">"one_gsva"</span></span>
<span id="cb19-130"><a href="#cb19-130" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb19-131"><a href="#cb19-131" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">inner_join</span>(</span>
<span id="cb19-132"><a href="#cb19-132" aria-hidden="true" tabindex="-1"></a>        merged_col_data <span class="sc">%&gt;%</span></span>
<span id="cb19-133"><a href="#cb19-133" aria-hidden="true" tabindex="-1"></a>            tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(</span>
<span id="cb19-134"><a href="#cb19-134" aria-hidden="true" tabindex="-1"></a>                <span class="at">cols =</span> dplyr<span class="sc">::</span><span class="fu">all_of</span>(<span class="fu">intersect</span>(</span>
<span id="cb19-135"><a href="#cb19-135" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">names</span>(gsva_one_pathway),</span>
<span id="cb19-136"><a href="#cb19-136" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">colnames</span>(merged_col_data)</span>
<span id="cb19-137"><a href="#cb19-137" aria-hidden="true" tabindex="-1"></a>                )),</span>
<span id="cb19-138"><a href="#cb19-138" aria-hidden="true" tabindex="-1"></a>                <span class="at">names_to =</span> <span class="st">"pathway"</span>,</span>
<span id="cb19-139"><a href="#cb19-139" aria-hidden="true" tabindex="-1"></a>                <span class="at">values_to =</span> <span class="st">"gsva"</span></span>
<span id="cb19-140"><a href="#cb19-140" aria-hidden="true" tabindex="-1"></a>            ),</span>
<span id="cb19-141"><a href="#cb19-141" aria-hidden="true" tabindex="-1"></a>        .,</span>
<span id="cb19-142"><a href="#cb19-142" aria-hidden="true" tabindex="-1"></a>        <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"sample_name"</span>, <span class="st">"pathway"</span>)</span>
<span id="cb19-143"><a href="#cb19-143" aria-hidden="true" tabindex="-1"></a>    ) </span>
<span id="cb19-144"><a href="#cb19-144" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-145"><a href="#cb19-145" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-146"><a href="#cb19-146" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=20, fig.height=16, dev="png"}</span></span>
<span id="cb19-147"><a href="#cb19-147" aria-hidden="true" tabindex="-1"></a>gsva_one_pathways <span class="sc">%&gt;%</span></span>
<span id="cb19-148"><a href="#cb19-148" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(</span>
<span id="cb19-149"><a href="#cb19-149" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> one_gsva, <span class="at">y =</span> gsva, <span class="at">color =</span> pam50</span>
<span id="cb19-150"><a href="#cb19-150" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">+</span></span>
<span id="cb19-151"><a href="#cb19-151" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb19-152"><a href="#cb19-152" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_abline</span>(<span class="at">intercept =</span> <span class="dv">0</span>, <span class="at">slope =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb19-153"><a href="#cb19-153" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>pathway, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb19-154"><a href="#cb19-154" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>()</span>
<span id="cb19-155"><a href="#cb19-155" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-156"><a href="#cb19-156" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-157"><a href="#cb19-157" aria-hidden="true" tabindex="-1"></a>It seems they are all positively correlated, so one could calculate the </span>
<span id="cb19-158"><a href="#cb19-158" aria-hidden="true" tabindex="-1"></a>scores basically using just one gene set with exception of some of the</span>
<span id="cb19-159"><a href="#cb19-159" aria-hidden="true" tabindex="-1"></a>gene sets. </span>
<span id="cb19-160"><a href="#cb19-160" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-161"><a href="#cb19-161" aria-hidden="true" tabindex="-1"></a><span class="fu">## Minimum number of samples for GSVA</span></span>
<span id="cb19-162"><a href="#cb19-162" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-163"><a href="#cb19-163" aria-hidden="true" tabindex="-1"></a>According to the original GSVA paper, a minimum of 10 samples are necessary</span>
<span id="cb19-164"><a href="#cb19-164" aria-hidden="true" tabindex="-1"></a>to have a good power when using GSVA. Since here we have at least 5 </span>
<span id="cb19-165"><a href="#cb19-165" aria-hidden="true" tabindex="-1"></a>molecular subgroups, we hypothesize we need at least 50 samples to get a </span>
<span id="cb19-166"><a href="#cb19-166" aria-hidden="true" tabindex="-1"></a>good score so we can compare across cohorts. We will try with</span>
<span id="cb19-167"><a href="#cb19-167" aria-hidden="true" tabindex="-1"></a>different numbers of samples using TCGA, we start with 10, 25, 50 and</span>
<span id="cb19-168"><a href="#cb19-168" aria-hidden="true" tabindex="-1"></a>100 samples and we compare with the scores using the full dataset.</span>
<span id="cb19-169"><a href="#cb19-169" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-172"><a href="#cb19-172" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-173"><a href="#cb19-173" aria-hidden="true" tabindex="-1"></a>nb_of_samples <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">10</span>, <span class="dv">25</span>, <span class="dv">40</span>, <span class="dv">50</span>, <span class="dv">100</span>, <span class="dv">150</span>)</span>
<span id="cb19-174"><a href="#cb19-174" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-175"><a href="#cb19-175" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-176"><a href="#cb19-176" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb19-177"><a href="#cb19-177" aria-hidden="true" tabindex="-1"></a>seeds <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">12</span>, <span class="dv">123</span>, <span class="dv">1234</span>, <span class="dv">21345</span>, <span class="dv">12</span>, <span class="dv">123</span>)</span>
<span id="cb19-178"><a href="#cb19-178" aria-hidden="true" tabindex="-1"></a>gsva_scores <span class="ot">&lt;-</span> <span class="fu">mapply</span>(</span>
<span id="cb19-179"><a href="#cb19-179" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(nb_samples, seed, df, gene_sets, col_data){</span>
<span id="cb19-180"><a href="#cb19-180" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb19-181"><a href="#cb19-181" aria-hidden="true" tabindex="-1"></a>        <span class="fu">set.seed</span>(seed)</span>
<span id="cb19-182"><a href="#cb19-182" aria-hidden="true" tabindex="-1"></a>        patients <span class="ot">&lt;-</span> col_data <span class="sc">%&gt;%</span></span>
<span id="cb19-183"><a href="#cb19-183" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">slice_sample</span>(<span class="at">n =</span> nb_samples) <span class="sc">%&gt;%</span></span>
<span id="cb19-184"><a href="#cb19-184" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">pull</span>(sample_name)</span>
<span id="cb19-185"><a href="#cb19-185" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb19-186"><a href="#cb19-186" aria-hidden="true" tabindex="-1"></a>        GSVA<span class="sc">::</span><span class="fu">gsva</span>(</span>
<span id="cb19-187"><a href="#cb19-187" aria-hidden="true" tabindex="-1"></a>            df[, patients],</span>
<span id="cb19-188"><a href="#cb19-188" aria-hidden="true" tabindex="-1"></a>            gene_sets,</span>
<span id="cb19-189"><a href="#cb19-189" aria-hidden="true" tabindex="-1"></a>            <span class="at">verbose =</span> <span class="cn">FALSE</span></span>
<span id="cb19-190"><a href="#cb19-190" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb19-191"><a href="#cb19-191" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb19-192"><a href="#cb19-192" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb19-193"><a href="#cb19-193" aria-hidden="true" tabindex="-1"></a>    <span class="at">nb_samples =</span> nb_of_samples,</span>
<span id="cb19-194"><a href="#cb19-194" aria-hidden="true" tabindex="-1"></a>    <span class="at">seed =</span> seeds,</span>
<span id="cb19-195"><a href="#cb19-195" aria-hidden="true" tabindex="-1"></a>    <span class="at">MoreArgs =</span> <span class="fu">list</span>(<span class="at">df =</span> gene_expression, <span class="at">gene_sets =</span> gene_sets_,</span>
<span id="cb19-196"><a href="#cb19-196" aria-hidden="true" tabindex="-1"></a>                    <span class="at">col_data =</span> <span class="fu">colData</span>(datasets<span class="sc">$</span>tcga) <span class="sc">%&gt;%</span> data.frame)</span>
<span id="cb19-197"><a href="#cb19-197" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-198"><a href="#cb19-198" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-199"><a href="#cb19-199" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb19-200"><a href="#cb19-200" aria-hidden="true" tabindex="-1"></a>    gsva_scores, </span>
<span id="cb19-201"><a href="#cb19-201" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../../results/rds_files/trying/gsva_scores_trying.rds"</span></span>
<span id="cb19-202"><a href="#cb19-202" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-203"><a href="#cb19-203" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-204"><a href="#cb19-204" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-207"><a href="#cb19-207" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-208"><a href="#cb19-208" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(gsva_scores) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"nb_"</span>, nb_of_samples)</span>
<span id="cb19-209"><a href="#cb19-209" aria-hidden="true" tabindex="-1"></a>gsva_scores_ <span class="ot">&lt;-</span> <span class="fu">lapply</span>(</span>
<span id="cb19-210"><a href="#cb19-210" aria-hidden="true" tabindex="-1"></a>    gsva_scores,</span>
<span id="cb19-211"><a href="#cb19-211" aria-hidden="true" tabindex="-1"></a>    \(x) <span class="fu">data.frame</span>(<span class="fu">t</span>(x)) <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"sample_name"</span>)</span>
<span id="cb19-212"><a href="#cb19-212" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb19-213"><a href="#cb19-213" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(<span class="at">.id =</span> <span class="st">"nb_of_samples"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-214"><a href="#cb19-214" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">nb_of_samples =</span> <span class="fu">factor</span>(nb_of_samples, <span class="fu">names</span>(gsva_scores))) <span class="sc">%&gt;%</span></span>
<span id="cb19-215"><a href="#cb19-215" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(</span>
<span id="cb19-216"><a href="#cb19-216" aria-hidden="true" tabindex="-1"></a>        <span class="at">cols =</span> <span class="fu">setdiff</span>(<span class="fu">names</span>(gene_sets_), p4_pathways<span class="sc">$</span>gs_name <span class="sc">%&gt;%</span> unique),</span>
<span id="cb19-217"><a href="#cb19-217" aria-hidden="true" tabindex="-1"></a>        <span class="at">names_to =</span> <span class="st">"pathway"</span>,</span>
<span id="cb19-218"><a href="#cb19-218" aria-hidden="true" tabindex="-1"></a>        <span class="at">values_to =</span> <span class="st">"gsva_sub"</span></span>
<span id="cb19-219"><a href="#cb19-219" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb19-220"><a href="#cb19-220" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">inner_join</span>(</span>
<span id="cb19-221"><a href="#cb19-221" aria-hidden="true" tabindex="-1"></a>        merged_col_data <span class="sc">%&gt;%</span></span>
<span id="cb19-222"><a href="#cb19-222" aria-hidden="true" tabindex="-1"></a>            tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(</span>
<span id="cb19-223"><a href="#cb19-223" aria-hidden="true" tabindex="-1"></a>            <span class="at">cols =</span> <span class="fu">setdiff</span>(<span class="fu">names</span>(gene_sets_), p4_pathways<span class="sc">$</span>gs_name <span class="sc">%&gt;%</span> unique),</span>
<span id="cb19-224"><a href="#cb19-224" aria-hidden="true" tabindex="-1"></a>            <span class="at">names_to =</span> <span class="st">"pathway"</span>,</span>
<span id="cb19-225"><a href="#cb19-225" aria-hidden="true" tabindex="-1"></a>            <span class="at">values_to =</span> <span class="st">"gsva_full"</span></span>
<span id="cb19-226"><a href="#cb19-226" aria-hidden="true" tabindex="-1"></a>        ),</span>
<span id="cb19-227"><a href="#cb19-227" aria-hidden="true" tabindex="-1"></a>        .,</span>
<span id="cb19-228"><a href="#cb19-228" aria-hidden="true" tabindex="-1"></a>        <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"sample_name"</span>, <span class="st">"pathway"</span>)</span>
<span id="cb19-229"><a href="#cb19-229" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb19-230"><a href="#cb19-230" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-231"><a href="#cb19-231" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-232"><a href="#cb19-232" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width = 50, fig.height=15}</span></span>
<span id="cb19-233"><a href="#cb19-233" aria-hidden="true" tabindex="-1"></a>gsva_scores_ <span class="sc">%&gt;%</span> </span>
<span id="cb19-234"><a href="#cb19-234" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(pathway <span class="sc">%in%</span> <span class="fu">names</span>(gene_sets_)[<span class="dv">1</span><span class="sc">:</span><span class="dv">25</span>]) <span class="sc">%&gt;%</span></span>
<span id="cb19-235"><a href="#cb19-235" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(</span>
<span id="cb19-236"><a href="#cb19-236" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> gsva_full, <span class="at">y =</span> gsva_sub, <span class="at">color =</span> pam50</span>
<span id="cb19-237"><a href="#cb19-237" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">+</span></span>
<span id="cb19-238"><a href="#cb19-238" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span> </span>
<span id="cb19-239"><a href="#cb19-239" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_abline</span>(<span class="at">intercept =</span> <span class="dv">0</span>, <span class="at">slope =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb19-240"><a href="#cb19-240" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_grid</span>(nb_of_samples <span class="sc">~</span> pathway, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb19-241"><a href="#cb19-241" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>()</span>
<span id="cb19-242"><a href="#cb19-242" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-243"><a href="#cb19-243" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-244"><a href="#cb19-244" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width = 50, fig.height=15}</span></span>
<span id="cb19-245"><a href="#cb19-245" aria-hidden="true" tabindex="-1"></a>gsva_scores_ <span class="sc">%&gt;%</span> </span>
<span id="cb19-246"><a href="#cb19-246" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(pathway <span class="sc">%in%</span> <span class="fu">names</span>(gene_sets_)[<span class="dv">25</span><span class="sc">:</span>(<span class="fu">length</span>(<span class="fu">names</span>(gene_sets_))<span class="sc">-</span><span class="dv">2</span>)]) <span class="sc">%&gt;%</span></span>
<span id="cb19-247"><a href="#cb19-247" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(</span>
<span id="cb19-248"><a href="#cb19-248" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> gsva_full, <span class="at">y =</span> gsva_sub, <span class="at">color =</span> pam50</span>
<span id="cb19-249"><a href="#cb19-249" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">+</span></span>
<span id="cb19-250"><a href="#cb19-250" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span> </span>
<span id="cb19-251"><a href="#cb19-251" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_abline</span>(<span class="at">intercept =</span> <span class="dv">0</span>, <span class="at">slope =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb19-252"><a href="#cb19-252" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_grid</span>(nb_of_samples <span class="sc">~</span> pathway, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb19-253"><a href="#cb19-253" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>()</span>
<span id="cb19-254"><a href="#cb19-254" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-255"><a href="#cb19-255" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-256"><a href="#cb19-256" aria-hidden="true" tabindex="-1"></a>The conclusion we can take here is that depending on the gene set not even 150</span>
<span id="cb19-257"><a href="#cb19-257" aria-hidden="true" tabindex="-1"></a>samples is enough. For some only 25 patients is already enough, such as SET </span>
<span id="cb19-258"><a href="#cb19-258" aria-hidden="true" tabindex="-1"></a>ER/PR, androgen response, estrogen response early and E2F targets.</span>
<span id="cb19-259"><a href="#cb19-259" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-260"><a href="#cb19-260" aria-hidden="true" tabindex="-1"></a>Could we mix around 20 samples of SCANB together with more samples from </span>
<span id="cb19-261"><a href="#cb19-261" aria-hidden="true" tabindex="-1"></a>TCGA and METABRIC to get the scores? Let us try to adress this question</span>
<span id="cb19-262"><a href="#cb19-262" aria-hidden="true" tabindex="-1"></a>again. We are using the regressed data now here as this is the only way.</span>
<span id="cb19-263"><a href="#cb19-263" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-264"><a href="#cb19-264" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb19-265"><a href="#cb19-265" aria-hidden="true" tabindex="-1"></a>scanb_samples <span class="ot">&lt;-</span> df_pca_coordinates <span class="sc">%&gt;%</span></span>
<span id="cb19-266"><a href="#cb19-266" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"scanb"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-267"><a href="#cb19-267" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">slice_sample</span>(<span class="at">n =</span> <span class="dv">10</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-268"><a href="#cb19-268" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">pull</span>(sample_name)</span>
<span id="cb19-269"><a href="#cb19-269" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-270"><a href="#cb19-270" aria-hidden="true" tabindex="-1"></a>gsva_scores_embeddings_scanb_within_tcga_regressed <span class="ot">&lt;-</span> GSVA<span class="sc">::</span><span class="fu">gsva</span>(</span>
<span id="cb19-271"><a href="#cb19-271" aria-hidden="true" tabindex="-1"></a>    df_pcs_regressed[</span>
<span id="cb19-272"><a href="#cb19-272" aria-hidden="true" tabindex="-1"></a>        , </span>
<span id="cb19-273"><a href="#cb19-273" aria-hidden="true" tabindex="-1"></a>        merged_col_data <span class="sc">%&gt;%</span></span>
<span id="cb19-274"><a href="#cb19-274" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"tcga"</span>, <span class="st">"metabric"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb19-275"><a href="#cb19-275" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">slice_sample</span>(<span class="at">n =</span> <span class="dv">500</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-276"><a href="#cb19-276" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">pull</span>(sample_name) <span class="sc">%&gt;%</span></span>
<span id="cb19-277"><a href="#cb19-277" aria-hidden="true" tabindex="-1"></a>            <span class="fu">c</span>(., scanb_samples)</span>
<span id="cb19-278"><a href="#cb19-278" aria-hidden="true" tabindex="-1"></a>    ],</span>
<span id="cb19-279"><a href="#cb19-279" aria-hidden="true" tabindex="-1"></a>    gene_sets_,</span>
<span id="cb19-280"><a href="#cb19-280" aria-hidden="true" tabindex="-1"></a>    <span class="at">parallel.sz =</span> <span class="dv">10</span>,</span>
<span id="cb19-281"><a href="#cb19-281" aria-hidden="true" tabindex="-1"></a>    <span class="at">verbose =</span> <span class="cn">FALSE</span></span>
<span id="cb19-282"><a href="#cb19-282" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-283"><a href="#cb19-283" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-284"><a href="#cb19-284" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb19-285"><a href="#cb19-285" aria-hidden="true" tabindex="-1"></a>    gsva_scores_embeddings_scanb_within_tcga_regressed,</span>
<span id="cb19-286"><a href="#cb19-286" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../../results/rds_files/trying/gsva_scores_embeddings_scanb_within_tcga_regressed.rds"</span></span>
<span id="cb19-287"><a href="#cb19-287" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-288"><a href="#cb19-288" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-289"><a href="#cb19-289" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-292"><a href="#cb19-292" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-293"><a href="#cb19-293" aria-hidden="true" tabindex="-1"></a>sub_scanb_scores <span class="ot">&lt;-</span> gsva_scores_embeddings_scanb_within_tcga_regressed <span class="sc">%&gt;%</span></span>
<span id="cb19-294"><a href="#cb19-294" aria-hidden="true" tabindex="-1"></a>    t <span class="sc">%&gt;%</span> </span>
<span id="cb19-295"><a href="#cb19-295" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span></span>
<span id="cb19-296"><a href="#cb19-296" aria-hidden="true" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"sample_name"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-297"><a href="#cb19-297" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(</span>
<span id="cb19-298"><a href="#cb19-298" aria-hidden="true" tabindex="-1"></a>        <span class="at">cols =</span> <span class="fu">setdiff</span>(<span class="fu">names</span>(gene_sets_), p4_pathways<span class="sc">$</span>gs_name <span class="sc">%&gt;%</span> unique),</span>
<span id="cb19-299"><a href="#cb19-299" aria-hidden="true" tabindex="-1"></a>        <span class="at">names_to =</span> <span class="st">"pathway"</span>,</span>
<span id="cb19-300"><a href="#cb19-300" aria-hidden="true" tabindex="-1"></a>        <span class="at">values_to =</span> <span class="st">"gsva_sub"</span></span>
<span id="cb19-301"><a href="#cb19-301" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb19-302"><a href="#cb19-302" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">inner_join</span>(</span>
<span id="cb19-303"><a href="#cb19-303" aria-hidden="true" tabindex="-1"></a>        merged_col_data <span class="sc">%&gt;%</span></span>
<span id="cb19-304"><a href="#cb19-304" aria-hidden="true" tabindex="-1"></a>            tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(</span>
<span id="cb19-305"><a href="#cb19-305" aria-hidden="true" tabindex="-1"></a>                <span class="at">cols =</span> <span class="fu">setdiff</span>(<span class="fu">names</span>(gene_sets_), p4_pathways<span class="sc">$</span>gs_name <span class="sc">%&gt;%</span> unique),</span>
<span id="cb19-306"><a href="#cb19-306" aria-hidden="true" tabindex="-1"></a>                <span class="at">names_to =</span> <span class="st">"pathway"</span>,</span>
<span id="cb19-307"><a href="#cb19-307" aria-hidden="true" tabindex="-1"></a>                <span class="at">values_to =</span> <span class="st">"gsva_full"</span></span>
<span id="cb19-308"><a href="#cb19-308" aria-hidden="true" tabindex="-1"></a>            ),</span>
<span id="cb19-309"><a href="#cb19-309" aria-hidden="true" tabindex="-1"></a>        .,</span>
<span id="cb19-310"><a href="#cb19-310" aria-hidden="true" tabindex="-1"></a>        <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"sample_name"</span>, <span class="st">"pathway"</span>)</span>
<span id="cb19-311"><a href="#cb19-311" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb19-312"><a href="#cb19-312" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-313"><a href="#cb19-313" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-314"><a href="#cb19-314" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=20, fig.height=16}</span></span>
<span id="cb19-315"><a href="#cb19-315" aria-hidden="true" tabindex="-1"></a>sub_scanb_scores <span class="sc">%&gt;%</span></span>
<span id="cb19-316"><a href="#cb19-316" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"scanb"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-317"><a href="#cb19-317" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(</span>
<span id="cb19-318"><a href="#cb19-318" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> gsva_sub,</span>
<span id="cb19-319"><a href="#cb19-319" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> gsva_full,</span>
<span id="cb19-320"><a href="#cb19-320" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> cohort</span>
<span id="cb19-321"><a href="#cb19-321" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">+</span></span>
<span id="cb19-322"><a href="#cb19-322" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb19-323"><a href="#cb19-323" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_abline</span>(<span class="at">intercept =</span> <span class="dv">0</span>, <span class="at">slope =</span> <span class="dv">1</span>) <span class="sc">+</span> </span>
<span id="cb19-324"><a href="#cb19-324" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>pathway) <span class="sc">+</span> </span>
<span id="cb19-325"><a href="#cb19-325" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>()</span>
<span id="cb19-326"><a href="#cb19-326" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-327"><a href="#cb19-327" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-328"><a href="#cb19-328" aria-hidden="true" tabindex="-1"></a>There is still some correlation but the scores are </span>
<span id="cb19-329"><a href="#cb19-329" aria-hidden="true" tabindex="-1"></a>shrinked in the end. </span>
<span id="cb19-330"><a href="#cb19-330" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-331"><a href="#cb19-331" aria-hidden="true" tabindex="-1"></a><span class="fu">## Normalization strategy</span></span>
<span id="cb19-332"><a href="#cb19-332" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-333"><a href="#cb19-333" aria-hidden="true" tabindex="-1"></a>We can also try another way of normalizing, namely for each sample we</span>
<span id="cb19-334"><a href="#cb19-334" aria-hidden="true" tabindex="-1"></a>can z scale it.</span>
<span id="cb19-335"><a href="#cb19-335" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-338"><a href="#cb19-338" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-339"><a href="#cb19-339" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1329</span>)</span>
<span id="cb19-340"><a href="#cb19-340" aria-hidden="true" tabindex="-1"></a>samples_for_training <span class="ot">&lt;-</span> merged_col_data <span class="sc">%&gt;%</span></span>
<span id="cb19-341"><a href="#cb19-341" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">%in%</span> which_cohorts_training) <span class="sc">%&gt;%</span></span>
<span id="cb19-342"><a href="#cb19-342" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">pull</span>(sample_name) <span class="sc">%&gt;%</span></span>
<span id="cb19-343"><a href="#cb19-343" aria-hidden="true" tabindex="-1"></a>    <span class="fu">sample</span>(., <span class="at">size =</span> <span class="dv">1000</span>) </span>
<span id="cb19-344"><a href="#cb19-344" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-345"><a href="#cb19-345" aria-hidden="true" tabindex="-1"></a>common_genes <span class="ot">&lt;-</span> <span class="fu">Reduce</span>(intersect, <span class="fu">lapply</span>(datasets, rownames))</span>
<span id="cb19-346"><a href="#cb19-346" aria-hidden="true" tabindex="-1"></a>stable_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(stable_genes, common_genes)</span>
<span id="cb19-347"><a href="#cb19-347" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-348"><a href="#cb19-348" aria-hidden="true" tabindex="-1"></a><span class="co"># and now we can perform the molecular embedding</span></span>
<span id="cb19-349"><a href="#cb19-349" aria-hidden="true" tabindex="-1"></a>training_set <span class="ot">&lt;-</span> <span class="fu">lapply</span>(</span>
<span id="cb19-350"><a href="#cb19-350" aria-hidden="true" tabindex="-1"></a>    datasets_normalized[which_cohorts_training], </span>
<span id="cb19-351"><a href="#cb19-351" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(sum_exp, i, genes_for_pca) </span>
<span id="cb19-352"><a href="#cb19-352" aria-hidden="true" tabindex="-1"></a>        <span class="fu">assay</span>(sum_exp[genes_for_pca, ], <span class="at">i =</span> i) <span class="sc">%&gt;%</span> </span>
<span id="cb19-353"><a href="#cb19-353" aria-hidden="true" tabindex="-1"></a>        <span class="fu">data.frame</span>(<span class="at">check.names =</span> <span class="cn">FALSE</span>), </span>
<span id="cb19-354"><a href="#cb19-354" aria-hidden="true" tabindex="-1"></a>    <span class="at">i =</span> <span class="st">"znorm"</span>,</span>
<span id="cb19-355"><a href="#cb19-355" aria-hidden="true" tabindex="-1"></a>    <span class="at">genes_for_pca =</span> common_genes</span>
<span id="cb19-356"><a href="#cb19-356" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-357"><a href="#cb19-357" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">bind_cols</span>() <span class="sc">%&gt;%</span></span>
<span id="cb19-358"><a href="#cb19-358" aria-hidden="true" tabindex="-1"></a>    .[, samples_for_training]</span>
<span id="cb19-359"><a href="#cb19-359" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-360"><a href="#cb19-360" aria-hidden="true" tabindex="-1"></a>pca_fit_znorm <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">pca</span>(</span>
<span id="cb19-361"><a href="#cb19-361" aria-hidden="true" tabindex="-1"></a>    training_set,</span>
<span id="cb19-362"><a href="#cb19-362" aria-hidden="true" tabindex="-1"></a>    <span class="at">metadata =</span> dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(</span>
<span id="cb19-363"><a href="#cb19-363" aria-hidden="true" tabindex="-1"></a>        <span class="fu">lapply</span>(</span>
<span id="cb19-364"><a href="#cb19-364" aria-hidden="true" tabindex="-1"></a>            datasets_normalized[which_cohorts_training],</span>
<span id="cb19-365"><a href="#cb19-365" aria-hidden="true" tabindex="-1"></a>            <span class="cf">function</span>(df){</span>
<span id="cb19-366"><a href="#cb19-366" aria-hidden="true" tabindex="-1"></a>                <span class="fu">colData</span>(df) <span class="sc">%&gt;%</span> data.frame <span class="sc">%&gt;%</span></span>
<span id="cb19-367"><a href="#cb19-367" aria-hidden="true" tabindex="-1"></a>                    dplyr<span class="sc">::</span><span class="fu">filter</span>(sample_name <span class="sc">%in%</span> <span class="fu">colnames</span>(training_set))</span>
<span id="cb19-368"><a href="#cb19-368" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb19-369"><a href="#cb19-369" aria-hidden="true" tabindex="-1"></a>        ),</span>
<span id="cb19-370"><a href="#cb19-370" aria-hidden="true" tabindex="-1"></a>        <span class="at">.id =</span> <span class="st">"cohort"</span></span>
<span id="cb19-371"><a href="#cb19-371" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span> .[<span class="fu">colnames</span>(training_set), ],</span>
<span id="cb19-372"><a href="#cb19-372" aria-hidden="true" tabindex="-1"></a>    <span class="at">center =</span> <span class="cn">FALSE</span>,</span>
<span id="cb19-373"><a href="#cb19-373" aria-hidden="true" tabindex="-1"></a>    <span class="at">scale =</span> <span class="cn">FALSE</span></span>
<span id="cb19-374"><a href="#cb19-374" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-375"><a href="#cb19-375" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-376"><a href="#cb19-376" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-377"><a href="#cb19-377" aria-hidden="true" tabindex="-1"></a>And now we can visualize the results.</span>
<span id="cb19-378"><a href="#cb19-378" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-379"><a href="#cb19-379" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=14, fig.height=16}</span></span>
<span id="cb19-380"><a href="#cb19-380" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-pca-embeddings-3</span></span>
<span id="cb19-381"><a href="#cb19-381" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: PCA projections colored by different factors and organized</span></span>
<span id="cb19-382"><a href="#cb19-382" aria-hidden="true" tabindex="-1"></a><span class="co">#|    by different components. (A) Plot of the first two components colored</span></span>
<span id="cb19-383"><a href="#cb19-383" aria-hidden="true" tabindex="-1"></a><span class="co">#|    by cohort. (B) Plot of the first two components colored by ER status.</span></span>
<span id="cb19-384"><a href="#cb19-384" aria-hidden="true" tabindex="-1"></a><span class="co">#|    (C) Plot of the second and third components colored by cohort. </span></span>
<span id="cb19-385"><a href="#cb19-385" aria-hidden="true" tabindex="-1"></a><span class="co">#|    (D) Plot of the second and third components colored by ER status.</span></span>
<span id="cb19-386"><a href="#cb19-386" aria-hidden="true" tabindex="-1"></a><span class="co">#|    (E) Plot of the second and third components colored by PAM50.</span></span>
<span id="cb19-387"><a href="#cb19-387" aria-hidden="true" tabindex="-1"></a><span class="co">#|    (F) Plot of the second and third components colored by INTCLUST, only</span></span>
<span id="cb19-388"><a href="#cb19-388" aria-hidden="true" tabindex="-1"></a><span class="co">#|    METABRIC has an assigned value for this variable, NAs are TCGA samples.</span></span>
<span id="cb19-389"><a href="#cb19-389" aria-hidden="true" tabindex="-1"></a>plots_pca_fit <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb19-390"><a href="#cb19-390" aria-hidden="true" tabindex="-1"></a>point_size <span class="ot">&lt;-</span> <span class="dv">2</span></span>
<span id="cb19-391"><a href="#cb19-391" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-392"><a href="#cb19-392" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>pc1_pc2_cohort <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb19-393"><a href="#cb19-393" aria-hidden="true" tabindex="-1"></a>    pca_fit_znorm,</span>
<span id="cb19-394"><a href="#cb19-394" aria-hidden="true" tabindex="-1"></a>    <span class="at">colby =</span> <span class="st">"cohort"</span>,</span>
<span id="cb19-395"><a href="#cb19-395" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> <span class="cn">NULL</span>, </span>
<span id="cb19-396"><a href="#cb19-396" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb19-397"><a href="#cb19-397" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC1"</span>,</span>
<span id="cb19-398"><a href="#cb19-398" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC2"</span>,</span>
<span id="cb19-399"><a href="#cb19-399" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"First two components colored by cohort"</span>,</span>
<span id="cb19-400"><a href="#cb19-400" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">"Only 1000 training samples"</span>,</span>
<span id="cb19-401"><a href="#cb19-401" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> point_size</span>
<span id="cb19-402"><a href="#cb19-402" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-403"><a href="#cb19-403" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-404"><a href="#cb19-404" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>pc1_pc2_er_status <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb19-405"><a href="#cb19-405" aria-hidden="true" tabindex="-1"></a>    pca_fit_znorm,</span>
<span id="cb19-406"><a href="#cb19-406" aria-hidden="true" tabindex="-1"></a>    <span class="at">colby =</span> <span class="st">"er_status"</span>,</span>
<span id="cb19-407"><a href="#cb19-407" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> <span class="cn">NULL</span>, </span>
<span id="cb19-408"><a href="#cb19-408" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb19-409"><a href="#cb19-409" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC1"</span>,</span>
<span id="cb19-410"><a href="#cb19-410" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC2"</span>,</span>
<span id="cb19-411"><a href="#cb19-411" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"First two components colored by ER status"</span>,</span>
<span id="cb19-412"><a href="#cb19-412" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">"Only 1000 training samples"</span>,</span>
<span id="cb19-413"><a href="#cb19-413" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> point_size</span>
<span id="cb19-414"><a href="#cb19-414" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-415"><a href="#cb19-415" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-416"><a href="#cb19-416" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>pc2_pc3_cohort <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb19-417"><a href="#cb19-417" aria-hidden="true" tabindex="-1"></a>    pca_fit_znorm,</span>
<span id="cb19-418"><a href="#cb19-418" aria-hidden="true" tabindex="-1"></a>    <span class="at">colby =</span> <span class="st">"cohort"</span>,</span>
<span id="cb19-419"><a href="#cb19-419" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> <span class="cn">NULL</span>, </span>
<span id="cb19-420"><a href="#cb19-420" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb19-421"><a href="#cb19-421" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC2"</span>,</span>
<span id="cb19-422"><a href="#cb19-422" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC3"</span>,</span>
<span id="cb19-423"><a href="#cb19-423" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Second and third components colored by cohort"</span>,</span>
<span id="cb19-424"><a href="#cb19-424" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">"Only 1000 training samples"</span>,</span>
<span id="cb19-425"><a href="#cb19-425" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> point_size</span>
<span id="cb19-426"><a href="#cb19-426" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-427"><a href="#cb19-427" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-428"><a href="#cb19-428" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>pc2_pc3_er_status <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb19-429"><a href="#cb19-429" aria-hidden="true" tabindex="-1"></a>    pca_fit_znorm,</span>
<span id="cb19-430"><a href="#cb19-430" aria-hidden="true" tabindex="-1"></a>    <span class="at">colby =</span> <span class="st">"er_status"</span>,</span>
<span id="cb19-431"><a href="#cb19-431" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> <span class="cn">NULL</span>, </span>
<span id="cb19-432"><a href="#cb19-432" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb19-433"><a href="#cb19-433" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC2"</span>,</span>
<span id="cb19-434"><a href="#cb19-434" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC3"</span>,</span>
<span id="cb19-435"><a href="#cb19-435" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Second and third components colored by ER status"</span>,</span>
<span id="cb19-436"><a href="#cb19-436" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">"Only 1000 training samples"</span>,</span>
<span id="cb19-437"><a href="#cb19-437" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> point_size</span>
<span id="cb19-438"><a href="#cb19-438" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-439"><a href="#cb19-439" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-440"><a href="#cb19-440" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>pc2_pc3_pam50 <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb19-441"><a href="#cb19-441" aria-hidden="true" tabindex="-1"></a>    pca_fit_znorm,</span>
<span id="cb19-442"><a href="#cb19-442" aria-hidden="true" tabindex="-1"></a>    <span class="at">colby =</span> <span class="st">"pam50"</span>,</span>
<span id="cb19-443"><a href="#cb19-443" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> <span class="cn">NULL</span>, </span>
<span id="cb19-444"><a href="#cb19-444" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb19-445"><a href="#cb19-445" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC2"</span>,</span>
<span id="cb19-446"><a href="#cb19-446" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC3"</span>,</span>
<span id="cb19-447"><a href="#cb19-447" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Second and third components colored by PAM50"</span>,</span>
<span id="cb19-448"><a href="#cb19-448" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">"Only 1000 training samples"</span>,</span>
<span id="cb19-449"><a href="#cb19-449" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> point_size</span>
<span id="cb19-450"><a href="#cb19-450" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-451"><a href="#cb19-451" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>pc2_pc3_pam50 <span class="ot">&lt;-</span> plots_pca_fit<span class="sc">$</span>pc2_pc3_pam50 <span class="sc">+</span></span>
<span id="cb19-452"><a href="#cb19-452" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">get_colors_pam50</span>(pca_fit<span class="sc">$</span>metadata))</span>
<span id="cb19-453"><a href="#cb19-453" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb19-454"><a href="#cb19-454" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-455"><a href="#cb19-455" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>pc2_pc3_intclust <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb19-456"><a href="#cb19-456" aria-hidden="true" tabindex="-1"></a>    pca_fit_znorm,</span>
<span id="cb19-457"><a href="#cb19-457" aria-hidden="true" tabindex="-1"></a>    <span class="at">colby =</span> <span class="st">"INTCLUST"</span>,</span>
<span id="cb19-458"><a href="#cb19-458" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> <span class="cn">NULL</span>, </span>
<span id="cb19-459"><a href="#cb19-459" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb19-460"><a href="#cb19-460" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC2"</span>,</span>
<span id="cb19-461"><a href="#cb19-461" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC3"</span>,</span>
<span id="cb19-462"><a href="#cb19-462" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Second and third components colored by INTCLUST"</span>,</span>
<span id="cb19-463"><a href="#cb19-463" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">"Only 1000 training samples"</span>,</span>
<span id="cb19-464"><a href="#cb19-464" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> point_size</span>
<span id="cb19-465"><a href="#cb19-465" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-466"><a href="#cb19-466" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-467"><a href="#cb19-467" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>pc3_pc4_pam50 <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb19-468"><a href="#cb19-468" aria-hidden="true" tabindex="-1"></a>    pca_fit_znorm,</span>
<span id="cb19-469"><a href="#cb19-469" aria-hidden="true" tabindex="-1"></a>    <span class="at">colby =</span> <span class="st">"pam50"</span>,</span>
<span id="cb19-470"><a href="#cb19-470" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> <span class="cn">NULL</span>, </span>
<span id="cb19-471"><a href="#cb19-471" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb19-472"><a href="#cb19-472" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC3"</span>,</span>
<span id="cb19-473"><a href="#cb19-473" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC4"</span>,</span>
<span id="cb19-474"><a href="#cb19-474" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Third and fourth components colored by PAM50"</span>,</span>
<span id="cb19-475"><a href="#cb19-475" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">"Only 1000 training samples"</span>,</span>
<span id="cb19-476"><a href="#cb19-476" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> point_size</span>
<span id="cb19-477"><a href="#cb19-477" aria-hidden="true" tabindex="-1"></a>) </span>
<span id="cb19-478"><a href="#cb19-478" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>pc3_pc4_pam50 <span class="ot">&lt;-</span> plots_pca_fit<span class="sc">$</span>pc3_pc4_pam50 <span class="sc">+</span></span>
<span id="cb19-479"><a href="#cb19-479" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">get_colors_pam50</span>(pca_fit<span class="sc">$</span>metadata))</span>
<span id="cb19-480"><a href="#cb19-480" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-481"><a href="#cb19-481" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>pc3_pc4_cohort <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb19-482"><a href="#cb19-482" aria-hidden="true" tabindex="-1"></a>    pca_fit_znorm,</span>
<span id="cb19-483"><a href="#cb19-483" aria-hidden="true" tabindex="-1"></a>    <span class="at">colby =</span> <span class="st">"cohort"</span>,</span>
<span id="cb19-484"><a href="#cb19-484" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> <span class="cn">NULL</span>, </span>
<span id="cb19-485"><a href="#cb19-485" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb19-486"><a href="#cb19-486" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC3"</span>,</span>
<span id="cb19-487"><a href="#cb19-487" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC4"</span>,</span>
<span id="cb19-488"><a href="#cb19-488" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Third and fourth components colored by cohort"</span>,</span>
<span id="cb19-489"><a href="#cb19-489" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">"Only 1000 training samples"</span>,</span>
<span id="cb19-490"><a href="#cb19-490" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> point_size</span>
<span id="cb19-491"><a href="#cb19-491" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-492"><a href="#cb19-492" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-493"><a href="#cb19-493" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(<span class="at">plotlist =</span> plots_pca_fit, <span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">labels =</span> <span class="st">"AUTO"</span>)</span>
<span id="cb19-494"><a href="#cb19-494" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-495"><a href="#cb19-495" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-498"><a href="#cb19-498" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-499"><a href="#cb19-499" aria-hidden="true" tabindex="-1"></a>datasets_pca_coordinates_znorm <span class="ot">&lt;-</span> <span class="fu">lapply</span>(</span>
<span id="cb19-500"><a href="#cb19-500" aria-hidden="true" tabindex="-1"></a>    datasets_normalized,</span>
<span id="cb19-501"><a href="#cb19-501" aria-hidden="true" tabindex="-1"></a>    get_pca_coordinates,</span>
<span id="cb19-502"><a href="#cb19-502" aria-hidden="true" tabindex="-1"></a>    <span class="at">pca_fit =</span> pca_fit_znorm,</span>
<span id="cb19-503"><a href="#cb19-503" aria-hidden="true" tabindex="-1"></a>    <span class="at">genes_for_pca =</span> <span class="fu">rownames</span>(pca_fit_znorm<span class="sc">$</span>loadings)</span>
<span id="cb19-504"><a href="#cb19-504" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-505"><a href="#cb19-505" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-506"><a href="#cb19-506" aria-hidden="true" tabindex="-1"></a>df_pca_coordinates_znorm <span class="ot">&lt;-</span> datasets_pca_coordinates_znorm <span class="sc">%&gt;%</span> </span>
<span id="cb19-507"><a href="#cb19-507" aria-hidden="true" tabindex="-1"></a>    <span class="fu">do.call</span>(rbind, .) <span class="sc">%&gt;%</span></span>
<span id="cb19-508"><a href="#cb19-508" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span> </span>
<span id="cb19-509"><a href="#cb19-509" aria-hidden="true" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"sample_name"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-510"><a href="#cb19-510" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">inner_join</span>(</span>
<span id="cb19-511"><a href="#cb19-511" aria-hidden="true" tabindex="-1"></a>        .,</span>
<span id="cb19-512"><a href="#cb19-512" aria-hidden="true" tabindex="-1"></a>        merged_col_data,</span>
<span id="cb19-513"><a href="#cb19-513" aria-hidden="true" tabindex="-1"></a>        <span class="at">by =</span> <span class="st">"sample_name"</span></span>
<span id="cb19-514"><a href="#cb19-514" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb19-515"><a href="#cb19-515" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-516"><a href="#cb19-516" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-519"><a href="#cb19-519" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-520"><a href="#cb19-520" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_pca_coordinates</span>(</span>
<span id="cb19-521"><a href="#cb19-521" aria-hidden="true" tabindex="-1"></a>    df_pca_coordinates_znorm <span class="sc">%&gt;%</span> </span>
<span id="cb19-522"><a href="#cb19-522" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">slice</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">n</span>()),</span>
<span id="cb19-523"><a href="#cb19-523" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"cohort"</span>,</span>
<span id="cb19-524"><a href="#cb19-524" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC3"</span>,</span>
<span id="cb19-525"><a href="#cb19-525" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC4"</span>,</span>
<span id="cb19-526"><a href="#cb19-526" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">1</span>,</span>
<span id="cb19-527"><a href="#cb19-527" aria-hidden="true" tabindex="-1"></a>    <span class="at">base_size =</span> <span class="dv">10</span>,</span>
<span id="cb19-528"><a href="#cb19-528" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="fu">paste0</span>(</span>
<span id="cb19-529"><a href="#cb19-529" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Embedding of all samples from TCGA,</span><span class="sc">\n</span><span class="st">"</span>,</span>
<span id="cb19-530"><a href="#cb19-530" aria-hidden="true" tabindex="-1"></a>        <span class="st">"METABRIC and SCANB"</span></span>
<span id="cb19-531"><a href="#cb19-531" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb19-532"><a href="#cb19-532" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span> ggplot2<span class="sc">::</span><span class="fu">scale_color_viridis_d</span>() <span class="sc">+</span></span>
<span id="cb19-533"><a href="#cb19-533" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">15</span>) <span class="sc">+</span></span>
<span id="cb19-534"><a href="#cb19-534" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span></span>
<span id="cb19-535"><a href="#cb19-535" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb19-536"><a href="#cb19-536" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-537"><a href="#cb19-537" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-538"><a href="#cb19-538" aria-hidden="true" tabindex="-1"></a>We see that samples are well embedded as well. </span>
<span id="cb19-539"><a href="#cb19-539" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-542"><a href="#cb19-542" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-543"><a href="#cb19-543" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-pca-scanb-pc1-2</span></span>
<span id="cb19-544"><a href="#cb19-544" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: PCA embedding colored by cohort. The components used are the</span></span>
<span id="cb19-545"><a href="#cb19-545" aria-hidden="true" tabindex="-1"></a><span class="co">#|     first and second components.</span></span>
<span id="cb19-546"><a href="#cb19-546" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_pca_coordinates</span>(</span>
<span id="cb19-547"><a href="#cb19-547" aria-hidden="true" tabindex="-1"></a>    df_pca_coordinates_znorm,</span>
<span id="cb19-548"><a href="#cb19-548" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"cohort"</span>,</span>
<span id="cb19-549"><a href="#cb19-549" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC1"</span>,</span>
<span id="cb19-550"><a href="#cb19-550" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC2"</span>,</span>
<span id="cb19-551"><a href="#cb19-551" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">1</span>,</span>
<span id="cb19-552"><a href="#cb19-552" aria-hidden="true" tabindex="-1"></a>    <span class="at">base_size =</span> <span class="dv">10</span>,</span>
<span id="cb19-553"><a href="#cb19-553" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="fu">paste0</span>(</span>
<span id="cb19-554"><a href="#cb19-554" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Embedding of all samples from TCGA,</span><span class="sc">\n</span><span class="st">"</span>,</span>
<span id="cb19-555"><a href="#cb19-555" aria-hidden="true" tabindex="-1"></a>        <span class="st">"METABRIC and SCANB"</span></span>
<span id="cb19-556"><a href="#cb19-556" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb19-557"><a href="#cb19-557" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span> </span>
<span id="cb19-558"><a href="#cb19-558" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">15</span>) <span class="sc">+</span></span>
<span id="cb19-559"><a href="#cb19-559" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span></span>
<span id="cb19-560"><a href="#cb19-560" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb19-561"><a href="#cb19-561" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-562"><a href="#cb19-562" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-563"><a href="#cb19-563" aria-hidden="true" tabindex="-1"></a>SCANB and POETIC are in between as expected and they are closer</span>
<span id="cb19-564"><a href="#cb19-564" aria-hidden="true" tabindex="-1"></a>to their sequencing technology. Interestingly the </span>
<span id="cb19-565"><a href="#cb19-565" aria-hidden="true" tabindex="-1"></a>variation for SCANB and TCGA are compressed in the PC1 vs PC2 axis compared</span>
<span id="cb19-566"><a href="#cb19-566" aria-hidden="true" tabindex="-1"></a>to METABRIC and POETIC.</span>
<span id="cb19-567"><a href="#cb19-567" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-568"><a href="#cb19-568" aria-hidden="true" tabindex="-1"></a>@fig-pca-scanb-er-pam50-2 shows that the SCANB samples are also well mixed</span>
<span id="cb19-569"><a href="#cb19-569" aria-hidden="true" tabindex="-1"></a>regarding the clinical factors, including the $SET_{ER/PR}$ signature.</span>
<span id="cb19-570"><a href="#cb19-570" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-571"><a href="#cb19-571" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=16, fig.height=14}</span></span>
<span id="cb19-572"><a href="#cb19-572" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-pca-scanb-er-pam50-2</span></span>
<span id="cb19-573"><a href="#cb19-573" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: PCA embedding of all samples from TCGA, SCANB and METABRIC.</span></span>
<span id="cb19-574"><a href="#cb19-574" aria-hidden="true" tabindex="-1"></a><span class="co">#|     (A) Colored by cohort,</span></span>
<span id="cb19-575"><a href="#cb19-575" aria-hidden="true" tabindex="-1"></a><span class="co">#|     (B) colored by ER status, (C) colored by PAM50 molecular subtype,</span></span>
<span id="cb19-576"><a href="#cb19-576" aria-hidden="true" tabindex="-1"></a><span class="co">#|     (D) colored by the $SET_{ER/PR}$ signature and only SCANB samples.</span></span>
<span id="cb19-577"><a href="#cb19-577" aria-hidden="true" tabindex="-1"></a>size <span class="ot">&lt;-</span> <span class="dv">2</span></span>
<span id="cb19-578"><a href="#cb19-578" aria-hidden="true" tabindex="-1"></a>base_size <span class="ot">&lt;-</span> <span class="dv">20</span></span>
<span id="cb19-579"><a href="#cb19-579" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-580"><a href="#cb19-580" aria-hidden="true" tabindex="-1"></a>plots_with_scanb <span class="ot">&lt;-</span> <span class="fu">sapply</span>(</span>
<span id="cb19-581"><a href="#cb19-581" aria-hidden="true" tabindex="-1"></a>    <span class="fu">c</span>(<span class="st">"cohort"</span>, <span class="st">"er_status"</span>, <span class="st">"pam50"</span>, <span class="st">"SET_ERPR"</span>, <span class="st">"HALLMARK_G2M_CHECKPOINT"</span>), </span>
<span id="cb19-582"><a href="#cb19-582" aria-hidden="true" tabindex="-1"></a>    plot_pca_coordinates,</span>
<span id="cb19-583"><a href="#cb19-583" aria-hidden="true" tabindex="-1"></a>    <span class="at">df_pca =</span> df_pca_coordinates_znorm <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb19-584"><a href="#cb19-584" aria-hidden="true" tabindex="-1"></a>        pam50 <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"luma"</span>, <span class="st">"lumb"</span>, <span class="st">"basal"</span>, <span class="st">"her2"</span>, <span class="st">"normal"</span>) <span class="sc">&amp;</span></span>
<span id="cb19-585"><a href="#cb19-585" aria-hidden="true" tabindex="-1"></a>            cohort <span class="sc">!=</span> <span class="st">"poetic"</span></span>
<span id="cb19-586"><a href="#cb19-586" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb19-587"><a href="#cb19-587" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">slice</span>(<span class="fu">sample</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">n</span>())),</span>
<span id="cb19-588"><a href="#cb19-588" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC3"</span>, </span>
<span id="cb19-589"><a href="#cb19-589" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC4"</span>,</span>
<span id="cb19-590"><a href="#cb19-590" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> size,</span>
<span id="cb19-591"><a href="#cb19-591" aria-hidden="true" tabindex="-1"></a>    <span class="at">base_size =</span> base_size,</span>
<span id="cb19-592"><a href="#cb19-592" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="fu">paste0</span>(</span>
<span id="cb19-593"><a href="#cb19-593" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Embedding of all samples from TCGA,</span><span class="sc">\n</span><span class="st">"</span>,</span>
<span id="cb19-594"><a href="#cb19-594" aria-hidden="true" tabindex="-1"></a>        <span class="st">"METABRIC and SCANB"</span></span>
<span id="cb19-595"><a href="#cb19-595" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb19-596"><a href="#cb19-596" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>, </span>
<span id="cb19-597"><a href="#cb19-597" aria-hidden="true" tabindex="-1"></a>    <span class="at">simplify =</span> <span class="cn">FALSE</span></span>
<span id="cb19-598"><a href="#cb19-598" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-599"><a href="#cb19-599" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-600"><a href="#cb19-600" aria-hidden="true" tabindex="-1"></a>plots_with_scanb<span class="sc">$</span>cohort <span class="ot">&lt;-</span> plots_with_scanb<span class="sc">$</span>cohort <span class="sc">+</span></span>
<span id="cb19-601"><a href="#cb19-601" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span></span>
<span id="cb19-602"><a href="#cb19-602" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb19-603"><a href="#cb19-603" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-604"><a href="#cb19-604" aria-hidden="true" tabindex="-1"></a>plots_with_scanb<span class="sc">$</span>pam50 <span class="ot">&lt;-</span> plots_with_scanb<span class="sc">$</span>pam50 <span class="sc">+</span></span>
<span id="cb19-605"><a href="#cb19-605" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_color_manual</span>(</span>
<span id="cb19-606"><a href="#cb19-606" aria-hidden="true" tabindex="-1"></a>        <span class="at">values =</span> <span class="fu">get_colors_pam50</span>(plots_with_scanb<span class="sc">$</span>pam50<span class="sc">$</span>data)</span>
<span id="cb19-607"><a href="#cb19-607" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb19-608"><a href="#cb19-608" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span></span>
<span id="cb19-609"><a href="#cb19-609" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb19-610"><a href="#cb19-610" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-611"><a href="#cb19-611" aria-hidden="true" tabindex="-1"></a>plots_with_scanb<span class="sc">$</span>er_status <span class="ot">&lt;-</span>  plots_with_scanb<span class="sc">$</span>er_status <span class="sc">+</span></span>
<span id="cb19-612"><a href="#cb19-612" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_color_viridis_d</span>() <span class="sc">+</span></span>
<span id="cb19-613"><a href="#cb19-613" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">color =</span> <span class="st">"ER status"</span>) <span class="sc">+</span></span>
<span id="cb19-614"><a href="#cb19-614" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span></span>
<span id="cb19-615"><a href="#cb19-615" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb19-616"><a href="#cb19-616" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-617"><a href="#cb19-617" aria-hidden="true" tabindex="-1"></a>plots_with_scanb<span class="sc">$</span>SET_ERPR <span class="ot">&lt;-</span> df_pca_coordinates_znorm <span class="sc">%&gt;%</span> </span>
<span id="cb19-618"><a href="#cb19-618" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"scanb"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb19-619"><a href="#cb19-619" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes_string</span>(<span class="at">x =</span> <span class="st">"PC3"</span>, <span class="at">y =</span> <span class="st">"PC4"</span>, <span class="at">z =</span> <span class="st">"SET_ERPR"</span>)) <span class="sc">+</span> </span>
<span id="cb19-620"><a href="#cb19-620" aria-hidden="true" tabindex="-1"></a>        ggplot2<span class="sc">::</span><span class="fu">stat_summary_hex</span>(<span class="at">bins =</span> <span class="dv">25</span>) <span class="sc">+</span></span>
<span id="cb19-621"><a href="#cb19-621" aria-hidden="true" tabindex="-1"></a>        ggplot2<span class="sc">::</span><span class="fu">scale_fill_viridis_c</span>() <span class="sc">+</span></span>
<span id="cb19-622"><a href="#cb19-622" aria-hidden="true" tabindex="-1"></a>        ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb19-623"><a href="#cb19-623" aria-hidden="true" tabindex="-1"></a>            <span class="at">title =</span> <span class="st">"Embedding of SCANB only"</span>,</span>
<span id="cb19-624"><a href="#cb19-624" aria-hidden="true" tabindex="-1"></a>            <span class="at">fill =</span> <span class="fu">expression</span>(SET[ER<span class="sc">/</span>PR])</span>
<span id="cb19-625"><a href="#cb19-625" aria-hidden="true" tabindex="-1"></a>        ) <span class="sc">+</span> </span>
<span id="cb19-626"><a href="#cb19-626" aria-hidden="true" tabindex="-1"></a>        ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> base_size) <span class="sc">+</span></span>
<span id="cb19-627"><a href="#cb19-627" aria-hidden="true" tabindex="-1"></a>        <span class="fu">change_plot_aes_point</span>()</span>
<span id="cb19-628"><a href="#cb19-628" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-629"><a href="#cb19-629" aria-hidden="true" tabindex="-1"></a>plots_with_scanb<span class="sc">$</span>HALLMARK_G2M_CHECKPOINT <span class="ot">&lt;-</span> df_pca_coordinates_znorm <span class="sc">%&gt;%</span> </span>
<span id="cb19-630"><a href="#cb19-630" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"scanb"</span>) <span class="sc">&amp;</span> pam50 <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"luma"</span>, <span class="st">"lumb"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb19-631"><a href="#cb19-631" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes_string</span>(<span class="at">x =</span> <span class="st">"PC3"</span>, <span class="at">y =</span> <span class="st">"PC4"</span>, <span class="at">z =</span> <span class="st">"HALLMARK_G2M_CHECKPOINT"</span>)) <span class="sc">+</span> </span>
<span id="cb19-632"><a href="#cb19-632" aria-hidden="true" tabindex="-1"></a>        ggplot2<span class="sc">::</span><span class="fu">stat_summary_hex</span>(<span class="at">bins =</span> <span class="dv">25</span>) <span class="sc">+</span></span>
<span id="cb19-633"><a href="#cb19-633" aria-hidden="true" tabindex="-1"></a>        ggplot2<span class="sc">::</span><span class="fu">scale_fill_viridis_c</span>() <span class="sc">+</span></span>
<span id="cb19-634"><a href="#cb19-634" aria-hidden="true" tabindex="-1"></a>        ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb19-635"><a href="#cb19-635" aria-hidden="true" tabindex="-1"></a>            <span class="at">title =</span> <span class="st">"Embedding of SCANB, luminal A and B only"</span>,</span>
<span id="cb19-636"><a href="#cb19-636" aria-hidden="true" tabindex="-1"></a>            <span class="at">fill =</span> <span class="st">"G2M"</span></span>
<span id="cb19-637"><a href="#cb19-637" aria-hidden="true" tabindex="-1"></a>        ) <span class="sc">+</span> </span>
<span id="cb19-638"><a href="#cb19-638" aria-hidden="true" tabindex="-1"></a>        ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> base_size) <span class="sc">+</span></span>
<span id="cb19-639"><a href="#cb19-639" aria-hidden="true" tabindex="-1"></a>        <span class="fu">change_plot_aes_point</span>()</span>
<span id="cb19-640"><a href="#cb19-640" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-641"><a href="#cb19-641" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(</span>
<span id="cb19-642"><a href="#cb19-642" aria-hidden="true" tabindex="-1"></a>    <span class="at">plotlist =</span> plots_with_scanb,</span>
<span id="cb19-643"><a href="#cb19-643" aria-hidden="true" tabindex="-1"></a>    <span class="at">ncol =</span> <span class="dv">2</span></span>
<span id="cb19-644"><a href="#cb19-644" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-645"><a href="#cb19-645" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-646"><a href="#cb19-646" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-647"><a href="#cb19-647" aria-hidden="true" tabindex="-1"></a>In the end normalizing by scaling the sample before doing any embedding</span>
<span id="cb19-648"><a href="#cb19-648" aria-hidden="true" tabindex="-1"></a>works as well. Though the embedding seems to be more compressed. </span>
<span id="cb19-649"><a href="#cb19-649" aria-hidden="true" tabindex="-1"></a>I would argue that the qPCR-like normalization gives a bit better</span>
<span id="cb19-650"><a href="#cb19-650" aria-hidden="true" tabindex="-1"></a>embedding. </span>
<span id="cb19-651"><a href="#cb19-651" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-652"><a href="#cb19-652" aria-hidden="true" tabindex="-1"></a><span class="fu">## genefu </span></span>
<span id="cb19-653"><a href="#cb19-653" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-654"><a href="#cb19-654" aria-hidden="true" tabindex="-1"></a>With genefu it is possible to calculate scores from comercially available </span>
<span id="cb19-655"><a href="#cb19-655" aria-hidden="true" tabindex="-1"></a>signatures. Here we try to use it on SCANB, METABRIC and TCGA. Moreover,</span>
<span id="cb19-656"><a href="#cb19-656" aria-hidden="true" tabindex="-1"></a>we can start with ABiM, as this dataset has the gold standard ROR score </span>
<span id="cb19-657"><a href="#cb19-657" aria-hidden="true" tabindex="-1"></a>obtained from nCounter data. This is a good way of validating the</span>
<span id="cb19-658"><a href="#cb19-658" aria-hidden="true" tabindex="-1"></a>pipeline. </span>
<span id="cb19-659"><a href="#cb19-659" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-662"><a href="#cb19-662" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-663"><a href="#cb19-663" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(pam50.robust)</span>
<span id="cb19-664"><a href="#cb19-664" aria-hidden="true" tabindex="-1"></a>abim_genefu_pam50 <span class="ot">&lt;-</span> genefu<span class="sc">::</span><span class="fu">molecular.subtyping</span>(</span>
<span id="cb19-665"><a href="#cb19-665" aria-hidden="true" tabindex="-1"></a>    <span class="at">sbt.model =</span> <span class="st">"pam50"</span>,</span>
<span id="cb19-666"><a href="#cb19-666" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> <span class="fu">t</span>(<span class="fu">assay</span>(abim_100, <span class="st">"log2fpkm"</span>)), </span>
<span id="cb19-667"><a href="#cb19-667" aria-hidden="true" tabindex="-1"></a>    <span class="at">annot =</span> annotables<span class="sc">::</span>grch38 <span class="sc">%&gt;%</span></span>
<span id="cb19-668"><a href="#cb19-668" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb19-669"><a href="#cb19-669" aria-hidden="true" tabindex="-1"></a>            <span class="at">EntrezGene.ID =</span> entrez,</span>
<span id="cb19-670"><a href="#cb19-670" aria-hidden="true" tabindex="-1"></a>            <span class="at">Gene.Symbol =</span> symbol</span>
<span id="cb19-671"><a href="#cb19-671" aria-hidden="true" tabindex="-1"></a>        ),</span>
<span id="cb19-672"><a href="#cb19-672" aria-hidden="true" tabindex="-1"></a>    <span class="at">do.mapping =</span> <span class="cn">FALSE</span></span>
<span id="cb19-673"><a href="#cb19-673" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-674"><a href="#cb19-674" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-675"><a href="#cb19-675" aria-hidden="true" tabindex="-1"></a>abim_100<span class="sc">$</span>GENEFU.PAM50 <span class="ot">&lt;-</span> abim_genefu_pam50<span class="sc">$</span>subtype</span>
<span id="cb19-676"><a href="#cb19-676" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-677"><a href="#cb19-677" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-678"><a href="#cb19-678" aria-hidden="true" tabindex="-1"></a>We start by comparing the molecular subtypes:</span>
<span id="cb19-679"><a href="#cb19-679" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-682"><a href="#cb19-682" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-683"><a href="#cb19-683" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(abim_100<span class="sc">$</span>GENEFU.PAM50 <span class="sc">==</span> abim_100<span class="sc">$</span>ProsignaFT.Subtype)</span>
<span id="cb19-684"><a href="#cb19-684" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-685"><a href="#cb19-685" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-686"><a href="#cb19-686" aria-hidden="true" tabindex="-1"></a>86 out of 100 were correctly called. Here we are using the gold standard wich</span>
<span id="cb19-687"><a href="#cb19-687" aria-hidden="true" tabindex="-1"></a>is the molecular subtype called from FT samples. The normal like subtype is not</span>
<span id="cb19-688"><a href="#cb19-688" aria-hidden="true" tabindex="-1"></a>available from the FT prosigna assay. So we now compare the results with the</span>
<span id="cb19-689"><a href="#cb19-689" aria-hidden="true" tabindex="-1"></a>SSP obtained from the SCANB team. </span>
<span id="cb19-690"><a href="#cb19-690" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-691"><a href="#cb19-691" aria-hidden="true" tabindex="-1"></a>Below is the confusion matrix to see which subtypes were misclassified. </span>
<span id="cb19-692"><a href="#cb19-692" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-695"><a href="#cb19-695" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-696"><a href="#cb19-696" aria-hidden="true" tabindex="-1"></a>caret<span class="sc">::</span><span class="fu">confusionMatrix</span>(</span>
<span id="cb19-697"><a href="#cb19-697" aria-hidden="true" tabindex="-1"></a>    abim_100<span class="sc">$</span>GENEFU.PAM50,</span>
<span id="cb19-698"><a href="#cb19-698" aria-hidden="true" tabindex="-1"></a>    <span class="fu">factor</span>(</span>
<span id="cb19-699"><a href="#cb19-699" aria-hidden="true" tabindex="-1"></a>        abim_100<span class="sc">$</span>ProsignaFT.Subtype, </span>
<span id="cb19-700"><a href="#cb19-700" aria-hidden="true" tabindex="-1"></a>        <span class="at">levels =</span> <span class="fu">levels</span>(abim_100<span class="sc">$</span>GENEFU.PAM50)</span>
<span id="cb19-701"><a href="#cb19-701" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb19-702"><a href="#cb19-702" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-703"><a href="#cb19-703" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-704"><a href="#cb19-704" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-705"><a href="#cb19-705" aria-hidden="true" tabindex="-1"></a>All luminal A were predicted correctly. Some normal like were predicted to</span>
<span id="cb19-706"><a href="#cb19-706" aria-hidden="true" tabindex="-1"></a>be luminal A or basal and Her2. There were some mismatches </span>
<span id="cb19-707"><a href="#cb19-707" aria-hidden="true" tabindex="-1"></a>for the LumB, that were misclassified as LumA. Still it is ok. </span>
<span id="cb19-708"><a href="#cb19-708" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-709"><a href="#cb19-709" aria-hidden="true" tabindex="-1"></a>Let us now calculate the ROR for the ABiM cohort.</span>
<span id="cb19-710"><a href="#cb19-710" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-713"><a href="#cb19-713" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-714"><a href="#cb19-714" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-ror-genefu-abim</span></span>
<span id="cb19-715"><a href="#cb19-715" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Correlation between ROR obtained from prosigna assay and </span></span>
<span id="cb19-716"><a href="#cb19-716" aria-hidden="true" tabindex="-1"></a><span class="co">#|     R package genefu</span></span>
<span id="cb19-717"><a href="#cb19-717" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(pam50)</span>
<span id="cb19-718"><a href="#cb19-718" aria-hidden="true" tabindex="-1"></a>ror_abim <span class="ot">&lt;-</span> genefu<span class="sc">::</span><span class="fu">rorS</span>(</span>
<span id="cb19-719"><a href="#cb19-719" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> <span class="fu">t</span>(<span class="fu">assay</span>(abim_100, <span class="st">"log2fpkm"</span>)), </span>
<span id="cb19-720"><a href="#cb19-720" aria-hidden="true" tabindex="-1"></a>    <span class="at">annot =</span> annotables<span class="sc">::</span>grch38 <span class="sc">%&gt;%</span></span>
<span id="cb19-721"><a href="#cb19-721" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb19-722"><a href="#cb19-722" aria-hidden="true" tabindex="-1"></a>            <span class="at">EntrezGene.ID =</span> entrez,</span>
<span id="cb19-723"><a href="#cb19-723" aria-hidden="true" tabindex="-1"></a>            <span class="at">Gene.Symbol =</span> symbol</span>
<span id="cb19-724"><a href="#cb19-724" aria-hidden="true" tabindex="-1"></a>        ),</span>
<span id="cb19-725"><a href="#cb19-725" aria-hidden="true" tabindex="-1"></a>    <span class="at">do.mapping =</span> <span class="cn">FALSE</span></span>
<span id="cb19-726"><a href="#cb19-726" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-727"><a href="#cb19-727" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-728"><a href="#cb19-728" aria-hidden="true" tabindex="-1"></a>abim_100<span class="sc">$</span>GENEFU.ROR <span class="ot">&lt;-</span> ror_abim<span class="sc">$</span>score</span>
<span id="cb19-729"><a href="#cb19-729" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-730"><a href="#cb19-730" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(abim_100) <span class="sc">%&gt;%</span></span>
<span id="cb19-731"><a href="#cb19-731" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span></span>
<span id="cb19-732"><a href="#cb19-732" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> GENEFU.ROR, <span class="at">y =</span> ProsignaFT.ROR, <span class="at">color =</span> ProsignaFT.Subtype)) <span class="sc">+</span></span>
<span id="cb19-733"><a href="#cb19-733" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span> </span>
<span id="cb19-734"><a href="#cb19-734" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_smooth</span>(</span>
<span id="cb19-735"><a href="#cb19-735" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">"black"</span>,</span>
<span id="cb19-736"><a href="#cb19-736" aria-hidden="true" tabindex="-1"></a>        <span class="at">method =</span> <span class="st">"loess"</span>,</span>
<span id="cb19-737"><a href="#cb19-737" aria-hidden="true" tabindex="-1"></a>        <span class="at">formula =</span> <span class="st">"y ~ x"</span></span>
<span id="cb19-738"><a href="#cb19-738" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb19-739"><a href="#cb19-739" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"ABiM 100 cohort"</span>) <span class="sc">+</span></span>
<span id="cb19-740"><a href="#cb19-740" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>) <span class="sc">+</span> </span>
<span id="cb19-741"><a href="#cb19-741" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span></span>
<span id="cb19-742"><a href="#cb19-742" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb19-743"><a href="#cb19-743" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-744"><a href="#cb19-744" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-745"><a href="#cb19-745" aria-hidden="true" tabindex="-1"></a>The correlation is relatively good and respects the molecular subtypes</span>
<span id="cb19-746"><a href="#cb19-746" aria-hidden="true" tabindex="-1"></a>in general. The ROR obtained from genefu for some luminal B are not </span>
<span id="cb19-747"><a href="#cb19-747" aria-hidden="true" tabindex="-1"></a>quite right as they are below 50. </span>
<span id="cb19-748"><a href="#cb19-748" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-749"><a href="#cb19-749" aria-hidden="true" tabindex="-1"></a>We now move on to the other signatures: Mammaprint, EndoPredict (EP) and </span>
<span id="cb19-750"><a href="#cb19-750" aria-hidden="true" tabindex="-1"></a>OncotypeDX Risk Score (RS).</span>
<span id="cb19-751"><a href="#cb19-751" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-754"><a href="#cb19-754" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-755"><a href="#cb19-755" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-ror-mammaprint-genefu-abim</span></span>
<span id="cb19-756"><a href="#cb19-756" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Correlation between ROR obtained from prosigna assay and </span></span>
<span id="cb19-757"><a href="#cb19-757" aria-hidden="true" tabindex="-1"></a><span class="co">#|     the mammaprint signature obtained from the R package genefu</span></span>
<span id="cb19-758"><a href="#cb19-758" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-759"><a href="#cb19-759" aria-hidden="true" tabindex="-1"></a>annot_abim <span class="ot">&lt;-</span> annotables<span class="sc">::</span>grch38 <span class="sc">%&gt;%</span></span>
<span id="cb19-760"><a href="#cb19-760" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb19-761"><a href="#cb19-761" aria-hidden="true" tabindex="-1"></a>            <span class="at">EntrezGene.ID =</span> entrez,</span>
<span id="cb19-762"><a href="#cb19-762" aria-hidden="true" tabindex="-1"></a>            <span class="at">Gene.Symbol =</span> symbol</span>
<span id="cb19-763"><a href="#cb19-763" aria-hidden="true" tabindex="-1"></a>        ) <span class="sc">%&gt;%</span></span>
<span id="cb19-764"><a href="#cb19-764" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">duplicated</span>(symbol)) <span class="sc">%&gt;%</span> </span>
<span id="cb19-765"><a href="#cb19-765" aria-hidden="true" tabindex="-1"></a>        data.frame <span class="sc">%&gt;%</span></span>
<span id="cb19-766"><a href="#cb19-766" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">rownames&lt;-</span><span class="st">`</span>(.<span class="sc">$</span>symbol)</span>
<span id="cb19-767"><a href="#cb19-767" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-768"><a href="#cb19-768" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(sig.gene70)</span>
<span id="cb19-769"><a href="#cb19-769" aria-hidden="true" tabindex="-1"></a>sig.gene70 <span class="ot">&lt;-</span> sig.gene70 <span class="sc">%&gt;%</span></span>
<span id="cb19-770"><a href="#cb19-770" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(HUGO.gene.symbol)) <span class="sc">%&gt;%</span></span>
<span id="cb19-771"><a href="#cb19-771" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">distinct</span>(HUGO.gene.symbol, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-772"><a href="#cb19-772" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">rownames&lt;-</span><span class="st">`</span>(.<span class="sc">$</span>HUGO.gene.symbol)</span>
<span id="cb19-773"><a href="#cb19-773" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-774"><a href="#cb19-774" aria-hidden="true" tabindex="-1"></a>erp_abim <span class="ot">&lt;-</span> <span class="fu">colData</span>(abim_100) <span class="sc">%&gt;%</span></span>
<span id="cb19-775"><a href="#cb19-775" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span></span>
<span id="cb19-776"><a href="#cb19-776" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(ER <span class="sc">==</span> <span class="st">"Positive"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-777"><a href="#cb19-777" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">pull</span>(sample_name)</span>
<span id="cb19-778"><a href="#cb19-778" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-779"><a href="#cb19-779" aria-hidden="true" tabindex="-1"></a>abim_100 <span class="ot">&lt;-</span> abim_100[, erp_abim]</span>
<span id="cb19-780"><a href="#cb19-780" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-781"><a href="#cb19-781" aria-hidden="true" tabindex="-1"></a>mammaprint_abim <span class="ot">&lt;-</span> genefu<span class="sc">::</span><span class="fu">gene70</span>(</span>
<span id="cb19-782"><a href="#cb19-782" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> <span class="fu">t</span>(<span class="fu">assay</span>(abim_100, <span class="st">"log2fpkm"</span>)), </span>
<span id="cb19-783"><a href="#cb19-783" aria-hidden="true" tabindex="-1"></a>    <span class="at">annot =</span> annot_abim,</span>
<span id="cb19-784"><a href="#cb19-784" aria-hidden="true" tabindex="-1"></a>    <span class="at">do.mapping =</span> <span class="cn">FALSE</span>,</span>
<span id="cb19-785"><a href="#cb19-785" aria-hidden="true" tabindex="-1"></a>    <span class="at">std =</span> <span class="st">"robust"</span>,</span>
<span id="cb19-786"><a href="#cb19-786" aria-hidden="true" tabindex="-1"></a>    <span class="at">verbose =</span> <span class="cn">TRUE</span></span>
<span id="cb19-787"><a href="#cb19-787" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-788"><a href="#cb19-788" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-789"><a href="#cb19-789" aria-hidden="true" tabindex="-1"></a>abim_100<span class="sc">$</span>GENEFU.GENE70 <span class="ot">&lt;-</span> mammaprint_abim<span class="sc">$</span>score</span>
<span id="cb19-790"><a href="#cb19-790" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-791"><a href="#cb19-791" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(abim_100) <span class="sc">%&gt;%</span></span>
<span id="cb19-792"><a href="#cb19-792" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span></span>
<span id="cb19-793"><a href="#cb19-793" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(</span>
<span id="cb19-794"><a href="#cb19-794" aria-hidden="true" tabindex="-1"></a>        <span class="fu">aes</span>(</span>
<span id="cb19-795"><a href="#cb19-795" aria-hidden="true" tabindex="-1"></a>            <span class="at">x =</span> GENEFU.GENE70, </span>
<span id="cb19-796"><a href="#cb19-796" aria-hidden="true" tabindex="-1"></a>            <span class="at">y =</span> ProsignaFT.ROR, </span>
<span id="cb19-797"><a href="#cb19-797" aria-hidden="true" tabindex="-1"></a>            <span class="at">color =</span> ProsignaFT.Subtype</span>
<span id="cb19-798"><a href="#cb19-798" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb19-799"><a href="#cb19-799" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb19-800"><a href="#cb19-800" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span> </span>
<span id="cb19-801"><a href="#cb19-801" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_smooth</span>(</span>
<span id="cb19-802"><a href="#cb19-802" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">"black"</span>,</span>
<span id="cb19-803"><a href="#cb19-803" aria-hidden="true" tabindex="-1"></a>        <span class="at">method =</span> <span class="st">"loess"</span>,</span>
<span id="cb19-804"><a href="#cb19-804" aria-hidden="true" tabindex="-1"></a>        <span class="at">formula =</span> <span class="st">"y ~ x"</span></span>
<span id="cb19-805"><a href="#cb19-805" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb19-806"><a href="#cb19-806" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"ABiM 100 cohort"</span>) <span class="sc">+</span></span>
<span id="cb19-807"><a href="#cb19-807" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>) <span class="sc">+</span> </span>
<span id="cb19-808"><a href="#cb19-808" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span></span>
<span id="cb19-809"><a href="#cb19-809" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb19-810"><a href="#cb19-810" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-811"><a href="#cb19-811" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-812"><a href="#cb19-812" aria-hidden="true" tabindex="-1"></a>@fig-ror-mammaprint-genefu-abim shows the correlation between </span>
<span id="cb19-813"><a href="#cb19-813" aria-hidden="true" tabindex="-1"></a>the mammaprint signature (named GENE70) and the ROR from prosigna. </span>
<span id="cb19-814"><a href="#cb19-814" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-815"><a href="#cb19-815" aria-hidden="true" tabindex="-1"></a>Let us now compare the GENE70 signature with the risk score developed in </span>
<span id="cb19-816"><a href="#cb19-816" aria-hidden="true" tabindex="-1"></a>the previous chapters. We use the ABiM 100 cohort to make the comparison. </span>
<span id="cb19-817"><a href="#cb19-817" aria-hidden="true" tabindex="-1"></a>One side note, we are using only 51 out of the 70 probes available in the</span>
<span id="cb19-818"><a href="#cb19-818" aria-hidden="true" tabindex="-1"></a>package. 15 probes don't have any corresponding symbol. Out of the available</span>
<span id="cb19-819"><a href="#cb19-819" aria-hidden="true" tabindex="-1"></a>genes, some are duplicated. For the duplicated genes, the correlation </span>
<span id="cb19-820"><a href="#cb19-820" aria-hidden="true" tabindex="-1"></a>coefficients are all similar, so we can select just the first ocurrence </span>
<span id="cb19-821"><a href="#cb19-821" aria-hidden="true" tabindex="-1"></a>in the list.</span>
<span id="cb19-822"><a href="#cb19-822" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-825"><a href="#cb19-825" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-826"><a href="#cb19-826" aria-hidden="true" tabindex="-1"></a>coefs_risk_model <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(</span>
<span id="cb19-827"><a href="#cb19-827" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../../results/rds_files/risk_score/coefs_risk_model.rds"</span></span>
<span id="cb19-828"><a href="#cb19-828" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-829"><a href="#cb19-829" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-830"><a href="#cb19-830" aria-hidden="true" tabindex="-1"></a>abim_100<span class="sc">$</span>risk_score <span class="ot">&lt;-</span> <span class="fu">calculate_risk_score</span>(</span>
<span id="cb19-831"><a href="#cb19-831" aria-hidden="true" tabindex="-1"></a>    coefs_risk_model,</span>
<span id="cb19-832"><a href="#cb19-832" aria-hidden="true" tabindex="-1"></a>    abim_100_df_pca[<span class="fu">colnames</span>(abim_100), ],</span>
<span id="cb19-833"><a href="#cb19-833" aria-hidden="true" tabindex="-1"></a>    <span class="at">is_node =</span> <span class="cn">TRUE</span></span>
<span id="cb19-834"><a href="#cb19-834" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-835"><a href="#cb19-835" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-836"><a href="#cb19-836" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(abim_100) <span class="sc">%&gt;%</span></span>
<span id="cb19-837"><a href="#cb19-837" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span></span>
<span id="cb19-838"><a href="#cb19-838" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(</span>
<span id="cb19-839"><a href="#cb19-839" aria-hidden="true" tabindex="-1"></a>        <span class="fu">aes</span>(</span>
<span id="cb19-840"><a href="#cb19-840" aria-hidden="true" tabindex="-1"></a>            <span class="at">x =</span> GENEFU.GENE70, </span>
<span id="cb19-841"><a href="#cb19-841" aria-hidden="true" tabindex="-1"></a>            <span class="at">y =</span> risk_score, </span>
<span id="cb19-842"><a href="#cb19-842" aria-hidden="true" tabindex="-1"></a>            <span class="at">color =</span> ProsignaFT.Subtype</span>
<span id="cb19-843"><a href="#cb19-843" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb19-844"><a href="#cb19-844" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb19-845"><a href="#cb19-845" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span> </span>
<span id="cb19-846"><a href="#cb19-846" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_smooth</span>(</span>
<span id="cb19-847"><a href="#cb19-847" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">"black"</span>,</span>
<span id="cb19-848"><a href="#cb19-848" aria-hidden="true" tabindex="-1"></a>        <span class="at">method =</span> <span class="st">"loess"</span>,</span>
<span id="cb19-849"><a href="#cb19-849" aria-hidden="true" tabindex="-1"></a>        <span class="at">formula =</span> <span class="st">"y ~ x"</span></span>
<span id="cb19-850"><a href="#cb19-850" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb19-851"><a href="#cb19-851" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>) <span class="sc">+</span> </span>
<span id="cb19-852"><a href="#cb19-852" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb19-853"><a href="#cb19-853" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">"ABiM 100 cohort, ER+ BC only"</span>,</span>
<span id="cb19-854"><a href="#cb19-854" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"Risk score"</span></span>
<span id="cb19-855"><a href="#cb19-855" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-856"><a href="#cb19-856" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span></span>
<span id="cb19-857"><a href="#cb19-857" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb19-858"><a href="#cb19-858" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-859"><a href="#cb19-859" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-860"><a href="#cb19-860" aria-hidden="true" tabindex="-1"></a>There is a good correlation between the two signatures, suggesting </span>
<span id="cb19-861"><a href="#cb19-861" aria-hidden="true" tabindex="-1"></a>that the Mammaprint signature is somehow based on the position of the samples in</span>
<span id="cb19-862"><a href="#cb19-862" aria-hidden="true" tabindex="-1"></a>the molecular landscape. </span>
<span id="cb19-863"><a href="#cb19-863" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-864"><a href="#cb19-864" aria-hidden="true" tabindex="-1"></a>We now proceed and calculate the mammaprint scores for SCANB and </span>
<span id="cb19-865"><a href="#cb19-865" aria-hidden="true" tabindex="-1"></a>METABRIC as well and only for ER+ BC samples.</span>
<span id="cb19-866"><a href="#cb19-866" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-869"><a href="#cb19-869" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-870"><a href="#cb19-870" aria-hidden="true" tabindex="-1"></a>mammaprint_datasets <span class="ot">&lt;-</span> <span class="fu">mapply</span>(</span>
<span id="cb19-871"><a href="#cb19-871" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(x, y){</span>
<span id="cb19-872"><a href="#cb19-872" aria-hidden="true" tabindex="-1"></a>        genefu<span class="sc">::</span><span class="fu">gene70</span>(</span>
<span id="cb19-873"><a href="#cb19-873" aria-hidden="true" tabindex="-1"></a>            <span class="at">data =</span> <span class="fu">t</span>(<span class="fu">assay</span>(x, y)),</span>
<span id="cb19-874"><a href="#cb19-874" aria-hidden="true" tabindex="-1"></a>            <span class="at">annot =</span> annot_abim,</span>
<span id="cb19-875"><a href="#cb19-875" aria-hidden="true" tabindex="-1"></a>            <span class="at">do.mapping =</span> <span class="cn">FALSE</span>,</span>
<span id="cb19-876"><a href="#cb19-876" aria-hidden="true" tabindex="-1"></a>            <span class="at">std =</span> <span class="st">"robust"</span>,</span>
<span id="cb19-877"><a href="#cb19-877" aria-hidden="true" tabindex="-1"></a>            <span class="at">verbose =</span> <span class="cn">TRUE</span></span>
<span id="cb19-878"><a href="#cb19-878" aria-hidden="true" tabindex="-1"></a>        )<span class="sc">$</span>score <span class="sc">%&gt;%</span></span>
<span id="cb19-879"><a href="#cb19-879" aria-hidden="true" tabindex="-1"></a>            data.frame <span class="sc">%&gt;%</span></span>
<span id="cb19-880"><a href="#cb19-880" aria-hidden="true" tabindex="-1"></a>            <span class="st">`</span><span class="at">colnames&lt;-</span><span class="st">`</span>(<span class="fu">c</span>(<span class="st">"mammaprint"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb19-881"><a href="#cb19-881" aria-hidden="true" tabindex="-1"></a>            tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"sample_name"</span>) </span>
<span id="cb19-882"><a href="#cb19-882" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb19-883"><a href="#cb19-883" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> datasets,</span>
<span id="cb19-884"><a href="#cb19-884" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="fu">c</span>(which_exp, <span class="st">"poetic"</span> <span class="ot">=</span> <span class="st">"normalized_intensity"</span>),</span>
<span id="cb19-885"><a href="#cb19-885" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>,</span>
<span id="cb19-886"><a href="#cb19-886" aria-hidden="true" tabindex="-1"></a>    <span class="at">SIMPLIFY =</span> <span class="cn">FALSE</span></span>
<span id="cb19-887"><a href="#cb19-887" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(<span class="at">.id =</span> <span class="st">"cohort"</span>)</span>
<span id="cb19-888"><a href="#cb19-888" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-889"><a href="#cb19-889" aria-hidden="true" tabindex="-1"></a>scanb_metabric_erp <span class="ot">&lt;-</span> df_pca <span class="sc">%&gt;%</span></span>
<span id="cb19-890"><a href="#cb19-890" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb19-891"><a href="#cb19-891" aria-hidden="true" tabindex="-1"></a>        cohort <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"scanb"</span>, <span class="st">"metabric"</span>) <span class="sc">&amp;</span></span>
<span id="cb19-892"><a href="#cb19-892" aria-hidden="true" tabindex="-1"></a>            er_status <span class="sc">==</span> <span class="st">"pos"</span></span>
<span id="cb19-893"><a href="#cb19-893" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span> </span>
<span id="cb19-894"><a href="#cb19-894" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">inner_join</span>(</span>
<span id="cb19-895"><a href="#cb19-895" aria-hidden="true" tabindex="-1"></a>        mammaprint_datasets,</span>
<span id="cb19-896"><a href="#cb19-896" aria-hidden="true" tabindex="-1"></a>        <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"sample_name"</span>, <span class="st">"cohort"</span>)</span>
<span id="cb19-897"><a href="#cb19-897" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb19-898"><a href="#cb19-898" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-899"><a href="#cb19-899" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-900"><a href="#cb19-900" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=5}</span></span>
<span id="cb19-901"><a href="#cb19-901" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-met-scanb-mammaprint</span></span>
<span id="cb19-902"><a href="#cb19-902" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Embedding of ER+ BC samples colored by the mammaprint </span></span>
<span id="cb19-903"><a href="#cb19-903" aria-hidden="true" tabindex="-1"></a><span class="co">#|     score.</span></span>
<span id="cb19-904"><a href="#cb19-904" aria-hidden="true" tabindex="-1"></a>scanb_metabric_erp <span class="sc">%&gt;%</span></span>
<span id="cb19-905"><a href="#cb19-905" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> PC3, <span class="at">y =</span> PC4, <span class="at">z =</span> mammaprint)) <span class="sc">+</span> </span>
<span id="cb19-906"><a href="#cb19-906" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">stat_summary_hex</span>(<span class="at">bins =</span> <span class="dv">20</span>) <span class="sc">+</span></span>
<span id="cb19-907"><a href="#cb19-907" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span> cohort) <span class="sc">+</span></span>
<span id="cb19-908"><a href="#cb19-908" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_fill_viridis_c</span>() <span class="sc">+</span></span>
<span id="cb19-909"><a href="#cb19-909" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">fill =</span> <span class="st">"Mammaprint"</span>) <span class="sc">+</span></span>
<span id="cb19-910"><a href="#cb19-910" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>)</span>
<span id="cb19-911"><a href="#cb19-911" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-912"><a href="#cb19-912" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-913"><a href="#cb19-913" aria-hidden="true" tabindex="-1"></a>We see that the pattern is very close to what we have from the risk score </span>
<span id="cb19-914"><a href="#cb19-914" aria-hidden="true" tabindex="-1"></a>created in the previous chapter. The figure below shows the correlation</span>
<span id="cb19-915"><a href="#cb19-915" aria-hidden="true" tabindex="-1"></a>between mammaprint and the risk score once again.</span>
<span id="cb19-916"><a href="#cb19-916" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-919"><a href="#cb19-919" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-920"><a href="#cb19-920" aria-hidden="true" tabindex="-1"></a>endo_only_scanb <span class="ot">&lt;-</span> df_pca <span class="sc">%&gt;%</span> </span>
<span id="cb19-921"><a href="#cb19-921" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"scanb"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-922"><a href="#cb19-922" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">node_status =</span> <span class="fu">ifelse</span>(</span>
<span id="cb19-923"><a href="#cb19-923" aria-hidden="true" tabindex="-1"></a>        node_group <span class="sc">==</span> <span class="st">"N0"</span>, </span>
<span id="cb19-924"><a href="#cb19-924" aria-hidden="true" tabindex="-1"></a>        <span class="st">"neg"</span>,</span>
<span id="cb19-925"><a href="#cb19-925" aria-hidden="true" tabindex="-1"></a>        <span class="st">"pos"</span></span>
<span id="cb19-926"><a href="#cb19-926" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">%&gt;%</span> </span>
<span id="cb19-927"><a href="#cb19-927" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">node_status =</span> <span class="fu">factor</span>(</span>
<span id="cb19-928"><a href="#cb19-928" aria-hidden="true" tabindex="-1"></a>        node_status, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"neg"</span>, <span class="st">"pos"</span>)</span>
<span id="cb19-929"><a href="#cb19-929" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">%&gt;%</span></span>
<span id="cb19-930"><a href="#cb19-930" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb19-931"><a href="#cb19-931" aria-hidden="true" tabindex="-1"></a>            TreatGroup <span class="sc">==</span> <span class="st">"Endo"</span> <span class="sc">&amp;</span> </span>
<span id="cb19-932"><a href="#cb19-932" aria-hidden="true" tabindex="-1"></a>            <span class="sc">!</span><span class="fu">is.na</span>(rfs_status) <span class="sc">&amp;</span> </span>
<span id="cb19-933"><a href="#cb19-933" aria-hidden="true" tabindex="-1"></a>            <span class="sc">!</span><span class="fu">is.na</span>(rfs_months) <span class="sc">&amp;</span></span>
<span id="cb19-934"><a href="#cb19-934" aria-hidden="true" tabindex="-1"></a>            er_status <span class="sc">==</span> <span class="st">"pos"</span> <span class="sc">&amp;</span></span>
<span id="cb19-935"><a href="#cb19-935" aria-hidden="true" tabindex="-1"></a>            <span class="sc">!</span><span class="fu">is.na</span>(age) <span class="sc">&amp;</span></span>
<span id="cb19-936"><a href="#cb19-936" aria-hidden="true" tabindex="-1"></a>            <span class="sc">!</span><span class="fu">is.na</span>(tumor_size) <span class="sc">&amp;</span></span>
<span id="cb19-937"><a href="#cb19-937" aria-hidden="true" tabindex="-1"></a>            <span class="sc">!</span><span class="fu">is.na</span>(node_status)</span>
<span id="cb19-938"><a href="#cb19-938" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb19-939"><a href="#cb19-939" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-940"><a href="#cb19-940" aria-hidden="true" tabindex="-1"></a>endo_only_metabric <span class="ot">&lt;-</span> df_pca <span class="sc">%&gt;%</span> </span>
<span id="cb19-941"><a href="#cb19-941" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"metabric"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-942"><a href="#cb19-942" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">TreatGroup =</span> dplyr<span class="sc">::</span><span class="fu">case_when</span>(</span>
<span id="cb19-943"><a href="#cb19-943" aria-hidden="true" tabindex="-1"></a>        HORMONE_THERAPY <span class="sc">==</span> <span class="st">"YES"</span> <span class="sc">&amp;</span> CHEMOTHERAPY <span class="sc">==</span> <span class="st">"YES"</span> <span class="sc">~</span> <span class="st">"ChemoEndo"</span>,</span>
<span id="cb19-944"><a href="#cb19-944" aria-hidden="true" tabindex="-1"></a>        HORMONE_THERAPY <span class="sc">==</span> <span class="st">"YES"</span> <span class="sc">&amp;</span> CHEMOTHERAPY <span class="sc">==</span> <span class="st">"NO"</span> <span class="sc">~</span> <span class="st">"Endo"</span>,</span>
<span id="cb19-945"><a href="#cb19-945" aria-hidden="true" tabindex="-1"></a>        <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"Other"</span></span>
<span id="cb19-946"><a href="#cb19-946" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">%&gt;%</span> </span>
<span id="cb19-947"><a href="#cb19-947" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">node_status =</span> <span class="fu">ifelse</span>(</span>
<span id="cb19-948"><a href="#cb19-948" aria-hidden="true" tabindex="-1"></a>        LYMPH_NODES_EXAMINED_POSITIVE <span class="sc">&gt;</span> <span class="dv">0</span>, </span>
<span id="cb19-949"><a href="#cb19-949" aria-hidden="true" tabindex="-1"></a>        <span class="st">"pos"</span>,</span>
<span id="cb19-950"><a href="#cb19-950" aria-hidden="true" tabindex="-1"></a>        <span class="st">"neg"</span></span>
<span id="cb19-951"><a href="#cb19-951" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">%&gt;%</span> </span>
<span id="cb19-952"><a href="#cb19-952" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb19-953"><a href="#cb19-953" aria-hidden="true" tabindex="-1"></a>        <span class="at">node_status =</span> <span class="fu">factor</span>(node_status, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"neg"</span>, <span class="st">"pos"</span>)),</span>
<span id="cb19-954"><a href="#cb19-954" aria-hidden="true" tabindex="-1"></a>        <span class="at">tumor_size =</span> TUMOR_SIZE</span>
<span id="cb19-955"><a href="#cb19-955" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span> </span>
<span id="cb19-956"><a href="#cb19-956" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb19-957"><a href="#cb19-957" aria-hidden="true" tabindex="-1"></a>            HORMONE_THERAPY <span class="sc">==</span> <span class="st">"YES"</span> <span class="sc">&amp;</span> </span>
<span id="cb19-958"><a href="#cb19-958" aria-hidden="true" tabindex="-1"></a>            CHEMOTHERAPY <span class="sc">==</span> <span class="st">"NO"</span> <span class="sc">&amp;</span></span>
<span id="cb19-959"><a href="#cb19-959" aria-hidden="true" tabindex="-1"></a>            <span class="sc">!</span><span class="fu">is.na</span>(rfs_status) <span class="sc">&amp;</span> </span>
<span id="cb19-960"><a href="#cb19-960" aria-hidden="true" tabindex="-1"></a>            <span class="sc">!</span><span class="fu">is.na</span>(rfs_months) <span class="sc">&amp;</span></span>
<span id="cb19-961"><a href="#cb19-961" aria-hidden="true" tabindex="-1"></a>            er_status <span class="sc">==</span> <span class="st">"pos"</span> <span class="sc">&amp;</span></span>
<span id="cb19-962"><a href="#cb19-962" aria-hidden="true" tabindex="-1"></a>            <span class="sc">!</span><span class="fu">is.na</span>(age) <span class="sc">&amp;</span></span>
<span id="cb19-963"><a href="#cb19-963" aria-hidden="true" tabindex="-1"></a>            <span class="sc">!</span><span class="fu">is.na</span>(tumor_size) <span class="sc">&amp;</span></span>
<span id="cb19-964"><a href="#cb19-964" aria-hidden="true" tabindex="-1"></a>            <span class="sc">!</span><span class="fu">is.na</span>(node_status)</span>
<span id="cb19-965"><a href="#cb19-965" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb19-966"><a href="#cb19-966" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-967"><a href="#cb19-967" aria-hidden="true" tabindex="-1"></a>endo_scanb_metabric <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(</span>
<span id="cb19-968"><a href="#cb19-968" aria-hidden="true" tabindex="-1"></a>    endo_only_metabric,</span>
<span id="cb19-969"><a href="#cb19-969" aria-hidden="true" tabindex="-1"></a>    endo_only_scanb</span>
<span id="cb19-970"><a href="#cb19-970" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb19-971"><a href="#cb19-971" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">inner_join</span>(</span>
<span id="cb19-972"><a href="#cb19-972" aria-hidden="true" tabindex="-1"></a>        mammaprint_datasets,</span>
<span id="cb19-973"><a href="#cb19-973" aria-hidden="true" tabindex="-1"></a>        <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"sample_name"</span>, <span class="st">"cohort"</span>)</span>
<span id="cb19-974"><a href="#cb19-974" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb19-975"><a href="#cb19-975" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-976"><a href="#cb19-976" aria-hidden="true" tabindex="-1"></a>endo_scanb_metabric<span class="sc">$</span>risk_score <span class="ot">&lt;-</span> <span class="fu">calculate_risk_score</span>(</span>
<span id="cb19-977"><a href="#cb19-977" aria-hidden="true" tabindex="-1"></a>    coefs_risk_model, endo_scanb_metabric</span>
<span id="cb19-978"><a href="#cb19-978" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-979"><a href="#cb19-979" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-980"><a href="#cb19-980" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-981"><a href="#cb19-981" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=14, fig.height=8}</span></span>
<span id="cb19-982"><a href="#cb19-982" aria-hidden="true" tabindex="-1"></a>endo_scanb_metabric <span class="sc">%&gt;%</span></span>
<span id="cb19-983"><a href="#cb19-983" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(pam50 <span class="sc">!=</span> <span class="st">"claudin-low"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-984"><a href="#cb19-984" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(</span>
<span id="cb19-985"><a href="#cb19-985" aria-hidden="true" tabindex="-1"></a>        <span class="at">cols =</span> dplyr<span class="sc">::</span><span class="fu">all_of</span>(<span class="fu">c</span>(</span>
<span id="cb19-986"><a href="#cb19-986" aria-hidden="true" tabindex="-1"></a>            <span class="st">"risk_score"</span>,</span>
<span id="cb19-987"><a href="#cb19-987" aria-hidden="true" tabindex="-1"></a>            <span class="st">"HALLMARK_G2M_CHECKPOINT"</span>,</span>
<span id="cb19-988"><a href="#cb19-988" aria-hidden="true" tabindex="-1"></a>            <span class="st">"SET_ERPR"</span></span>
<span id="cb19-989"><a href="#cb19-989" aria-hidden="true" tabindex="-1"></a>        )),</span>
<span id="cb19-990"><a href="#cb19-990" aria-hidden="true" tabindex="-1"></a>        <span class="at">values_to =</span> <span class="st">"score"</span>,</span>
<span id="cb19-991"><a href="#cb19-991" aria-hidden="true" tabindex="-1"></a>        <span class="at">names_to =</span> <span class="st">"pathway"</span></span>
<span id="cb19-992"><a href="#cb19-992" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb19-993"><a href="#cb19-993" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> mammaprint, <span class="at">y =</span> score, <span class="at">color =</span> pam50)) <span class="sc">+</span> </span>
<span id="cb19-994"><a href="#cb19-994" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb19-995"><a href="#cb19-995" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_smooth</span>(</span>
<span id="cb19-996"><a href="#cb19-996" aria-hidden="true" tabindex="-1"></a>        <span class="at">method =</span> <span class="st">"loess"</span>,</span>
<span id="cb19-997"><a href="#cb19-997" aria-hidden="true" tabindex="-1"></a>        <span class="at">formula =</span> y <span class="sc">~</span> x,</span>
<span id="cb19-998"><a href="#cb19-998" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">"black"</span></span>
<span id="cb19-999"><a href="#cb19-999" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb19-1000"><a href="#cb19-1000" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(</span>
<span id="cb19-1001"><a href="#cb19-1001" aria-hidden="true" tabindex="-1"></a>        cohort<span class="sc">~</span>pathway, </span>
<span id="cb19-1002"><a href="#cb19-1002" aria-hidden="true" tabindex="-1"></a>        <span class="at">scales =</span> <span class="st">"free_y"</span>,</span>
<span id="cb19-1003"><a href="#cb19-1003" aria-hidden="true" tabindex="-1"></a>        <span class="at">labeller =</span> <span class="fu">as_labeller</span>(<span class="fu">c</span>(</span>
<span id="cb19-1004"><a href="#cb19-1004" aria-hidden="true" tabindex="-1"></a>            <span class="st">"scanb"</span> <span class="ot">=</span> <span class="st">"SCANB"</span>,</span>
<span id="cb19-1005"><a href="#cb19-1005" aria-hidden="true" tabindex="-1"></a>            <span class="st">"metabric"</span> <span class="ot">=</span> <span class="st">"METABRIC"</span>,</span>
<span id="cb19-1006"><a href="#cb19-1006" aria-hidden="true" tabindex="-1"></a>            <span class="st">"HALLMARK_G2M_CHECKPOINT"</span> <span class="ot">=</span> <span class="st">"G2M"</span>,</span>
<span id="cb19-1007"><a href="#cb19-1007" aria-hidden="true" tabindex="-1"></a>            <span class="st">"SET_ERPR"</span> <span class="ot">=</span> <span class="st">"SET ER/PR"</span>,</span>
<span id="cb19-1008"><a href="#cb19-1008" aria-hidden="true" tabindex="-1"></a>            <span class="st">"risk_score"</span> <span class="ot">=</span> <span class="st">"Risk score"</span></span>
<span id="cb19-1009"><a href="#cb19-1009" aria-hidden="true" tabindex="-1"></a>        ))</span>
<span id="cb19-1010"><a href="#cb19-1010" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-1011"><a href="#cb19-1011" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>) <span class="sc">+</span> </span>
<span id="cb19-1012"><a href="#cb19-1012" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">get_colors_pam50</span>(endo_scanb_metabric)) <span class="sc">+</span></span>
<span id="cb19-1013"><a href="#cb19-1013" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span> </span>
<span id="cb19-1014"><a href="#cb19-1014" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb19-1015"><a href="#cb19-1015" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-1016"><a href="#cb19-1016" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1017"><a href="#cb19-1017" aria-hidden="true" tabindex="-1"></a>Now for OncotypeDX:</span>
<span id="cb19-1018"><a href="#cb19-1018" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1021"><a href="#cb19-1021" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-1022"><a href="#cb19-1022" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-ror-genefu-abim-oncotype</span></span>
<span id="cb19-1023"><a href="#cb19-1023" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Correlation between oncotypeDX and </span></span>
<span id="cb19-1024"><a href="#cb19-1024" aria-hidden="true" tabindex="-1"></a><span class="co">#|     R package genefu</span></span>
<span id="cb19-1025"><a href="#cb19-1025" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1026"><a href="#cb19-1026" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(sig.endoPredict)</span>
<span id="cb19-1027"><a href="#cb19-1027" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(sig.oncotypedx)</span>
<span id="cb19-1028"><a href="#cb19-1028" aria-hidden="true" tabindex="-1"></a>ep_abim <span class="ot">&lt;-</span> genefu<span class="sc">::</span><span class="fu">endoPredict</span>(</span>
<span id="cb19-1029"><a href="#cb19-1029" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> <span class="fu">t</span>(<span class="fu">assay</span>(abim_100, <span class="st">"log2fpkm"</span>)), </span>
<span id="cb19-1030"><a href="#cb19-1030" aria-hidden="true" tabindex="-1"></a>    <span class="at">annot =</span> annot_abim,</span>
<span id="cb19-1031"><a href="#cb19-1031" aria-hidden="true" tabindex="-1"></a>    <span class="at">do.mapping =</span> <span class="cn">TRUE</span>,</span>
<span id="cb19-1032"><a href="#cb19-1032" aria-hidden="true" tabindex="-1"></a>    <span class="at">verbose =</span> <span class="cn">TRUE</span></span>
<span id="cb19-1033"><a href="#cb19-1033" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-1034"><a href="#cb19-1034" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1035"><a href="#cb19-1035" aria-hidden="true" tabindex="-1"></a>oncotypedx_abim <span class="ot">&lt;-</span> genefu<span class="sc">::</span><span class="fu">oncotypedx</span>(</span>
<span id="cb19-1036"><a href="#cb19-1036" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> <span class="fu">t</span>(<span class="fu">assay</span>(abim_100, <span class="st">"log2fpkm"</span>)), </span>
<span id="cb19-1037"><a href="#cb19-1037" aria-hidden="true" tabindex="-1"></a>    <span class="at">annot =</span> annot_abim,</span>
<span id="cb19-1038"><a href="#cb19-1038" aria-hidden="true" tabindex="-1"></a>    <span class="at">do.mapping =</span> <span class="cn">TRUE</span>,</span>
<span id="cb19-1039"><a href="#cb19-1039" aria-hidden="true" tabindex="-1"></a>    <span class="at">verbose =</span> <span class="cn">TRUE</span>,</span>
<span id="cb19-1040"><a href="#cb19-1040" aria-hidden="true" tabindex="-1"></a>    <span class="at">do.scaling =</span> <span class="cn">FALSE</span></span>
<span id="cb19-1041"><a href="#cb19-1041" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-1042"><a href="#cb19-1042" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1043"><a href="#cb19-1043" aria-hidden="true" tabindex="-1"></a>abim_100<span class="sc">$</span>GENEFU.ONCOTYPEDX <span class="ot">&lt;-</span> oncotypedx_abim<span class="sc">$</span>score</span>
<span id="cb19-1044"><a href="#cb19-1044" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1045"><a href="#cb19-1045" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(abim_100) <span class="sc">%&gt;%</span></span>
<span id="cb19-1046"><a href="#cb19-1046" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span></span>
<span id="cb19-1047"><a href="#cb19-1047" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(</span>
<span id="cb19-1048"><a href="#cb19-1048" aria-hidden="true" tabindex="-1"></a>        <span class="fu">aes</span>(</span>
<span id="cb19-1049"><a href="#cb19-1049" aria-hidden="true" tabindex="-1"></a>            <span class="at">x =</span> GENEFU.ONCOTYPEDX, </span>
<span id="cb19-1050"><a href="#cb19-1050" aria-hidden="true" tabindex="-1"></a>            <span class="at">y =</span> risk_score, </span>
<span id="cb19-1051"><a href="#cb19-1051" aria-hidden="true" tabindex="-1"></a>            <span class="at">color =</span> ProsignaFT.Subtype</span>
<span id="cb19-1052"><a href="#cb19-1052" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb19-1053"><a href="#cb19-1053" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb19-1054"><a href="#cb19-1054" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span> </span>
<span id="cb19-1055"><a href="#cb19-1055" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_smooth</span>(</span>
<span id="cb19-1056"><a href="#cb19-1056" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">"black"</span>,</span>
<span id="cb19-1057"><a href="#cb19-1057" aria-hidden="true" tabindex="-1"></a>        <span class="at">method =</span> <span class="st">"loess"</span>,</span>
<span id="cb19-1058"><a href="#cb19-1058" aria-hidden="true" tabindex="-1"></a>        <span class="at">formula =</span> <span class="st">"y ~ x"</span></span>
<span id="cb19-1059"><a href="#cb19-1059" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb19-1060"><a href="#cb19-1060" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb19-1061"><a href="#cb19-1061" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span></span>
<span id="cb19-1062"><a href="#cb19-1062" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb19-1063"><a href="#cb19-1063" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-1064"><a href="#cb19-1064" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1065"><a href="#cb19-1065" aria-hidden="true" tabindex="-1"></a>There is also a correlation between the risk score and oncotypeDX.</span>
<span id="cb19-1066"><a href="#cb19-1066" aria-hidden="true" tabindex="-1"></a>The paper https://www.nature.com/articles/s41523-022-00492-0#Sec8</span>
<span id="cb19-1067"><a href="#cb19-1067" aria-hidden="true" tabindex="-1"></a>also used the genefu for calculating the oncotypeDX. </span>
<span id="cb19-1068"><a href="#cb19-1068" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1069"><a href="#cb19-1069" aria-hidden="true" tabindex="-1"></a><span class="fu">### genefu and AIMS</span></span>
<span id="cb19-1070"><a href="#cb19-1070" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1071"><a href="#cb19-1071" aria-hidden="true" tabindex="-1"></a>Here we try to use AIMS to calculate the PAM50 subtypes for the</span>
<span id="cb19-1072"><a href="#cb19-1072" aria-hidden="true" tabindex="-1"></a>and compare with the results obtained previously. We want to compare</span>
<span id="cb19-1073"><a href="#cb19-1073" aria-hidden="true" tabindex="-1"></a>the classifications with the molecular landscape.</span>
<span id="cb19-1074"><a href="#cb19-1074" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1077"><a href="#cb19-1077" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-1078"><a href="#cb19-1078" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(AIMSmodel)</span>
<span id="cb19-1079"><a href="#cb19-1079" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1080"><a href="#cb19-1080" aria-hidden="true" tabindex="-1"></a><span class="co"># we have to match the symbols</span></span>
<span id="cb19-1081"><a href="#cb19-1081" aria-hidden="true" tabindex="-1"></a>annot_aims <span class="ot">&lt;-</span> annotables<span class="sc">::</span>grch38 <span class="sc">%&gt;%</span> </span>
<span id="cb19-1082"><a href="#cb19-1082" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(entrez, symbol) <span class="sc">%&gt;%</span></span>
<span id="cb19-1083"><a href="#cb19-1083" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">EntrezGene.ID =</span> entrez) <span class="sc">%&gt;%</span></span>
<span id="cb19-1084"><a href="#cb19-1084" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">EntrezGene.ID =</span> <span class="fu">as.character</span>(EntrezGene.ID)) <span class="sc">%&gt;%</span></span>
<span id="cb19-1085"><a href="#cb19-1085" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(symbol <span class="sc">%in%</span> <span class="fu">rownames</span>(pdx_tpm)) <span class="sc">%&gt;%</span></span>
<span id="cb19-1086"><a href="#cb19-1086" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">duplicated</span>(symbol)) <span class="sc">%&gt;%</span></span>
<span id="cb19-1087"><a href="#cb19-1087" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span></span>
<span id="cb19-1088"><a href="#cb19-1088" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">rownames&lt;-</span><span class="st">`</span>(.<span class="sc">$</span>symbol)</span>
<span id="cb19-1089"><a href="#cb19-1089" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1090"><a href="#cb19-1090" aria-hidden="true" tabindex="-1"></a>pdx_tpm <span class="ot">&lt;-</span> pdx_tpm[annot_aims<span class="sc">$</span>symbol, ]</span>
<span id="cb19-1091"><a href="#cb19-1091" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1092"><a href="#cb19-1092" aria-hidden="true" tabindex="-1"></a>sbt_pdx_AIMS <span class="ot">&lt;-</span> genefu<span class="sc">::</span><span class="fu">molecular.subtyping</span>(</span>
<span id="cb19-1093"><a href="#cb19-1093" aria-hidden="true" tabindex="-1"></a>    <span class="at">sbt.model =</span> <span class="st">"AIMS"</span>, </span>
<span id="cb19-1094"><a href="#cb19-1094" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> <span class="fu">assay</span>(pdx_tpm, <span class="st">"log2tpm"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-1095"><a href="#cb19-1095" aria-hidden="true" tabindex="-1"></a>        t,</span>
<span id="cb19-1096"><a href="#cb19-1096" aria-hidden="true" tabindex="-1"></a>    <span class="at">annot =</span> annot_aims,</span>
<span id="cb19-1097"><a href="#cb19-1097" aria-hidden="true" tabindex="-1"></a>    <span class="at">do.mapping =</span> <span class="cn">TRUE</span></span>
<span id="cb19-1098"><a href="#cb19-1098" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-1099"><a href="#cb19-1099" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1100"><a href="#cb19-1100" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1101"><a href="#cb19-1101" aria-hidden="true" tabindex="-1"></a>annot_aims_abim_100 <span class="ot">&lt;-</span> annotables<span class="sc">::</span>grch38 <span class="sc">%&gt;%</span> </span>
<span id="cb19-1102"><a href="#cb19-1102" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(entrez, symbol) <span class="sc">%&gt;%</span></span>
<span id="cb19-1103"><a href="#cb19-1103" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">EntrezGene.ID =</span> entrez) <span class="sc">%&gt;%</span></span>
<span id="cb19-1104"><a href="#cb19-1104" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">EntrezGene.ID =</span> <span class="fu">as.character</span>(EntrezGene.ID)) <span class="sc">%&gt;%</span></span>
<span id="cb19-1105"><a href="#cb19-1105" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(symbol <span class="sc">%in%</span> <span class="fu">rownames</span>(abim_100)) <span class="sc">%&gt;%</span></span>
<span id="cb19-1106"><a href="#cb19-1106" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">duplicated</span>(symbol)) <span class="sc">%&gt;%</span></span>
<span id="cb19-1107"><a href="#cb19-1107" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span></span>
<span id="cb19-1108"><a href="#cb19-1108" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">rownames&lt;-</span><span class="st">`</span>(.<span class="sc">$</span>symbol)</span>
<span id="cb19-1109"><a href="#cb19-1109" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1110"><a href="#cb19-1110" aria-hidden="true" tabindex="-1"></a>abim_100 <span class="ot">&lt;-</span> abim_100[annot_aims_abim_100<span class="sc">$</span>symbol, ]</span>
<span id="cb19-1111"><a href="#cb19-1111" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1112"><a href="#cb19-1112" aria-hidden="true" tabindex="-1"></a>sbt_aim_100_AIMS <span class="ot">&lt;-</span> genefu<span class="sc">::</span><span class="fu">molecular.subtyping</span>(</span>
<span id="cb19-1113"><a href="#cb19-1113" aria-hidden="true" tabindex="-1"></a>    <span class="at">sbt.model =</span> <span class="st">"AIMS"</span>, </span>
<span id="cb19-1114"><a href="#cb19-1114" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> <span class="fu">assay</span>(abim_100, <span class="st">"log2fpkm"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-1115"><a href="#cb19-1115" aria-hidden="true" tabindex="-1"></a>        t,</span>
<span id="cb19-1116"><a href="#cb19-1116" aria-hidden="true" tabindex="-1"></a>    <span class="at">annot =</span> annot_aims_abim_100,</span>
<span id="cb19-1117"><a href="#cb19-1117" aria-hidden="true" tabindex="-1"></a>    <span class="at">do.mapping =</span> <span class="cn">TRUE</span></span>
<span id="cb19-1118"><a href="#cb19-1118" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-1119"><a href="#cb19-1119" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1120"><a href="#cb19-1120" aria-hidden="true" tabindex="-1"></a>abim_100<span class="sc">$</span>aims_pam50 <span class="ot">&lt;-</span> sbt_aim_100_AIMS<span class="sc">$</span>subtype <span class="sc">%&gt;%</span> <span class="fu">as.vector</span>()</span>
<span id="cb19-1121"><a href="#cb19-1121" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1122"><a href="#cb19-1122" aria-hidden="true" tabindex="-1"></a><span class="co"># try with swiss normal as we have the baseline values</span></span>
<span id="cb19-1123"><a href="#cb19-1123" aria-hidden="true" tabindex="-1"></a>normal_swiss <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../../results/rds_files/validation/normal_swiss.rds"</span>)</span>
<span id="cb19-1124"><a href="#cb19-1124" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1125"><a href="#cb19-1125" aria-hidden="true" tabindex="-1"></a>annot_aims_normal_swiss <span class="ot">&lt;-</span> annotables<span class="sc">::</span>grch38 <span class="sc">%&gt;%</span> </span>
<span id="cb19-1126"><a href="#cb19-1126" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(entrez, symbol) <span class="sc">%&gt;%</span></span>
<span id="cb19-1127"><a href="#cb19-1127" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">EntrezGene.ID =</span> entrez) <span class="sc">%&gt;%</span></span>
<span id="cb19-1128"><a href="#cb19-1128" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">EntrezGene.ID =</span> <span class="fu">as.character</span>(EntrezGene.ID)) <span class="sc">%&gt;%</span></span>
<span id="cb19-1129"><a href="#cb19-1129" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(symbol <span class="sc">%in%</span> <span class="fu">rownames</span>(normal_swiss)) <span class="sc">%&gt;%</span></span>
<span id="cb19-1130"><a href="#cb19-1130" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">duplicated</span>(symbol)) <span class="sc">%&gt;%</span></span>
<span id="cb19-1131"><a href="#cb19-1131" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span></span>
<span id="cb19-1132"><a href="#cb19-1132" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">rownames&lt;-</span><span class="st">`</span>(.<span class="sc">$</span>symbol)</span>
<span id="cb19-1133"><a href="#cb19-1133" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1134"><a href="#cb19-1134" aria-hidden="true" tabindex="-1"></a>normal_swiss <span class="ot">&lt;-</span> normal_swiss[annot_aims_normal_swiss<span class="sc">$</span>symbol, ]</span>
<span id="cb19-1135"><a href="#cb19-1135" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1136"><a href="#cb19-1136" aria-hidden="true" tabindex="-1"></a>sbt_normal_swiss_AIMS <span class="ot">&lt;-</span> genefu<span class="sc">::</span><span class="fu">molecular.subtyping</span>(</span>
<span id="cb19-1137"><a href="#cb19-1137" aria-hidden="true" tabindex="-1"></a>    <span class="at">sbt.model =</span> <span class="st">"AIMS"</span>, </span>
<span id="cb19-1138"><a href="#cb19-1138" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> <span class="fu">assay</span>(normal_swiss, <span class="st">"log2fpkm"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-1139"><a href="#cb19-1139" aria-hidden="true" tabindex="-1"></a>        t,</span>
<span id="cb19-1140"><a href="#cb19-1140" aria-hidden="true" tabindex="-1"></a>    <span class="at">annot =</span> annot_aims_normal_swiss,</span>
<span id="cb19-1141"><a href="#cb19-1141" aria-hidden="true" tabindex="-1"></a>    <span class="at">do.mapping =</span> <span class="cn">TRUE</span></span>
<span id="cb19-1142"><a href="#cb19-1142" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-1143"><a href="#cb19-1143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1144"><a href="#cb19-1144" aria-hidden="true" tabindex="-1"></a>normal_swiss<span class="sc">$</span>aims_pam50 <span class="ot">&lt;-</span> sbt_normal_swiss_AIMS<span class="sc">$</span>subtype <span class="sc">%&gt;%</span> <span class="fu">as.vector</span>()</span>
<span id="cb19-1145"><a href="#cb19-1145" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1146"><a href="#cb19-1146" aria-hidden="true" tabindex="-1"></a><span class="co"># and we try with scanb as well</span></span>
<span id="cb19-1147"><a href="#cb19-1147" aria-hidden="true" tabindex="-1"></a>annot_aims_scanb <span class="ot">&lt;-</span> annotables<span class="sc">::</span>grch38 <span class="sc">%&gt;%</span> </span>
<span id="cb19-1148"><a href="#cb19-1148" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(entrez, symbol) <span class="sc">%&gt;%</span></span>
<span id="cb19-1149"><a href="#cb19-1149" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">EntrezGene.ID =</span> entrez) <span class="sc">%&gt;%</span></span>
<span id="cb19-1150"><a href="#cb19-1150" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">EntrezGene.ID =</span> <span class="fu">as.character</span>(EntrezGene.ID)) <span class="sc">%&gt;%</span></span>
<span id="cb19-1151"><a href="#cb19-1151" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(symbol <span class="sc">%in%</span> <span class="fu">rownames</span>(datasets<span class="sc">$</span>scanb)) <span class="sc">%&gt;%</span></span>
<span id="cb19-1152"><a href="#cb19-1152" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">duplicated</span>(symbol)) <span class="sc">%&gt;%</span></span>
<span id="cb19-1153"><a href="#cb19-1153" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span></span>
<span id="cb19-1154"><a href="#cb19-1154" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">rownames&lt;-</span><span class="st">`</span>(.<span class="sc">$</span>symbol)</span>
<span id="cb19-1155"><a href="#cb19-1155" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1156"><a href="#cb19-1156" aria-hidden="true" tabindex="-1"></a>scanb <span class="ot">&lt;-</span> datasets<span class="sc">$</span>scanb[annot_aims_scanb<span class="sc">$</span>symbol, ]</span>
<span id="cb19-1157"><a href="#cb19-1157" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1158"><a href="#cb19-1158" aria-hidden="true" tabindex="-1"></a>sbt_scanb_AIMS <span class="ot">&lt;-</span> genefu<span class="sc">::</span><span class="fu">molecular.subtyping</span>(</span>
<span id="cb19-1159"><a href="#cb19-1159" aria-hidden="true" tabindex="-1"></a>    <span class="at">sbt.model =</span> <span class="st">"AIMS"</span>, </span>
<span id="cb19-1160"><a href="#cb19-1160" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> <span class="fu">assay</span>(scanb, <span class="st">"logFPKM"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-1161"><a href="#cb19-1161" aria-hidden="true" tabindex="-1"></a>        t,</span>
<span id="cb19-1162"><a href="#cb19-1162" aria-hidden="true" tabindex="-1"></a>    <span class="at">annot =</span> annot_aims_scanb,</span>
<span id="cb19-1163"><a href="#cb19-1163" aria-hidden="true" tabindex="-1"></a>    <span class="at">do.mapping =</span> <span class="cn">TRUE</span></span>
<span id="cb19-1164"><a href="#cb19-1164" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-1165"><a href="#cb19-1165" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1166"><a href="#cb19-1166" aria-hidden="true" tabindex="-1"></a>scanb<span class="sc">$</span>aims_pam50 <span class="ot">&lt;-</span> sbt_scanb_AIMS<span class="sc">$</span>subtype <span class="sc">%&gt;%</span> as.vector</span>
<span id="cb19-1167"><a href="#cb19-1167" aria-hidden="true" tabindex="-1"></a>scanb_coldata <span class="ot">&lt;-</span> <span class="fu">colData</span>(scanb) <span class="sc">%&gt;%</span></span>
<span id="cb19-1168"><a href="#cb19-1168" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span></span>
<span id="cb19-1169"><a href="#cb19-1169" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb19-1170"><a href="#cb19-1170" aria-hidden="true" tabindex="-1"></a>        <span class="at">aims_pam50 =</span> <span class="fu">factor</span>(<span class="fu">tolower</span>(aims_pam50), <span class="at">levels =</span> <span class="fu">unique</span>(pam50)),</span>
<span id="cb19-1171"><a href="#cb19-1171" aria-hidden="true" tabindex="-1"></a>        <span class="at">pam50 =</span> <span class="fu">factor</span>(pam50, <span class="at">levels =</span> <span class="fu">unique</span>(pam50))</span>
<span id="cb19-1172"><a href="#cb19-1172" aria-hidden="true" tabindex="-1"></a>    ) </span>
<span id="cb19-1173"><a href="#cb19-1173" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-1174"><a href="#cb19-1174" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1175"><a href="#cb19-1175" aria-hidden="true" tabindex="-1"></a>The confusion matrix below show the results of the AIMS PAM50 as the</span>
<span id="cb19-1176"><a href="#cb19-1176" aria-hidden="true" tabindex="-1"></a>predicted and the reference is the PAM50 subtypes obtained from the</span>
<span id="cb19-1177"><a href="#cb19-1177" aria-hidden="true" tabindex="-1"></a>SCANB team.</span>
<span id="cb19-1178"><a href="#cb19-1178" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1181"><a href="#cb19-1181" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-1182"><a href="#cb19-1182" aria-hidden="true" tabindex="-1"></a>caret<span class="sc">::</span><span class="fu">confusionMatrix</span>(scanb_coldata<span class="sc">$</span>aims_pam50, scanb_coldata<span class="sc">$</span>pam50)</span>
<span id="cb19-1183"><a href="#cb19-1183" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-1184"><a href="#cb19-1184" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1185"><a href="#cb19-1185" aria-hidden="true" tabindex="-1"></a>We see that there is a big misclassification from luminal A samples to </span>
<span id="cb19-1186"><a href="#cb19-1186" aria-hidden="true" tabindex="-1"></a>normal like samples and several luminal B are misclassified as luminal A.</span>
<span id="cb19-1187"><a href="#cb19-1187" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1188"><a href="#cb19-1188" aria-hidden="true" tabindex="-1"></a>The plot below shows the coloring by the reference PAM50 and the predicted</span>
<span id="cb19-1189"><a href="#cb19-1189" aria-hidden="true" tabindex="-1"></a>by AIMS.</span>
<span id="cb19-1190"><a href="#cb19-1190" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1191"><a href="#cb19-1191" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=7, fig.height=10}</span></span>
<span id="cb19-1192"><a href="#cb19-1192" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-scanb-aims</span></span>
<span id="cb19-1193"><a href="#cb19-1193" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: SCANB embedding stratified by PAM50 algorithms by both the ones</span></span>
<span id="cb19-1194"><a href="#cb19-1194" aria-hidden="true" tabindex="-1"></a><span class="co">#|     provided by the SCANB (SSP) and the AIMS algorithm</span></span>
<span id="cb19-1195"><a href="#cb19-1195" aria-hidden="true" tabindex="-1"></a>df_pca <span class="sc">%&gt;%</span></span>
<span id="cb19-1196"><a href="#cb19-1196" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"scanb"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-1197"><a href="#cb19-1197" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">inner_join</span>(</span>
<span id="cb19-1198"><a href="#cb19-1198" aria-hidden="true" tabindex="-1"></a>        ., </span>
<span id="cb19-1199"><a href="#cb19-1199" aria-hidden="true" tabindex="-1"></a>        scanb_coldata <span class="sc">%&gt;%</span></span>
<span id="cb19-1200"><a href="#cb19-1200" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">select</span>(aims_pam50, sample_name),</span>
<span id="cb19-1201"><a href="#cb19-1201" aria-hidden="true" tabindex="-1"></a>        <span class="at">by =</span> <span class="st">"sample_name"</span></span>
<span id="cb19-1202"><a href="#cb19-1202" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb19-1203"><a href="#cb19-1203" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb19-1204"><a href="#cb19-1204" aria-hidden="true" tabindex="-1"></a>        <span class="at">AIMS =</span> aims_pam50,</span>
<span id="cb19-1205"><a href="#cb19-1205" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">SSP SCANB</span><span class="st">`</span> <span class="ot">=</span> pam50</span>
<span id="cb19-1206"><a href="#cb19-1206" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb19-1207"><a href="#cb19-1207" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(</span>
<span id="cb19-1208"><a href="#cb19-1208" aria-hidden="true" tabindex="-1"></a>        <span class="at">cols =</span> dplyr<span class="sc">::</span><span class="fu">all_of</span>(<span class="fu">c</span>(<span class="st">"AIMS"</span>, <span class="st">"SSP SCANB"</span>)),</span>
<span id="cb19-1209"><a href="#cb19-1209" aria-hidden="true" tabindex="-1"></a>        <span class="at">names_to =</span> <span class="st">"pam50_algo"</span>,</span>
<span id="cb19-1210"><a href="#cb19-1210" aria-hidden="true" tabindex="-1"></a>        <span class="at">values_to =</span> <span class="st">"classes"</span></span>
<span id="cb19-1211"><a href="#cb19-1211" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb19-1212"><a href="#cb19-1212" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> PC3, <span class="at">y =</span> PC4, <span class="at">color =</span> classes)) <span class="sc">+</span></span>
<span id="cb19-1213"><a href="#cb19-1213" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb19-1214"><a href="#cb19-1214" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_grid</span>(classes<span class="sc">~</span>pam50_algo) <span class="sc">+</span></span>
<span id="cb19-1215"><a href="#cb19-1215" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb19-1216"><a href="#cb19-1216" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">"PAM50"</span>,</span>
<span id="cb19-1217"><a href="#cb19-1217" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-1218"><a href="#cb19-1218" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">15</span>) <span class="sc">+</span></span>
<span id="cb19-1219"><a href="#cb19-1219" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">get_colors_pam50</span>(</span>
<span id="cb19-1220"><a href="#cb19-1220" aria-hidden="true" tabindex="-1"></a>        df_pca <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(pam50 <span class="sc">%in%</span> scanb<span class="sc">$</span>pam50)</span>
<span id="cb19-1221"><a href="#cb19-1221" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">+</span> </span>
<span id="cb19-1222"><a href="#cb19-1222" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>) <span class="sc">+</span> </span>
<span id="cb19-1223"><a href="#cb19-1223" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span> </span>
<span id="cb19-1224"><a href="#cb19-1224" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb19-1225"><a href="#cb19-1225" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-1226"><a href="#cb19-1226" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1227"><a href="#cb19-1227" aria-hidden="true" tabindex="-1"></a>On the left there is the coloring for the AIMS algorithm and on the right</span>
<span id="cb19-1228"><a href="#cb19-1228" aria-hidden="true" tabindex="-1"></a>from the reference. We see the missclassified samples are not randomly</span>
<span id="cb19-1229"><a href="#cb19-1229" aria-hidden="true" tabindex="-1"></a>misclassified, they are on the boundaries. </span>
<span id="cb19-1230"><a href="#cb19-1230" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1231"><a href="#cb19-1231" aria-hidden="true" tabindex="-1"></a>The plot below shows the maximum probability score for each sample</span>
<span id="cb19-1232"><a href="#cb19-1232" aria-hidden="true" tabindex="-1"></a>individually.</span>
<span id="cb19-1233"><a href="#cb19-1233" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1234"><a href="#cb19-1234" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.height=10, fig.width=4}</span></span>
<span id="cb19-1235"><a href="#cb19-1235" aria-hidden="true" tabindex="-1"></a><span class="fu">data.frame</span>(</span>
<span id="cb19-1236"><a href="#cb19-1236" aria-hidden="true" tabindex="-1"></a>    <span class="at">max_prob =</span> <span class="fu">apply</span>(sbt_scanb_AIMS<span class="sc">$</span>subtype.proba, <span class="dv">1</span>, max),</span>
<span id="cb19-1237"><a href="#cb19-1237" aria-hidden="true" tabindex="-1"></a>    <span class="at">aims_pam50 =</span> sbt_scanb_AIMS<span class="sc">$</span>subtype <span class="sc">%&gt;%</span> as.vector</span>
<span id="cb19-1238"><a href="#cb19-1238" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb19-1239"><a href="#cb19-1239" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> max_prob)) <span class="sc">+</span></span>
<span id="cb19-1240"><a href="#cb19-1240" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_histogram</span>(<span class="at">bins =</span> <span class="dv">10</span>) <span class="sc">+</span> </span>
<span id="cb19-1241"><a href="#cb19-1241" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>aims_pam50, <span class="at">ncol =</span> <span class="dv">1</span>) <span class="sc">+</span> </span>
<span id="cb19-1242"><a href="#cb19-1242" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>)</span>
<span id="cb19-1243"><a href="#cb19-1243" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-1244"><a href="#cb19-1244" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1245"><a href="#cb19-1245" aria-hidden="true" tabindex="-1"></a>The majority of samples have probabilities of either 1 or </span>
<span id="cb19-1246"><a href="#cb19-1246" aria-hidden="true" tabindex="-1"></a><span class="ss">0. </span>Not so often in between. </span>
<span id="cb19-1247"><a href="#cb19-1247" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1248"><a href="#cb19-1248" aria-hidden="true" tabindex="-1"></a>And the figure below shows the confusion matrix for the ABIM cohort.</span>
<span id="cb19-1249"><a href="#cb19-1249" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1252"><a href="#cb19-1252" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-1253"><a href="#cb19-1253" aria-hidden="true" tabindex="-1"></a>abim_coldata <span class="ot">&lt;-</span> <span class="fu">colData</span>(abim_100) <span class="sc">%&gt;%</span></span>
<span id="cb19-1254"><a href="#cb19-1254" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span></span>
<span id="cb19-1255"><a href="#cb19-1255" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb19-1256"><a href="#cb19-1256" aria-hidden="true" tabindex="-1"></a>        <span class="at">aims_pam50 =</span> <span class="fu">factor</span>(<span class="fu">tolower</span>(aims_pam50), <span class="at">levels =</span> <span class="fu">unique</span>(pam50)),</span>
<span id="cb19-1257"><a href="#cb19-1257" aria-hidden="true" tabindex="-1"></a>        <span class="at">pam50 =</span> <span class="fu">factor</span>(pam50, <span class="at">levels =</span> <span class="fu">unique</span>(pam50))</span>
<span id="cb19-1258"><a href="#cb19-1258" aria-hidden="true" tabindex="-1"></a>    ) </span>
<span id="cb19-1259"><a href="#cb19-1259" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb19-1260"><a href="#cb19-1260" aria-hidden="true" tabindex="-1"></a>caret<span class="sc">::</span><span class="fu">confusionMatrix</span>(abim_coldata<span class="sc">$</span>aims_pam50, abim_coldata<span class="sc">$</span>pam50)</span>
<span id="cb19-1261"><a href="#cb19-1261" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-1262"><a href="#cb19-1262" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1263"><a href="#cb19-1263" aria-hidden="true" tabindex="-1"></a>We see here that in this case that there are several luminal B samples </span>
<span id="cb19-1264"><a href="#cb19-1264" aria-hidden="true" tabindex="-1"></a>missclassified as luminal A. </span>
<span id="cb19-1265"><a href="#cb19-1265" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1266"><a href="#cb19-1266" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=8, fig.height=10}</span></span>
<span id="cb19-1267"><a href="#cb19-1267" aria-hidden="true" tabindex="-1"></a>abim_100_df_pca <span class="sc">%&gt;%</span></span>
<span id="cb19-1268"><a href="#cb19-1268" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"abim_100"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-1269"><a href="#cb19-1269" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">inner_join</span>(</span>
<span id="cb19-1270"><a href="#cb19-1270" aria-hidden="true" tabindex="-1"></a>        ., </span>
<span id="cb19-1271"><a href="#cb19-1271" aria-hidden="true" tabindex="-1"></a>        abim_coldata <span class="sc">%&gt;%</span></span>
<span id="cb19-1272"><a href="#cb19-1272" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">select</span>(aims_pam50, sample_name),</span>
<span id="cb19-1273"><a href="#cb19-1273" aria-hidden="true" tabindex="-1"></a>        <span class="at">by =</span> <span class="st">"sample_name"</span></span>
<span id="cb19-1274"><a href="#cb19-1274" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb19-1275"><a href="#cb19-1275" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">pam50_scanb =</span> pam50) <span class="sc">%&gt;%</span></span>
<span id="cb19-1276"><a href="#cb19-1276" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(</span>
<span id="cb19-1277"><a href="#cb19-1277" aria-hidden="true" tabindex="-1"></a>        <span class="at">cols =</span> <span class="fu">c</span>(pam50_scanb, aims_pam50),</span>
<span id="cb19-1278"><a href="#cb19-1278" aria-hidden="true" tabindex="-1"></a>        <span class="at">names_to =</span> <span class="st">"pam50_algo"</span>,</span>
<span id="cb19-1279"><a href="#cb19-1279" aria-hidden="true" tabindex="-1"></a>        <span class="at">values_to =</span> <span class="st">"classes"</span></span>
<span id="cb19-1280"><a href="#cb19-1280" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb19-1281"><a href="#cb19-1281" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> PC3, <span class="at">y =</span> PC4, <span class="at">color =</span> classes)) <span class="sc">+</span></span>
<span id="cb19-1282"><a href="#cb19-1282" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb19-1283"><a href="#cb19-1283" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_grid</span>(classes<span class="sc">~</span>pam50_algo) <span class="sc">+</span></span>
<span id="cb19-1284"><a href="#cb19-1284" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">15</span>) <span class="sc">+</span></span>
<span id="cb19-1285"><a href="#cb19-1285" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">get_colors_pam50</span>(</span>
<span id="cb19-1286"><a href="#cb19-1286" aria-hidden="true" tabindex="-1"></a>        df_pca <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(pam50 <span class="sc">%in%</span> scanb<span class="sc">$</span>pam50)</span>
<span id="cb19-1287"><a href="#cb19-1287" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">+</span> </span>
<span id="cb19-1288"><a href="#cb19-1288" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span> </span>
<span id="cb19-1289"><a href="#cb19-1289" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb19-1290"><a href="#cb19-1290" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-1291"><a href="#cb19-1291" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1292"><a href="#cb19-1292" aria-hidden="true" tabindex="-1"></a>The misclassified samples from AIMS they really seem to be on the </span>
<span id="cb19-1293"><a href="#cb19-1293" aria-hidden="true" tabindex="-1"></a>top region together with the luminal A. Some of the normal</span>
<span id="cb19-1294"><a href="#cb19-1294" aria-hidden="true" tabindex="-1"></a>samples are even in the luminal A region. </span>
<span id="cb19-1295"><a href="#cb19-1295" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1296"><a href="#cb19-1296" aria-hidden="true" tabindex="-1"></a>Now we check the PDXs. </span>
<span id="cb19-1297"><a href="#cb19-1297" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1300"><a href="#cb19-1300" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-1301"><a href="#cb19-1301" aria-hidden="true" tabindex="-1"></a>coldata_pdx <span class="ot">&lt;-</span> <span class="fu">colData</span>(pdx_tpm) <span class="sc">%&gt;%</span> data.frame</span>
<span id="cb19-1302"><a href="#cb19-1302" aria-hidden="true" tabindex="-1"></a>coldata_pdx<span class="sc">$</span>aims_pam50 <span class="ot">&lt;-</span> sbt_pdx_AIMS<span class="sc">$</span>subtype <span class="sc">%&gt;%</span> as.vector</span>
<span id="cb19-1303"><a href="#cb19-1303" aria-hidden="true" tabindex="-1"></a>coldata_pdx<span class="sc">$</span>max_prob <span class="ot">&lt;-</span> <span class="fu">apply</span>(sbt_pdx_AIMS<span class="sc">$</span>subtype.proba, <span class="dv">1</span>, max)</span>
<span id="cb19-1304"><a href="#cb19-1304" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1305"><a href="#cb19-1305" aria-hidden="true" tabindex="-1"></a>coldata_pdx <span class="sc">%&gt;%</span></span>
<span id="cb19-1306"><a href="#cb19-1306" aria-hidden="true" tabindex="-1"></a>    janitor<span class="sc">::</span><span class="fu">tabyl</span>(treatment, aims_pam50, pdx)</span>
<span id="cb19-1307"><a href="#cb19-1307" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-1308"><a href="#cb19-1308" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1309"><a href="#cb19-1309" aria-hidden="true" tabindex="-1"></a>These molecular subtyping do not make much sense. We know that by </span>
<span id="cb19-1310"><a href="#cb19-1310" aria-hidden="true" tabindex="-1"></a>giving P4 estrogen signaling actually goes up, so a expected subtype</span>
<span id="cb19-1311"><a href="#cb19-1311" aria-hidden="true" tabindex="-1"></a>change is from normal to either luminal A and B. Moreover, these PDXs</span>
<span id="cb19-1312"><a href="#cb19-1312" aria-hidden="true" tabindex="-1"></a>are in an active state of proliferation due to the MIND model, so</span>
<span id="cb19-1313"><a href="#cb19-1313" aria-hidden="true" tabindex="-1"></a>being in the normal like region is not a good indication. Also this </span>
<span id="cb19-1314"><a href="#cb19-1314" aria-hidden="true" tabindex="-1"></a>results show that almost all PDXs would be normal like which does not </span>
<span id="cb19-1315"><a href="#cb19-1315" aria-hidden="true" tabindex="-1"></a>go in hand with the results we get from the lab.</span>
<span id="cb19-1316"><a href="#cb19-1316" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1317"><a href="#cb19-1317" aria-hidden="true" tabindex="-1"></a>And the probabilities are available below.</span>
<span id="cb19-1318"><a href="#cb19-1318" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1321"><a href="#cb19-1321" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-1322"><a href="#cb19-1322" aria-hidden="true" tabindex="-1"></a>coldata_pdx <span class="sc">%&gt;%</span></span>
<span id="cb19-1323"><a href="#cb19-1323" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> pdx, <span class="at">y =</span> max_prob)) <span class="sc">+</span> </span>
<span id="cb19-1324"><a href="#cb19-1324" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_boxplot</span>(<span class="at">outlier.shape =</span> <span class="cn">NA</span>) <span class="sc">+</span></span>
<span id="cb19-1325"><a href="#cb19-1325" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_jitter</span>() <span class="sc">+</span> </span>
<span id="cb19-1326"><a href="#cb19-1326" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>()</span>
<span id="cb19-1327"><a href="#cb19-1327" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-1328"><a href="#cb19-1328" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1329"><a href="#cb19-1329" aria-hidden="true" tabindex="-1"></a>And another validation is using the normal cohort. Below is the number of</span>
<span id="cb19-1330"><a href="#cb19-1330" aria-hidden="true" tabindex="-1"></a>all predicted molecular subtypes.</span>
<span id="cb19-1331"><a href="#cb19-1331" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1334"><a href="#cb19-1334" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-1335"><a href="#cb19-1335" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(normal_swiss<span class="sc">$</span>aims_pam50)</span>
<span id="cb19-1336"><a href="#cb19-1336" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-1337"><a href="#cb19-1337" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1338"><a href="#cb19-1338" aria-hidden="true" tabindex="-1"></a>All samples are normal-like as expected, since they are normal. So it could</span>
<span id="cb19-1339"><a href="#cb19-1339" aria-hidden="true" tabindex="-1"></a>be that the AIMS algorithm is not suited for PDXs but for clinical samples</span>
<span id="cb19-1340"><a href="#cb19-1340" aria-hidden="true" tabindex="-1"></a>it works quite well.</span>
<span id="cb19-1341"><a href="#cb19-1341" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1342"><a href="#cb19-1342" aria-hidden="true" tabindex="-1"></a><span class="fu">## nanostring signatures</span></span>
<span id="cb19-1343"><a href="#cb19-1343" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1344"><a href="#cb19-1344" aria-hidden="true" tabindex="-1"></a>In this section we try the nanostring panel signatures provided by the</span>
<span id="cb19-1345"><a href="#cb19-1345" aria-hidden="true" tabindex="-1"></a>Breast Cancer 360 test. We will compare some of the signatures</span>
<span id="cb19-1346"><a href="#cb19-1346" aria-hidden="true" tabindex="-1"></a>by using GSVA on the original datasets and by calculating the sum of the</span>
<span id="cb19-1347"><a href="#cb19-1347" aria-hidden="true" tabindex="-1"></a>gene expression levels of the genes when regressing out the 1st, 2nd and</span>
<span id="cb19-1348"><a href="#cb19-1348" aria-hidden="true" tabindex="-1"></a>5th components.</span>
<span id="cb19-1349"><a href="#cb19-1349" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1350"><a href="#cb19-1350" aria-hidden="true" tabindex="-1"></a>The table below shows the number of genes available for each signature</span>
<span id="cb19-1351"><a href="#cb19-1351" aria-hidden="true" tabindex="-1"></a>and each dataset. In general it seems that all signatures have over</span>
<span id="cb19-1352"><a href="#cb19-1352" aria-hidden="true" tabindex="-1"></a>70% of the genes available, a good indication.</span>
<span id="cb19-1353"><a href="#cb19-1353" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1356"><a href="#cb19-1356" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-1357"><a href="#cb19-1357" aria-hidden="true" tabindex="-1"></a>nanostring_panel <span class="ot">&lt;-</span> readxl<span class="sc">::</span><span class="fu">read_excel</span>(</span>
<span id="cb19-1358"><a href="#cb19-1358" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../../data/nanostring_panel.xlsx"</span>,</span>
<span id="cb19-1359"><a href="#cb19-1359" aria-hidden="true" tabindex="-1"></a>    <span class="at">sheet =</span> <span class="st">"Annotations"</span>, </span>
<span id="cb19-1360"><a href="#cb19-1360" aria-hidden="true" tabindex="-1"></a>    <span class="at">skip =</span> <span class="dv">1</span></span>
<span id="cb19-1361"><a href="#cb19-1361" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-1362"><a href="#cb19-1362" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>dplyr<span class="sc">::</span><span class="fu">all_of</span>(<span class="fu">c</span>(<span class="st">"Cell Type"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb19-1363"><a href="#cb19-1363" aria-hidden="true" tabindex="-1"></a>    janitor<span class="sc">::</span><span class="fu">clean_names</span>()</span>
<span id="cb19-1364"><a href="#cb19-1364" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1365"><a href="#cb19-1365" aria-hidden="true" tabindex="-1"></a>gene_sets_nanostring <span class="ot">&lt;-</span> <span class="fu">sapply</span>(</span>
<span id="cb19-1366"><a href="#cb19-1366" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(nanostring_panel)[<span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(nanostring_panel)], </span>
<span id="cb19-1367"><a href="#cb19-1367" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(signature){</span>
<span id="cb19-1368"><a href="#cb19-1368" aria-hidden="true" tabindex="-1"></a>        nanostring_panel <span class="sc">%&gt;%</span></span>
<span id="cb19-1369"><a href="#cb19-1369" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!!</span><span class="fu">sym</span>(signature) <span class="sc">==</span> <span class="st">"+"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-1370"><a href="#cb19-1370" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">pull</span>(gene)</span>
<span id="cb19-1371"><a href="#cb19-1371" aria-hidden="true" tabindex="-1"></a>    }, </span>
<span id="cb19-1372"><a href="#cb19-1372" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>, </span>
<span id="cb19-1373"><a href="#cb19-1373" aria-hidden="true" tabindex="-1"></a>    <span class="at">simplify =</span> <span class="cn">FALSE</span></span>
<span id="cb19-1374"><a href="#cb19-1374" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-1375"><a href="#cb19-1375" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1376"><a href="#cb19-1376" aria-hidden="true" tabindex="-1"></a>genes_each_dataset <span class="ot">&lt;-</span> <span class="fu">lapply</span>(datasets, rownames)</span>
<span id="cb19-1377"><a href="#cb19-1377" aria-hidden="true" tabindex="-1"></a>genes_intersection <span class="ot">&lt;-</span> <span class="fu">sapply</span>(</span>
<span id="cb19-1378"><a href="#cb19-1378" aria-hidden="true" tabindex="-1"></a>    <span class="fu">names</span>(gene_sets_nanostring),</span>
<span id="cb19-1379"><a href="#cb19-1379" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(set_name, gene_sets, genes_each_dataset){</span>
<span id="cb19-1380"><a href="#cb19-1380" aria-hidden="true" tabindex="-1"></a>        <span class="fu">lapply</span>(</span>
<span id="cb19-1381"><a href="#cb19-1381" aria-hidden="true" tabindex="-1"></a>            genes_each_dataset,</span>
<span id="cb19-1382"><a href="#cb19-1382" aria-hidden="true" tabindex="-1"></a>            intersect,</span>
<span id="cb19-1383"><a href="#cb19-1383" aria-hidden="true" tabindex="-1"></a>            <span class="at">y =</span> gene_sets[[set_name]]</span>
<span id="cb19-1384"><a href="#cb19-1384" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb19-1385"><a href="#cb19-1385" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb19-1386"><a href="#cb19-1386" aria-hidden="true" tabindex="-1"></a>    <span class="at">gene_sets =</span> gene_sets_nanostring,</span>
<span id="cb19-1387"><a href="#cb19-1387" aria-hidden="true" tabindex="-1"></a>    <span class="at">genes_each_dataset =</span> genes_each_dataset,</span>
<span id="cb19-1388"><a href="#cb19-1388" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>,</span>
<span id="cb19-1389"><a href="#cb19-1389" aria-hidden="true" tabindex="-1"></a>    <span class="at">simplify =</span> <span class="cn">FALSE</span></span>
<span id="cb19-1390"><a href="#cb19-1390" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-1391"><a href="#cb19-1391" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1392"><a href="#cb19-1392" aria-hidden="true" tabindex="-1"></a><span class="fu">sapply</span>(genes_intersection, <span class="cf">function</span>(x) <span class="fu">sapply</span>(x, length)) <span class="sc">%&gt;%</span> t <span class="sc">%&gt;%</span></span>
<span id="cb19-1393"><a href="#cb19-1393" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span></span>
<span id="cb19-1394"><a href="#cb19-1394" aria-hidden="true" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"pathway"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-1395"><a href="#cb19-1395" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">inner_join</span>(</span>
<span id="cb19-1396"><a href="#cb19-1396" aria-hidden="true" tabindex="-1"></a>        .,</span>
<span id="cb19-1397"><a href="#cb19-1397" aria-hidden="true" tabindex="-1"></a>        <span class="fu">lapply</span>(gene_sets_nanostring, data.frame) <span class="sc">%&gt;%</span> </span>
<span id="cb19-1398"><a href="#cb19-1398" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(<span class="at">.id =</span> <span class="st">"gs_name"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-1399"><a href="#cb19-1399" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">group_by</span>(gs_name) <span class="sc">%&gt;%</span></span>
<span id="cb19-1400"><a href="#cb19-1400" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">n =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb19-1401"><a href="#cb19-1401" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">pathway =</span> gs_name),</span>
<span id="cb19-1402"><a href="#cb19-1402" aria-hidden="true" tabindex="-1"></a>        <span class="at">by =</span> <span class="st">"pathway"</span></span>
<span id="cb19-1403"><a href="#cb19-1403" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span> </span>
<span id="cb19-1404"><a href="#cb19-1404" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb19-1405"><a href="#cb19-1405" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">average_percentage =</span> <span class="fu">format</span>(<span class="fu">mean</span>(</span>
<span id="cb19-1406"><a href="#cb19-1406" aria-hidden="true" tabindex="-1"></a>            <span class="fu">c</span>(tcga<span class="sc">/</span>n, scanb<span class="sc">/</span>n, metabric<span class="sc">/</span>n, poetic<span class="sc">/</span>n)</span>
<span id="cb19-1407"><a href="#cb19-1407" aria-hidden="true" tabindex="-1"></a>        ),</span>
<span id="cb19-1408"><a href="#cb19-1408" aria-hidden="true" tabindex="-1"></a>        <span class="at">digits =</span> <span class="dv">2</span></span>
<span id="cb19-1409"><a href="#cb19-1409" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">%&gt;%</span> </span>
<span id="cb19-1410"><a href="#cb19-1410" aria-hidden="true" tabindex="-1"></a>    kableExtra<span class="sc">::</span><span class="fu">kbl</span>() <span class="sc">%&gt;%</span></span>
<span id="cb19-1411"><a href="#cb19-1411" aria-hidden="true" tabindex="-1"></a>    kableExtra<span class="sc">::</span><span class="fu">kable_classic</span>(<span class="at">full_width =</span> <span class="cn">FALSE</span>)</span>
<span id="cb19-1412"><a href="#cb19-1412" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-1413"><a href="#cb19-1413" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1414"><a href="#cb19-1414" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb19-1415"><a href="#cb19-1415" aria-hidden="true" tabindex="-1"></a>gsva_scores_nanostring <span class="ot">&lt;-</span> <span class="fu">mapply</span>(</span>
<span id="cb19-1416"><a href="#cb19-1416" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(sum_exp, which_assay, gene_sets){</span>
<span id="cb19-1417"><a href="#cb19-1417" aria-hidden="true" tabindex="-1"></a>        GSVA<span class="sc">::</span><span class="fu">gsva</span>(</span>
<span id="cb19-1418"><a href="#cb19-1418" aria-hidden="true" tabindex="-1"></a>            <span class="at">expr =</span> <span class="fu">as.matrix</span>(<span class="fu">assay</span>(sum_exp, which_assay)),</span>
<span id="cb19-1419"><a href="#cb19-1419" aria-hidden="true" tabindex="-1"></a>            <span class="at">gset.idx.list =</span> gene_sets, </span>
<span id="cb19-1420"><a href="#cb19-1420" aria-hidden="true" tabindex="-1"></a>            <span class="at">parallel.sz =</span> <span class="dv">10</span>,</span>
<span id="cb19-1421"><a href="#cb19-1421" aria-hidden="true" tabindex="-1"></a>            <span class="at">verbose =</span> <span class="cn">FALSE</span></span>
<span id="cb19-1422"><a href="#cb19-1422" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb19-1423"><a href="#cb19-1423" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb19-1424"><a href="#cb19-1424" aria-hidden="true" tabindex="-1"></a>    <span class="at">sum_exp =</span> datasets,</span>
<span id="cb19-1425"><a href="#cb19-1425" aria-hidden="true" tabindex="-1"></a>    <span class="at">which_assay =</span> <span class="fu">c</span>(which_exp, <span class="st">"poetic"</span> <span class="ot">=</span> <span class="st">"normalized_intensity"</span>),</span>
<span id="cb19-1426"><a href="#cb19-1426" aria-hidden="true" tabindex="-1"></a>    <span class="at">MoreArgs =</span> <span class="fu">list</span>(<span class="at">gene_sets =</span> gene_sets_nanostring),</span>
<span id="cb19-1427"><a href="#cb19-1427" aria-hidden="true" tabindex="-1"></a>    <span class="at">SIMPLIFY =</span> <span class="cn">FALSE</span>,</span>
<span id="cb19-1428"><a href="#cb19-1428" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span></span>
<span id="cb19-1429"><a href="#cb19-1429" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-1430"><a href="#cb19-1430" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1431"><a href="#cb19-1431" aria-hidden="true" tabindex="-1"></a>datasets_nanostring <span class="ot">&lt;-</span> <span class="fu">mapply</span>(</span>
<span id="cb19-1432"><a href="#cb19-1432" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(gsva_score, dataset){</span>
<span id="cb19-1433"><a href="#cb19-1433" aria-hidden="true" tabindex="-1"></a>        <span class="fu">colData</span>(dataset)[, <span class="fu">rownames</span>(gsva_score)] <span class="ot">&lt;-</span> <span class="fu">t</span>(gsva_score)</span>
<span id="cb19-1434"><a href="#cb19-1434" aria-hidden="true" tabindex="-1"></a>        dataset</span>
<span id="cb19-1435"><a href="#cb19-1435" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb19-1436"><a href="#cb19-1436" aria-hidden="true" tabindex="-1"></a>    gsva_scores_nanostring,</span>
<span id="cb19-1437"><a href="#cb19-1437" aria-hidden="true" tabindex="-1"></a>    datasets,</span>
<span id="cb19-1438"><a href="#cb19-1438" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>,</span>
<span id="cb19-1439"><a href="#cb19-1439" aria-hidden="true" tabindex="-1"></a>    <span class="at">SIMPLIFY =</span> <span class="cn">FALSE</span></span>
<span id="cb19-1440"><a href="#cb19-1440" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-1441"><a href="#cb19-1441" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1442"><a href="#cb19-1442" aria-hidden="true" tabindex="-1"></a>scores_nanostring <span class="ot">&lt;-</span> <span class="fu">sapply</span>(</span>
<span id="cb19-1443"><a href="#cb19-1443" aria-hidden="true" tabindex="-1"></a>    gene_sets_nanostring,</span>
<span id="cb19-1444"><a href="#cb19-1444" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(gene_set, df_pcs_regressed){</span>
<span id="cb19-1445"><a href="#cb19-1445" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb19-1446"><a href="#cb19-1446" aria-hidden="true" tabindex="-1"></a>        gene_set <span class="ot">&lt;-</span> <span class="fu">intersect</span>(</span>
<span id="cb19-1447"><a href="#cb19-1447" aria-hidden="true" tabindex="-1"></a>            gene_set,</span>
<span id="cb19-1448"><a href="#cb19-1448" aria-hidden="true" tabindex="-1"></a>            <span class="fu">rownames</span>(df_pcs_regressed)</span>
<span id="cb19-1449"><a href="#cb19-1449" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb19-1450"><a href="#cb19-1450" aria-hidden="true" tabindex="-1"></a>        <span class="fu">colSums</span>(df_pcs_regressed[gene_set, ])</span>
<span id="cb19-1451"><a href="#cb19-1451" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb19-1452"><a href="#cb19-1452" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb19-1453"><a href="#cb19-1453" aria-hidden="true" tabindex="-1"></a>    <span class="at">df_pcs_regressed =</span> df_pcs_regressed</span>
<span id="cb19-1454"><a href="#cb19-1454" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-1455"><a href="#cb19-1455" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span></span>
<span id="cb19-1456"><a href="#cb19-1456" aria-hidden="true" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"sample_name"</span>)</span>
<span id="cb19-1457"><a href="#cb19-1457" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1458"><a href="#cb19-1458" aria-hidden="true" tabindex="-1"></a>scores_nanostring <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">inner_join</span>(</span>
<span id="cb19-1459"><a href="#cb19-1459" aria-hidden="true" tabindex="-1"></a>    scores_nanostring,</span>
<span id="cb19-1460"><a href="#cb19-1460" aria-hidden="true" tabindex="-1"></a>    df_pca,</span>
<span id="cb19-1461"><a href="#cb19-1461" aria-hidden="true" tabindex="-1"></a>    <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"sample_name"</span>)</span>
<span id="cb19-1462"><a href="#cb19-1462" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-1463"><a href="#cb19-1463" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1464"><a href="#cb19-1464" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(scores_nanostring) <span class="ot">&lt;-</span> scores_nanostring<span class="sc">$</span>sample_name</span>
<span id="cb19-1465"><a href="#cb19-1465" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1466"><a href="#cb19-1466" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb19-1467"><a href="#cb19-1467" aria-hidden="true" tabindex="-1"></a>    scores_nanostring,</span>
<span id="cb19-1468"><a href="#cb19-1468" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../../results/rds_files/trying/scores_nanostring_only.rds"</span></span>
<span id="cb19-1469"><a href="#cb19-1469" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-1470"><a href="#cb19-1470" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1471"><a href="#cb19-1471" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb19-1472"><a href="#cb19-1472" aria-hidden="true" tabindex="-1"></a>    gsva_scores_nanostring, </span>
<span id="cb19-1473"><a href="#cb19-1473" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../../results/rds_files/trying/gsva_scores_nanostring.rds"</span></span>
<span id="cb19-1474"><a href="#cb19-1474" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-1475"><a href="#cb19-1475" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1476"><a href="#cb19-1476" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb19-1477"><a href="#cb19-1477" aria-hidden="true" tabindex="-1"></a>    datasets_nanostring,</span>
<span id="cb19-1478"><a href="#cb19-1478" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../../results/rds_files/trying/datasets_with_scores_nanostring.rds"</span></span>
<span id="cb19-1479"><a href="#cb19-1479" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-1480"><a href="#cb19-1480" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-1481"><a href="#cb19-1481" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1482"><a href="#cb19-1482" aria-hidden="true" tabindex="-1"></a>We now compare the scores obtained by GSVA on the datasets and </span>
<span id="cb19-1483"><a href="#cb19-1483" aria-hidden="true" tabindex="-1"></a>see how they relate with nanostring scores obtained with the</span>
<span id="cb19-1484"><a href="#cb19-1484" aria-hidden="true" tabindex="-1"></a>regressed data. </span>
<span id="cb19-1485"><a href="#cb19-1485" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1486"><a href="#cb19-1486" aria-hidden="true" tabindex="-1"></a>First we compare GSVA SET ER/PR with ER signaling obtained by the</span>
<span id="cb19-1487"><a href="#cb19-1487" aria-hidden="true" tabindex="-1"></a>scoring procedure when regressing the data.</span>
<span id="cb19-1488"><a href="#cb19-1488" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1491"><a href="#cb19-1491" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-1492"><a href="#cb19-1492" aria-hidden="true" tabindex="-1"></a>scores_nanostring <span class="sc">%&gt;%</span></span>
<span id="cb19-1493"><a href="#cb19-1493" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(</span>
<span id="cb19-1494"><a href="#cb19-1494" aria-hidden="true" tabindex="-1"></a>        <span class="fu">aes</span>(</span>
<span id="cb19-1495"><a href="#cb19-1495" aria-hidden="true" tabindex="-1"></a>            <span class="at">x =</span> er_signaling, </span>
<span id="cb19-1496"><a href="#cb19-1496" aria-hidden="true" tabindex="-1"></a>            <span class="at">y =</span> SET_ERPR,</span>
<span id="cb19-1497"><a href="#cb19-1497" aria-hidden="true" tabindex="-1"></a>            <span class="at">color =</span> pam50</span>
<span id="cb19-1498"><a href="#cb19-1498" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb19-1499"><a href="#cb19-1499" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb19-1500"><a href="#cb19-1500" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb19-1501"><a href="#cb19-1501" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb19-1502"><a href="#cb19-1502" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>cohort)</span>
<span id="cb19-1503"><a href="#cb19-1503" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-1504"><a href="#cb19-1504" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1505"><a href="#cb19-1505" aria-hidden="true" tabindex="-1"></a>Next we compare EMT, G2M with proliferation and PI3K signaling. </span>
<span id="cb19-1506"><a href="#cb19-1506" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1509"><a href="#cb19-1509" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-1510"><a href="#cb19-1510" aria-hidden="true" tabindex="-1"></a>scores_nanostring <span class="sc">%&gt;%</span></span>
<span id="cb19-1511"><a href="#cb19-1511" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(</span>
<span id="cb19-1512"><a href="#cb19-1512" aria-hidden="true" tabindex="-1"></a>        <span class="fu">aes</span>(</span>
<span id="cb19-1513"><a href="#cb19-1513" aria-hidden="true" tabindex="-1"></a>            <span class="at">x =</span> emt, </span>
<span id="cb19-1514"><a href="#cb19-1514" aria-hidden="true" tabindex="-1"></a>            <span class="at">y =</span> HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION,</span>
<span id="cb19-1515"><a href="#cb19-1515" aria-hidden="true" tabindex="-1"></a>            <span class="at">color =</span> pam50</span>
<span id="cb19-1516"><a href="#cb19-1516" aria-hidden="true" tabindex="-1"></a>        )) <span class="sc">+</span></span>
<span id="cb19-1517"><a href="#cb19-1517" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb19-1518"><a href="#cb19-1518" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb19-1519"><a href="#cb19-1519" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>cohort)</span>
<span id="cb19-1520"><a href="#cb19-1520" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-1521"><a href="#cb19-1521" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1524"><a href="#cb19-1524" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-1525"><a href="#cb19-1525" aria-hidden="true" tabindex="-1"></a>scores_nanostring <span class="sc">%&gt;%</span></span>
<span id="cb19-1526"><a href="#cb19-1526" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> proliferation, <span class="at">y =</span> HALLMARK_G2M_CHECKPOINT,</span>
<span id="cb19-1527"><a href="#cb19-1527" aria-hidden="true" tabindex="-1"></a>                        <span class="at">color =</span> pam50)) <span class="sc">+</span></span>
<span id="cb19-1528"><a href="#cb19-1528" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb19-1529"><a href="#cb19-1529" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb19-1530"><a href="#cb19-1530" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>cohort)</span>
<span id="cb19-1531"><a href="#cb19-1531" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-1532"><a href="#cb19-1532" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1535"><a href="#cb19-1535" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-1536"><a href="#cb19-1536" aria-hidden="true" tabindex="-1"></a>scores_nanostring <span class="sc">%&gt;%</span></span>
<span id="cb19-1537"><a href="#cb19-1537" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> pi3k, <span class="at">y =</span> HALLMARK_PI3K_AKT_MTOR_SIGNALING,</span>
<span id="cb19-1538"><a href="#cb19-1538" aria-hidden="true" tabindex="-1"></a>                        <span class="at">color =</span> pam50)) <span class="sc">+</span></span>
<span id="cb19-1539"><a href="#cb19-1539" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb19-1540"><a href="#cb19-1540" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb19-1541"><a href="#cb19-1541" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>cohort)</span>
<span id="cb19-1542"><a href="#cb19-1542" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-1543"><a href="#cb19-1543" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1546"><a href="#cb19-1546" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-1547"><a href="#cb19-1547" aria-hidden="true" tabindex="-1"></a>scores_nanostring <span class="sc">%&gt;%</span></span>
<span id="cb19-1548"><a href="#cb19-1548" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> pam50, <span class="at">y =</span> proliferation, <span class="at">color =</span> pam50)) <span class="sc">+</span></span>
<span id="cb19-1549"><a href="#cb19-1549" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb19-1550"><a href="#cb19-1550" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb19-1551"><a href="#cb19-1551" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>cohort)</span>
<span id="cb19-1552"><a href="#cb19-1552" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-1553"><a href="#cb19-1553" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1554"><a href="#cb19-1554" aria-hidden="true" tabindex="-1"></a>In general there is a correlation but the scores are not so well correlated</span>
<span id="cb19-1555"><a href="#cb19-1555" aria-hidden="true" tabindex="-1"></a>and they are noisy. One that works fairly well is the ER signaling score,</span>
<span id="cb19-1556"><a href="#cb19-1556" aria-hidden="true" tabindex="-1"></a>but the others not so well. The proliferation score barely shows any</span>
<span id="cb19-1557"><a href="#cb19-1557" aria-hidden="true" tabindex="-1"></a>difference between luminal A and B patients in any of the cohorts. </span>
<span id="cb19-1558"><a href="#cb19-1558" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1559"><a href="#cb19-1559" aria-hidden="true" tabindex="-1"></a>Perhaps if we calculate the scores using the hallmarks pathways the scores are</span>
<span id="cb19-1560"><a href="#cb19-1560" aria-hidden="true" tabindex="-1"></a>better correlated, since they are the same gene sets.</span>
<span id="cb19-1561"><a href="#cb19-1561" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1562"><a href="#cb19-1562" aria-hidden="true" tabindex="-1"></a>Below we plot the correlation between the proliferation pathway using GSVA and</span>
<span id="cb19-1563"><a href="#cb19-1563" aria-hidden="true" tabindex="-1"></a>the regressed scoring strategy for the TCGA cohort.</span>
<span id="cb19-1564"><a href="#cb19-1564" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1567"><a href="#cb19-1567" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-1568"><a href="#cb19-1568" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(</span>
<span id="cb19-1569"><a href="#cb19-1569" aria-hidden="true" tabindex="-1"></a>    datasets_nanostring<span class="sc">$</span>tcga<span class="sc">$</span>proliferation,</span>
<span id="cb19-1570"><a href="#cb19-1570" aria-hidden="true" tabindex="-1"></a>    scores_nanostring[<span class="fu">colnames</span>(datasets_nanostring<span class="sc">$</span>tcga), <span class="st">"proliferation"</span>]</span>
<span id="cb19-1571"><a href="#cb19-1571" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-1572"><a href="#cb19-1572" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-1573"><a href="#cb19-1573" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1574"><a href="#cb19-1574" aria-hidden="true" tabindex="-1"></a>It is perfectly correlated.</span>
<span id="cb19-1575"><a href="#cb19-1575" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1576"><a href="#cb19-1576" aria-hidden="true" tabindex="-1"></a>And for ER signaling:</span>
<span id="cb19-1577"><a href="#cb19-1577" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1580"><a href="#cb19-1580" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-1581"><a href="#cb19-1581" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(</span>
<span id="cb19-1582"><a href="#cb19-1582" aria-hidden="true" tabindex="-1"></a>    datasets_nanostring<span class="sc">$</span>tcga<span class="sc">$</span>er_signaling,</span>
<span id="cb19-1583"><a href="#cb19-1583" aria-hidden="true" tabindex="-1"></a>    scores_nanostring[<span class="fu">colnames</span>(datasets_nanostring<span class="sc">$</span>tcga), <span class="st">"er_signaling"</span>]</span>
<span id="cb19-1584"><a href="#cb19-1584" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-1585"><a href="#cb19-1585" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-1586"><a href="#cb19-1586" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1587"><a href="#cb19-1587" aria-hidden="true" tabindex="-1"></a>Also it is very much correlated. So the problem in the end is comparing</span>
<span id="cb19-1588"><a href="#cb19-1588" aria-hidden="true" tabindex="-1"></a>the hallmarks signatures with the nanostring. But interestingly </span>
<span id="cb19-1589"><a href="#cb19-1589" aria-hidden="true" tabindex="-1"></a>the nanostring proliferation signature was not good to pick up the differences</span>
<span id="cb19-1590"><a href="#cb19-1590" aria-hidden="true" tabindex="-1"></a>between luminal A and B.</span>
<span id="cb19-1591"><a href="#cb19-1591" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1594"><a href="#cb19-1594" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-1595"><a href="#cb19-1595" aria-hidden="true" tabindex="-1"></a>datasets_nanostring<span class="sc">$</span>tcga <span class="sc">%&gt;%</span></span>
<span id="cb19-1596"><a href="#cb19-1596" aria-hidden="true" tabindex="-1"></a>    colData <span class="sc">%&gt;%</span></span>
<span id="cb19-1597"><a href="#cb19-1597" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span></span>
<span id="cb19-1598"><a href="#cb19-1598" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(</span>
<span id="cb19-1599"><a href="#cb19-1599" aria-hidden="true" tabindex="-1"></a>        <span class="at">cols =</span> <span class="fu">c</span>(<span class="st">"proliferation"</span>, <span class="st">"HALLMARK_G2M_CHECKPOINT"</span>),</span>
<span id="cb19-1600"><a href="#cb19-1600" aria-hidden="true" tabindex="-1"></a>        <span class="at">names_to =</span> <span class="st">"pathway"</span>,</span>
<span id="cb19-1601"><a href="#cb19-1601" aria-hidden="true" tabindex="-1"></a>        <span class="at">values_to =</span> <span class="st">"score"</span></span>
<span id="cb19-1602"><a href="#cb19-1602" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb19-1603"><a href="#cb19-1603" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> pam50, <span class="at">y =</span> score, <span class="at">color =</span> pam50)) <span class="sc">+</span> </span>
<span id="cb19-1604"><a href="#cb19-1604" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_boxplot</span>() <span class="sc">+</span> </span>
<span id="cb19-1605"><a href="#cb19-1605" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>pathway, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span> </span>
<span id="cb19-1606"><a href="#cb19-1606" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>()</span>
<span id="cb19-1607"><a href="#cb19-1607" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-1608"><a href="#cb19-1608" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1609"><a href="#cb19-1609" aria-hidden="true" tabindex="-1"></a><span class="fu">## Risk stratification and the molecular landscape</span></span>
<span id="cb19-1610"><a href="#cb19-1610" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1611"><a href="#cb19-1611" aria-hidden="true" tabindex="-1"></a>One common problem in the clinics is what to do with patients that are </span>
<span id="cb19-1612"><a href="#cb19-1612" aria-hidden="true" tabindex="-1"></a>in the so called intermediate risk group. Here we use the SSP </span>
<span id="cb19-1613"><a href="#cb19-1613" aria-hidden="true" tabindex="-1"></a>classification from the SCANB team to check what is the position of these</span>
<span id="cb19-1614"><a href="#cb19-1614" aria-hidden="true" tabindex="-1"></a>patients in the molecular landscape.</span>
<span id="cb19-1615"><a href="#cb19-1615" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1618"><a href="#cb19-1618" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-1619"><a href="#cb19-1619" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-scanb-risk-embedding</span></span>
<span id="cb19-1620"><a href="#cb19-1620" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Embedding of the SCANB samples colored by their predicted</span></span>
<span id="cb19-1621"><a href="#cb19-1621" aria-hidden="true" tabindex="-1"></a><span class="co">#|     ROR risk category.</span></span>
<span id="cb19-1622"><a href="#cb19-1622" aria-hidden="true" tabindex="-1"></a>df_pca <span class="sc">%&gt;%</span></span>
<span id="cb19-1623"><a href="#cb19-1623" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"scanb"</span> <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(SSP.ROR.risk.cat)) <span class="sc">%&gt;%</span></span>
<span id="cb19-1624"><a href="#cb19-1624" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">slice_sample</span>(<span class="at">n =</span> <span class="fu">nrow</span>(.)) <span class="sc">%&gt;%</span></span>
<span id="cb19-1625"><a href="#cb19-1625" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">sample_name =</span> <span class="fu">factor</span>(sample_name)) <span class="sc">%&gt;%</span></span>
<span id="cb19-1626"><a href="#cb19-1626" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> PC3, <span class="at">y =</span> PC4, <span class="at">color =</span> SSP.ROR.risk.cat)) <span class="sc">+</span></span>
<span id="cb19-1627"><a href="#cb19-1627" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span> </span>
<span id="cb19-1628"><a href="#cb19-1628" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_color_viridis_d</span>(<span class="at">option =</span> <span class="st">"H"</span>) <span class="sc">+</span> </span>
<span id="cb19-1629"><a href="#cb19-1629" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb19-1630"><a href="#cb19-1630" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span> </span>
<span id="cb19-1631"><a href="#cb19-1631" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb19-1632"><a href="#cb19-1632" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-1633"><a href="#cb19-1633" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1634"><a href="#cb19-1634" aria-hidden="true" tabindex="-1"></a>@fig-scanb-risk-embedding shows that the embedding detects the positions</span>
<span id="cb19-1635"><a href="#cb19-1635" aria-hidden="true" tabindex="-1"></a>where the intermediate risk patients are, namely they are in the</span>
<span id="cb19-1636"><a href="#cb19-1636" aria-hidden="true" tabindex="-1"></a>intersection of the luminal A and B subtypes.</span>
<span id="cb19-1637"><a href="#cb19-1637" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1638"><a href="#cb19-1638" aria-hidden="true" tabindex="-1"></a>And @fig-intermediate-pca-position below shows that there is a shift in</span>
<span id="cb19-1639"><a href="#cb19-1639" aria-hidden="true" tabindex="-1"></a>PC4 for the intermediate to the high risk group, which makes sense as this is</span>
<span id="cb19-1640"><a href="#cb19-1640" aria-hidden="true" tabindex="-1"></a>exactly where the distinction between luminal A and B are.</span>
<span id="cb19-1641"><a href="#cb19-1641" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1642"><a href="#cb19-1642" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.height=4, fig.width=8}</span></span>
<span id="cb19-1643"><a href="#cb19-1643" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-intermediate-pca-position</span></span>
<span id="cb19-1644"><a href="#cb19-1644" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Distribution of the principal components for the different</span></span>
<span id="cb19-1645"><a href="#cb19-1645" aria-hidden="true" tabindex="-1"></a><span class="co">#|     risk groups defined by the SSP ROR scores.</span></span>
<span id="cb19-1646"><a href="#cb19-1646" aria-hidden="true" tabindex="-1"></a>df_pca <span class="sc">%&gt;%</span></span>
<span id="cb19-1647"><a href="#cb19-1647" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"scanb"</span> <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(SSP.ROR.risk.cat)) <span class="sc">%&gt;%</span></span>
<span id="cb19-1648"><a href="#cb19-1648" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(</span>
<span id="cb19-1649"><a href="#cb19-1649" aria-hidden="true" tabindex="-1"></a>        <span class="at">cols =</span> <span class="fu">c</span>(PC3, PC4), </span>
<span id="cb19-1650"><a href="#cb19-1650" aria-hidden="true" tabindex="-1"></a>        <span class="at">names_to =</span> <span class="st">"pcs"</span>, </span>
<span id="cb19-1651"><a href="#cb19-1651" aria-hidden="true" tabindex="-1"></a>        <span class="at">values_to =</span> <span class="st">"pc_scores"</span></span>
<span id="cb19-1652"><a href="#cb19-1652" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb19-1653"><a href="#cb19-1653" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">y =</span> SSP.ROR.risk.cat, <span class="at">x =</span> pc_scores, <span class="at">color =</span> SSP.ROR.risk.cat)) <span class="sc">+</span></span>
<span id="cb19-1654"><a href="#cb19-1654" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_jitter</span>(<span class="at">alpha =</span> <span class="fl">0.4</span>, <span class="at">size =</span> <span class="fl">0.1</span>) <span class="sc">+</span> </span>
<span id="cb19-1655"><a href="#cb19-1655" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_violin</span>(<span class="at">alpha =</span> <span class="fl">0.1</span>, <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">linewidth =</span> <span class="fl">0.6</span>) <span class="sc">+</span> </span>
<span id="cb19-1656"><a href="#cb19-1656" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>pcs) <span class="sc">+</span></span>
<span id="cb19-1657"><a href="#cb19-1657" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_color_viridis_d</span>(<span class="at">option =</span> <span class="st">"H"</span>) <span class="sc">+</span></span>
<span id="cb19-1658"><a href="#cb19-1658" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"PC"</span>) <span class="sc">+</span> </span>
<span id="cb19-1659"><a href="#cb19-1659" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>) <span class="sc">+</span> </span>
<span id="cb19-1660"><a href="#cb19-1660" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>) <span class="sc">+</span> </span>
<span id="cb19-1661"><a href="#cb19-1661" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span> </span>
<span id="cb19-1662"><a href="#cb19-1662" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb19-1663"><a href="#cb19-1663" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-1664"><a href="#cb19-1664" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1665"><a href="#cb19-1665" aria-hidden="true" tabindex="-1"></a><span class="fu">## Late distant recurrence: trying to find biomarkers</span></span>
<span id="cb19-1666"><a href="#cb19-1666" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1667"><a href="#cb19-1667" aria-hidden="true" tabindex="-1"></a>One of the key problems in breast cancer research is finding patients that</span>
<span id="cb19-1668"><a href="#cb19-1668" aria-hidden="true" tabindex="-1"></a>will develop late recurrence. Here we use METABRIC as it has a long follow-up</span>
<span id="cb19-1669"><a href="#cb19-1669" aria-hidden="true" tabindex="-1"></a>history. We start by plotting the embedding of all METABRIC patients that</span>
<span id="cb19-1670"><a href="#cb19-1670" aria-hidden="true" tabindex="-1"></a>had a followup longer than 10 years and received only ET. We then try to </span>
<span id="cb19-1671"><a href="#cb19-1671" aria-hidden="true" tabindex="-1"></a>predict which patients recurr or not based on the molecular data by</span>
<span id="cb19-1672"><a href="#cb19-1672" aria-hidden="true" tabindex="-1"></a>using lasso regression. For this we use the package <span class="in">`glmnet`</span> that </span>
<span id="cb19-1673"><a href="#cb19-1673" aria-hidden="true" tabindex="-1"></a>provides such functionality. </span>
<span id="cb19-1674"><a href="#cb19-1674" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1677"><a href="#cb19-1677" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-1678"><a href="#cb19-1678" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">get_base_plot</span>(</span>
<span id="cb19-1679"><a href="#cb19-1679" aria-hidden="true" tabindex="-1"></a>    df_pca <span class="sc">%&gt;%</span> </span>
<span id="cb19-1680"><a href="#cb19-1680" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">!=</span> <span class="st">"poetic"</span>) </span>
<span id="cb19-1681"><a href="#cb19-1681" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span> </span>
<span id="cb19-1682"><a href="#cb19-1682" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb19-1683"><a href="#cb19-1683" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1684"><a href="#cb19-1684" aria-hidden="true" tabindex="-1"></a>et_metabric_late_data <span class="ot">&lt;-</span> df_pca <span class="sc">%&gt;%</span></span>
<span id="cb19-1685"><a href="#cb19-1685" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb19-1686"><a href="#cb19-1686" aria-hidden="true" tabindex="-1"></a>        cohort <span class="sc">==</span> <span class="st">"metabric"</span> <span class="sc">&amp;</span></span>
<span id="cb19-1687"><a href="#cb19-1687" aria-hidden="true" tabindex="-1"></a>            rfs_months <span class="sc">&gt;=</span> <span class="dv">10</span> <span class="sc">*</span> <span class="dv">12</span> <span class="sc">&amp;</span> </span>
<span id="cb19-1688"><a href="#cb19-1688" aria-hidden="true" tabindex="-1"></a>            <span class="fu">is.numeric</span>(rfs_status) <span class="sc">&amp;</span> </span>
<span id="cb19-1689"><a href="#cb19-1689" aria-hidden="true" tabindex="-1"></a>            HORMONE_THERAPY <span class="sc">==</span> <span class="st">"YES"</span> <span class="sc">&amp;</span> </span>
<span id="cb19-1690"><a href="#cb19-1690" aria-hidden="true" tabindex="-1"></a>            er_status <span class="sc">==</span> <span class="st">"pos"</span></span>
<span id="cb19-1691"><a href="#cb19-1691" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb19-1692"><a href="#cb19-1692" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">is_there_recurrence =</span> <span class="fu">ifelse</span>(</span>
<span id="cb19-1693"><a href="#cb19-1693" aria-hidden="true" tabindex="-1"></a>        rfs_status <span class="sc">==</span> <span class="dv">2</span>,</span>
<span id="cb19-1694"><a href="#cb19-1694" aria-hidden="true" tabindex="-1"></a>        <span class="st">"yes"</span>,</span>
<span id="cb19-1695"><a href="#cb19-1695" aria-hidden="true" tabindex="-1"></a>        <span class="st">"no"</span></span>
<span id="cb19-1696"><a href="#cb19-1696" aria-hidden="true" tabindex="-1"></a>    ))</span>
<span id="cb19-1697"><a href="#cb19-1697" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1698"><a href="#cb19-1698" aria-hidden="true" tabindex="-1"></a>ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(</span>
<span id="cb19-1699"><a href="#cb19-1699" aria-hidden="true" tabindex="-1"></a>    et_metabric_late_data,</span>
<span id="cb19-1700"><a href="#cb19-1700" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(</span>
<span id="cb19-1701"><a href="#cb19-1701" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> PC3, </span>
<span id="cb19-1702"><a href="#cb19-1702" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> PC4, </span>
<span id="cb19-1703"><a href="#cb19-1703" aria-hidden="true" tabindex="-1"></a>        <span class="at">shape =</span> is_there_recurrence, </span>
<span id="cb19-1704"><a href="#cb19-1704" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> is_there_recurrence</span>
<span id="cb19-1705"><a href="#cb19-1705" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb19-1706"><a href="#cb19-1706" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span> </span>
<span id="cb19-1707"><a href="#cb19-1707" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>(</span>
<span id="cb19-1708"><a href="#cb19-1708" aria-hidden="true" tabindex="-1"></a>        <span class="at">size =</span> <span class="dv">4</span>,</span>
<span id="cb19-1709"><a href="#cb19-1709" aria-hidden="true" tabindex="-1"></a>        <span class="at">alpha =</span> <span class="fl">0.5</span></span>
<span id="cb19-1710"><a href="#cb19-1710" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-1711"><a href="#cb19-1711" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">15</span>) <span class="sc">+</span></span>
<span id="cb19-1712"><a href="#cb19-1712" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb19-1713"><a href="#cb19-1713" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">"METABRIC"</span>,</span>
<span id="cb19-1714"><a href="#cb19-1714" aria-hidden="true" tabindex="-1"></a>        <span class="at">subtitle =</span> <span class="st">"ER+ BC, ET only, over 10 years of RFS follow-up"</span>,</span>
<span id="cb19-1715"><a href="#cb19-1715" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">"Recurrence?"</span>,</span>
<span id="cb19-1716"><a href="#cb19-1716" aria-hidden="true" tabindex="-1"></a>        <span class="at">shape =</span> <span class="st">"Recurrence?"</span></span>
<span id="cb19-1717"><a href="#cb19-1717" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb19-1718"><a href="#cb19-1718" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">coord_cartesian</span>(</span>
<span id="cb19-1719"><a href="#cb19-1719" aria-hidden="true" tabindex="-1"></a>        <span class="at">xlim =</span> <span class="fu">c</span>(<span class="fu">min</span>(df_pca<span class="sc">$</span>PC3), <span class="fu">max</span>(df_pca<span class="sc">$</span>PC3)),</span>
<span id="cb19-1720"><a href="#cb19-1720" aria-hidden="true" tabindex="-1"></a>        <span class="at">ylim =</span> <span class="fu">c</span>(<span class="fu">min</span>(df_pca<span class="sc">$</span>PC4), <span class="fu">max</span>(df_pca<span class="sc">$</span>PC4))</span>
<span id="cb19-1721"><a href="#cb19-1721" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-1722"><a href="#cb19-1722" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span></span>
<span id="cb19-1723"><a href="#cb19-1723" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb19-1724"><a href="#cb19-1724" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-1725"><a href="#cb19-1725" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1726"><a href="#cb19-1726" aria-hidden="true" tabindex="-1"></a>In general there is no distinction in terms of the molecular</span>
<span id="cb19-1727"><a href="#cb19-1727" aria-hidden="true" tabindex="-1"></a>landscape among the patients that recurred, both in terms of</span>
<span id="cb19-1728"><a href="#cb19-1728" aria-hidden="true" tabindex="-1"></a>molecular subtype and position in the molecular landscape. </span>
<span id="cb19-1729"><a href="#cb19-1729" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1730"><a href="#cb19-1730" aria-hidden="true" tabindex="-1"></a>We now compare their molecular scores. </span>
<span id="cb19-1731"><a href="#cb19-1731" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1732"><a href="#cb19-1732" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=20, fig.height=40}</span></span>
<span id="cb19-1733"><a href="#cb19-1733" aria-hidden="true" tabindex="-1"></a>pathways_to_compare <span class="ot">&lt;-</span> <span class="fu">colnames</span>(df_pca)[<span class="fu">grepl</span>(<span class="st">"^HALLMARK"</span>, <span class="fu">colnames</span>(df_pca))]</span>
<span id="cb19-1734"><a href="#cb19-1734" aria-hidden="true" tabindex="-1"></a>et_metabric_late_data <span class="sc">%&gt;%</span></span>
<span id="cb19-1735"><a href="#cb19-1735" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(</span>
<span id="cb19-1736"><a href="#cb19-1736" aria-hidden="true" tabindex="-1"></a>        <span class="at">cols =</span> dplyr<span class="sc">::</span><span class="fu">all_of</span>(pathways_to_compare),</span>
<span id="cb19-1737"><a href="#cb19-1737" aria-hidden="true" tabindex="-1"></a>        <span class="at">names_to =</span> <span class="st">"pathway"</span>,</span>
<span id="cb19-1738"><a href="#cb19-1738" aria-hidden="true" tabindex="-1"></a>        <span class="at">values_to =</span> <span class="st">"score"</span></span>
<span id="cb19-1739"><a href="#cb19-1739" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb19-1740"><a href="#cb19-1740" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> is_there_recurrence, <span class="at">y =</span> score)) <span class="sc">+</span> </span>
<span id="cb19-1741"><a href="#cb19-1741" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_jitter</span>(<span class="at">size =</span> <span class="fl">0.1</span>) <span class="sc">+</span> </span>
<span id="cb19-1742"><a href="#cb19-1742" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_boxplot</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="at">outlier.shape =</span> <span class="cn">NA</span>) <span class="sc">+</span> </span>
<span id="cb19-1743"><a href="#cb19-1743" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(</span>
<span id="cb19-1744"><a href="#cb19-1744" aria-hidden="true" tabindex="-1"></a>        <span class="sc">~</span>pathway, </span>
<span id="cb19-1745"><a href="#cb19-1745" aria-hidden="true" tabindex="-1"></a>        <span class="at">ncol =</span> <span class="dv">5</span>,</span>
<span id="cb19-1746"><a href="#cb19-1746" aria-hidden="true" tabindex="-1"></a>        <span class="at">labeller =</span> <span class="fu">as_labeller</span>(</span>
<span id="cb19-1747"><a href="#cb19-1747" aria-hidden="true" tabindex="-1"></a>            stringr<span class="sc">::</span><span class="fu">str_replace_all</span>(pathways_to_compare, <span class="st">"^HALLMARK_"</span>, <span class="st">""</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-1748"><a href="#cb19-1748" aria-hidden="true" tabindex="-1"></a>                <span class="st">`</span><span class="at">names&lt;-</span><span class="st">`</span>(pathways_to_compare)</span>
<span id="cb19-1749"><a href="#cb19-1749" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb19-1750"><a href="#cb19-1750" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-1751"><a href="#cb19-1751" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb19-1752"><a href="#cb19-1752" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"Recurrence?"</span>,</span>
<span id="cb19-1753"><a href="#cb19-1753" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">"METABRIC"</span>,</span>
<span id="cb19-1754"><a href="#cb19-1754" aria-hidden="true" tabindex="-1"></a>        <span class="at">subtitle =</span> <span class="st">"ER+ BC, ET only, over 10 years of RFS follow-up"</span>    </span>
<span id="cb19-1755"><a href="#cb19-1755" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-1756"><a href="#cb19-1756" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">15</span>)</span>
<span id="cb19-1757"><a href="#cb19-1757" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-1758"><a href="#cb19-1758" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb19-1759"><a href="#cb19-1759" aria-hidden="true" tabindex="-1"></a>Even here we don't see much difference. Unless there is some kind of </span>
<span id="cb19-1760"><a href="#cb19-1760" aria-hidden="true" tabindex="-1"></a>linear combination of pathways, I don't think it would be possible</span>
<span id="cb19-1761"><a href="#cb19-1761" aria-hidden="true" tabindex="-1"></a>to classify these two groups by using only transcriptomic data</span>
<span id="cb19-1762"><a href="#cb19-1762" aria-hidden="true" tabindex="-1"></a>and these pathways. In any case we proceed now with the </span>
<span id="cb19-1763"><a href="#cb19-1763" aria-hidden="true" tabindex="-1"></a>lasso regression using the 50 pathways. </span>
<span id="cb19-1764"><a href="#cb19-1764" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1767"><a href="#cb19-1767" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-1768"><a href="#cb19-1768" aria-hidden="true" tabindex="-1"></a>fit_recurrence <span class="ot">&lt;-</span> glmnet<span class="sc">::</span><span class="fu">cv.glmnet</span>(</span>
<span id="cb19-1769"><a href="#cb19-1769" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="fu">as.matrix</span>(et_metabric_late_data[, pathways_to_compare]),</span>
<span id="cb19-1770"><a href="#cb19-1770" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> et_metabric_late_data<span class="sc">$</span>is_there_recurrence,</span>
<span id="cb19-1771"><a href="#cb19-1771" aria-hidden="true" tabindex="-1"></a>    <span class="at">family =</span> <span class="st">"binomial"</span>,</span>
<span id="cb19-1772"><a href="#cb19-1772" aria-hidden="true" tabindex="-1"></a>    <span class="at">type.measure =</span> <span class="st">"class"</span></span>
<span id="cb19-1773"><a href="#cb19-1773" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-1774"><a href="#cb19-1774" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1775"><a href="#cb19-1775" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(fit_recurrence)</span>
<span id="cb19-1776"><a href="#cb19-1776" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-1777"><a href="#cb19-1777" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1778"><a href="#cb19-1778" aria-hidden="true" tabindex="-1"></a>We see that actually it does not work. And when we predict all the </span>
<span id="cb19-1779"><a href="#cb19-1779" aria-hidden="true" tabindex="-1"></a>samples are predicted as there is no recurrence.</span>
<span id="cb19-1780"><a href="#cb19-1780" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1783"><a href="#cb19-1783" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-1784"><a href="#cb19-1784" aria-hidden="true" tabindex="-1"></a><span class="fu">predict</span>(</span>
<span id="cb19-1785"><a href="#cb19-1785" aria-hidden="true" tabindex="-1"></a>    fit_recurrence, </span>
<span id="cb19-1786"><a href="#cb19-1786" aria-hidden="true" tabindex="-1"></a>    <span class="at">new =</span> et_metabric_late_data[, pathways_to_compare] <span class="sc">%&gt;%</span></span>
<span id="cb19-1787"><a href="#cb19-1787" aria-hidden="true" tabindex="-1"></a>            as.matrix,</span>
<span id="cb19-1788"><a href="#cb19-1788" aria-hidden="true" tabindex="-1"></a>    <span class="at">s =</span> <span class="st">"lambda.1se"</span>,</span>
<span id="cb19-1789"><a href="#cb19-1789" aria-hidden="true" tabindex="-1"></a>    <span class="at">type =</span> <span class="st">"class"</span></span>
<span id="cb19-1790"><a href="#cb19-1790" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> table</span>
<span id="cb19-1791"><a href="#cb19-1791" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-1792"><a href="#cb19-1792" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1793"><a href="#cb19-1793" aria-hidden="true" tabindex="-1"></a>This is probably because of the unbalanced dataset. Next we </span>
<span id="cb19-1794"><a href="#cb19-1794" aria-hidden="true" tabindex="-1"></a>try with random forest as well.</span>
<span id="cb19-1795"><a href="#cb19-1795" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1798"><a href="#cb19-1798" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-1799"><a href="#cb19-1799" aria-hidden="true" tabindex="-1"></a>rf <span class="ot">&lt;-</span> randomForest<span class="sc">::</span><span class="fu">randomForest</span>(</span>
<span id="cb19-1800"><a href="#cb19-1800" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.factor</span>(is_there_recurrence) <span class="sc">~</span> ., </span>
<span id="cb19-1801"><a href="#cb19-1801" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> et_metabric_late_data[, <span class="fu">c</span>(</span>
<span id="cb19-1802"><a href="#cb19-1802" aria-hidden="true" tabindex="-1"></a>        <span class="st">"is_there_recurrence"</span>,</span>
<span id="cb19-1803"><a href="#cb19-1803" aria-hidden="true" tabindex="-1"></a>        pathways_to_compare,</span>
<span id="cb19-1804"><a href="#cb19-1804" aria-hidden="true" tabindex="-1"></a>        <span class="st">"AGE_AT_DIAGNOSIS"</span>,</span>
<span id="cb19-1805"><a href="#cb19-1805" aria-hidden="true" tabindex="-1"></a>        <span class="st">"LYMPH_NODES_EXAMINED_POSITIVE"</span>, </span>
<span id="cb19-1806"><a href="#cb19-1806" aria-hidden="true" tabindex="-1"></a>        <span class="st">"npi"</span></span>
<span id="cb19-1807"><a href="#cb19-1807" aria-hidden="true" tabindex="-1"></a>    )],</span>
<span id="cb19-1808"><a href="#cb19-1808" aria-hidden="true" tabindex="-1"></a>    <span class="at">ntree =</span> <span class="dv">500</span></span>
<span id="cb19-1809"><a href="#cb19-1809" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-1810"><a href="#cb19-1810" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1811"><a href="#cb19-1811" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(rf)</span>
<span id="cb19-1812"><a href="#cb19-1812" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-1813"><a href="#cb19-1813" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1814"><a href="#cb19-1814" aria-hidden="true" tabindex="-1"></a>That is not the case, still we can't predict if a patient will develop</span>
<span id="cb19-1815"><a href="#cb19-1815" aria-hidden="true" tabindex="-1"></a>recurrence or not in this cohort. </span>
<span id="cb19-1816"><a href="#cb19-1816" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1817"><a href="#cb19-1817" aria-hidden="true" tabindex="-1"></a><span class="fu">## Correlation of pathways with proliferation</span></span>
<span id="cb19-1818"><a href="#cb19-1818" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1819"><a href="#cb19-1819" aria-hidden="true" tabindex="-1"></a>One thing that is very common is defining a gene signature and then </span>
<span id="cb19-1820"><a href="#cb19-1820" aria-hidden="true" tabindex="-1"></a>using this gene signature to calculate survival analysis. The thing is, </span>
<span id="cb19-1821"><a href="#cb19-1821" aria-hidden="true" tabindex="-1"></a>proliferation related genes are pretty much present in almost any signature</span>
<span id="cb19-1822"><a href="#cb19-1822" aria-hidden="true" tabindex="-1"></a>and hence all these pathways are correlated to G2M and E2F to some extent </span>
<span id="cb19-1823"><a href="#cb19-1823" aria-hidden="true" tabindex="-1"></a><span class="co">[</span><span class="ot">@Venet2011</span><span class="co">]</span>.</span>
<span id="cb19-1824"><a href="#cb19-1824" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1825"><a href="#cb19-1825" aria-hidden="true" tabindex="-1"></a>Here we calculate the pathway scores by using the regressed data and do</span>
<span id="cb19-1826"><a href="#cb19-1826" aria-hidden="true" tabindex="-1"></a>all pairwise comparison of some of the most important pathways to see</span>
<span id="cb19-1827"><a href="#cb19-1827" aria-hidden="true" tabindex="-1"></a>their correlations.</span>
<span id="cb19-1828"><a href="#cb19-1828" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1829"><a href="#cb19-1829" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=17, fig.height=13, dev='png'}</span></span>
<span id="cb19-1830"><a href="#cb19-1830" aria-hidden="true" tabindex="-1"></a>color_by <span class="ot">&lt;-</span> <span class="st">"pam50"</span></span>
<span id="cb19-1831"><a href="#cb19-1831" aria-hidden="true" tabindex="-1"></a>pathways_correlation <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb19-1832"><a href="#cb19-1832" aria-hidden="true" tabindex="-1"></a>    <span class="st">"G2M"</span>, <span class="st">"ER early"</span>, <span class="st">"ER late"</span>, <span class="st">"PI3K AKT MTOR"</span>,</span>
<span id="cb19-1833"><a href="#cb19-1833" aria-hidden="true" tabindex="-1"></a>    <span class="st">"AR"</span>, <span class="st">"E2F"</span>, <span class="st">"EMT"</span>, </span>
<span id="cb19-1834"><a href="#cb19-1834" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Biocarta HER2"</span>, </span>
<span id="cb19-1835"><a href="#cb19-1835" aria-hidden="true" tabindex="-1"></a>    <span class="st">"KEGG HER2"</span>,</span>
<span id="cb19-1836"><a href="#cb19-1836" aria-hidden="true" tabindex="-1"></a>    <span class="st">"SET ER/PR"</span>, <span class="st">"Random 18"</span>, <span class="st">"Random 200"</span></span>
<span id="cb19-1837"><a href="#cb19-1837" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-1838"><a href="#cb19-1838" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(pathways_correlation) <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb19-1839"><a href="#cb19-1839" aria-hidden="true" tabindex="-1"></a>    <span class="fu">paste0</span>(<span class="st">"HALLMARK_"</span>, <span class="fu">c</span>(</span>
<span id="cb19-1840"><a href="#cb19-1840" aria-hidden="true" tabindex="-1"></a>        <span class="st">"G2M_CHECKPOINT"</span>,</span>
<span id="cb19-1841"><a href="#cb19-1841" aria-hidden="true" tabindex="-1"></a>        <span class="st">"ESTROGEN_RESPONSE_EARLY"</span>,</span>
<span id="cb19-1842"><a href="#cb19-1842" aria-hidden="true" tabindex="-1"></a>        <span class="st">"ESTROGEN_RESPONSE_LATE"</span>,</span>
<span id="cb19-1843"><a href="#cb19-1843" aria-hidden="true" tabindex="-1"></a>        <span class="st">"PI3K_AKT_MTOR_SIGNALING"</span>,</span>
<span id="cb19-1844"><a href="#cb19-1844" aria-hidden="true" tabindex="-1"></a>        <span class="st">"ANDROGEN_RESPONSE"</span>,</span>
<span id="cb19-1845"><a href="#cb19-1845" aria-hidden="true" tabindex="-1"></a>        <span class="st">"E2F_TARGETS"</span>,</span>
<span id="cb19-1846"><a href="#cb19-1846" aria-hidden="true" tabindex="-1"></a>        <span class="st">"EPITHELIAL_MESENCHYMAL_TRANSITION"</span></span>
<span id="cb19-1847"><a href="#cb19-1847" aria-hidden="true" tabindex="-1"></a>    )),</span>
<span id="cb19-1848"><a href="#cb19-1848" aria-hidden="true" tabindex="-1"></a>    <span class="st">"BIOCARTA_HER2_PATHWAY"</span>,</span>
<span id="cb19-1849"><a href="#cb19-1849" aria-hidden="true" tabindex="-1"></a>    <span class="st">"KEGG_ERBB_SIGNALING_PATHWAY"</span>,</span>
<span id="cb19-1850"><a href="#cb19-1850" aria-hidden="true" tabindex="-1"></a>    <span class="st">"SET_ERPR"</span>,</span>
<span id="cb19-1851"><a href="#cb19-1851" aria-hidden="true" tabindex="-1"></a>    <span class="st">"random_18"</span>,</span>
<span id="cb19-1852"><a href="#cb19-1852" aria-hidden="true" tabindex="-1"></a>    <span class="st">"random_200"</span></span>
<span id="cb19-1853"><a href="#cb19-1853" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-1854"><a href="#cb19-1854" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1855"><a href="#cb19-1855" aria-hidden="true" tabindex="-1"></a>scores_her2 <span class="ot">&lt;-</span> <span class="fu">sapply</span>(</span>
<span id="cb19-1856"><a href="#cb19-1856" aria-hidden="true" tabindex="-1"></a>    gene_sets_[<span class="fu">names</span>(pathways_correlation)],</span>
<span id="cb19-1857"><a href="#cb19-1857" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(gene_set, df_pcs_regressed, avg_hkg_pcs_regressed){</span>
<span id="cb19-1858"><a href="#cb19-1858" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb19-1859"><a href="#cb19-1859" aria-hidden="true" tabindex="-1"></a>        gene_set <span class="ot">&lt;-</span> <span class="fu">intersect</span>(</span>
<span id="cb19-1860"><a href="#cb19-1860" aria-hidden="true" tabindex="-1"></a>            gene_set,</span>
<span id="cb19-1861"><a href="#cb19-1861" aria-hidden="true" tabindex="-1"></a>            <span class="fu">rownames</span>(df_pcs_regressed)</span>
<span id="cb19-1862"><a href="#cb19-1862" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb19-1863"><a href="#cb19-1863" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1864"><a href="#cb19-1864" aria-hidden="true" tabindex="-1"></a>        <span class="fu">colSums</span>(df_pcs_regressed[gene_set, ])</span>
<span id="cb19-1865"><a href="#cb19-1865" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb19-1866"><a href="#cb19-1866" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb19-1867"><a href="#cb19-1867" aria-hidden="true" tabindex="-1"></a>    <span class="at">df_pcs_regressed =</span> df_pcs_regressed,</span>
<span id="cb19-1868"><a href="#cb19-1868" aria-hidden="true" tabindex="-1"></a>    <span class="at">avg_hkg_pcs_regressed =</span> avg_hkg_pcs_regressed</span>
<span id="cb19-1869"><a href="#cb19-1869" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-1870"><a href="#cb19-1870" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span></span>
<span id="cb19-1871"><a href="#cb19-1871" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="fu">setNames</span>(<span class="fu">names</span>(pathways_correlation), pathways_correlation)) <span class="sc">%&gt;%</span></span>
<span id="cb19-1872"><a href="#cb19-1872" aria-hidden="true" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"sample_name"</span>)</span>
<span id="cb19-1873"><a href="#cb19-1873" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1874"><a href="#cb19-1874" aria-hidden="true" tabindex="-1"></a>p_pathways <span class="ot">&lt;-</span> GGally<span class="sc">::</span><span class="fu">ggpairs</span>(</span>
<span id="cb19-1875"><a href="#cb19-1875" aria-hidden="true" tabindex="-1"></a>    df_pca <span class="sc">%&gt;%</span></span>
<span id="cb19-1876"><a href="#cb19-1876" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"scanb"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-1877"><a href="#cb19-1877" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">inner_join</span>(., scores_her2, <span class="at">by =</span> <span class="st">"sample_name"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-1878"><a href="#cb19-1878" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">select</span>(dplyr<span class="sc">::</span><span class="fu">all_of</span>(<span class="fu">c</span>(</span>
<span id="cb19-1879"><a href="#cb19-1879" aria-hidden="true" tabindex="-1"></a>            <span class="fu">as.vector</span>(pathways_correlation), </span>
<span id="cb19-1880"><a href="#cb19-1880" aria-hidden="true" tabindex="-1"></a>            color_by</span>
<span id="cb19-1881"><a href="#cb19-1881" aria-hidden="true" tabindex="-1"></a>        ))),</span>
<span id="cb19-1882"><a href="#cb19-1882" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">color =</span> <span class="sc">!!</span><span class="fu">sym</span>(color_by), <span class="at">alpha =</span> <span class="fl">0.5</span>)</span>
<span id="cb19-1883"><a href="#cb19-1883" aria-hidden="true" tabindex="-1"></a>) </span>
<span id="cb19-1884"><a href="#cb19-1884" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1885"><a href="#cb19-1885" aria-hidden="true" tabindex="-1"></a>p_pathways <span class="sc">+</span> </span>
<span id="cb19-1886"><a href="#cb19-1886" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">10</span>) </span>
<span id="cb19-1887"><a href="#cb19-1887" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-1888"><a href="#cb19-1888" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1889"><a href="#cb19-1889" aria-hidden="true" tabindex="-1"></a>First we see that G2M and E2F are encoding basically the same information. </span>
<span id="cb19-1890"><a href="#cb19-1890" aria-hidden="true" tabindex="-1"></a>PI3K AKT MTOR, AR and the Biocarta HER2 and SET ER/PR are all correlated to</span>
<span id="cb19-1891"><a href="#cb19-1891" aria-hidden="true" tabindex="-1"></a>some extent to proliferation. Surprisingly the two HER2 signaling scores </span>
<span id="cb19-1892"><a href="#cb19-1892" aria-hidden="true" tabindex="-1"></a>are not able to distinguish the HER2 subtype.</span>
<span id="cb19-1893"><a href="#cb19-1893" aria-hidden="true" tabindex="-1"></a>The SET ER/PR correlation with G2M is possibly explained by the fact that</span>
<span id="cb19-1894"><a href="#cb19-1894" aria-hidden="true" tabindex="-1"></a>the basal molecular subtype has higher G2M and lower ER signaling compared to</span>
<span id="cb19-1895"><a href="#cb19-1895" aria-hidden="true" tabindex="-1"></a>luminal A and B.</span>
<span id="cb19-1896"><a href="#cb19-1896" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1897"><a href="#cb19-1897" aria-hidden="true" tabindex="-1"></a><span class="fu">## HER2 and their categories</span></span>
<span id="cb19-1898"><a href="#cb19-1898" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1899"><a href="#cb19-1899" aria-hidden="true" tabindex="-1"></a>Research on the benefits of giving trastuzumab together with antibody drug</span>
<span id="cb19-1900"><a href="#cb19-1900" aria-hidden="true" tabindex="-1"></a>conjugates (ADCs) has been</span>
<span id="cb19-1901"><a href="#cb19-1901" aria-hidden="true" tabindex="-1"></a>shown recently. Patients that are either HER2+ or even considered negative</span>
<span id="cb19-1902"><a href="#cb19-1902" aria-hidden="true" tabindex="-1"></a>but have some expression of HER2 (1+ and 2+) seem to benefit from ADC + </span>
<span id="cb19-1903"><a href="#cb19-1903" aria-hidden="true" tabindex="-1"></a>trastuzumab. </span>
<span id="cb19-1904"><a href="#cb19-1904" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1905"><a href="#cb19-1905" aria-hidden="true" tabindex="-1"></a>Here we first see what are the genes that are leading the differences </span>
<span id="cb19-1906"><a href="#cb19-1906" aria-hidden="true" tabindex="-1"></a>of the HER2 molecular subtype. We query 7 genes that are available in the </span>
<span id="cb19-1907"><a href="#cb19-1907" aria-hidden="true" tabindex="-1"></a>list of 8+ regions and are in the amplicon region of ERBB2 when this gene</span>
<span id="cb19-1908"><a href="#cb19-1908" aria-hidden="true" tabindex="-1"></a>is amplified (HER2+). </span>
<span id="cb19-1909"><a href="#cb19-1909" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1910"><a href="#cb19-1910" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=18, fig.height=8}</span></span>
<span id="cb19-1911"><a href="#cb19-1911" aria-hidden="true" tabindex="-1"></a>genes_her2_amp <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb19-1912"><a href="#cb19-1912" aria-hidden="true" tabindex="-1"></a>    <span class="st">"ERBB2"</span>, <span class="st">"CDK12"</span>, <span class="st">"FBXL20"</span>, <span class="st">"GRB7"</span>, </span>
<span id="cb19-1913"><a href="#cb19-1913" aria-hidden="true" tabindex="-1"></a>    <span class="st">"IKZF3"</span>, <span class="st">"MED1"</span>, <span class="st">"MIEN1"</span>, <span class="st">"NEUROD2"</span>,</span>
<span id="cb19-1914"><a href="#cb19-1914" aria-hidden="true" tabindex="-1"></a>    <span class="st">"PGAP3"</span>, <span class="st">"PNMT"</span>, <span class="st">"PPP1R1B"</span>, <span class="st">"STARD3"</span>, </span>
<span id="cb19-1915"><a href="#cb19-1915" aria-hidden="true" tabindex="-1"></a>    <span class="st">"TCAP"</span></span>
<span id="cb19-1916"><a href="#cb19-1916" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-1917"><a href="#cb19-1917" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1918"><a href="#cb19-1918" aria-hidden="true" tabindex="-1"></a><span class="co"># genes available to plot</span></span>
<span id="cb19-1919"><a href="#cb19-1919" aria-hidden="true" tabindex="-1"></a>genes_available_her2_amp <span class="ot">&lt;-</span> <span class="fu">intersect</span>(</span>
<span id="cb19-1920"><a href="#cb19-1920" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames</span>(df_pcs_regressed), </span>
<span id="cb19-1921"><a href="#cb19-1921" aria-hidden="true" tabindex="-1"></a>    genes_her2_amp</span>
<span id="cb19-1922"><a href="#cb19-1922" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-1923"><a href="#cb19-1923" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1924"><a href="#cb19-1924" aria-hidden="true" tabindex="-1"></a>df_pca_her2_amp <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">inner_join</span>(</span>
<span id="cb19-1925"><a href="#cb19-1925" aria-hidden="true" tabindex="-1"></a>    df_pcs_regressed[genes_available_her2_amp, ] <span class="sc">%&gt;%</span> </span>
<span id="cb19-1926"><a href="#cb19-1926" aria-hidden="true" tabindex="-1"></a>        t <span class="sc">%&gt;%</span></span>
<span id="cb19-1927"><a href="#cb19-1927" aria-hidden="true" tabindex="-1"></a>        data.frame <span class="sc">%&gt;%</span></span>
<span id="cb19-1928"><a href="#cb19-1928" aria-hidden="true" tabindex="-1"></a>        tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"sample_name"</span>),</span>
<span id="cb19-1929"><a href="#cb19-1929" aria-hidden="true" tabindex="-1"></a>    df_pca,</span>
<span id="cb19-1930"><a href="#cb19-1930" aria-hidden="true" tabindex="-1"></a>    <span class="at">by =</span> <span class="st">"sample_name"</span></span>
<span id="cb19-1931"><a href="#cb19-1931" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb19-1932"><a href="#cb19-1932" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"tcga"</span>, <span class="st">"scanb"</span>, <span class="st">"metabric"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb19-1933"><a href="#cb19-1933" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(</span>
<span id="cb19-1934"><a href="#cb19-1934" aria-hidden="true" tabindex="-1"></a>        <span class="at">cols =</span> dplyr<span class="sc">::</span><span class="fu">all_of</span>(genes_available_her2_amp),</span>
<span id="cb19-1935"><a href="#cb19-1935" aria-hidden="true" tabindex="-1"></a>        <span class="at">names_to =</span> <span class="st">"her2_amp_genes"</span>,</span>
<span id="cb19-1936"><a href="#cb19-1936" aria-hidden="true" tabindex="-1"></a>        <span class="at">values_to =</span> <span class="st">"exp_levels"</span></span>
<span id="cb19-1937"><a href="#cb19-1937" aria-hidden="true" tabindex="-1"></a>    ) </span>
<span id="cb19-1938"><a href="#cb19-1938" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1939"><a href="#cb19-1939" aria-hidden="true" tabindex="-1"></a>df_pca_her2_amp <span class="sc">%&gt;%</span></span>
<span id="cb19-1940"><a href="#cb19-1940" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(</span>
<span id="cb19-1941"><a href="#cb19-1941" aria-hidden="true" tabindex="-1"></a>        <span class="fu">aes</span>(<span class="at">x =</span> pam50, <span class="at">y =</span> exp_levels, <span class="at">color =</span> pam50)</span>
<span id="cb19-1942"><a href="#cb19-1942" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-1943"><a href="#cb19-1943" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_jitter</span>(<span class="at">size =</span> <span class="fl">0.1</span>) <span class="sc">+</span> </span>
<span id="cb19-1944"><a href="#cb19-1944" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_violin</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span> </span>
<span id="cb19-1945"><a href="#cb19-1945" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(</span>
<span id="cb19-1946"><a href="#cb19-1946" aria-hidden="true" tabindex="-1"></a>        cohort <span class="sc">~</span> her2_amp_genes, </span>
<span id="cb19-1947"><a href="#cb19-1947" aria-hidden="true" tabindex="-1"></a>        <span class="at">ncol =</span> <span class="fu">length</span>(genes_available_her2_amp)</span>
<span id="cb19-1948"><a href="#cb19-1948" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-1949"><a href="#cb19-1949" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_color_manual</span>(</span>
<span id="cb19-1950"><a href="#cb19-1950" aria-hidden="true" tabindex="-1"></a>        <span class="at">values =</span> <span class="fu">get_colors_pam50</span>(df_pca_her2_amp)</span>
<span id="cb19-1951"><a href="#cb19-1951" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-1952"><a href="#cb19-1952" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">coord_flip</span>() <span class="sc">+</span> </span>
<span id="cb19-1953"><a href="#cb19-1953" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb19-1954"><a href="#cb19-1954" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"PAM50"</span>,</span>
<span id="cb19-1955"><a href="#cb19-1955" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"Expression levels on regressed data"</span></span>
<span id="cb19-1956"><a href="#cb19-1956" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-1957"><a href="#cb19-1957" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>) <span class="sc">+</span> </span>
<span id="cb19-1958"><a href="#cb19-1958" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>(</span>
<span id="cb19-1959"><a href="#cb19-1959" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb19-1960"><a href="#cb19-1960" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.spacing =</span> <span class="fu">unit</span>(<span class="dv">1</span>, <span class="st">"lines"</span>)</span>
<span id="cb19-1961"><a href="#cb19-1961" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-1962"><a href="#cb19-1962" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb19-1963"><a href="#cb19-1963" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-1964"><a href="#cb19-1964" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1965"><a href="#cb19-1965" aria-hidden="true" tabindex="-1"></a>First we notice that ERBB2 is actually not the gene that is driving the </span>
<span id="cb19-1966"><a href="#cb19-1966" aria-hidden="true" tabindex="-1"></a>differences, perhaps because when the embedding was learned during the PCA</span>
<span id="cb19-1967"><a href="#cb19-1967" aria-hidden="true" tabindex="-1"></a>this gene was not used to make the differences between the molecular subtype.</span>
<span id="cb19-1968"><a href="#cb19-1968" aria-hidden="true" tabindex="-1"></a>Instead, the genes that are mostly capturing the difference of the </span>
<span id="cb19-1969"><a href="#cb19-1969" aria-hidden="true" tabindex="-1"></a>HER2-enriched PAM50 molecular subtype are FBXL20, GRB7, MED1 and STARD3. </span>
<span id="cb19-1970"><a href="#cb19-1970" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1971"><a href="#cb19-1971" aria-hidden="true" tabindex="-1"></a>And if we check by HER2 status according to the available datasets.</span>
<span id="cb19-1972"><a href="#cb19-1972" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-1973"><a href="#cb19-1973" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=16, fig.height=8}</span></span>
<span id="cb19-1974"><a href="#cb19-1974" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-her2-genes</span></span>
<span id="cb19-1975"><a href="#cb19-1975" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Regressed gene expression levels and comparison between the </span></span>
<span id="cb19-1976"><a href="#cb19-1976" aria-hidden="true" tabindex="-1"></a><span class="co">#|     HER2 groups.</span></span>
<span id="cb19-1977"><a href="#cb19-1977" aria-hidden="true" tabindex="-1"></a>p_scanb_her2_amp <span class="ot">&lt;-</span> df_pca_her2_amp <span class="sc">%&gt;%</span></span>
<span id="cb19-1978"><a href="#cb19-1978" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"scanb"</span> <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(HER2)) <span class="sc">%&gt;%</span></span>
<span id="cb19-1979"><a href="#cb19-1979" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(</span>
<span id="cb19-1980"><a href="#cb19-1980" aria-hidden="true" tabindex="-1"></a>        <span class="fu">aes</span>(<span class="at">x =</span> HER2, <span class="at">y =</span> exp_levels, <span class="at">color =</span> HER2)</span>
<span id="cb19-1981"><a href="#cb19-1981" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-1982"><a href="#cb19-1982" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_jitter</span>(<span class="at">size =</span> <span class="fl">0.1</span>) <span class="sc">+</span> </span>
<span id="cb19-1983"><a href="#cb19-1983" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_violin</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span> </span>
<span id="cb19-1984"><a href="#cb19-1984" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(</span>
<span id="cb19-1985"><a href="#cb19-1985" aria-hidden="true" tabindex="-1"></a>        <span class="sc">~</span> her2_amp_genes, </span>
<span id="cb19-1986"><a href="#cb19-1986" aria-hidden="true" tabindex="-1"></a>        <span class="at">ncol =</span> <span class="fu">length</span>(genes_available_her2_amp)</span>
<span id="cb19-1987"><a href="#cb19-1987" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-1988"><a href="#cb19-1988" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">coord_flip</span>() <span class="sc">+</span> </span>
<span id="cb19-1989"><a href="#cb19-1989" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb19-1990"><a href="#cb19-1990" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"HER2"</span>,</span>
<span id="cb19-1991"><a href="#cb19-1991" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">""</span>,</span>
<span id="cb19-1992"><a href="#cb19-1992" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">"SCANB"</span></span>
<span id="cb19-1993"><a href="#cb19-1993" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-1994"><a href="#cb19-1994" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">18</span>) <span class="sc">+</span> </span>
<span id="cb19-1995"><a href="#cb19-1995" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>(</span>
<span id="cb19-1996"><a href="#cb19-1996" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb19-1997"><a href="#cb19-1997" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.spacing =</span> <span class="fu">unit</span>(<span class="dv">1</span>, <span class="st">"lines"</span>)</span>
<span id="cb19-1998"><a href="#cb19-1998" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-1999"><a href="#cb19-1999" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb19-2000"><a href="#cb19-2000" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2001"><a href="#cb19-2001" aria-hidden="true" tabindex="-1"></a>p_metabric_her2_amp <span class="ot">&lt;-</span> df_pca_her2_amp <span class="sc">%&gt;%</span></span>
<span id="cb19-2002"><a href="#cb19-2002" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"metabric"</span> <span class="sc">&amp;</span> HER2_SNP6 <span class="sc">!=</span> <span class="st">"UNDEF"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-2003"><a href="#cb19-2003" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(</span>
<span id="cb19-2004"><a href="#cb19-2004" aria-hidden="true" tabindex="-1"></a>        <span class="fu">aes</span>(<span class="at">x =</span> HER2_SNP6, <span class="at">y =</span> exp_levels, <span class="at">color =</span> HER2_SNP6)</span>
<span id="cb19-2005"><a href="#cb19-2005" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-2006"><a href="#cb19-2006" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_jitter</span>(<span class="at">size =</span> <span class="fl">0.1</span>) <span class="sc">+</span> </span>
<span id="cb19-2007"><a href="#cb19-2007" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_violin</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span> </span>
<span id="cb19-2008"><a href="#cb19-2008" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(</span>
<span id="cb19-2009"><a href="#cb19-2009" aria-hidden="true" tabindex="-1"></a>        <span class="sc">~</span> her2_amp_genes, </span>
<span id="cb19-2010"><a href="#cb19-2010" aria-hidden="true" tabindex="-1"></a>        <span class="at">ncol =</span> <span class="fu">length</span>(genes_available_her2_amp)</span>
<span id="cb19-2011"><a href="#cb19-2011" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-2012"><a href="#cb19-2012" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">coord_flip</span>() <span class="sc">+</span> </span>
<span id="cb19-2013"><a href="#cb19-2013" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb19-2014"><a href="#cb19-2014" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"HER2 SNP6"</span>,</span>
<span id="cb19-2015"><a href="#cb19-2015" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"Expression levels on regressed data"</span>,</span>
<span id="cb19-2016"><a href="#cb19-2016" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">"METABRIC"</span></span>
<span id="cb19-2017"><a href="#cb19-2017" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-2018"><a href="#cb19-2018" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">18</span>) <span class="sc">+</span> </span>
<span id="cb19-2019"><a href="#cb19-2019" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>(</span>
<span id="cb19-2020"><a href="#cb19-2020" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb19-2021"><a href="#cb19-2021" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.spacing =</span> <span class="fu">unit</span>(<span class="dv">1</span>, <span class="st">"lines"</span>)</span>
<span id="cb19-2022"><a href="#cb19-2022" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-2023"><a href="#cb19-2023" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb19-2024"><a href="#cb19-2024" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2025"><a href="#cb19-2025" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(</span>
<span id="cb19-2026"><a href="#cb19-2026" aria-hidden="true" tabindex="-1"></a>    p_scanb_her2_amp,</span>
<span id="cb19-2027"><a href="#cb19-2027" aria-hidden="true" tabindex="-1"></a>    p_metabric_her2_amp,</span>
<span id="cb19-2028"><a href="#cb19-2028" aria-hidden="true" tabindex="-1"></a>    <span class="at">nrow =</span> <span class="dv">2</span></span>
<span id="cb19-2029"><a href="#cb19-2029" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-2030"><a href="#cb19-2030" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-2031"><a href="#cb19-2031" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2032"><a href="#cb19-2032" aria-hidden="true" tabindex="-1"></a>Similar to what was seen before for ER status you see differences for</span>
<span id="cb19-2033"><a href="#cb19-2033" aria-hidden="true" tabindex="-1"></a>the HER2+ patients. Note that several of the patients that have gain of</span>
<span id="cb19-2034"><a href="#cb19-2034" aria-hidden="true" tabindex="-1"></a>HER2 have similar gene expression levels to those that have loss</span>
<span id="cb19-2035"><a href="#cb19-2035" aria-hidden="true" tabindex="-1"></a>or are neutral. Moreover, the top genes that are differentiating the </span>
<span id="cb19-2036"><a href="#cb19-2036" aria-hidden="true" tabindex="-1"></a>HER2 status are STARD3, GRB7 and MED1. The differences in HER2 </span>
<span id="cb19-2037"><a href="#cb19-2037" aria-hidden="true" tabindex="-1"></a>are actually very small, but it looks like there is a consistent difference</span>
<span id="cb19-2038"><a href="#cb19-2038" aria-hidden="true" tabindex="-1"></a>still.</span>
<span id="cb19-2039"><a href="#cb19-2039" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2040"><a href="#cb19-2040" aria-hidden="true" tabindex="-1"></a>Just before we move on we compare the expression levels before and</span>
<span id="cb19-2041"><a href="#cb19-2041" aria-hidden="true" tabindex="-1"></a>after regressing the data on SCANB to see if we are not messing</span>
<span id="cb19-2042"><a href="#cb19-2042" aria-hidden="true" tabindex="-1"></a>up too much with the gene expression levels and getting just noise. </span>
<span id="cb19-2043"><a href="#cb19-2043" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2044"><a href="#cb19-2044" aria-hidden="true" tabindex="-1"></a>The plot below shows the correlation between ERBB2 FPKM and regressed</span>
<span id="cb19-2045"><a href="#cb19-2045" aria-hidden="true" tabindex="-1"></a>data and also for GRB7. We further show the correlation between</span>
<span id="cb19-2046"><a href="#cb19-2046" aria-hidden="true" tabindex="-1"></a>GRB7 and ERBB2 in the regressed data.</span>
<span id="cb19-2047"><a href="#cb19-2047" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2048"><a href="#cb19-2048" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=8, fig.height=6}</span></span>
<span id="cb19-2049"><a href="#cb19-2049" aria-hidden="true" tabindex="-1"></a>scores_her2_sub_genes <span class="ot">&lt;-</span> scores_her2 <span class="sc">%&gt;%</span></span>
<span id="cb19-2050"><a href="#cb19-2050" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">inner_join</span>(</span>
<span id="cb19-2051"><a href="#cb19-2051" aria-hidden="true" tabindex="-1"></a>        ., </span>
<span id="cb19-2052"><a href="#cb19-2052" aria-hidden="true" tabindex="-1"></a>        <span class="fu">data.frame</span>(</span>
<span id="cb19-2053"><a href="#cb19-2053" aria-hidden="true" tabindex="-1"></a>            <span class="at">sample_name =</span> <span class="fu">colnames</span>(datasets<span class="sc">$</span>scanb),</span>
<span id="cb19-2054"><a href="#cb19-2054" aria-hidden="true" tabindex="-1"></a>            <span class="at">grb7_fpkm =</span> <span class="fu">as.vector</span>(<span class="fu">assay</span>(datasets<span class="sc">$</span>scanb[<span class="st">"GRB7"</span>, ], <span class="st">"logFPKM"</span>)),</span>
<span id="cb19-2055"><a href="#cb19-2055" aria-hidden="true" tabindex="-1"></a>            <span class="at">her2_fpkm =</span> <span class="fu">as.vector</span>(<span class="fu">assay</span>(datasets<span class="sc">$</span>scanb[<span class="st">"ERBB2"</span>, ], <span class="st">"logFPKM"</span>)),</span>
<span id="cb19-2056"><a href="#cb19-2056" aria-hidden="true" tabindex="-1"></a>            <span class="at">grb7_reg =</span> <span class="fu">as.vector</span>(</span>
<span id="cb19-2057"><a href="#cb19-2057" aria-hidden="true" tabindex="-1"></a>                df_pcs_regressed[<span class="st">"GRB7"</span>, <span class="fu">colnames</span>(datasets<span class="sc">$</span>scanb)]</span>
<span id="cb19-2058"><a href="#cb19-2058" aria-hidden="true" tabindex="-1"></a>            ),</span>
<span id="cb19-2059"><a href="#cb19-2059" aria-hidden="true" tabindex="-1"></a>            <span class="at">her2_reg =</span> <span class="fu">as.vector</span>(</span>
<span id="cb19-2060"><a href="#cb19-2060" aria-hidden="true" tabindex="-1"></a>                df_pcs_regressed[<span class="st">"ERBB2"</span>, <span class="fu">colnames</span>(datasets<span class="sc">$</span>scanb)]</span>
<span id="cb19-2061"><a href="#cb19-2061" aria-hidden="true" tabindex="-1"></a>            )</span>
<span id="cb19-2062"><a href="#cb19-2062" aria-hidden="true" tabindex="-1"></a>        ),</span>
<span id="cb19-2063"><a href="#cb19-2063" aria-hidden="true" tabindex="-1"></a>        <span class="at">by =</span> <span class="st">"sample_name"</span></span>
<span id="cb19-2064"><a href="#cb19-2064" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb19-2065"><a href="#cb19-2065" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">inner_join</span>(</span>
<span id="cb19-2066"><a href="#cb19-2066" aria-hidden="true" tabindex="-1"></a>        .,</span>
<span id="cb19-2067"><a href="#cb19-2067" aria-hidden="true" tabindex="-1"></a>        df_pca,</span>
<span id="cb19-2068"><a href="#cb19-2068" aria-hidden="true" tabindex="-1"></a>        <span class="at">by =</span> <span class="st">"sample_name"</span></span>
<span id="cb19-2069"><a href="#cb19-2069" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb19-2070"><a href="#cb19-2070" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2071"><a href="#cb19-2071" aria-hidden="true" tabindex="-1"></a>p_her2 <span class="ot">&lt;-</span> scores_her2_sub_genes <span class="sc">%&gt;%</span> </span>
<span id="cb19-2072"><a href="#cb19-2072" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> her2_fpkm, <span class="at">y =</span> her2_reg, <span class="at">color =</span> pam50)) <span class="sc">+</span></span>
<span id="cb19-2073"><a href="#cb19-2073" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb19-2074"><a href="#cb19-2074" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_smooth</span>(</span>
<span id="cb19-2075"><a href="#cb19-2075" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">"black"</span>,</span>
<span id="cb19-2076"><a href="#cb19-2076" aria-hidden="true" tabindex="-1"></a>        <span class="at">method =</span> <span class="st">"gam"</span>,</span>
<span id="cb19-2077"><a href="#cb19-2077" aria-hidden="true" tabindex="-1"></a>        <span class="at">formula =</span> y <span class="sc">~</span> <span class="fu">s</span>(x, <span class="at">bs =</span> <span class="st">"cs"</span>)</span>
<span id="cb19-2078"><a href="#cb19-2078" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-2079"><a href="#cb19-2079" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">15</span>) <span class="sc">+</span> </span>
<span id="cb19-2080"><a href="#cb19-2080" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb19-2081"><a href="#cb19-2081" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2082"><a href="#cb19-2082" aria-hidden="true" tabindex="-1"></a>p_grb7 <span class="ot">&lt;-</span> scores_her2_sub_genes <span class="sc">%&gt;%</span> </span>
<span id="cb19-2083"><a href="#cb19-2083" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> grb7_fpkm, <span class="at">y =</span> grb7_reg, <span class="at">color =</span> pam50)) <span class="sc">+</span></span>
<span id="cb19-2084"><a href="#cb19-2084" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb19-2085"><a href="#cb19-2085" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_smooth</span>(</span>
<span id="cb19-2086"><a href="#cb19-2086" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">"black"</span>,</span>
<span id="cb19-2087"><a href="#cb19-2087" aria-hidden="true" tabindex="-1"></a>        <span class="at">method =</span> <span class="st">"gam"</span>,</span>
<span id="cb19-2088"><a href="#cb19-2088" aria-hidden="true" tabindex="-1"></a>        <span class="at">formula =</span> y <span class="sc">~</span> <span class="fu">s</span>(x, <span class="at">bs =</span> <span class="st">"cs"</span>)</span>
<span id="cb19-2089"><a href="#cb19-2089" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-2090"><a href="#cb19-2090" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">15</span>) <span class="sc">+</span> </span>
<span id="cb19-2091"><a href="#cb19-2091" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb19-2092"><a href="#cb19-2092" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2093"><a href="#cb19-2093" aria-hidden="true" tabindex="-1"></a>p_grb7_her2 <span class="ot">&lt;-</span> scores_her2_sub_genes <span class="sc">%&gt;%</span> </span>
<span id="cb19-2094"><a href="#cb19-2094" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> her2_reg, <span class="at">y =</span> grb7_reg, <span class="at">color =</span> pam50)) <span class="sc">+</span></span>
<span id="cb19-2095"><a href="#cb19-2095" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb19-2096"><a href="#cb19-2096" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_smooth</span>(</span>
<span id="cb19-2097"><a href="#cb19-2097" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">"black"</span>,</span>
<span id="cb19-2098"><a href="#cb19-2098" aria-hidden="true" tabindex="-1"></a>        <span class="at">method =</span> <span class="st">"gam"</span>,</span>
<span id="cb19-2099"><a href="#cb19-2099" aria-hidden="true" tabindex="-1"></a>        <span class="at">formula =</span> y <span class="sc">~</span> <span class="fu">s</span>(x, <span class="at">bs =</span> <span class="st">"cs"</span>)</span>
<span id="cb19-2100"><a href="#cb19-2100" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-2101"><a href="#cb19-2101" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">15</span>) <span class="sc">+</span> </span>
<span id="cb19-2102"><a href="#cb19-2102" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb19-2103"><a href="#cb19-2103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2104"><a href="#cb19-2104" aria-hidden="true" tabindex="-1"></a>p_grb7_her2_fpkm <span class="ot">&lt;-</span> scores_her2_sub_genes <span class="sc">%&gt;%</span> </span>
<span id="cb19-2105"><a href="#cb19-2105" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> her2_fpkm, <span class="at">y =</span> grb7_fpkm, <span class="at">color =</span> pam50)) <span class="sc">+</span></span>
<span id="cb19-2106"><a href="#cb19-2106" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb19-2107"><a href="#cb19-2107" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_smooth</span>(</span>
<span id="cb19-2108"><a href="#cb19-2108" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">"black"</span>,</span>
<span id="cb19-2109"><a href="#cb19-2109" aria-hidden="true" tabindex="-1"></a>        <span class="at">method =</span> <span class="st">"gam"</span>,</span>
<span id="cb19-2110"><a href="#cb19-2110" aria-hidden="true" tabindex="-1"></a>        <span class="at">formula =</span> y <span class="sc">~</span> <span class="fu">s</span>(x, <span class="at">bs =</span> <span class="st">"cs"</span>)</span>
<span id="cb19-2111"><a href="#cb19-2111" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-2112"><a href="#cb19-2112" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">15</span>) <span class="sc">+</span> </span>
<span id="cb19-2113"><a href="#cb19-2113" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb19-2114"><a href="#cb19-2114" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2115"><a href="#cb19-2115" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(p_her2, p_grb7, p_grb7_her2, p_grb7_her2_fpkm, <span class="at">nrow =</span> <span class="dv">2</span>)</span>
<span id="cb19-2116"><a href="#cb19-2116" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-2117"><a href="#cb19-2117" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2118"><a href="#cb19-2118" aria-hidden="true" tabindex="-1"></a>We see a good correlation between FPKM and regressed data. Moreover there is a good</span>
<span id="cb19-2119"><a href="#cb19-2119" aria-hidden="true" tabindex="-1"></a>correlation between GRB7 and HER2 in both regressed data and FPKM. We now</span>
<span id="cb19-2120"><a href="#cb19-2120" aria-hidden="true" tabindex="-1"></a>define a new score based on these genes available. The score is calculated</span>
<span id="cb19-2121"><a href="#cb19-2121" aria-hidden="true" tabindex="-1"></a>in the same way as before the regressed data, i.e., the sum of the </span>
<span id="cb19-2122"><a href="#cb19-2122" aria-hidden="true" tabindex="-1"></a>expression levels.</span>
<span id="cb19-2123"><a href="#cb19-2123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2124"><a href="#cb19-2124" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.height=8, fig.width=8}</span></span>
<span id="cb19-2125"><a href="#cb19-2125" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-her2-amplicon-her2subgroups</span></span>
<span id="cb19-2126"><a href="#cb19-2126" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: HER2 AMPLICON signature stratified by the HER2 status</span></span>
<span id="cb19-2127"><a href="#cb19-2127" aria-hidden="true" tabindex="-1"></a>df_pca<span class="sc">$</span>HER2_AMPLICON <span class="ot">&lt;-</span> <span class="fu">colSums</span>(</span>
<span id="cb19-2128"><a href="#cb19-2128" aria-hidden="true" tabindex="-1"></a>    df_pcs_regressed[genes_available_her2_amp, df_pca<span class="sc">$</span>sample_name]</span>
<span id="cb19-2129"><a href="#cb19-2129" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-2130"><a href="#cb19-2130" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2131"><a href="#cb19-2131" aria-hidden="true" tabindex="-1"></a>p_scanb_her2_amp_sig <span class="ot">&lt;-</span> df_pca <span class="sc">%&gt;%</span></span>
<span id="cb19-2132"><a href="#cb19-2132" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"scanb"</span> <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(HER2)) <span class="sc">%&gt;%</span></span>
<span id="cb19-2133"><a href="#cb19-2133" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(</span>
<span id="cb19-2134"><a href="#cb19-2134" aria-hidden="true" tabindex="-1"></a>        <span class="fu">aes</span>(<span class="at">x =</span> HER2, <span class="at">y =</span> HER2_AMPLICON, <span class="at">color =</span> HER2)</span>
<span id="cb19-2135"><a href="#cb19-2135" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-2136"><a href="#cb19-2136" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_jitter</span>(<span class="at">size =</span> <span class="fl">0.1</span>) <span class="sc">+</span> </span>
<span id="cb19-2137"><a href="#cb19-2137" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_violin</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span> </span>
<span id="cb19-2138"><a href="#cb19-2138" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">coord_flip</span>() <span class="sc">+</span> </span>
<span id="cb19-2139"><a href="#cb19-2139" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb19-2140"><a href="#cb19-2140" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"HER2"</span>,</span>
<span id="cb19-2141"><a href="#cb19-2141" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"HER2 AMPLICON"</span>,</span>
<span id="cb19-2142"><a href="#cb19-2142" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">"SCANB"</span></span>
<span id="cb19-2143"><a href="#cb19-2143" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-2144"><a href="#cb19-2144" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>) <span class="sc">+</span> </span>
<span id="cb19-2145"><a href="#cb19-2145" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>(</span>
<span id="cb19-2146"><a href="#cb19-2146" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb19-2147"><a href="#cb19-2147" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.spacing =</span> <span class="fu">unit</span>(<span class="dv">1</span>, <span class="st">"lines"</span>)</span>
<span id="cb19-2148"><a href="#cb19-2148" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-2149"><a href="#cb19-2149" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb19-2150"><a href="#cb19-2150" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2151"><a href="#cb19-2151" aria-hidden="true" tabindex="-1"></a>p_metabric_her2_amp_sig <span class="ot">&lt;-</span> df_pca <span class="sc">%&gt;%</span></span>
<span id="cb19-2152"><a href="#cb19-2152" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"metabric"</span> <span class="sc">&amp;</span> HER2_SNP6 <span class="sc">!=</span> <span class="st">"UNDEF"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-2153"><a href="#cb19-2153" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(</span>
<span id="cb19-2154"><a href="#cb19-2154" aria-hidden="true" tabindex="-1"></a>        <span class="fu">aes</span>(<span class="at">x =</span> HER2_SNP6, <span class="at">y =</span> HER2_AMPLICON, <span class="at">color =</span> HER2_SNP6)</span>
<span id="cb19-2155"><a href="#cb19-2155" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-2156"><a href="#cb19-2156" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_jitter</span>(<span class="at">size =</span> <span class="fl">0.1</span>) <span class="sc">+</span> </span>
<span id="cb19-2157"><a href="#cb19-2157" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_violin</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span> </span>
<span id="cb19-2158"><a href="#cb19-2158" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">coord_flip</span>() <span class="sc">+</span> </span>
<span id="cb19-2159"><a href="#cb19-2159" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb19-2160"><a href="#cb19-2160" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"HER2 SNP6"</span>,</span>
<span id="cb19-2161"><a href="#cb19-2161" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"HER2 AMPLICON"</span>,</span>
<span id="cb19-2162"><a href="#cb19-2162" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">"METABRIC"</span></span>
<span id="cb19-2163"><a href="#cb19-2163" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-2164"><a href="#cb19-2164" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>) <span class="sc">+</span> </span>
<span id="cb19-2165"><a href="#cb19-2165" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>(</span>
<span id="cb19-2166"><a href="#cb19-2166" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb19-2167"><a href="#cb19-2167" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.spacing =</span> <span class="fu">unit</span>(<span class="dv">1</span>, <span class="st">"lines"</span>)</span>
<span id="cb19-2168"><a href="#cb19-2168" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-2169"><a href="#cb19-2169" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb19-2170"><a href="#cb19-2170" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2171"><a href="#cb19-2171" aria-hidden="true" tabindex="-1"></a>p_poetic_her2_amp_sig <span class="ot">&lt;-</span> df_pca <span class="sc">%&gt;%</span></span>
<span id="cb19-2172"><a href="#cb19-2172" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"poetic"</span> <span class="sc">&amp;</span> timepoint <span class="sc">==</span> <span class="st">"baseline"</span> <span class="sc">&amp;</span> group <span class="sc">==</span> <span class="st">"treated"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-2173"><a href="#cb19-2173" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(</span>
<span id="cb19-2174"><a href="#cb19-2174" aria-hidden="true" tabindex="-1"></a>        <span class="fu">aes</span>(<span class="at">x =</span> her2_status, <span class="at">y =</span> HER2_AMPLICON, <span class="at">color =</span> her2_status)</span>
<span id="cb19-2175"><a href="#cb19-2175" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-2176"><a href="#cb19-2176" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_jitter</span>(<span class="at">size =</span> <span class="fl">0.1</span>) <span class="sc">+</span> </span>
<span id="cb19-2177"><a href="#cb19-2177" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_violin</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span> </span>
<span id="cb19-2178"><a href="#cb19-2178" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">coord_flip</span>() <span class="sc">+</span> </span>
<span id="cb19-2179"><a href="#cb19-2179" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb19-2180"><a href="#cb19-2180" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"HER2"</span>,</span>
<span id="cb19-2181"><a href="#cb19-2181" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"HER2 AMPLICON"</span>,</span>
<span id="cb19-2182"><a href="#cb19-2182" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">"POETIC"</span></span>
<span id="cb19-2183"><a href="#cb19-2183" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-2184"><a href="#cb19-2184" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>) <span class="sc">+</span> </span>
<span id="cb19-2185"><a href="#cb19-2185" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>(</span>
<span id="cb19-2186"><a href="#cb19-2186" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb19-2187"><a href="#cb19-2187" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.spacing =</span> <span class="fu">unit</span>(<span class="dv">1</span>, <span class="st">"lines"</span>)</span>
<span id="cb19-2188"><a href="#cb19-2188" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-2189"><a href="#cb19-2189" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb19-2190"><a href="#cb19-2190" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2191"><a href="#cb19-2191" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(</span>
<span id="cb19-2192"><a href="#cb19-2192" aria-hidden="true" tabindex="-1"></a>    p_scanb_her2_amp_sig,</span>
<span id="cb19-2193"><a href="#cb19-2193" aria-hidden="true" tabindex="-1"></a>    p_metabric_her2_amp_sig,</span>
<span id="cb19-2194"><a href="#cb19-2194" aria-hidden="true" tabindex="-1"></a>    <span class="at">nrow =</span> <span class="dv">2</span></span>
<span id="cb19-2195"><a href="#cb19-2195" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-2196"><a href="#cb19-2196" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-2197"><a href="#cb19-2197" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2198"><a href="#cb19-2198" aria-hidden="true" tabindex="-1"></a>And now plotting for the molecular subtypes specifically.</span>
<span id="cb19-2199"><a href="#cb19-2199" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2200"><a href="#cb19-2200" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.height=4, fig.width=10}</span></span>
<span id="cb19-2201"><a href="#cb19-2201" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-pam50-her2-amplicon</span></span>
<span id="cb19-2202"><a href="#cb19-2202" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: HER2 AMPLICON signature stratified by the PAM50 molecular</span></span>
<span id="cb19-2203"><a href="#cb19-2203" aria-hidden="true" tabindex="-1"></a><span class="co">#|     subtypes</span></span>
<span id="cb19-2204"><a href="#cb19-2204" aria-hidden="true" tabindex="-1"></a>df_pca <span class="sc">%&gt;%</span></span>
<span id="cb19-2205"><a href="#cb19-2205" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">!=</span> <span class="st">"poetic"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-2206"><a href="#cb19-2206" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(</span>
<span id="cb19-2207"><a href="#cb19-2207" aria-hidden="true" tabindex="-1"></a>        <span class="fu">aes</span>(<span class="at">x =</span> pam50, <span class="at">y =</span> HER2_AMPLICON, <span class="at">color =</span> pam50)</span>
<span id="cb19-2208"><a href="#cb19-2208" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-2209"><a href="#cb19-2209" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_jitter</span>(<span class="at">size =</span> <span class="fl">0.1</span>) <span class="sc">+</span> </span>
<span id="cb19-2210"><a href="#cb19-2210" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_violin</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span> </span>
<span id="cb19-2211"><a href="#cb19-2211" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">coord_flip</span>() <span class="sc">+</span> </span>
<span id="cb19-2212"><a href="#cb19-2212" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb19-2213"><a href="#cb19-2213" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"PAM50"</span>,</span>
<span id="cb19-2214"><a href="#cb19-2214" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"HER2 AMPLICON"</span></span>
<span id="cb19-2215"><a href="#cb19-2215" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-2216"><a href="#cb19-2216" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(</span>
<span id="cb19-2217"><a href="#cb19-2217" aria-hidden="true" tabindex="-1"></a>        <span class="sc">~</span>cohort,</span>
<span id="cb19-2218"><a href="#cb19-2218" aria-hidden="true" tabindex="-1"></a>        <span class="at">labeller =</span> <span class="fu">as_labeller</span>(<span class="fu">c</span>(</span>
<span id="cb19-2219"><a href="#cb19-2219" aria-hidden="true" tabindex="-1"></a>            <span class="st">"metabric"</span> <span class="ot">=</span> <span class="st">"METABRIC"</span>, </span>
<span id="cb19-2220"><a href="#cb19-2220" aria-hidden="true" tabindex="-1"></a>             <span class="st">"tcga"</span> <span class="ot">=</span> <span class="st">"TCGA"</span>,</span>
<span id="cb19-2221"><a href="#cb19-2221" aria-hidden="true" tabindex="-1"></a>             <span class="st">"scanb"</span> <span class="ot">=</span> <span class="st">"SCANB"</span></span>
<span id="cb19-2222"><a href="#cb19-2222" aria-hidden="true" tabindex="-1"></a>        ))</span>
<span id="cb19-2223"><a href="#cb19-2223" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-2224"><a href="#cb19-2224" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>) <span class="sc">+</span> </span>
<span id="cb19-2225"><a href="#cb19-2225" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_color_manual</span>(</span>
<span id="cb19-2226"><a href="#cb19-2226" aria-hidden="true" tabindex="-1"></a>        <span class="at">values =</span> <span class="fu">get_colors_pam50</span>(df_pca <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">!=</span> <span class="st">"poetic"</span>))</span>
<span id="cb19-2227"><a href="#cb19-2227" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-2228"><a href="#cb19-2228" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>(</span>
<span id="cb19-2229"><a href="#cb19-2229" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb19-2230"><a href="#cb19-2230" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.spacing =</span> <span class="fu">unit</span>(<span class="dv">1</span>, <span class="st">"lines"</span>)</span>
<span id="cb19-2231"><a href="#cb19-2231" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-2232"><a href="#cb19-2232" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb19-2233"><a href="#cb19-2233" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-2234"><a href="#cb19-2234" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2235"><a href="#cb19-2235" aria-hidden="true" tabindex="-1"></a>This is a signature that captures better the HER2 molecular subtype, but not in</span>
<span id="cb19-2236"><a href="#cb19-2236" aria-hidden="true" tabindex="-1"></a>its entirety. </span>
<span id="cb19-2237"><a href="#cb19-2237" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2238"><a href="#cb19-2238" aria-hidden="true" tabindex="-1"></a>Similar to what we did previously,</span>
<span id="cb19-2239"><a href="#cb19-2239" aria-hidden="true" tabindex="-1"></a>we check if this signature is correlated with proliferation. </span>
<span id="cb19-2240"><a href="#cb19-2240" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2243"><a href="#cb19-2243" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-2244"><a href="#cb19-2244" aria-hidden="true" tabindex="-1"></a>df_pca <span class="sc">%&gt;%</span> </span>
<span id="cb19-2245"><a href="#cb19-2245" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(</span>
<span id="cb19-2246"><a href="#cb19-2246" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> HER2_AMPLICON, </span>
<span id="cb19-2247"><a href="#cb19-2247" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> HALLMARK_G2M_CHECKPOINT, </span>
<span id="cb19-2248"><a href="#cb19-2248" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> pam50</span>
<span id="cb19-2249"><a href="#cb19-2249" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">+</span></span>
<span id="cb19-2250"><a href="#cb19-2250" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb19-2251"><a href="#cb19-2251" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_smooth</span>(<span class="at">color =</span> <span class="st">"black"</span>, <span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">formula =</span> <span class="st">"y~x"</span>) <span class="sc">+</span> </span>
<span id="cb19-2252"><a href="#cb19-2252" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>cohort) <span class="sc">+</span> </span>
<span id="cb19-2253"><a href="#cb19-2253" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">15</span>) <span class="sc">+</span> </span>
<span id="cb19-2254"><a href="#cb19-2254" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_color_manual</span>(</span>
<span id="cb19-2255"><a href="#cb19-2255" aria-hidden="true" tabindex="-1"></a>        <span class="at">values =</span> <span class="fu">get_colors_pam50</span>(</span>
<span id="cb19-2256"><a href="#cb19-2256" aria-hidden="true" tabindex="-1"></a>            df_pca <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">!=</span> <span class="st">"poetic"</span>)</span>
<span id="cb19-2257"><a href="#cb19-2257" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb19-2258"><a href="#cb19-2258" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-2259"><a href="#cb19-2259" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb19-2260"><a href="#cb19-2260" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-2261"><a href="#cb19-2261" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2262"><a href="#cb19-2262" aria-hidden="true" tabindex="-1"></a>If there is any correlation it is too small and not clear. For</span>
<span id="cb19-2263"><a href="#cb19-2263" aria-hidden="true" tabindex="-1"></a>comparison we do the same plot for the HER2 positive patients.</span>
<span id="cb19-2264"><a href="#cb19-2264" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2267"><a href="#cb19-2267" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-2268"><a href="#cb19-2268" aria-hidden="true" tabindex="-1"></a>df_pca <span class="sc">%&gt;%</span> </span>
<span id="cb19-2269"><a href="#cb19-2269" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(HER2 <span class="sc">==</span> <span class="st">"Positive"</span> <span class="sc">|</span> HER2_SNP6 <span class="sc">==</span> <span class="st">"GAIN"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-2270"><a href="#cb19-2270" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(</span>
<span id="cb19-2271"><a href="#cb19-2271" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> HER2_AMPLICON, </span>
<span id="cb19-2272"><a href="#cb19-2272" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> HALLMARK_G2M_CHECKPOINT, </span>
<span id="cb19-2273"><a href="#cb19-2273" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> pam50</span>
<span id="cb19-2274"><a href="#cb19-2274" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">+</span></span>
<span id="cb19-2275"><a href="#cb19-2275" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb19-2276"><a href="#cb19-2276" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_smooth</span>(<span class="at">color =</span> <span class="st">"black"</span>, <span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">formula =</span> <span class="st">"y~x"</span>) <span class="sc">+</span> </span>
<span id="cb19-2277"><a href="#cb19-2277" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>cohort) <span class="sc">+</span> </span>
<span id="cb19-2278"><a href="#cb19-2278" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">15</span>) <span class="sc">+</span> </span>
<span id="cb19-2279"><a href="#cb19-2279" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_color_manual</span>(</span>
<span id="cb19-2280"><a href="#cb19-2280" aria-hidden="true" tabindex="-1"></a>        <span class="at">values =</span> <span class="fu">get_colors_pam50</span>(</span>
<span id="cb19-2281"><a href="#cb19-2281" aria-hidden="true" tabindex="-1"></a>            df_pca <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">!=</span> <span class="st">"poetic"</span>)</span>
<span id="cb19-2282"><a href="#cb19-2282" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb19-2283"><a href="#cb19-2283" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-2284"><a href="#cb19-2284" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb19-2285"><a href="#cb19-2285" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-2286"><a href="#cb19-2286" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2287"><a href="#cb19-2287" aria-hidden="true" tabindex="-1"></a>Again if there is any kind of correlation it is weak. But notice how </span>
<span id="cb19-2288"><a href="#cb19-2288" aria-hidden="true" tabindex="-1"></a>the majority of the samples goes from 0 to 5. The maximum that a non </span>
<span id="cb19-2289"><a href="#cb19-2289" aria-hidden="true" tabindex="-1"></a>HER2 positive sample goes is 2. Below is the density plot for the </span>
<span id="cb19-2290"><a href="#cb19-2290" aria-hidden="true" tabindex="-1"></a>HER2+s.</span>
<span id="cb19-2291"><a href="#cb19-2291" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2294"><a href="#cb19-2294" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-2295"><a href="#cb19-2295" aria-hidden="true" tabindex="-1"></a>df_pca <span class="sc">%&gt;%</span> </span>
<span id="cb19-2296"><a href="#cb19-2296" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(HER2 <span class="sc">==</span> <span class="st">"Positive"</span> <span class="sc">|</span> HER2_SNP6 <span class="sc">==</span> <span class="st">"GAIN"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-2297"><a href="#cb19-2297" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> HER2_AMPLICON, <span class="at">fill =</span> cohort)) <span class="sc">+</span></span>
<span id="cb19-2298"><a href="#cb19-2298" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_histogram</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="at">position =</span> <span class="st">"identity"</span>, <span class="at">bins =</span> <span class="dv">30</span>) <span class="sc">+</span></span>
<span id="cb19-2299"><a href="#cb19-2299" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>cohort, <span class="at">nrow =</span> <span class="dv">2</span>, <span class="at">scale =</span> <span class="st">"free_y"</span>) <span class="sc">+</span> </span>
<span id="cb19-2300"><a href="#cb19-2300" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>)</span>
<span id="cb19-2301"><a href="#cb19-2301" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-2302"><a href="#cb19-2302" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2303"><a href="#cb19-2303" aria-hidden="true" tabindex="-1"></a>Since when giving aromatase inhibitors to patients we are shutting down</span>
<span id="cb19-2304"><a href="#cb19-2304" aria-hidden="true" tabindex="-1"></a>ER signaling or at least reducing it to some extent, so we expect to see</span>
<span id="cb19-2305"><a href="#cb19-2305" aria-hidden="true" tabindex="-1"></a>alternative pathways taking the lead so the cells can keep proliferating.</span>
<span id="cb19-2306"><a href="#cb19-2306" aria-hidden="true" tabindex="-1"></a>Here we calculate the difference between HER2 amplicon signature before and</span>
<span id="cb19-2307"><a href="#cb19-2307" aria-hidden="true" tabindex="-1"></a>after treatment for responders and non-responders.</span>
<span id="cb19-2308"><a href="#cb19-2308" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2311"><a href="#cb19-2311" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-2312"><a href="#cb19-2312" aria-hidden="true" tabindex="-1"></a>diff_her2_poetic <span class="ot">&lt;-</span> df_pca <span class="sc">%&gt;%</span></span>
<span id="cb19-2313"><a href="#cb19-2313" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb19-2314"><a href="#cb19-2314" aria-hidden="true" tabindex="-1"></a>        cohort <span class="sc">==</span> <span class="st">"poetic"</span> <span class="sc">&amp;</span> </span>
<span id="cb19-2315"><a href="#cb19-2315" aria-hidden="true" tabindex="-1"></a>            group <span class="sc">==</span> <span class="st">"treated"</span> <span class="sc">&amp;</span> </span>
<span id="cb19-2316"><a href="#cb19-2316" aria-hidden="true" tabindex="-1"></a>            is_responder <span class="sc">!=</span> <span class="st">"not_available"</span></span>
<span id="cb19-2317"><a href="#cb19-2317" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb19-2318"><a href="#cb19-2318" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">pivot_wider</span>(</span>
<span id="cb19-2319"><a href="#cb19-2319" aria-hidden="true" tabindex="-1"></a>        <span class="at">id_cols =</span> patient_nb, </span>
<span id="cb19-2320"><a href="#cb19-2320" aria-hidden="true" tabindex="-1"></a>        <span class="at">names_from =</span> timepoint, </span>
<span id="cb19-2321"><a href="#cb19-2321" aria-hidden="true" tabindex="-1"></a>        <span class="at">values_from =</span> HER2_AMPLICON</span>
<span id="cb19-2322"><a href="#cb19-2322" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span> </span>
<span id="cb19-2323"><a href="#cb19-2323" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">diff_HER2_AMPLICON =</span> surgery <span class="sc">-</span> baseline) <span class="sc">%&gt;%</span></span>
<span id="cb19-2324"><a href="#cb19-2324" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">inner_join</span>(</span>
<span id="cb19-2325"><a href="#cb19-2325" aria-hidden="true" tabindex="-1"></a>        ., df_pca,</span>
<span id="cb19-2326"><a href="#cb19-2326" aria-hidden="true" tabindex="-1"></a>        <span class="at">by =</span> <span class="st">"patient_nb"</span></span>
<span id="cb19-2327"><a href="#cb19-2327" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb19-2328"><a href="#cb19-2328" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">distinct</span>(patient_nb, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>) </span>
<span id="cb19-2329"><a href="#cb19-2329" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2330"><a href="#cb19-2330" aria-hidden="true" tabindex="-1"></a>diff_her2_poetic <span class="sc">%&gt;%</span></span>
<span id="cb19-2331"><a href="#cb19-2331" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> is_responder, <span class="at">y =</span> diff_HER2_AMPLICON)) <span class="sc">+</span></span>
<span id="cb19-2332"><a href="#cb19-2332" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_jitter</span>() <span class="sc">+</span> </span>
<span id="cb19-2333"><a href="#cb19-2333" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_boxplot</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="at">outlier.shape =</span> <span class="cn">NA</span>) <span class="sc">+</span> </span>
<span id="cb19-2334"><a href="#cb19-2334" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb19-2335"><a href="#cb19-2335" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"Responder or non-responder"</span>,</span>
<span id="cb19-2336"><a href="#cb19-2336" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"Difference in</span><span class="sc">\n</span><span class="st">HER2 AMPLICON"</span></span>
<span id="cb19-2337"><a href="#cb19-2337" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb19-2338"><a href="#cb19-2338" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-2339"><a href="#cb19-2339" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>)</span>
<span id="cb19-2340"><a href="#cb19-2340" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-2341"><a href="#cb19-2341" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2342"><a href="#cb19-2342" aria-hidden="true" tabindex="-1"></a>There is a very small difference between the two groups with responders</span>
<span id="cb19-2343"><a href="#cb19-2343" aria-hidden="true" tabindex="-1"></a>having a bit higher difference, meaning higher values of the HER2 AMPLICON</span>
<span id="cb19-2344"><a href="#cb19-2344" aria-hidden="true" tabindex="-1"></a>signature after treatment in average.</span>
<span id="cb19-2345"><a href="#cb19-2345" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2346"><a href="#cb19-2346" aria-hidden="true" tabindex="-1"></a><span class="fu">### HER2 Amplicon and survival analysis</span></span>
<span id="cb19-2347"><a href="#cb19-2347" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2348"><a href="#cb19-2348" aria-hidden="true" tabindex="-1"></a>Next we evaluate the HER2 amplicon signature in patients that received</span>
<span id="cb19-2349"><a href="#cb19-2349" aria-hidden="true" tabindex="-1"></a>trastuzumab + chemo + endo or just </span>
<span id="cb19-2350"><a href="#cb19-2350" aria-hidden="true" tabindex="-1"></a>trastuzumab + chemo, and have HER2+ and ER+ BC. The hypothesis here is that</span>
<span id="cb19-2351"><a href="#cb19-2351" aria-hidden="true" tabindex="-1"></a>the higher the signalling the better the outcome. In the SCAN-B dataset</span>
<span id="cb19-2352"><a href="#cb19-2352" aria-hidden="true" tabindex="-1"></a>we are making a big assumption in that the immunotherapy available on</span>
<span id="cb19-2353"><a href="#cb19-2353" aria-hidden="true" tabindex="-1"></a><span class="in">`TreatGroup`</span> corresponds to herceptin in this context for patients with</span>
<span id="cb19-2354"><a href="#cb19-2354" aria-hidden="true" tabindex="-1"></a>HER2+ BC.</span>
<span id="cb19-2355"><a href="#cb19-2355" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2358"><a href="#cb19-2358" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-2359"><a href="#cb19-2359" aria-hidden="true" tabindex="-1"></a>df_pca <span class="sc">%&gt;%</span></span>
<span id="cb19-2360"><a href="#cb19-2360" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"scanb"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-2361"><a href="#cb19-2361" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(HER2 <span class="sc">==</span> <span class="st">"Positive"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-2362"><a href="#cb19-2362" aria-hidden="true" tabindex="-1"></a>    janitor<span class="sc">::</span><span class="fu">tabyl</span>(TreatGroup, ER)</span>
<span id="cb19-2363"><a href="#cb19-2363" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-2364"><a href="#cb19-2364" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2365"><a href="#cb19-2365" aria-hidden="true" tabindex="-1"></a>Above the number of patients with HER2+ BC stratified by </span>
<span id="cb19-2366"><a href="#cb19-2366" aria-hidden="true" tabindex="-1"></a>ER status and treatment group. Majority is ER+ and received</span>
<span id="cb19-2367"><a href="#cb19-2367" aria-hidden="true" tabindex="-1"></a>endocrine therapy as well.</span>
<span id="cb19-2368"><a href="#cb19-2368" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2369"><a href="#cb19-2369" aria-hidden="true" tabindex="-1"></a>The table below shows the number of events ($rfs<span class="sc">\_</span>status = 2$) for </span>
<span id="cb19-2370"><a href="#cb19-2370" aria-hidden="true" tabindex="-1"></a>patients with HER2+ ER+ BC.</span>
<span id="cb19-2371"><a href="#cb19-2371" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2374"><a href="#cb19-2374" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-2375"><a href="#cb19-2375" aria-hidden="true" tabindex="-1"></a>df_pca <span class="sc">%&gt;%</span></span>
<span id="cb19-2376"><a href="#cb19-2376" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"scanb"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-2377"><a href="#cb19-2377" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(HER2 <span class="sc">==</span> <span class="st">"Positive"</span> <span class="sc">&amp;</span> ER <span class="sc">==</span> <span class="st">"Positive"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-2378"><a href="#cb19-2378" aria-hidden="true" tabindex="-1"></a>    janitor<span class="sc">::</span><span class="fu">tabyl</span>(TreatGroup, rfs_status)</span>
<span id="cb19-2379"><a href="#cb19-2379" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-2380"><a href="#cb19-2380" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2381"><a href="#cb19-2381" aria-hidden="true" tabindex="-1"></a>There is a considerable amount of patients with recurrence. The median </span>
<span id="cb19-2382"><a href="#cb19-2382" aria-hidden="true" tabindex="-1"></a>recurrence free survival for those patients that received immu + chemo + endo</span>
<span id="cb19-2383"><a href="#cb19-2383" aria-hidden="true" tabindex="-1"></a>is:</span>
<span id="cb19-2384"><a href="#cb19-2384" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2387"><a href="#cb19-2387" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-2388"><a href="#cb19-2388" aria-hidden="true" tabindex="-1"></a>df_pca <span class="sc">%&gt;%</span></span>
<span id="cb19-2389"><a href="#cb19-2389" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"scanb"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-2390"><a href="#cb19-2390" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(HER2 <span class="sc">==</span> <span class="st">"Positive"</span> <span class="sc">&amp;</span> ER <span class="sc">==</span> <span class="st">"Positive"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-2391"><a href="#cb19-2391" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(TreatGroup <span class="sc">==</span> <span class="st">"ImmuChemoEndo"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-2392"><a href="#cb19-2392" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">median =</span> <span class="fu">median</span>(rfs_months, <span class="at">na.rm =</span> <span class="cn">TRUE</span>))</span>
<span id="cb19-2393"><a href="#cb19-2393" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-2394"><a href="#cb19-2394" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2395"><a href="#cb19-2395" aria-hidden="true" tabindex="-1"></a>i.e., 67 months or 5.6 years. We now calculate the survival analysis. </span>
<span id="cb19-2396"><a href="#cb19-2396" aria-hidden="true" tabindex="-1"></a>And for the patients that have ER- BC and received only immuno + chemo:</span>
<span id="cb19-2397"><a href="#cb19-2397" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2400"><a href="#cb19-2400" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-2401"><a href="#cb19-2401" aria-hidden="true" tabindex="-1"></a>df_pca <span class="sc">%&gt;%</span></span>
<span id="cb19-2402"><a href="#cb19-2402" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"scanb"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-2403"><a href="#cb19-2403" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(HER2 <span class="sc">==</span> <span class="st">"Positive"</span> <span class="sc">&amp;</span> ER <span class="sc">==</span> <span class="st">"Negative"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-2404"><a href="#cb19-2404" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(TreatGroup <span class="sc">==</span> <span class="st">"ImmuChemo"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-2405"><a href="#cb19-2405" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">median =</span> <span class="fu">median</span>(rfs_months, <span class="at">na.rm =</span> <span class="cn">TRUE</span>))</span>
<span id="cb19-2406"><a href="#cb19-2406" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-2407"><a href="#cb19-2407" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2408"><a href="#cb19-2408" aria-hidden="true" tabindex="-1"></a>Similar as for the ER+ BC patients.</span>
<span id="cb19-2409"><a href="#cb19-2409" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2410"><a href="#cb19-2410" aria-hidden="true" tabindex="-1"></a>Below are the results from the analysis using the HER2 AMPLICON signature for</span>
<span id="cb19-2411"><a href="#cb19-2411" aria-hidden="true" tabindex="-1"></a>both ER- and ER+ BC patients separately.</span>
<span id="cb19-2412"><a href="#cb19-2412" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2415"><a href="#cb19-2415" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-2416"><a href="#cb19-2416" aria-hidden="true" tabindex="-1"></a>survival_her2_amplicon_erneg <span class="ot">&lt;-</span> survival<span class="sc">::</span><span class="fu">coxph</span>(</span>
<span id="cb19-2417"><a href="#cb19-2417" aria-hidden="true" tabindex="-1"></a>    <span class="fu">Surv</span>(rfs_months, rfs_status) <span class="sc">~</span> HER2_AMPLICON <span class="sc">+</span> age <span class="sc">+</span> </span>
<span id="cb19-2418"><a href="#cb19-2418" aria-hidden="true" tabindex="-1"></a>        node_stage <span class="sc">+</span> tumor_stage,</span>
<span id="cb19-2419"><a href="#cb19-2419" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> df_pca <span class="sc">%&gt;%</span></span>
<span id="cb19-2420"><a href="#cb19-2420" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"scanb"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-2421"><a href="#cb19-2421" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb19-2422"><a href="#cb19-2422" aria-hidden="true" tabindex="-1"></a>            HER2 <span class="sc">==</span> <span class="st">"Positive"</span> <span class="sc">&amp;</span> </span>
<span id="cb19-2423"><a href="#cb19-2423" aria-hidden="true" tabindex="-1"></a>                ER <span class="sc">==</span> <span class="st">"Negative"</span> <span class="sc">&amp;</span> </span>
<span id="cb19-2424"><a href="#cb19-2424" aria-hidden="true" tabindex="-1"></a>                TreatGroup <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"ImmuChemo"</span>) <span class="sc">&amp;</span> </span>
<span id="cb19-2425"><a href="#cb19-2425" aria-hidden="true" tabindex="-1"></a>                <span class="sc">!</span><span class="fu">is.na</span>(rfs_status)</span>
<span id="cb19-2426"><a href="#cb19-2426" aria-hidden="true" tabindex="-1"></a>        ) <span class="sc">%&gt;%</span></span>
<span id="cb19-2427"><a href="#cb19-2427" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">inner_join</span>(</span>
<span id="cb19-2428"><a href="#cb19-2428" aria-hidden="true" tabindex="-1"></a>            .,</span>
<span id="cb19-2429"><a href="#cb19-2429" aria-hidden="true" tabindex="-1"></a>            scores_her2_sub_genes <span class="sc">%&gt;%</span></span>
<span id="cb19-2430"><a href="#cb19-2430" aria-hidden="true" tabindex="-1"></a>                dplyr<span class="sc">::</span><span class="fu">select</span>(her2_fpkm, sample_name)</span>
<span id="cb19-2431"><a href="#cb19-2431" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb19-2432"><a href="#cb19-2432" aria-hidden="true" tabindex="-1"></a>) </span>
<span id="cb19-2433"><a href="#cb19-2433" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2434"><a href="#cb19-2434" aria-hidden="true" tabindex="-1"></a>survival_her2_fpkm_erneg <span class="ot">&lt;-</span> survival<span class="sc">::</span><span class="fu">coxph</span>(</span>
<span id="cb19-2435"><a href="#cb19-2435" aria-hidden="true" tabindex="-1"></a>    <span class="fu">Surv</span>(rfs_months, rfs_status) <span class="sc">~</span> her2_fpkm <span class="sc">+</span> age <span class="sc">+</span> </span>
<span id="cb19-2436"><a href="#cb19-2436" aria-hidden="true" tabindex="-1"></a>        node_stage <span class="sc">+</span> tumor_stage,</span>
<span id="cb19-2437"><a href="#cb19-2437" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> df_pca <span class="sc">%&gt;%</span></span>
<span id="cb19-2438"><a href="#cb19-2438" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"scanb"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-2439"><a href="#cb19-2439" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb19-2440"><a href="#cb19-2440" aria-hidden="true" tabindex="-1"></a>            HER2 <span class="sc">==</span> <span class="st">"Positive"</span> <span class="sc">&amp;</span> </span>
<span id="cb19-2441"><a href="#cb19-2441" aria-hidden="true" tabindex="-1"></a>                ER <span class="sc">==</span> <span class="st">"Negative"</span> <span class="sc">&amp;</span> </span>
<span id="cb19-2442"><a href="#cb19-2442" aria-hidden="true" tabindex="-1"></a>                TreatGroup <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"ImmuChemo"</span>) <span class="sc">&amp;</span> </span>
<span id="cb19-2443"><a href="#cb19-2443" aria-hidden="true" tabindex="-1"></a>                <span class="sc">!</span><span class="fu">is.na</span>(rfs_status)</span>
<span id="cb19-2444"><a href="#cb19-2444" aria-hidden="true" tabindex="-1"></a>        ) <span class="sc">%&gt;%</span></span>
<span id="cb19-2445"><a href="#cb19-2445" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">inner_join</span>(</span>
<span id="cb19-2446"><a href="#cb19-2446" aria-hidden="true" tabindex="-1"></a>            .,</span>
<span id="cb19-2447"><a href="#cb19-2447" aria-hidden="true" tabindex="-1"></a>            scores_her2_sub_genes <span class="sc">%&gt;%</span></span>
<span id="cb19-2448"><a href="#cb19-2448" aria-hidden="true" tabindex="-1"></a>                dplyr<span class="sc">::</span><span class="fu">select</span>(her2_fpkm, sample_name) <span class="sc">%&gt;%</span></span>
<span id="cb19-2449"><a href="#cb19-2449" aria-hidden="true" tabindex="-1"></a>                dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">her2_fpkm =</span> <span class="fu">scale</span>(her2_fpkm)),</span>
<span id="cb19-2450"><a href="#cb19-2450" aria-hidden="true" tabindex="-1"></a>            <span class="at">by =</span> <span class="st">"sample_name"</span></span>
<span id="cb19-2451"><a href="#cb19-2451" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb19-2452"><a href="#cb19-2452" aria-hidden="true" tabindex="-1"></a>) </span>
<span id="cb19-2453"><a href="#cb19-2453" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2454"><a href="#cb19-2454" aria-hidden="true" tabindex="-1"></a>survival_her2_amplicon_erpos <span class="ot">&lt;-</span> survival<span class="sc">::</span><span class="fu">coxph</span>(</span>
<span id="cb19-2455"><a href="#cb19-2455" aria-hidden="true" tabindex="-1"></a>    <span class="fu">Surv</span>(rfs_months, rfs_status) <span class="sc">~</span> HER2_AMPLICON <span class="sc">+</span> age <span class="sc">+</span> </span>
<span id="cb19-2456"><a href="#cb19-2456" aria-hidden="true" tabindex="-1"></a>        node_stage <span class="sc">+</span> tumor_stage,</span>
<span id="cb19-2457"><a href="#cb19-2457" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> df_pca <span class="sc">%&gt;%</span></span>
<span id="cb19-2458"><a href="#cb19-2458" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"scanb"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-2459"><a href="#cb19-2459" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb19-2460"><a href="#cb19-2460" aria-hidden="true" tabindex="-1"></a>            HER2 <span class="sc">==</span> <span class="st">"Positive"</span> <span class="sc">&amp;</span> </span>
<span id="cb19-2461"><a href="#cb19-2461" aria-hidden="true" tabindex="-1"></a>                ER <span class="sc">==</span> <span class="st">"Positive"</span> <span class="sc">&amp;</span> </span>
<span id="cb19-2462"><a href="#cb19-2462" aria-hidden="true" tabindex="-1"></a>                TreatGroup <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"ImmuChemoEndo"</span>) <span class="sc">&amp;</span> </span>
<span id="cb19-2463"><a href="#cb19-2463" aria-hidden="true" tabindex="-1"></a>                <span class="sc">!</span><span class="fu">is.na</span>(rfs_status)</span>
<span id="cb19-2464"><a href="#cb19-2464" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb19-2465"><a href="#cb19-2465" aria-hidden="true" tabindex="-1"></a>) </span>
<span id="cb19-2466"><a href="#cb19-2466" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2467"><a href="#cb19-2467" aria-hidden="true" tabindex="-1"></a>survival_her2_fpkm_erpos <span class="ot">&lt;-</span> survival<span class="sc">::</span><span class="fu">coxph</span>(</span>
<span id="cb19-2468"><a href="#cb19-2468" aria-hidden="true" tabindex="-1"></a>    <span class="fu">Surv</span>(rfs_months, rfs_status) <span class="sc">~</span> her2_fpkm <span class="sc">+</span> age <span class="sc">+</span> </span>
<span id="cb19-2469"><a href="#cb19-2469" aria-hidden="true" tabindex="-1"></a>        node_stage <span class="sc">+</span> tumor_stage,</span>
<span id="cb19-2470"><a href="#cb19-2470" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> df_pca <span class="sc">%&gt;%</span></span>
<span id="cb19-2471"><a href="#cb19-2471" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"scanb"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-2472"><a href="#cb19-2472" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb19-2473"><a href="#cb19-2473" aria-hidden="true" tabindex="-1"></a>            HER2 <span class="sc">==</span> <span class="st">"Positive"</span> <span class="sc">&amp;</span> </span>
<span id="cb19-2474"><a href="#cb19-2474" aria-hidden="true" tabindex="-1"></a>                ER <span class="sc">==</span> <span class="st">"Positive"</span> <span class="sc">&amp;</span> </span>
<span id="cb19-2475"><a href="#cb19-2475" aria-hidden="true" tabindex="-1"></a>                TreatGroup <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"ImmuChemoEndo"</span>) <span class="sc">&amp;</span> </span>
<span id="cb19-2476"><a href="#cb19-2476" aria-hidden="true" tabindex="-1"></a>                <span class="sc">!</span><span class="fu">is.na</span>(rfs_status)</span>
<span id="cb19-2477"><a href="#cb19-2477" aria-hidden="true" tabindex="-1"></a>        ) <span class="sc">%&gt;%</span></span>
<span id="cb19-2478"><a href="#cb19-2478" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">inner_join</span>(</span>
<span id="cb19-2479"><a href="#cb19-2479" aria-hidden="true" tabindex="-1"></a>            .,</span>
<span id="cb19-2480"><a href="#cb19-2480" aria-hidden="true" tabindex="-1"></a>            scores_her2_sub_genes <span class="sc">%&gt;%</span></span>
<span id="cb19-2481"><a href="#cb19-2481" aria-hidden="true" tabindex="-1"></a>                dplyr<span class="sc">::</span><span class="fu">select</span>(her2_fpkm, sample_name) <span class="sc">%&gt;%</span></span>
<span id="cb19-2482"><a href="#cb19-2482" aria-hidden="true" tabindex="-1"></a>                dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">her2_fpkm =</span> <span class="fu">scale</span>(her2_fpkm)),</span>
<span id="cb19-2483"><a href="#cb19-2483" aria-hidden="true" tabindex="-1"></a>            <span class="at">by =</span> <span class="st">"sample_name"</span></span>
<span id="cb19-2484"><a href="#cb19-2484" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb19-2485"><a href="#cb19-2485" aria-hidden="true" tabindex="-1"></a>) </span>
<span id="cb19-2486"><a href="#cb19-2486" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2487"><a href="#cb19-2487" aria-hidden="true" tabindex="-1"></a>survival_her2_amplicon_tidy_erneg <span class="ot">&lt;-</span> survival_her2_amplicon_erneg <span class="sc">%&gt;%</span> </span>
<span id="cb19-2488"><a href="#cb19-2488" aria-hidden="true" tabindex="-1"></a>    broom<span class="sc">::</span><span class="fu">tidy</span>(., <span class="at">exponentiate =</span> <span class="cn">TRUE</span>, <span class="at">conf.int =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-2489"><a href="#cb19-2489" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(term, estimate, conf.low, conf.high, std.error, p.value) <span class="sc">%&gt;%</span></span>
<span id="cb19-2490"><a href="#cb19-2490" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">pathway =</span> term) </span>
<span id="cb19-2491"><a href="#cb19-2491" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2492"><a href="#cb19-2492" aria-hidden="true" tabindex="-1"></a>survival_her2_amplicon_tidy_erpos <span class="ot">&lt;-</span> survival_her2_amplicon_erpos <span class="sc">%&gt;%</span> </span>
<span id="cb19-2493"><a href="#cb19-2493" aria-hidden="true" tabindex="-1"></a>    broom<span class="sc">::</span><span class="fu">tidy</span>(., <span class="at">exponentiate =</span> <span class="cn">TRUE</span>, <span class="at">conf.int =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-2494"><a href="#cb19-2494" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(term, estimate, conf.low, conf.high, std.error, p.value) <span class="sc">%&gt;%</span></span>
<span id="cb19-2495"><a href="#cb19-2495" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">pathway =</span> term) </span>
<span id="cb19-2496"><a href="#cb19-2496" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2497"><a href="#cb19-2497" aria-hidden="true" tabindex="-1"></a>survival_her2_fpkm_tidy_erneg <span class="ot">&lt;-</span> survival_her2_fpkm_erneg <span class="sc">%&gt;%</span> </span>
<span id="cb19-2498"><a href="#cb19-2498" aria-hidden="true" tabindex="-1"></a>    broom<span class="sc">::</span><span class="fu">tidy</span>(., <span class="at">exponentiate =</span> <span class="cn">TRUE</span>, <span class="at">conf.int =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-2499"><a href="#cb19-2499" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(term, estimate, conf.low, conf.high, std.error, p.value) <span class="sc">%&gt;%</span></span>
<span id="cb19-2500"><a href="#cb19-2500" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">pathway =</span> term) </span>
<span id="cb19-2501"><a href="#cb19-2501" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2502"><a href="#cb19-2502" aria-hidden="true" tabindex="-1"></a>survival_her2_fpkm_tidy_erpos <span class="ot">&lt;-</span> survival_her2_fpkm_erpos <span class="sc">%&gt;%</span> </span>
<span id="cb19-2503"><a href="#cb19-2503" aria-hidden="true" tabindex="-1"></a>    broom<span class="sc">::</span><span class="fu">tidy</span>(., <span class="at">exponentiate =</span> <span class="cn">TRUE</span>, <span class="at">conf.int =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-2504"><a href="#cb19-2504" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(term, estimate, conf.low, conf.high, std.error, p.value) <span class="sc">%&gt;%</span></span>
<span id="cb19-2505"><a href="#cb19-2505" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">pathway =</span> term) </span>
<span id="cb19-2506"><a href="#cb19-2506" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2507"><a href="#cb19-2507" aria-hidden="true" tabindex="-1"></a>survival_her2_amplicon_tidy <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(</span>
<span id="cb19-2508"><a href="#cb19-2508" aria-hidden="true" tabindex="-1"></a>    <span class="fu">list</span>(</span>
<span id="cb19-2509"><a href="#cb19-2509" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Negative"</span> <span class="ot">=</span> survival_her2_amplicon_tidy_erneg, </span>
<span id="cb19-2510"><a href="#cb19-2510" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Positive"</span> <span class="ot">=</span> survival_her2_amplicon_tidy_erpos</span>
<span id="cb19-2511"><a href="#cb19-2511" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb19-2512"><a href="#cb19-2512" aria-hidden="true" tabindex="-1"></a>    <span class="at">.id =</span> <span class="st">"er_status"</span></span>
<span id="cb19-2513"><a href="#cb19-2513" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-2514"><a href="#cb19-2514" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2515"><a href="#cb19-2515" aria-hidden="true" tabindex="-1"></a>survival_her2_fpkm_tidy <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(</span>
<span id="cb19-2516"><a href="#cb19-2516" aria-hidden="true" tabindex="-1"></a>    <span class="fu">list</span>(</span>
<span id="cb19-2517"><a href="#cb19-2517" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Negative"</span> <span class="ot">=</span> survival_her2_fpkm_tidy_erneg, </span>
<span id="cb19-2518"><a href="#cb19-2518" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Positive"</span> <span class="ot">=</span> survival_her2_fpkm_tidy_erpos</span>
<span id="cb19-2519"><a href="#cb19-2519" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb19-2520"><a href="#cb19-2520" aria-hidden="true" tabindex="-1"></a>    <span class="at">.id =</span> <span class="st">"er_status"</span></span>
<span id="cb19-2521"><a href="#cb19-2521" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-2522"><a href="#cb19-2522" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb19-2523"><a href="#cb19-2523" aria-hidden="true" tabindex="-1"></a>survival_her2_amplicon_tidy <span class="sc">%&gt;%</span></span>
<span id="cb19-2524"><a href="#cb19-2524" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(dplyr<span class="sc">::</span><span class="fu">across</span>(</span>
<span id="cb19-2525"><a href="#cb19-2525" aria-hidden="true" tabindex="-1"></a>        <span class="at">.cols =</span> dplyr<span class="sc">::</span><span class="fu">all_of</span>(<span class="fu">setdiff</span>(<span class="fu">colnames</span>(.), <span class="fu">c</span>(<span class="st">"er_status"</span>))), </span>
<span id="cb19-2526"><a href="#cb19-2526" aria-hidden="true" tabindex="-1"></a>        <span class="at">.fns =</span> \(x) <span class="fu">format</span>(x, <span class="at">digits =</span> <span class="dv">3</span>)</span>
<span id="cb19-2527"><a href="#cb19-2527" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">%&gt;%</span></span>
<span id="cb19-2528"><a href="#cb19-2528" aria-hidden="true" tabindex="-1"></a>    DT<span class="sc">::</span><span class="fu">datatable</span>(<span class="at">options =</span> <span class="fu">list</span>(<span class="at">scrollX =</span> <span class="cn">TRUE</span>))</span>
<span id="cb19-2529"><a href="#cb19-2529" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-2530"><a href="#cb19-2530" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2531"><a href="#cb19-2531" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2532"><a href="#cb19-2532" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=8.5, fig.height=5}</span></span>
<span id="cb19-2533"><a href="#cb19-2533" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-her2-amplicon-rfs-scanb</span></span>
<span id="cb19-2534"><a href="#cb19-2534" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Hazard ratio and confidence interval for the HER2 AMPLICON </span></span>
<span id="cb19-2535"><a href="#cb19-2535" aria-hidden="true" tabindex="-1"></a><span class="co">#|      signature.</span></span>
<span id="cb19-2536"><a href="#cb19-2536" aria-hidden="true" tabindex="-1"></a>survival_her2_amplicon_tidy <span class="sc">%&gt;%</span></span>
<span id="cb19-2537"><a href="#cb19-2537" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(pathway <span class="sc">==</span> <span class="st">"HER2_AMPLICON"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-2538"><a href="#cb19-2538" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(</span>
<span id="cb19-2539"><a href="#cb19-2539" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> pathway, </span>
<span id="cb19-2540"><a href="#cb19-2540" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> estimate, </span>
<span id="cb19-2541"><a href="#cb19-2541" aria-hidden="true" tabindex="-1"></a>        <span class="at">xmin =</span> conf.low, </span>
<span id="cb19-2542"><a href="#cb19-2542" aria-hidden="true" tabindex="-1"></a>        <span class="at">xmax =</span> conf.high</span>
<span id="cb19-2543"><a href="#cb19-2543" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">+</span></span>
<span id="cb19-2544"><a href="#cb19-2544" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_pointrange</span>() <span class="sc">+</span> </span>
<span id="cb19-2545"><a href="#cb19-2545" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">1</span>, <span class="at">lty =</span> <span class="dv">2</span>) <span class="sc">+</span> </span>
<span id="cb19-2546"><a href="#cb19-2546" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb19-2547"><a href="#cb19-2547" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"Hazard ratio (95% CI)"</span>,</span>
<span id="cb19-2548"><a href="#cb19-2548" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">""</span></span>
<span id="cb19-2549"><a href="#cb19-2549" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb19-2550"><a href="#cb19-2550" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(</span>
<span id="cb19-2551"><a href="#cb19-2551" aria-hidden="true" tabindex="-1"></a>        <span class="sc">~</span>er_status, </span>
<span id="cb19-2552"><a href="#cb19-2552" aria-hidden="true" tabindex="-1"></a>        <span class="at">ncol =</span> <span class="dv">1</span>,</span>
<span id="cb19-2553"><a href="#cb19-2553" aria-hidden="true" tabindex="-1"></a>        <span class="at">labeller =</span> <span class="fu">as_labeller</span>(<span class="fu">c</span>(</span>
<span id="cb19-2554"><a href="#cb19-2554" aria-hidden="true" tabindex="-1"></a>            <span class="st">"Negative"</span> <span class="ot">=</span> <span class="st">"ER status: Negative"</span>,</span>
<span id="cb19-2555"><a href="#cb19-2555" aria-hidden="true" tabindex="-1"></a>            <span class="st">"Positive"</span> <span class="ot">=</span> <span class="st">"ER status: Positive"</span></span>
<span id="cb19-2556"><a href="#cb19-2556" aria-hidden="true" tabindex="-1"></a>        ))</span>
<span id="cb19-2557"><a href="#cb19-2557" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb19-2558"><a href="#cb19-2558" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb19-2559"><a href="#cb19-2559" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="fu">paste0</span>(</span>
<span id="cb19-2560"><a href="#cb19-2560" aria-hidden="true" tabindex="-1"></a>            <span class="st">"RFS analysis for SCAN-B patients "</span>,</span>
<span id="cb19-2561"><a href="#cb19-2561" aria-hidden="true" tabindex="-1"></a>            <span class="st">"</span><span class="sc">\n</span><span class="st">with HER2+ BC stratified by ER status"</span></span>
<span id="cb19-2562"><a href="#cb19-2562" aria-hidden="true" tabindex="-1"></a>        ),</span>
<span id="cb19-2563"><a href="#cb19-2563" aria-hidden="true" tabindex="-1"></a>        <span class="at">caption =</span> <span class="fu">paste0</span>(</span>
<span id="cb19-2564"><a href="#cb19-2564" aria-hidden="true" tabindex="-1"></a>            <span class="st">"For ER+ tumors, only patients that received"</span>, </span>
<span id="cb19-2565"><a href="#cb19-2565" aria-hidden="true" tabindex="-1"></a>            <span class="st">" Immu + Chemo + Endo were selected. #pts: 434, #events: 48</span><span class="sc">\n</span><span class="st">"</span>,</span>
<span id="cb19-2566"><a href="#cb19-2566" aria-hidden="true" tabindex="-1"></a>            <span class="st">"For ER- tumors, only patients that received"</span>, </span>
<span id="cb19-2567"><a href="#cb19-2567" aria-hidden="true" tabindex="-1"></a>            <span class="st">" Immu + Chemo were selected. #pts: 177, #events: 26"</span></span>
<span id="cb19-2568"><a href="#cb19-2568" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb19-2569"><a href="#cb19-2569" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-2570"><a href="#cb19-2570" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_pointrange</span>(<span class="at">size =</span> <span class="dv">2</span>, <span class="at">linewidth =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb19-2571"><a href="#cb19-2571" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">coord_cartesian</span>(<span class="at">xlim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fl">1.25</span>)) <span class="sc">+</span> </span>
<span id="cb19-2572"><a href="#cb19-2572" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>) <span class="sc">+</span> </span>
<span id="cb19-2573"><a href="#cb19-2573" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>() <span class="sc">+</span></span>
<span id="cb19-2574"><a href="#cb19-2574" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span></span>
<span id="cb19-2575"><a href="#cb19-2575" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme</span>(</span>
<span id="cb19-2576"><a href="#cb19-2576" aria-hidden="true" tabindex="-1"></a>        <span class="at">plot.caption =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>)</span>
<span id="cb19-2577"><a href="#cb19-2577" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb19-2578"><a href="#cb19-2578" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-2579"><a href="#cb19-2579" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2580"><a href="#cb19-2580" aria-hidden="true" tabindex="-1"></a>For both ER- and ER+ BC patients, the higher the signature the better</span>
<span id="cb19-2581"><a href="#cb19-2581" aria-hidden="true" tabindex="-1"></a>the outcome. The signature is noisier for ER+ tumors, I wonder if it is</span>
<span id="cb19-2582"><a href="#cb19-2582" aria-hidden="true" tabindex="-1"></a>due to treatment selection or the fact we assumed that all immunotherapies </span>
<span id="cb19-2583"><a href="#cb19-2583" aria-hidden="true" tabindex="-1"></a>are herceptin when in fact they are not.</span>
<span id="cb19-2584"><a href="#cb19-2584" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2585"><a href="#cb19-2585" aria-hidden="true" tabindex="-1"></a>Next we compare to using just HER2 FPKM as a measure. </span>
<span id="cb19-2586"><a href="#cb19-2586" aria-hidden="true" tabindex="-1"></a>@fig-her2-amplicon-fpkm-rfs-scanb shows that the HER2 amplicon improves</span>
<span id="cb19-2587"><a href="#cb19-2587" aria-hidden="true" tabindex="-1"></a>in the estimation of the hazard ratio for such a signature and how </span>
<span id="cb19-2588"><a href="#cb19-2588" aria-hidden="true" tabindex="-1"></a>including more genes might be important.</span>
<span id="cb19-2589"><a href="#cb19-2589" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2590"><a href="#cb19-2590" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=8.5, fig.height=5}</span></span>
<span id="cb19-2591"><a href="#cb19-2591" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-her2-amplicon-fpkm-rfs-scanb</span></span>
<span id="cb19-2592"><a href="#cb19-2592" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Hazard ratio and confidence interval for the HER2 AMPLICON </span></span>
<span id="cb19-2593"><a href="#cb19-2593" aria-hidden="true" tabindex="-1"></a><span class="co">#|      signature.</span></span>
<span id="cb19-2594"><a href="#cb19-2594" aria-hidden="true" tabindex="-1"></a>survival_her2_fpkm_tidy <span class="sc">%&gt;%</span></span>
<span id="cb19-2595"><a href="#cb19-2595" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(., survival_her2_amplicon_tidy) <span class="sc">%&gt;%</span></span>
<span id="cb19-2596"><a href="#cb19-2596" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(pathway <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"her2_fpkm"</span>, <span class="st">"HER2_AMPLICON"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb19-2597"><a href="#cb19-2597" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">pathway =</span> dplyr<span class="sc">::</span><span class="fu">case_when</span>(</span>
<span id="cb19-2598"><a href="#cb19-2598" aria-hidden="true" tabindex="-1"></a>        pathway <span class="sc">==</span> <span class="st">"her2_fpkm"</span> <span class="sc">~</span> <span class="st">"HER2 FPKM"</span>,</span>
<span id="cb19-2599"><a href="#cb19-2599" aria-hidden="true" tabindex="-1"></a>        pathway <span class="sc">==</span> <span class="st">"HER2_AMPLICON"</span> <span class="sc">~</span> <span class="st">"HER2 amplicon"</span>,</span>
<span id="cb19-2600"><a href="#cb19-2600" aria-hidden="true" tabindex="-1"></a>        <span class="cn">TRUE</span> <span class="sc">~</span> <span class="cn">NA</span></span>
<span id="cb19-2601"><a href="#cb19-2601" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">%&gt;%</span> </span>
<span id="cb19-2602"><a href="#cb19-2602" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(</span>
<span id="cb19-2603"><a href="#cb19-2603" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> pathway, </span>
<span id="cb19-2604"><a href="#cb19-2604" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> estimate, </span>
<span id="cb19-2605"><a href="#cb19-2605" aria-hidden="true" tabindex="-1"></a>        <span class="at">xmin =</span> conf.low, </span>
<span id="cb19-2606"><a href="#cb19-2606" aria-hidden="true" tabindex="-1"></a>        <span class="at">xmax =</span> conf.high</span>
<span id="cb19-2607"><a href="#cb19-2607" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">+</span></span>
<span id="cb19-2608"><a href="#cb19-2608" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_pointrange</span>() <span class="sc">+</span> </span>
<span id="cb19-2609"><a href="#cb19-2609" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">1</span>, <span class="at">lty =</span> <span class="dv">2</span>) <span class="sc">+</span> </span>
<span id="cb19-2610"><a href="#cb19-2610" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb19-2611"><a href="#cb19-2611" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"Hazard ratio (95% CI)"</span>,</span>
<span id="cb19-2612"><a href="#cb19-2612" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">""</span></span>
<span id="cb19-2613"><a href="#cb19-2613" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb19-2614"><a href="#cb19-2614" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(</span>
<span id="cb19-2615"><a href="#cb19-2615" aria-hidden="true" tabindex="-1"></a>        <span class="sc">~</span>er_status, </span>
<span id="cb19-2616"><a href="#cb19-2616" aria-hidden="true" tabindex="-1"></a>        <span class="at">ncol =</span> <span class="dv">1</span>,</span>
<span id="cb19-2617"><a href="#cb19-2617" aria-hidden="true" tabindex="-1"></a>        <span class="at">labeller =</span> <span class="fu">as_labeller</span>(<span class="fu">c</span>(</span>
<span id="cb19-2618"><a href="#cb19-2618" aria-hidden="true" tabindex="-1"></a>            <span class="st">"Negative"</span> <span class="ot">=</span> <span class="st">"ER status: Negative"</span>,</span>
<span id="cb19-2619"><a href="#cb19-2619" aria-hidden="true" tabindex="-1"></a>            <span class="st">"Positive"</span> <span class="ot">=</span> <span class="st">"ER status: Positive"</span></span>
<span id="cb19-2620"><a href="#cb19-2620" aria-hidden="true" tabindex="-1"></a>        ))</span>
<span id="cb19-2621"><a href="#cb19-2621" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb19-2622"><a href="#cb19-2622" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb19-2623"><a href="#cb19-2623" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="fu">paste0</span>(</span>
<span id="cb19-2624"><a href="#cb19-2624" aria-hidden="true" tabindex="-1"></a>            <span class="st">"RFS analysis for SCAN-B patients "</span>,</span>
<span id="cb19-2625"><a href="#cb19-2625" aria-hidden="true" tabindex="-1"></a>            <span class="st">"</span><span class="sc">\n</span><span class="st">with HER2+ BC stratified by ER status"</span></span>
<span id="cb19-2626"><a href="#cb19-2626" aria-hidden="true" tabindex="-1"></a>        ),</span>
<span id="cb19-2627"><a href="#cb19-2627" aria-hidden="true" tabindex="-1"></a>        <span class="at">caption =</span> <span class="fu">paste0</span>(</span>
<span id="cb19-2628"><a href="#cb19-2628" aria-hidden="true" tabindex="-1"></a>            <span class="st">"For ER+ tumors, only patients that received"</span>, </span>
<span id="cb19-2629"><a href="#cb19-2629" aria-hidden="true" tabindex="-1"></a>            <span class="st">" Immu + Chemo + Endo were selected. #pts: 434, #events: 48</span><span class="sc">\n</span><span class="st">"</span>,</span>
<span id="cb19-2630"><a href="#cb19-2630" aria-hidden="true" tabindex="-1"></a>            <span class="st">"For ER- tumors, only patients that received"</span>, </span>
<span id="cb19-2631"><a href="#cb19-2631" aria-hidden="true" tabindex="-1"></a>            <span class="st">" Immu + Chemo were selected. #pts: 177, #events: 26"</span></span>
<span id="cb19-2632"><a href="#cb19-2632" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb19-2633"><a href="#cb19-2633" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-2634"><a href="#cb19-2634" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_pointrange</span>(<span class="at">size =</span> <span class="dv">2</span>, <span class="at">linewidth =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb19-2635"><a href="#cb19-2635" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">coord_cartesian</span>(<span class="at">xlim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fl">1.25</span>)) <span class="sc">+</span> </span>
<span id="cb19-2636"><a href="#cb19-2636" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>) <span class="sc">+</span> </span>
<span id="cb19-2637"><a href="#cb19-2637" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>() <span class="sc">+</span></span>
<span id="cb19-2638"><a href="#cb19-2638" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span></span>
<span id="cb19-2639"><a href="#cb19-2639" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme</span>(</span>
<span id="cb19-2640"><a href="#cb19-2640" aria-hidden="true" tabindex="-1"></a>        <span class="at">plot.caption =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>)</span>
<span id="cb19-2641"><a href="#cb19-2641" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb19-2642"><a href="#cb19-2642" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-2643"><a href="#cb19-2643" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2644"><a href="#cb19-2644" aria-hidden="true" tabindex="-1"></a>These scores lead to similar hazard ratios and this is due to how </span>
<span id="cb19-2645"><a href="#cb19-2645" aria-hidden="true" tabindex="-1"></a>correlated they are.</span>
<span id="cb19-2646"><a href="#cb19-2646" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2647"><a href="#cb19-2647" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=8, fig.height=5}</span></span>
<span id="cb19-2648"><a href="#cb19-2648" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-corr-her2-fpkm-amplicon</span></span>
<span id="cb19-2649"><a href="#cb19-2649" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Correlation between HER2 FPKM and HER2 amplicon among HER2+</span></span>
<span id="cb19-2650"><a href="#cb19-2650" aria-hidden="true" tabindex="-1"></a><span class="co">#|     BC tumors</span></span>
<span id="cb19-2651"><a href="#cb19-2651" aria-hidden="true" tabindex="-1"></a>her2_df_pca_scanb_comp <span class="ot">&lt;-</span> df_pca <span class="sc">%&gt;%</span></span>
<span id="cb19-2652"><a href="#cb19-2652" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb19-2653"><a href="#cb19-2653" aria-hidden="true" tabindex="-1"></a>        HER2 <span class="sc">==</span> <span class="st">"Positive"</span></span>
<span id="cb19-2654"><a href="#cb19-2654" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb19-2655"><a href="#cb19-2655" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">inner_join</span>(</span>
<span id="cb19-2656"><a href="#cb19-2656" aria-hidden="true" tabindex="-1"></a>        .,</span>
<span id="cb19-2657"><a href="#cb19-2657" aria-hidden="true" tabindex="-1"></a>        scores_her2_sub_genes <span class="sc">%&gt;%</span></span>
<span id="cb19-2658"><a href="#cb19-2658" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">select</span>(her2_fpkm, sample_name),</span>
<span id="cb19-2659"><a href="#cb19-2659" aria-hidden="true" tabindex="-1"></a>        <span class="at">by =</span> <span class="st">"sample_name"</span></span>
<span id="cb19-2660"><a href="#cb19-2660" aria-hidden="true" tabindex="-1"></a>    ) </span>
<span id="cb19-2661"><a href="#cb19-2661" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2662"><a href="#cb19-2662" aria-hidden="true" tabindex="-1"></a>her2_df_pca_scanb_comp <span class="sc">%&gt;%</span> </span>
<span id="cb19-2663"><a href="#cb19-2663" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(</span>
<span id="cb19-2664"><a href="#cb19-2664" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> HER2_AMPLICON,</span>
<span id="cb19-2665"><a href="#cb19-2665" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> her2_fpkm,</span>
<span id="cb19-2666"><a href="#cb19-2666" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> er_status</span>
<span id="cb19-2667"><a href="#cb19-2667" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">+</span> </span>
<span id="cb19-2668"><a href="#cb19-2668" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span> </span>
<span id="cb19-2669"><a href="#cb19-2669" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_smooth</span>(</span>
<span id="cb19-2670"><a href="#cb19-2670" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">"black"</span>,</span>
<span id="cb19-2671"><a href="#cb19-2671" aria-hidden="true" tabindex="-1"></a>        <span class="at">method =</span> <span class="st">"gam"</span>,</span>
<span id="cb19-2672"><a href="#cb19-2672" aria-hidden="true" tabindex="-1"></a>        <span class="at">formula =</span> y <span class="sc">~</span> <span class="fu">s</span>(x, <span class="at">bs =</span> <span class="st">"cs"</span>)</span>
<span id="cb19-2673"><a href="#cb19-2673" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-2674"><a href="#cb19-2674" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb19-2675"><a href="#cb19-2675" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"HER2 amplicon"</span>,</span>
<span id="cb19-2676"><a href="#cb19-2676" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"HER2 FPKM"</span>,</span>
<span id="cb19-2677"><a href="#cb19-2677" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">"SCAN-B HER2+ BC only"</span>,</span>
<span id="cb19-2678"><a href="#cb19-2678" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">"ER status"</span></span>
<span id="cb19-2679"><a href="#cb19-2679" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb19-2680"><a href="#cb19-2680" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>) <span class="sc">+</span> </span>
<span id="cb19-2681"><a href="#cb19-2681" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>() <span class="sc">+</span> </span>
<span id="cb19-2682"><a href="#cb19-2682" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>()</span>
<span id="cb19-2683"><a href="#cb19-2683" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-2684"><a href="#cb19-2684" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2685"><a href="#cb19-2685" aria-hidden="true" tabindex="-1"></a><span class="fu">## ILC and position in the molecular landscape</span></span>
<span id="cb19-2686"><a href="#cb19-2686" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2687"><a href="#cb19-2687" aria-hidden="true" tabindex="-1"></a>Invasive lobular carcinoma is a subtype of ER+ BC. They tend to respond</span>
<span id="cb19-2688"><a href="#cb19-2688" aria-hidden="true" tabindex="-1"></a>better to ET in the short term but they recurr after 20 years. Here we</span>
<span id="cb19-2689"><a href="#cb19-2689" aria-hidden="true" tabindex="-1"></a>just show where the ILC BC lie in the molecular landscape.</span>
<span id="cb19-2690"><a href="#cb19-2690" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2691"><a href="#cb19-2691" aria-hidden="true" tabindex="-1"></a><span class="in">```{r fig.width=9, fig.height=6, message=FALSE}</span></span>
<span id="cb19-2692"><a href="#cb19-2692" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-pca-lobular-scanb</span></span>
<span id="cb19-2693"><a href="#cb19-2693" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Embedding of the lobular samples from SCANB on top </span></span>
<span id="cb19-2694"><a href="#cb19-2694" aria-hidden="true" tabindex="-1"></a><span class="co">#|     METABRIC and TCGA</span></span>
<span id="cb19-2695"><a href="#cb19-2695" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">get_base_plot</span>(df_pca <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"tcga"</span>, <span class="st">"metabric"</span>)))</span>
<span id="cb19-2696"><a href="#cb19-2696" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2697"><a href="#cb19-2697" aria-hidden="true" tabindex="-1"></a>p <span class="sc">+</span></span>
<span id="cb19-2698"><a href="#cb19-2698" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>(</span>
<span id="cb19-2699"><a href="#cb19-2699" aria-hidden="true" tabindex="-1"></a>        df_pca <span class="sc">%&gt;%</span></span>
<span id="cb19-2700"><a href="#cb19-2700" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"scanb"</span> <span class="sc">&amp;</span> InvCa.type <span class="sc">==</span> <span class="st">"Lobular"</span>),</span>
<span id="cb19-2701"><a href="#cb19-2701" aria-hidden="true" tabindex="-1"></a>        <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> PC3, <span class="at">y =</span> PC4),</span>
<span id="cb19-2702"><a href="#cb19-2702" aria-hidden="true" tabindex="-1"></a>        <span class="at">size =</span> <span class="dv">2</span></span>
<span id="cb19-2703"><a href="#cb19-2703" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-2704"><a href="#cb19-2704" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">15</span>) <span class="sc">+</span></span>
<span id="cb19-2705"><a href="#cb19-2705" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb19-2706"><a href="#cb19-2706" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">"Lobular samples from SCANB</span><span class="sc">\n</span><span class="st">on top of METABRIC and TCGA samples"</span></span>
<span id="cb19-2707"><a href="#cb19-2707" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-2708"><a href="#cb19-2708" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_color_manual</span>(</span>
<span id="cb19-2709"><a href="#cb19-2709" aria-hidden="true" tabindex="-1"></a>        <span class="at">values =</span> <span class="fu">c</span>(<span class="fu">get_colors_pam50</span>(p<span class="sc">$</span>data))</span>
<span id="cb19-2710"><a href="#cb19-2710" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb19-2711"><a href="#cb19-2711" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span></span>
<span id="cb19-2712"><a href="#cb19-2712" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>(<span class="at">shape =</span> <span class="cn">FALSE</span>)</span>
<span id="cb19-2713"><a href="#cb19-2713" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-2714"><a href="#cb19-2714" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2715"><a href="#cb19-2715" aria-hidden="true" tabindex="-1"></a>We see that they are mostly in the normal and luminal A region. Still</span>
<span id="cb19-2716"><a href="#cb19-2716" aria-hidden="true" tabindex="-1"></a>there are several samples that are luminal B, meaning they are more</span>
<span id="cb19-2717"><a href="#cb19-2717" aria-hidden="true" tabindex="-1"></a>proliferative.</span>
<span id="cb19-2718"><a href="#cb19-2718" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2719"><a href="#cb19-2719" aria-hidden="true" tabindex="-1"></a>Is there any difference in the SET ER/PR for lobulars when comparing</span>
<span id="cb19-2720"><a href="#cb19-2720" aria-hidden="true" tabindex="-1"></a>to ER+ HER2- Ductal carcinomas? </span>
<span id="cb19-2721"><a href="#cb19-2721" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2724"><a href="#cb19-2724" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-2725"><a href="#cb19-2725" aria-hidden="true" tabindex="-1"></a>df_pca <span class="ot">&lt;-</span> df_pca <span class="sc">%&gt;%</span></span>
<span id="cb19-2726"><a href="#cb19-2726" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">inner_join</span>(</span>
<span id="cb19-2727"><a href="#cb19-2727" aria-hidden="true" tabindex="-1"></a>        .,</span>
<span id="cb19-2728"><a href="#cb19-2728" aria-hidden="true" tabindex="-1"></a>        scores_her2,</span>
<span id="cb19-2729"><a href="#cb19-2729" aria-hidden="true" tabindex="-1"></a>        <span class="at">by =</span> <span class="st">"sample_name"</span></span>
<span id="cb19-2730"><a href="#cb19-2730" aria-hidden="true" tabindex="-1"></a>    ) </span>
<span id="cb19-2731"><a href="#cb19-2731" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2732"><a href="#cb19-2732" aria-hidden="true" tabindex="-1"></a>coefs_risk_model <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(</span>
<span id="cb19-2733"><a href="#cb19-2733" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../../results/rds_files/risk_score/coefs_risk_model.rds"</span></span>
<span id="cb19-2734"><a href="#cb19-2734" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-2735"><a href="#cb19-2735" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2736"><a href="#cb19-2736" aria-hidden="true" tabindex="-1"></a><span class="co"># now we can calculate the scores for all the patients. For TCGA and POETIC</span></span>
<span id="cb19-2737"><a href="#cb19-2737" aria-hidden="true" tabindex="-1"></a><span class="co"># we do not have tumor size, so for those we calculate the risk score based</span></span>
<span id="cb19-2738"><a href="#cb19-2738" aria-hidden="true" tabindex="-1"></a><span class="co"># on the components only.</span></span>
<span id="cb19-2739"><a href="#cb19-2739" aria-hidden="true" tabindex="-1"></a>df_pca_scanb <span class="ot">&lt;-</span> df_pca <span class="sc">%&gt;%</span> </span>
<span id="cb19-2740"><a href="#cb19-2740" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"scanb"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-2741"><a href="#cb19-2741" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">node_status =</span> <span class="fu">ifelse</span>(</span>
<span id="cb19-2742"><a href="#cb19-2742" aria-hidden="true" tabindex="-1"></a>        node_group <span class="sc">==</span> <span class="st">"N0"</span>, </span>
<span id="cb19-2743"><a href="#cb19-2743" aria-hidden="true" tabindex="-1"></a>        <span class="st">"neg"</span>,</span>
<span id="cb19-2744"><a href="#cb19-2744" aria-hidden="true" tabindex="-1"></a>        <span class="st">"pos"</span></span>
<span id="cb19-2745"><a href="#cb19-2745" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">%&gt;%</span> </span>
<span id="cb19-2746"><a href="#cb19-2746" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">node_status =</span> <span class="fu">factor</span>(</span>
<span id="cb19-2747"><a href="#cb19-2747" aria-hidden="true" tabindex="-1"></a>        node_status, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"neg"</span>, <span class="st">"pos"</span>)</span>
<span id="cb19-2748"><a href="#cb19-2748" aria-hidden="true" tabindex="-1"></a>    ))</span>
<span id="cb19-2749"><a href="#cb19-2749" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2750"><a href="#cb19-2750" aria-hidden="true" tabindex="-1"></a>df_pca_scanb<span class="sc">$</span>risk_score <span class="ot">&lt;-</span> <span class="fu">calculate_risk_score</span>(</span>
<span id="cb19-2751"><a href="#cb19-2751" aria-hidden="true" tabindex="-1"></a>    coefs_risk_model, df_pca_scanb</span>
<span id="cb19-2752"><a href="#cb19-2752" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-2753"><a href="#cb19-2753" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2754"><a href="#cb19-2754" aria-hidden="true" tabindex="-1"></a>df_pca_scanb <span class="sc">%&gt;%</span></span>
<span id="cb19-2755"><a href="#cb19-2755" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb19-2756"><a href="#cb19-2756" aria-hidden="true" tabindex="-1"></a>        InvCa.type <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Lobular"</span>, <span class="st">"Ductal"</span>) <span class="sc">&amp;</span> </span>
<span id="cb19-2757"><a href="#cb19-2757" aria-hidden="true" tabindex="-1"></a>        er_status <span class="sc">==</span> <span class="st">"pos"</span> <span class="sc">&amp;</span> </span>
<span id="cb19-2758"><a href="#cb19-2758" aria-hidden="true" tabindex="-1"></a>        HER2 <span class="sc">==</span> <span class="st">"Negative"</span> <span class="sc">&amp;</span> </span>
<span id="cb19-2759"><a href="#cb19-2759" aria-hidden="true" tabindex="-1"></a>        <span class="sc">!</span><span class="fu">is.na</span>(risk_score)</span>
<span id="cb19-2760"><a href="#cb19-2760" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb19-2761"><a href="#cb19-2761" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> InvCa.type, <span class="at">y =</span> <span class="st">`</span><span class="at">SET ER/PR</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb19-2762"><a href="#cb19-2762" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_jitter</span>(<span class="at">size =</span> <span class="fl">0.4</span>, <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">color =</span> pam50)) <span class="sc">+</span> </span>
<span id="cb19-2763"><a href="#cb19-2763" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_violin</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb19-2764"><a href="#cb19-2764" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb19-2765"><a href="#cb19-2765" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">""</span>, </span>
<span id="cb19-2766"><a href="#cb19-2766" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">"SCANB, ER+ HER2- BC"</span></span>
<span id="cb19-2767"><a href="#cb19-2767" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-2768"><a href="#cb19-2768" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>) <span class="sc">+</span> </span>
<span id="cb19-2769"><a href="#cb19-2769" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span></span>
<span id="cb19-2770"><a href="#cb19-2770" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb19-2771"><a href="#cb19-2771" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-2772"><a href="#cb19-2772" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2773"><a href="#cb19-2773" aria-hidden="true" tabindex="-1"></a>They seem to be the same, also when checking the risk score calculated </span>
<span id="cb19-2774"><a href="#cb19-2774" aria-hidden="true" tabindex="-1"></a>for all patients, the lobular patients don't seem to have a decreased </span>
<span id="cb19-2775"><a href="#cb19-2775" aria-hidden="true" tabindex="-1"></a>risk overall, it is mostly because they are in the same region </span>
<span id="cb19-2776"><a href="#cb19-2776" aria-hidden="true" tabindex="-1"></a>of the molecular landscape. Interestingly there is way less</span>
<span id="cb19-2777"><a href="#cb19-2777" aria-hidden="true" tabindex="-1"></a>luminal B patients in the lobular, but still their distributions</span>
<span id="cb19-2778"><a href="#cb19-2778" aria-hidden="true" tabindex="-1"></a>are very similar.</span>
<span id="cb19-2779"><a href="#cb19-2779" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2782"><a href="#cb19-2782" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-2783"><a href="#cb19-2783" aria-hidden="true" tabindex="-1"></a>df_pca_scanb <span class="sc">%&gt;%</span></span>
<span id="cb19-2784"><a href="#cb19-2784" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb19-2785"><a href="#cb19-2785" aria-hidden="true" tabindex="-1"></a>        InvCa.type <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Lobular"</span>, <span class="st">"Ductal"</span>) <span class="sc">&amp;</span> </span>
<span id="cb19-2786"><a href="#cb19-2786" aria-hidden="true" tabindex="-1"></a>        er_status <span class="sc">==</span> <span class="st">"pos"</span> <span class="sc">&amp;</span> </span>
<span id="cb19-2787"><a href="#cb19-2787" aria-hidden="true" tabindex="-1"></a>        HER2 <span class="sc">==</span> <span class="st">"Negative"</span> <span class="sc">&amp;</span> </span>
<span id="cb19-2788"><a href="#cb19-2788" aria-hidden="true" tabindex="-1"></a>        <span class="sc">!</span><span class="fu">is.na</span>(risk_score)</span>
<span id="cb19-2789"><a href="#cb19-2789" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb19-2790"><a href="#cb19-2790" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> InvCa.type, <span class="at">y =</span> risk_score)) <span class="sc">+</span></span>
<span id="cb19-2791"><a href="#cb19-2791" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_jitter</span>(<span class="at">size =</span> <span class="fl">0.4</span>, <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">color =</span> pam50)) <span class="sc">+</span> </span>
<span id="cb19-2792"><a href="#cb19-2792" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_violin</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb19-2793"><a href="#cb19-2793" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb19-2794"><a href="#cb19-2794" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">""</span>, </span>
<span id="cb19-2795"><a href="#cb19-2795" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">"SCANB, ER+ HER2- BC"</span>,</span>
<span id="cb19-2796"><a href="#cb19-2796" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"Risk score"</span></span>
<span id="cb19-2797"><a href="#cb19-2797" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-2798"><a href="#cb19-2798" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>) <span class="sc">+</span> </span>
<span id="cb19-2799"><a href="#cb19-2799" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span></span>
<span id="cb19-2800"><a href="#cb19-2800" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb19-2801"><a href="#cb19-2801" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-2802"><a href="#cb19-2802" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2803"><a href="#cb19-2803" aria-hidden="true" tabindex="-1"></a>And we also plot by intrinsic molecular subtype. </span>
<span id="cb19-2804"><a href="#cb19-2804" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2805"><a href="#cb19-2805" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.height = 5, fig.width = 14}</span></span>
<span id="cb19-2806"><a href="#cb19-2806" aria-hidden="true" tabindex="-1"></a>df_pca_scanb <span class="sc">%&gt;%</span></span>
<span id="cb19-2807"><a href="#cb19-2807" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb19-2808"><a href="#cb19-2808" aria-hidden="true" tabindex="-1"></a>        InvCa.type <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Lobular"</span>, <span class="st">"Ductal"</span>) <span class="sc">&amp;</span> </span>
<span id="cb19-2809"><a href="#cb19-2809" aria-hidden="true" tabindex="-1"></a>        er_status <span class="sc">==</span> <span class="st">"pos"</span> <span class="sc">&amp;</span> </span>
<span id="cb19-2810"><a href="#cb19-2810" aria-hidden="true" tabindex="-1"></a>        HER2 <span class="sc">==</span> <span class="st">"Negative"</span> <span class="sc">&amp;</span> </span>
<span id="cb19-2811"><a href="#cb19-2811" aria-hidden="true" tabindex="-1"></a>        <span class="sc">!</span><span class="fu">is.na</span>(risk_score) <span class="sc">&amp;</span> </span>
<span id="cb19-2812"><a href="#cb19-2812" aria-hidden="true" tabindex="-1"></a>        risk_score <span class="sc">&lt;</span> <span class="dv">3</span></span>
<span id="cb19-2813"><a href="#cb19-2813" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb19-2814"><a href="#cb19-2814" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> InvCa.type, <span class="at">y =</span> risk_score)) <span class="sc">+</span></span>
<span id="cb19-2815"><a href="#cb19-2815" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_jitter</span>(<span class="at">size =</span> <span class="fl">0.4</span>) <span class="sc">+</span> </span>
<span id="cb19-2816"><a href="#cb19-2816" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_violin</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb19-2817"><a href="#cb19-2817" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb19-2818"><a href="#cb19-2818" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">""</span>, </span>
<span id="cb19-2819"><a href="#cb19-2819" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">"SCANB, ER+ HER2- BC"</span>,</span>
<span id="cb19-2820"><a href="#cb19-2820" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"Risk score"</span></span>
<span id="cb19-2821"><a href="#cb19-2821" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-2822"><a href="#cb19-2822" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>) <span class="sc">+</span> </span>
<span id="cb19-2823"><a href="#cb19-2823" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>pam50, <span class="at">nrow =</span> <span class="dv">1</span>) <span class="sc">+</span> </span>
<span id="cb19-2824"><a href="#cb19-2824" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span></span>
<span id="cb19-2825"><a href="#cb19-2825" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb19-2826"><a href="#cb19-2826" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-2827"><a href="#cb19-2827" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2828"><a href="#cb19-2828" aria-hidden="true" tabindex="-1"></a>We also compare EMT.</span>
<span id="cb19-2829"><a href="#cb19-2829" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2832"><a href="#cb19-2832" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-2833"><a href="#cb19-2833" aria-hidden="true" tabindex="-1"></a>df_pca_scanb <span class="sc">%&gt;%</span></span>
<span id="cb19-2834"><a href="#cb19-2834" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb19-2835"><a href="#cb19-2835" aria-hidden="true" tabindex="-1"></a>        InvCa.type <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Lobular"</span>, <span class="st">"Ductal"</span>) <span class="sc">&amp;</span> </span>
<span id="cb19-2836"><a href="#cb19-2836" aria-hidden="true" tabindex="-1"></a>        er_status <span class="sc">==</span> <span class="st">"pos"</span> <span class="sc">&amp;</span> </span>
<span id="cb19-2837"><a href="#cb19-2837" aria-hidden="true" tabindex="-1"></a>        HER2 <span class="sc">==</span> <span class="st">"Negative"</span></span>
<span id="cb19-2838"><a href="#cb19-2838" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb19-2839"><a href="#cb19-2839" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> InvCa.type, <span class="at">y =</span> <span class="st">`</span><span class="at">EMT</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb19-2840"><a href="#cb19-2840" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_jitter</span>(<span class="at">size =</span> <span class="fl">0.4</span>, <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">color =</span> pam50)) <span class="sc">+</span> </span>
<span id="cb19-2841"><a href="#cb19-2841" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_violin</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb19-2842"><a href="#cb19-2842" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">x =</span> <span class="st">""</span>, <span class="at">title =</span> <span class="st">"SCANB, ER+ HER2- BC"</span>) <span class="sc">+</span> </span>
<span id="cb19-2843"><a href="#cb19-2843" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>) <span class="sc">+</span> </span>
<span id="cb19-2844"><a href="#cb19-2844" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span></span>
<span id="cb19-2845"><a href="#cb19-2845" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb19-2846"><a href="#cb19-2846" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-2847"><a href="#cb19-2847" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2848"><a href="#cb19-2848" aria-hidden="true" tabindex="-1"></a>EMT is very similar across both histological subtypes. The plot </span>
<span id="cb19-2849"><a href="#cb19-2849" aria-hidden="true" tabindex="-1"></a>below shows the densities for ER percentage, notice how the two </span>
<span id="cb19-2850"><a href="#cb19-2850" aria-hidden="true" tabindex="-1"></a>overlap.</span>
<span id="cb19-2851"><a href="#cb19-2851" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2854"><a href="#cb19-2854" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-2855"><a href="#cb19-2855" aria-hidden="true" tabindex="-1"></a>df_pca_scanb <span class="sc">%&gt;%</span></span>
<span id="cb19-2856"><a href="#cb19-2856" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb19-2857"><a href="#cb19-2857" aria-hidden="true" tabindex="-1"></a>        InvCa.type <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Lobular"</span>, <span class="st">"Ductal"</span>) <span class="sc">&amp;</span> </span>
<span id="cb19-2858"><a href="#cb19-2858" aria-hidden="true" tabindex="-1"></a>        er_status <span class="sc">==</span> <span class="st">"pos"</span> <span class="sc">&amp;</span> </span>
<span id="cb19-2859"><a href="#cb19-2859" aria-hidden="true" tabindex="-1"></a>        HER2 <span class="sc">==</span> <span class="st">"Negative"</span> <span class="sc">&amp;</span> </span>
<span id="cb19-2860"><a href="#cb19-2860" aria-hidden="true" tabindex="-1"></a>        <span class="sc">!</span><span class="fu">is.na</span>(risk_score)</span>
<span id="cb19-2861"><a href="#cb19-2861" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb19-2862"><a href="#cb19-2862" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> ER.pct, <span class="at">fill =</span> InvCa.type)) <span class="sc">+</span></span>
<span id="cb19-2863"><a href="#cb19-2863" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_density</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span> </span>
<span id="cb19-2864"><a href="#cb19-2864" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb19-2865"><a href="#cb19-2865" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"Density"</span>,</span>
<span id="cb19-2866"><a href="#cb19-2866" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">"SCANB, ER+ HER2- BC"</span>,</span>
<span id="cb19-2867"><a href="#cb19-2867" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"ER percentage"</span>,</span>
<span id="cb19-2868"><a href="#cb19-2868" aria-hidden="true" tabindex="-1"></a>        <span class="at">fill =</span> <span class="st">"Histology"</span></span>
<span id="cb19-2869"><a href="#cb19-2869" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-2870"><a href="#cb19-2870" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>) <span class="sc">+</span> </span>
<span id="cb19-2871"><a href="#cb19-2871" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span></span>
<span id="cb19-2872"><a href="#cb19-2872" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb19-2873"><a href="#cb19-2873" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-2874"><a href="#cb19-2874" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2875"><a href="#cb19-2875" aria-hidden="true" tabindex="-1"></a>We now check the lobular signature LobSig that is touted to be superior</span>
<span id="cb19-2876"><a href="#cb19-2876" aria-hidden="true" tabindex="-1"></a>to OncotypeDX, Prosigna's PAM50 and other signatures. In total, 63%</span>
<span id="cb19-2877"><a href="#cb19-2877" aria-hidden="true" tabindex="-1"></a>of the genes are available here. Based on my experience this is </span>
<span id="cb19-2878"><a href="#cb19-2878" aria-hidden="true" tabindex="-1"></a>already a good number to make comparisons. For example, G2M checkpoint has</span>
<span id="cb19-2879"><a href="#cb19-2879" aria-hidden="true" tabindex="-1"></a>71% of the genes available, estrogen response early has 66%.</span>
<span id="cb19-2880"><a href="#cb19-2880" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2881"><a href="#cb19-2881" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=6}</span></span>
<span id="cb19-2882"><a href="#cb19-2882" aria-hidden="true" tabindex="-1"></a>lobsig <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"../../data/lobular_signature.txt"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-2883"><a href="#cb19-2883" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">colnames&lt;-</span><span class="st">`</span>(<span class="st">"genes"</span>)</span>
<span id="cb19-2884"><a href="#cb19-2884" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2885"><a href="#cb19-2885" aria-hidden="true" tabindex="-1"></a>lobsig_available <span class="ot">&lt;-</span> <span class="fu">intersect</span>(</span>
<span id="cb19-2886"><a href="#cb19-2886" aria-hidden="true" tabindex="-1"></a>    lobsig<span class="sc">$</span>genes,</span>
<span id="cb19-2887"><a href="#cb19-2887" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames</span>(df_pcs_regressed)</span>
<span id="cb19-2888"><a href="#cb19-2888" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-2889"><a href="#cb19-2889" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2890"><a href="#cb19-2890" aria-hidden="true" tabindex="-1"></a>df_pca_scanb<span class="sc">$</span>LobSig <span class="ot">&lt;-</span> <span class="fu">colSums</span>(</span>
<span id="cb19-2891"><a href="#cb19-2891" aria-hidden="true" tabindex="-1"></a>    df_pcs_regressed[lobsig_available, df_pca_scanb<span class="sc">$</span>sample_name]</span>
<span id="cb19-2892"><a href="#cb19-2892" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-2893"><a href="#cb19-2893" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2894"><a href="#cb19-2894" aria-hidden="true" tabindex="-1"></a>col_to_compare <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb19-2895"><a href="#cb19-2895" aria-hidden="true" tabindex="-1"></a>    <span class="st">"G2M"</span>, <span class="st">"EMT"</span>, <span class="st">"ER early"</span>, </span>
<span id="cb19-2896"><a href="#cb19-2896" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Random 200"</span>, <span class="st">"risk_score"</span>, <span class="st">"NCN.ROR.asT0"</span></span>
<span id="cb19-2897"><a href="#cb19-2897" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-2898"><a href="#cb19-2898" aria-hidden="true" tabindex="-1"></a>df_pca_scanb <span class="sc">%&gt;%</span></span>
<span id="cb19-2899"><a href="#cb19-2899" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(</span>
<span id="cb19-2900"><a href="#cb19-2900" aria-hidden="true" tabindex="-1"></a>        <span class="at">cols =</span> dplyr<span class="sc">::</span><span class="fu">all_of</span>(col_to_compare),</span>
<span id="cb19-2901"><a href="#cb19-2901" aria-hidden="true" tabindex="-1"></a>        <span class="at">names_to =</span> <span class="st">"pathway"</span>,</span>
<span id="cb19-2902"><a href="#cb19-2902" aria-hidden="true" tabindex="-1"></a>        <span class="at">values_to =</span> <span class="st">"score"</span> </span>
<span id="cb19-2903"><a href="#cb19-2903" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb19-2904"><a href="#cb19-2904" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb19-2905"><a href="#cb19-2905" aria-hidden="true" tabindex="-1"></a>        InvCa.type <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Lobular"</span>) <span class="sc">&amp;</span> </span>
<span id="cb19-2906"><a href="#cb19-2906" aria-hidden="true" tabindex="-1"></a>            <span class="sc">!</span><span class="fu">is.na</span>(score)</span>
<span id="cb19-2907"><a href="#cb19-2907" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb19-2908"><a href="#cb19-2908" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">pathway =</span> <span class="fu">ifelse</span>(</span>
<span id="cb19-2909"><a href="#cb19-2909" aria-hidden="true" tabindex="-1"></a>        pathway <span class="sc">==</span> <span class="st">"risk_score"</span>,</span>
<span id="cb19-2910"><a href="#cb19-2910" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Risk Score"</span>,</span>
<span id="cb19-2911"><a href="#cb19-2911" aria-hidden="true" tabindex="-1"></a>        pathway</span>
<span id="cb19-2912"><a href="#cb19-2912" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">%&gt;%</span></span>
<span id="cb19-2913"><a href="#cb19-2913" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(</span>
<span id="cb19-2914"><a href="#cb19-2914" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> LobSig, </span>
<span id="cb19-2915"><a href="#cb19-2915" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> score,</span>
<span id="cb19-2916"><a href="#cb19-2916" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> pam50</span>
<span id="cb19-2917"><a href="#cb19-2917" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">+</span> </span>
<span id="cb19-2918"><a href="#cb19-2918" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb19-2919"><a href="#cb19-2919" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">formula =</span> <span class="st">"y~x"</span>) <span class="sc">+</span></span>
<span id="cb19-2920"><a href="#cb19-2920" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>pathway, <span class="at">scales =</span> <span class="st">"free_y"</span>) <span class="sc">+</span></span>
<span id="cb19-2921"><a href="#cb19-2921" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_color_manual</span>(</span>
<span id="cb19-2922"><a href="#cb19-2922" aria-hidden="true" tabindex="-1"></a>        <span class="at">values =</span> <span class="fu">get_colors_pam50</span>(df_pca_scanb)</span>
<span id="cb19-2923"><a href="#cb19-2923" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-2924"><a href="#cb19-2924" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb19-2925"><a href="#cb19-2925" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"LobSig"</span>, </span>
<span id="cb19-2926"><a href="#cb19-2926" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"Score of signatures"</span>,</span>
<span id="cb19-2927"><a href="#cb19-2927" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">"All samples classified with lobular BC from SCANB"</span></span>
<span id="cb19-2928"><a href="#cb19-2928" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-2929"><a href="#cb19-2929" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">15</span>) <span class="sc">+</span> </span>
<span id="cb19-2930"><a href="#cb19-2930" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span> </span>
<span id="cb19-2931"><a href="#cb19-2931" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb19-2932"><a href="#cb19-2932" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-2933"><a href="#cb19-2933" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2934"><a href="#cb19-2934" aria-hidden="true" tabindex="-1"></a>We see that the LobSig is highgly correlated with G2M checkpoint and also</span>
<span id="cb19-2935"><a href="#cb19-2935" aria-hidden="true" tabindex="-1"></a>the risk of recurrence score obtained from PAM50. The random 200 is a signature</span>
<span id="cb19-2936"><a href="#cb19-2936" aria-hidden="true" tabindex="-1"></a>of 200 random genes that works as a negative control. </span>
<span id="cb19-2937"><a href="#cb19-2937" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2938"><a href="#cb19-2938" aria-hidden="true" tabindex="-1"></a>The figure below shows the distinction of the signature among the 3 different</span>
<span id="cb19-2939"><a href="#cb19-2939" aria-hidden="true" tabindex="-1"></a>risk categories from ROR.</span>
<span id="cb19-2940"><a href="#cb19-2940" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2943"><a href="#cb19-2943" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-2944"><a href="#cb19-2944" aria-hidden="true" tabindex="-1"></a>df_pca_scanb <span class="sc">%&gt;%</span></span>
<span id="cb19-2945"><a href="#cb19-2945" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(InvCa.type <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Lobular"</span>) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(NCN.ROR.risk.cat)) <span class="sc">%&gt;%</span></span>
<span id="cb19-2946"><a href="#cb19-2946" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">NCN.ROR.risk.cat =</span> <span class="fu">factor</span>(</span>
<span id="cb19-2947"><a href="#cb19-2947" aria-hidden="true" tabindex="-1"></a>        NCN.ROR.risk.cat,</span>
<span id="cb19-2948"><a href="#cb19-2948" aria-hidden="true" tabindex="-1"></a>        <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Low"</span>, <span class="st">"Intermediate"</span>, <span class="st">"High"</span>)</span>
<span id="cb19-2949"><a href="#cb19-2949" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">%&gt;%</span></span>
<span id="cb19-2950"><a href="#cb19-2950" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(</span>
<span id="cb19-2951"><a href="#cb19-2951" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> LobSig, </span>
<span id="cb19-2952"><a href="#cb19-2952" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> NCN.ROR.risk.cat,</span>
<span id="cb19-2953"><a href="#cb19-2953" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> pam50</span>
<span id="cb19-2954"><a href="#cb19-2954" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">+</span> </span>
<span id="cb19-2955"><a href="#cb19-2955" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_jitter</span>() <span class="sc">+</span> </span>
<span id="cb19-2956"><a href="#cb19-2956" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_boxplot</span>(<span class="at">alpha =</span> <span class="fl">0.4</span>, <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">outlier.shape =</span> <span class="cn">NA</span>) <span class="sc">+</span></span>
<span id="cb19-2957"><a href="#cb19-2957" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_color_manual</span>(</span>
<span id="cb19-2958"><a href="#cb19-2958" aria-hidden="true" tabindex="-1"></a>        <span class="at">values =</span> <span class="fu">get_colors_pam50</span>(df_pca_scanb)</span>
<span id="cb19-2959"><a href="#cb19-2959" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-2960"><a href="#cb19-2960" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb19-2961"><a href="#cb19-2961" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"NCN ROR risk category"</span>,</span>
<span id="cb19-2962"><a href="#cb19-2962" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">"PAM50"</span></span>
<span id="cb19-2963"><a href="#cb19-2963" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-2964"><a href="#cb19-2964" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>) <span class="sc">+</span></span>
<span id="cb19-2965"><a href="#cb19-2965" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span> </span>
<span id="cb19-2966"><a href="#cb19-2966" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb19-2967"><a href="#cb19-2967" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-2968"><a href="#cb19-2968" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2969"><a href="#cb19-2969" aria-hidden="true" tabindex="-1"></a>There is an increase in the score as expected based on the previous results</span>
<span id="cb19-2970"><a href="#cb19-2970" aria-hidden="true" tabindex="-1"></a>that show where the intermediate risk group is according to the molecular</span>
<span id="cb19-2971"><a href="#cb19-2971" aria-hidden="true" tabindex="-1"></a>landscape. </span>
<span id="cb19-2972"><a href="#cb19-2972" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2973"><a href="#cb19-2973" aria-hidden="true" tabindex="-1"></a>Lastly we check the CDH1 expression levels on the regressed data to </span>
<span id="cb19-2974"><a href="#cb19-2974" aria-hidden="true" tabindex="-1"></a>see if the differences are still captured upon the whole process. </span>
<span id="cb19-2975"><a href="#cb19-2975" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-2976"><a href="#cb19-2976" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.height = 5, fig.width = 14}</span></span>
<span id="cb19-2977"><a href="#cb19-2977" aria-hidden="true" tabindex="-1"></a>genes_of_interest <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"CDH1"</span>, <span class="st">"ESR1"</span>, <span class="st">"LOXL1"</span>)</span>
<span id="cb19-2978"><a href="#cb19-2978" aria-hidden="true" tabindex="-1"></a>df_pcs_regressed[genes_of_interest, ] <span class="sc">%&gt;%</span></span>
<span id="cb19-2979"><a href="#cb19-2979" aria-hidden="true" tabindex="-1"></a>    t <span class="sc">%&gt;%</span></span>
<span id="cb19-2980"><a href="#cb19-2980" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span></span>
<span id="cb19-2981"><a href="#cb19-2981" aria-hidden="true" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"sample_name"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-2982"><a href="#cb19-2982" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">inner_join</span>(</span>
<span id="cb19-2983"><a href="#cb19-2983" aria-hidden="true" tabindex="-1"></a>        .,</span>
<span id="cb19-2984"><a href="#cb19-2984" aria-hidden="true" tabindex="-1"></a>        df_pca_scanb,</span>
<span id="cb19-2985"><a href="#cb19-2985" aria-hidden="true" tabindex="-1"></a>        <span class="at">by =</span> <span class="st">"sample_name"</span></span>
<span id="cb19-2986"><a href="#cb19-2986" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb19-2987"><a href="#cb19-2987" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb19-2988"><a href="#cb19-2988" aria-hidden="true" tabindex="-1"></a>        InvCa.type <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Lobular"</span>, <span class="st">"Ductal"</span>) <span class="sc">&amp;</span> </span>
<span id="cb19-2989"><a href="#cb19-2989" aria-hidden="true" tabindex="-1"></a>        er_status <span class="sc">==</span> <span class="st">"pos"</span> <span class="sc">&amp;</span> </span>
<span id="cb19-2990"><a href="#cb19-2990" aria-hidden="true" tabindex="-1"></a>        HER2 <span class="sc">==</span> <span class="st">"Negative"</span></span>
<span id="cb19-2991"><a href="#cb19-2991" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb19-2992"><a href="#cb19-2992" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> InvCa.type, <span class="at">y =</span> CDH1)) <span class="sc">+</span></span>
<span id="cb19-2993"><a href="#cb19-2993" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_jitter</span>() <span class="sc">+</span> </span>
<span id="cb19-2994"><a href="#cb19-2994" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_violin</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span> </span>
<span id="cb19-2995"><a href="#cb19-2995" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>pam50, <span class="at">nrow =</span> <span class="dv">1</span>) <span class="sc">+</span> </span>
<span id="cb19-2996"><a href="#cb19-2996" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb19-2997"><a href="#cb19-2997" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">""</span>,</span>
<span id="cb19-2998"><a href="#cb19-2998" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"Regressed CDH1</span><span class="sc">\n</span><span class="st">expressioon levels"</span>,</span>
<span id="cb19-2999"><a href="#cb19-2999" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">"SCANB, ER+ HER2- BC samples only"</span></span>
<span id="cb19-3000"><a href="#cb19-3000" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-3001"><a href="#cb19-3001" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>)</span>
<span id="cb19-3002"><a href="#cb19-3002" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-3003"><a href="#cb19-3003" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-3004"><a href="#cb19-3004" aria-hidden="true" tabindex="-1"></a>The differences are still captured, and CDH1 is decreased when </span>
<span id="cb19-3005"><a href="#cb19-3005" aria-hidden="true" tabindex="-1"></a>comparing lobular vs NST. Note also that the difference seems to be</span>
<span id="cb19-3006"><a href="#cb19-3006" aria-hidden="true" tabindex="-1"></a>the biggest in the luminal B subtype.</span>
<span id="cb19-3007"><a href="#cb19-3007" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-3008"><a href="#cb19-3008" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.height = 5, fig.width = 14}</span></span>
<span id="cb19-3009"><a href="#cb19-3009" aria-hidden="true" tabindex="-1"></a>df_pcs_regressed[genes_of_interest, ] <span class="sc">%&gt;%</span></span>
<span id="cb19-3010"><a href="#cb19-3010" aria-hidden="true" tabindex="-1"></a>    t <span class="sc">%&gt;%</span></span>
<span id="cb19-3011"><a href="#cb19-3011" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span></span>
<span id="cb19-3012"><a href="#cb19-3012" aria-hidden="true" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"sample_name"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-3013"><a href="#cb19-3013" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">inner_join</span>(</span>
<span id="cb19-3014"><a href="#cb19-3014" aria-hidden="true" tabindex="-1"></a>        .,</span>
<span id="cb19-3015"><a href="#cb19-3015" aria-hidden="true" tabindex="-1"></a>        df_pca_scanb,</span>
<span id="cb19-3016"><a href="#cb19-3016" aria-hidden="true" tabindex="-1"></a>        <span class="at">by =</span> <span class="st">"sample_name"</span></span>
<span id="cb19-3017"><a href="#cb19-3017" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb19-3018"><a href="#cb19-3018" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb19-3019"><a href="#cb19-3019" aria-hidden="true" tabindex="-1"></a>        InvCa.type <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Lobular"</span>, <span class="st">"Ductal"</span>) <span class="sc">&amp;</span> </span>
<span id="cb19-3020"><a href="#cb19-3020" aria-hidden="true" tabindex="-1"></a>        er_status <span class="sc">==</span> <span class="st">"pos"</span> <span class="sc">&amp;</span> </span>
<span id="cb19-3021"><a href="#cb19-3021" aria-hidden="true" tabindex="-1"></a>        HER2 <span class="sc">==</span> <span class="st">"Negative"</span></span>
<span id="cb19-3022"><a href="#cb19-3022" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb19-3023"><a href="#cb19-3023" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> InvCa.type, <span class="at">y =</span> LOXL1)) <span class="sc">+</span></span>
<span id="cb19-3024"><a href="#cb19-3024" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_jitter</span>() <span class="sc">+</span> </span>
<span id="cb19-3025"><a href="#cb19-3025" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_violin</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span> </span>
<span id="cb19-3026"><a href="#cb19-3026" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>pam50, <span class="at">nrow =</span> <span class="dv">1</span>) <span class="sc">+</span> </span>
<span id="cb19-3027"><a href="#cb19-3027" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb19-3028"><a href="#cb19-3028" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">""</span>,</span>
<span id="cb19-3029"><a href="#cb19-3029" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"Regressed LOXL1</span><span class="sc">\n</span><span class="st">expressioon levels"</span>,</span>
<span id="cb19-3030"><a href="#cb19-3030" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">"SCANB, ER+ HER2- BC samples only"</span></span>
<span id="cb19-3031"><a href="#cb19-3031" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-3032"><a href="#cb19-3032" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>)</span>
<span id="cb19-3033"><a href="#cb19-3033" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-3034"><a href="#cb19-3034" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-3035"><a href="#cb19-3035" aria-hidden="true" tabindex="-1"></a>We also check the PC3 and PC4 loadings for CDH1, to see how important</span>
<span id="cb19-3036"><a href="#cb19-3036" aria-hidden="true" tabindex="-1"></a>this gene is in the molecular landscape. </span>
<span id="cb19-3037"><a href="#cb19-3037" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-3040"><a href="#cb19-3040" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-3041"><a href="#cb19-3041" aria-hidden="true" tabindex="-1"></a>pca_fit_all_genes_wo_outliers<span class="sc">$</span>loadings <span class="sc">%&gt;%</span></span>
<span id="cb19-3042"><a href="#cb19-3042" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(PC3, PC4) <span class="sc">%&gt;%</span></span>
<span id="cb19-3043"><a href="#cb19-3043" aria-hidden="true" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"gene"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-3044"><a href="#cb19-3044" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(gene <span class="sc">==</span> <span class="st">"CDH1"</span>)</span>
<span id="cb19-3045"><a href="#cb19-3045" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-3046"><a href="#cb19-3046" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-3047"><a href="#cb19-3047" aria-hidden="true" tabindex="-1"></a>PC4 has a relatively high value which puts it into the 95 percentile as </span>
<span id="cb19-3048"><a href="#cb19-3048" aria-hidden="true" tabindex="-1"></a>shown below.</span>
<span id="cb19-3049"><a href="#cb19-3049" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-3052"><a href="#cb19-3052" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-3053"><a href="#cb19-3053" aria-hidden="true" tabindex="-1"></a>ecdf_genes_pc4 <span class="ot">&lt;-</span> pca_fit_all_genes_wo_outliers<span class="sc">$</span>loadings <span class="sc">%&gt;%</span></span>
<span id="cb19-3054"><a href="#cb19-3054" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(PC3, PC4) <span class="sc">%&gt;%</span></span>
<span id="cb19-3055"><a href="#cb19-3055" aria-hidden="true" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"gene"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-3056"><a href="#cb19-3056" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">pull</span>(PC4) <span class="sc">%&gt;%</span></span>
<span id="cb19-3057"><a href="#cb19-3057" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ecdf</span>(.)</span>
<span id="cb19-3058"><a href="#cb19-3058" aria-hidden="true" tabindex="-1"></a><span class="fu">ecdf_genes_pc4</span>(<span class="fl">0.01509738</span>)</span>
<span id="cb19-3059"><a href="#cb19-3059" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-3060"><a href="#cb19-3060" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-3061"><a href="#cb19-3061" aria-hidden="true" tabindex="-1"></a>Next we can plot the lobular samples from Ciriello's paper</span>
<span id="cb19-3062"><a href="#cb19-3062" aria-hidden="true" tabindex="-1"></a>and color by the 3 subtypes they found. Note here that there is a mismatch</span>
<span id="cb19-3063"><a href="#cb19-3063" aria-hidden="true" tabindex="-1"></a>between the cbioportal data and what is found in the original papers</span>
<span id="cb19-3064"><a href="#cb19-3064" aria-hidden="true" tabindex="-1"></a><span class="co">[</span><span class="ot">@Curtis2012</span><span class="co">]</span>. We will use the Curtis 2012 annotation as this is what </span>
<span id="cb19-3065"><a href="#cb19-3065" aria-hidden="true" tabindex="-1"></a>matches Ciriello's paper. Surprisingly the mismatch is not a systematic</span>
<span id="cb19-3066"><a href="#cb19-3066" aria-hidden="true" tabindex="-1"></a>issue, rather 88 ILC samples are matching and the non-matching samples</span>
<span id="cb19-3067"><a href="#cb19-3067" aria-hidden="true" tabindex="-1"></a>are NST but the other info matches usually, such as NPI.</span>
<span id="cb19-3068"><a href="#cb19-3068" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-3071"><a href="#cb19-3071" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-3072"><a href="#cb19-3072" aria-hidden="true" tabindex="-1"></a>metabric_ilc_subtypes <span class="ot">&lt;-</span> <span class="fu">read.table</span>(</span>
<span id="cb19-3073"><a href="#cb19-3073" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../../data/2015_Ciriello_metabric_ilc_subtypes.txt"</span>,</span>
<span id="cb19-3074"><a href="#cb19-3074" aria-hidden="true" tabindex="-1"></a>    <span class="at">header =</span> <span class="cn">FALSE</span></span>
<span id="cb19-3075"><a href="#cb19-3075" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-3076"><a href="#cb19-3076" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-3077"><a href="#cb19-3077" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(metabric_ilc_subtypes) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"sample_name"</span>, <span class="st">"cohort"</span>, <span class="st">"subtype"</span>)</span>
<span id="cb19-3078"><a href="#cb19-3078" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-3079"><a href="#cb19-3079" aria-hidden="true" tabindex="-1"></a>metabric_lobular <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">inner_join</span>(</span>
<span id="cb19-3080"><a href="#cb19-3080" aria-hidden="true" tabindex="-1"></a>    df_pca, </span>
<span id="cb19-3081"><a href="#cb19-3081" aria-hidden="true" tabindex="-1"></a>    metabric_ilc_subtypes,</span>
<span id="cb19-3082"><a href="#cb19-3082" aria-hidden="true" tabindex="-1"></a>    <span class="at">by =</span> <span class="st">"sample_name"</span></span>
<span id="cb19-3083"><a href="#cb19-3083" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-3084"><a href="#cb19-3084" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-3085"><a href="#cb19-3085" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-3086"><a href="#cb19-3086" aria-hidden="true" tabindex="-1"></a><span class="in">```{r fig.width=9, fig.height=6, message=FALSE}</span></span>
<span id="cb19-3087"><a href="#cb19-3087" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-pca-lobular-metabric</span></span>
<span id="cb19-3088"><a href="#cb19-3088" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Embedding of the lobular samples from METABRIC on top </span></span>
<span id="cb19-3089"><a href="#cb19-3089" aria-hidden="true" tabindex="-1"></a><span class="co">#|     SCANB and TCGA</span></span>
<span id="cb19-3090"><a href="#cb19-3090" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">get_base_plot</span>(df_pca <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"tcga"</span>, <span class="st">"scanb"</span>)))</span>
<span id="cb19-3091"><a href="#cb19-3091" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-3092"><a href="#cb19-3092" aria-hidden="true" tabindex="-1"></a>p <span class="sc">+</span></span>
<span id="cb19-3093"><a href="#cb19-3093" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>(</span>
<span id="cb19-3094"><a href="#cb19-3094" aria-hidden="true" tabindex="-1"></a>        metabric_lobular,</span>
<span id="cb19-3095"><a href="#cb19-3095" aria-hidden="true" tabindex="-1"></a>        <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> PC3, <span class="at">y =</span> PC4),</span>
<span id="cb19-3096"><a href="#cb19-3096" aria-hidden="true" tabindex="-1"></a>        <span class="at">size =</span> <span class="dv">4</span></span>
<span id="cb19-3097"><a href="#cb19-3097" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-3098"><a href="#cb19-3098" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">15</span>) <span class="sc">+</span></span>
<span id="cb19-3099"><a href="#cb19-3099" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb19-3100"><a href="#cb19-3100" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">"Lobular samples from METABRIC</span><span class="sc">\n</span><span class="st">on top of SCANB and TCGA samples"</span></span>
<span id="cb19-3101"><a href="#cb19-3101" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-3102"><a href="#cb19-3102" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_color_manual</span>(</span>
<span id="cb19-3103"><a href="#cb19-3103" aria-hidden="true" tabindex="-1"></a>        <span class="at">values =</span> <span class="fu">c</span>(<span class="fu">get_colors_pam50</span>(df_pca <span class="sc">%&gt;%</span></span>
<span id="cb19-3104"><a href="#cb19-3104" aria-hidden="true" tabindex="-1"></a>                                        dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"metabric"</span>)))</span>
<span id="cb19-3105"><a href="#cb19-3105" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb19-3106"><a href="#cb19-3106" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span></span>
<span id="cb19-3107"><a href="#cb19-3107" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>(<span class="at">shape =</span> <span class="cn">FALSE</span>)</span>
<span id="cb19-3108"><a href="#cb19-3108" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-3109"><a href="#cb19-3109" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-3110"><a href="#cb19-3110" aria-hidden="true" tabindex="-1"></a>The table below shows the number of subtypes for each molecular subtype. </span>
<span id="cb19-3111"><a href="#cb19-3111" aria-hidden="true" tabindex="-1"></a>In general there are samples in each one of the molecular subtypes.</span>
<span id="cb19-3112"><a href="#cb19-3112" aria-hidden="true" tabindex="-1"></a>The reactive-like is actually more enriched in the normal region.</span>
<span id="cb19-3113"><a href="#cb19-3113" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-3116"><a href="#cb19-3116" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-3117"><a href="#cb19-3117" aria-hidden="true" tabindex="-1"></a>metabric_lobular <span class="sc">%&gt;%</span> </span>
<span id="cb19-3118"><a href="#cb19-3118" aria-hidden="true" tabindex="-1"></a>    janitor<span class="sc">::</span><span class="fu">tabyl</span>(subtype, pam50)</span>
<span id="cb19-3119"><a href="#cb19-3119" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-3120"><a href="#cb19-3120" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-3121"><a href="#cb19-3121" aria-hidden="true" tabindex="-1"></a>And below is a plot of the lobular samples from Ciriello/Curtis with</span>
<span id="cb19-3122"><a href="#cb19-3122" aria-hidden="true" tabindex="-1"></a>the molecular subtypes defined by Ciriello. </span>
<span id="cb19-3123"><a href="#cb19-3123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-3126"><a href="#cb19-3126" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-3127"><a href="#cb19-3127" aria-hidden="true" tabindex="-1"></a>ggplot2<span class="sc">::</span><span class="fu">ggplot</span>() <span class="sc">+</span> </span>
<span id="cb19-3128"><a href="#cb19-3128" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>(</span>
<span id="cb19-3129"><a href="#cb19-3129" aria-hidden="true" tabindex="-1"></a>        metabric_lobular,</span>
<span id="cb19-3130"><a href="#cb19-3130" aria-hidden="true" tabindex="-1"></a>        <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> PC3, <span class="at">y =</span> PC4, <span class="at">color =</span> subtype),</span>
<span id="cb19-3131"><a href="#cb19-3131" aria-hidden="true" tabindex="-1"></a>        <span class="at">size =</span> <span class="dv">4</span></span>
<span id="cb19-3132"><a href="#cb19-3132" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb19-3133"><a href="#cb19-3133" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">15</span>) <span class="sc">+</span></span>
<span id="cb19-3134"><a href="#cb19-3134" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb19-3135"><a href="#cb19-3135" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">"Lobular samples from METABRIC</span><span class="sc">\n</span><span class="st">on top of SCANB and TCGA samples"</span></span>
<span id="cb19-3136"><a href="#cb19-3136" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb19-3137"><a href="#cb19-3137" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span></span>
<span id="cb19-3138"><a href="#cb19-3138" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>(<span class="at">shape =</span> <span class="cn">FALSE</span>)</span>
<span id="cb19-3139"><a href="#cb19-3139" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-3140"><a href="#cb19-3140" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-3141"><a href="#cb19-3141" aria-hidden="true" tabindex="-1"></a>It seems that more proliferative saaples are in the luminal B </span>
<span id="cb19-3142"><a href="#cb19-3142" aria-hidden="true" tabindex="-1"></a>region whereas the Reactive-like and Immune-related are not so</span>
<span id="cb19-3143"><a href="#cb19-3143" aria-hidden="true" tabindex="-1"></a>distinguishable, they are in the luminal A region.</span>
<span id="cb19-3144"><a href="#cb19-3144" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-3145"><a href="#cb19-3145" aria-hidden="true" tabindex="-1"></a>Some of the samples are NST on cbioportal and Lobular on Curtis. </span>
<span id="cb19-3146"><a href="#cb19-3146" aria-hidden="true" tabindex="-1"></a>We selected the following samples and checked on </span>
<span id="cb19-3147"><a href="#cb19-3147" aria-hidden="true" tabindex="-1"></a>cbioPortal (https://www.cbioportal.org/study/clinicalData?id=brca_metabric)</span>
<span id="cb19-3148"><a href="#cb19-3148" aria-hidden="true" tabindex="-1"></a>their histologic subtype directly.</span>
<span id="cb19-3149"><a href="#cb19-3149" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-3152"><a href="#cb19-3152" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-3153"><a href="#cb19-3153" aria-hidden="true" tabindex="-1"></a>metabric_lobular <span class="sc">%&gt;%</span></span>
<span id="cb19-3154"><a href="#cb19-3154" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(sample_name, npi) <span class="sc">%&gt;%</span> </span>
<span id="cb19-3155"><a href="#cb19-3155" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">arrange</span>(sample_name) <span class="sc">%&gt;%</span></span>
<span id="cb19-3156"><a href="#cb19-3156" aria-hidden="true" tabindex="-1"></a>    <span class="fu">head</span>()</span>
<span id="cb19-3157"><a href="#cb19-3157" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-3158"><a href="#cb19-3158" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-3159"><a href="#cb19-3159" aria-hidden="true" tabindex="-1"></a>All the 6 samples are Ductal/NST and not Lobular.</span>
<span id="cb19-3160"><a href="#cb19-3160" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-3161"><a href="#cb19-3161" aria-hidden="true" tabindex="-1"></a><span class="fu">## Robustness to library protocol</span></span>
<span id="cb19-3162"><a href="#cb19-3162" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-3163"><a href="#cb19-3163" aria-hidden="true" tabindex="-1"></a>The figure below shows the first two components</span>
<span id="cb19-3164"><a href="#cb19-3164" aria-hidden="true" tabindex="-1"></a>for SCANB stratified by the library protocol used in the </span>
<span id="cb19-3165"><a href="#cb19-3165" aria-hidden="true" tabindex="-1"></a>study. We see that PC2 is partially capturing the</span>
<span id="cb19-3166"><a href="#cb19-3166" aria-hidden="true" tabindex="-1"></a>differences due to library protocol.</span>
<span id="cb19-3167"><a href="#cb19-3167" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-3170"><a href="#cb19-3170" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb19-3171"><a href="#cb19-3171" aria-hidden="true" tabindex="-1"></a>df_pca_scores <span class="sc">%&gt;%</span> </span>
<span id="cb19-3172"><a href="#cb19-3172" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"scanb"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-3173"><a href="#cb19-3173" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(</span>
<span id="cb19-3174"><a href="#cb19-3174" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> PC1,</span>
<span id="cb19-3175"><a href="#cb19-3175" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> PC2,</span>
<span id="cb19-3176"><a href="#cb19-3176" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> LibraryProtocol</span>
<span id="cb19-3177"><a href="#cb19-3177" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">+</span> </span>
<span id="cb19-3178"><a href="#cb19-3178" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span> </span>
<span id="cb19-3179"><a href="#cb19-3179" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>LibraryProtocol) <span class="sc">+</span></span>
<span id="cb19-3180"><a href="#cb19-3180" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>()</span>
<span id="cb19-3181"><a href="#cb19-3181" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-3182"><a href="#cb19-3182" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-3183"><a href="#cb19-3183" aria-hidden="true" tabindex="-1"></a><span class="fu">## Distance in 4th component for POETIC</span></span>
<span id="cb19-3184"><a href="#cb19-3184" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-3185"><a href="#cb19-3185" aria-hidden="true" tabindex="-1"></a>We have seen previosuly that there is a bigger difference in </span>
<span id="cb19-3186"><a href="#cb19-3186" aria-hidden="true" tabindex="-1"></a>between the matched samples in the responder than to non-responder </span>
<span id="cb19-3187"><a href="#cb19-3187" aria-hidden="true" tabindex="-1"></a>patients. Hallmark G2M checkpoint is one of the signatures</span>
<span id="cb19-3188"><a href="#cb19-3188" aria-hidden="true" tabindex="-1"></a>driving the PC4, meaning that the position wrt the fourth component</span>
<span id="cb19-3189"><a href="#cb19-3189" aria-hidden="true" tabindex="-1"></a>might be used to define a new response status. </span>
<span id="cb19-3190"><a href="#cb19-3190" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-3191"><a href="#cb19-3191" aria-hidden="true" tabindex="-1"></a>Below we plot the matched samples for all non responder patients</span>
<span id="cb19-3192"><a href="#cb19-3192" aria-hidden="true" tabindex="-1"></a>with no change at all in Ki67 (change &gt;= 0%)</span>
<span id="cb19-3193"><a href="#cb19-3193" aria-hidden="true" tabindex="-1"></a>and their respective positions in the molecular landscape.</span>
<span id="cb19-3194"><a href="#cb19-3194" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-3195"><a href="#cb19-3195" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=16, fig.height=10}</span></span>
<span id="cb19-3196"><a href="#cb19-3196" aria-hidden="true" tabindex="-1"></a>df_pca_scores <span class="sc">%&gt;%</span> </span>
<span id="cb19-3197"><a href="#cb19-3197" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb19-3198"><a href="#cb19-3198" aria-hidden="true" tabindex="-1"></a>        cohort <span class="sc">==</span> <span class="st">"poetic"</span> <span class="sc">&amp;</span> </span>
<span id="cb19-3199"><a href="#cb19-3199" aria-hidden="true" tabindex="-1"></a>            is_responder <span class="sc">==</span> <span class="st">"non_responder"</span> <span class="sc">&amp;</span></span>
<span id="cb19-3200"><a href="#cb19-3200" aria-hidden="true" tabindex="-1"></a>            change_ki67 <span class="sc">&gt;</span> <span class="dv">0</span></span>
<span id="cb19-3201"><a href="#cb19-3201" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb19-3202"><a href="#cb19-3202" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(</span>
<span id="cb19-3203"><a href="#cb19-3203" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> PC3,</span>
<span id="cb19-3204"><a href="#cb19-3204" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> PC4,</span>
<span id="cb19-3205"><a href="#cb19-3205" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> patient_nb,</span>
<span id="cb19-3206"><a href="#cb19-3206" aria-hidden="true" tabindex="-1"></a>        <span class="at">shape =</span> timepoint,</span>
<span id="cb19-3207"><a href="#cb19-3207" aria-hidden="true" tabindex="-1"></a>        <span class="at">label =</span> <span class="fu">paste</span>(change_ki67)</span>
<span id="cb19-3208"><a href="#cb19-3208" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">+</span> </span>
<span id="cb19-3209"><a href="#cb19-3209" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">4</span>) <span class="sc">+</span> </span>
<span id="cb19-3210"><a href="#cb19-3210" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">15</span>) <span class="sc">+</span> </span>
<span id="cb19-3211"><a href="#cb19-3211" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>patient_nb) <span class="sc">+</span> </span>
<span id="cb19-3212"><a href="#cb19-3212" aria-hidden="true" tabindex="-1"></a>    ggrepel<span class="sc">::</span><span class="fu">geom_label_repel</span>(<span class="at">max.overlaps =</span> <span class="dv">10</span>) <span class="sc">+</span></span>
<span id="cb19-3213"><a href="#cb19-3213" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">coord_cartesian</span>(</span>
<span id="cb19-3214"><a href="#cb19-3214" aria-hidden="true" tabindex="-1"></a>        <span class="at">xlim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">10</span>, <span class="dv">7</span>),</span>
<span id="cb19-3215"><a href="#cb19-3215" aria-hidden="true" tabindex="-1"></a>        <span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">6</span>, <span class="dv">6</span>)</span>
<span id="cb19-3216"><a href="#cb19-3216" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb19-3217"><a href="#cb19-3217" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-3218"><a href="#cb19-3218" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-3219"><a href="#cb19-3219" aria-hidden="true" tabindex="-1"></a>Now we do the same for the top responders (change in Ki67 $\leq$ -90%).</span>
<span id="cb19-3220"><a href="#cb19-3220" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-3221"><a href="#cb19-3221" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=16, fig.height=10}</span></span>
<span id="cb19-3222"><a href="#cb19-3222" aria-hidden="true" tabindex="-1"></a>df_pca_scores <span class="sc">%&gt;%</span> </span>
<span id="cb19-3223"><a href="#cb19-3223" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb19-3224"><a href="#cb19-3224" aria-hidden="true" tabindex="-1"></a>        cohort <span class="sc">==</span> <span class="st">"poetic"</span> <span class="sc">&amp;</span> </span>
<span id="cb19-3225"><a href="#cb19-3225" aria-hidden="true" tabindex="-1"></a>            is_responder <span class="sc">==</span> <span class="st">"responder"</span> <span class="sc">&amp;</span></span>
<span id="cb19-3226"><a href="#cb19-3226" aria-hidden="true" tabindex="-1"></a>            change_ki67 <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">90</span></span>
<span id="cb19-3227"><a href="#cb19-3227" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb19-3228"><a href="#cb19-3228" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(</span>
<span id="cb19-3229"><a href="#cb19-3229" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> PC3,</span>
<span id="cb19-3230"><a href="#cb19-3230" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> PC4,</span>
<span id="cb19-3231"><a href="#cb19-3231" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> patient_nb,</span>
<span id="cb19-3232"><a href="#cb19-3232" aria-hidden="true" tabindex="-1"></a>        <span class="at">shape =</span> timepoint</span>
<span id="cb19-3233"><a href="#cb19-3233" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">+</span> </span>
<span id="cb19-3234"><a href="#cb19-3234" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">4</span>) <span class="sc">+</span> </span>
<span id="cb19-3235"><a href="#cb19-3235" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">15</span>) <span class="sc">+</span> </span>
<span id="cb19-3236"><a href="#cb19-3236" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>patient_nb) <span class="sc">+</span> </span>
<span id="cb19-3237"><a href="#cb19-3237" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">coord_cartesian</span>(</span>
<span id="cb19-3238"><a href="#cb19-3238" aria-hidden="true" tabindex="-1"></a>        <span class="at">xlim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">10</span>, <span class="dv">7</span>),</span>
<span id="cb19-3239"><a href="#cb19-3239" aria-hidden="true" tabindex="-1"></a>        <span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">6</span>, <span class="dv">6</span>)</span>
<span id="cb19-3240"><a href="#cb19-3240" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb19-3241"><a href="#cb19-3241" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb19-3242"><a href="#cb19-3242" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-3243"><a href="#cb19-3243" aria-hidden="true" tabindex="-1"></a>In general the distance for the responder samples seem to be higher and not only</span>
<span id="cb19-3244"><a href="#cb19-3244" aria-hidden="true" tabindex="-1"></a>that we see that the surgery sample goes to the bottom part of the embedding, </span>
<span id="cb19-3245"><a href="#cb19-3245" aria-hidden="true" tabindex="-1"></a>suggesting that the reduction of the proliferation is reflected in the</span>
<span id="cb19-3246"><a href="#cb19-3246" aria-hidden="true" tabindex="-1"></a>molecular landscape.</span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
      &nbsp;
    </div>   
    <div class="nav-footer-center">
      &nbsp;
    </div>
    <div class="nav-footer-right">
      <ul class="footer-items list-unstyled">
    <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/chronchi">
      <i class="bi bi-github" role="img">
</i> 
    </a>
  </li>  
    <li class="nav-item compact">
    <a class="nav-link" href="https://twitter.com/chronchi">
      <i class="bi bi-twitter" role="img">
</i> 
    </a>
  </li>  
</ul>
    </div>
  </div>
</footer>



</body></html>