<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.554">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>EMBER creates a unified space for independent breast cancer transcriptomic datasets enabling precision oncology. - 3&nbsp; Validating the molecular landscape</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../chapters/scoring.html" rel="next">
<link href="../chapters/pca_merging.html" rel="prev">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script src="../site_libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<link href="../site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="../site_libs/datatables-binding-0.28/datatables.js"></script>
<script src="../site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<link href="../site_libs/dt-core-1.13.4/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="../site_libs/dt-core-1.13.4/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="../site_libs/dt-core-1.13.4/js/jquery.dataTables.min.js"></script>
<link href="../site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet">
<script src="../site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<script src="../site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="../site_libs/lightable-0.0.1/lightable.css" rel="stylesheet">
<link href="../site_libs/nouislider-7.0.10/jquery.nouislider.min.css" rel="stylesheet">
<script src="../site_libs/nouislider-7.0.10/jquery.nouislider.min.js"></script>
<link href="../site_libs/selectize-0.12.0/selectize.bootstrap3.css" rel="stylesheet">
<script src="../site_libs/selectize-0.12.0/selectize.min.js"></script>

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../chapters/validation.html"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Validating the molecular landscape</span></a></li></ol></nav>
        <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../">EMBER creates a unified space for independent breast cancer transcriptomic datasets enabling precision oncology.</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/chronchi/molecular_landscape" title="Source Code" class="quarto-navigation-tool px-1" aria-label="Source Code"><i class="bi bi-github"></i></a>
    <a href="https://twitter.com/intent/tweet?url=|url|" title="Twitter" class="quarto-navigation-tool px-1" aria-label="Twitter"><i class="bi bi-twitter"></i></a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Preface</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/intro.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Introduction</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/surv_analysis_estrogen.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Estrogen receptor status is continuous, not dichotomous</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/pca_merging.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">A new framework for personalized medicine: integrating</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/validation.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Validating the molecular landscape</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/scoring.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Pathway activity in a personalized context</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/risk_score.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Risk score model</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/expanding_ember.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Expanding embedding and removing batch effects</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/trying.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Tests and tries</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/revision.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Revision</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/discussion.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Discussion</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/references.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">References</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#scanb-embedding" id="toc-scanb-embedding" class="nav-link active" data-scroll-target="#scanb-embedding"><span class="header-section-number">3.1</span> SCANB embedding</a></li>
  <li><a href="#missing-genes" id="toc-missing-genes" class="nav-link" data-scroll-target="#missing-genes"><span class="header-section-number">3.2</span> Missing genes</a>
  <ul class="collapse">
  <li><a href="#fuzziness-score" id="toc-fuzziness-score" class="nav-link" data-scroll-target="#fuzziness-score"><span class="header-section-number">3.2.1</span> Fuzziness score</a></li>
  <li><a href="#validating-the-score" id="toc-validating-the-score" class="nav-link" data-scroll-target="#validating-the-score"><span class="header-section-number">3.2.2</span> Validating the score</a></li>
  <li><a href="#systematic-approach" id="toc-systematic-approach" class="nav-link" data-scroll-target="#systematic-approach"><span class="header-section-number">3.2.3</span> Systematic approach</a></li>
  </ul></li>
  <li><a href="#smc-embedding" id="toc-smc-embedding" class="nav-link" data-scroll-target="#smc-embedding"><span class="header-section-number">3.3</span> SMC embedding</a></li>
  <li><a href="#pdx" id="toc-pdx" class="nav-link" data-scroll-target="#pdx"><span class="header-section-number">3.4</span> PDX</a>
  <ul class="collapse">
  <li><a href="#growth-curves" id="toc-growth-curves" class="nav-link" data-scroll-target="#growth-curves"><span class="header-section-number">3.4.1</span> Growth curves</a></li>
  </ul></li>
  <li><a href="#normal-samples" id="toc-normal-samples" class="nav-link" data-scroll-target="#normal-samples"><span class="header-section-number">3.5</span> Normal samples</a></li>
  <li><a href="#third-component-as-a-prognostic-measure" id="toc-third-component-as-a-prognostic-measure" class="nav-link" data-scroll-target="#third-component-as-a-prognostic-measure"><span class="header-section-number">3.6</span> Third component as a prognostic measure?</a></li>
  <li><a href="#high-risk-patients" id="toc-high-risk-patients" class="nav-link" data-scroll-target="#high-risk-patients"><span class="header-section-number">3.7</span> High risk patients</a></li>
  </ul>
<div class="toc-actions"><ul><li><a href="https://github.com/chronchi/molecular_landscape/issues/new" class="toc-action"><i class="bi bi-github"></i>Report an issue</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Validating the molecular landscape</span></h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<p>In the last chapters only TCGA and METABRIC samples were used in the training and testing of the PCA projection. To completely validate, we use another big cohort, SCANB, that was not used in the PCA fitting and testing before. This way we ensure that the embedding is independent of the cohort. Moreover, we use the SMC cohort <span class="citation" data-cites="Kan2018">(<a href="references.html#ref-Kan2018" role="doc-biblioref">Kan et al. 2018</a>)</span>, whose patients are from South Korea. We show how well it is embedded, and that is not dependent on population characteristics.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>* The library is already synchronized with the lockfile.</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>    ../../results/plots/validation ../../results/rds_files/validation 
                             FALSE                              FALSE 
   ../../results/tables/validation 
                             FALSE </code></pre>
</div>
</div>
<section id="scanb-embedding" class="level2" data-number="3.1">
<h2 data-number="3.1" class="anchored" data-anchor-id="scanb-embedding"><span class="header-section-number">3.1</span> SCANB embedding</h2>
<p>The embedding was already calculate before when calculating the fit and tests, since the datasets were organized and loaded together. We also know already that all the 1044 genes are available in the SCANB cohort. SCANB is well mixed with all cohorts as <a href="#fig-pca-scanb" class="quarto-xref">Figure&nbsp;<span>3.1</span></a> shows.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pca-scanb" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-pca-scanb-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="validation_files/figure-html/fig-pca-scanb-1.png" class="img-fluid figure-img" width="672">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-pca-scanb-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3.1: PCA embedding colored by cohort. All samples from TCGA, METABRIC and SCANB were used.
</figcaption>
</figure>
</div>
</div>
</div>
<p>Ideally there would be some overlap between SCANB and TCGA already in the first component. Moreover, if there isn’t, the cohort should be closer to TCGA than to METABRIC, due to technologies and normalization procedures, as both of them are log FPKM. Even though SCANB was processed in a way to get directly FPKM values, by using cufflinks, TCGA was normalized to get log FPKM as well, as discussed previously. <a href="expanding_ember.html#fig-pca-scanb-pc1" class="quarto-xref">Figure&nbsp;<span>6.3</span></a> shows how close SCANB is from TCGA, moreover there is already some overlap. The variability in the first component is bigger for METABRIC compared to TCGA and SCANB. Probably this is due to the technology processing, since they have genes in the same scale. <a href="pca_merging.html#fig-pca-no-norm" class="quarto-xref">Figure&nbsp;<span>2.33</span></a> contrasts the different normalization procedures. If genes are not scaled correctly, the PCA shows an inverse picture.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pca-scanb-pc1" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-pca-scanb-pc1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="validation_files/figure-html/fig-pca-scanb-pc1-1.png" class="img-fluid figure-img" width="672">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-pca-scanb-pc1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3.2: PCA embedding colored by cohort. The components used are the first and second components.
</figcaption>
</figure>
</div>
</div>
</div>
<p><a href="#fig-pca-scanb-er-pam50" class="quarto-xref">Figure&nbsp;<span>3.3</span></a> shows that the SCANB samples are also well mixed regarding the clinical factors, including the <span class="math inline">\(SET_{ER/PR}\)</span> signature.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pca-scanb-er-pam50" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-pca-scanb-er-pam50-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="validation_files/figure-html/fig-pca-scanb-er-pam50-1.png" class="img-fluid figure-img" width="1728">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-pca-scanb-er-pam50-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3.3: PCA embedding of all samples from TCGA, SCANB and METABRIC. (A) Colored by cohort, (B) colored by ER status, (C) colored by PAM50 molecular subtype, (D) colored by the <span class="math inline">\(SET_{ER/PR}\)</span> signature and only SCANB samples.
</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="missing-genes" class="level2" data-number="3.2">
<h2 data-number="3.2" class="anchored" data-anchor-id="missing-genes"><span class="header-section-number">3.2</span> Missing genes</h2>
<p>Genes missing in publicly available datasets is very common. Usually this is due to processing pipelines or even data quality, therefore genes are removed. This should not be much of a problem, if the data is good enough, when calculating the qPCR-like normalization as it was shown before. The problem arises when multiplying the loadings obtained in the PCA with the normalized expression of the sample. If several genes are missing, 0s are added to the matrix and therefore the loadings for these genes cannot be used. Since 1000 genes were used, we investigate the loadings of genes and try to calculate a fuzziness score, indicating when the embedding should be trusted if genes are missing, and which genes are missing.</p>
<section id="fuzziness-score" class="level3" data-number="3.2.1">
<h3 data-number="3.2.1" class="anchored" data-anchor-id="fuzziness-score"><span class="header-section-number">3.2.1</span> Fuzziness score</h3>
<p>The idea of the fuzziness score is that we use the loadings of the missing genes and calculate the cumulative sum of their absolute values. The higher this value the more it indicates the missing genes are important. We start by removing random genes from random samples of the SCANB cohort. Each sample will have a random number of genes removed from the set of 1044 genes. Then two scores are calculated, one for PC3 and another for PC4. We also calculate the fuzziness score for the top loading genes and the low loadings genes, to compare these values.</p>
<p>We start by comparing the scores from top loadings and low loading genes. For this we remove the top 50 loadings and bottom 50 loadings, in terms of absolute values. <a href="#fig-loadings-top-bottom" class="quarto-xref">Figure&nbsp;<span>3.4</span></a> shows the cumulative sum of the absolute values from the top and bottom loadings respectively. See how important the top loadings are compared to the bottom loadings. A way of thinking of this is that the top loadings weights are much more important when calculating the embedding.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-loadings-top-bottom" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-loadings-top-bottom-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="validation_files/figure-html/fig-loadings-top-bottom-1.png" class="img-fluid figure-img" width="672">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-loadings-top-bottom-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3.4: Cumulative sum of absolute values for the top loadings and bottom loadings as a function of the number of selected genes.
</figcaption>
</figure>
</div>
</div>
</div>
<p>This was using the top loadings, so probably the fuzziness score of a sample with relatively good data will be between 0 and 12 maximum, meaning that they have less than 200 genes missing. When genes are missing, we hope that they are in the bottom part of the loadings, so the fuzziness score is as small as possible.</p>
<p>Suppose now that the genes missing in a sample are random. We will calculate random fuzziness scores based on this to have a feeling of the scores.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-fuzziness-random" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-fuzziness-random-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="validation_files/figure-html/fig-fuzziness-random-1.png" class="img-fluid figure-img" width="672">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-fuzziness-random-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3.5: Distribution of the fuzziness scores from random missing genes. Each color corresponds to a different distribution using a different number of genes.
</figcaption>
</figure>
</div>
</div>
</div>
<p><a href="#fig-fuzziness-random" class="quarto-xref">Figure&nbsp;<span>3.5</span></a> shows the results as a function of the number of missing genes. We see that in average, if 200 genes are missing, the fuzziness score is around 5. If less than a 100 genes are missing, the fuzziness score is around 3. In the next section we show how this influences the position where the sample is projected into the embedding.</p>
<p>Finally, we calculate the fuzziness score as a number of top loadings missing. Suppose 150 genes are missing. We will change the proportion of genes that are in the top loadings and in the missing list. Since the cumulative sum for PC2 and PC3 are very similar, we focus only on PC2.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-fuzziness-random-top" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-fuzziness-random-top-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="validation_files/figure-html/fig-fuzziness-random-top-1.png" class="img-fluid figure-img" width="672">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-fuzziness-random-top-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3.6: Distribution of the fuzziness scores from 150 random missing genes with different proportions of top loading genes in the list. Each color corresponds to a different distribution using a different proportion of genes.
</figcaption>
</figure>
</div>
</div>
</div>
<p><a href="#fig-fuzziness-random-top" class="quarto-xref">Figure&nbsp;<span>3.6</span></a> shows the fuzziness score with different proportions of the top loading genes. We see that they are actually very important when calculating the score. The top loading genes were defined as the top 150 genes in the list. So when calculating the number of missing genes it is important to check the number of top loadings in the list.</p>
</section>
<section id="validating-the-score" class="level3" data-number="3.2.2">
<h3 data-number="3.2.2" class="anchored" data-anchor-id="validating-the-score"><span class="header-section-number">3.2.2</span> Validating the score</h3>
<p>In this section we calculate the embeddings with different number of genes missing from a given sample and then plot the original embedding with the newly one. For this we use samples from TCGA. The genes are removed even before the normalization procedure, as they are usually not available there. We select randomly a number of top genes from PC3 and PC4 to compose the list of missing genes. The proportion of the top loading genes is varied.</p>
<div class="tabset-margin-container"></div><div class="panel-tabset">
<ul class="nav nav-tabs" role="tablist"><li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-1-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-1" role="tab" aria-controls="tabset-1-1" aria-selected="true">TCGA-OL-A66K-01A-11R-A29R-07</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-2" role="tab" aria-controls="tabset-1-2" aria-selected="false">TCGA-GI-A2C9-01A-11R-A21T-07</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-3-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-3" role="tab" aria-controls="tabset-1-3" aria-selected="false">TCGA-3C-AALK-01A-11R-A41B-07</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-4-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-4" role="tab" aria-controls="tabset-1-4" aria-selected="false">TCGA-BH-A0BW-01A-11R-A115-07</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-5-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-5" role="tab" aria-controls="tabset-1-5" aria-selected="false">TCGA-A1-A0SI-01A-11R-A144-07</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-6-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-6" role="tab" aria-controls="tabset-1-6" aria-selected="false">TCGA-D8-A27L-01A-11R-A16F-07</a></li></ul>
<div class="tab-content">
<div id="tabset-1-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-1-1-tab">
<div class="cell">
<div class="cell-output-display">
<div id="fig-missing-correction-10" class="quarto-figure quarto-figure-center quarto-float anchored" width="768">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-missing-correction-10-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="validation_files/figure-html/fig-missing-correction-10-1.png" id="fig-missing-correction-10" class="img-fluid figure-img" width="768">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig quarto-uncaptioned" id="fig-missing-correction-10-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3.7
</figcaption>
</figure>
</div>
</div>
</div>
</div>
<div id="tabset-1-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-2-tab">
<div class="cell">
<div class="cell-output-display">
<div id="fig-missing-correction-20" class="quarto-figure quarto-figure-center quarto-float anchored" width="768">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-missing-correction-20-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="validation_files/figure-html/fig-missing-correction-20-1.png" id="fig-missing-correction-20" class="img-fluid figure-img" width="768">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig quarto-uncaptioned" id="fig-missing-correction-20-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3.8
</figcaption>
</figure>
</div>
</div>
</div>
</div>
<div id="tabset-1-3" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-3-tab">
<div class="cell">
<div class="cell-output-display">
<div id="fig-missing-correction-50" class="quarto-figure quarto-figure-center quarto-float anchored" width="768">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-missing-correction-50-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="validation_files/figure-html/fig-missing-correction-50-1.png" id="fig-missing-correction-50" class="img-fluid figure-img" width="768">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig quarto-uncaptioned" id="fig-missing-correction-50-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3.9
</figcaption>
</figure>
</div>
</div>
</div>
</div>
<div id="tabset-1-4" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-4-tab">
<div class="cell">
<div class="cell-output-display">
<div id="fig-missing-correction-80" class="quarto-figure quarto-figure-center quarto-float anchored" width="768">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-missing-correction-80-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="validation_files/figure-html/fig-missing-correction-80-1.png" id="fig-missing-correction-80" class="img-fluid figure-img" width="768">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig quarto-uncaptioned" id="fig-missing-correction-80-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3.10
</figcaption>
</figure>
</div>
</div>
</div>
</div>
<div id="tabset-1-5" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-5-tab">
<div class="cell">
<div class="cell-output-display">
<div id="fig-missing-correction-182" class="quarto-figure quarto-figure-center quarto-float anchored" width="768">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-missing-correction-182-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="validation_files/figure-html/fig-missing-correction-182-1.png" id="fig-missing-correction-182" class="img-fluid figure-img" width="768">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig quarto-uncaptioned" id="fig-missing-correction-182-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3.11
</figcaption>
</figure>
</div>
</div>
</div>
</div>
<div id="tabset-1-6" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-6-tab">
<div class="cell">
<div class="cell-output-display">
<div id="fig-missing-correction-198" class="quarto-figure quarto-figure-center quarto-float anchored" width="768">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-missing-correction-198-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="validation_files/figure-html/fig-missing-correction-198-1.png" id="fig-missing-correction-198" class="img-fluid figure-img" width="768">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig quarto-uncaptioned" id="fig-missing-correction-198-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3.12
</figcaption>
</figure>
</div>
</div>
</div>
</div>
</div>
</div>
<p>When looking at the figures above, it seems that whenever there are genes missing, and the more of top loadings they are, the closer they are to the origin. This makes sense, since PCA is doing a simple linear combination of the loadings. This should be fine to correct, since we can fit a line going through the origin and goes through the points above. Then the adjustment will depend on the number of genes that are missing and are top loadings. The more they are, the more adjustment we will need.</p>
<p>For the adjustment we only know two things, its current position and how many of the top loadings are missing. Based on this information we should be able to correct the samples. Given the list of genes that are missing, we fit lines that would go through the genes that are missing in the TCGA and METABRIC samples. We then try to find the line that is closes to the point. After performing the fit and the line, we can then move the point to closer to the METABRIC and TCGA samples. The point should be close to the line and close to points that have the same proportion of top loadings missing.</p>
<p>These plots show how the fuzziness score might be indicative of how good the embedding will be depending on the number of missing genes. The video below shows the images for 20 different patients. Notice how all of them are connected to the origin as well.</p>
<iframe width="720" height="480" src="../plots/validation/random_loadings.mp4" align="middle" frameborder="0" allowfullscreen="">
</iframe>
</section>
<section id="systematic-approach" class="level3" data-number="3.2.3">
<h3 data-number="3.2.3" class="anchored" data-anchor-id="systematic-approach"><span class="header-section-number">3.2.3</span> Systematic approach</h3>
<p>We now use a more systematic approach to see the impact of the missing genes in the position in the embedding. The idea is that we extract some metrics from the TCGA samples as we increase the amount of the most important genes removed from the sample. What we simply do is for each patient we calculate the distance of the new embedding to the original embedding. Since for each proportion we ran the analysis 10 times we get 10 distances. We average over those. In the end we have 20 “distances” for each one of the 200 TCGA samples.</p>
<p><a href="#fig-proportion-missing-genes-distance" class="quarto-xref">Figure&nbsp;<span>3.13</span></a> shows the relationship between the proportion of missing top loading genes among the 200 genes that were selected out and the average distance to the original embedding. Each dot corresponds to the a sample. In total there are 200 dots in each proportion.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-proportion-missing-genes-distance" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-proportion-missing-genes-distance-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="validation_files/figure-html/fig-proportion-missing-genes-distance-1.png" class="img-fluid figure-img" width="960">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-proportion-missing-genes-distance-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3.13: Average distance to the original embedding
</figcaption>
</figure>
</div>
</div>
</div>
<p>We see that the more top loading genes that are missing the worse the embedding gets. Still the average distance is a bit difficult to interpret, so next we show the relationship between the variance and the proportion of missing genes.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-var-proportion-missing-genes" class="quarto-figure quarto-figure-center quarto-float anchored" width="672">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-var-proportion-missing-genes-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="validation_files/figure-html/fig-var-proportion-missing-genes-1.png" id="fig-var-proportion-missing-genes" class="img-fluid figure-img" width="672">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig quarto-uncaptioned" id="fig-var-proportion-missing-genes-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3.14
</figcaption>
</figure>
</div>
</div>
</div>
<p>Indeed as the proportion of the most important genes goes missing, there is a higher variance in the average distance.</p>
<p>Next we evaluate the number of samples in the neighborhood of samples, so we can understand a bit better what is distance in the molecular embedder.</p>
<p><a href="#fig-nb-samples-avg-neighborhood" class="quarto-xref">Figure&nbsp;<span>3.15</span></a> shows the proportion of samples in a neighborhood based on its radius. For samples within a radius of 1, in average the percentage of samples from the whole dataset including TCGA and METABRIC is of 2.5%.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-nb-samples-avg-neighborhood" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-nb-samples-avg-neighborhood-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="validation_files/figure-html/fig-nb-samples-avg-neighborhood-1.png" class="img-fluid figure-img" width="672">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-nb-samples-avg-neighborhood-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3.15: Percentage of number of samples in the neighborhood with varyiing radii. The total number of samples is 2873, including all samples from TCGA and METABRIC.
</figcaption>
</figure>
</div>
</div>
</div>
<p>This means that around 30% of top missing genes gives us neighborhoods that do not change much, since the average distance from the original embedding is below 1, so the neighborhood will not change much.</p>
</section>
</section>
<section id="smc-embedding" class="level2" data-number="3.3">
<h2 data-number="3.3" class="anchored" data-anchor-id="smc-embedding"><span class="header-section-number">3.3</span> SMC embedding</h2>
<p>By following the same procedures as previously, normalizing and getting the PC coordinates, <a href="#fig-smc-cohort" class="quarto-xref">Figure&nbsp;<span>3.16</span></a> shows the biplot of PC1 and PC2 when coloring by cohorts. The samples seem to be well mixed already in the RNA-seq samples. This is probably due to the fact that the SMC cohort has TPM values.</p>
<p>When checking the total number of genes missing, there are only 4 genes missing. And as seen before, this number is almost irrelevant, in the sense that the embedding will still be robust and locations will be precise enough.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-smc-cohort" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-smc-cohort-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="validation_files/figure-html/fig-smc-cohort-1.png" class="img-fluid figure-img" width="672">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-smc-cohort-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3.16: Biplot of first two PCA components from SMC, TCGA, SCANB and METABRIC. SMC is highlighted in the plot and has a bigger dot.
</figcaption>
</figure>
</div>
</div>
</div>
<p>As previously, <a href="#fig-pca-smc-er-pam50" class="quarto-xref">Figure&nbsp;<span>3.17</span></a> shows the biplot of PC3 and PC4, highlighting the well mixing of the cohorts. When coloring by ER status, samples are in the right group. Molecular subtype is also correct. This cohort is special in the sense that there are more luminal B patients, as discussed in the paper. That is why there are more samples in the luminal B region.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pca-smc-er-pam50" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-pca-smc-er-pam50-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="validation_files/figure-html/fig-pca-smc-er-pam50-1.png" class="img-fluid figure-img" width="1728">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-pca-smc-er-pam50-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3.17: PCA embedding of all samples from TCGA, SCANB and METABRIC including the SMC samples on top. (A) Colored by cohort, (B) colored by ER status, (C) colored by PAM50 molecular subtype.
</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="pdx" class="level2" data-number="3.4">
<h2 data-number="3.4" class="anchored" data-anchor-id="pdx"><span class="header-section-number">3.4</span> PDX</h2>
<p>Another way to validate the results is to use patient derived xenografts. In the Brisken lab the MIND model is used <span class="citation" data-cites="Sflomos2016 Scabia2022">(<a href="references.html#ref-Sflomos2016" role="doc-biblioref">Sflomos et al. 2016</a>; <a href="references.html#ref-Scabia2022" role="doc-biblioref">Scabia et al. 2022</a>)</span>. ER+ breast cancer samples coming from patients are processed in a specific way and injected into the mammary glands of mice. Based on that they can stablish themselves and proliferate. The environment should recapitulate what is seen in women as well, in the sense that estrogen (E2) levels are similar to those of post-menopausal women. Since these cells have to proliferate in order to establish and grow in the ducts, we expect them to be more proliferative in terms of RNA-seq components and also to lie in the luminal B region of the molecular landscape. We validate this now.</p>
<p>When checking the total number of genes missing, there are only 27 genes missing. And as seen before, this number is almost irrelevant, in the sense that the embedding will still be robust and locations will be precise enough.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pdx-cohort" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-pdx-cohort-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="validation_files/figure-html/fig-pdx-cohort-1.png" class="img-fluid figure-img" width="672">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-pdx-cohort-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3.18: Biplot of first two PCA components from PDX, TCGA, SCANB and METABRIC. PDX is highlighted in the plot and has bigger dot size.
</figcaption>
</figure>
</div>
</div>
</div>
<p>As previously, <a href="#fig-pca-pdx-cohort" class="quarto-xref">Figure&nbsp;<span>3.19</span></a> shows the biplot of PC3 and PC4, highlighting the well mixing of the cohorts. As expected the samples are overall in the luminal B region, since these are tumor cells in a proliferative state.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pca-pdx-cohort" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-pca-pdx-cohort-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="validation_files/figure-html/fig-pca-pdx-cohort-1.png" class="img-fluid figure-img" width="768">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-pca-pdx-cohort-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3.19: Embedding of all samples from TCGA and METABRIC including the PDX samples on top.
</figcaption>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pca-pdx-control-only" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-pca-pdx-control-only-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="validation_files/figure-html/fig-pca-pdx-control-only-1.png" class="img-fluid figure-img" width="864">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-pca-pdx-control-only-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3.20: Embedding of only PDX control samples on top of the METABRIC, SCANB and TCGA samples
</figcaption>
</figure>
</div>
</div>
</div>
<p>And we check the PDXs response to estrogen and see how much they move away from the untreated sample.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pdx-p4-control-components" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-pdx-p4-control-components-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="validation_files/figure-html/fig-pdx-p4-control-components-1.png" class="img-fluid figure-img" width="960">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-pdx-p4-control-components-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3.21: Comparison of position in the molecular landscape between CTRL and P4 treated samples
</figcaption>
</figure>
</div>
</div>
</div>
<p>It looks like some of the PDXs had a bigger shift towards the left when treated with estrogen or at least some stabilization. On the other hand, another set of PDXs, the last three, they didn’t see much change.</p>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="validation_files/figure-html/unnamed-chunk-39-1.png" class="img-fluid figure-img" width="1536"></p>
</figure>
</div>
</div>
</div>
<p>We now include as well the embedding for the P4 treated samples as well.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pca-pdx-only-embedding" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-pca-pdx-only-embedding-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="validation_files/figure-html/fig-pca-pdx-only-embedding-1.png" class="img-fluid figure-img" width="1152">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-pca-pdx-only-embedding-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3.22: Embedding of only PDX control and P4 samples
</figcaption>
</figure>
</div>
</div>
</div>
<p>We see that T110 has a change in position as well, it goes towards the center meaning that there is a decrease in proliferation and also ER signaling. T105 has a very strong effect, it goes up on the diagonal, meaning higher proliferation and lower ER signaling.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pdx-p4-control-pathways" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-pdx-p4-control-pathways-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="validation_files/figure-html/fig-pdx-p4-control-pathways-1.png" class="img-fluid figure-img" width="1536">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-pdx-p4-control-pathways-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3.23: Comparison of position in the molecular landscape between CTRL, E2 and P4 treated samples
</figcaption>
</figure>
</div>
</div>
</div>
<p>What we can see from these scores is that if there is an increase in G2M checkpoint and the control samples have scores below 0, then the fourth principal component will change it is value, it will go upwards and to the left in the molecular landscape, reflecting the biological response of the tumors to the given hormones.</p>
<p>Before we move on we also do the same using facet_grid instead of facet_wrap.</p>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="validation_files/figure-html/unnamed-chunk-42-1.png" class="img-fluid figure-img" width="1536"></p>
</figure>
</div>
</div>
</div>
<p>And the next two tables show the results for the differential expression analysis on the pathway level for each PDX individually. The estimate, i.e. difference, and the p-values are presented.</p>
<p>First for G2M:</p>
<div class="cell">
<div class="cell-output-display">
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-f39b8bb3c968a6aff95b" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-f39b8bb3c968a6aff95b">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6"],["T105","T111","T113","T109","METS15","T110"],[" 0.527"," 0.647"," 0.319"," 0.188","-0.037"," 0.036"],["6.6e-05","1.4e-04","3.9e-02","1.8e-01","5.9e-01","7.8e-01"],["HALLMARK_G2M_CHECKPOINT","HALLMARK_G2M_CHECKPOINT","HALLMARK_G2M_CHECKPOINT","HALLMARK_G2M_CHECKPOINT","HALLMARK_G2M_CHECKPOINT","HALLMARK_G2M_CHECKPOINT"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>pdx<\/th>\n      <th>estimate<\/th>\n      <th>p.value<\/th>\n      <th>pathway<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<p>Then to Estrogen Early:</p>
<div class="cell">
<div class="cell-output-display">
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-c691fbe77cc286ae1522" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-c691fbe77cc286ae1522">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6"],["T110","T105","T113","T111","METS15","T109"],["-0.237","-0.275","-0.179","-0.139","-0.192","-0.066"],["0.0020","0.0097","0.0988","0.1239","0.1493","0.3880"],["HALLMARK_ESTROGEN_RESPONSE_EARLY","HALLMARK_ESTROGEN_RESPONSE_EARLY","HALLMARK_ESTROGEN_RESPONSE_EARLY","HALLMARK_ESTROGEN_RESPONSE_EARLY","HALLMARK_ESTROGEN_RESPONSE_EARLY","HALLMARK_ESTROGEN_RESPONSE_EARLY"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>pdx<\/th>\n      <th>estimate<\/th>\n      <th>p.value<\/th>\n      <th>pathway<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<p>Next we add these p-values to the plot.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pdx-scores-with-pvalues" class="quarto-figure quarto-figure-center quarto-float anchored" width="1536">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-pdx-scores-with-pvalues-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="validation_files/figure-html/fig-pdx-scores-with-pvalues-1.png" id="fig-pdx-scores-with-pvalues" class="img-fluid figure-img" width="1536">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig quarto-uncaptioned" id="fig-pdx-scores-with-pvalues-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3.24
</figcaption>
</figure>
</div>
</div>
</div>
<p>We now further validate the results with the stainings performed on those samples. We will compare the differences with the stainings and the number of PR+ tumors cells.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-quantification-pdx-pr" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-quantification-pdx-pr-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="validation_files/figure-html/fig-quantification-pdx-pr-1.png" class="img-fluid figure-img" width="864">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-quantification-pdx-pr-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3.25: Quantification of PR+ cells based on DAB stainings performed on the patient derived xenografts.
</figcaption>
</figure>
</div>
</div>
</div>
<p>We see that T111 has a big increase in absolute values. T113 as well has a big increase, but has a baseline level already low and it does not go to higher values. Probably the reason there is not much difference in the fourth component of the PDX T113 is because of the low Ki67 index.</p>
<p>Based on the data shown here we see that there is an increase in proliferation upon P4 treatment for T111 and T113. Moreover, P4 reduced estrogen signaling and also change the third component of the three PDXs to a smaller value, towards a more basal like region. Overall, the molecular landscape is able to capture the differences between the treatments.</p>
<p>Lastly we check changes in the embedding upon E2 treatment.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pca-pdx-only-embedding-e2-ctrl" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-pca-pdx-only-embedding-e2-ctrl-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="validation_files/figure-html/fig-pca-pdx-only-embedding-e2-ctrl-1.png" class="img-fluid figure-img" width="1152">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-pca-pdx-only-embedding-e2-ctrl-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3.26: Embedding of only PDX control and E2 treated samples
</figcaption>
</figure>
</div>
</div>
</div>
<p>It looks like there is not big changes in the molecular landscape as there was for samples treated with progesterone.</p>
<p>And lastly we show for Ki67 as well.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-quantification-pdx-ki67" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-quantification-pdx-ki67-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="validation_files/figure-html/fig-quantification-pdx-ki67-1.png" class="img-fluid figure-img" width="672">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-quantification-pdx-ki67-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3.27: Quantification of Ki67+ cells based on DAB stainings performed on the patient derived xenografts.
</figcaption>
</figure>
</div>
</div>
</div>
<p>And below we plot both results together with the respective p-values.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-staining-quant-pr-ki67" class="quarto-figure quarto-figure-center quarto-float anchored" width="1536">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-staining-quant-pr-ki67-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="validation_files/figure-html/fig-staining-quant-pr-ki67-1.png" id="fig-staining-quant-pr-ki67" class="img-fluid figure-img" width="1536">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig quarto-uncaptioned" id="fig-staining-quant-pr-ki67-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3.28
</figcaption>
</figure>
</div>
</div>
</div>
<section id="growth-curves" class="level3" data-number="3.4.1">
<h3 data-number="3.4.1" class="anchored" data-anchor-id="growth-curves"><span class="header-section-number">3.4.1</span> Growth curves</h3>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pdx-estimates-ivis-together" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-pdx-estimates-ivis-together-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="validation_files/figure-html/fig-pdx-estimates-ivis-together-1.png" class="img-fluid figure-img" width="1248">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-pdx-estimates-ivis-together-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3.29: Estimates of the 6 PDXs together. Comparison between P4 vs CTRL.
</figcaption>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pdx-curves-ivis-together" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-pdx-curves-ivis-together-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="validation_files/figure-html/fig-pdx-curves-ivis-together-1.png" class="img-fluid figure-img" width="1152">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-pdx-curves-ivis-together-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3.30: Average growth curves for each PDX individually with their estimates. The credible intervals corresponds to the 25th and 50th CIs.
</figcaption>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="normal-samples" class="level2" data-number="3.5">
<h2 data-number="3.5" class="anchored" data-anchor-id="normal-samples"><span class="header-section-number">3.5</span> Normal samples</h2>
<p>We now further validate the molecular landscape on the normal samples that were sequenced using the same pipeline as the SCANB data. These are samples obtained from reduction mammoplasty of swiss women and were sent to Sweden for the tissue processing and sequecing. The hypotheses here is that the samples will be embedded in the region of the normal like BC samples.</p>
<p>When checking the total number of genes missing, there are only 0 genes missing, i.e., no genes were missing. This makes sense as these samples were processed in the same way as the ones from SCANB.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-normal-cohort" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-normal-cohort-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="validation_files/figure-html/fig-normal-cohort-1.png" class="img-fluid figure-img" width="672">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-normal-cohort-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3.31: Biplot of first two PCA components from Normal Swiss Cohort, TCGA, SCANB and METABRIC. The normal cohort is highlighted in the plot and has bigger dot size.
</figcaption>
</figure>
</div>
</div>
</div>
<p>As previously, <a href="#fig-pca-normal-cohort" class="quarto-xref">Figure&nbsp;<span>3.32</span></a> shows the biplot of PC3 and PC4, highlighting the well mixing of the cohorts. As expected the samples are overall in the luminal B region, since these are tumor cells in a proliferative state.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pca-normal-cohort" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-pca-normal-cohort-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="validation_files/figure-html/fig-pca-normal-cohort-1.png" class="img-fluid figure-img" width="864">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-pca-normal-cohort-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3.32: Embedding of all samples from TCGA, SCNAB and METABRIC including the Normal samples on top.
</figcaption>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pca-normal-pam50" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-pca-normal-pam50-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="validation_files/figure-html/fig-pca-normal-pam50-1.png" class="img-fluid figure-img" width="672">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-pca-normal-pam50-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3.33: Embedding of Normal Swiss cohort with the other cohorts colored by the PAM50.
</figcaption>
</figure>
</div>
</div>
</div>
<p>And since we finished getting all the plots for the second figure in the paper, we combine it below.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-all-fig2" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-all-fig2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="validation_files/figure-html/fig-all-fig2-1.png" class="img-fluid figure-img" width="2112">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-all-fig2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3.34: All figures combined for Fig2 in the paper.
</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="third-component-as-a-prognostic-measure" class="level2" data-number="3.6">
<h2 data-number="3.6" class="anchored" data-anchor-id="third-component-as-a-prognostic-measure"><span class="header-section-number">3.6</span> Third component as a prognostic measure?</h2>
<p>We have seen that estrogen signaling is a continuous measure across the third component. The farthest to the right you are the higher the ER signaling scores are also. Here we hypothesized then that the third component is correlated to the prognostic nature of the estrogen signaling score. For this we select only samples with PC3 higher than 0, as these would be the ones that are from luminal A and B subtypes and we use PC3 as a score in the survival analysis.</p>
<div class="cell">
<div class="cell-output-display">
<div>
<table class="lightable-classic table table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">cohort</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">number_patients</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">nb_events</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">SCANB</td>
<td style="text-align: right;">2777</td>
<td style="text-align: right;">456</td>
</tr>
<tr class="even">
<td style="text-align: left;">METABRIC</td>
<td style="text-align: right;">724</td>
<td style="text-align: right;">445</td>
</tr>
<tr class="odd">
<td style="text-align: left;">SCANB High ER</td>
<td style="text-align: right;">2253</td>
<td style="text-align: right;">347</td>
</tr>
</tbody>
</table>


</div>
</div>
</div>
<p>First we check the correlation of PC3 with estrogen response early and G2M checkpoint.</p>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="validation_files/figure-html/unnamed-chunk-61-1.png" class="img-fluid figure-img" width="1152"></p>
</figure>
</div>
</div>
</div>
<p>There is indeed the correlation with the components but still a lot of variability. So it will be unlikely to see such strong effects on the third component.</p>
<p>The table below shows the results for the survival analysis using all the samples from SCANB and METABRIC coming from patients that received only endocrine therapy and whose samples had a <span class="math inline">\(PC3 &gt; 0\)</span>. OS means overall survival and RFS means recurrence free survival.</p>
<div class="cell">
<div class="cell-output-display">
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-e495d9ce40cee3ca0240" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-e495d9ce40cee3ca0240">{"x":{"filter":"top","vertical":false,"filterHTML":"<tr>\n  <td><\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0.171874745942846\" data-max=\"0.988032640744915\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0.090750504939014\" data-max=\"0.916897422936799\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0.325518059792283\" data-max=\"1.07104187202532\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"4.5643692e-08\" data-max=\"0.75214967813996\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n<\/tr>","data":[["1","2","3","4","5","6","7","8","9","10","11","12"],["os","os","os","rfs","rfs","rfs","os","os","os","rfs","rfs","rfs"],["metabric","scanb","scanb_high_er","metabric","scanb","scanb_high_er","metabric","scanb","scanb_high_er","metabric","scanb","scanb_high_er"],["PC3","PC3","PC3","PC3","PC3","PC3","SET_ERPR","SET_ERPR","SET_ERPR","SET_ERPR","SET_ERPR","SET_ERPR"],[0.965196820738388,0.988032640744915,0.982828135587734,0.9276791213054461,0.778266395163085,0.7850264044905541,0.464246247797768,0.520546572781487,0.457731007834782,0.342276782189972,0.199605252620954,0.171874745942846],[0.905520711007731,0.916897422936799,0.901879907156442,0.853412922891882,0.676645079809247,0.671311330765721,0.318907386453501,0.37189423386866,0.312272360226255,0.217175112095403,0.112021321536825,0.0907505049390141],[1.02880573733839,1.06468670841128,1.07104187202532,1.0084081562649,0.895149613754518,0.9180039536118531,0.675821845931848,0.728617735251709,0.67094530999042,0.539442086599947,0.355666727791436,0.325518059792283],[0.276663895475407,0.75214967813996,0.692867249787021,0.0778512701200417,0.000445582620611866,0.00243324776660194,6.200072928721301e-05,0.00014162294268216,6.193225070653279e-05,3.85220744164788e-06,4.5643692033737e-08,6.50569533394187e-08]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>type_analysis<\/th>\n      <th>cohort<\/th>\n      <th>term<\/th>\n      <th>HR<\/th>\n      <th>conf.low<\/th>\n      <th>conf.high<\/th>\n      <th>p.value<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"pageLength":20,"columnDefs":[{"targets":7,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatSignif(data, 2, 3, \",\", \".\", null);\n  }"},{"targets":4,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 2, 3, \",\", \".\", null);\n  }"},{"targets":5,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 2, 3, \",\", \".\", null);\n  }"},{"targets":6,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 2, 3, \",\", \".\", null);\n  }"},{"className":"dt-right","targets":[4,5,6,7]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true,"lengthMenu":[10,20,25,50,100]},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":["options.columnDefs.0.render","options.columnDefs.1.render","options.columnDefs.2.render","options.columnDefs.3.render"],"jsHooks":[]}</script>
</div>
</div>
<p>We see that in all cases the HR for PC3 is below 1, for OS the values are 0.98 and 0.97 and the HR and for RFS from 0.80 to 0.93. What is interesting to note here is that in all cases the confidence interval is crossing 1, as expected due to the variability of the correlation with SET ER/PR and estrogen response early. Remember that the hazard ratio scale is not linear. Indeed it seems that the more a sample is on the right side of the plot the better is its outcome. Notice that the scale for HR is different for PC3 and SET ER/PR, since one is a positive score ranging from 0 to 8 and the other is a score ranging from -1 to 1. If we were to rescale SET ER/PR to an interval between 0 to 10 by multiplying by 5 and then summing 5, we would get hazard ratios in the range of 0.8. Moreover, estrogen signaling is not the only thing fully explaining the third component as it can be seen from <a href="pca_merging.html#fig-pca-mol-land-pathways" class="quarto-xref">Figure&nbsp;<span>2.35</span></a>. In this case, G2M Checkpoint and PI3K signaling have also a role in the third component.</p>
<p>The following figures correspond to the overall survival analysis and recurrence free survival analysis from the SCANB and METABRIC.</p>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="validation_files/figure-html/unnamed-chunk-65-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="validation_files/figure-html/unnamed-chunk-66-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="cell-output-display">
<div id="fig-os-rfs-pcs" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-os-rfs-pcs-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="validation_files/figure-html/fig-os-rfs-pcs-1.png" class="img-fluid figure-img" width="960">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-os-rfs-pcs-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3.35: Overall and recurrence free survival analysis of the principal components PC3 and PC4. The rows correspond to a different PC and cohort and if the case sub cohort.
</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="high-risk-patients" class="level2" data-number="3.7">
<h2 data-number="3.7" class="anchored" data-anchor-id="high-risk-patients"><span class="header-section-number">3.7</span> High risk patients</h2>
<p>The Prosigna risk of recurrence score takes into account the tumor stage and the molecular subtype, this is the ROR-C and the formula from the original paper is presented below:</p>
<p><span class="math display">\[
ROR-C = 0.05\times basal + 0.11 \times HER2 - 0.23 \times LumA + 0.09 \times
LumB  + 0.17 \times Tumor size
\]</span></p>
<p>We see that if a tumor sample has a high correlation to the luminal A subtype it will decrease the score, whereas for all other cases there is an increase in the risk. We want to evaluate if the position in the molecular landscape is somehow correlated with the risk score, specially in the forth component for luminal A patients.</p>
<p>In <span class="citation" data-cites="SCANB2022">(<a href="references.html#ref-SCANB2022" role="doc-biblioref">Staaf et al. 2022</a>)</span> they showed that they are able to predict the Prosigna binary categories high or low/intermediate risk for the patients based only on the RNA-seq provided by their pipeline. We use these categories here to understand the molecular scores of the samples and also their correlation with the position and risk groups. All the following analysis are done for ER+ BC patients <strong>only</strong>.</p>
<p>First we stratify the risk groups by the molecular subtype.</p>
<div class="cell">
<div class="cell-output-display">
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-ab62cd6a598f98c165fc" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-ab62cd6a598f98c165fc">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6"],["basal","her2","lumb","luma","normal","Total"],["88%    (50)","88%   (396)","85% (1,463)","14%   (399)","15%   (106)","41% (2,414)"],["12%     (7)","12%    (52)","15%   (257)","86% (2,549)","85%   (612)","59% (3,477)"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>pam50<\/th>\n      <th>High<\/th>\n      <th>Low/Intermediate<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<p>As expected the majority of luminal A patients are low/intermediate risk, on the other hand most of the luminal B patients are considered of high risk.</p>
<p>If we look more specifically for luminal A patients and stratify on the tumor stage we get the following numbers.</p>
<div class="cell">
<div class="cell-output-display">
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-bffde46c0314b1dde605" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-bffde46c0314b1dde605">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4"],["T1","T2","T3","Total"],["8% (184)","26% (181)","44%  (32)","13% (397)"],["92% (1,984)","74%   (519)","56%    (41)","87% (2,544)"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>tumor_stage<\/th>\n      <th>High<\/th>\n      <th>Low/Intermediate<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<p>We see that for tumor stage 1 the number of high individuals is lower in percentage than in tumor stage 3, as expected. We now plot (<a href="#fig-luma-scanb-g2m-ts" class="quarto-xref">Figure&nbsp;<span>3.36</span></a>) for all these patients the G2M molecular score, an index of proliferation, stratified by the tumor stage.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-luma-scanb-g2m-ts" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-luma-scanb-g2m-ts-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="validation_files/figure-html/fig-luma-scanb-g2m-ts-1.png" class="img-fluid figure-img" width="1056">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-luma-scanb-g2m-ts-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3.36: Luminal A BC patients G2M scores stratified by tumor stage.
</figcaption>
</figure>
</div>
</div>
</div>
<p>We see that in all three cases in average the G2M scores are higher in the high risk groups than in the Low/Intermediate.</p>
<p>We saw previously that some of the G2M checkpoint genes have higher loadings for the fourth component. We now plot the PC4 components stratified by the risk groups for the luminal A patients (<a href="#fig-luma-scanb-pc4" class="quarto-xref">Figure&nbsp;<span>3.37</span></a>).</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-luma-scanb-pc4" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-luma-scanb-pc4-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="validation_files/figure-html/fig-luma-scanb-pc4-1.png" class="img-fluid figure-img" width="576">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-luma-scanb-pc4-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3.37: Luminal A BC patients PC4 values stratified by risk category.
</figcaption>
</figure>
</div>
</div>
</div>
<p>Indeed, in average the higher the PC4, the more likely it is to be in the High risk group. And now we compare the G2M Checkpoint, Estrogen response early and random 200 signature for each molecular subtype separately (<a href="#fig-all-scanb-g2m-ere" class="quarto-xref">Figure&nbsp;<span>3.38</span></a>) of ER+ BC patients.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-all-scanb-g2m-ere" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-all-scanb-g2m-ere-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="validation_files/figure-html/fig-all-scanb-g2m-ere-1.png" class="img-fluid figure-img" width="1344">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-all-scanb-g2m-ere-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3.38: ER+ BC patients and molecular scores stratified by molecular subtype and binary risk categories.
</figcaption>
</figure>
</div>
</div>
</div>
<p>We see that in this case the binary risk category has distinct distributions for the G2M checkpoint for all molecular subtypes, in particular with the basal like.</p>


<!-- -->

<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list" style="display: none">
<div id="ref-Kan2018" class="csl-entry" role="listitem">
Kan, Zhengyan, Ying Ding, Jinho Kim, Hae Hyun Jung, Woosung Chung, Samir Lal, Soonweng Cho, et al. 2018. <span>“Multi-Omics Profiling of Younger Asian Breast Cancers Reveals Distinctive Molecular Signatures.”</span> <em>Nature Communications</em> 9 (1). <a href="https://doi.org/10.1038/s41467-018-04129-4">https://doi.org/10.1038/s41467-018-04129-4</a>.
</div>
<div id="ref-Scabia2022" class="csl-entry" role="listitem">
Scabia, Valentina, Ayyakkannu Ayyanan, Fabio De Martino, Andrea Agnoletto, Laura Battista, Csaba Laszlo, Assia Treboux, et al. 2022. <span>“Estrogen Receptor Positive Breast Cancers Have Patient Specific Hormone Sensitivities and Rely on Progesterone Receptor.”</span> <em>Nature Communications</em> 13 (1). <a href="https://doi.org/10.1038/s41467-022-30898-0">https://doi.org/10.1038/s41467-022-30898-0</a>.
</div>
<div id="ref-Sflomos2016" class="csl-entry" role="listitem">
Sflomos, George, Valerian Dormoy, Tauno Metsalu, Rachel Jeitziner, Laura Battista, Valentina Scabia, Wassim Raffoul, et al. 2016. <span>“A Preclinical Model for <span>ER</span><span class="math inline">\(\upalpha\)</span>-Positive Breast Cancer Points to the Epithelial Microenvironment as Determinant of Luminal Phenotype and Hormone Response.”</span> <em>Cancer Cell</em> 29 (3): 407–22. <a href="https://doi.org/10.1016/j.ccell.2016.02.002">https://doi.org/10.1016/j.ccell.2016.02.002</a>.
</div>
<div id="ref-SCANB2022" class="csl-entry" role="listitem">
Staaf, Johan, Jari Häkkinen, Cecilia Hegardt, Lao H. Saal, Siker Kimbung, Ingrid Hedenfalk, Tonje Lien, et al. 2022. <span>“<span>RNA</span> Sequencing-Based Single Sample Predictors of Molecular Subtype and Risk of Recurrence for Clinical Assessment of Early-Stage Breast Cancer.”</span> <em>Npj Breast Cancer</em> 8 (1). <a href="https://doi.org/10.1038/s41523-022-00465-3">https://doi.org/10.1038/s41523-022-00465-3</a>.
</div>
</div>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="../chapters/pca_merging.html" class="pagination-link" aria-label="A new framework for personalized medicine: integrating">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">A new framework for personalized medicine: integrating</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../chapters/scoring.html" class="pagination-link" aria-label="Pathway activity in a personalized context">
        <span class="nav-page-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Pathway activity in a personalized context</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb3" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu"># Validating the molecular landscape</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>In the last chapters only TCGA and METABRIC samples were used in the training</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>and testing of the PCA projection. To completely validate, we use another</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>big cohort, SCANB, that was not used in the PCA fitting and testing before.</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>This way we ensure that the embedding is independent of the cohort. Moreover,</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>we use the SMC cohort <span class="co">[</span><span class="ot">@Kan2018</span><span class="co">]</span>, whose patients are from South Korea.</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>We show how well it is embedded, and that </span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>is not dependent on population characteristics. </span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a><span class="in">```{r setup}</span></span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a><span class="in">renv::restore()</span></span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a><span class="in">library(tidyverse)</span></span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a><span class="in">library(ggplot2)</span></span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a><span class="in">library(PCAtools)</span></span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a><span class="in">library(singscore)</span></span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a><span class="in">library(SummarizedExperiment)</span></span>
<span id="cb3-21"><a href="#cb3-21" aria-hidden="true" tabindex="-1"></a><span class="in">library(tximport)</span></span>
<span id="cb3-22"><a href="#cb3-22" aria-hidden="true" tabindex="-1"></a><span class="in">library(survival)</span></span>
<span id="cb3-23"><a href="#cb3-23" aria-hidden="true" tabindex="-1"></a><span class="in">library(forestplot)</span></span>
<span id="cb3-24"><a href="#cb3-24" aria-hidden="true" tabindex="-1"></a><span class="in">library(ggplotify)</span></span>
<span id="cb3-25"><a href="#cb3-25" aria-hidden="true" tabindex="-1"></a><span class="in">library(gridExtra)</span></span>
<span id="cb3-26"><a href="#cb3-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-27"><a href="#cb3-27" aria-hidden="true" tabindex="-1"></a><span class="in">library(rstanarm)</span></span>
<span id="cb3-28"><a href="#cb3-28" aria-hidden="true" tabindex="-1"></a><span class="in">library(biogrowleR)</span></span>
<span id="cb3-29"><a href="#cb3-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-30"><a href="#cb3-30" aria-hidden="true" tabindex="-1"></a><span class="in">source("../R/utils.R")</span></span>
<span id="cb3-31"><a href="#cb3-31" aria-hidden="true" tabindex="-1"></a><span class="in">source("../R/first_run.R")</span></span>
<span id="cb3-32"><a href="#cb3-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-33"><a href="#cb3-33" aria-hidden="true" tabindex="-1"></a><span class="in"># Here we are using rds files from previous chapters, so we source</span></span>
<span id="cb3-34"><a href="#cb3-34" aria-hidden="true" tabindex="-1"></a><span class="in"># in any case the load_rds_files.R and exclude all the associated files</span></span>
<span id="cb3-35"><a href="#cb3-35" aria-hidden="true" tabindex="-1"></a><span class="in"># that are generated in this current chapter</span></span>
<span id="cb3-36"><a href="#cb3-36" aria-hidden="true" tabindex="-1"></a><span class="in">name_document &lt;- "validation"</span></span>
<span id="cb3-37"><a href="#cb3-37" aria-hidden="true" tabindex="-1"></a><span class="in">sapply(</span></span>
<span id="cb3-38"><a href="#cb3-38" aria-hidden="true" tabindex="-1"></a><span class="in">    paste0("../../results/", c("plots", "rds_files", "tables"), "/", name_document),</span></span>
<span id="cb3-39"><a href="#cb3-39" aria-hidden="true" tabindex="-1"></a><span class="in">    dir.create,</span></span>
<span id="cb3-40"><a href="#cb3-40" aria-hidden="true" tabindex="-1"></a><span class="in">    showWarnings = FALSE,</span></span>
<span id="cb3-41"><a href="#cb3-41" aria-hidden="true" tabindex="-1"></a><span class="in">    recursive = TRUE</span></span>
<span id="cb3-42"><a href="#cb3-42" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-43"><a href="#cb3-43" aria-hidden="true" tabindex="-1"></a><span class="in">source("../R/load_rds_files.R")</span></span>
<span id="cb3-44"><a href="#cb3-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-45"><a href="#cb3-45" aria-hidden="true" tabindex="-1"></a><span class="in"># by setting the dev to png and pdf, this saves the figures in a specific</span></span>
<span id="cb3-46"><a href="#cb3-46" aria-hidden="true" tabindex="-1"></a><span class="in"># folder in both formats. moreover, since png is coming first, it shows</span></span>
<span id="cb3-47"><a href="#cb3-47" aria-hidden="true" tabindex="-1"></a><span class="in"># this figure when rendering the html. What is nice about this is that it</span></span>
<span id="cb3-48"><a href="#cb3-48" aria-hidden="true" tabindex="-1"></a><span class="in"># inherits the properties from the chunk to save the figure, so no need</span></span>
<span id="cb3-49"><a href="#cb3-49" aria-hidden="true" tabindex="-1"></a><span class="in"># to use ggplot2::ggsave to save the plots. this also works </span></span>
<span id="cb3-50"><a href="#cb3-50" aria-hidden="true" tabindex="-1"></a><span class="in">knitr::opts_chunk$set(dev = c('png', 'pdf'))</span></span>
<span id="cb3-51"><a href="#cb3-51" aria-hidden="true" tabindex="-1"></a><span class="in">options(bitmapType = 'cairo')</span></span>
<span id="cb3-52"><a href="#cb3-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-53"><a href="#cb3-53" aria-hidden="true" tabindex="-1"></a><span class="in">mol_subs &lt;- c(</span></span>
<span id="cb3-54"><a href="#cb3-54" aria-hidden="true" tabindex="-1"></a><span class="in">    "basal" = "Basal-like", </span></span>
<span id="cb3-55"><a href="#cb3-55" aria-hidden="true" tabindex="-1"></a><span class="in">    "her2"= "HER2-enriched", </span></span>
<span id="cb3-56"><a href="#cb3-56" aria-hidden="true" tabindex="-1"></a><span class="in">    "lumb" = "LumB", </span></span>
<span id="cb3-57"><a href="#cb3-57" aria-hidden="true" tabindex="-1"></a><span class="in">    "luma" = "LumA", </span></span>
<span id="cb3-58"><a href="#cb3-58" aria-hidden="true" tabindex="-1"></a><span class="in">    "normal" = "Normal-like",</span></span>
<span id="cb3-59"><a href="#cb3-59" aria-hidden="true" tabindex="-1"></a><span class="in">    "claudin-low" = "Claudin-low"</span></span>
<span id="cb3-60"><a href="#cb3-60" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-61"><a href="#cb3-61" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-62"><a href="#cb3-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-63"><a href="#cb3-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-64"><a href="#cb3-64" aria-hidden="true" tabindex="-1"></a><span class="fu">## SCANB embedding</span></span>
<span id="cb3-65"><a href="#cb3-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-66"><a href="#cb3-66" aria-hidden="true" tabindex="-1"></a>The embedding was already calculate before when calculating the fit and</span>
<span id="cb3-67"><a href="#cb3-67" aria-hidden="true" tabindex="-1"></a>tests, since the datasets were organized and loaded together. We also</span>
<span id="cb3-68"><a href="#cb3-68" aria-hidden="true" tabindex="-1"></a>know already that all the 1044 genes are available in the SCANB cohort. </span>
<span id="cb3-69"><a href="#cb3-69" aria-hidden="true" tabindex="-1"></a>SCANB is well mixed with all cohorts as @fig-pca-scanb shows.</span>
<span id="cb3-70"><a href="#cb3-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-73"><a href="#cb3-73" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-74"><a href="#cb3-74" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-pca-scanb</span></span>
<span id="cb3-75"><a href="#cb3-75" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: PCA embedding colored by cohort. All samples from TCGA, METABRIC</span></span>
<span id="cb3-76"><a href="#cb3-76" aria-hidden="true" tabindex="-1"></a><span class="co">#|     and SCANB were used. </span></span>
<span id="cb3-77"><a href="#cb3-77" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_pca_coordinates</span>(</span>
<span id="cb3-78"><a href="#cb3-78" aria-hidden="true" tabindex="-1"></a>    df_pca_coordinates <span class="sc">%&gt;%</span> </span>
<span id="cb3-79"><a href="#cb3-79" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">slice</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">n</span>()),</span>
<span id="cb3-80"><a href="#cb3-80" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"cohort"</span>,</span>
<span id="cb3-81"><a href="#cb3-81" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC3"</span>,</span>
<span id="cb3-82"><a href="#cb3-82" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC4"</span>,</span>
<span id="cb3-83"><a href="#cb3-83" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">1</span>,</span>
<span id="cb3-84"><a href="#cb3-84" aria-hidden="true" tabindex="-1"></a>    <span class="at">base_size =</span> <span class="dv">10</span>,</span>
<span id="cb3-85"><a href="#cb3-85" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="fu">paste0</span>(</span>
<span id="cb3-86"><a href="#cb3-86" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Embedding of all samples from TCGA,</span><span class="sc">\n</span><span class="st">"</span>,</span>
<span id="cb3-87"><a href="#cb3-87" aria-hidden="true" tabindex="-1"></a>        <span class="st">"METABRIC and SCANB"</span></span>
<span id="cb3-88"><a href="#cb3-88" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb3-89"><a href="#cb3-89" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span> ggplot2<span class="sc">::</span><span class="fu">scale_color_viridis_d</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb3-90"><a href="#cb3-90" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">15</span>) <span class="sc">+</span></span>
<span id="cb3-91"><a href="#cb3-91" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span></span>
<span id="cb3-92"><a href="#cb3-92" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb3-93"><a href="#cb3-93" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-94"><a href="#cb3-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-95"><a href="#cb3-95" aria-hidden="true" tabindex="-1"></a>Ideally there would be some overlap between SCANB and TCGA already in the </span>
<span id="cb3-96"><a href="#cb3-96" aria-hidden="true" tabindex="-1"></a>first component. Moreover, if there isn't, the cohort should be closer</span>
<span id="cb3-97"><a href="#cb3-97" aria-hidden="true" tabindex="-1"></a>to TCGA than to METABRIC, due to technologies and normalization procedures,</span>
<span id="cb3-98"><a href="#cb3-98" aria-hidden="true" tabindex="-1"></a>as both of them are log FPKM. Even though SCANB was processed in a way </span>
<span id="cb3-99"><a href="#cb3-99" aria-hidden="true" tabindex="-1"></a>to get directly FPKM values, by using cufflinks, TCGA was normalized to get</span>
<span id="cb3-100"><a href="#cb3-100" aria-hidden="true" tabindex="-1"></a>log FPKM as well, as discussed previously. @fig-pca-scanb-pc1 shows</span>
<span id="cb3-101"><a href="#cb3-101" aria-hidden="true" tabindex="-1"></a>how close SCANB is from TCGA, moreover there is already some overlap. The </span>
<span id="cb3-102"><a href="#cb3-102" aria-hidden="true" tabindex="-1"></a>variability in the first component is bigger for METABRIC compared to</span>
<span id="cb3-103"><a href="#cb3-103" aria-hidden="true" tabindex="-1"></a>TCGA and SCANB. Probably this is due to the technology processing, since</span>
<span id="cb3-104"><a href="#cb3-104" aria-hidden="true" tabindex="-1"></a>they have genes in the same scale. @fig-pca-no-norm contrasts the different</span>
<span id="cb3-105"><a href="#cb3-105" aria-hidden="true" tabindex="-1"></a>normalization procedures. If genes are not scaled correctly, the PCA</span>
<span id="cb3-106"><a href="#cb3-106" aria-hidden="true" tabindex="-1"></a>shows an inverse picture.</span>
<span id="cb3-107"><a href="#cb3-107" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-110"><a href="#cb3-110" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-111"><a href="#cb3-111" aria-hidden="true" tabindex="-1"></a>plots_paper <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb3-112"><a href="#cb3-112" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-113"><a href="#cb3-113" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-114"><a href="#cb3-114" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-117"><a href="#cb3-117" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-118"><a href="#cb3-118" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-pca-scanb-pc1</span></span>
<span id="cb3-119"><a href="#cb3-119" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: PCA embedding colored by cohort. The components used are the</span></span>
<span id="cb3-120"><a href="#cb3-120" aria-hidden="true" tabindex="-1"></a><span class="co">#|     first and second components.</span></span>
<span id="cb3-121"><a href="#cb3-121" aria-hidden="true" tabindex="-1"></a>plots_paper<span class="sc">$</span>scanb_pc1_pc2 <span class="ot">&lt;-</span> <span class="fu">plot_pca_coordinates</span>(</span>
<span id="cb3-122"><a href="#cb3-122" aria-hidden="true" tabindex="-1"></a>    df_pca_coordinates <span class="sc">%&gt;%</span></span>
<span id="cb3-123"><a href="#cb3-123" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb3-124"><a href="#cb3-124" aria-hidden="true" tabindex="-1"></a>            <span class="at">cohort =</span> <span class="fu">toupper</span>(cohort),</span>
<span id="cb3-125"><a href="#cb3-125" aria-hidden="true" tabindex="-1"></a>            <span class="at">cohort =</span> <span class="fu">ifelse</span>(cohort <span class="sc">==</span> <span class="st">"SCANB"</span>, <span class="st">"SCAN-B"</span>, cohort)</span>
<span id="cb3-126"><a href="#cb3-126" aria-hidden="true" tabindex="-1"></a>        ),</span>
<span id="cb3-127"><a href="#cb3-127" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"cohort"</span>,</span>
<span id="cb3-128"><a href="#cb3-128" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC1"</span>,</span>
<span id="cb3-129"><a href="#cb3-129" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC2"</span>,</span>
<span id="cb3-130"><a href="#cb3-130" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">1</span>,</span>
<span id="cb3-131"><a href="#cb3-131" aria-hidden="true" tabindex="-1"></a>    <span class="at">base_size =</span> <span class="dv">10</span>,</span>
<span id="cb3-132"><a href="#cb3-132" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="cn">NULL</span></span>
<span id="cb3-133"><a href="#cb3-133" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span> </span>
<span id="cb3-134"><a href="#cb3-134" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">color =</span> <span class="st">"Cohort"</span>) <span class="sc">+</span> </span>
<span id="cb3-135"><a href="#cb3-135" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">15</span>) <span class="sc">+</span></span>
<span id="cb3-136"><a href="#cb3-136" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span></span>
<span id="cb3-137"><a href="#cb3-137" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb3-138"><a href="#cb3-138" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-139"><a href="#cb3-139" aria-hidden="true" tabindex="-1"></a>plots_paper<span class="sc">$</span>scanb_pc1_pc2</span>
<span id="cb3-140"><a href="#cb3-140" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-141"><a href="#cb3-141" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-142"><a href="#cb3-142" aria-hidden="true" tabindex="-1"></a>@fig-pca-scanb-er-pam50 shows that the SCANB samples are also well mixed</span>
<span id="cb3-143"><a href="#cb3-143" aria-hidden="true" tabindex="-1"></a>regarding the clinical factors, including the $SET_{ER/PR}$ signature.</span>
<span id="cb3-144"><a href="#cb3-144" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-145"><a href="#cb3-145" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=18, fig.height=12}</span></span>
<span id="cb3-146"><a href="#cb3-146" aria-hidden="true" tabindex="-1"></a><span class="in">#| label: fig-pca-scanb-er-pam50</span></span>
<span id="cb3-147"><a href="#cb3-147" aria-hidden="true" tabindex="-1"></a><span class="in">#| fig-cap: PCA embedding of all samples from TCGA, SCANB and METABRIC.</span></span>
<span id="cb3-148"><a href="#cb3-148" aria-hidden="true" tabindex="-1"></a><span class="in">#|     (A) Colored by cohort,</span></span>
<span id="cb3-149"><a href="#cb3-149" aria-hidden="true" tabindex="-1"></a><span class="in">#|     (B) colored by ER status, (C) colored by PAM50 molecular subtype,</span></span>
<span id="cb3-150"><a href="#cb3-150" aria-hidden="true" tabindex="-1"></a><span class="in">#|     (D) colored by the $SET_{ER/PR}$ signature and only SCANB samples.</span></span>
<span id="cb3-151"><a href="#cb3-151" aria-hidden="true" tabindex="-1"></a><span class="in">size &lt;- 2</span></span>
<span id="cb3-152"><a href="#cb3-152" aria-hidden="true" tabindex="-1"></a><span class="in">base_size &lt;- 20</span></span>
<span id="cb3-153"><a href="#cb3-153" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-154"><a href="#cb3-154" aria-hidden="true" tabindex="-1"></a><span class="in">plots_with_scanb &lt;- sapply(</span></span>
<span id="cb3-155"><a href="#cb3-155" aria-hidden="true" tabindex="-1"></a><span class="in">    c("cohort", "er_status", "pam50", "SET_ERPR"), </span></span>
<span id="cb3-156"><a href="#cb3-156" aria-hidden="true" tabindex="-1"></a><span class="in">    plot_pca_coordinates,</span></span>
<span id="cb3-157"><a href="#cb3-157" aria-hidden="true" tabindex="-1"></a><span class="in">    df_pca = df_pca_coordinates %&gt;% </span></span>
<span id="cb3-158"><a href="#cb3-158" aria-hidden="true" tabindex="-1"></a><span class="in">        dplyr::filter(</span></span>
<span id="cb3-159"><a href="#cb3-159" aria-hidden="true" tabindex="-1"></a><span class="in">            pam50 %in% c("luma", "lumb", "basal", "her2", "normal")</span></span>
<span id="cb3-160"><a href="#cb3-160" aria-hidden="true" tabindex="-1"></a><span class="in">        ) %&gt;%</span></span>
<span id="cb3-161"><a href="#cb3-161" aria-hidden="true" tabindex="-1"></a><span class="in">        dplyr::slice(sample(1:n())) %&gt;%</span></span>
<span id="cb3-162"><a href="#cb3-162" aria-hidden="true" tabindex="-1"></a><span class="in">        dplyr::mutate(</span></span>
<span id="cb3-163"><a href="#cb3-163" aria-hidden="true" tabindex="-1"></a><span class="in">            cohort = toupper(cohort),</span></span>
<span id="cb3-164"><a href="#cb3-164" aria-hidden="true" tabindex="-1"></a><span class="in">            cohort = ifelse(cohort == "SCANB", "SCAN-B", cohort)</span></span>
<span id="cb3-165"><a href="#cb3-165" aria-hidden="true" tabindex="-1"></a><span class="in">        ),</span></span>
<span id="cb3-166"><a href="#cb3-166" aria-hidden="true" tabindex="-1"></a><span class="in">    x = "PC3", </span></span>
<span id="cb3-167"><a href="#cb3-167" aria-hidden="true" tabindex="-1"></a><span class="in">    y = "PC4",</span></span>
<span id="cb3-168"><a href="#cb3-168" aria-hidden="true" tabindex="-1"></a><span class="in">    size = size,</span></span>
<span id="cb3-169"><a href="#cb3-169" aria-hidden="true" tabindex="-1"></a><span class="in">    base_size = base_size,</span></span>
<span id="cb3-170"><a href="#cb3-170" aria-hidden="true" tabindex="-1"></a><span class="in">    title = NULL,</span></span>
<span id="cb3-171"><a href="#cb3-171" aria-hidden="true" tabindex="-1"></a><span class="in">    USE.NAMES = TRUE, </span></span>
<span id="cb3-172"><a href="#cb3-172" aria-hidden="true" tabindex="-1"></a><span class="in">    simplify = FALSE</span></span>
<span id="cb3-173"><a href="#cb3-173" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-174"><a href="#cb3-174" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-175"><a href="#cb3-175" aria-hidden="true" tabindex="-1"></a><span class="in">plots_with_scanb$cohort &lt;- plots_with_scanb$cohort +</span></span>
<span id="cb3-176"><a href="#cb3-176" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::labs(color = "Cohort") +</span></span>
<span id="cb3-177"><a href="#cb3-177" aria-hidden="true" tabindex="-1"></a><span class="in">    change_plot_aes_point() +</span></span>
<span id="cb3-178"><a href="#cb3-178" aria-hidden="true" tabindex="-1"></a><span class="in">    change_guides_point()</span></span>
<span id="cb3-179"><a href="#cb3-179" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-180"><a href="#cb3-180" aria-hidden="true" tabindex="-1"></a><span class="in">plots_with_scanb$pam50 &lt;- plots_with_scanb$pam50 +</span></span>
<span id="cb3-181"><a href="#cb3-181" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::scale_color_manual(</span></span>
<span id="cb3-182"><a href="#cb3-182" aria-hidden="true" tabindex="-1"></a><span class="in">        values = get_colors_pam50(plots_with_scanb$pam50$data),</span></span>
<span id="cb3-183"><a href="#cb3-183" aria-hidden="true" tabindex="-1"></a><span class="in">        labels = mol_subs</span></span>
<span id="cb3-184"><a href="#cb3-184" aria-hidden="true" tabindex="-1"></a><span class="in">    ) +</span></span>
<span id="cb3-185"><a href="#cb3-185" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::labs(color = "Molecular\nsubtype") +</span></span>
<span id="cb3-186"><a href="#cb3-186" aria-hidden="true" tabindex="-1"></a><span class="in">    change_plot_aes_point() +</span></span>
<span id="cb3-187"><a href="#cb3-187" aria-hidden="true" tabindex="-1"></a><span class="in">    change_guides_point()</span></span>
<span id="cb3-188"><a href="#cb3-188" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-189"><a href="#cb3-189" aria-hidden="true" tabindex="-1"></a><span class="in">plots_with_scanb$er_status &lt;-  plots_with_scanb$er_status +</span></span>
<span id="cb3-190"><a href="#cb3-190" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::scale_color_viridis_d(</span></span>
<span id="cb3-191"><a href="#cb3-191" aria-hidden="true" tabindex="-1"></a><span class="in">        labels = c("neg" = "Negative", "pos" = "Positive")</span></span>
<span id="cb3-192"><a href="#cb3-192" aria-hidden="true" tabindex="-1"></a><span class="in">    ) +</span></span>
<span id="cb3-193"><a href="#cb3-193" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::labs(color = "ER IHC") +</span></span>
<span id="cb3-194"><a href="#cb3-194" aria-hidden="true" tabindex="-1"></a><span class="in">    change_plot_aes_point() +</span></span>
<span id="cb3-195"><a href="#cb3-195" aria-hidden="true" tabindex="-1"></a><span class="in">    change_guides_point()</span></span>
<span id="cb3-196"><a href="#cb3-196" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-197"><a href="#cb3-197" aria-hidden="true" tabindex="-1"></a><span class="in">plots_with_scanb$SET_ERPR &lt;- df_pca_coordinates %&gt;% </span></span>
<span id="cb3-198"><a href="#cb3-198" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::filter(cohort %in% c("scanb")) %&gt;%</span></span>
<span id="cb3-199"><a href="#cb3-199" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::ggplot(aes_string(x = "PC3", y = "PC4", z = "SET_ERPR")) + </span></span>
<span id="cb3-200"><a href="#cb3-200" aria-hidden="true" tabindex="-1"></a><span class="in">        ggplot2::stat_summary_hex(bins = 25) +</span></span>
<span id="cb3-201"><a href="#cb3-201" aria-hidden="true" tabindex="-1"></a><span class="in">        ggplot2::scale_fill_viridis_c() +</span></span>
<span id="cb3-202"><a href="#cb3-202" aria-hidden="true" tabindex="-1"></a><span class="in">        ggplot2::labs(</span></span>
<span id="cb3-203"><a href="#cb3-203" aria-hidden="true" tabindex="-1"></a><span class="in">            title = "Embedding of SCANB only",</span></span>
<span id="cb3-204"><a href="#cb3-204" aria-hidden="true" tabindex="-1"></a><span class="in">            fill = expression(SET[ER/PR])</span></span>
<span id="cb3-205"><a href="#cb3-205" aria-hidden="true" tabindex="-1"></a><span class="in">        ) + </span></span>
<span id="cb3-206"><a href="#cb3-206" aria-hidden="true" tabindex="-1"></a><span class="in">        ggplot2::theme_bw(base_size = base_size) +</span></span>
<span id="cb3-207"><a href="#cb3-207" aria-hidden="true" tabindex="-1"></a><span class="in">        change_plot_aes_point()</span></span>
<span id="cb3-208"><a href="#cb3-208" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-209"><a href="#cb3-209" aria-hidden="true" tabindex="-1"></a><span class="in">plots_paper$scanb_pc3_pc4 &lt;- plots_with_scanb$cohort</span></span>
<span id="cb3-210"><a href="#cb3-210" aria-hidden="true" tabindex="-1"></a><span class="in">plots_paper$er_ihc &lt;- plots_with_scanb$er_status</span></span>
<span id="cb3-211"><a href="#cb3-211" aria-hidden="true" tabindex="-1"></a><span class="in">plots_paper$mol_sub &lt;- plots_with_scanb$pam50</span></span>
<span id="cb3-212"><a href="#cb3-212" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-213"><a href="#cb3-213" aria-hidden="true" tabindex="-1"></a><span class="in">cowplot::plot_grid(</span></span>
<span id="cb3-214"><a href="#cb3-214" aria-hidden="true" tabindex="-1"></a><span class="in">    plotlist = plots_with_scanb,</span></span>
<span id="cb3-215"><a href="#cb3-215" aria-hidden="true" tabindex="-1"></a><span class="in">    ncol = 2</span></span>
<span id="cb3-216"><a href="#cb3-216" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-217"><a href="#cb3-217" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-218"><a href="#cb3-218" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-219"><a href="#cb3-219" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-220"><a href="#cb3-220" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-221"><a href="#cb3-221" aria-hidden="true" tabindex="-1"></a><span class="fu">## Missing genes</span></span>
<span id="cb3-222"><a href="#cb3-222" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-223"><a href="#cb3-223" aria-hidden="true" tabindex="-1"></a>Genes missing in publicly available datasets is very common. Usually this is</span>
<span id="cb3-224"><a href="#cb3-224" aria-hidden="true" tabindex="-1"></a>due to processing pipelines or even data quality, therefore genes are removed.</span>
<span id="cb3-225"><a href="#cb3-225" aria-hidden="true" tabindex="-1"></a>This should not be much of a problem, if the data is good enough, when </span>
<span id="cb3-226"><a href="#cb3-226" aria-hidden="true" tabindex="-1"></a>calculating the qPCR-like normalization as it was shown before. The problem</span>
<span id="cb3-227"><a href="#cb3-227" aria-hidden="true" tabindex="-1"></a>arises when multiplying the loadings obtained in the PCA with the </span>
<span id="cb3-228"><a href="#cb3-228" aria-hidden="true" tabindex="-1"></a>normalized expression of the sample. If several genes are missing, 0s are </span>
<span id="cb3-229"><a href="#cb3-229" aria-hidden="true" tabindex="-1"></a>added to the matrix and therefore the loadings for these genes cannot be </span>
<span id="cb3-230"><a href="#cb3-230" aria-hidden="true" tabindex="-1"></a>used. Since 1000 genes were used, we investigate the loadings of genes and </span>
<span id="cb3-231"><a href="#cb3-231" aria-hidden="true" tabindex="-1"></a>try to calculate a fuzziness score, indicating when the embedding should</span>
<span id="cb3-232"><a href="#cb3-232" aria-hidden="true" tabindex="-1"></a>be trusted if genes are missing, and which genes are missing. </span>
<span id="cb3-233"><a href="#cb3-233" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-234"><a href="#cb3-234" aria-hidden="true" tabindex="-1"></a><span class="fu">### Fuzziness score </span></span>
<span id="cb3-235"><a href="#cb3-235" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-236"><a href="#cb3-236" aria-hidden="true" tabindex="-1"></a>The idea of the fuzziness score is that we use the loadings of the missing</span>
<span id="cb3-237"><a href="#cb3-237" aria-hidden="true" tabindex="-1"></a>genes and calculate the cumulative sum of their absolute values. </span>
<span id="cb3-238"><a href="#cb3-238" aria-hidden="true" tabindex="-1"></a>The higher this value</span>
<span id="cb3-239"><a href="#cb3-239" aria-hidden="true" tabindex="-1"></a>the more it indicates the missing genes are important. We start by</span>
<span id="cb3-240"><a href="#cb3-240" aria-hidden="true" tabindex="-1"></a>removing random genes from random samples of the SCANB cohort. Each sample</span>
<span id="cb3-241"><a href="#cb3-241" aria-hidden="true" tabindex="-1"></a>will have a random number of genes removed from the set of 1044 genes. </span>
<span id="cb3-242"><a href="#cb3-242" aria-hidden="true" tabindex="-1"></a>Then two scores are calculated, one for PC3 and another for PC4. We also</span>
<span id="cb3-243"><a href="#cb3-243" aria-hidden="true" tabindex="-1"></a>calculate the fuzziness score for the top loading genes and the </span>
<span id="cb3-244"><a href="#cb3-244" aria-hidden="true" tabindex="-1"></a>low loadings genes, to compare these values.</span>
<span id="cb3-245"><a href="#cb3-245" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-246"><a href="#cb3-246" aria-hidden="true" tabindex="-1"></a>We start by comparing the scores from top loadings and low loading genes. For</span>
<span id="cb3-247"><a href="#cb3-247" aria-hidden="true" tabindex="-1"></a>this we remove the top 50 loadings and bottom 50 loadings, in terms of</span>
<span id="cb3-248"><a href="#cb3-248" aria-hidden="true" tabindex="-1"></a>absolute values. @fig-loadings-top-bottom shows the cumulative sum</span>
<span id="cb3-249"><a href="#cb3-249" aria-hidden="true" tabindex="-1"></a>of the absolute values from the top and bottom loadings respectively. See how </span>
<span id="cb3-250"><a href="#cb3-250" aria-hidden="true" tabindex="-1"></a>important the top loadings are compared to the bottom loadings. A way of</span>
<span id="cb3-251"><a href="#cb3-251" aria-hidden="true" tabindex="-1"></a>thinking of this is that the top loadings weights are much more important</span>
<span id="cb3-252"><a href="#cb3-252" aria-hidden="true" tabindex="-1"></a>when calculating the embedding.</span>
<span id="cb3-253"><a href="#cb3-253" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-256"><a href="#cb3-256" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-257"><a href="#cb3-257" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-loadings-top-bottom</span></span>
<span id="cb3-258"><a href="#cb3-258" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Cumulative sum of absolute values for the top loadings and</span></span>
<span id="cb3-259"><a href="#cb3-259" aria-hidden="true" tabindex="-1"></a><span class="co">#|     bottom loadings as a function of the number of selected genes.</span></span>
<span id="cb3-260"><a href="#cb3-260" aria-hidden="true" tabindex="-1"></a>which_loadings <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"PC3"</span>, <span class="st">"PC4"</span>)</span>
<span id="cb3-261"><a href="#cb3-261" aria-hidden="true" tabindex="-1"></a>n <span class="ot">&lt;-</span> <span class="dv">200</span></span>
<span id="cb3-262"><a href="#cb3-262" aria-hidden="true" tabindex="-1"></a>top_or_bottom <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"top"</span>, <span class="st">"bottom"</span>)</span>
<span id="cb3-263"><a href="#cb3-263" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-264"><a href="#cb3-264" aria-hidden="true" tabindex="-1"></a>fuzziness_scores_top <span class="ot">&lt;-</span> <span class="fu">sapply</span>(</span>
<span id="cb3-265"><a href="#cb3-265" aria-hidden="true" tabindex="-1"></a>    <span class="dv">1</span><span class="sc">:</span>n,</span>
<span id="cb3-266"><a href="#cb3-266" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(n, pca_fit){</span>
<span id="cb3-267"><a href="#cb3-267" aria-hidden="true" tabindex="-1"></a>        top_loadings <span class="ot">&lt;-</span> <span class="fu">apply</span>(</span>
<span id="cb3-268"><a href="#cb3-268" aria-hidden="true" tabindex="-1"></a>            pca_fit<span class="sc">$</span>loadings[, which_loadings],</span>
<span id="cb3-269"><a href="#cb3-269" aria-hidden="true" tabindex="-1"></a>            <span class="dv">2</span>,</span>
<span id="cb3-270"><a href="#cb3-270" aria-hidden="true" tabindex="-1"></a>            <span class="cf">function</span>(x, gene_names){</span>
<span id="cb3-271"><a href="#cb3-271" aria-hidden="true" tabindex="-1"></a>                gene_names[<span class="fu">order</span>(<span class="fu">abs</span>(x), <span class="at">decreasing =</span> <span class="cn">TRUE</span>)[<span class="dv">1</span><span class="sc">:</span>n]]</span>
<span id="cb3-272"><a href="#cb3-272" aria-hidden="true" tabindex="-1"></a>            },</span>
<span id="cb3-273"><a href="#cb3-273" aria-hidden="true" tabindex="-1"></a>            <span class="at">gene_names =</span> <span class="fu">rownames</span>(pca_fit<span class="sc">$</span>loadings)</span>
<span id="cb3-274"><a href="#cb3-274" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb3-275"><a href="#cb3-275" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb3-276"><a href="#cb3-276" aria-hidden="true" tabindex="-1"></a>        <span class="fu">get_fuzziness_score</span>(top_loadings, pca_fit, <span class="at">which_pcs =</span> <span class="dv">3</span><span class="sc">:</span><span class="dv">4</span>)</span>
<span id="cb3-277"><a href="#cb3-277" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb3-278"><a href="#cb3-278" aria-hidden="true" tabindex="-1"></a>    <span class="at">pca_fit =</span> pca_fit</span>
<span id="cb3-279"><a href="#cb3-279" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> t <span class="sc">%&gt;%</span> data.frame <span class="sc">%&gt;%</span> </span>
<span id="cb3-280"><a href="#cb3-280" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">colnames&lt;-</span><span class="st">`</span>(which_loadings) <span class="sc">%&gt;%</span></span>
<span id="cb3-281"><a href="#cb3-281" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">n =</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">n</span>())</span>
<span id="cb3-282"><a href="#cb3-282" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-283"><a href="#cb3-283" aria-hidden="true" tabindex="-1"></a>fuzziness_scores_bottom <span class="ot">&lt;-</span> <span class="fu">sapply</span>(</span>
<span id="cb3-284"><a href="#cb3-284" aria-hidden="true" tabindex="-1"></a>    <span class="dv">1</span><span class="sc">:</span>n,</span>
<span id="cb3-285"><a href="#cb3-285" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(n, pca_fit){</span>
<span id="cb3-286"><a href="#cb3-286" aria-hidden="true" tabindex="-1"></a>        top_loadings <span class="ot">&lt;-</span> <span class="fu">apply</span>(</span>
<span id="cb3-287"><a href="#cb3-287" aria-hidden="true" tabindex="-1"></a>            pca_fit<span class="sc">$</span>loadings[, which_loadings],</span>
<span id="cb3-288"><a href="#cb3-288" aria-hidden="true" tabindex="-1"></a>            <span class="dv">2</span>,</span>
<span id="cb3-289"><a href="#cb3-289" aria-hidden="true" tabindex="-1"></a>            <span class="cf">function</span>(x, gene_names){</span>
<span id="cb3-290"><a href="#cb3-290" aria-hidden="true" tabindex="-1"></a>                gene_names[<span class="fu">order</span>(<span class="fu">abs</span>(x), <span class="at">decreasing =</span> <span class="cn">FALSE</span>)[<span class="dv">1</span><span class="sc">:</span>n]]</span>
<span id="cb3-291"><a href="#cb3-291" aria-hidden="true" tabindex="-1"></a>            },</span>
<span id="cb3-292"><a href="#cb3-292" aria-hidden="true" tabindex="-1"></a>            <span class="at">gene_names =</span> <span class="fu">rownames</span>(pca_fit<span class="sc">$</span>loadings)</span>
<span id="cb3-293"><a href="#cb3-293" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb3-294"><a href="#cb3-294" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb3-295"><a href="#cb3-295" aria-hidden="true" tabindex="-1"></a>        <span class="fu">get_fuzziness_score</span>(top_loadings, pca_fit)</span>
<span id="cb3-296"><a href="#cb3-296" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb3-297"><a href="#cb3-297" aria-hidden="true" tabindex="-1"></a>    <span class="at">pca_fit =</span> pca_fit</span>
<span id="cb3-298"><a href="#cb3-298" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> t <span class="sc">%&gt;%</span> data.frame <span class="sc">%&gt;%</span> </span>
<span id="cb3-299"><a href="#cb3-299" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">colnames&lt;-</span><span class="st">`</span>(which_loadings) <span class="sc">%&gt;%</span></span>
<span id="cb3-300"><a href="#cb3-300" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">n =</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">n</span>())</span>
<span id="cb3-301"><a href="#cb3-301" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-302"><a href="#cb3-302" aria-hidden="true" tabindex="-1"></a>fuzziness_scores <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(</span>
<span id="cb3-303"><a href="#cb3-303" aria-hidden="true" tabindex="-1"></a>    <span class="fu">list</span>(<span class="at">top =</span> fuzziness_scores_top, <span class="at">bottom =</span> fuzziness_scores_bottom),</span>
<span id="cb3-304"><a href="#cb3-304" aria-hidden="true" tabindex="-1"></a>    <span class="at">.id =</span> <span class="st">"top_or_bottom"</span></span>
<span id="cb3-305"><a href="#cb3-305" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb3-306"><a href="#cb3-306" aria-hidden="true" tabindex="-1"></a>    <span class="at">top_or_bottom =</span> <span class="fu">factor</span>(top_or_bottom, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"top"</span>, <span class="st">"bottom"</span>))</span>
<span id="cb3-307"><a href="#cb3-307" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb3-308"><a href="#cb3-308" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-309"><a href="#cb3-309" aria-hidden="true" tabindex="-1"></a>fuzziness_scores <span class="sc">%&gt;%</span> tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(</span>
<span id="cb3-310"><a href="#cb3-310" aria-hidden="true" tabindex="-1"></a>    <span class="at">cols =</span> dplyr<span class="sc">::</span><span class="fu">all_of</span>(which_loadings),</span>
<span id="cb3-311"><a href="#cb3-311" aria-hidden="true" tabindex="-1"></a>    <span class="at">names_to =</span> <span class="st">"PC"</span>,</span>
<span id="cb3-312"><a href="#cb3-312" aria-hidden="true" tabindex="-1"></a>    <span class="at">values_to =</span> <span class="st">"fuzziness_score"</span></span>
<span id="cb3-313"><a href="#cb3-313" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb3-314"><a href="#cb3-314" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> n, <span class="at">y =</span> fuzziness_score, <span class="at">color =</span> PC)) <span class="sc">+</span></span>
<span id="cb3-315"><a href="#cb3-315" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_line</span>(<span class="at">size =</span> <span class="dv">3</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb3-316"><a href="#cb3-316" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_color_viridis_d</span>() <span class="sc">+</span></span>
<span id="cb3-317"><a href="#cb3-317" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>top_or_bottom) <span class="sc">+</span></span>
<span id="cb3-318"><a href="#cb3-318" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb3-319"><a href="#cb3-319" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"Number of genes"</span>,</span>
<span id="cb3-320"><a href="#cb3-320" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"Fuzziness score"</span>,</span>
<span id="cb3-321"><a href="#cb3-321" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">"Principal</span><span class="sc">\n</span><span class="st">component"</span></span>
<span id="cb3-322"><a href="#cb3-322" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb3-323"><a href="#cb3-323" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">15</span>) <span class="sc">+</span></span>
<span id="cb3-324"><a href="#cb3-324" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="fu">c</span>(<span class="fl">0.65</span>, <span class="fl">0.8</span>))</span>
<span id="cb3-325"><a href="#cb3-325" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-326"><a href="#cb3-326" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-327"><a href="#cb3-327" aria-hidden="true" tabindex="-1"></a>This was using the top loadings, so probably the fuzziness score of a sample</span>
<span id="cb3-328"><a href="#cb3-328" aria-hidden="true" tabindex="-1"></a>with relatively good data will be between 0 and 12 maximum, meaning that</span>
<span id="cb3-329"><a href="#cb3-329" aria-hidden="true" tabindex="-1"></a>they have less than 200 genes missing. When genes are missing, we hope that</span>
<span id="cb3-330"><a href="#cb3-330" aria-hidden="true" tabindex="-1"></a>they are in the bottom part of the loadings, so the fuzziness score</span>
<span id="cb3-331"><a href="#cb3-331" aria-hidden="true" tabindex="-1"></a>is as small as possible. </span>
<span id="cb3-332"><a href="#cb3-332" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-333"><a href="#cb3-333" aria-hidden="true" tabindex="-1"></a>Suppose now that the genes missing in a sample are random. We will calculate</span>
<span id="cb3-334"><a href="#cb3-334" aria-hidden="true" tabindex="-1"></a>random fuzziness scores based on this to have a feeling of the scores. </span>
<span id="cb3-335"><a href="#cb3-335" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-336"><a href="#cb3-336" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb3-337"><a href="#cb3-337" aria-hidden="true" tabindex="-1"></a><span class="in">nb_of_genes &lt;- seq(from = 2, to = 220, by = 20)</span></span>
<span id="cb3-338"><a href="#cb3-338" aria-hidden="true" tabindex="-1"></a><span class="in">list_of_random_genes &lt;- lapply(</span></span>
<span id="cb3-339"><a href="#cb3-339" aria-hidden="true" tabindex="-1"></a><span class="in">    nb_of_genes,</span></span>
<span id="cb3-340"><a href="#cb3-340" aria-hidden="true" tabindex="-1"></a><span class="in">    function(n, genes){</span></span>
<span id="cb3-341"><a href="#cb3-341" aria-hidden="true" tabindex="-1"></a><span class="in">        sapply(</span></span>
<span id="cb3-342"><a href="#cb3-342" aria-hidden="true" tabindex="-1"></a><span class="in">            1:1000,</span></span>
<span id="cb3-343"><a href="#cb3-343" aria-hidden="true" tabindex="-1"></a><span class="in">            function(x, genes, n){</span></span>
<span id="cb3-344"><a href="#cb3-344" aria-hidden="true" tabindex="-1"></a><span class="in">                sample(genes, size = n)</span></span>
<span id="cb3-345"><a href="#cb3-345" aria-hidden="true" tabindex="-1"></a><span class="in">            },</span></span>
<span id="cb3-346"><a href="#cb3-346" aria-hidden="true" tabindex="-1"></a><span class="in">            genes = genes,</span></span>
<span id="cb3-347"><a href="#cb3-347" aria-hidden="true" tabindex="-1"></a><span class="in">            n = n,</span></span>
<span id="cb3-348"><a href="#cb3-348" aria-hidden="true" tabindex="-1"></a><span class="in">            simplify = FALSE</span></span>
<span id="cb3-349"><a href="#cb3-349" aria-hidden="true" tabindex="-1"></a><span class="in">        )</span></span>
<span id="cb3-350"><a href="#cb3-350" aria-hidden="true" tabindex="-1"></a><span class="in">    },</span></span>
<span id="cb3-351"><a href="#cb3-351" aria-hidden="true" tabindex="-1"></a><span class="in">    genes = rownames(pca_fit$loadings)</span></span>
<span id="cb3-352"><a href="#cb3-352" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-353"><a href="#cb3-353" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-354"><a href="#cb3-354" aria-hidden="true" tabindex="-1"></a><span class="in">fuzziness_scores_random &lt;- lapply(</span></span>
<span id="cb3-355"><a href="#cb3-355" aria-hidden="true" tabindex="-1"></a><span class="in">    list_of_random_genes,</span></span>
<span id="cb3-356"><a href="#cb3-356" aria-hidden="true" tabindex="-1"></a><span class="in">    function(list_genes, pca_fit){</span></span>
<span id="cb3-357"><a href="#cb3-357" aria-hidden="true" tabindex="-1"></a><span class="in">        sapply(</span></span>
<span id="cb3-358"><a href="#cb3-358" aria-hidden="true" tabindex="-1"></a><span class="in">            list_genes,</span></span>
<span id="cb3-359"><a href="#cb3-359" aria-hidden="true" tabindex="-1"></a><span class="in">            get_fuzziness_score,</span></span>
<span id="cb3-360"><a href="#cb3-360" aria-hidden="true" tabindex="-1"></a><span class="in">            pca_fit = pca_fit</span></span>
<span id="cb3-361"><a href="#cb3-361" aria-hidden="true" tabindex="-1"></a><span class="in">        ) %&gt;% t %&gt;% data.frame %&gt;% </span></span>
<span id="cb3-362"><a href="#cb3-362" aria-hidden="true" tabindex="-1"></a><span class="in">            `colnames&lt;-`(which_loadings)</span></span>
<span id="cb3-363"><a href="#cb3-363" aria-hidden="true" tabindex="-1"></a><span class="in">    },</span></span>
<span id="cb3-364"><a href="#cb3-364" aria-hidden="true" tabindex="-1"></a><span class="in">    pca_fit = pca_fit</span></span>
<span id="cb3-365"><a href="#cb3-365" aria-hidden="true" tabindex="-1"></a><span class="in">) %&gt;% `names&lt;-`(paste0("#", nb_of_genes)) %&gt;%</span></span>
<span id="cb3-366"><a href="#cb3-366" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::bind_rows(., .id = "nb_genes") %&gt;%</span></span>
<span id="cb3-367"><a href="#cb3-367" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::mutate(</span></span>
<span id="cb3-368"><a href="#cb3-368" aria-hidden="true" tabindex="-1"></a><span class="in">        nb_genes = factor(</span></span>
<span id="cb3-369"><a href="#cb3-369" aria-hidden="true" tabindex="-1"></a><span class="in">            nb_genes,</span></span>
<span id="cb3-370"><a href="#cb3-370" aria-hidden="true" tabindex="-1"></a><span class="in">            levels = paste0("#", nb_of_genes)</span></span>
<span id="cb3-371"><a href="#cb3-371" aria-hidden="true" tabindex="-1"></a><span class="in">        )</span></span>
<span id="cb3-372"><a href="#cb3-372" aria-hidden="true" tabindex="-1"></a><span class="in">    ) %&gt;%</span></span>
<span id="cb3-373"><a href="#cb3-373" aria-hidden="true" tabindex="-1"></a><span class="in">    tidyr::pivot_longer(</span></span>
<span id="cb3-374"><a href="#cb3-374" aria-hidden="true" tabindex="-1"></a><span class="in">        which_loadings,</span></span>
<span id="cb3-375"><a href="#cb3-375" aria-hidden="true" tabindex="-1"></a><span class="in">        names_to = "PC",</span></span>
<span id="cb3-376"><a href="#cb3-376" aria-hidden="true" tabindex="-1"></a><span class="in">        values_to = "fuzziness_score"</span></span>
<span id="cb3-377"><a href="#cb3-377" aria-hidden="true" tabindex="-1"></a><span class="in">    )</span></span>
<span id="cb3-378"><a href="#cb3-378" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-379"><a href="#cb3-379" aria-hidden="true" tabindex="-1"></a><span class="in">saveRDS(</span></span>
<span id="cb3-380"><a href="#cb3-380" aria-hidden="true" tabindex="-1"></a><span class="in">    fuzziness_scores_random,</span></span>
<span id="cb3-381"><a href="#cb3-381" aria-hidden="true" tabindex="-1"></a><span class="in">    "../../results/rds_files/validation/fuzziness_scores_random.rds"</span></span>
<span id="cb3-382"><a href="#cb3-382" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-383"><a href="#cb3-383" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-384"><a href="#cb3-384" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-387"><a href="#cb3-387" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-388"><a href="#cb3-388" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-fuzziness-random</span></span>
<span id="cb3-389"><a href="#cb3-389" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Distribution of the fuzziness scores from random missing genes.</span></span>
<span id="cb3-390"><a href="#cb3-390" aria-hidden="true" tabindex="-1"></a><span class="co">#|     Each color corresponds to a different distribution using a different</span></span>
<span id="cb3-391"><a href="#cb3-391" aria-hidden="true" tabindex="-1"></a><span class="co">#|     number of genes.</span></span>
<span id="cb3-392"><a href="#cb3-392" aria-hidden="true" tabindex="-1"></a>fuzziness_scores_random <span class="sc">%&gt;%</span> </span>
<span id="cb3-393"><a href="#cb3-393" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(PC <span class="sc">==</span> <span class="st">"PC3"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb3-394"><a href="#cb3-394" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> fuzziness_score, <span class="at">fill =</span> nb_genes)) <span class="sc">+</span></span>
<span id="cb3-395"><a href="#cb3-395" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_histogram</span>(<span class="at">position =</span> <span class="st">"identity"</span>, <span class="at">bins =</span> <span class="dv">30</span>, <span class="at">alpha =</span> <span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb3-396"><a href="#cb3-396" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_fill_viridis_d</span>() <span class="sc">+</span></span>
<span id="cb3-397"><a href="#cb3-397" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb3-398"><a href="#cb3-398" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"Fuzziness score"</span>,</span>
<span id="cb3-399"><a href="#cb3-399" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"Count"</span>,</span>
<span id="cb3-400"><a href="#cb3-400" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">"Distribution of fuzziness score as a function of the number</span><span class="sc">\n</span><span class="st">of genes missing"</span>,</span>
<span id="cb3-401"><a href="#cb3-401" aria-hidden="true" tabindex="-1"></a>        <span class="at">subtitle =</span> <span class="st">"Only PC3 fuzziness scores"</span>,</span>
<span id="cb3-402"><a href="#cb3-402" aria-hidden="true" tabindex="-1"></a>        <span class="at">fill =</span> <span class="st">"# missing</span><span class="sc">\n</span><span class="st">genes"</span></span>
<span id="cb3-403"><a href="#cb3-403" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb3-404"><a href="#cb3-404" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>()</span>
<span id="cb3-405"><a href="#cb3-405" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-406"><a href="#cb3-406" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-407"><a href="#cb3-407" aria-hidden="true" tabindex="-1"></a>@fig-fuzziness-random shows the results as a function of the number of missing</span>
<span id="cb3-408"><a href="#cb3-408" aria-hidden="true" tabindex="-1"></a>genes. We see that in average, if 200 genes are missing, the fuzziness</span>
<span id="cb3-409"><a href="#cb3-409" aria-hidden="true" tabindex="-1"></a>score is around 5. If less than a 100 genes are missing, the fuzziness</span>
<span id="cb3-410"><a href="#cb3-410" aria-hidden="true" tabindex="-1"></a>score is around 3. In the next section we show how this influences the</span>
<span id="cb3-411"><a href="#cb3-411" aria-hidden="true" tabindex="-1"></a>position where the sample is projected into the embedding.</span>
<span id="cb3-412"><a href="#cb3-412" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-413"><a href="#cb3-413" aria-hidden="true" tabindex="-1"></a>Finally, we calculate the fuzziness score as a number of top loadings </span>
<span id="cb3-414"><a href="#cb3-414" aria-hidden="true" tabindex="-1"></a>missing. Suppose 150 genes are missing. We will change the proportion</span>
<span id="cb3-415"><a href="#cb3-415" aria-hidden="true" tabindex="-1"></a>of genes that are in the top loadings and in the missing list. Since</span>
<span id="cb3-416"><a href="#cb3-416" aria-hidden="true" tabindex="-1"></a>the cumulative sum for PC2 and PC3 are very similar, we focus only on</span>
<span id="cb3-417"><a href="#cb3-417" aria-hidden="true" tabindex="-1"></a>PC2.</span>
<span id="cb3-418"><a href="#cb3-418" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-421"><a href="#cb3-421" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-422"><a href="#cb3-422" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-fuzziness-random-top</span></span>
<span id="cb3-423"><a href="#cb3-423" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Distribution of the fuzziness scores from 150 random missing genes</span></span>
<span id="cb3-424"><a href="#cb3-424" aria-hidden="true" tabindex="-1"></a><span class="co">#|     with different proportions of top loading genes in the list.</span></span>
<span id="cb3-425"><a href="#cb3-425" aria-hidden="true" tabindex="-1"></a><span class="co">#|     Each color corresponds to a different distribution using a different</span></span>
<span id="cb3-426"><a href="#cb3-426" aria-hidden="true" tabindex="-1"></a><span class="co">#|     proportion of genes.</span></span>
<span id="cb3-427"><a href="#cb3-427" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-428"><a href="#cb3-428" aria-hidden="true" tabindex="-1"></a>nb_genes <span class="ot">&lt;-</span> <span class="dv">150</span></span>
<span id="cb3-429"><a href="#cb3-429" aria-hidden="true" tabindex="-1"></a>proportions <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="at">by =</span> <span class="fl">0.1</span>)</span>
<span id="cb3-430"><a href="#cb3-430" aria-hidden="true" tabindex="-1"></a>gene_names <span class="ot">&lt;-</span> <span class="fu">rownames</span>(pca_fit<span class="sc">$</span>loadings)</span>
<span id="cb3-431"><a href="#cb3-431" aria-hidden="true" tabindex="-1"></a>which_pcs <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">3</span>)</span>
<span id="cb3-432"><a href="#cb3-432" aria-hidden="true" tabindex="-1"></a>top_loadings_genes <span class="ot">&lt;-</span> gene_names[<span class="fu">order</span>(</span>
<span id="cb3-433"><a href="#cb3-433" aria-hidden="true" tabindex="-1"></a>    <span class="fu">abs</span>(pca_fit<span class="sc">$</span>loadings[, which_pcs]), </span>
<span id="cb3-434"><a href="#cb3-434" aria-hidden="true" tabindex="-1"></a>    <span class="at">decreasing =</span> <span class="cn">TRUE</span></span>
<span id="cb3-435"><a href="#cb3-435" aria-hidden="true" tabindex="-1"></a>)[<span class="dv">1</span><span class="sc">:</span>nb_genes]]</span>
<span id="cb3-436"><a href="#cb3-436" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-437"><a href="#cb3-437" aria-hidden="true" tabindex="-1"></a>proportions_top_fuzziness <span class="ot">&lt;-</span> <span class="fu">lapply</span>(</span>
<span id="cb3-438"><a href="#cb3-438" aria-hidden="true" tabindex="-1"></a>    proportions,</span>
<span id="cb3-439"><a href="#cb3-439" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(prop, gene_names, top_loadings_genes, pca_fit, nb_genes){</span>
<span id="cb3-440"><a href="#cb3-440" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb3-441"><a href="#cb3-441" aria-hidden="true" tabindex="-1"></a>        nb_genes_top <span class="ot">&lt;-</span> <span class="fu">floor</span>(prop <span class="sc">*</span> nb_genes)</span>
<span id="cb3-442"><a href="#cb3-442" aria-hidden="true" tabindex="-1"></a>        nb_genes_non_top <span class="ot">&lt;-</span> <span class="fu">floor</span>((<span class="dv">1</span> <span class="sc">-</span> prop) <span class="sc">*</span> nb_genes)</span>
<span id="cb3-443"><a href="#cb3-443" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb3-444"><a href="#cb3-444" aria-hidden="true" tabindex="-1"></a>        <span class="co"># run the fuzziness several times to get a distribution based on</span></span>
<span id="cb3-445"><a href="#cb3-445" aria-hidden="true" tabindex="-1"></a>        <span class="co"># the proportions</span></span>
<span id="cb3-446"><a href="#cb3-446" aria-hidden="true" tabindex="-1"></a>        list_genes <span class="ot">&lt;-</span> <span class="fu">sapply</span>(</span>
<span id="cb3-447"><a href="#cb3-447" aria-hidden="true" tabindex="-1"></a>            <span class="dv">1</span><span class="sc">:</span><span class="dv">1000</span>,</span>
<span id="cb3-448"><a href="#cb3-448" aria-hidden="true" tabindex="-1"></a>            <span class="cf">function</span>(</span>
<span id="cb3-449"><a href="#cb3-449" aria-hidden="true" tabindex="-1"></a>                i, </span>
<span id="cb3-450"><a href="#cb3-450" aria-hidden="true" tabindex="-1"></a>                genes_top, </span>
<span id="cb3-451"><a href="#cb3-451" aria-hidden="true" tabindex="-1"></a>                genes_non_top, </span>
<span id="cb3-452"><a href="#cb3-452" aria-hidden="true" tabindex="-1"></a>                nb_genes_top, </span>
<span id="cb3-453"><a href="#cb3-453" aria-hidden="true" tabindex="-1"></a>                nb_genes_non_top,</span>
<span id="cb3-454"><a href="#cb3-454" aria-hidden="true" tabindex="-1"></a>                pca_fit</span>
<span id="cb3-455"><a href="#cb3-455" aria-hidden="true" tabindex="-1"></a>            ){</span>
<span id="cb3-456"><a href="#cb3-456" aria-hidden="true" tabindex="-1"></a>                <span class="fu">get_fuzziness_score</span>(</span>
<span id="cb3-457"><a href="#cb3-457" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">c</span>(</span>
<span id="cb3-458"><a href="#cb3-458" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">sample</span>(genes_top, <span class="at">size =</span> nb_genes_top),</span>
<span id="cb3-459"><a href="#cb3-459" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">sample</span>(genes_non_top, <span class="at">size =</span> nb_genes_non_top)</span>
<span id="cb3-460"><a href="#cb3-460" aria-hidden="true" tabindex="-1"></a>                    ), </span>
<span id="cb3-461"><a href="#cb3-461" aria-hidden="true" tabindex="-1"></a>                    pca_fit,</span>
<span id="cb3-462"><a href="#cb3-462" aria-hidden="true" tabindex="-1"></a>                    <span class="at">which_pcs =</span> which_pcs</span>
<span id="cb3-463"><a href="#cb3-463" aria-hidden="true" tabindex="-1"></a>                )</span>
<span id="cb3-464"><a href="#cb3-464" aria-hidden="true" tabindex="-1"></a>            },</span>
<span id="cb3-465"><a href="#cb3-465" aria-hidden="true" tabindex="-1"></a>            <span class="at">genes_top =</span> top_loadings_genes,</span>
<span id="cb3-466"><a href="#cb3-466" aria-hidden="true" tabindex="-1"></a>            <span class="at">genes_non_top =</span> <span class="fu">setdiff</span>(gene_names, top_loadings_genes),</span>
<span id="cb3-467"><a href="#cb3-467" aria-hidden="true" tabindex="-1"></a>            <span class="at">nb_genes_top =</span> nb_genes_top,</span>
<span id="cb3-468"><a href="#cb3-468" aria-hidden="true" tabindex="-1"></a>            nb_genes_non_top,</span>
<span id="cb3-469"><a href="#cb3-469" aria-hidden="true" tabindex="-1"></a>            <span class="at">pca_fit =</span> pca_fit</span>
<span id="cb3-470"><a href="#cb3-470" aria-hidden="true" tabindex="-1"></a>        ) <span class="sc">%&gt;%</span> data.frame <span class="sc">%&gt;%</span> <span class="st">`</span><span class="at">colnames&lt;-</span><span class="st">`</span>(<span class="fu">paste0</span>(<span class="st">"PC"</span>, which_pcs))</span>
<span id="cb3-471"><a href="#cb3-471" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb3-472"><a href="#cb3-472" aria-hidden="true" tabindex="-1"></a>    <span class="at">gene_names =</span> gene_names,</span>
<span id="cb3-473"><a href="#cb3-473" aria-hidden="true" tabindex="-1"></a>    <span class="at">top_loadings_genes =</span> top_loadings_genes,</span>
<span id="cb3-474"><a href="#cb3-474" aria-hidden="true" tabindex="-1"></a>    <span class="at">pca_fit =</span> pca_fit,</span>
<span id="cb3-475"><a href="#cb3-475" aria-hidden="true" tabindex="-1"></a>    <span class="at">nb_genes =</span> nb_genes</span>
<span id="cb3-476"><a href="#cb3-476" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> <span class="st">`</span><span class="at">names&lt;-</span><span class="st">`</span>(<span class="fu">paste0</span>(proportions)) <span class="sc">%&gt;%</span></span>
<span id="cb3-477"><a href="#cb3-477" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(., <span class="at">.id =</span> <span class="st">"proportions"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb3-478"><a href="#cb3-478" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(</span>
<span id="cb3-479"><a href="#cb3-479" aria-hidden="true" tabindex="-1"></a>        <span class="at">cols =</span> <span class="fu">paste0</span>(<span class="st">"PC"</span>, which_pcs),</span>
<span id="cb3-480"><a href="#cb3-480" aria-hidden="true" tabindex="-1"></a>        <span class="at">names_to =</span> <span class="st">"PC"</span>,</span>
<span id="cb3-481"><a href="#cb3-481" aria-hidden="true" tabindex="-1"></a>        <span class="at">values_to =</span> <span class="st">"fuzziness_score"</span></span>
<span id="cb3-482"><a href="#cb3-482" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb3-483"><a href="#cb3-483" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-484"><a href="#cb3-484" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-485"><a href="#cb3-485" aria-hidden="true" tabindex="-1"></a>proportions_top_fuzziness <span class="sc">%&gt;%</span> </span>
<span id="cb3-486"><a href="#cb3-486" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(</span>
<span id="cb3-487"><a href="#cb3-487" aria-hidden="true" tabindex="-1"></a>        <span class="fu">aes</span>(<span class="at">x =</span> fuzziness_score, <span class="at">fill =</span> proportions)    </span>
<span id="cb3-488"><a href="#cb3-488" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb3-489"><a href="#cb3-489" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_histogram</span>(<span class="at">alpha =</span> <span class="fl">0.7</span>, <span class="at">bins =</span> <span class="dv">30</span>, <span class="at">position =</span> <span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb3-490"><a href="#cb3-490" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_fill_viridis_d</span>() <span class="sc">+</span></span>
<span id="cb3-491"><a href="#cb3-491" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb3-492"><a href="#cb3-492" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"Fuzziness score"</span>,</span>
<span id="cb3-493"><a href="#cb3-493" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"Count"</span>,</span>
<span id="cb3-494"><a href="#cb3-494" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="fu">paste0</span>(</span>
<span id="cb3-495"><a href="#cb3-495" aria-hidden="true" tabindex="-1"></a>            <span class="st">"Distribution of fuzziness score as a function of the</span><span class="sc">\n</span><span class="st">proportion "</span>,</span>
<span id="cb3-496"><a href="#cb3-496" aria-hidden="true" tabindex="-1"></a>            <span class="st">"of the top loading genes in the list of missing genes"</span></span>
<span id="cb3-497"><a href="#cb3-497" aria-hidden="true" tabindex="-1"></a>        ),</span>
<span id="cb3-498"><a href="#cb3-498" aria-hidden="true" tabindex="-1"></a>        <span class="at">subtitle =</span> <span class="st">"Only PC3 fuzziness scores"</span>,</span>
<span id="cb3-499"><a href="#cb3-499" aria-hidden="true" tabindex="-1"></a>        <span class="at">fill =</span> <span class="st">"Proportion of</span><span class="sc">\n</span><span class="st">top genes"</span></span>
<span id="cb3-500"><a href="#cb3-500" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb3-501"><a href="#cb3-501" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>()</span>
<span id="cb3-502"><a href="#cb3-502" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-503"><a href="#cb3-503" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-504"><a href="#cb3-504" aria-hidden="true" tabindex="-1"></a>@fig-fuzziness-random-top shows the fuzziness score with different proportions</span>
<span id="cb3-505"><a href="#cb3-505" aria-hidden="true" tabindex="-1"></a>of the top loading genes. We see that they are actually very important when</span>
<span id="cb3-506"><a href="#cb3-506" aria-hidden="true" tabindex="-1"></a>calculating the score. The top loading genes were defined as the top 150 genes</span>
<span id="cb3-507"><a href="#cb3-507" aria-hidden="true" tabindex="-1"></a>in the list. So when calculating the number of missing genes it is important</span>
<span id="cb3-508"><a href="#cb3-508" aria-hidden="true" tabindex="-1"></a>to check the number of top loadings in the list. </span>
<span id="cb3-509"><a href="#cb3-509" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-510"><a href="#cb3-510" aria-hidden="true" tabindex="-1"></a><span class="fu">### Validating the score</span></span>
<span id="cb3-511"><a href="#cb3-511" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-512"><a href="#cb3-512" aria-hidden="true" tabindex="-1"></a>In this section we calculate the embeddings with different number of genes</span>
<span id="cb3-513"><a href="#cb3-513" aria-hidden="true" tabindex="-1"></a>missing from a given sample and then plot the original embedding with</span>
<span id="cb3-514"><a href="#cb3-514" aria-hidden="true" tabindex="-1"></a>the newly one. For this</span>
<span id="cb3-515"><a href="#cb3-515" aria-hidden="true" tabindex="-1"></a>we use samples from TCGA. The genes are </span>
<span id="cb3-516"><a href="#cb3-516" aria-hidden="true" tabindex="-1"></a>removed even before the normalization procedure, as they are usually not</span>
<span id="cb3-517"><a href="#cb3-517" aria-hidden="true" tabindex="-1"></a>available there. We select randomly a number of top genes from PC3 and PC4</span>
<span id="cb3-518"><a href="#cb3-518" aria-hidden="true" tabindex="-1"></a>to compose the list of missing genes. The proportion of the top loading</span>
<span id="cb3-519"><a href="#cb3-519" aria-hidden="true" tabindex="-1"></a>genes is varied. </span>
<span id="cb3-520"><a href="#cb3-520" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-521"><a href="#cb3-521" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb3-522"><a href="#cb3-522" aria-hidden="true" tabindex="-1"></a><span class="in">nb_genes &lt;- 200</span></span>
<span id="cb3-523"><a href="#cb3-523" aria-hidden="true" tabindex="-1"></a><span class="in">proportions &lt;- seq(0, 1, by = 0.05)</span></span>
<span id="cb3-524"><a href="#cb3-524" aria-hidden="true" tabindex="-1"></a><span class="in">gene_names &lt;- rownames(pca_fit$loadings)</span></span>
<span id="cb3-525"><a href="#cb3-525" aria-hidden="true" tabindex="-1"></a><span class="in">which_pcs &lt;- 3:4</span></span>
<span id="cb3-526"><a href="#cb3-526" aria-hidden="true" tabindex="-1"></a><span class="in">top_loadings_genes_random &lt;- sapply(</span></span>
<span id="cb3-527"><a href="#cb3-527" aria-hidden="true" tabindex="-1"></a><span class="in">    which_pcs,</span></span>
<span id="cb3-528"><a href="#cb3-528" aria-hidden="true" tabindex="-1"></a><span class="in">    function(pc, gene_names, pca_fit, nb_genes){</span></span>
<span id="cb3-529"><a href="#cb3-529" aria-hidden="true" tabindex="-1"></a><span class="in">        gene_names[order(</span></span>
<span id="cb3-530"><a href="#cb3-530" aria-hidden="true" tabindex="-1"></a><span class="in">            abs(pca_fit$loadings[, paste0("PC", pc)]), </span></span>
<span id="cb3-531"><a href="#cb3-531" aria-hidden="true" tabindex="-1"></a><span class="in">            decreasing = TRUE</span></span>
<span id="cb3-532"><a href="#cb3-532" aria-hidden="true" tabindex="-1"></a><span class="in">        )[1:nb_genes]]   </span></span>
<span id="cb3-533"><a href="#cb3-533" aria-hidden="true" tabindex="-1"></a><span class="in">    },</span></span>
<span id="cb3-534"><a href="#cb3-534" aria-hidden="true" tabindex="-1"></a><span class="in">    gene_names = gene_names,</span></span>
<span id="cb3-535"><a href="#cb3-535" aria-hidden="true" tabindex="-1"></a><span class="in">    pca_fit = pca_fit,</span></span>
<span id="cb3-536"><a href="#cb3-536" aria-hidden="true" tabindex="-1"></a><span class="in">    nb_genes = nb_genes</span></span>
<span id="cb3-537"><a href="#cb3-537" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-538"><a href="#cb3-538" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-539"><a href="#cb3-539" aria-hidden="true" tabindex="-1"></a><span class="in">saveRDS(</span></span>
<span id="cb3-540"><a href="#cb3-540" aria-hidden="true" tabindex="-1"></a><span class="in">    top_loadings_genes_random,</span></span>
<span id="cb3-541"><a href="#cb3-541" aria-hidden="true" tabindex="-1"></a><span class="in">    "../../results/rds_files/validation/top_loadings_genes_random.rds"</span></span>
<span id="cb3-542"><a href="#cb3-542" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-543"><a href="#cb3-543" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-544"><a href="#cb3-544" aria-hidden="true" tabindex="-1"></a><span class="in"># we select a number of samples since it is not necessary to run</span></span>
<span id="cb3-545"><a href="#cb3-545" aria-hidden="true" tabindex="-1"></a><span class="in"># the analysis on all samples, but we want a representative sampling</span></span>
<span id="cb3-546"><a href="#cb3-546" aria-hidden="true" tabindex="-1"></a><span class="in"># so any sampling should suffice for the analysis</span></span>
<span id="cb3-547"><a href="#cb3-547" aria-hidden="true" tabindex="-1"></a><span class="in">set.seed(2039)</span></span>
<span id="cb3-548"><a href="#cb3-548" aria-hidden="true" tabindex="-1"></a><span class="in">samples_to_use &lt;- sample(colnames(datasets$tcga), size = 200)</span></span>
<span id="cb3-549"><a href="#cb3-549" aria-hidden="true" tabindex="-1"></a><span class="in">genes_proportions_top &lt;- lapply(</span></span>
<span id="cb3-550"><a href="#cb3-550" aria-hidden="true" tabindex="-1"></a><span class="in">    proportions,</span></span>
<span id="cb3-551"><a href="#cb3-551" aria-hidden="true" tabindex="-1"></a><span class="in">    function(prop, gene_names, top_loadings_genes, pca_fit, nb_genes){</span></span>
<span id="cb3-552"><a href="#cb3-552" aria-hidden="true" tabindex="-1"></a><span class="in">        </span></span>
<span id="cb3-553"><a href="#cb3-553" aria-hidden="true" tabindex="-1"></a><span class="in">        nb_genes_top &lt;- floor(prop * nb_genes)</span></span>
<span id="cb3-554"><a href="#cb3-554" aria-hidden="true" tabindex="-1"></a><span class="in">        nb_genes_non_top &lt;- floor((1 - prop) * nb_genes)</span></span>
<span id="cb3-555"><a href="#cb3-555" aria-hidden="true" tabindex="-1"></a><span class="in">        </span></span>
<span id="cb3-556"><a href="#cb3-556" aria-hidden="true" tabindex="-1"></a><span class="in">        # run the fuzziness several times to get a distribution based on</span></span>
<span id="cb3-557"><a href="#cb3-557" aria-hidden="true" tabindex="-1"></a><span class="in">        # the proportions</span></span>
<span id="cb3-558"><a href="#cb3-558" aria-hidden="true" tabindex="-1"></a><span class="in">        list_genes &lt;- lapply(</span></span>
<span id="cb3-559"><a href="#cb3-559" aria-hidden="true" tabindex="-1"></a><span class="in">            1:10,</span></span>
<span id="cb3-560"><a href="#cb3-560" aria-hidden="true" tabindex="-1"></a><span class="in">            function(</span></span>
<span id="cb3-561"><a href="#cb3-561" aria-hidden="true" tabindex="-1"></a><span class="in">                i, </span></span>
<span id="cb3-562"><a href="#cb3-562" aria-hidden="true" tabindex="-1"></a><span class="in">                genes_top, </span></span>
<span id="cb3-563"><a href="#cb3-563" aria-hidden="true" tabindex="-1"></a><span class="in">                genes_non_top, </span></span>
<span id="cb3-564"><a href="#cb3-564" aria-hidden="true" tabindex="-1"></a><span class="in">                nb_genes_top, </span></span>
<span id="cb3-565"><a href="#cb3-565" aria-hidden="true" tabindex="-1"></a><span class="in">                nb_genes_non_top</span></span>
<span id="cb3-566"><a href="#cb3-566" aria-hidden="true" tabindex="-1"></a><span class="in">            ){</span></span>
<span id="cb3-567"><a href="#cb3-567" aria-hidden="true" tabindex="-1"></a><span class="in">                c(</span></span>
<span id="cb3-568"><a href="#cb3-568" aria-hidden="true" tabindex="-1"></a><span class="in">                    sample(genes_top, size = nb_genes_top),</span></span>
<span id="cb3-569"><a href="#cb3-569" aria-hidden="true" tabindex="-1"></a><span class="in">                    sample(genes_non_top, size = nb_genes_non_top)</span></span>
<span id="cb3-570"><a href="#cb3-570" aria-hidden="true" tabindex="-1"></a><span class="in">                )</span></span>
<span id="cb3-571"><a href="#cb3-571" aria-hidden="true" tabindex="-1"></a><span class="in">            },</span></span>
<span id="cb3-572"><a href="#cb3-572" aria-hidden="true" tabindex="-1"></a><span class="in">            genes_top = top_loadings_genes,</span></span>
<span id="cb3-573"><a href="#cb3-573" aria-hidden="true" tabindex="-1"></a><span class="in">            genes_non_top = setdiff(gene_names, top_loadings_genes),</span></span>
<span id="cb3-574"><a href="#cb3-574" aria-hidden="true" tabindex="-1"></a><span class="in">            nb_genes_top = nb_genes_top,</span></span>
<span id="cb3-575"><a href="#cb3-575" aria-hidden="true" tabindex="-1"></a><span class="in">            nb_genes_non_top</span></span>
<span id="cb3-576"><a href="#cb3-576" aria-hidden="true" tabindex="-1"></a><span class="in">        ) </span></span>
<span id="cb3-577"><a href="#cb3-577" aria-hidden="true" tabindex="-1"></a><span class="in">    },</span></span>
<span id="cb3-578"><a href="#cb3-578" aria-hidden="true" tabindex="-1"></a><span class="in">    gene_names = gene_names,</span></span>
<span id="cb3-579"><a href="#cb3-579" aria-hidden="true" tabindex="-1"></a><span class="in">    top_loadings_genes = top_loadings_genes_random,</span></span>
<span id="cb3-580"><a href="#cb3-580" aria-hidden="true" tabindex="-1"></a><span class="in">    nb_genes = nb_genes</span></span>
<span id="cb3-581"><a href="#cb3-581" aria-hidden="true" tabindex="-1"></a><span class="in">) %&gt;% `names&lt;-`(paste0(proportions))</span></span>
<span id="cb3-582"><a href="#cb3-582" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-583"><a href="#cb3-583" aria-hidden="true" tabindex="-1"></a><span class="in">saveRDS(</span></span>
<span id="cb3-584"><a href="#cb3-584" aria-hidden="true" tabindex="-1"></a><span class="in">    samples_to_use,</span></span>
<span id="cb3-585"><a href="#cb3-585" aria-hidden="true" tabindex="-1"></a><span class="in">    "../../results/rds_files/validation/samples_to_use.rds"</span></span>
<span id="cb3-586"><a href="#cb3-586" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-587"><a href="#cb3-587" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-588"><a href="#cb3-588" aria-hidden="true" tabindex="-1"></a><span class="in">saveRDS(</span></span>
<span id="cb3-589"><a href="#cb3-589" aria-hidden="true" tabindex="-1"></a><span class="in">    genes_proportions_top,</span></span>
<span id="cb3-590"><a href="#cb3-590" aria-hidden="true" tabindex="-1"></a><span class="in">    "../../results/rds_files/validation/genes_proportions_top.rds"</span></span>
<span id="cb3-591"><a href="#cb3-591" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-592"><a href="#cb3-592" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-593"><a href="#cb3-593" aria-hidden="true" tabindex="-1"></a><span class="in">fuzziness_scores_random_top &lt;- rapply(</span></span>
<span id="cb3-594"><a href="#cb3-594" aria-hidden="true" tabindex="-1"></a><span class="in">    genes_proportions_top,</span></span>
<span id="cb3-595"><a href="#cb3-595" aria-hidden="true" tabindex="-1"></a><span class="in">    get_fuzziness_score,</span></span>
<span id="cb3-596"><a href="#cb3-596" aria-hidden="true" tabindex="-1"></a><span class="in">    pca_fit = pca_fit,</span></span>
<span id="cb3-597"><a href="#cb3-597" aria-hidden="true" tabindex="-1"></a><span class="in">    how = "list"</span></span>
<span id="cb3-598"><a href="#cb3-598" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-599"><a href="#cb3-599" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-600"><a href="#cb3-600" aria-hidden="true" tabindex="-1"></a><span class="in">saveRDS(</span></span>
<span id="cb3-601"><a href="#cb3-601" aria-hidden="true" tabindex="-1"></a><span class="in">    fuzziness_scores_random_top,</span></span>
<span id="cb3-602"><a href="#cb3-602" aria-hidden="true" tabindex="-1"></a><span class="in">    "../../results/rds_files/validation/fuzziness_scores_random_top.rds"</span></span>
<span id="cb3-603"><a href="#cb3-603" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-604"><a href="#cb3-604" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-605"><a href="#cb3-605" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-606"><a href="#cb3-606" aria-hidden="true" tabindex="-1"></a><span class="in">pca_coords_random &lt;- parallel::mclapply(</span></span>
<span id="cb3-607"><a href="#cb3-607" aria-hidden="true" tabindex="-1"></a><span class="in">    genes_proportions_top,</span></span>
<span id="cb3-608"><a href="#cb3-608" aria-hidden="true" tabindex="-1"></a><span class="in">    function(</span></span>
<span id="cb3-609"><a href="#cb3-609" aria-hidden="true" tabindex="-1"></a><span class="in">        lists_genes, </span></span>
<span id="cb3-610"><a href="#cb3-610" aria-hidden="true" tabindex="-1"></a><span class="in">        sum_exp, </span></span>
<span id="cb3-611"><a href="#cb3-611" aria-hidden="true" tabindex="-1"></a><span class="in">        pca_fit, </span></span>
<span id="cb3-612"><a href="#cb3-612" aria-hidden="true" tabindex="-1"></a><span class="in">        assay_to_use, </span></span>
<span id="cb3-613"><a href="#cb3-613" aria-hidden="true" tabindex="-1"></a><span class="in">        stable_genes, </span></span>
<span id="cb3-614"><a href="#cb3-614" aria-hidden="true" tabindex="-1"></a><span class="in">        most_variable_genes</span></span>
<span id="cb3-615"><a href="#cb3-615" aria-hidden="true" tabindex="-1"></a><span class="in">    ){</span></span>
<span id="cb3-616"><a href="#cb3-616" aria-hidden="true" tabindex="-1"></a><span class="in">        </span></span>
<span id="cb3-617"><a href="#cb3-617" aria-hidden="true" tabindex="-1"></a><span class="in">        # for each list of gene, remove the provided genes and proceed</span></span>
<span id="cb3-618"><a href="#cb3-618" aria-hidden="true" tabindex="-1"></a><span class="in">        # with the whole pipeline for all samples selected</span></span>
<span id="cb3-619"><a href="#cb3-619" aria-hidden="true" tabindex="-1"></a><span class="in">        lapply(</span></span>
<span id="cb3-620"><a href="#cb3-620" aria-hidden="true" tabindex="-1"></a><span class="in">            lists_genes,</span></span>
<span id="cb3-621"><a href="#cb3-621" aria-hidden="true" tabindex="-1"></a><span class="in">            function(</span></span>
<span id="cb3-622"><a href="#cb3-622" aria-hidden="true" tabindex="-1"></a><span class="in">                genes_to_remove, </span></span>
<span id="cb3-623"><a href="#cb3-623" aria-hidden="true" tabindex="-1"></a><span class="in">                sum_exp, </span></span>
<span id="cb3-624"><a href="#cb3-624" aria-hidden="true" tabindex="-1"></a><span class="in">                pca_fit,</span></span>
<span id="cb3-625"><a href="#cb3-625" aria-hidden="true" tabindex="-1"></a><span class="in">                assay_to_use, </span></span>
<span id="cb3-626"><a href="#cb3-626" aria-hidden="true" tabindex="-1"></a><span class="in">                stable_genes, </span></span>
<span id="cb3-627"><a href="#cb3-627" aria-hidden="true" tabindex="-1"></a><span class="in">                most_variable_genes</span></span>
<span id="cb3-628"><a href="#cb3-628" aria-hidden="true" tabindex="-1"></a><span class="in">            ){</span></span>
<span id="cb3-629"><a href="#cb3-629" aria-hidden="true" tabindex="-1"></a><span class="in">                </span></span>
<span id="cb3-630"><a href="#cb3-630" aria-hidden="true" tabindex="-1"></a><span class="in">                # remove genes</span></span>
<span id="cb3-631"><a href="#cb3-631" aria-hidden="true" tabindex="-1"></a><span class="in">                sum_exp &lt;- sum_exp[setdiff(rownames(sum_exp), genes_to_remove),]</span></span>
<span id="cb3-632"><a href="#cb3-632" aria-hidden="true" tabindex="-1"></a><span class="in">                </span></span>
<span id="cb3-633"><a href="#cb3-633" aria-hidden="true" tabindex="-1"></a><span class="in">                # now proceed with the normalization pipeline and PC retrieval</span></span>
<span id="cb3-634"><a href="#cb3-634" aria-hidden="true" tabindex="-1"></a><span class="in">                sum_exp &lt;- get_final_ranking_values(</span></span>
<span id="cb3-635"><a href="#cb3-635" aria-hidden="true" tabindex="-1"></a><span class="in">                    sum_exp,</span></span>
<span id="cb3-636"><a href="#cb3-636" aria-hidden="true" tabindex="-1"></a><span class="in">                    assay_to_use,</span></span>
<span id="cb3-637"><a href="#cb3-637" aria-hidden="true" tabindex="-1"></a><span class="in">                    stable_genes,</span></span>
<span id="cb3-638"><a href="#cb3-638" aria-hidden="true" tabindex="-1"></a><span class="in">                    most_variable_genes</span></span>
<span id="cb3-639"><a href="#cb3-639" aria-hidden="true" tabindex="-1"></a><span class="in">                )</span></span>
<span id="cb3-640"><a href="#cb3-640" aria-hidden="true" tabindex="-1"></a><span class="in">                pca_coords &lt;- get_pca_coordinates(sum_exp, pca_fit)</span></span>
<span id="cb3-641"><a href="#cb3-641" aria-hidden="true" tabindex="-1"></a><span class="in">             </span></span>
<span id="cb3-642"><a href="#cb3-642" aria-hidden="true" tabindex="-1"></a><span class="in">                pca_coords[, paste0("PC", 1:10)]</span></span>
<span id="cb3-643"><a href="#cb3-643" aria-hidden="true" tabindex="-1"></a><span class="in">                </span></span>
<span id="cb3-644"><a href="#cb3-644" aria-hidden="true" tabindex="-1"></a><span class="in">            },</span></span>
<span id="cb3-645"><a href="#cb3-645" aria-hidden="true" tabindex="-1"></a><span class="in">            sum_exp = sum_exp, </span></span>
<span id="cb3-646"><a href="#cb3-646" aria-hidden="true" tabindex="-1"></a><span class="in">            pca_fit = pca_fit, </span></span>
<span id="cb3-647"><a href="#cb3-647" aria-hidden="true" tabindex="-1"></a><span class="in">            assay_to_use = assay_to_use, </span></span>
<span id="cb3-648"><a href="#cb3-648" aria-hidden="true" tabindex="-1"></a><span class="in">            stable_genes = stable_genes,</span></span>
<span id="cb3-649"><a href="#cb3-649" aria-hidden="true" tabindex="-1"></a><span class="in">            most_variable_genes = most_variable_genes</span></span>
<span id="cb3-650"><a href="#cb3-650" aria-hidden="true" tabindex="-1"></a><span class="in">        )</span></span>
<span id="cb3-651"><a href="#cb3-651" aria-hidden="true" tabindex="-1"></a><span class="in">    },</span></span>
<span id="cb3-652"><a href="#cb3-652" aria-hidden="true" tabindex="-1"></a><span class="in">    sum_exp = datasets$tcga[, samples_to_use],</span></span>
<span id="cb3-653"><a href="#cb3-653" aria-hidden="true" tabindex="-1"></a><span class="in">    pca_fit = pca_fit, </span></span>
<span id="cb3-654"><a href="#cb3-654" aria-hidden="true" tabindex="-1"></a><span class="in">    assay_to_use = "logFPKM_TMM", </span></span>
<span id="cb3-655"><a href="#cb3-655" aria-hidden="true" tabindex="-1"></a><span class="in">    stable_genes = stable_genes, </span></span>
<span id="cb3-656"><a href="#cb3-656" aria-hidden="true" tabindex="-1"></a><span class="in">    most_variable_genes = setdiff(genes_for_pca, stable_genes),</span></span>
<span id="cb3-657"><a href="#cb3-657" aria-hidden="true" tabindex="-1"></a><span class="in">    mc.cores = 11</span></span>
<span id="cb3-658"><a href="#cb3-658" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-659"><a href="#cb3-659" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-660"><a href="#cb3-660" aria-hidden="true" tabindex="-1"></a><span class="in">saveRDS(</span></span>
<span id="cb3-661"><a href="#cb3-661" aria-hidden="true" tabindex="-1"></a><span class="in">    pca_coords_random,</span></span>
<span id="cb3-662"><a href="#cb3-662" aria-hidden="true" tabindex="-1"></a><span class="in">    "../../results/rds_files/validation/pca_coords_random.rds"</span></span>
<span id="cb3-663"><a href="#cb3-663" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-664"><a href="#cb3-664" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-665"><a href="#cb3-665" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-666"><a href="#cb3-666" aria-hidden="true" tabindex="-1"></a><span class="in">get_pca_random_genes &lt;- function(</span></span>
<span id="cb3-667"><a href="#cb3-667" aria-hidden="true" tabindex="-1"></a><span class="in">    use_sample,</span></span>
<span id="cb3-668"><a href="#cb3-668" aria-hidden="true" tabindex="-1"></a><span class="in">    og_pca,</span></span>
<span id="cb3-669"><a href="#cb3-669" aria-hidden="true" tabindex="-1"></a><span class="in">    pca_coords_random,</span></span>
<span id="cb3-670"><a href="#cb3-670" aria-hidden="true" tabindex="-1"></a><span class="in">    fuzziness_scores_random,</span></span>
<span id="cb3-671"><a href="#cb3-671" aria-hidden="true" tabindex="-1"></a><span class="in">    which_pcs = paste0("PC", 1:10)</span></span>
<span id="cb3-672"><a href="#cb3-672" aria-hidden="true" tabindex="-1"></a><span class="in">){</span></span>
<span id="cb3-673"><a href="#cb3-673" aria-hidden="true" tabindex="-1"></a><span class="in">    </span></span>
<span id="cb3-674"><a href="#cb3-674" aria-hidden="true" tabindex="-1"></a><span class="in">    # for each proportion, extract the coordinates</span></span>
<span id="cb3-675"><a href="#cb3-675" aria-hidden="true" tabindex="-1"></a><span class="in">    pca_coords &lt;- lapply(</span></span>
<span id="cb3-676"><a href="#cb3-676" aria-hidden="true" tabindex="-1"></a><span class="in">        pca_coords_random,</span></span>
<span id="cb3-677"><a href="#cb3-677" aria-hidden="true" tabindex="-1"></a><span class="in">        function(x){</span></span>
<span id="cb3-678"><a href="#cb3-678" aria-hidden="true" tabindex="-1"></a><span class="in">            sapply(</span></span>
<span id="cb3-679"><a href="#cb3-679" aria-hidden="true" tabindex="-1"></a><span class="in">                x,</span></span>
<span id="cb3-680"><a href="#cb3-680" aria-hidden="true" tabindex="-1"></a><span class="in">                function(y) y[use_sample, which_pcs]</span></span>
<span id="cb3-681"><a href="#cb3-681" aria-hidden="true" tabindex="-1"></a><span class="in">            ) %&gt;% t %&gt;% data.frame</span></span>
<span id="cb3-682"><a href="#cb3-682" aria-hidden="true" tabindex="-1"></a><span class="in">        }</span></span>
<span id="cb3-683"><a href="#cb3-683" aria-hidden="true" tabindex="-1"></a><span class="in">    ) %&gt;% dplyr::bind_rows(., .id = "proportion") %&gt;%</span></span>
<span id="cb3-684"><a href="#cb3-684" aria-hidden="true" tabindex="-1"></a><span class="in">        data.frame</span></span>
<span id="cb3-685"><a href="#cb3-685" aria-hidden="true" tabindex="-1"></a><span class="in">    </span></span>
<span id="cb3-686"><a href="#cb3-686" aria-hidden="true" tabindex="-1"></a><span class="in">    fuzziness_scores &lt;- lapply(</span></span>
<span id="cb3-687"><a href="#cb3-687" aria-hidden="true" tabindex="-1"></a><span class="in">        fuzziness_scores_random,</span></span>
<span id="cb3-688"><a href="#cb3-688" aria-hidden="true" tabindex="-1"></a><span class="in">        function(x){</span></span>
<span id="cb3-689"><a href="#cb3-689" aria-hidden="true" tabindex="-1"></a><span class="in">            sapply(</span></span>
<span id="cb3-690"><a href="#cb3-690" aria-hidden="true" tabindex="-1"></a><span class="in">                x,</span></span>
<span id="cb3-691"><a href="#cb3-691" aria-hidden="true" tabindex="-1"></a><span class="in">                function(y) y</span></span>
<span id="cb3-692"><a href="#cb3-692" aria-hidden="true" tabindex="-1"></a><span class="in">            ) %&gt;% t %&gt;% data.frame</span></span>
<span id="cb3-693"><a href="#cb3-693" aria-hidden="true" tabindex="-1"></a><span class="in">        }</span></span>
<span id="cb3-694"><a href="#cb3-694" aria-hidden="true" tabindex="-1"></a><span class="in">    ) %&gt;% dplyr::bind_rows(., .id = "proportion") %&gt;%</span></span>
<span id="cb3-695"><a href="#cb3-695" aria-hidden="true" tabindex="-1"></a><span class="in">        data.frame</span></span>
<span id="cb3-696"><a href="#cb3-696" aria-hidden="true" tabindex="-1"></a><span class="in">    </span></span>
<span id="cb3-697"><a href="#cb3-697" aria-hidden="true" tabindex="-1"></a><span class="in">    </span></span>
<span id="cb3-698"><a href="#cb3-698" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::bind_rows(</span></span>
<span id="cb3-699"><a href="#cb3-699" aria-hidden="true" tabindex="-1"></a><span class="in">            list(</span></span>
<span id="cb3-700"><a href="#cb3-700" aria-hidden="true" tabindex="-1"></a><span class="in">                random = pca_coords,</span></span>
<span id="cb3-701"><a href="#cb3-701" aria-hidden="true" tabindex="-1"></a><span class="in">                original = og_pca %&gt;% </span></span>
<span id="cb3-702"><a href="#cb3-702" aria-hidden="true" tabindex="-1"></a><span class="in">                    dplyr::filter(sample_name == use_sample) %&gt;%</span></span>
<span id="cb3-703"><a href="#cb3-703" aria-hidden="true" tabindex="-1"></a><span class="in">                    dplyr::select(all_of(which_pcs)) %&gt;%</span></span>
<span id="cb3-704"><a href="#cb3-704" aria-hidden="true" tabindex="-1"></a><span class="in">                    dplyr::mutate(proportion = "all_genes") %&gt;%</span></span>
<span id="cb3-705"><a href="#cb3-705" aria-hidden="true" tabindex="-1"></a><span class="in">                    data.frame %&gt;%</span></span>
<span id="cb3-706"><a href="#cb3-706" aria-hidden="true" tabindex="-1"></a><span class="in">                    .[, colnames(pca_coords)]</span></span>
<span id="cb3-707"><a href="#cb3-707" aria-hidden="true" tabindex="-1"></a><span class="in">            ),</span></span>
<span id="cb3-708"><a href="#cb3-708" aria-hidden="true" tabindex="-1"></a><span class="in">            .id = "embedding"</span></span>
<span id="cb3-709"><a href="#cb3-709" aria-hidden="true" tabindex="-1"></a><span class="in">        )</span></span>
<span id="cb3-710"><a href="#cb3-710" aria-hidden="true" tabindex="-1"></a><span class="in">}</span></span>
<span id="cb3-711"><a href="#cb3-711" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-712"><a href="#cb3-712" aria-hidden="true" tabindex="-1"></a><span class="in">pca_random_genes_patients &lt;- sapply(</span></span>
<span id="cb3-713"><a href="#cb3-713" aria-hidden="true" tabindex="-1"></a><span class="in">    samples_to_use, </span></span>
<span id="cb3-714"><a href="#cb3-714" aria-hidden="true" tabindex="-1"></a><span class="in">    get_pca_random_genes, </span></span>
<span id="cb3-715"><a href="#cb3-715" aria-hidden="true" tabindex="-1"></a><span class="in">    og_pca = df_pca_coordinates,</span></span>
<span id="cb3-716"><a href="#cb3-716" aria-hidden="true" tabindex="-1"></a><span class="in">    pca_coords_random = pca_coords_random, </span></span>
<span id="cb3-717"><a href="#cb3-717" aria-hidden="true" tabindex="-1"></a><span class="in">    fuzziness_scores_random = fuzziness_scores_random_top,</span></span>
<span id="cb3-718"><a href="#cb3-718" aria-hidden="true" tabindex="-1"></a><span class="in">    which_pcs = paste0("PC", 3:4),</span></span>
<span id="cb3-719"><a href="#cb3-719" aria-hidden="true" tabindex="-1"></a><span class="in">    USE.NAMES = TRUE, </span></span>
<span id="cb3-720"><a href="#cb3-720" aria-hidden="true" tabindex="-1"></a><span class="in">    simplify = FALSE</span></span>
<span id="cb3-721"><a href="#cb3-721" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-722"><a href="#cb3-722" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-723"><a href="#cb3-723" aria-hidden="true" tabindex="-1"></a><span class="in">saveRDS(</span></span>
<span id="cb3-724"><a href="#cb3-724" aria-hidden="true" tabindex="-1"></a><span class="in">    pca_random_genes_patients,</span></span>
<span id="cb3-725"><a href="#cb3-725" aria-hidden="true" tabindex="-1"></a><span class="in">    "../../results/rds_files/validation/pca_random_genes_patients.rds"</span></span>
<span id="cb3-726"><a href="#cb3-726" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-727"><a href="#cb3-727" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-728"><a href="#cb3-728" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-731"><a href="#cb3-731" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-732"><a href="#cb3-732" aria-hidden="true" tabindex="-1"></a>plots_random_genes <span class="ot">&lt;-</span> <span class="fu">sapply</span>(</span>
<span id="cb3-733"><a href="#cb3-733" aria-hidden="true" tabindex="-1"></a>    samples_to_use, </span>
<span id="cb3-734"><a href="#cb3-734" aria-hidden="true" tabindex="-1"></a>    plot_pca_random_genes,</span>
<span id="cb3-735"><a href="#cb3-735" aria-hidden="true" tabindex="-1"></a>    <span class="at">pca_random_genes_patients =</span> pca_random_genes_patients,</span>
<span id="cb3-736"><a href="#cb3-736" aria-hidden="true" tabindex="-1"></a>    <span class="at">df_pca_coordinates =</span> df_pca_coordinates,</span>
<span id="cb3-737"><a href="#cb3-737" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>, </span>
<span id="cb3-738"><a href="#cb3-738" aria-hidden="true" tabindex="-1"></a>    <span class="at">simplify =</span> <span class="cn">FALSE</span></span>
<span id="cb3-739"><a href="#cb3-739" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb3-740"><a href="#cb3-740" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-741"><a href="#cb3-741" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-742"><a href="#cb3-742" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-743"><a href="#cb3-743" aria-hidden="true" tabindex="-1"></a>::: {.panel-tabset}</span>
<span id="cb3-744"><a href="#cb3-744" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-747"><a href="#cb3-747" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-748"><a href="#cb3-748" aria-hidden="true" tabindex="-1"></a>patient_name <span class="ot">&lt;-</span> samples_to_use[<span class="dv">10</span>]</span>
<span id="cb3-749"><a href="#cb3-749" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-750"><a href="#cb3-750" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-751"><a href="#cb3-751" aria-hidden="true" tabindex="-1"></a><span class="fu">#### `r patient_name`</span></span>
<span id="cb3-752"><a href="#cb3-752" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-753"><a href="#cb3-753" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=8, fig.height=5}</span></span>
<span id="cb3-754"><a href="#cb3-754" aria-hidden="true" tabindex="-1"></a><span class="in">#| label: fig-missing-correction-10</span></span>
<span id="cb3-755"><a href="#cb3-755" aria-hidden="true" tabindex="-1"></a><span class="in">plots_random_genes[[patient_name]]</span></span>
<span id="cb3-756"><a href="#cb3-756" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-757"><a href="#cb3-757" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-760"><a href="#cb3-760" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-761"><a href="#cb3-761" aria-hidden="true" tabindex="-1"></a>patient_name <span class="ot">&lt;-</span> samples_to_use[<span class="dv">20</span>]</span>
<span id="cb3-762"><a href="#cb3-762" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-763"><a href="#cb3-763" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-764"><a href="#cb3-764" aria-hidden="true" tabindex="-1"></a><span class="fu">#### `r patient_name`</span></span>
<span id="cb3-765"><a href="#cb3-765" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-766"><a href="#cb3-766" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=8, fig.height=5}</span></span>
<span id="cb3-767"><a href="#cb3-767" aria-hidden="true" tabindex="-1"></a><span class="in">#| label: fig-missing-correction-20</span></span>
<span id="cb3-768"><a href="#cb3-768" aria-hidden="true" tabindex="-1"></a><span class="in">plots_random_genes[[patient_name]]</span></span>
<span id="cb3-769"><a href="#cb3-769" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-770"><a href="#cb3-770" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-773"><a href="#cb3-773" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-774"><a href="#cb3-774" aria-hidden="true" tabindex="-1"></a>patient_name <span class="ot">&lt;-</span> samples_to_use[<span class="dv">50</span>]</span>
<span id="cb3-775"><a href="#cb3-775" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-776"><a href="#cb3-776" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-777"><a href="#cb3-777" aria-hidden="true" tabindex="-1"></a><span class="fu">#### `r patient_name`</span></span>
<span id="cb3-778"><a href="#cb3-778" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-779"><a href="#cb3-779" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=8, fig.height=5}</span></span>
<span id="cb3-780"><a href="#cb3-780" aria-hidden="true" tabindex="-1"></a><span class="in">#| label: fig-missing-correction-50</span></span>
<span id="cb3-781"><a href="#cb3-781" aria-hidden="true" tabindex="-1"></a><span class="in">plots_random_genes[[patient_name]]</span></span>
<span id="cb3-782"><a href="#cb3-782" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-783"><a href="#cb3-783" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-786"><a href="#cb3-786" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-787"><a href="#cb3-787" aria-hidden="true" tabindex="-1"></a>patient_name <span class="ot">&lt;-</span> samples_to_use[<span class="dv">80</span>]</span>
<span id="cb3-788"><a href="#cb3-788" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-789"><a href="#cb3-789" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-790"><a href="#cb3-790" aria-hidden="true" tabindex="-1"></a><span class="fu">#### `r patient_name`</span></span>
<span id="cb3-791"><a href="#cb3-791" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-792"><a href="#cb3-792" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=8, fig.height=5}</span></span>
<span id="cb3-793"><a href="#cb3-793" aria-hidden="true" tabindex="-1"></a><span class="in">#| label: fig-missing-correction-80</span></span>
<span id="cb3-794"><a href="#cb3-794" aria-hidden="true" tabindex="-1"></a><span class="in">plots_random_genes[[patient_name]]</span></span>
<span id="cb3-795"><a href="#cb3-795" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-796"><a href="#cb3-796" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-799"><a href="#cb3-799" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-800"><a href="#cb3-800" aria-hidden="true" tabindex="-1"></a>patient_name <span class="ot">&lt;-</span> samples_to_use[<span class="dv">182</span>]</span>
<span id="cb3-801"><a href="#cb3-801" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-802"><a href="#cb3-802" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-803"><a href="#cb3-803" aria-hidden="true" tabindex="-1"></a><span class="fu">#### `r patient_name`</span></span>
<span id="cb3-804"><a href="#cb3-804" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-805"><a href="#cb3-805" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=8, fig.height=5}</span></span>
<span id="cb3-806"><a href="#cb3-806" aria-hidden="true" tabindex="-1"></a><span class="in">#| label: fig-missing-correction-182</span></span>
<span id="cb3-807"><a href="#cb3-807" aria-hidden="true" tabindex="-1"></a><span class="in">plots_random_genes[[patient_name]]</span></span>
<span id="cb3-808"><a href="#cb3-808" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-809"><a href="#cb3-809" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-812"><a href="#cb3-812" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-813"><a href="#cb3-813" aria-hidden="true" tabindex="-1"></a>patient_name <span class="ot">&lt;-</span> samples_to_use[<span class="dv">198</span>]</span>
<span id="cb3-814"><a href="#cb3-814" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-815"><a href="#cb3-815" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-816"><a href="#cb3-816" aria-hidden="true" tabindex="-1"></a><span class="fu">#### `r patient_name`</span></span>
<span id="cb3-817"><a href="#cb3-817" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-818"><a href="#cb3-818" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=8, fig.height=5}</span></span>
<span id="cb3-819"><a href="#cb3-819" aria-hidden="true" tabindex="-1"></a><span class="in">#| label: fig-missing-correction-198</span></span>
<span id="cb3-820"><a href="#cb3-820" aria-hidden="true" tabindex="-1"></a><span class="in">plots_random_genes[[patient_name]]</span></span>
<span id="cb3-821"><a href="#cb3-821" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-822"><a href="#cb3-822" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-823"><a href="#cb3-823" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb3-824"><a href="#cb3-824" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-825"><a href="#cb3-825" aria-hidden="true" tabindex="-1"></a>When looking at the figures above, it seems that whenever there are </span>
<span id="cb3-826"><a href="#cb3-826" aria-hidden="true" tabindex="-1"></a>genes missing,</span>
<span id="cb3-827"><a href="#cb3-827" aria-hidden="true" tabindex="-1"></a>and the more of top loadings they are, the closer they are to the origin.</span>
<span id="cb3-828"><a href="#cb3-828" aria-hidden="true" tabindex="-1"></a>This makes sense, since PCA is doing a simple linear combination of the loadings.</span>
<span id="cb3-829"><a href="#cb3-829" aria-hidden="true" tabindex="-1"></a>This should be fine to correct, since we can fit a line going through the </span>
<span id="cb3-830"><a href="#cb3-830" aria-hidden="true" tabindex="-1"></a>origin and goes through the points above. Then the adjustment will depend</span>
<span id="cb3-831"><a href="#cb3-831" aria-hidden="true" tabindex="-1"></a>on the number of genes that are missing and are top loadings. The more </span>
<span id="cb3-832"><a href="#cb3-832" aria-hidden="true" tabindex="-1"></a>they are, the more adjustment we will need. </span>
<span id="cb3-833"><a href="#cb3-833" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-834"><a href="#cb3-834" aria-hidden="true" tabindex="-1"></a>For the adjustment we only know two things, its current position and </span>
<span id="cb3-835"><a href="#cb3-835" aria-hidden="true" tabindex="-1"></a>how many of the top loadings are missing. Based on this information we</span>
<span id="cb3-836"><a href="#cb3-836" aria-hidden="true" tabindex="-1"></a>should be able to correct the samples. Given the list of genes that are missing,</span>
<span id="cb3-837"><a href="#cb3-837" aria-hidden="true" tabindex="-1"></a>we fit lines that would go through the genes that are missing in the TCGA</span>
<span id="cb3-838"><a href="#cb3-838" aria-hidden="true" tabindex="-1"></a>and METABRIC samples. We then try to find the line that is closes to the point.</span>
<span id="cb3-839"><a href="#cb3-839" aria-hidden="true" tabindex="-1"></a>After performing the fit and the line, we can then move the point to closer</span>
<span id="cb3-840"><a href="#cb3-840" aria-hidden="true" tabindex="-1"></a>to the METABRIC and TCGA samples. The point should be close to the line</span>
<span id="cb3-841"><a href="#cb3-841" aria-hidden="true" tabindex="-1"></a>and close to points that have the same proportion of top loadings </span>
<span id="cb3-842"><a href="#cb3-842" aria-hidden="true" tabindex="-1"></a>missing.</span>
<span id="cb3-843"><a href="#cb3-843" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-844"><a href="#cb3-844" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb3-845"><a href="#cb3-845" aria-hidden="true" tabindex="-1"></a><span class="in">folder_to_save &lt;- paste0(</span></span>
<span id="cb3-846"><a href="#cb3-846" aria-hidden="true" tabindex="-1"></a><span class="in">    "../../results/plots/validation/random_loadings"</span></span>
<span id="cb3-847"><a href="#cb3-847" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-848"><a href="#cb3-848" aria-hidden="true" tabindex="-1"></a><span class="in">dir.create(</span></span>
<span id="cb3-849"><a href="#cb3-849" aria-hidden="true" tabindex="-1"></a><span class="in">    folder_to_save, </span></span>
<span id="cb3-850"><a href="#cb3-850" aria-hidden="true" tabindex="-1"></a><span class="in">    showWarnings = FALSE, </span></span>
<span id="cb3-851"><a href="#cb3-851" aria-hidden="true" tabindex="-1"></a><span class="in">    recursive = TRUE</span></span>
<span id="cb3-852"><a href="#cb3-852" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-853"><a href="#cb3-853" aria-hidden="true" tabindex="-1"></a><span class="in">fig_width &lt;- 8</span></span>
<span id="cb3-854"><a href="#cb3-854" aria-hidden="true" tabindex="-1"></a><span class="in">fig_height &lt;- 5</span></span>
<span id="cb3-855"><a href="#cb3-855" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-856"><a href="#cb3-856" aria-hidden="true" tabindex="-1"></a><span class="in">sapply(</span></span>
<span id="cb3-857"><a href="#cb3-857" aria-hidden="true" tabindex="-1"></a><span class="in">    1:20,</span></span>
<span id="cb3-858"><a href="#cb3-858" aria-hidden="true" tabindex="-1"></a><span class="in">    function(i, final_plots){</span></span>
<span id="cb3-859"><a href="#cb3-859" aria-hidden="true" tabindex="-1"></a><span class="in">        ggsave(</span></span>
<span id="cb3-860"><a href="#cb3-860" aria-hidden="true" tabindex="-1"></a><span class="in">            filename = paste0(</span></span>
<span id="cb3-861"><a href="#cb3-861" aria-hidden="true" tabindex="-1"></a><span class="in">                folder_to_save, </span></span>
<span id="cb3-862"><a href="#cb3-862" aria-hidden="true" tabindex="-1"></a><span class="in">                "/", </span></span>
<span id="cb3-863"><a href="#cb3-863" aria-hidden="true" tabindex="-1"></a><span class="in">                ifelse(i &lt; 10, paste0("0",i), i), </span></span>
<span id="cb3-864"><a href="#cb3-864" aria-hidden="true" tabindex="-1"></a><span class="in">                ".png"</span></span>
<span id="cb3-865"><a href="#cb3-865" aria-hidden="true" tabindex="-1"></a><span class="in">            ),</span></span>
<span id="cb3-866"><a href="#cb3-866" aria-hidden="true" tabindex="-1"></a><span class="in">            plot = final_plots[[i]],</span></span>
<span id="cb3-867"><a href="#cb3-867" aria-hidden="true" tabindex="-1"></a><span class="in">            width = fig_width,</span></span>
<span id="cb3-868"><a href="#cb3-868" aria-hidden="true" tabindex="-1"></a><span class="in">            height = fig_height, </span></span>
<span id="cb3-869"><a href="#cb3-869" aria-hidden="true" tabindex="-1"></a><span class="in">            dpi = 320</span></span>
<span id="cb3-870"><a href="#cb3-870" aria-hidden="true" tabindex="-1"></a><span class="in">        )</span></span>
<span id="cb3-871"><a href="#cb3-871" aria-hidden="true" tabindex="-1"></a><span class="in">    },</span></span>
<span id="cb3-872"><a href="#cb3-872" aria-hidden="true" tabindex="-1"></a><span class="in">    final_plots = plots_random_genes</span></span>
<span id="cb3-873"><a href="#cb3-873" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-874"><a href="#cb3-874" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-875"><a href="#cb3-875" aria-hidden="true" tabindex="-1"></a><span class="in"># we then proceed to generate a mp4 by using the following command:</span></span>
<span id="cb3-876"><a href="#cb3-876" aria-hidden="true" tabindex="-1"></a><span class="in"># ffmpeg -framerate 1 -pattern_type glob -i 'random_loadings/*.png' random_loadings.mp4</span></span>
<span id="cb3-877"><a href="#cb3-877" aria-hidden="true" tabindex="-1"></a><span class="in">system(</span></span>
<span id="cb3-878"><a href="#cb3-878" aria-hidden="true" tabindex="-1"></a><span class="in">    paste0(</span></span>
<span id="cb3-879"><a href="#cb3-879" aria-hidden="true" tabindex="-1"></a><span class="in">        "ffmpeg -y -framerate 1 -pattern_type glob -i '",</span></span>
<span id="cb3-880"><a href="#cb3-880" aria-hidden="true" tabindex="-1"></a><span class="in">        folder_to_save, "/*.png' ", </span></span>
<span id="cb3-881"><a href="#cb3-881" aria-hidden="true" tabindex="-1"></a><span class="in">        folder_to_save, "/../random_loadings.mp4"</span></span>
<span id="cb3-882"><a href="#cb3-882" aria-hidden="true" tabindex="-1"></a><span class="in">    )</span></span>
<span id="cb3-883"><a href="#cb3-883" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-884"><a href="#cb3-884" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-885"><a href="#cb3-885" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-886"><a href="#cb3-886" aria-hidden="true" tabindex="-1"></a>These plots show how the fuzziness score might be indicative of how</span>
<span id="cb3-887"><a href="#cb3-887" aria-hidden="true" tabindex="-1"></a>good the embedding will be depending on the number of missing genes. The</span>
<span id="cb3-888"><a href="#cb3-888" aria-hidden="true" tabindex="-1"></a>video below shows the images for 20 different patients. Notice how all of</span>
<span id="cb3-889"><a href="#cb3-889" aria-hidden="true" tabindex="-1"></a>them are connected to the origin as well.</span>
<span id="cb3-890"><a href="#cb3-890" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-891"><a href="#cb3-891" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = !first_run, results = 'asis', echo = FALSE}</span></span>
<span id="cb3-892"><a href="#cb3-892" aria-hidden="true" tabindex="-1"></a><span class="in">embedding_video &lt;- paste0(</span></span>
<span id="cb3-893"><a href="#cb3-893" aria-hidden="true" tabindex="-1"></a><span class="in">    '&lt;iframe width="720" height="480" ',</span></span>
<span id="cb3-894"><a href="#cb3-894" aria-hidden="true" tabindex="-1"></a><span class="in">    'src="../plots/validation/random_loadings.mp4" ',</span></span>
<span id="cb3-895"><a href="#cb3-895" aria-hidden="true" tabindex="-1"></a><span class="in">    'align="middle" frameborder="0" allowfullscreen&gt;&lt;/iframe&gt;'</span></span>
<span id="cb3-896"><a href="#cb3-896" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-897"><a href="#cb3-897" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-898"><a href="#cb3-898" aria-hidden="true" tabindex="-1"></a><span class="in">cat(embedding_video)</span></span>
<span id="cb3-899"><a href="#cb3-899" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-900"><a href="#cb3-900" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-901"><a href="#cb3-901" aria-hidden="true" tabindex="-1"></a><span class="fu">### Systematic approach </span></span>
<span id="cb3-902"><a href="#cb3-902" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-903"><a href="#cb3-903" aria-hidden="true" tabindex="-1"></a>We now use a more systematic approach to see the impact of the missing</span>
<span id="cb3-904"><a href="#cb3-904" aria-hidden="true" tabindex="-1"></a>genes in the position in the embedding. The idea is that we extract</span>
<span id="cb3-905"><a href="#cb3-905" aria-hidden="true" tabindex="-1"></a>some metrics from the TCGA samples as we increase the amount of the most</span>
<span id="cb3-906"><a href="#cb3-906" aria-hidden="true" tabindex="-1"></a>important genes removed from the sample. What we simply do is </span>
<span id="cb3-907"><a href="#cb3-907" aria-hidden="true" tabindex="-1"></a>for each patient we calculate the distance of the new embedding to the</span>
<span id="cb3-908"><a href="#cb3-908" aria-hidden="true" tabindex="-1"></a>original embedding. Since for each proportion we ran the analysis 10 times </span>
<span id="cb3-909"><a href="#cb3-909" aria-hidden="true" tabindex="-1"></a>we get 10 distances. We average over those. In the end we have </span>
<span id="cb3-910"><a href="#cb3-910" aria-hidden="true" tabindex="-1"></a>20 "distances" for each one of the 200 TCGA samples.</span>
<span id="cb3-911"><a href="#cb3-911" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-914"><a href="#cb3-914" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-915"><a href="#cb3-915" aria-hidden="true" tabindex="-1"></a>distances_og_random <span class="ot">&lt;-</span> <span class="fu">sapply</span>(</span>
<span id="cb3-916"><a href="#cb3-916" aria-hidden="true" tabindex="-1"></a>    pca_random_genes_patients,</span>
<span id="cb3-917"><a href="#cb3-917" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(df){</span>
<span id="cb3-918"><a href="#cb3-918" aria-hidden="true" tabindex="-1"></a>        og_position <span class="ot">&lt;-</span> df <span class="sc">%&gt;%</span></span>
<span id="cb3-919"><a href="#cb3-919" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">filter</span>(embedding <span class="sc">==</span> <span class="st">"original"</span>)</span>
<span id="cb3-920"><a href="#cb3-920" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb3-921"><a href="#cb3-921" aria-hidden="true" tabindex="-1"></a>        df <span class="ot">&lt;-</span> df <span class="sc">%&gt;%</span></span>
<span id="cb3-922"><a href="#cb3-922" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">filter</span>(embedding <span class="sc">!=</span> <span class="st">"original"</span>)</span>
<span id="cb3-923"><a href="#cb3-923" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb3-924"><a href="#cb3-924" aria-hidden="true" tabindex="-1"></a>        df <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">bind_cols</span>(</span>
<span id="cb3-925"><a href="#cb3-925" aria-hidden="true" tabindex="-1"></a>            df,</span>
<span id="cb3-926"><a href="#cb3-926" aria-hidden="true" tabindex="-1"></a>            <span class="fu">data.frame</span>(</span>
<span id="cb3-927"><a href="#cb3-927" aria-hidden="true" tabindex="-1"></a>                <span class="at">PC3_og =</span> og_position<span class="sc">$</span>PC3,</span>
<span id="cb3-928"><a href="#cb3-928" aria-hidden="true" tabindex="-1"></a>                <span class="at">PC4_og =</span> og_position<span class="sc">$</span>PC4</span>
<span id="cb3-929"><a href="#cb3-929" aria-hidden="true" tabindex="-1"></a>            )</span>
<span id="cb3-930"><a href="#cb3-930" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb3-931"><a href="#cb3-931" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb3-932"><a href="#cb3-932" aria-hidden="true" tabindex="-1"></a>        df <span class="sc">%&gt;%</span></span>
<span id="cb3-933"><a href="#cb3-933" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb3-934"><a href="#cb3-934" aria-hidden="true" tabindex="-1"></a>                <span class="at">distance =</span> <span class="fu">sqrt</span>(</span>
<span id="cb3-935"><a href="#cb3-935" aria-hidden="true" tabindex="-1"></a>                    (PC3 <span class="sc">-</span> PC3_og)<span class="sc">^</span><span class="dv">2</span> <span class="sc">+</span> (PC4 <span class="sc">-</span> PC4_og)<span class="sc">^</span><span class="dv">2</span></span>
<span id="cb3-936"><a href="#cb3-936" aria-hidden="true" tabindex="-1"></a>                )</span>
<span id="cb3-937"><a href="#cb3-937" aria-hidden="true" tabindex="-1"></a>            ) <span class="sc">%&gt;%</span></span>
<span id="cb3-938"><a href="#cb3-938" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">group_by</span>(proportion) <span class="sc">%&gt;%</span></span>
<span id="cb3-939"><a href="#cb3-939" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">summarise</span>(</span>
<span id="cb3-940"><a href="#cb3-940" aria-hidden="true" tabindex="-1"></a>                <span class="at">distance =</span> <span class="fu">mean</span>(distance),</span>
<span id="cb3-941"><a href="#cb3-941" aria-hidden="true" tabindex="-1"></a>                <span class="at">proportion =</span> proportion[<span class="dv">1</span>]</span>
<span id="cb3-942"><a href="#cb3-942" aria-hidden="true" tabindex="-1"></a>            ) <span class="sc">%&gt;%</span></span>
<span id="cb3-943"><a href="#cb3-943" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">ungroup</span>()</span>
<span id="cb3-944"><a href="#cb3-944" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb3-945"><a href="#cb3-945" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>,</span>
<span id="cb3-946"><a href="#cb3-946" aria-hidden="true" tabindex="-1"></a>    <span class="at">simplify =</span> <span class="cn">FALSE</span></span>
<span id="cb3-947"><a href="#cb3-947" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb3-948"><a href="#cb3-948" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(<span class="at">.id =</span> <span class="st">"sample_name"</span>)</span>
<span id="cb3-949"><a href="#cb3-949" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-950"><a href="#cb3-950" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-951"><a href="#cb3-951" aria-hidden="true" tabindex="-1"></a>@fig-proportion-missing-genes-distance shows the relationship between</span>
<span id="cb3-952"><a href="#cb3-952" aria-hidden="true" tabindex="-1"></a>the proportion of missing top loading genes among the 200 genes that</span>
<span id="cb3-953"><a href="#cb3-953" aria-hidden="true" tabindex="-1"></a>were selected out and the average distance to the original embedding. Each</span>
<span id="cb3-954"><a href="#cb3-954" aria-hidden="true" tabindex="-1"></a>dot corresponds to the a sample. In total there are 200 dots in each</span>
<span id="cb3-955"><a href="#cb3-955" aria-hidden="true" tabindex="-1"></a>proportion.</span>
<span id="cb3-956"><a href="#cb3-956" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-957"><a href="#cb3-957" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=5}</span></span>
<span id="cb3-958"><a href="#cb3-958" aria-hidden="true" tabindex="-1"></a><span class="in">#| label: fig-proportion-missing-genes-distance</span></span>
<span id="cb3-959"><a href="#cb3-959" aria-hidden="true" tabindex="-1"></a><span class="in">#| fig-cap: Average distance to the original embedding </span></span>
<span id="cb3-960"><a href="#cb3-960" aria-hidden="true" tabindex="-1"></a><span class="in">distances_og_random %&gt;%</span></span>
<span id="cb3-961"><a href="#cb3-961" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::ggplot(</span></span>
<span id="cb3-962"><a href="#cb3-962" aria-hidden="true" tabindex="-1"></a><span class="in">        aes(</span></span>
<span id="cb3-963"><a href="#cb3-963" aria-hidden="true" tabindex="-1"></a><span class="in">            x = proportion, </span></span>
<span id="cb3-964"><a href="#cb3-964" aria-hidden="true" tabindex="-1"></a><span class="in">            y = distance, </span></span>
<span id="cb3-965"><a href="#cb3-965" aria-hidden="true" tabindex="-1"></a><span class="in">            color = proportion</span></span>
<span id="cb3-966"><a href="#cb3-966" aria-hidden="true" tabindex="-1"></a><span class="in">        )</span></span>
<span id="cb3-967"><a href="#cb3-967" aria-hidden="true" tabindex="-1"></a><span class="in">    ) + </span></span>
<span id="cb3-968"><a href="#cb3-968" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_boxplot(alpha = 0.4, color = "black", outlier.shape = NA) +</span></span>
<span id="cb3-969"><a href="#cb3-969" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_jitter(size = 0.1) +</span></span>
<span id="cb3-970"><a href="#cb3-970" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::labs(</span></span>
<span id="cb3-971"><a href="#cb3-971" aria-hidden="true" tabindex="-1"></a><span class="in">        x = "Proportion of missing genes among the top 200 loadings",</span></span>
<span id="cb3-972"><a href="#cb3-972" aria-hidden="true" tabindex="-1"></a><span class="in">        y = "Average distance"</span></span>
<span id="cb3-973"><a href="#cb3-973" aria-hidden="true" tabindex="-1"></a><span class="in">    ) +</span></span>
<span id="cb3-974"><a href="#cb3-974" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::scale_x_discrete(guide = ggplot2::guide_axis(angle = 45)) +</span></span>
<span id="cb3-975"><a href="#cb3-975" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme_bw(base_size = 20) +</span></span>
<span id="cb3-976"><a href="#cb3-976" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme(legend.position = "none")</span></span>
<span id="cb3-977"><a href="#cb3-977" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-978"><a href="#cb3-978" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-979"><a href="#cb3-979" aria-hidden="true" tabindex="-1"></a>We see that the more top loading genes that are</span>
<span id="cb3-980"><a href="#cb3-980" aria-hidden="true" tabindex="-1"></a>missing the worse the embedding gets. Still the average distance is </span>
<span id="cb3-981"><a href="#cb3-981" aria-hidden="true" tabindex="-1"></a>a bit difficult to interpret, so next we show the relationship between </span>
<span id="cb3-982"><a href="#cb3-982" aria-hidden="true" tabindex="-1"></a>the variance and the proportion of missing genes. </span>
<span id="cb3-983"><a href="#cb3-983" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-986"><a href="#cb3-986" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-987"><a href="#cb3-987" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-var-proportion-missing-genes</span></span>
<span id="cb3-988"><a href="#cb3-988" aria-hidden="true" tabindex="-1"></a>distances_og_random <span class="sc">%&gt;%</span></span>
<span id="cb3-989"><a href="#cb3-989" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">group_by</span>(proportion) <span class="sc">%&gt;%</span></span>
<span id="cb3-990"><a href="#cb3-990" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">summarise</span>(</span>
<span id="cb3-991"><a href="#cb3-991" aria-hidden="true" tabindex="-1"></a>        <span class="at">mean =</span> <span class="fu">mean</span>(distance),</span>
<span id="cb3-992"><a href="#cb3-992" aria-hidden="true" tabindex="-1"></a>        <span class="at">var =</span> <span class="fu">var</span>(distance)</span>
<span id="cb3-993"><a href="#cb3-993" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb3-994"><a href="#cb3-994" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">proportion =</span> <span class="fu">as.numeric</span>(proportion)) <span class="sc">%&gt;%</span></span>
<span id="cb3-995"><a href="#cb3-995" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> proportion, <span class="at">y =</span> var)) <span class="sc">+</span></span>
<span id="cb3-996"><a href="#cb3-996" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span> </span>
<span id="cb3-997"><a href="#cb3-997" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">formula =</span> <span class="st">"y~x"</span>) <span class="sc">+</span></span>
<span id="cb3-998"><a href="#cb3-998" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb3-999"><a href="#cb3-999" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"Proportion of top loading missing genes"</span>,</span>
<span id="cb3-1000"><a href="#cb3-1000" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"Variance of the average distance"</span></span>
<span id="cb3-1001"><a href="#cb3-1001" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb3-1002"><a href="#cb3-1002" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">18</span>)</span>
<span id="cb3-1003"><a href="#cb3-1003" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-1004"><a href="#cb3-1004" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1005"><a href="#cb3-1005" aria-hidden="true" tabindex="-1"></a>Indeed as the proportion of the most important genes goes missing, there is</span>
<span id="cb3-1006"><a href="#cb3-1006" aria-hidden="true" tabindex="-1"></a>a higher variance in the average distance.</span>
<span id="cb3-1007"><a href="#cb3-1007" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1008"><a href="#cb3-1008" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb3-1009"><a href="#cb3-1009" aria-hidden="true" tabindex="-1"></a><span class="in"># number of samples in a neighborhood as the radius increases</span></span>
<span id="cb3-1010"><a href="#cb3-1010" aria-hidden="true" tabindex="-1"></a><span class="in">patients_tcga_distances_avg &lt;- mapply(</span></span>
<span id="cb3-1011"><a href="#cb3-1011" aria-hidden="true" tabindex="-1"></a><span class="in">    function(df, sample_name, df_pca){</span></span>
<span id="cb3-1012"><a href="#cb3-1012" aria-hidden="true" tabindex="-1"></a><span class="in">        </span></span>
<span id="cb3-1013"><a href="#cb3-1013" aria-hidden="true" tabindex="-1"></a><span class="in">        radii &lt;- seq(0.1, 5, by = 0.3)</span></span>
<span id="cb3-1014"><a href="#cb3-1014" aria-hidden="true" tabindex="-1"></a><span class="in">        </span></span>
<span id="cb3-1015"><a href="#cb3-1015" aria-hidden="true" tabindex="-1"></a><span class="in">        components_sample &lt;- df %&gt;%</span></span>
<span id="cb3-1016"><a href="#cb3-1016" aria-hidden="true" tabindex="-1"></a><span class="in">            dplyr::filter(embedding == "original") %&gt;%</span></span>
<span id="cb3-1017"><a href="#cb3-1017" aria-hidden="true" tabindex="-1"></a><span class="in">            data.frame %&gt;%</span></span>
<span id="cb3-1018"><a href="#cb3-1018" aria-hidden="true" tabindex="-1"></a><span class="in">            dplyr::select(PC3, PC4)</span></span>
<span id="cb3-1019"><a href="#cb3-1019" aria-hidden="true" tabindex="-1"></a><span class="in">        </span></span>
<span id="cb3-1020"><a href="#cb3-1020" aria-hidden="true" tabindex="-1"></a><span class="in">        components_sample &lt;- components_sample %&gt;% as.matrix</span></span>
<span id="cb3-1021"><a href="#cb3-1021" aria-hidden="true" tabindex="-1"></a><span class="in">        components_sample &lt;- components_sample[1, ]</span></span>
<span id="cb3-1022"><a href="#cb3-1022" aria-hidden="true" tabindex="-1"></a><span class="in">        </span></span>
<span id="cb3-1023"><a href="#cb3-1023" aria-hidden="true" tabindex="-1"></a><span class="in">        sapply(</span></span>
<span id="cb3-1024"><a href="#cb3-1024" aria-hidden="true" tabindex="-1"></a><span class="in">            radii,</span></span>
<span id="cb3-1025"><a href="#cb3-1025" aria-hidden="true" tabindex="-1"></a><span class="in">            get_samples_neighborhood,</span></span>
<span id="cb3-1026"><a href="#cb3-1026" aria-hidden="true" tabindex="-1"></a><span class="in">            components_sample = components_sample,</span></span>
<span id="cb3-1027"><a href="#cb3-1027" aria-hidden="true" tabindex="-1"></a><span class="in">            df_pca = df_pca %&gt;%</span></span>
<span id="cb3-1028"><a href="#cb3-1028" aria-hidden="true" tabindex="-1"></a><span class="in">                dplyr::filter(cohort %in% c("tcga", "metabric")),</span></span>
<span id="cb3-1029"><a href="#cb3-1029" aria-hidden="true" tabindex="-1"></a><span class="in">            sample_name = sample_name,</span></span>
<span id="cb3-1030"><a href="#cb3-1030" aria-hidden="true" tabindex="-1"></a><span class="in">            USE.NAMES = TRUE,</span></span>
<span id="cb3-1031"><a href="#cb3-1031" aria-hidden="true" tabindex="-1"></a><span class="in">            simplify = FALSE</span></span>
<span id="cb3-1032"><a href="#cb3-1032" aria-hidden="true" tabindex="-1"></a><span class="in">        ) %&gt;%</span></span>
<span id="cb3-1033"><a href="#cb3-1033" aria-hidden="true" tabindex="-1"></a><span class="in">            `names&lt;-`(radii)</span></span>
<span id="cb3-1034"><a href="#cb3-1034" aria-hidden="true" tabindex="-1"></a><span class="in">        </span></span>
<span id="cb3-1035"><a href="#cb3-1035" aria-hidden="true" tabindex="-1"></a><span class="in">    },</span></span>
<span id="cb3-1036"><a href="#cb3-1036" aria-hidden="true" tabindex="-1"></a><span class="in">    df = pca_random_genes_patients,</span></span>
<span id="cb3-1037"><a href="#cb3-1037" aria-hidden="true" tabindex="-1"></a><span class="in">    sample_name = names(pca_random_genes_patients),</span></span>
<span id="cb3-1038"><a href="#cb3-1038" aria-hidden="true" tabindex="-1"></a><span class="in">    MoreArgs = list(df_pca = df_pca_coordinates),</span></span>
<span id="cb3-1039"><a href="#cb3-1039" aria-hidden="true" tabindex="-1"></a><span class="in">    USE.NAMES = TRUE,</span></span>
<span id="cb3-1040"><a href="#cb3-1040" aria-hidden="true" tabindex="-1"></a><span class="in">    SIMPLIFY = FALSE</span></span>
<span id="cb3-1041"><a href="#cb3-1041" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-1042"><a href="#cb3-1042" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1043"><a href="#cb3-1043" aria-hidden="true" tabindex="-1"></a><span class="in">patients_tcga_nb_samples_neighborhood &lt;- sapply(</span></span>
<span id="cb3-1044"><a href="#cb3-1044" aria-hidden="true" tabindex="-1"></a><span class="in">    patients_tcga_distances_avg,</span></span>
<span id="cb3-1045"><a href="#cb3-1045" aria-hidden="true" tabindex="-1"></a><span class="in">    function(x){sapply(x, function(y) length(y$samples))}</span></span>
<span id="cb3-1046"><a href="#cb3-1046" aria-hidden="true" tabindex="-1"></a><span class="in">) %&gt;%</span></span>
<span id="cb3-1047"><a href="#cb3-1047" aria-hidden="true" tabindex="-1"></a><span class="in">    data.frame(check.names = FALSE) %&gt;%</span></span>
<span id="cb3-1048"><a href="#cb3-1048" aria-hidden="true" tabindex="-1"></a><span class="in">    tibble::rownames_to_column(var = "radii") %&gt;%</span></span>
<span id="cb3-1049"><a href="#cb3-1049" aria-hidden="true" tabindex="-1"></a><span class="in">    tidyr::pivot_longer(</span></span>
<span id="cb3-1050"><a href="#cb3-1050" aria-hidden="true" tabindex="-1"></a><span class="in">        cols = -dplyr::all_of(c("radii")), </span></span>
<span id="cb3-1051"><a href="#cb3-1051" aria-hidden="true" tabindex="-1"></a><span class="in">        names_to = "sample_name",</span></span>
<span id="cb3-1052"><a href="#cb3-1052" aria-hidden="true" tabindex="-1"></a><span class="in">        values_to = "nb_samples"</span></span>
<span id="cb3-1053"><a href="#cb3-1053" aria-hidden="true" tabindex="-1"></a><span class="in">    )</span></span>
<span id="cb3-1054"><a href="#cb3-1054" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1055"><a href="#cb3-1055" aria-hidden="true" tabindex="-1"></a><span class="in">nb_samples_avg &lt;- patients_tcga_nb_samples_neighborhood %&gt;%</span></span>
<span id="cb3-1056"><a href="#cb3-1056" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::group_by(radii) %&gt;%</span></span>
<span id="cb3-1057"><a href="#cb3-1057" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::summarise(</span></span>
<span id="cb3-1058"><a href="#cb3-1058" aria-hidden="true" tabindex="-1"></a><span class="in">        nb_samples = mean(nb_samples)</span></span>
<span id="cb3-1059"><a href="#cb3-1059" aria-hidden="true" tabindex="-1"></a><span class="in">    ) %&gt;%</span></span>
<span id="cb3-1060"><a href="#cb3-1060" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::ungroup() %&gt;%</span></span>
<span id="cb3-1061"><a href="#cb3-1061" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::mutate(nb_samples_perc = </span></span>
<span id="cb3-1062"><a href="#cb3-1062" aria-hidden="true" tabindex="-1"></a><span class="in">        nb_samples/nrow(</span></span>
<span id="cb3-1063"><a href="#cb3-1063" aria-hidden="true" tabindex="-1"></a><span class="in">            df_pca_coordinates %&gt;% </span></span>
<span id="cb3-1064"><a href="#cb3-1064" aria-hidden="true" tabindex="-1"></a><span class="in">                dplyr::filter(cohort %in% c("tcga", "metabric"))</span></span>
<span id="cb3-1065"><a href="#cb3-1065" aria-hidden="true" tabindex="-1"></a><span class="in">        )*100</span></span>
<span id="cb3-1066"><a href="#cb3-1066" aria-hidden="true" tabindex="-1"></a><span class="in">    )</span></span>
<span id="cb3-1067"><a href="#cb3-1067" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1068"><a href="#cb3-1068" aria-hidden="true" tabindex="-1"></a><span class="in">saveRDS(</span></span>
<span id="cb3-1069"><a href="#cb3-1069" aria-hidden="true" tabindex="-1"></a><span class="in">    nb_samples_avg,</span></span>
<span id="cb3-1070"><a href="#cb3-1070" aria-hidden="true" tabindex="-1"></a><span class="in">    "../../results/rds_files/validation/nb_samples_avg.rds"   </span></span>
<span id="cb3-1071"><a href="#cb3-1071" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-1072"><a href="#cb3-1072" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-1073"><a href="#cb3-1073" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1074"><a href="#cb3-1074" aria-hidden="true" tabindex="-1"></a>Next we evaluate the number of samples in the neighborhood of samples, so</span>
<span id="cb3-1075"><a href="#cb3-1075" aria-hidden="true" tabindex="-1"></a>we can understand a bit better what is distance in the molecular embedder. </span>
<span id="cb3-1076"><a href="#cb3-1076" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1077"><a href="#cb3-1077" aria-hidden="true" tabindex="-1"></a>@fig-nb-samples-avg-neighborhood shows the proportion of samples in a </span>
<span id="cb3-1078"><a href="#cb3-1078" aria-hidden="true" tabindex="-1"></a>neighborhood based on its radius. For samples within a radius of 1, in average</span>
<span id="cb3-1079"><a href="#cb3-1079" aria-hidden="true" tabindex="-1"></a>the percentage of samples from the whole dataset including TCGA and METABRIC</span>
<span id="cb3-1080"><a href="#cb3-1080" aria-hidden="true" tabindex="-1"></a>is of 2.5%.</span>
<span id="cb3-1081"><a href="#cb3-1081" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1084"><a href="#cb3-1084" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-1085"><a href="#cb3-1085" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-nb-samples-avg-neighborhood</span></span>
<span id="cb3-1086"><a href="#cb3-1086" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Percentage of number of samples in the neighborhood with </span></span>
<span id="cb3-1087"><a href="#cb3-1087" aria-hidden="true" tabindex="-1"></a><span class="co">#|     varyiing radii. The total number of samples is 2873, including all</span></span>
<span id="cb3-1088"><a href="#cb3-1088" aria-hidden="true" tabindex="-1"></a><span class="co">#|     samples from TCGA and METABRIC.</span></span>
<span id="cb3-1089"><a href="#cb3-1089" aria-hidden="true" tabindex="-1"></a>nb_samples_avg <span class="sc">%&gt;%</span></span>
<span id="cb3-1090"><a href="#cb3-1090" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">radii =</span> <span class="fu">as.numeric</span>(radii)) <span class="sc">%&gt;%</span></span>
<span id="cb3-1091"><a href="#cb3-1091" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> radii, <span class="at">y =</span> nb_samples_perc)) <span class="sc">+</span> </span>
<span id="cb3-1092"><a href="#cb3-1092" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb3-1093"><a href="#cb3-1093" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb3-1094"><a href="#cb3-1094" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"Radii"</span>, </span>
<span id="cb3-1095"><a href="#cb3-1095" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"Percentage of samples</span><span class="sc">\n</span><span class="st">in the neighborhood"</span></span>
<span id="cb3-1096"><a href="#cb3-1096" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb3-1097"><a href="#cb3-1097" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">1</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb3-1098"><a href="#cb3-1098" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="fl">2.5</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb3-1099"><a href="#cb3-1099" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_y_continuous</span>(</span>
<span id="cb3-1100"><a href="#cb3-1100" aria-hidden="true" tabindex="-1"></a>        <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">40</span>, <span class="at">by =</span> <span class="fl">2.5</span>),</span>
<span id="cb3-1101"><a href="#cb3-1101" aria-hidden="true" tabindex="-1"></a>        <span class="at">labels =</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">40</span>, <span class="at">by =</span> <span class="fl">2.5</span>)</span>
<span id="cb3-1102"><a href="#cb3-1102" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb3-1103"><a href="#cb3-1103" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>)</span>
<span id="cb3-1104"><a href="#cb3-1104" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-1105"><a href="#cb3-1105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1106"><a href="#cb3-1106" aria-hidden="true" tabindex="-1"></a>This means that around 30% of top missing genes gives us neighborhoods</span>
<span id="cb3-1107"><a href="#cb3-1107" aria-hidden="true" tabindex="-1"></a>that do not change much, since the average distance from the original embedding</span>
<span id="cb3-1108"><a href="#cb3-1108" aria-hidden="true" tabindex="-1"></a>is below 1, so the neighborhood will not change much.</span>
<span id="cb3-1109"><a href="#cb3-1109" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1110"><a href="#cb3-1110" aria-hidden="true" tabindex="-1"></a><span class="fu">## SMC embedding</span></span>
<span id="cb3-1111"><a href="#cb3-1111" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1112"><a href="#cb3-1112" aria-hidden="true" tabindex="-1"></a>By following the same procedures as previously, normalizing and getting</span>
<span id="cb3-1113"><a href="#cb3-1113" aria-hidden="true" tabindex="-1"></a>the PC coordinates, @fig-smc-cohort shows the biplot of PC1 and PC2</span>
<span id="cb3-1114"><a href="#cb3-1114" aria-hidden="true" tabindex="-1"></a>when coloring by cohorts. The samples seem to be well mixed already in</span>
<span id="cb3-1115"><a href="#cb3-1115" aria-hidden="true" tabindex="-1"></a>the RNA-seq samples. This is probably due to the fact that the SMC cohort</span>
<span id="cb3-1116"><a href="#cb3-1116" aria-hidden="true" tabindex="-1"></a>has TPM values. </span>
<span id="cb3-1117"><a href="#cb3-1117" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1118"><a href="#cb3-1118" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb3-1119"><a href="#cb3-1119" aria-hidden="true" tabindex="-1"></a><span class="in">clin_data_smc &lt;- read.table(</span></span>
<span id="cb3-1120"><a href="#cb3-1120" aria-hidden="true" tabindex="-1"></a><span class="in">    "../../data/2018_SMC/data_clinical_sample.txt",</span></span>
<span id="cb3-1121"><a href="#cb3-1121" aria-hidden="true" tabindex="-1"></a><span class="in">    sep = "\t",</span></span>
<span id="cb3-1122"><a href="#cb3-1122" aria-hidden="true" tabindex="-1"></a><span class="in">    header = TRUE</span></span>
<span id="cb3-1123"><a href="#cb3-1123" aria-hidden="true" tabindex="-1"></a><span class="in">) %&gt;% dplyr::filter(</span></span>
<span id="cb3-1124"><a href="#cb3-1124" aria-hidden="true" tabindex="-1"></a><span class="in">    !is.na(PAM50_SUBTYPE)</span></span>
<span id="cb3-1125"><a href="#cb3-1125" aria-hidden="true" tabindex="-1"></a><span class="in">) %&gt;%</span></span>
<span id="cb3-1126"><a href="#cb3-1126" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::mutate(</span></span>
<span id="cb3-1127"><a href="#cb3-1127" aria-hidden="true" tabindex="-1"></a><span class="in">        pam50 = tolower(PAM50_SUBTYPE)</span></span>
<span id="cb3-1128"><a href="#cb3-1128" aria-hidden="true" tabindex="-1"></a><span class="in">    ) %&gt;%</span></span>
<span id="cb3-1129"><a href="#cb3-1129" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::mutate(</span></span>
<span id="cb3-1130"><a href="#cb3-1130" aria-hidden="true" tabindex="-1"></a><span class="in">        pam50 = stringr::str_replace_all(pam50, "luminal", "lum")</span></span>
<span id="cb3-1131"><a href="#cb3-1131" aria-hidden="true" tabindex="-1"></a><span class="in">    ) %&gt;%</span></span>
<span id="cb3-1132"><a href="#cb3-1132" aria-hidden="true" tabindex="-1"></a><span class="in">    `rownames&lt;-`(.$SAMPLE_ID) %&gt;%</span></span>
<span id="cb3-1133"><a href="#cb3-1133" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::mutate(sample_name = SAMPLE_ID)</span></span>
<span id="cb3-1134"><a href="#cb3-1134" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1135"><a href="#cb3-1135" aria-hidden="true" tabindex="-1"></a><span class="in">smc_df &lt;- read.table(</span></span>
<span id="cb3-1136"><a href="#cb3-1136" aria-hidden="true" tabindex="-1"></a><span class="in">    "../../data/2018_SMC/data_mrna_seq_tpm.txt",</span></span>
<span id="cb3-1137"><a href="#cb3-1137" aria-hidden="true" tabindex="-1"></a><span class="in">    sep = "\t",</span></span>
<span id="cb3-1138"><a href="#cb3-1138" aria-hidden="true" tabindex="-1"></a><span class="in">    header = TRUE</span></span>
<span id="cb3-1139"><a href="#cb3-1139" aria-hidden="true" tabindex="-1"></a><span class="in">) %&gt;% </span></span>
<span id="cb3-1140"><a href="#cb3-1140" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::filter(!duplicated(Hugo_Symbol)) %&gt;%</span></span>
<span id="cb3-1141"><a href="#cb3-1141" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::mutate(Entrez_Gene_Id = NULL) %&gt;%</span></span>
<span id="cb3-1142"><a href="#cb3-1142" aria-hidden="true" tabindex="-1"></a><span class="in">    `rownames&lt;-`(.$Hugo_Symbol) %&gt;%</span></span>
<span id="cb3-1143"><a href="#cb3-1143" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::mutate(Hugo_Symbol = NULL)</span></span>
<span id="cb3-1144"><a href="#cb3-1144" aria-hidden="true" tabindex="-1"></a><span class="in">smc_df &lt;- smc_df[, rownames(clin_data_smc)]</span></span>
<span id="cb3-1145"><a href="#cb3-1145" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1146"><a href="#cb3-1146" aria-hidden="true" tabindex="-1"></a><span class="in"># almost all genes are available, only 4 are missing</span></span>
<span id="cb3-1147"><a href="#cb3-1147" aria-hidden="true" tabindex="-1"></a><span class="in">smc_df &lt;- SummarizedExperiment::SummarizedExperiment(</span></span>
<span id="cb3-1148"><a href="#cb3-1148" aria-hidden="true" tabindex="-1"></a><span class="in">    assays = list(</span></span>
<span id="cb3-1149"><a href="#cb3-1149" aria-hidden="true" tabindex="-1"></a><span class="in">        "tpm" = smc_df,</span></span>
<span id="cb3-1150"><a href="#cb3-1150" aria-hidden="true" tabindex="-1"></a><span class="in">        "logtpm" = log2(smc_df + 0.01)</span></span>
<span id="cb3-1151"><a href="#cb3-1151" aria-hidden="true" tabindex="-1"></a><span class="in">    ),</span></span>
<span id="cb3-1152"><a href="#cb3-1152" aria-hidden="true" tabindex="-1"></a><span class="in">    colData = clin_data_smc</span></span>
<span id="cb3-1153"><a href="#cb3-1153" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-1154"><a href="#cb3-1154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1155"><a href="#cb3-1155" aria-hidden="true" tabindex="-1"></a><span class="in"># get the normalization performed </span></span>
<span id="cb3-1156"><a href="#cb3-1156" aria-hidden="true" tabindex="-1"></a><span class="in">smc_normalized &lt;- get_final_ranking_values(</span></span>
<span id="cb3-1157"><a href="#cb3-1157" aria-hidden="true" tabindex="-1"></a><span class="in">    sum_exp = smc_df,</span></span>
<span id="cb3-1158"><a href="#cb3-1158" aria-hidden="true" tabindex="-1"></a><span class="in">    assay_to_use = "logtpm",</span></span>
<span id="cb3-1159"><a href="#cb3-1159" aria-hidden="true" tabindex="-1"></a><span class="in">    stable_genes = stable_genes,</span></span>
<span id="cb3-1160"><a href="#cb3-1160" aria-hidden="true" tabindex="-1"></a><span class="in">    most_variable_genes = setdiff(rownames(pca_fit$loadings), stable_genes)</span></span>
<span id="cb3-1161"><a href="#cb3-1161" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-1162"><a href="#cb3-1162" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1163"><a href="#cb3-1163" aria-hidden="true" tabindex="-1"></a><span class="in"># calculate the embedding</span></span>
<span id="cb3-1164"><a href="#cb3-1164" aria-hidden="true" tabindex="-1"></a><span class="in">smc_df_pca &lt;- get_pca_coordinates(smc_normalized, pca_fit) %&gt;%</span></span>
<span id="cb3-1165"><a href="#cb3-1165" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::bind_cols(</span></span>
<span id="cb3-1166"><a href="#cb3-1166" aria-hidden="true" tabindex="-1"></a><span class="in">        ., </span></span>
<span id="cb3-1167"><a href="#cb3-1167" aria-hidden="true" tabindex="-1"></a><span class="in">        colData(smc_normalized) %&gt;% data.frame %&gt;% </span></span>
<span id="cb3-1168"><a href="#cb3-1168" aria-hidden="true" tabindex="-1"></a><span class="in">            dplyr::mutate(</span></span>
<span id="cb3-1169"><a href="#cb3-1169" aria-hidden="true" tabindex="-1"></a><span class="in">                er_status = ifelse(</span></span>
<span id="cb3-1170"><a href="#cb3-1170" aria-hidden="true" tabindex="-1"></a><span class="in">                    str_detect(IHC_SUBTYPE, "ER\\+"), "pos", "neg"</span></span>
<span id="cb3-1171"><a href="#cb3-1171" aria-hidden="true" tabindex="-1"></a><span class="in">                )</span></span>
<span id="cb3-1172"><a href="#cb3-1172" aria-hidden="true" tabindex="-1"></a><span class="in">            ) %&gt;% </span></span>
<span id="cb3-1173"><a href="#cb3-1173" aria-hidden="true" tabindex="-1"></a><span class="in">            dplyr::mutate(cohort = "smc")</span></span>
<span id="cb3-1174"><a href="#cb3-1174" aria-hidden="true" tabindex="-1"></a><span class="in">    ) %&gt;% </span></span>
<span id="cb3-1175"><a href="#cb3-1175" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::bind_rows(., df_pca_coordinates)</span></span>
<span id="cb3-1176"><a href="#cb3-1176" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1177"><a href="#cb3-1177" aria-hidden="true" tabindex="-1"></a><span class="in"># save all the results</span></span>
<span id="cb3-1178"><a href="#cb3-1178" aria-hidden="true" tabindex="-1"></a><span class="in">saveRDS(</span></span>
<span id="cb3-1179"><a href="#cb3-1179" aria-hidden="true" tabindex="-1"></a><span class="in">    smc_df,</span></span>
<span id="cb3-1180"><a href="#cb3-1180" aria-hidden="true" tabindex="-1"></a><span class="in">    "../../results/rds_files/validation/smc_df.rds"</span></span>
<span id="cb3-1181"><a href="#cb3-1181" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-1182"><a href="#cb3-1182" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1183"><a href="#cb3-1183" aria-hidden="true" tabindex="-1"></a><span class="in">saveRDS(</span></span>
<span id="cb3-1184"><a href="#cb3-1184" aria-hidden="true" tabindex="-1"></a><span class="in">    smc_df_pca %&gt;% data.frame,</span></span>
<span id="cb3-1185"><a href="#cb3-1185" aria-hidden="true" tabindex="-1"></a><span class="in">    "../../results/rds_files/validation/smc_df_pca.rds"</span></span>
<span id="cb3-1186"><a href="#cb3-1186" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-1187"><a href="#cb3-1187" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1188"><a href="#cb3-1188" aria-hidden="true" tabindex="-1"></a><span class="in">saveRDS(</span></span>
<span id="cb3-1189"><a href="#cb3-1189" aria-hidden="true" tabindex="-1"></a><span class="in">    smc_normalized,</span></span>
<span id="cb3-1190"><a href="#cb3-1190" aria-hidden="true" tabindex="-1"></a><span class="in">    "../../results/rds_files/validation/smc_normalized.rds"</span></span>
<span id="cb3-1191"><a href="#cb3-1191" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-1192"><a href="#cb3-1192" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-1193"><a href="#cb3-1193" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1194"><a href="#cb3-1194" aria-hidden="true" tabindex="-1"></a>When checking the total number of genes missing, there are only </span>
<span id="cb3-1195"><a href="#cb3-1195" aria-hidden="true" tabindex="-1"></a><span class="in">`r nrow(datasets_normalized$tcga) - nrow(smc_normalized)`</span> genes missing. </span>
<span id="cb3-1196"><a href="#cb3-1196" aria-hidden="true" tabindex="-1"></a>And as seen before, this number is almost irrelevant, in the sense that</span>
<span id="cb3-1197"><a href="#cb3-1197" aria-hidden="true" tabindex="-1"></a>the embedding will still be robust and locations will be precise enough.</span>
<span id="cb3-1198"><a href="#cb3-1198" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1201"><a href="#cb3-1201" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-1202"><a href="#cb3-1202" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-smc-cohort</span></span>
<span id="cb3-1203"><a href="#cb3-1203" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Biplot of first two PCA components from SMC, TCGA, SCANB and </span></span>
<span id="cb3-1204"><a href="#cb3-1204" aria-hidden="true" tabindex="-1"></a><span class="co">#|     METABRIC. SMC is highlighted in the plot and has a bigger dot.</span></span>
<span id="cb3-1205"><a href="#cb3-1205" aria-hidden="true" tabindex="-1"></a>smc_df_pca <span class="sc">%&gt;%</span> </span>
<span id="cb3-1206"><a href="#cb3-1206" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(</span>
<span id="cb3-1207"><a href="#cb3-1207" aria-hidden="true" tabindex="-1"></a>        <span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> cohort)</span>
<span id="cb3-1208"><a href="#cb3-1208" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb3-1209"><a href="#cb3-1209" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">alpha =</span> cohort, <span class="at">size =</span> cohort)) <span class="sc">+</span></span>
<span id="cb3-1210"><a href="#cb3-1210" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_size_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(</span>
<span id="cb3-1211"><a href="#cb3-1211" aria-hidden="true" tabindex="-1"></a>        <span class="st">"smc"</span> <span class="ot">=</span> <span class="dv">2</span>, </span>
<span id="cb3-1212"><a href="#cb3-1212" aria-hidden="true" tabindex="-1"></a>        <span class="st">"tcga"</span> <span class="ot">=</span> <span class="dv">1</span>,</span>
<span id="cb3-1213"><a href="#cb3-1213" aria-hidden="true" tabindex="-1"></a>        <span class="st">"scanb"</span> <span class="ot">=</span> <span class="dv">1</span>,</span>
<span id="cb3-1214"><a href="#cb3-1214" aria-hidden="true" tabindex="-1"></a>        <span class="st">"metabric"</span> <span class="ot">=</span> <span class="dv">1</span></span>
<span id="cb3-1215"><a href="#cb3-1215" aria-hidden="true" tabindex="-1"></a>    ), <span class="at">guide =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb3-1216"><a href="#cb3-1216" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_alpha_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(</span>
<span id="cb3-1217"><a href="#cb3-1217" aria-hidden="true" tabindex="-1"></a>        <span class="st">"smc"</span> <span class="ot">=</span> <span class="dv">1</span>, </span>
<span id="cb3-1218"><a href="#cb3-1218" aria-hidden="true" tabindex="-1"></a>        <span class="st">"tcga"</span> <span class="ot">=</span> <span class="fl">0.1</span>,</span>
<span id="cb3-1219"><a href="#cb3-1219" aria-hidden="true" tabindex="-1"></a>        <span class="st">"scanb"</span> <span class="ot">=</span> <span class="fl">0.1</span>,</span>
<span id="cb3-1220"><a href="#cb3-1220" aria-hidden="true" tabindex="-1"></a>        <span class="st">"metabric"</span> <span class="ot">=</span> <span class="fl">0.1</span></span>
<span id="cb3-1221"><a href="#cb3-1221" aria-hidden="true" tabindex="-1"></a>    ), <span class="at">guide =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb3-1222"><a href="#cb3-1222" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb3-1223"><a href="#cb3-1223" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">"Cohort"</span>,</span>
<span id="cb3-1224"><a href="#cb3-1224" aria-hidden="true" tabindex="-1"></a>        <span class="at">alpha =</span> <span class="st">"Cohort"</span>,</span>
<span id="cb3-1225"><a href="#cb3-1225" aria-hidden="true" tabindex="-1"></a>        <span class="at">size =</span> <span class="st">"Cohort"</span>,</span>
<span id="cb3-1226"><a href="#cb3-1226" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="fu">paste0</span>(</span>
<span id="cb3-1227"><a href="#cb3-1227" aria-hidden="true" tabindex="-1"></a>            <span class="st">"First two components of SMC overlayed on all the three</span><span class="sc">\n</span><span class="st">"</span>,</span>
<span id="cb3-1228"><a href="#cb3-1228" aria-hidden="true" tabindex="-1"></a>            <span class="st">"big cohorts in the the molecular landscape"</span></span>
<span id="cb3-1229"><a href="#cb3-1229" aria-hidden="true" tabindex="-1"></a>        ),</span>
<span id="cb3-1230"><a href="#cb3-1230" aria-hidden="true" tabindex="-1"></a>        <span class="at">subtitle =</span> <span class="st">"All samples from TCGA, METABRIC and SCANB are plotted"</span></span>
<span id="cb3-1231"><a href="#cb3-1231" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb3-1232"><a href="#cb3-1232" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">15</span>) <span class="sc">+</span> </span>
<span id="cb3-1233"><a href="#cb3-1233" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span></span>
<span id="cb3-1234"><a href="#cb3-1234" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb3-1235"><a href="#cb3-1235" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-1236"><a href="#cb3-1236" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1237"><a href="#cb3-1237" aria-hidden="true" tabindex="-1"></a>As previously, @fig-pca-smc-er-pam50 shows the biplot of PC3 and PC4, </span>
<span id="cb3-1238"><a href="#cb3-1238" aria-hidden="true" tabindex="-1"></a>highlighting the well mixing of the cohorts. When coloring by ER status,</span>
<span id="cb3-1239"><a href="#cb3-1239" aria-hidden="true" tabindex="-1"></a>samples are in the right group. Molecular subtype is also correct. This</span>
<span id="cb3-1240"><a href="#cb3-1240" aria-hidden="true" tabindex="-1"></a>cohort is special in the sense that there are more luminal B patients,</span>
<span id="cb3-1241"><a href="#cb3-1241" aria-hidden="true" tabindex="-1"></a>as discussed in the paper. That is why there are more samples in the </span>
<span id="cb3-1242"><a href="#cb3-1242" aria-hidden="true" tabindex="-1"></a>luminal B region.</span>
<span id="cb3-1243"><a href="#cb3-1243" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1244"><a href="#cb3-1244" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=18, fig.height=11}</span></span>
<span id="cb3-1245"><a href="#cb3-1245" aria-hidden="true" tabindex="-1"></a><span class="in">#| label: fig-pca-smc-er-pam50</span></span>
<span id="cb3-1246"><a href="#cb3-1246" aria-hidden="true" tabindex="-1"></a><span class="in">#| fig-cap: PCA embedding of all samples from TCGA, SCANB and METABRIC including</span></span>
<span id="cb3-1247"><a href="#cb3-1247" aria-hidden="true" tabindex="-1"></a><span class="in">#|     the SMC samples on top.</span></span>
<span id="cb3-1248"><a href="#cb3-1248" aria-hidden="true" tabindex="-1"></a><span class="in">#|     (A) Colored by cohort,</span></span>
<span id="cb3-1249"><a href="#cb3-1249" aria-hidden="true" tabindex="-1"></a><span class="in">#|     (B) colored by ER status, (C) colored by PAM50 molecular subtype.</span></span>
<span id="cb3-1250"><a href="#cb3-1250" aria-hidden="true" tabindex="-1"></a><span class="in">plots_smc &lt;- sapply(</span></span>
<span id="cb3-1251"><a href="#cb3-1251" aria-hidden="true" tabindex="-1"></a><span class="in">    c("cohort", "er_status", "pam50"), </span></span>
<span id="cb3-1252"><a href="#cb3-1252" aria-hidden="true" tabindex="-1"></a><span class="in">    get_plot_new_samples,</span></span>
<span id="cb3-1253"><a href="#cb3-1253" aria-hidden="true" tabindex="-1"></a><span class="in">    name_cohort = "smc",</span></span>
<span id="cb3-1254"><a href="#cb3-1254" aria-hidden="true" tabindex="-1"></a><span class="in">    df_pca = smc_df_pca %&gt;% dplyr::filter(cohort != "scanb"),</span></span>
<span id="cb3-1255"><a href="#cb3-1255" aria-hidden="true" tabindex="-1"></a><span class="in">    x = "PC3",</span></span>
<span id="cb3-1256"><a href="#cb3-1256" aria-hidden="true" tabindex="-1"></a><span class="in">    y = "PC4",</span></span>
<span id="cb3-1257"><a href="#cb3-1257" aria-hidden="true" tabindex="-1"></a><span class="in">    USE.NAMES = TRUE, </span></span>
<span id="cb3-1258"><a href="#cb3-1258" aria-hidden="true" tabindex="-1"></a><span class="in">    simplify = FALSE</span></span>
<span id="cb3-1259"><a href="#cb3-1259" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-1260"><a href="#cb3-1260" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1261"><a href="#cb3-1261" aria-hidden="true" tabindex="-1"></a><span class="in">plots_smc$pam50 &lt;- plots_smc$pam50 +</span></span>
<span id="cb3-1262"><a href="#cb3-1262" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::scale_color_manual(</span></span>
<span id="cb3-1263"><a href="#cb3-1263" aria-hidden="true" tabindex="-1"></a><span class="in">        values = get_colors_pam50(plots_smc$pam50$data)</span></span>
<span id="cb3-1264"><a href="#cb3-1264" aria-hidden="true" tabindex="-1"></a><span class="in">    )</span></span>
<span id="cb3-1265"><a href="#cb3-1265" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1266"><a href="#cb3-1266" aria-hidden="true" tabindex="-1"></a><span class="in">cowplot::plot_grid(plotlist = plots_smc)</span></span>
<span id="cb3-1267"><a href="#cb3-1267" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-1268"><a href="#cb3-1268" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1269"><a href="#cb3-1269" aria-hidden="true" tabindex="-1"></a><span class="fu">## PDX </span></span>
<span id="cb3-1270"><a href="#cb3-1270" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1271"><a href="#cb3-1271" aria-hidden="true" tabindex="-1"></a>Another way to validate the results is to use patient derived xenografts.</span>
<span id="cb3-1272"><a href="#cb3-1272" aria-hidden="true" tabindex="-1"></a>In the Brisken lab the MIND model is used <span class="co">[</span><span class="ot">@Sflomos2016; @Scabia2022</span><span class="co">]</span>. </span>
<span id="cb3-1273"><a href="#cb3-1273" aria-hidden="true" tabindex="-1"></a>ER+ breast cancer</span>
<span id="cb3-1274"><a href="#cb3-1274" aria-hidden="true" tabindex="-1"></a>samples coming from patients are processed in a specific way and</span>
<span id="cb3-1275"><a href="#cb3-1275" aria-hidden="true" tabindex="-1"></a>injected into the mammary glands of mice. Based on that they can</span>
<span id="cb3-1276"><a href="#cb3-1276" aria-hidden="true" tabindex="-1"></a>stablish themselves and proliferate. The environment should recapitulate</span>
<span id="cb3-1277"><a href="#cb3-1277" aria-hidden="true" tabindex="-1"></a>what is seen in women as well, in the sense that estrogen (E2) levels </span>
<span id="cb3-1278"><a href="#cb3-1278" aria-hidden="true" tabindex="-1"></a>are similar to those of post-menopausal women. Since these cells have</span>
<span id="cb3-1279"><a href="#cb3-1279" aria-hidden="true" tabindex="-1"></a>to proliferate in order to establish and grow in the ducts, we expect</span>
<span id="cb3-1280"><a href="#cb3-1280" aria-hidden="true" tabindex="-1"></a>them to be more proliferative in terms of RNA-seq components and also</span>
<span id="cb3-1281"><a href="#cb3-1281" aria-hidden="true" tabindex="-1"></a>to lie in the luminal B region of the molecular landscape. We validate</span>
<span id="cb3-1282"><a href="#cb3-1282" aria-hidden="true" tabindex="-1"></a>this now. </span>
<span id="cb3-1283"><a href="#cb3-1283" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1284"><a href="#cb3-1284" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval=first_run}</span></span>
<span id="cb3-1285"><a href="#cb3-1285" aria-hidden="true" tabindex="-1"></a><span class="in">path_data &lt;- "../../data/2019_pdx_brisken"</span></span>
<span id="cb3-1286"><a href="#cb3-1286" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1287"><a href="#cb3-1287" aria-hidden="true" tabindex="-1"></a><span class="in">tx2gene &lt;- readr::read_table(</span></span>
<span id="cb3-1288"><a href="#cb3-1288" aria-hidden="true" tabindex="-1"></a><span class="in">    file.path(path_data, "transcripts_to_genes.txt"), </span></span>
<span id="cb3-1289"><a href="#cb3-1289" aria-hidden="true" tabindex="-1"></a><span class="in">    col_names = c("transcript", "gene", "symbol")</span></span>
<span id="cb3-1290"><a href="#cb3-1290" aria-hidden="true" tabindex="-1"></a><span class="in">)[, c(1, 3)]</span></span>
<span id="cb3-1291"><a href="#cb3-1291" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1292"><a href="#cb3-1292" aria-hidden="true" tabindex="-1"></a><span class="in">files &lt;- list.files(file.path(path_data, "abundance"), full.names = TRUE)</span></span>
<span id="cb3-1293"><a href="#cb3-1293" aria-hidden="true" tabindex="-1"></a><span class="in">files &lt;- file.path(files, "abundance.h5")</span></span>
<span id="cb3-1294"><a href="#cb3-1294" aria-hidden="true" tabindex="-1"></a><span class="in">names(files) &lt;- list.files(file.path(path_data, "abundance"))</span></span>
<span id="cb3-1295"><a href="#cb3-1295" aria-hidden="true" tabindex="-1"></a><span class="in">txi &lt;- tximport::tximport(</span></span>
<span id="cb3-1296"><a href="#cb3-1296" aria-hidden="true" tabindex="-1"></a><span class="in">    files, </span></span>
<span id="cb3-1297"><a href="#cb3-1297" aria-hidden="true" tabindex="-1"></a><span class="in">    type = "kallisto", </span></span>
<span id="cb3-1298"><a href="#cb3-1298" aria-hidden="true" tabindex="-1"></a><span class="in">    tx2gene = tx2gene, </span></span>
<span id="cb3-1299"><a href="#cb3-1299" aria-hidden="true" tabindex="-1"></a><span class="in">    countsFromAbundance = "lengthScaledTPM"</span></span>
<span id="cb3-1300"><a href="#cb3-1300" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-1301"><a href="#cb3-1301" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1302"><a href="#cb3-1302" aria-hidden="true" tabindex="-1"></a><span class="in">pdx_tpm &lt;- txi$abundance</span></span>
<span id="cb3-1303"><a href="#cb3-1303" aria-hidden="true" tabindex="-1"></a><span class="in">log2_pdx_tpm &lt;- log2(pdx_tpm + 1)</span></span>
<span id="cb3-1304"><a href="#cb3-1304" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1305"><a href="#cb3-1305" aria-hidden="true" tabindex="-1"></a><span class="in"># filter the samples to remove lowly expressed genes</span></span>
<span id="cb3-1306"><a href="#cb3-1306" aria-hidden="true" tabindex="-1"></a><span class="in">highly_expressed &lt;- rowSums(log2_pdx_tpm &gt; 1) &gt; 0.3</span></span>
<span id="cb3-1307"><a href="#cb3-1307" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1308"><a href="#cb3-1308" aria-hidden="true" tabindex="-1"></a><span class="in"># load the metadata so we can create later the summarized</span></span>
<span id="cb3-1309"><a href="#cb3-1309" aria-hidden="true" tabindex="-1"></a><span class="in"># experiment object for the downstream analysis</span></span>
<span id="cb3-1310"><a href="#cb3-1310" aria-hidden="true" tabindex="-1"></a><span class="in">metadata &lt;- readr::read_csv(file.path(path_data, "metadata.csv")) %&gt;%</span></span>
<span id="cb3-1311"><a href="#cb3-1311" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::mutate(</span></span>
<span id="cb3-1312"><a href="#cb3-1312" aria-hidden="true" tabindex="-1"></a><span class="in">        pdx = sapply(</span></span>
<span id="cb3-1313"><a href="#cb3-1313" aria-hidden="true" tabindex="-1"></a><span class="in">            sample_name,</span></span>
<span id="cb3-1314"><a href="#cb3-1314" aria-hidden="true" tabindex="-1"></a><span class="in">            function(x){</span></span>
<span id="cb3-1315"><a href="#cb3-1315" aria-hidden="true" tabindex="-1"></a><span class="in">                stringr::str_split(x, "_", simplify = TRUE)[1] %&gt;%</span></span>
<span id="cb3-1316"><a href="#cb3-1316" aria-hidden="true" tabindex="-1"></a><span class="in">                as.character</span></span>
<span id="cb3-1317"><a href="#cb3-1317" aria-hidden="true" tabindex="-1"></a><span class="in">            }</span></span>
<span id="cb3-1318"><a href="#cb3-1318" aria-hidden="true" tabindex="-1"></a><span class="in">        ),</span></span>
<span id="cb3-1319"><a href="#cb3-1319" aria-hidden="true" tabindex="-1"></a><span class="in">        treatment = ifelse(treatment == "P20", "P4", treatment)</span></span>
<span id="cb3-1320"><a href="#cb3-1320" aria-hidden="true" tabindex="-1"></a><span class="in">    )</span></span>
<span id="cb3-1321"><a href="#cb3-1321" aria-hidden="true" tabindex="-1"></a><span class="in">rownames(metadata) &lt;- metadata$sample_name</span></span>
<span id="cb3-1322"><a href="#cb3-1322" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1323"><a href="#cb3-1323" aria-hidden="true" tabindex="-1"></a><span class="in">pdx_tpm &lt;- SummarizedExperiment::SummarizedExperiment(</span></span>
<span id="cb3-1324"><a href="#cb3-1324" aria-hidden="true" tabindex="-1"></a><span class="in">    assays = list(</span></span>
<span id="cb3-1325"><a href="#cb3-1325" aria-hidden="true" tabindex="-1"></a><span class="in">            tpm = pdx_tpm[highly_expressed, ],</span></span>
<span id="cb3-1326"><a href="#cb3-1326" aria-hidden="true" tabindex="-1"></a><span class="in">            log2tpm = log2_pdx_tpm[highly_expressed, ]</span></span>
<span id="cb3-1327"><a href="#cb3-1327" aria-hidden="true" tabindex="-1"></a><span class="in">        ),</span></span>
<span id="cb3-1328"><a href="#cb3-1328" aria-hidden="true" tabindex="-1"></a><span class="in">    colData = metadata[colnames(pdx_tpm), ]</span></span>
<span id="cb3-1329"><a href="#cb3-1329" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-1330"><a href="#cb3-1330" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1331"><a href="#cb3-1331" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1332"><a href="#cb3-1332" aria-hidden="true" tabindex="-1"></a><span class="in">p4_pathways &lt;- msigdbr::msigdbr() %&gt;%</span></span>
<span id="cb3-1333"><a href="#cb3-1333" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::filter(gs_name %in% c(</span></span>
<span id="cb3-1334"><a href="#cb3-1334" aria-hidden="true" tabindex="-1"></a><span class="in">        "WILCOX_RESPONSE_TO_PROGESTERONE_UP",</span></span>
<span id="cb3-1335"><a href="#cb3-1335" aria-hidden="true" tabindex="-1"></a><span class="in">        "GOBP_RESPONSE_TO_PROGESTERONE"</span></span>
<span id="cb3-1336"><a href="#cb3-1336" aria-hidden="true" tabindex="-1"></a><span class="in">    ))</span></span>
<span id="cb3-1337"><a href="#cb3-1337" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1338"><a href="#cb3-1338" aria-hidden="true" tabindex="-1"></a><span class="in">gene_sets_prog &lt;- dplyr::bind_rows(</span></span>
<span id="cb3-1339"><a href="#cb3-1339" aria-hidden="true" tabindex="-1"></a><span class="in">    gene_sets, </span></span>
<span id="cb3-1340"><a href="#cb3-1340" aria-hidden="true" tabindex="-1"></a><span class="in">    p4_pathways</span></span>
<span id="cb3-1341"><a href="#cb3-1341" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-1342"><a href="#cb3-1342" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1343"><a href="#cb3-1343" aria-hidden="true" tabindex="-1"></a><span class="in">gene_sets_ &lt;- sapply(</span></span>
<span id="cb3-1344"><a href="#cb3-1344" aria-hidden="true" tabindex="-1"></a><span class="in">    gene_sets_prog$gs_name %&gt;% unique,</span></span>
<span id="cb3-1345"><a href="#cb3-1345" aria-hidden="true" tabindex="-1"></a><span class="in">    function(x, gene_sets) gene_sets %&gt;% dplyr::filter(gs_name == x) %&gt;% </span></span>
<span id="cb3-1346"><a href="#cb3-1346" aria-hidden="true" tabindex="-1"></a><span class="in">        dplyr::pull(gene_symbol),</span></span>
<span id="cb3-1347"><a href="#cb3-1347" aria-hidden="true" tabindex="-1"></a><span class="in">    gene_sets = gene_sets_prog,</span></span>
<span id="cb3-1348"><a href="#cb3-1348" aria-hidden="true" tabindex="-1"></a><span class="in">    USE.NAMES = TRUE,</span></span>
<span id="cb3-1349"><a href="#cb3-1349" aria-hidden="true" tabindex="-1"></a><span class="in">    simplify = FALSE</span></span>
<span id="cb3-1350"><a href="#cb3-1350" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-1351"><a href="#cb3-1351" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1352"><a href="#cb3-1352" aria-hidden="true" tabindex="-1"></a><span class="in">gsva_pdx &lt;- GSVA::gsva(</span></span>
<span id="cb3-1353"><a href="#cb3-1353" aria-hidden="true" tabindex="-1"></a><span class="in">    assay(pdx_tpm, "log2tpm"),</span></span>
<span id="cb3-1354"><a href="#cb3-1354" aria-hidden="true" tabindex="-1"></a><span class="in">    gset.idx.list = gene_sets_,</span></span>
<span id="cb3-1355"><a href="#cb3-1355" aria-hidden="true" tabindex="-1"></a><span class="in">    parallel.sz = 10,</span></span>
<span id="cb3-1356"><a href="#cb3-1356" aria-hidden="true" tabindex="-1"></a><span class="in">    verbose = FALSE</span></span>
<span id="cb3-1357"><a href="#cb3-1357" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-1358"><a href="#cb3-1358" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1359"><a href="#cb3-1359" aria-hidden="true" tabindex="-1"></a><span class="in">colData(pdx_tpm)[, rownames(gsva_pdx)] &lt;- t(gsva_pdx)</span></span>
<span id="cb3-1360"><a href="#cb3-1360" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1361"><a href="#cb3-1361" aria-hidden="true" tabindex="-1"></a><span class="in">pdx_normalized &lt;- get_final_ranking_values(</span></span>
<span id="cb3-1362"><a href="#cb3-1362" aria-hidden="true" tabindex="-1"></a><span class="in">    sum_exp = pdx_tpm,</span></span>
<span id="cb3-1363"><a href="#cb3-1363" aria-hidden="true" tabindex="-1"></a><span class="in">    assay_to_use = "log2tpm",</span></span>
<span id="cb3-1364"><a href="#cb3-1364" aria-hidden="true" tabindex="-1"></a><span class="in">    stable_genes = stable_genes,</span></span>
<span id="cb3-1365"><a href="#cb3-1365" aria-hidden="true" tabindex="-1"></a><span class="in">    most_variable_genes = setdiff(rownames(pca_fit$loadings), stable_genes)</span></span>
<span id="cb3-1366"><a href="#cb3-1366" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-1367"><a href="#cb3-1367" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1368"><a href="#cb3-1368" aria-hidden="true" tabindex="-1"></a><span class="in"># calculate the embedding</span></span>
<span id="cb3-1369"><a href="#cb3-1369" aria-hidden="true" tabindex="-1"></a><span class="in">pdx_df_pca &lt;- get_pca_coordinates(pdx_normalized, pca_fit) %&gt;%</span></span>
<span id="cb3-1370"><a href="#cb3-1370" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::bind_cols(</span></span>
<span id="cb3-1371"><a href="#cb3-1371" aria-hidden="true" tabindex="-1"></a><span class="in">        ., </span></span>
<span id="cb3-1372"><a href="#cb3-1372" aria-hidden="true" tabindex="-1"></a><span class="in">        colData(pdx_tpm) %&gt;% </span></span>
<span id="cb3-1373"><a href="#cb3-1373" aria-hidden="true" tabindex="-1"></a><span class="in">            data.frame %&gt;% </span></span>
<span id="cb3-1374"><a href="#cb3-1374" aria-hidden="true" tabindex="-1"></a><span class="in">            dplyr::mutate(cohort = "pdx")</span></span>
<span id="cb3-1375"><a href="#cb3-1375" aria-hidden="true" tabindex="-1"></a><span class="in">    ) %&gt;% </span></span>
<span id="cb3-1376"><a href="#cb3-1376" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::bind_rows(., df_pca_coordinates)</span></span>
<span id="cb3-1377"><a href="#cb3-1377" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1378"><a href="#cb3-1378" aria-hidden="true" tabindex="-1"></a><span class="in"># save all the results</span></span>
<span id="cb3-1379"><a href="#cb3-1379" aria-hidden="true" tabindex="-1"></a><span class="in">saveRDS(</span></span>
<span id="cb3-1380"><a href="#cb3-1380" aria-hidden="true" tabindex="-1"></a><span class="in">    pdx_tpm,</span></span>
<span id="cb3-1381"><a href="#cb3-1381" aria-hidden="true" tabindex="-1"></a><span class="in">    "../../results/rds_files/validation/pdx_tpm.rds"</span></span>
<span id="cb3-1382"><a href="#cb3-1382" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-1383"><a href="#cb3-1383" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1384"><a href="#cb3-1384" aria-hidden="true" tabindex="-1"></a><span class="in">saveRDS(</span></span>
<span id="cb3-1385"><a href="#cb3-1385" aria-hidden="true" tabindex="-1"></a><span class="in">    pdx_df_pca %&gt;% data.frame,</span></span>
<span id="cb3-1386"><a href="#cb3-1386" aria-hidden="true" tabindex="-1"></a><span class="in">    "../../results/rds_files/validation/pdx_df_pca.rds"</span></span>
<span id="cb3-1387"><a href="#cb3-1387" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-1388"><a href="#cb3-1388" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1389"><a href="#cb3-1389" aria-hidden="true" tabindex="-1"></a><span class="in">saveRDS(</span></span>
<span id="cb3-1390"><a href="#cb3-1390" aria-hidden="true" tabindex="-1"></a><span class="in">    pdx_normalized,</span></span>
<span id="cb3-1391"><a href="#cb3-1391" aria-hidden="true" tabindex="-1"></a><span class="in">    "../../results/rds_files/validation/pdx_normalized.rds"</span></span>
<span id="cb3-1392"><a href="#cb3-1392" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-1393"><a href="#cb3-1393" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-1394"><a href="#cb3-1394" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1395"><a href="#cb3-1395" aria-hidden="true" tabindex="-1"></a>When checking the total number of genes missing, there are only </span>
<span id="cb3-1396"><a href="#cb3-1396" aria-hidden="true" tabindex="-1"></a><span class="in">`r nrow(datasets_normalized$tcga) - nrow(pdx_normalized)`</span> genes missing. </span>
<span id="cb3-1397"><a href="#cb3-1397" aria-hidden="true" tabindex="-1"></a>And as seen before, this number is almost irrelevant, in the sense that</span>
<span id="cb3-1398"><a href="#cb3-1398" aria-hidden="true" tabindex="-1"></a>the embedding will still be robust and locations will be precise enough.</span>
<span id="cb3-1399"><a href="#cb3-1399" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1402"><a href="#cb3-1402" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-1403"><a href="#cb3-1403" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-pdx-cohort</span></span>
<span id="cb3-1404"><a href="#cb3-1404" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Biplot of first two PCA components from PDX, TCGA, SCANB and </span></span>
<span id="cb3-1405"><a href="#cb3-1405" aria-hidden="true" tabindex="-1"></a><span class="co">#|     METABRIC. PDX is highlighted in the plot and has bigger dot size.</span></span>
<span id="cb3-1406"><a href="#cb3-1406" aria-hidden="true" tabindex="-1"></a>pdx_df_pca <span class="sc">%&gt;%</span> </span>
<span id="cb3-1407"><a href="#cb3-1407" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(</span>
<span id="cb3-1408"><a href="#cb3-1408" aria-hidden="true" tabindex="-1"></a>        <span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> cohort)</span>
<span id="cb3-1409"><a href="#cb3-1409" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb3-1410"><a href="#cb3-1410" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">alpha =</span> cohort, <span class="at">size =</span> cohort)) <span class="sc">+</span></span>
<span id="cb3-1411"><a href="#cb3-1411" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_size_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(</span>
<span id="cb3-1412"><a href="#cb3-1412" aria-hidden="true" tabindex="-1"></a>        <span class="st">"pdx"</span> <span class="ot">=</span> <span class="dv">2</span>, </span>
<span id="cb3-1413"><a href="#cb3-1413" aria-hidden="true" tabindex="-1"></a>        <span class="st">"tcga"</span> <span class="ot">=</span> <span class="dv">1</span>,</span>
<span id="cb3-1414"><a href="#cb3-1414" aria-hidden="true" tabindex="-1"></a>        <span class="st">"scanb"</span> <span class="ot">=</span> <span class="dv">1</span>,</span>
<span id="cb3-1415"><a href="#cb3-1415" aria-hidden="true" tabindex="-1"></a>        <span class="st">"metabric"</span> <span class="ot">=</span> <span class="dv">1</span></span>
<span id="cb3-1416"><a href="#cb3-1416" aria-hidden="true" tabindex="-1"></a>    ), <span class="at">guide =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb3-1417"><a href="#cb3-1417" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_alpha_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(</span>
<span id="cb3-1418"><a href="#cb3-1418" aria-hidden="true" tabindex="-1"></a>        <span class="st">"pdx"</span> <span class="ot">=</span> <span class="dv">1</span>, </span>
<span id="cb3-1419"><a href="#cb3-1419" aria-hidden="true" tabindex="-1"></a>        <span class="st">"tcga"</span> <span class="ot">=</span> <span class="fl">0.1</span>,</span>
<span id="cb3-1420"><a href="#cb3-1420" aria-hidden="true" tabindex="-1"></a>        <span class="st">"scanb"</span> <span class="ot">=</span> <span class="fl">0.1</span>,</span>
<span id="cb3-1421"><a href="#cb3-1421" aria-hidden="true" tabindex="-1"></a>        <span class="st">"metabric"</span> <span class="ot">=</span> <span class="fl">0.1</span></span>
<span id="cb3-1422"><a href="#cb3-1422" aria-hidden="true" tabindex="-1"></a>    ), <span class="at">guide =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb3-1423"><a href="#cb3-1423" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb3-1424"><a href="#cb3-1424" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">"Cohort"</span>,</span>
<span id="cb3-1425"><a href="#cb3-1425" aria-hidden="true" tabindex="-1"></a>        <span class="at">alpha =</span> <span class="st">"Cohort"</span>,</span>
<span id="cb3-1426"><a href="#cb3-1426" aria-hidden="true" tabindex="-1"></a>        <span class="at">size =</span> <span class="st">"Cohort"</span></span>
<span id="cb3-1427"><a href="#cb3-1427" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb3-1428"><a href="#cb3-1428" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">14</span>) <span class="sc">+</span></span>
<span id="cb3-1429"><a href="#cb3-1429" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span></span>
<span id="cb3-1430"><a href="#cb3-1430" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb3-1431"><a href="#cb3-1431" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-1432"><a href="#cb3-1432" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1433"><a href="#cb3-1433" aria-hidden="true" tabindex="-1"></a>As previously, @fig-pca-pdx-cohort shows the biplot of PC3 and PC4, </span>
<span id="cb3-1434"><a href="#cb3-1434" aria-hidden="true" tabindex="-1"></a>highlighting the well mixing of the cohorts. As expected the samples are </span>
<span id="cb3-1435"><a href="#cb3-1435" aria-hidden="true" tabindex="-1"></a>overall in the luminal B region, since these are tumor cells in a </span>
<span id="cb3-1436"><a href="#cb3-1436" aria-hidden="true" tabindex="-1"></a>proliferative state.</span>
<span id="cb3-1437"><a href="#cb3-1437" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1438"><a href="#cb3-1438" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=8, fig.height=6}</span></span>
<span id="cb3-1439"><a href="#cb3-1439" aria-hidden="true" tabindex="-1"></a><span class="in">#| label: fig-pca-pdx-cohort</span></span>
<span id="cb3-1440"><a href="#cb3-1440" aria-hidden="true" tabindex="-1"></a><span class="in">#| fig-cap: Embedding of all samples from TCGA and METABRIC including</span></span>
<span id="cb3-1441"><a href="#cb3-1441" aria-hidden="true" tabindex="-1"></a><span class="in">#|     the PDX samples on top.</span></span>
<span id="cb3-1442"><a href="#cb3-1442" aria-hidden="true" tabindex="-1"></a><span class="in">pdx_df_pca %&gt;%</span></span>
<span id="cb3-1443"><a href="#cb3-1443" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::ggplot(</span></span>
<span id="cb3-1444"><a href="#cb3-1444" aria-hidden="true" tabindex="-1"></a><span class="in">        aes(x = PC3, y = PC4, color = cohort)</span></span>
<span id="cb3-1445"><a href="#cb3-1445" aria-hidden="true" tabindex="-1"></a><span class="in">    ) +</span></span>
<span id="cb3-1446"><a href="#cb3-1446" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_point(aes(alpha = cohort, size = cohort)) +</span></span>
<span id="cb3-1447"><a href="#cb3-1447" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::scale_size_manual(values = c(</span></span>
<span id="cb3-1448"><a href="#cb3-1448" aria-hidden="true" tabindex="-1"></a><span class="in">        "pdx" = 3, </span></span>
<span id="cb3-1449"><a href="#cb3-1449" aria-hidden="true" tabindex="-1"></a><span class="in">        "tcga" = 1,</span></span>
<span id="cb3-1450"><a href="#cb3-1450" aria-hidden="true" tabindex="-1"></a><span class="in">        "scanb" = 1,</span></span>
<span id="cb3-1451"><a href="#cb3-1451" aria-hidden="true" tabindex="-1"></a><span class="in">        "metabric" = 1</span></span>
<span id="cb3-1452"><a href="#cb3-1452" aria-hidden="true" tabindex="-1"></a><span class="in">    ), guide = "none") +</span></span>
<span id="cb3-1453"><a href="#cb3-1453" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::scale_alpha_manual(values = c(</span></span>
<span id="cb3-1454"><a href="#cb3-1454" aria-hidden="true" tabindex="-1"></a><span class="in">        "pdx" = 1, </span></span>
<span id="cb3-1455"><a href="#cb3-1455" aria-hidden="true" tabindex="-1"></a><span class="in">        "tcga" = 0.1,</span></span>
<span id="cb3-1456"><a href="#cb3-1456" aria-hidden="true" tabindex="-1"></a><span class="in">        "scanb" = 0.1,</span></span>
<span id="cb3-1457"><a href="#cb3-1457" aria-hidden="true" tabindex="-1"></a><span class="in">        "metabric" = 0.1</span></span>
<span id="cb3-1458"><a href="#cb3-1458" aria-hidden="true" tabindex="-1"></a><span class="in">    ), guide = "none") +</span></span>
<span id="cb3-1459"><a href="#cb3-1459" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::labs(</span></span>
<span id="cb3-1460"><a href="#cb3-1460" aria-hidden="true" tabindex="-1"></a><span class="in">        color = "Cohort",</span></span>
<span id="cb3-1461"><a href="#cb3-1461" aria-hidden="true" tabindex="-1"></a><span class="in">        alpha = "Cohort",</span></span>
<span id="cb3-1462"><a href="#cb3-1462" aria-hidden="true" tabindex="-1"></a><span class="in">        size = "Cohort",</span></span>
<span id="cb3-1463"><a href="#cb3-1463" aria-hidden="true" tabindex="-1"></a><span class="in">        title = paste0(</span></span>
<span id="cb3-1464"><a href="#cb3-1464" aria-hidden="true" tabindex="-1"></a><span class="in">            "PDX cohort embedding"</span></span>
<span id="cb3-1465"><a href="#cb3-1465" aria-hidden="true" tabindex="-1"></a><span class="in">        ),</span></span>
<span id="cb3-1466"><a href="#cb3-1466" aria-hidden="true" tabindex="-1"></a><span class="in">        subtitle = "All samples from TCGA, SCANB and METABRIC are plotted"</span></span>
<span id="cb3-1467"><a href="#cb3-1467" aria-hidden="true" tabindex="-1"></a><span class="in">    ) + </span></span>
<span id="cb3-1468"><a href="#cb3-1468" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme_bw(base_size = 18) +</span></span>
<span id="cb3-1469"><a href="#cb3-1469" aria-hidden="true" tabindex="-1"></a><span class="in">    change_plot_aes_point() +</span></span>
<span id="cb3-1470"><a href="#cb3-1470" aria-hidden="true" tabindex="-1"></a><span class="in">    change_guides_point()</span></span>
<span id="cb3-1471"><a href="#cb3-1471" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-1472"><a href="#cb3-1472" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1473"><a href="#cb3-1473" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1474"><a href="#cb3-1474" aria-hidden="true" tabindex="-1"></a><span class="in">```{r fig.width=9, fig.height=6}</span></span>
<span id="cb3-1475"><a href="#cb3-1475" aria-hidden="true" tabindex="-1"></a><span class="in">#| label: fig-pca-pdx-control-only</span></span>
<span id="cb3-1476"><a href="#cb3-1476" aria-hidden="true" tabindex="-1"></a><span class="in">#| fig-cap: Embedding of only PDX control samples on top of the </span></span>
<span id="cb3-1477"><a href="#cb3-1477" aria-hidden="true" tabindex="-1"></a><span class="in">#|     METABRIC, SCANB and TCGA samples</span></span>
<span id="cb3-1478"><a href="#cb3-1478" aria-hidden="true" tabindex="-1"></a><span class="in">p &lt;- get_base_plot(</span></span>
<span id="cb3-1479"><a href="#cb3-1479" aria-hidden="true" tabindex="-1"></a><span class="in">    df_pca_coordinates %&gt;%</span></span>
<span id="cb3-1480"><a href="#cb3-1480" aria-hidden="true" tabindex="-1"></a><span class="in">        dplyr::filter(pam50 != "claudin-low")</span></span>
<span id="cb3-1481"><a href="#cb3-1481" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-1482"><a href="#cb3-1482" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1483"><a href="#cb3-1483" aria-hidden="true" tabindex="-1"></a><span class="in">pdx_df_pca_sub &lt;- pdx_df_pca %&gt;%</span></span>
<span id="cb3-1484"><a href="#cb3-1484" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::filter(cohort == "pdx" &amp; treatment == "CTRL") %&gt;%</span></span>
<span id="cb3-1485"><a href="#cb3-1485" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::mutate(pam50 = "not available") %&gt;%</span></span>
<span id="cb3-1486"><a href="#cb3-1486" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::mutate(pdx = factor(</span></span>
<span id="cb3-1487"><a href="#cb3-1487" aria-hidden="true" tabindex="-1"></a><span class="in">        pdx, </span></span>
<span id="cb3-1488"><a href="#cb3-1488" aria-hidden="true" tabindex="-1"></a><span class="in">        levels = c("METS15", "T105", "T109", "T110", "T111", "T113")</span></span>
<span id="cb3-1489"><a href="#cb3-1489" aria-hidden="true" tabindex="-1"></a><span class="in">    ))</span></span>
<span id="cb3-1490"><a href="#cb3-1490" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1491"><a href="#cb3-1491" aria-hidden="true" tabindex="-1"></a><span class="in">pdx_ctrl_plot &lt;- p +</span></span>
<span id="cb3-1492"><a href="#cb3-1492" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_point(</span></span>
<span id="cb3-1493"><a href="#cb3-1493" aria-hidden="true" tabindex="-1"></a><span class="in">        pdx_df_pca_sub,</span></span>
<span id="cb3-1494"><a href="#cb3-1494" aria-hidden="true" tabindex="-1"></a><span class="in">        mapping = aes(x = PC3, y = PC4, shape = pdx),</span></span>
<span id="cb3-1495"><a href="#cb3-1495" aria-hidden="true" tabindex="-1"></a><span class="in">        size = 5,</span></span>
<span id="cb3-1496"><a href="#cb3-1496" aria-hidden="true" tabindex="-1"></a><span class="in">        alpha = 1,</span></span>
<span id="cb3-1497"><a href="#cb3-1497" aria-hidden="true" tabindex="-1"></a><span class="in">        stroke =  1</span></span>
<span id="cb3-1498"><a href="#cb3-1498" aria-hidden="true" tabindex="-1"></a><span class="in">    ) + </span></span>
<span id="cb3-1499"><a href="#cb3-1499" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme_bw(base_size = 15) +</span></span>
<span id="cb3-1500"><a href="#cb3-1500" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::labs(</span></span>
<span id="cb3-1501"><a href="#cb3-1501" aria-hidden="true" tabindex="-1"></a><span class="in">        caption = "Grey points correspond to the PDX cohort",</span></span>
<span id="cb3-1502"><a href="#cb3-1502" aria-hidden="true" tabindex="-1"></a><span class="in">        title = "PDX cohort, only control samples",</span></span>
<span id="cb3-1503"><a href="#cb3-1503" aria-hidden="true" tabindex="-1"></a><span class="in">        color = "Molecular\nsubtype",</span></span>
<span id="cb3-1504"><a href="#cb3-1504" aria-hidden="true" tabindex="-1"></a><span class="in">        shape = "PDX"</span></span>
<span id="cb3-1505"><a href="#cb3-1505" aria-hidden="true" tabindex="-1"></a><span class="in">    ) + </span></span>
<span id="cb3-1506"><a href="#cb3-1506" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::scale_color_manual(</span></span>
<span id="cb3-1507"><a href="#cb3-1507" aria-hidden="true" tabindex="-1"></a><span class="in">        values = c(</span></span>
<span id="cb3-1508"><a href="#cb3-1508" aria-hidden="true" tabindex="-1"></a><span class="in">            get_colors_pam50(p$data),</span></span>
<span id="cb3-1509"><a href="#cb3-1509" aria-hidden="true" tabindex="-1"></a><span class="in">            "not available" = "black"</span></span>
<span id="cb3-1510"><a href="#cb3-1510" aria-hidden="true" tabindex="-1"></a><span class="in">        ),</span></span>
<span id="cb3-1511"><a href="#cb3-1511" aria-hidden="true" tabindex="-1"></a><span class="in">        labels = c("Basal-like", "HER2-enriched", </span></span>
<span id="cb3-1512"><a href="#cb3-1512" aria-hidden="true" tabindex="-1"></a><span class="in">                   "LumB", "LumA", "Normal-like", </span></span>
<span id="cb3-1513"><a href="#cb3-1513" aria-hidden="true" tabindex="-1"></a><span class="in">                   "Not available")</span></span>
<span id="cb3-1514"><a href="#cb3-1514" aria-hidden="true" tabindex="-1"></a><span class="in">    ) +</span></span>
<span id="cb3-1515"><a href="#cb3-1515" aria-hidden="true" tabindex="-1"></a><span class="in">    change_plot_aes_point() +</span></span>
<span id="cb3-1516"><a href="#cb3-1516" aria-hidden="true" tabindex="-1"></a><span class="in">    change_guides_point(shape = TRUE)</span></span>
<span id="cb3-1517"><a href="#cb3-1517" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1518"><a href="#cb3-1518" aria-hidden="true" tabindex="-1"></a><span class="in">plots_paper$pdx_ctrl_plot &lt;- pdx_ctrl_plot</span></span>
<span id="cb3-1519"><a href="#cb3-1519" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1520"><a href="#cb3-1520" aria-hidden="true" tabindex="-1"></a><span class="in">pdx_ctrl_plot</span></span>
<span id="cb3-1521"><a href="#cb3-1521" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-1522"><a href="#cb3-1522" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1525"><a href="#cb3-1525" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-1526"><a href="#cb3-1526" aria-hidden="true" tabindex="-1"></a>plots_paper<span class="sc">$</span>pdx_ctrl_plot <span class="ot">&lt;-</span> plots_paper<span class="sc">$</span>pdx_ctrl_plot <span class="sc">+</span></span>
<span id="cb3-1527"><a href="#cb3-1527" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">title =</span> <span class="cn">NULL</span>, <span class="at">caption =</span> <span class="cn">NULL</span>)</span>
<span id="cb3-1528"><a href="#cb3-1528" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-1529"><a href="#cb3-1529" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1530"><a href="#cb3-1530" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1531"><a href="#cb3-1531" aria-hidden="true" tabindex="-1"></a>And we check the PDXs response to estrogen and see how much they move</span>
<span id="cb3-1532"><a href="#cb3-1532" aria-hidden="true" tabindex="-1"></a>away from the untreated sample.</span>
<span id="cb3-1533"><a href="#cb3-1533" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1534"><a href="#cb3-1534" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=8}</span></span>
<span id="cb3-1535"><a href="#cb3-1535" aria-hidden="true" tabindex="-1"></a><span class="in">#| label: fig-pdx-p4-control-components</span></span>
<span id="cb3-1536"><a href="#cb3-1536" aria-hidden="true" tabindex="-1"></a><span class="in">#| fig-cap: Comparison of position in the molecular landscape between</span></span>
<span id="cb3-1537"><a href="#cb3-1537" aria-hidden="true" tabindex="-1"></a><span class="in">#|     CTRL and P4 treated samples</span></span>
<span id="cb3-1538"><a href="#cb3-1538" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1539"><a href="#cb3-1539" aria-hidden="true" tabindex="-1"></a><span class="in">pdxs_to_use &lt;- c("METS15", "T113", "T111") #, "T105", "T109", "T110")</span></span>
<span id="cb3-1540"><a href="#cb3-1540" aria-hidden="true" tabindex="-1"></a><span class="in">names(pdxs_to_use) &lt;- pdxs_to_use</span></span>
<span id="cb3-1541"><a href="#cb3-1541" aria-hidden="true" tabindex="-1"></a><span class="in">pdx_df_pca_sub &lt;- pdx_df_pca %&gt;%</span></span>
<span id="cb3-1542"><a href="#cb3-1542" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::filter(</span></span>
<span id="cb3-1543"><a href="#cb3-1543" aria-hidden="true" tabindex="-1"></a><span class="in">        cohort == "pdx" &amp; </span></span>
<span id="cb3-1544"><a href="#cb3-1544" aria-hidden="true" tabindex="-1"></a><span class="in">            treatment %in% c("CTRL", "P4", "P20")</span></span>
<span id="cb3-1545"><a href="#cb3-1545" aria-hidden="true" tabindex="-1"></a><span class="in">    ) %&gt;%</span></span>
<span id="cb3-1546"><a href="#cb3-1546" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::mutate(pam50 = "nc") %&gt;%</span></span>
<span id="cb3-1547"><a href="#cb3-1547" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::mutate(pdx = factor(pdx, levels = c(</span></span>
<span id="cb3-1548"><a href="#cb3-1548" aria-hidden="true" tabindex="-1"></a><span class="in">        "METS15", "T113", "T109",</span></span>
<span id="cb3-1549"><a href="#cb3-1549" aria-hidden="true" tabindex="-1"></a><span class="in">        "T105", "T111", "T110"</span></span>
<span id="cb3-1550"><a href="#cb3-1550" aria-hidden="true" tabindex="-1"></a><span class="in">    ))) %&gt;%</span></span>
<span id="cb3-1551"><a href="#cb3-1551" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::filter(pdx %in% pdxs_to_use)</span></span>
<span id="cb3-1552"><a href="#cb3-1552" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1553"><a href="#cb3-1553" aria-hidden="true" tabindex="-1"></a><span class="in">pdx_df_pca_sub %&gt;%</span></span>
<span id="cb3-1554"><a href="#cb3-1554" aria-hidden="true" tabindex="-1"></a><span class="in">    tidyr::pivot_longer(</span></span>
<span id="cb3-1555"><a href="#cb3-1555" aria-hidden="true" tabindex="-1"></a><span class="in">        cols = dplyr::all_of(c("PC3", "PC4")),</span></span>
<span id="cb3-1556"><a href="#cb3-1556" aria-hidden="true" tabindex="-1"></a><span class="in">        names_to = "pc",</span></span>
<span id="cb3-1557"><a href="#cb3-1557" aria-hidden="true" tabindex="-1"></a><span class="in">        values_to = "components"</span></span>
<span id="cb3-1558"><a href="#cb3-1558" aria-hidden="true" tabindex="-1"></a><span class="in">    ) %&gt;%</span></span>
<span id="cb3-1559"><a href="#cb3-1559" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::ggplot(aes(x = treatment, y = components)) +</span></span>
<span id="cb3-1560"><a href="#cb3-1560" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_boxplot(outlier.shape = NA) + </span></span>
<span id="cb3-1561"><a href="#cb3-1561" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_jitter(</span></span>
<span id="cb3-1562"><a href="#cb3-1562" aria-hidden="true" tabindex="-1"></a><span class="in">        size = 3, </span></span>
<span id="cb3-1563"><a href="#cb3-1563" aria-hidden="true" tabindex="-1"></a><span class="in">        alpha = 0.9, </span></span>
<span id="cb3-1564"><a href="#cb3-1564" aria-hidden="true" tabindex="-1"></a><span class="in">        mapping = aes(color = treatment)</span></span>
<span id="cb3-1565"><a href="#cb3-1565" aria-hidden="true" tabindex="-1"></a><span class="in">    ) +</span></span>
<span id="cb3-1566"><a href="#cb3-1566" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::scale_color_viridis_d(option = "H") +</span></span>
<span id="cb3-1567"><a href="#cb3-1567" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::facet_wrap(</span></span>
<span id="cb3-1568"><a href="#cb3-1568" aria-hidden="true" tabindex="-1"></a><span class="in">        ~ pc + pdx, </span></span>
<span id="cb3-1569"><a href="#cb3-1569" aria-hidden="true" tabindex="-1"></a><span class="in">        nrow = 2, </span></span>
<span id="cb3-1570"><a href="#cb3-1570" aria-hidden="true" tabindex="-1"></a><span class="in">        scales = "free_y"</span></span>
<span id="cb3-1571"><a href="#cb3-1571" aria-hidden="true" tabindex="-1"></a><span class="in">    ) +</span></span>
<span id="cb3-1572"><a href="#cb3-1572" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::labs(</span></span>
<span id="cb3-1573"><a href="#cb3-1573" aria-hidden="true" tabindex="-1"></a><span class="in">        y = "Component", </span></span>
<span id="cb3-1574"><a href="#cb3-1574" aria-hidden="true" tabindex="-1"></a><span class="in">        x = "Treatment"</span></span>
<span id="cb3-1575"><a href="#cb3-1575" aria-hidden="true" tabindex="-1"></a><span class="in">    ) +</span></span>
<span id="cb3-1576"><a href="#cb3-1576" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme_bw(base_size = 20) +</span></span>
<span id="cb3-1577"><a href="#cb3-1577" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme(legend.position = "none") +</span></span>
<span id="cb3-1578"><a href="#cb3-1578" aria-hidden="true" tabindex="-1"></a><span class="in">    change_guides_point() +</span></span>
<span id="cb3-1579"><a href="#cb3-1579" aria-hidden="true" tabindex="-1"></a><span class="in">    change_plot_aes_point()</span></span>
<span id="cb3-1580"><a href="#cb3-1580" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-1581"><a href="#cb3-1581" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1582"><a href="#cb3-1582" aria-hidden="true" tabindex="-1"></a>It looks like some of the PDXs had a bigger shift towards the left</span>
<span id="cb3-1583"><a href="#cb3-1583" aria-hidden="true" tabindex="-1"></a>when treated with estrogen or at least some stabilization. On the</span>
<span id="cb3-1584"><a href="#cb3-1584" aria-hidden="true" tabindex="-1"></a>other hand, another set of PDXs, the last three, they didn't see</span>
<span id="cb3-1585"><a href="#cb3-1585" aria-hidden="true" tabindex="-1"></a>much change. </span>
<span id="cb3-1586"><a href="#cb3-1586" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1587"><a href="#cb3-1587" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=16, fig.height=8}</span></span>
<span id="cb3-1588"><a href="#cb3-1588" aria-hidden="true" tabindex="-1"></a><span class="in">pdxs_to_use &lt;- c("METS15", "T113", "T111", "T105", "T109", "T110")</span></span>
<span id="cb3-1589"><a href="#cb3-1589" aria-hidden="true" tabindex="-1"></a><span class="in">names(pdxs_to_use) &lt;- pdxs_to_use</span></span>
<span id="cb3-1590"><a href="#cb3-1590" aria-hidden="true" tabindex="-1"></a><span class="in">pdx_df_pca_sub &lt;- pdx_df_pca %&gt;%</span></span>
<span id="cb3-1591"><a href="#cb3-1591" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::filter(</span></span>
<span id="cb3-1592"><a href="#cb3-1592" aria-hidden="true" tabindex="-1"></a><span class="in">        cohort == "pdx" &amp; </span></span>
<span id="cb3-1593"><a href="#cb3-1593" aria-hidden="true" tabindex="-1"></a><span class="in">            treatment %in% c("CTRL", "P4", "P20")</span></span>
<span id="cb3-1594"><a href="#cb3-1594" aria-hidden="true" tabindex="-1"></a><span class="in">    ) %&gt;%</span></span>
<span id="cb3-1595"><a href="#cb3-1595" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::mutate(pam50 = "nc") %&gt;%</span></span>
<span id="cb3-1596"><a href="#cb3-1596" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::mutate(pdx = factor(pdx, levels = c(</span></span>
<span id="cb3-1597"><a href="#cb3-1597" aria-hidden="true" tabindex="-1"></a><span class="in">        "METS15", "T105", "T109",</span></span>
<span id="cb3-1598"><a href="#cb3-1598" aria-hidden="true" tabindex="-1"></a><span class="in">        "T110", "T111", "T113"</span></span>
<span id="cb3-1599"><a href="#cb3-1599" aria-hidden="true" tabindex="-1"></a><span class="in">    ))) %&gt;%</span></span>
<span id="cb3-1600"><a href="#cb3-1600" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::filter(pdx %in% pdxs_to_use)</span></span>
<span id="cb3-1601"><a href="#cb3-1601" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1602"><a href="#cb3-1602" aria-hidden="true" tabindex="-1"></a><span class="in">pdx_df_pca_sub %&gt;%</span></span>
<span id="cb3-1603"><a href="#cb3-1603" aria-hidden="true" tabindex="-1"></a><span class="in">    tidyr::pivot_longer(</span></span>
<span id="cb3-1604"><a href="#cb3-1604" aria-hidden="true" tabindex="-1"></a><span class="in">        cols = dplyr::all_of(c("PC3", "PC4")),</span></span>
<span id="cb3-1605"><a href="#cb3-1605" aria-hidden="true" tabindex="-1"></a><span class="in">        names_to = "pc",</span></span>
<span id="cb3-1606"><a href="#cb3-1606" aria-hidden="true" tabindex="-1"></a><span class="in">        values_to = "components"</span></span>
<span id="cb3-1607"><a href="#cb3-1607" aria-hidden="true" tabindex="-1"></a><span class="in">    ) %&gt;%</span></span>
<span id="cb3-1608"><a href="#cb3-1608" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::ggplot(aes(x = treatment, y = components)) +</span></span>
<span id="cb3-1609"><a href="#cb3-1609" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_boxplot(outlier.shape = NA) + </span></span>
<span id="cb3-1610"><a href="#cb3-1610" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_jitter(</span></span>
<span id="cb3-1611"><a href="#cb3-1611" aria-hidden="true" tabindex="-1"></a><span class="in">        size = 3, </span></span>
<span id="cb3-1612"><a href="#cb3-1612" aria-hidden="true" tabindex="-1"></a><span class="in">        alpha = 0.9, </span></span>
<span id="cb3-1613"><a href="#cb3-1613" aria-hidden="true" tabindex="-1"></a><span class="in">        mapping = aes(color = treatment)</span></span>
<span id="cb3-1614"><a href="#cb3-1614" aria-hidden="true" tabindex="-1"></a><span class="in">    ) +</span></span>
<span id="cb3-1615"><a href="#cb3-1615" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::scale_color_viridis_d(option = "H") +</span></span>
<span id="cb3-1616"><a href="#cb3-1616" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::facet_wrap(</span></span>
<span id="cb3-1617"><a href="#cb3-1617" aria-hidden="true" tabindex="-1"></a><span class="in">        ~ pc + pdx, </span></span>
<span id="cb3-1618"><a href="#cb3-1618" aria-hidden="true" tabindex="-1"></a><span class="in">        nrow = 2, </span></span>
<span id="cb3-1619"><a href="#cb3-1619" aria-hidden="true" tabindex="-1"></a><span class="in">        scales = "free_y"</span></span>
<span id="cb3-1620"><a href="#cb3-1620" aria-hidden="true" tabindex="-1"></a><span class="in">    ) +</span></span>
<span id="cb3-1621"><a href="#cb3-1621" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::labs(</span></span>
<span id="cb3-1622"><a href="#cb3-1622" aria-hidden="true" tabindex="-1"></a><span class="in">        y = "Component", </span></span>
<span id="cb3-1623"><a href="#cb3-1623" aria-hidden="true" tabindex="-1"></a><span class="in">        x = "Treatment"</span></span>
<span id="cb3-1624"><a href="#cb3-1624" aria-hidden="true" tabindex="-1"></a><span class="in">    ) +</span></span>
<span id="cb3-1625"><a href="#cb3-1625" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme_bw(base_size = 20) +</span></span>
<span id="cb3-1626"><a href="#cb3-1626" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme(legend.position = "none") +</span></span>
<span id="cb3-1627"><a href="#cb3-1627" aria-hidden="true" tabindex="-1"></a><span class="in">    change_guides_point() +</span></span>
<span id="cb3-1628"><a href="#cb3-1628" aria-hidden="true" tabindex="-1"></a><span class="in">    change_plot_aes_point()</span></span>
<span id="cb3-1629"><a href="#cb3-1629" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-1630"><a href="#cb3-1630" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1631"><a href="#cb3-1631" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1632"><a href="#cb3-1632" aria-hidden="true" tabindex="-1"></a>We now include as well the embedding for the P4 treated samples</span>
<span id="cb3-1633"><a href="#cb3-1633" aria-hidden="true" tabindex="-1"></a>as well. </span>
<span id="cb3-1634"><a href="#cb3-1634" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1635"><a href="#cb3-1635" aria-hidden="true" tabindex="-1"></a><span class="in">```{r fig.width=12, fig.height=9}</span></span>
<span id="cb3-1636"><a href="#cb3-1636" aria-hidden="true" tabindex="-1"></a><span class="in">#| label: fig-pca-pdx-only-embedding</span></span>
<span id="cb3-1637"><a href="#cb3-1637" aria-hidden="true" tabindex="-1"></a><span class="in">#| fig-cap: Embedding of only PDX control and P4 samples </span></span>
<span id="cb3-1638"><a href="#cb3-1638" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1639"><a href="#cb3-1639" aria-hidden="true" tabindex="-1"></a><span class="in"># we first do by PDX individually the positions so we can combine </span></span>
<span id="cb3-1640"><a href="#cb3-1640" aria-hidden="true" tabindex="-1"></a><span class="in"># them into a single plot and include in the figure instead of the </span></span>
<span id="cb3-1641"><a href="#cb3-1641" aria-hidden="true" tabindex="-1"></a><span class="in"># boxplots for PC3 and PC4 components</span></span>
<span id="cb3-1642"><a href="#cb3-1642" aria-hidden="true" tabindex="-1"></a><span class="in">pdx_ctrl_plot_build &lt;- ggplot2::ggplot_build(pdx_ctrl_plot)</span></span>
<span id="cb3-1643"><a href="#cb3-1643" aria-hidden="true" tabindex="-1"></a><span class="in">shape_pdx &lt;- pdx_ctrl_plot_build$data[[2]]$shape %&gt;% unique</span></span>
<span id="cb3-1644"><a href="#cb3-1644" aria-hidden="true" tabindex="-1"></a><span class="in">names(shape_pdx) &lt;-  c("METS15", "T105", "T109", "T110", "T111", "T113") %&gt;%</span></span>
<span id="cb3-1645"><a href="#cb3-1645" aria-hidden="true" tabindex="-1"></a><span class="in">    sort</span></span>
<span id="cb3-1646"><a href="#cb3-1646" aria-hidden="true" tabindex="-1"></a><span class="in">pdx_only_embedding &lt;- function(</span></span>
<span id="cb3-1647"><a href="#cb3-1647" aria-hidden="true" tabindex="-1"></a><span class="in">    pdx_name, </span></span>
<span id="cb3-1648"><a href="#cb3-1648" aria-hidden="true" tabindex="-1"></a><span class="in">    pdx_df_pca, </span></span>
<span id="cb3-1649"><a href="#cb3-1649" aria-hidden="true" tabindex="-1"></a><span class="in">    treatments_pdx,</span></span>
<span id="cb3-1650"><a href="#cb3-1650" aria-hidden="true" tabindex="-1"></a><span class="in">    shape_pdx</span></span>
<span id="cb3-1651"><a href="#cb3-1651" aria-hidden="true" tabindex="-1"></a><span class="in">){</span></span>
<span id="cb3-1652"><a href="#cb3-1652" aria-hidden="true" tabindex="-1"></a><span class="in">    </span></span>
<span id="cb3-1653"><a href="#cb3-1653" aria-hidden="true" tabindex="-1"></a><span class="in">    pdx_df_pca_sub &lt;- pdx_df_pca %&gt;%</span></span>
<span id="cb3-1654"><a href="#cb3-1654" aria-hidden="true" tabindex="-1"></a><span class="in">        dplyr::filter(cohort == "pdx" &amp; treatment %in% treatments_pdx) %&gt;%</span></span>
<span id="cb3-1655"><a href="#cb3-1655" aria-hidden="true" tabindex="-1"></a><span class="in">        dplyr::filter(pdx %in% c(pdx_name)) %&gt;%</span></span>
<span id="cb3-1656"><a href="#cb3-1656" aria-hidden="true" tabindex="-1"></a><span class="in">        dplyr::mutate(pam50 = "nc")</span></span>
<span id="cb3-1657"><a href="#cb3-1657" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1658"><a href="#cb3-1658" aria-hidden="true" tabindex="-1"></a><span class="in">    </span></span>
<span id="cb3-1659"><a href="#cb3-1659" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::ggplot(</span></span>
<span id="cb3-1660"><a href="#cb3-1660" aria-hidden="true" tabindex="-1"></a><span class="in">        pdx_df_pca_sub,</span></span>
<span id="cb3-1661"><a href="#cb3-1661" aria-hidden="true" tabindex="-1"></a><span class="in">        mapping = aes(x = PC3, y = PC4, shape = treatment, color = treatment)</span></span>
<span id="cb3-1662"><a href="#cb3-1662" aria-hidden="true" tabindex="-1"></a><span class="in">    ) + </span></span>
<span id="cb3-1663"><a href="#cb3-1663" aria-hidden="true" tabindex="-1"></a><span class="in">        ggplot2::geom_point(</span></span>
<span id="cb3-1664"><a href="#cb3-1664" aria-hidden="true" tabindex="-1"></a><span class="in">            size = 5, </span></span>
<span id="cb3-1665"><a href="#cb3-1665" aria-hidden="true" tabindex="-1"></a><span class="in">            alpha = 0.9,</span></span>
<span id="cb3-1666"><a href="#cb3-1666" aria-hidden="true" tabindex="-1"></a><span class="in">            shape = shape_pdx[pdx_name],</span></span>
<span id="cb3-1667"><a href="#cb3-1667" aria-hidden="true" tabindex="-1"></a><span class="in">            stroke = 2.5</span></span>
<span id="cb3-1668"><a href="#cb3-1668" aria-hidden="true" tabindex="-1"></a><span class="in">        ) + </span></span>
<span id="cb3-1669"><a href="#cb3-1669" aria-hidden="true" tabindex="-1"></a><span class="in">        ggplot2::theme_bw(base_size = 15) +</span></span>
<span id="cb3-1670"><a href="#cb3-1670" aria-hidden="true" tabindex="-1"></a><span class="in">        ggplot2::labs(</span></span>
<span id="cb3-1671"><a href="#cb3-1671" aria-hidden="true" tabindex="-1"></a><span class="in">            title = pdx_name,</span></span>
<span id="cb3-1672"><a href="#cb3-1672" aria-hidden="true" tabindex="-1"></a><span class="in">            color = "Treatment"</span></span>
<span id="cb3-1673"><a href="#cb3-1673" aria-hidden="true" tabindex="-1"></a><span class="in">        ) +</span></span>
<span id="cb3-1674"><a href="#cb3-1674" aria-hidden="true" tabindex="-1"></a><span class="in">        ggplot2::scale_color_manual(</span></span>
<span id="cb3-1675"><a href="#cb3-1675" aria-hidden="true" tabindex="-1"></a><span class="in">            values = c("black", "blue")</span></span>
<span id="cb3-1676"><a href="#cb3-1676" aria-hidden="true" tabindex="-1"></a><span class="in">        ) + </span></span>
<span id="cb3-1677"><a href="#cb3-1677" aria-hidden="true" tabindex="-1"></a><span class="in">        change_plot_aes_point() +</span></span>
<span id="cb3-1678"><a href="#cb3-1678" aria-hidden="true" tabindex="-1"></a><span class="in">        change_guides_point(shape = TRUE) + </span></span>
<span id="cb3-1679"><a href="#cb3-1679" aria-hidden="true" tabindex="-1"></a><span class="in">        ggplot2::theme(legend.position = "top")</span></span>
<span id="cb3-1680"><a href="#cb3-1680" aria-hidden="true" tabindex="-1"></a><span class="in">}</span></span>
<span id="cb3-1681"><a href="#cb3-1681" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1682"><a href="#cb3-1682" aria-hidden="true" tabindex="-1"></a><span class="in">pdx_plots_embedding_p4 &lt;- sapply(</span></span>
<span id="cb3-1683"><a href="#cb3-1683" aria-hidden="true" tabindex="-1"></a><span class="in">    c("METS15", "T105", "T109", "T110", "T111", "T113"), </span></span>
<span id="cb3-1684"><a href="#cb3-1684" aria-hidden="true" tabindex="-1"></a><span class="in">    pdx_only_embedding,</span></span>
<span id="cb3-1685"><a href="#cb3-1685" aria-hidden="true" tabindex="-1"></a><span class="in">    pdx_df_pca = pdx_df_pca,</span></span>
<span id="cb3-1686"><a href="#cb3-1686" aria-hidden="true" tabindex="-1"></a><span class="in">    treatments_pdx = c("CTRL", "P4"),</span></span>
<span id="cb3-1687"><a href="#cb3-1687" aria-hidden="true" tabindex="-1"></a><span class="in">    shape_pdx = shape_pdx,</span></span>
<span id="cb3-1688"><a href="#cb3-1688" aria-hidden="true" tabindex="-1"></a><span class="in">    USE.NAMES = TRUE, </span></span>
<span id="cb3-1689"><a href="#cb3-1689" aria-hidden="true" tabindex="-1"></a><span class="in">    simplify = FALSE</span></span>
<span id="cb3-1690"><a href="#cb3-1690" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-1691"><a href="#cb3-1691" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1692"><a href="#cb3-1692" aria-hidden="true" tabindex="-1"></a><span class="in">cowplot::plot_grid(plotlist = pdx_plots_embedding_p4, ncol = 3)</span></span>
<span id="cb3-1693"><a href="#cb3-1693" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-1694"><a href="#cb3-1694" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1695"><a href="#cb3-1695" aria-hidden="true" tabindex="-1"></a>We see that T110 has a change in position as well, it goes</span>
<span id="cb3-1696"><a href="#cb3-1696" aria-hidden="true" tabindex="-1"></a>towards the center meaning that there is a decrease in </span>
<span id="cb3-1697"><a href="#cb3-1697" aria-hidden="true" tabindex="-1"></a>proliferation and also ER signaling. T105 has a very</span>
<span id="cb3-1698"><a href="#cb3-1698" aria-hidden="true" tabindex="-1"></a>strong effect, it goes up on the diagonal, meaning higher</span>
<span id="cb3-1699"><a href="#cb3-1699" aria-hidden="true" tabindex="-1"></a>proliferation and lower ER signaling.</span>
<span id="cb3-1700"><a href="#cb3-1700" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1701"><a href="#cb3-1701" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=16, fig.height=12}</span></span>
<span id="cb3-1702"><a href="#cb3-1702" aria-hidden="true" tabindex="-1"></a><span class="in">#| label: fig-pdx-p4-control-pathways</span></span>
<span id="cb3-1703"><a href="#cb3-1703" aria-hidden="true" tabindex="-1"></a><span class="in">#| fig-cap: Comparison of position in the molecular landscape between</span></span>
<span id="cb3-1704"><a href="#cb3-1704" aria-hidden="true" tabindex="-1"></a><span class="in">#|     CTRL, E2 and P4 treated samples</span></span>
<span id="cb3-1705"><a href="#cb3-1705" aria-hidden="true" tabindex="-1"></a><span class="in">pathways &lt;- c(</span></span>
<span id="cb3-1706"><a href="#cb3-1706" aria-hidden="true" tabindex="-1"></a><span class="in">    #"HALLMARK_ANDROGEN_RESPONSE" = "Androgen Response",</span></span>
<span id="cb3-1707"><a href="#cb3-1707" aria-hidden="true" tabindex="-1"></a><span class="in">    #"HALLMARK_E2F_TARGETS" = "E2F",</span></span>
<span id="cb3-1708"><a href="#cb3-1708" aria-hidden="true" tabindex="-1"></a><span class="in">    "HALLMARK_G2M_CHECKPOINT" = "G2M",</span></span>
<span id="cb3-1709"><a href="#cb3-1709" aria-hidden="true" tabindex="-1"></a><span class="in">    "HALLMARK_ESTROGEN_RESPONSE_EARLY" = "Estrogen Early",</span></span>
<span id="cb3-1710"><a href="#cb3-1710" aria-hidden="true" tabindex="-1"></a><span class="in">    #"SET_ERPR" = "SET ER/PR",</span></span>
<span id="cb3-1711"><a href="#cb3-1711" aria-hidden="true" tabindex="-1"></a><span class="in">    #"HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION" = "EMT",</span></span>
<span id="cb3-1712"><a href="#cb3-1712" aria-hidden="true" tabindex="-1"></a><span class="in">    "GOBP_RESPONSE_TO_PROGESTERONE" = "GO Progesterone"</span></span>
<span id="cb3-1713"><a href="#cb3-1713" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-1714"><a href="#cb3-1714" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1715"><a href="#cb3-1715" aria-hidden="true" tabindex="-1"></a><span class="in">pdxs_to_use &lt;- c("METS15", "T113", "T111", "T105", "T109", "T110")</span></span>
<span id="cb3-1716"><a href="#cb3-1716" aria-hidden="true" tabindex="-1"></a><span class="in">names(pdxs_to_use) &lt;- pdxs_to_use</span></span>
<span id="cb3-1717"><a href="#cb3-1717" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1718"><a href="#cb3-1718" aria-hidden="true" tabindex="-1"></a><span class="in">pdx_df_pca_sub &lt;- pdx_df_pca %&gt;%</span></span>
<span id="cb3-1719"><a href="#cb3-1719" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::filter(</span></span>
<span id="cb3-1720"><a href="#cb3-1720" aria-hidden="true" tabindex="-1"></a><span class="in">        cohort == "pdx" &amp; </span></span>
<span id="cb3-1721"><a href="#cb3-1721" aria-hidden="true" tabindex="-1"></a><span class="in">            treatment %in% c("CTRL", "P4")</span></span>
<span id="cb3-1722"><a href="#cb3-1722" aria-hidden="true" tabindex="-1"></a><span class="in">    ) %&gt;%</span></span>
<span id="cb3-1723"><a href="#cb3-1723" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::mutate(pam50 = "nc") %&gt;%</span></span>
<span id="cb3-1724"><a href="#cb3-1724" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::mutate(pdx = factor(pdx, levels = c(</span></span>
<span id="cb3-1725"><a href="#cb3-1725" aria-hidden="true" tabindex="-1"></a><span class="in">        "METS15", "T105", "T109",</span></span>
<span id="cb3-1726"><a href="#cb3-1726" aria-hidden="true" tabindex="-1"></a><span class="in">        "T110", "T111", "T113"</span></span>
<span id="cb3-1727"><a href="#cb3-1727" aria-hidden="true" tabindex="-1"></a><span class="in">    ))) %&gt;% </span></span>
<span id="cb3-1728"><a href="#cb3-1728" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::filter(pdx %in% pdxs_to_use)</span></span>
<span id="cb3-1729"><a href="#cb3-1729" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1730"><a href="#cb3-1730" aria-hidden="true" tabindex="-1"></a><span class="in">plots_pathways &lt;- mapply(</span></span>
<span id="cb3-1731"><a href="#cb3-1731" aria-hidden="true" tabindex="-1"></a><span class="in">    function(pathways, pathway_names, i){</span></span>
<span id="cb3-1732"><a href="#cb3-1732" aria-hidden="true" tabindex="-1"></a><span class="in">        names(pathways) &lt;- pathway_names </span></span>
<span id="cb3-1733"><a href="#cb3-1733" aria-hidden="true" tabindex="-1"></a><span class="in">        pdx_df_pca_sub %&gt;%</span></span>
<span id="cb3-1734"><a href="#cb3-1734" aria-hidden="true" tabindex="-1"></a><span class="in">            tidyr::pivot_longer(</span></span>
<span id="cb3-1735"><a href="#cb3-1735" aria-hidden="true" tabindex="-1"></a><span class="in">                cols = dplyr::all_of(pathway_names),</span></span>
<span id="cb3-1736"><a href="#cb3-1736" aria-hidden="true" tabindex="-1"></a><span class="in">                names_to = "pathway",</span></span>
<span id="cb3-1737"><a href="#cb3-1737" aria-hidden="true" tabindex="-1"></a><span class="in">                values_to = "score"</span></span>
<span id="cb3-1738"><a href="#cb3-1738" aria-hidden="true" tabindex="-1"></a><span class="in">            ) %&gt;%</span></span>
<span id="cb3-1739"><a href="#cb3-1739" aria-hidden="true" tabindex="-1"></a><span class="in">            dplyr::mutate(pathway = factor(pathway, levels = pathway_names)) %&gt;%</span></span>
<span id="cb3-1740"><a href="#cb3-1740" aria-hidden="true" tabindex="-1"></a><span class="in">            ggplot2::ggplot(aes(x = treatment, y = score)) +</span></span>
<span id="cb3-1741"><a href="#cb3-1741" aria-hidden="true" tabindex="-1"></a><span class="in">            ggplot2::geom_boxplot(outlier.shape = NA) + </span></span>
<span id="cb3-1742"><a href="#cb3-1742" aria-hidden="true" tabindex="-1"></a><span class="in">            ggplot2::geom_jitter(</span></span>
<span id="cb3-1743"><a href="#cb3-1743" aria-hidden="true" tabindex="-1"></a><span class="in">                size = 3, </span></span>
<span id="cb3-1744"><a href="#cb3-1744" aria-hidden="true" tabindex="-1"></a><span class="in">                alpha = 0.9, </span></span>
<span id="cb3-1745"><a href="#cb3-1745" aria-hidden="true" tabindex="-1"></a><span class="in">                mapping = aes(color = treatment)</span></span>
<span id="cb3-1746"><a href="#cb3-1746" aria-hidden="true" tabindex="-1"></a><span class="in">            ) + </span></span>
<span id="cb3-1747"><a href="#cb3-1747" aria-hidden="true" tabindex="-1"></a><span class="in">            ggplot2::scale_color_manual(values = c(</span></span>
<span id="cb3-1748"><a href="#cb3-1748" aria-hidden="true" tabindex="-1"></a><span class="in">                "black",</span></span>
<span id="cb3-1749"><a href="#cb3-1749" aria-hidden="true" tabindex="-1"></a><span class="in">                "blue"</span></span>
<span id="cb3-1750"><a href="#cb3-1750" aria-hidden="true" tabindex="-1"></a><span class="in">            )) + </span></span>
<span id="cb3-1751"><a href="#cb3-1751" aria-hidden="true" tabindex="-1"></a><span class="in">            ggplot2::facet_wrap(</span></span>
<span id="cb3-1752"><a href="#cb3-1752" aria-hidden="true" tabindex="-1"></a><span class="in">                ~ pdx + pathway, </span></span>
<span id="cb3-1753"><a href="#cb3-1753" aria-hidden="true" tabindex="-1"></a><span class="in">                scales = "fixed", </span></span>
<span id="cb3-1754"><a href="#cb3-1754" aria-hidden="true" tabindex="-1"></a><span class="in">                nrow = length(pathway_names),</span></span>
<span id="cb3-1755"><a href="#cb3-1755" aria-hidden="true" tabindex="-1"></a><span class="in">                labeller = as_labeller(c(pathways, pdxs_to_use))</span></span>
<span id="cb3-1756"><a href="#cb3-1756" aria-hidden="true" tabindex="-1"></a><span class="in">            ) +</span></span>
<span id="cb3-1757"><a href="#cb3-1757" aria-hidden="true" tabindex="-1"></a><span class="in">            ggplot2::labs(</span></span>
<span id="cb3-1758"><a href="#cb3-1758" aria-hidden="true" tabindex="-1"></a><span class="in">                x = ifelse(i == 4, "Treatment", ""),</span></span>
<span id="cb3-1759"><a href="#cb3-1759" aria-hidden="true" tabindex="-1"></a><span class="in">                y = "GSVA scores"</span></span>
<span id="cb3-1760"><a href="#cb3-1760" aria-hidden="true" tabindex="-1"></a><span class="in">            ) +</span></span>
<span id="cb3-1761"><a href="#cb3-1761" aria-hidden="true" tabindex="-1"></a><span class="in">            ggplot2::scale_y_continuous(labels = scales::number_format(</span></span>
<span id="cb3-1762"><a href="#cb3-1762" aria-hidden="true" tabindex="-1"></a><span class="in">                accuracy = 0.01,</span></span>
<span id="cb3-1763"><a href="#cb3-1763" aria-hidden="true" tabindex="-1"></a><span class="in">                decimal.mark = '.'</span></span>
<span id="cb3-1764"><a href="#cb3-1764" aria-hidden="true" tabindex="-1"></a><span class="in">            )) + </span></span>
<span id="cb3-1765"><a href="#cb3-1765" aria-hidden="true" tabindex="-1"></a><span class="in">            ggplot2::theme_bw(base_size = 20) +</span></span>
<span id="cb3-1766"><a href="#cb3-1766" aria-hidden="true" tabindex="-1"></a><span class="in">            ggplot2::theme(legend.position="none") +</span></span>
<span id="cb3-1767"><a href="#cb3-1767" aria-hidden="true" tabindex="-1"></a><span class="in">            change_guides_point() +</span></span>
<span id="cb3-1768"><a href="#cb3-1768" aria-hidden="true" tabindex="-1"></a><span class="in">            change_plot_aes_point()</span></span>
<span id="cb3-1769"><a href="#cb3-1769" aria-hidden="true" tabindex="-1"></a><span class="in">    },</span></span>
<span id="cb3-1770"><a href="#cb3-1770" aria-hidden="true" tabindex="-1"></a><span class="in">    pathways = pathways,</span></span>
<span id="cb3-1771"><a href="#cb3-1771" aria-hidden="true" tabindex="-1"></a><span class="in">    pathway_name = names(pathways),</span></span>
<span id="cb3-1772"><a href="#cb3-1772" aria-hidden="true" tabindex="-1"></a><span class="in">    i = 1:length(pathways),</span></span>
<span id="cb3-1773"><a href="#cb3-1773" aria-hidden="true" tabindex="-1"></a><span class="in">    SIMPLIFY = FALSE,</span></span>
<span id="cb3-1774"><a href="#cb3-1774" aria-hidden="true" tabindex="-1"></a><span class="in">    USE.NAMES = TRUE</span></span>
<span id="cb3-1775"><a href="#cb3-1775" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-1776"><a href="#cb3-1776" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1777"><a href="#cb3-1777" aria-hidden="true" tabindex="-1"></a><span class="in">cowplot::plot_grid(</span></span>
<span id="cb3-1778"><a href="#cb3-1778" aria-hidden="true" tabindex="-1"></a><span class="in">    plotlist = plots_pathways,</span></span>
<span id="cb3-1779"><a href="#cb3-1779" aria-hidden="true" tabindex="-1"></a><span class="in">    nrow = length(plots_pathways) </span></span>
<span id="cb3-1780"><a href="#cb3-1780" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-1781"><a href="#cb3-1781" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-1782"><a href="#cb3-1782" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1783"><a href="#cb3-1783" aria-hidden="true" tabindex="-1"></a>What we can see from these scores is that if there is an increase</span>
<span id="cb3-1784"><a href="#cb3-1784" aria-hidden="true" tabindex="-1"></a>in G2M checkpoint and the control samples have scores below 0,</span>
<span id="cb3-1785"><a href="#cb3-1785" aria-hidden="true" tabindex="-1"></a>then the fourth principal component will change it is value, it </span>
<span id="cb3-1786"><a href="#cb3-1786" aria-hidden="true" tabindex="-1"></a>will go upwards and to the left in the molecular landscape,</span>
<span id="cb3-1787"><a href="#cb3-1787" aria-hidden="true" tabindex="-1"></a>reflecting the biological response of the tumors to the</span>
<span id="cb3-1788"><a href="#cb3-1788" aria-hidden="true" tabindex="-1"></a>given hormones. </span>
<span id="cb3-1789"><a href="#cb3-1789" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1790"><a href="#cb3-1790" aria-hidden="true" tabindex="-1"></a>Before we move on we also do the same using facet_grid instead of</span>
<span id="cb3-1791"><a href="#cb3-1791" aria-hidden="true" tabindex="-1"></a>facet_wrap.</span>
<span id="cb3-1792"><a href="#cb3-1792" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1793"><a href="#cb3-1793" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=16, fig.height=9}</span></span>
<span id="cb3-1794"><a href="#cb3-1794" aria-hidden="true" tabindex="-1"></a><span class="in">pdx_df_pca_sub %&gt;%</span></span>
<span id="cb3-1795"><a href="#cb3-1795" aria-hidden="true" tabindex="-1"></a><span class="in">    tidyr::pivot_longer(</span></span>
<span id="cb3-1796"><a href="#cb3-1796" aria-hidden="true" tabindex="-1"></a><span class="in">        cols = dplyr::all_of(names(pathways)),</span></span>
<span id="cb3-1797"><a href="#cb3-1797" aria-hidden="true" tabindex="-1"></a><span class="in">        names_to = "pathway",</span></span>
<span id="cb3-1798"><a href="#cb3-1798" aria-hidden="true" tabindex="-1"></a><span class="in">        values_to = "score"</span></span>
<span id="cb3-1799"><a href="#cb3-1799" aria-hidden="true" tabindex="-1"></a><span class="in">    ) %&gt;%</span></span>
<span id="cb3-1800"><a href="#cb3-1800" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::mutate(pathway = factor(pathway, levels = names(pathways))) %&gt;%</span></span>
<span id="cb3-1801"><a href="#cb3-1801" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::ggplot(aes(x = treatment, y = score)) +</span></span>
<span id="cb3-1802"><a href="#cb3-1802" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_boxplot(outlier.shape = NA) + </span></span>
<span id="cb3-1803"><a href="#cb3-1803" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_jitter(</span></span>
<span id="cb3-1804"><a href="#cb3-1804" aria-hidden="true" tabindex="-1"></a><span class="in">        size = 3, </span></span>
<span id="cb3-1805"><a href="#cb3-1805" aria-hidden="true" tabindex="-1"></a><span class="in">        alpha = 0.9, </span></span>
<span id="cb3-1806"><a href="#cb3-1806" aria-hidden="true" tabindex="-1"></a><span class="in">        mapping = aes(color = treatment)</span></span>
<span id="cb3-1807"><a href="#cb3-1807" aria-hidden="true" tabindex="-1"></a><span class="in">    ) + </span></span>
<span id="cb3-1808"><a href="#cb3-1808" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::scale_color_manual(values = c(</span></span>
<span id="cb3-1809"><a href="#cb3-1809" aria-hidden="true" tabindex="-1"></a><span class="in">        "black",</span></span>
<span id="cb3-1810"><a href="#cb3-1810" aria-hidden="true" tabindex="-1"></a><span class="in">        "blue"</span></span>
<span id="cb3-1811"><a href="#cb3-1811" aria-hidden="true" tabindex="-1"></a><span class="in">    )) + </span></span>
<span id="cb3-1812"><a href="#cb3-1812" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::facet_grid(</span></span>
<span id="cb3-1813"><a href="#cb3-1813" aria-hidden="true" tabindex="-1"></a><span class="in">        pathway ~ pdx, </span></span>
<span id="cb3-1814"><a href="#cb3-1814" aria-hidden="true" tabindex="-1"></a><span class="in">        #scales = "fixed", </span></span>
<span id="cb3-1815"><a href="#cb3-1815" aria-hidden="true" tabindex="-1"></a><span class="in">        #nrow = length(pathways),</span></span>
<span id="cb3-1816"><a href="#cb3-1816" aria-hidden="true" tabindex="-1"></a><span class="in">        labeller = as_labeller(c(pathways, pdxs_to_use))</span></span>
<span id="cb3-1817"><a href="#cb3-1817" aria-hidden="true" tabindex="-1"></a><span class="in">    ) +</span></span>
<span id="cb3-1818"><a href="#cb3-1818" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::labs(</span></span>
<span id="cb3-1819"><a href="#cb3-1819" aria-hidden="true" tabindex="-1"></a><span class="in">        x = "Treatment",</span></span>
<span id="cb3-1820"><a href="#cb3-1820" aria-hidden="true" tabindex="-1"></a><span class="in">        y = "GSVA score"</span></span>
<span id="cb3-1821"><a href="#cb3-1821" aria-hidden="true" tabindex="-1"></a><span class="in">    ) +</span></span>
<span id="cb3-1822"><a href="#cb3-1822" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::scale_y_continuous(labels = scales::number_format(</span></span>
<span id="cb3-1823"><a href="#cb3-1823" aria-hidden="true" tabindex="-1"></a><span class="in">        accuracy = 0.1,</span></span>
<span id="cb3-1824"><a href="#cb3-1824" aria-hidden="true" tabindex="-1"></a><span class="in">        decimal.mark = '.'</span></span>
<span id="cb3-1825"><a href="#cb3-1825" aria-hidden="true" tabindex="-1"></a><span class="in">    )) + </span></span>
<span id="cb3-1826"><a href="#cb3-1826" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme_bw(base_size = 25) +</span></span>
<span id="cb3-1827"><a href="#cb3-1827" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme(legend.position="none") +</span></span>
<span id="cb3-1828"><a href="#cb3-1828" aria-hidden="true" tabindex="-1"></a><span class="in">    change_guides_point() +</span></span>
<span id="cb3-1829"><a href="#cb3-1829" aria-hidden="true" tabindex="-1"></a><span class="in">    change_plot_aes_point()</span></span>
<span id="cb3-1830"><a href="#cb3-1830" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-1831"><a href="#cb3-1831" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1832"><a href="#cb3-1832" aria-hidden="true" tabindex="-1"></a>And the next two tables show the results for the differential expression</span>
<span id="cb3-1833"><a href="#cb3-1833" aria-hidden="true" tabindex="-1"></a>analysis on the pathway level for each PDX individually. The estimate, i.e.</span>
<span id="cb3-1834"><a href="#cb3-1834" aria-hidden="true" tabindex="-1"></a>difference, and the p-values are presented.</span>
<span id="cb3-1835"><a href="#cb3-1835" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1836"><a href="#cb3-1836" aria-hidden="true" tabindex="-1"></a>First for G2M: </span>
<span id="cb3-1837"><a href="#cb3-1837" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1840"><a href="#cb3-1840" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-1841"><a href="#cb3-1841" aria-hidden="true" tabindex="-1"></a><span class="co"># And below we perform differential expression analysis for each PDX</span></span>
<span id="cb3-1842"><a href="#cb3-1842" aria-hidden="true" tabindex="-1"></a><span class="co"># individually.</span></span>
<span id="cb3-1843"><a href="#cb3-1843" aria-hidden="true" tabindex="-1"></a>pathway <span class="ot">&lt;-</span> <span class="st">"HALLMARK_G2M_CHECKPOINT"</span></span>
<span id="cb3-1844"><a href="#cb3-1844" aria-hidden="true" tabindex="-1"></a>g2m_pvalues <span class="ot">&lt;-</span> pdx_df_pca_sub <span class="sc">%&gt;%</span></span>
<span id="cb3-1845"><a href="#cb3-1845" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(treatment <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"CTRL"</span>, <span class="st">"P4"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb3-1846"><a href="#cb3-1846" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">group_by</span>(pdx) <span class="sc">%&gt;%</span></span>
<span id="cb3-1847"><a href="#cb3-1847" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">group_modify</span>(</span>
<span id="cb3-1848"><a href="#cb3-1848" aria-hidden="true" tabindex="-1"></a>        <span class="sc">~</span> broom<span class="sc">::</span><span class="fu">tidy</span>(<span class="fu">lm</span>(<span class="fu">as.formula</span>(<span class="fu">paste0</span>(pathway, <span class="st">"~ treatment"</span>)), <span class="at">data =</span> .))</span>
<span id="cb3-1849"><a href="#cb3-1849" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb3-1850"><a href="#cb3-1850" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(term <span class="sc">==</span> <span class="st">"treatmentP4"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb3-1851"><a href="#cb3-1851" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb3-1852"><a href="#cb3-1852" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">pathway =</span> <span class="fu">c</span>(pathway)) <span class="sc">%&gt;%</span></span>
<span id="cb3-1853"><a href="#cb3-1853" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">arrange</span>(p.value) <span class="sc">%&gt;%</span></span>
<span id="cb3-1854"><a href="#cb3-1854" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb3-1855"><a href="#cb3-1855" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">across</span>(</span>
<span id="cb3-1856"><a href="#cb3-1856" aria-hidden="true" tabindex="-1"></a>            <span class="sc">-</span>dplyr<span class="sc">::</span><span class="fu">all_of</span>(<span class="fu">c</span>(<span class="st">"pdx"</span>, <span class="st">"term"</span>)), </span>
<span id="cb3-1857"><a href="#cb3-1857" aria-hidden="true" tabindex="-1"></a>            <span class="at">.fns =</span> \(x) <span class="fu">format</span>(x, <span class="at">digits =</span> <span class="dv">2</span>)</span>
<span id="cb3-1858"><a href="#cb3-1858" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb3-1859"><a href="#cb3-1859" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb3-1860"><a href="#cb3-1860" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(pdx, estimate, p.value, pathway) </span>
<span id="cb3-1861"><a href="#cb3-1861" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1862"><a href="#cb3-1862" aria-hidden="true" tabindex="-1"></a>g2m_pvalues <span class="sc">%&gt;%</span></span>
<span id="cb3-1863"><a href="#cb3-1863" aria-hidden="true" tabindex="-1"></a>    DT<span class="sc">::</span><span class="fu">datatable</span>()</span>
<span id="cb3-1864"><a href="#cb3-1864" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-1865"><a href="#cb3-1865" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1866"><a href="#cb3-1866" aria-hidden="true" tabindex="-1"></a>Then to Estrogen Early: </span>
<span id="cb3-1867"><a href="#cb3-1867" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1870"><a href="#cb3-1870" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-1871"><a href="#cb3-1871" aria-hidden="true" tabindex="-1"></a><span class="co"># And below we perform differential expression analysis for each PDX</span></span>
<span id="cb3-1872"><a href="#cb3-1872" aria-hidden="true" tabindex="-1"></a><span class="co"># individually.</span></span>
<span id="cb3-1873"><a href="#cb3-1873" aria-hidden="true" tabindex="-1"></a>pathway <span class="ot">&lt;-</span> <span class="st">"HALLMARK_ESTROGEN_RESPONSE_EARLY"</span></span>
<span id="cb3-1874"><a href="#cb3-1874" aria-hidden="true" tabindex="-1"></a>estrogen_early_pvalues <span class="ot">&lt;-</span> pdx_df_pca_sub <span class="sc">%&gt;%</span></span>
<span id="cb3-1875"><a href="#cb3-1875" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(treatment <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"CTRL"</span>, <span class="st">"P4"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb3-1876"><a href="#cb3-1876" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">group_by</span>(pdx) <span class="sc">%&gt;%</span></span>
<span id="cb3-1877"><a href="#cb3-1877" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">group_modify</span>(</span>
<span id="cb3-1878"><a href="#cb3-1878" aria-hidden="true" tabindex="-1"></a>        <span class="sc">~</span> broom<span class="sc">::</span><span class="fu">tidy</span>(<span class="fu">lm</span>(<span class="fu">as.formula</span>(<span class="fu">paste0</span>(pathway, <span class="st">"~ treatment"</span>)), <span class="at">data =</span> .))</span>
<span id="cb3-1879"><a href="#cb3-1879" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb3-1880"><a href="#cb3-1880" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(term <span class="sc">==</span> <span class="st">"treatmentP4"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb3-1881"><a href="#cb3-1881" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb3-1882"><a href="#cb3-1882" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">pathway =</span> <span class="fu">c</span>(pathway)) <span class="sc">%&gt;%</span></span>
<span id="cb3-1883"><a href="#cb3-1883" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">arrange</span>(p.value) <span class="sc">%&gt;%</span></span>
<span id="cb3-1884"><a href="#cb3-1884" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb3-1885"><a href="#cb3-1885" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">across</span>(</span>
<span id="cb3-1886"><a href="#cb3-1886" aria-hidden="true" tabindex="-1"></a>            <span class="sc">-</span>dplyr<span class="sc">::</span><span class="fu">all_of</span>(<span class="fu">c</span>(<span class="st">"pdx"</span>, <span class="st">"term"</span>)), </span>
<span id="cb3-1887"><a href="#cb3-1887" aria-hidden="true" tabindex="-1"></a>            <span class="at">.fns =</span> \(x) <span class="fu">format</span>(x, <span class="at">digits =</span> <span class="dv">2</span>)</span>
<span id="cb3-1888"><a href="#cb3-1888" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb3-1889"><a href="#cb3-1889" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb3-1890"><a href="#cb3-1890" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(pdx, estimate, p.value, pathway)</span>
<span id="cb3-1891"><a href="#cb3-1891" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb3-1892"><a href="#cb3-1892" aria-hidden="true" tabindex="-1"></a>estrogen_early_pvalues <span class="sc">%&gt;%</span></span>
<span id="cb3-1893"><a href="#cb3-1893" aria-hidden="true" tabindex="-1"></a>    DT<span class="sc">::</span><span class="fu">datatable</span>()</span>
<span id="cb3-1894"><a href="#cb3-1894" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-1895"><a href="#cb3-1895" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1896"><a href="#cb3-1896" aria-hidden="true" tabindex="-1"></a>Next we add these p-values to the plot.</span>
<span id="cb3-1897"><a href="#cb3-1897" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1898"><a href="#cb3-1898" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=16, fig.height=7}</span></span>
<span id="cb3-1899"><a href="#cb3-1899" aria-hidden="true" tabindex="-1"></a><span class="in">#| label: fig-pdx-scores-with-pvalues</span></span>
<span id="cb3-1900"><a href="#cb3-1900" aria-hidden="true" tabindex="-1"></a><span class="in">pvalues_pdxs &lt;- rbind(</span></span>
<span id="cb3-1901"><a href="#cb3-1901" aria-hidden="true" tabindex="-1"></a><span class="in">    estrogen_early_pvalues,</span></span>
<span id="cb3-1902"><a href="#cb3-1902" aria-hidden="true" tabindex="-1"></a><span class="in">    g2m_pvalues</span></span>
<span id="cb3-1903"><a href="#cb3-1903" aria-hidden="true" tabindex="-1"></a><span class="in">) %&gt;%</span></span>
<span id="cb3-1904"><a href="#cb3-1904" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::mutate(dplyr::across(</span></span>
<span id="cb3-1905"><a href="#cb3-1905" aria-hidden="true" tabindex="-1"></a><span class="in">        c(estimate, p.value),</span></span>
<span id="cb3-1906"><a href="#cb3-1906" aria-hidden="true" tabindex="-1"></a><span class="in">        as.numeric</span></span>
<span id="cb3-1907"><a href="#cb3-1907" aria-hidden="true" tabindex="-1"></a><span class="in">    )) %&gt;%</span></span>
<span id="cb3-1908"><a href="#cb3-1908" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::mutate(</span></span>
<span id="cb3-1909"><a href="#cb3-1909" aria-hidden="true" tabindex="-1"></a><span class="in">        pdx = factor(</span></span>
<span id="cb3-1910"><a href="#cb3-1910" aria-hidden="true" tabindex="-1"></a><span class="in">            pdx, </span></span>
<span id="cb3-1911"><a href="#cb3-1911" aria-hidden="true" tabindex="-1"></a><span class="in">            levels = c("METS15", "T105", "T109", "T110","T111","T113")</span></span>
<span id="cb3-1912"><a href="#cb3-1912" aria-hidden="true" tabindex="-1"></a><span class="in">        )</span></span>
<span id="cb3-1913"><a href="#cb3-1913" aria-hidden="true" tabindex="-1"></a><span class="in">    ) %&gt;%</span></span>
<span id="cb3-1914"><a href="#cb3-1914" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::mutate(</span></span>
<span id="cb3-1915"><a href="#cb3-1915" aria-hidden="true" tabindex="-1"></a><span class="in">        p.value = format.pval(p.value, digits = 1)</span></span>
<span id="cb3-1916"><a href="#cb3-1916" aria-hidden="true" tabindex="-1"></a><span class="in">    )</span></span>
<span id="cb3-1917"><a href="#cb3-1917" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1918"><a href="#cb3-1918" aria-hidden="true" tabindex="-1"></a><span class="in">pdx_df_pca_sub %&gt;%</span></span>
<span id="cb3-1919"><a href="#cb3-1919" aria-hidden="true" tabindex="-1"></a><span class="in">    tidyr::pivot_longer(</span></span>
<span id="cb3-1920"><a href="#cb3-1920" aria-hidden="true" tabindex="-1"></a><span class="in">        cols = dplyr::all_of(names(pathways)),</span></span>
<span id="cb3-1921"><a href="#cb3-1921" aria-hidden="true" tabindex="-1"></a><span class="in">        names_to = "pathway",</span></span>
<span id="cb3-1922"><a href="#cb3-1922" aria-hidden="true" tabindex="-1"></a><span class="in">        values_to = "score"</span></span>
<span id="cb3-1923"><a href="#cb3-1923" aria-hidden="true" tabindex="-1"></a><span class="in">    ) %&gt;%</span></span>
<span id="cb3-1924"><a href="#cb3-1924" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::filter(pathway != "GOBP_RESPONSE_TO_PROGESTERONE") %&gt;%</span></span>
<span id="cb3-1925"><a href="#cb3-1925" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::mutate(pathway = factor(pathway, levels = names(pathways))) %&gt;%</span></span>
<span id="cb3-1926"><a href="#cb3-1926" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::ggplot(aes(x = treatment, y = score)) +</span></span>
<span id="cb3-1927"><a href="#cb3-1927" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_boxplot(outlier.shape = NA) + </span></span>
<span id="cb3-1928"><a href="#cb3-1928" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_jitter(</span></span>
<span id="cb3-1929"><a href="#cb3-1929" aria-hidden="true" tabindex="-1"></a><span class="in">        size = 3, </span></span>
<span id="cb3-1930"><a href="#cb3-1930" aria-hidden="true" tabindex="-1"></a><span class="in">        alpha = 0.9, </span></span>
<span id="cb3-1931"><a href="#cb3-1931" aria-hidden="true" tabindex="-1"></a><span class="in">        mapping = aes(color = treatment)</span></span>
<span id="cb3-1932"><a href="#cb3-1932" aria-hidden="true" tabindex="-1"></a><span class="in">    ) + </span></span>
<span id="cb3-1933"><a href="#cb3-1933" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::scale_color_manual(values = c(</span></span>
<span id="cb3-1934"><a href="#cb3-1934" aria-hidden="true" tabindex="-1"></a><span class="in">        "black",</span></span>
<span id="cb3-1935"><a href="#cb3-1935" aria-hidden="true" tabindex="-1"></a><span class="in">        "blue"</span></span>
<span id="cb3-1936"><a href="#cb3-1936" aria-hidden="true" tabindex="-1"></a><span class="in">    )) + </span></span>
<span id="cb3-1937"><a href="#cb3-1937" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_text(</span></span>
<span id="cb3-1938"><a href="#cb3-1938" aria-hidden="true" tabindex="-1"></a><span class="in">        data = pvalues_pdxs, </span></span>
<span id="cb3-1939"><a href="#cb3-1939" aria-hidden="true" tabindex="-1"></a><span class="in">        mapping = aes(label = p.value),</span></span>
<span id="cb3-1940"><a href="#cb3-1940" aria-hidden="true" tabindex="-1"></a><span class="in">        x = -Inf,</span></span>
<span id="cb3-1941"><a href="#cb3-1941" aria-hidden="true" tabindex="-1"></a><span class="in">        y = +Inf,</span></span>
<span id="cb3-1942"><a href="#cb3-1942" aria-hidden="true" tabindex="-1"></a><span class="in">        hjust = -0.1,</span></span>
<span id="cb3-1943"><a href="#cb3-1943" aria-hidden="true" tabindex="-1"></a><span class="in">        vjust = 1.5,</span></span>
<span id="cb3-1944"><a href="#cb3-1944" aria-hidden="true" tabindex="-1"></a><span class="in">        size = 7</span></span>
<span id="cb3-1945"><a href="#cb3-1945" aria-hidden="true" tabindex="-1"></a><span class="in">    ) + </span></span>
<span id="cb3-1946"><a href="#cb3-1946" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::facet_grid(</span></span>
<span id="cb3-1947"><a href="#cb3-1947" aria-hidden="true" tabindex="-1"></a><span class="in">        pathway ~ pdx, </span></span>
<span id="cb3-1948"><a href="#cb3-1948" aria-hidden="true" tabindex="-1"></a><span class="in">        labeller = as_labeller(c(pathways, pdxs_to_use))</span></span>
<span id="cb3-1949"><a href="#cb3-1949" aria-hidden="true" tabindex="-1"></a><span class="in">    ) +</span></span>
<span id="cb3-1950"><a href="#cb3-1950" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::labs(</span></span>
<span id="cb3-1951"><a href="#cb3-1951" aria-hidden="true" tabindex="-1"></a><span class="in">        x = "Treatment",</span></span>
<span id="cb3-1952"><a href="#cb3-1952" aria-hidden="true" tabindex="-1"></a><span class="in">        y = "GSVA score"</span></span>
<span id="cb3-1953"><a href="#cb3-1953" aria-hidden="true" tabindex="-1"></a><span class="in">    ) +</span></span>
<span id="cb3-1954"><a href="#cb3-1954" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::coord_cartesian(</span></span>
<span id="cb3-1955"><a href="#cb3-1955" aria-hidden="true" tabindex="-1"></a><span class="in">        ylim = c(-0.6, 0.5)</span></span>
<span id="cb3-1956"><a href="#cb3-1956" aria-hidden="true" tabindex="-1"></a><span class="in">    ) + </span></span>
<span id="cb3-1957"><a href="#cb3-1957" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::scale_y_continuous(labels = scales::number_format(</span></span>
<span id="cb3-1958"><a href="#cb3-1958" aria-hidden="true" tabindex="-1"></a><span class="in">        accuracy = 0.1,</span></span>
<span id="cb3-1959"><a href="#cb3-1959" aria-hidden="true" tabindex="-1"></a><span class="in">        decimal.mark = '.'</span></span>
<span id="cb3-1960"><a href="#cb3-1960" aria-hidden="true" tabindex="-1"></a><span class="in">    )) + </span></span>
<span id="cb3-1961"><a href="#cb3-1961" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme_bw(base_size = 25) +</span></span>
<span id="cb3-1962"><a href="#cb3-1962" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme(legend.position="none") +</span></span>
<span id="cb3-1963"><a href="#cb3-1963" aria-hidden="true" tabindex="-1"></a><span class="in">    change_guides_point() +</span></span>
<span id="cb3-1964"><a href="#cb3-1964" aria-hidden="true" tabindex="-1"></a><span class="in">    change_plot_aes_point()</span></span>
<span id="cb3-1965"><a href="#cb3-1965" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-1966"><a href="#cb3-1966" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1967"><a href="#cb3-1967" aria-hidden="true" tabindex="-1"></a>We now further validate the results with the stainings performed</span>
<span id="cb3-1968"><a href="#cb3-1968" aria-hidden="true" tabindex="-1"></a>on those samples. We will compare the differences with the stainings and</span>
<span id="cb3-1969"><a href="#cb3-1969" aria-hidden="true" tabindex="-1"></a>the number of PR+ tumors cells.</span>
<span id="cb3-1970"><a href="#cb3-1970" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1973"><a href="#cb3-1973" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-1974"><a href="#cb3-1974" aria-hidden="true" tabindex="-1"></a>staining_pdxs <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../../data/quantification.csv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb3-1975"><a href="#cb3-1975" aria-hidden="true" tabindex="-1"></a>    janitor<span class="sc">::</span><span class="fu">clean_names</span>() <span class="sc">%&gt;%</span></span>
<span id="cb3-1976"><a href="#cb3-1976" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>dplyr<span class="sc">::</span><span class="fu">all_of</span>(<span class="fu">c</span>(<span class="st">"type_corrected"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb3-1977"><a href="#cb3-1977" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">pdx =</span> tumor) <span class="sc">%&gt;%</span></span>
<span id="cb3-1978"><a href="#cb3-1978" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">treatment =</span> dplyr<span class="sc">::</span><span class="fu">case_when</span>(</span>
<span id="cb3-1979"><a href="#cb3-1979" aria-hidden="true" tabindex="-1"></a>        treatment <span class="sc">==</span> <span class="st">"P20"</span> <span class="sc">~</span> <span class="st">"P4"</span>,</span>
<span id="cb3-1980"><a href="#cb3-1980" aria-hidden="true" tabindex="-1"></a>        treatment <span class="sc">==</span> <span class="st">"P20+E2"</span> <span class="sc">~</span> <span class="st">"E2P4"</span>,</span>
<span id="cb3-1981"><a href="#cb3-1981" aria-hidden="true" tabindex="-1"></a>        treatment <span class="sc">==</span> <span class="st">"P4+E2"</span> <span class="sc">~</span> <span class="st">"E2P4"</span>,</span>
<span id="cb3-1982"><a href="#cb3-1982" aria-hidden="true" tabindex="-1"></a>        treatment <span class="sc">==</span> <span class="st">"E2+P4"</span> <span class="sc">~</span> <span class="st">"E2P4"</span>,</span>
<span id="cb3-1983"><a href="#cb3-1983" aria-hidden="true" tabindex="-1"></a>        treatment <span class="sc">==</span> <span class="st">"E2+P20"</span> <span class="sc">~</span> <span class="st">"E2P4"</span>,</span>
<span id="cb3-1984"><a href="#cb3-1984" aria-hidden="true" tabindex="-1"></a>        <span class="cn">TRUE</span> <span class="sc">~</span> <span class="fu">as.character</span>(.<span class="sc">$</span>treatment)</span>
<span id="cb3-1985"><a href="#cb3-1985" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">%&gt;%</span></span>
<span id="cb3-1986"><a href="#cb3-1986" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb3-1987"><a href="#cb3-1987" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">positive_percentage =</span> </span>
<span id="cb3-1988"><a href="#cb3-1988" aria-hidden="true" tabindex="-1"></a>        <span class="fu">sum</span>(<span class="fu">c</span>(num1, num2, num3))<span class="sc">/</span><span class="fu">sum</span>(<span class="fu">c</span>(num1, num2, num3, num_negative))</span>
<span id="cb3-1989"><a href="#cb3-1989" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb3-1990"><a href="#cb3-1990" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(positive_percentage))</span>
<span id="cb3-1991"><a href="#cb3-1991" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-1992"><a href="#cb3-1992" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1993"><a href="#cb3-1993" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.height=7, fig.width=9}</span></span>
<span id="cb3-1994"><a href="#cb3-1994" aria-hidden="true" tabindex="-1"></a><span class="in">#| label: fig-quantification-pdx-pr</span></span>
<span id="cb3-1995"><a href="#cb3-1995" aria-hidden="true" tabindex="-1"></a><span class="in">#| fig-cap: Quantification of PR+ cells based on DAB stainings performed</span></span>
<span id="cb3-1996"><a href="#cb3-1996" aria-hidden="true" tabindex="-1"></a><span class="in">#|     on the patient derived xenografts.</span></span>
<span id="cb3-1997"><a href="#cb3-1997" aria-hidden="true" tabindex="-1"></a><span class="in">plot_quantification(</span></span>
<span id="cb3-1998"><a href="#cb3-1998" aria-hidden="true" tabindex="-1"></a><span class="in">    staining_pdxs %&gt;%</span></span>
<span id="cb3-1999"><a href="#cb3-1999" aria-hidden="true" tabindex="-1"></a><span class="in">        dplyr::filter(treatment %in% c("CTRL", "P4")),</span></span>
<span id="cb3-2000"><a href="#cb3-2000" aria-hidden="true" tabindex="-1"></a><span class="in">    "PR",</span></span>
<span id="cb3-2001"><a href="#cb3-2001" aria-hidden="true" tabindex="-1"></a><span class="in">    base_size = 18</span></span>
<span id="cb3-2002"><a href="#cb3-2002" aria-hidden="true" tabindex="-1"></a><span class="in">) +</span></span>
<span id="cb3-2003"><a href="#cb3-2003" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::scale_color_manual(values = c("black", "blue"))</span></span>
<span id="cb3-2004"><a href="#cb3-2004" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-2005"><a href="#cb3-2005" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2006"><a href="#cb3-2006" aria-hidden="true" tabindex="-1"></a>We see that T111 has a big increase in absolute values. T113 as well</span>
<span id="cb3-2007"><a href="#cb3-2007" aria-hidden="true" tabindex="-1"></a>has a big increase, but has a baseline level already low and it does</span>
<span id="cb3-2008"><a href="#cb3-2008" aria-hidden="true" tabindex="-1"></a>not go to higher values. Probably the reason there is not much</span>
<span id="cb3-2009"><a href="#cb3-2009" aria-hidden="true" tabindex="-1"></a>difference in the fourth component of the PDX T113 is because of the</span>
<span id="cb3-2010"><a href="#cb3-2010" aria-hidden="true" tabindex="-1"></a>low Ki67 index. </span>
<span id="cb3-2011"><a href="#cb3-2011" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2012"><a href="#cb3-2012" aria-hidden="true" tabindex="-1"></a>Based on the data shown here we see that there is an increase in </span>
<span id="cb3-2013"><a href="#cb3-2013" aria-hidden="true" tabindex="-1"></a>proliferation upon P4 treatment for T111 and T113. Moreover,</span>
<span id="cb3-2014"><a href="#cb3-2014" aria-hidden="true" tabindex="-1"></a>P4 reduced estrogen signaling and also change the third component of the</span>
<span id="cb3-2015"><a href="#cb3-2015" aria-hidden="true" tabindex="-1"></a>three PDXs to a smaller value, towards a more basal like region. Overall,</span>
<span id="cb3-2016"><a href="#cb3-2016" aria-hidden="true" tabindex="-1"></a>the molecular landscape is able to capture the differences between</span>
<span id="cb3-2017"><a href="#cb3-2017" aria-hidden="true" tabindex="-1"></a>the treatments.</span>
<span id="cb3-2018"><a href="#cb3-2018" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2019"><a href="#cb3-2019" aria-hidden="true" tabindex="-1"></a>Lastly we check changes in the embedding upon E2 treatment.</span>
<span id="cb3-2020"><a href="#cb3-2020" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2021"><a href="#cb3-2021" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=12, fig.height=9}</span></span>
<span id="cb3-2022"><a href="#cb3-2022" aria-hidden="true" tabindex="-1"></a><span class="in">#| label: fig-pca-pdx-only-embedding-e2-ctrl</span></span>
<span id="cb3-2023"><a href="#cb3-2023" aria-hidden="true" tabindex="-1"></a><span class="in">#| fig-cap: Embedding of only PDX control and E2 treated samples </span></span>
<span id="cb3-2024"><a href="#cb3-2024" aria-hidden="true" tabindex="-1"></a><span class="in">pdx_plots_embedding_e2 &lt;- sapply(</span></span>
<span id="cb3-2025"><a href="#cb3-2025" aria-hidden="true" tabindex="-1"></a><span class="in">    c("METS15", "T113", "T111", "T105", "T109", "T110"), </span></span>
<span id="cb3-2026"><a href="#cb3-2026" aria-hidden="true" tabindex="-1"></a><span class="in">    pdx_only_embedding,</span></span>
<span id="cb3-2027"><a href="#cb3-2027" aria-hidden="true" tabindex="-1"></a><span class="in">    pdx_df_pca = pdx_df_pca,</span></span>
<span id="cb3-2028"><a href="#cb3-2028" aria-hidden="true" tabindex="-1"></a><span class="in">    treatments_pdx = c("CTRL", "E2"),</span></span>
<span id="cb3-2029"><a href="#cb3-2029" aria-hidden="true" tabindex="-1"></a><span class="in">    shape_pdx = shape_pdx,</span></span>
<span id="cb3-2030"><a href="#cb3-2030" aria-hidden="true" tabindex="-1"></a><span class="in">    USE.NAMES = TRUE, </span></span>
<span id="cb3-2031"><a href="#cb3-2031" aria-hidden="true" tabindex="-1"></a><span class="in">    simplify = FALSE</span></span>
<span id="cb3-2032"><a href="#cb3-2032" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-2033"><a href="#cb3-2033" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2034"><a href="#cb3-2034" aria-hidden="true" tabindex="-1"></a><span class="in">cowplot::plot_grid(plotlist = pdx_plots_embedding_e2, ncol = 3)</span></span>
<span id="cb3-2035"><a href="#cb3-2035" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-2036"><a href="#cb3-2036" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2037"><a href="#cb3-2037" aria-hidden="true" tabindex="-1"></a>It looks like there is not big changes in the molecular landscape </span>
<span id="cb3-2038"><a href="#cb3-2038" aria-hidden="true" tabindex="-1"></a>as there was for samples treated with progesterone.</span>
<span id="cb3-2039"><a href="#cb3-2039" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2040"><a href="#cb3-2040" aria-hidden="true" tabindex="-1"></a>And lastly we show for Ki67 as well.</span>
<span id="cb3-2041"><a href="#cb3-2041" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2042"><a href="#cb3-2042" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.height=6}</span></span>
<span id="cb3-2043"><a href="#cb3-2043" aria-hidden="true" tabindex="-1"></a><span class="in">#| label: fig-quantification-pdx-ki67</span></span>
<span id="cb3-2044"><a href="#cb3-2044" aria-hidden="true" tabindex="-1"></a><span class="in">#| fig-cap: Quantification of Ki67+ cells based on DAB stainings performed</span></span>
<span id="cb3-2045"><a href="#cb3-2045" aria-hidden="true" tabindex="-1"></a><span class="in">#|     on the patient derived xenografts.</span></span>
<span id="cb3-2046"><a href="#cb3-2046" aria-hidden="true" tabindex="-1"></a><span class="in">plot_quantification(</span></span>
<span id="cb3-2047"><a href="#cb3-2047" aria-hidden="true" tabindex="-1"></a><span class="in">    staining_pdxs %&gt;%</span></span>
<span id="cb3-2048"><a href="#cb3-2048" aria-hidden="true" tabindex="-1"></a><span class="in">        dplyr::filter(treatment %in% c("CTRL", "P4")),</span></span>
<span id="cb3-2049"><a href="#cb3-2049" aria-hidden="true" tabindex="-1"></a><span class="in">    "Ki67",</span></span>
<span id="cb3-2050"><a href="#cb3-2050" aria-hidden="true" tabindex="-1"></a><span class="in">    base_size = 18</span></span>
<span id="cb3-2051"><a href="#cb3-2051" aria-hidden="true" tabindex="-1"></a><span class="in">) +</span></span>
<span id="cb3-2052"><a href="#cb3-2052" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::scale_color_manual(</span></span>
<span id="cb3-2053"><a href="#cb3-2053" aria-hidden="true" tabindex="-1"></a><span class="in">        values = c("black", "blue")</span></span>
<span id="cb3-2054"><a href="#cb3-2054" aria-hidden="true" tabindex="-1"></a><span class="in">    )</span></span>
<span id="cb3-2055"><a href="#cb3-2055" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-2056"><a href="#cb3-2056" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2057"><a href="#cb3-2057" aria-hidden="true" tabindex="-1"></a>And below we plot both results together with the respective p-values.</span>
<span id="cb3-2058"><a href="#cb3-2058" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2061"><a href="#cb3-2061" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-2062"><a href="#cb3-2062" aria-hidden="true" tabindex="-1"></a>staining_pvalues <span class="ot">&lt;-</span> <span class="fu">sapply</span>(</span>
<span id="cb3-2063"><a href="#cb3-2063" aria-hidden="true" tabindex="-1"></a>    <span class="fu">c</span>(<span class="st">"PR"</span>, <span class="st">"Ki67"</span>), </span>
<span id="cb3-2064"><a href="#cb3-2064" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(staining_protein, staining_pdxs){</span>
<span id="cb3-2065"><a href="#cb3-2065" aria-hidden="true" tabindex="-1"></a>        staining_pdxs <span class="sc">%&gt;%</span></span>
<span id="cb3-2066"><a href="#cb3-2066" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">filter</span>(treatment <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"CTRL"</span>, <span class="st">"P4"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb3-2067"><a href="#cb3-2067" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">filter</span>(staining <span class="sc">==</span> staining_protein) <span class="sc">%&gt;%</span></span>
<span id="cb3-2068"><a href="#cb3-2068" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">group_by</span>(pdx) <span class="sc">%&gt;%</span></span>
<span id="cb3-2069"><a href="#cb3-2069" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">group_modify</span>(</span>
<span id="cb3-2070"><a href="#cb3-2070" aria-hidden="true" tabindex="-1"></a>                <span class="sc">~</span> broom<span class="sc">::</span><span class="fu">tidy</span>(<span class="fu">glm</span>(</span>
<span id="cb3-2071"><a href="#cb3-2071" aria-hidden="true" tabindex="-1"></a>                    positive_percentage <span class="sc">~</span> treatment, </span>
<span id="cb3-2072"><a href="#cb3-2072" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data =</span> .,</span>
<span id="cb3-2073"><a href="#cb3-2073" aria-hidden="true" tabindex="-1"></a>                    <span class="at">family =</span> <span class="fu">Gamma</span>(<span class="at">link =</span> <span class="st">"identity"</span>)</span>
<span id="cb3-2074"><a href="#cb3-2074" aria-hidden="true" tabindex="-1"></a>                ))</span>
<span id="cb3-2075"><a href="#cb3-2075" aria-hidden="true" tabindex="-1"></a>            ) <span class="sc">%&gt;%</span></span>
<span id="cb3-2076"><a href="#cb3-2076" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">filter</span>(term <span class="sc">==</span> <span class="st">"treatmentP4"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb3-2077"><a href="#cb3-2077" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb3-2078"><a href="#cb3-2078" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">staining =</span> <span class="fu">c</span>(staining_protein)) <span class="sc">%&gt;%</span></span>
<span id="cb3-2079"><a href="#cb3-2079" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">arrange</span>(estimate) <span class="sc">%&gt;%</span></span>
<span id="cb3-2080"><a href="#cb3-2080" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">select</span>(pdx, estimate, p.value, staining) </span>
<span id="cb3-2081"><a href="#cb3-2081" aria-hidden="true" tabindex="-1"></a>    }, </span>
<span id="cb3-2082"><a href="#cb3-2082" aria-hidden="true" tabindex="-1"></a>    <span class="at">staining_pdxs =</span> staining_pdxs,</span>
<span id="cb3-2083"><a href="#cb3-2083" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>, </span>
<span id="cb3-2084"><a href="#cb3-2084" aria-hidden="true" tabindex="-1"></a>    <span class="at">simplify =</span> <span class="cn">FALSE</span></span>
<span id="cb3-2085"><a href="#cb3-2085" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb3-2086"><a href="#cb3-2086" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">bind_rows</span>() <span class="sc">%&gt;%</span></span>
<span id="cb3-2087"><a href="#cb3-2087" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">p.value =</span> <span class="fu">format.pval</span>(p.value, <span class="at">digits =</span> <span class="dv">1</span>))</span>
<span id="cb3-2088"><a href="#cb3-2088" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-2089"><a href="#cb3-2089" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2090"><a href="#cb3-2090" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=16, fig.height=7}</span></span>
<span id="cb3-2091"><a href="#cb3-2091" aria-hidden="true" tabindex="-1"></a><span class="in">#| label: fig-staining-quant-pr-ki67</span></span>
<span id="cb3-2092"><a href="#cb3-2092" aria-hidden="true" tabindex="-1"></a><span class="in">staining_pdxs %&gt;%</span></span>
<span id="cb3-2093"><a href="#cb3-2093" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::filter(</span></span>
<span id="cb3-2094"><a href="#cb3-2094" aria-hidden="true" tabindex="-1"></a><span class="in">        staining %in% c("PR", "Ki67") &amp; </span></span>
<span id="cb3-2095"><a href="#cb3-2095" aria-hidden="true" tabindex="-1"></a><span class="in">            treatment %in% c("CTRL", "P4")</span></span>
<span id="cb3-2096"><a href="#cb3-2096" aria-hidden="true" tabindex="-1"></a><span class="in">    ) %&gt;%</span></span>
<span id="cb3-2097"><a href="#cb3-2097" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::ggplot(aes(x = treatment, y = positive_percentage*100)) +</span></span>
<span id="cb3-2098"><a href="#cb3-2098" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_boxplot(outlier.shape = NA) + </span></span>
<span id="cb3-2099"><a href="#cb3-2099" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_jitter(</span></span>
<span id="cb3-2100"><a href="#cb3-2100" aria-hidden="true" tabindex="-1"></a><span class="in">        size = 3, </span></span>
<span id="cb3-2101"><a href="#cb3-2101" aria-hidden="true" tabindex="-1"></a><span class="in">        alpha = 0.9, </span></span>
<span id="cb3-2102"><a href="#cb3-2102" aria-hidden="true" tabindex="-1"></a><span class="in">        mapping = aes(color = treatment)</span></span>
<span id="cb3-2103"><a href="#cb3-2103" aria-hidden="true" tabindex="-1"></a><span class="in">    ) + </span></span>
<span id="cb3-2104"><a href="#cb3-2104" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::scale_color_manual(values = c("black", "blue")) +</span></span>
<span id="cb3-2105"><a href="#cb3-2105" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::facet_grid(staining~pdx, scales = "free_y") +</span></span>
<span id="cb3-2106"><a href="#cb3-2106" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::labs(</span></span>
<span id="cb3-2107"><a href="#cb3-2107" aria-hidden="true" tabindex="-1"></a><span class="in">        x = "Treatment",</span></span>
<span id="cb3-2108"><a href="#cb3-2108" aria-hidden="true" tabindex="-1"></a><span class="in">        y = paste0("% (Positive cells)")</span></span>
<span id="cb3-2109"><a href="#cb3-2109" aria-hidden="true" tabindex="-1"></a><span class="in">    ) +</span></span>
<span id="cb3-2110"><a href="#cb3-2110" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_text(</span></span>
<span id="cb3-2111"><a href="#cb3-2111" aria-hidden="true" tabindex="-1"></a><span class="in">        data = staining_pvalues, </span></span>
<span id="cb3-2112"><a href="#cb3-2112" aria-hidden="true" tabindex="-1"></a><span class="in">        mapping = aes(label = p.value),</span></span>
<span id="cb3-2113"><a href="#cb3-2113" aria-hidden="true" tabindex="-1"></a><span class="in">        x = -Inf,</span></span>
<span id="cb3-2114"><a href="#cb3-2114" aria-hidden="true" tabindex="-1"></a><span class="in">        y = +Inf,</span></span>
<span id="cb3-2115"><a href="#cb3-2115" aria-hidden="true" tabindex="-1"></a><span class="in">        hjust = -2.5,</span></span>
<span id="cb3-2116"><a href="#cb3-2116" aria-hidden="true" tabindex="-1"></a><span class="in">        vjust = 1.5,</span></span>
<span id="cb3-2117"><a href="#cb3-2117" aria-hidden="true" tabindex="-1"></a><span class="in">        size = 7</span></span>
<span id="cb3-2118"><a href="#cb3-2118" aria-hidden="true" tabindex="-1"></a><span class="in">    ) + </span></span>
<span id="cb3-2119"><a href="#cb3-2119" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme_bw(base_size = 25) + </span></span>
<span id="cb3-2120"><a href="#cb3-2120" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme(legend.position = "none") + </span></span>
<span id="cb3-2121"><a href="#cb3-2121" aria-hidden="true" tabindex="-1"></a><span class="in">    change_guides_point() + </span></span>
<span id="cb3-2122"><a href="#cb3-2122" aria-hidden="true" tabindex="-1"></a><span class="in">    change_plot_aes_point()</span></span>
<span id="cb3-2123"><a href="#cb3-2123" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-2124"><a href="#cb3-2124" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2125"><a href="#cb3-2125" aria-hidden="true" tabindex="-1"></a><span class="fu">### Growth curves</span></span>
<span id="cb3-2126"><a href="#cb3-2126" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2127"><a href="#cb3-2127" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, message=FALSE, eval = first_run}</span></span>
<span id="cb3-2128"><a href="#cb3-2128" aria-hidden="true" tabindex="-1"></a><span class="in"># run all fittings together</span></span>
<span id="cb3-2129"><a href="#cb3-2129" aria-hidden="true" tabindex="-1"></a><span class="in">tables &lt;- readRDS("../../data/2019_pdx_brisken/ivis_growth_curves.rds")</span></span>
<span id="cb3-2130"><a href="#cb3-2130" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2131"><a href="#cb3-2131" aria-hidden="true" tabindex="-1"></a><span class="in">all_fits &lt;- lapply(</span></span>
<span id="cb3-2132"><a href="#cb3-2132" aria-hidden="true" tabindex="-1"></a><span class="in">    tables,</span></span>
<span id="cb3-2133"><a href="#cb3-2133" aria-hidden="true" tabindex="-1"></a><span class="in">    function(table_pdx){</span></span>
<span id="cb3-2134"><a href="#cb3-2134" aria-hidden="true" tabindex="-1"></a><span class="in">        rstanarm::stan_glmer(</span></span>
<span id="cb3-2135"><a href="#cb3-2135" aria-hidden="true" tabindex="-1"></a><span class="in">            log10tf ~ days_after_treatment * treatment + (1 | id),</span></span>
<span id="cb3-2136"><a href="#cb3-2136" aria-hidden="true" tabindex="-1"></a><span class="in">            data = table_pdx,</span></span>
<span id="cb3-2137"><a href="#cb3-2137" aria-hidden="true" tabindex="-1"></a><span class="in">            cores = 4,</span></span>
<span id="cb3-2138"><a href="#cb3-2138" aria-hidden="true" tabindex="-1"></a><span class="in">            iter = 8000,</span></span>
<span id="cb3-2139"><a href="#cb3-2139" aria-hidden="true" tabindex="-1"></a><span class="in">            prior_intercept = rstanarm::normal(6.5, 3),</span></span>
<span id="cb3-2140"><a href="#cb3-2140" aria-hidden="true" tabindex="-1"></a><span class="in">            prior = rstanarm::normal(0, 4),</span></span>
<span id="cb3-2141"><a href="#cb3-2141" aria-hidden="true" tabindex="-1"></a><span class="in">            refresh = 0</span></span>
<span id="cb3-2142"><a href="#cb3-2142" aria-hidden="true" tabindex="-1"></a><span class="in">        )</span></span>
<span id="cb3-2143"><a href="#cb3-2143" aria-hidden="true" tabindex="-1"></a><span class="in">    }</span></span>
<span id="cb3-2144"><a href="#cb3-2144" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-2145"><a href="#cb3-2145" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2146"><a href="#cb3-2146" aria-hidden="true" tabindex="-1"></a><span class="in">plot_estimates_all &lt;- mapply(</span></span>
<span id="cb3-2147"><a href="#cb3-2147" aria-hidden="true" tabindex="-1"></a><span class="in">    function(table_pdx, fit_pdx){</span></span>
<span id="cb3-2148"><a href="#cb3-2148" aria-hidden="true" tabindex="-1"></a><span class="in">        </span></span>
<span id="cb3-2149"><a href="#cb3-2149" aria-hidden="true" tabindex="-1"></a><span class="in">        new_df_predict &lt;- biogrowleR::get_df_predict(table_pdx)</span></span>
<span id="cb3-2150"><a href="#cb3-2150" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2151"><a href="#cb3-2151" aria-hidden="true" tabindex="-1"></a><span class="in">        results_estimates_global &lt;- tidybayes::add_linpred_draws(</span></span>
<span id="cb3-2152"><a href="#cb3-2152" aria-hidden="true" tabindex="-1"></a><span class="in">            object = fit_pdx, </span></span>
<span id="cb3-2153"><a href="#cb3-2153" aria-hidden="true" tabindex="-1"></a><span class="in">            newdata = new_df_predict,</span></span>
<span id="cb3-2154"><a href="#cb3-2154" aria-hidden="true" tabindex="-1"></a><span class="in">            re_formula = NA,</span></span>
<span id="cb3-2155"><a href="#cb3-2155" aria-hidden="true" tabindex="-1"></a><span class="in">            value = "log10tf"</span></span>
<span id="cb3-2156"><a href="#cb3-2156" aria-hidden="true" tabindex="-1"></a><span class="in">        )</span></span>
<span id="cb3-2157"><a href="#cb3-2157" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2158"><a href="#cb3-2158" aria-hidden="true" tabindex="-1"></a><span class="in">        stats_from_data &lt;- biogrowleR::get_stats_from_data(table_pdx)</span></span>
<span id="cb3-2159"><a href="#cb3-2159" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2160"><a href="#cb3-2160" aria-hidden="true" tabindex="-1"></a><span class="in">        plot_estimates &lt;- biogrowleR::plot_posterior_estimates(</span></span>
<span id="cb3-2161"><a href="#cb3-2161" aria-hidden="true" tabindex="-1"></a><span class="in">            results_estimates_global, </span></span>
<span id="cb3-2162"><a href="#cb3-2162" aria-hidden="true" tabindex="-1"></a><span class="in">            stats_from_data, </span></span>
<span id="cb3-2163"><a href="#cb3-2163" aria-hidden="true" tabindex="-1"></a><span class="in">            base_size = 15</span></span>
<span id="cb3-2164"><a href="#cb3-2164" aria-hidden="true" tabindex="-1"></a><span class="in">        )</span></span>
<span id="cb3-2165"><a href="#cb3-2165" aria-hidden="true" tabindex="-1"></a><span class="in">        </span></span>
<span id="cb3-2166"><a href="#cb3-2166" aria-hidden="true" tabindex="-1"></a><span class="in">        combination_treatments &lt;- combn(</span></span>
<span id="cb3-2167"><a href="#cb3-2167" aria-hidden="true" tabindex="-1"></a><span class="in">            table_pdx$treatment %&gt;% unique %&gt;% as.vector, </span></span>
<span id="cb3-2168"><a href="#cb3-2168" aria-hidden="true" tabindex="-1"></a><span class="in">            m = 2</span></span>
<span id="cb3-2169"><a href="#cb3-2169" aria-hidden="true" tabindex="-1"></a><span class="in">        )</span></span>
<span id="cb3-2170"><a href="#cb3-2170" aria-hidden="true" tabindex="-1"></a><span class="in">        </span></span>
<span id="cb3-2171"><a href="#cb3-2171" aria-hidden="true" tabindex="-1"></a><span class="in">        # first get the pooled sds that will be used in the next function.</span></span>
<span id="cb3-2172"><a href="#cb3-2172" aria-hidden="true" tabindex="-1"></a><span class="in">        pooled_sds &lt;- biogrowleR::get_pooled_sds(df = table_pdx)</span></span>
<span id="cb3-2173"><a href="#cb3-2173" aria-hidden="true" tabindex="-1"></a><span class="in">        </span></span>
<span id="cb3-2174"><a href="#cb3-2174" aria-hidden="true" tabindex="-1"></a><span class="in">        # days correspond to the first and last days of treatment in the case </span></span>
<span id="cb3-2175"><a href="#cb3-2175" aria-hidden="true" tabindex="-1"></a><span class="in">        # of using linear regression to model the data</span></span>
<span id="cb3-2176"><a href="#cb3-2176" aria-hidden="true" tabindex="-1"></a><span class="in">        days &lt;- c(</span></span>
<span id="cb3-2177"><a href="#cb3-2177" aria-hidden="true" tabindex="-1"></a><span class="in">            min(results_estimates_global$days_after_treatment),</span></span>
<span id="cb3-2178"><a href="#cb3-2178" aria-hidden="true" tabindex="-1"></a><span class="in">            max(results_estimates_global$days_after_treatment)</span></span>
<span id="cb3-2179"><a href="#cb3-2179" aria-hidden="true" tabindex="-1"></a><span class="in">        )</span></span>
<span id="cb3-2180"><a href="#cb3-2180" aria-hidden="true" tabindex="-1"></a><span class="in">        </span></span>
<span id="cb3-2181"><a href="#cb3-2181" aria-hidden="true" tabindex="-1"></a><span class="in">        # combination_conditions is the same as used before for the frequentist</span></span>
<span id="cb3-2182"><a href="#cb3-2182" aria-hidden="true" tabindex="-1"></a><span class="in">        # approach.</span></span>
<span id="cb3-2183"><a href="#cb3-2183" aria-hidden="true" tabindex="-1"></a><span class="in">        results_differences &lt;- biogrowleR::get_slope_differences(</span></span>
<span id="cb3-2184"><a href="#cb3-2184" aria-hidden="true" tabindex="-1"></a><span class="in">            results_estimates = results_estimates_global,</span></span>
<span id="cb3-2185"><a href="#cb3-2185" aria-hidden="true" tabindex="-1"></a><span class="in">            combination_conditions = combination_treatments,</span></span>
<span id="cb3-2186"><a href="#cb3-2186" aria-hidden="true" tabindex="-1"></a><span class="in">            days = days,</span></span>
<span id="cb3-2187"><a href="#cb3-2187" aria-hidden="true" tabindex="-1"></a><span class="in">            pooled_sds = pooled_sds</span></span>
<span id="cb3-2188"><a href="#cb3-2188" aria-hidden="true" tabindex="-1"></a><span class="in">        )</span></span>
<span id="cb3-2189"><a href="#cb3-2189" aria-hidden="true" tabindex="-1"></a><span class="in">        </span></span>
<span id="cb3-2190"><a href="#cb3-2190" aria-hidden="true" tabindex="-1"></a><span class="in">        effect_size_plot &lt;- biogrowleR::plot_eff_size_dist(</span></span>
<span id="cb3-2191"><a href="#cb3-2191" aria-hidden="true" tabindex="-1"></a><span class="in">            results_differences, </span></span>
<span id="cb3-2192"><a href="#cb3-2192" aria-hidden="true" tabindex="-1"></a><span class="in">            days</span></span>
<span id="cb3-2193"><a href="#cb3-2193" aria-hidden="true" tabindex="-1"></a><span class="in">        ) +</span></span>
<span id="cb3-2194"><a href="#cb3-2194" aria-hidden="true" tabindex="-1"></a><span class="in">            ggplot2::theme_bw(base_size = 15)</span></span>
<span id="cb3-2195"><a href="#cb3-2195" aria-hidden="true" tabindex="-1"></a><span class="in">        </span></span>
<span id="cb3-2196"><a href="#cb3-2196" aria-hidden="true" tabindex="-1"></a><span class="in">        </span></span>
<span id="cb3-2197"><a href="#cb3-2197" aria-hidden="true" tabindex="-1"></a><span class="in">        list(</span></span>
<span id="cb3-2198"><a href="#cb3-2198" aria-hidden="true" tabindex="-1"></a><span class="in">            effect_size_plot = effect_size_plot,</span></span>
<span id="cb3-2199"><a href="#cb3-2199" aria-hidden="true" tabindex="-1"></a><span class="in">            results_differences = results_differences,</span></span>
<span id="cb3-2200"><a href="#cb3-2200" aria-hidden="true" tabindex="-1"></a><span class="in">            plot_estimates = plot_estimates</span></span>
<span id="cb3-2201"><a href="#cb3-2201" aria-hidden="true" tabindex="-1"></a><span class="in">        )</span></span>
<span id="cb3-2202"><a href="#cb3-2202" aria-hidden="true" tabindex="-1"></a><span class="in">    },</span></span>
<span id="cb3-2203"><a href="#cb3-2203" aria-hidden="true" tabindex="-1"></a><span class="in">    table_pdx = tables,</span></span>
<span id="cb3-2204"><a href="#cb3-2204" aria-hidden="true" tabindex="-1"></a><span class="in">    fit_pdx = all_fits,</span></span>
<span id="cb3-2205"><a href="#cb3-2205" aria-hidden="true" tabindex="-1"></a><span class="in">    USE.NAMES = TRUE,</span></span>
<span id="cb3-2206"><a href="#cb3-2206" aria-hidden="true" tabindex="-1"></a><span class="in">    SIMPLIFY = FALSE</span></span>
<span id="cb3-2207"><a href="#cb3-2207" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-2208"><a href="#cb3-2208" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2209"><a href="#cb3-2209" aria-hidden="true" tabindex="-1"></a><span class="in">saveRDS(</span></span>
<span id="cb3-2210"><a href="#cb3-2210" aria-hidden="true" tabindex="-1"></a><span class="in">    all_fits,</span></span>
<span id="cb3-2211"><a href="#cb3-2211" aria-hidden="true" tabindex="-1"></a><span class="in">    "../../results/rds_files/validation/all_fits.rds"</span></span>
<span id="cb3-2212"><a href="#cb3-2212" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-2213"><a href="#cb3-2213" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2214"><a href="#cb3-2214" aria-hidden="true" tabindex="-1"></a><span class="in">saveRDS(</span></span>
<span id="cb3-2215"><a href="#cb3-2215" aria-hidden="true" tabindex="-1"></a><span class="in">    plot_estimates_all,</span></span>
<span id="cb3-2216"><a href="#cb3-2216" aria-hidden="true" tabindex="-1"></a><span class="in">    "../../results/rds_files/validation/plot_estimates_all.rds"</span></span>
<span id="cb3-2217"><a href="#cb3-2217" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-2218"><a href="#cb3-2218" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-2219"><a href="#cb3-2219" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2220"><a href="#cb3-2220" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2221"><a href="#cb3-2221" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.height=8, fig.width=13}</span></span>
<span id="cb3-2222"><a href="#cb3-2222" aria-hidden="true" tabindex="-1"></a><span class="in">#| label: fig-pdx-estimates-ivis-together</span></span>
<span id="cb3-2223"><a href="#cb3-2223" aria-hidden="true" tabindex="-1"></a><span class="in">#| fig-cap: Estimates of the 6 PDXs together. Comparison between P4 vs CTRL.</span></span>
<span id="cb3-2224"><a href="#cb3-2224" aria-hidden="true" tabindex="-1"></a><span class="in">lapply(plot_estimates_all, \(x) x$results_differences) %&gt;%</span></span>
<span id="cb3-2225"><a href="#cb3-2225" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::bind_rows(.id = "pdx") %&gt;%</span></span>
<span id="cb3-2226"><a href="#cb3-2226" aria-hidden="true" tabindex="-1"></a><span class="in">    tidyr::separate(</span></span>
<span id="cb3-2227"><a href="#cb3-2227" aria-hidden="true" tabindex="-1"></a><span class="in">        pdx, </span></span>
<span id="cb3-2228"><a href="#cb3-2228" aria-hidden="true" tabindex="-1"></a><span class="in">        into = c("date", "pdx")</span></span>
<span id="cb3-2229"><a href="#cb3-2229" aria-hidden="true" tabindex="-1"></a><span class="in">    ) %&gt;%</span></span>
<span id="cb3-2230"><a href="#cb3-2230" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::filter(pdx %in% c("METS15", "T111", "T113", "T110", "T109", "T105")) %&gt;%</span></span>
<span id="cb3-2231"><a href="#cb3-2231" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::mutate(pdx = dplyr::case_when(</span></span>
<span id="cb3-2232"><a href="#cb3-2232" aria-hidden="true" tabindex="-1"></a><span class="in">        pdx == "METS15" ~ "METS15 (**)",</span></span>
<span id="cb3-2233"><a href="#cb3-2233" aria-hidden="true" tabindex="-1"></a><span class="in">        pdx == "T105" ~ "T105 (*)",</span></span>
<span id="cb3-2234"><a href="#cb3-2234" aria-hidden="true" tabindex="-1"></a><span class="in">        pdx == "T109" ~ "T109 (n.s.)",</span></span>
<span id="cb3-2235"><a href="#cb3-2235" aria-hidden="true" tabindex="-1"></a><span class="in">        pdx == "T110" ~ "T110 (n.s.)",</span></span>
<span id="cb3-2236"><a href="#cb3-2236" aria-hidden="true" tabindex="-1"></a><span class="in">        pdx == "T111" ~ "T111 (***)",</span></span>
<span id="cb3-2237"><a href="#cb3-2237" aria-hidden="true" tabindex="-1"></a><span class="in">        pdx == "T113" ~ "T113 (****)"</span></span>
<span id="cb3-2238"><a href="#cb3-2238" aria-hidden="true" tabindex="-1"></a><span class="in">    )) %&gt;%</span></span>
<span id="cb3-2239"><a href="#cb3-2239" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::ggplot(aes(x = difference, fill = pdx)) +</span></span>
<span id="cb3-2240"><a href="#cb3-2240" aria-hidden="true" tabindex="-1"></a><span class="in">    ggdist::stat_halfeye(alpha = 0.5) +</span></span>
<span id="cb3-2241"><a href="#cb3-2241" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_vline(xintercept = 0, linetype = "dashed") +</span></span>
<span id="cb3-2242"><a href="#cb3-2242" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::labs(</span></span>
<span id="cb3-2243"><a href="#cb3-2243" aria-hidden="true" tabindex="-1"></a><span class="in">        x = "Effect size",</span></span>
<span id="cb3-2244"><a href="#cb3-2244" aria-hidden="true" tabindex="-1"></a><span class="in">        y = "Density",</span></span>
<span id="cb3-2245"><a href="#cb3-2245" aria-hidden="true" tabindex="-1"></a><span class="in">        fill = "PDX",</span></span>
<span id="cb3-2246"><a href="#cb3-2246" aria-hidden="true" tabindex="-1"></a><span class="in">        title = paste0(</span></span>
<span id="cb3-2247"><a href="#cb3-2247" aria-hidden="true" tabindex="-1"></a><span class="in">            "In vivo growth analysis P4 vs CTRL\n",</span></span>
<span id="cb3-2248"><a href="#cb3-2248" aria-hidden="true" tabindex="-1"></a><span class="in">            "Distributions of the effect sizes"</span></span>
<span id="cb3-2249"><a href="#cb3-2249" aria-hidden="true" tabindex="-1"></a><span class="in">        )</span></span>
<span id="cb3-2250"><a href="#cb3-2250" aria-hidden="true" tabindex="-1"></a><span class="in">    ) +</span></span>
<span id="cb3-2251"><a href="#cb3-2251" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::scale_fill_viridis_d(option = "H") + </span></span>
<span id="cb3-2252"><a href="#cb3-2252" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme_bw(base_size = 30) + </span></span>
<span id="cb3-2253"><a href="#cb3-2253" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme(</span></span>
<span id="cb3-2254"><a href="#cb3-2254" aria-hidden="true" tabindex="-1"></a><span class="in">        legend.position = c(0.8, 0.6), </span></span>
<span id="cb3-2255"><a href="#cb3-2255" aria-hidden="true" tabindex="-1"></a><span class="in">        legend.background = element_blank()</span></span>
<span id="cb3-2256"><a href="#cb3-2256" aria-hidden="true" tabindex="-1"></a><span class="in">    )</span></span>
<span id="cb3-2257"><a href="#cb3-2257" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-2258"><a href="#cb3-2258" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2259"><a href="#cb3-2259" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2260"><a href="#cb3-2260" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=12, fig.height=8}</span></span>
<span id="cb3-2261"><a href="#cb3-2261" aria-hidden="true" tabindex="-1"></a><span class="in">#| label: fig-pdx-curves-ivis-together</span></span>
<span id="cb3-2262"><a href="#cb3-2262" aria-hidden="true" tabindex="-1"></a><span class="in">#| fig-cap: Average growth curves for each PDX individually with their </span></span>
<span id="cb3-2263"><a href="#cb3-2263" aria-hidden="true" tabindex="-1"></a><span class="in">#|     estimates. The credible intervals corresponds to the 25th and 50th CIs.</span></span>
<span id="cb3-2264"><a href="#cb3-2264" aria-hidden="true" tabindex="-1"></a><span class="in">pdxs_responders &lt;- c("T111", "T113", "METS15", "T105", "T110", "T109")</span></span>
<span id="cb3-2265"><a href="#cb3-2265" aria-hidden="true" tabindex="-1"></a><span class="in">responder_non_responder &lt;- c(</span></span>
<span id="cb3-2266"><a href="#cb3-2266" aria-hidden="true" tabindex="-1"></a><span class="in">    "Responder", </span></span>
<span id="cb3-2267"><a href="#cb3-2267" aria-hidden="true" tabindex="-1"></a><span class="in">    "Responder",</span></span>
<span id="cb3-2268"><a href="#cb3-2268" aria-hidden="true" tabindex="-1"></a><span class="in">    "Partial responder",</span></span>
<span id="cb3-2269"><a href="#cb3-2269" aria-hidden="true" tabindex="-1"></a><span class="in">    "Partial responder",</span></span>
<span id="cb3-2270"><a href="#cb3-2270" aria-hidden="true" tabindex="-1"></a><span class="in">    "Non-responder",</span></span>
<span id="cb3-2271"><a href="#cb3-2271" aria-hidden="true" tabindex="-1"></a><span class="in">    "Non-responder"</span></span>
<span id="cb3-2272"><a href="#cb3-2272" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-2273"><a href="#cb3-2273" aria-hidden="true" tabindex="-1"></a><span class="in">plot_estimates_responders &lt;- lapply(</span></span>
<span id="cb3-2274"><a href="#cb3-2274" aria-hidden="true" tabindex="-1"></a><span class="in">    plot_estimates_all[paste0("20190801_", pdxs_responders)],</span></span>
<span id="cb3-2275"><a href="#cb3-2275" aria-hidden="true" tabindex="-1"></a><span class="in">    \(x) x$plot_estimates</span></span>
<span id="cb3-2276"><a href="#cb3-2276" aria-hidden="true" tabindex="-1"></a><span class="in">) %&gt;% </span></span>
<span id="cb3-2277"><a href="#cb3-2277" aria-hidden="true" tabindex="-1"></a><span class="in">    mapply(</span></span>
<span id="cb3-2278"><a href="#cb3-2278" aria-hidden="true" tabindex="-1"></a><span class="in">        function(p, pdx, responder_non_responder, i){</span></span>
<span id="cb3-2279"><a href="#cb3-2279" aria-hidden="true" tabindex="-1"></a><span class="in">            </span></span>
<span id="cb3-2280"><a href="#cb3-2280" aria-hidden="true" tabindex="-1"></a><span class="in">            values_col &lt;- viridis::turbo(n = 2)</span></span>
<span id="cb3-2281"><a href="#cb3-2281" aria-hidden="true" tabindex="-1"></a><span class="in">            </span></span>
<span id="cb3-2282"><a href="#cb3-2282" aria-hidden="true" tabindex="-1"></a><span class="in">            if (i &gt; 1){</span></span>
<span id="cb3-2283"><a href="#cb3-2283" aria-hidden="true" tabindex="-1"></a><span class="in">                p &lt;- p + ggplot2::ylab("")</span></span>
<span id="cb3-2284"><a href="#cb3-2284" aria-hidden="true" tabindex="-1"></a><span class="in">            }</span></span>
<span id="cb3-2285"><a href="#cb3-2285" aria-hidden="true" tabindex="-1"></a><span class="in">            </span></span>
<span id="cb3-2286"><a href="#cb3-2286" aria-hidden="true" tabindex="-1"></a><span class="in">            p + </span></span>
<span id="cb3-2287"><a href="#cb3-2287" aria-hidden="true" tabindex="-1"></a><span class="in">                ggplot2::labs(</span></span>
<span id="cb3-2288"><a href="#cb3-2288" aria-hidden="true" tabindex="-1"></a><span class="in">                    title = pdx, </span></span>
<span id="cb3-2289"><a href="#cb3-2289" aria-hidden="true" tabindex="-1"></a><span class="in">                    subtitle = responder_non_responder,</span></span>
<span id="cb3-2290"><a href="#cb3-2290" aria-hidden="true" tabindex="-1"></a><span class="in">                    fill = "Treatment", </span></span>
<span id="cb3-2291"><a href="#cb3-2291" aria-hidden="true" tabindex="-1"></a><span class="in">                    linetype = "Treatment",</span></span>
<span id="cb3-2292"><a href="#cb3-2292" aria-hidden="true" tabindex="-1"></a><span class="in">                    color = "Treatment"</span></span>
<span id="cb3-2293"><a href="#cb3-2293" aria-hidden="true" tabindex="-1"></a><span class="in">                ) + </span></span>
<span id="cb3-2294"><a href="#cb3-2294" aria-hidden="true" tabindex="-1"></a><span class="in">                ggplot2::scale_color_manual(</span></span>
<span id="cb3-2295"><a href="#cb3-2295" aria-hidden="true" tabindex="-1"></a><span class="in">                    labels = c("CTRL", "P4"), </span></span>
<span id="cb3-2296"><a href="#cb3-2296" aria-hidden="true" tabindex="-1"></a><span class="in">                    values = values_col</span></span>
<span id="cb3-2297"><a href="#cb3-2297" aria-hidden="true" tabindex="-1"></a><span class="in">                ) +</span></span>
<span id="cb3-2298"><a href="#cb3-2298" aria-hidden="true" tabindex="-1"></a><span class="in">                ggplot2::scale_fill_manual(</span></span>
<span id="cb3-2299"><a href="#cb3-2299" aria-hidden="true" tabindex="-1"></a><span class="in">                    labels = c("CTRL", "P4"), </span></span>
<span id="cb3-2300"><a href="#cb3-2300" aria-hidden="true" tabindex="-1"></a><span class="in">                    values = values_col</span></span>
<span id="cb3-2301"><a href="#cb3-2301" aria-hidden="true" tabindex="-1"></a><span class="in">                ) +</span></span>
<span id="cb3-2302"><a href="#cb3-2302" aria-hidden="true" tabindex="-1"></a><span class="in">                ggplot2::scale_linetype_manual(</span></span>
<span id="cb3-2303"><a href="#cb3-2303" aria-hidden="true" tabindex="-1"></a><span class="in">                    labels = c("CTRL", "P4"), </span></span>
<span id="cb3-2304"><a href="#cb3-2304" aria-hidden="true" tabindex="-1"></a><span class="in">                    values = c(2, 4)</span></span>
<span id="cb3-2305"><a href="#cb3-2305" aria-hidden="true" tabindex="-1"></a><span class="in">                ) +</span></span>
<span id="cb3-2306"><a href="#cb3-2306" aria-hidden="true" tabindex="-1"></a><span class="in">                ggplot2::theme(</span></span>
<span id="cb3-2307"><a href="#cb3-2307" aria-hidden="true" tabindex="-1"></a><span class="in">                    legend.position = c(0.3, 0.82)</span></span>
<span id="cb3-2308"><a href="#cb3-2308" aria-hidden="true" tabindex="-1"></a><span class="in">                )</span></span>
<span id="cb3-2309"><a href="#cb3-2309" aria-hidden="true" tabindex="-1"></a><span class="in">        },</span></span>
<span id="cb3-2310"><a href="#cb3-2310" aria-hidden="true" tabindex="-1"></a><span class="in">        p = .,</span></span>
<span id="cb3-2311"><a href="#cb3-2311" aria-hidden="true" tabindex="-1"></a><span class="in">        pdx = pdxs_responders,</span></span>
<span id="cb3-2312"><a href="#cb3-2312" aria-hidden="true" tabindex="-1"></a><span class="in">        responder_non_responder = responder_non_responder,</span></span>
<span id="cb3-2313"><a href="#cb3-2313" aria-hidden="true" tabindex="-1"></a><span class="in">        i = 1:length(pdxs_responders),</span></span>
<span id="cb3-2314"><a href="#cb3-2314" aria-hidden="true" tabindex="-1"></a><span class="in">        USE.NAMES = TRUE,</span></span>
<span id="cb3-2315"><a href="#cb3-2315" aria-hidden="true" tabindex="-1"></a><span class="in">        SIMPLIFY = FALSE</span></span>
<span id="cb3-2316"><a href="#cb3-2316" aria-hidden="true" tabindex="-1"></a><span class="in">    )</span></span>
<span id="cb3-2317"><a href="#cb3-2317" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2318"><a href="#cb3-2318" aria-hidden="true" tabindex="-1"></a><span class="in">cowplot::plot_grid(</span></span>
<span id="cb3-2319"><a href="#cb3-2319" aria-hidden="true" tabindex="-1"></a><span class="in">    plotlist = plot_estimates_responders, </span></span>
<span id="cb3-2320"><a href="#cb3-2320" aria-hidden="true" tabindex="-1"></a><span class="in">    nrow = 2</span></span>
<span id="cb3-2321"><a href="#cb3-2321" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-2322"><a href="#cb3-2322" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-2323"><a href="#cb3-2323" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2324"><a href="#cb3-2324" aria-hidden="true" tabindex="-1"></a><span class="fu">## Normal samples</span></span>
<span id="cb3-2325"><a href="#cb3-2325" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2326"><a href="#cb3-2326" aria-hidden="true" tabindex="-1"></a>We now further validate the molecular landscape on the normal samples</span>
<span id="cb3-2327"><a href="#cb3-2327" aria-hidden="true" tabindex="-1"></a>that were sequenced using the same pipeline as the SCANB data. These</span>
<span id="cb3-2328"><a href="#cb3-2328" aria-hidden="true" tabindex="-1"></a>are samples obtained from reduction mammoplasty of swiss women and were</span>
<span id="cb3-2329"><a href="#cb3-2329" aria-hidden="true" tabindex="-1"></a>sent to Sweden for the tissue processing and sequecing. The hypotheses </span>
<span id="cb3-2330"><a href="#cb3-2330" aria-hidden="true" tabindex="-1"></a>here is that the samples will be embedded in the region of the normal</span>
<span id="cb3-2331"><a href="#cb3-2331" aria-hidden="true" tabindex="-1"></a>like BC samples.</span>
<span id="cb3-2332"><a href="#cb3-2332" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2333"><a href="#cb3-2333" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb3-2334"><a href="#cb3-2334" aria-hidden="true" tabindex="-1"></a><span class="in"># load the data and perform the normalization + embedding</span></span>
<span id="cb3-2335"><a href="#cb3-2335" aria-hidden="true" tabindex="-1"></a><span class="in">load("../../data/scanb_2022/Normal.66.mymatrix.Rdata")</span></span>
<span id="cb3-2336"><a href="#cb3-2336" aria-hidden="true" tabindex="-1"></a><span class="in">load("../../data/scanb_2022/Gene.ID.ann.Rdata")</span></span>
<span id="cb3-2337"><a href="#cb3-2337" aria-hidden="true" tabindex="-1"></a><span class="in">normal_swiss &lt;- Normal.66.mymatrix</span></span>
<span id="cb3-2338"><a href="#cb3-2338" aria-hidden="true" tabindex="-1"></a><span class="in">gene_id_ann &lt;- Gene.ID.ann</span></span>
<span id="cb3-2339"><a href="#cb3-2339" aria-hidden="true" tabindex="-1"></a><span class="in">rownames(normal_swiss) &lt;- gene_id_ann[rownames(normal_swiss), "Gene.Name"]</span></span>
<span id="cb3-2340"><a href="#cb3-2340" aria-hidden="true" tabindex="-1"></a><span class="in">normal_swiss &lt;- normal_swiss[!duplicated(rownames(normal_swiss)), ]</span></span>
<span id="cb3-2341"><a href="#cb3-2341" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2342"><a href="#cb3-2342" aria-hidden="true" tabindex="-1"></a><span class="in">normal_swiss &lt;- SummarizedExperiment::SummarizedExperiment(</span></span>
<span id="cb3-2343"><a href="#cb3-2343" aria-hidden="true" tabindex="-1"></a><span class="in">    assays = list(</span></span>
<span id="cb3-2344"><a href="#cb3-2344" aria-hidden="true" tabindex="-1"></a><span class="in">            fpkm = normal_swiss,</span></span>
<span id="cb3-2345"><a href="#cb3-2345" aria-hidden="true" tabindex="-1"></a><span class="in">            log2fpkm = log2(normal_swiss + 1)</span></span>
<span id="cb3-2346"><a href="#cb3-2346" aria-hidden="true" tabindex="-1"></a><span class="in">        )</span></span>
<span id="cb3-2347"><a href="#cb3-2347" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-2348"><a href="#cb3-2348" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2349"><a href="#cb3-2349" aria-hidden="true" tabindex="-1"></a><span class="in">normal_swiss_normalized &lt;- get_final_ranking_values(</span></span>
<span id="cb3-2350"><a href="#cb3-2350" aria-hidden="true" tabindex="-1"></a><span class="in">    sum_exp = normal_swiss,</span></span>
<span id="cb3-2351"><a href="#cb3-2351" aria-hidden="true" tabindex="-1"></a><span class="in">    assay_to_use = "log2fpkm",</span></span>
<span id="cb3-2352"><a href="#cb3-2352" aria-hidden="true" tabindex="-1"></a><span class="in">    stable_genes = stable_genes,</span></span>
<span id="cb3-2353"><a href="#cb3-2353" aria-hidden="true" tabindex="-1"></a><span class="in">    most_variable_genes = setdiff(rownames(pca_fit$loadings), stable_genes)</span></span>
<span id="cb3-2354"><a href="#cb3-2354" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-2355"><a href="#cb3-2355" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2356"><a href="#cb3-2356" aria-hidden="true" tabindex="-1"></a><span class="in"># calculate the embedding</span></span>
<span id="cb3-2357"><a href="#cb3-2357" aria-hidden="true" tabindex="-1"></a><span class="in">normal_swiss_df_pca &lt;- get_pca_coordinates(normal_swiss_normalized, pca_fit) %&gt;%</span></span>
<span id="cb3-2358"><a href="#cb3-2358" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::bind_cols(</span></span>
<span id="cb3-2359"><a href="#cb3-2359" aria-hidden="true" tabindex="-1"></a><span class="in">        ., </span></span>
<span id="cb3-2360"><a href="#cb3-2360" aria-hidden="true" tabindex="-1"></a><span class="in">        data.frame(cohort = rep("normal_swiss", ncol(normal_swiss_normalized)))</span></span>
<span id="cb3-2361"><a href="#cb3-2361" aria-hidden="true" tabindex="-1"></a><span class="in">    ) %&gt;% </span></span>
<span id="cb3-2362"><a href="#cb3-2362" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::bind_rows(., df_pca_coordinates)</span></span>
<span id="cb3-2363"><a href="#cb3-2363" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2364"><a href="#cb3-2364" aria-hidden="true" tabindex="-1"></a><span class="in"># save all the results</span></span>
<span id="cb3-2365"><a href="#cb3-2365" aria-hidden="true" tabindex="-1"></a><span class="in">saveRDS(</span></span>
<span id="cb3-2366"><a href="#cb3-2366" aria-hidden="true" tabindex="-1"></a><span class="in">    normal_swiss,</span></span>
<span id="cb3-2367"><a href="#cb3-2367" aria-hidden="true" tabindex="-1"></a><span class="in">    "../../results/rds_files/validation/normal_swiss.rds"</span></span>
<span id="cb3-2368"><a href="#cb3-2368" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-2369"><a href="#cb3-2369" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2370"><a href="#cb3-2370" aria-hidden="true" tabindex="-1"></a><span class="in">saveRDS(</span></span>
<span id="cb3-2371"><a href="#cb3-2371" aria-hidden="true" tabindex="-1"></a><span class="in">    normal_swiss_df_pca %&gt;% data.frame,</span></span>
<span id="cb3-2372"><a href="#cb3-2372" aria-hidden="true" tabindex="-1"></a><span class="in">    "../../results/rds_files/validation/normal_swiss_df_pca.rds"</span></span>
<span id="cb3-2373"><a href="#cb3-2373" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-2374"><a href="#cb3-2374" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2375"><a href="#cb3-2375" aria-hidden="true" tabindex="-1"></a><span class="in">saveRDS(</span></span>
<span id="cb3-2376"><a href="#cb3-2376" aria-hidden="true" tabindex="-1"></a><span class="in">    normal_swiss_normalized,</span></span>
<span id="cb3-2377"><a href="#cb3-2377" aria-hidden="true" tabindex="-1"></a><span class="in">    "../../results/rds_files/validation/normal_swiss_normalized.rds"</span></span>
<span id="cb3-2378"><a href="#cb3-2378" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-2379"><a href="#cb3-2379" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-2380"><a href="#cb3-2380" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2381"><a href="#cb3-2381" aria-hidden="true" tabindex="-1"></a>When checking the total number of genes missing, there are only </span>
<span id="cb3-2382"><a href="#cb3-2382" aria-hidden="true" tabindex="-1"></a><span class="in">`r nrow(datasets_normalized$tcga) - nrow(normal_swiss_normalized)`</span> genes</span>
<span id="cb3-2383"><a href="#cb3-2383" aria-hidden="true" tabindex="-1"></a>missing, i.e., no genes were missing. This makes sense as these samples were</span>
<span id="cb3-2384"><a href="#cb3-2384" aria-hidden="true" tabindex="-1"></a>processed in the same way as the ones from SCANB.</span>
<span id="cb3-2385"><a href="#cb3-2385" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2388"><a href="#cb3-2388" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-2389"><a href="#cb3-2389" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-normal-cohort</span></span>
<span id="cb3-2390"><a href="#cb3-2390" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Biplot of first two PCA components from Normal Swiss Cohort, </span></span>
<span id="cb3-2391"><a href="#cb3-2391" aria-hidden="true" tabindex="-1"></a><span class="co">#|     TCGA, SCANB and </span></span>
<span id="cb3-2392"><a href="#cb3-2392" aria-hidden="true" tabindex="-1"></a><span class="co">#|     METABRIC. The normal cohort is highlighted in the plot and</span></span>
<span id="cb3-2393"><a href="#cb3-2393" aria-hidden="true" tabindex="-1"></a><span class="co">#|     has bigger dot size.</span></span>
<span id="cb3-2394"><a href="#cb3-2394" aria-hidden="true" tabindex="-1"></a>normal_swiss_df_pca <span class="sc">%&gt;%</span> </span>
<span id="cb3-2395"><a href="#cb3-2395" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(</span>
<span id="cb3-2396"><a href="#cb3-2396" aria-hidden="true" tabindex="-1"></a>        <span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> cohort)</span>
<span id="cb3-2397"><a href="#cb3-2397" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb3-2398"><a href="#cb3-2398" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">alpha =</span> cohort, <span class="at">size =</span> cohort)) <span class="sc">+</span></span>
<span id="cb3-2399"><a href="#cb3-2399" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_size_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(</span>
<span id="cb3-2400"><a href="#cb3-2400" aria-hidden="true" tabindex="-1"></a>        <span class="st">"normal_swiss"</span> <span class="ot">=</span> <span class="dv">2</span>, </span>
<span id="cb3-2401"><a href="#cb3-2401" aria-hidden="true" tabindex="-1"></a>        <span class="st">"tcga"</span> <span class="ot">=</span> <span class="dv">1</span>,</span>
<span id="cb3-2402"><a href="#cb3-2402" aria-hidden="true" tabindex="-1"></a>        <span class="st">"scanb"</span> <span class="ot">=</span> <span class="dv">1</span>,</span>
<span id="cb3-2403"><a href="#cb3-2403" aria-hidden="true" tabindex="-1"></a>        <span class="st">"metabric"</span> <span class="ot">=</span> <span class="dv">1</span></span>
<span id="cb3-2404"><a href="#cb3-2404" aria-hidden="true" tabindex="-1"></a>    ), <span class="at">guide =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb3-2405"><a href="#cb3-2405" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_alpha_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(</span>
<span id="cb3-2406"><a href="#cb3-2406" aria-hidden="true" tabindex="-1"></a>        <span class="st">"normal_swiss"</span> <span class="ot">=</span> <span class="dv">1</span>, </span>
<span id="cb3-2407"><a href="#cb3-2407" aria-hidden="true" tabindex="-1"></a>        <span class="st">"tcga"</span> <span class="ot">=</span> <span class="fl">0.1</span>,</span>
<span id="cb3-2408"><a href="#cb3-2408" aria-hidden="true" tabindex="-1"></a>        <span class="st">"scanb"</span> <span class="ot">=</span> <span class="fl">0.1</span>,</span>
<span id="cb3-2409"><a href="#cb3-2409" aria-hidden="true" tabindex="-1"></a>        <span class="st">"metabric"</span> <span class="ot">=</span> <span class="fl">0.1</span></span>
<span id="cb3-2410"><a href="#cb3-2410" aria-hidden="true" tabindex="-1"></a>    ), <span class="at">guide =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb3-2411"><a href="#cb3-2411" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb3-2412"><a href="#cb3-2412" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">"Cohort"</span>,</span>
<span id="cb3-2413"><a href="#cb3-2413" aria-hidden="true" tabindex="-1"></a>        <span class="at">alpha =</span> <span class="st">"Cohort"</span>,</span>
<span id="cb3-2414"><a href="#cb3-2414" aria-hidden="true" tabindex="-1"></a>        <span class="at">size =</span> <span class="st">"Cohort"</span>,</span>
<span id="cb3-2415"><a href="#cb3-2415" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="fu">paste0</span>(</span>
<span id="cb3-2416"><a href="#cb3-2416" aria-hidden="true" tabindex="-1"></a>            <span class="st">"First two components of Normal Swiss cohort overlayed on all</span><span class="sc">\n</span><span class="st">the three "</span>,</span>
<span id="cb3-2417"><a href="#cb3-2417" aria-hidden="true" tabindex="-1"></a>            <span class="st">"big cohorts in the the molecular landscape"</span></span>
<span id="cb3-2418"><a href="#cb3-2418" aria-hidden="true" tabindex="-1"></a>        ),</span>
<span id="cb3-2419"><a href="#cb3-2419" aria-hidden="true" tabindex="-1"></a>        <span class="at">subtitle =</span> <span class="st">"All samples from TCGA, METABRIC and SCANB are plotted"</span></span>
<span id="cb3-2420"><a href="#cb3-2420" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb3-2421"><a href="#cb3-2421" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">14</span>) <span class="sc">+</span></span>
<span id="cb3-2422"><a href="#cb3-2422" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span></span>
<span id="cb3-2423"><a href="#cb3-2423" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb3-2424"><a href="#cb3-2424" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-2425"><a href="#cb3-2425" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2426"><a href="#cb3-2426" aria-hidden="true" tabindex="-1"></a>As previously, @fig-pca-normal-cohort shows the biplot of PC3 and PC4, </span>
<span id="cb3-2427"><a href="#cb3-2427" aria-hidden="true" tabindex="-1"></a>highlighting the well mixing of the cohorts. As expected the samples are </span>
<span id="cb3-2428"><a href="#cb3-2428" aria-hidden="true" tabindex="-1"></a>overall in the luminal B region, since these are tumor cells in a </span>
<span id="cb3-2429"><a href="#cb3-2429" aria-hidden="true" tabindex="-1"></a>proliferative state.</span>
<span id="cb3-2430"><a href="#cb3-2430" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2431"><a href="#cb3-2431" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=9, fig.height=6}</span></span>
<span id="cb3-2432"><a href="#cb3-2432" aria-hidden="true" tabindex="-1"></a><span class="in">#| label: fig-pca-normal-cohort</span></span>
<span id="cb3-2433"><a href="#cb3-2433" aria-hidden="true" tabindex="-1"></a><span class="in">#| fig-cap: Embedding of all samples from TCGA, SCNAB and METABRIC including</span></span>
<span id="cb3-2434"><a href="#cb3-2434" aria-hidden="true" tabindex="-1"></a><span class="in">#|     the Normal samples on top. </span></span>
<span id="cb3-2435"><a href="#cb3-2435" aria-hidden="true" tabindex="-1"></a><span class="in">normal_swiss_df_pca %&gt;%</span></span>
<span id="cb3-2436"><a href="#cb3-2436" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::ggplot(</span></span>
<span id="cb3-2437"><a href="#cb3-2437" aria-hidden="true" tabindex="-1"></a><span class="in">        aes(x = PC3, y = PC4, color = cohort)</span></span>
<span id="cb3-2438"><a href="#cb3-2438" aria-hidden="true" tabindex="-1"></a><span class="in">    ) +</span></span>
<span id="cb3-2439"><a href="#cb3-2439" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_point(aes(alpha = cohort, size = cohort)) +</span></span>
<span id="cb3-2440"><a href="#cb3-2440" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::scale_size_manual(values = c(</span></span>
<span id="cb3-2441"><a href="#cb3-2441" aria-hidden="true" tabindex="-1"></a><span class="in">        "normal_swiss" = 3, </span></span>
<span id="cb3-2442"><a href="#cb3-2442" aria-hidden="true" tabindex="-1"></a><span class="in">        "tcga" = 1,</span></span>
<span id="cb3-2443"><a href="#cb3-2443" aria-hidden="true" tabindex="-1"></a><span class="in">        "scanb" = 1,</span></span>
<span id="cb3-2444"><a href="#cb3-2444" aria-hidden="true" tabindex="-1"></a><span class="in">        "metabric" = 1</span></span>
<span id="cb3-2445"><a href="#cb3-2445" aria-hidden="true" tabindex="-1"></a><span class="in">    ), guide = "none") +</span></span>
<span id="cb3-2446"><a href="#cb3-2446" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::scale_alpha_manual(values = c(</span></span>
<span id="cb3-2447"><a href="#cb3-2447" aria-hidden="true" tabindex="-1"></a><span class="in">        "normal_swiss" = 1, </span></span>
<span id="cb3-2448"><a href="#cb3-2448" aria-hidden="true" tabindex="-1"></a><span class="in">        "tcga" = 0.1,</span></span>
<span id="cb3-2449"><a href="#cb3-2449" aria-hidden="true" tabindex="-1"></a><span class="in">        "scanb" = 0.1,</span></span>
<span id="cb3-2450"><a href="#cb3-2450" aria-hidden="true" tabindex="-1"></a><span class="in">        "metabric" = 0.1</span></span>
<span id="cb3-2451"><a href="#cb3-2451" aria-hidden="true" tabindex="-1"></a><span class="in">    ), guide = "none") +</span></span>
<span id="cb3-2452"><a href="#cb3-2452" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::labs(</span></span>
<span id="cb3-2453"><a href="#cb3-2453" aria-hidden="true" tabindex="-1"></a><span class="in">        color = "Cohort",</span></span>
<span id="cb3-2454"><a href="#cb3-2454" aria-hidden="true" tabindex="-1"></a><span class="in">        alpha = "Cohort",</span></span>
<span id="cb3-2455"><a href="#cb3-2455" aria-hidden="true" tabindex="-1"></a><span class="in">        size = "Cohort",</span></span>
<span id="cb3-2456"><a href="#cb3-2456" aria-hidden="true" tabindex="-1"></a><span class="in">        title = paste0(</span></span>
<span id="cb3-2457"><a href="#cb3-2457" aria-hidden="true" tabindex="-1"></a><span class="in">            "Normal Swiss cohort embedding"</span></span>
<span id="cb3-2458"><a href="#cb3-2458" aria-hidden="true" tabindex="-1"></a><span class="in">        ),</span></span>
<span id="cb3-2459"><a href="#cb3-2459" aria-hidden="true" tabindex="-1"></a><span class="in">        subtitle = "All samples from TCGA, SCANB and METABRIC are plotted"</span></span>
<span id="cb3-2460"><a href="#cb3-2460" aria-hidden="true" tabindex="-1"></a><span class="in">    ) + </span></span>
<span id="cb3-2461"><a href="#cb3-2461" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme_bw(base_size = 18) +</span></span>
<span id="cb3-2462"><a href="#cb3-2462" aria-hidden="true" tabindex="-1"></a><span class="in">    change_plot_aes_point() +</span></span>
<span id="cb3-2463"><a href="#cb3-2463" aria-hidden="true" tabindex="-1"></a><span class="in">    change_guides_point()</span></span>
<span id="cb3-2464"><a href="#cb3-2464" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-2465"><a href="#cb3-2465" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2466"><a href="#cb3-2466" aria-hidden="true" tabindex="-1"></a><span class="in">```{r fig.width=7, fig.height=4}</span></span>
<span id="cb3-2467"><a href="#cb3-2467" aria-hidden="true" tabindex="-1"></a><span class="in">#| label: fig-pca-normal-pam50</span></span>
<span id="cb3-2468"><a href="#cb3-2468" aria-hidden="true" tabindex="-1"></a><span class="in">#| fig-cap: Embedding of Normal Swiss cohort with the </span></span>
<span id="cb3-2469"><a href="#cb3-2469" aria-hidden="true" tabindex="-1"></a><span class="in">#|     other cohorts colored by the PAM50. </span></span>
<span id="cb3-2470"><a href="#cb3-2470" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2471"><a href="#cb3-2471" aria-hidden="true" tabindex="-1"></a><span class="in">p &lt;- get_base_plot(</span></span>
<span id="cb3-2472"><a href="#cb3-2472" aria-hidden="true" tabindex="-1"></a><span class="in">    df_pca_coordinates %&gt;%</span></span>
<span id="cb3-2473"><a href="#cb3-2473" aria-hidden="true" tabindex="-1"></a><span class="in">        dplyr::filter(pam50 != "claudin-low")</span></span>
<span id="cb3-2474"><a href="#cb3-2474" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-2475"><a href="#cb3-2475" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2476"><a href="#cb3-2476" aria-hidden="true" tabindex="-1"></a><span class="in">normal_df_pca_sub &lt;- normal_swiss_df_pca %&gt;%</span></span>
<span id="cb3-2477"><a href="#cb3-2477" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::filter(cohort == "normal_swiss") %&gt;%</span></span>
<span id="cb3-2478"><a href="#cb3-2478" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::mutate(pam50 = "nc")</span></span>
<span id="cb3-2479"><a href="#cb3-2479" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2480"><a href="#cb3-2480" aria-hidden="true" tabindex="-1"></a><span class="in">normal_plot &lt;- p +</span></span>
<span id="cb3-2481"><a href="#cb3-2481" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_point(</span></span>
<span id="cb3-2482"><a href="#cb3-2482" aria-hidden="true" tabindex="-1"></a><span class="in">        normal_df_pca_sub,</span></span>
<span id="cb3-2483"><a href="#cb3-2483" aria-hidden="true" tabindex="-1"></a><span class="in">        mapping = aes(x = PC3, y = PC4),</span></span>
<span id="cb3-2484"><a href="#cb3-2484" aria-hidden="true" tabindex="-1"></a><span class="in">        size = 3</span></span>
<span id="cb3-2485"><a href="#cb3-2485" aria-hidden="true" tabindex="-1"></a><span class="in">    ) + </span></span>
<span id="cb3-2486"><a href="#cb3-2486" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::labs(</span></span>
<span id="cb3-2487"><a href="#cb3-2487" aria-hidden="true" tabindex="-1"></a><span class="in">        #title = paste0(</span></span>
<span id="cb3-2488"><a href="#cb3-2488" aria-hidden="true" tabindex="-1"></a><span class="in">        #    "Healthy Swiss Cohort"</span></span>
<span id="cb3-2489"><a href="#cb3-2489" aria-hidden="true" tabindex="-1"></a><span class="in">        #),</span></span>
<span id="cb3-2490"><a href="#cb3-2490" aria-hidden="true" tabindex="-1"></a><span class="in">        #subtitle = "All samples from TCGA, SCANB and METABRIC are plotted",</span></span>
<span id="cb3-2491"><a href="#cb3-2491" aria-hidden="true" tabindex="-1"></a><span class="in">        #caption = "Grey points correspond to the Swiss cohort",</span></span>
<span id="cb3-2492"><a href="#cb3-2492" aria-hidden="true" tabindex="-1"></a><span class="in">        color = "Molecular\nsubtype"</span></span>
<span id="cb3-2493"><a href="#cb3-2493" aria-hidden="true" tabindex="-1"></a><span class="in">    ) +</span></span>
<span id="cb3-2494"><a href="#cb3-2494" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::scale_color_manual(</span></span>
<span id="cb3-2495"><a href="#cb3-2495" aria-hidden="true" tabindex="-1"></a><span class="in">        values = c(</span></span>
<span id="cb3-2496"><a href="#cb3-2496" aria-hidden="true" tabindex="-1"></a><span class="in">            get_colors_pam50(p$data),</span></span>
<span id="cb3-2497"><a href="#cb3-2497" aria-hidden="true" tabindex="-1"></a><span class="in">            "not available" = "grey"</span></span>
<span id="cb3-2498"><a href="#cb3-2498" aria-hidden="true" tabindex="-1"></a><span class="in">        ),</span></span>
<span id="cb3-2499"><a href="#cb3-2499" aria-hidden="true" tabindex="-1"></a><span class="in">        labels = c("Basal-like", "HER2-enriched", </span></span>
<span id="cb3-2500"><a href="#cb3-2500" aria-hidden="true" tabindex="-1"></a><span class="in">                   "LumB", "LumA", "Normal-like")</span></span>
<span id="cb3-2501"><a href="#cb3-2501" aria-hidden="true" tabindex="-1"></a><span class="in">    ) +</span></span>
<span id="cb3-2502"><a href="#cb3-2502" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme_bw(base_size = 20) + </span></span>
<span id="cb3-2503"><a href="#cb3-2503" aria-hidden="true" tabindex="-1"></a><span class="in">    change_plot_aes_point() +</span></span>
<span id="cb3-2504"><a href="#cb3-2504" aria-hidden="true" tabindex="-1"></a><span class="in">    change_guides_point()</span></span>
<span id="cb3-2505"><a href="#cb3-2505" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2506"><a href="#cb3-2506" aria-hidden="true" tabindex="-1"></a><span class="in">plots_paper$normal_samples &lt;- normal_plot</span></span>
<span id="cb3-2507"><a href="#cb3-2507" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2508"><a href="#cb3-2508" aria-hidden="true" tabindex="-1"></a><span class="in">normal_plot</span></span>
<span id="cb3-2509"><a href="#cb3-2509" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-2510"><a href="#cb3-2510" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2511"><a href="#cb3-2511" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2512"><a href="#cb3-2512" aria-hidden="true" tabindex="-1"></a>And since we finished getting all the plots for the second figure in the </span>
<span id="cb3-2513"><a href="#cb3-2513" aria-hidden="true" tabindex="-1"></a>paper, we combine it below.</span>
<span id="cb3-2514"><a href="#cb3-2514" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2515"><a href="#cb3-2515" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=22, fig.height=8}</span></span>
<span id="cb3-2516"><a href="#cb3-2516" aria-hidden="true" tabindex="-1"></a><span class="in">#| label: fig-all-fig2</span></span>
<span id="cb3-2517"><a href="#cb3-2517" aria-hidden="true" tabindex="-1"></a><span class="in">#| fig-cap: All figures combined for Fig2 in the paper.</span></span>
<span id="cb3-2518"><a href="#cb3-2518" aria-hidden="true" tabindex="-1"></a><span class="in">plots_aligned &lt;- cowplot::align_plots(</span></span>
<span id="cb3-2519"><a href="#cb3-2519" aria-hidden="true" tabindex="-1"></a><span class="in">    plots_paper$scanb_pc1_pc2,</span></span>
<span id="cb3-2520"><a href="#cb3-2520" aria-hidden="true" tabindex="-1"></a><span class="in">    plots_paper$mol_sub,</span></span>
<span id="cb3-2521"><a href="#cb3-2521" aria-hidden="true" tabindex="-1"></a><span class="in">    align = "hv", greedy = TRUE</span></span>
<span id="cb3-2522"><a href="#cb3-2522" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-2523"><a href="#cb3-2523" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2524"><a href="#cb3-2524" aria-hidden="true" tabindex="-1"></a><span class="in">plots_aligned_2 &lt;- cowplot::align_plots(</span></span>
<span id="cb3-2525"><a href="#cb3-2525" aria-hidden="true" tabindex="-1"></a><span class="in">    plots_paper$er_ihc,</span></span>
<span id="cb3-2526"><a href="#cb3-2526" aria-hidden="true" tabindex="-1"></a><span class="in">    plots_paper$pdx_ctrl_plot,</span></span>
<span id="cb3-2527"><a href="#cb3-2527" aria-hidden="true" tabindex="-1"></a><span class="in">    align = "hv", greedy = TRUE</span></span>
<span id="cb3-2528"><a href="#cb3-2528" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-2529"><a href="#cb3-2529" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2530"><a href="#cb3-2530" aria-hidden="true" tabindex="-1"></a><span class="in">plots_aligned_3 &lt;- cowplot::align_plots(</span></span>
<span id="cb3-2531"><a href="#cb3-2531" aria-hidden="true" tabindex="-1"></a><span class="in">    plots_paper$scanb_pc3_pc4,</span></span>
<span id="cb3-2532"><a href="#cb3-2532" aria-hidden="true" tabindex="-1"></a><span class="in">    plots_paper$normal_samples,</span></span>
<span id="cb3-2533"><a href="#cb3-2533" aria-hidden="true" tabindex="-1"></a><span class="in">    align = "hv", greedy = TRUE</span></span>
<span id="cb3-2534"><a href="#cb3-2534" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-2535"><a href="#cb3-2535" aria-hidden="true" tabindex="-1"></a><span class="in">cowplot::plot_grid(</span></span>
<span id="cb3-2536"><a href="#cb3-2536" aria-hidden="true" tabindex="-1"></a><span class="in">    plots_aligned[[1]],</span></span>
<span id="cb3-2537"><a href="#cb3-2537" aria-hidden="true" tabindex="-1"></a><span class="in">    plots_aligned_3[[1]],</span></span>
<span id="cb3-2538"><a href="#cb3-2538" aria-hidden="true" tabindex="-1"></a><span class="in">    plots_aligned_2[[1]],</span></span>
<span id="cb3-2539"><a href="#cb3-2539" aria-hidden="true" tabindex="-1"></a><span class="in">    plots_aligned[[2]],</span></span>
<span id="cb3-2540"><a href="#cb3-2540" aria-hidden="true" tabindex="-1"></a><span class="in">    plots_aligned_3[[2]],</span></span>
<span id="cb3-2541"><a href="#cb3-2541" aria-hidden="true" tabindex="-1"></a><span class="in">    plots_aligned_2[[2]],</span></span>
<span id="cb3-2542"><a href="#cb3-2542" aria-hidden="true" tabindex="-1"></a><span class="in">    nrow = 2, </span></span>
<span id="cb3-2543"><a href="#cb3-2543" aria-hidden="true" tabindex="-1"></a><span class="in">    ncol = 3,</span></span>
<span id="cb3-2544"><a href="#cb3-2544" aria-hidden="true" tabindex="-1"></a><span class="in">    labels = letters[1:length(plots_paper)],</span></span>
<span id="cb3-2545"><a href="#cb3-2545" aria-hidden="true" tabindex="-1"></a><span class="in">    label_size = 20</span></span>
<span id="cb3-2546"><a href="#cb3-2546" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-2547"><a href="#cb3-2547" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-2548"><a href="#cb3-2548" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2549"><a href="#cb3-2549" aria-hidden="true" tabindex="-1"></a><span class="fu">## Third component as a prognostic measure?</span></span>
<span id="cb3-2550"><a href="#cb3-2550" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2551"><a href="#cb3-2551" aria-hidden="true" tabindex="-1"></a>We have seen that estrogen signaling is a continuous measure across</span>
<span id="cb3-2552"><a href="#cb3-2552" aria-hidden="true" tabindex="-1"></a>the third component. The farthest to the right you are the</span>
<span id="cb3-2553"><a href="#cb3-2553" aria-hidden="true" tabindex="-1"></a>higher the ER signaling scores are also. Here we hypothesized then</span>
<span id="cb3-2554"><a href="#cb3-2554" aria-hidden="true" tabindex="-1"></a>that the third component is correlated to the prognostic nature</span>
<span id="cb3-2555"><a href="#cb3-2555" aria-hidden="true" tabindex="-1"></a>of the estrogen signaling score. For this we select only samples</span>
<span id="cb3-2556"><a href="#cb3-2556" aria-hidden="true" tabindex="-1"></a>with PC3 higher than 0, as these would be the ones that are from</span>
<span id="cb3-2557"><a href="#cb3-2557" aria-hidden="true" tabindex="-1"></a>luminal A and B subtypes and we use PC3 as a score in the survival analysis.</span>
<span id="cb3-2558"><a href="#cb3-2558" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2561"><a href="#cb3-2561" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-2562"><a href="#cb3-2562" aria-hidden="true" tabindex="-1"></a>scanb_samples <span class="ot">&lt;-</span> df_pca_coordinates <span class="sc">%&gt;%</span> </span>
<span id="cb3-2563"><a href="#cb3-2563" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"scanb"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb3-2564"><a href="#cb3-2564" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb3-2565"><a href="#cb3-2565" aria-hidden="true" tabindex="-1"></a>        node_stage <span class="sc">!=</span> <span class="st">"NX"</span> <span class="sc">&amp;</span></span>
<span id="cb3-2566"><a href="#cb3-2566" aria-hidden="true" tabindex="-1"></a>        <span class="sc">!</span>(tumor_stage <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"T4"</span>, <span class="st">"Tis"</span>, <span class="st">"TX"</span>)) <span class="sc">&amp;</span> </span>
<span id="cb3-2567"><a href="#cb3-2567" aria-hidden="true" tabindex="-1"></a>        er_status <span class="sc">==</span> <span class="st">"pos"</span> <span class="sc">&amp;</span> </span>
<span id="cb3-2568"><a href="#cb3-2568" aria-hidden="true" tabindex="-1"></a>        TreatGroup <span class="sc">==</span> <span class="st">"Endo"</span> <span class="sc">&amp;</span> </span>
<span id="cb3-2569"><a href="#cb3-2569" aria-hidden="true" tabindex="-1"></a>        PC3 <span class="sc">&gt;</span> <span class="dv">0</span></span>
<span id="cb3-2570"><a href="#cb3-2570" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span> </span>
<span id="cb3-2571"><a href="#cb3-2571" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">pull</span>(sample_name)</span>
<span id="cb3-2572"><a href="#cb3-2572" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2573"><a href="#cb3-2573" aria-hidden="true" tabindex="-1"></a>scanb_high_er <span class="ot">&lt;-</span> df_pca_coordinates <span class="sc">%&gt;%</span> </span>
<span id="cb3-2574"><a href="#cb3-2574" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"scanb"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb3-2575"><a href="#cb3-2575" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb3-2576"><a href="#cb3-2576" aria-hidden="true" tabindex="-1"></a>        node_stage <span class="sc">!=</span> <span class="st">"NX"</span> <span class="sc">&amp;</span></span>
<span id="cb3-2577"><a href="#cb3-2577" aria-hidden="true" tabindex="-1"></a>        <span class="sc">!</span>(tumor_stage <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"T4"</span>, <span class="st">"Tis"</span>, <span class="st">"TX"</span>)) <span class="sc">&amp;</span> </span>
<span id="cb3-2578"><a href="#cb3-2578" aria-hidden="true" tabindex="-1"></a>        er_status <span class="sc">==</span> <span class="st">"pos"</span> <span class="sc">&amp;</span> </span>
<span id="cb3-2579"><a href="#cb3-2579" aria-hidden="true" tabindex="-1"></a>        TreatGroup <span class="sc">==</span> <span class="st">"Endo"</span> <span class="sc">&amp;</span> </span>
<span id="cb3-2580"><a href="#cb3-2580" aria-hidden="true" tabindex="-1"></a>        PC3 <span class="sc">&gt;</span> <span class="dv">0</span> <span class="sc">&amp;</span> </span>
<span id="cb3-2581"><a href="#cb3-2581" aria-hidden="true" tabindex="-1"></a>        ER.pct <span class="sc">&gt;=</span> <span class="dv">90</span></span>
<span id="cb3-2582"><a href="#cb3-2582" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span> </span>
<span id="cb3-2583"><a href="#cb3-2583" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">pull</span>(sample_name)</span>
<span id="cb3-2584"><a href="#cb3-2584" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2585"><a href="#cb3-2585" aria-hidden="true" tabindex="-1"></a>metabric_samples <span class="ot">&lt;-</span> df_pca_coordinates <span class="sc">%&gt;%</span> </span>
<span id="cb3-2586"><a href="#cb3-2586" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"metabric"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb3-2587"><a href="#cb3-2587" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb3-2588"><a href="#cb3-2588" aria-hidden="true" tabindex="-1"></a>        CHEMOTHERAPY <span class="sc">==</span> <span class="st">"NO"</span> <span class="sc">&amp;</span></span>
<span id="cb3-2589"><a href="#cb3-2589" aria-hidden="true" tabindex="-1"></a>        er_status <span class="sc">==</span> <span class="st">"pos"</span> <span class="sc">&amp;</span> </span>
<span id="cb3-2590"><a href="#cb3-2590" aria-hidden="true" tabindex="-1"></a>        HORMONE_THERAPY <span class="sc">==</span> <span class="st">"YES"</span> <span class="sc">&amp;</span> </span>
<span id="cb3-2591"><a href="#cb3-2591" aria-hidden="true" tabindex="-1"></a>        PC3 <span class="sc">&gt;</span> <span class="dv">0</span></span>
<span id="cb3-2592"><a href="#cb3-2592" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span> </span>
<span id="cb3-2593"><a href="#cb3-2593" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">pull</span>(sample_name)</span>
<span id="cb3-2594"><a href="#cb3-2594" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2595"><a href="#cb3-2595" aria-hidden="true" tabindex="-1"></a>nb_pts_events <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb3-2596"><a href="#cb3-2596" aria-hidden="true" tabindex="-1"></a>    <span class="at">cohort =</span> <span class="fu">c</span>(<span class="st">"SCANB"</span>, <span class="st">"METABRIC"</span>, <span class="st">"SCANB High ER"</span>),</span>
<span id="cb3-2597"><a href="#cb3-2597" aria-hidden="true" tabindex="-1"></a>    <span class="at">number_patients =</span> <span class="fu">c</span>(</span>
<span id="cb3-2598"><a href="#cb3-2598" aria-hidden="true" tabindex="-1"></a>        <span class="fu">length</span>(scanb_samples),</span>
<span id="cb3-2599"><a href="#cb3-2599" aria-hidden="true" tabindex="-1"></a>        <span class="fu">length</span>(metabric_samples),</span>
<span id="cb3-2600"><a href="#cb3-2600" aria-hidden="true" tabindex="-1"></a>        <span class="fu">length</span>(scanb_high_er)</span>
<span id="cb3-2601"><a href="#cb3-2601" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb3-2602"><a href="#cb3-2602" aria-hidden="true" tabindex="-1"></a>    <span class="at">nb_events =</span> <span class="fu">c</span>(</span>
<span id="cb3-2603"><a href="#cb3-2603" aria-hidden="true" tabindex="-1"></a>       datasets<span class="sc">$</span>scanb <span class="sc">%&gt;%</span> colData <span class="sc">%&gt;%</span> data.frame <span class="sc">%&gt;%</span></span>
<span id="cb3-2604"><a href="#cb3-2604" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">filter</span>(sample_name <span class="sc">%in%</span> scanb_samples <span class="sc">&amp;</span> os_status <span class="sc">==</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb3-2605"><a href="#cb3-2605" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">n =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb3-2606"><a href="#cb3-2606" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">pull</span>(n),</span>
<span id="cb3-2607"><a href="#cb3-2607" aria-hidden="true" tabindex="-1"></a>       datasets<span class="sc">$</span>metabric <span class="sc">%&gt;%</span> colData <span class="sc">%&gt;%</span> data.frame <span class="sc">%&gt;%</span></span>
<span id="cb3-2608"><a href="#cb3-2608" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">filter</span>(sample_name <span class="sc">%in%</span> metabric_samples <span class="sc">&amp;</span> os_status <span class="sc">==</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb3-2609"><a href="#cb3-2609" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">n =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb3-2610"><a href="#cb3-2610" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">pull</span>(n),</span>
<span id="cb3-2611"><a href="#cb3-2611" aria-hidden="true" tabindex="-1"></a>       datasets<span class="sc">$</span>scanb <span class="sc">%&gt;%</span> colData <span class="sc">%&gt;%</span> data.frame <span class="sc">%&gt;%</span></span>
<span id="cb3-2612"><a href="#cb3-2612" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">filter</span>(sample_name <span class="sc">%in%</span> scanb_high_er <span class="sc">&amp;</span> os_status <span class="sc">==</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb3-2613"><a href="#cb3-2613" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">n =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb3-2614"><a href="#cb3-2614" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">pull</span>(n)</span>
<span id="cb3-2615"><a href="#cb3-2615" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb3-2616"><a href="#cb3-2616" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> kableExtra<span class="sc">::</span><span class="fu">kbl</span>() <span class="sc">%&gt;%</span></span>
<span id="cb3-2617"><a href="#cb3-2617" aria-hidden="true" tabindex="-1"></a>    kableExtra<span class="sc">::</span><span class="fu">kable_classic</span>(<span class="at">full_width =</span> <span class="cn">FALSE</span>)</span>
<span id="cb3-2618"><a href="#cb3-2618" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2619"><a href="#cb3-2619" aria-hidden="true" tabindex="-1"></a>nb_pts_events</span>
<span id="cb3-2620"><a href="#cb3-2620" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-2621"><a href="#cb3-2621" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2622"><a href="#cb3-2622" aria-hidden="true" tabindex="-1"></a>First we check the correlation of PC3 with estrogen response early and </span>
<span id="cb3-2623"><a href="#cb3-2623" aria-hidden="true" tabindex="-1"></a>G2M checkpoint.</span>
<span id="cb3-2624"><a href="#cb3-2624" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2625"><a href="#cb3-2625" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=12, fig.height=10}</span></span>
<span id="cb3-2626"><a href="#cb3-2626" aria-hidden="true" tabindex="-1"></a><span class="in">df_pca_coordinates %&gt;%</span></span>
<span id="cb3-2627"><a href="#cb3-2627" aria-hidden="true" tabindex="-1"></a><span class="in">    tidyr::pivot_longer(</span></span>
<span id="cb3-2628"><a href="#cb3-2628" aria-hidden="true" tabindex="-1"></a><span class="in">        cols = c(</span></span>
<span id="cb3-2629"><a href="#cb3-2629" aria-hidden="true" tabindex="-1"></a><span class="in">            "HALLMARK_ESTROGEN_RESPONSE_EARLY",</span></span>
<span id="cb3-2630"><a href="#cb3-2630" aria-hidden="true" tabindex="-1"></a><span class="in">            "HALLMARK_G2M_CHECKPOINT",</span></span>
<span id="cb3-2631"><a href="#cb3-2631" aria-hidden="true" tabindex="-1"></a><span class="in">            "SET_ERPR"</span></span>
<span id="cb3-2632"><a href="#cb3-2632" aria-hidden="true" tabindex="-1"></a><span class="in">        ),</span></span>
<span id="cb3-2633"><a href="#cb3-2633" aria-hidden="true" tabindex="-1"></a><span class="in">        names_to = "pathway",</span></span>
<span id="cb3-2634"><a href="#cb3-2634" aria-hidden="true" tabindex="-1"></a><span class="in">        values_to = "score"</span></span>
<span id="cb3-2635"><a href="#cb3-2635" aria-hidden="true" tabindex="-1"></a><span class="in">    ) %&gt;%</span></span>
<span id="cb3-2636"><a href="#cb3-2636" aria-hidden="true" tabindex="-1"></a><span class="in">    tidyr::pivot_longer(</span></span>
<span id="cb3-2637"><a href="#cb3-2637" aria-hidden="true" tabindex="-1"></a><span class="in">        cols = c("PC3", "PC4"),</span></span>
<span id="cb3-2638"><a href="#cb3-2638" aria-hidden="true" tabindex="-1"></a><span class="in">        names_to = "component",</span></span>
<span id="cb3-2639"><a href="#cb3-2639" aria-hidden="true" tabindex="-1"></a><span class="in">        values_to = "score_pc"</span></span>
<span id="cb3-2640"><a href="#cb3-2640" aria-hidden="true" tabindex="-1"></a><span class="in">    ) %&gt;%</span></span>
<span id="cb3-2641"><a href="#cb3-2641" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::ggplot(aes(</span></span>
<span id="cb3-2642"><a href="#cb3-2642" aria-hidden="true" tabindex="-1"></a><span class="in">        x = score_pc, </span></span>
<span id="cb3-2643"><a href="#cb3-2643" aria-hidden="true" tabindex="-1"></a><span class="in">        y = score,</span></span>
<span id="cb3-2644"><a href="#cb3-2644" aria-hidden="true" tabindex="-1"></a><span class="in">        color = pam50,</span></span>
<span id="cb3-2645"><a href="#cb3-2645" aria-hidden="true" tabindex="-1"></a><span class="in">        shape = cohort</span></span>
<span id="cb3-2646"><a href="#cb3-2646" aria-hidden="true" tabindex="-1"></a><span class="in">    )) +</span></span>
<span id="cb3-2647"><a href="#cb3-2647" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_point(size = 2, alpha = 0.8) + #, mapping = aes(shape = pam50)) +</span></span>
<span id="cb3-2648"><a href="#cb3-2648" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_smooth(</span></span>
<span id="cb3-2649"><a href="#cb3-2649" aria-hidden="true" tabindex="-1"></a><span class="in">        formula = y ~ s(x, bs = "cs"),</span></span>
<span id="cb3-2650"><a href="#cb3-2650" aria-hidden="true" tabindex="-1"></a><span class="in">        method = "gam",</span></span>
<span id="cb3-2651"><a href="#cb3-2651" aria-hidden="true" tabindex="-1"></a><span class="in">        color = "black",</span></span>
<span id="cb3-2652"><a href="#cb3-2652" aria-hidden="true" tabindex="-1"></a><span class="in">        mapping = aes(linetype = cohort)</span></span>
<span id="cb3-2653"><a href="#cb3-2653" aria-hidden="true" tabindex="-1"></a><span class="in">    ) +</span></span>
<span id="cb3-2654"><a href="#cb3-2654" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::labs(</span></span>
<span id="cb3-2655"><a href="#cb3-2655" aria-hidden="true" tabindex="-1"></a><span class="in">        x = "Principal Components",</span></span>
<span id="cb3-2656"><a href="#cb3-2656" aria-hidden="true" tabindex="-1"></a><span class="in">        y = "Molecular score",</span></span>
<span id="cb3-2657"><a href="#cb3-2657" aria-hidden="true" tabindex="-1"></a><span class="in">        shape = "Cohort",</span></span>
<span id="cb3-2658"><a href="#cb3-2658" aria-hidden="true" tabindex="-1"></a><span class="in">        color = "PAM50",</span></span>
<span id="cb3-2659"><a href="#cb3-2659" aria-hidden="true" tabindex="-1"></a><span class="in">        title = paste0(</span></span>
<span id="cb3-2660"><a href="#cb3-2660" aria-hidden="true" tabindex="-1"></a><span class="in">            "Correlation between principal components\nand",</span></span>
<span id="cb3-2661"><a href="#cb3-2661" aria-hidden="true" tabindex="-1"></a><span class="in">            " molecular scores"</span></span>
<span id="cb3-2662"><a href="#cb3-2662" aria-hidden="true" tabindex="-1"></a><span class="in">        )</span></span>
<span id="cb3-2663"><a href="#cb3-2663" aria-hidden="true" tabindex="-1"></a><span class="in">    ) + </span></span>
<span id="cb3-2664"><a href="#cb3-2664" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::facet_grid(</span></span>
<span id="cb3-2665"><a href="#cb3-2665" aria-hidden="true" tabindex="-1"></a><span class="in">        pathway ~ component, </span></span>
<span id="cb3-2666"><a href="#cb3-2666" aria-hidden="true" tabindex="-1"></a><span class="in">        scales = "free_y",</span></span>
<span id="cb3-2667"><a href="#cb3-2667" aria-hidden="true" tabindex="-1"></a><span class="in">        labeller = as_labeller(c(</span></span>
<span id="cb3-2668"><a href="#cb3-2668" aria-hidden="true" tabindex="-1"></a><span class="in">            "PC3" = "PC3",</span></span>
<span id="cb3-2669"><a href="#cb3-2669" aria-hidden="true" tabindex="-1"></a><span class="in">            "PC4" = "PC4",</span></span>
<span id="cb3-2670"><a href="#cb3-2670" aria-hidden="true" tabindex="-1"></a><span class="in">            "HALLMARK_ESTROGEN_RESPONSE_EARLY" = "Estrogen early",</span></span>
<span id="cb3-2671"><a href="#cb3-2671" aria-hidden="true" tabindex="-1"></a><span class="in">            "HALLMARK_G2M_CHECKPOINT" = "G2M",</span></span>
<span id="cb3-2672"><a href="#cb3-2672" aria-hidden="true" tabindex="-1"></a><span class="in">            "SET_ERPR" = "SET ER/PR"</span></span>
<span id="cb3-2673"><a href="#cb3-2673" aria-hidden="true" tabindex="-1"></a><span class="in">        ))</span></span>
<span id="cb3-2674"><a href="#cb3-2674" aria-hidden="true" tabindex="-1"></a><span class="in">    ) +</span></span>
<span id="cb3-2675"><a href="#cb3-2675" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme_bw(base_size = 18) +</span></span>
<span id="cb3-2676"><a href="#cb3-2676" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::scale_color_manual(</span></span>
<span id="cb3-2677"><a href="#cb3-2677" aria-hidden="true" tabindex="-1"></a><span class="in">        values = get_colors_pam50(df_pca_coordinates)</span></span>
<span id="cb3-2678"><a href="#cb3-2678" aria-hidden="true" tabindex="-1"></a><span class="in">    ) +</span></span>
<span id="cb3-2679"><a href="#cb3-2679" aria-hidden="true" tabindex="-1"></a><span class="in">    change_plot_aes_point() +</span></span>
<span id="cb3-2680"><a href="#cb3-2680" aria-hidden="true" tabindex="-1"></a><span class="in">    change_guides_point()</span></span>
<span id="cb3-2681"><a href="#cb3-2681" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-2682"><a href="#cb3-2682" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2683"><a href="#cb3-2683" aria-hidden="true" tabindex="-1"></a>There is indeed the correlation with the components but still a lot of </span>
<span id="cb3-2684"><a href="#cb3-2684" aria-hidden="true" tabindex="-1"></a>variability. So it will be unlikely to see such strong effects on the</span>
<span id="cb3-2685"><a href="#cb3-2685" aria-hidden="true" tabindex="-1"></a>third component.</span>
<span id="cb3-2686"><a href="#cb3-2686" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2687"><a href="#cb3-2687" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb3-2688"><a href="#cb3-2688" aria-hidden="true" tabindex="-1"></a><span class="in"># run the survival analysis for each dataset with its own formula.</span></span>
<span id="cb3-2689"><a href="#cb3-2689" aria-hidden="true" tabindex="-1"></a><span class="in"># we start by defining the base formulas, because later we sapply and </span></span>
<span id="cb3-2690"><a href="#cb3-2690" aria-hidden="true" tabindex="-1"></a><span class="in"># add the scores. this way if we need to change the formulas, we only</span></span>
<span id="cb3-2691"><a href="#cb3-2691" aria-hidden="true" tabindex="-1"></a><span class="in"># need to change once and all together.</span></span>
<span id="cb3-2692"><a href="#cb3-2692" aria-hidden="true" tabindex="-1"></a><span class="in">formulas_survival &lt;- c(</span></span>
<span id="cb3-2693"><a href="#cb3-2693" aria-hidden="true" tabindex="-1"></a><span class="in">    "os_scanb" = "Surv(os_months, os_status) ~ age + node_stage + tumor_stage",</span></span>
<span id="cb3-2694"><a href="#cb3-2694" aria-hidden="true" tabindex="-1"></a><span class="in">    "os_scanb_high_er" = "Surv(os_months, os_status) ~ age + node_stage + tumor_stage",</span></span>
<span id="cb3-2695"><a href="#cb3-2695" aria-hidden="true" tabindex="-1"></a><span class="in">    "os_metabric" = "Surv(os_months, os_status) ~ age + npi",</span></span>
<span id="cb3-2696"><a href="#cb3-2696" aria-hidden="true" tabindex="-1"></a><span class="in">    "rfs_metabric" = "Surv(rfs_months, rfs_status) ~ age + npi",</span></span>
<span id="cb3-2697"><a href="#cb3-2697" aria-hidden="true" tabindex="-1"></a><span class="in">    "rfs_scanb" = "Surv(rfs_months, rfs_status) ~ age + node_stage + tumor_stage",</span></span>
<span id="cb3-2698"><a href="#cb3-2698" aria-hidden="true" tabindex="-1"></a><span class="in">    "rfs_scanb_high_er" = "Surv(rfs_months, rfs_status) ~ age + node_stage + tumor_stage"</span></span>
<span id="cb3-2699"><a href="#cb3-2699" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-2700"><a href="#cb3-2700" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2701"><a href="#cb3-2701" aria-hidden="true" tabindex="-1"></a><span class="in">type_survival &lt;- c("os", "rfs")</span></span>
<span id="cb3-2702"><a href="#cb3-2702" aria-hidden="true" tabindex="-1"></a><span class="in">which_scores &lt;- c(</span></span>
<span id="cb3-2703"><a href="#cb3-2703" aria-hidden="true" tabindex="-1"></a><span class="in">    "PC3", </span></span>
<span id="cb3-2704"><a href="#cb3-2704" aria-hidden="true" tabindex="-1"></a><span class="in">    "PC4",</span></span>
<span id="cb3-2705"><a href="#cb3-2705" aria-hidden="true" tabindex="-1"></a><span class="in">    "SET_ERPR",</span></span>
<span id="cb3-2706"><a href="#cb3-2706" aria-hidden="true" tabindex="-1"></a><span class="in">    "SET_ERPR_scaled"</span></span>
<span id="cb3-2707"><a href="#cb3-2707" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-2708"><a href="#cb3-2708" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2709"><a href="#cb3-2709" aria-hidden="true" tabindex="-1"></a><span class="in">df_pca_coordinates$SET_ERPR_scaled &lt;- df_pca_coordinates$SET_ERPR*5 + 5</span></span>
<span id="cb3-2710"><a href="#cb3-2710" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2711"><a href="#cb3-2711" aria-hidden="true" tabindex="-1"></a><span class="in">survival_results &lt;- sapply(</span></span>
<span id="cb3-2712"><a href="#cb3-2712" aria-hidden="true" tabindex="-1"></a><span class="in">    c("os", "rfs"),</span></span>
<span id="cb3-2713"><a href="#cb3-2713" aria-hidden="true" tabindex="-1"></a><span class="in">    function(type_analysis, datasets, name_scores, formulas){</span></span>
<span id="cb3-2714"><a href="#cb3-2714" aria-hidden="true" tabindex="-1"></a><span class="in">        mapply(</span></span>
<span id="cb3-2715"><a href="#cb3-2715" aria-hidden="true" tabindex="-1"></a><span class="in">            function(</span></span>
<span id="cb3-2716"><a href="#cb3-2716" aria-hidden="true" tabindex="-1"></a><span class="in">                dataset, name_dataset, name_scores, formulas, type_analysis</span></span>
<span id="cb3-2717"><a href="#cb3-2717" aria-hidden="true" tabindex="-1"></a><span class="in">            ){</span></span>
<span id="cb3-2718"><a href="#cb3-2718" aria-hidden="true" tabindex="-1"></a><span class="in">                </span></span>
<span id="cb3-2719"><a href="#cb3-2719" aria-hidden="true" tabindex="-1"></a><span class="in">                if (type_analysis == "rfs" &amp; name_dataset == "tcga"){</span></span>
<span id="cb3-2720"><a href="#cb3-2720" aria-hidden="true" tabindex="-1"></a><span class="in">                    return()</span></span>
<span id="cb3-2721"><a href="#cb3-2721" aria-hidden="true" tabindex="-1"></a><span class="in">                } else { </span></span>
<span id="cb3-2722"><a href="#cb3-2722" aria-hidden="true" tabindex="-1"></a><span class="in">                </span></span>
<span id="cb3-2723"><a href="#cb3-2723" aria-hidden="true" tabindex="-1"></a><span class="in">                    sapply(</span></span>
<span id="cb3-2724"><a href="#cb3-2724" aria-hidden="true" tabindex="-1"></a><span class="in">                        name_scores,</span></span>
<span id="cb3-2725"><a href="#cb3-2725" aria-hidden="true" tabindex="-1"></a><span class="in">                        function(name_score, col_data, formula_str){</span></span>
<span id="cb3-2726"><a href="#cb3-2726" aria-hidden="true" tabindex="-1"></a><span class="in">                            </span></span>
<span id="cb3-2727"><a href="#cb3-2727" aria-hidden="true" tabindex="-1"></a><span class="in">                            if (name_score %in% colnames(col_data)){</span></span>
<span id="cb3-2728"><a href="#cb3-2728" aria-hidden="true" tabindex="-1"></a><span class="in">                                </span></span>
<span id="cb3-2729"><a href="#cb3-2729" aria-hidden="true" tabindex="-1"></a><span class="in">                                survival::coxph(</span></span>
<span id="cb3-2730"><a href="#cb3-2730" aria-hidden="true" tabindex="-1"></a><span class="in">                                    as.formula(</span></span>
<span id="cb3-2731"><a href="#cb3-2731" aria-hidden="true" tabindex="-1"></a><span class="in">                                        paste(</span></span>
<span id="cb3-2732"><a href="#cb3-2732" aria-hidden="true" tabindex="-1"></a><span class="in">                                            formula_str, </span></span>
<span id="cb3-2733"><a href="#cb3-2733" aria-hidden="true" tabindex="-1"></a><span class="in">                                            name_score, </span></span>
<span id="cb3-2734"><a href="#cb3-2734" aria-hidden="true" tabindex="-1"></a><span class="in">                                            sep = "+"</span></span>
<span id="cb3-2735"><a href="#cb3-2735" aria-hidden="true" tabindex="-1"></a><span class="in">                                        )[1]</span></span>
<span id="cb3-2736"><a href="#cb3-2736" aria-hidden="true" tabindex="-1"></a><span class="in">                                    ),</span></span>
<span id="cb3-2737"><a href="#cb3-2737" aria-hidden="true" tabindex="-1"></a><span class="in">                                    data = col_data, </span></span>
<span id="cb3-2738"><a href="#cb3-2738" aria-hidden="true" tabindex="-1"></a><span class="in">                                    y = FALSE,</span></span>
<span id="cb3-2739"><a href="#cb3-2739" aria-hidden="true" tabindex="-1"></a><span class="in">                                    x = FALSE</span></span>
<span id="cb3-2740"><a href="#cb3-2740" aria-hidden="true" tabindex="-1"></a><span class="in">                                )    </span></span>
<span id="cb3-2741"><a href="#cb3-2741" aria-hidden="true" tabindex="-1"></a><span class="in">                            } else {</span></span>
<span id="cb3-2742"><a href="#cb3-2742" aria-hidden="true" tabindex="-1"></a><span class="in">                                return()</span></span>
<span id="cb3-2743"><a href="#cb3-2743" aria-hidden="true" tabindex="-1"></a><span class="in">                            }</span></span>
<span id="cb3-2744"><a href="#cb3-2744" aria-hidden="true" tabindex="-1"></a><span class="in">                            </span></span>
<span id="cb3-2745"><a href="#cb3-2745" aria-hidden="true" tabindex="-1"></a><span class="in">                        },</span></span>
<span id="cb3-2746"><a href="#cb3-2746" aria-hidden="true" tabindex="-1"></a><span class="in">                        col_data = dataset,</span></span>
<span id="cb3-2747"><a href="#cb3-2747" aria-hidden="true" tabindex="-1"></a><span class="in">                        formula = formulas[</span></span>
<span id="cb3-2748"><a href="#cb3-2748" aria-hidden="true" tabindex="-1"></a><span class="in">                            grepl(</span></span>
<span id="cb3-2749"><a href="#cb3-2749" aria-hidden="true" tabindex="-1"></a><span class="in">                                paste(type_analysis, name_dataset, sep = "_"),</span></span>
<span id="cb3-2750"><a href="#cb3-2750" aria-hidden="true" tabindex="-1"></a><span class="in">                                names(formulas)</span></span>
<span id="cb3-2751"><a href="#cb3-2751" aria-hidden="true" tabindex="-1"></a><span class="in">                            )</span></span>
<span id="cb3-2752"><a href="#cb3-2752" aria-hidden="true" tabindex="-1"></a><span class="in">                        ],</span></span>
<span id="cb3-2753"><a href="#cb3-2753" aria-hidden="true" tabindex="-1"></a><span class="in">                        USE.NAMES = TRUE,</span></span>
<span id="cb3-2754"><a href="#cb3-2754" aria-hidden="true" tabindex="-1"></a><span class="in">                        simplify = FALSE</span></span>
<span id="cb3-2755"><a href="#cb3-2755" aria-hidden="true" tabindex="-1"></a><span class="in">                    )</span></span>
<span id="cb3-2756"><a href="#cb3-2756" aria-hidden="true" tabindex="-1"></a><span class="in">                }</span></span>
<span id="cb3-2757"><a href="#cb3-2757" aria-hidden="true" tabindex="-1"></a><span class="in">            },</span></span>
<span id="cb3-2758"><a href="#cb3-2758" aria-hidden="true" tabindex="-1"></a><span class="in">            dataset = datasets,</span></span>
<span id="cb3-2759"><a href="#cb3-2759" aria-hidden="true" tabindex="-1"></a><span class="in">            name_dataset = names(datasets),</span></span>
<span id="cb3-2760"><a href="#cb3-2760" aria-hidden="true" tabindex="-1"></a><span class="in">            MoreArgs = list(</span></span>
<span id="cb3-2761"><a href="#cb3-2761" aria-hidden="true" tabindex="-1"></a><span class="in">                formulas = formulas, </span></span>
<span id="cb3-2762"><a href="#cb3-2762" aria-hidden="true" tabindex="-1"></a><span class="in">                name_scores = name_scores,</span></span>
<span id="cb3-2763"><a href="#cb3-2763" aria-hidden="true" tabindex="-1"></a><span class="in">                type_analysis = type_analysis</span></span>
<span id="cb3-2764"><a href="#cb3-2764" aria-hidden="true" tabindex="-1"></a><span class="in">            ),</span></span>
<span id="cb3-2765"><a href="#cb3-2765" aria-hidden="true" tabindex="-1"></a><span class="in">            USE.NAMES = TRUE,</span></span>
<span id="cb3-2766"><a href="#cb3-2766" aria-hidden="true" tabindex="-1"></a><span class="in">            SIMPLIFY = FALSE</span></span>
<span id="cb3-2767"><a href="#cb3-2767" aria-hidden="true" tabindex="-1"></a><span class="in">        )</span></span>
<span id="cb3-2768"><a href="#cb3-2768" aria-hidden="true" tabindex="-1"></a><span class="in">    },</span></span>
<span id="cb3-2769"><a href="#cb3-2769" aria-hidden="true" tabindex="-1"></a><span class="in">    datasets = list(</span></span>
<span id="cb3-2770"><a href="#cb3-2770" aria-hidden="true" tabindex="-1"></a><span class="in">        scanb = df_pca_coordinates %&gt;%</span></span>
<span id="cb3-2771"><a href="#cb3-2771" aria-hidden="true" tabindex="-1"></a><span class="in">            dplyr::filter(sample_name %in% scanb_samples),</span></span>
<span id="cb3-2772"><a href="#cb3-2772" aria-hidden="true" tabindex="-1"></a><span class="in">        metabric = df_pca_coordinates %&gt;%</span></span>
<span id="cb3-2773"><a href="#cb3-2773" aria-hidden="true" tabindex="-1"></a><span class="in">            dplyr::filter(sample_name %in% metabric_samples),</span></span>
<span id="cb3-2774"><a href="#cb3-2774" aria-hidden="true" tabindex="-1"></a><span class="in">        scanb_high_er = df_pca_coordinates %&gt;%</span></span>
<span id="cb3-2775"><a href="#cb3-2775" aria-hidden="true" tabindex="-1"></a><span class="in">            dplyr::filter(sample_name %in% scanb_high_er)</span></span>
<span id="cb3-2776"><a href="#cb3-2776" aria-hidden="true" tabindex="-1"></a><span class="in">    ),</span></span>
<span id="cb3-2777"><a href="#cb3-2777" aria-hidden="true" tabindex="-1"></a><span class="in">    name_scores = which_scores,</span></span>
<span id="cb3-2778"><a href="#cb3-2778" aria-hidden="true" tabindex="-1"></a><span class="in">    formulas = formulas_survival,</span></span>
<span id="cb3-2779"><a href="#cb3-2779" aria-hidden="true" tabindex="-1"></a><span class="in">    USE.NAMES = TRUE,</span></span>
<span id="cb3-2780"><a href="#cb3-2780" aria-hidden="true" tabindex="-1"></a><span class="in">    simplify = FALSE</span></span>
<span id="cb3-2781"><a href="#cb3-2781" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-2782"><a href="#cb3-2782" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2783"><a href="#cb3-2783" aria-hidden="true" tabindex="-1"></a><span class="in">survival_results$rfs &lt;- survival_results$rfs[</span></span>
<span id="cb3-2784"><a href="#cb3-2784" aria-hidden="true" tabindex="-1"></a><span class="in">    !sapply(survival_results$rfs, is.null)</span></span>
<span id="cb3-2785"><a href="#cb3-2785" aria-hidden="true" tabindex="-1"></a><span class="in">]</span></span>
<span id="cb3-2786"><a href="#cb3-2786" aria-hidden="true" tabindex="-1"></a><span class="in">survival_results$rfs$metabric &lt;- survival_results$rfs$metabric[</span></span>
<span id="cb3-2787"><a href="#cb3-2787" aria-hidden="true" tabindex="-1"></a><span class="in">    !sapply(survival_results$rfs$metabric, is.null)</span></span>
<span id="cb3-2788"><a href="#cb3-2788" aria-hidden="true" tabindex="-1"></a><span class="in">]</span></span>
<span id="cb3-2789"><a href="#cb3-2789" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2790"><a href="#cb3-2790" aria-hidden="true" tabindex="-1"></a><span class="in">survival_results$os &lt;- purrr::discard(</span></span>
<span id="cb3-2791"><a href="#cb3-2791" aria-hidden="true" tabindex="-1"></a><span class="in">    purrr::map(</span></span>
<span id="cb3-2792"><a href="#cb3-2792" aria-hidden="true" tabindex="-1"></a><span class="in">        survival_results$os, </span></span>
<span id="cb3-2793"><a href="#cb3-2793" aria-hidden="true" tabindex="-1"></a><span class="in">        ~ purrr::discard(.x, is.null),</span></span>
<span id="cb3-2794"><a href="#cb3-2794" aria-hidden="true" tabindex="-1"></a><span class="in">    ),</span></span>
<span id="cb3-2795"><a href="#cb3-2795" aria-hidden="true" tabindex="-1"></a><span class="in">    is.null</span></span>
<span id="cb3-2796"><a href="#cb3-2796" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-2797"><a href="#cb3-2797" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2798"><a href="#cb3-2798" aria-hidden="true" tabindex="-1"></a><span class="in"># we cannot simply save the models as each model has some </span></span>
<span id="cb3-2799"><a href="#cb3-2799" aria-hidden="true" tabindex="-1"></a><span class="in"># environment variables, which adds up to over 20GB. </span></span>
<span id="cb3-2800"><a href="#cb3-2800" aria-hidden="true" tabindex="-1"></a><span class="in"># check this stack exchange thread to read more on it:</span></span>
<span id="cb3-2801"><a href="#cb3-2801" aria-hidden="true" tabindex="-1"></a><span class="in"># https://stackoverflow.com/questions/42230920/saverds-inflating-size-of-object/52372480</span></span>
<span id="cb3-2802"><a href="#cb3-2802" aria-hidden="true" tabindex="-1"></a><span class="in"># the solution is to basically clear the environment from the terms </span></span>
<span id="cb3-2803"><a href="#cb3-2803" aria-hidden="true" tabindex="-1"></a><span class="in"># object. since it is pretty fast to run the survival analysis</span></span>
<span id="cb3-2804"><a href="#cb3-2804" aria-hidden="true" tabindex="-1"></a><span class="in"># we will not save the objects.</span></span>
<span id="cb3-2805"><a href="#cb3-2805" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2806"><a href="#cb3-2806" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2807"><a href="#cb3-2807" aria-hidden="true" tabindex="-1"></a><span class="in"># we now proceed to prepare the plots. since the survival result is </span></span>
<span id="cb3-2808"><a href="#cb3-2808" aria-hidden="true" tabindex="-1"></a><span class="in"># actually a large file, it is best to do all the plottings, save</span></span>
<span id="cb3-2809"><a href="#cb3-2809" aria-hidden="true" tabindex="-1"></a><span class="in"># them in other rds/pdf files and then use on the quarto markdown.</span></span>
<span id="cb3-2810"><a href="#cb3-2810" aria-hidden="true" tabindex="-1"></a><span class="in"># otherwise it is too slow to render the book. </span></span>
<span id="cb3-2811"><a href="#cb3-2811" aria-hidden="true" tabindex="-1"></a><span class="in"># the same is true for the statistics and coefficients available.</span></span>
<span id="cb3-2812"><a href="#cb3-2812" aria-hidden="true" tabindex="-1"></a><span class="in">names_signatures &lt;- c(</span></span>
<span id="cb3-2813"><a href="#cb3-2813" aria-hidden="true" tabindex="-1"></a><span class="in">    "HALLMARK_ESTROGEN_RESPONSE_EARLY" = "Estrogen Early",</span></span>
<span id="cb3-2814"><a href="#cb3-2814" aria-hidden="true" tabindex="-1"></a><span class="in">    "HALLMARK_ESTROGEN_RESPONSE_LATE" = "Estrogen Late",</span></span>
<span id="cb3-2815"><a href="#cb3-2815" aria-hidden="true" tabindex="-1"></a><span class="in">    "SET_ERPR" = "SET ER/PR",</span></span>
<span id="cb3-2816"><a href="#cb3-2816" aria-hidden="true" tabindex="-1"></a><span class="in">    "random_200" = "random 200 genes",</span></span>
<span id="cb3-2817"><a href="#cb3-2817" aria-hidden="true" tabindex="-1"></a><span class="in">    "random_18" = "random 18 genes",</span></span>
<span id="cb3-2818"><a href="#cb3-2818" aria-hidden="true" tabindex="-1"></a><span class="in">    "ER.pct" = "ER percentage",</span></span>
<span id="cb3-2819"><a href="#cb3-2819" aria-hidden="true" tabindex="-1"></a><span class="in">    "PC3" = "PC3",</span></span>
<span id="cb3-2820"><a href="#cb3-2820" aria-hidden="true" tabindex="-1"></a><span class="in">    "PC4" = "PC4"</span></span>
<span id="cb3-2821"><a href="#cb3-2821" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-2822"><a href="#cb3-2822" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2823"><a href="#cb3-2823" aria-hidden="true" tabindex="-1"></a><span class="in">names_coefficients &lt;- list(</span></span>
<span id="cb3-2824"><a href="#cb3-2824" aria-hidden="true" tabindex="-1"></a><span class="in">    metabric = c(</span></span>
<span id="cb3-2825"><a href="#cb3-2825" aria-hidden="true" tabindex="-1"></a><span class="in">        "age" = "Age", </span></span>
<span id="cb3-2826"><a href="#cb3-2826" aria-hidden="true" tabindex="-1"></a><span class="in">        "npi" = "NPI", </span></span>
<span id="cb3-2827"><a href="#cb3-2827" aria-hidden="true" tabindex="-1"></a><span class="in">        names_signatures</span></span>
<span id="cb3-2828"><a href="#cb3-2828" aria-hidden="true" tabindex="-1"></a><span class="in">    ),</span></span>
<span id="cb3-2829"><a href="#cb3-2829" aria-hidden="true" tabindex="-1"></a><span class="in">    scanb = c(</span></span>
<span id="cb3-2830"><a href="#cb3-2830" aria-hidden="true" tabindex="-1"></a><span class="in">        "age" = "age",</span></span>
<span id="cb3-2831"><a href="#cb3-2831" aria-hidden="true" tabindex="-1"></a><span class="in">        names_signatures,</span></span>
<span id="cb3-2832"><a href="#cb3-2832" aria-hidden="true" tabindex="-1"></a><span class="in">        "node_stageN1" = "N1",</span></span>
<span id="cb3-2833"><a href="#cb3-2833" aria-hidden="true" tabindex="-1"></a><span class="in">        "node_stageN2and3" = "N2 and N3",</span></span>
<span id="cb3-2834"><a href="#cb3-2834" aria-hidden="true" tabindex="-1"></a><span class="in">        "tumor_stageT1" = "T1",</span></span>
<span id="cb3-2835"><a href="#cb3-2835" aria-hidden="true" tabindex="-1"></a><span class="in">        "tumor_stageT2" = "T2",</span></span>
<span id="cb3-2836"><a href="#cb3-2836" aria-hidden="true" tabindex="-1"></a><span class="in">        "tumor_stageT3" = "T3"</span></span>
<span id="cb3-2837"><a href="#cb3-2837" aria-hidden="true" tabindex="-1"></a><span class="in">    ),</span></span>
<span id="cb3-2838"><a href="#cb3-2838" aria-hidden="true" tabindex="-1"></a><span class="in">    tcga = c(</span></span>
<span id="cb3-2839"><a href="#cb3-2839" aria-hidden="true" tabindex="-1"></a><span class="in">        "age" = "age",</span></span>
<span id="cb3-2840"><a href="#cb3-2840" aria-hidden="true" tabindex="-1"></a><span class="in">        "tumor_stage" = "Tumor Stage",</span></span>
<span id="cb3-2841"><a href="#cb3-2841" aria-hidden="true" tabindex="-1"></a><span class="in">        names_signatures,</span></span>
<span id="cb3-2842"><a href="#cb3-2842" aria-hidden="true" tabindex="-1"></a><span class="in">        "node_stageN1" = "N1",</span></span>
<span id="cb3-2843"><a href="#cb3-2843" aria-hidden="true" tabindex="-1"></a><span class="in">        "node_stageN2" = "N2",</span></span>
<span id="cb3-2844"><a href="#cb3-2844" aria-hidden="true" tabindex="-1"></a><span class="in">        "node_stageN3" = "N3",</span></span>
<span id="cb3-2845"><a href="#cb3-2845" aria-hidden="true" tabindex="-1"></a><span class="in">        "tumor_stagestage_ii" = "T2",</span></span>
<span id="cb3-2846"><a href="#cb3-2846" aria-hidden="true" tabindex="-1"></a><span class="in">        "tumor_stagestage_iii" = "T3"</span></span>
<span id="cb3-2847"><a href="#cb3-2847" aria-hidden="true" tabindex="-1"></a><span class="in">    ),</span></span>
<span id="cb3-2848"><a href="#cb3-2848" aria-hidden="true" tabindex="-1"></a><span class="in">    scanb_high_er = c(</span></span>
<span id="cb3-2849"><a href="#cb3-2849" aria-hidden="true" tabindex="-1"></a><span class="in">        "age" = "age",</span></span>
<span id="cb3-2850"><a href="#cb3-2850" aria-hidden="true" tabindex="-1"></a><span class="in">        names_signatures,</span></span>
<span id="cb3-2851"><a href="#cb3-2851" aria-hidden="true" tabindex="-1"></a><span class="in">        "node_stageN1" = "N1",</span></span>
<span id="cb3-2852"><a href="#cb3-2852" aria-hidden="true" tabindex="-1"></a><span class="in">        "node_stageN2and3" = "N2 and N3",</span></span>
<span id="cb3-2853"><a href="#cb3-2853" aria-hidden="true" tabindex="-1"></a><span class="in">        "tumor_stageT1" = "T1",</span></span>
<span id="cb3-2854"><a href="#cb3-2854" aria-hidden="true" tabindex="-1"></a><span class="in">        "tumor_stageT2" = "T2",</span></span>
<span id="cb3-2855"><a href="#cb3-2855" aria-hidden="true" tabindex="-1"></a><span class="in">        "tumor_stageT3" = "T3"</span></span>
<span id="cb3-2856"><a href="#cb3-2856" aria-hidden="true" tabindex="-1"></a><span class="in">    )</span></span>
<span id="cb3-2857"><a href="#cb3-2857" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-2858"><a href="#cb3-2858" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2859"><a href="#cb3-2859" aria-hidden="true" tabindex="-1"></a><span class="in">patients &lt;- list(</span></span>
<span id="cb3-2860"><a href="#cb3-2860" aria-hidden="true" tabindex="-1"></a><span class="in">    tcga = "Lum A/B, ER+ BC",</span></span>
<span id="cb3-2861"><a href="#cb3-2861" aria-hidden="true" tabindex="-1"></a><span class="in">    scanb = "Endo Only, ER+ BC",</span></span>
<span id="cb3-2862"><a href="#cb3-2862" aria-hidden="true" tabindex="-1"></a><span class="in">    metabric = "Endo Only, ER+ BC",</span></span>
<span id="cb3-2863"><a href="#cb3-2863" aria-hidden="true" tabindex="-1"></a><span class="in">    scanb_high_er = "Endo Only, ER+ BC, ER% &gt;= 90"</span></span>
<span id="cb3-2864"><a href="#cb3-2864" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-2865"><a href="#cb3-2865" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2866"><a href="#cb3-2866" aria-hidden="true" tabindex="-1"></a><span class="in">path_to_save &lt;- "../../results/plots/validation/forest_plots/"</span></span>
<span id="cb3-2867"><a href="#cb3-2867" aria-hidden="true" tabindex="-1"></a><span class="in">sapply(</span></span>
<span id="cb3-2868"><a href="#cb3-2868" aria-hidden="true" tabindex="-1"></a><span class="in">    paste0(</span></span>
<span id="cb3-2869"><a href="#cb3-2869" aria-hidden="true" tabindex="-1"></a><span class="in">        path_to_save,</span></span>
<span id="cb3-2870"><a href="#cb3-2870" aria-hidden="true" tabindex="-1"></a><span class="in">        c("pdf", "png")</span></span>
<span id="cb3-2871"><a href="#cb3-2871" aria-hidden="true" tabindex="-1"></a><span class="in">    ), </span></span>
<span id="cb3-2872"><a href="#cb3-2872" aria-hidden="true" tabindex="-1"></a><span class="in">    dir.create,</span></span>
<span id="cb3-2873"><a href="#cb3-2873" aria-hidden="true" tabindex="-1"></a><span class="in">    showWarnings = FALSE, </span></span>
<span id="cb3-2874"><a href="#cb3-2874" aria-hidden="true" tabindex="-1"></a><span class="in">    recursive = TRUE</span></span>
<span id="cb3-2875"><a href="#cb3-2875" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-2876"><a href="#cb3-2876" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2877"><a href="#cb3-2877" aria-hidden="true" tabindex="-1"></a><span class="in">forest_plots &lt;- mapply(</span></span>
<span id="cb3-2878"><a href="#cb3-2878" aria-hidden="true" tabindex="-1"></a><span class="in">    function(fits, type_analysis){</span></span>
<span id="cb3-2879"><a href="#cb3-2879" aria-hidden="true" tabindex="-1"></a><span class="in">        mapply(</span></span>
<span id="cb3-2880"><a href="#cb3-2880" aria-hidden="true" tabindex="-1"></a><span class="in">            function(cohort_fits, name_cohort, type_analysis){</span></span>
<span id="cb3-2881"><a href="#cb3-2881" aria-hidden="true" tabindex="-1"></a><span class="in">                mapply(</span></span>
<span id="cb3-2882"><a href="#cb3-2882" aria-hidden="true" tabindex="-1"></a><span class="in">                    forest_plot_fits,</span></span>
<span id="cb3-2883"><a href="#cb3-2883" aria-hidden="true" tabindex="-1"></a><span class="in">                    fit = cohort_fits,</span></span>
<span id="cb3-2884"><a href="#cb3-2884" aria-hidden="true" tabindex="-1"></a><span class="in">                    name_signature = names(cohort_fits),</span></span>
<span id="cb3-2885"><a href="#cb3-2885" aria-hidden="true" tabindex="-1"></a><span class="in">                    MoreArgs = list(</span></span>
<span id="cb3-2886"><a href="#cb3-2886" aria-hidden="true" tabindex="-1"></a><span class="in">                        cohort = name_cohort,</span></span>
<span id="cb3-2887"><a href="#cb3-2887" aria-hidden="true" tabindex="-1"></a><span class="in">                        names_coefficients = names_coefficients,</span></span>
<span id="cb3-2888"><a href="#cb3-2888" aria-hidden="true" tabindex="-1"></a><span class="in">                        type_survival = toupper(type_analysis),</span></span>
<span id="cb3-2889"><a href="#cb3-2889" aria-hidden="true" tabindex="-1"></a><span class="in">                        patients = patients[[name_cohort]],</span></span>
<span id="cb3-2890"><a href="#cb3-2890" aria-hidden="true" tabindex="-1"></a><span class="in">                        #clip = c(0.1, 2),</span></span>
<span id="cb3-2891"><a href="#cb3-2891" aria-hidden="true" tabindex="-1"></a><span class="in">                        width = 6, </span></span>
<span id="cb3-2892"><a href="#cb3-2892" aria-hidden="true" tabindex="-1"></a><span class="in">                        height = 4, </span></span>
<span id="cb3-2893"><a href="#cb3-2893" aria-hidden="true" tabindex="-1"></a><span class="in">                        path_to_save = path_to_save</span></span>
<span id="cb3-2894"><a href="#cb3-2894" aria-hidden="true" tabindex="-1"></a><span class="in">                    ),</span></span>
<span id="cb3-2895"><a href="#cb3-2895" aria-hidden="true" tabindex="-1"></a><span class="in">                    USE.NAMES = TRUE,</span></span>
<span id="cb3-2896"><a href="#cb3-2896" aria-hidden="true" tabindex="-1"></a><span class="in">                    SIMPLIFY = FALSE</span></span>
<span id="cb3-2897"><a href="#cb3-2897" aria-hidden="true" tabindex="-1"></a><span class="in">                )</span></span>
<span id="cb3-2898"><a href="#cb3-2898" aria-hidden="true" tabindex="-1"></a><span class="in">            </span></span>
<span id="cb3-2899"><a href="#cb3-2899" aria-hidden="true" tabindex="-1"></a><span class="in">            },</span></span>
<span id="cb3-2900"><a href="#cb3-2900" aria-hidden="true" tabindex="-1"></a><span class="in">            cohort_fits = fits,</span></span>
<span id="cb3-2901"><a href="#cb3-2901" aria-hidden="true" tabindex="-1"></a><span class="in">            name_cohort = names(fits),</span></span>
<span id="cb3-2902"><a href="#cb3-2902" aria-hidden="true" tabindex="-1"></a><span class="in">            MoreArgs = list(type_analysis = type_analysis),</span></span>
<span id="cb3-2903"><a href="#cb3-2903" aria-hidden="true" tabindex="-1"></a><span class="in">            USE.NAMES = TRUE,</span></span>
<span id="cb3-2904"><a href="#cb3-2904" aria-hidden="true" tabindex="-1"></a><span class="in">            SIMPLIFY = FALSE</span></span>
<span id="cb3-2905"><a href="#cb3-2905" aria-hidden="true" tabindex="-1"></a><span class="in">        )</span></span>
<span id="cb3-2906"><a href="#cb3-2906" aria-hidden="true" tabindex="-1"></a><span class="in">    },</span></span>
<span id="cb3-2907"><a href="#cb3-2907" aria-hidden="true" tabindex="-1"></a><span class="in">    survival_results,</span></span>
<span id="cb3-2908"><a href="#cb3-2908" aria-hidden="true" tabindex="-1"></a><span class="in">    names(survival_results),</span></span>
<span id="cb3-2909"><a href="#cb3-2909" aria-hidden="true" tabindex="-1"></a><span class="in">    USE.NAMES = TRUE,</span></span>
<span id="cb3-2910"><a href="#cb3-2910" aria-hidden="true" tabindex="-1"></a><span class="in">    SIMPLIFY = FALSE</span></span>
<span id="cb3-2911"><a href="#cb3-2911" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-2912"><a href="#cb3-2912" aria-hidden="true" tabindex="-1"></a><span class="in"># all forest plots are saved as a pdf in the </span></span>
<span id="cb3-2913"><a href="#cb3-2913" aria-hidden="true" tabindex="-1"></a><span class="in"># results/plots/surv_analysis_estrogen folder and</span></span>
<span id="cb3-2914"><a href="#cb3-2914" aria-hidden="true" tabindex="-1"></a><span class="in"># as a rds file in the folder below.</span></span>
<span id="cb3-2915"><a href="#cb3-2915" aria-hidden="true" tabindex="-1"></a><span class="in">saveRDS(</span></span>
<span id="cb3-2916"><a href="#cb3-2916" aria-hidden="true" tabindex="-1"></a><span class="in">    forest_plots,</span></span>
<span id="cb3-2917"><a href="#cb3-2917" aria-hidden="true" tabindex="-1"></a><span class="in">    "../../results/rds_files/validation/forest_plots.rds"</span></span>
<span id="cb3-2918"><a href="#cb3-2918" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-2919"><a href="#cb3-2919" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2920"><a href="#cb3-2920" aria-hidden="true" tabindex="-1"></a><span class="in">analysis_code &lt;- list(</span></span>
<span id="cb3-2921"><a href="#cb3-2921" aria-hidden="true" tabindex="-1"></a><span class="in">    os = "Overall survival",</span></span>
<span id="cb3-2922"><a href="#cb3-2922" aria-hidden="true" tabindex="-1"></a><span class="in">    rfs = "Recurrence free survival"</span></span>
<span id="cb3-2923"><a href="#cb3-2923" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-2924"><a href="#cb3-2924" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2925"><a href="#cb3-2925" aria-hidden="true" tabindex="-1"></a><span class="in"># get all the results together</span></span>
<span id="cb3-2926"><a href="#cb3-2926" aria-hidden="true" tabindex="-1"></a><span class="in">tidy_survival &lt;- survival_results %&gt;%</span></span>
<span id="cb3-2927"><a href="#cb3-2927" aria-hidden="true" tabindex="-1"></a><span class="in">    lapply(</span></span>
<span id="cb3-2928"><a href="#cb3-2928" aria-hidden="true" tabindex="-1"></a><span class="in">        ., </span></span>
<span id="cb3-2929"><a href="#cb3-2929" aria-hidden="true" tabindex="-1"></a><span class="in">        function(x){</span></span>
<span id="cb3-2930"><a href="#cb3-2930" aria-hidden="true" tabindex="-1"></a><span class="in">            lapply(</span></span>
<span id="cb3-2931"><a href="#cb3-2931" aria-hidden="true" tabindex="-1"></a><span class="in">                x,</span></span>
<span id="cb3-2932"><a href="#cb3-2932" aria-hidden="true" tabindex="-1"></a><span class="in">                get_df_survival,</span></span>
<span id="cb3-2933"><a href="#cb3-2933" aria-hidden="true" tabindex="-1"></a><span class="in">                pathways_of_interest = c("PC3", "PC4")</span></span>
<span id="cb3-2934"><a href="#cb3-2934" aria-hidden="true" tabindex="-1"></a><span class="in">            ) %&gt;%</span></span>
<span id="cb3-2935"><a href="#cb3-2935" aria-hidden="true" tabindex="-1"></a><span class="in">                dplyr::bind_rows(.id = "cohort")</span></span>
<span id="cb3-2936"><a href="#cb3-2936" aria-hidden="true" tabindex="-1"></a><span class="in">        }</span></span>
<span id="cb3-2937"><a href="#cb3-2937" aria-hidden="true" tabindex="-1"></a><span class="in">    ) %&gt;%</span></span>
<span id="cb3-2938"><a href="#cb3-2938" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::bind_rows(.id = "type_analysis")</span></span>
<span id="cb3-2939"><a href="#cb3-2939" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2940"><a href="#cb3-2940" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2941"><a href="#cb3-2941" aria-hidden="true" tabindex="-1"></a><span class="in"># and now we fetch the tables that will be used later on including the</span></span>
<span id="cb3-2942"><a href="#cb3-2942" aria-hidden="true" tabindex="-1"></a><span class="in"># confidence intervals and other parameters from the fit</span></span>
<span id="cb3-2943"><a href="#cb3-2943" aria-hidden="true" tabindex="-1"></a><span class="in">tables_survival &lt;- lapply(</span></span>
<span id="cb3-2944"><a href="#cb3-2944" aria-hidden="true" tabindex="-1"></a><span class="in">    survival_results, </span></span>
<span id="cb3-2945"><a href="#cb3-2945" aria-hidden="true" tabindex="-1"></a><span class="in">    function(x) </span></span>
<span id="cb3-2946"><a href="#cb3-2946" aria-hidden="true" tabindex="-1"></a><span class="in">        lapply(</span></span>
<span id="cb3-2947"><a href="#cb3-2947" aria-hidden="true" tabindex="-1"></a><span class="in">            x,</span></span>
<span id="cb3-2948"><a href="#cb3-2948" aria-hidden="true" tabindex="-1"></a><span class="in">            function(y)</span></span>
<span id="cb3-2949"><a href="#cb3-2949" aria-hidden="true" tabindex="-1"></a><span class="in">                lapply(</span></span>
<span id="cb3-2950"><a href="#cb3-2950" aria-hidden="true" tabindex="-1"></a><span class="in">                    y,</span></span>
<span id="cb3-2951"><a href="#cb3-2951" aria-hidden="true" tabindex="-1"></a><span class="in">                    function(z) broom::tidy(</span></span>
<span id="cb3-2952"><a href="#cb3-2952" aria-hidden="true" tabindex="-1"></a><span class="in">                        z, </span></span>
<span id="cb3-2953"><a href="#cb3-2953" aria-hidden="true" tabindex="-1"></a><span class="in">                        exponentiate = TRUE, </span></span>
<span id="cb3-2954"><a href="#cb3-2954" aria-hidden="true" tabindex="-1"></a><span class="in">                        conf.int = TRUE</span></span>
<span id="cb3-2955"><a href="#cb3-2955" aria-hidden="true" tabindex="-1"></a><span class="in">                    )</span></span>
<span id="cb3-2956"><a href="#cb3-2956" aria-hidden="true" tabindex="-1"></a><span class="in">                ) %&gt;% dplyr::bind_rows(.id = "score")</span></span>
<span id="cb3-2957"><a href="#cb3-2957" aria-hidden="true" tabindex="-1"></a><span class="in">        ) %&gt;% dplyr::bind_rows(.id = "cohort")</span></span>
<span id="cb3-2958"><a href="#cb3-2958" aria-hidden="true" tabindex="-1"></a><span class="in">) %&gt;% dplyr::bind_rows(.id = "type_analysis") %&gt;%</span></span>
<span id="cb3-2959"><a href="#cb3-2959" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::mutate(HR = estimate)</span></span>
<span id="cb3-2960"><a href="#cb3-2960" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2961"><a href="#cb3-2961" aria-hidden="true" tabindex="-1"></a><span class="in">dir.create(</span></span>
<span id="cb3-2962"><a href="#cb3-2962" aria-hidden="true" tabindex="-1"></a><span class="in">    "../../results/tables/validation/",</span></span>
<span id="cb3-2963"><a href="#cb3-2963" aria-hidden="true" tabindex="-1"></a><span class="in">    showWarnings = FALSE</span></span>
<span id="cb3-2964"><a href="#cb3-2964" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-2965"><a href="#cb3-2965" aria-hidden="true" tabindex="-1"></a><span class="in">write.csv(</span></span>
<span id="cb3-2966"><a href="#cb3-2966" aria-hidden="true" tabindex="-1"></a><span class="in">    tables_survival,</span></span>
<span id="cb3-2967"><a href="#cb3-2967" aria-hidden="true" tabindex="-1"></a><span class="in">    "../../results/tables/validation/survival_results.csv",</span></span>
<span id="cb3-2968"><a href="#cb3-2968" aria-hidden="true" tabindex="-1"></a><span class="in">    row.names = FALSE</span></span>
<span id="cb3-2969"><a href="#cb3-2969" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-2970"><a href="#cb3-2970" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2971"><a href="#cb3-2971" aria-hidden="true" tabindex="-1"></a><span class="in">write.csv(</span></span>
<span id="cb3-2972"><a href="#cb3-2972" aria-hidden="true" tabindex="-1"></a><span class="in">    tidy_survival,</span></span>
<span id="cb3-2973"><a href="#cb3-2973" aria-hidden="true" tabindex="-1"></a><span class="in">    "../../results/tables/validation/tidy_survival.csv",</span></span>
<span id="cb3-2974"><a href="#cb3-2974" aria-hidden="true" tabindex="-1"></a><span class="in">    row.names = FALSE</span></span>
<span id="cb3-2975"><a href="#cb3-2975" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb3-2976"><a href="#cb3-2976" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-2977"><a href="#cb3-2977" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2980"><a href="#cb3-2980" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-2981"><a href="#cb3-2981" aria-hidden="true" tabindex="-1"></a>tables_survival <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(</span>
<span id="cb3-2982"><a href="#cb3-2982" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../../results/tables/validation/survival_results.csv"</span></span>
<span id="cb3-2983"><a href="#cb3-2983" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb3-2984"><a href="#cb3-2984" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2985"><a href="#cb3-2985" aria-hidden="true" tabindex="-1"></a>tidy_survival <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(</span>
<span id="cb3-2986"><a href="#cb3-2986" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../../results/tables/validation/tidy_survival.csv"</span></span>
<span id="cb3-2987"><a href="#cb3-2987" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb3-2988"><a href="#cb3-2988" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2989"><a href="#cb3-2989" aria-hidden="true" tabindex="-1"></a>forest_plots <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(</span>
<span id="cb3-2990"><a href="#cb3-2990" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../../results/rds_files/validation/forest_plots.rds"</span></span>
<span id="cb3-2991"><a href="#cb3-2991" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb3-2992"><a href="#cb3-2992" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-2993"><a href="#cb3-2993" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-2994"><a href="#cb3-2994" aria-hidden="true" tabindex="-1"></a>The table below shows the results for the survival analysis using all</span>
<span id="cb3-2995"><a href="#cb3-2995" aria-hidden="true" tabindex="-1"></a>the samples from SCANB and METABRIC coming from patients that received</span>
<span id="cb3-2996"><a href="#cb3-2996" aria-hidden="true" tabindex="-1"></a>only endocrine therapy and whose samples had a $PC3 &gt; 0$. OS means</span>
<span id="cb3-2997"><a href="#cb3-2997" aria-hidden="true" tabindex="-1"></a>overall survival and RFS means recurrence free survival.</span>
<span id="cb3-2998"><a href="#cb3-2998" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3001"><a href="#cb3-3001" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-3002"><a href="#cb3-3002" aria-hidden="true" tabindex="-1"></a>tables_survival <span class="sc">%&gt;%</span> </span>
<span id="cb3-3003"><a href="#cb3-3003" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(term <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"PC3"</span>, <span class="st">"SET_ERPR"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb3-3004"><a href="#cb3-3004" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(</span>
<span id="cb3-3005"><a href="#cb3-3005" aria-hidden="true" tabindex="-1"></a>        type_analysis, </span>
<span id="cb3-3006"><a href="#cb3-3006" aria-hidden="true" tabindex="-1"></a>        cohort, term, </span>
<span id="cb3-3007"><a href="#cb3-3007" aria-hidden="true" tabindex="-1"></a>        HR, conf.low, </span>
<span id="cb3-3008"><a href="#cb3-3008" aria-hidden="true" tabindex="-1"></a>        conf.high, p.value</span>
<span id="cb3-3009"><a href="#cb3-3009" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb3-3010"><a href="#cb3-3010" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">arrange</span>(term, type_analysis, cohort) <span class="sc">%&gt;%</span></span>
<span id="cb3-3011"><a href="#cb3-3011" aria-hidden="true" tabindex="-1"></a>    DT<span class="sc">::</span><span class="fu">datatable</span>(</span>
<span id="cb3-3012"><a href="#cb3-3012" aria-hidden="true" tabindex="-1"></a>        <span class="at">options =</span> <span class="fu">list</span>(</span>
<span id="cb3-3013"><a href="#cb3-3013" aria-hidden="true" tabindex="-1"></a>            <span class="at">scrollX =</span> <span class="cn">TRUE</span>, </span>
<span id="cb3-3014"><a href="#cb3-3014" aria-hidden="true" tabindex="-1"></a>            <span class="at">pageLength =</span> <span class="dv">20</span></span>
<span id="cb3-3015"><a href="#cb3-3015" aria-hidden="true" tabindex="-1"></a>        ), </span>
<span id="cb3-3016"><a href="#cb3-3016" aria-hidden="true" tabindex="-1"></a>        <span class="at">filter =</span> <span class="st">'top'</span></span>
<span id="cb3-3017"><a href="#cb3-3017" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span> </span>
<span id="cb3-3018"><a href="#cb3-3018" aria-hidden="true" tabindex="-1"></a>    DT<span class="sc">::</span><span class="fu">formatRound</span>(</span>
<span id="cb3-3019"><a href="#cb3-3019" aria-hidden="true" tabindex="-1"></a>        <span class="at">columns =</span> <span class="fu">c</span>(<span class="st">"HR"</span>, <span class="st">"conf.low"</span>, <span class="st">"conf.high"</span>), </span>
<span id="cb3-3020"><a href="#cb3-3020" aria-hidden="true" tabindex="-1"></a>        <span class="at">digits =</span> <span class="dv">2</span></span>
<span id="cb3-3021"><a href="#cb3-3021" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span> </span>
<span id="cb3-3022"><a href="#cb3-3022" aria-hidden="true" tabindex="-1"></a>    DT<span class="sc">::</span><span class="fu">formatSignif</span>(<span class="at">columns =</span> <span class="fu">c</span>(<span class="st">"p.value"</span>))</span>
<span id="cb3-3023"><a href="#cb3-3023" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-3024"><a href="#cb3-3024" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3025"><a href="#cb3-3025" aria-hidden="true" tabindex="-1"></a>We see that in all cases the HR for PC3 is below 1, for OS the values are</span>
<span id="cb3-3026"><a href="#cb3-3026" aria-hidden="true" tabindex="-1"></a>0.98 and 0.97 and the HR and for RFS from 0.80 to 0.93. What is </span>
<span id="cb3-3027"><a href="#cb3-3027" aria-hidden="true" tabindex="-1"></a>interesting to note here is that in all cases the confidence interval</span>
<span id="cb3-3028"><a href="#cb3-3028" aria-hidden="true" tabindex="-1"></a>is crossing 1, as expected due to the variability of the </span>
<span id="cb3-3029"><a href="#cb3-3029" aria-hidden="true" tabindex="-1"></a>correlation with SET ER/PR and estrogen response early. </span>
<span id="cb3-3030"><a href="#cb3-3030" aria-hidden="true" tabindex="-1"></a>Remember that the hazard ratio scale is not</span>
<span id="cb3-3031"><a href="#cb3-3031" aria-hidden="true" tabindex="-1"></a>linear. Indeed it seems that the more a sample is on the right side</span>
<span id="cb3-3032"><a href="#cb3-3032" aria-hidden="true" tabindex="-1"></a>of the plot the better is its outcome. Notice that the </span>
<span id="cb3-3033"><a href="#cb3-3033" aria-hidden="true" tabindex="-1"></a>scale for HR is different for PC3 and SET ER/PR, since one is a positive</span>
<span id="cb3-3034"><a href="#cb3-3034" aria-hidden="true" tabindex="-1"></a>score ranging from 0 to 8 and the other is a score ranging from</span>
<span id="cb3-3035"><a href="#cb3-3035" aria-hidden="true" tabindex="-1"></a>-1 to 1. If we were to rescale SET ER/PR to an interval between</span>
<span id="cb3-3036"><a href="#cb3-3036" aria-hidden="true" tabindex="-1"></a>0 to 10 by multiplying by 5 and then summing 5, we would get </span>
<span id="cb3-3037"><a href="#cb3-3037" aria-hidden="true" tabindex="-1"></a>hazard ratios in the range of 0.8. Moreover, estrogen signaling</span>
<span id="cb3-3038"><a href="#cb3-3038" aria-hidden="true" tabindex="-1"></a>is not the only thing fully explaining the third component as</span>
<span id="cb3-3039"><a href="#cb3-3039" aria-hidden="true" tabindex="-1"></a>it can be seen from @fig-pca-mol-land-pathways. In this case,</span>
<span id="cb3-3040"><a href="#cb3-3040" aria-hidden="true" tabindex="-1"></a>G2M Checkpoint and PI3K signaling have also a role in the third</span>
<span id="cb3-3041"><a href="#cb3-3041" aria-hidden="true" tabindex="-1"></a>component.</span>
<span id="cb3-3042"><a href="#cb3-3042" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3043"><a href="#cb3-3043" aria-hidden="true" tabindex="-1"></a>The following figures correspond to the overall survival analysis and</span>
<span id="cb3-3044"><a href="#cb3-3044" aria-hidden="true" tabindex="-1"></a>recurrence free survival analysis from the SCANB and METABRIC.</span>
<span id="cb3-3045"><a href="#cb3-3045" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3046"><a href="#cb3-3046" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=8,fig.height=6}</span></span>
<span id="cb3-3047"><a href="#cb3-3047" aria-hidden="true" tabindex="-1"></a><span class="in">plot_combined_scores(</span></span>
<span id="cb3-3048"><a href="#cb3-3048" aria-hidden="true" tabindex="-1"></a><span class="in">    tidy_survival %&gt;%</span></span>
<span id="cb3-3049"><a href="#cb3-3049" aria-hidden="true" tabindex="-1"></a><span class="in">        dplyr::filter(type_analysis == "os") %&gt;%</span></span>
<span id="cb3-3050"><a href="#cb3-3050" aria-hidden="true" tabindex="-1"></a><span class="in">        dplyr::mutate(pathway = paste0(pathway, "_", cohort)) %&gt;%</span></span>
<span id="cb3-3051"><a href="#cb3-3051" aria-hidden="true" tabindex="-1"></a><span class="in">        dplyr::arrange(rev(pathway)) %&gt;%</span></span>
<span id="cb3-3052"><a href="#cb3-3052" aria-hidden="true" tabindex="-1"></a><span class="in">        dplyr::mutate(pathway = factor(pathway)),</span></span>
<span id="cb3-3053"><a href="#cb3-3053" aria-hidden="true" tabindex="-1"></a><span class="in">    range_hr = 1.1,</span></span>
<span id="cb3-3054"><a href="#cb3-3054" aria-hidden="true" tabindex="-1"></a><span class="in">    range_min = 0.9</span></span>
<span id="cb3-3055"><a href="#cb3-3055" aria-hidden="true" tabindex="-1"></a><span class="in">) + </span></span>
<span id="cb3-3056"><a href="#cb3-3056" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme_bw(base_size = 15) + </span></span>
<span id="cb3-3057"><a href="#cb3-3057" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::labs(</span></span>
<span id="cb3-3058"><a href="#cb3-3058" aria-hidden="true" tabindex="-1"></a><span class="in">        title = paste0("Overall survival analysis"),</span></span>
<span id="cb3-3059"><a href="#cb3-3059" aria-hidden="true" tabindex="-1"></a><span class="in">        subtitle = "SCANB, METABRIC and SCANB high ER% (&gt;= 90%)"  </span></span>
<span id="cb3-3060"><a href="#cb3-3060" aria-hidden="true" tabindex="-1"></a><span class="in">    ) + </span></span>
<span id="cb3-3061"><a href="#cb3-3061" aria-hidden="true" tabindex="-1"></a><span class="in">    change_plot_aes_point()</span></span>
<span id="cb3-3062"><a href="#cb3-3062" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-3063"><a href="#cb3-3063" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3064"><a href="#cb3-3064" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=8,fig.height=6}</span></span>
<span id="cb3-3065"><a href="#cb3-3065" aria-hidden="true" tabindex="-1"></a><span class="in">plot_combined_scores(</span></span>
<span id="cb3-3066"><a href="#cb3-3066" aria-hidden="true" tabindex="-1"></a><span class="in">    tidy_survival %&gt;%</span></span>
<span id="cb3-3067"><a href="#cb3-3067" aria-hidden="true" tabindex="-1"></a><span class="in">        dplyr::filter(type_analysis == "rfs") %&gt;%</span></span>
<span id="cb3-3068"><a href="#cb3-3068" aria-hidden="true" tabindex="-1"></a><span class="in">        dplyr::mutate(pathway = paste0(pathway, "_", cohort)) %&gt;%</span></span>
<span id="cb3-3069"><a href="#cb3-3069" aria-hidden="true" tabindex="-1"></a><span class="in">        dplyr::arrange(rev(pathway)) %&gt;%</span></span>
<span id="cb3-3070"><a href="#cb3-3070" aria-hidden="true" tabindex="-1"></a><span class="in">        dplyr::mutate(pathway = factor(pathway)),</span></span>
<span id="cb3-3071"><a href="#cb3-3071" aria-hidden="true" tabindex="-1"></a><span class="in">    range_hr = 1.12,</span></span>
<span id="cb3-3072"><a href="#cb3-3072" aria-hidden="true" tabindex="-1"></a><span class="in">    range_min = 0.7</span></span>
<span id="cb3-3073"><a href="#cb3-3073" aria-hidden="true" tabindex="-1"></a><span class="in">) + </span></span>
<span id="cb3-3074"><a href="#cb3-3074" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme_bw(base_size = 15) + </span></span>
<span id="cb3-3075"><a href="#cb3-3075" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::labs(</span></span>
<span id="cb3-3076"><a href="#cb3-3076" aria-hidden="true" tabindex="-1"></a><span class="in">        title = paste0("Recurrence free survival (RFS) analysis"),</span></span>
<span id="cb3-3077"><a href="#cb3-3077" aria-hidden="true" tabindex="-1"></a><span class="in">        subtitle = "SCANB, METABRIC and SCANB high ER% (&gt;= 90%)"  </span></span>
<span id="cb3-3078"><a href="#cb3-3078" aria-hidden="true" tabindex="-1"></a><span class="in">    ) + </span></span>
<span id="cb3-3079"><a href="#cb3-3079" aria-hidden="true" tabindex="-1"></a><span class="in">    change_plot_aes_point()</span></span>
<span id="cb3-3080"><a href="#cb3-3080" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-3081"><a href="#cb3-3081" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3082"><a href="#cb3-3082" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10,fig.height=6}</span></span>
<span id="cb3-3083"><a href="#cb3-3083" aria-hidden="true" tabindex="-1"></a><span class="in">#| label: fig-os-rfs-pcs</span></span>
<span id="cb3-3084"><a href="#cb3-3084" aria-hidden="true" tabindex="-1"></a><span class="in">#| fig-cap: Overall and recurrence free survival analysis of the</span></span>
<span id="cb3-3085"><a href="#cb3-3085" aria-hidden="true" tabindex="-1"></a><span class="in">#|     principal components PC3 and PC4. The rows correspond to a different</span></span>
<span id="cb3-3086"><a href="#cb3-3086" aria-hidden="true" tabindex="-1"></a><span class="in">#|     PC and cohort and if the case sub cohort. </span></span>
<span id="cb3-3087"><a href="#cb3-3087" aria-hidden="true" tabindex="-1"></a><span class="in">tidy_survival %&gt;%</span></span>
<span id="cb3-3088"><a href="#cb3-3088" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::mutate(pathway = paste0(pathway, "_", cohort)) %&gt;%</span></span>
<span id="cb3-3089"><a href="#cb3-3089" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::arrange(rev(pathway)) %&gt;%</span></span>
<span id="cb3-3090"><a href="#cb3-3090" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::mutate(pathway = factor(pathway)) %&gt;%</span></span>
<span id="cb3-3091"><a href="#cb3-3091" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::ggplot(aes(</span></span>
<span id="cb3-3092"><a href="#cb3-3092" aria-hidden="true" tabindex="-1"></a><span class="in">        y = pathway, </span></span>
<span id="cb3-3093"><a href="#cb3-3093" aria-hidden="true" tabindex="-1"></a><span class="in">        x = estimate, </span></span>
<span id="cb3-3094"><a href="#cb3-3094" aria-hidden="true" tabindex="-1"></a><span class="in">        xmin = conf.low, </span></span>
<span id="cb3-3095"><a href="#cb3-3095" aria-hidden="true" tabindex="-1"></a><span class="in">        xmax = conf.high</span></span>
<span id="cb3-3096"><a href="#cb3-3096" aria-hidden="true" tabindex="-1"></a><span class="in">    )) +</span></span>
<span id="cb3-3097"><a href="#cb3-3097" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_pointrange() + </span></span>
<span id="cb3-3098"><a href="#cb3-3098" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_vline(xintercept = 1, lty = 2) + </span></span>
<span id="cb3-3099"><a href="#cb3-3099" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::labs(</span></span>
<span id="cb3-3100"><a href="#cb3-3100" aria-hidden="true" tabindex="-1"></a><span class="in">        x = "Hazard ratio (95% CI)",</span></span>
<span id="cb3-3101"><a href="#cb3-3101" aria-hidden="true" tabindex="-1"></a><span class="in">        y = ""</span></span>
<span id="cb3-3102"><a href="#cb3-3102" aria-hidden="true" tabindex="-1"></a><span class="in">    ) +</span></span>
<span id="cb3-3103"><a href="#cb3-3103" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::facet_wrap(</span></span>
<span id="cb3-3104"><a href="#cb3-3104" aria-hidden="true" tabindex="-1"></a><span class="in">        ~type_analysis, </span></span>
<span id="cb3-3105"><a href="#cb3-3105" aria-hidden="true" tabindex="-1"></a><span class="in">        scales = "free_x", </span></span>
<span id="cb3-3106"><a href="#cb3-3106" aria-hidden="true" tabindex="-1"></a><span class="in">        labeller = as_labeller(c(</span></span>
<span id="cb3-3107"><a href="#cb3-3107" aria-hidden="true" tabindex="-1"></a><span class="in">            "os" = "Overall survival (OS)",</span></span>
<span id="cb3-3108"><a href="#cb3-3108" aria-hidden="true" tabindex="-1"></a><span class="in">            "rfs" = "Recurrence free survival (RFS)"</span></span>
<span id="cb3-3109"><a href="#cb3-3109" aria-hidden="true" tabindex="-1"></a><span class="in">        ))</span></span>
<span id="cb3-3110"><a href="#cb3-3110" aria-hidden="true" tabindex="-1"></a><span class="in">    ) +</span></span>
<span id="cb3-3111"><a href="#cb3-3111" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme_bw(base_size = 15) + </span></span>
<span id="cb3-3112"><a href="#cb3-3112" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::labs(</span></span>
<span id="cb3-3113"><a href="#cb3-3113" aria-hidden="true" tabindex="-1"></a><span class="in">        title = paste0("Survival (RFS) analysis of the principal components"),</span></span>
<span id="cb3-3114"><a href="#cb3-3114" aria-hidden="true" tabindex="-1"></a><span class="in">        subtitle = paste0(</span></span>
<span id="cb3-3115"><a href="#cb3-3115" aria-hidden="true" tabindex="-1"></a><span class="in">            "SCANB, METABRIC and SCANB high ER% (&gt;= 90%).\n",</span></span>
<span id="cb3-3116"><a href="#cb3-3116" aria-hidden="true" tabindex="-1"></a><span class="in">            "ER+ BC, ET treated only patients."</span></span>
<span id="cb3-3117"><a href="#cb3-3117" aria-hidden="true" tabindex="-1"></a><span class="in">        )</span></span>
<span id="cb3-3118"><a href="#cb3-3118" aria-hidden="true" tabindex="-1"></a><span class="in">    ) + </span></span>
<span id="cb3-3119"><a href="#cb3-3119" aria-hidden="true" tabindex="-1"></a><span class="in">    change_plot_aes_point()</span></span>
<span id="cb3-3120"><a href="#cb3-3120" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-3121"><a href="#cb3-3121" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3122"><a href="#cb3-3122" aria-hidden="true" tabindex="-1"></a><span class="fu">## High risk patients</span></span>
<span id="cb3-3123"><a href="#cb3-3123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3124"><a href="#cb3-3124" aria-hidden="true" tabindex="-1"></a>The Prosigna risk of recurrence score takes into account the </span>
<span id="cb3-3125"><a href="#cb3-3125" aria-hidden="true" tabindex="-1"></a>tumor stage and the molecular subtype, this is the ROR-C and </span>
<span id="cb3-3126"><a href="#cb3-3126" aria-hidden="true" tabindex="-1"></a>the formula from the original paper is presented below:</span>
<span id="cb3-3127"><a href="#cb3-3127" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3128"><a href="#cb3-3128" aria-hidden="true" tabindex="-1"></a>$$</span>
<span id="cb3-3129"><a href="#cb3-3129" aria-hidden="true" tabindex="-1"></a>ROR-C = 0.05\times basal + 0.11 \times HER2 - 0.23 \times LumA + 0.09 \times </span>
<span id="cb3-3130"><a href="#cb3-3130" aria-hidden="true" tabindex="-1"></a>LumB  + 0.17 \times Tumor size</span>
<span id="cb3-3131"><a href="#cb3-3131" aria-hidden="true" tabindex="-1"></a>$$</span>
<span id="cb3-3132"><a href="#cb3-3132" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3133"><a href="#cb3-3133" aria-hidden="true" tabindex="-1"></a>We see that if a tumor sample has a high correlation to the</span>
<span id="cb3-3134"><a href="#cb3-3134" aria-hidden="true" tabindex="-1"></a>luminal A subtype it will decrease the score,</span>
<span id="cb3-3135"><a href="#cb3-3135" aria-hidden="true" tabindex="-1"></a>whereas for all other cases there is an increase in the risk. We </span>
<span id="cb3-3136"><a href="#cb3-3136" aria-hidden="true" tabindex="-1"></a>want to evaluate if the position in the molecular landscape is </span>
<span id="cb3-3137"><a href="#cb3-3137" aria-hidden="true" tabindex="-1"></a>somehow correlated with the risk score, specially in the forth component</span>
<span id="cb3-3138"><a href="#cb3-3138" aria-hidden="true" tabindex="-1"></a>for luminal A patients.</span>
<span id="cb3-3139"><a href="#cb3-3139" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3140"><a href="#cb3-3140" aria-hidden="true" tabindex="-1"></a>In <span class="co">[</span><span class="ot">@SCANB2022</span><span class="co">]</span> they showed that they are able to predict the </span>
<span id="cb3-3141"><a href="#cb3-3141" aria-hidden="true" tabindex="-1"></a>Prosigna binary categories high or low/intermediate risk for </span>
<span id="cb3-3142"><a href="#cb3-3142" aria-hidden="true" tabindex="-1"></a>the patients based only on the RNA-seq provided by their pipeline.</span>
<span id="cb3-3143"><a href="#cb3-3143" aria-hidden="true" tabindex="-1"></a>We use these categories here to understand the molecular scores </span>
<span id="cb3-3144"><a href="#cb3-3144" aria-hidden="true" tabindex="-1"></a>of the samples and also their correlation with the position and</span>
<span id="cb3-3145"><a href="#cb3-3145" aria-hidden="true" tabindex="-1"></a>risk groups.  All the following analysis are done for ER+ BC patients </span>
<span id="cb3-3146"><a href="#cb3-3146" aria-hidden="true" tabindex="-1"></a>**only**.</span>
<span id="cb3-3147"><a href="#cb3-3147" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3148"><a href="#cb3-3148" aria-hidden="true" tabindex="-1"></a>First we stratify the risk groups by the molecular subtype.</span>
<span id="cb3-3149"><a href="#cb3-3149" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3152"><a href="#cb3-3152" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-3153"><a href="#cb3-3153" aria-hidden="true" tabindex="-1"></a>sub_df_scanb <span class="ot">&lt;-</span> df_pca_coordinates <span class="sc">%&gt;%</span></span>
<span id="cb3-3154"><a href="#cb3-3154" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb3-3155"><a href="#cb3-3155" aria-hidden="true" tabindex="-1"></a>        cohort <span class="sc">==</span> <span class="st">"scanb"</span> <span class="sc">&amp;</span> </span>
<span id="cb3-3156"><a href="#cb3-3156" aria-hidden="true" tabindex="-1"></a>            er_status <span class="sc">==</span> <span class="st">"pos"</span> <span class="sc">&amp;</span> </span>
<span id="cb3-3157"><a href="#cb3-3157" aria-hidden="true" tabindex="-1"></a>            <span class="sc">!</span><span class="fu">is.na</span>(SSP.ROR.binary.risk.cat)</span>
<span id="cb3-3158"><a href="#cb3-3158" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb3-3159"><a href="#cb3-3159" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3160"><a href="#cb3-3160" aria-hidden="true" tabindex="-1"></a>sub_df_scanb <span class="sc">%&gt;%</span> </span>
<span id="cb3-3161"><a href="#cb3-3161" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">pam50 =</span> <span class="fu">droplevels</span>(pam50)) <span class="sc">%&gt;%</span></span>
<span id="cb3-3162"><a href="#cb3-3162" aria-hidden="true" tabindex="-1"></a>    janitor<span class="sc">::</span><span class="fu">tabyl</span>(pam50, SSP.ROR.binary.risk.cat) <span class="sc">%&gt;%</span></span>
<span id="cb3-3163"><a href="#cb3-3163" aria-hidden="true" tabindex="-1"></a>    janitor<span class="sc">::</span><span class="fu">adorn_totals</span>() <span class="sc">%&gt;%</span></span>
<span id="cb3-3164"><a href="#cb3-3164" aria-hidden="true" tabindex="-1"></a>    janitor<span class="sc">::</span><span class="fu">adorn_percentages</span>(<span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb3-3165"><a href="#cb3-3165" aria-hidden="true" tabindex="-1"></a>    janitor<span class="sc">::</span><span class="fu">adorn_pct_formatting</span>(<span class="at">digits =</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb3-3166"><a href="#cb3-3166" aria-hidden="true" tabindex="-1"></a>    janitor<span class="sc">::</span><span class="fu">adorn_ns</span>() <span class="sc">%&gt;%</span></span>
<span id="cb3-3167"><a href="#cb3-3167" aria-hidden="true" tabindex="-1"></a>    DT<span class="sc">::</span><span class="fu">datatable</span>()</span>
<span id="cb3-3168"><a href="#cb3-3168" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-3169"><a href="#cb3-3169" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3170"><a href="#cb3-3170" aria-hidden="true" tabindex="-1"></a>As expected the majority of luminal A patients are low/intermediate</span>
<span id="cb3-3171"><a href="#cb3-3171" aria-hidden="true" tabindex="-1"></a>risk, on the other hand most of the luminal B patients are considered</span>
<span id="cb3-3172"><a href="#cb3-3172" aria-hidden="true" tabindex="-1"></a>of high risk. </span>
<span id="cb3-3173"><a href="#cb3-3173" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3174"><a href="#cb3-3174" aria-hidden="true" tabindex="-1"></a>If we look more specifically for luminal A patients and stratify </span>
<span id="cb3-3175"><a href="#cb3-3175" aria-hidden="true" tabindex="-1"></a>on the tumor stage we get the following numbers.</span>
<span id="cb3-3176"><a href="#cb3-3176" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3179"><a href="#cb3-3179" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-3180"><a href="#cb3-3180" aria-hidden="true" tabindex="-1"></a>sub_df_scanb <span class="sc">%&gt;%</span> </span>
<span id="cb3-3181"><a href="#cb3-3181" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>( </span>
<span id="cb3-3182"><a href="#cb3-3182" aria-hidden="true" tabindex="-1"></a>            pam50 <span class="sc">==</span> <span class="st">"luma"</span> <span class="sc">&amp;</span> </span>
<span id="cb3-3183"><a href="#cb3-3183" aria-hidden="true" tabindex="-1"></a>            <span class="sc">!</span><span class="fu">is.na</span>(tumor_stage)</span>
<span id="cb3-3184"><a href="#cb3-3184" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb3-3185"><a href="#cb3-3185" aria-hidden="true" tabindex="-1"></a>    janitor<span class="sc">::</span><span class="fu">tabyl</span>(tumor_stage, SSP.ROR.binary.risk.cat) <span class="sc">%&gt;%</span></span>
<span id="cb3-3186"><a href="#cb3-3186" aria-hidden="true" tabindex="-1"></a>    janitor<span class="sc">::</span><span class="fu">adorn_totals</span>() <span class="sc">%&gt;%</span></span>
<span id="cb3-3187"><a href="#cb3-3187" aria-hidden="true" tabindex="-1"></a>    janitor<span class="sc">::</span><span class="fu">adorn_percentages</span>(<span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb3-3188"><a href="#cb3-3188" aria-hidden="true" tabindex="-1"></a>    janitor<span class="sc">::</span><span class="fu">adorn_pct_formatting</span>(<span class="at">digits =</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb3-3189"><a href="#cb3-3189" aria-hidden="true" tabindex="-1"></a>    janitor<span class="sc">::</span><span class="fu">adorn_ns</span>() <span class="sc">%&gt;%</span></span>
<span id="cb3-3190"><a href="#cb3-3190" aria-hidden="true" tabindex="-1"></a>    DT<span class="sc">::</span><span class="fu">datatable</span>()</span>
<span id="cb3-3191"><a href="#cb3-3191" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-3192"><a href="#cb3-3192" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3193"><a href="#cb3-3193" aria-hidden="true" tabindex="-1"></a>We see that for tumor stage 1 the number of high individuals is</span>
<span id="cb3-3194"><a href="#cb3-3194" aria-hidden="true" tabindex="-1"></a>lower in percentage than in tumor stage 3, as expected. We </span>
<span id="cb3-3195"><a href="#cb3-3195" aria-hidden="true" tabindex="-1"></a>now plot (@fig-luma-scanb-g2m-ts) for all these patients the G2M </span>
<span id="cb3-3196"><a href="#cb3-3196" aria-hidden="true" tabindex="-1"></a>molecular score,</span>
<span id="cb3-3197"><a href="#cb3-3197" aria-hidden="true" tabindex="-1"></a>an index of proliferation, stratified by the tumor stage. </span>
<span id="cb3-3198"><a href="#cb3-3198" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3199"><a href="#cb3-3199" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=11, fig.height=6}</span></span>
<span id="cb3-3200"><a href="#cb3-3200" aria-hidden="true" tabindex="-1"></a><span class="in">#| label: fig-luma-scanb-g2m-ts</span></span>
<span id="cb3-3201"><a href="#cb3-3201" aria-hidden="true" tabindex="-1"></a><span class="in">#| fig-cap: Luminal A BC patients G2M scores stratified by tumor stage.</span></span>
<span id="cb3-3202"><a href="#cb3-3202" aria-hidden="true" tabindex="-1"></a><span class="in">sub_df_scanb %&gt;% </span></span>
<span id="cb3-3203"><a href="#cb3-3203" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::filter(</span></span>
<span id="cb3-3204"><a href="#cb3-3204" aria-hidden="true" tabindex="-1"></a><span class="in">            pam50 == "luma" &amp; </span></span>
<span id="cb3-3205"><a href="#cb3-3205" aria-hidden="true" tabindex="-1"></a><span class="in">            !is.na(tumor_stage)</span></span>
<span id="cb3-3206"><a href="#cb3-3206" aria-hidden="true" tabindex="-1"></a><span class="in">    ) %&gt;%</span></span>
<span id="cb3-3207"><a href="#cb3-3207" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::ggplot(aes(</span></span>
<span id="cb3-3208"><a href="#cb3-3208" aria-hidden="true" tabindex="-1"></a><span class="in">        fill = SSP.ROR.binary.risk.cat, </span></span>
<span id="cb3-3209"><a href="#cb3-3209" aria-hidden="true" tabindex="-1"></a><span class="in">        y = HALLMARK_G2M_CHECKPOINT,</span></span>
<span id="cb3-3210"><a href="#cb3-3210" aria-hidden="true" tabindex="-1"></a><span class="in">        x = SSP.ROR.binary.risk.cat </span></span>
<span id="cb3-3211"><a href="#cb3-3211" aria-hidden="true" tabindex="-1"></a><span class="in">    )) +</span></span>
<span id="cb3-3212"><a href="#cb3-3212" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_jitter() + </span></span>
<span id="cb3-3213"><a href="#cb3-3213" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_violin(alpha = 0.3) +</span></span>
<span id="cb3-3214"><a href="#cb3-3214" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::facet_wrap(~tumor_stage) +</span></span>
<span id="cb3-3215"><a href="#cb3-3215" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme_bw(base_size = 20) +</span></span>
<span id="cb3-3216"><a href="#cb3-3216" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::labs(</span></span>
<span id="cb3-3217"><a href="#cb3-3217" aria-hidden="true" tabindex="-1"></a><span class="in">        x = "ROR binary risk category (SSP)",</span></span>
<span id="cb3-3218"><a href="#cb3-3218" aria-hidden="true" tabindex="-1"></a><span class="in">        title = paste0(</span></span>
<span id="cb3-3219"><a href="#cb3-3219" aria-hidden="true" tabindex="-1"></a><span class="in">            "G2M scores stratified by risk category.\n",</span></span>
<span id="cb3-3220"><a href="#cb3-3220" aria-hidden="true" tabindex="-1"></a><span class="in">            "Only luminal A, ER+ BC, ET treated only patients from SCANB"</span></span>
<span id="cb3-3221"><a href="#cb3-3221" aria-hidden="true" tabindex="-1"></a><span class="in">        ),</span></span>
<span id="cb3-3222"><a href="#cb3-3222" aria-hidden="true" tabindex="-1"></a><span class="in">        y = "G2M"</span></span>
<span id="cb3-3223"><a href="#cb3-3223" aria-hidden="true" tabindex="-1"></a><span class="in">    ) + </span></span>
<span id="cb3-3224"><a href="#cb3-3224" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme(legend.position = "none") + </span></span>
<span id="cb3-3225"><a href="#cb3-3225" aria-hidden="true" tabindex="-1"></a><span class="in">    change_plot_aes_point()</span></span>
<span id="cb3-3226"><a href="#cb3-3226" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-3227"><a href="#cb3-3227" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3228"><a href="#cb3-3228" aria-hidden="true" tabindex="-1"></a>We see that in all three cases in average the G2M scores are </span>
<span id="cb3-3229"><a href="#cb3-3229" aria-hidden="true" tabindex="-1"></a>higher in the high risk groups than in the Low/Intermediate.</span>
<span id="cb3-3230"><a href="#cb3-3230" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3231"><a href="#cb3-3231" aria-hidden="true" tabindex="-1"></a>We saw previously that some of the G2M checkpoint genes have higher</span>
<span id="cb3-3232"><a href="#cb3-3232" aria-hidden="true" tabindex="-1"></a>loadings for the fourth component. We now plot the PC4 components stratified</span>
<span id="cb3-3233"><a href="#cb3-3233" aria-hidden="true" tabindex="-1"></a>by the risk groups for the luminal A patients (@fig-luma-scanb-pc4).</span>
<span id="cb3-3234"><a href="#cb3-3234" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3235"><a href="#cb3-3235" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=6, fig.height=4}</span></span>
<span id="cb3-3236"><a href="#cb3-3236" aria-hidden="true" tabindex="-1"></a><span class="in">#| label: fig-luma-scanb-pc4</span></span>
<span id="cb3-3237"><a href="#cb3-3237" aria-hidden="true" tabindex="-1"></a><span class="in">#| fig-cap: Luminal A BC patients PC4 values stratified by risk category.</span></span>
<span id="cb3-3238"><a href="#cb3-3238" aria-hidden="true" tabindex="-1"></a><span class="in">sub_df_scanb %&gt;% </span></span>
<span id="cb3-3239"><a href="#cb3-3239" aria-hidden="true" tabindex="-1"></a><span class="in">    dplyr::filter(pam50 == "luma") %&gt;%</span></span>
<span id="cb3-3240"><a href="#cb3-3240" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::ggplot(aes(x = SSP.ROR.binary.risk.cat, y = PC4)) +</span></span>
<span id="cb3-3241"><a href="#cb3-3241" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_jitter() + </span></span>
<span id="cb3-3242"><a href="#cb3-3242" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_violin(alpha = 0.3) +</span></span>
<span id="cb3-3243"><a href="#cb3-3243" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_boxplot(</span></span>
<span id="cb3-3244"><a href="#cb3-3244" aria-hidden="true" tabindex="-1"></a><span class="in">        width=0.1, </span></span>
<span id="cb3-3245"><a href="#cb3-3245" aria-hidden="true" tabindex="-1"></a><span class="in">        color = "black", </span></span>
<span id="cb3-3246"><a href="#cb3-3246" aria-hidden="true" tabindex="-1"></a><span class="in">        alpha = 0.3,</span></span>
<span id="cb3-3247"><a href="#cb3-3247" aria-hidden="true" tabindex="-1"></a><span class="in">        outlier.shape = NA</span></span>
<span id="cb3-3248"><a href="#cb3-3248" aria-hidden="true" tabindex="-1"></a><span class="in">    ) +</span></span>
<span id="cb3-3249"><a href="#cb3-3249" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme_bw(base_size = 15) +</span></span>
<span id="cb3-3250"><a href="#cb3-3250" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::labs(</span></span>
<span id="cb3-3251"><a href="#cb3-3251" aria-hidden="true" tabindex="-1"></a><span class="in">        x = "ROR binary risk category (SSP)",</span></span>
<span id="cb3-3252"><a href="#cb3-3252" aria-hidden="true" tabindex="-1"></a><span class="in">        title = paste0(</span></span>
<span id="cb3-3253"><a href="#cb3-3253" aria-hidden="true" tabindex="-1"></a><span class="in">            "Molecular scores stratified by risk category.\n",</span></span>
<span id="cb3-3254"><a href="#cb3-3254" aria-hidden="true" tabindex="-1"></a><span class="in">            "Only luminal A, ER+ BC patients from SCANB"</span></span>
<span id="cb3-3255"><a href="#cb3-3255" aria-hidden="true" tabindex="-1"></a><span class="in">        )</span></span>
<span id="cb3-3256"><a href="#cb3-3256" aria-hidden="true" tabindex="-1"></a><span class="in">    ) + </span></span>
<span id="cb3-3257"><a href="#cb3-3257" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme(legend.position = "bottom") + </span></span>
<span id="cb3-3258"><a href="#cb3-3258" aria-hidden="true" tabindex="-1"></a><span class="in">    change_plot_aes_point()</span></span>
<span id="cb3-3259"><a href="#cb3-3259" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-3260"><a href="#cb3-3260" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3261"><a href="#cb3-3261" aria-hidden="true" tabindex="-1"></a>Indeed, in average the higher the PC4, the more likely it is to be</span>
<span id="cb3-3262"><a href="#cb3-3262" aria-hidden="true" tabindex="-1"></a>in the High risk group. And now we compare the </span>
<span id="cb3-3263"><a href="#cb3-3263" aria-hidden="true" tabindex="-1"></a>G2M Checkpoint, Estrogen response early and random 200 signature </span>
<span id="cb3-3264"><a href="#cb3-3264" aria-hidden="true" tabindex="-1"></a>for each molecular subtype separately (@fig-all-scanb-g2m-ere) of </span>
<span id="cb3-3265"><a href="#cb3-3265" aria-hidden="true" tabindex="-1"></a>ER+ BC patients.</span>
<span id="cb3-3266"><a href="#cb3-3266" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3267"><a href="#cb3-3267" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=14,fig.height=10}</span></span>
<span id="cb3-3268"><a href="#cb3-3268" aria-hidden="true" tabindex="-1"></a><span class="in">#| label: fig-all-scanb-g2m-ere</span></span>
<span id="cb3-3269"><a href="#cb3-3269" aria-hidden="true" tabindex="-1"></a><span class="in">#| fig-cap: ER+ BC patients and molecular scores stratified by</span></span>
<span id="cb3-3270"><a href="#cb3-3270" aria-hidden="true" tabindex="-1"></a><span class="in">#|     molecular subtype and binary risk categories.</span></span>
<span id="cb3-3271"><a href="#cb3-3271" aria-hidden="true" tabindex="-1"></a><span class="in">sub_df_scanb %&gt;% </span></span>
<span id="cb3-3272"><a href="#cb3-3272" aria-hidden="true" tabindex="-1"></a><span class="in">    tidyr::pivot_longer(</span></span>
<span id="cb3-3273"><a href="#cb3-3273" aria-hidden="true" tabindex="-1"></a><span class="in">        cols = c(</span></span>
<span id="cb3-3274"><a href="#cb3-3274" aria-hidden="true" tabindex="-1"></a><span class="in">            "HALLMARK_G2M_CHECKPOINT", </span></span>
<span id="cb3-3275"><a href="#cb3-3275" aria-hidden="true" tabindex="-1"></a><span class="in">            "HALLMARK_ESTROGEN_RESPONSE_EARLY",</span></span>
<span id="cb3-3276"><a href="#cb3-3276" aria-hidden="true" tabindex="-1"></a><span class="in">            "random_200"</span></span>
<span id="cb3-3277"><a href="#cb3-3277" aria-hidden="true" tabindex="-1"></a><span class="in">        ),</span></span>
<span id="cb3-3278"><a href="#cb3-3278" aria-hidden="true" tabindex="-1"></a><span class="in">        names_to = "pathway",</span></span>
<span id="cb3-3279"><a href="#cb3-3279" aria-hidden="true" tabindex="-1"></a><span class="in">        values_to = "score"</span></span>
<span id="cb3-3280"><a href="#cb3-3280" aria-hidden="true" tabindex="-1"></a><span class="in">    ) %&gt;%</span></span>
<span id="cb3-3281"><a href="#cb3-3281" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::ggplot(aes(</span></span>
<span id="cb3-3282"><a href="#cb3-3282" aria-hidden="true" tabindex="-1"></a><span class="in">        fill = SSP.ROR.binary.risk.cat, </span></span>
<span id="cb3-3283"><a href="#cb3-3283" aria-hidden="true" tabindex="-1"></a><span class="in">        x = score</span></span>
<span id="cb3-3284"><a href="#cb3-3284" aria-hidden="true" tabindex="-1"></a><span class="in">    )) +</span></span>
<span id="cb3-3285"><a href="#cb3-3285" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::geom_density(alpha = 0.3) + </span></span>
<span id="cb3-3286"><a href="#cb3-3286" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::facet_grid(</span></span>
<span id="cb3-3287"><a href="#cb3-3287" aria-hidden="true" tabindex="-1"></a><span class="in">        pathway~pam50, </span></span>
<span id="cb3-3288"><a href="#cb3-3288" aria-hidden="true" tabindex="-1"></a><span class="in">        scales = "free_y",</span></span>
<span id="cb3-3289"><a href="#cb3-3289" aria-hidden="true" tabindex="-1"></a><span class="in">        labeller = as_labeller(c(</span></span>
<span id="cb3-3290"><a href="#cb3-3290" aria-hidden="true" tabindex="-1"></a><span class="in">            "HALLMARK_ESTROGEN_RESPONSE_EARLY" = "Estrogen early",</span></span>
<span id="cb3-3291"><a href="#cb3-3291" aria-hidden="true" tabindex="-1"></a><span class="in">            "HALLMARK_G2M_CHECKPOINT" = "G2M",</span></span>
<span id="cb3-3292"><a href="#cb3-3292" aria-hidden="true" tabindex="-1"></a><span class="in">            "random_200" = "Random 200",</span></span>
<span id="cb3-3293"><a href="#cb3-3293" aria-hidden="true" tabindex="-1"></a><span class="in">            "luma" = "luma",</span></span>
<span id="cb3-3294"><a href="#cb3-3294" aria-hidden="true" tabindex="-1"></a><span class="in">            "lumb" = "lumb",</span></span>
<span id="cb3-3295"><a href="#cb3-3295" aria-hidden="true" tabindex="-1"></a><span class="in">            "her2" = "her2",</span></span>
<span id="cb3-3296"><a href="#cb3-3296" aria-hidden="true" tabindex="-1"></a><span class="in">            "normal" = "normal",</span></span>
<span id="cb3-3297"><a href="#cb3-3297" aria-hidden="true" tabindex="-1"></a><span class="in">            "basal" = "basal"</span></span>
<span id="cb3-3298"><a href="#cb3-3298" aria-hidden="true" tabindex="-1"></a><span class="in">        ))</span></span>
<span id="cb3-3299"><a href="#cb3-3299" aria-hidden="true" tabindex="-1"></a><span class="in">    ) +</span></span>
<span id="cb3-3300"><a href="#cb3-3300" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme_bw(base_size = 20) +</span></span>
<span id="cb3-3301"><a href="#cb3-3301" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::labs(</span></span>
<span id="cb3-3302"><a href="#cb3-3302" aria-hidden="true" tabindex="-1"></a><span class="in">        fill = "ROR binary risk category (SSP)",</span></span>
<span id="cb3-3303"><a href="#cb3-3303" aria-hidden="true" tabindex="-1"></a><span class="in">        x = "Molecular score",</span></span>
<span id="cb3-3304"><a href="#cb3-3304" aria-hidden="true" tabindex="-1"></a><span class="in">        title = paste0(</span></span>
<span id="cb3-3305"><a href="#cb3-3305" aria-hidden="true" tabindex="-1"></a><span class="in">            "Molecular scores stratified by risk category and PAM50.\n",</span></span>
<span id="cb3-3306"><a href="#cb3-3306" aria-hidden="true" tabindex="-1"></a><span class="in">            "Only ER+ BC patients from SCANB"</span></span>
<span id="cb3-3307"><a href="#cb3-3307" aria-hidden="true" tabindex="-1"></a><span class="in">        ),</span></span>
<span id="cb3-3308"><a href="#cb3-3308" aria-hidden="true" tabindex="-1"></a><span class="in">        y = "Density"</span></span>
<span id="cb3-3309"><a href="#cb3-3309" aria-hidden="true" tabindex="-1"></a><span class="in">    ) + </span></span>
<span id="cb3-3310"><a href="#cb3-3310" aria-hidden="true" tabindex="-1"></a><span class="in">    ggplot2::theme(legend.position = "bottom") + </span></span>
<span id="cb3-3311"><a href="#cb3-3311" aria-hidden="true" tabindex="-1"></a><span class="in">    change_plot_aes_point()</span></span>
<span id="cb3-3312"><a href="#cb3-3312" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-3313"><a href="#cb3-3313" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3314"><a href="#cb3-3314" aria-hidden="true" tabindex="-1"></a>We see that in this case the binary risk category has distinct distributions</span>
<span id="cb3-3315"><a href="#cb3-3315" aria-hidden="true" tabindex="-1"></a>for the G2M checkpoint for all molecular subtypes, in particular with the </span>
<span id="cb3-3316"><a href="#cb3-3316" aria-hidden="true" tabindex="-1"></a>basal like. </span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
      &nbsp;
    </div>   
    <div class="nav-footer-center">
      &nbsp;
    <div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/chronchi/molecular_landscape/issues/new" class="toc-action"><i class="bi bi-github"></i>Report an issue</a></li></ul></div></div>
    <div class="nav-footer-right">
      <ul class="footer-items list-unstyled">
    <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/chronchi">
      <i class="bi bi-github" role="img">
</i> 
    </a>
  </li>  
    <li class="nav-item compact">
    <a class="nav-link" href="https://twitter.com/chronchi">
      <i class="bi bi-twitter" role="img">
</i> 
    </a>
  </li>  
</ul>
    </div>
  </div>
</footer>




</body></html>