<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.0.37">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Molecular landscape: a new framework for personalized medicine in Breast Cancer. - 2&nbsp; A new framework for personalized medicine: integrating</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./validation.html" rel="next">
<link href="./surv_analysis_estrogen.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>

<script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="site_libs/datatables-binding-0.22/datatables.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="site_libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet">
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="site_libs/nouislider-7.0.10/jquery.nouislider.min.css" rel="stylesheet">
<script src="site_libs/nouislider-7.0.10/jquery.nouislider.min.js"></script>
<link href="site_libs/selectize-0.12.0/selectize.bootstrap3.css" rel="stylesheet">
<script src="site_libs/selectize-0.12.0/selectize.min.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet">

  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
    <div class="container-fluid d-flex justify-content-between">
      <h1 class="quarto-secondary-nav-title"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">A new framework for personalized medicine: integrating</span></h1>
      <button type="button" class="quarto-btn-toggle btn" aria-label="Show secondary navigation">
        <i class="bi bi-chevron-right"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Molecular landscape: a new framework for personalized medicine in Breast Cancer.</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/chronchi/molecular_landscape" title="Source Code" class="sidebar-tool px-1"><i class="bi bi-github"></i></a>
    <a href="https://twitter.com/intent/tweet?url=|url|" title="Twitter" class="sidebar-tool px-1"><i class="bi bi-twitter"></i></a>
</div>
    </div>
      </div>
      <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">Preface</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./intro.html" class="sidebar-item-text sidebar-link">Introduction</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./surv_analysis_estrogen.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Estrogen receptor status is continuous, not dichotomous</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./pca_merging.html" class="sidebar-item-text sidebar-link active"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">A new framework for personalized medicine: integrating</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./validation.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Validating the molecular landscape</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./scoring.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Pathway activity in a personalized context</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./discussion.html" class="sidebar-item-text sidebar-link">Discussion</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./references.html" class="sidebar-item-text sidebar-link">References</a>
  </div>
</li>
    </ul>
    </div>
</nav>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#transcriptomics-and-qpcr-normalization" id="toc-transcriptomics-and-qpcr-normalization" class="nav-link active" data-scroll-target="#transcriptomics-and-qpcr-normalization"> <span class="header-section-number">2.1</span> Transcriptomics and qPCR normalization</a>
  <ul class="collapse">
  <li><a href="#selecting-stable-genes" id="toc-selecting-stable-genes" class="nav-link" data-scroll-target="#selecting-stable-genes"> <span class="header-section-number">2.1.1</span> Selecting stable genes</a></li>
  <li><a href="#calculating-the-normalization" id="toc-calculating-the-normalization" class="nav-link" data-scroll-target="#calculating-the-normalization"> <span class="header-section-number">2.1.2</span> Calculating the normalization</a></li>
  <li><a href="#checking-robustness-with-respect-of-number-of-houesekeeping-genes-available" id="toc-checking-robustness-with-respect-of-number-of-houesekeeping-genes-available" class="nav-link" data-scroll-target="#checking-robustness-with-respect-of-number-of-houesekeeping-genes-available"> <span class="header-section-number">2.1.3</span> Checking robustness with respect of number of houesekeeping genes available</a></li>
  </ul></li>
  <li><a href="#integrating-tcga-and-metabric-using-pca" id="toc-integrating-tcga-and-metabric-using-pca" class="nav-link" data-scroll-target="#integrating-tcga-and-metabric-using-pca"> <span class="header-section-number">2.2</span> Integrating TCGA and METABRIC using PCA</a></li>
  <li><a href="#the-integration-is-robust-with-respect-of-selection-of-samples" id="toc-the-integration-is-robust-with-respect-of-selection-of-samples" class="nav-link" data-scroll-target="#the-integration-is-robust-with-respect-of-selection-of-samples"> <span class="header-section-number">2.3</span> The integration is robust with respect of selection of samples</a></li>
  <li><a href="#normalization-is-necessary-to-reduce-batch-effects-in-other-components" id="toc-normalization-is-necessary-to-reduce-batch-effects-in-other-components" class="nav-link" data-scroll-target="#normalization-is-necessary-to-reduce-batch-effects-in-other-components"> <span class="header-section-number">2.4</span> Normalization is necessary to reduce batch effects in other components</a></li>
  <li><a href="#molecular-landscape-is-characterized-by-several-pathways" id="toc-molecular-landscape-is-characterized-by-several-pathways" class="nav-link" data-scroll-target="#molecular-landscape-is-characterized-by-several-pathways"> <span class="header-section-number">2.5</span> Molecular landscape is characterized by several pathways</a></li>
  </ul>
<div class="toc-actions"><div><i class="bi bi-github"></i></div><div class="action-links"><p><a href="https://github.com/chronchi/molecular_landscape/issues/new" class="toc-action">Report an issue</a></p></div></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title d-none d-lg-block"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">A new framework for personalized medicine: integrating</span></h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
</div>



<div class="quarto-title-meta">

    
    
  </div>
  

</header>

<p>In the next few chapters we will show a new framework for personalized medicine. This is motivated by the fact that we know already some patients are more sensitive to endocrine therapy than others, but we don’t know what are possible treatments. Moreover, it is very difficult to compare, molecularly, patients among themselves.</p>
<p>The PREDICT tool <span class="citation" data-cites="Wishart2010">(<a href="references.html#ref-Wishart2010" role="doc-biblioref">Wishart et al. 2010</a>)</span> provides a tool for practitioners to calculate survival risks among patients with similar clinical characteristics. This is very important as it shows how to combine data from previous patients to guide further treatments of new patients. However, no additional information is given, such as alternative pathways to target or what might be different from other patients.</p>
<p>Other tools that help in the clinics are the molecular signatures developed by companies <span class="citation" data-cites="Wallden2015 vandeVijver2002 Paik2004">(<a href="references.html#ref-Wallden2015" role="doc-biblioref">Wallden et al. 2015</a>; <a href="references.html#ref-vandeVijver2002" role="doc-biblioref">Vijver et al. 2002</a>; <a href="references.html#ref-Paik2004" role="doc-biblioref">Paik et al. 2004</a>)</span>. These signatures are intended to be used with ER+ BC patients and depending on the tool, on node negative, post-menopausal women. Based on the set of genes, a risk score is assigned or each patient. The score means if a patient would benefit from additional chemotherapy besides the usual endocrine therapy received for either 5 or 10 years. They also lack the additional information of what are possible pathways involved and alternative treatments beside chemotherapy.</p>
<p>There are a few challenges to overcome pathway analysis for patients individually and how to compare patients molecularly. There is no tool that allows to integrate patients in a continuous way. Usually integration is a one step procedure and cannot be updated. The usual tools are <span class="citation" data-cites="Risso2014 Zhang2020 Fei2018">(<a href="references.html#ref-Risso2014" role="doc-biblioref">Risso et al. 2014</a>; <a href="references.html#ref-Zhang2020" role="doc-biblioref">Zhang, Parmigiani, and Johnson 2020</a>; <a href="references.html#ref-Fei2018" role="doc-biblioref">Fei et al. 2018</a>)</span> and they do not provide batch effect removal for new samples. The only way is to re-run the procedure together with the new sample. The problem of doing this is usually you don’t have enough samples to estimate batch effects across groups and therefore correct it.</p>
<p>This chapter show how to perform PCA projection using samples from different datasets and how new samples can be introduced without retraining of the PCA. We first introduce the notion of a new normalization that depends on housekeeping genes. After using this new normalization, we proceed with PCA on a subset of two cohorts using the 1000 most variable genes. We then validate the integration with new samples from a completely different cohort.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>* The library is already synchronized with the lockfile.</code></pre>
</div>
</div>
<section id="transcriptomics-and-qpcr-normalization" class="level2" data-number="2.1">
<h2 data-number="2.1" class="anchored" data-anchor-id="transcriptomics-and-qpcr-normalization"><span class="header-section-number">2.1</span> Transcriptomics and qPCR normalization</h2>
<p>In this first step we introduce a new normalization. The rational is that we need a normalization procedure that is applied in a single sample only, similar to logCPM, but that can be applied to microarray too. In this case, given a set of samples, we start by ranking the genes in each sample. We then normalize the ranking to be between 0 and 1, i.e., we divide by the total number of genes available in that sample. Then, given a list of 50 stable genes across cancer <span class="citation" data-cites="Bhuva2020">(<a href="references.html#ref-Bhuva2020" role="doc-biblioref">Bhuva, Cursons, and Davis 2020</a>)</span>, we calculate the average ranking of these genes in the sample and then for each gene in the sample we divide by this average.</p>
<section id="selecting-stable-genes" class="level3" data-number="2.1.1">
<h3 data-number="2.1.1" class="anchored" data-anchor-id="selecting-stable-genes"><span class="header-section-number">2.1.1</span> Selecting stable genes</h3>
<div class="cell">

</div>
<div class="cell">

</div>
<p>The first step is to check what number of these 50 genes are available on TCGA and METABRIC, as these datasets will be used for training the PCA projection. The total number of available genes is 44. <a href="#fig-stable-genes">Table&nbsp;<span>2.1</span></a> shows the name of the genes available in TCGA and METABRIC.</p>
<p>Now we select the 1000 most variable genes across METABRIC and TCGA. Reducing the number of genes used when calculating the PCA helps in the future. Several publicly available datasets don’t have many genes available, therefore reducing the number of genes increases the chances of using this approach in other datasets.</p>
<p>Moreover, the use of a large number of genes is used because if some of them are not available. We show in later sections that if some number of stable genes are missing, it will not affect the normalization.</p>
<div class="cell">
<div id="fig-stable-genes" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">

<div id="htmlwidget-c7088ce56ec4b1792f1c" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-c7088ce56ec4b1792f1c">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44"],["ASB7","C2orf42","CNOT4","RBM45","NRF1","ZNF143","AMMECR1L","SMARCAL1","WDR33","ZNF394","UBE3B","SFSWAP","PPHLN1","SART3","CPSF6","CELF1","IWS1","STAMBP","ERCC3","RBM22","TOR1A","CNOT2","TIAL1","HDAC3","RSRC2","DDX42","TARDBP","RNF10","SRSF1","KHDRBS1","ZNF207","DDB1","SF3B1","SF1","UBE2I","UBE2D3","RAB1A","SRSF3","SFPQ","EIF4H","PCBP1","HNRNPK","DYNLL1","HNRNPA2B1"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>genes<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p></p><figcaption class="figure-caption">Figure 2.1: (<a href="#fig-stable-genes">Table&nbsp;<span>2.1</span></a>) Available stable genes in both TCGA and METABRIC cohorts.</figcaption><p></p>
</figure>
</div>
</div>
</section>
<section id="calculating-the-normalization" class="level3" data-number="2.1.2">
<h3 data-number="2.1.2" class="anchored" data-anchor-id="calculating-the-normalization"><span class="header-section-number">2.1.2</span> Calculating the normalization</h3>
<p>We will create subset of the summarised experiment objects to save these information. Moreover, since a summarized experiment object is being used, we can save the new normalization in the <code>assays</code> and the average ranking scores of the housekeeping genes in the column data <code>colData</code>. This way it is very easy to retrieve the information for the patients.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-comparison-sd-cv" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="pca_merging_files/figure-html/fig-comparison-sd-cv-1.png" class="img-fluid figure-img" width="672"></p>
<p></p><figcaption class="figure-caption">Figure 2.2: Comparison of coefficient of variation and standard deviation scores.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>They are highly correlated values, so we stick to the coefficient of variation as it makes more sense in this kind of analysis.</p>
<div class="cell">

</div>
<p><a href="#fig-sd-stable-genes">Table&nbsp;<span>2.3</span></a> shows the top 1000 genes along with their standard deviation. The stable genes are also added to the list.</p>
<div class="cell">
<div id="fig-sd-stable-genes" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">

<div id="htmlwidget-1e4eb5a961e25a236b4e" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-1e4eb5a961e25a236b4e">{"x":{"filter":"top","vertical":false,"filterHTML":"<tr>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0.018382631544801\" data-max=\"0.368915306302523\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0.028911914173166\" data-max=\"0.332773405977009\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0.026776265846626\" data-max=\"0.348900294758046\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n<\/tr>","data":[[0.0940914238185255,0.117693563442433,0.0948097122810194,0.112005049273451,0.220017274674532,0.163894940799199,0.150627739027483,0.0962987209766597,0.0955762652841267,0.128886664178946,0.100035987701397,0.160710120454481,0.114463619672776,0.0906908990829836,0.117875812408174,0.153106092543963,0.0243240963262126,0.143505693454104,0.098586614124342,0.117782254662222,0.10011224107914,0.166225791340768,0.256217188091341,0.0837193222410926,0.109930258755372,0.179170537029975,0.164509561608192,0.0997977332390348,0.126940477982802,0.162898690374686,0.123470490899357,0.0265234063643107,0.110261671506693,0.0964486841135979,0.0985221102404845,0.096118149011729,0.111983400833197,0.112565966481571,0.120741689707563,0.169121481380003,0.126308833099605,0.148597930720569,0.12090672706907,0.106552203882421,0.148841049974628,0.0950434238905944,0.103928669454351,0.170371122492248,0.154034252594713,0.103950367685875,0.189011515811912,0.11245557225661,0.108249809187914,0.125894807529986,0.166437792236518,0.182750273072006,0.114071200357164,0.11323515832792,0.0360556307877073,0.164475998436122,0.144864696252243,0.138537813435389,0.102241786546309,0.128806404941264,0.108901460740426,0.149536233446081,0.12090891468476,0.2174282518082,0.214636457097756,0.133462518838899,0.0947209854778981,0.109774035149822,0.142627663358269,0.0951001578408046,0.120508424336605,0.107425870073659,0.129591570731113,0.162753956903769,0.147340314036085,0.0903762941560711,0.124649712617043,0.111587367476398,0.10377608728125,0.106352692784925,0.146172298485385,0.105161795712176,0.100448970634991,0.114765582115001,0.168030714668868,0.0961968876042237,0.153303098121652,0.109141449158468,0.100120047636832,0.0297007297049953,0.11273399538818,0.118474918508162,0.101065893142892,0.104483892554086,0.136340853670329,0.112110941459443,0.140709810907178,0.126810334038769,0.124735900075262,0.129978894435638,0.15312738055794,0.128685835080148,0.034553997051454,0.118574070542645,0.125403021935201,0.101596861781221,0.138709361757426,0.0947427509257063,0.107689406003718,0.106036255099807,0.119139806175419,0.109890921648198,0.0292620510788451,0.106491186539421,0.19551172098629,0.0925300039594485,0.112140955991095,0.115395828529056,0.16074395376581,0.107458072143173,0.111401073167608,0.127979457517884,0.0368403513836977,0.113220089952273,0.125274881087115,0.118546768845502,0.140463886856771,0.168274850439294,0.114415058225268,0.119129213515525,0.219403316296829,0.109321898628815,0.103817708216984,0.121278586209397,0.111561269676535,0.123047344049389,0.147067805164057,0.122295764653616,0.126387001783711,0.101073505949878,0.0922605924798418,0.110271209588324,0.125323016347087,0.130394118211075,0.117638683941253,0.128490325528829,0.131355136995481,0.150911705846701,0.156298184356284,0.0957378149484573,0.137745800237785,0.139668180179532,0.11766726135835,0.14509022884351,0.0904829361154789,0.104615137027548,0.105828951928046,0.0970535752708678,0.115869688812056,0.165599619916773,0.241821706036916,0.131491762578634,0.142980393959422,0.112156594487787,0.222479053163541,0.130098015488272,0.136159066478683,0.116711514911238,0.101218611694772,0.0925454986446155,0.103154488893509,0.1038199650526,0.107720536717903,0.097622530293157,0.103956208142446,0.0959196371040997,0.132669334519419,0.134700845795706,0.133833045436034,0.033395408776306,0.193644306423635,0.104677612383038,0.1181346066202,0.162004978681107,0.216743220199856,0.0953367695180632,0.117322909936114,0.108813392426149,0.0987234081233707,0.176598994449596,0.238326765164089,0.0965987323445077,0.132703463692239,0.136225107926355,0.097152931272864,0.123662670164687,0.109393825133466,0.161309513487767,0.119218803436353,0.104717749170916,0.109871104441864,0.215756493638122,0.161163310186792,0.101028169640301,0.114099540551543,0.12605470070605,0.116327938633452,0.125007043642211,0.11114301658003,0.158734147066166,0.165444695111098,0.125972567134464,0.177083740788042,0.092610171700586,0.132686890914364,0.117269545018702,0.099358911669144,0.118676689919886,0.109189881696736,0.0226391673146897,0.142874246227988,0.113192004841013,0.0809259748700947,0.11365743401028,0.225095576324755,0.102835794481446,0.0256660625452076,0.21607797818483,0.20506398024882,0.111655495190076,0.177643208231709,0.139600160023776,0.0938901027022193,0.13315916726307,0.194928036917702,0.222529456117632,0.12517389929639,0.160019901337241,0.121919823954411,0.184259397827458,0.0770471636609752,0.14377196205974,0.0253008696630999,0.13048572706246,0.131281837258753,0.0994742337254656,0.147666520054188,0.126006077088509,0.154717902113448,0.091505846287213,0.176802151771479,0.0873238743407052,0.0871998152526244,0.105274862128836,0.105735689757332,0.368915306302523,0.104331732045395,0.13978513836517,0.0260026646056224,0.122095319898414,0.106742102827297,0.114988375437831,0.126332388960205,0.0357529158339022,0.0275135616940665,0.117410227247805,0.0890775745553097,0.118070463313864,0.110682857688159,0.0986672851728163,0.142836695093669,0.118415404733371,0.101083877304725,0.116529770171226,0.0231851033970141,0.102526508531426,0.0859511121831538,0.162086225888432,0.111880492983217,0.140084126300863,0.135757081825028,0.115568338405311,0.11343026375388,0.105690821261541,0.177611610336988,0.180279744534393,0.101438690349457,0.173202314497338,0.166615414356134,0.137175833853938,0.10235519018509,0.131504763997494,0.113378862919399,0.109687598912195,0.130531964183761,0.172135471969247,0.122274176317931,0.127395893756601,0.115406035900724,0.102480033620406,0.0278487878123212,0.107408338114084,0.15957047724892,0.124521567298217,0.108974779964276,0.103630823498497,0.0301486637906608,0.118524309297681,0.104880405376708,0.106378956696421,0.0221381659406993,0.159627837101031,0.100780472195886,0.117429288988899,0.129334066984342,0.15454325291307,0.127752505372749,0.169174216561703,0.1090085541857,0.134808301057683,0.19052556999388,0.117547997952708,0.106318291164174,0.126944701202373,0.194178886470743,0.133683849064295,0.100986431918567,0.11150236064097,0.100056908386465,0.1328428686145,0.137746341480592,0.19010576992384,0.127595726521222,0.119573675849609,0.117306921355014,0.177354089814678,0.159247713562946,0.104683105274887,0.163262746082484,0.116616236696037,0.111752779932686,0.110226320818539,0.144109158904589,0.143212596045434,0.108191937854813,0.0200628118193525,0.187605072408682,0.138246300409366,0.107349649590442,0.0207496801653205,0.0967424874381317,0.11639975051885,0.0832489803291029,0.10757958451449,0.118442818397217,0.022221426602762,0.0890653670365049,0.136367350382224,0.126742010902721,0.101484042944783,0.088366623787312,0.113879805032124,0.124671452569795,0.130036505258475,0.115569671392401,0.153425255326697,0.101428440260975,0.179329680562297,0.15786548549799,0.23405282045882,0.107567798835894,0.105556540047821,0.189924459106997,0.0315828371204833,0.137166309130335,0.16751412654644,0.10268613970256,0.365027183539083,0.209991461118408,0.161155341866337,0.108675052015722,0.0954090589244482,0.18965423782311,0.129152289370882,0.110510926662761,0.123034801466012,0.132522319388781,0.117255832336314,0.138516187215793,0.0960498683131828,0.135306154661725,0.092763473171975,0.170449329773552,0.137828333332568,0.116162899683994,0.153676519280531,0.115773883568349,0.126771818746895,0.110397452039994,0.115078796077079,0.0923071474631764,0.095599911790889,0.115016975243033,0.0988607283773852,0.245209944762613,0.155594744817209,0.112259725874733,0.128293940713135,0.0983625102565504,0.185424494341868,0.0880869462433834,0.106662687182644,0.135715805816757,0.137339857610105,0.14353547426534,0.108082711179741,0.133671429991719,0.112817481412847,0.186162822007504,0.155419226885632,0.0908968350868524,0.141165220280907,0.104813810921904,0.105213398707637,0.10930383020151,0.0950948883600947,0.255083819761424,0.120408786307839,0.150325943645711,0.148015424865067,0.117876120018859,0.174450458098116,0.106293997337863,0.127120140342683,0.141916763222907,0.0967811626827068,0.121659227041102,0.157076190862818,0.109555464077371,0.0971135900790241,0.146052361080562,0.126906890115681,0.154179808349768,0.0903988282596518,0.115757277902905,0.209813677030624,0.0283777444550292,0.135245687172618,0.128242129511378,0.0995841906276212,0.15048463196985,0.181032822351986,0.104763880319267,0.142006902054369,0.109378226755843,0.137575879328868,0.134009489907009,0.148458254117448,0.117209580331278,0.112052028283133,0.105765365604431,0.164996155911515,0.120920003422245,0.152688595765257,0.177433378907609,0.167741983041465,0.12396726284195,0.156687612129297,0.148305532080881,0.178250829724416,0.112546370714059,0.102511318183983,0.115246569563905,0.0985807037111683,0.106026106958107,0.0256851849488699,0.141650181883704,0.127939860204728,0.0212111359877665,0.10456629529231,0.107364995649655,0.0307685594361872,0.0913667262655693,0.120650476348218,0.0981012382909036,0.228679449227459,0.120490547124565,0.0986796040178079,0.123835123979873,0.120654169593787,0.12920154967339,0.0957148896114074,0.102830173804317,0.209372144684352,0.121443238236079,0.0965348301735988,0.126196519421615,0.144463614728796,0.123765170101247,0.120873188700119,0.0284572508542065,0.117304757126786,0.152624652841041,0.0289669374910534,0.149209556425227,0.161052807629749,0.118933497992884,0.195942571759594,0.157521889951014,0.108746159455451,0.160566217050473,0.103900079168962,0.154110465929289,0.143941859142438,0.0988780818524892,0.0945237429190714,0.104788451535496,0.194472816795871,0.117165024652192,0.146993079227132,0.0966216509023176,0.113848158732329,0.144244129723322,0.105893365478982,0.114842370706652,0.118588911597371,0.108210248054825,0.12660667660239,0.102853806340499,0.123092368090947,0.123197834037469,0.123339176900792,0.120212015564345,0.119202654596287,0.226920606742516,0.109513354900114,0.0963185958672447,0.0926196925306708,0.147838701894169,0.125112326679781,0.105521424358533,0.119546508010458,0.127607192058009,0.143529862498299,0.0877128028119911,0.0964499092194632,0.133406552162717,0.148453859506749,0.287121139382167,0.123859309155607,0.127663024211912,0.0983608020612752,0.104268410625474,0.197680545069791,0.034555808417985,0.129213500135798,0.134657690839155,0.14648302208588,0.1126665666467,0.12456863987867,0.107897943621862,0.106241277624355,0.127066440658638,0.113124448998045,0.113100078686497,0.0997245491443908,0.154373876261315,0.123303220596203,0.190972668369635,0.142217933723261,0.101943845115809,0.150690509041694,0.112723037358287,0.114193073475576,0.176844153834333,0.141743291201626,0.110207482948063,0.117140795625508,0.164934383457502,0.172590371141162,0.0976176802247606,0.135492408297581,0.129295851914073,0.134203426792427,0.193275069217096,0.0888930375735711,0.151679027696016,0.0255510052497129,0.119933518064534,0.163877827399754,0.117354285437082,0.110077253688625,0.118735176145076,0.17621547983861,0.173992621531515,0.143845476316372,0.108562765032855,0.0397590179129245,0.0731101675824166,0.108536371042868,0.104574581130214,0.100412721465515,0.0933795559676658,0.119249244220334,0.104781129412431,0.243568387432787,0.0264178906996927,0.182785405749851,0.108860439539987,0.127950260825742,0.126136273221574,0.122272367817225,0.132754967619639,0.130467671237755,0.116913666676764,0.144806731355342,0.122513488962747,0.101226113031527,0.147381322125119,0.164395512454005,0.137784047718195,0.176558929971097,0.111542409667105,0.134073347028523,0.143487350887172,0.110062729958394,0.115395401428415,0.108997504434337,0.131280617092731,0.0268919550504291,0.114817852738119,0.0974560694062233,0.102094053498111,0.149096184641041,0.14247746291488,0.119842756457873,0.130368802761722,0.124300467913282,0.0885885215075061,0.187663129324628,0.161233698480024,0.163125935602606,0.114062286548883,0.129401395422785,0.119733516262967,0.203506522319229,0.122816194227589,0.121984552686154,0.172305320207967,0.104501347875017,0.149828113504474,0.151314011412326,0.106467588289091,0.278458062769674,0.116468171245021,0.0980136348458865,0.0791260269920429,0.133890162947989,0.332793724235416,0.123555656374358,0.258927151631277,0.324244099811826,0.211994050330764,0.13990764214414,0.135965314069035,0.122068819134815,0.109398154329963,0.131887524029843,0.0384297650060977,0.10573920599616,0.166249101295453,0.122673325884812,0.106307338208293,0.153962649823319,0.197383071052804,0.125716189452765,0.133067514936078,0.216057943743017,0.109734621920922,0.160483532329,0.115055923903436,0.134450368237014,0.115968868060467,0.123172916580733,0.101752398272603,0.112510457447928,0.153114839784476,0.132868479644513,0.12914911476357,0.115336919916841,0.125699300521904,0.171863435259866,0.138663551009042,0.209637564428009,0.173317797890786,0.144226005639663,0.137924075287286,0.154747782950165,0.0290608809820278,0.0311119120457381,0.26062562215781,0.1672887279104,0.102878447726907,0.146966481659054,0.109606360012404,0.118389230952977,0.116641101222228,0.18701967502549,0.117019204074406,0.123795052462418,0.129827490460339,0.188927079776973,0.108385239642698,0.0971323390357696,0.109033513313372,0.162482030752551,0.122463012940349,0.119696787005887,0.112610011303364,0.159163069316094,0.181708155385861,0.112743158276174,0.123894817671126,0.115888696810416,0.118824420167256,0.141119774069738,0.126436402846951,0.0867726827161202,0.143393385703809,0.0931833861607868,0.110965407960421,0.121177303114259,0.106478535975266,0.0183826315448015,0.132045158816851,0.128217215806913,0.105534452978945,0.122377812008822,0.127899156169227,0.130147967950053,0.158850139920489,0.148246794029721,0.11532643579867,0.130900087768056,0.102360094740222,0.111777540514441,0.138252358002079,0.105020856018404,0.154950310234984,0.105541378806013,0.121056500970588,0.0913447771706274,0.122410650620309,0.0966305418723977,0.0332296792831,0.152633487353506,0.135472558559268,0.114441143782565,0.139707664048956,0.156904167635171,0.111827897116134,0.149232522032952,0.112018951228389,0.14127423807977,0.13156216511282,0.119956606192643,0.107943482243617,0.139638251965881,0.15972833504251,0.122975612184199,0.140454650100632,0.218660629143602,0.112205474018094,0.159992208035408,0.133147528411881,0.168536350796077,0.118903499567649,0.0966474448914523,0.0892690016535822,0.032730814076289,0.0207077495757628,0.103786374646595,0.105083667977414,0.105741627465998,0.188145707074539,0.1584251487075,0.179854845659584,0.128988211219327,0.1635936411177,0.139009929918589,0.126373216545477,0.113110988134648,0.17192718345578,0.112275807377043,0.119715051094239,0.103405870183171,0.171619285787976,0.143570693216653,0.124256276290797,0.0320118532094051,0.112788520726608,0.148435551705584,0.12338039158191,0.130365655642941,0.111162471734099,0.10216726251404,0.0996518574392652,0.0970580311298052,0.099682875564833,0.257899885621472,0.13018627978889,0.252899088376041,0.132798242715425,0.128900102905516,0.173182494359428,0.120638399626527,0.161936652681707,0.102762455338942,0.104444306784386,0.112776007836842,0.218546973906874,0.224260326214741,0.109182396853238,0.10585119492351,0.0312491046728922,0.12170031346082,0.12554989697036,0.116846198821319,0.0823683826567809,0.0874595663600392,0.142906645470188,0.2024057830024,0.104356474491386,0.114305626714432,0.178764281413372,0.13839660596293,0.104211989768125,0.153110328926713,0.147288758526627,0.143758059780128,0.227956655454659,0.128953236272108,0.131966421777244,0.141600221668433,0.152225383534164,0.183714094634913,0.0949726402484264,0.21923046447991,0.270782727060278,0.10858612138513,0.120599549725464,0.163677669178528,0.124076575112458,0.118174972758321,0.148455788203668,0.13191994689881,0.120483452697706,0.159088514764416,0.183851149681319,0.220054347229936,0.15562967733168,0.135089965327777,0.149182121907454,0.120209613749178,0.183491952071257,0.108036120604493,0.118703105274942,0.11885280054388,0.105991352988789,0.179815479214893,0.0963556512057279,0.126635432172374,0.108973638954445,0.114849523142129,0.111367796168641,0.120047821131191,0.13455858195131,0.108843028477743,0.103934500095749,0.165291865407358,0.163843154836911,0.105756858685355,0.158586337267095,0.162970184965483,0.122429191493044,0.116559289937864,0.152471497370774,0.101876177256425,0.159462038912391,0.135112017685159,0.0980893926926393,0.196399035376597,0.110029002812387,0.17035517314858,0.122811647345422,0.134056677147849,0.115038204457903,0.0993290741826718,0.106856737691341,0.0913637659504166,0.101402218491081,0.102660497694999,0.10566156024049,0.100652392015733,0.186345961281885,0.150969640633545,0.100058915416606,0.153908947282641,0.142539100952075,0.107581468558191,0.117378269905261,0.0469742550376633,0.102276331557281,0.189414346650361,0.125959740077177,0.159301155693674,0.107131503429468,0.137190311159604,0.106814400821041,0.0977797449634017,0.125078134852153,0.135079376656943,0.220783789627036,0.149553636281284,0.127383071962338,0.1509993239632,0.126380176394945,0.137459803968229,0.121377333302735,0.116648592790642,0.144370152424126,0.107587250528354,0.137038553581329,0.116555435623541,0.266326545711024,0.0993640112591245,0.177440919883767,0.0981780030663569,0.226957367294604,0.106359424695875,0.274776459450201,0.189408986258743,0.150531879874421,0.152429792149151,0.10244786744169,0.125295871052904,0.116598601113872,0.115956005492328,0.116771719888944,0.151644784937943,0.0950788767448713,0.155272262216396,0.140275531674607,0.10829294521572,0.141413578724767,0.135608322206381,0.128385632061388,0.149463981519314,0.232518881888637,0.0446760742218352,0.171165425305825,0.186155612055219,0.138452228648368,0.103697395781206,0.175758083156874,0.143631183499009,0.123184420611597,0.136034172815872,0.137185738463345,0.118975616442096,0.110214786091865,0.146416127744669,0.129955975188517,0.111762500053888,0.102862189937976,0.151341935812955,0.106937594125446,0.152416561692411,0.115998340284897,0.115110012813012,0.13089681642532,0.11703559247627,0.176045704958121,0.102927710479972,0.112523825211929,0.0975819027713601,0.120519775566732,0.0319333778583423,0.116877605867704,0.108675823106915,0.114118534837399,0.17301991363947,0.159811167340593,0.0946613553373381,0.0908916712931144,0.230292080837098,0.128996342037967,0.10738957524914,0.204671328623232,0.105381627777869,0.0702900348981964,0.120861418158343,0.142941455315412,0.105034500223996,0.135234427492657,0.140900163191156,0.207733169711448,0.105740029791955,0.101988618149111,0.182396079065308,0.140565306430686,0.184612020075795,0.123957895717558,0.100805401448799,0.17768263219231,0.107885382213659,0.0986094828365959,0.109599031543749,0.112576494423253,0.170440410278777,0.110931905387364,0.184266434018637,0.169112321421422,0.192129826087525,0.168039890219596,0.115538832757915,0.112948794158465,0.103499634655839,0.120114093642085,0.122004060869102,0.132854440636179,0.142801884516157,0.117710923324473,0.124636547249607,0.102307395857996,0.17407202162592,0.104784764286802,0.20444406129399,0.163942707483413,0.11946591317954],[0.116771502402258,0.110568477944759,0.101597329944268,0.11008666604144,0.221899667673518,0.126251647789346,0.134401109670467,0.112767522708528,0.113248849762843,0.148942947364902,0.122549502077574,0.152874292564407,0.143816640444656,0.105204917985658,0.10854778488169,0.106028876374705,0.0313966648210779,0.119711778622898,0.121414533708129,0.0952761785559999,0.158899799082937,0.103533235977313,0.264016982930087,0.114324306503232,0.12009809777583,0.118401705655323,0.0943405200798448,0.101778504497718,0.113510104211859,0.133552900901032,0.119572441674651,0.0505846427270687,0.0955856340192848,0.10858303610713,0.119504785516107,0.107235705643542,0.127162118410186,0.109036613162737,0.142784070126865,0.0886288860774667,0.122027699467887,0.135389996013287,0.111962337198851,0.0978629489043894,0.127350272346364,0.101753607283899,0.120662994037216,0.115633500530739,0.139452815122628,0.107025239548914,0.178162409208312,0.109017562953373,0.138499395223609,0.114216443685489,0.139561948064507,0.183296293205807,0.0939609048272608,0.0824661620203921,0.0452644867224631,0.170291070500452,0.0770768061173889,0.115640843141724,0.110820168387715,0.108247816714317,0.126636116313168,0.136599466349499,0.12235777905925,0.141941592846525,0.175073256075199,0.0840822011487221,0.114859574693683,0.119890083066663,0.109217073377591,0.113319244014027,0.124117881421702,0.0994273949959401,0.0849119857066792,0.153464486585323,0.148237118706971,0.106484674420863,0.107818027179812,0.119682561078615,0.118225910673307,0.102222080705658,0.0955081783935327,0.112111111193297,0.0965055905444108,0.0993786145682779,0.170271946646883,0.106006608415673,0.155103388283522,0.118272084131693,0.11273512606532,0.0347062269588929,0.114604878468961,0.136570947936964,0.119031774873164,0.106556939097748,0.105396079277157,0.113679926751803,0.121092115529577,0.133805557781613,0.0832300546502343,0.12861394378599,0.073919410149548,0.111150866964339,0.0386560853977512,0.080633494076928,0.075250563745769,0.0939605928820277,0.103578446403191,0.101239809875901,0.106784947272934,0.106430604653699,0.105643517380579,0.103999464617636,0.0319439841071596,0.0941896324889782,0.175865362929789,0.103456063870771,0.115244284094589,0.100696933832084,0.122472161075844,0.113511650747595,0.0915932233982704,0.105347238860632,0.0415189071013226,0.128620006655878,0.0807974821235224,0.142466580291046,0.184578192406554,0.133362995466177,0.0881259603478402,0.125615720422682,0.222158096522817,0.107872864780828,0.096501528190062,0.112980974489534,0.110566201330127,0.133205597474743,0.132724855027051,0.075481101682725,0.129055178520098,0.109362847806745,0.115506610649687,0.0977158054398825,0.0897918402803258,0.11419975421786,0.0978300330893128,0.142593364404467,0.138400722873115,0.163179622955774,0.0854795692395554,0.116325659812654,0.0921777344205534,0.114460141343649,0.106225339900406,0.133971725689908,0.106342724548743,0.112985694666723,0.105709069504205,0.100118854893511,0.106515362855138,0.1666170542833,0.230836351550745,0.112063992455578,0.0861646130346451,0.0892406336334422,0.0515343089864258,0.11311178498756,0.11899669083977,0.133683643913238,0.109288862567605,0.117254779498745,0.101262047953098,0.127642385022873,0.113129702386147,0.121295102936702,0.104719517713574,0.104523903058231,0.100341223736494,0.155658606074091,0.112901946400636,0.037929125774154,0.115172816209325,0.120802016547155,0.0942116464470963,0.145278409743163,0.1032058208234,0.105908684697829,0.118591452992713,0.122607234969632,0.0993813743869494,0.192014507037053,0.228269855221284,0.112011243658553,0.155074147230357,0.103592207490029,0.110625320219324,0.116654224275218,0.130039791991288,0.0796679744452529,0.113518491637281,0.123047814633304,0.103379093948383,0.189607286389049,0.150096435579456,0.0994955626145976,0.0882229714119931,0.105133084985271,0.0931541174285453,0.125374339589627,0.0962985895675046,0.155095393766723,0.137182660047309,0.121321936573576,0.0853765177854892,0.113175222962468,0.0752116385903276,0.108099536841457,0.112906875426388,0.121422654639114,0.110095853359869,0.0309133643785638,0.158995197233485,0.0876102550313025,0.121975103133374,0.103161187350874,0.189912400456142,0.104014387923627,0.109859877999999,0.145332796212337,0.17004789495823,0.11580445881831,0.18021913733574,0.158036453755825,0.126479868498843,0.147910270205293,0.196643916485542,0.132302796922155,0.088735716087888,0.162723967466877,0.124572780473673,0.12736770897201,0.133098298385327,0.122827034224921,0.0433989701247487,0.145499797184287,0.109475604363182,0.121846231924572,0.0844956397355323,0.121122705495296,0.146377405739101,0.104647534781445,0.171615626137262,0.111814750147977,0.113294157932872,0.0933645046828479,0.094301876568705,0.289062538726589,0.0951921990315707,0.134702646116194,0.0456300626959534,0.115840346680648,0.103621496247597,0.105434420243204,0.104965125990954,0.0442089772949382,0.0347922901946812,0.115116026348819,0.110877629746979,0.122128509786509,0.129842523102549,0.115442706320281,0.10999804464687,0.13533278757715,0.102950987417288,0.0804659297948919,0.0496005027186403,0.108078380766753,0.122962226450887,0.0531315909277289,0.12181927456006,0.151192812015612,0.126946040229928,0.111249062347795,0.136865148567477,0.113414647214902,0.173882974962456,0.179344069370603,0.0953131177436525,0.198126538458763,0.157421019817268,0.10724982316222,0.107868850203473,0.111555731535456,0.115524933216265,0.0948161887027908,0.0877177983004394,0.155935899646849,0.148656733602873,0.143738043245103,0.106496849125589,0.0989145915092226,0.0605387318274443,0.124807840137437,0.148865615163395,0.119040659935595,0.112448217115781,0.113587155863951,0.0373065184683414,0.0858579466974664,0.103889726188979,0.123071311970829,0.0441240533267615,0.128809437575013,0.107923704471955,0.128407485098222,0.12604795573047,0.154191913526337,0.138840962255051,0.0361127654174525,0.102988181293116,0.108456723746525,0.162064924737921,0.136089783174055,0.0990809416427258,0.13402988166526,0.160239303743969,0.0822818124557687,0.104901597605226,0.12335585001456,0.111450901551181,0.127332973565135,0.177091811415821,0.197552067954228,0.114812359225728,0.129685649582243,0.1132476532161,0.16964233862758,0.11938739996087,0.132953625181913,0.055419785534232,0.127825971146424,0.0902358057444071,0.115797056293777,0.141896650842592,0.128682763997652,0.103490027057678,0.0519157177186306,0.141772896461142,0.146178739520738,0.10430968697921,0.0389679025121715,0.105623311403891,0.113232414880061,0.138805042580849,0.0929332331155527,0.116434791590729,0.071994489909982,0.126864856838641,0.134853709831474,0.088200367709691,0.110275072991756,0.126907265874504,0.0958079355906358,0.127355350421618,0.119223409896841,0.105532911108397,0.17161550150649,0.0997054065187144,0.115340931842711,0.139832025915102,0.210538450039279,0.106329756705856,0.109304567613632,0.064223444205673,0.049519999025339,0.141068519256565,0.145112905601803,0.102554348558981,0.332773405977008,0.199239606936306,0.144660645697151,0.120037764958364,0.13179867099959,0.198383009593718,0.0860834563689093,0.0929317124499529,0.105051996659885,0.159749370464802,0.091114627689094,0.148563701938302,0.10484196597915,0.128208135189675,0.108153429123159,0.133902263663427,0.128974127611457,0.0927092909245623,0.137790442354063,0.115453490194801,0.118786085450446,0.116629746026097,0.127201751727696,0.104624404922253,0.104714440743893,0.116919123716696,0.101017489330603,0.219837605248964,0.145953179833289,0.114514982917923,0.113559981999771,0.100326313931236,0.162410453587301,0.117332348669807,0.0903173796273471,0.0601115369841358,0.125838198305019,0.108024490978526,0.101977021198875,0.103670587074601,0.129562505051058,0.160019856844333,0.142736699878288,0.107345370687529,0.158136051744037,0.113531165583267,0.112544183067503,0.123433760716757,0.1162250152752,0.24930925474527,0.102078478359643,0.094103328282295,0.129329813605058,0.10947663929827,0.148665051982881,0.122063656268463,0.109008977264329,0.141574522772363,0.105204897201176,0.0922253524869275,0.145952487166186,0.120775395801967,0.145507109691091,0.13992565047168,0.107147310813352,0.162988336507939,0.122199042499403,0.263764961283645,0.214525415305083,0.0428421741051879,0.122500907006038,0.0769180172525106,0.116492461625309,0.130042175192324,0.180649296883974,0.0961211271252254,0.155175350487147,0.101036657069177,0.13966203811716,0.116561420190744,0.12267047377802,0.0906931767428806,0.0993600058078082,0.0914912408976635,0.138632480636716,0.117273485079709,0.127947582710044,0.177942708790413,0.132412085083336,0.104873252774711,0.158281341844185,0.151808454758927,0.168162991343694,0.122005726371961,0.11697581242785,0.107716013006759,0.100716888058697,0.0986202047892523,0.0367876925844577,0.101056212531273,0.0811396655315679,0.0541441561756336,0.0909440608539939,0.124904933793802,0.0382967063197363,0.10567302504624,0.132784923592711,0.123779690959109,0.170450968655036,0.0808576098097355,0.107010084712633,0.114281415793931,0.120160383278082,0.141142889167137,0.1087265994966,0.098453981155588,0.212719929217361,0.0815306433629043,0.100822747873214,0.129227301101157,0.148886396209074,0.113426069874755,0.132921549214916,0.0393717808938805,0.104185045987767,0.127581499335805,0.0370525893568099,0.138587640822604,0.157034331112794,0.115518280716687,0.207035377870661,0.0996293779518357,0.107881964853424,0.156109715164631,0.105042389079192,0.12370852985112,0.109136169134863,0.126075213989699,0.120497178291595,0.0961750590466942,0.0915868613552993,0.145858338213621,0.118397854365234,0.108910376935474,0.110867068212102,0.123122774602574,0.0983840972823507,0.114958645653621,0.175919059450659,0.098055490914324,0.117704970264571,0.125715885396355,0.169220072162122,0.0823720101322249,0.0899280726385061,0.142900859777696,0.111010587031856,0.166466965299769,0.0968710545362063,0.136038589670093,0.108607470969505,0.136684863730884,0.0739299680567595,0.121518029262548,0.123706200699436,0.0900912344659294,0.160125923435567,0.111075369512195,0.110790242801024,0.145411363294311,0.12615224697922,0.264148712552481,0.119054575326773,0.0925417813568662,0.11129685314712,0.108653999040678,0.181208218502398,0.0493135712702361,0.112266616015065,0.122342562547481,0.13291014657579,0.116122496945965,0.0777101022105151,0.0986905244437043,0.094081041517606,0.0989976197082543,0.101667126193208,0.129483017376008,0.103558082670422,0.143238407076818,0.105641455764672,0.189397907667141,0.133437919350933,0.112112481193022,0.138844366437757,0.118548154418202,0.10883241063218,0.134515149035767,0.139058416939454,0.105877037296763,0.122405357939922,0.124265947648252,0.153457062035973,0.101773345871816,0.111348100659827,0.141147339035722,0.124508791932087,0.0638899084775402,0.12178523020968,0.14780999018057,0.0447447952720336,0.081626796175695,0.104510929540531,0.104364666295534,0.107737004940958,0.107539763795126,0.128138303777139,0.149697519458143,0.0757096791257108,0.113109758969343,0.0528797881332591,0.147007427969199,0.106598520348094,0.0966552080143631,0.115105188933184,0.110519043103109,0.105701208761171,0.0977695003686577,0.221852162155695,0.0576555949714285,0.156916312913555,0.110417722986913,0.0826371811293454,0.122516838478174,0.168026386231079,0.108294088720195,0.137198804602449,0.102439741335631,0.147373662937113,0.0935248304755724,0.0995132010055277,0.116542759405137,0.133031376409222,0.065720426549183,0.150526372829965,0.117384246524756,0.138841842986808,0.133459764762493,0.106597821580632,0.0973886428590472,0.0963234385862775,0.081312011870537,0.0369169564604649,0.0945817146941288,0.109997129556389,0.118361368562556,0.169648156658407,0.116264859765675,0.091464096447068,0.131805122607309,0.100077051276833,0.119908381939061,0.179830764108636,0.071956233836237,0.150937501291643,0.119765988503938,0.117447398575019,0.0876801407907058,0.144091126966952,0.128191710376989,0.120655025562338,0.176851319737674,0.091150948829425,0.159982178813484,0.170943259066174,0.122540728730602,0.267427633289075,0.0977870301998304,0.112040233846269,0.141923171193811,0.0986340006715519,0.278564219778822,0.0899265413578235,0.208360577719156,0.301523113426627,0.1225679750128,0.108875064244686,0.111840007090189,0.120313117936438,0.105151374635657,0.148539286211755,0.0400074649693687,0.0938891316740634,0.13345479543491,0.1085424679114,0.131408130198001,0.152995369904804,0.14462904707843,0.130703068402519,0.122416644924434,0.136158585073409,0.100190060884095,0.0386460934320191,0.122270077345778,0.135389552754277,0.108210406305791,0.129939305984515,0.106423934555734,0.113517143811684,0.143335410629899,0.124983605583671,0.146684630335722,0.123842656373359,0.0703199209745427,0.159780416041334,0.117300729725328,0.190791933805177,0.178520666045908,0.114159194308229,0.136669395655843,0.153419226842434,0.0369640863828418,0.0352182072920893,0.268055013009687,0.149749113769226,0.113926948711739,0.150090860511319,0.0914970800369357,0.0853810258331822,0.114927151033582,0.179320167793533,0.126302317420422,0.0937729831908519,0.108414253528029,0.165721340302581,0.111490083744985,0.100547533487517,0.0945058316864476,0.103753149349742,0.116775569330888,0.105372731446644,0.114426882028465,0.13662840091047,0.143826136844893,0.105314874070076,0.121888733678949,0.0814382868869533,0.0947388103310309,0.130663998223091,0.145518416357087,0.120202452599405,0.103098715799384,0.149256342573141,0.122431784226532,0.131322089077996,0.14711038858358,0.0493558581961923,0.124727445318911,0.109203927688041,0.105163648518872,0.116532616665132,0.127135637572602,0.124169717508217,0.15671218610532,0.0523158450843372,0.0851635015813716,0.11189050330397,0.110979901013593,0.0960688557276665,0.117045279253693,0.112341797337787,0.136592057269596,0.0970421252043985,0.101887687507174,0.110953036753762,0.110865628983667,0.109814935643982,0.0430971040072754,0.149796342211023,0.133198800837095,0.0853365576522177,0.110188433904571,0.168308957795264,0.118488584849291,0.1452926246518,0.118631689890126,0.115535053905968,0.121672366347929,0.100599847308582,0.0907641822385765,0.0627650255447793,0.161693897293781,0.102198567162009,0.0839255441539103,0.217330343896606,0.100599837395913,0.111346735952306,0.136949447865495,0.135641640480839,0.109401861783389,0.106951900787319,0.114935785879601,0.0664086314246233,0.0484803324070489,0.105958512500166,0.0984994881026274,0.101447879787261,0.185315989066516,0.151167293268775,0.144644347277656,0.108983448107883,0.155024326868546,0.125131698677401,0.124052022455535,0.110343147434788,0.134867522129766,0.105357551684954,0.103068452079154,0.0976678619072457,0.171425346839928,0.15803938176386,0.133702690224982,0.031523353839505,0.108388717700455,0.117971207847294,0.10330126246172,0.142170324906683,0.0957205978251694,0.112752905123756,0.116242624423539,0.101586316134761,0.113754423316941,0.209248614329155,0.165896101866891,0.226151302416338,0.133054720870314,0.118091460672524,0.172637655667584,0.0877512617299473,0.17466309866508,0.115832085339796,0.0930421954459274,0.097416306514651,0.236094292583213,0.225897454378923,0.127657057264169,0.115527265281896,0.0313338653658062,0.113384402284397,0.123072800820977,0.11348273171942,0.11719308709443,0.12346330076583,0.102702288904829,0.171564843629441,0.102070682989014,0.0921276987392435,0.172689275989529,0.140307596851549,0.122286532962209,0.0767356130038373,0.0939137964378668,0.0831869928902498,0.211755835813216,0.0868374513423915,0.0877740753643153,0.0933849114619032,0.0742138097494739,0.133748943152278,0.103015663687012,0.186231080584317,0.234141176356771,0.0922326143202301,0.0775778387804062,0.154036651586614,0.12032249954317,0.108458321111413,0.159441999461461,0.134602838524525,0.105827682515174,0.147669984289325,0.111423197223801,0.230112346713603,0.134719667738891,0.0931164182450057,0.142267069515505,0.108212381328378,0.119302900257624,0.0941629021756402,0.0919482538572355,0.0888409894747425,0.0990453140259643,0.0457104054985517,0.109141086986233,0.105124792621358,0.105505427669526,0.13114577628228,0.0883204929279026,0.0965490178437816,0.118605433891334,0.117700595429289,0.167149269267387,0.0870246431038894,0.0863560342603137,0.0980245122299276,0.164642765147707,0.157296348556655,0.117776153324889,0.126067476347391,0.142442952092091,0.129191498425822,0.108874003130104,0.100192279560061,0.117874877321179,0.0685542580425841,0.118972608673045,0.15943464588024,0.142638473250317,0.0915549930851287,0.0977471154483225,0.116434793547816,0.0997720351186035,0.11037816214955,0.0982835705703005,0.0980801242256042,0.0954683862541849,0.104056783338112,0.162477813482447,0.0505993353680777,0.110179229872301,0.10408957748123,0.0737734644440564,0.112960611987855,0.107173228843963,0.043536787135796,0.104314864032244,0.228016538011492,0.102685579128234,0.129393815320394,0.0939216584745902,0.121129884965663,0.0962131135995426,0.104777194253898,0.0887856604015405,0.133873411944684,0.20525904367703,0.125871120057912,0.103869080736021,0.142216334730451,0.0721277197922425,0.14827088027648,0.117684431642199,0.108742698903912,0.142352611172868,0.114438322233171,0.0589530767897708,0.125858606195319,0.192869182729213,0.107405007083023,0.146623048627706,0.103475414671592,0.223321770998052,0.0972541545459827,0.209933715892885,0.190211602244125,0.0838158457625563,0.11826152262033,0.101556271707562,0.114741250783674,0.106612830822504,0.113986432226091,0.0983717734708579,0.140253667047638,0.101445204247515,0.0830394292218949,0.149187272574088,0.0882167370664583,0.102747761371858,0.13199415557426,0.0724727752314695,0.133329136696365,0.230124303842609,0.164466998160777,0.153588839157193,0.172189122496686,0.128842610091462,0.106278282926378,0.176706594535674,0.119175336267877,0.101901903385985,0.0990758185379135,0.13395520607369,0.0878927104136289,0.111437293603701,0.121164741407703,0.107667757665081,0.0924779611625733,0.126154870196191,0.125574834921506,0.100220542303305,0.175641103156051,0.0936851451624622,0.0868712448107657,0.124949888508808,0.100389659886049,0.143110025205502,0.1050023795427,0.124915393522871,0.103627842168379,0.113250327271379,0.0386203335703449,0.105758114599942,0.0870786973406758,0.0905051155753009,0.131966361476338,0.087071815819429,0.102906032184983,0.105431866547035,0.208806539724436,0.120091627674813,0.0972081038567599,0.157030249414079,0.102927021770169,0.245890367146799,0.1203849285182,0.127436843784476,0.115704437159528,0.101324311203431,0.130341826267604,0.19802760309482,0.132596801663161,0.111409327419914,0.145031113005609,0.122174783338137,0.167770332772844,0.0752554722069568,0.110866205125852,0.028911914173167,0.107448641777443,0.115216863991052,0.110594592486659,0.119480834228449,0.157400382866691,0.0867210079550541,0.072303866540019,0.182393550055925,0.105195756797961,0.153586079048704,0.085747909513013,0.102598221485426,0.108312378528112,0.110678400324294,0.118536252695695,0.133607700933962,0.137848985619991,0.131794826816274,0.128414017418216,0.102745498466335,0.182694145572844,0.108301122343364,0.179082299256105,0.12441924382373,0.127388282204298],[0.105431463110392,0.114131020693596,0.0982035211126435,0.111045857657446,0.220958471174025,0.145073294294272,0.142514424348975,0.104533121842594,0.104412557523485,0.138914805771924,0.111292744889486,0.156792206509444,0.129140130058716,0.0979479085343209,0.113211798644932,0.129567484459334,0.0278603805736452,0.131608736038501,0.110000573916235,0.106529216609111,0.129506020081039,0.134879513659041,0.260117085510714,0.0990218143721624,0.115014178265601,0.148786121342649,0.129425040844018,0.100788118868376,0.120225291097331,0.148225795637859,0.121521466287004,0.0385540245456897,0.102923652762989,0.102515860110364,0.109013447878296,0.101676927327635,0.119572759621691,0.110801289822154,0.131762879917214,0.128875183728735,0.124168266283746,0.141993963366928,0.116434532133961,0.102207576393405,0.138095661160496,0.0983985155872465,0.112295831745783,0.143002311511493,0.14674353385867,0.105487803617394,0.183586962510112,0.110736567604991,0.123374602205762,0.120055625607738,0.152999870150512,0.183023283138906,0.104016052592213,0.0978506601741561,0.0406600587550852,0.167383534468287,0.110970751184816,0.127089328288556,0.106530977467012,0.11852711082779,0.117768788526797,0.14306784989779,0.121633346872005,0.179684922327363,0.194854856586478,0.108772359993811,0.104790280085791,0.114832059108243,0.12592236836793,0.104209700927416,0.122313152879153,0.1034266325348,0.107251778218896,0.158109221744546,0.147788716371528,0.0984304842884673,0.116233869898427,0.115634964277507,0.111000998977278,0.104287386745292,0.120840238439459,0.108636453452737,0.0984772805897008,0.107072098341639,0.169151330657876,0.101101748009948,0.154203243202587,0.113706766645081,0.106427586851076,0.0322034783319441,0.113669436928571,0.127522933222563,0.110048834008028,0.105520415825917,0.120868466473743,0.112895434105623,0.130900963218377,0.130307945910191,0.103982977362748,0.129296419110814,0.113523395353744,0.119918351022244,0.0366050412246026,0.0996037823097863,0.100326792840485,0.0977787273316244,0.121143904080308,0.0979912804008036,0.107237176638326,0.106233429876753,0.112391661777999,0.106945193132917,0.0306030175930023,0.100340409514199,0.185688541958039,0.0979930339151097,0.113692620042842,0.10804638118057,0.141608057420827,0.110484861445384,0.101497148282939,0.116663348189258,0.0391796292425102,0.120920048304075,0.103036181605319,0.130506674568274,0.162521039631663,0.150818922952735,0.101270509286554,0.122372466969103,0.220780706409823,0.108597381704821,0.100159618203523,0.117129780349466,0.111063735503331,0.128126470762066,0.139896330095554,0.0988884331681705,0.127721090151904,0.105218176878311,0.103883601564764,0.103993507514103,0.107557428313706,0.122296936214467,0.107734358515283,0.135541844966648,0.134877929934298,0.157045664401238,0.12088887679792,0.106031737380555,0.114961767329169,0.12706416076159,0.111946300629378,0.139530977266709,0.0984128303321111,0.108800415847136,0.105769010716126,0.0985862150821896,0.111192525833597,0.166108337100037,0.23632902879383,0.121777877517106,0.114572503497033,0.100698614060615,0.137006681074984,0.121604900237916,0.127577878659227,0.125197579412238,0.105253737131188,0.10490013907168,0.102208268423303,0.115731175037736,0.110425119552025,0.109458816614929,0.10433786292801,0.100221770081165,0.116505279127956,0.145179725934898,0.123367495918335,0.03566226727523,0.15440856131648,0.112739814465096,0.106173126533648,0.153641694212135,0.159974520511628,0.100622727107946,0.117957181464413,0.11571031369789,0.09905239125516,0.184306750743325,0.233298310192686,0.104304988001531,0.143888805461298,0.119908657708192,0.103889125746094,0.120158447219952,0.119716808562377,0.12048874396651,0.116368647536817,0.11388278190211,0.106625099195124,0.202681890013585,0.155629872883124,0.100261866127449,0.101161255981768,0.11559389284566,0.104741028030999,0.125190691615919,0.103720803073767,0.156914770416445,0.151313677579203,0.12364725185402,0.131230129286766,0.102892697331527,0.103949264752346,0.11268454093008,0.106132893547766,0.1200496722795,0.109642867528302,0.0267762658466268,0.150934721730737,0.100401129936158,0.101450539001734,0.108409310680577,0.207503988390448,0.103425091202536,0.0677629702726031,0.180705387198584,0.187555937603525,0.113729977004193,0.178931172783724,0.1488183068898,0.110184985600531,0.140534718734182,0.195785976701622,0.177416126519894,0.106954807692139,0.161371934402059,0.123246302214042,0.155813553399734,0.105072731023151,0.133299498142331,0.0343499198939243,0.137992762123374,0.120378720810967,0.110660232825019,0.11608107989486,0.123564391291902,0.150547653926275,0.0980766905343292,0.17420888895437,0.0995693122443411,0.100246986592748,0.0993196834058419,0.100018783163018,0.328988922514556,0.0997619655384828,0.137243892240682,0.0358163636507879,0.118967833289531,0.105181799537447,0.110211397840518,0.11564875747558,0.0399809465644202,0.0311529259443738,0.116263126798312,0.0999776021511441,0.120099486550186,0.120262690395354,0.107054995746549,0.126417369870269,0.12687409615526,0.102017432361006,0.0984978499830587,0.0363928030578272,0.105302444649089,0.10445666931702,0.107608908408081,0.116849883771638,0.145638469158238,0.131351561027478,0.113408700376553,0.125147706160678,0.109552734238221,0.175747292649722,0.179811906952498,0.0983759040465549,0.18566442647805,0.162018217086701,0.122212828508079,0.105112020194281,0.121530247766475,0.114451898067832,0.102251893807493,0.1091248812421,0.164035685808048,0.135465454960402,0.135566968500852,0.110951442513156,0.100697312564814,0.0441937598198828,0.116108089125761,0.154218046206158,0.121781113616906,0.110711498540029,0.108608989681224,0.0337275911295011,0.102191127997574,0.104385065782844,0.114725134333625,0.0331311096337304,0.144218637338022,0.10435208833392,0.12291838704356,0.127691011357406,0.154367583219703,0.1332967338139,0.102643490989578,0.105998367739408,0.121632512402104,0.176295247365901,0.126818890563381,0.10269961640345,0.130487291433816,0.177209095107356,0.107982830760032,0.102944014761896,0.117429105327765,0.105753904968823,0.130087921089817,0.157419076448206,0.193828918939034,0.121204042873475,0.124629662715926,0.115277287285557,0.173498214221129,0.139317556761908,0.1188183652284,0.109341265808358,0.12222110392123,0.100994292838547,0.113011688556158,0.143002904873591,0.135947680021543,0.105840982456246,0.0359892647689916,0.164688984434912,0.142212519965052,0.105829668284826,0.029858791338746,0.101182899421011,0.114816082699455,0.111027011454976,0.100256408815021,0.117438804993973,0.047107958256372,0.107965111937573,0.135610530106849,0.107471189306206,0.105879557968269,0.107636944830908,0.10484387031138,0.126013401495706,0.124629957577658,0.110551291250399,0.162520378416594,0.100566923389845,0.147335306202504,0.148848755706546,0.222295635249049,0.106948777770875,0.107430553830726,0.127073951656335,0.0405514180729111,0.13911741419345,0.156313516074121,0.102620244130771,0.348900294758046,0.204615534027357,0.152907993781744,0.114356408487043,0.113603864962019,0.194018623708414,0.107617872869896,0.101721319556357,0.114043399062949,0.146135844926791,0.104185230012704,0.143539944577048,0.100445917146166,0.1317571449257,0.100458451147567,0.152175796718489,0.133401230472012,0.104436095304278,0.145733480817297,0.115613686881575,0.12277895209867,0.113513599033046,0.121140273902387,0.0984657761927145,0.100157176267391,0.115968049479864,0.0999391088539942,0.232523775005788,0.150773962325249,0.113387354396328,0.120926961356453,0.0993444120938934,0.173917473964585,0.102709647456595,0.0984900334049957,0.0979136714004466,0.131589027957562,0.125779982621933,0.105029866189308,0.11867100853316,0.121189993231952,0.173091339425919,0.14907796338196,0.0991211028871908,0.149650636012472,0.109172488252586,0.10887879088757,0.116368795459133,0.105659951817647,0.252196537253347,0.111243632333741,0.122214635964003,0.138672619235063,0.113676379658565,0.161557755040499,0.114178826803163,0.118064558803506,0.141745642997635,0.100993029941941,0.106942289764015,0.151514339014502,0.115165429939669,0.121310349885058,0.142989005776121,0.117027100464516,0.158584072428853,0.106298935379527,0.189761119593275,0.212169546167853,0.0356099592801085,0.128873297089328,0.102580073381944,0.108038326126465,0.140263403581087,0.18084105961798,0.100442503722246,0.148591126270758,0.10520744191251,0.138618958723014,0.125285455048877,0.135564363947734,0.103951378537079,0.105706017045471,0.098628303251047,0.151814318274115,0.119096744250977,0.14031808923765,0.177688043849011,0.150077034062401,0.11442025780833,0.157484476986741,0.150056993419904,0.173206910534055,0.11727604854301,0.109743565305917,0.111481291285332,0.0996487958849325,0.10232315587368,0.0312364387666638,0.121353197207488,0.104539762868148,0.0376776460817001,0.0977551780731519,0.116134964721729,0.0345326328779617,0.0985198756559046,0.126717699970464,0.110940464625006,0.199565208941247,0.100674078467151,0.102844844365221,0.119058269886902,0.120407276435934,0.135172219420264,0.102220744554004,0.100642077479953,0.211046036950856,0.101486940799491,0.0986787890234061,0.127711910261386,0.146675005468935,0.118595619988001,0.126897368957518,0.0339145158740435,0.110744901557277,0.140103076088423,0.0330097634239316,0.143898598623915,0.159043569371272,0.117225889354785,0.201488974815128,0.128575633951425,0.108314062154437,0.158337966107552,0.104471234124077,0.138909497890205,0.12653901413865,0.112476647921094,0.107510460605333,0.100481755291095,0.143029839075585,0.131511681432906,0.132695466796183,0.102766013918896,0.112357613472215,0.133683452162948,0.102138731380667,0.114900508180137,0.147253985524015,0.103132869484575,0.122155823433481,0.114284845868427,0.146156220126535,0.102784922084847,0.106633624769649,0.13155643767102,0.115106620814071,0.196693786021142,0.10319220471816,0.116178592768669,0.100613581750088,0.142261782812527,0.0995211473682703,0.113519726810541,0.121626354354947,0.108849213261969,0.151827892966933,0.0993940861620931,0.103620076010244,0.139408957728514,0.137303053242985,0.275634925967324,0.12145694224119,0.110102402784389,0.104828827604198,0.106461204833076,0.189444381786094,0.0419346898441106,0.120740058075431,0.128500126693318,0.139696584330835,0.114394531796332,0.101139371044593,0.103294234032783,0.10016115957098,0.113032030183446,0.107395787595627,0.121291548031252,0.101641315907406,0.148806141669066,0.114472338180437,0.190185288018388,0.137827926537097,0.107028163154415,0.144767437739726,0.115635595888244,0.111512742053878,0.15567965143505,0.14040085407054,0.108042260122413,0.119773076782715,0.144600165552877,0.163023716588568,0.0996955130482882,0.123420254478704,0.135221595474898,0.129356109362257,0.128582488847318,0.105339133891625,0.149744508938293,0.0351479002608733,0.100780157120115,0.134194378470142,0.110859475866308,0.108907129314791,0.113137469970101,0.152176891807874,0.161845070494829,0.109777577721042,0.110836262001099,0.0463194030230918,0.110058797775808,0.107567445695481,0.100614894572289,0.107758955199349,0.101949299535388,0.112475226490753,0.101275314890544,0.232710274794241,0.0420367428355606,0.169850859331703,0.10963908126345,0.105293720977544,0.124326555849874,0.145149377024152,0.120524528169917,0.133833237920102,0.109676704006197,0.146090197146227,0.108019159719159,0.100369657018527,0.131962040765128,0.148713444431614,0.101752237133689,0.163542651400531,0.114463328095931,0.136457595007665,0.138473557824832,0.108330275769513,0.106392022143731,0.102660471510307,0.106296314481634,0.031904455755447,0.104699783716124,0.103726599481306,0.110227711030334,0.159372170649724,0.129371161340277,0.105653426452471,0.131086962684516,0.112188759595057,0.104248451723284,0.183746946716632,0.11659496615813,0.157031718447124,0.116914137526411,0.123424396998902,0.103706828526837,0.173798824643091,0.125503952302289,0.121319789124246,0.174578319972821,0.0978261483522211,0.154905146158979,0.16112863523925,0.114504158509846,0.272942848029375,0.107127600722426,0.105026934346078,0.110524599092927,0.11626208180977,0.305678972007119,0.106741098866091,0.233643864675217,0.312883606619226,0.167281012671782,0.124391353194413,0.123902660579612,0.121190968535627,0.10727476448281,0.140213405120799,0.0392186149877332,0.0998141688351115,0.149851948365182,0.115607896898106,0.118857734203147,0.153479009864062,0.171006059065617,0.128209628927642,0.127742079930256,0.176108264408213,0.104962341402509,0.0995648128805096,0.118663000624607,0.134919960495646,0.112089637183129,0.126556111282624,0.104088166414168,0.113013800629806,0.148225125207188,0.128926042614092,0.137916872549646,0.1195897881451,0.0980096107482235,0.1658219256506,0.127982140367185,0.200214749116593,0.175919231968347,0.129192599973946,0.137296735471564,0.1540835048963,0.0330124836824348,0.0331650596689137,0.264340317583749,0.158518920839813,0.108402698219323,0.148528671085186,0.10055172002467,0.101885128393079,0.115784126127905,0.183169921409511,0.121660760747414,0.108784017826635,0.119120871994184,0.177324210039777,0.109937661693842,0.0988399362616435,0.10176967249991,0.133117590051147,0.119619291135619,0.112534759226266,0.113518446665915,0.147895735113282,0.162767146115377,0.109029016173125,0.122891775675038,0.0986634918486848,0.106781615249143,0.135891886146414,0.135977409602019,0.103487567657762,0.123246050751597,0.121219864366964,0.116698596093476,0.126249696096128,0.126794462279423,0.0338692448704969,0.128386302067881,0.118710571747477,0.105349050748909,0.119455214336977,0.127517396870914,0.127158842729135,0.157781163012905,0.100281319557029,0.100244968690021,0.121395295536013,0.106669997876908,0.103923198121054,0.127648818627886,0.108681326678095,0.14577118375229,0.101291752005206,0.111472094238881,0.101148906962195,0.116638139801988,0.10322273875819,0.0381633916451877,0.151214914782264,0.134335679698181,0.0998888507173916,0.124948048976764,0.162606562715218,0.115158240982713,0.147262573342376,0.115325320559257,0.128404645992869,0.126617265730375,0.110278226750612,0.0993538322410969,0.10120163875533,0.160711116168146,0.112587089673104,0.112190097127271,0.217995486520104,0.106402655707003,0.135669471993857,0.135048488138688,0.152088995638458,0.114152680675519,0.101799672839385,0.102102393766592,0.0495697227504561,0.0345940409914058,0.10487244357338,0.101791578040021,0.103594753626629,0.186730848070527,0.154796220988138,0.16224959646862,0.118985829663605,0.159308983993123,0.132070814297995,0.125212619500506,0.111727067784718,0.153397352792773,0.108816679530998,0.111391751586697,0.100536866045208,0.171522316313952,0.150805037490257,0.12897948325789,0.0317676035244551,0.110588619213532,0.133203379776439,0.113340827021815,0.136267990274812,0.103441534779634,0.107460083818898,0.107947240931402,0.0993221736322832,0.106718649440887,0.233574249975313,0.148041190827891,0.239525195396189,0.132926481792869,0.12349578178902,0.172910075013506,0.104194830678237,0.168299875673393,0.109297270339369,0.0987432511151566,0.105096157175747,0.227320633245044,0.225078890296832,0.118419727058703,0.110689230102703,0.0312914850193492,0.117542357872608,0.124311348895669,0.11516446527037,0.0997807348756052,0.105461433562935,0.122804467187509,0.186985313315921,0.1032135787402,0.103216662726838,0.175726778701451,0.13935210140724,0.113249261365167,0.114922970965275,0.120601277482247,0.113472526335189,0.219856245633937,0.10789534380725,0.10987024857078,0.117492566565168,0.113219596641819,0.158731518893595,0.0989941519677192,0.202730772532113,0.252461951708525,0.10040936785268,0.0990886942529348,0.158857160382571,0.122199537327814,0.113316646934867,0.153948893832565,0.133261392711668,0.11315556760644,0.153379249526871,0.14763717345256,0.22508334697177,0.145174672535286,0.114103191786391,0.145724595711479,0.114210997538778,0.151397426164441,0.101099511390067,0.105325679566089,0.103846895009311,0.102518333507377,0.112762942356723,0.102748369095981,0.115880112396866,0.107239533311986,0.122997649712204,0.0998441445482716,0.108298419487486,0.126582007921322,0.113271811953516,0.135541884681568,0.126158254255624,0.125099594548612,0.101890685457641,0.161614551207401,0.160133266761069,0.120102672408967,0.121313383142627,0.147457224731432,0.115533837841123,0.134168021021247,0.11765214862261,0.107982135006909,0.132476646709591,0.114500805742716,0.16489490951441,0.13272506029787,0.112805835116489,0.106392659953113,0.107881933865244,0.103314386404973,0.100870964049984,0.0998428945306909,0.100370310960302,0.100564973247338,0.102354587676922,0.174411887382166,0.100784488000811,0.105119072644453,0.128999262381935,0.108156282698066,0.110271040273023,0.112275749374612,0.0452555210867296,0.103295597794763,0.208715442330927,0.114322659602705,0.144347485507034,0.100526580952029,0.129160098062633,0.101513757210292,0.10127846960865,0.106931897626847,0.134476394300814,0.213021416652033,0.137712378169598,0.11562607634918,0.146607829346826,0.0992539480935938,0.142865342122355,0.119530882472467,0.112695645847277,0.143361381798497,0.111012786380763,0.0979958151855498,0.12120702090943,0.229597864220119,0.103384509171074,0.162031984255736,0.100826708868974,0.225139569146328,0.101806789620929,0.242355087671543,0.189810294251434,0.117173862818488,0.13534565738474,0.102002069574626,0.120018560918289,0.111605715968188,0.114971218859209,0.107571746679901,0.14594922599279,0.0982620404961932,0.119155845719145,0.144731402124348,0.0982548411410892,0.122080670048312,0.13380123889032,0.100429203646429,0.14139655910784,0.231321592865623,0.104571536191306,0.162377132231509,0.179172367275952,0.133647419369915,0.104987839353792,0.176232338846274,0.131403259883443,0.112543161998791,0.117554995676893,0.135570472268517,0.103434163427862,0.110826039847783,0.133790434576186,0.118811866426799,0.102120230608231,0.114508530067084,0.13845838536723,0.103579068214375,0.164028832424231,0.10484174272368,0.100990628811889,0.127923352467064,0.10871262618116,0.159577865081812,0.103965045011336,0.1187196093674,0.10060487246987,0.116885051419056,0.0352768557143436,0.111317860233823,0.0978772602237952,0.10231182520635,0.152493137557904,0.123441491580011,0.0987836937611603,0.0981617689200746,0.219549310280767,0.12454398485639,0.10229883955295,0.180850789018656,0.104154324774019,0.158090201022498,0.120623173338271,0.135189149549944,0.110369468691762,0.118279369348044,0.13562099472938,0.202880386403134,0.119168415727558,0.106698972784512,0.163713596035458,0.131370044884411,0.17619117642432,0.0996066839622572,0.105835803287326,0.103297273182739,0.107667011995551,0.106913173413824,0.110096812015204,0.116028664325851,0.163920396572734,0.098826456671209,0.128285150279328,0.175752935738674,0.148662791442743,0.16081298463415,0.100643371135464,0.107773507821945,0.105906006591975,0.11539624698319,0.120270156782399,0.133231070785071,0.140325435068074,0.124752875070374,0.126525282333911,0.102526447162165,0.178383083599382,0.106542943315083,0.191763180275047,0.144180975653571,0.123427097691919],["CFH","TMEM176A","DBNDD1","TFPI","SLC7A2","HSPB6","PDK4","TMEM98","TMEM132A","CX3CL1","TNFRSF12A","CD79B","MGST1","NFIX","IL32","MAPK8IP2","ZNF207","FMO1","SYT7","ANLN","DCN","SEMA3B","LTF","ELOVL5","GPRC5A","IL20RA","CNTN1","SNAI2","RUNX3","PLEKHB1","FHL1","RNF10","RRAGD","HSD17B6","HMGB3","VCAN","CDH1","RAB27B","TNC","BARX2","GPM6B","COL9A2","KITLG","EPN3","ELN","DKK3","NFE2L3","PTGER3","LAMA3","OPN3","FOXC1","CYFIP2","IL17RB","GYG2","LAMC2","COL11A1","BCAT1","SPAG4","SFSWAP","CDH3","DGAT2","CA11","LIMCH1","ATP2C2","SLC12A2","EYA2","FAR2","BCAS1","CHI3L2","NTN1","MYLK","GSTO2","ME1","ISOC1","PFKP","TRO","PRR11","REEP1","TGFBR3","PFN2","ST6GALNAC2","TRIP13","CYBRD1","HMMR","SIDT1","ST6GAL1","TEAD2","ATP2A3","CA12","MGLL","NTN4","SEMA3C","MOCOS","SART3","ACACB","GPC4","RAP1GAP","UBE2T","CST7","FAP","EDN1","ITM2A","CDC14A","CEACAM1","LIPE","MOXD1","CNOT4","NDC80","PTPRC","CADPS2","C1QTNF3","EPB41L3","SERTAD4","FAT1","GSTP1","EPDR1","RBM22","NOX4","PTHLH","TFAP2C","AURKA","PIR","SULT2B1","TPX2","SMOX","CPXM1","DYNLL1","OAS1","BIRC5","RGS1","LYZ","CHRD","KIF4A","SEL1L3","ESR1","TMEM101","CCDC80","SLC7A8","SLC22A17","SEMA6A","PHGDH","UNC13D","CRAT","SORBS1","DSP","SCD","PRTFDC1","H2AFY2","PALM","MMP11","SUSD2","SEC14L2","GGT1","KDELR3","RAB36","TIMP3","BIK","RASD2","SYNGR1","PYGL","CDKN3","SIX4","EFS","MMP9","EEF1A2","ISM1","PLCB4","PXMP4","WFDC2","PPP1R16B","FAM83D","SRPX","SMARCA1","PIM2","SRPX2","SYTL4","TNFSF13B","ACP5","CORO1A","CRISPLD2","HAGHL","SLC7A5","METRN","UBE2I","TOX3","XYLT1","PYCARD","SYT17","TMC5","HOMER2","SGK3","PDGFRL","BRF2","CA2","SFRP1","LAPTM4B","PLAT","GDAP1","NDRG1","NCALD","TUSC3","KCNN4","MAP4K1","CD37","RASAL3","CD79A","NKG7","ETFB","MEIS3","CNFN","CLEC11A","CAPS","PLEKHA4","COMP","HPN","CDK6","PTN","CAV2","WNT2","LFNG","CPVL","COBL","SERPINE1","NRF1","SFRP4","MEST","TSPAN13","RARRES2","AGR2","GLI3","EIF4H","SLC1A1","OGN","ECM2","PTGDS","GATA3","CXCL12","SPOCK2","DNAJC12","KRT23","RNF43","RASD1","CCL2","SLC16A6","PMP22","ALDOC","UBE2D3","TBC1D9","INPP4B","CPE","SOD3","TRIM2","GLRB","UGDH","CRYAB","CTSC","MS4A6A","CCND1","IL10RA","SCGB2A2","SLC35F2","CD69","RSRC2","FOXM1","RAD51AP1","OAS2","MGP","CNOT2","CPSF6","CDCA3","CHPT1","ENO2","LDHB","COL12A1","GCNT2","TPD52L1","NCOA7","UST","SRSF3","CAP2","PERP","MDFI","SMOC2","ENPP5","GHR","KIF20A","MRPS30","LOX","GZMK","THBS4","PRLR","STC2","RBP1","WNT5A","C3orf14","C3orf52","PODXL2","VIPR1","CYTIP","REEP6","EFEMP1","IGFBP2","IGFBP5","EFHD1","SF3B1","FHL2","MLPH","QPCT","PLEK","RND3","C2orf42","EPHA4","GPX7","KIAA1324","SFPQ","FBXO2","C1orf21","WLS","RGS2","CD2","CD48","PADI2","MFAP2","RGS4","IGSF21","GBP1","CDK18","CDC20","TSPAN1","F3","MAN1C1","NEK2","CENPF","MYB","SPP1","RARRES1","CSF3R","IFIT3","IFIT2","PPP1R3C","HOXB3","DUSP1","MSX2","EGR1","PLS1","GLT8D2","SOCS2","DUSP4","EPHX2","TARDBP","CRISPLD1","RDH10","MAPK8IP1","KHDRBS1","TNFSF10","CPXM2","CXCR4","SASH3","FMOD","HNRNPA2B1","ARL4A","RAMP3","TWIST1","PLAU","SRGN","CHST3","EGR2","ATP7B","ARHGAP9","COL10A1","TNFAIP6","G0S2","RAB38","SLPI","PREX1","BCAS4","KCNK15","STAMBP","F13A1","SPDEF","RAB17","SCGB1D2","MT1G","BMP4","SOX9","C3","FOSB","SDCBP2","ID1","HCST","IFI6","DLGAP5","HSPA2","EVI2A","AIF1L","ARMCX1","OMD","FGD3","PKMYT1","FBXL16","TUBA4A","TNFRSF19","FGL2","SGCE","CASD1","ZFP36","RASL11B","RAC2","KRT17","LRRC17","CGNL1","LRRC6","ADCY4","FOXA1","RNASE1","CDKN1C","SHC2","GAMT","CNN1","APOC1","GADD45G","BST2","GDF15","OLFM1","COL5A1","ASS1","GINS2","ANO1","TOP2A","STARD3","PPP1R1B","KHDRBS3","RAI2","MATN3","PPARG","RAMP1","ITGB4","XAF1","MATN2","ALOX5AP","WASF3","DCLK1","EPSTI1","POSTN","BEX2","RARRES3","MYH11","ACTR3B","VAV3","VTCN1","PPHLN1","RSAD2","CMPK2","IL6ST","FST","GRP","DOCK2","RERG","CDCA8","CYP2J2","DSC2","DTNA","FHOD3","PDGFRA","SLC37A2","FAM189A2","TBX3","OASL","CD36","EHF","TROAP","KRT7","PKIB","KCNK1","FAM129A","ITM2C","SERPINE2","CYP27A1","SPRY2","SRSF1","RTP4","IL1RN","WDR33","BIN1","KLF4","TOR1A","ANGPTL2","ENPP2","MYC","CCL21","GMPR","IER3","SYTL2","RAB30","GGH","SULF1","DDX60","MMP7","TMPRSS13","NUSAP1","ITGA11","IFI44L","IFI44","CYP1B1","RAB1A","CEP55","AOX1","SMARCAL1","SLC40A1","CILP","PDE5A","CXCL9","ANXA3","LEF1","ERP27","PRICKLE1","CD27","RBP5","PHLDA1","LUM","CSAD","TTC6","IGF1R","HAPLN3","CDH11","IRF8","SKAP1","ARSG","IMPA2","SLC39A6","SLC16A3","PMAIP1","ERBB2","GRB7","MISP3","CBLC","PRKACB","KIF2C","CYP4B1","LMO4","CELSR2","ATP1B1","DPT","NUF2","RGS5","CRABP2","RORC","ECM1","CGN","CTSK","ANXA9","SUSD4","S100A8","ETNK2","PTPN7","SFXN5","THNSL2","AFF3","AMMECR1L","SPAG16","HES6","NFKBIZ","COL8A1","BOC","EAF2","SLIT2","SCD5","CCNA2","SFRP2","SRD5A1","GZMA","IQGAP2","CXCL14","PLA2G7","TNFRSF21","EGFR","CDCA5","TMEM47","GPC3","SYBU","VLDLR","NFIB","CDKN2A","NTRK2","UGCG","ASTN2","PTGES","FAM171A1","ST8SIA6","ADD3","ADM","CELF1","SESN3","GLB1L2","FADS1","MPZL2","TM7SF2","MMP3","LYPD6B","ADRA2A","ITPR1","UBE3B","NDUFC2","ME3","CCDC3","QDPR","ACSL1","SERP2","PARP8","GFRA1","TIAL1","HSPB8","SETBP1","TCF7L1","SPOCK1","GJA1","ANKRD22","NR4A2","CMTM7","CD8A","PID1","CABYR","ABI3BP","UCHL1","TNIK","GBP5","PDLIM3","SORBS2","CXADR","BTG3","ADAMTS1","APCDD1","PPM1J","RBM45","VSIG4","FZD7","RMND1","PSD3","KCNMA1","BATF","TSPAN7","RAB11FIP1","FBXO32","SUSD3","IL34","KIT","CCNB2","MX1","KRTCAP3","CLSTN2","COLEC12","HIST1H2BD","HIST1H4H","GDPD5","LAD1","STC1","RCAN1","CLIC6","CBR3","BTG2","C1R","TPPP3","PIP","CCDC117","TFF3","TFF1","TMPRSS3","RSPH1","PDE9A","C9orf116","TLCD1","AZGP1","ZNF394","RECQL4","DMKN","COX7A1","PGAP3","ZG16B","PDZK1IP1","AK4","RBP7","AKR7A3","PDPN","UBXN10","CAMK2N1","ALPL","HENMT1","GBP4","AGL","VCAM1","VANGL2","KLHDC9","ATF3","C1orf115","PKDCC","CAPN13","FRZB","ACTG2","CCDC74A","SPATA18","NOSTRIN","BNIPL","ERCC3","IWS1","S100A9","CLDN1","FBLN2","SERPINI1","PTPN13","PPM1K","PRRT3","CPA3","TM4SF18","CLEC3B","NMNAT3","DNALI1","RPL39L","SPRY1","MST1R","F2RL2","CMBL","ESM1","RHOBTB3","ERAP2","SHROOM1","PTTG1","STEAP1","CDCA7L","KIAA1324L","FABP5","FNDC1","DLC1","OSR2","CTHRC1","GEM","ALDH1A1","RASEF","HNRNPK","SVEP1","FBP1","CYBB","CLDN3","MELK","FOLR2","GJB2","TMEM63C","COMTD1","ZNF503","NDRG2","SALL2","IFI27","GPT2","CKB","C2","PLEKHF1","CYB5A","CYB5R2","CENPN","ZNF143","MFAP4","CENPV","ATF7IP2","C16orf45","ANPEP","NAV2","C15orf48","SCG5","GREM1","MS4A7","PRRX2","PRR15L","IGF2","CD3D","CDC42EP5","PPP1R14A","C19orf33","KIFC2","GGT6","KRT80","ANGPTL4","IGFBP6","TK1","ABCA3","DDB1","SF1","SCARA3","DDIT4","KCTD6","ACOX2","CX3CR1","DEGS2","DYNLRB2","IL7R","NPNT","TSPAN5","IRS1","AR","PARM1","FAM110B","AFAP1L2","CXCL10","CD52","GPR183","PCBP1","BUB1","MT1E","PCDH7","REPS2","TM4SF1","ALCAM","UBE2E3","CMTM8","CDK1","FABP4","FOS","CST1","LRRN2","GPRC5C","MZB1","NPAS2","IRX2","KCNS3","PDGFD","FPR1","KRT15","NAT1","HOPX","GPR34","HDAC3","GATM","BCL2","COL8A2","PRNP","ORMDL3","LRRC15","CALB2","ISG20","MBOAT1","TPSAB1","ARNT2","TCEAL1","CTSW","RASGRP1","RND1","CCL19","NBEA","HSPA6","ABLIM3","SYT12","SNCG","C1QA","SAA1","AGR3","LRFN4","SPTBN2","HOXB2","PODN","GOLT1A","SEZ6L2","UBE2C","CSRP2","CHST1","PHYHD1","SCUBE2","LPL","P2RY2","NR2F1","SFN","LRRN1","SPHK1","ZNF135","HSD11B2","PLA2G16","PRR15","TYMS","FZD8","ACOT4","CD163","RPP25","ZNF552","AURKB","GIMAP7","HLA-DQB1","GATA2","EGR3","FJX1","FCER1A","CITED4","SSC5D","GAS1","TSPYL5","HIST1H2AC","HIST1H2BC","TMEM64","HOXC9","HOXC10","NQO1","HIST3H2A","TMEM45A","PHLDA2","FAM89A","BACE2","B4GALNT4","C1S","EPHB3","NDN","NTM","TSKU","C1orf116","LCK","CADM1","ABAT","GPC6","TMEM119","PRR36","ASB7","MX2","SERHL2","ZNF703","TMPRSS2","ACOT1","ALDH1A3","SLIT3","PROS1","BEX5","PDE4B","NELL2","ZBTB7C","SLC24A3","INPP5J","SOCS1","MUC1","AHNAK2","NDUFA4L2","IFIT1","EVI2B","RTN4RL1","IRS2","KRT5","WWOX","EDARADD","THBS2","CYP4X1","LILRB4","KRT14","MAPT","EMID1","ESPN","ENTPD5","MT1X","GCNT1","FPR3","C11orf96","ISG15","C17orf97","THSD4","HIST1H1C","EIF4EBP1","COL14A1","COL4A5","HES4","SELL","SERPINA5","RPL14","C9orf152","APOD","PLAC9","PCDH18","MAOA","S100A14","GSPT2","SYCP2","AKR1C3","ACADSB","SLC6A9","MME","LAMA2","TLR7","HLA-DQA1","LAMB3","ACSL5","SERPINA1","STMN3","ENPP1","MFAP5","NMB","CFD","KIF13B","HIST1H2BK","PLCG2","MB","DDX42","HOXC4","MT1F","ZNF521","CES1","CD3E","PRC1","TGM2","CAPN8","CENPW","HLA-DOA","SLC44A4","HLA-F","HLA-A","GPX3","DIO2","DENND1B","CRIP1","GSTM2","UBD","ANG","STARD10","C4B","C14orf132","LTB","TWIST2","C12orf75","HLA-DQA2","KIFC1","TXNDC5","APOBEC3G","PSMB9","CFB","SCARF2","C4A","HBB","TMEM150C","TMEM158","TRNP1","CLDN23","TRIL","HOXB7","EPPK1","SPON1","CCL5","CD24","CCL4","PRAG1","FCGBP","UHRF1","DACH1","HIST1H2AE","CCL3"],["no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","yes","no","no","no","no","no","no","no","no","no","no","no","no","no","no","yes","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","yes","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","yes","no","no","no","no","no","no","no","no","no","no","no","no","yes","no","no","no","no","no","no","no","no","no","yes","no","no","no","no","no","no","no","no","no","yes","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","yes","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","yes","no","no","no","no","no","no","yes","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","yes","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","yes","no","no","no","no","yes","yes","no","no","no","no","no","no","no","no","no","yes","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","yes","no","no","no","no","no","yes","no","no","no","yes","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","yes","no","no","no","yes","no","no","no","no","no","yes","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","yes","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","yes","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","yes","no","no","yes","no","no","yes","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","yes","no","no","yes","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","yes","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","yes","no","no","no","no","no","no","no","no","no","yes","no","no","no","no","no","no","no","no","yes","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","yes","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","yes","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","yes","yes","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","yes","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","yes","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","yes","yes","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","yes","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","yes","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","yes","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","yes","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no","no"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>tcga<\/th>\n      <th>metabric<\/th>\n      <th>average<\/th>\n      <th>symbol<\/th>\n      <th>is_housekeeping<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"targets":0,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 2, 3, \",\", \".\", null);\n  }"},{"targets":1,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 2, 3, \",\", \".\", null);\n  }"},{"targets":2,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 2, 3, \",\", \".\", null);\n  }"},{"className":"dt-center","targets":[0,1,2,3,4]}],"scrollX":true,"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true}},"evals":["options.columnDefs.0.render","options.columnDefs.1.render","options.columnDefs.2.render"],"jsHooks":[]}</script>
<p></p><figcaption class="figure-caption">Figure 2.3: (<a href="#fig-sd-stable-genes">Table&nbsp;<span>2.3</span></a>) Standard deviation of each gene in each condition and pooled standard deviation.</figcaption><p></p>
</figure>
</div>
</div>
<p>Overall we can see that housekeeping genes have very low standard deviation across cohorts, which is good.</p>
<div class="cell">

</div>
<div class="cell">

</div>
<p><a href="#fig-avg-ranking-hkg">Figure&nbsp;<span>2.4</span></a> shows a histogram with the distribution of the average ranking of the housekeeping genes for each sample in each cohort.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-avg-ranking-hkg" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="pca_merging_files/figure-html/fig-avg-ranking-hkg-1.png" class="img-fluid figure-img" width="960"></p>
<p></p><figcaption class="figure-caption">Figure 2.4: Distribution of the average ranking of the housekeeping genes defined previously for each cohort separately.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>The average ranking are somewhat similar. For the two experiments where there are RNA-seq samples, the distribution is the same. For METABRIC, a microarray experiment, the distribution is shifted a bit below.</p>
<p><a href="#fig-gene-bc-imp">Figure&nbsp;<span>2.5</span></a> shows histograms of gene expression from some important genes in breast cancer.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-gene-bc-imp" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="pca_merging_files/figure-html/fig-gene-bc-imp-1.png" class="img-fluid figure-img" width="1920"></p>
<p></p><figcaption class="figure-caption">Figure 2.5: Distribution of the gene expression levels of ESR1 and TFF1 in each one of the cohorts</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>ESR1 and TFF1 expressions are divided across ER status, as expected. The proliferation marker CCND1 is also highly expressed in all conditions.</p>
</section>
<section id="checking-robustness-with-respect-of-number-of-houesekeeping-genes-available" class="level3" data-number="2.1.3">
<h3 data-number="2.1.3" class="anchored" data-anchor-id="checking-robustness-with-respect-of-number-of-houesekeeping-genes-available"><span class="header-section-number">2.1.3</span> Checking robustness with respect of number of houesekeeping genes available</h3>
<p>When calculating the normalized data, a total of 44 housekeeping genes were used. Sometimes not all of these genes are available, therefore we will calculate the normalization using random subsets of 44 housekeeping genes with different random sizes. The sets will have a minimum of 20 genes and maximum of 44. The sampling of the set size is obtained by using a uniform distribution. We focus just on TCGA for simplicity. The correlation used in this case was the Pearson correlation.</p>
<div class="cell">

</div>
<div class="cell">

</div>
<div class="cell">
<div class="cell-output-display">
<div id="fig-correlation" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="pca_merging_files/figure-html/fig-correlation-1.png" class="img-fluid figure-img" width="2016"></p>
<p></p><figcaption class="figure-caption">Figure 2.6: Correlation distribution for different subsets of housekeeping genes (HKGs). Each different subset has a different number of HKGs.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p><a href="#fig-correlation">Figure&nbsp;<span>2.6</span></a> shows the different distributions given the subsets of housekeeping genes. All the correlation coefficients are close to 1. This means that if a subset of housekeeping genes are missing it is fine, as the others will compensate.</p>
</section>
</section>
<section id="integrating-tcga-and-metabric-using-pca" class="level2" data-number="2.2">
<h2 data-number="2.2" class="anchored" data-anchor-id="integrating-tcga-and-metabric-using-pca"><span class="header-section-number">2.2</span> Integrating TCGA and METABRIC using PCA</h2>
<p>The next step is creating the framework where we can integrate any patient sample and compare each other, at least spatially. The approach is similar to some sort of unsupervised learning, where no labels or groups are specified, which is something common in algorithms for batch removal. By using PCA in some subset of TCGA and METABRIC, we should learn the idiosyncrasies of the RNA-seq and microarray technologies as well as batch effects general to datasets. Since these two cohorts are big ones, we assume that if there are common batch effects, they are represented on them.</p>
<p>The practical steps are: first subselect 1000 samples in total from METABRIC <strong>and</strong> TCGA. Train PCA using these 1000 samples. Validate the results by projecting the other samples. In the next section we show how robust the approach is when selecting the samples. In the next chapters we validate the projection in unseen cohorts.</p>
<p>Integrating as laid out here only uses molecular data, therefore patients might be close to each other but they might have very different clinical features. This should always be kept in mind.</p>
<div class="cell">

</div>
<p>The table below shows the molecular characteristics of the patients selected for the PCA training.</p>
<div class="cell">
<div class="cell-output-display">

<table class=" lightable-classic" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
 <thead>
  <tr>
   <th style="text-align:left;"> cohort </th>
   <th style="text-align:right;"> basal </th>
   <th style="text-align:right;"> claudin-low </th>
   <th style="text-align:right;"> her2 </th>
   <th style="text-align:right;"> luma </th>
   <th style="text-align:right;"> lumb </th>
   <th style="text-align:right;"> nc </th>
   <th style="text-align:right;"> normal </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:left;"> metabric </td>
   <td style="text-align:right;"> 83 </td>
   <td style="text-align:right;"> 78 </td>
   <td style="text-align:right;"> 71 </td>
   <td style="text-align:right;"> 215 </td>
   <td style="text-align:right;"> 158 </td>
   <td style="text-align:right;"> 2 </td>
   <td style="text-align:right;"> 41 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> tcga </td>
   <td style="text-align:right;"> 69 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 29 </td>
   <td style="text-align:right;"> 177 </td>
   <td style="text-align:right;"> 61 </td>
   <td style="text-align:right;"> 0 </td>
   <td style="text-align:right;"> 16 </td>
  </tr>
</tbody>
</table>

</div>
</div>
<div class="cell">

</div>
<div class="cell">

</div>
<p><a href="#fig-pca-embeddings">Figure&nbsp;<span>2.7</span></a> shows the embedding of the two datasets together of the 1000 samples used for training. We select several PCs of interest to show how the embedding works.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pca-embeddings" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="pca_merging_files/figure-html/fig-pca-embeddings-1.png" class="img-fluid figure-img" width="1920"></p>
<p></p><figcaption class="figure-caption">Figure 2.7: PCA projections colored by different factors and organized by different components. (A) Plot of the first two components colored by cohort. (B) Plot of the first two components colored by ER status. (C) Plot of the second and third components colored by cohort. (D) Plot of the second and third components colored by ER status. (E) Plot of the second and third components colored by PAM50. (F) Plot of the second and third components colored by INTCLUST, only METABRIC has an assigned value for this variable, NAs are TCGA samples.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>The first three components correspond to the batch effect. The third component explains the ER status and the two cohorts are well inter mingled. When plotting the third and fourth components and coloring by the molecular subtype, we see a clear distinction, recapitulating the biology.</p>
<p>The plot below shows how the loadings of the first component are really concentrated around zero, so there are not actually specific genes that are accounting for the batch effects we see in the first component. We show latter that the genes that are the most important are the ones with high loading, so the tail of the distributions. In this sense there are really no genes in the tail of the loadings for PC1.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-loadings-pcs" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="pca_merging_files/figure-html/fig-loadings-pcs-1.png" class="img-fluid figure-img" width="672"></p>
<p></p><figcaption class="figure-caption">Figure 2.8: Different loadings for the first 4 components.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<div class="cell">

</div>
<div class="cell">

</div>
<p><a href="#fig-pca-og">Figure&nbsp;<span>2.9</span></a> shows all the samples from TCGA and cohort projected into the molecular landscape. The samples are mixing well, the only thing is that METABRIC is a bit offset to the left bottom.</p>
<div class="cell">

</div>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pca-og" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="pca_merging_files/figure-html/fig-pca-og-1.png" class="img-fluid figure-img" width="672"></p>
<p></p><figcaption class="figure-caption">Figure 2.9: Original PC3 and PC4 of all samples, including test and train samples, of TCGA and METABRIC.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>And <a href="#fig-pca-pam50">Figure&nbsp;<span>2.10</span></a>.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pca-pam50" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="pca_merging_files/figure-html/fig-pca-pam50-1.png" class="img-fluid figure-img" width="672"></p>
<p></p><figcaption class="figure-caption">Figure 2.10: Original PC3 and PC4 of all samples, including test and train samples, of TCGA and METABRIC.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>In order to check if the integration made sense, it is possible to color by ER status, NPI for METABRIC samples and tumor/node stage for TCGA. To check ER signaling scores, the pathways Estrogen response early/late and <span class="math inline">\(SET_{ER/PR}\)</span> are used to color the projection.</p>
<p><a href="#fig-pca-age">Figure&nbsp;<span>2.11</span></a> shows the coloring by age.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pca-age" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="pca_merging_files/figure-html/fig-pca-age-1.png" class="img-fluid figure-img" width="672"></p>
<p></p><figcaption class="figure-caption">Figure 2.11: PC3 and PC4 of samples from TCGA and METABRIC colored by age.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>Age seems to be well mixed. <a href="#fig-pca-er-status">Figure&nbsp;<span>2.12</span></a> shows the PCA colored by ER status.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pca-er-status" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="pca_merging_files/figure-html/fig-pca-er-status-1.png" class="img-fluid figure-img" width="672"></p>
<p></p><figcaption class="figure-caption">Figure 2.12: PC3 and PC4 of samples from TCGA and METABRIC colored by ER status.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>It has a very good separation among the different patients. <a href="#fig-pca-npi">Figure&nbsp;<span>2.13</span></a> is the PCA embedding colored by NPI. Only METABRIC samples have this information.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pca-npi" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="pca_merging_files/figure-html/fig-pca-npi-1.png" class="img-fluid figure-img" width="672"></p>
<p></p><figcaption class="figure-caption">Figure 2.13: PC3 and PC4 of samples from METABRIC only colored by NPI.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>It looks good, ER+ BC patients have lower NPI, which makes sense. <a href="#fig-pca-tumor-purity">Figure&nbsp;<span>2.14</span></a> shows the biplot colored by tumor purity.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pca-tumor-purity" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="pca_merging_files/figure-html/fig-pca-tumor-purity-1.png" class="img-fluid figure-img" width="672"></p>
<p></p><figcaption class="figure-caption">Figure 2.14: PC3 and PC4 of samples from METABRIC only colored by tumor purity.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>The tumor purity is not a factor that is affecting the embedding as samples with low cellularity (&lt; 40% of tumor cells) and high cellularity (&gt; 70% of tumor cells) intermingle.</p>
<p>METABRIC has also inferred menopausal status for the patients, we use this information to check if there is any enrichment based on the patient menopausal status (<a href="#fig-pca-men-status">Figure&nbsp;<span>2.15</span></a>)</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pca-men-status" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="pca_merging_files/figure-html/fig-pca-men-status-1.png" class="img-fluid figure-img" width="672"></p>
<p></p><figcaption class="figure-caption">Figure 2.15: PC3 and PC4 of samples from METABRIC only colored by menopausal status.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>We don’t see any difference between the two status. We proceed now with the analysis using clinical factors from TCGA.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pca-tumor-node" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="pca_merging_files/figure-html/fig-pca-tumor-node-1.png" class="img-fluid figure-img" width="1152"></p>
<p></p><figcaption class="figure-caption">Figure 2.16: PC3 and PC4 of samples from TCGA only colored by tumor stage and node stage.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>For both tumor and node stage the samples seem to have mixed well, indicating the projection is not a function of the stages, as shown in <a href="#fig-pca-tumor-node">Figure&nbsp;<span>2.16</span></a>.</p>
<p>And finally <a href="#fig-pca-pam50-2">Figure&nbsp;<span>2.17</span></a> below shows the color by PAM50 molecular subtype.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pca-pam50-2" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="pca_merging_files/figure-html/fig-pca-pam50-2-1.png" class="img-fluid figure-img" width="672"></p>
<p></p><figcaption class="figure-caption">Figure 2.17: Corrected PC3 and PC4 of samples from TCGA and METABRIC colored by PAM50 molecular subtype.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>The molecular subtypes are projected as expected, going from left to right there are the basal-like BC patients, going to HER2 and then they are split between luminal A and B. The normal like patients are in the top and closer to the luminal A, reflecting the biology.</p>
<p>The next figures show the ER signaling measures coloring the projections.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pca-seterpr" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="pca_merging_files/figure-html/fig-pca-seterpr-1.png" class="img-fluid figure-img" width="672"></p>
<p></p><figcaption class="figure-caption">Figure 2.18: biplot of PC3 and PC4 of samples from TCGA and METABRIC colored by the SET ER/PR score.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pca-early" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="pca_merging_files/figure-html/fig-pca-early-1.png" class="img-fluid figure-img" width="672"></p>
<p></p><figcaption class="figure-caption">Figure 2.19: biplot of PC3 and PC4 of samples from TCGA and METABRIC colored by the hallmark estrogen response early.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pca-late" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="pca_merging_files/figure-html/fig-pca-late-1.png" class="img-fluid figure-img" width="672"></p>
<p></p><figcaption class="figure-caption">Figure 2.20: biplot of PC3 and PC4 of samples from TCGA and METABRIC colored by the hallmark estrogen response late</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>In all cases the mixing is relatively good. For <span class="math inline">\(SET_{ER/PR}\)</span> you have a gradient going from the right to the left, showing again that patients on the far top right probably are those patients with better prognosis.</p>
</section>
<section id="the-integration-is-robust-with-respect-of-selection-of-samples" class="level2" data-number="2.3">
<h2 data-number="2.3" class="anchored" data-anchor-id="the-integration-is-robust-with-respect-of-selection-of-samples"><span class="header-section-number">2.3</span> The integration is robust with respect of selection of samples</h2>
<p>The embedding could be due to the patients selected and samples that are close to each other in one embedding would be far away from them in another embedding. Ten random sets of 1000 patients are sampled and the embedding is calculate. A total of 10 PCA embeddings are obtained.</p>
<div class="cell">

</div>
<div class="cell">

</div>
<div class="cell">
<div class="cell-output-display">
<div id="fig-plots-pcas-random" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="pca_merging_files/figure-html/fig-plots-pcas-random-1.png" class="img-fluid figure-img" width="1728"></p>
<p></p><figcaption class="figure-caption">Figure 2.21: Selecting a random set of 1000 patients an rerunning the pipeline to obtain other embeddings. Red color corresponds to the embedding using the new fitted PCA. Blue corresponds to the embedding of the original PCA. Each dot corresponds to a patient.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p><a href="#fig-plots-pcas-random">Figure&nbsp;<span>2.21</span></a> shows that the embedding is invariant for rotation, translation and reflection. Therefore, the subset of samples used for the embedding does not really matter.</p>
</section>
<section id="normalization-is-necessary-to-reduce-batch-effects-in-other-components" class="level2" data-number="2.4">
<h2 data-number="2.4" class="anchored" data-anchor-id="normalization-is-necessary-to-reduce-batch-effects-in-other-components"><span class="header-section-number">2.4</span> Normalization is necessary to reduce batch effects in other components</h2>
<p>Since PCA is removing batch effects from the datasets, one could argue it could also remove platform dependent normalization values and therefore one would need to look only at lower principal components. In this section we show that doing the normalization is necessary to get a good integration in the datasets.</p>
<p>The idea is similar as before, but instead of performing the <em>qPCR</em>-like normalization on the datasets, one would use the logFPKM values for TCGA and median intensity for microarray. The problem with this kind of approach also includes the fact that only other samples that went through this processing could be used in the pipeline.</p>
<div class="cell">

</div>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pca-no-norm" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="pca_merging_files/figure-html/fig-pca-no-norm-1.png" class="img-fluid figure-img" width="1920"></p>
<p></p><figcaption class="figure-caption">Figure 2.22: PCA projections colored by different factors and organized by different components. (A) Plot of the first two components colored by cohort. (B) Plot of the first two components colored by ER status. (C) Plot of the second and third components colored by cohort. (D) Plot of the second and third components colored by ER status. (E) Plot of the second and third components colored by PAM50.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p><a href="#fig-pca-no-norm">Figure&nbsp;<span>2.22</span></a> shows the contrast with the other method including the normalization. Without the normalization, the samples from METABRIC tend to concentrate close to the center, they are not so well mixed.</p>
<div class="cell">

</div>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pca-norm-no-norm" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="pca_merging_files/figure-html/fig-pca-norm-no-norm-1.png" class="img-fluid figure-img" width="960"></p>
<p></p><figcaption class="figure-caption">Figure 2.23: PC3 and PC4 from all METABRIC and TCGA samples. (A) Embedding using the qPCR-like normalization. (B) Embedding using the original normalization from each dataset.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p><a href="#fig-pca-norm-no-norm">Figure&nbsp;<span>2.23</span></a> shows side by side the differences when using the qPCR-like normalization and when not using. The left plot is using the qPCR normalization and shows how well the mixing gets, putting samples in the same scale. If the scaling in a sample level is not performed, samples do not mix well.</p>
</section>
<section id="molecular-landscape-is-characterized-by-several-pathways" class="level2" data-number="2.5">
<h2 data-number="2.5" class="anchored" data-anchor-id="molecular-landscape-is-characterized-by-several-pathways"><span class="header-section-number">2.5</span> Molecular landscape is characterized by several pathways</h2>
<p>Among the 1000 genes that are used when calculating the molecular landscape, several might be involved in biological process. <a href="#fig-pca-mol-land-pathways">Figure&nbsp;<span>2.24</span></a> shows that several pathways are associated with the position in the molecular landscape. Something that is particularly remarkable here is the association of the epithelial mesenchymal transition and DNA repair pathways with PC4.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pca-mol-land-pathways" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="pca_merging_files/figure-html/fig-pca-mol-land-pathways-1.png" class="img-fluid figure-img" width="576"></p>
<p></p><figcaption class="figure-caption">Figure 2.24: Hex biplots colored by different pathways. Each hex is colored based on the average score in the hex.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>We see that these pathways are somehow relevant to the embedding, but are they also prognostic of overall survival in the two cohorts? In the next chunks we evaluate this hypothesis. Mostly the focus is on ER+ BC patients that received some kind of hormonal therapy. As seen before, for the TCGA dataset we select ER+ BC patients that have the molecular subtype luminal A and B instead.</p>
<div class="cell">

</div>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pca-mol-land-survival" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="pca_merging_files/figure-html/fig-pca-mol-land-survival-1.png" class="img-fluid figure-img" width="576"></p>
<p></p><figcaption class="figure-caption">Figure 2.25: Overall survival analysis from all the three big cohorts and their scores. The formulas used were the same as for the estrogen signaling analysis.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>We see that G2M checkpoint has hazard ratio higher than 1. The other scores are a bit more variable and can change the sign (type S error).</p>


<!-- -->

<div id="refs" class="references csl-bib-body hanging-indent" role="doc-bibliography" style="display: none">
<div id="ref-Bhuva2020" class="csl-entry" role="doc-biblioentry">
Bhuva, Dharmesh D, Joseph Cursons, and Melissa J Davis. 2020. <span>“Stable Gene Expression for Normalisation and Single-Sample Scoring.”</span> <em>Nucleic Acids Research</em> 48 (19): e113–13. <a href="https://doi.org/10.1093/nar/gkaa802">https://doi.org/10.1093/nar/gkaa802</a>.
</div>
<div id="ref-Fei2018" class="csl-entry" role="doc-biblioentry">
Fei, Teng, Tengjiao Zhang, Weiyang Shi, and Tianwei Yu. 2018. <span>“Mitigating the Adverse Impact of Batch Effects in Sample Pattern Detection.”</span> Edited by Inanc Birol. <em>Bioinformatics</em> 34 (15): 2634–41. <a href="https://doi.org/10.1093/bioinformatics/bty117">https://doi.org/10.1093/bioinformatics/bty117</a>.
</div>
<div id="ref-Paik2004" class="csl-entry" role="doc-biblioentry">
Paik, Soonmyung, Steven Shak, Gong Tang, Chungyeul Kim, Joffre Baker, Maureen Cronin, Frederick L. Baehner, et al. 2004. <span>“A Multigene Assay to Predict Recurrence of Tamoxifen-Treated, Node-Negative Breast Cancer.”</span> <em>New England Journal of Medicine</em> 351 (27): 2817–26. <a href="https://doi.org/10.1056/nejmoa041588">https://doi.org/10.1056/nejmoa041588</a>.
</div>
<div id="ref-Risso2014" class="csl-entry" role="doc-biblioentry">
Risso, Davide, John Ngai, Terence P Speed, and Sandrine Dudoit. 2014. <span>“Normalization of <span>RNA</span>-Seq Data Using Factor Analysis of Control Genes or Samples.”</span> <em>Nature Biotechnology</em> 32 (9): 896–902. <a href="https://doi.org/10.1038/nbt.2931">https://doi.org/10.1038/nbt.2931</a>.
</div>
<div id="ref-vandeVijver2002" class="csl-entry" role="doc-biblioentry">
Vijver, Marc J. van de, Yudong D. He, Laura J. van t Veer, Hongyue Dai, Augustinus A. M. Hart, Dorien W. Voskuil, George J. Schreiber, et al. 2002. <span>“A Gene-Expression Signature as a Predictor of Survival in Breast Cancer.”</span> <em>New England Journal of Medicine</em> 347 (25): 1999–2009. <a href="https://doi.org/10.1056/nejmoa021967">https://doi.org/10.1056/nejmoa021967</a>.
</div>
<div id="ref-Wallden2015" class="csl-entry" role="doc-biblioentry">
Wallden, Brett, James Storhoff, Torsten Nielsen, Naeem Dowidar, Carl Schaper, Sean Ferree, Shuzhen Liu, et al. 2015. <span>“Development and Verification of the <span>PAM</span>50-Based Prosigna Breast Cancer Gene Signature Assay.”</span> <em><span>BMC</span> Medical Genomics</em> 8 (1). <a href="https://doi.org/10.1186/s12920-015-0129-6">https://doi.org/10.1186/s12920-015-0129-6</a>.
</div>
<div id="ref-Wishart2010" class="csl-entry" role="doc-biblioentry">
Wishart, Gordon C, Elizabeth M Azzato, David C Greenberg, Jem Rashbass, Olive Kearins, Gill Lawrence, Carlos Caldas, and Paul DP Pharoah. 2010. <span>“<span>PREDICT</span>: A New <span>UK</span> Prognostic Model That Predicts Survival Following Surgery for Invasive Breast Cancer.”</span> <em>Breast Cancer Research</em> 12 (1). <a href="https://doi.org/10.1186/bcr2464">https://doi.org/10.1186/bcr2464</a>.
</div>
<div id="ref-Zhang2020" class="csl-entry" role="doc-biblioentry">
Zhang, Yuqing, Giovanni Parmigiani, and W Evan Johnson. 2020. <span>“<span>ComBat</span>-Seq: Batch Effect Adjustment for <span>RNA</span>-Seq Count Data.”</span> <em><span>NAR</span> Genomics and Bioinformatics</em> 2 (3). <a href="https://doi.org/10.1093/nargab/lqaa078">https://doi.org/10.1093/nargab/lqaa078</a>.
</div>
</div>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      let href = ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./surv_analysis_estrogen.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Estrogen receptor status is continuous, not dichotomous</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./validation.html" class="pagination-link">
        <span class="nav-page-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Validating the molecular landscape</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb2" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="an">editor_options:</span><span class="co"> </span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="co">  markdown: </span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="co">    wrap: 80</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a><span class="fu"># A new framework for personalized medicine: integrating</span></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>In the next few chapters we will show a new framework for personalized medicine.</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>This is motivated by the fact that we know already some patients are more</span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>sensitive to endocrine therapy than others, but we don't know what are possible</span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>treatments. Moreover, it is very difficult to compare, molecularly, patients</span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>among themselves.</span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>The PREDICT tool <span class="co">[</span><span class="ot">@Wishart2010</span><span class="co">]</span> provides a tool for practitioners to calculate</span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>survival risks among patients with similar clinical characteristics. This is</span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a>very important as it shows how to combine data from previous patients to guide</span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a>further treatments of new patients. However, no additional information is given,</span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a>such as alternative pathways to target or what might be different from other</span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a>patients.</span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a>Other tools that help in the clinics are the molecular signatures developed by</span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a>companies <span class="co">[</span><span class="ot">@Wallden2015; @vandeVijver2002; @Paik2004</span><span class="co">]</span>. These signatures are</span>
<span id="cb2-24"><a href="#cb2-24" aria-hidden="true" tabindex="-1"></a>intended to be used with ER+ BC patients and depending on the tool, on node</span>
<span id="cb2-25"><a href="#cb2-25" aria-hidden="true" tabindex="-1"></a>negative, post-menopausal women. Based on the set of genes, a risk score is</span>
<span id="cb2-26"><a href="#cb2-26" aria-hidden="true" tabindex="-1"></a>assigned or each patient. The score means if a patient would benefit from</span>
<span id="cb2-27"><a href="#cb2-27" aria-hidden="true" tabindex="-1"></a>additional chemotherapy besides the usual endocrine therapy received for either</span>
<span id="cb2-28"><a href="#cb2-28" aria-hidden="true" tabindex="-1"></a>5 or 10 years. They also lack the additional information of what are possible</span>
<span id="cb2-29"><a href="#cb2-29" aria-hidden="true" tabindex="-1"></a>pathways involved and alternative treatments beside chemotherapy.</span>
<span id="cb2-30"><a href="#cb2-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-31"><a href="#cb2-31" aria-hidden="true" tabindex="-1"></a>There are a few challenges to overcome pathway analysis for patients</span>
<span id="cb2-32"><a href="#cb2-32" aria-hidden="true" tabindex="-1"></a>individually and how to compare patients molecularly. There is no tool that</span>
<span id="cb2-33"><a href="#cb2-33" aria-hidden="true" tabindex="-1"></a>allows to integrate patients in a continuous way. Usually integration is a one</span>
<span id="cb2-34"><a href="#cb2-34" aria-hidden="true" tabindex="-1"></a>step procedure and cannot be updated. The usual tools are [@Risso2014;</span>
<span id="cb2-35"><a href="#cb2-35" aria-hidden="true" tabindex="-1"></a>@Zhang2020; @Fei2018] and they do not provide batch effect removal for new</span>
<span id="cb2-36"><a href="#cb2-36" aria-hidden="true" tabindex="-1"></a>samples. The only way is to re-run the procedure together with the new sample.</span>
<span id="cb2-37"><a href="#cb2-37" aria-hidden="true" tabindex="-1"></a>The problem of doing this is usually you don't have enough samples to estimate</span>
<span id="cb2-38"><a href="#cb2-38" aria-hidden="true" tabindex="-1"></a>batch effects across groups and therefore correct it.</span>
<span id="cb2-39"><a href="#cb2-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-40"><a href="#cb2-40" aria-hidden="true" tabindex="-1"></a>This chapter show how to perform PCA projection using samples from different</span>
<span id="cb2-41"><a href="#cb2-41" aria-hidden="true" tabindex="-1"></a>datasets and how new samples can be introduced without retraining of the PCA. We</span>
<span id="cb2-42"><a href="#cb2-42" aria-hidden="true" tabindex="-1"></a>first introduce the notion of a new normalization that depends on housekeeping</span>
<span id="cb2-43"><a href="#cb2-43" aria-hidden="true" tabindex="-1"></a>genes. After using this new normalization, we proceed with PCA on a subset of</span>
<span id="cb2-44"><a href="#cb2-44" aria-hidden="true" tabindex="-1"></a>two cohorts using the 1000 most variable genes. We then validate the integration</span>
<span id="cb2-45"><a href="#cb2-45" aria-hidden="true" tabindex="-1"></a>with new samples from a completely different cohort.</span>
<span id="cb2-46"><a href="#cb2-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-47"><a href="#cb2-47" aria-hidden="true" tabindex="-1"></a><span class="in">```{r setup}</span></span>
<span id="cb2-48"><a href="#cb2-48" aria-hidden="true" tabindex="-1"></a>renv<span class="sc">::</span><span class="fu">restore</span>()</span>
<span id="cb2-49"><a href="#cb2-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-50"><a href="#cb2-50" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb2-51"><a href="#cb2-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-52"><a href="#cb2-52" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb2-53"><a href="#cb2-53" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(PCAtools)</span>
<span id="cb2-54"><a href="#cb2-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-55"><a href="#cb2-55" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(singscore)</span>
<span id="cb2-56"><a href="#cb2-56" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SummarizedExperiment)</span>
<span id="cb2-57"><a href="#cb2-57" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(survival)</span>
<span id="cb2-58"><a href="#cb2-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-59"><a href="#cb2-59" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"utils.R"</span>)</span>
<span id="cb2-60"><a href="#cb2-60" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"first_run.R"</span>)</span>
<span id="cb2-61"><a href="#cb2-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-62"><a href="#cb2-62" aria-hidden="true" tabindex="-1"></a><span class="co"># the following script load all data necessary to run the chunks.</span></span>
<span id="cb2-63"><a href="#cb2-63" aria-hidden="true" tabindex="-1"></a><span class="co"># the data is generated from this quarto document itself, therefore</span></span>
<span id="cb2-64"><a href="#cb2-64" aria-hidden="true" tabindex="-1"></a><span class="co"># if you are running this documents the first time and don't have the</span></span>
<span id="cb2-65"><a href="#cb2-65" aria-hidden="true" tabindex="-1"></a><span class="co"># files, comment the following lines. Moreover, if this is your first</span></span>
<span id="cb2-66"><a href="#cb2-66" aria-hidden="true" tabindex="-1"></a><span class="co"># time running the document, you should run all chunks, to generate </span></span>
<span id="cb2-67"><a href="#cb2-67" aria-hidden="true" tabindex="-1"></a><span class="co"># all the necessary files, if you don't have them. Once all files </span></span>
<span id="cb2-68"><a href="#cb2-68" aria-hidden="true" tabindex="-1"></a><span class="co"># are saved and available in the respective folder, the following</span></span>
<span id="cb2-69"><a href="#cb2-69" aria-hidden="true" tabindex="-1"></a><span class="co"># lines can be executed. </span></span>
<span id="cb2-70"><a href="#cb2-70" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (first_run){</span>
<span id="cb2-71"><a href="#cb2-71" aria-hidden="true" tabindex="-1"></a>    load_at_setup <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb2-72"><a href="#cb2-72" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb2-73"><a href="#cb2-73" aria-hidden="true" tabindex="-1"></a>    load_at_setup <span class="ot">&lt;-</span> <span class="cn">TRUE</span></span>
<span id="cb2-74"><a href="#cb2-74" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb2-75"><a href="#cb2-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-76"><a href="#cb2-76" aria-hidden="true" tabindex="-1"></a>name_document <span class="ot">&lt;-</span> <span class="st">"pca_merging"</span></span>
<span id="cb2-77"><a href="#cb2-77" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"load_rds_files.R"</span>)</span>
<span id="cb2-78"><a href="#cb2-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-79"><a href="#cb2-79" aria-hidden="true" tabindex="-1"></a><span class="co"># by setting the dev to png and pdf, this saves the figures in a specific</span></span>
<span id="cb2-80"><a href="#cb2-80" aria-hidden="true" tabindex="-1"></a><span class="co"># folder in both formats. moreover, since png is coming first, it shows</span></span>
<span id="cb2-81"><a href="#cb2-81" aria-hidden="true" tabindex="-1"></a><span class="co"># this figure when rendering the html. What is nice about this is that it</span></span>
<span id="cb2-82"><a href="#cb2-82" aria-hidden="true" tabindex="-1"></a><span class="co"># inherits the properties from the chunk to save the figure, so no need</span></span>
<span id="cb2-83"><a href="#cb2-83" aria-hidden="true" tabindex="-1"></a><span class="co"># to use ggplot2::ggsave to save the plots. this also works </span></span>
<span id="cb2-84"><a href="#cb2-84" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">dev =</span> <span class="fu">c</span>(<span class="st">'png'</span>, <span class="st">'pdf'</span>, <span class="st">'svg'</span>))</span>
<span id="cb2-85"><a href="#cb2-85" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">bitmapType =</span> <span class="st">'cairo'</span>)</span>
<span id="cb2-86"><a href="#cb2-86" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb2-87"><a href="#cb2-87" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-88"><a href="#cb2-88" aria-hidden="true" tabindex="-1"></a><span class="fu">## Transcriptomics and qPCR normalization</span></span>
<span id="cb2-89"><a href="#cb2-89" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-90"><a href="#cb2-90" aria-hidden="true" tabindex="-1"></a>In this first step we introduce a new normalization. The rational is that we</span>
<span id="cb2-91"><a href="#cb2-91" aria-hidden="true" tabindex="-1"></a>need a normalization procedure that is applied in a single sample only, similar</span>
<span id="cb2-92"><a href="#cb2-92" aria-hidden="true" tabindex="-1"></a>to logCPM, but that can be applied to microarray too. In this case, given a set</span>
<span id="cb2-93"><a href="#cb2-93" aria-hidden="true" tabindex="-1"></a>of samples, we start by ranking the genes in each sample. We then normalize the</span>
<span id="cb2-94"><a href="#cb2-94" aria-hidden="true" tabindex="-1"></a>ranking to be between 0 and 1, i.e., we divide by the total number of genes</span>
<span id="cb2-95"><a href="#cb2-95" aria-hidden="true" tabindex="-1"></a>available in that sample. Then, given a list of 50 stable genes across cancer</span>
<span id="cb2-96"><a href="#cb2-96" aria-hidden="true" tabindex="-1"></a><span class="co">[</span><span class="ot">@Bhuva2020</span><span class="co">]</span>, we calculate the average ranking of these genes in the sample and</span>
<span id="cb2-97"><a href="#cb2-97" aria-hidden="true" tabindex="-1"></a>then for each gene in the sample we divide by this average.</span>
<span id="cb2-98"><a href="#cb2-98" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-99"><a href="#cb2-99" aria-hidden="true" tabindex="-1"></a><span class="fu">### Selecting stable genes</span></span>
<span id="cb2-100"><a href="#cb2-100" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-101"><a href="#cb2-101" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb2-102"><a href="#cb2-102" aria-hidden="true" tabindex="-1"></a>datasets <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(</span>
<span id="cb2-103"><a href="#cb2-103" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/surv_analysis_estrogen/datasets_with_scores.rds"</span></span>
<span id="cb2-104"><a href="#cb2-104" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-105"><a href="#cb2-105" aria-hidden="true" tabindex="-1"></a>stable_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(</span>
<span id="cb2-106"><a href="#cb2-106" aria-hidden="true" tabindex="-1"></a>    singscore<span class="sc">::</span><span class="fu">getStableGenes</span>(<span class="at">n_stable =</span> <span class="dv">50</span>, <span class="at">type =</span> <span class="st">"carcinoma"</span>),</span>
<span id="cb2-107"><a href="#cb2-107" aria-hidden="true" tabindex="-1"></a>    <span class="fu">Reduce</span>(intersect, <span class="fu">lapply</span>(datasets[<span class="fu">c</span>(<span class="st">"tcga"</span>, <span class="st">"metabric"</span>)], rownames))</span>
<span id="cb2-108"><a href="#cb2-108" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-109"><a href="#cb2-109" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb2-110"><a href="#cb2-110" aria-hidden="true" tabindex="-1"></a>    stable_genes,</span>
<span id="cb2-111"><a href="#cb2-111" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/pca_merging/stable_genes.rds"</span></span>
<span id="cb2-112"><a href="#cb2-112" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-113"><a href="#cb2-113" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb2-114"><a href="#cb2-114" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-117"><a href="#cb2-117" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb2-118"><a href="#cb2-118" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span>load_at_setup){</span>
<span id="cb2-119"><a href="#cb2-119" aria-hidden="true" tabindex="-1"></a>    datasets <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(</span>
<span id="cb2-120"><a href="#cb2-120" aria-hidden="true" tabindex="-1"></a>        <span class="st">"../results/rds_files/surv_analysis_estrogen/datasets_with_scores.rds"</span></span>
<span id="cb2-121"><a href="#cb2-121" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb2-122"><a href="#cb2-122" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb2-123"><a href="#cb2-123" aria-hidden="true" tabindex="-1"></a>    stable_genes <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(</span>
<span id="cb2-124"><a href="#cb2-124" aria-hidden="true" tabindex="-1"></a>        <span class="st">"../results/rds_files/pca_merging/stable_genes.rds"</span></span>
<span id="cb2-125"><a href="#cb2-125" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb2-126"><a href="#cb2-126" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb2-127"><a href="#cb2-127" aria-hidden="true" tabindex="-1"></a>    which_exp <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(</span>
<span id="cb2-128"><a href="#cb2-128" aria-hidden="true" tabindex="-1"></a>        <span class="st">"../results/rds_files/surv_analysis_estrogen/which_exp.rds"</span></span>
<span id="cb2-129"><a href="#cb2-129" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb2-130"><a href="#cb2-130" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb2-131"><a href="#cb2-131" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb2-132"><a href="#cb2-132" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-133"><a href="#cb2-133" aria-hidden="true" tabindex="-1"></a>The first step is to check what number of these 50 genes are available on TCGA</span>
<span id="cb2-134"><a href="#cb2-134" aria-hidden="true" tabindex="-1"></a>and METABRIC, as these datasets will be used for training the PCA projection.</span>
<span id="cb2-135"><a href="#cb2-135" aria-hidden="true" tabindex="-1"></a>The total number of available genes is <span class="in">`r length(stable_genes)`</span>.</span>
<span id="cb2-136"><a href="#cb2-136" aria-hidden="true" tabindex="-1"></a><span class="co">[</span><span class="ot">Table @fig-stable-genes</span><span class="co">]</span> shows the name of the genes </span>
<span id="cb2-137"><a href="#cb2-137" aria-hidden="true" tabindex="-1"></a>available in TCGA and METABRIC.</span>
<span id="cb2-138"><a href="#cb2-138" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-139"><a href="#cb2-139" aria-hidden="true" tabindex="-1"></a>Now we select the 1000 most variable genes across METABRIC and TCGA. Reducing</span>
<span id="cb2-140"><a href="#cb2-140" aria-hidden="true" tabindex="-1"></a>the number of genes used when calculating the PCA helps in the future. Several</span>
<span id="cb2-141"><a href="#cb2-141" aria-hidden="true" tabindex="-1"></a>publicly available datasets don't have many genes available, therefore reducing</span>
<span id="cb2-142"><a href="#cb2-142" aria-hidden="true" tabindex="-1"></a>the number of genes increases the chances of using this approach in other</span>
<span id="cb2-143"><a href="#cb2-143" aria-hidden="true" tabindex="-1"></a>datasets.</span>
<span id="cb2-144"><a href="#cb2-144" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-145"><a href="#cb2-145" aria-hidden="true" tabindex="-1"></a>Moreover, the use of a large number of genes is used because if some of them</span>
<span id="cb2-146"><a href="#cb2-146" aria-hidden="true" tabindex="-1"></a>are not available. We show in later sections that if some number of stable</span>
<span id="cb2-147"><a href="#cb2-147" aria-hidden="true" tabindex="-1"></a>genes are missing, it will not affect the normalization. </span>
<span id="cb2-148"><a href="#cb2-148" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-151"><a href="#cb2-151" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb2-152"><a href="#cb2-152" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-stable-genes</span></span>
<span id="cb2-153"><a href="#cb2-153" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: ([Table @fig-stable-genes]) Available stable genes </span></span>
<span id="cb2-154"><a href="#cb2-154" aria-hidden="true" tabindex="-1"></a><span class="co">#|     in both TCGA and METABRIC cohorts.</span></span>
<span id="cb2-155"><a href="#cb2-155" aria-hidden="true" tabindex="-1"></a>DT<span class="sc">::</span><span class="fu">datatable</span>(</span>
<span id="cb2-156"><a href="#cb2-156" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>(<span class="at">genes =</span> stable_genes),</span>
<span id="cb2-157"><a href="#cb2-157" aria-hidden="true" tabindex="-1"></a>    <span class="at">options =</span> <span class="fu">list</span>(<span class="at">scrollX =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-158"><a href="#cb2-158" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-159"><a href="#cb2-159" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb2-160"><a href="#cb2-160" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-161"><a href="#cb2-161" aria-hidden="true" tabindex="-1"></a><span class="fu">### Calculating the normalization</span></span>
<span id="cb2-162"><a href="#cb2-162" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-163"><a href="#cb2-163" aria-hidden="true" tabindex="-1"></a>We will create subset of the summarised experiment objects to save these</span>
<span id="cb2-164"><a href="#cb2-164" aria-hidden="true" tabindex="-1"></a>information. Moreover, since a summarized experiment object is being used, we</span>
<span id="cb2-165"><a href="#cb2-165" aria-hidden="true" tabindex="-1"></a>can save the new normalization in the <span class="in">`assays`</span> and the average ranking scores of</span>
<span id="cb2-166"><a href="#cb2-166" aria-hidden="true" tabindex="-1"></a>the housekeeping genes in the column data <span class="in">`colData`</span>. This way it is very easy to</span>
<span id="cb2-167"><a href="#cb2-167" aria-hidden="true" tabindex="-1"></a>retrieve the information for the patients.</span>
<span id="cb2-168"><a href="#cb2-168" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-171"><a href="#cb2-171" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb2-172"><a href="#cb2-172" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-comparison-sd-cv</span></span>
<span id="cb2-173"><a href="#cb2-173" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Comparison of coefficient of variation and standard deviation</span></span>
<span id="cb2-174"><a href="#cb2-174" aria-hidden="true" tabindex="-1"></a><span class="co">#|     scores. </span></span>
<span id="cb2-175"><a href="#cb2-175" aria-hidden="true" tabindex="-1"></a><span class="co"># the strategy is to calculate the standard deviation for each cohort separately</span></span>
<span id="cb2-176"><a href="#cb2-176" aria-hidden="true" tabindex="-1"></a><span class="co"># and then select the genes with the highest average standard deviation and</span></span>
<span id="cb2-177"><a href="#cb2-177" aria-hidden="true" tabindex="-1"></a><span class="co"># then we include the stable genes as well</span></span>
<span id="cb2-178"><a href="#cb2-178" aria-hidden="true" tabindex="-1"></a>common_genes <span class="ot">&lt;-</span> <span class="fu">Reduce</span>(intersect, <span class="fu">lapply</span>(datasets, rownames))</span>
<span id="cb2-179"><a href="#cb2-179" aria-hidden="true" tabindex="-1"></a>nb_tcga <span class="ot">&lt;-</span> <span class="fu">ncol</span>(datasets<span class="sc">$</span>tcga)</span>
<span id="cb2-180"><a href="#cb2-180" aria-hidden="true" tabindex="-1"></a>nb_metabric <span class="ot">&lt;-</span> <span class="fu">ncol</span>(datasets<span class="sc">$</span>metabric)</span>
<span id="cb2-181"><a href="#cb2-181" aria-hidden="true" tabindex="-1"></a>cv_genes <span class="ot">&lt;-</span> <span class="fu">mapply</span>(</span>
<span id="cb2-182"><a href="#cb2-182" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(df, assay_to_use, common_genes){</span>
<span id="cb2-183"><a href="#cb2-183" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb2-184"><a href="#cb2-184" aria-hidden="true" tabindex="-1"></a>        <span class="co"># select first common genes across the two cohorts, so the identified</span></span>
<span id="cb2-185"><a href="#cb2-185" aria-hidden="true" tabindex="-1"></a>        <span class="co"># genes are the same</span></span>
<span id="cb2-186"><a href="#cb2-186" aria-hidden="true" tabindex="-1"></a>        df <span class="ot">&lt;-</span> df[common_genes, ]</span>
<span id="cb2-187"><a href="#cb2-187" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb2-188"><a href="#cb2-188" aria-hidden="true" tabindex="-1"></a>        <span class="fu">rowSds</span>(<span class="fu">as.matrix</span>(<span class="fu">assay</span>(df, assay_to_use)))<span class="sc">/</span></span>
<span id="cb2-189"><a href="#cb2-189" aria-hidden="true" tabindex="-1"></a>            <span class="fu">rowMeans</span>(<span class="fu">as.matrix</span>(<span class="fu">assay</span>(df, assay_to_use)))</span>
<span id="cb2-190"><a href="#cb2-190" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb2-191"><a href="#cb2-191" aria-hidden="true" tabindex="-1"></a>    <span class="at">df =</span> datasets[<span class="fu">c</span>(<span class="st">"tcga"</span>, <span class="st">"metabric"</span>)],</span>
<span id="cb2-192"><a href="#cb2-192" aria-hidden="true" tabindex="-1"></a>    <span class="at">assay_to_use =</span> <span class="fu">list</span>(</span>
<span id="cb2-193"><a href="#cb2-193" aria-hidden="true" tabindex="-1"></a>        <span class="at">tcga =</span> <span class="st">"vst"</span>,</span>
<span id="cb2-194"><a href="#cb2-194" aria-hidden="true" tabindex="-1"></a>        <span class="at">metabric =</span> <span class="st">"median_intensity"</span></span>
<span id="cb2-195"><a href="#cb2-195" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb2-196"><a href="#cb2-196" aria-hidden="true" tabindex="-1"></a>    <span class="at">MoreArgs =</span> <span class="fu">list</span>(<span class="at">common_genes =</span> common_genes)</span>
<span id="cb2-197"><a href="#cb2-197" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> data.frame <span class="sc">%&gt;%</span></span>
<span id="cb2-198"><a href="#cb2-198" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb2-199"><a href="#cb2-199" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">average =</span> <span class="fu">mean</span>(<span class="fu">c</span>(tcga, metabric))) <span class="sc">%&gt;%</span></span>
<span id="cb2-200"><a href="#cb2-200" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb2-201"><a href="#cb2-201" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">symbol =</span> common_genes) </span>
<span id="cb2-202"><a href="#cb2-202" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-203"><a href="#cb2-203" aria-hidden="true" tabindex="-1"></a>sd_genes <span class="ot">&lt;-</span> <span class="fu">mapply</span>(</span>
<span id="cb2-204"><a href="#cb2-204" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(df, assay_to_use, common_genes){</span>
<span id="cb2-205"><a href="#cb2-205" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb2-206"><a href="#cb2-206" aria-hidden="true" tabindex="-1"></a>        <span class="co"># select first common genes across the two cohorts, so the identified</span></span>
<span id="cb2-207"><a href="#cb2-207" aria-hidden="true" tabindex="-1"></a>        <span class="co"># genes are the same</span></span>
<span id="cb2-208"><a href="#cb2-208" aria-hidden="true" tabindex="-1"></a>        df <span class="ot">&lt;-</span> df[common_genes, ]</span>
<span id="cb2-209"><a href="#cb2-209" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb2-210"><a href="#cb2-210" aria-hidden="true" tabindex="-1"></a>        <span class="fu">rowSds</span>(<span class="fu">as.matrix</span>(<span class="fu">assay</span>(df, assay_to_use)))</span>
<span id="cb2-211"><a href="#cb2-211" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb2-212"><a href="#cb2-212" aria-hidden="true" tabindex="-1"></a>    <span class="at">df =</span> datasets[<span class="fu">c</span>(<span class="st">"tcga"</span>, <span class="st">"metabric"</span>)],</span>
<span id="cb2-213"><a href="#cb2-213" aria-hidden="true" tabindex="-1"></a>    <span class="at">assay_to_use =</span> <span class="fu">list</span>(</span>
<span id="cb2-214"><a href="#cb2-214" aria-hidden="true" tabindex="-1"></a>        <span class="at">tcga =</span> <span class="st">"vst"</span>,</span>
<span id="cb2-215"><a href="#cb2-215" aria-hidden="true" tabindex="-1"></a>        <span class="at">metabric =</span> <span class="st">"median_intensity"</span></span>
<span id="cb2-216"><a href="#cb2-216" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb2-217"><a href="#cb2-217" aria-hidden="true" tabindex="-1"></a>    <span class="at">MoreArgs =</span> <span class="fu">list</span>(<span class="at">common_genes =</span> common_genes)</span>
<span id="cb2-218"><a href="#cb2-218" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> data.frame <span class="sc">%&gt;%</span></span>
<span id="cb2-219"><a href="#cb2-219" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb2-220"><a href="#cb2-220" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">average =</span> <span class="fu">sqrt</span>(</span>
<span id="cb2-221"><a href="#cb2-221" aria-hidden="true" tabindex="-1"></a>        ((nb_tcga<span class="dv">-1</span>)<span class="sc">*</span>tcga<span class="sc">^</span><span class="dv">2</span> <span class="sc">+</span> (nb_metabric<span class="dv">-1</span>)<span class="sc">*</span>metabric<span class="sc">^</span><span class="dv">2</span>)<span class="sc">/</span></span>
<span id="cb2-222"><a href="#cb2-222" aria-hidden="true" tabindex="-1"></a>            (nb_tcga <span class="sc">+</span> nb_metabric <span class="sc">-</span> <span class="dv">2</span>) <span class="co"># this is the formula to calculate the</span></span>
<span id="cb2-223"><a href="#cb2-223" aria-hidden="true" tabindex="-1"></a>                                        <span class="co"># average for standard deviations</span></span>
<span id="cb2-224"><a href="#cb2-224" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">%&gt;%</span></span>
<span id="cb2-225"><a href="#cb2-225" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb2-226"><a href="#cb2-226" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">symbol =</span> common_genes) </span>
<span id="cb2-227"><a href="#cb2-227" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-228"><a href="#cb2-228" aria-hidden="true" tabindex="-1"></a>most_variable_genes_sd <span class="ot">&lt;-</span> sd_genes <span class="sc">%&gt;%</span> </span>
<span id="cb2-229"><a href="#cb2-229" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">slice_max</span>(<span class="at">n =</span> <span class="dv">1000</span>, <span class="at">order_by =</span> average) </span>
<span id="cb2-230"><a href="#cb2-230" aria-hidden="true" tabindex="-1"></a>most_variable_genes_cv <span class="ot">&lt;-</span> cv_genes <span class="sc">%&gt;%</span> </span>
<span id="cb2-231"><a href="#cb2-231" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">slice_max</span>(<span class="at">n =</span> <span class="dv">1000</span>, <span class="at">order_by =</span> average) </span>
<span id="cb2-232"><a href="#cb2-232" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-233"><a href="#cb2-233" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">inner_join</span>(cv_genes, sd_genes, <span class="at">by =</span> <span class="st">"symbol"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb2-234"><a href="#cb2-234" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> average.x, <span class="at">y =</span> average.y)) <span class="sc">+</span> </span>
<span id="cb2-235"><a href="#cb2-235" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb2-236"><a href="#cb2-236" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">formula =</span> <span class="st">"y~x"</span>) <span class="sc">+</span> </span>
<span id="cb2-237"><a href="#cb2-237" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb2-238"><a href="#cb2-238" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"Average coefficient of variation (CV)"</span>,</span>
<span id="cb2-239"><a href="#cb2-239" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"Average of standard deviation (SD)"</span>,</span>
<span id="cb2-240"><a href="#cb2-240" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">"Comparison of scores when selecting most variable genes"</span></span>
<span id="cb2-241"><a href="#cb2-241" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb2-242"><a href="#cb2-242" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>()</span>
<span id="cb2-243"><a href="#cb2-243" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-244"><a href="#cb2-244" aria-hidden="true" tabindex="-1"></a>p</span>
<span id="cb2-245"><a href="#cb2-245" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb2-246"><a href="#cb2-246" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-247"><a href="#cb2-247" aria-hidden="true" tabindex="-1"></a>They are highly correlated values, so we stick to the coefficient of </span>
<span id="cb2-248"><a href="#cb2-248" aria-hidden="true" tabindex="-1"></a>variation as it makes more sense in this kind of analysis.</span>
<span id="cb2-249"><a href="#cb2-249" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-252"><a href="#cb2-252" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb2-253"><a href="#cb2-253" aria-hidden="true" tabindex="-1"></a>most_variable_genes <span class="ot">&lt;-</span> most_variable_genes_cv</span>
<span id="cb2-254"><a href="#cb2-254" aria-hidden="true" tabindex="-1"></a>genes_for_pca <span class="ot">&lt;-</span> most_variable_genes_cv<span class="sc">$</span>symbol</span>
<span id="cb2-255"><a href="#cb2-255" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-256"><a href="#cb2-256" aria-hidden="true" tabindex="-1"></a>genes_for_pca <span class="ot">&lt;-</span> <span class="fu">c</span>(genes_for_pca, stable_genes) <span class="sc">%&gt;%</span> unique</span>
<span id="cb2-257"><a href="#cb2-257" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb2-258"><a href="#cb2-258" aria-hidden="true" tabindex="-1"></a>    genes_for_pca,</span>
<span id="cb2-259"><a href="#cb2-259" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/pca_merging/genes_for_pca.rds"</span></span>
<span id="cb2-260"><a href="#cb2-260" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-261"><a href="#cb2-261" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb2-262"><a href="#cb2-262" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-263"><a href="#cb2-263" aria-hidden="true" tabindex="-1"></a><span class="co">[</span><span class="ot">Table @fig-sd-stable-genes</span><span class="co">]</span> shows the top 1000 genes along with their standard</span>
<span id="cb2-264"><a href="#cb2-264" aria-hidden="true" tabindex="-1"></a>deviation. The stable genes are also added to the list.</span>
<span id="cb2-265"><a href="#cb2-265" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-268"><a href="#cb2-268" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb2-269"><a href="#cb2-269" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-sd-stable-genes</span></span>
<span id="cb2-270"><a href="#cb2-270" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: ([Table @fig-sd-stable-genes]) Standard deviation of each gene in</span></span>
<span id="cb2-271"><a href="#cb2-271" aria-hidden="true" tabindex="-1"></a><span class="co">#|     each condition and </span></span>
<span id="cb2-272"><a href="#cb2-272" aria-hidden="true" tabindex="-1"></a><span class="co">#|     pooled standard deviation. </span></span>
<span id="cb2-273"><a href="#cb2-273" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-274"><a href="#cb2-274" aria-hidden="true" tabindex="-1"></a>cv_genes <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(symbol <span class="sc">%in%</span> genes_for_pca) <span class="sc">%&gt;%</span></span>
<span id="cb2-275"><a href="#cb2-275" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb2-276"><a href="#cb2-276" aria-hidden="true" tabindex="-1"></a>        <span class="at">is_housekeeping =</span> <span class="fu">ifelse</span>(symbol <span class="sc">%in%</span> stable_genes, <span class="st">"yes"</span>, <span class="st">"no"</span>)</span>
<span id="cb2-277"><a href="#cb2-277" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb2-278"><a href="#cb2-278" aria-hidden="true" tabindex="-1"></a>    DT<span class="sc">::</span><span class="fu">datatable</span>(</span>
<span id="cb2-279"><a href="#cb2-279" aria-hidden="true" tabindex="-1"></a>        <span class="at">rownames =</span> <span class="cn">FALSE</span>,</span>
<span id="cb2-280"><a href="#cb2-280" aria-hidden="true" tabindex="-1"></a>        <span class="at">options =</span> <span class="fu">list</span>(</span>
<span id="cb2-281"><a href="#cb2-281" aria-hidden="true" tabindex="-1"></a>            <span class="at">columnDefs =</span> <span class="fu">list</span>(</span>
<span id="cb2-282"><a href="#cb2-282" aria-hidden="true" tabindex="-1"></a>                <span class="fu">list</span>(<span class="at">className =</span> <span class="st">'dt-center'</span>, <span class="at">targets =</span> (<span class="dv">0</span><span class="sc">:</span><span class="fu">ncol</span>(sd_genes)))</span>
<span id="cb2-283"><a href="#cb2-283" aria-hidden="true" tabindex="-1"></a>            ),</span>
<span id="cb2-284"><a href="#cb2-284" aria-hidden="true" tabindex="-1"></a>            <span class="at">scrollX =</span> <span class="cn">TRUE</span></span>
<span id="cb2-285"><a href="#cb2-285" aria-hidden="true" tabindex="-1"></a>        ),</span>
<span id="cb2-286"><a href="#cb2-286" aria-hidden="true" tabindex="-1"></a>        <span class="at">filter =</span> <span class="st">'top'</span></span>
<span id="cb2-287"><a href="#cb2-287" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb2-288"><a href="#cb2-288" aria-hidden="true" tabindex="-1"></a>    DT<span class="sc">::</span><span class="fu">formatRound</span>(<span class="fu">c</span>(<span class="st">"tcga"</span>, <span class="st">"metabric"</span>, <span class="st">"average"</span>))</span>
<span id="cb2-289"><a href="#cb2-289" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb2-290"><a href="#cb2-290" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-291"><a href="#cb2-291" aria-hidden="true" tabindex="-1"></a>Overall we can see that housekeeping genes have very low standard deviation</span>
<span id="cb2-292"><a href="#cb2-292" aria-hidden="true" tabindex="-1"></a>across cohorts, which is good.</span>
<span id="cb2-293"><a href="#cb2-293" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-294"><a href="#cb2-294" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb2-295"><a href="#cb2-295" aria-hidden="true" tabindex="-1"></a>datasets_normalized <span class="ot">&lt;-</span> <span class="fu">mapply</span>(</span>
<span id="cb2-296"><a href="#cb2-296" aria-hidden="true" tabindex="-1"></a>    get_final_ranking_values,</span>
<span id="cb2-297"><a href="#cb2-297" aria-hidden="true" tabindex="-1"></a>    <span class="at">sum_exp =</span> datasets,</span>
<span id="cb2-298"><a href="#cb2-298" aria-hidden="true" tabindex="-1"></a>    <span class="at">assay_to_use =</span> which_exp,</span>
<span id="cb2-299"><a href="#cb2-299" aria-hidden="true" tabindex="-1"></a>    <span class="at">MoreArgs =</span> <span class="fu">list</span>(</span>
<span id="cb2-300"><a href="#cb2-300" aria-hidden="true" tabindex="-1"></a>        <span class="at">stable_genes =</span> stable_genes,</span>
<span id="cb2-301"><a href="#cb2-301" aria-hidden="true" tabindex="-1"></a>        <span class="at">most_variable_genes =</span> <span class="fu">setdiff</span>(genes_for_pca, stable_genes)</span>
<span id="cb2-302"><a href="#cb2-302" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb2-303"><a href="#cb2-303" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>,</span>
<span id="cb2-304"><a href="#cb2-304" aria-hidden="true" tabindex="-1"></a>    <span class="at">SIMPLIFY =</span> <span class="cn">FALSE</span></span>
<span id="cb2-305"><a href="#cb2-305" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-306"><a href="#cb2-306" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-307"><a href="#cb2-307" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb2-308"><a href="#cb2-308" aria-hidden="true" tabindex="-1"></a>    datasets_normalized,</span>
<span id="cb2-309"><a href="#cb2-309" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/pca_merging/datasets_normalized.rds"</span></span>
<span id="cb2-310"><a href="#cb2-310" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-311"><a href="#cb2-311" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb2-312"><a href="#cb2-312" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-315"><a href="#cb2-315" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb2-316"><a href="#cb2-316" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span>load_at_setup){</span>
<span id="cb2-317"><a href="#cb2-317" aria-hidden="true" tabindex="-1"></a>    datasets_normalized <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(</span>
<span id="cb2-318"><a href="#cb2-318" aria-hidden="true" tabindex="-1"></a>        <span class="st">"../results/rds_files/pca_merging/datasets_normalized.rds"</span></span>
<span id="cb2-319"><a href="#cb2-319" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb2-320"><a href="#cb2-320" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb2-321"><a href="#cb2-321" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb2-322"><a href="#cb2-322" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-323"><a href="#cb2-323" aria-hidden="true" tabindex="-1"></a>@fig-avg-ranking-hkg shows a histogram with</span>
<span id="cb2-324"><a href="#cb2-324" aria-hidden="true" tabindex="-1"></a>the distribution of the average ranking of the</span>
<span id="cb2-325"><a href="#cb2-325" aria-hidden="true" tabindex="-1"></a>housekeeping genes for each sample in each cohort.</span>
<span id="cb2-326"><a href="#cb2-326" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-327"><a href="#cb2-327" aria-hidden="true" tabindex="-1"></a><span class="in">```{r avg-rankings-hkg, fig.height=4, fig.width=10}</span></span>
<span id="cb2-328"><a href="#cb2-328" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-avg-ranking-hkg</span></span>
<span id="cb2-329"><a href="#cb2-329" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Distribution of the average ranking of the housekeeping genes</span></span>
<span id="cb2-330"><a href="#cb2-330" aria-hidden="true" tabindex="-1"></a><span class="co">#|     defined previously for each cohort separately.</span></span>
<span id="cb2-331"><a href="#cb2-331" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-332"><a href="#cb2-332" aria-hidden="true" tabindex="-1"></a>avg_rankings <span class="ot">&lt;-</span> <span class="fu">lapply</span>(</span>
<span id="cb2-333"><a href="#cb2-333" aria-hidden="true" tabindex="-1"></a>    datasets_normalized, </span>
<span id="cb2-334"><a href="#cb2-334" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(x) <span class="fu">colData</span>(x) <span class="sc">%&gt;%</span> data.frame <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(avg_ranking)</span>
<span id="cb2-335"><a href="#cb2-335" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(<span class="at">.id =</span> <span class="st">"cohort"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb2-336"><a href="#cb2-336" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">cohort =</span> <span class="fu">toupper</span>(cohort))</span>
<span id="cb2-337"><a href="#cb2-337" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-338"><a href="#cb2-338" aria-hidden="true" tabindex="-1"></a>avg_rankings <span class="sc">%&gt;%</span> ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(</span>
<span id="cb2-339"><a href="#cb2-339" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">x =</span> avg_ranking, <span class="at">fill =</span> cohort)</span>
<span id="cb2-340"><a href="#cb2-340" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span> </span>
<span id="cb2-341"><a href="#cb2-341" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_histogram</span>(<span class="at">alpha =</span> <span class="fl">0.9</span>, <span class="at">bins =</span> <span class="dv">30</span>) <span class="sc">+</span></span>
<span id="cb2-342"><a href="#cb2-342" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Average ranking of housekeeping genes"</span>) <span class="sc">+</span> </span>
<span id="cb2-343"><a href="#cb2-343" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>cohort, <span class="at">scales =</span> <span class="st">"free_y"</span>) <span class="sc">+</span></span>
<span id="cb2-344"><a href="#cb2-344" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>()</span>
<span id="cb2-345"><a href="#cb2-345" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb2-346"><a href="#cb2-346" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-347"><a href="#cb2-347" aria-hidden="true" tabindex="-1"></a>The average ranking are somewhat similar. For the two experiments where there</span>
<span id="cb2-348"><a href="#cb2-348" aria-hidden="true" tabindex="-1"></a>are RNA-seq samples, the distribution is the same. For METABRIC, a microarray</span>
<span id="cb2-349"><a href="#cb2-349" aria-hidden="true" tabindex="-1"></a>experiment, the distribution is shifted a bit below.</span>
<span id="cb2-350"><a href="#cb2-350" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-351"><a href="#cb2-351" aria-hidden="true" tabindex="-1"></a>@fig-gene-bc-imp shows histograms of gene expression from some</span>
<span id="cb2-352"><a href="#cb2-352" aria-hidden="true" tabindex="-1"></a>important genes in</span>
<span id="cb2-353"><a href="#cb2-353" aria-hidden="true" tabindex="-1"></a>breast cancer.</span>
<span id="cb2-354"><a href="#cb2-354" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-355"><a href="#cb2-355" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width = 20, fig.height=12}</span></span>
<span id="cb2-356"><a href="#cb2-356" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-gene-bc-imp</span></span>
<span id="cb2-357"><a href="#cb2-357" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Distribution of the gene expression levels of ESR1 and TFF1</span></span>
<span id="cb2-358"><a href="#cb2-358" aria-hidden="true" tabindex="-1"></a><span class="co">#|     in each one of the cohorts</span></span>
<span id="cb2-359"><a href="#cb2-359" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-360"><a href="#cb2-360" aria-hidden="true" tabindex="-1"></a>genes <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"TFF1"</span>, <span class="st">"ESR1"</span>)</span>
<span id="cb2-361"><a href="#cb2-361" aria-hidden="true" tabindex="-1"></a>col_data_genes <span class="ot">&lt;-</span> <span class="fu">mapply</span>(</span>
<span id="cb2-362"><a href="#cb2-362" aria-hidden="true" tabindex="-1"></a>    get_gene_col_data,</span>
<span id="cb2-363"><a href="#cb2-363" aria-hidden="true" tabindex="-1"></a>    <span class="at">sum_exp =</span> datasets_normalized,</span>
<span id="cb2-364"><a href="#cb2-364" aria-hidden="true" tabindex="-1"></a>    <span class="at">assay_to_use =</span> <span class="st">"avg_ranking"</span>,</span>
<span id="cb2-365"><a href="#cb2-365" aria-hidden="true" tabindex="-1"></a>    <span class="at">MoreArgs =</span> <span class="fu">list</span>(<span class="at">genes =</span> genes),</span>
<span id="cb2-366"><a href="#cb2-366" aria-hidden="true" tabindex="-1"></a>    <span class="at">SIMPLIFY =</span> <span class="cn">FALSE</span>,</span>
<span id="cb2-367"><a href="#cb2-367" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span></span>
<span id="cb2-368"><a href="#cb2-368" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> </span>
<span id="cb2-369"><a href="#cb2-369" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(<span class="at">.id =</span> <span class="st">"cohort"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb2-370"><a href="#cb2-370" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">cohort =</span> <span class="fu">toupper</span>(cohort)) <span class="sc">%&gt;%</span></span>
<span id="cb2-371"><a href="#cb2-371" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(</span>
<span id="cb2-372"><a href="#cb2-372" aria-hidden="true" tabindex="-1"></a>        <span class="at">cols =</span> <span class="fu">all_of</span>(genes),</span>
<span id="cb2-373"><a href="#cb2-373" aria-hidden="true" tabindex="-1"></a>        <span class="at">names_to =</span> <span class="st">"gene"</span>,</span>
<span id="cb2-374"><a href="#cb2-374" aria-hidden="true" tabindex="-1"></a>        <span class="at">values_to =</span> <span class="st">"avg_ranking_genes"</span></span>
<span id="cb2-375"><a href="#cb2-375" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb2-376"><a href="#cb2-376" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-377"><a href="#cb2-377" aria-hidden="true" tabindex="-1"></a>ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(</span>
<span id="cb2-378"><a href="#cb2-378" aria-hidden="true" tabindex="-1"></a>    col_data_genes,</span>
<span id="cb2-379"><a href="#cb2-379" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">x =</span> avg_ranking_genes, <span class="at">fill =</span> er_status)</span>
<span id="cb2-380"><a href="#cb2-380" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span> </span>
<span id="cb2-381"><a href="#cb2-381" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_histogram</span>(<span class="at">bins =</span> <span class="dv">30</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="at">position =</span> <span class="st">"identity"</span>) <span class="sc">+</span> </span>
<span id="cb2-382"><a href="#cb2-382" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>cohort <span class="sc">+</span> gene, <span class="at">scales =</span> <span class="st">"free_y"</span>) <span class="sc">+</span></span>
<span id="cb2-383"><a href="#cb2-383" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb2-384"><a href="#cb2-384" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"Normalized average ranking by using housekeeping genes"</span></span>
<span id="cb2-385"><a href="#cb2-385" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb2-386"><a href="#cb2-386" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">30</span>)</span>
<span id="cb2-387"><a href="#cb2-387" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb2-388"><a href="#cb2-388" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-389"><a href="#cb2-389" aria-hidden="true" tabindex="-1"></a>ESR1 and TFF1 expressions are divided across ER status, as expected. The</span>
<span id="cb2-390"><a href="#cb2-390" aria-hidden="true" tabindex="-1"></a>proliferation marker CCND1 is also highly expressed in all conditions.</span>
<span id="cb2-391"><a href="#cb2-391" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-392"><a href="#cb2-392" aria-hidden="true" tabindex="-1"></a><span class="fu">### Checking robustness with respect of number of houesekeeping genes available</span></span>
<span id="cb2-393"><a href="#cb2-393" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-394"><a href="#cb2-394" aria-hidden="true" tabindex="-1"></a>When calculating the normalized data, a total of 44 housekeeping genes were</span>
<span id="cb2-395"><a href="#cb2-395" aria-hidden="true" tabindex="-1"></a>used. Sometimes not all of these genes are available, therefore we will</span>
<span id="cb2-396"><a href="#cb2-396" aria-hidden="true" tabindex="-1"></a>calculate the normalization using random subsets of 44 housekeeping genes with</span>
<span id="cb2-397"><a href="#cb2-397" aria-hidden="true" tabindex="-1"></a>different random sizes. The sets will have a minimum of 20 genes and maximum of</span>
<span id="cb2-398"><a href="#cb2-398" aria-hidden="true" tabindex="-1"></a><span class="ss">44. </span>The sampling of the set size is obtained by using a uniform distribution. We</span>
<span id="cb2-399"><a href="#cb2-399" aria-hidden="true" tabindex="-1"></a>focus just on TCGA for simplicity. The correlation used in this case</span>
<span id="cb2-400"><a href="#cb2-400" aria-hidden="true" tabindex="-1"></a>was the Pearson correlation.</span>
<span id="cb2-401"><a href="#cb2-401" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-402"><a href="#cb2-402" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb2-403"><a href="#cb2-403" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">02943</span>)</span>
<span id="cb2-404"><a href="#cb2-404" aria-hidden="true" tabindex="-1"></a>sizes_sets <span class="ot">&lt;-</span> <span class="fu">floor</span>(<span class="fu">runif</span>(<span class="dv">10</span>, <span class="at">min =</span> <span class="dv">20</span>, <span class="at">max =</span> <span class="dv">44</span>))</span>
<span id="cb2-405"><a href="#cb2-405" aria-hidden="true" tabindex="-1"></a>random_subsets <span class="ot">&lt;-</span> <span class="fu">sapply</span>(</span>
<span id="cb2-406"><a href="#cb2-406" aria-hidden="true" tabindex="-1"></a>    sizes_sets,</span>
<span id="cb2-407"><a href="#cb2-407" aria-hidden="true" tabindex="-1"></a>    sample,</span>
<span id="cb2-408"><a href="#cb2-408" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> stable_genes,</span>
<span id="cb2-409"><a href="#cb2-409" aria-hidden="true" tabindex="-1"></a>    <span class="at">replace =</span> <span class="cn">FALSE</span>,</span>
<span id="cb2-410"><a href="#cb2-410" aria-hidden="true" tabindex="-1"></a>    <span class="at">simplify =</span> <span class="cn">FALSE</span></span>
<span id="cb2-411"><a href="#cb2-411" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-412"><a href="#cb2-412" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-413"><a href="#cb2-413" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(random_subsets, <span class="st">"../results/rds_files/pca_merging/random_subsets.rds"</span>)</span>
<span id="cb2-414"><a href="#cb2-414" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-415"><a href="#cb2-415" aria-hidden="true" tabindex="-1"></a>random_normalization <span class="ot">&lt;-</span> parallel<span class="sc">::</span><span class="fu">mclapply</span>(</span>
<span id="cb2-416"><a href="#cb2-416" aria-hidden="true" tabindex="-1"></a>    random_subsets,</span>
<span id="cb2-417"><a href="#cb2-417" aria-hidden="true" tabindex="-1"></a>    get_final_ranking_values,</span>
<span id="cb2-418"><a href="#cb2-418" aria-hidden="true" tabindex="-1"></a>    <span class="at">sum_exp =</span> datasets<span class="sc">$</span>tcga,</span>
<span id="cb2-419"><a href="#cb2-419" aria-hidden="true" tabindex="-1"></a>    <span class="at">assay_to_use =</span> which_exp<span class="sc">$</span>tcga,</span>
<span id="cb2-420"><a href="#cb2-420" aria-hidden="true" tabindex="-1"></a>    <span class="at">most_variable_genes =</span> <span class="fu">setdiff</span>(genes_for_pca, stable_genes),</span>
<span id="cb2-421"><a href="#cb2-421" aria-hidden="true" tabindex="-1"></a>    <span class="at">mc.cores =</span> <span class="dv">10</span></span>
<span id="cb2-422"><a href="#cb2-422" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-423"><a href="#cb2-423" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-424"><a href="#cb2-424" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb2-425"><a href="#cb2-425" aria-hidden="true" tabindex="-1"></a>    random_normalization, </span>
<span id="cb2-426"><a href="#cb2-426" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/pca_merging/random_normalization.rds"</span></span>
<span id="cb2-427"><a href="#cb2-427" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-428"><a href="#cb2-428" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-429"><a href="#cb2-429" aria-hidden="true" tabindex="-1"></a><span class="co"># fetch the average ranking for each calculation. we will plot the scores</span></span>
<span id="cb2-430"><a href="#cb2-430" aria-hidden="true" tabindex="-1"></a><span class="co"># for some genes and calculate correlation for all genes as well. </span></span>
<span id="cb2-431"><a href="#cb2-431" aria-hidden="true" tabindex="-1"></a><span class="co"># correlation is calculate by using the original list with 44 genes</span></span>
<span id="cb2-432"><a href="#cb2-432" aria-hidden="true" tabindex="-1"></a>avg_ranking_random <span class="ot">&lt;-</span> <span class="fu">lapply</span>(</span>
<span id="cb2-433"><a href="#cb2-433" aria-hidden="true" tabindex="-1"></a>    random_normalization,</span>
<span id="cb2-434"><a href="#cb2-434" aria-hidden="true" tabindex="-1"></a>    assay,</span>
<span id="cb2-435"><a href="#cb2-435" aria-hidden="true" tabindex="-1"></a>    <span class="at">i =</span> <span class="st">"avg_ranking"</span></span>
<span id="cb2-436"><a href="#cb2-436" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-437"><a href="#cb2-437" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-438"><a href="#cb2-438" aria-hidden="true" tabindex="-1"></a>correlation_scores <span class="ot">&lt;-</span> <span class="fu">lapply</span>(</span>
<span id="cb2-439"><a href="#cb2-439" aria-hidden="true" tabindex="-1"></a>    avg_ranking_random,</span>
<span id="cb2-440"><a href="#cb2-440" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(df, og_sum_exp, most_variable_genes){</span>
<span id="cb2-441"><a href="#cb2-441" aria-hidden="true" tabindex="-1"></a>        <span class="fu">sapply</span>(</span>
<span id="cb2-442"><a href="#cb2-442" aria-hidden="true" tabindex="-1"></a>            most_variable_genes,</span>
<span id="cb2-443"><a href="#cb2-443" aria-hidden="true" tabindex="-1"></a>            <span class="cf">function</span>(gene, df, og_values){</span>
<span id="cb2-444"><a href="#cb2-444" aria-hidden="true" tabindex="-1"></a>                <span class="fu">cor</span>(df[gene, ], og_values[gene, ])</span>
<span id="cb2-445"><a href="#cb2-445" aria-hidden="true" tabindex="-1"></a>            },</span>
<span id="cb2-446"><a href="#cb2-446" aria-hidden="true" tabindex="-1"></a>            <span class="at">df =</span> df,</span>
<span id="cb2-447"><a href="#cb2-447" aria-hidden="true" tabindex="-1"></a>            <span class="at">og_values =</span> <span class="fu">assay</span>(og_sum_exp, <span class="st">"avg_ranking"</span>)</span>
<span id="cb2-448"><a href="#cb2-448" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb2-449"><a href="#cb2-449" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb2-450"><a href="#cb2-450" aria-hidden="true" tabindex="-1"></a>    <span class="at">og_sum_exp =</span> datasets_normalized<span class="sc">$</span>tcga,</span>
<span id="cb2-451"><a href="#cb2-451" aria-hidden="true" tabindex="-1"></a>    <span class="at">most_variable_genes =</span> <span class="fu">setdiff</span>(genes_for_pca, stable_genes)</span>
<span id="cb2-452"><a href="#cb2-452" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> <span class="st">`</span><span class="at">names&lt;-</span><span class="st">`</span>(</span>
<span id="cb2-453"><a href="#cb2-453" aria-hidden="true" tabindex="-1"></a>    <span class="fu">paste0</span>(<span class="st">"nb_genes_"</span>, <span class="fu">sapply</span>(random_subsets, length))</span>
<span id="cb2-454"><a href="#cb2-454" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-455"><a href="#cb2-455" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-456"><a href="#cb2-456" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb2-457"><a href="#cb2-457" aria-hidden="true" tabindex="-1"></a>    correlation_scores,</span>
<span id="cb2-458"><a href="#cb2-458" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/pca_merging/correlation_scores.rds"</span></span>
<span id="cb2-459"><a href="#cb2-459" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-460"><a href="#cb2-460" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb2-461"><a href="#cb2-461" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-464"><a href="#cb2-464" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb2-465"><a href="#cb2-465" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span>load_at_setup){</span>
<span id="cb2-466"><a href="#cb2-466" aria-hidden="true" tabindex="-1"></a>    correlation_scores <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(</span>
<span id="cb2-467"><a href="#cb2-467" aria-hidden="true" tabindex="-1"></a>        <span class="st">"../results/rds_files/pca_merging/correlation_scores.rds"</span></span>
<span id="cb2-468"><a href="#cb2-468" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb2-469"><a href="#cb2-469" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb2-470"><a href="#cb2-470" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb2-471"><a href="#cb2-471" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-472"><a href="#cb2-472" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=21, fig.height=12}</span></span>
<span id="cb2-473"><a href="#cb2-473" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-correlation</span></span>
<span id="cb2-474"><a href="#cb2-474" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Correlation distribution for different subsets of housekeeping</span></span>
<span id="cb2-475"><a href="#cb2-475" aria-hidden="true" tabindex="-1"></a><span class="co">#|     genes (HKGs). Each different subset has a different number of HKGs.</span></span>
<span id="cb2-476"><a href="#cb2-476" aria-hidden="true" tabindex="-1"></a>correlation_df <span class="ot">&lt;-</span> <span class="fu">lapply</span>(</span>
<span id="cb2-477"><a href="#cb2-477" aria-hidden="true" tabindex="-1"></a>    correlation_scores,</span>
<span id="cb2-478"><a href="#cb2-478" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(x) <span class="fu">data.frame</span>(<span class="at">correlation =</span> x, <span class="at">gene =</span> <span class="fu">names</span>(x), <span class="at">row.names =</span> <span class="cn">NULL</span>)</span>
<span id="cb2-479"><a href="#cb2-479" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(<span class="at">.id =</span> <span class="st">"nb_genes"</span>)</span>
<span id="cb2-480"><a href="#cb2-480" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-481"><a href="#cb2-481" aria-hidden="true" tabindex="-1"></a>correlation_df <span class="sc">%&gt;%</span> ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(</span>
<span id="cb2-482"><a href="#cb2-482" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">x =</span> correlation) </span>
<span id="cb2-483"><a href="#cb2-483" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span> </span>
<span id="cb2-484"><a href="#cb2-484" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_histogram</span>(<span class="at">bins =</span> <span class="dv">30</span>) <span class="sc">+</span></span>
<span id="cb2-485"><a href="#cb2-485" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb2-486"><a href="#cb2-486" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"Correlation"</span>, </span>
<span id="cb2-487"><a href="#cb2-487" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="fu">paste0</span>(</span>
<span id="cb2-488"><a href="#cb2-488" aria-hidden="true" tabindex="-1"></a>            <span class="st">"Correlation distribution for different"</span>,</span>
<span id="cb2-489"><a href="#cb2-489" aria-hidden="true" tabindex="-1"></a>            <span class="st">" subsets of the housekeeping genes"</span></span>
<span id="cb2-490"><a href="#cb2-490" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb2-491"><a href="#cb2-491" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb2-492"><a href="#cb2-492" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>nb_genes) <span class="sc">+</span> </span>
<span id="cb2-493"><a href="#cb2-493" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">30</span>) <span class="sc">+</span></span>
<span id="cb2-494"><a href="#cb2-494" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme</span>(<span class="at">panel.spacing =</span> <span class="fu">unit</span>(<span class="dv">3</span>, <span class="st">"lines"</span>))</span>
<span id="cb2-495"><a href="#cb2-495" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb2-496"><a href="#cb2-496" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-497"><a href="#cb2-497" aria-hidden="true" tabindex="-1"></a>@fig-correlation shows the different distributions given the subsets of</span>
<span id="cb2-498"><a href="#cb2-498" aria-hidden="true" tabindex="-1"></a>housekeeping genes. All the correlation coefficients are close to 1. This means</span>
<span id="cb2-499"><a href="#cb2-499" aria-hidden="true" tabindex="-1"></a>that if a subset of housekeeping genes are missing it is fine, as the others</span>
<span id="cb2-500"><a href="#cb2-500" aria-hidden="true" tabindex="-1"></a>will compensate.</span>
<span id="cb2-501"><a href="#cb2-501" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-502"><a href="#cb2-502" aria-hidden="true" tabindex="-1"></a><span class="fu">## Integrating TCGA and METABRIC using PCA</span></span>
<span id="cb2-503"><a href="#cb2-503" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-504"><a href="#cb2-504" aria-hidden="true" tabindex="-1"></a>The next step is creating the framework where we can integrate any patient</span>
<span id="cb2-505"><a href="#cb2-505" aria-hidden="true" tabindex="-1"></a>sample and compare each other, at least spatially. The approach is similar to</span>
<span id="cb2-506"><a href="#cb2-506" aria-hidden="true" tabindex="-1"></a>some sort of unsupervised learning, where no labels or groups are specified,</span>
<span id="cb2-507"><a href="#cb2-507" aria-hidden="true" tabindex="-1"></a>which is something common in algorithms for batch removal. By using PCA in some</span>
<span id="cb2-508"><a href="#cb2-508" aria-hidden="true" tabindex="-1"></a>subset of TCGA and METABRIC, we should learn the idiosyncrasies of the RNA-seq</span>
<span id="cb2-509"><a href="#cb2-509" aria-hidden="true" tabindex="-1"></a>and microarray technologies as well as batch effects general to datasets. Since</span>
<span id="cb2-510"><a href="#cb2-510" aria-hidden="true" tabindex="-1"></a>these two cohorts are big ones, we assume that if there are common batch</span>
<span id="cb2-511"><a href="#cb2-511" aria-hidden="true" tabindex="-1"></a>effects, they are represented on them.</span>
<span id="cb2-512"><a href="#cb2-512" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-513"><a href="#cb2-513" aria-hidden="true" tabindex="-1"></a>The practical steps are: first subselect 1000 samples in total from METABRIC</span>
<span id="cb2-514"><a href="#cb2-514" aria-hidden="true" tabindex="-1"></a>**and** TCGA. Train PCA using these 1000 samples. Validate the results by</span>
<span id="cb2-515"><a href="#cb2-515" aria-hidden="true" tabindex="-1"></a>projecting the other samples. In the next section we show how robust the</span>
<span id="cb2-516"><a href="#cb2-516" aria-hidden="true" tabindex="-1"></a>approach is when selecting the samples. In the next chapters we validate the</span>
<span id="cb2-517"><a href="#cb2-517" aria-hidden="true" tabindex="-1"></a>projection in unseen cohorts.</span>
<span id="cb2-518"><a href="#cb2-518" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-519"><a href="#cb2-519" aria-hidden="true" tabindex="-1"></a>Integrating as laid out here only uses molecular data, therefore patients might</span>
<span id="cb2-520"><a href="#cb2-520" aria-hidden="true" tabindex="-1"></a>be close to each other but they might have very different clinical features.</span>
<span id="cb2-521"><a href="#cb2-521" aria-hidden="true" tabindex="-1"></a>This should always be kept in mind.</span>
<span id="cb2-522"><a href="#cb2-522" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-525"><a href="#cb2-525" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb2-526"><a href="#cb2-526" aria-hidden="true" tabindex="-1"></a><span class="co"># we start by selecting the patients that will be used in the training </span></span>
<span id="cb2-527"><a href="#cb2-527" aria-hidden="true" tabindex="-1"></a><span class="co"># set. </span></span>
<span id="cb2-528"><a href="#cb2-528" aria-hidden="true" tabindex="-1"></a>merged_col_data <span class="ot">&lt;-</span> <span class="fu">lapply</span>(datasets_normalized, colData) <span class="sc">%&gt;%</span> </span>
<span id="cb2-529"><a href="#cb2-529" aria-hidden="true" tabindex="-1"></a>    <span class="fu">lapply</span>(., data.frame) <span class="sc">%&gt;%</span></span>
<span id="cb2-530"><a href="#cb2-530" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(<span class="at">.id =</span> <span class="st">"cohort"</span>)</span>
<span id="cb2-531"><a href="#cb2-531" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-532"><a href="#cb2-532" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb2-533"><a href="#cb2-533" aria-hidden="true" tabindex="-1"></a>    merged_col_data,</span>
<span id="cb2-534"><a href="#cb2-534" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/pca_merging/merged_col_data.rds"</span></span>
<span id="cb2-535"><a href="#cb2-535" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-536"><a href="#cb2-536" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-537"><a href="#cb2-537" aria-hidden="true" tabindex="-1"></a>which_cohorts_training <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"tcga"</span>, <span class="st">"metabric"</span>)</span>
<span id="cb2-538"><a href="#cb2-538" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-539"><a href="#cb2-539" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb2-540"><a href="#cb2-540" aria-hidden="true" tabindex="-1"></a>    which_cohorts_training,</span>
<span id="cb2-541"><a href="#cb2-541" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/pca_merging/which_cohorts_training.rds"</span></span>
<span id="cb2-542"><a href="#cb2-542" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-543"><a href="#cb2-543" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-544"><a href="#cb2-544" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">132904</span>)</span>
<span id="cb2-545"><a href="#cb2-545" aria-hidden="true" tabindex="-1"></a>samples_for_training <span class="ot">&lt;-</span> merged_col_data <span class="sc">%&gt;%</span></span>
<span id="cb2-546"><a href="#cb2-546" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">%in%</span> which_cohorts_training) <span class="sc">%&gt;%</span></span>
<span id="cb2-547"><a href="#cb2-547" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">pull</span>(sample_name) <span class="sc">%&gt;%</span></span>
<span id="cb2-548"><a href="#cb2-548" aria-hidden="true" tabindex="-1"></a>    <span class="fu">sample</span>(., <span class="at">size =</span> <span class="dv">1000</span>)</span>
<span id="cb2-549"><a href="#cb2-549" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb2-550"><a href="#cb2-550" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-551"><a href="#cb2-551" aria-hidden="true" tabindex="-1"></a>The table below shows the molecular characteristics of the patients selected for</span>
<span id="cb2-552"><a href="#cb2-552" aria-hidden="true" tabindex="-1"></a>the PCA training.</span>
<span id="cb2-553"><a href="#cb2-553" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-556"><a href="#cb2-556" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb2-557"><a href="#cb2-557" aria-hidden="true" tabindex="-1"></a>merged_col_data <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb2-558"><a href="#cb2-558" aria-hidden="true" tabindex="-1"></a>    sample_name <span class="sc">%in%</span> samples_for_training</span>
<span id="cb2-559"><a href="#cb2-559" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> janitor<span class="sc">::</span><span class="fu">tabyl</span>(cohort, pam50) <span class="sc">%&gt;%</span> </span>
<span id="cb2-560"><a href="#cb2-560" aria-hidden="true" tabindex="-1"></a>    kableExtra<span class="sc">::</span><span class="fu">kbl</span>() <span class="sc">%&gt;%</span></span>
<span id="cb2-561"><a href="#cb2-561" aria-hidden="true" tabindex="-1"></a>    kableExtra<span class="sc">::</span><span class="fu">kable_classic</span>(<span class="at">full_width =</span> <span class="cn">FALSE</span>)</span>
<span id="cb2-562"><a href="#cb2-562" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb2-563"><a href="#cb2-563" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-564"><a href="#cb2-564" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb2-565"><a href="#cb2-565" aria-hidden="true" tabindex="-1"></a>training_set <span class="ot">&lt;-</span> <span class="fu">lapply</span>(</span>
<span id="cb2-566"><a href="#cb2-566" aria-hidden="true" tabindex="-1"></a>    datasets_normalized[which_cohorts_training], </span>
<span id="cb2-567"><a href="#cb2-567" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(sum_exp, i, genes_for_pca) </span>
<span id="cb2-568"><a href="#cb2-568" aria-hidden="true" tabindex="-1"></a>        <span class="fu">assay</span>(sum_exp[genes_for_pca, ], <span class="at">i =</span> i) <span class="sc">%&gt;%</span> </span>
<span id="cb2-569"><a href="#cb2-569" aria-hidden="true" tabindex="-1"></a>        <span class="fu">data.frame</span>(<span class="at">check.names =</span> <span class="cn">FALSE</span>), </span>
<span id="cb2-570"><a href="#cb2-570" aria-hidden="true" tabindex="-1"></a>    <span class="at">i =</span> <span class="st">"avg_ranking"</span>,</span>
<span id="cb2-571"><a href="#cb2-571" aria-hidden="true" tabindex="-1"></a>    <span class="at">genes_for_pca =</span> genes_for_pca</span>
<span id="cb2-572"><a href="#cb2-572" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> </span>
<span id="cb2-573"><a href="#cb2-573" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">bind_cols</span>() <span class="sc">%&gt;%</span></span>
<span id="cb2-574"><a href="#cb2-574" aria-hidden="true" tabindex="-1"></a>    .[, samples_for_training]</span>
<span id="cb2-575"><a href="#cb2-575" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-576"><a href="#cb2-576" aria-hidden="true" tabindex="-1"></a>pca_fit <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">pca</span>(</span>
<span id="cb2-577"><a href="#cb2-577" aria-hidden="true" tabindex="-1"></a>    training_set,</span>
<span id="cb2-578"><a href="#cb2-578" aria-hidden="true" tabindex="-1"></a>    <span class="at">metadata =</span> dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(</span>
<span id="cb2-579"><a href="#cb2-579" aria-hidden="true" tabindex="-1"></a>        <span class="fu">lapply</span>(</span>
<span id="cb2-580"><a href="#cb2-580" aria-hidden="true" tabindex="-1"></a>            datasets_normalized[which_cohorts_training],</span>
<span id="cb2-581"><a href="#cb2-581" aria-hidden="true" tabindex="-1"></a>            <span class="cf">function</span>(df){</span>
<span id="cb2-582"><a href="#cb2-582" aria-hidden="true" tabindex="-1"></a>                <span class="fu">colData</span>(df) <span class="sc">%&gt;%</span> data.frame <span class="sc">%&gt;%</span></span>
<span id="cb2-583"><a href="#cb2-583" aria-hidden="true" tabindex="-1"></a>                    dplyr<span class="sc">::</span><span class="fu">filter</span>(sample_name <span class="sc">%in%</span> <span class="fu">colnames</span>(training_set))</span>
<span id="cb2-584"><a href="#cb2-584" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb2-585"><a href="#cb2-585" aria-hidden="true" tabindex="-1"></a>        ),</span>
<span id="cb2-586"><a href="#cb2-586" aria-hidden="true" tabindex="-1"></a>        <span class="at">.id =</span> <span class="st">"cohort"</span></span>
<span id="cb2-587"><a href="#cb2-587" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span> .[<span class="fu">colnames</span>(training_set), ],</span>
<span id="cb2-588"><a href="#cb2-588" aria-hidden="true" tabindex="-1"></a>    <span class="at">center =</span> <span class="cn">FALSE</span>,</span>
<span id="cb2-589"><a href="#cb2-589" aria-hidden="true" tabindex="-1"></a>    <span class="at">scale =</span> <span class="cn">FALSE</span></span>
<span id="cb2-590"><a href="#cb2-590" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-591"><a href="#cb2-591" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-592"><a href="#cb2-592" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb2-593"><a href="#cb2-593" aria-hidden="true" tabindex="-1"></a>    pca_fit,</span>
<span id="cb2-594"><a href="#cb2-594" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/pca_merging/pca_fit.rds"</span></span>
<span id="cb2-595"><a href="#cb2-595" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-596"><a href="#cb2-596" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb2-597"><a href="#cb2-597" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-600"><a href="#cb2-600" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb2-601"><a href="#cb2-601" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span>load_at_setup){</span>
<span id="cb2-602"><a href="#cb2-602" aria-hidden="true" tabindex="-1"></a>    pca_fit <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(</span>
<span id="cb2-603"><a href="#cb2-603" aria-hidden="true" tabindex="-1"></a>        <span class="st">"../results/rds_files/pca_merging/pca_fit.rds"</span></span>
<span id="cb2-604"><a href="#cb2-604" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb2-605"><a href="#cb2-605" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb2-606"><a href="#cb2-606" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb2-607"><a href="#cb2-607" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-608"><a href="#cb2-608" aria-hidden="true" tabindex="-1"></a>@fig-pca-embeddings shows the embedding of the two datasets together of the</span>
<span id="cb2-609"><a href="#cb2-609" aria-hidden="true" tabindex="-1"></a>1000 samples used for training. We select several PCs of interest to show how</span>
<span id="cb2-610"><a href="#cb2-610" aria-hidden="true" tabindex="-1"></a>the embedding works.</span>
<span id="cb2-611"><a href="#cb2-611" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-612"><a href="#cb2-612" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=20, fig.height=16}</span></span>
<span id="cb2-613"><a href="#cb2-613" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-pca-embeddings</span></span>
<span id="cb2-614"><a href="#cb2-614" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: PCA projections colored by different factors and organized</span></span>
<span id="cb2-615"><a href="#cb2-615" aria-hidden="true" tabindex="-1"></a><span class="co">#|    by different components. (A) Plot of the first two components colored</span></span>
<span id="cb2-616"><a href="#cb2-616" aria-hidden="true" tabindex="-1"></a><span class="co">#|    by cohort. (B) Plot of the first two components colored by ER status.</span></span>
<span id="cb2-617"><a href="#cb2-617" aria-hidden="true" tabindex="-1"></a><span class="co">#|    (C) Plot of the second and third components colored by cohort. </span></span>
<span id="cb2-618"><a href="#cb2-618" aria-hidden="true" tabindex="-1"></a><span class="co">#|    (D) Plot of the second and third components colored by ER status.</span></span>
<span id="cb2-619"><a href="#cb2-619" aria-hidden="true" tabindex="-1"></a><span class="co">#|    (E) Plot of the second and third components colored by PAM50.</span></span>
<span id="cb2-620"><a href="#cb2-620" aria-hidden="true" tabindex="-1"></a><span class="co">#|    (F) Plot of the second and third components colored by INTCLUST, only</span></span>
<span id="cb2-621"><a href="#cb2-621" aria-hidden="true" tabindex="-1"></a><span class="co">#|    METABRIC has an assigned value for this variable, NAs are TCGA samples.</span></span>
<span id="cb2-622"><a href="#cb2-622" aria-hidden="true" tabindex="-1"></a>plots_pca_fit <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb2-623"><a href="#cb2-623" aria-hidden="true" tabindex="-1"></a>point_size <span class="ot">&lt;-</span> <span class="dv">2</span></span>
<span id="cb2-624"><a href="#cb2-624" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-625"><a href="#cb2-625" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>pc1_pc2_cohort <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb2-626"><a href="#cb2-626" aria-hidden="true" tabindex="-1"></a>    pca_fit,</span>
<span id="cb2-627"><a href="#cb2-627" aria-hidden="true" tabindex="-1"></a>    <span class="at">colby =</span> <span class="st">"cohort"</span>,</span>
<span id="cb2-628"><a href="#cb2-628" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> <span class="cn">NULL</span>, </span>
<span id="cb2-629"><a href="#cb2-629" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb2-630"><a href="#cb2-630" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC1"</span>,</span>
<span id="cb2-631"><a href="#cb2-631" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC2"</span>,</span>
<span id="cb2-632"><a href="#cb2-632" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"First two components colored by cohort"</span>,</span>
<span id="cb2-633"><a href="#cb2-633" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">"Only 1000 training samples"</span>,</span>
<span id="cb2-634"><a href="#cb2-634" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> point_size</span>
<span id="cb2-635"><a href="#cb2-635" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-636"><a href="#cb2-636" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-637"><a href="#cb2-637" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>pc1_pc2_er_status <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb2-638"><a href="#cb2-638" aria-hidden="true" tabindex="-1"></a>    pca_fit,</span>
<span id="cb2-639"><a href="#cb2-639" aria-hidden="true" tabindex="-1"></a>    <span class="at">colby =</span> <span class="st">"er_status"</span>,</span>
<span id="cb2-640"><a href="#cb2-640" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> <span class="cn">NULL</span>, </span>
<span id="cb2-641"><a href="#cb2-641" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb2-642"><a href="#cb2-642" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC1"</span>,</span>
<span id="cb2-643"><a href="#cb2-643" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC2"</span>,</span>
<span id="cb2-644"><a href="#cb2-644" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"First two components colored by ER status"</span>,</span>
<span id="cb2-645"><a href="#cb2-645" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">"Only 1000 training samples"</span>,</span>
<span id="cb2-646"><a href="#cb2-646" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> point_size</span>
<span id="cb2-647"><a href="#cb2-647" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-648"><a href="#cb2-648" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-649"><a href="#cb2-649" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>pc2_pc3_cohort <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb2-650"><a href="#cb2-650" aria-hidden="true" tabindex="-1"></a>    pca_fit,</span>
<span id="cb2-651"><a href="#cb2-651" aria-hidden="true" tabindex="-1"></a>    <span class="at">colby =</span> <span class="st">"cohort"</span>,</span>
<span id="cb2-652"><a href="#cb2-652" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> <span class="cn">NULL</span>, </span>
<span id="cb2-653"><a href="#cb2-653" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb2-654"><a href="#cb2-654" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC2"</span>,</span>
<span id="cb2-655"><a href="#cb2-655" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC3"</span>,</span>
<span id="cb2-656"><a href="#cb2-656" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Second and third components colored by cohort"</span>,</span>
<span id="cb2-657"><a href="#cb2-657" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">"Only 1000 training samples"</span>,</span>
<span id="cb2-658"><a href="#cb2-658" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> point_size</span>
<span id="cb2-659"><a href="#cb2-659" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-660"><a href="#cb2-660" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-661"><a href="#cb2-661" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>pc2_pc3_er_status <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb2-662"><a href="#cb2-662" aria-hidden="true" tabindex="-1"></a>    pca_fit,</span>
<span id="cb2-663"><a href="#cb2-663" aria-hidden="true" tabindex="-1"></a>    <span class="at">colby =</span> <span class="st">"er_status"</span>,</span>
<span id="cb2-664"><a href="#cb2-664" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> <span class="cn">NULL</span>, </span>
<span id="cb2-665"><a href="#cb2-665" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb2-666"><a href="#cb2-666" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC2"</span>,</span>
<span id="cb2-667"><a href="#cb2-667" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC3"</span>,</span>
<span id="cb2-668"><a href="#cb2-668" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Second and third components colored by ER status"</span>,</span>
<span id="cb2-669"><a href="#cb2-669" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">"Only 1000 training samples"</span>,</span>
<span id="cb2-670"><a href="#cb2-670" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> point_size</span>
<span id="cb2-671"><a href="#cb2-671" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-672"><a href="#cb2-672" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-673"><a href="#cb2-673" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>pc2_pc3_pam50 <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb2-674"><a href="#cb2-674" aria-hidden="true" tabindex="-1"></a>    pca_fit,</span>
<span id="cb2-675"><a href="#cb2-675" aria-hidden="true" tabindex="-1"></a>    <span class="at">colby =</span> <span class="st">"pam50"</span>,</span>
<span id="cb2-676"><a href="#cb2-676" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> <span class="cn">NULL</span>, </span>
<span id="cb2-677"><a href="#cb2-677" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb2-678"><a href="#cb2-678" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC2"</span>,</span>
<span id="cb2-679"><a href="#cb2-679" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC3"</span>,</span>
<span id="cb2-680"><a href="#cb2-680" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Second and third components colored by PAM50"</span>,</span>
<span id="cb2-681"><a href="#cb2-681" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">"Only 1000 training samples"</span>,</span>
<span id="cb2-682"><a href="#cb2-682" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> point_size</span>
<span id="cb2-683"><a href="#cb2-683" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-684"><a href="#cb2-684" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-685"><a href="#cb2-685" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>pc2_pc3_intclust <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb2-686"><a href="#cb2-686" aria-hidden="true" tabindex="-1"></a>    pca_fit,</span>
<span id="cb2-687"><a href="#cb2-687" aria-hidden="true" tabindex="-1"></a>    <span class="at">colby =</span> <span class="st">"INTCLUST"</span>,</span>
<span id="cb2-688"><a href="#cb2-688" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> <span class="cn">NULL</span>, </span>
<span id="cb2-689"><a href="#cb2-689" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb2-690"><a href="#cb2-690" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC2"</span>,</span>
<span id="cb2-691"><a href="#cb2-691" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC3"</span>,</span>
<span id="cb2-692"><a href="#cb2-692" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Second and third components colored by INTCLUST"</span>,</span>
<span id="cb2-693"><a href="#cb2-693" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">"Only 1000 training samples"</span>,</span>
<span id="cb2-694"><a href="#cb2-694" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> point_size</span>
<span id="cb2-695"><a href="#cb2-695" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-696"><a href="#cb2-696" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-697"><a href="#cb2-697" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>pc3_pc4_pam50 <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb2-698"><a href="#cb2-698" aria-hidden="true" tabindex="-1"></a>    pca_fit,</span>
<span id="cb2-699"><a href="#cb2-699" aria-hidden="true" tabindex="-1"></a>    <span class="at">colby =</span> <span class="st">"pam50"</span>,</span>
<span id="cb2-700"><a href="#cb2-700" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> <span class="cn">NULL</span>, </span>
<span id="cb2-701"><a href="#cb2-701" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb2-702"><a href="#cb2-702" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC3"</span>,</span>
<span id="cb2-703"><a href="#cb2-703" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC4"</span>,</span>
<span id="cb2-704"><a href="#cb2-704" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Third and fourth components colored by PAM50"</span>,</span>
<span id="cb2-705"><a href="#cb2-705" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">"Only 1000 training samples"</span>,</span>
<span id="cb2-706"><a href="#cb2-706" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> point_size</span>
<span id="cb2-707"><a href="#cb2-707" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-708"><a href="#cb2-708" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-709"><a href="#cb2-709" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>pc3_pc4_cohort <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb2-710"><a href="#cb2-710" aria-hidden="true" tabindex="-1"></a>    pca_fit,</span>
<span id="cb2-711"><a href="#cb2-711" aria-hidden="true" tabindex="-1"></a>    <span class="at">colby =</span> <span class="st">"cohort"</span>,</span>
<span id="cb2-712"><a href="#cb2-712" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> <span class="cn">NULL</span>, </span>
<span id="cb2-713"><a href="#cb2-713" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb2-714"><a href="#cb2-714" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC3"</span>,</span>
<span id="cb2-715"><a href="#cb2-715" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC4"</span>,</span>
<span id="cb2-716"><a href="#cb2-716" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Third and fourth components colored by cohort"</span>,</span>
<span id="cb2-717"><a href="#cb2-717" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">"Only 1000 training samples"</span>,</span>
<span id="cb2-718"><a href="#cb2-718" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> point_size</span>
<span id="cb2-719"><a href="#cb2-719" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-720"><a href="#cb2-720" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-721"><a href="#cb2-721" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(<span class="at">plotlist =</span> plots_pca_fit, <span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">labels =</span> <span class="st">"AUTO"</span>)</span>
<span id="cb2-722"><a href="#cb2-722" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb2-723"><a href="#cb2-723" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-724"><a href="#cb2-724" aria-hidden="true" tabindex="-1"></a>The first three components correspond to the batch effect. The third component</span>
<span id="cb2-725"><a href="#cb2-725" aria-hidden="true" tabindex="-1"></a>explains the ER status and the two cohorts are well inter mingled. When plotting</span>
<span id="cb2-726"><a href="#cb2-726" aria-hidden="true" tabindex="-1"></a>the third and fourth components and coloring by the molecular subtype, we see a</span>
<span id="cb2-727"><a href="#cb2-727" aria-hidden="true" tabindex="-1"></a>clear distinction, recapitulating the biology.</span>
<span id="cb2-728"><a href="#cb2-728" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-729"><a href="#cb2-729" aria-hidden="true" tabindex="-1"></a>The plot below shows how the loadings of the first component are really</span>
<span id="cb2-730"><a href="#cb2-730" aria-hidden="true" tabindex="-1"></a>concentrated around zero, so there are not actually specific genes that</span>
<span id="cb2-731"><a href="#cb2-731" aria-hidden="true" tabindex="-1"></a>are accounting for the batch effects we see in the first component. We show </span>
<span id="cb2-732"><a href="#cb2-732" aria-hidden="true" tabindex="-1"></a>latter that the genes that are the most important are the ones with high loading,</span>
<span id="cb2-733"><a href="#cb2-733" aria-hidden="true" tabindex="-1"></a>so the tail of the distributions. In this sense there are really no genes </span>
<span id="cb2-734"><a href="#cb2-734" aria-hidden="true" tabindex="-1"></a>in the tail of the loadings for PC1. </span>
<span id="cb2-735"><a href="#cb2-735" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-738"><a href="#cb2-738" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb2-739"><a href="#cb2-739" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-loadings-pcs</span></span>
<span id="cb2-740"><a href="#cb2-740" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Different loadings for the first 4 components. </span></span>
<span id="cb2-741"><a href="#cb2-741" aria-hidden="true" tabindex="-1"></a>pca_fit<span class="sc">$</span>loadings <span class="sc">%&gt;%</span></span>
<span id="cb2-742"><a href="#cb2-742" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span></span>
<span id="cb2-743"><a href="#cb2-743" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(PC1, PC2, PC3, PC4) <span class="sc">%&gt;%</span></span>
<span id="cb2-744"><a href="#cb2-744" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(</span>
<span id="cb2-745"><a href="#cb2-745" aria-hidden="true" tabindex="-1"></a>        <span class="at">cols =</span> <span class="fu">c</span>(PC1, PC2, PC3, PC4),</span>
<span id="cb2-746"><a href="#cb2-746" aria-hidden="true" tabindex="-1"></a>        <span class="at">names_to =</span> <span class="st">"component"</span>, </span>
<span id="cb2-747"><a href="#cb2-747" aria-hidden="true" tabindex="-1"></a>        <span class="at">values_to =</span> <span class="st">"loadings"</span></span>
<span id="cb2-748"><a href="#cb2-748" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb2-749"><a href="#cb2-749" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(</span>
<span id="cb2-750"><a href="#cb2-750" aria-hidden="true" tabindex="-1"></a>        <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">abs</span>(loadings))</span>
<span id="cb2-751"><a href="#cb2-751" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb2-752"><a href="#cb2-752" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_histogram</span>(<span class="at">bins =</span> <span class="dv">30</span>) <span class="sc">+</span></span>
<span id="cb2-753"><a href="#cb2-753" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>component) <span class="sc">+</span> </span>
<span id="cb2-754"><a href="#cb2-754" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb2-755"><a href="#cb2-755" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">"Histogram of the loadings from the first four components"</span></span>
<span id="cb2-756"><a href="#cb2-756" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb2-757"><a href="#cb2-757" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>() </span>
<span id="cb2-758"><a href="#cb2-758" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb2-759"><a href="#cb2-759" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-760"><a href="#cb2-760" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb2-761"><a href="#cb2-761" aria-hidden="true" tabindex="-1"></a>datasets_pca_coordinates <span class="ot">&lt;-</span> <span class="fu">lapply</span>(</span>
<span id="cb2-762"><a href="#cb2-762" aria-hidden="true" tabindex="-1"></a>    datasets_normalized,</span>
<span id="cb2-763"><a href="#cb2-763" aria-hidden="true" tabindex="-1"></a>    get_pca_coordinates,</span>
<span id="cb2-764"><a href="#cb2-764" aria-hidden="true" tabindex="-1"></a>    <span class="at">pca_fit =</span> pca_fit</span>
<span id="cb2-765"><a href="#cb2-765" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-766"><a href="#cb2-766" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-767"><a href="#cb2-767" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb2-768"><a href="#cb2-768" aria-hidden="true" tabindex="-1"></a>    datasets_pca_coordinates,</span>
<span id="cb2-769"><a href="#cb2-769" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/pca_merging/datasets_pca_coordinates.rds"</span></span>
<span id="cb2-770"><a href="#cb2-770" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-771"><a href="#cb2-771" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb2-772"><a href="#cb2-772" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-775"><a href="#cb2-775" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb2-776"><a href="#cb2-776" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span>load_at_setup){</span>
<span id="cb2-777"><a href="#cb2-777" aria-hidden="true" tabindex="-1"></a>    datasets_pca_coordinates <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(</span>
<span id="cb2-778"><a href="#cb2-778" aria-hidden="true" tabindex="-1"></a>        <span class="st">"../results/rds_files/pca_merging/datasets_pca_coordinates.rds"</span></span>
<span id="cb2-779"><a href="#cb2-779" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb2-780"><a href="#cb2-780" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb2-781"><a href="#cb2-781" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb2-782"><a href="#cb2-782" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-783"><a href="#cb2-783" aria-hidden="true" tabindex="-1"></a>@fig-pca-og shows all the samples from TCGA and cohort projected into the</span>
<span id="cb2-784"><a href="#cb2-784" aria-hidden="true" tabindex="-1"></a>molecular landscape. The samples are mixing well, the only thing is that</span>
<span id="cb2-785"><a href="#cb2-785" aria-hidden="true" tabindex="-1"></a>METABRIC is a bit offset to the left bottom.</span>
<span id="cb2-786"><a href="#cb2-786" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-787"><a href="#cb2-787" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb2-788"><a href="#cb2-788" aria-hidden="true" tabindex="-1"></a>df_pca_coordinates <span class="ot">&lt;-</span> datasets_pca_coordinates <span class="sc">%&gt;%</span> </span>
<span id="cb2-789"><a href="#cb2-789" aria-hidden="true" tabindex="-1"></a>    <span class="fu">do.call</span>(rbind, .) <span class="sc">%&gt;%</span></span>
<span id="cb2-790"><a href="#cb2-790" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span> </span>
<span id="cb2-791"><a href="#cb2-791" aria-hidden="true" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"sample_name"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb2-792"><a href="#cb2-792" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">inner_join</span>(</span>
<span id="cb2-793"><a href="#cb2-793" aria-hidden="true" tabindex="-1"></a>        .,</span>
<span id="cb2-794"><a href="#cb2-794" aria-hidden="true" tabindex="-1"></a>        merged_col_data,</span>
<span id="cb2-795"><a href="#cb2-795" aria-hidden="true" tabindex="-1"></a>        <span class="at">by =</span> <span class="st">"sample_name"</span></span>
<span id="cb2-796"><a href="#cb2-796" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb2-797"><a href="#cb2-797" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-798"><a href="#cb2-798" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb2-799"><a href="#cb2-799" aria-hidden="true" tabindex="-1"></a>    df_pca_coordinates,</span>
<span id="cb2-800"><a href="#cb2-800" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/pca_merging/df_pca_coordinates.rds"</span></span>
<span id="cb2-801"><a href="#cb2-801" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-802"><a href="#cb2-802" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb2-803"><a href="#cb2-803" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-806"><a href="#cb2-806" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb2-807"><a href="#cb2-807" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-pca-og</span></span>
<span id="cb2-808"><a href="#cb2-808" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Original PC3 and PC4 of all samples, including test and </span></span>
<span id="cb2-809"><a href="#cb2-809" aria-hidden="true" tabindex="-1"></a><span class="co">#|     train samples, of TCGA and METABRIC.</span></span>
<span id="cb2-810"><a href="#cb2-810" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_pca_coordinates</span>(</span>
<span id="cb2-811"><a href="#cb2-811" aria-hidden="true" tabindex="-1"></a>    df_pca_coordinates <span class="sc">%&gt;%</span> </span>
<span id="cb2-812"><a href="#cb2-812" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"tcga"</span>, <span class="st">"metabric"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb2-813"><a href="#cb2-813" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">slice</span>(<span class="fu">sample</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">n</span>())),</span>
<span id="cb2-814"><a href="#cb2-814" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"cohort"</span>,</span>
<span id="cb2-815"><a href="#cb2-815" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC3"</span>,</span>
<span id="cb2-816"><a href="#cb2-816" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC4"</span>,</span>
<span id="cb2-817"><a href="#cb2-817" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">1</span>,</span>
<span id="cb2-818"><a href="#cb2-818" aria-hidden="true" tabindex="-1"></a>    <span class="at">base_size =</span> <span class="dv">10</span>,</span>
<span id="cb2-819"><a href="#cb2-819" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="fu">paste0</span>(</span>
<span id="cb2-820"><a href="#cb2-820" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Embedding of all samples from TCGA"</span>,</span>
<span id="cb2-821"><a href="#cb2-821" aria-hidden="true" tabindex="-1"></a>        <span class="st">" and METABRIC"</span></span>
<span id="cb2-822"><a href="#cb2-822" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb2-823"><a href="#cb2-823" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-824"><a href="#cb2-824" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb2-825"><a href="#cb2-825" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-826"><a href="#cb2-826" aria-hidden="true" tabindex="-1"></a>And @fig-pca-pam50.</span>
<span id="cb2-827"><a href="#cb2-827" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-830"><a href="#cb2-830" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb2-831"><a href="#cb2-831" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-pca-pam50</span></span>
<span id="cb2-832"><a href="#cb2-832" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Original PC3 and PC4 of all samples, including test and </span></span>
<span id="cb2-833"><a href="#cb2-833" aria-hidden="true" tabindex="-1"></a><span class="co">#|     train samples, of TCGA and METABRIC.</span></span>
<span id="cb2-834"><a href="#cb2-834" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_pca_coordinates</span>(</span>
<span id="cb2-835"><a href="#cb2-835" aria-hidden="true" tabindex="-1"></a>    df_pca_coordinates <span class="sc">%&gt;%</span> </span>
<span id="cb2-836"><a href="#cb2-836" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"tcga"</span>, <span class="st">"metabric"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb2-837"><a href="#cb2-837" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb2-838"><a href="#cb2-838" aria-hidden="true" tabindex="-1"></a>            pam50 <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"basal"</span>, <span class="st">"her2"</span>, <span class="st">"lumb"</span>, <span class="st">"luma"</span>, <span class="st">"normal"</span>)</span>
<span id="cb2-839"><a href="#cb2-839" aria-hidden="true" tabindex="-1"></a>        ) <span class="sc">%&gt;%</span></span>
<span id="cb2-840"><a href="#cb2-840" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">slice</span>(<span class="fu">sample</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">n</span>())),</span>
<span id="cb2-841"><a href="#cb2-841" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"pam50"</span>,</span>
<span id="cb2-842"><a href="#cb2-842" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC3"</span>,</span>
<span id="cb2-843"><a href="#cb2-843" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC4"</span>,</span>
<span id="cb2-844"><a href="#cb2-844" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">1</span>,</span>
<span id="cb2-845"><a href="#cb2-845" aria-hidden="true" tabindex="-1"></a>    <span class="at">base_size =</span> <span class="dv">10</span>,</span>
<span id="cb2-846"><a href="#cb2-846" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="fu">paste0</span>(</span>
<span id="cb2-847"><a href="#cb2-847" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Embedding of all samples from TCGA"</span>,</span>
<span id="cb2-848"><a href="#cb2-848" aria-hidden="true" tabindex="-1"></a>        <span class="st">" and METABRIC"</span></span>
<span id="cb2-849"><a href="#cb2-849" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb2-850"><a href="#cb2-850" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-851"><a href="#cb2-851" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb2-852"><a href="#cb2-852" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-853"><a href="#cb2-853" aria-hidden="true" tabindex="-1"></a>In order to check if the integration made sense, it is possible to color by ER</span>
<span id="cb2-854"><a href="#cb2-854" aria-hidden="true" tabindex="-1"></a>status, NPI for METABRIC samples and tumor/node stage for TCGA. To check ER</span>
<span id="cb2-855"><a href="#cb2-855" aria-hidden="true" tabindex="-1"></a>signaling scores, the pathways Estrogen response early/late and $SET_{ER/PR}$</span>
<span id="cb2-856"><a href="#cb2-856" aria-hidden="true" tabindex="-1"></a>are used to color the projection.</span>
<span id="cb2-857"><a href="#cb2-857" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-858"><a href="#cb2-858" aria-hidden="true" tabindex="-1"></a>@fig-pca-age shows the coloring by age.</span>
<span id="cb2-859"><a href="#cb2-859" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-862"><a href="#cb2-862" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb2-863"><a href="#cb2-863" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-pca-age</span></span>
<span id="cb2-864"><a href="#cb2-864" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: PC3 and PC4 of samples from TCGA and METABRIC</span></span>
<span id="cb2-865"><a href="#cb2-865" aria-hidden="true" tabindex="-1"></a><span class="co">#|     colored by age.</span></span>
<span id="cb2-866"><a href="#cb2-866" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_pca_coordinates</span>(</span>
<span id="cb2-867"><a href="#cb2-867" aria-hidden="true" tabindex="-1"></a>    df_pca_coordinates <span class="sc">%&gt;%</span> </span>
<span id="cb2-868"><a href="#cb2-868" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"tcga"</span>, <span class="st">"metabric"</span>)),</span>
<span id="cb2-869"><a href="#cb2-869" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"age"</span>,</span>
<span id="cb2-870"><a href="#cb2-870" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC3"</span>,</span>
<span id="cb2-871"><a href="#cb2-871" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC4"</span>,</span>
<span id="cb2-872"><a href="#cb2-872" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">1</span>,</span>
<span id="cb2-873"><a href="#cb2-873" aria-hidden="true" tabindex="-1"></a>    <span class="at">base_size =</span> <span class="dv">10</span>,</span>
<span id="cb2-874"><a href="#cb2-874" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="fu">paste0</span>(</span>
<span id="cb2-875"><a href="#cb2-875" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Embedding of all samples from TCGA"</span>,</span>
<span id="cb2-876"><a href="#cb2-876" aria-hidden="true" tabindex="-1"></a>        <span class="st">" and METABRIC"</span></span>
<span id="cb2-877"><a href="#cb2-877" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb2-878"><a href="#cb2-878" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span> ggplot2<span class="sc">::</span><span class="fu">scale_color_viridis_c</span>() <span class="sc">+</span></span>
<span id="cb2-879"><a href="#cb2-879" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">color =</span> <span class="st">"Age"</span>)</span>
<span id="cb2-880"><a href="#cb2-880" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb2-881"><a href="#cb2-881" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-882"><a href="#cb2-882" aria-hidden="true" tabindex="-1"></a>Age seems to be well mixed. @fig-pca-er-status shows the</span>
<span id="cb2-883"><a href="#cb2-883" aria-hidden="true" tabindex="-1"></a>PCA colored by ER status.</span>
<span id="cb2-884"><a href="#cb2-884" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-887"><a href="#cb2-887" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb2-888"><a href="#cb2-888" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-pca-er-status</span></span>
<span id="cb2-889"><a href="#cb2-889" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: PC3 and PC4 of samples from TCGA and METABRIC</span></span>
<span id="cb2-890"><a href="#cb2-890" aria-hidden="true" tabindex="-1"></a><span class="co">#|     colored by ER status.</span></span>
<span id="cb2-891"><a href="#cb2-891" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_pca_coordinates</span>(</span>
<span id="cb2-892"><a href="#cb2-892" aria-hidden="true" tabindex="-1"></a>    df_pca_coordinates <span class="sc">%&gt;%</span> </span>
<span id="cb2-893"><a href="#cb2-893" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"tcga"</span>, <span class="st">"metabric"</span>)),</span>
<span id="cb2-894"><a href="#cb2-894" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"er_status"</span>,</span>
<span id="cb2-895"><a href="#cb2-895" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC3"</span>,</span>
<span id="cb2-896"><a href="#cb2-896" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC4"</span>,</span>
<span id="cb2-897"><a href="#cb2-897" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">1</span>,</span>
<span id="cb2-898"><a href="#cb2-898" aria-hidden="true" tabindex="-1"></a>    <span class="at">base_size =</span> <span class="dv">10</span>,</span>
<span id="cb2-899"><a href="#cb2-899" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="fu">paste0</span>(</span>
<span id="cb2-900"><a href="#cb2-900" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Embedding of all samples from TCGA"</span>,</span>
<span id="cb2-901"><a href="#cb2-901" aria-hidden="true" tabindex="-1"></a>        <span class="st">" and METABRIC"</span></span>
<span id="cb2-902"><a href="#cb2-902" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb2-903"><a href="#cb2-903" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span> ggplot2<span class="sc">::</span><span class="fu">scale_color_viridis_d</span>() <span class="sc">+</span></span>
<span id="cb2-904"><a href="#cb2-904" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">color =</span> <span class="st">"ER status"</span>)</span>
<span id="cb2-905"><a href="#cb2-905" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb2-906"><a href="#cb2-906" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-907"><a href="#cb2-907" aria-hidden="true" tabindex="-1"></a>It has a very good separation among the different patients.</span>
<span id="cb2-908"><a href="#cb2-908" aria-hidden="true" tabindex="-1"></a>@fig-pca-npi is the PCA embedding colored by NPI. Only METABRIC samples</span>
<span id="cb2-909"><a href="#cb2-909" aria-hidden="true" tabindex="-1"></a>have this information.</span>
<span id="cb2-910"><a href="#cb2-910" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-913"><a href="#cb2-913" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb2-914"><a href="#cb2-914" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-pca-npi</span></span>
<span id="cb2-915"><a href="#cb2-915" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: PC3 and PC4 of samples from METABRIC only</span></span>
<span id="cb2-916"><a href="#cb2-916" aria-hidden="true" tabindex="-1"></a><span class="co">#|     colored by NPI.</span></span>
<span id="cb2-917"><a href="#cb2-917" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_pca_coordinates</span>(</span>
<span id="cb2-918"><a href="#cb2-918" aria-hidden="true" tabindex="-1"></a>    df_pca_coordinates <span class="sc">%&gt;%</span> </span>
<span id="cb2-919"><a href="#cb2-919" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"metabric"</span>)),</span>
<span id="cb2-920"><a href="#cb2-920" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"npi"</span>,</span>
<span id="cb2-921"><a href="#cb2-921" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC3"</span>,</span>
<span id="cb2-922"><a href="#cb2-922" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC4"</span>,</span>
<span id="cb2-923"><a href="#cb2-923" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">1</span>,</span>
<span id="cb2-924"><a href="#cb2-924" aria-hidden="true" tabindex="-1"></a>    <span class="at">base_size =</span> <span class="dv">10</span>,</span>
<span id="cb2-925"><a href="#cb2-925" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="fu">paste0</span>(</span>
<span id="cb2-926"><a href="#cb2-926" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Embedding of all samples from METABRIC"</span></span>
<span id="cb2-927"><a href="#cb2-927" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb2-928"><a href="#cb2-928" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span> ggplot2<span class="sc">::</span><span class="fu">scale_color_viridis_c</span>(<span class="at">na.value =</span> <span class="dv">0</span>) <span class="sc">+</span></span>
<span id="cb2-929"><a href="#cb2-929" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">color =</span> <span class="st">"NPI"</span>)</span>
<span id="cb2-930"><a href="#cb2-930" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb2-931"><a href="#cb2-931" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-932"><a href="#cb2-932" aria-hidden="true" tabindex="-1"></a>It looks good, ER+ BC patients have lower NPI, which makes sense. </span>
<span id="cb2-933"><a href="#cb2-933" aria-hidden="true" tabindex="-1"></a>@fig-pca-tumor-purity shows the biplot colored by tumor purity.</span>
<span id="cb2-934"><a href="#cb2-934" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-937"><a href="#cb2-937" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb2-938"><a href="#cb2-938" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-pca-tumor-purity</span></span>
<span id="cb2-939"><a href="#cb2-939" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: PC3 and PC4 of samples from METABRIC only</span></span>
<span id="cb2-940"><a href="#cb2-940" aria-hidden="true" tabindex="-1"></a><span class="co">#|     colored by tumor purity.</span></span>
<span id="cb2-941"><a href="#cb2-941" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_pca_coordinates</span>(</span>
<span id="cb2-942"><a href="#cb2-942" aria-hidden="true" tabindex="-1"></a>    df_pca_coordinates <span class="sc">%&gt;%</span> </span>
<span id="cb2-943"><a href="#cb2-943" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"metabric"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb2-944"><a href="#cb2-944" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">filter</span>(CELLULARITY <span class="sc">!=</span> <span class="st">""</span>),</span>
<span id="cb2-945"><a href="#cb2-945" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"CELLULARITY"</span>,</span>
<span id="cb2-946"><a href="#cb2-946" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC3"</span>,</span>
<span id="cb2-947"><a href="#cb2-947" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC4"</span>,</span>
<span id="cb2-948"><a href="#cb2-948" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">1</span>,</span>
<span id="cb2-949"><a href="#cb2-949" aria-hidden="true" tabindex="-1"></a>    <span class="at">base_size =</span> <span class="dv">10</span>,</span>
<span id="cb2-950"><a href="#cb2-950" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="fu">paste0</span>(</span>
<span id="cb2-951"><a href="#cb2-951" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Embedding of all samples from METABRIC"</span></span>
<span id="cb2-952"><a href="#cb2-952" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb2-953"><a href="#cb2-953" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span> ggplot2<span class="sc">::</span><span class="fu">scale_color_viridis_d</span>(<span class="at">na.value =</span> <span class="dv">0</span>) <span class="sc">+</span></span>
<span id="cb2-954"><a href="#cb2-954" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">color =</span> <span class="st">"Tumor purity"</span>)</span>
<span id="cb2-955"><a href="#cb2-955" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb2-956"><a href="#cb2-956" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-957"><a href="#cb2-957" aria-hidden="true" tabindex="-1"></a>The tumor purity is not a factor that is affecting the embedding as samples</span>
<span id="cb2-958"><a href="#cb2-958" aria-hidden="true" tabindex="-1"></a>with low cellularity (&lt; 40% of tumor cells) and high cellularity</span>
<span id="cb2-959"><a href="#cb2-959" aria-hidden="true" tabindex="-1"></a>(&gt; 70% of tumor cells) intermingle. </span>
<span id="cb2-960"><a href="#cb2-960" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-961"><a href="#cb2-961" aria-hidden="true" tabindex="-1"></a>METABRIC has also inferred menopausal status for the patients, we use</span>
<span id="cb2-962"><a href="#cb2-962" aria-hidden="true" tabindex="-1"></a>this information to check if there is any enrichment based on the </span>
<span id="cb2-963"><a href="#cb2-963" aria-hidden="true" tabindex="-1"></a>patient menopausal status (@fig-pca-men-status)</span>
<span id="cb2-964"><a href="#cb2-964" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-967"><a href="#cb2-967" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb2-968"><a href="#cb2-968" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-pca-men-status</span></span>
<span id="cb2-969"><a href="#cb2-969" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: PC3 and PC4 of samples from METABRIC only</span></span>
<span id="cb2-970"><a href="#cb2-970" aria-hidden="true" tabindex="-1"></a><span class="co">#|     colored by menopausal status.</span></span>
<span id="cb2-971"><a href="#cb2-971" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_pca_coordinates</span>(</span>
<span id="cb2-972"><a href="#cb2-972" aria-hidden="true" tabindex="-1"></a>    df_pca_coordinates <span class="sc">%&gt;%</span> </span>
<span id="cb2-973"><a href="#cb2-973" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"metabric"</span>)),</span>
<span id="cb2-974"><a href="#cb2-974" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"INFERRED_MENOPAUSAL_STATE"</span>,</span>
<span id="cb2-975"><a href="#cb2-975" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC3"</span>,</span>
<span id="cb2-976"><a href="#cb2-976" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC4"</span>,</span>
<span id="cb2-977"><a href="#cb2-977" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">1</span>,</span>
<span id="cb2-978"><a href="#cb2-978" aria-hidden="true" tabindex="-1"></a>    <span class="at">base_size =</span> <span class="dv">10</span>,</span>
<span id="cb2-979"><a href="#cb2-979" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="fu">paste0</span>(</span>
<span id="cb2-980"><a href="#cb2-980" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Embedding of all samples from METABRIC"</span></span>
<span id="cb2-981"><a href="#cb2-981" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb2-982"><a href="#cb2-982" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span> ggplot2<span class="sc">::</span><span class="fu">scale_color_viridis_d</span>(<span class="at">na.value =</span> <span class="dv">0</span>) <span class="sc">+</span></span>
<span id="cb2-983"><a href="#cb2-983" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">color =</span> <span class="st">"Menopausal</span><span class="sc">\n</span><span class="st">status"</span>)</span>
<span id="cb2-984"><a href="#cb2-984" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb2-985"><a href="#cb2-985" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-986"><a href="#cb2-986" aria-hidden="true" tabindex="-1"></a>We don't see any difference between the two status. We proceed now with </span>
<span id="cb2-987"><a href="#cb2-987" aria-hidden="true" tabindex="-1"></a>the analysis using clinical factors from TCGA.</span>
<span id="cb2-988"><a href="#cb2-988" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-989"><a href="#cb2-989" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=12, fig.height=4}</span></span>
<span id="cb2-990"><a href="#cb2-990" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-pca-tumor-node</span></span>
<span id="cb2-991"><a href="#cb2-991" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: PC3 and PC4 of samples from TCGA only</span></span>
<span id="cb2-992"><a href="#cb2-992" aria-hidden="true" tabindex="-1"></a><span class="co">#|     colored by tumor stage and node stage.</span></span>
<span id="cb2-993"><a href="#cb2-993" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-994"><a href="#cb2-994" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">plot_pca_coordinates</span>(</span>
<span id="cb2-995"><a href="#cb2-995" aria-hidden="true" tabindex="-1"></a>    df_pca_coordinates <span class="sc">%&gt;%</span> </span>
<span id="cb2-996"><a href="#cb2-996" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"tcga"</span>)),</span>
<span id="cb2-997"><a href="#cb2-997" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"tumor_stage"</span>,</span>
<span id="cb2-998"><a href="#cb2-998" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC3"</span>,</span>
<span id="cb2-999"><a href="#cb2-999" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC4"</span>,</span>
<span id="cb2-1000"><a href="#cb2-1000" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">1</span>,</span>
<span id="cb2-1001"><a href="#cb2-1001" aria-hidden="true" tabindex="-1"></a>    <span class="at">base_size =</span> <span class="dv">10</span>,</span>
<span id="cb2-1002"><a href="#cb2-1002" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="fu">paste0</span>(</span>
<span id="cb2-1003"><a href="#cb2-1003" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Embedding of all samples from TCGA"</span></span>
<span id="cb2-1004"><a href="#cb2-1004" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb2-1005"><a href="#cb2-1005" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span> ggplot2<span class="sc">::</span><span class="fu">scale_color_viridis_d</span>() <span class="sc">+</span></span>
<span id="cb2-1006"><a href="#cb2-1006" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span> </span>
<span id="cb2-1007"><a href="#cb2-1007" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">color =</span> <span class="st">"Tumor</span><span class="sc">\n</span><span class="st">stage"</span>)</span>
<span id="cb2-1008"><a href="#cb2-1008" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1009"><a href="#cb2-1009" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">plot_pca_coordinates</span>(</span>
<span id="cb2-1010"><a href="#cb2-1010" aria-hidden="true" tabindex="-1"></a>    df_pca_coordinates <span class="sc">%&gt;%</span> </span>
<span id="cb2-1011"><a href="#cb2-1011" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"tcga"</span>)),</span>
<span id="cb2-1012"><a href="#cb2-1012" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"node_stage"</span>,</span>
<span id="cb2-1013"><a href="#cb2-1013" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC3"</span>,</span>
<span id="cb2-1014"><a href="#cb2-1014" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC4"</span>,</span>
<span id="cb2-1015"><a href="#cb2-1015" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">1</span>,</span>
<span id="cb2-1016"><a href="#cb2-1016" aria-hidden="true" tabindex="-1"></a>    <span class="at">base_size =</span> <span class="dv">10</span>,</span>
<span id="cb2-1017"><a href="#cb2-1017" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="fu">paste0</span>(</span>
<span id="cb2-1018"><a href="#cb2-1018" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Embedding of all samples from TCGA"</span></span>
<span id="cb2-1019"><a href="#cb2-1019" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb2-1020"><a href="#cb2-1020" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span> ggplot2<span class="sc">::</span><span class="fu">scale_color_viridis_d</span>() <span class="sc">+</span></span>
<span id="cb2-1021"><a href="#cb2-1021" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span> </span>
<span id="cb2-1022"><a href="#cb2-1022" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">color =</span> <span class="st">"Node</span><span class="sc">\n</span><span class="st">stage"</span>)</span>
<span id="cb2-1023"><a href="#cb2-1023" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1024"><a href="#cb2-1024" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(p1, p2, <span class="at">ncol =</span> <span class="dv">2</span>)</span>
<span id="cb2-1025"><a href="#cb2-1025" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb2-1026"><a href="#cb2-1026" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1027"><a href="#cb2-1027" aria-hidden="true" tabindex="-1"></a>For both tumor and node stage the samples seem to have mixed well, indicating</span>
<span id="cb2-1028"><a href="#cb2-1028" aria-hidden="true" tabindex="-1"></a>the projection is not a function of the stages, as shown in </span>
<span id="cb2-1029"><a href="#cb2-1029" aria-hidden="true" tabindex="-1"></a>@fig-pca-tumor-node.</span>
<span id="cb2-1030"><a href="#cb2-1030" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1031"><a href="#cb2-1031" aria-hidden="true" tabindex="-1"></a>And finally @fig-pca-pam50-2 below shows the color by PAM50 molecular</span>
<span id="cb2-1032"><a href="#cb2-1032" aria-hidden="true" tabindex="-1"></a>subtype.</span>
<span id="cb2-1033"><a href="#cb2-1033" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1036"><a href="#cb2-1036" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb2-1037"><a href="#cb2-1037" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-pca-pam50-2</span></span>
<span id="cb2-1038"><a href="#cb2-1038" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Corrected PC3 and PC4 of samples from TCGA and METABRIC</span></span>
<span id="cb2-1039"><a href="#cb2-1039" aria-hidden="true" tabindex="-1"></a><span class="co">#|     colored by PAM50 molecular subtype.</span></span>
<span id="cb2-1040"><a href="#cb2-1040" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_pca_coordinates</span>(</span>
<span id="cb2-1041"><a href="#cb2-1041" aria-hidden="true" tabindex="-1"></a>    df_pca_coordinates <span class="sc">%&gt;%</span> </span>
<span id="cb2-1042"><a href="#cb2-1042" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"tcga"</span>, <span class="st">"metabric"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb2-1043"><a href="#cb2-1043" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span>(pam50 <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"claudin-low"</span>, <span class="st">"nc"</span>))),</span>
<span id="cb2-1044"><a href="#cb2-1044" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"pam50"</span>,</span>
<span id="cb2-1045"><a href="#cb2-1045" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC3"</span>,</span>
<span id="cb2-1046"><a href="#cb2-1046" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC4"</span>,</span>
<span id="cb2-1047"><a href="#cb2-1047" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">1</span>,</span>
<span id="cb2-1048"><a href="#cb2-1048" aria-hidden="true" tabindex="-1"></a>    <span class="at">base_size =</span> <span class="dv">10</span>,</span>
<span id="cb2-1049"><a href="#cb2-1049" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="fu">paste0</span>(</span>
<span id="cb2-1050"><a href="#cb2-1050" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Embedding of all samples from TCGA and METABRIC"</span></span>
<span id="cb2-1051"><a href="#cb2-1051" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb2-1052"><a href="#cb2-1052" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb2-1053"><a href="#cb2-1053" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span> </span>
<span id="cb2-1054"><a href="#cb2-1054" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">color =</span> <span class="st">"PAM50"</span>)</span>
<span id="cb2-1055"><a href="#cb2-1055" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb2-1056"><a href="#cb2-1056" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1057"><a href="#cb2-1057" aria-hidden="true" tabindex="-1"></a>The molecular subtypes are projected as expected, going from left to right there</span>
<span id="cb2-1058"><a href="#cb2-1058" aria-hidden="true" tabindex="-1"></a>are the basal-like BC patients, going to HER2 and then they are split between</span>
<span id="cb2-1059"><a href="#cb2-1059" aria-hidden="true" tabindex="-1"></a>luminal A and B. The normal like patients are in the top and closer to the</span>
<span id="cb2-1060"><a href="#cb2-1060" aria-hidden="true" tabindex="-1"></a>luminal A, reflecting the biology.</span>
<span id="cb2-1061"><a href="#cb2-1061" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1062"><a href="#cb2-1062" aria-hidden="true" tabindex="-1"></a>The next figures show the ER signaling measures coloring the projections.</span>
<span id="cb2-1063"><a href="#cb2-1063" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1066"><a href="#cb2-1066" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb2-1067"><a href="#cb2-1067" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-pca-seterpr</span></span>
<span id="cb2-1068"><a href="#cb2-1068" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: biplot of PC3 and PC4 of samples from TCGA and METABRIC</span></span>
<span id="cb2-1069"><a href="#cb2-1069" aria-hidden="true" tabindex="-1"></a><span class="co">#|     colored by the SET ER/PR score.</span></span>
<span id="cb2-1070"><a href="#cb2-1070" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1071"><a href="#cb2-1071" aria-hidden="true" tabindex="-1"></a>color <span class="ot">&lt;-</span> <span class="st">"SET_ERPR"</span></span>
<span id="cb2-1072"><a href="#cb2-1072" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="st">"PC3"</span></span>
<span id="cb2-1073"><a href="#cb2-1073" aria-hidden="true" tabindex="-1"></a>y <span class="ot">&lt;-</span> <span class="st">"PC4"</span></span>
<span id="cb2-1074"><a href="#cb2-1074" aria-hidden="true" tabindex="-1"></a>size <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb2-1075"><a href="#cb2-1075" aria-hidden="true" tabindex="-1"></a>base_size <span class="ot">&lt;-</span> <span class="dv">10</span></span>
<span id="cb2-1076"><a href="#cb2-1076" aria-hidden="true" tabindex="-1"></a>title <span class="ot">&lt;-</span> <span class="st">"Embedding of all samples from TCGA and METABRIC"</span></span>
<span id="cb2-1077"><a href="#cb2-1077" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1078"><a href="#cb2-1078" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> df_pca_coordinates <span class="sc">%&gt;%</span> </span>
<span id="cb2-1079"><a href="#cb2-1079" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"tcga"</span>, <span class="st">"metabric"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb2-1080"><a href="#cb2-1080" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes_string</span>(<span class="at">x =</span> x, <span class="at">y =</span> y, <span class="at">z =</span> color)) <span class="sc">+</span> </span>
<span id="cb2-1081"><a href="#cb2-1081" aria-hidden="true" tabindex="-1"></a>        ggplot2<span class="sc">::</span><span class="fu">stat_summary_hex</span>(<span class="at">bins =</span> <span class="dv">35</span>) <span class="sc">+</span></span>
<span id="cb2-1082"><a href="#cb2-1082" aria-hidden="true" tabindex="-1"></a>        ggplot2<span class="sc">::</span><span class="fu">scale_fill_viridis_c</span>() <span class="sc">+</span></span>
<span id="cb2-1083"><a href="#cb2-1083" aria-hidden="true" tabindex="-1"></a>        ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb2-1084"><a href="#cb2-1084" aria-hidden="true" tabindex="-1"></a>            <span class="at">title =</span> title,</span>
<span id="cb2-1085"><a href="#cb2-1085" aria-hidden="true" tabindex="-1"></a>            <span class="at">fill =</span> <span class="fu">expression</span>(SET[ER<span class="sc">/</span>PR])</span>
<span id="cb2-1086"><a href="#cb2-1086" aria-hidden="true" tabindex="-1"></a>        ) <span class="sc">+</span> </span>
<span id="cb2-1087"><a href="#cb2-1087" aria-hidden="true" tabindex="-1"></a>        ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> base_size)</span>
<span id="cb2-1088"><a href="#cb2-1088" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1089"><a href="#cb2-1089" aria-hidden="true" tabindex="-1"></a>p1</span>
<span id="cb2-1090"><a href="#cb2-1090" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb2-1091"><a href="#cb2-1091" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1094"><a href="#cb2-1094" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb2-1095"><a href="#cb2-1095" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-pca-early</span></span>
<span id="cb2-1096"><a href="#cb2-1096" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: biplot of PC3 and PC4 of samples from TCGA and METABRIC</span></span>
<span id="cb2-1097"><a href="#cb2-1097" aria-hidden="true" tabindex="-1"></a><span class="co">#|     colored by the hallmark estrogen response early.</span></span>
<span id="cb2-1098"><a href="#cb2-1098" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1099"><a href="#cb2-1099" aria-hidden="true" tabindex="-1"></a>color <span class="ot">&lt;-</span> <span class="st">"HALLMARK_ESTROGEN_RESPONSE_EARLY"</span></span>
<span id="cb2-1100"><a href="#cb2-1100" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="st">"PC3"</span></span>
<span id="cb2-1101"><a href="#cb2-1101" aria-hidden="true" tabindex="-1"></a>y <span class="ot">&lt;-</span> <span class="st">"PC4"</span></span>
<span id="cb2-1102"><a href="#cb2-1102" aria-hidden="true" tabindex="-1"></a>size <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb2-1103"><a href="#cb2-1103" aria-hidden="true" tabindex="-1"></a>base_size <span class="ot">&lt;-</span> <span class="dv">10</span></span>
<span id="cb2-1104"><a href="#cb2-1104" aria-hidden="true" tabindex="-1"></a>title <span class="ot">&lt;-</span> <span class="st">"Embedding of all samples from TCGA and METABRIC"</span></span>
<span id="cb2-1105"><a href="#cb2-1105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1106"><a href="#cb2-1106" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> df_pca_coordinates <span class="sc">%&gt;%</span> </span>
<span id="cb2-1107"><a href="#cb2-1107" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"tcga"</span>, <span class="st">"metabric"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb2-1108"><a href="#cb2-1108" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes_string</span>(<span class="at">x =</span> x, <span class="at">y =</span> y, <span class="at">z =</span> color)) <span class="sc">+</span> </span>
<span id="cb2-1109"><a href="#cb2-1109" aria-hidden="true" tabindex="-1"></a>        ggplot2<span class="sc">::</span><span class="fu">stat_summary_hex</span>(<span class="at">bins =</span> <span class="dv">35</span>) <span class="sc">+</span></span>
<span id="cb2-1110"><a href="#cb2-1110" aria-hidden="true" tabindex="-1"></a>        ggplot2<span class="sc">::</span><span class="fu">scale_fill_viridis_c</span>() <span class="sc">+</span></span>
<span id="cb2-1111"><a href="#cb2-1111" aria-hidden="true" tabindex="-1"></a>        ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb2-1112"><a href="#cb2-1112" aria-hidden="true" tabindex="-1"></a>            <span class="at">title =</span> title,</span>
<span id="cb2-1113"><a href="#cb2-1113" aria-hidden="true" tabindex="-1"></a>            <span class="at">fill =</span> <span class="st">"Estrogen</span><span class="sc">\n</span><span class="st">early"</span></span>
<span id="cb2-1114"><a href="#cb2-1114" aria-hidden="true" tabindex="-1"></a>        ) <span class="sc">+</span> </span>
<span id="cb2-1115"><a href="#cb2-1115" aria-hidden="true" tabindex="-1"></a>        ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> base_size)</span>
<span id="cb2-1116"><a href="#cb2-1116" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1117"><a href="#cb2-1117" aria-hidden="true" tabindex="-1"></a>p1</span>
<span id="cb2-1118"><a href="#cb2-1118" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb2-1119"><a href="#cb2-1119" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1122"><a href="#cb2-1122" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb2-1123"><a href="#cb2-1123" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-pca-late</span></span>
<span id="cb2-1124"><a href="#cb2-1124" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: biplot of PC3 and PC4 of samples from TCGA and METABRIC</span></span>
<span id="cb2-1125"><a href="#cb2-1125" aria-hidden="true" tabindex="-1"></a><span class="co">#|     colored by the hallmark estrogen response late</span></span>
<span id="cb2-1126"><a href="#cb2-1126" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1127"><a href="#cb2-1127" aria-hidden="true" tabindex="-1"></a>color <span class="ot">&lt;-</span> <span class="st">"HALLMARK_ESTROGEN_RESPONSE_LATE"</span></span>
<span id="cb2-1128"><a href="#cb2-1128" aria-hidden="true" tabindex="-1"></a><span class="co">#color &lt;- "HALLMARK_G2M_CHECKPOINT"</span></span>
<span id="cb2-1129"><a href="#cb2-1129" aria-hidden="true" tabindex="-1"></a><span class="co">#color &lt;- "HALLMARK_TGF_BETA_SIGNALING"</span></span>
<span id="cb2-1130"><a href="#cb2-1130" aria-hidden="true" tabindex="-1"></a><span class="co">#color &lt;- "HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION"</span></span>
<span id="cb2-1131"><a href="#cb2-1131" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="st">"PC3"</span></span>
<span id="cb2-1132"><a href="#cb2-1132" aria-hidden="true" tabindex="-1"></a>y <span class="ot">&lt;-</span> <span class="st">"PC4"</span></span>
<span id="cb2-1133"><a href="#cb2-1133" aria-hidden="true" tabindex="-1"></a>size <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb2-1134"><a href="#cb2-1134" aria-hidden="true" tabindex="-1"></a>base_size <span class="ot">&lt;-</span> <span class="dv">10</span></span>
<span id="cb2-1135"><a href="#cb2-1135" aria-hidden="true" tabindex="-1"></a>title <span class="ot">&lt;-</span> <span class="st">"Embedding of all samples from TCGA and METABRIC"</span></span>
<span id="cb2-1136"><a href="#cb2-1136" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1137"><a href="#cb2-1137" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> df_pca_coordinates <span class="sc">%&gt;%</span> </span>
<span id="cb2-1138"><a href="#cb2-1138" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"tcga"</span>, <span class="st">"metabric"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb2-1139"><a href="#cb2-1139" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes_string</span>(<span class="at">x =</span> x, <span class="at">y =</span> y, <span class="at">z =</span> color)) <span class="sc">+</span> </span>
<span id="cb2-1140"><a href="#cb2-1140" aria-hidden="true" tabindex="-1"></a>        ggplot2<span class="sc">::</span><span class="fu">stat_summary_hex</span>(<span class="at">bins =</span> <span class="dv">35</span>) <span class="sc">+</span></span>
<span id="cb2-1141"><a href="#cb2-1141" aria-hidden="true" tabindex="-1"></a>        ggplot2<span class="sc">::</span><span class="fu">scale_fill_viridis_c</span>() <span class="sc">+</span></span>
<span id="cb2-1142"><a href="#cb2-1142" aria-hidden="true" tabindex="-1"></a>        ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb2-1143"><a href="#cb2-1143" aria-hidden="true" tabindex="-1"></a>            <span class="at">title =</span> title,</span>
<span id="cb2-1144"><a href="#cb2-1144" aria-hidden="true" tabindex="-1"></a>            <span class="at">fill =</span> <span class="st">"Estrogen</span><span class="sc">\n</span><span class="st">late"</span></span>
<span id="cb2-1145"><a href="#cb2-1145" aria-hidden="true" tabindex="-1"></a>        ) <span class="sc">+</span> </span>
<span id="cb2-1146"><a href="#cb2-1146" aria-hidden="true" tabindex="-1"></a>        ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> base_size)</span>
<span id="cb2-1147"><a href="#cb2-1147" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1148"><a href="#cb2-1148" aria-hidden="true" tabindex="-1"></a>p1</span>
<span id="cb2-1149"><a href="#cb2-1149" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb2-1150"><a href="#cb2-1150" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1151"><a href="#cb2-1151" aria-hidden="true" tabindex="-1"></a>In all cases the mixing is relatively good. For $SET_{ER/PR}$ you have a</span>
<span id="cb2-1152"><a href="#cb2-1152" aria-hidden="true" tabindex="-1"></a>gradient going from the right to the left, showing again that</span>
<span id="cb2-1153"><a href="#cb2-1153" aria-hidden="true" tabindex="-1"></a>patients on the far top right probably are those patients with better prognosis.</span>
<span id="cb2-1154"><a href="#cb2-1154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1155"><a href="#cb2-1155" aria-hidden="true" tabindex="-1"></a><span class="fu">## The integration is robust with respect of selection of samples</span></span>
<span id="cb2-1156"><a href="#cb2-1156" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1157"><a href="#cb2-1157" aria-hidden="true" tabindex="-1"></a>The embedding could be due to the patients selected and samples that are close</span>
<span id="cb2-1158"><a href="#cb2-1158" aria-hidden="true" tabindex="-1"></a>to each other in one embedding would be far away from them in another embedding.</span>
<span id="cb2-1159"><a href="#cb2-1159" aria-hidden="true" tabindex="-1"></a>Ten random sets of 1000 patients are sampled and the embedding is calculate. A</span>
<span id="cb2-1160"><a href="#cb2-1160" aria-hidden="true" tabindex="-1"></a>total of 10 PCA embeddings are obtained.</span>
<span id="cb2-1161"><a href="#cb2-1161" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1162"><a href="#cb2-1162" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, warning = FALSE, eval = first_run}</span></span>
<span id="cb2-1163"><a href="#cb2-1163" aria-hidden="true" tabindex="-1"></a>seeds <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">500</span>, <span class="at">by =</span> <span class="dv">29</span>)</span>
<span id="cb2-1164"><a href="#cb2-1164" aria-hidden="true" tabindex="-1"></a>new_pcas <span class="ot">&lt;-</span> <span class="fu">lapply</span>(</span>
<span id="cb2-1165"><a href="#cb2-1165" aria-hidden="true" tabindex="-1"></a>    seeds, </span>
<span id="cb2-1166"><a href="#cb2-1166" aria-hidden="true" tabindex="-1"></a>    get_new_pca, </span>
<span id="cb2-1167"><a href="#cb2-1167" aria-hidden="true" tabindex="-1"></a>    <span class="at">datasets_normalized =</span> datasets_normalized,</span>
<span id="cb2-1168"><a href="#cb2-1168" aria-hidden="true" tabindex="-1"></a>    <span class="at">merged_col_data =</span> merged_col_data,</span>
<span id="cb2-1169"><a href="#cb2-1169" aria-hidden="true" tabindex="-1"></a>    <span class="at">which_cohorts_training =</span> which_cohorts_training</span>
<span id="cb2-1170"><a href="#cb2-1170" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-1171"><a href="#cb2-1171" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(new_pcas) <span class="ot">&lt;-</span> <span class="fu">as.character</span>(seeds)</span>
<span id="cb2-1172"><a href="#cb2-1172" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb2-1173"><a href="#cb2-1173" aria-hidden="true" tabindex="-1"></a>    new_pcas,</span>
<span id="cb2-1174"><a href="#cb2-1174" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/pca_merging/new_pcas.rds"</span></span>
<span id="cb2-1175"><a href="#cb2-1175" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-1176"><a href="#cb2-1176" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb2-1177"><a href="#cb2-1177" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1180"><a href="#cb2-1180" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb2-1181"><a href="#cb2-1181" aria-hidden="true" tabindex="-1"></a><span class="co"># plot PCA of random 50 samples using the new and original embeddings.</span></span>
<span id="cb2-1182"><a href="#cb2-1182" aria-hidden="true" tabindex="-1"></a>plot_pcas_together <span class="ot">&lt;-</span> <span class="cf">function</span>(</span>
<span id="cb2-1183"><a href="#cb2-1183" aria-hidden="true" tabindex="-1"></a>    new_pca_results, </span>
<span id="cb2-1184"><a href="#cb2-1184" aria-hidden="true" tabindex="-1"></a>    which_seed, </span>
<span id="cb2-1185"><a href="#cb2-1185" aria-hidden="true" tabindex="-1"></a>    test_og_pca, </span>
<span id="cb2-1186"><a href="#cb2-1186" aria-hidden="true" tabindex="-1"></a>    seed_nb,</span>
<span id="cb2-1187"><a href="#cb2-1187" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC3"</span>,</span>
<span id="cb2-1188"><a href="#cb2-1188" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC4"</span>,</span>
<span id="cb2-1189"><a href="#cb2-1189" aria-hidden="true" tabindex="-1"></a>    <span class="at">size_point =</span> <span class="dv">3</span>,</span>
<span id="cb2-1190"><a href="#cb2-1190" aria-hidden="true" tabindex="-1"></a>    <span class="at">base_size =</span> <span class="dv">30</span>,</span>
<span id="cb2-1191"><a href="#cb2-1191" aria-hidden="true" tabindex="-1"></a>    <span class="at">nb_samples =</span> <span class="dv">50</span></span>
<span id="cb2-1192"><a href="#cb2-1192" aria-hidden="true" tabindex="-1"></a>){</span>
<span id="cb2-1193"><a href="#cb2-1193" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb2-1194"><a href="#cb2-1194" aria-hidden="true" tabindex="-1"></a>    <span class="fu">set.seed</span>(seed_nb)</span>
<span id="cb2-1195"><a href="#cb2-1195" aria-hidden="true" tabindex="-1"></a>    random_samples <span class="ot">&lt;-</span> <span class="fu">sample</span>(<span class="at">x =</span> test_og_pca<span class="sc">$</span>sample_name, <span class="at">size =</span> nb_samples)</span>
<span id="cb2-1196"><a href="#cb2-1196" aria-hidden="true" tabindex="-1"></a>    new_pca_results <span class="ot">&lt;-</span> new_pca_results[[which_seed]]<span class="sc">$</span>test <span class="sc">%&gt;%</span></span>
<span id="cb2-1197"><a href="#cb2-1197" aria-hidden="true" tabindex="-1"></a>        <span class="fu">do.call</span>(rbind, .)</span>
<span id="cb2-1198"><a href="#cb2-1198" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb2-1199"><a href="#cb2-1199" aria-hidden="true" tabindex="-1"></a>    <span class="fu">list</span>(</span>
<span id="cb2-1200"><a href="#cb2-1200" aria-hidden="true" tabindex="-1"></a>        <span class="st">"og_pca"</span> <span class="ot">=</span> test_og_pca[random_samples, <span class="fu">c</span>(<span class="st">"PC3"</span>, <span class="st">"PC4"</span>)] <span class="sc">%&gt;%</span> </span>
<span id="cb2-1201"><a href="#cb2-1201" aria-hidden="true" tabindex="-1"></a>            data.frame <span class="sc">%&gt;%</span></span>
<span id="cb2-1202"><a href="#cb2-1202" aria-hidden="true" tabindex="-1"></a>            tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"sample_name"</span>),</span>
<span id="cb2-1203"><a href="#cb2-1203" aria-hidden="true" tabindex="-1"></a>        <span class="st">"new_pca"</span> <span class="ot">=</span> new_pca_results[random_samples, <span class="fu">c</span>(<span class="st">"PC3"</span>, <span class="st">"PC4"</span>)] <span class="sc">%&gt;%</span></span>
<span id="cb2-1204"><a href="#cb2-1204" aria-hidden="true" tabindex="-1"></a>            data.frame <span class="sc">%&gt;%</span></span>
<span id="cb2-1205"><a href="#cb2-1205" aria-hidden="true" tabindex="-1"></a>            tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"sample_name"</span>)</span>
<span id="cb2-1206"><a href="#cb2-1206" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(</span>
<span id="cb2-1207"><a href="#cb2-1207" aria-hidden="true" tabindex="-1"></a>        <span class="at">.id =</span> <span class="st">"type_data"</span></span>
<span id="cb2-1208"><a href="#cb2-1208" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span> </span>
<span id="cb2-1209"><a href="#cb2-1209" aria-hidden="true" tabindex="-1"></a>        ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes_string</span>(</span>
<span id="cb2-1210"><a href="#cb2-1210" aria-hidden="true" tabindex="-1"></a>            <span class="at">x =</span> x, </span>
<span id="cb2-1211"><a href="#cb2-1211" aria-hidden="true" tabindex="-1"></a>            <span class="at">y =</span> y, </span>
<span id="cb2-1212"><a href="#cb2-1212" aria-hidden="true" tabindex="-1"></a>            <span class="at">color =</span> <span class="st">"type_data"</span>, </span>
<span id="cb2-1213"><a href="#cb2-1213" aria-hidden="true" tabindex="-1"></a>            <span class="at">group =</span> <span class="st">"sample_name"</span></span>
<span id="cb2-1214"><a href="#cb2-1214" aria-hidden="true" tabindex="-1"></a>        )) <span class="sc">+</span> </span>
<span id="cb2-1215"><a href="#cb2-1215" aria-hidden="true" tabindex="-1"></a>        ggplot2<span class="sc">::</span><span class="fu">geom_point</span>(<span class="at">size =</span> size_point) <span class="sc">+</span> </span>
<span id="cb2-1216"><a href="#cb2-1216" aria-hidden="true" tabindex="-1"></a>        ggplot2<span class="sc">::</span><span class="fu">geom_line</span>(<span class="at">color =</span> <span class="st">"gray"</span>) <span class="sc">+</span> </span>
<span id="cb2-1217"><a href="#cb2-1217" aria-hidden="true" tabindex="-1"></a>        ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste0</span>(<span class="st">"Seed nb: "</span>, which_seed)) <span class="sc">+</span> </span>
<span id="cb2-1218"><a href="#cb2-1218" aria-hidden="true" tabindex="-1"></a>        ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> base_size) <span class="sc">+</span> </span>
<span id="cb2-1219"><a href="#cb2-1219" aria-hidden="true" tabindex="-1"></a>        ggplot2<span class="sc">::</span><span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="fu">c</span>(<span class="fl">0.2</span>, <span class="fl">0.7</span>))</span>
<span id="cb2-1220"><a href="#cb2-1220" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb2-1221"><a href="#cb2-1221" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb2-1222"><a href="#cb2-1222" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1223"><a href="#cb2-1223" aria-hidden="true" tabindex="-1"></a><span class="in">```{r plots-pcas-random, fig.width=18, fig.height=20}</span></span>
<span id="cb2-1224"><a href="#cb2-1224" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-plots-pcas-random</span></span>
<span id="cb2-1225"><a href="#cb2-1225" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Selecting a random set of 1000 patients an rerunning the pipeline</span></span>
<span id="cb2-1226"><a href="#cb2-1226" aria-hidden="true" tabindex="-1"></a><span class="co">#|     to obtain other embeddings. Red color corresponds to the embedding using</span></span>
<span id="cb2-1227"><a href="#cb2-1227" aria-hidden="true" tabindex="-1"></a><span class="co">#|     the new fitted PCA. Blue corresponds to the embedding of the original</span></span>
<span id="cb2-1228"><a href="#cb2-1228" aria-hidden="true" tabindex="-1"></a><span class="co">#|     PCA. Each dot corresponds to a patient.  </span></span>
<span id="cb2-1229"><a href="#cb2-1229" aria-hidden="true" tabindex="-1"></a>plots_new_pcas <span class="ot">&lt;-</span> <span class="fu">sapply</span>(</span>
<span id="cb2-1230"><a href="#cb2-1230" aria-hidden="true" tabindex="-1"></a>    <span class="fu">names</span>(new_pcas),</span>
<span id="cb2-1231"><a href="#cb2-1231" aria-hidden="true" tabindex="-1"></a>    plot_pcas_together,</span>
<span id="cb2-1232"><a href="#cb2-1232" aria-hidden="true" tabindex="-1"></a>    <span class="at">new_pca_results =</span> new_pcas,</span>
<span id="cb2-1233"><a href="#cb2-1233" aria-hidden="true" tabindex="-1"></a>    <span class="at">test_og_pca =</span> df_pca_coordinates <span class="sc">%&gt;%</span></span>
<span id="cb2-1234"><a href="#cb2-1234" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">rownames&lt;-</span><span class="st">`</span>(df_pca_coordinates<span class="sc">$</span>sample_name),</span>
<span id="cb2-1235"><a href="#cb2-1235" aria-hidden="true" tabindex="-1"></a>    <span class="at">seed_nb =</span> <span class="dv">0923</span>, </span>
<span id="cb2-1236"><a href="#cb2-1236" aria-hidden="true" tabindex="-1"></a>    <span class="at">size_point =</span> <span class="dv">3</span>,</span>
<span id="cb2-1237"><a href="#cb2-1237" aria-hidden="true" tabindex="-1"></a>    <span class="at">base_size =</span> <span class="dv">15</span>,</span>
<span id="cb2-1238"><a href="#cb2-1238" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>,</span>
<span id="cb2-1239"><a href="#cb2-1239" aria-hidden="true" tabindex="-1"></a>    <span class="at">simplify =</span> <span class="cn">FALSE</span></span>
<span id="cb2-1240"><a href="#cb2-1240" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-1241"><a href="#cb2-1241" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1242"><a href="#cb2-1242" aria-hidden="true" tabindex="-1"></a>plots_new_pcas_all <span class="ot">&lt;-</span> cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(</span>
<span id="cb2-1243"><a href="#cb2-1243" aria-hidden="true" tabindex="-1"></a>    <span class="at">plotlist =</span> plots_new_pcas, </span>
<span id="cb2-1244"><a href="#cb2-1244" aria-hidden="true" tabindex="-1"></a>    <span class="at">ncol =</span> <span class="dv">3</span></span>
<span id="cb2-1245"><a href="#cb2-1245" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-1246"><a href="#cb2-1246" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1247"><a href="#cb2-1247" aria-hidden="true" tabindex="-1"></a>plots_new_pcas_all</span>
<span id="cb2-1248"><a href="#cb2-1248" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb2-1249"><a href="#cb2-1249" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1250"><a href="#cb2-1250" aria-hidden="true" tabindex="-1"></a>@fig-plots-pcas-random shows that the embedding is invariant for rotation,</span>
<span id="cb2-1251"><a href="#cb2-1251" aria-hidden="true" tabindex="-1"></a>translation and reflection. Therefore, the subset of samples used for the</span>
<span id="cb2-1252"><a href="#cb2-1252" aria-hidden="true" tabindex="-1"></a>embedding does not really matter.</span>
<span id="cb2-1253"><a href="#cb2-1253" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1254"><a href="#cb2-1254" aria-hidden="true" tabindex="-1"></a><span class="fu">## Normalization is necessary to reduce batch effects in other components</span></span>
<span id="cb2-1255"><a href="#cb2-1255" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1256"><a href="#cb2-1256" aria-hidden="true" tabindex="-1"></a>Since PCA is removing batch effects from the datasets, one could argue it could</span>
<span id="cb2-1257"><a href="#cb2-1257" aria-hidden="true" tabindex="-1"></a>also remove platform dependent normalization values and therefore one would need</span>
<span id="cb2-1258"><a href="#cb2-1258" aria-hidden="true" tabindex="-1"></a>to look only at lower principal components. In this section we show that doing</span>
<span id="cb2-1259"><a href="#cb2-1259" aria-hidden="true" tabindex="-1"></a>the normalization is necessary to get a good integration in the datasets.</span>
<span id="cb2-1260"><a href="#cb2-1260" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1261"><a href="#cb2-1261" aria-hidden="true" tabindex="-1"></a>The idea is similar as before, but instead of performing the *qPCR*-like</span>
<span id="cb2-1262"><a href="#cb2-1262" aria-hidden="true" tabindex="-1"></a>normalization on the datasets, one would use the logFPKM values for TCGA and</span>
<span id="cb2-1263"><a href="#cb2-1263" aria-hidden="true" tabindex="-1"></a>median intensity for microarray. The problem with this kind of approach also</span>
<span id="cb2-1264"><a href="#cb2-1264" aria-hidden="true" tabindex="-1"></a>includes the fact that only other samples that went through this processing</span>
<span id="cb2-1265"><a href="#cb2-1265" aria-hidden="true" tabindex="-1"></a>could be used in the pipeline.</span>
<span id="cb2-1266"><a href="#cb2-1266" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1269"><a href="#cb2-1269" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb2-1270"><a href="#cb2-1270" aria-hidden="true" tabindex="-1"></a>training_set <span class="ot">&lt;-</span> <span class="fu">mapply</span>( </span>
<span id="cb2-1271"><a href="#cb2-1271" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(sum_exp, i, genes_for_pca){</span>
<span id="cb2-1272"><a href="#cb2-1272" aria-hidden="true" tabindex="-1"></a>        <span class="fu">assay</span>(sum_exp[genes_for_pca, ], <span class="at">i =</span> i) <span class="sc">%&gt;%</span> </span>
<span id="cb2-1273"><a href="#cb2-1273" aria-hidden="true" tabindex="-1"></a>            <span class="fu">data.frame</span>(<span class="at">check.names =</span> <span class="cn">FALSE</span>)    </span>
<span id="cb2-1274"><a href="#cb2-1274" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb2-1275"><a href="#cb2-1275" aria-hidden="true" tabindex="-1"></a>    datasets_normalized[which_cohorts_training],</span>
<span id="cb2-1276"><a href="#cb2-1276" aria-hidden="true" tabindex="-1"></a>    <span class="at">i =</span> which_exp[which_cohorts_training],</span>
<span id="cb2-1277"><a href="#cb2-1277" aria-hidden="true" tabindex="-1"></a>    <span class="at">MoreArgs =</span> <span class="fu">list</span>(<span class="at">genes_for_pca =</span> genes_for_pca),</span>
<span id="cb2-1278"><a href="#cb2-1278" aria-hidden="true" tabindex="-1"></a>    <span class="at">SIMPLIFY =</span> <span class="cn">FALSE</span>,</span>
<span id="cb2-1279"><a href="#cb2-1279" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span></span>
<span id="cb2-1280"><a href="#cb2-1280" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> </span>
<span id="cb2-1281"><a href="#cb2-1281" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">bind_cols</span>() <span class="sc">%&gt;%</span></span>
<span id="cb2-1282"><a href="#cb2-1282" aria-hidden="true" tabindex="-1"></a>    .[, samples_for_training]</span>
<span id="cb2-1283"><a href="#cb2-1283" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1284"><a href="#cb2-1284" aria-hidden="true" tabindex="-1"></a>pca_fit_no_norm <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">pca</span>(</span>
<span id="cb2-1285"><a href="#cb2-1285" aria-hidden="true" tabindex="-1"></a>    training_set,</span>
<span id="cb2-1286"><a href="#cb2-1286" aria-hidden="true" tabindex="-1"></a>    <span class="at">metadata =</span> merged_col_data <span class="sc">%&gt;%</span> </span>
<span id="cb2-1287"><a href="#cb2-1287" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">filter</span>(sample_name <span class="sc">%in%</span> <span class="fu">colnames</span>(training_set)) <span class="sc">%&gt;%</span></span>
<span id="cb2-1288"><a href="#cb2-1288" aria-hidden="true" tabindex="-1"></a>        .[<span class="fu">colnames</span>(training_set), ],</span>
<span id="cb2-1289"><a href="#cb2-1289" aria-hidden="true" tabindex="-1"></a>    <span class="at">center =</span> <span class="cn">FALSE</span>,</span>
<span id="cb2-1290"><a href="#cb2-1290" aria-hidden="true" tabindex="-1"></a>    <span class="at">scale =</span> <span class="cn">FALSE</span></span>
<span id="cb2-1291"><a href="#cb2-1291" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-1292"><a href="#cb2-1292" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1293"><a href="#cb2-1293" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb2-1294"><a href="#cb2-1294" aria-hidden="true" tabindex="-1"></a>    pca_fit_no_norm,</span>
<span id="cb2-1295"><a href="#cb2-1295" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/pca_merging/pca_fit_no_norm.rds"</span></span>
<span id="cb2-1296"><a href="#cb2-1296" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-1297"><a href="#cb2-1297" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb2-1298"><a href="#cb2-1298" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1299"><a href="#cb2-1299" aria-hidden="true" tabindex="-1"></a><span class="in">```{r pca-no-norm, fig.width=20, fig.height=16}</span></span>
<span id="cb2-1300"><a href="#cb2-1300" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-pca-no-norm</span></span>
<span id="cb2-1301"><a href="#cb2-1301" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: PCA projections colored by different factors and organized</span></span>
<span id="cb2-1302"><a href="#cb2-1302" aria-hidden="true" tabindex="-1"></a><span class="co">#|    by different components. (A) Plot of the first two components colored</span></span>
<span id="cb2-1303"><a href="#cb2-1303" aria-hidden="true" tabindex="-1"></a><span class="co">#|    by cohort. (B) Plot of the first two components colored by ER status.</span></span>
<span id="cb2-1304"><a href="#cb2-1304" aria-hidden="true" tabindex="-1"></a><span class="co">#|    (C) Plot of the second and third components colored by cohort. </span></span>
<span id="cb2-1305"><a href="#cb2-1305" aria-hidden="true" tabindex="-1"></a><span class="co">#|    (D) Plot of the second and third components colored by ER status.</span></span>
<span id="cb2-1306"><a href="#cb2-1306" aria-hidden="true" tabindex="-1"></a><span class="co">#|    (E) Plot of the second and third components colored by PAM50.</span></span>
<span id="cb2-1307"><a href="#cb2-1307" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1308"><a href="#cb2-1308" aria-hidden="true" tabindex="-1"></a>plots_pca_fit <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb2-1309"><a href="#cb2-1309" aria-hidden="true" tabindex="-1"></a>point_size <span class="ot">&lt;-</span> <span class="dv">2</span></span>
<span id="cb2-1310"><a href="#cb2-1310" aria-hidden="true" tabindex="-1"></a>which_pca_fit <span class="ot">&lt;-</span> pca_fit_no_norm</span>
<span id="cb2-1311"><a href="#cb2-1311" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>pc1_pc2_cohort <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb2-1312"><a href="#cb2-1312" aria-hidden="true" tabindex="-1"></a>    pca_fit_no_norm,</span>
<span id="cb2-1313"><a href="#cb2-1313" aria-hidden="true" tabindex="-1"></a>    <span class="at">colby =</span> <span class="st">"cohort"</span>,</span>
<span id="cb2-1314"><a href="#cb2-1314" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> <span class="cn">NULL</span>, </span>
<span id="cb2-1315"><a href="#cb2-1315" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb2-1316"><a href="#cb2-1316" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC1"</span>,</span>
<span id="cb2-1317"><a href="#cb2-1317" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC2"</span>,</span>
<span id="cb2-1318"><a href="#cb2-1318" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"First two components colored by cohort"</span>,</span>
<span id="cb2-1319"><a href="#cb2-1319" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">"Only 1000 training samples, no qPCR normalization"</span>,</span>
<span id="cb2-1320"><a href="#cb2-1320" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> point_size</span>
<span id="cb2-1321"><a href="#cb2-1321" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-1322"><a href="#cb2-1322" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1323"><a href="#cb2-1323" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>pc1_pc2_er_status <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb2-1324"><a href="#cb2-1324" aria-hidden="true" tabindex="-1"></a>    pca_fit_no_norm,</span>
<span id="cb2-1325"><a href="#cb2-1325" aria-hidden="true" tabindex="-1"></a>    <span class="at">colby =</span> <span class="st">"er_status"</span>,</span>
<span id="cb2-1326"><a href="#cb2-1326" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> <span class="cn">NULL</span>, </span>
<span id="cb2-1327"><a href="#cb2-1327" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb2-1328"><a href="#cb2-1328" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC1"</span>,</span>
<span id="cb2-1329"><a href="#cb2-1329" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC2"</span>,</span>
<span id="cb2-1330"><a href="#cb2-1330" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"First two components colored by ER status"</span>,</span>
<span id="cb2-1331"><a href="#cb2-1331" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">"Only 1000 training samples"</span>,</span>
<span id="cb2-1332"><a href="#cb2-1332" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> point_size</span>
<span id="cb2-1333"><a href="#cb2-1333" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-1334"><a href="#cb2-1334" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1335"><a href="#cb2-1335" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>pc2_pc3_cohort <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb2-1336"><a href="#cb2-1336" aria-hidden="true" tabindex="-1"></a>    pca_fit_no_norm,</span>
<span id="cb2-1337"><a href="#cb2-1337" aria-hidden="true" tabindex="-1"></a>    <span class="at">colby =</span> <span class="st">"cohort"</span>,</span>
<span id="cb2-1338"><a href="#cb2-1338" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> <span class="cn">NULL</span>, </span>
<span id="cb2-1339"><a href="#cb2-1339" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb2-1340"><a href="#cb2-1340" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC2"</span>,</span>
<span id="cb2-1341"><a href="#cb2-1341" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC3"</span>,</span>
<span id="cb2-1342"><a href="#cb2-1342" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Second and third components colored by cohort"</span>,</span>
<span id="cb2-1343"><a href="#cb2-1343" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">"Only 1000 training samples"</span>,</span>
<span id="cb2-1344"><a href="#cb2-1344" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> point_size</span>
<span id="cb2-1345"><a href="#cb2-1345" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-1346"><a href="#cb2-1346" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1347"><a href="#cb2-1347" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>pc2_pc3_er_status <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb2-1348"><a href="#cb2-1348" aria-hidden="true" tabindex="-1"></a>    pca_fit_no_norm,</span>
<span id="cb2-1349"><a href="#cb2-1349" aria-hidden="true" tabindex="-1"></a>    <span class="at">colby =</span> <span class="st">"er_status"</span>,</span>
<span id="cb2-1350"><a href="#cb2-1350" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> <span class="cn">NULL</span>, </span>
<span id="cb2-1351"><a href="#cb2-1351" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb2-1352"><a href="#cb2-1352" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC2"</span>,</span>
<span id="cb2-1353"><a href="#cb2-1353" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC3"</span>,</span>
<span id="cb2-1354"><a href="#cb2-1354" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Second and third components colored by ER status"</span>,</span>
<span id="cb2-1355"><a href="#cb2-1355" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">"Only 1000 training samples"</span>,</span>
<span id="cb2-1356"><a href="#cb2-1356" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> point_size</span>
<span id="cb2-1357"><a href="#cb2-1357" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-1358"><a href="#cb2-1358" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1359"><a href="#cb2-1359" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>pc2_pc3_pam50 <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb2-1360"><a href="#cb2-1360" aria-hidden="true" tabindex="-1"></a>    pca_fit_no_norm,</span>
<span id="cb2-1361"><a href="#cb2-1361" aria-hidden="true" tabindex="-1"></a>    <span class="at">colby =</span> <span class="st">"pam50"</span>,</span>
<span id="cb2-1362"><a href="#cb2-1362" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> <span class="cn">NULL</span>, </span>
<span id="cb2-1363"><a href="#cb2-1363" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb2-1364"><a href="#cb2-1364" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC2"</span>,</span>
<span id="cb2-1365"><a href="#cb2-1365" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC3"</span>,</span>
<span id="cb2-1366"><a href="#cb2-1366" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Second and third components colored by PAM50"</span>,</span>
<span id="cb2-1367"><a href="#cb2-1367" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">"Only 1000 training samples"</span>,</span>
<span id="cb2-1368"><a href="#cb2-1368" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> point_size</span>
<span id="cb2-1369"><a href="#cb2-1369" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-1370"><a href="#cb2-1370" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1371"><a href="#cb2-1371" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>pc3_pc4_pam50 <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb2-1372"><a href="#cb2-1372" aria-hidden="true" tabindex="-1"></a>    pca_fit_no_norm,</span>
<span id="cb2-1373"><a href="#cb2-1373" aria-hidden="true" tabindex="-1"></a>    <span class="at">colby =</span> <span class="st">"pam50"</span>,</span>
<span id="cb2-1374"><a href="#cb2-1374" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> <span class="cn">NULL</span>, </span>
<span id="cb2-1375"><a href="#cb2-1375" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb2-1376"><a href="#cb2-1376" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC3"</span>,</span>
<span id="cb2-1377"><a href="#cb2-1377" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC4"</span>,</span>
<span id="cb2-1378"><a href="#cb2-1378" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Third and fourth components colored by PAM50"</span>,</span>
<span id="cb2-1379"><a href="#cb2-1379" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">"Only 1000 training samples"</span>,</span>
<span id="cb2-1380"><a href="#cb2-1380" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> point_size</span>
<span id="cb2-1381"><a href="#cb2-1381" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-1382"><a href="#cb2-1382" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1383"><a href="#cb2-1383" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>pc3_pc4_cohort <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb2-1384"><a href="#cb2-1384" aria-hidden="true" tabindex="-1"></a>    pca_fit_no_norm,</span>
<span id="cb2-1385"><a href="#cb2-1385" aria-hidden="true" tabindex="-1"></a>    <span class="at">colby =</span> <span class="st">"cohort"</span>,</span>
<span id="cb2-1386"><a href="#cb2-1386" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> <span class="cn">NULL</span>, </span>
<span id="cb2-1387"><a href="#cb2-1387" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb2-1388"><a href="#cb2-1388" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC3"</span>,</span>
<span id="cb2-1389"><a href="#cb2-1389" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC4"</span>,</span>
<span id="cb2-1390"><a href="#cb2-1390" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Third and fourth components colored by cohort"</span>,</span>
<span id="cb2-1391"><a href="#cb2-1391" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">"Only 1000 training samples"</span>,</span>
<span id="cb2-1392"><a href="#cb2-1392" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> point_size</span>
<span id="cb2-1393"><a href="#cb2-1393" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-1394"><a href="#cb2-1394" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1395"><a href="#cb2-1395" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(<span class="at">plotlist =</span> plots_pca_fit, <span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">labels =</span> <span class="st">"AUTO"</span>)</span>
<span id="cb2-1396"><a href="#cb2-1396" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb2-1397"><a href="#cb2-1397" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1398"><a href="#cb2-1398" aria-hidden="true" tabindex="-1"></a>@fig-pca-no-norm shows the contrast with the other method including the</span>
<span id="cb2-1399"><a href="#cb2-1399" aria-hidden="true" tabindex="-1"></a>normalization. Without the normalization, the samples from METABRIC tend to</span>
<span id="cb2-1400"><a href="#cb2-1400" aria-hidden="true" tabindex="-1"></a>concentrate close to the center, they are not so well mixed. </span>
<span id="cb2-1401"><a href="#cb2-1401" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1404"><a href="#cb2-1404" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb2-1405"><a href="#cb2-1405" aria-hidden="true" tabindex="-1"></a>datasets_pca_coordinates_no_norm <span class="ot">&lt;-</span> <span class="fu">mapply</span>(</span>
<span id="cb2-1406"><a href="#cb2-1406" aria-hidden="true" tabindex="-1"></a>    get_pca_coordinates,</span>
<span id="cb2-1407"><a href="#cb2-1407" aria-hidden="true" tabindex="-1"></a>    datasets_normalized,</span>
<span id="cb2-1408"><a href="#cb2-1408" aria-hidden="true" tabindex="-1"></a>    <span class="at">which_exp =</span> which_exp,</span>
<span id="cb2-1409"><a href="#cb2-1409" aria-hidden="true" tabindex="-1"></a>    <span class="at">MoreArgs =</span> <span class="fu">list</span>(<span class="at">pca_fit =</span> pca_fit_no_norm)</span>
<span id="cb2-1410"><a href="#cb2-1410" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-1411"><a href="#cb2-1411" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1412"><a href="#cb2-1412" aria-hidden="true" tabindex="-1"></a>df_pca_coordinates_og_no_norm <span class="ot">&lt;-</span> datasets_pca_coordinates_no_norm <span class="sc">%&gt;%</span> </span>
<span id="cb2-1413"><a href="#cb2-1413" aria-hidden="true" tabindex="-1"></a>    <span class="fu">do.call</span>(rbind, .) <span class="sc">%&gt;%</span></span>
<span id="cb2-1414"><a href="#cb2-1414" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span> </span>
<span id="cb2-1415"><a href="#cb2-1415" aria-hidden="true" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"sample_name"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb2-1416"><a href="#cb2-1416" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">inner_join</span>(</span>
<span id="cb2-1417"><a href="#cb2-1417" aria-hidden="true" tabindex="-1"></a>        .,</span>
<span id="cb2-1418"><a href="#cb2-1418" aria-hidden="true" tabindex="-1"></a>        merged_col_data,</span>
<span id="cb2-1419"><a href="#cb2-1419" aria-hidden="true" tabindex="-1"></a>        <span class="at">by =</span> <span class="st">"sample_name"</span></span>
<span id="cb2-1420"><a href="#cb2-1420" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb2-1421"><a href="#cb2-1421" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1422"><a href="#cb2-1422" aria-hidden="true" tabindex="-1"></a>plot_corrected <span class="ot">&lt;-</span> <span class="fu">plot_pca_coordinates</span>(</span>
<span id="cb2-1423"><a href="#cb2-1423" aria-hidden="true" tabindex="-1"></a>    df_pca_coordinates <span class="sc">%&gt;%</span> </span>
<span id="cb2-1424"><a href="#cb2-1424" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"tcga"</span>, <span class="st">"metabric"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb2-1425"><a href="#cb2-1425" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">slice</span>(<span class="fu">sample</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">n</span>())),</span>
<span id="cb2-1426"><a href="#cb2-1426" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"cohort"</span>,</span>
<span id="cb2-1427"><a href="#cb2-1427" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC3"</span>,</span>
<span id="cb2-1428"><a href="#cb2-1428" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC4"</span>,</span>
<span id="cb2-1429"><a href="#cb2-1429" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">1</span>,</span>
<span id="cb2-1430"><a href="#cb2-1430" aria-hidden="true" tabindex="-1"></a>    <span class="at">base_size =</span> <span class="dv">10</span>,</span>
<span id="cb2-1431"><a href="#cb2-1431" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="fu">paste0</span>(</span>
<span id="cb2-1432"><a href="#cb2-1432" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Embedding of all samples from TCGA"</span>,</span>
<span id="cb2-1433"><a href="#cb2-1433" aria-hidden="true" tabindex="-1"></a>        <span class="st">" and METABRIC</span><span class="sc">\n</span><span class="st">"</span>,</span>
<span id="cb2-1434"><a href="#cb2-1434" aria-hidden="true" tabindex="-1"></a>        <span class="st">"qPCR normalization performed in the samples"</span></span>
<span id="cb2-1435"><a href="#cb2-1435" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb2-1436"><a href="#cb2-1436" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-1437"><a href="#cb2-1437" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1438"><a href="#cb2-1438" aria-hidden="true" tabindex="-1"></a>plot_no_norm <span class="ot">&lt;-</span> <span class="fu">plot_pca_coordinates</span>(</span>
<span id="cb2-1439"><a href="#cb2-1439" aria-hidden="true" tabindex="-1"></a>    df_pca_coordinates_og_no_norm <span class="sc">%&gt;%</span> </span>
<span id="cb2-1440"><a href="#cb2-1440" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"tcga"</span>, <span class="st">"metabric"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb2-1441"><a href="#cb2-1441" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">slice</span>(<span class="fu">sample</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">n</span>())),</span>
<span id="cb2-1442"><a href="#cb2-1442" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"cohort"</span>,</span>
<span id="cb2-1443"><a href="#cb2-1443" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC3"</span>,</span>
<span id="cb2-1444"><a href="#cb2-1444" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC4"</span>,</span>
<span id="cb2-1445"><a href="#cb2-1445" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">1</span>,</span>
<span id="cb2-1446"><a href="#cb2-1446" aria-hidden="true" tabindex="-1"></a>    <span class="at">base_size =</span> <span class="dv">10</span>,</span>
<span id="cb2-1447"><a href="#cb2-1447" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="fu">paste0</span>(</span>
<span id="cb2-1448"><a href="#cb2-1448" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Embedding of all samples from TCGA"</span>,</span>
<span id="cb2-1449"><a href="#cb2-1449" aria-hidden="true" tabindex="-1"></a>        <span class="st">" and METABRIC</span><span class="sc">\n</span><span class="st">"</span>,</span>
<span id="cb2-1450"><a href="#cb2-1450" aria-hidden="true" tabindex="-1"></a>        <span class="st">"No qPCR normalization performed in the samples"</span></span>
<span id="cb2-1451"><a href="#cb2-1451" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb2-1452"><a href="#cb2-1452" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-1453"><a href="#cb2-1453" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb2-1454"><a href="#cb2-1454" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1455"><a href="#cb2-1455" aria-hidden="true" tabindex="-1"></a><span class="in">```{r pca-norm-no-norm, fig.width=10, fig.height=4}</span></span>
<span id="cb2-1456"><a href="#cb2-1456" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-pca-norm-no-norm</span></span>
<span id="cb2-1457"><a href="#cb2-1457" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: PC3 and PC4 from all METABRIC and TCGA samples. (A) Embedding</span></span>
<span id="cb2-1458"><a href="#cb2-1458" aria-hidden="true" tabindex="-1"></a><span class="co">#|     using the qPCR-like normalization. (B) Embedding using the original</span></span>
<span id="cb2-1459"><a href="#cb2-1459" aria-hidden="true" tabindex="-1"></a><span class="co">#|     normalization from each dataset.</span></span>
<span id="cb2-1460"><a href="#cb2-1460" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1461"><a href="#cb2-1461" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(</span>
<span id="cb2-1462"><a href="#cb2-1462" aria-hidden="true" tabindex="-1"></a>    plot_corrected,</span>
<span id="cb2-1463"><a href="#cb2-1463" aria-hidden="true" tabindex="-1"></a>    plot_no_norm</span>
<span id="cb2-1464"><a href="#cb2-1464" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-1465"><a href="#cb2-1465" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb2-1466"><a href="#cb2-1466" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1467"><a href="#cb2-1467" aria-hidden="true" tabindex="-1"></a>@fig-pca-norm-no-norm shows side by side the differences when using the</span>
<span id="cb2-1468"><a href="#cb2-1468" aria-hidden="true" tabindex="-1"></a>qPCR-like normalization and when not using. The left plot is using the qPCR</span>
<span id="cb2-1469"><a href="#cb2-1469" aria-hidden="true" tabindex="-1"></a>normalization and shows how well the mixing gets, putting samples in the same</span>
<span id="cb2-1470"><a href="#cb2-1470" aria-hidden="true" tabindex="-1"></a>scale. If the scaling in a sample level is not performed, samples do not mix</span>
<span id="cb2-1471"><a href="#cb2-1471" aria-hidden="true" tabindex="-1"></a>well.</span>
<span id="cb2-1472"><a href="#cb2-1472" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1473"><a href="#cb2-1473" aria-hidden="true" tabindex="-1"></a><span class="fu">## Molecular landscape is characterized by several pathways</span></span>
<span id="cb2-1474"><a href="#cb2-1474" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1475"><a href="#cb2-1475" aria-hidden="true" tabindex="-1"></a>Among the 1000 genes that are used when calculating the molecular landscape,</span>
<span id="cb2-1476"><a href="#cb2-1476" aria-hidden="true" tabindex="-1"></a>several might be involved in biological process. @fig-pca-mol-land-pathways</span>
<span id="cb2-1477"><a href="#cb2-1477" aria-hidden="true" tabindex="-1"></a>shows that several pathways are associated with the position in the </span>
<span id="cb2-1478"><a href="#cb2-1478" aria-hidden="true" tabindex="-1"></a>molecular landscape. Something that is particularly remarkable here is the </span>
<span id="cb2-1479"><a href="#cb2-1479" aria-hidden="true" tabindex="-1"></a>association of the epithelial mesenchymal transition and DNA repair</span>
<span id="cb2-1480"><a href="#cb2-1480" aria-hidden="true" tabindex="-1"></a>pathways with PC4. </span>
<span id="cb2-1481"><a href="#cb2-1481" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1482"><a href="#cb2-1482" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width = 6, fig.height=20}</span></span>
<span id="cb2-1483"><a href="#cb2-1483" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-pca-mol-land-pathways</span></span>
<span id="cb2-1484"><a href="#cb2-1484" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Hex biplots colored by different pathways. Each hex is colored</span></span>
<span id="cb2-1485"><a href="#cb2-1485" aria-hidden="true" tabindex="-1"></a><span class="co">#|     based on the average score in the hex.</span></span>
<span id="cb2-1486"><a href="#cb2-1486" aria-hidden="true" tabindex="-1"></a>plots_pathways <span class="ot">&lt;-</span> <span class="fu">sapply</span>(</span>
<span id="cb2-1487"><a href="#cb2-1487" aria-hidden="true" tabindex="-1"></a>    <span class="fu">c</span>(</span>
<span id="cb2-1488"><a href="#cb2-1488" aria-hidden="true" tabindex="-1"></a>        <span class="st">"HALLMARK_ESTROGEN_RESPONSE_EARLY"</span>, </span>
<span id="cb2-1489"><a href="#cb2-1489" aria-hidden="true" tabindex="-1"></a>        <span class="st">"HALLMARK_G2M_CHECKPOINT"</span>,</span>
<span id="cb2-1490"><a href="#cb2-1490" aria-hidden="true" tabindex="-1"></a>        <span class="st">"HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION"</span>,</span>
<span id="cb2-1491"><a href="#cb2-1491" aria-hidden="true" tabindex="-1"></a>        <span class="st">"SET_ERPR"</span>,</span>
<span id="cb2-1492"><a href="#cb2-1492" aria-hidden="true" tabindex="-1"></a>        <span class="st">"HALLMARK_DNA_REPAIR"</span>,</span>
<span id="cb2-1493"><a href="#cb2-1493" aria-hidden="true" tabindex="-1"></a>        <span class="st">"HALLMARK_ANDROGEN_RESPONSE"</span>,</span>
<span id="cb2-1494"><a href="#cb2-1494" aria-hidden="true" tabindex="-1"></a>        <span class="st">"HALLMARK_PI3K_AKT_MTOR_SIGNALING"</span>,</span>
<span id="cb2-1495"><a href="#cb2-1495" aria-hidden="true" tabindex="-1"></a>        <span class="st">"random_200"</span>,</span>
<span id="cb2-1496"><a href="#cb2-1496" aria-hidden="true" tabindex="-1"></a>        <span class="st">"random_18"</span></span>
<span id="cb2-1497"><a href="#cb2-1497" aria-hidden="true" tabindex="-1"></a>    ), </span>
<span id="cb2-1498"><a href="#cb2-1498" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(pathway, df_pca_coordinates){</span>
<span id="cb2-1499"><a href="#cb2-1499" aria-hidden="true" tabindex="-1"></a>        color <span class="ot">&lt;-</span> pathway</span>
<span id="cb2-1500"><a href="#cb2-1500" aria-hidden="true" tabindex="-1"></a>        x <span class="ot">&lt;-</span> <span class="st">"PC3"</span></span>
<span id="cb2-1501"><a href="#cb2-1501" aria-hidden="true" tabindex="-1"></a>        y <span class="ot">&lt;-</span> <span class="st">"PC4"</span></span>
<span id="cb2-1502"><a href="#cb2-1502" aria-hidden="true" tabindex="-1"></a>        size <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb2-1503"><a href="#cb2-1503" aria-hidden="true" tabindex="-1"></a>        base_size <span class="ot">&lt;-</span> <span class="dv">10</span></span>
<span id="cb2-1504"><a href="#cb2-1504" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb2-1505"><a href="#cb2-1505" aria-hidden="true" tabindex="-1"></a>        p1 <span class="ot">&lt;-</span> df_pca_coordinates <span class="sc">%&gt;%</span> </span>
<span id="cb2-1506"><a href="#cb2-1506" aria-hidden="true" tabindex="-1"></a>            ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes_string</span>(<span class="at">x =</span> x, <span class="at">y =</span> y, <span class="at">z =</span> color)) <span class="sc">+</span> </span>
<span id="cb2-1507"><a href="#cb2-1507" aria-hidden="true" tabindex="-1"></a>                ggplot2<span class="sc">::</span><span class="fu">stat_summary_hex</span>(<span class="at">bins =</span> <span class="dv">35</span>) <span class="sc">+</span></span>
<span id="cb2-1508"><a href="#cb2-1508" aria-hidden="true" tabindex="-1"></a>                ggplot2<span class="sc">::</span><span class="fu">scale_fill_viridis_c</span>() <span class="sc">+</span></span>
<span id="cb2-1509"><a href="#cb2-1509" aria-hidden="true" tabindex="-1"></a>                ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>cohort) <span class="sc">+</span> </span>
<span id="cb2-1510"><a href="#cb2-1510" aria-hidden="true" tabindex="-1"></a>                ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb2-1511"><a href="#cb2-1511" aria-hidden="true" tabindex="-1"></a>                    <span class="at">title =</span> color,</span>
<span id="cb2-1512"><a href="#cb2-1512" aria-hidden="true" tabindex="-1"></a>                    <span class="at">fill =</span> <span class="st">"Score"</span></span>
<span id="cb2-1513"><a href="#cb2-1513" aria-hidden="true" tabindex="-1"></a>                ) <span class="sc">+</span> </span>
<span id="cb2-1514"><a href="#cb2-1514" aria-hidden="true" tabindex="-1"></a>                ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> base_size)</span>
<span id="cb2-1515"><a href="#cb2-1515" aria-hidden="true" tabindex="-1"></a>    }, </span>
<span id="cb2-1516"><a href="#cb2-1516" aria-hidden="true" tabindex="-1"></a>    <span class="at">df_pca_coordinates =</span> df_pca_coordinates <span class="sc">%&gt;%</span> </span>
<span id="cb2-1517"><a href="#cb2-1517" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"tcga"</span>, <span class="st">"metabric"</span>, <span class="st">"scanb"</span>)),</span>
<span id="cb2-1518"><a href="#cb2-1518" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>, </span>
<span id="cb2-1519"><a href="#cb2-1519" aria-hidden="true" tabindex="-1"></a>    <span class="at">simplify =</span> <span class="cn">FALSE</span></span>
<span id="cb2-1520"><a href="#cb2-1520" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-1521"><a href="#cb2-1521" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1522"><a href="#cb2-1522" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(</span>
<span id="cb2-1523"><a href="#cb2-1523" aria-hidden="true" tabindex="-1"></a>    <span class="at">plotlist =</span> plots_pathways,</span>
<span id="cb2-1524"><a href="#cb2-1524" aria-hidden="true" tabindex="-1"></a>    <span class="at">ncol =</span> <span class="dv">1</span></span>
<span id="cb2-1525"><a href="#cb2-1525" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-1526"><a href="#cb2-1526" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb2-1527"><a href="#cb2-1527" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1528"><a href="#cb2-1528" aria-hidden="true" tabindex="-1"></a>We see that these pathways are somehow relevant to the embedding, but are they</span>
<span id="cb2-1529"><a href="#cb2-1529" aria-hidden="true" tabindex="-1"></a>also prognostic of overall survival in the two cohorts? In the next chunks</span>
<span id="cb2-1530"><a href="#cb2-1530" aria-hidden="true" tabindex="-1"></a>we evaluate this hypothesis. Mostly the focus is on ER+ BC patients that</span>
<span id="cb2-1531"><a href="#cb2-1531" aria-hidden="true" tabindex="-1"></a>received some kind of hormonal therapy. As seen before, for the TCGA</span>
<span id="cb2-1532"><a href="#cb2-1532" aria-hidden="true" tabindex="-1"></a>dataset we select ER+ BC patients that have the molecular subtype luminal</span>
<span id="cb2-1533"><a href="#cb2-1533" aria-hidden="true" tabindex="-1"></a>A and B instead. </span>
<span id="cb2-1534"><a href="#cb2-1534" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1535"><a href="#cb2-1535" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb2-1536"><a href="#cb2-1536" aria-hidden="true" tabindex="-1"></a>tcga_samples <span class="ot">&lt;-</span> <span class="fu">colData</span>(datasets<span class="sc">$</span>tcga) <span class="sc">%&gt;%</span> data.frame <span class="sc">%&gt;%</span></span>
<span id="cb2-1537"><a href="#cb2-1537" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb2-1538"><a href="#cb2-1538" aria-hidden="true" tabindex="-1"></a>        pam50 <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"luma"</span>, <span class="st">"lumb"</span>) <span class="sc">&amp;</span></span>
<span id="cb2-1539"><a href="#cb2-1539" aria-hidden="true" tabindex="-1"></a>        node_stage <span class="sc">!=</span> <span class="st">"NX"</span> <span class="sc">&amp;</span></span>
<span id="cb2-1540"><a href="#cb2-1540" aria-hidden="true" tabindex="-1"></a>        <span class="sc">!</span>(tumor_stage <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"stage_iv"</span>, <span class="st">"na"</span>)) <span class="sc">&amp;</span> </span>
<span id="cb2-1541"><a href="#cb2-1541" aria-hidden="true" tabindex="-1"></a>        er_status <span class="sc">==</span> <span class="st">"pos"</span></span>
<span id="cb2-1542"><a href="#cb2-1542" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span> </span>
<span id="cb2-1543"><a href="#cb2-1543" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">pull</span>(sample_name)</span>
<span id="cb2-1544"><a href="#cb2-1544" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1545"><a href="#cb2-1545" aria-hidden="true" tabindex="-1"></a>tcga_samples <span class="ot">&lt;-</span> df_pca_coordinates <span class="sc">%&gt;%</span> </span>
<span id="cb2-1546"><a href="#cb2-1546" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(sample_name <span class="sc">%in%</span> tcga_samples <span class="sc">&amp;</span> PC3 <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb2-1547"><a href="#cb2-1547" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">pull</span>(sample_name)</span>
<span id="cb2-1548"><a href="#cb2-1548" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1549"><a href="#cb2-1549" aria-hidden="true" tabindex="-1"></a>scanb_samples <span class="ot">&lt;-</span> <span class="fu">colData</span>(datasets<span class="sc">$</span>scanb) <span class="sc">%&gt;%</span> data.frame <span class="sc">%&gt;%</span></span>
<span id="cb2-1550"><a href="#cb2-1550" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb2-1551"><a href="#cb2-1551" aria-hidden="true" tabindex="-1"></a>        node_stage <span class="sc">!=</span> <span class="st">"NX"</span> <span class="sc">&amp;</span></span>
<span id="cb2-1552"><a href="#cb2-1552" aria-hidden="true" tabindex="-1"></a>        <span class="sc">!</span>(tumor_stage <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"T4"</span>, <span class="st">"Tis"</span>, <span class="st">"TX"</span>)) <span class="sc">&amp;</span> </span>
<span id="cb2-1553"><a href="#cb2-1553" aria-hidden="true" tabindex="-1"></a>        er_status <span class="sc">==</span> <span class="st">"pos"</span> <span class="sc">&amp;</span> </span>
<span id="cb2-1554"><a href="#cb2-1554" aria-hidden="true" tabindex="-1"></a>        TreatGroup <span class="sc">==</span> <span class="st">"Endo"</span></span>
<span id="cb2-1555"><a href="#cb2-1555" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span> </span>
<span id="cb2-1556"><a href="#cb2-1556" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">pull</span>(sample_name)</span>
<span id="cb2-1557"><a href="#cb2-1557" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1558"><a href="#cb2-1558" aria-hidden="true" tabindex="-1"></a>scanb_samples <span class="ot">&lt;-</span> df_pca_coordinates <span class="sc">%&gt;%</span> </span>
<span id="cb2-1559"><a href="#cb2-1559" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(sample_name <span class="sc">%in%</span> scanb_samples <span class="sc">&amp;</span> PC3 <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb2-1560"><a href="#cb2-1560" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">pull</span>(sample_name)</span>
<span id="cb2-1561"><a href="#cb2-1561" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1562"><a href="#cb2-1562" aria-hidden="true" tabindex="-1"></a>metabric_samples <span class="ot">&lt;-</span> <span class="fu">colData</span>(datasets<span class="sc">$</span>metabric) <span class="sc">%&gt;%</span> data.frame <span class="sc">%&gt;%</span></span>
<span id="cb2-1563"><a href="#cb2-1563" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb2-1564"><a href="#cb2-1564" aria-hidden="true" tabindex="-1"></a>        CHEMOTHERAPY <span class="sc">==</span> <span class="st">"NO"</span> <span class="sc">&amp;</span></span>
<span id="cb2-1565"><a href="#cb2-1565" aria-hidden="true" tabindex="-1"></a>        er_status <span class="sc">==</span> <span class="st">"pos"</span> <span class="sc">&amp;</span> </span>
<span id="cb2-1566"><a href="#cb2-1566" aria-hidden="true" tabindex="-1"></a>        HORMONE_THERAPY <span class="sc">==</span> <span class="st">"YES"</span></span>
<span id="cb2-1567"><a href="#cb2-1567" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span> </span>
<span id="cb2-1568"><a href="#cb2-1568" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">pull</span>(sample_name)</span>
<span id="cb2-1569"><a href="#cb2-1569" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1570"><a href="#cb2-1570" aria-hidden="true" tabindex="-1"></a>metabric_samples <span class="ot">&lt;-</span> df_pca_coordinates <span class="sc">%&gt;%</span> </span>
<span id="cb2-1571"><a href="#cb2-1571" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(sample_name <span class="sc">%in%</span> metabric_samples <span class="sc">&amp;</span> PC3 <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb2-1572"><a href="#cb2-1572" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">pull</span>(sample_name)</span>
<span id="cb2-1573"><a href="#cb2-1573" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1574"><a href="#cb2-1574" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1575"><a href="#cb2-1575" aria-hidden="true" tabindex="-1"></a>formulas_survival <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb2-1576"><a href="#cb2-1576" aria-hidden="true" tabindex="-1"></a>    <span class="st">"os_tcga"</span> <span class="ot">=</span> <span class="st">"Surv(os_months, os_status) ~ age + node_stage + tumor_stage"</span>,</span>
<span id="cb2-1577"><a href="#cb2-1577" aria-hidden="true" tabindex="-1"></a>    <span class="st">"os_scanb"</span> <span class="ot">=</span> <span class="st">"Surv(os_months, os_status) ~ age + node_stage + tumor_stage"</span>,</span>
<span id="cb2-1578"><a href="#cb2-1578" aria-hidden="true" tabindex="-1"></a>    <span class="st">"os_metabric"</span> <span class="ot">=</span> <span class="st">"Surv(os_months, os_status) ~ age + npi"</span>,</span>
<span id="cb2-1579"><a href="#cb2-1579" aria-hidden="true" tabindex="-1"></a>    <span class="st">"rfs_metabric"</span> <span class="ot">=</span> <span class="st">"Surv(rfs_months, rfs_status) ~ age + npi"</span>,</span>
<span id="cb2-1580"><a href="#cb2-1580" aria-hidden="true" tabindex="-1"></a>    <span class="st">"rfs_scanb"</span> <span class="ot">=</span> <span class="st">"Surv(rfs_months, rfs_status) ~ age + node_stage + tumor_stage"</span></span>
<span id="cb2-1581"><a href="#cb2-1581" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-1582"><a href="#cb2-1582" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1583"><a href="#cb2-1583" aria-hidden="true" tabindex="-1"></a>type_survival <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"os"</span>, <span class="st">"rfs"</span>)</span>
<span id="cb2-1584"><a href="#cb2-1584" aria-hidden="true" tabindex="-1"></a>which_scores <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb2-1585"><a href="#cb2-1585" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_G2M_CHECKPOINT"</span>,</span>
<span id="cb2-1586"><a href="#cb2-1586" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION"</span>,</span>
<span id="cb2-1587"><a href="#cb2-1587" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_DNA_REPAIR"</span>,</span>
<span id="cb2-1588"><a href="#cb2-1588" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_ANDROGEN_RESPONSE"</span>,</span>
<span id="cb2-1589"><a href="#cb2-1589" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_PI3K_AKT_MTOR_SIGNALING"</span></span>
<span id="cb2-1590"><a href="#cb2-1590" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-1591"><a href="#cb2-1591" aria-hidden="true" tabindex="-1"></a>survival_results <span class="ot">&lt;-</span> <span class="fu">sapply</span>(</span>
<span id="cb2-1592"><a href="#cb2-1592" aria-hidden="true" tabindex="-1"></a>    <span class="fu">c</span>(<span class="st">"os"</span>, <span class="st">"rfs"</span>),</span>
<span id="cb2-1593"><a href="#cb2-1593" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(type_analysis, datasets, name_scores, formulas){</span>
<span id="cb2-1594"><a href="#cb2-1594" aria-hidden="true" tabindex="-1"></a>        <span class="fu">mapply</span>(</span>
<span id="cb2-1595"><a href="#cb2-1595" aria-hidden="true" tabindex="-1"></a>            <span class="cf">function</span>(</span>
<span id="cb2-1596"><a href="#cb2-1596" aria-hidden="true" tabindex="-1"></a>                dataset, name_dataset, name_scores, formulas, type_analysis</span>
<span id="cb2-1597"><a href="#cb2-1597" aria-hidden="true" tabindex="-1"></a>            ){</span>
<span id="cb2-1598"><a href="#cb2-1598" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb2-1599"><a href="#cb2-1599" aria-hidden="true" tabindex="-1"></a>                <span class="cf">if</span> (type_analysis <span class="sc">==</span> <span class="st">"rfs"</span> <span class="sc">&amp;</span> name_dataset <span class="sc">==</span> <span class="st">"tcga"</span>){</span>
<span id="cb2-1600"><a href="#cb2-1600" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">return</span>()</span>
<span id="cb2-1601"><a href="#cb2-1601" aria-hidden="true" tabindex="-1"></a>                } <span class="cf">else</span> { </span>
<span id="cb2-1602"><a href="#cb2-1602" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb2-1603"><a href="#cb2-1603" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">sapply</span>(</span>
<span id="cb2-1604"><a href="#cb2-1604" aria-hidden="true" tabindex="-1"></a>                        name_scores,</span>
<span id="cb2-1605"><a href="#cb2-1605" aria-hidden="true" tabindex="-1"></a>                        <span class="cf">function</span>(name_score, col_data, formula){</span>
<span id="cb2-1606"><a href="#cb2-1606" aria-hidden="true" tabindex="-1"></a>                            </span>
<span id="cb2-1607"><a href="#cb2-1607" aria-hidden="true" tabindex="-1"></a>                            survival<span class="sc">::</span><span class="fu">coxph</span>(</span>
<span id="cb2-1608"><a href="#cb2-1608" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">as.formula</span>(</span>
<span id="cb2-1609"><a href="#cb2-1609" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">paste</span>(formula, name_score, <span class="at">sep =</span> <span class="st">"+"</span>)</span>
<span id="cb2-1610"><a href="#cb2-1610" aria-hidden="true" tabindex="-1"></a>                                ),</span>
<span id="cb2-1611"><a href="#cb2-1611" aria-hidden="true" tabindex="-1"></a>                                <span class="at">data =</span> col_data, </span>
<span id="cb2-1612"><a href="#cb2-1612" aria-hidden="true" tabindex="-1"></a>                                <span class="at">y =</span> <span class="cn">FALSE</span>,</span>
<span id="cb2-1613"><a href="#cb2-1613" aria-hidden="true" tabindex="-1"></a>                                <span class="at">x =</span> <span class="cn">FALSE</span></span>
<span id="cb2-1614"><a href="#cb2-1614" aria-hidden="true" tabindex="-1"></a>                            )</span>
<span id="cb2-1615"><a href="#cb2-1615" aria-hidden="true" tabindex="-1"></a>                        },</span>
<span id="cb2-1616"><a href="#cb2-1616" aria-hidden="true" tabindex="-1"></a>                        <span class="at">col_data =</span> <span class="fu">colData</span>(dataset),</span>
<span id="cb2-1617"><a href="#cb2-1617" aria-hidden="true" tabindex="-1"></a>                        <span class="at">formula =</span> formulas[</span>
<span id="cb2-1618"><a href="#cb2-1618" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">grepl</span>(</span>
<span id="cb2-1619"><a href="#cb2-1619" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">paste</span>(type_analysis, name_dataset, <span class="at">sep =</span> <span class="st">"_"</span>),</span>
<span id="cb2-1620"><a href="#cb2-1620" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">names</span>(formulas)</span>
<span id="cb2-1621"><a href="#cb2-1621" aria-hidden="true" tabindex="-1"></a>                            )</span>
<span id="cb2-1622"><a href="#cb2-1622" aria-hidden="true" tabindex="-1"></a>                        ],</span>
<span id="cb2-1623"><a href="#cb2-1623" aria-hidden="true" tabindex="-1"></a>                        <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>,</span>
<span id="cb2-1624"><a href="#cb2-1624" aria-hidden="true" tabindex="-1"></a>                        <span class="at">simplify =</span> <span class="cn">FALSE</span></span>
<span id="cb2-1625"><a href="#cb2-1625" aria-hidden="true" tabindex="-1"></a>                    )</span>
<span id="cb2-1626"><a href="#cb2-1626" aria-hidden="true" tabindex="-1"></a>                }</span>
<span id="cb2-1627"><a href="#cb2-1627" aria-hidden="true" tabindex="-1"></a>            },</span>
<span id="cb2-1628"><a href="#cb2-1628" aria-hidden="true" tabindex="-1"></a>            <span class="at">dataset =</span> datasets,</span>
<span id="cb2-1629"><a href="#cb2-1629" aria-hidden="true" tabindex="-1"></a>            <span class="at">name_dataset =</span> <span class="fu">names</span>(datasets),</span>
<span id="cb2-1630"><a href="#cb2-1630" aria-hidden="true" tabindex="-1"></a>            <span class="at">MoreArgs =</span> <span class="fu">list</span>(</span>
<span id="cb2-1631"><a href="#cb2-1631" aria-hidden="true" tabindex="-1"></a>                <span class="at">formulas =</span> formulas, <span class="at">name_scores =</span> name_scores,</span>
<span id="cb2-1632"><a href="#cb2-1632" aria-hidden="true" tabindex="-1"></a>                <span class="at">type_analysis =</span> type_analysis</span>
<span id="cb2-1633"><a href="#cb2-1633" aria-hidden="true" tabindex="-1"></a>            ),</span>
<span id="cb2-1634"><a href="#cb2-1634" aria-hidden="true" tabindex="-1"></a>            <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>,</span>
<span id="cb2-1635"><a href="#cb2-1635" aria-hidden="true" tabindex="-1"></a>            <span class="at">SIMPLIFY =</span> <span class="cn">FALSE</span></span>
<span id="cb2-1636"><a href="#cb2-1636" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb2-1637"><a href="#cb2-1637" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb2-1638"><a href="#cb2-1638" aria-hidden="true" tabindex="-1"></a>    <span class="at">datasets =</span> <span class="fu">mapply</span>(</span>
<span id="cb2-1639"><a href="#cb2-1639" aria-hidden="true" tabindex="-1"></a>        <span class="cf">function</span>(dataset, patients) dataset[, patients],</span>
<span id="cb2-1640"><a href="#cb2-1640" aria-hidden="true" tabindex="-1"></a>        <span class="at">dataset =</span> datasets,</span>
<span id="cb2-1641"><a href="#cb2-1641" aria-hidden="true" tabindex="-1"></a>        <span class="at">patients =</span> <span class="fu">list</span>(tcga_samples, scanb_samples, metabric_samples),</span>
<span id="cb2-1642"><a href="#cb2-1642" aria-hidden="true" tabindex="-1"></a>        <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>,</span>
<span id="cb2-1643"><a href="#cb2-1643" aria-hidden="true" tabindex="-1"></a>        <span class="at">SIMPLIFY =</span> <span class="cn">FALSE</span></span>
<span id="cb2-1644"><a href="#cb2-1644" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb2-1645"><a href="#cb2-1645" aria-hidden="true" tabindex="-1"></a>    <span class="at">name_scores =</span> which_scores,</span>
<span id="cb2-1646"><a href="#cb2-1646" aria-hidden="true" tabindex="-1"></a>    <span class="at">formulas =</span> formulas_survival,</span>
<span id="cb2-1647"><a href="#cb2-1647" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>,</span>
<span id="cb2-1648"><a href="#cb2-1648" aria-hidden="true" tabindex="-1"></a>    <span class="at">simplify =</span> <span class="cn">FALSE</span></span>
<span id="cb2-1649"><a href="#cb2-1649" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-1650"><a href="#cb2-1650" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1651"><a href="#cb2-1651" aria-hidden="true" tabindex="-1"></a>survival_results<span class="sc">$</span>rfs <span class="ot">&lt;-</span> survival_results<span class="sc">$</span>rfs[</span>
<span id="cb2-1652"><a href="#cb2-1652" aria-hidden="true" tabindex="-1"></a>    <span class="sc">!</span><span class="fu">sapply</span>(survival_results<span class="sc">$</span>rfs, is.null)</span>
<span id="cb2-1653"><a href="#cb2-1653" aria-hidden="true" tabindex="-1"></a>]</span>
<span id="cb2-1654"><a href="#cb2-1654" aria-hidden="true" tabindex="-1"></a>survival_results<span class="sc">$</span>rfs<span class="sc">$</span>metabric <span class="ot">&lt;-</span> survival_results<span class="sc">$</span>rfs<span class="sc">$</span>metabric[</span>
<span id="cb2-1655"><a href="#cb2-1655" aria-hidden="true" tabindex="-1"></a>    <span class="sc">!</span><span class="fu">sapply</span>(survival_results<span class="sc">$</span>rfs<span class="sc">$</span>metabric, is.null)</span>
<span id="cb2-1656"><a href="#cb2-1656" aria-hidden="true" tabindex="-1"></a>]</span>
<span id="cb2-1657"><a href="#cb2-1657" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1658"><a href="#cb2-1658" aria-hidden="true" tabindex="-1"></a>survival_results<span class="sc">$</span>os <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">discard</span>(</span>
<span id="cb2-1659"><a href="#cb2-1659" aria-hidden="true" tabindex="-1"></a>    purrr<span class="sc">::</span><span class="fu">map</span>(</span>
<span id="cb2-1660"><a href="#cb2-1660" aria-hidden="true" tabindex="-1"></a>        survival_results<span class="sc">$</span>os, </span>
<span id="cb2-1661"><a href="#cb2-1661" aria-hidden="true" tabindex="-1"></a>        <span class="sc">~</span> purrr<span class="sc">::</span><span class="fu">discard</span>(.x, is.null),</span>
<span id="cb2-1662"><a href="#cb2-1662" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb2-1663"><a href="#cb2-1663" aria-hidden="true" tabindex="-1"></a>    is.null</span>
<span id="cb2-1664"><a href="#cb2-1664" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-1665"><a href="#cb2-1665" aria-hidden="true" tabindex="-1"></a><span class="co"># we cannot simply save the models as each model has some </span></span>
<span id="cb2-1666"><a href="#cb2-1666" aria-hidden="true" tabindex="-1"></a><span class="co"># environment variables, which adds up to over 20GB. </span></span>
<span id="cb2-1667"><a href="#cb2-1667" aria-hidden="true" tabindex="-1"></a><span class="co"># check this stack exchange thread to read more on it:</span></span>
<span id="cb2-1668"><a href="#cb2-1668" aria-hidden="true" tabindex="-1"></a><span class="co"># https://stackoverflow.com/questions/42230920/saverds-inflating-size-of-object/52372480</span></span>
<span id="cb2-1669"><a href="#cb2-1669" aria-hidden="true" tabindex="-1"></a><span class="co"># the solution is to basically clear the environment from the terms </span></span>
<span id="cb2-1670"><a href="#cb2-1670" aria-hidden="true" tabindex="-1"></a><span class="co"># object. since it is pretty fast to run the survival analysis</span></span>
<span id="cb2-1671"><a href="#cb2-1671" aria-hidden="true" tabindex="-1"></a><span class="co"># we will not save the objects.</span></span>
<span id="cb2-1672"><a href="#cb2-1672" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1673"><a href="#cb2-1673" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1674"><a href="#cb2-1674" aria-hidden="true" tabindex="-1"></a><span class="co"># we now proceed to prepare the plots. since the survival result is </span></span>
<span id="cb2-1675"><a href="#cb2-1675" aria-hidden="true" tabindex="-1"></a><span class="co"># actually a large file, it is best to do all the plottings, save</span></span>
<span id="cb2-1676"><a href="#cb2-1676" aria-hidden="true" tabindex="-1"></a><span class="co"># them in other rds/pdf files and then use on the quarto markdown.</span></span>
<span id="cb2-1677"><a href="#cb2-1677" aria-hidden="true" tabindex="-1"></a><span class="co"># otherwise it is too slow to render the book. </span></span>
<span id="cb2-1678"><a href="#cb2-1678" aria-hidden="true" tabindex="-1"></a><span class="co"># the same is true for the statistics and coefficients available.</span></span>
<span id="cb2-1679"><a href="#cb2-1679" aria-hidden="true" tabindex="-1"></a>names_signatures <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb2-1680"><a href="#cb2-1680" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_G2M_CHECKPOINT"</span> <span class="ot">=</span> <span class="st">"G2M checkpoint"</span>,</span>
<span id="cb2-1681"><a href="#cb2-1681" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION"</span> <span class="ot">=</span> <span class="st">"EMT"</span>,</span>
<span id="cb2-1682"><a href="#cb2-1682" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_DNA_REPAIR"</span> <span class="ot">=</span> <span class="st">"DNA repair"</span>,</span>
<span id="cb2-1683"><a href="#cb2-1683" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_ANDROGEN_RESPONSE"</span> <span class="ot">=</span> <span class="st">"Androgen response"</span>,</span>
<span id="cb2-1684"><a href="#cb2-1684" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_PI3K_AKT_MTOR_SIGNALING"</span> <span class="ot">=</span> <span class="st">"PI3K AKT MTOR signaling"</span></span>
<span id="cb2-1685"><a href="#cb2-1685" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-1686"><a href="#cb2-1686" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1687"><a href="#cb2-1687" aria-hidden="true" tabindex="-1"></a>names_coefficients <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb2-1688"><a href="#cb2-1688" aria-hidden="true" tabindex="-1"></a>    <span class="at">metabric =</span> <span class="fu">c</span>(</span>
<span id="cb2-1689"><a href="#cb2-1689" aria-hidden="true" tabindex="-1"></a>        <span class="st">"age"</span> <span class="ot">=</span> <span class="st">"Age"</span>, </span>
<span id="cb2-1690"><a href="#cb2-1690" aria-hidden="true" tabindex="-1"></a>        <span class="st">"npi"</span> <span class="ot">=</span> <span class="st">"NPI"</span>, </span>
<span id="cb2-1691"><a href="#cb2-1691" aria-hidden="true" tabindex="-1"></a>        names_signatures</span>
<span id="cb2-1692"><a href="#cb2-1692" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb2-1693"><a href="#cb2-1693" aria-hidden="true" tabindex="-1"></a>    <span class="at">scanb =</span> <span class="fu">c</span>(</span>
<span id="cb2-1694"><a href="#cb2-1694" aria-hidden="true" tabindex="-1"></a>        <span class="st">"age"</span> <span class="ot">=</span> <span class="st">"age"</span>,</span>
<span id="cb2-1695"><a href="#cb2-1695" aria-hidden="true" tabindex="-1"></a>        names_signatures,</span>
<span id="cb2-1696"><a href="#cb2-1696" aria-hidden="true" tabindex="-1"></a>        <span class="st">"node_stageN1"</span> <span class="ot">=</span> <span class="st">"N1"</span>,</span>
<span id="cb2-1697"><a href="#cb2-1697" aria-hidden="true" tabindex="-1"></a>        <span class="st">"node_stageN2and3"</span> <span class="ot">=</span> <span class="st">"N2 and N3"</span>,</span>
<span id="cb2-1698"><a href="#cb2-1698" aria-hidden="true" tabindex="-1"></a>        <span class="st">"tumor_stageT1"</span> <span class="ot">=</span> <span class="st">"T1"</span>,</span>
<span id="cb2-1699"><a href="#cb2-1699" aria-hidden="true" tabindex="-1"></a>        <span class="st">"tumor_stageT2"</span> <span class="ot">=</span> <span class="st">"T2"</span>,</span>
<span id="cb2-1700"><a href="#cb2-1700" aria-hidden="true" tabindex="-1"></a>        <span class="st">"tumor_stageT3"</span> <span class="ot">=</span> <span class="st">"T3"</span></span>
<span id="cb2-1701"><a href="#cb2-1701" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb2-1702"><a href="#cb2-1702" aria-hidden="true" tabindex="-1"></a>    <span class="at">tcga =</span> <span class="fu">c</span>(</span>
<span id="cb2-1703"><a href="#cb2-1703" aria-hidden="true" tabindex="-1"></a>        <span class="st">"age"</span> <span class="ot">=</span> <span class="st">"age"</span>,</span>
<span id="cb2-1704"><a href="#cb2-1704" aria-hidden="true" tabindex="-1"></a>        <span class="st">"tumor_stage"</span> <span class="ot">=</span> <span class="st">"Tumor Stage"</span>,</span>
<span id="cb2-1705"><a href="#cb2-1705" aria-hidden="true" tabindex="-1"></a>        names_signatures,</span>
<span id="cb2-1706"><a href="#cb2-1706" aria-hidden="true" tabindex="-1"></a>        <span class="st">"node_stageN1"</span> <span class="ot">=</span> <span class="st">"N1"</span>,</span>
<span id="cb2-1707"><a href="#cb2-1707" aria-hidden="true" tabindex="-1"></a>        <span class="st">"node_stageN2"</span> <span class="ot">=</span> <span class="st">"N2"</span>,</span>
<span id="cb2-1708"><a href="#cb2-1708" aria-hidden="true" tabindex="-1"></a>        <span class="st">"node_stageN3"</span> <span class="ot">=</span> <span class="st">"N3"</span>,</span>
<span id="cb2-1709"><a href="#cb2-1709" aria-hidden="true" tabindex="-1"></a>        <span class="st">"tumor_stagestage_ii"</span> <span class="ot">=</span> <span class="st">"T2"</span>,</span>
<span id="cb2-1710"><a href="#cb2-1710" aria-hidden="true" tabindex="-1"></a>        <span class="st">"tumor_stagestage_iii"</span> <span class="ot">=</span> <span class="st">"T3"</span></span>
<span id="cb2-1711"><a href="#cb2-1711" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb2-1712"><a href="#cb2-1712" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-1713"><a href="#cb2-1713" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1714"><a href="#cb2-1714" aria-hidden="true" tabindex="-1"></a>patients <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb2-1715"><a href="#cb2-1715" aria-hidden="true" tabindex="-1"></a>    <span class="at">tcga =</span> <span class="st">"Lum A/B, ER+ BC"</span>,</span>
<span id="cb2-1716"><a href="#cb2-1716" aria-hidden="true" tabindex="-1"></a>    <span class="at">scanb =</span> <span class="st">"Endo Only, ER+ BC"</span>,</span>
<span id="cb2-1717"><a href="#cb2-1717" aria-hidden="true" tabindex="-1"></a>    <span class="at">metabric =</span> <span class="st">"Endo Only, ER+ BC"</span></span>
<span id="cb2-1718"><a href="#cb2-1718" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-1719"><a href="#cb2-1719" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1720"><a href="#cb2-1720" aria-hidden="true" tabindex="-1"></a>analysis_code <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb2-1721"><a href="#cb2-1721" aria-hidden="true" tabindex="-1"></a>    <span class="at">os =</span> <span class="st">"Overall survival"</span>,</span>
<span id="cb2-1722"><a href="#cb2-1722" aria-hidden="true" tabindex="-1"></a>    <span class="at">rfs =</span> <span class="st">"Recurrence free survival"</span></span>
<span id="cb2-1723"><a href="#cb2-1723" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-1724"><a href="#cb2-1724" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1725"><a href="#cb2-1725" aria-hidden="true" tabindex="-1"></a>forest_plots <span class="ot">&lt;-</span> <span class="fu">mapply</span>(</span>
<span id="cb2-1726"><a href="#cb2-1726" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(fits, type_analysis){</span>
<span id="cb2-1727"><a href="#cb2-1727" aria-hidden="true" tabindex="-1"></a>        <span class="fu">mapply</span>(</span>
<span id="cb2-1728"><a href="#cb2-1728" aria-hidden="true" tabindex="-1"></a>            <span class="cf">function</span>(cohort_fits, name_cohort, type_analysis){</span>
<span id="cb2-1729"><a href="#cb2-1729" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb2-1730"><a href="#cb2-1730" aria-hidden="true" tabindex="-1"></a>                tidy_survival <span class="ot">&lt;-</span> <span class="fu">get_df_survival</span>(</span>
<span id="cb2-1731"><a href="#cb2-1731" aria-hidden="true" tabindex="-1"></a>                    cohort_fits,</span>
<span id="cb2-1732"><a href="#cb2-1732" aria-hidden="true" tabindex="-1"></a>                    <span class="at">pathways_of_interest =</span> <span class="fu">names</span>(names_signatures)</span>
<span id="cb2-1733"><a href="#cb2-1733" aria-hidden="true" tabindex="-1"></a>                )</span>
<span id="cb2-1734"><a href="#cb2-1734" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb2-1735"><a href="#cb2-1735" aria-hidden="true" tabindex="-1"></a>                <span class="fu">plot_combined_scores</span>(tidy_survival, <span class="at">range_hr =</span> <span class="fl">4.5</span>) <span class="sc">+</span> </span>
<span id="cb2-1736"><a href="#cb2-1736" aria-hidden="true" tabindex="-1"></a>                    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb2-1737"><a href="#cb2-1737" aria-hidden="true" tabindex="-1"></a>                        <span class="at">title =</span> <span class="fu">paste0</span>(</span>
<span id="cb2-1738"><a href="#cb2-1738" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">toupper</span>(name_cohort), <span class="st">", PC3 &gt; 0</span><span class="sc">\n</span><span class="st">"</span>, </span>
<span id="cb2-1739"><a href="#cb2-1739" aria-hidden="true" tabindex="-1"></a>                            patients[[name_cohort]], <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>,</span>
<span id="cb2-1740"><a href="#cb2-1740" aria-hidden="true" tabindex="-1"></a>                            analysis_code[[type_analysis]]</span>
<span id="cb2-1741"><a href="#cb2-1741" aria-hidden="true" tabindex="-1"></a>                        )</span>
<span id="cb2-1742"><a href="#cb2-1742" aria-hidden="true" tabindex="-1"></a>                    )</span>
<span id="cb2-1743"><a href="#cb2-1743" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb2-1744"><a href="#cb2-1744" aria-hidden="true" tabindex="-1"></a>            },</span>
<span id="cb2-1745"><a href="#cb2-1745" aria-hidden="true" tabindex="-1"></a>            <span class="at">cohort_fits =</span> fits,</span>
<span id="cb2-1746"><a href="#cb2-1746" aria-hidden="true" tabindex="-1"></a>            <span class="at">name_cohort =</span> <span class="fu">names</span>(fits),</span>
<span id="cb2-1747"><a href="#cb2-1747" aria-hidden="true" tabindex="-1"></a>            <span class="at">MoreArgs =</span> <span class="fu">list</span>(<span class="at">type_analysis =</span> type_analysis),</span>
<span id="cb2-1748"><a href="#cb2-1748" aria-hidden="true" tabindex="-1"></a>            <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>,</span>
<span id="cb2-1749"><a href="#cb2-1749" aria-hidden="true" tabindex="-1"></a>            <span class="at">SIMPLIFY =</span> <span class="cn">FALSE</span></span>
<span id="cb2-1750"><a href="#cb2-1750" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb2-1751"><a href="#cb2-1751" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb2-1752"><a href="#cb2-1752" aria-hidden="true" tabindex="-1"></a>    survival_results,</span>
<span id="cb2-1753"><a href="#cb2-1753" aria-hidden="true" tabindex="-1"></a>    <span class="fu">names</span>(survival_results),</span>
<span id="cb2-1754"><a href="#cb2-1754" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>,</span>
<span id="cb2-1755"><a href="#cb2-1755" aria-hidden="true" tabindex="-1"></a>    <span class="at">SIMPLIFY =</span> <span class="cn">FALSE</span></span>
<span id="cb2-1756"><a href="#cb2-1756" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-1757"><a href="#cb2-1757" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1758"><a href="#cb2-1758" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb2-1759"><a href="#cb2-1759" aria-hidden="true" tabindex="-1"></a>    forest_plots,</span>
<span id="cb2-1760"><a href="#cb2-1760" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/pca_merging/forest_plots.rds"</span></span>
<span id="cb2-1761"><a href="#cb2-1761" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-1762"><a href="#cb2-1762" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1763"><a href="#cb2-1763" aria-hidden="true" tabindex="-1"></a><span class="co"># and now we fetch the tables that will be used later on including the</span></span>
<span id="cb2-1764"><a href="#cb2-1764" aria-hidden="true" tabindex="-1"></a><span class="co"># confidence intervals and other parameters from the fit</span></span>
<span id="cb2-1765"><a href="#cb2-1765" aria-hidden="true" tabindex="-1"></a>tables_survival <span class="ot">&lt;-</span> <span class="fu">lapply</span>(</span>
<span id="cb2-1766"><a href="#cb2-1766" aria-hidden="true" tabindex="-1"></a>    survival_results, </span>
<span id="cb2-1767"><a href="#cb2-1767" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(x) </span>
<span id="cb2-1768"><a href="#cb2-1768" aria-hidden="true" tabindex="-1"></a>        <span class="fu">lapply</span>(</span>
<span id="cb2-1769"><a href="#cb2-1769" aria-hidden="true" tabindex="-1"></a>            x,</span>
<span id="cb2-1770"><a href="#cb2-1770" aria-hidden="true" tabindex="-1"></a>            <span class="cf">function</span>(y)</span>
<span id="cb2-1771"><a href="#cb2-1771" aria-hidden="true" tabindex="-1"></a>                <span class="fu">lapply</span>(</span>
<span id="cb2-1772"><a href="#cb2-1772" aria-hidden="true" tabindex="-1"></a>                    y,</span>
<span id="cb2-1773"><a href="#cb2-1773" aria-hidden="true" tabindex="-1"></a>                    <span class="cf">function</span>(z) broom<span class="sc">::</span><span class="fu">tidy</span>(z)</span>
<span id="cb2-1774"><a href="#cb2-1774" aria-hidden="true" tabindex="-1"></a>                ) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(<span class="at">.id =</span> <span class="st">"score"</span>)</span>
<span id="cb2-1775"><a href="#cb2-1775" aria-hidden="true" tabindex="-1"></a>        ) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(<span class="at">.id =</span> <span class="st">"cohort"</span>)</span>
<span id="cb2-1776"><a href="#cb2-1776" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(<span class="at">.id =</span> <span class="st">"type_analysis"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb2-1777"><a href="#cb2-1777" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">HR =</span> <span class="fu">exp</span>(estimate))</span>
<span id="cb2-1778"><a href="#cb2-1778" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1779"><a href="#cb2-1779" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(</span>
<span id="cb2-1780"><a href="#cb2-1780" aria-hidden="true" tabindex="-1"></a>    tables_survival,</span>
<span id="cb2-1781"><a href="#cb2-1781" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/tables/pca_merging/survival_results.csv"</span>,</span>
<span id="cb2-1782"><a href="#cb2-1782" aria-hidden="true" tabindex="-1"></a>    <span class="at">row.names =</span> <span class="cn">FALSE</span></span>
<span id="cb2-1783"><a href="#cb2-1783" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-1784"><a href="#cb2-1784" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb2-1785"><a href="#cb2-1785" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1786"><a href="#cb2-1786" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width = 6, fig.height=10}</span></span>
<span id="cb2-1787"><a href="#cb2-1787" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-pca-mol-land-survival</span></span>
<span id="cb2-1788"><a href="#cb2-1788" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Overall survival analysis from all the three big cohorts and their</span></span>
<span id="cb2-1789"><a href="#cb2-1789" aria-hidden="true" tabindex="-1"></a><span class="co">#|     scores. The formulas used were the same as for the estrogen</span></span>
<span id="cb2-1790"><a href="#cb2-1790" aria-hidden="true" tabindex="-1"></a><span class="co">#|     signaling analysis.</span></span>
<span id="cb2-1791"><a href="#cb2-1791" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1792"><a href="#cb2-1792" aria-hidden="true" tabindex="-1"></a>forest_plots <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../results/rds_files/pca_merging/forest_plots.rds"</span>)</span>
<span id="cb2-1793"><a href="#cb2-1793" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(<span class="at">plotlist =</span> <span class="fu">c</span>(</span>
<span id="cb2-1794"><a href="#cb2-1794" aria-hidden="true" tabindex="-1"></a>    forest_plots<span class="sc">$</span>os,</span>
<span id="cb2-1795"><a href="#cb2-1795" aria-hidden="true" tabindex="-1"></a>    forest_plots<span class="sc">$</span>rfs</span>
<span id="cb2-1796"><a href="#cb2-1796" aria-hidden="true" tabindex="-1"></a>), <span class="at">ncol =</span> <span class="dv">1</span>)</span>
<span id="cb2-1797"><a href="#cb2-1797" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb2-1798"><a href="#cb2-1798" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-1799"><a href="#cb2-1799" aria-hidden="true" tabindex="-1"></a>We see that G2M checkpoint has hazard ratio higher than 1.</span>
<span id="cb2-1800"><a href="#cb2-1800" aria-hidden="true" tabindex="-1"></a>The other scores are a bit more</span>
<span id="cb2-1801"><a href="#cb2-1801" aria-hidden="true" tabindex="-1"></a>variable and can change the sign (type S error).</span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-right">
      <ul class="footer-items list-unstyled">
    <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/chronchi">
      <i class="bi bi-github" role="img">
</i> 
    </a>
  </li>  
    <li class="nav-item compact">
    <a class="nav-link" href="https://twitter.com/chronchi">
      <i class="bi bi-twitter" role="img">
</i> 
    </a>
  </li>  
</ul>
    </div>
  </div>
</footer>



</body></html>