<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.0.37">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Molecular landscape: a new framework for personalized medicine in Breast Cancer. - 5&nbsp; Risk score model</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./discussion.html" rel="next">
<link href="./scoring.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
    <div class="container-fluid d-flex justify-content-between">
      <h1 class="quarto-secondary-nav-title"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Risk score model</span></h1>
      <button type="button" class="quarto-btn-toggle btn" aria-label="Show secondary navigation">
        <i class="bi bi-chevron-right"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Molecular landscape: a new framework for personalized medicine in Breast Cancer.</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/chronchi/molecular_landscape" title="Source Code" class="sidebar-tool px-1"><i class="bi bi-github"></i></a>
    <a href="https://twitter.com/intent/tweet?url=|url|" title="Twitter" class="sidebar-tool px-1"><i class="bi bi-twitter"></i></a>
</div>
    </div>
      </div>
      <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">Preface</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./intro.html" class="sidebar-item-text sidebar-link">Introduction</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./surv_analysis_estrogen.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Estrogen receptor status is continuous, not dichotomous</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./pca_merging.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">A new framework for personalized medicine: integrating</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./validation.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Validating the molecular landscape</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./scoring.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Pathway activity in a personalized context</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./risk_score.html" class="sidebar-item-text sidebar-link active"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Risk score model</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./discussion.html" class="sidebar-item-text sidebar-link">Discussion</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./references.html" class="sidebar-item-text sidebar-link">References</a>
  </div>
</li>
    </ul>
    </div>
</nav>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#generating-the-risk-score" id="toc-generating-the-risk-score" class="nav-link active" data-scroll-target="#generating-the-risk-score"> <span class="header-section-number">5.1</span> Generating the risk score</a></li>
  <li><a href="#validating-the-risk-score-on-scanb" id="toc-validating-the-risk-score-on-scanb" class="nav-link" data-scroll-target="#validating-the-risk-score-on-scanb"> <span class="header-section-number">5.2</span> Validating the risk score on SCANB</a></li>
  <li><a href="#risk-score-and-poetic-trial" id="toc-risk-score-and-poetic-trial" class="nav-link" data-scroll-target="#risk-score-and-poetic-trial"> <span class="header-section-number">5.3</span> Risk score and POETIC trial</a></li>
  </ul>
<div class="toc-actions"><div><i class="bi bi-github"></i></div><div class="action-links"><p><a href="https://github.com/chronchi/molecular_landscape/issues/new" class="toc-action">Report an issue</a></p></div></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title d-none d-lg-block"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Risk score model</span></h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
</div>



<div class="quarto-title-meta">

    
    
  </div>
  

</header>

<p>Several risk signatures were developed over the years using transcriptomic datasets. Some examples were given in the previous chapters, such as OncotypeDX, Prosigna and EndoPredict. They were all derived in a similar manner, subsetting genes related to usually estrogen and proliferation signaling and then using recurrence free survival data to calculate the risk scores.</p>
<p>In this chapter we introduce a new risk score that does not subset genes and uses the information of the molecular landscape. The concept is simple, we will use the third and fourth components as covariates in the survival modeling along with tumor size and node status. This way we avoid subsetting and selecting specific genes and we use all the information the molecular landscape is providing us. Also it avoids any kind of binning for patients based on molecular subtype, thus not being necessary calculating correlations with pre-defined molecular subtypes.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>* The library is already synchronized with the lockfile.</code></pre>
</div>
</div>
<p>We develop a risk score for early stage breast cancer patients that received only endocrine therapy (ET). The idea is similar to what was done with the EndoPredict signatures, but here we extend it by using the molecular landscape information. This provides a full framework where we can analyse the risk of recurrence even for patients that have high ER+ percentage.</p>
<section id="generating-the-risk-score" class="level2" data-number="5.1">
<h2 data-number="5.1" class="anchored" data-anchor-id="generating-the-risk-score"><span class="header-section-number">5.1</span> Generating the risk score</h2>
<p>We focus only on the endocrine treated patients. For the training part of the algorithm we use a subset of patients from the METABRIC cohort. We then test the risk scores in the test set and we further validate the score on SCANB.</p>
<p>We will generate two risk scores: one including only the principal components and another one including the clinical features of the patients when valiable, namely tumor size, node status and age (either having one or more lymph node with breast cancer cells or none).</p>
<div class="cell">

</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>Call:
survival::coxph(formula = as.formula(paste0("Surv(time = rfs_months, event = rfs_status) ~ ", 
    paste(c(pcs_to_use, clin_vars), collapse = "+"))), data = training_set)

  n= 556, number of events= 216 

                    coef exp(coef)  se(coef)      z Pr(&gt;|z|)    
PC3            -0.091540  0.912525  0.027255 -3.359 0.000783 ***
PC4             0.080512  1.083842  0.022588  3.564 0.000365 ***
tumor_size      0.021809  1.022049  0.004292  5.082 3.74e-07 ***
node_statuspos  0.582681  1.790834  0.147602  3.948 7.89e-05 ***
age             0.003150  1.003155  0.007030  0.448 0.654061    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

               exp(coef) exp(-coef) lower .95 upper .95
PC3               0.9125     1.0959    0.8651    0.9626
PC4               1.0838     0.9226    1.0369    1.1329
tumor_size        1.0220     0.9784    1.0135    1.0307
node_statuspos    1.7908     0.5584    1.3410    2.3916
age               1.0032     0.9969    0.9894    1.0171

Concordance= 0.676  (se = 0.019 )
Likelihood ratio test= 66.18  on 5 df,   p=6e-13
Wald test            = 71.92  on 5 df,   p=4e-14
Score (logrank) test = 72.75  on 5 df,   p=3e-14</code></pre>
</div>
</div>
<p>We see that all coefficients have less variability than age. We still keep all the variables when calculating the risk score, since age has a coefficient relatively small we can keep it.</p>
<p>Now we calculate only the risk score based on the components.</p>
<div class="cell">

</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>Call:
survival::coxph(formula = as.formula(paste0("Surv(time = rfs_months, event = rfs_status) ~ ", 
    paste(c(pcs_to_use), collapse = "+"))), data = training_set)

  n= 556, number of events= 216 

        coef exp(coef) se(coef)      z Pr(&gt;|z|)    
PC3 -0.08657   0.91707  0.02699 -3.207  0.00134 ** 
PC4  0.09068   1.09492  0.02216  4.091 4.29e-05 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

    exp(coef) exp(-coef) lower .95 upper .95
PC3    0.9171     1.0904    0.8698    0.9669
PC4    1.0949     0.9133    1.0484    1.1435

Concordance= 0.626  (se = 0.019 )
Likelihood ratio test= 22.5  on 2 df,   p=1e-05
Wald test            = 22.49  on 2 df,   p=1e-05
Score (logrank) test = 22.78  on 2 df,   p=1e-05</code></pre>
</div>
</div>
<p>We see that the coefficients are very similar to what was found before, it didn’t change much. Also when we include the principal components on the top of the clinical variables the log likelihood increases, which is a good indication:</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>Analysis of Deviance Table
 Cox model: response is  Surv(time = rfs_months, event = rfs_status)
 Model 1: ~ tumor_size + node_status + age
 Model 2: ~ PC3 + PC4 + tumor_size + node_status + age
   loglik Chisq Df P(&gt;|Chi|)    
1 -1231.1                       
2 -1221.2 19.88  2 4.821e-05 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
</div>
<div class="cell">

</div>
<p>Now we use the test set to calculate the score.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>Call:
survival::coxph(formula = Surv(time = rfs_months, event = rfs_status) ~ 
    risk_score, data = test_set)

  n= 371, number of events= 140 

             coef exp(coef) se(coef)     z Pr(&gt;|z|)    
risk_score 0.8632    2.3707   0.1583 5.453 4.96e-08 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

           exp(coef) exp(-coef) lower .95 upper .95
risk_score     2.371     0.4218     1.738     3.233

Concordance= 0.655  (se = 0.025 )
Likelihood ratio test= 28.56  on 1 df,   p=9e-08
Wald test            = 29.73  on 1 df,   p=5e-08
Score (logrank) test = 29.78  on 1 df,   p=5e-08</code></pre>
</div>
</div>
<p>We see that the higher the risk score the worse it is for the patients treated only with endocrine therapy.</p>
</section>
<section id="validating-the-risk-score-on-scanb" class="level2" data-number="5.2">
<h2 data-number="5.2" class="anchored" data-anchor-id="validating-the-risk-score-on-scanb"><span class="header-section-number">5.2</span> Validating the risk score on SCANB</h2>
<p>We can now validate the risk score on the SCANB dataset.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>Call:
survival::coxph(formula = as.formula(paste0("Surv(time = rfs_months, event = rfs_status) ~ ", 
    paste(c("risk_score"), collapse = "+"))), data = endo_only_scanb)

  n= 2829, number of events= 212 

             coef exp(coef) se(coef)     z Pr(&gt;|z|)    
risk_score 1.0568    2.8772   0.1001 10.56   &lt;2e-16 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

           exp(coef) exp(-coef) lower .95 upper .95
risk_score     2.877     0.3476     2.365     3.501

Concordance= 0.699  (se = 0.02 )
Likelihood ratio test= 87.08  on 1 df,   p=&lt;2e-16
Wald test            = 111.4  on 1 df,   p=&lt;2e-16
Score (logrank) test = 108.8  on 1 df,   p=&lt;2e-16</code></pre>
</div>
</div>
<p>Again the risk score is predictive with a high coefficient, meaning that the higher the score the worse it is for the patient.</p>
<p>We now visualize a bit the risk scores developed here in terms of the molecular landscape. <a href="#fig-risk-endo-pam50">Figure&nbsp;<span>5.1</span></a></p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-risk-endo-pam50" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="risk_score_files/figure-html/fig-risk-endo-pam50-1.png" class="img-fluid figure-img" width="672"></p>
<p></p><figcaption class="figure-caption">Figure 5.1: Molecular landscape of SCANB patients treated with only endocrine therapy and that have ER+ breast cancer. Only patients with risk score less than 2 are plotted here. There are only 26 patients out of 2829 with risk score higher than 2 that would skew the color scale.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>We now investigate the risk score among patients that have high ER IHC percentage as well. We have seen that these patients have different outcomes based on the molecular estrogen signaling. We now investigate the effect of the risk score.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>Call:
survival::coxph(formula = as.formula(paste0("Surv(time = rfs_months, event = rfs_status) ~ ", 
    paste(c("risk_score"), collapse = "+"))), data = endo_only_scanb %&gt;% 
    dplyr::filter(ER.pct &gt;= 90 &amp; risk_score &lt; 2))

  n= 2109, number of events= 153 

             coef exp(coef) se(coef)     z Pr(&gt;|z|)    
risk_score 1.1937    3.2994   0.1498 7.967 1.63e-15 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

           exp(coef) exp(-coef) lower .95 upper .95
risk_score     3.299     0.3031      2.46     4.426

Concordance= 0.695  (se = 0.023 )
Likelihood ratio test= 58.92  on 1 df,   p=2e-14
Wald test            = 63.47  on 1 df,   p=2e-15
Score (logrank) test = 66.98  on 1 df,   p=3e-16</code></pre>
</div>
</div>
<p>We see that the score is still predictive, meaning that even patients that have high ER IHC percentage, there are those with higher risk. Note though that the analysis still is a bit limited due to the followup time that in median is 5.3 years.</p>
</section>
<section id="risk-score-and-poetic-trial" class="level2" data-number="5.3">
<h2 data-number="5.3" class="anchored" data-anchor-id="risk-score-and-poetic-trial"><span class="header-section-number">5.3</span> Risk score and POETIC trial</h2>
<p>We can evaluate one of the two risk scores, the one using only the principal components, in the POETIC trial data. The first question we can address is if there is difference in scores in the baseline tissue (<a href="#fig-poetic-risk-score">Figure&nbsp;<span>5.2</span></a>).</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-poetic-risk-score" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="risk_score_files/figure-html/fig-poetic-risk-score-1.png" class="img-fluid figure-img" width="672"></p>
<p></p><figcaption class="figure-caption">Figure 5.2: Risk scores for responders and non-responders in the baseline tumor tissue.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>We see in the <a href="#fig-poetic-risk-score">Figure&nbsp;<span>5.2</span></a> that in average the non responders have higher risk scores. Now if we compare the risk scores between baseline and surgery, the slope of the line that fits for each group separately, responder and non responder, is different. The slope for the non responders is parallel to the identity line going through the origin with a slope of 1, meaning that there is not much change of risk upon the treatment. On other hand, for the responder group the slope is decreased and has a value lower than 1, meaning that the risk is decreased upon aromatase inhibition after two weeks (<a href="#fig-poetic-risk-base-surgery">Figure&nbsp;<span>5.3</span></a>).</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-poetic-risk-base-surgery" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="risk_score_files/figure-html/fig-poetic-risk-base-surgery-1.png" class="img-fluid figure-img" width="672"></p>
<p></p><figcaption class="figure-caption">Figure 5.3: Comparison of risk scores using the PCs only before and after two weeks of aromatase inhibitors in post-menopausal women. The colored lines correspond do the linear regression for each response status group separately.</figcaption><p></p>
</figure>
</div>
</div>
</div>


<!-- -->

</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      let href = ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./scoring.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Pathway activity in a personalized context</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./discussion.html" class="pagination-link">
        <span class="nav-page-text">Discussion</span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb8" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu"># Risk score model</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>Several risk signatures were developed over the years using transcriptomic</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>datasets. Some examples were given in the previous chapters, such as </span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>OncotypeDX, Prosigna and EndoPredict. They were all derived in a similar</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>manner, subsetting genes related to usually estrogen and proliferation </span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>signaling and then using recurrence free survival data to calculate</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>the risk scores. </span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>In this chapter we introduce a new risk score that does not subset</span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>genes and uses the information of the molecular landscape. The concept</span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a>is simple, we will use the third and fourth components as covariates in</span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a>the survival modeling along with tumor size and node status. This</span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a>way we avoid subsetting and selecting specific genes and we use all</span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a>the information the molecular landscape is providing us. Also it avoids</span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a>any kind of binning for patients based on molecular subtype, thus not being</span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a>necessary calculating correlations with pre-defined molecular subtypes.</span>
<span id="cb8-18"><a href="#cb8-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-21"><a href="#cb8-21" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb8-22"><a href="#cb8-22" aria-hidden="true" tabindex="-1"></a>renv<span class="sc">::</span><span class="fu">restore</span>()</span>
<span id="cb8-23"><a href="#cb8-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-24"><a href="#cb8-24" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb8-25"><a href="#cb8-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-26"><a href="#cb8-26" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb8-27"><a href="#cb8-27" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(PCAtools)</span>
<span id="cb8-28"><a href="#cb8-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-29"><a href="#cb8-29" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(singscore)</span>
<span id="cb8-30"><a href="#cb8-30" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SummarizedExperiment)</span>
<span id="cb8-31"><a href="#cb8-31" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(survival)</span>
<span id="cb8-32"><a href="#cb8-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-33"><a href="#cb8-33" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"utils.R"</span>)</span>
<span id="cb8-34"><a href="#cb8-34" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"first_run.R"</span>)</span>
<span id="cb8-35"><a href="#cb8-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-36"><a href="#cb8-36" aria-hidden="true" tabindex="-1"></a><span class="co"># the following script load all data necessary to run the chunks.</span></span>
<span id="cb8-37"><a href="#cb8-37" aria-hidden="true" tabindex="-1"></a><span class="co"># the data is generated from this quarto document itself, therefore</span></span>
<span id="cb8-38"><a href="#cb8-38" aria-hidden="true" tabindex="-1"></a><span class="co"># if you are running this documents the first time and don't have the</span></span>
<span id="cb8-39"><a href="#cb8-39" aria-hidden="true" tabindex="-1"></a><span class="co"># files, comment the following lines. Moreover, if this is your first</span></span>
<span id="cb8-40"><a href="#cb8-40" aria-hidden="true" tabindex="-1"></a><span class="co"># time running the document, you should run all chunks, to generate </span></span>
<span id="cb8-41"><a href="#cb8-41" aria-hidden="true" tabindex="-1"></a><span class="co"># all the necessary files, if you don't have them. Once all files </span></span>
<span id="cb8-42"><a href="#cb8-42" aria-hidden="true" tabindex="-1"></a><span class="co"># are saved and available in the respective folder, the following</span></span>
<span id="cb8-43"><a href="#cb8-43" aria-hidden="true" tabindex="-1"></a><span class="co"># lines can be executed. </span></span>
<span id="cb8-44"><a href="#cb8-44" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (first_run){</span>
<span id="cb8-45"><a href="#cb8-45" aria-hidden="true" tabindex="-1"></a>    load_at_setup <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb8-46"><a href="#cb8-46" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb8-47"><a href="#cb8-47" aria-hidden="true" tabindex="-1"></a>    load_at_setup <span class="ot">&lt;-</span> <span class="cn">TRUE</span></span>
<span id="cb8-48"><a href="#cb8-48" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb8-49"><a href="#cb8-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-50"><a href="#cb8-50" aria-hidden="true" tabindex="-1"></a>name_document <span class="ot">&lt;-</span> <span class="st">"risk_score"</span></span>
<span id="cb8-51"><a href="#cb8-51" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"load_rds_files.R"</span>)</span>
<span id="cb8-52"><a href="#cb8-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-53"><a href="#cb8-53" aria-hidden="true" tabindex="-1"></a><span class="co"># by setting the dev to png and pdf, this saves the figures in a specific</span></span>
<span id="cb8-54"><a href="#cb8-54" aria-hidden="true" tabindex="-1"></a><span class="co"># folder in both formats. moreover, since png is coming first, it shows</span></span>
<span id="cb8-55"><a href="#cb8-55" aria-hidden="true" tabindex="-1"></a><span class="co"># this figure when rendering the html. What is nice about this is that it</span></span>
<span id="cb8-56"><a href="#cb8-56" aria-hidden="true" tabindex="-1"></a><span class="co"># inherits the properties from the chunk to save the figure, so no need</span></span>
<span id="cb8-57"><a href="#cb8-57" aria-hidden="true" tabindex="-1"></a><span class="co"># to use ggplot2::ggsave to save the plots. this also works </span></span>
<span id="cb8-58"><a href="#cb8-58" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">dev =</span> <span class="fu">c</span>(<span class="st">'png'</span>, <span class="st">'pdf'</span>, <span class="st">'svg'</span>))</span>
<span id="cb8-59"><a href="#cb8-59" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">bitmapType =</span> <span class="st">'cairo'</span>)</span>
<span id="cb8-60"><a href="#cb8-60" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb8-61"><a href="#cb8-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-62"><a href="#cb8-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-63"><a href="#cb8-63" aria-hidden="true" tabindex="-1"></a>We develop a risk score for early stage breast cancer patients that </span>
<span id="cb8-64"><a href="#cb8-64" aria-hidden="true" tabindex="-1"></a>received only endocrine therapy (ET). The idea is similar to what was</span>
<span id="cb8-65"><a href="#cb8-65" aria-hidden="true" tabindex="-1"></a>done with the EndoPredict signatures, but here we extend it by using the</span>
<span id="cb8-66"><a href="#cb8-66" aria-hidden="true" tabindex="-1"></a>molecular landscape information. This provides a full framework where</span>
<span id="cb8-67"><a href="#cb8-67" aria-hidden="true" tabindex="-1"></a>we can analyse the risk of recurrence even for patients that have </span>
<span id="cb8-68"><a href="#cb8-68" aria-hidden="true" tabindex="-1"></a>high ER+ percentage.</span>
<span id="cb8-69"><a href="#cb8-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-70"><a href="#cb8-70" aria-hidden="true" tabindex="-1"></a><span class="fu">## Generating the risk score</span></span>
<span id="cb8-71"><a href="#cb8-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-72"><a href="#cb8-72" aria-hidden="true" tabindex="-1"></a>We focus only on the endocrine treated patients. For the training part</span>
<span id="cb8-73"><a href="#cb8-73" aria-hidden="true" tabindex="-1"></a>of the algorithm we use a subset of patients from the METABRIC cohort.</span>
<span id="cb8-74"><a href="#cb8-74" aria-hidden="true" tabindex="-1"></a>We then test the risk scores in the test set and we further validate</span>
<span id="cb8-75"><a href="#cb8-75" aria-hidden="true" tabindex="-1"></a>the score on SCANB. </span>
<span id="cb8-76"><a href="#cb8-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-77"><a href="#cb8-77" aria-hidden="true" tabindex="-1"></a>We will generate two risk scores: one including only the principal </span>
<span id="cb8-78"><a href="#cb8-78" aria-hidden="true" tabindex="-1"></a>components and another one including the clinical features of the </span>
<span id="cb8-79"><a href="#cb8-79" aria-hidden="true" tabindex="-1"></a>patients when valiable, namely tumor size, node status and age (either having</span>
<span id="cb8-80"><a href="#cb8-80" aria-hidden="true" tabindex="-1"></a>one or more lymph node with breast cancer cells or none). </span>
<span id="cb8-81"><a href="#cb8-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-84"><a href="#cb8-84" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb8-85"><a href="#cb8-85" aria-hidden="true" tabindex="-1"></a><span class="co"># start by preparing the datasets so it is easily used later</span></span>
<span id="cb8-86"><a href="#cb8-86" aria-hidden="true" tabindex="-1"></a>df_pca_metabric <span class="ot">&lt;-</span> df_pca <span class="sc">%&gt;%</span> </span>
<span id="cb8-87"><a href="#cb8-87" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"metabric"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb8-88"><a href="#cb8-88" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">TreatGroup =</span> dplyr<span class="sc">::</span><span class="fu">case_when</span>(</span>
<span id="cb8-89"><a href="#cb8-89" aria-hidden="true" tabindex="-1"></a>        HORMONE_THERAPY <span class="sc">==</span> <span class="st">"YES"</span> <span class="sc">&amp;</span> CHEMOTHERAPY <span class="sc">==</span> <span class="st">"YES"</span> <span class="sc">~</span> <span class="st">"ChemoEndo"</span>,</span>
<span id="cb8-90"><a href="#cb8-90" aria-hidden="true" tabindex="-1"></a>        HORMONE_THERAPY <span class="sc">==</span> <span class="st">"YES"</span> <span class="sc">&amp;</span> CHEMOTHERAPY <span class="sc">==</span> <span class="st">"NO"</span> <span class="sc">~</span> <span class="st">"Endo"</span>,</span>
<span id="cb8-91"><a href="#cb8-91" aria-hidden="true" tabindex="-1"></a>        <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"Other"</span></span>
<span id="cb8-92"><a href="#cb8-92" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">%&gt;%</span> </span>
<span id="cb8-93"><a href="#cb8-93" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">node_status =</span> <span class="fu">ifelse</span>(</span>
<span id="cb8-94"><a href="#cb8-94" aria-hidden="true" tabindex="-1"></a>        LYMPH_NODES_EXAMINED_POSITIVE <span class="sc">&gt;</span> <span class="dv">0</span>, </span>
<span id="cb8-95"><a href="#cb8-95" aria-hidden="true" tabindex="-1"></a>        <span class="st">"pos"</span>,</span>
<span id="cb8-96"><a href="#cb8-96" aria-hidden="true" tabindex="-1"></a>        <span class="st">"neg"</span></span>
<span id="cb8-97"><a href="#cb8-97" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">%&gt;%</span> </span>
<span id="cb8-98"><a href="#cb8-98" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb8-99"><a href="#cb8-99" aria-hidden="true" tabindex="-1"></a>        <span class="at">node_status =</span> <span class="fu">factor</span>(node_status, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"neg"</span>, <span class="st">"pos"</span>)),</span>
<span id="cb8-100"><a href="#cb8-100" aria-hidden="true" tabindex="-1"></a>        <span class="at">tumor_size =</span> TUMOR_SIZE</span>
<span id="cb8-101"><a href="#cb8-101" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb8-102"><a href="#cb8-102" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-103"><a href="#cb8-103" aria-hidden="true" tabindex="-1"></a>df_pca_scanb <span class="ot">&lt;-</span> df_pca <span class="sc">%&gt;%</span> </span>
<span id="cb8-104"><a href="#cb8-104" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"scanb"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb8-105"><a href="#cb8-105" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">node_status =</span> <span class="fu">ifelse</span>(</span>
<span id="cb8-106"><a href="#cb8-106" aria-hidden="true" tabindex="-1"></a>        node_group <span class="sc">==</span> <span class="st">"N0"</span>, </span>
<span id="cb8-107"><a href="#cb8-107" aria-hidden="true" tabindex="-1"></a>        <span class="st">"neg"</span>,</span>
<span id="cb8-108"><a href="#cb8-108" aria-hidden="true" tabindex="-1"></a>        <span class="st">"pos"</span></span>
<span id="cb8-109"><a href="#cb8-109" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">%&gt;%</span> </span>
<span id="cb8-110"><a href="#cb8-110" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">node_status =</span> <span class="fu">factor</span>(</span>
<span id="cb8-111"><a href="#cb8-111" aria-hidden="true" tabindex="-1"></a>        node_status, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"neg"</span>, <span class="st">"pos"</span>)</span>
<span id="cb8-112"><a href="#cb8-112" aria-hidden="true" tabindex="-1"></a>    ))</span>
<span id="cb8-113"><a href="#cb8-113" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-114"><a href="#cb8-114" aria-hidden="true" tabindex="-1"></a><span class="co"># select only the endocrine treated patients</span></span>
<span id="cb8-115"><a href="#cb8-115" aria-hidden="true" tabindex="-1"></a>endo_only_scanb <span class="ot">&lt;-</span> df_pca_scanb <span class="sc">%&gt;%</span> </span>
<span id="cb8-116"><a href="#cb8-116" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb8-117"><a href="#cb8-117" aria-hidden="true" tabindex="-1"></a>            TreatGroup <span class="sc">==</span> <span class="st">"Endo"</span> <span class="sc">&amp;</span> </span>
<span id="cb8-118"><a href="#cb8-118" aria-hidden="true" tabindex="-1"></a>            <span class="sc">!</span><span class="fu">is.na</span>(rfs_status) <span class="sc">&amp;</span> </span>
<span id="cb8-119"><a href="#cb8-119" aria-hidden="true" tabindex="-1"></a>            <span class="sc">!</span><span class="fu">is.na</span>(rfs_months) <span class="sc">&amp;</span></span>
<span id="cb8-120"><a href="#cb8-120" aria-hidden="true" tabindex="-1"></a>            er_status <span class="sc">==</span> <span class="st">"pos"</span> <span class="sc">&amp;</span></span>
<span id="cb8-121"><a href="#cb8-121" aria-hidden="true" tabindex="-1"></a>            <span class="sc">!</span><span class="fu">is.na</span>(age) <span class="sc">&amp;</span></span>
<span id="cb8-122"><a href="#cb8-122" aria-hidden="true" tabindex="-1"></a>            <span class="sc">!</span><span class="fu">is.na</span>(tumor_size) <span class="sc">&amp;</span></span>
<span id="cb8-123"><a href="#cb8-123" aria-hidden="true" tabindex="-1"></a>            <span class="sc">!</span><span class="fu">is.na</span>(node_status)</span>
<span id="cb8-124"><a href="#cb8-124" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb8-125"><a href="#cb8-125" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-126"><a href="#cb8-126" aria-hidden="true" tabindex="-1"></a>endo_only_metabric <span class="ot">&lt;-</span> df_pca_metabric <span class="sc">%&gt;%</span> </span>
<span id="cb8-127"><a href="#cb8-127" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb8-128"><a href="#cb8-128" aria-hidden="true" tabindex="-1"></a>            HORMONE_THERAPY <span class="sc">==</span> <span class="st">"YES"</span> <span class="sc">&amp;</span> </span>
<span id="cb8-129"><a href="#cb8-129" aria-hidden="true" tabindex="-1"></a>            CHEMOTHERAPY <span class="sc">==</span> <span class="st">"NO"</span> <span class="sc">&amp;</span></span>
<span id="cb8-130"><a href="#cb8-130" aria-hidden="true" tabindex="-1"></a>            <span class="sc">!</span><span class="fu">is.na</span>(rfs_status) <span class="sc">&amp;</span> </span>
<span id="cb8-131"><a href="#cb8-131" aria-hidden="true" tabindex="-1"></a>            <span class="sc">!</span><span class="fu">is.na</span>(rfs_months) <span class="sc">&amp;</span></span>
<span id="cb8-132"><a href="#cb8-132" aria-hidden="true" tabindex="-1"></a>            er_status <span class="sc">==</span> <span class="st">"pos"</span> <span class="sc">&amp;</span></span>
<span id="cb8-133"><a href="#cb8-133" aria-hidden="true" tabindex="-1"></a>            <span class="sc">!</span><span class="fu">is.na</span>(age) <span class="sc">&amp;</span></span>
<span id="cb8-134"><a href="#cb8-134" aria-hidden="true" tabindex="-1"></a>            <span class="sc">!</span><span class="fu">is.na</span>(tumor_size) <span class="sc">&amp;</span></span>
<span id="cb8-135"><a href="#cb8-135" aria-hidden="true" tabindex="-1"></a>            <span class="sc">!</span><span class="fu">is.na</span>(node_status)</span>
<span id="cb8-136"><a href="#cb8-136" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb8-137"><a href="#cb8-137" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-138"><a href="#cb8-138" aria-hidden="true" tabindex="-1"></a><span class="co"># set the seed so we can reproduce the results later and then</span></span>
<span id="cb8-139"><a href="#cb8-139" aria-hidden="true" tabindex="-1"></a><span class="co"># select the training cohort</span></span>
<span id="cb8-140"><a href="#cb8-140" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">12938</span>)</span>
<span id="cb8-141"><a href="#cb8-141" aria-hidden="true" tabindex="-1"></a>training_set <span class="ot">&lt;-</span> endo_only_metabric <span class="sc">%&gt;%</span></span>
<span id="cb8-142"><a href="#cb8-142" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">slice_sample</span>(<span class="at">prop =</span> <span class="fl">0.6</span>)</span>
<span id="cb8-143"><a href="#cb8-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-144"><a href="#cb8-144" aria-hidden="true" tabindex="-1"></a><span class="co"># we use the two PCs we have been investigating, but that does not mean</span></span>
<span id="cb8-145"><a href="#cb8-145" aria-hidden="true" tabindex="-1"></a><span class="co"># other PCs can't be used</span></span>
<span id="cb8-146"><a href="#cb8-146" aria-hidden="true" tabindex="-1"></a>pcs_to_use <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"PC"</span>, <span class="fu">c</span>(<span class="dv">3</span><span class="sc">:</span><span class="dv">4</span>))</span>
<span id="cb8-147"><a href="#cb8-147" aria-hidden="true" tabindex="-1"></a>clin_vars <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"tumor_size"</span>, <span class="st">"node_status"</span>, <span class="st">"age"</span>)</span>
<span id="cb8-148"><a href="#cb8-148" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-149"><a href="#cb8-149" aria-hidden="true" tabindex="-1"></a>risk_score_model <span class="ot">&lt;-</span> survival<span class="sc">::</span><span class="fu">coxph</span>(</span>
<span id="cb8-150"><a href="#cb8-150" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.formula</span>(<span class="fu">paste0</span>(</span>
<span id="cb8-151"><a href="#cb8-151" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Surv(time = rfs_months, event = rfs_status) ~ "</span>,</span>
<span id="cb8-152"><a href="#cb8-152" aria-hidden="true" tabindex="-1"></a>        <span class="fu">paste</span>(<span class="fu">c</span>(pcs_to_use, clin_vars), <span class="at">collapse =</span> <span class="st">"+"</span>)</span>
<span id="cb8-153"><a href="#cb8-153" aria-hidden="true" tabindex="-1"></a>    )),</span>
<span id="cb8-154"><a href="#cb8-154" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> training_set</span>
<span id="cb8-155"><a href="#cb8-155" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb8-156"><a href="#cb8-156" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-157"><a href="#cb8-157" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb8-158"><a href="#cb8-158" aria-hidden="true" tabindex="-1"></a>    risk_score_model,</span>
<span id="cb8-159"><a href="#cb8-159" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/risk_score/risk_score_model.rds"</span></span>
<span id="cb8-160"><a href="#cb8-160" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb8-161"><a href="#cb8-161" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb8-162"><a href="#cb8-162" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-165"><a href="#cb8-165" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb8-166"><a href="#cb8-166" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(risk_score_model)</span>
<span id="cb8-167"><a href="#cb8-167" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb8-168"><a href="#cb8-168" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-169"><a href="#cb8-169" aria-hidden="true" tabindex="-1"></a>We see that all coefficients have less variability than age. We still keep</span>
<span id="cb8-170"><a href="#cb8-170" aria-hidden="true" tabindex="-1"></a>all the variables when calculating the risk score, since age has a coefficient</span>
<span id="cb8-171"><a href="#cb8-171" aria-hidden="true" tabindex="-1"></a>relatively small we can keep it. </span>
<span id="cb8-172"><a href="#cb8-172" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-173"><a href="#cb8-173" aria-hidden="true" tabindex="-1"></a>Now we calculate only the risk score based on the components. </span>
<span id="cb8-174"><a href="#cb8-174" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-177"><a href="#cb8-177" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb8-178"><a href="#cb8-178" aria-hidden="true" tabindex="-1"></a>risk_score_model_pcs <span class="ot">&lt;-</span> survival<span class="sc">::</span><span class="fu">coxph</span>(</span>
<span id="cb8-179"><a href="#cb8-179" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.formula</span>(<span class="fu">paste0</span>(</span>
<span id="cb8-180"><a href="#cb8-180" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Surv(time = rfs_months, event = rfs_status) ~ "</span>,</span>
<span id="cb8-181"><a href="#cb8-181" aria-hidden="true" tabindex="-1"></a>        <span class="fu">paste</span>(<span class="fu">c</span>(pcs_to_use), <span class="at">collapse =</span> <span class="st">"+"</span>)</span>
<span id="cb8-182"><a href="#cb8-182" aria-hidden="true" tabindex="-1"></a>    )),</span>
<span id="cb8-183"><a href="#cb8-183" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> training_set</span>
<span id="cb8-184"><a href="#cb8-184" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb8-185"><a href="#cb8-185" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-186"><a href="#cb8-186" aria-hidden="true" tabindex="-1"></a>risk_score_model_clins <span class="ot">&lt;-</span> survival<span class="sc">::</span><span class="fu">coxph</span>(</span>
<span id="cb8-187"><a href="#cb8-187" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.formula</span>(<span class="fu">paste0</span>(</span>
<span id="cb8-188"><a href="#cb8-188" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Surv(time = rfs_months, event = rfs_status) ~ "</span>,</span>
<span id="cb8-189"><a href="#cb8-189" aria-hidden="true" tabindex="-1"></a>        <span class="fu">paste</span>(<span class="fu">c</span>(clin_vars), <span class="at">collapse =</span> <span class="st">"+"</span>)</span>
<span id="cb8-190"><a href="#cb8-190" aria-hidden="true" tabindex="-1"></a>    )),</span>
<span id="cb8-191"><a href="#cb8-191" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> training_set</span>
<span id="cb8-192"><a href="#cb8-192" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb8-193"><a href="#cb8-193" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-194"><a href="#cb8-194" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb8-195"><a href="#cb8-195" aria-hidden="true" tabindex="-1"></a>    risk_score_model_pcs,</span>
<span id="cb8-196"><a href="#cb8-196" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/risk_score/risk_score_model_pcs.rds"</span></span>
<span id="cb8-197"><a href="#cb8-197" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb8-198"><a href="#cb8-198" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb8-199"><a href="#cb8-199" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-200"><a href="#cb8-200" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-203"><a href="#cb8-203" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb8-204"><a href="#cb8-204" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(risk_score_model_pcs)</span>
<span id="cb8-205"><a href="#cb8-205" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb8-206"><a href="#cb8-206" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-207"><a href="#cb8-207" aria-hidden="true" tabindex="-1"></a>We see that the coefficients are very similar to what was found before,</span>
<span id="cb8-208"><a href="#cb8-208" aria-hidden="true" tabindex="-1"></a>it didn't change much. Also when we include the principal components on the</span>
<span id="cb8-209"><a href="#cb8-209" aria-hidden="true" tabindex="-1"></a>top of the clinical variables the log likelihood increases, which is </span>
<span id="cb8-210"><a href="#cb8-210" aria-hidden="true" tabindex="-1"></a>a good indication: </span>
<span id="cb8-211"><a href="#cb8-211" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-214"><a href="#cb8-214" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb8-215"><a href="#cb8-215" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(risk_score_model_clins, risk_score_model)</span>
<span id="cb8-216"><a href="#cb8-216" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb8-217"><a href="#cb8-217" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-220"><a href="#cb8-220" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb8-221"><a href="#cb8-221" aria-hidden="true" tabindex="-1"></a><span class="co"># first get the coefficients </span></span>
<span id="cb8-222"><a href="#cb8-222" aria-hidden="true" tabindex="-1"></a>coefs_risk_model <span class="ot">&lt;-</span> <span class="fu">coef</span>(risk_score_model)</span>
<span id="cb8-223"><a href="#cb8-223" aria-hidden="true" tabindex="-1"></a>coefs_risk_model_pcs <span class="ot">&lt;-</span> <span class="fu">coef</span>(risk_score_model_pcs)</span>
<span id="cb8-224"><a href="#cb8-224" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-225"><a href="#cb8-225" aria-hidden="true" tabindex="-1"></a><span class="co"># let us save them so they can be used later as well</span></span>
<span id="cb8-226"><a href="#cb8-226" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb8-227"><a href="#cb8-227" aria-hidden="true" tabindex="-1"></a>    coefs_risk_model,</span>
<span id="cb8-228"><a href="#cb8-228" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/risk_score/coefs_risk_model.rds"</span></span>
<span id="cb8-229"><a href="#cb8-229" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb8-230"><a href="#cb8-230" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb8-231"><a href="#cb8-231" aria-hidden="true" tabindex="-1"></a>    coefs_risk_model_pcs,</span>
<span id="cb8-232"><a href="#cb8-232" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/risk_score/coefs_risk_model_pcs.rds"</span></span>
<span id="cb8-233"><a href="#cb8-233" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb8-234"><a href="#cb8-234" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-235"><a href="#cb8-235" aria-hidden="true" tabindex="-1"></a><span class="co"># now we can calculate the scores for all the patients. For TCGA and POETIC</span></span>
<span id="cb8-236"><a href="#cb8-236" aria-hidden="true" tabindex="-1"></a><span class="co"># we do not have tumor size, so for those we calculate the risk score based</span></span>
<span id="cb8-237"><a href="#cb8-237" aria-hidden="true" tabindex="-1"></a><span class="co"># on the components only.</span></span>
<span id="cb8-238"><a href="#cb8-238" aria-hidden="true" tabindex="-1"></a>endo_only_scanb<span class="sc">$</span>risk_score <span class="ot">&lt;-</span> <span class="fu">calculate_risk_score</span>(coefs_risk_model, endo_only_scanb)</span>
<span id="cb8-239"><a href="#cb8-239" aria-hidden="true" tabindex="-1"></a>endo_only_metabric<span class="sc">$</span>risk_score <span class="ot">&lt;-</span> <span class="fu">calculate_risk_score</span>(coefs_risk_model, endo_only_metabric)</span>
<span id="cb8-240"><a href="#cb8-240" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-241"><a href="#cb8-241" aria-hidden="true" tabindex="-1"></a>df_pca_metabric<span class="sc">$</span>risk_score <span class="ot">&lt;-</span> <span class="fu">calculate_risk_score</span>(coefs_risk_model, df_pca_metabric) </span>
<span id="cb8-242"><a href="#cb8-242" aria-hidden="true" tabindex="-1"></a>df_pca_scanb<span class="sc">$</span>risk_score <span class="ot">&lt;-</span> <span class="fu">calculate_risk_score</span>(coefs_risk_model, df_pca_scanb)</span>
<span id="cb8-243"><a href="#cb8-243" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-244"><a href="#cb8-244" aria-hidden="true" tabindex="-1"></a><span class="co"># we can now calculate the score for all samples based on PC3 and PC4</span></span>
<span id="cb8-245"><a href="#cb8-245" aria-hidden="true" tabindex="-1"></a>df_pca<span class="sc">$</span>risk_score_pcs <span class="ot">&lt;-</span> <span class="fu">calculate_risk_score</span>(coefs_risk_model_pcs, df_pca, <span class="at">is_node =</span> <span class="cn">FALSE</span>)</span>
<span id="cb8-246"><a href="#cb8-246" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-247"><a href="#cb8-247" aria-hidden="true" tabindex="-1"></a>test_set <span class="ot">&lt;-</span> endo_only_metabric <span class="sc">%&gt;%</span></span>
<span id="cb8-248"><a href="#cb8-248" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span>(sample_name <span class="sc">%in%</span> training_set<span class="sc">$</span>sample_name))</span>
<span id="cb8-249"><a href="#cb8-249" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-250"><a href="#cb8-250" aria-hidden="true" tabindex="-1"></a>risk_score_test <span class="ot">&lt;-</span> survival<span class="sc">::</span><span class="fu">coxph</span>(</span>
<span id="cb8-251"><a href="#cb8-251" aria-hidden="true" tabindex="-1"></a>    <span class="fu">Surv</span>(<span class="at">time =</span> rfs_months, <span class="at">event =</span> rfs_status) <span class="sc">~</span> risk_score,</span>
<span id="cb8-252"><a href="#cb8-252" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> test_set</span>
<span id="cb8-253"><a href="#cb8-253" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb8-254"><a href="#cb8-254" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-255"><a href="#cb8-255" aria-hidden="true" tabindex="-1"></a>risk_score_test_clins <span class="ot">&lt;-</span> survival<span class="sc">::</span><span class="fu">coxph</span>(</span>
<span id="cb8-256"><a href="#cb8-256" aria-hidden="true" tabindex="-1"></a>    <span class="fu">Surv</span>(<span class="at">time =</span> rfs_months, <span class="at">event =</span> rfs_status) <span class="sc">~</span> risk_score <span class="sc">+</span> </span>
<span id="cb8-257"><a href="#cb8-257" aria-hidden="true" tabindex="-1"></a>        tumor_size,</span>
<span id="cb8-258"><a href="#cb8-258" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> test_set</span>
<span id="cb8-259"><a href="#cb8-259" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb8-260"><a href="#cb8-260" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb8-261"><a href="#cb8-261" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-262"><a href="#cb8-262" aria-hidden="true" tabindex="-1"></a>Now we use the test set to calculate the score.</span>
<span id="cb8-263"><a href="#cb8-263" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-266"><a href="#cb8-266" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb8-267"><a href="#cb8-267" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(risk_score_test)</span>
<span id="cb8-268"><a href="#cb8-268" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb8-269"><a href="#cb8-269" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-270"><a href="#cb8-270" aria-hidden="true" tabindex="-1"></a>We see that the higher the risk score the worse it is for the patients </span>
<span id="cb8-271"><a href="#cb8-271" aria-hidden="true" tabindex="-1"></a>treated only with endocrine therapy. </span>
<span id="cb8-272"><a href="#cb8-272" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-273"><a href="#cb8-273" aria-hidden="true" tabindex="-1"></a><span class="fu">## Validating the risk score on SCANB</span></span>
<span id="cb8-274"><a href="#cb8-274" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-275"><a href="#cb8-275" aria-hidden="true" tabindex="-1"></a>We can now validate the risk score on the SCANB dataset. </span>
<span id="cb8-276"><a href="#cb8-276" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-279"><a href="#cb8-279" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb8-280"><a href="#cb8-280" aria-hidden="true" tabindex="-1"></a>risk_score_scanb <span class="ot">&lt;-</span> survival<span class="sc">::</span><span class="fu">coxph</span>(</span>
<span id="cb8-281"><a href="#cb8-281" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.formula</span>(<span class="fu">paste0</span>(</span>
<span id="cb8-282"><a href="#cb8-282" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Surv(time = rfs_months, event = rfs_status) ~ "</span>,</span>
<span id="cb8-283"><a href="#cb8-283" aria-hidden="true" tabindex="-1"></a>        <span class="fu">paste</span>(<span class="fu">c</span>(<span class="st">"risk_score"</span>), <span class="at">collapse =</span> <span class="st">"+"</span>)</span>
<span id="cb8-284"><a href="#cb8-284" aria-hidden="true" tabindex="-1"></a>    )),</span>
<span id="cb8-285"><a href="#cb8-285" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> endo_only_scanb</span>
<span id="cb8-286"><a href="#cb8-286" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb8-287"><a href="#cb8-287" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-288"><a href="#cb8-288" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(risk_score_scanb)</span>
<span id="cb8-289"><a href="#cb8-289" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb8-290"><a href="#cb8-290" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-291"><a href="#cb8-291" aria-hidden="true" tabindex="-1"></a>Again the risk score is predictive with a high coefficient, meaning</span>
<span id="cb8-292"><a href="#cb8-292" aria-hidden="true" tabindex="-1"></a>that the higher the score the worse it is for the patient. </span>
<span id="cb8-293"><a href="#cb8-293" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-294"><a href="#cb8-294" aria-hidden="true" tabindex="-1"></a>We now visualize a bit the risk scores developed here in terms </span>
<span id="cb8-295"><a href="#cb8-295" aria-hidden="true" tabindex="-1"></a>of the molecular landscape. @fig-risk-endo-pam50</span>
<span id="cb8-296"><a href="#cb8-296" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-299"><a href="#cb8-299" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb8-300"><a href="#cb8-300" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-risk-endo-pam50</span></span>
<span id="cb8-301"><a href="#cb8-301" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Molecular landscape of SCANB patients treated with only</span></span>
<span id="cb8-302"><a href="#cb8-302" aria-hidden="true" tabindex="-1"></a><span class="co">#|     endocrine therapy and that have ER+ breast cancer. Only patients</span></span>
<span id="cb8-303"><a href="#cb8-303" aria-hidden="true" tabindex="-1"></a><span class="co">#|     with risk score less than 2 are plotted here. There are only 26</span></span>
<span id="cb8-304"><a href="#cb8-304" aria-hidden="true" tabindex="-1"></a><span class="co">#|     patients out of 2829 with risk score higher than 2 that would skew </span></span>
<span id="cb8-305"><a href="#cb8-305" aria-hidden="true" tabindex="-1"></a><span class="co">#|     the color scale.</span></span>
<span id="cb8-306"><a href="#cb8-306" aria-hidden="true" tabindex="-1"></a>endo_only_scanb <span class="sc">%&gt;%</span></span>
<span id="cb8-307"><a href="#cb8-307" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(risk_score <span class="sc">&lt;</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb8-308"><a href="#cb8-308" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> PC3, <span class="at">y =</span> PC4, <span class="at">color =</span> risk_score)) <span class="sc">+</span></span>
<span id="cb8-309"><a href="#cb8-309" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span> </span>
<span id="cb8-310"><a href="#cb8-310" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_color_viridis_c</span>() <span class="sc">+</span> </span>
<span id="cb8-311"><a href="#cb8-311" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb8-312"><a href="#cb8-312" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">color =</span> <span class="st">"Risk score"</span>) <span class="sc">+</span></span>
<span id="cb8-313"><a href="#cb8-313" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>()</span>
<span id="cb8-314"><a href="#cb8-314" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb8-315"><a href="#cb8-315" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-316"><a href="#cb8-316" aria-hidden="true" tabindex="-1"></a>We now investigate the risk score among patients that have high ER IHC</span>
<span id="cb8-317"><a href="#cb8-317" aria-hidden="true" tabindex="-1"></a>percentage as well. We have seen that these patients have different outcomes</span>
<span id="cb8-318"><a href="#cb8-318" aria-hidden="true" tabindex="-1"></a>based on the molecular estrogen signaling. We now investigate the effect</span>
<span id="cb8-319"><a href="#cb8-319" aria-hidden="true" tabindex="-1"></a>of the risk score. </span>
<span id="cb8-320"><a href="#cb8-320" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-323"><a href="#cb8-323" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb8-324"><a href="#cb8-324" aria-hidden="true" tabindex="-1"></a>risk_score_scanb_highER <span class="ot">&lt;-</span> survival<span class="sc">::</span><span class="fu">coxph</span>(</span>
<span id="cb8-325"><a href="#cb8-325" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.formula</span>(<span class="fu">paste0</span>(</span>
<span id="cb8-326"><a href="#cb8-326" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Surv(time = rfs_months, event = rfs_status) ~ "</span>,</span>
<span id="cb8-327"><a href="#cb8-327" aria-hidden="true" tabindex="-1"></a>        <span class="fu">paste</span>(<span class="fu">c</span>(<span class="st">"risk_score"</span>), <span class="at">collapse =</span> <span class="st">"+"</span>)</span>
<span id="cb8-328"><a href="#cb8-328" aria-hidden="true" tabindex="-1"></a>    )),</span>
<span id="cb8-329"><a href="#cb8-329" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> endo_only_scanb <span class="sc">%&gt;%</span></span>
<span id="cb8-330"><a href="#cb8-330" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">filter</span>(ER.pct <span class="sc">&gt;=</span> <span class="dv">90</span> <span class="sc">&amp;</span> risk_score <span class="sc">&lt;</span> <span class="dv">2</span>)</span>
<span id="cb8-331"><a href="#cb8-331" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb8-332"><a href="#cb8-332" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-333"><a href="#cb8-333" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(risk_score_scanb_highER)</span>
<span id="cb8-334"><a href="#cb8-334" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb8-335"><a href="#cb8-335" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-336"><a href="#cb8-336" aria-hidden="true" tabindex="-1"></a>We see that the score is still predictive, meaning that even patients</span>
<span id="cb8-337"><a href="#cb8-337" aria-hidden="true" tabindex="-1"></a>that have high ER IHC percentage, there are those with higher risk. Note</span>
<span id="cb8-338"><a href="#cb8-338" aria-hidden="true" tabindex="-1"></a>though that the analysis still is a bit limited due to the followup time</span>
<span id="cb8-339"><a href="#cb8-339" aria-hidden="true" tabindex="-1"></a>that in median is <span class="in">`r format(median(endo_only_scanb$rfs_month)/12, digits = 2)`</span> </span>
<span id="cb8-340"><a href="#cb8-340" aria-hidden="true" tabindex="-1"></a>years.</span>
<span id="cb8-341"><a href="#cb8-341" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-342"><a href="#cb8-342" aria-hidden="true" tabindex="-1"></a><span class="fu">## Risk score and POETIC trial</span></span>
<span id="cb8-343"><a href="#cb8-343" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-344"><a href="#cb8-344" aria-hidden="true" tabindex="-1"></a>We can evaluate one of the two risk scores, the one using only the </span>
<span id="cb8-345"><a href="#cb8-345" aria-hidden="true" tabindex="-1"></a>principal components, in the POETIC trial data. The first question we</span>
<span id="cb8-346"><a href="#cb8-346" aria-hidden="true" tabindex="-1"></a>can address is if there is difference in scores in the baseline tissue </span>
<span id="cb8-347"><a href="#cb8-347" aria-hidden="true" tabindex="-1"></a>(@fig-poetic-risk-score).</span>
<span id="cb8-348"><a href="#cb8-348" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-351"><a href="#cb8-351" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb8-352"><a href="#cb8-352" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-poetic-risk-score</span></span>
<span id="cb8-353"><a href="#cb8-353" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Risk scores for responders and non-responders in the baseline</span></span>
<span id="cb8-354"><a href="#cb8-354" aria-hidden="true" tabindex="-1"></a><span class="co">#|     tumor tissue. </span></span>
<span id="cb8-355"><a href="#cb8-355" aria-hidden="true" tabindex="-1"></a>df_pca <span class="sc">%&gt;%</span></span>
<span id="cb8-356"><a href="#cb8-356" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb8-357"><a href="#cb8-357" aria-hidden="true" tabindex="-1"></a>        cohort <span class="sc">==</span> <span class="st">"poetic"</span> <span class="sc">&amp;</span> </span>
<span id="cb8-358"><a href="#cb8-358" aria-hidden="true" tabindex="-1"></a>            timepoint <span class="sc">==</span> <span class="st">"baseline"</span> <span class="sc">&amp;</span> </span>
<span id="cb8-359"><a href="#cb8-359" aria-hidden="true" tabindex="-1"></a>            is_responder <span class="sc">!=</span> <span class="st">"not_available"</span></span>
<span id="cb8-360"><a href="#cb8-360" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb8-361"><a href="#cb8-361" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> is_responder, <span class="at">y =</span> risk_score_pcs)) <span class="sc">+</span></span>
<span id="cb8-362"><a href="#cb8-362" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_violin</span>(<span class="at">alpha =</span> <span class="fl">0.3</span>) <span class="sc">+</span> </span>
<span id="cb8-363"><a href="#cb8-363" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_jitter</span>() <span class="sc">+</span> </span>
<span id="cb8-364"><a href="#cb8-364" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb8-365"><a href="#cb8-365" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"Response status"</span>,</span>
<span id="cb8-366"><a href="#cb8-366" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"Risk score (PCs only)"</span>,</span>
<span id="cb8-367"><a href="#cb8-367" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">"Risk score (PCs only) based on response status"</span></span>
<span id="cb8-368"><a href="#cb8-368" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb8-369"><a href="#cb8-369" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>) <span class="sc">+</span></span>
<span id="cb8-370"><a href="#cb8-370" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>()</span>
<span id="cb8-371"><a href="#cb8-371" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb8-372"><a href="#cb8-372" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-373"><a href="#cb8-373" aria-hidden="true" tabindex="-1"></a>We see in the @fig-poetic-risk-score that in average the non responders</span>
<span id="cb8-374"><a href="#cb8-374" aria-hidden="true" tabindex="-1"></a>have higher risk scores. Now if we compare the risk scores between</span>
<span id="cb8-375"><a href="#cb8-375" aria-hidden="true" tabindex="-1"></a>baseline and surgery, the slope of the line that fits for each group</span>
<span id="cb8-376"><a href="#cb8-376" aria-hidden="true" tabindex="-1"></a>separately, responder and non responder, is different. The slope</span>
<span id="cb8-377"><a href="#cb8-377" aria-hidden="true" tabindex="-1"></a>for the non responders is parallel to the identity line going through</span>
<span id="cb8-378"><a href="#cb8-378" aria-hidden="true" tabindex="-1"></a>the origin with a slope of 1, meaning that there is not much change of risk</span>
<span id="cb8-379"><a href="#cb8-379" aria-hidden="true" tabindex="-1"></a>upon the treatment. On other hand, for the responder group the slope is</span>
<span id="cb8-380"><a href="#cb8-380" aria-hidden="true" tabindex="-1"></a>decreased and has a value lower than 1, meaning that the risk is decreased</span>
<span id="cb8-381"><a href="#cb8-381" aria-hidden="true" tabindex="-1"></a>upon aromatase inhibition after two weeks (@fig-poetic-risk-base-surgery).</span>
<span id="cb8-382"><a href="#cb8-382" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-385"><a href="#cb8-385" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb8-386"><a href="#cb8-386" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-poetic-risk-base-surgery</span></span>
<span id="cb8-387"><a href="#cb8-387" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Comparison of risk scores using the PCs only before and after</span></span>
<span id="cb8-388"><a href="#cb8-388" aria-hidden="true" tabindex="-1"></a><span class="co">#|     two weeks of aromatase inhibitors in post-menopausal women. The colored</span></span>
<span id="cb8-389"><a href="#cb8-389" aria-hidden="true" tabindex="-1"></a><span class="co">#|     lines correspond do the linear regression for each response status</span></span>
<span id="cb8-390"><a href="#cb8-390" aria-hidden="true" tabindex="-1"></a><span class="co">#|     group separately. </span></span>
<span id="cb8-391"><a href="#cb8-391" aria-hidden="true" tabindex="-1"></a>df_pca <span class="sc">%&gt;%</span></span>
<span id="cb8-392"><a href="#cb8-392" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb8-393"><a href="#cb8-393" aria-hidden="true" tabindex="-1"></a>        cohort <span class="sc">==</span> <span class="st">"poetic"</span> <span class="sc">&amp;</span> </span>
<span id="cb8-394"><a href="#cb8-394" aria-hidden="true" tabindex="-1"></a>            is_responder <span class="sc">!=</span> <span class="st">"not_available"</span></span>
<span id="cb8-395"><a href="#cb8-395" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb8-396"><a href="#cb8-396" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">pivot_wider</span>(</span>
<span id="cb8-397"><a href="#cb8-397" aria-hidden="true" tabindex="-1"></a>        <span class="at">id_cols =</span> <span class="fu">c</span>(<span class="st">"patient_nb"</span>, <span class="st">"is_responder"</span>),</span>
<span id="cb8-398"><a href="#cb8-398" aria-hidden="true" tabindex="-1"></a>        <span class="at">names_from =</span> <span class="st">"timepoint"</span>,</span>
<span id="cb8-399"><a href="#cb8-399" aria-hidden="true" tabindex="-1"></a>        <span class="at">values_from =</span> <span class="st">"risk_score_pcs"</span></span>
<span id="cb8-400"><a href="#cb8-400" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb8-401"><a href="#cb8-401" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> baseline, <span class="at">y =</span> surgery, <span class="at">color =</span> is_responder)) <span class="sc">+</span></span>
<span id="cb8-402"><a href="#cb8-402" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb8-403"><a href="#cb8-403" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_abline</span>(<span class="at">slope =</span> <span class="dv">1</span>, <span class="at">intercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span> </span>
<span id="cb8-404"><a href="#cb8-404" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">formula =</span> <span class="st">"y ~ x"</span>) <span class="sc">+</span></span>
<span id="cb8-405"><a href="#cb8-405" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb8-406"><a href="#cb8-406" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"Baseline risk score"</span>,</span>
<span id="cb8-407"><a href="#cb8-407" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"Surgery risk score"</span>,</span>
<span id="cb8-408"><a href="#cb8-408" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">"Risk score (PCs only)"</span>,</span>
<span id="cb8-409"><a href="#cb8-409" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">"Is responder?"</span></span>
<span id="cb8-410"><a href="#cb8-410" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb8-411"><a href="#cb8-411" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>) <span class="sc">+</span></span>
<span id="cb8-412"><a href="#cb8-412" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>()</span>
<span id="cb8-413"><a href="#cb8-413" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-right">
      <ul class="footer-items list-unstyled">
    <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/chronchi">
      <i class="bi bi-github" role="img">
</i> 
    </a>
  </li>  
    <li class="nav-item compact">
    <a class="nav-link" href="https://twitter.com/chronchi">
      <i class="bi bi-twitter" role="img">
</i> 
    </a>
  </li>  
</ul>
    </div>
  </div>
</footer>



</body></html>