<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.0.37">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Molecular landscape: a new framework for personalized medicine in Breast Cancer. - 4&nbsp; Pathway activity in a personalized context</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./risk_score.html" rel="next">
<link href="./validation.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>

<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet">

  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
    <div class="container-fluid d-flex justify-content-between">
      <h1 class="quarto-secondary-nav-title"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Pathway activity in a personalized context</span></h1>
      <button type="button" class="quarto-btn-toggle btn" aria-label="Show secondary navigation">
        <i class="bi bi-chevron-right"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Molecular landscape: a new framework for personalized medicine in Breast Cancer.</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/chronchi/molecular_landscape" title="Source Code" class="sidebar-tool px-1"><i class="bi bi-github"></i></a>
    <a href="https://twitter.com/intent/tweet?url=|url|" title="Twitter" class="sidebar-tool px-1"><i class="bi bi-twitter"></i></a>
</div>
    </div>
      </div>
      <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">Preface</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./intro.html" class="sidebar-item-text sidebar-link">Introduction</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./surv_analysis_estrogen.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Estrogen receptor status is continuous, not dichotomous</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./pca_merging.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">A new framework for personalized medicine: integrating</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./validation.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Validating the molecular landscape</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./scoring.html" class="sidebar-item-text sidebar-link active"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Pathway activity in a personalized context</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./risk_score.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Risk score model</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./discussion.html" class="sidebar-item-text sidebar-link">Discussion</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./references.html" class="sidebar-item-text sidebar-link">References</a>
  </div>
</li>
    </ul>
    </div>
</nav>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#calculating-scores-in-neighborhoods" id="toc-calculating-scores-in-neighborhoods" class="nav-link active" data-scroll-target="#calculating-scores-in-neighborhoods"> <span class="header-section-number">4.1</span> Calculating scores in neighborhoods</a></li>
  <li><a href="#poetic-embedding" id="toc-poetic-embedding" class="nav-link" data-scroll-target="#poetic-embedding"> <span class="header-section-number">4.2</span> POETIC embedding</a>
  <ul class="collapse">
  <li><a href="#number-of-samples" id="toc-number-of-samples" class="nav-link" data-scroll-target="#number-of-samples"> <span class="header-section-number">4.2.1</span> Number of samples</a></li>
  <li><a href="#proportion-of-top-loadings" id="toc-proportion-of-top-loadings" class="nav-link" data-scroll-target="#proportion-of-top-loadings"> <span class="header-section-number">4.2.2</span> Proportion of top loadings</a></li>
  <li><a href="#embedding" id="toc-embedding" class="nav-link" data-scroll-target="#embedding"> <span class="header-section-number">4.2.3</span> Embedding</a></li>
  <li><a href="#differences-in-pc3-for-responders-and-non-responders" id="toc-differences-in-pc3-for-responders-and-non-responders" class="nav-link" data-scroll-target="#differences-in-pc3-for-responders-and-non-responders"> <span class="header-section-number">4.2.4</span> Differences in PC3 for responders and non responders</a></li>
  </ul></li>
  <li><a href="#scoring-all-samples" id="toc-scoring-all-samples" class="nav-link" data-scroll-target="#scoring-all-samples"> <span class="header-section-number">4.3</span> Scoring all samples</a></li>
  <li><a href="#analysing-patient-neighborhoods" id="toc-analysing-patient-neighborhoods" class="nav-link" data-scroll-target="#analysing-patient-neighborhoods"> <span class="header-section-number">4.4</span> Analysing patient neighborhoods</a></li>
  </ul>
<div class="toc-actions"><div><i class="bi bi-github"></i></div><div class="action-links"><p><a href="https://github.com/chronchi/molecular_landscape/issues/new" class="toc-action">Report an issue</a></p></div></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title d-none d-lg-block"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Pathway activity in a personalized context</span></h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
</div>



<div class="quarto-title-meta">

    
    
  </div>
  

</header>

<p>This chapter we present a new way to think about the personalized medicine, by incorporating patient molecular information and its context. We saw previously that by using different cohorts, we can embed new patients into a point cloud. The idea is that the neighborhood of the patients are similar in the molecular level, and we can draw conclusions based on this.</p>
<p>Each patient can be assigned a score. Scores represent a biological pathway, or in other words, a biological activity. Given a biological process, there is usually a set of genes that represent it. We can then use these set of genes to calculate scores that are proxies of this biological activity.</p>
<p>For example, in the previous chapters we used the ER signaling scores to motivate the continuity of ER signaling. This is not restricted to this specific pathway, we could have used any other list of genes representing a process.</p>
<p>By combining a score and the neighborhood of a patient, we can make direct comparisons and questions. Given a patient, how different is its score for a pathway compared to its neighbors? Is there also an alternative pathway that could be target and has a higher signaling? In this case specifically, is ER signaling more or less expressed compared to its neighbors?</p>
<p>The POETIC trial <span class="citation" data-cites="Gao2019">(<a href="references.html#ref-Gao2019" role="doc-biblioref">Gao et al. 2019</a>)</span> is a breast cancer trial that had as hypothesis neoadjuvant therapy could improve overall and recurrence free survival. They gave aromatase inhibitors (AIs) for ER+ BC patients for 2 weeks prior to surgery and then 2 weeks after surgery. Moreover, they collected a needle biopsy before the treatment started and a biopsy in the surgery. They also measured the Ki67 levels, therefore we have a proxy on how well these patients responded in the short term to AIs. Also, they sent the tissues for sequencing, so microarray data is available. This is a unique cohort, in the sense that we can understand what are the responders or not, in terms of Ki67 levels, and use the molecular data to try to understand the responsiveness. The definition of responders are those patients that have a reduction of at least 60% in Ki67 levels when comparing pre-treatment versus surgery.</p>
<p>Here we use this cohort to draw conclusions on responders and non responders and how the molecular landscape can be used. We show that some patients that are close to each other in the molecular landscape can have different pathway scores and therefore have different responses.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>* The library is already synchronized with the lockfile.</code></pre>
</div>
</div>
<section id="calculating-scores-in-neighborhoods" class="level2" data-number="4.1">
<h2 data-number="4.1" class="anchored" data-anchor-id="calculating-scores-in-neighborhoods"><span class="header-section-number">4.1</span> Calculating scores in neighborhoods</h2>
<p>Patients close to each other in the molecular landscape are somewhat molecularly close, due to how the embedding works. Therefore, we can look at patients in a neighborhood and calculate the average scores, to see what it means to be in a specific neighborhood.</p>
<p>To do this, we define a radius for a patient where all the patients within the euclidean ball with this radius will be selected and an average score distribution is calculated.</p>
<p>To calculate the average score distribution we use the <code>rstanarm</code> package. We set the prior distribution for the intercept to be a normal distribution centered at 0 with standard deviation of 1. All scores are in a range of -1 to 1, so this is a relatively flat prior for our use case.</p>
<p>The video below shows how scores change in average when comparing different neighborhoods.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code> [1] "../results/plots/scoring/movie/01.png"
 [2] "../results/plots/scoring/movie/02.png"
 [3] "../results/plots/scoring/movie/03.png"
 [4] "../results/plots/scoring/movie/04.png"
 [5] "../results/plots/scoring/movie/05.png"
 [6] "../results/plots/scoring/movie/06.png"
 [7] "../results/plots/scoring/movie/07.png"
 [8] "../results/plots/scoring/movie/08.png"
 [9] "../results/plots/scoring/movie/09.png"
[10] "../results/plots/scoring/movie/10.png"
[11] "../results/plots/scoring/movie/11.png"
[12] "../results/plots/scoring/movie/12.png"
[13] "../results/plots/scoring/movie/13.png"
[14] "../results/plots/scoring/movie/14.png"
[15] "../results/plots/scoring/movie/15.png"
[16] "../results/plots/scoring/movie/16.png"
[17] "../results/plots/scoring/movie/17.png"
[18] "../results/plots/scoring/movie/18.png"
[19] "../results/plots/scoring/movie/19.png"
[20] "../results/plots/scoring/movie/20.png"</code></pre>
</div>
</div>
<p>Going from right to left it shows how ER signaling is decreased. From top to bottom there is a reduction of proliferation, E2F targets, and a change in EMT and TGFb signaling. These are representative scores, other pathways could be used as well.</p>
</section>
<section id="poetic-embedding" class="level2" data-number="4.2">
<h2 data-number="4.2" class="anchored" data-anchor-id="poetic-embedding"><span class="header-section-number">4.2</span> POETIC embedding</h2>
<p>In this section we show the embedding of the POETIC trial samples. But first we analyse some basic properties of the dataset. The data for this dataset was downloaded and processed as described at <a href="https://chronchi.github.io/transcriptomics/">chronchi.github.io/transcriptomics</a> in the chapter AI - GSE105777.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>Total number of stable genes: 39
Total number of genes: 1008
Number of samples: 426</code></pre>
</div>
</div>
<section id="number-of-samples" class="level3" data-number="4.2.1">
<h3 data-number="4.2.1" class="anchored" data-anchor-id="number-of-samples"><span class="header-section-number">4.2.1</span> Number of samples</h3>
<p>In this dataset there are matched samples from patients that are treated and untreated. <a href="#tbl-pt-poetic-nb">Table&nbsp;<span>4.1</span></a></p>
<div class="cell">
<div class="cell-output-display">
<div id="tbl-pt-poetic-nb" class="anchored">

<table class=" lightable-classic" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; margin-left: auto; margin-right: auto;"><caption>Table 4.1:  Table showing the number of samples for treated and untreated patients </caption>
 <thead>
  <tr>
   <th style="text-align:left;"> group </th>
   <th style="text-align:right;"> n </th>
   <th style="text-align:right;"> percent </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:left;"> treated </td>
   <td style="text-align:right;"> 157 </td>
   <td style="text-align:right;"> 74 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> untreated </td>
   <td style="text-align:right;"> 56 </td>
   <td style="text-align:right;"> 26 </td>
  </tr>
</tbody>
</table>

</div>
</div>
</div>
<p>In this cohort, there is the PAM50 molecular subtype available for the untreated. <a href="#tbl-pt-poetic-pam50">Table&nbsp;<span>4.2</span></a> shows the number of patients for each subtype.</p>
<div class="cell">
<div class="cell-output-display">
<div id="tbl-pt-poetic-pam50" class="anchored">

<table class=" lightable-classic" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; margin-left: auto; margin-right: auto;"><caption>Table 4.2:  Table showing the number of molecular subtypes for each group. </caption>
 <thead>
  <tr>
   <th style="text-align:left;"> pam50 </th>
   <th style="text-align:right;"> n </th>
   <th style="text-align:right;"> percent </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:left;"> her2 </td>
   <td style="text-align:right;"> 1 </td>
   <td style="text-align:right;"> 0.235 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> luma </td>
   <td style="text-align:right;"> 77 </td>
   <td style="text-align:right;"> 18.075 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> lumb </td>
   <td style="text-align:right;"> 31 </td>
   <td style="text-align:right;"> 7.277 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> normal </td>
   <td style="text-align:right;"> 3 </td>
   <td style="text-align:right;"> 0.704 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> not_available </td>
   <td style="text-align:right;"> 314 </td>
   <td style="text-align:right;"> 73.709 </td>
  </tr>
</tbody>
</table>

</div>
</div>
</div>
<p>The numbers differ from previously, since the molecular subtype is calculated for each sample, so two molecular subtypes for each patient.</p>
</section>
<section id="proportion-of-top-loadings" class="level3" data-number="4.2.2">
<h3 data-number="4.2.2" class="anchored" data-anchor-id="proportion-of-top-loadings"><span class="header-section-number">4.2.2</span> Proportion of top loadings</h3>
<div class="cell">

</div>
<p>There are in total 1008 genes available out of the . Among them, 39 are the housekeeping genes. Out of the 36 genes missing, a total number of 8 are missing from the top 200 PC3 loadings, a proportion of 4%. A total number of 5 are missing from the top 200 PC4 loadings, a proportion of 2%. These are very small proportions, and we have seen from the last chapter that they will not affect the position of the patients in the embedding.</p>
</section>
<section id="embedding" class="level3" data-number="4.2.3">
<h3 data-number="4.2.3" class="anchored" data-anchor-id="embedding"><span class="header-section-number">4.2.3</span> Embedding</h3>
<p><a href="#fig-pca-poetic-er-pam50">Figure&nbsp;<span>4.1</span></a> shows a good mixing of the POETIC sample and how all the patients are scattered across the whole landscape. Here all samples are plotted, meaning that every two dots correspond to a single patient.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pca-poetic-er-pam50" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="scoring_files/figure-html/fig-pca-poetic-er-pam50-1.png" class="img-fluid figure-img" width="1824"></p>
<p></p><figcaption class="figure-caption">Figure 4.1: PCA embedding of all samples from TCGA, SCANB and METABRIC including the POETIC samples on top. (A) Colored by cohort, (B) colored by ER status, (C) colored by PAM50 molecular subtype.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p><a href="#fig-pca-responders-baseline">Figure&nbsp;<span>4.2</span></a> shows the embedding of baseline samples from patients that received endocrine therapy prior to surgery. Patients on the left part of the molecular landscape are considered to be non responders, this coincides with the fact this region corresponds to the ER- BC patients.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pca-responders-baseline" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="scoring_files/figure-html/fig-pca-responders-baseline-1.png" class="img-fluid figure-img" width="960"></p>
<p></p><figcaption class="figure-caption">Figure 4.2: PCA embedding of all samples from TCGA, SCANB and METABRIC including the POETIC samples on top. POETIC samples correspond to only baseline treated patients.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>And when checking the first two components, the POETIC data is closer to METABRIC, which makes sense since both are microarrays, as it can be seen in <a href="#fig-poetic-cohort">Figure&nbsp;<span>4.3</span></a>.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-poetic-cohort" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="scoring_files/figure-html/fig-poetic-cohort-1.png" class="img-fluid figure-img" width="672"></p>
<p></p><figcaption class="figure-caption">Figure 4.3: Biplot of first two PCA components from POETIC, TCGA, SCANB and METABRIC. POETIC is highlighted in the plot and has a bigger point.</figcaption><p></p>
</figure>
</div>
</div>
</div>
</section>
<section id="differences-in-pc3-for-responders-and-non-responders" class="level3" data-number="4.2.4">
<h3 data-number="4.2.4" class="anchored" data-anchor-id="differences-in-pc3-for-responders-and-non-responders"><span class="header-section-number">4.2.4</span> Differences in PC3 for responders and non responders</h3>
<p>We now also compare the differences in non responders vs responders according to the third component.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="scoring_files/figure-html/unnamed-chunk-10-1.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>List of 93
 $ line                      :List of 6
  ..$ colour       : chr "black"
  ..$ size         : num 0.5
  ..$ linetype     : num 1
  ..$ lineend      : chr "butt"
  ..$ arrow        : logi FALSE
  ..$ inherit.blank: logi TRUE
  ..- attr(*, "class")= chr [1:2] "element_line" "element"
 $ rect                      :List of 5
  ..$ fill         : chr "white"
  ..$ colour       : chr "black"
  ..$ size         : num 0.5
  ..$ linetype     : num 1
  ..$ inherit.blank: logi TRUE
  ..- attr(*, "class")= chr [1:2] "element_rect" "element"
 $ text                      :List of 11
  ..$ family       : chr ""
  ..$ face         : chr "plain"
  ..$ colour       : chr "black"
  ..$ size         : num 11
  ..$ hjust        : num 0.5
  ..$ vjust        : num 0.5
  ..$ angle        : num 0
  ..$ lineheight   : num 0.9
  ..$ margin       : 'margin' num [1:4] 0points 0points 0points 0points
  .. ..- attr(*, "unit")= int 8
  ..$ debug        : logi FALSE
  ..$ inherit.blank: logi TRUE
  ..- attr(*, "class")= chr [1:2] "element_text" "element"
 $ title                     : NULL
 $ aspect.ratio              : NULL
 $ axis.title                : NULL
 $ axis.title.x              :List of 11
  ..$ family       : NULL
  ..$ face         : NULL
  ..$ colour       : NULL
  ..$ size         : NULL
  ..$ hjust        : NULL
  ..$ vjust        : num 1
  ..$ angle        : NULL
  ..$ lineheight   : NULL
  ..$ margin       : 'margin' num [1:4] 2.75points 0points 0points 0points
  .. ..- attr(*, "unit")= int 8
  ..$ debug        : NULL
  ..$ inherit.blank: logi TRUE
  ..- attr(*, "class")= chr [1:2] "element_text" "element"
 $ axis.title.x.top          :List of 11
  ..$ family       : NULL
  ..$ face         : NULL
  ..$ colour       : NULL
  ..$ size         : NULL
  ..$ hjust        : NULL
  ..$ vjust        : num 0
  ..$ angle        : NULL
  ..$ lineheight   : NULL
  ..$ margin       : 'margin' num [1:4] 0points 0points 2.75points 0points
  .. ..- attr(*, "unit")= int 8
  ..$ debug        : NULL
  ..$ inherit.blank: logi TRUE
  ..- attr(*, "class")= chr [1:2] "element_text" "element"
 $ axis.title.x.bottom       : NULL
 $ axis.title.y              :List of 11
  ..$ family       : NULL
  ..$ face         : NULL
  ..$ colour       : NULL
  ..$ size         : NULL
  ..$ hjust        : NULL
  ..$ vjust        : num 1
  ..$ angle        : num 90
  ..$ lineheight   : NULL
  ..$ margin       : 'margin' num [1:4] 0points 2.75points 0points 0points
  .. ..- attr(*, "unit")= int 8
  ..$ debug        : NULL
  ..$ inherit.blank: logi TRUE
  ..- attr(*, "class")= chr [1:2] "element_text" "element"
 $ axis.title.y.left         : NULL
 $ axis.title.y.right        :List of 11
  ..$ family       : NULL
  ..$ face         : NULL
  ..$ colour       : NULL
  ..$ size         : NULL
  ..$ hjust        : NULL
  ..$ vjust        : num 0
  ..$ angle        : num -90
  ..$ lineheight   : NULL
  ..$ margin       : 'margin' num [1:4] 0points 0points 0points 2.75points
  .. ..- attr(*, "unit")= int 8
  ..$ debug        : NULL
  ..$ inherit.blank: logi TRUE
  ..- attr(*, "class")= chr [1:2] "element_text" "element"
 $ axis.text                 :List of 11
  ..$ family       : NULL
  ..$ face         : NULL
  ..$ colour       : chr "grey30"
  ..$ size         : 'rel' num 0.8
  ..$ hjust        : NULL
  ..$ vjust        : NULL
  ..$ angle        : NULL
  ..$ lineheight   : NULL
  ..$ margin       : NULL
  ..$ debug        : NULL
  ..$ inherit.blank: logi TRUE
  ..- attr(*, "class")= chr [1:2] "element_text" "element"
 $ axis.text.x               :List of 11
  ..$ family       : NULL
  ..$ face         : NULL
  ..$ colour       : NULL
  ..$ size         : NULL
  ..$ hjust        : NULL
  ..$ vjust        : num 1
  ..$ angle        : NULL
  ..$ lineheight   : NULL
  ..$ margin       : 'margin' num [1:4] 2.2points 0points 0points 0points
  .. ..- attr(*, "unit")= int 8
  ..$ debug        : NULL
  ..$ inherit.blank: logi TRUE
  ..- attr(*, "class")= chr [1:2] "element_text" "element"
 $ axis.text.x.top           :List of 11
  ..$ family       : NULL
  ..$ face         : NULL
  ..$ colour       : NULL
  ..$ size         : NULL
  ..$ hjust        : NULL
  ..$ vjust        : num 0
  ..$ angle        : NULL
  ..$ lineheight   : NULL
  ..$ margin       : 'margin' num [1:4] 0points 0points 2.2points 0points
  .. ..- attr(*, "unit")= int 8
  ..$ debug        : NULL
  ..$ inherit.blank: logi TRUE
  ..- attr(*, "class")= chr [1:2] "element_text" "element"
 $ axis.text.x.bottom        : NULL
 $ axis.text.y               :List of 11
  ..$ family       : NULL
  ..$ face         : NULL
  ..$ colour       : NULL
  ..$ size         : NULL
  ..$ hjust        : num 1
  ..$ vjust        : NULL
  ..$ angle        : NULL
  ..$ lineheight   : NULL
  ..$ margin       : 'margin' num [1:4] 0points 2.2points 0points 0points
  .. ..- attr(*, "unit")= int 8
  ..$ debug        : NULL
  ..$ inherit.blank: logi TRUE
  ..- attr(*, "class")= chr [1:2] "element_text" "element"
 $ axis.text.y.left          : NULL
 $ axis.text.y.right         :List of 11
  ..$ family       : NULL
  ..$ face         : NULL
  ..$ colour       : NULL
  ..$ size         : NULL
  ..$ hjust        : num 0
  ..$ vjust        : NULL
  ..$ angle        : NULL
  ..$ lineheight   : NULL
  ..$ margin       : 'margin' num [1:4] 0points 0points 0points 2.2points
  .. ..- attr(*, "unit")= int 8
  ..$ debug        : NULL
  ..$ inherit.blank: logi TRUE
  ..- attr(*, "class")= chr [1:2] "element_text" "element"
 $ axis.ticks                :List of 6
  ..$ colour       : chr "grey20"
  ..$ size         : NULL
  ..$ linetype     : NULL
  ..$ lineend      : NULL
  ..$ arrow        : logi FALSE
  ..$ inherit.blank: logi TRUE
  ..- attr(*, "class")= chr [1:2] "element_line" "element"
 $ axis.ticks.x              : NULL
 $ axis.ticks.x.top          : NULL
 $ axis.ticks.x.bottom       : NULL
 $ axis.ticks.y              : NULL
 $ axis.ticks.y.left         : NULL
 $ axis.ticks.y.right        : NULL
 $ axis.ticks.length         : 'simpleUnit' num 2.75points
  ..- attr(*, "unit")= int 8
 $ axis.ticks.length.x       : NULL
 $ axis.ticks.length.x.top   : NULL
 $ axis.ticks.length.x.bottom: NULL
 $ axis.ticks.length.y       : NULL
 $ axis.ticks.length.y.left  : NULL
 $ axis.ticks.length.y.right : NULL
 $ axis.line                 : list()
  ..- attr(*, "class")= chr [1:2] "element_blank" "element"
 $ axis.line.x               : NULL
 $ axis.line.x.top           : NULL
 $ axis.line.x.bottom        : NULL
 $ axis.line.y               : NULL
 $ axis.line.y.left          : NULL
 $ axis.line.y.right         : NULL
 $ legend.background         :List of 5
  ..$ fill         : NULL
  ..$ colour       : logi NA
  ..$ size         : NULL
  ..$ linetype     : NULL
  ..$ inherit.blank: logi TRUE
  ..- attr(*, "class")= chr [1:2] "element_rect" "element"
 $ legend.margin             : 'margin' num [1:4] 5.5points 5.5points 5.5points 5.5points
  ..- attr(*, "unit")= int 8
 $ legend.spacing            : 'simpleUnit' num 11points
  ..- attr(*, "unit")= int 8
 $ legend.spacing.x          : NULL
 $ legend.spacing.y          : NULL
 $ legend.key                :List of 5
  ..$ fill         : chr "white"
  ..$ colour       : logi NA
  ..$ size         : NULL
  ..$ linetype     : NULL
  ..$ inherit.blank: logi TRUE
  ..- attr(*, "class")= chr [1:2] "element_rect" "element"
 $ legend.key.size           : 'simpleUnit' num 1.2lines
  ..- attr(*, "unit")= int 3
 $ legend.key.height         : NULL
 $ legend.key.width          : NULL
 $ legend.text               :List of 11
  ..$ family       : NULL
  ..$ face         : NULL
  ..$ colour       : NULL
  ..$ size         : 'rel' num 0.8
  ..$ hjust        : NULL
  ..$ vjust        : NULL
  ..$ angle        : NULL
  ..$ lineheight   : NULL
  ..$ margin       : NULL
  ..$ debug        : NULL
  ..$ inherit.blank: logi TRUE
  ..- attr(*, "class")= chr [1:2] "element_text" "element"
 $ legend.text.align         : NULL
 $ legend.title              :List of 11
  ..$ family       : NULL
  ..$ face         : NULL
  ..$ colour       : NULL
  ..$ size         : NULL
  ..$ hjust        : num 0
  ..$ vjust        : NULL
  ..$ angle        : NULL
  ..$ lineheight   : NULL
  ..$ margin       : NULL
  ..$ debug        : NULL
  ..$ inherit.blank: logi TRUE
  ..- attr(*, "class")= chr [1:2] "element_text" "element"
 $ legend.title.align        : NULL
 $ legend.position           : chr "right"
 $ legend.direction          : NULL
 $ legend.justification      : chr "center"
 $ legend.box                : NULL
 $ legend.box.just           : NULL
 $ legend.box.margin         : 'margin' num [1:4] 0cm 0cm 0cm 0cm
  ..- attr(*, "unit")= int 1
 $ legend.box.background     : list()
  ..- attr(*, "class")= chr [1:2] "element_blank" "element"
 $ legend.box.spacing        : 'simpleUnit' num 11points
  ..- attr(*, "unit")= int 8
 $ panel.background          :List of 5
  ..$ fill         : chr "white"
  ..$ colour       : logi NA
  ..$ size         : NULL
  ..$ linetype     : NULL
  ..$ inherit.blank: logi TRUE
  ..- attr(*, "class")= chr [1:2] "element_rect" "element"
 $ panel.border              :List of 5
  ..$ fill         : logi NA
  ..$ colour       : chr "grey20"
  ..$ size         : NULL
  ..$ linetype     : NULL
  ..$ inherit.blank: logi TRUE
  ..- attr(*, "class")= chr [1:2] "element_rect" "element"
 $ panel.spacing             : 'simpleUnit' num 5.5points
  ..- attr(*, "unit")= int 8
 $ panel.spacing.x           : NULL
 $ panel.spacing.y           : NULL
 $ panel.grid                :List of 6
  ..$ colour       : chr "grey92"
  ..$ size         : NULL
  ..$ linetype     : NULL
  ..$ lineend      : NULL
  ..$ arrow        : logi FALSE
  ..$ inherit.blank: logi TRUE
  ..- attr(*, "class")= chr [1:2] "element_line" "element"
 $ panel.grid.major          : NULL
 $ panel.grid.minor          :List of 6
  ..$ colour       : NULL
  ..$ size         : 'rel' num 0.5
  ..$ linetype     : NULL
  ..$ lineend      : NULL
  ..$ arrow        : logi FALSE
  ..$ inherit.blank: logi TRUE
  ..- attr(*, "class")= chr [1:2] "element_line" "element"
 $ panel.grid.major.x        : NULL
 $ panel.grid.major.y        : NULL
 $ panel.grid.minor.x        : NULL
 $ panel.grid.minor.y        : NULL
 $ panel.ontop               : logi FALSE
 $ plot.background           :List of 5
  ..$ fill         : NULL
  ..$ colour       : chr "white"
  ..$ size         : NULL
  ..$ linetype     : NULL
  ..$ inherit.blank: logi TRUE
  ..- attr(*, "class")= chr [1:2] "element_rect" "element"
 $ plot.title                :List of 11
  ..$ family       : NULL
  ..$ face         : NULL
  ..$ colour       : NULL
  ..$ size         : 'rel' num 1.2
  ..$ hjust        : num 0
  ..$ vjust        : num 1
  ..$ angle        : NULL
  ..$ lineheight   : NULL
  ..$ margin       : 'margin' num [1:4] 0points 0points 5.5points 0points
  .. ..- attr(*, "unit")= int 8
  ..$ debug        : NULL
  ..$ inherit.blank: logi TRUE
  ..- attr(*, "class")= chr [1:2] "element_text" "element"
 $ plot.title.position       : chr "panel"
 $ plot.subtitle             :List of 11
  ..$ family       : NULL
  ..$ face         : NULL
  ..$ colour       : NULL
  ..$ size         : NULL
  ..$ hjust        : num 0
  ..$ vjust        : num 1
  ..$ angle        : NULL
  ..$ lineheight   : NULL
  ..$ margin       : 'margin' num [1:4] 0points 0points 5.5points 0points
  .. ..- attr(*, "unit")= int 8
  ..$ debug        : NULL
  ..$ inherit.blank: logi TRUE
  ..- attr(*, "class")= chr [1:2] "element_text" "element"
 $ plot.caption              :List of 11
  ..$ family       : NULL
  ..$ face         : NULL
  ..$ colour       : NULL
  ..$ size         : 'rel' num 0.8
  ..$ hjust        : num 1
  ..$ vjust        : num 1
  ..$ angle        : NULL
  ..$ lineheight   : NULL
  ..$ margin       : 'margin' num [1:4] 5.5points 0points 0points 0points
  .. ..- attr(*, "unit")= int 8
  ..$ debug        : NULL
  ..$ inherit.blank: logi TRUE
  ..- attr(*, "class")= chr [1:2] "element_text" "element"
 $ plot.caption.position     : chr "panel"
 $ plot.tag                  :List of 11
  ..$ family       : NULL
  ..$ face         : NULL
  ..$ colour       : NULL
  ..$ size         : 'rel' num 1.2
  ..$ hjust        : num 0.5
  ..$ vjust        : num 0.5
  ..$ angle        : NULL
  ..$ lineheight   : NULL
  ..$ margin       : NULL
  ..$ debug        : NULL
  ..$ inherit.blank: logi TRUE
  ..- attr(*, "class")= chr [1:2] "element_text" "element"
 $ plot.tag.position         : chr "topleft"
 $ plot.margin               : 'margin' num [1:4] 5.5points 5.5points 5.5points 5.5points
  ..- attr(*, "unit")= int 8
 $ strip.background          :List of 5
  ..$ fill         : chr "grey85"
  ..$ colour       : chr "grey20"
  ..$ size         : NULL
  ..$ linetype     : NULL
  ..$ inherit.blank: logi TRUE
  ..- attr(*, "class")= chr [1:2] "element_rect" "element"
 $ strip.background.x        : NULL
 $ strip.background.y        : NULL
 $ strip.placement           : chr "inside"
 $ strip.text                :List of 11
  ..$ family       : NULL
  ..$ face         : NULL
  ..$ colour       : chr "grey10"
  ..$ size         : 'rel' num 0.8
  ..$ hjust        : NULL
  ..$ vjust        : NULL
  ..$ angle        : NULL
  ..$ lineheight   : NULL
  ..$ margin       : 'margin' num [1:4] 4.4points 4.4points 4.4points 4.4points
  .. ..- attr(*, "unit")= int 8
  ..$ debug        : NULL
  ..$ inherit.blank: logi TRUE
  ..- attr(*, "class")= chr [1:2] "element_text" "element"
 $ strip.text.x              : NULL
 $ strip.text.y              :List of 11
  ..$ family       : NULL
  ..$ face         : NULL
  ..$ colour       : NULL
  ..$ size         : NULL
  ..$ hjust        : NULL
  ..$ vjust        : NULL
  ..$ angle        : num -90
  ..$ lineheight   : NULL
  ..$ margin       : NULL
  ..$ debug        : NULL
  ..$ inherit.blank: logi TRUE
  ..- attr(*, "class")= chr [1:2] "element_text" "element"
 $ strip.switch.pad.grid     : 'simpleUnit' num 2.75points
  ..- attr(*, "unit")= int 8
 $ strip.switch.pad.wrap     : 'simpleUnit' num 2.75points
  ..- attr(*, "unit")= int 8
 $ strip.text.y.left         :List of 11
  ..$ family       : NULL
  ..$ face         : NULL
  ..$ colour       : NULL
  ..$ size         : NULL
  ..$ hjust        : NULL
  ..$ vjust        : NULL
  ..$ angle        : num 90
  ..$ lineheight   : NULL
  ..$ margin       : NULL
  ..$ debug        : NULL
  ..$ inherit.blank: logi TRUE
  ..- attr(*, "class")= chr [1:2] "element_text" "element"
 - attr(*, "class")= chr [1:2] "theme" "gg"
 - attr(*, "complete")= logi TRUE
 - attr(*, "validate")= logi TRUE</code></pre>
</div>
</div>
</section>
</section>
<section id="scoring-all-samples" class="level2" data-number="4.3">
<h2 data-number="4.3" class="anchored" data-anchor-id="scoring-all-samples"><span class="header-section-number">4.3</span> Scoring all samples</h2>
<p>In the last chapters scores were showed for TCGA, SCANB and METABRIC. <a href="pca_merging.html#fig-pca-seterpr">Figure&nbsp;<span>2.18</span></a> shows how ER signaling changes as patients move across the molecular landscape. We intend to use the scores now for the POETIC samples. We first start by calculating using GSVA as well and include in the dataframe with the other cohorts. Before moving to the next chapter, we will analyse and compare the scores across the responders and non responders.</p>
<p><a href="#fig-scores-poetic-er">Figure&nbsp;<span>4.4</span></a> shows that when using the scores, it looks like in average at baseline <span class="math inline">\(SET_{ER/PR}\)</span> is lower in non responders than in responders. For the other scores there is no clear difference.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>Setting parallel calculations through a MulticoreParam back-end
with workers=20 and tasks=100.
Estimating GSVA scores for 53 gene sets.
Estimating ECDFs with Gaussian kernels
Estimating ECDFs in parallel on 20 cores

  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |=                                                                     |   2%
  |                                                                            
  |===                                                                   |   4%
  |                                                                            
  |====                                                                  |   6%
  |                                                                            
  |=====                                                                 |   8%
  |                                                                            
  |=======                                                               |   9%
  |                                                                            
  |========                                                              |  11%
  |                                                                            
  |=========                                                             |  13%
  |                                                                            
  |===========                                                           |  15%
  |                                                                            
  |============                                                          |  17%
  |                                                                            
  |=============                                                         |  19%
  |                                                                            
  |===============                                                       |  21%
  |                                                                            
  |================                                                      |  23%
  |                                                                            
  |=================                                                     |  25%
  |                                                                            
  |==================                                                    |  26%
  |                                                                            
  |====================                                                  |  28%
  |                                                                            
  |=====================                                                 |  30%
  |                                                                            
  |======================                                                |  32%
  |                                                                            
  |========================                                              |  34%
  |                                                                            
  |=========================                                             |  36%
  |                                                                            
  |==========================                                            |  38%
  |                                                                            
  |============================                                          |  40%
  |                                                                            
  |=============================                                         |  42%
  |                                                                            
  |==============================                                        |  43%
  |                                                                            
  |================================                                      |  45%
  |                                                                            
  |=================================                                     |  47%
  |                                                                            
  |==================================                                    |  49%
  |                                                                            
  |====================================                                  |  51%
  |                                                                            
  |=====================================                                 |  53%
  |                                                                            
  |======================================                                |  55%
  |                                                                            
  |========================================                              |  57%
  |                                                                            
  |=========================================                             |  58%
  |                                                                            
  |==========================================                            |  60%
  |                                                                            
  |============================================                          |  62%
  |                                                                            
  |=============================================                         |  64%
  |                                                                            
  |==============================================                        |  66%
  |                                                                            
  |================================================                      |  68%
  |                                                                            
  |=================================================                     |  70%
  |                                                                            
  |==================================================                    |  72%
  |                                                                            
  |====================================================                  |  74%
  |                                                                            
  |=====================================================                 |  75%
  |                                                                            
  |======================================================                |  77%
  |                                                                            
  |=======================================================               |  79%
  |                                                                            
  |=========================================================             |  81%
  |                                                                            
  |==========================================================            |  83%
  |                                                                            
  |===========================================================           |  85%
  |                                                                            
  |=============================================================         |  87%
  |                                                                            
  |==============================================================        |  89%
  |                                                                            
  |===============================================================       |  91%
  |                                                                            
  |=================================================================     |  92%
  |                                                                            
  |==================================================================    |  94%
  |                                                                            
  |===================================================================   |  96%
  |                                                                            
  |===================================================================== |  98%
  |                                                                            
  |======================================================================| 100%</code></pre>
</div>
</div>
<div class="cell">

</div>
<div class="cell">
<div class="cell-output-display">
<div id="fig-scores-poetic-er" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="scoring_files/figure-html/fig-scores-poetic-er-1.png" class="img-fluid figure-img" width="1248"></p>
<p></p><figcaption class="figure-caption">Figure 4.4: Baseline scores for all treated patients in the POETIC trial. G2M corresponds to a proliferation score, EMT to epithelial to mesenchymal transition score and the others are related to ER signaling.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>We see that for PC3 and SET ER/PR There are some differences in terms of the responders and non responders.</p>
<p>Let us take a closer look at those patients. We start by plotting the scatter plot of PC3 vs SET ER/PR and color by response status defined by the POETIC trial.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-scores-poetic-responders" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="scoring_files/figure-html/fig-scores-poetic-responders-1.png" class="img-fluid figure-img" width="672"></p>
<p></p><figcaption class="figure-caption">Figure 4.5: Comparison of the third component with SET ER/PR scores colored by response status.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>This figure shows that there is a correlation between third component and SET ER/PR for the POETIC trial data as well as it was already seen from the other cohorts. Moreover, it seems that the responders have higher PC scores in general. We now correlate for each subgroup (responder and non responders) the PC3 and the change in percentage for Ki67.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-scores-corr-poetic" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="scoring_files/figure-html/fig-scores-corr-poetic-1.png" class="img-fluid figure-img" width="960"></p>
<p></p><figcaption class="figure-caption">Figure 4.6: Correlation between change in Ki67 versus PC3 stratified by response group.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>Now another metric we can calculate from our dataset is the difference of the embedding positions between the different samples of the same patient. The idea is that if a patient responded well, there were bigger changes as well in the transcriptomics and that is reflected in the embedding.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-diff-embedding-poetic" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="scoring_files/figure-html/fig-diff-embedding-poetic-1.png" class="img-fluid figure-img" width="960"></p>
<p></p><figcaption class="figure-caption">Figure 4.7: Differences of the embeddings for each patient individually stratified by response status. Each dot corresponds to a single patient.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>And now we perform a comparison of these positions by using rstanarm to get the posterior distributions of the differences.</p>
<div class="cell">

</div>
<p>The figure below shows the distribution of the averages of the differences in the embedding of the projections.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-diff-embedding-poetic-averages" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="scoring_files/figure-html/fig-diff-embedding-poetic-averages-1.png" class="img-fluid figure-img" width="672"></p>
<p></p><figcaption class="figure-caption">Figure 4.8: Differences of the embeddings for each patient individually stratified by response status. Each dot corresponds to a single patient.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>And the figure below shows the posterior distribution of the differences in average.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-diff-embedding-poetic-averages-diff" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="scoring_files/figure-html/fig-diff-embedding-poetic-averages-diff-1.png" class="img-fluid figure-img" width="672"></p>
<p></p><figcaption class="figure-caption">Figure 4.9: Differences of the average differences in the embedding for each patient individually. The comparison made was non responder vs responder, which means that if non responder has a lower difference in average, the difference in difference will be negative.</figcaption><p></p>
</figure>
</div>
</div>
</div>
</section>
<section id="analysing-patient-neighborhoods" class="level2" data-number="4.4">
<h2 data-number="4.4" class="anchored" data-anchor-id="analysing-patient-neighborhoods"><span class="header-section-number">4.4</span> Analysing patient neighborhoods</h2>
<p>There are two ways to interpret the embedding. If the patient has ER+ BC and its embedding is close to the ER- BC patients, we can infer that the endocrine response might not work so well, as it was shown in <a href="#fig-pca-responders-baseline">Figure&nbsp;<span>4.2</span></a>. The other option is to compare the patient’s score to the average score of its neighborhood.</p>
<p>What is a neighborhood? When performing the embedding, global structures are not preserved usually. In this sense we only compare patients that are close to each other in an euclidean neighborhood, i.e., we calculate a radius around each sample and see what other samples are within this ball given the euclidean distance.</p>
<p>Given a neighborhood, one can calculate the posterior average score of all samples from METABRIC, SCANB and TCGA and use that as an indication of average score that we can compare other patients to. The concept is that if an ER+ BC patient has a much lower ER signaling score, it means the patients will not respond so well to endocrine therapy, as they are very different from their neighboors.</p>
<p>To showcase the ability to use the scores and infer results, we use the POETIC trial patients to compare the scores. This cohort is special in the sense that the patients are filtered based on selection criterias, meaning that the patients are clinically similar.</p>
<p>Two random patients in very close neighborhoods were selected to showcase the ability of the molecular landscape. The id of the patients are 63 and 236. <a href="#tbl-pt63-236">Table&nbsp;<span>4.3</span></a> shows the clinical features of these patients. One of them is HER2+. Patient 63 had a higher Ki67 baseline level but a very good response to endocrine therapy. Patient 236 did not respond to endocrine therapy in terms of reduction of Ki67 levels.</p>
<div class="cell">
<div class="cell-output-display">
<div id="tbl-pt63-236" class="anchored">

<table class=" lightable-classic" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; margin-left: auto; margin-right: auto;"><caption>Table 4.3:  Clinical features from patients 63 and 236. </caption>
 <thead>
  <tr>
   <th style="text-align:left;"> patient_nb </th>
   <th style="text-align:left;"> er_status </th>
   <th style="text-align:left;"> her2_status </th>
   <th style="text-align:right;"> ki67 </th>
   <th style="text-align:right;"> change_ki67 </th>
   <th style="text-align:left;"> ccca_surgery_ki67_2_7 </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:left;"> 63 </td>
   <td style="text-align:left;"> pos </td>
   <td style="text-align:left;"> Positive </td>
   <td style="text-align:right;"> 16.60702 </td>
   <td style="text-align:right;"> -69.958509 </td>
   <td style="text-align:left;"> noCCCA </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 236 </td>
   <td style="text-align:left;"> pos </td>
   <td style="text-align:left;"> Negative </td>
   <td style="text-align:right;"> 11.30906 </td>
   <td style="text-align:right;"> 5.593711 </td>
   <td style="text-align:left;"> noCCCA </td>
  </tr>
</tbody>
</table>

</div>
</div>
</div>
<p><a href="#fig-pt63-236">Figure&nbsp;<span>4.10</span></a> shows the embedding of two distinct patients in a similar neighborhood.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pt63-236" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="scoring_files/figure-html/fig-pt63-236-1.png" class="img-fluid figure-img" width="960"></p>
<p></p><figcaption class="figure-caption">Figure 4.10: Plot of two selected patients in the molecular landscape. One is a responder and the other is a non responder. The patient id numbers are 63 and 236.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>And now we calculate the average scores for each patient neighborhoods. We start by defining a radius value. After that we select the patients in the neighborhood and use <code>rstanarm</code> to get the posterior distribution of the average score in the neighborhood. For this we use the function <code>stan_glm</code> to calculate the average. This is effective because it can be paired with the package <code>tidybayes</code> for plotting. It makes extremely easy to deal with posterior data. For the average we use a normal prior centered at 0 with 1 standard deviation, since all scores are between -1 and 1.</p>
<div class="cell">

</div>
<p>Patient 63 <a href="#fig-pt63">Figure&nbsp;<span>4.11</span></a> is considered a responder. According to the scores, when comparing the estrogen early signature to its average distribution in the neighborhood, the score is higher.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pt63" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="scoring_files/figure-html/fig-pt63-1.png" class="img-fluid figure-img" width="960"></p>
<p></p><figcaption class="figure-caption">Figure 4.11: Posterior distribution of the average scores in the neighborhood of patient 236 from POETIC trial. Each dot corresponds to a 1% quantile.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>Patient 236 <a href="#fig-pt236">Figure&nbsp;<span>4.12</span></a> was considered a non responder. According to the scores, when comparing the estrogen early signature to its average distribution in the neighborhood, the score is lower, being in the 5% quantile.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pt236" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="scoring_files/figure-html/fig-pt236-1.png" class="img-fluid figure-img" width="960"></p>
<p></p><figcaption class="figure-caption">Figure 4.12: Posterior distribution of the average scores in the neighborhood of patient 236 from POETIC trial. Each dot corresponds to a 1% quantile.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>When comparing these two patients, there is also a difference in the androgen response score, which could be a reflection of different estrogen signaling.</p>


<!-- -->

<div id="refs" class="references csl-bib-body hanging-indent" role="doc-bibliography" style="display: none">
<div id="ref-Gao2019" class="csl-entry" role="doc-biblioentry">
Gao, Qiong, and Elena López-Knowles, Maggie Chon U. Cheang, James Morden, Ricardo Ribas, Kally Sidhu, David Evans, et al. 2019. <span>“Impact of Aromatase Inhibitor Treatment on Global Gene Expression and Its Association with Antiproliferative Response in <span>ER</span><span class="math inline">\(\mathplus\)</span> Breast Cancer in Postmenopausal Patients.”</span> <em>Breast Cancer Research</em> 22 (1). <a href="https://doi.org/10.1186/s13058-019-1223-z">https://doi.org/10.1186/s13058-019-1223-z</a>.
</div>
</div>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      let href = ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./validation.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Validating the molecular landscape</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./risk_score.html" class="pagination-link">
        <span class="nav-page-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Risk score model</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb6" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu"># Pathway activity in a personalized context</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>This chapter we present a new way to think about the personalized medicine,</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>by incorporating patient molecular information and its context. </span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>We saw previously that by using different cohorts, we can embed new patients</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>into a point cloud. The idea is that the neighborhood of the patients </span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>are similar in the molecular level, and we can draw conclusions based on</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>this. </span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>Each patient can be assigned a score. Scores represent a biological</span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a>pathway, or in other words, a biological activity. Given a biological</span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>process, there is usually a set of genes that represent it. We</span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a>can then use these set of genes to calculate scores that are </span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a>proxies of this biological activity.</span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a>For example, in the previous chapters we used the ER signaling</span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a>scores to motivate the continuity of ER signaling. This is not</span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a>restricted to this specific pathway, we could have used any</span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a>other list of genes representing a process. </span>
<span id="cb6-20"><a href="#cb6-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-21"><a href="#cb6-21" aria-hidden="true" tabindex="-1"></a>By combining a score and the neighborhood of a patient, we </span>
<span id="cb6-22"><a href="#cb6-22" aria-hidden="true" tabindex="-1"></a>can make direct comparisons and questions. Given a patient,</span>
<span id="cb6-23"><a href="#cb6-23" aria-hidden="true" tabindex="-1"></a>how different is its score for a pathway compared to its</span>
<span id="cb6-24"><a href="#cb6-24" aria-hidden="true" tabindex="-1"></a>neighbors? Is there also an alternative pathway that could</span>
<span id="cb6-25"><a href="#cb6-25" aria-hidden="true" tabindex="-1"></a>be target and has a higher signaling? In this case </span>
<span id="cb6-26"><a href="#cb6-26" aria-hidden="true" tabindex="-1"></a>specifically, is ER signaling more or less expressed compared</span>
<span id="cb6-27"><a href="#cb6-27" aria-hidden="true" tabindex="-1"></a>to its neighbors? </span>
<span id="cb6-28"><a href="#cb6-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-29"><a href="#cb6-29" aria-hidden="true" tabindex="-1"></a>The POETIC trial <span class="co">[</span><span class="ot">@Gao2019</span><span class="co">]</span> </span>
<span id="cb6-30"><a href="#cb6-30" aria-hidden="true" tabindex="-1"></a>is a breast cancer trial that had as hypothesis</span>
<span id="cb6-31"><a href="#cb6-31" aria-hidden="true" tabindex="-1"></a>neoadjuvant therapy could improve overall and recurrence free</span>
<span id="cb6-32"><a href="#cb6-32" aria-hidden="true" tabindex="-1"></a>survival. They gave aromatase inhibitors (AIs) for ER+ BC patients</span>
<span id="cb6-33"><a href="#cb6-33" aria-hidden="true" tabindex="-1"></a>for 2 weeks prior to surgery and then 2 weeks after surgery.</span>
<span id="cb6-34"><a href="#cb6-34" aria-hidden="true" tabindex="-1"></a>Moreover, they collected a needle biopsy before the</span>
<span id="cb6-35"><a href="#cb6-35" aria-hidden="true" tabindex="-1"></a>treatment started and a biopsy in the surgery. They</span>
<span id="cb6-36"><a href="#cb6-36" aria-hidden="true" tabindex="-1"></a>also measured the Ki67 levels, therefore we have a proxy</span>
<span id="cb6-37"><a href="#cb6-37" aria-hidden="true" tabindex="-1"></a>on how well these patients responded in the short term to </span>
<span id="cb6-38"><a href="#cb6-38" aria-hidden="true" tabindex="-1"></a>AIs. Also, they sent the tissues for sequencing, so </span>
<span id="cb6-39"><a href="#cb6-39" aria-hidden="true" tabindex="-1"></a>microarray data is available. This is a unique </span>
<span id="cb6-40"><a href="#cb6-40" aria-hidden="true" tabindex="-1"></a>cohort, in the sense that we can understand what are the</span>
<span id="cb6-41"><a href="#cb6-41" aria-hidden="true" tabindex="-1"></a>responders or not, in terms of Ki67 levels, and use</span>
<span id="cb6-42"><a href="#cb6-42" aria-hidden="true" tabindex="-1"></a>the molecular data to try to understand the responsiveness. </span>
<span id="cb6-43"><a href="#cb6-43" aria-hidden="true" tabindex="-1"></a>The definition of responders are those patients that have a </span>
<span id="cb6-44"><a href="#cb6-44" aria-hidden="true" tabindex="-1"></a>reduction of at least 60\%</span>
<span id="cb6-45"><a href="#cb6-45" aria-hidden="true" tabindex="-1"></a>in Ki67 levels when comparing pre-treatment versus surgery. </span>
<span id="cb6-46"><a href="#cb6-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-47"><a href="#cb6-47" aria-hidden="true" tabindex="-1"></a>Here we use this cohort to draw conclusions on responders</span>
<span id="cb6-48"><a href="#cb6-48" aria-hidden="true" tabindex="-1"></a>and non responders and how the molecular landscape can be </span>
<span id="cb6-49"><a href="#cb6-49" aria-hidden="true" tabindex="-1"></a>used. We show that some patients that are close to each</span>
<span id="cb6-50"><a href="#cb6-50" aria-hidden="true" tabindex="-1"></a>other in the molecular landscape can have different</span>
<span id="cb6-51"><a href="#cb6-51" aria-hidden="true" tabindex="-1"></a>pathway scores and therefore have different responses.</span>
<span id="cb6-52"><a href="#cb6-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-53"><a href="#cb6-53" aria-hidden="true" tabindex="-1"></a><span class="in">```{r setup}</span></span>
<span id="cb6-54"><a href="#cb6-54" aria-hidden="true" tabindex="-1"></a>renv<span class="sc">::</span><span class="fu">restore</span>()</span>
<span id="cb6-55"><a href="#cb6-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-56"><a href="#cb6-56" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb6-57"><a href="#cb6-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-58"><a href="#cb6-58" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb6-59"><a href="#cb6-59" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(PCAtools)</span>
<span id="cb6-60"><a href="#cb6-60" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidybayes)</span>
<span id="cb6-61"><a href="#cb6-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-62"><a href="#cb6-62" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GSVA)</span>
<span id="cb6-63"><a href="#cb6-63" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SummarizedExperiment)</span>
<span id="cb6-64"><a href="#cb6-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-65"><a href="#cb6-65" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(rstanarm)</span>
<span id="cb6-66"><a href="#cb6-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-67"><a href="#cb6-67" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"utils.R"</span>)</span>
<span id="cb6-68"><a href="#cb6-68" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"first_run.R"</span>)</span>
<span id="cb6-69"><a href="#cb6-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-70"><a href="#cb6-70" aria-hidden="true" tabindex="-1"></a><span class="co"># Here we are using rds files from previous chapters, so we source</span></span>
<span id="cb6-71"><a href="#cb6-71" aria-hidden="true" tabindex="-1"></a><span class="co"># in any case the load_rds_files.R and exclude all the associated files</span></span>
<span id="cb6-72"><a href="#cb6-72" aria-hidden="true" tabindex="-1"></a><span class="co"># that are generated in this current chapter</span></span>
<span id="cb6-73"><a href="#cb6-73" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (first_run){</span>
<span id="cb6-74"><a href="#cb6-74" aria-hidden="true" tabindex="-1"></a>    load_at_setup <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb6-75"><a href="#cb6-75" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb6-76"><a href="#cb6-76" aria-hidden="true" tabindex="-1"></a>    load_at_setup <span class="ot">&lt;-</span> <span class="cn">TRUE</span></span>
<span id="cb6-77"><a href="#cb6-77" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb6-78"><a href="#cb6-78" aria-hidden="true" tabindex="-1"></a>name_document <span class="ot">&lt;-</span> <span class="st">"scoring"</span></span>
<span id="cb6-79"><a href="#cb6-79" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"load_rds_files.R"</span>)</span>
<span id="cb6-80"><a href="#cb6-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-81"><a href="#cb6-81" aria-hidden="true" tabindex="-1"></a><span class="co"># by setting the dev to png and pdf, this saves the figures in a specific</span></span>
<span id="cb6-82"><a href="#cb6-82" aria-hidden="true" tabindex="-1"></a><span class="co"># folder in both formats. moreover, since png is coming first, it shows</span></span>
<span id="cb6-83"><a href="#cb6-83" aria-hidden="true" tabindex="-1"></a><span class="co"># this figure when rendering the html. What is nice about this is that it</span></span>
<span id="cb6-84"><a href="#cb6-84" aria-hidden="true" tabindex="-1"></a><span class="co"># inherits the properties from the chunk to save the figure, so no need</span></span>
<span id="cb6-85"><a href="#cb6-85" aria-hidden="true" tabindex="-1"></a><span class="co"># to use ggplot2::ggsave to save the plots. this also works </span></span>
<span id="cb6-86"><a href="#cb6-86" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">dev =</span> <span class="fu">c</span>(<span class="st">'png'</span>, <span class="st">'pdf'</span>, <span class="st">'svg'</span>))</span>
<span id="cb6-87"><a href="#cb6-87" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">bitmapType =</span> <span class="st">'cairo'</span>)</span>
<span id="cb6-88"><a href="#cb6-88" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb6-89"><a href="#cb6-89" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-90"><a href="#cb6-90" aria-hidden="true" tabindex="-1"></a><span class="fu">## Calculating scores in neighborhoods</span></span>
<span id="cb6-91"><a href="#cb6-91" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-92"><a href="#cb6-92" aria-hidden="true" tabindex="-1"></a>Patients close to each other in the molecular landscape are somewhat</span>
<span id="cb6-93"><a href="#cb6-93" aria-hidden="true" tabindex="-1"></a>molecularly close, due to how the embedding works. Therefore, we can </span>
<span id="cb6-94"><a href="#cb6-94" aria-hidden="true" tabindex="-1"></a>look at patients in a neighborhood and calculate the average scores,</span>
<span id="cb6-95"><a href="#cb6-95" aria-hidden="true" tabindex="-1"></a>to see what it means to be in a specific neighborhood. </span>
<span id="cb6-96"><a href="#cb6-96" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-97"><a href="#cb6-97" aria-hidden="true" tabindex="-1"></a>To do this, we define a radius for a patient where all the patients</span>
<span id="cb6-98"><a href="#cb6-98" aria-hidden="true" tabindex="-1"></a>within the euclidean ball with this radius will be selected and an </span>
<span id="cb6-99"><a href="#cb6-99" aria-hidden="true" tabindex="-1"></a>average score distribution is calculated.</span>
<span id="cb6-100"><a href="#cb6-100" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-101"><a href="#cb6-101" aria-hidden="true" tabindex="-1"></a>To calculate the average score distribution we use the <span class="in">`rstanarm`</span> package. </span>
<span id="cb6-102"><a href="#cb6-102" aria-hidden="true" tabindex="-1"></a>We set the prior distribution for the intercept to be a normal distribution</span>
<span id="cb6-103"><a href="#cb6-103" aria-hidden="true" tabindex="-1"></a>centered at 0 with standard deviation of 1. All scores are in a range of</span>
<span id="cb6-104"><a href="#cb6-104" aria-hidden="true" tabindex="-1"></a>-1 to 1, so this is a relatively flat prior for our use case.</span>
<span id="cb6-105"><a href="#cb6-105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-106"><a href="#cb6-106" aria-hidden="true" tabindex="-1"></a>The video below shows how scores change in average when comparing different</span>
<span id="cb6-107"><a href="#cb6-107" aria-hidden="true" tabindex="-1"></a>neighborhoods. </span>
<span id="cb6-108"><a href="#cb6-108" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-109"><a href="#cb6-109" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb6-110"><a href="#cb6-110" aria-hidden="true" tabindex="-1"></a>scores_to_use <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb6-111"><a href="#cb6-111" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Estrogen response early"</span>,</span>
<span id="cb6-112"><a href="#cb6-112" aria-hidden="true" tabindex="-1"></a>    <span class="st">"E2F targets"</span>,</span>
<span id="cb6-113"><a href="#cb6-113" aria-hidden="true" tabindex="-1"></a>    <span class="st">"P53 pathway"</span>,</span>
<span id="cb6-114"><a href="#cb6-114" aria-hidden="true" tabindex="-1"></a>    <span class="st">"EMT"</span>,</span>
<span id="cb6-115"><a href="#cb6-115" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Androgen response"</span>,</span>
<span id="cb6-116"><a href="#cb6-116" aria-hidden="true" tabindex="-1"></a>    <span class="st">"TGFb signaling"</span>,</span>
<span id="cb6-117"><a href="#cb6-117" aria-hidden="true" tabindex="-1"></a>    <span class="st">"SET ER/PR"</span></span>
<span id="cb6-118"><a href="#cb6-118" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> <span class="st">`</span><span class="at">names&lt;-</span><span class="st">`</span>(<span class="fu">c</span>(</span>
<span id="cb6-119"><a href="#cb6-119" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_ESTROGEN_RESPONSE_EARLY"</span>,</span>
<span id="cb6-120"><a href="#cb6-120" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_E2F_TARGETS"</span>,</span>
<span id="cb6-121"><a href="#cb6-121" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_P53_PATHWAY"</span>,</span>
<span id="cb6-122"><a href="#cb6-122" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION"</span>,</span>
<span id="cb6-123"><a href="#cb6-123" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_ANDROGEN_RESPONSE"</span>,</span>
<span id="cb6-124"><a href="#cb6-124" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_TGF_BETA_SIGNALING"</span>,</span>
<span id="cb6-125"><a href="#cb6-125" aria-hidden="true" tabindex="-1"></a>    <span class="st">"SET_ERPR"</span></span>
<span id="cb6-126"><a href="#cb6-126" aria-hidden="true" tabindex="-1"></a>))</span>
<span id="cb6-127"><a href="#cb6-127" aria-hidden="true" tabindex="-1"></a>df_pca <span class="ot">&lt;-</span> df_pca <span class="sc">%&gt;%</span> </span>
<span id="cb6-128"><a href="#cb6-128" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb6-129"><a href="#cb6-129" aria-hidden="true" tabindex="-1"></a>        pam50 <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"normal"</span>, <span class="st">"luma"</span>, <span class="st">"lumb"</span>, <span class="st">"basal"</span>, <span class="st">"her2"</span>)</span>
<span id="cb6-130"><a href="#cb6-130" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span> </span>
<span id="cb6-131"><a href="#cb6-131" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">rownames&lt;-</span><span class="st">`</span>(.[, <span class="st">"sample_name"</span>])</span>
<span id="cb6-132"><a href="#cb6-132" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-133"><a href="#cb6-133" aria-hidden="true" tabindex="-1"></a><span class="co"># we first get random samples from the path determined below</span></span>
<span id="cb6-134"><a href="#cb6-134" aria-hidden="true" tabindex="-1"></a>pc3_values <span class="ot">&lt;-</span> <span class="fl">2.5</span> <span class="sc">+</span> <span class="fu">rnorm</span>(<span class="dv">0</span>, <span class="fl">0.1</span>, <span class="at">n =</span> <span class="dv">20</span>)</span>
<span id="cb6-135"><a href="#cb6-135" aria-hidden="true" tabindex="-1"></a>pc4_values <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="at">from =</span> <span class="sc">-</span><span class="dv">4</span>, <span class="at">to =</span> <span class="dv">4</span>, <span class="at">length.out =</span> <span class="dv">20</span>)</span>
<span id="cb6-136"><a href="#cb6-136" aria-hidden="true" tabindex="-1"></a>pca_values <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">PC3 =</span> pc3_values, <span class="at">PC4 =</span> pc4_values) <span class="sc">%&gt;%</span> as.matrix</span>
<span id="cb6-137"><a href="#cb6-137" aria-hidden="true" tabindex="-1"></a>samples_top_down <span class="ot">&lt;-</span> <span class="fu">apply</span>(pca_values, <span class="dv">1</span>, get_closest_sample, <span class="at">df_pca =</span> df_pca)</span>
<span id="cb6-138"><a href="#cb6-138" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-139"><a href="#cb6-139" aria-hidden="true" tabindex="-1"></a>pc3_values <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="at">from =</span> <span class="sc">-</span><span class="fl">8.5</span>, <span class="at">to =</span> <span class="dv">1</span>, <span class="at">length.out =</span> <span class="dv">20</span>)</span>
<span id="cb6-140"><a href="#cb6-140" aria-hidden="true" tabindex="-1"></a>pc4_values <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(<span class="dv">1</span>, <span class="fl">0.1</span>, <span class="at">n =</span> <span class="dv">20</span>)</span>
<span id="cb6-141"><a href="#cb6-141" aria-hidden="true" tabindex="-1"></a>pca_values <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">PC3 =</span> pc3_values, <span class="at">PC4 =</span> pc4_values) <span class="sc">%&gt;%</span> as.matrix</span>
<span id="cb6-142"><a href="#cb6-142" aria-hidden="true" tabindex="-1"></a>samples_left_right <span class="ot">&lt;-</span> <span class="fu">apply</span>(pca_values, <span class="dv">1</span>, get_closest_sample, <span class="at">df_pca =</span> df_pca)</span>
<span id="cb6-143"><a href="#cb6-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-144"><a href="#cb6-144" aria-hidden="true" tabindex="-1"></a>all_samples <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb6-145"><a href="#cb6-145" aria-hidden="true" tabindex="-1"></a>    <span class="at">top_down =</span> samples_top_down,</span>
<span id="cb6-146"><a href="#cb6-146" aria-hidden="true" tabindex="-1"></a>    <span class="at">left_right =</span> <span class="fu">rev</span>(samples_left_right)</span>
<span id="cb6-147"><a href="#cb6-147" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb6-148"><a href="#cb6-148" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-149"><a href="#cb6-149" aria-hidden="true" tabindex="-1"></a>pca_values <span class="ot">&lt;-</span> <span class="fu">lapply</span>(</span>
<span id="cb6-150"><a href="#cb6-150" aria-hidden="true" tabindex="-1"></a>    all_samples, </span>
<span id="cb6-151"><a href="#cb6-151" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(x, df_pca){</span>
<span id="cb6-152"><a href="#cb6-152" aria-hidden="true" tabindex="-1"></a>        df_pca <span class="sc">%&gt;%</span> </span>
<span id="cb6-153"><a href="#cb6-153" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">select</span>(PC3, PC4) <span class="sc">%&gt;%</span></span>
<span id="cb6-154"><a href="#cb6-154" aria-hidden="true" tabindex="-1"></a>            as.matrix <span class="sc">%&gt;%</span></span>
<span id="cb6-155"><a href="#cb6-155" aria-hidden="true" tabindex="-1"></a>            .[x, ]</span>
<span id="cb6-156"><a href="#cb6-156" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb6-157"><a href="#cb6-157" aria-hidden="true" tabindex="-1"></a>    <span class="at">df_pca =</span> df_pca</span>
<span id="cb6-158"><a href="#cb6-158" aria-hidden="true" tabindex="-1"></a>) </span>
<span id="cb6-159"><a href="#cb6-159" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-160"><a href="#cb6-160" aria-hidden="true" tabindex="-1"></a>scores_for_movie <span class="ot">&lt;-</span> <span class="fu">mapply</span>(</span>
<span id="cb6-161"><a href="#cb6-161" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(</span>
<span id="cb6-162"><a href="#cb6-162" aria-hidden="true" tabindex="-1"></a>        pca_value, </span>
<span id="cb6-163"><a href="#cb6-163" aria-hidden="true" tabindex="-1"></a>        which_direction, </span>
<span id="cb6-164"><a href="#cb6-164" aria-hidden="true" tabindex="-1"></a>        df_pca, </span>
<span id="cb6-165"><a href="#cb6-165" aria-hidden="true" tabindex="-1"></a>        scores_to_use,</span>
<span id="cb6-166"><a href="#cb6-166" aria-hidden="true" tabindex="-1"></a>        radius,</span>
<span id="cb6-167"><a href="#cb6-167" aria-hidden="true" tabindex="-1"></a>        base_size</span>
<span id="cb6-168"><a href="#cb6-168" aria-hidden="true" tabindex="-1"></a>    ){</span>
<span id="cb6-169"><a href="#cb6-169" aria-hidden="true" tabindex="-1"></a>        <span class="fu">sapply</span>(</span>
<span id="cb6-170"><a href="#cb6-170" aria-hidden="true" tabindex="-1"></a>            <span class="fu">rownames</span>(pca_value), </span>
<span id="cb6-171"><a href="#cb6-171" aria-hidden="true" tabindex="-1"></a>            get_patient_scores_distributions,</span>
<span id="cb6-172"><a href="#cb6-172" aria-hidden="true" tabindex="-1"></a>            <span class="at">df_pca =</span> df_pca,</span>
<span id="cb6-173"><a href="#cb6-173" aria-hidden="true" tabindex="-1"></a>            <span class="at">scores_to_use =</span> scores_to_use,</span>
<span id="cb6-174"><a href="#cb6-174" aria-hidden="true" tabindex="-1"></a>            <span class="at">which_direction =</span> which_direction,</span>
<span id="cb6-175"><a href="#cb6-175" aria-hidden="true" tabindex="-1"></a>            <span class="at">radius =</span> radius,</span>
<span id="cb6-176"><a href="#cb6-176" aria-hidden="true" tabindex="-1"></a>            <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>, </span>
<span id="cb6-177"><a href="#cb6-177" aria-hidden="true" tabindex="-1"></a>            <span class="at">simplify =</span> <span class="cn">FALSE</span></span>
<span id="cb6-178"><a href="#cb6-178" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb6-179"><a href="#cb6-179" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb6-180"><a href="#cb6-180" aria-hidden="true" tabindex="-1"></a>    <span class="at">pca_value =</span> pca_values,</span>
<span id="cb6-181"><a href="#cb6-181" aria-hidden="true" tabindex="-1"></a>    <span class="at">which_direction =</span> <span class="fu">names</span>(pca_values),</span>
<span id="cb6-182"><a href="#cb6-182" aria-hidden="true" tabindex="-1"></a>    <span class="at">MoreArgs =</span> <span class="fu">list</span>(</span>
<span id="cb6-183"><a href="#cb6-183" aria-hidden="true" tabindex="-1"></a>        <span class="at">df_pca =</span> df_pca, </span>
<span id="cb6-184"><a href="#cb6-184" aria-hidden="true" tabindex="-1"></a>        <span class="at">scores_to_use =</span> scores_to_use,</span>
<span id="cb6-185"><a href="#cb6-185" aria-hidden="true" tabindex="-1"></a>        <span class="at">radius =</span> <span class="dv">1</span></span>
<span id="cb6-186"><a href="#cb6-186" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb6-187"><a href="#cb6-187" aria-hidden="true" tabindex="-1"></a>    <span class="at">SIMPLIFY =</span> <span class="cn">FALSE</span>,</span>
<span id="cb6-188"><a href="#cb6-188" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span></span>
<span id="cb6-189"><a href="#cb6-189" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb6-190"><a href="#cb6-190" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-191"><a href="#cb6-191" aria-hidden="true" tabindex="-1"></a>plots_scores_for_movie <span class="ot">&lt;-</span> <span class="fu">mapply</span>(</span>
<span id="cb6-192"><a href="#cb6-192" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(</span>
<span id="cb6-193"><a href="#cb6-193" aria-hidden="true" tabindex="-1"></a>        avg_rstan_samples, </span>
<span id="cb6-194"><a href="#cb6-194" aria-hidden="true" tabindex="-1"></a>        which_direction,</span>
<span id="cb6-195"><a href="#cb6-195" aria-hidden="true" tabindex="-1"></a>        ...</span>
<span id="cb6-196"><a href="#cb6-196" aria-hidden="true" tabindex="-1"></a>    ){</span>
<span id="cb6-197"><a href="#cb6-197" aria-hidden="true" tabindex="-1"></a>        <span class="fu">sapply</span>(</span>
<span id="cb6-198"><a href="#cb6-198" aria-hidden="true" tabindex="-1"></a>            avg_rstan_samples, </span>
<span id="cb6-199"><a href="#cb6-199" aria-hidden="true" tabindex="-1"></a>            get_plot_patient_distribution,</span>
<span id="cb6-200"><a href="#cb6-200" aria-hidden="true" tabindex="-1"></a>            <span class="at">which_direction =</span> which_direction,</span>
<span id="cb6-201"><a href="#cb6-201" aria-hidden="true" tabindex="-1"></a>            ...,</span>
<span id="cb6-202"><a href="#cb6-202" aria-hidden="true" tabindex="-1"></a>            <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>, </span>
<span id="cb6-203"><a href="#cb6-203" aria-hidden="true" tabindex="-1"></a>            <span class="at">simplify =</span> <span class="cn">FALSE</span></span>
<span id="cb6-204"><a href="#cb6-204" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb6-205"><a href="#cb6-205" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb6-206"><a href="#cb6-206" aria-hidden="true" tabindex="-1"></a>    <span class="at">avg_rstan_samples =</span> scores_for_movie,</span>
<span id="cb6-207"><a href="#cb6-207" aria-hidden="true" tabindex="-1"></a>    <span class="at">which_direction =</span> <span class="fu">names</span>(scores_for_movie),</span>
<span id="cb6-208"><a href="#cb6-208" aria-hidden="true" tabindex="-1"></a>    <span class="at">MoreArgs =</span> <span class="fu">list</span>(</span>
<span id="cb6-209"><a href="#cb6-209" aria-hidden="true" tabindex="-1"></a>        <span class="at">scores_to_use =</span> scores_to_use,</span>
<span id="cb6-210"><a href="#cb6-210" aria-hidden="true" tabindex="-1"></a>        <span class="at">base_size =</span> <span class="dv">15</span>,</span>
<span id="cb6-211"><a href="#cb6-211" aria-hidden="true" tabindex="-1"></a>        <span class="at">size_dots =</span> <span class="dv">3</span></span>
<span id="cb6-212"><a href="#cb6-212" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb6-213"><a href="#cb6-213" aria-hidden="true" tabindex="-1"></a>    <span class="at">SIMPLIFY =</span> <span class="cn">FALSE</span>,</span>
<span id="cb6-214"><a href="#cb6-214" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span></span>
<span id="cb6-215"><a href="#cb6-215" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb6-216"><a href="#cb6-216" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-217"><a href="#cb6-217" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-218"><a href="#cb6-218" aria-hidden="true" tabindex="-1"></a>plots_molecular_landscape <span class="ot">&lt;-</span> <span class="fu">plot_selected_samples</span>(</span>
<span id="cb6-219"><a href="#cb6-219" aria-hidden="true" tabindex="-1"></a>    df_pca,</span>
<span id="cb6-220"><a href="#cb6-220" aria-hidden="true" tabindex="-1"></a>    pca_values,</span>
<span id="cb6-221"><a href="#cb6-221" aria-hidden="true" tabindex="-1"></a>    <span class="at">scores_plots_movie =</span> plots_scores_for_movie,</span>
<span id="cb6-222"><a href="#cb6-222" aria-hidden="true" tabindex="-1"></a>    <span class="at">size_line =</span> <span class="fl">0.5</span>,</span>
<span id="cb6-223"><a href="#cb6-223" aria-hidden="true" tabindex="-1"></a>    <span class="at">radius =</span> <span class="dv">1</span>,</span>
<span id="cb6-224"><a href="#cb6-224" aria-hidden="true" tabindex="-1"></a>    <span class="at">size_points =</span> <span class="dv">3</span>,</span>
<span id="cb6-225"><a href="#cb6-225" aria-hidden="true" tabindex="-1"></a>    <span class="at">title_plot =</span> <span class="st">"Molecular landscape"</span></span>
<span id="cb6-226"><a href="#cb6-226" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb6-227"><a href="#cb6-227" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-228"><a href="#cb6-228" aria-hidden="true" tabindex="-1"></a>plots_estimates_tog <span class="ot">&lt;-</span> <span class="fu">mapply</span>(</span>
<span id="cb6-229"><a href="#cb6-229" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(plot1, plot2){</span>
<span id="cb6-230"><a href="#cb6-230" aria-hidden="true" tabindex="-1"></a>        cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(plot1, plot2, <span class="at">nrow=</span><span class="dv">2</span>)</span>
<span id="cb6-231"><a href="#cb6-231" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb6-232"><a href="#cb6-232" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot1 =</span> plots_scores_for_movie<span class="sc">$</span>top_down,</span>
<span id="cb6-233"><a href="#cb6-233" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot2 =</span> plots_scores_for_movie<span class="sc">$</span>left_right,</span>
<span id="cb6-234"><a href="#cb6-234" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">FALSE</span>,</span>
<span id="cb6-235"><a href="#cb6-235" aria-hidden="true" tabindex="-1"></a>    <span class="at">SIMPLIFY =</span> <span class="cn">FALSE</span></span>
<span id="cb6-236"><a href="#cb6-236" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb6-237"><a href="#cb6-237" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-238"><a href="#cb6-238" aria-hidden="true" tabindex="-1"></a>final_plots <span class="ot">&lt;-</span> <span class="fu">lapply</span>(</span>
<span id="cb6-239"><a href="#cb6-239" aria-hidden="true" tabindex="-1"></a>    <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(plots_estimates_tog),</span>
<span id="cb6-240"><a href="#cb6-240" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(i) {</span>
<span id="cb6-241"><a href="#cb6-241" aria-hidden="true" tabindex="-1"></a>        cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(</span>
<span id="cb6-242"><a href="#cb6-242" aria-hidden="true" tabindex="-1"></a>            plots_molecular_landscape[[i]],</span>
<span id="cb6-243"><a href="#cb6-243" aria-hidden="true" tabindex="-1"></a>            plots_estimates_tog[[i]], </span>
<span id="cb6-244"><a href="#cb6-244" aria-hidden="true" tabindex="-1"></a>            <span class="at">ncol =</span> <span class="dv">2</span>, </span>
<span id="cb6-245"><a href="#cb6-245" aria-hidden="true" tabindex="-1"></a>            <span class="at">rel_widths =</span> <span class="fu">c</span>(<span class="fl">1.8</span>, <span class="dv">1</span>)</span>
<span id="cb6-246"><a href="#cb6-246" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb6-247"><a href="#cb6-247" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb6-248"><a href="#cb6-248" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb6-249"><a href="#cb6-249" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-250"><a href="#cb6-250" aria-hidden="true" tabindex="-1"></a>folder_to_save <span class="ot">&lt;-</span> <span class="st">"../results/plots/scoring/movie"</span></span>
<span id="cb6-251"><a href="#cb6-251" aria-hidden="true" tabindex="-1"></a><span class="fu">dir.create</span>(</span>
<span id="cb6-252"><a href="#cb6-252" aria-hidden="true" tabindex="-1"></a>    folder_to_save, </span>
<span id="cb6-253"><a href="#cb6-253" aria-hidden="true" tabindex="-1"></a>    <span class="at">showWarnings =</span> <span class="cn">FALSE</span>, </span>
<span id="cb6-254"><a href="#cb6-254" aria-hidden="true" tabindex="-1"></a>    <span class="at">recursive =</span> <span class="cn">TRUE</span></span>
<span id="cb6-255"><a href="#cb6-255" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb6-256"><a href="#cb6-256" aria-hidden="true" tabindex="-1"></a>fig_width <span class="ot">&lt;-</span> <span class="dv">22</span></span>
<span id="cb6-257"><a href="#cb6-257" aria-hidden="true" tabindex="-1"></a>fig_height <span class="ot">&lt;-</span> <span class="dv">10</span></span>
<span id="cb6-258"><a href="#cb6-258" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-259"><a href="#cb6-259" aria-hidden="true" tabindex="-1"></a><span class="fu">sapply</span>(</span>
<span id="cb6-260"><a href="#cb6-260" aria-hidden="true" tabindex="-1"></a>    <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(final_plots),</span>
<span id="cb6-261"><a href="#cb6-261" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(i, final_plots){</span>
<span id="cb6-262"><a href="#cb6-262" aria-hidden="true" tabindex="-1"></a>        ggplot2<span class="sc">::</span><span class="fu">ggsave</span>(</span>
<span id="cb6-263"><a href="#cb6-263" aria-hidden="true" tabindex="-1"></a>            <span class="at">filename =</span> <span class="fu">paste0</span>(</span>
<span id="cb6-264"><a href="#cb6-264" aria-hidden="true" tabindex="-1"></a>                folder_to_save, </span>
<span id="cb6-265"><a href="#cb6-265" aria-hidden="true" tabindex="-1"></a>                <span class="st">"/"</span>, </span>
<span id="cb6-266"><a href="#cb6-266" aria-hidden="true" tabindex="-1"></a>                <span class="fu">ifelse</span>(i <span class="sc">&lt;</span> <span class="dv">10</span>, <span class="fu">paste0</span>(<span class="st">"0"</span>,i), i), <span class="st">".png"</span></span>
<span id="cb6-267"><a href="#cb6-267" aria-hidden="true" tabindex="-1"></a>            ),</span>
<span id="cb6-268"><a href="#cb6-268" aria-hidden="true" tabindex="-1"></a>            <span class="at">plot =</span> final_plots[[i]],</span>
<span id="cb6-269"><a href="#cb6-269" aria-hidden="true" tabindex="-1"></a>            <span class="at">width =</span> fig_width,</span>
<span id="cb6-270"><a href="#cb6-270" aria-hidden="true" tabindex="-1"></a>            <span class="at">height =</span> fig_height, </span>
<span id="cb6-271"><a href="#cb6-271" aria-hidden="true" tabindex="-1"></a>            <span class="at">dpi =</span> <span class="dv">320</span></span>
<span id="cb6-272"><a href="#cb6-272" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb6-273"><a href="#cb6-273" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb6-274"><a href="#cb6-274" aria-hidden="true" tabindex="-1"></a>    <span class="at">final_plots =</span> final_plots</span>
<span id="cb6-275"><a href="#cb6-275" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb6-276"><a href="#cb6-276" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-277"><a href="#cb6-277" aria-hidden="true" tabindex="-1"></a><span class="co"># the original command used in the command line is the one below:</span></span>
<span id="cb6-278"><a href="#cb6-278" aria-hidden="true" tabindex="-1"></a><span class="co"># ffmpeg -y -framerate 2 -pattern_type glob -i 'movie/*.png' mol_land.mp4</span></span>
<span id="cb6-279"><a href="#cb6-279" aria-hidden="true" tabindex="-1"></a><span class="co"># we can then call it directly from R instead of having to do it manually</span></span>
<span id="cb6-280"><a href="#cb6-280" aria-hidden="true" tabindex="-1"></a><span class="co"># after rendering the document</span></span>
<span id="cb6-281"><a href="#cb6-281" aria-hidden="true" tabindex="-1"></a><span class="fu">system</span>(</span>
<span id="cb6-282"><a href="#cb6-282" aria-hidden="true" tabindex="-1"></a>    <span class="fu">paste0</span>(</span>
<span id="cb6-283"><a href="#cb6-283" aria-hidden="true" tabindex="-1"></a>        <span class="st">"ffmpeg -y -framerate 2 -pattern_type glob -i '"</span>,</span>
<span id="cb6-284"><a href="#cb6-284" aria-hidden="true" tabindex="-1"></a>        folder_to_save, <span class="st">"/*.png' "</span>, </span>
<span id="cb6-285"><a href="#cb6-285" aria-hidden="true" tabindex="-1"></a>        folder_to_save, <span class="st">"/../mol_land.mp4"</span></span>
<span id="cb6-286"><a href="#cb6-286" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb6-287"><a href="#cb6-287" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb6-288"><a href="#cb6-288" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb6-289"><a href="#cb6-289" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-290"><a href="#cb6-290" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = !first_run, results = 'asis', echo = FALSE}</span></span>
<span id="cb6-291"><a href="#cb6-291" aria-hidden="true" tabindex="-1"></a>embedding_video <span class="ot">&lt;-</span> <span class="fu">paste0</span>(</span>
<span id="cb6-292"><a href="#cb6-292" aria-hidden="true" tabindex="-1"></a>    <span class="st">'&lt;iframe width="720" height="480" '</span>,</span>
<span id="cb6-293"><a href="#cb6-293" aria-hidden="true" tabindex="-1"></a>    <span class="st">'src="../plots/scoring/mol_land.mp4" align="middle"'</span>,</span>
<span id="cb6-294"><a href="#cb6-294" aria-hidden="true" tabindex="-1"></a>    <span class="st">'frameborder="0" allowfullscreen&gt;&lt;/iframe&gt;'</span></span>
<span id="cb6-295"><a href="#cb6-295" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb6-296"><a href="#cb6-296" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-297"><a href="#cb6-297" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(embedding_video)</span>
<span id="cb6-298"><a href="#cb6-298" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb6-299"><a href="#cb6-299" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-300"><a href="#cb6-300" aria-hidden="true" tabindex="-1"></a>Going from right to left it shows how ER signaling is decreased. From top to</span>
<span id="cb6-301"><a href="#cb6-301" aria-hidden="true" tabindex="-1"></a>bottom there is a reduction of proliferation, E2F targets, and a change in</span>
<span id="cb6-302"><a href="#cb6-302" aria-hidden="true" tabindex="-1"></a>EMT and TGFb signaling. These are representative scores, other pathways could</span>
<span id="cb6-303"><a href="#cb6-303" aria-hidden="true" tabindex="-1"></a>be used as well. </span>
<span id="cb6-304"><a href="#cb6-304" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-305"><a href="#cb6-305" aria-hidden="true" tabindex="-1"></a><span class="fu">## POETIC embedding</span></span>
<span id="cb6-306"><a href="#cb6-306" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-307"><a href="#cb6-307" aria-hidden="true" tabindex="-1"></a>In this section we show the embedding of the POETIC trial samples.</span>
<span id="cb6-308"><a href="#cb6-308" aria-hidden="true" tabindex="-1"></a>But first we analyse some basic properties of the dataset.</span>
<span id="cb6-309"><a href="#cb6-309" aria-hidden="true" tabindex="-1"></a>The data for this dataset was downloaded and processed</span>
<span id="cb6-310"><a href="#cb6-310" aria-hidden="true" tabindex="-1"></a>as described at </span>
<span id="cb6-311"><a href="#cb6-311" aria-hidden="true" tabindex="-1"></a><span class="co">[</span><span class="ot">chronchi.github.io/transcriptomics</span><span class="co">](https://chronchi.github.io/transcriptomics/)</span></span>
<span id="cb6-312"><a href="#cb6-312" aria-hidden="true" tabindex="-1"></a>in the chapter AI - GSE105777. </span>
<span id="cb6-313"><a href="#cb6-313" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-314"><a href="#cb6-314" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb6-315"><a href="#cb6-315" aria-hidden="true" tabindex="-1"></a>datasets <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(</span>
<span id="cb6-316"><a href="#cb6-316" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/surv_analysis_estrogen/datasets_with_scores.rds"</span></span>
<span id="cb6-317"><a href="#cb6-317" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb6-318"><a href="#cb6-318" aria-hidden="true" tabindex="-1"></a>poetic <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../data/poetic.rds"</span>)</span>
<span id="cb6-319"><a href="#cb6-319" aria-hidden="true" tabindex="-1"></a>df_pca_coordinates <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(</span>
<span id="cb6-320"><a href="#cb6-320" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/pca_merging/df_pca_coordinates.rds"</span></span>
<span id="cb6-321"><a href="#cb6-321" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb6-322"><a href="#cb6-322" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-323"><a href="#cb6-323" aria-hidden="true" tabindex="-1"></a>datasets<span class="sc">$</span>poetic <span class="ot">&lt;-</span> poetic</span>
<span id="cb6-324"><a href="#cb6-324" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-325"><a href="#cb6-325" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb6-326"><a href="#cb6-326" aria-hidden="true" tabindex="-1"></a>    datasets,</span>
<span id="cb6-327"><a href="#cb6-327" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/scoring/datasets_with_poetic.rds"</span></span>
<span id="cb6-328"><a href="#cb6-328" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb6-329"><a href="#cb6-329" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-330"><a href="#cb6-330" aria-hidden="true" tabindex="-1"></a><span class="co"># get the normalization performed </span></span>
<span id="cb6-331"><a href="#cb6-331" aria-hidden="true" tabindex="-1"></a>poetic_normalized <span class="ot">&lt;-</span> <span class="fu">get_final_ranking_values</span>(</span>
<span id="cb6-332"><a href="#cb6-332" aria-hidden="true" tabindex="-1"></a>    <span class="at">sum_exp =</span> poetic,</span>
<span id="cb6-333"><a href="#cb6-333" aria-hidden="true" tabindex="-1"></a>    <span class="at">assay_to_use =</span> <span class="st">"normalized_intensity"</span>,</span>
<span id="cb6-334"><a href="#cb6-334" aria-hidden="true" tabindex="-1"></a>    <span class="at">stable_genes =</span> stable_genes,</span>
<span id="cb6-335"><a href="#cb6-335" aria-hidden="true" tabindex="-1"></a>    <span class="at">most_variable_genes =</span> <span class="fu">setdiff</span>(<span class="fu">rownames</span>(pca_fit<span class="sc">$</span>loadings), stable_genes)</span>
<span id="cb6-336"><a href="#cb6-336" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb6-337"><a href="#cb6-337" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-338"><a href="#cb6-338" aria-hidden="true" tabindex="-1"></a><span class="co"># calculate the embedding</span></span>
<span id="cb6-339"><a href="#cb6-339" aria-hidden="true" tabindex="-1"></a>poetic_df_pca <span class="ot">&lt;-</span> <span class="fu">get_pca_coordinates</span>(poetic_normalized, pca_fit) <span class="sc">%&gt;%</span></span>
<span id="cb6-340"><a href="#cb6-340" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span> </span>
<span id="cb6-341"><a href="#cb6-341" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">bind_cols</span>(</span>
<span id="cb6-342"><a href="#cb6-342" aria-hidden="true" tabindex="-1"></a>        ., </span>
<span id="cb6-343"><a href="#cb6-343" aria-hidden="true" tabindex="-1"></a>        <span class="fu">colData</span>(poetic_normalized) <span class="sc">%&gt;%</span> data.frame <span class="sc">%&gt;%</span> </span>
<span id="cb6-344"><a href="#cb6-344" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb6-345"><a href="#cb6-345" aria-hidden="true" tabindex="-1"></a>                <span class="at">er_status =</span> <span class="fu">ifelse</span>(</span>
<span id="cb6-346"><a href="#cb6-346" aria-hidden="true" tabindex="-1"></a>                    er_status <span class="sc">==</span> <span class="st">"Positive"</span>, </span>
<span id="cb6-347"><a href="#cb6-347" aria-hidden="true" tabindex="-1"></a>                    <span class="st">"pos"</span>, </span>
<span id="cb6-348"><a href="#cb6-348" aria-hidden="true" tabindex="-1"></a>                    <span class="st">"neg"</span></span>
<span id="cb6-349"><a href="#cb6-349" aria-hidden="true" tabindex="-1"></a>                )</span>
<span id="cb6-350"><a href="#cb6-350" aria-hidden="true" tabindex="-1"></a>            ) <span class="sc">%&gt;%</span> </span>
<span id="cb6-351"><a href="#cb6-351" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">cohort =</span> <span class="st">"poetic"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb6-352"><a href="#cb6-352" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">is_responder =</span></span>
<span id="cb6-353"><a href="#cb6-353" aria-hidden="true" tabindex="-1"></a>                r_or_no_r_change_ki67_60_and_baseline_ki67_5_percent</span>
<span id="cb6-354"><a href="#cb6-354" aria-hidden="true" tabindex="-1"></a>            ) <span class="sc">%&gt;%</span> </span>
<span id="cb6-355"><a href="#cb6-355" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb6-356"><a href="#cb6-356" aria-hidden="true" tabindex="-1"></a>                <span class="at">sample_name =</span> <span class="fu">colnames</span>(poetic_normalized)</span>
<span id="cb6-357"><a href="#cb6-357" aria-hidden="true" tabindex="-1"></a>            )</span>
<span id="cb6-358"><a href="#cb6-358" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span> </span>
<span id="cb6-359"><a href="#cb6-359" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(., df_pca_coordinates)</span>
<span id="cb6-360"><a href="#cb6-360" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-361"><a href="#cb6-361" aria-hidden="true" tabindex="-1"></a><span class="co"># save all the results</span></span>
<span id="cb6-362"><a href="#cb6-362" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb6-363"><a href="#cb6-363" aria-hidden="true" tabindex="-1"></a>    poetic_df_pca <span class="sc">%&gt;%</span> data.frame,</span>
<span id="cb6-364"><a href="#cb6-364" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/scoring/poetic_df_pca.rds"</span></span>
<span id="cb6-365"><a href="#cb6-365" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb6-366"><a href="#cb6-366" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-367"><a href="#cb6-367" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb6-368"><a href="#cb6-368" aria-hidden="true" tabindex="-1"></a>    poetic_normalized,</span>
<span id="cb6-369"><a href="#cb6-369" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/scoring/poetic_normalized.rds"</span></span>
<span id="cb6-370"><a href="#cb6-370" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb6-371"><a href="#cb6-371" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb6-372"><a href="#cb6-372" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-373"><a href="#cb6-373" aria-hidden="true" tabindex="-1"></a><span class="fu">### Number of samples</span></span>
<span id="cb6-374"><a href="#cb6-374" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-375"><a href="#cb6-375" aria-hidden="true" tabindex="-1"></a>In this dataset there are matched samples from patients that </span>
<span id="cb6-376"><a href="#cb6-376" aria-hidden="true" tabindex="-1"></a>are treated and untreated. @tbl-pt-poetic-nb</span>
<span id="cb6-377"><a href="#cb6-377" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-380"><a href="#cb6-380" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb6-381"><a href="#cb6-381" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: tbl-pt-poetic-nb</span></span>
<span id="cb6-382"><a href="#cb6-382" aria-hidden="true" tabindex="-1"></a><span class="co">#| tbl-cap: Table showing the number of samples for treated and untreated</span></span>
<span id="cb6-383"><a href="#cb6-383" aria-hidden="true" tabindex="-1"></a><span class="co">#|     patients</span></span>
<span id="cb6-384"><a href="#cb6-384" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(poetic_normalized) <span class="sc">%&gt;%</span> data.frame <span class="sc">%&gt;%</span></span>
<span id="cb6-385"><a href="#cb6-385" aria-hidden="true" tabindex="-1"></a>    janitor<span class="sc">::</span><span class="fu">tabyl</span>(group) <span class="sc">%&gt;%</span> </span>
<span id="cb6-386"><a href="#cb6-386" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">n =</span> n<span class="sc">/</span><span class="dv">2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb6-387"><a href="#cb6-387" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">percent =</span> <span class="fu">round</span>(n<span class="sc">/</span><span class="fu">sum</span>(n) <span class="sc">*</span> <span class="dv">100</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb6-388"><a href="#cb6-388" aria-hidden="true" tabindex="-1"></a>    kableExtra<span class="sc">::</span><span class="fu">kbl</span>() <span class="sc">%&gt;%</span></span>
<span id="cb6-389"><a href="#cb6-389" aria-hidden="true" tabindex="-1"></a>    kableExtra<span class="sc">::</span><span class="fu">kable_classic</span>()</span>
<span id="cb6-390"><a href="#cb6-390" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb6-391"><a href="#cb6-391" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-392"><a href="#cb6-392" aria-hidden="true" tabindex="-1"></a>In this cohort, there is the PAM50 molecular subtype available</span>
<span id="cb6-393"><a href="#cb6-393" aria-hidden="true" tabindex="-1"></a>for the untreated. @tbl-pt-poetic-pam50 shows the</span>
<span id="cb6-394"><a href="#cb6-394" aria-hidden="true" tabindex="-1"></a>number of patients for each subtype.</span>
<span id="cb6-395"><a href="#cb6-395" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-398"><a href="#cb6-398" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb6-399"><a href="#cb6-399" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: tbl-pt-poetic-pam50</span></span>
<span id="cb6-400"><a href="#cb6-400" aria-hidden="true" tabindex="-1"></a><span class="co">#| tbl-cap: Table showing the number of molecular subtypes for each </span></span>
<span id="cb6-401"><a href="#cb6-401" aria-hidden="true" tabindex="-1"></a><span class="co">#|     group.</span></span>
<span id="cb6-402"><a href="#cb6-402" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(poetic_normalized) <span class="sc">%&gt;%</span> data.frame <span class="sc">%&gt;%</span></span>
<span id="cb6-403"><a href="#cb6-403" aria-hidden="true" tabindex="-1"></a>    janitor<span class="sc">::</span><span class="fu">tabyl</span>(pam50) <span class="sc">%&gt;%</span> </span>
<span id="cb6-404"><a href="#cb6-404" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">percent =</span> <span class="fu">round</span>(n<span class="sc">/</span><span class="fu">sum</span>(n) <span class="sc">*</span> <span class="dv">100</span>, <span class="at">digits =</span> <span class="dv">3</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb6-405"><a href="#cb6-405" aria-hidden="true" tabindex="-1"></a>    kableExtra<span class="sc">::</span><span class="fu">kbl</span>() <span class="sc">%&gt;%</span></span>
<span id="cb6-406"><a href="#cb6-406" aria-hidden="true" tabindex="-1"></a>    kableExtra<span class="sc">::</span><span class="fu">kable_classic</span>()</span>
<span id="cb6-407"><a href="#cb6-407" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb6-408"><a href="#cb6-408" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-409"><a href="#cb6-409" aria-hidden="true" tabindex="-1"></a>The numbers differ from previously, since the molecular subtype is </span>
<span id="cb6-410"><a href="#cb6-410" aria-hidden="true" tabindex="-1"></a>calculated for each sample, so two molecular subtypes for each patient. </span>
<span id="cb6-411"><a href="#cb6-411" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-412"><a href="#cb6-412" aria-hidden="true" tabindex="-1"></a><span class="fu">### Proportion of top loadings</span></span>
<span id="cb6-413"><a href="#cb6-413" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-416"><a href="#cb6-416" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb6-417"><a href="#cb6-417" aria-hidden="true" tabindex="-1"></a>nb_genes <span class="ot">&lt;-</span> <span class="dv">200</span></span>
<span id="cb6-418"><a href="#cb6-418" aria-hidden="true" tabindex="-1"></a>proportions <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="at">by =</span> <span class="fl">0.05</span>)</span>
<span id="cb6-419"><a href="#cb6-419" aria-hidden="true" tabindex="-1"></a>gene_names <span class="ot">&lt;-</span> <span class="fu">rownames</span>(pca_fit<span class="sc">$</span>loadings)</span>
<span id="cb6-420"><a href="#cb6-420" aria-hidden="true" tabindex="-1"></a>which_pcs <span class="ot">&lt;-</span> <span class="dv">3</span><span class="sc">:</span><span class="dv">4</span></span>
<span id="cb6-421"><a href="#cb6-421" aria-hidden="true" tabindex="-1"></a>top_loadings_genes <span class="ot">&lt;-</span> <span class="fu">sapply</span>(</span>
<span id="cb6-422"><a href="#cb6-422" aria-hidden="true" tabindex="-1"></a>    which_pcs,</span>
<span id="cb6-423"><a href="#cb6-423" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(pc, gene_names, pca_fit, nb_genes){</span>
<span id="cb6-424"><a href="#cb6-424" aria-hidden="true" tabindex="-1"></a>        gene_names[<span class="fu">order</span>(</span>
<span id="cb6-425"><a href="#cb6-425" aria-hidden="true" tabindex="-1"></a>            <span class="fu">abs</span>(pca_fit<span class="sc">$</span>loadings[, <span class="fu">paste0</span>(<span class="st">"PC"</span>, pc)]), </span>
<span id="cb6-426"><a href="#cb6-426" aria-hidden="true" tabindex="-1"></a>            <span class="at">decreasing =</span> <span class="cn">TRUE</span></span>
<span id="cb6-427"><a href="#cb6-427" aria-hidden="true" tabindex="-1"></a>        )[<span class="dv">1</span><span class="sc">:</span>nb_genes]]   </span>
<span id="cb6-428"><a href="#cb6-428" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb6-429"><a href="#cb6-429" aria-hidden="true" tabindex="-1"></a>    <span class="at">gene_names =</span> gene_names,</span>
<span id="cb6-430"><a href="#cb6-430" aria-hidden="true" tabindex="-1"></a>    <span class="at">pca_fit =</span> pca_fit,</span>
<span id="cb6-431"><a href="#cb6-431" aria-hidden="true" tabindex="-1"></a>    <span class="at">nb_genes =</span> nb_genes</span>
<span id="cb6-432"><a href="#cb6-432" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> data.frame <span class="sc">%&gt;%</span> </span>
<span id="cb6-433"><a href="#cb6-433" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">colnames&lt;-</span><span class="st">`</span>(<span class="fu">paste0</span>(<span class="st">"PC"</span>, which_pcs))</span>
<span id="cb6-434"><a href="#cb6-434" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-435"><a href="#cb6-435" aria-hidden="true" tabindex="-1"></a>missing_genes_pcx <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(</span>
<span id="cb6-436"><a href="#cb6-436" aria-hidden="true" tabindex="-1"></a>    top_loadings_genes<span class="sc">$</span>PC3,</span>
<span id="cb6-437"><a href="#cb6-437" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames</span>(poetic_normalized)</span>
<span id="cb6-438"><a href="#cb6-438" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb6-439"><a href="#cb6-439" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-440"><a href="#cb6-440" aria-hidden="true" tabindex="-1"></a>missing_genes_pcy <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(</span>
<span id="cb6-441"><a href="#cb6-441" aria-hidden="true" tabindex="-1"></a>    top_loadings_genes<span class="sc">$</span>PC4,</span>
<span id="cb6-442"><a href="#cb6-442" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames</span>(poetic_normalized)</span>
<span id="cb6-443"><a href="#cb6-443" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb6-444"><a href="#cb6-444" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb6-445"><a href="#cb6-445" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-446"><a href="#cb6-446" aria-hidden="true" tabindex="-1"></a>There are in total <span class="in">`r nrow(poetic_normalized)`</span> genes available out</span>
<span id="cb6-447"><a href="#cb6-447" aria-hidden="true" tabindex="-1"></a>of the <span class="in">`r nrow(datasets_normalized)`</span>. Among them, </span>
<span id="cb6-448"><a href="#cb6-448" aria-hidden="true" tabindex="-1"></a><span class="in">`r length(intersect(rownames(poetic_normalized), stable_genes))`</span> are</span>
<span id="cb6-449"><a href="#cb6-449" aria-hidden="true" tabindex="-1"></a>the housekeeping genes. Out of the</span>
<span id="cb6-450"><a href="#cb6-450" aria-hidden="true" tabindex="-1"></a><span class="in">`r nrow(pca_fit$loadings) - nrow(poetic_normalized)`</span> genes missing,</span>
<span id="cb6-451"><a href="#cb6-451" aria-hidden="true" tabindex="-1"></a>a total number of <span class="in">`r length(missing_genes_pcx)`</span> are missing from the</span>
<span id="cb6-452"><a href="#cb6-452" aria-hidden="true" tabindex="-1"></a>top <span class="in">`r nb_genes`</span> PC3 loadings, a proportion of </span>
<span id="cb6-453"><a href="#cb6-453" aria-hidden="true" tabindex="-1"></a><span class="in">`r format(length(missing_genes_pcx)/nb_genes * 100, digits = 1)`</span>\%. </span>
<span id="cb6-454"><a href="#cb6-454" aria-hidden="true" tabindex="-1"></a>A total number of <span class="in">`r length(missing_genes_pcy)`</span> are missing from the</span>
<span id="cb6-455"><a href="#cb6-455" aria-hidden="true" tabindex="-1"></a>top <span class="in">`r nb_genes`</span> PC4 loadings, a proportion of </span>
<span id="cb6-456"><a href="#cb6-456" aria-hidden="true" tabindex="-1"></a><span class="in">`r format(length(missing_genes_pcy)/nb_genes * 100, digits = 1)`</span>\%. These</span>
<span id="cb6-457"><a href="#cb6-457" aria-hidden="true" tabindex="-1"></a>are very small proportions, and we have seen from the last chapter that</span>
<span id="cb6-458"><a href="#cb6-458" aria-hidden="true" tabindex="-1"></a>they will not affect the position of the patients in the embedding.</span>
<span id="cb6-459"><a href="#cb6-459" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-460"><a href="#cb6-460" aria-hidden="true" tabindex="-1"></a><span class="fu">### Embedding </span></span>
<span id="cb6-461"><a href="#cb6-461" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-462"><a href="#cb6-462" aria-hidden="true" tabindex="-1"></a>@fig-pca-poetic-er-pam50 shows a good mixing of the POETIC sample and how</span>
<span id="cb6-463"><a href="#cb6-463" aria-hidden="true" tabindex="-1"></a>all the patients are scattered across the whole landscape. Here all samples</span>
<span id="cb6-464"><a href="#cb6-464" aria-hidden="true" tabindex="-1"></a>are plotted, meaning that every two dots correspond to a single patient.</span>
<span id="cb6-465"><a href="#cb6-465" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-466"><a href="#cb6-466" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=19, fig.height=12}</span></span>
<span id="cb6-467"><a href="#cb6-467" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-pca-poetic-er-pam50</span></span>
<span id="cb6-468"><a href="#cb6-468" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: PCA embedding of all samples from TCGA, SCANB and METABRIC including</span></span>
<span id="cb6-469"><a href="#cb6-469" aria-hidden="true" tabindex="-1"></a><span class="co">#|     the POETIC samples on top.</span></span>
<span id="cb6-470"><a href="#cb6-470" aria-hidden="true" tabindex="-1"></a><span class="co">#|     (A) Colored by cohort,</span></span>
<span id="cb6-471"><a href="#cb6-471" aria-hidden="true" tabindex="-1"></a><span class="co">#|     (B) colored by ER status, (C) colored by PAM50 molecular subtype.</span></span>
<span id="cb6-472"><a href="#cb6-472" aria-hidden="true" tabindex="-1"></a>plots_poetic <span class="ot">&lt;-</span> <span class="fu">sapply</span>(</span>
<span id="cb6-473"><a href="#cb6-473" aria-hidden="true" tabindex="-1"></a>    <span class="fu">c</span>(<span class="st">"cohort"</span>, <span class="st">"er_status"</span>, <span class="st">"pam50"</span>), </span>
<span id="cb6-474"><a href="#cb6-474" aria-hidden="true" tabindex="-1"></a>    get_plot_new_samples,</span>
<span id="cb6-475"><a href="#cb6-475" aria-hidden="true" tabindex="-1"></a>    <span class="at">name_cohort =</span> <span class="st">"poetic"</span>,</span>
<span id="cb6-476"><a href="#cb6-476" aria-hidden="true" tabindex="-1"></a>    <span class="at">df_pca =</span> poetic_df_pca,</span>
<span id="cb6-477"><a href="#cb6-477" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"POETIC samples projected on the molecular landscape"</span>,</span>
<span id="cb6-478"><a href="#cb6-478" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>, </span>
<span id="cb6-479"><a href="#cb6-479" aria-hidden="true" tabindex="-1"></a>    <span class="at">simplify =</span> <span class="cn">FALSE</span>,</span>
<span id="cb6-480"><a href="#cb6-480" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC3"</span>,</span>
<span id="cb6-481"><a href="#cb6-481" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC4"</span></span>
<span id="cb6-482"><a href="#cb6-482" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb6-483"><a href="#cb6-483" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-484"><a href="#cb6-484" aria-hidden="true" tabindex="-1"></a>plots_poetic<span class="sc">$</span>pam50 <span class="ot">&lt;-</span> plots_poetic<span class="sc">$</span>pam50 <span class="sc">+</span></span>
<span id="cb6-485"><a href="#cb6-485" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_color_manual</span>(</span>
<span id="cb6-486"><a href="#cb6-486" aria-hidden="true" tabindex="-1"></a>        <span class="at">values =</span> <span class="fu">c</span>(</span>
<span id="cb6-487"><a href="#cb6-487" aria-hidden="true" tabindex="-1"></a>            <span class="fu">get_colors_pam50</span>(plots_poetic<span class="sc">$</span>pam50<span class="sc">$</span>data),</span>
<span id="cb6-488"><a href="#cb6-488" aria-hidden="true" tabindex="-1"></a>            <span class="st">"not_available"</span> <span class="ot">=</span> <span class="st">"black"</span></span>
<span id="cb6-489"><a href="#cb6-489" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb6-490"><a href="#cb6-490" aria-hidden="true" tabindex="-1"></a>    ) </span>
<span id="cb6-491"><a href="#cb6-491" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-492"><a href="#cb6-492" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(<span class="at">plotlist =</span> plots_poetic, <span class="at">labels =</span> <span class="st">"AUTO"</span>)</span>
<span id="cb6-493"><a href="#cb6-493" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb6-494"><a href="#cb6-494" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-495"><a href="#cb6-495" aria-hidden="true" tabindex="-1"></a>@fig-pca-responders-baseline shows the embedding of baseline samples</span>
<span id="cb6-496"><a href="#cb6-496" aria-hidden="true" tabindex="-1"></a>from patients that received endocrine therapy prior to surgery. </span>
<span id="cb6-497"><a href="#cb6-497" aria-hidden="true" tabindex="-1"></a>Patients on the left part of the molecular landscape are considered to</span>
<span id="cb6-498"><a href="#cb6-498" aria-hidden="true" tabindex="-1"></a>be non responders, this coincides with the fact this region corresponds</span>
<span id="cb6-499"><a href="#cb6-499" aria-hidden="true" tabindex="-1"></a>to the ER- BC patients.</span>
<span id="cb6-500"><a href="#cb6-500" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-501"><a href="#cb6-501" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=6}</span></span>
<span id="cb6-502"><a href="#cb6-502" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-pca-responders-baseline</span></span>
<span id="cb6-503"><a href="#cb6-503" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: PCA embedding of all samples from TCGA, SCANB and METABRIC including</span></span>
<span id="cb6-504"><a href="#cb6-504" aria-hidden="true" tabindex="-1"></a><span class="co">#|     the POETIC samples on top. POETIC samples correspond to only baseline</span></span>
<span id="cb6-505"><a href="#cb6-505" aria-hidden="true" tabindex="-1"></a><span class="co">#|     treated patients.</span></span>
<span id="cb6-506"><a href="#cb6-506" aria-hidden="true" tabindex="-1"></a>poetic_df_pca_baseline <span class="ot">&lt;-</span> poetic_df_pca <span class="sc">%&gt;%</span> </span>
<span id="cb6-507"><a href="#cb6-507" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span>(cohort <span class="sc">==</span> <span class="st">"poetic"</span> <span class="sc">&amp;</span> group <span class="sc">==</span> <span class="st">"untreated"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb6-508"><a href="#cb6-508" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb6-509"><a href="#cb6-509" aria-hidden="true" tabindex="-1"></a>        <span class="at">timepoint =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(timepoint), <span class="st">"baseline"</span>, timepoint)</span>
<span id="cb6-510"><a href="#cb6-510" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb6-511"><a href="#cb6-511" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(timepoint <span class="sc">==</span> <span class="st">"baseline"</span>)</span>
<span id="cb6-512"><a href="#cb6-512" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb6-513"><a href="#cb6-513" aria-hidden="true" tabindex="-1"></a>plots_poetic <span class="ot">&lt;-</span> <span class="fu">sapply</span>(</span>
<span id="cb6-514"><a href="#cb6-514" aria-hidden="true" tabindex="-1"></a>    <span class="fu">c</span>(<span class="st">"is_responder"</span>), </span>
<span id="cb6-515"><a href="#cb6-515" aria-hidden="true" tabindex="-1"></a>    get_plot_new_samples,</span>
<span id="cb6-516"><a href="#cb6-516" aria-hidden="true" tabindex="-1"></a>    <span class="at">name_cohort =</span> <span class="st">"poetic"</span>,</span>
<span id="cb6-517"><a href="#cb6-517" aria-hidden="true" tabindex="-1"></a>    <span class="at">df_pca =</span> poetic_df_pca_baseline,</span>
<span id="cb6-518"><a href="#cb6-518" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC3"</span>,</span>
<span id="cb6-519"><a href="#cb6-519" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC4"</span>,</span>
<span id="cb6-520"><a href="#cb6-520" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"POETIC baseline patients colored by response status"</span>,</span>
<span id="cb6-521"><a href="#cb6-521" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>, </span>
<span id="cb6-522"><a href="#cb6-522" aria-hidden="true" tabindex="-1"></a>    <span class="at">simplify =</span> <span class="cn">FALSE</span></span>
<span id="cb6-523"><a href="#cb6-523" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb6-524"><a href="#cb6-524" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-525"><a href="#cb6-525" aria-hidden="true" tabindex="-1"></a>plots_poetic<span class="sc">$</span>is_responder</span>
<span id="cb6-526"><a href="#cb6-526" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb6-527"><a href="#cb6-527" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-528"><a href="#cb6-528" aria-hidden="true" tabindex="-1"></a>And when checking the first two components, the POETIC data is closer to</span>
<span id="cb6-529"><a href="#cb6-529" aria-hidden="true" tabindex="-1"></a>METABRIC, which makes sense since both are microarrays, as it can be seen</span>
<span id="cb6-530"><a href="#cb6-530" aria-hidden="true" tabindex="-1"></a>in @fig-poetic-cohort.</span>
<span id="cb6-531"><a href="#cb6-531" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-534"><a href="#cb6-534" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb6-535"><a href="#cb6-535" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-poetic-cohort</span></span>
<span id="cb6-536"><a href="#cb6-536" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Biplot of first two PCA components from POETIC, TCGA, SCANB and </span></span>
<span id="cb6-537"><a href="#cb6-537" aria-hidden="true" tabindex="-1"></a><span class="co">#|     METABRIC. POETIC is highlighted in the plot and has a bigger point.</span></span>
<span id="cb6-538"><a href="#cb6-538" aria-hidden="true" tabindex="-1"></a>poetic_df_pca <span class="sc">%&gt;%</span> </span>
<span id="cb6-539"><a href="#cb6-539" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(</span>
<span id="cb6-540"><a href="#cb6-540" aria-hidden="true" tabindex="-1"></a>        <span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> cohort)</span>
<span id="cb6-541"><a href="#cb6-541" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb6-542"><a href="#cb6-542" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">alpha =</span> cohort, <span class="at">size =</span> cohort)) <span class="sc">+</span></span>
<span id="cb6-543"><a href="#cb6-543" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_size_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(</span>
<span id="cb6-544"><a href="#cb6-544" aria-hidden="true" tabindex="-1"></a>        <span class="st">"poetic"</span> <span class="ot">=</span> <span class="dv">2</span>, </span>
<span id="cb6-545"><a href="#cb6-545" aria-hidden="true" tabindex="-1"></a>        <span class="st">"tcga"</span> <span class="ot">=</span> <span class="dv">1</span>,</span>
<span id="cb6-546"><a href="#cb6-546" aria-hidden="true" tabindex="-1"></a>        <span class="st">"scanb"</span> <span class="ot">=</span> <span class="dv">1</span>,</span>
<span id="cb6-547"><a href="#cb6-547" aria-hidden="true" tabindex="-1"></a>        <span class="st">"metabric"</span> <span class="ot">=</span> <span class="dv">1</span></span>
<span id="cb6-548"><a href="#cb6-548" aria-hidden="true" tabindex="-1"></a>    ), <span class="at">guide =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb6-549"><a href="#cb6-549" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_alpha_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(</span>
<span id="cb6-550"><a href="#cb6-550" aria-hidden="true" tabindex="-1"></a>        <span class="st">"poetic"</span> <span class="ot">=</span> <span class="dv">1</span>, </span>
<span id="cb6-551"><a href="#cb6-551" aria-hidden="true" tabindex="-1"></a>        <span class="st">"tcga"</span> <span class="ot">=</span> <span class="fl">0.1</span>,</span>
<span id="cb6-552"><a href="#cb6-552" aria-hidden="true" tabindex="-1"></a>        <span class="st">"scanb"</span> <span class="ot">=</span> <span class="fl">0.1</span>,</span>
<span id="cb6-553"><a href="#cb6-553" aria-hidden="true" tabindex="-1"></a>        <span class="st">"metabric"</span> <span class="ot">=</span> <span class="fl">0.1</span></span>
<span id="cb6-554"><a href="#cb6-554" aria-hidden="true" tabindex="-1"></a>    ), <span class="at">guide =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb6-555"><a href="#cb6-555" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb6-556"><a href="#cb6-556" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">"Cohort"</span>,</span>
<span id="cb6-557"><a href="#cb6-557" aria-hidden="true" tabindex="-1"></a>        <span class="at">alpha =</span> <span class="st">"Cohort"</span>,</span>
<span id="cb6-558"><a href="#cb6-558" aria-hidden="true" tabindex="-1"></a>        <span class="at">size =</span> <span class="st">"Cohort"</span>,</span>
<span id="cb6-559"><a href="#cb6-559" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="fu">paste0</span>(</span>
<span id="cb6-560"><a href="#cb6-560" aria-hidden="true" tabindex="-1"></a>            <span class="st">"First two components of POETIC overlayed on all the three</span><span class="sc">\n</span><span class="st">"</span>,</span>
<span id="cb6-561"><a href="#cb6-561" aria-hidden="true" tabindex="-1"></a>            <span class="st">"big cohorts in the the molecular landscape"</span></span>
<span id="cb6-562"><a href="#cb6-562" aria-hidden="true" tabindex="-1"></a>        ),</span>
<span id="cb6-563"><a href="#cb6-563" aria-hidden="true" tabindex="-1"></a>        <span class="at">subtitle =</span> <span class="st">"All samples from TCGA, METABRIC and SCANB are plotted"</span></span>
<span id="cb6-564"><a href="#cb6-564" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb6-565"><a href="#cb6-565" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">14</span>) <span class="sc">+</span></span>
<span id="cb6-566"><a href="#cb6-566" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span></span>
<span id="cb6-567"><a href="#cb6-567" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb6-568"><a href="#cb6-568" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb6-569"><a href="#cb6-569" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-570"><a href="#cb6-570" aria-hidden="true" tabindex="-1"></a><span class="fu">### Differences in PC3 for responders and non responders</span></span>
<span id="cb6-571"><a href="#cb6-571" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-572"><a href="#cb6-572" aria-hidden="true" tabindex="-1"></a>We now also compare the differences in non responders vs responders</span>
<span id="cb6-573"><a href="#cb6-573" aria-hidden="true" tabindex="-1"></a>according to the third component. </span>
<span id="cb6-574"><a href="#cb6-574" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-577"><a href="#cb6-577" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb6-578"><a href="#cb6-578" aria-hidden="true" tabindex="-1"></a>poetic_df_pca_baseline <span class="sc">%&gt;%</span></span>
<span id="cb6-579"><a href="#cb6-579" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"poetic"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb6-580"><a href="#cb6-580" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> is_responder, <span class="at">y =</span> PC3, <span class="at">color =</span> is_responder)) <span class="sc">+</span></span>
<span id="cb6-581"><a href="#cb6-581" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb6-582"><a href="#cb6-582" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_jitter</span>() <span class="sc">+</span></span>
<span id="cb6-583"><a href="#cb6-583" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Third component score"</span>)</span>
<span id="cb6-584"><a href="#cb6-584" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>()</span>
<span id="cb6-585"><a href="#cb6-585" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb6-586"><a href="#cb6-586" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-587"><a href="#cb6-587" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-588"><a href="#cb6-588" aria-hidden="true" tabindex="-1"></a><span class="fu">## Scoring all samples </span></span>
<span id="cb6-589"><a href="#cb6-589" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-590"><a href="#cb6-590" aria-hidden="true" tabindex="-1"></a>In the last chapters scores were showed for TCGA, SCANB and </span>
<span id="cb6-591"><a href="#cb6-591" aria-hidden="true" tabindex="-1"></a>METABRIC. @fig-pca-seterpr shows how ER signaling changes as patients</span>
<span id="cb6-592"><a href="#cb6-592" aria-hidden="true" tabindex="-1"></a>move across the molecular landscape. We intend to use the scores now for the</span>
<span id="cb6-593"><a href="#cb6-593" aria-hidden="true" tabindex="-1"></a>POETIC samples. We first start by calculating using GSVA as well and include</span>
<span id="cb6-594"><a href="#cb6-594" aria-hidden="true" tabindex="-1"></a>in the dataframe with the other cohorts. Before moving to the next chapter,</span>
<span id="cb6-595"><a href="#cb6-595" aria-hidden="true" tabindex="-1"></a>we will analyse and compare the scores across the responders and non </span>
<span id="cb6-596"><a href="#cb6-596" aria-hidden="true" tabindex="-1"></a>responders. </span>
<span id="cb6-597"><a href="#cb6-597" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-598"><a href="#cb6-598" aria-hidden="true" tabindex="-1"></a>@fig-scores-poetic-er shows that when using the scores, it looks like</span>
<span id="cb6-599"><a href="#cb6-599" aria-hidden="true" tabindex="-1"></a>in average at baseline $SET_{ER/PR}$ is lower in non responders than</span>
<span id="cb6-600"><a href="#cb6-600" aria-hidden="true" tabindex="-1"></a>in responders. For the other scores there is no clear difference.</span>
<span id="cb6-601"><a href="#cb6-601" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-602"><a href="#cb6-602" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb6-603"><a href="#cb6-603" aria-hidden="true" tabindex="-1"></a><span class="co"># before we calculate the scores we filter the poetic dataset</span></span>
<span id="cb6-604"><a href="#cb6-604" aria-hidden="true" tabindex="-1"></a>keep_genes <span class="ot">&lt;-</span> <span class="fu">rowMeans</span>(<span class="fu">assay</span>(datasets<span class="sc">$</span>poetic, <span class="st">"normalized_intensity"</span>) <span class="sc">&gt;</span> <span class="fl">6.4</span>)</span>
<span id="cb6-605"><a href="#cb6-605" aria-hidden="true" tabindex="-1"></a>keep_genes <span class="ot">&lt;-</span> keep_genes <span class="sc">&gt;</span> <span class="fl">0.7</span></span>
<span id="cb6-606"><a href="#cb6-606" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-607"><a href="#cb6-607" aria-hidden="true" tabindex="-1"></a>gene_sets <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(</span>
<span id="cb6-608"><a href="#cb6-608" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/surv_analysis_estrogen/gene_sets.rds"</span></span>
<span id="cb6-609"><a href="#cb6-609" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb6-610"><a href="#cb6-610" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-611"><a href="#cb6-611" aria-hidden="true" tabindex="-1"></a>gene_sets <span class="ot">&lt;-</span> <span class="fu">sapply</span>(</span>
<span id="cb6-612"><a href="#cb6-612" aria-hidden="true" tabindex="-1"></a>    gene_sets<span class="sc">$</span>gs_name <span class="sc">%&gt;%</span> unique,</span>
<span id="cb6-613"><a href="#cb6-613" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(x, gene_sets) gene_sets <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(gs_name <span class="sc">==</span> x) <span class="sc">%&gt;%</span> </span>
<span id="cb6-614"><a href="#cb6-614" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">pull</span>(gene_symbol),</span>
<span id="cb6-615"><a href="#cb6-615" aria-hidden="true" tabindex="-1"></a>    <span class="at">gene_sets =</span> gene_sets,</span>
<span id="cb6-616"><a href="#cb6-616" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>,</span>
<span id="cb6-617"><a href="#cb6-617" aria-hidden="true" tabindex="-1"></a>    <span class="at">simplify =</span> <span class="cn">FALSE</span></span>
<span id="cb6-618"><a href="#cb6-618" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb6-619"><a href="#cb6-619" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-620"><a href="#cb6-620" aria-hidden="true" tabindex="-1"></a>gsva_scores <span class="ot">&lt;-</span> <span class="fu">mapply</span>(</span>
<span id="cb6-621"><a href="#cb6-621" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(sum_exp, which_assay, gene_sets){</span>
<span id="cb6-622"><a href="#cb6-622" aria-hidden="true" tabindex="-1"></a>        GSVA<span class="sc">::</span><span class="fu">gsva</span>(</span>
<span id="cb6-623"><a href="#cb6-623" aria-hidden="true" tabindex="-1"></a>            <span class="at">expr =</span> <span class="fu">as.matrix</span>(<span class="fu">assay</span>(sum_exp, which_assay)),</span>
<span id="cb6-624"><a href="#cb6-624" aria-hidden="true" tabindex="-1"></a>            <span class="at">gset.idx.list =</span> gene_sets, </span>
<span id="cb6-625"><a href="#cb6-625" aria-hidden="true" tabindex="-1"></a>            <span class="at">parallel.sz =</span> <span class="dv">20</span></span>
<span id="cb6-626"><a href="#cb6-626" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb6-627"><a href="#cb6-627" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb6-628"><a href="#cb6-628" aria-hidden="true" tabindex="-1"></a>    <span class="at">sum_exp =</span> <span class="fu">list</span>(<span class="at">poetic =</span> datasets<span class="sc">$</span>poetic[keep_genes, ]),</span>
<span id="cb6-629"><a href="#cb6-629" aria-hidden="true" tabindex="-1"></a>    <span class="at">which_assay =</span> <span class="fu">list</span>(<span class="at">poetic =</span> <span class="st">"normalized_intensity"</span>),</span>
<span id="cb6-630"><a href="#cb6-630" aria-hidden="true" tabindex="-1"></a>    <span class="at">MoreArgs =</span> <span class="fu">list</span>(<span class="at">gene_sets =</span> gene_sets),</span>
<span id="cb6-631"><a href="#cb6-631" aria-hidden="true" tabindex="-1"></a>    <span class="at">SIMPLIFY =</span> <span class="cn">FALSE</span>,</span>
<span id="cb6-632"><a href="#cb6-632" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span></span>
<span id="cb6-633"><a href="#cb6-633" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb6-634"><a href="#cb6-634" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-635"><a href="#cb6-635" aria-hidden="true" tabindex="-1"></a>poetic_df_pca[</span>
<span id="cb6-636"><a href="#cb6-636" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(gsva_scores<span class="sc">$</span>poetic), </span>
<span id="cb6-637"><a href="#cb6-637" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames</span>(gsva_scores<span class="sc">$</span>poetic)</span>
<span id="cb6-638"><a href="#cb6-638" aria-hidden="true" tabindex="-1"></a>] <span class="ot">&lt;-</span> gsva_scores<span class="sc">$</span>poetic <span class="sc">%&gt;%</span> t</span>
<span id="cb6-639"><a href="#cb6-639" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-640"><a href="#cb6-640" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb6-641"><a href="#cb6-641" aria-hidden="true" tabindex="-1"></a>    poetic_df_pca,</span>
<span id="cb6-642"><a href="#cb6-642" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/scoring/poetic_df_pca_with_scores.rds"</span></span>
<span id="cb6-643"><a href="#cb6-643" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb6-644"><a href="#cb6-644" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-645"><a href="#cb6-645" aria-hidden="true" tabindex="-1"></a>df_pca_coordinates <span class="ot">&lt;-</span> poetic_df_pca</span>
<span id="cb6-646"><a href="#cb6-646" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb6-647"><a href="#cb6-647" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-650"><a href="#cb6-650" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb6-651"><a href="#cb6-651" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span>load_at_setup){</span>
<span id="cb6-652"><a href="#cb6-652" aria-hidden="true" tabindex="-1"></a>    df_pca_coordinates <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(</span>
<span id="cb6-653"><a href="#cb6-653" aria-hidden="true" tabindex="-1"></a>        <span class="st">"../results/rds_files/scoring/poetic_df_pca_with_scores.rds"</span></span>
<span id="cb6-654"><a href="#cb6-654" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb6-655"><a href="#cb6-655" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb6-656"><a href="#cb6-656" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb6-657"><a href="#cb6-657" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-658"><a href="#cb6-658" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=13, fig.height=8}</span></span>
<span id="cb6-659"><a href="#cb6-659" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-scores-poetic-er</span></span>
<span id="cb6-660"><a href="#cb6-660" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Baseline scores for all treated patients in the POETIC trial. </span></span>
<span id="cb6-661"><a href="#cb6-661" aria-hidden="true" tabindex="-1"></a><span class="co">#|     G2M corresponds to a proliferation score, EMT to epithelial to </span></span>
<span id="cb6-662"><a href="#cb6-662" aria-hidden="true" tabindex="-1"></a><span class="co">#|     mesenchymal transition score and the others are related to ER </span></span>
<span id="cb6-663"><a href="#cb6-663" aria-hidden="true" tabindex="-1"></a><span class="co">#|     signaling. </span></span>
<span id="cb6-664"><a href="#cb6-664" aria-hidden="true" tabindex="-1"></a>which_scores <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb6-665"><a href="#cb6-665" aria-hidden="true" tabindex="-1"></a>    <span class="st">"SET_ERPR"</span> <span class="ot">=</span> <span class="st">"SET ER/PR"</span>,</span>
<span id="cb6-666"><a href="#cb6-666" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_ESTROGEN_RESPONSE_EARLY"</span> <span class="ot">=</span> <span class="st">"Estrogen Early"</span>,</span>
<span id="cb6-667"><a href="#cb6-667" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_ESTROGEN_RESPONSE_LATE"</span> <span class="ot">=</span> <span class="st">"Estrogen Late"</span>,</span>
<span id="cb6-668"><a href="#cb6-668" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_G2M_CHECKPOINT"</span> <span class="ot">=</span> <span class="st">"G2M"</span>,</span>
<span id="cb6-669"><a href="#cb6-669" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION"</span> <span class="ot">=</span> <span class="st">"EMT"</span>,</span>
<span id="cb6-670"><a href="#cb6-670" aria-hidden="true" tabindex="-1"></a>    <span class="st">"PC3"</span> <span class="ot">=</span> <span class="st">"PC3"</span></span>
<span id="cb6-671"><a href="#cb6-671" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb6-672"><a href="#cb6-672" aria-hidden="true" tabindex="-1"></a>df_pca_coordinates <span class="sc">%&gt;%</span> </span>
<span id="cb6-673"><a href="#cb6-673" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"poetic"</span> <span class="sc">&amp;</span> timepoint <span class="sc">==</span> <span class="st">"baseline"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb6-674"><a href="#cb6-674" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(</span>
<span id="cb6-675"><a href="#cb6-675" aria-hidden="true" tabindex="-1"></a>        <span class="fu">names</span>(which_scores),</span>
<span id="cb6-676"><a href="#cb6-676" aria-hidden="true" tabindex="-1"></a>        <span class="at">names_to =</span> <span class="st">"pathway"</span>,</span>
<span id="cb6-677"><a href="#cb6-677" aria-hidden="true" tabindex="-1"></a>        <span class="at">values_to =</span> <span class="st">"score"</span></span>
<span id="cb6-678"><a href="#cb6-678" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb6-679"><a href="#cb6-679" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb6-680"><a href="#cb6-680" aria-hidden="true" tabindex="-1"></a>        <span class="at">pathway =</span> which_scores[pathway]</span>
<span id="cb6-681"><a href="#cb6-681" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb6-682"><a href="#cb6-682" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> is_responder, <span class="at">y =</span> score, <span class="at">fill =</span> is_responder)) <span class="sc">+</span></span>
<span id="cb6-683"><a href="#cb6-683" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_violin</span>(<span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb6-684"><a href="#cb6-684" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_jitter</span>(<span class="at">size =</span> <span class="fl">0.5</span>) <span class="sc">+</span> </span>
<span id="cb6-685"><a href="#cb6-685" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb6-686"><a href="#cb6-686" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>pathway, <span class="at">scale =</span> <span class="st">"free_x"</span>) <span class="sc">+</span></span>
<span id="cb6-687"><a href="#cb6-687" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>) <span class="sc">+</span> </span>
<span id="cb6-688"><a href="#cb6-688" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb6-689"><a href="#cb6-689" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"Is responder?"</span>, <span class="at">y =</span> <span class="st">"Score"</span>, <span class="at">fill =</span> <span class="st">"Is responder?"</span>,</span>
<span id="cb6-690"><a href="#cb6-690" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">"Distribution of some scores for samples at baseline timepoint"</span></span>
<span id="cb6-691"><a href="#cb6-691" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb6-692"><a href="#cb6-692" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span> </span>
<span id="cb6-693"><a href="#cb6-693" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb6-694"><a href="#cb6-694" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb6-695"><a href="#cb6-695" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-696"><a href="#cb6-696" aria-hidden="true" tabindex="-1"></a>We see that for PC3 and SET ER/PR There are some</span>
<span id="cb6-697"><a href="#cb6-697" aria-hidden="true" tabindex="-1"></a>differences in terms of the responders and non responders.</span>
<span id="cb6-698"><a href="#cb6-698" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-699"><a href="#cb6-699" aria-hidden="true" tabindex="-1"></a>Let us take a closer look at those patients. We start</span>
<span id="cb6-700"><a href="#cb6-700" aria-hidden="true" tabindex="-1"></a>by plotting the scatter plot of PC3 vs SET ER/PR and color</span>
<span id="cb6-701"><a href="#cb6-701" aria-hidden="true" tabindex="-1"></a>by response status defined by the POETIC trial. </span>
<span id="cb6-702"><a href="#cb6-702" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-705"><a href="#cb6-705" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb6-706"><a href="#cb6-706" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-scores-poetic-responders</span></span>
<span id="cb6-707"><a href="#cb6-707" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Comparison of the third component with SET ER/PR scores colored</span></span>
<span id="cb6-708"><a href="#cb6-708" aria-hidden="true" tabindex="-1"></a><span class="co">#|     by response status.</span></span>
<span id="cb6-709"><a href="#cb6-709" aria-hidden="true" tabindex="-1"></a>df_pca_coordinates <span class="sc">%&gt;%</span> </span>
<span id="cb6-710"><a href="#cb6-710" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"poetic"</span> <span class="sc">&amp;</span> timepoint <span class="sc">==</span> <span class="st">"baseline"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb6-711"><a href="#cb6-711" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(is_responder <span class="sc">!=</span> <span class="st">"not_available"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb6-712"><a href="#cb6-712" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> PC3, <span class="at">y =</span> SET_ERPR, <span class="at">color =</span> is_responder)) <span class="sc">+</span></span>
<span id="cb6-713"><a href="#cb6-713" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span> </span>
<span id="cb6-714"><a href="#cb6-714" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_smooth</span>(</span>
<span id="cb6-715"><a href="#cb6-715" aria-hidden="true" tabindex="-1"></a>        <span class="at">method =</span> <span class="st">"gam"</span>,</span>
<span id="cb6-716"><a href="#cb6-716" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">"black"</span></span>
<span id="cb6-717"><a href="#cb6-717" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb6-718"><a href="#cb6-718" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>) <span class="sc">+</span> </span>
<span id="cb6-719"><a href="#cb6-719" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb6-720"><a href="#cb6-720" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"PC3"</span>, </span>
<span id="cb6-721"><a href="#cb6-721" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"SET ER/PR"</span>, </span>
<span id="cb6-722"><a href="#cb6-722" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">"Is responder?"</span>,</span>
<span id="cb6-723"><a href="#cb6-723" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="fu">paste0</span>(</span>
<span id="cb6-724"><a href="#cb6-724" aria-hidden="true" tabindex="-1"></a>            <span class="st">"Scores for responders and non</span><span class="sc">\n</span><span class="st">responders of "</span>,</span>
<span id="cb6-725"><a href="#cb6-725" aria-hidden="true" tabindex="-1"></a>            <span class="st">"POETIC trial samples"</span></span>
<span id="cb6-726"><a href="#cb6-726" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb6-727"><a href="#cb6-727" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb6-728"><a href="#cb6-728" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span> </span>
<span id="cb6-729"><a href="#cb6-729" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb6-730"><a href="#cb6-730" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb6-731"><a href="#cb6-731" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-732"><a href="#cb6-732" aria-hidden="true" tabindex="-1"></a>This figure shows that there is a correlation between</span>
<span id="cb6-733"><a href="#cb6-733" aria-hidden="true" tabindex="-1"></a>third component and SET ER/PR for the POETIC trial data</span>
<span id="cb6-734"><a href="#cb6-734" aria-hidden="true" tabindex="-1"></a>as well as it was already seen from the other cohorts. </span>
<span id="cb6-735"><a href="#cb6-735" aria-hidden="true" tabindex="-1"></a>Moreover, it seems that the responders have higher PC scores</span>
<span id="cb6-736"><a href="#cb6-736" aria-hidden="true" tabindex="-1"></a>in general. We now correlate for each subgroup (responder and non </span>
<span id="cb6-737"><a href="#cb6-737" aria-hidden="true" tabindex="-1"></a>responders) the PC3 and the change in percentage for Ki67.</span>
<span id="cb6-738"><a href="#cb6-738" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-739"><a href="#cb6-739" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=6}</span></span>
<span id="cb6-740"><a href="#cb6-740" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-scores-corr-poetic</span></span>
<span id="cb6-741"><a href="#cb6-741" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Correlation between change in Ki67 versus PC3 stratified by</span></span>
<span id="cb6-742"><a href="#cb6-742" aria-hidden="true" tabindex="-1"></a><span class="co">#|     response group.</span></span>
<span id="cb6-743"><a href="#cb6-743" aria-hidden="true" tabindex="-1"></a>responder_non_responder_base <span class="ot">&lt;-</span> df_pca_coordinates <span class="sc">%&gt;%</span> </span>
<span id="cb6-744"><a href="#cb6-744" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb6-745"><a href="#cb6-745" aria-hidden="true" tabindex="-1"></a>        cohort <span class="sc">==</span> <span class="st">"poetic"</span> <span class="sc">&amp;</span> timepoint <span class="sc">==</span> <span class="st">"baseline"</span> <span class="sc">&amp;</span></span>
<span id="cb6-746"><a href="#cb6-746" aria-hidden="true" tabindex="-1"></a>            is_responder <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"non_responder"</span>, <span class="st">"responder"</span>)</span>
<span id="cb6-747"><a href="#cb6-747" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span> </span>
<span id="cb6-748"><a href="#cb6-748" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">is_responder =</span> <span class="fu">factor</span>(</span>
<span id="cb6-749"><a href="#cb6-749" aria-hidden="true" tabindex="-1"></a>        is_responder, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"responder"</span>, <span class="st">"non_responder"</span>)</span>
<span id="cb6-750"><a href="#cb6-750" aria-hidden="true" tabindex="-1"></a>    ))</span>
<span id="cb6-751"><a href="#cb6-751" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-752"><a href="#cb6-752" aria-hidden="true" tabindex="-1"></a><span class="co"># calculates the correlation and also get the p-values</span></span>
<span id="cb6-753"><a href="#cb6-753" aria-hidden="true" tabindex="-1"></a>cor_tests <span class="ot">&lt;-</span> responder_non_responder_base <span class="sc">%&gt;%</span> </span>
<span id="cb6-754"><a href="#cb6-754" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">nest</span>(<span class="at">data =</span> <span class="sc">-</span>is_responder) <span class="sc">%&gt;%</span></span>
<span id="cb6-755"><a href="#cb6-755" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb6-756"><a href="#cb6-756" aria-hidden="true" tabindex="-1"></a>        <span class="at">test =</span> <span class="fu">map</span>(data, <span class="sc">~</span> <span class="fu">cor.test</span>(.x<span class="sc">$</span>change_ki67, .x<span class="sc">$</span>PC3)),</span>
<span id="cb6-757"><a href="#cb6-757" aria-hidden="true" tabindex="-1"></a>        <span class="at">tidied =</span> <span class="fu">map</span>(test, broom<span class="sc">::</span>tidy)</span>
<span id="cb6-758"><a href="#cb6-758" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span> </span>
<span id="cb6-759"><a href="#cb6-759" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">unnest</span>(tidied) <span class="sc">%&gt;%</span> </span>
<span id="cb6-760"><a href="#cb6-760" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>dplyr<span class="sc">::</span><span class="fu">all_of</span>(<span class="st">"data"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb6-761"><a href="#cb6-761" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb6-762"><a href="#cb6-762" aria-hidden="true" tabindex="-1"></a>        <span class="at">label =</span> <span class="fu">paste0</span>(</span>
<span id="cb6-763"><a href="#cb6-763" aria-hidden="true" tabindex="-1"></a>            <span class="st">"\U03C1: "</span>, </span>
<span id="cb6-764"><a href="#cb6-764" aria-hidden="true" tabindex="-1"></a>            <span class="fu">format</span>(estimate, <span class="at">digits =</span> <span class="dv">2</span>), </span>
<span id="cb6-765"><a href="#cb6-765" aria-hidden="true" tabindex="-1"></a>            <span class="st">"</span><span class="sc">\n</span><span class="st"> p-val: "</span>, </span>
<span id="cb6-766"><a href="#cb6-766" aria-hidden="true" tabindex="-1"></a>            <span class="fu">format</span>(p.value, <span class="at">digits =</span> <span class="dv">2</span>)</span>
<span id="cb6-767"><a href="#cb6-767" aria-hidden="true" tabindex="-1"></a>        ),</span>
<span id="cb6-768"><a href="#cb6-768" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">90</span>, <span class="sc">-</span><span class="dv">20</span>),</span>
<span id="cb6-769"><a href="#cb6-769" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">5</span>, <span class="dv">4</span>)</span>
<span id="cb6-770"><a href="#cb6-770" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb6-771"><a href="#cb6-771" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-772"><a href="#cb6-772" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> responder_non_responder_base <span class="sc">%&gt;%</span></span>
<span id="cb6-773"><a href="#cb6-773" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> change_ki67, <span class="at">y =</span> PC3)) <span class="sc">+</span></span>
<span id="cb6-774"><a href="#cb6-774" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb6-775"><a href="#cb6-775" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>) <span class="sc">+</span> </span>
<span id="cb6-776"><a href="#cb6-776" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>is_responder, <span class="at">scale =</span> <span class="st">"free_x"</span>) <span class="sc">+</span></span>
<span id="cb6-777"><a href="#cb6-777" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb6-778"><a href="#cb6-778" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"Change in Ki67"</span>,</span>
<span id="cb6-779"><a href="#cb6-779" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">"Pearson correlation for each subgroup separately"</span></span>
<span id="cb6-780"><a href="#cb6-780" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb6-781"><a href="#cb6-781" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>) <span class="sc">+</span> </span>
<span id="cb6-782"><a href="#cb6-782" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span> </span>
<span id="cb6-783"><a href="#cb6-783" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb6-784"><a href="#cb6-784" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-785"><a href="#cb6-785" aria-hidden="true" tabindex="-1"></a>p <span class="sc">+</span> ggplot2<span class="sc">::</span><span class="fu">geom_label</span>(</span>
<span id="cb6-786"><a href="#cb6-786" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> cor_tests,</span>
<span id="cb6-787"><a href="#cb6-787" aria-hidden="true" tabindex="-1"></a>    <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> y, <span class="at">label =</span> label)</span>
<span id="cb6-788"><a href="#cb6-788" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb6-789"><a href="#cb6-789" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb6-790"><a href="#cb6-790" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-791"><a href="#cb6-791" aria-hidden="true" tabindex="-1"></a>Now another metric we can calculate from our dataset is the difference</span>
<span id="cb6-792"><a href="#cb6-792" aria-hidden="true" tabindex="-1"></a>of the embedding positions </span>
<span id="cb6-793"><a href="#cb6-793" aria-hidden="true" tabindex="-1"></a>between the different samples of the same patient. The idea is that if a </span>
<span id="cb6-794"><a href="#cb6-794" aria-hidden="true" tabindex="-1"></a>patient responded well, there were bigger changes as well in the</span>
<span id="cb6-795"><a href="#cb6-795" aria-hidden="true" tabindex="-1"></a>transcriptomics and that is reflected in the embedding. </span>
<span id="cb6-796"><a href="#cb6-796" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-797"><a href="#cb6-797" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=7}</span></span>
<span id="cb6-798"><a href="#cb6-798" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-diff-embedding-poetic</span></span>
<span id="cb6-799"><a href="#cb6-799" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Differences of the embeddings for each patient individually</span></span>
<span id="cb6-800"><a href="#cb6-800" aria-hidden="true" tabindex="-1"></a><span class="co">#|     stratified by response status. Each dot corresponds to a single patient.</span></span>
<span id="cb6-801"><a href="#cb6-801" aria-hidden="true" tabindex="-1"></a>diff_positions_poetic <span class="ot">&lt;-</span> df_pca_coordinates <span class="sc">%&gt;%</span></span>
<span id="cb6-802"><a href="#cb6-802" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"poetic"</span> <span class="sc">&amp;</span> </span>
<span id="cb6-803"><a href="#cb6-803" aria-hidden="true" tabindex="-1"></a>                      is_responder <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"responder"</span>, <span class="st">"non_responder"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb6-804"><a href="#cb6-804" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">pivot_wider</span>(</span>
<span id="cb6-805"><a href="#cb6-805" aria-hidden="true" tabindex="-1"></a>        <span class="at">id_cols =</span> <span class="fu">c</span>(patient_nb, is_responder, change_ki67),</span>
<span id="cb6-806"><a href="#cb6-806" aria-hidden="true" tabindex="-1"></a>        <span class="at">names_from =</span> timepoint, </span>
<span id="cb6-807"><a href="#cb6-807" aria-hidden="true" tabindex="-1"></a>        <span class="at">values_from =</span> <span class="fu">c</span>(PC3, PC4)</span>
<span id="cb6-808"><a href="#cb6-808" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb6-809"><a href="#cb6-809" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">summarise</span>(</span>
<span id="cb6-810"><a href="#cb6-810" aria-hidden="true" tabindex="-1"></a>        <span class="at">diff_positions =</span> <span class="fu">sqrt</span>(</span>
<span id="cb6-811"><a href="#cb6-811" aria-hidden="true" tabindex="-1"></a>            (PC3_baseline <span class="sc">-</span> PC3_surgery)<span class="sc">^</span><span class="dv">2</span> <span class="sc">+</span> </span>
<span id="cb6-812"><a href="#cb6-812" aria-hidden="true" tabindex="-1"></a>                (PC4_baseline <span class="sc">-</span> PC4_surgery)<span class="sc">^</span><span class="dv">2</span> </span>
<span id="cb6-813"><a href="#cb6-813" aria-hidden="true" tabindex="-1"></a>        ),</span>
<span id="cb6-814"><a href="#cb6-814" aria-hidden="true" tabindex="-1"></a>        <span class="at">is_responder =</span> is_responder,</span>
<span id="cb6-815"><a href="#cb6-815" aria-hidden="true" tabindex="-1"></a>        <span class="at">change_ki67 =</span> change_ki67</span>
<span id="cb6-816"><a href="#cb6-816" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb6-817"><a href="#cb6-817" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">is_responder =</span> <span class="fu">factor</span>(</span>
<span id="cb6-818"><a href="#cb6-818" aria-hidden="true" tabindex="-1"></a>        is_responder, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"responder"</span>, <span class="st">"non_responder"</span>)</span>
<span id="cb6-819"><a href="#cb6-819" aria-hidden="true" tabindex="-1"></a>    ))</span>
<span id="cb6-820"><a href="#cb6-820" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-821"><a href="#cb6-821" aria-hidden="true" tabindex="-1"></a>diff_positions_poetic <span class="sc">%&gt;%</span></span>
<span id="cb6-822"><a href="#cb6-822" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> is_responder, <span class="at">y =</span> diff_positions)) <span class="sc">+</span></span>
<span id="cb6-823"><a href="#cb6-823" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb6-824"><a href="#cb6-824" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_jitter</span>() <span class="sc">+</span> </span>
<span id="cb6-825"><a href="#cb6-825" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb6-826"><a href="#cb6-826" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"Is responder?"</span>,</span>
<span id="cb6-827"><a href="#cb6-827" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"Difference in positions of the embedding"</span></span>
<span id="cb6-828"><a href="#cb6-828" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb6-829"><a href="#cb6-829" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>) <span class="sc">+</span></span>
<span id="cb6-830"><a href="#cb6-830" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span></span>
<span id="cb6-831"><a href="#cb6-831" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb6-832"><a href="#cb6-832" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb6-833"><a href="#cb6-833" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-834"><a href="#cb6-834" aria-hidden="true" tabindex="-1"></a>And now we perform a comparison of these positions by using</span>
<span id="cb6-835"><a href="#cb6-835" aria-hidden="true" tabindex="-1"></a>rstanarm to get the posterior distributions of the differences.</span>
<span id="cb6-836"><a href="#cb6-836" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-839"><a href="#cb6-839" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb6-840"><a href="#cb6-840" aria-hidden="true" tabindex="-1"></a>fit_differences <span class="ot">&lt;-</span> rstanarm<span class="sc">::</span><span class="fu">stan_glm</span>(</span>
<span id="cb6-841"><a href="#cb6-841" aria-hidden="true" tabindex="-1"></a>    <span class="at">formula =</span> diff_positions <span class="sc">~</span> is_responder,</span>
<span id="cb6-842"><a href="#cb6-842" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> diff_positions_poetic, </span>
<span id="cb6-843"><a href="#cb6-843" aria-hidden="true" tabindex="-1"></a>    <span class="at">refresh =</span> <span class="dv">0</span> </span>
<span id="cb6-844"><a href="#cb6-844" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb6-845"><a href="#cb6-845" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-846"><a href="#cb6-846" aria-hidden="true" tabindex="-1"></a>diff_means <span class="ot">&lt;-</span> fit_differences <span class="sc">%&gt;%</span></span>
<span id="cb6-847"><a href="#cb6-847" aria-hidden="true" tabindex="-1"></a>    tidybayes<span class="sc">::</span><span class="fu">spread_draws</span>(is_respondernon_responder)</span>
<span id="cb6-848"><a href="#cb6-848" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-849"><a href="#cb6-849" aria-hidden="true" tabindex="-1"></a>means <span class="ot">&lt;-</span> diff_positions_poetic <span class="sc">%&gt;%</span></span>
<span id="cb6-850"><a href="#cb6-850" aria-hidden="true" tabindex="-1"></a>    modelr<span class="sc">::</span><span class="fu">data_grid</span>(is_responder) <span class="sc">%&gt;%</span></span>
<span id="cb6-851"><a href="#cb6-851" aria-hidden="true" tabindex="-1"></a>    tidybayes<span class="sc">::</span><span class="fu">add_epred_draws</span>(fit_differences)</span>
<span id="cb6-852"><a href="#cb6-852" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb6-853"><a href="#cb6-853" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-854"><a href="#cb6-854" aria-hidden="true" tabindex="-1"></a>The figure below shows the distribution of the averages of the differences</span>
<span id="cb6-855"><a href="#cb6-855" aria-hidden="true" tabindex="-1"></a>in the embedding of the projections.</span>
<span id="cb6-856"><a href="#cb6-856" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-859"><a href="#cb6-859" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb6-860"><a href="#cb6-860" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-diff-embedding-poetic-averages</span></span>
<span id="cb6-861"><a href="#cb6-861" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Differences of the embeddings for each patient individually</span></span>
<span id="cb6-862"><a href="#cb6-862" aria-hidden="true" tabindex="-1"></a><span class="co">#|     stratified by response status. Each dot corresponds to a single patient.</span></span>
<span id="cb6-863"><a href="#cb6-863" aria-hidden="true" tabindex="-1"></a>means <span class="sc">%&gt;%</span></span>
<span id="cb6-864"><a href="#cb6-864" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> .epred, <span class="at">y =</span> is_responder)) <span class="sc">+</span></span>
<span id="cb6-865"><a href="#cb6-865" aria-hidden="true" tabindex="-1"></a>    tidybayes<span class="sc">::</span><span class="fu">stat_dotsinterval</span>(<span class="at">quantiles =</span> <span class="dv">100</span>) <span class="sc">+</span> </span>
<span id="cb6-866"><a href="#cb6-866" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb6-867"><a href="#cb6-867" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"Average distance of embedding positions"</span>,</span>
<span id="cb6-868"><a href="#cb6-868" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">""</span></span>
<span id="cb6-869"><a href="#cb6-869" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb6-870"><a href="#cb6-870" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb6-871"><a href="#cb6-871" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>()</span>
<span id="cb6-872"><a href="#cb6-872" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb6-873"><a href="#cb6-873" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-874"><a href="#cb6-874" aria-hidden="true" tabindex="-1"></a>And the figure below shows the posterior distribution of the differences</span>
<span id="cb6-875"><a href="#cb6-875" aria-hidden="true" tabindex="-1"></a>in average.</span>
<span id="cb6-876"><a href="#cb6-876" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-879"><a href="#cb6-879" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb6-880"><a href="#cb6-880" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-diff-embedding-poetic-averages-diff</span></span>
<span id="cb6-881"><a href="#cb6-881" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Differences of the average differences in the embedding for </span></span>
<span id="cb6-882"><a href="#cb6-882" aria-hidden="true" tabindex="-1"></a><span class="co">#|    each patient individually. The comparison made was non responder vs </span></span>
<span id="cb6-883"><a href="#cb6-883" aria-hidden="true" tabindex="-1"></a><span class="co">#|    responder, which means that if non responder has a lower difference in</span></span>
<span id="cb6-884"><a href="#cb6-884" aria-hidden="true" tabindex="-1"></a><span class="co">#|    average, the difference in difference will be negative.</span></span>
<span id="cb6-885"><a href="#cb6-885" aria-hidden="true" tabindex="-1"></a>diff_means <span class="sc">%&gt;%</span></span>
<span id="cb6-886"><a href="#cb6-886" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> is_respondernon_responder)) <span class="sc">+</span></span>
<span id="cb6-887"><a href="#cb6-887" aria-hidden="true" tabindex="-1"></a>    tidybayes<span class="sc">::</span><span class="fu">stat_dotsinterval</span>(<span class="at">quantiles =</span> <span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb6-888"><a href="#cb6-888" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb6-889"><a href="#cb6-889" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb6-890"><a href="#cb6-890" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"Average difference of the embedding differences"</span>,</span>
<span id="cb6-891"><a href="#cb6-891" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"Density"</span>,</span>
<span id="cb6-892"><a href="#cb6-892" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">""</span></span>
<span id="cb6-893"><a href="#cb6-893" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb6-894"><a href="#cb6-894" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>()</span>
<span id="cb6-895"><a href="#cb6-895" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb6-896"><a href="#cb6-896" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-897"><a href="#cb6-897" aria-hidden="true" tabindex="-1"></a><span class="fu">## Analysing patient neighborhoods</span></span>
<span id="cb6-898"><a href="#cb6-898" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-899"><a href="#cb6-899" aria-hidden="true" tabindex="-1"></a>There are two ways to interpret the embedding. If the patient has</span>
<span id="cb6-900"><a href="#cb6-900" aria-hidden="true" tabindex="-1"></a>ER+ BC and its embedding is close to the ER- BC patients, we can</span>
<span id="cb6-901"><a href="#cb6-901" aria-hidden="true" tabindex="-1"></a>infer that the endocrine response might not work so well, as </span>
<span id="cb6-902"><a href="#cb6-902" aria-hidden="true" tabindex="-1"></a>it was shown in @fig-pca-responders-baseline. The other option</span>
<span id="cb6-903"><a href="#cb6-903" aria-hidden="true" tabindex="-1"></a>is to compare the patient's score to the average score of </span>
<span id="cb6-904"><a href="#cb6-904" aria-hidden="true" tabindex="-1"></a>its neighborhood.</span>
<span id="cb6-905"><a href="#cb6-905" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-906"><a href="#cb6-906" aria-hidden="true" tabindex="-1"></a>What is a neighborhood? When performing the embedding, global</span>
<span id="cb6-907"><a href="#cb6-907" aria-hidden="true" tabindex="-1"></a>structures are not preserved usually. In this sense we only</span>
<span id="cb6-908"><a href="#cb6-908" aria-hidden="true" tabindex="-1"></a>compare patients that are close to each other in an euclidean </span>
<span id="cb6-909"><a href="#cb6-909" aria-hidden="true" tabindex="-1"></a>neighborhood, i.e., we calculate a radius around each sample and</span>
<span id="cb6-910"><a href="#cb6-910" aria-hidden="true" tabindex="-1"></a>see what other samples are within this ball given the euclidean</span>
<span id="cb6-911"><a href="#cb6-911" aria-hidden="true" tabindex="-1"></a>distance.</span>
<span id="cb6-912"><a href="#cb6-912" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-913"><a href="#cb6-913" aria-hidden="true" tabindex="-1"></a>Given a neighborhood, one can calculate the posterior average score</span>
<span id="cb6-914"><a href="#cb6-914" aria-hidden="true" tabindex="-1"></a>of all samples from METABRIC, SCANB and TCGA and use that as an </span>
<span id="cb6-915"><a href="#cb6-915" aria-hidden="true" tabindex="-1"></a>indication of average score that we can compare other patients to.</span>
<span id="cb6-916"><a href="#cb6-916" aria-hidden="true" tabindex="-1"></a>The concept is that if an ER+ BC patient has a much lower ER signaling</span>
<span id="cb6-917"><a href="#cb6-917" aria-hidden="true" tabindex="-1"></a>score, it means the patients will not respond so well to endocrine </span>
<span id="cb6-918"><a href="#cb6-918" aria-hidden="true" tabindex="-1"></a>therapy, as they are very different from their neighboors.</span>
<span id="cb6-919"><a href="#cb6-919" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-920"><a href="#cb6-920" aria-hidden="true" tabindex="-1"></a>To showcase the ability to use the scores and infer results, we </span>
<span id="cb6-921"><a href="#cb6-921" aria-hidden="true" tabindex="-1"></a>use the POETIC trial patients to compare the scores. This cohort</span>
<span id="cb6-922"><a href="#cb6-922" aria-hidden="true" tabindex="-1"></a>is special in the sense that the patients are filtered based on</span>
<span id="cb6-923"><a href="#cb6-923" aria-hidden="true" tabindex="-1"></a>selection criterias, meaning that the patients are clinically</span>
<span id="cb6-924"><a href="#cb6-924" aria-hidden="true" tabindex="-1"></a>similar. </span>
<span id="cb6-925"><a href="#cb6-925" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-926"><a href="#cb6-926" aria-hidden="true" tabindex="-1"></a>Two random patients in very close neighborhoods were selected</span>
<span id="cb6-927"><a href="#cb6-927" aria-hidden="true" tabindex="-1"></a>to showcase the ability of the molecular landscape.</span>
<span id="cb6-928"><a href="#cb6-928" aria-hidden="true" tabindex="-1"></a>The id of the patients are 63 and 236.</span>
<span id="cb6-929"><a href="#cb6-929" aria-hidden="true" tabindex="-1"></a>@tbl-pt63-236 shows the clinical features of these patients. One</span>
<span id="cb6-930"><a href="#cb6-930" aria-hidden="true" tabindex="-1"></a>of them is HER2+. Patient 63 had a higher Ki67 baseline level but</span>
<span id="cb6-931"><a href="#cb6-931" aria-hidden="true" tabindex="-1"></a>a very good response to endocrine therapy. Patient 236 did not </span>
<span id="cb6-932"><a href="#cb6-932" aria-hidden="true" tabindex="-1"></a>respond to endocrine therapy in terms of reduction of Ki67 levels.</span>
<span id="cb6-933"><a href="#cb6-933" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-936"><a href="#cb6-936" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb6-937"><a href="#cb6-937" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: tbl-pt63-236</span></span>
<span id="cb6-938"><a href="#cb6-938" aria-hidden="true" tabindex="-1"></a><span class="co">#| tbl-cap: Clinical features from patients 63 and 236. </span></span>
<span id="cb6-939"><a href="#cb6-939" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-940"><a href="#cb6-940" aria-hidden="true" tabindex="-1"></a>patients <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"treated_nb63_baseline"</span>, <span class="st">"treated_nb236_baseline"</span>)</span>
<span id="cb6-941"><a href="#cb6-941" aria-hidden="true" tabindex="-1"></a>poetic_df_pca <span class="sc">%&gt;%</span> </span>
<span id="cb6-942"><a href="#cb6-942" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span></span>
<span id="cb6-943"><a href="#cb6-943" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(sample_name <span class="sc">%in%</span> patients) <span class="sc">%&gt;%</span></span>
<span id="cb6-944"><a href="#cb6-944" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(</span>
<span id="cb6-945"><a href="#cb6-945" aria-hidden="true" tabindex="-1"></a>        patient_nb, er_status, her2_status, ki67, change_ki67, </span>
<span id="cb6-946"><a href="#cb6-946" aria-hidden="true" tabindex="-1"></a>        ccca_surgery_ki67_2_7</span>
<span id="cb6-947"><a href="#cb6-947" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb6-948"><a href="#cb6-948" aria-hidden="true" tabindex="-1"></a>    kableExtra<span class="sc">::</span><span class="fu">kbl</span>(<span class="at">row.names =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb6-949"><a href="#cb6-949" aria-hidden="true" tabindex="-1"></a>    kableExtra<span class="sc">::</span><span class="fu">kable_classic</span>()</span>
<span id="cb6-950"><a href="#cb6-950" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb6-951"><a href="#cb6-951" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-952"><a href="#cb6-952" aria-hidden="true" tabindex="-1"></a>@fig-pt63-236 shows the embedding of two distinct patients </span>
<span id="cb6-953"><a href="#cb6-953" aria-hidden="true" tabindex="-1"></a>in a similar neighborhood.</span>
<span id="cb6-954"><a href="#cb6-954" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-955"><a href="#cb6-955" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=7}</span></span>
<span id="cb6-956"><a href="#cb6-956" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-pt63-236</span></span>
<span id="cb6-957"><a href="#cb6-957" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Plot of two selected patients in the molecular landscape. One</span></span>
<span id="cb6-958"><a href="#cb6-958" aria-hidden="true" tabindex="-1"></a><span class="co">#|     is a responder and the other is a non responder. The patient id numbers</span></span>
<span id="cb6-959"><a href="#cb6-959" aria-hidden="true" tabindex="-1"></a><span class="co">#|     are 63 and 236.</span></span>
<span id="cb6-960"><a href="#cb6-960" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-961"><a href="#cb6-961" aria-hidden="true" tabindex="-1"></a>patients <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"treated_nb63_baseline"</span>, <span class="st">"treated_nb236_baseline"</span>)</span>
<span id="cb6-962"><a href="#cb6-962" aria-hidden="true" tabindex="-1"></a>remove_patients <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(datasets<span class="sc">$</span>poetic <span class="sc">%&gt;%</span> colnames, patients)</span>
<span id="cb6-963"><a href="#cb6-963" aria-hidden="true" tabindex="-1"></a>poetic_df_pca_baseline <span class="ot">&lt;-</span> poetic_df_pca <span class="sc">%&gt;%</span> </span>
<span id="cb6-964"><a href="#cb6-964" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span>(sample_name <span class="sc">%in%</span> remove_patients))</span>
<span id="cb6-965"><a href="#cb6-965" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-966"><a href="#cb6-966" aria-hidden="true" tabindex="-1"></a>base_plot <span class="ot">&lt;-</span> <span class="fu">get_base_plot</span>(</span>
<span id="cb6-967"><a href="#cb6-967" aria-hidden="true" tabindex="-1"></a>    poetic_df_pca <span class="sc">%&gt;%</span> </span>
<span id="cb6-968"><a href="#cb6-968" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"tcga"</span>, <span class="st">"metabric"</span>, <span class="st">"scanb"</span>)),</span>
<span id="cb6-969"><a href="#cb6-969" aria-hidden="true" tabindex="-1"></a>    <span class="at">size_dots =</span> <span class="dv">2</span>,</span>
<span id="cb6-970"><a href="#cb6-970" aria-hidden="true" tabindex="-1"></a>    <span class="at">alpha_val =</span> <span class="fl">0.1</span>,</span>
<span id="cb6-971"><a href="#cb6-971" aria-hidden="true" tabindex="-1"></a>    <span class="at">size_legend =</span> <span class="dv">4</span>,</span>
<span id="cb6-972"><a href="#cb6-972" aria-hidden="true" tabindex="-1"></a>    <span class="at">base_size =</span> <span class="dv">10</span></span>
<span id="cb6-973"><a href="#cb6-973" aria-hidden="true" tabindex="-1"></a>)    </span>
<span id="cb6-974"><a href="#cb6-974" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> poetic_df_pca_baseline <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"poetic"</span>)</span>
<span id="cb6-975"><a href="#cb6-975" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-976"><a href="#cb6-976" aria-hidden="true" tabindex="-1"></a>plot_patients <span class="ot">&lt;-</span> base_plot <span class="sc">+</span> </span>
<span id="cb6-977"><a href="#cb6-977" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>(</span>
<span id="cb6-978"><a href="#cb6-978" aria-hidden="true" tabindex="-1"></a>        <span class="at">data =</span> df,</span>
<span id="cb6-979"><a href="#cb6-979" aria-hidden="true" tabindex="-1"></a>        <span class="at">mapping =</span> <span class="fu">aes</span>(</span>
<span id="cb6-980"><a href="#cb6-980" aria-hidden="true" tabindex="-1"></a>            <span class="at">shape =</span> timepoint,</span>
<span id="cb6-981"><a href="#cb6-981" aria-hidden="true" tabindex="-1"></a>            <span class="at">color =</span> <span class="cn">NA</span>,</span>
<span id="cb6-982"><a href="#cb6-982" aria-hidden="true" tabindex="-1"></a>            <span class="at">fill =</span> is_responder</span>
<span id="cb6-983"><a href="#cb6-983" aria-hidden="true" tabindex="-1"></a>        ),</span>
<span id="cb6-984"><a href="#cb6-984" aria-hidden="true" tabindex="-1"></a>        <span class="at">size =</span> <span class="dv">7</span>,</span>
<span id="cb6-985"><a href="#cb6-985" aria-hidden="true" tabindex="-1"></a>        <span class="at">shape =</span> <span class="dv">22</span></span>
<span id="cb6-986"><a href="#cb6-986" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb6-987"><a href="#cb6-987" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_fill_viridis_d</span>() <span class="sc">+</span></span>
<span id="cb6-988"><a href="#cb6-988" aria-hidden="true" tabindex="-1"></a>    ggrepel<span class="sc">::</span><span class="fu">geom_label_repel</span>(</span>
<span id="cb6-989"><a href="#cb6-989" aria-hidden="true" tabindex="-1"></a>        <span class="at">data =</span> df <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">patient_nb =</span> <span class="fu">paste0</span>(<span class="st">"#"</span>, patient_nb)),</span>
<span id="cb6-990"><a href="#cb6-990" aria-hidden="true" tabindex="-1"></a>        <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> PC3, <span class="at">y =</span> PC4, <span class="at">label =</span> patient_nb),</span>
<span id="cb6-991"><a href="#cb6-991" aria-hidden="true" tabindex="-1"></a>        <span class="at">inherit.aes =</span> <span class="cn">FALSE</span>,</span>
<span id="cb6-992"><a href="#cb6-992" aria-hidden="true" tabindex="-1"></a>        <span class="at">size =</span> <span class="dv">10</span>,</span>
<span id="cb6-993"><a href="#cb6-993" aria-hidden="true" tabindex="-1"></a>        <span class="at">box.padding =</span> <span class="fl">0.5</span></span>
<span id="cb6-994"><a href="#cb6-994" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb6-995"><a href="#cb6-995" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb6-996"><a href="#cb6-996" aria-hidden="true" tabindex="-1"></a>        <span class="at">fill =</span> <span class="st">"Is responder?"</span>,</span>
<span id="cb6-997"><a href="#cb6-997" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="fu">paste0</span>(</span>
<span id="cb6-998"><a href="#cb6-998" aria-hidden="true" tabindex="-1"></a>            <span class="st">"Responder status for each patient in the baseline</span><span class="sc">\n</span><span class="st">"</span>,</span>
<span id="cb6-999"><a href="#cb6-999" aria-hidden="true" tabindex="-1"></a>            <span class="st">"and treated group"</span></span>
<span id="cb6-1000"><a href="#cb6-1000" aria-hidden="true" tabindex="-1"></a>        ),</span>
<span id="cb6-1001"><a href="#cb6-1001" aria-hidden="true" tabindex="-1"></a>        <span class="at">caption =</span> <span class="fu">paste0</span>(</span>
<span id="cb6-1002"><a href="#cb6-1002" aria-hidden="true" tabindex="-1"></a>            <span class="st">"Responder is defined as having at least 5% baseline Ki67</span><span class="sc">\n</span><span class="st">and"</span>,</span>
<span id="cb6-1003"><a href="#cb6-1003" aria-hidden="true" tabindex="-1"></a>            <span class="st">" reduction of at least 60% in Ki67 levels"</span></span>
<span id="cb6-1004"><a href="#cb6-1004" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb6-1005"><a href="#cb6-1005" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb6-1006"><a href="#cb6-1006" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">18</span>) <span class="sc">+</span></span>
<span id="cb6-1007"><a href="#cb6-1007" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>()</span>
<span id="cb6-1008"><a href="#cb6-1008" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-1009"><a href="#cb6-1009" aria-hidden="true" tabindex="-1"></a>plot_patients</span>
<span id="cb6-1010"><a href="#cb6-1010" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb6-1011"><a href="#cb6-1011" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-1012"><a href="#cb6-1012" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-1013"><a href="#cb6-1013" aria-hidden="true" tabindex="-1"></a>And now we calculate the average scores for each patient neighborhoods. We </span>
<span id="cb6-1014"><a href="#cb6-1014" aria-hidden="true" tabindex="-1"></a>start by defining a radius value. After that we select the patients in the</span>
<span id="cb6-1015"><a href="#cb6-1015" aria-hidden="true" tabindex="-1"></a>neighborhood and use <span class="in">`rstanarm`</span> to get the posterior distribution of the</span>
<span id="cb6-1016"><a href="#cb6-1016" aria-hidden="true" tabindex="-1"></a>average score in the neighborhood. For this we use the function</span>
<span id="cb6-1017"><a href="#cb6-1017" aria-hidden="true" tabindex="-1"></a><span class="in">`stan_glm`</span> to calculate the average. This is effective because it can</span>
<span id="cb6-1018"><a href="#cb6-1018" aria-hidden="true" tabindex="-1"></a>be paired with the package <span class="in">`tidybayes`</span> for plotting. It makes extremely easy</span>
<span id="cb6-1019"><a href="#cb6-1019" aria-hidden="true" tabindex="-1"></a>to deal with posterior data. For the average we use a </span>
<span id="cb6-1020"><a href="#cb6-1020" aria-hidden="true" tabindex="-1"></a>normal prior centered at 0 with 1 standard deviation, </span>
<span id="cb6-1021"><a href="#cb6-1021" aria-hidden="true" tabindex="-1"></a>since all scores are between -1 and 1.</span>
<span id="cb6-1022"><a href="#cb6-1022" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-1023"><a href="#cb6-1023" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb6-1024"><a href="#cb6-1024" aria-hidden="true" tabindex="-1"></a>patients <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"treated_nb63_baseline"</span>, <span class="st">"treated_nb236_baseline"</span>)</span>
<span id="cb6-1025"><a href="#cb6-1025" aria-hidden="true" tabindex="-1"></a>scores_to_use <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb6-1026"><a href="#cb6-1026" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Estrogen response early"</span>,</span>
<span id="cb6-1027"><a href="#cb6-1027" aria-hidden="true" tabindex="-1"></a>    <span class="st">"E2F targets"</span>,</span>
<span id="cb6-1028"><a href="#cb6-1028" aria-hidden="true" tabindex="-1"></a>    <span class="st">"P53 pathway"</span>,</span>
<span id="cb6-1029"><a href="#cb6-1029" aria-hidden="true" tabindex="-1"></a>    <span class="st">"EMT"</span>,</span>
<span id="cb6-1030"><a href="#cb6-1030" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Androgen response"</span>,</span>
<span id="cb6-1031"><a href="#cb6-1031" aria-hidden="true" tabindex="-1"></a>    <span class="st">"TGFb signaling"</span></span>
<span id="cb6-1032"><a href="#cb6-1032" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> <span class="st">`</span><span class="at">names&lt;-</span><span class="st">`</span>(<span class="fu">c</span>(</span>
<span id="cb6-1033"><a href="#cb6-1033" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_ESTROGEN_RESPONSE_EARLY"</span>,</span>
<span id="cb6-1034"><a href="#cb6-1034" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_E2F_TARGETS"</span>,</span>
<span id="cb6-1035"><a href="#cb6-1035" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_P53_PATHWAY"</span>,</span>
<span id="cb6-1036"><a href="#cb6-1036" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION"</span>,</span>
<span id="cb6-1037"><a href="#cb6-1037" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_ANDROGEN_RESPONSE"</span>,</span>
<span id="cb6-1038"><a href="#cb6-1038" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_TGF_BETA_SIGNALING"</span></span>
<span id="cb6-1039"><a href="#cb6-1039" aria-hidden="true" tabindex="-1"></a>))</span>
<span id="cb6-1040"><a href="#cb6-1040" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-1041"><a href="#cb6-1041" aria-hidden="true" tabindex="-1"></a>pipeline_scores_plots <span class="ot">&lt;-</span> <span class="fu">sapply</span>(</span>
<span id="cb6-1042"><a href="#cb6-1042" aria-hidden="true" tabindex="-1"></a>    patients, </span>
<span id="cb6-1043"><a href="#cb6-1043" aria-hidden="true" tabindex="-1"></a>    get_patient_scores_distributions_all,</span>
<span id="cb6-1044"><a href="#cb6-1044" aria-hidden="true" tabindex="-1"></a>    <span class="at">df_pca =</span> poetic_df_pca,</span>
<span id="cb6-1045"><a href="#cb6-1045" aria-hidden="true" tabindex="-1"></a>    <span class="at">scores_to_use =</span> scores_to_use,</span>
<span id="cb6-1046"><a href="#cb6-1046" aria-hidden="true" tabindex="-1"></a>    <span class="at">base_size =</span> <span class="dv">20</span>,</span>
<span id="cb6-1047"><a href="#cb6-1047" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>, </span>
<span id="cb6-1048"><a href="#cb6-1048" aria-hidden="true" tabindex="-1"></a>    <span class="at">simplify =</span> <span class="cn">FALSE</span></span>
<span id="cb6-1049"><a href="#cb6-1049" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb6-1050"><a href="#cb6-1050" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-1051"><a href="#cb6-1051" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-1052"><a href="#cb6-1052" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb6-1053"><a href="#cb6-1053" aria-hidden="true" tabindex="-1"></a>    pipeline_scores_plots,</span>
<span id="cb6-1054"><a href="#cb6-1054" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/scoring/pipeline_scores_plots.rds"</span></span>
<span id="cb6-1055"><a href="#cb6-1055" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb6-1056"><a href="#cb6-1056" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb6-1057"><a href="#cb6-1057" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-1058"><a href="#cb6-1058" aria-hidden="true" tabindex="-1"></a>Patient 63 @fig-pt63 is considered a responder. According to </span>
<span id="cb6-1059"><a href="#cb6-1059" aria-hidden="true" tabindex="-1"></a>the scores, when comparing the estrogen early signature to its average </span>
<span id="cb6-1060"><a href="#cb6-1060" aria-hidden="true" tabindex="-1"></a>distribution in the neighborhood, the score is higher.</span>
<span id="cb6-1061"><a href="#cb6-1061" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-1062"><a href="#cb6-1062" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.height=6, fig.width=10}</span></span>
<span id="cb6-1063"><a href="#cb6-1063" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-pt63</span></span>
<span id="cb6-1064"><a href="#cb6-1064" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Posterior distribution of the average scores in the neighborhood</span></span>
<span id="cb6-1065"><a href="#cb6-1065" aria-hidden="true" tabindex="-1"></a><span class="co">#|     of patient 236 from POETIC trial. Each dot corresponds to a 1% quantile.</span></span>
<span id="cb6-1066"><a href="#cb6-1066" aria-hidden="true" tabindex="-1"></a>pipeline_scores_plots<span class="sc">$</span>treated_nb63_baseline<span class="sc">$</span>plot <span class="sc">+</span></span>
<span id="cb6-1067"><a href="#cb6-1067" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>()</span>
<span id="cb6-1068"><a href="#cb6-1068" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb6-1069"><a href="#cb6-1069" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-1070"><a href="#cb6-1070" aria-hidden="true" tabindex="-1"></a>Patient 236 @fig-pt236 was considered a non responder. According to </span>
<span id="cb6-1071"><a href="#cb6-1071" aria-hidden="true" tabindex="-1"></a>the scores, when comparing the estrogen early signature to its average </span>
<span id="cb6-1072"><a href="#cb6-1072" aria-hidden="true" tabindex="-1"></a>distribution in the neighborhood, the score is lower, being in the</span>
<span id="cb6-1073"><a href="#cb6-1073" aria-hidden="true" tabindex="-1"></a>5% quantile.</span>
<span id="cb6-1074"><a href="#cb6-1074" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-1075"><a href="#cb6-1075" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.height=6, fig.width=10}</span></span>
<span id="cb6-1076"><a href="#cb6-1076" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-pt236</span></span>
<span id="cb6-1077"><a href="#cb6-1077" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Posterior distribution of the average scores in the neighborhood</span></span>
<span id="cb6-1078"><a href="#cb6-1078" aria-hidden="true" tabindex="-1"></a><span class="co">#|     of patient 236 from POETIC trial. Each dot corresponds to a 1% quantile.</span></span>
<span id="cb6-1079"><a href="#cb6-1079" aria-hidden="true" tabindex="-1"></a>pipeline_scores_plots<span class="sc">$</span>treated_nb236_baseline<span class="sc">$</span>plot <span class="sc">+</span></span>
<span id="cb6-1080"><a href="#cb6-1080" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>()</span>
<span id="cb6-1081"><a href="#cb6-1081" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb6-1082"><a href="#cb6-1082" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-1083"><a href="#cb6-1083" aria-hidden="true" tabindex="-1"></a>When comparing these two patients, there is also a difference in the androgen</span>
<span id="cb6-1084"><a href="#cb6-1084" aria-hidden="true" tabindex="-1"></a>response score, which could be a reflection of different estrogen signaling.</span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-right">
      <ul class="footer-items list-unstyled">
    <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/chronchi">
      <i class="bi bi-github" role="img">
</i> 
    </a>
  </li>  
    <li class="nav-item compact">
    <a class="nav-link" href="https://twitter.com/chronchi">
      <i class="bi bi-twitter" role="img">
</i> 
    </a>
  </li>  
</ul>
    </div>
  </div>
</footer>



</body></html>