<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.0.37">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Molecular landscape: a new framework for personalized medicine in Breast Cancer. - 4&nbsp; Pathway activity in a personalized context</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./risk_score.html" rel="next">
<link href="./validation.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>

<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet">
<script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="site_libs/datatables-binding-0.22/datatables.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="site_libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet">
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>

  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
    <div class="container-fluid d-flex justify-content-between">
      <h1 class="quarto-secondary-nav-title"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Pathway activity in a personalized context</span></h1>
      <button type="button" class="quarto-btn-toggle btn" aria-label="Show secondary navigation">
        <i class="bi bi-chevron-right"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Molecular landscape: a new framework for personalized medicine in Breast Cancer.</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/chronchi/molecular_landscape" title="Source Code" class="sidebar-tool px-1"><i class="bi bi-github"></i></a>
    <a href="https://twitter.com/intent/tweet?url=|url|" title="Twitter" class="sidebar-tool px-1"><i class="bi bi-twitter"></i></a>
</div>
    </div>
      </div>
      <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">Preface</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./intro.html" class="sidebar-item-text sidebar-link">Introduction</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./surv_analysis_estrogen.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Estrogen receptor status is continuous, not dichotomous</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./pca_merging.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">A new framework for personalized medicine: integrating</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./validation.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Validating the molecular landscape</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./scoring.html" class="sidebar-item-text sidebar-link active"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Pathway activity in a personalized context</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./risk_score.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Risk score model</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./trying.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Tests and tries</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./discussion.html" class="sidebar-item-text sidebar-link">Discussion</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./references.html" class="sidebar-item-text sidebar-link">References</a>
  </div>
</li>
    </ul>
    </div>
</nav>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#calculating-scores-in-neighborhoods" id="toc-calculating-scores-in-neighborhoods" class="nav-link active" data-scroll-target="#calculating-scores-in-neighborhoods"> <span class="header-section-number">4.1</span> Calculating scores in neighborhoods</a></li>
  <li><a href="#poetic-embedding" id="toc-poetic-embedding" class="nav-link" data-scroll-target="#poetic-embedding"> <span class="header-section-number">4.2</span> POETIC embedding</a>
  <ul class="collapse">
  <li><a href="#number-of-samples" id="toc-number-of-samples" class="nav-link" data-scroll-target="#number-of-samples"> <span class="header-section-number">4.2.1</span> Number of samples</a></li>
  <li><a href="#proportion-of-top-loadings" id="toc-proportion-of-top-loadings" class="nav-link" data-scroll-target="#proportion-of-top-loadings"> <span class="header-section-number">4.2.2</span> Proportion of top loadings</a></li>
  <li><a href="#embedding" id="toc-embedding" class="nav-link" data-scroll-target="#embedding"> <span class="header-section-number">4.2.3</span> Embedding</a></li>
  <li><a href="#differences-in-pc3-for-responders-and-non-responders" id="toc-differences-in-pc3-for-responders-and-non-responders" class="nav-link" data-scroll-target="#differences-in-pc3-for-responders-and-non-responders"> <span class="header-section-number">4.2.4</span> Differences in PC3 for responders and non responders</a></li>
  </ul></li>
  <li><a href="#scoring-all-samples" id="toc-scoring-all-samples" class="nav-link" data-scroll-target="#scoring-all-samples"> <span class="header-section-number">4.3</span> Scoring all samples</a></li>
  <li><a href="#analysing-patient-neighborhoods" id="toc-analysing-patient-neighborhoods" class="nav-link" data-scroll-target="#analysing-patient-neighborhoods"> <span class="header-section-number">4.4</span> Analysing patient neighborhoods</a>
  <ul class="collapse">
  <li><a href="#analysing-all-patients" id="toc-analysing-all-patients" class="nav-link" data-scroll-target="#analysing-all-patients"> <span class="header-section-number">4.4.1</span> Analysing all patients</a></li>
  </ul></li>
  <li><a href="#predicting-response-to-endocrine-therapy" id="toc-predicting-response-to-endocrine-therapy" class="nav-link" data-scroll-target="#predicting-response-to-endocrine-therapy"> <span class="header-section-number">4.5</span> Predicting response to endocrine therapy</a></li>
  </ul>
<div class="toc-actions"><div><i class="bi bi-github"></i></div><div class="action-links"><p><a href="https://github.com/chronchi/molecular_landscape/issues/new" class="toc-action">Report an issue</a></p></div></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title d-none d-lg-block"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Pathway activity in a personalized context</span></h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
</div>



<div class="quarto-title-meta">

    
    
  </div>
  

</header>

<p>This chapter we present a new way to think about the personalized medicine, by incorporating patient molecular information and its context. We saw previously that by using different cohorts, we can embed new patients into a point cloud. The idea is that the neighborhood of the patients are similar in the molecular level, and we can draw conclusions based on this.</p>
<p>Each patient can be assigned a score. Scores represent a biological pathway, or in other words, a biological activity. Given a biological process, there is usually a set of genes that represent it. We can then use these set of genes to calculate scores that are proxies of this biological activity.</p>
<p>For example, in the previous chapters we used the ER signaling scores to motivate the continuity of ER signaling. This is not restricted to this specific pathway, we could have used any other list of genes representing a process.</p>
<p>By combining a score and the neighborhood of a patient, we can make direct comparisons and questions. Given a patient, how different is its score for a pathway compared to its neighbors? Is there also an alternative pathway that could be target and has a higher signaling? In this case specifically, is ER signaling more or less expressed compared to its neighbors?</p>
<p>The POETIC trial <span class="citation" data-cites="Gao2019">(<a href="references.html#ref-Gao2019" role="doc-biblioref">Gao et al. 2019</a>)</span> is a breast cancer trial that had as hypothesis neoadjuvant therapy could improve overall and recurrence free survival. They gave aromatase inhibitors (AIs) for ER+ BC patients for 2 weeks prior to surgery and then 2 weeks after surgery. Moreover, they collected a needle biopsy before the treatment started and a biopsy in the surgery. They also measured the Ki67 levels, therefore we have a proxy on how well these patients responded in the short term to AIs. Also, they sent the tissues for sequencing, so microarray data is available. This is a unique cohort, in the sense that we can understand what are the responders or not, in terms of Ki67 levels, and use the molecular data to try to understand the responsiveness. The definition of responders are those patients that have a reduction of at least 60% in Ki67 levels when comparing pre-treatment versus surgery.</p>
<p>Here we use this cohort to draw conclusions on responders and non responders and how the molecular landscape can be used. We show that some patients that are close to each other in the molecular landscape can have different pathway scores and therefore have different responses.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>* The library is already synchronized with the lockfile.</code></pre>
</div>
</div>
<section id="calculating-scores-in-neighborhoods" class="level2" data-number="4.1">
<h2 data-number="4.1" class="anchored" data-anchor-id="calculating-scores-in-neighborhoods"><span class="header-section-number">4.1</span> Calculating scores in neighborhoods</h2>
<p>Patients close to each other in the molecular landscape are somewhat molecularly close, due to how the embedding works. Therefore, we can look at patients in a neighborhood and calculate the average scores, to see what it means to be in a specific neighborhood.</p>
<p>To do this, we define a radius for a patient where all the patients within the euclidean ball with this radius will be selected and an average score distribution is calculated.</p>
<p>To calculate the average score distribution we use the <code>rstanarm</code> package. We set the prior distribution for the intercept to be a normal distribution centered at 0 with standard deviation of 1. All scores are in a range of -1 to 1, so this is a relatively flat prior for our use case.</p>
<p>The video below shows how scores change in average when comparing different neighborhoods.</p>
<div class="cell">

</div>
<iframe width="720" height="480" src="../plots/scoring/mol_land.mp4" align="middle" frameborder="0" allowfullscreen="">
</iframe>
<p>Going from right to left it shows how ER signaling is decreased. From top to bottom there is a reduction of proliferation, E2F targets, and a change in EMT and TGFb signaling. These are representative scores, other pathways could be used as well.</p>
</section>
<section id="poetic-embedding" class="level2" data-number="4.2">
<h2 data-number="4.2" class="anchored" data-anchor-id="poetic-embedding"><span class="header-section-number">4.2</span> POETIC embedding</h2>
<p>In this section we show the embedding of the POETIC trial samples. But first we analyse some basic properties of the dataset. The data for this dataset was downloaded and processed as described at <a href="https://chronchi.github.io/transcriptomics/">chronchi.github.io/transcriptomics</a> in the chapter AI - GSE105777.</p>
<div class="cell">

</div>
<section id="number-of-samples" class="level3" data-number="4.2.1">
<h3 data-number="4.2.1" class="anchored" data-anchor-id="number-of-samples"><span class="header-section-number">4.2.1</span> Number of samples</h3>
<p>In this dataset there are matched samples from patients that are treated and untreated. <a href="#tbl-pt-poetic-nb">Table&nbsp;<span>4.1</span></a></p>
<div class="cell">
<div class="cell-output-display">
<div id="tbl-pt-poetic-nb" class="anchored">

<table class=" lightable-classic" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; margin-left: auto; margin-right: auto;"><caption>Table 4.1:  Table showing the number of samples for treated and untreated patients </caption>
 <thead>
  <tr>
   <th style="text-align:left;"> group </th>
   <th style="text-align:right;"> n </th>
   <th style="text-align:right;"> percent </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:left;"> treated </td>
   <td style="text-align:right;"> 157 </td>
   <td style="text-align:right;"> 74 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> untreated </td>
   <td style="text-align:right;"> 56 </td>
   <td style="text-align:right;"> 26 </td>
  </tr>
</tbody>
</table>

</div>
</div>
</div>
<p>In this cohort, there is the PAM50 molecular subtype available for the untreated. <a href="#tbl-pt-poetic-pam50">Table&nbsp;<span>4.2</span></a> shows the number of patients for each subtype.</p>
<div class="cell">
<div class="cell-output-display">
<div id="tbl-pt-poetic-pam50" class="anchored">

<table class=" lightable-classic" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; margin-left: auto; margin-right: auto;"><caption>Table 4.2:  Table showing the number of molecular subtypes for each group. </caption>
 <thead>
  <tr>
   <th style="text-align:left;"> pam50 </th>
   <th style="text-align:right;"> n </th>
   <th style="text-align:right;"> percent </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:left;"> her2 </td>
   <td style="text-align:right;"> 1 </td>
   <td style="text-align:right;"> 0.235 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> luma </td>
   <td style="text-align:right;"> 77 </td>
   <td style="text-align:right;"> 18.075 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> lumb </td>
   <td style="text-align:right;"> 31 </td>
   <td style="text-align:right;"> 7.277 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> normal </td>
   <td style="text-align:right;"> 3 </td>
   <td style="text-align:right;"> 0.704 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> not_available </td>
   <td style="text-align:right;"> 314 </td>
   <td style="text-align:right;"> 73.709 </td>
  </tr>
</tbody>
</table>

</div>
</div>
</div>
<p>The numbers differ from previously, since the molecular subtype is calculated for each sample, so two molecular subtypes for each patient.</p>
</section>
<section id="proportion-of-top-loadings" class="level3" data-number="4.2.2">
<h3 data-number="4.2.2" class="anchored" data-anchor-id="proportion-of-top-loadings"><span class="header-section-number">4.2.2</span> Proportion of top loadings</h3>
<div class="cell">

</div>
<p>There are in total 1008 genes available out of the . Among them, 39 are the housekeeping genes. Out of the 36 genes missing, a total number of 8 are missing from the top 200 PC3 loadings, a proportion of 4%. A total number of 5 are missing from the top 200 PC4 loadings, a proportion of 2%. These are very small proportions, and we have seen from the last chapter that they will not affect the position of the patients in the embedding.</p>
</section>
<section id="embedding" class="level3" data-number="4.2.3">
<h3 data-number="4.2.3" class="anchored" data-anchor-id="embedding"><span class="header-section-number">4.2.3</span> Embedding</h3>
<p><a href="#fig-pca-poetic-er-pam50">Figure&nbsp;<span>4.1</span></a> shows a good mixing of the POETIC sample and how all the patients are scattered across the whole landscape. Here all samples are plotted, meaning that every two dots correspond to a single patient.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pca-poetic-er-pam50" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="scoring_files/figure-html/fig-pca-poetic-er-pam50-1.png" class="img-fluid figure-img" width="1824"></p>
<p></p><figcaption class="figure-caption">Figure 4.1: PCA embedding of all samples from TCGA, SCANB and METABRIC including the POETIC samples on top. (A) Colored by cohort, (B) colored by ER status, (C) colored by PAM50 molecular subtype.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p><a href="#fig-pca-responders-baseline">Figure&nbsp;<span>4.2</span></a> shows the embedding of baseline samples from patients that received endocrine therapy prior to surgery. Patients on the left part of the molecular landscape are considered to be non responders, this coincides with the fact this region corresponds to the ER- BC patients.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pca-responders-baseline" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="scoring_files/figure-html/fig-pca-responders-baseline-1.png" class="img-fluid figure-img" width="960"></p>
<p></p><figcaption class="figure-caption">Figure 4.2: PCA embedding of all samples from TCGA, SCANB and METABRIC including the POETIC samples on top. POETIC samples correspond to only baseline treated patients.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>And when checking the first two components, the POETIC data is closer to METABRIC, which makes sense since both are microarrays, as it can be seen in <a href="#fig-poetic-cohort">Figure&nbsp;<span>4.3</span></a>.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-poetic-cohort" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="scoring_files/figure-html/fig-poetic-cohort-1.png" class="img-fluid figure-img" width="672"></p>
<p></p><figcaption class="figure-caption">Figure 4.3: Biplot of first two PCA components from POETIC, TCGA, SCANB and METABRIC. POETIC is highlighted in the plot and has a bigger point.</figcaption><p></p>
</figure>
</div>
</div>
</div>
</section>
<section id="differences-in-pc3-for-responders-and-non-responders" class="level3" data-number="4.2.4">
<h3 data-number="4.2.4" class="anchored" data-anchor-id="differences-in-pc3-for-responders-and-non-responders"><span class="header-section-number">4.2.4</span> Differences in PC3 for responders and non responders</h3>
<p>We now also compare the differences in non responders vs responders according to the third component.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="scoring_files/figure-html/unnamed-chunk-10-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
</section>
<section id="scoring-all-samples" class="level2" data-number="4.3">
<h2 data-number="4.3" class="anchored" data-anchor-id="scoring-all-samples"><span class="header-section-number">4.3</span> Scoring all samples</h2>
<p>In the last chapters scores were showed for TCGA, SCANB and METABRIC. <a href="pca_merging.html#fig-pca-seterpr">Figure&nbsp;<span>2.18</span></a> shows how ER signaling changes as patients move across the molecular landscape. We intend to use the scores now for the POETIC samples. We first start by calculating using GSVA as well and include in the dataframe with the other cohorts. Before moving to the next chapter, we will analyse and compare the scores across the responders and non responders.</p>
<p><a href="#fig-scores-poetic-er">Figure&nbsp;<span>4.4</span></a> shows that when using the scores, it looks like in average at baseline <span class="math inline">\(SET_{ER/PR}\)</span> is lower in non responders than in responders. For the other scores there is no clear difference.</p>
<div class="cell">

</div>
<div class="cell">

</div>
<div class="cell">
<div class="cell-output-display">
<div id="fig-scores-poetic-er" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="scoring_files/figure-html/fig-scores-poetic-er-1.png" class="img-fluid figure-img" width="1248"></p>
<p></p><figcaption class="figure-caption">Figure 4.4: Baseline scores for all treated patients in the POETIC trial. G2M corresponds to a proliferation score, EMT to epithelial to mesenchymal transition score and the others are related to ER signaling.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>We see that for PC3 and SET ER/PR There are some differences in terms of the responders and non responders.</p>
<p>Let us take a closer look at those patients. We start by plotting the scatter plot of PC3 vs SET ER/PR and color by response status defined by the POETIC trial.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-scores-poetic-responders" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="scoring_files/figure-html/fig-scores-poetic-responders-1.png" class="img-fluid figure-img" width="672"></p>
<p></p><figcaption class="figure-caption">Figure 4.5: Comparison of the third component with SET ER/PR scores colored by response status.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>This figure shows that there is a correlation between third component and SET ER/PR for the POETIC trial data as well as it was already seen from the other cohorts. Moreover, it seems that the responders have higher PC scores in general. We now correlate for each subgroup (responder and non responders) the PC3 and the change in percentage for Ki67.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-scores-corr-poetic" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="scoring_files/figure-html/fig-scores-corr-poetic-1.png" class="img-fluid figure-img" width="960"></p>
<p></p><figcaption class="figure-caption">Figure 4.6: Correlation between change in Ki67 versus PC3 stratified by response group.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>Now another metric we can calculate from our dataset is the difference of the embedding positions between the different samples of the same patient. The idea is that if a patient responded well, there were bigger changes as well in the transcriptomics and that is reflected in the embedding.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-diff-embedding-poetic" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="scoring_files/figure-html/fig-diff-embedding-poetic-1.png" class="img-fluid figure-img" width="960"></p>
<p></p><figcaption class="figure-caption">Figure 4.7: Differences of the embeddings for each patient individually stratified by response status. Each dot corresponds to a single patient.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>And now we perform a comparison of these positions by using rstanarm to get the posterior distributions of the differences.</p>
<div class="cell">

</div>
<p>The figure below shows the distribution of the averages of the differences in the embedding of the projections.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-diff-embedding-poetic-averages" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="scoring_files/figure-html/fig-diff-embedding-poetic-averages-1.png" class="img-fluid figure-img" width="672"></p>
<p></p><figcaption class="figure-caption">Figure 4.8: Differences of the embeddings for each patient individually stratified by response status. Each dot corresponds to a single patient.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>And the figure below shows the posterior distribution of the differences in average.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-diff-embedding-poetic-averages-diff" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="scoring_files/figure-html/fig-diff-embedding-poetic-averages-diff-1.png" class="img-fluid figure-img" width="672"></p>
<p></p><figcaption class="figure-caption">Figure 4.9: Differences of the average differences in the embedding for each patient individually. The comparison made was non responder vs responder, which means that if non responder has a lower difference in average, the difference in difference will be negative.</figcaption><p></p>
</figure>
</div>
</div>
</div>
</section>
<section id="analysing-patient-neighborhoods" class="level2" data-number="4.4">
<h2 data-number="4.4" class="anchored" data-anchor-id="analysing-patient-neighborhoods"><span class="header-section-number">4.4</span> Analysing patient neighborhoods</h2>
<p>There are two ways to interpret the embedding. If the patient has ER+ BC and its embedding is close to the ER- BC patients, we can infer that the endocrine response might not work so well, as it was shown in <a href="#fig-pca-responders-baseline">Figure&nbsp;<span>4.2</span></a>. The other option is to compare the patient’s score to the average score of its neighborhood.</p>
<p>What is a neighborhood? When performing the embedding, global structures are not preserved usually. In this sense we only compare patients that are close to each other in an euclidean neighborhood, i.e., we calculate a radius around each sample and see what other samples are within this ball given the euclidean distance.</p>
<p>Given a neighborhood, one can calculate the posterior average score of all samples from METABRIC, SCANB and TCGA and use that as an indication of average score that we can compare other patients to. The concept is that if an ER+ BC patient has a much lower ER signaling score, it means the patients will not respond so well to endocrine therapy, as they are very different from their neighboors.</p>
<p>To showcase the ability to use the scores and infer results, we use the POETIC trial patients to compare the scores. This cohort is special in the sense that the patients are filtered based on selection criterias, meaning that the patients are clinically similar.</p>
<p>Two random patients in very close neighborhoods were selected to showcase the ability of the molecular landscape. The id of the patients are 63 and 236. <a href="#tbl-pt63-236">Table&nbsp;<span>4.3</span></a> shows the clinical features of these patients. One of them is HER2+. Patient 63 had a higher Ki67 baseline level but a very good response to endocrine therapy. Patient 236 did not respond to endocrine therapy in terms of reduction of Ki67 levels.</p>
<div class="cell">
<div class="cell-output-display">
<div id="tbl-pt63-236" class="anchored">

<table class=" lightable-classic" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; margin-left: auto; margin-right: auto;"><caption>Table 4.3:  Clinical features from patients 63 and 236. </caption>
 <thead>
  <tr>
   <th style="text-align:left;"> patient_nb </th>
   <th style="text-align:left;"> er_status </th>
   <th style="text-align:left;"> her2_status </th>
   <th style="text-align:right;"> ki67 </th>
   <th style="text-align:right;"> change_ki67 </th>
   <th style="text-align:left;"> ccca_surgery_ki67_2_7 </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:left;"> 63 </td>
   <td style="text-align:left;"> pos </td>
   <td style="text-align:left;"> Positive </td>
   <td style="text-align:right;"> 16.60702 </td>
   <td style="text-align:right;"> -69.958509 </td>
   <td style="text-align:left;"> noCCCA </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 236 </td>
   <td style="text-align:left;"> pos </td>
   <td style="text-align:left;"> Negative </td>
   <td style="text-align:right;"> 11.30906 </td>
   <td style="text-align:right;"> 5.593711 </td>
   <td style="text-align:left;"> noCCCA </td>
  </tr>
</tbody>
</table>

</div>
</div>
</div>
<p><a href="#fig-pt63-236">Figure&nbsp;<span>4.10</span></a> shows the embedding of two distinct patients in a similar neighborhood.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pt63-236" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="scoring_files/figure-html/fig-pt63-236-1.png" class="img-fluid figure-img" width="960"></p>
<p></p><figcaption class="figure-caption">Figure 4.10: Plot of two selected patients in the molecular landscape. One is a responder and the other is a non responder. The patient id numbers are 63 and 236.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>And now we calculate the average scores for each patient neighborhoods. We start by defining a radius value. After that we select the patients in the neighborhood and use <code>rstanarm</code> to get the posterior distribution of the average score in the neighborhood. For this we use the function <code>stan_glm</code> to calculate the average. This is effective because it can be paired with the package <code>tidybayes</code> for plotting. It makes extremely easy to deal with posterior data. For the average we use a normal prior centered at 0 with 1 standard deviation, since all scores are between -1 and 1.</p>
<div class="cell">

</div>
<div class="cell">

</div>
<p>Patient 63 <a href="trying.html#fig-pt63">Figure&nbsp;<span>6.8</span></a> is considered a responder. According to the scores, when comparing the estrogen early signature to its average distribution in the neighborhood, the score is higher.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pt63" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="scoring_files/figure-html/fig-pt63-1.png" class="img-fluid figure-img" width="960"></p>
<p></p><figcaption class="figure-caption">Figure 4.11: Posterior distribution of the average scores in the neighborhood of patient 236 from POETIC trial. Each dot corresponds to a 1% quantile.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>Patient 236 <a href="trying.html#fig-pt236">Figure&nbsp;<span>6.9</span></a> was considered a non responder. According to the scores, when comparing the estrogen early signature to its average distribution in the neighborhood, the score is lower, being in the 5% quantile.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pt236" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="scoring_files/figure-html/fig-pt236-1.png" class="img-fluid figure-img" width="960"></p>
<p></p><figcaption class="figure-caption">Figure 4.12: Posterior distribution of the average scores in the neighborhood of patient 236 from POETIC trial. Each dot corresponds to a 1% quantile.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>When comparing these two patients, there is also a difference in the androgen response score, which could be a reflection of different estrogen signaling.</p>
<section id="analysing-all-patients" class="level3" data-number="4.4.1">
<h3 data-number="4.4.1" class="anchored" data-anchor-id="analysing-all-patients"><span class="header-section-number">4.4.1</span> Analysing all patients</h3>
<p>We now try to extend this comparison to something more systematic. Instead of looking just for these two patients we will compare the neighborhoods for the top responders and non-responder patients. The first thing we have to have in mind is that some non responders they did respond to the drug, just the reduction in Ki67 was not so big. For example patient 209 has a reduction of 40% in Ki67 going from 16% to 9%. Still this patient is considered a non-responder.</p>
<div class="cell">

</div>
<p>We don’t select all patients for the comparisons. We select only patients that are at least in the HER2 enriched region to the right, so we exclude the basal like patients. The reason for this as well is because they all either have low Estrogen response early scores or low SET ER/PR.</p>
<div class="cell">
<div class="cell-output-display">

<table class=" lightable-classic" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
 <thead>
  <tr>
   <th style="text-align:left;">   </th>
   <th style="text-align:left;"> HALLMARK_ESTROGEN_RESPONSE_EARLY </th>
   <th style="text-align:left;"> SET_ERPR </th>
   <th style="text-align:left;"> PC3 </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:left;"> treated_nb69_baseline </td>
   <td style="text-align:left;"> -0.265 </td>
   <td style="text-align:left;"> -0.50 </td>
   <td style="text-align:left;"> -4.5 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> treated_nb76_baseline </td>
   <td style="text-align:left;"> -0.275 </td>
   <td style="text-align:left;"> -0.19 </td>
   <td style="text-align:left;"> -4.1 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> treated_nb80_baseline </td>
   <td style="text-align:left;"> -0.013 </td>
   <td style="text-align:left;"> -0.49 </td>
   <td style="text-align:left;"> -5.6 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> treated_nb229_baseline </td>
   <td style="text-align:left;"> -0.234 </td>
   <td style="text-align:left;"> -0.54 </td>
   <td style="text-align:left;"> -4.7 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> treated_nb230_baseline </td>
   <td style="text-align:left;"> -0.308 </td>
   <td style="text-align:left;"> -0.50 </td>
   <td style="text-align:left;"> -7.5 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> treated_nb237_baseline </td>
   <td style="text-align:left;"> -0.377 </td>
   <td style="text-align:left;"> -0.29 </td>
   <td style="text-align:left;"> -7.9 </td>
  </tr>
</tbody>
</table>

</div>
</div>
<div class="cell">

</div>
<section id="non-responders" class="level4" data-number="4.4.1.1">
<h4 data-number="4.4.1.1" class="anchored" data-anchor-id="non-responders"><span class="header-section-number">4.4.1.1</span> Non-responders</h4>
<div class="cell">
<div class="cell-output-display">
<p><img src="scoring_files/figure-html/unnamed-chunk-29-1.png" class="img-fluid" width="960"></p>
</div>
</div>
<p>Estrogen score still close to 0, high proliferation at baseline and higher androgen score. But in this case SET ER/PR is lower than average. There is a mismatch.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="scoring_files/figure-html/unnamed-chunk-30-1.png" class="img-fluid" width="960"></p>
</div>
</div>
<p>This patient has a score close to 0 still but low proliferation in general. This is a case where the EMT signaling is really at average and TGFb is super low. Again SET ER/PR is much lower, mismatch with ER signaling.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="scoring_files/figure-html/unnamed-chunk-31-1.png" class="img-fluid" width="960"></p>
</div>
</div>
<p>Estrogen signaling close to 0, high androgen and E2F targets and high EMT. Intriguing patient, both ER and SET ER/PR are going in the same direction and above average.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="scoring_files/figure-html/unnamed-chunk-32-1.png" class="img-fluid" width="960"></p>
</div>
</div>
<p>Low ER signaling and super high EMT with an increase P53 pathway expression. Super low ER signaling here in terms of SET ER/PR.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="scoring_files/figure-html/unnamed-chunk-33-1.png" class="img-fluid" width="960"></p>
</div>
</div>
<p>Low ER score, even lower than average and low EMT signaling. Average E2F targets and lower than average for SET ER/PR.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="scoring_files/figure-html/unnamed-chunk-34-1.png" class="img-fluid" width="960"></p>
</div>
</div>
<p>This case is a bit different, the ER signaling is higher, E2F targets at baseline is not so high, even lower than the average, but in this case androgen response is much higher and EMT is extremely high. There is a relative mismatch between ER early and SET ER/PR. SET ER/PR is relatively small and close to the average.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="scoring_files/figure-html/unnamed-chunk-35-1.png" class="img-fluid" width="960"></p>
</div>
</div>
<p>Androgen score lower than the average and high E2F targets at baseline. In this case we see a very small EMT signature. In this case there is again a mismatch between ER signaling and SET ER/PR.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="scoring_files/figure-html/unnamed-chunk-36-1.png" class="img-fluid" width="960"></p>
</div>
</div>
<p>This one has a low ER signaling as well and higher E2F targets score. Massive difference in SET ER/PR here when comparing to the average.</p>
<p>In general all these non responders with no change in Ki67 had a low ER signaling score or a mismatch between ER early and SET ER/PR</p>
</section>
<section id="top-responders" class="level4" data-number="4.4.1.2">
<h4 data-number="4.4.1.2" class="anchored" data-anchor-id="top-responders"><span class="header-section-number">4.4.1.2</span> Top responders</h4>
<p>We now investigate the top responders with the biggest decrease in Ki67.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="scoring_files/figure-html/unnamed-chunk-37-1.png" class="img-fluid" width="960"></p>
</div>
</div>
<p>First thing is that this patient had already a positive score and a very low Androgen signaling score. Moreover the Ki67 is low. This is one of the furthest patients to the right and still has a low ER signaling score.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="scoring_files/figure-html/unnamed-chunk-38-1.png" class="img-fluid" width="960"></p>
</div>
</div>
<p>A bit high E2F targets but higher than average of the estrogen signaling score. High SET ER/PR.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="scoring_files/figure-html/unnamed-chunk-39-1.png" class="img-fluid" width="960"></p>
</div>
</div>
<p>Here the ER signaling is above average and androgen response is similar to the average. E2F targets is high. Also super high SET ER/PR.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="scoring_files/figure-html/unnamed-chunk-40-1.png" class="img-fluid" width="960"></p>
</div>
</div>
<p>Very low E2F targets and low Androgen response. ER is about the average. SET ER/PR is still positive. This sample is in the luminal A region also.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="scoring_files/figure-html/unnamed-chunk-41-1.png" class="img-fluid" width="960"></p>
</div>
</div>
<p>In this case ER signaling is also above average and E2F targets is similar to the average of neighbors. SET ER/PR on average and in the expected direction.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="scoring_files/figure-html/unnamed-chunk-42-1.png" class="img-fluid" width="960"></p>
</div>
</div>
<p>This one is intriguing, ER signaling is extremely low, E2F targets is very low as well but the patient responded pretty well. But there is a big mismatch between Ki67 and transcriptomics for this patient. Further investigation should be performed. Actually probably the samples have mixed names. Below are the values of the pathways after treatment:</p>
<div class="cell">
<div class="cell-output-display">

<div id="htmlwidget-239d30b10b5ec6490a0f" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-239d30b10b5ec6490a0f">{"x":{"filter":"none","vertical":false,"data":[["treated_nb89_baseline","treated_nb89_surgery"],[49.3116395494368,1.86104218362283],[-0.556062047578453,0.383861479152084],[0.0570299959923077,0.279317135063836],[-0.324343902994388,-0.174999482790282],[0.737080017429381,0.0409566810926228],[-2.03855921324292,1.88122496727946]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>ki67<\/th>\n      <th>HALLMARK_E2F_TARGETS<\/th>\n      <th>SET_ERPR<\/th>\n      <th>HALLMARK_ESTROGEN_RESPONSE_EARLY<\/th>\n      <th>PC3<\/th>\n      <th>PC4<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"columnDefs":[{"className":"dt-right","targets":[1,2,3,4,5,6]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<p>It looks super weird that there is a big increase in proliferation based on the transcriptomic data considering the reduction in the Ki67 percentage. Not only that, ER signaling increases. It is very likely that the sample has a mixed label.</p>
<p>Moving to the next one.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="scoring_files/figure-html/unnamed-chunk-44-1.png" class="img-fluid" width="960"></p>
</div>
</div>
<p>This one also is intriguing, estrogen signaling is very low but the patient is a responder with a good decrease in Ki67. Below is a table with the values for surgery and baseline. We see that there is indeed a decrease in proliferation and also in estrogen signaling. Also this sample is closer to the normal like and luminal A region.</p>
<div class="cell">
<div class="cell-output-display">

<div id="htmlwidget-fc52d003806914f69bec" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-fc52d003806914f69bec">{"x":{"filter":"none","vertical":false,"data":[["treated_nb90_baseline","treated_nb90_surgery"],[0.4062863196128,2.49153864759479],[-3.02836676173437,-2.06070153823311],[-0.324759889064836,-0.218895973428806],[0.0642536744292175,-0.378936381935116],[29.3718166383701,1.11111111111111]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>PC3<\/th>\n      <th>PC4<\/th>\n      <th>HALLMARK_ESTROGEN_RESPONSE_EARLY<\/th>\n      <th>HALLMARK_E2F_TARGETS<\/th>\n      <th>ki67<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"columnDefs":[{"className":"dt-right","targets":[1,2,3,4,5]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<p>And last patient.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="scoring_files/figure-html/unnamed-chunk-46-1.png" class="img-fluid" width="960"></p>
</div>
</div>
<p>This patient has a score slightly higher than the average, but still negative. Nonetheless, the patient responded well to the therapy. One note is that this patient is close to the HER2 enriched region.</p>
<p>In general non responders had more mismatches between SET ER/PR and below average values than responders.</p>
</section>
<section id="responders-in-the-middle-of-embedding" class="level4" data-number="4.4.1.3">
<h4 data-number="4.4.1.3" class="anchored" data-anchor-id="responders-in-the-middle-of-embedding"><span class="header-section-number">4.4.1.3</span> Responders in the middle of embedding</h4>
<p>We lastly check the responders that are in the middle of the molecular landscape to see how they are.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="scoring_files/figure-html/unnamed-chunk-47-1.png" class="img-fluid" width="960"></p>
</div>
</div>
<p>Low ER signaling in both cases. This patient specifically had very high Ki67 levels at baseline.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="scoring_files/figure-html/unnamed-chunk-48-1.png" class="img-fluid" width="960"></p>
</div>
</div>
<p>Also ER signaling is low here compared to the average, but a very low Ki67 proliferation index at baseline.</p>
</section>
<section id="conclusion" class="level4" data-number="4.4.1.4">
<h4 data-number="4.4.1.4" class="anchored" data-anchor-id="conclusion"><span class="header-section-number">4.4.1.4</span> Conclusion</h4>
<p>In general what we can see is that when comparing responders and non responders the main difference is in the position of the molecular landscape. If we select the top responders and the top non-responders in terms of decrease of Ki67 index, we see that non responders have way lower third principal component (<a href="#fig-high-respnon">Figure&nbsp;<span>4.13</span></a>).</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-high-respnon" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="scoring_files/figure-html/fig-high-respnon-1.png" class="img-fluid figure-img" width="576"></p>
<p></p><figcaption class="figure-caption">Figure 4.13: PC3 comparison of the top non responders and responders</figcaption><p></p>
</figure>
</div>
</div>
</div>
</section>
</section>
</section>
<section id="predicting-response-to-endocrine-therapy" class="level2" data-number="4.5">
<h2 data-number="4.5" class="anchored" data-anchor-id="predicting-response-to-endocrine-therapy"><span class="header-section-number">4.5</span> Predicting response to endocrine therapy</h2>
<p>We now try to use a penalized logistic regression to predict patient response, as defined by the POETIC trialists using the principal components. For this we already saw that in average the patients further to the left are non-responders and further to the right would be responders, suggesting a predictive power of the molecular landscape. The landscape still is not perfect as there is a lot of overlap between the responders and non-responders. So we try to use more principal components to capture the differences.</p>
<p>Just when randomizing the train test split, we need to make sure that we include some of the samples that are in the basal like region in both splits, since they are in small numbers. Due to the small sample size, we use the components starting from 3 and going up to 20.</p>
<div class="cell">

</div>
<p>In this dataset the total of responders vs non responder is:</p>
<div class="cell">
<div class="cell-output-display">

<div id="htmlwidget-3b67eb253c36ba810e33" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-3b67eb253c36ba810e33">{"x":{"filter":"none","vertical":false,"data":[["1","2"],["non_responder","responder"],[43,94],[0.313868613138686,0.686131386861314]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>is_responder<\/th>\n      <th>n<\/th>\n      <th>percent<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"targets":3,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatPercentage(data, 0, 3, \",\", \".\", null);\n  }"},{"className":"dt-right","targets":[2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":["options.columnDefs.0.render"],"jsHooks":[]}</script>
</div>
</div>
<p>More than 2/3 are responders. Using a set seed for the randomization, the numbers of responders and non responders are shown below for the training set.</p>
<div class="cell">
<div class="cell-output-display">

<div id="htmlwidget-72e9f85625d3553ae9f3" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-72e9f85625d3553ae9f3">{"x":{"filter":"none","vertical":false,"data":[["1","2"],["non_responder","responder"],[28,63],[0.307692307692308,0.692307692307692]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>is_responder<\/th>\n      <th>n<\/th>\n      <th>percent<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"targets":3,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatPercentage(data, 0, 3, \",\", \".\", null);\n  }"},{"className":"dt-right","targets":[2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":["options.columnDefs.0.render"],"jsHooks":[]}</script>
</div>
</div>
<p>And for the test set.</p>
<div class="cell">
<div class="cell-output-display">

<div id="htmlwidget-19fbc2a323475bbd0992" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-19fbc2a323475bbd0992">{"x":{"filter":"none","vertical":false,"data":[["1","2"],["non_responder","responder"],[15,31],[0.326086956521739,0.673913043478261]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>is_responder<\/th>\n      <th>n<\/th>\n      <th>percent<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"targets":3,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatPercentage(data, 0, 3, \",\", \".\", null);\n  }"},{"className":"dt-right","targets":[2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":["options.columnDefs.0.render"],"jsHooks":[]}</script>
</div>
</div>
<p>And we compare the PC3 values distributions to see if they are equally distributed.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="scoring_files/figure-html/unnamed-chunk-54-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>They are similarly distributed, so we can proceed with the training and testing strategy.</p>
<div class="cell">

</div>
<p>The plot below shows the misclassification error versus the lambda used for the penalization term in the logistic regression.</p>
<p>The plot below shows the misclassification error for different <span class="math inline">\(\lambda\)</span> values and number of components.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="scoring_files/figure-html/unnamed-chunk-56-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Overall the misclassification is pretty similar for all number of variables. We select the variables obtained with the lambda that is 1 SE away from the minimum. So in total we have three variables: The intercept and PC3, PC10, PC20.</p>
<p>And below we show the confusion matrix.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>         True
Predicted  0  1 Total
    0      1  3     4
    1     14 28    42
    Total 15 31    46

 Percent Correct:  0.6304 </code></pre>
</div>
</div>
<p>We see that the accuracy is very low and most of the patients are classified as responders. This could be potentially due to the class imbalance, but the results are still underwhelming.</p>
<p>Let us take a look on the patients that were predicted to be non responders and check their Ki67 differences in baseline and surgery.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>                             PC3 change_ki67     ki67
treated_nb64_baseline  -2.733038   -66.53449 44.46653
treated_nb80_baseline  -5.649423    26.83238 43.24122
treated_nb129_baseline -3.468263   -86.59865 21.55340
treated_nb147_baseline -3.421153   -81.93744 42.51592</code></pre>
</div>
</div>
<p>All the patients that were misclassified as non responders they have a very low third principal component and their baseline Ki67 is quite high, so the decrease in proliferation is not actually big enough. It is still considered a responder.</p>


<!-- -->

<div id="refs" class="references csl-bib-body hanging-indent" role="doc-bibliography" style="display: none">
<div id="ref-Gao2019" class="csl-entry" role="doc-biblioentry">
Gao, Qiong, and Elena López-Knowles, Maggie Chon U. Cheang, James Morden, Ricardo Ribas, Kally Sidhu, David Evans, et al. 2019. <span>“Impact of Aromatase Inhibitor Treatment on Global Gene Expression and Its Association with Antiproliferative Response in <span>ER</span><span class="math inline">\(\mathplus\)</span> Breast Cancer in Postmenopausal Patients.”</span> <em>Breast Cancer Research</em> 22 (1). <a href="https://doi.org/10.1186/s13058-019-1223-z">https://doi.org/10.1186/s13058-019-1223-z</a>.
</div>
</div>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      let href = ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./validation.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Validating the molecular landscape</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./risk_score.html" class="pagination-link">
        <span class="nav-page-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Risk score model</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb4" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu"># Pathway activity in a personalized context</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>This chapter we present a new way to think about the personalized medicine,</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>by incorporating patient molecular information and its context. </span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>We saw previously that by using different cohorts, we can embed new patients</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>into a point cloud. The idea is that the neighborhood of the patients </span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>are similar in the molecular level, and we can draw conclusions based on</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>this. </span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>Each patient can be assigned a score. Scores represent a biological</span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>pathway, or in other words, a biological activity. Given a biological</span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>process, there is usually a set of genes that represent it. We</span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a>can then use these set of genes to calculate scores that are </span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a>proxies of this biological activity.</span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a>For example, in the previous chapters we used the ER signaling</span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a>scores to motivate the continuity of ER signaling. This is not</span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a>restricted to this specific pathway, we could have used any</span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a>other list of genes representing a process. </span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a>By combining a score and the neighborhood of a patient, we </span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a>can make direct comparisons and questions. Given a patient,</span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a>how different is its score for a pathway compared to its</span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a>neighbors? Is there also an alternative pathway that could</span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a>be target and has a higher signaling? In this case </span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true" tabindex="-1"></a>specifically, is ER signaling more or less expressed compared</span>
<span id="cb4-27"><a href="#cb4-27" aria-hidden="true" tabindex="-1"></a>to its neighbors? </span>
<span id="cb4-28"><a href="#cb4-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-29"><a href="#cb4-29" aria-hidden="true" tabindex="-1"></a>The POETIC trial <span class="co">[</span><span class="ot">@Gao2019</span><span class="co">]</span> </span>
<span id="cb4-30"><a href="#cb4-30" aria-hidden="true" tabindex="-1"></a>is a breast cancer trial that had as hypothesis</span>
<span id="cb4-31"><a href="#cb4-31" aria-hidden="true" tabindex="-1"></a>neoadjuvant therapy could improve overall and recurrence free</span>
<span id="cb4-32"><a href="#cb4-32" aria-hidden="true" tabindex="-1"></a>survival. They gave aromatase inhibitors (AIs) for ER+ BC patients</span>
<span id="cb4-33"><a href="#cb4-33" aria-hidden="true" tabindex="-1"></a>for 2 weeks prior to surgery and then 2 weeks after surgery.</span>
<span id="cb4-34"><a href="#cb4-34" aria-hidden="true" tabindex="-1"></a>Moreover, they collected a needle biopsy before the</span>
<span id="cb4-35"><a href="#cb4-35" aria-hidden="true" tabindex="-1"></a>treatment started and a biopsy in the surgery. They</span>
<span id="cb4-36"><a href="#cb4-36" aria-hidden="true" tabindex="-1"></a>also measured the Ki67 levels, therefore we have a proxy</span>
<span id="cb4-37"><a href="#cb4-37" aria-hidden="true" tabindex="-1"></a>on how well these patients responded in the short term to </span>
<span id="cb4-38"><a href="#cb4-38" aria-hidden="true" tabindex="-1"></a>AIs. Also, they sent the tissues for sequencing, so </span>
<span id="cb4-39"><a href="#cb4-39" aria-hidden="true" tabindex="-1"></a>microarray data is available. This is a unique </span>
<span id="cb4-40"><a href="#cb4-40" aria-hidden="true" tabindex="-1"></a>cohort, in the sense that we can understand what are the</span>
<span id="cb4-41"><a href="#cb4-41" aria-hidden="true" tabindex="-1"></a>responders or not, in terms of Ki67 levels, and use</span>
<span id="cb4-42"><a href="#cb4-42" aria-hidden="true" tabindex="-1"></a>the molecular data to try to understand the responsiveness. </span>
<span id="cb4-43"><a href="#cb4-43" aria-hidden="true" tabindex="-1"></a>The definition of responders are those patients that have a </span>
<span id="cb4-44"><a href="#cb4-44" aria-hidden="true" tabindex="-1"></a>reduction of at least 60\%</span>
<span id="cb4-45"><a href="#cb4-45" aria-hidden="true" tabindex="-1"></a>in Ki67 levels when comparing pre-treatment versus surgery. </span>
<span id="cb4-46"><a href="#cb4-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-47"><a href="#cb4-47" aria-hidden="true" tabindex="-1"></a>Here we use this cohort to draw conclusions on responders</span>
<span id="cb4-48"><a href="#cb4-48" aria-hidden="true" tabindex="-1"></a>and non responders and how the molecular landscape can be </span>
<span id="cb4-49"><a href="#cb4-49" aria-hidden="true" tabindex="-1"></a>used. We show that some patients that are close to each</span>
<span id="cb4-50"><a href="#cb4-50" aria-hidden="true" tabindex="-1"></a>other in the molecular landscape can have different</span>
<span id="cb4-51"><a href="#cb4-51" aria-hidden="true" tabindex="-1"></a>pathway scores and therefore have different responses.</span>
<span id="cb4-52"><a href="#cb4-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-53"><a href="#cb4-53" aria-hidden="true" tabindex="-1"></a><span class="in">```{r setup}</span></span>
<span id="cb4-54"><a href="#cb4-54" aria-hidden="true" tabindex="-1"></a>renv<span class="sc">::</span><span class="fu">restore</span>()</span>
<span id="cb4-55"><a href="#cb4-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-56"><a href="#cb4-56" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb4-57"><a href="#cb4-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-58"><a href="#cb4-58" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb4-59"><a href="#cb4-59" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(PCAtools)</span>
<span id="cb4-60"><a href="#cb4-60" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidybayes)</span>
<span id="cb4-61"><a href="#cb4-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-62"><a href="#cb4-62" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GSVA)</span>
<span id="cb4-63"><a href="#cb4-63" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SummarizedExperiment)</span>
<span id="cb4-64"><a href="#cb4-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-65"><a href="#cb4-65" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(rstanarm)</span>
<span id="cb4-66"><a href="#cb4-66" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(glmnet)</span>
<span id="cb4-67"><a href="#cb4-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-68"><a href="#cb4-68" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"utils.R"</span>)</span>
<span id="cb4-69"><a href="#cb4-69" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"first_run.R"</span>)</span>
<span id="cb4-70"><a href="#cb4-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-71"><a href="#cb4-71" aria-hidden="true" tabindex="-1"></a><span class="co"># Here we are using rds files from previous chapters, so we source</span></span>
<span id="cb4-72"><a href="#cb4-72" aria-hidden="true" tabindex="-1"></a><span class="co"># in any case the load_rds_files.R and exclude all the associated files</span></span>
<span id="cb4-73"><a href="#cb4-73" aria-hidden="true" tabindex="-1"></a><span class="co"># that are generated in this current chapter</span></span>
<span id="cb4-74"><a href="#cb4-74" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (first_run){</span>
<span id="cb4-75"><a href="#cb4-75" aria-hidden="true" tabindex="-1"></a>    load_at_setup <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb4-76"><a href="#cb4-76" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb4-77"><a href="#cb4-77" aria-hidden="true" tabindex="-1"></a>    load_at_setup <span class="ot">&lt;-</span> <span class="cn">TRUE</span></span>
<span id="cb4-78"><a href="#cb4-78" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb4-79"><a href="#cb4-79" aria-hidden="true" tabindex="-1"></a>name_document <span class="ot">&lt;-</span> <span class="st">"scoring"</span></span>
<span id="cb4-80"><a href="#cb4-80" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"load_rds_files.R"</span>)</span>
<span id="cb4-81"><a href="#cb4-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-82"><a href="#cb4-82" aria-hidden="true" tabindex="-1"></a><span class="co"># by setting the dev to png and pdf, this saves the figures in a specific</span></span>
<span id="cb4-83"><a href="#cb4-83" aria-hidden="true" tabindex="-1"></a><span class="co"># folder in both formats. moreover, since png is coming first, it shows</span></span>
<span id="cb4-84"><a href="#cb4-84" aria-hidden="true" tabindex="-1"></a><span class="co"># this figure when rendering the html. What is nice about this is that it</span></span>
<span id="cb4-85"><a href="#cb4-85" aria-hidden="true" tabindex="-1"></a><span class="co"># inherits the properties from the chunk to save the figure, so no need</span></span>
<span id="cb4-86"><a href="#cb4-86" aria-hidden="true" tabindex="-1"></a><span class="co"># to use ggplot2::ggsave to save the plots. this also works </span></span>
<span id="cb4-87"><a href="#cb4-87" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">dev =</span> <span class="fu">c</span>(<span class="st">'png'</span>, <span class="st">'pdf'</span>, <span class="st">'svg'</span>))</span>
<span id="cb4-88"><a href="#cb4-88" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">bitmapType =</span> <span class="st">'cairo'</span>)</span>
<span id="cb4-89"><a href="#cb4-89" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-90"><a href="#cb4-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-91"><a href="#cb4-91" aria-hidden="true" tabindex="-1"></a><span class="fu">## Calculating scores in neighborhoods</span></span>
<span id="cb4-92"><a href="#cb4-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-93"><a href="#cb4-93" aria-hidden="true" tabindex="-1"></a>Patients close to each other in the molecular landscape are somewhat</span>
<span id="cb4-94"><a href="#cb4-94" aria-hidden="true" tabindex="-1"></a>molecularly close, due to how the embedding works. Therefore, we can </span>
<span id="cb4-95"><a href="#cb4-95" aria-hidden="true" tabindex="-1"></a>look at patients in a neighborhood and calculate the average scores,</span>
<span id="cb4-96"><a href="#cb4-96" aria-hidden="true" tabindex="-1"></a>to see what it means to be in a specific neighborhood. </span>
<span id="cb4-97"><a href="#cb4-97" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-98"><a href="#cb4-98" aria-hidden="true" tabindex="-1"></a>To do this, we define a radius for a patient where all the patients</span>
<span id="cb4-99"><a href="#cb4-99" aria-hidden="true" tabindex="-1"></a>within the euclidean ball with this radius will be selected and an </span>
<span id="cb4-100"><a href="#cb4-100" aria-hidden="true" tabindex="-1"></a>average score distribution is calculated.</span>
<span id="cb4-101"><a href="#cb4-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-102"><a href="#cb4-102" aria-hidden="true" tabindex="-1"></a>To calculate the average score distribution we use the <span class="in">`rstanarm`</span> package. </span>
<span id="cb4-103"><a href="#cb4-103" aria-hidden="true" tabindex="-1"></a>We set the prior distribution for the intercept to be a normal distribution</span>
<span id="cb4-104"><a href="#cb4-104" aria-hidden="true" tabindex="-1"></a>centered at 0 with standard deviation of 1. All scores are in a range of</span>
<span id="cb4-105"><a href="#cb4-105" aria-hidden="true" tabindex="-1"></a>-1 to 1, so this is a relatively flat prior for our use case.</span>
<span id="cb4-106"><a href="#cb4-106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-107"><a href="#cb4-107" aria-hidden="true" tabindex="-1"></a>The video below shows how scores change in average when comparing different</span>
<span id="cb4-108"><a href="#cb4-108" aria-hidden="true" tabindex="-1"></a>neighborhoods. </span>
<span id="cb4-109"><a href="#cb4-109" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-110"><a href="#cb4-110" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb4-111"><a href="#cb4-111" aria-hidden="true" tabindex="-1"></a>scores_to_use <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb4-112"><a href="#cb4-112" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Estrogen response early"</span>,</span>
<span id="cb4-113"><a href="#cb4-113" aria-hidden="true" tabindex="-1"></a>    <span class="st">"E2F targets"</span>,</span>
<span id="cb4-114"><a href="#cb4-114" aria-hidden="true" tabindex="-1"></a>    <span class="st">"P53 pathway"</span>,</span>
<span id="cb4-115"><a href="#cb4-115" aria-hidden="true" tabindex="-1"></a>    <span class="st">"EMT"</span>,</span>
<span id="cb4-116"><a href="#cb4-116" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Androgen response"</span>,</span>
<span id="cb4-117"><a href="#cb4-117" aria-hidden="true" tabindex="-1"></a>    <span class="st">"TGFb signaling"</span>,</span>
<span id="cb4-118"><a href="#cb4-118" aria-hidden="true" tabindex="-1"></a>    <span class="st">"SET ER/PR"</span></span>
<span id="cb4-119"><a href="#cb4-119" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> <span class="st">`</span><span class="at">names&lt;-</span><span class="st">`</span>(<span class="fu">c</span>(</span>
<span id="cb4-120"><a href="#cb4-120" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_ESTROGEN_RESPONSE_EARLY"</span>,</span>
<span id="cb4-121"><a href="#cb4-121" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_E2F_TARGETS"</span>,</span>
<span id="cb4-122"><a href="#cb4-122" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_P53_PATHWAY"</span>,</span>
<span id="cb4-123"><a href="#cb4-123" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION"</span>,</span>
<span id="cb4-124"><a href="#cb4-124" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_ANDROGEN_RESPONSE"</span>,</span>
<span id="cb4-125"><a href="#cb4-125" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_TGF_BETA_SIGNALING"</span>,</span>
<span id="cb4-126"><a href="#cb4-126" aria-hidden="true" tabindex="-1"></a>    <span class="st">"SET_ERPR"</span></span>
<span id="cb4-127"><a href="#cb4-127" aria-hidden="true" tabindex="-1"></a>))</span>
<span id="cb4-128"><a href="#cb4-128" aria-hidden="true" tabindex="-1"></a>df_pca <span class="ot">&lt;-</span> df_pca <span class="sc">%&gt;%</span> </span>
<span id="cb4-129"><a href="#cb4-129" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb4-130"><a href="#cb4-130" aria-hidden="true" tabindex="-1"></a>        pam50 <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"normal"</span>, <span class="st">"luma"</span>, <span class="st">"lumb"</span>, <span class="st">"basal"</span>, <span class="st">"her2"</span>)</span>
<span id="cb4-131"><a href="#cb4-131" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span> </span>
<span id="cb4-132"><a href="#cb4-132" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">rownames&lt;-</span><span class="st">`</span>(.[, <span class="st">"sample_name"</span>])</span>
<span id="cb4-133"><a href="#cb4-133" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-134"><a href="#cb4-134" aria-hidden="true" tabindex="-1"></a><span class="co"># we first get random samples from the path determined below</span></span>
<span id="cb4-135"><a href="#cb4-135" aria-hidden="true" tabindex="-1"></a>pc3_values <span class="ot">&lt;-</span> <span class="fl">2.5</span> <span class="sc">+</span> <span class="fu">rnorm</span>(<span class="dv">0</span>, <span class="fl">0.1</span>, <span class="at">n =</span> <span class="dv">20</span>)</span>
<span id="cb4-136"><a href="#cb4-136" aria-hidden="true" tabindex="-1"></a>pc4_values <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="at">from =</span> <span class="sc">-</span><span class="dv">4</span>, <span class="at">to =</span> <span class="dv">4</span>, <span class="at">length.out =</span> <span class="dv">20</span>)</span>
<span id="cb4-137"><a href="#cb4-137" aria-hidden="true" tabindex="-1"></a>pca_values <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">PC3 =</span> pc3_values, <span class="at">PC4 =</span> pc4_values) <span class="sc">%&gt;%</span> as.matrix</span>
<span id="cb4-138"><a href="#cb4-138" aria-hidden="true" tabindex="-1"></a>samples_top_down <span class="ot">&lt;-</span> <span class="fu">apply</span>(pca_values, <span class="dv">1</span>, get_closest_sample, <span class="at">df_pca =</span> df_pca)</span>
<span id="cb4-139"><a href="#cb4-139" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-140"><a href="#cb4-140" aria-hidden="true" tabindex="-1"></a>pc3_values <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="at">from =</span> <span class="sc">-</span><span class="fl">8.5</span>, <span class="at">to =</span> <span class="dv">1</span>, <span class="at">length.out =</span> <span class="dv">20</span>)</span>
<span id="cb4-141"><a href="#cb4-141" aria-hidden="true" tabindex="-1"></a>pc4_values <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(<span class="dv">1</span>, <span class="fl">0.1</span>, <span class="at">n =</span> <span class="dv">20</span>)</span>
<span id="cb4-142"><a href="#cb4-142" aria-hidden="true" tabindex="-1"></a>pca_values <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">PC3 =</span> pc3_values, <span class="at">PC4 =</span> pc4_values) <span class="sc">%&gt;%</span> as.matrix</span>
<span id="cb4-143"><a href="#cb4-143" aria-hidden="true" tabindex="-1"></a>samples_left_right <span class="ot">&lt;-</span> <span class="fu">apply</span>(pca_values, <span class="dv">1</span>, get_closest_sample, <span class="at">df_pca =</span> df_pca)</span>
<span id="cb4-144"><a href="#cb4-144" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-145"><a href="#cb4-145" aria-hidden="true" tabindex="-1"></a>all_samples <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb4-146"><a href="#cb4-146" aria-hidden="true" tabindex="-1"></a>    <span class="at">top_down =</span> samples_top_down,</span>
<span id="cb4-147"><a href="#cb4-147" aria-hidden="true" tabindex="-1"></a>    <span class="at">left_right =</span> <span class="fu">rev</span>(samples_left_right)</span>
<span id="cb4-148"><a href="#cb4-148" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb4-149"><a href="#cb4-149" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-150"><a href="#cb4-150" aria-hidden="true" tabindex="-1"></a>pca_values <span class="ot">&lt;-</span> <span class="fu">lapply</span>(</span>
<span id="cb4-151"><a href="#cb4-151" aria-hidden="true" tabindex="-1"></a>    all_samples, </span>
<span id="cb4-152"><a href="#cb4-152" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(x, df_pca){</span>
<span id="cb4-153"><a href="#cb4-153" aria-hidden="true" tabindex="-1"></a>        df_pca <span class="sc">%&gt;%</span> </span>
<span id="cb4-154"><a href="#cb4-154" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">select</span>(PC3, PC4) <span class="sc">%&gt;%</span></span>
<span id="cb4-155"><a href="#cb4-155" aria-hidden="true" tabindex="-1"></a>            as.matrix <span class="sc">%&gt;%</span></span>
<span id="cb4-156"><a href="#cb4-156" aria-hidden="true" tabindex="-1"></a>            .[x, ]</span>
<span id="cb4-157"><a href="#cb4-157" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb4-158"><a href="#cb4-158" aria-hidden="true" tabindex="-1"></a>    <span class="at">df_pca =</span> df_pca</span>
<span id="cb4-159"><a href="#cb4-159" aria-hidden="true" tabindex="-1"></a>) </span>
<span id="cb4-160"><a href="#cb4-160" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-161"><a href="#cb4-161" aria-hidden="true" tabindex="-1"></a>scores_for_movie <span class="ot">&lt;-</span> <span class="fu">mapply</span>(</span>
<span id="cb4-162"><a href="#cb4-162" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(</span>
<span id="cb4-163"><a href="#cb4-163" aria-hidden="true" tabindex="-1"></a>        pca_value, </span>
<span id="cb4-164"><a href="#cb4-164" aria-hidden="true" tabindex="-1"></a>        which_direction, </span>
<span id="cb4-165"><a href="#cb4-165" aria-hidden="true" tabindex="-1"></a>        df_pca, </span>
<span id="cb4-166"><a href="#cb4-166" aria-hidden="true" tabindex="-1"></a>        scores_to_use,</span>
<span id="cb4-167"><a href="#cb4-167" aria-hidden="true" tabindex="-1"></a>        radius,</span>
<span id="cb4-168"><a href="#cb4-168" aria-hidden="true" tabindex="-1"></a>        base_size</span>
<span id="cb4-169"><a href="#cb4-169" aria-hidden="true" tabindex="-1"></a>    ){</span>
<span id="cb4-170"><a href="#cb4-170" aria-hidden="true" tabindex="-1"></a>        <span class="fu">sapply</span>(</span>
<span id="cb4-171"><a href="#cb4-171" aria-hidden="true" tabindex="-1"></a>            <span class="fu">rownames</span>(pca_value), </span>
<span id="cb4-172"><a href="#cb4-172" aria-hidden="true" tabindex="-1"></a>            get_patient_scores_distributions,</span>
<span id="cb4-173"><a href="#cb4-173" aria-hidden="true" tabindex="-1"></a>            <span class="at">df_pca =</span> df_pca,</span>
<span id="cb4-174"><a href="#cb4-174" aria-hidden="true" tabindex="-1"></a>            <span class="at">scores_to_use =</span> scores_to_use,</span>
<span id="cb4-175"><a href="#cb4-175" aria-hidden="true" tabindex="-1"></a>            <span class="at">which_direction =</span> which_direction,</span>
<span id="cb4-176"><a href="#cb4-176" aria-hidden="true" tabindex="-1"></a>            <span class="at">radius =</span> radius,</span>
<span id="cb4-177"><a href="#cb4-177" aria-hidden="true" tabindex="-1"></a>            <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>, </span>
<span id="cb4-178"><a href="#cb4-178" aria-hidden="true" tabindex="-1"></a>            <span class="at">simplify =</span> <span class="cn">FALSE</span></span>
<span id="cb4-179"><a href="#cb4-179" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb4-180"><a href="#cb4-180" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb4-181"><a href="#cb4-181" aria-hidden="true" tabindex="-1"></a>    <span class="at">pca_value =</span> pca_values,</span>
<span id="cb4-182"><a href="#cb4-182" aria-hidden="true" tabindex="-1"></a>    <span class="at">which_direction =</span> <span class="fu">names</span>(pca_values),</span>
<span id="cb4-183"><a href="#cb4-183" aria-hidden="true" tabindex="-1"></a>    <span class="at">MoreArgs =</span> <span class="fu">list</span>(</span>
<span id="cb4-184"><a href="#cb4-184" aria-hidden="true" tabindex="-1"></a>        <span class="at">df_pca =</span> df_pca, </span>
<span id="cb4-185"><a href="#cb4-185" aria-hidden="true" tabindex="-1"></a>        <span class="at">scores_to_use =</span> scores_to_use,</span>
<span id="cb4-186"><a href="#cb4-186" aria-hidden="true" tabindex="-1"></a>        <span class="at">radius =</span> <span class="dv">1</span></span>
<span id="cb4-187"><a href="#cb4-187" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb4-188"><a href="#cb4-188" aria-hidden="true" tabindex="-1"></a>    <span class="at">SIMPLIFY =</span> <span class="cn">FALSE</span>,</span>
<span id="cb4-189"><a href="#cb4-189" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span></span>
<span id="cb4-190"><a href="#cb4-190" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb4-191"><a href="#cb4-191" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-192"><a href="#cb4-192" aria-hidden="true" tabindex="-1"></a>plots_scores_for_movie <span class="ot">&lt;-</span> <span class="fu">mapply</span>(</span>
<span id="cb4-193"><a href="#cb4-193" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(</span>
<span id="cb4-194"><a href="#cb4-194" aria-hidden="true" tabindex="-1"></a>        avg_rstan_samples, </span>
<span id="cb4-195"><a href="#cb4-195" aria-hidden="true" tabindex="-1"></a>        which_direction,</span>
<span id="cb4-196"><a href="#cb4-196" aria-hidden="true" tabindex="-1"></a>        ...</span>
<span id="cb4-197"><a href="#cb4-197" aria-hidden="true" tabindex="-1"></a>    ){</span>
<span id="cb4-198"><a href="#cb4-198" aria-hidden="true" tabindex="-1"></a>        <span class="fu">sapply</span>(</span>
<span id="cb4-199"><a href="#cb4-199" aria-hidden="true" tabindex="-1"></a>            avg_rstan_samples, </span>
<span id="cb4-200"><a href="#cb4-200" aria-hidden="true" tabindex="-1"></a>            get_plot_patient_distribution,</span>
<span id="cb4-201"><a href="#cb4-201" aria-hidden="true" tabindex="-1"></a>            <span class="at">which_direction =</span> which_direction,</span>
<span id="cb4-202"><a href="#cb4-202" aria-hidden="true" tabindex="-1"></a>            ...,</span>
<span id="cb4-203"><a href="#cb4-203" aria-hidden="true" tabindex="-1"></a>            <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>, </span>
<span id="cb4-204"><a href="#cb4-204" aria-hidden="true" tabindex="-1"></a>            <span class="at">simplify =</span> <span class="cn">FALSE</span></span>
<span id="cb4-205"><a href="#cb4-205" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb4-206"><a href="#cb4-206" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb4-207"><a href="#cb4-207" aria-hidden="true" tabindex="-1"></a>    <span class="at">avg_rstan_samples =</span> scores_for_movie,</span>
<span id="cb4-208"><a href="#cb4-208" aria-hidden="true" tabindex="-1"></a>    <span class="at">which_direction =</span> <span class="fu">names</span>(scores_for_movie),</span>
<span id="cb4-209"><a href="#cb4-209" aria-hidden="true" tabindex="-1"></a>    <span class="at">MoreArgs =</span> <span class="fu">list</span>(</span>
<span id="cb4-210"><a href="#cb4-210" aria-hidden="true" tabindex="-1"></a>        <span class="at">scores_to_use =</span> scores_to_use,</span>
<span id="cb4-211"><a href="#cb4-211" aria-hidden="true" tabindex="-1"></a>        <span class="at">base_size =</span> <span class="dv">15</span>,</span>
<span id="cb4-212"><a href="#cb4-212" aria-hidden="true" tabindex="-1"></a>        <span class="at">size_dots =</span> <span class="dv">3</span></span>
<span id="cb4-213"><a href="#cb4-213" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb4-214"><a href="#cb4-214" aria-hidden="true" tabindex="-1"></a>    <span class="at">SIMPLIFY =</span> <span class="cn">FALSE</span>,</span>
<span id="cb4-215"><a href="#cb4-215" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span></span>
<span id="cb4-216"><a href="#cb4-216" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb4-217"><a href="#cb4-217" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-218"><a href="#cb4-218" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-219"><a href="#cb4-219" aria-hidden="true" tabindex="-1"></a>plots_molecular_landscape <span class="ot">&lt;-</span> <span class="fu">plot_selected_samples</span>(</span>
<span id="cb4-220"><a href="#cb4-220" aria-hidden="true" tabindex="-1"></a>    df_pca,</span>
<span id="cb4-221"><a href="#cb4-221" aria-hidden="true" tabindex="-1"></a>    pca_values,</span>
<span id="cb4-222"><a href="#cb4-222" aria-hidden="true" tabindex="-1"></a>    <span class="at">scores_plots_movie =</span> plots_scores_for_movie,</span>
<span id="cb4-223"><a href="#cb4-223" aria-hidden="true" tabindex="-1"></a>    <span class="at">size_line =</span> <span class="fl">0.5</span>,</span>
<span id="cb4-224"><a href="#cb4-224" aria-hidden="true" tabindex="-1"></a>    <span class="at">radius =</span> <span class="dv">1</span>,</span>
<span id="cb4-225"><a href="#cb4-225" aria-hidden="true" tabindex="-1"></a>    <span class="at">size_points =</span> <span class="dv">3</span>,</span>
<span id="cb4-226"><a href="#cb4-226" aria-hidden="true" tabindex="-1"></a>    <span class="at">title_plot =</span> <span class="st">"Molecular landscape"</span></span>
<span id="cb4-227"><a href="#cb4-227" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb4-228"><a href="#cb4-228" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-229"><a href="#cb4-229" aria-hidden="true" tabindex="-1"></a>plots_estimates_tog <span class="ot">&lt;-</span> <span class="fu">mapply</span>(</span>
<span id="cb4-230"><a href="#cb4-230" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(plot1, plot2){</span>
<span id="cb4-231"><a href="#cb4-231" aria-hidden="true" tabindex="-1"></a>        cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(plot1, plot2, <span class="at">nrow=</span><span class="dv">2</span>)</span>
<span id="cb4-232"><a href="#cb4-232" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb4-233"><a href="#cb4-233" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot1 =</span> plots_scores_for_movie<span class="sc">$</span>top_down,</span>
<span id="cb4-234"><a href="#cb4-234" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot2 =</span> plots_scores_for_movie<span class="sc">$</span>left_right,</span>
<span id="cb4-235"><a href="#cb4-235" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">FALSE</span>,</span>
<span id="cb4-236"><a href="#cb4-236" aria-hidden="true" tabindex="-1"></a>    <span class="at">SIMPLIFY =</span> <span class="cn">FALSE</span></span>
<span id="cb4-237"><a href="#cb4-237" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb4-238"><a href="#cb4-238" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-239"><a href="#cb4-239" aria-hidden="true" tabindex="-1"></a>final_plots <span class="ot">&lt;-</span> <span class="fu">lapply</span>(</span>
<span id="cb4-240"><a href="#cb4-240" aria-hidden="true" tabindex="-1"></a>    <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(plots_estimates_tog),</span>
<span id="cb4-241"><a href="#cb4-241" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(i) {</span>
<span id="cb4-242"><a href="#cb4-242" aria-hidden="true" tabindex="-1"></a>        cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(</span>
<span id="cb4-243"><a href="#cb4-243" aria-hidden="true" tabindex="-1"></a>            plots_molecular_landscape[[i]],</span>
<span id="cb4-244"><a href="#cb4-244" aria-hidden="true" tabindex="-1"></a>            plots_estimates_tog[[i]], </span>
<span id="cb4-245"><a href="#cb4-245" aria-hidden="true" tabindex="-1"></a>            <span class="at">ncol =</span> <span class="dv">2</span>, </span>
<span id="cb4-246"><a href="#cb4-246" aria-hidden="true" tabindex="-1"></a>            <span class="at">rel_widths =</span> <span class="fu">c</span>(<span class="fl">1.8</span>, <span class="dv">1</span>)</span>
<span id="cb4-247"><a href="#cb4-247" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb4-248"><a href="#cb4-248" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb4-249"><a href="#cb4-249" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb4-250"><a href="#cb4-250" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-251"><a href="#cb4-251" aria-hidden="true" tabindex="-1"></a>folder_to_save <span class="ot">&lt;-</span> <span class="st">"../results/plots/scoring/movie"</span></span>
<span id="cb4-252"><a href="#cb4-252" aria-hidden="true" tabindex="-1"></a><span class="fu">dir.create</span>(</span>
<span id="cb4-253"><a href="#cb4-253" aria-hidden="true" tabindex="-1"></a>    folder_to_save, </span>
<span id="cb4-254"><a href="#cb4-254" aria-hidden="true" tabindex="-1"></a>    <span class="at">showWarnings =</span> <span class="cn">FALSE</span>, </span>
<span id="cb4-255"><a href="#cb4-255" aria-hidden="true" tabindex="-1"></a>    <span class="at">recursive =</span> <span class="cn">TRUE</span></span>
<span id="cb4-256"><a href="#cb4-256" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb4-257"><a href="#cb4-257" aria-hidden="true" tabindex="-1"></a>fig_width <span class="ot">&lt;-</span> <span class="dv">22</span></span>
<span id="cb4-258"><a href="#cb4-258" aria-hidden="true" tabindex="-1"></a>fig_height <span class="ot">&lt;-</span> <span class="dv">10</span></span>
<span id="cb4-259"><a href="#cb4-259" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-260"><a href="#cb4-260" aria-hidden="true" tabindex="-1"></a><span class="fu">sapply</span>(</span>
<span id="cb4-261"><a href="#cb4-261" aria-hidden="true" tabindex="-1"></a>    <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(final_plots),</span>
<span id="cb4-262"><a href="#cb4-262" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(i, final_plots){</span>
<span id="cb4-263"><a href="#cb4-263" aria-hidden="true" tabindex="-1"></a>        ggplot2<span class="sc">::</span><span class="fu">ggsave</span>(</span>
<span id="cb4-264"><a href="#cb4-264" aria-hidden="true" tabindex="-1"></a>            <span class="at">filename =</span> <span class="fu">paste0</span>(</span>
<span id="cb4-265"><a href="#cb4-265" aria-hidden="true" tabindex="-1"></a>                folder_to_save, </span>
<span id="cb4-266"><a href="#cb4-266" aria-hidden="true" tabindex="-1"></a>                <span class="st">"/"</span>, </span>
<span id="cb4-267"><a href="#cb4-267" aria-hidden="true" tabindex="-1"></a>                <span class="fu">ifelse</span>(i <span class="sc">&lt;</span> <span class="dv">10</span>, <span class="fu">paste0</span>(<span class="st">"0"</span>,i), i), <span class="st">".png"</span></span>
<span id="cb4-268"><a href="#cb4-268" aria-hidden="true" tabindex="-1"></a>            ),</span>
<span id="cb4-269"><a href="#cb4-269" aria-hidden="true" tabindex="-1"></a>            <span class="at">plot =</span> final_plots[[i]],</span>
<span id="cb4-270"><a href="#cb4-270" aria-hidden="true" tabindex="-1"></a>            <span class="at">width =</span> fig_width,</span>
<span id="cb4-271"><a href="#cb4-271" aria-hidden="true" tabindex="-1"></a>            <span class="at">height =</span> fig_height, </span>
<span id="cb4-272"><a href="#cb4-272" aria-hidden="true" tabindex="-1"></a>            <span class="at">dpi =</span> <span class="dv">320</span></span>
<span id="cb4-273"><a href="#cb4-273" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb4-274"><a href="#cb4-274" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb4-275"><a href="#cb4-275" aria-hidden="true" tabindex="-1"></a>    <span class="at">final_plots =</span> final_plots</span>
<span id="cb4-276"><a href="#cb4-276" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb4-277"><a href="#cb4-277" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-278"><a href="#cb4-278" aria-hidden="true" tabindex="-1"></a><span class="co"># the original command used in the command line is the one below:</span></span>
<span id="cb4-279"><a href="#cb4-279" aria-hidden="true" tabindex="-1"></a><span class="co"># ffmpeg -y -framerate 2 -pattern_type glob -i 'movie/*.png' mol_land.mp4</span></span>
<span id="cb4-280"><a href="#cb4-280" aria-hidden="true" tabindex="-1"></a><span class="co"># we can then call it directly from R instead of having to do it manually</span></span>
<span id="cb4-281"><a href="#cb4-281" aria-hidden="true" tabindex="-1"></a><span class="co"># after rendering the document</span></span>
<span id="cb4-282"><a href="#cb4-282" aria-hidden="true" tabindex="-1"></a><span class="fu">system</span>(</span>
<span id="cb4-283"><a href="#cb4-283" aria-hidden="true" tabindex="-1"></a>    <span class="fu">paste0</span>(</span>
<span id="cb4-284"><a href="#cb4-284" aria-hidden="true" tabindex="-1"></a>        <span class="st">"ffmpeg -y -framerate 2 -pattern_type glob -i '"</span>,</span>
<span id="cb4-285"><a href="#cb4-285" aria-hidden="true" tabindex="-1"></a>        folder_to_save, <span class="st">"/*.png' "</span>, </span>
<span id="cb4-286"><a href="#cb4-286" aria-hidden="true" tabindex="-1"></a>        folder_to_save, <span class="st">"/../mol_land.mp4"</span></span>
<span id="cb4-287"><a href="#cb4-287" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb4-288"><a href="#cb4-288" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb4-289"><a href="#cb4-289" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-290"><a href="#cb4-290" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-291"><a href="#cb4-291" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = !first_run, results = 'asis', echo = FALSE}</span></span>
<span id="cb4-292"><a href="#cb4-292" aria-hidden="true" tabindex="-1"></a>embedding_video <span class="ot">&lt;-</span> <span class="fu">paste0</span>(</span>
<span id="cb4-293"><a href="#cb4-293" aria-hidden="true" tabindex="-1"></a>    <span class="st">'&lt;iframe width="720" height="480" '</span>,</span>
<span id="cb4-294"><a href="#cb4-294" aria-hidden="true" tabindex="-1"></a>    <span class="st">'src="../plots/scoring/mol_land.mp4" align="middle"'</span>,</span>
<span id="cb4-295"><a href="#cb4-295" aria-hidden="true" tabindex="-1"></a>    <span class="st">'frameborder="0" allowfullscreen&gt;&lt;/iframe&gt;'</span></span>
<span id="cb4-296"><a href="#cb4-296" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb4-297"><a href="#cb4-297" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-298"><a href="#cb4-298" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(embedding_video)</span>
<span id="cb4-299"><a href="#cb4-299" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-300"><a href="#cb4-300" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-301"><a href="#cb4-301" aria-hidden="true" tabindex="-1"></a>Going from right to left it shows how ER signaling is decreased. From top to</span>
<span id="cb4-302"><a href="#cb4-302" aria-hidden="true" tabindex="-1"></a>bottom there is a reduction of proliferation, E2F targets, and a change in</span>
<span id="cb4-303"><a href="#cb4-303" aria-hidden="true" tabindex="-1"></a>EMT and TGFb signaling. These are representative scores, other pathways could</span>
<span id="cb4-304"><a href="#cb4-304" aria-hidden="true" tabindex="-1"></a>be used as well. </span>
<span id="cb4-305"><a href="#cb4-305" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-306"><a href="#cb4-306" aria-hidden="true" tabindex="-1"></a><span class="fu">## POETIC embedding</span></span>
<span id="cb4-307"><a href="#cb4-307" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-308"><a href="#cb4-308" aria-hidden="true" tabindex="-1"></a>In this section we show the embedding of the POETIC trial samples.</span>
<span id="cb4-309"><a href="#cb4-309" aria-hidden="true" tabindex="-1"></a>But first we analyse some basic properties of the dataset.</span>
<span id="cb4-310"><a href="#cb4-310" aria-hidden="true" tabindex="-1"></a>The data for this dataset was downloaded and processed</span>
<span id="cb4-311"><a href="#cb4-311" aria-hidden="true" tabindex="-1"></a>as described at </span>
<span id="cb4-312"><a href="#cb4-312" aria-hidden="true" tabindex="-1"></a><span class="co">[</span><span class="ot">chronchi.github.io/transcriptomics</span><span class="co">](https://chronchi.github.io/transcriptomics/)</span></span>
<span id="cb4-313"><a href="#cb4-313" aria-hidden="true" tabindex="-1"></a>in the chapter AI - GSE105777. </span>
<span id="cb4-314"><a href="#cb4-314" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-315"><a href="#cb4-315" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb4-316"><a href="#cb4-316" aria-hidden="true" tabindex="-1"></a>datasets <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(</span>
<span id="cb4-317"><a href="#cb4-317" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/surv_analysis_estrogen/datasets_with_scores.rds"</span></span>
<span id="cb4-318"><a href="#cb4-318" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb4-319"><a href="#cb4-319" aria-hidden="true" tabindex="-1"></a>poetic <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../data/poetic.rds"</span>)</span>
<span id="cb4-320"><a href="#cb4-320" aria-hidden="true" tabindex="-1"></a>df_pca_coordinates <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(</span>
<span id="cb4-321"><a href="#cb4-321" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/pca_merging/df_pca_coordinates.rds"</span></span>
<span id="cb4-322"><a href="#cb4-322" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb4-323"><a href="#cb4-323" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-324"><a href="#cb4-324" aria-hidden="true" tabindex="-1"></a>datasets<span class="sc">$</span>poetic <span class="ot">&lt;-</span> poetic</span>
<span id="cb4-325"><a href="#cb4-325" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-326"><a href="#cb4-326" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb4-327"><a href="#cb4-327" aria-hidden="true" tabindex="-1"></a>    datasets,</span>
<span id="cb4-328"><a href="#cb4-328" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/scoring/datasets_with_poetic.rds"</span></span>
<span id="cb4-329"><a href="#cb4-329" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb4-330"><a href="#cb4-330" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-331"><a href="#cb4-331" aria-hidden="true" tabindex="-1"></a><span class="co"># get the normalization performed </span></span>
<span id="cb4-332"><a href="#cb4-332" aria-hidden="true" tabindex="-1"></a>poetic_normalized <span class="ot">&lt;-</span> <span class="fu">get_final_ranking_values</span>(</span>
<span id="cb4-333"><a href="#cb4-333" aria-hidden="true" tabindex="-1"></a>    <span class="at">sum_exp =</span> poetic,</span>
<span id="cb4-334"><a href="#cb4-334" aria-hidden="true" tabindex="-1"></a>    <span class="at">assay_to_use =</span> <span class="st">"normalized_intensity"</span>,</span>
<span id="cb4-335"><a href="#cb4-335" aria-hidden="true" tabindex="-1"></a>    <span class="at">stable_genes =</span> stable_genes,</span>
<span id="cb4-336"><a href="#cb4-336" aria-hidden="true" tabindex="-1"></a>    <span class="at">most_variable_genes =</span> <span class="fu">setdiff</span>(<span class="fu">rownames</span>(pca_fit<span class="sc">$</span>loadings), stable_genes)</span>
<span id="cb4-337"><a href="#cb4-337" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb4-338"><a href="#cb4-338" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-339"><a href="#cb4-339" aria-hidden="true" tabindex="-1"></a><span class="co"># calculate the embedding</span></span>
<span id="cb4-340"><a href="#cb4-340" aria-hidden="true" tabindex="-1"></a>poetic_df_pca <span class="ot">&lt;-</span> <span class="fu">get_pca_coordinates</span>(poetic_normalized, pca_fit) <span class="sc">%&gt;%</span></span>
<span id="cb4-341"><a href="#cb4-341" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span> </span>
<span id="cb4-342"><a href="#cb4-342" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">bind_cols</span>(</span>
<span id="cb4-343"><a href="#cb4-343" aria-hidden="true" tabindex="-1"></a>        ., </span>
<span id="cb4-344"><a href="#cb4-344" aria-hidden="true" tabindex="-1"></a>        <span class="fu">colData</span>(poetic_normalized) <span class="sc">%&gt;%</span> data.frame <span class="sc">%&gt;%</span> </span>
<span id="cb4-345"><a href="#cb4-345" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb4-346"><a href="#cb4-346" aria-hidden="true" tabindex="-1"></a>                <span class="at">er_status =</span> <span class="fu">ifelse</span>(</span>
<span id="cb4-347"><a href="#cb4-347" aria-hidden="true" tabindex="-1"></a>                    er_status <span class="sc">==</span> <span class="st">"Positive"</span>, </span>
<span id="cb4-348"><a href="#cb4-348" aria-hidden="true" tabindex="-1"></a>                    <span class="st">"pos"</span>, </span>
<span id="cb4-349"><a href="#cb4-349" aria-hidden="true" tabindex="-1"></a>                    <span class="st">"neg"</span></span>
<span id="cb4-350"><a href="#cb4-350" aria-hidden="true" tabindex="-1"></a>                )</span>
<span id="cb4-351"><a href="#cb4-351" aria-hidden="true" tabindex="-1"></a>            ) <span class="sc">%&gt;%</span> </span>
<span id="cb4-352"><a href="#cb4-352" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">cohort =</span> <span class="st">"poetic"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb4-353"><a href="#cb4-353" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">is_responder =</span></span>
<span id="cb4-354"><a href="#cb4-354" aria-hidden="true" tabindex="-1"></a>                r_or_no_r_change_ki67_60_and_baseline_ki67_5_percent</span>
<span id="cb4-355"><a href="#cb4-355" aria-hidden="true" tabindex="-1"></a>            ) <span class="sc">%&gt;%</span> </span>
<span id="cb4-356"><a href="#cb4-356" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb4-357"><a href="#cb4-357" aria-hidden="true" tabindex="-1"></a>                <span class="at">sample_name =</span> <span class="fu">colnames</span>(poetic_normalized)</span>
<span id="cb4-358"><a href="#cb4-358" aria-hidden="true" tabindex="-1"></a>            )</span>
<span id="cb4-359"><a href="#cb4-359" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span> </span>
<span id="cb4-360"><a href="#cb4-360" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(., df_pca_coordinates)</span>
<span id="cb4-361"><a href="#cb4-361" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-362"><a href="#cb4-362" aria-hidden="true" tabindex="-1"></a><span class="co"># save all the results</span></span>
<span id="cb4-363"><a href="#cb4-363" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb4-364"><a href="#cb4-364" aria-hidden="true" tabindex="-1"></a>    poetic_df_pca <span class="sc">%&gt;%</span> data.frame,</span>
<span id="cb4-365"><a href="#cb4-365" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/scoring/poetic_df_pca.rds"</span></span>
<span id="cb4-366"><a href="#cb4-366" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb4-367"><a href="#cb4-367" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-368"><a href="#cb4-368" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb4-369"><a href="#cb4-369" aria-hidden="true" tabindex="-1"></a>    poetic_normalized,</span>
<span id="cb4-370"><a href="#cb4-370" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/scoring/poetic_normalized.rds"</span></span>
<span id="cb4-371"><a href="#cb4-371" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb4-372"><a href="#cb4-372" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-373"><a href="#cb4-373" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-374"><a href="#cb4-374" aria-hidden="true" tabindex="-1"></a><span class="fu">### Number of samples</span></span>
<span id="cb4-375"><a href="#cb4-375" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-376"><a href="#cb4-376" aria-hidden="true" tabindex="-1"></a>In this dataset there are matched samples from patients that </span>
<span id="cb4-377"><a href="#cb4-377" aria-hidden="true" tabindex="-1"></a>are treated and untreated. @tbl-pt-poetic-nb</span>
<span id="cb4-378"><a href="#cb4-378" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-381"><a href="#cb4-381" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb4-382"><a href="#cb4-382" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: tbl-pt-poetic-nb</span></span>
<span id="cb4-383"><a href="#cb4-383" aria-hidden="true" tabindex="-1"></a><span class="co">#| tbl-cap: Table showing the number of samples for treated and untreated</span></span>
<span id="cb4-384"><a href="#cb4-384" aria-hidden="true" tabindex="-1"></a><span class="co">#|     patients</span></span>
<span id="cb4-385"><a href="#cb4-385" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(poetic_normalized) <span class="sc">%&gt;%</span> data.frame <span class="sc">%&gt;%</span></span>
<span id="cb4-386"><a href="#cb4-386" aria-hidden="true" tabindex="-1"></a>    janitor<span class="sc">::</span><span class="fu">tabyl</span>(group) <span class="sc">%&gt;%</span> </span>
<span id="cb4-387"><a href="#cb4-387" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">n =</span> n<span class="sc">/</span><span class="dv">2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-388"><a href="#cb4-388" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">percent =</span> <span class="fu">round</span>(n<span class="sc">/</span><span class="fu">sum</span>(n) <span class="sc">*</span> <span class="dv">100</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb4-389"><a href="#cb4-389" aria-hidden="true" tabindex="-1"></a>    kableExtra<span class="sc">::</span><span class="fu">kbl</span>() <span class="sc">%&gt;%</span></span>
<span id="cb4-390"><a href="#cb4-390" aria-hidden="true" tabindex="-1"></a>    kableExtra<span class="sc">::</span><span class="fu">kable_classic</span>()</span>
<span id="cb4-391"><a href="#cb4-391" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-392"><a href="#cb4-392" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-393"><a href="#cb4-393" aria-hidden="true" tabindex="-1"></a>In this cohort, there is the PAM50 molecular subtype available</span>
<span id="cb4-394"><a href="#cb4-394" aria-hidden="true" tabindex="-1"></a>for the untreated. @tbl-pt-poetic-pam50 shows the</span>
<span id="cb4-395"><a href="#cb4-395" aria-hidden="true" tabindex="-1"></a>number of patients for each subtype.</span>
<span id="cb4-396"><a href="#cb4-396" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-399"><a href="#cb4-399" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb4-400"><a href="#cb4-400" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: tbl-pt-poetic-pam50</span></span>
<span id="cb4-401"><a href="#cb4-401" aria-hidden="true" tabindex="-1"></a><span class="co">#| tbl-cap: Table showing the number of molecular subtypes for each </span></span>
<span id="cb4-402"><a href="#cb4-402" aria-hidden="true" tabindex="-1"></a><span class="co">#|     group.</span></span>
<span id="cb4-403"><a href="#cb4-403" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(poetic_normalized) <span class="sc">%&gt;%</span> data.frame <span class="sc">%&gt;%</span></span>
<span id="cb4-404"><a href="#cb4-404" aria-hidden="true" tabindex="-1"></a>    janitor<span class="sc">::</span><span class="fu">tabyl</span>(pam50) <span class="sc">%&gt;%</span> </span>
<span id="cb4-405"><a href="#cb4-405" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">percent =</span> <span class="fu">round</span>(n<span class="sc">/</span><span class="fu">sum</span>(n) <span class="sc">*</span> <span class="dv">100</span>, <span class="at">digits =</span> <span class="dv">3</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb4-406"><a href="#cb4-406" aria-hidden="true" tabindex="-1"></a>    kableExtra<span class="sc">::</span><span class="fu">kbl</span>() <span class="sc">%&gt;%</span></span>
<span id="cb4-407"><a href="#cb4-407" aria-hidden="true" tabindex="-1"></a>    kableExtra<span class="sc">::</span><span class="fu">kable_classic</span>()</span>
<span id="cb4-408"><a href="#cb4-408" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-409"><a href="#cb4-409" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-410"><a href="#cb4-410" aria-hidden="true" tabindex="-1"></a>The numbers differ from previously, since the molecular subtype is </span>
<span id="cb4-411"><a href="#cb4-411" aria-hidden="true" tabindex="-1"></a>calculated for each sample, so two molecular subtypes for each patient. </span>
<span id="cb4-412"><a href="#cb4-412" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-413"><a href="#cb4-413" aria-hidden="true" tabindex="-1"></a><span class="fu">### Proportion of top loadings</span></span>
<span id="cb4-414"><a href="#cb4-414" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-417"><a href="#cb4-417" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb4-418"><a href="#cb4-418" aria-hidden="true" tabindex="-1"></a>nb_genes <span class="ot">&lt;-</span> <span class="dv">200</span></span>
<span id="cb4-419"><a href="#cb4-419" aria-hidden="true" tabindex="-1"></a>proportions <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="at">by =</span> <span class="fl">0.05</span>)</span>
<span id="cb4-420"><a href="#cb4-420" aria-hidden="true" tabindex="-1"></a>gene_names <span class="ot">&lt;-</span> <span class="fu">rownames</span>(pca_fit<span class="sc">$</span>loadings)</span>
<span id="cb4-421"><a href="#cb4-421" aria-hidden="true" tabindex="-1"></a>which_pcs <span class="ot">&lt;-</span> <span class="dv">3</span><span class="sc">:</span><span class="dv">4</span></span>
<span id="cb4-422"><a href="#cb4-422" aria-hidden="true" tabindex="-1"></a>top_loadings_genes <span class="ot">&lt;-</span> <span class="fu">sapply</span>(</span>
<span id="cb4-423"><a href="#cb4-423" aria-hidden="true" tabindex="-1"></a>    which_pcs,</span>
<span id="cb4-424"><a href="#cb4-424" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(pc, gene_names, pca_fit, nb_genes){</span>
<span id="cb4-425"><a href="#cb4-425" aria-hidden="true" tabindex="-1"></a>        gene_names[<span class="fu">order</span>(</span>
<span id="cb4-426"><a href="#cb4-426" aria-hidden="true" tabindex="-1"></a>            <span class="fu">abs</span>(pca_fit<span class="sc">$</span>loadings[, <span class="fu">paste0</span>(<span class="st">"PC"</span>, pc)]), </span>
<span id="cb4-427"><a href="#cb4-427" aria-hidden="true" tabindex="-1"></a>            <span class="at">decreasing =</span> <span class="cn">TRUE</span></span>
<span id="cb4-428"><a href="#cb4-428" aria-hidden="true" tabindex="-1"></a>        )[<span class="dv">1</span><span class="sc">:</span>nb_genes]]   </span>
<span id="cb4-429"><a href="#cb4-429" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb4-430"><a href="#cb4-430" aria-hidden="true" tabindex="-1"></a>    <span class="at">gene_names =</span> gene_names,</span>
<span id="cb4-431"><a href="#cb4-431" aria-hidden="true" tabindex="-1"></a>    <span class="at">pca_fit =</span> pca_fit,</span>
<span id="cb4-432"><a href="#cb4-432" aria-hidden="true" tabindex="-1"></a>    <span class="at">nb_genes =</span> nb_genes</span>
<span id="cb4-433"><a href="#cb4-433" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> data.frame <span class="sc">%&gt;%</span> </span>
<span id="cb4-434"><a href="#cb4-434" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">colnames&lt;-</span><span class="st">`</span>(<span class="fu">paste0</span>(<span class="st">"PC"</span>, which_pcs))</span>
<span id="cb4-435"><a href="#cb4-435" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-436"><a href="#cb4-436" aria-hidden="true" tabindex="-1"></a>missing_genes_pcx <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(</span>
<span id="cb4-437"><a href="#cb4-437" aria-hidden="true" tabindex="-1"></a>    top_loadings_genes<span class="sc">$</span>PC3,</span>
<span id="cb4-438"><a href="#cb4-438" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames</span>(poetic_normalized)</span>
<span id="cb4-439"><a href="#cb4-439" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb4-440"><a href="#cb4-440" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-441"><a href="#cb4-441" aria-hidden="true" tabindex="-1"></a>missing_genes_pcy <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(</span>
<span id="cb4-442"><a href="#cb4-442" aria-hidden="true" tabindex="-1"></a>    top_loadings_genes<span class="sc">$</span>PC4,</span>
<span id="cb4-443"><a href="#cb4-443" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames</span>(poetic_normalized)</span>
<span id="cb4-444"><a href="#cb4-444" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb4-445"><a href="#cb4-445" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-446"><a href="#cb4-446" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-447"><a href="#cb4-447" aria-hidden="true" tabindex="-1"></a>There are in total <span class="in">`r nrow(poetic_normalized)`</span> genes available out</span>
<span id="cb4-448"><a href="#cb4-448" aria-hidden="true" tabindex="-1"></a>of the <span class="in">`r nrow(datasets_normalized)`</span>. Among them, </span>
<span id="cb4-449"><a href="#cb4-449" aria-hidden="true" tabindex="-1"></a><span class="in">`r length(intersect(rownames(poetic_normalized), stable_genes))`</span> are</span>
<span id="cb4-450"><a href="#cb4-450" aria-hidden="true" tabindex="-1"></a>the housekeeping genes. Out of the</span>
<span id="cb4-451"><a href="#cb4-451" aria-hidden="true" tabindex="-1"></a><span class="in">`r nrow(pca_fit$loadings) - nrow(poetic_normalized)`</span> genes missing,</span>
<span id="cb4-452"><a href="#cb4-452" aria-hidden="true" tabindex="-1"></a>a total number of <span class="in">`r length(missing_genes_pcx)`</span> are missing from the</span>
<span id="cb4-453"><a href="#cb4-453" aria-hidden="true" tabindex="-1"></a>top <span class="in">`r nb_genes`</span> PC3 loadings, a proportion of </span>
<span id="cb4-454"><a href="#cb4-454" aria-hidden="true" tabindex="-1"></a><span class="in">`r format(length(missing_genes_pcx)/nb_genes * 100, digits = 1)`</span>\%. </span>
<span id="cb4-455"><a href="#cb4-455" aria-hidden="true" tabindex="-1"></a>A total number of <span class="in">`r length(missing_genes_pcy)`</span> are missing from the</span>
<span id="cb4-456"><a href="#cb4-456" aria-hidden="true" tabindex="-1"></a>top <span class="in">`r nb_genes`</span> PC4 loadings, a proportion of </span>
<span id="cb4-457"><a href="#cb4-457" aria-hidden="true" tabindex="-1"></a><span class="in">`r format(length(missing_genes_pcy)/nb_genes * 100, digits = 1)`</span>\%. These</span>
<span id="cb4-458"><a href="#cb4-458" aria-hidden="true" tabindex="-1"></a>are very small proportions, and we have seen from the last chapter that</span>
<span id="cb4-459"><a href="#cb4-459" aria-hidden="true" tabindex="-1"></a>they will not affect the position of the patients in the embedding.</span>
<span id="cb4-460"><a href="#cb4-460" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-461"><a href="#cb4-461" aria-hidden="true" tabindex="-1"></a><span class="fu">### Embedding </span></span>
<span id="cb4-462"><a href="#cb4-462" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-463"><a href="#cb4-463" aria-hidden="true" tabindex="-1"></a>@fig-pca-poetic-er-pam50 shows a good mixing of the POETIC sample and how</span>
<span id="cb4-464"><a href="#cb4-464" aria-hidden="true" tabindex="-1"></a>all the patients are scattered across the whole landscape. Here all samples</span>
<span id="cb4-465"><a href="#cb4-465" aria-hidden="true" tabindex="-1"></a>are plotted, meaning that every two dots correspond to a single patient.</span>
<span id="cb4-466"><a href="#cb4-466" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-467"><a href="#cb4-467" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=19, fig.height=12}</span></span>
<span id="cb4-468"><a href="#cb4-468" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-pca-poetic-er-pam50</span></span>
<span id="cb4-469"><a href="#cb4-469" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: PCA embedding of all samples from TCGA, SCANB and METABRIC including</span></span>
<span id="cb4-470"><a href="#cb4-470" aria-hidden="true" tabindex="-1"></a><span class="co">#|     the POETIC samples on top.</span></span>
<span id="cb4-471"><a href="#cb4-471" aria-hidden="true" tabindex="-1"></a><span class="co">#|     (A) Colored by cohort,</span></span>
<span id="cb4-472"><a href="#cb4-472" aria-hidden="true" tabindex="-1"></a><span class="co">#|     (B) colored by ER status, (C) colored by PAM50 molecular subtype.</span></span>
<span id="cb4-473"><a href="#cb4-473" aria-hidden="true" tabindex="-1"></a>plots_poetic <span class="ot">&lt;-</span> <span class="fu">sapply</span>(</span>
<span id="cb4-474"><a href="#cb4-474" aria-hidden="true" tabindex="-1"></a>    <span class="fu">c</span>(<span class="st">"cohort"</span>, <span class="st">"er_status"</span>, <span class="st">"pam50"</span>), </span>
<span id="cb4-475"><a href="#cb4-475" aria-hidden="true" tabindex="-1"></a>    get_plot_new_samples,</span>
<span id="cb4-476"><a href="#cb4-476" aria-hidden="true" tabindex="-1"></a>    <span class="at">name_cohort =</span> <span class="st">"poetic"</span>,</span>
<span id="cb4-477"><a href="#cb4-477" aria-hidden="true" tabindex="-1"></a>    <span class="at">df_pca =</span> poetic_df_pca,</span>
<span id="cb4-478"><a href="#cb4-478" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"POETIC samples projected on the molecular landscape"</span>,</span>
<span id="cb4-479"><a href="#cb4-479" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>, </span>
<span id="cb4-480"><a href="#cb4-480" aria-hidden="true" tabindex="-1"></a>    <span class="at">simplify =</span> <span class="cn">FALSE</span>,</span>
<span id="cb4-481"><a href="#cb4-481" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC3"</span>,</span>
<span id="cb4-482"><a href="#cb4-482" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC4"</span></span>
<span id="cb4-483"><a href="#cb4-483" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb4-484"><a href="#cb4-484" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-485"><a href="#cb4-485" aria-hidden="true" tabindex="-1"></a>plots_poetic<span class="sc">$</span>pam50 <span class="ot">&lt;-</span> plots_poetic<span class="sc">$</span>pam50 <span class="sc">+</span></span>
<span id="cb4-486"><a href="#cb4-486" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_color_manual</span>(</span>
<span id="cb4-487"><a href="#cb4-487" aria-hidden="true" tabindex="-1"></a>        <span class="at">values =</span> <span class="fu">c</span>(</span>
<span id="cb4-488"><a href="#cb4-488" aria-hidden="true" tabindex="-1"></a>            <span class="fu">get_colors_pam50</span>(plots_poetic<span class="sc">$</span>pam50<span class="sc">$</span>data),</span>
<span id="cb4-489"><a href="#cb4-489" aria-hidden="true" tabindex="-1"></a>            <span class="st">"not_available"</span> <span class="ot">=</span> <span class="st">"black"</span></span>
<span id="cb4-490"><a href="#cb4-490" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb4-491"><a href="#cb4-491" aria-hidden="true" tabindex="-1"></a>    ) </span>
<span id="cb4-492"><a href="#cb4-492" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-493"><a href="#cb4-493" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(<span class="at">plotlist =</span> plots_poetic, <span class="at">labels =</span> <span class="st">"AUTO"</span>)</span>
<span id="cb4-494"><a href="#cb4-494" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-495"><a href="#cb4-495" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-496"><a href="#cb4-496" aria-hidden="true" tabindex="-1"></a>@fig-pca-responders-baseline shows the embedding of baseline samples</span>
<span id="cb4-497"><a href="#cb4-497" aria-hidden="true" tabindex="-1"></a>from patients that received endocrine therapy prior to surgery. </span>
<span id="cb4-498"><a href="#cb4-498" aria-hidden="true" tabindex="-1"></a>Patients on the left part of the molecular landscape are considered to</span>
<span id="cb4-499"><a href="#cb4-499" aria-hidden="true" tabindex="-1"></a>be non responders, this coincides with the fact this region corresponds</span>
<span id="cb4-500"><a href="#cb4-500" aria-hidden="true" tabindex="-1"></a>to the ER- BC patients.</span>
<span id="cb4-501"><a href="#cb4-501" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-502"><a href="#cb4-502" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=6}</span></span>
<span id="cb4-503"><a href="#cb4-503" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-pca-responders-baseline</span></span>
<span id="cb4-504"><a href="#cb4-504" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: PCA embedding of all samples from TCGA, SCANB and METABRIC including</span></span>
<span id="cb4-505"><a href="#cb4-505" aria-hidden="true" tabindex="-1"></a><span class="co">#|     the POETIC samples on top. POETIC samples correspond to only baseline</span></span>
<span id="cb4-506"><a href="#cb4-506" aria-hidden="true" tabindex="-1"></a><span class="co">#|     treated patients.</span></span>
<span id="cb4-507"><a href="#cb4-507" aria-hidden="true" tabindex="-1"></a>poetic_df_pca_baseline <span class="ot">&lt;-</span> poetic_df_pca <span class="sc">%&gt;%</span> </span>
<span id="cb4-508"><a href="#cb4-508" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span>(cohort <span class="sc">==</span> <span class="st">"poetic"</span> <span class="sc">&amp;</span> group <span class="sc">==</span> <span class="st">"untreated"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb4-509"><a href="#cb4-509" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb4-510"><a href="#cb4-510" aria-hidden="true" tabindex="-1"></a>        <span class="at">timepoint =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(timepoint), <span class="st">"baseline"</span>, timepoint)</span>
<span id="cb4-511"><a href="#cb4-511" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb4-512"><a href="#cb4-512" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(timepoint <span class="sc">==</span> <span class="st">"baseline"</span>)</span>
<span id="cb4-513"><a href="#cb4-513" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb4-514"><a href="#cb4-514" aria-hidden="true" tabindex="-1"></a>plots_poetic <span class="ot">&lt;-</span> <span class="fu">sapply</span>(</span>
<span id="cb4-515"><a href="#cb4-515" aria-hidden="true" tabindex="-1"></a>    <span class="fu">c</span>(<span class="st">"is_responder"</span>), </span>
<span id="cb4-516"><a href="#cb4-516" aria-hidden="true" tabindex="-1"></a>    get_plot_new_samples,</span>
<span id="cb4-517"><a href="#cb4-517" aria-hidden="true" tabindex="-1"></a>    <span class="at">name_cohort =</span> <span class="st">"poetic"</span>,</span>
<span id="cb4-518"><a href="#cb4-518" aria-hidden="true" tabindex="-1"></a>    <span class="at">df_pca =</span> poetic_df_pca_baseline,</span>
<span id="cb4-519"><a href="#cb4-519" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC3"</span>,</span>
<span id="cb4-520"><a href="#cb4-520" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC4"</span>,</span>
<span id="cb4-521"><a href="#cb4-521" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"POETIC baseline patients colored by response status"</span>,</span>
<span id="cb4-522"><a href="#cb4-522" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>, </span>
<span id="cb4-523"><a href="#cb4-523" aria-hidden="true" tabindex="-1"></a>    <span class="at">simplify =</span> <span class="cn">FALSE</span></span>
<span id="cb4-524"><a href="#cb4-524" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb4-525"><a href="#cb4-525" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-526"><a href="#cb4-526" aria-hidden="true" tabindex="-1"></a>plots_poetic<span class="sc">$</span>is_responder</span>
<span id="cb4-527"><a href="#cb4-527" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-528"><a href="#cb4-528" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-529"><a href="#cb4-529" aria-hidden="true" tabindex="-1"></a>And when checking the first two components, the POETIC data is closer to</span>
<span id="cb4-530"><a href="#cb4-530" aria-hidden="true" tabindex="-1"></a>METABRIC, which makes sense since both are microarrays, as it can be seen</span>
<span id="cb4-531"><a href="#cb4-531" aria-hidden="true" tabindex="-1"></a>in @fig-poetic-cohort.</span>
<span id="cb4-532"><a href="#cb4-532" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-535"><a href="#cb4-535" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb4-536"><a href="#cb4-536" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-poetic-cohort</span></span>
<span id="cb4-537"><a href="#cb4-537" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Biplot of first two PCA components from POETIC, TCGA, SCANB and </span></span>
<span id="cb4-538"><a href="#cb4-538" aria-hidden="true" tabindex="-1"></a><span class="co">#|     METABRIC. POETIC is highlighted in the plot and has a bigger point.</span></span>
<span id="cb4-539"><a href="#cb4-539" aria-hidden="true" tabindex="-1"></a>poetic_df_pca <span class="sc">%&gt;%</span> </span>
<span id="cb4-540"><a href="#cb4-540" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(</span>
<span id="cb4-541"><a href="#cb4-541" aria-hidden="true" tabindex="-1"></a>        <span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> cohort)</span>
<span id="cb4-542"><a href="#cb4-542" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb4-543"><a href="#cb4-543" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">alpha =</span> cohort, <span class="at">size =</span> cohort)) <span class="sc">+</span></span>
<span id="cb4-544"><a href="#cb4-544" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_size_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(</span>
<span id="cb4-545"><a href="#cb4-545" aria-hidden="true" tabindex="-1"></a>        <span class="st">"poetic"</span> <span class="ot">=</span> <span class="dv">2</span>, </span>
<span id="cb4-546"><a href="#cb4-546" aria-hidden="true" tabindex="-1"></a>        <span class="st">"tcga"</span> <span class="ot">=</span> <span class="dv">1</span>,</span>
<span id="cb4-547"><a href="#cb4-547" aria-hidden="true" tabindex="-1"></a>        <span class="st">"scanb"</span> <span class="ot">=</span> <span class="dv">1</span>,</span>
<span id="cb4-548"><a href="#cb4-548" aria-hidden="true" tabindex="-1"></a>        <span class="st">"metabric"</span> <span class="ot">=</span> <span class="dv">1</span></span>
<span id="cb4-549"><a href="#cb4-549" aria-hidden="true" tabindex="-1"></a>    ), <span class="at">guide =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb4-550"><a href="#cb4-550" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_alpha_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(</span>
<span id="cb4-551"><a href="#cb4-551" aria-hidden="true" tabindex="-1"></a>        <span class="st">"poetic"</span> <span class="ot">=</span> <span class="dv">1</span>, </span>
<span id="cb4-552"><a href="#cb4-552" aria-hidden="true" tabindex="-1"></a>        <span class="st">"tcga"</span> <span class="ot">=</span> <span class="fl">0.1</span>,</span>
<span id="cb4-553"><a href="#cb4-553" aria-hidden="true" tabindex="-1"></a>        <span class="st">"scanb"</span> <span class="ot">=</span> <span class="fl">0.1</span>,</span>
<span id="cb4-554"><a href="#cb4-554" aria-hidden="true" tabindex="-1"></a>        <span class="st">"metabric"</span> <span class="ot">=</span> <span class="fl">0.1</span></span>
<span id="cb4-555"><a href="#cb4-555" aria-hidden="true" tabindex="-1"></a>    ), <span class="at">guide =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb4-556"><a href="#cb4-556" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb4-557"><a href="#cb4-557" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">"Cohort"</span>,</span>
<span id="cb4-558"><a href="#cb4-558" aria-hidden="true" tabindex="-1"></a>        <span class="at">alpha =</span> <span class="st">"Cohort"</span>,</span>
<span id="cb4-559"><a href="#cb4-559" aria-hidden="true" tabindex="-1"></a>        <span class="at">size =</span> <span class="st">"Cohort"</span>,</span>
<span id="cb4-560"><a href="#cb4-560" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="fu">paste0</span>(</span>
<span id="cb4-561"><a href="#cb4-561" aria-hidden="true" tabindex="-1"></a>            <span class="st">"First two components of POETIC overlayed on all the three</span><span class="sc">\n</span><span class="st">"</span>,</span>
<span id="cb4-562"><a href="#cb4-562" aria-hidden="true" tabindex="-1"></a>            <span class="st">"big cohorts in the the molecular landscape"</span></span>
<span id="cb4-563"><a href="#cb4-563" aria-hidden="true" tabindex="-1"></a>        ),</span>
<span id="cb4-564"><a href="#cb4-564" aria-hidden="true" tabindex="-1"></a>        <span class="at">subtitle =</span> <span class="st">"All samples from TCGA, METABRIC and SCANB are plotted"</span></span>
<span id="cb4-565"><a href="#cb4-565" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb4-566"><a href="#cb4-566" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">14</span>) <span class="sc">+</span></span>
<span id="cb4-567"><a href="#cb4-567" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span></span>
<span id="cb4-568"><a href="#cb4-568" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb4-569"><a href="#cb4-569" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-570"><a href="#cb4-570" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-571"><a href="#cb4-571" aria-hidden="true" tabindex="-1"></a><span class="fu">### Differences in PC3 for responders and non responders</span></span>
<span id="cb4-572"><a href="#cb4-572" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-573"><a href="#cb4-573" aria-hidden="true" tabindex="-1"></a>We now also compare the differences in non responders vs responders</span>
<span id="cb4-574"><a href="#cb4-574" aria-hidden="true" tabindex="-1"></a>according to the third component. </span>
<span id="cb4-575"><a href="#cb4-575" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-578"><a href="#cb4-578" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb4-579"><a href="#cb4-579" aria-hidden="true" tabindex="-1"></a>poetic_df_pca_baseline <span class="sc">%&gt;%</span></span>
<span id="cb4-580"><a href="#cb4-580" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"poetic"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-581"><a href="#cb4-581" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> is_responder, <span class="at">y =</span> PC3, <span class="at">color =</span> is_responder)) <span class="sc">+</span></span>
<span id="cb4-582"><a href="#cb4-582" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb4-583"><a href="#cb4-583" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_jitter</span>() <span class="sc">+</span></span>
<span id="cb4-584"><a href="#cb4-584" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Third component score"</span>) <span class="sc">+</span></span>
<span id="cb4-585"><a href="#cb4-585" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>()</span>
<span id="cb4-586"><a href="#cb4-586" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-587"><a href="#cb4-587" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-588"><a href="#cb4-588" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-589"><a href="#cb4-589" aria-hidden="true" tabindex="-1"></a><span class="fu">## Scoring all samples </span></span>
<span id="cb4-590"><a href="#cb4-590" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-591"><a href="#cb4-591" aria-hidden="true" tabindex="-1"></a>In the last chapters scores were showed for TCGA, SCANB and </span>
<span id="cb4-592"><a href="#cb4-592" aria-hidden="true" tabindex="-1"></a>METABRIC. @fig-pca-seterpr shows how ER signaling changes as patients</span>
<span id="cb4-593"><a href="#cb4-593" aria-hidden="true" tabindex="-1"></a>move across the molecular landscape. We intend to use the scores now for the</span>
<span id="cb4-594"><a href="#cb4-594" aria-hidden="true" tabindex="-1"></a>POETIC samples. We first start by calculating using GSVA as well and include</span>
<span id="cb4-595"><a href="#cb4-595" aria-hidden="true" tabindex="-1"></a>in the dataframe with the other cohorts. Before moving to the next chapter,</span>
<span id="cb4-596"><a href="#cb4-596" aria-hidden="true" tabindex="-1"></a>we will analyse and compare the scores across the responders and non </span>
<span id="cb4-597"><a href="#cb4-597" aria-hidden="true" tabindex="-1"></a>responders. </span>
<span id="cb4-598"><a href="#cb4-598" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-599"><a href="#cb4-599" aria-hidden="true" tabindex="-1"></a>@fig-scores-poetic-er shows that when using the scores, it looks like</span>
<span id="cb4-600"><a href="#cb4-600" aria-hidden="true" tabindex="-1"></a>in average at baseline $SET_{ER/PR}$ is lower in non responders than</span>
<span id="cb4-601"><a href="#cb4-601" aria-hidden="true" tabindex="-1"></a>in responders. For the other scores there is no clear difference.</span>
<span id="cb4-602"><a href="#cb4-602" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-603"><a href="#cb4-603" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb4-604"><a href="#cb4-604" aria-hidden="true" tabindex="-1"></a><span class="co"># before we calculate the scores we filter the poetic dataset</span></span>
<span id="cb4-605"><a href="#cb4-605" aria-hidden="true" tabindex="-1"></a>keep_genes <span class="ot">&lt;-</span> <span class="fu">rowMeans</span>(<span class="fu">assay</span>(datasets<span class="sc">$</span>poetic, <span class="st">"normalized_intensity"</span>) <span class="sc">&gt;</span> <span class="fl">6.4</span>)</span>
<span id="cb4-606"><a href="#cb4-606" aria-hidden="true" tabindex="-1"></a>keep_genes <span class="ot">&lt;-</span> keep_genes <span class="sc">&gt;</span> <span class="fl">0.7</span></span>
<span id="cb4-607"><a href="#cb4-607" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-608"><a href="#cb4-608" aria-hidden="true" tabindex="-1"></a>gene_sets <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(</span>
<span id="cb4-609"><a href="#cb4-609" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/surv_analysis_estrogen/gene_sets.rds"</span></span>
<span id="cb4-610"><a href="#cb4-610" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb4-611"><a href="#cb4-611" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-612"><a href="#cb4-612" aria-hidden="true" tabindex="-1"></a>gene_sets <span class="ot">&lt;-</span> <span class="fu">sapply</span>(</span>
<span id="cb4-613"><a href="#cb4-613" aria-hidden="true" tabindex="-1"></a>    gene_sets<span class="sc">$</span>gs_name <span class="sc">%&gt;%</span> unique,</span>
<span id="cb4-614"><a href="#cb4-614" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(x, gene_sets) gene_sets <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(gs_name <span class="sc">==</span> x) <span class="sc">%&gt;%</span> </span>
<span id="cb4-615"><a href="#cb4-615" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">pull</span>(gene_symbol),</span>
<span id="cb4-616"><a href="#cb4-616" aria-hidden="true" tabindex="-1"></a>    <span class="at">gene_sets =</span> gene_sets,</span>
<span id="cb4-617"><a href="#cb4-617" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>,</span>
<span id="cb4-618"><a href="#cb4-618" aria-hidden="true" tabindex="-1"></a>    <span class="at">simplify =</span> <span class="cn">FALSE</span></span>
<span id="cb4-619"><a href="#cb4-619" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb4-620"><a href="#cb4-620" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-621"><a href="#cb4-621" aria-hidden="true" tabindex="-1"></a>gsva_scores <span class="ot">&lt;-</span> <span class="fu">mapply</span>(</span>
<span id="cb4-622"><a href="#cb4-622" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(sum_exp, which_assay, gene_sets){</span>
<span id="cb4-623"><a href="#cb4-623" aria-hidden="true" tabindex="-1"></a>        GSVA<span class="sc">::</span><span class="fu">gsva</span>(</span>
<span id="cb4-624"><a href="#cb4-624" aria-hidden="true" tabindex="-1"></a>            <span class="at">expr =</span> <span class="fu">as.matrix</span>(<span class="fu">assay</span>(sum_exp, which_assay)),</span>
<span id="cb4-625"><a href="#cb4-625" aria-hidden="true" tabindex="-1"></a>            <span class="at">gset.idx.list =</span> gene_sets, </span>
<span id="cb4-626"><a href="#cb4-626" aria-hidden="true" tabindex="-1"></a>            <span class="at">parallel.sz =</span> <span class="dv">20</span></span>
<span id="cb4-627"><a href="#cb4-627" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb4-628"><a href="#cb4-628" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb4-629"><a href="#cb4-629" aria-hidden="true" tabindex="-1"></a>    <span class="at">sum_exp =</span> <span class="fu">list</span>(<span class="at">poetic =</span> datasets<span class="sc">$</span>poetic[keep_genes, ]),</span>
<span id="cb4-630"><a href="#cb4-630" aria-hidden="true" tabindex="-1"></a>    <span class="at">which_assay =</span> <span class="fu">list</span>(<span class="at">poetic =</span> <span class="st">"normalized_intensity"</span>),</span>
<span id="cb4-631"><a href="#cb4-631" aria-hidden="true" tabindex="-1"></a>    <span class="at">MoreArgs =</span> <span class="fu">list</span>(<span class="at">gene_sets =</span> gene_sets),</span>
<span id="cb4-632"><a href="#cb4-632" aria-hidden="true" tabindex="-1"></a>    <span class="at">SIMPLIFY =</span> <span class="cn">FALSE</span>,</span>
<span id="cb4-633"><a href="#cb4-633" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span></span>
<span id="cb4-634"><a href="#cb4-634" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb4-635"><a href="#cb4-635" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-636"><a href="#cb4-636" aria-hidden="true" tabindex="-1"></a>poetic_df_pca[</span>
<span id="cb4-637"><a href="#cb4-637" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(gsva_scores<span class="sc">$</span>poetic), </span>
<span id="cb4-638"><a href="#cb4-638" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames</span>(gsva_scores<span class="sc">$</span>poetic)</span>
<span id="cb4-639"><a href="#cb4-639" aria-hidden="true" tabindex="-1"></a>] <span class="ot">&lt;-</span> gsva_scores<span class="sc">$</span>poetic <span class="sc">%&gt;%</span> t</span>
<span id="cb4-640"><a href="#cb4-640" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-641"><a href="#cb4-641" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb4-642"><a href="#cb4-642" aria-hidden="true" tabindex="-1"></a>    poetic_df_pca,</span>
<span id="cb4-643"><a href="#cb4-643" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/scoring/poetic_df_pca_with_scores.rds"</span></span>
<span id="cb4-644"><a href="#cb4-644" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb4-645"><a href="#cb4-645" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-646"><a href="#cb4-646" aria-hidden="true" tabindex="-1"></a>df_pca_coordinates <span class="ot">&lt;-</span> poetic_df_pca</span>
<span id="cb4-647"><a href="#cb4-647" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-648"><a href="#cb4-648" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-651"><a href="#cb4-651" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb4-652"><a href="#cb4-652" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span>load_at_setup){</span>
<span id="cb4-653"><a href="#cb4-653" aria-hidden="true" tabindex="-1"></a>    df_pca_coordinates <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(</span>
<span id="cb4-654"><a href="#cb4-654" aria-hidden="true" tabindex="-1"></a>        <span class="st">"../results/rds_files/scoring/poetic_df_pca_with_scores.rds"</span></span>
<span id="cb4-655"><a href="#cb4-655" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb4-656"><a href="#cb4-656" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb4-657"><a href="#cb4-657" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-658"><a href="#cb4-658" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-659"><a href="#cb4-659" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=13, fig.height=8}</span></span>
<span id="cb4-660"><a href="#cb4-660" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-scores-poetic-er</span></span>
<span id="cb4-661"><a href="#cb4-661" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Baseline scores for all treated patients in the POETIC trial. </span></span>
<span id="cb4-662"><a href="#cb4-662" aria-hidden="true" tabindex="-1"></a><span class="co">#|     G2M corresponds to a proliferation score, EMT to epithelial to </span></span>
<span id="cb4-663"><a href="#cb4-663" aria-hidden="true" tabindex="-1"></a><span class="co">#|     mesenchymal transition score and the others are related to ER </span></span>
<span id="cb4-664"><a href="#cb4-664" aria-hidden="true" tabindex="-1"></a><span class="co">#|     signaling. </span></span>
<span id="cb4-665"><a href="#cb4-665" aria-hidden="true" tabindex="-1"></a>which_scores <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb4-666"><a href="#cb4-666" aria-hidden="true" tabindex="-1"></a>    <span class="st">"SET_ERPR"</span> <span class="ot">=</span> <span class="st">"SET ER/PR"</span>,</span>
<span id="cb4-667"><a href="#cb4-667" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_ESTROGEN_RESPONSE_EARLY"</span> <span class="ot">=</span> <span class="st">"Estrogen Early"</span>,</span>
<span id="cb4-668"><a href="#cb4-668" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_ESTROGEN_RESPONSE_LATE"</span> <span class="ot">=</span> <span class="st">"Estrogen Late"</span>,</span>
<span id="cb4-669"><a href="#cb4-669" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_G2M_CHECKPOINT"</span> <span class="ot">=</span> <span class="st">"G2M"</span>,</span>
<span id="cb4-670"><a href="#cb4-670" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION"</span> <span class="ot">=</span> <span class="st">"EMT"</span>,</span>
<span id="cb4-671"><a href="#cb4-671" aria-hidden="true" tabindex="-1"></a>    <span class="st">"PC3"</span> <span class="ot">=</span> <span class="st">"PC3"</span></span>
<span id="cb4-672"><a href="#cb4-672" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb4-673"><a href="#cb4-673" aria-hidden="true" tabindex="-1"></a>df_pca_coordinates <span class="sc">%&gt;%</span> </span>
<span id="cb4-674"><a href="#cb4-674" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"poetic"</span> <span class="sc">&amp;</span> timepoint <span class="sc">==</span> <span class="st">"baseline"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-675"><a href="#cb4-675" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(</span>
<span id="cb4-676"><a href="#cb4-676" aria-hidden="true" tabindex="-1"></a>        <span class="fu">names</span>(which_scores),</span>
<span id="cb4-677"><a href="#cb4-677" aria-hidden="true" tabindex="-1"></a>        <span class="at">names_to =</span> <span class="st">"pathway"</span>,</span>
<span id="cb4-678"><a href="#cb4-678" aria-hidden="true" tabindex="-1"></a>        <span class="at">values_to =</span> <span class="st">"score"</span></span>
<span id="cb4-679"><a href="#cb4-679" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb4-680"><a href="#cb4-680" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb4-681"><a href="#cb4-681" aria-hidden="true" tabindex="-1"></a>        <span class="at">pathway =</span> which_scores[pathway]</span>
<span id="cb4-682"><a href="#cb4-682" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb4-683"><a href="#cb4-683" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> is_responder, <span class="at">y =</span> score, <span class="at">fill =</span> is_responder)) <span class="sc">+</span></span>
<span id="cb4-684"><a href="#cb4-684" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_violin</span>(<span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb4-685"><a href="#cb4-685" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_jitter</span>(<span class="at">size =</span> <span class="fl">0.5</span>) <span class="sc">+</span> </span>
<span id="cb4-686"><a href="#cb4-686" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb4-687"><a href="#cb4-687" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>pathway, <span class="at">scale =</span> <span class="st">"free_x"</span>) <span class="sc">+</span></span>
<span id="cb4-688"><a href="#cb4-688" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>) <span class="sc">+</span> </span>
<span id="cb4-689"><a href="#cb4-689" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb4-690"><a href="#cb4-690" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"Is responder?"</span>, <span class="at">y =</span> <span class="st">"Score"</span>, <span class="at">fill =</span> <span class="st">"Is responder?"</span>,</span>
<span id="cb4-691"><a href="#cb4-691" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">"Distribution of some scores for samples at baseline timepoint"</span></span>
<span id="cb4-692"><a href="#cb4-692" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb4-693"><a href="#cb4-693" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span> </span>
<span id="cb4-694"><a href="#cb4-694" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb4-695"><a href="#cb4-695" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-696"><a href="#cb4-696" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-697"><a href="#cb4-697" aria-hidden="true" tabindex="-1"></a>We see that for PC3 and SET ER/PR There are some</span>
<span id="cb4-698"><a href="#cb4-698" aria-hidden="true" tabindex="-1"></a>differences in terms of the responders and non responders.</span>
<span id="cb4-699"><a href="#cb4-699" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-700"><a href="#cb4-700" aria-hidden="true" tabindex="-1"></a>Let us take a closer look at those patients. We start</span>
<span id="cb4-701"><a href="#cb4-701" aria-hidden="true" tabindex="-1"></a>by plotting the scatter plot of PC3 vs SET ER/PR and color</span>
<span id="cb4-702"><a href="#cb4-702" aria-hidden="true" tabindex="-1"></a>by response status defined by the POETIC trial. </span>
<span id="cb4-703"><a href="#cb4-703" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-706"><a href="#cb4-706" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb4-707"><a href="#cb4-707" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-scores-poetic-responders</span></span>
<span id="cb4-708"><a href="#cb4-708" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Comparison of the third component with SET ER/PR scores colored</span></span>
<span id="cb4-709"><a href="#cb4-709" aria-hidden="true" tabindex="-1"></a><span class="co">#|     by response status.</span></span>
<span id="cb4-710"><a href="#cb4-710" aria-hidden="true" tabindex="-1"></a>df_pca_coordinates <span class="sc">%&gt;%</span> </span>
<span id="cb4-711"><a href="#cb4-711" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"poetic"</span> <span class="sc">&amp;</span> timepoint <span class="sc">==</span> <span class="st">"baseline"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-712"><a href="#cb4-712" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(is_responder <span class="sc">!=</span> <span class="st">"not_available"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-713"><a href="#cb4-713" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> PC3, <span class="at">y =</span> SET_ERPR, <span class="at">color =</span> is_responder)) <span class="sc">+</span></span>
<span id="cb4-714"><a href="#cb4-714" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span> </span>
<span id="cb4-715"><a href="#cb4-715" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_smooth</span>(</span>
<span id="cb4-716"><a href="#cb4-716" aria-hidden="true" tabindex="-1"></a>        <span class="at">method =</span> <span class="st">"gam"</span>,</span>
<span id="cb4-717"><a href="#cb4-717" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">"black"</span></span>
<span id="cb4-718"><a href="#cb4-718" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb4-719"><a href="#cb4-719" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>) <span class="sc">+</span> </span>
<span id="cb4-720"><a href="#cb4-720" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb4-721"><a href="#cb4-721" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"PC3"</span>, </span>
<span id="cb4-722"><a href="#cb4-722" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"SET ER/PR"</span>, </span>
<span id="cb4-723"><a href="#cb4-723" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">"Is responder?"</span>,</span>
<span id="cb4-724"><a href="#cb4-724" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="fu">paste0</span>(</span>
<span id="cb4-725"><a href="#cb4-725" aria-hidden="true" tabindex="-1"></a>            <span class="st">"Scores for responders and non</span><span class="sc">\n</span><span class="st">responders of "</span>,</span>
<span id="cb4-726"><a href="#cb4-726" aria-hidden="true" tabindex="-1"></a>            <span class="st">"POETIC trial samples"</span></span>
<span id="cb4-727"><a href="#cb4-727" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb4-728"><a href="#cb4-728" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb4-729"><a href="#cb4-729" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span> </span>
<span id="cb4-730"><a href="#cb4-730" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb4-731"><a href="#cb4-731" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-732"><a href="#cb4-732" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-733"><a href="#cb4-733" aria-hidden="true" tabindex="-1"></a>This figure shows that there is a correlation between</span>
<span id="cb4-734"><a href="#cb4-734" aria-hidden="true" tabindex="-1"></a>third component and SET ER/PR for the POETIC trial data</span>
<span id="cb4-735"><a href="#cb4-735" aria-hidden="true" tabindex="-1"></a>as well as it was already seen from the other cohorts. </span>
<span id="cb4-736"><a href="#cb4-736" aria-hidden="true" tabindex="-1"></a>Moreover, it seems that the responders have higher PC scores</span>
<span id="cb4-737"><a href="#cb4-737" aria-hidden="true" tabindex="-1"></a>in general. We now correlate for each subgroup (responder and non </span>
<span id="cb4-738"><a href="#cb4-738" aria-hidden="true" tabindex="-1"></a>responders) the PC3 and the change in percentage for Ki67.</span>
<span id="cb4-739"><a href="#cb4-739" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-740"><a href="#cb4-740" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=6}</span></span>
<span id="cb4-741"><a href="#cb4-741" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-scores-corr-poetic</span></span>
<span id="cb4-742"><a href="#cb4-742" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Correlation between change in Ki67 versus PC3 stratified by</span></span>
<span id="cb4-743"><a href="#cb4-743" aria-hidden="true" tabindex="-1"></a><span class="co">#|     response group.</span></span>
<span id="cb4-744"><a href="#cb4-744" aria-hidden="true" tabindex="-1"></a>responder_non_responder_base <span class="ot">&lt;-</span> df_pca_coordinates <span class="sc">%&gt;%</span> </span>
<span id="cb4-745"><a href="#cb4-745" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb4-746"><a href="#cb4-746" aria-hidden="true" tabindex="-1"></a>        cohort <span class="sc">==</span> <span class="st">"poetic"</span> <span class="sc">&amp;</span> timepoint <span class="sc">==</span> <span class="st">"baseline"</span> <span class="sc">&amp;</span></span>
<span id="cb4-747"><a href="#cb4-747" aria-hidden="true" tabindex="-1"></a>            is_responder <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"non_responder"</span>, <span class="st">"responder"</span>)</span>
<span id="cb4-748"><a href="#cb4-748" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span> </span>
<span id="cb4-749"><a href="#cb4-749" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">is_responder =</span> <span class="fu">factor</span>(</span>
<span id="cb4-750"><a href="#cb4-750" aria-hidden="true" tabindex="-1"></a>        is_responder, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"responder"</span>, <span class="st">"non_responder"</span>)</span>
<span id="cb4-751"><a href="#cb4-751" aria-hidden="true" tabindex="-1"></a>    ))</span>
<span id="cb4-752"><a href="#cb4-752" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-753"><a href="#cb4-753" aria-hidden="true" tabindex="-1"></a><span class="co"># calculates the correlation and also get the p-values</span></span>
<span id="cb4-754"><a href="#cb4-754" aria-hidden="true" tabindex="-1"></a>cor_tests <span class="ot">&lt;-</span> responder_non_responder_base <span class="sc">%&gt;%</span> </span>
<span id="cb4-755"><a href="#cb4-755" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">nest</span>(<span class="at">data =</span> <span class="sc">-</span>is_responder) <span class="sc">%&gt;%</span></span>
<span id="cb4-756"><a href="#cb4-756" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb4-757"><a href="#cb4-757" aria-hidden="true" tabindex="-1"></a>        <span class="at">test =</span> <span class="fu">map</span>(data, <span class="sc">~</span> <span class="fu">cor.test</span>(.x<span class="sc">$</span>change_ki67, .x<span class="sc">$</span>PC3)),</span>
<span id="cb4-758"><a href="#cb4-758" aria-hidden="true" tabindex="-1"></a>        <span class="at">tidied =</span> <span class="fu">map</span>(test, broom<span class="sc">::</span>tidy)</span>
<span id="cb4-759"><a href="#cb4-759" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span> </span>
<span id="cb4-760"><a href="#cb4-760" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">unnest</span>(tidied) <span class="sc">%&gt;%</span> </span>
<span id="cb4-761"><a href="#cb4-761" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>dplyr<span class="sc">::</span><span class="fu">all_of</span>(<span class="st">"data"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb4-762"><a href="#cb4-762" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb4-763"><a href="#cb4-763" aria-hidden="true" tabindex="-1"></a>        <span class="at">label =</span> <span class="fu">paste0</span>(</span>
<span id="cb4-764"><a href="#cb4-764" aria-hidden="true" tabindex="-1"></a>            <span class="st">"\U03C1: "</span>, </span>
<span id="cb4-765"><a href="#cb4-765" aria-hidden="true" tabindex="-1"></a>            <span class="fu">format</span>(estimate, <span class="at">digits =</span> <span class="dv">2</span>), </span>
<span id="cb4-766"><a href="#cb4-766" aria-hidden="true" tabindex="-1"></a>            <span class="st">"</span><span class="sc">\n</span><span class="st"> p-val: "</span>, </span>
<span id="cb4-767"><a href="#cb4-767" aria-hidden="true" tabindex="-1"></a>            <span class="fu">format</span>(p.value, <span class="at">digits =</span> <span class="dv">2</span>)</span>
<span id="cb4-768"><a href="#cb4-768" aria-hidden="true" tabindex="-1"></a>        ),</span>
<span id="cb4-769"><a href="#cb4-769" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">90</span>, <span class="sc">-</span><span class="dv">20</span>),</span>
<span id="cb4-770"><a href="#cb4-770" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">5</span>, <span class="dv">4</span>)</span>
<span id="cb4-771"><a href="#cb4-771" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb4-772"><a href="#cb4-772" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-773"><a href="#cb4-773" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> responder_non_responder_base <span class="sc">%&gt;%</span></span>
<span id="cb4-774"><a href="#cb4-774" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> change_ki67, <span class="at">y =</span> PC3)) <span class="sc">+</span></span>
<span id="cb4-775"><a href="#cb4-775" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb4-776"><a href="#cb4-776" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>) <span class="sc">+</span> </span>
<span id="cb4-777"><a href="#cb4-777" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>is_responder, <span class="at">scale =</span> <span class="st">"free_x"</span>) <span class="sc">+</span></span>
<span id="cb4-778"><a href="#cb4-778" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb4-779"><a href="#cb4-779" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"Change in Ki67"</span>,</span>
<span id="cb4-780"><a href="#cb4-780" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">"Pearson correlation for each subgroup separately"</span></span>
<span id="cb4-781"><a href="#cb4-781" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb4-782"><a href="#cb4-782" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>) <span class="sc">+</span> </span>
<span id="cb4-783"><a href="#cb4-783" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span> </span>
<span id="cb4-784"><a href="#cb4-784" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb4-785"><a href="#cb4-785" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-786"><a href="#cb4-786" aria-hidden="true" tabindex="-1"></a>p <span class="sc">+</span> ggplot2<span class="sc">::</span><span class="fu">geom_label</span>(</span>
<span id="cb4-787"><a href="#cb4-787" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> cor_tests,</span>
<span id="cb4-788"><a href="#cb4-788" aria-hidden="true" tabindex="-1"></a>    <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> y, <span class="at">label =</span> label)</span>
<span id="cb4-789"><a href="#cb4-789" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb4-790"><a href="#cb4-790" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-791"><a href="#cb4-791" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-792"><a href="#cb4-792" aria-hidden="true" tabindex="-1"></a>Now another metric we can calculate from our dataset is the difference</span>
<span id="cb4-793"><a href="#cb4-793" aria-hidden="true" tabindex="-1"></a>of the embedding positions </span>
<span id="cb4-794"><a href="#cb4-794" aria-hidden="true" tabindex="-1"></a>between the different samples of the same patient. The idea is that if a </span>
<span id="cb4-795"><a href="#cb4-795" aria-hidden="true" tabindex="-1"></a>patient responded well, there were bigger changes as well in the</span>
<span id="cb4-796"><a href="#cb4-796" aria-hidden="true" tabindex="-1"></a>transcriptomics and that is reflected in the embedding. </span>
<span id="cb4-797"><a href="#cb4-797" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-798"><a href="#cb4-798" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=7}</span></span>
<span id="cb4-799"><a href="#cb4-799" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-diff-embedding-poetic</span></span>
<span id="cb4-800"><a href="#cb4-800" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Differences of the embeddings for each patient individually</span></span>
<span id="cb4-801"><a href="#cb4-801" aria-hidden="true" tabindex="-1"></a><span class="co">#|     stratified by response status. Each dot corresponds to a single patient.</span></span>
<span id="cb4-802"><a href="#cb4-802" aria-hidden="true" tabindex="-1"></a>diff_positions_poetic <span class="ot">&lt;-</span> df_pca_coordinates <span class="sc">%&gt;%</span></span>
<span id="cb4-803"><a href="#cb4-803" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"poetic"</span> <span class="sc">&amp;</span> </span>
<span id="cb4-804"><a href="#cb4-804" aria-hidden="true" tabindex="-1"></a>                      is_responder <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"responder"</span>, <span class="st">"non_responder"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb4-805"><a href="#cb4-805" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">pivot_wider</span>(</span>
<span id="cb4-806"><a href="#cb4-806" aria-hidden="true" tabindex="-1"></a>        <span class="at">id_cols =</span> <span class="fu">c</span>(patient_nb, is_responder, change_ki67),</span>
<span id="cb4-807"><a href="#cb4-807" aria-hidden="true" tabindex="-1"></a>        <span class="at">names_from =</span> timepoint, </span>
<span id="cb4-808"><a href="#cb4-808" aria-hidden="true" tabindex="-1"></a>        <span class="at">values_from =</span> <span class="fu">c</span>(PC3, PC4, ki67, SET_ERPR, HALLMARK_G2M_CHECKPOINT)</span>
<span id="cb4-809"><a href="#cb4-809" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb4-810"><a href="#cb4-810" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">summarise</span>(</span>
<span id="cb4-811"><a href="#cb4-811" aria-hidden="true" tabindex="-1"></a>        <span class="at">diff_positions =</span> <span class="fu">sqrt</span>(</span>
<span id="cb4-812"><a href="#cb4-812" aria-hidden="true" tabindex="-1"></a>            (PC3_baseline <span class="sc">-</span> PC3_surgery)<span class="sc">^</span><span class="dv">2</span> <span class="sc">+</span> </span>
<span id="cb4-813"><a href="#cb4-813" aria-hidden="true" tabindex="-1"></a>                (PC4_baseline <span class="sc">-</span> PC4_surgery)<span class="sc">^</span><span class="dv">2</span> </span>
<span id="cb4-814"><a href="#cb4-814" aria-hidden="true" tabindex="-1"></a>        ),</span>
<span id="cb4-815"><a href="#cb4-815" aria-hidden="true" tabindex="-1"></a>        <span class="at">is_responder =</span> is_responder,</span>
<span id="cb4-816"><a href="#cb4-816" aria-hidden="true" tabindex="-1"></a>        <span class="at">change_ki67 =</span> change_ki67,</span>
<span id="cb4-817"><a href="#cb4-817" aria-hidden="true" tabindex="-1"></a>        <span class="at">ki67_baseline =</span> ki67_baseline,</span>
<span id="cb4-818"><a href="#cb4-818" aria-hidden="true" tabindex="-1"></a>        <span class="at">diff_pc4 =</span> PC4_baseline <span class="sc">-</span> PC4_surgery,</span>
<span id="cb4-819"><a href="#cb4-819" aria-hidden="true" tabindex="-1"></a>        <span class="at">SET_ERPR =</span> SET_ERPR_baseline,</span>
<span id="cb4-820"><a href="#cb4-820" aria-hidden="true" tabindex="-1"></a>        <span class="at">G2M =</span> HALLMARK_G2M_CHECKPOINT_baseline,</span>
<span id="cb4-821"><a href="#cb4-821" aria-hidden="true" tabindex="-1"></a>        <span class="at">diff_G2M =</span> HALLMARK_G2M_CHECKPOINT_baseline <span class="sc">-</span> </span>
<span id="cb4-822"><a href="#cb4-822" aria-hidden="true" tabindex="-1"></a>            HALLMARK_G2M_CHECKPOINT_surgery</span>
<span id="cb4-823"><a href="#cb4-823" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb4-824"><a href="#cb4-824" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">is_responder =</span> <span class="fu">factor</span>(</span>
<span id="cb4-825"><a href="#cb4-825" aria-hidden="true" tabindex="-1"></a>        is_responder, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"responder"</span>, <span class="st">"non_responder"</span>)</span>
<span id="cb4-826"><a href="#cb4-826" aria-hidden="true" tabindex="-1"></a>    ))</span>
<span id="cb4-827"><a href="#cb4-827" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-828"><a href="#cb4-828" aria-hidden="true" tabindex="-1"></a>diff_positions_poetic <span class="sc">%&gt;%</span></span>
<span id="cb4-829"><a href="#cb4-829" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> is_responder, <span class="at">y =</span> diff_positions)) <span class="sc">+</span></span>
<span id="cb4-830"><a href="#cb4-830" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb4-831"><a href="#cb4-831" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_jitter</span>() <span class="sc">+</span> </span>
<span id="cb4-832"><a href="#cb4-832" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb4-833"><a href="#cb4-833" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"Is responder?"</span>,</span>
<span id="cb4-834"><a href="#cb4-834" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"Difference in positions of the embedding"</span></span>
<span id="cb4-835"><a href="#cb4-835" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb4-836"><a href="#cb4-836" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>) <span class="sc">+</span></span>
<span id="cb4-837"><a href="#cb4-837" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span></span>
<span id="cb4-838"><a href="#cb4-838" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb4-839"><a href="#cb4-839" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-840"><a href="#cb4-840" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-841"><a href="#cb4-841" aria-hidden="true" tabindex="-1"></a>And now we perform a comparison of these positions by using</span>
<span id="cb4-842"><a href="#cb4-842" aria-hidden="true" tabindex="-1"></a>rstanarm to get the posterior distributions of the differences.</span>
<span id="cb4-843"><a href="#cb4-843" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-846"><a href="#cb4-846" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb4-847"><a href="#cb4-847" aria-hidden="true" tabindex="-1"></a>fit_differences <span class="ot">&lt;-</span> rstanarm<span class="sc">::</span><span class="fu">stan_glm</span>(</span>
<span id="cb4-848"><a href="#cb4-848" aria-hidden="true" tabindex="-1"></a>    <span class="at">formula =</span> diff_positions <span class="sc">~</span> is_responder,</span>
<span id="cb4-849"><a href="#cb4-849" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> diff_positions_poetic, </span>
<span id="cb4-850"><a href="#cb4-850" aria-hidden="true" tabindex="-1"></a>    <span class="at">refresh =</span> <span class="dv">0</span> </span>
<span id="cb4-851"><a href="#cb4-851" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb4-852"><a href="#cb4-852" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-853"><a href="#cb4-853" aria-hidden="true" tabindex="-1"></a>diff_means <span class="ot">&lt;-</span> fit_differences <span class="sc">%&gt;%</span></span>
<span id="cb4-854"><a href="#cb4-854" aria-hidden="true" tabindex="-1"></a>    tidybayes<span class="sc">::</span><span class="fu">spread_draws</span>(is_respondernon_responder)</span>
<span id="cb4-855"><a href="#cb4-855" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-856"><a href="#cb4-856" aria-hidden="true" tabindex="-1"></a>means <span class="ot">&lt;-</span> diff_positions_poetic <span class="sc">%&gt;%</span></span>
<span id="cb4-857"><a href="#cb4-857" aria-hidden="true" tabindex="-1"></a>    modelr<span class="sc">::</span><span class="fu">data_grid</span>(is_responder) <span class="sc">%&gt;%</span></span>
<span id="cb4-858"><a href="#cb4-858" aria-hidden="true" tabindex="-1"></a>    tidybayes<span class="sc">::</span><span class="fu">add_epred_draws</span>(fit_differences)</span>
<span id="cb4-859"><a href="#cb4-859" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-860"><a href="#cb4-860" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-861"><a href="#cb4-861" aria-hidden="true" tabindex="-1"></a>The figure below shows the distribution of the averages of the differences</span>
<span id="cb4-862"><a href="#cb4-862" aria-hidden="true" tabindex="-1"></a>in the embedding of the projections.</span>
<span id="cb4-863"><a href="#cb4-863" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-866"><a href="#cb4-866" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb4-867"><a href="#cb4-867" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-diff-embedding-poetic-averages</span></span>
<span id="cb4-868"><a href="#cb4-868" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Differences of the embeddings for each patient individually</span></span>
<span id="cb4-869"><a href="#cb4-869" aria-hidden="true" tabindex="-1"></a><span class="co">#|     stratified by response status. Each dot corresponds to a single patient.</span></span>
<span id="cb4-870"><a href="#cb4-870" aria-hidden="true" tabindex="-1"></a>means <span class="sc">%&gt;%</span></span>
<span id="cb4-871"><a href="#cb4-871" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> .epred, <span class="at">y =</span> is_responder)) <span class="sc">+</span></span>
<span id="cb4-872"><a href="#cb4-872" aria-hidden="true" tabindex="-1"></a>    tidybayes<span class="sc">::</span><span class="fu">stat_dotsinterval</span>(<span class="at">quantiles =</span> <span class="dv">100</span>) <span class="sc">+</span> </span>
<span id="cb4-873"><a href="#cb4-873" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb4-874"><a href="#cb4-874" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"Average distance of embedding positions"</span>,</span>
<span id="cb4-875"><a href="#cb4-875" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">""</span></span>
<span id="cb4-876"><a href="#cb4-876" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb4-877"><a href="#cb4-877" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb4-878"><a href="#cb4-878" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>()</span>
<span id="cb4-879"><a href="#cb4-879" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-880"><a href="#cb4-880" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-881"><a href="#cb4-881" aria-hidden="true" tabindex="-1"></a>And the figure below shows the posterior distribution of the differences</span>
<span id="cb4-882"><a href="#cb4-882" aria-hidden="true" tabindex="-1"></a>in average.</span>
<span id="cb4-883"><a href="#cb4-883" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-886"><a href="#cb4-886" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb4-887"><a href="#cb4-887" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-diff-embedding-poetic-averages-diff</span></span>
<span id="cb4-888"><a href="#cb4-888" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Differences of the average differences in the embedding for </span></span>
<span id="cb4-889"><a href="#cb4-889" aria-hidden="true" tabindex="-1"></a><span class="co">#|    each patient individually. The comparison made was non responder vs </span></span>
<span id="cb4-890"><a href="#cb4-890" aria-hidden="true" tabindex="-1"></a><span class="co">#|    responder, which means that if non responder has a lower difference in</span></span>
<span id="cb4-891"><a href="#cb4-891" aria-hidden="true" tabindex="-1"></a><span class="co">#|    average, the difference in difference will be negative.</span></span>
<span id="cb4-892"><a href="#cb4-892" aria-hidden="true" tabindex="-1"></a>diff_means <span class="sc">%&gt;%</span></span>
<span id="cb4-893"><a href="#cb4-893" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> is_respondernon_responder)) <span class="sc">+</span></span>
<span id="cb4-894"><a href="#cb4-894" aria-hidden="true" tabindex="-1"></a>    tidybayes<span class="sc">::</span><span class="fu">stat_dotsinterval</span>(<span class="at">quantiles =</span> <span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb4-895"><a href="#cb4-895" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb4-896"><a href="#cb4-896" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb4-897"><a href="#cb4-897" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"Average difference of the embedding differences"</span>,</span>
<span id="cb4-898"><a href="#cb4-898" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"Density"</span>,</span>
<span id="cb4-899"><a href="#cb4-899" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">""</span></span>
<span id="cb4-900"><a href="#cb4-900" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb4-901"><a href="#cb4-901" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>()</span>
<span id="cb4-902"><a href="#cb4-902" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-903"><a href="#cb4-903" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-904"><a href="#cb4-904" aria-hidden="true" tabindex="-1"></a><span class="fu">## Analysing patient neighborhoods</span></span>
<span id="cb4-905"><a href="#cb4-905" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-906"><a href="#cb4-906" aria-hidden="true" tabindex="-1"></a>There are two ways to interpret the embedding. If the patient has</span>
<span id="cb4-907"><a href="#cb4-907" aria-hidden="true" tabindex="-1"></a>ER+ BC and its embedding is close to the ER- BC patients, we can</span>
<span id="cb4-908"><a href="#cb4-908" aria-hidden="true" tabindex="-1"></a>infer that the endocrine response might not work so well, as </span>
<span id="cb4-909"><a href="#cb4-909" aria-hidden="true" tabindex="-1"></a>it was shown in @fig-pca-responders-baseline. The other option</span>
<span id="cb4-910"><a href="#cb4-910" aria-hidden="true" tabindex="-1"></a>is to compare the patient's score to the average score of </span>
<span id="cb4-911"><a href="#cb4-911" aria-hidden="true" tabindex="-1"></a>its neighborhood.</span>
<span id="cb4-912"><a href="#cb4-912" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-913"><a href="#cb4-913" aria-hidden="true" tabindex="-1"></a>What is a neighborhood? When performing the embedding, global</span>
<span id="cb4-914"><a href="#cb4-914" aria-hidden="true" tabindex="-1"></a>structures are not preserved usually. In this sense we only</span>
<span id="cb4-915"><a href="#cb4-915" aria-hidden="true" tabindex="-1"></a>compare patients that are close to each other in an euclidean </span>
<span id="cb4-916"><a href="#cb4-916" aria-hidden="true" tabindex="-1"></a>neighborhood, i.e., we calculate a radius around each sample and</span>
<span id="cb4-917"><a href="#cb4-917" aria-hidden="true" tabindex="-1"></a>see what other samples are within this ball given the euclidean</span>
<span id="cb4-918"><a href="#cb4-918" aria-hidden="true" tabindex="-1"></a>distance.</span>
<span id="cb4-919"><a href="#cb4-919" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-920"><a href="#cb4-920" aria-hidden="true" tabindex="-1"></a>Given a neighborhood, one can calculate the posterior average score</span>
<span id="cb4-921"><a href="#cb4-921" aria-hidden="true" tabindex="-1"></a>of all samples from METABRIC, SCANB and TCGA and use that as an </span>
<span id="cb4-922"><a href="#cb4-922" aria-hidden="true" tabindex="-1"></a>indication of average score that we can compare other patients to.</span>
<span id="cb4-923"><a href="#cb4-923" aria-hidden="true" tabindex="-1"></a>The concept is that if an ER+ BC patient has a much lower ER signaling</span>
<span id="cb4-924"><a href="#cb4-924" aria-hidden="true" tabindex="-1"></a>score, it means the patients will not respond so well to endocrine </span>
<span id="cb4-925"><a href="#cb4-925" aria-hidden="true" tabindex="-1"></a>therapy, as they are very different from their neighboors.</span>
<span id="cb4-926"><a href="#cb4-926" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-927"><a href="#cb4-927" aria-hidden="true" tabindex="-1"></a>To showcase the ability to use the scores and infer results, we </span>
<span id="cb4-928"><a href="#cb4-928" aria-hidden="true" tabindex="-1"></a>use the POETIC trial patients to compare the scores. This cohort</span>
<span id="cb4-929"><a href="#cb4-929" aria-hidden="true" tabindex="-1"></a>is special in the sense that the patients are filtered based on</span>
<span id="cb4-930"><a href="#cb4-930" aria-hidden="true" tabindex="-1"></a>selection criterias, meaning that the patients are clinically</span>
<span id="cb4-931"><a href="#cb4-931" aria-hidden="true" tabindex="-1"></a>similar. </span>
<span id="cb4-932"><a href="#cb4-932" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-933"><a href="#cb4-933" aria-hidden="true" tabindex="-1"></a>Two random patients in very close neighborhoods were selected</span>
<span id="cb4-934"><a href="#cb4-934" aria-hidden="true" tabindex="-1"></a>to showcase the ability of the molecular landscape.</span>
<span id="cb4-935"><a href="#cb4-935" aria-hidden="true" tabindex="-1"></a>The id of the patients are 63 and 236.</span>
<span id="cb4-936"><a href="#cb4-936" aria-hidden="true" tabindex="-1"></a>@tbl-pt63-236 shows the clinical features of these patients. One</span>
<span id="cb4-937"><a href="#cb4-937" aria-hidden="true" tabindex="-1"></a>of them is HER2+. Patient 63 had a higher Ki67 baseline level but</span>
<span id="cb4-938"><a href="#cb4-938" aria-hidden="true" tabindex="-1"></a>a very good response to endocrine therapy. Patient 236 did not </span>
<span id="cb4-939"><a href="#cb4-939" aria-hidden="true" tabindex="-1"></a>respond to endocrine therapy in terms of reduction of Ki67 levels.</span>
<span id="cb4-940"><a href="#cb4-940" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-943"><a href="#cb4-943" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb4-944"><a href="#cb4-944" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: tbl-pt63-236</span></span>
<span id="cb4-945"><a href="#cb4-945" aria-hidden="true" tabindex="-1"></a><span class="co">#| tbl-cap: Clinical features from patients 63 and 236. </span></span>
<span id="cb4-946"><a href="#cb4-946" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-947"><a href="#cb4-947" aria-hidden="true" tabindex="-1"></a>patients <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"treated_nb63_baseline"</span>, <span class="st">"treated_nb236_baseline"</span>)</span>
<span id="cb4-948"><a href="#cb4-948" aria-hidden="true" tabindex="-1"></a>poetic_df_pca <span class="sc">%&gt;%</span> </span>
<span id="cb4-949"><a href="#cb4-949" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span></span>
<span id="cb4-950"><a href="#cb4-950" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(sample_name <span class="sc">%in%</span> patients) <span class="sc">%&gt;%</span></span>
<span id="cb4-951"><a href="#cb4-951" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(</span>
<span id="cb4-952"><a href="#cb4-952" aria-hidden="true" tabindex="-1"></a>        patient_nb, er_status, her2_status, ki67, change_ki67, </span>
<span id="cb4-953"><a href="#cb4-953" aria-hidden="true" tabindex="-1"></a>        ccca_surgery_ki67_2_7</span>
<span id="cb4-954"><a href="#cb4-954" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb4-955"><a href="#cb4-955" aria-hidden="true" tabindex="-1"></a>    kableExtra<span class="sc">::</span><span class="fu">kbl</span>(<span class="at">row.names =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-956"><a href="#cb4-956" aria-hidden="true" tabindex="-1"></a>    kableExtra<span class="sc">::</span><span class="fu">kable_classic</span>()</span>
<span id="cb4-957"><a href="#cb4-957" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-958"><a href="#cb4-958" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-959"><a href="#cb4-959" aria-hidden="true" tabindex="-1"></a>@fig-pt63-236 shows the embedding of two distinct patients </span>
<span id="cb4-960"><a href="#cb4-960" aria-hidden="true" tabindex="-1"></a>in a similar neighborhood.</span>
<span id="cb4-961"><a href="#cb4-961" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-962"><a href="#cb4-962" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=7}</span></span>
<span id="cb4-963"><a href="#cb4-963" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-pt63-236</span></span>
<span id="cb4-964"><a href="#cb4-964" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Plot of two selected patients in the molecular landscape. One</span></span>
<span id="cb4-965"><a href="#cb4-965" aria-hidden="true" tabindex="-1"></a><span class="co">#|     is a responder and the other is a non responder. The patient id numbers</span></span>
<span id="cb4-966"><a href="#cb4-966" aria-hidden="true" tabindex="-1"></a><span class="co">#|     are 63 and 236.</span></span>
<span id="cb4-967"><a href="#cb4-967" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-968"><a href="#cb4-968" aria-hidden="true" tabindex="-1"></a>patients <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"treated_nb63_baseline"</span>, <span class="st">"treated_nb236_baseline"</span>)</span>
<span id="cb4-969"><a href="#cb4-969" aria-hidden="true" tabindex="-1"></a>remove_patients <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(datasets<span class="sc">$</span>poetic <span class="sc">%&gt;%</span> colnames, patients)</span>
<span id="cb4-970"><a href="#cb4-970" aria-hidden="true" tabindex="-1"></a>poetic_df_pca_baseline <span class="ot">&lt;-</span> poetic_df_pca <span class="sc">%&gt;%</span> </span>
<span id="cb4-971"><a href="#cb4-971" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span>(sample_name <span class="sc">%in%</span> remove_patients))</span>
<span id="cb4-972"><a href="#cb4-972" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-973"><a href="#cb4-973" aria-hidden="true" tabindex="-1"></a>base_plot <span class="ot">&lt;-</span> <span class="fu">get_base_plot</span>(</span>
<span id="cb4-974"><a href="#cb4-974" aria-hidden="true" tabindex="-1"></a>    poetic_df_pca <span class="sc">%&gt;%</span> </span>
<span id="cb4-975"><a href="#cb4-975" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"tcga"</span>, <span class="st">"metabric"</span>, <span class="st">"scanb"</span>)),</span>
<span id="cb4-976"><a href="#cb4-976" aria-hidden="true" tabindex="-1"></a>    <span class="at">size_dots =</span> <span class="dv">2</span>,</span>
<span id="cb4-977"><a href="#cb4-977" aria-hidden="true" tabindex="-1"></a>    <span class="at">alpha_val =</span> <span class="fl">0.1</span>,</span>
<span id="cb4-978"><a href="#cb4-978" aria-hidden="true" tabindex="-1"></a>    <span class="at">size_legend =</span> <span class="dv">4</span>,</span>
<span id="cb4-979"><a href="#cb4-979" aria-hidden="true" tabindex="-1"></a>    <span class="at">base_size =</span> <span class="dv">10</span></span>
<span id="cb4-980"><a href="#cb4-980" aria-hidden="true" tabindex="-1"></a>)    </span>
<span id="cb4-981"><a href="#cb4-981" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> poetic_df_pca_baseline <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"poetic"</span>)</span>
<span id="cb4-982"><a href="#cb4-982" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-983"><a href="#cb4-983" aria-hidden="true" tabindex="-1"></a>plot_patients <span class="ot">&lt;-</span> base_plot <span class="sc">+</span> </span>
<span id="cb4-984"><a href="#cb4-984" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>(</span>
<span id="cb4-985"><a href="#cb4-985" aria-hidden="true" tabindex="-1"></a>        <span class="at">data =</span> df,</span>
<span id="cb4-986"><a href="#cb4-986" aria-hidden="true" tabindex="-1"></a>        <span class="at">mapping =</span> <span class="fu">aes</span>(</span>
<span id="cb4-987"><a href="#cb4-987" aria-hidden="true" tabindex="-1"></a>            <span class="at">shape =</span> timepoint,</span>
<span id="cb4-988"><a href="#cb4-988" aria-hidden="true" tabindex="-1"></a>            <span class="at">color =</span> <span class="cn">NA</span>,</span>
<span id="cb4-989"><a href="#cb4-989" aria-hidden="true" tabindex="-1"></a>            <span class="at">fill =</span> is_responder</span>
<span id="cb4-990"><a href="#cb4-990" aria-hidden="true" tabindex="-1"></a>        ),</span>
<span id="cb4-991"><a href="#cb4-991" aria-hidden="true" tabindex="-1"></a>        <span class="at">size =</span> <span class="dv">7</span>,</span>
<span id="cb4-992"><a href="#cb4-992" aria-hidden="true" tabindex="-1"></a>        <span class="at">shape =</span> <span class="dv">22</span></span>
<span id="cb4-993"><a href="#cb4-993" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb4-994"><a href="#cb4-994" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_fill_viridis_d</span>() <span class="sc">+</span></span>
<span id="cb4-995"><a href="#cb4-995" aria-hidden="true" tabindex="-1"></a>    ggrepel<span class="sc">::</span><span class="fu">geom_label_repel</span>(</span>
<span id="cb4-996"><a href="#cb4-996" aria-hidden="true" tabindex="-1"></a>        <span class="at">data =</span> df <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">patient_nb =</span> <span class="fu">paste0</span>(<span class="st">"#"</span>, patient_nb)),</span>
<span id="cb4-997"><a href="#cb4-997" aria-hidden="true" tabindex="-1"></a>        <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> PC3, <span class="at">y =</span> PC4, <span class="at">label =</span> patient_nb),</span>
<span id="cb4-998"><a href="#cb4-998" aria-hidden="true" tabindex="-1"></a>        <span class="at">inherit.aes =</span> <span class="cn">FALSE</span>,</span>
<span id="cb4-999"><a href="#cb4-999" aria-hidden="true" tabindex="-1"></a>        <span class="at">size =</span> <span class="dv">10</span>,</span>
<span id="cb4-1000"><a href="#cb4-1000" aria-hidden="true" tabindex="-1"></a>        <span class="at">box.padding =</span> <span class="fl">0.5</span></span>
<span id="cb4-1001"><a href="#cb4-1001" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb4-1002"><a href="#cb4-1002" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb4-1003"><a href="#cb4-1003" aria-hidden="true" tabindex="-1"></a>        <span class="at">fill =</span> <span class="st">"Is responder?"</span>,</span>
<span id="cb4-1004"><a href="#cb4-1004" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="fu">paste0</span>(</span>
<span id="cb4-1005"><a href="#cb4-1005" aria-hidden="true" tabindex="-1"></a>            <span class="st">"Responder status for each patient in the baseline</span><span class="sc">\n</span><span class="st">"</span>,</span>
<span id="cb4-1006"><a href="#cb4-1006" aria-hidden="true" tabindex="-1"></a>            <span class="st">"and treated group"</span></span>
<span id="cb4-1007"><a href="#cb4-1007" aria-hidden="true" tabindex="-1"></a>        ),</span>
<span id="cb4-1008"><a href="#cb4-1008" aria-hidden="true" tabindex="-1"></a>        <span class="at">caption =</span> <span class="fu">paste0</span>(</span>
<span id="cb4-1009"><a href="#cb4-1009" aria-hidden="true" tabindex="-1"></a>            <span class="st">"Responder is defined as having at least 5% baseline Ki67</span><span class="sc">\n</span><span class="st">and"</span>,</span>
<span id="cb4-1010"><a href="#cb4-1010" aria-hidden="true" tabindex="-1"></a>            <span class="st">" reduction of at least 60% in Ki67 levels"</span></span>
<span id="cb4-1011"><a href="#cb4-1011" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb4-1012"><a href="#cb4-1012" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb4-1013"><a href="#cb4-1013" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">18</span>) <span class="sc">+</span></span>
<span id="cb4-1014"><a href="#cb4-1014" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>()</span>
<span id="cb4-1015"><a href="#cb4-1015" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1016"><a href="#cb4-1016" aria-hidden="true" tabindex="-1"></a>plot_patients</span>
<span id="cb4-1017"><a href="#cb4-1017" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-1018"><a href="#cb4-1018" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1019"><a href="#cb4-1019" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1020"><a href="#cb4-1020" aria-hidden="true" tabindex="-1"></a>And now we calculate the average scores for each patient neighborhoods. We </span>
<span id="cb4-1021"><a href="#cb4-1021" aria-hidden="true" tabindex="-1"></a>start by defining a radius value. After that we select the patients in the</span>
<span id="cb4-1022"><a href="#cb4-1022" aria-hidden="true" tabindex="-1"></a>neighborhood and use <span class="in">`rstanarm`</span> to get the posterior distribution of the</span>
<span id="cb4-1023"><a href="#cb4-1023" aria-hidden="true" tabindex="-1"></a>average score in the neighborhood. For this we use the function</span>
<span id="cb4-1024"><a href="#cb4-1024" aria-hidden="true" tabindex="-1"></a><span class="in">`stan_glm`</span> to calculate the average. This is effective because it can</span>
<span id="cb4-1025"><a href="#cb4-1025" aria-hidden="true" tabindex="-1"></a>be paired with the package <span class="in">`tidybayes`</span> for plotting. It makes extremely easy</span>
<span id="cb4-1026"><a href="#cb4-1026" aria-hidden="true" tabindex="-1"></a>to deal with posterior data. For the average we use a </span>
<span id="cb4-1027"><a href="#cb4-1027" aria-hidden="true" tabindex="-1"></a>normal prior centered at 0 with 1 standard deviation, </span>
<span id="cb4-1028"><a href="#cb4-1028" aria-hidden="true" tabindex="-1"></a>since all scores are between -1 and 1.</span>
<span id="cb4-1029"><a href="#cb4-1029" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1032"><a href="#cb4-1032" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb4-1033"><a href="#cb4-1033" aria-hidden="true" tabindex="-1"></a>scores_to_use <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb4-1034"><a href="#cb4-1034" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Estrogen response early"</span>,</span>
<span id="cb4-1035"><a href="#cb4-1035" aria-hidden="true" tabindex="-1"></a>    <span class="st">"E2F targets"</span>,</span>
<span id="cb4-1036"><a href="#cb4-1036" aria-hidden="true" tabindex="-1"></a>    <span class="st">"P53 pathway"</span>,</span>
<span id="cb4-1037"><a href="#cb4-1037" aria-hidden="true" tabindex="-1"></a>    <span class="st">"EMT"</span>,</span>
<span id="cb4-1038"><a href="#cb4-1038" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Androgen response"</span>,</span>
<span id="cb4-1039"><a href="#cb4-1039" aria-hidden="true" tabindex="-1"></a>    <span class="st">"TGFb signaling"</span>,</span>
<span id="cb4-1040"><a href="#cb4-1040" aria-hidden="true" tabindex="-1"></a>    <span class="st">"SET ER/PR"</span></span>
<span id="cb4-1041"><a href="#cb4-1041" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> <span class="st">`</span><span class="at">names&lt;-</span><span class="st">`</span>(<span class="fu">c</span>(</span>
<span id="cb4-1042"><a href="#cb4-1042" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_ESTROGEN_RESPONSE_EARLY"</span>,</span>
<span id="cb4-1043"><a href="#cb4-1043" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_E2F_TARGETS"</span>,</span>
<span id="cb4-1044"><a href="#cb4-1044" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_P53_PATHWAY"</span>,</span>
<span id="cb4-1045"><a href="#cb4-1045" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION"</span>,</span>
<span id="cb4-1046"><a href="#cb4-1046" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_ANDROGEN_RESPONSE"</span>,</span>
<span id="cb4-1047"><a href="#cb4-1047" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_TGF_BETA_SIGNALING"</span>,</span>
<span id="cb4-1048"><a href="#cb4-1048" aria-hidden="true" tabindex="-1"></a>    <span class="st">"SET_ERPR"</span></span>
<span id="cb4-1049"><a href="#cb4-1049" aria-hidden="true" tabindex="-1"></a>))</span>
<span id="cb4-1050"><a href="#cb4-1050" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-1051"><a href="#cb4-1051" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1052"><a href="#cb4-1052" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1053"><a href="#cb4-1053" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb4-1054"><a href="#cb4-1054" aria-hidden="true" tabindex="-1"></a>patients_all <span class="ot">&lt;-</span> df_pca_coordinates <span class="sc">%&gt;%</span></span>
<span id="cb4-1055"><a href="#cb4-1055" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(timepoint <span class="sc">==</span> <span class="st">"baseline"</span> <span class="sc">&amp;</span> </span>
<span id="cb4-1056"><a href="#cb4-1056" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">grepl</span>(<span class="st">"^treated"</span>, sample_name)) <span class="sc">%&gt;%</span></span>
<span id="cb4-1057"><a href="#cb4-1057" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">pull</span>(sample_name)</span>
<span id="cb4-1058"><a href="#cb4-1058" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1059"><a href="#cb4-1059" aria-hidden="true" tabindex="-1"></a>patients <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"treated_nb63_baseline"</span>, <span class="st">"treated_nb236_baseline"</span>)</span>
<span id="cb4-1060"><a href="#cb4-1060" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1061"><a href="#cb4-1061" aria-hidden="true" tabindex="-1"></a>pipeline_scores_plots <span class="ot">&lt;-</span> <span class="fu">sapply</span>(</span>
<span id="cb4-1062"><a href="#cb4-1062" aria-hidden="true" tabindex="-1"></a>    patients_all, </span>
<span id="cb4-1063"><a href="#cb4-1063" aria-hidden="true" tabindex="-1"></a>    get_patient_scores_distributions_all,</span>
<span id="cb4-1064"><a href="#cb4-1064" aria-hidden="true" tabindex="-1"></a>    <span class="at">df_pca =</span> poetic_df_pca,</span>
<span id="cb4-1065"><a href="#cb4-1065" aria-hidden="true" tabindex="-1"></a>    <span class="at">scores_to_use =</span> scores_to_use,</span>
<span id="cb4-1066"><a href="#cb4-1066" aria-hidden="true" tabindex="-1"></a>    <span class="at">base_size =</span> <span class="dv">20</span>,</span>
<span id="cb4-1067"><a href="#cb4-1067" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>, </span>
<span id="cb4-1068"><a href="#cb4-1068" aria-hidden="true" tabindex="-1"></a>    <span class="at">simplify =</span> <span class="cn">FALSE</span></span>
<span id="cb4-1069"><a href="#cb4-1069" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb4-1070"><a href="#cb4-1070" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1071"><a href="#cb4-1071" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb4-1072"><a href="#cb4-1072" aria-hidden="true" tabindex="-1"></a>    pipeline_scores_plots,</span>
<span id="cb4-1073"><a href="#cb4-1073" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/scoring/pipeline_scores_plots.rds"</span></span>
<span id="cb4-1074"><a href="#cb4-1074" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb4-1075"><a href="#cb4-1075" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-1076"><a href="#cb4-1076" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1077"><a href="#cb4-1077" aria-hidden="true" tabindex="-1"></a>Patient 63 @fig-pt63 is considered a responder. According to </span>
<span id="cb4-1078"><a href="#cb4-1078" aria-hidden="true" tabindex="-1"></a>the scores, when comparing the estrogen early signature to its average </span>
<span id="cb4-1079"><a href="#cb4-1079" aria-hidden="true" tabindex="-1"></a>distribution in the neighborhood, the score is higher.</span>
<span id="cb4-1080"><a href="#cb4-1080" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1081"><a href="#cb4-1081" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.height=7, fig.width=10}</span></span>
<span id="cb4-1082"><a href="#cb4-1082" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-pt63</span></span>
<span id="cb4-1083"><a href="#cb4-1083" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Posterior distribution of the average scores in the neighborhood</span></span>
<span id="cb4-1084"><a href="#cb4-1084" aria-hidden="true" tabindex="-1"></a><span class="co">#|     of patient 236 from POETIC trial. Each dot corresponds to a 1% quantile.</span></span>
<span id="cb4-1085"><a href="#cb4-1085" aria-hidden="true" tabindex="-1"></a>pipeline_scores_plots<span class="sc">$</span>treated_nb63_baseline<span class="sc">$</span>plot <span class="sc">+</span></span>
<span id="cb4-1086"><a href="#cb4-1086" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>()</span>
<span id="cb4-1087"><a href="#cb4-1087" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-1088"><a href="#cb4-1088" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1089"><a href="#cb4-1089" aria-hidden="true" tabindex="-1"></a>Patient 236 @fig-pt236 was considered a non responder. According to </span>
<span id="cb4-1090"><a href="#cb4-1090" aria-hidden="true" tabindex="-1"></a>the scores, when comparing the estrogen early signature to its average </span>
<span id="cb4-1091"><a href="#cb4-1091" aria-hidden="true" tabindex="-1"></a>distribution in the neighborhood, the score is lower, being in the</span>
<span id="cb4-1092"><a href="#cb4-1092" aria-hidden="true" tabindex="-1"></a>5% quantile.</span>
<span id="cb4-1093"><a href="#cb4-1093" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1094"><a href="#cb4-1094" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.height=7, fig.width=10}</span></span>
<span id="cb4-1095"><a href="#cb4-1095" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-pt236</span></span>
<span id="cb4-1096"><a href="#cb4-1096" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Posterior distribution of the average scores in the neighborhood</span></span>
<span id="cb4-1097"><a href="#cb4-1097" aria-hidden="true" tabindex="-1"></a><span class="co">#|     of patient 236 from POETIC trial. Each dot corresponds to a 1% quantile.</span></span>
<span id="cb4-1098"><a href="#cb4-1098" aria-hidden="true" tabindex="-1"></a>pipeline_scores_plots<span class="sc">$</span>treated_nb236_baseline<span class="sc">$</span>plot <span class="sc">+</span></span>
<span id="cb4-1099"><a href="#cb4-1099" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>()</span>
<span id="cb4-1100"><a href="#cb4-1100" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-1101"><a href="#cb4-1101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1102"><a href="#cb4-1102" aria-hidden="true" tabindex="-1"></a>When comparing these two patients, there is also a difference in the androgen</span>
<span id="cb4-1103"><a href="#cb4-1103" aria-hidden="true" tabindex="-1"></a>response score, which could be a reflection of different estrogen signaling.</span>
<span id="cb4-1104"><a href="#cb4-1104" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1105"><a href="#cb4-1105" aria-hidden="true" tabindex="-1"></a><span class="fu">### Analysing all patients</span></span>
<span id="cb4-1106"><a href="#cb4-1106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1107"><a href="#cb4-1107" aria-hidden="true" tabindex="-1"></a>We now try to extend this comparison to something more systematic. Instead</span>
<span id="cb4-1108"><a href="#cb4-1108" aria-hidden="true" tabindex="-1"></a>of looking just for these two patients we will compare the neighborhoods</span>
<span id="cb4-1109"><a href="#cb4-1109" aria-hidden="true" tabindex="-1"></a>for the top responders and </span>
<span id="cb4-1110"><a href="#cb4-1110" aria-hidden="true" tabindex="-1"></a>non-responder patients. The first thing we</span>
<span id="cb4-1111"><a href="#cb4-1111" aria-hidden="true" tabindex="-1"></a>have to have in mind is that some non </span>
<span id="cb4-1112"><a href="#cb4-1112" aria-hidden="true" tabindex="-1"></a>responders they did respond to the drug, just the reduction in Ki67</span>
<span id="cb4-1113"><a href="#cb4-1113" aria-hidden="true" tabindex="-1"></a>was not so big. For example patient 209 has a reduction of 40% in Ki67</span>
<span id="cb4-1114"><a href="#cb4-1114" aria-hidden="true" tabindex="-1"></a>going from 16% to 9%. Still this patient is considered a non-responder.</span>
<span id="cb4-1115"><a href="#cb4-1115" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1118"><a href="#cb4-1118" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb4-1119"><a href="#cb4-1119" aria-hidden="true" tabindex="-1"></a><span class="co"># first we calculate the distance to distributions of the scores</span></span>
<span id="cb4-1120"><a href="#cb4-1120" aria-hidden="true" tabindex="-1"></a>distance_to_distribution <span class="ot">&lt;-</span> <span class="fu">lapply</span>(</span>
<span id="cb4-1121"><a href="#cb4-1121" aria-hidden="true" tabindex="-1"></a>    pipeline_scores_plots,</span>
<span id="cb4-1122"><a href="#cb4-1122" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(patient_data){</span>
<span id="cb4-1123"><a href="#cb4-1123" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb4-1124"><a href="#cb4-1124" aria-hidden="true" tabindex="-1"></a>        <span class="co"># get the draws from the pathway of interest and then subtract </span></span>
<span id="cb4-1125"><a href="#cb4-1125" aria-hidden="true" tabindex="-1"></a>        <span class="co"># the score of the patient. we will then compare the results for</span></span>
<span id="cb4-1126"><a href="#cb4-1126" aria-hidden="true" tabindex="-1"></a>        <span class="co"># responders and non responders</span></span>
<span id="cb4-1127"><a href="#cb4-1127" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb4-1128"><a href="#cb4-1128" aria-hidden="true" tabindex="-1"></a>        <span class="fu">mapply</span>(</span>
<span id="cb4-1129"><a href="#cb4-1129" aria-hidden="true" tabindex="-1"></a>            <span class="cf">function</span>(pathway, pathway_name){</span>
<span id="cb4-1130"><a href="#cb4-1130" aria-hidden="true" tabindex="-1"></a>                score_patient <span class="ot">&lt;-</span> patient_data<span class="sc">$</span>scores_patient <span class="sc">%&gt;%</span></span>
<span id="cb4-1131"><a href="#cb4-1131" aria-hidden="true" tabindex="-1"></a>                    dplyr<span class="sc">::</span><span class="fu">filter</span>(pathway <span class="sc">==</span> pathway_name) <span class="sc">%&gt;%</span></span>
<span id="cb4-1132"><a href="#cb4-1132" aria-hidden="true" tabindex="-1"></a>                    dplyr<span class="sc">::</span><span class="fu">pull</span>(score)</span>
<span id="cb4-1133"><a href="#cb4-1133" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb4-1134"><a href="#cb4-1134" aria-hidden="true" tabindex="-1"></a>                <span class="co"># we split it in two lines here so the code is easier to read</span></span>
<span id="cb4-1135"><a href="#cb4-1135" aria-hidden="true" tabindex="-1"></a>                <span class="co"># afterwards</span></span>
<span id="cb4-1136"><a href="#cb4-1136" aria-hidden="true" tabindex="-1"></a>                scores_neighborhood <span class="ot">&lt;-</span> patient_data<span class="sc">$</span>average_scores_neighborhood</span>
<span id="cb4-1137"><a href="#cb4-1137" aria-hidden="true" tabindex="-1"></a>                scores_neighborhood <span class="ot">&lt;-</span> scores_neighborhood[[pathway]]</span>
<span id="cb4-1138"><a href="#cb4-1138" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb4-1139"><a href="#cb4-1139" aria-hidden="true" tabindex="-1"></a>                scores_neighborhood <span class="sc">%&gt;%</span></span>
<span id="cb4-1140"><a href="#cb4-1140" aria-hidden="true" tabindex="-1"></a>                    tidybayes<span class="sc">::</span><span class="fu">spread_draws</span>(<span class="st">`</span><span class="at">(Intercept)</span><span class="st">`</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-1141"><a href="#cb4-1141" aria-hidden="true" tabindex="-1"></a>                    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">intercept =</span> <span class="st">`</span><span class="at">(Intercept)</span><span class="st">`</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-1142"><a href="#cb4-1142" aria-hidden="true" tabindex="-1"></a>                    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">intercept_diff =</span> score_patient <span class="sc">-</span> intercept) <span class="sc">%&gt;%</span></span>
<span id="cb4-1143"><a href="#cb4-1143" aria-hidden="true" tabindex="-1"></a>                    dplyr<span class="sc">::</span><span class="fu">select</span>(intercept, intercept_diff)       </span>
<span id="cb4-1144"><a href="#cb4-1144" aria-hidden="true" tabindex="-1"></a>            }, </span>
<span id="cb4-1145"><a href="#cb4-1145" aria-hidden="true" tabindex="-1"></a>            <span class="at">pathway =</span> <span class="fu">names</span>(patient_data<span class="sc">$</span>scores_patient<span class="sc">$</span>pathway),</span>
<span id="cb4-1146"><a href="#cb4-1146" aria-hidden="true" tabindex="-1"></a>            <span class="at">pathway_name =</span> patient_data<span class="sc">$</span>scores_patient<span class="sc">$</span>pathway,</span>
<span id="cb4-1147"><a href="#cb4-1147" aria-hidden="true" tabindex="-1"></a>            <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>, </span>
<span id="cb4-1148"><a href="#cb4-1148" aria-hidden="true" tabindex="-1"></a>            <span class="at">SIMPLIFY =</span> <span class="cn">FALSE</span></span>
<span id="cb4-1149"><a href="#cb4-1149" aria-hidden="true" tabindex="-1"></a>        ) <span class="sc">%&gt;%</span> </span>
<span id="cb4-1150"><a href="#cb4-1150" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(<span class="at">.id =</span> <span class="st">"pathway"</span>)</span>
<span id="cb4-1151"><a href="#cb4-1151" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb4-1152"><a href="#cb4-1152" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb4-1153"><a href="#cb4-1153" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1154"><a href="#cb4-1154" aria-hidden="true" tabindex="-1"></a>is_mismatch <span class="ot">&lt;-</span> <span class="fu">lapply</span>(</span>
<span id="cb4-1155"><a href="#cb4-1155" aria-hidden="true" tabindex="-1"></a>    pipeline_scores_plots,</span>
<span id="cb4-1156"><a href="#cb4-1156" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(patient_data){</span>
<span id="cb4-1157"><a href="#cb4-1157" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb4-1158"><a href="#cb4-1158" aria-hidden="true" tabindex="-1"></a>        <span class="co"># get the draws from the pathway of interest and then subtract </span></span>
<span id="cb4-1159"><a href="#cb4-1159" aria-hidden="true" tabindex="-1"></a>        <span class="co"># the score of the patient. we will then compare the results for</span></span>
<span id="cb4-1160"><a href="#cb4-1160" aria-hidden="true" tabindex="-1"></a>        <span class="co"># responders and non responders</span></span>
<span id="cb4-1161"><a href="#cb4-1161" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb4-1162"><a href="#cb4-1162" aria-hidden="true" tabindex="-1"></a>        is_mismatch <span class="ot">&lt;-</span> <span class="fu">mapply</span>(</span>
<span id="cb4-1163"><a href="#cb4-1163" aria-hidden="true" tabindex="-1"></a>            <span class="cf">function</span>(pathway, pathway_name){</span>
<span id="cb4-1164"><a href="#cb4-1164" aria-hidden="true" tabindex="-1"></a>                score_patient <span class="ot">&lt;-</span> patient_data<span class="sc">$</span>scores_patient <span class="sc">%&gt;%</span></span>
<span id="cb4-1165"><a href="#cb4-1165" aria-hidden="true" tabindex="-1"></a>                    dplyr<span class="sc">::</span><span class="fu">filter</span>(pathway <span class="sc">==</span> pathway_name) <span class="sc">%&gt;%</span></span>
<span id="cb4-1166"><a href="#cb4-1166" aria-hidden="true" tabindex="-1"></a>                    dplyr<span class="sc">::</span><span class="fu">pull</span>(score)</span>
<span id="cb4-1167"><a href="#cb4-1167" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb4-1168"><a href="#cb4-1168" aria-hidden="true" tabindex="-1"></a>                <span class="co"># we split it in two lines here so the code is easier to read</span></span>
<span id="cb4-1169"><a href="#cb4-1169" aria-hidden="true" tabindex="-1"></a>                <span class="co"># afterwards</span></span>
<span id="cb4-1170"><a href="#cb4-1170" aria-hidden="true" tabindex="-1"></a>                scores_neighborhood <span class="ot">&lt;-</span> patient_data<span class="sc">$</span>average_scores_neighborhood</span>
<span id="cb4-1171"><a href="#cb4-1171" aria-hidden="true" tabindex="-1"></a>                scores_neighborhood <span class="ot">&lt;-</span> scores_neighborhood[[pathway]]</span>
<span id="cb4-1172"><a href="#cb4-1172" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb4-1173"><a href="#cb4-1173" aria-hidden="true" tabindex="-1"></a>                top_quantiles <span class="ot">&lt;-</span> scores_neighborhood <span class="sc">%&gt;%</span></span>
<span id="cb4-1174"><a href="#cb4-1174" aria-hidden="true" tabindex="-1"></a>                    tidybayes<span class="sc">::</span><span class="fu">spread_draws</span>(<span class="st">`</span><span class="at">(Intercept)</span><span class="st">`</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-1175"><a href="#cb4-1175" aria-hidden="true" tabindex="-1"></a>                    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">intercept =</span> <span class="st">`</span><span class="at">(Intercept)</span><span class="st">`</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-1176"><a href="#cb4-1176" aria-hidden="true" tabindex="-1"></a>                    dplyr<span class="sc">::</span><span class="fu">pull</span>(intercept) <span class="sc">%&gt;%</span></span>
<span id="cb4-1177"><a href="#cb4-1177" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">quantile</span>(., <span class="at">prob =</span> <span class="fu">c</span>(.<span class="dv">05</span>, .<span class="dv">95</span>))</span>
<span id="cb4-1178"><a href="#cb4-1178" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb4-1179"><a href="#cb4-1179" aria-hidden="true" tabindex="-1"></a>                <span class="fu">data.frame</span>(</span>
<span id="cb4-1180"><a href="#cb4-1180" aria-hidden="true" tabindex="-1"></a>                    <span class="at">lower =</span> top_quantiles[<span class="dv">1</span>],</span>
<span id="cb4-1181"><a href="#cb4-1181" aria-hidden="true" tabindex="-1"></a>                    <span class="at">upper =</span> top_quantiles[<span class="dv">2</span>]</span>
<span id="cb4-1182"><a href="#cb4-1182" aria-hidden="true" tabindex="-1"></a>                ) <span class="sc">%&gt;%</span></span>
<span id="cb4-1183"><a href="#cb4-1183" aria-hidden="true" tabindex="-1"></a>                    dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb4-1184"><a href="#cb4-1184" aria-hidden="true" tabindex="-1"></a>                        <span class="at">is_between =</span> dplyr<span class="sc">::</span><span class="fu">between</span>(</span>
<span id="cb4-1185"><a href="#cb4-1185" aria-hidden="true" tabindex="-1"></a>                            <span class="at">x =</span> score_patient,</span>
<span id="cb4-1186"><a href="#cb4-1186" aria-hidden="true" tabindex="-1"></a>                            <span class="at">left =</span> lower,</span>
<span id="cb4-1187"><a href="#cb4-1187" aria-hidden="true" tabindex="-1"></a>                            <span class="at">right =</span> upper</span>
<span id="cb4-1188"><a href="#cb4-1188" aria-hidden="true" tabindex="-1"></a>                        ),</span>
<span id="cb4-1189"><a href="#cb4-1189" aria-hidden="true" tabindex="-1"></a>                        <span class="at">is_above =</span> score_patient <span class="sc">&gt;</span> upper,</span>
<span id="cb4-1190"><a href="#cb4-1190" aria-hidden="true" tabindex="-1"></a>                        <span class="at">is_below =</span> score_patient <span class="sc">&lt;</span> lower</span>
<span id="cb4-1191"><a href="#cb4-1191" aria-hidden="true" tabindex="-1"></a>                    )</span>
<span id="cb4-1192"><a href="#cb4-1192" aria-hidden="true" tabindex="-1"></a>            }, </span>
<span id="cb4-1193"><a href="#cb4-1193" aria-hidden="true" tabindex="-1"></a>            <span class="at">pathway =</span> <span class="fu">names</span>(patient_data<span class="sc">$</span>scores_patient<span class="sc">$</span>pathway),</span>
<span id="cb4-1194"><a href="#cb4-1194" aria-hidden="true" tabindex="-1"></a>            <span class="at">pathway_name =</span> patient_data<span class="sc">$</span>scores_patient<span class="sc">$</span>pathway,</span>
<span id="cb4-1195"><a href="#cb4-1195" aria-hidden="true" tabindex="-1"></a>            <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>, </span>
<span id="cb4-1196"><a href="#cb4-1196" aria-hidden="true" tabindex="-1"></a>            <span class="at">SIMPLIFY =</span> <span class="cn">FALSE</span></span>
<span id="cb4-1197"><a href="#cb4-1197" aria-hidden="true" tabindex="-1"></a>        ) <span class="sc">%&gt;%</span> </span>
<span id="cb4-1198"><a href="#cb4-1198" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(<span class="at">.id =</span> <span class="st">"pathway"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-1199"><a href="#cb4-1199" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">mutate</span>(dplyr<span class="sc">::</span><span class="fu">across</span>(</span>
<span id="cb4-1200"><a href="#cb4-1200" aria-hidden="true" tabindex="-1"></a>                <span class="at">.cols =</span> <span class="fu">c</span>(is_between, is_above, is_below),</span>
<span id="cb4-1201"><a href="#cb4-1201" aria-hidden="true" tabindex="-1"></a>                <span class="at">.fns =</span> as.numeric</span>
<span id="cb4-1202"><a href="#cb4-1202" aria-hidden="true" tabindex="-1"></a>            ))</span>
<span id="cb4-1203"><a href="#cb4-1203" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb4-1204"><a href="#cb4-1204" aria-hidden="true" tabindex="-1"></a>        estrogen_early_above <span class="ot">&lt;-</span> is_mismatch <span class="sc">%&gt;%</span> </span>
<span id="cb4-1205"><a href="#cb4-1205" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">filter</span>(pathway <span class="sc">==</span> <span class="st">"HALLMARK_ESTROGEN_RESPONSE_EARLY"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-1206"><a href="#cb4-1206" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">pull</span>(is_above)</span>
<span id="cb4-1207"><a href="#cb4-1207" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb4-1208"><a href="#cb4-1208" aria-hidden="true" tabindex="-1"></a>        estrogen_early_below <span class="ot">&lt;-</span> is_mismatch <span class="sc">%&gt;%</span> </span>
<span id="cb4-1209"><a href="#cb4-1209" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">filter</span>(pathway <span class="sc">==</span> <span class="st">"HALLMARK_ESTROGEN_RESPONSE_EARLY"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-1210"><a href="#cb4-1210" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">pull</span>(is_below)</span>
<span id="cb4-1211"><a href="#cb4-1211" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb4-1212"><a href="#cb4-1212" aria-hidden="true" tabindex="-1"></a>        set_erpr_above <span class="ot">&lt;-</span> is_mismatch <span class="sc">%&gt;%</span> </span>
<span id="cb4-1213"><a href="#cb4-1213" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">filter</span>(pathway <span class="sc">==</span> <span class="st">"SET_ERPR"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-1214"><a href="#cb4-1214" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">pull</span>(is_above)</span>
<span id="cb4-1215"><a href="#cb4-1215" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb4-1216"><a href="#cb4-1216" aria-hidden="true" tabindex="-1"></a>        <span class="fu">list</span>(</span>
<span id="cb4-1217"><a href="#cb4-1217" aria-hidden="true" tabindex="-1"></a>            <span class="at">mismatch =</span> estrogen_early_above <span class="sc">!=</span> set_erpr_above,</span>
<span id="cb4-1218"><a href="#cb4-1218" aria-hidden="true" tabindex="-1"></a>            <span class="at">is_above =</span> estrogen_early_above,</span>
<span id="cb4-1219"><a href="#cb4-1219" aria-hidden="true" tabindex="-1"></a>            <span class="at">df =</span> is_mismatch,</span>
<span id="cb4-1220"><a href="#cb4-1220" aria-hidden="true" tabindex="-1"></a>            <span class="at">is_below =</span> estrogen_early_below</span>
<span id="cb4-1221"><a href="#cb4-1221" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb4-1222"><a href="#cb4-1222" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb4-1223"><a href="#cb4-1223" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb4-1224"><a href="#cb4-1224" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-1225"><a href="#cb4-1225" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1226"><a href="#cb4-1226" aria-hidden="true" tabindex="-1"></a>We don't select all patients for the comparisons. We select only patients that </span>
<span id="cb4-1227"><a href="#cb4-1227" aria-hidden="true" tabindex="-1"></a>are at least in the HER2 enriched region to the right, so we </span>
<span id="cb4-1228"><a href="#cb4-1228" aria-hidden="true" tabindex="-1"></a>exclude the basal like patients. The reason for this as well is because</span>
<span id="cb4-1229"><a href="#cb4-1229" aria-hidden="true" tabindex="-1"></a>they all either have low Estrogen response early scores or low SET ER/PR.</span>
<span id="cb4-1230"><a href="#cb4-1230" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1233"><a href="#cb4-1233" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb4-1234"><a href="#cb4-1234" aria-hidden="true" tabindex="-1"></a>remove_patients <span class="ot">&lt;-</span> df_pca_coordinates <span class="sc">%&gt;%</span> </span>
<span id="cb4-1235"><a href="#cb4-1235" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb4-1236"><a href="#cb4-1236" aria-hidden="true" tabindex="-1"></a>        cohort <span class="sc">==</span> <span class="st">"poetic"</span> <span class="sc">&amp;</span> </span>
<span id="cb4-1237"><a href="#cb4-1237" aria-hidden="true" tabindex="-1"></a>            PC3 <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">4</span> <span class="sc">&amp;</span> </span>
<span id="cb4-1238"><a href="#cb4-1238" aria-hidden="true" tabindex="-1"></a>            is_responder <span class="sc">==</span> <span class="st">"non_responder"</span> <span class="sc">&amp;</span></span>
<span id="cb4-1239"><a href="#cb4-1239" aria-hidden="true" tabindex="-1"></a>            timepoint <span class="sc">==</span> <span class="st">"baseline"</span></span>
<span id="cb4-1240"><a href="#cb4-1240" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb4-1241"><a href="#cb4-1241" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">pull</span>(sample_name)</span>
<span id="cb4-1242"><a href="#cb4-1242" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb4-1243"><a href="#cb4-1243" aria-hidden="true" tabindex="-1"></a>df_pca_coordinates <span class="sc">%&gt;%</span> </span>
<span id="cb4-1244"><a href="#cb4-1244" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(sample_name <span class="sc">%in%</span> remove_patients) <span class="sc">%&gt;%</span></span>
<span id="cb4-1245"><a href="#cb4-1245" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(dplyr<span class="sc">::</span><span class="fu">all_of</span>(<span class="fu">c</span>(</span>
<span id="cb4-1246"><a href="#cb4-1246" aria-hidden="true" tabindex="-1"></a>        <span class="st">"HALLMARK_ESTROGEN_RESPONSE_EARLY"</span>, <span class="st">"SET_ERPR"</span>, <span class="st">"PC3"</span></span>
<span id="cb4-1247"><a href="#cb4-1247" aria-hidden="true" tabindex="-1"></a>    ))) <span class="sc">%&gt;%</span></span>
<span id="cb4-1248"><a href="#cb4-1248" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(dplyr<span class="sc">::</span><span class="fu">across</span>(<span class="at">.fns =</span> format, <span class="at">digits =</span> <span class="dv">2</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb4-1249"><a href="#cb4-1249" aria-hidden="true" tabindex="-1"></a>    kableExtra<span class="sc">::</span><span class="fu">kbl</span>() <span class="sc">%&gt;%</span></span>
<span id="cb4-1250"><a href="#cb4-1250" aria-hidden="true" tabindex="-1"></a>    kableExtra<span class="sc">::</span><span class="fu">kable_classic</span>(<span class="at">full_width =</span> <span class="cn">FALSE</span>)</span>
<span id="cb4-1251"><a href="#cb4-1251" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-1252"><a href="#cb4-1252" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1255"><a href="#cb4-1255" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb4-1256"><a href="#cb4-1256" aria-hidden="true" tabindex="-1"></a>mismatch_analysis <span class="ot">&lt;-</span> <span class="fu">lapply</span>(</span>
<span id="cb4-1257"><a href="#cb4-1257" aria-hidden="true" tabindex="-1"></a>    is_mismatch,</span>
<span id="cb4-1258"><a href="#cb4-1258" aria-hidden="true" tabindex="-1"></a>    \(x) <span class="fu">c</span>(</span>
<span id="cb4-1259"><a href="#cb4-1259" aria-hidden="true" tabindex="-1"></a>        purrr<span class="sc">::</span><span class="fu">pluck</span>(x, <span class="st">"mismatch"</span>), </span>
<span id="cb4-1260"><a href="#cb4-1260" aria-hidden="true" tabindex="-1"></a>        purrr<span class="sc">::</span><span class="fu">pluck</span>(x, <span class="st">"is_above"</span>),</span>
<span id="cb4-1261"><a href="#cb4-1261" aria-hidden="true" tabindex="-1"></a>        purrr<span class="sc">::</span><span class="fu">pluck</span>(x, <span class="st">"is_below"</span>)</span>
<span id="cb4-1262"><a href="#cb4-1262" aria-hidden="true" tabindex="-1"></a>    )  <span class="sc">%&gt;%</span></span>
<span id="cb4-1263"><a href="#cb4-1263" aria-hidden="true" tabindex="-1"></a>        as.numeric</span>
<span id="cb4-1264"><a href="#cb4-1264" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb4-1265"><a href="#cb4-1265" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(<span class="at">.id =</span> <span class="st">"sample_name"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-1266"><a href="#cb4-1266" aria-hidden="true" tabindex="-1"></a>    t <span class="sc">%&gt;%</span></span>
<span id="cb4-1267"><a href="#cb4-1267" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>(<span class="at">sample_name =</span> <span class="fu">rownames</span>(.)) <span class="sc">%&gt;%</span></span>
<span id="cb4-1268"><a href="#cb4-1268" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">mismatch =</span> <span class="st">"X1"</span>, <span class="at">is_above =</span> <span class="st">"X2"</span>, <span class="at">is_below =</span> <span class="st">"X3"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-1269"><a href="#cb4-1269" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span>(sample_name <span class="sc">%in%</span> remove_patients)) <span class="sc">%&gt;%</span></span>
<span id="cb4-1270"><a href="#cb4-1270" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">inner_join</span>(</span>
<span id="cb4-1271"><a href="#cb4-1271" aria-hidden="true" tabindex="-1"></a>        .,</span>
<span id="cb4-1272"><a href="#cb4-1272" aria-hidden="true" tabindex="-1"></a>        df_pca_coordinates,</span>
<span id="cb4-1273"><a href="#cb4-1273" aria-hidden="true" tabindex="-1"></a>        <span class="at">by =</span> <span class="st">"sample_name"</span></span>
<span id="cb4-1274"><a href="#cb4-1274" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb4-1275"><a href="#cb4-1275" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1276"><a href="#cb4-1276" aria-hidden="true" tabindex="-1"></a>average_distances <span class="ot">&lt;-</span> <span class="fu">lapply</span>(</span>
<span id="cb4-1277"><a href="#cb4-1277" aria-hidden="true" tabindex="-1"></a>    distance_to_distribution, </span>
<span id="cb4-1278"><a href="#cb4-1278" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(x){</span>
<span id="cb4-1279"><a href="#cb4-1279" aria-hidden="true" tabindex="-1"></a>        x <span class="sc">%&gt;%</span></span>
<span id="cb4-1280"><a href="#cb4-1280" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">group_by</span>(pathway) <span class="sc">%&gt;%</span></span>
<span id="cb4-1281"><a href="#cb4-1281" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">summarise</span>(</span>
<span id="cb4-1282"><a href="#cb4-1282" aria-hidden="true" tabindex="-1"></a>                <span class="at">avg_distance =</span> <span class="fu">mean</span>(intercept_diff),</span>
<span id="cb4-1283"><a href="#cb4-1283" aria-hidden="true" tabindex="-1"></a>                <span class="at">position =</span> <span class="fu">mean</span>(intercept)</span>
<span id="cb4-1284"><a href="#cb4-1284" aria-hidden="true" tabindex="-1"></a>            )</span>
<span id="cb4-1285"><a href="#cb4-1285" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb4-1286"><a href="#cb4-1286" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb4-1287"><a href="#cb4-1287" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(<span class="at">.id =</span> <span class="st">"sample_name"</span>)</span>
<span id="cb4-1288"><a href="#cb4-1288" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1289"><a href="#cb4-1289" aria-hidden="true" tabindex="-1"></a>poetic_distances <span class="ot">&lt;-</span> df_pca_coordinates <span class="sc">%&gt;%</span></span>
<span id="cb4-1290"><a href="#cb4-1290" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span>(sample_name <span class="sc">%in%</span> remove_patients)) <span class="sc">%&gt;%</span></span>
<span id="cb4-1291"><a href="#cb4-1291" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">inner_join</span>(., average_distances, <span class="at">by =</span> <span class="st">"sample_name"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-1292"><a href="#cb4-1292" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(is_responder <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"responder"</span>, <span class="st">"non_responder"</span>)) </span>
<span id="cb4-1293"><a href="#cb4-1293" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1294"><a href="#cb4-1294" aria-hidden="true" tabindex="-1"></a>non_responders_sub <span class="ot">&lt;-</span> poetic_distances <span class="sc">%&gt;%</span></span>
<span id="cb4-1295"><a href="#cb4-1295" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(change_ki67 <span class="sc">&gt;=</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-1296"><a href="#cb4-1296" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">pull</span>(sample_name) <span class="sc">%&gt;%</span></span>
<span id="cb4-1297"><a href="#cb4-1297" aria-hidden="true" tabindex="-1"></a>    unique</span>
<span id="cb4-1298"><a href="#cb4-1298" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1299"><a href="#cb4-1299" aria-hidden="true" tabindex="-1"></a>responders_sub <span class="ot">&lt;-</span> poetic_distances <span class="sc">%&gt;%</span></span>
<span id="cb4-1300"><a href="#cb4-1300" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">arrange</span>(change_ki67) <span class="sc">%&gt;%</span></span>
<span id="cb4-1301"><a href="#cb4-1301" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">distinct</span>(sample_name) <span class="sc">%&gt;%</span></span>
<span id="cb4-1302"><a href="#cb4-1302" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">8</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb4-1303"><a href="#cb4-1303" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">pull</span>(sample_name) <span class="sc">%&gt;%</span></span>
<span id="cb4-1304"><a href="#cb4-1304" aria-hidden="true" tabindex="-1"></a>    unique</span>
<span id="cb4-1305"><a href="#cb4-1305" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1306"><a href="#cb4-1306" aria-hidden="true" tabindex="-1"></a>responders_middle <span class="ot">&lt;-</span> poetic_distances <span class="sc">%&gt;%</span></span>
<span id="cb4-1307"><a href="#cb4-1307" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">arrange</span>(change_ki67) <span class="sc">%&gt;%</span></span>
<span id="cb4-1308"><a href="#cb4-1308" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">distinct</span>(sample_name, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-1309"><a href="#cb4-1309" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(is_responder <span class="sc">==</span> <span class="st">"responder"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-1310"><a href="#cb4-1310" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(PC3 <span class="sc">&lt;</span> <span class="dv">0</span> <span class="sc">&amp;</span> PC4 <span class="sc">&gt;</span> <span class="sc">-</span><span class="fl">0.5</span> <span class="sc">&amp;</span> PC4 <span class="sc">&lt;</span> <span class="fl">0.5</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-1311"><a href="#cb4-1311" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">pull</span>(sample_name) <span class="sc">%&gt;%</span></span>
<span id="cb4-1312"><a href="#cb4-1312" aria-hidden="true" tabindex="-1"></a>    unique</span>
<span id="cb4-1313"><a href="#cb4-1313" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1314"><a href="#cb4-1314" aria-hidden="true" tabindex="-1"></a>plot_scores_patient <span class="ot">&lt;-</span> <span class="cf">function</span>(nb_patient, list_patients, distances){</span>
<span id="cb4-1315"><a href="#cb4-1315" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb4-1316"><a href="#cb4-1316" aria-hidden="true" tabindex="-1"></a>    ki67_patient <span class="ot">&lt;-</span> distances <span class="sc">%&gt;%</span></span>
<span id="cb4-1317"><a href="#cb4-1317" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">filter</span>(sample_name <span class="sc">==</span> list_patients[[nb]]) <span class="sc">%&gt;%</span></span>
<span id="cb4-1318"><a href="#cb4-1318" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">select</span>(change_ki67, ki67, PC3, PC4) <span class="sc">%&gt;%</span></span>
<span id="cb4-1319"><a href="#cb4-1319" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">slice_head</span>(<span class="at">n=</span><span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-1320"><a href="#cb4-1320" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">mutate</span>(dplyr<span class="sc">::</span><span class="fu">across</span>(<span class="at">.fns =</span> format, <span class="at">digit =</span> <span class="dv">3</span>))</span>
<span id="cb4-1321"><a href="#cb4-1321" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb4-1322"><a href="#cb4-1322" aria-hidden="true" tabindex="-1"></a>    pipeline_scores_plots[[list_patients[nb]]]<span class="sc">$</span>plot <span class="sc">+</span></span>
<span id="cb4-1323"><a href="#cb4-1323" aria-hidden="true" tabindex="-1"></a>        ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb4-1324"><a href="#cb4-1324" aria-hidden="true" tabindex="-1"></a>            <span class="at">subtitle =</span> <span class="fu">paste0</span>(</span>
<span id="cb4-1325"><a href="#cb4-1325" aria-hidden="true" tabindex="-1"></a>                <span class="st">"Ki67: "</span>, ki67_patient<span class="sc">$</span>ki67, </span>
<span id="cb4-1326"><a href="#cb4-1326" aria-hidden="true" tabindex="-1"></a>                <span class="st">", change Ki67: "</span>, ki67_patient<span class="sc">$</span>change_ki67,</span>
<span id="cb4-1327"><a href="#cb4-1327" aria-hidden="true" tabindex="-1"></a>                <span class="st">", PC3: "</span>, ki67_patient<span class="sc">$</span>PC3, <span class="st">", PC4: "</span>, ki67_patient<span class="sc">$</span>PC4</span>
<span id="cb4-1328"><a href="#cb4-1328" aria-hidden="true" tabindex="-1"></a>            )</span>
<span id="cb4-1329"><a href="#cb4-1329" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb4-1330"><a href="#cb4-1330" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb4-1331"><a href="#cb4-1331" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-1332"><a href="#cb4-1332" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1333"><a href="#cb4-1333" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Non-responders</span></span>
<span id="cb4-1334"><a href="#cb4-1334" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1335"><a href="#cb4-1335" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=7}</span></span>
<span id="cb4-1336"><a href="#cb4-1336" aria-hidden="true" tabindex="-1"></a>nb <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb4-1337"><a href="#cb4-1337" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_scores_patient</span>(nb, non_responders_sub, poetic_distances)</span>
<span id="cb4-1338"><a href="#cb4-1338" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-1339"><a href="#cb4-1339" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1340"><a href="#cb4-1340" aria-hidden="true" tabindex="-1"></a>Estrogen score still close to 0, high proliferation at baseline and higher</span>
<span id="cb4-1341"><a href="#cb4-1341" aria-hidden="true" tabindex="-1"></a>androgen score. But in this case SET ER/PR is lower than average. There is</span>
<span id="cb4-1342"><a href="#cb4-1342" aria-hidden="true" tabindex="-1"></a>a mismatch.</span>
<span id="cb4-1343"><a href="#cb4-1343" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1344"><a href="#cb4-1344" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=7}</span></span>
<span id="cb4-1345"><a href="#cb4-1345" aria-hidden="true" tabindex="-1"></a>nb <span class="ot">&lt;-</span> <span class="dv">2</span></span>
<span id="cb4-1346"><a href="#cb4-1346" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_scores_patient</span>(nb, non_responders_sub, poetic_distances)</span>
<span id="cb4-1347"><a href="#cb4-1347" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-1348"><a href="#cb4-1348" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1349"><a href="#cb4-1349" aria-hidden="true" tabindex="-1"></a>This patient has a score close to 0 still but low proliferation in general. </span>
<span id="cb4-1350"><a href="#cb4-1350" aria-hidden="true" tabindex="-1"></a>This is a case where the EMT signaling is really at average and </span>
<span id="cb4-1351"><a href="#cb4-1351" aria-hidden="true" tabindex="-1"></a>TGFb is super low. Again SET ER/PR is much lower, mismatch with ER </span>
<span id="cb4-1352"><a href="#cb4-1352" aria-hidden="true" tabindex="-1"></a>signaling.</span>
<span id="cb4-1353"><a href="#cb4-1353" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1354"><a href="#cb4-1354" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=7}</span></span>
<span id="cb4-1355"><a href="#cb4-1355" aria-hidden="true" tabindex="-1"></a>nb <span class="ot">&lt;-</span> <span class="dv">3</span></span>
<span id="cb4-1356"><a href="#cb4-1356" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_scores_patient</span>(nb, non_responders_sub, poetic_distances)</span>
<span id="cb4-1357"><a href="#cb4-1357" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-1358"><a href="#cb4-1358" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1359"><a href="#cb4-1359" aria-hidden="true" tabindex="-1"></a>Estrogen signaling close to 0, high androgen and E2F targets and high</span>
<span id="cb4-1360"><a href="#cb4-1360" aria-hidden="true" tabindex="-1"></a>EMT. Intriguing patient, both ER and SET ER/PR are going in </span>
<span id="cb4-1361"><a href="#cb4-1361" aria-hidden="true" tabindex="-1"></a>the same direction and above average. </span>
<span id="cb4-1362"><a href="#cb4-1362" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1363"><a href="#cb4-1363" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=7}</span></span>
<span id="cb4-1364"><a href="#cb4-1364" aria-hidden="true" tabindex="-1"></a>nb <span class="ot">&lt;-</span> <span class="dv">4</span></span>
<span id="cb4-1365"><a href="#cb4-1365" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_scores_patient</span>(nb, non_responders_sub, poetic_distances)</span>
<span id="cb4-1366"><a href="#cb4-1366" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-1367"><a href="#cb4-1367" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1368"><a href="#cb4-1368" aria-hidden="true" tabindex="-1"></a>Low ER signaling and super high EMT with an increase P53 pathway</span>
<span id="cb4-1369"><a href="#cb4-1369" aria-hidden="true" tabindex="-1"></a>expression. Super low ER signaling here in terms of SET ER/PR. </span>
<span id="cb4-1370"><a href="#cb4-1370" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1371"><a href="#cb4-1371" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=7}</span></span>
<span id="cb4-1372"><a href="#cb4-1372" aria-hidden="true" tabindex="-1"></a>nb <span class="ot">&lt;-</span> <span class="dv">5</span></span>
<span id="cb4-1373"><a href="#cb4-1373" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_scores_patient</span>(nb, non_responders_sub, poetic_distances)</span>
<span id="cb4-1374"><a href="#cb4-1374" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-1375"><a href="#cb4-1375" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1376"><a href="#cb4-1376" aria-hidden="true" tabindex="-1"></a>Low ER score, even lower than average and low EMT signaling. Average </span>
<span id="cb4-1377"><a href="#cb4-1377" aria-hidden="true" tabindex="-1"></a>E2F targets and lower than average for SET ER/PR.</span>
<span id="cb4-1378"><a href="#cb4-1378" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1379"><a href="#cb4-1379" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=7}</span></span>
<span id="cb4-1380"><a href="#cb4-1380" aria-hidden="true" tabindex="-1"></a>nb <span class="ot">&lt;-</span> <span class="dv">6</span></span>
<span id="cb4-1381"><a href="#cb4-1381" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_scores_patient</span>(nb, non_responders_sub, poetic_distances)</span>
<span id="cb4-1382"><a href="#cb4-1382" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-1383"><a href="#cb4-1383" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1384"><a href="#cb4-1384" aria-hidden="true" tabindex="-1"></a>This case is a bit different, the ER signaling is higher, E2F targets at</span>
<span id="cb4-1385"><a href="#cb4-1385" aria-hidden="true" tabindex="-1"></a>baseline is not so high, even lower than the average, but in this</span>
<span id="cb4-1386"><a href="#cb4-1386" aria-hidden="true" tabindex="-1"></a>case androgen response is much higher and EMT is extremely high. There is</span>
<span id="cb4-1387"><a href="#cb4-1387" aria-hidden="true" tabindex="-1"></a>a relative mismatch between ER early and SET ER/PR. SET ER/PR is </span>
<span id="cb4-1388"><a href="#cb4-1388" aria-hidden="true" tabindex="-1"></a>relatively small and close to the average.</span>
<span id="cb4-1389"><a href="#cb4-1389" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1390"><a href="#cb4-1390" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=7}</span></span>
<span id="cb4-1391"><a href="#cb4-1391" aria-hidden="true" tabindex="-1"></a>nb <span class="ot">&lt;-</span> <span class="dv">7</span></span>
<span id="cb4-1392"><a href="#cb4-1392" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_scores_patient</span>(nb, non_responders_sub, poetic_distances)</span>
<span id="cb4-1393"><a href="#cb4-1393" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-1394"><a href="#cb4-1394" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1395"><a href="#cb4-1395" aria-hidden="true" tabindex="-1"></a>Androgen score lower than the average and high E2F targets at baseline. </span>
<span id="cb4-1396"><a href="#cb4-1396" aria-hidden="true" tabindex="-1"></a>In this case we see a very small EMT signature. In this case there is </span>
<span id="cb4-1397"><a href="#cb4-1397" aria-hidden="true" tabindex="-1"></a>again a mismatch between ER signaling and SET ER/PR.</span>
<span id="cb4-1398"><a href="#cb4-1398" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1399"><a href="#cb4-1399" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=7}</span></span>
<span id="cb4-1400"><a href="#cb4-1400" aria-hidden="true" tabindex="-1"></a>nb <span class="ot">&lt;-</span> <span class="dv">8</span></span>
<span id="cb4-1401"><a href="#cb4-1401" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_scores_patient</span>(nb, non_responders_sub, poetic_distances)</span>
<span id="cb4-1402"><a href="#cb4-1402" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-1403"><a href="#cb4-1403" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1404"><a href="#cb4-1404" aria-hidden="true" tabindex="-1"></a>This one has a low ER signaling as well and higher E2F targets score. </span>
<span id="cb4-1405"><a href="#cb4-1405" aria-hidden="true" tabindex="-1"></a>Massive difference in SET ER/PR here when comparing to the average.</span>
<span id="cb4-1406"><a href="#cb4-1406" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1407"><a href="#cb4-1407" aria-hidden="true" tabindex="-1"></a>In general all these non responders with no change in Ki67 had a low</span>
<span id="cb4-1408"><a href="#cb4-1408" aria-hidden="true" tabindex="-1"></a>ER signaling score or a mismatch between ER early and SET ER/PR</span>
<span id="cb4-1409"><a href="#cb4-1409" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1410"><a href="#cb4-1410" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Top responders</span></span>
<span id="cb4-1411"><a href="#cb4-1411" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1412"><a href="#cb4-1412" aria-hidden="true" tabindex="-1"></a>We now investigate the top responders with the</span>
<span id="cb4-1413"><a href="#cb4-1413" aria-hidden="true" tabindex="-1"></a>biggest decrease in Ki67.</span>
<span id="cb4-1414"><a href="#cb4-1414" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1415"><a href="#cb4-1415" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=7}</span></span>
<span id="cb4-1416"><a href="#cb4-1416" aria-hidden="true" tabindex="-1"></a>nb <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb4-1417"><a href="#cb4-1417" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_scores_patient</span>(nb, responders_sub, poetic_distances)</span>
<span id="cb4-1418"><a href="#cb4-1418" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-1419"><a href="#cb4-1419" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1420"><a href="#cb4-1420" aria-hidden="true" tabindex="-1"></a>First thing is that this patient had already a positive score</span>
<span id="cb4-1421"><a href="#cb4-1421" aria-hidden="true" tabindex="-1"></a>and a very low Androgen signaling score. Moreover the Ki67</span>
<span id="cb4-1422"><a href="#cb4-1422" aria-hidden="true" tabindex="-1"></a>is low. This is one of the furthest patients to the</span>
<span id="cb4-1423"><a href="#cb4-1423" aria-hidden="true" tabindex="-1"></a>right and still has a low ER signaling score.</span>
<span id="cb4-1424"><a href="#cb4-1424" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1425"><a href="#cb4-1425" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=7}</span></span>
<span id="cb4-1426"><a href="#cb4-1426" aria-hidden="true" tabindex="-1"></a>nb <span class="ot">&lt;-</span> <span class="dv">2</span></span>
<span id="cb4-1427"><a href="#cb4-1427" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_scores_patient</span>(nb, responders_sub, poetic_distances)</span>
<span id="cb4-1428"><a href="#cb4-1428" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-1429"><a href="#cb4-1429" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1430"><a href="#cb4-1430" aria-hidden="true" tabindex="-1"></a>A bit high E2F targets but higher than average of the estrogen</span>
<span id="cb4-1431"><a href="#cb4-1431" aria-hidden="true" tabindex="-1"></a>signaling score. High SET ER/PR.</span>
<span id="cb4-1432"><a href="#cb4-1432" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1433"><a href="#cb4-1433" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=7}</span></span>
<span id="cb4-1434"><a href="#cb4-1434" aria-hidden="true" tabindex="-1"></a>nb <span class="ot">&lt;-</span> <span class="dv">3</span></span>
<span id="cb4-1435"><a href="#cb4-1435" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_scores_patient</span>(nb, responders_sub, poetic_distances)</span>
<span id="cb4-1436"><a href="#cb4-1436" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-1437"><a href="#cb4-1437" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1438"><a href="#cb4-1438" aria-hidden="true" tabindex="-1"></a>Here the ER signaling is above average and androgen response is similar</span>
<span id="cb4-1439"><a href="#cb4-1439" aria-hidden="true" tabindex="-1"></a>to the average. E2F targets is high. Also super high SET ER/PR.</span>
<span id="cb4-1440"><a href="#cb4-1440" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1441"><a href="#cb4-1441" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=7}</span></span>
<span id="cb4-1442"><a href="#cb4-1442" aria-hidden="true" tabindex="-1"></a>nb <span class="ot">&lt;-</span> <span class="dv">4</span></span>
<span id="cb4-1443"><a href="#cb4-1443" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_scores_patient</span>(nb, responders_sub, poetic_distances)</span>
<span id="cb4-1444"><a href="#cb4-1444" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-1445"><a href="#cb4-1445" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1446"><a href="#cb4-1446" aria-hidden="true" tabindex="-1"></a>Very low E2F targets and low Androgen response. ER is about the</span>
<span id="cb4-1447"><a href="#cb4-1447" aria-hidden="true" tabindex="-1"></a>average. SET ER/PR is still positive. This sample is in </span>
<span id="cb4-1448"><a href="#cb4-1448" aria-hidden="true" tabindex="-1"></a>the luminal A region also.</span>
<span id="cb4-1449"><a href="#cb4-1449" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1450"><a href="#cb4-1450" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=7}</span></span>
<span id="cb4-1451"><a href="#cb4-1451" aria-hidden="true" tabindex="-1"></a>nb <span class="ot">&lt;-</span> <span class="dv">5</span></span>
<span id="cb4-1452"><a href="#cb4-1452" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_scores_patient</span>(nb, responders_sub, poetic_distances)</span>
<span id="cb4-1453"><a href="#cb4-1453" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-1454"><a href="#cb4-1454" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1455"><a href="#cb4-1455" aria-hidden="true" tabindex="-1"></a>In this case ER signaling is also above average and E2F targets is </span>
<span id="cb4-1456"><a href="#cb4-1456" aria-hidden="true" tabindex="-1"></a>similar to the average of neighbors. SET ER/PR on average and in the </span>
<span id="cb4-1457"><a href="#cb4-1457" aria-hidden="true" tabindex="-1"></a>expected direction.</span>
<span id="cb4-1458"><a href="#cb4-1458" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1459"><a href="#cb4-1459" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=7}</span></span>
<span id="cb4-1460"><a href="#cb4-1460" aria-hidden="true" tabindex="-1"></a>nb <span class="ot">&lt;-</span> <span class="dv">6</span></span>
<span id="cb4-1461"><a href="#cb4-1461" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_scores_patient</span>(nb, responders_sub, poetic_distances)</span>
<span id="cb4-1462"><a href="#cb4-1462" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-1463"><a href="#cb4-1463" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1464"><a href="#cb4-1464" aria-hidden="true" tabindex="-1"></a>This one is intriguing, ER signaling is extremely low, E2F targets is</span>
<span id="cb4-1465"><a href="#cb4-1465" aria-hidden="true" tabindex="-1"></a>very low as well but the patient responded pretty well. But there is a </span>
<span id="cb4-1466"><a href="#cb4-1466" aria-hidden="true" tabindex="-1"></a>big mismatch between Ki67 and transcriptomics for this patient. Further</span>
<span id="cb4-1467"><a href="#cb4-1467" aria-hidden="true" tabindex="-1"></a>investigation should be performed. Actually probably the samples have mixed</span>
<span id="cb4-1468"><a href="#cb4-1468" aria-hidden="true" tabindex="-1"></a>names. Below are the values of the pathways after treatment:</span>
<span id="cb4-1469"><a href="#cb4-1469" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1472"><a href="#cb4-1472" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb4-1473"><a href="#cb4-1473" aria-hidden="true" tabindex="-1"></a>df_pca_coordinates <span class="sc">%&gt;%</span> </span>
<span id="cb4-1474"><a href="#cb4-1474" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(patient_nb <span class="sc">==</span> <span class="st">"89"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb4-1475"><a href="#cb4-1475" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(</span>
<span id="cb4-1476"><a href="#cb4-1476" aria-hidden="true" tabindex="-1"></a>        ki67,</span>
<span id="cb4-1477"><a href="#cb4-1477" aria-hidden="true" tabindex="-1"></a>        HALLMARK_E2F_TARGETS,</span>
<span id="cb4-1478"><a href="#cb4-1478" aria-hidden="true" tabindex="-1"></a>        SET_ERPR,</span>
<span id="cb4-1479"><a href="#cb4-1479" aria-hidden="true" tabindex="-1"></a>        HALLMARK_ESTROGEN_RESPONSE_EARLY,</span>
<span id="cb4-1480"><a href="#cb4-1480" aria-hidden="true" tabindex="-1"></a>        PC3, </span>
<span id="cb4-1481"><a href="#cb4-1481" aria-hidden="true" tabindex="-1"></a>        PC4</span>
<span id="cb4-1482"><a href="#cb4-1482" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb4-1483"><a href="#cb4-1483" aria-hidden="true" tabindex="-1"></a>    DT<span class="sc">::</span><span class="fu">datatable</span>(<span class="at">options =</span> <span class="fu">list</span>(<span class="at">scrollX =</span> <span class="cn">TRUE</span>))</span>
<span id="cb4-1484"><a href="#cb4-1484" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-1485"><a href="#cb4-1485" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1486"><a href="#cb4-1486" aria-hidden="true" tabindex="-1"></a>It looks super weird that there is a big increase in proliferation based</span>
<span id="cb4-1487"><a href="#cb4-1487" aria-hidden="true" tabindex="-1"></a>on the transcriptomic data considering the reduction in the Ki67 percentage.</span>
<span id="cb4-1488"><a href="#cb4-1488" aria-hidden="true" tabindex="-1"></a>Not only that, ER signaling increases. It is very likely that the sample </span>
<span id="cb4-1489"><a href="#cb4-1489" aria-hidden="true" tabindex="-1"></a>has a mixed label. </span>
<span id="cb4-1490"><a href="#cb4-1490" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1491"><a href="#cb4-1491" aria-hidden="true" tabindex="-1"></a>Moving to the next one.</span>
<span id="cb4-1492"><a href="#cb4-1492" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1493"><a href="#cb4-1493" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=7}</span></span>
<span id="cb4-1494"><a href="#cb4-1494" aria-hidden="true" tabindex="-1"></a>nb <span class="ot">&lt;-</span> <span class="dv">7</span></span>
<span id="cb4-1495"><a href="#cb4-1495" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_scores_patient</span>(nb, responders_sub, poetic_distances)</span>
<span id="cb4-1496"><a href="#cb4-1496" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-1497"><a href="#cb4-1497" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1498"><a href="#cb4-1498" aria-hidden="true" tabindex="-1"></a>This one also is intriguing, estrogen signaling is very low but the patient</span>
<span id="cb4-1499"><a href="#cb4-1499" aria-hidden="true" tabindex="-1"></a>is a responder with a good decrease in Ki67. Below is a table with the values</span>
<span id="cb4-1500"><a href="#cb4-1500" aria-hidden="true" tabindex="-1"></a>for surgery and baseline. We see that there is indeed a decrease in </span>
<span id="cb4-1501"><a href="#cb4-1501" aria-hidden="true" tabindex="-1"></a>proliferation and also in estrogen signaling. Also this sample </span>
<span id="cb4-1502"><a href="#cb4-1502" aria-hidden="true" tabindex="-1"></a>is closer to the normal like and luminal A region.</span>
<span id="cb4-1503"><a href="#cb4-1503" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1506"><a href="#cb4-1506" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb4-1507"><a href="#cb4-1507" aria-hidden="true" tabindex="-1"></a>df_pca_coordinates <span class="sc">%&gt;%</span> </span>
<span id="cb4-1508"><a href="#cb4-1508" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(patient_nb <span class="sc">==</span> <span class="st">"90"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb4-1509"><a href="#cb4-1509" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(</span>
<span id="cb4-1510"><a href="#cb4-1510" aria-hidden="true" tabindex="-1"></a>        PC3, </span>
<span id="cb4-1511"><a href="#cb4-1511" aria-hidden="true" tabindex="-1"></a>        PC4, </span>
<span id="cb4-1512"><a href="#cb4-1512" aria-hidden="true" tabindex="-1"></a>        HALLMARK_ESTROGEN_RESPONSE_EARLY, </span>
<span id="cb4-1513"><a href="#cb4-1513" aria-hidden="true" tabindex="-1"></a>        HALLMARK_E2F_TARGETS, </span>
<span id="cb4-1514"><a href="#cb4-1514" aria-hidden="true" tabindex="-1"></a>        ki67</span>
<span id="cb4-1515"><a href="#cb4-1515" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb4-1516"><a href="#cb4-1516" aria-hidden="true" tabindex="-1"></a>    DT<span class="sc">::</span><span class="fu">datatable</span>(<span class="at">options =</span> <span class="fu">list</span>(<span class="at">scrollX =</span> <span class="cn">TRUE</span>))</span>
<span id="cb4-1517"><a href="#cb4-1517" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-1518"><a href="#cb4-1518" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1519"><a href="#cb4-1519" aria-hidden="true" tabindex="-1"></a>And last patient.</span>
<span id="cb4-1520"><a href="#cb4-1520" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1521"><a href="#cb4-1521" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=7}</span></span>
<span id="cb4-1522"><a href="#cb4-1522" aria-hidden="true" tabindex="-1"></a>nb <span class="ot">&lt;-</span> <span class="dv">8</span></span>
<span id="cb4-1523"><a href="#cb4-1523" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_scores_patient</span>(nb, responders_sub, poetic_distances)</span>
<span id="cb4-1524"><a href="#cb4-1524" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-1525"><a href="#cb4-1525" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1526"><a href="#cb4-1526" aria-hidden="true" tabindex="-1"></a>This patient has a score slightly higher than the average, but still negative.</span>
<span id="cb4-1527"><a href="#cb4-1527" aria-hidden="true" tabindex="-1"></a>Nonetheless, the patient responded well to the therapy. One note is </span>
<span id="cb4-1528"><a href="#cb4-1528" aria-hidden="true" tabindex="-1"></a>that this patient is close to the HER2 enriched region. </span>
<span id="cb4-1529"><a href="#cb4-1529" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1530"><a href="#cb4-1530" aria-hidden="true" tabindex="-1"></a>In general non responders had more mismatches between SET ER/PR and </span>
<span id="cb4-1531"><a href="#cb4-1531" aria-hidden="true" tabindex="-1"></a>below average values than responders.</span>
<span id="cb4-1532"><a href="#cb4-1532" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1533"><a href="#cb4-1533" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Responders in the middle of embedding </span></span>
<span id="cb4-1534"><a href="#cb4-1534" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1535"><a href="#cb4-1535" aria-hidden="true" tabindex="-1"></a>We lastly check the responders that are in the middle of the </span>
<span id="cb4-1536"><a href="#cb4-1536" aria-hidden="true" tabindex="-1"></a>molecular landscape to see how they are. </span>
<span id="cb4-1537"><a href="#cb4-1537" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1538"><a href="#cb4-1538" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=7}</span></span>
<span id="cb4-1539"><a href="#cb4-1539" aria-hidden="true" tabindex="-1"></a>nb <span class="ot">&lt;-</span> <span class="dv">2</span></span>
<span id="cb4-1540"><a href="#cb4-1540" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_scores_patient</span>(nb, responders_middle, poetic_distances)</span>
<span id="cb4-1541"><a href="#cb4-1541" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-1542"><a href="#cb4-1542" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1543"><a href="#cb4-1543" aria-hidden="true" tabindex="-1"></a>Low ER signaling in both cases. This patient</span>
<span id="cb4-1544"><a href="#cb4-1544" aria-hidden="true" tabindex="-1"></a>specifically had very high Ki67 levels at baseline.</span>
<span id="cb4-1545"><a href="#cb4-1545" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1546"><a href="#cb4-1546" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=7}</span></span>
<span id="cb4-1547"><a href="#cb4-1547" aria-hidden="true" tabindex="-1"></a>nb <span class="ot">&lt;-</span> <span class="dv">3</span></span>
<span id="cb4-1548"><a href="#cb4-1548" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_scores_patient</span>(nb, responders_middle, poetic_distances)</span>
<span id="cb4-1549"><a href="#cb4-1549" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-1550"><a href="#cb4-1550" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1551"><a href="#cb4-1551" aria-hidden="true" tabindex="-1"></a>Also ER signaling is low here compared to the average, but a very low</span>
<span id="cb4-1552"><a href="#cb4-1552" aria-hidden="true" tabindex="-1"></a>Ki67 proliferation index at baseline.</span>
<span id="cb4-1553"><a href="#cb4-1553" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1554"><a href="#cb4-1554" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Conclusion</span></span>
<span id="cb4-1555"><a href="#cb4-1555" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1556"><a href="#cb4-1556" aria-hidden="true" tabindex="-1"></a>In general what we can see is that when comparing responders and</span>
<span id="cb4-1557"><a href="#cb4-1557" aria-hidden="true" tabindex="-1"></a>non responders the main difference is in the position of </span>
<span id="cb4-1558"><a href="#cb4-1558" aria-hidden="true" tabindex="-1"></a>the molecular landscape. If we select the top responders and the</span>
<span id="cb4-1559"><a href="#cb4-1559" aria-hidden="true" tabindex="-1"></a>top non-responders in terms of decrease of Ki67 index, we see</span>
<span id="cb4-1560"><a href="#cb4-1560" aria-hidden="true" tabindex="-1"></a>that non responders have way lower third principal </span>
<span id="cb4-1561"><a href="#cb4-1561" aria-hidden="true" tabindex="-1"></a>component (@fig-high-respnon).</span>
<span id="cb4-1562"><a href="#cb4-1562" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1563"><a href="#cb4-1563" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=6, fig.height=5}</span></span>
<span id="cb4-1564"><a href="#cb4-1564" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-high-respnon</span></span>
<span id="cb4-1565"><a href="#cb4-1565" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: PC3 comparison of the top non responders and responders</span></span>
<span id="cb4-1566"><a href="#cb4-1566" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1567"><a href="#cb4-1567" aria-hidden="true" tabindex="-1"></a>df_pca_coordinates <span class="sc">%&gt;%</span> </span>
<span id="cb4-1568"><a href="#cb4-1568" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(sample_name <span class="sc">%in%</span> <span class="fu">c</span>(responders_sub, non_responders_sub)) <span class="sc">%&gt;%</span> </span>
<span id="cb4-1569"><a href="#cb4-1569" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> is_responder, <span class="at">y =</span> PC3, <span class="at">color =</span> is_responder)) <span class="sc">+</span> </span>
<span id="cb4-1570"><a href="#cb4-1570" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_boxplot</span>() <span class="sc">+</span> </span>
<span id="cb4-1571"><a href="#cb4-1571" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_jitter</span>(<span class="at">size =</span> <span class="dv">3</span>, <span class="at">alpha =</span> <span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb4-1572"><a href="#cb4-1572" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_color_viridis_d</span>(<span class="at">option =</span> <span class="st">"H"</span>) <span class="sc">+</span></span>
<span id="cb4-1573"><a href="#cb4-1573" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb4-1574"><a href="#cb4-1574" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"Is responder?"</span>,</span>
<span id="cb4-1575"><a href="#cb4-1575" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">"Top non-responders and responders"</span></span>
<span id="cb4-1576"><a href="#cb4-1576" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb4-1577"><a href="#cb4-1577" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">18</span>) <span class="sc">+</span></span>
<span id="cb4-1578"><a href="#cb4-1578" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb4-1579"><a href="#cb4-1579" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span></span>
<span id="cb4-1580"><a href="#cb4-1580" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb4-1581"><a href="#cb4-1581" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-1582"><a href="#cb4-1582" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1583"><a href="#cb4-1583" aria-hidden="true" tabindex="-1"></a><span class="fu">## Predicting response to endocrine therapy</span></span>
<span id="cb4-1584"><a href="#cb4-1584" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1585"><a href="#cb4-1585" aria-hidden="true" tabindex="-1"></a>We now try to use a penalized logistic regression to predict patient</span>
<span id="cb4-1586"><a href="#cb4-1586" aria-hidden="true" tabindex="-1"></a>response, as defined by the POETIC trialists using the principal</span>
<span id="cb4-1587"><a href="#cb4-1587" aria-hidden="true" tabindex="-1"></a>components. For this we already saw that in average the patients further</span>
<span id="cb4-1588"><a href="#cb4-1588" aria-hidden="true" tabindex="-1"></a>to the left are non-responders and further to the right would be responders,</span>
<span id="cb4-1589"><a href="#cb4-1589" aria-hidden="true" tabindex="-1"></a>suggesting a predictive power of the molecular landscape. The landscape still</span>
<span id="cb4-1590"><a href="#cb4-1590" aria-hidden="true" tabindex="-1"></a>is not perfect as there is a lot of overlap between the responders and </span>
<span id="cb4-1591"><a href="#cb4-1591" aria-hidden="true" tabindex="-1"></a>non-responders. So we try to use more principal components to </span>
<span id="cb4-1592"><a href="#cb4-1592" aria-hidden="true" tabindex="-1"></a>capture the differences. </span>
<span id="cb4-1593"><a href="#cb4-1593" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1594"><a href="#cb4-1594" aria-hidden="true" tabindex="-1"></a>Just when randomizing the train test split, we need to make sure that</span>
<span id="cb4-1595"><a href="#cb4-1595" aria-hidden="true" tabindex="-1"></a>we include some of the samples that are in the basal like region in </span>
<span id="cb4-1596"><a href="#cb4-1596" aria-hidden="true" tabindex="-1"></a>both splits, since they are in small numbers. Due to the small sample</span>
<span id="cb4-1597"><a href="#cb4-1597" aria-hidden="true" tabindex="-1"></a>size, we use the components starting from 3 and going up to 20.</span>
<span id="cb4-1598"><a href="#cb4-1598" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1601"><a href="#cb4-1601" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb4-1602"><a href="#cb4-1602" aria-hidden="true" tabindex="-1"></a>nb_components <span class="ot">&lt;-</span> <span class="dv">3</span><span class="sc">:</span><span class="dv">20</span></span>
<span id="cb4-1603"><a href="#cb4-1603" aria-hidden="true" tabindex="-1"></a>poetic_pcs_response <span class="ot">&lt;-</span> poetic_df_pca <span class="sc">%&gt;%</span></span>
<span id="cb4-1604"><a href="#cb4-1604" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb4-1605"><a href="#cb4-1605" aria-hidden="true" tabindex="-1"></a>        cohort <span class="sc">==</span> <span class="st">"poetic"</span> <span class="sc">&amp;</span> </span>
<span id="cb4-1606"><a href="#cb4-1606" aria-hidden="true" tabindex="-1"></a>            timepoint <span class="sc">==</span> <span class="st">"baseline"</span> <span class="sc">&amp;</span> </span>
<span id="cb4-1607"><a href="#cb4-1607" aria-hidden="true" tabindex="-1"></a>            is_responder <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"responder"</span>, <span class="st">"non_responder"</span>)</span>
<span id="cb4-1608"><a href="#cb4-1608" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb4-1609"><a href="#cb4-1609" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(dplyr<span class="sc">::</span><span class="fu">all_of</span>(<span class="fu">c</span>(</span>
<span id="cb4-1610"><a href="#cb4-1610" aria-hidden="true" tabindex="-1"></a>        <span class="st">"sample_name"</span>,</span>
<span id="cb4-1611"><a href="#cb4-1611" aria-hidden="true" tabindex="-1"></a>        <span class="st">"is_responder"</span>,</span>
<span id="cb4-1612"><a href="#cb4-1612" aria-hidden="true" tabindex="-1"></a>        <span class="fu">paste0</span>(<span class="st">"PC"</span>, nb_components)</span>
<span id="cb4-1613"><a href="#cb4-1613" aria-hidden="true" tabindex="-1"></a>    )))</span>
<span id="cb4-1614"><a href="#cb4-1614" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-1615"><a href="#cb4-1615" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1616"><a href="#cb4-1616" aria-hidden="true" tabindex="-1"></a>In this dataset the total of responders vs non responder is:</span>
<span id="cb4-1617"><a href="#cb4-1617" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1620"><a href="#cb4-1620" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb4-1621"><a href="#cb4-1621" aria-hidden="true" tabindex="-1"></a>poetic_pcs_response <span class="sc">%&gt;%</span> </span>
<span id="cb4-1622"><a href="#cb4-1622" aria-hidden="true" tabindex="-1"></a>    janitor<span class="sc">::</span><span class="fu">tabyl</span>(is_responder) <span class="sc">%&gt;%</span></span>
<span id="cb4-1623"><a href="#cb4-1623" aria-hidden="true" tabindex="-1"></a>    DT<span class="sc">::</span><span class="fu">datatable</span>() <span class="sc">%&gt;%</span></span>
<span id="cb4-1624"><a href="#cb4-1624" aria-hidden="true" tabindex="-1"></a>    DT<span class="sc">::</span><span class="fu">formatPercentage</span>(<span class="st">"percent"</span>)</span>
<span id="cb4-1625"><a href="#cb4-1625" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-1626"><a href="#cb4-1626" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1627"><a href="#cb4-1627" aria-hidden="true" tabindex="-1"></a>More than 2/3 are responders. Using a set seed for the randomization, </span>
<span id="cb4-1628"><a href="#cb4-1628" aria-hidden="true" tabindex="-1"></a>the numbers of responders and non responders are shown below for the training</span>
<span id="cb4-1629"><a href="#cb4-1629" aria-hidden="true" tabindex="-1"></a>set.</span>
<span id="cb4-1630"><a href="#cb4-1630" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1633"><a href="#cb4-1633" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb4-1634"><a href="#cb4-1634" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb4-1635"><a href="#cb4-1635" aria-hidden="true" tabindex="-1"></a>train_poetic <span class="ot">&lt;-</span> <span class="fu">sample</span>(</span>
<span id="cb4-1636"><a href="#cb4-1636" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames</span>(poetic_pcs_response), </span>
<span id="cb4-1637"><a href="#cb4-1637" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="fu">floor</span>(<span class="dv">2</span><span class="sc">/</span><span class="dv">3</span> <span class="sc">*</span> <span class="fu">nrow</span>(poetic_pcs_response))</span>
<span id="cb4-1638"><a href="#cb4-1638" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb4-1639"><a href="#cb4-1639" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1640"><a href="#cb4-1640" aria-hidden="true" tabindex="-1"></a>poetic_pcs_response <span class="sc">%&gt;%</span> </span>
<span id="cb4-1641"><a href="#cb4-1641" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(sample_name <span class="sc">%in%</span> train_poetic) <span class="sc">%&gt;%</span></span>
<span id="cb4-1642"><a href="#cb4-1642" aria-hidden="true" tabindex="-1"></a>    janitor<span class="sc">::</span><span class="fu">tabyl</span>(is_responder) <span class="sc">%&gt;%</span></span>
<span id="cb4-1643"><a href="#cb4-1643" aria-hidden="true" tabindex="-1"></a>    DT<span class="sc">::</span><span class="fu">datatable</span>() <span class="sc">%&gt;%</span></span>
<span id="cb4-1644"><a href="#cb4-1644" aria-hidden="true" tabindex="-1"></a>    DT<span class="sc">::</span><span class="fu">formatPercentage</span>(<span class="st">"percent"</span>)</span>
<span id="cb4-1645"><a href="#cb4-1645" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-1646"><a href="#cb4-1646" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1647"><a href="#cb4-1647" aria-hidden="true" tabindex="-1"></a>And for the test set.</span>
<span id="cb4-1648"><a href="#cb4-1648" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1651"><a href="#cb4-1651" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb4-1652"><a href="#cb4-1652" aria-hidden="true" tabindex="-1"></a>poetic_pcs_response <span class="sc">%&gt;%</span> </span>
<span id="cb4-1653"><a href="#cb4-1653" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span>(sample_name <span class="sc">%in%</span> train_poetic)) <span class="sc">%&gt;%</span></span>
<span id="cb4-1654"><a href="#cb4-1654" aria-hidden="true" tabindex="-1"></a>    janitor<span class="sc">::</span><span class="fu">tabyl</span>(is_responder) <span class="sc">%&gt;%</span></span>
<span id="cb4-1655"><a href="#cb4-1655" aria-hidden="true" tabindex="-1"></a>    DT<span class="sc">::</span><span class="fu">datatable</span>() <span class="sc">%&gt;%</span></span>
<span id="cb4-1656"><a href="#cb4-1656" aria-hidden="true" tabindex="-1"></a>    DT<span class="sc">::</span><span class="fu">formatPercentage</span>(<span class="st">"percent"</span>)</span>
<span id="cb4-1657"><a href="#cb4-1657" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-1658"><a href="#cb4-1658" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1659"><a href="#cb4-1659" aria-hidden="true" tabindex="-1"></a>And we compare the PC3 values distributions to see if they are </span>
<span id="cb4-1660"><a href="#cb4-1660" aria-hidden="true" tabindex="-1"></a>equally distributed.</span>
<span id="cb4-1661"><a href="#cb4-1661" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1664"><a href="#cb4-1664" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb4-1665"><a href="#cb4-1665" aria-hidden="true" tabindex="-1"></a>poetic_pcs_response <span class="ot">&lt;-</span> poetic_pcs_response <span class="sc">%&gt;%</span></span>
<span id="cb4-1666"><a href="#cb4-1666" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">train_or_test =</span> <span class="fu">ifelse</span>(</span>
<span id="cb4-1667"><a href="#cb4-1667" aria-hidden="true" tabindex="-1"></a>        sample_name <span class="sc">%in%</span> train_poetic,</span>
<span id="cb4-1668"><a href="#cb4-1668" aria-hidden="true" tabindex="-1"></a>        <span class="st">"train"</span>,</span>
<span id="cb4-1669"><a href="#cb4-1669" aria-hidden="true" tabindex="-1"></a>        <span class="st">"test"</span></span>
<span id="cb4-1670"><a href="#cb4-1670" aria-hidden="true" tabindex="-1"></a>    ))</span>
<span id="cb4-1671"><a href="#cb4-1671" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1672"><a href="#cb4-1672" aria-hidden="true" tabindex="-1"></a>poetic_pcs_response <span class="sc">%&gt;%</span></span>
<span id="cb4-1673"><a href="#cb4-1673" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(</span>
<span id="cb4-1674"><a href="#cb4-1674" aria-hidden="true" tabindex="-1"></a>        <span class="at">cols =</span> <span class="fu">paste0</span>(<span class="st">"PC"</span>, <span class="dv">3</span><span class="sc">:</span><span class="dv">4</span>),</span>
<span id="cb4-1675"><a href="#cb4-1675" aria-hidden="true" tabindex="-1"></a>        <span class="at">names_to =</span> <span class="st">"pc"</span>,</span>
<span id="cb4-1676"><a href="#cb4-1676" aria-hidden="true" tabindex="-1"></a>        <span class="at">values_to =</span> <span class="st">"position"</span></span>
<span id="cb4-1677"><a href="#cb4-1677" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb4-1678"><a href="#cb4-1678" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> position, <span class="at">fill =</span> train_or_test)) <span class="sc">+</span></span>
<span id="cb4-1679"><a href="#cb4-1679" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_histogram</span>(</span>
<span id="cb4-1680"><a href="#cb4-1680" aria-hidden="true" tabindex="-1"></a>        <span class="at">position =</span> <span class="st">"identity"</span>, </span>
<span id="cb4-1681"><a href="#cb4-1681" aria-hidden="true" tabindex="-1"></a>        <span class="at">alpha =</span> <span class="fl">0.4</span>,</span>
<span id="cb4-1682"><a href="#cb4-1682" aria-hidden="true" tabindex="-1"></a>        <span class="at">bins =</span> <span class="dv">20</span></span>
<span id="cb4-1683"><a href="#cb4-1683" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb4-1684"><a href="#cb4-1684" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>pc, <span class="at">scales =</span> <span class="st">"free_x"</span>) <span class="sc">+</span></span>
<span id="cb4-1685"><a href="#cb4-1685" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>()</span>
<span id="cb4-1686"><a href="#cb4-1686" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-1687"><a href="#cb4-1687" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1688"><a href="#cb4-1688" aria-hidden="true" tabindex="-1"></a>They are similarly distributed, so we can proceed with the training and</span>
<span id="cb4-1689"><a href="#cb4-1689" aria-hidden="true" tabindex="-1"></a>testing strategy. </span>
<span id="cb4-1690"><a href="#cb4-1690" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1693"><a href="#cb4-1693" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb4-1694"><a href="#cb4-1694" aria-hidden="true" tabindex="-1"></a>x_train <span class="ot">&lt;-</span> poetic_pcs_response <span class="sc">%&gt;%</span></span>
<span id="cb4-1695"><a href="#cb4-1695" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(train_or_test <span class="sc">==</span> <span class="st">"train"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-1696"><a href="#cb4-1696" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(dplyr<span class="sc">::</span><span class="fu">all_of</span>(</span>
<span id="cb4-1697"><a href="#cb4-1697" aria-hidden="true" tabindex="-1"></a>        <span class="fu">paste0</span>(<span class="st">"PC"</span>, nb_components)</span>
<span id="cb4-1698"><a href="#cb4-1698" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">%&gt;%</span></span>
<span id="cb4-1699"><a href="#cb4-1699" aria-hidden="true" tabindex="-1"></a>    as.matrix</span>
<span id="cb4-1700"><a href="#cb4-1700" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1701"><a href="#cb4-1701" aria-hidden="true" tabindex="-1"></a>x_test <span class="ot">&lt;-</span> poetic_pcs_response <span class="sc">%&gt;%</span></span>
<span id="cb4-1702"><a href="#cb4-1702" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(train_or_test <span class="sc">==</span> <span class="st">"test"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-1703"><a href="#cb4-1703" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(dplyr<span class="sc">::</span><span class="fu">all_of</span>(</span>
<span id="cb4-1704"><a href="#cb4-1704" aria-hidden="true" tabindex="-1"></a>        <span class="fu">paste0</span>(<span class="st">"PC"</span>, nb_components)</span>
<span id="cb4-1705"><a href="#cb4-1705" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">%&gt;%</span></span>
<span id="cb4-1706"><a href="#cb4-1706" aria-hidden="true" tabindex="-1"></a>    as.matrix</span>
<span id="cb4-1707"><a href="#cb4-1707" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1708"><a href="#cb4-1708" aria-hidden="true" tabindex="-1"></a>y_all <span class="ot">&lt;-</span> poetic_pcs_response <span class="sc">%&gt;%</span></span>
<span id="cb4-1709"><a href="#cb4-1709" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">is_responder =</span> <span class="fu">ifelse</span>(</span>
<span id="cb4-1710"><a href="#cb4-1710" aria-hidden="true" tabindex="-1"></a>        is_responder <span class="sc">==</span> <span class="st">"responder"</span>,</span>
<span id="cb4-1711"><a href="#cb4-1711" aria-hidden="true" tabindex="-1"></a>        <span class="dv">1</span>,</span>
<span id="cb4-1712"><a href="#cb4-1712" aria-hidden="true" tabindex="-1"></a>        <span class="dv">0</span></span>
<span id="cb4-1713"><a href="#cb4-1713" aria-hidden="true" tabindex="-1"></a>    ))</span>
<span id="cb4-1714"><a href="#cb4-1714" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1715"><a href="#cb4-1715" aria-hidden="true" tabindex="-1"></a>y_train <span class="ot">&lt;-</span> y_all <span class="sc">%&gt;%</span></span>
<span id="cb4-1716"><a href="#cb4-1716" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(train_or_test <span class="sc">==</span> <span class="st">"train"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-1717"><a href="#cb4-1717" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">pull</span>(is_responder)</span>
<span id="cb4-1718"><a href="#cb4-1718" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1719"><a href="#cb4-1719" aria-hidden="true" tabindex="-1"></a>y_test <span class="ot">&lt;-</span> y_all <span class="sc">%&gt;%</span></span>
<span id="cb4-1720"><a href="#cb4-1720" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(train_or_test <span class="sc">==</span> <span class="st">"test"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-1721"><a href="#cb4-1721" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">pull</span>(is_responder)</span>
<span id="cb4-1722"><a href="#cb4-1722" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb4-1723"><a href="#cb4-1723" aria-hidden="true" tabindex="-1"></a>fit_responders <span class="ot">&lt;-</span> glmnet<span class="sc">::</span><span class="fu">cv.glmnet</span>(</span>
<span id="cb4-1724"><a href="#cb4-1724" aria-hidden="true" tabindex="-1"></a>    x_train,</span>
<span id="cb4-1725"><a href="#cb4-1725" aria-hidden="true" tabindex="-1"></a>    y_train, </span>
<span id="cb4-1726"><a href="#cb4-1726" aria-hidden="true" tabindex="-1"></a>    <span class="at">family =</span> <span class="st">"binomial"</span>, </span>
<span id="cb4-1727"><a href="#cb4-1727" aria-hidden="true" tabindex="-1"></a>    <span class="at">type.measure =</span> <span class="st">"class"</span></span>
<span id="cb4-1728"><a href="#cb4-1728" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb4-1729"><a href="#cb4-1729" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1730"><a href="#cb4-1730" aria-hidden="true" tabindex="-1"></a>coefs_vals <span class="ot">&lt;-</span> <span class="fu">coef</span>(fit_responders, <span class="at">s =</span> <span class="st">"lambda.1se"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-1731"><a href="#cb4-1731" aria-hidden="true" tabindex="-1"></a>    as.matrix</span>
<span id="cb4-1732"><a href="#cb4-1732" aria-hidden="true" tabindex="-1"></a>coefs_vals <span class="ot">&lt;-</span> coefs_vals[coefs_vals[, <span class="dv">1</span>] <span class="sc">!=</span> <span class="dv">0</span>, ]</span>
<span id="cb4-1733"><a href="#cb4-1733" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-1734"><a href="#cb4-1734" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1735"><a href="#cb4-1735" aria-hidden="true" tabindex="-1"></a>The plot below shows the misclassification error versus the </span>
<span id="cb4-1736"><a href="#cb4-1736" aria-hidden="true" tabindex="-1"></a>lambda used for the penalization term in the logistic regression. </span>
<span id="cb4-1737"><a href="#cb4-1737" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1738"><a href="#cb4-1738" aria-hidden="true" tabindex="-1"></a>The plot below shows the misclassification error for different $\lambda$</span>
<span id="cb4-1739"><a href="#cb4-1739" aria-hidden="true" tabindex="-1"></a>values and number of components.</span>
<span id="cb4-1740"><a href="#cb4-1740" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1743"><a href="#cb4-1743" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb4-1744"><a href="#cb4-1744" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(fit_responders, <span class="at">xlab =</span> <span class="st">"log(lambda)"</span>)</span>
<span id="cb4-1745"><a href="#cb4-1745" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-1746"><a href="#cb4-1746" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1747"><a href="#cb4-1747" aria-hidden="true" tabindex="-1"></a>Overall the misclassification is pretty similar for all number of variables.</span>
<span id="cb4-1748"><a href="#cb4-1748" aria-hidden="true" tabindex="-1"></a>We select the variables obtained with the lambda that is 1 SE away </span>
<span id="cb4-1749"><a href="#cb4-1749" aria-hidden="true" tabindex="-1"></a>from the minimum. So in total we have three variables: </span>
<span id="cb4-1750"><a href="#cb4-1750" aria-hidden="true" tabindex="-1"></a>The intercept and <span class="in">`r names(coefs_vals)[2:length(coefs_vals)]`</span>. </span>
<span id="cb4-1751"><a href="#cb4-1751" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1752"><a href="#cb4-1752" aria-hidden="true" tabindex="-1"></a>And below we show the confusion matrix.</span>
<span id="cb4-1753"><a href="#cb4-1753" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1756"><a href="#cb4-1756" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb4-1757"><a href="#cb4-1757" aria-hidden="true" tabindex="-1"></a>cnf_test <span class="ot">&lt;-</span> glmnet<span class="sc">::</span><span class="fu">confusion.glmnet</span>(</span>
<span id="cb4-1758"><a href="#cb4-1758" aria-hidden="true" tabindex="-1"></a>    fit_responders, </span>
<span id="cb4-1759"><a href="#cb4-1759" aria-hidden="true" tabindex="-1"></a>    <span class="at">newx =</span> x_test, </span>
<span id="cb4-1760"><a href="#cb4-1760" aria-hidden="true" tabindex="-1"></a>    <span class="at">newy =</span> y_test,</span>
<span id="cb4-1761"><a href="#cb4-1761" aria-hidden="true" tabindex="-1"></a>    <span class="at">s =</span> <span class="st">"lambda.1se"</span></span>
<span id="cb4-1762"><a href="#cb4-1762" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb4-1763"><a href="#cb4-1763" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1764"><a href="#cb4-1764" aria-hidden="true" tabindex="-1"></a>cnf_test</span>
<span id="cb4-1765"><a href="#cb4-1765" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-1766"><a href="#cb4-1766" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1767"><a href="#cb4-1767" aria-hidden="true" tabindex="-1"></a>We see that the accuracy is very low and most of the patients are </span>
<span id="cb4-1768"><a href="#cb4-1768" aria-hidden="true" tabindex="-1"></a>classified as responders. This could be potentially due to the</span>
<span id="cb4-1769"><a href="#cb4-1769" aria-hidden="true" tabindex="-1"></a>class imbalance, but the results are still underwhelming. </span>
<span id="cb4-1770"><a href="#cb4-1770" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1771"><a href="#cb4-1771" aria-hidden="true" tabindex="-1"></a>Let us take a look on the patients that were predicted to be</span>
<span id="cb4-1772"><a href="#cb4-1772" aria-hidden="true" tabindex="-1"></a>non responders and check their Ki67 differences in baseline and</span>
<span id="cb4-1773"><a href="#cb4-1773" aria-hidden="true" tabindex="-1"></a>surgery.</span>
<span id="cb4-1774"><a href="#cb4-1774" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1777"><a href="#cb4-1777" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb4-1778"><a href="#cb4-1778" aria-hidden="true" tabindex="-1"></a>y_pred <span class="ot">&lt;-</span> <span class="fu">predict</span>(</span>
<span id="cb4-1779"><a href="#cb4-1779" aria-hidden="true" tabindex="-1"></a>    fit_responders, </span>
<span id="cb4-1780"><a href="#cb4-1780" aria-hidden="true" tabindex="-1"></a>    <span class="at">newx =</span> x_test, </span>
<span id="cb4-1781"><a href="#cb4-1781" aria-hidden="true" tabindex="-1"></a>    <span class="at">type =</span> <span class="st">"class"</span>,</span>
<span id="cb4-1782"><a href="#cb4-1782" aria-hidden="true" tabindex="-1"></a>    <span class="at">lambda =</span> <span class="st">"lambda.1se"</span></span>
<span id="cb4-1783"><a href="#cb4-1783" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb4-1784"><a href="#cb4-1784" aria-hidden="true" tabindex="-1"></a>patients_non_responder <span class="ot">&lt;-</span> <span class="fu">rownames</span>(x_test)[<span class="fu">which</span>(y_pred <span class="sc">==</span> <span class="st">"0"</span>)]</span>
<span id="cb4-1785"><a href="#cb4-1785" aria-hidden="true" tabindex="-1"></a>poetic_df_pca <span class="sc">%&gt;%</span></span>
<span id="cb4-1786"><a href="#cb4-1786" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(sample_name <span class="sc">%in%</span> patients_non_responder) <span class="sc">%&gt;%</span></span>
<span id="cb4-1787"><a href="#cb4-1787" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(PC3, change_ki67, ki67)</span>
<span id="cb4-1788"><a href="#cb4-1788" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb4-1789"><a href="#cb4-1789" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1790"><a href="#cb4-1790" aria-hidden="true" tabindex="-1"></a>All the patients that were misclassified as non responders they have a </span>
<span id="cb4-1791"><a href="#cb4-1791" aria-hidden="true" tabindex="-1"></a>very low third principal component and their baseline Ki67 is quite high, so</span>
<span id="cb4-1792"><a href="#cb4-1792" aria-hidden="true" tabindex="-1"></a>the decrease in proliferation is not actually big enough. It is still</span>
<span id="cb4-1793"><a href="#cb4-1793" aria-hidden="true" tabindex="-1"></a>considered a responder. </span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-right">
      <ul class="footer-items list-unstyled">
    <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/chronchi">
      <i class="bi bi-github" role="img">
</i> 
    </a>
  </li>  
    <li class="nav-item compact">
    <a class="nav-link" href="https://twitter.com/chronchi">
      <i class="bi bi-twitter" role="img">
</i> 
    </a>
  </li>  
</ul>
    </div>
  </div>
</footer>



</body></html>