<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.0.37">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Molecular landscape: a new framework for personalized medicine in Breast Cancer. - 1&nbsp; Estrogen receptor status is continuous, not dichotomous</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./pca_merging.html" rel="next">
<link href="./intro.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>

<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet">
<script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="site_libs/datatables-binding-0.22/datatables.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="site_libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="site_libs/nouislider-7.0.10/jquery.nouislider.min.css" rel="stylesheet">
<script src="site_libs/nouislider-7.0.10/jquery.nouislider.min.js"></script>
<link href="site_libs/selectize-0.12.0/selectize.bootstrap3.css" rel="stylesheet">
<script src="site_libs/selectize-0.12.0/selectize.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet">
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>

  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

</head>

<body class="nav-sidebar floating slimcontent">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
    <div class="container-fluid d-flex justify-content-between">
      <h1 class="quarto-secondary-nav-title"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Estrogen receptor status is continuous, not dichotomous</span></h1>
      <button type="button" class="quarto-btn-toggle btn" aria-label="Show secondary navigation">
        <i class="bi bi-chevron-right"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Molecular landscape: a new framework for personalized medicine in Breast Cancer.</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/chronchi/molecular_landscape" title="Source Code" class="sidebar-tool px-1"><i class="bi bi-github"></i></a>
    <a href="https://twitter.com/intent/tweet?url=|url|" title="Twitter" class="sidebar-tool px-1"><i class="bi bi-twitter"></i></a>
</div>
    </div>
      </div>
      <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">Preface</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./intro.html" class="sidebar-item-text sidebar-link">Introduction</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./surv_analysis_estrogen.html" class="sidebar-item-text sidebar-link active"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Estrogen receptor status is continuous, not dichotomous</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./pca_merging.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">A new framework for personalized medicine: integrating</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./validation.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Validating the molecular landscape</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./scoring.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Pathway activity in a personalized context</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./discussion.html" class="sidebar-item-text sidebar-link">Discussion</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./references.html" class="sidebar-item-text sidebar-link">References</a>
  </div>
</li>
    </ul>
    </div>
</nav>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#loading-and-filtering-the-datasets" id="toc-loading-and-filtering-the-datasets" class="nav-link active" data-scroll-target="#loading-and-filtering-the-datasets"> <span class="header-section-number">1.1</span> Loading and filtering the datasets</a></li>
  <li><a href="#umap-projection-of-the-datasets" id="toc-umap-projection-of-the-datasets" class="nav-link" data-scroll-target="#umap-projection-of-the-datasets"> <span class="header-section-number">1.2</span> UMAP projection of the datasets</a></li>
  <li><a href="#calculating-scores" id="toc-calculating-scores" class="nav-link" data-scroll-target="#calculating-scores"> <span class="header-section-number">1.3</span> Calculating scores</a></li>
  <li><a href="#survival-analysis" id="toc-survival-analysis" class="nav-link" data-scroll-target="#survival-analysis"> <span class="header-section-number">1.4</span> Survival analysis</a>
  <ul class="collapse">
  <li><a href="#results" id="toc-results" class="nav-link" data-scroll-target="#results"> <span class="header-section-number">1.4.1</span> Results</a></li>
  </ul></li>
  <li><a href="#conclusion" id="toc-conclusion" class="nav-link" data-scroll-target="#conclusion"> <span class="header-section-number">1.5</span> Conclusion</a></li>
  </ul>
<div class="toc-actions"><div><i class="bi bi-github"></i></div><div class="action-links"><p><a href="https://github.com/chronchi/molecular_landscape/issues/new" class="toc-action">Report an issue</a></p></div></div></nav>
    </div>
<!-- main -->
<main class="content page-columns page-full" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title d-none d-lg-block"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Estrogen receptor status is continuous, not dichotomous</span></h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
</div>



<div class="quarto-title-meta">

    
    
  </div>
  

</header>

<p>Currently in the clinics estrogen receptor (ER) status is treated as dichotomous condition. Either a breast cancer (BC) is ER positive (ER+) or ER negative (ER-). The threshold for ER+ cells usually is 1% or 10% of the cells positive in a IHC staining. The idea is that ER+ BC patients will receive endocrine therapy, usually tamoxifen or some aromatase inhibitor, for treatment. The problem is not all patients respond the same to these drugs and they also have different proportions of ER+ cells.</p>
<p>In this document we show how leveraging molecular information distinguishes ER+ BC patients and how one should look more carefully on ER status. In order to do this, estrogen related signatures: estrogen response early and late from MSigDB<span class="citation" data-cites="Subramanian2005">(<a href="references.html#ref-Subramanian2005" role="doc-biblioref">Subramanian et al. 2005</a>)</span> and <span class="math inline">\(SET_{ER/PR}\)</span><span class="citation" data-cites="Sinn2019">(<a href="references.html#ref-Sinn2019" role="doc-biblioref">Sinn et al. 2019</a>)</span>, are used to calculate scores for each BC patient. These scores then are used to calculate associations with survival analysis. When performing cox regression, we try to adjust for sensible covariates in order to reduce bias. Even though we are adjusting, it is very difficult to know if a covariate is missing in the regression. Thus, care should be always taken when interpreting these results.</p>
<p>This chapter is structure in the following way. First section corresponds to loading the datasets and then filtering them. Estrogen signatures scores are calculated using GSVA <span class="citation" data-cites="Hnzelmann2013">(<a href="references.html#ref-Hnzelmann2013" role="doc-biblioref">Hänzelmann, Castelo, and Guinney 2013</a>)</span>. Given the scores, cox regression can be performed adjusting for clinical variables. After this the Hazard ratios can be computed and interpreted along with their confidence intervals.</p>
<section id="loading-and-filtering-the-datasets" class="level2" data-number="1.1">
<h2 data-number="1.1" class="anchored" data-anchor-id="loading-and-filtering-the-datasets"><span class="header-section-number">1.1</span> Loading and filtering the datasets</h2>
<p>The preprocessing of the datasets is described in the website below: &gt; https://chronchi.github.io/transcriptomics</p>
<p>To check the code used here either click in the code button on the top right part of the page or check the github page (github.com/chronchi/molecular_landscape).</p>
<p>The TCGA, METABRIC and SCANB cohorts are used in this section here. They are the biggest cohort of Breast cancer patients in the world. Each datasets has an overall equal distribution of ER+ and ER- patients and similar age distribution.</p>
<div class="cell">

</div>
<div class="cell">

</div>
<div class="cell">

</div>
</section>
<section id="umap-projection-of-the-datasets" class="level2" data-number="1.2">
<h2 data-number="1.2" class="anchored" data-anchor-id="umap-projection-of-the-datasets"><span class="header-section-number">1.2</span> UMAP projection of the datasets</h2>
<p>The plots below show how each patient is different in a molecular sense, and even inside each molecular subtype there are some differences. This indicates how different patients, at least molecularly. We only do the umap of samples that have a PAM50 molecular subtype assigned.</p>
<div class="cell">

</div>
<div class="cell">

</div>
<div class="cell">
<div class="cell-output-display">
<p><img src="surv_analysis_estrogen_files/figure-html/unnamed-chunk-5-1.png" class="img-fluid" width="960"></p>
</div>
</div>
<div class="cell">

</div>
<p>From the plots above we see a distinction of the different molecular subtypes.</p>
</section>
<section id="calculating-scores" class="level2" data-number="1.3">
<h2 data-number="1.3" class="anchored" data-anchor-id="calculating-scores"><span class="header-section-number">1.3</span> Calculating scores</h2>
<p>In order to calculate the scores, the package <code>msigdb</code> is used to load the hallmark data into R. The <span class="math inline">\(SET_{ER/PR}\)</span> is made of the following genes:</p>
<div class="cell">
<div class="cell-output-display">

<table class=" lightable-classic" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
 <thead>
  <tr>
   <th style="text-align:left;"> Affy </th>
   <th style="text-align:left;"> ID </th>
   <th style="text-align:left;"> is_target </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:left;"> 202089_s_at </td>
   <td style="text-align:left;"> SLC39A6 </td>
   <td style="text-align:left;"> yes </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 203438_at </td>
   <td style="text-align:left;"> STC2 </td>
   <td style="text-align:left;"> yes </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 204508_s_at </td>
   <td style="text-align:left;"> CA12 </td>
   <td style="text-align:left;"> yes </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 205225_at </td>
   <td style="text-align:left;"> ESR1 </td>
   <td style="text-align:left;"> yes </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 205380_at </td>
   <td style="text-align:left;"> PDZK1 </td>
   <td style="text-align:left;"> yes </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 205440_s_at </td>
   <td style="text-align:left;"> NPY1R </td>
   <td style="text-align:left;"> yes </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 205831_at </td>
   <td style="text-align:left;"> CD2 </td>
   <td style="text-align:left;"> yes </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 206401_s_at </td>
   <td style="text-align:left;"> MAPT </td>
   <td style="text-align:left;"> yes </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 209123_at </td>
   <td style="text-align:left;"> QDPR </td>
   <td style="text-align:left;"> yes </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 209309_at </td>
   <td style="text-align:left;"> AZGP1 </td>
   <td style="text-align:left;"> yes </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 209459_s_at </td>
   <td style="text-align:left;"> ABAT </td>
   <td style="text-align:left;"> yes </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 213245_at </td>
   <td style="text-align:left;"> ADCY1 </td>
   <td style="text-align:left;"> yes </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 213539_at </td>
   <td style="text-align:left;"> CD3D </td>
   <td style="text-align:left;"> yes </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 214440_at </td>
   <td style="text-align:left;"> NAT1 </td>
   <td style="text-align:left;"> yes </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 218398_at </td>
   <td style="text-align:left;"> MRPS30 </td>
   <td style="text-align:left;"> yes </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 218976_at </td>
   <td style="text-align:left;"> DNAJC12 </td>
   <td style="text-align:left;"> yes </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 219197_s_at </td>
   <td style="text-align:left;"> SCUBE2 </td>
   <td style="text-align:left;"> yes </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 222379_at </td>
   <td style="text-align:left;"> KCNE4 </td>
   <td style="text-align:left;"> yes </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 200650_s_at </td>
   <td style="text-align:left;"> LDHA </td>
   <td style="text-align:left;"> no </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 202961_s_at </td>
   <td style="text-align:left;"> ATP5J2 </td>
   <td style="text-align:left;"> no </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 211662_s_at </td>
   <td style="text-align:left;"> VDAC2 </td>
   <td style="text-align:left;"> no </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 201623_s_at </td>
   <td style="text-align:left;"> DARS </td>
   <td style="text-align:left;"> no </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 205480_s_at </td>
   <td style="text-align:left;"> UGP2 </td>
   <td style="text-align:left;"> no </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 217750_s_at </td>
   <td style="text-align:left;"> UBE2Z </td>
   <td style="text-align:left;"> no </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 212175_s_at </td>
   <td style="text-align:left;"> AK2 </td>
   <td style="text-align:left;"> no </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 212050_at </td>
   <td style="text-align:left;"> WIPF2 </td>
   <td style="text-align:left;"> no </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 202631_s_at </td>
   <td style="text-align:left;"> APPBP2 </td>
   <td style="text-align:left;"> no </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 202342_s_at </td>
   <td style="text-align:left;"> TRIM2 </td>
   <td style="text-align:left;"> no </td>
  </tr>
</tbody>
</table>

</div>
</div>
<p>The first 18 genes are considered to be the target genes, the last 10 genes are the genes used for reference. According to their paper, the score is calculated in the following way:</p>
<p><span class="math display">\[
SET_{ER/PR} = \sum_{i = 1}^{18} \frac{T_i}{18} - \sum_{j=1}^{10}\frac{R_j}{10} + 2
\]</span></p>
<p>Where <span class="math inline">\(T_i\)</span> are the expression levels of target genes and <span class="math inline">\(R_j\)</span> are the expression levels of the reference genes. Here we use GSVA to calculate the scores, even when using their genes.</p>
<div class="cell">

</div>
<p>Before we calculate any score, let us check the number of genes available for each pathway in each dataset. This is important, in other to have robust scores, most of the genes should be available in the datasets. We also add signatures of 18 and 200 random genes for control.</p>
<div class="cell">

</div>
<div class="cell">
<div class="cell-output-display">

<table class=" lightable-classic" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
 <thead>
  <tr>
   <th style="text-align:left;"> pathway </th>
   <th style="text-align:right;"> tcga </th>
   <th style="text-align:right;"> scanb </th>
   <th style="text-align:right;"> metabric </th>
   <th style="text-align:right;"> n </th>
   <th style="text-align:left;"> average_percentage </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:left;"> HALLMARK_ADIPOGENESIS </td>
   <td style="text-align:right;"> 188 </td>
   <td style="text-align:right;"> 183 </td>
   <td style="text-align:right;"> 164 </td>
   <td style="text-align:right;"> 210 </td>
   <td style="text-align:left;"> 0.85 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> HALLMARK_ALLOGRAFT_REJECTION </td>
   <td style="text-align:right;"> 141 </td>
   <td style="text-align:right;"> 139 </td>
   <td style="text-align:right;"> 139 </td>
   <td style="text-align:right;"> 335 </td>
   <td style="text-align:left;"> 0.42 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> HALLMARK_ANDROGEN_RESPONSE </td>
   <td style="text-align:right;"> 93 </td>
   <td style="text-align:right;"> 91 </td>
   <td style="text-align:right;"> 83 </td>
   <td style="text-align:right;"> 102 </td>
   <td style="text-align:left;"> 0.87 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> HALLMARK_ANGIOGENESIS </td>
   <td style="text-align:right;"> 30 </td>
   <td style="text-align:right;"> 30 </td>
   <td style="text-align:right;"> 28 </td>
   <td style="text-align:right;"> 36 </td>
   <td style="text-align:left;"> 0.81 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> HALLMARK_APICAL_JUNCTION </td>
   <td style="text-align:right;"> 155 </td>
   <td style="text-align:right;"> 146 </td>
   <td style="text-align:right;"> 151 </td>
   <td style="text-align:right;"> 231 </td>
   <td style="text-align:left;"> 0.65 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> HALLMARK_APICAL_SURFACE </td>
   <td style="text-align:right;"> 31 </td>
   <td style="text-align:right;"> 30 </td>
   <td style="text-align:right;"> 31 </td>
   <td style="text-align:right;"> 46 </td>
   <td style="text-align:left;"> 0.67 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> HALLMARK_APOPTOSIS </td>
   <td style="text-align:right;"> 141 </td>
   <td style="text-align:right;"> 142 </td>
   <td style="text-align:right;"> 130 </td>
   <td style="text-align:right;"> 183 </td>
   <td style="text-align:left;"> 0.75 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> HALLMARK_BILE_ACID_METABOLISM </td>
   <td style="text-align:right;"> 75 </td>
   <td style="text-align:right;"> 74 </td>
   <td style="text-align:right;"> 77 </td>
   <td style="text-align:right;"> 114 </td>
   <td style="text-align:left;"> 0.66 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> HALLMARK_CHOLESTEROL_HOMEOSTASIS </td>
   <td style="text-align:right;"> 67 </td>
   <td style="text-align:right;"> 67 </td>
   <td style="text-align:right;"> 60 </td>
   <td style="text-align:right;"> 77 </td>
   <td style="text-align:left;"> 0.84 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> HALLMARK_COAGULATION </td>
   <td style="text-align:right;"> 92 </td>
   <td style="text-align:right;"> 91 </td>
   <td style="text-align:right;"> 88 </td>
   <td style="text-align:right;"> 162 </td>
   <td style="text-align:left;"> 0.56 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> HALLMARK_COMPLEMENT </td>
   <td style="text-align:right;"> 152 </td>
   <td style="text-align:right;"> 149 </td>
   <td style="text-align:right;"> 144 </td>
   <td style="text-align:right;"> 237 </td>
   <td style="text-align:left;"> 0.63 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> HALLMARK_DNA_REPAIR </td>
   <td style="text-align:right;"> 145 </td>
   <td style="text-align:right;"> 145 </td>
   <td style="text-align:right;"> 127 </td>
   <td style="text-align:right;"> 170 </td>
   <td style="text-align:left;"> 0.82 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> HALLMARK_E2F_TARGETS </td>
   <td style="text-align:right;"> 196 </td>
   <td style="text-align:right;"> 181 </td>
   <td style="text-align:right;"> 174 </td>
   <td style="text-align:right;"> 218 </td>
   <td style="text-align:left;"> 0.84 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION </td>
   <td style="text-align:right;"> 179 </td>
   <td style="text-align:right;"> 172 </td>
   <td style="text-align:right;"> 172 </td>
   <td style="text-align:right;"> 204 </td>
   <td style="text-align:left;"> 0.85 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> HALLMARK_ESTROGEN_RESPONSE_EARLY </td>
   <td style="text-align:right;"> 189 </td>
   <td style="text-align:right;"> 175 </td>
   <td style="text-align:right;"> 162 </td>
   <td style="text-align:right;"> 216 </td>
   <td style="text-align:left;"> 0.81 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> HALLMARK_ESTROGEN_RESPONSE_LATE </td>
   <td style="text-align:right;"> 176 </td>
   <td style="text-align:right;"> 165 </td>
   <td style="text-align:right;"> 162 </td>
   <td style="text-align:right;"> 218 </td>
   <td style="text-align:left;"> 0.77 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> HALLMARK_FATTY_ACID_METABOLISM </td>
   <td style="text-align:right;"> 128 </td>
   <td style="text-align:right;"> 126 </td>
   <td style="text-align:right;"> 127 </td>
   <td style="text-align:right;"> 165 </td>
   <td style="text-align:left;"> 0.77 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> HALLMARK_G2M_CHECKPOINT </td>
   <td style="text-align:right;"> 187 </td>
   <td style="text-align:right;"> 172 </td>
   <td style="text-align:right;"> 173 </td>
   <td style="text-align:right;"> 204 </td>
   <td style="text-align:left;"> 0.87 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> HALLMARK_GLYCOLYSIS </td>
   <td style="text-align:right;"> 173 </td>
   <td style="text-align:right;"> 169 </td>
   <td style="text-align:right;"> 167 </td>
   <td style="text-align:right;"> 215 </td>
   <td style="text-align:left;"> 0.79 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> HALLMARK_HEDGEHOG_SIGNALING </td>
   <td style="text-align:right;"> 24 </td>
   <td style="text-align:right;"> 22 </td>
   <td style="text-align:right;"> 25 </td>
   <td style="text-align:right;"> 36 </td>
   <td style="text-align:left;"> 0.66 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> HALLMARK_HEME_METABOLISM </td>
   <td style="text-align:right;"> 152 </td>
   <td style="text-align:right;"> 152 </td>
   <td style="text-align:right;"> 138 </td>
   <td style="text-align:right;"> 214 </td>
   <td style="text-align:left;"> 0.69 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> HALLMARK_HYPOXIA </td>
   <td style="text-align:right;"> 168 </td>
   <td style="text-align:right;"> 163 </td>
   <td style="text-align:right;"> 163 </td>
   <td style="text-align:right;"> 215 </td>
   <td style="text-align:left;"> 0.77 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> HALLMARK_IL2_STAT5_SIGNALING </td>
   <td style="text-align:right;"> 163 </td>
   <td style="text-align:right;"> 156 </td>
   <td style="text-align:right;"> 163 </td>
   <td style="text-align:right;"> 216 </td>
   <td style="text-align:left;"> 0.74 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> HALLMARK_IL6_JAK_STAT3_SIGNALING </td>
   <td style="text-align:right;"> 65 </td>
   <td style="text-align:right;"> 65 </td>
   <td style="text-align:right;"> 63 </td>
   <td style="text-align:right;"> 103 </td>
   <td style="text-align:left;"> 0.62 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> HALLMARK_INFLAMMATORY_RESPONSE </td>
   <td style="text-align:right;"> 142 </td>
   <td style="text-align:right;"> 136 </td>
   <td style="text-align:right;"> 144 </td>
   <td style="text-align:right;"> 222 </td>
   <td style="text-align:left;"> 0.63 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> HALLMARK_INTERFERON_ALPHA_RESPONSE </td>
   <td style="text-align:right;"> 92 </td>
   <td style="text-align:right;"> 90 </td>
   <td style="text-align:right;"> 81 </td>
   <td style="text-align:right;"> 140 </td>
   <td style="text-align:left;"> 0.63 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> HALLMARK_INTERFERON_GAMMA_RESPONSE </td>
   <td style="text-align:right;"> 178 </td>
   <td style="text-align:right;"> 176 </td>
   <td style="text-align:right;"> 164 </td>
   <td style="text-align:right;"> 286 </td>
   <td style="text-align:left;"> 0.6 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> HALLMARK_KRAS_SIGNALING_DN </td>
   <td style="text-align:right;"> 71 </td>
   <td style="text-align:right;"> 56 </td>
   <td style="text-align:right;"> 82 </td>
   <td style="text-align:right;"> 220 </td>
   <td style="text-align:left;"> 0.32 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> HALLMARK_KRAS_SIGNALING_UP </td>
   <td style="text-align:right;"> 156 </td>
   <td style="text-align:right;"> 148 </td>
   <td style="text-align:right;"> 146 </td>
   <td style="text-align:right;"> 220 </td>
   <td style="text-align:left;"> 0.68 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> HALLMARK_MITOTIC_SPINDLE </td>
   <td style="text-align:right;"> 194 </td>
   <td style="text-align:right;"> 190 </td>
   <td style="text-align:right;"> 168 </td>
   <td style="text-align:right;"> 215 </td>
   <td style="text-align:left;"> 0.86 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> HALLMARK_MTORC1_SIGNALING </td>
   <td style="text-align:right;"> 190 </td>
   <td style="text-align:right;"> 189 </td>
   <td style="text-align:right;"> 176 </td>
   <td style="text-align:right;"> 211 </td>
   <td style="text-align:left;"> 0.88 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> HALLMARK_MYC_TARGETS_V1 </td>
   <td style="text-align:right;"> 194 </td>
   <td style="text-align:right;"> 192 </td>
   <td style="text-align:right;"> 185 </td>
   <td style="text-align:right;"> 236 </td>
   <td style="text-align:left;"> 0.81 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> HALLMARK_MYC_TARGETS_V2 </td>
   <td style="text-align:right;"> 58 </td>
   <td style="text-align:right;"> 57 </td>
   <td style="text-align:right;"> 52 </td>
   <td style="text-align:right;"> 60 </td>
   <td style="text-align:left;"> 0.93 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> HALLMARK_MYOGENESIS </td>
   <td style="text-align:right;"> 126 </td>
   <td style="text-align:right;"> 132 </td>
   <td style="text-align:right;"> 139 </td>
   <td style="text-align:right;"> 212 </td>
   <td style="text-align:left;"> 0.62 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> HALLMARK_NOTCH_SIGNALING </td>
   <td style="text-align:right;"> 29 </td>
   <td style="text-align:right;"> 29 </td>
   <td style="text-align:right;"> 29 </td>
   <td style="text-align:right;"> 34 </td>
   <td style="text-align:left;"> 0.85 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> HALLMARK_OXIDATIVE_PHOSPHORYLATION </td>
   <td style="text-align:right;"> 199 </td>
   <td style="text-align:right;"> 184 </td>
   <td style="text-align:right;"> 169 </td>
   <td style="text-align:right;"> 220 </td>
   <td style="text-align:left;"> 0.84 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> HALLMARK_P53_PATHWAY </td>
   <td style="text-align:right;"> 177 </td>
   <td style="text-align:right;"> 171 </td>
   <td style="text-align:right;"> 160 </td>
   <td style="text-align:right;"> 215 </td>
   <td style="text-align:left;"> 0.79 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> HALLMARK_PANCREAS_BETA_CELLS </td>
   <td style="text-align:right;"> 14 </td>
   <td style="text-align:right;"> 13 </td>
   <td style="text-align:right;"> 17 </td>
   <td style="text-align:right;"> 44 </td>
   <td style="text-align:left;"> 0.33 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> HALLMARK_PEROXISOME </td>
   <td style="text-align:right;"> 85 </td>
   <td style="text-align:right;"> 85 </td>
   <td style="text-align:right;"> 87 </td>
   <td style="text-align:right;"> 110 </td>
   <td style="text-align:left;"> 0.78 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> HALLMARK_PI3K_AKT_MTOR_SIGNALING </td>
   <td style="text-align:right;"> 91 </td>
   <td style="text-align:right;"> 89 </td>
   <td style="text-align:right;"> 84 </td>
   <td style="text-align:right;"> 118 </td>
   <td style="text-align:left;"> 0.75 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> HALLMARK_PROTEIN_SECRETION </td>
   <td style="text-align:right;"> 94 </td>
   <td style="text-align:right;"> 92 </td>
   <td style="text-align:right;"> 90 </td>
   <td style="text-align:right;"> 98 </td>
   <td style="text-align:left;"> 0.94 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY </td>
   <td style="text-align:right;"> 46 </td>
   <td style="text-align:right;"> 45 </td>
   <td style="text-align:right;"> 43 </td>
   <td style="text-align:right;"> 58 </td>
   <td style="text-align:left;"> 0.77 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> HALLMARK_SPERMATOGENESIS </td>
   <td style="text-align:right;"> 62 </td>
   <td style="text-align:right;"> 58 </td>
   <td style="text-align:right;"> 71 </td>
   <td style="text-align:right;"> 144 </td>
   <td style="text-align:left;"> 0.44 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> HALLMARK_TGF_BETA_SIGNALING </td>
   <td style="text-align:right;"> 49 </td>
   <td style="text-align:right;"> 51 </td>
   <td style="text-align:right;"> 45 </td>
   <td style="text-align:right;"> 59 </td>
   <td style="text-align:left;"> 0.82 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> HALLMARK_TNFA_SIGNALING_VIA_NFKB </td>
   <td style="text-align:right;"> 175 </td>
   <td style="text-align:right;"> 174 </td>
   <td style="text-align:right;"> 166 </td>
   <td style="text-align:right;"> 228 </td>
   <td style="text-align:left;"> 0.75 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> HALLMARK_UNFOLDED_PROTEIN_RESPONSE </td>
   <td style="text-align:right;"> 106 </td>
   <td style="text-align:right;"> 104 </td>
   <td style="text-align:right;"> 98 </td>
   <td style="text-align:right;"> 115 </td>
   <td style="text-align:left;"> 0.89 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> HALLMARK_UV_RESPONSE_DN </td>
   <td style="text-align:right;"> 129 </td>
   <td style="text-align:right;"> 127 </td>
   <td style="text-align:right;"> 121 </td>
   <td style="text-align:right;"> 152 </td>
   <td style="text-align:left;"> 0.83 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> HALLMARK_UV_RESPONSE_UP </td>
   <td style="text-align:right;"> 129 </td>
   <td style="text-align:right;"> 128 </td>
   <td style="text-align:right;"> 127 </td>
   <td style="text-align:right;"> 191 </td>
   <td style="text-align:left;"> 0.67 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> HALLMARK_WNT_BETA_CATENIN_SIGNALING </td>
   <td style="text-align:right;"> 36 </td>
   <td style="text-align:right;"> 36 </td>
   <td style="text-align:right;"> 36 </td>
   <td style="text-align:right;"> 50 </td>
   <td style="text-align:left;"> 0.72 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> HALLMARK_XENOBIOTIC_METABOLISM </td>
   <td style="text-align:right;"> 142 </td>
   <td style="text-align:right;"> 141 </td>
   <td style="text-align:right;"> 141 </td>
   <td style="text-align:right;"> 224 </td>
   <td style="text-align:left;"> 0.63 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> SET_ERPR </td>
   <td style="text-align:right;"> 18 </td>
   <td style="text-align:right;"> 16 </td>
   <td style="text-align:right;"> 16 </td>
   <td style="text-align:right;"> 18 </td>
   <td style="text-align:left;"> 0.93 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> random_200 </td>
   <td style="text-align:right;"> 147 </td>
   <td style="text-align:right;"> 142 </td>
   <td style="text-align:right;"> 150 </td>
   <td style="text-align:right;"> 200 </td>
   <td style="text-align:left;"> 0.73 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> random_18 </td>
   <td style="text-align:right;"> 10 </td>
   <td style="text-align:right;"> 10 </td>
   <td style="text-align:right;"> 15 </td>
   <td style="text-align:right;"> 18 </td>
   <td style="text-align:left;"> 0.65 </td>
  </tr>
</tbody>
</table>

</div>
</div>
<p>Most of the genes are available in all datasets. When calculating scores it is always good to check the availability of the genes. Otherwise this can make the score unstable, since too many of the genes are missing. For example, HALLMARK_PANCREAS_BETA_CELLS might have an unstable score, due to a lot of genes missing.</p>
<div class="cell">

</div>
<div class="cell">

</div>
<div class="cell">

</div>
<p>For each dataset one can plot the differences in scores for ER+ and ER- BC patients. This should be already an indication that the scores are meaningful. The next sections shows the results for each dataset individually.</p>
<div class="cell">

</div>
<div class="panel-tabset">
<ul class="nav nav-tabs" role="tablist"><li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-1-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-1" role="tab" aria-controls="tabset-1-1" aria-selected="true">TCGA</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-2" role="tab" aria-controls="tabset-1-2" aria-selected="false">METABRIC</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-3-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-3" role="tab" aria-controls="tabset-1-3" aria-selected="false">SCANB</a></li></ul>
<div class="tab-content">
<div id="tabset-1-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-1-1-tab">
<div class="cell">

</div>
<div class="cell">
<div class="cell-output-display">
<p><img src="surv_analysis_estrogen_files/figure-html/unnamed-chunk-16-1.png" class="img-fluid" width="960"></p>
</div>
</div>
</div>
<div id="tabset-1-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-2-tab">
<div class="cell">

</div>
<div class="cell">
<div class="cell-output-display">
<p><img src="surv_analysis_estrogen_files/figure-html/unnamed-chunk-18-1.png" class="img-fluid" width="960"></p>
</div>
</div>
</div>
<div id="tabset-1-3" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-3-tab">
<div class="cell">

</div>
<div class="cell">
<div class="cell-output-display">
<p><img src="surv_analysis_estrogen_files/figure-html/unnamed-chunk-20-1.png" class="img-fluid" width="960"></p>
</div>
</div>
</div>
</div>
</div>
<p>From the plots above one can conclude that three different estrogen pathways capture the differences between ER status and also molecular subtypes. There are two plots with random genes for control and we can see that there is no difference between the ER status when using those gene sets.</p>
<p>Another way to look at the data is to plot by molecular subtype instead of ER status.</p>
<div class="panel-tabset">
<ul class="nav nav-tabs" role="tablist"><li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-2-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-2-1" role="tab" aria-controls="tabset-2-1" aria-selected="true">TCGA</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-2-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-2-2" role="tab" aria-controls="tabset-2-2" aria-selected="false">METABRIC</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-2-3-tab" data-bs-toggle="tab" data-bs-target="#tabset-2-3" role="tab" aria-controls="tabset-2-3" aria-selected="false">SCANB</a></li></ul>
<div class="tab-content">
<div id="tabset-2-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-2-1-tab">
<div class="cell">

</div>
<div class="cell">

</div>
<div class="cell">
<div class="cell-output-display">
<p><img src="surv_analysis_estrogen_files/figure-html/unnamed-chunk-23-1.png" class="img-fluid" width="960"></p>
</div>
</div>
</div>
<div id="tabset-2-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-2-2-tab">
<div class="cell">

</div>
<div class="cell">
<div class="cell-output-display">
<p><img src="surv_analysis_estrogen_files/figure-html/unnamed-chunk-25-1.png" class="img-fluid" width="960"></p>
</div>
</div>
</div>
<div id="tabset-2-3" class="tab-pane" role="tabpanel" aria-labelledby="tabset-2-3-tab">
<div class="cell">

</div>
<div class="cell">
<div class="cell-output-display">
<p><img src="surv_analysis_estrogen_files/figure-html/unnamed-chunk-27-1.png" class="img-fluid" width="960"></p>
</div>
</div>
</div>
</div>
</div>
<p>In all cohorts the luminal A and B patients have a similar score. Also the distinction is very clear between the basal and HER2-like patients versus luminal A and B.</p>
<p>One question that usually arises when calculating scores from gene sets is if proliferation associated genes (PAG) are driving the distinctions. These signatures are highly curated and they have close or no PAGs. Therefore, the scores are not affected by PAGs and they really reflect the biology.</p>
<p>In the next section we will show how these scores are also prognostic for ER+ BC patients.</p>
</section>
<section id="survival-analysis" class="level2 page-columns page-full" data-number="1.4">
<h2 data-number="1.4" class="anchored" data-anchor-id="survival-analysis"><span class="header-section-number">1.4</span> Survival analysis</h2>
<p>Since the scores are continuous variables and they are already scaled due to the output of GSVA, cox regression <span class="citation" data-cites="Cox1972">(<a href="references.html#ref-Cox1972" role="doc-biblioref">Cox 1972</a>)</span> can be used. The advantages of using cox regression is that one can control for other variables. You might ask, why should one control for clinical variables? One of the reasons is because the data being dealt here is observational data. There are several confounders, for example, a score might be up because patients with a higher tumor grade have higher expression of some genes. Thus the score is confounded by the tumor grade and the interpretation changes.</p>
<p>There are limitations still when dealing with observational data. A strong hypothesis for performing survival analysis with observational data is that we have measured all the confounder variables. This is pretty strong and in practice we never know if a confounder is missing or not. For a more thorough overview of the causal framework for observational data, check the books <span class="citation" data-cites="Gelman2020-uh McElreath2020-jn">(<a href="references.html#ref-Gelman2020-uh" role="doc-biblioref">Gelman, Hill, and Vehtari 2020</a>; <a href="references.html#ref-McElreath2020-jn" role="doc-biblioref">McElreath 2020</a>)</span>.</p>
<div class="page-columns page-full"><p>All the cohorts have a different set of clinical variables available. Therefore, the regression will be done by adjusting a different set of variables. Below is the description of the variables used for each cohort.<a href="#fn1" class="footnote-ref" id="fnref1" role="doc-noteref"><sup>1</sup></a></p><div class="no-row-height column-margin column-container"><li id="fn1" role="doc-endnote"><p><sup>1</sup>&nbsp;This <a href="https://web.archive.org/web/20220630083539/https://www.cancer.net/cancer-types/breast-cancer/stages">webpage</a> explains with more details the different tumor stages and how BC are classified.</p></li></div></div>
<ul>
<li><strong>Age</strong>: age is one of the most important factors to adjust, specially in breast cancer. This covariate is used for all cohorts.</li>
<li><strong>NPI</strong>: the Nottingham prognostic index scores each tumor based on tumor grade, tumor size and number of lymph nodes. Only METABRIC has this information.</li>
<li><strong>Tumor Size</strong>: as name describes. Only SCANB has this information.</li>
<li><strong>Tumor Stage</strong>: tumors are usually described in terms of stage, it reflects the tumor size and location. SCANB and TCGA have this information.</li>
<li><strong>Node Stage</strong>: similar to tumor stage, but encodes the number of lymph nodes where breast cancer cells can be found. SCANB and TCGA have this information.</li>
</ul>
<p>Thus the models we are going to use for the survival analysis of each dataset is shown below.</p>
<ul>
<li>TCGA: <code>~ score + age + node_stage + tumor_stage</code>,</li>
<li>SCANB: <code>~ score + age + node_stage + tumor_stage</code>,</li>
<li>METABRIC: <code>~ score + age + NPI</code>,</li>
</ul>
<p>where score is one of the scores calculated earlier with GSVA. Note here that since each node stage and tumor stage have sub classifications they will be grouped together, otherwise there will be too many variables with few points. We will also subselect specific tumor stages for TCGA and SCANB, since there are very few patients with tumor stage 4.</p>
<p>To be very specific in the survival analysis, only endocrine treated patients should be used in the analysis, as that is what are interested in. METABRIC and SCANB has this kind of annotation, but TCGA not. So the survival analysis on SCANB and METABRIC will be performed on endocrine only treated, ER+ BC patients. On TCGA, to mitigate this effect, we subselect only luminal A and B patients, and we keep in mind that they might have been treated with chemotherapy as well. Moreover, in Sweden the guidelines for BC treatment is to use 10% as the threshold for ER status. Therefore, ER+ BC patients used on SCANB are those that are above the 10% threshold.</p>
<p>The table below shows the number of patients for each cohort.</p>
<div class="cell">
<div class="cell-output-display">

<table class=" lightable-classic" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
 <thead>
  <tr>
   <th style="text-align:left;"> cohort </th>
   <th style="text-align:right;"> number_patients </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:left;"> TCGA </td>
   <td style="text-align:right;"> 684 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> SCANB </td>
   <td style="text-align:right;"> 1319 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> METABRIC </td>
   <td style="text-align:right;"> 938 </td>
  </tr>
</tbody>
</table>

</div>
</div>
<div class="cell">

</div>
<div class="cell">

</div>
<section id="results" class="level3" data-number="1.4.1">
<h3 data-number="1.4.1" class="anchored" data-anchor-id="results"><span class="header-section-number">1.4.1</span> Results</h3>
<p>The table below shows the results for each analysis performed for a specific term. In order to understand the table the user can filter based on the term, cohort and type of analysis. Since METABRIC has recurrence free survival (RFS) and overall survival (OS), the results for both analysis are presented here.</p>
<div class="cell">
<div class="cell-output-display">

<div id="htmlwidget-7512c87e2c1b128ebd52" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-7512c87e2c1b128ebd52">{"x":{"filter":"top","vertical":false,"filterHTML":"<tr>\n  <td><\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"-2.27148265520376\" data-max=\"1.14067956714181\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0.004528300477021\" data-max=\"1.3603405844892\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"-3.71530103169947\" data-max=\"12.1646114627659\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0\" data-max=\"0.99513910655546\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0.103159117239438\" data-max=\"3.12889393637955\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n<\/tr>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100"],["os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","os","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs","rfs"],["tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","tcga","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","scanb","metabric","metabric","metabric","metabric","metabric","metabric","metabric","metabric","metabric","metabric","metabric","metabric","metabric","metabric","metabric","metabric","metabric","metabric","metabric","metabric","metabric","metabric","metabric","metabric","metabric","metabric","metabric","metabric","metabric","metabric"],["HALLMARK_ESTROGEN_RESPONSE_EARLY","HALLMARK_ESTROGEN_RESPONSE_EARLY","HALLMARK_ESTROGEN_RESPONSE_EARLY","HALLMARK_ESTROGEN_RESPONSE_EARLY","HALLMARK_ESTROGEN_RESPONSE_EARLY","HALLMARK_ESTROGEN_RESPONSE_EARLY","HALLMARK_ESTROGEN_RESPONSE_EARLY","HALLMARK_ESTROGEN_RESPONSE_LATE","HALLMARK_ESTROGEN_RESPONSE_LATE","HALLMARK_ESTROGEN_RESPONSE_LATE","HALLMARK_ESTROGEN_RESPONSE_LATE","HALLMARK_ESTROGEN_RESPONSE_LATE","HALLMARK_ESTROGEN_RESPONSE_LATE","HALLMARK_ESTROGEN_RESPONSE_LATE","SET_ERPR","SET_ERPR","SET_ERPR","SET_ERPR","SET_ERPR","SET_ERPR","SET_ERPR","random_200","random_200","random_200","random_200","random_200","random_200","random_200","random_18","random_18","random_18","random_18","random_18","random_18","random_18","HALLMARK_ESTROGEN_RESPONSE_EARLY","HALLMARK_ESTROGEN_RESPONSE_EARLY","HALLMARK_ESTROGEN_RESPONSE_EARLY","HALLMARK_ESTROGEN_RESPONSE_EARLY","HALLMARK_ESTROGEN_RESPONSE_EARLY","HALLMARK_ESTROGEN_RESPONSE_EARLY","HALLMARK_ESTROGEN_RESPONSE_EARLY","HALLMARK_ESTROGEN_RESPONSE_LATE","HALLMARK_ESTROGEN_RESPONSE_LATE","HALLMARK_ESTROGEN_RESPONSE_LATE","HALLMARK_ESTROGEN_RESPONSE_LATE","HALLMARK_ESTROGEN_RESPONSE_LATE","HALLMARK_ESTROGEN_RESPONSE_LATE","HALLMARK_ESTROGEN_RESPONSE_LATE","SET_ERPR","SET_ERPR","SET_ERPR","SET_ERPR","SET_ERPR","SET_ERPR","SET_ERPR","random_200","random_200","random_200","random_200","random_200","random_200","random_200","random_18","random_18","random_18","random_18","random_18","random_18","random_18","HALLMARK_ESTROGEN_RESPONSE_EARLY","HALLMARK_ESTROGEN_RESPONSE_EARLY","HALLMARK_ESTROGEN_RESPONSE_EARLY","HALLMARK_ESTROGEN_RESPONSE_LATE","HALLMARK_ESTROGEN_RESPONSE_LATE","HALLMARK_ESTROGEN_RESPONSE_LATE","SET_ERPR","SET_ERPR","SET_ERPR","random_200","random_200","random_200","random_18","random_18","random_18","HALLMARK_ESTROGEN_RESPONSE_EARLY","HALLMARK_ESTROGEN_RESPONSE_EARLY","HALLMARK_ESTROGEN_RESPONSE_EARLY","HALLMARK_ESTROGEN_RESPONSE_LATE","HALLMARK_ESTROGEN_RESPONSE_LATE","HALLMARK_ESTROGEN_RESPONSE_LATE","SET_ERPR","SET_ERPR","SET_ERPR","random_200","random_200","random_200","random_18","random_18","random_18"],["age","node_stageN1","node_stageN2","node_stageN3","tumor_stagestage_ii","tumor_stagestage_iii","HALLMARK_ESTROGEN_RESPONSE_EARLY","age","node_stageN1","node_stageN2","node_stageN3","tumor_stagestage_ii","tumor_stagestage_iii","HALLMARK_ESTROGEN_RESPONSE_LATE","age","node_stageN1","node_stageN2","node_stageN3","tumor_stagestage_ii","tumor_stagestage_iii","SET_ERPR","age","node_stageN1","node_stageN2","node_stageN3","tumor_stagestage_ii","tumor_stagestage_iii","random_200","age","node_stageN1","node_stageN2","node_stageN3","tumor_stagestage_ii","tumor_stagestage_iii","random_18","age","node_stageN1","node_stageN2and3","tumor_stageT1","tumor_stageT2","tumor_stageT3","HALLMARK_ESTROGEN_RESPONSE_EARLY","age","node_stageN1","node_stageN2and3","tumor_stageT1","tumor_stageT2","tumor_stageT3","HALLMARK_ESTROGEN_RESPONSE_LATE","age","node_stageN1","node_stageN2and3","tumor_stageT1","tumor_stageT2","tumor_stageT3","SET_ERPR","age","node_stageN1","node_stageN2and3","tumor_stageT1","tumor_stageT2","tumor_stageT3","random_200","age","node_stageN1","node_stageN2and3","tumor_stageT1","tumor_stageT2","tumor_stageT3","random_18","age","npi","HALLMARK_ESTROGEN_RESPONSE_EARLY","age","npi","HALLMARK_ESTROGEN_RESPONSE_LATE","age","npi","SET_ERPR","age","npi","random_200","age","npi","random_18","age","npi","HALLMARK_ESTROGEN_RESPONSE_EARLY","age","npi","HALLMARK_ESTROGEN_RESPONSE_LATE","age","npi","SET_ERPR","age","npi","random_200","age","npi","random_18"],[0.0567152337524123,0.476200705597867,1.09466101517939,0.824240143237846,0.276215858435805,0.265666942198468,-1.45316423474658,0.0543516491298734,0.487664887426536,1.14067956714181,0.912130694798677,0.276274392756463,0.199016757637233,-2.27148265520376,0.0598999745222749,0.491249760541679,1.08950853606705,0.921117875079167,0.32765013562628,0.343683673490105,-1.46820338127532,0.0617501978887594,0.430529347580059,1.04705509410593,0.836298876228518,0.338004180413856,0.448465346981723,-1.49435203995335,0.0592780477420013,0.416063362235048,1.06853723080715,0.773165033705162,0.341021718880365,0.416229045658303,-0.257755784910888,0.0981084420872003,-0.308453113640183,0.571917369779115,-0.13856905781893,0.211595221431306,-0.0117303643555566,-0.950668940382365,0.0977632882010961,-0.276736876957246,0.584175815037848,-0.131952438978738,0.207955867081594,-0.0392444079114511,-0.558395334440776,0.100873688511352,-0.314292941112492,0.564453056074096,-0.151153897532418,0.181128066893171,-0.172996644369302,-0.77450011618949,0.098806248974535,-0.269868160922949,0.592593714945517,-0.139500331254453,0.191880054652138,-0.0289049840707443,-1.20784275752657,0.0980542429809453,-0.301080426597193,0.552725848201119,-0.160053224097509,0.173978683261871,-0.00288717345698174,-0.592610867483573,0.055693069494223,0.219265759722507,-0.551169549172122,0.054826688758258,0.223450055554535,-0.629299270717519,0.0565713643382181,0.205555729610991,-0.490383719893686,0.0536633468535495,0.22629918928716,-1.23755094380535,0.0542788088946874,0.223095708473899,0.341055117265038,0.0470581999919049,0.222068095072552,-0.507065224798242,0.0463106274350261,0.226404320824185,-0.602854161841783,0.0479765714719412,0.206960135917346,-0.495213833598908,0.0449914068469724,0.233000704214954,-1.33735134063678,0.0455426887084237,0.226357927139767,0.422215729484354],[0.010886554098736,0.336308315475216,0.581772588932351,0.706394279596297,0.383952215910169,0.554979764556066,0.737296332838902,0.0109480240355903,0.337299024443268,0.587824947442621,0.711413855912346,0.384106656681585,0.558764016198543,0.859143019111638,0.0105804953292004,0.337921019941039,0.582593873549099,0.708764448198166,0.38489129431499,0.551452484417036,0.39763624663053,0.0107561237562366,0.334785819863858,0.579319934765939,0.706251112738912,0.38169373390972,0.548231515797347,1.3603405844892,0.0107137983950725,0.335717135991992,0.580961647223953,0.704279390969456,0.38229042686991,0.54934057548468,0.528605349297393,0.00878217417070111,0.180635047041078,0.217243789619677,0.23644170147081,0.244957074128653,0.474427849624775,0.40243983904664,0.00876745417132055,0.179759566759836,0.216890943824417,0.236160734870106,0.244853924015564,0.473514796289845,0.468540482121164,0.00895843387983846,0.179894807597288,0.216785814142985,0.236856662734481,0.245447956172112,0.475974112217719,0.227548870273586,0.00882838194620167,0.179032020832331,0.216611123914275,0.236141692172384,0.244802111907716,0.473656549563072,0.812047106474735,0.00874805469031544,0.180298307484441,0.21756238266932,0.23687415839976,0.246135146547033,0.473908120370753,0.277273415817037,0.00463957429423414,0.0402818721583936,0.23919750023664,0.00461849132027626,0.0401220485404947,0.294401791614505,0.00465048674274348,0.0406772539486767,0.134888824759695,0.00460980927366192,0.0401892501826606,0.507296586774614,0.00462562400264107,0.0401006874447158,0.172498317732871,0.00454938828402498,0.0407729178263534,0.238287021744976,0.00453084857530772,0.0405772222367598,0.292245452177103,0.00456270759404006,0.0412326206029048,0.133290365807151,0.00452830047702107,0.0406576587280166,0.507069755395868,0.00454329942611444,0.0405218308739033,0.173819578729616],[5.20965892770396,1.41596470763733,1.88159606692415,1.16682731874457,0.719401652054613,0.478696628535597,-1.97093647428203,4.9645167888913,1.44579394568797,1.94050894250818,1.28213793872334,0.719264787398588,0.35617318200125,-2.64389351327384,5.6613582501153,1.45374135242428,1.87009954195,1.29961072034701,0.851279674198437,0.623233520932321,-3.69232783408581,5.74093412163981,1.28598441760507,1.8073866119055,1.18413813605938,0.885537671660605,0.818021828477839,-1.0985131642709,5.53286944145453,1.23932715262105,1.83925606090007,1.09781010720885,0.892048806119886,0.757688516438217,-0.487614787200866,11.1713159156542,-1.70760391570102,2.63260630271805,-0.586060144876927,0.863805310313978,-0.0247252861838404,-2.36226349417705,11.1507042170682,-1.53948344416615,2.69340805446798,-0.558739957560327,0.849305837828336,-0.0828789474351062,-1.19177607004804,11.2601923354455,-1.74709290006896,2.60373612685653,-0.638166120333559,0.737948971822611,-0.363458095574261,-3.4036649589088,11.1918865287705,-1.50737370705149,2.73574922763449,-0.590748418761298,0.783816990616777,-0.0610251966269821,-1.48740479203241,11.2086911264394,-1.66990156922674,2.54053959797464,-0.675688834859711,0.706842097532894,-0.00609226416024064,-2.13727978838987,12.0039180239954,5.44328622215783,-2.30424460383928,11.8711252130227,5.56925839240261,-2.13755244921041,12.1646114627659,5.05333348879315,-3.63546587915867,11.6411208507376,5.63083880039134,-2.43950181426152,11.7343754839771,5.56338862722657,1.97715039629077,10.3438521959421,5.44646071243445,-2.12795989091224,10.2211818967887,5.57959141468979,-2.06283504961593,10.514934495169,5.01933015392103,-3.71530103169947,9.93560543857062,5.73079492288613,-2.63741098025598,10.0241442258128,5.58607353759884,2.42904586796363],[1.8918809239639e-07,0.156785837767849,0.0598908806554443,0.24328011221383,0.471893479242516,0.632154464640894,0.0487311424674724,6.88723286247422e-07,0.148235004742027,0.0523178684532471,0.199794262294553,0.471977787602689,0.721710874415636,0.00819584358466826,1.50179531285201e-08,0.146018028736397,0.0614699960909584,0.193734423545292,0.394614013062295,0.533131076698918,0.000222210765540212,9.4155717225859e-09,0.198448499956816,0.0707020199440809,0.236358386099275,0.375866711000773,0.413344723167035,0.271980472505376,3.15033968214652e-08,0.215224367108381,0.0658775324915814,0.272287417088517,0.372366772942491,0.448637475275488,0.625822719539068,5.63399548438551e-29,0.0877098613357082,0.00847325025480468,0.557835097890749,0.387694850205822,0.980274085785622,0.0181637269796194,7.10418805398548e-29,0.123686316618528,0.00707256499044029,0.576339205106161,0.395711133278859,0.933947793952985,0.233349061241365,2.06307949043091e-29,0.0806212247717774,0.00922137225679203,0.523365548805385,0.46054545831525,0.716262710523858,0.00066488252678568,4.46824228706481e-29,0.131714894359006,0.00622384360682512,0.554689000622669,0.433147502749004,0.951339142441631,0.136907932026688,3.69618573962091e-29,0.0949388398277234,0.011068155958812,0.499238226244576,0.479664610178863,0.99513910655546,0.0325752417652802,3.38866560979539e-33,5.23064983567251e-08,0.0212089166682267,1.67203484996365e-32,2.55825861234783e-08,0.0325530845446077,4.79722447594097e-34,4.3416483739151e-07,0.000277478720564909,2.54646894281451e-31,1.79335283942469e-08,0.0147075288191497,8.49510383683592e-32,2.64585750496536e-08,0.0480246326451877,4.46231289401466e-25,5.13819456576677e-08,0.0333404092204405,1.5938556192957e-24,2.41084217440367e-08,0.0391283027390872,7.37320254053452e-26,5.18519674344682e-07,0.000202961919366514,2.91400314619971e-23,9.9961061994468e-09,0.00835415492024733,1.19389348523947e-23,2.32260917420216e-08,0.0151386169881163],[1.05835438387297,1.60994610873037,2.98816956756282,2.2801475213913,1.31813236331054,1.30430057885417,0.233829225703695,1.05585582766376,1.62850902467106,3.12889393637955,2.48962150973089,1.31820952155115,1.22020241343794,0.103159117239438,1.06173034114924,1.63435749964834,2.97281268326916,2.5120970315747,1.3877033793748,1.41013250265652,0.230338944922228,1.06369659788148,1.5380714825129,2.84924798373528,2.30780966150974,1.40214636494516,1.5659072163297,0.224393955952724,1.06107022790898,1.51598192183033,2.9111180901483,2.16661281585786,1.40638378561166,1.51623311571349,0.77278393456144,1.10308239918051,0.734582393286162,1.77166072566408,0.870603127240002,1.23564762016198,0.98833816813629,0.386482403276779,1.10270173170195,0.758253984960135,1.79351219054639,0.876382675743252,1.23115883371752,0.961515678387066,0.572126399172048,1.10613691513867,0.730305060444119,1.75848572656348,0.859715380402114,1.19856866643004,0.841140437396561,0.460934136931332,1.10385240630285,0.763480144219474,1.80867352039931,0.869792735081634,1.21152519143898,0.971508768889796,0.298841257625352,1.1030226147205,0.740018253203297,1.73798404759327,0.852098435589053,1.19003019798789,0.997116990420061,0.55288189733008,1.05727312455574,1.2451621466662,0.576275433642648,1.05635752017624,1.25038318902194,0.532965134465039,1.05820213000036,1.22820742657673,0.61239136234565,1.05512932974788,1.25395077779991,0.290093804704894,1.05577892167791,1.24994019788018,1.40643075734271,1.04818301154725,1.24865640231769,0.602260486883201,1.04739971151652,1.25408261450251,0.547247472114371,1.04914607508827,1.22993354063115,0.609440574490676,1.04601887126987,1.26238236831549,0.262540127849161,1.04659568147823,1.25402443433906,1.52533754947484]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>type_analysis<\/th>\n      <th>cohort<\/th>\n      <th>score<\/th>\n      <th>term<\/th>\n      <th>estimate<\/th>\n      <th>std.error<\/th>\n      <th>statistic<\/th>\n      <th>p.value<\/th>\n      <th>HR<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"columnDefs":[{"targets":8,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatSignif(data, 2, 3, \",\", \".\", null);\n  }"},{"targets":5,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 2, 3, \",\", \".\", null);\n  }"},{"targets":6,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 2, 3, \",\", \".\", null);\n  }"},{"targets":7,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 2, 3, \",\", \".\", null);\n  }"},{"targets":9,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 2, 3, \",\", \".\", null);\n  }"},{"className":"dt-right","targets":[5,6,7,8,9]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true}},"evals":["options.columnDefs.0.render","options.columnDefs.1.render","options.columnDefs.2.render","options.columnDefs.3.render","options.columnDefs.4.render"],"jsHooks":[]}</script>
</div>
</div>
<p>The table above shows that <span class="math inline">\(SET_{ER/PR}\)</span> had a small hazard ratio (&lt; 1) in all 4 analysis performed. Moreover, for all cases where a measure of estrogen signaling was used, the hazard ratio was below 1, indicating that the higher the score, the less likely the patient is to suffer the event in a specific timepoint. This indicates that ER signaling is actually something continuous and not dichotomous.</p>
<p><a href="#fig-forest-plot">Figure&nbsp;<span>1.1</span></a> shows the forest plot of <span class="math inline">\(SET_{ER/PR}\)</span> for all three cohorts.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-forest-plot" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="surv_analysis_estrogen_files/figure-html/fig-forest-plot-1.png" class="img-fluid figure-img" width="672"></p>
<p></p><figcaption class="figure-caption">Figure 1.1: Forest plots of the different cohorts and their hazard ratios. The bars correspond to 95% confidence interval.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>The hazard ratios are all below 1 and small for <span class="math inline">\(SET_{ER/PR}\)</span>. The variability changes depending on the cohort, specially because they have different follow-up times, SCANB being the shortest. TCGA has very few events, but using the 10 to 1 rule of thumb, meaning 10 events for each covariate added, the regression satisfies the rule of thumb.</p>
</section>
</section>
<section id="conclusion" class="level2" data-number="1.5">
<h2 data-number="1.5" class="anchored" data-anchor-id="conclusion"><span class="header-section-number">1.5</span> Conclusion</h2>
<p>In this chapter we’ve shown that ER+ BC patients are very distinct from each other, as it can be seen from the umap projections and the subtypes. These patients might respond differently for endocrine therapy as well, and this might depend on the ER signaling, how active it is. Therefore, when deciding a treatment, more care should be taken with ER+ BC patients and check their signaling scores somehow. The <span class="math inline">\(SET_{ER/PR}\)</span> signature is a good signature showing very good hazard ratios across the different cohorts. This signature has also been validated on the clinics for use.</p>
<p>Knowing the ER signaling for a patient is very important when deciding treatment, but not enough. What could be other alternatives for patients that have low ER signaling and are still considered ER+? Should they use only endocrine therapy or supplement it with something else? In the next chapters we present a framework where we can take a look at a more personalised approach for treatments.</p>


<!-- -->

<div id="refs" class="references csl-bib-body hanging-indent" role="doc-bibliography" style="display: none">
<div id="ref-Cox1972" class="csl-entry" role="doc-biblioentry">
Cox, D. R. 1972. <span>“Regression Models and Life-Tables.”</span> <em>Journal of the Royal Statistical Society: Series B (Methodological)</em> 34 (2): 187–202. <a href="https://doi.org/10.1111/j.2517-6161.1972.tb00899.x">https://doi.org/10.1111/j.2517-6161.1972.tb00899.x</a>.
</div>
<div id="ref-Gelman2020-uh" class="csl-entry" role="doc-biblioentry">
Gelman, Andrew, Jennifer Hill, and Aki Vehtari. 2020. <em>Regression and Other Stories</em>. Analytical Methods for Social Research. Cambridge, England: Cambridge University Press.
</div>
<div id="ref-Hnzelmann2013" class="csl-entry" role="doc-biblioentry">
Hänzelmann, Sonja, Robert Castelo, and Justin Guinney. 2013. <span>“<span>GSVA</span>: Gene Set Variation Analysis for Microarray and <span>RNA</span>-Seq Data.”</span> <em><span>BMC</span> Bioinformatics</em> 14 (1). <a href="https://doi.org/10.1186/1471-2105-14-7">https://doi.org/10.1186/1471-2105-14-7</a>.
</div>
<div id="ref-McElreath2020-jn" class="csl-entry" role="doc-biblioentry">
McElreath, Richard. 2020. <em>Statistical Rethinking</em>. 2nd ed. Chapman &amp; Hall/CRC Texts in Statistical Science. London, England: CRC Press.
</div>
<div id="ref-Sinn2019" class="csl-entry" role="doc-biblioentry">
Sinn, Bruno V., Chunxiao Fu, Rosanna Lau, Jennifer Litton, Tsung-Heng Tsai, Rashmi Murthy, Alda Tam, et al. 2019. <span>“<span>SETER</span>/<span>PR</span>: A Robust 18-Gene Predictor for Sensitivity to Endocrine Therapy for Metastatic Breast Cancer.”</span> <em>Npj Breast Cancer</em> 5 (1). <a href="https://doi.org/10.1038/s41523-019-0111-0">https://doi.org/10.1038/s41523-019-0111-0</a>.
</div>
<div id="ref-Subramanian2005" class="csl-entry" role="doc-biblioentry">
Subramanian, Aravind, Pablo Tamayo, Vamsi K. Mootha, Sayan Mukherjee, Benjamin L. Ebert, Michael A. Gillette, Amanda Paulovich, et al. 2005. <span>“Gene Set Enrichment Analysis: A Knowledge-Based Approach for Interpreting Genome-Wide Expression Profiles.”</span> <em>Proceedings of the National Academy of Sciences</em> 102 (43): 15545–50. <a href="https://doi.org/10.1073/pnas.0506580102">https://doi.org/10.1073/pnas.0506580102</a>.
</div>
</div>
</section>


</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      let href = ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./intro.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">Introduction</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./pca_merging.html" class="pagination-link">
        <span class="nav-page-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">A new framework for personalized medicine: integrating</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb1" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu"># Estrogen receptor status is continuous, not dichotomous</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>Currently in the clinics estrogen receptor (ER) status is treated </span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>as dichotomous condition. Either a breast cancer (BC) is ER positive (ER+) or</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>ER negative (ER-). The threshold for ER+ cells usually is 1% or 10% of the </span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>cells positive in a IHC staining. The idea is that ER+ BC patients</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>will receive endocrine therapy, usually tamoxifen or some aromatase inhibitor,</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>for treatment. The problem is not all patients respond the same to these</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>drugs and they also have different proportions of ER+ cells. </span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>In this document we show how leveraging molecular information distinguishes</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>ER+ BC patients and how one should look more carefully on ER status. In order</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a>to do this, estrogen related signatures: estrogen response early and late from</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>MSigDB<span class="co">[</span><span class="ot">@Subramanian2005</span><span class="co">]</span> and $SET_{ER/PR}$<span class="co">[</span><span class="ot">@Sinn2019</span><span class="co">]</span>, </span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a>are used to calculate scores for each</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a>BC patient. These scores then are used to calculate associations with survival</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a>analysis. When performing cox regression, we try to adjust for sensible </span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a>covariates in order to reduce bias. Even though we are adjusting, it is </span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a>very difficult to know if a covariate is missing in the regression. Thus,</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a>care should be always taken when interpreting these results.</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a>This chapter is structure in the following way. First section corresponds</span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a>to loading the datasets and then filtering them. Estrogen signatures scores</span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a>are calculated using GSVA <span class="co">[</span><span class="ot">@Hnzelmann2013</span><span class="co">]</span>. Given the scores, cox regression</span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a>can be performed adjusting for clinical variables. After this the Hazard</span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a>ratios can be computed and interpreted along with their confidence</span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a>intervals.</span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a><span class="fu">## Loading and filtering the datasets</span></span>
<span id="cb1-30"><a href="#cb1-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-31"><a href="#cb1-31" aria-hidden="true" tabindex="-1"></a>The preprocessing of the datasets is described in the website below:</span>
<span id="cb1-32"><a href="#cb1-32" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; https://chronchi.github.io/transcriptomics</span></span>
<span id="cb1-33"><a href="#cb1-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-34"><a href="#cb1-34" aria-hidden="true" tabindex="-1"></a>To check the code used here either click in the code button on the top right</span>
<span id="cb1-35"><a href="#cb1-35" aria-hidden="true" tabindex="-1"></a>part of the page or check the github page </span>
<span id="cb1-36"><a href="#cb1-36" aria-hidden="true" tabindex="-1"></a>(github.com/chronchi/molecular_landscape).</span>
<span id="cb1-37"><a href="#cb1-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-38"><a href="#cb1-38" aria-hidden="true" tabindex="-1"></a>The TCGA, METABRIC and SCANB cohorts are used in this section here. They</span>
<span id="cb1-39"><a href="#cb1-39" aria-hidden="true" tabindex="-1"></a>are the biggest cohort of Breast cancer patients in the world. Each datasets</span>
<span id="cb1-40"><a href="#cb1-40" aria-hidden="true" tabindex="-1"></a>has an overall equal distribution of ER+ and ER- patients and similar age</span>
<span id="cb1-41"><a href="#cb1-41" aria-hidden="true" tabindex="-1"></a>distribution. </span>
<span id="cb1-42"><a href="#cb1-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-43"><a href="#cb1-43" aria-hidden="true" tabindex="-1"></a><span class="in">```{r setup}</span></span>
<span id="cb1-44"><a href="#cb1-44" aria-hidden="true" tabindex="-1"></a><span class="co"># first load the packages</span></span>
<span id="cb1-45"><a href="#cb1-45" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-46"><a href="#cb1-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-47"><a href="#cb1-47" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb1-48"><a href="#cb1-48" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(forestplot)</span>
<span id="cb1-49"><a href="#cb1-49" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplotify)</span>
<span id="cb1-50"><a href="#cb1-50" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gridExtra)</span>
<span id="cb1-51"><a href="#cb1-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-52"><a href="#cb1-52" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SummarizedExperiment)</span>
<span id="cb1-53"><a href="#cb1-53" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(survival)</span>
<span id="cb1-54"><a href="#cb1-54" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(msigdbr)</span>
<span id="cb1-55"><a href="#cb1-55" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GSVA)</span>
<span id="cb1-56"><a href="#cb1-56" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(uwot)</span>
<span id="cb1-57"><a href="#cb1-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-58"><a href="#cb1-58" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"utils.R"</span>)</span>
<span id="cb1-59"><a href="#cb1-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-60"><a href="#cb1-60" aria-hidden="true" tabindex="-1"></a><span class="co"># by setting the dev to png and pdf, this saves the figures in a specific</span></span>
<span id="cb1-61"><a href="#cb1-61" aria-hidden="true" tabindex="-1"></a><span class="co"># folder in both formats. moreover, since png is coming first, it shows</span></span>
<span id="cb1-62"><a href="#cb1-62" aria-hidden="true" tabindex="-1"></a><span class="co"># this figure when rendering the html. What is nice about this is that it</span></span>
<span id="cb1-63"><a href="#cb1-63" aria-hidden="true" tabindex="-1"></a><span class="co"># inherits the properties from the chunk to save the figure, so no need</span></span>
<span id="cb1-64"><a href="#cb1-64" aria-hidden="true" tabindex="-1"></a><span class="co"># to use ggplot2::ggsave to save the plots. this also works </span></span>
<span id="cb1-65"><a href="#cb1-65" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">dev =</span> <span class="fu">c</span>(<span class="st">'png'</span>, <span class="st">'pdf'</span>, <span class="st">'svg'</span>))</span>
<span id="cb1-66"><a href="#cb1-66" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-67"><a href="#cb1-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-70"><a href="#cb1-70" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-71"><a href="#cb1-71" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb1-72"><a href="#cb1-72" aria-hidden="true" tabindex="-1"></a><span class="co"># load the different datasets. we will do the analysis </span></span>
<span id="cb1-73"><a href="#cb1-73" aria-hidden="true" tabindex="-1"></a><span class="co"># and save the intermediate results so loading all the dataset again</span></span>
<span id="cb1-74"><a href="#cb1-74" aria-hidden="true" tabindex="-1"></a><span class="co"># is not necessary</span></span>
<span id="cb1-75"><a href="#cb1-75" aria-hidden="true" tabindex="-1"></a>tcga <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../data/tcga_brca_tumor_filtered.rds"</span>)</span>
<span id="cb1-76"><a href="#cb1-76" aria-hidden="true" tabindex="-1"></a>metabric <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../data/metabric_filtered.rds"</span>)</span>
<span id="cb1-77"><a href="#cb1-77" aria-hidden="true" tabindex="-1"></a>scanb <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../data/scanb.rds"</span>)</span>
<span id="cb1-78"><a href="#cb1-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-79"><a href="#cb1-79" aria-hidden="true" tabindex="-1"></a><span class="co"># we want to have common variables across datasets. so for tcga</span></span>
<span id="cb1-80"><a href="#cb1-80" aria-hidden="true" tabindex="-1"></a><span class="co"># we convert the ensembl ids to gene symbol in the rownames,</span></span>
<span id="cb1-81"><a href="#cb1-81" aria-hidden="true" tabindex="-1"></a><span class="co"># for scanb and metabric we perform a heavier filtering as well.</span></span>
<span id="cb1-82"><a href="#cb1-82" aria-hidden="true" tabindex="-1"></a><span class="co"># also for the clinical data we want to have common column names,</span></span>
<span id="cb1-83"><a href="#cb1-83" aria-hidden="true" tabindex="-1"></a><span class="co"># such as for pam50, events, time to event, tumor stage,</span></span>
<span id="cb1-84"><a href="#cb1-84" aria-hidden="true" tabindex="-1"></a><span class="co"># tumor grade and others.</span></span>
<span id="cb1-85"><a href="#cb1-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-86"><a href="#cb1-86" aria-hidden="true" tabindex="-1"></a><span class="co"># first convert ensembl to symbol. this information is readily available</span></span>
<span id="cb1-87"><a href="#cb1-87" aria-hidden="true" tabindex="-1"></a><span class="co"># from the tcga filtered data. we just need to remove duplicated symbols</span></span>
<span id="cb1-88"><a href="#cb1-88" aria-hidden="true" tabindex="-1"></a><span class="co"># first.</span></span>
<span id="cb1-89"><a href="#cb1-89" aria-hidden="true" tabindex="-1"></a>symbol_names <span class="ot">&lt;-</span> <span class="fu">mcols</span>(<span class="fu">rowRanges</span>(tcga)) <span class="sc">%&gt;%</span> </span>
<span id="cb1-90"><a href="#cb1-90" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span> </span>
<span id="cb1-91"><a href="#cb1-91" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">duplicated</span>(external_gene_name))</span>
<span id="cb1-92"><a href="#cb1-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-93"><a href="#cb1-93" aria-hidden="true" tabindex="-1"></a>tcga <span class="ot">&lt;-</span> tcga[symbol_names<span class="sc">$</span>ensembl_gene_id, ]</span>
<span id="cb1-94"><a href="#cb1-94" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(tcga) <span class="ot">&lt;-</span> symbol_names<span class="sc">$</span>external_gene_name</span>
<span id="cb1-95"><a href="#cb1-95" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-96"><a href="#cb1-96" aria-hidden="true" tabindex="-1"></a>fpkm_tcga <span class="ot">&lt;-</span> <span class="dv">2</span><span class="sc">^</span><span class="fu">assay</span>(tcga, <span class="st">"logFPKM_TMM"</span>)</span>
<span id="cb1-97"><a href="#cb1-97" aria-hidden="true" tabindex="-1"></a>prop_expressed <span class="ot">&lt;-</span> <span class="fu">rowMeans</span>(fpkm_tcga <span class="sc">&gt;</span> <span class="dv">1</span>)</span>
<span id="cb1-98"><a href="#cb1-98" aria-hidden="true" tabindex="-1"></a>genes_to_keep <span class="ot">&lt;-</span> prop_expressed <span class="sc">&gt;</span> <span class="fl">0.5</span></span>
<span id="cb1-99"><a href="#cb1-99" aria-hidden="true" tabindex="-1"></a>tcga <span class="ot">&lt;-</span> tcga[genes_to_keep, ]</span>
<span id="cb1-100"><a href="#cb1-100" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-101"><a href="#cb1-101" aria-hidden="true" tabindex="-1"></a><span class="co"># we now filter down the list of genes from metabric and scanb.</span></span>
<span id="cb1-102"><a href="#cb1-102" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-103"><a href="#cb1-103" aria-hidden="true" tabindex="-1"></a><span class="co"># remove first genes with na in the table</span></span>
<span id="cb1-104"><a href="#cb1-104" aria-hidden="true" tabindex="-1"></a>keep_genes <span class="ot">&lt;-</span> <span class="fu">rowMeans</span>(<span class="fu">assay</span>(metabric, <span class="st">"median_intensity"</span>) <span class="sc">&gt;</span> <span class="fl">5.5</span>)</span>
<span id="cb1-105"><a href="#cb1-105" aria-hidden="true" tabindex="-1"></a>keep_genes <span class="ot">&lt;-</span> keep_genes <span class="sc">&gt;</span> <span class="fl">0.7</span> <span class="sc">%&gt;%</span> unname <span class="sc">%&gt;%</span> unlist</span>
<span id="cb1-106"><a href="#cb1-106" aria-hidden="true" tabindex="-1"></a>keep_genes <span class="ot">&lt;-</span> <span class="fu">which</span>(<span class="fu">is.na</span>(keep_genes))</span>
<span id="cb1-107"><a href="#cb1-107" aria-hidden="true" tabindex="-1"></a>metabric <span class="ot">&lt;-</span> metabric[<span class="sc">-</span>keep_genes, ]</span>
<span id="cb1-108"><a href="#cb1-108" aria-hidden="true" tabindex="-1"></a><span class="co"># now remove genes with low expression across samples</span></span>
<span id="cb1-109"><a href="#cb1-109" aria-hidden="true" tabindex="-1"></a>keep_genes <span class="ot">&lt;-</span> <span class="fu">rowMeans</span>(<span class="fu">assay</span>(metabric, <span class="st">"median_intensity"</span>) <span class="sc">&gt;</span> <span class="fl">5.5</span>)</span>
<span id="cb1-110"><a href="#cb1-110" aria-hidden="true" tabindex="-1"></a>keep_genes <span class="ot">&lt;-</span> keep_genes <span class="sc">&gt;</span> <span class="fl">0.7</span></span>
<span id="cb1-111"><a href="#cb1-111" aria-hidden="true" tabindex="-1"></a>metabric <span class="ot">&lt;-</span> metabric[keep_genes, ]</span>
<span id="cb1-112"><a href="#cb1-112" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-113"><a href="#cb1-113" aria-hidden="true" tabindex="-1"></a><span class="co"># convert back to non log values and then filter.</span></span>
<span id="cb1-114"><a href="#cb1-114" aria-hidden="true" tabindex="-1"></a>fpkm_scanb <span class="ot">&lt;-</span> <span class="dv">2</span><span class="sc">^</span><span class="fu">assay</span>(scanb, <span class="st">"logFPKM"</span>)</span>
<span id="cb1-115"><a href="#cb1-115" aria-hidden="true" tabindex="-1"></a>prop_expressed <span class="ot">&lt;-</span> <span class="fu">rowMeans</span>(fpkm_scanb <span class="sc">&gt;</span> <span class="dv">1</span>)</span>
<span id="cb1-116"><a href="#cb1-116" aria-hidden="true" tabindex="-1"></a>genes_to_keep <span class="ot">&lt;-</span> prop_expressed <span class="sc">&gt;</span> <span class="fl">0.8</span></span>
<span id="cb1-117"><a href="#cb1-117" aria-hidden="true" tabindex="-1"></a>scanb <span class="ot">&lt;-</span> scanb[genes_to_keep, ]</span>
<span id="cb1-118"><a href="#cb1-118" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-119"><a href="#cb1-119" aria-hidden="true" tabindex="-1"></a><span class="co"># we now convert the clinical data to some common names and values</span></span>
<span id="cb1-120"><a href="#cb1-120" aria-hidden="true" tabindex="-1"></a>tcga <span class="ot">&lt;-</span> tcga[, tcga<span class="sc">$</span>er_status <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Positive"</span>, <span class="st">"Negative"</span>)]</span>
<span id="cb1-121"><a href="#cb1-121" aria-hidden="true" tabindex="-1"></a>tcga<span class="sc">$</span>er_status <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Positive"</span> <span class="ot">=</span> <span class="st">"pos"</span>, <span class="st">"Negative"</span> <span class="ot">=</span> <span class="st">"neg"</span>)[tcga<span class="sc">$</span>er_status]</span>
<span id="cb1-122"><a href="#cb1-122" aria-hidden="true" tabindex="-1"></a>tcga<span class="sc">$</span>pam50 <span class="ot">&lt;-</span> <span class="fu">tolower</span>(tcga<span class="sc">$</span>molecular_subtype)</span>
<span id="cb1-123"><a href="#cb1-123" aria-hidden="true" tabindex="-1"></a>tcga<span class="sc">$</span>os_days <span class="ot">&lt;-</span> tcga<span class="sc">$</span>time</span>
<span id="cb1-124"><a href="#cb1-124" aria-hidden="true" tabindex="-1"></a>tcga<span class="sc">$</span>os_months <span class="ot">&lt;-</span> tcga<span class="sc">$</span>os_days<span class="sc">/</span><span class="dv">30</span></span>
<span id="cb1-125"><a href="#cb1-125" aria-hidden="true" tabindex="-1"></a>tcga<span class="sc">$</span>os_status <span class="ot">&lt;-</span> tcga<span class="sc">$</span>status</span>
<span id="cb1-126"><a href="#cb1-126" aria-hidden="true" tabindex="-1"></a>tcga<span class="sc">$</span>tumor_stage <span class="ot">&lt;-</span> <span class="fu">tolower</span>(tcga<span class="sc">$</span>paper_pathologic_stage)</span>
<span id="cb1-127"><a href="#cb1-127" aria-hidden="true" tabindex="-1"></a>tcga<span class="sc">$</span>age <span class="ot">&lt;-</span> tcga<span class="sc">$</span>age_at_index</span>
<span id="cb1-128"><a href="#cb1-128" aria-hidden="true" tabindex="-1"></a>tcga<span class="sc">$</span>sample_name <span class="ot">&lt;-</span> <span class="fu">colnames</span>(tcga)</span>
<span id="cb1-129"><a href="#cb1-129" aria-hidden="true" tabindex="-1"></a>node_stage <span class="ot">&lt;-</span> tcga<span class="sc">$</span>ajcc_pathologic_n</span>
<span id="cb1-130"><a href="#cb1-130" aria-hidden="true" tabindex="-1"></a>tcga<span class="sc">$</span>node_stage <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(</span>
<span id="cb1-131"><a href="#cb1-131" aria-hidden="true" tabindex="-1"></a>    <span class="fu">grepl</span>(<span class="st">"^N0"</span>, node_stage),</span>
<span id="cb1-132"><a href="#cb1-132" aria-hidden="true" tabindex="-1"></a>    <span class="st">"N0"</span>,</span>
<span id="cb1-133"><a href="#cb1-133" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ifelse</span>(</span>
<span id="cb1-134"><a href="#cb1-134" aria-hidden="true" tabindex="-1"></a>        <span class="fu">grepl</span>(<span class="st">"^N1"</span>, node_stage),</span>
<span id="cb1-135"><a href="#cb1-135" aria-hidden="true" tabindex="-1"></a>        <span class="st">"N1"</span>,</span>
<span id="cb1-136"><a href="#cb1-136" aria-hidden="true" tabindex="-1"></a>        <span class="fu">ifelse</span>(</span>
<span id="cb1-137"><a href="#cb1-137" aria-hidden="true" tabindex="-1"></a>            <span class="fu">grepl</span>(<span class="st">"^N2"</span>, node_stage),</span>
<span id="cb1-138"><a href="#cb1-138" aria-hidden="true" tabindex="-1"></a>            <span class="st">"N2"</span>,</span>
<span id="cb1-139"><a href="#cb1-139" aria-hidden="true" tabindex="-1"></a>            <span class="fu">ifelse</span>(</span>
<span id="cb1-140"><a href="#cb1-140" aria-hidden="true" tabindex="-1"></a>                <span class="fu">grepl</span>(<span class="st">"^N3"</span>, node_stage),</span>
<span id="cb1-141"><a href="#cb1-141" aria-hidden="true" tabindex="-1"></a>                <span class="st">"N3"</span>,</span>
<span id="cb1-142"><a href="#cb1-142" aria-hidden="true" tabindex="-1"></a>                <span class="st">"NX"</span></span>
<span id="cb1-143"><a href="#cb1-143" aria-hidden="true" tabindex="-1"></a>            ) </span>
<span id="cb1-144"><a href="#cb1-144" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb1-145"><a href="#cb1-145" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb1-146"><a href="#cb1-146" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-147"><a href="#cb1-147" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-148"><a href="#cb1-148" aria-hidden="true" tabindex="-1"></a>metabric <span class="ot">&lt;-</span> metabric[, metabric<span class="sc">$</span>ER_IHC <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Positve"</span>, <span class="st">"Negative"</span>)]</span>
<span id="cb1-149"><a href="#cb1-149" aria-hidden="true" tabindex="-1"></a>metabric<span class="sc">$</span>er_status <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Positve"</span> <span class="ot">=</span> <span class="st">"pos"</span>, <span class="st">"Negative"</span> <span class="ot">=</span> <span class="st">"neg"</span>)[metabric<span class="sc">$</span>ER_IHC]</span>
<span id="cb1-150"><a href="#cb1-150" aria-hidden="true" tabindex="-1"></a>metabric<span class="sc">$</span>pam50 <span class="ot">&lt;-</span> <span class="fu">tolower</span>(metabric<span class="sc">$</span>CLAUDIN_SUBTYPE)</span>
<span id="cb1-151"><a href="#cb1-151" aria-hidden="true" tabindex="-1"></a>metabric<span class="sc">$</span>os_months <span class="ot">&lt;-</span> metabric<span class="sc">$</span>OS_MONTHS</span>
<span id="cb1-152"><a href="#cb1-152" aria-hidden="true" tabindex="-1"></a>metabric<span class="sc">$</span>rfs_months <span class="ot">&lt;-</span> metabric<span class="sc">$</span>RFS_MONTHS</span>
<span id="cb1-153"><a href="#cb1-153" aria-hidden="true" tabindex="-1"></a>metabric<span class="sc">$</span>os_status <span class="ot">&lt;-</span> <span class="fu">substr</span>(metabric<span class="sc">$</span>OS_STATUS, <span class="dv">1</span>, <span class="dv">1</span>) <span class="sc">%&gt;%</span> as.numeric</span>
<span id="cb1-154"><a href="#cb1-154" aria-hidden="true" tabindex="-1"></a>metabric<span class="sc">$</span>os_status <span class="ot">&lt;-</span> metabric<span class="sc">$</span>os_status <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb1-155"><a href="#cb1-155" aria-hidden="true" tabindex="-1"></a>metabric<span class="sc">$</span>rfs_status <span class="ot">&lt;-</span> <span class="fu">substr</span>(metabric<span class="sc">$</span>RFS_STATUS, <span class="dv">1</span>, <span class="dv">1</span>) <span class="sc">%&gt;%</span> as.numeric</span>
<span id="cb1-156"><a href="#cb1-156" aria-hidden="true" tabindex="-1"></a>metabric<span class="sc">$</span>rfs_status <span class="ot">&lt;-</span> metabric<span class="sc">$</span>rfs_status <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb1-157"><a href="#cb1-157" aria-hidden="true" tabindex="-1"></a>metabric<span class="sc">$</span>age <span class="ot">&lt;-</span> metabric<span class="sc">$</span>AGE_AT_DIAGNOSIS</span>
<span id="cb1-158"><a href="#cb1-158" aria-hidden="true" tabindex="-1"></a>metabric<span class="sc">$</span>npi <span class="ot">&lt;-</span> metabric<span class="sc">$</span>NPI </span>
<span id="cb1-159"><a href="#cb1-159" aria-hidden="true" tabindex="-1"></a>metabric<span class="sc">$</span>NPI <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb1-160"><a href="#cb1-160" aria-hidden="true" tabindex="-1"></a>metabric<span class="sc">$</span>sample_name <span class="ot">&lt;-</span> <span class="fu">colnames</span>(metabric)</span>
<span id="cb1-161"><a href="#cb1-161" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-162"><a href="#cb1-162" aria-hidden="true" tabindex="-1"></a>scanb <span class="ot">&lt;-</span> scanb[, scanb<span class="sc">$</span>ER_1perc <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"NEG"</span>, <span class="st">"POS"</span>)]</span>
<span id="cb1-163"><a href="#cb1-163" aria-hidden="true" tabindex="-1"></a>scanb<span class="sc">$</span>er_status <span class="ot">&lt;-</span> <span class="fu">tolower</span>(scanb<span class="sc">$</span>ER_1perc)</span>
<span id="cb1-164"><a href="#cb1-164" aria-hidden="true" tabindex="-1"></a>scanb<span class="sc">$</span>pam50 <span class="ot">&lt;-</span> <span class="fu">tolower</span>(scanb<span class="sc">$</span>PAM50)</span>
<span id="cb1-165"><a href="#cb1-165" aria-hidden="true" tabindex="-1"></a>scanb<span class="sc">$</span>PAM50 <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb1-166"><a href="#cb1-166" aria-hidden="true" tabindex="-1"></a>scanb<span class="sc">$</span>nhg <span class="ot">&lt;-</span> scanb<span class="sc">$</span>NHG</span>
<span id="cb1-167"><a href="#cb1-167" aria-hidden="true" tabindex="-1"></a>scanb<span class="sc">$</span>NHG <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb1-168"><a href="#cb1-168" aria-hidden="true" tabindex="-1"></a>scanb<span class="sc">$</span>age <span class="ot">&lt;-</span> scanb<span class="sc">$</span>Age</span>
<span id="cb1-169"><a href="#cb1-169" aria-hidden="true" tabindex="-1"></a>scanb<span class="sc">$</span>os_months <span class="ot">&lt;-</span> scanb<span class="sc">$</span>OS_months</span>
<span id="cb1-170"><a href="#cb1-170" aria-hidden="true" tabindex="-1"></a>scanb<span class="sc">$</span>os_status <span class="ot">&lt;-</span> scanb<span class="sc">$</span>OS_event <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb1-171"><a href="#cb1-171" aria-hidden="true" tabindex="-1"></a>scanb<span class="sc">$</span>tumor_size <span class="ot">&lt;-</span> scanb<span class="sc">$</span>TumorSize</span>
<span id="cb1-172"><a href="#cb1-172" aria-hidden="true" tabindex="-1"></a>scanb<span class="sc">$</span>tumor_stage <span class="ot">&lt;-</span> scanb<span class="sc">$</span>TStage</span>
<span id="cb1-173"><a href="#cb1-173" aria-hidden="true" tabindex="-1"></a>scanb<span class="sc">$</span>node_group <span class="ot">&lt;-</span> scanb<span class="sc">$</span>Node_group</span>
<span id="cb1-174"><a href="#cb1-174" aria-hidden="true" tabindex="-1"></a>scanb<span class="sc">$</span>Node_group <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb1-175"><a href="#cb1-175" aria-hidden="true" tabindex="-1"></a>scanb<span class="sc">$</span>sample_name <span class="ot">&lt;-</span> <span class="fu">colnames</span>(scanb)</span>
<span id="cb1-176"><a href="#cb1-176" aria-hidden="true" tabindex="-1"></a>node_stage <span class="ot">&lt;-</span> scanb<span class="sc">$</span>node_group</span>
<span id="cb1-177"><a href="#cb1-177" aria-hidden="true" tabindex="-1"></a>scanb<span class="sc">$</span>node_stage <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(</span>
<span id="cb1-178"><a href="#cb1-178" aria-hidden="true" tabindex="-1"></a>    <span class="fu">grepl</span>(<span class="st">"1to3"</span>, node_stage),</span>
<span id="cb1-179"><a href="#cb1-179" aria-hidden="true" tabindex="-1"></a>    <span class="st">"N1"</span>,</span>
<span id="cb1-180"><a href="#cb1-180" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ifelse</span>(</span>
<span id="cb1-181"><a href="#cb1-181" aria-hidden="true" tabindex="-1"></a>        <span class="fu">grepl</span>(<span class="st">"4toX"</span>, node_stage),</span>
<span id="cb1-182"><a href="#cb1-182" aria-hidden="true" tabindex="-1"></a>        <span class="st">"N2and3"</span>,</span>
<span id="cb1-183"><a href="#cb1-183" aria-hidden="true" tabindex="-1"></a>        <span class="fu">ifelse</span>(</span>
<span id="cb1-184"><a href="#cb1-184" aria-hidden="true" tabindex="-1"></a>            <span class="fu">grepl</span>(<span class="st">"NodeNegative"</span>, node_stage),</span>
<span id="cb1-185"><a href="#cb1-185" aria-hidden="true" tabindex="-1"></a>            <span class="st">"N0"</span>,</span>
<span id="cb1-186"><a href="#cb1-186" aria-hidden="true" tabindex="-1"></a>            <span class="st">"NX"</span></span>
<span id="cb1-187"><a href="#cb1-187" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb1-188"><a href="#cb1-188" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb1-189"><a href="#cb1-189" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-190"><a href="#cb1-190" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb1-191"><a href="#cb1-191" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb1-192"><a href="#cb1-192" aria-hidden="true" tabindex="-1"></a>    <span class="fu">list</span>(<span class="at">tcga =</span> tcga, <span class="at">scanb =</span> scanb, <span class="at">metabric =</span> metabric),</span>
<span id="cb1-193"><a href="#cb1-193" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/surv_analysis_estrogen/datasets.rds"</span></span>
<span id="cb1-194"><a href="#cb1-194" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-195"><a href="#cb1-195" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-196"><a href="#cb1-196" aria-hidden="true" tabindex="-1"></a>which_exp <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb1-197"><a href="#cb1-197" aria-hidden="true" tabindex="-1"></a>    <span class="at">tcga =</span> <span class="st">"logFPKM_TMM"</span>,</span>
<span id="cb1-198"><a href="#cb1-198" aria-hidden="true" tabindex="-1"></a>    <span class="at">scanb =</span> <span class="st">"logFPKM"</span>,</span>
<span id="cb1-199"><a href="#cb1-199" aria-hidden="true" tabindex="-1"></a>    <span class="at">metabric =</span> <span class="st">"median_intensity"</span></span>
<span id="cb1-200"><a href="#cb1-200" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-201"><a href="#cb1-201" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-202"><a href="#cb1-202" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb1-203"><a href="#cb1-203" aria-hidden="true" tabindex="-1"></a>    which_exp,</span>
<span id="cb1-204"><a href="#cb1-204" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/surv_analysis_estrogen/which_exp.rds"</span></span>
<span id="cb1-205"><a href="#cb1-205" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-206"><a href="#cb1-206" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-207"><a href="#cb1-207" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-210"><a href="#cb1-210" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-211"><a href="#cb1-211" aria-hidden="true" tabindex="-1"></a>datasets <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../results/rds_files/surv_analysis_estrogen/datasets.rds"</span>)</span>
<span id="cb1-212"><a href="#cb1-212" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-213"><a href="#cb1-213" aria-hidden="true" tabindex="-1"></a><span class="co"># these are some global parameters for each dataset. important for</span></span>
<span id="cb1-214"><a href="#cb1-214" aria-hidden="true" tabindex="-1"></a><span class="co"># when calculating scores and embeddings.</span></span>
<span id="cb1-215"><a href="#cb1-215" aria-hidden="true" tabindex="-1"></a>which_exp <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../results/rds_files/surv_analysis_estrogen/which_exp.rds"</span>)</span>
<span id="cb1-216"><a href="#cb1-216" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-217"><a href="#cb1-217" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-218"><a href="#cb1-218" aria-hidden="true" tabindex="-1"></a><span class="fu">## UMAP projection of the datasets</span></span>
<span id="cb1-219"><a href="#cb1-219" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-220"><a href="#cb1-220" aria-hidden="true" tabindex="-1"></a>The plots below show how each patient is different in a molecular sense, </span>
<span id="cb1-221"><a href="#cb1-221" aria-hidden="true" tabindex="-1"></a>and even inside each molecular subtype there are some differences. This</span>
<span id="cb1-222"><a href="#cb1-222" aria-hidden="true" tabindex="-1"></a>indicates how different patients, at least molecularly. We only do the umap</span>
<span id="cb1-223"><a href="#cb1-223" aria-hidden="true" tabindex="-1"></a>of samples that have a PAM50 molecular subtype assigned.</span>
<span id="cb1-224"><a href="#cb1-224" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-227"><a href="#cb1-227" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-228"><a href="#cb1-228" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb1-229"><a href="#cb1-229" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-230"><a href="#cb1-230" aria-hidden="true" tabindex="-1"></a><span class="co"># first one uses the library uwot to calculate the umap projection.</span></span>
<span id="cb1-231"><a href="#cb1-231" aria-hidden="true" tabindex="-1"></a><span class="co"># using uwot is better than umap as it allows you to make parallel computations</span></span>
<span id="cb1-232"><a href="#cb1-232" aria-hidden="true" tabindex="-1"></a><span class="co"># in a reproducible way.</span></span>
<span id="cb1-233"><a href="#cb1-233" aria-hidden="true" tabindex="-1"></a>umap_projections <span class="ot">&lt;-</span> <span class="fu">mapply</span>(</span>
<span id="cb1-234"><a href="#cb1-234" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(dataset, name_assay){</span>
<span id="cb1-235"><a href="#cb1-235" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb1-236"><a href="#cb1-236" aria-hidden="true" tabindex="-1"></a>        <span class="fu">cat</span>(<span class="st">"Assay being used:"</span>, name_assay, <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb1-237"><a href="#cb1-237" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb1-238"><a href="#cb1-238" aria-hidden="true" tabindex="-1"></a>        samples_to_use <span class="ot">&lt;-</span> <span class="fu">colData</span>(dataset) <span class="sc">%&gt;%</span> data.frame <span class="sc">%&gt;%</span></span>
<span id="cb1-239"><a href="#cb1-239" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">filter</span>(pam50 <span class="sc">%in%</span> </span>
<span id="cb1-240"><a href="#cb1-240" aria-hidden="true" tabindex="-1"></a>                <span class="fu">c</span>(<span class="st">"basal"</span>, <span class="st">"her2"</span>, <span class="st">"luma"</span>, <span class="st">"lumb"</span>, <span class="st">"normal"</span>)</span>
<span id="cb1-241"><a href="#cb1-241" aria-hidden="true" tabindex="-1"></a>            ) <span class="sc">%&gt;%</span> </span>
<span id="cb1-242"><a href="#cb1-242" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">pull</span>(sample_name)</span>
<span id="cb1-243"><a href="#cb1-243" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb1-244"><a href="#cb1-244" aria-hidden="true" tabindex="-1"></a>        <span class="co"># select most variable genes first</span></span>
<span id="cb1-245"><a href="#cb1-245" aria-hidden="true" tabindex="-1"></a>        df <span class="ot">&lt;-</span> <span class="fu">assay</span>(dataset[, samples_to_use], name_assay) <span class="sc">%&gt;%</span> as.matrix</span>
<span id="cb1-246"><a href="#cb1-246" aria-hidden="true" tabindex="-1"></a>        df <span class="ot">&lt;-</span> df[<span class="fu">order</span>(<span class="fu">rowVars</span>(df), <span class="at">decreasing =</span> <span class="cn">TRUE</span>)[<span class="dv">1</span><span class="sc">:</span><span class="dv">1000</span>], ]</span>
<span id="cb1-247"><a href="#cb1-247" aria-hidden="true" tabindex="-1"></a>        df <span class="ot">&lt;-</span> <span class="fu">t</span>(df)</span>
<span id="cb1-248"><a href="#cb1-248" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb1-249"><a href="#cb1-249" aria-hidden="true" tabindex="-1"></a>        uwot<span class="sc">::</span><span class="fu">umap</span>(</span>
<span id="cb1-250"><a href="#cb1-250" aria-hidden="true" tabindex="-1"></a>            df,</span>
<span id="cb1-251"><a href="#cb1-251" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_threads =</span> <span class="dv">20</span>,</span>
<span id="cb1-252"><a href="#cb1-252" aria-hidden="true" tabindex="-1"></a>            <span class="at">n_sgd_threads =</span> <span class="st">"auto"</span>, </span>
<span id="cb1-253"><a href="#cb1-253" aria-hidden="true" tabindex="-1"></a>            <span class="at">batch =</span> <span class="cn">TRUE</span></span>
<span id="cb1-254"><a href="#cb1-254" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb1-255"><a href="#cb1-255" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb1-256"><a href="#cb1-256" aria-hidden="true" tabindex="-1"></a>    datasets,</span>
<span id="cb1-257"><a href="#cb1-257" aria-hidden="true" tabindex="-1"></a>    which_exp,</span>
<span id="cb1-258"><a href="#cb1-258" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>,</span>
<span id="cb1-259"><a href="#cb1-259" aria-hidden="true" tabindex="-1"></a>    <span class="at">SIMPLIFY =</span> <span class="cn">FALSE</span></span>
<span id="cb1-260"><a href="#cb1-260" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-261"><a href="#cb1-261" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-262"><a href="#cb1-262" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb1-263"><a href="#cb1-263" aria-hidden="true" tabindex="-1"></a>    umap_projections, </span>
<span id="cb1-264"><a href="#cb1-264" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/surv_analysis_estrogen/umap_projections.rds"</span></span>
<span id="cb1-265"><a href="#cb1-265" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-266"><a href="#cb1-266" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-267"><a href="#cb1-267" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-270"><a href="#cb1-270" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-271"><a href="#cb1-271" aria-hidden="true" tabindex="-1"></a>umap_projections <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(</span>
<span id="cb1-272"><a href="#cb1-272" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/surv_analysis_estrogen/umap_projections.rds"</span></span>
<span id="cb1-273"><a href="#cb1-273" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-274"><a href="#cb1-274" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-275"><a href="#cb1-275" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-276"><a href="#cb1-276" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=20}</span></span>
<span id="cb1-277"><a href="#cb1-277" aria-hidden="true" tabindex="-1"></a>plots_umap <span class="ot">&lt;-</span> <span class="fu">mapply</span>(</span>
<span id="cb1-278"><a href="#cb1-278" aria-hidden="true" tabindex="-1"></a>    plot_umap,</span>
<span id="cb1-279"><a href="#cb1-279" aria-hidden="true" tabindex="-1"></a>    <span class="at">sum_exp =</span> datasets,</span>
<span id="cb1-280"><a href="#cb1-280" aria-hidden="true" tabindex="-1"></a>    <span class="at">umap_projection =</span> umap_projections,</span>
<span id="cb1-281"><a href="#cb1-281" aria-hidden="true" tabindex="-1"></a>    <span class="at">cohort =</span> <span class="fu">names</span>(datasets),</span>
<span id="cb1-282"><a href="#cb1-282" aria-hidden="true" tabindex="-1"></a>    <span class="at">MoreArgs =</span> <span class="fu">list</span>(<span class="at">color_by =</span> <span class="st">"pam50"</span>, <span class="at">point_size =</span> <span class="dv">2</span>, <span class="at">base_size =</span> <span class="dv">20</span>),</span>
<span id="cb1-283"><a href="#cb1-283" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>,</span>
<span id="cb1-284"><a href="#cb1-284" aria-hidden="true" tabindex="-1"></a>    <span class="at">SIMPLIFY =</span> <span class="cn">FALSE</span></span>
<span id="cb1-285"><a href="#cb1-285" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-286"><a href="#cb1-286" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-287"><a href="#cb1-287" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(<span class="at">plotlist =</span> plots_umap, <span class="at">ncol =</span> <span class="dv">1</span>)</span>
<span id="cb1-288"><a href="#cb1-288" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-289"><a href="#cb1-289" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-292"><a href="#cb1-292" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-293"><a href="#cb1-293" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb1-294"><a href="#cb1-294" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-295"><a href="#cb1-295" aria-hidden="true" tabindex="-1"></a>width_plot <span class="ot">&lt;-</span> <span class="dv">12</span></span>
<span id="cb1-296"><a href="#cb1-296" aria-hidden="true" tabindex="-1"></a>height_plot <span class="ot">&lt;-</span> width_plot<span class="sc">/</span><span class="fl">1.6</span></span>
<span id="cb1-297"><a href="#cb1-297" aria-hidden="true" tabindex="-1"></a>a <span class="ot">&lt;-</span> <span class="fu">mapply</span>(</span>
<span id="cb1-298"><a href="#cb1-298" aria-hidden="true" tabindex="-1"></a>    ggsave,</span>
<span id="cb1-299"><a href="#cb1-299" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot =</span> plots_umap,</span>
<span id="cb1-300"><a href="#cb1-300" aria-hidden="true" tabindex="-1"></a>    <span class="at">filename =</span> <span class="fu">paste0</span>(</span>
<span id="cb1-301"><a href="#cb1-301" aria-hidden="true" tabindex="-1"></a>        <span class="st">"../results/plots/surv_analysis_estrogen/umap/"</span>,</span>
<span id="cb1-302"><a href="#cb1-302" aria-hidden="true" tabindex="-1"></a>        <span class="fu">names</span>(plots_umap),</span>
<span id="cb1-303"><a href="#cb1-303" aria-hidden="true" tabindex="-1"></a>        <span class="st">".pdf"</span></span>
<span id="cb1-304"><a href="#cb1-304" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb1-305"><a href="#cb1-305" aria-hidden="true" tabindex="-1"></a>    <span class="at">MoreArgs =</span> <span class="fu">list</span>(<span class="at">width =</span> width_plot, <span class="at">height =</span> height_plot)</span>
<span id="cb1-306"><a href="#cb1-306" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-307"><a href="#cb1-307" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-308"><a href="#cb1-308" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-309"><a href="#cb1-309" aria-hidden="true" tabindex="-1"></a>From the plots above we see a distinction of the different molecular subtypes.</span>
<span id="cb1-310"><a href="#cb1-310" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-311"><a href="#cb1-311" aria-hidden="true" tabindex="-1"></a><span class="fu">## Calculating scores</span></span>
<span id="cb1-312"><a href="#cb1-312" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-313"><a href="#cb1-313" aria-hidden="true" tabindex="-1"></a>In order to calculate the scores, the package <span class="in">`msigdb`</span> is used to load</span>
<span id="cb1-314"><a href="#cb1-314" aria-hidden="true" tabindex="-1"></a>the hallmark data into R. The $SET_{ER/PR}$ is made of the following genes:</span>
<span id="cb1-315"><a href="#cb1-315" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-318"><a href="#cb1-318" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-319"><a href="#cb1-319" aria-hidden="true" tabindex="-1"></a><span class="co"># data is available in the supplementary material of the paper </span></span>
<span id="cb1-320"><a href="#cb1-320" aria-hidden="true" tabindex="-1"></a>seterpr <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"../data/seterpr.tsv"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb1-321"><a href="#cb1-321" aria-hidden="true" tabindex="-1"></a>seterpr<span class="sc">$</span>is_target <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">"yes"</span>, <span class="dv">18</span>), <span class="fu">rep</span>(<span class="st">"no"</span>, <span class="dv">10</span>))</span>
<span id="cb1-322"><a href="#cb1-322" aria-hidden="true" tabindex="-1"></a>seterpr <span class="sc">%&gt;%</span> kableExtra<span class="sc">::</span><span class="fu">kbl</span>() <span class="sc">%&gt;%</span></span>
<span id="cb1-323"><a href="#cb1-323" aria-hidden="true" tabindex="-1"></a>    kableExtra<span class="sc">::</span><span class="fu">kable_classic</span>(<span class="at">full_width =</span> <span class="cn">FALSE</span>)</span>
<span id="cb1-324"><a href="#cb1-324" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-325"><a href="#cb1-325" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-326"><a href="#cb1-326" aria-hidden="true" tabindex="-1"></a>The first 18 genes are considered to be the target genes, the last 10 genes</span>
<span id="cb1-327"><a href="#cb1-327" aria-hidden="true" tabindex="-1"></a>are the genes used for reference. According to their paper, the score is </span>
<span id="cb1-328"><a href="#cb1-328" aria-hidden="true" tabindex="-1"></a>calculated in the following way:</span>
<span id="cb1-329"><a href="#cb1-329" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-330"><a href="#cb1-330" aria-hidden="true" tabindex="-1"></a>$$</span>
<span id="cb1-331"><a href="#cb1-331" aria-hidden="true" tabindex="-1"></a>SET_{ER/PR} = \sum_{i = 1}^{18} \frac{T_i}{18} - \sum_{j=1}^{10}\frac{R_j}{10} + 2</span>
<span id="cb1-332"><a href="#cb1-332" aria-hidden="true" tabindex="-1"></a>$$</span>
<span id="cb1-333"><a href="#cb1-333" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-334"><a href="#cb1-334" aria-hidden="true" tabindex="-1"></a>Where $T_i$ are the expression levels of target genes and $R_j$ are the</span>
<span id="cb1-335"><a href="#cb1-335" aria-hidden="true" tabindex="-1"></a>expression levels of the reference genes. Here we use GSVA to calculate</span>
<span id="cb1-336"><a href="#cb1-336" aria-hidden="true" tabindex="-1"></a>the scores, even when using their genes. </span>
<span id="cb1-337"><a href="#cb1-337" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-340"><a href="#cb1-340" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-341"><a href="#cb1-341" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb1-342"><a href="#cb1-342" aria-hidden="true" tabindex="-1"></a>all_genes <span class="ot">&lt;-</span> <span class="fu">sapply</span>(datasets, rownames) <span class="sc">%&gt;%</span> unlist <span class="sc">%&gt;%</span> unique</span>
<span id="cb1-343"><a href="#cb1-343" aria-hidden="true" tabindex="-1"></a><span class="co"># now load the msigdbr and concatenate with set erpr</span></span>
<span id="cb1-344"><a href="#cb1-344" aria-hidden="true" tabindex="-1"></a>gene_sets <span class="ot">&lt;-</span> msigdbr<span class="sc">::</span><span class="fu">msigdbr</span>(<span class="at">species =</span> <span class="st">"Homo sapiens"</span>, <span class="at">category =</span> <span class="st">"H"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb1-345"><a href="#cb1-345" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(gs_name, gene_symbol) <span class="sc">%&gt;%</span></span>
<span id="cb1-346"><a href="#cb1-346" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(</span>
<span id="cb1-347"><a href="#cb1-347" aria-hidden="true" tabindex="-1"></a>        .,</span>
<span id="cb1-348"><a href="#cb1-348" aria-hidden="true" tabindex="-1"></a>        seterpr <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(is_target <span class="sc">==</span> <span class="st">"yes"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb1-349"><a href="#cb1-349" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">select</span>(ID) <span class="sc">%&gt;%</span></span>
<span id="cb1-350"><a href="#cb1-350" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">gene_symbol =</span> ID) <span class="sc">%&gt;%</span></span>
<span id="cb1-351"><a href="#cb1-351" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">gs_name =</span> <span class="st">"SET_ERPR"</span>)</span>
<span id="cb1-352"><a href="#cb1-352" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb1-353"><a href="#cb1-353" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(</span>
<span id="cb1-354"><a href="#cb1-354" aria-hidden="true" tabindex="-1"></a>        .,</span>
<span id="cb1-355"><a href="#cb1-355" aria-hidden="true" tabindex="-1"></a>        <span class="fu">data.frame</span>(<span class="at">gene_symbol =</span> <span class="fu">sample</span>(all_genes, <span class="dv">200</span>), <span class="at">gs_name =</span> <span class="st">"random_200"</span>)</span>
<span id="cb1-356"><a href="#cb1-356" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb1-357"><a href="#cb1-357" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(</span>
<span id="cb1-358"><a href="#cb1-358" aria-hidden="true" tabindex="-1"></a>        .,</span>
<span id="cb1-359"><a href="#cb1-359" aria-hidden="true" tabindex="-1"></a>        <span class="fu">data.frame</span>(<span class="at">gene_symbol =</span> <span class="fu">sample</span>(all_genes, <span class="dv">18</span>), <span class="at">gs_name =</span> <span class="st">"random_18"</span>)</span>
<span id="cb1-360"><a href="#cb1-360" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb1-361"><a href="#cb1-361" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-362"><a href="#cb1-362" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(gene_sets, <span class="st">"../results/rds_files/surv_analysis_estrogen/gene_sets.rds"</span>)</span>
<span id="cb1-363"><a href="#cb1-363" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-364"><a href="#cb1-364" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-365"><a href="#cb1-365" aria-hidden="true" tabindex="-1"></a>Before we calculate any score, let us check the number of genes available</span>
<span id="cb1-366"><a href="#cb1-366" aria-hidden="true" tabindex="-1"></a>for each pathway in each dataset. This is important, in other to have</span>
<span id="cb1-367"><a href="#cb1-367" aria-hidden="true" tabindex="-1"></a>robust scores, most of the genes should be available in the datasets. We also</span>
<span id="cb1-368"><a href="#cb1-368" aria-hidden="true" tabindex="-1"></a>add signatures of 18 and 200 random genes for control. </span>
<span id="cb1-369"><a href="#cb1-369" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-372"><a href="#cb1-372" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-373"><a href="#cb1-373" aria-hidden="true" tabindex="-1"></a>gene_sets <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../results/rds_files/surv_analysis_estrogen/gene_sets.rds"</span>)</span>
<span id="cb1-374"><a href="#cb1-374" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-375"><a href="#cb1-375" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-378"><a href="#cb1-378" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-379"><a href="#cb1-379" aria-hidden="true" tabindex="-1"></a>genes_each_dataset <span class="ot">&lt;-</span> <span class="fu">lapply</span>(datasets, rownames)</span>
<span id="cb1-380"><a href="#cb1-380" aria-hidden="true" tabindex="-1"></a>genes_intersection <span class="ot">&lt;-</span> <span class="fu">sapply</span>(</span>
<span id="cb1-381"><a href="#cb1-381" aria-hidden="true" tabindex="-1"></a>    gene_sets<span class="sc">$</span>gs_name <span class="sc">%&gt;%</span> unique,</span>
<span id="cb1-382"><a href="#cb1-382" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(set_name, gene_sets, genes_each_dataset){</span>
<span id="cb1-383"><a href="#cb1-383" aria-hidden="true" tabindex="-1"></a>        <span class="fu">lapply</span>(</span>
<span id="cb1-384"><a href="#cb1-384" aria-hidden="true" tabindex="-1"></a>            genes_each_dataset,</span>
<span id="cb1-385"><a href="#cb1-385" aria-hidden="true" tabindex="-1"></a>            intersect,</span>
<span id="cb1-386"><a href="#cb1-386" aria-hidden="true" tabindex="-1"></a>            <span class="at">y =</span> gene_sets <span class="sc">%&gt;%</span> </span>
<span id="cb1-387"><a href="#cb1-387" aria-hidden="true" tabindex="-1"></a>                dplyr<span class="sc">::</span><span class="fu">filter</span>(gs_name <span class="sc">==</span> set_name) <span class="sc">%&gt;%</span></span>
<span id="cb1-388"><a href="#cb1-388" aria-hidden="true" tabindex="-1"></a>                dplyr<span class="sc">::</span><span class="fu">pull</span>(gene_symbol)</span>
<span id="cb1-389"><a href="#cb1-389" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb1-390"><a href="#cb1-390" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb1-391"><a href="#cb1-391" aria-hidden="true" tabindex="-1"></a>    <span class="at">gene_sets =</span> gene_sets,</span>
<span id="cb1-392"><a href="#cb1-392" aria-hidden="true" tabindex="-1"></a>    <span class="at">genes_each_dataset =</span> genes_each_dataset,</span>
<span id="cb1-393"><a href="#cb1-393" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>,</span>
<span id="cb1-394"><a href="#cb1-394" aria-hidden="true" tabindex="-1"></a>    <span class="at">simplify =</span> <span class="cn">FALSE</span></span>
<span id="cb1-395"><a href="#cb1-395" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-396"><a href="#cb1-396" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-397"><a href="#cb1-397" aria-hidden="true" tabindex="-1"></a><span class="fu">sapply</span>(genes_intersection, <span class="cf">function</span>(x) <span class="fu">sapply</span>(x, length)) <span class="sc">%&gt;%</span> t <span class="sc">%&gt;%</span></span>
<span id="cb1-398"><a href="#cb1-398" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span></span>
<span id="cb1-399"><a href="#cb1-399" aria-hidden="true" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"pathway"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb1-400"><a href="#cb1-400" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">inner_join</span>(</span>
<span id="cb1-401"><a href="#cb1-401" aria-hidden="true" tabindex="-1"></a>        .,</span>
<span id="cb1-402"><a href="#cb1-402" aria-hidden="true" tabindex="-1"></a>        gene_sets <span class="sc">%&gt;%</span> </span>
<span id="cb1-403"><a href="#cb1-403" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">group_by</span>(gs_name) <span class="sc">%&gt;%</span></span>
<span id="cb1-404"><a href="#cb1-404" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">n =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb1-405"><a href="#cb1-405" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">pathway =</span> gs_name),</span>
<span id="cb1-406"><a href="#cb1-406" aria-hidden="true" tabindex="-1"></a>        <span class="at">by =</span> <span class="st">"pathway"</span></span>
<span id="cb1-407"><a href="#cb1-407" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span> </span>
<span id="cb1-408"><a href="#cb1-408" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb1-409"><a href="#cb1-409" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">average_percentage =</span> <span class="fu">format</span>(<span class="fu">mean</span>(</span>
<span id="cb1-410"><a href="#cb1-410" aria-hidden="true" tabindex="-1"></a>            <span class="fu">c</span>(tcga<span class="sc">/</span>n, scanb<span class="sc">/</span>n, metabric<span class="sc">/</span>n)</span>
<span id="cb1-411"><a href="#cb1-411" aria-hidden="true" tabindex="-1"></a>        ),</span>
<span id="cb1-412"><a href="#cb1-412" aria-hidden="true" tabindex="-1"></a>        <span class="at">digits =</span> <span class="dv">2</span></span>
<span id="cb1-413"><a href="#cb1-413" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">%&gt;%</span> </span>
<span id="cb1-414"><a href="#cb1-414" aria-hidden="true" tabindex="-1"></a>    kableExtra<span class="sc">::</span><span class="fu">kbl</span>() <span class="sc">%&gt;%</span></span>
<span id="cb1-415"><a href="#cb1-415" aria-hidden="true" tabindex="-1"></a>    kableExtra<span class="sc">::</span><span class="fu">kable_classic</span>(<span class="at">full_width =</span> <span class="cn">FALSE</span>)</span>
<span id="cb1-416"><a href="#cb1-416" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-417"><a href="#cb1-417" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-418"><a href="#cb1-418" aria-hidden="true" tabindex="-1"></a>Most of the genes are available in all datasets. When calculating scores</span>
<span id="cb1-419"><a href="#cb1-419" aria-hidden="true" tabindex="-1"></a>it is always good to check the availability of the genes. Otherwise this</span>
<span id="cb1-420"><a href="#cb1-420" aria-hidden="true" tabindex="-1"></a>can make the score unstable, since too many of the genes are missing. </span>
<span id="cb1-421"><a href="#cb1-421" aria-hidden="true" tabindex="-1"></a>For example, </span>
<span id="cb1-422"><a href="#cb1-422" aria-hidden="true" tabindex="-1"></a>HALLMARK_PANCREAS_BETA_CELLS might have an unstable score, due to a lot </span>
<span id="cb1-423"><a href="#cb1-423" aria-hidden="true" tabindex="-1"></a>of genes missing. </span>
<span id="cb1-424"><a href="#cb1-424" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-427"><a href="#cb1-427" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-428"><a href="#cb1-428" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb1-429"><a href="#cb1-429" aria-hidden="true" tabindex="-1"></a>which_exp <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb1-430"><a href="#cb1-430" aria-hidden="true" tabindex="-1"></a>    <span class="at">tcga =</span> <span class="st">"logFPKM_TMM"</span>,</span>
<span id="cb1-431"><a href="#cb1-431" aria-hidden="true" tabindex="-1"></a>    <span class="at">scanb =</span> <span class="st">"logFPKM"</span>,</span>
<span id="cb1-432"><a href="#cb1-432" aria-hidden="true" tabindex="-1"></a>    <span class="at">metabric =</span> <span class="st">"median_intensity"</span></span>
<span id="cb1-433"><a href="#cb1-433" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-434"><a href="#cb1-434" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-435"><a href="#cb1-435" aria-hidden="true" tabindex="-1"></a>gene_sets <span class="ot">&lt;-</span> <span class="fu">sapply</span>(</span>
<span id="cb1-436"><a href="#cb1-436" aria-hidden="true" tabindex="-1"></a>    gene_sets<span class="sc">$</span>gs_name <span class="sc">%&gt;%</span> unique,</span>
<span id="cb1-437"><a href="#cb1-437" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(x, gene_sets) gene_sets <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(gs_name <span class="sc">==</span> x) <span class="sc">%&gt;%</span> </span>
<span id="cb1-438"><a href="#cb1-438" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">pull</span>(gene_symbol),</span>
<span id="cb1-439"><a href="#cb1-439" aria-hidden="true" tabindex="-1"></a>    <span class="at">gene_sets =</span> gene_sets,</span>
<span id="cb1-440"><a href="#cb1-440" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>,</span>
<span id="cb1-441"><a href="#cb1-441" aria-hidden="true" tabindex="-1"></a>    <span class="at">simplify =</span> <span class="cn">FALSE</span></span>
<span id="cb1-442"><a href="#cb1-442" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-443"><a href="#cb1-443" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-444"><a href="#cb1-444" aria-hidden="true" tabindex="-1"></a>gsva_scores <span class="ot">&lt;-</span> <span class="fu">mapply</span>(</span>
<span id="cb1-445"><a href="#cb1-445" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(sum_exp, which_assay, gene_sets){</span>
<span id="cb1-446"><a href="#cb1-446" aria-hidden="true" tabindex="-1"></a>        GSVA<span class="sc">::</span><span class="fu">gsva</span>(</span>
<span id="cb1-447"><a href="#cb1-447" aria-hidden="true" tabindex="-1"></a>            <span class="at">expr =</span> <span class="fu">as.matrix</span>(<span class="fu">assay</span>(sum_exp, which_assay)),</span>
<span id="cb1-448"><a href="#cb1-448" aria-hidden="true" tabindex="-1"></a>            <span class="at">gset.idx.list =</span> gene_sets, </span>
<span id="cb1-449"><a href="#cb1-449" aria-hidden="true" tabindex="-1"></a>            <span class="at">parallel.sz =</span> <span class="dv">20</span></span>
<span id="cb1-450"><a href="#cb1-450" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb1-451"><a href="#cb1-451" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb1-452"><a href="#cb1-452" aria-hidden="true" tabindex="-1"></a>    <span class="at">sum_exp =</span> datasets,</span>
<span id="cb1-453"><a href="#cb1-453" aria-hidden="true" tabindex="-1"></a>    <span class="at">which_assay =</span> which_exp,</span>
<span id="cb1-454"><a href="#cb1-454" aria-hidden="true" tabindex="-1"></a>    <span class="at">MoreArgs =</span> <span class="fu">list</span>(<span class="at">gene_sets =</span> gene_sets),</span>
<span id="cb1-455"><a href="#cb1-455" aria-hidden="true" tabindex="-1"></a>    <span class="at">SIMPLIFY =</span> <span class="cn">FALSE</span>,</span>
<span id="cb1-456"><a href="#cb1-456" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span></span>
<span id="cb1-457"><a href="#cb1-457" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-458"><a href="#cb1-458" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-459"><a href="#cb1-459" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb1-460"><a href="#cb1-460" aria-hidden="true" tabindex="-1"></a>    gsva_scores, </span>
<span id="cb1-461"><a href="#cb1-461" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/surv_analysis_estrogen/gsva_scores.rds"</span></span>
<span id="cb1-462"><a href="#cb1-462" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-463"><a href="#cb1-463" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-464"><a href="#cb1-464" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-467"><a href="#cb1-467" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-468"><a href="#cb1-468" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb1-469"><a href="#cb1-469" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-470"><a href="#cb1-470" aria-hidden="true" tabindex="-1"></a>gsva_scores <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(</span>
<span id="cb1-471"><a href="#cb1-471" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/surv_analysis_estrogen/gsva_scores.rds"</span></span>
<span id="cb1-472"><a href="#cb1-472" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-473"><a href="#cb1-473" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-474"><a href="#cb1-474" aria-hidden="true" tabindex="-1"></a>datasets <span class="ot">&lt;-</span> <span class="fu">mapply</span>(</span>
<span id="cb1-475"><a href="#cb1-475" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(gsva_score, dataset){</span>
<span id="cb1-476"><a href="#cb1-476" aria-hidden="true" tabindex="-1"></a>        <span class="fu">colData</span>(dataset)[, <span class="fu">rownames</span>(gsva_score)] <span class="ot">&lt;-</span> <span class="fu">t</span>(gsva_score)</span>
<span id="cb1-477"><a href="#cb1-477" aria-hidden="true" tabindex="-1"></a>        dataset</span>
<span id="cb1-478"><a href="#cb1-478" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb1-479"><a href="#cb1-479" aria-hidden="true" tabindex="-1"></a>    gsva_scores,</span>
<span id="cb1-480"><a href="#cb1-480" aria-hidden="true" tabindex="-1"></a>    datasets,</span>
<span id="cb1-481"><a href="#cb1-481" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>,</span>
<span id="cb1-482"><a href="#cb1-482" aria-hidden="true" tabindex="-1"></a>    <span class="at">SIMPLIFY =</span> <span class="cn">FALSE</span></span>
<span id="cb1-483"><a href="#cb1-483" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-484"><a href="#cb1-484" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-485"><a href="#cb1-485" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb1-486"><a href="#cb1-486" aria-hidden="true" tabindex="-1"></a>    datasets,</span>
<span id="cb1-487"><a href="#cb1-487" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/surv_analysis_estrogen/datasets_with_scores.rds"</span></span>
<span id="cb1-488"><a href="#cb1-488" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-489"><a href="#cb1-489" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-490"><a href="#cb1-490" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-493"><a href="#cb1-493" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-494"><a href="#cb1-494" aria-hidden="true" tabindex="-1"></a>gsva_scores <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(</span>
<span id="cb1-495"><a href="#cb1-495" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/surv_analysis_estrogen/gsva_scores.rds"</span></span>
<span id="cb1-496"><a href="#cb1-496" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-497"><a href="#cb1-497" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-498"><a href="#cb1-498" aria-hidden="true" tabindex="-1"></a>datasets <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(</span>
<span id="cb1-499"><a href="#cb1-499" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/surv_analysis_estrogen/datasets_with_scores.rds"</span></span>
<span id="cb1-500"><a href="#cb1-500" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-501"><a href="#cb1-501" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-502"><a href="#cb1-502" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-503"><a href="#cb1-503" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-504"><a href="#cb1-504" aria-hidden="true" tabindex="-1"></a>For each dataset one can plot the differences in scores for ER+ and ER- BC</span>
<span id="cb1-505"><a href="#cb1-505" aria-hidden="true" tabindex="-1"></a>patients. This should be already an indication that the scores are meaningful.</span>
<span id="cb1-506"><a href="#cb1-506" aria-hidden="true" tabindex="-1"></a>The next sections shows the results for each dataset individually.</span>
<span id="cb1-507"><a href="#cb1-507" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-510"><a href="#cb1-510" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-511"><a href="#cb1-511" aria-hidden="true" tabindex="-1"></a>plot_scores <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb1-512"><a href="#cb1-512" aria-hidden="true" tabindex="-1"></a>which_scores <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb1-513"><a href="#cb1-513" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_ESTROGEN_RESPONSE_EARLY"</span>,</span>
<span id="cb1-514"><a href="#cb1-514" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_ESTROGEN_RESPONSE_LATE"</span>,</span>
<span id="cb1-515"><a href="#cb1-515" aria-hidden="true" tabindex="-1"></a>    <span class="st">"SET_ERPR"</span>,</span>
<span id="cb1-516"><a href="#cb1-516" aria-hidden="true" tabindex="-1"></a>    <span class="st">"random_200"</span>,</span>
<span id="cb1-517"><a href="#cb1-517" aria-hidden="true" tabindex="-1"></a>    <span class="st">"random_18"</span></span>
<span id="cb1-518"><a href="#cb1-518" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-519"><a href="#cb1-519" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-520"><a href="#cb1-520" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-521"><a href="#cb1-521" aria-hidden="true" tabindex="-1"></a>::: {.panel-tabset}</span>
<span id="cb1-522"><a href="#cb1-522" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-523"><a href="#cb1-523" aria-hidden="true" tabindex="-1"></a><span class="fu">### TCGA</span></span>
<span id="cb1-524"><a href="#cb1-524" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-527"><a href="#cb1-527" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-528"><a href="#cb1-528" aria-hidden="true" tabindex="-1"></a>which_dataset <span class="ot">&lt;-</span> <span class="st">"tcga"</span></span>
<span id="cb1-529"><a href="#cb1-529" aria-hidden="true" tabindex="-1"></a>plot_scores[[which_dataset]] <span class="ot">&lt;-</span> <span class="fu">mapply</span>(</span>
<span id="cb1-530"><a href="#cb1-530" aria-hidden="true" tabindex="-1"></a>    plot_scores_vs_clinics,</span>
<span id="cb1-531"><a href="#cb1-531" aria-hidden="true" tabindex="-1"></a>    <span class="at">which_score =</span> which_scores,</span>
<span id="cb1-532"><a href="#cb1-532" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> which_scores,</span>
<span id="cb1-533"><a href="#cb1-533" aria-hidden="true" tabindex="-1"></a>    <span class="at">MoreArgs =</span> <span class="fu">list</span>(</span>
<span id="cb1-534"><a href="#cb1-534" aria-hidden="true" tabindex="-1"></a>        <span class="at">sum_exp =</span> datasets[[which_dataset]],</span>
<span id="cb1-535"><a href="#cb1-535" aria-hidden="true" tabindex="-1"></a>        <span class="at">clinical_variable =</span> <span class="st">"er_status"</span>, </span>
<span id="cb1-536"><a href="#cb1-536" aria-hidden="true" tabindex="-1"></a>        <span class="at">color_by =</span> <span class="st">"pam50"</span>,</span>
<span id="cb1-537"><a href="#cb1-537" aria-hidden="true" tabindex="-1"></a>        <span class="at">base_size =</span> <span class="dv">10</span>,</span>
<span id="cb1-538"><a href="#cb1-538" aria-hidden="true" tabindex="-1"></a>        <span class="at">point_size =</span> <span class="fl">0.5</span></span>
<span id="cb1-539"><a href="#cb1-539" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb1-540"><a href="#cb1-540" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>,</span>
<span id="cb1-541"><a href="#cb1-541" aria-hidden="true" tabindex="-1"></a>    <span class="at">SIMPLIFY =</span> <span class="cn">FALSE</span></span>
<span id="cb1-542"><a href="#cb1-542" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-543"><a href="#cb1-543" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-544"><a href="#cb1-544" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-545"><a href="#cb1-545" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=8}</span></span>
<span id="cb1-546"><a href="#cb1-546" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(<span class="at">plotlist =</span> plot_scores[[which_dataset]], <span class="at">ncol =</span> <span class="dv">2</span>)</span>
<span id="cb1-547"><a href="#cb1-547" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-548"><a href="#cb1-548" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-549"><a href="#cb1-549" aria-hidden="true" tabindex="-1"></a><span class="fu">### METABRIC</span></span>
<span id="cb1-550"><a href="#cb1-550" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-553"><a href="#cb1-553" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-554"><a href="#cb1-554" aria-hidden="true" tabindex="-1"></a>which_dataset <span class="ot">&lt;-</span> <span class="st">"metabric"</span></span>
<span id="cb1-555"><a href="#cb1-555" aria-hidden="true" tabindex="-1"></a>plot_scores[[which_dataset]] <span class="ot">&lt;-</span> <span class="fu">mapply</span>(</span>
<span id="cb1-556"><a href="#cb1-556" aria-hidden="true" tabindex="-1"></a>    plot_scores_vs_clinics,</span>
<span id="cb1-557"><a href="#cb1-557" aria-hidden="true" tabindex="-1"></a>    <span class="at">which_score =</span> which_scores,</span>
<span id="cb1-558"><a href="#cb1-558" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> which_scores,</span>
<span id="cb1-559"><a href="#cb1-559" aria-hidden="true" tabindex="-1"></a>    <span class="at">MoreArgs =</span> <span class="fu">list</span>(</span>
<span id="cb1-560"><a href="#cb1-560" aria-hidden="true" tabindex="-1"></a>        <span class="at">sum_exp =</span> datasets[[which_dataset]],</span>
<span id="cb1-561"><a href="#cb1-561" aria-hidden="true" tabindex="-1"></a>        <span class="at">clinical_variable =</span> <span class="st">"er_status"</span>, </span>
<span id="cb1-562"><a href="#cb1-562" aria-hidden="true" tabindex="-1"></a>        <span class="at">color_by =</span> <span class="st">"pam50"</span>,</span>
<span id="cb1-563"><a href="#cb1-563" aria-hidden="true" tabindex="-1"></a>        <span class="at">base_size =</span> <span class="dv">10</span>,</span>
<span id="cb1-564"><a href="#cb1-564" aria-hidden="true" tabindex="-1"></a>        <span class="at">point_size =</span> <span class="fl">0.5</span></span>
<span id="cb1-565"><a href="#cb1-565" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb1-566"><a href="#cb1-566" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>,</span>
<span id="cb1-567"><a href="#cb1-567" aria-hidden="true" tabindex="-1"></a>    <span class="at">SIMPLIFY =</span> <span class="cn">FALSE</span></span>
<span id="cb1-568"><a href="#cb1-568" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-569"><a href="#cb1-569" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-570"><a href="#cb1-570" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-571"><a href="#cb1-571" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=8}</span></span>
<span id="cb1-572"><a href="#cb1-572" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(<span class="at">plotlist =</span> plot_scores[[which_dataset]], <span class="at">ncol =</span> <span class="dv">2</span>)</span>
<span id="cb1-573"><a href="#cb1-573" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-574"><a href="#cb1-574" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-575"><a href="#cb1-575" aria-hidden="true" tabindex="-1"></a><span class="fu">### SCANB</span></span>
<span id="cb1-576"><a href="#cb1-576" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-579"><a href="#cb1-579" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-580"><a href="#cb1-580" aria-hidden="true" tabindex="-1"></a>which_dataset <span class="ot">&lt;-</span> <span class="st">"scanb"</span></span>
<span id="cb1-581"><a href="#cb1-581" aria-hidden="true" tabindex="-1"></a>plot_scores[[which_dataset]] <span class="ot">&lt;-</span> <span class="fu">mapply</span>(</span>
<span id="cb1-582"><a href="#cb1-582" aria-hidden="true" tabindex="-1"></a>    plot_scores_vs_clinics,</span>
<span id="cb1-583"><a href="#cb1-583" aria-hidden="true" tabindex="-1"></a>    <span class="at">which_score =</span> which_scores,</span>
<span id="cb1-584"><a href="#cb1-584" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> which_scores,</span>
<span id="cb1-585"><a href="#cb1-585" aria-hidden="true" tabindex="-1"></a>    <span class="at">MoreArgs =</span> <span class="fu">list</span>(</span>
<span id="cb1-586"><a href="#cb1-586" aria-hidden="true" tabindex="-1"></a>        <span class="at">sum_exp =</span> datasets[[which_dataset]],</span>
<span id="cb1-587"><a href="#cb1-587" aria-hidden="true" tabindex="-1"></a>        <span class="at">clinical_variable =</span> <span class="st">"er_status"</span>, </span>
<span id="cb1-588"><a href="#cb1-588" aria-hidden="true" tabindex="-1"></a>        <span class="at">color_by =</span> <span class="st">"pam50"</span>,</span>
<span id="cb1-589"><a href="#cb1-589" aria-hidden="true" tabindex="-1"></a>        <span class="at">base_size =</span> <span class="dv">10</span>,</span>
<span id="cb1-590"><a href="#cb1-590" aria-hidden="true" tabindex="-1"></a>        <span class="at">point_size =</span> <span class="fl">0.5</span></span>
<span id="cb1-591"><a href="#cb1-591" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb1-592"><a href="#cb1-592" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>,</span>
<span id="cb1-593"><a href="#cb1-593" aria-hidden="true" tabindex="-1"></a>    <span class="at">SIMPLIFY =</span> <span class="cn">FALSE</span></span>
<span id="cb1-594"><a href="#cb1-594" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-595"><a href="#cb1-595" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-596"><a href="#cb1-596" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-597"><a href="#cb1-597" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=8}</span></span>
<span id="cb1-598"><a href="#cb1-598" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(<span class="at">plotlist =</span> plot_scores[[which_dataset]], <span class="at">ncol =</span> <span class="dv">2</span>)</span>
<span id="cb1-599"><a href="#cb1-599" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-600"><a href="#cb1-600" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-601"><a href="#cb1-601" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb1-602"><a href="#cb1-602" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-603"><a href="#cb1-603" aria-hidden="true" tabindex="-1"></a>From the plots above one can conclude that three different estrogen pathways</span>
<span id="cb1-604"><a href="#cb1-604" aria-hidden="true" tabindex="-1"></a>capture the differences between ER status and also molecular subtypes.</span>
<span id="cb1-605"><a href="#cb1-605" aria-hidden="true" tabindex="-1"></a>There are two plots with random genes for control and we can see that</span>
<span id="cb1-606"><a href="#cb1-606" aria-hidden="true" tabindex="-1"></a>there is no difference between the ER status when using those gene sets.</span>
<span id="cb1-607"><a href="#cb1-607" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-608"><a href="#cb1-608" aria-hidden="true" tabindex="-1"></a>Another way to look at the data is to plot by molecular subtype instead</span>
<span id="cb1-609"><a href="#cb1-609" aria-hidden="true" tabindex="-1"></a>of ER status.</span>
<span id="cb1-610"><a href="#cb1-610" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-611"><a href="#cb1-611" aria-hidden="true" tabindex="-1"></a>::: {.panel-tabset}</span>
<span id="cb1-612"><a href="#cb1-612" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-613"><a href="#cb1-613" aria-hidden="true" tabindex="-1"></a><span class="fu">### TCGA</span></span>
<span id="cb1-614"><a href="#cb1-614" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-617"><a href="#cb1-617" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-618"><a href="#cb1-618" aria-hidden="true" tabindex="-1"></a>clinical_variable <span class="ot">&lt;-</span> <span class="st">"pam50"</span></span>
<span id="cb1-619"><a href="#cb1-619" aria-hidden="true" tabindex="-1"></a>color_by <span class="ot">&lt;-</span> <span class="st">"er_status"</span></span>
<span id="cb1-620"><a href="#cb1-620" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-621"><a href="#cb1-621" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-622"><a href="#cb1-622" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-625"><a href="#cb1-625" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-626"><a href="#cb1-626" aria-hidden="true" tabindex="-1"></a>which_dataset <span class="ot">&lt;-</span> <span class="st">"tcga"</span></span>
<span id="cb1-627"><a href="#cb1-627" aria-hidden="true" tabindex="-1"></a>plot_scores[[which_dataset]] <span class="ot">&lt;-</span> <span class="fu">mapply</span>(</span>
<span id="cb1-628"><a href="#cb1-628" aria-hidden="true" tabindex="-1"></a>    plot_scores_vs_clinics,</span>
<span id="cb1-629"><a href="#cb1-629" aria-hidden="true" tabindex="-1"></a>    <span class="at">which_score =</span> which_scores,</span>
<span id="cb1-630"><a href="#cb1-630" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> which_scores,</span>
<span id="cb1-631"><a href="#cb1-631" aria-hidden="true" tabindex="-1"></a>    <span class="at">MoreArgs =</span> <span class="fu">list</span>(</span>
<span id="cb1-632"><a href="#cb1-632" aria-hidden="true" tabindex="-1"></a>        <span class="at">sum_exp =</span> datasets[[which_dataset]],</span>
<span id="cb1-633"><a href="#cb1-633" aria-hidden="true" tabindex="-1"></a>        <span class="at">clinical_variable =</span> clinical_variable, </span>
<span id="cb1-634"><a href="#cb1-634" aria-hidden="true" tabindex="-1"></a>        <span class="at">color_by =</span> color_by,</span>
<span id="cb1-635"><a href="#cb1-635" aria-hidden="true" tabindex="-1"></a>        <span class="at">base_size =</span> <span class="dv">10</span>,</span>
<span id="cb1-636"><a href="#cb1-636" aria-hidden="true" tabindex="-1"></a>        <span class="at">point_size =</span> <span class="fl">0.5</span></span>
<span id="cb1-637"><a href="#cb1-637" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb1-638"><a href="#cb1-638" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>,</span>
<span id="cb1-639"><a href="#cb1-639" aria-hidden="true" tabindex="-1"></a>    <span class="at">SIMPLIFY =</span> <span class="cn">FALSE</span></span>
<span id="cb1-640"><a href="#cb1-640" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-641"><a href="#cb1-641" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-642"><a href="#cb1-642" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-643"><a href="#cb1-643" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=8}</span></span>
<span id="cb1-644"><a href="#cb1-644" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(<span class="at">plotlist =</span> plot_scores[[which_dataset]], <span class="at">ncol =</span> <span class="dv">2</span>)</span>
<span id="cb1-645"><a href="#cb1-645" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-646"><a href="#cb1-646" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-647"><a href="#cb1-647" aria-hidden="true" tabindex="-1"></a><span class="fu">### METABRIC</span></span>
<span id="cb1-648"><a href="#cb1-648" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-651"><a href="#cb1-651" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-652"><a href="#cb1-652" aria-hidden="true" tabindex="-1"></a>which_dataset <span class="ot">&lt;-</span> <span class="st">"metabric"</span></span>
<span id="cb1-653"><a href="#cb1-653" aria-hidden="true" tabindex="-1"></a>plot_scores[[which_dataset]] <span class="ot">&lt;-</span> <span class="fu">mapply</span>(</span>
<span id="cb1-654"><a href="#cb1-654" aria-hidden="true" tabindex="-1"></a>    plot_scores_vs_clinics,</span>
<span id="cb1-655"><a href="#cb1-655" aria-hidden="true" tabindex="-1"></a>    <span class="at">which_score =</span> which_scores,</span>
<span id="cb1-656"><a href="#cb1-656" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> which_scores,</span>
<span id="cb1-657"><a href="#cb1-657" aria-hidden="true" tabindex="-1"></a>    <span class="at">MoreArgs =</span> <span class="fu">list</span>(</span>
<span id="cb1-658"><a href="#cb1-658" aria-hidden="true" tabindex="-1"></a>        <span class="at">sum_exp =</span> datasets[[which_dataset]],</span>
<span id="cb1-659"><a href="#cb1-659" aria-hidden="true" tabindex="-1"></a>        <span class="at">clinical_variable =</span> clinical_variable, </span>
<span id="cb1-660"><a href="#cb1-660" aria-hidden="true" tabindex="-1"></a>        <span class="at">color_by =</span> color_by,</span>
<span id="cb1-661"><a href="#cb1-661" aria-hidden="true" tabindex="-1"></a>        <span class="at">base_size =</span> <span class="dv">10</span>,</span>
<span id="cb1-662"><a href="#cb1-662" aria-hidden="true" tabindex="-1"></a>        <span class="at">point_size =</span> <span class="fl">0.5</span></span>
<span id="cb1-663"><a href="#cb1-663" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb1-664"><a href="#cb1-664" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>,</span>
<span id="cb1-665"><a href="#cb1-665" aria-hidden="true" tabindex="-1"></a>    <span class="at">SIMPLIFY =</span> <span class="cn">FALSE</span></span>
<span id="cb1-666"><a href="#cb1-666" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-667"><a href="#cb1-667" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-668"><a href="#cb1-668" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-669"><a href="#cb1-669" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=8}</span></span>
<span id="cb1-670"><a href="#cb1-670" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(<span class="at">plotlist =</span> plot_scores[[which_dataset]], <span class="at">ncol =</span> <span class="dv">2</span>)</span>
<span id="cb1-671"><a href="#cb1-671" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-672"><a href="#cb1-672" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-673"><a href="#cb1-673" aria-hidden="true" tabindex="-1"></a><span class="fu">### SCANB</span></span>
<span id="cb1-674"><a href="#cb1-674" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-677"><a href="#cb1-677" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-678"><a href="#cb1-678" aria-hidden="true" tabindex="-1"></a>which_dataset <span class="ot">&lt;-</span> <span class="st">"scanb"</span></span>
<span id="cb1-679"><a href="#cb1-679" aria-hidden="true" tabindex="-1"></a>plot_scores[[which_dataset]] <span class="ot">&lt;-</span> <span class="fu">mapply</span>(</span>
<span id="cb1-680"><a href="#cb1-680" aria-hidden="true" tabindex="-1"></a>    plot_scores_vs_clinics,</span>
<span id="cb1-681"><a href="#cb1-681" aria-hidden="true" tabindex="-1"></a>    <span class="at">which_score =</span> which_scores,</span>
<span id="cb1-682"><a href="#cb1-682" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> which_scores,</span>
<span id="cb1-683"><a href="#cb1-683" aria-hidden="true" tabindex="-1"></a>    <span class="at">MoreArgs =</span> <span class="fu">list</span>(</span>
<span id="cb1-684"><a href="#cb1-684" aria-hidden="true" tabindex="-1"></a>        <span class="at">sum_exp =</span> datasets[[which_dataset]],</span>
<span id="cb1-685"><a href="#cb1-685" aria-hidden="true" tabindex="-1"></a>        <span class="at">clinical_variable =</span> clinical_variable, </span>
<span id="cb1-686"><a href="#cb1-686" aria-hidden="true" tabindex="-1"></a>        <span class="at">color_by =</span> color_by,</span>
<span id="cb1-687"><a href="#cb1-687" aria-hidden="true" tabindex="-1"></a>        <span class="at">base_size =</span> <span class="dv">10</span>,</span>
<span id="cb1-688"><a href="#cb1-688" aria-hidden="true" tabindex="-1"></a>        <span class="at">point_size =</span> <span class="fl">0.5</span></span>
<span id="cb1-689"><a href="#cb1-689" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb1-690"><a href="#cb1-690" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>,</span>
<span id="cb1-691"><a href="#cb1-691" aria-hidden="true" tabindex="-1"></a>    <span class="at">SIMPLIFY =</span> <span class="cn">FALSE</span></span>
<span id="cb1-692"><a href="#cb1-692" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-693"><a href="#cb1-693" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-694"><a href="#cb1-694" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-695"><a href="#cb1-695" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=8}</span></span>
<span id="cb1-696"><a href="#cb1-696" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(<span class="at">plotlist =</span> plot_scores[[which_dataset]], <span class="at">ncol =</span> <span class="dv">2</span>)</span>
<span id="cb1-697"><a href="#cb1-697" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-698"><a href="#cb1-698" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-699"><a href="#cb1-699" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb1-700"><a href="#cb1-700" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-701"><a href="#cb1-701" aria-hidden="true" tabindex="-1"></a>In all cohorts the luminal A and B patients have a similar score. Also</span>
<span id="cb1-702"><a href="#cb1-702" aria-hidden="true" tabindex="-1"></a>the distinction is very clear between the basal and HER2-like patients</span>
<span id="cb1-703"><a href="#cb1-703" aria-hidden="true" tabindex="-1"></a>versus luminal A and B.</span>
<span id="cb1-704"><a href="#cb1-704" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-705"><a href="#cb1-705" aria-hidden="true" tabindex="-1"></a>One question that usually arises when calculating scores from gene sets is</span>
<span id="cb1-706"><a href="#cb1-706" aria-hidden="true" tabindex="-1"></a>if proliferation associated genes (PAG) are driving the distinctions. </span>
<span id="cb1-707"><a href="#cb1-707" aria-hidden="true" tabindex="-1"></a>These signatures are highly curated and they have close or no PAGs. </span>
<span id="cb1-708"><a href="#cb1-708" aria-hidden="true" tabindex="-1"></a>Therefore, the scores are not affected by PAGs and they really reflect the</span>
<span id="cb1-709"><a href="#cb1-709" aria-hidden="true" tabindex="-1"></a>biology.</span>
<span id="cb1-710"><a href="#cb1-710" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-711"><a href="#cb1-711" aria-hidden="true" tabindex="-1"></a>In the next section we will show how these scores are also prognostic for</span>
<span id="cb1-712"><a href="#cb1-712" aria-hidden="true" tabindex="-1"></a>ER+ BC patients.</span>
<span id="cb1-713"><a href="#cb1-713" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-714"><a href="#cb1-714" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-715"><a href="#cb1-715" aria-hidden="true" tabindex="-1"></a><span class="fu">## Survival analysis</span></span>
<span id="cb1-716"><a href="#cb1-716" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-717"><a href="#cb1-717" aria-hidden="true" tabindex="-1"></a>Since the scores are continuous variables and they are already scaled due</span>
<span id="cb1-718"><a href="#cb1-718" aria-hidden="true" tabindex="-1"></a>to the output of GSVA, cox regression <span class="co">[</span><span class="ot">@Cox1972</span><span class="co">]</span> can be used. The advantages </span>
<span id="cb1-719"><a href="#cb1-719" aria-hidden="true" tabindex="-1"></a>of using cox regression is that one can control for other variables. You might</span>
<span id="cb1-720"><a href="#cb1-720" aria-hidden="true" tabindex="-1"></a>ask, why should one control for clinical variables? One of the reasons is </span>
<span id="cb1-721"><a href="#cb1-721" aria-hidden="true" tabindex="-1"></a>because the data being dealt here is observational data. There are several</span>
<span id="cb1-722"><a href="#cb1-722" aria-hidden="true" tabindex="-1"></a>confounders, for example, a score might be up because patients with a higher</span>
<span id="cb1-723"><a href="#cb1-723" aria-hidden="true" tabindex="-1"></a>tumor grade have higher expression of some genes. Thus the score is confounded</span>
<span id="cb1-724"><a href="#cb1-724" aria-hidden="true" tabindex="-1"></a>by the tumor grade and the interpretation changes. </span>
<span id="cb1-725"><a href="#cb1-725" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-726"><a href="#cb1-726" aria-hidden="true" tabindex="-1"></a>There are limitations still when dealing with observational data. A strong </span>
<span id="cb1-727"><a href="#cb1-727" aria-hidden="true" tabindex="-1"></a>hypothesis for performing survival analysis with observational data is that</span>
<span id="cb1-728"><a href="#cb1-728" aria-hidden="true" tabindex="-1"></a>we have measured all the confounder variables. This is pretty strong and in</span>
<span id="cb1-729"><a href="#cb1-729" aria-hidden="true" tabindex="-1"></a>practice we never know if a confounder is missing or not. For a more</span>
<span id="cb1-730"><a href="#cb1-730" aria-hidden="true" tabindex="-1"></a>thorough overview of the causal framework for observational data, check</span>
<span id="cb1-731"><a href="#cb1-731" aria-hidden="true" tabindex="-1"></a>the books <span class="co">[</span><span class="ot">@Gelman2020-uh; @McElreath2020-jn</span><span class="co">]</span>.</span>
<span id="cb1-732"><a href="#cb1-732" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-733"><a href="#cb1-733" aria-hidden="true" tabindex="-1"></a>All the cohorts have a different set of clinical variables available. Therefore,</span>
<span id="cb1-734"><a href="#cb1-734" aria-hidden="true" tabindex="-1"></a>the regression will be done by adjusting a different set of variables. Below</span>
<span id="cb1-735"><a href="#cb1-735" aria-hidden="true" tabindex="-1"></a>is the description of the variables used for each cohort.^<span class="co">[</span><span class="ot">This [webpage</span><span class="co">](https://web.archive.org/web/20220630083539/https://www.cancer.net/cancer-types/breast-cancer/stages)</span> explains with more details the different tumor stages and how</span>
<span id="cb1-736"><a href="#cb1-736" aria-hidden="true" tabindex="-1"></a>BC are classified.]</span>
<span id="cb1-737"><a href="#cb1-737" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-738"><a href="#cb1-738" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>**Age**: age is one of the most important factors to adjust, specially in</span>
<span id="cb1-739"><a href="#cb1-739" aria-hidden="true" tabindex="-1"></a>breast cancer. This covariate is used for all cohorts.</span>
<span id="cb1-740"><a href="#cb1-740" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>**NPI**: the Nottingham prognostic index scores each tumor</span>
<span id="cb1-741"><a href="#cb1-741" aria-hidden="true" tabindex="-1"></a>based on tumor grade, tumor size and number of lymph nodes. Only METABRIC</span>
<span id="cb1-742"><a href="#cb1-742" aria-hidden="true" tabindex="-1"></a>has this information. </span>
<span id="cb1-743"><a href="#cb1-743" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>**Tumor Size**: as name describes. Only SCANB has this information. </span>
<span id="cb1-744"><a href="#cb1-744" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>**Tumor Stage**: tumors are usually described in terms of stage, it reflects</span>
<span id="cb1-745"><a href="#cb1-745" aria-hidden="true" tabindex="-1"></a>the tumor size and location. SCANB and TCGA have this information.</span>
<span id="cb1-746"><a href="#cb1-746" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>**Node Stage**: similar to tumor stage, but encodes the number of lymph nodes</span>
<span id="cb1-747"><a href="#cb1-747" aria-hidden="true" tabindex="-1"></a>where breast cancer cells can be found. SCANB and TCGA have this information.</span>
<span id="cb1-748"><a href="#cb1-748" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-749"><a href="#cb1-749" aria-hidden="true" tabindex="-1"></a>Thus the models we are going to use for the survival analysis of each dataset</span>
<span id="cb1-750"><a href="#cb1-750" aria-hidden="true" tabindex="-1"></a>is shown below.</span>
<span id="cb1-751"><a href="#cb1-751" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-752"><a href="#cb1-752" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>TCGA: <span class="in">`~ score + age + node_stage + tumor_stage`</span>,</span>
<span id="cb1-753"><a href="#cb1-753" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>SCANB: <span class="in">`~ score + age + node_stage + tumor_stage`</span>,</span>
<span id="cb1-754"><a href="#cb1-754" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>METABRIC: <span class="in">`~ score + age + NPI`</span>,</span>
<span id="cb1-755"><a href="#cb1-755" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-756"><a href="#cb1-756" aria-hidden="true" tabindex="-1"></a>where score is one of the scores calculated earlier with GSVA. Note here</span>
<span id="cb1-757"><a href="#cb1-757" aria-hidden="true" tabindex="-1"></a>that since each node stage and tumor stage have sub classifications they </span>
<span id="cb1-758"><a href="#cb1-758" aria-hidden="true" tabindex="-1"></a>will be grouped together, otherwise there will be too many variables with</span>
<span id="cb1-759"><a href="#cb1-759" aria-hidden="true" tabindex="-1"></a>few points. We will also subselect specific tumor stages for TCGA and SCANB,</span>
<span id="cb1-760"><a href="#cb1-760" aria-hidden="true" tabindex="-1"></a>since there are very few patients with tumor stage 4.</span>
<span id="cb1-761"><a href="#cb1-761" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-762"><a href="#cb1-762" aria-hidden="true" tabindex="-1"></a>To be very specific in the survival analysis, only endocrine treated patients</span>
<span id="cb1-763"><a href="#cb1-763" aria-hidden="true" tabindex="-1"></a>should be used in the analysis, as that is what are interested in. METABRIC</span>
<span id="cb1-764"><a href="#cb1-764" aria-hidden="true" tabindex="-1"></a>and SCANB has this kind of annotation, but TCGA not. So the survival analysis</span>
<span id="cb1-765"><a href="#cb1-765" aria-hidden="true" tabindex="-1"></a>on SCANB and METABRIC will be performed on endocrine only treated, ER+ BC</span>
<span id="cb1-766"><a href="#cb1-766" aria-hidden="true" tabindex="-1"></a>patients. On TCGA, to mitigate this effect, we subselect only luminal A and </span>
<span id="cb1-767"><a href="#cb1-767" aria-hidden="true" tabindex="-1"></a>B patients, and we keep in mind that they might have been treated with</span>
<span id="cb1-768"><a href="#cb1-768" aria-hidden="true" tabindex="-1"></a>chemotherapy as well. Moreover, in Sweden the guidelines for BC treatment is to</span>
<span id="cb1-769"><a href="#cb1-769" aria-hidden="true" tabindex="-1"></a>use 10% as the threshold for ER status. Therefore, ER+ BC patients used on</span>
<span id="cb1-770"><a href="#cb1-770" aria-hidden="true" tabindex="-1"></a>SCANB are those that are above the 10% threshold.</span>
<span id="cb1-771"><a href="#cb1-771" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-772"><a href="#cb1-772" aria-hidden="true" tabindex="-1"></a>The table below shows the number of patients for each cohort.</span>
<span id="cb1-773"><a href="#cb1-773" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-776"><a href="#cb1-776" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-777"><a href="#cb1-777" aria-hidden="true" tabindex="-1"></a>tcga_samples <span class="ot">&lt;-</span> <span class="fu">colData</span>(datasets<span class="sc">$</span>tcga) <span class="sc">%&gt;%</span> data.frame <span class="sc">%&gt;%</span></span>
<span id="cb1-778"><a href="#cb1-778" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb1-779"><a href="#cb1-779" aria-hidden="true" tabindex="-1"></a>        pam50 <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"luma"</span>, <span class="st">"lumb"</span>) <span class="sc">&amp;</span></span>
<span id="cb1-780"><a href="#cb1-780" aria-hidden="true" tabindex="-1"></a>        node_stage <span class="sc">!=</span> <span class="st">"NX"</span> <span class="sc">&amp;</span></span>
<span id="cb1-781"><a href="#cb1-781" aria-hidden="true" tabindex="-1"></a>        <span class="sc">!</span>(tumor_stage <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"stage_iv"</span>, <span class="st">"na"</span>)) <span class="sc">&amp;</span> </span>
<span id="cb1-782"><a href="#cb1-782" aria-hidden="true" tabindex="-1"></a>        er_status <span class="sc">==</span> <span class="st">"pos"</span></span>
<span id="cb1-783"><a href="#cb1-783" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span> </span>
<span id="cb1-784"><a href="#cb1-784" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">pull</span>(sample_name)</span>
<span id="cb1-785"><a href="#cb1-785" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-786"><a href="#cb1-786" aria-hidden="true" tabindex="-1"></a>scanb_samples <span class="ot">&lt;-</span> <span class="fu">colData</span>(datasets<span class="sc">$</span>scanb) <span class="sc">%&gt;%</span> data.frame <span class="sc">%&gt;%</span></span>
<span id="cb1-787"><a href="#cb1-787" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb1-788"><a href="#cb1-788" aria-hidden="true" tabindex="-1"></a>        node_stage <span class="sc">!=</span> <span class="st">"NX"</span> <span class="sc">&amp;</span></span>
<span id="cb1-789"><a href="#cb1-789" aria-hidden="true" tabindex="-1"></a>        <span class="sc">!</span>(tumor_stage <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"T4"</span>, <span class="st">"Tis"</span>, <span class="st">"TX"</span>)) <span class="sc">&amp;</span> </span>
<span id="cb1-790"><a href="#cb1-790" aria-hidden="true" tabindex="-1"></a>        ER_10perc <span class="sc">==</span> <span class="st">"POS"</span> <span class="sc">&amp;</span> </span>
<span id="cb1-791"><a href="#cb1-791" aria-hidden="true" tabindex="-1"></a>        Adjuvant_Endo_only <span class="sc">==</span> <span class="dv">1</span></span>
<span id="cb1-792"><a href="#cb1-792" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span> </span>
<span id="cb1-793"><a href="#cb1-793" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">pull</span>(sample_name)</span>
<span id="cb1-794"><a href="#cb1-794" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-795"><a href="#cb1-795" aria-hidden="true" tabindex="-1"></a>metabric_samples <span class="ot">&lt;-</span> <span class="fu">colData</span>(datasets<span class="sc">$</span>metabric) <span class="sc">%&gt;%</span> data.frame <span class="sc">%&gt;%</span></span>
<span id="cb1-796"><a href="#cb1-796" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb1-797"><a href="#cb1-797" aria-hidden="true" tabindex="-1"></a>        CHEMOTHERAPY <span class="sc">==</span> <span class="st">"NO"</span> <span class="sc">&amp;</span></span>
<span id="cb1-798"><a href="#cb1-798" aria-hidden="true" tabindex="-1"></a>        er_status <span class="sc">==</span> <span class="st">"pos"</span> <span class="sc">&amp;</span> </span>
<span id="cb1-799"><a href="#cb1-799" aria-hidden="true" tabindex="-1"></a>        HORMONE_THERAPY <span class="sc">==</span> <span class="st">"YES"</span></span>
<span id="cb1-800"><a href="#cb1-800" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span> </span>
<span id="cb1-801"><a href="#cb1-801" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">pull</span>(sample_name)</span>
<span id="cb1-802"><a href="#cb1-802" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-803"><a href="#cb1-803" aria-hidden="true" tabindex="-1"></a><span class="fu">data.frame</span>(</span>
<span id="cb1-804"><a href="#cb1-804" aria-hidden="true" tabindex="-1"></a>    <span class="at">cohort =</span> <span class="fu">c</span>(<span class="st">"TCGA"</span>, <span class="st">"SCANB"</span>, <span class="st">"METABRIC"</span>),</span>
<span id="cb1-805"><a href="#cb1-805" aria-hidden="true" tabindex="-1"></a>    <span class="at">number_patients =</span> <span class="fu">c</span>(</span>
<span id="cb1-806"><a href="#cb1-806" aria-hidden="true" tabindex="-1"></a>        <span class="fu">length</span>(tcga_samples),</span>
<span id="cb1-807"><a href="#cb1-807" aria-hidden="true" tabindex="-1"></a>        <span class="fu">length</span>(scanb_samples),</span>
<span id="cb1-808"><a href="#cb1-808" aria-hidden="true" tabindex="-1"></a>        <span class="fu">length</span>(metabric_samples)</span>
<span id="cb1-809"><a href="#cb1-809" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb1-810"><a href="#cb1-810" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> kableExtra<span class="sc">::</span><span class="fu">kbl</span>() <span class="sc">%&gt;%</span></span>
<span id="cb1-811"><a href="#cb1-811" aria-hidden="true" tabindex="-1"></a>    kableExtra<span class="sc">::</span><span class="fu">kable_classic</span>(<span class="at">full_width =</span> <span class="cn">FALSE</span>)</span>
<span id="cb1-812"><a href="#cb1-812" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-813"><a href="#cb1-813" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-816"><a href="#cb1-816" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-817"><a href="#cb1-817" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb1-818"><a href="#cb1-818" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-819"><a href="#cb1-819" aria-hidden="true" tabindex="-1"></a><span class="co"># run the survival analysis for each dataset with its own formula.</span></span>
<span id="cb1-820"><a href="#cb1-820" aria-hidden="true" tabindex="-1"></a><span class="co"># we start by defining the base formulas, because later we sapply and </span></span>
<span id="cb1-821"><a href="#cb1-821" aria-hidden="true" tabindex="-1"></a><span class="co"># add the scores. this way if we need to change the formulas, we only</span></span>
<span id="cb1-822"><a href="#cb1-822" aria-hidden="true" tabindex="-1"></a><span class="co"># need to change once and all together.</span></span>
<span id="cb1-823"><a href="#cb1-823" aria-hidden="true" tabindex="-1"></a>formulas_survival <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb1-824"><a href="#cb1-824" aria-hidden="true" tabindex="-1"></a>    <span class="st">"os_tcga"</span> <span class="ot">=</span> <span class="st">"Surv(os_months, os_status) ~ age + node_stage + tumor_stage"</span>,</span>
<span id="cb1-825"><a href="#cb1-825" aria-hidden="true" tabindex="-1"></a>    <span class="st">"os_scanb"</span> <span class="ot">=</span> <span class="st">"Surv(os_months, os_status) ~ age + node_stage + tumor_stage"</span>,</span>
<span id="cb1-826"><a href="#cb1-826" aria-hidden="true" tabindex="-1"></a>    <span class="st">"os_metabric"</span> <span class="ot">=</span> <span class="st">"Surv(os_months, os_status) ~ age + npi"</span>,</span>
<span id="cb1-827"><a href="#cb1-827" aria-hidden="true" tabindex="-1"></a>    <span class="st">"rfs_metabric"</span> <span class="ot">=</span> <span class="st">"Surv(rfs_months, os_status) ~ age + npi"</span></span>
<span id="cb1-828"><a href="#cb1-828" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-829"><a href="#cb1-829" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-830"><a href="#cb1-830" aria-hidden="true" tabindex="-1"></a>type_survival <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"os"</span>, <span class="st">"rfs"</span>)</span>
<span id="cb1-831"><a href="#cb1-831" aria-hidden="true" tabindex="-1"></a>which_scores <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb1-832"><a href="#cb1-832" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_ESTROGEN_RESPONSE_EARLY"</span>,</span>
<span id="cb1-833"><a href="#cb1-833" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_ESTROGEN_RESPONSE_LATE"</span>,</span>
<span id="cb1-834"><a href="#cb1-834" aria-hidden="true" tabindex="-1"></a>    <span class="st">"SET_ERPR"</span>,</span>
<span id="cb1-835"><a href="#cb1-835" aria-hidden="true" tabindex="-1"></a>    <span class="st">"random_200"</span>,</span>
<span id="cb1-836"><a href="#cb1-836" aria-hidden="true" tabindex="-1"></a>    <span class="st">"random_18"</span></span>
<span id="cb1-837"><a href="#cb1-837" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-838"><a href="#cb1-838" aria-hidden="true" tabindex="-1"></a>survival_results <span class="ot">&lt;-</span> <span class="fu">sapply</span>(</span>
<span id="cb1-839"><a href="#cb1-839" aria-hidden="true" tabindex="-1"></a>    <span class="fu">c</span>(<span class="st">"os"</span>, <span class="st">"rfs"</span>),</span>
<span id="cb1-840"><a href="#cb1-840" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(type_analysis, datasets, name_scores, formulas){</span>
<span id="cb1-841"><a href="#cb1-841" aria-hidden="true" tabindex="-1"></a>        <span class="fu">mapply</span>(</span>
<span id="cb1-842"><a href="#cb1-842" aria-hidden="true" tabindex="-1"></a>            <span class="cf">function</span>(</span>
<span id="cb1-843"><a href="#cb1-843" aria-hidden="true" tabindex="-1"></a>                dataset, name_dataset, name_scores, formulas, type_analysis</span>
<span id="cb1-844"><a href="#cb1-844" aria-hidden="true" tabindex="-1"></a>            ){</span>
<span id="cb1-845"><a href="#cb1-845" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb1-846"><a href="#cb1-846" aria-hidden="true" tabindex="-1"></a>                <span class="cf">if</span> (type_analysis <span class="sc">==</span> <span class="st">"rfs"</span> <span class="sc">&amp;</span> name_dataset <span class="sc">!=</span> <span class="st">"metabric"</span>){</span>
<span id="cb1-847"><a href="#cb1-847" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">return</span>()</span>
<span id="cb1-848"><a href="#cb1-848" aria-hidden="true" tabindex="-1"></a>                } <span class="cf">else</span> { </span>
<span id="cb1-849"><a href="#cb1-849" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb1-850"><a href="#cb1-850" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">sapply</span>(</span>
<span id="cb1-851"><a href="#cb1-851" aria-hidden="true" tabindex="-1"></a>                        name_scores,</span>
<span id="cb1-852"><a href="#cb1-852" aria-hidden="true" tabindex="-1"></a>                        <span class="cf">function</span>(name_score, col_data, formula){</span>
<span id="cb1-853"><a href="#cb1-853" aria-hidden="true" tabindex="-1"></a>                            </span>
<span id="cb1-854"><a href="#cb1-854" aria-hidden="true" tabindex="-1"></a>                            survival<span class="sc">::</span><span class="fu">coxph</span>(</span>
<span id="cb1-855"><a href="#cb1-855" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">as.formula</span>(</span>
<span id="cb1-856"><a href="#cb1-856" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">paste</span>(formula, name_score, <span class="at">sep =</span> <span class="st">"+"</span>)</span>
<span id="cb1-857"><a href="#cb1-857" aria-hidden="true" tabindex="-1"></a>                                ),</span>
<span id="cb1-858"><a href="#cb1-858" aria-hidden="true" tabindex="-1"></a>                                <span class="at">data =</span> col_data, </span>
<span id="cb1-859"><a href="#cb1-859" aria-hidden="true" tabindex="-1"></a>                                <span class="at">y =</span> <span class="cn">FALSE</span>,</span>
<span id="cb1-860"><a href="#cb1-860" aria-hidden="true" tabindex="-1"></a>                                <span class="at">x =</span> <span class="cn">FALSE</span></span>
<span id="cb1-861"><a href="#cb1-861" aria-hidden="true" tabindex="-1"></a>                            )</span>
<span id="cb1-862"><a href="#cb1-862" aria-hidden="true" tabindex="-1"></a>                        },</span>
<span id="cb1-863"><a href="#cb1-863" aria-hidden="true" tabindex="-1"></a>                        <span class="at">col_data =</span> <span class="fu">colData</span>(dataset) <span class="sc">%&gt;%</span> data.frame,</span>
<span id="cb1-864"><a href="#cb1-864" aria-hidden="true" tabindex="-1"></a>                        <span class="at">formula =</span> formulas[</span>
<span id="cb1-865"><a href="#cb1-865" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">grepl</span>(</span>
<span id="cb1-866"><a href="#cb1-866" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">paste</span>(type_analysis, name_dataset, <span class="at">sep =</span> <span class="st">"_"</span>),</span>
<span id="cb1-867"><a href="#cb1-867" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">names</span>(formulas)</span>
<span id="cb1-868"><a href="#cb1-868" aria-hidden="true" tabindex="-1"></a>                            )</span>
<span id="cb1-869"><a href="#cb1-869" aria-hidden="true" tabindex="-1"></a>                        ],</span>
<span id="cb1-870"><a href="#cb1-870" aria-hidden="true" tabindex="-1"></a>                        <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>,</span>
<span id="cb1-871"><a href="#cb1-871" aria-hidden="true" tabindex="-1"></a>                        <span class="at">simplify =</span> <span class="cn">FALSE</span></span>
<span id="cb1-872"><a href="#cb1-872" aria-hidden="true" tabindex="-1"></a>                    )</span>
<span id="cb1-873"><a href="#cb1-873" aria-hidden="true" tabindex="-1"></a>                }</span>
<span id="cb1-874"><a href="#cb1-874" aria-hidden="true" tabindex="-1"></a>            },</span>
<span id="cb1-875"><a href="#cb1-875" aria-hidden="true" tabindex="-1"></a>            <span class="at">dataset =</span> datasets,</span>
<span id="cb1-876"><a href="#cb1-876" aria-hidden="true" tabindex="-1"></a>            <span class="at">name_dataset =</span> <span class="fu">names</span>(datasets),</span>
<span id="cb1-877"><a href="#cb1-877" aria-hidden="true" tabindex="-1"></a>            <span class="at">MoreArgs =</span> <span class="fu">list</span>(</span>
<span id="cb1-878"><a href="#cb1-878" aria-hidden="true" tabindex="-1"></a>                <span class="at">formulas =</span> formulas, <span class="at">name_scores =</span> name_scores,</span>
<span id="cb1-879"><a href="#cb1-879" aria-hidden="true" tabindex="-1"></a>                <span class="at">type_analysis =</span> type_analysis</span>
<span id="cb1-880"><a href="#cb1-880" aria-hidden="true" tabindex="-1"></a>            ),</span>
<span id="cb1-881"><a href="#cb1-881" aria-hidden="true" tabindex="-1"></a>            <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>,</span>
<span id="cb1-882"><a href="#cb1-882" aria-hidden="true" tabindex="-1"></a>            <span class="at">SIMPLIFY =</span> <span class="cn">FALSE</span></span>
<span id="cb1-883"><a href="#cb1-883" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb1-884"><a href="#cb1-884" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb1-885"><a href="#cb1-885" aria-hidden="true" tabindex="-1"></a>    <span class="at">datasets =</span> <span class="fu">mapply</span>(</span>
<span id="cb1-886"><a href="#cb1-886" aria-hidden="true" tabindex="-1"></a>        <span class="cf">function</span>(dataset, patients) dataset[, patients],</span>
<span id="cb1-887"><a href="#cb1-887" aria-hidden="true" tabindex="-1"></a>        <span class="at">dataset =</span> datasets,</span>
<span id="cb1-888"><a href="#cb1-888" aria-hidden="true" tabindex="-1"></a>        <span class="at">patients =</span> <span class="fu">list</span>(tcga_samples, scanb_samples, metabric_samples),</span>
<span id="cb1-889"><a href="#cb1-889" aria-hidden="true" tabindex="-1"></a>        <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>,</span>
<span id="cb1-890"><a href="#cb1-890" aria-hidden="true" tabindex="-1"></a>        <span class="at">SIMPLIFY =</span> <span class="cn">FALSE</span></span>
<span id="cb1-891"><a href="#cb1-891" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb1-892"><a href="#cb1-892" aria-hidden="true" tabindex="-1"></a>    <span class="at">name_scores =</span> which_scores,</span>
<span id="cb1-893"><a href="#cb1-893" aria-hidden="true" tabindex="-1"></a>    <span class="at">formulas =</span> formulas_survival,</span>
<span id="cb1-894"><a href="#cb1-894" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>,</span>
<span id="cb1-895"><a href="#cb1-895" aria-hidden="true" tabindex="-1"></a>    <span class="at">simplify =</span> <span class="cn">FALSE</span></span>
<span id="cb1-896"><a href="#cb1-896" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-897"><a href="#cb1-897" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-898"><a href="#cb1-898" aria-hidden="true" tabindex="-1"></a>survival_results<span class="sc">$</span>rfs <span class="ot">&lt;-</span> survival_results<span class="sc">$</span>rfs[</span>
<span id="cb1-899"><a href="#cb1-899" aria-hidden="true" tabindex="-1"></a>    <span class="sc">!</span><span class="fu">sapply</span>(survival_results<span class="sc">$</span>rfs, is.null)</span>
<span id="cb1-900"><a href="#cb1-900" aria-hidden="true" tabindex="-1"></a>]</span>
<span id="cb1-901"><a href="#cb1-901" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-902"><a href="#cb1-902" aria-hidden="true" tabindex="-1"></a><span class="co"># we cannot simply save the models as each model has some </span></span>
<span id="cb1-903"><a href="#cb1-903" aria-hidden="true" tabindex="-1"></a><span class="co"># environment variables, which adds up to over 20GB. </span></span>
<span id="cb1-904"><a href="#cb1-904" aria-hidden="true" tabindex="-1"></a><span class="co"># check this stack exchange thread to read more on it:</span></span>
<span id="cb1-905"><a href="#cb1-905" aria-hidden="true" tabindex="-1"></a><span class="co"># https://stackoverflow.com/questions/42230920/saverds-inflating-size-of-object/52372480</span></span>
<span id="cb1-906"><a href="#cb1-906" aria-hidden="true" tabindex="-1"></a><span class="co"># the solution is to basically clear the environment from the terms </span></span>
<span id="cb1-907"><a href="#cb1-907" aria-hidden="true" tabindex="-1"></a><span class="co"># object. since it is pretty fast to run the survival analysis</span></span>
<span id="cb1-908"><a href="#cb1-908" aria-hidden="true" tabindex="-1"></a><span class="co"># we will not save the objects.</span></span>
<span id="cb1-909"><a href="#cb1-909" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-910"><a href="#cb1-910" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-911"><a href="#cb1-911" aria-hidden="true" tabindex="-1"></a><span class="co"># we now proceed to prepare the plots. since the survival result is </span></span>
<span id="cb1-912"><a href="#cb1-912" aria-hidden="true" tabindex="-1"></a><span class="co"># actually a large file, it is best to do all the plottings, save</span></span>
<span id="cb1-913"><a href="#cb1-913" aria-hidden="true" tabindex="-1"></a><span class="co"># them in other rds/pdf files and then use on the quarto markdown.</span></span>
<span id="cb1-914"><a href="#cb1-914" aria-hidden="true" tabindex="-1"></a><span class="co"># otherwise it is too slow to render the book. </span></span>
<span id="cb1-915"><a href="#cb1-915" aria-hidden="true" tabindex="-1"></a><span class="co"># the same is true for the statistics and coefficients available.</span></span>
<span id="cb1-916"><a href="#cb1-916" aria-hidden="true" tabindex="-1"></a>names_signatures <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb1-917"><a href="#cb1-917" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_ESTROGEN_RESPONSE_EARLY"</span> <span class="ot">=</span> <span class="st">"Estrogen Early"</span>,</span>
<span id="cb1-918"><a href="#cb1-918" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_ESTROGEN_RESPONSE_LATE"</span> <span class="ot">=</span> <span class="st">"Estrogen Late"</span>,</span>
<span id="cb1-919"><a href="#cb1-919" aria-hidden="true" tabindex="-1"></a>    <span class="st">"SET_ERPR"</span> <span class="ot">=</span> <span class="st">"SET ER/PR"</span>,</span>
<span id="cb1-920"><a href="#cb1-920" aria-hidden="true" tabindex="-1"></a>    <span class="st">"random_200"</span> <span class="ot">=</span> <span class="st">"random 200 genes"</span>,</span>
<span id="cb1-921"><a href="#cb1-921" aria-hidden="true" tabindex="-1"></a>    <span class="st">"random_18"</span> <span class="ot">=</span> <span class="st">"random 18 genes"</span></span>
<span id="cb1-922"><a href="#cb1-922" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-923"><a href="#cb1-923" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-924"><a href="#cb1-924" aria-hidden="true" tabindex="-1"></a>names_coefficients <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb1-925"><a href="#cb1-925" aria-hidden="true" tabindex="-1"></a>    <span class="at">metabric =</span> <span class="fu">c</span>(</span>
<span id="cb1-926"><a href="#cb1-926" aria-hidden="true" tabindex="-1"></a>        <span class="st">"age"</span> <span class="ot">=</span> <span class="st">"Age"</span>, </span>
<span id="cb1-927"><a href="#cb1-927" aria-hidden="true" tabindex="-1"></a>        <span class="st">"npi"</span> <span class="ot">=</span> <span class="st">"NPI"</span>, </span>
<span id="cb1-928"><a href="#cb1-928" aria-hidden="true" tabindex="-1"></a>        names_signatures</span>
<span id="cb1-929"><a href="#cb1-929" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb1-930"><a href="#cb1-930" aria-hidden="true" tabindex="-1"></a>    <span class="at">scanb =</span> <span class="fu">c</span>(</span>
<span id="cb1-931"><a href="#cb1-931" aria-hidden="true" tabindex="-1"></a>        <span class="st">"age"</span> <span class="ot">=</span> <span class="st">"age"</span>,</span>
<span id="cb1-932"><a href="#cb1-932" aria-hidden="true" tabindex="-1"></a>        names_signatures,</span>
<span id="cb1-933"><a href="#cb1-933" aria-hidden="true" tabindex="-1"></a>        <span class="st">"node_stageN1"</span> <span class="ot">=</span> <span class="st">"N1"</span>,</span>
<span id="cb1-934"><a href="#cb1-934" aria-hidden="true" tabindex="-1"></a>        <span class="st">"node_stageN2and3"</span> <span class="ot">=</span> <span class="st">"N2 and N3"</span>,</span>
<span id="cb1-935"><a href="#cb1-935" aria-hidden="true" tabindex="-1"></a>        <span class="st">"tumor_stageT1"</span> <span class="ot">=</span> <span class="st">"T1"</span>,</span>
<span id="cb1-936"><a href="#cb1-936" aria-hidden="true" tabindex="-1"></a>        <span class="st">"tumor_stageT2"</span> <span class="ot">=</span> <span class="st">"T2"</span>,</span>
<span id="cb1-937"><a href="#cb1-937" aria-hidden="true" tabindex="-1"></a>        <span class="st">"tumor_stageT3"</span> <span class="ot">=</span> <span class="st">"T3"</span></span>
<span id="cb1-938"><a href="#cb1-938" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb1-939"><a href="#cb1-939" aria-hidden="true" tabindex="-1"></a>    <span class="at">tcga =</span> <span class="fu">c</span>(</span>
<span id="cb1-940"><a href="#cb1-940" aria-hidden="true" tabindex="-1"></a>        <span class="st">"age"</span> <span class="ot">=</span> <span class="st">"age"</span>,</span>
<span id="cb1-941"><a href="#cb1-941" aria-hidden="true" tabindex="-1"></a>        <span class="st">"tumor_stage"</span> <span class="ot">=</span> <span class="st">"Tumor Stage"</span>,</span>
<span id="cb1-942"><a href="#cb1-942" aria-hidden="true" tabindex="-1"></a>        names_signatures,</span>
<span id="cb1-943"><a href="#cb1-943" aria-hidden="true" tabindex="-1"></a>        <span class="st">"node_stageN1"</span> <span class="ot">=</span> <span class="st">"N1"</span>,</span>
<span id="cb1-944"><a href="#cb1-944" aria-hidden="true" tabindex="-1"></a>        <span class="st">"node_stageN2"</span> <span class="ot">=</span> <span class="st">"N2"</span>,</span>
<span id="cb1-945"><a href="#cb1-945" aria-hidden="true" tabindex="-1"></a>        <span class="st">"node_stageN3"</span> <span class="ot">=</span> <span class="st">"N3"</span>,</span>
<span id="cb1-946"><a href="#cb1-946" aria-hidden="true" tabindex="-1"></a>        <span class="st">"tumor_stagestage_ii"</span> <span class="ot">=</span> <span class="st">"T2"</span>,</span>
<span id="cb1-947"><a href="#cb1-947" aria-hidden="true" tabindex="-1"></a>        <span class="st">"tumor_stagestage_iii"</span> <span class="ot">=</span> <span class="st">"T3"</span></span>
<span id="cb1-948"><a href="#cb1-948" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb1-949"><a href="#cb1-949" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-950"><a href="#cb1-950" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-951"><a href="#cb1-951" aria-hidden="true" tabindex="-1"></a>patients <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb1-952"><a href="#cb1-952" aria-hidden="true" tabindex="-1"></a>    <span class="at">tcga =</span> <span class="st">"Lum A/B, ER+ BC"</span>,</span>
<span id="cb1-953"><a href="#cb1-953" aria-hidden="true" tabindex="-1"></a>    <span class="at">scanb =</span> <span class="st">"Endo Only, ER+ BC"</span>,</span>
<span id="cb1-954"><a href="#cb1-954" aria-hidden="true" tabindex="-1"></a>    <span class="at">metabric =</span> <span class="st">"Endo Only, ER+ BC"</span></span>
<span id="cb1-955"><a href="#cb1-955" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-956"><a href="#cb1-956" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-957"><a href="#cb1-957" aria-hidden="true" tabindex="-1"></a>forest_plots <span class="ot">&lt;-</span> <span class="fu">mapply</span>(</span>
<span id="cb1-958"><a href="#cb1-958" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(fits, type_analysis){</span>
<span id="cb1-959"><a href="#cb1-959" aria-hidden="true" tabindex="-1"></a>        <span class="fu">mapply</span>(</span>
<span id="cb1-960"><a href="#cb1-960" aria-hidden="true" tabindex="-1"></a>            <span class="cf">function</span>(cohort_fits, name_cohort, type_analysis){</span>
<span id="cb1-961"><a href="#cb1-961" aria-hidden="true" tabindex="-1"></a>                <span class="fu">mapply</span>(</span>
<span id="cb1-962"><a href="#cb1-962" aria-hidden="true" tabindex="-1"></a>                    forest_plot_fits,</span>
<span id="cb1-963"><a href="#cb1-963" aria-hidden="true" tabindex="-1"></a>                    <span class="at">fit =</span> cohort_fits,</span>
<span id="cb1-964"><a href="#cb1-964" aria-hidden="true" tabindex="-1"></a>                    <span class="at">name_signature =</span> <span class="fu">names</span>(cohort_fits),</span>
<span id="cb1-965"><a href="#cb1-965" aria-hidden="true" tabindex="-1"></a>                    <span class="at">MoreArgs =</span> <span class="fu">list</span>(</span>
<span id="cb1-966"><a href="#cb1-966" aria-hidden="true" tabindex="-1"></a>                        <span class="at">cohort =</span> name_cohort,</span>
<span id="cb1-967"><a href="#cb1-967" aria-hidden="true" tabindex="-1"></a>                        <span class="at">names_coefficients =</span> names_coefficients,</span>
<span id="cb1-968"><a href="#cb1-968" aria-hidden="true" tabindex="-1"></a>                        <span class="at">type_survival =</span> <span class="fu">toupper</span>(type_analysis),</span>
<span id="cb1-969"><a href="#cb1-969" aria-hidden="true" tabindex="-1"></a>                        <span class="at">patients =</span> patients[[name_cohort]],</span>
<span id="cb1-970"><a href="#cb1-970" aria-hidden="true" tabindex="-1"></a>                        <span class="co">#clip = c(0.1, 2),</span></span>
<span id="cb1-971"><a href="#cb1-971" aria-hidden="true" tabindex="-1"></a>                        <span class="at">width =</span> <span class="fl">6.5</span>, </span>
<span id="cb1-972"><a href="#cb1-972" aria-hidden="true" tabindex="-1"></a>                        <span class="at">height =</span> <span class="dv">3</span>, </span>
<span id="cb1-973"><a href="#cb1-973" aria-hidden="true" tabindex="-1"></a>                        <span class="at">path_to_save =</span> <span class="fu">paste0</span>(</span>
<span id="cb1-974"><a href="#cb1-974" aria-hidden="true" tabindex="-1"></a>                            <span class="st">"../results/plots/surv_analysis_estrogen"</span>,</span>
<span id="cb1-975"><a href="#cb1-975" aria-hidden="true" tabindex="-1"></a>                            <span class="st">"/forest_plots/"</span>  </span>
<span id="cb1-976"><a href="#cb1-976" aria-hidden="true" tabindex="-1"></a>                        ) </span>
<span id="cb1-977"><a href="#cb1-977" aria-hidden="true" tabindex="-1"></a>                    ),</span>
<span id="cb1-978"><a href="#cb1-978" aria-hidden="true" tabindex="-1"></a>                    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>,</span>
<span id="cb1-979"><a href="#cb1-979" aria-hidden="true" tabindex="-1"></a>                    <span class="at">SIMPLIFY =</span> <span class="cn">FALSE</span></span>
<span id="cb1-980"><a href="#cb1-980" aria-hidden="true" tabindex="-1"></a>                )</span>
<span id="cb1-981"><a href="#cb1-981" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb1-982"><a href="#cb1-982" aria-hidden="true" tabindex="-1"></a>            },</span>
<span id="cb1-983"><a href="#cb1-983" aria-hidden="true" tabindex="-1"></a>            <span class="at">cohort_fits =</span> fits,</span>
<span id="cb1-984"><a href="#cb1-984" aria-hidden="true" tabindex="-1"></a>            <span class="at">name_cohort =</span> <span class="fu">names</span>(fits),</span>
<span id="cb1-985"><a href="#cb1-985" aria-hidden="true" tabindex="-1"></a>            <span class="at">MoreArgs =</span> <span class="fu">list</span>(<span class="at">type_analysis =</span> type_analysis),</span>
<span id="cb1-986"><a href="#cb1-986" aria-hidden="true" tabindex="-1"></a>            <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>,</span>
<span id="cb1-987"><a href="#cb1-987" aria-hidden="true" tabindex="-1"></a>            <span class="at">SIMPLIFY =</span> <span class="cn">FALSE</span></span>
<span id="cb1-988"><a href="#cb1-988" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb1-989"><a href="#cb1-989" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb1-990"><a href="#cb1-990" aria-hidden="true" tabindex="-1"></a>    survival_results,</span>
<span id="cb1-991"><a href="#cb1-991" aria-hidden="true" tabindex="-1"></a>    <span class="fu">names</span>(survival_results),</span>
<span id="cb1-992"><a href="#cb1-992" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>,</span>
<span id="cb1-993"><a href="#cb1-993" aria-hidden="true" tabindex="-1"></a>    <span class="at">SIMPLIFY =</span> <span class="cn">FALSE</span></span>
<span id="cb1-994"><a href="#cb1-994" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-995"><a href="#cb1-995" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-996"><a href="#cb1-996" aria-hidden="true" tabindex="-1"></a><span class="co"># all forest plots are saved as a pdf in the </span></span>
<span id="cb1-997"><a href="#cb1-997" aria-hidden="true" tabindex="-1"></a><span class="co"># results/plots/surv_analysis_estrogen folder and</span></span>
<span id="cb1-998"><a href="#cb1-998" aria-hidden="true" tabindex="-1"></a><span class="co"># as a rds file in the folder below.</span></span>
<span id="cb1-999"><a href="#cb1-999" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1000"><a href="#cb1-1000" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb1-1001"><a href="#cb1-1001" aria-hidden="true" tabindex="-1"></a>    forest_plots,</span>
<span id="cb1-1002"><a href="#cb1-1002" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/surv_analysis_estrogen/forest_plots.rds"</span></span>
<span id="cb1-1003"><a href="#cb1-1003" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-1004"><a href="#cb1-1004" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1005"><a href="#cb1-1005" aria-hidden="true" tabindex="-1"></a><span class="co"># and now we fetch the tables that will be used later on including the</span></span>
<span id="cb1-1006"><a href="#cb1-1006" aria-hidden="true" tabindex="-1"></a><span class="co"># confidence intervals and other parameters from the fit</span></span>
<span id="cb1-1007"><a href="#cb1-1007" aria-hidden="true" tabindex="-1"></a>tables_survival <span class="ot">&lt;-</span> <span class="fu">lapply</span>(</span>
<span id="cb1-1008"><a href="#cb1-1008" aria-hidden="true" tabindex="-1"></a>    survival_results, </span>
<span id="cb1-1009"><a href="#cb1-1009" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(x) </span>
<span id="cb1-1010"><a href="#cb1-1010" aria-hidden="true" tabindex="-1"></a>        <span class="fu">lapply</span>(</span>
<span id="cb1-1011"><a href="#cb1-1011" aria-hidden="true" tabindex="-1"></a>            x,</span>
<span id="cb1-1012"><a href="#cb1-1012" aria-hidden="true" tabindex="-1"></a>            <span class="cf">function</span>(y)</span>
<span id="cb1-1013"><a href="#cb1-1013" aria-hidden="true" tabindex="-1"></a>                <span class="fu">lapply</span>(</span>
<span id="cb1-1014"><a href="#cb1-1014" aria-hidden="true" tabindex="-1"></a>                    y,</span>
<span id="cb1-1015"><a href="#cb1-1015" aria-hidden="true" tabindex="-1"></a>                    <span class="cf">function</span>(z) broom<span class="sc">::</span><span class="fu">tidy</span>(z)</span>
<span id="cb1-1016"><a href="#cb1-1016" aria-hidden="true" tabindex="-1"></a>                ) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(<span class="at">.id =</span> <span class="st">"score"</span>)</span>
<span id="cb1-1017"><a href="#cb1-1017" aria-hidden="true" tabindex="-1"></a>        ) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(<span class="at">.id =</span> <span class="st">"cohort"</span>)</span>
<span id="cb1-1018"><a href="#cb1-1018" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(<span class="at">.id =</span> <span class="st">"type_analysis"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb1-1019"><a href="#cb1-1019" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">HR =</span> <span class="fu">exp</span>(estimate))</span>
<span id="cb1-1020"><a href="#cb1-1020" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1021"><a href="#cb1-1021" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(</span>
<span id="cb1-1022"><a href="#cb1-1022" aria-hidden="true" tabindex="-1"></a>    tables_survival,</span>
<span id="cb1-1023"><a href="#cb1-1023" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/tables/surv_analysis_estrogen/survival_results.csv"</span>,</span>
<span id="cb1-1024"><a href="#cb1-1024" aria-hidden="true" tabindex="-1"></a>    <span class="at">row.names =</span> <span class="cn">FALSE</span></span>
<span id="cb1-1025"><a href="#cb1-1025" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-1026"><a href="#cb1-1026" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-1027"><a href="#cb1-1027" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1030"><a href="#cb1-1030" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-1031"><a href="#cb1-1031" aria-hidden="true" tabindex="-1"></a>tables_survival <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(</span>
<span id="cb1-1032"><a href="#cb1-1032" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/tables/surv_analysis_estrogen/survival_results.csv"</span></span>
<span id="cb1-1033"><a href="#cb1-1033" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-1034"><a href="#cb1-1034" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1035"><a href="#cb1-1035" aria-hidden="true" tabindex="-1"></a>forest_plots <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(</span>
<span id="cb1-1036"><a href="#cb1-1036" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/surv_analysis_estrogen/forest_plots.rds"</span></span>
<span id="cb1-1037"><a href="#cb1-1037" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-1038"><a href="#cb1-1038" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-1039"><a href="#cb1-1039" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1040"><a href="#cb1-1040" aria-hidden="true" tabindex="-1"></a><span class="fu">### Results</span></span>
<span id="cb1-1041"><a href="#cb1-1041" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1042"><a href="#cb1-1042" aria-hidden="true" tabindex="-1"></a>The table below shows the results for each analysis performed for a specific</span>
<span id="cb1-1043"><a href="#cb1-1043" aria-hidden="true" tabindex="-1"></a>term. In order to understand the table the user can filter based on the term,</span>
<span id="cb1-1044"><a href="#cb1-1044" aria-hidden="true" tabindex="-1"></a>cohort and type of analysis. Since METABRIC has recurrence free survival (RFS)</span>
<span id="cb1-1045"><a href="#cb1-1045" aria-hidden="true" tabindex="-1"></a>and overall survival (OS), the results for both analysis are presented here. </span>
<span id="cb1-1046"><a href="#cb1-1046" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1049"><a href="#cb1-1049" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb1-1050"><a href="#cb1-1050" aria-hidden="true" tabindex="-1"></a>tables_survival <span class="sc">%&gt;%</span> </span>
<span id="cb1-1051"><a href="#cb1-1051" aria-hidden="true" tabindex="-1"></a>    DT<span class="sc">::</span><span class="fu">datatable</span>(<span class="at">options =</span> <span class="fu">list</span>(<span class="at">scrollX =</span> <span class="cn">TRUE</span>), <span class="at">filter =</span> <span class="st">'top'</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb1-1052"><a href="#cb1-1052" aria-hidden="true" tabindex="-1"></a>    DT<span class="sc">::</span><span class="fu">formatRound</span>(</span>
<span id="cb1-1053"><a href="#cb1-1053" aria-hidden="true" tabindex="-1"></a>        <span class="at">columns=</span><span class="fu">c</span>(</span>
<span id="cb1-1054"><a href="#cb1-1054" aria-hidden="true" tabindex="-1"></a>            <span class="st">"estimate"</span>, <span class="st">"std.error"</span>, <span class="st">"statistic"</span>, <span class="st">"HR"</span></span>
<span id="cb1-1055"><a href="#cb1-1055" aria-hidden="true" tabindex="-1"></a>        ), </span>
<span id="cb1-1056"><a href="#cb1-1056" aria-hidden="true" tabindex="-1"></a>        <span class="at">digits =</span> <span class="dv">2</span></span>
<span id="cb1-1057"><a href="#cb1-1057" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span> </span>
<span id="cb1-1058"><a href="#cb1-1058" aria-hidden="true" tabindex="-1"></a>    DT<span class="sc">::</span><span class="fu">formatSignif</span>(<span class="at">columns =</span> <span class="fu">c</span>(<span class="st">"p.value"</span>))</span>
<span id="cb1-1059"><a href="#cb1-1059" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-1060"><a href="#cb1-1060" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1061"><a href="#cb1-1061" aria-hidden="true" tabindex="-1"></a>The table above shows that $SET_{ER/PR}$ had a small hazard ratio (&lt; 1) in all</span>
<span id="cb1-1062"><a href="#cb1-1062" aria-hidden="true" tabindex="-1"></a>4 analysis performed. Moreover, for all cases where a measure of estrogen</span>
<span id="cb1-1063"><a href="#cb1-1063" aria-hidden="true" tabindex="-1"></a>signaling was used, the hazard ratio was below 1, indicating that the higher</span>
<span id="cb1-1064"><a href="#cb1-1064" aria-hidden="true" tabindex="-1"></a>the score, the less likely the patient is to suffer the event in a specific</span>
<span id="cb1-1065"><a href="#cb1-1065" aria-hidden="true" tabindex="-1"></a>timepoint. This indicates that ER signaling is actually something continuous</span>
<span id="cb1-1066"><a href="#cb1-1066" aria-hidden="true" tabindex="-1"></a>and not dichotomous.</span>
<span id="cb1-1067"><a href="#cb1-1067" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1068"><a href="#cb1-1068" aria-hidden="true" tabindex="-1"></a>@fig-forest-plot shows the forest plot of $SET_{ER/PR}$ for all </span>
<span id="cb1-1069"><a href="#cb1-1069" aria-hidden="true" tabindex="-1"></a>three cohorts. </span>
<span id="cb1-1070"><a href="#cb1-1070" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1071"><a href="#cb1-1071" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.height=10, fig.width = 7}</span></span>
<span id="cb1-1072"><a href="#cb1-1072" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-forest-plot</span></span>
<span id="cb1-1073"><a href="#cb1-1073" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Forest plots of the different cohorts and their hazard ratios.</span></span>
<span id="cb1-1074"><a href="#cb1-1074" aria-hidden="true" tabindex="-1"></a><span class="co">#|     The bars correspond to 95% confidence interval.</span></span>
<span id="cb1-1075"><a href="#cb1-1075" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1076"><a href="#cb1-1076" aria-hidden="true" tabindex="-1"></a><span class="co"># we use the workaround described here:</span></span>
<span id="cb1-1077"><a href="#cb1-1077" aria-hidden="true" tabindex="-1"></a><span class="co"># https://stackoverflow.com/questions/71266791/combine-several-forestplot-object-in-one-graph-in-r</span></span>
<span id="cb1-1078"><a href="#cb1-1078" aria-hidden="true" tabindex="-1"></a><span class="co"># we convert the forest plots to grob and then used gridExtra to </span></span>
<span id="cb1-1079"><a href="#cb1-1079" aria-hidden="true" tabindex="-1"></a><span class="co"># plot into the output</span></span>
<span id="cb1-1080"><a href="#cb1-1080" aria-hidden="true" tabindex="-1"></a>which_pathway <span class="ot">&lt;-</span> <span class="st">"SET_ERPR"</span></span>
<span id="cb1-1081"><a href="#cb1-1081" aria-hidden="true" tabindex="-1"></a>plots_seterpr <span class="ot">&lt;-</span> <span class="fu">lapply</span>(</span>
<span id="cb1-1082"><a href="#cb1-1082" aria-hidden="true" tabindex="-1"></a>    forest_plots<span class="sc">$</span>os, </span>
<span id="cb1-1083"><a href="#cb1-1083" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(x){</span>
<span id="cb1-1084"><a href="#cb1-1084" aria-hidden="true" tabindex="-1"></a>        ggplotify<span class="sc">::</span><span class="fu">grid2grob</span>(<span class="fu">print</span>(x[[which_pathway]]))</span>
<span id="cb1-1085"><a href="#cb1-1085" aria-hidden="true" tabindex="-1"></a>    } </span>
<span id="cb1-1086"><a href="#cb1-1086" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-1087"><a href="#cb1-1087" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1088"><a href="#cb1-1088" aria-hidden="true" tabindex="-1"></a>gridExtra<span class="sc">::</span><span class="fu">grid.arrange</span>(</span>
<span id="cb1-1089"><a href="#cb1-1089" aria-hidden="true" tabindex="-1"></a>    plots_seterpr<span class="sc">$</span>tcga, </span>
<span id="cb1-1090"><a href="#cb1-1090" aria-hidden="true" tabindex="-1"></a>    plots_seterpr<span class="sc">$</span>scanb,</span>
<span id="cb1-1091"><a href="#cb1-1091" aria-hidden="true" tabindex="-1"></a>    plots_seterpr<span class="sc">$</span>metabric</span>
<span id="cb1-1092"><a href="#cb1-1092" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-1093"><a href="#cb1-1093" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-1094"><a href="#cb1-1094" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1095"><a href="#cb1-1095" aria-hidden="true" tabindex="-1"></a>The hazard ratios are all below 1 and small for $SET_{ER/PR}$.</span>
<span id="cb1-1096"><a href="#cb1-1096" aria-hidden="true" tabindex="-1"></a>The variability changes depending on the cohort, </span>
<span id="cb1-1097"><a href="#cb1-1097" aria-hidden="true" tabindex="-1"></a>specially because they have different follow-up times, </span>
<span id="cb1-1098"><a href="#cb1-1098" aria-hidden="true" tabindex="-1"></a>SCANB being the shortest. TCGA has very few events, but using the 10</span>
<span id="cb1-1099"><a href="#cb1-1099" aria-hidden="true" tabindex="-1"></a>to 1 rule of thumb, </span>
<span id="cb1-1100"><a href="#cb1-1100" aria-hidden="true" tabindex="-1"></a>meaning 10 events for each covariate added, the regression</span>
<span id="cb1-1101"><a href="#cb1-1101" aria-hidden="true" tabindex="-1"></a>satisfies the rule of thumb.</span>
<span id="cb1-1102"><a href="#cb1-1102" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1103"><a href="#cb1-1103" aria-hidden="true" tabindex="-1"></a><span class="fu">## Conclusion</span></span>
<span id="cb1-1104"><a href="#cb1-1104" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1105"><a href="#cb1-1105" aria-hidden="true" tabindex="-1"></a>In this chapter we've shown that ER+ BC patients are very distinct</span>
<span id="cb1-1106"><a href="#cb1-1106" aria-hidden="true" tabindex="-1"></a>from each other, as it can be seen from the umap projections and </span>
<span id="cb1-1107"><a href="#cb1-1107" aria-hidden="true" tabindex="-1"></a>the subtypes. These patients might respond differently for endocrine therapy</span>
<span id="cb1-1108"><a href="#cb1-1108" aria-hidden="true" tabindex="-1"></a>as well, and this might depend on the ER signaling, how active it is. </span>
<span id="cb1-1109"><a href="#cb1-1109" aria-hidden="true" tabindex="-1"></a>Therefore, when deciding a treatment, more care should be taken with</span>
<span id="cb1-1110"><a href="#cb1-1110" aria-hidden="true" tabindex="-1"></a>ER+ BC patients and check their signaling scores somehow. The $SET_{ER/PR}$</span>
<span id="cb1-1111"><a href="#cb1-1111" aria-hidden="true" tabindex="-1"></a>signature is a good signature showing very good hazard ratios across</span>
<span id="cb1-1112"><a href="#cb1-1112" aria-hidden="true" tabindex="-1"></a>the different cohorts. This signature has also been validated on the clinics</span>
<span id="cb1-1113"><a href="#cb1-1113" aria-hidden="true" tabindex="-1"></a>for use.</span>
<span id="cb1-1114"><a href="#cb1-1114" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-1115"><a href="#cb1-1115" aria-hidden="true" tabindex="-1"></a>Knowing the ER signaling for a patient is very important when deciding</span>
<span id="cb1-1116"><a href="#cb1-1116" aria-hidden="true" tabindex="-1"></a>treatment, but not enough. What could be other alternatives for patients</span>
<span id="cb1-1117"><a href="#cb1-1117" aria-hidden="true" tabindex="-1"></a>that have low ER signaling and are still considered ER+? Should they use</span>
<span id="cb1-1118"><a href="#cb1-1118" aria-hidden="true" tabindex="-1"></a>only endocrine therapy or supplement it with something else? In the</span>
<span id="cb1-1119"><a href="#cb1-1119" aria-hidden="true" tabindex="-1"></a>next chapters we present a framework where we can take a look</span>
<span id="cb1-1120"><a href="#cb1-1120" aria-hidden="true" tabindex="-1"></a>at a more personalised approach for treatments. </span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-right">
      <ul class="footer-items list-unstyled">
    <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/chronchi">
      <i class="bi bi-github" role="img">
</i> 
    </a>
  </li>  
    <li class="nav-item compact">
    <a class="nav-link" href="https://twitter.com/chronchi">
      <i class="bi bi-twitter" role="img">
</i> 
    </a>
  </li>  
</ul>
    </div>
  </div>
</footer>



</body></html>