<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.0.37">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Molecular landscape: a new framework for personalized medicine in Breast Cancer. - 6&nbsp; Tests and tries</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./discussion.html" rel="next">
<link href="./risk_score.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>

<script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="site_libs/datatables-binding-0.22/datatables.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="site_libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet">
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet">

  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
    <div class="container-fluid d-flex justify-content-between">
      <h1 class="quarto-secondary-nav-title"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Tests and tries</span></h1>
      <button type="button" class="quarto-btn-toggle btn" aria-label="Show secondary navigation">
        <i class="bi bi-chevron-right"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Molecular landscape: a new framework for personalized medicine in Breast Cancer.</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/chronchi/molecular_landscape" title="Source Code" class="sidebar-tool px-1"><i class="bi bi-github"></i></a>
    <a href="https://twitter.com/intent/tweet?url=|url|" title="Twitter" class="sidebar-tool px-1"><i class="bi bi-twitter"></i></a>
</div>
    </div>
      </div>
      <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">Preface</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./intro.html" class="sidebar-item-text sidebar-link">Introduction</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./surv_analysis_estrogen.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Estrogen receptor status is continuous, not dichotomous</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./pca_merging.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">A new framework for personalized medicine: integrating</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./validation.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Validating the molecular landscape</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./scoring.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Pathway activity in a personalized context</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./risk_score.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Risk score model</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./trying.html" class="sidebar-item-text sidebar-link active"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Tests and tries</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./discussion.html" class="sidebar-item-text sidebar-link">Discussion</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./references.html" class="sidebar-item-text sidebar-link">References</a>
  </div>
</li>
    </ul>
    </div>
</nav>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#embedding-with-more-than-1000-genes" id="toc-embedding-with-more-than-1000-genes" class="nav-link active" data-scroll-target="#embedding-with-more-than-1000-genes"> <span class="header-section-number">6.1</span> Embedding with more than 1000 genes</a></li>
  <li><a href="#removing-the-batch-effects" id="toc-removing-the-batch-effects" class="nav-link" data-scroll-target="#removing-the-batch-effects"> <span class="header-section-number">6.2</span> Removing the batch effects</a></li>
  <li><a href="#scoring-strategy-with-regressed-data" id="toc-scoring-strategy-with-regressed-data" class="nav-link" data-scroll-target="#scoring-strategy-with-regressed-data"> <span class="header-section-number">6.3</span> Scoring strategy with regressed data</a>
  <ul class="collapse">
  <li><a href="#checking-the-patients-from-poetic-with-new-scores" id="toc-checking-the-patients-from-poetic-with-new-scores" class="nav-link" data-scroll-target="#checking-the-patients-from-poetic-with-new-scores"> <span class="header-section-number">6.3.1</span> Checking the patients from POETIC with new scores</a></li>
  <li><a href="#responder-vs-non-responder-comparison-of-new-scores" id="toc-responder-vs-non-responder-comparison-of-new-scores" class="nav-link" data-scroll-target="#responder-vs-non-responder-comparison-of-new-scores"> <span class="header-section-number">6.3.2</span> Responder vs non responder comparison of new scores</a></li>
  <li><a href="#survival-analysis-with-the-new-scores" id="toc-survival-analysis-with-the-new-scores" class="nav-link" data-scroll-target="#survival-analysis-with-the-new-scores"> <span class="header-section-number">6.3.3</span> Survival analysis with the new scores</a></li>
  </ul></li>
  <li><a href="#regressing-the-pdxs" id="toc-regressing-the-pdxs" class="nav-link" data-scroll-target="#regressing-the-pdxs"> <span class="header-section-number">6.4</span> Regressing the PDXs</a></li>
  <li><a href="#singscore" id="toc-singscore" class="nav-link" data-scroll-target="#singscore"> <span class="header-section-number">6.5</span> singscore</a></li>
  <li><a href="#using-several-pathways-or-just-one-pathway-for-gsva" id="toc-using-several-pathways-or-just-one-pathway-for-gsva" class="nav-link" data-scroll-target="#using-several-pathways-or-just-one-pathway-for-gsva"> <span class="header-section-number">6.6</span> Using several pathways or just one pathway for GSVA</a>
  <ul class="collapse">
  <li><a href="#one-vs-all" id="toc-one-vs-all" class="nav-link" data-scroll-target="#one-vs-all"> <span class="header-section-number">6.6.1</span> One vs all</a></li>
  </ul></li>
  <li><a href="#minimum-number-of-samples-for-gsva" id="toc-minimum-number-of-samples-for-gsva" class="nav-link" data-scroll-target="#minimum-number-of-samples-for-gsva"> <span class="header-section-number">6.7</span> Minimum number of samples for GSVA</a></li>
  <li><a href="#normalization-strategy" id="toc-normalization-strategy" class="nav-link" data-scroll-target="#normalization-strategy"> <span class="header-section-number">6.8</span> Normalization strategy</a></li>
  <li><a href="#ilc-and-position-in-the-molecular-landscape" id="toc-ilc-and-position-in-the-molecular-landscape" class="nav-link" data-scroll-target="#ilc-and-position-in-the-molecular-landscape"> <span class="header-section-number">6.9</span> ILC and position in the molecular landscape</a></li>
  <li><a href="#genefu" id="toc-genefu" class="nav-link" data-scroll-target="#genefu"> <span class="header-section-number">6.10</span> genefu</a>
  <ul class="collapse">
  <li><a href="#genefu-and-aims" id="toc-genefu-and-aims" class="nav-link" data-scroll-target="#genefu-and-aims"> <span class="header-section-number">6.10.1</span> genefu and AIMS</a></li>
  </ul></li>
  <li><a href="#nanostring-signatures" id="toc-nanostring-signatures" class="nav-link" data-scroll-target="#nanostring-signatures"> <span class="header-section-number">6.11</span> nanostring signatures</a></li>
  <li><a href="#risk-stratification-and-the-molecular-landscape" id="toc-risk-stratification-and-the-molecular-landscape" class="nav-link" data-scroll-target="#risk-stratification-and-the-molecular-landscape"> <span class="header-section-number">6.12</span> Risk stratification and the molecular landscape</a></li>
  <li><a href="#late-distant-recurrence-trying-to-find-biomarkers" id="toc-late-distant-recurrence-trying-to-find-biomarkers" class="nav-link" data-scroll-target="#late-distant-recurrence-trying-to-find-biomarkers"> <span class="header-section-number">6.13</span> Late distant recurrence: trying to find biomarkers</a></li>
  </ul>
<div class="toc-actions"><div><i class="bi bi-github"></i></div><div class="action-links"><p><a href="https://github.com/chronchi/molecular_landscape/issues/new" class="toc-action">Report an issue</a></p></div></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title d-none d-lg-block"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Tests and tries</span></h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
</div>



<div class="quarto-title-meta">

    
    
  </div>
  

</header>

<p>We have used some numbers of genes for doing the embedding. Here we try to perform the embedding with all the genes (~10000 genes) and then we try to regress out the components.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>* The library is already synchronized with the lockfile.</code></pre>
</div>
</div>
<section id="embedding-with-more-than-1000-genes" class="level2" data-number="6.1">
<h2 data-number="6.1" class="anchored" data-anchor-id="embedding-with-more-than-1000-genes"><span class="header-section-number">6.1</span> Embedding with more than 1000 genes</h2>
<div class="cell">

</div>
<p>And now we can visualize the results.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pca-embeddings-1" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="trying_files/figure-html/fig-pca-embeddings-1-1.png" class="img-fluid figure-img" width="1344"></p>
<p></p><figcaption class="figure-caption">Figure 6.1: PCA projections colored by different factors and organized by different components. (A) Plot of the first two components colored by cohort. (B) Plot of the first two components colored by ER status. (C) Plot of the second and third components colored by cohort. (D) Plot of the second and third components colored by ER status. (E) Plot of the second and third components colored by PAM50. (F) Plot of the second and third components colored by INTCLUST, only METABRIC has an assigned value for this variable, NAs are TCGA samples.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<div class="cell">

</div>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-4-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">

</div>
<p>And now we can visualize the results.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pca-embeddings-2" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="trying_files/figure-html/fig-pca-embeddings-2-1.png" class="img-fluid figure-img" width="1344"></p>
<p></p><figcaption class="figure-caption">Figure 6.2: PCA projections colored by different factors and organized by different components. (A) Plot of the first two components colored by cohort. (B) Plot of the first two components colored by ER status. (C) Plot of the second and third components colored by cohort. (D) Plot of the second and third components colored by ER status. (E) Plot of the second and third components colored by PAM50. (F) Plot of the second and third components colored by INTCLUST, only METABRIC has an assigned value for this variable, NAs are TCGA samples.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<div class="cell">

</div>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-8-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pca-scanb-pc1" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="trying_files/figure-html/fig-pca-scanb-pc1-1.png" class="img-fluid figure-img" width="672"></p>
<p></p><figcaption class="figure-caption">Figure 6.3: PCA embedding colored by cohort. The components used are the first and second components.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p><a href="#fig-pca-scanb-er-pam50-all-genes">Figure&nbsp;<span>6.4</span></a> shows that the SCANB samples are also well mixed regarding the clinical factors, including the <span class="math inline">\(SET_{ER/PR}\)</span> signature.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pca-scanb-er-pam50-all-genes" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="trying_files/figure-html/fig-pca-scanb-er-pam50-all-genes-1.png" class="img-fluid figure-img" width="1536"></p>
<p></p><figcaption class="figure-caption">Figure 6.4: PCA embedding of all samples from TCGA, SCANB and METABRIC. (A) Colored by cohort, (B) colored by ER status, (C) colored by PAM50 molecular subtype, (D) colored by the <span class="math inline">\(SET_{ER/PR}\)</span> signature and only SCANB samples.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-11-1.png" class="img-fluid" width="1152"></p>
</div>
</div>
<p>Let us also check the other components in the biplots.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid" width="1536"></p>
</div>
</div>
<p>We see that the pam50 subtypes are being separated using the components 3, 4 and also 8. In the component 8 there is some expression that is able to differentiate between HER2-like from all other samples.</p>
<p>We now plot by cohort.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-13-1.png" class="img-fluid" width="1536"></p>
</div>
</div>
<p>In PC1, 2 and 5 there are some small differences in the differences of the distributions from the datasets. We will then remove these 3 components up to component 10 with exception of components 3, 4 and 5.</p>
</section>
<section id="removing-the-batch-effects" class="level2" data-number="6.2">
<h2 data-number="6.2" class="anchored" data-anchor-id="removing-the-batch-effects"><span class="header-section-number">6.2</span> Removing the batch effects</h2>
<p>We now try to remove the batch effects by removing all the PCs that are not of interest, so we remove the first two components and the fifth as well. We then calculate the scores for each cohort separately and then compare to what was calculated before. We can then try to calculate the scores using samples coming from all cohorts.</p>
<div class="cell">

</div>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-15-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-16-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>In general it looks good, the luminal B population is a bit compressed towards the luminal A population though.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-17-1.png" class="img-fluid" width="576"></p>
</div>
</div>
<p>There is a very good merging of the datasets. We see mostly green points because they are the majority.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-18-1.png" class="img-fluid" width="960"></p>
</div>
</div>
<p>Now we see that the first loadings has a tail in a similar fashio as the other loadings.</p>
<p>We now try to calculate the scores of the samples using these new embeddings only from TCGA samples as we will compare the results later on.</p>
<div class="cell">

</div>
<p>And we can now check and compare the scores.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-20-1.png" class="img-fluid" width="960"></p>
</div>
</div>
<p>We see that for the three pathways the correlation is positive and strong. The difference is that it is a bit noisy, so the correlation is not perfect. What is interesting to see is that at the extremes the scores have less variation usually.</p>
<p>We now calculate the scores from SCANB and compare the results. Remember that SCANB was not used for the PCA training.</p>
<div class="cell">

</div>
<p>And we can now check and compare the scores.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-22-1.png" class="img-fluid" width="960"></p>
</div>
</div>
<p>The same phenomenon happened. Samples in the endges have lower variability, but samples in the middle are the ones that have high variability. For example there are several samples from the regressed data with SET_ERPR score close to 0.25 and negative scores.</p>
<p>What happens now if we use another dataset and then use only one sample to calculate the score. Let us include SCANB samples in the pipeline. For this we select 50 scanb samples and 100 random samples from tcga and metabric. We then combine all these samples together and calculate the GSVA scores.</p>
<div class="cell">

</div>
<p>We now plot only the scores from the SCANB samples.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-24-1.png" class="img-fluid" width="960"></p>
</div>
</div>
<p>So it seems there are still dataset related problems there when calculating the scores. The scores are positively correlated, but they are not quite the same as only using the SCANB dataset. Moreover, for G2M checkpoint the scores are concentrated around -0.4 from the regressed data but they are in a range going from - 0.4 to 0.1 in the unregressed data.</p>
<p>If we try to instead add some samples from the SCANB dataset as well randomly into the mix. How well does the scores get compared to the “ground truth”?</p>
<div class="cell">

</div>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-26-1.png" class="img-fluid" width="960"></p>
</div>
</div>
<p>It does not work so well when including some SCANB samples together with the METABRIC and TCGA samples. We are trying to achieve a correlation of almost 1 with very low variability since we want to use the scores clinically.</p>
<p>Let us check the distribution of some genes for SCANB and TCGA after regressing out the first two PCs.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-27-1.png" class="img-fluid" width="960"></p>
</div>
</div>
<p>We see that even though the expression profile of the samples are overlapping there are discrepancies. That is why when using GSVA it does not work so well perhaps.</p>
</section>
<section id="scoring-strategy-with-regressed-data" class="level2" data-number="6.3">
<h2 data-number="6.3" class="anchored" data-anchor-id="scoring-strategy-with-regressed-data"><span class="header-section-number">6.3</span> Scoring strategy with regressed data</h2>
<p>We now try to use a slightly different scoring strategy. We can think of our data in a log scale, so what we do now instead is to sum all of our genes of interest and then subtract an average of the housekeeping genes expression levels.</p>
<p>First we check the distribution of this average for each cohort separately.</p>
<div class="cell">

</div>
<div class="cell">
<div class="cell-output-display">
<div id="fig-hkg-regressed" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="trying_files/figure-html/fig-hkg-regressed-1.png" class="img-fluid figure-img" width="768"></p>
<p></p><figcaption class="figure-caption">Figure 6.5: Distribution of the average of the housekeeping genes for distinct cohorts.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>Indeed the tcga and metabric cohorts have an overlapping distribution, but the SCANB does not overlap completely, it is actually shifted. One should still take into account that the distributions are very close to 0, meaning that the housekeeping genes have very close to 0 values, as expected here in this case. We now calculate the score by taking this into account.</p>
<div class="cell">

</div>
<div class="cell">

</div>
<div class="cell">
<div class="cell-output-display">
<div id="fig-scores-regressed" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="trying_files/figure-html/fig-scores-regressed-1.png" class="img-fluid figure-img" width="1344"></p>
<p></p><figcaption class="figure-caption">Figure 6.6: Scores obtained from regressed data of all the four cohorts TCGA, SCANB, METABRIC and POETIC compared to the original GSVA scores.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>The scores are actually highly correlated, specially the estrogen related pathways. Also the correlation between the scores in general follows the same trends.</p>
<p>The plot below is a subset of the scores above to include in the publication.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-scores-regressed-subset" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="trying_files/figure-html/fig-scores-regressed-subset-1.png" class="img-fluid figure-img" width="1440"></p>
<p></p><figcaption class="figure-caption">Figure 6.7: Scores obtained from regressed data of all the four cohorts TCGA, SCANB, METABRIC and POETIC compared to the original GSVA scores.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>We now compare the distributions of these scores. Across the different cohorts and pathways of interest.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-34-1.png" class="img-fluid" width="1536"></p>
</div>
</div>
<p>And with all molecular subtypes together.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-35-1.png" class="img-fluid" width="960"></p>
</div>
</div>
<p>We see that the overlap is actually pretty good in general, even for SCANB. The only part where there is a bigger discrepancy is for G2M checkpoint. We see below that by calculating the GSVA score on the whole scanb cohort we don’t have this problem.</p>
<p>And as a comparison we plot with the original scores using GSVA on the whole cohort individually.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-36-1.png" class="img-fluid" width="1536"></p>
</div>
</div>
<p>And all the subtypes together for the original score:</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-37-1.png" class="img-fluid" width="960"></p>
</div>
</div>
<p>We see that by doing GSVA the distributions they overlap much more than using the regressed data and the simple sum of the values for those genes. Still for estrogen signaling it seems that the estrogen early signature is pretty robust.</p>
<section id="checking-the-patients-from-poetic-with-new-scores" class="level3" data-number="6.3.1">
<h3 data-number="6.3.1" class="anchored" data-anchor-id="checking-the-patients-from-poetic-with-new-scores"><span class="header-section-number">6.3.1</span> Checking the patients from POETIC with new scores</h3>
<p>We now use the new scores to evaluate the comparison with the neighboorhood, just as before in the previous chapter where we used the POETIC trial data.</p>
<div class="cell">

</div>
<p>Patient 63 <a href="#fig-pt63">Figure&nbsp;<span>6.8</span></a> is considered a responder. According to the scores, when comparing the estrogen early signature to its average distribution in the neighborhood, the score is higher.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pt63" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="trying_files/figure-html/fig-pt63-1.png" class="img-fluid figure-img" width="960"></p>
<p></p><figcaption class="figure-caption">Figure 6.8: Posterior distribution of the average scores in the neighborhood of patient 63 from POETIC trial. Each dot corresponds to a 1% quantile.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>Patient 236 <a href="#fig-pt236">Figure&nbsp;<span>6.9</span></a> was considered a non responder. According to the scores, when comparing the estrogen early signature to its average distribution in the neighborhood, the score is lower, being in the 5% quantile.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pt236" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="trying_files/figure-html/fig-pt236-1.png" class="img-fluid figure-img" width="960"></p>
<p></p><figcaption class="figure-caption">Figure 6.9: Posterior distribution of the average scores in the neighborhood of patient 236 from POETIC trial. Each dot corresponds to a 1% quantile.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>When comparing these two patients, there is also a difference in the androgen response score, which could be a reflection of different estrogen signaling.</p>
<p>It reflects what we saw previously as well.</p>
</section>
<section id="responder-vs-non-responder-comparison-of-new-scores" class="level3" data-number="6.3.2">
<h3 data-number="6.3.2" class="anchored" data-anchor-id="responder-vs-non-responder-comparison-of-new-scores"><span class="header-section-number">6.3.2</span> Responder vs non responder comparison of new scores</h3>
<p>We now compare the scores for responders and non responders.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-41-1.png" class="img-fluid" width="768"></p>
</div>
</div>
<p>The higher E2F targets in average the lower the estrogen early score is for non responders, whereas this correlation does not exist for responder patients. This difference is mostly drive by the high E2F target non responder tumors.</p>
<p>Now the plot below shows the correlation for baselien Ki67% and E2F targets for the score calculated in the regressed data in a single sample manner. We see a positive correlation between the two scores. Moreover, it is interesting to see that when the tumor has low Ki67, there is still a spectrum on the E2F targets score. Also there is no difference in the correlation between responders and non responders.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-42-1.png" class="img-fluid" width="768"></p>
</div>
</div>
<p>And lastly we compare the estrogen signaling signatures between responders and non-responder.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-43-1.png" class="img-fluid" width="960"></p>
</div>
</div>
<p>We see that the estrogen response early has a wider range when compared to the SET ER/PR score. That makes sense as this signature has over 100 genes, whereas SET ER/PR has only 18. Also the responder was very good in discriminating the responders and non responders. What we can conclude from this figure is that the samples with very low scores are usually non responders, which reflect their position in the molecular landscape, also in general responders have higher ER signaling than non responders.</p>
<p>Below is the figure with the GSVA scores before regressing the data, we see that SET ER/PR is not able to really distinguish completely the non-responders with very low score similar to what was done with GSVA. Even estrogen early had some differences.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-44-1.png" class="img-fluid" width="960"></p>
</div>
</div>
<p>And below is a figure with both GSVA and regressed scores together.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-comparison-scoring-strategy-poetic" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="trying_files/figure-html/fig-comparison-scoring-strategy-poetic-1.png" class="img-fluid figure-img" width="960"></p>
<p></p><figcaption class="figure-caption">Figure 6.10: Comparing scoring strategies for responders and non responders in the POETIC dataset.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>Below we show the correlation between these scores for all cohorts.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-46-1.png" class="img-fluid" width="576"></p>
</div>
</div>
<p>These scores are highly correlated but in different absolute scales. We now check the change in Ki67 for all patients and compare the scores with the original ones obtained with GSVA.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-comparison-og-regressed-er" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="trying_files/figure-html/fig-comparison-og-regressed-er-1.png" class="img-fluid figure-img" width="1152"></p>
<p></p><figcaption class="figure-caption">Figure 6.11: Comparison of original GSVA and regressed scores between responders and non responders compared to change in Ki67.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>We see that the bigger the change the higher the score is when using the regressed data. Also the non responders tend to have scores mostly in the negative scale, whereas the responders are more shifted to the right. SET ER/PR is also a good measure here.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-48-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>There is a very good match between the two scores in general, meaning that the regression scoring strategy capture relatively well the scores. There is one patient that is a non-responder and has a low regressed estrogen early but a GSVA score close to 0, let us check this patient.</p>
<div class="cell">
<div class="cell-output-display">

<div id="htmlwidget-4de5c1843110208f8d96" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-4de5c1843110208f8d96">{"x":{"filter":"none","vertical":false,"data":[["1"],["80"],[43.2412247946229],[-5.64942273914443],[26.832381276351],["pos"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>patient_nb<\/th>\n      <th>ki67<\/th>\n      <th>PC3<\/th>\n      <th>change_ki67<\/th>\n      <th>er_status<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<p>It is a patient that is far to the left on the molecular landscape, among the basal like, and had an increase in Ki67 upon ET. So the new scoring system was able to capture this patient as with low score.</p>
</section>
<section id="survival-analysis-with-the-new-scores" class="level3" data-number="6.3.3">
<h3 data-number="6.3.3" class="anchored" data-anchor-id="survival-analysis-with-the-new-scores"><span class="header-section-number">6.3.3</span> Survival analysis with the new scores</h3>
<p>We can check how robust these scores are by performing survival analysis.</p>
<div class="cell">

</div>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pca-mol-land-survival-regressed" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="trying_files/figure-html/fig-pca-mol-land-survival-regressed-1.png" class="img-fluid figure-img" width="864"></p>
<p></p><figcaption class="figure-caption">Figure 6.12: Overall survival analysis from SCANB and METABRIC cohorts and their scores. The formulas used were the same as for the estrogen signaling analysis. Coefficients were scaled before performing cox regression</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>The plot with TCGA is shown below.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-52-1.png" class="img-fluid" width="432"></p>
</div>
</div>
<p>Scores for TCGA are noisier than SCANB and METABRIC. Though the scaled scores they have comparable hazard ratios and also they follow what we saw previously. The only pathway that is a bit different is the PI3K AKT MTOR signaling. This pathway had a noisier correlation with the GSVA score as well.</p>
</section>
</section>
<section id="regressing-the-pdxs" class="level2" data-number="6.4">
<h2 data-number="6.4" class="anchored" data-anchor-id="regressing-the-pdxs"><span class="header-section-number">6.4</span> Regressing the PDXs</h2>
<p>We can use the regressing strategy to score the PDXs and then evaluate their scores.</p>
<div class="cell">

</div>
<p>We start by comparing the scores across the different control samples and their positions in the molecular landscape.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pca-pdx-control-only-regressed" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="trying_files/figure-html/fig-pca-pdx-control-only-regressed-1.png" class="img-fluid figure-img" width="864"></p>
<p></p><figcaption class="figure-caption">Figure 6.13: Embedding of only regressed PDX control samples on top of the METABRIC, SCANB and TCGA samples</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>The results are jus twhat we got previously as well, meaning that by using all these genes the information of the PDXs are still captured. The main differences is that the luminal B region is more compact compare to <a href="validation.html#fig-pca-pdx-control-only">Figure&nbsp;<span>3.17</span></a>.</p>
<p>And now that we have the results we can compare the PDXs as well.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pdx-regressed-scores" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="trying_files/figure-html/fig-pdx-regressed-scores-1.png" class="img-fluid figure-img" width="960"></p>
<p></p><figcaption class="figure-caption">Figure 6.14: Scores obtained from the regressed data of the PDXs.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>The random 200 is just a gene set with 200 random genes that works as a control and serves as a comparison to the other pathways.</p>
<p>And below we show the distribution of some of the scores calculated for all the other cohorts as a comparison.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-56-1.png" class="img-fluid" width="768"></p>
</div>
</div>
<p>We see that METS15 is the one with the highest proliferation and lowest EMT, it is a PDX that comes from a pleural efusion. Also it is the PDX with the lowest apoptosis and highest DNA repair, probably due to a higher replication rate. Moreover, all these PDXs they have a estrogen response early scores higher than 0 in average, which is about the average of the scores in the ER+ BC samples.</p>
<p>Now we check what happens with some of the PDXs when they are treated with P4, to compare with <a href="validation.html#fig-pdx-p4-control-pathways">Figure&nbsp;<span>3.19</span></a>.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pdx-p4-control-pathways-regressed." class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="trying_files/figure-html/fig-pdx-p4-control-pathways-regressed.-1.png" class="img-fluid figure-img" width="1248"></p>
<p></p><figcaption class="figure-caption">Figure 6.15: Comparison of position in the molecular landscape between CTRL, E2 and P4 treated samples with the regressed data</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>The results are very similar to what was seen previously using the GSVA. What is interesting here is that we have an absolute scale, so we can compare the scores.</p>
</section>
<section id="singscore" class="level2" data-number="6.5">
<h2 data-number="6.5" class="anchored" data-anchor-id="singscore"><span class="header-section-number">6.5</span> singscore</h2>
<p>We can try to use singscore instead, it is a sample dependent measure so it does not estimate the distribution of the gene expression levels before calculating the enrichment score.</p>
<div class="cell">

</div>
<p>We start by comparing with the GSVA scores.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-59-1.png" class="img-fluid" width="1920"></p>
</div>
</div>
<p>We see that in all cases there is a correlation, but the interpretation changes. In some of the pathways there is difference in the singscores based on the cohort, this is not seen on GSVA.</p>
<p>The next two figures are the distributions of the singscore and GSVA filled by cohort.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-60-1.png" class="img-fluid" width="1920"></p>
</div>
</div>
<p>For several pathways in the singscore the distributions are similar, but that is not always the case.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-61-1.png" class="img-fluid" width="1920"></p>
</div>
</div>
<p>On the other hand GSVA gives very similar distributions for all the different cohorts. Therefore we actually need to use GSVA. It is important in the end to calculate scores based on the cohort.</p>
<p>We now move on and try the same analysis on the regressed data.</p>
<div class="cell">

</div>
<p>We start by comparing with the GSVA scores.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-63-1.png" class="img-fluid" width="1920"></p>
</div>
</div>
<p>We see that in all cases there is a correlation, but the interpretation changes. In some of the pathways there is difference in the singscores based on the cohort, this is not seen on GSVA.</p>
<p>The next two figures are the distributions of the singscore and GSVA filled by cohort.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-64-1.png" class="img-fluid" width="1920"></p>
</div>
</div>
<p>In the end the best way of scoring seems to be using GSVA on the unregressed data though it depends on the pathways. For example the hallmark estrogen response early and set ER/PR pathways seem to have been regressed quite well, so we can actually compare the results using the scores. On the other hand we can’t use the pathways G2M checkpoint, E2F targets and EMT. For the PI3K signaling score it seems it is comparable to SCANB only.</p>
</section>
<section id="using-several-pathways-or-just-one-pathway-for-gsva" class="level2" data-number="6.6">
<h2 data-number="6.6" class="anchored" data-anchor-id="using-several-pathways-or-just-one-pathway-for-gsva"><span class="header-section-number">6.6</span> Using several pathways or just one pathway for GSVA</h2>
<p>We now evaluate the effect of the scores for each pathway when using all the 53 pathways combined and each one individually. We use only TCGA for this step.</p>
<div class="cell">

</div>
<section id="one-vs-all" class="level3" data-number="6.6.1">
<h3 data-number="6.6.1" class="anchored" data-anchor-id="one-vs-all"><span class="header-section-number">6.6.1</span> One vs all</h3>
<p>We now calculate the scores for each gene set individually.</p>
<div class="cell">

</div>
<div class="cell">

</div>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-68-1.png" class="img-fluid" width="1920"></p>
</div>
</div>
<p>It seems they are all positively correlated, so one could calculate the scores basically using just one gene set with exception of some of the gene sets.</p>
</section>
</section>
<section id="minimum-number-of-samples-for-gsva" class="level2" data-number="6.7">
<h2 data-number="6.7" class="anchored" data-anchor-id="minimum-number-of-samples-for-gsva"><span class="header-section-number">6.7</span> Minimum number of samples for GSVA</h2>
<p>According to the original GSVA paper, a minimum of 10 samples are necessary to have a good power when using GSVA. Since here we have at least 5 molecular subgroups, we hypothesize we need at least 50 samples to get a good score so we can compare across cohorts. We will try with different numbers of samples using TCGA, we start with 10, 25, 50 and 100 samples and we compare with the scores using the full dataset.</p>
<div class="cell">

</div>
<div class="cell">

</div>
<div class="cell">

</div>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-72-1.png" class="img-fluid" width="4800"></p>
</div>
</div>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-73-1.png" class="img-fluid" width="4800"></p>
</div>
</div>
<p>The conclusion we can take here is that depending on the gene set not even 150 samples is enough. For some only 25 patients is already enough, such as SET ER/PR, androgen response, estrogen response early and E2F targets.</p>
<p>Could we mix around 20 samples of SCANB together with more samples from TCGA and METABRIC to get the scores? Let us try to adress this question again. We are using the regressed data now here as this is the only way.</p>
<div class="cell">

</div>
<div class="cell">

</div>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-76-1.png" class="img-fluid" width="1920"></p>
</div>
</div>
<p>There is still some correlation but the scores are shrinked in the end.</p>
</section>
<section id="normalization-strategy" class="level2" data-number="6.8">
<h2 data-number="6.8" class="anchored" data-anchor-id="normalization-strategy"><span class="header-section-number">6.8</span> Normalization strategy</h2>
<p>We can also try another way of normalizing, namely for each sample we can z scale it.</p>
<div class="cell">

</div>
<p>And now we can visualize the results.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pca-embeddings-3" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="trying_files/figure-html/fig-pca-embeddings-3-1.png" class="img-fluid figure-img" width="1344"></p>
<p></p><figcaption class="figure-caption">Figure 6.16: PCA projections colored by different factors and organized by different components. (A) Plot of the first two components colored by cohort. (B) Plot of the first two components colored by ER status. (C) Plot of the second and third components colored by cohort. (D) Plot of the second and third components colored by ER status. (E) Plot of the second and third components colored by PAM50. (F) Plot of the second and third components colored by INTCLUST, only METABRIC has an assigned value for this variable, NAs are TCGA samples.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<div class="cell">

</div>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-80-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>We see that samples are well embedded as well.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pca-scanb-pc1-2" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="trying_files/figure-html/fig-pca-scanb-pc1-2-1.png" class="img-fluid figure-img" width="672"></p>
<p></p><figcaption class="figure-caption">Figure 6.17: PCA embedding colored by cohort. The components used are the first and second components.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>SCANB and POETIC are in between as expected and they are closer to their sequencing technology. Interestingly the variation for SCANB and TCGA are compressed in the PC1 vs PC2 axis compared to METABRIC and POETIC.</p>
<p><a href="#fig-pca-scanb-er-pam50-2">Figure&nbsp;<span>6.18</span></a> shows that the SCANB samples are also well mixed regarding the clinical factors, including the <span class="math inline">\(SET_{ER/PR}\)</span> signature.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pca-scanb-er-pam50-2" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="trying_files/figure-html/fig-pca-scanb-er-pam50-2-1.png" class="img-fluid figure-img" width="1536"></p>
<p></p><figcaption class="figure-caption">Figure 6.18: PCA embedding of all samples from TCGA, SCANB and METABRIC. (A) Colored by cohort, (B) colored by ER status, (C) colored by PAM50 molecular subtype, (D) colored by the <span class="math inline">\(SET_{ER/PR}\)</span> signature and only SCANB samples.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>In the end normalizing by scaling the sample before doing any embedding works as well. Though the embedding seems to be more compressed. I would argue that the qPCR-like normalization gives a bit better embedding.</p>
</section>
<section id="ilc-and-position-in-the-molecular-landscape" class="level2" data-number="6.9">
<h2 data-number="6.9" class="anchored" data-anchor-id="ilc-and-position-in-the-molecular-landscape"><span class="header-section-number">6.9</span> ILC and position in the molecular landscape</h2>
<p>Invasive lobular carcinoma is a subtype of ER+ BC. They tend to respond better to ET in the short term but they recurr after 20 years. Here we just show where the ILC BC lie in the molecular landscape.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-pca-lobular-scanb" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="trying_files/figure-html/fig-pca-lobular-scanb-1.png" class="img-fluid figure-img" width="864"></p>
<p></p><figcaption class="figure-caption">Figure 6.19: Embedding of the lobular samples from SCANB on top METABRIC and TCGA</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>We see that they are mostly in the normal and luminal A region. Still there are several samples that are luminal B, meaning they are more proliferative.</p>
</section>
<section id="genefu" class="level2" data-number="6.10">
<h2 data-number="6.10" class="anchored" data-anchor-id="genefu"><span class="header-section-number">6.10</span> genefu</h2>
<p>With genefu it is possible to calculate scores from comercially available signatures. Here we try to use it on SCANB, METABRIC and TCGA. Moreover, we can start with ABiM, as this dataset has the gold standard ROR score obtained from nCounter data. This is a good way of validating the pipeline.</p>
<div class="cell">

</div>
<p>We start by comparing the molecular subtypes:</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
FALSE  TRUE 
   14    86 </code></pre>
</div>
</div>
<p>86 out of 100 were correctly called. Here we are using the gold standard wich is the molecular subtype called from FT samples. The normal like subtype is not available from the FT prosigna assay. So we now compare the results with the SSP obtained from the SCANB team.</p>
<p>Below is the confusion matrix to see which subtypes were misclassified.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>Confusion Matrix and Statistics

          Reference
Prediction Basal Her2 LumB LumA Normal
    Basal     13    0    0    0      0
    Her2       0    8    0    0      0
    LumB       1    1   33    1      0
    LumA       1    0    7   32      0
    Normal     1    0    0    2      0

Overall Statistics
                                          
               Accuracy : 0.86            
                 95% CI : (0.7763, 0.9213)
    No Information Rate : 0.4             
    P-Value [Acc &gt; NIR] : &lt; 2.2e-16       
                                          
                  Kappa : 0.7965          
                                          
 Mcnemar's Test P-Value : NA              

Statistics by Class:

                     Class: Basal Class: Her2 Class: LumB Class: LumA
Sensitivity                0.8125      0.8889      0.8250      0.9143
Specificity                1.0000      1.0000      0.9500      0.8769
Pos Pred Value             1.0000      1.0000      0.9167      0.8000
Neg Pred Value             0.9655      0.9891      0.8906      0.9500
Prevalence                 0.1600      0.0900      0.4000      0.3500
Detection Rate             0.1300      0.0800      0.3300      0.3200
Detection Prevalence       0.1300      0.0800      0.3600      0.4000
Balanced Accuracy          0.9062      0.9444      0.8875      0.8956
                     Class: Normal
Sensitivity                     NA
Specificity                   0.97
Pos Pred Value                  NA
Neg Pred Value                  NA
Prevalence                    0.00
Detection Rate                0.00
Detection Prevalence          0.03
Balanced Accuracy               NA</code></pre>
</div>
</div>
<p>All luminal A were predicted correctly. Some normal like were predicted to be luminal A or basal and Her2. There were some mismatches for the LumB, that were misclassified as LumA. Still it is ok.</p>
<p>Let us now calculate the ROR for the ABiM cohort.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-ror-genefu-abim" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="trying_files/figure-html/fig-ror-genefu-abim-1.png" class="img-fluid figure-img" width="672"></p>
<p></p><figcaption class="figure-caption">Figure 6.20: Correlation between ROR obtained from prosigna assay and R package genefu</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>The correlation is relatively good and respects the molecular subtypes in general. The ROR obtained from genefu for some luminal B are not quite right as they are below 50.</p>
<p>We now move on to the other signatures: Mammaprint, EndoPredict (EP) and OncotypeDX Risk Score (RS).</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-ror-mammaprint-genefu-abim" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="trying_files/figure-html/fig-ror-mammaprint-genefu-abim-1.png" class="img-fluid figure-img" width="672"></p>
<p></p><figcaption class="figure-caption">Figure 6.21: Correlation between ROR obtained from prosigna assay and the mammaprint signature obtained from the R package genefu</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p><a href="#fig-ror-mammaprint-genefu-abim">Figure&nbsp;<span>6.21</span></a> shows the correlation between the mammaprint signature (named GENE70) and the ROR from prosigna.</p>
<p>Let us now compare the GENE70 signature with the risk score developed in the previous chapters. We use the ABiM 100 cohort to make the comparison. One side note, we are using only 51 out of the 70 probes available in the package. 15 probes don’t have any corresponding symbol. Out of the available genes, some are duplicated. For the duplicated genes, the correlation coefficients are all similar, so we can select just the first ocurrence in the list.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-89-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>There is a good correlation between the two signatures, suggesting that the Mammaprint signature is somehow based on the position of the samples in the molecular landscape.</p>
<p>We now proceed and calculate the mammaprint scores for SCANB and METABRIC as well and only for ER+ BC samples.</p>
<div class="cell">

</div>
<div class="cell">
<div class="cell-output-display">
<div id="fig-met-scanb-mammaprint" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="trying_files/figure-html/fig-met-scanb-mammaprint-1.png" class="img-fluid figure-img" width="960"></p>
<p></p><figcaption class="figure-caption">Figure 6.22: Embedding of ER+ BC samples colored by the mammaprint score.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>We see that the pattern is very close to what we have from the risk score created in the previous chapter. The figure below shows the correlation between mammaprint and the risk score once again.</p>
<div class="cell">

</div>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-93-1.png" class="img-fluid" width="1344"></p>
</div>
</div>
<p>Now for OncotypeDX:</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-ror-genefu-abim-oncotype" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="trying_files/figure-html/fig-ror-genefu-abim-oncotype-1.png" class="img-fluid figure-img" width="672"></p>
<p></p><figcaption class="figure-caption">Figure 6.23: Correlation between oncotypeDX and R package genefu</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>There is also a correlation between the risk score and oncotypeDX. The paper https://www.nature.com/articles/s41523-022-00492-0#Sec8 also used the genefu for calculating the oncotypeDX.</p>
<section id="genefu-and-aims" class="level3" data-number="6.10.1">
<h3 data-number="6.10.1" class="anchored" data-anchor-id="genefu-and-aims"><span class="header-section-number">6.10.1</span> genefu and AIMS</h3>
<p>Here we try to use AIMS to calculate the PAM50 subtypes for the and compare with the results obtained previously. We want to compare the classifications with the molecular landscape.</p>
<div class="cell">

</div>
<p>The confusion matrix below show the results of the AIMS PAM50 as the predicted and the reference is the PAM50 subtypes obtained from the SCANB team.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>Confusion Matrix and Statistics

          Reference
Prediction luma basal lumb her2 normal
    luma   2645     0  778   73     68
    basal     0   739    0   95     16
    lumb     19     0 1122   20      0
    her2     10     7  237  729     17
    normal  827    14    7   48   1004

Overall Statistics
                                          
               Accuracy : 0.7362          
                 95% CI : (0.7266, 0.7455)
    No Information Rate : 0.4131          
    P-Value [Acc &gt; NIR] : &lt; 2.2e-16       
                                          
                  Kappa : 0.6434          
                                          
 Mcnemar's Test P-Value : NA              

Statistics by Class:

                     Class: luma Class: basal Class: lumb Class: her2
Sensitivity               0.7555      0.97237      0.5233     0.75544
Specificity               0.8152      0.98561      0.9938     0.96391
Pos Pred Value            0.7421      0.86941      0.9664     0.72900
Neg Pred Value            0.8257      0.99725      0.8603     0.96843
Prevalence                0.4131      0.08968      0.2530     0.11386
Detection Rate            0.3121      0.08720      0.1324     0.08602
Detection Prevalence      0.4205      0.10029      0.1370     0.11799
Balanced Accuracy         0.7854      0.97899      0.7586     0.85968
                     Class: normal
Sensitivity                 0.9086
Specificity                 0.8784
Pos Pred Value              0.5284
Neg Pred Value              0.9846
Prevalence                  0.1304
Detection Rate              0.1185
Detection Prevalence        0.2242
Balanced Accuracy           0.8935</code></pre>
</div>
</div>
<p>We see that there is a big misclassification from luminal A samples to normal like samples and several luminal B are misclassified as luminal A.</p>
<p>The plot below shows the coloring by the reference PAM50 and the predicted by AIMS.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-scanb-aims" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="trying_files/figure-html/fig-scanb-aims-1.png" class="img-fluid figure-img" width="672"></p>
<p></p><figcaption class="figure-caption">Figure 6.24: SCANB embedding stratified by PAM50 algorithms by both the ones provided by the SCANB (SSP) and the AIMS algorithm</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>On the left there is the coloring for the AIMS algorithm and on the right from the reference. We see the missclassified samples are not randomly misclassified, they are on the boundaries.</p>
<p>The plot below shows the maximum probability score for each sample individually.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-98-1.png" class="img-fluid" width="384"></p>
</div>
</div>
<p>The majority of samples have probabilities of either 1 or 0. Not so often in between.</p>
<p>And the figure below shows the confusion matrix for the ABIM cohort.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>Confusion Matrix and Statistics

          Reference
Prediction luma basal lumb her2 normal
    luma     28     0   17    1      3
    basal     0    11    0    2      1
    lumb      0     0   16    0      0
    her2      0     0    0   13      0
    normal    2     0    0    2      4

Overall Statistics
                                          
               Accuracy : 0.72            
                 95% CI : (0.6213, 0.8052)
    No Information Rate : 0.33            
    P-Value [Acc &gt; NIR] : 1.786e-15       
                                          
                  Kappa : 0.6291          
                                          
 Mcnemar's Test P-Value : NA              

Statistics by Class:

                     Class: luma Class: basal Class: lumb Class: her2
Sensitivity               0.9333       1.0000      0.4848      0.7222
Specificity               0.7000       0.9663      1.0000      1.0000
Pos Pred Value            0.5714       0.7857      1.0000      1.0000
Neg Pred Value            0.9608       1.0000      0.7976      0.9425
Prevalence                0.3000       0.1100      0.3300      0.1800
Detection Rate            0.2800       0.1100      0.1600      0.1300
Detection Prevalence      0.4900       0.1400      0.1600      0.1300
Balanced Accuracy         0.8167       0.9831      0.7424      0.8611
                     Class: normal
Sensitivity                 0.5000
Specificity                 0.9565
Pos Pred Value              0.5000
Neg Pred Value              0.9565
Prevalence                  0.0800
Detection Rate              0.0400
Detection Prevalence        0.0800
Balanced Accuracy           0.7283</code></pre>
</div>
</div>
<p>We see here that in this case that there are several luminal B samples missclassified as luminal A.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-100-1.png" class="img-fluid" width="768"></p>
</div>
</div>
<p>The misclassified samples from AIMS they really seem to be on the top region together with the luminal A. Some of the normal samples are even in the luminal A region.</p>
<p>Now we check the PDXs.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>$METS15
 treatment Basal Her2 LumA LumB Normal
      CTRL     0    0    0    3      0
       DHT     0    0    0    0      0
        E2     0    0    1    3      0
      E2P4     0    0    1    2      0
        P4     0    0    1    3      0

$T105
 treatment Basal Her2 LumA LumB Normal
      CTRL     0    0    0    0      5
       DHT     0    0    0    0      0
        E2     0    0    0    0      4
      E2P4     0    0    0    0      0
        P4     0    4    0    0      0

$T109
 treatment Basal Her2 LumA LumB Normal
      CTRL     0    0    3    0      1
       DHT     0    0    0    0      0
        E2     0    0    1    0      3
      E2P4     0    0    0    0      0
        P4     0    0    4    0      0

$T110
 treatment Basal Her2 LumA LumB Normal
      CTRL     0    1    0    0      3
       DHT     0    0    0    0      0
        E2     0    0    0    0      2
      E2P4     0    1    0    0      2
        P4     0    0    0    0      4

$T111
 treatment Basal Her2 LumA LumB Normal
      CTRL     0    0    0    0      3
       DHT     0    0    0    0      3
        E2     0    0    0    0      3
      E2P4     0    0    0    0      2
        P4     1    0    0    0      2

$T113
 treatment Basal Her2 LumA LumB Normal
      CTRL     0    0    2    0      1
       DHT     0    0    0    0      0
        E2     0    0    0    0      3
      E2P4     0    0    0    0      0
        P4     0    0    2    0      2</code></pre>
</div>
</div>
<p>These molecular subtyping do not make much sense. We know that by giving P4 estrogen signaling actually goes up, so a expected subtype change is from normal to either luminal A and B. Moreover, these PDXs are in an active state of proliferation due to the MIND model, so being in the normal like region is not a good indication. Also this results show that almost all PDXs would be normal like which does not go in hand with the results we get from the lab.</p>
<p>And the probabilities are available below.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-102-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>And another validation is using the normal cohort. Below is the number of all predicted molecular subtypes.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
Normal 
    66 </code></pre>
</div>
</div>
<p>All samples are normal-like as expected, since they are normal. So it could be that the AIMS algorithm is not suited for PDXs but for clinical samples it works quite well.</p>
</section>
</section>
<section id="nanostring-signatures" class="level2" data-number="6.11">
<h2 data-number="6.11" class="anchored" data-anchor-id="nanostring-signatures"><span class="header-section-number">6.11</span> nanostring signatures</h2>
<p>In this section we try the nanostring panel signatures provided by the Breast Cancer 360 test. We will compare some of the signatures by using GSVA on the original datasets and by calculating the sum of the gene expression levels of the genes when regressing out the 1st, 2nd and 5th components.</p>
<p>The table below shows the number of genes available for each signature and each dataset. In general it seems that all signatures have over 70% of the genes available, a good indication.</p>
<div class="cell">
<div class="cell-output-display">

<table class=" lightable-classic" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
 <thead>
  <tr>
   <th style="text-align:left;"> pathway </th>
   <th style="text-align:right;"> tcga </th>
   <th style="text-align:right;"> scanb </th>
   <th style="text-align:right;"> metabric </th>
   <th style="text-align:right;"> poetic </th>
   <th style="text-align:right;"> n </th>
   <th style="text-align:left;"> average_percentage </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:left;"> adhesion_and_migration </td>
   <td style="text-align:right;"> 65 </td>
   <td style="text-align:right;"> 61 </td>
   <td style="text-align:right;"> 68 </td>
   <td style="text-align:right;"> 83 </td>
   <td style="text-align:right;"> 83 </td>
   <td style="text-align:left;"> 0.83 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> angiogenesis </td>
   <td style="text-align:right;"> 27 </td>
   <td style="text-align:right;"> 27 </td>
   <td style="text-align:right;"> 29 </td>
   <td style="text-align:right;"> 33 </td>
   <td style="text-align:right;"> 34 </td>
   <td style="text-align:left;"> 0.85 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> antigen_presentation </td>
   <td style="text-align:right;"> 20 </td>
   <td style="text-align:right;"> 19 </td>
   <td style="text-align:right;"> 19 </td>
   <td style="text-align:right;"> 19 </td>
   <td style="text-align:right;"> 21 </td>
   <td style="text-align:left;"> 0.92 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> apoptosis </td>
   <td style="text-align:right;"> 7 </td>
   <td style="text-align:right;"> 7 </td>
   <td style="text-align:right;"> 5 </td>
   <td style="text-align:right;"> 9 </td>
   <td style="text-align:right;"> 9 </td>
   <td style="text-align:left;"> 0.78 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> cytokine_and_chemokine_signaling </td>
   <td style="text-align:right;"> 38 </td>
   <td style="text-align:right;"> 29 </td>
   <td style="text-align:right;"> 37 </td>
   <td style="text-align:right;"> 47 </td>
   <td style="text-align:right;"> 50 </td>
   <td style="text-align:left;"> 0.76 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> dna_damage_repair </td>
   <td style="text-align:right;"> 126 </td>
   <td style="text-align:right;"> 108 </td>
   <td style="text-align:right;"> 118 </td>
   <td style="text-align:right;"> 133 </td>
   <td style="text-align:right;"> 143 </td>
   <td style="text-align:left;"> 0.85 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> emt </td>
   <td style="text-align:right;"> 72 </td>
   <td style="text-align:right;"> 71 </td>
   <td style="text-align:right;"> 63 </td>
   <td style="text-align:right;"> 82 </td>
   <td style="text-align:right;"> 85 </td>
   <td style="text-align:left;"> 0.85 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> er_signaling </td>
   <td style="text-align:right;"> 27 </td>
   <td style="text-align:right;"> 25 </td>
   <td style="text-align:right;"> 24 </td>
   <td style="text-align:right;"> 25 </td>
   <td style="text-align:right;"> 27 </td>
   <td style="text-align:left;"> 0.94 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> epigenetic_regulation </td>
   <td style="text-align:right;"> 16 </td>
   <td style="text-align:right;"> 16 </td>
   <td style="text-align:right;"> 14 </td>
   <td style="text-align:right;"> 17 </td>
   <td style="text-align:right;"> 18 </td>
   <td style="text-align:left;"> 0.88 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> hedgehog </td>
   <td style="text-align:right;"> 14 </td>
   <td style="text-align:right;"> 10 </td>
   <td style="text-align:right;"> 17 </td>
   <td style="text-align:right;"> 20 </td>
   <td style="text-align:right;"> 20 </td>
   <td style="text-align:left;"> 0.76 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> immune_infiltration </td>
   <td style="text-align:right;"> 24 </td>
   <td style="text-align:right;"> 21 </td>
   <td style="text-align:right;"> 23 </td>
   <td style="text-align:right;"> 34 </td>
   <td style="text-align:right;"> 34 </td>
   <td style="text-align:left;"> 0.75 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> jak_stat </td>
   <td style="text-align:right;"> 37 </td>
   <td style="text-align:right;"> 36 </td>
   <td style="text-align:right;"> 35 </td>
   <td style="text-align:right;"> 45 </td>
   <td style="text-align:right;"> 47 </td>
   <td style="text-align:left;"> 0.81 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> mapk </td>
   <td style="text-align:right;"> 67 </td>
   <td style="text-align:right;"> 61 </td>
   <td style="text-align:right;"> 68 </td>
   <td style="text-align:right;"> 95 </td>
   <td style="text-align:right;"> 100 </td>
   <td style="text-align:left;"> 0.73 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> notch </td>
   <td style="text-align:right;"> 21 </td>
   <td style="text-align:right;"> 19 </td>
   <td style="text-align:right;"> 16 </td>
   <td style="text-align:right;"> 21 </td>
   <td style="text-align:right;"> 22 </td>
   <td style="text-align:left;"> 0.88 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> pi3k </td>
   <td style="text-align:right;"> 72 </td>
   <td style="text-align:right;"> 67 </td>
   <td style="text-align:right;"> 63 </td>
   <td style="text-align:right;"> 91 </td>
   <td style="text-align:right;"> 96 </td>
   <td style="text-align:left;"> 0.76 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> proliferation </td>
   <td style="text-align:right;"> 121 </td>
   <td style="text-align:right;"> 113 </td>
   <td style="text-align:right;"> 114 </td>
   <td style="text-align:right;"> 136 </td>
   <td style="text-align:right;"> 144 </td>
   <td style="text-align:left;"> 0.84 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> stromal_markers </td>
   <td style="text-align:right;"> 6 </td>
   <td style="text-align:right;"> 6 </td>
   <td style="text-align:right;"> 6 </td>
   <td style="text-align:right;"> 6 </td>
   <td style="text-align:right;"> 6 </td>
   <td style="text-align:left;"> 1 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> subtypes </td>
   <td style="text-align:right;"> 69 </td>
   <td style="text-align:right;"> 63 </td>
   <td style="text-align:right;"> 62 </td>
   <td style="text-align:right;"> 69 </td>
   <td style="text-align:right;"> 70 </td>
   <td style="text-align:left;"> 0.94 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> tgf_beta </td>
   <td style="text-align:right;"> 47 </td>
   <td style="text-align:right;"> 47 </td>
   <td style="text-align:right;"> 42 </td>
   <td style="text-align:right;"> 53 </td>
   <td style="text-align:right;"> 57 </td>
   <td style="text-align:left;"> 0.83 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> transcriptional_misregulation </td>
   <td style="text-align:right;"> 44 </td>
   <td style="text-align:right;"> 37 </td>
   <td style="text-align:right;"> 44 </td>
   <td style="text-align:right;"> 61 </td>
   <td style="text-align:right;"> 63 </td>
   <td style="text-align:left;"> 0.74 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> triple_negative_biology </td>
   <td style="text-align:right;"> 38 </td>
   <td style="text-align:right;"> 32 </td>
   <td style="text-align:right;"> 41 </td>
   <td style="text-align:right;"> 49 </td>
   <td style="text-align:right;"> 50 </td>
   <td style="text-align:left;"> 0.8 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> tumor_metabolism </td>
   <td style="text-align:right;"> 15 </td>
   <td style="text-align:right;"> 15 </td>
   <td style="text-align:right;"> 14 </td>
   <td style="text-align:right;"> 14 </td>
   <td style="text-align:right;"> 15 </td>
   <td style="text-align:left;"> 0.97 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> wnt </td>
   <td style="text-align:right;"> 37 </td>
   <td style="text-align:right;"> 34 </td>
   <td style="text-align:right;"> 44 </td>
   <td style="text-align:right;"> 51 </td>
   <td style="text-align:right;"> 51 </td>
   <td style="text-align:left;"> 0.81 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> internal_reference_gene </td>
   <td style="text-align:right;"> 18 </td>
   <td style="text-align:right;"> 18 </td>
   <td style="text-align:right;"> 16 </td>
   <td style="text-align:right;"> 17 </td>
   <td style="text-align:right;"> 18 </td>
   <td style="text-align:left;"> 0.96 </td>
  </tr>
</tbody>
</table>

</div>
</div>
<div class="cell">

</div>
<p>We now compare the scores obtained by GSVA on the datasets and see how they relate with nanostring scores obtained with the regressed data.</p>
<p>First we compare GSVA SET ER/PR with ER signaling obtained by the scoring procedure when regressing the data.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-106-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Next we compare EMT, G2M with proliferation and PI3K signaling.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-107-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-108-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-109-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-110-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>In general there is a correlation but the scores are not so well correlated and they are noisy. One that works fairly well is the ER signaling score, but the others not so well. The proliferation score barely shows any difference between luminal A and B patients in any of the cohorts.</p>
<p>Perhaps if we calculate the scores using the hallmarks pathways the scores are better correlated, since they are the same gene sets.</p>
<p>Below we plot the correlation between the proliferation pathway using GSVA and the regressed scoring strategy for the TCGA cohort.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-111-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>It is perfectly correlated.</p>
<p>And for ER signaling:</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-112-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Also it is very much correlated. So the problem in the end is comparing the hallmarks signatures with the nanostring. But interestingly the nanostring proliferation signature was not good to pick up the differences between luminal A and B.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-113-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="risk-stratification-and-the-molecular-landscape" class="level2" data-number="6.12">
<h2 data-number="6.12" class="anchored" data-anchor-id="risk-stratification-and-the-molecular-landscape"><span class="header-section-number">6.12</span> Risk stratification and the molecular landscape</h2>
<p>One common problem in the clinics is what to do with patients that are in the so called intermediate risk group. Here we use the SSP classification from the SCANB team to check what is the position of these patients in the molecular landscape.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-scanb-risk-embedding" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="trying_files/figure-html/fig-scanb-risk-embedding-1.png" class="img-fluid figure-img" width="672"></p>
<p></p><figcaption class="figure-caption">Figure 6.25: Embedding of the SCANB samples colored by their predicted ROR risk category.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p><a href="#fig-scanb-risk-embedding">Figure&nbsp;<span>6.25</span></a> shows that the embedding detects the positions where the intermediate risk patients are, namely they are in the intersection of the luminal A and B subtypes.</p>
<p>And <a href="#fig-intermediate-pca-position">Figure&nbsp;<span>6.26</span></a> below shows that there is a shift in PC4 for the intermediate to the high risk group, which makes sense as this is exactly where the distinction between luminal A and B are.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-intermediate-pca-position" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="trying_files/figure-html/fig-intermediate-pca-position-1.png" class="img-fluid figure-img" width="768"></p>
<p></p><figcaption class="figure-caption">Figure 6.26: Distribution of the principal components for the different risk groups defined by the SSP ROR scores.</figcaption><p></p>
</figure>
</div>
</div>
</div>
</section>
<section id="late-distant-recurrence-trying-to-find-biomarkers" class="level2" data-number="6.13">
<h2 data-number="6.13" class="anchored" data-anchor-id="late-distant-recurrence-trying-to-find-biomarkers"><span class="header-section-number">6.13</span> Late distant recurrence: trying to find biomarkers</h2>
<p>One of the key problems in breast cancer research is finding patients that will develop late recurrence. Here we use METABRIC as it has a long follow-up history. We start by plotting the embedding of all METABRIC patients that had a followup longer than 10 years and received only ET. We then try to predict which patients recurr or not based on the molecular data by using lasso regression. For this we use the package <code>glmnet</code> that provides such functionality.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-116-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>In general there is no distinction in terms of the molecular landscape among the patients that recurred, both in terms of molecular subtype and position in the molecular landscape.</p>
<p>We now compare their molecular scores.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-117-1.png" class="img-fluid" width="1920"></p>
</div>
</div>
<p>Even here we don’t see much difference. Unless there is some kind of linear combination of pathways, I don’t think it would be possible to classify these two groups by using only transcriptomic data and these pathways. In any case we proceed now with the lasso regression using the 50 pathways.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="trying_files/figure-html/unnamed-chunk-118-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>We see that actually it does not work. And when we predict all the samples are predicted as there is no recurrence.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>.
 no 
407 </code></pre>
</div>
</div>
<p>This is probably because of the unbalanced dataset. Next we try with random forest as well.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
 randomForest(formula = as.factor(is_there_recurrence) ~ ., data = et_metabric_late_data[,      c("is_there_recurrence", pathways_to_compare, "AGE_AT_DIAGNOSIS",          "LYMPH_NODES_EXAMINED_POSITIVE", "npi")], ntree = 500) 
               Type of random forest: classification
                     Number of trees: 500
No. of variables tried at each split: 7

        OOB estimate of  error rate: 17.2%
Confusion matrix:
     no yes class.error
no  337   2 0.005899705
yes  68   0 1.000000000</code></pre>
</div>
</div>
<p>That is not the case, still we can’t predict if a patient will develop recurrence or not in this cohort.</p>


<!-- -->

</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      let href = ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./risk_score.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Risk score model</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./discussion.html" class="pagination-link">
        <span class="nav-page-text">Discussion</span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb10" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu"># Tests and tries </span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>We have used some numbers of genes for doing the embedding. Here we try to </span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>perform the embedding with all the genes (~10000 genes) and then</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>we try to regress out the components. </span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a><span class="in">```{r setup}</span></span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>renv<span class="sc">::</span><span class="fu">restore</span>()</span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(PCAtools)</span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(singscore)</span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SummarizedExperiment)</span>
<span id="cb10-17"><a href="#cb10-17" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(survival)</span>
<span id="cb10-18"><a href="#cb10-18" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(genefu)</span>
<span id="cb10-19"><a href="#cb10-19" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(caret)</span>
<span id="cb10-20"><a href="#cb10-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-21"><a href="#cb10-21" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"utils.R"</span>)</span>
<span id="cb10-22"><a href="#cb10-22" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"first_run.R"</span>)</span>
<span id="cb10-23"><a href="#cb10-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-24"><a href="#cb10-24" aria-hidden="true" tabindex="-1"></a><span class="co"># the following script load all data necessary to run the chunks.</span></span>
<span id="cb10-25"><a href="#cb10-25" aria-hidden="true" tabindex="-1"></a><span class="co"># the data is generated from this quarto document itself, therefore</span></span>
<span id="cb10-26"><a href="#cb10-26" aria-hidden="true" tabindex="-1"></a><span class="co"># if you are running this documents the first time and don't have the</span></span>
<span id="cb10-27"><a href="#cb10-27" aria-hidden="true" tabindex="-1"></a><span class="co"># files, comment the following lines. Moreover, if this is your first</span></span>
<span id="cb10-28"><a href="#cb10-28" aria-hidden="true" tabindex="-1"></a><span class="co"># time running the document, you should run all chunks, to generate </span></span>
<span id="cb10-29"><a href="#cb10-29" aria-hidden="true" tabindex="-1"></a><span class="co"># all the necessary files, if you don't have them. Once all files </span></span>
<span id="cb10-30"><a href="#cb10-30" aria-hidden="true" tabindex="-1"></a><span class="co"># are saved and available in the respective folder, the following</span></span>
<span id="cb10-31"><a href="#cb10-31" aria-hidden="true" tabindex="-1"></a><span class="co"># lines can be executed. </span></span>
<span id="cb10-32"><a href="#cb10-32" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (first_run){</span>
<span id="cb10-33"><a href="#cb10-33" aria-hidden="true" tabindex="-1"></a>    load_at_setup <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb10-34"><a href="#cb10-34" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb10-35"><a href="#cb10-35" aria-hidden="true" tabindex="-1"></a>    load_at_setup <span class="ot">&lt;-</span> <span class="cn">TRUE</span></span>
<span id="cb10-36"><a href="#cb10-36" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb10-37"><a href="#cb10-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-38"><a href="#cb10-38" aria-hidden="true" tabindex="-1"></a>name_document <span class="ot">&lt;-</span> <span class="st">"trying"</span></span>
<span id="cb10-39"><a href="#cb10-39" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"load_rds_files.R"</span>)</span>
<span id="cb10-40"><a href="#cb10-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-41"><a href="#cb10-41" aria-hidden="true" tabindex="-1"></a><span class="co"># by setting the dev to png and pdf, this saves the figures in a specific</span></span>
<span id="cb10-42"><a href="#cb10-42" aria-hidden="true" tabindex="-1"></a><span class="co"># folder in both formats. moreover, since png is coming first, it shows</span></span>
<span id="cb10-43"><a href="#cb10-43" aria-hidden="true" tabindex="-1"></a><span class="co"># this figure when rendering the html. What is nice about this is that it</span></span>
<span id="cb10-44"><a href="#cb10-44" aria-hidden="true" tabindex="-1"></a><span class="co"># inherits the properties from the chunk to save the figure, so no need</span></span>
<span id="cb10-45"><a href="#cb10-45" aria-hidden="true" tabindex="-1"></a><span class="co"># to use ggplot2::ggsave to save the plots. this also works </span></span>
<span id="cb10-46"><a href="#cb10-46" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">dev =</span> <span class="fu">c</span>(<span class="st">'png'</span>, <span class="st">'pdf'</span>, <span class="st">'svg'</span>))</span>
<span id="cb10-47"><a href="#cb10-47" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">bitmapType =</span> <span class="st">'cairo'</span>)</span>
<span id="cb10-48"><a href="#cb10-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-49"><a href="#cb10-49" aria-hidden="true" tabindex="-1"></a>p4_pathways <span class="ot">&lt;-</span> msigdbr<span class="sc">::</span><span class="fu">msigdbr</span>() <span class="sc">%&gt;%</span></span>
<span id="cb10-50"><a href="#cb10-50" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(gs_name <span class="sc">%in%</span> <span class="fu">c</span>(</span>
<span id="cb10-51"><a href="#cb10-51" aria-hidden="true" tabindex="-1"></a>        <span class="st">"WILCOX_RESPONSE_TO_PROGESTERONE_UP"</span>,</span>
<span id="cb10-52"><a href="#cb10-52" aria-hidden="true" tabindex="-1"></a>        <span class="st">"GOBP_RESPONSE_TO_PROGESTERONE"</span></span>
<span id="cb10-53"><a href="#cb10-53" aria-hidden="true" tabindex="-1"></a>    ))</span>
<span id="cb10-54"><a href="#cb10-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-55"><a href="#cb10-55" aria-hidden="true" tabindex="-1"></a>gene_sets_prog <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(</span>
<span id="cb10-56"><a href="#cb10-56" aria-hidden="true" tabindex="-1"></a>    gene_sets, </span>
<span id="cb10-57"><a href="#cb10-57" aria-hidden="true" tabindex="-1"></a>    p4_pathways</span>
<span id="cb10-58"><a href="#cb10-58" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-59"><a href="#cb10-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-60"><a href="#cb10-60" aria-hidden="true" tabindex="-1"></a>gene_sets_ <span class="ot">&lt;-</span> <span class="fu">sapply</span>(</span>
<span id="cb10-61"><a href="#cb10-61" aria-hidden="true" tabindex="-1"></a>    gene_sets_prog<span class="sc">$</span>gs_name <span class="sc">%&gt;%</span> unique,</span>
<span id="cb10-62"><a href="#cb10-62" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(x, gene_sets) gene_sets <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(gs_name <span class="sc">==</span> x) <span class="sc">%&gt;%</span> </span>
<span id="cb10-63"><a href="#cb10-63" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">pull</span>(gene_symbol),</span>
<span id="cb10-64"><a href="#cb10-64" aria-hidden="true" tabindex="-1"></a>    <span class="at">gene_sets =</span> gene_sets_prog,</span>
<span id="cb10-65"><a href="#cb10-65" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>,</span>
<span id="cb10-66"><a href="#cb10-66" aria-hidden="true" tabindex="-1"></a>    <span class="at">simplify =</span> <span class="cn">FALSE</span></span>
<span id="cb10-67"><a href="#cb10-67" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-68"><a href="#cb10-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-69"><a href="#cb10-69" aria-hidden="true" tabindex="-1"></a>datasets<span class="sc">$</span>poetic<span class="sc">$</span>sample_name <span class="ot">&lt;-</span> <span class="fu">colnames</span>(datasets<span class="sc">$</span>poetic)</span>
<span id="cb10-70"><a href="#cb10-70" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-71"><a href="#cb10-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-72"><a href="#cb10-72" aria-hidden="true" tabindex="-1"></a><span class="fu">## Embedding with more than 1000 genes</span></span>
<span id="cb10-73"><a href="#cb10-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-74"><a href="#cb10-74" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb10-75"><a href="#cb10-75" aria-hidden="true" tabindex="-1"></a><span class="co"># we perform now the normalization for all genes</span></span>
<span id="cb10-76"><a href="#cb10-76" aria-hidden="true" tabindex="-1"></a>common_genes <span class="ot">&lt;-</span> <span class="fu">Reduce</span>(intersect, <span class="fu">lapply</span>(datasets, rownames))</span>
<span id="cb10-77"><a href="#cb10-77" aria-hidden="true" tabindex="-1"></a>stable_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(stable_genes, common_genes)</span>
<span id="cb10-78"><a href="#cb10-78" aria-hidden="true" tabindex="-1"></a>datasets_normalized <span class="ot">&lt;-</span> <span class="fu">mapply</span>(</span>
<span id="cb10-79"><a href="#cb10-79" aria-hidden="true" tabindex="-1"></a>    get_final_ranking_values,</span>
<span id="cb10-80"><a href="#cb10-80" aria-hidden="true" tabindex="-1"></a>    <span class="at">sum_exp =</span> datasets,</span>
<span id="cb10-81"><a href="#cb10-81" aria-hidden="true" tabindex="-1"></a>    <span class="at">assay_to_use =</span> <span class="fu">c</span>(which_exp, <span class="fu">list</span>(<span class="at">poetic =</span> <span class="st">"normalized_intensity"</span>)),</span>
<span id="cb10-82"><a href="#cb10-82" aria-hidden="true" tabindex="-1"></a>    <span class="at">MoreArgs =</span> <span class="fu">list</span>(</span>
<span id="cb10-83"><a href="#cb10-83" aria-hidden="true" tabindex="-1"></a>        <span class="at">stable_genes =</span> stable_genes,</span>
<span id="cb10-84"><a href="#cb10-84" aria-hidden="true" tabindex="-1"></a>        <span class="at">most_variable_genes =</span> common_genes</span>
<span id="cb10-85"><a href="#cb10-85" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb10-86"><a href="#cb10-86" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>,</span>
<span id="cb10-87"><a href="#cb10-87" aria-hidden="true" tabindex="-1"></a>    <span class="at">SIMPLIFY =</span> <span class="cn">FALSE</span></span>
<span id="cb10-88"><a href="#cb10-88" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-89"><a href="#cb10-89" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-90"><a href="#cb10-90" aria-hidden="true" tabindex="-1"></a>merged_col_data <span class="ot">&lt;-</span> <span class="fu">lapply</span>(datasets_normalized, colData) <span class="sc">%&gt;%</span> </span>
<span id="cb10-91"><a href="#cb10-91" aria-hidden="true" tabindex="-1"></a>    <span class="fu">lapply</span>(., data.frame) <span class="sc">%&gt;%</span></span>
<span id="cb10-92"><a href="#cb10-92" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(<span class="at">.id =</span> <span class="st">"cohort"</span>)</span>
<span id="cb10-93"><a href="#cb10-93" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-94"><a href="#cb10-94" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1329</span>)</span>
<span id="cb10-95"><a href="#cb10-95" aria-hidden="true" tabindex="-1"></a>samples_for_training <span class="ot">&lt;-</span> merged_col_data <span class="sc">%&gt;%</span></span>
<span id="cb10-96"><a href="#cb10-96" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">%in%</span> which_cohorts_training) <span class="sc">%&gt;%</span></span>
<span id="cb10-97"><a href="#cb10-97" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">pull</span>(sample_name) <span class="sc">%&gt;%</span></span>
<span id="cb10-98"><a href="#cb10-98" aria-hidden="true" tabindex="-1"></a>    <span class="fu">sample</span>(., <span class="at">size =</span> <span class="dv">1000</span>) </span>
<span id="cb10-99"><a href="#cb10-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-100"><a href="#cb10-100" aria-hidden="true" tabindex="-1"></a><span class="co"># and now we can perform the molecular embedding</span></span>
<span id="cb10-101"><a href="#cb10-101" aria-hidden="true" tabindex="-1"></a>training_set <span class="ot">&lt;-</span> <span class="fu">lapply</span>(</span>
<span id="cb10-102"><a href="#cb10-102" aria-hidden="true" tabindex="-1"></a>    datasets_normalized[which_cohorts_training], </span>
<span id="cb10-103"><a href="#cb10-103" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(sum_exp, i, genes_for_pca) </span>
<span id="cb10-104"><a href="#cb10-104" aria-hidden="true" tabindex="-1"></a>        <span class="fu">assay</span>(sum_exp[genes_for_pca, ], <span class="at">i =</span> i) <span class="sc">%&gt;%</span> </span>
<span id="cb10-105"><a href="#cb10-105" aria-hidden="true" tabindex="-1"></a>        <span class="fu">data.frame</span>(<span class="at">check.names =</span> <span class="cn">FALSE</span>), </span>
<span id="cb10-106"><a href="#cb10-106" aria-hidden="true" tabindex="-1"></a>    <span class="at">i =</span> <span class="st">"avg_ranking"</span>,</span>
<span id="cb10-107"><a href="#cb10-107" aria-hidden="true" tabindex="-1"></a>    <span class="at">genes_for_pca =</span> common_genes</span>
<span id="cb10-108"><a href="#cb10-108" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-109"><a href="#cb10-109" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">bind_cols</span>() <span class="sc">%&gt;%</span></span>
<span id="cb10-110"><a href="#cb10-110" aria-hidden="true" tabindex="-1"></a>    .[, samples_for_training]</span>
<span id="cb10-111"><a href="#cb10-111" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-112"><a href="#cb10-112" aria-hidden="true" tabindex="-1"></a>pca_fit_all_genes <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">pca</span>(</span>
<span id="cb10-113"><a href="#cb10-113" aria-hidden="true" tabindex="-1"></a>    training_set,</span>
<span id="cb10-114"><a href="#cb10-114" aria-hidden="true" tabindex="-1"></a>    <span class="at">metadata =</span> dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(</span>
<span id="cb10-115"><a href="#cb10-115" aria-hidden="true" tabindex="-1"></a>        <span class="fu">lapply</span>(</span>
<span id="cb10-116"><a href="#cb10-116" aria-hidden="true" tabindex="-1"></a>            datasets_normalized[which_cohorts_training],</span>
<span id="cb10-117"><a href="#cb10-117" aria-hidden="true" tabindex="-1"></a>            <span class="cf">function</span>(df){</span>
<span id="cb10-118"><a href="#cb10-118" aria-hidden="true" tabindex="-1"></a>                <span class="fu">colData</span>(df) <span class="sc">%&gt;%</span> data.frame <span class="sc">%&gt;%</span></span>
<span id="cb10-119"><a href="#cb10-119" aria-hidden="true" tabindex="-1"></a>                    dplyr<span class="sc">::</span><span class="fu">filter</span>(sample_name <span class="sc">%in%</span> <span class="fu">colnames</span>(training_set))</span>
<span id="cb10-120"><a href="#cb10-120" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb10-121"><a href="#cb10-121" aria-hidden="true" tabindex="-1"></a>        ),</span>
<span id="cb10-122"><a href="#cb10-122" aria-hidden="true" tabindex="-1"></a>        <span class="at">.id =</span> <span class="st">"cohort"</span></span>
<span id="cb10-123"><a href="#cb10-123" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span> .[<span class="fu">colnames</span>(training_set), ],</span>
<span id="cb10-124"><a href="#cb10-124" aria-hidden="true" tabindex="-1"></a>    <span class="at">center =</span> <span class="cn">FALSE</span>,</span>
<span id="cb10-125"><a href="#cb10-125" aria-hidden="true" tabindex="-1"></a>    <span class="at">scale =</span> <span class="cn">FALSE</span></span>
<span id="cb10-126"><a href="#cb10-126" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-127"><a href="#cb10-127" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-128"><a href="#cb10-128" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb10-129"><a href="#cb10-129" aria-hidden="true" tabindex="-1"></a>    pca_fit_all_genes,</span>
<span id="cb10-130"><a href="#cb10-130" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/trying/pca_fit_all_genes.rds"</span></span>
<span id="cb10-131"><a href="#cb10-131" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-132"><a href="#cb10-132" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-133"><a href="#cb10-133" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb10-134"><a href="#cb10-134" aria-hidden="true" tabindex="-1"></a>    datasets_normalized,</span>
<span id="cb10-135"><a href="#cb10-135" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/trying/datasets_normalized_all_genes.rds"</span></span>
<span id="cb10-136"><a href="#cb10-136" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-137"><a href="#cb10-137" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-138"><a href="#cb10-138" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb10-139"><a href="#cb10-139" aria-hidden="true" tabindex="-1"></a>    merged_col_data,</span>
<span id="cb10-140"><a href="#cb10-140" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/trying/merged_col_data_trying.rds"</span></span>
<span id="cb10-141"><a href="#cb10-141" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-142"><a href="#cb10-142" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-143"><a href="#cb10-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-144"><a href="#cb10-144" aria-hidden="true" tabindex="-1"></a>And now we can visualize the results.</span>
<span id="cb10-145"><a href="#cb10-145" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-146"><a href="#cb10-146" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=14, fig.height=16}</span></span>
<span id="cb10-147"><a href="#cb10-147" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-pca-embeddings-1</span></span>
<span id="cb10-148"><a href="#cb10-148" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: PCA projections colored by different factors and organized</span></span>
<span id="cb10-149"><a href="#cb10-149" aria-hidden="true" tabindex="-1"></a><span class="co">#|    by different components. (A) Plot of the first two components colored</span></span>
<span id="cb10-150"><a href="#cb10-150" aria-hidden="true" tabindex="-1"></a><span class="co">#|    by cohort. (B) Plot of the first two components colored by ER status.</span></span>
<span id="cb10-151"><a href="#cb10-151" aria-hidden="true" tabindex="-1"></a><span class="co">#|    (C) Plot of the second and third components colored by cohort. </span></span>
<span id="cb10-152"><a href="#cb10-152" aria-hidden="true" tabindex="-1"></a><span class="co">#|    (D) Plot of the second and third components colored by ER status.</span></span>
<span id="cb10-153"><a href="#cb10-153" aria-hidden="true" tabindex="-1"></a><span class="co">#|    (E) Plot of the second and third components colored by PAM50.</span></span>
<span id="cb10-154"><a href="#cb10-154" aria-hidden="true" tabindex="-1"></a><span class="co">#|    (F) Plot of the second and third components colored by INTCLUST, only</span></span>
<span id="cb10-155"><a href="#cb10-155" aria-hidden="true" tabindex="-1"></a><span class="co">#|    METABRIC has an assigned value for this variable, NAs are TCGA samples.</span></span>
<span id="cb10-156"><a href="#cb10-156" aria-hidden="true" tabindex="-1"></a>plots_pca_fit <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb10-157"><a href="#cb10-157" aria-hidden="true" tabindex="-1"></a>point_size <span class="ot">&lt;-</span> <span class="dv">2</span></span>
<span id="cb10-158"><a href="#cb10-158" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-159"><a href="#cb10-159" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>pc1_pc2_cohort <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb10-160"><a href="#cb10-160" aria-hidden="true" tabindex="-1"></a>    pca_fit_all_genes,</span>
<span id="cb10-161"><a href="#cb10-161" aria-hidden="true" tabindex="-1"></a>    <span class="at">colby =</span> <span class="st">"cohort"</span>,</span>
<span id="cb10-162"><a href="#cb10-162" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> <span class="cn">NULL</span>, </span>
<span id="cb10-163"><a href="#cb10-163" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb10-164"><a href="#cb10-164" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC1"</span>,</span>
<span id="cb10-165"><a href="#cb10-165" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC2"</span>,</span>
<span id="cb10-166"><a href="#cb10-166" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"First two components colored by cohort"</span>,</span>
<span id="cb10-167"><a href="#cb10-167" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">"Only 1000 training samples"</span>,</span>
<span id="cb10-168"><a href="#cb10-168" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> point_size</span>
<span id="cb10-169"><a href="#cb10-169" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-170"><a href="#cb10-170" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-171"><a href="#cb10-171" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>pc1_pc2_er_status <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb10-172"><a href="#cb10-172" aria-hidden="true" tabindex="-1"></a>    pca_fit_all_genes,</span>
<span id="cb10-173"><a href="#cb10-173" aria-hidden="true" tabindex="-1"></a>    <span class="at">colby =</span> <span class="st">"er_status"</span>,</span>
<span id="cb10-174"><a href="#cb10-174" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> <span class="cn">NULL</span>, </span>
<span id="cb10-175"><a href="#cb10-175" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb10-176"><a href="#cb10-176" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC1"</span>,</span>
<span id="cb10-177"><a href="#cb10-177" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC2"</span>,</span>
<span id="cb10-178"><a href="#cb10-178" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"First two components colored by ER status"</span>,</span>
<span id="cb10-179"><a href="#cb10-179" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">"Only 1000 training samples"</span>,</span>
<span id="cb10-180"><a href="#cb10-180" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> point_size</span>
<span id="cb10-181"><a href="#cb10-181" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-182"><a href="#cb10-182" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-183"><a href="#cb10-183" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>pc2_pc3_cohort <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb10-184"><a href="#cb10-184" aria-hidden="true" tabindex="-1"></a>    pca_fit_all_genes,</span>
<span id="cb10-185"><a href="#cb10-185" aria-hidden="true" tabindex="-1"></a>    <span class="at">colby =</span> <span class="st">"cohort"</span>,</span>
<span id="cb10-186"><a href="#cb10-186" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> <span class="cn">NULL</span>, </span>
<span id="cb10-187"><a href="#cb10-187" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb10-188"><a href="#cb10-188" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC2"</span>,</span>
<span id="cb10-189"><a href="#cb10-189" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC3"</span>,</span>
<span id="cb10-190"><a href="#cb10-190" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Second and third components colored by cohort"</span>,</span>
<span id="cb10-191"><a href="#cb10-191" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">"Only 1000 training samples"</span>,</span>
<span id="cb10-192"><a href="#cb10-192" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> point_size</span>
<span id="cb10-193"><a href="#cb10-193" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-194"><a href="#cb10-194" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-195"><a href="#cb10-195" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>pc2_pc3_er_status <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb10-196"><a href="#cb10-196" aria-hidden="true" tabindex="-1"></a>    pca_fit_all_genes,</span>
<span id="cb10-197"><a href="#cb10-197" aria-hidden="true" tabindex="-1"></a>    <span class="at">colby =</span> <span class="st">"er_status"</span>,</span>
<span id="cb10-198"><a href="#cb10-198" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> <span class="cn">NULL</span>, </span>
<span id="cb10-199"><a href="#cb10-199" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb10-200"><a href="#cb10-200" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC2"</span>,</span>
<span id="cb10-201"><a href="#cb10-201" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC3"</span>,</span>
<span id="cb10-202"><a href="#cb10-202" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Second and third components colored by ER status"</span>,</span>
<span id="cb10-203"><a href="#cb10-203" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">"Only 1000 training samples"</span>,</span>
<span id="cb10-204"><a href="#cb10-204" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> point_size</span>
<span id="cb10-205"><a href="#cb10-205" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-206"><a href="#cb10-206" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-207"><a href="#cb10-207" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>pc2_pc3_pam50 <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb10-208"><a href="#cb10-208" aria-hidden="true" tabindex="-1"></a>    pca_fit_all_genes,</span>
<span id="cb10-209"><a href="#cb10-209" aria-hidden="true" tabindex="-1"></a>    <span class="at">colby =</span> <span class="st">"pam50"</span>,</span>
<span id="cb10-210"><a href="#cb10-210" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> <span class="cn">NULL</span>, </span>
<span id="cb10-211"><a href="#cb10-211" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb10-212"><a href="#cb10-212" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC2"</span>,</span>
<span id="cb10-213"><a href="#cb10-213" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC3"</span>,</span>
<span id="cb10-214"><a href="#cb10-214" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Second and third components colored by PAM50"</span>,</span>
<span id="cb10-215"><a href="#cb10-215" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">"Only 1000 training samples"</span>,</span>
<span id="cb10-216"><a href="#cb10-216" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> point_size</span>
<span id="cb10-217"><a href="#cb10-217" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-218"><a href="#cb10-218" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>pc2_pc3_pam50 <span class="ot">&lt;-</span> plots_pca_fit<span class="sc">$</span>pc2_pc3_pam50 <span class="sc">+</span></span>
<span id="cb10-219"><a href="#cb10-219" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_color_manual</span>(</span>
<span id="cb10-220"><a href="#cb10-220" aria-hidden="true" tabindex="-1"></a>        <span class="at">values =</span> <span class="fu">get_colors_pam50</span>(pca_fit_all_genes<span class="sc">$</span>metadata)</span>
<span id="cb10-221"><a href="#cb10-221" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb10-222"><a href="#cb10-222" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb10-223"><a href="#cb10-223" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-224"><a href="#cb10-224" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>pc2_pc3_intclust <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb10-225"><a href="#cb10-225" aria-hidden="true" tabindex="-1"></a>    pca_fit_all_genes,</span>
<span id="cb10-226"><a href="#cb10-226" aria-hidden="true" tabindex="-1"></a>    <span class="at">colby =</span> <span class="st">"INTCLUST"</span>,</span>
<span id="cb10-227"><a href="#cb10-227" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> <span class="cn">NULL</span>, </span>
<span id="cb10-228"><a href="#cb10-228" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb10-229"><a href="#cb10-229" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC2"</span>,</span>
<span id="cb10-230"><a href="#cb10-230" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC3"</span>,</span>
<span id="cb10-231"><a href="#cb10-231" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Second and third components colored by INTCLUST"</span>,</span>
<span id="cb10-232"><a href="#cb10-232" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">"Only 1000 training samples"</span>,</span>
<span id="cb10-233"><a href="#cb10-233" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> point_size</span>
<span id="cb10-234"><a href="#cb10-234" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-235"><a href="#cb10-235" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-236"><a href="#cb10-236" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>pc3_pc4_pam50 <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb10-237"><a href="#cb10-237" aria-hidden="true" tabindex="-1"></a>    pca_fit_all_genes,</span>
<span id="cb10-238"><a href="#cb10-238" aria-hidden="true" tabindex="-1"></a>    <span class="at">colby =</span> <span class="st">"pam50"</span>,</span>
<span id="cb10-239"><a href="#cb10-239" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> <span class="cn">NULL</span>, </span>
<span id="cb10-240"><a href="#cb10-240" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb10-241"><a href="#cb10-241" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC3"</span>,</span>
<span id="cb10-242"><a href="#cb10-242" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC4"</span>,</span>
<span id="cb10-243"><a href="#cb10-243" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Third and fourth components colored by PAM50"</span>,</span>
<span id="cb10-244"><a href="#cb10-244" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">"Only 1000 training samples"</span>,</span>
<span id="cb10-245"><a href="#cb10-245" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> point_size</span>
<span id="cb10-246"><a href="#cb10-246" aria-hidden="true" tabindex="-1"></a>) </span>
<span id="cb10-247"><a href="#cb10-247" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>pc3_pc4_pam50 <span class="ot">&lt;-</span> plots_pca_fit<span class="sc">$</span>pc3_pc4_pam50 <span class="sc">+</span></span>
<span id="cb10-248"><a href="#cb10-248" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_color_manual</span>(</span>
<span id="cb10-249"><a href="#cb10-249" aria-hidden="true" tabindex="-1"></a>        <span class="at">values =</span> <span class="fu">get_colors_pam50</span>(pca_fit_all_genes<span class="sc">$</span>metadata)</span>
<span id="cb10-250"><a href="#cb10-250" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb10-251"><a href="#cb10-251" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-252"><a href="#cb10-252" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>pc3_pc4_cohort <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb10-253"><a href="#cb10-253" aria-hidden="true" tabindex="-1"></a>    pca_fit_all_genes,</span>
<span id="cb10-254"><a href="#cb10-254" aria-hidden="true" tabindex="-1"></a>    <span class="at">colby =</span> <span class="st">"cohort"</span>,</span>
<span id="cb10-255"><a href="#cb10-255" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> <span class="cn">NULL</span>, </span>
<span id="cb10-256"><a href="#cb10-256" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb10-257"><a href="#cb10-257" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC3"</span>,</span>
<span id="cb10-258"><a href="#cb10-258" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC4"</span>,</span>
<span id="cb10-259"><a href="#cb10-259" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Third and fourth components colored by cohort"</span>,</span>
<span id="cb10-260"><a href="#cb10-260" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">"Only 1000 training samples"</span>,</span>
<span id="cb10-261"><a href="#cb10-261" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> point_size</span>
<span id="cb10-262"><a href="#cb10-262" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-263"><a href="#cb10-263" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-264"><a href="#cb10-264" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(<span class="at">plotlist =</span> plots_pca_fit, <span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">labels =</span> <span class="st">"AUTO"</span>)</span>
<span id="cb10-265"><a href="#cb10-265" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-266"><a href="#cb10-266" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-267"><a href="#cb10-267" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb10-268"><a href="#cb10-268" aria-hidden="true" tabindex="-1"></a>datasets_pca_coordinates <span class="ot">&lt;-</span> <span class="fu">lapply</span>(</span>
<span id="cb10-269"><a href="#cb10-269" aria-hidden="true" tabindex="-1"></a>    datasets_normalized,</span>
<span id="cb10-270"><a href="#cb10-270" aria-hidden="true" tabindex="-1"></a>    get_pca_coordinates,</span>
<span id="cb10-271"><a href="#cb10-271" aria-hidden="true" tabindex="-1"></a>    <span class="at">pca_fit =</span> pca_fit_all_genes,</span>
<span id="cb10-272"><a href="#cb10-272" aria-hidden="true" tabindex="-1"></a>    <span class="at">genes_for_pca =</span> <span class="fu">rownames</span>(pca_fit_all_genes<span class="sc">$</span>loadings)</span>
<span id="cb10-273"><a href="#cb10-273" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-274"><a href="#cb10-274" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-275"><a href="#cb10-275" aria-hidden="true" tabindex="-1"></a>df_pca_coordinates <span class="ot">&lt;-</span> datasets_pca_coordinates <span class="sc">%&gt;%</span> </span>
<span id="cb10-276"><a href="#cb10-276" aria-hidden="true" tabindex="-1"></a>    <span class="fu">do.call</span>(rbind, .) <span class="sc">%&gt;%</span></span>
<span id="cb10-277"><a href="#cb10-277" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span> </span>
<span id="cb10-278"><a href="#cb10-278" aria-hidden="true" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"sample_name"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-279"><a href="#cb10-279" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">inner_join</span>(</span>
<span id="cb10-280"><a href="#cb10-280" aria-hidden="true" tabindex="-1"></a>        .,</span>
<span id="cb10-281"><a href="#cb10-281" aria-hidden="true" tabindex="-1"></a>        merged_col_data,</span>
<span id="cb10-282"><a href="#cb10-282" aria-hidden="true" tabindex="-1"></a>        <span class="at">by =</span> <span class="st">"sample_name"</span></span>
<span id="cb10-283"><a href="#cb10-283" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb10-284"><a href="#cb10-284" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-285"><a href="#cb10-285" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb10-286"><a href="#cb10-286" aria-hidden="true" tabindex="-1"></a>    df_pca_coordinates,</span>
<span id="cb10-287"><a href="#cb10-287" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/trying/df_pca_coordinates_all_genes.rds"</span></span>
<span id="cb10-288"><a href="#cb10-288" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-289"><a href="#cb10-289" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-290"><a href="#cb10-290" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-293"><a href="#cb10-293" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-294"><a href="#cb10-294" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_pca_coordinates</span>(</span>
<span id="cb10-295"><a href="#cb10-295" aria-hidden="true" tabindex="-1"></a>    df_pca_coordinates <span class="sc">%&gt;%</span> </span>
<span id="cb10-296"><a href="#cb10-296" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">slice</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">n</span>()),</span>
<span id="cb10-297"><a href="#cb10-297" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"cohort"</span>,</span>
<span id="cb10-298"><a href="#cb10-298" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC3"</span>,</span>
<span id="cb10-299"><a href="#cb10-299" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC4"</span>,</span>
<span id="cb10-300"><a href="#cb10-300" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">1</span>,</span>
<span id="cb10-301"><a href="#cb10-301" aria-hidden="true" tabindex="-1"></a>    <span class="at">base_size =</span> <span class="dv">10</span>,</span>
<span id="cb10-302"><a href="#cb10-302" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="fu">paste0</span>(</span>
<span id="cb10-303"><a href="#cb10-303" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Embedding of all samples from TCGA,</span><span class="sc">\n</span><span class="st">"</span>,</span>
<span id="cb10-304"><a href="#cb10-304" aria-hidden="true" tabindex="-1"></a>        <span class="st">"METABRIC and SCANB"</span></span>
<span id="cb10-305"><a href="#cb10-305" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb10-306"><a href="#cb10-306" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span> ggplot2<span class="sc">::</span><span class="fu">scale_color_viridis_d</span>() <span class="sc">+</span></span>
<span id="cb10-307"><a href="#cb10-307" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">15</span>) <span class="sc">+</span></span>
<span id="cb10-308"><a href="#cb10-308" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span></span>
<span id="cb10-309"><a href="#cb10-309" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb10-310"><a href="#cb10-310" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-311"><a href="#cb10-311" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-312"><a href="#cb10-312" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb10-313"><a href="#cb10-313" aria-hidden="true" tabindex="-1"></a>remove_metabric <span class="ot">&lt;-</span> df_pca_coordinates <span class="sc">%&gt;%</span></span>
<span id="cb10-314"><a href="#cb10-314" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(PC4 <span class="sc">&gt;</span> <span class="dv">10</span> <span class="sc">&amp;</span> cohort <span class="sc">==</span> <span class="st">"metabric"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-315"><a href="#cb10-315" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">pull</span>(sample_name)</span>
<span id="cb10-316"><a href="#cb10-316" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-317"><a href="#cb10-317" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1329</span>)</span>
<span id="cb10-318"><a href="#cb10-318" aria-hidden="true" tabindex="-1"></a>samples_for_training <span class="ot">&lt;-</span> merged_col_data <span class="sc">%&gt;%</span></span>
<span id="cb10-319"><a href="#cb10-319" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">%in%</span> which_cohorts_training) <span class="sc">%&gt;%</span></span>
<span id="cb10-320"><a href="#cb10-320" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">pull</span>(sample_name) <span class="sc">%&gt;%</span></span>
<span id="cb10-321"><a href="#cb10-321" aria-hidden="true" tabindex="-1"></a>    <span class="fu">sample</span>(., <span class="at">size =</span> <span class="dv">1000</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-322"><a href="#cb10-322" aria-hidden="true" tabindex="-1"></a>    <span class="fu">setdiff</span>(., remove_metabric)</span>
<span id="cb10-323"><a href="#cb10-323" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-324"><a href="#cb10-324" aria-hidden="true" tabindex="-1"></a><span class="co"># and now we can perform the molecular embedding</span></span>
<span id="cb10-325"><a href="#cb10-325" aria-hidden="true" tabindex="-1"></a>training_set <span class="ot">&lt;-</span> <span class="fu">lapply</span>(</span>
<span id="cb10-326"><a href="#cb10-326" aria-hidden="true" tabindex="-1"></a>    datasets_normalized[which_cohorts_training], </span>
<span id="cb10-327"><a href="#cb10-327" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(sum_exp, i, genes_for_pca) </span>
<span id="cb10-328"><a href="#cb10-328" aria-hidden="true" tabindex="-1"></a>        <span class="fu">assay</span>(sum_exp[genes_for_pca, ], <span class="at">i =</span> i) <span class="sc">%&gt;%</span> </span>
<span id="cb10-329"><a href="#cb10-329" aria-hidden="true" tabindex="-1"></a>        <span class="fu">data.frame</span>(<span class="at">check.names =</span> <span class="cn">FALSE</span>), </span>
<span id="cb10-330"><a href="#cb10-330" aria-hidden="true" tabindex="-1"></a>    <span class="at">i =</span> <span class="st">"avg_ranking"</span>,</span>
<span id="cb10-331"><a href="#cb10-331" aria-hidden="true" tabindex="-1"></a>    <span class="at">genes_for_pca =</span> common_genes</span>
<span id="cb10-332"><a href="#cb10-332" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-333"><a href="#cb10-333" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">bind_cols</span>() <span class="sc">%&gt;%</span></span>
<span id="cb10-334"><a href="#cb10-334" aria-hidden="true" tabindex="-1"></a>    .[, samples_for_training]</span>
<span id="cb10-335"><a href="#cb10-335" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-336"><a href="#cb10-336" aria-hidden="true" tabindex="-1"></a>pca_fit_all_genes_wo_outliers <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">pca</span>(</span>
<span id="cb10-337"><a href="#cb10-337" aria-hidden="true" tabindex="-1"></a>    training_set,</span>
<span id="cb10-338"><a href="#cb10-338" aria-hidden="true" tabindex="-1"></a>    <span class="at">metadata =</span> dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(</span>
<span id="cb10-339"><a href="#cb10-339" aria-hidden="true" tabindex="-1"></a>        <span class="fu">lapply</span>(</span>
<span id="cb10-340"><a href="#cb10-340" aria-hidden="true" tabindex="-1"></a>            datasets_normalized[which_cohorts_training],</span>
<span id="cb10-341"><a href="#cb10-341" aria-hidden="true" tabindex="-1"></a>            <span class="cf">function</span>(df){</span>
<span id="cb10-342"><a href="#cb10-342" aria-hidden="true" tabindex="-1"></a>                <span class="fu">colData</span>(df) <span class="sc">%&gt;%</span> data.frame <span class="sc">%&gt;%</span></span>
<span id="cb10-343"><a href="#cb10-343" aria-hidden="true" tabindex="-1"></a>                    dplyr<span class="sc">::</span><span class="fu">filter</span>(sample_name <span class="sc">%in%</span> <span class="fu">colnames</span>(training_set))</span>
<span id="cb10-344"><a href="#cb10-344" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb10-345"><a href="#cb10-345" aria-hidden="true" tabindex="-1"></a>        ),</span>
<span id="cb10-346"><a href="#cb10-346" aria-hidden="true" tabindex="-1"></a>        <span class="at">.id =</span> <span class="st">"cohort"</span></span>
<span id="cb10-347"><a href="#cb10-347" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span> .[<span class="fu">colnames</span>(training_set), ],</span>
<span id="cb10-348"><a href="#cb10-348" aria-hidden="true" tabindex="-1"></a>    <span class="at">center =</span> <span class="cn">FALSE</span>,</span>
<span id="cb10-349"><a href="#cb10-349" aria-hidden="true" tabindex="-1"></a>    <span class="at">scale =</span> <span class="cn">FALSE</span></span>
<span id="cb10-350"><a href="#cb10-350" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-351"><a href="#cb10-351" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-352"><a href="#cb10-352" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb10-353"><a href="#cb10-353" aria-hidden="true" tabindex="-1"></a>    pca_fit_all_genes_wo_outliers, </span>
<span id="cb10-354"><a href="#cb10-354" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/trying/pca_fit_all_genes_wo_outliers.rds"</span></span>
<span id="cb10-355"><a href="#cb10-355" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-356"><a href="#cb10-356" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-357"><a href="#cb10-357" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-358"><a href="#cb10-358" aria-hidden="true" tabindex="-1"></a>And now we can visualize the results.</span>
<span id="cb10-359"><a href="#cb10-359" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-360"><a href="#cb10-360" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=14, fig.height=16}</span></span>
<span id="cb10-361"><a href="#cb10-361" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-pca-embeddings-2</span></span>
<span id="cb10-362"><a href="#cb10-362" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: PCA projections colored by different factors and organized</span></span>
<span id="cb10-363"><a href="#cb10-363" aria-hidden="true" tabindex="-1"></a><span class="co">#|    by different components. (A) Plot of the first two components colored</span></span>
<span id="cb10-364"><a href="#cb10-364" aria-hidden="true" tabindex="-1"></a><span class="co">#|    by cohort. (B) Plot of the first two components colored by ER status.</span></span>
<span id="cb10-365"><a href="#cb10-365" aria-hidden="true" tabindex="-1"></a><span class="co">#|    (C) Plot of the second and third components colored by cohort. </span></span>
<span id="cb10-366"><a href="#cb10-366" aria-hidden="true" tabindex="-1"></a><span class="co">#|    (D) Plot of the second and third components colored by ER status.</span></span>
<span id="cb10-367"><a href="#cb10-367" aria-hidden="true" tabindex="-1"></a><span class="co">#|    (E) Plot of the second and third components colored by PAM50.</span></span>
<span id="cb10-368"><a href="#cb10-368" aria-hidden="true" tabindex="-1"></a><span class="co">#|    (F) Plot of the second and third components colored by INTCLUST, only</span></span>
<span id="cb10-369"><a href="#cb10-369" aria-hidden="true" tabindex="-1"></a><span class="co">#|    METABRIC has an assigned value for this variable, NAs are TCGA samples.</span></span>
<span id="cb10-370"><a href="#cb10-370" aria-hidden="true" tabindex="-1"></a>plots_pca_fit <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb10-371"><a href="#cb10-371" aria-hidden="true" tabindex="-1"></a>point_size <span class="ot">&lt;-</span> <span class="dv">2</span></span>
<span id="cb10-372"><a href="#cb10-372" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-373"><a href="#cb10-373" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>pc1_pc2_cohort <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb10-374"><a href="#cb10-374" aria-hidden="true" tabindex="-1"></a>    pca_fit_all_genes_wo_outliers,</span>
<span id="cb10-375"><a href="#cb10-375" aria-hidden="true" tabindex="-1"></a>    <span class="at">colby =</span> <span class="st">"cohort"</span>,</span>
<span id="cb10-376"><a href="#cb10-376" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> <span class="cn">NULL</span>, </span>
<span id="cb10-377"><a href="#cb10-377" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb10-378"><a href="#cb10-378" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC1"</span>,</span>
<span id="cb10-379"><a href="#cb10-379" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC2"</span>,</span>
<span id="cb10-380"><a href="#cb10-380" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"First two components colored by cohort"</span>,</span>
<span id="cb10-381"><a href="#cb10-381" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">"Only 1000 training samples"</span>,</span>
<span id="cb10-382"><a href="#cb10-382" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> point_size</span>
<span id="cb10-383"><a href="#cb10-383" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-384"><a href="#cb10-384" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-385"><a href="#cb10-385" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>pc1_pc2_er_status <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb10-386"><a href="#cb10-386" aria-hidden="true" tabindex="-1"></a>    pca_fit_all_genes_wo_outliers,</span>
<span id="cb10-387"><a href="#cb10-387" aria-hidden="true" tabindex="-1"></a>    <span class="at">colby =</span> <span class="st">"er_status"</span>,</span>
<span id="cb10-388"><a href="#cb10-388" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> <span class="cn">NULL</span>, </span>
<span id="cb10-389"><a href="#cb10-389" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb10-390"><a href="#cb10-390" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC1"</span>,</span>
<span id="cb10-391"><a href="#cb10-391" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC2"</span>,</span>
<span id="cb10-392"><a href="#cb10-392" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"First two components colored by ER status"</span>,</span>
<span id="cb10-393"><a href="#cb10-393" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">"Only 1000 training samples"</span>,</span>
<span id="cb10-394"><a href="#cb10-394" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> point_size</span>
<span id="cb10-395"><a href="#cb10-395" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-396"><a href="#cb10-396" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-397"><a href="#cb10-397" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>pc2_pc3_cohort <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb10-398"><a href="#cb10-398" aria-hidden="true" tabindex="-1"></a>    pca_fit_all_genes_wo_outliers,</span>
<span id="cb10-399"><a href="#cb10-399" aria-hidden="true" tabindex="-1"></a>    <span class="at">colby =</span> <span class="st">"cohort"</span>,</span>
<span id="cb10-400"><a href="#cb10-400" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> <span class="cn">NULL</span>, </span>
<span id="cb10-401"><a href="#cb10-401" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb10-402"><a href="#cb10-402" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC2"</span>,</span>
<span id="cb10-403"><a href="#cb10-403" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC3"</span>,</span>
<span id="cb10-404"><a href="#cb10-404" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Second and third components colored by cohort"</span>,</span>
<span id="cb10-405"><a href="#cb10-405" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">"Only 1000 training samples"</span>,</span>
<span id="cb10-406"><a href="#cb10-406" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> point_size</span>
<span id="cb10-407"><a href="#cb10-407" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-408"><a href="#cb10-408" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-409"><a href="#cb10-409" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>pc2_pc3_er_status <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb10-410"><a href="#cb10-410" aria-hidden="true" tabindex="-1"></a>    pca_fit_all_genes_wo_outliers,</span>
<span id="cb10-411"><a href="#cb10-411" aria-hidden="true" tabindex="-1"></a>    <span class="at">colby =</span> <span class="st">"er_status"</span>,</span>
<span id="cb10-412"><a href="#cb10-412" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> <span class="cn">NULL</span>, </span>
<span id="cb10-413"><a href="#cb10-413" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb10-414"><a href="#cb10-414" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC2"</span>,</span>
<span id="cb10-415"><a href="#cb10-415" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC3"</span>,</span>
<span id="cb10-416"><a href="#cb10-416" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Second and third components colored by ER status"</span>,</span>
<span id="cb10-417"><a href="#cb10-417" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">"Only 1000 training samples"</span>,</span>
<span id="cb10-418"><a href="#cb10-418" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> point_size</span>
<span id="cb10-419"><a href="#cb10-419" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-420"><a href="#cb10-420" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-421"><a href="#cb10-421" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>pc2_pc3_pam50 <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb10-422"><a href="#cb10-422" aria-hidden="true" tabindex="-1"></a>    pca_fit_all_genes_wo_outliers,</span>
<span id="cb10-423"><a href="#cb10-423" aria-hidden="true" tabindex="-1"></a>    <span class="at">colby =</span> <span class="st">"pam50"</span>,</span>
<span id="cb10-424"><a href="#cb10-424" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> <span class="cn">NULL</span>, </span>
<span id="cb10-425"><a href="#cb10-425" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb10-426"><a href="#cb10-426" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC2"</span>,</span>
<span id="cb10-427"><a href="#cb10-427" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC3"</span>,</span>
<span id="cb10-428"><a href="#cb10-428" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Second and third components colored by PAM50"</span>,</span>
<span id="cb10-429"><a href="#cb10-429" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">"Only 1000 training samples"</span>,</span>
<span id="cb10-430"><a href="#cb10-430" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> point_size</span>
<span id="cb10-431"><a href="#cb10-431" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-432"><a href="#cb10-432" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>pc2_pc3_pam50 <span class="ot">&lt;-</span> plots_pca_fit<span class="sc">$</span>pc2_pc3_pam50 <span class="sc">+</span></span>
<span id="cb10-433"><a href="#cb10-433" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_color_manual</span>(</span>
<span id="cb10-434"><a href="#cb10-434" aria-hidden="true" tabindex="-1"></a>        <span class="at">values =</span> <span class="fu">get_colors_pam50</span>(pca_fit_all_genes_wo_outliers<span class="sc">$</span>metadata)</span>
<span id="cb10-435"><a href="#cb10-435" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb10-436"><a href="#cb10-436" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb10-437"><a href="#cb10-437" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-438"><a href="#cb10-438" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>pc2_pc3_intclust <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb10-439"><a href="#cb10-439" aria-hidden="true" tabindex="-1"></a>    pca_fit_all_genes_wo_outliers,</span>
<span id="cb10-440"><a href="#cb10-440" aria-hidden="true" tabindex="-1"></a>    <span class="at">colby =</span> <span class="st">"INTCLUST"</span>,</span>
<span id="cb10-441"><a href="#cb10-441" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> <span class="cn">NULL</span>, </span>
<span id="cb10-442"><a href="#cb10-442" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb10-443"><a href="#cb10-443" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC2"</span>,</span>
<span id="cb10-444"><a href="#cb10-444" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC3"</span>,</span>
<span id="cb10-445"><a href="#cb10-445" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Second and third components colored by INTCLUST"</span>,</span>
<span id="cb10-446"><a href="#cb10-446" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">"Only 1000 training samples"</span>,</span>
<span id="cb10-447"><a href="#cb10-447" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> point_size</span>
<span id="cb10-448"><a href="#cb10-448" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-449"><a href="#cb10-449" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-450"><a href="#cb10-450" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>pc3_pc4_pam50 <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb10-451"><a href="#cb10-451" aria-hidden="true" tabindex="-1"></a>    pca_fit_all_genes_wo_outliers,</span>
<span id="cb10-452"><a href="#cb10-452" aria-hidden="true" tabindex="-1"></a>    <span class="at">colby =</span> <span class="st">"pam50"</span>,</span>
<span id="cb10-453"><a href="#cb10-453" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> <span class="cn">NULL</span>, </span>
<span id="cb10-454"><a href="#cb10-454" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb10-455"><a href="#cb10-455" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC3"</span>,</span>
<span id="cb10-456"><a href="#cb10-456" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC4"</span>,</span>
<span id="cb10-457"><a href="#cb10-457" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Third and fourth components colored by PAM50"</span>,</span>
<span id="cb10-458"><a href="#cb10-458" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">"Only 1000 training samples"</span>,</span>
<span id="cb10-459"><a href="#cb10-459" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> point_size</span>
<span id="cb10-460"><a href="#cb10-460" aria-hidden="true" tabindex="-1"></a>) </span>
<span id="cb10-461"><a href="#cb10-461" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>pc3_pc4_pam50 <span class="ot">&lt;-</span> plots_pca_fit<span class="sc">$</span>pc3_pc4_pam50 <span class="sc">+</span></span>
<span id="cb10-462"><a href="#cb10-462" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_color_manual</span>(</span>
<span id="cb10-463"><a href="#cb10-463" aria-hidden="true" tabindex="-1"></a>        <span class="at">values =</span> <span class="fu">get_colors_pam50</span>(pca_fit_all_genes_wo_outliers<span class="sc">$</span>metadata)</span>
<span id="cb10-464"><a href="#cb10-464" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb10-465"><a href="#cb10-465" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-466"><a href="#cb10-466" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>pc3_pc4_cohort <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb10-467"><a href="#cb10-467" aria-hidden="true" tabindex="-1"></a>    pca_fit_all_genes_wo_outliers,</span>
<span id="cb10-468"><a href="#cb10-468" aria-hidden="true" tabindex="-1"></a>    <span class="at">colby =</span> <span class="st">"cohort"</span>,</span>
<span id="cb10-469"><a href="#cb10-469" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> <span class="cn">NULL</span>, </span>
<span id="cb10-470"><a href="#cb10-470" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb10-471"><a href="#cb10-471" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC3"</span>,</span>
<span id="cb10-472"><a href="#cb10-472" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC4"</span>,</span>
<span id="cb10-473"><a href="#cb10-473" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Third and fourth components colored by cohort"</span>,</span>
<span id="cb10-474"><a href="#cb10-474" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">"Only 1000 training samples"</span>,</span>
<span id="cb10-475"><a href="#cb10-475" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> point_size</span>
<span id="cb10-476"><a href="#cb10-476" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-477"><a href="#cb10-477" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-478"><a href="#cb10-478" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(<span class="at">plotlist =</span> plots_pca_fit, <span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">labels =</span> <span class="st">"AUTO"</span>)</span>
<span id="cb10-479"><a href="#cb10-479" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-480"><a href="#cb10-480" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-481"><a href="#cb10-481" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb10-482"><a href="#cb10-482" aria-hidden="true" tabindex="-1"></a>datasets_pca_coordinates <span class="ot">&lt;-</span> <span class="fu">lapply</span>(</span>
<span id="cb10-483"><a href="#cb10-483" aria-hidden="true" tabindex="-1"></a>    datasets_normalized,</span>
<span id="cb10-484"><a href="#cb10-484" aria-hidden="true" tabindex="-1"></a>    get_pca_coordinates,</span>
<span id="cb10-485"><a href="#cb10-485" aria-hidden="true" tabindex="-1"></a>    <span class="at">pca_fit =</span> pca_fit_all_genes_wo_outliers,</span>
<span id="cb10-486"><a href="#cb10-486" aria-hidden="true" tabindex="-1"></a>    <span class="at">genes_for_pca =</span> <span class="fu">rownames</span>(pca_fit_all_genes_wo_outliers<span class="sc">$</span>loadings)</span>
<span id="cb10-487"><a href="#cb10-487" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-488"><a href="#cb10-488" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-489"><a href="#cb10-489" aria-hidden="true" tabindex="-1"></a>df_pca_coordinates_wo_outliers <span class="ot">&lt;-</span> datasets_pca_coordinates <span class="sc">%&gt;%</span> </span>
<span id="cb10-490"><a href="#cb10-490" aria-hidden="true" tabindex="-1"></a>    <span class="fu">do.call</span>(rbind, .) <span class="sc">%&gt;%</span></span>
<span id="cb10-491"><a href="#cb10-491" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span> </span>
<span id="cb10-492"><a href="#cb10-492" aria-hidden="true" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"sample_name"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-493"><a href="#cb10-493" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">inner_join</span>(</span>
<span id="cb10-494"><a href="#cb10-494" aria-hidden="true" tabindex="-1"></a>        .,</span>
<span id="cb10-495"><a href="#cb10-495" aria-hidden="true" tabindex="-1"></a>        merged_col_data,</span>
<span id="cb10-496"><a href="#cb10-496" aria-hidden="true" tabindex="-1"></a>        <span class="at">by =</span> <span class="st">"sample_name"</span></span>
<span id="cb10-497"><a href="#cb10-497" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb10-498"><a href="#cb10-498" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-499"><a href="#cb10-499" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb10-500"><a href="#cb10-500" aria-hidden="true" tabindex="-1"></a>    df_pca_coordinates_wo_outliers,</span>
<span id="cb10-501"><a href="#cb10-501" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/trying/df_pca_coordinates_wo_outliers.rds"</span></span>
<span id="cb10-502"><a href="#cb10-502" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-503"><a href="#cb10-503" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-504"><a href="#cb10-504" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-507"><a href="#cb10-507" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-508"><a href="#cb10-508" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_pca_coordinates</span>(</span>
<span id="cb10-509"><a href="#cb10-509" aria-hidden="true" tabindex="-1"></a>    df_pca_coordinates_wo_outliers <span class="sc">%&gt;%</span> </span>
<span id="cb10-510"><a href="#cb10-510" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">slice</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">n</span>()),</span>
<span id="cb10-511"><a href="#cb10-511" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"cohort"</span>,</span>
<span id="cb10-512"><a href="#cb10-512" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC3"</span>,</span>
<span id="cb10-513"><a href="#cb10-513" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC4"</span>,</span>
<span id="cb10-514"><a href="#cb10-514" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">1</span>,</span>
<span id="cb10-515"><a href="#cb10-515" aria-hidden="true" tabindex="-1"></a>    <span class="at">base_size =</span> <span class="dv">10</span>,</span>
<span id="cb10-516"><a href="#cb10-516" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="fu">paste0</span>(</span>
<span id="cb10-517"><a href="#cb10-517" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Embedding of all samples from TCGA,</span><span class="sc">\n</span><span class="st">"</span>,</span>
<span id="cb10-518"><a href="#cb10-518" aria-hidden="true" tabindex="-1"></a>        <span class="st">"METABRIC and SCANB"</span></span>
<span id="cb10-519"><a href="#cb10-519" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb10-520"><a href="#cb10-520" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span> ggplot2<span class="sc">::</span><span class="fu">scale_color_viridis_d</span>() <span class="sc">+</span></span>
<span id="cb10-521"><a href="#cb10-521" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">15</span>) <span class="sc">+</span></span>
<span id="cb10-522"><a href="#cb10-522" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span></span>
<span id="cb10-523"><a href="#cb10-523" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb10-524"><a href="#cb10-524" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-525"><a href="#cb10-525" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-528"><a href="#cb10-528" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-529"><a href="#cb10-529" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-pca-scanb-pc1</span></span>
<span id="cb10-530"><a href="#cb10-530" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: PCA embedding colored by cohort. The components used are the</span></span>
<span id="cb10-531"><a href="#cb10-531" aria-hidden="true" tabindex="-1"></a><span class="co">#|     first and second components.</span></span>
<span id="cb10-532"><a href="#cb10-532" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_pca_coordinates</span>(</span>
<span id="cb10-533"><a href="#cb10-533" aria-hidden="true" tabindex="-1"></a>    df_pca_coordinates_wo_outliers,</span>
<span id="cb10-534"><a href="#cb10-534" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"cohort"</span>,</span>
<span id="cb10-535"><a href="#cb10-535" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC1"</span>,</span>
<span id="cb10-536"><a href="#cb10-536" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC2"</span>,</span>
<span id="cb10-537"><a href="#cb10-537" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">1</span>,</span>
<span id="cb10-538"><a href="#cb10-538" aria-hidden="true" tabindex="-1"></a>    <span class="at">base_size =</span> <span class="dv">10</span>,</span>
<span id="cb10-539"><a href="#cb10-539" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="fu">paste0</span>(</span>
<span id="cb10-540"><a href="#cb10-540" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Embedding of all samples from TCGA,</span><span class="sc">\n</span><span class="st">"</span>,</span>
<span id="cb10-541"><a href="#cb10-541" aria-hidden="true" tabindex="-1"></a>        <span class="st">"METABRIC and SCANB"</span></span>
<span id="cb10-542"><a href="#cb10-542" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb10-543"><a href="#cb10-543" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span> </span>
<span id="cb10-544"><a href="#cb10-544" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">15</span>) <span class="sc">+</span></span>
<span id="cb10-545"><a href="#cb10-545" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span></span>
<span id="cb10-546"><a href="#cb10-546" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb10-547"><a href="#cb10-547" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-548"><a href="#cb10-548" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-549"><a href="#cb10-549" aria-hidden="true" tabindex="-1"></a>@fig-pca-scanb-er-pam50-all-genes shows that the SCANB samples are also well mixed</span>
<span id="cb10-550"><a href="#cb10-550" aria-hidden="true" tabindex="-1"></a>regarding the clinical factors, including the $SET_{ER/PR}$ signature.</span>
<span id="cb10-551"><a href="#cb10-551" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-552"><a href="#cb10-552" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=16, fig.height=12}</span></span>
<span id="cb10-553"><a href="#cb10-553" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-pca-scanb-er-pam50-all-genes</span></span>
<span id="cb10-554"><a href="#cb10-554" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: PCA embedding of all samples from TCGA, SCANB and METABRIC.</span></span>
<span id="cb10-555"><a href="#cb10-555" aria-hidden="true" tabindex="-1"></a><span class="co">#|     (A) Colored by cohort,</span></span>
<span id="cb10-556"><a href="#cb10-556" aria-hidden="true" tabindex="-1"></a><span class="co">#|     (B) colored by ER status, (C) colored by PAM50 molecular subtype,</span></span>
<span id="cb10-557"><a href="#cb10-557" aria-hidden="true" tabindex="-1"></a><span class="co">#|     (D) colored by the $SET_{ER/PR}$ signature and only SCANB samples.</span></span>
<span id="cb10-558"><a href="#cb10-558" aria-hidden="true" tabindex="-1"></a>size <span class="ot">&lt;-</span> <span class="dv">2</span></span>
<span id="cb10-559"><a href="#cb10-559" aria-hidden="true" tabindex="-1"></a>base_size <span class="ot">&lt;-</span> <span class="dv">20</span></span>
<span id="cb10-560"><a href="#cb10-560" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-561"><a href="#cb10-561" aria-hidden="true" tabindex="-1"></a>plots_with_scanb <span class="ot">&lt;-</span> <span class="fu">sapply</span>(</span>
<span id="cb10-562"><a href="#cb10-562" aria-hidden="true" tabindex="-1"></a>    <span class="fu">c</span>(<span class="st">"cohort"</span>, <span class="st">"er_status"</span>, <span class="st">"pam50"</span>, <span class="st">"SET_ERPR"</span>), </span>
<span id="cb10-563"><a href="#cb10-563" aria-hidden="true" tabindex="-1"></a>    plot_pca_coordinates,</span>
<span id="cb10-564"><a href="#cb10-564" aria-hidden="true" tabindex="-1"></a>    <span class="at">df_pca =</span> df_pca_coordinates_wo_outliers <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb10-565"><a href="#cb10-565" aria-hidden="true" tabindex="-1"></a>        pam50 <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"luma"</span>, <span class="st">"lumb"</span>, <span class="st">"basal"</span>, <span class="st">"her2"</span>, <span class="st">"normal"</span>)</span>
<span id="cb10-566"><a href="#cb10-566" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb10-567"><a href="#cb10-567" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span>(cohort <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"poetic"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb10-568"><a href="#cb10-568" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">slice</span>(<span class="fu">sample</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">n</span>())),</span>
<span id="cb10-569"><a href="#cb10-569" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC3"</span>, </span>
<span id="cb10-570"><a href="#cb10-570" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC4"</span>,</span>
<span id="cb10-571"><a href="#cb10-571" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> size,</span>
<span id="cb10-572"><a href="#cb10-572" aria-hidden="true" tabindex="-1"></a>    <span class="at">base_size =</span> base_size,</span>
<span id="cb10-573"><a href="#cb10-573" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="fu">paste0</span>(</span>
<span id="cb10-574"><a href="#cb10-574" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Embedding of all samples from TCGA,</span><span class="sc">\n</span><span class="st">"</span>,</span>
<span id="cb10-575"><a href="#cb10-575" aria-hidden="true" tabindex="-1"></a>        <span class="st">"METABRIC and SCANB"</span></span>
<span id="cb10-576"><a href="#cb10-576" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb10-577"><a href="#cb10-577" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>, </span>
<span id="cb10-578"><a href="#cb10-578" aria-hidden="true" tabindex="-1"></a>    <span class="at">simplify =</span> <span class="cn">FALSE</span></span>
<span id="cb10-579"><a href="#cb10-579" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-580"><a href="#cb10-580" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-581"><a href="#cb10-581" aria-hidden="true" tabindex="-1"></a>plots_with_scanb<span class="sc">$</span>cohort <span class="ot">&lt;-</span> plots_with_scanb<span class="sc">$</span>cohort <span class="sc">+</span></span>
<span id="cb10-582"><a href="#cb10-582" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span></span>
<span id="cb10-583"><a href="#cb10-583" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb10-584"><a href="#cb10-584" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-585"><a href="#cb10-585" aria-hidden="true" tabindex="-1"></a>plots_with_scanb<span class="sc">$</span>pam50 <span class="ot">&lt;-</span> plots_with_scanb<span class="sc">$</span>pam50 <span class="sc">+</span></span>
<span id="cb10-586"><a href="#cb10-586" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_color_manual</span>(</span>
<span id="cb10-587"><a href="#cb10-587" aria-hidden="true" tabindex="-1"></a>        <span class="at">values =</span> <span class="fu">get_colors_pam50</span>(plots_with_scanb<span class="sc">$</span>pam50<span class="sc">$</span>data)</span>
<span id="cb10-588"><a href="#cb10-588" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb10-589"><a href="#cb10-589" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">color =</span> <span class="st">"PAM50"</span>) <span class="sc">+</span> </span>
<span id="cb10-590"><a href="#cb10-590" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span></span>
<span id="cb10-591"><a href="#cb10-591" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb10-592"><a href="#cb10-592" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-593"><a href="#cb10-593" aria-hidden="true" tabindex="-1"></a>plots_with_scanb<span class="sc">$</span>er_status <span class="ot">&lt;-</span>  plots_with_scanb<span class="sc">$</span>er_status <span class="sc">+</span></span>
<span id="cb10-594"><a href="#cb10-594" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_color_viridis_d</span>() <span class="sc">+</span></span>
<span id="cb10-595"><a href="#cb10-595" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">color =</span> <span class="st">"ER status"</span>) <span class="sc">+</span></span>
<span id="cb10-596"><a href="#cb10-596" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span></span>
<span id="cb10-597"><a href="#cb10-597" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb10-598"><a href="#cb10-598" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-599"><a href="#cb10-599" aria-hidden="true" tabindex="-1"></a>plots_with_scanb<span class="sc">$</span>SET_ERPR <span class="ot">&lt;-</span> df_pca_coordinates_wo_outliers <span class="sc">%&gt;%</span> </span>
<span id="cb10-600"><a href="#cb10-600" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"scanb"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb10-601"><a href="#cb10-601" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes_string</span>(<span class="at">x =</span> <span class="st">"PC3"</span>, <span class="at">y =</span> <span class="st">"PC4"</span>, <span class="at">z =</span> <span class="st">"SET_ERPR"</span>)) <span class="sc">+</span> </span>
<span id="cb10-602"><a href="#cb10-602" aria-hidden="true" tabindex="-1"></a>        ggplot2<span class="sc">::</span><span class="fu">stat_summary_hex</span>(<span class="at">bins =</span> <span class="dv">25</span>) <span class="sc">+</span></span>
<span id="cb10-603"><a href="#cb10-603" aria-hidden="true" tabindex="-1"></a>        ggplot2<span class="sc">::</span><span class="fu">scale_fill_viridis_c</span>() <span class="sc">+</span></span>
<span id="cb10-604"><a href="#cb10-604" aria-hidden="true" tabindex="-1"></a>        ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb10-605"><a href="#cb10-605" aria-hidden="true" tabindex="-1"></a>            <span class="at">title =</span> <span class="st">"Embedding of SCANB only"</span>,</span>
<span id="cb10-606"><a href="#cb10-606" aria-hidden="true" tabindex="-1"></a>            <span class="at">fill =</span> <span class="fu">expression</span>(SET[ER<span class="sc">/</span>PR])</span>
<span id="cb10-607"><a href="#cb10-607" aria-hidden="true" tabindex="-1"></a>        ) <span class="sc">+</span> </span>
<span id="cb10-608"><a href="#cb10-608" aria-hidden="true" tabindex="-1"></a>        ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> base_size) <span class="sc">+</span></span>
<span id="cb10-609"><a href="#cb10-609" aria-hidden="true" tabindex="-1"></a>        <span class="fu">change_plot_aes_point</span>()</span>
<span id="cb10-610"><a href="#cb10-610" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-611"><a href="#cb10-611" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(</span>
<span id="cb10-612"><a href="#cb10-612" aria-hidden="true" tabindex="-1"></a>    <span class="at">plotlist =</span> plots_with_scanb,</span>
<span id="cb10-613"><a href="#cb10-613" aria-hidden="true" tabindex="-1"></a>    <span class="at">ncol =</span> <span class="dv">2</span></span>
<span id="cb10-614"><a href="#cb10-614" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-615"><a href="#cb10-615" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-616"><a href="#cb10-616" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-617"><a href="#cb10-617" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=12, fig.height=8}</span></span>
<span id="cb10-618"><a href="#cb10-618" aria-hidden="true" tabindex="-1"></a>select_pcs <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"PC"</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">30</span>)</span>
<span id="cb10-619"><a href="#cb10-619" aria-hidden="true" tabindex="-1"></a>pca_fit_all_genes_wo_outliers<span class="sc">$</span>loadings <span class="sc">%&gt;%</span></span>
<span id="cb10-620"><a href="#cb10-620" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span></span>
<span id="cb10-621"><a href="#cb10-621" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(dplyr<span class="sc">::</span><span class="fu">all_of</span>(select_pcs)) <span class="sc">%&gt;%</span></span>
<span id="cb10-622"><a href="#cb10-622" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(</span>
<span id="cb10-623"><a href="#cb10-623" aria-hidden="true" tabindex="-1"></a>        <span class="at">cols =</span> dplyr<span class="sc">::</span><span class="fu">all_of</span>(select_pcs),</span>
<span id="cb10-624"><a href="#cb10-624" aria-hidden="true" tabindex="-1"></a>        <span class="at">names_to =</span> <span class="st">"component"</span>, </span>
<span id="cb10-625"><a href="#cb10-625" aria-hidden="true" tabindex="-1"></a>        <span class="at">values_to =</span> <span class="st">"loadings"</span></span>
<span id="cb10-626"><a href="#cb10-626" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb10-627"><a href="#cb10-627" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(</span>
<span id="cb10-628"><a href="#cb10-628" aria-hidden="true" tabindex="-1"></a>        <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">abs</span>(loadings))</span>
<span id="cb10-629"><a href="#cb10-629" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb10-630"><a href="#cb10-630" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_histogram</span>(<span class="at">bins =</span> <span class="dv">30</span>) <span class="sc">+</span></span>
<span id="cb10-631"><a href="#cb10-631" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>component) <span class="sc">+</span> </span>
<span id="cb10-632"><a href="#cb10-632" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb10-633"><a href="#cb10-633" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">"Histogram of the loadings from the first four components"</span></span>
<span id="cb10-634"><a href="#cb10-634" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb10-635"><a href="#cb10-635" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>) <span class="sc">+</span></span>
<span id="cb10-636"><a href="#cb10-636" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>()</span>
<span id="cb10-637"><a href="#cb10-637" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-638"><a href="#cb10-638" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-639"><a href="#cb10-639" aria-hidden="true" tabindex="-1"></a>Let us also check the other components in the biplots.</span>
<span id="cb10-640"><a href="#cb10-640" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-641"><a href="#cb10-641" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=16, fig.height=12}</span></span>
<span id="cb10-642"><a href="#cb10-642" aria-hidden="true" tabindex="-1"></a>color_by <span class="ot">&lt;-</span> <span class="st">"pam50"</span></span>
<span id="cb10-643"><a href="#cb10-643" aria-hidden="true" tabindex="-1"></a>GGally<span class="sc">::</span><span class="fu">ggpairs</span>(</span>
<span id="cb10-644"><a href="#cb10-644" aria-hidden="true" tabindex="-1"></a>    df_pca_coordinates_wo_outliers <span class="sc">%&gt;%</span></span>
<span id="cb10-645"><a href="#cb10-645" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">filter</span>(sample_name <span class="sc">%in%</span></span>
<span id="cb10-646"><a href="#cb10-646" aria-hidden="true" tabindex="-1"></a>                          pca_fit_all_genes_wo_outliers<span class="sc">$</span>metadata<span class="sc">$</span>sample_name</span>
<span id="cb10-647"><a href="#cb10-647" aria-hidden="true" tabindex="-1"></a>        ) <span class="sc">%&gt;%</span></span>
<span id="cb10-648"><a href="#cb10-648" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">select</span>(dplyr<span class="sc">::</span><span class="fu">all_of</span>(<span class="fu">c</span>(<span class="fu">paste0</span>(<span class="st">"PC"</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>), color_by))),</span>
<span id="cb10-649"><a href="#cb10-649" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">color =</span> <span class="sc">!!</span><span class="fu">sym</span>(color_by), <span class="at">alpha =</span> <span class="fl">0.5</span>)</span>
<span id="cb10-650"><a href="#cb10-650" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-651"><a href="#cb10-651" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-652"><a href="#cb10-652" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-653"><a href="#cb10-653" aria-hidden="true" tabindex="-1"></a>We see that the pam50 subtypes are being separated using the components 3, 4</span>
<span id="cb10-654"><a href="#cb10-654" aria-hidden="true" tabindex="-1"></a>and also 8. In the component 8 there is some expression that is able </span>
<span id="cb10-655"><a href="#cb10-655" aria-hidden="true" tabindex="-1"></a>to differentiate between HER2-like from all other samples.</span>
<span id="cb10-656"><a href="#cb10-656" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-657"><a href="#cb10-657" aria-hidden="true" tabindex="-1"></a>We now plot by cohort.</span>
<span id="cb10-658"><a href="#cb10-658" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-659"><a href="#cb10-659" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=16, fig.height=12}</span></span>
<span id="cb10-660"><a href="#cb10-660" aria-hidden="true" tabindex="-1"></a>color_by <span class="ot">&lt;-</span> <span class="st">"cohort"</span></span>
<span id="cb10-661"><a href="#cb10-661" aria-hidden="true" tabindex="-1"></a>GGally<span class="sc">::</span><span class="fu">ggpairs</span>(</span>
<span id="cb10-662"><a href="#cb10-662" aria-hidden="true" tabindex="-1"></a>    df_pca_coordinates_wo_outliers <span class="sc">%&gt;%</span></span>
<span id="cb10-663"><a href="#cb10-663" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">filter</span>(sample_name <span class="sc">%in%</span> </span>
<span id="cb10-664"><a href="#cb10-664" aria-hidden="true" tabindex="-1"></a>                          pca_fit_all_genes_wo_outliers<span class="sc">$</span>metadata<span class="sc">$</span>sample_name</span>
<span id="cb10-665"><a href="#cb10-665" aria-hidden="true" tabindex="-1"></a>        ) <span class="sc">%&gt;%</span></span>
<span id="cb10-666"><a href="#cb10-666" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">select</span>(dplyr<span class="sc">::</span><span class="fu">all_of</span>(<span class="fu">c</span>(<span class="fu">paste0</span>(<span class="st">"PC"</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>), color_by))),</span>
<span id="cb10-667"><a href="#cb10-667" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">color =</span> <span class="sc">!!</span><span class="fu">sym</span>(color_by), <span class="at">alpha =</span> <span class="fl">0.5</span>)</span>
<span id="cb10-668"><a href="#cb10-668" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-669"><a href="#cb10-669" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-670"><a href="#cb10-670" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-671"><a href="#cb10-671" aria-hidden="true" tabindex="-1"></a>In PC1, 2 and 5 there are some small differences in the differences of the </span>
<span id="cb10-672"><a href="#cb10-672" aria-hidden="true" tabindex="-1"></a>distributions from the datasets. We will then remove these 3 components</span>
<span id="cb10-673"><a href="#cb10-673" aria-hidden="true" tabindex="-1"></a>up to component 10 with exception of components 3, 4 and 5.</span>
<span id="cb10-674"><a href="#cb10-674" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-675"><a href="#cb10-675" aria-hidden="true" tabindex="-1"></a><span class="fu">## Removing the batch effects</span></span>
<span id="cb10-676"><a href="#cb10-676" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-677"><a href="#cb10-677" aria-hidden="true" tabindex="-1"></a>We now try to remove the batch effects by removing all the PCs that are not</span>
<span id="cb10-678"><a href="#cb10-678" aria-hidden="true" tabindex="-1"></a>of interest, so we remove the first two components and the fifth as well. We</span>
<span id="cb10-679"><a href="#cb10-679" aria-hidden="true" tabindex="-1"></a>then calculate the scores for each cohort separately and then compare</span>
<span id="cb10-680"><a href="#cb10-680" aria-hidden="true" tabindex="-1"></a>to what was calculated before. We can then try to calculate the scores</span>
<span id="cb10-681"><a href="#cb10-681" aria-hidden="true" tabindex="-1"></a>using samples coming from all cohorts.</span>
<span id="cb10-682"><a href="#cb10-682" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-683"><a href="#cb10-683" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb10-684"><a href="#cb10-684" aria-hidden="true" tabindex="-1"></a>data_together <span class="ot">&lt;-</span> <span class="fu">lapply</span>(</span>
<span id="cb10-685"><a href="#cb10-685" aria-hidden="true" tabindex="-1"></a>    datasets_normalized,</span>
<span id="cb10-686"><a href="#cb10-686" aria-hidden="true" tabindex="-1"></a>    \(x, y) <span class="fu">assay</span>(x, y) <span class="sc">%&gt;%</span> as.matrix,</span>
<span id="cb10-687"><a href="#cb10-687" aria-hidden="true" tabindex="-1"></a>    <span class="st">"avg_ranking"</span></span>
<span id="cb10-688"><a href="#cb10-688" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> <span class="fu">Reduce</span>(cbind, .)</span>
<span id="cb10-689"><a href="#cb10-689" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-690"><a href="#cb10-690" aria-hidden="true" tabindex="-1"></a>samples_for_training <span class="ot">&lt;-</span> pca_fit_all_genes_wo_outliers<span class="sc">$</span>metadata<span class="sc">$</span>sample_name</span>
<span id="cb10-691"><a href="#cb10-691" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-692"><a href="#cb10-692" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(df_pca_coordinates_wo_outliers) <span class="ot">&lt;-</span> df_pca_coordinates_wo_outliers<span class="sc">$</span>sample_name</span>
<span id="cb10-693"><a href="#cb10-693" aria-hidden="true" tabindex="-1"></a>name_components <span class="ot">&lt;-</span> <span class="fu">colnames</span>(pca_fit_all_genes_wo_outliers<span class="sc">$</span>loadings)</span>
<span id="cb10-694"><a href="#cb10-694" aria-hidden="true" tabindex="-1"></a><span class="co"># from previous analysis we decided that the component 5 still included</span></span>
<span id="cb10-695"><a href="#cb10-695" aria-hidden="true" tabindex="-1"></a><span class="co"># differences between the two datasets, so we decided to remove it as well </span></span>
<span id="cb10-696"><a href="#cb10-696" aria-hidden="true" tabindex="-1"></a><span class="co"># besides component 1 and 2</span></span>
<span id="cb10-697"><a href="#cb10-697" aria-hidden="true" tabindex="-1"></a>remove_components <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"PC"</span>, <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">5</span>))</span>
<span id="cb10-698"><a href="#cb10-698" aria-hidden="true" tabindex="-1"></a>keep_components <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(name_components, remove_components)</span>
<span id="cb10-699"><a href="#cb10-699" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-700"><a href="#cb10-700" aria-hidden="true" tabindex="-1"></a><span class="co"># the code below calculates the regressed dataframe for all samples from</span></span>
<span id="cb10-701"><a href="#cb10-701" aria-hidden="true" tabindex="-1"></a><span class="co"># scanb, tcga and metabric</span></span>
<span id="cb10-702"><a href="#cb10-702" aria-hidden="true" tabindex="-1"></a>df_pcs_regressed <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(</span>
<span id="cb10-703"><a href="#cb10-703" aria-hidden="true" tabindex="-1"></a>    df_pca_coordinates_wo_outliers[<span class="fu">colnames</span>(data_together), keep_components]</span>
<span id="cb10-704"><a href="#cb10-704" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%*%</span> <span class="fu">t</span>(pca_fit_all_genes_wo_outliers<span class="sc">$</span>loadings[, keep_components]) <span class="sc">%&gt;%</span></span>
<span id="cb10-705"><a href="#cb10-705" aria-hidden="true" tabindex="-1"></a>    t</span>
<span id="cb10-706"><a href="#cb10-706" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-707"><a href="#cb10-707" aria-hidden="true" tabindex="-1"></a>pca_removed_pcs <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">pca</span>(</span>
<span id="cb10-708"><a href="#cb10-708" aria-hidden="true" tabindex="-1"></a>    <span class="at">mat =</span> df_pcs_regressed[, samples_for_training],</span>
<span id="cb10-709"><a href="#cb10-709" aria-hidden="true" tabindex="-1"></a>    <span class="at">metadata =</span> pca_fit_all_genes_wo_outliers<span class="sc">$</span>metadata</span>
<span id="cb10-710"><a href="#cb10-710" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-711"><a href="#cb10-711" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-712"><a href="#cb10-712" aria-hidden="true" tabindex="-1"></a>pca_all_samples_wo_pcs <span class="ot">&lt;-</span> <span class="fu">t</span>(df_pcs_regressed <span class="sc">%&gt;%</span> as.matrix) <span class="sc">%*%</span> </span>
<span id="cb10-713"><a href="#cb10-713" aria-hidden="true" tabindex="-1"></a>    (pca_removed_pcs<span class="sc">$</span>loadings <span class="sc">%&gt;%</span> as.matrix)</span>
<span id="cb10-714"><a href="#cb10-714" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-715"><a href="#cb10-715" aria-hidden="true" tabindex="-1"></a>df_pca_wo_pcs <span class="ot">&lt;-</span> pca_all_samples_wo_pcs <span class="sc">%&gt;%</span></span>
<span id="cb10-716"><a href="#cb10-716" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span></span>
<span id="cb10-717"><a href="#cb10-717" aria-hidden="true" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"sample_name"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-718"><a href="#cb10-718" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">inner_join</span>(</span>
<span id="cb10-719"><a href="#cb10-719" aria-hidden="true" tabindex="-1"></a>        df_pca[, <span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"PC"</span>, <span class="fu">colnames</span>(df_pca))],</span>
<span id="cb10-720"><a href="#cb10-720" aria-hidden="true" tabindex="-1"></a>        <span class="at">by =</span> <span class="st">"sample_name"</span></span>
<span id="cb10-721"><a href="#cb10-721" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb10-722"><a href="#cb10-722" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-723"><a href="#cb10-723" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb10-724"><a href="#cb10-724" aria-hidden="true" tabindex="-1"></a>    df_pca_wo_pcs, </span>
<span id="cb10-725"><a href="#cb10-725" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/trying/df_pca_wo_pcs.rds"</span></span>
<span id="cb10-726"><a href="#cb10-726" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-727"><a href="#cb10-727" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-728"><a href="#cb10-728" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb10-729"><a href="#cb10-729" aria-hidden="true" tabindex="-1"></a>    pca_removed_pcs,</span>
<span id="cb10-730"><a href="#cb10-730" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/trying/pca_removed_pcs.rds"</span></span>
<span id="cb10-731"><a href="#cb10-731" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-732"><a href="#cb10-732" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-733"><a href="#cb10-733" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb10-734"><a href="#cb10-734" aria-hidden="true" tabindex="-1"></a>    df_pcs_regressed, </span>
<span id="cb10-735"><a href="#cb10-735" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/trying/df_pcs_regressed.rds"</span></span>
<span id="cb10-736"><a href="#cb10-736" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-737"><a href="#cb10-737" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-738"><a href="#cb10-738" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-741"><a href="#cb10-741" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-742"><a href="#cb10-742" aria-hidden="true" tabindex="-1"></a>PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb10-743"><a href="#cb10-743" aria-hidden="true" tabindex="-1"></a>    pca_removed_pcs,</span>
<span id="cb10-744"><a href="#cb10-744" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> <span class="cn">NULL</span>,</span>
<span id="cb10-745"><a href="#cb10-745" aria-hidden="true" tabindex="-1"></a>    <span class="at">colby =</span> <span class="st">"cohort"</span>,</span>
<span id="cb10-746"><a href="#cb10-746" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">"right"</span></span>
<span id="cb10-747"><a href="#cb10-747" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-748"><a href="#cb10-748" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-749"><a href="#cb10-749" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-752"><a href="#cb10-752" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-753"><a href="#cb10-753" aria-hidden="true" tabindex="-1"></a>PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb10-754"><a href="#cb10-754" aria-hidden="true" tabindex="-1"></a>    pca_removed_pcs,</span>
<span id="cb10-755"><a href="#cb10-755" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> <span class="cn">NULL</span>,</span>
<span id="cb10-756"><a href="#cb10-756" aria-hidden="true" tabindex="-1"></a>    <span class="at">colby =</span> <span class="st">"pam50"</span>,</span>
<span id="cb10-757"><a href="#cb10-757" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">"right"</span></span>
<span id="cb10-758"><a href="#cb10-758" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-759"><a href="#cb10-759" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-760"><a href="#cb10-760" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-761"><a href="#cb10-761" aria-hidden="true" tabindex="-1"></a>In general it looks good, the luminal B population is a bit compressed </span>
<span id="cb10-762"><a href="#cb10-762" aria-hidden="true" tabindex="-1"></a>towards the luminal A population though.</span>
<span id="cb10-763"><a href="#cb10-763" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-764"><a href="#cb10-764" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width = 6, fig.height = 4}</span></span>
<span id="cb10-765"><a href="#cb10-765" aria-hidden="true" tabindex="-1"></a>size_dots <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb10-766"><a href="#cb10-766" aria-hidden="true" tabindex="-1"></a>alpha_val <span class="ot">&lt;-</span> <span class="fl">0.8</span></span>
<span id="cb10-767"><a href="#cb10-767" aria-hidden="true" tabindex="-1"></a>base_size <span class="ot">&lt;-</span> <span class="dv">10</span></span>
<span id="cb10-768"><a href="#cb10-768" aria-hidden="true" tabindex="-1"></a>size_legend <span class="ot">&lt;-</span> <span class="dv">5</span></span>
<span id="cb10-769"><a href="#cb10-769" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-770"><a href="#cb10-770" aria-hidden="true" tabindex="-1"></a>ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(df_pca_wo_pcs, <span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> cohort)) <span class="sc">+</span> </span>
<span id="cb10-771"><a href="#cb10-771" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>(<span class="at">size =</span> size_dots, <span class="at">alpha =</span> alpha_val) <span class="sc">+</span></span>
<span id="cb10-772"><a href="#cb10-772" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_alpha</span>(<span class="at">guide =</span> <span class="st">'none'</span>) <span class="sc">+</span></span>
<span id="cb10-773"><a href="#cb10-773" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb10-774"><a href="#cb10-774" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">"PAM50"</span></span>
<span id="cb10-775"><a href="#cb10-775" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb10-776"><a href="#cb10-776" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> base_size)</span>
<span id="cb10-777"><a href="#cb10-777" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-778"><a href="#cb10-778" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-779"><a href="#cb10-779" aria-hidden="true" tabindex="-1"></a>There is a very good merging of the datasets. We see mostly</span>
<span id="cb10-780"><a href="#cb10-780" aria-hidden="true" tabindex="-1"></a>green points because they are the majority. </span>
<span id="cb10-781"><a href="#cb10-781" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-782"><a href="#cb10-782" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=8}</span></span>
<span id="cb10-783"><a href="#cb10-783" aria-hidden="true" tabindex="-1"></a>pcs_loadings_plot <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"PC"</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>)</span>
<span id="cb10-784"><a href="#cb10-784" aria-hidden="true" tabindex="-1"></a>pca_removed_pcs<span class="sc">$</span>loadings <span class="sc">%&gt;%</span></span>
<span id="cb10-785"><a href="#cb10-785" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span></span>
<span id="cb10-786"><a href="#cb10-786" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(dplyr<span class="sc">::</span><span class="fu">all_of</span>(pcs_loadings_plot)) <span class="sc">%&gt;%</span></span>
<span id="cb10-787"><a href="#cb10-787" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(</span>
<span id="cb10-788"><a href="#cb10-788" aria-hidden="true" tabindex="-1"></a>        <span class="at">cols =</span> dplyr<span class="sc">::</span><span class="fu">all_of</span>(pcs_loadings_plot),</span>
<span id="cb10-789"><a href="#cb10-789" aria-hidden="true" tabindex="-1"></a>        <span class="at">names_to =</span> <span class="st">"component"</span>, </span>
<span id="cb10-790"><a href="#cb10-790" aria-hidden="true" tabindex="-1"></a>        <span class="at">values_to =</span> <span class="st">"loadings"</span></span>
<span id="cb10-791"><a href="#cb10-791" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb10-792"><a href="#cb10-792" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">component =</span> <span class="fu">factor</span>(component, <span class="at">levels =</span> pcs_loadings_plot)) <span class="sc">%&gt;%</span></span>
<span id="cb10-793"><a href="#cb10-793" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(</span>
<span id="cb10-794"><a href="#cb10-794" aria-hidden="true" tabindex="-1"></a>        <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">abs</span>(loadings))</span>
<span id="cb10-795"><a href="#cb10-795" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb10-796"><a href="#cb10-796" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_histogram</span>(<span class="at">bins =</span> <span class="dv">30</span>) <span class="sc">+</span></span>
<span id="cb10-797"><a href="#cb10-797" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>component) <span class="sc">+</span> </span>
<span id="cb10-798"><a href="#cb10-798" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb10-799"><a href="#cb10-799" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">"Histogram of the loadings from the first four components"</span></span>
<span id="cb10-800"><a href="#cb10-800" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb10-801"><a href="#cb10-801" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>)</span>
<span id="cb10-802"><a href="#cb10-802" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-803"><a href="#cb10-803" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-804"><a href="#cb10-804" aria-hidden="true" tabindex="-1"></a>Now we see that the first loadings has a tail in a similar fashio as the</span>
<span id="cb10-805"><a href="#cb10-805" aria-hidden="true" tabindex="-1"></a>other loadings. </span>
<span id="cb10-806"><a href="#cb10-806" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-807"><a href="#cb10-807" aria-hidden="true" tabindex="-1"></a>We now try to calculate the scores of the samples using these new embeddings</span>
<span id="cb10-808"><a href="#cb10-808" aria-hidden="true" tabindex="-1"></a>only from TCGA samples as we will compare the results later on. </span>
<span id="cb10-809"><a href="#cb10-809" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-810"><a href="#cb10-810" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb10-811"><a href="#cb10-811" aria-hidden="true" tabindex="-1"></a>gsva_scores_embeddings_tcga <span class="ot">&lt;-</span> GSVA<span class="sc">::</span><span class="fu">gsva</span>(</span>
<span id="cb10-812"><a href="#cb10-812" aria-hidden="true" tabindex="-1"></a>    df_pcs_regressed[, merged_col_data <span class="sc">%&gt;%</span></span>
<span id="cb10-813"><a href="#cb10-813" aria-hidden="true" tabindex="-1"></a>                         dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"tcga"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-814"><a href="#cb10-814" aria-hidden="true" tabindex="-1"></a>                         dplyr<span class="sc">::</span><span class="fu">pull</span>(sample_name)],</span>
<span id="cb10-815"><a href="#cb10-815" aria-hidden="true" tabindex="-1"></a>    gene_sets_, </span>
<span id="cb10-816"><a href="#cb10-816" aria-hidden="true" tabindex="-1"></a>    <span class="at">parallel.sz =</span> <span class="dv">10</span></span>
<span id="cb10-817"><a href="#cb10-817" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-818"><a href="#cb10-818" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-819"><a href="#cb10-819" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb10-820"><a href="#cb10-820" aria-hidden="true" tabindex="-1"></a>    gsva_scores_embeddings_tcga,</span>
<span id="cb10-821"><a href="#cb10-821" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/trying/gsva_scores_embeddings_tcga.rds"</span></span>
<span id="cb10-822"><a href="#cb10-822" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-823"><a href="#cb10-823" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-824"><a href="#cb10-824" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-825"><a href="#cb10-825" aria-hidden="true" tabindex="-1"></a>And we can now check and compare the scores.</span>
<span id="cb10-826"><a href="#cb10-826" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-827"><a href="#cb10-827" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=5}</span></span>
<span id="cb10-828"><a href="#cb10-828" aria-hidden="true" tabindex="-1"></a>pathways_of_interest <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb10-829"><a href="#cb10-829" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_ESTROGEN_RESPONSE_EARLY"</span>,</span>
<span id="cb10-830"><a href="#cb10-830" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_G2M_CHECKPOINT"</span>, </span>
<span id="cb10-831"><a href="#cb10-831" aria-hidden="true" tabindex="-1"></a>    <span class="st">"SET_ERPR"</span></span>
<span id="cb10-832"><a href="#cb10-832" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-833"><a href="#cb10-833" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-834"><a href="#cb10-834" aria-hidden="true" tabindex="-1"></a><span class="fu">sapply</span>(</span>
<span id="cb10-835"><a href="#cb10-835" aria-hidden="true" tabindex="-1"></a>    pathways_of_interest, </span>
<span id="cb10-836"><a href="#cb10-836" aria-hidden="true" tabindex="-1"></a>    get_merged_df_scores,</span>
<span id="cb10-837"><a href="#cb10-837" aria-hidden="true" tabindex="-1"></a>    <span class="at">df1 =</span> gsva_scores_embeddings_tcga,</span>
<span id="cb10-838"><a href="#cb10-838" aria-hidden="true" tabindex="-1"></a>    <span class="at">df2 =</span> df_pca,</span>
<span id="cb10-839"><a href="#cb10-839" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>, </span>
<span id="cb10-840"><a href="#cb10-840" aria-hidden="true" tabindex="-1"></a>    <span class="at">simplify =</span> <span class="cn">FALSE</span></span>
<span id="cb10-841"><a href="#cb10-841" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-842"><a href="#cb10-842" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(<span class="at">.id =</span> <span class="st">"pathway"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-843"><a href="#cb10-843" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> df1, <span class="at">y =</span> df2)) <span class="sc">+</span></span>
<span id="cb10-844"><a href="#cb10-844" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span> </span>
<span id="cb10-845"><a href="#cb10-845" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>pathway) <span class="sc">+</span></span>
<span id="cb10-846"><a href="#cb10-846" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_abline</span>(<span class="at">slope =</span> <span class="dv">1</span>, <span class="at">intercept =</span> <span class="dv">0</span>, <span class="at">color =</span> <span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb10-847"><a href="#cb10-847" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb10-848"><a href="#cb10-848" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"GSVA scores from regressed data"</span>,</span>
<span id="cb10-849"><a href="#cb10-849" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"GSVA scores from unregressed data"</span>,</span>
<span id="cb10-850"><a href="#cb10-850" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">"TCGA scores"</span></span>
<span id="cb10-851"><a href="#cb10-851" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb10-852"><a href="#cb10-852" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>()</span>
<span id="cb10-853"><a href="#cb10-853" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-854"><a href="#cb10-854" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-855"><a href="#cb10-855" aria-hidden="true" tabindex="-1"></a>We see that for the three pathways the correlation is positive and</span>
<span id="cb10-856"><a href="#cb10-856" aria-hidden="true" tabindex="-1"></a>strong. The difference is that it is a bit noisy, so the correlation</span>
<span id="cb10-857"><a href="#cb10-857" aria-hidden="true" tabindex="-1"></a>is not perfect. What is interesting to see is that at the extremes</span>
<span id="cb10-858"><a href="#cb10-858" aria-hidden="true" tabindex="-1"></a>the scores have less variation usually. </span>
<span id="cb10-859"><a href="#cb10-859" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-860"><a href="#cb10-860" aria-hidden="true" tabindex="-1"></a>We now calculate the scores from SCANB and compare the results. Remember</span>
<span id="cb10-861"><a href="#cb10-861" aria-hidden="true" tabindex="-1"></a>that SCANB was not used for the PCA training.</span>
<span id="cb10-862"><a href="#cb10-862" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-863"><a href="#cb10-863" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb10-864"><a href="#cb10-864" aria-hidden="true" tabindex="-1"></a>gsva_scores_embeddings_scanb <span class="ot">&lt;-</span> GSVA<span class="sc">::</span><span class="fu">gsva</span>(</span>
<span id="cb10-865"><a href="#cb10-865" aria-hidden="true" tabindex="-1"></a>    df_pcs_regressed[, merged_col_data <span class="sc">%&gt;%</span></span>
<span id="cb10-866"><a href="#cb10-866" aria-hidden="true" tabindex="-1"></a>                         dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"scanb"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-867"><a href="#cb10-867" aria-hidden="true" tabindex="-1"></a>                         dplyr<span class="sc">::</span><span class="fu">pull</span>(sample_name)],</span>
<span id="cb10-868"><a href="#cb10-868" aria-hidden="true" tabindex="-1"></a>    gene_sets_, </span>
<span id="cb10-869"><a href="#cb10-869" aria-hidden="true" tabindex="-1"></a>    <span class="at">parallel.sz =</span> <span class="dv">10</span></span>
<span id="cb10-870"><a href="#cb10-870" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-871"><a href="#cb10-871" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-872"><a href="#cb10-872" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb10-873"><a href="#cb10-873" aria-hidden="true" tabindex="-1"></a>    gsva_scores_embeddings_scanb,</span>
<span id="cb10-874"><a href="#cb10-874" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/trying/gsva_scores_embeddings_scanb.rds"</span></span>
<span id="cb10-875"><a href="#cb10-875" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-876"><a href="#cb10-876" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-877"><a href="#cb10-877" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-878"><a href="#cb10-878" aria-hidden="true" tabindex="-1"></a>And we can now check and compare the scores.</span>
<span id="cb10-879"><a href="#cb10-879" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-880"><a href="#cb10-880" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=5}</span></span>
<span id="cb10-881"><a href="#cb10-881" aria-hidden="true" tabindex="-1"></a><span class="fu">sapply</span>(</span>
<span id="cb10-882"><a href="#cb10-882" aria-hidden="true" tabindex="-1"></a>    pathways_of_interest, </span>
<span id="cb10-883"><a href="#cb10-883" aria-hidden="true" tabindex="-1"></a>    get_merged_df_scores,</span>
<span id="cb10-884"><a href="#cb10-884" aria-hidden="true" tabindex="-1"></a>    <span class="at">df1 =</span> gsva_scores_embeddings_scanb,</span>
<span id="cb10-885"><a href="#cb10-885" aria-hidden="true" tabindex="-1"></a>    <span class="at">df2 =</span> df_pca,</span>
<span id="cb10-886"><a href="#cb10-886" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>, </span>
<span id="cb10-887"><a href="#cb10-887" aria-hidden="true" tabindex="-1"></a>    <span class="at">simplify =</span> <span class="cn">FALSE</span></span>
<span id="cb10-888"><a href="#cb10-888" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-889"><a href="#cb10-889" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(<span class="at">.id =</span> <span class="st">"pathway"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-890"><a href="#cb10-890" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> df1, <span class="at">y =</span> df2)) <span class="sc">+</span></span>
<span id="cb10-891"><a href="#cb10-891" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span> </span>
<span id="cb10-892"><a href="#cb10-892" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>pathway) <span class="sc">+</span></span>
<span id="cb10-893"><a href="#cb10-893" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_abline</span>(<span class="at">slope =</span> <span class="dv">1</span>, <span class="at">intercept =</span> <span class="dv">0</span>, <span class="at">color =</span> <span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb10-894"><a href="#cb10-894" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb10-895"><a href="#cb10-895" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"GSVA scores from regressed data"</span>,</span>
<span id="cb10-896"><a href="#cb10-896" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"GSVA scores from unregressed data"</span>,</span>
<span id="cb10-897"><a href="#cb10-897" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">"TCGA scores"</span></span>
<span id="cb10-898"><a href="#cb10-898" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb10-899"><a href="#cb10-899" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>()</span>
<span id="cb10-900"><a href="#cb10-900" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-901"><a href="#cb10-901" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-902"><a href="#cb10-902" aria-hidden="true" tabindex="-1"></a>The same phenomenon happened. Samples in the endges have lower variability,</span>
<span id="cb10-903"><a href="#cb10-903" aria-hidden="true" tabindex="-1"></a>but samples in the middle are the ones that have high variability.</span>
<span id="cb10-904"><a href="#cb10-904" aria-hidden="true" tabindex="-1"></a>For example there are several samples from the regressed data</span>
<span id="cb10-905"><a href="#cb10-905" aria-hidden="true" tabindex="-1"></a>with SET_ERPR score close to 0.25 and negative scores.</span>
<span id="cb10-906"><a href="#cb10-906" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-907"><a href="#cb10-907" aria-hidden="true" tabindex="-1"></a>What happens now if we use another dataset and then use only one sample to </span>
<span id="cb10-908"><a href="#cb10-908" aria-hidden="true" tabindex="-1"></a>calculate the score. Let us include SCANB samples in the pipeline. For this </span>
<span id="cb10-909"><a href="#cb10-909" aria-hidden="true" tabindex="-1"></a>we select 50 scanb samples and 100 random samples from tcga and metabric.</span>
<span id="cb10-910"><a href="#cb10-910" aria-hidden="true" tabindex="-1"></a>We then combine all these samples together and calculate the GSVA scores.</span>
<span id="cb10-911"><a href="#cb10-911" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-912"><a href="#cb10-912" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb10-913"><a href="#cb10-913" aria-hidden="true" tabindex="-1"></a>scanb_samples <span class="ot">&lt;-</span> df_pca <span class="sc">%&gt;%</span></span>
<span id="cb10-914"><a href="#cb10-914" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"scanb"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-915"><a href="#cb10-915" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">slice_sample</span>(<span class="at">n =</span> <span class="dv">50</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-916"><a href="#cb10-916" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">pull</span>(sample_name)</span>
<span id="cb10-917"><a href="#cb10-917" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-918"><a href="#cb10-918" aria-hidden="true" tabindex="-1"></a>gsva_scores_embeddings_scanb_within_tcga <span class="ot">&lt;-</span> parallel<span class="sc">::</span><span class="fu">mclapply</span>(</span>
<span id="cb10-919"><a href="#cb10-919" aria-hidden="true" tabindex="-1"></a>    scanb_samples,</span>
<span id="cb10-920"><a href="#cb10-920" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(x){</span>
<span id="cb10-921"><a href="#cb10-921" aria-hidden="true" tabindex="-1"></a>        GSVA<span class="sc">::</span><span class="fu">gsva</span>(</span>
<span id="cb10-922"><a href="#cb10-922" aria-hidden="true" tabindex="-1"></a>            df_pcs_regressed[</span>
<span id="cb10-923"><a href="#cb10-923" aria-hidden="true" tabindex="-1"></a>                , </span>
<span id="cb10-924"><a href="#cb10-924" aria-hidden="true" tabindex="-1"></a>                merged_col_data <span class="sc">%&gt;%</span></span>
<span id="cb10-925"><a href="#cb10-925" aria-hidden="true" tabindex="-1"></a>                    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"tcga"</span>, <span class="st">"metabric"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb10-926"><a href="#cb10-926" aria-hidden="true" tabindex="-1"></a>                    dplyr<span class="sc">::</span><span class="fu">slice_sample</span>(<span class="at">n =</span> <span class="dv">100</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-927"><a href="#cb10-927" aria-hidden="true" tabindex="-1"></a>                    dplyr<span class="sc">::</span><span class="fu">pull</span>(sample_name) <span class="sc">%&gt;%</span></span>
<span id="cb10-928"><a href="#cb10-928" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">c</span>(., x)</span>
<span id="cb10-929"><a href="#cb10-929" aria-hidden="true" tabindex="-1"></a>            ],</span>
<span id="cb10-930"><a href="#cb10-930" aria-hidden="true" tabindex="-1"></a>            gene_sets_</span>
<span id="cb10-931"><a href="#cb10-931" aria-hidden="true" tabindex="-1"></a>        )   </span>
<span id="cb10-932"><a href="#cb10-932" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb10-933"><a href="#cb10-933" aria-hidden="true" tabindex="-1"></a>    <span class="at">mc.cores =</span> <span class="dv">10</span></span>
<span id="cb10-934"><a href="#cb10-934" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-935"><a href="#cb10-935" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-936"><a href="#cb10-936" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb10-937"><a href="#cb10-937" aria-hidden="true" tabindex="-1"></a>    gsva_scores_embeddings_scanb_within_tcga,</span>
<span id="cb10-938"><a href="#cb10-938" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/trying/gsva_scores_embeddings_scanb_within_tcga.rds"</span>    </span>
<span id="cb10-939"><a href="#cb10-939" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-940"><a href="#cb10-940" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-941"><a href="#cb10-941" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-942"><a href="#cb10-942" aria-hidden="true" tabindex="-1"></a>We now plot only the scores from the SCANB samples.</span>
<span id="cb10-943"><a href="#cb10-943" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-944"><a href="#cb10-944" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=5}</span></span>
<span id="cb10-945"><a href="#cb10-945" aria-hidden="true" tabindex="-1"></a><span class="fu">lapply</span>(</span>
<span id="cb10-946"><a href="#cb10-946" aria-hidden="true" tabindex="-1"></a>    gsva_scores_embeddings_scanb_within_tcga,</span>
<span id="cb10-947"><a href="#cb10-947" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(x){</span>
<span id="cb10-948"><a href="#cb10-948" aria-hidden="true" tabindex="-1"></a>        <span class="fu">sapply</span>(</span>
<span id="cb10-949"><a href="#cb10-949" aria-hidden="true" tabindex="-1"></a>            pathways_of_interest, </span>
<span id="cb10-950"><a href="#cb10-950" aria-hidden="true" tabindex="-1"></a>            get_merged_df_scores,</span>
<span id="cb10-951"><a href="#cb10-951" aria-hidden="true" tabindex="-1"></a>            <span class="at">df1 =</span> x,</span>
<span id="cb10-952"><a href="#cb10-952" aria-hidden="true" tabindex="-1"></a>            <span class="at">df2 =</span> df_pca <span class="sc">%&gt;%</span></span>
<span id="cb10-953"><a href="#cb10-953" aria-hidden="true" tabindex="-1"></a>                dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"scanb"</span>),</span>
<span id="cb10-954"><a href="#cb10-954" aria-hidden="true" tabindex="-1"></a>            <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>, </span>
<span id="cb10-955"><a href="#cb10-955" aria-hidden="true" tabindex="-1"></a>            <span class="at">simplify =</span> <span class="cn">FALSE</span></span>
<span id="cb10-956"><a href="#cb10-956" aria-hidden="true" tabindex="-1"></a>        ) <span class="sc">%&gt;%</span> </span>
<span id="cb10-957"><a href="#cb10-957" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(<span class="at">.id =</span> <span class="st">"pathway"</span>)</span>
<span id="cb10-958"><a href="#cb10-958" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb10-959"><a href="#cb10-959" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb10-960"><a href="#cb10-960" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(<span class="at">.id =</span> <span class="st">"sample_name"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-961"><a href="#cb10-961" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> df1, <span class="at">y =</span> df2)) <span class="sc">+</span></span>
<span id="cb10-962"><a href="#cb10-962" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span> </span>
<span id="cb10-963"><a href="#cb10-963" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>pathway) <span class="sc">+</span></span>
<span id="cb10-964"><a href="#cb10-964" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_abline</span>(<span class="at">slope =</span> <span class="dv">1</span>, <span class="at">intercept =</span> <span class="dv">0</span>, <span class="at">color =</span> <span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb10-965"><a href="#cb10-965" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb10-966"><a href="#cb10-966" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"GSVA scores from regressed data"</span>,</span>
<span id="cb10-967"><a href="#cb10-967" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"GSVA scores from unregressed data"</span>,</span>
<span id="cb10-968"><a href="#cb10-968" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">"TCGA scores"</span></span>
<span id="cb10-969"><a href="#cb10-969" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb10-970"><a href="#cb10-970" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>()</span>
<span id="cb10-971"><a href="#cb10-971" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-972"><a href="#cb10-972" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-973"><a href="#cb10-973" aria-hidden="true" tabindex="-1"></a>So it seems there are still dataset related problems there when calculating</span>
<span id="cb10-974"><a href="#cb10-974" aria-hidden="true" tabindex="-1"></a>the scores. The scores are positively correlated, but they are not quite</span>
<span id="cb10-975"><a href="#cb10-975" aria-hidden="true" tabindex="-1"></a>the same as only using the SCANB dataset. Moreover, for G2M checkpoint the </span>
<span id="cb10-976"><a href="#cb10-976" aria-hidden="true" tabindex="-1"></a>scores are concentrated around -0.4 from the regressed data but </span>
<span id="cb10-977"><a href="#cb10-977" aria-hidden="true" tabindex="-1"></a>they are in a range going from - 0.4 to 0.1 in the unregressed data. </span>
<span id="cb10-978"><a href="#cb10-978" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-979"><a href="#cb10-979" aria-hidden="true" tabindex="-1"></a>If we try to instead add some samples from the SCANB dataset as well </span>
<span id="cb10-980"><a href="#cb10-980" aria-hidden="true" tabindex="-1"></a>randomly into the mix. How well does the scores get compared to the </span>
<span id="cb10-981"><a href="#cb10-981" aria-hidden="true" tabindex="-1"></a>"ground truth"? </span>
<span id="cb10-982"><a href="#cb10-982" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-983"><a href="#cb10-983" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb10-984"><a href="#cb10-984" aria-hidden="true" tabindex="-1"></a>scanb_samples <span class="ot">&lt;-</span> df_pca <span class="sc">%&gt;%</span></span>
<span id="cb10-985"><a href="#cb10-985" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"scanb"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-986"><a href="#cb10-986" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">slice_sample</span>(<span class="at">n =</span> <span class="dv">30</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-987"><a href="#cb10-987" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">pull</span>(sample_name)</span>
<span id="cb10-988"><a href="#cb10-988" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-989"><a href="#cb10-989" aria-hidden="true" tabindex="-1"></a>gsva_scores_embeddings_scanb_within_tcga_scanb <span class="ot">&lt;-</span> parallel<span class="sc">::</span><span class="fu">mclapply</span>(</span>
<span id="cb10-990"><a href="#cb10-990" aria-hidden="true" tabindex="-1"></a>    scanb_samples,</span>
<span id="cb10-991"><a href="#cb10-991" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(x){</span>
<span id="cb10-992"><a href="#cb10-992" aria-hidden="true" tabindex="-1"></a>        GSVA<span class="sc">::</span><span class="fu">gsva</span>(</span>
<span id="cb10-993"><a href="#cb10-993" aria-hidden="true" tabindex="-1"></a>            df_pcs_regressed[, merged_col_data <span class="sc">%&gt;%</span></span>
<span id="cb10-994"><a href="#cb10-994" aria-hidden="true" tabindex="-1"></a>                                dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb10-995"><a href="#cb10-995" aria-hidden="true" tabindex="-1"></a>                                    cohort <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"tcga"</span>, <span class="st">"metabric"</span>, <span class="st">"scanb"</span>)</span>
<span id="cb10-996"><a href="#cb10-996" aria-hidden="true" tabindex="-1"></a>                                ) <span class="sc">%&gt;%</span></span>
<span id="cb10-997"><a href="#cb10-997" aria-hidden="true" tabindex="-1"></a>                                dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">weights =</span> dplyr<span class="sc">::</span><span class="fu">case_when</span>(</span>
<span id="cb10-998"><a href="#cb10-998" aria-hidden="true" tabindex="-1"></a>                                    cohort <span class="sc">==</span> <span class="st">"tcga"</span> <span class="sc">~</span> <span class="fl">0.45</span>,</span>
<span id="cb10-999"><a href="#cb10-999" aria-hidden="true" tabindex="-1"></a>                                    cohort <span class="sc">==</span> <span class="st">"metabric"</span> <span class="sc">~</span> <span class="fl">0.45</span>,</span>
<span id="cb10-1000"><a href="#cb10-1000" aria-hidden="true" tabindex="-1"></a>                                    cohort <span class="sc">==</span> <span class="st">"scanb"</span> <span class="sc">~</span> <span class="fl">0.1</span></span>
<span id="cb10-1001"><a href="#cb10-1001" aria-hidden="true" tabindex="-1"></a>                                )) <span class="sc">%&gt;%</span></span>
<span id="cb10-1002"><a href="#cb10-1002" aria-hidden="true" tabindex="-1"></a>                                dplyr<span class="sc">::</span><span class="fu">slice_sample</span>(<span class="at">n =</span> <span class="dv">150</span>, <span class="at">weight_by =</span> weights) <span class="sc">%&gt;%</span></span>
<span id="cb10-1003"><a href="#cb10-1003" aria-hidden="true" tabindex="-1"></a>                                dplyr<span class="sc">::</span><span class="fu">pull</span>(sample_name) <span class="sc">%&gt;%</span></span>
<span id="cb10-1004"><a href="#cb10-1004" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">c</span>(., x)],</span>
<span id="cb10-1005"><a href="#cb10-1005" aria-hidden="true" tabindex="-1"></a>            gene_sets_</span>
<span id="cb10-1006"><a href="#cb10-1006" aria-hidden="true" tabindex="-1"></a>        )   </span>
<span id="cb10-1007"><a href="#cb10-1007" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb10-1008"><a href="#cb10-1008" aria-hidden="true" tabindex="-1"></a>    <span class="at">mc.cores =</span> <span class="dv">20</span></span>
<span id="cb10-1009"><a href="#cb10-1009" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-1010"><a href="#cb10-1010" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1011"><a href="#cb10-1011" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb10-1012"><a href="#cb10-1012" aria-hidden="true" tabindex="-1"></a>    gsva_scores_embeddings_scanb_within_tcga_scanb,</span>
<span id="cb10-1013"><a href="#cb10-1013" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/trying/gsva_scores_embeddings_scanb_within_tcga_scanb.rds"</span></span>
<span id="cb10-1014"><a href="#cb10-1014" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-1015"><a href="#cb10-1015" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-1016"><a href="#cb10-1016" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1017"><a href="#cb10-1017" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=5}</span></span>
<span id="cb10-1018"><a href="#cb10-1018" aria-hidden="true" tabindex="-1"></a><span class="fu">lapply</span>(</span>
<span id="cb10-1019"><a href="#cb10-1019" aria-hidden="true" tabindex="-1"></a>    gsva_scores_embeddings_scanb_within_tcga_scanb,</span>
<span id="cb10-1020"><a href="#cb10-1020" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(x){</span>
<span id="cb10-1021"><a href="#cb10-1021" aria-hidden="true" tabindex="-1"></a>        <span class="fu">sapply</span>(</span>
<span id="cb10-1022"><a href="#cb10-1022" aria-hidden="true" tabindex="-1"></a>            pathways_of_interest, </span>
<span id="cb10-1023"><a href="#cb10-1023" aria-hidden="true" tabindex="-1"></a>            get_merged_df_scores,</span>
<span id="cb10-1024"><a href="#cb10-1024" aria-hidden="true" tabindex="-1"></a>            <span class="at">df1 =</span> x,</span>
<span id="cb10-1025"><a href="#cb10-1025" aria-hidden="true" tabindex="-1"></a>            <span class="at">df2 =</span> df_pca <span class="sc">%&gt;%</span></span>
<span id="cb10-1026"><a href="#cb10-1026" aria-hidden="true" tabindex="-1"></a>                dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"scanb"</span>),</span>
<span id="cb10-1027"><a href="#cb10-1027" aria-hidden="true" tabindex="-1"></a>            <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>, </span>
<span id="cb10-1028"><a href="#cb10-1028" aria-hidden="true" tabindex="-1"></a>            <span class="at">simplify =</span> <span class="cn">FALSE</span></span>
<span id="cb10-1029"><a href="#cb10-1029" aria-hidden="true" tabindex="-1"></a>        ) <span class="sc">%&gt;%</span> </span>
<span id="cb10-1030"><a href="#cb10-1030" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(<span class="at">.id =</span> <span class="st">"pathway"</span>)</span>
<span id="cb10-1031"><a href="#cb10-1031" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb10-1032"><a href="#cb10-1032" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb10-1033"><a href="#cb10-1033" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(<span class="at">.id =</span> <span class="st">"sample_name"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-1034"><a href="#cb10-1034" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> df1, <span class="at">y =</span> df2)) <span class="sc">+</span></span>
<span id="cb10-1035"><a href="#cb10-1035" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span> </span>
<span id="cb10-1036"><a href="#cb10-1036" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>pathway) <span class="sc">+</span></span>
<span id="cb10-1037"><a href="#cb10-1037" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_abline</span>(<span class="at">slope =</span> <span class="dv">1</span>, <span class="at">intercept =</span> <span class="dv">0</span>, <span class="at">color =</span> <span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb10-1038"><a href="#cb10-1038" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb10-1039"><a href="#cb10-1039" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"GSVA scores from regressed data"</span>,</span>
<span id="cb10-1040"><a href="#cb10-1040" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"GSVA scores from unregressed data"</span>,</span>
<span id="cb10-1041"><a href="#cb10-1041" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">"TCGA scores"</span></span>
<span id="cb10-1042"><a href="#cb10-1042" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb10-1043"><a href="#cb10-1043" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>()</span>
<span id="cb10-1044"><a href="#cb10-1044" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-1045"><a href="#cb10-1045" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1046"><a href="#cb10-1046" aria-hidden="true" tabindex="-1"></a>It does not work so well when including some SCANB samples together </span>
<span id="cb10-1047"><a href="#cb10-1047" aria-hidden="true" tabindex="-1"></a>with the METABRIC and TCGA samples. We are trying to achieve a</span>
<span id="cb10-1048"><a href="#cb10-1048" aria-hidden="true" tabindex="-1"></a>correlation of almost 1 with very low variability since we want to use</span>
<span id="cb10-1049"><a href="#cb10-1049" aria-hidden="true" tabindex="-1"></a>the scores clinically. </span>
<span id="cb10-1050"><a href="#cb10-1050" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1051"><a href="#cb10-1051" aria-hidden="true" tabindex="-1"></a>Let us check the distribution of some genes for SCANB and TCGA after</span>
<span id="cb10-1052"><a href="#cb10-1052" aria-hidden="true" tabindex="-1"></a>regressing out the first two PCs.</span>
<span id="cb10-1053"><a href="#cb10-1053" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1054"><a href="#cb10-1054" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=8}</span></span>
<span id="cb10-1055"><a href="#cb10-1055" aria-hidden="true" tabindex="-1"></a>genes_histogram <span class="ot">&lt;-</span> gene_sets_<span class="sc">$</span>HALLMARK_G2M_CHECKPOINT</span>
<span id="cb10-1056"><a href="#cb10-1056" aria-hidden="true" tabindex="-1"></a>genes_histogram <span class="ot">&lt;-</span> <span class="fu">intersect</span>(genes_histogram, <span class="fu">rownames</span>(df_pcs_regressed))</span>
<span id="cb10-1057"><a href="#cb10-1057" aria-hidden="true" tabindex="-1"></a>genes_histogram <span class="ot">&lt;-</span> <span class="fu">sample</span>(genes_histogram, <span class="dv">30</span>)</span>
<span id="cb10-1058"><a href="#cb10-1058" aria-hidden="true" tabindex="-1"></a>genes_histogram <span class="ot">&lt;-</span> <span class="fu">c</span>(genes_histogram, <span class="st">"MKI67"</span>) <span class="sc">%&gt;%</span> unique</span>
<span id="cb10-1059"><a href="#cb10-1059" aria-hidden="true" tabindex="-1"></a>df_pcs_regressed <span class="sc">%&gt;%</span> </span>
<span id="cb10-1060"><a href="#cb10-1060" aria-hidden="true" tabindex="-1"></a>    t <span class="sc">%&gt;%</span></span>
<span id="cb10-1061"><a href="#cb10-1061" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span></span>
<span id="cb10-1062"><a href="#cb10-1062" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(dplyr<span class="sc">::</span><span class="fu">all_of</span>(genes_histogram)) <span class="sc">%&gt;%</span></span>
<span id="cb10-1063"><a href="#cb10-1063" aria-hidden="true" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"sample_name"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-1064"><a href="#cb10-1064" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">inner_join</span>(</span>
<span id="cb10-1065"><a href="#cb10-1065" aria-hidden="true" tabindex="-1"></a>        .,</span>
<span id="cb10-1066"><a href="#cb10-1066" aria-hidden="true" tabindex="-1"></a>        merged_col_data,</span>
<span id="cb10-1067"><a href="#cb10-1067" aria-hidden="true" tabindex="-1"></a>        <span class="at">by =</span> <span class="st">"sample_name"</span></span>
<span id="cb10-1068"><a href="#cb10-1068" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb10-1069"><a href="#cb10-1069" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(</span>
<span id="cb10-1070"><a href="#cb10-1070" aria-hidden="true" tabindex="-1"></a>        <span class="at">cols =</span> dplyr<span class="sc">::</span><span class="fu">all_of</span>(genes_histogram),</span>
<span id="cb10-1071"><a href="#cb10-1071" aria-hidden="true" tabindex="-1"></a>        <span class="at">values_to =</span> <span class="st">"expression"</span>,</span>
<span id="cb10-1072"><a href="#cb10-1072" aria-hidden="true" tabindex="-1"></a>        <span class="at">names_to =</span> <span class="st">"gene"</span></span>
<span id="cb10-1073"><a href="#cb10-1073" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span> </span>
<span id="cb10-1074"><a href="#cb10-1074" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(</span>
<span id="cb10-1075"><a href="#cb10-1075" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> expression,</span>
<span id="cb10-1076"><a href="#cb10-1076" aria-hidden="true" tabindex="-1"></a>        <span class="at">fill =</span> cohort,</span>
<span id="cb10-1077"><a href="#cb10-1077" aria-hidden="true" tabindex="-1"></a>        <span class="at">alpha =</span> <span class="fl">0.5</span></span>
<span id="cb10-1078"><a href="#cb10-1078" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">+</span> </span>
<span id="cb10-1079"><a href="#cb10-1079" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_density</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb10-1080"><a href="#cb10-1080" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>gene, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span> </span>
<span id="cb10-1081"><a href="#cb10-1081" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>()</span>
<span id="cb10-1082"><a href="#cb10-1082" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-1083"><a href="#cb10-1083" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1084"><a href="#cb10-1084" aria-hidden="true" tabindex="-1"></a>We see that even though the expression profile of the samples are </span>
<span id="cb10-1085"><a href="#cb10-1085" aria-hidden="true" tabindex="-1"></a>overlapping there are discrepancies. That is why when using GSVA </span>
<span id="cb10-1086"><a href="#cb10-1086" aria-hidden="true" tabindex="-1"></a>it does not work so well perhaps.</span>
<span id="cb10-1087"><a href="#cb10-1087" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1088"><a href="#cb10-1088" aria-hidden="true" tabindex="-1"></a><span class="fu">## Scoring strategy with regressed data</span></span>
<span id="cb10-1089"><a href="#cb10-1089" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1090"><a href="#cb10-1090" aria-hidden="true" tabindex="-1"></a>We now try to use a slightly different scoring strategy. We can think of </span>
<span id="cb10-1091"><a href="#cb10-1091" aria-hidden="true" tabindex="-1"></a>our data in a log scale, so what we do now instead is to sum all of </span>
<span id="cb10-1092"><a href="#cb10-1092" aria-hidden="true" tabindex="-1"></a>our genes of interest and then subtract an average of the housekeeping</span>
<span id="cb10-1093"><a href="#cb10-1093" aria-hidden="true" tabindex="-1"></a>genes expression levels.</span>
<span id="cb10-1094"><a href="#cb10-1094" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1095"><a href="#cb10-1095" aria-hidden="true" tabindex="-1"></a>First we check the distribution of this average for each cohort</span>
<span id="cb10-1096"><a href="#cb10-1096" aria-hidden="true" tabindex="-1"></a>separately.</span>
<span id="cb10-1097"><a href="#cb10-1097" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1100"><a href="#cb10-1100" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-1101"><a href="#cb10-1101" aria-hidden="true" tabindex="-1"></a>hk_genes <span class="ot">&lt;-</span> singscore<span class="sc">::</span><span class="fu">getStableGenes</span>(</span>
<span id="cb10-1102"><a href="#cb10-1102" aria-hidden="true" tabindex="-1"></a>    <span class="dv">50</span>, </span>
<span id="cb10-1103"><a href="#cb10-1103" aria-hidden="true" tabindex="-1"></a>    <span class="at">type =</span> <span class="st">"carcinoma"</span>, </span>
<span id="cb10-1104"><a href="#cb10-1104" aria-hidden="true" tabindex="-1"></a>    <span class="at">id =</span> <span class="st">"geneid"</span></span>
<span id="cb10-1105"><a href="#cb10-1105" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-1106"><a href="#cb10-1106" aria-hidden="true" tabindex="-1"></a>    <span class="fu">intersect</span>(., <span class="fu">rownames</span>(df_pcs_regressed))</span>
<span id="cb10-1107"><a href="#cb10-1107" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1108"><a href="#cb10-1108" aria-hidden="true" tabindex="-1"></a>avg_hkg_pcs_regressed <span class="ot">&lt;-</span> df_pcs_regressed <span class="sc">%&gt;%</span> </span>
<span id="cb10-1109"><a href="#cb10-1109" aria-hidden="true" tabindex="-1"></a>    t <span class="sc">%&gt;%</span></span>
<span id="cb10-1110"><a href="#cb10-1110" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span></span>
<span id="cb10-1111"><a href="#cb10-1111" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(dplyr<span class="sc">::</span><span class="fu">all_of</span>(hk_genes)) <span class="sc">%&gt;%</span></span>
<span id="cb10-1112"><a href="#cb10-1112" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">avg_hkg =</span> <span class="fu">rowMeans</span>(dplyr<span class="sc">::</span><span class="fu">select</span>(</span>
<span id="cb10-1113"><a href="#cb10-1113" aria-hidden="true" tabindex="-1"></a>        ., </span>
<span id="cb10-1114"><a href="#cb10-1114" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">all_of</span>(hk_genes)</span>
<span id="cb10-1115"><a href="#cb10-1115" aria-hidden="true" tabindex="-1"></a>    ))) <span class="sc">%&gt;%</span></span>
<span id="cb10-1116"><a href="#cb10-1116" aria-hidden="true" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"sample_name"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-1117"><a href="#cb10-1117" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">inner_join</span>(</span>
<span id="cb10-1118"><a href="#cb10-1118" aria-hidden="true" tabindex="-1"></a>        .,</span>
<span id="cb10-1119"><a href="#cb10-1119" aria-hidden="true" tabindex="-1"></a>        merged_col_data,</span>
<span id="cb10-1120"><a href="#cb10-1120" aria-hidden="true" tabindex="-1"></a>        <span class="at">by =</span> <span class="st">"sample_name"</span></span>
<span id="cb10-1121"><a href="#cb10-1121" aria-hidden="true" tabindex="-1"></a>    ) </span>
<span id="cb10-1122"><a href="#cb10-1122" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-1123"><a href="#cb10-1123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1124"><a href="#cb10-1124" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width = 8, fig.height=6}</span></span>
<span id="cb10-1125"><a href="#cb10-1125" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-hkg-regressed</span></span>
<span id="cb10-1126"><a href="#cb10-1126" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Distribution of the average of the housekeeping genes</span></span>
<span id="cb10-1127"><a href="#cb10-1127" aria-hidden="true" tabindex="-1"></a><span class="co">#|     for distinct cohorts. </span></span>
<span id="cb10-1128"><a href="#cb10-1128" aria-hidden="true" tabindex="-1"></a>avg_hkg_pcs_regressed <span class="sc">%&gt;%</span> </span>
<span id="cb10-1129"><a href="#cb10-1129" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(</span>
<span id="cb10-1130"><a href="#cb10-1130" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> avg_hkg,</span>
<span id="cb10-1131"><a href="#cb10-1131" aria-hidden="true" tabindex="-1"></a>        <span class="at">fill =</span> cohort,</span>
<span id="cb10-1132"><a href="#cb10-1132" aria-hidden="true" tabindex="-1"></a>        <span class="at">alpha =</span> <span class="fl">0.5</span></span>
<span id="cb10-1133"><a href="#cb10-1133" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">+</span> </span>
<span id="cb10-1134"><a href="#cb10-1134" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_density</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb10-1135"><a href="#cb10-1135" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb10-1136"><a href="#cb10-1136" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"Average of housekeeping genes on regressed data"</span>,</span>
<span id="cb10-1137"><a href="#cb10-1137" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"Density"</span></span>
<span id="cb10-1138"><a href="#cb10-1138" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb10-1139"><a href="#cb10-1139" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">30</span>) <span class="sc">+</span></span>
<span id="cb10-1140"><a href="#cb10-1140" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme</span>(</span>
<span id="cb10-1141"><a href="#cb10-1141" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="fu">c</span>(<span class="fl">0.8</span>, <span class="fl">0.6</span>),</span>
<span id="cb10-1142"><a href="#cb10-1142" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.background =</span> <span class="fu">element_blank</span>()</span>
<span id="cb10-1143"><a href="#cb10-1143" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb10-1144"><a href="#cb10-1144" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>() <span class="sc">+</span></span>
<span id="cb10-1145"><a href="#cb10-1145" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>()</span>
<span id="cb10-1146"><a href="#cb10-1146" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-1147"><a href="#cb10-1147" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1148"><a href="#cb10-1148" aria-hidden="true" tabindex="-1"></a>Indeed the tcga and metabric cohorts have an overlapping </span>
<span id="cb10-1149"><a href="#cb10-1149" aria-hidden="true" tabindex="-1"></a>distribution, but the SCANB does not overlap completely, it is</span>
<span id="cb10-1150"><a href="#cb10-1150" aria-hidden="true" tabindex="-1"></a>actually shifted. One should still take into account that </span>
<span id="cb10-1151"><a href="#cb10-1151" aria-hidden="true" tabindex="-1"></a>the distributions are very close to 0, meaning that the </span>
<span id="cb10-1152"><a href="#cb10-1152" aria-hidden="true" tabindex="-1"></a>housekeeping genes have very close to 0 values, as expected here in this case.</span>
<span id="cb10-1153"><a href="#cb10-1153" aria-hidden="true" tabindex="-1"></a>We now calculate the score by taking this into account. </span>
<span id="cb10-1154"><a href="#cb10-1154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1157"><a href="#cb10-1157" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-1158"><a href="#cb10-1158" aria-hidden="true" tabindex="-1"></a>pathways_of_interest <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb10-1159"><a href="#cb10-1159" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_ESTROGEN_RESPONSE_EARLY"</span> <span class="ot">=</span> <span class="st">"Estrogen Early"</span>,</span>
<span id="cb10-1160"><a href="#cb10-1160" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_G2M_CHECKPOINT"</span> <span class="ot">=</span> <span class="st">"G2M"</span>, </span>
<span id="cb10-1161"><a href="#cb10-1161" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_E2F_TARGETS"</span> <span class="ot">=</span> <span class="st">"E2F"</span>, </span>
<span id="cb10-1162"><a href="#cb10-1162" aria-hidden="true" tabindex="-1"></a>    <span class="st">"SET_ERPR"</span> <span class="ot">=</span> <span class="st">"SET ER/PR"</span>,</span>
<span id="cb10-1163"><a href="#cb10-1163" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION"</span> <span class="ot">=</span> <span class="st">"EMT"</span>,</span>
<span id="cb10-1164"><a href="#cb10-1164" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_PI3K_AKT_MTOR_SIGNALING"</span> <span class="ot">=</span> <span class="st">"PI3K AKT MTOR"</span>,</span>
<span id="cb10-1165"><a href="#cb10-1165" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_DNA_REPAIR"</span> <span class="ot">=</span> <span class="st">"DNA Repair"</span> ,</span>
<span id="cb10-1166"><a href="#cb10-1166" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_INTERFERON_GAMMA_RESPONSE"</span> <span class="ot">=</span> <span class="st">"Interferon Gamma"</span>,</span>
<span id="cb10-1167"><a href="#cb10-1167" aria-hidden="true" tabindex="-1"></a>    <span class="st">"random_200"</span> <span class="ot">=</span> <span class="st">"Random 200"</span></span>
<span id="cb10-1168"><a href="#cb10-1168" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-1169"><a href="#cb10-1169" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-1170"><a href="#cb10-1170" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1171"><a href="#cb10-1171" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb10-1172"><a href="#cb10-1172" aria-hidden="true" tabindex="-1"></a>scores <span class="ot">&lt;-</span> <span class="fu">sapply</span>(</span>
<span id="cb10-1173"><a href="#cb10-1173" aria-hidden="true" tabindex="-1"></a>    gene_sets_[<span class="fu">names</span>(pathways_of_interest)],</span>
<span id="cb10-1174"><a href="#cb10-1174" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(gene_set, df_pcs_regressed, avg_hkg_pcs_regressed){</span>
<span id="cb10-1175"><a href="#cb10-1175" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb10-1176"><a href="#cb10-1176" aria-hidden="true" tabindex="-1"></a>        proportion <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb10-1177"><a href="#cb10-1177" aria-hidden="true" tabindex="-1"></a>        nb_genes <span class="ot">&lt;-</span> <span class="fu">floor</span>(proportion <span class="sc">*</span> <span class="fu">length</span>(gene_set))</span>
<span id="cb10-1178"><a href="#cb10-1178" aria-hidden="true" tabindex="-1"></a>        gene_set <span class="ot">&lt;-</span> <span class="fu">intersect</span>(</span>
<span id="cb10-1179"><a href="#cb10-1179" aria-hidden="true" tabindex="-1"></a>            <span class="fu">sample</span>(gene_set, nb_genes),</span>
<span id="cb10-1180"><a href="#cb10-1180" aria-hidden="true" tabindex="-1"></a>            <span class="fu">rownames</span>(df_pcs_regressed)</span>
<span id="cb10-1181"><a href="#cb10-1181" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb10-1182"><a href="#cb10-1182" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb10-1183"><a href="#cb10-1183" aria-hidden="true" tabindex="-1"></a>        <span class="co"># the code below was tried and it didnt work</span></span>
<span id="cb10-1184"><a href="#cb10-1184" aria-hidden="true" tabindex="-1"></a>        <span class="co"># 1) Mean of the genes and subtract the average of housekeeping genes</span></span>
<span id="cb10-1185"><a href="#cb10-1185" aria-hidden="true" tabindex="-1"></a>        <span class="co"># colMeans(df_pcs_regressed[gene_set, ]) - avg_hkg_pcs_regressed$avg_hkg</span></span>
<span id="cb10-1186"><a href="#cb10-1186" aria-hidden="true" tabindex="-1"></a>        <span class="co"># 2) top quantile. this one works surprisingly well considering </span></span>
<span id="cb10-1187"><a href="#cb10-1187" aria-hidden="true" tabindex="-1"></a>        <span class="co"># what was done. It could be further developped</span></span>
<span id="cb10-1188"><a href="#cb10-1188" aria-hidden="true" tabindex="-1"></a>        <span class="co"># apply(</span></span>
<span id="cb10-1189"><a href="#cb10-1189" aria-hidden="true" tabindex="-1"></a>        <span class="co">#     X = df_pcs_regressed[gene_set, ], </span></span>
<span id="cb10-1190"><a href="#cb10-1190" aria-hidden="true" tabindex="-1"></a>        <span class="co">#     MARGIN = 2,</span></span>
<span id="cb10-1191"><a href="#cb10-1191" aria-hidden="true" tabindex="-1"></a>        <span class="co">#     quantile,</span></span>
<span id="cb10-1192"><a href="#cb10-1192" aria-hidden="true" tabindex="-1"></a>        <span class="co">#     probs = 0.75</span></span>
<span id="cb10-1193"><a href="#cb10-1193" aria-hidden="true" tabindex="-1"></a>        <span class="co"># )</span></span>
<span id="cb10-1194"><a href="#cb10-1194" aria-hidden="true" tabindex="-1"></a>        <span class="co"># 3) pairwise differences of all genes. Small differences means genes</span></span>
<span id="cb10-1195"><a href="#cb10-1195" aria-hidden="true" tabindex="-1"></a>        <span class="co"># are going in the same direction. Big differences means genes are</span></span>
<span id="cb10-1196"><a href="#cb10-1196" aria-hidden="true" tabindex="-1"></a>        <span class="co"># going in opposite directions.</span></span>
<span id="cb10-1197"><a href="#cb10-1197" aria-hidden="true" tabindex="-1"></a>        <span class="co"># apply(</span></span>
<span id="cb10-1198"><a href="#cb10-1198" aria-hidden="true" tabindex="-1"></a>        <span class="co">#     X = df_pcs_regressed[gene_set, ],</span></span>
<span id="cb10-1199"><a href="#cb10-1199" aria-hidden="true" tabindex="-1"></a>        <span class="co">#     MARGIN = 2,</span></span>
<span id="cb10-1200"><a href="#cb10-1200" aria-hidden="true" tabindex="-1"></a>        <span class="co">#     function(v){</span></span>
<span id="cb10-1201"><a href="#cb10-1201" aria-hidden="true" tabindex="-1"></a>        <span class="co">#         quantile(as.vector(dist(v)), probs = 0.95)</span></span>
<span id="cb10-1202"><a href="#cb10-1202" aria-hidden="true" tabindex="-1"></a>        <span class="co">#     }</span></span>
<span id="cb10-1203"><a href="#cb10-1203" aria-hidden="true" tabindex="-1"></a>        <span class="co"># )</span></span>
<span id="cb10-1204"><a href="#cb10-1204" aria-hidden="true" tabindex="-1"></a>        <span class="co"># 4) Scale all genes for each sample and then take the average of </span></span>
<span id="cb10-1205"><a href="#cb10-1205" aria-hidden="true" tabindex="-1"></a>        <span class="co"># the genes. Works quite well, still problem with SCANB</span></span>
<span id="cb10-1206"><a href="#cb10-1206" aria-hidden="true" tabindex="-1"></a>        <span class="co"># apply(</span></span>
<span id="cb10-1207"><a href="#cb10-1207" aria-hidden="true" tabindex="-1"></a>        <span class="co">#     X = df_pcs_regressed,</span></span>
<span id="cb10-1208"><a href="#cb10-1208" aria-hidden="true" tabindex="-1"></a>        <span class="co">#     MARGIN = 2,</span></span>
<span id="cb10-1209"><a href="#cb10-1209" aria-hidden="true" tabindex="-1"></a>        <span class="co">#     function(v, name_genes, gene_set){</span></span>
<span id="cb10-1210"><a href="#cb10-1210" aria-hidden="true" tabindex="-1"></a>        <span class="co">#         mean(scale(v)[which(name_genes %in% gene_set)])</span></span>
<span id="cb10-1211"><a href="#cb10-1211" aria-hidden="true" tabindex="-1"></a>        <span class="co">#     },</span></span>
<span id="cb10-1212"><a href="#cb10-1212" aria-hidden="true" tabindex="-1"></a>        <span class="co">#     name_genes = rownames(df_pcs_regressed),</span></span>
<span id="cb10-1213"><a href="#cb10-1213" aria-hidden="true" tabindex="-1"></a>        <span class="co">#     gene_set = gene_set</span></span>
<span id="cb10-1214"><a href="#cb10-1214" aria-hidden="true" tabindex="-1"></a>        <span class="co"># )</span></span>
<span id="cb10-1215"><a href="#cb10-1215" aria-hidden="true" tabindex="-1"></a>        <span class="co"># 5) Scale and then calculate 100 times the score with subset of </span></span>
<span id="cb10-1216"><a href="#cb10-1216" aria-hidden="true" tabindex="-1"></a>        <span class="co"># genes and then take average value. </span></span>
<span id="cb10-1217"><a href="#cb10-1217" aria-hidden="true" tabindex="-1"></a>        <span class="co"># apply(</span></span>
<span id="cb10-1218"><a href="#cb10-1218" aria-hidden="true" tabindex="-1"></a>        <span class="co">#     X = df_pcs_regressed[, sample(colnames(df_pcs_regressed), size = 500)],</span></span>
<span id="cb10-1219"><a href="#cb10-1219" aria-hidden="true" tabindex="-1"></a>        <span class="co">#     MARGIN = 2,</span></span>
<span id="cb10-1220"><a href="#cb10-1220" aria-hidden="true" tabindex="-1"></a>        <span class="co">#     function(v, name_genes, gene_set){</span></span>
<span id="cb10-1221"><a href="#cb10-1221" aria-hidden="true" tabindex="-1"></a>        <span class="co">#         </span></span>
<span id="cb10-1222"><a href="#cb10-1222" aria-hidden="true" tabindex="-1"></a>        <span class="co">#         mean(sapply(1:100,</span></span>
<span id="cb10-1223"><a href="#cb10-1223" aria-hidden="true" tabindex="-1"></a>        <span class="co">#             function(i, x, gene_set, name_genes){</span></span>
<span id="cb10-1224"><a href="#cb10-1224" aria-hidden="true" tabindex="-1"></a>        <span class="co">#                 proportion &lt;- 0.2</span></span>
<span id="cb10-1225"><a href="#cb10-1225" aria-hidden="true" tabindex="-1"></a>        <span class="co">#                 nb_genes &lt;- floor(proportion * length(gene_set))</span></span>
<span id="cb10-1226"><a href="#cb10-1226" aria-hidden="true" tabindex="-1"></a>        <span class="co">#                 gene_set &lt;- sample(gene_set, nb_genes)</span></span>
<span id="cb10-1227"><a href="#cb10-1227" aria-hidden="true" tabindex="-1"></a>        <span class="co">#                 mean(x[which(name_genes %in% gene_set)])</span></span>
<span id="cb10-1228"><a href="#cb10-1228" aria-hidden="true" tabindex="-1"></a>        <span class="co">#             },</span></span>
<span id="cb10-1229"><a href="#cb10-1229" aria-hidden="true" tabindex="-1"></a>        <span class="co">#             x = scale(v),</span></span>
<span id="cb10-1230"><a href="#cb10-1230" aria-hidden="true" tabindex="-1"></a>        <span class="co">#             gene_set = gene_set,</span></span>
<span id="cb10-1231"><a href="#cb10-1231" aria-hidden="true" tabindex="-1"></a>        <span class="co">#             name_genes = name_genes</span></span>
<span id="cb10-1232"><a href="#cb10-1232" aria-hidden="true" tabindex="-1"></a>        <span class="co">#         ))</span></span>
<span id="cb10-1233"><a href="#cb10-1233" aria-hidden="true" tabindex="-1"></a>        <span class="co">#     },</span></span>
<span id="cb10-1234"><a href="#cb10-1234" aria-hidden="true" tabindex="-1"></a>        <span class="co">#     name_genes = rownames(df_pcs_regressed),</span></span>
<span id="cb10-1235"><a href="#cb10-1235" aria-hidden="true" tabindex="-1"></a>        <span class="co">#     gene_set = gene_set</span></span>
<span id="cb10-1236"><a href="#cb10-1236" aria-hidden="true" tabindex="-1"></a>        <span class="co"># )</span></span>
<span id="cb10-1237"><a href="#cb10-1237" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb10-1238"><a href="#cb10-1238" aria-hidden="true" tabindex="-1"></a>        <span class="fu">colSums</span>(df_pcs_regressed[gene_set, ])</span>
<span id="cb10-1239"><a href="#cb10-1239" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb10-1240"><a href="#cb10-1240" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb10-1241"><a href="#cb10-1241" aria-hidden="true" tabindex="-1"></a>    <span class="at">df_pcs_regressed =</span> df_pcs_regressed,</span>
<span id="cb10-1242"><a href="#cb10-1242" aria-hidden="true" tabindex="-1"></a>    <span class="at">avg_hkg_pcs_regressed =</span> avg_hkg_pcs_regressed</span>
<span id="cb10-1243"><a href="#cb10-1243" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-1244"><a href="#cb10-1244" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span></span>
<span id="cb10-1245"><a href="#cb10-1245" aria-hidden="true" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"sample_name"</span>)</span>
<span id="cb10-1246"><a href="#cb10-1246" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1247"><a href="#cb10-1247" aria-hidden="true" tabindex="-1"></a>scores <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">inner_join</span>(</span>
<span id="cb10-1248"><a href="#cb10-1248" aria-hidden="true" tabindex="-1"></a>    scores <span class="sc">%&gt;%</span> </span>
<span id="cb10-1249"><a href="#cb10-1249" aria-hidden="true" tabindex="-1"></a>        tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(</span>
<span id="cb10-1250"><a href="#cb10-1250" aria-hidden="true" tabindex="-1"></a>            <span class="at">cols =</span> dplyr<span class="sc">::</span><span class="fu">all_of</span>(<span class="fu">names</span>(pathways_of_interest)),</span>
<span id="cb10-1251"><a href="#cb10-1251" aria-hidden="true" tabindex="-1"></a>            <span class="at">values_to =</span> <span class="st">"score_hkg"</span>,</span>
<span id="cb10-1252"><a href="#cb10-1252" aria-hidden="true" tabindex="-1"></a>            <span class="at">names_to =</span> <span class="st">"pathway"</span></span>
<span id="cb10-1253"><a href="#cb10-1253" aria-hidden="true" tabindex="-1"></a>        ),</span>
<span id="cb10-1254"><a href="#cb10-1254" aria-hidden="true" tabindex="-1"></a>    df_pca <span class="sc">%&gt;%</span></span>
<span id="cb10-1255"><a href="#cb10-1255" aria-hidden="true" tabindex="-1"></a>        tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(</span>
<span id="cb10-1256"><a href="#cb10-1256" aria-hidden="true" tabindex="-1"></a>            <span class="at">cols =</span> dplyr<span class="sc">::</span><span class="fu">all_of</span>(<span class="fu">names</span>(pathways_of_interest)),</span>
<span id="cb10-1257"><a href="#cb10-1257" aria-hidden="true" tabindex="-1"></a>            <span class="at">values_to =</span> <span class="st">"score_og"</span>,</span>
<span id="cb10-1258"><a href="#cb10-1258" aria-hidden="true" tabindex="-1"></a>            <span class="at">names_to =</span> <span class="st">"pathway"</span></span>
<span id="cb10-1259"><a href="#cb10-1259" aria-hidden="true" tabindex="-1"></a>        ),</span>
<span id="cb10-1260"><a href="#cb10-1260" aria-hidden="true" tabindex="-1"></a>    <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"sample_name"</span>, <span class="st">"pathway"</span>)</span>
<span id="cb10-1261"><a href="#cb10-1261" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-1262"><a href="#cb10-1262" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1263"><a href="#cb10-1263" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb10-1264"><a href="#cb10-1264" aria-hidden="true" tabindex="-1"></a>    scores, </span>
<span id="cb10-1265"><a href="#cb10-1265" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/trying/scores_regressed_only.rds"</span></span>
<span id="cb10-1266"><a href="#cb10-1266" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-1267"><a href="#cb10-1267" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-1268"><a href="#cb10-1268" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1269"><a href="#cb10-1269" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=14, fig.height=7}</span></span>
<span id="cb10-1270"><a href="#cb10-1270" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-scores-regressed</span></span>
<span id="cb10-1271"><a href="#cb10-1271" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Scores obtained from regressed data</span></span>
<span id="cb10-1272"><a href="#cb10-1272" aria-hidden="true" tabindex="-1"></a><span class="co">#|     of all the four cohorts TCGA, SCANB, METABRIC and POETIC compared to the </span></span>
<span id="cb10-1273"><a href="#cb10-1273" aria-hidden="true" tabindex="-1"></a><span class="co">#|     original GSVA scores.</span></span>
<span id="cb10-1274"><a href="#cb10-1274" aria-hidden="true" tabindex="-1"></a>scores <span class="sc">%&gt;%</span> </span>
<span id="cb10-1275"><a href="#cb10-1275" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(pathway <span class="sc">%in%</span> <span class="fu">names</span>(pathways_of_interest)) <span class="sc">%&gt;%</span></span>
<span id="cb10-1276"><a href="#cb10-1276" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">pathway =</span> <span class="fu">factor</span>(pathway, <span class="at">levels =</span> <span class="fu">names</span>(pathways_of_interest))) <span class="sc">%&gt;%</span></span>
<span id="cb10-1277"><a href="#cb10-1277" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> score_og, <span class="at">y =</span> score_hkg)) <span class="sc">+</span></span>
<span id="cb10-1278"><a href="#cb10-1278" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb10-1279"><a href="#cb10-1279" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(</span>
<span id="cb10-1280"><a href="#cb10-1280" aria-hidden="true" tabindex="-1"></a>        cohort <span class="sc">~</span> pathway, </span>
<span id="cb10-1281"><a href="#cb10-1281" aria-hidden="true" tabindex="-1"></a>        <span class="at">scales =</span> <span class="st">"free"</span>, </span>
<span id="cb10-1282"><a href="#cb10-1282" aria-hidden="true" tabindex="-1"></a>        <span class="at">ncol =</span> <span class="fu">length</span>(pathways_of_interest),</span>
<span id="cb10-1283"><a href="#cb10-1283" aria-hidden="true" tabindex="-1"></a>        <span class="at">labeller =</span> <span class="fu">as_labeller</span>(<span class="fu">c</span>(</span>
<span id="cb10-1284"><a href="#cb10-1284" aria-hidden="true" tabindex="-1"></a>            pathways_of_interest, </span>
<span id="cb10-1285"><a href="#cb10-1285" aria-hidden="true" tabindex="-1"></a>            <span class="st">"metabric"</span> <span class="ot">=</span> <span class="st">"METABRIC"</span>,</span>
<span id="cb10-1286"><a href="#cb10-1286" aria-hidden="true" tabindex="-1"></a>            <span class="st">"scanb"</span> <span class="ot">=</span> <span class="st">"SCANB"</span>,</span>
<span id="cb10-1287"><a href="#cb10-1287" aria-hidden="true" tabindex="-1"></a>            <span class="st">"tcga"</span> <span class="ot">=</span> <span class="st">"TCGA"</span>,</span>
<span id="cb10-1288"><a href="#cb10-1288" aria-hidden="true" tabindex="-1"></a>            <span class="st">"poetic"</span> <span class="ot">=</span> <span class="st">"POETIC"</span></span>
<span id="cb10-1289"><a href="#cb10-1289" aria-hidden="true" tabindex="-1"></a>        ))</span>
<span id="cb10-1290"><a href="#cb10-1290" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb10-1291"><a href="#cb10-1291" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb10-1292"><a href="#cb10-1292" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"GSVA score"</span>,</span>
<span id="cb10-1293"><a href="#cb10-1293" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"Regressed score"</span></span>
<span id="cb10-1294"><a href="#cb10-1294" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb10-1295"><a href="#cb10-1295" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>()</span>
<span id="cb10-1296"><a href="#cb10-1296" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-1297"><a href="#cb10-1297" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1298"><a href="#cb10-1298" aria-hidden="true" tabindex="-1"></a>The scores are actually highly correlated, specially the estrogen related </span>
<span id="cb10-1299"><a href="#cb10-1299" aria-hidden="true" tabindex="-1"></a>pathways. Also the correlation between the scores in general follows the </span>
<span id="cb10-1300"><a href="#cb10-1300" aria-hidden="true" tabindex="-1"></a>same trends.</span>
<span id="cb10-1301"><a href="#cb10-1301" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1302"><a href="#cb10-1302" aria-hidden="true" tabindex="-1"></a>The plot below is a subset of the scores above to include in the </span>
<span id="cb10-1303"><a href="#cb10-1303" aria-hidden="true" tabindex="-1"></a>publication.</span>
<span id="cb10-1304"><a href="#cb10-1304" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1305"><a href="#cb10-1305" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=15, fig.height=7}</span></span>
<span id="cb10-1306"><a href="#cb10-1306" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-scores-regressed-subset</span></span>
<span id="cb10-1307"><a href="#cb10-1307" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Scores obtained from regressed data</span></span>
<span id="cb10-1308"><a href="#cb10-1308" aria-hidden="true" tabindex="-1"></a><span class="co">#|     of all the four cohorts TCGA, SCANB, METABRIC and POETIC compared to the </span></span>
<span id="cb10-1309"><a href="#cb10-1309" aria-hidden="true" tabindex="-1"></a><span class="co">#|     original GSVA scores.</span></span>
<span id="cb10-1310"><a href="#cb10-1310" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1311"><a href="#cb10-1311" aria-hidden="true" tabindex="-1"></a>pathways_of_interest_sub <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb10-1312"><a href="#cb10-1312" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_ESTROGEN_RESPONSE_EARLY"</span> <span class="ot">=</span> <span class="st">"Estrogen Early"</span>,</span>
<span id="cb10-1313"><a href="#cb10-1313" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_G2M_CHECKPOINT"</span> <span class="ot">=</span> <span class="st">"G2M"</span>, </span>
<span id="cb10-1314"><a href="#cb10-1314" aria-hidden="true" tabindex="-1"></a>    <span class="st">"SET_ERPR"</span> <span class="ot">=</span> <span class="st">"SET ER/PR"</span>,</span>
<span id="cb10-1315"><a href="#cb10-1315" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION"</span> <span class="ot">=</span> <span class="st">"EMT"</span>,</span>
<span id="cb10-1316"><a href="#cb10-1316" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_DNA_REPAIR"</span> <span class="ot">=</span> <span class="st">"DNA Repair"</span></span>
<span id="cb10-1317"><a href="#cb10-1317" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-1318"><a href="#cb10-1318" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1319"><a href="#cb10-1319" aria-hidden="true" tabindex="-1"></a>scores <span class="sc">%&gt;%</span> </span>
<span id="cb10-1320"><a href="#cb10-1320" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(pathway <span class="sc">%in%</span> <span class="fu">names</span>(pathways_of_interest_sub)) <span class="sc">%&gt;%</span></span>
<span id="cb10-1321"><a href="#cb10-1321" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="fu">c</span>(<span class="st">"scanb"</span>, <span class="st">"metabric"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb10-1322"><a href="#cb10-1322" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">pathway =</span> <span class="fu">factor</span>(pathway, <span class="at">levels =</span> <span class="fu">names</span>(pathways_of_interest_sub))) <span class="sc">%&gt;%</span></span>
<span id="cb10-1323"><a href="#cb10-1323" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> score_og, <span class="at">y =</span> score_hkg)) <span class="sc">+</span></span>
<span id="cb10-1324"><a href="#cb10-1324" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb10-1325"><a href="#cb10-1325" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"loess"</span>, <span class="at">formula =</span> y<span class="sc">~</span>x) <span class="sc">+</span> </span>
<span id="cb10-1326"><a href="#cb10-1326" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(</span>
<span id="cb10-1327"><a href="#cb10-1327" aria-hidden="true" tabindex="-1"></a>        cohort <span class="sc">~</span> pathway, </span>
<span id="cb10-1328"><a href="#cb10-1328" aria-hidden="true" tabindex="-1"></a>        <span class="at">scales =</span> <span class="st">"free"</span>, </span>
<span id="cb10-1329"><a href="#cb10-1329" aria-hidden="true" tabindex="-1"></a>        <span class="at">ncol =</span> <span class="fu">length</span>(pathways_of_interest_sub),</span>
<span id="cb10-1330"><a href="#cb10-1330" aria-hidden="true" tabindex="-1"></a>        <span class="at">labeller =</span> <span class="fu">as_labeller</span>(<span class="fu">c</span>(</span>
<span id="cb10-1331"><a href="#cb10-1331" aria-hidden="true" tabindex="-1"></a>            pathways_of_interest_sub, </span>
<span id="cb10-1332"><a href="#cb10-1332" aria-hidden="true" tabindex="-1"></a>            <span class="st">"metabric"</span> <span class="ot">=</span> <span class="st">"METABRIC"</span>,</span>
<span id="cb10-1333"><a href="#cb10-1333" aria-hidden="true" tabindex="-1"></a>            <span class="st">"scanb"</span> <span class="ot">=</span> <span class="st">"SCANB"</span></span>
<span id="cb10-1334"><a href="#cb10-1334" aria-hidden="true" tabindex="-1"></a>        ))</span>
<span id="cb10-1335"><a href="#cb10-1335" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb10-1336"><a href="#cb10-1336" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb10-1337"><a href="#cb10-1337" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"GSVA score"</span>,</span>
<span id="cb10-1338"><a href="#cb10-1338" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"Regressed score"</span></span>
<span id="cb10-1339"><a href="#cb10-1339" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb10-1340"><a href="#cb10-1340" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_x_continuous</span>(</span>
<span id="cb10-1341"><a href="#cb10-1341" aria-hidden="true" tabindex="-1"></a>        <span class="at">labels =</span> scales<span class="sc">::</span><span class="fu">number_format</span>(<span class="at">accuracy =</span> <span class="fl">0.1</span>)</span>
<span id="cb10-1342"><a href="#cb10-1342" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb10-1343"><a href="#cb10-1343" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>) <span class="sc">+</span></span>
<span id="cb10-1344"><a href="#cb10-1344" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>() <span class="sc">+</span></span>
<span id="cb10-1345"><a href="#cb10-1345" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>()</span>
<span id="cb10-1346"><a href="#cb10-1346" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-1347"><a href="#cb10-1347" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1348"><a href="#cb10-1348" aria-hidden="true" tabindex="-1"></a>We now compare the distributions</span>
<span id="cb10-1349"><a href="#cb10-1349" aria-hidden="true" tabindex="-1"></a>of these scores. Across the different cohorts and pathways of interest.</span>
<span id="cb10-1350"><a href="#cb10-1350" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1351"><a href="#cb10-1351" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=16, fig.height=8}</span></span>
<span id="cb10-1352"><a href="#cb10-1352" aria-hidden="true" tabindex="-1"></a>scores <span class="sc">%&gt;%</span></span>
<span id="cb10-1353"><a href="#cb10-1353" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(pathway <span class="sc">%in%</span> <span class="fu">names</span>(pathways_of_interest)) <span class="sc">%&gt;%</span></span>
<span id="cb10-1354"><a href="#cb10-1354" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">!=</span> <span class="st">"poetic"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-1355"><a href="#cb10-1355" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(pam50 <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"luma"</span>, <span class="st">"lumb"</span>, <span class="st">"her2"</span>, <span class="st">"basal"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb10-1356"><a href="#cb10-1356" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> score_hkg, <span class="at">fill =</span> cohort)) <span class="sc">+</span></span>
<span id="cb10-1357"><a href="#cb10-1357" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_density</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb10-1358"><a href="#cb10-1358" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(</span>
<span id="cb10-1359"><a href="#cb10-1359" aria-hidden="true" tabindex="-1"></a>        pam50<span class="sc">~</span>pathway, </span>
<span id="cb10-1360"><a href="#cb10-1360" aria-hidden="true" tabindex="-1"></a>        <span class="at">scales =</span> <span class="st">"free"</span>, </span>
<span id="cb10-1361"><a href="#cb10-1361" aria-hidden="true" tabindex="-1"></a>        <span class="at">ncol =</span> <span class="fu">length</span>(pathways_of_interest),</span>
<span id="cb10-1362"><a href="#cb10-1362" aria-hidden="true" tabindex="-1"></a>        <span class="at">labeller =</span> <span class="fu">as_labeller</span>(<span class="fu">c</span>(</span>
<span id="cb10-1363"><a href="#cb10-1363" aria-hidden="true" tabindex="-1"></a>            pathways_of_interest, </span>
<span id="cb10-1364"><a href="#cb10-1364" aria-hidden="true" tabindex="-1"></a>            <span class="st">"luma"</span> <span class="ot">=</span> <span class="st">"Luminal A"</span>,</span>
<span id="cb10-1365"><a href="#cb10-1365" aria-hidden="true" tabindex="-1"></a>            <span class="st">"lumb"</span> <span class="ot">=</span> <span class="st">"Luminal B"</span>,</span>
<span id="cb10-1366"><a href="#cb10-1366" aria-hidden="true" tabindex="-1"></a>            <span class="st">"her2"</span> <span class="ot">=</span> <span class="st">"HER2"</span>,</span>
<span id="cb10-1367"><a href="#cb10-1367" aria-hidden="true" tabindex="-1"></a>            <span class="st">"basal"</span> <span class="ot">=</span> <span class="st">"Basal"</span></span>
<span id="cb10-1368"><a href="#cb10-1368" aria-hidden="true" tabindex="-1"></a>        ))</span>
<span id="cb10-1369"><a href="#cb10-1369" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb10-1370"><a href="#cb10-1370" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>()</span>
<span id="cb10-1371"><a href="#cb10-1371" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-1372"><a href="#cb10-1372" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1373"><a href="#cb10-1373" aria-hidden="true" tabindex="-1"></a>And with all molecular subtypes together.</span>
<span id="cb10-1374"><a href="#cb10-1374" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1375"><a href="#cb10-1375" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=6}</span></span>
<span id="cb10-1376"><a href="#cb10-1376" aria-hidden="true" tabindex="-1"></a>scores <span class="sc">%&gt;%</span></span>
<span id="cb10-1377"><a href="#cb10-1377" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(pathway <span class="sc">%in%</span> <span class="fu">names</span>(pathways_of_interest)) <span class="sc">%&gt;%</span></span>
<span id="cb10-1378"><a href="#cb10-1378" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> score_hkg, <span class="at">fill =</span> cohort)) <span class="sc">+</span></span>
<span id="cb10-1379"><a href="#cb10-1379" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_density</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb10-1380"><a href="#cb10-1380" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(</span>
<span id="cb10-1381"><a href="#cb10-1381" aria-hidden="true" tabindex="-1"></a>        <span class="sc">~</span>pathway, </span>
<span id="cb10-1382"><a href="#cb10-1382" aria-hidden="true" tabindex="-1"></a>        <span class="at">scales =</span> <span class="st">"free"</span>, </span>
<span id="cb10-1383"><a href="#cb10-1383" aria-hidden="true" tabindex="-1"></a>        <span class="at">labeller =</span> <span class="fu">as_labeller</span>(pathways_of_interest)</span>
<span id="cb10-1384"><a href="#cb10-1384" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb10-1385"><a href="#cb10-1385" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>()</span>
<span id="cb10-1386"><a href="#cb10-1386" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-1387"><a href="#cb10-1387" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1388"><a href="#cb10-1388" aria-hidden="true" tabindex="-1"></a>We see that the overlap is actually pretty good in general, even for </span>
<span id="cb10-1389"><a href="#cb10-1389" aria-hidden="true" tabindex="-1"></a>SCANB. The only part where there is a bigger discrepancy is for</span>
<span id="cb10-1390"><a href="#cb10-1390" aria-hidden="true" tabindex="-1"></a>G2M checkpoint. We see below that by calculating the GSVA score </span>
<span id="cb10-1391"><a href="#cb10-1391" aria-hidden="true" tabindex="-1"></a>on the whole scanb cohort we don't have this problem.</span>
<span id="cb10-1392"><a href="#cb10-1392" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1393"><a href="#cb10-1393" aria-hidden="true" tabindex="-1"></a>And as a comparison we plot with the original scores using GSVA on the </span>
<span id="cb10-1394"><a href="#cb10-1394" aria-hidden="true" tabindex="-1"></a>whole cohort individually.</span>
<span id="cb10-1395"><a href="#cb10-1395" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1396"><a href="#cb10-1396" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=16, fig.height=8}</span></span>
<span id="cb10-1397"><a href="#cb10-1397" aria-hidden="true" tabindex="-1"></a>scores <span class="sc">%&gt;%</span></span>
<span id="cb10-1398"><a href="#cb10-1398" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(pathway <span class="sc">%in%</span> <span class="fu">names</span>(pathways_of_interest)) <span class="sc">%&gt;%</span></span>
<span id="cb10-1399"><a href="#cb10-1399" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(pam50 <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"luma"</span>, <span class="st">"lumb"</span>, <span class="st">"her2"</span>, <span class="st">"basal"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb10-1400"><a href="#cb10-1400" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> score_og, <span class="at">fill =</span> cohort)) <span class="sc">+</span></span>
<span id="cb10-1401"><a href="#cb10-1401" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_density</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb10-1402"><a href="#cb10-1402" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(</span>
<span id="cb10-1403"><a href="#cb10-1403" aria-hidden="true" tabindex="-1"></a>        pam50<span class="sc">~</span>pathway, </span>
<span id="cb10-1404"><a href="#cb10-1404" aria-hidden="true" tabindex="-1"></a>        <span class="at">scales =</span> <span class="st">"free"</span>, </span>
<span id="cb10-1405"><a href="#cb10-1405" aria-hidden="true" tabindex="-1"></a>        <span class="at">ncol =</span> <span class="fu">length</span>(pathways_of_interest),</span>
<span id="cb10-1406"><a href="#cb10-1406" aria-hidden="true" tabindex="-1"></a>        <span class="at">labeller =</span> <span class="fu">as_labeller</span>(<span class="fu">c</span>(</span>
<span id="cb10-1407"><a href="#cb10-1407" aria-hidden="true" tabindex="-1"></a>            pathways_of_interest, </span>
<span id="cb10-1408"><a href="#cb10-1408" aria-hidden="true" tabindex="-1"></a>            <span class="st">"luma"</span> <span class="ot">=</span> <span class="st">"Luminal A"</span>,</span>
<span id="cb10-1409"><a href="#cb10-1409" aria-hidden="true" tabindex="-1"></a>            <span class="st">"lumb"</span> <span class="ot">=</span> <span class="st">"Luminal B"</span>,</span>
<span id="cb10-1410"><a href="#cb10-1410" aria-hidden="true" tabindex="-1"></a>            <span class="st">"her2"</span> <span class="ot">=</span> <span class="st">"HER2"</span>,</span>
<span id="cb10-1411"><a href="#cb10-1411" aria-hidden="true" tabindex="-1"></a>            <span class="st">"basal"</span> <span class="ot">=</span> <span class="st">"Basal"</span></span>
<span id="cb10-1412"><a href="#cb10-1412" aria-hidden="true" tabindex="-1"></a>        ))</span>
<span id="cb10-1413"><a href="#cb10-1413" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb10-1414"><a href="#cb10-1414" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>()</span>
<span id="cb10-1415"><a href="#cb10-1415" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-1416"><a href="#cb10-1416" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1417"><a href="#cb10-1417" aria-hidden="true" tabindex="-1"></a>And all the subtypes together for the original score:</span>
<span id="cb10-1418"><a href="#cb10-1418" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1419"><a href="#cb10-1419" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=6}</span></span>
<span id="cb10-1420"><a href="#cb10-1420" aria-hidden="true" tabindex="-1"></a>scores <span class="sc">%&gt;%</span></span>
<span id="cb10-1421"><a href="#cb10-1421" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(pathway <span class="sc">%in%</span> <span class="fu">names</span>(pathways_of_interest)) <span class="sc">%&gt;%</span></span>
<span id="cb10-1422"><a href="#cb10-1422" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> score_og, <span class="at">fill =</span> cohort)) <span class="sc">+</span></span>
<span id="cb10-1423"><a href="#cb10-1423" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_density</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb10-1424"><a href="#cb10-1424" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(</span>
<span id="cb10-1425"><a href="#cb10-1425" aria-hidden="true" tabindex="-1"></a>        <span class="sc">~</span>pathway, </span>
<span id="cb10-1426"><a href="#cb10-1426" aria-hidden="true" tabindex="-1"></a>        <span class="at">scales =</span> <span class="st">"free"</span>, </span>
<span id="cb10-1427"><a href="#cb10-1427" aria-hidden="true" tabindex="-1"></a>        <span class="at">labeller =</span> <span class="fu">as_labeller</span>(pathways_of_interest)</span>
<span id="cb10-1428"><a href="#cb10-1428" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb10-1429"><a href="#cb10-1429" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>()</span>
<span id="cb10-1430"><a href="#cb10-1430" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-1431"><a href="#cb10-1431" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1432"><a href="#cb10-1432" aria-hidden="true" tabindex="-1"></a>We see that by doing GSVA the distributions they overlap much more than </span>
<span id="cb10-1433"><a href="#cb10-1433" aria-hidden="true" tabindex="-1"></a>using the regressed data and the simple sum of the values for those genes.</span>
<span id="cb10-1434"><a href="#cb10-1434" aria-hidden="true" tabindex="-1"></a>Still for estrogen signaling it seems that the estrogen early signature</span>
<span id="cb10-1435"><a href="#cb10-1435" aria-hidden="true" tabindex="-1"></a>is pretty robust.</span>
<span id="cb10-1436"><a href="#cb10-1436" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1437"><a href="#cb10-1437" aria-hidden="true" tabindex="-1"></a><span class="fu">### Checking the patients from POETIC with new scores</span></span>
<span id="cb10-1438"><a href="#cb10-1438" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1439"><a href="#cb10-1439" aria-hidden="true" tabindex="-1"></a>We now use the new scores to evaluate the comparison with the neighboorhood,</span>
<span id="cb10-1440"><a href="#cb10-1440" aria-hidden="true" tabindex="-1"></a>just as before in the previous chapter where we used the POETIC trial data.</span>
<span id="cb10-1441"><a href="#cb10-1441" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1442"><a href="#cb10-1442" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb10-1443"><a href="#cb10-1443" aria-hidden="true" tabindex="-1"></a><span class="co"># get scores in the wide format so we can join with the original df_pca</span></span>
<span id="cb10-1444"><a href="#cb10-1444" aria-hidden="true" tabindex="-1"></a><span class="co"># dataframe and use the same code as in the previous scoring chapter</span></span>
<span id="cb10-1445"><a href="#cb10-1445" aria-hidden="true" tabindex="-1"></a>scores_to_use <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb10-1446"><a href="#cb10-1446" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Estrogen response early"</span>,</span>
<span id="cb10-1447"><a href="#cb10-1447" aria-hidden="true" tabindex="-1"></a>    <span class="st">"E2F targets"</span>,</span>
<span id="cb10-1448"><a href="#cb10-1448" aria-hidden="true" tabindex="-1"></a>    <span class="st">"G2M"</span>,</span>
<span id="cb10-1449"><a href="#cb10-1449" aria-hidden="true" tabindex="-1"></a>    <span class="st">"P53 pathway"</span>,</span>
<span id="cb10-1450"><a href="#cb10-1450" aria-hidden="true" tabindex="-1"></a>    <span class="st">"EMT"</span>,</span>
<span id="cb10-1451"><a href="#cb10-1451" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Androgen response"</span>,</span>
<span id="cb10-1452"><a href="#cb10-1452" aria-hidden="true" tabindex="-1"></a>    <span class="st">"TGFb signaling"</span>,</span>
<span id="cb10-1453"><a href="#cb10-1453" aria-hidden="true" tabindex="-1"></a>    <span class="st">"SET ER/PR"</span></span>
<span id="cb10-1454"><a href="#cb10-1454" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> <span class="st">`</span><span class="at">names&lt;-</span><span class="st">`</span>(<span class="fu">paste0</span>(<span class="fu">c</span>(</span>
<span id="cb10-1455"><a href="#cb10-1455" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_ESTROGEN_RESPONSE_EARLY"</span>,</span>
<span id="cb10-1456"><a href="#cb10-1456" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_E2F_TARGETS"</span>,</span>
<span id="cb10-1457"><a href="#cb10-1457" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_G2M_CHECKPOINT"</span>,</span>
<span id="cb10-1458"><a href="#cb10-1458" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_P53_PATHWAY"</span>,</span>
<span id="cb10-1459"><a href="#cb10-1459" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION"</span>,</span>
<span id="cb10-1460"><a href="#cb10-1460" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_ANDROGEN_RESPONSE"</span>,</span>
<span id="cb10-1461"><a href="#cb10-1461" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_TGF_BETA_SIGNALING"</span>,</span>
<span id="cb10-1462"><a href="#cb10-1462" aria-hidden="true" tabindex="-1"></a>    <span class="st">"SET_ERPR"</span></span>
<span id="cb10-1463"><a href="#cb10-1463" aria-hidden="true" tabindex="-1"></a>)))</span>
<span id="cb10-1464"><a href="#cb10-1464" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1465"><a href="#cb10-1465" aria-hidden="true" tabindex="-1"></a>scores_wide <span class="ot">&lt;-</span> <span class="fu">sapply</span>(</span>
<span id="cb10-1466"><a href="#cb10-1466" aria-hidden="true" tabindex="-1"></a>    gene_sets_,</span>
<span id="cb10-1467"><a href="#cb10-1467" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(gene_set, df_pcs_regressed, avg_hkg_pcs_regressed){</span>
<span id="cb10-1468"><a href="#cb10-1468" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb10-1469"><a href="#cb10-1469" aria-hidden="true" tabindex="-1"></a>        proportion <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb10-1470"><a href="#cb10-1470" aria-hidden="true" tabindex="-1"></a>        nb_genes <span class="ot">&lt;-</span> <span class="fu">floor</span>(proportion <span class="sc">*</span> <span class="fu">length</span>(gene_set))</span>
<span id="cb10-1471"><a href="#cb10-1471" aria-hidden="true" tabindex="-1"></a>        gene_set <span class="ot">&lt;-</span> <span class="fu">intersect</span>(</span>
<span id="cb10-1472"><a href="#cb10-1472" aria-hidden="true" tabindex="-1"></a>            <span class="fu">sample</span>(gene_set, nb_genes),</span>
<span id="cb10-1473"><a href="#cb10-1473" aria-hidden="true" tabindex="-1"></a>            <span class="fu">rownames</span>(df_pcs_regressed)</span>
<span id="cb10-1474"><a href="#cb10-1474" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb10-1475"><a href="#cb10-1475" aria-hidden="true" tabindex="-1"></a>        <span class="fu">colSums</span>(df_pcs_regressed[gene_set, ])</span>
<span id="cb10-1476"><a href="#cb10-1476" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb10-1477"><a href="#cb10-1477" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb10-1478"><a href="#cb10-1478" aria-hidden="true" tabindex="-1"></a>    <span class="at">df_pcs_regressed =</span> df_pcs_regressed,</span>
<span id="cb10-1479"><a href="#cb10-1479" aria-hidden="true" tabindex="-1"></a>    <span class="at">avg_hkg_pcs_regressed =</span> avg_hkg_pcs_regressed</span>
<span id="cb10-1480"><a href="#cb10-1480" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-1481"><a href="#cb10-1481" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span></span>
<span id="cb10-1482"><a href="#cb10-1482" aria-hidden="true" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"sample_name"</span>)</span>
<span id="cb10-1483"><a href="#cb10-1483" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1484"><a href="#cb10-1484" aria-hidden="true" tabindex="-1"></a>df_pca_scores <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">inner_join</span>(</span>
<span id="cb10-1485"><a href="#cb10-1485" aria-hidden="true" tabindex="-1"></a>    df_pca, </span>
<span id="cb10-1486"><a href="#cb10-1486" aria-hidden="true" tabindex="-1"></a>    scores_wide, </span>
<span id="cb10-1487"><a href="#cb10-1487" aria-hidden="true" tabindex="-1"></a>    <span class="at">by =</span> <span class="st">"sample_name"</span>, <span class="at">suffix =</span> <span class="fu">c</span>(<span class="st">"_og"</span>, <span class="st">"_new"</span>)</span>
<span id="cb10-1488"><a href="#cb10-1488" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-1489"><a href="#cb10-1489" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1490"><a href="#cb10-1490" aria-hidden="true" tabindex="-1"></a><span class="co"># arguments for getting the neighborhood score</span></span>
<span id="cb10-1491"><a href="#cb10-1491" aria-hidden="true" tabindex="-1"></a>patients <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"treated_nb63_baseline"</span>, <span class="st">"treated_nb236_baseline"</span>)</span>
<span id="cb10-1492"><a href="#cb10-1492" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(scores_to_use) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">names</span>(scores_to_use), <span class="st">"_new"</span>)</span>
<span id="cb10-1493"><a href="#cb10-1493" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1494"><a href="#cb10-1494" aria-hidden="true" tabindex="-1"></a>pipeline_scores_plots_new_scores <span class="ot">&lt;-</span> <span class="fu">sapply</span>(</span>
<span id="cb10-1495"><a href="#cb10-1495" aria-hidden="true" tabindex="-1"></a>    patients, </span>
<span id="cb10-1496"><a href="#cb10-1496" aria-hidden="true" tabindex="-1"></a>    get_patient_scores_distributions_all,</span>
<span id="cb10-1497"><a href="#cb10-1497" aria-hidden="true" tabindex="-1"></a>    <span class="at">df_pca =</span> df_pca_scores,</span>
<span id="cb10-1498"><a href="#cb10-1498" aria-hidden="true" tabindex="-1"></a>    <span class="at">scores_to_use =</span> scores_to_use,</span>
<span id="cb10-1499"><a href="#cb10-1499" aria-hidden="true" tabindex="-1"></a>    <span class="at">base_size =</span> <span class="dv">20</span>,</span>
<span id="cb10-1500"><a href="#cb10-1500" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>, </span>
<span id="cb10-1501"><a href="#cb10-1501" aria-hidden="true" tabindex="-1"></a>    <span class="at">simplify =</span> <span class="cn">FALSE</span></span>
<span id="cb10-1502"><a href="#cb10-1502" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-1503"><a href="#cb10-1503" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1504"><a href="#cb10-1504" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb10-1505"><a href="#cb10-1505" aria-hidden="true" tabindex="-1"></a>    pipeline_scores_plots_new_scores,</span>
<span id="cb10-1506"><a href="#cb10-1506" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/trying/pipeline_scores_plots_new_scores.rds"</span></span>
<span id="cb10-1507"><a href="#cb10-1507" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-1508"><a href="#cb10-1508" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1509"><a href="#cb10-1509" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb10-1510"><a href="#cb10-1510" aria-hidden="true" tabindex="-1"></a>    df_pca_scores,</span>
<span id="cb10-1511"><a href="#cb10-1511" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/trying/df_pca_scores.rds"</span></span>
<span id="cb10-1512"><a href="#cb10-1512" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-1513"><a href="#cb10-1513" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-1514"><a href="#cb10-1514" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1515"><a href="#cb10-1515" aria-hidden="true" tabindex="-1"></a>Patient 63 @fig-pt63 is considered a responder. According to </span>
<span id="cb10-1516"><a href="#cb10-1516" aria-hidden="true" tabindex="-1"></a>the scores, when comparing the estrogen early signature to its average </span>
<span id="cb10-1517"><a href="#cb10-1517" aria-hidden="true" tabindex="-1"></a>distribution in the neighborhood, the score is higher.</span>
<span id="cb10-1518"><a href="#cb10-1518" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1519"><a href="#cb10-1519" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.height=8, fig.width=10}</span></span>
<span id="cb10-1520"><a href="#cb10-1520" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-pt63</span></span>
<span id="cb10-1521"><a href="#cb10-1521" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Posterior distribution of the average scores in the neighborhood</span></span>
<span id="cb10-1522"><a href="#cb10-1522" aria-hidden="true" tabindex="-1"></a><span class="co">#|     of patient 63 from POETIC trial. Each dot corresponds to a 1% quantile.</span></span>
<span id="cb10-1523"><a href="#cb10-1523" aria-hidden="true" tabindex="-1"></a>pipeline_scores_plots_new_scores<span class="sc">$</span>treated_nb63_baseline<span class="sc">$</span>plot <span class="sc">+</span></span>
<span id="cb10-1524"><a href="#cb10-1524" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>()</span>
<span id="cb10-1525"><a href="#cb10-1525" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-1526"><a href="#cb10-1526" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1527"><a href="#cb10-1527" aria-hidden="true" tabindex="-1"></a>Patient 236 @fig-pt236 was considered a non responder. According to </span>
<span id="cb10-1528"><a href="#cb10-1528" aria-hidden="true" tabindex="-1"></a>the scores, when comparing the estrogen early signature to its average </span>
<span id="cb10-1529"><a href="#cb10-1529" aria-hidden="true" tabindex="-1"></a>distribution in the neighborhood, the score is lower, being in the</span>
<span id="cb10-1530"><a href="#cb10-1530" aria-hidden="true" tabindex="-1"></a>5% quantile.</span>
<span id="cb10-1531"><a href="#cb10-1531" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1532"><a href="#cb10-1532" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.height=8, fig.width=10}</span></span>
<span id="cb10-1533"><a href="#cb10-1533" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-pt236</span></span>
<span id="cb10-1534"><a href="#cb10-1534" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Posterior distribution of the average scores in the neighborhood</span></span>
<span id="cb10-1535"><a href="#cb10-1535" aria-hidden="true" tabindex="-1"></a><span class="co">#|     of patient 236 from POETIC trial. Each dot corresponds to a 1% quantile.</span></span>
<span id="cb10-1536"><a href="#cb10-1536" aria-hidden="true" tabindex="-1"></a>pipeline_scores_plots_new_scores<span class="sc">$</span>treated_nb236_baseline<span class="sc">$</span>plot <span class="sc">+</span></span>
<span id="cb10-1537"><a href="#cb10-1537" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>()</span>
<span id="cb10-1538"><a href="#cb10-1538" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-1539"><a href="#cb10-1539" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1540"><a href="#cb10-1540" aria-hidden="true" tabindex="-1"></a>When comparing these two patients, there is also a difference in the androgen</span>
<span id="cb10-1541"><a href="#cb10-1541" aria-hidden="true" tabindex="-1"></a>response score, which could be a reflection of different estrogen signaling.</span>
<span id="cb10-1542"><a href="#cb10-1542" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1543"><a href="#cb10-1543" aria-hidden="true" tabindex="-1"></a>It reflects what we saw previously as well. </span>
<span id="cb10-1544"><a href="#cb10-1544" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1545"><a href="#cb10-1545" aria-hidden="true" tabindex="-1"></a><span class="fu">### Responder vs non responder comparison of new scores</span></span>
<span id="cb10-1546"><a href="#cb10-1546" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1547"><a href="#cb10-1547" aria-hidden="true" tabindex="-1"></a>We now compare the scores for responders and non responders.</span>
<span id="cb10-1548"><a href="#cb10-1548" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1549"><a href="#cb10-1549" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=8, fig.height=5}</span></span>
<span id="cb10-1550"><a href="#cb10-1550" aria-hidden="true" tabindex="-1"></a>df_pca_scores <span class="sc">%&gt;%</span></span>
<span id="cb10-1551"><a href="#cb10-1551" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb10-1552"><a href="#cb10-1552" aria-hidden="true" tabindex="-1"></a>        cohort <span class="sc">==</span> <span class="st">"poetic"</span> <span class="sc">&amp;</span> </span>
<span id="cb10-1553"><a href="#cb10-1553" aria-hidden="true" tabindex="-1"></a>            timepoint <span class="sc">==</span> <span class="st">"baseline"</span> <span class="sc">&amp;</span> </span>
<span id="cb10-1554"><a href="#cb10-1554" aria-hidden="true" tabindex="-1"></a>            is_responder <span class="sc">!=</span> <span class="st">"not_available"</span></span>
<span id="cb10-1555"><a href="#cb10-1555" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb10-1556"><a href="#cb10-1556" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(</span>
<span id="cb10-1557"><a href="#cb10-1557" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> is_responder, </span>
<span id="cb10-1558"><a href="#cb10-1558" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> HALLMARK_ESTROGEN_RESPONSE_EARLY_new,</span>
<span id="cb10-1559"><a href="#cb10-1559" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> HALLMARK_E2F_TARGETS_new</span>
<span id="cb10-1560"><a href="#cb10-1560" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">+</span></span>
<span id="cb10-1561"><a href="#cb10-1561" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb10-1562"><a href="#cb10-1562" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">formula =</span> y<span class="sc">~</span>x) <span class="sc">+</span></span>
<span id="cb10-1563"><a href="#cb10-1563" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb10-1564"><a href="#cb10-1564" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"Estrogen Early"</span>,</span>
<span id="cb10-1565"><a href="#cb10-1565" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"E2F targets"</span>,</span>
<span id="cb10-1566"><a href="#cb10-1566" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">"Scores calculated after regressing the data"</span></span>
<span id="cb10-1567"><a href="#cb10-1567" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb10-1568"><a href="#cb10-1568" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>) <span class="sc">+</span></span>
<span id="cb10-1569"><a href="#cb10-1569" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>()</span>
<span id="cb10-1570"><a href="#cb10-1570" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-1571"><a href="#cb10-1571" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1572"><a href="#cb10-1572" aria-hidden="true" tabindex="-1"></a>The higher E2F targets in average the lower the estrogen early score is</span>
<span id="cb10-1573"><a href="#cb10-1573" aria-hidden="true" tabindex="-1"></a>for non responders, whereas this correlation does not exist for responder</span>
<span id="cb10-1574"><a href="#cb10-1574" aria-hidden="true" tabindex="-1"></a>patients. This difference is mostly drive by the high E2F target non responder</span>
<span id="cb10-1575"><a href="#cb10-1575" aria-hidden="true" tabindex="-1"></a>tumors. </span>
<span id="cb10-1576"><a href="#cb10-1576" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1577"><a href="#cb10-1577" aria-hidden="true" tabindex="-1"></a>Now the plot below shows the correlation for baselien Ki67% and E2F targets</span>
<span id="cb10-1578"><a href="#cb10-1578" aria-hidden="true" tabindex="-1"></a>for the score calculated in the regressed data in a single sample manner. </span>
<span id="cb10-1579"><a href="#cb10-1579" aria-hidden="true" tabindex="-1"></a>We see a positive correlation between the two scores. Moreover, it is interesting</span>
<span id="cb10-1580"><a href="#cb10-1580" aria-hidden="true" tabindex="-1"></a>to see that when the tumor has low Ki67, there is still a spectrum on </span>
<span id="cb10-1581"><a href="#cb10-1581" aria-hidden="true" tabindex="-1"></a>the E2F targets score. Also there is no difference in the correlation </span>
<span id="cb10-1582"><a href="#cb10-1582" aria-hidden="true" tabindex="-1"></a>between responders and non responders.</span>
<span id="cb10-1583"><a href="#cb10-1583" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1584"><a href="#cb10-1584" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=8, fig.height=5}</span></span>
<span id="cb10-1585"><a href="#cb10-1585" aria-hidden="true" tabindex="-1"></a>df_pca_scores <span class="sc">%&gt;%</span></span>
<span id="cb10-1586"><a href="#cb10-1586" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb10-1587"><a href="#cb10-1587" aria-hidden="true" tabindex="-1"></a>        cohort <span class="sc">==</span> <span class="st">"poetic"</span> <span class="sc">&amp;</span> </span>
<span id="cb10-1588"><a href="#cb10-1588" aria-hidden="true" tabindex="-1"></a>            timepoint <span class="sc">==</span> <span class="st">"baseline"</span> <span class="sc">&amp;</span> </span>
<span id="cb10-1589"><a href="#cb10-1589" aria-hidden="true" tabindex="-1"></a>            is_responder <span class="sc">!=</span> <span class="st">"not_available"</span></span>
<span id="cb10-1590"><a href="#cb10-1590" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb10-1591"><a href="#cb10-1591" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(</span>
<span id="cb10-1592"><a href="#cb10-1592" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> is_responder, </span>
<span id="cb10-1593"><a href="#cb10-1593" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> ki67,</span>
<span id="cb10-1594"><a href="#cb10-1594" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> HALLMARK_G2M_CHECKPOINT_new</span>
<span id="cb10-1595"><a href="#cb10-1595" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">+</span></span>
<span id="cb10-1596"><a href="#cb10-1596" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb10-1597"><a href="#cb10-1597" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"gam"</span>, <span class="at">formula =</span> y <span class="sc">~</span> <span class="fu">s</span>(x, <span class="at">bs =</span> <span class="st">"cs"</span>)) <span class="sc">+</span></span>
<span id="cb10-1598"><a href="#cb10-1598" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb10-1599"><a href="#cb10-1599" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"Ki67%"</span>,</span>
<span id="cb10-1600"><a href="#cb10-1600" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"E2F targets"</span>,</span>
<span id="cb10-1601"><a href="#cb10-1601" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">"Scores calculated after regressing the data"</span></span>
<span id="cb10-1602"><a href="#cb10-1602" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb10-1603"><a href="#cb10-1603" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>) <span class="sc">+</span></span>
<span id="cb10-1604"><a href="#cb10-1604" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>()</span>
<span id="cb10-1605"><a href="#cb10-1605" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-1606"><a href="#cb10-1606" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1607"><a href="#cb10-1607" aria-hidden="true" tabindex="-1"></a>And lastly we compare the estrogen signaling signatures between</span>
<span id="cb10-1608"><a href="#cb10-1608" aria-hidden="true" tabindex="-1"></a>responders and non-responder.</span>
<span id="cb10-1609"><a href="#cb10-1609" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1610"><a href="#cb10-1610" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=6}</span></span>
<span id="cb10-1611"><a href="#cb10-1611" aria-hidden="true" tabindex="-1"></a>plot_responders_non_responder_comparison <span class="ot">&lt;-</span> <span class="cf">function</span>(</span>
<span id="cb10-1612"><a href="#cb10-1612" aria-hidden="true" tabindex="-1"></a>    df_pca_scores, </span>
<span id="cb10-1613"><a href="#cb10-1613" aria-hidden="true" tabindex="-1"></a>    <span class="at">og_or_new =</span> <span class="st">"new"</span>,</span>
<span id="cb10-1614"><a href="#cb10-1614" aria-hidden="true" tabindex="-1"></a>    <span class="at">pathways =</span> <span class="fu">c</span>(</span>
<span id="cb10-1615"><a href="#cb10-1615" aria-hidden="true" tabindex="-1"></a>        <span class="st">"SET_ERPR"</span> <span class="ot">=</span> <span class="st">"SET ER/PR"</span>, </span>
<span id="cb10-1616"><a href="#cb10-1616" aria-hidden="true" tabindex="-1"></a>        <span class="st">"HALLMARK_ESTROGEN_RESPONSE_EARLY"</span> <span class="ot">=</span> <span class="st">"Estrogen early"</span></span>
<span id="cb10-1617"><a href="#cb10-1617" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb10-1618"><a href="#cb10-1618" aria-hidden="true" tabindex="-1"></a>){</span>
<span id="cb10-1619"><a href="#cb10-1619" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb10-1620"><a href="#cb10-1620" aria-hidden="true" tabindex="-1"></a>    pathways_labeller <span class="ot">&lt;-</span> pathways</span>
<span id="cb10-1621"><a href="#cb10-1621" aria-hidden="true" tabindex="-1"></a>    <span class="fu">names</span>(pathways_labeller) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(</span>
<span id="cb10-1622"><a href="#cb10-1622" aria-hidden="true" tabindex="-1"></a>        <span class="fu">names</span>(pathways), <span class="st">"_"</span>, og_or_new</span>
<span id="cb10-1623"><a href="#cb10-1623" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb10-1624"><a href="#cb10-1624" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb10-1625"><a href="#cb10-1625" aria-hidden="true" tabindex="-1"></a>    df_pca_scores <span class="sc">%&gt;%</span></span>
<span id="cb10-1626"><a href="#cb10-1626" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb10-1627"><a href="#cb10-1627" aria-hidden="true" tabindex="-1"></a>            cohort <span class="sc">==</span> <span class="st">"poetic"</span> <span class="sc">&amp;</span> </span>
<span id="cb10-1628"><a href="#cb10-1628" aria-hidden="true" tabindex="-1"></a>                timepoint <span class="sc">==</span> <span class="st">"baseline"</span> <span class="sc">&amp;</span> </span>
<span id="cb10-1629"><a href="#cb10-1629" aria-hidden="true" tabindex="-1"></a>                is_responder <span class="sc">!=</span> <span class="st">"not_available"</span></span>
<span id="cb10-1630"><a href="#cb10-1630" aria-hidden="true" tabindex="-1"></a>        ) <span class="sc">%&gt;%</span></span>
<span id="cb10-1631"><a href="#cb10-1631" aria-hidden="true" tabindex="-1"></a>        tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(</span>
<span id="cb10-1632"><a href="#cb10-1632" aria-hidden="true" tabindex="-1"></a>            <span class="at">cols =</span> dplyr<span class="sc">::</span><span class="fu">all_of</span>(</span>
<span id="cb10-1633"><a href="#cb10-1633" aria-hidden="true" tabindex="-1"></a>                <span class="fu">paste0</span>(</span>
<span id="cb10-1634"><a href="#cb10-1634" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">names</span>(pathways),</span>
<span id="cb10-1635"><a href="#cb10-1635" aria-hidden="true" tabindex="-1"></a>                    <span class="st">"_"</span>,</span>
<span id="cb10-1636"><a href="#cb10-1636" aria-hidden="true" tabindex="-1"></a>                    og_or_new</span>
<span id="cb10-1637"><a href="#cb10-1637" aria-hidden="true" tabindex="-1"></a>                )</span>
<span id="cb10-1638"><a href="#cb10-1638" aria-hidden="true" tabindex="-1"></a>            ),</span>
<span id="cb10-1639"><a href="#cb10-1639" aria-hidden="true" tabindex="-1"></a>            <span class="at">names_to =</span> <span class="st">"pathway"</span>,</span>
<span id="cb10-1640"><a href="#cb10-1640" aria-hidden="true" tabindex="-1"></a>            <span class="at">values_to =</span> <span class="st">"score"</span></span>
<span id="cb10-1641"><a href="#cb10-1641" aria-hidden="true" tabindex="-1"></a>        ) <span class="sc">%&gt;%</span></span>
<span id="cb10-1642"><a href="#cb10-1642" aria-hidden="true" tabindex="-1"></a>        ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(</span>
<span id="cb10-1643"><a href="#cb10-1643" aria-hidden="true" tabindex="-1"></a>            <span class="at">color =</span> is_responder, </span>
<span id="cb10-1644"><a href="#cb10-1644" aria-hidden="true" tabindex="-1"></a>            <span class="at">y =</span> score,</span>
<span id="cb10-1645"><a href="#cb10-1645" aria-hidden="true" tabindex="-1"></a>            <span class="at">x =</span> is_responder</span>
<span id="cb10-1646"><a href="#cb10-1646" aria-hidden="true" tabindex="-1"></a>        )) <span class="sc">+</span></span>
<span id="cb10-1647"><a href="#cb10-1647" aria-hidden="true" tabindex="-1"></a>        ggplot2<span class="sc">::</span><span class="fu">geom_boxplot</span>(<span class="at">outlier.shape =</span> <span class="cn">NA</span>) <span class="sc">+</span> </span>
<span id="cb10-1648"><a href="#cb10-1648" aria-hidden="true" tabindex="-1"></a>        ggplot2<span class="sc">::</span><span class="fu">geom_jitter</span>() <span class="sc">+</span></span>
<span id="cb10-1649"><a href="#cb10-1649" aria-hidden="true" tabindex="-1"></a>        ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb10-1650"><a href="#cb10-1650" aria-hidden="true" tabindex="-1"></a>            <span class="at">x =</span> <span class="st">"Is responder?"</span>,</span>
<span id="cb10-1651"><a href="#cb10-1651" aria-hidden="true" tabindex="-1"></a>            <span class="at">y =</span> <span class="fu">ifelse</span>(og_or_new <span class="sc">==</span> <span class="st">"new"</span>, </span>
<span id="cb10-1652"><a href="#cb10-1652" aria-hidden="true" tabindex="-1"></a>                       <span class="st">"Scores with regressed data"</span>,</span>
<span id="cb10-1653"><a href="#cb10-1653" aria-hidden="true" tabindex="-1"></a>                       <span class="st">"GSVA scores"</span>)</span>
<span id="cb10-1654"><a href="#cb10-1654" aria-hidden="true" tabindex="-1"></a>        ) <span class="sc">+</span></span>
<span id="cb10-1655"><a href="#cb10-1655" aria-hidden="true" tabindex="-1"></a>        ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(</span>
<span id="cb10-1656"><a href="#cb10-1656" aria-hidden="true" tabindex="-1"></a>            <span class="sc">~</span>pathway,</span>
<span id="cb10-1657"><a href="#cb10-1657" aria-hidden="true" tabindex="-1"></a>            <span class="at">labeller =</span> <span class="fu">as_labeller</span>(pathways_labeller),</span>
<span id="cb10-1658"><a href="#cb10-1658" aria-hidden="true" tabindex="-1"></a>            <span class="at">scales =</span> <span class="st">"free_y"</span></span>
<span id="cb10-1659"><a href="#cb10-1659" aria-hidden="true" tabindex="-1"></a>        ) <span class="sc">+</span> </span>
<span id="cb10-1660"><a href="#cb10-1660" aria-hidden="true" tabindex="-1"></a>        ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">30</span>) <span class="sc">+</span></span>
<span id="cb10-1661"><a href="#cb10-1661" aria-hidden="true" tabindex="-1"></a>        ggplot2<span class="sc">::</span><span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>) <span class="sc">+</span> </span>
<span id="cb10-1662"><a href="#cb10-1662" aria-hidden="true" tabindex="-1"></a>        <span class="fu">change_plot_aes_point</span>()</span>
<span id="cb10-1663"><a href="#cb10-1663" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb10-1664"><a href="#cb10-1664" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1665"><a href="#cb10-1665" aria-hidden="true" tabindex="-1"></a>regressed_plot_er <span class="ot">&lt;-</span> <span class="fu">plot_responders_non_responder_comparison</span>(df_pca_scores)</span>
<span id="cb10-1666"><a href="#cb10-1666" aria-hidden="true" tabindex="-1"></a>regressed_plot_er</span>
<span id="cb10-1667"><a href="#cb10-1667" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-1668"><a href="#cb10-1668" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1669"><a href="#cb10-1669" aria-hidden="true" tabindex="-1"></a>We see that the estrogen response early has a wider range when compared to the</span>
<span id="cb10-1670"><a href="#cb10-1670" aria-hidden="true" tabindex="-1"></a>SET ER/PR score. That makes sense as this signature has over 100 genes,</span>
<span id="cb10-1671"><a href="#cb10-1671" aria-hidden="true" tabindex="-1"></a>whereas SET ER/PR has only 18. Also the responder</span>
<span id="cb10-1672"><a href="#cb10-1672" aria-hidden="true" tabindex="-1"></a>was very good in discriminating the responders and non responders.</span>
<span id="cb10-1673"><a href="#cb10-1673" aria-hidden="true" tabindex="-1"></a>What we can conclude from this figure is that the samples</span>
<span id="cb10-1674"><a href="#cb10-1674" aria-hidden="true" tabindex="-1"></a>with very low scores are usually non responders, which reflect</span>
<span id="cb10-1675"><a href="#cb10-1675" aria-hidden="true" tabindex="-1"></a>their position in the molecular landscape, also in general </span>
<span id="cb10-1676"><a href="#cb10-1676" aria-hidden="true" tabindex="-1"></a>responders have higher ER signaling than non responders.</span>
<span id="cb10-1677"><a href="#cb10-1677" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1678"><a href="#cb10-1678" aria-hidden="true" tabindex="-1"></a>Below is the figure with the GSVA scores before regressing the data, </span>
<span id="cb10-1679"><a href="#cb10-1679" aria-hidden="true" tabindex="-1"></a>we see that SET ER/PR is not able to really distinguish </span>
<span id="cb10-1680"><a href="#cb10-1680" aria-hidden="true" tabindex="-1"></a>completely the non-responders with very low score similar</span>
<span id="cb10-1681"><a href="#cb10-1681" aria-hidden="true" tabindex="-1"></a>to what was done with GSVA. Even estrogen early had</span>
<span id="cb10-1682"><a href="#cb10-1682" aria-hidden="true" tabindex="-1"></a>some differences.</span>
<span id="cb10-1683"><a href="#cb10-1683" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1684"><a href="#cb10-1684" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=6}</span></span>
<span id="cb10-1685"><a href="#cb10-1685" aria-hidden="true" tabindex="-1"></a>gsva_plot_er <span class="ot">&lt;-</span> <span class="fu">plot_responders_non_responder_comparison</span>(</span>
<span id="cb10-1686"><a href="#cb10-1686" aria-hidden="true" tabindex="-1"></a>    df_pca_scores,</span>
<span id="cb10-1687"><a href="#cb10-1687" aria-hidden="true" tabindex="-1"></a>    <span class="at">og_or_new =</span> <span class="st">"og"</span></span>
<span id="cb10-1688"><a href="#cb10-1688" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-1689"><a href="#cb10-1689" aria-hidden="true" tabindex="-1"></a>gsva_plot_er</span>
<span id="cb10-1690"><a href="#cb10-1690" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-1691"><a href="#cb10-1691" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1692"><a href="#cb10-1692" aria-hidden="true" tabindex="-1"></a>And below is a figure with both GSVA and regressed scores together.</span>
<span id="cb10-1693"><a href="#cb10-1693" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1694"><a href="#cb10-1694" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=10}</span></span>
<span id="cb10-1695"><a href="#cb10-1695" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-comparison-scoring-strategy-poetic</span></span>
<span id="cb10-1696"><a href="#cb10-1696" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Comparing scoring strategies for responders and non responders</span></span>
<span id="cb10-1697"><a href="#cb10-1697" aria-hidden="true" tabindex="-1"></a><span class="co">#|     in the POETIC dataset. </span></span>
<span id="cb10-1698"><a href="#cb10-1698" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(</span>
<span id="cb10-1699"><a href="#cb10-1699" aria-hidden="true" tabindex="-1"></a>    regressed_plot_er,</span>
<span id="cb10-1700"><a href="#cb10-1700" aria-hidden="true" tabindex="-1"></a>    gsva_plot_er,</span>
<span id="cb10-1701"><a href="#cb10-1701" aria-hidden="true" tabindex="-1"></a>    <span class="at">ncol =</span> <span class="dv">1</span></span>
<span id="cb10-1702"><a href="#cb10-1702" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-1703"><a href="#cb10-1703" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-1704"><a href="#cb10-1704" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1705"><a href="#cb10-1705" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1706"><a href="#cb10-1706" aria-hidden="true" tabindex="-1"></a>Below we show the correlation between</span>
<span id="cb10-1707"><a href="#cb10-1707" aria-hidden="true" tabindex="-1"></a>these scores for all cohorts.</span>
<span id="cb10-1708"><a href="#cb10-1708" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1709"><a href="#cb10-1709" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=6, fig.height=8}</span></span>
<span id="cb10-1710"><a href="#cb10-1710" aria-hidden="true" tabindex="-1"></a>df_pca_scores <span class="sc">%&gt;%</span></span>
<span id="cb10-1711"><a href="#cb10-1711" aria-hidden="true" tabindex="-1"></a>    <span class="co">#dplyr::filter(cohort != "poetic") %&gt;%</span></span>
<span id="cb10-1712"><a href="#cb10-1712" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(</span>
<span id="cb10-1713"><a href="#cb10-1713" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> HALLMARK_ESTROGEN_RESPONSE_EARLY_new, </span>
<span id="cb10-1714"><a href="#cb10-1714" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> SET_ERPR_new,</span>
<span id="cb10-1715"><a href="#cb10-1715" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> pam50</span>
<span id="cb10-1716"><a href="#cb10-1716" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">+</span></span>
<span id="cb10-1717"><a href="#cb10-1717" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span> </span>
<span id="cb10-1718"><a href="#cb10-1718" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_smooth</span>(</span>
<span id="cb10-1719"><a href="#cb10-1719" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">"black"</span>, </span>
<span id="cb10-1720"><a href="#cb10-1720" aria-hidden="true" tabindex="-1"></a>        <span class="at">alpha =</span> <span class="fl">0.5</span>,</span>
<span id="cb10-1721"><a href="#cb10-1721" aria-hidden="true" tabindex="-1"></a>        <span class="at">method =</span> <span class="st">"gam"</span>,</span>
<span id="cb10-1722"><a href="#cb10-1722" aria-hidden="true" tabindex="-1"></a>        <span class="at">formula =</span> y <span class="sc">~</span> <span class="fu">s</span>(x, <span class="at">bs =</span> <span class="st">"cs"</span>)</span>
<span id="cb10-1723"><a href="#cb10-1723" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb10-1724"><a href="#cb10-1724" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>cohort, <span class="at">ncol =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb10-1725"><a href="#cb10-1725" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>) <span class="sc">+</span> </span>
<span id="cb10-1726"><a href="#cb10-1726" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_color_manual</span>(</span>
<span id="cb10-1727"><a href="#cb10-1727" aria-hidden="true" tabindex="-1"></a>        <span class="at">values =</span> <span class="fu">get_colors_pam50</span>(</span>
<span id="cb10-1728"><a href="#cb10-1728" aria-hidden="true" tabindex="-1"></a>            df_pca_scores <span class="sc">%&gt;%</span></span>
<span id="cb10-1729"><a href="#cb10-1729" aria-hidden="true" tabindex="-1"></a>                dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">!=</span> <span class="st">"poetic"</span>)</span>
<span id="cb10-1730"><a href="#cb10-1730" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb10-1731"><a href="#cb10-1731" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb10-1732"><a href="#cb10-1732" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Estrogen Early"</span>, <span class="at">y =</span> <span class="st">"SET ER/PR"</span>) <span class="sc">+</span> </span>
<span id="cb10-1733"><a href="#cb10-1733" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>() <span class="sc">+</span> </span>
<span id="cb10-1734"><a href="#cb10-1734" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>()</span>
<span id="cb10-1735"><a href="#cb10-1735" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-1736"><a href="#cb10-1736" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1737"><a href="#cb10-1737" aria-hidden="true" tabindex="-1"></a>These scores are highly correlated but in different absolute</span>
<span id="cb10-1738"><a href="#cb10-1738" aria-hidden="true" tabindex="-1"></a>scales. We now check the change in Ki67 for all patients and </span>
<span id="cb10-1739"><a href="#cb10-1739" aria-hidden="true" tabindex="-1"></a>compare the scores with the original ones obtained with GSVA.</span>
<span id="cb10-1740"><a href="#cb10-1740" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1741"><a href="#cb10-1741" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=12, fig.height=6}</span></span>
<span id="cb10-1742"><a href="#cb10-1742" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-comparison-og-regressed-er</span></span>
<span id="cb10-1743"><a href="#cb10-1743" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Comparison of original GSVA and regressed scores between responders</span></span>
<span id="cb10-1744"><a href="#cb10-1744" aria-hidden="true" tabindex="-1"></a><span class="co">#|     and non responders compared to change in Ki67.</span></span>
<span id="cb10-1745"><a href="#cb10-1745" aria-hidden="true" tabindex="-1"></a>df_pca_scores <span class="sc">%&gt;%</span></span>
<span id="cb10-1746"><a href="#cb10-1746" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb10-1747"><a href="#cb10-1747" aria-hidden="true" tabindex="-1"></a>        cohort <span class="sc">==</span> <span class="st">"poetic"</span> <span class="sc">&amp;</span> </span>
<span id="cb10-1748"><a href="#cb10-1748" aria-hidden="true" tabindex="-1"></a>            timepoint <span class="sc">==</span> <span class="st">"baseline"</span> <span class="sc">&amp;</span> </span>
<span id="cb10-1749"><a href="#cb10-1749" aria-hidden="true" tabindex="-1"></a>            is_responder <span class="sc">!=</span> <span class="st">"not_available"</span></span>
<span id="cb10-1750"><a href="#cb10-1750" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb10-1751"><a href="#cb10-1751" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(</span>
<span id="cb10-1752"><a href="#cb10-1752" aria-hidden="true" tabindex="-1"></a>        <span class="at">cols =</span> <span class="fu">c</span>(</span>
<span id="cb10-1753"><a href="#cb10-1753" aria-hidden="true" tabindex="-1"></a>            <span class="st">"HALLMARK_ESTROGEN_RESPONSE_EARLY_og"</span>,</span>
<span id="cb10-1754"><a href="#cb10-1754" aria-hidden="true" tabindex="-1"></a>            <span class="st">"HALLMARK_ESTROGEN_RESPONSE_EARLY_new"</span>,</span>
<span id="cb10-1755"><a href="#cb10-1755" aria-hidden="true" tabindex="-1"></a>            <span class="st">"SET_ERPR_og"</span>,</span>
<span id="cb10-1756"><a href="#cb10-1756" aria-hidden="true" tabindex="-1"></a>            <span class="st">"SET_ERPR_new"</span></span>
<span id="cb10-1757"><a href="#cb10-1757" aria-hidden="true" tabindex="-1"></a>        ),</span>
<span id="cb10-1758"><a href="#cb10-1758" aria-hidden="true" tabindex="-1"></a>        <span class="at">names_to =</span> <span class="st">"pathway"</span>,</span>
<span id="cb10-1759"><a href="#cb10-1759" aria-hidden="true" tabindex="-1"></a>        <span class="at">values_to =</span> <span class="st">"score"</span></span>
<span id="cb10-1760"><a href="#cb10-1760" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb10-1761"><a href="#cb10-1761" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(</span>
<span id="cb10-1762"><a href="#cb10-1762" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> score,</span>
<span id="cb10-1763"><a href="#cb10-1763" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> change_ki67</span>
<span id="cb10-1764"><a href="#cb10-1764" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">+</span> </span>
<span id="cb10-1765"><a href="#cb10-1765" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">formula =</span> <span class="st">"y~x"</span>) <span class="sc">+</span></span>
<span id="cb10-1766"><a href="#cb10-1766" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb10-1767"><a href="#cb10-1767" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(</span>
<span id="cb10-1768"><a href="#cb10-1768" aria-hidden="true" tabindex="-1"></a>        pathway<span class="sc">~</span>is_responder, </span>
<span id="cb10-1769"><a href="#cb10-1769" aria-hidden="true" tabindex="-1"></a>        <span class="at">scales =</span> <span class="st">"free"</span>,</span>
<span id="cb10-1770"><a href="#cb10-1770" aria-hidden="true" tabindex="-1"></a>        <span class="at">ncol =</span> <span class="dv">4</span>,</span>
<span id="cb10-1771"><a href="#cb10-1771" aria-hidden="true" tabindex="-1"></a>        <span class="at">labeller =</span> <span class="fu">as_labeller</span>(<span class="fu">c</span>(</span>
<span id="cb10-1772"><a href="#cb10-1772" aria-hidden="true" tabindex="-1"></a>            <span class="st">"non_responder"</span> <span class="ot">=</span> <span class="st">"Non-responder"</span>,</span>
<span id="cb10-1773"><a href="#cb10-1773" aria-hidden="true" tabindex="-1"></a>            <span class="st">"responder"</span> <span class="ot">=</span> <span class="st">"Responder"</span>,</span>
<span id="cb10-1774"><a href="#cb10-1774" aria-hidden="true" tabindex="-1"></a>            <span class="st">"HALLMARK_ESTROGEN_RESPONSE_EARLY_og"</span> <span class="ot">=</span> <span class="st">"GSVA estrogen early"</span>,</span>
<span id="cb10-1775"><a href="#cb10-1775" aria-hidden="true" tabindex="-1"></a>            <span class="st">"HALLMARK_ESTROGEN_RESPONSE_EARLY_new"</span> <span class="ot">=</span> <span class="st">"Regressed estrogen early"</span>,</span>
<span id="cb10-1776"><a href="#cb10-1776" aria-hidden="true" tabindex="-1"></a>            <span class="st">"SET_ERPR_og"</span> <span class="ot">=</span> <span class="st">"GSVA SET ER/PR"</span>,</span>
<span id="cb10-1777"><a href="#cb10-1777" aria-hidden="true" tabindex="-1"></a>            <span class="st">"SET_ERPR_new"</span> <span class="ot">=</span> <span class="st">"Regressed SET ER/PR"</span></span>
<span id="cb10-1778"><a href="#cb10-1778" aria-hidden="true" tabindex="-1"></a>        ))</span>
<span id="cb10-1779"><a href="#cb10-1779" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb10-1780"><a href="#cb10-1780" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb10-1781"><a href="#cb10-1781" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"Scores"</span>,</span>
<span id="cb10-1782"><a href="#cb10-1782" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"Change in Ki67 (%)"</span></span>
<span id="cb10-1783"><a href="#cb10-1783" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb10-1784"><a href="#cb10-1784" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">15</span>) <span class="sc">+</span> </span>
<span id="cb10-1785"><a href="#cb10-1785" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>() <span class="sc">+</span> </span>
<span id="cb10-1786"><a href="#cb10-1786" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>()</span>
<span id="cb10-1787"><a href="#cb10-1787" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-1788"><a href="#cb10-1788" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1789"><a href="#cb10-1789" aria-hidden="true" tabindex="-1"></a>We see that the bigger the change the higher the score is when using the</span>
<span id="cb10-1790"><a href="#cb10-1790" aria-hidden="true" tabindex="-1"></a>regressed data. Also the non responders tend to have scores mostly in</span>
<span id="cb10-1791"><a href="#cb10-1791" aria-hidden="true" tabindex="-1"></a>the negative scale, whereas the responders are more shifted to the</span>
<span id="cb10-1792"><a href="#cb10-1792" aria-hidden="true" tabindex="-1"></a>right. SET ER/PR is also a good measure here. </span>
<span id="cb10-1793"><a href="#cb10-1793" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1796"><a href="#cb10-1796" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-1797"><a href="#cb10-1797" aria-hidden="true" tabindex="-1"></a>df_pca_scores <span class="sc">%&gt;%</span></span>
<span id="cb10-1798"><a href="#cb10-1798" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb10-1799"><a href="#cb10-1799" aria-hidden="true" tabindex="-1"></a>        cohort <span class="sc">==</span> <span class="st">"poetic"</span> <span class="sc">&amp;</span> </span>
<span id="cb10-1800"><a href="#cb10-1800" aria-hidden="true" tabindex="-1"></a>            timepoint <span class="sc">==</span> <span class="st">"baseline"</span> <span class="sc">&amp;</span> </span>
<span id="cb10-1801"><a href="#cb10-1801" aria-hidden="true" tabindex="-1"></a>            is_responder <span class="sc">!=</span> <span class="st">"not_available"</span></span>
<span id="cb10-1802"><a href="#cb10-1802" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb10-1803"><a href="#cb10-1803" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(</span>
<span id="cb10-1804"><a href="#cb10-1804" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> HALLMARK_ESTROGEN_RESPONSE_EARLY_og,</span>
<span id="cb10-1805"><a href="#cb10-1805" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> HALLMARK_ESTROGEN_RESPONSE_EARLY_new,</span>
<span id="cb10-1806"><a href="#cb10-1806" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> is_responder</span>
<span id="cb10-1807"><a href="#cb10-1807" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">+</span> </span>
<span id="cb10-1808"><a href="#cb10-1808" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_smooth</span>(</span>
<span id="cb10-1809"><a href="#cb10-1809" aria-hidden="true" tabindex="-1"></a>        <span class="at">method =</span> <span class="st">"gam"</span>,</span>
<span id="cb10-1810"><a href="#cb10-1810" aria-hidden="true" tabindex="-1"></a>        <span class="at">formula =</span> y <span class="sc">~</span> <span class="fu">s</span>(x, <span class="at">bs =</span> <span class="st">"cs"</span>), </span>
<span id="cb10-1811"><a href="#cb10-1811" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">"black"</span></span>
<span id="cb10-1812"><a href="#cb10-1812" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb10-1813"><a href="#cb10-1813" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb10-1814"><a href="#cb10-1814" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb10-1815"><a href="#cb10-1815" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"GSVA Estrogen early"</span>,</span>
<span id="cb10-1816"><a href="#cb10-1816" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"Regressed Estrogen early"</span></span>
<span id="cb10-1817"><a href="#cb10-1817" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb10-1818"><a href="#cb10-1818" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">15</span>) <span class="sc">+</span> </span>
<span id="cb10-1819"><a href="#cb10-1819" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>() <span class="sc">+</span> </span>
<span id="cb10-1820"><a href="#cb10-1820" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>()</span>
<span id="cb10-1821"><a href="#cb10-1821" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-1822"><a href="#cb10-1822" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1823"><a href="#cb10-1823" aria-hidden="true" tabindex="-1"></a>There is a very good match between the two scores in general,</span>
<span id="cb10-1824"><a href="#cb10-1824" aria-hidden="true" tabindex="-1"></a>meaning that the regression scoring strategy capture relatively</span>
<span id="cb10-1825"><a href="#cb10-1825" aria-hidden="true" tabindex="-1"></a>well the scores. There is one patient that is a non-responder</span>
<span id="cb10-1826"><a href="#cb10-1826" aria-hidden="true" tabindex="-1"></a>and has a low regressed estrogen early but a GSVA score </span>
<span id="cb10-1827"><a href="#cb10-1827" aria-hidden="true" tabindex="-1"></a>close to 0, let us check this patient.</span>
<span id="cb10-1828"><a href="#cb10-1828" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1831"><a href="#cb10-1831" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-1832"><a href="#cb10-1832" aria-hidden="true" tabindex="-1"></a>df_pca_scores <span class="sc">%&gt;%</span></span>
<span id="cb10-1833"><a href="#cb10-1833" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb10-1834"><a href="#cb10-1834" aria-hidden="true" tabindex="-1"></a>        is_responder <span class="sc">==</span> <span class="st">"non_responder"</span> <span class="sc">&amp;</span> </span>
<span id="cb10-1835"><a href="#cb10-1835" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">between</span>(HALLMARK_ESTROGEN_RESPONSE_EARLY_og, <span class="sc">-</span><span class="fl">0.1</span>, <span class="fl">0.1</span>) <span class="sc">&amp;</span></span>
<span id="cb10-1836"><a href="#cb10-1836" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">between</span>(HALLMARK_ESTROGEN_RESPONSE_EARLY_new, <span class="sc">-</span><span class="dv">20</span>, <span class="sc">-</span><span class="dv">15</span>) </span>
<span id="cb10-1837"><a href="#cb10-1837" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb10-1838"><a href="#cb10-1838" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(patient_nb, ki67, PC3, change_ki67, er_status) <span class="sc">%&gt;%</span></span>
<span id="cb10-1839"><a href="#cb10-1839" aria-hidden="true" tabindex="-1"></a>    DT<span class="sc">::</span><span class="fu">datatable</span>()</span>
<span id="cb10-1840"><a href="#cb10-1840" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-1841"><a href="#cb10-1841" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1842"><a href="#cb10-1842" aria-hidden="true" tabindex="-1"></a>It is a patient that is far to the left on the molecular landscape, among the</span>
<span id="cb10-1843"><a href="#cb10-1843" aria-hidden="true" tabindex="-1"></a>basal like, and had an increase in Ki67 upon ET. So the </span>
<span id="cb10-1844"><a href="#cb10-1844" aria-hidden="true" tabindex="-1"></a>new scoring system was able to capture this patient as with low score. </span>
<span id="cb10-1845"><a href="#cb10-1845" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1846"><a href="#cb10-1846" aria-hidden="true" tabindex="-1"></a><span class="fu">### Survival analysis with the new scores</span></span>
<span id="cb10-1847"><a href="#cb10-1847" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1848"><a href="#cb10-1848" aria-hidden="true" tabindex="-1"></a>We can check how robust these scores are by performing survival analysis.</span>
<span id="cb10-1849"><a href="#cb10-1849" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1850"><a href="#cb10-1850" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb10-1851"><a href="#cb10-1851" aria-hidden="true" tabindex="-1"></a>tcga_samples <span class="ot">&lt;-</span> <span class="fu">colData</span>(datasets<span class="sc">$</span>tcga) <span class="sc">%&gt;%</span> data.frame <span class="sc">%&gt;%</span></span>
<span id="cb10-1852"><a href="#cb10-1852" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb10-1853"><a href="#cb10-1853" aria-hidden="true" tabindex="-1"></a>        pam50 <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"luma"</span>, <span class="st">"lumb"</span>) <span class="sc">&amp;</span></span>
<span id="cb10-1854"><a href="#cb10-1854" aria-hidden="true" tabindex="-1"></a>        node_stage <span class="sc">!=</span> <span class="st">"NX"</span> <span class="sc">&amp;</span></span>
<span id="cb10-1855"><a href="#cb10-1855" aria-hidden="true" tabindex="-1"></a>        <span class="sc">!</span>(tumor_stage <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"stage_iv"</span>, <span class="st">"na"</span>)) <span class="sc">&amp;</span> </span>
<span id="cb10-1856"><a href="#cb10-1856" aria-hidden="true" tabindex="-1"></a>        er_status <span class="sc">==</span> <span class="st">"pos"</span></span>
<span id="cb10-1857"><a href="#cb10-1857" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span> </span>
<span id="cb10-1858"><a href="#cb10-1858" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">pull</span>(sample_name)</span>
<span id="cb10-1859"><a href="#cb10-1859" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1860"><a href="#cb10-1860" aria-hidden="true" tabindex="-1"></a>scanb_samples <span class="ot">&lt;-</span> <span class="fu">colData</span>(datasets<span class="sc">$</span>scanb) <span class="sc">%&gt;%</span> data.frame <span class="sc">%&gt;%</span></span>
<span id="cb10-1861"><a href="#cb10-1861" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb10-1862"><a href="#cb10-1862" aria-hidden="true" tabindex="-1"></a>        node_stage <span class="sc">!=</span> <span class="st">"NX"</span> <span class="sc">&amp;</span></span>
<span id="cb10-1863"><a href="#cb10-1863" aria-hidden="true" tabindex="-1"></a>        <span class="sc">!</span>(tumor_stage <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"T4"</span>, <span class="st">"Tis"</span>, <span class="st">"TX"</span>)) <span class="sc">&amp;</span> </span>
<span id="cb10-1864"><a href="#cb10-1864" aria-hidden="true" tabindex="-1"></a>        er_status <span class="sc">==</span> <span class="st">"pos"</span> <span class="sc">&amp;</span> </span>
<span id="cb10-1865"><a href="#cb10-1865" aria-hidden="true" tabindex="-1"></a>        TreatGroup <span class="sc">==</span> <span class="st">"Endo"</span></span>
<span id="cb10-1866"><a href="#cb10-1866" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span> </span>
<span id="cb10-1867"><a href="#cb10-1867" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">pull</span>(sample_name)</span>
<span id="cb10-1868"><a href="#cb10-1868" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1869"><a href="#cb10-1869" aria-hidden="true" tabindex="-1"></a>metabric_samples <span class="ot">&lt;-</span> <span class="fu">colData</span>(datasets<span class="sc">$</span>metabric) <span class="sc">%&gt;%</span> data.frame <span class="sc">%&gt;%</span></span>
<span id="cb10-1870"><a href="#cb10-1870" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb10-1871"><a href="#cb10-1871" aria-hidden="true" tabindex="-1"></a>        CHEMOTHERAPY <span class="sc">==</span> <span class="st">"NO"</span> <span class="sc">&amp;</span></span>
<span id="cb10-1872"><a href="#cb10-1872" aria-hidden="true" tabindex="-1"></a>        er_status <span class="sc">==</span> <span class="st">"pos"</span> <span class="sc">&amp;</span> </span>
<span id="cb10-1873"><a href="#cb10-1873" aria-hidden="true" tabindex="-1"></a>        HORMONE_THERAPY <span class="sc">==</span> <span class="st">"YES"</span></span>
<span id="cb10-1874"><a href="#cb10-1874" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span> </span>
<span id="cb10-1875"><a href="#cb10-1875" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">pull</span>(sample_name)</span>
<span id="cb10-1876"><a href="#cb10-1876" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1877"><a href="#cb10-1877" aria-hidden="true" tabindex="-1"></a>formulas_survival <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb10-1878"><a href="#cb10-1878" aria-hidden="true" tabindex="-1"></a>    <span class="st">"os_tcga"</span> <span class="ot">=</span> <span class="st">"Surv(os_months, os_status) ~ age + node_stage + tumor_stage"</span>,</span>
<span id="cb10-1879"><a href="#cb10-1879" aria-hidden="true" tabindex="-1"></a>    <span class="st">"os_scanb"</span> <span class="ot">=</span> <span class="st">"Surv(os_months, os_status) ~ age + node_stage + tumor_stage"</span>,</span>
<span id="cb10-1880"><a href="#cb10-1880" aria-hidden="true" tabindex="-1"></a>    <span class="st">"os_metabric"</span> <span class="ot">=</span> <span class="st">"Surv(os_months, os_status) ~ age + npi"</span>,</span>
<span id="cb10-1881"><a href="#cb10-1881" aria-hidden="true" tabindex="-1"></a>    <span class="st">"rfs_metabric"</span> <span class="ot">=</span> <span class="st">"Surv(rfs_months, rfs_status) ~ age + npi"</span>,</span>
<span id="cb10-1882"><a href="#cb10-1882" aria-hidden="true" tabindex="-1"></a>    <span class="st">"rfs_scanb"</span> <span class="ot">=</span> <span class="st">"Surv(rfs_months, rfs_status) ~ age + node_stage + tumor_stage"</span></span>
<span id="cb10-1883"><a href="#cb10-1883" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-1884"><a href="#cb10-1884" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1885"><a href="#cb10-1885" aria-hidden="true" tabindex="-1"></a>type_survival <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"os"</span>, <span class="st">"rfs"</span>)</span>
<span id="cb10-1886"><a href="#cb10-1886" aria-hidden="true" tabindex="-1"></a>which_scores <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb10-1887"><a href="#cb10-1887" aria-hidden="true" tabindex="-1"></a>    <span class="fu">paste0</span>(<span class="fu">c</span>(</span>
<span id="cb10-1888"><a href="#cb10-1888" aria-hidden="true" tabindex="-1"></a>        <span class="st">"HALLMARK_G2M_CHECKPOINT"</span>,</span>
<span id="cb10-1889"><a href="#cb10-1889" aria-hidden="true" tabindex="-1"></a>        <span class="st">"HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION"</span>,</span>
<span id="cb10-1890"><a href="#cb10-1890" aria-hidden="true" tabindex="-1"></a>        <span class="st">"HALLMARK_DNA_REPAIR"</span>,</span>
<span id="cb10-1891"><a href="#cb10-1891" aria-hidden="true" tabindex="-1"></a>        <span class="st">"HALLMARK_ANDROGEN_RESPONSE"</span>,</span>
<span id="cb10-1892"><a href="#cb10-1892" aria-hidden="true" tabindex="-1"></a>        <span class="st">"HALLMARK_PI3K_AKT_MTOR_SIGNALING"</span>,</span>
<span id="cb10-1893"><a href="#cb10-1893" aria-hidden="true" tabindex="-1"></a>        <span class="st">"SET_ERPR"</span>,</span>
<span id="cb10-1894"><a href="#cb10-1894" aria-hidden="true" tabindex="-1"></a>        <span class="st">"HALLMARK_ESTROGEN_RESPONSE_EARLY"</span></span>
<span id="cb10-1895"><a href="#cb10-1895" aria-hidden="true" tabindex="-1"></a>    ), <span class="st">"_new"</span>)</span>
<span id="cb10-1896"><a href="#cb10-1896" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-1897"><a href="#cb10-1897" aria-hidden="true" tabindex="-1"></a>survival_results <span class="ot">&lt;-</span> <span class="fu">sapply</span>(</span>
<span id="cb10-1898"><a href="#cb10-1898" aria-hidden="true" tabindex="-1"></a>    <span class="fu">c</span>(<span class="st">"os"</span>, <span class="st">"rfs"</span>),</span>
<span id="cb10-1899"><a href="#cb10-1899" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(type_analysis, datasets, name_scores, formulas){</span>
<span id="cb10-1900"><a href="#cb10-1900" aria-hidden="true" tabindex="-1"></a>        <span class="fu">mapply</span>(</span>
<span id="cb10-1901"><a href="#cb10-1901" aria-hidden="true" tabindex="-1"></a>            <span class="cf">function</span>(</span>
<span id="cb10-1902"><a href="#cb10-1902" aria-hidden="true" tabindex="-1"></a>                dataset, name_dataset, name_scores, formulas, type_analysis</span>
<span id="cb10-1903"><a href="#cb10-1903" aria-hidden="true" tabindex="-1"></a>            ){</span>
<span id="cb10-1904"><a href="#cb10-1904" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb10-1905"><a href="#cb10-1905" aria-hidden="true" tabindex="-1"></a>                <span class="cf">if</span> (type_analysis <span class="sc">==</span> <span class="st">"rfs"</span> <span class="sc">&amp;</span> name_dataset <span class="sc">==</span> <span class="st">"tcga"</span>){</span>
<span id="cb10-1906"><a href="#cb10-1906" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">return</span>()</span>
<span id="cb10-1907"><a href="#cb10-1907" aria-hidden="true" tabindex="-1"></a>                } <span class="cf">else</span> { </span>
<span id="cb10-1908"><a href="#cb10-1908" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb10-1909"><a href="#cb10-1909" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">sapply</span>(</span>
<span id="cb10-1910"><a href="#cb10-1910" aria-hidden="true" tabindex="-1"></a>                        name_scores,</span>
<span id="cb10-1911"><a href="#cb10-1911" aria-hidden="true" tabindex="-1"></a>                        <span class="cf">function</span>(name_score, col_data, formula){</span>
<span id="cb10-1912"><a href="#cb10-1912" aria-hidden="true" tabindex="-1"></a>                            </span>
<span id="cb10-1913"><a href="#cb10-1913" aria-hidden="true" tabindex="-1"></a>                            col_data[, name_score] <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">scale</span>(</span>
<span id="cb10-1914"><a href="#cb10-1914" aria-hidden="true" tabindex="-1"></a>                                col_data[, name_score]</span>
<span id="cb10-1915"><a href="#cb10-1915" aria-hidden="true" tabindex="-1"></a>                            )</span>
<span id="cb10-1916"><a href="#cb10-1916" aria-hidden="true" tabindex="-1"></a>                            </span>
<span id="cb10-1917"><a href="#cb10-1917" aria-hidden="true" tabindex="-1"></a>                            survival<span class="sc">::</span><span class="fu">coxph</span>(</span>
<span id="cb10-1918"><a href="#cb10-1918" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">as.formula</span>(</span>
<span id="cb10-1919"><a href="#cb10-1919" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">paste</span>(formula, name_score, <span class="at">sep =</span> <span class="st">"+"</span>)</span>
<span id="cb10-1920"><a href="#cb10-1920" aria-hidden="true" tabindex="-1"></a>                                ),</span>
<span id="cb10-1921"><a href="#cb10-1921" aria-hidden="true" tabindex="-1"></a>                                <span class="at">data =</span> col_data, </span>
<span id="cb10-1922"><a href="#cb10-1922" aria-hidden="true" tabindex="-1"></a>                                <span class="at">y =</span> <span class="cn">FALSE</span>,</span>
<span id="cb10-1923"><a href="#cb10-1923" aria-hidden="true" tabindex="-1"></a>                                <span class="at">x =</span> <span class="cn">FALSE</span></span>
<span id="cb10-1924"><a href="#cb10-1924" aria-hidden="true" tabindex="-1"></a>                            )</span>
<span id="cb10-1925"><a href="#cb10-1925" aria-hidden="true" tabindex="-1"></a>                        },</span>
<span id="cb10-1926"><a href="#cb10-1926" aria-hidden="true" tabindex="-1"></a>                        <span class="at">col_data =</span> dataset,</span>
<span id="cb10-1927"><a href="#cb10-1927" aria-hidden="true" tabindex="-1"></a>                        <span class="at">formula =</span> formulas[</span>
<span id="cb10-1928"><a href="#cb10-1928" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">grepl</span>(</span>
<span id="cb10-1929"><a href="#cb10-1929" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">paste</span>(type_analysis, name_dataset, <span class="at">sep =</span> <span class="st">"_"</span>),</span>
<span id="cb10-1930"><a href="#cb10-1930" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">names</span>(formulas)</span>
<span id="cb10-1931"><a href="#cb10-1931" aria-hidden="true" tabindex="-1"></a>                            )</span>
<span id="cb10-1932"><a href="#cb10-1932" aria-hidden="true" tabindex="-1"></a>                        ],</span>
<span id="cb10-1933"><a href="#cb10-1933" aria-hidden="true" tabindex="-1"></a>                        <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>,</span>
<span id="cb10-1934"><a href="#cb10-1934" aria-hidden="true" tabindex="-1"></a>                        <span class="at">simplify =</span> <span class="cn">FALSE</span></span>
<span id="cb10-1935"><a href="#cb10-1935" aria-hidden="true" tabindex="-1"></a>                    )</span>
<span id="cb10-1936"><a href="#cb10-1936" aria-hidden="true" tabindex="-1"></a>                }</span>
<span id="cb10-1937"><a href="#cb10-1937" aria-hidden="true" tabindex="-1"></a>            },</span>
<span id="cb10-1938"><a href="#cb10-1938" aria-hidden="true" tabindex="-1"></a>            <span class="at">dataset =</span> datasets,</span>
<span id="cb10-1939"><a href="#cb10-1939" aria-hidden="true" tabindex="-1"></a>            <span class="at">name_dataset =</span> <span class="fu">names</span>(datasets),</span>
<span id="cb10-1940"><a href="#cb10-1940" aria-hidden="true" tabindex="-1"></a>            <span class="at">MoreArgs =</span> <span class="fu">list</span>(</span>
<span id="cb10-1941"><a href="#cb10-1941" aria-hidden="true" tabindex="-1"></a>                <span class="at">formulas =</span> formulas, <span class="at">name_scores =</span> name_scores,</span>
<span id="cb10-1942"><a href="#cb10-1942" aria-hidden="true" tabindex="-1"></a>                <span class="at">type_analysis =</span> type_analysis</span>
<span id="cb10-1943"><a href="#cb10-1943" aria-hidden="true" tabindex="-1"></a>            ),</span>
<span id="cb10-1944"><a href="#cb10-1944" aria-hidden="true" tabindex="-1"></a>            <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>,</span>
<span id="cb10-1945"><a href="#cb10-1945" aria-hidden="true" tabindex="-1"></a>            <span class="at">SIMPLIFY =</span> <span class="cn">FALSE</span></span>
<span id="cb10-1946"><a href="#cb10-1946" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb10-1947"><a href="#cb10-1947" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb10-1948"><a href="#cb10-1948" aria-hidden="true" tabindex="-1"></a>    <span class="at">datasets =</span> <span class="fu">lapply</span>(</span>
<span id="cb10-1949"><a href="#cb10-1949" aria-hidden="true" tabindex="-1"></a>        <span class="fu">list</span>(</span>
<span id="cb10-1950"><a href="#cb10-1950" aria-hidden="true" tabindex="-1"></a>            <span class="st">"tcga"</span> <span class="ot">=</span> tcga_samples, </span>
<span id="cb10-1951"><a href="#cb10-1951" aria-hidden="true" tabindex="-1"></a>            <span class="st">"scanb"</span> <span class="ot">=</span> scanb_samples, </span>
<span id="cb10-1952"><a href="#cb10-1952" aria-hidden="true" tabindex="-1"></a>            <span class="st">"metabric"</span> <span class="ot">=</span> metabric_samples</span>
<span id="cb10-1953"><a href="#cb10-1953" aria-hidden="true" tabindex="-1"></a>        ),</span>
<span id="cb10-1954"><a href="#cb10-1954" aria-hidden="true" tabindex="-1"></a>        <span class="cf">function</span>(x) df_pca_scores <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(sample_name <span class="sc">%in%</span> x) </span>
<span id="cb10-1955"><a href="#cb10-1955" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb10-1956"><a href="#cb10-1956" aria-hidden="true" tabindex="-1"></a>    <span class="at">name_scores =</span> which_scores,</span>
<span id="cb10-1957"><a href="#cb10-1957" aria-hidden="true" tabindex="-1"></a>    <span class="at">formulas =</span> formulas_survival,</span>
<span id="cb10-1958"><a href="#cb10-1958" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>,</span>
<span id="cb10-1959"><a href="#cb10-1959" aria-hidden="true" tabindex="-1"></a>    <span class="at">simplify =</span> <span class="cn">FALSE</span></span>
<span id="cb10-1960"><a href="#cb10-1960" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-1961"><a href="#cb10-1961" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1962"><a href="#cb10-1962" aria-hidden="true" tabindex="-1"></a>survival_results<span class="sc">$</span>rfs <span class="ot">&lt;-</span> survival_results<span class="sc">$</span>rfs[</span>
<span id="cb10-1963"><a href="#cb10-1963" aria-hidden="true" tabindex="-1"></a>    <span class="sc">!</span><span class="fu">sapply</span>(survival_results<span class="sc">$</span>rfs, is.null)</span>
<span id="cb10-1964"><a href="#cb10-1964" aria-hidden="true" tabindex="-1"></a>]</span>
<span id="cb10-1965"><a href="#cb10-1965" aria-hidden="true" tabindex="-1"></a>survival_results<span class="sc">$</span>rfs<span class="sc">$</span>metabric <span class="ot">&lt;-</span> survival_results<span class="sc">$</span>rfs<span class="sc">$</span>metabric[</span>
<span id="cb10-1966"><a href="#cb10-1966" aria-hidden="true" tabindex="-1"></a>    <span class="sc">!</span><span class="fu">sapply</span>(survival_results<span class="sc">$</span>rfs<span class="sc">$</span>metabric, is.null)</span>
<span id="cb10-1967"><a href="#cb10-1967" aria-hidden="true" tabindex="-1"></a>]</span>
<span id="cb10-1968"><a href="#cb10-1968" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1969"><a href="#cb10-1969" aria-hidden="true" tabindex="-1"></a>survival_results<span class="sc">$</span>os <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">discard</span>(</span>
<span id="cb10-1970"><a href="#cb10-1970" aria-hidden="true" tabindex="-1"></a>    purrr<span class="sc">::</span><span class="fu">map</span>(</span>
<span id="cb10-1971"><a href="#cb10-1971" aria-hidden="true" tabindex="-1"></a>        survival_results<span class="sc">$</span>os, </span>
<span id="cb10-1972"><a href="#cb10-1972" aria-hidden="true" tabindex="-1"></a>        <span class="sc">~</span> purrr<span class="sc">::</span><span class="fu">discard</span>(.x, is.null),</span>
<span id="cb10-1973"><a href="#cb10-1973" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb10-1974"><a href="#cb10-1974" aria-hidden="true" tabindex="-1"></a>    is.null</span>
<span id="cb10-1975"><a href="#cb10-1975" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-1976"><a href="#cb10-1976" aria-hidden="true" tabindex="-1"></a><span class="co"># we cannot simply save the models as each model has some </span></span>
<span id="cb10-1977"><a href="#cb10-1977" aria-hidden="true" tabindex="-1"></a><span class="co"># environment variables, which adds up to over 20GB. </span></span>
<span id="cb10-1978"><a href="#cb10-1978" aria-hidden="true" tabindex="-1"></a><span class="co"># check this stack exchange thread to read more on it:</span></span>
<span id="cb10-1979"><a href="#cb10-1979" aria-hidden="true" tabindex="-1"></a><span class="co"># https://stackoverflow.com/questions/42230920/saverds-inflating-size-of-object/52372480</span></span>
<span id="cb10-1980"><a href="#cb10-1980" aria-hidden="true" tabindex="-1"></a><span class="co"># the solution is to basically clear the environment from the terms </span></span>
<span id="cb10-1981"><a href="#cb10-1981" aria-hidden="true" tabindex="-1"></a><span class="co"># object. since it is pretty fast to run the survival analysis</span></span>
<span id="cb10-1982"><a href="#cb10-1982" aria-hidden="true" tabindex="-1"></a><span class="co"># we will not save the objects.</span></span>
<span id="cb10-1983"><a href="#cb10-1983" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1984"><a href="#cb10-1984" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-1985"><a href="#cb10-1985" aria-hidden="true" tabindex="-1"></a><span class="co"># we now proceed to prepare the plots. since the survival result is </span></span>
<span id="cb10-1986"><a href="#cb10-1986" aria-hidden="true" tabindex="-1"></a><span class="co"># actually a large file, it is best to do all the plottings, save</span></span>
<span id="cb10-1987"><a href="#cb10-1987" aria-hidden="true" tabindex="-1"></a><span class="co"># them in other rds/pdf files and then use on the quarto markdown.</span></span>
<span id="cb10-1988"><a href="#cb10-1988" aria-hidden="true" tabindex="-1"></a><span class="co"># otherwise it is too slow to render the book. </span></span>
<span id="cb10-1989"><a href="#cb10-1989" aria-hidden="true" tabindex="-1"></a><span class="co"># the same is true for the statistics and coefficients available.</span></span>
<span id="cb10-1990"><a href="#cb10-1990" aria-hidden="true" tabindex="-1"></a>names_signatures <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb10-1991"><a href="#cb10-1991" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_G2M_CHECKPOINT"</span> <span class="ot">=</span> <span class="st">"G2M checkpoint"</span>,</span>
<span id="cb10-1992"><a href="#cb10-1992" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION"</span> <span class="ot">=</span> <span class="st">"EMT"</span>,</span>
<span id="cb10-1993"><a href="#cb10-1993" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_DNA_REPAIR"</span> <span class="ot">=</span> <span class="st">"DNA repair"</span>,</span>
<span id="cb10-1994"><a href="#cb10-1994" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_ANDROGEN_RESPONSE"</span> <span class="ot">=</span> <span class="st">"Androgen response"</span>,</span>
<span id="cb10-1995"><a href="#cb10-1995" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_PI3K_AKT_MTOR_SIGNALING"</span> <span class="ot">=</span> <span class="st">"PI3K AKT MTOR signaling"</span>,</span>
<span id="cb10-1996"><a href="#cb10-1996" aria-hidden="true" tabindex="-1"></a>    <span class="st">"SET_ERPR"</span> <span class="ot">=</span> <span class="st">"SET ER/PR"</span>,</span>
<span id="cb10-1997"><a href="#cb10-1997" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_ESTROGEN_RESPONSE_EARLY"</span> <span class="ot">=</span> <span class="st">"Estrogen early"</span></span>
<span id="cb10-1998"><a href="#cb10-1998" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-1999"><a href="#cb10-1999" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(names_signatures) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">names</span>(names_signatures), <span class="st">"_new"</span>)</span>
<span id="cb10-2000"><a href="#cb10-2000" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2001"><a href="#cb10-2001" aria-hidden="true" tabindex="-1"></a>names_coefficients <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb10-2002"><a href="#cb10-2002" aria-hidden="true" tabindex="-1"></a>    <span class="at">metabric =</span> <span class="fu">c</span>(</span>
<span id="cb10-2003"><a href="#cb10-2003" aria-hidden="true" tabindex="-1"></a>        <span class="st">"age"</span> <span class="ot">=</span> <span class="st">"Age"</span>, </span>
<span id="cb10-2004"><a href="#cb10-2004" aria-hidden="true" tabindex="-1"></a>        <span class="st">"npi"</span> <span class="ot">=</span> <span class="st">"NPI"</span>, </span>
<span id="cb10-2005"><a href="#cb10-2005" aria-hidden="true" tabindex="-1"></a>        names_signatures</span>
<span id="cb10-2006"><a href="#cb10-2006" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb10-2007"><a href="#cb10-2007" aria-hidden="true" tabindex="-1"></a>    <span class="at">scanb =</span> <span class="fu">c</span>(</span>
<span id="cb10-2008"><a href="#cb10-2008" aria-hidden="true" tabindex="-1"></a>        <span class="st">"age"</span> <span class="ot">=</span> <span class="st">"age"</span>,</span>
<span id="cb10-2009"><a href="#cb10-2009" aria-hidden="true" tabindex="-1"></a>        names_signatures,</span>
<span id="cb10-2010"><a href="#cb10-2010" aria-hidden="true" tabindex="-1"></a>        <span class="st">"node_stageN1"</span> <span class="ot">=</span> <span class="st">"N1"</span>,</span>
<span id="cb10-2011"><a href="#cb10-2011" aria-hidden="true" tabindex="-1"></a>        <span class="st">"node_stageN2and3"</span> <span class="ot">=</span> <span class="st">"N2 and N3"</span>,</span>
<span id="cb10-2012"><a href="#cb10-2012" aria-hidden="true" tabindex="-1"></a>        <span class="st">"tumor_stageT1"</span> <span class="ot">=</span> <span class="st">"T1"</span>,</span>
<span id="cb10-2013"><a href="#cb10-2013" aria-hidden="true" tabindex="-1"></a>        <span class="st">"tumor_stageT2"</span> <span class="ot">=</span> <span class="st">"T2"</span>,</span>
<span id="cb10-2014"><a href="#cb10-2014" aria-hidden="true" tabindex="-1"></a>        <span class="st">"tumor_stageT3"</span> <span class="ot">=</span> <span class="st">"T3"</span></span>
<span id="cb10-2015"><a href="#cb10-2015" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb10-2016"><a href="#cb10-2016" aria-hidden="true" tabindex="-1"></a>    <span class="at">tcga =</span> <span class="fu">c</span>(</span>
<span id="cb10-2017"><a href="#cb10-2017" aria-hidden="true" tabindex="-1"></a>        <span class="st">"age"</span> <span class="ot">=</span> <span class="st">"age"</span>,</span>
<span id="cb10-2018"><a href="#cb10-2018" aria-hidden="true" tabindex="-1"></a>        <span class="st">"tumor_stage"</span> <span class="ot">=</span> <span class="st">"Tumor Stage"</span>,</span>
<span id="cb10-2019"><a href="#cb10-2019" aria-hidden="true" tabindex="-1"></a>        names_signatures,</span>
<span id="cb10-2020"><a href="#cb10-2020" aria-hidden="true" tabindex="-1"></a>        <span class="st">"node_stageN1"</span> <span class="ot">=</span> <span class="st">"N1"</span>,</span>
<span id="cb10-2021"><a href="#cb10-2021" aria-hidden="true" tabindex="-1"></a>        <span class="st">"node_stageN2"</span> <span class="ot">=</span> <span class="st">"N2"</span>,</span>
<span id="cb10-2022"><a href="#cb10-2022" aria-hidden="true" tabindex="-1"></a>        <span class="st">"node_stageN3"</span> <span class="ot">=</span> <span class="st">"N3"</span>,</span>
<span id="cb10-2023"><a href="#cb10-2023" aria-hidden="true" tabindex="-1"></a>        <span class="st">"tumor_stagestage_ii"</span> <span class="ot">=</span> <span class="st">"T2"</span>,</span>
<span id="cb10-2024"><a href="#cb10-2024" aria-hidden="true" tabindex="-1"></a>        <span class="st">"tumor_stagestage_iii"</span> <span class="ot">=</span> <span class="st">"T3"</span></span>
<span id="cb10-2025"><a href="#cb10-2025" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb10-2026"><a href="#cb10-2026" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-2027"><a href="#cb10-2027" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2028"><a href="#cb10-2028" aria-hidden="true" tabindex="-1"></a>patients <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb10-2029"><a href="#cb10-2029" aria-hidden="true" tabindex="-1"></a>    <span class="at">tcga =</span> <span class="st">"Lum A/B, ER+ BC"</span>,</span>
<span id="cb10-2030"><a href="#cb10-2030" aria-hidden="true" tabindex="-1"></a>    <span class="at">scanb =</span> <span class="st">"Endo Only, ER+ BC"</span>,</span>
<span id="cb10-2031"><a href="#cb10-2031" aria-hidden="true" tabindex="-1"></a>    <span class="at">metabric =</span> <span class="st">"Endo Only, ER+ BC"</span></span>
<span id="cb10-2032"><a href="#cb10-2032" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-2033"><a href="#cb10-2033" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2034"><a href="#cb10-2034" aria-hidden="true" tabindex="-1"></a>analysis_code <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb10-2035"><a href="#cb10-2035" aria-hidden="true" tabindex="-1"></a>    <span class="at">os =</span> <span class="st">"Overall survival"</span>,</span>
<span id="cb10-2036"><a href="#cb10-2036" aria-hidden="true" tabindex="-1"></a>    <span class="at">rfs =</span> <span class="st">"Recurrence free survival"</span></span>
<span id="cb10-2037"><a href="#cb10-2037" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-2038"><a href="#cb10-2038" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2039"><a href="#cb10-2039" aria-hidden="true" tabindex="-1"></a>forest_plots <span class="ot">&lt;-</span> <span class="fu">mapply</span>(</span>
<span id="cb10-2040"><a href="#cb10-2040" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(fits, type_analysis, names_signatures){</span>
<span id="cb10-2041"><a href="#cb10-2041" aria-hidden="true" tabindex="-1"></a>        <span class="fu">mapply</span>(</span>
<span id="cb10-2042"><a href="#cb10-2042" aria-hidden="true" tabindex="-1"></a>            <span class="cf">function</span>(cohort_fits, name_cohort, type_analysis, names_signatures){</span>
<span id="cb10-2043"><a href="#cb10-2043" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb10-2044"><a href="#cb10-2044" aria-hidden="true" tabindex="-1"></a>                tidy_survival <span class="ot">&lt;-</span> <span class="fu">get_df_survival</span>(</span>
<span id="cb10-2045"><a href="#cb10-2045" aria-hidden="true" tabindex="-1"></a>                    cohort_fits,</span>
<span id="cb10-2046"><a href="#cb10-2046" aria-hidden="true" tabindex="-1"></a>                    <span class="at">pathways_of_interest =</span> <span class="fu">names</span>(names_signatures)</span>
<span id="cb10-2047"><a href="#cb10-2047" aria-hidden="true" tabindex="-1"></a>                )</span>
<span id="cb10-2048"><a href="#cb10-2048" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb10-2049"><a href="#cb10-2049" aria-hidden="true" tabindex="-1"></a>                <span class="fu">plot_combined_scores</span>(</span>
<span id="cb10-2050"><a href="#cb10-2050" aria-hidden="true" tabindex="-1"></a>                    tidy_survival, </span>
<span id="cb10-2051"><a href="#cb10-2051" aria-hidden="true" tabindex="-1"></a>                    <span class="at">range_min =</span> <span class="fl">0.6</span>, </span>
<span id="cb10-2052"><a href="#cb10-2052" aria-hidden="true" tabindex="-1"></a>                    <span class="at">range_hr =</span> <span class="fl">1.5</span>,</span>
<span id="cb10-2053"><a href="#cb10-2053" aria-hidden="true" tabindex="-1"></a>                    <span class="at">names_signatures =</span> names_signatures</span>
<span id="cb10-2054"><a href="#cb10-2054" aria-hidden="true" tabindex="-1"></a>                ) <span class="sc">+</span> </span>
<span id="cb10-2055"><a href="#cb10-2055" aria-hidden="true" tabindex="-1"></a>                    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb10-2056"><a href="#cb10-2056" aria-hidden="true" tabindex="-1"></a>                        <span class="at">title =</span> <span class="fu">paste0</span>(</span>
<span id="cb10-2057"><a href="#cb10-2057" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">toupper</span>(name_cohort), <span class="st">" "</span>,</span>
<span id="cb10-2058"><a href="#cb10-2058" aria-hidden="true" tabindex="-1"></a>                            patients[[name_cohort]], <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>,</span>
<span id="cb10-2059"><a href="#cb10-2059" aria-hidden="true" tabindex="-1"></a>                            analysis_code[[type_analysis]]</span>
<span id="cb10-2060"><a href="#cb10-2060" aria-hidden="true" tabindex="-1"></a>                        )</span>
<span id="cb10-2061"><a href="#cb10-2061" aria-hidden="true" tabindex="-1"></a>                    )</span>
<span id="cb10-2062"><a href="#cb10-2062" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb10-2063"><a href="#cb10-2063" aria-hidden="true" tabindex="-1"></a>            },</span>
<span id="cb10-2064"><a href="#cb10-2064" aria-hidden="true" tabindex="-1"></a>            <span class="at">cohort_fits =</span> fits,</span>
<span id="cb10-2065"><a href="#cb10-2065" aria-hidden="true" tabindex="-1"></a>            <span class="at">name_cohort =</span> <span class="fu">names</span>(fits),</span>
<span id="cb10-2066"><a href="#cb10-2066" aria-hidden="true" tabindex="-1"></a>            <span class="at">MoreArgs =</span> <span class="fu">list</span>(</span>
<span id="cb10-2067"><a href="#cb10-2067" aria-hidden="true" tabindex="-1"></a>                <span class="at">type_analysis =</span> type_analysis,</span>
<span id="cb10-2068"><a href="#cb10-2068" aria-hidden="true" tabindex="-1"></a>                <span class="at">names_signatures =</span> names_signatures</span>
<span id="cb10-2069"><a href="#cb10-2069" aria-hidden="true" tabindex="-1"></a>            ),</span>
<span id="cb10-2070"><a href="#cb10-2070" aria-hidden="true" tabindex="-1"></a>            <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>,</span>
<span id="cb10-2071"><a href="#cb10-2071" aria-hidden="true" tabindex="-1"></a>            <span class="at">SIMPLIFY =</span> <span class="cn">FALSE</span></span>
<span id="cb10-2072"><a href="#cb10-2072" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb10-2073"><a href="#cb10-2073" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb10-2074"><a href="#cb10-2074" aria-hidden="true" tabindex="-1"></a>    survival_results,</span>
<span id="cb10-2075"><a href="#cb10-2075" aria-hidden="true" tabindex="-1"></a>    <span class="fu">names</span>(survival_results),</span>
<span id="cb10-2076"><a href="#cb10-2076" aria-hidden="true" tabindex="-1"></a>    <span class="at">MoreArgs =</span> <span class="fu">list</span>(<span class="at">names_signatures =</span> names_signatures),</span>
<span id="cb10-2077"><a href="#cb10-2077" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>,</span>
<span id="cb10-2078"><a href="#cb10-2078" aria-hidden="true" tabindex="-1"></a>    <span class="at">SIMPLIFY =</span> <span class="cn">FALSE</span></span>
<span id="cb10-2079"><a href="#cb10-2079" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-2080"><a href="#cb10-2080" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2081"><a href="#cb10-2081" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb10-2082"><a href="#cb10-2082" aria-hidden="true" tabindex="-1"></a>    forest_plots,</span>
<span id="cb10-2083"><a href="#cb10-2083" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/trying/forest_plots.rds"</span></span>
<span id="cb10-2084"><a href="#cb10-2084" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-2085"><a href="#cb10-2085" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2086"><a href="#cb10-2086" aria-hidden="true" tabindex="-1"></a><span class="co"># and now we fetch the tables that will be used later on including the</span></span>
<span id="cb10-2087"><a href="#cb10-2087" aria-hidden="true" tabindex="-1"></a><span class="co"># confidence intervals and other parameters from the fit</span></span>
<span id="cb10-2088"><a href="#cb10-2088" aria-hidden="true" tabindex="-1"></a>tables_survival <span class="ot">&lt;-</span> <span class="fu">lapply</span>(</span>
<span id="cb10-2089"><a href="#cb10-2089" aria-hidden="true" tabindex="-1"></a>    survival_results, </span>
<span id="cb10-2090"><a href="#cb10-2090" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(x) </span>
<span id="cb10-2091"><a href="#cb10-2091" aria-hidden="true" tabindex="-1"></a>        <span class="fu">lapply</span>(</span>
<span id="cb10-2092"><a href="#cb10-2092" aria-hidden="true" tabindex="-1"></a>            x,</span>
<span id="cb10-2093"><a href="#cb10-2093" aria-hidden="true" tabindex="-1"></a>            <span class="cf">function</span>(y)</span>
<span id="cb10-2094"><a href="#cb10-2094" aria-hidden="true" tabindex="-1"></a>                <span class="fu">lapply</span>(</span>
<span id="cb10-2095"><a href="#cb10-2095" aria-hidden="true" tabindex="-1"></a>                    y,</span>
<span id="cb10-2096"><a href="#cb10-2096" aria-hidden="true" tabindex="-1"></a>                    <span class="cf">function</span>(z) broom<span class="sc">::</span><span class="fu">tidy</span>(z)</span>
<span id="cb10-2097"><a href="#cb10-2097" aria-hidden="true" tabindex="-1"></a>                ) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(<span class="at">.id =</span> <span class="st">"score"</span>)</span>
<span id="cb10-2098"><a href="#cb10-2098" aria-hidden="true" tabindex="-1"></a>        ) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(<span class="at">.id =</span> <span class="st">"cohort"</span>)</span>
<span id="cb10-2099"><a href="#cb10-2099" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(<span class="at">.id =</span> <span class="st">"type_analysis"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-2100"><a href="#cb10-2100" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">HR =</span> <span class="fu">exp</span>(estimate))</span>
<span id="cb10-2101"><a href="#cb10-2101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2102"><a href="#cb10-2102" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(</span>
<span id="cb10-2103"><a href="#cb10-2103" aria-hidden="true" tabindex="-1"></a>    tables_survival,</span>
<span id="cb10-2104"><a href="#cb10-2104" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/tables/trying/survival_results.csv"</span>,</span>
<span id="cb10-2105"><a href="#cb10-2105" aria-hidden="true" tabindex="-1"></a>    <span class="at">row.names =</span> <span class="cn">FALSE</span></span>
<span id="cb10-2106"><a href="#cb10-2106" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-2107"><a href="#cb10-2107" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-2108"><a href="#cb10-2108" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2109"><a href="#cb10-2109" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width = 9, fig.height=4, warning = FALSE}</span></span>
<span id="cb10-2110"><a href="#cb10-2110" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-pca-mol-land-survival-regressed</span></span>
<span id="cb10-2111"><a href="#cb10-2111" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Overall survival analysis from SCANB and METABRIC cohorts and their</span></span>
<span id="cb10-2112"><a href="#cb10-2112" aria-hidden="true" tabindex="-1"></a><span class="co">#|     scores. The formulas used were the same as for the estrogen</span></span>
<span id="cb10-2113"><a href="#cb10-2113" aria-hidden="true" tabindex="-1"></a><span class="co">#|     signaling analysis. Coefficients were scaled before performing</span></span>
<span id="cb10-2114"><a href="#cb10-2114" aria-hidden="true" tabindex="-1"></a><span class="co">#|     cox regression</span></span>
<span id="cb10-2115"><a href="#cb10-2115" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2116"><a href="#cb10-2116" aria-hidden="true" tabindex="-1"></a>forest_plots <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../results/rds_files/trying/forest_plots.rds"</span>)</span>
<span id="cb10-2117"><a href="#cb10-2117" aria-hidden="true" tabindex="-1"></a><span class="co"># we remove the TCGA plot so we have actually 4 plots and they are </span></span>
<span id="cb10-2118"><a href="#cb10-2118" aria-hidden="true" tabindex="-1"></a><span class="co"># ordered nicely. </span></span>
<span id="cb10-2119"><a href="#cb10-2119" aria-hidden="true" tabindex="-1"></a>forest_plots<span class="sc">$</span>os_wo_tcga <span class="ot">&lt;-</span> forest_plots<span class="sc">$</span>os</span>
<span id="cb10-2120"><a href="#cb10-2120" aria-hidden="true" tabindex="-1"></a>forest_plots<span class="sc">$</span>os_wo_tcga<span class="sc">$</span>tcga <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb10-2121"><a href="#cb10-2121" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(<span class="at">plotlist =</span> <span class="fu">c</span>(</span>
<span id="cb10-2122"><a href="#cb10-2122" aria-hidden="true" tabindex="-1"></a>    forest_plots<span class="sc">$</span>os_wo_tcga,</span>
<span id="cb10-2123"><a href="#cb10-2123" aria-hidden="true" tabindex="-1"></a>    forest_plots<span class="sc">$</span>rfs</span>
<span id="cb10-2124"><a href="#cb10-2124" aria-hidden="true" tabindex="-1"></a>), <span class="at">nrow =</span> <span class="dv">2</span>)</span>
<span id="cb10-2125"><a href="#cb10-2125" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-2126"><a href="#cb10-2126" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2127"><a href="#cb10-2127" aria-hidden="true" tabindex="-1"></a>The plot with TCGA is shown below.</span>
<span id="cb10-2128"><a href="#cb10-2128" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2129"><a href="#cb10-2129" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=4.5, fig.height=10}</span></span>
<span id="cb10-2130"><a href="#cb10-2130" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(<span class="at">plotlist =</span> <span class="fu">c</span>(</span>
<span id="cb10-2131"><a href="#cb10-2131" aria-hidden="true" tabindex="-1"></a>    forest_plots<span class="sc">$</span>os,</span>
<span id="cb10-2132"><a href="#cb10-2132" aria-hidden="true" tabindex="-1"></a>    forest_plots<span class="sc">$</span>rfs</span>
<span id="cb10-2133"><a href="#cb10-2133" aria-hidden="true" tabindex="-1"></a>), <span class="at">ncol =</span> <span class="dv">1</span>)</span>
<span id="cb10-2134"><a href="#cb10-2134" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-2135"><a href="#cb10-2135" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2136"><a href="#cb10-2136" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2137"><a href="#cb10-2137" aria-hidden="true" tabindex="-1"></a>Scores for TCGA are noisier than SCANB and METABRIC. Though the</span>
<span id="cb10-2138"><a href="#cb10-2138" aria-hidden="true" tabindex="-1"></a>scaled scores they have comparable hazard ratios and also </span>
<span id="cb10-2139"><a href="#cb10-2139" aria-hidden="true" tabindex="-1"></a>they follow what we saw previously. The only pathway that is </span>
<span id="cb10-2140"><a href="#cb10-2140" aria-hidden="true" tabindex="-1"></a>a bit different is the PI3K AKT MTOR signaling. This pathway had a noisier</span>
<span id="cb10-2141"><a href="#cb10-2141" aria-hidden="true" tabindex="-1"></a>correlation with the GSVA score as well.</span>
<span id="cb10-2142"><a href="#cb10-2142" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2143"><a href="#cb10-2143" aria-hidden="true" tabindex="-1"></a><span class="fu">## Regressing the PDXs</span></span>
<span id="cb10-2144"><a href="#cb10-2144" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2145"><a href="#cb10-2145" aria-hidden="true" tabindex="-1"></a>We can use the regressing strategy to score the PDXs and then evaluate</span>
<span id="cb10-2146"><a href="#cb10-2146" aria-hidden="true" tabindex="-1"></a>their scores. </span>
<span id="cb10-2147"><a href="#cb10-2147" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2148"><a href="#cb10-2148" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb10-2149"><a href="#cb10-2149" aria-hidden="true" tabindex="-1"></a><span class="co"># first get the regressed data</span></span>
<span id="cb10-2150"><a href="#cb10-2150" aria-hidden="true" tabindex="-1"></a>pdx_regressed <span class="ot">&lt;-</span> <span class="fu">get_regressed_data</span>(</span>
<span id="cb10-2151"><a href="#cb10-2151" aria-hidden="true" tabindex="-1"></a>    <span class="at">sum_exp =</span> pdx_tpm, </span>
<span id="cb10-2152"><a href="#cb10-2152" aria-hidden="true" tabindex="-1"></a>    <span class="at">which_exp =</span> <span class="st">"log2tpm"</span>,</span>
<span id="cb10-2153"><a href="#cb10-2153" aria-hidden="true" tabindex="-1"></a>    <span class="at">pca_fit =</span> pca_fit_all_genes_wo_outliers, </span>
<span id="cb10-2154"><a href="#cb10-2154" aria-hidden="true" tabindex="-1"></a>    <span class="at">stable_genes =</span> stable_genes</span>
<span id="cb10-2155"><a href="#cb10-2155" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb10-2156"><a href="#cb10-2156" aria-hidden="true" tabindex="-1"></a>    <span class="co"># calculate the scores for the regressed data</span></span>
<span id="cb10-2157"><a href="#cb10-2157" aria-hidden="true" tabindex="-1"></a>    <span class="fu">get_scores_regressed</span>(</span>
<span id="cb10-2158"><a href="#cb10-2158" aria-hidden="true" tabindex="-1"></a>        <span class="at">sum_exp =</span> .,</span>
<span id="cb10-2159"><a href="#cb10-2159" aria-hidden="true" tabindex="-1"></a>        <span class="at">gene_sets =</span> gene_sets_</span>
<span id="cb10-2160"><a href="#cb10-2160" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb10-2161"><a href="#cb10-2161" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2162"><a href="#cb10-2162" aria-hidden="true" tabindex="-1"></a>pdx_regressed<span class="sc">$</span>cohort <span class="ot">&lt;-</span> <span class="st">"pdx"</span></span>
<span id="cb10-2163"><a href="#cb10-2163" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2164"><a href="#cb10-2164" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb10-2165"><a href="#cb10-2165" aria-hidden="true" tabindex="-1"></a>    pdx_regressed,</span>
<span id="cb10-2166"><a href="#cb10-2166" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/trying/pdx_regressed.rds"</span></span>
<span id="cb10-2167"><a href="#cb10-2167" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-2168"><a href="#cb10-2168" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-2169"><a href="#cb10-2169" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2170"><a href="#cb10-2170" aria-hidden="true" tabindex="-1"></a>We start by comparing the scores across the different </span>
<span id="cb10-2171"><a href="#cb10-2171" aria-hidden="true" tabindex="-1"></a>control samples and their positions in the molecular landscape.</span>
<span id="cb10-2172"><a href="#cb10-2172" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2173"><a href="#cb10-2173" aria-hidden="true" tabindex="-1"></a><span class="in">```{r fig.width=9, fig.height=6}</span></span>
<span id="cb10-2174"><a href="#cb10-2174" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-pca-pdx-control-only-regressed</span></span>
<span id="cb10-2175"><a href="#cb10-2175" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Embedding of only regressed PDX control samples on top of the </span></span>
<span id="cb10-2176"><a href="#cb10-2176" aria-hidden="true" tabindex="-1"></a><span class="co">#|     METABRIC, SCANB and TCGA samples</span></span>
<span id="cb10-2177"><a href="#cb10-2177" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">get_base_plot</span>(</span>
<span id="cb10-2178"><a href="#cb10-2178" aria-hidden="true" tabindex="-1"></a>    df_pca_coordinates_wo_outliers <span class="sc">%&gt;%</span> </span>
<span id="cb10-2179"><a href="#cb10-2179" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">!=</span> <span class="st">"poetic"</span>)</span>
<span id="cb10-2180"><a href="#cb10-2180" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-2181"><a href="#cb10-2181" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2182"><a href="#cb10-2182" aria-hidden="true" tabindex="-1"></a>pdx_df_pca_regressed_sub <span class="ot">&lt;-</span> <span class="fu">colData</span>(pdx_regressed) <span class="sc">%&gt;%</span></span>
<span id="cb10-2183"><a href="#cb10-2183" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span></span>
<span id="cb10-2184"><a href="#cb10-2184" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"pdx"</span> <span class="sc">&amp;</span> treatment <span class="sc">==</span> <span class="st">"CTRL"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-2185"><a href="#cb10-2185" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">pam50 =</span> <span class="st">"nc"</span>)</span>
<span id="cb10-2186"><a href="#cb10-2186" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2187"><a href="#cb10-2187" aria-hidden="true" tabindex="-1"></a>p <span class="sc">+</span></span>
<span id="cb10-2188"><a href="#cb10-2188" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>(</span>
<span id="cb10-2189"><a href="#cb10-2189" aria-hidden="true" tabindex="-1"></a>        pdx_df_pca_regressed_sub,</span>
<span id="cb10-2190"><a href="#cb10-2190" aria-hidden="true" tabindex="-1"></a>        <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> PC3, <span class="at">y =</span> PC4, <span class="at">shape =</span> pdx),</span>
<span id="cb10-2191"><a href="#cb10-2191" aria-hidden="true" tabindex="-1"></a>        <span class="at">size =</span> <span class="dv">5</span></span>
<span id="cb10-2192"><a href="#cb10-2192" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb10-2193"><a href="#cb10-2193" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">15</span>) <span class="sc">+</span></span>
<span id="cb10-2194"><a href="#cb10-2194" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb10-2195"><a href="#cb10-2195" aria-hidden="true" tabindex="-1"></a>        <span class="at">caption =</span> <span class="st">"Grey points correspond to the PDX cohort"</span>,</span>
<span id="cb10-2196"><a href="#cb10-2196" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">"PDX cohort, only control samples"</span></span>
<span id="cb10-2197"><a href="#cb10-2197" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb10-2198"><a href="#cb10-2198" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_color_manual</span>(</span>
<span id="cb10-2199"><a href="#cb10-2199" aria-hidden="true" tabindex="-1"></a>        <span class="at">values =</span> <span class="fu">c</span>(</span>
<span id="cb10-2200"><a href="#cb10-2200" aria-hidden="true" tabindex="-1"></a>            <span class="fu">get_colors_pam50</span>(p<span class="sc">$</span>data),</span>
<span id="cb10-2201"><a href="#cb10-2201" aria-hidden="true" tabindex="-1"></a>            <span class="st">"not available"</span> <span class="ot">=</span> <span class="st">"grey"</span></span>
<span id="cb10-2202"><a href="#cb10-2202" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb10-2203"><a href="#cb10-2203" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb10-2204"><a href="#cb10-2204" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span></span>
<span id="cb10-2205"><a href="#cb10-2205" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>(<span class="at">shape =</span> <span class="cn">TRUE</span>)</span>
<span id="cb10-2206"><a href="#cb10-2206" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-2207"><a href="#cb10-2207" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2208"><a href="#cb10-2208" aria-hidden="true" tabindex="-1"></a>The results are jus twhat we got previously as well, meaning that</span>
<span id="cb10-2209"><a href="#cb10-2209" aria-hidden="true" tabindex="-1"></a>by using all these genes the information of the PDXs are still </span>
<span id="cb10-2210"><a href="#cb10-2210" aria-hidden="true" tabindex="-1"></a>captured. The main differences is that the luminal B region is more</span>
<span id="cb10-2211"><a href="#cb10-2211" aria-hidden="true" tabindex="-1"></a>compact compare to @fig-pca-pdx-control-only.</span>
<span id="cb10-2212"><a href="#cb10-2212" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2213"><a href="#cb10-2213" aria-hidden="true" tabindex="-1"></a>And now that we have the results we can compare the PDXs as well. </span>
<span id="cb10-2214"><a href="#cb10-2214" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2215"><a href="#cb10-2215" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=6}</span></span>
<span id="cb10-2216"><a href="#cb10-2216" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-pdx-regressed-scores</span></span>
<span id="cb10-2217"><a href="#cb10-2217" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Scores obtained from the regressed data of the PDXs.</span></span>
<span id="cb10-2218"><a href="#cb10-2218" aria-hidden="true" tabindex="-1"></a>col_data_pdx_regressed <span class="ot">&lt;-</span> <span class="fu">colData</span>(pdx_regressed) <span class="sc">%&gt;%</span> data.frame</span>
<span id="cb10-2219"><a href="#cb10-2219" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2220"><a href="#cb10-2220" aria-hidden="true" tabindex="-1"></a>pathways_pdx <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"regressed_"</span>, </span>
<span id="cb10-2221"><a href="#cb10-2221" aria-hidden="true" tabindex="-1"></a>    <span class="fu">c</span>(<span class="fu">paste0</span>(<span class="st">"HALLMARK_"</span>, <span class="fu">c</span>(</span>
<span id="cb10-2222"><a href="#cb10-2222" aria-hidden="true" tabindex="-1"></a>        <span class="st">"ESTROGEN_RESPONSE_EARLY"</span>,</span>
<span id="cb10-2223"><a href="#cb10-2223" aria-hidden="true" tabindex="-1"></a>        <span class="st">"G2M_CHECKPOINT"</span>,</span>
<span id="cb10-2224"><a href="#cb10-2224" aria-hidden="true" tabindex="-1"></a>        <span class="st">"E2F_TARGETS"</span>,</span>
<span id="cb10-2225"><a href="#cb10-2225" aria-hidden="true" tabindex="-1"></a>        <span class="st">"ANDROGEN_RESPONSE"</span>,</span>
<span id="cb10-2226"><a href="#cb10-2226" aria-hidden="true" tabindex="-1"></a>        <span class="st">"APOPTOSIS"</span>,</span>
<span id="cb10-2227"><a href="#cb10-2227" aria-hidden="true" tabindex="-1"></a>        <span class="st">"MTORC1_SIGNALING"</span>,</span>
<span id="cb10-2228"><a href="#cb10-2228" aria-hidden="true" tabindex="-1"></a>        <span class="st">"EPITHELIAL_MESENCHYMAL_TRANSITION"</span>,</span>
<span id="cb10-2229"><a href="#cb10-2229" aria-hidden="true" tabindex="-1"></a>        <span class="st">"DNA_REPAIR"</span></span>
<span id="cb10-2230"><a href="#cb10-2230" aria-hidden="true" tabindex="-1"></a>    )),</span>
<span id="cb10-2231"><a href="#cb10-2231" aria-hidden="true" tabindex="-1"></a>    <span class="st">"SET_ERPR"</span>,</span>
<span id="cb10-2232"><a href="#cb10-2232" aria-hidden="true" tabindex="-1"></a>    <span class="st">"random_200"</span>)</span>
<span id="cb10-2233"><a href="#cb10-2233" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-2234"><a href="#cb10-2234" aria-hidden="true" tabindex="-1"></a>labeller_names <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb10-2235"><a href="#cb10-2235" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Estrogen early"</span>,</span>
<span id="cb10-2236"><a href="#cb10-2236" aria-hidden="true" tabindex="-1"></a>    <span class="st">"G2M"</span>,</span>
<span id="cb10-2237"><a href="#cb10-2237" aria-hidden="true" tabindex="-1"></a>    <span class="st">"E2F"</span>,</span>
<span id="cb10-2238"><a href="#cb10-2238" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Androgen response"</span>,</span>
<span id="cb10-2239"><a href="#cb10-2239" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Apoptosis"</span>,</span>
<span id="cb10-2240"><a href="#cb10-2240" aria-hidden="true" tabindex="-1"></a>    <span class="st">"MTORC1 signaling"</span>,</span>
<span id="cb10-2241"><a href="#cb10-2241" aria-hidden="true" tabindex="-1"></a>    <span class="st">"EMT"</span>,</span>
<span id="cb10-2242"><a href="#cb10-2242" aria-hidden="true" tabindex="-1"></a>    <span class="st">"DNA Repair"</span>,</span>
<span id="cb10-2243"><a href="#cb10-2243" aria-hidden="true" tabindex="-1"></a>    <span class="st">"SET ER/PR"</span>,</span>
<span id="cb10-2244"><a href="#cb10-2244" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Random 200"</span></span>
<span id="cb10-2245"><a href="#cb10-2245" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb10-2246"><a href="#cb10-2246" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">names&lt;-</span><span class="st">`</span>(pathways_pdx)</span>
<span id="cb10-2247"><a href="#cb10-2247" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2248"><a href="#cb10-2248" aria-hidden="true" tabindex="-1"></a>col_data_pdx_regressed <span class="sc">%&gt;%</span></span>
<span id="cb10-2249"><a href="#cb10-2249" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(treatment <span class="sc">==</span> <span class="st">"CTRL"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-2250"><a href="#cb10-2250" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(</span>
<span id="cb10-2251"><a href="#cb10-2251" aria-hidden="true" tabindex="-1"></a>        <span class="at">cols =</span> dplyr<span class="sc">::</span><span class="fu">all_of</span>(pathways_pdx),</span>
<span id="cb10-2252"><a href="#cb10-2252" aria-hidden="true" tabindex="-1"></a>        <span class="at">names_to =</span> <span class="st">"pathway"</span>,</span>
<span id="cb10-2253"><a href="#cb10-2253" aria-hidden="true" tabindex="-1"></a>        <span class="at">values_to =</span> <span class="st">"score"</span></span>
<span id="cb10-2254"><a href="#cb10-2254" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb10-2255"><a href="#cb10-2255" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">y =</span> pdx, <span class="at">x =</span> score)) <span class="sc">+</span> </span>
<span id="cb10-2256"><a href="#cb10-2256" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb10-2257"><a href="#cb10-2257" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_jitter</span>() <span class="sc">+</span> </span>
<span id="cb10-2258"><a href="#cb10-2258" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Regressed score"</span>, <span class="at">y =</span> <span class="st">"PDX"</span>) <span class="sc">+</span></span>
<span id="cb10-2259"><a href="#cb10-2259" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>pathway, <span class="at">labeller =</span> <span class="fu">as_labeller</span>(labeller_names)) <span class="sc">+</span> </span>
<span id="cb10-2260"><a href="#cb10-2260" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">15</span>)</span>
<span id="cb10-2261"><a href="#cb10-2261" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-2262"><a href="#cb10-2262" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2263"><a href="#cb10-2263" aria-hidden="true" tabindex="-1"></a>The random 200 is just a gene set with 200 random genes that works as a </span>
<span id="cb10-2264"><a href="#cb10-2264" aria-hidden="true" tabindex="-1"></a>control and serves as a comparison to the other pathways.</span>
<span id="cb10-2265"><a href="#cb10-2265" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2266"><a href="#cb10-2266" aria-hidden="true" tabindex="-1"></a>And below we show the distribution of some of the scores calculated </span>
<span id="cb10-2267"><a href="#cb10-2267" aria-hidden="true" tabindex="-1"></a>for all the other cohorts as a comparison. </span>
<span id="cb10-2268"><a href="#cb10-2268" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2269"><a href="#cb10-2269" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=8, fig.height=5}</span></span>
<span id="cb10-2270"><a href="#cb10-2270" aria-hidden="true" tabindex="-1"></a>scores_wide <span class="ot">&lt;-</span> tidyr<span class="sc">::</span><span class="fu">pivot_wider</span>(</span>
<span id="cb10-2271"><a href="#cb10-2271" aria-hidden="true" tabindex="-1"></a>    scores,</span>
<span id="cb10-2272"><a href="#cb10-2272" aria-hidden="true" tabindex="-1"></a>    <span class="at">id_cols =</span> <span class="fu">c</span>(<span class="st">"sample_name"</span>, <span class="st">"cohort"</span>),</span>
<span id="cb10-2273"><a href="#cb10-2273" aria-hidden="true" tabindex="-1"></a>    <span class="at">values_from =</span> <span class="st">"score_hkg"</span>,</span>
<span id="cb10-2274"><a href="#cb10-2274" aria-hidden="true" tabindex="-1"></a>    <span class="at">names_from =</span> <span class="st">"pathway"</span></span>
<span id="cb10-2275"><a href="#cb10-2275" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-2276"><a href="#cb10-2276" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2277"><a href="#cb10-2277" aria-hidden="true" tabindex="-1"></a>pathways_density <span class="ot">&lt;-</span> stringr<span class="sc">::</span><span class="fu">str_split</span>(</span>
<span id="cb10-2278"><a href="#cb10-2278" aria-hidden="true" tabindex="-1"></a>    pathways_pdx, </span>
<span id="cb10-2279"><a href="#cb10-2279" aria-hidden="true" tabindex="-1"></a>    <span class="at">pattern =</span> <span class="st">"_"</span>,</span>
<span id="cb10-2280"><a href="#cb10-2280" aria-hidden="true" tabindex="-1"></a>    <span class="at">n =</span> <span class="dv">2</span>, </span>
<span id="cb10-2281"><a href="#cb10-2281" aria-hidden="true" tabindex="-1"></a>    <span class="at">simplify =</span> <span class="cn">TRUE</span></span>
<span id="cb10-2282"><a href="#cb10-2282" aria-hidden="true" tabindex="-1"></a>)[, <span class="dv">2</span>]</span>
<span id="cb10-2283"><a href="#cb10-2283" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2284"><a href="#cb10-2284" aria-hidden="true" tabindex="-1"></a>scores <span class="sc">%&gt;%</span></span>
<span id="cb10-2285"><a href="#cb10-2285" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">!=</span> <span class="st">"poetic"</span> <span class="sc">&amp;</span> pathway <span class="sc">%in%</span> pathways_density) <span class="sc">%&gt;%</span></span>
<span id="cb10-2286"><a href="#cb10-2286" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>score_hkg, <span class="at">fill =</span> cohort)) <span class="sc">+</span></span>
<span id="cb10-2287"><a href="#cb10-2287" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_density</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb10-2288"><a href="#cb10-2288" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>pathway, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb10-2289"><a href="#cb10-2289" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>()</span>
<span id="cb10-2290"><a href="#cb10-2290" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-2291"><a href="#cb10-2291" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2292"><a href="#cb10-2292" aria-hidden="true" tabindex="-1"></a>We see that METS15 is the one with the highest proliferation and</span>
<span id="cb10-2293"><a href="#cb10-2293" aria-hidden="true" tabindex="-1"></a>lowest EMT, it is a PDX that comes from a pleural efusion. Also</span>
<span id="cb10-2294"><a href="#cb10-2294" aria-hidden="true" tabindex="-1"></a>it is the PDX with the lowest apoptosis and highest DNA repair,</span>
<span id="cb10-2295"><a href="#cb10-2295" aria-hidden="true" tabindex="-1"></a>probably due to a higher replication rate. Moreover, all these</span>
<span id="cb10-2296"><a href="#cb10-2296" aria-hidden="true" tabindex="-1"></a>PDXs they have a estrogen response early scores higher than 0</span>
<span id="cb10-2297"><a href="#cb10-2297" aria-hidden="true" tabindex="-1"></a>in average, which is about the average of the scores in the</span>
<span id="cb10-2298"><a href="#cb10-2298" aria-hidden="true" tabindex="-1"></a>ER+ BC samples.</span>
<span id="cb10-2299"><a href="#cb10-2299" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2300"><a href="#cb10-2300" aria-hidden="true" tabindex="-1"></a>Now we check what happens with some of the PDXs when they</span>
<span id="cb10-2301"><a href="#cb10-2301" aria-hidden="true" tabindex="-1"></a>are treated with P4, to compare with @fig-pdx-p4-control-pathways.</span>
<span id="cb10-2302"><a href="#cb10-2302" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2303"><a href="#cb10-2303" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=13, fig.height=15}</span></span>
<span id="cb10-2304"><a href="#cb10-2304" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-pdx-p4-control-pathways-regressed.</span></span>
<span id="cb10-2305"><a href="#cb10-2305" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Comparison of position in the molecular landscape between</span></span>
<span id="cb10-2306"><a href="#cb10-2306" aria-hidden="true" tabindex="-1"></a><span class="co">#|     CTRL, E2 and P4 treated samples with the regressed data</span></span>
<span id="cb10-2307"><a href="#cb10-2307" aria-hidden="true" tabindex="-1"></a>pathways <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb10-2308"><a href="#cb10-2308" aria-hidden="true" tabindex="-1"></a>    <span class="co">#"HALLMARK_ANDROGEN_RESPONSE" = "Androgen Response",</span></span>
<span id="cb10-2309"><a href="#cb10-2309" aria-hidden="true" tabindex="-1"></a>    <span class="co">#"HALLMARK_E2F_TARGETS" = "E2F",</span></span>
<span id="cb10-2310"><a href="#cb10-2310" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_G2M_CHECKPOINT"</span> <span class="ot">=</span> <span class="st">"G2M"</span>,</span>
<span id="cb10-2311"><a href="#cb10-2311" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_ESTROGEN_RESPONSE_EARLY"</span> <span class="ot">=</span> <span class="st">"Estrogen Early"</span>,</span>
<span id="cb10-2312"><a href="#cb10-2312" aria-hidden="true" tabindex="-1"></a>    <span class="co">#"SET_ERPR" = "SET ER/PR",</span></span>
<span id="cb10-2313"><a href="#cb10-2313" aria-hidden="true" tabindex="-1"></a>    <span class="st">"HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION"</span> <span class="ot">=</span> <span class="st">"EMT"</span>,</span>
<span id="cb10-2314"><a href="#cb10-2314" aria-hidden="true" tabindex="-1"></a>    <span class="st">"GOBP_RESPONSE_TO_PROGESTERONE"</span> <span class="ot">=</span> <span class="st">"GO Progesterone"</span>,</span>
<span id="cb10-2315"><a href="#cb10-2315" aria-hidden="true" tabindex="-1"></a>    <span class="st">"WILCOX_RESPONSE_TO_PROGESTERONE_UP"</span> <span class="ot">=</span> <span class="st">"W. Prog UP"</span></span>
<span id="cb10-2316"><a href="#cb10-2316" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-2317"><a href="#cb10-2317" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(pathways) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"regressed_"</span>, <span class="fu">names</span>(pathways))</span>
<span id="cb10-2318"><a href="#cb10-2318" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2319"><a href="#cb10-2319" aria-hidden="true" tabindex="-1"></a>pdxs_to_use <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"METS15"</span>, <span class="st">"T113"</span>, <span class="st">"T111"</span>, <span class="st">"T105"</span>, <span class="st">"T109"</span>, <span class="st">"T110"</span>)</span>
<span id="cb10-2320"><a href="#cb10-2320" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(pdxs_to_use) <span class="ot">&lt;-</span> pdxs_to_use</span>
<span id="cb10-2321"><a href="#cb10-2321" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2322"><a href="#cb10-2322" aria-hidden="true" tabindex="-1"></a>pdx_df_pca_sub <span class="ot">&lt;-</span> col_data_pdx_regressed <span class="sc">%&gt;%</span></span>
<span id="cb10-2323"><a href="#cb10-2323" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb10-2324"><a href="#cb10-2324" aria-hidden="true" tabindex="-1"></a>        cohort <span class="sc">==</span> <span class="st">"pdx"</span> <span class="sc">&amp;</span> </span>
<span id="cb10-2325"><a href="#cb10-2325" aria-hidden="true" tabindex="-1"></a>            treatment <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"CTRL"</span>, <span class="st">"P4"</span>, <span class="st">"E2"</span>)</span>
<span id="cb10-2326"><a href="#cb10-2326" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb10-2327"><a href="#cb10-2327" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">pam50 =</span> <span class="st">"nc"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-2328"><a href="#cb10-2328" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">pdx =</span> <span class="fu">factor</span>(pdx, <span class="at">levels =</span> <span class="fu">c</span>(</span>
<span id="cb10-2329"><a href="#cb10-2329" aria-hidden="true" tabindex="-1"></a>        <span class="st">"METS15"</span>, <span class="st">"T113"</span>, <span class="st">"T109"</span>,</span>
<span id="cb10-2330"><a href="#cb10-2330" aria-hidden="true" tabindex="-1"></a>        <span class="st">"T105"</span>, <span class="st">"T111"</span>, <span class="st">"T110"</span></span>
<span id="cb10-2331"><a href="#cb10-2331" aria-hidden="true" tabindex="-1"></a>    ))) <span class="sc">%&gt;%</span> </span>
<span id="cb10-2332"><a href="#cb10-2332" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(pdx <span class="sc">%in%</span> pdxs_to_use)</span>
<span id="cb10-2333"><a href="#cb10-2333" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2334"><a href="#cb10-2334" aria-hidden="true" tabindex="-1"></a>plots_pathways <span class="ot">&lt;-</span> <span class="fu">mapply</span>(</span>
<span id="cb10-2335"><a href="#cb10-2335" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(pathways, pathway_names, i){</span>
<span id="cb10-2336"><a href="#cb10-2336" aria-hidden="true" tabindex="-1"></a>        <span class="fu">names</span>(pathways) <span class="ot">&lt;-</span> pathway_names </span>
<span id="cb10-2337"><a href="#cb10-2337" aria-hidden="true" tabindex="-1"></a>        pdx_df_pca_sub <span class="sc">%&gt;%</span></span>
<span id="cb10-2338"><a href="#cb10-2338" aria-hidden="true" tabindex="-1"></a>            tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(</span>
<span id="cb10-2339"><a href="#cb10-2339" aria-hidden="true" tabindex="-1"></a>                <span class="at">cols =</span> dplyr<span class="sc">::</span><span class="fu">all_of</span>(pathway_names),</span>
<span id="cb10-2340"><a href="#cb10-2340" aria-hidden="true" tabindex="-1"></a>                <span class="at">names_to =</span> <span class="st">"pathway"</span>,</span>
<span id="cb10-2341"><a href="#cb10-2341" aria-hidden="true" tabindex="-1"></a>                <span class="at">values_to =</span> <span class="st">"score"</span></span>
<span id="cb10-2342"><a href="#cb10-2342" aria-hidden="true" tabindex="-1"></a>            ) <span class="sc">%&gt;%</span></span>
<span id="cb10-2343"><a href="#cb10-2343" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">pathway =</span> <span class="fu">factor</span>(pathway, <span class="at">levels =</span> pathway_names)) <span class="sc">%&gt;%</span></span>
<span id="cb10-2344"><a href="#cb10-2344" aria-hidden="true" tabindex="-1"></a>            ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> treatment, <span class="at">y =</span> score)) <span class="sc">+</span></span>
<span id="cb10-2345"><a href="#cb10-2345" aria-hidden="true" tabindex="-1"></a>            ggplot2<span class="sc">::</span><span class="fu">geom_boxplot</span>(<span class="at">outlier.shape =</span> <span class="cn">NA</span>) <span class="sc">+</span> </span>
<span id="cb10-2346"><a href="#cb10-2346" aria-hidden="true" tabindex="-1"></a>            ggplot2<span class="sc">::</span><span class="fu">geom_jitter</span>(</span>
<span id="cb10-2347"><a href="#cb10-2347" aria-hidden="true" tabindex="-1"></a>                <span class="at">size =</span> <span class="dv">3</span>, </span>
<span id="cb10-2348"><a href="#cb10-2348" aria-hidden="true" tabindex="-1"></a>                <span class="at">alpha =</span> <span class="fl">0.9</span>, </span>
<span id="cb10-2349"><a href="#cb10-2349" aria-hidden="true" tabindex="-1"></a>                <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">color =</span> treatment)</span>
<span id="cb10-2350"><a href="#cb10-2350" aria-hidden="true" tabindex="-1"></a>            ) <span class="sc">+</span> </span>
<span id="cb10-2351"><a href="#cb10-2351" aria-hidden="true" tabindex="-1"></a>            ggplot2<span class="sc">::</span><span class="fu">scale_color_viridis_d</span>(<span class="at">option =</span> <span class="st">"H"</span>) <span class="sc">+</span></span>
<span id="cb10-2352"><a href="#cb10-2352" aria-hidden="true" tabindex="-1"></a>            ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(</span>
<span id="cb10-2353"><a href="#cb10-2353" aria-hidden="true" tabindex="-1"></a>                <span class="sc">~</span> pdx <span class="sc">+</span> pathway, </span>
<span id="cb10-2354"><a href="#cb10-2354" aria-hidden="true" tabindex="-1"></a>                <span class="at">scales =</span> <span class="st">"fixed"</span>, </span>
<span id="cb10-2355"><a href="#cb10-2355" aria-hidden="true" tabindex="-1"></a>                <span class="at">nrow =</span> <span class="fu">length</span>(pathway_names),</span>
<span id="cb10-2356"><a href="#cb10-2356" aria-hidden="true" tabindex="-1"></a>                <span class="at">labeller =</span> <span class="fu">as_labeller</span>(<span class="fu">c</span>(pathways, pdxs_to_use))</span>
<span id="cb10-2357"><a href="#cb10-2357" aria-hidden="true" tabindex="-1"></a>            ) <span class="sc">+</span></span>
<span id="cb10-2358"><a href="#cb10-2358" aria-hidden="true" tabindex="-1"></a>            ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb10-2359"><a href="#cb10-2359" aria-hidden="true" tabindex="-1"></a>                <span class="at">x =</span> <span class="fu">ifelse</span>(i <span class="sc">==</span> <span class="dv">4</span>, <span class="st">"Treatment"</span>, <span class="st">""</span>),</span>
<span id="cb10-2360"><a href="#cb10-2360" aria-hidden="true" tabindex="-1"></a>                <span class="at">y =</span> <span class="st">"Regressed scores"</span></span>
<span id="cb10-2361"><a href="#cb10-2361" aria-hidden="true" tabindex="-1"></a>            ) <span class="sc">+</span></span>
<span id="cb10-2362"><a href="#cb10-2362" aria-hidden="true" tabindex="-1"></a>            ggplot2<span class="sc">::</span><span class="fu">scale_y_continuous</span>(<span class="at">labels =</span> scales<span class="sc">::</span><span class="fu">number_format</span>(</span>
<span id="cb10-2363"><a href="#cb10-2363" aria-hidden="true" tabindex="-1"></a>                <span class="at">accuracy =</span> <span class="fl">0.01</span>,</span>
<span id="cb10-2364"><a href="#cb10-2364" aria-hidden="true" tabindex="-1"></a>                <span class="at">decimal.mark =</span> <span class="st">'.'</span></span>
<span id="cb10-2365"><a href="#cb10-2365" aria-hidden="true" tabindex="-1"></a>            )) <span class="sc">+</span> </span>
<span id="cb10-2366"><a href="#cb10-2366" aria-hidden="true" tabindex="-1"></a>            ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>) <span class="sc">+</span></span>
<span id="cb10-2367"><a href="#cb10-2367" aria-hidden="true" tabindex="-1"></a>            ggplot2<span class="sc">::</span><span class="fu">theme</span>(<span class="at">legend.position=</span><span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb10-2368"><a href="#cb10-2368" aria-hidden="true" tabindex="-1"></a>            <span class="fu">change_guides_point</span>() <span class="sc">+</span></span>
<span id="cb10-2369"><a href="#cb10-2369" aria-hidden="true" tabindex="-1"></a>            <span class="fu">change_plot_aes_point</span>()</span>
<span id="cb10-2370"><a href="#cb10-2370" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb10-2371"><a href="#cb10-2371" aria-hidden="true" tabindex="-1"></a>    <span class="at">pathways =</span> pathways,</span>
<span id="cb10-2372"><a href="#cb10-2372" aria-hidden="true" tabindex="-1"></a>    <span class="at">pathway_name =</span> <span class="fu">names</span>(pathways),</span>
<span id="cb10-2373"><a href="#cb10-2373" aria-hidden="true" tabindex="-1"></a>    <span class="at">i =</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(pathways),</span>
<span id="cb10-2374"><a href="#cb10-2374" aria-hidden="true" tabindex="-1"></a>    <span class="at">SIMPLIFY =</span> <span class="cn">FALSE</span>,</span>
<span id="cb10-2375"><a href="#cb10-2375" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span></span>
<span id="cb10-2376"><a href="#cb10-2376" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-2377"><a href="#cb10-2377" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2378"><a href="#cb10-2378" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(</span>
<span id="cb10-2379"><a href="#cb10-2379" aria-hidden="true" tabindex="-1"></a>    <span class="at">plotlist =</span> plots_pathways,</span>
<span id="cb10-2380"><a href="#cb10-2380" aria-hidden="true" tabindex="-1"></a>    <span class="at">nrow =</span> <span class="fu">length</span>(plots_pathways) </span>
<span id="cb10-2381"><a href="#cb10-2381" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-2382"><a href="#cb10-2382" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-2383"><a href="#cb10-2383" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2384"><a href="#cb10-2384" aria-hidden="true" tabindex="-1"></a>The results are very similar to what was seen previously using the GSVA. What is </span>
<span id="cb10-2385"><a href="#cb10-2385" aria-hidden="true" tabindex="-1"></a>interesting here is that we have an absolute scale, so we can compare the scores.</span>
<span id="cb10-2386"><a href="#cb10-2386" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2387"><a href="#cb10-2387" aria-hidden="true" tabindex="-1"></a><span class="fu">## singscore</span></span>
<span id="cb10-2388"><a href="#cb10-2388" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2389"><a href="#cb10-2389" aria-hidden="true" tabindex="-1"></a>We can try to use singscore instead, it is a sample dependent measure so </span>
<span id="cb10-2390"><a href="#cb10-2390" aria-hidden="true" tabindex="-1"></a>it does not estimate the distribution of the gene expression levels before</span>
<span id="cb10-2391"><a href="#cb10-2391" aria-hidden="true" tabindex="-1"></a>calculating the enrichment score.</span>
<span id="cb10-2392"><a href="#cb10-2392" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2393"><a href="#cb10-2393" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb10-2394"><a href="#cb10-2394" aria-hidden="true" tabindex="-1"></a>gene_sets_gsea <span class="ot">&lt;-</span> <span class="fu">mapply</span>(</span>
<span id="cb10-2395"><a href="#cb10-2395" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(gene_set, name_gs){</span>
<span id="cb10-2396"><a href="#cb10-2396" aria-hidden="true" tabindex="-1"></a>        GSEABase<span class="sc">::</span><span class="fu">GeneSet</span>(</span>
<span id="cb10-2397"><a href="#cb10-2397" aria-hidden="true" tabindex="-1"></a>            gene_set,</span>
<span id="cb10-2398"><a href="#cb10-2398" aria-hidden="true" tabindex="-1"></a>            <span class="at">setName =</span> name_gs</span>
<span id="cb10-2399"><a href="#cb10-2399" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb10-2400"><a href="#cb10-2400" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb10-2401"><a href="#cb10-2401" aria-hidden="true" tabindex="-1"></a>    <span class="at">gene_set =</span> gene_sets_,</span>
<span id="cb10-2402"><a href="#cb10-2402" aria-hidden="true" tabindex="-1"></a>    <span class="at">name_gs =</span> <span class="fu">setdiff</span>(<span class="fu">names</span>(gene_sets_), p4_pathways<span class="sc">$</span>gs_name <span class="sc">%&gt;%</span> unique)</span>
<span id="cb10-2403"><a href="#cb10-2403" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-2404"><a href="#cb10-2404" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb10-2405"><a href="#cb10-2405" aria-hidden="true" tabindex="-1"></a>singscore_dfs <span class="ot">&lt;-</span> <span class="fu">mapply</span>(</span>
<span id="cb10-2406"><a href="#cb10-2406" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(sum_exp, gene_sets, which_exp){</span>
<span id="cb10-2407"><a href="#cb10-2407" aria-hidden="true" tabindex="-1"></a>        singscore<span class="sc">::</span><span class="fu">multiScore</span>(</span>
<span id="cb10-2408"><a href="#cb10-2408" aria-hidden="true" tabindex="-1"></a>            singscore<span class="sc">::</span><span class="fu">rankGenes</span>(<span class="fu">assay</span>(sum_exp,which_exp)),</span>
<span id="cb10-2409"><a href="#cb10-2409" aria-hidden="true" tabindex="-1"></a>            <span class="at">upSetColc =</span> gene_sets</span>
<span id="cb10-2410"><a href="#cb10-2410" aria-hidden="true" tabindex="-1"></a>        )[[<span class="dv">1</span>]]</span>
<span id="cb10-2411"><a href="#cb10-2411" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb10-2412"><a href="#cb10-2412" aria-hidden="true" tabindex="-1"></a>    <span class="at">sum_exp =</span> datasets,</span>
<span id="cb10-2413"><a href="#cb10-2413" aria-hidden="true" tabindex="-1"></a>    <span class="at">which_exp =</span> <span class="fu">c</span>(which_exp, <span class="st">"poetic"</span> <span class="ot">=</span> <span class="st">"normalized_intensity"</span>),</span>
<span id="cb10-2414"><a href="#cb10-2414" aria-hidden="true" tabindex="-1"></a>    <span class="at">MoreArgs =</span> <span class="fu">list</span>(<span class="at">gene_sets =</span> gene_sets_gsea)</span>
<span id="cb10-2415"><a href="#cb10-2415" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> <span class="fu">Reduce</span>(cbind, .)</span>
<span id="cb10-2416"><a href="#cb10-2416" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2417"><a href="#cb10-2417" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb10-2418"><a href="#cb10-2418" aria-hidden="true" tabindex="-1"></a>    singscore_dfs,</span>
<span id="cb10-2419"><a href="#cb10-2419" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/trying/singscore_dfs.rds"</span></span>
<span id="cb10-2420"><a href="#cb10-2420" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-2421"><a href="#cb10-2421" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2422"><a href="#cb10-2422" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb10-2423"><a href="#cb10-2423" aria-hidden="true" tabindex="-1"></a>    gene_sets_gsea,</span>
<span id="cb10-2424"><a href="#cb10-2424" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/trying/gene_sets_gsea.rds"</span></span>
<span id="cb10-2425"><a href="#cb10-2425" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-2426"><a href="#cb10-2426" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-2427"><a href="#cb10-2427" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2428"><a href="#cb10-2428" aria-hidden="true" tabindex="-1"></a>We start by comparing with the GSVA scores.</span>
<span id="cb10-2429"><a href="#cb10-2429" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2430"><a href="#cb10-2430" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=20, fig.height=16, dev='png'}</span></span>
<span id="cb10-2431"><a href="#cb10-2431" aria-hidden="true" tabindex="-1"></a>df_all_scores <span class="ot">&lt;-</span> singscore_dfs <span class="sc">%&gt;%</span> </span>
<span id="cb10-2432"><a href="#cb10-2432" aria-hidden="true" tabindex="-1"></a>    t <span class="sc">%&gt;%</span></span>
<span id="cb10-2433"><a href="#cb10-2433" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span></span>
<span id="cb10-2434"><a href="#cb10-2434" aria-hidden="true" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"sample_name"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-2435"><a href="#cb10-2435" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(</span>
<span id="cb10-2436"><a href="#cb10-2436" aria-hidden="true" tabindex="-1"></a>        <span class="at">cols =</span> dplyr<span class="sc">::</span><span class="fu">all_of</span>(<span class="fu">names</span>(gene_sets_gsea)),</span>
<span id="cb10-2437"><a href="#cb10-2437" aria-hidden="true" tabindex="-1"></a>        <span class="at">names_to =</span> <span class="st">"pathway"</span>,</span>
<span id="cb10-2438"><a href="#cb10-2438" aria-hidden="true" tabindex="-1"></a>        <span class="at">values_to =</span> <span class="st">"singscore"</span></span>
<span id="cb10-2439"><a href="#cb10-2439" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb10-2440"><a href="#cb10-2440" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">inner_join</span>(</span>
<span id="cb10-2441"><a href="#cb10-2441" aria-hidden="true" tabindex="-1"></a>        merged_col_data <span class="sc">%&gt;%</span></span>
<span id="cb10-2442"><a href="#cb10-2442" aria-hidden="true" tabindex="-1"></a>            tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(</span>
<span id="cb10-2443"><a href="#cb10-2443" aria-hidden="true" tabindex="-1"></a>                <span class="at">cols =</span> dplyr<span class="sc">::</span><span class="fu">all_of</span>(<span class="fu">names</span>(gene_sets_gsea)),</span>
<span id="cb10-2444"><a href="#cb10-2444" aria-hidden="true" tabindex="-1"></a>                <span class="at">names_to =</span> <span class="st">"pathway"</span>,</span>
<span id="cb10-2445"><a href="#cb10-2445" aria-hidden="true" tabindex="-1"></a>                <span class="at">values_to =</span> <span class="st">"gsva"</span></span>
<span id="cb10-2446"><a href="#cb10-2446" aria-hidden="true" tabindex="-1"></a>            ),</span>
<span id="cb10-2447"><a href="#cb10-2447" aria-hidden="true" tabindex="-1"></a>        .,</span>
<span id="cb10-2448"><a href="#cb10-2448" aria-hidden="true" tabindex="-1"></a>        <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"sample_name"</span>, <span class="st">"pathway"</span>)</span>
<span id="cb10-2449"><a href="#cb10-2449" aria-hidden="true" tabindex="-1"></a>    ) </span>
<span id="cb10-2450"><a href="#cb10-2450" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2451"><a href="#cb10-2451" aria-hidden="true" tabindex="-1"></a>df_all_scores <span class="sc">%&gt;%</span></span>
<span id="cb10-2452"><a href="#cb10-2452" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(</span>
<span id="cb10-2453"><a href="#cb10-2453" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> singscore, <span class="at">y =</span> gsva, <span class="at">color =</span> cohort</span>
<span id="cb10-2454"><a href="#cb10-2454" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">+</span></span>
<span id="cb10-2455"><a href="#cb10-2455" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb10-2456"><a href="#cb10-2456" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>pathway, <span class="at">scales =</span> <span class="st">"free"</span>)</span>
<span id="cb10-2457"><a href="#cb10-2457" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-2458"><a href="#cb10-2458" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2459"><a href="#cb10-2459" aria-hidden="true" tabindex="-1"></a>We see that in all cases there is a correlation, but the interpretation</span>
<span id="cb10-2460"><a href="#cb10-2460" aria-hidden="true" tabindex="-1"></a>changes. In some of the pathways there is difference in the singscores</span>
<span id="cb10-2461"><a href="#cb10-2461" aria-hidden="true" tabindex="-1"></a>based on the cohort, this is not seen on GSVA. </span>
<span id="cb10-2462"><a href="#cb10-2462" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2463"><a href="#cb10-2463" aria-hidden="true" tabindex="-1"></a>The next two figures are the distributions of the singscore and GSVA</span>
<span id="cb10-2464"><a href="#cb10-2464" aria-hidden="true" tabindex="-1"></a>filled by cohort.</span>
<span id="cb10-2465"><a href="#cb10-2465" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2466"><a href="#cb10-2466" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=20, fig.height=16, dev='png'}</span></span>
<span id="cb10-2467"><a href="#cb10-2467" aria-hidden="true" tabindex="-1"></a>df_all_scores <span class="sc">%&gt;%</span></span>
<span id="cb10-2468"><a href="#cb10-2468" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(</span>
<span id="cb10-2469"><a href="#cb10-2469" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> singscore, <span class="at">fill =</span> cohort, <span class="at">alpha =</span> <span class="fl">0.5</span></span>
<span id="cb10-2470"><a href="#cb10-2470" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">+</span></span>
<span id="cb10-2471"><a href="#cb10-2471" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_density</span>(<span class="at">position =</span> <span class="st">"identity"</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb10-2472"><a href="#cb10-2472" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>pathway, <span class="at">scales =</span> <span class="st">"free"</span>)</span>
<span id="cb10-2473"><a href="#cb10-2473" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-2474"><a href="#cb10-2474" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2475"><a href="#cb10-2475" aria-hidden="true" tabindex="-1"></a>For several pathways in the singscore the distributions are similar,</span>
<span id="cb10-2476"><a href="#cb10-2476" aria-hidden="true" tabindex="-1"></a>but that is not always the case.</span>
<span id="cb10-2477"><a href="#cb10-2477" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2478"><a href="#cb10-2478" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=20, fig.height=16, dev='png'}</span></span>
<span id="cb10-2479"><a href="#cb10-2479" aria-hidden="true" tabindex="-1"></a>df_all_scores <span class="sc">%&gt;%</span></span>
<span id="cb10-2480"><a href="#cb10-2480" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(</span>
<span id="cb10-2481"><a href="#cb10-2481" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> gsva, <span class="at">fill =</span> cohort</span>
<span id="cb10-2482"><a href="#cb10-2482" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">+</span></span>
<span id="cb10-2483"><a href="#cb10-2483" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_density</span>(<span class="at">position =</span> <span class="st">"identity"</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb10-2484"><a href="#cb10-2484" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>pathway, <span class="at">scales =</span> <span class="st">"free"</span>)</span>
<span id="cb10-2485"><a href="#cb10-2485" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-2486"><a href="#cb10-2486" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2487"><a href="#cb10-2487" aria-hidden="true" tabindex="-1"></a>On the other hand GSVA gives very similar distributions for all the </span>
<span id="cb10-2488"><a href="#cb10-2488" aria-hidden="true" tabindex="-1"></a>different cohorts. Therefore we actually need to use GSVA. It is important in </span>
<span id="cb10-2489"><a href="#cb10-2489" aria-hidden="true" tabindex="-1"></a>the end to calculate scores based on the cohort. </span>
<span id="cb10-2490"><a href="#cb10-2490" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2491"><a href="#cb10-2491" aria-hidden="true" tabindex="-1"></a>We now move on and try the same analysis on the regressed data.</span>
<span id="cb10-2492"><a href="#cb10-2492" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2493"><a href="#cb10-2493" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb10-2494"><a href="#cb10-2494" aria-hidden="true" tabindex="-1"></a>singscore_dfs_regressed <span class="ot">&lt;-</span> singscore<span class="sc">::</span><span class="fu">multiScore</span>(</span>
<span id="cb10-2495"><a href="#cb10-2495" aria-hidden="true" tabindex="-1"></a>    singscore<span class="sc">::</span><span class="fu">rankGenes</span>(df_pcs_regressed),</span>
<span id="cb10-2496"><a href="#cb10-2496" aria-hidden="true" tabindex="-1"></a>    <span class="at">upSetColc =</span> gene_sets_gsea</span>
<span id="cb10-2497"><a href="#cb10-2497" aria-hidden="true" tabindex="-1"></a>)[[<span class="dv">1</span>]]</span>
<span id="cb10-2498"><a href="#cb10-2498" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2499"><a href="#cb10-2499" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb10-2500"><a href="#cb10-2500" aria-hidden="true" tabindex="-1"></a>    singscore_dfs_regressed,</span>
<span id="cb10-2501"><a href="#cb10-2501" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/trying/singscore_dfs_regressed.rds"</span></span>
<span id="cb10-2502"><a href="#cb10-2502" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-2503"><a href="#cb10-2503" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-2504"><a href="#cb10-2504" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2505"><a href="#cb10-2505" aria-hidden="true" tabindex="-1"></a>We start by comparing with the GSVA scores.</span>
<span id="cb10-2506"><a href="#cb10-2506" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2507"><a href="#cb10-2507" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=20, fig.height=16, dev='png'}</span></span>
<span id="cb10-2508"><a href="#cb10-2508" aria-hidden="true" tabindex="-1"></a>df_all_scores_regressed <span class="ot">&lt;-</span> singscore_dfs_regressed <span class="sc">%&gt;%</span> </span>
<span id="cb10-2509"><a href="#cb10-2509" aria-hidden="true" tabindex="-1"></a>    t <span class="sc">%&gt;%</span></span>
<span id="cb10-2510"><a href="#cb10-2510" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span></span>
<span id="cb10-2511"><a href="#cb10-2511" aria-hidden="true" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"sample_name"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-2512"><a href="#cb10-2512" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(</span>
<span id="cb10-2513"><a href="#cb10-2513" aria-hidden="true" tabindex="-1"></a>        <span class="at">cols =</span> dplyr<span class="sc">::</span><span class="fu">all_of</span>(<span class="fu">names</span>(gene_sets_gsea)),</span>
<span id="cb10-2514"><a href="#cb10-2514" aria-hidden="true" tabindex="-1"></a>        <span class="at">names_to =</span> <span class="st">"pathway"</span>,</span>
<span id="cb10-2515"><a href="#cb10-2515" aria-hidden="true" tabindex="-1"></a>        <span class="at">values_to =</span> <span class="st">"singscore"</span></span>
<span id="cb10-2516"><a href="#cb10-2516" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb10-2517"><a href="#cb10-2517" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">inner_join</span>(</span>
<span id="cb10-2518"><a href="#cb10-2518" aria-hidden="true" tabindex="-1"></a>        merged_col_data <span class="sc">%&gt;%</span></span>
<span id="cb10-2519"><a href="#cb10-2519" aria-hidden="true" tabindex="-1"></a>            tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(</span>
<span id="cb10-2520"><a href="#cb10-2520" aria-hidden="true" tabindex="-1"></a>                <span class="at">cols =</span> dplyr<span class="sc">::</span><span class="fu">all_of</span>(<span class="fu">names</span>(gene_sets_gsea)),</span>
<span id="cb10-2521"><a href="#cb10-2521" aria-hidden="true" tabindex="-1"></a>                <span class="at">names_to =</span> <span class="st">"pathway"</span>,</span>
<span id="cb10-2522"><a href="#cb10-2522" aria-hidden="true" tabindex="-1"></a>                <span class="at">values_to =</span> <span class="st">"gsva"</span></span>
<span id="cb10-2523"><a href="#cb10-2523" aria-hidden="true" tabindex="-1"></a>            ),</span>
<span id="cb10-2524"><a href="#cb10-2524" aria-hidden="true" tabindex="-1"></a>        .,</span>
<span id="cb10-2525"><a href="#cb10-2525" aria-hidden="true" tabindex="-1"></a>        <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"sample_name"</span>, <span class="st">"pathway"</span>)</span>
<span id="cb10-2526"><a href="#cb10-2526" aria-hidden="true" tabindex="-1"></a>    ) </span>
<span id="cb10-2527"><a href="#cb10-2527" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2528"><a href="#cb10-2528" aria-hidden="true" tabindex="-1"></a>df_all_scores_regressed <span class="sc">%&gt;%</span></span>
<span id="cb10-2529"><a href="#cb10-2529" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(</span>
<span id="cb10-2530"><a href="#cb10-2530" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> singscore, <span class="at">y =</span> gsva, <span class="at">color =</span> cohort</span>
<span id="cb10-2531"><a href="#cb10-2531" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">+</span></span>
<span id="cb10-2532"><a href="#cb10-2532" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb10-2533"><a href="#cb10-2533" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>pathway, <span class="at">scales =</span> <span class="st">"free"</span>)</span>
<span id="cb10-2534"><a href="#cb10-2534" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-2535"><a href="#cb10-2535" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2536"><a href="#cb10-2536" aria-hidden="true" tabindex="-1"></a>We see that in all cases there is a correlation, but the interpretation</span>
<span id="cb10-2537"><a href="#cb10-2537" aria-hidden="true" tabindex="-1"></a>changes. In some of the pathways there is difference in the singscores</span>
<span id="cb10-2538"><a href="#cb10-2538" aria-hidden="true" tabindex="-1"></a>based on the cohort, this is not seen on GSVA. </span>
<span id="cb10-2539"><a href="#cb10-2539" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2540"><a href="#cb10-2540" aria-hidden="true" tabindex="-1"></a>The next two figures are the distributions of the singscore and GSVA</span>
<span id="cb10-2541"><a href="#cb10-2541" aria-hidden="true" tabindex="-1"></a>filled by cohort.</span>
<span id="cb10-2542"><a href="#cb10-2542" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2543"><a href="#cb10-2543" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=20, fig.height=16, dev='png'}</span></span>
<span id="cb10-2544"><a href="#cb10-2544" aria-hidden="true" tabindex="-1"></a>df_all_scores_regressed <span class="sc">%&gt;%</span></span>
<span id="cb10-2545"><a href="#cb10-2545" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(</span>
<span id="cb10-2546"><a href="#cb10-2546" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> singscore, <span class="at">fill =</span> cohort</span>
<span id="cb10-2547"><a href="#cb10-2547" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">+</span></span>
<span id="cb10-2548"><a href="#cb10-2548" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_density</span>(<span class="at">position =</span> <span class="st">"identity"</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb10-2549"><a href="#cb10-2549" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>pathway, <span class="at">scales =</span> <span class="st">"free"</span>)</span>
<span id="cb10-2550"><a href="#cb10-2550" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-2551"><a href="#cb10-2551" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2552"><a href="#cb10-2552" aria-hidden="true" tabindex="-1"></a>In the end the best way of scoring seems to be using GSVA on the </span>
<span id="cb10-2553"><a href="#cb10-2553" aria-hidden="true" tabindex="-1"></a>unregressed data though it depends on the pathways. For example the</span>
<span id="cb10-2554"><a href="#cb10-2554" aria-hidden="true" tabindex="-1"></a>hallmark estrogen response early and set ER/PR pathways seem to </span>
<span id="cb10-2555"><a href="#cb10-2555" aria-hidden="true" tabindex="-1"></a>have been regressed quite well, so we can actually compare the results using </span>
<span id="cb10-2556"><a href="#cb10-2556" aria-hidden="true" tabindex="-1"></a>the scores. On the other hand we can't use the pathways G2M checkpoint,</span>
<span id="cb10-2557"><a href="#cb10-2557" aria-hidden="true" tabindex="-1"></a>E2F targets and EMT. For the PI3K signaling score it seems it is </span>
<span id="cb10-2558"><a href="#cb10-2558" aria-hidden="true" tabindex="-1"></a>comparable to SCANB only.</span>
<span id="cb10-2559"><a href="#cb10-2559" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2560"><a href="#cb10-2560" aria-hidden="true" tabindex="-1"></a><span class="fu">## Using several pathways or just one pathway for GSVA</span></span>
<span id="cb10-2561"><a href="#cb10-2561" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2562"><a href="#cb10-2562" aria-hidden="true" tabindex="-1"></a>We now evaluate the effect of the scores for each pathway when using all the</span>
<span id="cb10-2563"><a href="#cb10-2563" aria-hidden="true" tabindex="-1"></a>53 pathways combined and each one individually. We use only TCGA for </span>
<span id="cb10-2564"><a href="#cb10-2564" aria-hidden="true" tabindex="-1"></a>this step.</span>
<span id="cb10-2565"><a href="#cb10-2565" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2568"><a href="#cb10-2568" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-2569"><a href="#cb10-2569" aria-hidden="true" tabindex="-1"></a>gene_expression <span class="ot">&lt;-</span> <span class="fu">assay</span>(datasets<span class="sc">$</span>tcga, which_exp<span class="sc">$</span>tcga) <span class="sc">%&gt;%</span> as.matrix</span>
<span id="cb10-2570"><a href="#cb10-2570" aria-hidden="true" tabindex="-1"></a>all_scores <span class="ot">&lt;-</span> <span class="fu">colData</span>(datasets<span class="sc">$</span>tcga) <span class="sc">%&gt;%</span> data.frame</span>
<span id="cb10-2571"><a href="#cb10-2571" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-2572"><a href="#cb10-2572" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2573"><a href="#cb10-2573" aria-hidden="true" tabindex="-1"></a><span class="fu">### One vs all</span></span>
<span id="cb10-2574"><a href="#cb10-2574" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2575"><a href="#cb10-2575" aria-hidden="true" tabindex="-1"></a>We now calculate the scores for each gene set individually.</span>
<span id="cb10-2576"><a href="#cb10-2576" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2577"><a href="#cb10-2577" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb10-2578"><a href="#cb10-2578" aria-hidden="true" tabindex="-1"></a>gsva_one_pathway <span class="ot">&lt;-</span> parallel<span class="sc">::</span><span class="fu">mclapply</span>(</span>
<span id="cb10-2579"><a href="#cb10-2579" aria-hidden="true" tabindex="-1"></a>    gene_sets_,</span>
<span id="cb10-2580"><a href="#cb10-2580" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(x){</span>
<span id="cb10-2581"><a href="#cb10-2581" aria-hidden="true" tabindex="-1"></a>        GSVA<span class="sc">::</span><span class="fu">gsva</span>(</span>
<span id="cb10-2582"><a href="#cb10-2582" aria-hidden="true" tabindex="-1"></a>            gene_expression, </span>
<span id="cb10-2583"><a href="#cb10-2583" aria-hidden="true" tabindex="-1"></a>            <span class="fu">list</span>(x)</span>
<span id="cb10-2584"><a href="#cb10-2584" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb10-2585"><a href="#cb10-2585" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb10-2586"><a href="#cb10-2586" aria-hidden="true" tabindex="-1"></a>    <span class="at">mc.cores =</span> <span class="dv">5</span></span>
<span id="cb10-2587"><a href="#cb10-2587" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-2588"><a href="#cb10-2588" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2589"><a href="#cb10-2589" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb10-2590"><a href="#cb10-2590" aria-hidden="true" tabindex="-1"></a>    gsva_one_pathway,</span>
<span id="cb10-2591"><a href="#cb10-2591" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/trying/gsva_one_pathway.rds"</span></span>
<span id="cb10-2592"><a href="#cb10-2592" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-2593"><a href="#cb10-2593" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-2594"><a href="#cb10-2594" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2597"><a href="#cb10-2597" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-2598"><a href="#cb10-2598" aria-hidden="true" tabindex="-1"></a>gsva_one_pathways <span class="ot">&lt;-</span> <span class="fu">Reduce</span>(rbind, gsva_one_pathway) <span class="sc">%&gt;%</span></span>
<span id="cb10-2599"><a href="#cb10-2599" aria-hidden="true" tabindex="-1"></a>    t <span class="sc">%&gt;%</span></span>
<span id="cb10-2600"><a href="#cb10-2600" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span> </span>
<span id="cb10-2601"><a href="#cb10-2601" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">colnames&lt;-</span><span class="st">`</span>(<span class="fu">names</span>(gsva_one_pathway)) <span class="sc">%&gt;%</span></span>
<span id="cb10-2602"><a href="#cb10-2602" aria-hidden="true" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"sample_name"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-2603"><a href="#cb10-2603" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(</span>
<span id="cb10-2604"><a href="#cb10-2604" aria-hidden="true" tabindex="-1"></a>        <span class="at">cols =</span> dplyr<span class="sc">::</span><span class="fu">all_of</span>(<span class="fu">names</span>(gsva_one_pathway)),</span>
<span id="cb10-2605"><a href="#cb10-2605" aria-hidden="true" tabindex="-1"></a>        <span class="at">names_to =</span> <span class="st">"pathway"</span>,</span>
<span id="cb10-2606"><a href="#cb10-2606" aria-hidden="true" tabindex="-1"></a>        <span class="at">values_to =</span> <span class="st">"one_gsva"</span></span>
<span id="cb10-2607"><a href="#cb10-2607" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb10-2608"><a href="#cb10-2608" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">inner_join</span>(</span>
<span id="cb10-2609"><a href="#cb10-2609" aria-hidden="true" tabindex="-1"></a>        merged_col_data <span class="sc">%&gt;%</span></span>
<span id="cb10-2610"><a href="#cb10-2610" aria-hidden="true" tabindex="-1"></a>            tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(</span>
<span id="cb10-2611"><a href="#cb10-2611" aria-hidden="true" tabindex="-1"></a>                <span class="at">cols =</span> dplyr<span class="sc">::</span><span class="fu">all_of</span>(<span class="fu">names</span>(gsva_one_pathway)),</span>
<span id="cb10-2612"><a href="#cb10-2612" aria-hidden="true" tabindex="-1"></a>                <span class="at">names_to =</span> <span class="st">"pathway"</span>,</span>
<span id="cb10-2613"><a href="#cb10-2613" aria-hidden="true" tabindex="-1"></a>                <span class="at">values_to =</span> <span class="st">"gsva"</span></span>
<span id="cb10-2614"><a href="#cb10-2614" aria-hidden="true" tabindex="-1"></a>            ),</span>
<span id="cb10-2615"><a href="#cb10-2615" aria-hidden="true" tabindex="-1"></a>        .,</span>
<span id="cb10-2616"><a href="#cb10-2616" aria-hidden="true" tabindex="-1"></a>        <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"sample_name"</span>, <span class="st">"pathway"</span>)</span>
<span id="cb10-2617"><a href="#cb10-2617" aria-hidden="true" tabindex="-1"></a>    ) </span>
<span id="cb10-2618"><a href="#cb10-2618" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-2619"><a href="#cb10-2619" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2620"><a href="#cb10-2620" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=20, fig.height=16}</span></span>
<span id="cb10-2621"><a href="#cb10-2621" aria-hidden="true" tabindex="-1"></a>gsva_one_pathways <span class="sc">%&gt;%</span></span>
<span id="cb10-2622"><a href="#cb10-2622" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(</span>
<span id="cb10-2623"><a href="#cb10-2623" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> one_gsva, <span class="at">y =</span> gsva, <span class="at">color =</span> pam50</span>
<span id="cb10-2624"><a href="#cb10-2624" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">+</span></span>
<span id="cb10-2625"><a href="#cb10-2625" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb10-2626"><a href="#cb10-2626" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_abline</span>(<span class="at">intercept =</span> <span class="dv">0</span>, <span class="at">slope =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb10-2627"><a href="#cb10-2627" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>pathway, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb10-2628"><a href="#cb10-2628" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>()</span>
<span id="cb10-2629"><a href="#cb10-2629" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-2630"><a href="#cb10-2630" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2631"><a href="#cb10-2631" aria-hidden="true" tabindex="-1"></a>It seems they are all positively correlated, so one could calculate the </span>
<span id="cb10-2632"><a href="#cb10-2632" aria-hidden="true" tabindex="-1"></a>scores basically using just one gene set with exception of some of the</span>
<span id="cb10-2633"><a href="#cb10-2633" aria-hidden="true" tabindex="-1"></a>gene sets. </span>
<span id="cb10-2634"><a href="#cb10-2634" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2635"><a href="#cb10-2635" aria-hidden="true" tabindex="-1"></a><span class="fu">## Minimum number of samples for GSVA</span></span>
<span id="cb10-2636"><a href="#cb10-2636" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2637"><a href="#cb10-2637" aria-hidden="true" tabindex="-1"></a>According to the original GSVA paper, a minimum of 10 samples are necessary</span>
<span id="cb10-2638"><a href="#cb10-2638" aria-hidden="true" tabindex="-1"></a>to have a good power when using GSVA. Since here we have at least 5 </span>
<span id="cb10-2639"><a href="#cb10-2639" aria-hidden="true" tabindex="-1"></a>molecular subgroups, we hypothesize we need at least 50 samples to get a </span>
<span id="cb10-2640"><a href="#cb10-2640" aria-hidden="true" tabindex="-1"></a>good score so we can compare across cohorts. We will try with</span>
<span id="cb10-2641"><a href="#cb10-2641" aria-hidden="true" tabindex="-1"></a>different numbers of samples using TCGA, we start with 10, 25, 50 and</span>
<span id="cb10-2642"><a href="#cb10-2642" aria-hidden="true" tabindex="-1"></a>100 samples and we compare with the scores using the full dataset.</span>
<span id="cb10-2643"><a href="#cb10-2643" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2646"><a href="#cb10-2646" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-2647"><a href="#cb10-2647" aria-hidden="true" tabindex="-1"></a>nb_of_samples <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">10</span>, <span class="dv">25</span>, <span class="dv">40</span>, <span class="dv">50</span>, <span class="dv">100</span>, <span class="dv">150</span>)</span>
<span id="cb10-2648"><a href="#cb10-2648" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-2649"><a href="#cb10-2649" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2650"><a href="#cb10-2650" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb10-2651"><a href="#cb10-2651" aria-hidden="true" tabindex="-1"></a>seeds <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">12</span>, <span class="dv">123</span>, <span class="dv">1234</span>, <span class="dv">21345</span>, <span class="dv">12</span>, <span class="dv">123</span>)</span>
<span id="cb10-2652"><a href="#cb10-2652" aria-hidden="true" tabindex="-1"></a>gsva_scores <span class="ot">&lt;-</span> <span class="fu">mapply</span>(</span>
<span id="cb10-2653"><a href="#cb10-2653" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(nb_samples, seed, df, gene_sets, col_data){</span>
<span id="cb10-2654"><a href="#cb10-2654" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb10-2655"><a href="#cb10-2655" aria-hidden="true" tabindex="-1"></a>        <span class="fu">set.seed</span>(seed)</span>
<span id="cb10-2656"><a href="#cb10-2656" aria-hidden="true" tabindex="-1"></a>        patients <span class="ot">&lt;-</span> col_data <span class="sc">%&gt;%</span></span>
<span id="cb10-2657"><a href="#cb10-2657" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">slice_sample</span>(<span class="at">n =</span> nb_samples) <span class="sc">%&gt;%</span></span>
<span id="cb10-2658"><a href="#cb10-2658" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">pull</span>(sample_name)</span>
<span id="cb10-2659"><a href="#cb10-2659" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb10-2660"><a href="#cb10-2660" aria-hidden="true" tabindex="-1"></a>        GSVA<span class="sc">::</span><span class="fu">gsva</span>(</span>
<span id="cb10-2661"><a href="#cb10-2661" aria-hidden="true" tabindex="-1"></a>            df[, patients],</span>
<span id="cb10-2662"><a href="#cb10-2662" aria-hidden="true" tabindex="-1"></a>            gene_sets</span>
<span id="cb10-2663"><a href="#cb10-2663" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb10-2664"><a href="#cb10-2664" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb10-2665"><a href="#cb10-2665" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb10-2666"><a href="#cb10-2666" aria-hidden="true" tabindex="-1"></a>    <span class="at">nb_samples =</span> nb_of_samples,</span>
<span id="cb10-2667"><a href="#cb10-2667" aria-hidden="true" tabindex="-1"></a>    <span class="at">seed =</span> seeds,</span>
<span id="cb10-2668"><a href="#cb10-2668" aria-hidden="true" tabindex="-1"></a>    <span class="at">MoreArgs =</span> <span class="fu">list</span>(<span class="at">df =</span> gene_expression, <span class="at">gene_sets =</span> gene_sets_,</span>
<span id="cb10-2669"><a href="#cb10-2669" aria-hidden="true" tabindex="-1"></a>                    <span class="at">col_data =</span> <span class="fu">colData</span>(datasets<span class="sc">$</span>tcga) <span class="sc">%&gt;%</span> data.frame)</span>
<span id="cb10-2670"><a href="#cb10-2670" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-2671"><a href="#cb10-2671" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2672"><a href="#cb10-2672" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb10-2673"><a href="#cb10-2673" aria-hidden="true" tabindex="-1"></a>    gsva_scores, </span>
<span id="cb10-2674"><a href="#cb10-2674" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/trying/gsva_scores_trying.rds"</span></span>
<span id="cb10-2675"><a href="#cb10-2675" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-2676"><a href="#cb10-2676" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-2677"><a href="#cb10-2677" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2680"><a href="#cb10-2680" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-2681"><a href="#cb10-2681" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(gsva_scores) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"nb_"</span>, nb_of_samples)</span>
<span id="cb10-2682"><a href="#cb10-2682" aria-hidden="true" tabindex="-1"></a>gsva_scores_ <span class="ot">&lt;-</span> <span class="fu">lapply</span>(</span>
<span id="cb10-2683"><a href="#cb10-2683" aria-hidden="true" tabindex="-1"></a>    gsva_scores,</span>
<span id="cb10-2684"><a href="#cb10-2684" aria-hidden="true" tabindex="-1"></a>    \(x) <span class="fu">data.frame</span>(<span class="fu">t</span>(x)) <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"sample_name"</span>)</span>
<span id="cb10-2685"><a href="#cb10-2685" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb10-2686"><a href="#cb10-2686" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(<span class="at">.id =</span> <span class="st">"nb_of_samples"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-2687"><a href="#cb10-2687" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">nb_of_samples =</span> <span class="fu">factor</span>(nb_of_samples, <span class="fu">names</span>(gsva_scores))) <span class="sc">%&gt;%</span></span>
<span id="cb10-2688"><a href="#cb10-2688" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(</span>
<span id="cb10-2689"><a href="#cb10-2689" aria-hidden="true" tabindex="-1"></a>        <span class="at">cols =</span> <span class="fu">setdiff</span>(<span class="fu">names</span>(gene_sets_), p4_pathways<span class="sc">$</span>gs_name <span class="sc">%&gt;%</span> unique),</span>
<span id="cb10-2690"><a href="#cb10-2690" aria-hidden="true" tabindex="-1"></a>        <span class="at">names_to =</span> <span class="st">"pathway"</span>,</span>
<span id="cb10-2691"><a href="#cb10-2691" aria-hidden="true" tabindex="-1"></a>        <span class="at">values_to =</span> <span class="st">"gsva_sub"</span></span>
<span id="cb10-2692"><a href="#cb10-2692" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb10-2693"><a href="#cb10-2693" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">inner_join</span>(</span>
<span id="cb10-2694"><a href="#cb10-2694" aria-hidden="true" tabindex="-1"></a>        merged_col_data <span class="sc">%&gt;%</span></span>
<span id="cb10-2695"><a href="#cb10-2695" aria-hidden="true" tabindex="-1"></a>            tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(</span>
<span id="cb10-2696"><a href="#cb10-2696" aria-hidden="true" tabindex="-1"></a>            <span class="at">cols =</span> <span class="fu">setdiff</span>(<span class="fu">names</span>(gene_sets_), p4_pathways<span class="sc">$</span>gs_name <span class="sc">%&gt;%</span> unique),</span>
<span id="cb10-2697"><a href="#cb10-2697" aria-hidden="true" tabindex="-1"></a>            <span class="at">names_to =</span> <span class="st">"pathway"</span>,</span>
<span id="cb10-2698"><a href="#cb10-2698" aria-hidden="true" tabindex="-1"></a>            <span class="at">values_to =</span> <span class="st">"gsva_full"</span></span>
<span id="cb10-2699"><a href="#cb10-2699" aria-hidden="true" tabindex="-1"></a>        ),</span>
<span id="cb10-2700"><a href="#cb10-2700" aria-hidden="true" tabindex="-1"></a>        .,</span>
<span id="cb10-2701"><a href="#cb10-2701" aria-hidden="true" tabindex="-1"></a>        <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"sample_name"</span>, <span class="st">"pathway"</span>)</span>
<span id="cb10-2702"><a href="#cb10-2702" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb10-2703"><a href="#cb10-2703" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-2704"><a href="#cb10-2704" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2705"><a href="#cb10-2705" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width = 50, fig.height=15}</span></span>
<span id="cb10-2706"><a href="#cb10-2706" aria-hidden="true" tabindex="-1"></a>gsva_scores_ <span class="sc">%&gt;%</span> </span>
<span id="cb10-2707"><a href="#cb10-2707" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(pathway <span class="sc">%in%</span> <span class="fu">names</span>(gene_sets_)[<span class="dv">1</span><span class="sc">:</span><span class="dv">25</span>]) <span class="sc">%&gt;%</span></span>
<span id="cb10-2708"><a href="#cb10-2708" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(</span>
<span id="cb10-2709"><a href="#cb10-2709" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> gsva_full, <span class="at">y =</span> gsva_sub, <span class="at">color =</span> pam50</span>
<span id="cb10-2710"><a href="#cb10-2710" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">+</span></span>
<span id="cb10-2711"><a href="#cb10-2711" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span> </span>
<span id="cb10-2712"><a href="#cb10-2712" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_abline</span>(<span class="at">intercept =</span> <span class="dv">0</span>, <span class="at">slope =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb10-2713"><a href="#cb10-2713" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_grid</span>(nb_of_samples <span class="sc">~</span> pathway, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb10-2714"><a href="#cb10-2714" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>()</span>
<span id="cb10-2715"><a href="#cb10-2715" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-2716"><a href="#cb10-2716" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2717"><a href="#cb10-2717" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width = 50, fig.height=15}</span></span>
<span id="cb10-2718"><a href="#cb10-2718" aria-hidden="true" tabindex="-1"></a>gsva_scores_ <span class="sc">%&gt;%</span> </span>
<span id="cb10-2719"><a href="#cb10-2719" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(pathway <span class="sc">%in%</span> <span class="fu">names</span>(gene_sets_)[<span class="dv">25</span><span class="sc">:</span>(<span class="fu">length</span>(<span class="fu">names</span>(gene_sets_))<span class="sc">-</span><span class="dv">2</span>)]) <span class="sc">%&gt;%</span></span>
<span id="cb10-2720"><a href="#cb10-2720" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(</span>
<span id="cb10-2721"><a href="#cb10-2721" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> gsva_full, <span class="at">y =</span> gsva_sub, <span class="at">color =</span> pam50</span>
<span id="cb10-2722"><a href="#cb10-2722" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">+</span></span>
<span id="cb10-2723"><a href="#cb10-2723" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span> </span>
<span id="cb10-2724"><a href="#cb10-2724" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_abline</span>(<span class="at">intercept =</span> <span class="dv">0</span>, <span class="at">slope =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb10-2725"><a href="#cb10-2725" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_grid</span>(nb_of_samples <span class="sc">~</span> pathway, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb10-2726"><a href="#cb10-2726" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>()</span>
<span id="cb10-2727"><a href="#cb10-2727" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-2728"><a href="#cb10-2728" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2729"><a href="#cb10-2729" aria-hidden="true" tabindex="-1"></a>The conclusion we can take here is that depending on the gene set not even 150</span>
<span id="cb10-2730"><a href="#cb10-2730" aria-hidden="true" tabindex="-1"></a>samples is enough. For some only 25 patients is already enough, such as SET </span>
<span id="cb10-2731"><a href="#cb10-2731" aria-hidden="true" tabindex="-1"></a>ER/PR, androgen response, estrogen response early and E2F targets.</span>
<span id="cb10-2732"><a href="#cb10-2732" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2733"><a href="#cb10-2733" aria-hidden="true" tabindex="-1"></a>Could we mix around 20 samples of SCANB together with more samples from </span>
<span id="cb10-2734"><a href="#cb10-2734" aria-hidden="true" tabindex="-1"></a>TCGA and METABRIC to get the scores? Let us try to adress this question</span>
<span id="cb10-2735"><a href="#cb10-2735" aria-hidden="true" tabindex="-1"></a>again. We are using the regressed data now here as this is the only way.</span>
<span id="cb10-2736"><a href="#cb10-2736" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2737"><a href="#cb10-2737" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb10-2738"><a href="#cb10-2738" aria-hidden="true" tabindex="-1"></a>scanb_samples <span class="ot">&lt;-</span> df_pca_coordinates <span class="sc">%&gt;%</span></span>
<span id="cb10-2739"><a href="#cb10-2739" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"scanb"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-2740"><a href="#cb10-2740" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">slice_sample</span>(<span class="at">n =</span> <span class="dv">10</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-2741"><a href="#cb10-2741" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">pull</span>(sample_name)</span>
<span id="cb10-2742"><a href="#cb10-2742" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2743"><a href="#cb10-2743" aria-hidden="true" tabindex="-1"></a>gsva_scores_embeddings_scanb_within_tcga_regressed <span class="ot">&lt;-</span> GSVA<span class="sc">::</span><span class="fu">gsva</span>(</span>
<span id="cb10-2744"><a href="#cb10-2744" aria-hidden="true" tabindex="-1"></a>    df_pcs_regressed[</span>
<span id="cb10-2745"><a href="#cb10-2745" aria-hidden="true" tabindex="-1"></a>        , </span>
<span id="cb10-2746"><a href="#cb10-2746" aria-hidden="true" tabindex="-1"></a>        merged_col_data <span class="sc">%&gt;%</span></span>
<span id="cb10-2747"><a href="#cb10-2747" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"tcga"</span>, <span class="st">"metabric"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb10-2748"><a href="#cb10-2748" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">slice_sample</span>(<span class="at">n =</span> <span class="dv">500</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-2749"><a href="#cb10-2749" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">pull</span>(sample_name) <span class="sc">%&gt;%</span></span>
<span id="cb10-2750"><a href="#cb10-2750" aria-hidden="true" tabindex="-1"></a>            <span class="fu">c</span>(., scanb_samples)</span>
<span id="cb10-2751"><a href="#cb10-2751" aria-hidden="true" tabindex="-1"></a>    ],</span>
<span id="cb10-2752"><a href="#cb10-2752" aria-hidden="true" tabindex="-1"></a>    gene_sets_,</span>
<span id="cb10-2753"><a href="#cb10-2753" aria-hidden="true" tabindex="-1"></a>    <span class="at">parallel.sz =</span> <span class="dv">10</span></span>
<span id="cb10-2754"><a href="#cb10-2754" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-2755"><a href="#cb10-2755" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2756"><a href="#cb10-2756" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb10-2757"><a href="#cb10-2757" aria-hidden="true" tabindex="-1"></a>    gsva_scores_embeddings_scanb_within_tcga_regressed,</span>
<span id="cb10-2758"><a href="#cb10-2758" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/trying/gsva_scores_embeddings_scanb_within_tcga_regressed.rds"</span></span>
<span id="cb10-2759"><a href="#cb10-2759" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-2760"><a href="#cb10-2760" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-2761"><a href="#cb10-2761" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2764"><a href="#cb10-2764" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-2765"><a href="#cb10-2765" aria-hidden="true" tabindex="-1"></a>sub_scanb_scores <span class="ot">&lt;-</span> gsva_scores_embeddings_scanb_within_tcga_regressed <span class="sc">%&gt;%</span></span>
<span id="cb10-2766"><a href="#cb10-2766" aria-hidden="true" tabindex="-1"></a>    t <span class="sc">%&gt;%</span> </span>
<span id="cb10-2767"><a href="#cb10-2767" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span></span>
<span id="cb10-2768"><a href="#cb10-2768" aria-hidden="true" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"sample_name"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-2769"><a href="#cb10-2769" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(</span>
<span id="cb10-2770"><a href="#cb10-2770" aria-hidden="true" tabindex="-1"></a>        <span class="at">cols =</span> <span class="fu">setdiff</span>(<span class="fu">names</span>(gene_sets_), p4_pathways<span class="sc">$</span>gs_name <span class="sc">%&gt;%</span> unique),</span>
<span id="cb10-2771"><a href="#cb10-2771" aria-hidden="true" tabindex="-1"></a>        <span class="at">names_to =</span> <span class="st">"pathway"</span>,</span>
<span id="cb10-2772"><a href="#cb10-2772" aria-hidden="true" tabindex="-1"></a>        <span class="at">values_to =</span> <span class="st">"gsva_sub"</span></span>
<span id="cb10-2773"><a href="#cb10-2773" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb10-2774"><a href="#cb10-2774" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">inner_join</span>(</span>
<span id="cb10-2775"><a href="#cb10-2775" aria-hidden="true" tabindex="-1"></a>        merged_col_data <span class="sc">%&gt;%</span></span>
<span id="cb10-2776"><a href="#cb10-2776" aria-hidden="true" tabindex="-1"></a>            tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(</span>
<span id="cb10-2777"><a href="#cb10-2777" aria-hidden="true" tabindex="-1"></a>                <span class="at">cols =</span> <span class="fu">setdiff</span>(<span class="fu">names</span>(gene_sets_), p4_pathways<span class="sc">$</span>gs_name <span class="sc">%&gt;%</span> unique),</span>
<span id="cb10-2778"><a href="#cb10-2778" aria-hidden="true" tabindex="-1"></a>                <span class="at">names_to =</span> <span class="st">"pathway"</span>,</span>
<span id="cb10-2779"><a href="#cb10-2779" aria-hidden="true" tabindex="-1"></a>                <span class="at">values_to =</span> <span class="st">"gsva_full"</span></span>
<span id="cb10-2780"><a href="#cb10-2780" aria-hidden="true" tabindex="-1"></a>            ),</span>
<span id="cb10-2781"><a href="#cb10-2781" aria-hidden="true" tabindex="-1"></a>        .,</span>
<span id="cb10-2782"><a href="#cb10-2782" aria-hidden="true" tabindex="-1"></a>        <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"sample_name"</span>, <span class="st">"pathway"</span>)</span>
<span id="cb10-2783"><a href="#cb10-2783" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb10-2784"><a href="#cb10-2784" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-2785"><a href="#cb10-2785" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2786"><a href="#cb10-2786" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=20, fig.height=16}</span></span>
<span id="cb10-2787"><a href="#cb10-2787" aria-hidden="true" tabindex="-1"></a>sub_scanb_scores <span class="sc">%&gt;%</span></span>
<span id="cb10-2788"><a href="#cb10-2788" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"scanb"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-2789"><a href="#cb10-2789" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(</span>
<span id="cb10-2790"><a href="#cb10-2790" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> gsva_sub,</span>
<span id="cb10-2791"><a href="#cb10-2791" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> gsva_full,</span>
<span id="cb10-2792"><a href="#cb10-2792" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> cohort</span>
<span id="cb10-2793"><a href="#cb10-2793" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">+</span></span>
<span id="cb10-2794"><a href="#cb10-2794" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb10-2795"><a href="#cb10-2795" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_abline</span>(<span class="at">intercept =</span> <span class="dv">0</span>, <span class="at">slope =</span> <span class="dv">1</span>) <span class="sc">+</span> </span>
<span id="cb10-2796"><a href="#cb10-2796" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>pathway) <span class="sc">+</span> </span>
<span id="cb10-2797"><a href="#cb10-2797" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>()</span>
<span id="cb10-2798"><a href="#cb10-2798" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-2799"><a href="#cb10-2799" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2800"><a href="#cb10-2800" aria-hidden="true" tabindex="-1"></a>There is still some correlation but the scores are </span>
<span id="cb10-2801"><a href="#cb10-2801" aria-hidden="true" tabindex="-1"></a>shrinked in the end. </span>
<span id="cb10-2802"><a href="#cb10-2802" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2803"><a href="#cb10-2803" aria-hidden="true" tabindex="-1"></a><span class="fu">## Normalization strategy</span></span>
<span id="cb10-2804"><a href="#cb10-2804" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2805"><a href="#cb10-2805" aria-hidden="true" tabindex="-1"></a>We can also try another way of normalizing, namely for each sample we</span>
<span id="cb10-2806"><a href="#cb10-2806" aria-hidden="true" tabindex="-1"></a>can z scale it.</span>
<span id="cb10-2807"><a href="#cb10-2807" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2810"><a href="#cb10-2810" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-2811"><a href="#cb10-2811" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1329</span>)</span>
<span id="cb10-2812"><a href="#cb10-2812" aria-hidden="true" tabindex="-1"></a>samples_for_training <span class="ot">&lt;-</span> merged_col_data <span class="sc">%&gt;%</span></span>
<span id="cb10-2813"><a href="#cb10-2813" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">%in%</span> which_cohorts_training) <span class="sc">%&gt;%</span></span>
<span id="cb10-2814"><a href="#cb10-2814" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">pull</span>(sample_name) <span class="sc">%&gt;%</span></span>
<span id="cb10-2815"><a href="#cb10-2815" aria-hidden="true" tabindex="-1"></a>    <span class="fu">sample</span>(., <span class="at">size =</span> <span class="dv">1000</span>) </span>
<span id="cb10-2816"><a href="#cb10-2816" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2817"><a href="#cb10-2817" aria-hidden="true" tabindex="-1"></a>common_genes <span class="ot">&lt;-</span> <span class="fu">Reduce</span>(intersect, <span class="fu">lapply</span>(datasets, rownames))</span>
<span id="cb10-2818"><a href="#cb10-2818" aria-hidden="true" tabindex="-1"></a>stable_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(stable_genes, common_genes)</span>
<span id="cb10-2819"><a href="#cb10-2819" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2820"><a href="#cb10-2820" aria-hidden="true" tabindex="-1"></a><span class="co"># and now we can perform the molecular embedding</span></span>
<span id="cb10-2821"><a href="#cb10-2821" aria-hidden="true" tabindex="-1"></a>training_set <span class="ot">&lt;-</span> <span class="fu">lapply</span>(</span>
<span id="cb10-2822"><a href="#cb10-2822" aria-hidden="true" tabindex="-1"></a>    datasets_normalized[which_cohorts_training], </span>
<span id="cb10-2823"><a href="#cb10-2823" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(sum_exp, i, genes_for_pca) </span>
<span id="cb10-2824"><a href="#cb10-2824" aria-hidden="true" tabindex="-1"></a>        <span class="fu">assay</span>(sum_exp[genes_for_pca, ], <span class="at">i =</span> i) <span class="sc">%&gt;%</span> </span>
<span id="cb10-2825"><a href="#cb10-2825" aria-hidden="true" tabindex="-1"></a>        <span class="fu">data.frame</span>(<span class="at">check.names =</span> <span class="cn">FALSE</span>), </span>
<span id="cb10-2826"><a href="#cb10-2826" aria-hidden="true" tabindex="-1"></a>    <span class="at">i =</span> <span class="st">"znorm"</span>,</span>
<span id="cb10-2827"><a href="#cb10-2827" aria-hidden="true" tabindex="-1"></a>    <span class="at">genes_for_pca =</span> common_genes</span>
<span id="cb10-2828"><a href="#cb10-2828" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-2829"><a href="#cb10-2829" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">bind_cols</span>() <span class="sc">%&gt;%</span></span>
<span id="cb10-2830"><a href="#cb10-2830" aria-hidden="true" tabindex="-1"></a>    .[, samples_for_training]</span>
<span id="cb10-2831"><a href="#cb10-2831" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2832"><a href="#cb10-2832" aria-hidden="true" tabindex="-1"></a>pca_fit_znorm <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">pca</span>(</span>
<span id="cb10-2833"><a href="#cb10-2833" aria-hidden="true" tabindex="-1"></a>    training_set,</span>
<span id="cb10-2834"><a href="#cb10-2834" aria-hidden="true" tabindex="-1"></a>    <span class="at">metadata =</span> dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(</span>
<span id="cb10-2835"><a href="#cb10-2835" aria-hidden="true" tabindex="-1"></a>        <span class="fu">lapply</span>(</span>
<span id="cb10-2836"><a href="#cb10-2836" aria-hidden="true" tabindex="-1"></a>            datasets_normalized[which_cohorts_training],</span>
<span id="cb10-2837"><a href="#cb10-2837" aria-hidden="true" tabindex="-1"></a>            <span class="cf">function</span>(df){</span>
<span id="cb10-2838"><a href="#cb10-2838" aria-hidden="true" tabindex="-1"></a>                <span class="fu">colData</span>(df) <span class="sc">%&gt;%</span> data.frame <span class="sc">%&gt;%</span></span>
<span id="cb10-2839"><a href="#cb10-2839" aria-hidden="true" tabindex="-1"></a>                    dplyr<span class="sc">::</span><span class="fu">filter</span>(sample_name <span class="sc">%in%</span> <span class="fu">colnames</span>(training_set))</span>
<span id="cb10-2840"><a href="#cb10-2840" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb10-2841"><a href="#cb10-2841" aria-hidden="true" tabindex="-1"></a>        ),</span>
<span id="cb10-2842"><a href="#cb10-2842" aria-hidden="true" tabindex="-1"></a>        <span class="at">.id =</span> <span class="st">"cohort"</span></span>
<span id="cb10-2843"><a href="#cb10-2843" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span> .[<span class="fu">colnames</span>(training_set), ],</span>
<span id="cb10-2844"><a href="#cb10-2844" aria-hidden="true" tabindex="-1"></a>    <span class="at">center =</span> <span class="cn">FALSE</span>,</span>
<span id="cb10-2845"><a href="#cb10-2845" aria-hidden="true" tabindex="-1"></a>    <span class="at">scale =</span> <span class="cn">FALSE</span></span>
<span id="cb10-2846"><a href="#cb10-2846" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-2847"><a href="#cb10-2847" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-2848"><a href="#cb10-2848" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2849"><a href="#cb10-2849" aria-hidden="true" tabindex="-1"></a>And now we can visualize the results.</span>
<span id="cb10-2850"><a href="#cb10-2850" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2851"><a href="#cb10-2851" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=14, fig.height=16}</span></span>
<span id="cb10-2852"><a href="#cb10-2852" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-pca-embeddings-3</span></span>
<span id="cb10-2853"><a href="#cb10-2853" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: PCA projections colored by different factors and organized</span></span>
<span id="cb10-2854"><a href="#cb10-2854" aria-hidden="true" tabindex="-1"></a><span class="co">#|    by different components. (A) Plot of the first two components colored</span></span>
<span id="cb10-2855"><a href="#cb10-2855" aria-hidden="true" tabindex="-1"></a><span class="co">#|    by cohort. (B) Plot of the first two components colored by ER status.</span></span>
<span id="cb10-2856"><a href="#cb10-2856" aria-hidden="true" tabindex="-1"></a><span class="co">#|    (C) Plot of the second and third components colored by cohort. </span></span>
<span id="cb10-2857"><a href="#cb10-2857" aria-hidden="true" tabindex="-1"></a><span class="co">#|    (D) Plot of the second and third components colored by ER status.</span></span>
<span id="cb10-2858"><a href="#cb10-2858" aria-hidden="true" tabindex="-1"></a><span class="co">#|    (E) Plot of the second and third components colored by PAM50.</span></span>
<span id="cb10-2859"><a href="#cb10-2859" aria-hidden="true" tabindex="-1"></a><span class="co">#|    (F) Plot of the second and third components colored by INTCLUST, only</span></span>
<span id="cb10-2860"><a href="#cb10-2860" aria-hidden="true" tabindex="-1"></a><span class="co">#|    METABRIC has an assigned value for this variable, NAs are TCGA samples.</span></span>
<span id="cb10-2861"><a href="#cb10-2861" aria-hidden="true" tabindex="-1"></a>plots_pca_fit <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb10-2862"><a href="#cb10-2862" aria-hidden="true" tabindex="-1"></a>point_size <span class="ot">&lt;-</span> <span class="dv">2</span></span>
<span id="cb10-2863"><a href="#cb10-2863" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2864"><a href="#cb10-2864" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>pc1_pc2_cohort <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb10-2865"><a href="#cb10-2865" aria-hidden="true" tabindex="-1"></a>    pca_fit_znorm,</span>
<span id="cb10-2866"><a href="#cb10-2866" aria-hidden="true" tabindex="-1"></a>    <span class="at">colby =</span> <span class="st">"cohort"</span>,</span>
<span id="cb10-2867"><a href="#cb10-2867" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> <span class="cn">NULL</span>, </span>
<span id="cb10-2868"><a href="#cb10-2868" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb10-2869"><a href="#cb10-2869" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC1"</span>,</span>
<span id="cb10-2870"><a href="#cb10-2870" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC2"</span>,</span>
<span id="cb10-2871"><a href="#cb10-2871" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"First two components colored by cohort"</span>,</span>
<span id="cb10-2872"><a href="#cb10-2872" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">"Only 1000 training samples"</span>,</span>
<span id="cb10-2873"><a href="#cb10-2873" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> point_size</span>
<span id="cb10-2874"><a href="#cb10-2874" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-2875"><a href="#cb10-2875" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2876"><a href="#cb10-2876" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>pc1_pc2_er_status <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb10-2877"><a href="#cb10-2877" aria-hidden="true" tabindex="-1"></a>    pca_fit_znorm,</span>
<span id="cb10-2878"><a href="#cb10-2878" aria-hidden="true" tabindex="-1"></a>    <span class="at">colby =</span> <span class="st">"er_status"</span>,</span>
<span id="cb10-2879"><a href="#cb10-2879" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> <span class="cn">NULL</span>, </span>
<span id="cb10-2880"><a href="#cb10-2880" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb10-2881"><a href="#cb10-2881" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC1"</span>,</span>
<span id="cb10-2882"><a href="#cb10-2882" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC2"</span>,</span>
<span id="cb10-2883"><a href="#cb10-2883" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"First two components colored by ER status"</span>,</span>
<span id="cb10-2884"><a href="#cb10-2884" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">"Only 1000 training samples"</span>,</span>
<span id="cb10-2885"><a href="#cb10-2885" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> point_size</span>
<span id="cb10-2886"><a href="#cb10-2886" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-2887"><a href="#cb10-2887" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2888"><a href="#cb10-2888" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>pc2_pc3_cohort <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb10-2889"><a href="#cb10-2889" aria-hidden="true" tabindex="-1"></a>    pca_fit_znorm,</span>
<span id="cb10-2890"><a href="#cb10-2890" aria-hidden="true" tabindex="-1"></a>    <span class="at">colby =</span> <span class="st">"cohort"</span>,</span>
<span id="cb10-2891"><a href="#cb10-2891" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> <span class="cn">NULL</span>, </span>
<span id="cb10-2892"><a href="#cb10-2892" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb10-2893"><a href="#cb10-2893" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC2"</span>,</span>
<span id="cb10-2894"><a href="#cb10-2894" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC3"</span>,</span>
<span id="cb10-2895"><a href="#cb10-2895" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Second and third components colored by cohort"</span>,</span>
<span id="cb10-2896"><a href="#cb10-2896" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">"Only 1000 training samples"</span>,</span>
<span id="cb10-2897"><a href="#cb10-2897" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> point_size</span>
<span id="cb10-2898"><a href="#cb10-2898" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-2899"><a href="#cb10-2899" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2900"><a href="#cb10-2900" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>pc2_pc3_er_status <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb10-2901"><a href="#cb10-2901" aria-hidden="true" tabindex="-1"></a>    pca_fit_znorm,</span>
<span id="cb10-2902"><a href="#cb10-2902" aria-hidden="true" tabindex="-1"></a>    <span class="at">colby =</span> <span class="st">"er_status"</span>,</span>
<span id="cb10-2903"><a href="#cb10-2903" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> <span class="cn">NULL</span>, </span>
<span id="cb10-2904"><a href="#cb10-2904" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb10-2905"><a href="#cb10-2905" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC2"</span>,</span>
<span id="cb10-2906"><a href="#cb10-2906" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC3"</span>,</span>
<span id="cb10-2907"><a href="#cb10-2907" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Second and third components colored by ER status"</span>,</span>
<span id="cb10-2908"><a href="#cb10-2908" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">"Only 1000 training samples"</span>,</span>
<span id="cb10-2909"><a href="#cb10-2909" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> point_size</span>
<span id="cb10-2910"><a href="#cb10-2910" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-2911"><a href="#cb10-2911" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2912"><a href="#cb10-2912" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>pc2_pc3_pam50 <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb10-2913"><a href="#cb10-2913" aria-hidden="true" tabindex="-1"></a>    pca_fit_znorm,</span>
<span id="cb10-2914"><a href="#cb10-2914" aria-hidden="true" tabindex="-1"></a>    <span class="at">colby =</span> <span class="st">"pam50"</span>,</span>
<span id="cb10-2915"><a href="#cb10-2915" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> <span class="cn">NULL</span>, </span>
<span id="cb10-2916"><a href="#cb10-2916" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb10-2917"><a href="#cb10-2917" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC2"</span>,</span>
<span id="cb10-2918"><a href="#cb10-2918" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC3"</span>,</span>
<span id="cb10-2919"><a href="#cb10-2919" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Second and third components colored by PAM50"</span>,</span>
<span id="cb10-2920"><a href="#cb10-2920" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">"Only 1000 training samples"</span>,</span>
<span id="cb10-2921"><a href="#cb10-2921" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> point_size</span>
<span id="cb10-2922"><a href="#cb10-2922" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-2923"><a href="#cb10-2923" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>pc2_pc3_pam50 <span class="ot">&lt;-</span> plots_pca_fit<span class="sc">$</span>pc2_pc3_pam50 <span class="sc">+</span></span>
<span id="cb10-2924"><a href="#cb10-2924" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">get_colors_pam50</span>(pca_fit<span class="sc">$</span>metadata))</span>
<span id="cb10-2925"><a href="#cb10-2925" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb10-2926"><a href="#cb10-2926" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2927"><a href="#cb10-2927" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>pc2_pc3_intclust <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb10-2928"><a href="#cb10-2928" aria-hidden="true" tabindex="-1"></a>    pca_fit_znorm,</span>
<span id="cb10-2929"><a href="#cb10-2929" aria-hidden="true" tabindex="-1"></a>    <span class="at">colby =</span> <span class="st">"INTCLUST"</span>,</span>
<span id="cb10-2930"><a href="#cb10-2930" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> <span class="cn">NULL</span>, </span>
<span id="cb10-2931"><a href="#cb10-2931" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb10-2932"><a href="#cb10-2932" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC2"</span>,</span>
<span id="cb10-2933"><a href="#cb10-2933" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC3"</span>,</span>
<span id="cb10-2934"><a href="#cb10-2934" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Second and third components colored by INTCLUST"</span>,</span>
<span id="cb10-2935"><a href="#cb10-2935" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">"Only 1000 training samples"</span>,</span>
<span id="cb10-2936"><a href="#cb10-2936" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> point_size</span>
<span id="cb10-2937"><a href="#cb10-2937" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-2938"><a href="#cb10-2938" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2939"><a href="#cb10-2939" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>pc3_pc4_pam50 <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb10-2940"><a href="#cb10-2940" aria-hidden="true" tabindex="-1"></a>    pca_fit_znorm,</span>
<span id="cb10-2941"><a href="#cb10-2941" aria-hidden="true" tabindex="-1"></a>    <span class="at">colby =</span> <span class="st">"pam50"</span>,</span>
<span id="cb10-2942"><a href="#cb10-2942" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> <span class="cn">NULL</span>, </span>
<span id="cb10-2943"><a href="#cb10-2943" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb10-2944"><a href="#cb10-2944" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC3"</span>,</span>
<span id="cb10-2945"><a href="#cb10-2945" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC4"</span>,</span>
<span id="cb10-2946"><a href="#cb10-2946" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Third and fourth components colored by PAM50"</span>,</span>
<span id="cb10-2947"><a href="#cb10-2947" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">"Only 1000 training samples"</span>,</span>
<span id="cb10-2948"><a href="#cb10-2948" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> point_size</span>
<span id="cb10-2949"><a href="#cb10-2949" aria-hidden="true" tabindex="-1"></a>) </span>
<span id="cb10-2950"><a href="#cb10-2950" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>pc3_pc4_pam50 <span class="ot">&lt;-</span> plots_pca_fit<span class="sc">$</span>pc3_pc4_pam50 <span class="sc">+</span></span>
<span id="cb10-2951"><a href="#cb10-2951" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">get_colors_pam50</span>(pca_fit<span class="sc">$</span>metadata))</span>
<span id="cb10-2952"><a href="#cb10-2952" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2953"><a href="#cb10-2953" aria-hidden="true" tabindex="-1"></a>plots_pca_fit<span class="sc">$</span>pc3_pc4_cohort <span class="ot">&lt;-</span> PCAtools<span class="sc">::</span><span class="fu">biplot</span>(</span>
<span id="cb10-2954"><a href="#cb10-2954" aria-hidden="true" tabindex="-1"></a>    pca_fit_znorm,</span>
<span id="cb10-2955"><a href="#cb10-2955" aria-hidden="true" tabindex="-1"></a>    <span class="at">colby =</span> <span class="st">"cohort"</span>,</span>
<span id="cb10-2956"><a href="#cb10-2956" aria-hidden="true" tabindex="-1"></a>    <span class="at">lab =</span> <span class="cn">NULL</span>, </span>
<span id="cb10-2957"><a href="#cb10-2957" aria-hidden="true" tabindex="-1"></a>    <span class="at">legendPosition =</span> <span class="st">"right"</span>,</span>
<span id="cb10-2958"><a href="#cb10-2958" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC3"</span>,</span>
<span id="cb10-2959"><a href="#cb10-2959" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC4"</span>,</span>
<span id="cb10-2960"><a href="#cb10-2960" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Third and fourth components colored by cohort"</span>,</span>
<span id="cb10-2961"><a href="#cb10-2961" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="st">"Only 1000 training samples"</span>,</span>
<span id="cb10-2962"><a href="#cb10-2962" aria-hidden="true" tabindex="-1"></a>    <span class="at">pointSize =</span> point_size</span>
<span id="cb10-2963"><a href="#cb10-2963" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-2964"><a href="#cb10-2964" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2965"><a href="#cb10-2965" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(<span class="at">plotlist =</span> plots_pca_fit, <span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">labels =</span> <span class="st">"AUTO"</span>)</span>
<span id="cb10-2966"><a href="#cb10-2966" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-2967"><a href="#cb10-2967" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2970"><a href="#cb10-2970" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-2971"><a href="#cb10-2971" aria-hidden="true" tabindex="-1"></a>datasets_pca_coordinates_znorm <span class="ot">&lt;-</span> <span class="fu">lapply</span>(</span>
<span id="cb10-2972"><a href="#cb10-2972" aria-hidden="true" tabindex="-1"></a>    datasets_normalized,</span>
<span id="cb10-2973"><a href="#cb10-2973" aria-hidden="true" tabindex="-1"></a>    get_pca_coordinates,</span>
<span id="cb10-2974"><a href="#cb10-2974" aria-hidden="true" tabindex="-1"></a>    <span class="at">pca_fit =</span> pca_fit_znorm,</span>
<span id="cb10-2975"><a href="#cb10-2975" aria-hidden="true" tabindex="-1"></a>    <span class="at">genes_for_pca =</span> <span class="fu">rownames</span>(pca_fit_znorm<span class="sc">$</span>loadings)</span>
<span id="cb10-2976"><a href="#cb10-2976" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-2977"><a href="#cb10-2977" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2978"><a href="#cb10-2978" aria-hidden="true" tabindex="-1"></a>df_pca_coordinates_znorm <span class="ot">&lt;-</span> datasets_pca_coordinates_znorm <span class="sc">%&gt;%</span> </span>
<span id="cb10-2979"><a href="#cb10-2979" aria-hidden="true" tabindex="-1"></a>    <span class="fu">do.call</span>(rbind, .) <span class="sc">%&gt;%</span></span>
<span id="cb10-2980"><a href="#cb10-2980" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span> </span>
<span id="cb10-2981"><a href="#cb10-2981" aria-hidden="true" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"sample_name"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-2982"><a href="#cb10-2982" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">inner_join</span>(</span>
<span id="cb10-2983"><a href="#cb10-2983" aria-hidden="true" tabindex="-1"></a>        .,</span>
<span id="cb10-2984"><a href="#cb10-2984" aria-hidden="true" tabindex="-1"></a>        merged_col_data,</span>
<span id="cb10-2985"><a href="#cb10-2985" aria-hidden="true" tabindex="-1"></a>        <span class="at">by =</span> <span class="st">"sample_name"</span></span>
<span id="cb10-2986"><a href="#cb10-2986" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb10-2987"><a href="#cb10-2987" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-2988"><a href="#cb10-2988" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2991"><a href="#cb10-2991" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-2992"><a href="#cb10-2992" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_pca_coordinates</span>(</span>
<span id="cb10-2993"><a href="#cb10-2993" aria-hidden="true" tabindex="-1"></a>    df_pca_coordinates_znorm <span class="sc">%&gt;%</span> </span>
<span id="cb10-2994"><a href="#cb10-2994" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">slice</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">n</span>()),</span>
<span id="cb10-2995"><a href="#cb10-2995" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"cohort"</span>,</span>
<span id="cb10-2996"><a href="#cb10-2996" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC3"</span>,</span>
<span id="cb10-2997"><a href="#cb10-2997" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC4"</span>,</span>
<span id="cb10-2998"><a href="#cb10-2998" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">1</span>,</span>
<span id="cb10-2999"><a href="#cb10-2999" aria-hidden="true" tabindex="-1"></a>    <span class="at">base_size =</span> <span class="dv">10</span>,</span>
<span id="cb10-3000"><a href="#cb10-3000" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="fu">paste0</span>(</span>
<span id="cb10-3001"><a href="#cb10-3001" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Embedding of all samples from TCGA,</span><span class="sc">\n</span><span class="st">"</span>,</span>
<span id="cb10-3002"><a href="#cb10-3002" aria-hidden="true" tabindex="-1"></a>        <span class="st">"METABRIC and SCANB"</span></span>
<span id="cb10-3003"><a href="#cb10-3003" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb10-3004"><a href="#cb10-3004" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span> ggplot2<span class="sc">::</span><span class="fu">scale_color_viridis_d</span>() <span class="sc">+</span></span>
<span id="cb10-3005"><a href="#cb10-3005" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">15</span>) <span class="sc">+</span></span>
<span id="cb10-3006"><a href="#cb10-3006" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span></span>
<span id="cb10-3007"><a href="#cb10-3007" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb10-3008"><a href="#cb10-3008" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-3009"><a href="#cb10-3009" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3010"><a href="#cb10-3010" aria-hidden="true" tabindex="-1"></a>We see that samples are well embedded as well. </span>
<span id="cb10-3011"><a href="#cb10-3011" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3014"><a href="#cb10-3014" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-3015"><a href="#cb10-3015" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-pca-scanb-pc1-2</span></span>
<span id="cb10-3016"><a href="#cb10-3016" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: PCA embedding colored by cohort. The components used are the</span></span>
<span id="cb10-3017"><a href="#cb10-3017" aria-hidden="true" tabindex="-1"></a><span class="co">#|     first and second components.</span></span>
<span id="cb10-3018"><a href="#cb10-3018" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_pca_coordinates</span>(</span>
<span id="cb10-3019"><a href="#cb10-3019" aria-hidden="true" tabindex="-1"></a>    df_pca_coordinates_znorm,</span>
<span id="cb10-3020"><a href="#cb10-3020" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"cohort"</span>,</span>
<span id="cb10-3021"><a href="#cb10-3021" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC1"</span>,</span>
<span id="cb10-3022"><a href="#cb10-3022" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC2"</span>,</span>
<span id="cb10-3023"><a href="#cb10-3023" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">1</span>,</span>
<span id="cb10-3024"><a href="#cb10-3024" aria-hidden="true" tabindex="-1"></a>    <span class="at">base_size =</span> <span class="dv">10</span>,</span>
<span id="cb10-3025"><a href="#cb10-3025" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="fu">paste0</span>(</span>
<span id="cb10-3026"><a href="#cb10-3026" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Embedding of all samples from TCGA,</span><span class="sc">\n</span><span class="st">"</span>,</span>
<span id="cb10-3027"><a href="#cb10-3027" aria-hidden="true" tabindex="-1"></a>        <span class="st">"METABRIC and SCANB"</span></span>
<span id="cb10-3028"><a href="#cb10-3028" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb10-3029"><a href="#cb10-3029" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span> </span>
<span id="cb10-3030"><a href="#cb10-3030" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">15</span>) <span class="sc">+</span></span>
<span id="cb10-3031"><a href="#cb10-3031" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span></span>
<span id="cb10-3032"><a href="#cb10-3032" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb10-3033"><a href="#cb10-3033" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-3034"><a href="#cb10-3034" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3035"><a href="#cb10-3035" aria-hidden="true" tabindex="-1"></a>SCANB and POETIC are in between as expected and they are closer</span>
<span id="cb10-3036"><a href="#cb10-3036" aria-hidden="true" tabindex="-1"></a>to their sequencing technology. Interestingly the </span>
<span id="cb10-3037"><a href="#cb10-3037" aria-hidden="true" tabindex="-1"></a>variation for SCANB and TCGA are compressed in the PC1 vs PC2 axis compared</span>
<span id="cb10-3038"><a href="#cb10-3038" aria-hidden="true" tabindex="-1"></a>to METABRIC and POETIC.</span>
<span id="cb10-3039"><a href="#cb10-3039" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3040"><a href="#cb10-3040" aria-hidden="true" tabindex="-1"></a>@fig-pca-scanb-er-pam50-2 shows that the SCANB samples are also well mixed</span>
<span id="cb10-3041"><a href="#cb10-3041" aria-hidden="true" tabindex="-1"></a>regarding the clinical factors, including the $SET_{ER/PR}$ signature.</span>
<span id="cb10-3042"><a href="#cb10-3042" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3043"><a href="#cb10-3043" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=16, fig.height=14}</span></span>
<span id="cb10-3044"><a href="#cb10-3044" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-pca-scanb-er-pam50-2</span></span>
<span id="cb10-3045"><a href="#cb10-3045" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: PCA embedding of all samples from TCGA, SCANB and METABRIC.</span></span>
<span id="cb10-3046"><a href="#cb10-3046" aria-hidden="true" tabindex="-1"></a><span class="co">#|     (A) Colored by cohort,</span></span>
<span id="cb10-3047"><a href="#cb10-3047" aria-hidden="true" tabindex="-1"></a><span class="co">#|     (B) colored by ER status, (C) colored by PAM50 molecular subtype,</span></span>
<span id="cb10-3048"><a href="#cb10-3048" aria-hidden="true" tabindex="-1"></a><span class="co">#|     (D) colored by the $SET_{ER/PR}$ signature and only SCANB samples.</span></span>
<span id="cb10-3049"><a href="#cb10-3049" aria-hidden="true" tabindex="-1"></a>size <span class="ot">&lt;-</span> <span class="dv">2</span></span>
<span id="cb10-3050"><a href="#cb10-3050" aria-hidden="true" tabindex="-1"></a>base_size <span class="ot">&lt;-</span> <span class="dv">20</span></span>
<span id="cb10-3051"><a href="#cb10-3051" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3052"><a href="#cb10-3052" aria-hidden="true" tabindex="-1"></a>plots_with_scanb <span class="ot">&lt;-</span> <span class="fu">sapply</span>(</span>
<span id="cb10-3053"><a href="#cb10-3053" aria-hidden="true" tabindex="-1"></a>    <span class="fu">c</span>(<span class="st">"cohort"</span>, <span class="st">"er_status"</span>, <span class="st">"pam50"</span>, <span class="st">"SET_ERPR"</span>, <span class="st">"HALLMARK_G2M_CHECKPOINT"</span>), </span>
<span id="cb10-3054"><a href="#cb10-3054" aria-hidden="true" tabindex="-1"></a>    plot_pca_coordinates,</span>
<span id="cb10-3055"><a href="#cb10-3055" aria-hidden="true" tabindex="-1"></a>    <span class="at">df_pca =</span> df_pca_coordinates_znorm <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb10-3056"><a href="#cb10-3056" aria-hidden="true" tabindex="-1"></a>        pam50 <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"luma"</span>, <span class="st">"lumb"</span>, <span class="st">"basal"</span>, <span class="st">"her2"</span>, <span class="st">"normal"</span>) <span class="sc">&amp;</span></span>
<span id="cb10-3057"><a href="#cb10-3057" aria-hidden="true" tabindex="-1"></a>            cohort <span class="sc">!=</span> <span class="st">"poetic"</span></span>
<span id="cb10-3058"><a href="#cb10-3058" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb10-3059"><a href="#cb10-3059" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">slice</span>(<span class="fu">sample</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">n</span>())),</span>
<span id="cb10-3060"><a href="#cb10-3060" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"PC3"</span>, </span>
<span id="cb10-3061"><a href="#cb10-3061" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"PC4"</span>,</span>
<span id="cb10-3062"><a href="#cb10-3062" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> size,</span>
<span id="cb10-3063"><a href="#cb10-3063" aria-hidden="true" tabindex="-1"></a>    <span class="at">base_size =</span> base_size,</span>
<span id="cb10-3064"><a href="#cb10-3064" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="fu">paste0</span>(</span>
<span id="cb10-3065"><a href="#cb10-3065" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Embedding of all samples from TCGA,</span><span class="sc">\n</span><span class="st">"</span>,</span>
<span id="cb10-3066"><a href="#cb10-3066" aria-hidden="true" tabindex="-1"></a>        <span class="st">"METABRIC and SCANB"</span></span>
<span id="cb10-3067"><a href="#cb10-3067" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb10-3068"><a href="#cb10-3068" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>, </span>
<span id="cb10-3069"><a href="#cb10-3069" aria-hidden="true" tabindex="-1"></a>    <span class="at">simplify =</span> <span class="cn">FALSE</span></span>
<span id="cb10-3070"><a href="#cb10-3070" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-3071"><a href="#cb10-3071" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3072"><a href="#cb10-3072" aria-hidden="true" tabindex="-1"></a>plots_with_scanb<span class="sc">$</span>cohort <span class="ot">&lt;-</span> plots_with_scanb<span class="sc">$</span>cohort <span class="sc">+</span></span>
<span id="cb10-3073"><a href="#cb10-3073" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span></span>
<span id="cb10-3074"><a href="#cb10-3074" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb10-3075"><a href="#cb10-3075" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3076"><a href="#cb10-3076" aria-hidden="true" tabindex="-1"></a>plots_with_scanb<span class="sc">$</span>pam50 <span class="ot">&lt;-</span> plots_with_scanb<span class="sc">$</span>pam50 <span class="sc">+</span></span>
<span id="cb10-3077"><a href="#cb10-3077" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_color_manual</span>(</span>
<span id="cb10-3078"><a href="#cb10-3078" aria-hidden="true" tabindex="-1"></a>        <span class="at">values =</span> <span class="fu">get_colors_pam50</span>(plots_with_scanb<span class="sc">$</span>pam50<span class="sc">$</span>data)</span>
<span id="cb10-3079"><a href="#cb10-3079" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb10-3080"><a href="#cb10-3080" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span></span>
<span id="cb10-3081"><a href="#cb10-3081" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb10-3082"><a href="#cb10-3082" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3083"><a href="#cb10-3083" aria-hidden="true" tabindex="-1"></a>plots_with_scanb<span class="sc">$</span>er_status <span class="ot">&lt;-</span>  plots_with_scanb<span class="sc">$</span>er_status <span class="sc">+</span></span>
<span id="cb10-3084"><a href="#cb10-3084" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_color_viridis_d</span>() <span class="sc">+</span></span>
<span id="cb10-3085"><a href="#cb10-3085" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">color =</span> <span class="st">"ER status"</span>) <span class="sc">+</span></span>
<span id="cb10-3086"><a href="#cb10-3086" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span></span>
<span id="cb10-3087"><a href="#cb10-3087" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb10-3088"><a href="#cb10-3088" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3089"><a href="#cb10-3089" aria-hidden="true" tabindex="-1"></a>plots_with_scanb<span class="sc">$</span>SET_ERPR <span class="ot">&lt;-</span> df_pca_coordinates_znorm <span class="sc">%&gt;%</span> </span>
<span id="cb10-3090"><a href="#cb10-3090" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"scanb"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb10-3091"><a href="#cb10-3091" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes_string</span>(<span class="at">x =</span> <span class="st">"PC3"</span>, <span class="at">y =</span> <span class="st">"PC4"</span>, <span class="at">z =</span> <span class="st">"SET_ERPR"</span>)) <span class="sc">+</span> </span>
<span id="cb10-3092"><a href="#cb10-3092" aria-hidden="true" tabindex="-1"></a>        ggplot2<span class="sc">::</span><span class="fu">stat_summary_hex</span>(<span class="at">bins =</span> <span class="dv">25</span>) <span class="sc">+</span></span>
<span id="cb10-3093"><a href="#cb10-3093" aria-hidden="true" tabindex="-1"></a>        ggplot2<span class="sc">::</span><span class="fu">scale_fill_viridis_c</span>() <span class="sc">+</span></span>
<span id="cb10-3094"><a href="#cb10-3094" aria-hidden="true" tabindex="-1"></a>        ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb10-3095"><a href="#cb10-3095" aria-hidden="true" tabindex="-1"></a>            <span class="at">title =</span> <span class="st">"Embedding of SCANB only"</span>,</span>
<span id="cb10-3096"><a href="#cb10-3096" aria-hidden="true" tabindex="-1"></a>            <span class="at">fill =</span> <span class="fu">expression</span>(SET[ER<span class="sc">/</span>PR])</span>
<span id="cb10-3097"><a href="#cb10-3097" aria-hidden="true" tabindex="-1"></a>        ) <span class="sc">+</span> </span>
<span id="cb10-3098"><a href="#cb10-3098" aria-hidden="true" tabindex="-1"></a>        ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> base_size) <span class="sc">+</span></span>
<span id="cb10-3099"><a href="#cb10-3099" aria-hidden="true" tabindex="-1"></a>        <span class="fu">change_plot_aes_point</span>()</span>
<span id="cb10-3100"><a href="#cb10-3100" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3101"><a href="#cb10-3101" aria-hidden="true" tabindex="-1"></a>plots_with_scanb<span class="sc">$</span>HALLMARK_G2M_CHECKPOINT <span class="ot">&lt;-</span> df_pca_coordinates_znorm <span class="sc">%&gt;%</span> </span>
<span id="cb10-3102"><a href="#cb10-3102" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"scanb"</span>) <span class="sc">&amp;</span> pam50 <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"luma"</span>, <span class="st">"lumb"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb10-3103"><a href="#cb10-3103" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes_string</span>(<span class="at">x =</span> <span class="st">"PC3"</span>, <span class="at">y =</span> <span class="st">"PC4"</span>, <span class="at">z =</span> <span class="st">"HALLMARK_G2M_CHECKPOINT"</span>)) <span class="sc">+</span> </span>
<span id="cb10-3104"><a href="#cb10-3104" aria-hidden="true" tabindex="-1"></a>        ggplot2<span class="sc">::</span><span class="fu">stat_summary_hex</span>(<span class="at">bins =</span> <span class="dv">25</span>) <span class="sc">+</span></span>
<span id="cb10-3105"><a href="#cb10-3105" aria-hidden="true" tabindex="-1"></a>        ggplot2<span class="sc">::</span><span class="fu">scale_fill_viridis_c</span>() <span class="sc">+</span></span>
<span id="cb10-3106"><a href="#cb10-3106" aria-hidden="true" tabindex="-1"></a>        ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb10-3107"><a href="#cb10-3107" aria-hidden="true" tabindex="-1"></a>            <span class="at">title =</span> <span class="st">"Embedding of SCANB, luminal A and B only"</span>,</span>
<span id="cb10-3108"><a href="#cb10-3108" aria-hidden="true" tabindex="-1"></a>            <span class="at">fill =</span> <span class="st">"G2M"</span></span>
<span id="cb10-3109"><a href="#cb10-3109" aria-hidden="true" tabindex="-1"></a>        ) <span class="sc">+</span> </span>
<span id="cb10-3110"><a href="#cb10-3110" aria-hidden="true" tabindex="-1"></a>        ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> base_size) <span class="sc">+</span></span>
<span id="cb10-3111"><a href="#cb10-3111" aria-hidden="true" tabindex="-1"></a>        <span class="fu">change_plot_aes_point</span>()</span>
<span id="cb10-3112"><a href="#cb10-3112" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3113"><a href="#cb10-3113" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(</span>
<span id="cb10-3114"><a href="#cb10-3114" aria-hidden="true" tabindex="-1"></a>    <span class="at">plotlist =</span> plots_with_scanb,</span>
<span id="cb10-3115"><a href="#cb10-3115" aria-hidden="true" tabindex="-1"></a>    <span class="at">ncol =</span> <span class="dv">2</span></span>
<span id="cb10-3116"><a href="#cb10-3116" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-3117"><a href="#cb10-3117" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-3118"><a href="#cb10-3118" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3119"><a href="#cb10-3119" aria-hidden="true" tabindex="-1"></a>In the end normalizing by scaling the sample before doing any embedding</span>
<span id="cb10-3120"><a href="#cb10-3120" aria-hidden="true" tabindex="-1"></a>works as well. Though the embedding seems to be more compressed. </span>
<span id="cb10-3121"><a href="#cb10-3121" aria-hidden="true" tabindex="-1"></a>I would argue that the qPCR-like normalization gives a bit better</span>
<span id="cb10-3122"><a href="#cb10-3122" aria-hidden="true" tabindex="-1"></a>embedding. </span>
<span id="cb10-3123"><a href="#cb10-3123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3124"><a href="#cb10-3124" aria-hidden="true" tabindex="-1"></a><span class="fu">## ILC and position in the molecular landscape</span></span>
<span id="cb10-3125"><a href="#cb10-3125" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3126"><a href="#cb10-3126" aria-hidden="true" tabindex="-1"></a>Invasive lobular carcinoma is a subtype of ER+ BC. They tend to respond</span>
<span id="cb10-3127"><a href="#cb10-3127" aria-hidden="true" tabindex="-1"></a>better to ET in the short term but they recurr after 20 years. Here we</span>
<span id="cb10-3128"><a href="#cb10-3128" aria-hidden="true" tabindex="-1"></a>just show where the ILC BC lie in the molecular landscape.</span>
<span id="cb10-3129"><a href="#cb10-3129" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3130"><a href="#cb10-3130" aria-hidden="true" tabindex="-1"></a><span class="in">```{r fig.width=9, fig.height=6, message=FALSE}</span></span>
<span id="cb10-3131"><a href="#cb10-3131" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-pca-lobular-scanb</span></span>
<span id="cb10-3132"><a href="#cb10-3132" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Embedding of the lobular samples from SCANB on top </span></span>
<span id="cb10-3133"><a href="#cb10-3133" aria-hidden="true" tabindex="-1"></a><span class="co">#|     METABRIC and TCGA</span></span>
<span id="cb10-3134"><a href="#cb10-3134" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">get_base_plot</span>(df_pca <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"tcga"</span>, <span class="st">"metabric"</span>)))</span>
<span id="cb10-3135"><a href="#cb10-3135" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3136"><a href="#cb10-3136" aria-hidden="true" tabindex="-1"></a>df_pca_lob <span class="ot">&lt;-</span> df_pca <span class="sc">%&gt;%</span></span>
<span id="cb10-3137"><a href="#cb10-3137" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"scanb"</span> <span class="sc">&amp;</span> InvCa.type <span class="sc">==</span> <span class="st">"Lobular"</span>)</span>
<span id="cb10-3138"><a href="#cb10-3138" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3139"><a href="#cb10-3139" aria-hidden="true" tabindex="-1"></a>p <span class="sc">+</span></span>
<span id="cb10-3140"><a href="#cb10-3140" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>(</span>
<span id="cb10-3141"><a href="#cb10-3141" aria-hidden="true" tabindex="-1"></a>        df_pca_lob,</span>
<span id="cb10-3142"><a href="#cb10-3142" aria-hidden="true" tabindex="-1"></a>        <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> PC3, <span class="at">y =</span> PC4),</span>
<span id="cb10-3143"><a href="#cb10-3143" aria-hidden="true" tabindex="-1"></a>        <span class="at">size =</span> <span class="dv">2</span></span>
<span id="cb10-3144"><a href="#cb10-3144" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb10-3145"><a href="#cb10-3145" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">15</span>) <span class="sc">+</span></span>
<span id="cb10-3146"><a href="#cb10-3146" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb10-3147"><a href="#cb10-3147" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">"Lobular samples from SCANB</span><span class="sc">\n</span><span class="st">on top of METABRIC and TCGA samples"</span></span>
<span id="cb10-3148"><a href="#cb10-3148" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb10-3149"><a href="#cb10-3149" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_color_manual</span>(</span>
<span id="cb10-3150"><a href="#cb10-3150" aria-hidden="true" tabindex="-1"></a>        <span class="at">values =</span> <span class="fu">c</span>(<span class="fu">get_colors_pam50</span>(p<span class="sc">$</span>data))</span>
<span id="cb10-3151"><a href="#cb10-3151" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb10-3152"><a href="#cb10-3152" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span></span>
<span id="cb10-3153"><a href="#cb10-3153" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>(<span class="at">shape =</span> <span class="cn">FALSE</span>)</span>
<span id="cb10-3154"><a href="#cb10-3154" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-3155"><a href="#cb10-3155" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3156"><a href="#cb10-3156" aria-hidden="true" tabindex="-1"></a>We see that they are mostly in the normal and luminal A region. Still</span>
<span id="cb10-3157"><a href="#cb10-3157" aria-hidden="true" tabindex="-1"></a>there are several samples that are luminal B, meaning they are more</span>
<span id="cb10-3158"><a href="#cb10-3158" aria-hidden="true" tabindex="-1"></a>proliferative.</span>
<span id="cb10-3159"><a href="#cb10-3159" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3160"><a href="#cb10-3160" aria-hidden="true" tabindex="-1"></a><span class="fu">## genefu </span></span>
<span id="cb10-3161"><a href="#cb10-3161" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3162"><a href="#cb10-3162" aria-hidden="true" tabindex="-1"></a>With genefu it is possible to calculate scores from comercially available </span>
<span id="cb10-3163"><a href="#cb10-3163" aria-hidden="true" tabindex="-1"></a>signatures. Here we try to use it on SCANB, METABRIC and TCGA. Moreover,</span>
<span id="cb10-3164"><a href="#cb10-3164" aria-hidden="true" tabindex="-1"></a>we can start with ABiM, as this dataset has the gold standard ROR score </span>
<span id="cb10-3165"><a href="#cb10-3165" aria-hidden="true" tabindex="-1"></a>obtained from nCounter data. This is a good way of validating the</span>
<span id="cb10-3166"><a href="#cb10-3166" aria-hidden="true" tabindex="-1"></a>pipeline. </span>
<span id="cb10-3167"><a href="#cb10-3167" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3170"><a href="#cb10-3170" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-3171"><a href="#cb10-3171" aria-hidden="true" tabindex="-1"></a>abim_100 <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../results/rds_files/validation/abim_100.rds"</span>)</span>
<span id="cb10-3172"><a href="#cb10-3172" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3173"><a href="#cb10-3173" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(pam50.robust)</span>
<span id="cb10-3174"><a href="#cb10-3174" aria-hidden="true" tabindex="-1"></a>abim_genefu_pam50 <span class="ot">&lt;-</span> genefu<span class="sc">::</span><span class="fu">molecular.subtyping</span>(</span>
<span id="cb10-3175"><a href="#cb10-3175" aria-hidden="true" tabindex="-1"></a>    <span class="at">sbt.model =</span> <span class="st">"pam50"</span>,</span>
<span id="cb10-3176"><a href="#cb10-3176" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> <span class="fu">t</span>(<span class="fu">assay</span>(abim_100, <span class="st">"log2fpkm"</span>)), </span>
<span id="cb10-3177"><a href="#cb10-3177" aria-hidden="true" tabindex="-1"></a>    <span class="at">annot =</span> annotables<span class="sc">::</span>grch38 <span class="sc">%&gt;%</span></span>
<span id="cb10-3178"><a href="#cb10-3178" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb10-3179"><a href="#cb10-3179" aria-hidden="true" tabindex="-1"></a>            <span class="at">EntrezGene.ID =</span> entrez,</span>
<span id="cb10-3180"><a href="#cb10-3180" aria-hidden="true" tabindex="-1"></a>            <span class="at">Gene.Symbol =</span> symbol</span>
<span id="cb10-3181"><a href="#cb10-3181" aria-hidden="true" tabindex="-1"></a>        ),</span>
<span id="cb10-3182"><a href="#cb10-3182" aria-hidden="true" tabindex="-1"></a>    <span class="at">do.mapping =</span> <span class="cn">FALSE</span></span>
<span id="cb10-3183"><a href="#cb10-3183" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-3184"><a href="#cb10-3184" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3185"><a href="#cb10-3185" aria-hidden="true" tabindex="-1"></a>abim_100<span class="sc">$</span>GENEFU.PAM50 <span class="ot">&lt;-</span> abim_genefu_pam50<span class="sc">$</span>subtype</span>
<span id="cb10-3186"><a href="#cb10-3186" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-3187"><a href="#cb10-3187" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3188"><a href="#cb10-3188" aria-hidden="true" tabindex="-1"></a>We start by comparing the molecular subtypes:</span>
<span id="cb10-3189"><a href="#cb10-3189" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3192"><a href="#cb10-3192" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-3193"><a href="#cb10-3193" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(abim_100<span class="sc">$</span>GENEFU.PAM50 <span class="sc">==</span> abim_100<span class="sc">$</span>ProsignaFT.Subtype)</span>
<span id="cb10-3194"><a href="#cb10-3194" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-3195"><a href="#cb10-3195" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3196"><a href="#cb10-3196" aria-hidden="true" tabindex="-1"></a>86 out of 100 were correctly called. Here we are using the gold standard wich</span>
<span id="cb10-3197"><a href="#cb10-3197" aria-hidden="true" tabindex="-1"></a>is the molecular subtype called from FT samples. The normal like subtype is not</span>
<span id="cb10-3198"><a href="#cb10-3198" aria-hidden="true" tabindex="-1"></a>available from the FT prosigna assay. So we now compare the results with the</span>
<span id="cb10-3199"><a href="#cb10-3199" aria-hidden="true" tabindex="-1"></a>SSP obtained from the SCANB team. </span>
<span id="cb10-3200"><a href="#cb10-3200" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3201"><a href="#cb10-3201" aria-hidden="true" tabindex="-1"></a>Below is the confusion matrix to see which subtypes were misclassified. </span>
<span id="cb10-3202"><a href="#cb10-3202" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3205"><a href="#cb10-3205" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-3206"><a href="#cb10-3206" aria-hidden="true" tabindex="-1"></a>caret<span class="sc">::</span><span class="fu">confusionMatrix</span>(</span>
<span id="cb10-3207"><a href="#cb10-3207" aria-hidden="true" tabindex="-1"></a>    abim_100<span class="sc">$</span>GENEFU.PAM50,</span>
<span id="cb10-3208"><a href="#cb10-3208" aria-hidden="true" tabindex="-1"></a>    <span class="fu">factor</span>(</span>
<span id="cb10-3209"><a href="#cb10-3209" aria-hidden="true" tabindex="-1"></a>        abim_100<span class="sc">$</span>ProsignaFT.Subtype, </span>
<span id="cb10-3210"><a href="#cb10-3210" aria-hidden="true" tabindex="-1"></a>        <span class="at">levels =</span> <span class="fu">levels</span>(abim_100<span class="sc">$</span>GENEFU.PAM50)</span>
<span id="cb10-3211"><a href="#cb10-3211" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb10-3212"><a href="#cb10-3212" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-3213"><a href="#cb10-3213" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-3214"><a href="#cb10-3214" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3215"><a href="#cb10-3215" aria-hidden="true" tabindex="-1"></a>All luminal A were predicted correctly. Some normal like were predicted to</span>
<span id="cb10-3216"><a href="#cb10-3216" aria-hidden="true" tabindex="-1"></a>be luminal A or basal and Her2. There were some mismatches </span>
<span id="cb10-3217"><a href="#cb10-3217" aria-hidden="true" tabindex="-1"></a>for the LumB, that were misclassified as LumA. Still it is ok. </span>
<span id="cb10-3218"><a href="#cb10-3218" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3219"><a href="#cb10-3219" aria-hidden="true" tabindex="-1"></a>Let us now calculate the ROR for the ABiM cohort.</span>
<span id="cb10-3220"><a href="#cb10-3220" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3223"><a href="#cb10-3223" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-3224"><a href="#cb10-3224" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-ror-genefu-abim</span></span>
<span id="cb10-3225"><a href="#cb10-3225" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Correlation between ROR obtained from prosigna assay and </span></span>
<span id="cb10-3226"><a href="#cb10-3226" aria-hidden="true" tabindex="-1"></a><span class="co">#|     R package genefu</span></span>
<span id="cb10-3227"><a href="#cb10-3227" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(pam50)</span>
<span id="cb10-3228"><a href="#cb10-3228" aria-hidden="true" tabindex="-1"></a>ror_abim <span class="ot">&lt;-</span> genefu<span class="sc">::</span><span class="fu">rorS</span>(</span>
<span id="cb10-3229"><a href="#cb10-3229" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> <span class="fu">t</span>(<span class="fu">assay</span>(abim_100, <span class="st">"log2fpkm"</span>)), </span>
<span id="cb10-3230"><a href="#cb10-3230" aria-hidden="true" tabindex="-1"></a>    <span class="at">annot =</span> annotables<span class="sc">::</span>grch38 <span class="sc">%&gt;%</span></span>
<span id="cb10-3231"><a href="#cb10-3231" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb10-3232"><a href="#cb10-3232" aria-hidden="true" tabindex="-1"></a>            <span class="at">EntrezGene.ID =</span> entrez,</span>
<span id="cb10-3233"><a href="#cb10-3233" aria-hidden="true" tabindex="-1"></a>            <span class="at">Gene.Symbol =</span> symbol</span>
<span id="cb10-3234"><a href="#cb10-3234" aria-hidden="true" tabindex="-1"></a>        ),</span>
<span id="cb10-3235"><a href="#cb10-3235" aria-hidden="true" tabindex="-1"></a>    <span class="at">do.mapping =</span> <span class="cn">FALSE</span></span>
<span id="cb10-3236"><a href="#cb10-3236" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-3237"><a href="#cb10-3237" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3238"><a href="#cb10-3238" aria-hidden="true" tabindex="-1"></a>abim_100<span class="sc">$</span>GENEFU.ROR <span class="ot">&lt;-</span> ror_abim<span class="sc">$</span>score</span>
<span id="cb10-3239"><a href="#cb10-3239" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3240"><a href="#cb10-3240" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(abim_100) <span class="sc">%&gt;%</span></span>
<span id="cb10-3241"><a href="#cb10-3241" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span></span>
<span id="cb10-3242"><a href="#cb10-3242" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> GENEFU.ROR, <span class="at">y =</span> ProsignaFT.ROR, <span class="at">color =</span> ProsignaFT.Subtype)) <span class="sc">+</span></span>
<span id="cb10-3243"><a href="#cb10-3243" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span> </span>
<span id="cb10-3244"><a href="#cb10-3244" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_smooth</span>(</span>
<span id="cb10-3245"><a href="#cb10-3245" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">"black"</span>,</span>
<span id="cb10-3246"><a href="#cb10-3246" aria-hidden="true" tabindex="-1"></a>        <span class="at">method =</span> <span class="st">"loess"</span>,</span>
<span id="cb10-3247"><a href="#cb10-3247" aria-hidden="true" tabindex="-1"></a>        <span class="at">formula =</span> <span class="st">"y ~ x"</span></span>
<span id="cb10-3248"><a href="#cb10-3248" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb10-3249"><a href="#cb10-3249" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"ABiM 100 cohort"</span>) <span class="sc">+</span></span>
<span id="cb10-3250"><a href="#cb10-3250" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>) <span class="sc">+</span> </span>
<span id="cb10-3251"><a href="#cb10-3251" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span></span>
<span id="cb10-3252"><a href="#cb10-3252" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb10-3253"><a href="#cb10-3253" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-3254"><a href="#cb10-3254" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3255"><a href="#cb10-3255" aria-hidden="true" tabindex="-1"></a>The correlation is relatively good and respects the molecular subtypes</span>
<span id="cb10-3256"><a href="#cb10-3256" aria-hidden="true" tabindex="-1"></a>in general. The ROR obtained from genefu for some luminal B are not </span>
<span id="cb10-3257"><a href="#cb10-3257" aria-hidden="true" tabindex="-1"></a>quite right as they are below 50. </span>
<span id="cb10-3258"><a href="#cb10-3258" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3259"><a href="#cb10-3259" aria-hidden="true" tabindex="-1"></a>We now move on to the other signatures: Mammaprint, EndoPredict (EP) and </span>
<span id="cb10-3260"><a href="#cb10-3260" aria-hidden="true" tabindex="-1"></a>OncotypeDX Risk Score (RS).</span>
<span id="cb10-3261"><a href="#cb10-3261" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3264"><a href="#cb10-3264" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-3265"><a href="#cb10-3265" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-ror-mammaprint-genefu-abim</span></span>
<span id="cb10-3266"><a href="#cb10-3266" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Correlation between ROR obtained from prosigna assay and </span></span>
<span id="cb10-3267"><a href="#cb10-3267" aria-hidden="true" tabindex="-1"></a><span class="co">#|     the mammaprint signature obtained from the R package genefu</span></span>
<span id="cb10-3268"><a href="#cb10-3268" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3269"><a href="#cb10-3269" aria-hidden="true" tabindex="-1"></a>annot_abim <span class="ot">&lt;-</span> annotables<span class="sc">::</span>grch38 <span class="sc">%&gt;%</span></span>
<span id="cb10-3270"><a href="#cb10-3270" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb10-3271"><a href="#cb10-3271" aria-hidden="true" tabindex="-1"></a>            <span class="at">EntrezGene.ID =</span> entrez,</span>
<span id="cb10-3272"><a href="#cb10-3272" aria-hidden="true" tabindex="-1"></a>            <span class="at">Gene.Symbol =</span> symbol</span>
<span id="cb10-3273"><a href="#cb10-3273" aria-hidden="true" tabindex="-1"></a>        ) <span class="sc">%&gt;%</span></span>
<span id="cb10-3274"><a href="#cb10-3274" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">duplicated</span>(symbol)) <span class="sc">%&gt;%</span> </span>
<span id="cb10-3275"><a href="#cb10-3275" aria-hidden="true" tabindex="-1"></a>        data.frame <span class="sc">%&gt;%</span></span>
<span id="cb10-3276"><a href="#cb10-3276" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">rownames&lt;-</span><span class="st">`</span>(.<span class="sc">$</span>symbol)</span>
<span id="cb10-3277"><a href="#cb10-3277" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3278"><a href="#cb10-3278" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(sig.gene70)</span>
<span id="cb10-3279"><a href="#cb10-3279" aria-hidden="true" tabindex="-1"></a>sig.gene70 <span class="ot">&lt;-</span> sig.gene70 <span class="sc">%&gt;%</span></span>
<span id="cb10-3280"><a href="#cb10-3280" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(HUGO.gene.symbol)) <span class="sc">%&gt;%</span></span>
<span id="cb10-3281"><a href="#cb10-3281" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">distinct</span>(HUGO.gene.symbol, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-3282"><a href="#cb10-3282" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">rownames&lt;-</span><span class="st">`</span>(.<span class="sc">$</span>HUGO.gene.symbol)</span>
<span id="cb10-3283"><a href="#cb10-3283" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3284"><a href="#cb10-3284" aria-hidden="true" tabindex="-1"></a>erp_abim <span class="ot">&lt;-</span> <span class="fu">colData</span>(abim_100) <span class="sc">%&gt;%</span></span>
<span id="cb10-3285"><a href="#cb10-3285" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span></span>
<span id="cb10-3286"><a href="#cb10-3286" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(ER <span class="sc">==</span> <span class="st">"Positive"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-3287"><a href="#cb10-3287" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">pull</span>(sample_name)</span>
<span id="cb10-3288"><a href="#cb10-3288" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3289"><a href="#cb10-3289" aria-hidden="true" tabindex="-1"></a>abim_100 <span class="ot">&lt;-</span> abim_100[, erp_abim]</span>
<span id="cb10-3290"><a href="#cb10-3290" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3291"><a href="#cb10-3291" aria-hidden="true" tabindex="-1"></a>mammaprint_abim <span class="ot">&lt;-</span> genefu<span class="sc">::</span><span class="fu">gene70</span>(</span>
<span id="cb10-3292"><a href="#cb10-3292" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> <span class="fu">t</span>(<span class="fu">assay</span>(abim_100, <span class="st">"log2fpkm"</span>)), </span>
<span id="cb10-3293"><a href="#cb10-3293" aria-hidden="true" tabindex="-1"></a>    <span class="at">annot =</span> annot_abim,</span>
<span id="cb10-3294"><a href="#cb10-3294" aria-hidden="true" tabindex="-1"></a>    <span class="at">do.mapping =</span> <span class="cn">FALSE</span>,</span>
<span id="cb10-3295"><a href="#cb10-3295" aria-hidden="true" tabindex="-1"></a>    <span class="at">std =</span> <span class="st">"robust"</span>,</span>
<span id="cb10-3296"><a href="#cb10-3296" aria-hidden="true" tabindex="-1"></a>    <span class="at">verbose =</span> <span class="cn">TRUE</span></span>
<span id="cb10-3297"><a href="#cb10-3297" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-3298"><a href="#cb10-3298" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3299"><a href="#cb10-3299" aria-hidden="true" tabindex="-1"></a>abim_100<span class="sc">$</span>GENEFU.GENE70 <span class="ot">&lt;-</span> mammaprint_abim<span class="sc">$</span>score</span>
<span id="cb10-3300"><a href="#cb10-3300" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3301"><a href="#cb10-3301" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(abim_100) <span class="sc">%&gt;%</span></span>
<span id="cb10-3302"><a href="#cb10-3302" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span></span>
<span id="cb10-3303"><a href="#cb10-3303" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(</span>
<span id="cb10-3304"><a href="#cb10-3304" aria-hidden="true" tabindex="-1"></a>        <span class="fu">aes</span>(</span>
<span id="cb10-3305"><a href="#cb10-3305" aria-hidden="true" tabindex="-1"></a>            <span class="at">x =</span> GENEFU.GENE70, </span>
<span id="cb10-3306"><a href="#cb10-3306" aria-hidden="true" tabindex="-1"></a>            <span class="at">y =</span> ProsignaFT.ROR, </span>
<span id="cb10-3307"><a href="#cb10-3307" aria-hidden="true" tabindex="-1"></a>            <span class="at">color =</span> ProsignaFT.Subtype</span>
<span id="cb10-3308"><a href="#cb10-3308" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb10-3309"><a href="#cb10-3309" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb10-3310"><a href="#cb10-3310" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span> </span>
<span id="cb10-3311"><a href="#cb10-3311" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_smooth</span>(</span>
<span id="cb10-3312"><a href="#cb10-3312" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">"black"</span>,</span>
<span id="cb10-3313"><a href="#cb10-3313" aria-hidden="true" tabindex="-1"></a>        <span class="at">method =</span> <span class="st">"loess"</span>,</span>
<span id="cb10-3314"><a href="#cb10-3314" aria-hidden="true" tabindex="-1"></a>        <span class="at">formula =</span> <span class="st">"y ~ x"</span></span>
<span id="cb10-3315"><a href="#cb10-3315" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb10-3316"><a href="#cb10-3316" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"ABiM 100 cohort"</span>) <span class="sc">+</span></span>
<span id="cb10-3317"><a href="#cb10-3317" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>) <span class="sc">+</span> </span>
<span id="cb10-3318"><a href="#cb10-3318" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span></span>
<span id="cb10-3319"><a href="#cb10-3319" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb10-3320"><a href="#cb10-3320" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-3321"><a href="#cb10-3321" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3322"><a href="#cb10-3322" aria-hidden="true" tabindex="-1"></a>@fig-ror-mammaprint-genefu-abim shows the correlation between </span>
<span id="cb10-3323"><a href="#cb10-3323" aria-hidden="true" tabindex="-1"></a>the mammaprint signature (named GENE70) and the ROR from prosigna. </span>
<span id="cb10-3324"><a href="#cb10-3324" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3325"><a href="#cb10-3325" aria-hidden="true" tabindex="-1"></a>Let us now compare the GENE70 signature with the risk score developed in </span>
<span id="cb10-3326"><a href="#cb10-3326" aria-hidden="true" tabindex="-1"></a>the previous chapters. We use the ABiM 100 cohort to make the comparison. </span>
<span id="cb10-3327"><a href="#cb10-3327" aria-hidden="true" tabindex="-1"></a>One side note, we are using only 51 out of the 70 probes available in the</span>
<span id="cb10-3328"><a href="#cb10-3328" aria-hidden="true" tabindex="-1"></a>package. 15 probes don't have any corresponding symbol. Out of the available</span>
<span id="cb10-3329"><a href="#cb10-3329" aria-hidden="true" tabindex="-1"></a>genes, some are duplicated. For the duplicated genes, the correlation </span>
<span id="cb10-3330"><a href="#cb10-3330" aria-hidden="true" tabindex="-1"></a>coefficients are all similar, so we can select just the first ocurrence </span>
<span id="cb10-3331"><a href="#cb10-3331" aria-hidden="true" tabindex="-1"></a>in the list.</span>
<span id="cb10-3332"><a href="#cb10-3332" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3335"><a href="#cb10-3335" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-3336"><a href="#cb10-3336" aria-hidden="true" tabindex="-1"></a>abim_100_df_pca <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(</span>
<span id="cb10-3337"><a href="#cb10-3337" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/validation/abim_100_df_pca.rds"</span></span>
<span id="cb10-3338"><a href="#cb10-3338" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb10-3339"><a href="#cb10-3339" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">rownames&lt;-</span><span class="st">`</span>(.<span class="sc">$</span>sample_name)</span>
<span id="cb10-3340"><a href="#cb10-3340" aria-hidden="true" tabindex="-1"></a>coefs_risk_model <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(</span>
<span id="cb10-3341"><a href="#cb10-3341" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/risk_score/coefs_risk_model.rds"</span></span>
<span id="cb10-3342"><a href="#cb10-3342" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-3343"><a href="#cb10-3343" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3344"><a href="#cb10-3344" aria-hidden="true" tabindex="-1"></a>abim_100<span class="sc">$</span>risk_score <span class="ot">&lt;-</span> <span class="fu">calculate_risk_score</span>(</span>
<span id="cb10-3345"><a href="#cb10-3345" aria-hidden="true" tabindex="-1"></a>    coefs_risk_model,</span>
<span id="cb10-3346"><a href="#cb10-3346" aria-hidden="true" tabindex="-1"></a>    abim_100_df_pca[<span class="fu">colnames</span>(abim_100), ],</span>
<span id="cb10-3347"><a href="#cb10-3347" aria-hidden="true" tabindex="-1"></a>    <span class="at">is_node =</span> <span class="cn">TRUE</span></span>
<span id="cb10-3348"><a href="#cb10-3348" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-3349"><a href="#cb10-3349" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3350"><a href="#cb10-3350" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(abim_100) <span class="sc">%&gt;%</span></span>
<span id="cb10-3351"><a href="#cb10-3351" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span></span>
<span id="cb10-3352"><a href="#cb10-3352" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(</span>
<span id="cb10-3353"><a href="#cb10-3353" aria-hidden="true" tabindex="-1"></a>        <span class="fu">aes</span>(</span>
<span id="cb10-3354"><a href="#cb10-3354" aria-hidden="true" tabindex="-1"></a>            <span class="at">x =</span> GENEFU.GENE70, </span>
<span id="cb10-3355"><a href="#cb10-3355" aria-hidden="true" tabindex="-1"></a>            <span class="at">y =</span> risk_score, </span>
<span id="cb10-3356"><a href="#cb10-3356" aria-hidden="true" tabindex="-1"></a>            <span class="at">color =</span> ProsignaFT.Subtype</span>
<span id="cb10-3357"><a href="#cb10-3357" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb10-3358"><a href="#cb10-3358" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb10-3359"><a href="#cb10-3359" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span> </span>
<span id="cb10-3360"><a href="#cb10-3360" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_smooth</span>(</span>
<span id="cb10-3361"><a href="#cb10-3361" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">"black"</span>,</span>
<span id="cb10-3362"><a href="#cb10-3362" aria-hidden="true" tabindex="-1"></a>        <span class="at">method =</span> <span class="st">"loess"</span>,</span>
<span id="cb10-3363"><a href="#cb10-3363" aria-hidden="true" tabindex="-1"></a>        <span class="at">formula =</span> <span class="st">"y ~ x"</span></span>
<span id="cb10-3364"><a href="#cb10-3364" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb10-3365"><a href="#cb10-3365" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>) <span class="sc">+</span> </span>
<span id="cb10-3366"><a href="#cb10-3366" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb10-3367"><a href="#cb10-3367" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">"ABiM 100 cohort, ER+ BC only"</span>,</span>
<span id="cb10-3368"><a href="#cb10-3368" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="st">"Risk score"</span></span>
<span id="cb10-3369"><a href="#cb10-3369" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb10-3370"><a href="#cb10-3370" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span></span>
<span id="cb10-3371"><a href="#cb10-3371" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb10-3372"><a href="#cb10-3372" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-3373"><a href="#cb10-3373" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3374"><a href="#cb10-3374" aria-hidden="true" tabindex="-1"></a>There is a good correlation between the two signatures, suggesting </span>
<span id="cb10-3375"><a href="#cb10-3375" aria-hidden="true" tabindex="-1"></a>that the Mammaprint signature is somehow based on the position of the samples in</span>
<span id="cb10-3376"><a href="#cb10-3376" aria-hidden="true" tabindex="-1"></a>the molecular landscape. </span>
<span id="cb10-3377"><a href="#cb10-3377" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3378"><a href="#cb10-3378" aria-hidden="true" tabindex="-1"></a>We now proceed and calculate the mammaprint scores for SCANB and </span>
<span id="cb10-3379"><a href="#cb10-3379" aria-hidden="true" tabindex="-1"></a>METABRIC as well and only for ER+ BC samples.</span>
<span id="cb10-3380"><a href="#cb10-3380" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3383"><a href="#cb10-3383" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-3384"><a href="#cb10-3384" aria-hidden="true" tabindex="-1"></a>mammaprint_datasets <span class="ot">&lt;-</span> <span class="fu">mapply</span>(</span>
<span id="cb10-3385"><a href="#cb10-3385" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(x, y){</span>
<span id="cb10-3386"><a href="#cb10-3386" aria-hidden="true" tabindex="-1"></a>        genefu<span class="sc">::</span><span class="fu">gene70</span>(</span>
<span id="cb10-3387"><a href="#cb10-3387" aria-hidden="true" tabindex="-1"></a>            <span class="at">data =</span> <span class="fu">t</span>(<span class="fu">assay</span>(x, y)),</span>
<span id="cb10-3388"><a href="#cb10-3388" aria-hidden="true" tabindex="-1"></a>            <span class="at">annot =</span> annot_abim,</span>
<span id="cb10-3389"><a href="#cb10-3389" aria-hidden="true" tabindex="-1"></a>            <span class="at">do.mapping =</span> <span class="cn">FALSE</span>,</span>
<span id="cb10-3390"><a href="#cb10-3390" aria-hidden="true" tabindex="-1"></a>            <span class="at">std =</span> <span class="st">"robust"</span>,</span>
<span id="cb10-3391"><a href="#cb10-3391" aria-hidden="true" tabindex="-1"></a>            <span class="at">verbose =</span> <span class="cn">TRUE</span></span>
<span id="cb10-3392"><a href="#cb10-3392" aria-hidden="true" tabindex="-1"></a>        )<span class="sc">$</span>score <span class="sc">%&gt;%</span></span>
<span id="cb10-3393"><a href="#cb10-3393" aria-hidden="true" tabindex="-1"></a>            data.frame <span class="sc">%&gt;%</span></span>
<span id="cb10-3394"><a href="#cb10-3394" aria-hidden="true" tabindex="-1"></a>            <span class="st">`</span><span class="at">colnames&lt;-</span><span class="st">`</span>(<span class="fu">c</span>(<span class="st">"mammaprint"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb10-3395"><a href="#cb10-3395" aria-hidden="true" tabindex="-1"></a>            tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"sample_name"</span>) </span>
<span id="cb10-3396"><a href="#cb10-3396" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb10-3397"><a href="#cb10-3397" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> datasets,</span>
<span id="cb10-3398"><a href="#cb10-3398" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="fu">c</span>(which_exp, <span class="st">"poetic"</span> <span class="ot">=</span> <span class="st">"normalized_intensity"</span>),</span>
<span id="cb10-3399"><a href="#cb10-3399" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>,</span>
<span id="cb10-3400"><a href="#cb10-3400" aria-hidden="true" tabindex="-1"></a>    <span class="at">SIMPLIFY =</span> <span class="cn">FALSE</span></span>
<span id="cb10-3401"><a href="#cb10-3401" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(<span class="at">.id =</span> <span class="st">"cohort"</span>)</span>
<span id="cb10-3402"><a href="#cb10-3402" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3403"><a href="#cb10-3403" aria-hidden="true" tabindex="-1"></a>scanb_metabric_erp <span class="ot">&lt;-</span> df_pca <span class="sc">%&gt;%</span></span>
<span id="cb10-3404"><a href="#cb10-3404" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb10-3405"><a href="#cb10-3405" aria-hidden="true" tabindex="-1"></a>        cohort <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"scanb"</span>, <span class="st">"metabric"</span>) <span class="sc">&amp;</span></span>
<span id="cb10-3406"><a href="#cb10-3406" aria-hidden="true" tabindex="-1"></a>            er_status <span class="sc">==</span> <span class="st">"pos"</span></span>
<span id="cb10-3407"><a href="#cb10-3407" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span> </span>
<span id="cb10-3408"><a href="#cb10-3408" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">inner_join</span>(</span>
<span id="cb10-3409"><a href="#cb10-3409" aria-hidden="true" tabindex="-1"></a>        mammaprint_datasets,</span>
<span id="cb10-3410"><a href="#cb10-3410" aria-hidden="true" tabindex="-1"></a>        <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"sample_name"</span>, <span class="st">"cohort"</span>)</span>
<span id="cb10-3411"><a href="#cb10-3411" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb10-3412"><a href="#cb10-3412" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-3413"><a href="#cb10-3413" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3414"><a href="#cb10-3414" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=10, fig.height=5}</span></span>
<span id="cb10-3415"><a href="#cb10-3415" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-met-scanb-mammaprint</span></span>
<span id="cb10-3416"><a href="#cb10-3416" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Embedding of ER+ BC samples colored by the mammaprint </span></span>
<span id="cb10-3417"><a href="#cb10-3417" aria-hidden="true" tabindex="-1"></a><span class="co">#|     score.</span></span>
<span id="cb10-3418"><a href="#cb10-3418" aria-hidden="true" tabindex="-1"></a>scanb_metabric_erp <span class="sc">%&gt;%</span></span>
<span id="cb10-3419"><a href="#cb10-3419" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> PC3, <span class="at">y =</span> PC4, <span class="at">z =</span> mammaprint)) <span class="sc">+</span> </span>
<span id="cb10-3420"><a href="#cb10-3420" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">stat_summary_hex</span>(<span class="at">bins =</span> <span class="dv">20</span>) <span class="sc">+</span></span>
<span id="cb10-3421"><a href="#cb10-3421" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span> cohort) <span class="sc">+</span></span>
<span id="cb10-3422"><a href="#cb10-3422" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_fill_viridis_c</span>() <span class="sc">+</span></span>
<span id="cb10-3423"><a href="#cb10-3423" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">fill =</span> <span class="st">"Mammaprint"</span>) <span class="sc">+</span></span>
<span id="cb10-3424"><a href="#cb10-3424" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>)</span>
<span id="cb10-3425"><a href="#cb10-3425" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-3426"><a href="#cb10-3426" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3427"><a href="#cb10-3427" aria-hidden="true" tabindex="-1"></a>We see that the pattern is very close to what we have from the risk score </span>
<span id="cb10-3428"><a href="#cb10-3428" aria-hidden="true" tabindex="-1"></a>created in the previous chapter. The figure below shows the correlation</span>
<span id="cb10-3429"><a href="#cb10-3429" aria-hidden="true" tabindex="-1"></a>between mammaprint and the risk score once again.</span>
<span id="cb10-3430"><a href="#cb10-3430" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3433"><a href="#cb10-3433" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-3434"><a href="#cb10-3434" aria-hidden="true" tabindex="-1"></a>endo_only_scanb <span class="ot">&lt;-</span> df_pca <span class="sc">%&gt;%</span> </span>
<span id="cb10-3435"><a href="#cb10-3435" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"scanb"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-3436"><a href="#cb10-3436" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">node_status =</span> <span class="fu">ifelse</span>(</span>
<span id="cb10-3437"><a href="#cb10-3437" aria-hidden="true" tabindex="-1"></a>        node_group <span class="sc">==</span> <span class="st">"N0"</span>, </span>
<span id="cb10-3438"><a href="#cb10-3438" aria-hidden="true" tabindex="-1"></a>        <span class="st">"neg"</span>,</span>
<span id="cb10-3439"><a href="#cb10-3439" aria-hidden="true" tabindex="-1"></a>        <span class="st">"pos"</span></span>
<span id="cb10-3440"><a href="#cb10-3440" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">%&gt;%</span> </span>
<span id="cb10-3441"><a href="#cb10-3441" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">node_status =</span> <span class="fu">factor</span>(</span>
<span id="cb10-3442"><a href="#cb10-3442" aria-hidden="true" tabindex="-1"></a>        node_status, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"neg"</span>, <span class="st">"pos"</span>)</span>
<span id="cb10-3443"><a href="#cb10-3443" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">%&gt;%</span></span>
<span id="cb10-3444"><a href="#cb10-3444" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb10-3445"><a href="#cb10-3445" aria-hidden="true" tabindex="-1"></a>            TreatGroup <span class="sc">==</span> <span class="st">"Endo"</span> <span class="sc">&amp;</span> </span>
<span id="cb10-3446"><a href="#cb10-3446" aria-hidden="true" tabindex="-1"></a>            <span class="sc">!</span><span class="fu">is.na</span>(rfs_status) <span class="sc">&amp;</span> </span>
<span id="cb10-3447"><a href="#cb10-3447" aria-hidden="true" tabindex="-1"></a>            <span class="sc">!</span><span class="fu">is.na</span>(rfs_months) <span class="sc">&amp;</span></span>
<span id="cb10-3448"><a href="#cb10-3448" aria-hidden="true" tabindex="-1"></a>            er_status <span class="sc">==</span> <span class="st">"pos"</span> <span class="sc">&amp;</span></span>
<span id="cb10-3449"><a href="#cb10-3449" aria-hidden="true" tabindex="-1"></a>            <span class="sc">!</span><span class="fu">is.na</span>(age) <span class="sc">&amp;</span></span>
<span id="cb10-3450"><a href="#cb10-3450" aria-hidden="true" tabindex="-1"></a>            <span class="sc">!</span><span class="fu">is.na</span>(tumor_size) <span class="sc">&amp;</span></span>
<span id="cb10-3451"><a href="#cb10-3451" aria-hidden="true" tabindex="-1"></a>            <span class="sc">!</span><span class="fu">is.na</span>(node_status)</span>
<span id="cb10-3452"><a href="#cb10-3452" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb10-3453"><a href="#cb10-3453" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3454"><a href="#cb10-3454" aria-hidden="true" tabindex="-1"></a>endo_only_metabric <span class="ot">&lt;-</span> df_pca <span class="sc">%&gt;%</span> </span>
<span id="cb10-3455"><a href="#cb10-3455" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"metabric"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-3456"><a href="#cb10-3456" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">TreatGroup =</span> dplyr<span class="sc">::</span><span class="fu">case_when</span>(</span>
<span id="cb10-3457"><a href="#cb10-3457" aria-hidden="true" tabindex="-1"></a>        HORMONE_THERAPY <span class="sc">==</span> <span class="st">"YES"</span> <span class="sc">&amp;</span> CHEMOTHERAPY <span class="sc">==</span> <span class="st">"YES"</span> <span class="sc">~</span> <span class="st">"ChemoEndo"</span>,</span>
<span id="cb10-3458"><a href="#cb10-3458" aria-hidden="true" tabindex="-1"></a>        HORMONE_THERAPY <span class="sc">==</span> <span class="st">"YES"</span> <span class="sc">&amp;</span> CHEMOTHERAPY <span class="sc">==</span> <span class="st">"NO"</span> <span class="sc">~</span> <span class="st">"Endo"</span>,</span>
<span id="cb10-3459"><a href="#cb10-3459" aria-hidden="true" tabindex="-1"></a>        <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"Other"</span></span>
<span id="cb10-3460"><a href="#cb10-3460" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">%&gt;%</span> </span>
<span id="cb10-3461"><a href="#cb10-3461" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">node_status =</span> <span class="fu">ifelse</span>(</span>
<span id="cb10-3462"><a href="#cb10-3462" aria-hidden="true" tabindex="-1"></a>        LYMPH_NODES_EXAMINED_POSITIVE <span class="sc">&gt;</span> <span class="dv">0</span>, </span>
<span id="cb10-3463"><a href="#cb10-3463" aria-hidden="true" tabindex="-1"></a>        <span class="st">"pos"</span>,</span>
<span id="cb10-3464"><a href="#cb10-3464" aria-hidden="true" tabindex="-1"></a>        <span class="st">"neg"</span></span>
<span id="cb10-3465"><a href="#cb10-3465" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">%&gt;%</span> </span>
<span id="cb10-3466"><a href="#cb10-3466" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb10-3467"><a href="#cb10-3467" aria-hidden="true" tabindex="-1"></a>        <span class="at">node_status =</span> <span class="fu">factor</span>(node_status, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"neg"</span>, <span class="st">"pos"</span>)),</span>
<span id="cb10-3468"><a href="#cb10-3468" aria-hidden="true" tabindex="-1"></a>        <span class="at">tumor_size =</span> TUMOR_SIZE</span>
<span id="cb10-3469"><a href="#cb10-3469" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span> </span>
<span id="cb10-3470"><a href="#cb10-3470" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb10-3471"><a href="#cb10-3471" aria-hidden="true" tabindex="-1"></a>            HORMONE_THERAPY <span class="sc">==</span> <span class="st">"YES"</span> <span class="sc">&amp;</span> </span>
<span id="cb10-3472"><a href="#cb10-3472" aria-hidden="true" tabindex="-1"></a>            CHEMOTHERAPY <span class="sc">==</span> <span class="st">"NO"</span> <span class="sc">&amp;</span></span>
<span id="cb10-3473"><a href="#cb10-3473" aria-hidden="true" tabindex="-1"></a>            <span class="sc">!</span><span class="fu">is.na</span>(rfs_status) <span class="sc">&amp;</span> </span>
<span id="cb10-3474"><a href="#cb10-3474" aria-hidden="true" tabindex="-1"></a>            <span class="sc">!</span><span class="fu">is.na</span>(rfs_months) <span class="sc">&amp;</span></span>
<span id="cb10-3475"><a href="#cb10-3475" aria-hidden="true" tabindex="-1"></a>            er_status <span class="sc">==</span> <span class="st">"pos"</span> <span class="sc">&amp;</span></span>
<span id="cb10-3476"><a href="#cb10-3476" aria-hidden="true" tabindex="-1"></a>            <span class="sc">!</span><span class="fu">is.na</span>(age) <span class="sc">&amp;</span></span>
<span id="cb10-3477"><a href="#cb10-3477" aria-hidden="true" tabindex="-1"></a>            <span class="sc">!</span><span class="fu">is.na</span>(tumor_size) <span class="sc">&amp;</span></span>
<span id="cb10-3478"><a href="#cb10-3478" aria-hidden="true" tabindex="-1"></a>            <span class="sc">!</span><span class="fu">is.na</span>(node_status)</span>
<span id="cb10-3479"><a href="#cb10-3479" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb10-3480"><a href="#cb10-3480" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3481"><a href="#cb10-3481" aria-hidden="true" tabindex="-1"></a>endo_scanb_metabric <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(</span>
<span id="cb10-3482"><a href="#cb10-3482" aria-hidden="true" tabindex="-1"></a>    endo_only_metabric,</span>
<span id="cb10-3483"><a href="#cb10-3483" aria-hidden="true" tabindex="-1"></a>    endo_only_scanb</span>
<span id="cb10-3484"><a href="#cb10-3484" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb10-3485"><a href="#cb10-3485" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">inner_join</span>(</span>
<span id="cb10-3486"><a href="#cb10-3486" aria-hidden="true" tabindex="-1"></a>        mammaprint_datasets,</span>
<span id="cb10-3487"><a href="#cb10-3487" aria-hidden="true" tabindex="-1"></a>        <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"sample_name"</span>, <span class="st">"cohort"</span>)</span>
<span id="cb10-3488"><a href="#cb10-3488" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb10-3489"><a href="#cb10-3489" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3490"><a href="#cb10-3490" aria-hidden="true" tabindex="-1"></a>endo_scanb_metabric<span class="sc">$</span>risk_score <span class="ot">&lt;-</span> <span class="fu">calculate_risk_score</span>(</span>
<span id="cb10-3491"><a href="#cb10-3491" aria-hidden="true" tabindex="-1"></a>    coefs_risk_model, endo_scanb_metabric</span>
<span id="cb10-3492"><a href="#cb10-3492" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-3493"><a href="#cb10-3493" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-3494"><a href="#cb10-3494" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3495"><a href="#cb10-3495" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=14, fig.height=8}</span></span>
<span id="cb10-3496"><a href="#cb10-3496" aria-hidden="true" tabindex="-1"></a>endo_scanb_metabric <span class="sc">%&gt;%</span></span>
<span id="cb10-3497"><a href="#cb10-3497" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(pam50 <span class="sc">!=</span> <span class="st">"claudin-low"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-3498"><a href="#cb10-3498" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(</span>
<span id="cb10-3499"><a href="#cb10-3499" aria-hidden="true" tabindex="-1"></a>        <span class="at">cols =</span> dplyr<span class="sc">::</span><span class="fu">all_of</span>(<span class="fu">c</span>(</span>
<span id="cb10-3500"><a href="#cb10-3500" aria-hidden="true" tabindex="-1"></a>            <span class="st">"risk_score"</span>,</span>
<span id="cb10-3501"><a href="#cb10-3501" aria-hidden="true" tabindex="-1"></a>            <span class="st">"HALLMARK_G2M_CHECKPOINT"</span>,</span>
<span id="cb10-3502"><a href="#cb10-3502" aria-hidden="true" tabindex="-1"></a>            <span class="st">"SET_ERPR"</span></span>
<span id="cb10-3503"><a href="#cb10-3503" aria-hidden="true" tabindex="-1"></a>        )),</span>
<span id="cb10-3504"><a href="#cb10-3504" aria-hidden="true" tabindex="-1"></a>        <span class="at">values_to =</span> <span class="st">"score"</span>,</span>
<span id="cb10-3505"><a href="#cb10-3505" aria-hidden="true" tabindex="-1"></a>        <span class="at">names_to =</span> <span class="st">"pathway"</span></span>
<span id="cb10-3506"><a href="#cb10-3506" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb10-3507"><a href="#cb10-3507" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> mammaprint, <span class="at">y =</span> score, <span class="at">color =</span> pam50)) <span class="sc">+</span> </span>
<span id="cb10-3508"><a href="#cb10-3508" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb10-3509"><a href="#cb10-3509" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_smooth</span>(</span>
<span id="cb10-3510"><a href="#cb10-3510" aria-hidden="true" tabindex="-1"></a>        <span class="at">method =</span> <span class="st">"loess"</span>,</span>
<span id="cb10-3511"><a href="#cb10-3511" aria-hidden="true" tabindex="-1"></a>        <span class="at">formula =</span> y <span class="sc">~</span> x,</span>
<span id="cb10-3512"><a href="#cb10-3512" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">"black"</span></span>
<span id="cb10-3513"><a href="#cb10-3513" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb10-3514"><a href="#cb10-3514" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(</span>
<span id="cb10-3515"><a href="#cb10-3515" aria-hidden="true" tabindex="-1"></a>        cohort<span class="sc">~</span>pathway, </span>
<span id="cb10-3516"><a href="#cb10-3516" aria-hidden="true" tabindex="-1"></a>        <span class="at">scales =</span> <span class="st">"free_y"</span>,</span>
<span id="cb10-3517"><a href="#cb10-3517" aria-hidden="true" tabindex="-1"></a>        <span class="at">labeller =</span> <span class="fu">as_labeller</span>(<span class="fu">c</span>(</span>
<span id="cb10-3518"><a href="#cb10-3518" aria-hidden="true" tabindex="-1"></a>            <span class="st">"scanb"</span> <span class="ot">=</span> <span class="st">"SCANB"</span>,</span>
<span id="cb10-3519"><a href="#cb10-3519" aria-hidden="true" tabindex="-1"></a>            <span class="st">"metabric"</span> <span class="ot">=</span> <span class="st">"METABRIC"</span>,</span>
<span id="cb10-3520"><a href="#cb10-3520" aria-hidden="true" tabindex="-1"></a>            <span class="st">"HALLMARK_G2M_CHECKPOINT"</span> <span class="ot">=</span> <span class="st">"G2M"</span>,</span>
<span id="cb10-3521"><a href="#cb10-3521" aria-hidden="true" tabindex="-1"></a>            <span class="st">"SET_ERPR"</span> <span class="ot">=</span> <span class="st">"SET ER/PR"</span>,</span>
<span id="cb10-3522"><a href="#cb10-3522" aria-hidden="true" tabindex="-1"></a>            <span class="st">"risk_score"</span> <span class="ot">=</span> <span class="st">"Risk score"</span></span>
<span id="cb10-3523"><a href="#cb10-3523" aria-hidden="true" tabindex="-1"></a>        ))</span>
<span id="cb10-3524"><a href="#cb10-3524" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb10-3525"><a href="#cb10-3525" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>) <span class="sc">+</span> </span>
<span id="cb10-3526"><a href="#cb10-3526" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">get_colors_pam50</span>(endo_scanb_metabric)) <span class="sc">+</span></span>
<span id="cb10-3527"><a href="#cb10-3527" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span> </span>
<span id="cb10-3528"><a href="#cb10-3528" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb10-3529"><a href="#cb10-3529" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-3530"><a href="#cb10-3530" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3531"><a href="#cb10-3531" aria-hidden="true" tabindex="-1"></a>Now for OncotypeDX:</span>
<span id="cb10-3532"><a href="#cb10-3532" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3535"><a href="#cb10-3535" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-3536"><a href="#cb10-3536" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-ror-genefu-abim-oncotype</span></span>
<span id="cb10-3537"><a href="#cb10-3537" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Correlation between oncotypeDX and </span></span>
<span id="cb10-3538"><a href="#cb10-3538" aria-hidden="true" tabindex="-1"></a><span class="co">#|     R package genefu</span></span>
<span id="cb10-3539"><a href="#cb10-3539" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3540"><a href="#cb10-3540" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(sig.endoPredict)</span>
<span id="cb10-3541"><a href="#cb10-3541" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(sig.oncotypedx)</span>
<span id="cb10-3542"><a href="#cb10-3542" aria-hidden="true" tabindex="-1"></a>ep_abim <span class="ot">&lt;-</span> genefu<span class="sc">::</span><span class="fu">endoPredict</span>(</span>
<span id="cb10-3543"><a href="#cb10-3543" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> <span class="fu">t</span>(<span class="fu">assay</span>(abim_100, <span class="st">"log2fpkm"</span>)), </span>
<span id="cb10-3544"><a href="#cb10-3544" aria-hidden="true" tabindex="-1"></a>    <span class="at">annot =</span> annot_abim,</span>
<span id="cb10-3545"><a href="#cb10-3545" aria-hidden="true" tabindex="-1"></a>    <span class="at">do.mapping =</span> <span class="cn">TRUE</span>,</span>
<span id="cb10-3546"><a href="#cb10-3546" aria-hidden="true" tabindex="-1"></a>    <span class="at">verbose =</span> <span class="cn">TRUE</span></span>
<span id="cb10-3547"><a href="#cb10-3547" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-3548"><a href="#cb10-3548" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3549"><a href="#cb10-3549" aria-hidden="true" tabindex="-1"></a>oncotypedx_abim <span class="ot">&lt;-</span> genefu<span class="sc">::</span><span class="fu">oncotypedx</span>(</span>
<span id="cb10-3550"><a href="#cb10-3550" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> <span class="fu">t</span>(<span class="fu">assay</span>(abim_100, <span class="st">"log2fpkm"</span>)), </span>
<span id="cb10-3551"><a href="#cb10-3551" aria-hidden="true" tabindex="-1"></a>    <span class="at">annot =</span> annot_abim,</span>
<span id="cb10-3552"><a href="#cb10-3552" aria-hidden="true" tabindex="-1"></a>    <span class="at">do.mapping =</span> <span class="cn">TRUE</span>,</span>
<span id="cb10-3553"><a href="#cb10-3553" aria-hidden="true" tabindex="-1"></a>    <span class="at">verbose =</span> <span class="cn">TRUE</span>,</span>
<span id="cb10-3554"><a href="#cb10-3554" aria-hidden="true" tabindex="-1"></a>    <span class="at">do.scaling =</span> <span class="cn">FALSE</span></span>
<span id="cb10-3555"><a href="#cb10-3555" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-3556"><a href="#cb10-3556" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3557"><a href="#cb10-3557" aria-hidden="true" tabindex="-1"></a>abim_100<span class="sc">$</span>GENEFU.ONCOTYPEDX <span class="ot">&lt;-</span> oncotypedx_abim<span class="sc">$</span>score</span>
<span id="cb10-3558"><a href="#cb10-3558" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3559"><a href="#cb10-3559" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(abim_100) <span class="sc">%&gt;%</span></span>
<span id="cb10-3560"><a href="#cb10-3560" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span></span>
<span id="cb10-3561"><a href="#cb10-3561" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(</span>
<span id="cb10-3562"><a href="#cb10-3562" aria-hidden="true" tabindex="-1"></a>        <span class="fu">aes</span>(</span>
<span id="cb10-3563"><a href="#cb10-3563" aria-hidden="true" tabindex="-1"></a>            <span class="at">x =</span> GENEFU.ONCOTYPEDX, </span>
<span id="cb10-3564"><a href="#cb10-3564" aria-hidden="true" tabindex="-1"></a>            <span class="at">y =</span> risk_score, </span>
<span id="cb10-3565"><a href="#cb10-3565" aria-hidden="true" tabindex="-1"></a>            <span class="at">color =</span> ProsignaFT.Subtype</span>
<span id="cb10-3566"><a href="#cb10-3566" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb10-3567"><a href="#cb10-3567" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb10-3568"><a href="#cb10-3568" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span> </span>
<span id="cb10-3569"><a href="#cb10-3569" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_smooth</span>(</span>
<span id="cb10-3570"><a href="#cb10-3570" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">"black"</span>,</span>
<span id="cb10-3571"><a href="#cb10-3571" aria-hidden="true" tabindex="-1"></a>        <span class="at">method =</span> <span class="st">"loess"</span>,</span>
<span id="cb10-3572"><a href="#cb10-3572" aria-hidden="true" tabindex="-1"></a>        <span class="at">formula =</span> <span class="st">"y ~ x"</span></span>
<span id="cb10-3573"><a href="#cb10-3573" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb10-3574"><a href="#cb10-3574" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb10-3575"><a href="#cb10-3575" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span></span>
<span id="cb10-3576"><a href="#cb10-3576" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb10-3577"><a href="#cb10-3577" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-3578"><a href="#cb10-3578" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3579"><a href="#cb10-3579" aria-hidden="true" tabindex="-1"></a>There is also a correlation between the risk score and oncotypeDX.</span>
<span id="cb10-3580"><a href="#cb10-3580" aria-hidden="true" tabindex="-1"></a>The paper https://www.nature.com/articles/s41523-022-00492-0#Sec8</span>
<span id="cb10-3581"><a href="#cb10-3581" aria-hidden="true" tabindex="-1"></a>also used the genefu for calculating the oncotypeDX. </span>
<span id="cb10-3582"><a href="#cb10-3582" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3583"><a href="#cb10-3583" aria-hidden="true" tabindex="-1"></a><span class="fu">### genefu and AIMS</span></span>
<span id="cb10-3584"><a href="#cb10-3584" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3585"><a href="#cb10-3585" aria-hidden="true" tabindex="-1"></a>Here we try to use AIMS to calculate the PAM50 subtypes for the</span>
<span id="cb10-3586"><a href="#cb10-3586" aria-hidden="true" tabindex="-1"></a>and compare with the results obtained previously. We want to compare</span>
<span id="cb10-3587"><a href="#cb10-3587" aria-hidden="true" tabindex="-1"></a>the classifications with the molecular landscape.</span>
<span id="cb10-3588"><a href="#cb10-3588" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3591"><a href="#cb10-3591" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-3592"><a href="#cb10-3592" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(AIMSmodel)</span>
<span id="cb10-3593"><a href="#cb10-3593" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3594"><a href="#cb10-3594" aria-hidden="true" tabindex="-1"></a><span class="co"># we have to match the symbols</span></span>
<span id="cb10-3595"><a href="#cb10-3595" aria-hidden="true" tabindex="-1"></a>annot_aims <span class="ot">&lt;-</span> annotables<span class="sc">::</span>grch38 <span class="sc">%&gt;%</span> </span>
<span id="cb10-3596"><a href="#cb10-3596" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(entrez, symbol) <span class="sc">%&gt;%</span></span>
<span id="cb10-3597"><a href="#cb10-3597" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">EntrezGene.ID =</span> entrez) <span class="sc">%&gt;%</span></span>
<span id="cb10-3598"><a href="#cb10-3598" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">EntrezGene.ID =</span> <span class="fu">as.character</span>(EntrezGene.ID)) <span class="sc">%&gt;%</span></span>
<span id="cb10-3599"><a href="#cb10-3599" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(symbol <span class="sc">%in%</span> <span class="fu">rownames</span>(pdx_tpm)) <span class="sc">%&gt;%</span></span>
<span id="cb10-3600"><a href="#cb10-3600" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">duplicated</span>(symbol)) <span class="sc">%&gt;%</span></span>
<span id="cb10-3601"><a href="#cb10-3601" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span></span>
<span id="cb10-3602"><a href="#cb10-3602" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">rownames&lt;-</span><span class="st">`</span>(.<span class="sc">$</span>symbol)</span>
<span id="cb10-3603"><a href="#cb10-3603" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3604"><a href="#cb10-3604" aria-hidden="true" tabindex="-1"></a>pdx_tpm <span class="ot">&lt;-</span> pdx_tpm[annot_aims<span class="sc">$</span>symbol, ]</span>
<span id="cb10-3605"><a href="#cb10-3605" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3606"><a href="#cb10-3606" aria-hidden="true" tabindex="-1"></a>sbt_pdx_AIMS <span class="ot">&lt;-</span> genefu<span class="sc">::</span><span class="fu">molecular.subtyping</span>(</span>
<span id="cb10-3607"><a href="#cb10-3607" aria-hidden="true" tabindex="-1"></a>    <span class="at">sbt.model =</span> <span class="st">"AIMS"</span>, </span>
<span id="cb10-3608"><a href="#cb10-3608" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> <span class="fu">assay</span>(pdx_tpm, <span class="st">"log2tpm"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-3609"><a href="#cb10-3609" aria-hidden="true" tabindex="-1"></a>        t,</span>
<span id="cb10-3610"><a href="#cb10-3610" aria-hidden="true" tabindex="-1"></a>    <span class="at">annot =</span> annot_aims,</span>
<span id="cb10-3611"><a href="#cb10-3611" aria-hidden="true" tabindex="-1"></a>    <span class="at">do.mapping =</span> <span class="cn">TRUE</span></span>
<span id="cb10-3612"><a href="#cb10-3612" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-3613"><a href="#cb10-3613" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3614"><a href="#cb10-3614" aria-hidden="true" tabindex="-1"></a><span class="co"># try with abim as we have the baseline values</span></span>
<span id="cb10-3615"><a href="#cb10-3615" aria-hidden="true" tabindex="-1"></a>abim_100 <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../results/rds_files/validation/abim_100.rds"</span>)</span>
<span id="cb10-3616"><a href="#cb10-3616" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3617"><a href="#cb10-3617" aria-hidden="true" tabindex="-1"></a>annot_aims_abim_100 <span class="ot">&lt;-</span> annotables<span class="sc">::</span>grch38 <span class="sc">%&gt;%</span> </span>
<span id="cb10-3618"><a href="#cb10-3618" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(entrez, symbol) <span class="sc">%&gt;%</span></span>
<span id="cb10-3619"><a href="#cb10-3619" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">EntrezGene.ID =</span> entrez) <span class="sc">%&gt;%</span></span>
<span id="cb10-3620"><a href="#cb10-3620" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">EntrezGene.ID =</span> <span class="fu">as.character</span>(EntrezGene.ID)) <span class="sc">%&gt;%</span></span>
<span id="cb10-3621"><a href="#cb10-3621" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(symbol <span class="sc">%in%</span> <span class="fu">rownames</span>(abim_100)) <span class="sc">%&gt;%</span></span>
<span id="cb10-3622"><a href="#cb10-3622" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">duplicated</span>(symbol)) <span class="sc">%&gt;%</span></span>
<span id="cb10-3623"><a href="#cb10-3623" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span></span>
<span id="cb10-3624"><a href="#cb10-3624" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">rownames&lt;-</span><span class="st">`</span>(.<span class="sc">$</span>symbol)</span>
<span id="cb10-3625"><a href="#cb10-3625" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3626"><a href="#cb10-3626" aria-hidden="true" tabindex="-1"></a>abim_100 <span class="ot">&lt;-</span> abim_100[annot_aims_abim_100<span class="sc">$</span>symbol, ]</span>
<span id="cb10-3627"><a href="#cb10-3627" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3628"><a href="#cb10-3628" aria-hidden="true" tabindex="-1"></a>sbt_aim_100_AIMS <span class="ot">&lt;-</span> genefu<span class="sc">::</span><span class="fu">molecular.subtyping</span>(</span>
<span id="cb10-3629"><a href="#cb10-3629" aria-hidden="true" tabindex="-1"></a>    <span class="at">sbt.model =</span> <span class="st">"AIMS"</span>, </span>
<span id="cb10-3630"><a href="#cb10-3630" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> <span class="fu">assay</span>(abim_100, <span class="st">"log2fpkm"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-3631"><a href="#cb10-3631" aria-hidden="true" tabindex="-1"></a>        t,</span>
<span id="cb10-3632"><a href="#cb10-3632" aria-hidden="true" tabindex="-1"></a>    <span class="at">annot =</span> annot_aims_abim_100,</span>
<span id="cb10-3633"><a href="#cb10-3633" aria-hidden="true" tabindex="-1"></a>    <span class="at">do.mapping =</span> <span class="cn">TRUE</span></span>
<span id="cb10-3634"><a href="#cb10-3634" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-3635"><a href="#cb10-3635" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3636"><a href="#cb10-3636" aria-hidden="true" tabindex="-1"></a>abim_100<span class="sc">$</span>aims_pam50 <span class="ot">&lt;-</span> sbt_aim_100_AIMS<span class="sc">$</span>subtype <span class="sc">%&gt;%</span> <span class="fu">as.vector</span>()</span>
<span id="cb10-3637"><a href="#cb10-3637" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3638"><a href="#cb10-3638" aria-hidden="true" tabindex="-1"></a><span class="co"># try with swiss normal as we have the baseline values</span></span>
<span id="cb10-3639"><a href="#cb10-3639" aria-hidden="true" tabindex="-1"></a>normal_swiss <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../results/rds_files/validation/normal_swiss.rds"</span>)</span>
<span id="cb10-3640"><a href="#cb10-3640" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3641"><a href="#cb10-3641" aria-hidden="true" tabindex="-1"></a>annot_aims_normal_swiss <span class="ot">&lt;-</span> annotables<span class="sc">::</span>grch38 <span class="sc">%&gt;%</span> </span>
<span id="cb10-3642"><a href="#cb10-3642" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(entrez, symbol) <span class="sc">%&gt;%</span></span>
<span id="cb10-3643"><a href="#cb10-3643" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">EntrezGene.ID =</span> entrez) <span class="sc">%&gt;%</span></span>
<span id="cb10-3644"><a href="#cb10-3644" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">EntrezGene.ID =</span> <span class="fu">as.character</span>(EntrezGene.ID)) <span class="sc">%&gt;%</span></span>
<span id="cb10-3645"><a href="#cb10-3645" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(symbol <span class="sc">%in%</span> <span class="fu">rownames</span>(normal_swiss)) <span class="sc">%&gt;%</span></span>
<span id="cb10-3646"><a href="#cb10-3646" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">duplicated</span>(symbol)) <span class="sc">%&gt;%</span></span>
<span id="cb10-3647"><a href="#cb10-3647" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span></span>
<span id="cb10-3648"><a href="#cb10-3648" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">rownames&lt;-</span><span class="st">`</span>(.<span class="sc">$</span>symbol)</span>
<span id="cb10-3649"><a href="#cb10-3649" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3650"><a href="#cb10-3650" aria-hidden="true" tabindex="-1"></a>normal_swiss <span class="ot">&lt;-</span> normal_swiss[annot_aims_normal_swiss<span class="sc">$</span>symbol, ]</span>
<span id="cb10-3651"><a href="#cb10-3651" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3652"><a href="#cb10-3652" aria-hidden="true" tabindex="-1"></a>sbt_normal_swiss_AIMS <span class="ot">&lt;-</span> genefu<span class="sc">::</span><span class="fu">molecular.subtyping</span>(</span>
<span id="cb10-3653"><a href="#cb10-3653" aria-hidden="true" tabindex="-1"></a>    <span class="at">sbt.model =</span> <span class="st">"AIMS"</span>, </span>
<span id="cb10-3654"><a href="#cb10-3654" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> <span class="fu">assay</span>(normal_swiss, <span class="st">"log2fpkm"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-3655"><a href="#cb10-3655" aria-hidden="true" tabindex="-1"></a>        t,</span>
<span id="cb10-3656"><a href="#cb10-3656" aria-hidden="true" tabindex="-1"></a>    <span class="at">annot =</span> annot_aims_normal_swiss,</span>
<span id="cb10-3657"><a href="#cb10-3657" aria-hidden="true" tabindex="-1"></a>    <span class="at">do.mapping =</span> <span class="cn">TRUE</span></span>
<span id="cb10-3658"><a href="#cb10-3658" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-3659"><a href="#cb10-3659" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3660"><a href="#cb10-3660" aria-hidden="true" tabindex="-1"></a>normal_swiss<span class="sc">$</span>aims_pam50 <span class="ot">&lt;-</span> sbt_normal_swiss_AIMS<span class="sc">$</span>subtype <span class="sc">%&gt;%</span> <span class="fu">as.vector</span>()</span>
<span id="cb10-3661"><a href="#cb10-3661" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3662"><a href="#cb10-3662" aria-hidden="true" tabindex="-1"></a><span class="co"># and we try with scanb as well</span></span>
<span id="cb10-3663"><a href="#cb10-3663" aria-hidden="true" tabindex="-1"></a>annot_aims_scanb <span class="ot">&lt;-</span> annotables<span class="sc">::</span>grch38 <span class="sc">%&gt;%</span> </span>
<span id="cb10-3664"><a href="#cb10-3664" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(entrez, symbol) <span class="sc">%&gt;%</span></span>
<span id="cb10-3665"><a href="#cb10-3665" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">EntrezGene.ID =</span> entrez) <span class="sc">%&gt;%</span></span>
<span id="cb10-3666"><a href="#cb10-3666" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">EntrezGene.ID =</span> <span class="fu">as.character</span>(EntrezGene.ID)) <span class="sc">%&gt;%</span></span>
<span id="cb10-3667"><a href="#cb10-3667" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(symbol <span class="sc">%in%</span> <span class="fu">rownames</span>(datasets<span class="sc">$</span>scanb)) <span class="sc">%&gt;%</span></span>
<span id="cb10-3668"><a href="#cb10-3668" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">duplicated</span>(symbol)) <span class="sc">%&gt;%</span></span>
<span id="cb10-3669"><a href="#cb10-3669" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span></span>
<span id="cb10-3670"><a href="#cb10-3670" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">rownames&lt;-</span><span class="st">`</span>(.<span class="sc">$</span>symbol)</span>
<span id="cb10-3671"><a href="#cb10-3671" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3672"><a href="#cb10-3672" aria-hidden="true" tabindex="-1"></a>scanb <span class="ot">&lt;-</span> datasets<span class="sc">$</span>scanb[annot_aims_scanb<span class="sc">$</span>symbol, ]</span>
<span id="cb10-3673"><a href="#cb10-3673" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3674"><a href="#cb10-3674" aria-hidden="true" tabindex="-1"></a>sbt_scanb_AIMS <span class="ot">&lt;-</span> genefu<span class="sc">::</span><span class="fu">molecular.subtyping</span>(</span>
<span id="cb10-3675"><a href="#cb10-3675" aria-hidden="true" tabindex="-1"></a>    <span class="at">sbt.model =</span> <span class="st">"AIMS"</span>, </span>
<span id="cb10-3676"><a href="#cb10-3676" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> <span class="fu">assay</span>(scanb, <span class="st">"logFPKM"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-3677"><a href="#cb10-3677" aria-hidden="true" tabindex="-1"></a>        t,</span>
<span id="cb10-3678"><a href="#cb10-3678" aria-hidden="true" tabindex="-1"></a>    <span class="at">annot =</span> annot_aims_scanb,</span>
<span id="cb10-3679"><a href="#cb10-3679" aria-hidden="true" tabindex="-1"></a>    <span class="at">do.mapping =</span> <span class="cn">TRUE</span></span>
<span id="cb10-3680"><a href="#cb10-3680" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-3681"><a href="#cb10-3681" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3682"><a href="#cb10-3682" aria-hidden="true" tabindex="-1"></a>scanb<span class="sc">$</span>aims_pam50 <span class="ot">&lt;-</span> sbt_scanb_AIMS<span class="sc">$</span>subtype <span class="sc">%&gt;%</span> as.vector</span>
<span id="cb10-3683"><a href="#cb10-3683" aria-hidden="true" tabindex="-1"></a>scanb_coldata <span class="ot">&lt;-</span> <span class="fu">colData</span>(scanb) <span class="sc">%&gt;%</span></span>
<span id="cb10-3684"><a href="#cb10-3684" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span></span>
<span id="cb10-3685"><a href="#cb10-3685" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb10-3686"><a href="#cb10-3686" aria-hidden="true" tabindex="-1"></a>        <span class="at">aims_pam50 =</span> <span class="fu">factor</span>(<span class="fu">tolower</span>(aims_pam50), <span class="at">levels =</span> <span class="fu">unique</span>(pam50)),</span>
<span id="cb10-3687"><a href="#cb10-3687" aria-hidden="true" tabindex="-1"></a>        <span class="at">pam50 =</span> <span class="fu">factor</span>(pam50, <span class="at">levels =</span> <span class="fu">unique</span>(pam50))</span>
<span id="cb10-3688"><a href="#cb10-3688" aria-hidden="true" tabindex="-1"></a>    ) </span>
<span id="cb10-3689"><a href="#cb10-3689" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3690"><a href="#cb10-3690" aria-hidden="true" tabindex="-1"></a><span class="co"># trying with MCF7</span></span>
<span id="cb10-3691"><a href="#cb10-3691" aria-hidden="true" tabindex="-1"></a>mcf7 <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../results/rds_files/trying/mcf7_scabia.rds"</span>)</span>
<span id="cb10-3692"><a href="#cb10-3692" aria-hidden="true" tabindex="-1"></a>mcf7 <span class="ot">&lt;-</span> mcf7[<span class="fu">rowSums</span>(<span class="fu">assay</span>(mcf7, <span class="st">"counts"</span>)) <span class="sc">&gt;</span> <span class="dv">100</span>, ]</span>
<span id="cb10-3693"><a href="#cb10-3693" aria-hidden="true" tabindex="-1"></a><span class="fu">assay</span>(mcf7, <span class="st">"logcpm"</span>) <span class="ot">&lt;-</span> edgeR<span class="sc">::</span><span class="fu">cpm</span>(<span class="fu">assay</span>(mcf7, <span class="st">"counts"</span>), <span class="at">log =</span> <span class="cn">TRUE</span>)</span>
<span id="cb10-3694"><a href="#cb10-3694" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3695"><a href="#cb10-3695" aria-hidden="true" tabindex="-1"></a>annot_aims_mcf7 <span class="ot">&lt;-</span> annotables<span class="sc">::</span>grch38 <span class="sc">%&gt;%</span> </span>
<span id="cb10-3696"><a href="#cb10-3696" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(entrez, symbol) <span class="sc">%&gt;%</span></span>
<span id="cb10-3697"><a href="#cb10-3697" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">EntrezGene.ID =</span> entrez) <span class="sc">%&gt;%</span></span>
<span id="cb10-3698"><a href="#cb10-3698" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">EntrezGene.ID =</span> <span class="fu">as.character</span>(EntrezGene.ID)) <span class="sc">%&gt;%</span></span>
<span id="cb10-3699"><a href="#cb10-3699" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(symbol <span class="sc">%in%</span> <span class="fu">rownames</span>(mcf7)) <span class="sc">%&gt;%</span></span>
<span id="cb10-3700"><a href="#cb10-3700" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">duplicated</span>(symbol)) <span class="sc">%&gt;%</span></span>
<span id="cb10-3701"><a href="#cb10-3701" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span></span>
<span id="cb10-3702"><a href="#cb10-3702" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">rownames&lt;-</span><span class="st">`</span>(.<span class="sc">$</span>symbol)</span>
<span id="cb10-3703"><a href="#cb10-3703" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3704"><a href="#cb10-3704" aria-hidden="true" tabindex="-1"></a>mcf7 <span class="ot">&lt;-</span> mcf7[annot_aims_mcf7<span class="sc">$</span>symbol, ]</span>
<span id="cb10-3705"><a href="#cb10-3705" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3706"><a href="#cb10-3706" aria-hidden="true" tabindex="-1"></a>sbt_mcf7_AIMS <span class="ot">&lt;-</span> genefu<span class="sc">::</span><span class="fu">molecular.subtyping</span>(</span>
<span id="cb10-3707"><a href="#cb10-3707" aria-hidden="true" tabindex="-1"></a>    <span class="at">sbt.model =</span> <span class="st">"AIMS"</span>, </span>
<span id="cb10-3708"><a href="#cb10-3708" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> <span class="fu">assay</span>(mcf7, <span class="st">"logcpm"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-3709"><a href="#cb10-3709" aria-hidden="true" tabindex="-1"></a>        t,</span>
<span id="cb10-3710"><a href="#cb10-3710" aria-hidden="true" tabindex="-1"></a>    <span class="at">annot =</span> annot_aims_mcf7,</span>
<span id="cb10-3711"><a href="#cb10-3711" aria-hidden="true" tabindex="-1"></a>    <span class="at">do.mapping =</span> <span class="cn">TRUE</span></span>
<span id="cb10-3712"><a href="#cb10-3712" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-3713"><a href="#cb10-3713" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3714"><a href="#cb10-3714" aria-hidden="true" tabindex="-1"></a>mcf7<span class="sc">$</span>aims_pam50 <span class="ot">&lt;-</span> sbt_mcf7_AIMS<span class="sc">$</span>subtype <span class="sc">%&gt;%</span> <span class="fu">as.vector</span>()</span>
<span id="cb10-3715"><a href="#cb10-3715" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-3716"><a href="#cb10-3716" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3717"><a href="#cb10-3717" aria-hidden="true" tabindex="-1"></a>The confusion matrix below show the results of the AIMS PAM50 as the</span>
<span id="cb10-3718"><a href="#cb10-3718" aria-hidden="true" tabindex="-1"></a>predicted and the reference is the PAM50 subtypes obtained from the</span>
<span id="cb10-3719"><a href="#cb10-3719" aria-hidden="true" tabindex="-1"></a>SCANB team.</span>
<span id="cb10-3720"><a href="#cb10-3720" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3723"><a href="#cb10-3723" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-3724"><a href="#cb10-3724" aria-hidden="true" tabindex="-1"></a>caret<span class="sc">::</span><span class="fu">confusionMatrix</span>(scanb_coldata<span class="sc">$</span>aims_pam50, scanb_coldata<span class="sc">$</span>pam50)</span>
<span id="cb10-3725"><a href="#cb10-3725" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-3726"><a href="#cb10-3726" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3727"><a href="#cb10-3727" aria-hidden="true" tabindex="-1"></a>We see that there is a big misclassification from luminal A samples to </span>
<span id="cb10-3728"><a href="#cb10-3728" aria-hidden="true" tabindex="-1"></a>normal like samples and several luminal B are misclassified as luminal A.</span>
<span id="cb10-3729"><a href="#cb10-3729" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3730"><a href="#cb10-3730" aria-hidden="true" tabindex="-1"></a>The plot below shows the coloring by the reference PAM50 and the predicted</span>
<span id="cb10-3731"><a href="#cb10-3731" aria-hidden="true" tabindex="-1"></a>by AIMS.</span>
<span id="cb10-3732"><a href="#cb10-3732" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3733"><a href="#cb10-3733" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=7, fig.height=10}</span></span>
<span id="cb10-3734"><a href="#cb10-3734" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-scanb-aims</span></span>
<span id="cb10-3735"><a href="#cb10-3735" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: SCANB embedding stratified by PAM50 algorithms by both the ones</span></span>
<span id="cb10-3736"><a href="#cb10-3736" aria-hidden="true" tabindex="-1"></a><span class="co">#|     provided by the SCANB (SSP) and the AIMS algorithm</span></span>
<span id="cb10-3737"><a href="#cb10-3737" aria-hidden="true" tabindex="-1"></a>df_pca <span class="sc">%&gt;%</span></span>
<span id="cb10-3738"><a href="#cb10-3738" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"scanb"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-3739"><a href="#cb10-3739" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">inner_join</span>(</span>
<span id="cb10-3740"><a href="#cb10-3740" aria-hidden="true" tabindex="-1"></a>        ., </span>
<span id="cb10-3741"><a href="#cb10-3741" aria-hidden="true" tabindex="-1"></a>        scanb_coldata <span class="sc">%&gt;%</span></span>
<span id="cb10-3742"><a href="#cb10-3742" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">select</span>(aims_pam50, sample_name),</span>
<span id="cb10-3743"><a href="#cb10-3743" aria-hidden="true" tabindex="-1"></a>        <span class="at">by =</span> <span class="st">"sample_name"</span></span>
<span id="cb10-3744"><a href="#cb10-3744" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb10-3745"><a href="#cb10-3745" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb10-3746"><a href="#cb10-3746" aria-hidden="true" tabindex="-1"></a>        <span class="at">AIMS =</span> aims_pam50,</span>
<span id="cb10-3747"><a href="#cb10-3747" aria-hidden="true" tabindex="-1"></a>        <span class="st">`</span><span class="at">SSP SCANB</span><span class="st">`</span> <span class="ot">=</span> pam50</span>
<span id="cb10-3748"><a href="#cb10-3748" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb10-3749"><a href="#cb10-3749" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(</span>
<span id="cb10-3750"><a href="#cb10-3750" aria-hidden="true" tabindex="-1"></a>        <span class="at">cols =</span> dplyr<span class="sc">::</span><span class="fu">all_of</span>(<span class="fu">c</span>(<span class="st">"AIMS"</span>, <span class="st">"SSP SCANB"</span>)),</span>
<span id="cb10-3751"><a href="#cb10-3751" aria-hidden="true" tabindex="-1"></a>        <span class="at">names_to =</span> <span class="st">"pam50_algo"</span>,</span>
<span id="cb10-3752"><a href="#cb10-3752" aria-hidden="true" tabindex="-1"></a>        <span class="at">values_to =</span> <span class="st">"classes"</span></span>
<span id="cb10-3753"><a href="#cb10-3753" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb10-3754"><a href="#cb10-3754" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> PC3, <span class="at">y =</span> PC4, <span class="at">color =</span> classes)) <span class="sc">+</span></span>
<span id="cb10-3755"><a href="#cb10-3755" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb10-3756"><a href="#cb10-3756" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_grid</span>(classes<span class="sc">~</span>pam50_algo) <span class="sc">+</span></span>
<span id="cb10-3757"><a href="#cb10-3757" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb10-3758"><a href="#cb10-3758" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">"PAM50"</span>,</span>
<span id="cb10-3759"><a href="#cb10-3759" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb10-3760"><a href="#cb10-3760" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">15</span>) <span class="sc">+</span></span>
<span id="cb10-3761"><a href="#cb10-3761" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">get_colors_pam50</span>(</span>
<span id="cb10-3762"><a href="#cb10-3762" aria-hidden="true" tabindex="-1"></a>        df_pca <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(pam50 <span class="sc">%in%</span> scanb<span class="sc">$</span>pam50)</span>
<span id="cb10-3763"><a href="#cb10-3763" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">+</span> </span>
<span id="cb10-3764"><a href="#cb10-3764" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>) <span class="sc">+</span> </span>
<span id="cb10-3765"><a href="#cb10-3765" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span> </span>
<span id="cb10-3766"><a href="#cb10-3766" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb10-3767"><a href="#cb10-3767" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-3768"><a href="#cb10-3768" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3769"><a href="#cb10-3769" aria-hidden="true" tabindex="-1"></a>On the left there is the coloring for the AIMS algorithm and on the right</span>
<span id="cb10-3770"><a href="#cb10-3770" aria-hidden="true" tabindex="-1"></a>from the reference. We see the missclassified samples are not randomly</span>
<span id="cb10-3771"><a href="#cb10-3771" aria-hidden="true" tabindex="-1"></a>misclassified, they are on the boundaries. </span>
<span id="cb10-3772"><a href="#cb10-3772" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3773"><a href="#cb10-3773" aria-hidden="true" tabindex="-1"></a>The plot below shows the maximum probability score for each sample</span>
<span id="cb10-3774"><a href="#cb10-3774" aria-hidden="true" tabindex="-1"></a>individually.</span>
<span id="cb10-3775"><a href="#cb10-3775" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3776"><a href="#cb10-3776" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.height=10, fig.width=4}</span></span>
<span id="cb10-3777"><a href="#cb10-3777" aria-hidden="true" tabindex="-1"></a><span class="fu">data.frame</span>(</span>
<span id="cb10-3778"><a href="#cb10-3778" aria-hidden="true" tabindex="-1"></a>    <span class="at">max_prob =</span> <span class="fu">apply</span>(sbt_scanb_AIMS<span class="sc">$</span>subtype.proba, <span class="dv">1</span>, max),</span>
<span id="cb10-3779"><a href="#cb10-3779" aria-hidden="true" tabindex="-1"></a>    <span class="at">aims_pam50 =</span> sbt_scanb_AIMS<span class="sc">$</span>subtype <span class="sc">%&gt;%</span> as.vector</span>
<span id="cb10-3780"><a href="#cb10-3780" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb10-3781"><a href="#cb10-3781" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> max_prob)) <span class="sc">+</span></span>
<span id="cb10-3782"><a href="#cb10-3782" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_histogram</span>(<span class="at">bins =</span> <span class="dv">10</span>) <span class="sc">+</span> </span>
<span id="cb10-3783"><a href="#cb10-3783" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>aims_pam50, <span class="at">ncol =</span> <span class="dv">1</span>) <span class="sc">+</span> </span>
<span id="cb10-3784"><a href="#cb10-3784" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>)</span>
<span id="cb10-3785"><a href="#cb10-3785" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-3786"><a href="#cb10-3786" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3787"><a href="#cb10-3787" aria-hidden="true" tabindex="-1"></a>The majority of samples have probabilities of either 1 or </span>
<span id="cb10-3788"><a href="#cb10-3788" aria-hidden="true" tabindex="-1"></a><span class="ss">0. </span>Not so often in between. </span>
<span id="cb10-3789"><a href="#cb10-3789" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3790"><a href="#cb10-3790" aria-hidden="true" tabindex="-1"></a>And the figure below shows the confusion matrix for the ABIM cohort.</span>
<span id="cb10-3791"><a href="#cb10-3791" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3794"><a href="#cb10-3794" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-3795"><a href="#cb10-3795" aria-hidden="true" tabindex="-1"></a>abim_coldata <span class="ot">&lt;-</span> <span class="fu">colData</span>(abim_100) <span class="sc">%&gt;%</span></span>
<span id="cb10-3796"><a href="#cb10-3796" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span></span>
<span id="cb10-3797"><a href="#cb10-3797" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb10-3798"><a href="#cb10-3798" aria-hidden="true" tabindex="-1"></a>        <span class="at">aims_pam50 =</span> <span class="fu">factor</span>(<span class="fu">tolower</span>(aims_pam50), <span class="at">levels =</span> <span class="fu">unique</span>(pam50)),</span>
<span id="cb10-3799"><a href="#cb10-3799" aria-hidden="true" tabindex="-1"></a>        <span class="at">pam50 =</span> <span class="fu">factor</span>(pam50, <span class="at">levels =</span> <span class="fu">unique</span>(pam50))</span>
<span id="cb10-3800"><a href="#cb10-3800" aria-hidden="true" tabindex="-1"></a>    ) </span>
<span id="cb10-3801"><a href="#cb10-3801" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb10-3802"><a href="#cb10-3802" aria-hidden="true" tabindex="-1"></a>caret<span class="sc">::</span><span class="fu">confusionMatrix</span>(abim_coldata<span class="sc">$</span>aims_pam50, abim_coldata<span class="sc">$</span>pam50)</span>
<span id="cb10-3803"><a href="#cb10-3803" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-3804"><a href="#cb10-3804" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3805"><a href="#cb10-3805" aria-hidden="true" tabindex="-1"></a>We see here that in this case that there are several luminal B samples </span>
<span id="cb10-3806"><a href="#cb10-3806" aria-hidden="true" tabindex="-1"></a>missclassified as luminal A. </span>
<span id="cb10-3807"><a href="#cb10-3807" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3808"><a href="#cb10-3808" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=8, fig.height=10}</span></span>
<span id="cb10-3809"><a href="#cb10-3809" aria-hidden="true" tabindex="-1"></a>abim_100_df_pca <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(</span>
<span id="cb10-3810"><a href="#cb10-3810" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/validation/abim_100_df_pca.rds"</span></span>
<span id="cb10-3811"><a href="#cb10-3811" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-3812"><a href="#cb10-3812" aria-hidden="true" tabindex="-1"></a>abim_100_df_pca <span class="sc">%&gt;%</span></span>
<span id="cb10-3813"><a href="#cb10-3813" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"abim_100"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-3814"><a href="#cb10-3814" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">inner_join</span>(</span>
<span id="cb10-3815"><a href="#cb10-3815" aria-hidden="true" tabindex="-1"></a>        ., </span>
<span id="cb10-3816"><a href="#cb10-3816" aria-hidden="true" tabindex="-1"></a>        abim_coldata <span class="sc">%&gt;%</span></span>
<span id="cb10-3817"><a href="#cb10-3817" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">select</span>(aims_pam50, sample_name),</span>
<span id="cb10-3818"><a href="#cb10-3818" aria-hidden="true" tabindex="-1"></a>        <span class="at">by =</span> <span class="st">"sample_name"</span></span>
<span id="cb10-3819"><a href="#cb10-3819" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb10-3820"><a href="#cb10-3820" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">pam50_scanb =</span> pam50) <span class="sc">%&gt;%</span></span>
<span id="cb10-3821"><a href="#cb10-3821" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(</span>
<span id="cb10-3822"><a href="#cb10-3822" aria-hidden="true" tabindex="-1"></a>        <span class="at">cols =</span> <span class="fu">c</span>(pam50_scanb, aims_pam50),</span>
<span id="cb10-3823"><a href="#cb10-3823" aria-hidden="true" tabindex="-1"></a>        <span class="at">names_to =</span> <span class="st">"pam50_algo"</span>,</span>
<span id="cb10-3824"><a href="#cb10-3824" aria-hidden="true" tabindex="-1"></a>        <span class="at">values_to =</span> <span class="st">"classes"</span></span>
<span id="cb10-3825"><a href="#cb10-3825" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb10-3826"><a href="#cb10-3826" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> PC3, <span class="at">y =</span> PC4, <span class="at">color =</span> classes)) <span class="sc">+</span></span>
<span id="cb10-3827"><a href="#cb10-3827" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb10-3828"><a href="#cb10-3828" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_grid</span>(classes<span class="sc">~</span>pam50_algo) <span class="sc">+</span></span>
<span id="cb10-3829"><a href="#cb10-3829" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">15</span>) <span class="sc">+</span></span>
<span id="cb10-3830"><a href="#cb10-3830" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">get_colors_pam50</span>(</span>
<span id="cb10-3831"><a href="#cb10-3831" aria-hidden="true" tabindex="-1"></a>        df_pca <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(pam50 <span class="sc">%in%</span> scanb<span class="sc">$</span>pam50)</span>
<span id="cb10-3832"><a href="#cb10-3832" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">+</span> </span>
<span id="cb10-3833"><a href="#cb10-3833" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span> </span>
<span id="cb10-3834"><a href="#cb10-3834" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb10-3835"><a href="#cb10-3835" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-3836"><a href="#cb10-3836" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3837"><a href="#cb10-3837" aria-hidden="true" tabindex="-1"></a>The misclassified samples from AIMS they really seem to be on the </span>
<span id="cb10-3838"><a href="#cb10-3838" aria-hidden="true" tabindex="-1"></a>top region together with the luminal A. Some of the normal</span>
<span id="cb10-3839"><a href="#cb10-3839" aria-hidden="true" tabindex="-1"></a>samples are even in the luminal A region. </span>
<span id="cb10-3840"><a href="#cb10-3840" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3841"><a href="#cb10-3841" aria-hidden="true" tabindex="-1"></a>Now we check the PDXs. </span>
<span id="cb10-3842"><a href="#cb10-3842" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3845"><a href="#cb10-3845" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-3846"><a href="#cb10-3846" aria-hidden="true" tabindex="-1"></a>coldata_pdx <span class="ot">&lt;-</span> <span class="fu">colData</span>(pdx_tpm) <span class="sc">%&gt;%</span> data.frame</span>
<span id="cb10-3847"><a href="#cb10-3847" aria-hidden="true" tabindex="-1"></a>coldata_pdx<span class="sc">$</span>aims_pam50 <span class="ot">&lt;-</span> sbt_pdx_AIMS<span class="sc">$</span>subtype <span class="sc">%&gt;%</span> as.vector</span>
<span id="cb10-3848"><a href="#cb10-3848" aria-hidden="true" tabindex="-1"></a>coldata_pdx<span class="sc">$</span>max_prob <span class="ot">&lt;-</span> <span class="fu">apply</span>(sbt_pdx_AIMS<span class="sc">$</span>subtype.proba, <span class="dv">1</span>, max)</span>
<span id="cb10-3849"><a href="#cb10-3849" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3850"><a href="#cb10-3850" aria-hidden="true" tabindex="-1"></a>coldata_pdx <span class="sc">%&gt;%</span></span>
<span id="cb10-3851"><a href="#cb10-3851" aria-hidden="true" tabindex="-1"></a>    janitor<span class="sc">::</span><span class="fu">tabyl</span>(treatment, aims_pam50, pdx)</span>
<span id="cb10-3852"><a href="#cb10-3852" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-3853"><a href="#cb10-3853" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3854"><a href="#cb10-3854" aria-hidden="true" tabindex="-1"></a>These molecular subtyping do not make much sense. We know that by </span>
<span id="cb10-3855"><a href="#cb10-3855" aria-hidden="true" tabindex="-1"></a>giving P4 estrogen signaling actually goes up, so a expected subtype</span>
<span id="cb10-3856"><a href="#cb10-3856" aria-hidden="true" tabindex="-1"></a>change is from normal to either luminal A and B. Moreover, these PDXs</span>
<span id="cb10-3857"><a href="#cb10-3857" aria-hidden="true" tabindex="-1"></a>are in an active state of proliferation due to the MIND model, so</span>
<span id="cb10-3858"><a href="#cb10-3858" aria-hidden="true" tabindex="-1"></a>being in the normal like region is not a good indication. Also this </span>
<span id="cb10-3859"><a href="#cb10-3859" aria-hidden="true" tabindex="-1"></a>results show that almost all PDXs would be normal like which does not </span>
<span id="cb10-3860"><a href="#cb10-3860" aria-hidden="true" tabindex="-1"></a>go in hand with the results we get from the lab.</span>
<span id="cb10-3861"><a href="#cb10-3861" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3862"><a href="#cb10-3862" aria-hidden="true" tabindex="-1"></a>And the probabilities are available below.</span>
<span id="cb10-3863"><a href="#cb10-3863" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3866"><a href="#cb10-3866" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-3867"><a href="#cb10-3867" aria-hidden="true" tabindex="-1"></a>coldata_pdx <span class="sc">%&gt;%</span></span>
<span id="cb10-3868"><a href="#cb10-3868" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> pdx, <span class="at">y =</span> max_prob)) <span class="sc">+</span> </span>
<span id="cb10-3869"><a href="#cb10-3869" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_boxplot</span>(<span class="at">outlier.shape =</span> <span class="cn">NA</span>) <span class="sc">+</span></span>
<span id="cb10-3870"><a href="#cb10-3870" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_jitter</span>() <span class="sc">+</span> </span>
<span id="cb10-3871"><a href="#cb10-3871" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>()</span>
<span id="cb10-3872"><a href="#cb10-3872" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-3873"><a href="#cb10-3873" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3874"><a href="#cb10-3874" aria-hidden="true" tabindex="-1"></a>And another validation is using the normal cohort. Below is the number of</span>
<span id="cb10-3875"><a href="#cb10-3875" aria-hidden="true" tabindex="-1"></a>all predicted molecular subtypes.</span>
<span id="cb10-3876"><a href="#cb10-3876" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3879"><a href="#cb10-3879" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-3880"><a href="#cb10-3880" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(normal_swiss<span class="sc">$</span>aims_pam50)</span>
<span id="cb10-3881"><a href="#cb10-3881" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-3882"><a href="#cb10-3882" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3883"><a href="#cb10-3883" aria-hidden="true" tabindex="-1"></a>All samples are normal-like as expected, since they are normal. So it could</span>
<span id="cb10-3884"><a href="#cb10-3884" aria-hidden="true" tabindex="-1"></a>be that the AIMS algorithm is not suited for PDXs but for clinical samples</span>
<span id="cb10-3885"><a href="#cb10-3885" aria-hidden="true" tabindex="-1"></a>it works quite well.</span>
<span id="cb10-3886"><a href="#cb10-3886" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3887"><a href="#cb10-3887" aria-hidden="true" tabindex="-1"></a><span class="fu">## nanostring signatures</span></span>
<span id="cb10-3888"><a href="#cb10-3888" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3889"><a href="#cb10-3889" aria-hidden="true" tabindex="-1"></a>In this section we try the nanostring panel signatures provided by the</span>
<span id="cb10-3890"><a href="#cb10-3890" aria-hidden="true" tabindex="-1"></a>Breast Cancer 360 test. We will compare some of the signatures</span>
<span id="cb10-3891"><a href="#cb10-3891" aria-hidden="true" tabindex="-1"></a>by using GSVA on the original datasets and by calculating the sum of the</span>
<span id="cb10-3892"><a href="#cb10-3892" aria-hidden="true" tabindex="-1"></a>gene expression levels of the genes when regressing out the 1st, 2nd and</span>
<span id="cb10-3893"><a href="#cb10-3893" aria-hidden="true" tabindex="-1"></a>5th components.</span>
<span id="cb10-3894"><a href="#cb10-3894" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3895"><a href="#cb10-3895" aria-hidden="true" tabindex="-1"></a>The table below shows the number of genes available for each signature</span>
<span id="cb10-3896"><a href="#cb10-3896" aria-hidden="true" tabindex="-1"></a>and each dataset. In general it seems that all signatures have over</span>
<span id="cb10-3897"><a href="#cb10-3897" aria-hidden="true" tabindex="-1"></a>70% of the genes available, a good indication.</span>
<span id="cb10-3898"><a href="#cb10-3898" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3901"><a href="#cb10-3901" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-3902"><a href="#cb10-3902" aria-hidden="true" tabindex="-1"></a>nanostring_panel <span class="ot">&lt;-</span> readxl<span class="sc">::</span><span class="fu">read_excel</span>(</span>
<span id="cb10-3903"><a href="#cb10-3903" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../data/nanostring_panel.xlsx"</span>,</span>
<span id="cb10-3904"><a href="#cb10-3904" aria-hidden="true" tabindex="-1"></a>    <span class="at">sheet =</span> <span class="st">"Annotations"</span>, </span>
<span id="cb10-3905"><a href="#cb10-3905" aria-hidden="true" tabindex="-1"></a>    <span class="at">skip =</span> <span class="dv">1</span></span>
<span id="cb10-3906"><a href="#cb10-3906" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-3907"><a href="#cb10-3907" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>dplyr<span class="sc">::</span><span class="fu">all_of</span>(<span class="fu">c</span>(<span class="st">"Cell Type"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb10-3908"><a href="#cb10-3908" aria-hidden="true" tabindex="-1"></a>    janitor<span class="sc">::</span><span class="fu">clean_names</span>()</span>
<span id="cb10-3909"><a href="#cb10-3909" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3910"><a href="#cb10-3910" aria-hidden="true" tabindex="-1"></a>gene_sets_nanostring <span class="ot">&lt;-</span> <span class="fu">sapply</span>(</span>
<span id="cb10-3911"><a href="#cb10-3911" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(nanostring_panel)[<span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(nanostring_panel)], </span>
<span id="cb10-3912"><a href="#cb10-3912" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(signature){</span>
<span id="cb10-3913"><a href="#cb10-3913" aria-hidden="true" tabindex="-1"></a>        nanostring_panel <span class="sc">%&gt;%</span></span>
<span id="cb10-3914"><a href="#cb10-3914" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!!</span><span class="fu">sym</span>(signature) <span class="sc">==</span> <span class="st">"+"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-3915"><a href="#cb10-3915" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">pull</span>(gene)</span>
<span id="cb10-3916"><a href="#cb10-3916" aria-hidden="true" tabindex="-1"></a>    }, </span>
<span id="cb10-3917"><a href="#cb10-3917" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>, </span>
<span id="cb10-3918"><a href="#cb10-3918" aria-hidden="true" tabindex="-1"></a>    <span class="at">simplify =</span> <span class="cn">FALSE</span></span>
<span id="cb10-3919"><a href="#cb10-3919" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-3920"><a href="#cb10-3920" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3921"><a href="#cb10-3921" aria-hidden="true" tabindex="-1"></a>genes_each_dataset <span class="ot">&lt;-</span> <span class="fu">lapply</span>(datasets, rownames)</span>
<span id="cb10-3922"><a href="#cb10-3922" aria-hidden="true" tabindex="-1"></a>genes_intersection <span class="ot">&lt;-</span> <span class="fu">sapply</span>(</span>
<span id="cb10-3923"><a href="#cb10-3923" aria-hidden="true" tabindex="-1"></a>    <span class="fu">names</span>(gene_sets_nanostring),</span>
<span id="cb10-3924"><a href="#cb10-3924" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(set_name, gene_sets, genes_each_dataset){</span>
<span id="cb10-3925"><a href="#cb10-3925" aria-hidden="true" tabindex="-1"></a>        <span class="fu">lapply</span>(</span>
<span id="cb10-3926"><a href="#cb10-3926" aria-hidden="true" tabindex="-1"></a>            genes_each_dataset,</span>
<span id="cb10-3927"><a href="#cb10-3927" aria-hidden="true" tabindex="-1"></a>            intersect,</span>
<span id="cb10-3928"><a href="#cb10-3928" aria-hidden="true" tabindex="-1"></a>            <span class="at">y =</span> gene_sets[[set_name]]</span>
<span id="cb10-3929"><a href="#cb10-3929" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb10-3930"><a href="#cb10-3930" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb10-3931"><a href="#cb10-3931" aria-hidden="true" tabindex="-1"></a>    <span class="at">gene_sets =</span> gene_sets_nanostring,</span>
<span id="cb10-3932"><a href="#cb10-3932" aria-hidden="true" tabindex="-1"></a>    <span class="at">genes_each_dataset =</span> genes_each_dataset,</span>
<span id="cb10-3933"><a href="#cb10-3933" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>,</span>
<span id="cb10-3934"><a href="#cb10-3934" aria-hidden="true" tabindex="-1"></a>    <span class="at">simplify =</span> <span class="cn">FALSE</span></span>
<span id="cb10-3935"><a href="#cb10-3935" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-3936"><a href="#cb10-3936" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3937"><a href="#cb10-3937" aria-hidden="true" tabindex="-1"></a><span class="fu">sapply</span>(genes_intersection, <span class="cf">function</span>(x) <span class="fu">sapply</span>(x, length)) <span class="sc">%&gt;%</span> t <span class="sc">%&gt;%</span></span>
<span id="cb10-3938"><a href="#cb10-3938" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span></span>
<span id="cb10-3939"><a href="#cb10-3939" aria-hidden="true" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"pathway"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-3940"><a href="#cb10-3940" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">inner_join</span>(</span>
<span id="cb10-3941"><a href="#cb10-3941" aria-hidden="true" tabindex="-1"></a>        .,</span>
<span id="cb10-3942"><a href="#cb10-3942" aria-hidden="true" tabindex="-1"></a>        <span class="fu">lapply</span>(gene_sets_nanostring, data.frame) <span class="sc">%&gt;%</span> </span>
<span id="cb10-3943"><a href="#cb10-3943" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(<span class="at">.id =</span> <span class="st">"gs_name"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-3944"><a href="#cb10-3944" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">group_by</span>(gs_name) <span class="sc">%&gt;%</span></span>
<span id="cb10-3945"><a href="#cb10-3945" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">n =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb10-3946"><a href="#cb10-3946" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">pathway =</span> gs_name),</span>
<span id="cb10-3947"><a href="#cb10-3947" aria-hidden="true" tabindex="-1"></a>        <span class="at">by =</span> <span class="st">"pathway"</span></span>
<span id="cb10-3948"><a href="#cb10-3948" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span> </span>
<span id="cb10-3949"><a href="#cb10-3949" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb10-3950"><a href="#cb10-3950" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">average_percentage =</span> <span class="fu">format</span>(<span class="fu">mean</span>(</span>
<span id="cb10-3951"><a href="#cb10-3951" aria-hidden="true" tabindex="-1"></a>            <span class="fu">c</span>(tcga<span class="sc">/</span>n, scanb<span class="sc">/</span>n, metabric<span class="sc">/</span>n, poetic<span class="sc">/</span>n)</span>
<span id="cb10-3952"><a href="#cb10-3952" aria-hidden="true" tabindex="-1"></a>        ),</span>
<span id="cb10-3953"><a href="#cb10-3953" aria-hidden="true" tabindex="-1"></a>        <span class="at">digits =</span> <span class="dv">2</span></span>
<span id="cb10-3954"><a href="#cb10-3954" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">%&gt;%</span> </span>
<span id="cb10-3955"><a href="#cb10-3955" aria-hidden="true" tabindex="-1"></a>    kableExtra<span class="sc">::</span><span class="fu">kbl</span>() <span class="sc">%&gt;%</span></span>
<span id="cb10-3956"><a href="#cb10-3956" aria-hidden="true" tabindex="-1"></a>    kableExtra<span class="sc">::</span><span class="fu">kable_classic</span>(<span class="at">full_width =</span> <span class="cn">FALSE</span>)</span>
<span id="cb10-3957"><a href="#cb10-3957" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-3958"><a href="#cb10-3958" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3959"><a href="#cb10-3959" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = first_run}</span></span>
<span id="cb10-3960"><a href="#cb10-3960" aria-hidden="true" tabindex="-1"></a>gsva_scores_nanostring <span class="ot">&lt;-</span> <span class="fu">mapply</span>(</span>
<span id="cb10-3961"><a href="#cb10-3961" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(sum_exp, which_assay, gene_sets){</span>
<span id="cb10-3962"><a href="#cb10-3962" aria-hidden="true" tabindex="-1"></a>        GSVA<span class="sc">::</span><span class="fu">gsva</span>(</span>
<span id="cb10-3963"><a href="#cb10-3963" aria-hidden="true" tabindex="-1"></a>            <span class="at">expr =</span> <span class="fu">as.matrix</span>(<span class="fu">assay</span>(sum_exp, which_assay)),</span>
<span id="cb10-3964"><a href="#cb10-3964" aria-hidden="true" tabindex="-1"></a>            <span class="at">gset.idx.list =</span> gene_sets, </span>
<span id="cb10-3965"><a href="#cb10-3965" aria-hidden="true" tabindex="-1"></a>            <span class="at">parallel.sz =</span> <span class="dv">10</span></span>
<span id="cb10-3966"><a href="#cb10-3966" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb10-3967"><a href="#cb10-3967" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb10-3968"><a href="#cb10-3968" aria-hidden="true" tabindex="-1"></a>    <span class="at">sum_exp =</span> datasets,</span>
<span id="cb10-3969"><a href="#cb10-3969" aria-hidden="true" tabindex="-1"></a>    <span class="at">which_assay =</span> <span class="fu">c</span>(which_exp, <span class="st">"poetic"</span> <span class="ot">=</span> <span class="st">"normalized_intensity"</span>),</span>
<span id="cb10-3970"><a href="#cb10-3970" aria-hidden="true" tabindex="-1"></a>    <span class="at">MoreArgs =</span> <span class="fu">list</span>(<span class="at">gene_sets =</span> gene_sets_nanostring),</span>
<span id="cb10-3971"><a href="#cb10-3971" aria-hidden="true" tabindex="-1"></a>    <span class="at">SIMPLIFY =</span> <span class="cn">FALSE</span>,</span>
<span id="cb10-3972"><a href="#cb10-3972" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span></span>
<span id="cb10-3973"><a href="#cb10-3973" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-3974"><a href="#cb10-3974" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3975"><a href="#cb10-3975" aria-hidden="true" tabindex="-1"></a>datasets_nanostring <span class="ot">&lt;-</span> <span class="fu">mapply</span>(</span>
<span id="cb10-3976"><a href="#cb10-3976" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(gsva_score, dataset){</span>
<span id="cb10-3977"><a href="#cb10-3977" aria-hidden="true" tabindex="-1"></a>        <span class="fu">colData</span>(dataset)[, <span class="fu">rownames</span>(gsva_score)] <span class="ot">&lt;-</span> <span class="fu">t</span>(gsva_score)</span>
<span id="cb10-3978"><a href="#cb10-3978" aria-hidden="true" tabindex="-1"></a>        dataset</span>
<span id="cb10-3979"><a href="#cb10-3979" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb10-3980"><a href="#cb10-3980" aria-hidden="true" tabindex="-1"></a>    gsva_scores_nanostring,</span>
<span id="cb10-3981"><a href="#cb10-3981" aria-hidden="true" tabindex="-1"></a>    datasets,</span>
<span id="cb10-3982"><a href="#cb10-3982" aria-hidden="true" tabindex="-1"></a>    <span class="at">USE.NAMES =</span> <span class="cn">TRUE</span>,</span>
<span id="cb10-3983"><a href="#cb10-3983" aria-hidden="true" tabindex="-1"></a>    <span class="at">SIMPLIFY =</span> <span class="cn">FALSE</span></span>
<span id="cb10-3984"><a href="#cb10-3984" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-3985"><a href="#cb10-3985" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3986"><a href="#cb10-3986" aria-hidden="true" tabindex="-1"></a>scores_nanostring <span class="ot">&lt;-</span> <span class="fu">sapply</span>(</span>
<span id="cb10-3987"><a href="#cb10-3987" aria-hidden="true" tabindex="-1"></a>    gene_sets_nanostring,</span>
<span id="cb10-3988"><a href="#cb10-3988" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(gene_set, df_pcs_regressed){</span>
<span id="cb10-3989"><a href="#cb10-3989" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb10-3990"><a href="#cb10-3990" aria-hidden="true" tabindex="-1"></a>        gene_set <span class="ot">&lt;-</span> <span class="fu">intersect</span>(</span>
<span id="cb10-3991"><a href="#cb10-3991" aria-hidden="true" tabindex="-1"></a>            gene_set,</span>
<span id="cb10-3992"><a href="#cb10-3992" aria-hidden="true" tabindex="-1"></a>            <span class="fu">rownames</span>(df_pcs_regressed)</span>
<span id="cb10-3993"><a href="#cb10-3993" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb10-3994"><a href="#cb10-3994" aria-hidden="true" tabindex="-1"></a>        <span class="fu">colSums</span>(df_pcs_regressed[gene_set, ])</span>
<span id="cb10-3995"><a href="#cb10-3995" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb10-3996"><a href="#cb10-3996" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb10-3997"><a href="#cb10-3997" aria-hidden="true" tabindex="-1"></a>    <span class="at">df_pcs_regressed =</span> df_pcs_regressed</span>
<span id="cb10-3998"><a href="#cb10-3998" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-3999"><a href="#cb10-3999" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span></span>
<span id="cb10-4000"><a href="#cb10-4000" aria-hidden="true" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"sample_name"</span>)</span>
<span id="cb10-4001"><a href="#cb10-4001" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4002"><a href="#cb10-4002" aria-hidden="true" tabindex="-1"></a>scores_nanostring <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">inner_join</span>(</span>
<span id="cb10-4003"><a href="#cb10-4003" aria-hidden="true" tabindex="-1"></a>    scores,</span>
<span id="cb10-4004"><a href="#cb10-4004" aria-hidden="true" tabindex="-1"></a>    df_pca,</span>
<span id="cb10-4005"><a href="#cb10-4005" aria-hidden="true" tabindex="-1"></a>    <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"sample_name"</span>)</span>
<span id="cb10-4006"><a href="#cb10-4006" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-4007"><a href="#cb10-4007" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4008"><a href="#cb10-4008" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(scores_nanostring) <span class="ot">&lt;-</span> scores_nanostring<span class="sc">$</span>sample_name</span>
<span id="cb10-4009"><a href="#cb10-4009" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4010"><a href="#cb10-4010" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb10-4011"><a href="#cb10-4011" aria-hidden="true" tabindex="-1"></a>    scores_nanostring,</span>
<span id="cb10-4012"><a href="#cb10-4012" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/trying/scores_nanostring_only.rds"</span></span>
<span id="cb10-4013"><a href="#cb10-4013" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-4014"><a href="#cb10-4014" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4015"><a href="#cb10-4015" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb10-4016"><a href="#cb10-4016" aria-hidden="true" tabindex="-1"></a>    gsva_scores_nanostring, </span>
<span id="cb10-4017"><a href="#cb10-4017" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/trying/gsva_scores_nanostring.rds"</span></span>
<span id="cb10-4018"><a href="#cb10-4018" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-4019"><a href="#cb10-4019" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4020"><a href="#cb10-4020" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(</span>
<span id="cb10-4021"><a href="#cb10-4021" aria-hidden="true" tabindex="-1"></a>    datasets_nanostring,</span>
<span id="cb10-4022"><a href="#cb10-4022" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../results/rds_files/trying/datasets_with_scores_nanostring.rds"</span></span>
<span id="cb10-4023"><a href="#cb10-4023" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-4024"><a href="#cb10-4024" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-4025"><a href="#cb10-4025" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4026"><a href="#cb10-4026" aria-hidden="true" tabindex="-1"></a>We now compare the scores obtained by GSVA on the datasets and </span>
<span id="cb10-4027"><a href="#cb10-4027" aria-hidden="true" tabindex="-1"></a>see how they relate with nanostring scores obtained with the</span>
<span id="cb10-4028"><a href="#cb10-4028" aria-hidden="true" tabindex="-1"></a>regressed data. </span>
<span id="cb10-4029"><a href="#cb10-4029" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4030"><a href="#cb10-4030" aria-hidden="true" tabindex="-1"></a>First we compare GSVA SET ER/PR with ER signaling obtained by the</span>
<span id="cb10-4031"><a href="#cb10-4031" aria-hidden="true" tabindex="-1"></a>scoring procedure when regressing the data.</span>
<span id="cb10-4032"><a href="#cb10-4032" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4035"><a href="#cb10-4035" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-4036"><a href="#cb10-4036" aria-hidden="true" tabindex="-1"></a>scores_nanostring <span class="sc">%&gt;%</span></span>
<span id="cb10-4037"><a href="#cb10-4037" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(</span>
<span id="cb10-4038"><a href="#cb10-4038" aria-hidden="true" tabindex="-1"></a>        <span class="fu">aes</span>(</span>
<span id="cb10-4039"><a href="#cb10-4039" aria-hidden="true" tabindex="-1"></a>            <span class="at">x =</span> er_signaling, </span>
<span id="cb10-4040"><a href="#cb10-4040" aria-hidden="true" tabindex="-1"></a>            <span class="at">y =</span> SET_ERPR,</span>
<span id="cb10-4041"><a href="#cb10-4041" aria-hidden="true" tabindex="-1"></a>            <span class="at">color =</span> pam50</span>
<span id="cb10-4042"><a href="#cb10-4042" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb10-4043"><a href="#cb10-4043" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb10-4044"><a href="#cb10-4044" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb10-4045"><a href="#cb10-4045" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb10-4046"><a href="#cb10-4046" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>cohort)</span>
<span id="cb10-4047"><a href="#cb10-4047" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-4048"><a href="#cb10-4048" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4049"><a href="#cb10-4049" aria-hidden="true" tabindex="-1"></a>Next we compare EMT, G2M with proliferation and PI3K signaling. </span>
<span id="cb10-4050"><a href="#cb10-4050" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4053"><a href="#cb10-4053" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-4054"><a href="#cb10-4054" aria-hidden="true" tabindex="-1"></a>scores_nanostring <span class="sc">%&gt;%</span></span>
<span id="cb10-4055"><a href="#cb10-4055" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(</span>
<span id="cb10-4056"><a href="#cb10-4056" aria-hidden="true" tabindex="-1"></a>        <span class="fu">aes</span>(</span>
<span id="cb10-4057"><a href="#cb10-4057" aria-hidden="true" tabindex="-1"></a>            <span class="at">x =</span> emt, </span>
<span id="cb10-4058"><a href="#cb10-4058" aria-hidden="true" tabindex="-1"></a>            <span class="at">y =</span> HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION,</span>
<span id="cb10-4059"><a href="#cb10-4059" aria-hidden="true" tabindex="-1"></a>            <span class="at">color =</span> pam50</span>
<span id="cb10-4060"><a href="#cb10-4060" aria-hidden="true" tabindex="-1"></a>        )) <span class="sc">+</span></span>
<span id="cb10-4061"><a href="#cb10-4061" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb10-4062"><a href="#cb10-4062" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb10-4063"><a href="#cb10-4063" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>cohort)</span>
<span id="cb10-4064"><a href="#cb10-4064" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-4065"><a href="#cb10-4065" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4068"><a href="#cb10-4068" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-4069"><a href="#cb10-4069" aria-hidden="true" tabindex="-1"></a>scores_nanostring <span class="sc">%&gt;%</span></span>
<span id="cb10-4070"><a href="#cb10-4070" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> proliferation, <span class="at">y =</span> HALLMARK_G2M_CHECKPOINT,</span>
<span id="cb10-4071"><a href="#cb10-4071" aria-hidden="true" tabindex="-1"></a>                        <span class="at">color =</span> pam50)) <span class="sc">+</span></span>
<span id="cb10-4072"><a href="#cb10-4072" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb10-4073"><a href="#cb10-4073" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb10-4074"><a href="#cb10-4074" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>cohort)</span>
<span id="cb10-4075"><a href="#cb10-4075" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-4076"><a href="#cb10-4076" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4079"><a href="#cb10-4079" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-4080"><a href="#cb10-4080" aria-hidden="true" tabindex="-1"></a>scores_nanostring <span class="sc">%&gt;%</span></span>
<span id="cb10-4081"><a href="#cb10-4081" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> pi3k, <span class="at">y =</span> HALLMARK_PI3K_AKT_MTOR_SIGNALING,</span>
<span id="cb10-4082"><a href="#cb10-4082" aria-hidden="true" tabindex="-1"></a>                        <span class="at">color =</span> pam50)) <span class="sc">+</span></span>
<span id="cb10-4083"><a href="#cb10-4083" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb10-4084"><a href="#cb10-4084" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb10-4085"><a href="#cb10-4085" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>cohort)</span>
<span id="cb10-4086"><a href="#cb10-4086" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-4087"><a href="#cb10-4087" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4090"><a href="#cb10-4090" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-4091"><a href="#cb10-4091" aria-hidden="true" tabindex="-1"></a>scores_nanostring <span class="sc">%&gt;%</span></span>
<span id="cb10-4092"><a href="#cb10-4092" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> pam50, <span class="at">y =</span> proliferation, <span class="at">color =</span> pam50)) <span class="sc">+</span></span>
<span id="cb10-4093"><a href="#cb10-4093" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb10-4094"><a href="#cb10-4094" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb10-4095"><a href="#cb10-4095" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>cohort)</span>
<span id="cb10-4096"><a href="#cb10-4096" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-4097"><a href="#cb10-4097" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4098"><a href="#cb10-4098" aria-hidden="true" tabindex="-1"></a>In general there is a correlation but the scores are not so well correlated</span>
<span id="cb10-4099"><a href="#cb10-4099" aria-hidden="true" tabindex="-1"></a>and they are noisy. One that works fairly well is the ER signaling score,</span>
<span id="cb10-4100"><a href="#cb10-4100" aria-hidden="true" tabindex="-1"></a>but the others not so well. The proliferation score barely shows any</span>
<span id="cb10-4101"><a href="#cb10-4101" aria-hidden="true" tabindex="-1"></a>difference between luminal A and B patients in any of the cohorts. </span>
<span id="cb10-4102"><a href="#cb10-4102" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4103"><a href="#cb10-4103" aria-hidden="true" tabindex="-1"></a>Perhaps if we calculate the scores using the hallmarks pathways the scores are</span>
<span id="cb10-4104"><a href="#cb10-4104" aria-hidden="true" tabindex="-1"></a>better correlated, since they are the same gene sets.</span>
<span id="cb10-4105"><a href="#cb10-4105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4106"><a href="#cb10-4106" aria-hidden="true" tabindex="-1"></a>Below we plot the correlation between the proliferation pathway using GSVA and</span>
<span id="cb10-4107"><a href="#cb10-4107" aria-hidden="true" tabindex="-1"></a>the regressed scoring strategy for the TCGA cohort.</span>
<span id="cb10-4108"><a href="#cb10-4108" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4111"><a href="#cb10-4111" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-4112"><a href="#cb10-4112" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(</span>
<span id="cb10-4113"><a href="#cb10-4113" aria-hidden="true" tabindex="-1"></a>    datasets_nanostring<span class="sc">$</span>tcga<span class="sc">$</span>proliferation,</span>
<span id="cb10-4114"><a href="#cb10-4114" aria-hidden="true" tabindex="-1"></a>    scores_nanostring[<span class="fu">colnames</span>(datasets_nanostring<span class="sc">$</span>tcga), <span class="st">"proliferation"</span>]</span>
<span id="cb10-4115"><a href="#cb10-4115" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-4116"><a href="#cb10-4116" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-4117"><a href="#cb10-4117" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4118"><a href="#cb10-4118" aria-hidden="true" tabindex="-1"></a>It is perfectly correlated.</span>
<span id="cb10-4119"><a href="#cb10-4119" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4120"><a href="#cb10-4120" aria-hidden="true" tabindex="-1"></a>And for ER signaling:</span>
<span id="cb10-4121"><a href="#cb10-4121" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4124"><a href="#cb10-4124" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-4125"><a href="#cb10-4125" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(</span>
<span id="cb10-4126"><a href="#cb10-4126" aria-hidden="true" tabindex="-1"></a>    datasets_nanostring<span class="sc">$</span>tcga<span class="sc">$</span>er_signaling,</span>
<span id="cb10-4127"><a href="#cb10-4127" aria-hidden="true" tabindex="-1"></a>    scores_nanostring[<span class="fu">colnames</span>(datasets_nanostring<span class="sc">$</span>tcga), <span class="st">"er_signaling"</span>]</span>
<span id="cb10-4128"><a href="#cb10-4128" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-4129"><a href="#cb10-4129" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-4130"><a href="#cb10-4130" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4131"><a href="#cb10-4131" aria-hidden="true" tabindex="-1"></a>Also it is very much correlated. So the problem in the end is comparing</span>
<span id="cb10-4132"><a href="#cb10-4132" aria-hidden="true" tabindex="-1"></a>the hallmarks signatures with the nanostring. But interestingly </span>
<span id="cb10-4133"><a href="#cb10-4133" aria-hidden="true" tabindex="-1"></a>the nanostring proliferation signature was not good to pick up the differences</span>
<span id="cb10-4134"><a href="#cb10-4134" aria-hidden="true" tabindex="-1"></a>between luminal A and B.</span>
<span id="cb10-4135"><a href="#cb10-4135" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4138"><a href="#cb10-4138" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-4139"><a href="#cb10-4139" aria-hidden="true" tabindex="-1"></a>datasets_nanostring<span class="sc">$</span>tcga <span class="sc">%&gt;%</span></span>
<span id="cb10-4140"><a href="#cb10-4140" aria-hidden="true" tabindex="-1"></a>    colData <span class="sc">%&gt;%</span></span>
<span id="cb10-4141"><a href="#cb10-4141" aria-hidden="true" tabindex="-1"></a>    data.frame <span class="sc">%&gt;%</span></span>
<span id="cb10-4142"><a href="#cb10-4142" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(</span>
<span id="cb10-4143"><a href="#cb10-4143" aria-hidden="true" tabindex="-1"></a>        <span class="at">cols =</span> <span class="fu">c</span>(<span class="st">"proliferation"</span>, <span class="st">"HALLMARK_G2M_CHECKPOINT"</span>),</span>
<span id="cb10-4144"><a href="#cb10-4144" aria-hidden="true" tabindex="-1"></a>        <span class="at">names_to =</span> <span class="st">"pathway"</span>,</span>
<span id="cb10-4145"><a href="#cb10-4145" aria-hidden="true" tabindex="-1"></a>        <span class="at">values_to =</span> <span class="st">"score"</span></span>
<span id="cb10-4146"><a href="#cb10-4146" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb10-4147"><a href="#cb10-4147" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> pam50, <span class="at">y =</span> score, <span class="at">color =</span> pam50)) <span class="sc">+</span> </span>
<span id="cb10-4148"><a href="#cb10-4148" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_boxplot</span>() <span class="sc">+</span> </span>
<span id="cb10-4149"><a href="#cb10-4149" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>pathway, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span> </span>
<span id="cb10-4150"><a href="#cb10-4150" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>()</span>
<span id="cb10-4151"><a href="#cb10-4151" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-4152"><a href="#cb10-4152" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4153"><a href="#cb10-4153" aria-hidden="true" tabindex="-1"></a><span class="fu">## Risk stratification and the molecular landscape</span></span>
<span id="cb10-4154"><a href="#cb10-4154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4155"><a href="#cb10-4155" aria-hidden="true" tabindex="-1"></a>One common problem in the clinics is what to do with patients that are </span>
<span id="cb10-4156"><a href="#cb10-4156" aria-hidden="true" tabindex="-1"></a>in the so called intermediate risk group. Here we use the SSP </span>
<span id="cb10-4157"><a href="#cb10-4157" aria-hidden="true" tabindex="-1"></a>classification from the SCANB team to check what is the position of these</span>
<span id="cb10-4158"><a href="#cb10-4158" aria-hidden="true" tabindex="-1"></a>patients in the molecular landscape.</span>
<span id="cb10-4159"><a href="#cb10-4159" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4162"><a href="#cb10-4162" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-4163"><a href="#cb10-4163" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-scanb-risk-embedding</span></span>
<span id="cb10-4164"><a href="#cb10-4164" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Embedding of the SCANB samples colored by their predicted</span></span>
<span id="cb10-4165"><a href="#cb10-4165" aria-hidden="true" tabindex="-1"></a><span class="co">#|     ROR risk category.</span></span>
<span id="cb10-4166"><a href="#cb10-4166" aria-hidden="true" tabindex="-1"></a>df_pca <span class="sc">%&gt;%</span></span>
<span id="cb10-4167"><a href="#cb10-4167" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"scanb"</span> <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(SSP.ROR.risk.cat)) <span class="sc">%&gt;%</span></span>
<span id="cb10-4168"><a href="#cb10-4168" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">slice_sample</span>(<span class="at">n =</span> <span class="fu">nrow</span>(.)) <span class="sc">%&gt;%</span></span>
<span id="cb10-4169"><a href="#cb10-4169" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">sample_name =</span> <span class="fu">factor</span>(sample_name)) <span class="sc">%&gt;%</span></span>
<span id="cb10-4170"><a href="#cb10-4170" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> PC3, <span class="at">y =</span> PC4, <span class="at">color =</span> SSP.ROR.risk.cat)) <span class="sc">+</span></span>
<span id="cb10-4171"><a href="#cb10-4171" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span> </span>
<span id="cb10-4172"><a href="#cb10-4172" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_color_viridis_d</span>(<span class="at">option =</span> <span class="st">"H"</span>) <span class="sc">+</span> </span>
<span id="cb10-4173"><a href="#cb10-4173" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb10-4174"><a href="#cb10-4174" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span> </span>
<span id="cb10-4175"><a href="#cb10-4175" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb10-4176"><a href="#cb10-4176" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-4177"><a href="#cb10-4177" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4178"><a href="#cb10-4178" aria-hidden="true" tabindex="-1"></a>@fig-scanb-risk-embedding shows that the embedding detects the positions</span>
<span id="cb10-4179"><a href="#cb10-4179" aria-hidden="true" tabindex="-1"></a>where the intermediate risk patients are, namely they are in the</span>
<span id="cb10-4180"><a href="#cb10-4180" aria-hidden="true" tabindex="-1"></a>intersection of the luminal A and B subtypes.</span>
<span id="cb10-4181"><a href="#cb10-4181" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4182"><a href="#cb10-4182" aria-hidden="true" tabindex="-1"></a>And @fig-intermediate-pca-position below shows that there is a shift in</span>
<span id="cb10-4183"><a href="#cb10-4183" aria-hidden="true" tabindex="-1"></a>PC4 for the intermediate to the high risk group, which makes sense as this is</span>
<span id="cb10-4184"><a href="#cb10-4184" aria-hidden="true" tabindex="-1"></a>exactly where the distinction between luminal A and B are.</span>
<span id="cb10-4185"><a href="#cb10-4185" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4186"><a href="#cb10-4186" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.height=4, fig.width=8}</span></span>
<span id="cb10-4187"><a href="#cb10-4187" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-intermediate-pca-position</span></span>
<span id="cb10-4188"><a href="#cb10-4188" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Distribution of the principal components for the different</span></span>
<span id="cb10-4189"><a href="#cb10-4189" aria-hidden="true" tabindex="-1"></a><span class="co">#|     risk groups defined by the SSP ROR scores.</span></span>
<span id="cb10-4190"><a href="#cb10-4190" aria-hidden="true" tabindex="-1"></a>df_pca <span class="sc">%&gt;%</span></span>
<span id="cb10-4191"><a href="#cb10-4191" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">==</span> <span class="st">"scanb"</span> <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(SSP.ROR.risk.cat)) <span class="sc">%&gt;%</span></span>
<span id="cb10-4192"><a href="#cb10-4192" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(</span>
<span id="cb10-4193"><a href="#cb10-4193" aria-hidden="true" tabindex="-1"></a>        <span class="at">cols =</span> <span class="fu">c</span>(PC3, PC4), </span>
<span id="cb10-4194"><a href="#cb10-4194" aria-hidden="true" tabindex="-1"></a>        <span class="at">names_to =</span> <span class="st">"pcs"</span>, </span>
<span id="cb10-4195"><a href="#cb10-4195" aria-hidden="true" tabindex="-1"></a>        <span class="at">values_to =</span> <span class="st">"pc_scores"</span></span>
<span id="cb10-4196"><a href="#cb10-4196" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb10-4197"><a href="#cb10-4197" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">y =</span> SSP.ROR.risk.cat, <span class="at">x =</span> pc_scores, <span class="at">color =</span> SSP.ROR.risk.cat)) <span class="sc">+</span></span>
<span id="cb10-4198"><a href="#cb10-4198" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_jitter</span>(<span class="at">alpha =</span> <span class="fl">0.4</span>, <span class="at">size =</span> <span class="fl">0.1</span>) <span class="sc">+</span> </span>
<span id="cb10-4199"><a href="#cb10-4199" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_violin</span>(<span class="at">alpha =</span> <span class="fl">0.1</span>, <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">linewidth =</span> <span class="fl">0.6</span>) <span class="sc">+</span> </span>
<span id="cb10-4200"><a href="#cb10-4200" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(<span class="sc">~</span>pcs) <span class="sc">+</span></span>
<span id="cb10-4201"><a href="#cb10-4201" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">scale_color_viridis_d</span>(<span class="at">option =</span> <span class="st">"H"</span>) <span class="sc">+</span></span>
<span id="cb10-4202"><a href="#cb10-4202" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"PC"</span>) <span class="sc">+</span> </span>
<span id="cb10-4203"><a href="#cb10-4203" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">20</span>) <span class="sc">+</span> </span>
<span id="cb10-4204"><a href="#cb10-4204" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>) <span class="sc">+</span> </span>
<span id="cb10-4205"><a href="#cb10-4205" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span> </span>
<span id="cb10-4206"><a href="#cb10-4206" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb10-4207"><a href="#cb10-4207" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-4208"><a href="#cb10-4208" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4209"><a href="#cb10-4209" aria-hidden="true" tabindex="-1"></a><span class="fu">## Late distant recurrence: trying to find biomarkers</span></span>
<span id="cb10-4210"><a href="#cb10-4210" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4211"><a href="#cb10-4211" aria-hidden="true" tabindex="-1"></a>One of the key problems in breast cancer research is finding patients that</span>
<span id="cb10-4212"><a href="#cb10-4212" aria-hidden="true" tabindex="-1"></a>will develop late recurrence. Here we use METABRIC as it has a long follow-up</span>
<span id="cb10-4213"><a href="#cb10-4213" aria-hidden="true" tabindex="-1"></a>history. We start by plotting the embedding of all METABRIC patients that</span>
<span id="cb10-4214"><a href="#cb10-4214" aria-hidden="true" tabindex="-1"></a>had a followup longer than 10 years and received only ET. We then try to </span>
<span id="cb10-4215"><a href="#cb10-4215" aria-hidden="true" tabindex="-1"></a>predict which patients recurr or not based on the molecular data by</span>
<span id="cb10-4216"><a href="#cb10-4216" aria-hidden="true" tabindex="-1"></a>using lasso regression. For this we use the package <span class="in">`glmnet`</span> that </span>
<span id="cb10-4217"><a href="#cb10-4217" aria-hidden="true" tabindex="-1"></a>provides such functionality. </span>
<span id="cb10-4218"><a href="#cb10-4218" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4221"><a href="#cb10-4221" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-4222"><a href="#cb10-4222" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">get_base_plot</span>(</span>
<span id="cb10-4223"><a href="#cb10-4223" aria-hidden="true" tabindex="-1"></a>    df_pca <span class="sc">%&gt;%</span> </span>
<span id="cb10-4224"><a href="#cb10-4224" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">filter</span>(cohort <span class="sc">!=</span> <span class="st">"poetic"</span>) </span>
<span id="cb10-4225"><a href="#cb10-4225" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span> </span>
<span id="cb10-4226"><a href="#cb10-4226" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb10-4227"><a href="#cb10-4227" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4228"><a href="#cb10-4228" aria-hidden="true" tabindex="-1"></a>et_metabric_late_data <span class="ot">&lt;-</span> df_pca <span class="sc">%&gt;%</span></span>
<span id="cb10-4229"><a href="#cb10-4229" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb10-4230"><a href="#cb10-4230" aria-hidden="true" tabindex="-1"></a>        cohort <span class="sc">==</span> <span class="st">"metabric"</span> <span class="sc">&amp;</span></span>
<span id="cb10-4231"><a href="#cb10-4231" aria-hidden="true" tabindex="-1"></a>            rfs_months <span class="sc">&gt;=</span> <span class="dv">10</span> <span class="sc">*</span> <span class="dv">12</span> <span class="sc">&amp;</span> </span>
<span id="cb10-4232"><a href="#cb10-4232" aria-hidden="true" tabindex="-1"></a>            <span class="fu">is.numeric</span>(rfs_status) <span class="sc">&amp;</span> </span>
<span id="cb10-4233"><a href="#cb10-4233" aria-hidden="true" tabindex="-1"></a>            HORMONE_THERAPY <span class="sc">==</span> <span class="st">"YES"</span> <span class="sc">&amp;</span> </span>
<span id="cb10-4234"><a href="#cb10-4234" aria-hidden="true" tabindex="-1"></a>            er_status <span class="sc">==</span> <span class="st">"pos"</span></span>
<span id="cb10-4235"><a href="#cb10-4235" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb10-4236"><a href="#cb10-4236" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">is_there_recurrence =</span> <span class="fu">ifelse</span>(</span>
<span id="cb10-4237"><a href="#cb10-4237" aria-hidden="true" tabindex="-1"></a>        rfs_status <span class="sc">==</span> <span class="dv">2</span>,</span>
<span id="cb10-4238"><a href="#cb10-4238" aria-hidden="true" tabindex="-1"></a>        <span class="st">"yes"</span>,</span>
<span id="cb10-4239"><a href="#cb10-4239" aria-hidden="true" tabindex="-1"></a>        <span class="st">"no"</span></span>
<span id="cb10-4240"><a href="#cb10-4240" aria-hidden="true" tabindex="-1"></a>    ))</span>
<span id="cb10-4241"><a href="#cb10-4241" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4242"><a href="#cb10-4242" aria-hidden="true" tabindex="-1"></a>ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(</span>
<span id="cb10-4243"><a href="#cb10-4243" aria-hidden="true" tabindex="-1"></a>    et_metabric_late_data,</span>
<span id="cb10-4244"><a href="#cb10-4244" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(</span>
<span id="cb10-4245"><a href="#cb10-4245" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> PC3, </span>
<span id="cb10-4246"><a href="#cb10-4246" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> PC4, </span>
<span id="cb10-4247"><a href="#cb10-4247" aria-hidden="true" tabindex="-1"></a>        <span class="at">shape =</span> is_there_recurrence, </span>
<span id="cb10-4248"><a href="#cb10-4248" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> is_there_recurrence</span>
<span id="cb10-4249"><a href="#cb10-4249" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb10-4250"><a href="#cb10-4250" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span> </span>
<span id="cb10-4251"><a href="#cb10-4251" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_point</span>(</span>
<span id="cb10-4252"><a href="#cb10-4252" aria-hidden="true" tabindex="-1"></a>        <span class="at">size =</span> <span class="dv">4</span>,</span>
<span id="cb10-4253"><a href="#cb10-4253" aria-hidden="true" tabindex="-1"></a>        <span class="at">alpha =</span> <span class="fl">0.5</span></span>
<span id="cb10-4254"><a href="#cb10-4254" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb10-4255"><a href="#cb10-4255" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">15</span>) <span class="sc">+</span></span>
<span id="cb10-4256"><a href="#cb10-4256" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb10-4257"><a href="#cb10-4257" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">"METABRIC"</span>,</span>
<span id="cb10-4258"><a href="#cb10-4258" aria-hidden="true" tabindex="-1"></a>        <span class="at">subtitle =</span> <span class="st">"ER+ BC, ET only, over 10 years of RFS follow-up"</span>,</span>
<span id="cb10-4259"><a href="#cb10-4259" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">"Recurrence?"</span>,</span>
<span id="cb10-4260"><a href="#cb10-4260" aria-hidden="true" tabindex="-1"></a>        <span class="at">shape =</span> <span class="st">"Recurrence?"</span></span>
<span id="cb10-4261"><a href="#cb10-4261" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb10-4262"><a href="#cb10-4262" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">coord_cartesian</span>(</span>
<span id="cb10-4263"><a href="#cb10-4263" aria-hidden="true" tabindex="-1"></a>        <span class="at">xlim =</span> <span class="fu">c</span>(<span class="fu">min</span>(df_pca<span class="sc">$</span>PC3), <span class="fu">max</span>(df_pca<span class="sc">$</span>PC3)),</span>
<span id="cb10-4264"><a href="#cb10-4264" aria-hidden="true" tabindex="-1"></a>        <span class="at">ylim =</span> <span class="fu">c</span>(<span class="fu">min</span>(df_pca<span class="sc">$</span>PC4), <span class="fu">max</span>(df_pca<span class="sc">$</span>PC4))</span>
<span id="cb10-4265"><a href="#cb10-4265" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb10-4266"><a href="#cb10-4266" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_plot_aes_point</span>() <span class="sc">+</span></span>
<span id="cb10-4267"><a href="#cb10-4267" aria-hidden="true" tabindex="-1"></a>    <span class="fu">change_guides_point</span>()</span>
<span id="cb10-4268"><a href="#cb10-4268" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-4269"><a href="#cb10-4269" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4270"><a href="#cb10-4270" aria-hidden="true" tabindex="-1"></a>In general there is no distinction in terms of the molecular</span>
<span id="cb10-4271"><a href="#cb10-4271" aria-hidden="true" tabindex="-1"></a>landscape among the patients that recurred, both in terms of</span>
<span id="cb10-4272"><a href="#cb10-4272" aria-hidden="true" tabindex="-1"></a>molecular subtype and position in the molecular landscape. </span>
<span id="cb10-4273"><a href="#cb10-4273" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4274"><a href="#cb10-4274" aria-hidden="true" tabindex="-1"></a>We now compare their molecular scores. </span>
<span id="cb10-4275"><a href="#cb10-4275" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4276"><a href="#cb10-4276" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, fig.width=20, fig.height=40}</span></span>
<span id="cb10-4277"><a href="#cb10-4277" aria-hidden="true" tabindex="-1"></a>pathways_to_compare <span class="ot">&lt;-</span> <span class="fu">colnames</span>(df_pca)[<span class="fu">grepl</span>(<span class="st">"^HALLMARK"</span>, <span class="fu">colnames</span>(df_pca))]</span>
<span id="cb10-4278"><a href="#cb10-4278" aria-hidden="true" tabindex="-1"></a>et_metabric_late_data <span class="sc">%&gt;%</span></span>
<span id="cb10-4279"><a href="#cb10-4279" aria-hidden="true" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(</span>
<span id="cb10-4280"><a href="#cb10-4280" aria-hidden="true" tabindex="-1"></a>        <span class="at">cols =</span> dplyr<span class="sc">::</span><span class="fu">all_of</span>(pathways_to_compare),</span>
<span id="cb10-4281"><a href="#cb10-4281" aria-hidden="true" tabindex="-1"></a>        <span class="at">names_to =</span> <span class="st">"pathway"</span>,</span>
<span id="cb10-4282"><a href="#cb10-4282" aria-hidden="true" tabindex="-1"></a>        <span class="at">values_to =</span> <span class="st">"score"</span></span>
<span id="cb10-4283"><a href="#cb10-4283" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb10-4284"><a href="#cb10-4284" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> is_there_recurrence, <span class="at">y =</span> score)) <span class="sc">+</span> </span>
<span id="cb10-4285"><a href="#cb10-4285" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_jitter</span>(<span class="at">size =</span> <span class="fl">0.1</span>) <span class="sc">+</span> </span>
<span id="cb10-4286"><a href="#cb10-4286" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">geom_boxplot</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="at">outlier.shape =</span> <span class="cn">NA</span>) <span class="sc">+</span> </span>
<span id="cb10-4287"><a href="#cb10-4287" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">facet_wrap</span>(</span>
<span id="cb10-4288"><a href="#cb10-4288" aria-hidden="true" tabindex="-1"></a>        <span class="sc">~</span>pathway, </span>
<span id="cb10-4289"><a href="#cb10-4289" aria-hidden="true" tabindex="-1"></a>        <span class="at">ncol =</span> <span class="dv">5</span>,</span>
<span id="cb10-4290"><a href="#cb10-4290" aria-hidden="true" tabindex="-1"></a>        <span class="at">labeller =</span> <span class="fu">as_labeller</span>(</span>
<span id="cb10-4291"><a href="#cb10-4291" aria-hidden="true" tabindex="-1"></a>            stringr<span class="sc">::</span><span class="fu">str_replace_all</span>(pathways_to_compare, <span class="st">"^HALLMARK_"</span>, <span class="st">""</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-4292"><a href="#cb10-4292" aria-hidden="true" tabindex="-1"></a>                <span class="st">`</span><span class="at">names&lt;-</span><span class="st">`</span>(pathways_to_compare)</span>
<span id="cb10-4293"><a href="#cb10-4293" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb10-4294"><a href="#cb10-4294" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb10-4295"><a href="#cb10-4295" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">labs</span>(</span>
<span id="cb10-4296"><a href="#cb10-4296" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">"Recurrence?"</span>,</span>
<span id="cb10-4297"><a href="#cb10-4297" aria-hidden="true" tabindex="-1"></a>        <span class="at">title =</span> <span class="st">"METABRIC"</span>,</span>
<span id="cb10-4298"><a href="#cb10-4298" aria-hidden="true" tabindex="-1"></a>        <span class="at">subtitle =</span> <span class="st">"ER+ BC, ET only, over 10 years of RFS follow-up"</span>    </span>
<span id="cb10-4299"><a href="#cb10-4299" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span> </span>
<span id="cb10-4300"><a href="#cb10-4300" aria-hidden="true" tabindex="-1"></a>    ggplot2<span class="sc">::</span><span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">15</span>)</span>
<span id="cb10-4301"><a href="#cb10-4301" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-4302"><a href="#cb10-4302" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb10-4303"><a href="#cb10-4303" aria-hidden="true" tabindex="-1"></a>Even here we don't see much difference. Unless there is some kind of </span>
<span id="cb10-4304"><a href="#cb10-4304" aria-hidden="true" tabindex="-1"></a>linear combination of pathways, I don't think it would be possible</span>
<span id="cb10-4305"><a href="#cb10-4305" aria-hidden="true" tabindex="-1"></a>to classify these two groups by using only transcriptomic data</span>
<span id="cb10-4306"><a href="#cb10-4306" aria-hidden="true" tabindex="-1"></a>and these pathways. In any case we proceed now with the </span>
<span id="cb10-4307"><a href="#cb10-4307" aria-hidden="true" tabindex="-1"></a>lasso regression using the 50 pathways. </span>
<span id="cb10-4308"><a href="#cb10-4308" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4311"><a href="#cb10-4311" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-4312"><a href="#cb10-4312" aria-hidden="true" tabindex="-1"></a>fit_recurrence <span class="ot">&lt;-</span> glmnet<span class="sc">::</span><span class="fu">cv.glmnet</span>(</span>
<span id="cb10-4313"><a href="#cb10-4313" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="fu">as.matrix</span>(et_metabric_late_data[, pathways_to_compare]),</span>
<span id="cb10-4314"><a href="#cb10-4314" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> et_metabric_late_data<span class="sc">$</span>is_there_recurrence,</span>
<span id="cb10-4315"><a href="#cb10-4315" aria-hidden="true" tabindex="-1"></a>    <span class="at">family =</span> <span class="st">"binomial"</span>,</span>
<span id="cb10-4316"><a href="#cb10-4316" aria-hidden="true" tabindex="-1"></a>    <span class="at">type.measure =</span> <span class="st">"class"</span></span>
<span id="cb10-4317"><a href="#cb10-4317" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-4318"><a href="#cb10-4318" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4319"><a href="#cb10-4319" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(fit_recurrence)</span>
<span id="cb10-4320"><a href="#cb10-4320" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-4321"><a href="#cb10-4321" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4322"><a href="#cb10-4322" aria-hidden="true" tabindex="-1"></a>We see that actually it does not work. And when we predict all the </span>
<span id="cb10-4323"><a href="#cb10-4323" aria-hidden="true" tabindex="-1"></a>samples are predicted as there is no recurrence.</span>
<span id="cb10-4324"><a href="#cb10-4324" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4327"><a href="#cb10-4327" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-4328"><a href="#cb10-4328" aria-hidden="true" tabindex="-1"></a><span class="fu">predict</span>(</span>
<span id="cb10-4329"><a href="#cb10-4329" aria-hidden="true" tabindex="-1"></a>    fit_recurrence, </span>
<span id="cb10-4330"><a href="#cb10-4330" aria-hidden="true" tabindex="-1"></a>    <span class="at">new =</span> et_metabric_late_data[, pathways_to_compare] <span class="sc">%&gt;%</span></span>
<span id="cb10-4331"><a href="#cb10-4331" aria-hidden="true" tabindex="-1"></a>            as.matrix,</span>
<span id="cb10-4332"><a href="#cb10-4332" aria-hidden="true" tabindex="-1"></a>    <span class="at">s =</span> <span class="st">"lambda.1se"</span>,</span>
<span id="cb10-4333"><a href="#cb10-4333" aria-hidden="true" tabindex="-1"></a>    <span class="at">type =</span> <span class="st">"class"</span></span>
<span id="cb10-4334"><a href="#cb10-4334" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> table</span>
<span id="cb10-4335"><a href="#cb10-4335" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-4336"><a href="#cb10-4336" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4337"><a href="#cb10-4337" aria-hidden="true" tabindex="-1"></a>This is probably because of the unbalanced dataset. Next we </span>
<span id="cb10-4338"><a href="#cb10-4338" aria-hidden="true" tabindex="-1"></a>try with random forest as well.</span>
<span id="cb10-4339"><a href="#cb10-4339" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4342"><a href="#cb10-4342" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb10-4343"><a href="#cb10-4343" aria-hidden="true" tabindex="-1"></a>rf <span class="ot">&lt;-</span> randomForest<span class="sc">::</span><span class="fu">randomForest</span>(</span>
<span id="cb10-4344"><a href="#cb10-4344" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.factor</span>(is_there_recurrence) <span class="sc">~</span> ., </span>
<span id="cb10-4345"><a href="#cb10-4345" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> et_metabric_late_data[, <span class="fu">c</span>(</span>
<span id="cb10-4346"><a href="#cb10-4346" aria-hidden="true" tabindex="-1"></a>        <span class="st">"is_there_recurrence"</span>,</span>
<span id="cb10-4347"><a href="#cb10-4347" aria-hidden="true" tabindex="-1"></a>        pathways_to_compare,</span>
<span id="cb10-4348"><a href="#cb10-4348" aria-hidden="true" tabindex="-1"></a>        <span class="st">"AGE_AT_DIAGNOSIS"</span>,</span>
<span id="cb10-4349"><a href="#cb10-4349" aria-hidden="true" tabindex="-1"></a>        <span class="st">"LYMPH_NODES_EXAMINED_POSITIVE"</span>, </span>
<span id="cb10-4350"><a href="#cb10-4350" aria-hidden="true" tabindex="-1"></a>        <span class="st">"npi"</span></span>
<span id="cb10-4351"><a href="#cb10-4351" aria-hidden="true" tabindex="-1"></a>    )],</span>
<span id="cb10-4352"><a href="#cb10-4352" aria-hidden="true" tabindex="-1"></a>    <span class="at">ntree =</span> <span class="dv">500</span></span>
<span id="cb10-4353"><a href="#cb10-4353" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-4354"><a href="#cb10-4354" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4355"><a href="#cb10-4355" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(rf)</span>
<span id="cb10-4356"><a href="#cb10-4356" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-4357"><a href="#cb10-4357" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4358"><a href="#cb10-4358" aria-hidden="true" tabindex="-1"></a>That is not the case, still we can't predict if a patient will develop</span>
<span id="cb10-4359"><a href="#cb10-4359" aria-hidden="true" tabindex="-1"></a>recurrence or not in this cohort. </span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-right">
      <ul class="footer-items list-unstyled">
    <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/chronchi">
      <i class="bi bi-github" role="img">
</i> 
    </a>
  </li>  
    <li class="nav-item compact">
    <a class="nav-link" href="https://twitter.com/chronchi">
      <i class="bi bi-twitter" role="img">
</i> 
    </a>
  </li>  
</ul>
    </div>
  </div>
</footer>



</body></html>