<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.1.175">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Carlos Ronchi, Cathrin Brisken">

<title>Molecular landscape: a new framework for personalized medicine in Breast Cancer</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>


<script src="main_files/libs/clipboard/clipboard.min.js"></script>
<script src="main_files/libs/quarto-html/quarto.js"></script>
<script src="main_files/libs/quarto-html/popper.min.js"></script>
<script src="main_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="main_files/libs/quarto-html/anchor.min.js"></script>
<link href="main_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="main_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="main_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="main_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="main_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">

  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title"><p>Molecular landscape: a new framework for personalized medicine in Breast Cancer</p></h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Carlos Ronchi, Cathrin Brisken </p>
          </div>
  </div>
    
    
  </div>
  

</header>

<section id="abstract" class="level1">
<h1>Abstract</h1>
</section>
<section id="introduction" class="level1">
<h1>Introduction</h1>
<p>Breast cancer is the most commonly diagnose tumor disease in the world <span class="citation" data-cites="whocancer">(<a href="#ref-whocancer" role="doc-biblioref">1</a>)</span>. It is also a very heterogeneous disease that can be subdivided in different subtypes. Patients that are diagnosed with breast cancer are classified based on the expression of the estrogen, progesterone receptors and HER2 membrane protein. The estrogen receptor positive (ER+) breast cancer subtype is the most frequent found in clinical practice <span class="citation" data-cites="Anderson2002">(<a href="#ref-Anderson2002" role="doc-biblioref">2</a>)</span>. Tumor cells of this subtype contain the estrogen receptor whose main mechanism is the target of selective estrogen receptor modulator (SERM), selective estrogen receptor degrader (SERD) and aromatase inhibitors (AI). The degree of the expression of the protein can range from 1% to 100% according to ASCO guidelines <span class="citation" data-cites="Allison2020">(<a href="#ref-Allison2020" role="doc-biblioref">3</a>)</span>. It has been shown that the patients in the low spectrum of ER positivity, from 1% to 10%, do not benefit as much as patients with 10% to 100% of ER positive cells in the tumor <span class="citation" data-cites="Dieci2021">(<a href="#ref-Dieci2021" role="doc-biblioref">4</a>)</span>.</p>
<p>Transcriptomics and genomics have revolutionazed how cancer research is done. The new related tools opened the option to better understand the molecular underpinnings of cancer biology. In breast cancer, microarray technologies and RNA-sequencing have been used to develop gene signatures that are able to assign a risk score, usually for node-negative, post-menopausal and ER+ BC patients. The risk score is based on survival data found in the literature and the higher the risk score of a patient, the higher the benefit of additional chemotherapy besides the first-line therapy, such as aromatase inhibitors. Several of these signatures are already available to be used in hospitals <span class="citation" data-cites="mammaprint pam50 oncotypedx endopredict bci">(<a href="#ref-mammaprint" role="doc-biblioref">5</a>–<a href="#ref-bci" role="doc-biblioref">9</a>)</span>. Moreover, some of the risk scores are already recommended in international guidelines <span class="citation" data-cites="Andre2022">(<a href="#ref-Andre2022" role="doc-biblioref">10</a>)</span>. These signatures though do not provide a possible explanation on why a patient should receive additional therapy. These signatures are composed of several submodules, they are in most of the cases related to estrogen, proliferation, HER2 and invasion. In some of the cases the risk score is more strongly associated with the estrogen module and in other cases to the proliferation module <span class="citation" data-cites="Buus2021">(<a href="#ref-Buus2021" role="doc-biblioref">11</a>)</span>. There is still a lack of understanding of the patient’s tumor molecular biology. There are challenges to overcome in the pathway analysis for patients individually and how to compare patients molecularly when considering complete transcriptomics data. There is no tool that allows to integrate patients in a continuous way. Usually integration is a one step procedure and cannot be updated. The usual tools are <span class="citation" data-cites="Risso2014 Zhang2020 Fei2018">(<a href="#ref-Risso2014" role="doc-biblioref">12</a>–<a href="#ref-Fei2018" role="doc-biblioref">14</a>)</span> and they do not provide batch effect removal for new samples. The only way is to re-run the procedure together with the new sample. The problem of doing this is usually you don’t have enough samples to estimate batch effects across groups and therefore correct it.</p>
<p>We propose here a framework for personalized medicine and the understanding of the signaling pathways in a patient level. By developing a normalization and projection technique, we show that it is possible to integrate publicly available molecular datasets, such as TCGA, SCAN-B, METABRIC and microarray data of some patients from the POETIC trial <span class="citation" data-cites="tcgabrca Saal2015 Curtis2012 Gao2019">(<a href="#ref-tcgabrca" role="doc-biblioref">15</a>–<a href="#ref-Gao2019" role="doc-biblioref">18</a>)</span>, and that it is possible to compare patients in a neighborhood of the projected data, providing a context to compare patients and better understand the biology of the tumor. Moreover, we propose to use the different gene sets from the hallmarks collection of the Molecular Signature Database (MSigDB) <span class="citation" data-cites="Subramanian2005 Liberzon2011">(<a href="#ref-Subramanian2005" role="doc-biblioref">19</a>,<a href="#ref-Liberzon2011" role="doc-biblioref">20</a>)</span> to understand the biology of the tumors in patient and neighborhood levels.</p>
</section>
<section id="methods" class="level1">
<h1>Methods</h1>
<section id="cohorts" class="level2">
<h2 class="anchored" data-anchor-id="cohorts">Cohorts</h2>
<p>The breast cancer cohorts TCGA, SCANB, METABRIC and POETIC <span class="citation" data-cites="tcgabrca Saal2015 Curtis2012 Gao2019">(<a href="#ref-tcgabrca" role="doc-biblioref">15</a>–<a href="#ref-Gao2019" role="doc-biblioref">18</a>)</span> were used to calculate the PCA embeddings and scores. The samples are of primary breast cancer samples from different molecular subtypes and age distribution. TCGA was download from Firebrowse, SCANB data was downloaded from GEO (accession code GSE96058) using the package GEOquery <span class="citation" data-cites="geoquery">(<a href="#ref-geoquery" role="doc-biblioref">21</a>)</span> in R along the clinical data from the https://oncogenomics.bmc.lu.se/MutationExplorer/ website. METABRIC was downloaded directly from cBioPortal. POETIC data was downloaded from GEO (accession code GSE105777) using the R package GEOquery. More details about the download and preprocessing steps are described in https://chronchi.github.io/transcriptomics.</p>
</section>
<section id="selection-of-highly-variable-genes" class="level2">
<h2 class="anchored" data-anchor-id="selection-of-highly-variable-genes">Selection of highly variable genes</h2>
<p>In order to perform the PCA, we sub selected 1000 common and highly variable genes in the TCGA and METABRIC cohorts. For each gene and in each cohort separately, the standard deviation of that gene was calculated. Then an average standard deviation was calculated using the formula below.</p>
<p><span class="math display">\[
\text{average sd} = \sqrt{\frac{(sd_{TCGA}^2 + sd_{METABRIC}^2)}{k}}
\]</span></p>
</section>
<section id="qpcr-like-normalization" class="level2">
<h2 class="anchored" data-anchor-id="qpcr-like-normalization">qPCR-like normalization</h2>
<p>Since samples are coming from different platforms, they need to be scaled in a way that they are comparable. For this we developed a new way to scale the data based on the ranking of the samples. Given a list of 44 stable genes across different cancers <span class="citation" data-cites="Bhuva2020">(<a href="#ref-Bhuva2020" role="doc-biblioref">22</a>)</span> and the 1000 genes selected previously, all genes were ranked from lowest to highest expression for each sample separately and the rankings were divided by the average ranking of the stable genes. Stable genes are considered the housekeeping genes.</p>
</section>
<section id="pca-embedding" class="level2">
<h2 class="anchored" data-anchor-id="pca-embedding">PCA embedding</h2>
<p>Using the normalized data, a total of 1000 random samples coming from TCGA and METABRIC were selected to perform the initial PCA. This is an unsupervised learning method, therefore there is no need to label the samples with respect to some category. The package PCAtools <span class="citation" data-cites="Blighe2022">(<a href="#ref-Blighe2022" role="doc-biblioref">23</a>)</span> in R was used to perform the PCA and to obtain the loadings for downstream analysis. PCA was used with centering and scaling set to <code>FALSE</code>. The embedding for new individual samples is obtained by multiplying the loadings matrix with the normalized data for that sample. Since the normalization procedure is performed sample-wise, this step is independent of the number of samples. If there are missing genes in a sample, the normalization is performed and the missing genes are padded with 0.</p>
</section>
<section id="scoring-strategies" class="level2">
<h2 class="anchored" data-anchor-id="scoring-strategies">Scoring strategies</h2>
<p>For the 4 big cohorts, TCGA, SCANB, METABRIC and POETIC, GSVA <span class="citation" data-cites="Hnzelmann2013">(<a href="#ref-Hnzelmann2013" role="doc-biblioref">24</a>)</span> was applied along with the <span class="math inline">\(SET_{ER/PR}\)</span> signature <span class="citation" data-cites="Sinn2019">(<a href="#ref-Sinn2019" role="doc-biblioref">25</a>)</span> and the hallmark collection from the molecular signature database <span class="citation" data-cites="Subramanian2005 Mootha2003">(<a href="#ref-Subramanian2005" role="doc-biblioref">19</a>,<a href="#ref-Mootha2003" role="doc-biblioref">26</a>)</span>. Default parameters were used in the <code>gsva</code> function from the GSVA package.</p>
</section>
<section id="average-neighborhood-scores" class="level2">
<h2 class="anchored" data-anchor-id="average-neighborhood-scores">Average neighborhood scores</h2>
<p>In order to calculate the posterior distribution of the average scores in each neighborhood, a linear regression witho only intercept was fit using <code>rstanarm</code> <span class="citation" data-cites="Goodrich2022">(<a href="#ref-Goodrich2022" role="doc-biblioref">27</a>)</span>. The package <code>tidybayes</code> <span class="citation" data-cites="Kay2022">(<a href="#ref-Kay2022" role="doc-biblioref">28</a>)</span> was used to extract the draws and put them in a tidy format.</p>
</section>
<section id="code-availability" class="level2">
<h2 class="anchored" data-anchor-id="code-availability">Code availability</h2>
<p>The code used to generate all the analysis is available on <a href="https://github.com/chronchi/molecular_landscape">https://github.com/chronchi/molecular_landscape</a>. Descriptions for a docker image to reproduce the analysis are available on the github repository. An online version with a website containing all the analysis can be found on <a href="https://chronchi.github.io/molecular_landscape">https://chronchi.github.io/molecular_landscape</a></p>
</section>
</section>
<section id="results" class="level1">
<h1>Results</h1>
<section id="estrogen-receptor-is-a-clinical-continuous-variable" class="level2">
<h2 class="anchored" data-anchor-id="estrogen-receptor-is-a-clinical-continuous-variable">Estrogen receptor is a clinical continuous variable</h2>
<p>We used three independent breast cancer molecular datasets <span class="citation" data-cites="Saal2015 tcgabrca Curtis2012">(<a href="#ref-tcgabrca" role="doc-biblioref">15</a>–<a href="#ref-Curtis2012" role="doc-biblioref">17</a>)</span> to calculate estrogen signaling scores. The estrogen signatures HALLMARK_ESTROGEN_RESPONSE_EARLY and HALLMARK_ESTROGEN_RESPONSE_LATE were extracted from the molecular signature database <span class="citation" data-cites="Subramanian2005">(<a href="#ref-Subramanian2005" role="doc-biblioref">19</a>)</span> and <span class="math inline">\(SET_{ER/PR}\)</span> from <span class="citation" data-cites="Sinn2019">(<a href="#ref-Sinn2019" role="doc-biblioref">25</a>)</span>. The individual scores for each patient sample are shown in <a href="#fig-01">Figure&nbsp;1</a> (a) for each cohort stratified by estrogen receptor status. It shows the scores capture the differences between the two breast cancer subtypes as expected. Moreover, there is a wide range of values in the estrogen receptor positive (ER+) subgroup.</p>
<p>Cox regression was used to determine the hazard ratio of the estrogen signaling in overall survival (OS) for TCGA, SCANB and METABRIC and recurrence free survival (RFS) for METABRIC. Each survival analysis was done independently and adjusted for available clinical variables. Tumor size and number of lymph nodes were used for TCGA and SCANB cohorts. The Nottingham prognostic index (NPI) was used for METABRIC. Age was used in all cohorts as a clinical variable for adjustment. Only ER+ BC patients were used and when possible only those that received endocrine therapy. <a href="#fig-01">Figure&nbsp;1</a> (b) shows the forest plots for each cohort individually when calculating the hazard ratio for the <span class="math inline">\(SET_{ER/PR}\)</span> estrogen signaling signature. In all the three cases, the hazard ratio for <span class="math inline">\(SET_{ER/PR}\)</span> was below 1, with values ranging from 0.23 to 0.61. There is moderate variability for each hazard ratio. This shows the continuous aspect of estrogen receptor status. For both estrogen pathways of the hallmark collection of gene sets, hazard ratios were below 1 in all cases (Table 1, supplementary material).</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-01" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="images/png/figure01.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">Figure&nbsp;1: Scores and survival analysis results from TCGA, SCANB and METABRIC cohorts. (a) GSVA scores for the SET ER/PR signature for each cohort. Each point corresponds to a patient sample and they are divided by estrogen receptor status. (b) Forest plot of the survival analysis for each cohort separately. NPI: Nottingham prognostic index. Ti: i-th stage of tumor. Ni: i lymph nodes with breast cancer cells.</figcaption><p></p>
</figure>
</div>
</div>
</div>
</section>
<section id="single-sample-integration-preserves-relevant-breast-cancer-properties" class="level2">
<h2 class="anchored" data-anchor-id="single-sample-integration-preserves-relevant-breast-cancer-properties">Single sample integration preserves relevant breast cancer properties</h2>
<p>Since each patient has a different ER signaling score, we assumed that patients should be treated individually, not just binned in two big subgroups as ER+ and ER-. Therefore, it is important to consider each patient individually. We developed a single sample batch effect removal method (see methods section) to integrate microarray and bulk RNA-seq and create a molecular landscape. The advantage of the method is that given a new sample, it can easily be integrated with all the other previous samples without any retraining.</p>
<p>The first step is to normalize the samples to the same scale. The average ranking of housekeeping genes distribution is similar among the two RNA-seq datasets, SCAN-B and TCGA and their mean values are higher than the mean of the distribution on METABRIC (Fig S1a). Moreover the normalization preserves essential characteristics of the breast cancer biology (Fig S1b).</p>
<p>The biplot in <a href="#fig-02">Figure&nbsp;2</a> (a) with the third and fourth components from TCGA and METABRIC samples shows that the samples are well integrated. All samples, including those using for training and validation, are plotted. The third components corresponds to the separation between ER+ and ER- BC patients in both cohorts (<a href="#fig-02">Figure&nbsp;2</a> (b)). A combination of the third and fourth components shows a good distinction among the PAM50 molecular subtypes (<a href="#fig-02">Figure&nbsp;2</a> (c)). The fourth component is mostly dividing the luminal A and luminal B subtypes, whereas the normal-like subtype is spread across the third and fourth component. This also highlights the fact that one cannot interpret the PCA locations globally, rather when comparing samples one should consider only its neighborhood.</p>
<p>As pointed out before, ER status should be considered continuous and not dichotomous, <a href="#fig-02">Figure&nbsp;2</a> (d) shows a gradient of the ER signaling score <span class="math inline">\(SET_{ER/PR}\)</span>. The higher values are on the far right of the third component, going to negative values as one goes from right to left, i.e., moving from a more ER+ status to the ER-. There are other pathways driving the distinction between samples, such as G2M checkpoint, epithelial to mesenchymal transition, DNA repair and others (supplementary figure 2; 2.20 from book).</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-02" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="images/png/figure02.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">Figure&nbsp;2: (a) Biplot using the third and fourth components on TCGA and METABRIC samples. Colored by cohort. (b) Same as <strong>a</strong>, colored by ER status. (c) Same as <strong>a</strong>, colored by PAM50 molecular subtype. (d) Hex grid calculated on the biplot of the fourth and third component. Each hex is colored based on its average value of the SET ER/PR signature.</figcaption><p></p>
</figure>
</div>
</div>
</div>
</section>
<section id="embedding-is-robust-to-missing-genes-and-generalized-to-a-validation-cohort" class="level2">
<h2 class="anchored" data-anchor-id="embedding-is-robust-to-missing-genes-and-generalized-to-a-validation-cohort">Embedding is robust to missing genes and generalized to a validation cohort</h2>
<p>METABRIC and TCGA were used to train and validate the projections. SCANB was used as an external validation cohort. SCANB is well mixed with both METABRIC and TCGA samples (<a href="#fig-03">Figure&nbsp;3</a> (a)). ER+ and ER- BC patients are well separated (<a href="#fig-03">Figure&nbsp;3</a> (b)) and the procedure can also distinguish the molecular subtypes (<a href="#fig-03">Figure&nbsp;3</a> (c)). As an RNA-seq cohort, it is expected that SCANB samples will be closer to TCGA than to METABRIC when removing batch effects, due to platform biases and initial scale of the genes. Biplot of PC1 and PC2 (<a href="#fig-03">Figure&nbsp;3</a> (d)) shows that SCANB is closer to TCGA than to METABRIC. It is also in between the two cohorts. We also show that the embedding works for the SMC cohort <span class="citation" data-cites="Kan2018">(<a href="#ref-Kan2018" role="doc-biblioref">29</a>)</span> (supplementary figure 3; 3.14 from the book).</p>
<p>In order to check the robustness of the procedure, we redid all the pipeline 10 times with 10 random sets of patient samples from TCGA and METABRIC, simulating a cross validation process. The PCA embedding is invariant to rotation, translation and reflection (<a href="#fig-03">Figure&nbsp;3</a> (e)). Another problem that arises with publicly available datasets, is the fact that there are missing genes. We try to understand the effect of missing genes in the embedding based on their loading values. Ideally if a low amount of genes with high loadings are missing, this should not affect very much the embedding. On the other hand, the more genes missing with high loadings, the more it will impact the embedding. We removed 200 genes in total with a varying proportion of top loading genes (ranging from 0 to 100% in a 5% step). The number of top loading genes missing from the dataset is key for the embedding (<a href="#fig-03">Figure&nbsp;3</a> (f)). The higher the proportion the less precise the embedding is.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-03" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="images/png/figure03.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">Figure&nbsp;3: Validation of the molecular landscape with an external cohort (a) Biplot using the third and fourth components and now including all samples from the three cohorts: TCGA, METABRIC and SCANB. (b) Same as <strong>a</strong>, colored by ER status. (c) Same as <strong>a</strong>, colored by PAM50 molecular subtype. (d) Biplot using the first and second component of TCGA, METABRIC and SCANB. (e) Embedding of random samples given different training sets for PCA. Blue dots correspond to the original embedding of a sample and red dots correspond to the new embedding given the new training set. (f) Biplot of all possible embeddings of sample given a certain proportion of top loadings missing in the dataset.</figcaption><p></p>
</figure>
</div>
</div>
</div>
</section>
<section id="molecular-landscape-is-a-tool-to-understand-and-explore-patient-heterogeneity" class="level2">
<h2 class="anchored" data-anchor-id="molecular-landscape-is-a-tool-to-understand-and-explore-patient-heterogeneity">Molecular landscape is a tool to understand and explore patient heterogeneity</h2>
<p>Since the molecular landscape relies in a single sample embedding, we can add samples from any cohort with relative good data. The POETIC trial <span class="citation" data-cites="Dowsett2011">(<a href="#ref-Dowsett2011" role="doc-biblioref">30</a>)</span> was a trial that evaluated the use of perioperative aromatase inhibitors in ER+, postmenopausal BC patients. Its primary endpoint was time to recurrence. They sent for microarray hybridization matched samples from baseline (before treatment) and at surgery (after an average of 14 days of treatment) <span class="citation" data-cites="Gao2019">(<a href="#ref-Gao2019" role="doc-biblioref">18</a>)</span>. There are also untreated patients, used to control for sample processing artefacts. Moreover, the patients have matched Ki67 percentage levels, which can be considered an indication of how well a patient responded to the endocrine therapy. Patients with more than 5% of baseline Ki67 and a reduction of 60% upon endocrine therapy are considered responders, otherwise they are called non responders.</p>
<p>The molecular landscape can shed light on the differences between responders and non responders. We embedded the POETIC trial samples using the procedure (<a href="#fig-04">Figure&nbsp;4</a> (a) left). The samples are spread across the whole molecular landscape, showing that patients indeed have different molecular biological properties. Furthermore, the POETIC samples are embedded closer to the METABRIC samples (supplementary figure 4; 4.3 from the book). Given the available information, the patients that are ER+ and in the left part of the landscape (ER- patients), are all non responders (<a href="#fig-04">Figure&nbsp;4</a> (a) right). This highlights the importance to look more carefully to ER+ patients. We selected two patients, a responder and non responder that are close in the embedding (<a href="#fig-04">Figure&nbsp;4</a> (b)) to highlight their molecular differences and see what is their context. <a href="#fig-04">Figure&nbsp;4</a> (c) shows the average posterior distribution of the neighborhood for the responder patient. The responder patient has a ER signaling score higher than the average. On the other hand, the non responder has a smaller ER signaling score than the average (<a href="#fig-04">Figure&nbsp;4</a> (d)) and also a higher androgen signaling score (Androgen response). Other pathways and their average posterior distributions are shown for both patients.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-04" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="images/png/figure04.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">Figure&nbsp;4: Embedding of the POETIC cohort into the molecular landscape and pathway analysis for patient samples. (a) Biplots of the POETIC samples (baseline and surgery) into the molecular landscape. Left plot is colored by ER status and right plot is colored by molecular subtype PAM50 when available. (b) Biplot highlighting two patients with similar embedding and different response status. (c) Posterior distributions of the average scores in the neighborhood of the responder patient. Red line corresponds to the patient score. (d) Posterior distributions of the average scores in the neighborhood of the <strong>non</strong>-responder patient. Red line corresponds to the patient score.</figcaption><p></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="discussion" class="level1">
<h1>Discussion</h1>
<p>Personalized medicine is a key topic in medicine. The goal of better understanding the molecular underpinnings of the diseases leads to a better allocation of treatments and resources in the patient care. We have shown here a possible framework to deal with personalized medicine in breast cancer in general with a focus on ER+ BC patients.</p>
<p>In general, estrogen receptor status is considered either a dichotomous or trichotomous state <span class="citation" data-cites="Allison2020">(<a href="#ref-Allison2020" role="doc-biblioref">3</a>)</span>. Breast cancers are classified either in ER+ or ER- based on their protein expression levels and IHC. For those tumors that are ER+, they can be subdivided into low ER positive (1% &lt; IHC% &lt; 10%) and simply ER positive. Those that are into the low ER positive do not usually benefit as much on endocrine therapy. Here we show using cox regression and multiple big study cohorts <span class="citation" data-cites="tcgabrca Curtis2012 Saal2015">(<a href="#ref-tcgabrca" role="doc-biblioref">15</a>–<a href="#ref-Curtis2012" role="doc-biblioref">17</a>)</span> with both RNA-seq and microarray data, that ER status is more of a continuous rather than a categorical state. This can aid in treatment decision when evaluating the breast cancer. Several gene signatures already are being used, such as OncotypeDX, Mammaprint and Prosigna to assign chemotherapy for those patients with higher risk of recurrence <span class="citation" data-cites="oncotypedx mammaprint pam50">(<a href="#ref-mammaprint" role="doc-biblioref">5</a>–<a href="#ref-oncotypedx" role="doc-biblioref">7</a>)</span>. This score might be associated with the commercial signatures, as it has been shown that OncotypeDX’s estrogen module is highly correlated with the signature itself <span class="citation" data-cites="Buus2021">(<a href="#ref-Buus2021" role="doc-biblioref">11</a>)</span>.</p>
<p>Integrating molecular data steaming from different sources is a challenge. On one hand batch effect tools are usually able to remove the batch effects across the different sources of variability <span class="citation" data-cites="Zhang2020 Risso2014">(<a href="#ref-Risso2014" role="doc-biblioref">12</a>,<a href="#ref-Zhang2020" role="doc-biblioref">13</a>)</span>, on the other hand they are not single sample based, meaning each time a new sample comes the algorithm runs. It is also based on the fact one has enough data in the different datasets, otherwise it skews the possible integration towards one of the datasets. Here we show by using TCGA, METABRIC and SCANB that it is possible to integrate the samples from these cohorts in a single sample manner. The samples show good mixing when using test samples not seen during the training stage. The embeddings preserve key molecular features of breast cancer. PC3 is clearly driven by estrogen signaling where from right to left there is a gradient of ER positivity. On the other hand, PC4 is what makes a difference between the molecular subtypes luminal A and B, which in practice differ by proliferation status in terms of Ki67 levels <span class="citation" data-cites="Arima2018">(<a href="#ref-Arima2018" role="doc-biblioref">31</a>)</span>.</p>
<p>Moreover, the embeddings in a validation set preserve the key features of breast cancer. Samples are projected by their different PAM50 molecular subtypes and there is a gradient of estrogen signaling pathway from ER+ BC towards ER- BC patients. The first two components are the batch effect removal components and SCANB is projected closer to TCGA, since both datasets are RNA-seq. On the other hand, part of the POETIC trial was sequenced using microarray dataset <span class="citation" data-cites="Dowsett2011 Gao2019">(<a href="#ref-Gao2019" role="doc-biblioref">18</a>,<a href="#ref-Dowsett2011" role="doc-biblioref">30</a>)</span>, and we show that it is projected closer to METABRIC as expected according to the first two components. We also show that the embedding is independent of the training set and that when changing the training set the embedding is only rotated or translated, thus not affecting the global structure of the projection.</p>
<p>Sometimes when dealing with publicly available datasets, not all of the genes are available due to ethical <span class="citation" data-cites="Saal2015">(<a href="#ref-Saal2015" role="doc-biblioref">16</a>)</span>, pre-processing or technological reasons. Therefore we showed how robust the projection is to missing genes with high loadings in the projection. If less than 20% of the genes we can recover almost surely the position of the embedding if all the genes are available. The more genes that are missing, the closer the projection will be to the origin, i.e., the (0,0) coordinate in the plane.</p>
<p>To show the clinical validity of the projection, we used a subset of patients from the POETIC trial with microarray and clinical information <span class="citation" data-cites="Gao2019">(<a href="#ref-Gao2019" role="doc-biblioref">18</a>)</span>. The samples are projected as expected and surprisingly the samples that are considered to be non responders upon 2 weeks of aromatase inhibition are projected among the ER- BC patients. When looking further upon two different patients that have similar embedding but different response to endocrine therapy, we see that the responder has a higher value of estrogen signaling than the average. On the other hand, the non responder patient has a smaller estrogen signaling score than the average, which suggests a possible explanation for the difference in response. Moreover, estrogen and androgen receptor signaling have been show to be tightly linked <span class="citation" data-cites="Hickey2021">(<a href="#ref-Hickey2021" role="doc-biblioref">32</a>)</span> and these two patients have different androgen signaling scores. The non responder has a higher score than the average compared to the responder, whose score is just the same as the average.</p>
<p>Some weaknesses of the proposed method is that we rely on GSVA scores, which can be used and compared across different cohorts since they have a representation of all molecular subtypes. A possible way to circumvent this problem is by using singscore or stingscore <span class="citation" data-cites="Foroutan2018 Bhuva2020">(<a href="#ref-Bhuva2020" role="doc-biblioref">22</a>,<a href="#ref-Foroutan2018" role="doc-biblioref">33</a>)</span>, a single sample pathway scorer. There is still a cost barrier for using RNA-seq dataset in the clinical setting, but efforts are being made to reduce costs by using shallow sequencing of multiple samples at the same time, such as done by Alithea Genomics.</p>
<p>In conclusion we provide some evidence that ER status should be considered a continuous marker rather than a categorical one. We also extend this notion to a framework for personalized medicine, where each patient is embedded in some context that can be used for the interpretation of its molecular underpinnings. In this paper we only discussed some pathways, but when analysing patient data, several pathways can be considered and should be taken into account when deciding tumor treatment.</p>
</section>
<section id="references" class="level1">


</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body" role="doc-bibliography">
<div id="ref-whocancer" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">1. </div><div class="csl-right-inline">World Health Organization: Regional Office for Europe. World cancer report. IARC; 2020. </div>
</div>
<div id="ref-Anderson2002" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">2. </div><div class="csl-right-inline">Anderson WF, Chatterjee N, Ershler WB, Brawley OW. Estrogen receptor breast cancer phenotypes in the surveillance, epidemiology, and end results database. Breast Cancer Research and Treatment [Internet]. Springer Science; Business Media <span>LLC</span>; 2002;76:27–36. Available from: <a href="https://doi.org/10.1023/a:1020299707510">https://doi.org/10.1023/a:1020299707510</a></div>
</div>
<div id="ref-Allison2020" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">3. </div><div class="csl-right-inline">Allison KH, Hammond MEH, Dowsett M, McKernin SE, Carey LA, Fitzgibbons PL, et al. Estrogen and progesterone receptor testing in breast cancer: <span>ASCO</span>/<span>CAP</span> guideline update. Journal of Clinical Oncology [Internet]. American Society of Clinical Oncology (<span>ASCO</span>); 2020;38:1346–66. Available from: <a href="https://doi.org/10.1200/jco.19.02309">https://doi.org/10.1200/jco.19.02309</a></div>
</div>
<div id="ref-Dieci2021" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">4. </div><div class="csl-right-inline">Dieci MV, Griguolo G, Bottosso M, Tsvetkova V, Giorgi CA, Vernaci G, et al. Impact of estrogen receptor levels on outcome in non-metastatic triple negative breast cancer patients treated with neoadjuvant/adjuvant chemotherapy. npj Breast Cancer [Internet]. Springer Science; Business Media <span>LLC</span>; 2021;7. Available from: <a href="https://doi.org/10.1038/s41523-021-00308-7">https://doi.org/10.1038/s41523-021-00308-7</a></div>
</div>
<div id="ref-mammaprint" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">5. </div><div class="csl-right-inline">Cardoso F, Veer LJ van’t, Bogaerts J, Slaets L, Viale G, Delaloge S, et al. 70-gene signature as an aid to treatment decisions in early-stage breast cancer. New England Journal of Medicine [Internet]. Massachusetts Medical Society; 2016;375:717–29. Available from: <a href="https://doi.org/10.1056/nejmoa1602253">https://doi.org/10.1056/nejmoa1602253</a></div>
</div>
<div id="ref-pam50" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">6. </div><div class="csl-right-inline">Parker JS, Mullins M, Cheang MCU, Leung S, Voduc D, Vickery T, et al. Supervised risk predictor of breast cancer based on intrinsic subtypes. Journal of Clinical Oncology [Internet]. American Society of Clinical Oncology (<span>ASCO</span>); 2009;27:1160–7. Available from: <a href="https://doi.org/10.1200/jco.2008.18.1370">https://doi.org/10.1200/jco.2008.18.1370</a></div>
</div>
<div id="ref-oncotypedx" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">7. </div><div class="csl-right-inline">Sparano JA, Gray RJ, Makower DF, Pritchard KI, Albain KS, Hayes DF, et al. Adjuvant chemotherapy guided by a 21-gene expression assay in breast cancer. New England Journal of Medicine [Internet]. Massachusetts Medical Society; 2018;379:111–21. Available from: <a href="https://doi.org/10.1056/nejmoa1804710">https://doi.org/10.1056/nejmoa1804710</a></div>
</div>
<div id="ref-endopredict" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">8. </div><div class="csl-right-inline">Filipits M, Rudas M, Jakesz R, Dubsky P, Fitzal F, Singer CF, et al. A new molecular predictor of distant recurrence in <span>ER</span>-positive, <span>HER</span>2-negative breast cancer adds independent information to conventional clinical risk factors. Clinical Cancer Research [Internet]. American Association for Cancer Research (<span>AACR</span>); 2011;17:6012–20. Available from: <a href="https://doi.org/10.1158/1078-0432.ccr-11-0926">https://doi.org/10.1158/1078-0432.ccr-11-0926</a></div>
</div>
<div id="ref-bci" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">9. </div><div class="csl-right-inline">Jankowitz RC, Cooper K, Erlander MG, Ma X-J, Kesty NC, Li H, et al. Prognostic utility of the breast cancer index and comparison to adjuvant! Online in a clinical case series of early breast cancer. Breast Cancer Research [Internet]. Springer Science; Business Media <span>LLC</span>; 2011;13. Available from: <a href="https://doi.org/10.1186/bcr3038">https://doi.org/10.1186/bcr3038</a></div>
</div>
<div id="ref-Andre2022" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">10. </div><div class="csl-right-inline">Andre F, Ismaila N, Allison KH, Barlow WE, Collyar DE, Damodaran S, et al. Biomarkers for adjuvant endocrine and chemotherapy in early-stage breast cancer: <span>ASCO</span> guideline update. Journal of Clinical Oncology [Internet]. American Society of Clinical Oncology (<span>ASCO</span>); 2022;40:1816–37. Available from: <a href="https://doi.org/10.1200/jco.22.00069">https://doi.org/10.1200/jco.22.00069</a></div>
</div>
<div id="ref-Buus2021" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">11. </div><div class="csl-right-inline">Buus R, Sestak I, Kronenwett R, Ferree S, Schnabel CA, Baehner FL, et al. Molecular drivers of onco<span class="math inline">\(\less\)</span>i<span class="math inline">\(\greater\)</span>type<span class="math inline">\(\less\)</span>/i<span class="math inline">\(\greater\)</span> <span>DX</span>, prosigna, <span>EndoPredict</span>, and the breast cancer index: A <span>TransATAC</span> study. Journal of Clinical Oncology [Internet]. American Society of Clinical Oncology (<span>ASCO</span>); 2021;39:126–35. Available from: <a href="https://doi.org/10.1200/jco.20.00853">https://doi.org/10.1200/jco.20.00853</a></div>
</div>
<div id="ref-Risso2014" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">12. </div><div class="csl-right-inline">Risso D, Ngai J, Speed TP, Dudoit S. Normalization of <span>RNA</span>-seq data using factor analysis of control genes or samples. Nature Biotechnology [Internet]. Springer Science; Business Media <span>LLC</span>; 2014;32:896–902. Available from: <a href="https://doi.org/10.1038/nbt.2931">https://doi.org/10.1038/nbt.2931</a></div>
</div>
<div id="ref-Zhang2020" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">13. </div><div class="csl-right-inline">Zhang Y, Parmigiani G, Johnson WE. <span>ComBat</span>-seq: Batch effect adjustment for <span>RNA</span>-seq count data. <span>NAR</span> Genomics and Bioinformatics [Internet]. Oxford University Press (<span>OUP</span>); 2020;2. Available from: <a href="https://doi.org/10.1093/nargab/lqaa078">https://doi.org/10.1093/nargab/lqaa078</a></div>
</div>
<div id="ref-Fei2018" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">14. </div><div class="csl-right-inline">Fei T, Zhang T, Shi W, Yu T. Mitigating the adverse impact of batch effects in sample pattern detection. Birol I, editor. Bioinformatics [Internet]. Oxford University Press (<span>OUP</span>); 2018;34:2634–41. Available from: <a href="https://doi.org/10.1093/bioinformatics/bty117">https://doi.org/10.1093/bioinformatics/bty117</a></div>
</div>
<div id="ref-tcgabrca" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">15. </div><div class="csl-right-inline">Comprehensive molecular portraits of human breast tumours. Nature [Internet]. Springer Science; Business Media <span>LLC</span>; 2012;490:61–70. Available from: <a href="https://doi.org/10.1038/nature11412">https://doi.org/10.1038/nature11412</a></div>
</div>
<div id="ref-Saal2015" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">16. </div><div class="csl-right-inline">Saal LH, Vallon-Christersson J, Häkkinen J, Hegardt C, Grabau D, Winter C, et al. The sweden cancerome analysis network - breast (<span>SCAN</span>-b) initiative: A large-scale multicenter infrastructure towards implementation of breast cancer genomic analyses in the clinical routine. Genome Medicine [Internet]. Springer Science; Business Media <span>LLC</span>; 2015;7:20. Available from: <a href="https://doi.org/10.1186/s13073-015-0131-9">https://doi.org/10.1186/s13073-015-0131-9</a></div>
</div>
<div id="ref-Curtis2012" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">17. </div><div class="csl-right-inline">Curtis C, Sohrab P. Shah and, Chin S-F, Turashvili G, Rueda OM, Dunning MJ, et al. The genomic and transcriptomic architecture of 2, 000 breast tumours reveals novel subgroups. Nature [Internet]. Springer Science; Business Media <span>LLC</span>; 2012;486:346–52. Available from: <a href="https://doi.org/10.1038/nature10983">https://doi.org/10.1038/nature10983</a></div>
</div>
<div id="ref-Gao2019" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">18. </div><div class="csl-right-inline">Gao Q, Elena López-Knowles and, Cheang MCU, Morden J, Ribas R, Sidhu K, et al. Impact of aromatase inhibitor treatment on global gene expression and its association with antiproliferative response in <span>ER</span><span class="math inline">\(\mathplus\)</span> breast cancer in postmenopausal patients. Breast Cancer Research [Internet]. Springer Science; Business Media <span>LLC</span>; 2019;22. Available from: <a href="https://doi.org/10.1186/s13058-019-1223-z">https://doi.org/10.1186/s13058-019-1223-z</a></div>
</div>
<div id="ref-Subramanian2005" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">19. </div><div class="csl-right-inline">Subramanian A, Tamayo P, Mootha VK, Mukherjee S, Ebert BL, Gillette MA, et al. Gene set enrichment analysis: A knowledge-based approach for interpreting genome-wide expression profiles. Proceedings of the National Academy of Sciences [Internet]. Proceedings of the National Academy of Sciences; 2005;102:15545–50. Available from: <a href="https://doi.org/10.1073/pnas.0506580102">https://doi.org/10.1073/pnas.0506580102</a></div>
</div>
<div id="ref-Liberzon2011" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">20. </div><div class="csl-right-inline">Liberzon A, Subramanian A, Pinchback R, Thorvaldsdottir H, Tamayo P, Mesirov JP. Molecular signatures database (<span>MSigDB</span>) 3.0. Bioinformatics [Internet]. Oxford University Press (<span>OUP</span>); 2011;27:1739–40. Available from: <a href="https://doi.org/10.1093/bioinformatics/btr260">https://doi.org/10.1093/bioinformatics/btr260</a></div>
</div>
<div id="ref-geoquery" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">21. </div><div class="csl-right-inline">Davis S, Meltzer P. GEOquery: A bridge between the gene expression omnibus (GEO) and BioConductor. Bioinformatics. 2007;14:1846–7. </div>
</div>
<div id="ref-Bhuva2020" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">22. </div><div class="csl-right-inline">Bhuva DD, Cursons J, Davis MJ. Stable gene expression for normalisation and single-sample scoring. Nucleic Acids Research [Internet]. Oxford University Press (<span>OUP</span>); 2020;48:e113–3. Available from: <a href="https://doi.org/10.1093/nar/gkaa802">https://doi.org/10.1093/nar/gkaa802</a></div>
</div>
<div id="ref-Blighe2022" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">23. </div><div class="csl-right-inline">Blighe K, Lun A. PCAtools: PCAtools: Everything principal components analysis [Internet]. 2022. Available from: <a href="https://github.com/kevinblighe/PCAtools">https://github.com/kevinblighe/PCAtools</a></div>
</div>
<div id="ref-Hnzelmann2013" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">24. </div><div class="csl-right-inline">Hänzelmann S, Castelo R, Guinney J. <span>GSVA</span>: Gene set variation analysis for microarray and <span>RNA</span>-seq data. <span>BMC</span> Bioinformatics [Internet]. Springer Science; Business Media <span>LLC</span>; 2013;14. Available from: <a href="https://doi.org/10.1186/1471-2105-14-7">https://doi.org/10.1186/1471-2105-14-7</a></div>
</div>
<div id="ref-Sinn2019" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">25. </div><div class="csl-right-inline">Sinn BV, Fu C, Lau R, Litton J, Tsai T-H, Murthy R, et al. <span>SETER</span>/<span>PR</span>: A robust 18-gene predictor for sensitivity to endocrine therapy for metastatic breast cancer. npj Breast Cancer [Internet]. Springer Science; Business Media <span>LLC</span>; 2019;5. Available from: <a href="https://doi.org/10.1038/s41523-019-0111-0">https://doi.org/10.1038/s41523-019-0111-0</a></div>
</div>
<div id="ref-Mootha2003" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">26. </div><div class="csl-right-inline">Mootha VK, Lindgren CM, Eriksson K-F, Subramanian A, Sihag S, Lehar J, et al. <span>PGC</span>-1<span class="math inline">\(\upalpha\)</span>-responsive genes involved in oxidative phosphorylation are coordinately downregulated in human diabetes. Nature Genetics [Internet]. Springer Science; Business Media <span>LLC</span>; 2003;34:267–73. Available from: <a href="https://doi.org/10.1038/ng1180">https://doi.org/10.1038/ng1180</a></div>
</div>
<div id="ref-Goodrich2022" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">27. </div><div class="csl-right-inline">Goodrich B, Gabry J, Ali I, Brilleman S. Rstanarm: <span>Bayesian</span> applied regression modeling via <span>Stan</span>. [Internet]. 2022. Available from: <a href="https://mc-stan.org/rstanarm/">https://mc-stan.org/rstanarm/</a></div>
</div>
<div id="ref-Kay2022" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">28. </div><div class="csl-right-inline">Kay M. <span class="nocase">tidybayes</span>: Tidy data and geoms for <span>Bayesian</span> models [Internet]. 2022. Available from: <a href="http://mjskay.github.io/tidybayes/">http://mjskay.github.io/tidybayes/</a></div>
</div>
<div id="ref-Kan2018" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">29. </div><div class="csl-right-inline">Kan Z, Ding Y, Kim J, Jung HH, Chung W, Lal S, et al. Multi-omics profiling of younger asian breast cancers reveals distinctive molecular signatures. Nature Communications [Internet]. Springer Science; Business Media <span>LLC</span>; 2018;9. Available from: <a href="https://doi.org/10.1038/s41467-018-04129-4">https://doi.org/10.1038/s41467-018-04129-4</a></div>
</div>
<div id="ref-Dowsett2011" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">30. </div><div class="csl-right-inline">Dowsett M, Smith I, Robertson J, Robison L, Pinhel I, Johnson L, et al. Endocrine therapy, new biologicals, and new study designs for presurgical studies in breast cancer. <span>JNCI</span> Monographs [Internet]. Oxford University Press (<span>OUP</span>); 2011;2011:120–3. Available from: <a href="https://doi.org/10.1093/jncimonographs/lgr034">https://doi.org/10.1093/jncimonographs/lgr034</a></div>
</div>
<div id="ref-Arima2018" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">31. </div><div class="csl-right-inline">Arima N, Nishimura R, Osako T, Okumura Y, Nakano M, Fujisue M, et al. Ki‑67 index value and progesterone receptor status can predict prognosis and suitable treatment in node‑negative breast cancer patients with estrogen receptor‑positive and <span>HER</span>2‑negative tumors. Oncology Letters [Internet]. Spandidos Publications; 2018; Available from: <a href="https://doi.org/10.3892/ol.2018.9633">https://doi.org/10.3892/ol.2018.9633</a></div>
</div>
<div id="ref-Hickey2021" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">32. </div><div class="csl-right-inline">Hickey TE, Selth LA, Chia KM, Laven-Law G, Milioli HH, Roden D, et al. The androgen receptor is a tumor suppressor in estrogen receptor<span></span>positive breast cancer. Nature Medicine [Internet]. Springer Science; Business Media <span>LLC</span>; 2021;27:310–20. Available from: <a href="https://doi.org/10.1038/s41591-020-01168-7">https://doi.org/10.1038/s41591-020-01168-7</a></div>
</div>
<div id="ref-Foroutan2018" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">33. </div><div class="csl-right-inline">Foroutan M, Bhuva DD, Lyu R, Horan K, Cursons J, Davis MJ. Single sample scoring of molecular phenotypes. <span>BMC</span> Bioinformatics [Internet]. Springer Science; Business Media <span>LLC</span>; 2018;19. Available from: <a href="https://doi.org/10.1186/s12859-018-2435-4">https://doi.org/10.1186/s12859-018-2435-4</a></div>
</div>
</div></section></div></main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
</div> <!-- /content -->



</body></html>